--- a/cmake/modules/AddMLIR.cmake
+++ b/cmake/modules/AddMLIR.cmake
@@ -300,7 +300,7 @@
   endforeach()
 
   list(APPEND ARG_DEPENDS mlir-generic-headers)
-  llvm_add_library(${name} ${LIBTYPE} ${ARG_UNPARSED_ARGUMENTS} ${srcs} DEPENDS ${ARG_DEPENDS} LINK_COMPONENTS ${ARG_LINK_COMPONENTS} LINK_LIBS ${ARG_LINK_LIBS})
+  llvm_add_library(${name} ${LIBTYPE} DISABLE_LLVM_LINK_LLVM_DYLIB ${ARG_UNPARSED_ARGUMENTS} ${srcs} DEPENDS ${ARG_DEPENDS} LINK_COMPONENTS ${ARG_LINK_COMPONENTS} LINK_LIBS ${ARG_LINK_LIBS})
 
   if(TARGET ${name})
     target_link_libraries(${name} INTERFACE ${LLVM_COMMON_LIBS})
@@ -352,7 +352,7 @@
 endfunction(add_mlir_library)
 
 macro(add_mlir_tool name)
-  llvm_add_tool(MLIR ${ARGV})
+  llvm_add_tool(MLIR ${ARGV} DISABLE_LLVM_LINK_LLVM_DYLIB)
 endmacro()
 
 # Sets a variable with a transformed list of link libraries such individual
--- a/tools/mlir-parser-fuzzer/bytecode/CMakeLists.txt
+++ b/tools/mlir-parser-fuzzer/bytecode/CMakeLists.txt
@@ -3,6 +3,7 @@
   Support
 )
 add_llvm_fuzzer(mlir-bytecode-parser-fuzzer
+  DISABLE_LLVM_LINK_LLVM_DYLIB
   mlir-bytecode-parser-fuzzer.cpp
   DUMMY_MAIN DummyParserFuzzer.cpp
 )
--- a/tools/mlir-parser-fuzzer/text/CMakeLists.txt
+++ b/tools/mlir-parser-fuzzer/text/CMakeLists.txt
@@ -3,6 +3,7 @@
   Support
 )
 add_llvm_fuzzer(mlir-text-parser-fuzzer
+  DISABLE_LLVM_LINK_LLVM_DYLIB
   mlir-text-parser-fuzzer.cpp
   DUMMY_MAIN DummyParserFuzzer.cpp
 )
--- a/test/CAPI/CMakeLists.txt
+++ b/test/CAPI/CMakeLists.txt
@@ -7,6 +7,7 @@
   set(LLVM_LINK_COMPONENTS
     )
   add_llvm_executable(${name}
+    DISABLE_LLVM_LINK_LLVM_DYLIB
     PARTIAL_SOURCES_INTENDED
     ${ARG_UNPARSED_ARGUMENTS})
   llvm_update_compile_flags(${name})
