diff -urN libiconv-1.17/configure.orig libiconv-1.17/configure
--- libiconv-1.17/configure.orig	2022-06-09 17:17:33.325129700 +0200
+++ libiconv-1.17/configure	2022-06-09 17:41:31.158194700 +0200
@@ -31929,7 +31929,7 @@
 fi
 ac_cs_awk_cr=`$AWK 'BEGIN { print "a\rb" }' </dev/null 2>/dev/null`
 if test "$ac_cs_awk_cr" = "a${ac_cr}b"; then
-  ac_cs_awk_cr='\\r'
+  ac_cs_awk_cr='\r'
 else
   ac_cs_awk_cr=$ac_cr
 fi
