diff --git a/libcharset/lib/localcharset.c "b/libcharset/lib/localcharset.c"
index 3aceb42..21fba38 100644
--- a/libcharset/lib/localcharset.c
+++ b/libcharset/lib/localcharset.c
@@ -466,7 +466,7 @@ locale_charset (void)
      But in GUI programs and for output sent to files and pipes, GetACP()
      encoding is the best bet.  */
   sprintf (buf, "CP%u", GetACP ());
-  codeset = buf;
+  codeset = "UTF-8";
 
 #elif defined OS2
 
