--- a/lib/iconv.c	2018-05-03 23:18:55.997221700 -0400
+++ b/lib/iconv.c	2018-05-03 23:26:47.611682700 -0400
@@ -170,12 +170,12 @@ static const struct stringpool2_t string
 #include "aliases2.h"
 #undef S
 };
 #define stringpool2 ((const char *) &stringpool2_contents)
 static const struct alias sysdep_aliases[] = {
+#define S(tag,name,encoding_index) { (intptr_t)&((struct stringpool2_t *)0)->stringpool_##tag, encoding_index },
-#define S(tag,name,encoding_index) { (int)(long)&((struct stringpool2_t *)0)->stringpool_##tag, encoding_index },
 #include "aliases2.h"
 #undef S
 };
 #ifdef __GNUC__
 __inline
 #else
