From ce83996a2c634a7db30946e7eb275f7e4f688db3 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgads@gmail.com>
Date: Thu, 13 Feb 2020 09:25:04 +0200
Subject: [PATCH 4/5] Python: Configure path fixes

---
 gdb/configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/configure b/gdb/configure
index 5d89635c043..f25b51e5e6e 100755
--- a/gdb/configure
+++ b/gdb/configure
@@ -11181,7 +11181,7 @@ fi
 	as_fn_error $? "failure running python-config --ldflags" "$LINENO" 5
       fi
     fi
-    python_prefix=`${python_prog} ${srcdir}/python/python-config.py --exec-prefix`
+    python_prefix=`cygpath -u "$(${python_prog} ${srcdir}/python/python-config.py --exec-prefix)"`
     if test $? != 0; then
       have_python_config=failed
       if test "${with_python}" != auto; then
-- 
2.33.1.windows.1

