diff -rupN old/tools/cygwin_windows.go new/tools/cygwin_windows.go
--- old/tools/cygwin_windows.go	2017-04-21 22:27:30.893330900 -0500
+++ new/tools/cygwin_windows.go	2017-04-21 22:35:24.320047700 -0500
@@ -43,7 +43,10 @@ func isCygwin() bool {
 		return false
 	}
 
-	if bytes.Contains(out, []byte("CYGWIN")) || bytes.Contains(out, []byte("MSYS")) {
+	if bytes.Contains(out, []byte("CYGWIN")) ||
+	   bytes.Contains(out, []byte("MINGW64")) ||
+	   bytes.Contains(out, []byte("MINGW32")) ||
+	   bytes.Contains(out, []byte("MSYS")) {
 		cygwinState = cygwinStateEnabled
 	} else {
 		cygwinState = cygwinStateDisabled
