From 46c000c4f87f06fe9c67292a2c673afeed48a1b4 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 21 Sep 2021 20:52:42 +0200
Subject: [PATCH 016/N] sysconfig: MINGW build extensions with GCC

---
 Lib/sysconfig.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index 8122a7a..3710a83 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -782,6 +782,8 @@ def get_platform():
 
     """
     if os.name == 'nt':
+        if 'GCC' in sys.version:
+            return 'mingw'
         if 'amd64' in sys.version.lower():
             return 'win-amd64'
         if '(arm)' in sys.version.lower():
