From 68bc9c9ec8ffd42ca82e413547ab12c8a0586c7e Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Mon, 19 Feb 2024 19:55:55 +0530
Subject: [PATCH 112/N] mingw_smoketests: skip test_c_ext_build for in-tree
 builds

---
 mingw_smoketests.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 0ab5d3b..dc31423 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -283,6 +283,10 @@ class Tests(unittest.TestCase):
         self.assertEqual(len(site.getsitepackages()), 1)
 
     def test_c_ext_build(self):
+        # This will not work in in-tree build
+        if sysconfig.is_python_build():
+            raise unittest.SkipTest("in-tree build")
+
         import tempfile
         import sys
         import subprocess
