diff --git a/gnuradio-runtime/lib/CMakeLists.txt b/gnuradio-runtime/lib/CMakeLists.txt
index 577262df2..553731167 100644
--- a/gnuradio-runtime/lib/CMakeLists.txt
+++ b/gnuradio-runtime/lib/CMakeLists.txt
@@ -129,7 +129,8 @@ target_link_libraries(
         MPLib::mplib
     PRIVATE
         Boost::program_options
-        libunwind::libunwind)
+        libunwind::libunwind
+        ws2_32)
 
 # Address linker issues with std::filesystem on Centos 8 and Debian
 target_link_libraries(
