--- libcerf-v2.0-46d5887030419861cd68bf6b8f7c96c9f18c551a/lib/CMakeLists.txt.orig	2020-10-14 09:26:19.870186900 +0300
+++ libcerf-v2.0-46d5887030419861cd68bf6b8f7c96c9f18c551a/lib/CMakeLists.txt	2020-10-14 09:27:21.958923400 +0300
@@ -16,7 +16,7 @@
 install(
     TARGETS ${lib}
     LIBRARY DESTINATION lib
-    RUNTIME DESTINATION lib
+    RUNTIME DESTINATION bin
     ARCHIVE DESTINATION lib
     COMPONENT Libraries)
 install(
