diff -Naur libcerf-1.6.orig/lib/CMakeLists.txt libcerf-1.6/lib/CMakeLists.txt
--- libcerf-1.6.orig/lib/CMakeLists.txt	2018-09-21 20:38:53.117797400 -0400
+++ libcerf-1.6/lib/CMakeLists.txt	2018-09-21 20:47:43.024933500 -0400
@@ -16,8 +16,10 @@
 
 if (${LIB_INSTALL})
     install(
-        TARGETS ${lib} LIBRARY
-        DESTINATION ${destination}/lib
+        TARGETS ${lib}
+        RUNTIME DESTINATION ${destination}/bin
+        LIBRARY DESTINATION ${destination}/lib
+        ARCHIVE DESTINATION ${destination}/lib
         COMPONENT Libraries)
     install(
         FILES ${inc_files}
diff -Naur libcerf-1.6.orig/man/CMakeLists.txt libcerf-1.6/man/CMakeLists.txt
--- libcerf-1.6.orig/man/CMakeLists.txt	2018-09-21 20:38:53.135797800 -0400
+++ libcerf-1.6/man/CMakeLists.txt	2018-09-21 20:45:30.899574600 -0400
@@ -15,7 +15,7 @@
         )
     install(
         FILES ${CMAKE_CURRENT_BINARY_DIR}/${pname}.${section}
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/man/man${section}"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/share/man/man${section}"
         )
     set(TARGETS_MAN "${TARGETS_MAN};${pname}.${section}" PARENT_SCOPE)
     set(TARGETS_HTM "${TARGETS_HTM};${pname}.html"       PARENT_SCOPE)
