diff -Naur libcerf-1.13-orig/CMakeLists.txt libcerf-1.13/CMakeLists.txt
--- libcerf-1.13-orig/CMakeLists.txt	2019-02-28 17:36:35.000000000 +0300
+++ libcerf-1.13/CMakeLists.txt	2019-06-10 10:55:58.632523900 +0300
@@ -54,9 +54,6 @@
     set(CTEST_CONFIGURATION_TYPE "${JOB_BUILD_CONFIGURATION}")
     # set(DEFAULT_BUILD_SHARED_LIBS OFF)
     # set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin/$<CONFIG>)
-    if(BUILD_SHARED_LIBS)
-        set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
-    endif()
 else()
     add_compile_options(-O2 -Wno-sign-compare -fno-omit-frame-pointer)
     if(PEDANTIC)
@@ -66,6 +63,10 @@
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3 -w")
 endif()
 
+if(BUILD_SHARED_LIBS)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
 include(CTest)
 
 if(LIB_INSTALL)
diff -Naur libcerf-1.13-orig/lib/w_of_z.c libcerf-1.13/lib/w_of_z.c
--- libcerf-1.13-orig/lib/w_of_z.c	2019-02-12 20:07:55.000000000 +0300
+++ libcerf-1.13/lib/w_of_z.c	2019-06-10 10:59:23.640820900 +0300
@@ -66,7 +66,7 @@
 #include <math.h>
 #include "defs.h" // defines _cerf_cmplx, NaN, C, cexp, ...
 
-#if _WIN32
+#if _MSC_VER
 #define EXPORT __declspec(dllexport)
 #else
 #define EXPORT
diff -Naur libcerf-1.13-orig/run/run_w_of_z.c libcerf-1.13/run/run_w_of_z.c
--- libcerf-1.13-orig/run/run_w_of_z.c	2019-02-12 20:08:13.000000000 +0300
+++ libcerf-1.13/run/run_w_of_z.c	2019-06-10 10:59:12.590805400 +0300
@@ -23,7 +23,7 @@
 #include "cerf.h"
 #include "defs.h"
 
-#if _WIN32
+#ifdef _MSC_VER
 #define IMPORT __declspec(dllimport)
 #else
 #define IMPORT
