--- gst-python-1.4.0/configure.ac.orig	2015-07-21 14:34:06.163625200 +0300
+++ gst-python-1.4.0/configure.ac	2015-07-21 14:34:25.981643200 +0300
@@ -101,11 +101,7 @@
 
 AC_MSG_CHECKING(for pygobject overrides directory)
 if test "x$with_pygi_overrides_dir" = "x" ; then
-        overrides_dir="`$PYTHON -c 'import gi; print(gi._overridesdir)' 2>/dev/null`"
-    # fallback if the previous failed
-    if test "x$overrides_dir" = "x" ; then
         overrides_dir="${pyexecdir}/gi/overrides"
-    fi
 else
         overrides_dir="$with_pygi_overrides_dir"
 fi
