--- a/Makefile.am
+++ b/Makefile.am
@@ -9,7 +9,7 @@
 # When rolling a release, remember to adjust the version info.
 # It's current:release:age.
 libieee1284_la_LDFLAGS = -version-info 5:2:2 \
-	-export-symbols $(top_srcdir)/libieee1284.sym
+	-no-undefined -export-symbols $(top_srcdir)/libieee1284.sym
 include_HEADERS = include/ieee1284.h
 INCLUDES = -I$(top_srcdir)/include
 
@@ -39,43 +39,6 @@
 src_ieee1284module_la_CFLAGS = -I/usr/include/python$(PYTHON_VERSION) -fpic
 endif
 
-man3_MANS = doc/libieee1284.3 \
-	doc/parport.3 \
-	doc/parport_list.3 \
-	doc/ieee1284_find_ports.3 \
-	doc/ieee1284_free_ports.3 \
-	doc/ieee1284_get_deviceid.3 \
-	doc/ieee1284_open.3 \
-	doc/ieee1284_close.3 \
-	doc/ieee1284_ref.3 \
-	doc/ieee1284_unref.3 \
-	doc/ieee1284_claim.3 \
-	doc/ieee1284_release.3 \
-	doc/ieee1284_read_data.3 \
-	doc/ieee1284_write_data.3 \
-	doc/ieee1284_data_dir.3 \
-	doc/ieee1284_wait_data.3 \
-	doc/ieee1284_read_status.3 \
-	doc/ieee1284_wait_status.3 \
-	doc/ieee1284_read_control.3 \
-	doc/ieee1284_write_control.3 \
-	doc/ieee1284_frob_control.3 \
-	doc/ieee1284_do_nack_handshake.3 \
-	doc/ieee1284_negotiate.3 doc/ieee1284_terminate.3 \
-	doc/ieee1284_ecp_fwd_to_rev.3 doc/ieee1284_ecp_rev_to_fwd.3 \
-	doc/ieee1284_nibble_read.3 \
-	doc/ieee1284_compat_write.3 \
-	doc/ieee1284_byte_read.3 \
-	doc/ieee1284_epp_read_data.3 doc/ieee1284_epp_write_data.3 \
-	doc/ieee1284_epp_read_addr.3 doc/ieee1284_epp_write_addr.3 \
-	doc/ieee1284_ecp_read_data.3 doc/ieee1284_ecp_write_data.3 \
-	doc/ieee1284_ecp_read_addr.3 doc/ieee1284_ecp_write_addr.3 \
-	doc/ieee1284_get_irq_fd.3 \
-	doc/ieee1284_set_timeout.3
-
-$(man3_MANS): $(top_srcdir)/doc/interface.xml
-	xmlto man -o doc $<
-
 doc/interface.pdf: $(top_srcdir)/doc/interface.xml
 	xmlto pdf -o doc --extensions $<
 
@@ -89,7 +52,5 @@
 DISTCLEANFILES = include/stamp-h include/stamp-h[0-9]* \
 	doc/manpage.links doc/manpage.refs
 
-MAINTAINERCLEANFILES = $(man3_MANS)
-
 tag:
 	cvs tag -c `echo V@VERSION@ | tr . _`
