--- clazy-1.16/CMakeLists.txt.orig	2025-09-12 08:58:59.103364100 +0300
+++ clazy-1.16/CMakeLists.txt	2025-09-12 08:59:08.367962300 +0300
@@ -311,7 +311,7 @@
     message(FATAL_ERROR "Could not find ClangTidyModule.h header")
   endif()
   add_clang_plugin(ClazyClangTidy src/ClazyClangTidyModule.cpp)
-  target_link_libraries(ClazyClangTidy ClazyPlugin)
+  target_link_libraries(ClazyClangTidy ClazyPlugin clangTidy)
   set_target_properties(ClazyClangTidy PROPERTIES
     LINKER_LANGUAGE CXX
     PREFIX ""
