--- ghc-8.6.5/libraries/integer-gmp/gmp/ghc.mk.orig	2019-04-08 03:39:58.000000000 +0800
+++ ghc-8.6.5/libraries/integer-gmp/gmp/ghc.mk	2026-06-29 20:27:55.692507600 +0800
@@ -42,7 +42,6 @@
 ifeq "$(Windows_Host)" "YES"
 # Apparently building on Windows fails when there is a system gmp
 # available, so we never try to use the system gmp on Windows
-libraries/integer-gmp_CONFIGURE_OPTS += --configure-option=--with-intree-gmp
 endif
 
 ifeq "$(GMP_PREFER_FRAMEWORK)" "YES"
