From 3b3fcc71bec389adbe7f6545ccb4765d7da5ea65 Mon Sep 17 00:00:00 2001
From: Ben Gamari <ben@smart-cactus.org>
Date: Fri, 1 Jan 2021 20:43:38 -0500
Subject: [PATCH] rts/Linker: Add noreturn to loadNativeObj on non-ELF
 platforms

---
 rts/Linker.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/rts/Linker.c b/rts/Linker.c
index 176127da24..5502965355 100644
--- a/rts/Linker.c
+++ b/rts/Linker.c
@@ -2012,19 +2012,23 @@ addSection (Section *s, SectionKind kind, SectionAlloc alloc,
 
 #define UNUSED(x) (void)(x)
 
+#if defined(OBJFORMAT_ELF)
 void * loadNativeObj (pathchar *path, char **errmsg)
 {
-#if defined(OBJFORMAT_ELF)
    ACQUIRE_LOCK(&linker_mutex);
    void *r = loadNativeObj_ELF(path, errmsg);
    RELEASE_LOCK(&linker_mutex);
    return r;
+}
 #else
+void * GNU_ATTRIBUTE(__noreturn__)
+loadNativeObj (pathchar *path, char **errmsg)
+{
    UNUSED(path);
    UNUSED(errmsg);
    barf("loadNativeObj: not implemented on this platform");
-#endif
 }
+#endif
 
 HsInt unloadNativeObj (void *handle)
 {
-- 
2.30.0

