From 8bed49759ae23f6fdfcc89e24bd1743b4c2b4285 Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Tue, 21 Feb 2017 02:53:10 -0500
Subject: [PATCH] Handle CRLF properly for C dependencies

---
 rules/build-dependencies.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rules/build-dependencies.mk b/rules/build-dependencies.mk
index cdc1edf..60cfd6c 100644
--- a/rules/build-dependencies.mk
+++ b/rules/build-dependencies.mk
@@ -145,7 +145,7 @@ endef
 define addCFileDeps
 
 	$(CPP) $($1_$2_MKDEPENDC_OPTS) $($1_$2_$(firstword $($1_$2_WAYS))_ALL_CC_OPTS) $($(basename $4)_CC_OPTS) -MM -x c $4 -MF $3.bit
-	$(foreach w,$5,sed -e 's|\\|/|g' -e 's| /$$| \\|' -e "1s|\.o|\.$($w_osuf)|" -e "1s|^|$(dir $4)|" -e "1s|$1/|$1/$2/build/|" -e "1s|$2/build/$2/build|$2/build|g" -e "s|$(TOP)/||g$(CASE_INSENSITIVE_SED)" $3.bit >> $3.tmp &&) true
+	$(foreach w,$5,sed -e 's|\\|/|g' -e 's|\r$$||' -e 's| /$$| \\|' -e "1s|\.o|\.$($w_osuf)|" -e "1s|^|$(dir $4)|" -e "1s|$1/|$1/$2/build/|" -e "1s|$2/build/$2/build|$2/build|g" -e "s|$(TOP)/||g$(CASE_INSENSITIVE_SED)" $3.bit >> $3.tmp &&) true
 endef
 
 ifeq "$(Windows_Host)" "YES"
-- 
2.11.1

