--- ghc-8.6.5/ghc.mk.orig	2019-04-08 03:39:58.000000000 +0800
+++ ghc-8.6.5/ghc.mk	2026-06-30 02:55:18.543108600 +0800
@@ -819,20 +819,6 @@
 # -----------------------------------------------------------------------------
 # Creating a local mingw copy on Windows
 
-ifeq "$(Windows_Host)" "YES"
-
-install : install_mingw
-.PHONY: install_mingw
-install_mingw : $(INPLACE_MINGW)
-	"$(CP)" -Rp $(INPLACE_MINGW) $(prefix)
-
-install : install_perl
-.PHONY: install_perl
-install_perl : $(INPLACE_PERL)
-	"$(CP)" -Rp $(INPLACE_PERL) $(prefix)
-
-endif # Windows_Host
-
 ifneq "$(BINDIST)" "YES"
 $(ghc-prim-$(libraries/ghc-prim_dist-install_VERSION)_HADDOCK_FILE): \
     libraries/ghc-prim/dist-install/build/autogen/GHC/Prim.hs
--- ghc-8.6.5/compiler/main/SysTools/BaseDir.hs.orig	2019-04-08 03:39:58.000000000 +0800
+++ ghc-8.6.5/compiler/main/SysTools/BaseDir.hs	2026-06-30 03:03:20.276421800 +0800
@@ -92,12 +92,7 @@
 -- | Expand occurrences of the @$tooldir@ interpolation in a string
 -- on Windows, leave the string untouched otherwise.
 expandToolDir :: Maybe FilePath -> String -> String
-#if defined(mingw32_HOST_OS)
-expandToolDir (Just tool_dir) s = expandPathVar "tooldir" tool_dir s
-expandToolDir Nothing         _ = panic "Could not determine $tooldir"
-#else
 expandToolDir _ s = s
-#endif
 
 -- | @expandPathVar var value str@
 --
@@ -230,18 +230,4 @@
 findToolDir
   :: FilePath -- ^ topdir
   -> IO (Maybe FilePath)
-#if defined(mingw32_HOST_OS)
-findToolDir top_dir = go 0 (top_dir </> "..")
-  where maxDepth = 2
-        go :: Int -> FilePath -> IO (Maybe FilePath)
-        go k path
-          | k == maxDepth = throwGhcExceptionIO $
-              InstallationError "could not detect mingw toolchain"
-          | otherwise = do
-              oneLevel <- doesDirectoryExist (path </> "mingw")
-              if oneLevel
-                then return (Just path)
-                else go (k+1) (path </> "..")
-#else
 findToolDir _ = return Nothing
-#endif
