diff --git a/rts/rts.cabal.in b/rts/rts.cabal.in
index 9860951226..e6b6e84dcb 100644
--- a/rts/rts.cabal.in
+++ b/rts/rts.cabal.in
@@ -104,8 +104,8 @@ library
        -- If we're compiling on windows, enforce that we only support Windows 7+
        -- Adding this here means it doesn't have to be done in individual .c files
        -- and also centralizes the versioning.
-       cpp-options: -D_WIN32_WINNT=0x06010000
-       cc-options: -D_WIN32_WINNT=0x06010000
+       cpp-options: -D_WIN32_WINNT=0x06010000 -w
+       cc-options: -D_WIN32_WINNT=0x06010000 -w
     if flag(need-pthread)
        -- for pthread_getthreadid_np, pthread_create, ...
        extra-libraries: pthread
