--- ghc-8.6.5/mk/config.mk.in.orig
+++ ghc-8.6.5/mk/config.mk.in
@@ -111,7 +111,6 @@
 # Some platforms don't support shared libraries
 NoSharedLibsPlatformList = \
 	powerpc-ibm-aix \
-	x86_64-unknown-mingw32 \
 	i386-unknown-mingw32
 
 ifeq "$(SOLARIS_BROKEN_SHLD)" "YES"
@@ -129,9 +128,7 @@
 # Currently this isn't possible on Windows, and we have not yet enabled
 # it on i386 while we consider the performance implications.
 #
-ifeq "$(TargetOS_CPP)" "mingw32"
-DYNAMIC_BY_DEFAULT = NO
-else ifeq "$(TargetArch_CPP)" "i386"
+ifeq "$(TargetArch_CPP)" "i386"
 DYNAMIC_BY_DEFAULT = NO
 else
 DYNAMIC_BY_DEFAULT = YES
@@ -144,15 +141,14 @@
 # If building both v and dyn ways, then use -dynamic-too to build them.
 # This makes the build faster.
 DYNAMIC_TOO = YES
+ifeq "$(TargetOS_CPP)" "mingw32"
+DYNAMIC_TOO = NO
+endif
 
 # Use the dynamic way when building programs in the GHC tree. In
 # particular, this means that GHCi will use DLLs rather than loading
 # object files directly.
-ifeq "$(TargetOS_CPP)" "mingw32"
-# This doesn't work on Windows yet
-DYNAMIC_GHC_PROGRAMS = NO
-DYNAMIC_TOO=NO
-else ifeq "$(PlatformSupportsSharedLibs)" "NO"
+ifeq "$(PlatformSupportsSharedLibs)" "NO"
 DYNAMIC_GHC_PROGRAMS = NO
 else
 DYNAMIC_GHC_PROGRAMS = YES
--- ghc-8.6.5/configure.ac.orig
+++ ghc-8.6.5/configure.ac
@@ -636,7 +636,7 @@
 # tools for Windows. See #14274.
 if test "$HostOS" = "mingw32"
 then
-    AC_PATH_PROG([LIBTOOL],[libtool])
+    AC_PATH_PROG([LIBTOOL],[dlltool])
     LibtoolCmd="$(cygpath -m $LIBTOOL)"
 else
     AC_CHECK_TARGET_TOOL([LIBTOOL], [libtool])
--- ghc-8.6.5/configure.orig
+++ ghc-8.6.5/configure
@@ -8575,8 +8575,8 @@
 # tools for Windows. See #14274.
 if test "$HostOS" = "mingw32"
 then
-    # Extract the first word of "libtool", so it can be a program name with args.
-set dummy libtool; ac_word=$2
+    # Extract the first word of "dlltool", so it can be a program name with args.
+set dummy dlltool; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_path_LIBTOOL+:} false; then :
--- ghc-8.6.5/rules/build-package-way.mk.orig
+++ ghc-8.6.5/rules/build-package-way.mk
@@ -58,7 +58,7 @@
 # On windows we have to supply the extra libs this one links to when building it.
 ifeq "$$(TargetOS_CPP)" "mingw32"
 $$($1_$2_$3_LIB) : $$($1_$2_$3_ALL_OBJS) $$(ALL_RTS_LIBS) $$($1_$2_$3_DEPS_LIBS)
-	$$(call build-dll,$1,$2,$3,-L$1/$2/build,,$$($1_$2_$3_HS_OBJS) $$($1_$2_$3_NON_HS_OBJS),"$$@","NO","$$($1_PACKAGE)","$$($1_$2_VERSION)")
+	$$(call build-dll,$1,$2,$3,-L$1/$2/build $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES)) $$(addprefix -L,$$($1_$2_EXTRA_LIBDIRS)),,$$($1_$2_$3_HS_OBJS) $$($1_$2_$3_NON_HS_OBJS),"$$@","NO","$$($1_PACKAGE)","$$($1_$2_VERSION)")
 
 else # ifneq "$$(TargetOS_CPP)" "mingw32"
 $$($1_$2_$3_LIB) : $$($1_$2_$3_ALL_OBJS) $$(ALL_RTS_LIBS) $$($1_$2_$3_DEPS_LIBS)
@@ -150,7 +150,7 @@
 # 9  = create delay load import lib
 # 10 = SxS Name
 # 11 = SxS Version
-$(gen-dll_INPLACE) link "$1" "$2" "$3" "$4" "$5" "$6" "$7" "$(call cmd,$1_$2_HC) $(subst -no-hs-main,,$($1_$2_$3_ALL_HC_OPTS) $($1_$2_$3_GHC_LD_OPTS)) \
+$(gen-dll_INPLACE) link "$1" "$2" "$3" "$4" "$5" "$6" "$7" "$($1_$2_HC) $(subst -no-hs-main,,$($1_$2_$3_ALL_HC_OPTS) $($1_$2_$3_GHC_LD_OPTS)) \
            -shared -dynamic -dynload deploy \
            $(addprefix -l,$($1_$2_EXTRA_LIBRARIES)) \
            -no-auto-link-packages" "$8" \
--- ghc-8.6.5/rules/build-prog.mk.orig
+++ ghc-8.6.5/rules/build-prog.mk
@@ -261,7 +261,7 @@
 	$$(call cmd,$1_$2_HC) -no-hs-main -no-auto-link-packages -optc-g -optc-O0 -Iincludes $$< -o $$@
 
 $1/$2/build/tmp/$$($1_$2_PROG).dll : $$($1_$2_$$($1_$2_PROGRAM_WAY)_HS_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_C_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_S_OBJS) $$($1_$2_OTHER_OBJS) | $$$$(dir $$$$@)/.
-	$$(call build-dll,$1,$2,$$($1_$2_PROGRAM_WAY),,$$($1_$2_$$($1_$2_PROGRAM_WAY)_HS_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_C_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_S_OBJS) $$($1_$2_OTHER_OBJS),$$@)
+	$$(call build-dll,$1,$2,$$($1_$2_PROGRAM_WAY),,,$$($1_$2_$$($1_$2_PROGRAM_WAY)_HS_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_C_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_S_OBJS) $$($1_$2_OTHER_OBJS),$$@,"NO","$$($1_$2_PROG)","$$($1_$2_VERSION)")
 else # $1_$2_PROG_NEEDS_C_WRAPPER=NO
 ifeq "$$($1_$2_LINK_WITH_GCC)" "NO"
 $1/$2/build/tmp/$$($1_$2_PROG) : $$($1_$2_$$($1_$2_PROGRAM_WAY)_HS_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_C_OBJS) $$($1_$2_$$($1_$2_PROGRAM_WAY)_S_OBJS) $$($1_$2_OTHER_OBJS) | $$$$(dir $$$$@)/.
