diff --git a/utils/ghc-cabal/ghc.mk b/utils/ghc-cabal/ghc.mk
index 76b083684f7d84cd1a2ac628e28e765636766928..63de1785675197e4b926bdec2461d33696dab59f 100644
--- a/utils/ghc-cabal/ghc.mk
+++ b/utils/ghc-cabal/ghc.mk
@@ -59,7 +59,8 @@ $(ghc-cabal_DIST_BINARY): $(CABAL_LEXER_DEP) utils/ghc-cabal/Main.hs $(TOUCH_DEP
 	"$(GHC)" $(SRC_HC_OPTS) \
 	       $(addprefix -optc, $(SRC_CC_OPTS) $(CONF_CC_OPTS_STAGE0)) \
 	       $(addprefix -optl, $(SRC_LD_OPTS) $(CONF_GCC_LINKER_OPTS_STAGE0)) \
-				 -O0 \
+		   -O0 \
+		   -XHaskell2010 \
 	       -hide-all-packages \
 	       -package-env - \
 	       $(addprefix -package , $(CABAL_BUILD_DEPS)) \
