From 9de34ccf0c97d7b5b27503981236f992532a6ef9 Mon Sep 17 00:00:00 2001
From: Sylvain Henry <sylvain@haskus.fr>
Date: Wed, 24 Mar 2021 18:29:45 +0100
Subject: [PATCH] Fix Windres call (#16780)

---
 compiler/GHC/SysTools/Tasks.hs | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/compiler/GHC/SysTools/Tasks.hs b/compiler/GHC/SysTools/Tasks.hs
index 7dc40cef04..08065c0048 100644
--- a/compiler/GHC/SysTools/Tasks.hs
+++ b/compiler/GHC/SysTools/Tasks.hs
@@ -389,21 +389,21 @@ runWindres dflags args = traceToolCommand dflags "windres" $ do
   let cc = pgm_c dflags
       cc_args = map Option (sOpt_c (settings dflags))
       windres = pgm_windres dflags
-      opts = map Option (getOpts dflags opt_windres)
+      opts = getOpts dflags opt_windres ++ ["-E", "-xc", "-DRC_INVOKED"]
       quote x = "\"" ++ x ++ "\""
-      args' = -- If windres.exe and gcc.exe are in a directory containing
-              -- spaces then windres fails to run gcc. We therefore need
-              -- to tell it what command to use...
-              Option ("--preprocessor=" ++
-                      unwords (map quote (cc :
-                                          map showOpt opts ++
-                                          ["-E", "-xc", "-DRC_INVOKED"])))
-              -- ...but if we do that then if windres calls popen then
-              -- it can't understand the quoting, so we have to use
-              -- --use-temp-file so that it interprets it correctly.
-              -- See #1828.
-            : Option "--use-temp-file"
-            : args
+      args' = mconcat
+                -- If windres.exe and gcc.exe are in a directory containing
+                -- spaces then windres fails to run gcc. We therefore need
+                -- to tell it what command to use...
+              [ [Option ("--preprocessor=" ++ quote cc)]
+              , map (\o -> Option ("--preprocessor-arg=" ++ o)) opts
+                -- ...but if we do that then if windres calls popen then
+                -- it can't understand the quoting, so we have to use
+                -- --use-temp-file so that it interprets it correctly.
+                -- See #1828.
+              , [Option "--use-temp-file"]
+              , args
+              ]
   mb_env <- getGccEnv cc_args
   runSomethingFiltered dflags id "Windres" windres args' Nothing mb_env
 
-- 
2.31.0