diff --git a/compiler/GHC/SysTools/Tasks.hs b/compiler/GHC/SysTools/Tasks.hs
index 32c49d9099..e505305395 100644
--- a/compiler/GHC/SysTools/Tasks.hs
+++ b/compiler/GHC/SysTools/Tasks.hs
@@ -365,9 +365,9 @@ runWindres logger dflags args = traceToolCommand logger dflags "windres" $ do
       args' = -- If windres.exe and gcc.exe are in a directory containing
               -- spaces then windres fails to run gcc. We therefore need
               -- to tell it what command to use...
-              Option ("--preprocessor=" ++
-                      unwords (map quote (cc :
-                                          map showOpt opts ++
+              Option ("--preprocessor=" ++ quote cc)
+            : Option ("--preprocessor-arg=" ++
+                      unwords ( (map showOpt opts ++
                                           ["-E", "-xc", "-DRC_INVOKED"])))
               -- ...but if we do that then if windres calls popen then
               -- it can't understand the quoting, so we have to use


