--- src/ghc-9.8.2/configure.ac.orig	2024-03-31 14:34:05.434256300 +0200
+++ src/ghc-9.8.2/configure.ac	2024-03-31 14:34:11.733112000 +0200
@@ -375,12 +375,6 @@
 FP_GMP
 FP_CURSES
 
-dnl On Windows we force in-tree GMP build until we support dynamic linking
-if test "$HostOS" = "mingw32"
-then
-  GMP_FORCE_INTREE="YES"
-fi
-
 dnl ** Building a cross compiler?
 dnl --------------------------------------------------------------
 CrossCompiling=NO
