diff -u ghc-9.8.2/rts/win32/ConsoleHandler.c src/ghc-9.8.2/rts/win32/ConsoleHandler.c
--- ghc-9.8.2/rts/win32/ConsoleHandler.c	2024-02-22 21:59:45.000000000 +0100
+++ src/ghc-9.8.2/rts/win32/ConsoleHandler.c	2024-03-27 19:51:31.311213400 +0100
@@ -11,6 +11,8 @@
 #include "RtsUtils.h"
 #include "AsyncMIO.h"
 #include "RtsSignals.h"
+#include "win32/MIOManager.h"
+#include "ThreadLabels.h"
 
 extern int stg_InstallConsoleEvent(int action, StgStablePtr *handler);
 
diff -u ghc-9.8.2/rts/win32/ThrIOManager.c src/ghc-9.8.2/rts/win32/ThrIOManager.c
--- ghc-9.8.2/rts/win32/ThrIOManager.c	2024-02-22 21:59:45.000000000 +0100
+++ src/ghc-9.8.2/rts/win32/ThrIOManager.c	2024-03-27 19:46:45.932658100 +0100
@@ -12,6 +12,7 @@
 #include "ThrIOManager.h"
 #include "MIOManager.h"
 #include "rts/OSThreads.h"
+#include "RtsFlags.h"
 #include "Prelude.h"
 #include <windows.h>
 
