--- a/python/setup.py
+++ b/python/setup.py
@@ -146,7 +146,7 @@
     def initialize_options(self):
         _build_ext.initialize_options(self)
         self.cmake_generator = os.environ.get('PYARROW_CMAKE_GENERATOR')
-        if not self.cmake_generator and sys.platform == 'win32':
+        if not self.cmake_generator and sys.platform == 'win32' and 'MSC' in sys.version:
             self.cmake_generator = 'Visual Studio 15 2017 Win64'
         self.extra_cmake_args = os.environ.get('PYARROW_CMAKE_OPTIONS', '')
         self.build_type = os.environ.get('PYARROW_BUILD_TYPE',
