--- a/src/Vector/BLF/FileStatistics.h
+++ b/src/Vector/BLF/FileStatistics.h
@@ -7,6 +7,7 @@
 #include <Vector/BLF/platform.h>
 
 #include <array>
+#include <cstdint>
 
 #include <Vector/BLF/AbstractFile.h>
 
--- a/src/Vector/BLF/ObjectHeaderBase.h
+++ b/src/Vector/BLF/ObjectHeaderBase.h
@@ -10,6 +10,8 @@
 
 #include <Vector/BLF/vector_blf_export.h>
 
+#include <cstdint>
+
 namespace Vector {
 namespace BLF {
 
