diff --git a/src/vmime/platforms/windows/windowsHandler.cpp b/src/vmime/platforms/windows/windowsHandler.cpp
index 551f6726..2777f65e 100644
--- a/src/vmime/platforms/windows/windowsHandler.cpp
+++ b/src/vmime/platforms/windows/windowsHandler.cpp
@@ -42,7 +42,9 @@
 #include <wincrypt.h>
 
 #if VMIME_HAVE_MLANG
+#   include <initguid.h>
 #   include <mlang.h>
+DEFINE_GUID(IID_IMultiLanguage, 0x275c23e1, 0x3747, 0x11d0, 0x9f,0xea, 0x00,0xaa,0x00,0x3f,0x86,0x46);
 #endif
 
 
