--- libtre/configure.ac.orig	2024-10-16 07:56:30.608197900 +0200
+++ libtre/configure.ac	2024-10-16 07:58:37.253298700 +0200
@@ -435,16 +435,6 @@
   fi
 fi
 
-case $host in
-  *-mingw*)
-    dnl wcsrtombs and wcstombs don't seem to work at all on MinGW.
-    if test "$tre_libutf8" != "yes"; then
-      tre_wchar_reason="Not supported on MinGW"
-      tre_wchar="no ($tre_wchar_reason)"
-    fi
-    ;;
-esac
-
 # Fail if wide character support was specifically requested but is
 # not supported on this system.
 if test "$tre_enable_wchar" = "yes"; then
