diff -ru libbluray-0.4.0/src/Makefile.am patched/src/Makefile.am
--- libbluray-0.4.0/src/Makefile.am	2013-09-17 10:05:56.000000000 +0200
+++ patched/src/Makefile.am	2013-11-02 14:35:10.123235113 +0100
@@ -109,7 +109,7 @@
 endif
 
 
-libbluray_la_LDFLAGS= -version-info $(LT_VERSION_INFO)
+libbluray_la_LDFLAGS= -no-undefined -version-info $(LT_VERSION_INFO)
 libbluray_la_LIBADD= $(LIBXML2_LIBS) $(FT2_LIBS) $(FONTCONFIG_LIBS)
 
 pkginclude_HEADERS = \
@@ -149,6 +149,7 @@
 	libbluray/bdj/native/register_native.c
 
 AM_CFLAGS += $(BDJAVA_CFLAGS)
+libbluray_la_LDFLAGS= -no-undefined
 
 all-local:
 	ant -f $(top_srcdir)/src/libbluray/bdj/build.xml \
