diff -ru libbluray-0.8.0/Makefile.am.orig libbluray-0.8.0/Makefile.am
--- libbluray-0.8.0/Makefile.am.orig	2013-09-17 10:05:56.000000000 +0200
+++ libbluray-0.8.0/Makefile.am	2013-11-02 14:35:10.123235113 +0100
@@ -149,7 +149,7 @@
 endif
 endif
 
-libbluray_la_LDFLAGS= -version-info $(LT_VERSION_INFO) -export-symbols-regex "^bd_"
+libbluray_la_LDFLAGS= -no-undefined -version-info $(LT_VERSION_INFO) -export-symbols-regex "^bd_"
 libbluray_la_LIBADD= $(LIBXML2_LIBS) $(FT2_LIBS) $(FONTCONFIG_LIBS)
 
 noinst_HEADERS = \
