--- gtk+-2.24.20/gdk/Makefile.am.orig	2013-07-28 23:30:01.983082000 +0000
+++ gtk+-2.24.20/gdk/Makefile.am	2013-07-29 00:06:24.167684200 +0000
@@ -207,7 +207,9 @@
 Gdk_2_0_gir_FILES = $(introspection_files)
 Gdk_2_0_gir_CFLAGS = $(INCLUDES)
 INTROSPECTION_GIRS += Gdk-2.0.gir
-
+if OS_WIN32
+Gdk_2_0_gir_CFLAGS += -I$(srcdir)/win32 -I$(srcdir)
+endif
 if USE_X11
 x11_introspection_files = 		\
 	x11/checksettings.c		\
@@ -253,6 +255,53 @@
 
 endif # USE_X11
 
+if OS_WIN32
+w32_introspection_files = 		\
+	win32/gdkapplaunchcontext-win32.c \
+	win32/gdkcolor-win32.c \
+	win32/gdkcursor-win32.c \
+	win32/gdkdisplay-win32.c \
+	win32/gdkdnd-win32.c \
+	win32/gdkdrawable-win32.c \
+	win32/gdkdrawable-win32.h \
+	win32/gdkevents-win32.c \
+	win32/gdkfont-win32.c \
+	win32/gdkgc-win32.c \
+	win32/gdkgeometry-win32.c \
+	win32/gdkglobals-win32.c \
+	win32/gdkim-win32.c \
+	win32/gdkimage-win32.c \
+	win32/gdkinput.c \
+	win32/gdkinput-win32.c \
+	win32/gdkinput-win32.h \
+	win32/gdkkeys-win32.c \
+	win32/gdkmain-win32.c \
+	win32/gdkpixmap-win32.c \
+	win32/gdkpixmap-win32.h \
+	win32/gdkprivate-win32.h \
+	win32/gdkproperty-win32.c \
+	win32/gdkscreen-win32.c \
+	win32/gdkselection-win32.c \
+	win32/gdkspawn-win32.c \
+	win32/gdktestutils-win32.c \
+	win32/gdkvisual-win32.c \
+	win32/gdkwin32.h \
+	win32/gdkwin32id.c \
+	win32/gdkwindow-win32.c \
+	win32/gdkwindow-win32.h
+
+GdkWin32-2.0.gir: $(gdktargetlib) Gdk-2.0.gir Makefile
+GdkWin32_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk
+GdkWin32_2_0_gir_INCLUDES = Gio-2.0 Gdk-2.0 GdkPixbuf-2.0 Pango-1.0
+GdkWin32_2_0_gir_LIBS = $(gdktargetlib)
+GdkWin32_2_0_gir_FILES = $(w32_introspection_files)
+GdkWin32_2_0_gir_CFLAGS = $(INCLUDES) -L$(top_builddir)/gdk -I$(srcdir)/win32 -I$(srcdir)
+INTROSPECTION_GIRS += GdkWin32-2.0.gir
+
+introspection_files += $(filter-out x11/gdkwin32.h, $(w32_introspection_files))
+
+endif # OS_WIN32
+
 girdir = $(datadir)/gir-1.0
 gir_DATA = $(INTROSPECTION_GIRS)
 
