diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cb6f3f5..e9412ee 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -137,9 +137,7 @@ set(ktouch_SRCS
     undocommands/coursecommands.cpp
     undocommands/coursecommands.h
     undocommands/keyboardlayoutcommands.cpp
-    undocommands/keyboardlayoutcommands.h
-    x11_helper.cpp
-    x11_helper.h)
+    undocommands/keyboardlayoutcommands.h)
 
 qt_add_resources(ktouch_imgs_SRCS images/images.qrc)
 
