--- origsrc/ocl-icd-2.3.2/Makefile.am	2023-09-10 19:42:53.360404200 +0900
+++ src/ocl-icd-2.3.2/Makefile.am	2023-09-10 19:54:18.819354400 +0900
@@ -48,7 +48,7 @@ libOpenCL_la_CFLAGS= $(NO_DEPRECATED_DEC
 
 libOpenCL_la_SOURCES = ocl_icd_loader.c ocl_icd_loader.h ocl_icd_debug.h
 nodist_libOpenCL_la_SOURCES = ocl_icd_loader_gen.c ocl_icd.h
-libOpenCL_la_LDFLAGS = -version-info 1:0:0 $(PTHREAD_CFLAGS)
+libOpenCL_la_LDFLAGS = -no-undefined -version-info 1:0:0 $(PTHREAD_CFLAGS)
 libOpenCL_la_LIBADD = $(PTHREAD_LIBS)
 if USE_MAP
 libOpenCL_la_LDFLAGS += -Wl,--version-script,ocl_icd_loader_gen.map
@@ -103,25 +103,23 @@ stamp-generator-dummy: icd_generator.rb
 	$(RUBY) $< --mode generate --database $(srcdir)/ocl_interface.yaml
 	echo "timestamp" > $@
 
-# noinst_LTLIBRARIES would be the correct thing but then libtool
-# only built non shared version :-( So, declaring the libs as
-# pkglib_LTLIBRARIES and using an install hook to remove them.
-pkglib_LTLIBRARIES = libdummycl.la libdummycl2.la libdummycl-noext.la libdummylayer.la
-install-exec-hook::
-	$(RM) -r $(DESTDIR)$(pkglibdir)
+noinst_LTLIBRARIES = libdummycl.la libdummycl2.la libdummycl-noext.la
 libdummycl_la_SOURCES = libdummy_icd.c libdummy_icd.h
 nodist_libdummycl_la_SOURCES = libdummy_icd_gen.c libdummy_icd_gen.h
 libdummycl_la_LDFLAGS = -module
+libdummycl_la_LDFLAGS += -rpath `pwd` -no-undefined -avoid-version
 libdummycl_noext_la_SOURCES = $(libdummycl_la_SOURCES)
 nodist_libdummycl_noext_la_SOURCES = $(nodist_libdummycl_la_SOURCES)
 libdummycl_noext_la_LDFLAGS = -module
+libdummycl_noext_la_LDFLAGS += -rpath `pwd` -no-undefined -avoid-version
 libdummycl_noext_la_CPPFLAGS = $(AM_CPPFLAGS) -DICD_WITHOUT_EXTENSION
 libdummycl2_la_SOURCES = $(libdummycl_la_SOURCES)
 nodist_libdummycl2_la_SOURCES = $(nodist_libdummycl_la_SOURCES)
 libdummycl2_la_LDFLAGS = -module
+libdummycl2_la_LDFLAGS += -rpath `pwd` -no-undefined -avoid-version
 libdummycl2_la_CPPFLAGS = $(AM_CPPFLAGS) -DICD_SUFFIX=\"2\"
-nodist_libdummylayer_la_SOURCES = dummy_icd_layer_gen.c
-libdummylayer_la_LDFLAGS = -module
+#nodist_libdummylayer_la_SOURCES = dummy_icd_layer_gen.c
+#libdummylayer_la_LDFLAGS = -module
 
 noinst_PROGRAMS=run_dummy_icd_through_our_ICDL
 run_dummy_icd_through_our_ICDL_SOURCES = run_dummy_icd.c
