diff --git a/lib/wrappers/sqlite3.nim b/lib/wrappers/sqlite3.nim
index 0276a0a65..91a0dd91b 100644
--- a/lib/wrappers/sqlite3.nim
+++ b/lib/wrappers/sqlite3.nim
@@ -12,9 +12,9 @@ when defined(windows):
   when defined(nimOldDlls):
     const Lib = "sqlite3.dll"
   elif defined(cpu64):
-    const Lib = "sqlite3_64.dll"
+    const Lib = "(sqlite3_64|libsqlite3-0).dll"
   else:
-    const Lib = "sqlite3_32.dll"
+    const Lib = "(sqlite3_32|libsqlite3-0).dll"
 elif defined(macosx):
   const
     Lib = "libsqlite3(|.0).dylib"
