--- gnu-cobol/extras/Makefile.am.orig	2014-04-04 23:43:39.838600000 +0400
+++ gnu-cobol/extras/Makefile.am	2014-04-05 00:07:39.107200000 +0400
@@ -26,5 +26,5 @@
 
 CBL_OC_DUMP.$(COB_MODULE_EXT): CBL_OC_DUMP.cob $(top_builddir)/cobc/cobc
 	. $(top_builddir)/tests/atconfig; . $(top_builddir)/tests/atlocal; \
-		cobc -m -Wall -std=mf -O $(top_srcdir)/extras/CBL_OC_DUMP.cob
+		cobc -m -Wall -std=mf -I$(top_srcdir) -O $(top_srcdir)/extras/CBL_OC_DUMP.cob
 
