diff -Naur webview-0.12.0.orig/docs/api/CMakeLists.txt webview-0.12.0/docs/api/CMakeLists.txt
--- webview-0.12.0.orig/docs/api/CMakeLists.txt	2026-02-11 23:10:37.488417600 +0300
+++ webview-0.12.0/docs/api/CMakeLists.txt	2026-02-11 23:47:28.559280000 +0300
@@ -5,6 +5,7 @@
 set(DOXYFILE_HAVE_DOT YES)
 set(DOXYFILE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/output${CONFIG_SUFFIX}")
 set(DOXYFILE_CLANG_DATABASE_PATH "${PROJECT_BINARY_DIR}")
+set(DOXYFILE_USE_MDFILE_AS_MAINPAGE "${PROJECT_SOURCE_DIR}/README.md")
 
 get_target_property(CORE_SOURCE_DIR webview::core SOURCE_DIR)
 set(DOXYFILE_INPUT "${CORE_SOURCE_DIR}/include")
@@ -22,6 +23,7 @@
 set(DOXYFILE_HAVE_DOT \"${DOXYFILE_HAVE_DOT}\")\n\
 set(DOXYFILE_OUTPUT_DIRECTORY \"${DOXYFILE_OUTPUT_DIRECTORY}\")\n\
 set(DOXYFILE_INPUT \"${DOXYFILE_INPUT}\")\n\
+set(DOXYFILE_USE_MDFILE_AS_MAINPAGE \"${DOXYFILE_USE_MDFILE_AS_MAINPAGE}\")\n\
 set(DOXYFILE_CLANG_DATABASE_PATH \"${DOXYFILE_CLANG_DATABASE_PATH}\")\n\
 configure_file(\"${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in\" \"${CMAKE_CURRENT_BINARY_DIR}/Doxyfile${CONFIG_SUFFIX}\" @ONLY)\n\
 execute_process(COMMAND \"${DOXYGEN_EXECUTABLE}\" \"${CMAKE_CURRENT_BINARY_DIR}/Doxyfile${CONFIG_SUFFIX}\")"
diff -Naur webview-0.12.0.orig/docs/api/Doxyfile.in webview-0.12.0/docs/api/Doxyfile.in
--- webview-0.12.0.orig/docs/api/Doxyfile.in	2024-09-11 19:21:07.000000000 +0300
+++ webview-0.12.0/docs/api/Doxyfile.in	2026-02-11 23:50:06.651645500 +0300
@@ -864,7 +864,8 @@
 # spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = "@DOXYFILE_INPUT@"
+INPUT                  = "@DOXYFILE_INPUT@" \
+                         "@DOXYFILE_USE_MDFILE_AS_MAINPAGE@"
 
 # This tag can be used to specify the character encoding of the source files
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding. Doxygen uses
@@ -1024,7 +1025,7 @@
 # (index.html). This can be useful if you have a project on for instance GitHub
 # and want to reuse the introduction page also for the doxygen output.
 
-USE_MDFILE_AS_MAINPAGE =
+USE_MDFILE_AS_MAINPAGE = "@DOXYFILE_USE_MDFILE_AS_MAINPAGE@"
 
 #---------------------------------------------------------------------------
 # Configuration options related to source browsing
