From 602a3d497b934f28eb55ea15151bac9a747584dd Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Wed, 21 Apr 2021 13:22:57 -0400
Subject: [PATCH 3/8] issue-4107

---
 cmake/OpenCVCompilerOptions.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/OpenCVCompilerOptions.cmake b/cmake/OpenCVCompilerOptions.cmake
index 40a058d..bd93e99 100644
--- a/cmake/OpenCVCompilerOptions.cmake
+++ b/cmake/OpenCVCompilerOptions.cmake
@@ -120,7 +120,6 @@ if(CV_GCC OR CV_CLANG)
     add_extra_compiler_option(-Wall)
   endif()
   add_extra_compiler_option(-Werror=return-type)
-  add_extra_compiler_option(-Werror=non-virtual-dtor)
   add_extra_compiler_option(-Werror=address)
   add_extra_compiler_option(-Werror=sequence-point)
   add_extra_compiler_option(-Wformat)
-- 
2.31.1

