--- opencv-3.0.0/cmake/OpenCVGenPkgconfig.cmake.orig	2015-07-15 12:59:24.381598000 +0300
+++ opencv-3.0.0/cmake/OpenCVGenPkgconfig.cmake	2015-07-15 13:09:57.506279200 +0300
@@ -17,6 +17,7 @@
       if("${isown}")
         get_filename_component(item "${item}" NAME_WE)
         string(REGEX REPLACE "^lib(.*)" "\\1" item "${item}")
+        string(REGEX REPLACE "${OPENCV_DLLVERSION}" "" item "${item}")
       endif()
     endif()
     if(item MATCHES "^-l")
@@ -25,8 +26,10 @@
       get_filename_component(libdir "${item}" PATH)
       get_filename_component(libname "${item}" NAME_WE)
       string(REGEX REPLACE "^lib(.*)" "\\1" libname "${libname}")
+      string(REGEX REPLACE "${OPENCV_DLLVERSION}" "" item "${item}")
       list(APPEND _lst "-L${libdir}" "-l${libname}")
     else()
+      string(REGEX REPLACE "${OPENCV_DLLVERSION}" "" item "${item}")
       list(APPEND _lst "-l${item}")
     endif()
   endforeach()
