--- opencv-3.4.2/cmake/templates/OpenCVConfig.root-WIN32.cmake.in.orig	2018-08-07 08:36:12.076932700 +0300
+++ opencv-3.4.2/cmake/templates/OpenCVConfig.root-WIN32.cmake.in	2018-08-07 08:38:01.530193100 +0300
@@ -63,6 +63,7 @@
     return()
   endif()
   set(candidates)
+  if(MSVC)
   if(OpenCV_STATIC)
     list(APPEND candidates "${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib")
   endif()
@@ -73,6 +74,9 @@
     list(APPEND candidates "gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib")
   endif()
   list(APPEND candidates "${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib")
+  else()
+  list(APPEND candidates "lib")
+  endif()
   foreach(c ${candidates})
     set(p "${OpenCV_CONFIG_PATH}/${c}")
     if(EXISTS "${p}/OpenCVConfig.cmake")
