--- opencv-3.3.0/modules/imgcodecs/src/grfmt_png.cpp.orig	2017-09-16 22:17:59.122629700 +0300
+++ opencv-3.3.0/modules/imgcodecs/src/grfmt_png.cpp	2017-09-16 22:24:12.164966500 +0300
@@ -59,7 +59,7 @@
 #endif
 
 #ifdef HAVE_LIBPNG_PNG_H
-#include <libpng/png.h>
+#include <libpng16/png.h>
 #else
 #include <png.h>
 #endif
--- opencv-3.3.0/cmake/OpenCVFindLibsGrfmt.cmake.orig	2017-09-16 22:17:59.122629700 +0300
+++ opencv-3.3.0/cmake/OpenCVFindLibsGrfmt.cmake	2017-09-16 22:24:12.164966500 +0300
@@ -163,9 +163,9 @@
     include(FindPNG)
     if(PNG_FOUND)
       include(CheckIncludeFile)
-      check_include_file("${PNG_PNG_INCLUDE_DIR}/libpng/png.h" HAVE_LIBPNG_PNG_H)
+      check_include_file("${PNG_PNG_INCLUDE_DIR}/libpng16/png.h" HAVE_LIBPNG_PNG_H)
       if(HAVE_LIBPNG_PNG_H)
-        ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/libpng/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
+        ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/libpng16/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
       else()
         ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
       endif()