--- opencv-3.3.0/modules/videoio/src/cap_dshow.cpp.orig	2017-09-16 22:17:59.122629700 +0300
+++ opencv-3.3.0/modules/videoio/src/cap_dshow.cpp	2017-09-16 22:24:12.164966500 +0300
@@ -108,6 +108,9 @@
 #include <vector>
 
 //Include Directshow stuff here so we don't worry about needing all the h files.
+#ifdef __MINGW32__
+#define STRSAFE_NO_DEPRECATE
+#endif
 #include "DShow.h"
 #include "strmif.h"
 #include "Aviriff.h"
