--- opencv_contrib-3.3.0/modules/dnn_modern/src/caffe_converter.cpp.org	2017-07-31 22:58:38.000000000 +0800
+++ opencv_contrib-3.3.0/modules/dnn_modern/src/caffe_converter.cpp	2017-10-05 14:52:19.512831000 +0800
@@ -41,7 +41,7 @@
   or tort (including negligence or otherwise) arising in any way out of
   the use of this software, even if advised of the possibility of such damage.
  */
-
+#define NO_STRICT 
 #include "precomp.hpp"
 #include <opencv2/imgproc.hpp>
 
