From 9a9feb4b496a92318e6d0ef2594793a0bb063165 Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Wed, 21 Apr 2021 13:31:50 -0400
Subject: [PATCH 7/8] python-install-path

---
 modules/python/common.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/python/common.cmake b/modules/python/common.cmake
index 6a438fd..080bec8 100644
--- a/modules/python/common.cmake
+++ b/modules/python/common.cmake
@@ -128,7 +128,7 @@ else()
   set(PYTHON_INSTALL_CONFIGURATIONS "")
 endif()
 
-if(WIN32)
+if(MSVC)
   set(PYTHON_INSTALL_ARCHIVE "")
 else()
   set(PYTHON_INSTALL_ARCHIVE ARCHIVE DESTINATION ${${PYTHON}_PACKAGES_PATH} COMPONENT python)
-- 
2.31.1

