--- a/modules/core/CMakeLists.txt	2019-12-24 11:16:17.940771900 +0300
+++ b/modules/core/CMakeLists.txt	2019-12-24 11:14:48.489214700 +0300
@@ -94,7 +94,7 @@
   endif()
  if(DEFINED OPENCV_ALLOCATOR_STATS_COUNTER_TYPE)
     message(STATUS "Allocator metrics storage type: '${OPENCV_ALLOCATOR_STATS_COUNTER_TYPE}'")
-    add_definitions("-DOPENCV_ALLOCATOR_STATS_COUNTER_TYPE=${OPENCV_ALLOCATOR_STATS_COUNTER_TYPE}")
+    # add_definitions("-DOPENCV_ALLOCATOR_STATS_COUNTER_TYPE=\"${OPENCV_ALLOCATOR_STATS_COUNTER_TYPE}\"")
   endif()
 endif()
 
