From c5861878c330edfc806c31d164bf14801e613403 Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Wed, 21 Apr 2021 13:32:43 -0400
Subject: [PATCH 8/8] windres-cant-handle-spaces-in-defines

---
 modules/core/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/core/CMakeLists.txt b/modules/core/CMakeLists.txt
index b2797ab..794b2e8 100644
--- a/modules/core/CMakeLists.txt
+++ b/modules/core/CMakeLists.txt
@@ -127,7 +127,7 @@ elseif(HAVE_CXX11 OR DEFINED OPENCV_ALLOCATOR_STATS_COUNTER_TYPE)
   endif()
   if(DEFINED OPENCV_ALLOCATOR_STATS_COUNTER_TYPE)
     message(STATUS "Allocator metrics storage type: '${OPENCV_ALLOCATOR_STATS_COUNTER_TYPE}'")
-    add_definitions("-DOPENCV_ALLOCATOR_STATS_COUNTER_TYPE=${OPENCV_ALLOCATOR_STATS_COUNTER_TYPE}")
+    # add_definitions("-DOPENCV_ALLOCATOR_STATS_COUNTER_TYPE=\"${OPENCV_ALLOCATOR_STATS_COUNTER_TYPE}\"")
   endif()
 endif()
 
-- 
2.31.1

