--- opencv_contrib-4.5.3/modules/sfm/src/libmv_light/libmv/correspondence/CMakeLists.txt.orig
+++ opencv_contrib-4.5.3/modules/sfm/src/libmv_light/libmv/correspondence/CMakeLists.txt
@@ -12,6 +12,6 @@ TARGET_LINK_LIBRARIES(correspondence LINK_PRIVATE ${GLOG_LIBRARIES} multiview)
 IF(TARGET Eigen3::Eigen)
   TARGET_LINK_LIBRARIES(correspondence LINK_PUBLIC Eigen3::Eigen)
 ENDIF()
-
+TARGET_LINK_LIBRARIES(correspondence LINK_PRIVATE opencv_imgcodecs)

 LIBMV_INSTALL_LIB(correspondence)
