--- opencv-4.11.0/modules/highgui/CMakeLists.txt.orig	2025-01-08 13:47:46.000000000 +0100
+++ opencv-4.11.0/modules/highgui/CMakeLists.txt	2025-04-11 21:20:28.236694900 +0200
@@ -125,7 +125,7 @@
     endif()
 
     foreach(dt_dep ${qt_deps})
-      add_definitions(${Qt${QT_VERSION_MAJOR}${dt_dep}_DEFINITIONS})
+      link_libraries(${Qt${QT_VERSION_MAJOR}${dt_dep}})
       include_directories(${Qt${QT_VERSION_MAJOR}${dt_dep}_INCLUDE_DIRS})
       list(APPEND HIGHGUI_LIBRARIES ${Qt${QT_VERSION_MAJOR}${dt_dep}_LIBRARIES})
     endforeach()
