diff -baur opencv-2.4.7.orig/cmake/OpenCVGenPkgconfig.cmake opencv-2.4.7/cmake/OpenCVGenPkgconfig.cmake
--- opencv-2.4.7.orig/cmake/OpenCVGenPkgconfig.cmake	2013-11-12 20:31:27.253763202 +0000
+++ opencv-2.4.7/cmake/OpenCVGenPkgconfig.cmake	2013-11-12 20:31:54.923577113 +0000
@@ -46,7 +46,7 @@
 
   #need better solution....
   if(libpath MATCHES "3rdparty")
-    set(installDir "share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH}")
+    set(installDir "share/opencv/3rdparty/${OPENCV_LIB_INSTALL_PATH}")
   else()
     set(installDir "${OPENCV_LIB_INSTALL_PATH}")
   endif()
diff -baur opencv-2.4.7.orig/cmake/OpenCVModule.cmake opencv-2.4.7/cmake/OpenCVModule.cmake
--- opencv-2.4.7.orig/cmake/OpenCVModule.cmake	2013-11-12 20:31:27.253763202 +0000
+++ opencv-2.4.7/cmake/OpenCVModule.cmake	2013-11-12 20:35:37.805415871 +0000
@@ -807,7 +807,7 @@
   if(INSTALL_C_EXAMPLES AND NOT WIN32 AND EXISTS "${samples_path}")
     file(GLOB sample_files "${samples_path}/*")
     install(FILES ${sample_files}
-            DESTINATION share/OpenCV/samples/${module_id}
+            DESTINATION share/opencv/samples/${module_id}
             PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
   endif()
 endfunction()
diff -baur opencv-2.4.7.orig/CMakeLists.txt opencv-2.4.7/CMakeLists.txt
--- opencv-2.4.7.orig/CMakeLists.txt	2013-11-12 20:31:27.437095684 +0000
+++ opencv-2.4.7/CMakeLists.txt	2013-11-12 20:40:21.160171499 +0000
@@ -243,9 +243,9 @@
 if(ANDROID OR WIN32)
   set(OPENCV_DOC_INSTALL_PATH doc)
 elseif(INSTALL_TO_MANGLED_PATHS)
-  set(OPENCV_DOC_INSTALL_PATH share/OpenCV-${OPENCV_VERSION}/doc)
+  set(OPENCV_DOC_INSTALL_PATH share/doc/opencv-${OPENCV_VERSION})
 else()
-  set(OPENCV_DOC_INSTALL_PATH share/OpenCV/doc)
+  set(OPENCV_DOC_INSTALL_PATH share/doc/opencv)
 endif()
 
 if(WIN32)
@@ -282,7 +282,7 @@
     set(OPENCV_3P_LIB_INSTALL_PATH  "${OpenCV_INSTALL_BINARIES_PREFIX}staticlib${LIB_SUFFIX}")
   else()
     set(OPENCV_LIB_INSTALL_PATH     lib${LIB_SUFFIX})
-    set(OPENCV_3P_LIB_INSTALL_PATH  share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH})
+    set(OPENCV_3P_LIB_INSTALL_PATH  share/opencv/3rdparty/${OPENCV_LIB_INSTALL_PATH})
   endif()
   set(OPENCV_INCLUDE_INSTALL_PATH "include")
 
@@ -290,7 +290,7 @@
   if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
     set(OPENCV_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/cmake/opencv)
   else()
-    set(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
+    set(OPENCV_CONFIG_INSTALL_PATH share/opencv)
   endif()
 endif()
 
diff -baur opencv-2.4.7.orig/data/CMakeLists.txt opencv-2.4.7/data/CMakeLists.txt
--- opencv-2.4.7.orig/data/CMakeLists.txt	2013-11-12 20:31:27.280430213 +0000
+++ opencv-2.4.7/data/CMakeLists.txt	2013-11-12 20:31:54.923577113 +0000
@@ -5,6 +5,6 @@
   install(FILES ${HAAR_CASCADES} DESTINATION sdk/etc/haarcascades COMPONENT main)
   install(FILES ${LBP_CASCADES}  DESTINATION sdk/etc/lbpcascades  COMPONENT main)
 elseif(NOT WIN32)
-  install(FILES ${HAAR_CASCADES} DESTINATION share/OpenCV/haarcascades COMPONENT main)
-  install(FILES ${LBP_CASCADES}  DESTINATION share/OpenCV/lbpcascades  COMPONENT main)
+  install(FILES ${HAAR_CASCADES} DESTINATION share/opencv/haarcascades COMPONENT main)
+  install(FILES ${LBP_CASCADES}  DESTINATION share/opencv/lbpcascades  COMPONENT main)
 endif()
diff -baur opencv-2.4.7.orig/modules/java/CMakeLists.txt opencv-2.4.7/modules/java/CMakeLists.txt
--- opencv-2.4.7.orig/modules/java/CMakeLists.txt	2013-11-12 20:31:27.360429532 +0000
+++ opencv-2.4.7/modules/java/CMakeLists.txt	2013-11-12 20:36:41.984985273 +0000
@@ -279,7 +279,7 @@
   if(WIN32)
     set(JAR_INSTALL_DIR java)
   else(WIN32)
-    set(JAR_INSTALL_DIR share/OpenCV/java)
+    set(JAR_INSTALL_DIR share/opencv/java)
   endif(WIN32)
   install(FILES ${JAR_FILE} DESTINATION ${JAR_INSTALL_DIR} COMPONENT main)
 endif(ANDROID)
diff -baur opencv-2.4.7.orig/samples/c/CMakeLists.txt opencv-2.4.7/samples/c/CMakeLists.txt
--- opencv-2.4.7.orig/samples/c/CMakeLists.txt	2013-11-12 20:31:27.433762373 +0000
+++ opencv-2.4.7/samples/c/CMakeLists.txt	2013-11-12 20:31:54.923577113 +0000
@@ -54,6 +54,6 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB C_SAMPLES *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${C_SAMPLES}
-          DESTINATION share/OpenCV/samples/c
+          DESTINATION share/opencv/samples/c
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif ()
diff -baur opencv-2.4.7.orig/samples/cpp/CMakeLists.txt opencv-2.4.7/samples/cpp/CMakeLists.txt
--- opencv-2.4.7.orig/samples/cpp/CMakeLists.txt	2013-11-12 20:31:27.417095818 +0000
+++ opencv-2.4.7/samples/cpp/CMakeLists.txt	2013-11-12 20:31:55.110242982 +0000
@@ -83,6 +83,6 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB C_SAMPLES *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${C_SAMPLES}
-          DESTINATION share/OpenCV/samples/cpp
+          DESTINATION share/opencv/samples/cpp
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
diff -baur opencv-2.4.7.orig/samples/gpu/CMakeLists.txt opencv-2.4.7/samples/gpu/CMakeLists.txt
--- opencv-2.4.7.orig/samples/gpu/CMakeLists.txt	2013-11-12 20:31:27.407095885 +0000
+++ opencv-2.4.7/samples/gpu/CMakeLists.txt	2013-11-12 20:31:55.110242982 +0000
@@ -83,6 +83,6 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB install_list *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${install_list}
-          DESTINATION share/OpenCV/samples/${project}
+          DESTINATION share/opencv/samples/${project}
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
diff -baur opencv-2.4.7.orig/samples/gpu/performance/CMakeLists.txt opencv-2.4.7/samples/gpu/performance/CMakeLists.txt
--- opencv-2.4.7.orig/samples/gpu/performance/CMakeLists.txt	2013-11-12 20:31:27.407095885 +0000
+++ opencv-2.4.7/samples/gpu/performance/CMakeLists.txt	2013-11-12 20:31:55.110242982 +0000
@@ -29,6 +29,6 @@
 if(INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB GPU_FILES performance/*.cpp performance/*.h)
   install(FILES ${GPU_FILES}
-          DESTINATION share/OpenCV/samples/gpu/performance
+          DESTINATION share/opencv/samples/gpu/performance
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
diff -baur opencv-2.4.7.orig/samples/ocl/CMakeLists.txt opencv-2.4.7/samples/ocl/CMakeLists.txt
--- opencv-2.4.7.orig/samples/ocl/CMakeLists.txt	2013-11-12 20:31:27.397095952 +0000
+++ opencv-2.4.7/samples/ocl/CMakeLists.txt	2013-11-12 20:39:29.227200043 +0000
@@ -54,6 +54,6 @@
 if (INSTALL_C_EXAMPLES AND NOT WIN32)
   file(GLOB install_list *.c *.cpp *.jpg *.png *.data makefile.* build_all.sh *.dsp *.cmd )
   install(FILES ${install_list}
-          DESTINATION share/OpenCV/samples/${project}
+          DESTINATION share/opencv/samples/${project}
           PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
 endif()
