http://code.opencv.org/issues/4107

diff -Naur a/cmake/OpenCVCompilerOptions.cmake b/cmake/OpenCVCompilerOptions.cmake
--- a/cmake/OpenCVCompilerOptions.cmake	2015-02-25 18:10:31.000000000 +0600
+++ b/cmake/OpenCVCompilerOptions.cmake	2015-03-05 22:18:10.532856700 +0600
@@ -64,7 +64,6 @@
   add_extra_compiler_option(-W)
   add_extra_compiler_option(-Wall)
   add_extra_compiler_option(-Werror=return-type)
-  add_extra_compiler_option(-Werror=non-virtual-dtor)
   add_extra_compiler_option(-Werror=address)
   add_extra_compiler_option(-Werror=sequence-point)
   add_extra_compiler_option(-Wformat)
