diff --git a/StandAlone/CMakeLists.txt b/StandAlone/CMakeLists.txt
index 2a9a3c47..e3d46595 100644
--- a/StandAlone/CMakeLists.txt
+++ b/StandAlone/CMakeLists.txt
@@ -45,4 +45,7 @@ install(TARGETS glslangValidator
 
     install(TARGETS spirv-remap
             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+    install(TARGETS glslang-default-resource-limits
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif(ENABLE_GLSLANG_INSTALL)
