diff -ur b/glslang/Include/visibility.h a/glslang/Include/visibility.h
--- b/glslang/Include/visibility.h	2025-06-27 09:05:50.000000000 -0600
+++ a/glslang/Include/visibility.h	2025-09-21 19:07:03.064261500 -0600
@@ -51,4 +51,8 @@
 // Symbols marked with this macro are only meant for public use by the test suite
 // and do not appear in publicly installed headers. They are not considered to be
 // part of the glslang library ABI.
-#define GLSLANG_EXPORT_FOR_TESTS GLSLANG_EXPORT
+#if GLSLANG_EXPORTING
+    #define GLSLANG_EXPORT_FOR_TESTS GLSLANG_EXPORT
+#else
+    #define GLSLANG_EXPORT_FOR_TESTS
+#endif
diff -ur b/glslang/Public/ShaderLang.h a/glslang/Public/ShaderLang.h
--- b/glslang/Public/ShaderLang.h	2025-06-27 09:05:50.000000000 -0600
+++ a/glslang/Public/ShaderLang.h	2025-09-21 19:00:31.753263300 -0600
@@ -463,7 +463,7 @@
 //
 // N.B.: Destruct a linked program *before* destructing the shaders linked into it.
 //
-class TShader {
+class GLSLANG_EXPORT TShader {
 public:
     GLSLANG_EXPORT explicit TShader(EShLanguage);
     GLSLANG_EXPORT virtual ~TShader();
