--- glfw-3.3/src/CMakeLists.txt.orig	2019-04-16 19:10:44.731518900 -0400
+++ glfw-3.3/src/CMakeLists.txt	2019-04-16 20:05:45.961541300 -0400
@@ -158,7 +158,7 @@
 if (GLFW_INSTALL)
     install(TARGETS glfw
             EXPORT glfwTargets
-            RUNTIME DESTINATION "bin"
+            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
             ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
             LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 endif()
