--- libplist-2.0.0/cython/Makefile.am.orig	2015-01-17 19:17:30.724800000 +0300
+++ libplist-2.0.0/cython/Makefile.am	2015-01-17 19:17:43.813200000 +0300
@@ -20,8 +20,10 @@
 plist_LTLIBRARIES = plist.la
 plist_la_SOURCES = plist_util.c plist_util.h plist.pyx
 plist_la_CFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src $(PYTHON_CPPFLAGS) $(AM_CFLAGS) -Wno-shadow -Wno-redundant-decls -Wno-switch-default -Wno-strict-aliasing -Wno-implicit-function-declaration -fvisibility=default
-plist_la_LDFLAGS = -module -avoid-version $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
+plist_la_LDFLAGS = -module -avoid-version -no-undefined $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
 plist_la_LIBADD = $(top_builddir)/src/libplist.la
+otherincludedir = $(includedir)/plist/cython
+otherinclude_HEADERS = plist_util.h plist.pxd
 
 plist.c: plist.pyx $(PXDINCLUDES) $(PXIINCLUDES)
 
