--- libplist-2.2.0/cython/Makefile.am.orig	2015-01-17 19:17:30.724800000 +0300
+++ libplist-2.2.0/cython/Makefile.am	2015-01-17 19:17:43.813200000 +0300
@@ -39,8 +39,10 @@
 	-Wno-implicit-function-declaration \
 	-fvisibility=default
 
-plist_la_LDFLAGS = -module -avoid-version $(PYTHON_LIBS) $(AM_LDFLAGS)
+plist_la_LDFLAGS = -module -avoid-version -no-undefined $(PYTHON_LIBS) $(AM_LDFLAGS)
 plist_la_LIBADD = $(top_builddir)/src/libplist-2.0.la
+otherincludedir = $(includedir)/plist/cython
+otherinclude_HEADERS = plist_util.h plist.pxd 
 
 plist.c: plist.pyx $(PXDINCLUDES) $(PXIINCLUDES)
 
