--- libplist-2.1.0/configure.ac.orig	2020-04-26 14:50:09.337844300 +0200
+++ libplist-2.1.0/configure.ac	2020-04-26 14:50:22.958195000 +0200
@@ -115,6 +115,9 @@
             AM_PATH_PYTHON(2.3)
             AC_PROG_CYTHON(0.17.0)
             CYTHON_PYTHON
+
+            PYTHON_SO=`$PYTHON -c "import distutils.sysconfig, sys; get = distutils.sysconfig.get_config_var; sys.stdout.write(get('EXT_SUFFIX') or get('SO') or '.so');"`
+            AC_SUBST(PYTHON_SO)
 else
             CYTHON=false
 fi
--- libplist-2.2.0/cython/Makefile.am.orig	2020-04-26 14:50:32.957877900 +0200
+++ libplist-2.2.0/cython/Makefile.am	2020-04-26 14:50:50.787782000 +0200
@@ -39,7 +39,7 @@
 	-Wno-implicit-function-declaration \
 	-fvisibility=default
 
-plist_la_LDFLAGS = -module -avoid-version -no-undefined $(PYTHON_LIBS) $(AM_LDFLAGS)
+plist_la_LDFLAGS = -module -avoid-version -no-undefined $(PYTHON_LIBS) $(AM_LDFLAGS) -shrext $(PYTHON_SO)
 plist_la_LIBADD = $(top_builddir)/src/libplist-2.0.la
 otherincludedir = $(includedir)/plist/cython
 otherinclude_HEADERS = plist_util.h plist.pxd
