--- a/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/thirdparty/diy/vtkmdiy/include/vtkmdiy/mpi/no-mpi.hpp
+++ b/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/thirdparty/diy/vtkmdiy/include/vtkmdiy/mpi/no-mpi.hpp
@@ -1,9 +1,9 @@
 #ifndef VTKMDIY_MPI_NO_MPI_HPP
 #define VTKMDIY_MPI_NO_MPI_HPP
 
+#include <cstdint> // uintptr_t
 #include <stdexcept> // std::runtime_error
 
-
 static const int MPI_SUCCESS = 0;
 static const int MPI_ANY_SOURCE = -1;
