diff -Naur libsoup-2.48.0-orig/libsoup/soup-cache.c libsoup-2.48.0/libsoup/soup-cache.c
--- libsoup-2.48.0-orig/libsoup/soup-cache.c	2014-09-23 03:02:43.000000000 +0500
+++ libsoup-2.48.0/libsoup/soup-cache.c	2014-09-23 15:15:23.836600000 +0500
@@ -51,7 +51,7 @@
 static void soup_cache_session_feature_init (SoupSessionFeatureInterface *feature_interface, gpointer interface_data);
 
 static SoupContentProcessorInterface *soup_cache_default_content_processor_interface;
-static void soup_cache_content_processor_init (SoupContentProcessorInterface *interface, gpointer interface_data);
+static void soup_cache_content_processor_init (SoupContentProcessorInterface *interface_name, gpointer interface_data);
 
 #define DEFAULT_MAX_SIZE 50 * 1024 * 1024
 #define MAX_ENTRY_DATA_PERCENTAGE 10 /* Percentage of the total size
diff -Naur libsoup-2.48.0-orig/libsoup/soup-content-decoder.c libsoup-2.48.0/libsoup/soup-content-decoder.c
--- libsoup-2.48.0-orig/libsoup/soup-content-decoder.c	2014-06-18 19:42:51.000000000 +0500
+++ libsoup-2.48.0/libsoup/soup-content-decoder.c	2014-09-23 15:15:43.508200000 +0500
@@ -59,7 +59,7 @@
 static void soup_content_decoder_session_feature_init (SoupSessionFeatureInterface *feature_interface, gpointer interface_data);
 
 static SoupContentProcessorInterface *soup_content_decoder_default_content_processor_interface;
-static void soup_content_decoder_content_processor_init (SoupContentProcessorInterface *interface, gpointer interface_data);
+static void soup_content_decoder_content_processor_init (SoupContentProcessorInterface *interface_name, gpointer interface_data);
 
 
 G_DEFINE_TYPE_WITH_CODE (SoupContentDecoder, soup_content_decoder, G_TYPE_OBJECT,
diff -Naur libsoup-2.48.0-orig/libsoup/soup-content-processor.c libsoup-2.48.0/libsoup/soup-content-processor.c
--- libsoup-2.48.0-orig/libsoup/soup-content-processor.c	2014-06-18 19:42:51.000000000 +0500
+++ libsoup-2.48.0/libsoup/soup-content-processor.c	2014-09-23 15:16:35.081800000 +0500
@@ -11,7 +11,7 @@
 #include "soup-content-processor.h"
 #include "soup.h"
 
-static void soup_content_processor_default_init (SoupContentProcessorInterface *interface);
+static void soup_content_processor_default_init (SoupContentProcessorInterface *interface_name);
 
 G_DEFINE_INTERFACE (SoupContentProcessor, soup_content_processor, G_TYPE_OBJECT)
 
@@ -25,10 +25,10 @@
 }
 
 static void
-soup_content_processor_default_init (SoupContentProcessorInterface *interface)
+soup_content_processor_default_init (SoupContentProcessorInterface *interface_name)
 {
-	interface->processing_stage = SOUP_STAGE_INVALID;
-	interface->wrap_input = soup_content_processor_real_wrap_input;
+	interface_name->processing_stage = SOUP_STAGE_INVALID;
+	interface_name->wrap_input = soup_content_processor_real_wrap_input;
 }
 
 GInputStream *
diff -Naur libsoup-2.48.0-orig/libsoup/soup-content-sniffer.c libsoup-2.48.0/libsoup/soup-content-sniffer.c
--- libsoup-2.48.0-orig/libsoup/soup-content-sniffer.c	2014-06-18 19:42:51.000000000 +0500
+++ libsoup-2.48.0/libsoup/soup-content-sniffer.c	2014-09-23 15:16:58.232200000 +0500
@@ -38,7 +38,7 @@
 static void soup_content_sniffer_session_feature_init (SoupSessionFeatureInterface *feature_interface, gpointer interface_data);
 
 static SoupContentProcessorInterface *soup_content_sniffer_default_content_processor_interface;
-static void soup_content_sniffer_content_processor_init (SoupContentProcessorInterface *interface, gpointer interface_data);
+static void soup_content_sniffer_content_processor_init (SoupContentProcessorInterface *interface_name, gpointer interface_data);
 
 
 G_DEFINE_TYPE_WITH_CODE (SoupContentSniffer, soup_content_sniffer, G_TYPE_OBJECT,
diff -Naur libsoup-2.48.0-orig/libsoup/soup-server.h libsoup-2.48.0/libsoup/soup-server.h
--- libsoup-2.48.0-orig/libsoup/soup-server.h	2014-09-23 03:02:43.000000000 +0500
+++ libsoup-2.48.0/libsoup/soup-server.h	2014-09-23 15:13:34.844000000 +0500
@@ -80,7 +80,7 @@
 
 SOUP_AVAILABLE_IN_2_48
 gboolean        soup_server_listen             (SoupServer               *server,
-					        GSocketAddress           *interface,
+					        GSocketAddress           *interface_name,
 					        SoupServerListenOptions   options,
 					        GError                  **error);
 SOUP_AVAILABLE_IN_2_48
