--- libsoup-2.52.0/m4/vapigen.m4.orig	2015-01-14 14:51:42.117600000 +0300
+++ libsoup-2.52.0/m4/vapigen.m4	2015-01-14 14:52:07.297000000 +0300
@@ -85,11 +85,11 @@
 		  [yes], [
 		      VAPIGEN=`$PKG_CONFIG --variable=vapigen vapigen`
 		      VAPIGEN_MAKEFILE=`$PKG_CONFIG --variable=datadir vapigen`/vala/Makefile.vapigen
-		      VAPIDIR=`$PKG_CONFIG --variable=vapidir vapigen`
+		      VAPIDIR=`$PKG_CONFIG --dont-define-prefix --variable=vapidir vapigen`
 		      AS_IF([ test "x$2" = "x"], [
-		          VAPIGEN_VAPIDIR=`$PKG_CONFIG --variable=vapidir vapigen`
+		          VAPIGEN_VAPIDIR=`$PKG_CONFIG --dont-define-prefix --variable=vapidir $vapigen_pkg_name`
 		        ], [
-		          VAPIGEN_VAPIDIR=`$PKG_CONFIG --variable=vapidir_versioned vapigen`
+		          VAPIGEN_VAPIDIR=`$PKG_CONFIG --dont-define-prefix --variable=vapidir_versioned $vapigen_pkg_name`
 		        ])
 		    ],
 		 [no], [
