diff --git a/libsoup/soup-init.c b/libsoup/soup-init.c
index 92eb837..c4950ce 100644
--- a/libsoup/soup-init.c
+++ b/libsoup/soup-init.c
@@ -38,6 +38,10 @@ soup_init (void)
 
 #if defined (G_OS_WIN32)
 
+BOOL WINAPI DllMain (HINSTANCE hinstDLL,
+                     DWORD     fdwReason,
+                     LPVOID    lpvReserved);
+
 BOOL WINAPI
 DllMain (HINSTANCE hinstDLL,
          DWORD     fdwReason,
