diff --git a/libsoup/soup-session.c b/libsoup/soup-session.c
index 6420183..ca8d80c 100644
--- a/libsoup/soup-session.c
+++ b/libsoup/soup-session.c
@@ -77,36 +77,11 @@
  * subtypes) have a #SoupContentDecoder by default.
  **/
 
-#if defined (G_OS_WIN32)
-#define WIN32_LEAN_AND_MEAN
-#include <windows.h>
-
-HMODULE soup_dll;
-
-BOOL WINAPI
-DllMain (HINSTANCE hinstDLL,
-         DWORD     fdwReason,
-         LPVOID    lpvReserved)
-{
-	switch (fdwReason) {
-	case DLL_PROCESS_ATTACH:
-		soup_dll = hinstDLL;
-		break;
-
-	case DLL_THREAD_DETACH:
-
-	default:
-		/* do nothing */
-		;
-	}
-}
-#endif
-
 static void
 soup_init (void)
 {
 #ifdef G_OS_WIN32
-	char *basedir = g_win32_get_package_installation_directory_of_module (soup_dll);
+	char *basedir = g_win32_get_package_installation_directory_of_module (NULL);
 	char *localedir = g_build_filename (basedir, "share", "locale", NULL);
 	bindtextdomain (GETTEXT_PACKAGE, localedir);
 	g_free (localedir);
-- 
cgit v0.10.2
