diff --git a/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.cc b/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.cc
index 8ed9ace..204d5f4 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.cc
+++ b/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.cc
@@ -14,7 +14,7 @@
 #include "gpu/command_buffer/common/shared_image_usage.h"
 #include "ui/gfx/buffer_format_util.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "ui/gfx/win/d3d_shared_fence.h"
 #endif
 
@@ -171,14 +171,14 @@ void SharedImageInterface::Release() const {
   }
 }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void SharedImageInterface::UpdateSharedImage(
     const SyncToken& sync_token,
     scoped_refptr<gfx::D3DSharedFence> d3d_shared_fence,
     const Mailbox& mailbox) {
   NOTIMPLEMENTED_LOG_ONCE();
 }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
 void SharedImageInterface::CreateSharedImagePool(
     const SharedImagePoolId& pool_id,
diff --git a/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.h b/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.h
index efac09c..8e47d47 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.h
+++ b/src/3rdparty/chromium/gpu/command_buffer/client/shared_image_interface.h
@@ -41,7 +41,7 @@ namespace gfx {
 class GpuFence;
 class Size;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 class D3DSharedFence;
 #endif
 }  // namespace gfx
@@ -327,7 +327,7 @@ class GPU_EXPORT SharedImageInterface
       bool register_with_image_pipe) = 0;
 #endif  // BUILDFLAG(IS_FUCHSIA)
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   // Update fence between processes. Register D3DSharedFence in GPU process
   // first and then use DXGIHandleToken to identify the fence between processes
   // and pass signaled fence value from current process to GPU process.
@@ -335,7 +335,7 @@ class GPU_EXPORT SharedImageInterface
       const SyncToken& sync_token,
       scoped_refptr<gfx::D3DSharedFence> d3d_shared_fence,
       const Mailbox& mailbox);
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
   // Generates an unverified SyncToken that is released after all previous
   // commands on this interface have executed on the service side.
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/BUILD.gn b/src/3rdparty/chromium/gpu/command_buffer/service/BUILD.gn
index 40c4f28..0d0deba 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/BUILD.gn
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/BUILD.gn
@@ -604,7 +604,18 @@ target(link_target_type, "gles2_sources") {
     }
   }
 
-  if (is_win) {
+  if (is_mingw) {
+    sources += [
+      "shared_image/d3d_image_representation.cc",
+      "shared_image/d3d_image_representation.h",
+      "shared_image/d3d_image_utils.cc",
+      "shared_image/d3d_image_utils.h",
+    ]
+    libs = [ "dxguid" ]
+    all_dependent_configs = [ ":service_win_linker_flags" ]
+  }
+
+  if (is_msvc) {
     sources += [
       "dxgi_shared_handle_manager.cc",
       "dxgi_shared_handle_manager.h",
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/d3d_image_backing.h b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/d3d_image_backing.h
index 90bb3a4..42c5af2 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/d3d_image_backing.h
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/d3d_image_backing.h
@@ -115,8 +115,10 @@ class GPU_GLES2_EXPORT D3DImageBacking final
       std::vector<wgpu::TextureFormat> view_formats,
       scoped_refptr<SharedContextState> context_state) override;
 #endif
+#if BUILDFLAG(IS_MSVC)
   void UpdateExternalFence(
       scoped_refptr<gfx::D3DSharedFence> external_fence) override;
+#endif
 
   bool BeginAccessD3D11(Microsoft::WRL::ComPtr<ID3D11Device> d3d11_device,
                         bool write_access);
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.cc b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.cc
index 6878fb8..888aa44 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.cc
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.cc
@@ -20,7 +20,7 @@
 #include "gpu/command_buffer/service/shared_image/shared_image_representation.h"
 #include "third_party/skia/include/core/SkColorSpace.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "ui/gfx/win/d3d_shared_fence.h"
 #endif
 
@@ -299,7 +299,7 @@ SharedImageBacking::ProduceLegacyOverlay(SharedImageManager* manager,
 }
 #endif
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void SharedImageBacking::UpdateExternalFence(
     scoped_refptr<gfx::D3DSharedFence> external_fence) {
   NOTIMPLEMENTED_LOG_ONCE();
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.h b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.h
index 9440141..0982009 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.h
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_backing.h
@@ -363,7 +363,7 @@ class GPU_GLES2_EXPORT SharedImageBacking {
   ProduceLegacyOverlay(SharedImageManager* manager, MemoryTypeTracker* tracker);
 #endif
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   virtual void UpdateExternalFence(
       scoped_refptr<gfx::D3DSharedFence> external_fence);
 #endif
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.cc b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.cc
index 77a72e4..39b472f 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.cc
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.cc
@@ -255,7 +255,7 @@ SharedImageFactory::SharedImageFactory(
     factories_.push_back(std::move(gl_texture_backing_factory));
   }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   if (gl::DirectCompositionSupported()) {
     factories_.push_back(
         std::make_unique<DCompImageBackingFactory>(context_state_));
@@ -276,7 +276,7 @@ SharedImageFactory::SharedImageFactory(
             /*supports_cpu_upload=*/true);
     factories_.push_back(std::move(gl_texture_backing_factory));
   }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
 #if BUILDFLAG(ENABLE_VULKAN)
   // If Chrome and ANGLE are sharing the same vulkan device queue, AngleVulkan
@@ -692,7 +692,7 @@ bool SharedImageFactory::PresentSwapChain(const Mailbox& mailbox) {
   (*it)->PresentSwapChain();
   return true;
 }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
 #if BUILDFLAG(IS_FUCHSIA)
 void SharedImageFactory::RegisterSysmemBufferCollection(
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.h b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.h
index 24fce45..534c006 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.h
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_factory.h
@@ -105,7 +105,7 @@ class GPU_GLES2_EXPORT SharedImageFactory {
   bool HasImages() const { return !shared_images_.empty(); }
   void DestroyAllSharedImages(bool have_context);
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   bool CreateSwapChain(const Mailbox& front_buffer_mailbox,
                        const Mailbox& back_buffer_mailbox,
                        viz::SharedImageFormat format,
@@ -115,7 +115,7 @@ class GPU_GLES2_EXPORT SharedImageFactory {
                        SkAlphaType alpha_type,
                        gpu::SharedImageUsageSet usage);
   bool PresentSwapChain(const Mailbox& mailbox);
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
 #if BUILDFLAG(IS_FUCHSIA)
   void RegisterSysmemBufferCollection(zx::eventpair service_handle,
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.cc b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.cc
index b9bfa8c..39d3df5 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.cc
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.cc
@@ -28,7 +28,7 @@
 #include "gpu/command_buffer/service/shared_image/shared_image_representation.h"
 #include "ui/gl/trace_util.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "gpu/command_buffer/service/dxgi_shared_handle_manager.h"
 #include "ui/gfx/win/d3d_shared_fence.h"
 #include "ui/gl/direct_composition_support.h"
@@ -262,7 +262,7 @@ class SCOPED_LOCKABLE SharedImageManager::AutoLock {
 SharedImageManager::SharedImageManager(bool thread_safe,
                                        bool display_context_on_another_thread)
     : display_context_on_another_thread_(display_context_on_another_thread)
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
       ,
       dxgi_shared_handle_manager_(
           base::MakeRefCounted<DXGISharedHandleManager>())
@@ -607,7 +607,7 @@ SharedImageManager::ProduceLegacyOverlay(const Mailbox& mailbox,
 }
 #endif
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void SharedImageManager::UpdateExternalFence(
     const Mailbox& mailbox,
     scoped_refptr<gfx::D3DSharedFence> external_fence) {
diff --git a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.h b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.h
index 531c12e..f330d36 100644
--- a/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.h
+++ b/src/3rdparty/chromium/gpu/command_buffer/service/shared_image/shared_image_manager.h
@@ -19,7 +19,7 @@
 #include "gpu/gpu_gles2_export.h"
 #include "gpu/vulkan/buildflags.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 namespace gfx {
 class D3DSharedFence;
 }
@@ -128,7 +128,7 @@ class GPU_GLES2_EXPORT SharedImageManager
       MemoryTypeTracker* ref);
 #endif
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   void UpdateExternalFence(const Mailbox& mailbox,
                            scoped_refptr<gfx::D3DSharedFence> external_fence);
 #endif
@@ -158,7 +158,7 @@ class GPU_GLES2_EXPORT SharedImageManager
   // method is to facilitate pageflip testing on the viz thread.
   scoped_refptr<gfx::NativePixmap> GetNativePixmap(const gpu::Mailbox& mailbox);
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   const scoped_refptr<DXGISharedHandleManager>& dxgi_shared_handle_manager()
       const {
     return dxgi_shared_handle_manager_;
@@ -176,7 +176,7 @@ class GPU_GLES2_EXPORT SharedImageManager
 
   bool is_registered_as_memory_dump_provider_ = false;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   scoped_refptr<DXGISharedHandleManager> dxgi_shared_handle_manager_;
 #endif
 
diff --git a/src/3rdparty/chromium/gpu/config/BUILD.gn b/src/3rdparty/chromium/gpu/config/BUILD.gn
index f49a550..28c58e3 100644
--- a/src/3rdparty/chromium/gpu/config/BUILD.gn
+++ b/src/3rdparty/chromium/gpu/config/BUILD.gn
@@ -244,7 +244,11 @@ source_set("config_sources") {
     if (!is_qtwebengine) {
       deps += [ "//third_party/microsoft_dxheaders:dxguids" ]
     }
-    libs = [ "dxgi.lib" ]
+    if (is_msvc) {
+      libs = [ "dxgi.lib" ]
+    } else {
+      libs = [ "dxgi" ]
+    }
   }
   if (is_mac) {
     sources += [ "gpu_info_collector_mac.mm" ]
diff --git a/src/3rdparty/chromium/gpu/config/gpu_info_collector_win.cc b/src/3rdparty/chromium/gpu/config/gpu_info_collector_win.cc
index 4bd1190..0e96b92 100644
--- a/src/3rdparty/chromium/gpu/config/gpu_info_collector_win.cc
+++ b/src/3rdparty/chromium/gpu/config/gpu_info_collector_win.cc
@@ -5,9 +5,11 @@
 #include "gpu/config/gpu_info_collector.h"
 
 // C system before C++ system.
+//#if BUILDFLAG(USE_ML)
-#include <DirectML.h>
+#include <directx/DirectML.h>
+//#endif
 #include <d3d11.h>
 #include <d3d11_3.h>
 #include <dxgi.h>
 #include <dxgi1_6.h>
 #include <stddef.h>
@@ -35,7 +37,7 @@
 #include "third_party/microsoft_dxheaders/src/include/directx/dxcore.h"
 #else
 #include <d3d12.h>
-#include <dxcore.h>
+#include <directx/dxcore.h>
 #endif
 #include "third_party/re2/src/re2/re2.h"
 #include "ui/gl/direct_composition_support.h"
@@ -453,7 +455,7 @@ void GetGpuSupportedDirectXVersion(uint32_t& d3d12_feature_level,
     // Check to see if the adapter supports Direct3D 12.
     for (auto level : feature_levels) {
-      if (SUCCEEDED(d3d12_create_device_proc(dxgi_adapter.Get(), level,
-                                             _uuidof(ID3D12Device),
-                                             &d3d12_device))) {
+      if (SUCCEEDED(d3d12_create_device_proc(
+              dxgi_adapter.Get(), level, IID_ID3D12Device,
+              reinterpret_cast<void**>(d3d12_device.GetAddressOf())))) {
         d3d12_feature_level = level;
         break;
diff --git a/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.cc b/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.cc
index 40801d4..2a2e901 100644
--- a/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.cc
+++ b/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.cc
@@ -14,7 +14,7 @@
 #include "ui/gfx/gpu_fence.h"
 #include "ui/gfx/gpu_memory_buffer.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "ui/gfx/win/d3d_shared_fence.h"
 #endif
 
@@ -228,7 +228,7 @@ void ClientSharedImageInterface::CopyToGpuMemoryBuffer(
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void ClientSharedImageInterface::CopyToGpuMemoryBufferAsync(
     const SyncToken& sync_token,
     const Mailbox& mailbox,
     base::OnceCallback<void(bool)> callback) {
   proxy_->CopyToGpuMemoryBufferAsync(sync_token, mailbox, std::move(callback));
 }
diff --git a/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.h b/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.h
index 88cbde8..cc1bb8e 100644
--- a/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.h
+++ b/src/3rdparty/chromium/gpu/ipc/client/client_shared_image_interface.h
@@ -78,7 +78,7 @@ class GPU_EXPORT ClientSharedImageInterface : public SharedImageInterface {
       const SharedImageInfo& si_info) override;
   void CopyToGpuMemoryBuffer(const SyncToken& sync_token,
                              const Mailbox& mailbox) override;
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   void CopyToGpuMemoryBufferAsync(
       const SyncToken& sync_token,
       const Mailbox& mailbox,
diff --git a/src/3rdparty/chromium/gpu/ipc/client/shared_image_interface_proxy.cc b/src/3rdparty/chromium/gpu/ipc/client/shared_image_interface_proxy.cc
index 5b02cc3..74e61ed 100644
--- a/src/3rdparty/chromium/gpu/ipc/client/shared_image_interface_proxy.cc
+++ b/src/3rdparty/chromium/gpu/ipc/client/shared_image_interface_proxy.cc
@@ -21,7 +21,7 @@
 #include "ui/gfx/buffer_types.h"
 #include "ui/gfx/gpu_fence.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "ui/gfx/win/d3d_shared_fence.h"
 #endif
 
@@ -243,7 +243,7 @@ void SharedImageInterfaceProxy::CopyToGpuMemoryBuffer(
   }
 }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void SharedImageInterfaceProxy::CopyToGpuMemoryBufferAsync(
     const SyncToken& sync_token,
     const Mailbox& mailbox,
@@ -307,7 +307,7 @@ void SharedImageInterfaceProxy::CopyNativeGmbToSharedMemoryAsync(
 bool SharedImageInterfaceProxy::IsConnected() {
   return host_->IsConnected();
 }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
 void SharedImageInterfaceProxy::UpdateSharedImage(const SyncToken& sync_token,
                                                   const Mailbox& mailbox) {
@@ -499,7 +499,7 @@ SharedImageInterfaceProxy::CreateSwapChain(viz::SharedImageFormat format,
                                            GrSurfaceOrigin surface_origin,
                                            SkAlphaType alpha_type,
                                            gpu::SharedImageUsageSet usage) {
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   const SwapChainMailboxes mailboxes = {Mailbox::Generate(),
                                         Mailbox::Generate()};
   auto params = mojom::CreateSwapChainParams::New();
@@ -526,12 +526,12 @@ SharedImageInterfaceProxy::CreateSwapChain(viz::SharedImageFormat format,
   return mailboxes;
 #else
   NOTREACHED();
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 }
 
 void SharedImageInterfaceProxy::PresentSwapChain(const SyncToken& sync_token,
                                                  const Mailbox& mailbox) {
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   std::vector<SyncToken> dependencies =
       GenerateDependenciesFromSyncToken(std::move(sync_token), host_);
   {
@@ -545,7 +545,7 @@ void SharedImageInterfaceProxy::PresentSwapChain(const SyncToken& sync_token,
   }
 #else
   NOTREACHED();
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 }
 
 #if BUILDFLAG(IS_FUCHSIA)
diff --git a/src/3rdparty/chromium/gpu/ipc/common/BUILD.gn b/src/3rdparty/chromium/gpu/ipc/common/BUILD.gn
index 2886b9a..de10731 100644
--- a/src/3rdparty/chromium/gpu/ipc/common/BUILD.gn
+++ b/src/3rdparty/chromium/gpu/ipc/common/BUILD.gn
diff --git a/src/3rdparty/chromium/gpu/ipc/service/BUILD.gn b/src/3rdparty/chromium/gpu/ipc/service/BUILD.gn
index 973cd33..73c19a3 100644
--- a/src/3rdparty/chromium/gpu/ipc/service/BUILD.gn
+++ b/src/3rdparty/chromium/gpu/ipc/service/BUILD.gn
@@ -100,11 +100,19 @@ component("service") {
       "gpu_memory_buffer_factory_dxgi.h",
       "image_transport_surface_win.cc",
     ]
-    libs += [
-      "d3d11.lib",
-      "dcomp.lib",
-      "dxguid.lib",
-    ]
+    if (is_msvc) {
+      libs += [
+        "d3d11.lib",
+        "dcomp.lib",
+        "dxguid.lib",
+      ]
+    } else {
+      libs += [
+        "d3d11",
+        "dcomp",
+        "dxguid",
+      ]
+    }
   }
   if (is_apple) {
     sources += [
diff --git a/src/3rdparty/chromium/gpu/ipc/service/gpu_channel.cc b/src/3rdparty/chromium/gpu/ipc/service/gpu_channel.cc
index d4df75d..a000f8b 100644
--- a/src/3rdparty/chromium/gpu/ipc/service/gpu_channel.cc
+++ b/src/3rdparty/chromium/gpu/ipc/service/gpu_channel.cc
@@ -695,10 +695,15 @@ void GpuChannelMessageFilter::CopyToGpuMemoryBufferAsync(
          CopyToGpuMemoryBufferAsyncCallback callback) {
         if (!channel) {
           std::move(callback).Run(false);
+          return;
         }
+#if BUILDFLAG(IS_MSVC)
         channel->shared_image_stub()->CopyToGpuMemoryBufferAsync(
             mailbox, std::move(callback));
+#else
+        std::move(callback).Run(false);
+#endif
       },
       gpu_channel_->AsWeakPtr(), mailbox,
       base::BindPostTask(base::SequencedTaskRunner::GetCurrentDefault(),
                         std::move(callback)));
diff --git a/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.cc b/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.cc
index 91caf7f..1541e52 100644
--- a/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.cc
+++ b/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.cc
@@ -28,7 +28,7 @@
 #include "ui/gfx/gpu_memory_buffer.h"
 #include "ui/gl/gl_context.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "ui/gfx/win/d3d_shared_fence.h"
 #endif
 
@@ -160,7 +160,7 @@ void SharedImageStub::ExecuteDeferredRequest(
           std::move(request->get_destroy_shared_image_pool()));
       break;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
     case mojom::DeferredSharedImageRequest::Tag::kCreateSwapChain:
       OnCreateSwapChain(std::move(request->get_create_swap_chain()));
       break;
@@ -184,7 +184,7 @@ void SharedImageStub::ExecuteDeferredRequest(
       OnUnregisterDxgiFence(unregister.mailbox, unregister.dxgi_token);
       break;
     }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
   }
 }
 
@@ -414,7 +414,7 @@ void SharedImageStub::OnDestroySharedImage(const Mailbox& mailbox) {
     return;
   }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   registered_dxgi_fences_.erase(mailbox);
 #endif
 }
@@ -432,7 +432,7 @@ void SharedImageStub::OnCopyToGpuMemoryBuffer(const Mailbox& mailbox) {
   }
 }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void SharedImageStub::CopyToGpuMemoryBufferAsync(
     const Mailbox& mailbox,
     base::OnceCallback<void(bool)> callback) {
@@ -563,7 +563,7 @@ void SharedImageStub::OnUnregisterDxgiFence(const Mailbox& mailbox,
   mailbox_fences.erase(fence_it);
 }
 
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
 #if BUILDFLAG(IS_FUCHSIA)
 void SharedImageStub::RegisterSysmemBufferCollection(
diff --git a/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.h b/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.h
index 6b0e287..2437c6e 100644
--- a/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.h
+++ b/src/3rdparty/chromium/gpu/ipc/service/shared_image_stub.h
@@ -18,7 +18,7 @@
 #include "ui/gfx/gpu_extra_info.h"
 
 namespace gfx {
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 class D3DSharedFence;
 #endif
 
@@ -85,7 +85,7 @@ class GPU_IPC_SERVICE_EXPORT SharedImageStub : public MemoryTracker {
   bool UpdateSharedImage(const Mailbox& mailbox,
                          gfx::GpuFenceHandle in_fence_handle);
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   void CopyToGpuMemoryBufferAsync(const Mailbox& mailbox,
                                   base::OnceCallback<void(bool)> callback);
 #endif
@@ -117,7 +117,7 @@ class GPU_IPC_SERVICE_EXPORT SharedImageStub : public MemoryTracker {
   void OnDestroySharedImage(const Mailbox& mailbox);
   void OnRegisterSharedImageUploadBuffer(base::ReadOnlySharedMemoryRegion shm);
   void OnCopyToGpuMemoryBuffer(const Mailbox& mailbox);
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   void OnCreateSwapChain(mojom::CreateSwapChainParamsPtr params);
   void OnPresentSwapChain(const Mailbox& mailbox);
   void OnRegisterDxgiFence(const Mailbox& mailbox,
@@ -128,7 +128,7 @@ class GPU_IPC_SERVICE_EXPORT SharedImageStub : public MemoryTracker {
                          uint64_t fence_value);
   void OnUnregisterDxgiFence(const Mailbox& mailbox,
                              gfx::DXGIHandleToken dxgi_token);
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
   void OnCreateSharedImagePool(mojom::CreateSharedImagePoolParamsPtr params);
   void OnDestroySharedImagePool(mojom::DestroySharedImagePoolParamsPtr params);
@@ -159,7 +159,7 @@ class GPU_IPC_SERVICE_EXPORT SharedImageStub : public MemoryTracker {
   base::ReadOnlySharedMemoryRegion upload_memory_;
   base::ReadOnlySharedMemoryMapping upload_memory_mapping_;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   // Fences held by external processes. Registered and signaled from ipc
   // channel. Using DXGIHandleToken to identify the fence.
   base::flat_map<Mailbox, base::flat_set<scoped_refptr<gfx::D3DSharedFence>>>
diff --git a/src/3rdparty/chromium/gpu/ipc/common/device_perf_info_mojom_traits.cc b/src/3rdparty/chromium/gpu/ipc/common/device_perf_info_mojom_traits.cc
index 98cb19e..2e9c4d2 100644
--- a/src/3rdparty/chromium/gpu/ipc/common/device_perf_info_mojom_traits.cc
+++ b/src/3rdparty/chromium/gpu/ipc/common/device_perf_info_mojom_traits.cc
@@ -7,6 +7,10 @@
 
 #include "build/build_config.h"
 
+#if BUILDFLAG(IS_WIN) && !defined(D3D_FEATURE_LEVEL_1_0_GENERIC)
+#define D3D_FEATURE_LEVEL_1_0_GENERIC static_cast<D3D_FEATURE_LEVEL>(0x1001)
+#endif
+
 namespace mojo {
 
 #if BUILDFLAG(IS_WIN)
