diff --git a/src/3rdparty/chromium/third_party/webrtc/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/BUILD.gn
index 253f5d0..6778fc0 100644
--- a/src/3rdparty/chromium/third_party/webrtc/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/BUILD.gn
@@ -422,7 +422,7 @@ config("common_config") {
       ]
     }
 
-    if (is_win && !is_clang) {
+    if (is_msvc && !is_clang) {
       # MSVC warning suppressions (needed to use Abseil).
       # TODO(bugs.webrtc.org/9274): Remove these warnings as soon as MSVC allows
       # external headers warning suppression (or fix them upstream).
@@ -486,8 +486,8 @@ config("common_config") {
 
   if (!build_with_chromium && rtc_win_undef_unicode) {
     cflags += [
-      "/UUNICODE",
-      "/U_UNICODE",
+      "-UUNICODE",
+      "-U_UNICODE",
     ]
   }
 
diff --git a/src/3rdparty/chromium/third_party/webrtc/common_audio/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/common_audio/BUILD.gn
index 4845bc7..8b86311 100644
--- a/src/3rdparty/chromium/third_party/webrtc/common_audio/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/common_audio/BUILD.gn
@@ -248,7 +248,7 @@ if (current_cpu == "x86" || current_cpu == "x64") {
       "resampler/sinc_resampler_sse.cc",
     ]
 
-    if (is_posix || is_fuchsia) {
+    if (is_posix || is_fuchsia || is_mingw) {
       cflags = [ "-msse2" ]
     }
 
@@ -267,7 +267,7 @@ if (current_cpu == "x86" || current_cpu == "x64") {
       "resampler/sinc_resampler_avx2.cc",
     ]
 
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX2" ]
     } else {
       cflags = [
diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/BUILD.gn
index a0bda5f..5835b05 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/BUILD.gn
@@ -261,7 +261,7 @@ if (current_cpu == "x86" || current_cpu == "x64") {
       "vector_math_avx2.cc",
     ]
 
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX2" ]
     } else {
       cflags = [
diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/BUILD.gn
index 025794d..8cab438 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/BUILD.gn
@@ -121,7 +121,7 @@ rtc_source_set("vector_math") {
 if (current_cpu == "x86" || current_cpu == "x64") {
   rtc_library("vector_math_avx2") {
     sources = [ "vector_math_avx2.cc" ]
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX2" ]
     } else {
       cflags = [
diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/BUILD.gn
index eaedf62..a5309f4 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/BUILD.gn
@@ -530,11 +530,19 @@ rtc_library("desktop_capture") {
       "window_finder_win.cc",
       "window_finder_win.h",
     ]
-    libs = [
-      "d3d11.lib",
-      "dxgi.lib",
-      "shcore.lib",
-    ]
+    if (is_mingw) {
+      libs = [
+        "d3d11",
+        "dxgi",
+        "shcore",
+      ]
+    } else {
+      libs = [
+        "d3d11.lib",
+        "dxgi.lib",
+        "shcore.lib",
+      ]
+    }
     deps += [
       "../../rtc_base:win32",
       "../../rtc_base/containers:flat_map",
@@ -688,7 +696,7 @@ if (use_desktop_capture_differ_sse2) {
       "differ_vector_sse2.h",
     ]
 
-    if (is_posix || is_fuchsia) {
+    if (is_posix || is_fuchsia || is_mingw) {
       cflags = [ "-msse2" ]
     }
   }
diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/desktop_capture_utils.cc b/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/desktop_capture_utils.cc
index 476ddc4..80a3ded 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/desktop_capture_utils.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/desktop_capture_utils.cc
@@ -10,7 +10,12 @@
 
 #include "modules/desktop_capture/win/desktop_capture_utils.h"
 
+#ifdef _MSC_VER
 #include "rtc_base/strings/string_builder.h"
+#else
+#include <cstdio>
+#include <cstdlib>
+#endif
 
 namespace webrtc {
 namespace desktop_capture {
@@ -27,10 +27,22 @@ namespace utils {
 // Generates a human-readable string from a COM error.
 std::string ComErrorToString(const _com_error& error) {
   char buffer[1024];
+#ifdef _MSC_VER
   webrtc::SimpleStringBuilder string_builder(buffer);
   // Use _bstr_t to simplify the wchar to char conversion for ErrorMessage().
   _bstr_t error_message(error.ErrorMessage());
   string_builder.AppendFormat("HRESULT: 0x%08X, Message: %s", error.Error(),
                               static_cast<const char*>(error_message));
   return string_builder.str();
+#else
+  int offset = snprintf(buffer, sizeof(buffer),
+                        "HRESULT: 0x%08lX, Message: ", error.Error());
+#ifdef _UNICODE
+  wcstombs(buffer + offset, error.ErrorMessage(), sizeof(buffer) - offset);
+#else
+  strcpy(buffer + offset, error.ErrorMessage(), sizeof(buffer) - offset);
+#endif
+  buffer[sizeof(buffer) - 1] = 0;
+  return buffer;
+#endif
 }
 
 }  // namespace utils
diff --git a/src/3rdparty/chromium/third_party/webrtc/rtc_base/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/rtc_base/BUILD.gn
index ccdecd0..848445b 100644
--- a/src/3rdparty/chromium/third_party/webrtc/rtc_base/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/rtc_base/BUILD.gn
@@ -575,7 +575,11 @@ rtc_library("timeutils") {
 
   libs = []
   if (is_win) {
-    libs += [ "winmm.lib" ]
+    if (is_mingw) {
+      libs += [ "winmm" ]
+    } else {
+      libs += [ "winmm.lib" ]
+    }
     deps += [ ":win32" ]
   }
 }
@@ -1094,11 +1098,19 @@ if (is_win) {
       ":stringutils",
     ]
 
-    libs = [
-      "crypt32.lib",
-      "iphlpapi.lib",
-      "secur32.lib",
-    ]
+    if (is_mingw) {
+      libs = [
+        "crypt32",
+        "iphlpapi",
+        "secur32",
+      ]
+    } else {
+      libs = [
+        "crypt32.lib",
+        "iphlpapi.lib",
+        "secur32.lib",
+      ]
+    }
 
     defines = [ "_CRT_NONSTDC_NO_DEPRECATE" ]
   }
diff --git a/src/3rdparty/chromium/third_party/webrtc/rtc_base/platform_thread_types.cc b/src/3rdparty/chromium/third_party/webrtc/rtc_base/platform_thread_types.cc
index d64ea68..f943485 100644
--- a/src/3rdparty/chromium/third_party/webrtc/rtc_base/platform_thread_types.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/rtc_base/platform_thread_types.cc
@@ -24,5 +24,6 @@
 #endif
 
 #if defined(WEBRTC_WIN)
+#include "build/build_config.h"
 #include <iterator>
 
 // The SetThreadDescription API was brought in version 1607 of Windows 10.
diff --git a/src/3rdparty/chromium/third_party/webrtc/rtc_base/system/rtc_export_template.h b/src/3rdparty/chromium/third_party/webrtc/rtc_base/system/rtc_export_template.h
index 4ac7043..050dc23 100644
--- a/src/3rdparty/chromium/third_party/webrtc/rtc_base/system/rtc_export_template.h
+++ b/src/3rdparty/chromium/third_party/webrtc/rtc_base/system/rtc_export_template.h
@@ -185,8 +185,10 @@
 
 RTC_EXPORT_TEMPLATE_TEST(DEFAULT, );  // NOLINT
 RTC_EXPORT_TEMPLATE_TEST(DEFAULT, __attribute__((visibility("default"))));
+#ifdef _MSC_VER
 RTC_EXPORT_TEMPLATE_TEST(MSVC_HACK, __declspec(dllexport));
 RTC_EXPORT_TEMPLATE_TEST(DEFAULT, __declspec(dllimport));
+#endif
 
 #undef RTC_EXPORT_TEMPLATE_TEST
 #undef RTC_EXPORT_TEMPLATE_TEST_DEFAULT_DEFAULT
diff --git a/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/create_direct3d_device.h b/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/create_direct3d_device.h
index 7c21f87..66bc9a3 100644
--- a/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/create_direct3d_device.h
+++ b/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/create_direct3d_device.h
@@ -11,8 +11,20 @@
 #ifndef RTC_BASE_WIN_CREATE_DIRECT3D_DEVICE_H_
 #define RTC_BASE_WIN_CREATE_DIRECT3D_DEVICE_H_
 
+#ifdef _MSC_VER
 #include <windows.graphics.directX.direct3d11.h>
 #include <windows.graphics.directX.direct3d11.interop.h>
+#else
+#include <windows.graphics.directx.direct3d11.h>
+#include <dxgi.h>
+#include <inspectable.h>
+
+extern "C" {
+// This function is only used in decltype(..)
+HRESULT __stdcall CreateDirect3D11DeviceFromDXGIDevice(
+    ::IDXGIDevice* dxgiDevice, ::IInspectable** graphicsDevice);
+}
+#endif
 #include <winerror.h>
 #include <wrl/client.h>
 
diff --git a/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/windows_version.cc b/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/windows_version.cc
index 93af137..6914722 100644
--- a/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/windows_version.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/rtc_base/win/windows_version.cc
@@ -17,7 +17,7 @@
 #include "rtc_base/checks.h"
 #include "rtc_base/string_utils.h"
 
-#if !defined(__clang__) && _MSC_FULL_VER < 191125507
+#if !defined(__clang__) && defined(_MSC_VER) && _MSC_FULL_VER < 191125507
 #error VS 2017 Update 3.2 or higher is required
 #endif
 
diff --git a/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn b/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn
index 786c469..65c2377 100644
--- a/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn
@@ -67,7 +67,11 @@ rtc_library("system_wrappers") {
   }
 
   if (is_win) {
-    libs += [ "winmm.lib" ]
+    if (is_mingw) {
+      libs += [ "winmm" ]
+    } else {
+      libs += [ "winmm.lib" ]
+    }
 
     # Windows needs ../rtc_base due to include of
     # webrtc/rtc_base/win32.h in source/clock.cc.
diff --git a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
index c5eeba5..7314e5c 100644
--- a/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
+++ b/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
@@ -232,7 +232,7 @@ declare_args() {
   # When set to true, a capturer implementation that uses the
   # Windows.Graphics.Capture APIs will be available for use. This introduces a
   # dependency on the Win 10 SDK v10.0.17763.0.
-  rtc_enable_win_wgc = is_win
+  rtc_enable_win_wgc = is_msvc
 
   # Includes the dav1d decoder in the internal decoder factory when set to true.
  rtc_include_dav1d_in_internal_decoder_factory = true
diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/window_capture_utils.h b/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/window_capture_utils.h
index 2bb5caf..892124e 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/window_capture_utils.h
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/desktop_capture/win/window_capture_utils.h
@@ -11,7 +11,7 @@
 #ifndef MODULES_DESKTOP_CAPTURE_WIN_WINDOW_CAPTURE_UTILS_H_
 #define MODULES_DESKTOP_CAPTURE_WIN_WINDOW_CAPTURE_UTILS_H_
 
-#include <shlobj_core.h>
+#include <shlobj.h>
 #include <windows.h>
 #include <wrl/client.h>
 
