diff --git a/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc b/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc
index 9e92226a42f..29af0238df5 100644
--- a/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc
+++ b/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc
@@ -507,6 +507,9 @@ bool DelegatedInkPointRendererGpu::DrawDelegatedInkPoint(
     return false;
   }
 
+#if defined(COMPILER_GCC)
+  return false;
+#else
   DCHECK(delegated_ink_trail_);
 
   DCompositionInkTrailPoint ink_point;
@@ -543,6 +547,7 @@ bool DelegatedInkPointRendererGpu::DrawDelegatedInkPoint(
   }
   delegated_ink_points_[point.pointer_id()][point] = token;
   return true;
+#endif
 }
 
 void DelegatedInkPointRendererGpu::InitializeForTesting(
