--- a/src/3rdparty/chromium/third_party/protobuf/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/protobuf/BUILD.gn
@@ -43,6 +43,30 @@
   include_dirs = [ "src" ]
 }

+if (qtwebengine_private_tool_dir != "") {
+  config("system_protobuf_lite_config") {
+    defines = [
+      "GOOGLE_PROTOBUF_NO_RTTI",
+      "NOMINMAX",
+      "PROTOBUF_USE_DLLS",
+    ]
+    libs = [ "protobuf-lite" ]
+  }
+
+  config("system_protobuf_full_config") {
+    defines = [
+      "GOOGLE_PROTOBUF_NO_RTTI",
+      "NOMINMAX",
+      "PROTOBUF_USE_DLLS",
+    ]
+    libs = [ "protobuf" ]
+  }
+
+  source_set("protobuf_lite") {
+    public = protobuf_headers
+    public_configs = [ ":system_protobuf_lite_config" ]
+  }
+} else {
 component("protobuf_lite") {
   sources = protobuf_lite_sources
   public = protobuf_headers
@@ -81,10 +105,18 @@
     defines = [ "LIBPROTOBUF_EXPORTS" ]
   }
 }
+}

 # This is the full, heavy protobuf lib that's needed for c++ .protos that don't
 # specify the LITE_RUNTIME option. The protocol compiler itself (protoc) falls
 # into that category. Do not use in Chrome code.
+if (qtwebengine_private_tool_dir != "") {
+  source_set("protobuf_full") {
+    public = protobuf_headers
+    public_configs = [ ":system_protobuf_full_config" ]
+    public_deps = [ ":protobuf_lite" ]
+  }
+} else {
 static_library("protobuf_full") {
   if (defined(build_with_chromium) && build_with_chromium) {
     # Prevent people from depending on this outside our file.
@@ -187,6 +219,7 @@
   }
   public_configs = [ ":protobuf_config" ]
 }
+}

 # Only compile the compiler for the host architecture.
 if (current_toolchain == host_toolchain) {
--- a/src/3rdparty/chromium/components/resources/ssl/ssl_error_assistant/BUILD.gn
+++ b/src/3rdparty/chromium/components/resources/ssl/ssl_error_assistant/BUILD.gn
@@ -25,8 +25,11 @@

   outputs = [ "$output_dir/$output_basename" ]

-  args = [
-    "-w",
+  args = []
+  if (qtwebengine_private_tool_dir == "") {
+    args += [ "-w" ]
+  }
+  args += [
     "-i",
     rebase_path(input_filename, root_build_dir),
     "-d",
