diff --git a/src/3rdparty/chromium/tools/metrics/histograms/generate_allowlist_from_histograms_file.py b/src/3rdparty/chromium/tools/metrics/histograms/generate_allowlist_from_histograms_file.py
index 1a16e4f1f3d..ed247e207a3 100755
--- a/src/3rdparty/chromium/tools/metrics/histograms/generate_allowlist_from_histograms_file.py
+++ b/src/3rdparty/chromium/tools/metrics/histograms/generate_allowlist_from_histograms_file.py
@@ -6,6 +6,7 @@
 import argparse
 import os
 import sys
+sys.path.insert(0, os.path.dirname(__file__))
 
 import extract_histograms
 import xml.dom.minidom
