--- cocos2d-x-cocos2d-x-3.14.1/cmake/Modules/BuildModules.cmake.orig	2017-04-05 09:52:34.908921600 +0300
+++ cocos2d-x-cocos2d-x-3.14.1/cmake/Modules/BuildModules.cmake	2017-04-05 09:52:53.560988400 +0300
@@ -71,7 +71,7 @@
 	    set(BULLET_LIBRARIES bullet)
 	  else()
 	    cocos_find_package(bullet BULLET REQUIRED)
-	    set(BULLET_LIBRARIES bullet)
+	    set(BULLET_LIBRARIES BulletSoftBody BulletDynamics BulletCollision LinearMath)
 	  endif()
 	  message(STATUS "Bullet include dirs: ${BULLET_INCLUDE_DIRS}")
 	endif(USE_BULLET)
