From 100f4ee4b22f93061bacd31d7479a7c8d7107db1 Mon Sep 17 00:00:00 2001
From: Xiretza <xiretza@xiretza.xyz>
Date: Fri, 6 Aug 2021 13:54:10 +0200
Subject: [PATCH 2/3] Build dynamic library

---
 c_lib/c_libfswatch.gpr | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/c_lib/c_libfswatch.gpr b/c_lib/c_libfswatch.gpr
index a032dca..9210f6b 100644
--- a/c_lib/c_libfswatch.gpr
+++ b/c_lib/c_libfswatch.gpr
@@ -10,7 +10,7 @@ library project C_Libfswatch is
    for Externally_Built use "true";
    for Library_Name use "fswatch";
-   for Library_Dir use libfswatch_prefix & "/lib";
-   for Library_Kind use "static";
+   for Library_Dir use libfswatch_prefix & "/bin";
+   for Library_Kind use "dynamic";
 
    package Linker is
       case OS is
-- 
2.32.0
