diff --git a/win32/GNUmakefile b/win32/GNUmakefile
index 86453a2..3fb67dd 100644
--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -1719,7 +1719,7 @@ endif
 
 doc: $(PERLEXE) $(PERLDLL) ..\pod\perltoc.pod
 	$(PERLEXE) -I..\lib ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
-	    --podpath=pod:lib:utils --htmlroot="file://$(subst :,|,$(INST_HTML))"\
+	    --podpath=pod:lib:utils --htmlroot="file://$(INST_HTML)" \
 	    --recurse
 
 ..\utils\Makefile: $(CONFIGPM) ..\utils\Makefile.PL
diff --git a/win32/makefile.mk b/win32/makefile.mk
index ff5befa..6a7fff6 100644
--- a/win32/makefile.mk
+++ b/win32/makefile.mk
@@ -1637,8 +1637,8 @@ rebasePE : Extensions $(PERLDLL) $(NORMALIZE_DYN) $(PERLEXE)
 
 
 doc: $(PERLEXE) $(PERLDLL) ..\pod\perltoc.pod
-	$(PERLEXE) ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
-	    --podpath=pod:lib:utils --htmlroot="file://$(INST_HTML:s,:,|,)"\
+	$(PERLEXE) -I..\lib ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
+	    --podpath=pod:lib:utils --htmlroot="file://$(INST_HTML)" \
 	    --recurse
 
 ..\utils\Makefile: $(CONFIGPM) ..\utils\Makefile.PL
-- 
2.31.1.windows.1

