diff -aur 001/win32/makefile.mk 002/win32/makefile.mk
--- 001/win32/makefile.mk	2015-01-05 20:37:05.274414000 -0200
+++ 002/win32/makefile.mk	2015-01-05 20:37:54.890625000 -0200
@@ -1349,7 +1349,7 @@
 
 doc: $(PERLEXE) ..\pod\perltoc.pod
-	$(PERLEXE) ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
-	    --podpath=pod:lib:utils --htmlroot="file://$(INST_HTML:s,:,|,)"\
+	$(PERLEXE) -I..\lib ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
+	    --podpath=pod:lib:utils --htmlroot="@PERL_RELOCATE_HTML@/share/doc/perl5" \
 	    --recurse
 
 ..\utils\Makefile: $(CONFIGPM) ..\utils\Makefile.PL
