diff --git 001/win32/GNUmakefile 002/win32/GNUmakefile
index 17b07d1..ff851c5 100644
--- 001/win32/GNUmakefile
+++ 002/win32/GNUmakefile
@@ -1641,7 +1641,7 @@ endif
 
 doc: $(PERLEXE) $(PERLDLL) ..\pod\perltoc.pod
 	$(PERLEXE) -I..\lib ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
-	    --podpath=pod:lib:utils --htmlroot="file://$(subst :,|,$(INST_HTML))"\
+	    --podpath=pod:lib:utils --htmlroot="@PERL_RELOCATE_HTML@/share/doc/perl5" \
 	    --recurse
 
 ..\utils\Makefile: $(CONFIGPM) ..\utils\Makefile.PL
diff --git 001/win32/makefile.mk 002/win32/makefile.mk
index f098f76..a8a2536 100644
--- 001/win32/makefile.mk
+++ 002/win32/makefile.mk
@@ -1555,8 +1555,8 @@ rebasePE : Extensions $(PERLDLL) Extensions_normalize $(PERLEXE)
 
 
 doc: $(PERLEXE) $(PERLDLL) ..\pod\perltoc.pod
-	$(PERLEXE) ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
-	    --podpath=pod:lib:utils --htmlroot="file://$(INST_HTML:s,:,|,)"\
+	$(PERLEXE) -I..\lib ..\installhtml --podroot=.. --htmldir=$(HTMLDIR) \
+	    --podpath=pod:lib:utils --htmlroot="@PERL_RELOCATE_HTML@/share/doc/perl5" \
 	    --recurse
 
 ..\utils\Makefile: $(CONFIGPM) ..\utils\Makefile.PL
