diff --git 003/cpan/CPAN/lib/CPAN/FirstTime.pm 004/cpan/CPAN/lib/CPAN/FirstTime.pm
index 531c115..6ee1642 100644
--- 003/cpan/CPAN/lib/CPAN/FirstTime.pm
+++ 004/cpan/CPAN/lib/CPAN/FirstTime.pm
@@ -1492,6 +1492,21 @@ sub _init_external_progs {
                 next;
             }
 
+            # Let's not store full paths to programs in MSYS2 because
+            # paths change across shells. For example let's not use 32-bit
+            # programs in the 64-bit shell.
+
+            if ($^O eq 'MSWin32') {
+                if ($progname eq 'make') {
+                    # Make doesn't like the cmd.exe-friendly makefiles that are
+                    # created when installing modules, so let's stick to dmake.
+                    $CPAN::Config->{$progname} = 'dmake';
+                } else {
+                    $CPAN::Config->{$progname} = $progname;
+                }
+                next;
+            }
+
             my $progcall = $progname;
             unless ($matcher) {
                 # we really don't need ncftp if we have ncftpget, but
