diff -aur 003/cpan/CPAN/lib/CPAN/FirstTime.pm 004/cpan/CPAN/lib/CPAN/FirstTime.pm
--- 003/cpan/CPAN/lib/CPAN/FirstTime.pm	2015-01-06 21:07:13.509562500 -0200
+++ 004/cpan/CPAN/lib/CPAN/FirstTime.pm	2015-01-07 09:17:32.727227100 -0200
@@ -1470,6 +1470,21 @@
                 next;
             }
 
+            # Let's not store full paths to programs in MSYS2 because
+            # paths change across shells. For example let's not use 32-bit
+            # programs in the 64-bit shell.
+
+            if ($^O eq 'MSWin32') {
+                if ($progname eq 'make') {
+                    # Make doesn't like the cmd.exe-friendly makefiles that are
+                    # created when installing modules, so let's stick to dmake.
+                    $CPAN::Config->{$progname} = 'dmake';
+                } else {
+                    $CPAN::Config->{$progname} = $progname;
+                }
+                next;
+            }
+
             my $progcall = $progname;
             unless ($matcher) {
                 # we really don't need ncftp if we have ncftpget, but
