Author: Alexey Pavlov <Alexpux@gmail.com>
diff -aur 004/win32/config.gc 005/win32/config.gc
--- 004/win32/config.gc
+++ 005/win32/config.gc
@@ -310,7 +310,7 @@
 d_mkdir='define'
 d_mkdtemp='undef'
 d_mkfifo='undef'
-d_mkstemp='undef'
+d_mkstemp='define'
 d_mkstemps='undef'
 d_mktime64='undef'
 d_mktime='define'
diff -aur 004/win32/config_H.gc 005/win32/config_H.gc
--- 004/win32/config_H.gc
+++ 005/win32/config_H.gc
@@ -1973,7 +1973,7 @@
  *	available to exclusively create and open a uniquely named
  *	temporary file.
  */
-/*#define HAS_MKSTEMP		/ **/
+#define HAS_MKSTEMP		/**/
 
 /* HAS_MMAP:
  *	This symbol, if defined, indicates that the mmap system call is
diff -aur 004/win32/win32.c 005/win32/win32.c
--- 004/win32/win32.c
+++ 005/win32/win32.c
@@ -1142,6 +1142,7 @@
  * XXX this needs strengthening  (for PerlIO)
  *   -- BKS, 11-11-200
 */
+#if !defined(HAS_MKSTEMP)
 int mkstemp(const char *path)
 {
     dTHX;
@@ -1162,6 +1163,7 @@
 	goto retry;
     return fd;
 }
+#endif
 
 static long
 find_pid(pTHX_ int pid)
diff -aur 004/win32/win32.h 005/win32/win32.h
--- 004/win32/win32.h
+++ 005/win32/win32.h
@@ -331,8 +331,10 @@
 #endif
 extern	char *	getlogin(void);
 extern	int	chown(const char *p, uid_t o, gid_t g);
+#if !defined(HAS_MKSTEMP)
 extern  int	mkstemp(const char *path);
 #endif
+#endif
 
 #undef	 Stat
 #define  Stat		win32_stat
