diff --git 005/installman 006/installman
index 53117ab..5b85abd 100644
--- 005/installman
+++ 006/installman
@@ -1,7 +1,9 @@
 #!./perl -w
 
 BEGIN {
-    @INC = qw(lib);
+    chdir '..' if !-d 'lib' and -d '../lib';
+    @INC = 'lib';
+    $ENV{PERL5LIB} = 'lib';
 
     # This needs to be at BEGIN time, before any use of Config
     # install_lib itself loads and imports Config into main::
@@ -146,7 +148,7 @@ sub pod2man {
 	    next;
 	}
 
-	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'uwin' || $^O eq 'cygwin') {
+	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'uwin' || $^O eq 'cygwin' || $^O eq 'MSWin32' ) {
             $manpage =~ s#::#.#g;
 	}
         my $tmp = "${mandir}/${manpage}.tmp";
diff --git 005/win32/GNUmakefile 006/win32/GNUmakefile
index 5781bc3..f31dfbe 100644
--- 005/win32/GNUmakefile
+++ 006/win32/GNUmakefile
@@ -1800,7 +1800,14 @@ distclean: realclean
 	-if exist $(HTMLDIR) rmdir /s /q $(HTMLDIR)
 	-del /f ..\t\test_state
 
-install : all installbare installhtml
+install : all installbare installhtml installman
+
+install.bare : all installbare
+
+install.doc : all installhtml installman
+
+installman : ..\pod\perltoc.pod
+	$(PERLEXE) ..\installman
 
 installbare : utils ..\pod\perltoc.pod
 	$(PERLEXE) ..\installperl
diff --git 005/win32/makefile.mk 006/win32/makefile.mk
index d1911e6..2305272 100644
--- 005/win32/makefile.mk
+++ 006/win32/makefile.mk
@@ -1716,7 +1716,14 @@ distclean: realclean
 	-if exist $(HTMLDIR) rmdir /s /q $(HTMLDIR)
 	-del /f ..\t\test_state
 
-install : all installbare installhtml
+install : all installbare installhtml installman
+
+install.bare : all installbare
+
+install.doc : all installhtml installman
+
+installman : ..\pod\perltoc.pod
+	$(PERLEXE) ..\installman
 
 installbare : utils ..\pod\perltoc.pod
 	$(PERLEXE) ..\installperl
