--- a/installman
+++ b/installman
@@ -1,7 +1,9 @@
 #!./perl -w
 
 BEGIN {
-    @INC = qw(lib);
+    chdir '..' if !-d 'lib' and -d '../lib';
+    @INC = 'lib';
+    $ENV{PERL5LIB} = 'lib';
 
     # This needs to be at BEGIN time, before any use of Config
     # install_lib itself loads and imports Config into main::
@@ -154,7 +156,7 @@
 	    next;
 	}
 
-	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'cygwin') {
+	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'cygwin' || $^O eq 'MSWin32') {
             $manpage =~ s#::#.#g;
 	}
         my $tmp = "${mandir}/${manpage}.tmp";
diff --git a/win32/GNUmakefile b/win32/GNUmakefile
index 15a1658..4f241a1 100644
--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -1772,7 +1772,14 @@
 	-if exist $(HTMLDIR) rmdir /s /q $(HTMLDIR)
 	-del /f ..\t\test_state
 
-install : all installbare installhtml
+install : all installbare installhtml installman
+
+install.bare : all installbare
+
+install.doc : all installhtml installman
+
+installman : ..\pod\perltoc.pod
+	$(PERLEXE) ..\installman
 
 installbare : utils ..\pod\perltoc.pod
 	$(PERLEXE) ..\installperl
--- a/win32/Makefile
+++ b/win32/Makefile
@@ -1296,7 +1296,14 @@
 	-if exist $(HTMLDIR) rmdir /s /q $(HTMLDIR)
 	-del /f ..\t\test_state
 
-install : all installbare installhtml
+install : all installbare installhtml installman
+
+install.bare : all installbare
+
+install.doc : all installhtml installman
+
+installman : ..\pod\perltoc.pod
+	$(PERLEXE) ..\installman
 
 installbare : utils ..\pod\perltoc.pod
 	$(PERLEXE) ..\installperl
-- 
2.31.1.windows.1

