--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm
@@ -291,7 +291,7 @@
     my @extralibs;
     my $cc = $Config{cc} || '';
     my $VC = $cc =~ /\bcl\b/i;
-    my $GC = $cc =~ /\bgcc\b/i;
+    my $GC = $cc =~ /\bcc\b/i;
 
     my $libext     = _win32_lib_extensions();
     my @searchpath = ( '' );                                    # from "-L/path" entries in $potential_libs
--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_NW5.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_NW5.pm
@@ -34,7 +34,7 @@
 $ENV{EMXSHELL} = 'sh'; # to run `commands`
 
 my $BORLAND  = $Config{'cc'} =~ /\bbcc/i;
-my $GCC      = $Config{'cc'} =~ /\bgcc/i;
+my $GCC      = $Config{'cc'} =~ /\bcc/i;
 
 
 =item os_flavor
--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Win32.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Win32.pm
@@ -38,7 +38,7 @@
 	my ( $config ) = @_;
 
 	my $BORLAND = $config->{cc} =~ /\bbcc/i ? 1 : 0;
-	my $GCC     = $config->{cc} =~ /\bgcc\b/i ? 1 : 0;
+	my $GCC     = $config->{cc} =~ /\bcc\b/i ? 1 : 0;
 	my $MSVC    = $config->{cc} =~ /\b(?:cl|icl)/i ? 1 : 0; # MSVC can come as clarm.exe, icl=Intel C
 
 	return ( $BORLAND, $GCC, $MSVC );
--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Mksymlists.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Mksymlists.pm
@@ -137,7 +137,7 @@
     open( my $def, ">", "$data->{FILE}.def" )
         or croak("Can't create $data->{FILE}.def: $!\n");
     # put library name in quotes (it could be a keyword, like 'Alias')
-    if ($Config::Config{'cc'} !~ /\bgcc/i) {
+    if ($Config::Config{'cc'} !~ /\bcc/i) {
         print $def "LIBRARY \"$data->{DLBASE}\"\n";
     }
     print $def "EXPORTS\n  ";
--- a/cpan/ExtUtils-MakeMaker/t/MM_NW5.t
+++ b/cpan/ExtUtils-MakeMaker/t/MM_NW5.t
@@ -160,7 +160,7 @@
 
 # clean()
 {
-    my $clean = $Config{cc} =~ /^gcc/i ? 'dll.base dll.exp' : '*.pdb';
+    my $clean = $Config{cc} =~ /^cc/i ? 'dll.base dll.exp' : '*.pdb';
     like( $MM->clean(), qr/^clean ::\s+\Q-$(RM_F) $clean\E\s+$/m,
           'clean() Makefile target' );
 }
--- a/cpan/ExtUtils-MakeMaker/t/MM_Win32.t
+++ b/cpan/ExtUtils-MakeMaker/t/MM_Win32.t
@@ -306,9 +306,9 @@
             desc => 'empty cc is not recognized as gcc',
         },
         {
-            config => { cc => 'gcc' },
+            config => { cc => 'cc' },
             key => 'GCC', expect => 1,
-            desc => 'plain "gcc" is recognized',
+            desc => 'plain "cc" is recognized',
         },
         {
             config => { cc => 'C:/MinGW/bin/gcc.exe' },
--- a/t/win32/system.t
+++ b/t/win32/system.t
@@ -99,7 +99,7 @@
 }
 else {
     my $minus_o = '';
-    if ($Config{cc} =~ /\bgcc/i)
+    if ($Config{cc} =~ /\bcc/i)
      {
       $minus_o = "-o $exename.exe";
      }
--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -27,12 +27,12 @@
 # possible values: gcc
 #                  i686-w64-mingw32-gcc
 #                  x86_64-w64-mingw32-gcc
-GCCBIN := gcc
+GCCBIN := cc
 
-ifeq ($(GCCBIN),x86_64-w64-mingw32-gcc)
+ifeq ($(GCCBIN),x86_64-w64-mingw32-cc)
 GCCCROSS := x86_64-w64-mingw32
 endif
-ifeq ($(GCCBIN),i686-w64-mingw32-gcc)
+ifeq ($(GCCBIN),i686-w64-mingw32-cc)
 GCCCROSS := i686-w64-mingw32
 endif
 
@@ -565,8 +565,8 @@
 
 ifeq ($(CCTYPE),GCC)
 
-CC		= $(ARCHPREFIX)gcc
-LINK32		= $(ARCHPREFIX)g++
+CC		= $(ARCHPREFIX)cc
+LINK32		= $(ARCHPREFIX)c++
 LIB32		= $(ARCHPREFIX)ar rc
 IMPLIB		= $(ARCHPREFIX)dlltool
 RSC		= $(ARCHPREFIX)windres
--- a/win32/config.gc
+++ b/win32/config.gc
@@ -40,7 +40,7 @@
 c=''
 castflags='0'
 cat='type'
-cc='~ARCHPREFIX~gcc'
+cc='~ARCHPREFIX~cc'
 cccdlflags=' '
 ccdlflags=' '
 ccflags='-MD -DWIN32'
@@ -868,7 +868,7 @@
 ivtype='long'
 known_extensions='~static_ext~ ~dynamic_ext~ ~nonxs_ext~'
 ksh=''
-ld='~ARCHPREFIX~g++'
+ld='~ARCHPREFIX~c++'
 lddlflags='-shared ~LINK_FLAGS~'
 ldflags='~LINK_FLAGS~'
 ldflags_uselargefiles=''
--- a/win32/config_H.gc
+++ b/win32/config_H.gc
@@ -1417,9 +1417,9 @@
  *	This symbol is intended to be used along with CPPRUN in the same manner
  *	symbol CPPMINUS is used with CPPSTDIN. It contains either "-" or "".
  */
-#define CPPSTDIN "gcc -E"
+#define CPPSTDIN "cc -E"
 #define CPPMINUS "-"
-#define CPPRUN "gcc -E"
+#define CPPRUN "cc -E"
 #define CPPLAST ""
 
 /* HAS_ACCESS:
--- a/win32/config_h.PL
+++ b/win32/config_h.PL
@@ -68,9 +68,9 @@
    {
      $_ = "/*#define ". $1 . "_EXP \"\"\t/ **/\n";
    }
-  elsif (/^\s*#define\s+CPP(STDIN|RUN)\s+"gcc(.*)"\s*$/)
+  elsif (/^\s*#define\s+CPP(STDIN|RUN)\s+"cc(.*)"\s*$/)
    {
-     $_ = "#define CPP" . $1 . " \"" . $opt{ARCHPREFIX} . "gcc" . $2 . "\"\n";
+     $_ = "#define CPP" . $1 . " \"" . $opt{ARCHPREFIX} . "cc" . $2 . "\"\n";
    }
   print H;
  }
--- a/win32/config_sh.PL
+++ b/win32/config_sh.PL
@@ -153,7 +153,7 @@
         my $output = `$opt{cc} 2>&1`;
         $opt{ccversion} = $output =~ /\b(\d+(?:\.\d+)+)\b/ ? $1 : '?';
     }
-    elsif ($opt{cc} =~ /\bgcc\b/) {
+    elsif ($opt{cc} =~ /\bcc\b/) {
         chomp($opt{gccversion} = `$opt{cc} -dumpversion`);
         chomp($opt{libc} = `$opt{cc} -o rt.exe configure/rt.c 1>nul 2>&1 && rt`);
         if(-e 'rt.exe') { unlink 'rt.exe' } # rt.exe no longer needed
@@ -182,7 +182,7 @@
 if ($opt{cc} =~ /\b(?:cl|icl)/) {
     $int64  = '__int64';
 }
-elsif ($opt{cc} =~ /\bgcc\b/) {
+elsif ($opt{cc} =~ /\bcc\b/) {
     $int64  = 'long long';
 }
 
