--- build-x86_64-perl-5.32.1/win32/perlhost.h.orig	2021-07-17 21:13:21.426911700 -0700
+++ build-x86_64-perl-5.32.1/win32/perlhost.h	2021-07-17 21:13:56.254963800 -0700
@@ -1524,13 +1524,19 @@
 }
 
 PERL_CALLCONV_NO_RET void
-PerlProcExit(struct IPerlProc* piPerl, int status)
+PerlProcExit(struct IPerlProc* piPerl, int status)
+#ifdef __clang__
+__attribute__noreturn__
+#endif
 {
     exit(status);
 }
 
 PERL_CALLCONV_NO_RET void
-PerlProc_Exit(struct IPerlProc* piPerl, int status)
+PerlProc_Exit(struct IPerlProc* piPerl, int status)
+#ifdef __clang__
+__attribute__noreturn__
+#endif
 {
     _exit(status);
 }
