--- build-x86_64-perl-5.32.1/win32/GNUmakefile.orig	2021-07-19 11:39:52.573976600 -0700
+++ build-x86_64-perl-5.32.1/win32/GNUmakefile	2021-07-19 11:47:05.542748000 -0700
@@ -416,7 +416,7 @@
 # Check processor architecture from target triplet
 # possible values:
 # gcc: i686-w64-mingw32, x86_64-w64-mingw32
-# clang: i686-w64-windows-gnu, x86_64-w64-windows-gnu
+# clang: i686-w64-windows-gnu, x86_64-w64-windows-gnu, aarch64-w64-windows-gnu
 ifeq ($(CCTYPE),GCC)
 ifeq (x86_64, $(findstring x86_64, $(GCCTARGET)))
 WIN64 := define
@@ -426,6 +426,10 @@
 WIN64 := undef
 PROCESSOR_ARCHITECTURE := x86
 endif
+ifeq (aarch64, $(findstring aarch64, $(GCCTARGET)))
+WIN64 := define
+PROCESSOR_ARCHITECTURE := arm64
+endif
 endif
 
 PROCESSOR_ARCHITECTURE  ?= x86
