--- a/perl.h
+++ b/perl.h
@@ -2759,7 +2759,7 @@
 #   ifndef Perl_isfinite
 #     ifdef HAS_ISFINITE
 #       define Perl_isfinite(x) isfinite(x)
-#     elif defined(HAS_FINITE)
+#     elif defined(HAS_FINITE) && !defined(__MINGW32__)
 #       define Perl_isfinite(x) finite(x)
 #     endif
 #   endif
@@ -2935,7 +2935,7 @@
 #endif
 
 /* Win32: _fpclass(), _isnan(), _finite(). */
-#ifdef _MSC_VER
+#ifdef _WIN32
 #  ifndef Perl_isnan
 #    define Perl_isnan(x) _isnan(x)
 #  endif
