--- perl-5.40.4/perl.h.orig	2026-07-06 08:24:47.695135900 +0200
+++ perl-5.40.4/perl.h	2026-07-06 08:24:23.505542300 +0200
@@ -2790,7 +2790,7 @@
 #   ifndef Perl_isfinite
 #     ifdef HAS_ISFINITE
 #       define Perl_isfinite(x) isfinite(x)
-#     elif defined(HAS_FINITE)
+#     elif defined(HAS_FINITE) && !defined(__MINGW32__)
 #       define Perl_isfinite(x) finite(x)
 #     endif
 #   endif
@@ -2966,7 +2966,7 @@
 #endif
 
 /* Win32: _fpclass(), _isnan(), _finite(). */
-#ifdef _MSC_VER
+#ifdef _WIN32
 #  ifndef Perl_isnan
 #    define Perl_isnan(x) _isnan(x)
 #  endif
--- perl-5.40.4/ext/POSIX/POSIX.xs.orig	2026-07-06 08:26:34.688097000 +0200
+++ perl-5.40.4/ext/POSIX/POSIX.xs	2026-07-06 09:01:31.470301800 +0200
@@ -572,7 +572,7 @@
 #  undef c99_trunc
 #endif
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 
 /* Some APIs exist under Win32 with "underbar" names. */
 #  undef c99_hypot
