--- perl-5.16.2/win32/makefile.mk.orig	2012-11-27 21:31:21 +0400
+++ perl-5.16.2/win32/makefile.mk	2012-11-27 21:34:26 +0400
@@ -21,13 +21,13 @@
 # Set these to wherever you want "dmake install" to put your
 # newly built perl.
 #
-INST_DRV	*= c:
-INST_TOP	*= $(INST_DRV)\perl
+INST_DRV	*= %DRV%
+INST_TOP	*= $(INST_DRV)%NODRV%
 
 #
 # Uncomment if you want to build a 32-bit Perl using a 32-bit compiler
 # on a 64-bit version of Windows.
-#WIN64		*= undef
+%COMW64%WIN64		*= undef
 
 #
 # Comment this out if you DON'T want your perl installation to be versioned.
@@ -177,7 +177,7 @@
 # not be quoted)
 #
 .IF "$(CCTYPE)" == "GCC"
-CCHOME		*= C:\MinGW
+CCHOME		*= %MINGWHOME%
 .ELSE
 CCHOME		*= $(MSVCDIR)
 .ENDIF
@@ -228,7 +228,7 @@
 # specify semicolon-separated list of extra directories that modules will
 # look for libraries (spaces in path names need not be quoted)
 #
-EXTRALIBDIRS	*=
+EXTRALIBDIRS	*= %THIRDPARTY_LIBS%
 
 #
 # set this to point to cmd.exe (only needed if you use some
