From 9345eddc1b31aef3530359809cb6b5ac6d4a4a79 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 06:18:02 +0300
Subject: [PATCH 02/27] Always use Win32 threads for mingw.

---
 src/jrd/common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jrd/common.h b/src/jrd/common.h
index ae54880929..49e8b607d8 100644
--- a/src/jrd/common.h
+++ b/src/jrd/common.h
@@ -65,7 +65,7 @@
 #include <unistd.h>
 #endif
 
-#if defined(_POSIX_THREADS) && _POSIX_THREADS >= 200112L
+#if defined(_POSIX_THREADS) && _POSIX_THREADS >= 200112L && !defined(WIN_NT)
 // above check is generally true. However, we use pthreads on some platforms
 // where _POSIX_THREADS is defined to "1" or not even defined at all!
 #define USE_POSIX_THREADS
-- 
2.13.0

