From 94a88c81c24c948f163109a86aea62ad04ee1ba6 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 09:57:27 +0300
Subject: [PATCH 05/27] Fix cast error.

---
 src/remote/inet.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/remote/inet.cpp b/src/remote/inet.cpp
index 2f1f63dea0..9a3e9a120a 100644
--- a/src/remote/inet.cpp
+++ b/src/remote/inet.cpp
@@ -1888,7 +1888,7 @@ static void wsaExitHandler(void*)
 }
 
 
-static int fork(SOCKET old_handle, USHORT flag)
+static int fork(SOCKET old_handle, DWORD flag)
 {
 /**************************************
  *
@@ -1945,7 +1945,7 @@ static int fork(SOCKET old_handle, USHORT flag)
 
 THREAD_ENTRY_DECLARE forkThread(THREAD_ENTRY_PARAM arg)
 {
-	const USHORT flag = (USHORT) arg;
+	const DWORD flag = (uintptr_t) arg;
 
 	while (!INET_shutting_down)
 	{
-- 
2.13.0

