diff -urN freetype-d911cb5392bd558642fcd6e85d6224618b8e1421.a/meson.build freetype-d911cb5392bd558642fcd6e85d6224618b8e1421.b/meson.build
--- freetype-d911cb5392bd558642fcd6e85d6224618b8e1421.a/meson.build	2021-05-02 05:49:11.000000000 +0200
+++ freetype-d911cb5392bd558642fcd6e85d6224618b8e1421.b/meson.build	2021-05-21 06:29:43.356074600 +0200
@@ -124,6 +124,7 @@
   'include/freetype/ftbzip2.h',
   'include/freetype/ftcache.h',
   'include/freetype/ftchapters.h',
+  'include/freetype/ftcid.h',
   'include/freetype/ftcolor.h',
   'include/freetype/ftdriver.h',
   'include/freetype/fterrdef.h',
