From 47586605368ef89502c4cdee2d8a58e8d87becb7 Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Sun, 3 Aug 2014 14:47:26 +0200
Subject: [PATCH 2/2] msys2: add CC flags to gir-scanner

---
 clutter/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clutter/Makefile.am b/clutter/Makefile.am
index a48010b..e23021c 100644
--- a/clutter/Makefile.am
+++ b/clutter/Makefile.am
@@ -977,6 +977,7 @@ Clutter_@CLUTTER_API_VERSION_AM@_gir_SCANNERFLAGS = \
 	--pkg-export=clutter-@CLUTTER_API_VERSION@
 
 INTROSPECTION_GIRS += Clutter-@CLUTTER_API_VERSION@.gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 Cally-@CLUTTER_API_VERSION@.gir: Makefile Clutter-@CLUTTER_API_VERSION@.gir
 
-- 
2.0.4

