diff -rupN openjpeg-1.5.1/configure.ac openjpeg-1.5.1-new/configure.ac
--- openjpeg-1.5.1/configure.ac	2012-09-13 09:58:39.000000000 +0200
+++ openjpeg-1.5.1-new/configure.ac	2013-05-10 15:56:02.458721728 +0200
@@ -267,17 +267,7 @@ AC_MSG_CHECKING([whether to build with T
 AC_MSG_RESULT([${want_tiff}])
 
 if test "x${want_tiff}" = "xyes" ; then
-
-   OPJ_CHECK_LIB(
-      [tiff.h],
-      [tiff],
-      [TIFFOpen],
-      [have_libtiff="yes"],
-      [have_libtiff="no"])
-
-   if test "x${have_libtiff}" = "xno" ; then
-      AC_MSG_WARN([Can not find a usuable TIFF library. Make sure that CPPFLAGS and LDFLAGS are correctly set.])
-   fi
+   PKG_CHECK_MODULES([TIFF], libtiff-4)
 fi
 
 AC_MSG_CHECKING([whether TIFF is available])
