--- schroedinger-1.0.11/testsuite/Makefile.am.orig	2012-04-15 08:18:57 +0400
+++ schroedinger-1.0.11/testsuite/Makefile.am	2012-04-15 08:18:46 +0400
@@ -97,7 +97,7 @@
 #wavelet_2d_CFLAGS = $(SCHRO_CFLAGS)
 #wavelet_2d_LDADD = $(SCHRO_LIBS)
 wavelet_2d_SOURCES = wavelet_2d.c
-nodist_wavelet_2d_SOURCES = $(top_srcdir)/schroedinger/tmp-orc.c
+nodist_wavelet_2d_SOURCES = $(top_builddir)/schroedinger/tmp-orc.c
 
 wavelet_max_gain_SOURCES = wavelet_max_gain.c
 
@@ -118,7 +118,7 @@
 #quant_CFLAGS = 
 #quant_LDADD = -lm
 quant_SOURCES = quant.c
-nodist_quant_SOURCES = $(top_srcdir)/schroedinger/tmp-orc.c
+nodist_quant_SOURCES = $(top_builddir)/schroedinger/tmp-orc.c
 
 #sort_CFLAGS = $(SCHRO_CFLAGS)
 #sort_LDADD = $(SCHRO_LIBS)
