--- a/CMakeLists.txt	2021-10-04 13:49:12.582319800 -0700
+++ b/CMakeLists.txt	2021-10-04 13:54:57.833484200 -0700
@@ -110,7 +110,9 @@
   # LLVM_INCLUDE_DIRS when merging in the monorepo (Warning from flang headers
   # should not be suppressed).
   include_directories(SYSTEM ${LLVM_INCLUDE_DIRS})
-  add_definitions(${LLVM_DEFINITIONS})
+  # https://gitlab.kitware.com/cmake/cmake/-/issues/22162
+  separate_arguments(LLVM_DEFINITIONS_LIST NATIVE_COMMAND ${LLVM_DEFINITIONS})
+  add_definitions(${LLVM_DEFINITIONS_LIST})
 
   # LLVM's cmake configuration files currently sneak in a c++11 flag.
   # We look for it here and remove it from Flang's compile flags to
