--- a/cmake/modules/AddFlang.cmake
+++ b/cmake/modules/AddFlang.cmake
@@ -61,7 +61,7 @@
     set_property(GLOBAL APPEND PROPERTY FLANG_STATIC_LIBS ${name})
   endif()
 
-  llvm_add_library(${name} ${LIBTYPE} ${ARG_UNPARSED_ARGUMENTS} ${srcs})
+  llvm_add_library(${name} ${LIBTYPE} ${ARG_UNPARSED_ARGUMENTS} DISABLE_LLVM_LINK_LLVM_DYLIB ${srcs})
 
   if (TARGET ${name})
 
@@ -94,7 +94,7 @@
 endmacro(add_flang_library)
 
 macro(add_flang_executable name)
-  add_llvm_executable(${name} ${ARGN})
+  add_llvm_executable(${name} ${ARGN} DISABLE_LLVM_LINK_LLVM_DYLIB)
   set_target_properties(${name} PROPERTIES FOLDER "Flang executables")
   set_flang_windows_version_resource_properties(${name})
 endmacro(add_flang_executable)
