--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -76,6 +76,11 @@
 
 append(${NO_LTO_FLAGS} CMAKE_C_FLAGS)
 append(${NO_LTO_FLAGS} CMAKE_CXX_FLAGS)
+
+if (MINGW)
+  append("-municode" CMAKE_C_FLAGS)
+  append("-municode" CMAKE_CXX_FLAGS)
+endif()
 
 # Disable libstdc++/libc++ assertions, even in an LLVM_ENABLE_ASSERTIONS build,
 # to avoid an unwanted dependency on libstdc++/libc++.so.
