--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -478,7 +478,7 @@
           target_link_libraries (${target_name} ${CMAKE_THREAD_LIBS_INIT})
         endif ()
         if (HAVE_SHLWAPI_H)
-          target_link_libraries (${target_name} shlwapi.lib)
+          target_link_libraries (${target_name} shlwapi)
         endif ()
         list (APPEND TARGETS ${target_name})
         # add convenience make target for build of both shared and static libraries
