diff -r -c gflags-2.2.0.orig/CMakeLists.txt gflags-2.2.0/CMakeLists.txt
*** gflags-2.2.0.orig/CMakeLists.txt	2016-11-25 16:54:37.000000000 -0200
--- gflags-2.2.0/CMakeLists.txt	2017-02-26 12:05:09.782590500 -0300
***************
*** 396,402 ****
  
  # ----------------------------------------------------------------------------
  # installation directories
! if (OS_WINDOWS)
    set (RUNTIME_INSTALL_DIR Bin)
    set (LIBRARY_INSTALL_DIR Lib)
    set (INCLUDE_INSTALL_DIR Include)
--- 396,402 ----
  
  # ----------------------------------------------------------------------------
  # installation directories
! if (OS_WINDOWS AND NOT MINGW)
    set (RUNTIME_INSTALL_DIR Bin)
    set (LIBRARY_INSTALL_DIR Lib)
    set (INCLUDE_INSTALL_DIR Include)
***************
*** 501,514 ****
  if (BUILD_SHARED_LIBS AND INSTALL_SHARED_LIBS)
    foreach (opts IN ITEMS "" _nothreads)
      if (BUILD_gflags${opts}_LIB)
!       install (TARGETS gflags${opts}_shared DESTINATION ${LIBRARY_INSTALL_DIR} EXPORT ${EXPORT_NAME})
      endif ()
    endforeach ()
  endif ()
  if (BUILD_STATIC_LIBS AND INSTALL_STATIC_LIBS)
    foreach (opts IN ITEMS "" _nothreads)
      if (BUILD_gflags${opts}_LIB)
!       install (TARGETS gflags${opts}_static DESTINATION ${LIBRARY_INSTALL_DIR} EXPORT ${EXPORT_NAME})
      endif ()
    endforeach ()
  endif ()
--- 501,520 ----
  if (BUILD_SHARED_LIBS AND INSTALL_SHARED_LIBS)
    foreach (opts IN ITEMS "" _nothreads)
      if (BUILD_gflags${opts}_LIB)
!       install (TARGETS gflags${opts}_shared EXPORT ${EXPORT_NAME}
!                RUNTIME DESTINATION ${RUNTIME_INSTALL_DIR}
!                LIBRARY DESTINATION ${LIBRARY_INSTALL_DIR}
!                ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR})
      endif ()
    endforeach ()
  endif ()
  if (BUILD_STATIC_LIBS AND INSTALL_STATIC_LIBS)
    foreach (opts IN ITEMS "" _nothreads)
      if (BUILD_gflags${opts}_LIB)
!       install (TARGETS gflags${opts}_static EXPORT ${EXPORT_NAME}
!                RUNTIME DESTINATION ${RUNTIME_INSTALL_DIR}
!                LIBRARY DESTINATION ${LIBRARY_INSTALL_DIR}
!                ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR})
      endif ()
    endforeach ()
  endif ()
diff -r -c gflags-2.2.0.orig/src/config.h.in gflags-2.2.0/src/config.h.in
*** gflags-2.2.0.orig/src/config.h.in	2016-11-25 16:54:37.000000000 -0200
--- gflags-2.2.0/src/config.h.in	2017-02-26 12:03:04.182584900 -0300
***************
*** 77,83 ****
  // ---------------------------------------------------------------------------
  // Path separator
  #ifndef PATH_SEPARATOR
! #  ifdef OS_WINDOWS
  #    define PATH_SEPARATOR  '\\'
  #  else
  #    define PATH_SEPARATOR  '/'
--- 77,83 ----
  // ---------------------------------------------------------------------------
  // Path separator
  #ifndef PATH_SEPARATOR
! #  if defined(OS_WINDOWS) && not defined(__MINGW32__)
  #    define PATH_SEPARATOR  '\\'
  #  else
  #    define PATH_SEPARATOR  '/'
