diff -c -r gflags-2.1.2.orig/CMakeLists.txt gflags-2.1.2/CMakeLists.txt
*** gflags-2.1.2.orig/CMakeLists.txt	2015-03-24 13:19:41.000000000 -0300
--- gflags-2.1.2/CMakeLists.txt	2016-03-10 19:34:47.991173900 -0300
***************
*** 321,327 ****
  
  # ----------------------------------------------------------------------------
  # installation
! if (OS_WINDOWS)
    set (RUNTIME_INSTALL_DIR Bin)
    set (LIBRARY_INSTALL_DIR Lib)
    set (INCLUDE_INSTALL_DIR Include)
--- 321,327 ----
  
  # ----------------------------------------------------------------------------
  # installation
! if (OS_WINDOWS AND NOT MINGW)
    set (RUNTIME_INSTALL_DIR Bin)
    set (LIBRARY_INSTALL_DIR Lib)
    set (INCLUDE_INSTALL_DIR Include)
***************
*** 346,352 ****
  configure_file (cmake/config.cmake.in  "${PROJECT_BINARY_DIR}/${PACKAGE_NAME}-config-install.cmake" @ONLY)
  configure_file (cmake/version.cmake.in "${PROJECT_BINARY_DIR}/${PACKAGE_NAME}-config-version.cmake" @ONLY)
  
! install (TARGETS ${TARGETS} DESTINATION ${LIBRARY_INSTALL_DIR} EXPORT gflags-lib)
  if (INSTALL_HEADERS)
    install (FILES ${PUBLIC_HDRS} DESTINATION ${INCLUDE_INSTALL_DIR}/${GFLAGS_INCLUDE_DIR})
    install (
--- 346,356 ----
  configure_file (cmake/config.cmake.in  "${PROJECT_BINARY_DIR}/${PACKAGE_NAME}-config-install.cmake" @ONLY)
  configure_file (cmake/version.cmake.in "${PROJECT_BINARY_DIR}/${PACKAGE_NAME}-config-version.cmake" @ONLY)
  
! install (TARGETS ${TARGETS} EXPORT gflags-lib
!   RUNTIME DESTINATION ${RUNTIME_INSTALL_DIR}
!   LIBRARY DESTINATION ${LIBRARY_INSTALL_DIR}
!   ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
!   )
  if (INSTALL_HEADERS)
    install (FILES ${PUBLIC_HDRS} DESTINATION ${INCLUDE_INSTALL_DIR}/${GFLAGS_INCLUDE_DIR})
    install (
diff -c -r gflags-2.1.2.orig/src/config.h.in gflags-2.1.2/src/config.h.in
*** gflags-2.1.2.orig/src/config.h.in	2015-03-24 13:19:41.000000000 -0300
--- gflags-2.1.2/src/config.h.in	2016-03-10 19:35:37.039204900 -0300
***************
*** 73,79 ****
  // ---------------------------------------------------------------------------
  // Path separator
  #ifndef PATH_SEPARATOR
! #  ifdef OS_WINDOWS
  #    define PATH_SEPARATOR  '\\'
  #  else
  #    define PATH_SEPARATOR  '/'
--- 73,79 ----
  // ---------------------------------------------------------------------------
  // Path separator
  #ifndef PATH_SEPARATOR
! #  if defined(OS_WINDOWS) && not defined(__MINGW32__)
  #    define PATH_SEPARATOR  '\\'
  #  else
  #    define PATH_SEPARATOR  '/'
