diff --git a/MinGW/Options.td b/MinGW/Options.td
index 0604b458193..efccf2d48ce 100644
--- a/MinGW/Options.td
+++ b/MinGW/Options.td
@@ -114,6 +114,8 @@ def alias_undefined_u: JoinedOrSeparate<["-"], "u">, Alias<undefined>;
 // Ignored options
 def: Joined<["-"], "O">;
 def: F<"build-id">;
+def: F<"default-image-base-high">;
+def: F<"default-image-base-low">;
 def: F<"disable-auto-image-base">;
 def: F<"enable-auto-image-base">;
 def: F<"end-group">;
