--- libmysofa-1.3.4/src/CMakeLists.txt.orig	2026-06-24 15:30:13.776230200 +0800
+++ libmysofa-1.3.4/src/CMakeLists.txt	2026-06-24 15:30:34.006088800 +0800
@@ -102,10 +102,12 @@
   target_compile_definitions(mysofa-shared PRIVATE mysofa_shared_EXPORTS)
   set_target_properties(mysofa-shared
                         PROPERTIES OUTPUT_NAME mysofa CLEAN_DIRECT_OUTPUT 1)
+  if(MSVC)
   # Rename the Windows import library (stub) of the DLL to prevent a name clash
   # with the mysofa-static library.
   set_target_properties(mysofa-shared
                         PROPERTIES ARCHIVE_OUTPUT_NAME mysofa_shared)
+  endif()
   set_target_properties(mysofa-shared
                         PROPERTIES PUBLIC_HEADER "${public-headers}")
 
