From f06eaaf3ebfdfab194466cf060d0484ae05e37ce Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Thu, 24 Jul 2014 15:59:40 +0200
Subject: [PATCH] Do not build clone example

---
 examples/Makefile.am | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/examples/Makefile.am b/examples/Makefile.am
index d21f1ea..149d26b 100644
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -11,8 +11,4 @@ EXAMPLE_PROGS		+= walk
 walk_SOURCES		= walk.c
 walk_LDADD		= $(progs_ldadd)
 
-EXAMPLE_PROGS		+= clone
-clone_SOURCES		= clone.c
-clone_LDADD		= $(progs_ldadd)
-
 # ex:set ts=8 noet:
-- 
1.9.3

