From 1c27627e75bbf1c028d3e5fec504744070ee0621 Mon Sep 17 00:00:00 2001
From: Alberto Fanjul <albertofanjul@gmail.com>
Date: Fri, 19 Oct 2018 11:20:13 +0200
Subject: [PATCH] Fix vapi generation on msys2

---
 meson_vapi_link.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson_vapi_link.py b/meson_vapi_link.py
index 435cd95..a419fc0 100644
--- a/meson_vapi_link.py
+++ b/meson_vapi_link.py
@@ -4,7 +4,7 @@ import os
 import subprocess
 import sys
 
-vapidir = os.path.join(sys.argv[1], 'vala', 'vapi')
+vapidir = os.path.join('share', 'vala', 'vapi')
 
 destdir = os.environ.get('DESTDIR', '')
 dest_vapidir = os.path.normpath(destdir + os.sep + vapidir)
-- 
2.14.2.windows.3

