--- a/meson_vapi_link.py
+++ b/meson_vapi_link.py
@@ -19,9 +19,12 @@
 for ext in ['vapi', 'deps']:
   src = '{}.{}'.format(new, ext)
   dest = '{}.{}'.format(old, ext)
-  try:
-      os.symlink(src, dest)
-  except FileExistsError:
-      pass
-  except OSError:
-      shutil.copy(src, dest)
+  if os.name == 'nt':
+    shutil.copy(src, dest)
+  else:
+    try:
+        os.symlink(src, dest)
+    except FileExistsError:
+        pass
+    except OSError:
+        shutil.copy(src, dest)
