--- libcddb-1.3.2/configure.ac.orig	2013-04-07 03:34:43 +0400
+++ libcddb-1.3.2/configure.ac	2013-04-07 03:36:58 +0400
@@ -13,7 +13,7 @@
                  examples/Makefile     \
                  tests/Makefile        \
                  tests/settings.sh])
-AM_INIT_AUTOMAKE([dist-bzip2])
+AM_INIT_AUTOMAKE([silent-rules dist-bzip2])
 AM_MAINTAINER_MODE
 
 LIBCDDB_VERSION_NUM=131
--- libcddb-1.3.2/configure.ac.orig	2013-04-07 03:37:21 +0400
+++ libcddb-1.3.2/configure.ac	2013-04-07 03:38:02 +0400
@@ -2,7 +2,7 @@
 AC_PREREQ(2.55)
 AC_INIT(libcddb, 1.3.2)
 AC_CONFIG_SRCDIR(configure.ac)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 AC_CONFIG_FILES([Makefile              \
                  libcddb.pc            \
                  include/Makefile      \
