--- a/share/qbs/modules/cpp/CppModule.qbs
+++ b/share/qbs/modules/cpp/CppModule.qbs
@@ -189,7 +189,7 @@
     property string debugInfoSuffix: ""
     property string debugInfoBundleSuffix: ""
     property string variantSuffix: ""
-    property string dynamicLibraryImportSuffix: ".lib"
+    property string dynamicLibraryImportSuffix: ".dll.a"
     property string objectSuffix: ".o"
     property string linkerMapSuffix: ".map"
     property string compilerListingSuffix: ".lst"
--- a/share/qbs/imports/qbs/PathTools/path-tools.js
+++ b/share/qbs/imports/qbs/PathTools/path-tools.js
@@ -240,7 +240,7 @@
     if (targetOS.includes("windows")) {
         if (forImport) {
             return []
-                .concat(types.includes("shared") ? [".lib"] : [])
+                .concat(types.includes("shared") ? [".lib", ".dll.a"] : [])
                 // mingw uses .a for static libs
                 .concat(types.includes("static") ? [".lib", ".a"] : []);
         }
--- a/share/qbs/imports/qbs/Probes/qmake-probe.js
+++ b/share/qbs/imports/qbs/Probes/qmake-probe.js
@@ -209,8 +209,6 @@
 
     // Some Linux distributions rename the qtmain library.
     var qtMainCandidates = ["qtmain"];
-    if (isMinGW && qtProps.qtMajorVersion === 5)
-        qtMainCandidates.push("qt5main");
     if (qtProps.qtMajorVersion === 6)
         qtMainCandidates.push("Qt6EntryPoint");
 
--- a/tests/auto/blackbox/testdata/setup-run-environment/setup-run-environment.qbs
+++ b/tests/auto/blackbox/testdata/setup-run-environment/setup-run-environment.qbs
@@ -69,7 +69,7 @@
         Properties {
             condition: Host.os().includes("windows") && qbs.toolchain.includes("clang")
             cpp.dynamicLibraryPrefix: "lib"
-            cpp.dynamicLibraryImportSuffix: ".a"
+            cpp.dynamicLibraryImportSuffix: ".dll.a"
         }
 
         install: true
