--- a/share/qbs/imports/qbs/Probes/LibraryProbe.qbs
+++ b/share/qbs/imports/qbs/Probes/LibraryProbe.qbs
@@ -32,7 +32,7 @@
     property string endianness
     nameSuffixes: {
         if (qbs.targetOS.contains("windows"))
-            return [".lib"];
+            return [".dll.a", ".a"];
         if (qbs.targetOS.contains("macos"))
             return [".dylib", ".a"];
         return [".so", ".a"];
--- a/share/qbs/imports/qbs/Probes/qmake-probe.js
+++ b/share/qbs/imports/qbs/Probes/qmake-probe.js
@@ -183,8 +183,6 @@
 
     // Some Linux distributions rename the qtmain library.
     var qtMainCandidates = ["qtmain"];
-    if (isMinGW && qtProps.qtMajorVersion === 5)
-        qtMainCandidates.push("qt5main");
     if (qtProps.qtMajorVersion === 6)
         qtMainCandidates.push("Qt6EntryPoint");
 
--- a/tests/auto/blackbox/testdata/setup-run-environment/setup-run-environment.qbs
+++ b/tests/auto/blackbox/testdata/setup-run-environment/setup-run-environment.qbs
@@ -67,7 +67,7 @@
         Properties {
             condition: Host.os().includes("windows") && qbs.toolchain.includes("clang")
             cpp.dynamicLibraryPrefix: "lib"
-            cpp.dynamicLibraryImportSuffix: ".a"
+            cpp.dynamicLibraryImportSuffix: ".dll.a"
         }
         cpp.dynamicLibraryPrefix: original
         cpp.dynamicLibraryImportSuffix: original
