--- plasma-framework-5.59.0/src/desktoptheme/CMakeLists.txt.orig	2019-06-14 00:34:12.375538500 +0300
+++ plasma-framework-5.59.0/src/desktoptheme/CMakeLists.txt	2019-06-14 00:35:11.819938500 +0300
@@ -2,7 +2,7 @@
 option(GZIP_DESKTOPTHEME_SVG "Install Desktop Theme SVG files as .svgz." ON)
 
 if (GZIP_DESKTOPTHEME_SVG)
-    if(WIN32)
+    if(MSVC)
         find_package(7z)
         set_package_properties(7z PROPERTIES
             TYPE REQUIRED
@@ -62,7 +62,7 @@
         if (GZIP_DESKTOPTHEME_SVG)
             set(_gzipped_file_displayname "${desktoptheme_COMPONENTDIR}/${_fileName}z")
             set(_gzipped_file "${CMAKE_CURRENT_BINARY_DIR}/${desktoptheme_GZIPDIR}/${_fileName}z")
-            if(WIN32)
+            if(MSVC)
                 add_custom_command(
                     OUTPUT ${_gzipped_file}
                     COMMAND ${7z_EXECUTABLE}
