--- gdk-pixbuf-2.26.0/gdk-pixbuf/Makefile.am.orig	2012-02-05 03:21:30 +0400
+++ gdk-pixbuf-2.26.0/gdk-pixbuf/Makefile.am	2012-05-06 14:09:29 +0400
@@ -11,7 +11,7 @@
 
 if OS_WIN32
 gdk_pixbuf_def = gdk_pixbuf.def
-gdk_pixbuf_symbols = -export-symbols $(srcdir)/gdk_pixbuf.def
+gdk_pixbuf_symbols = 
 
 gdk_pixbuf_win32_res = gdk_pixbuf-win32-res.o
 gdk_pixbuf_win32_res_ldflag = -Wl,gdk_pixbuf-win32-res.o
