--- gdk-pixbuf-2.26.0/configure.ac.orig	2012-03-20 20:46:27 +0400
+++ gdk-pixbuf-2.26.0/configure.ac	2012-05-06 14:05:36 +0400
@@ -264,7 +264,7 @@
 ## annoying to construct
 PKG_CHECK_MODULES(BASE_DEPENDENCIES, [glib-2.0 >= glib_required_version])
 
-if test "$os_win32" != yes; then
+if test "$os_win32" != no_idea_why_using_export_symbols_regex_on_w32_is_bad ; then
     # libtool option to control which symbols are exported
     # right now, symbols starting with _ are not exported
     LIBTOOL_EXPORT_OPTIONS='-export-symbols-regex "^[[^_]].*"'
@@ -516,7 +516,7 @@
 AM_CONDITIONAL(BUILD_GDIPLUS_LOADERS, [ test x$os_win32 = xyes && test x$with_gdiplus != xno ])
 
 dnl Test for libtiff
-if test x$os_win32 = xno || test x$with_gdiplus = xno; then
+if test x$os_win32 = xno_idea_why_libtiff_is_not_used_on_w32 || test x$with_gdiplus = xforget_about_gdiplus_it_sux || test always_build = always_build ; then
   if test x$with_libtiff != xno && test -z "$LIBTIFF"; then
     AC_CHECK_LIB(tiff, TIFFReadRGBAImageOriented,
       [AC_CHECK_HEADER(tiffio.h,
@@ -542,7 +542,7 @@
 fi
 
 dnl Test for libjpeg
-if test x$os_win32 = xno || test x$with_gdiplus = xno; then
+if test x$os_win32 = xno_idea_why_libjpeg_is_not_used_on_w32 || test x$with_gdiplus = xforget_about_gdiplus_it_sux || test always_build = always_build ; then
   if test x$with_libjpeg != xno && test -z "$LIBJPEG"; then
     AC_CHECK_LIB(jpeg, jpeg_destroy_decompress,
       jpeg_ok=yes,
@@ -676,7 +676,13 @@
 fi
 if test x$os_win32 = xyes && test x$with_gdiplus != xno; then
   # Skip PNG, see comment above
-  gdip_formats="bmp emf gif ico jpeg tiff wmf"
+  gdip_formats="bmp emf gif ico wmf"
+  if test x$with_libtiff = xno; then
+    gdip_formats="$gdip_formats tiff"
+  fi
+  if test x$with_libjpeg = xno; then
+    gdip_formats="$gdip_formats jpeg"
+  fi
   for f in $gdip_formats; do
     all_loaders="$all_loaders,gdip-$f"
   done
--- gdk-pixbuf-2.26.0/gdk-pixbuf/Makefile.am.orig	2012-05-06 14:10:43 +0400
+++ gdk-pixbuf-2.26.0/gdk-pixbuf/Makefile.am	2012-05-06 14:17:33 +0400
@@ -366,6 +366,9 @@
 ICO_LIB = libpixbufloader-ico.la
 endif
 
+# End of loaders not built if building GDI+ loader
+endif
+
 if HAVE_JPEG
 if INCLUDE_JPEG
 STATIC_JPEG_LIB = libstatic-pixbufloader-jpeg.la
@@ -382,9 +385,6 @@
 endif
 endif
 
-# End of loaders not built if building GDI+ loader
-endif
-
 if HAVE_PNG
 if INCLUDE_PNG
 STATIC_PNG_LIB = libstatic-pixbufloader-png.la
--- gdk-pixbuf-2.26.0/gdk-pixbuf/Makefile.am.orig	2012-05-06 14:18:03 +0400
+++ gdk-pixbuf-2.26.0/gdk-pixbuf/Makefile.am	2012-05-06 14:23:37 +0400
@@ -223,9 +223,7 @@
 	libstatic-pixbufloader-gdip-wmf.la \
 	libstatic-pixbufloader-gdip-emf.la \
 	libstatic-pixbufloader-gdip-bmp.la \
-	libstatic-pixbufloader-gdip-gif.la \
-	libstatic-pixbufloader-gdip-jpeg.la \
-	libstatic-pixbufloader-gdip-tiff.la
+	libstatic-pixbufloader-gdip-gif.la
 
 libstatic_pixbufloader_gdip_ico_la_SOURCES = 	\
 	io-gdip-native.h		\
@@ -261,9 +259,7 @@
 	libpixbufloader-gdip-wmf.la \
 	libpixbufloader-gdip-emf.la \
 	libpixbufloader-gdip-bmp.la \
-	libpixbufloader-gdip-gif.la \
-	libpixbufloader-gdip-jpeg.la \
-	libpixbufloader-gdip-tiff.la
+	libpixbufloader-gdip-gif.la
 
 libpixbufloader_gdip_ico_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
 libpixbufloader_gdip_ico_la_SOURCES = 	\
