diff -Naur gdk-pixbuf-2.36.0.orig/gdk-pixbuf/gdk-pixbuf-loader.c gdk-pixbuf-2.36.0/gdk-pixbuf/gdk-pixbuf-loader.c
--- gdk-pixbuf-2.36.0.orig/gdk-pixbuf/gdk-pixbuf-loader.c	2016-08-02 18:50:41.000000000 +0200
+++ gdk-pixbuf-2.36.0/gdk-pixbuf/gdk-pixbuf-loader.c	2016-10-20 18:55:04.046471400 +0200
@@ -735,7 +735,7 @@
         if (priv->animation)
                 return gdk_pixbuf_animation_get_static_image (priv->animation);
         else
-                return NULL;
+                return priv;
 }
 
 /**
