diff -Naur geos-3.6.2.orig/CMakeLists.txt geos-3.6.2/CMakeLists.txt
--- geos-3.6.2.orig/CMakeLists.txt	2018-05-23 19:46:15.542316000 -0400
+++ geos-3.6.2/CMakeLists.txt	2018-05-23 20:36:33.943193000 -0400
@@ -116,7 +116,7 @@
     "Forcing IEEE 754 using flag -ffloat-store - ${GEOS_ENABLE_FLOATSTORE}")
 
   # Warnings specification
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-long-long")
+  #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-long-long")
 
   # Turn on Position Independent Code generation for GEOS C shared library
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
@@ -191,7 +191,10 @@
 include(CheckPrototypeExists)
 include(CheckSymbolExists)
 
-check_prototype_exists(isnan cmath HAVE_STD_ISNAN)
+check_cxx_source_compiles(
+  "#include <cmath>
+  int main() { double x=0.1; return std::isnan(x)?1:0; }"
+  HAVE_STD_ISNAN)
 if(NOT HAVE_STD_ISNAN)
   if(MSVC)
     check_prototype_exists(_isnan float.h HAVE_ISNAN)
@@ -205,7 +208,10 @@
   endif()
 endif()
 
-check_prototype_exists(isfinite cmath HAVE_STD_ISFINITE)
+check_cxx_source_compiles(
+  "#include <cmath>
+  int main() { double x=0.1; return std::isfinite(x)?1:0; }"
+  HAVE_STD_ISFINITE)
 
 if(NOT HAVE_STD_ISFINITE)
   if(MSVC)
@@ -347,7 +353,7 @@
 #################################################################################
 # Configure subdirectories
 #################################################################################
-include(GenerateSourceGroups)
+#include(GenerateSourceGroups)
 
 add_subdirectory(include)
 add_subdirectory(src)
diff -Naur geos-3.6.2.orig/include/CMakeLists.txt geos-3.6.2/include/CMakeLists.txt
--- geos-3.6.2.orig/include/CMakeLists.txt	2018-05-23 19:46:15.521315800 -0400
+++ geos-3.6.2/include/CMakeLists.txt	2018-05-23 20:37:32.614369700 -0400
@@ -54,4 +54,4 @@
 #################################################################################
 # Group source files for IDE source explorers (e.g. Visual Studio)
 #################################################################################
-GenerateSourceGroups(include)
+#GenerateSourceGroups(include)
diff -Naur geos-3.6.2.orig/src/CMakeLists.txt geos-3.6.2/src/CMakeLists.txt
--- geos-3.6.2.orig/src/CMakeLists.txt	2018-05-23 19:46:15.848312300 -0400
+++ geos-3.6.2/src/CMakeLists.txt	2018-05-23 20:37:52.022699800 -0400
@@ -105,4 +105,4 @@
 #################################################################################
 # Group source files for IDE source explorers (e.g. Visual Studio)
 #################################################################################
-GenerateSourceGroups(src)
+#GenerateSourceGroups(src)
diff -Naur geos-3.6.2.orig/tests/unit/CMakeLists.txt geos-3.6.2/tests/unit/CMakeLists.txt
--- geos-3.6.2.orig/tests/unit/CMakeLists.txt	2018-05-23 19:46:14.736324600 -0400
+++ geos-3.6.2/tests/unit/CMakeLists.txt	2018-05-23 20:38:21.110716300 -0400
@@ -38,4 +38,4 @@
 #################################################################################
 # Group source files for IDE source explorers (e.g. Visual Studio)
 #################################################################################
-GenerateSourceGroups(tests/unit)
\ No newline at end of file
+#GenerateSourceGroups(tests/unit)
\ No newline at end of file
