From 84ea10dc61fb156341bc7eba4d0b2025db801e9b Mon Sep 17 00:00:00 2001
From: Robert Roth <robert.roth.off@gmail.com>
Date: Wed, 15 Jun 2022 18:40:03 +0300
Subject: [PATCH] Remove unnecessary problematic XML parser cleanup call (fixes
 #283)

---
 lib/currency-provider.vala | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/currency-provider.vala b/lib/currency-provider.vala
index bd471598..bcef3630 100644
--- a/lib/currency-provider.vala
+++ b/lib/currency-provider.vala
@@ -428,7 +428,6 @@ public class EcbCurrencyProvider : AbstractCurrencyProvider {
                 node = null;
         }
 
-        Xml.Parser.cleanup ();
         base.do_load_rates ();
     }
 
-- 
GitLab

