diff --git a/meson.build b/meson.build
index 1aec95adc..872ae73f5 100644
--- a/meson.build
+++ b/meson.build
@@ -205,9 +205,9 @@ elif host_machine.system() == 'windows'
   winsock_dep = meson.get_compiler('c').find_library('wsock32')
   ole32_dep = meson.get_compiler('c').find_library('ole32')
   ssp_dep = meson.get_compiler('c').find_library('ssp')
-  pcreposix_dep = meson.get_compiler('c').find_library('pcreposix')
+  pcre2_posix_dep = meson.get_compiler('c').find_library('pcre2-posix')
   platform_socket_dep = [ws2_32_dep, winsock_dep]
-  platform_dep = [ole32_dep, ssp_dep, pcreposix_dep]
+  platform_dep = [ole32_dep, ssp_dep, pcre2_posix_dep]
 elif host_machine.system() == 'freebsd'
   cdata.set('OS_IS_FREEBSD', 1)
 #elif host_machine.system() == 'solaris'
@@ -241,10 +241,10 @@ check_headers = [
   'netinet/in_systm.h',
   'netinet/ip.h',
   'netinet/tcp.h',
-  'pcreposix.h',
+  'pcre2posix.h',
   'poll.h',
   'pwd.h',
-  'regex.h',
+#  'regex.h',
   'sched.h',
   'stdint.h',
   'sys/atomic.h',
diff --git a/src/modules/module-match.c b/src/modules/module-match.c
index 9b5e76cd7..1882b7746 100644
--- a/src/modules/module-match.c
+++ b/src/modules/module-match.c
@@ -30,8 +30,8 @@
 
 #if defined(HAVE_REGEX_H)
 #include <regex.h>
-#elif defined(HAVE_PCREPOSIX_H)
-#include <pcreposix.h>
+#elif defined(HAVE_PCRE2POSIX_H)
+#include <pcre2posix.h>
 #endif
 
 #include <pulse/xmalloc.h>
diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index 5d2bae92f..ee7184f0a 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -48,8 +48,8 @@
 
 #if defined(HAVE_REGEX_H)
 #include <regex.h>
-#elif defined(HAVE_PCREPOSIX_H)
-#include <pcreposix.h>
+#elif defined(HAVE_PCRE2POSIX_H)
+#include <pcre2posix.h>
 #endif
 
 #ifdef HAVE_STRTOD_L
@@ -801,7 +801,7 @@ void pa_reset_priority(void) {
 
 /* Check whenever any substring in v matches the provided regex. */
 int pa_match(const char *expr, const char *v) {
-#if defined(HAVE_REGEX_H) || defined(HAVE_PCREPOSIX_H)
+#if defined(HAVE_REGEX_H) || defined(HAVE_PCRE2POSIX_H)
     int k;
     regex_t re;
     int r;
@@ -835,7 +835,7 @@ int pa_match(const char *expr, const char *v) {
 
 /* Check whenever the provided regex pattern is valid. */
 bool pa_is_regex_valid(const char *expr) {
-#if defined(HAVE_REGEX_H) || defined(HAVE_PCREPOSIX_H)
+#if defined(HAVE_REGEX_H) || defined(HAVE_PCRE2POSIX_H)
     regex_t re;
 
     if (expr == NULL || regcomp(&re, expr, REG_NOSUB|REG_EXTENDED) != 0) {
