--- src/src.pro.orig	2013-09-08 13:01:33.000000000 +0300
+++ src/src.pro	2014-07-16 20:02:20.842493000 +0300
@@ -7,7 +7,7 @@
 TEMPLATE = lib
 QT += network
 CONFIG += \
-    create_prl
+    create_prl release
 
 !macx: CONFIG += static_and_shared
 
@@ -16,6 +16,7 @@
 INC_DIR = ../include
 
 INCLUDEPATH += .
+LIBS += -L$$DESTDIR
 
 PUBLIC_HEADERS += kqoauthmanager.h \
                   kqoauthrequest.h \
@@ -67,12 +68,13 @@
     postinstall.path =  target.path
     postinstall.extra = install_name_tool -id $${target.path}/$${QMAKE_FRAMEWORK_BUNDLE_NAME}.framework/Versions/0/$${TARGET} $${target.path}/$${QMAKE_FRAMEWORK_BUNDLE_NAME}.framework/Versions/0/$${TARGET}
 }
-else:unix {
+else:if(unix|win32-g++) {
     isEmpty( PREFIX ):INSTALL_PREFIX = /usr
     else:INSTALL_PREFIX = $${PREFIX}
 
     # this creates a pkgconfig file
-    system( ./pcfile.sh $${INSTALL_PREFIX} $${VERSION} )
+    #system( ./pcfile.sh $${INSTALL_PREFIX} $${VERSION} )
+    features.path = $${INSTALL_PREFIX}/share/qt4/mkspecs/features
     pkgconfig.files = kqoauth.pc
     
     target.path = $${INSTALL_PREFIX}/lib
