--- libsmi-0.5.0/lib/Makefile.am	2014-10-13 14:20:46.000000000 +0100
+++ libsmi-0.5.0/lib/Makefile.am	2023-01-24 19:24:37.817251400 +0000
@@ -43,7 +43,7 @@ libsmi_la_SOURCES	= common.c error.c uti
 		  	  parser-smi.c scanner-smi.c \
 		  	  parser-sming.c scanner-sming.c \
 		  	  parser-yang.c scanner-yang.c
-libsmi_la_LDFLAGS	= -version-info @VERSION_LIBTOOL@
+libsmi_la_LDFLAGS	= -no-undefined -version-info @VERSION_LIBTOOL@
 
 parser-smi.c parser-smi.tab.h: parser-smi.y scanner-smi.h parser-smi.h error.h
 	$(BISON) --defines=parser-smi.tab.h -t -psmi -o parser-smi.c parser-smi.y
