diff -aur 002/configure 003/configure
--- 002/configure	2014-12-17 16:53:14.776367100 -0200
+++ 003/configure	2014-12-17 16:59:46.995117100 -0200
@@ -14624,9 +14624,9 @@
 _ACEOF
 
       LIBS="$LIBS -liconv"
-      LDFLAGS="$LDFLAGS -L/usr/lib"
-      CFLAGS="$CFLAGS -I/usr/include"
-      CPPLAGS="$CPPFLAGS -I/usr/include"
+      LDFLAGS="$LDFLAGS -L${prefix}/lib"
+      CFLAGS="$CFLAGS -I${prefix}/include"
+      CPPLAGS="$CPPFLAGS -I${prefix}/include"
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for iconv in -liconv" >&5
 $as_echo_n "checking for iconv in -liconv... " >&6; }
diff -aur 002/configure.ac 003/configure.ac
--- 002/configure.ac	2014-12-17 16:53:04.545898400 -0200
+++ 003/configure.ac	2014-12-17 17:00:02.793945300 -0200
@@ -869,9 +869,9 @@
   AC_CHECK_HEADERS(iconv.h,
     [
       LIBS="$LIBS -liconv"
-      LDFLAGS="$LDFLAGS -L/usr/lib"
-      CFLAGS="$CFLAGS -I/usr/include"
-      CPPLAGS="$CPPFLAGS -I/usr/include"
+      LDFLAGS="$LDFLAGS -L${prefix}/lib"
+      CFLAGS="$CFLAGS -I${prefix}/include"
+      CPPLAGS="$CPPFLAGS -I${prefix}/include"
 
       AC_MSG_CHECKING(for iconv in -liconv)
       AC_TRY_LINK(
