diff -aur 003/includes/silcwin32.h 004/includes/silcwin32.h
--- 003/includes/silcwin32.h 2014-12-17 16:55:41.595703100 -0200
+++ 004/includes/silcwin32.h 2014-12-17 17:46:48.268554600 -0200
@@ -71,7 +71,7 @@
 #undef sleep
 #define sleep(x) Sleep((x) * 1000)
 
-#if _MSC_VER < 1300
+#if (defined(_MSC_VER) && (_MSC_VER < 1300))
 #define SetWindowLongPtr SetWindowLong
 #define GetWindowLongPtr GetWindowLong
 #endif /* _MSC_VER < 1300 */
