diff --git a/CMakeLists.txt b/CMakeLists.txt
index 647e3c7d..4ca3150b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ include(ECMQtDeclareLoggingCategory)
 
 find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Sql)
 
-find_package(KF5 ${KF5_MIN_VERSION} REQUIRED I18n TextWidgets KIO XmlGui Notifications Config CoreAddons DocTools Sonnet DBusAddons Crash)
+find_package(KF5 ${KF5_MIN_VERSION} REQUIRED I18n TextWidgets KIO XmlGui Notifications Config CoreAddons Sonnet DBusAddons Crash)
 
 add_definitions(
     -DQT_USE_QSTRINGBUILDER
@@ -41,7 +41,6 @@ set_package_properties(
     TYPE        REQUIRED 
     PURPOSE     "Required to build Lokalize.")
 
-add_subdirectory(doc)
 add_subdirectory(src)
 add_subdirectory(scripts)
 add_subdirectory(icons)
