diff --git a/src/tm/jobs.cpp b/src/tm/jobs.cpp
index 44abc16a..f69fc7ee 100644
--- a/src/tm/jobs.cpp
+++ b/src/tm/jobs.cpp
@@ -60,7 +60,7 @@ void TM::cancelAllJobs()
 }
 static const QString getConnectionName(const QString dbName)
 {
-    return dbName + QString::number((long)QThread::currentThreadId());
+    return dbName + QString::number((std::intptr_t)QThread::currentThreadId());
 }
 
 static qlonglong newTMSourceEntryCount = 0;
