--- pycairo-1.10.0/src/context.c.orig	2014-02-21 20:26:47.585600000 +0400
+++ pycairo-1.10.0/src/context.c	2014-02-21 20:27:06.711200000 +0400
@@ -1329,9 +1329,9 @@
 };
 
 PyTypeObject PycairoContext_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //  PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Context",                    /* tp_name */
   sizeof(PycairoContext),             /* tp_basicsize */
   0,                                  /* tp_itemsize */
--- pycairo-1.10.0/src/font.c.orig	2014-02-21 20:26:47.585600000 +0400
+++ pycairo-1.10.0/src/font.c	2014-02-21 20:27:06.711200000 +0400
@@ -93,9 +93,9 @@
 */
 
 PyTypeObject PycairoFontFace_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.FontFace",                   /* tp_name */
   sizeof(PycairoFontFace),            /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -178,9 +178,9 @@
 
 
 PyTypeObject PycairoToyFontFace_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.ToyFontFace",                /* tp_name */
   sizeof(PycairoToyFontFace),         /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -340,9 +340,9 @@
 
 
 PyTypeObject PycairoScaledFont_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.ScaledFont",                 /* tp_name */
   sizeof(PycairoScaledFont),          /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -527,9 +527,9 @@
 
 
 PyTypeObject PycairoFontOptions_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.FontOptions",                /* tp_name */
   sizeof(PycairoFontOptions),         /* tp_basicsize */
   0,                                  /* tp_itemsize */
--- pycairo-1.10.0/src/path.c.orig	2014-02-21 20:26:47.585600000 +0400
+++ pycairo-1.10.0/src/path.c	2014-02-21 20:27:06.711200000 +0400
@@ -165,9 +165,9 @@
 
 
 PyTypeObject PycairoPath_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,				        /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,				        /* ob_size */
   "cairo.Path",			/* tp_name */
   sizeof(PycairoPath),		/* tp_basicsize */
   0,					/* tp_itemsize */
@@ -286,9 +286,9 @@
 }
 
 PyTypeObject PycairoPathiter_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,				        /* ob_size */
   "cairo.Pathiter",                   /* tp_name */
   sizeof(PycairoPathiter),            /* tp_basicsize */
   0,                                  /* tp_itemsize */
--- pycairo-1.10.0/src/matrix.c.orig	2014-02-21 20:26:47.585600000 +0400
+++ pycairo-1.10.0/src/matrix.c	2014-02-21 20:27:06.711200000 +0400
@@ -284,9 +284,9 @@
 };
 
 PyTypeObject PycairoMatrix_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Matrix",                     /* tp_name */
   sizeof(PycairoMatrix),              /* tp_basicsize */
   0,                                  /* tp_itemsize */
--- pycairo-1.10.0/src/pattern.c.orig	2014-02-21 20:26:47.585600000 +0400
+++ pycairo-1.10.0/src/pattern.c	2014-02-21 20:27:06.711200000 +0400
@@ -153,9 +153,9 @@
 };
 
 PyTypeObject PycairoPattern_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Pattern",                    /* tp_name */
   sizeof(PycairoPattern),             /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -223,9 +223,9 @@
 };
 
 PyTypeObject PycairoSolidPattern_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.SolidPattern",               /* tp_name */
   sizeof(PycairoSolidPattern),        /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -316,9 +316,9 @@
 };
 
 PyTypeObject PycairoSurfacePattern_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.SurfacePattern",             /* tp_name */
   sizeof(PycairoSurfacePattern),      /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -402,9 +402,9 @@
 };
 
 PyTypeObject PycairoGradient_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Gradient",                   /* tp_name */
   sizeof(PycairoGradient),            /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -473,9 +473,9 @@
 };
 
 PyTypeObject PycairoLinearGradient_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.LinearGradient",             /* tp_name */
   sizeof(PycairoLinearGradient),      /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -546,9 +546,9 @@
 };
 
 PyTypeObject PycairoRadialGradient_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.RadialGradient",             /* tp_name */
   sizeof(PycairoRadialGradient),      /* tp_basicsize */
   0,                                  /* tp_itemsize */
--- pycairo-1.10.0/src/surface.c.orig	2014-02-21 20:26:47.585600000 +0400
+++ pycairo-1.10.0/src/surface.c	2014-02-21 20:27:06.711200000 +0400
@@ -347,9 +347,9 @@
 
 
 PyTypeObject PycairoSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Surface",                    /* tp_name */
   sizeof(PycairoSurface),             /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -646,9 +646,9 @@
 
 
 PyTypeObject PycairoImageSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //  PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.ImageSurface",               /* tp_name */
   sizeof(PycairoImageSurface),        /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -771,9 +771,9 @@
 };
 
 PyTypeObject PycairoPDFSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //  PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.PDFSurface",                 /* tp_name */
   sizeof(PycairoPDFSurface),          /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -974,9 +974,9 @@
 };
 
 PyTypeObject PycairoPSSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.PSSurface",                  /* tp_name */
   sizeof(PycairoPSSurface),           /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -1089,9 +1089,9 @@
 };
 
 PyTypeObject PycairoSVGSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.SVGSurface",                 /* tp_name */
   sizeof(PycairoSVGSurface),          /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -1155,9 +1155,9 @@
 };
 
 PyTypeObject PycairoWin32Surface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Win32Surface",               /* tp_name */
   sizeof(PycairoWin32Surface),        /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -1218,9 +1218,9 @@
 };
 
 PyTypeObject PycairoWin32PrintingSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.Win32PrintingSurface",       /* tp_name */
   sizeof(PycairoWin32PrintingSurface), /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -1368,9 +1368,9 @@
 };
 
 PyTypeObject PycairoXCBSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.XCBSurface",                 /* tp_name */
   sizeof(PycairoXCBSurface),          /* tp_basicsize */
   0,                                  /* tp_itemsize */
@@ -1449,9 +1449,9 @@
 };
 
 PyTypeObject PycairoXlibSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
-  //PyObject_HEAD_INIT(NULL)
-  //0,                                  /* ob_size */
+  //PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyObject_HEAD_INIT(NULL)
+  0,                                  /* ob_size */
   "cairo.XlibSurface",                /* tp_name */
   sizeof(PycairoXlibSurface),         /* tp_basicsize */
   0,                                  /* tp_itemsize */
