diff -Nur pycairo-1.10.0/src/context.c pycairo-1.10.0-new/src/context.c
--- pycairo-1.10.0/src/context.c	2010-09-19 05:06:09.000000000 +0200
+++ pycairo-1.10.0-new/src/context.c	2016-12-01 20:18:27.516585100 +0100
@@ -1329,7 +1329,7 @@
 };
 
 PyTypeObject PycairoContext_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //  PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Context",                    /* tp_name */
diff -Nur pycairo-1.10.0/src/font.c pycairo-1.10.0-new/src/font.c
--- pycairo-1.10.0/src/font.c	2010-09-19 05:09:23.000000000 +0200
+++ pycairo-1.10.0-new/src/font.c	2016-12-01 20:24:01.738701500 +0100
@@ -90,7 +90,7 @@
 */
 
 PyTypeObject PycairoFontFace_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.FontFace",                   /* tp_name */
@@ -178,7 +178,7 @@
 
 
 PyTypeObject PycairoToyFontFace_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.ToyFontFace",                /* tp_name */
@@ -340,7 +340,7 @@
 
 
 PyTypeObject PycairoScaledFont_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.ScaledFont",                 /* tp_name */
@@ -527,7 +527,7 @@
 
 
 PyTypeObject PycairoFontOptions_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.FontOptions",                /* tp_name */
diff -Nur pycairo-1.10.0/src/matrix.c pycairo-1.10.0-new/src/matrix.c
--- pycairo-1.10.0/src/matrix.c	2010-09-11 16:13:19.000000000 +0200
+++ pycairo-1.10.0-new/src/matrix.c	2016-12-01 20:22:42.873190700 +0100
@@ -284,7 +284,7 @@
 };
 
 PyTypeObject PycairoMatrix_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Matrix",                     /* tp_name */
diff -Nur pycairo-1.10.0/src/path.c pycairo-1.10.0-new/src/path.c
--- pycairo-1.10.0/src/path.c	2010-09-11 16:06:57.000000000 +0200
+++ pycairo-1.10.0-new/src/path.c	2016-12-01 20:22:39.493997400 +0100
@@ -165,7 +165,7 @@
 
 
 PyTypeObject PycairoPath_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,				        /* ob_size */
   "cairo.Path",			/* tp_name */
@@ -286,7 +286,7 @@
 }
 
 PyTypeObject PycairoPathiter_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Pathiter",                   /* tp_name */
diff -Nur pycairo-1.10.0/src/pattern.c pycairo-1.10.0-new/src/pattern.c
--- pycairo-1.10.0/src/pattern.c	2011-03-28 05:23:25.000000000 +0200
+++ pycairo-1.10.0-new/src/pattern.c	2016-12-01 20:23:50.691069600 +0100
@@ -153,7 +153,7 @@
 };
 
 PyTypeObject PycairoPattern_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Pattern",                    /* tp_name */
@@ -223,7 +223,7 @@
 };
 
 PyTypeObject PycairoSolidPattern_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.SolidPattern",               /* tp_name */
@@ -316,7 +316,7 @@
 };
 
 PyTypeObject PycairoSurfacePattern_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.SurfacePattern",             /* tp_name */
@@ -402,7 +402,7 @@
 };
 
 PyTypeObject PycairoGradient_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Gradient",                   /* tp_name */
@@ -473,7 +473,7 @@
 };
 
 PyTypeObject PycairoLinearGradient_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.LinearGradient",             /* tp_name */
@@ -546,7 +546,7 @@
 };
 
 PyTypeObject PycairoRadialGradient_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.RadialGradient",             /* tp_name */
diff -Nur pycairo-1.10.0/src/surface.c pycairo-1.10.0-new/src/surface.c
--- pycairo-1.10.0/src/surface.c	2010-09-19 11:36:54.000000000 +0200
+++ pycairo-1.10.0-new/src/surface.c	2016-12-01 20:22:06.819128500 +0100
@@ -347,7 +347,7 @@
 
 
 PyTypeObject PycairoSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Surface",                    /* tp_name */
@@ -646,7 +646,7 @@
 
 
 PyTypeObject PycairoImageSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //  PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.ImageSurface",               /* tp_name */
@@ -771,7 +771,7 @@
 };
 
 PyTypeObject PycairoPDFSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //  PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.PDFSurface",                 /* tp_name */
@@ -974,7 +974,7 @@
 };
 
 PyTypeObject PycairoPSSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.PSSurface",                  /* tp_name */
@@ -1089,7 +1089,7 @@
 };
 
 PyTypeObject PycairoSVGSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.SVGSurface",                 /* tp_name */
@@ -1155,7 +1155,7 @@
 };
 
 PyTypeObject PycairoWin32Surface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Win32Surface",               /* tp_name */
@@ -1218,7 +1218,7 @@
 };
 
 PyTypeObject PycairoWin32PrintingSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.Win32PrintingSurface",       /* tp_name */
@@ -1368,7 +1368,7 @@
 };
 
 PyTypeObject PycairoXCBSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.XCBSurface",                 /* tp_name */
@@ -1449,7 +1449,7 @@
 };
 
 PyTypeObject PycairoXlibSurface_Type = {
-  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  PyVarObject_HEAD_INIT(NULL, 0)
   //PyObject_HEAD_INIT(NULL)
   //0,                                  /* ob_size */
   "cairo.XlibSurface",                /* tp_name */
