--- setup.py    2016-12-01 18:38:00.167840500 +0100
+++ setup_new.py        2016-12-01 19:51:56.800601300 +0100
@@ -114,10 +114,10 @@
   version = pycairo_version,
   description = "python interface for cairo",
   ext_modules = [cairo],
+  package_dir = {"cairo": "src"},
+  packages = ["cairo"],
   data_files = [
     ('include/pycairo', ['src/py3cairo.h']),
     ('lib/pkgconfig', [pkgconfig_file]),
-    (os.path.join(dsy.get_python_lib(), 'cairo'),
-     ['src/__init__.py']),
     ],
   )

