diff -bur enter-tex-3.48.0-orig/src/gtex/meson.build enter-tex-3.48.0/src/gtex/meson.build
--- enter-tex-3.48.0-orig/src/gtex/meson.build	2025-12-28 08:18:43.073988800 -0700
+++ enter-tex-3.48.0/src/gtex/meson.build	2025-12-28 08:19:52.466809000 -0700
@@ -132,7 +132,7 @@
 # GObject Introspection
 
 GNOME.generate_gir(
-  GTEX_STATIC_LIB,
+  gtex_shared_lib,
   header: 'gtex.h',
   identifier_prefix: 'Gtex',
   includes: ['Gio-2.0', 'Gtk-3.0', 'Gspell-1', 'Tepl-6'],
