diff -urN precice-2.3.0/src/com/ConnectionInfoPublisher.cpp.orig precice-2.3.0/src/com//ConnectionInfoPublisher.cpp
--- precice-2.3.0/src/com/ConnectionInfoPublisher.cpp.orig	2021-10-06 12:55:21.000000000 +0200
+++ precice-2.3.0/src/com//ConnectionInfoPublisher.cpp	2022-06-13 13:10:15.253682600 +0200
@@ -4,7 +4,7 @@
 #include <boost/uuid/string_generator.hpp>
 #include <boost/uuid/uuid_io.hpp>
 #include <chrono>
-#include <istream>
+#include <fstream>
 #include <thread>
 
 #include "ConnectionInfoPublisher.hpp"
