diff -urN reportlab-3.4.0.orig/setup.py reportlab-3.4.0/setup.py
--- reportlab-3.4.0.orig/setup.py	2017-06-21 05:59:58.000000000 -0500
+++ reportlab-3.4.0/setup.py	2017-06-21 06:00:17.000000000 -0500
@@ -404,7 +404,7 @@
                     if len(l)>1 and l[0].strip() in K:
                         D[l[0].strip()] = l[1].strip()
                         if len(D)==3: break
-                return (sys.platform == 'win32' and '\\"%s\\"' or '"%s"') % '.'.join(map(lambda k,D=D: D.get(k,'?'),K))
+                return (sys.platform == 'win32' and '"%s"' or '"%s"') % '.'.join(map(lambda k,D=D: D.get(k,'?'),K))
             LIBART_VERSION = libart_version()
             infoline('will use package libart %s' % LIBART_VERSION.replace('"',''))
 
('"',''))
 
