diff -bur gimp3-2.99.12.r3.g02739dd6e6-c/app/tests/meson.build gimp3-2.99.12.r3.g02739dd6e6/app/tests/meson.build
--- gimp3-2.99.12.r3.g02739dd6e6-c/app/tests/meson.build	2022-08-23 22:34:33.774184400 -0600
+++ gimp3-2.99.12.r3.g02739dd6e6/app/tests/meson.build	2022-08-23 22:35:38.433049400 -0600
@@ -44,10 +44,10 @@
   'xcf',
 ]
 
-cmd = run_command('create_test_env.sh', check: false)
-if cmd.returncode() != 0
- error(cmd.stderr().strip())
-endif
+# cmd = run_command('create_test_env.sh', check: false)
+# if cmd.returncode() != 0
+# error(cmd.stderr().strip())
+# endif
 
 foreach test_name : app_tests
   test_exe = executable(test_name,
