diff -bur gimp-2.99.16-orig/meson.build gimp-2.99.16/meson.build
--- gimp-2.99.16-orig/meson.build	2023-10-12 11:17:37.929301600 -0600
+++ gimp-2.99.16/meson.build	2023-10-12 11:18:09.028568700 -0600
@@ -1135,11 +1135,11 @@
 
 ## Lua
 
-lua = find_program('luajit', required: get_option('lua'))
+lua = find_program('lua5.1', required: get_option('lua'))
 have_lua = get_option('lua').enabled() or (lua.found() and get_option('lua').auto())
 if not lua.found() and have_lua
   lua_warning = '''
-  Luajit was not found.
+  Lua was not found.
   Lua plug-ins will be installed anyway but you should make sure that
   luajit and LGI are available at installation, otherwise
   installed plug-ins won't be usable.
