diff -bur gimp-3.0.0-RC1-orig/meson.build gimp-3.0.0-RC1/meson.build
--- gimp-3.0.0-RC1-orig/meson.build	2024-11-06 16:57:53.191116600 -0700
+++ gimp-3.0.0-RC1/meson.build	2024-11-06 16:58:14.258445000 -0700
@@ -2086,13 +2086,6 @@
 #  install_win_debug_script = find_program('./meson_install_win_debug.sh')
 #  meson.add_install_script(install_win_debug_script)
 #endif
-
-# On Windows, install deps making a bundle in the prefix
-if platform_windows and relocatable_bundle
-  install_win_bundling_script = find_program('build/windows/2_bundle-gimp-uni_base.sh')
-  meson.add_install_script(install_win_bundling_script)
-endif
-
 ################################################################################
 # Subdir installations
 
