--- libspectre-0.2.9/libspectre/Makefile.am.orig	2020-10-27 19:07:47.468300400 +0300
+++ libspectre-0.2.9/libspectre/Makefile.am	2020-10-27 19:09:57.381452100 +0300
@@ -42,7 +42,7 @@
 	$(gv_sources)
 
 libspectre_la_CPPFLAGS = 		\
-	$(SPECTRE_CFLAGS)
+	$(SPECTRE_CFLAGS) -I$(top_srcdir)
 
 libspectre_la_CFLAGS = 			\
 	$(VISIBILITY_CFLAGS)
--- libspectre-0.2.9/test/Makefile.am.orig	2020-10-27 19:28:35.820526000 +0300
+++ libspectre-0.2.9/test/Makefile.am	2020-10-27 19:29:01.918369800 +0300
@@ -6,6 +6,7 @@
 	$(top_srcdir)/libspectre/spectre-utils.h
 
 spectre_test_CPPFLAGS =			\
+	-I$(top_srcdir)	\
 	-I$(top_srcdir)/libspectre	\
 	$(SPECTRE_CFLAGS)		\
 	$(CAIRO_CFLAGS)
@@ -22,6 +23,7 @@
 	$(top_srcdir)/libspectre/spectre-utils.h
 
 parser_test_CPPFLAGS = 			\
+	-I$(top_srcdir)	\
 	-I$(top_srcdir)/libspectre	\
 	$(SPECTRE_CFLAGS)
 
@@ -36,6 +38,7 @@
 	$(top_srcdir)/libspectre/spectre-utils.h
 	
 fuzz_test_CPPFLAGS = \
+	-I$(top_srcdir)	\
 	-I$(top_srcdir)/libspectre	\
 	$(SPECTRE_CFLAGS)
 	
