--- libspectre-0.2.9/configure.ac.orig	2020-10-27 18:52:36.139323500 +0300
+++ libspectre-0.2.9/configure.ac	2020-10-27 18:52:51.682308100 +0300
@@ -299,7 +299,7 @@
 case "$host" in
   *-*-mingw*)
     dnl on mingw32 we do -fvisibility=hidden and __declspec(dllexport)
-    AC_DEFINE([SPECTRE_PUBLIC], [__attribute__((visibility("default"))) __declspec(dllexport) extern],
+    AC_DEFINE([SPECTRE_PUBLIC], [__declspec(dllexport)],
               [defines how to decorate public symbols while building])
     CFLAGS="${CFLAGS} -fvisibility=hidden"
     ;;
