From 83f9ea2c0ae8dbe4faa2e6c1f427add3602aa9e7 Mon Sep 17 00:00:00 2001
From: Liu Hao <lh_mouse@126.com>
Date: Sun, 10 Jun 2018 22:50:37 +0800
Subject: [PATCH] Revert changes in 5045b3d on doc/gdb.texinfo

Signed-off-by: Liu Hao <lh_mouse@126.com>
---
 gdb/doc/gdb.texinfo | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/gdb/doc/gdb.texinfo b/gdb/doc/gdb.texinfo
index 2c0ac33f8b..4968b374af 100644
--- a/gdb/doc/gdb.texinfo
+++ b/gdb/doc/gdb.texinfo
@@ -35547,15 +35547,6 @@ modify XML target descriptions.
 Check that the target descriptions dynamically created by @value{GDBN}
 equal the descriptions created from XML files found in @var{dir}.
 
-@kindex maint check libthread-db
-@item maint check libthread-db
-Run integrity checks on the current inferior's thread debugging
-library.  This exercises all @code{libthread_db} functionality used by
-@value{GDBN} on GNU/Linux systems, and by extension also exercises the
-@code{proc_service} functions provided by @value{GDBN} that
-@code{libthread_db} uses.  Note that parts of the test may be skipped
-on some platforms when debugging core files.
-
 @kindex maint print dummy-frames
 @item maint print dummy-frames
 Prints the contents of @value{GDBN}'s internal dummy-frame stack.
@@ -35863,17 +35854,6 @@ number of blocks in the blockvector
 @end enumerate
 @end table
 
-@kindex maint set check-libthread-db
-@kindex maint show check-libthread-db
-@item maint set check-libthread-db [on|off]
-@itemx maint show check-libthread-db
-Control whether @value{GDBN} should run integrity checks on inferior
-specific thread debugging libraries as they are loaded.  The default
-is not to perform such checks.  If any check fails @value{GDBN} will
-unload the library and continue searching for a suitable candidate as
-described in @ref{set libthread-db-search-path}.  For more information
-about the tests, see @ref{maint check libthread-db}.
-
 @kindex maint space
 @cindex memory used by commands
 @item maint space @var{value}
-- 
2.17.1

