--- geoclue-2.2.0/configure.ac.orig	2015-07-10 12:50:52.864282200 +0300
+++ geoclue-2.2.0/configure.ac	2015-07-10 12:51:12.097358700 +0300
@@ -53,7 +53,7 @@
 PKG_CHECK_MODULES(GEOCLUE, [
   glib-2.0 >= $GLIB_MIN_VERSION
   gio-2.0 >= $GLIB_MIN_VERSION
-  gio-unix-2.0 >= $GLIB_MIN_VERSION
+  gio-windows-2.0 >= $GLIB_MIN_VERSION
   json-glib-1.0 >= $JSON_GLIB_MIN_VERSION
   libsoup-2.4 >= $LIBSOUP_MIN_VERSION
 ])
@@ -115,7 +115,7 @@
     PKG_CHECK_MODULES(DEMO_AGENT, [
                       glib-2.0 >= $GLIB_MIN_VERSION
                       gio-2.0 >= $GLIB_MIN_VERSION
-                      gio-unix-2.0 >= $GLIB_MIN_VERSION
+                      gio-windows-2.0 >= $GLIB_MIN_VERSION
                       libnotify])
 else
     AC_DEFINE([BUILD_DEMO_AGENT], [0], [Build demo agent?])
