--- a/setup.py
+++ b/setup.py
@@ -316,7 +316,7 @@ def make_release_tree(self, base_dir, files):
     setup_requires=[
         "certifi>=2020.06.20",
         "numpy>=1.17",
-        "setuptools_scm>=4,<7",
+        "setuptools_scm>=4",
         "setuptools_scm_git_archive",
     ],
     install_requires=[
@@ -330,7 +330,7 @@ def make_release_tree(self, base_dir, files):
         "python-dateutil>=2.7",
     ] + (
         # Installing from a git checkout that is not producing a wheel.
-        ["setuptools_scm>=4,<7"] if (
+        ["setuptools_scm>=4"] if (
             Path(__file__).with_name(".git").exists() and
             os.environ.get("CIBUILDWHEEL", "0") != "1"
         ) else []