diff -u -r ruby-2.3.0-orig/tool/rbinstall.rb ruby-2.3.0/tool/rbinstall.rb
--- ruby-2.3.0-orig/tool/rbinstall.rb	2016-02-15 09:00:50.829489800 -0800
+++ ruby-2.3.0/tool/rbinstall.rb	2016-02-15 09:02:10.899131700 -0800
@@ -282,7 +282,6 @@
 
 def without_destdir(dir)
   return dir if !$destdir or $destdir.empty? or !dir.start_with?($destdir)
-  dir = dir.sub(/\A\w:/, '') if File::PATH_SEPARATOR == ';'
   dir[$destdir.size..-1]
 end
 
