diff -Naur jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_malloc.h jasper-version-2.0.10/src/libjasper/include/jasper/jas_malloc.h
--- jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_malloc.h	2016-12-30 19:40:24.000000000 +0300
+++ jasper-version-2.0.10/src/libjasper/include/jasper/jas_malloc.h	2017-02-22 19:18:12.162859500 +0300
@@ -89,25 +89,25 @@
 \******************************************************************************/
 
 /* Allocate memory. */
-void *jas_malloc(size_t size);
+JAS_DLLEXPORT void *jas_malloc(size_t size);
 
 /* Free memory. */
-void jas_free(void *ptr);
+JAS_DLLEXPORT void jas_free(void *ptr);
 
 /* Resize a block of allocated memory. */
-void *jas_realloc(void *ptr, size_t size);
+JAS_DLLEXPORT void *jas_realloc(void *ptr, size_t size);
 
 /* Allocate a block of memory and initialize the contents to zero. */
-void *jas_calloc(size_t num_elements, size_t element_size);
+JAS_DLLEXPORT void *jas_calloc(size_t num_elements, size_t element_size);
 
 /* Allocate array (with overflow checking) . */
-void *jas_alloc2(size_t num_elements, size_t element_size);
+JAS_DLLEXPORT void *jas_alloc2(size_t num_elements, size_t element_size);
 
 /* Allocate array of arrays (with overflow checking) . */
-void *jas_alloc3(size_t num_arrays, size_t array_size, size_t element_size);
+JAS_DLLEXPORT void *jas_alloc3(size_t num_arrays, size_t array_size, size_t element_size);
 
 /* Resize a block of allocated memory (with overflow checking) . */
-void *jas_realloc2(void *ptr, size_t num_elements, size_t element_size);
+JAS_DLLEXPORT void *jas_realloc2(void *ptr, size_t num_elements, size_t element_size);
 
 #if defined(JAS_DEFAULT_MAX_MEM_USAGE)
 
