diff -Naur jasper-1.900.1/configure.ac jasper-1.900.1.new/configure.ac
--- jasper-1.900.1/configure.ac	2007-01-19 21:47:11.000000000 +0000
+++ jasper-1.900.1.new/configure.ac	2009-10-29 13:37:02.000000000 +0000
@@ -399,6 +399,8 @@
 src/libjasper/pnm/Makefile
 src/libjasper/ras/Makefile
 src/msvc/Makefile
+pkgconfig/Makefile
+pkgconfig/jasper.pc
 jasper.spec
 ])
 AC_OUTPUT
diff -Naur jasper-1.900.1/Makefile.am jasper-1.900.1.new/Makefile.am
--- jasper-1.900.1/Makefile.am	2007-01-19 21:43:14.000000000 +0000
+++ jasper-1.900.1.new/Makefile.am	2009-10-29 13:36:28.000000000 +0000
@@ -64,7 +64,7 @@
 # Note: We need to put the derived file "jasper.spec" in the distribution
 # in order to facilitate RPM building.
 
-SUBDIRS = src
+SUBDIRS = src pkgconfig
 
 rpm: dist
 	for i in BUILD RPMS SRPMS SOURCES SPECS; do \
diff -Naur jasper-1.900.1/pkgconfig/jasper.pc.in jasper-1.900.1.new/pkgconfig/jasper.pc.in
--- jasper-1.900.1/pkgconfig/jasper.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ jasper-1.900.1.new/pkgconfig/jasper.pc.in	2009-10-29 16:30:43.000000000 +0000
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: Jasper
+Description: JPEG 2000 encoding and decoding library
+Version: @VERSION@
+
+Libs: -L${libdir} -ljasper
+Cflags: -I${includedir}/jasper
diff -Naur jasper-1.900.1/pkgconfig/Makefile.am jasper-1.900.1.new/pkgconfig/Makefile.am
--- jasper-1.900.1/pkgconfig/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ jasper-1.900.1.new/pkgconfig/Makefile.am	2009-10-29 13:38:59.000000000 +0000
@@ -0,0 +1,5 @@
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = jasper.pc
+
+EXTRA_DIST = \
+        jasper.pc.in
