--- goom2k4-0/src/mmx.h.orig	2021-08-25 20:52:28.962510200 -0700
+++ goom2k4-0/src/mmx.h	2021-08-25 20:54:36.790526200 -0700
@@ -715,13 +715,13 @@
 	{ \
 		printf("emms()\n"); \
 		__asm__ __volatile__ ("emms" \
-                        "st(1)","st(2)","st(3)","st(4)","st(5)","st(6)","st(7)"); \
+                        "st", "st(1)","st(2)","st(3)","st(4)","st(5)","st(6)","st(7)"); \
 	}
 
 #else
 
 #define	emms() __asm__ __volatile__ ("emms"::: \
-                      "st(1)","st(2)","st(3)","st(4)","st(5)","st(6)","st(7)")
+                      "st", "st(1)","st(2)","st(3)","st(4)","st(5)","st(6)","st(7)")
 
 #endif
 
