diff --git a/src/Makefile.am b/src/Makefile.am
index de4fc65..7b67684 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -16,7 +16,7 @@ goom2_lib_LTLIBRARIES = libgoom2.la
 goom2_libdir = $(libdir)
 
 goom2_library_includedir=$(includedir)/goom
-goom2_library_include_HEADERS = goom.h goom_plugin_info.h goom_typedefs.h goom_graphic.h goom_config_param.h goom_visual_fx.h goom_filters.h goom_tools.h goomsl.h goomsl_hash.h goomsl_heap.h goom_tools.h goom_config.h
+goom2_library_include_HEADERS = goom.h goom_plugin_info.h goom_typedefs.h goom_graphic.h goom_config_param.h goom_visual_fx.h goom_filters.h goom_tools.h goomsl.h goomsl_hash.h goomsl_heap.h goom_config.h
 libgoom2_la_LDFLAGS = -export-dynamic -export-symbols-regex "goom.*" -no-undefined
 libgoom2_la_SOURCES = \
 	goomsl_yacc.y goomsl_lex.l goomsl.c goomsl_hash.c goomsl_heap.c \
