--- qhull-2020.2/CMakeLists.txt.orig	2020-10-27 11:43:50.575775500 +0300
+++ qhull-2020.2/CMakeLists.txt	2020-10-27 11:44:03.758022100 +0300
@@ -485,7 +485,7 @@
 # ---------------------------------------
 # Define options for linking to shared libaries
 # ---------------------------------------
-if(MSVC) 
+if(WIN32) 
     set(qconvex_DEFINES  qh_dllimport)
     set(qdelaunay_DEFINES qh_dllimport)
     set(qhalf_DEFINES    qh_dllimport)
