diff -Naur qtwebkit-5.212.0-alpha4.a/Source/cmake/OptionsQt.cmake qtwebkit-5.212.0-alpha4.b/Source/cmake/OptionsQt.cmake
--- qtwebkit-5.212.0-alpha4.a/Source/cmake/OptionsQt.cmake	2021-08-18 16:23:36.216770900 +0200
+++ qtwebkit-5.212.0-alpha4.b/Source/cmake/OptionsQt.cmake	2021-08-18 21:20:32.642677100 +0200
@@ -717,7 +717,7 @@
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-expansion-to-defined")
 endif ()
 
-if (WIN32 AND COMPILER_IS_GCC_OR_CLANG)
+if (WIN32 AND NOT COMPILER_IS_CLANG)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-keep-inline-dllexport")
 endif ()
 
diff -Naur qtwebkit-5.212.0-alpha4.a/Source/ThirdParty/gtest/include/gtest/internal/gtest-port.h qtwebkit-5.212.0-alpha4.b/Source/ThirdParty/gtest/include/gtest/internal/gtest-port.h
--- qtwebkit-5.212.0-alpha4.a/Source/ThirdParty/gtest/include/gtest/internal/gtest-port.h	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4.b/Source/ThirdParty/gtest/include/gtest/internal/gtest-port.h	2021-08-18 21:26:38.374692400 +0200
@@ -932,6 +932,10 @@
   return NULL;
 }
 
+// gtest-port.h guarantees to #include <pthread.h> when GTEST_HAS_PTHREAD is
+// true.
+#include <pthread.h>
+
 // Helper class for testing Google Test's multi-threading constructs.
 // To use it, write:
 //
@@ -988,10 +992,6 @@
   GTEST_DISALLOW_COPY_AND_ASSIGN_(ThreadWithParam);
 };
 
-// gtest-port.h guarantees to #include <pthread.h> when GTEST_HAS_PTHREAD is
-// true.
-#include <pthread.h>
-
 // MutexBase and Mutex implement mutex on pthreads-based platforms. They
 // are used in conjunction with class MutexLock:
 //
diff -Naur qtwebkit-5.212.0-alpha4.a/Source/WebKit/win/Plugins/PluginViewWin.cpp qtwebkit-5.212.0-alpha4.b/Source/WebKit/win/Plugins/PluginViewWin.cpp
--- qtwebkit-5.212.0-alpha4.a/Source/WebKit/win/Plugins/PluginViewWin.cpp	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4.b/Source/WebKit/win/Plugins/PluginViewWin.cpp	2021-08-18 21:20:32.687353200 +0200
@@ -153,7 +153,7 @@
         return pluginView->m_wmPrintHDC;
     }
 
-#if COMPILER(GCC)
+#if COMPILER(GCC_OR_CLANG)
     HDC result;
     asm ("push    %2\n"
          "push    %3\n"
@@ -183,7 +183,7 @@
         return TRUE;
     }
 
-#if COMPILER(GCC)
+#if COMPILER(GCC_OR_CLANG)
     BOOL result;
     asm ("push   %2\n"
          "push   %3\n"
