--- a/Source/JavaScriptCore/bytecode/StructureSet.h
+++ b/Source/JavaScriptCore/bytecode/StructureSet.h
@@ -57,11 +57,6 @@
     {
     }
     
-    ALWAYS_INLINE StructureSet(const StructureSet& other)
-        : TinyPtrSet(other)
-    {
-    }
-    
     Structure* onlyStructure() const
     {
         return onlyEntry();
--- a/Source/JavaScriptCore/debugger/Breakpoint.h
+++ b/Source/JavaScriptCore/debugger/Breakpoint.h
@@ -48,18 +48,6 @@
     {
     }
 
-    Breakpoint(const Breakpoint& other)
-        : id(other.id)
-        , sourceID(other.sourceID)
-        , line(other.line)
-        , column(other.column)
-        , condition(other.condition)
-        , autoContinue(other.autoContinue)
-        , ignoreCount(other.ignoreCount)
-        , hitCount(other.hitCount)
-    {
-    }
-
     BreakpointID id { noBreakpointID };
     SourceID sourceID { noSourceID };
     unsigned line { 0 };
--- a/Source/JavaScriptCore/parser/ParserTokens.h
+++ b/Source/JavaScriptCore/parser/ParserTokens.h
@@ -169,7 +169,6 @@
 struct JSTextPosition {
     JSTextPosition() : line(0), offset(0), lineStartOffset(0) { }
     JSTextPosition(int _line, int _offset, int _lineStartOffset) : line(_line), offset(_offset), lineStartOffset(_lineStartOffset) { }
-    JSTextPosition(const JSTextPosition& other) : line(other.line), offset(other.offset), lineStartOffset(other.lineStartOffset) { }
 
     JSTextPosition operator+(int adjustment) const { return JSTextPosition(line, offset + adjustment, lineStartOffset); }
     JSTextPosition operator+(unsigned adjustment) const { return *this + static_cast<int>(adjustment); }
@@ -200,13 +199,6 @@
 
 struct JSTokenLocation {
     JSTokenLocation() : line(0), lineStartOffset(0), startOffset(0) { }
-    JSTokenLocation(const JSTokenLocation& location)
-    {
-        line = location.line;
-        lineStartOffset = location.lineStartOffset;
-        startOffset = location.startOffset;
-        endOffset = location.endOffset;
-    }
 
     int line;
     unsigned lineStartOffset;
--- a/Source/WebCore/dom/RangeBoundaryPoint.h
+++ b/Source/WebCore/dom/RangeBoundaryPoint.h
@@ -35,8 +35,6 @@
 public:
     explicit RangeBoundaryPoint(Node* container);
 
-    explicit RangeBoundaryPoint(const RangeBoundaryPoint&);
-
     const Position toPosition() const;
 
     Node* container() const;
@@ -71,13 +69,6 @@
     ASSERT(m_containerNode);
 }
 
-inline RangeBoundaryPoint::RangeBoundaryPoint(const RangeBoundaryPoint& other)
-    : m_containerNode(other.container())
-    , m_offsetInContainer(other.offset())
-    , m_childBeforeBoundary(other.childBefore())
-{
-}
-
 inline Node* RangeBoundaryPoint::container() const
 {
     return m_containerNode.get();
--- a/Source/WebCore/Modules/indexeddb/shared/IDBError.cpp
+++ b/Source/WebCore/Modules/indexeddb/shared/IDBError.cpp
@@ -48,13 +48,6 @@
     return { m_code, m_message.isolatedCopy() };
 }
 
-IDBError& IDBError::operator=(const IDBError& other)
-{
-    m_code = other.m_code;
-    m_message = other.m_message;
-    return *this;
-}
-
 String IDBError::name() const
 {
     return IDBDatabaseException::getErrorName(m_code);
--- a/Source/WebCore/Modules/indexeddb/shared/IDBError.h
+++ b/Source/WebCore/Modules/indexeddb/shared/IDBError.h
@@ -39,8 +39,6 @@
     IDBError(ExceptionCode);
     IDBError(ExceptionCode, const String& message);
 
-    IDBError& operator=(const IDBError&);
-
     ExceptionCode code() const { return m_code; }
     String name() const;
     String message() const;
--- a/Source/WebCore/platform/graphics/FloatPoint3D.h
+++ b/Source/WebCore/platform/graphics/FloatPoint3D.h
@@ -50,13 +50,6 @@
     {
     }
 
-    FloatPoint3D(const FloatPoint3D& p)
-        : m_x(p.x())
-        , m_y(p.y())
-        , m_z(p.z())
-    {
-    }
-
     float x() const { return m_x; }
     void setX(float x) { m_x = x; }
 
--- a/Source/WebCore/platform/graphics/GraphicsContext.cpp
+++ b/Source/WebCore/platform/graphics/GraphicsContext.cpp
@@ -53,12 +53,6 @@
     {
     }
 
-    TextRunIterator(const TextRunIterator& other)
-        : m_textRun(other.m_textRun)
-        , m_offset(other.m_offset)
-    {
-    }
-
     unsigned offset() const { return m_offset; }
     void increment() { m_offset++; }
     bool atEnd() const { return !m_textRun || m_offset >= m_textRun->length(); }
--- a/Source/WebCore/platform/graphics/texmap/TextureMapperAnimation.cpp
+++ b/Source/WebCore/platform/graphics/texmap/TextureMapperAnimation.cpp
@@ -218,20 +218,6 @@
 {
 }
 
-TextureMapperAnimation::TextureMapperAnimation(const TextureMapperAnimation& other)
-    : m_name(other.m_name.isSafeToSendToAnotherThread() ? other.m_name : other.m_name.isolatedCopy())
-    , m_keyframes(other.m_keyframes)
-    , m_boxSize(other.m_boxSize)
-    , m_animation(Animation::create(*other.m_animation))
-    , m_listsMatch(other.m_listsMatch)
-    , m_startTime(other.m_startTime)
-    , m_pauseTime(other.m_pauseTime)
-    , m_totalRunningTime(other.m_totalRunningTime)
-    , m_lastRefreshedTime(other.m_lastRefreshedTime)
-    , m_state(other.m_state)
-{
-}
-
 void TextureMapperAnimation::apply(Client& client)
 {
     if (!isActive())
--- a/Source/WebCore/platform/graphics/texmap/TextureMapperAnimation.h
+++ b/Source/WebCore/platform/graphics/texmap/TextureMapperAnimation.h
@@ -41,7 +41,6 @@
         : m_keyframes(AnimatedPropertyInvalid)
     { }
     TextureMapperAnimation(const String&, const KeyframeValueList&, const FloatSize&, const Animation&, bool, double, double, AnimationState);
-    TextureMapperAnimation(const TextureMapperAnimation&);
 
     void apply(Client&);
     void pause(double);
--- a/Source/WebCore/rendering/mathml/RenderMathMLOperator.h
+++ b/Source/WebCore/rendering/mathml/RenderMathMLOperator.h
@@ -106,20 +106,6 @@
         }
         StretchyData()
             : m_mode(DrawNormal) { }
-        StretchyData(const StretchyData& data)
-        {
-            switch (data.m_mode) {
-            case DrawNormal:
-                setNormalMode();
-                break;
-            case DrawSizeVariant:
-                setSizeVariantMode(data.variant());
-                break;
-            case DrawGlyphAssembly:
-                setGlyphAssemblyMode(data.top(), data.extension(), data.bottom(), data.middle());
-                break;
-            }
-        }
 
     private:
         DrawMode m_mode;
--- a/Source/WebCore/rendering/style/BasicShapes.h
+++ b/Source/WebCore/rendering/style/BasicShapes.h
@@ -90,13 +90,6 @@
         updateComputedLength();
     }
 
-    BasicShapeCenterCoordinate(const BasicShapeCenterCoordinate& other)
-        : m_direction(other.direction())
-        , m_length(other.length())
-        , m_computedLength(other.m_computedLength)
-    {
-    }
-
     Direction direction() const { return m_direction; }
     const Length& length() const { return m_length; }
     const Length& computedLength() const { return m_computedLength; }
@@ -141,10 +134,6 @@
         : m_value(Undefined)
         , m_type(t)
     { }
-    BasicShapeRadius(const BasicShapeRadius& other)
-        : m_value(other.value())
-        , m_type(other.type())
-    { }
 
     const Length& value() const { return m_value; }
     Type type() const { return m_type; }
--- a/Source/WebCore/svg/SVGLength.cpp
+++ b/Source/WebCore/svg/SVGLength.cpp
@@ -133,12 +133,6 @@
     setValue(value, context, ASSERT_NO_EXCEPTION);
 }
 
-SVGLength::SVGLength(const SVGLength& other)
-    : m_valueInSpecifiedUnits(other.m_valueInSpecifiedUnits)
-    , m_unit(other.m_unit)
-{
-}
-
 void SVGLength::setValueAsString(const String& valueAsString, SVGLengthMode mode, ExceptionCode& ec)
 {
     m_valueInSpecifiedUnits = 0;
--- a/Source/WebCore/svg/SVGLength.h
+++ b/Source/WebCore/svg/SVGLength.h
@@ -59,7 +59,6 @@
     // FIXME: Once all SVGLength users use Length internally, we make this a wrapper for Length.
     SVGLength(SVGLengthMode = LengthModeOther, const String& valueAsString = String());
     SVGLength(const SVGLengthContext&, float, SVGLengthMode = LengthModeOther, SVGLengthType = LengthTypeNumber);
-    SVGLength(const SVGLength&);
 
     SVGLengthType unitType() const;
     SVGLengthMode unitMode() const;
--- a/Source/WTF/wtf/CheckedArithmetic.h
+++ b/Source/WTF/wtf/CheckedArithmetic.h
@@ -493,15 +493,6 @@
         m_value = static_cast<T>(rhs.m_value);
     }
     
-    const Checked& operator=(Checked rhs)
-    {
-        this->clearOverflow();
-        if (rhs.hasOverflowed())
-            this->overflowed();
-        m_value = static_cast<T>(rhs.m_value);
-        return *this;
-    }
-    
     template <typename U> const Checked& operator=(U value)
     {
         return *this = Checked(value);
--- a/Source/WTF/wtf/PageBlock.h
+++ b/Source/WTF/wtf/PageBlock.h
@@ -37,7 +37,6 @@
 class PageBlock {
 public:
     PageBlock();
-    PageBlock(const PageBlock&);
     PageBlock(void*, size_t, bool hasGuardPages);
     
     void* base() const { return m_base; }
@@ -64,13 +63,6 @@
 {
 }
 
-inline PageBlock::PageBlock(const PageBlock& other)
-    : m_realBase(other.m_realBase)
-    , m_base(other.m_base)
-    , m_size(other.m_size)
-{
-}
-
 inline PageBlock::PageBlock(void* base, size_t size, bool hasGuardPages)
     : m_realBase(base)
     , m_base(static_cast<char*>(base) + ((base && hasGuardPages) ? pageSize() : 0))
