--- ogitor/OgreOfsPlugin/CMakeLists.txt.orig	2018-07-24 13:16:44.892880100 +0300
+++ ogitor/OgreOfsPlugin/CMakeLists.txt	2018-07-24 13:16:48.324886100 +0300
@@ -19,6 +19,10 @@
 
 set_target_properties(OgreOfsPlugin PROPERTIES COMPILE_DEFINITIONS "PLUGIN_EXPORT")
 
+if(MINGW)
+    set_target_properties(OgreOfsPlugin PROPERTIES PREFIX "")
+endif()
+
 target_link_libraries(OgreOfsPlugin ${OGRE_LIBRARIES} OFS ${Boost_REGEX_LIBRARY})
 
 install(TARGETS OgreOfsPlugin
