From ba0444146c35848917101b28f3463cd0df08a02d Mon Sep 17 00:00:00 2001
From: David Macek <david.macek.0@gmail.com>
Date: Tue, 10 Mar 2015 00:22:35 +0100
Subject: [PATCH] Do not invoke cmd in makefile

---
 makefile.gcc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/makefile.gcc b/makefile.gcc
index 7000335..281cbe7 100644
--- a/makefile.gcc
+++ b/makefile.gcc
@@ -90,7 +90,7 @@ ansicon32: x86 x86/ansicon.exe x86/ANSI32.dll x64 x64/ANSI32.dll
 ansicon64: x64 x64/ansicon.exe x64/ANSI64.dll
 
 x86:
-	cmd /c "mkdir x86"
+	mkdir x86
 
 x86/ansicon.exe: x86/ansicon.o $(X86OBJS) x86/procrva.o x86/ansiconv.o
 	$(LDmsg)$(CC) -m32 $+ -s -o $@ $(IVER)
@@ -99,7 +99,7 @@ x86/ANSI32.dll: x86/ANSI.o $(X86OBJS) x86/ansiv.o
 	$(LDmsg)$(CC) -m32 $+ -s -o $@ -mdll -Wl,-shared,--image-base,0xAC0000
 
 x64:
-	cmd /c "mkdir x64"
+	mkdir x64
 
 x64/ansicon.exe: x64/ansicon.o $(X64OBJS) x64/ansiconv.o
 	$(LDmsg)$(CC) -m64 $+ -s -o $@ $(IVER)
@@ -121,5 +121,5 @@ x64/util.o:	version.h
 # Need two commands, because if the directory doesn't exist, it won't delete
 # anything at all.
 clean:
-	-cmd /c "del x86\*.o 2>nul"
-	-cmd /c "del x64\*.o 2>nul"
+	rm -f x86\*.o
+	rm -f x64\*.o
-- 
2.3.1

