--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.11)
-project (Lerc)
+project (Lerc VERSION 3.0.0)
 
 include(GNUInstallDirs)
 
@@ -32,3 +32,9 @@
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
+
+if(NOT MSVC AND NOT APPLE)
+    configure_file(Lerc.pc.in ${CMAKE_CURRENT_BINARY_DIR}/Lerc.pc @ONLY)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/Lerc.pc
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+endif()
