diff --git a/src/pcre2test.c b/src/pcre2test.c
--- a/src/pcre2test.c
+++ b/src/pcre2test.c
@@ -2026,17 +2026,21 @@ for (;;)
     char promptbuf[80];
     int snprintf_rc;
     char *s;
+#if defined(RL_PROMPT_START_IGNORE) && defined(RL_PROMPT_END_IGNORE)
     if (should_print_colour(clr_prompt, stdout))
       /* libedit cannot handle ANSI escapes as the final character of a prompt,
       so we ensure a trailing space comes after the last escape. */
       snprintf_rc = snprintf(promptbuf, sizeof(promptbuf),
                              "%c\x1b[%dm%c%s%c\x1b[0m%c ",
                              RL_PROMPT_START_IGNORE, clr_prompt,
                              RL_PROMPT_END_IGNORE, prompt,
                              RL_PROMPT_START_IGNORE,
                              RL_PROMPT_END_IGNORE);
     else
       snprintf_rc = snprintf(promptbuf, sizeof(promptbuf), "%s ", prompt);
+#else
+    snprintf_rc = snprintf(promptbuf, sizeof(promptbuf), "%s ", prompt);
+#endif
     if (snprintf_rc <= 0 || snprintf_rc >= (int)sizeof(promptbuf))
       {
       cfprintf(clr_test_error, outfile, "** Interactive prompt exceeds buffer space\n");
