diff --git a/external/fmt/include/fmt/format.h b/external/fmt/include/fmt/format.h
index 50e5714..0256393 100644
--- a/external/fmt/include/fmt/format.h
+++ b/external/fmt/include/fmt/format.h
@@ -45,6 +45,7 @@
 #  include <cstddef>  // std::byte
 #  include <cstdint>  // uint32_t
 #  include <cstring>  // std::memcpy
+#  include <cstdlib>  // std::malloc, std::free
 #  include <limits>   // std::numeric_limits
 #  include <new>      // std::bad_alloc
 #  if defined(__GLIBCXX__) && !defined(_GLIBCXX_USE_DUAL_ABI)
