--- pywin32-219-x86_64/setup.py.orig	2014-10-27 00:17:26.065460800 +0000
+++ pywin32-219-x86_64/setup.py	2014-10-27 00:31:35.736355300 +0000
@@ -97,7 +97,10 @@
     # py24 and earlier
     bdist_msi = None
 
-from distutils.msvccompiler import get_build_version
+if (sys.platform == "win32" and sys.version.find("GCC") >= 0):
+    from distutils.mingwcompiler import get_build_version
+else:
+    from distutils.msvccompiler import get_build_version
 from distutils import log
 
 # some modules need a static CRT to avoid problems caused by them having a
