diff -aur old/setup.py 001/setup.py
--- old/setup.py	2014-05-03 11:37:18.000000000 -0300
+++ 001/setup.py	2015-03-25 20:05:46.116630400 -0300
@@ -224,7 +224,7 @@
 # languishing and will probably never be fixed for Python 2.6...
 if sys.version_info > (2,6):
     from distutils.spawn import spawn
-    from distutils.msvc9compiler import MSVCCompiler
+    from distutils.msvccompiler import MSVCCompiler
     MSVCCompiler._orig_spawn = MSVCCompiler.spawn
     MSVCCompiler._orig_link = MSVCCompiler.link
 
@@ -293,7 +293,7 @@
     MSVCCompiler.link = monkeypatched_link
 
 
-sdk_dir = find_platform_sdk_dir()
+sdk_dir = find_platform_sdk_dir() or ''
 
 class WinExt (Extension):
     # Base class for all win32 extensions, with some predefined
