diff -aur 000/setup.py 001/setup.py
--- 000/setup.py	2015-04-09 21:00:48.725278100 -0300
+++ 001/setup.py	2015-04-09 21:02:02.350399800 -0300
@@ -1500,6 +1504,10 @@
         "win32/src/PySecurityObjects.h",
         ],
     extra_compile_args = ['-DBUILD_PYWINTYPES'],
+    extra_link_args = ['-Wl,--out-implib,build/temp.mingw-%d.%d/libpywintypes.dll.a' % (
+        sys.version_info.major,
+        sys.version_info.minor,
+    )],
     libraries = "advapi32 user32 ole32 oleaut32",
     pch_header = "PyWinTypes.h",
     )
@@ -1608,7 +1606,7 @@
         sources = info[4].split()
     extra_compile_args = []
     ext = WinExt_win32(name,
-                 libraries=lib_names,
+                 libraries=(lib_names + ' pywintypes'),
                  extra_compile_args = extra_compile_args,
                  windows_h_version = windows_h_ver,
                  sources = sources,
@@ -1621,7 +1619,7 @@
             sources = """
                 win32\\src\\win32evtlog_messages.mc win32\\src\\win32evtlog.i
                 """.split(),
-                libraries="advapi32 oleaut32",
+                libraries="advapi32 oleaut32 pywintypes",
                 delay_load_libraries="wevtapi",
                 windows_h_version=0x0600
         ),
@@ -1629,7 +1627,7 @@
            sources = """
                 win32/src/win32apimodule.cpp win32/src/win32api_display.cpp
                 """.split(),
-           libraries="user32 advapi32 shell32 version",
+           libraries="user32 advapi32 shell32 version pywintypes",
            delay_load_libraries="powrprof",
            windows_h_version=0x0500,
         ),
@@ -1637,7 +1637,7 @@
                 win32/src/win32gui.i
                """.split(),
            windows_h_version=0x0500,
-           libraries="gdi32 user32 comdlg32 comctl32 shell32",
+           libraries="gdi32 user32 comdlg32 comctl32 shell32 pywintypes",
            define_macros = [("WIN32GUI", None)],
         ),
     # winxpgui is built from win32gui.i, but sets up different #defines before
@@ -1649,7 +1647,7 @@
                 win32/src/winxpgui.rc win32/src/win32dynamicdialog.cpp
                 win32/src/win32gui.i
                """.split(),
-           libraries="gdi32 user32 comdlg32 comctl32 shell32",
+           libraries="gdi32 user32 comdlg32 comctl32 shell32 pywintypes",
            windows_h_version=0x0500,
            define_macros = [("WIN32GUI",None), ("WINXPGUI",None)],
            extra_swig_commands=["-DWINXPGUI"],
@@ -1657,7 +1655,7 @@
     # winxptheme
     WinExt_win32("_winxptheme",
            sources = ["win32/src/_winxptheme.i"],
-           libraries="gdi32 user32 comdlg32 comctl32 shell32 Uxtheme",
+           libraries="gdi32 user32 comdlg32 comctl32 shell32 Uxtheme pywintypes",
            windows_h_version=0x0500,
         ),
 ]
@@ -1665,7 +1663,7 @@
     WinExt_win32('servicemanager',
            sources = ["win32/src/PythonServiceMessages.mc", "win32/src/PythonService.cpp"],
            extra_compile_args = ['-DPYSERVICE_BUILD_DLL'],
-           libraries = "user32 ole32 advapi32 shell32",
+           libraries = "user32 ole32 advapi32 shell32 pywintypes",
            windows_h_version = 0x500,
            unicode_mode=True,),
 ]
@@ -1764,9 +1762,13 @@
                         %(win32com)s/include\\PyIEnumContextProps.h     %(win32com)s/include\\PyIClientSecurity.h
                         %(win32com)s/include\\PyIServerSecurity.h
                         """ % dirs).split(),
-                   libraries = "oleaut32 ole32 user32 urlmon",
-                   export_symbol_file = 'com/win32com/src/PythonCOM.def',
+                   libraries = "oleaut32 ole32 user32 urlmon uuid pywintypes",
+                   # export_symbol_file = 'com/win32com/src/PythonCOM.def',
                    extra_compile_args = ['-DBUILD_PYTHONCOM'],
+                   extra_link_args = ['-Wl,--out-implib,build/temp.mingw-%d.%d/libpythoncom.dll.a' % (
+                       sys.version_info.major,
+                       sys.version_info.minor,
+                   )],
                    pch_header = "stdafx.h",
                    windows_h_version = 0x500,
                    base_address = dll_base_address,
@@ -1788,7 +1786,7 @@
                         %(adsi)s/PyADSIUtil.cpp         %(adsi)s/PyDSOPObjects.cpp
                         %(adsi)s/PyIADs.cpp
                         """ % dirs).split()),
-    WinExt_win32com('axcontrol', pch_header="axcontrol_pch.h",
+    WinExt_win32com('axcontrol', pch_header="axcontrol_pch.h", libraries = "uuid pywintypes pythoncom",
                     sources=("""
                         %(axcontrol)s/AXControl.cpp
                         %(axcontrol)s/PyIOleControl.cpp          %(axcontrol)s/PyIOleControlSite.cpp
@@ -1800,7 +1800,7 @@
                         %(axcontrol)s/PyIOleObject.cpp           %(axcontrol)s/PyIViewObject2.cpp
                         %(axcontrol)s/PyIOleCommandTarget.cpp
                         """ % dirs).split()),
-    WinExt_win32com('axscript',
+    WinExt_win32com('axscript', libraries = "uuid pywintypes pythoncom",
                     sources=("""
                         %(axscript)s/AXScript.cpp
                         %(axscript)s/GUIDS.cpp                   %(axscript)s/PyGActiveScript.cpp
@@ -1873,7 +1871,7 @@
                     %(axdebug)s/stdafx.cpp
                      """ % dirs).split(),
     ),
-    WinExt_win32com('internet', pch_header="internet_pch.h",
+    WinExt_win32com('internet', pch_header="internet_pch.h", libraries = "uuid pywintypes pythoncom",
                     sources=("""
                         %(internet)s/internet.cpp                   %(internet)s/PyIDocHostUIHandler.cpp
                         %(internet)s/PyIHTMLOMWindowServices.cpp    %(internet)s/PyIInternetBindInfo.cpp
@@ -1930,7 +1928,7 @@
                                   %(mapi)s/mapi_stub_library/MapiStubLibrary.cpp
                                   %(mapi)s/mapi_stub_library/StubUtils.cpp
                                   """ % dirs).split()),
-    WinExt_win32com('shell', libraries='shell32', pch_header="shell_pch.h",
+    WinExt_win32com('shell', libraries='shell32 uuid pywintypes pythoncom', pch_header="shell_pch.h",
                     windows_h_version = 0x600,
                     sources=("""
                         %(shell)s/PyIActiveDesktop.cpp
@@ -2033,7 +2031,7 @@
                     ),


-    WinExt_win32com('taskscheduler', libraries='mstask',
+    WinExt_win32com('taskscheduler', libraries='mstask uuid pywintypes pythoncom',
                     sources=("""
                         %(taskscheduler)s/taskscheduler.cpp
                         %(taskscheduler)s/PyIProvideTaskPage.cpp
@@ -2078,8 +2076,8 @@
                         %(directsound)s/PyIDirectSoundCaptureBuffer.h %(directsound)s/PyIDirectSoundNotify.h
                         """ % dirs).split(),
                     optional_headers = ['dsound.h'],
-                    libraries='user32 dsound dxguid'),
-    WinExt_win32com('authorization', libraries='aclui advapi32',
+                    libraries='user32 dsound dxguid uuid pywintypes pythoncom'),
+    WinExt_win32com('authorization', libraries='aclui advapi32 pywintypes pythoncom',
                     sources=("""
                         %(authorization)s/authorization.cpp
                         %(authorization)s/PyGSecurityInformation.cpp
@@ -2273,7 +2273,7 @@
          sources=[os.path.join("win32", "src", s) for s in
                   "PythonService.cpp PythonService.rc".split()],
          unicode_mode = True,
-         libraries = "user32 advapi32 ole32 shell32"),
+         libraries = "user32 advapi32 ole32 shell32 pywintypes"),
     WinExt_pythonwin_subsys_win("Pythonwin",
         sources = [
             "Pythonwin/pythonwin.cpp",
