diff -Naur pywin32-b301-orig/isapi/src/PythonEng.h pywin32-b301/isapi/src/PythonEng.h
--- pywin32-b301-orig/isapi/src/PythonEng.h	2021-05-30 16:31:38 +0800
+++ pywin32-b301/isapi/src/PythonEng.h	2021-10-05 19:07:38 +0800
@@ -63,7 +63,7 @@
     PyObject *DoCallback(HANDLER_TYPE typ, PyObject *args);
 
     bool LoadHandler(bool reload);
-    bool CPythonHandler::CheckCallback(const char *cbname, PyObject **cb);
+    bool CheckCallback(const char *cbname, PyObject **cb);
     const char *m_namefactory;
     const char *m_nameinit;
     const char *m_namedo;
diff -Naur pywin32-b301-orig/isapi/src/StdAfx.h pywin32-b301/isapi/src/StdAfx.h
--- pywin32-b301-orig/isapi/src/StdAfx.h	2021-05-30 16:31:38 +0800
+++ pywin32-b301/isapi/src/StdAfx.h	2021-10-05 19:08:26 +0800
@@ -51,6 +51,10 @@
 typedef int Py_ssize_t;
 #endif
 
+#ifndef _MSC_VER
+#define HSE_REQ_SET_FLUSH_FLAG (HSE_REQ_END_RESERVED+43)
+#endif
+
 // ***** py3k support *****
 // Note that when built for py3k, 'UNICODE' is defined, which conveniently
 // means TCHAR is the same size as the native unicode object in all versions.
