diff -urN pywin32-b306-orig/isapi/src/PyExtensionObjects.cpp pywin32-b306/isapi/src/PyExtensionObjects.cpp
--- pywin32-b306-orig/isapi/src/PyExtensionObjects.cpp	2023-03-26 07:53:26.000000000 +0800
+++ pywin32-b306/isapi/src/PyExtensionObjects.cpp	2023-04-07 09:31:10.813179500 +0800
@@ -24,8 +24,8 @@
  */
 
 //#define PY_SSIZE_T_CLEAN  // defined by isapi\src\StdAfx.h
-#include "stdafx.h"
-#include "pywintypes.h"
+#include "StdAfx.h"
+#include "PyWinTypes.h"
 #include "Utils.h"
 #include "PyExtensionObjects.h"
 #include "PythonEng.h"
@@ -806,7 +806,7 @@
 
     BOOL bRes;
     Py_BEGIN_ALLOW_THREADS bRes =
-        ecb->ServerSupportFunction(ecb->ConnID, HSE_REQ_IO_COMPLETION, DoIOCallback, NULL, NULL);
+        ecb->ServerSupportFunction(ecb->ConnID, HSE_REQ_IO_COMPLETION, (LPVOID)DoIOCallback, NULL, NULL);
     Py_END_ALLOW_THREADS if (!bRes) return SetPyECBError("ServerSupportFunction(HSE_REQ_IO_COMPLETION)");
     Py_RETURN_NONE;
 }
diff -urN pywin32-b306-orig/isapi/src/PyFilterObjects.cpp pywin32-b306/isapi/src/PyFilterObjects.cpp
--- pywin32-b306-orig/isapi/src/PyFilterObjects.cpp	2023-03-26 07:53:26.000000000 +0800
+++ pywin32-b306/isapi/src/PyFilterObjects.cpp	2023-04-07 09:31:10.839162200 +0800
@@ -307,7 +307,7 @@
         // The Java code passes "\r\n" as first DWORD, and header in second,
         // but docs clearly have second as unused.  Either way, I can't see the
         // specific header!
-        bRes = phfc->m_pfc->ServerSupportFunction(SF_REQ_SEND_RESPONSE_HEADER, status, (DWORD)header, 0);
+        bRes = phfc->m_pfc->ServerSupportFunction(SF_REQ_SEND_RESPONSE_HEADER, status, (DWORD_PTR)header, 0);
     Py_END_ALLOW_THREADS if (!bRes) return SetPyHFCError("SendResponseHeader");
     Py_INCREF(Py_None);
     return Py_None;
