diff --git a/isapi/src/PyExtensionObjects.cpp b/isapi/src/PyExtensionObjects.cpp
index 19208c1..3bf7223 100644
--- a/isapi/src/PyExtensionObjects.cpp
+++ b/isapi/src/PyExtensionObjects.cpp
@@ -24,8 +24,8 @@
  */
 
 // #define PY_SSIZE_T_CLEAN  // defined by isapi\src\StdAfx.h
-#include "stdafx.h"
-#include "pywintypes.h"
+#include "StdAfx.h"
+#include "PyWinTypes.h"
 #include "Utils.h"
 #include "PyExtensionObjects.h"
 #include "PythonEng.h"
@@ -815,7 +815,7 @@ PyObject *PyECB::IOCompletion(PyObject *self, PyObject *args)
 
     BOOL bRes;
     Py_BEGIN_ALLOW_THREADS bRes =
-        ecb->ServerSupportFunction(ecb->ConnID, HSE_REQ_IO_COMPLETION, DoIOCallback, NULL, NULL);
+        ecb->ServerSupportFunction(ecb->ConnID, HSE_REQ_IO_COMPLETION, (LPVOID)DoIOCallback, NULL, NULL);
     Py_END_ALLOW_THREADS if (!bRes) return SetPyECBError("ServerSupportFunction(HSE_REQ_IO_COMPLETION)");
     Py_RETURN_NONE;
 }
diff --git a/isapi/src/PyFilterObjects.cpp b/isapi/src/PyFilterObjects.cpp
index 0239313..06fc210 100644
--- a/isapi/src/PyFilterObjects.cpp
+++ b/isapi/src/PyFilterObjects.cpp
@@ -311,7 +311,7 @@ PyObject *PyHFC::SendResponseHeader(PyObject *self, PyObject *args)
         // The Java code passes "\r\n" as first DWORD, and header in second,
         // but docs clearly have second as unused.  Either way, I can't see the
         // specific header!
-        bRes = phfc->m_pfc->ServerSupportFunction(SF_REQ_SEND_RESPONSE_HEADER, status, (DWORD)header, 0);
+        bRes = phfc->m_pfc->ServerSupportFunction(SF_REQ_SEND_RESPONSE_HEADER, status, (DWORD_PTR)header, 0);
     Py_END_ALLOW_THREADS if (!bRes) return SetPyHFCError("SendResponseHeader");
     Py_INCREF(Py_None);
     return Py_None;
