diff --git a/com/win32comext/mapi/src/mapi_headers/EdkMdb.h b/com/win32comext/mapi/src/mapi_headers/EdkMdb.h
index 6de3859..39ac0c3 100644
--- a/com/win32comext/mapi/src/mapi_headers/EdkMdb.h
+++ b/com/win32comext/mapi/src/mapi_headers/EdkMdb.h
@@ -76,7 +76,7 @@
 #define PR_PROFILE_RECONNECT_INTERVAL PROP_TAG(PT_LONG, pidProfileMin + 0x1a)  // dup tag of PR_USER_NAME
 #define PR_PROFILE_SERVER_VERSION PROP_TAG(PT_LONG, pidProfileMin + 0x1b)
 
-/* SE 233155 - MarkH: EMSABP DCR /*
+/* SE 233155 - MarkH: EMSABP DCR */
 /* Properties in the abp section - I got these values from AlecDun (Outlook team) */
 #define PR_PROFILE_ABP_ALLOW_RECONNECT PROP_TAG(PT_LONG, pidProfileMin + 0x39)
 #define PR_PROFILE_ABP_MTHREAD_TIMEOUT_SECS PROP_TAG(PT_LONG, pidProfileMin + 0x3A)
diff --git a/com/win32comext/mapi/src/mapi_headers/MAPI.h b/com/win32comext/mapi/src/mapi_headers/MAPI.h
index 038b58f..dbc0e40 100644
--- a/com/win32comext/mapi/src/mapi_headers/MAPI.h
+++ b/com/win32comext/mapi/src/mapi_headers/MAPI.h
@@ -20,6 +20,13 @@
 #pragma once
 #endif
 
+#ifndef _MSC_VER
+#define __in
+#define __in_opt
+#define __out
+#define __deref_out_ecount_full(x)
+#endif
+
 /*
  *  Types.
  */
@@ -160,7 +167,7 @@ typedef struct
 
 /* #define MAPI_LOGON_UI        0x00000001     Display logon UI             */
 /* #define MAPI_NEW_SESSION     0x00000002     Don't use shared session     */
-/* #define MAPI_LONG_MSGID		0x00004000	/* allow 512 char returned ID	*/
+/* #define MAPI_LONG_MSGID      0x00004000     allow 512 char returned ID	*/
 
 /* MAPIAddress() flags.     */
 
diff --git a/com/win32comext/mapi/src/mapi_headers/MAPICode.h b/com/win32comext/mapi/src/mapi_headers/MAPICode.h
index 467023a..a27609d 100644
--- a/com/win32comext/mapi/src/mapi_headers/MAPICode.h
+++ b/com/win32comext/mapi/src/mapi_headers/MAPICode.h
@@ -216,5 +216,4 @@
 #endif
 
 #endif	/* MAPICODE_H */
-
 
diff --git a/com/win32comext/mapi/src/mapi_headers/MAPIDefS.h b/com/win32comext/mapi/src/mapi_headers/MAPIDefS.h
index 159935b..a44ba3c 100644
--- a/com/win32comext/mapi/src/mapi_headers/MAPIDefS.h
+++ b/com/win32comext/mapi/src/mapi_headers/MAPIDefS.h
@@ -51,6 +51,12 @@
 #include <stddef.h>
 #endif
 
+#ifndef _MSC_VER
+#define __in
+#define __out
+#define __deref_out_ecount_full(x)
+#endif
+
 /* Array dimension for structures with variable-sized arrays at the end. */
 
 #ifndef MAPI_DIM
@@ -2728,5 +2734,4 @@ DECLARE_MAPI_INTERFACE_(IMAPIProviderShutdown, IUnknown)
 #endif
 
 #endif /* MAPIDEFS_H */
-
 
diff --git a/com/win32comext/mapi/src/mapi_headers/MAPIForm.h b/com/win32comext/mapi/src/mapi_headers/MAPIForm.h
index cfdd84a..02b9669 100644
--- a/com/win32comext/mapi/src/mapi_headers/MAPIForm.h
+++ b/com/win32comext/mapi/src/mapi_headers/MAPIForm.h
@@ -631,5 +631,4 @@ DECLARE_MAPI_INTERFACE_(IMAPIFormFactory, IUnknown)
 
 #endif							/* MAPIFORM_H */
 
-
 
diff --git a/com/win32comext/mapi/src/mapi_headers/MAPIWin.h b/com/win32comext/mapi/src/mapi_headers/MAPIWin.h
index cc9ef14..9e913e9 100644
--- a/com/win32comext/mapi/src/mapi_headers/MAPIWin.h
+++ b/com/win32comext/mapi/src/mapi_headers/MAPIWin.h
@@ -272,5 +272,4 @@ BOOL WINAPI IsBadBoundedStringPtr(const void FAR* lpsz, UINT cchMax);
 #endif
 
 #endif /* __MAPIWIN_H__ */
-
 
diff --git a/com/win32comext/mapi/src/mapi_headers/MSPST.h b/com/win32comext/mapi/src/mapi_headers/MSPST.h
index dae5b31..9ca667d 100644
--- a/com/win32comext/mapi/src/mapi_headers/MSPST.h
+++ b/com/win32comext/mapi/src/mapi_headers/MSPST.h
@@ -97,5 +97,4 @@
                                 0xd9, 0x6e, 0x00, 0x00 }
 
 #endif  /* _MSPST_H_ */
-
 
