diff -rup gdlmm-2.30.0-org/gdl/gdlmm/Makefile.am gdlmm-2.30.0/gdl/gdlmm/Makefile.am
--- gdlmm-2.30.0-org/gdl/gdlmm/Makefile.am	2010-04-26 04:21:18.000000000 +0800
+++ gdlmm-2.30.0/gdl/gdlmm/Makefile.am	2015-10-22 22:06:15.021489400 +0800
@@ -23,8 +23,8 @@ include $(top_srcdir)/build/compile-bind
 
 local_cppflags = $(binding_includes) $(binding_cppflags) -DGDLMM_BUILD=1
 
-AM_CPPFLAGS = $(local_cppflags) $(GTHREAD_CFLAGS) $(GDLMM_CFLAGS)
-AM_CXXFLAGS = $(GDLMM_WXXFLAGS)
+AM_CPPFLAGS = $(local_cppflags) $(GTHREAD_CFLAGS) $(GDLMM_CFLAGS) -std=c++11
+AM_CXXFLAGS = $(GDLMM_WXXFLAGS) -std=c++11
 
 libgdlmm_@GDLMM_API_VERSION@_la_SOURCES = $(binding_sources)
 libgdlmm_@GDLMM_API_VERSION@_la_LDFLAGS = -no-undefined -version-info $(GDLMM_SO_VERSION)
diff -rup gdlmm-2.30.0-org/gdl/gdlmm/Makefile.in gdlmm-2.30.0/gdl/gdlmm/Makefile.in
--- gdlmm-2.30.0-org/gdl/gdlmm/Makefile.in	2010-04-26 04:32:22.000000000 +0800
+++ gdlmm-2.30.0/gdl/gdlmm/Makefile.in	2015-10-22 22:06:07.821479300 +0800
@@ -300,8 +300,8 @@ binding_private_HEADERS = $(files_built_
 binding_built_sources = $(files_built_cc) $(files_built_h) $(files_built_ph)
 MAINTAINERCLEANFILES = $(binding_built_sources) $(addprefix $(srcdir)/,$(binding_built_sources))
 local_cppflags = $(binding_includes) $(binding_cppflags) -DGDLMM_BUILD=1
-AM_CPPFLAGS = $(local_cppflags) $(GTHREAD_CFLAGS) $(GDLMM_CFLAGS)
-AM_CXXFLAGS = $(GDLMM_WXXFLAGS)
+AM_CPPFLAGS = $(local_cppflags) $(GTHREAD_CFLAGS) $(GDLMM_CFLAGS) -std=c++11
+AM_CXXFLAGS = $(GDLMM_WXXFLAGS) -std=c++11
 libgdlmm_@GDLMM_API_VERSION@_la_SOURCES = $(binding_sources)
 libgdlmm_@GDLMM_API_VERSION@_la_LDFLAGS = -no-undefined -version-info $(GDLMM_SO_VERSION)
 libgdlmm_@GDLMM_API_VERSION@_la_LIBADD = $(GDLMM_LIBS)
