diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2015-04-08 01:00:08.000000000 +0600
+++ b/configure.ac	2015-04-09 18:44:40.857793700 +0600
@@ -467,7 +467,7 @@
 	  dnl Note double square brackets, for extra m4 quoting.
 	  tmp_bits=`grep GMP_NUMB_BITS "$srcdir/$asm_dir/$tmp_h" \
             | sed 's/^.*GMP_NUMB_BITS(\([[0-9]]*\)).*$/\1/'`
-	  if test "$tmp_bits" && test "$tmp_bits" != '${GMP_NUMB_BITS}' ; then
+	  if test "$tmp_bits" && test "$tmp_bits" != "${GMP_NUMB_BITS}" ; then
 	     AC_MSG_WARN([skipping $tmp_h, because GMP_NUMB_BITS != $tmp_bits])
 	     continue
 	  fi
