--- llvm-10.0.0.src/include/llvm/ADT/Triple.h.orig	2020-05-15 15:49:40.073000000 +0200
+++ llvm-10.0.0.src/include/llvm/ADT/Triple.h	2020-05-15 20:51:29.844520400 +0200
@@ -809,7 +809,7 @@

   /// Tests whether the target uses emulated TLS as default.
   bool hasDefaultEmulatedTLS() const {
-    return isAndroid() || isOSOpenBSD() || isWindowsCygwinEnvironment();
+    return isAndroid() || isOSOpenBSD() || isOSCygMing();
   }

   /// Tests whether the target uses -data-sections as default.
