--- kcoreaddons-5.24.0/src/lib/util/kuser_win.cpp.orig	2016-07-03 02:03:33.000000000 +0200
+++ kcoreaddons-5.24.0/src/lib/util/kuser_win.cpp	2016-07-20 17:18:09.564831500 +0200
@@ -853,7 +853,7 @@
     HANDLE _token;
     if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &_token)) {
         qWarning("Failed to get the token for the current process: %d", (int)GetLastError());
-        return false;
+        return nullptr;
     }
     ScopedHANDLE token(_token, handleCloser);
     // query required size
