--- plplot-5.13.0/src/CMakeLists.txt.orig
+++ plplot-5.13.0/src/CMakeLists.txt
@@ -263,9 +263,9 @@
 if(PL_HAVE_QHULL)
   list(APPEND libplplot_LINK_LIBRARIES PLPLOT::csironn)
   if(QHULL_RPATH)
-    list(APPEND pc_libplplot_LINK_FLAGS -l${WRITEABLE_TARGET}csironn -L${QHULL_RPATH} -lqhull)
+    list(APPEND pc_libplplot_LINK_FLAGS -l${WRITEABLE_TARGET}csironn -L${QHULL_RPATH})
   else(QHULL_RPATH)
-    list(APPEND pc_libplplot_LINK_FLAGS -l${WRITEABLE_TARGET}csironn -lqhull)
+    list(APPEND pc_libplplot_LINK_FLAGS -l${WRITEABLE_TARGET}csironn)
   endif(QHULL_RPATH)
 
   # Needed by plgridd.c.
