diff -urN plplot-5.15.0/drivers/wingdi.c plplot-5.15.0-wingdi-patch/drivers/wingdi.c
--- plplot-5.15.0/drivers/wingdi.c	2019-06-02 01:24:33.000000000 +0200
+++ plplot-5.15.0-wingdi-patch/drivers/wingdi.c	2021-01-08 22:52:16.754725400 +0100
@@ -198,7 +198,7 @@
     HCURSOR cursor;
 
     cursor = LoadCursor( NULL, IDC_CROSS );
-    SetClassLongPtr( dev->plot, GCL_HCURSOR, (long) cursor );
+    SetClassLongPtr( dev->plot, GCLP_HCURSOR, (long) cursor );
     return SetCursor( cursor );
 }
 
@@ -208,7 +208,7 @@
     HCURSOR cursor;
 
     cursor = LoadCursor( NULL, IDC_ARROW );
-    SetClassLongPtr( dev->plot, GCL_HCURSOR, (LONG_PTR) cursor );
+    SetClassLongPtr( dev->plot, GCLP_HCURSOR, (LONG_PTR) cursor );
     SetCursor( cursor );
 }
 
@@ -218,7 +218,7 @@
     HCURSOR cursor;
 
     cursor = LoadCursor( NULL, IDC_WAIT );
-    SetClassLongPtr( dev->plot, GCL_HCURSOR, (LONG_PTR) cursor );
+    SetClassLongPtr( dev->plot, GCLP_HCURSOR, (LONG_PTR) cursor );
     SetCursor( cursor );
 }
 
diff -urN plplot-5.15.0/drivers/wingdi.c plplot-5.15.0-wingdi-patch/drivers/wingdi.c
--- plplot-5.15.0/drivers/wingdi.c	2021-01-08 22:57:01.930905200 +0100
+++ plplot-5.15.0-wingdi-patch/drivers/wingdi.c	2021-01-08 23:46:11.503662600 +0100
@@ -1278,7 +1278,7 @@
     int    programlength;
 #endif
 
-    pls->debug = 1;
+    // pls->debug = 1;
     pldebug( "wingdi", "Device Init\n" );
 
     // Initialize the module
