diff -urN plplot-5.15.0/CMakeLists.txt plplot-5.15.0-cmake-patched/CMakeLists.txt
--- plplot-5.15.0/CMakeLists.txt	2019-06-02 01:24:33.000000000 +0200
+++ plplot-5.15.0-cmake-patched/CMakeLists.txt	2021-01-10 17:11:14.257972900 +0100
@@ -140,7 +140,7 @@
 # the import libraries for system libraries.
 if(MINGW)
   get_filename_component(MINGWBINPATH ${CMAKE_C_COMPILER} PATH)
-  set(MINGWLIBPATH ${MINGWBINPATH}/../lib
+  set(MINGWLIBPATH ${MINGWBINPATH}/../lib;${MINGWBINPATH}/../$ENV{MINGW_CHOST}/lib
     CACHE FILEPATH
     "Path to MinGW import libraries")
 endif(MINGW)
