From d5c628a8b9a263a74874612638ff5426c4eff36a Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Sat, 4 Apr 2015 19:08:51 +0800
Subject: [PATCH] Remove the usages of stdout to workaround the missing of
 __getreent

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 test/dump.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/dump.c b/test/dump.c
index 0d6d700..bb4abe7 100644
--- a/test/dump.c
+++ b/test/dump.c
@@ -25,9 +25,9 @@ int main(int argc, char **argv)
   int i;
   torun *torun;

-  printf("INIT\n"); fflush(stdout);
+  printf("INIT\n");
   flexdll_dump_exports(NULL);
-  printf("OK\n"); fflush(stdout);
+  printf("OK\n");
   for (i = 1; i < argc; i++) {
     printf("** Loading %s\n", argv[i]);
     handle = flexdll_dlopen(argv[i], FLEXDLL_RTLD_GLOBAL);
@@ -39,7 +39,7 @@ int main(int argc, char **argv)
     if (NULL == handle) { printf("error: %s\n", flexdll_dlerror()); exit(2); }

     torun = flexdll_dlsym(handle, "torun");
-    printf("Now running %p...\n", torun); fflush(stdout);
+    printf("Now running %p...\n", torun);
     if (torun) torun();
   }
   exit(0);
--
2.3.4
