diff --git a/glib/adbc-arrow-glib/meson.build b/glib/adbc-arrow-glib/meson.build
index 732e27033..10dd80537 100644
--- a/glib/adbc-arrow-glib/meson.build
+++ b/glib/adbc-arrow-glib/meson.build
@@ -82,6 +82,7 @@ pkgconfig.generate(
     version: meson.project_version(),
 )
 
+arrow_version_major = arrow_glib.version().split('.')[0]
 adbc_arrow_glib_gir = \
     gnome.generate_gir(
     libadbc_arrow_glib,
@@ -91,7 +92,7 @@ adbc_arrow_glib_gir = \
     fatal_warnings: gi_fatal_warnings,
     header: 'adbc-arrow-glib/adbc-arrow-glib.h',
     identifier_prefix: 'GADBCArrow',
-    includes: ['ADBC-1.0', 'Arrow-1.0'],
+    includes: ['ADBC-1.0', 'Arrow-@0@.0'.format(arrow_version_major)],
     install: true,
     namespace: 'ADBCArrow',
     nsversion: api_version,
