--- scipy-1.9.0/scipy/meson.build.orig	2022-08-02 08:50:49.182475300 +0200
+++ scipy-1.9.0/scipy/meson.build	2022-08-03 08:08:17.971413600 +0200
@@ -10,7 +10,6 @@
     add_global_link_arguments(gcc_link_args, language: ['c', 'cpp'])
     # Force gcc to float64 long doubles for compatibility with MSVC
     # builds, for C only.
-    add_global_arguments('-mlong-double-64', language: 'c')
     # Make fprintf("%zd") work (see https://github.com/rgommers/scipy/issues/118)
     add_global_arguments('-D__USE_MINGW_ANSI_STDIO=1', language: ['c', 'cpp'])
     # Manual add of MS_WIN64 macro when not using MSVC.
