--- a/scipy/meson.build
+++ b/scipy/meson.build
@@ -11,7 +11,6 @@
   add_project_link_arguments(gcc_link_args, language: ['c', 'cpp', 'fortran'])
   # Force gcc to float64 long doubles for compatibility with MSVC
   # builds, for C only.
-  add_project_arguments('-mlong-double-64', language: 'c')
   # Make fprintf("%zd") work (see https://github.com/rgommers/scipy/issues/118)
   add_project_arguments('-D__USE_MINGW_ANSI_STDIO=1', language: ['c', 'cpp'])
   # Silence warnings emitted by PyOS_snprintf for (%zd), see
