diff -Naur scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/complex16/second.F scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/complex16/second.F
--- scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/complex16/second.F	2023-02-24 19:27:22.070327100 +0100
+++ scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/complex16/second.F	2023-02-24 19:34:43.250667300 +0100
@@ -39,8 +39,7 @@
       T1 = ETIME_( TARRAY )
       T = T1
 #else
-      T1 = ETIME( TARRAY )
-      T  = TARRAY( 1 )
+      CALL CPU_TIME(T)
 #endif
 
       RETURN
diff -Naur scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/complex8/second.F scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/complex8/second.F
--- scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/complex8/second.F	2023-02-24 19:27:22.263811300 +0100
+++ scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/complex8/second.F	2023-02-24 19:35:00.037285900 +0100
@@ -39,8 +39,7 @@
       T1 = ETIME_( TARRAY )
       T = T1
 #else
-      T1 = ETIME( TARRAY )
-      T  = TARRAY( 1 )
+      CALL CPU_TIME(T)
 #endif
 
       RETURN
diff -Naur scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/double/second.F scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/double/second.F
--- scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/double/second.F	2023-02-24 19:27:22.347849900 +0100
+++ scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/double/second.F	2023-02-24 19:34:55.537199900 +0100
@@ -39,8 +39,7 @@
       T1 = ETIME_( TARRAY )
       T = T1
 #else
-      T1 = ETIME( TARRAY )
-      T  = TARRAY( 1 )
+      CALL CPU_TIME(T)
 #endif
 
       RETURN
diff -Naur scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/single/second.F scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/single/second.F
--- scipy-1.10.1-orig/scipy/sparse/linalg/_propack/PROPACK/single/second.F	2023-02-24 19:27:22.425959300 +0100
+++ scipy-1.10.1/scipy/sparse/linalg/_propack/PROPACK/single/second.F	2023-02-24 19:34:51.285499000 +0100
@@ -39,8 +39,7 @@
       T1 = ETIME_( TARRAY )
       T = T1
 #else
-      T1 = ETIME( TARRAY )
-      T  = TARRAY( 1 )
+      CALL CPU_TIME(T)
 #endif
 
       RETURN
