From d7cd42677f9c59590127badb4c0972df5a65f6bc Mon Sep 17 00:00:00 2001
From: Roberto Bagnara <roberto.bagnara@bugseng.com>
Date: Sun, 9 Aug 2020 08:08:05 +0200
Subject: [PATCH] Make test immune from LTO.

---
 m4/ac_check_fpu_control.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/m4/ac_check_fpu_control.m4 b/m4/ac_check_fpu_control.m4
index 707dc7981..b69fa802d 100644
--- a/m4/ac_check_fpu_control.m4
+++ b/m4/ac_check_fpu_control.m4
@@ -49,9 +49,9 @@ main() {
 
 #else
 
-     float  nf1 =  -3, pf1 = 3,  f2 =  5;
-     double nd1 =  -7, pd1 = 7,  d2 = 11;
-long double nl1 = -13, pl1 = 13, l2 = 17;
+     volatile float  nf1 =  -3, pf1 = 3,  f2 =  5;
+     volatile double nd1 =  -7, pd1 = 7,  d2 = 11;
+volatile long double nl1 = -13, pl1 = 13, l2 = 17;
 
       float nf[2], pf[2];
      double nd[2], pd[2];
