--- a/configure.ac
+++ b/configure.ac
@@ -536,22 +536,7 @@
   then
     OPT_FLAGS="$OPT_FLAGS -fp-model strict -fp-speculation off"
   else
-    # -frounding-math is not yet supported in clang++ or llvm-g++.
-    if test x"$CLANGXX" = xyes || test x"$LLVM_GXX" = xyes
-    then
-      if test x"$use_fpmath" = xyes
-      then
-        AC_MSG_WARN([CANNOT RELY ON DIRECTED ROUNDING:
-*** DISABLED ALL ABSTRACTIONS BASED ON MACHINE FLOATING-POINT NUMBERS.
-*** A compiler has been detected that does not provide support for
-*** -frounding-math (or any other option with the same semantics).
-*** Hence, we cannot rely on floating-point computations to happen
-*** in agreement with the rounding direction(s) used by the PPL.])
-        use_fpmath=no
-      fi
-    else
-      OPT_FLAGS="$OPT_FLAGS -frounding-math"
-    fi
+    OPT_FLAGS="$OPT_FLAGS -frounding-math"
   fi
 fi
 
