diff -Naur zziplib-0.13.69.orig/configure.ac zziplib-0.13.69/configure.ac
--- zziplib-0.13.69.orig/configure.ac	2018-05-01 19:44:36.387886000 -0400
+++ zziplib-0.13.69/configure.ac	2018-05-01 19:46:24.715029400 -0400
@@ -317,7 +317,7 @@
 ZZIPLIB_LDFLAGS=""
 case "$host_os" in mingw*)
     # we may assume that the backend is gnu-ld but DLLs need full resolution
-    ZZIPLIB_LDFLAGS="${wl}-no-undefined ${wl}-export-all-symbols -mconsole"
+    ZZIPLIB_LDFLAGS="-no-undefined ${wl}-no-undefined ${wl}-export-all-symbols -mconsole"
     RESOLVES=' '
 ;; darwin*)
     # darwin never uses gnu-ld but the linker can do it anyway
diff -Naur zziplib-0.13.69.orig/m4/libtool.m4 zziplib-0.13.69/m4/libtool.m4
--- zziplib-0.13.69.orig/m4/libtool.m4	2018-05-01 19:44:36.325386000 -0400
+++ zziplib-0.13.69/m4/libtool.m4	2018-05-01 19:45:41.528967100 -0400
@@ -3467,7 +3467,8 @@
   # Base MSYS/MinGW do not provide the 'file' command needed by
   # func_win32_libid shell function, so use a weaker test based on 'objdump',
   # unless we find 'file', for example because we are cross-compiling.
-  if ( file / ) >/dev/null 2>&1; then
+  # func_win32_libid assumes BSD nm, so disallow it if using MS dumpbin.
+  if ( test "$lt_cv_nm_interface" = "BSD nm" && file / ) >/dev/null 2>&1; then
     lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
     lt_cv_file_magic_cmd='func_win32_libid'
   else
