diff -ru zziplib-0.13.62/configure patched/configure
--- zziplib-0.13.62/configure	2012-03-11 21:28:52.000000000 +0100
+++ patched/configure	2013-08-02 23:28:48.122420796 +0200
@@ -5908,11 +5908,13 @@
   # Base MSYS/MinGW do not provide the 'file' command needed by
   # func_win32_libid shell function, so use a weaker test based on 'objdump',
   # unless we find 'file', for example because we are cross-compiling.
-  if ( file / ) >/dev/null 2>&1; then
+  # func_win32_libid assumes BSD nm, so disallow it if using MS dumpbin.
+  if ( test "$lt_cv_nm_interface" = "BSD nm" && file / ) >/dev/null 2>&1; then
     lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
     lt_cv_file_magic_cmd='func_win32_libid'
   else
-    lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
+    # Keep this pattern in sync with the one in func_win32_libid.
+    lt_cv_deplibs_check_method='file_magic file format (pei*-i386(.*architecture: i386)?|pe-arm-wince|pe-x86-64)'
     lt_cv_file_magic_cmd='$OBJDUMP -f'
   fi
   ;;
@@ -13724,7 +13726,7 @@
 ZZIPLIB_LDFLAGS=""
 case "$host_os" in mingw*)
     # we may assume that the backend is gnu-ld but DLLs need full resolution
-    ZZIPLIB_LDFLAGS="${wl}-no-undefined ${wl}-export-all-symbols -mconsole"
+    ZZIPLIB_LDFLAGS="-no-undefined ${wl}-no-undefined ${wl}-export-all-symbols -mconsole"
     RESOLVES=' '
 ;; darwin*)
     # darwin never uses gnu-ld but the linker can do it anyway
diff -ru zziplib-0.13.62/configure.ac patched/configure.ac
--- zziplib-0.13.62/configure.ac	2012-03-11 21:28:35.000000000 +0100
+++ patched/configure.ac	2013-08-03 00:25:42.305761319 +0200
@@ -294,7 +294,7 @@
 ZZIPLIB_LDFLAGS=""
 case "$host_os" in mingw*)
     # we may assume that the backend is gnu-ld but DLLs need full resolution
-    ZZIPLIB_LDFLAGS="${wl}-no-undefined ${wl}-export-all-symbols -mconsole"
+    ZZIPLIB_LDFLAGS="-no-undefined ${wl}-no-undefined ${wl}-export-all-symbols -mconsole"
     RESOLVES=' '
 ;; darwin*)
     # darwin never uses gnu-ld but the linker can do it anyway
diff -ru zziplib-0.13.62/m4/libtool.m4 patched/m4/libtool.m4
--- zziplib-0.13.62/m4/libtool.m4	2009-08-23 13:38:19.000000000 +0200
+++ patched/m4/libtool.m4	2013-08-02 22:57:24.381209038 +0200
@@ -2999,11 +2999,13 @@
   # Base MSYS/MinGW do not provide the 'file' command needed by
   # func_win32_libid shell function, so use a weaker test based on 'objdump',
   # unless we find 'file', for example because we are cross-compiling.
-  if ( file / ) >/dev/null 2>&1; then
+  # func_win32_libid assumes BSD nm, so disallow it if using MS dumpbin.
+  if ( test "$lt_cv_nm_interface" = "BSD nm" && file / ) >/dev/null 2>&1; then
     lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
     lt_cv_file_magic_cmd='func_win32_libid'
   else
-    lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
+    # Keep this pattern in sync with the one in func_win32_libid.
+    lt_cv_deplibs_check_method='file_magic file format (pei*-i386(.*architecture: i386)?|pe-arm-wince|pe-x86-64)'
     lt_cv_file_magic_cmd='$OBJDUMP -f'
   fi
   ;;
diff -ru zziplib-0.13.62/uses/ltmain.sh patched/uses/ltmain.sh
--- zziplib-0.13.62/uses/ltmain.sh	2009-08-23 13:38:18.000000000 +0200
+++ patched/uses/ltmain.sh	2013-08-02 23:36:33.609957248 +0200
@@ -2560,7 +2560,7 @@
     ;;
   *ar\ archive*) # could be an import, or static
     if eval $OBJDUMP -f $1 | $SED -e '10q' 2>/dev/null |
-       $EGREP 'file format pe-i386(.*architecture: i386)?' >/dev/null ; then
+       $EGREP 'file format (pei*-i386(.*architecture: i386)?|pe-arm-wince|pe-x86-64)' >/dev/null ; then
       win32_nmres=`eval $NM -f posix -A $1 |
 	$SED -n -e '
 	    1,100{