From 2a401e94c6b056cf4143f3a56519840614e816cc Mon Sep 17 00:00:00 2001
From: kreijstal <rainb@tfwno.gf>
Date: Sun, 25 Aug 2024 11:19:44 +0200
Subject: [PATCH] Use msys2 perl, since it doesn't work with windows perl

---
 bin/redirect | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/redirect b/bin/redirect
index 0e7becf..d767de3 100644
--- a/bin/redirect
+++ b/bin/redirect
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 ######################################################################
 #
 # Copyright 2003-2024 by Wilson Snyder. This program is free software; you
-- 
2.46.0

