---
 src/V3Os.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/V3Os.cpp b/src/V3Os.cpp
index b6d7375e..2fdcecca 100644
--- a/src/V3Os.cpp
+++ b/src/V3Os.cpp
@@ -53,8 +53,10 @@
 # ifndef WIFEXITED
 #  ifdef __MINGW32__
 #   define WIFEXITED(w)	(((w) & 0xC0000000) == 0)
+#   define WEXITSTATUS(w)	((w) & ~0xC0000000)
 #  else
 #   define WIFEXITED(w)	(((w) & 0377) == 0)
+#   define WEXITSTATUS(w)	(((w) >> 8) & 0377)
 #  endif
 # endif
 #else
--
