--- ./doc/Makefile.am.orig	2015-03-28 11:22:54.627520700 +0100
+++ ./doc/Makefile.am	2015-03-28 11:23:10.053773500 +0100
@@ -1,21 +1,4 @@
 SUBDIRS = reference
 
-MAN1_TXT = chezdav.txt
-
-if HAVE_DOCTOOLS
-# Manual pages are generated from .txt via Docbook
-man1_MANS = ${MAN1_TXT:%.txt=%.1}
-
-SUFFIXES = .txt .xml .1
-
-.xml.1:
-	$(XMLTO) man $< 2>&1
-
-.txt.xml:
-	$(ASCIIDOC) --backend=docbook --doctype=manpage -o $@ $<
-endif
-
-EXTRA_DIST = $(MAN1_TXT)
-CLEANFILES = $(man1_MANS)
 
 -include $(top_srcdir)/git.mk
--- ./m4/intltool.m4.orig	2015-03-28 11:13:05.423072100 +0100
+++ ./m4/intltool.m4	2015-03-28 11:24:53.078804700 +0100
@@ -155,31 +155,6 @@
 # Substitute ALL_LINGUAS so we can use it in po/Makefile
 AC_SUBST(ALL_LINGUAS)
 
-# Set DATADIRNAME correctly if it is not set yet
-# (copied from glib-gettext.m4)
-if test -z "$DATADIRNAME"; then
-  AC_LINK_IFELSE(
-    [AC_LANG_PROGRAM([[]],
-                     [[extern int _nl_msg_cat_cntr;
-                       return _nl_msg_cat_cntr]])],
-    [DATADIRNAME=share],
-    [case $host in
-    *-*-solaris*)
-    dnl On Solaris, if bind_textdomain_codeset is in libc,
-    dnl GNU format message catalog is always supported,
-    dnl since both are added to the libc all together.
-    dnl Hence, we'd like to go with DATADIRNAME=share
-    dnl in this case.
-    AC_CHECK_FUNC(bind_textdomain_codeset,
-      [DATADIRNAME=share], [DATADIRNAME=lib])
-    ;;
-    *)
-    [DATADIRNAME=lib]
-    ;;
-    esac])
-fi
-AC_SUBST(DATADIRNAME)
-
 IT_PO_SUBDIR([po])
 
 ])
--- ./po/Makefile.in.in.orig	2015-03-28 11:14:58.472814600 +0100
+++ ./po/Makefile.in.in	2015-03-28 11:24:28.814784300 +0100
@@ -34,7 +34,7 @@
 datarootdir = @datarootdir@
 libdir = @libdir@
 DATADIRNAME = @DATADIRNAME@
-itlocaledir = $(prefix)/$(DATADIRNAME)/locale
+localedir = @localedir@
 subdir = po
 install_sh = @install_sh@
 # Automake >= 1.8 provides @mkdir_p@.
@@ -108,7 +108,7 @@
 install-data-yes: all
 	linguas="$(USE_LINGUAS)"; \
 	for lang in $$linguas; do \
-	  dir=$(DESTDIR)$(itlocaledir)/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$(localedir)/$$lang/LC_MESSAGES; \
 	  $(mkdir_p) $$dir; \
 	  if test -r $$lang.gmo; then \
 	    $(INSTALL_DATA) $$lang.gmo $$dir/$(GETTEXT_PACKAGE).mo; \
@@ -142,8 +142,8 @@
 uninstall:
 	linguas="$(USE_LINGUAS)"; \
 	for lang in $$linguas; do \
-	  rm -f $(DESTDIR)$(itlocaledir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo; \
-	  rm -f $(DESTDIR)$(itlocaledir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo.m; \
+	  rm -f $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo; \
+	  rm -f $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo.m; \
 	done
 
 check: all $(GETTEXT_PACKAGE).pot
