diff -Naur tcl8.6.5-orig/unix/installManPage tcl8.6.5/unix/installManPage
--- tcl8.6.5-orig/unix/installManPage	2016-03-01 04:59:35.000000000 +0300
+++ tcl8.6.5/unix/installManPage	2016-03-03 08:47:43.725911500 +0300
@@ -103,7 +103,7 @@
     Target=$Target.$Section$Suffix
     rm -f $Dir/$Target $Dir/$Target.*
     if test -z "$First" ; then
-	First=$Target
+	First=$(echo $Target | sed -e 's/:/-/g')
 	sed -e "/man\.macros/r $SrcDir/man.macros" -e "/man\.macros/d" \
 	    $ManPage > $Dir/$First
 	chmod 444 $Dir/$First
