diff -Naur tcl8.6.5-orig/win/configure.in tcl8.6.5/win/configure.in
--- tcl8.6.5-orig/win/configure.in	2016-03-03 08:47:47.861497900 +0300
+++ tcl8.6.5/win/configure.in	2016-03-03 08:47:47.952488800 +0300
@@ -105,15 +105,7 @@
 
 SC_CONFIG_CFLAGS
 
-# Cross-compiling
-case ${host_alias} in
-*mingw32*)
-    TCL_EXE="tclsh"
-    ;;
-*)
     TCL_EXE="TCL_LIBRARY=\"\${LIBRARY_DIR}\"; export TCL_LIBRARY; ./\${TCLSH}"
-    ;;
-esac
 
 #------------------------------------------------------------------------
 #	Add stuff for zlib; note that this is mostly done in the makefile now
