diff -Naur tcl8.6.5-orig/generic/tclInt.h tcl8.6.5/generic/tclInt.h
--- tcl8.6.5-orig/generic/tclInt.h	2016-02-25 23:12:37.000000000 +0300
+++ tcl8.6.5/generic/tclInt.h	2016-03-03 08:47:50.393244700 +0300
@@ -3204,7 +3204,7 @@
 MODULE_SCOPE int	TclClockOldscanObjCmd(
 			    ClientData clientData, Tcl_Interp *interp,
 			    int objc, Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_CloseObjCmd(ClientData clientData,
+extern int	Tcl_CloseObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ConcatObjCmd(ClientData clientData,
@@ -3387,7 +3387,7 @@
 MODULE_SCOPE int	Tcl_RepresentationCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_ReturnObjCmd(ClientData clientData,
+extern int	Tcl_ReturnObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ScanObjCmd(ClientData clientData,
diff -Naur tcl8.6.5-orig/generic/tclPort.h tcl8.6.5/generic/tclPort.h
--- tcl8.6.5-orig/generic/tclPort.h	2016-02-25 23:12:37.000000000 +0300
+++ tcl8.6.5/generic/tclPort.h	2016-03-03 08:47:50.394244600 +0300
@@ -20,7 +20,7 @@
 #if defined(_WIN32)
 #   include "tclWinPort.h"
 #else
-#   include "tclUnixPort.h"
+#   include "../unix/tclUnixPort.h"
 #endif
 #include "tcl.h"
 
