--- pidgin-2.14.8/configure.ac.orig	2022-03-14 21:52:26.098261600 -0700
+++ pidgin-2.14.8/configure.ac	2022-03-14 21:52:48.004503900 -0700
@@ -361,7 +361,7 @@
 AC_SUBST(GLIB_LIBS)
 
 PKG_CHECK_MODULES(GIO_UNIX, [gio-unix-2.0 >= 2.26],
-			    AC_DEFINE(HAVE_GIOUNIX, 1, [Have gio-unix]), [])
+			    AC_DEFINE(HAVE_GIOUNIX, 1, [Have gio-unix]), :)
 
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
 AC_SUBST(GLIB_GENMARSHAL)
--- pidgin-2.14.8/pidgin/gtkrequest.c.orig	2022-03-14 22:27:31.371093000 -0700
+++ pidgin-2.14.8/pidgin/gtkrequest.c	2022-03-14 22:27:59.667532500 -0700
@@ -2174,7 +2174,7 @@
 	return ret;
 }
 #elif defined (_WIN32)
-static GstElement *create_screensrc_cb(PPurpleMedia *media, const gchar *session_id,
+static GstElement *create_screensrc_cb(PurpleMedia *media, const gchar *session_id,
 				       const gchar *participant)
 {
 	GObject *info;
