--- sundials-5.7.0.orig/include/cvodes/cvodes.h	2021-01-29 15:16:13.000000000 -0600
+++ sundials-5.7.0/include/cvodes/cvodes.h	2021-07-07 20:55:24.723436000 -0500
@@ -567,7 +567,7 @@
                                                CVadjCheckPointRec *ckpnt);
 
 /* CVLS interface function that depends on CVRhsFn */
-int CVodeSetJacTimesRhsFnB(void *cvode_mem, int which, CVRhsFn jtimesRhsFn);
+SUNDIALS_EXPORT int CVodeSetJacTimesRhsFnB(void *cvode_mem, int which, CVRhsFn jtimesRhsFn);
 
 
 /* Undocumented Optional Output Functions For Backward Problems */
