--- a/LiteEditor/OutputDebugStringThread.h	2015-06-01 23:22:47.501600000 +0300
+++ b/Outline/OutputDebugStringThread.h	2015-06-01 23:39:02.750800000 +0300
@@ -3,6 +3,10 @@
 
 #include <wx/thread.h> // Base class: wxThread
 #include "cl_command_event.h"
+#ifdef __WXMSW__
+#include <windows.h>
+#include <wx/msw/winundef.h>
+#endif
 
 wxDECLARE_EVENT(wxEVT_OUTPUT_DEBUG_STRING, clCommandEvent);
 class OutputDebugStringThread : public wxThread
--- a/Outline/outline_tab.cpp	2015-06-01 23:22:47.501600000 +0300
+++ b/Outline/outline_tab.cpp	2015-06-01 23:39:02.750800000 +0300
@@ -25,6 +25,9 @@
 #include "bitmap_loader.h"
 #include "clToolBar.h"
 #include "cl_config.h"
+#ifdef __WXMSW__
+#include <wx/msw/winundef.h>
+#endif
 #include "event_notifier.h"
 #include "file_logger.h"
 #include "fileextmanager.h"
--- a/CodeLite/SocketAPI/clSocketBase.h	2015-07-15 18:25:00.504000000 +0300
+++ b/CodeLite/SocketAPI/clSocketBase.h	2015-07-15 18:34:16.400800000 +0300
@@ -31,11 +31,12 @@
 #include <wx/string.h>
 #include "codelite_exports.h"
 
-#ifdef _WIN32
+#ifdef __WXMSW__
 #include <winsock2.h>
 
 typedef SOCKET socket_t;
 typedef int socklen_t;
+#include <wx/msw/winundef.h>
 #else
 typedef int socket_t;
 #define INVALID_SOCKET -1
