diff --git a/CodeLite/fileutils.cpp b/CodeLite/fileutils.cpp
index 46bbac151..3a42f2a56 100644
--- a/CodeLite/fileutils.cpp
+++ b/CodeLite/fileutils.cpp
@@ -587,7 +587,7 @@ std::string FileUtils::ToStdString(const wxString& str) { return StringUtils::To
 bool FileUtils::ReadBufferFromFile(const wxFileName& fn, wxString& data, size_t bufferSize)
 {
     if(!fn.FileExists()) { return false; }
-    std::wifstream fin(fn.GetFullPath().c_str(), std::ios::binary);
+    std::wifstream fin(std::string(fn.GetFullPath().c_str()), std::ios::binary);
     if(fin.bad()) {
         clERROR() << "Failed to open file:" << fn;
         return false;
