--- codelite/cmake/Modules/FindClang.cmake.orig	2014-09-06 22:05:50.675200000 +0400
+++ codelite/cmake/Modules/FindClang.cmake	2014-09-06 22:06:42.763600000 +0400
@@ -36,6 +36,7 @@
 FIND_AND_ADD_CLANG_LIB(clangIndex)
 FIND_AND_ADD_CLANG_LIB(clangLex)
 FIND_AND_ADD_CLANG_LIB(clangParse)
+FIND_AND_ADD_CLANG_LIB(clangRewrite)
 FIND_AND_ADD_CLANG_LIB(clangRewriteCore)
 FIND_AND_ADD_CLANG_LIB(clangRewriteFrontend)
 FIND_AND_ADD_CLANG_LIB(clangSema)
@@ -58,6 +58,9 @@
   message(STATUS "Found Clang: ${CLANG_INCLUDE_DIRS}")
   set(CLANG_LDFLAGS -L${LLVM_LIBRARY_DIRS} ${CLANG_LIBS})
   set(CLANG_LDFLAGS "-Wl,--start-group ${CLANG_LIBS} ${LLVM_MODULE_LIBS} -Wl,--end-group ${LLVM_LDFLAGS}")
+  if (MINGW)
+    set(CLANG_LDFLAGS "${CLANG_LDFLAGS} ${LLVM_SYSTEM_LIBS}")
+  endif()
   
 else (CLANG_FOUND)
   if (CLANG_FIND_REQUIRED)
--- codelite/cmake/Modules/FindLLVM.cmake.orig	2014-09-06 22:41:11.427600000 +0400
+++ codelite/cmake/Modules/FindLLVM.cmake	2014-09-06 22:43:29.378400000 +0400
@@ -7,6 +7,7 @@
 #  LLVM_CFLAGS       - llvm compiler flags
 #  LLVM_LDFLAGS      - llvm linker flags
 #  LLVM_MODULE_LIBS  - list of llvm libs for working with modules.
+#  LLVM_SYSTEM_LIBS  - list of 3rdparty libraries needed to link.
 
 find_program(LLVM_CONFIG_EXECUTABLE llvm-config DOC "llvm-config executable")
 
@@ -68,6 +69,12 @@
   OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 
+execute_process(
+  COMMAND ${LLVM_CONFIG_EXECUTABLE} --system-libs
+  OUTPUT_VARIABLE LLVM_SYSTEM_LIBS
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
 if (LLVM_CONFIG_EXECUTABLE)
   set(LLVM_FOUND TRUE)
 endif (LLVM_CONFIG_EXECUTABLE)
