--- codelite/CodeLite/cl_standard_paths.h.orig	2015-12-24 15:52:24.680223100 +0300
+++ codelite/CodeLite/cl_standard_paths.h	2015-12-24 15:53:38.368672300 +0300
@@ -118,6 +118,8 @@
      * OSX: /Users/name/Documents
      */
     wxString GetDocumentsDir() const;
+    
+    void IgnoreAppSubDir(const wxString & subdirPattern);
 };
 
 #endif // CLSTANDARDPATHS_H
--- codelite/CodeLite/cl_standard_paths.cpp.orig	2015-12-24 15:53:47.553753700 +0300
+++ codelite/CodeLite/cl_standard_paths.cpp	2015-12-24 15:57:59.523953000 +0300
@@ -29,9 +29,7 @@
 
 clStandardPaths::clStandardPaths()
 {
-#ifdef USE_POSIX_LAYOUT
-    wxStandardPaths::Get().IgnoreAppSubDir("bin");
-#endif
+    IgnoreAppSubDir("bin");
 }
 
 clStandardPaths::~clStandardPaths() {}
@@ -164,3 +162,10 @@
 #endif
     return path;
 }
+
+void clStandardPaths::IgnoreAppSubDir(const wxString & subdirPattern)
+{
+#ifdef USE_POSIX_LAYOUT
+    wxStandardPaths::Get().IgnoreAppSubDir(subdirPattern);
+#endif
+}
