--- codelite/LiteEditor/clang_pch_maker_thread.cpp.orig	2014-07-06 14:54:17.618600000 +0400
+++ codelite/LiteEditor/clang_pch_maker_thread.cpp	2014-07-06 14:55:02.094200000 +0400
@@ -434,7 +434,7 @@
     // First time, need to create it
     unsigned flags;
     if(reparse) {
-        flags = CXTranslationUnit_CXXPrecompiledPreamble | CXTranslationUnit_CacheCompletionResults |
+        flags = CXTranslationUnit_CacheCompletionResults |
                 CXTranslationUnit_PrecompiledPreamble | CXTranslationUnit_Incomplete |
                 CXTranslationUnit_DetailedPreprocessingRecord | CXTranslationUnit_CXXChainedPCH;
     } else {
