diff -Naur codelite-orig/CMakeLists.txt codelite-mingw/CMakeLists.txt
--- codelite-orig/CMakeLists.txt	2014-09-04 18:16:53.916000000 +0400
+++ codelite-mingw/CMakeLists.txt	2014-09-04 19:17:03.429000000 +0400
@@ -208,8 +208,10 @@
 
 set( PLUGINS_DIR "${CL_PREFIX}/${CL_INSTALL_LIBDIR}/codelite")
 add_definitions(-DYY_NEVER_INTERACTIVE=1)
+if (NOT MINGW)
 add_definitions(-DINSTALL_DIR=\"${CL_PREFIX}/share/codelite\")
 add_definitions(-DPLUGINS_DIR=\"${PLUGINS_DIR}\")
+endif()
 
 message("-- PLUGINS_DIR is set to ${PLUGINS_DIR}")
 
diff -Naur codelite-orig/codelitephp/CMakeLists.txt codelite-mingw/codelitephp/CMakeLists.txt
--- codelite-orig/codelitephp/CMakeLists.txt	2014-09-04 18:16:53.916000000 +0400
+++ codelite-mingw/codelitephp/CMakeLists.txt	2014-09-04 19:17:03.429000000 +0400
@@ -4,7 +4,9 @@
 # set the plugin name here
 set( PLUGIN_NAME "codelitephp")
 
+if (NOT MINGW)
 add_definitions(-DPLUGINS_DIR=\"${PLUGINS_DIR}\")
+endif()
 add_definitions(-DUSE_SFTP=1)
 
 # Our project is called 'plugin' this is how it will be called in
diff -Naur codelite-orig/CodeLite/CMakeLists.txt codelite-mingw/CodeLite/CMakeLists.txt
--- codelite-orig/CodeLite/CMakeLists.txt	2014-09-04 18:16:53.916000000 +0400
+++ codelite-mingw/CodeLite/CMakeLists.txt	2014-09-04 19:17:03.429000000 +0400
@@ -15,7 +15,8 @@
 include_directories("${CL_SRC_ROOT}/sdk/wxsqlite3/include" 
                     "${CL_SRC_ROOT}/sdk/codelite_indexer/network" 
                     "${CL_SRC_ROOT}/CodeLite" 
-                    "${CL_SRC_ROOT}/PCH")
+                    "${CL_SRC_ROOT}/PCH"
+                    "${CL_SRC_ROOT}/Interfaces")
 
 set( ADDITIONAL_LIBRARIES "" )
 
diff -Naur codelite-orig/CodeLite/cl_standard_paths.h codelite-mingw/CodeLite/cl_standard_paths.h
--- codelite-orig/CodeLite/cl_standard_paths.h	2014-09-04 18:16:53.916000000 +0400
+++ codelite-mingw/CodeLite/cl_standard_paths.h	2014-09-04 19:17:03.429000000 +0400
@@ -28,6 +28,7 @@
 
 #include <wx/string.h>
 #include "codelite_exports.h"
+#include "cl_defs.h"
 
 class WXDLLIMPEXP_CL clStandardPaths
 {
diff -Naur codelite-orig/CodeLite/cl_standard_paths.cpp codelite-mingw/CodeLite/cl_standard_paths.cpp
--- codelite-orig/CodeLite/cl_standard_paths.cpp	2014-09-04 18:16:53.916000000 +0400
+++ codelite-mingw/CodeLite/cl_standard_paths.cpp	2014-09-04 19:17:03.429000000 +0400
@@ -29,6 +29,7 @@
 
 clStandardPaths::clStandardPaths()
 {
+    wxStandardPaths::Get().IgnoreAppSubDir("bin");
 }
 
 clStandardPaths::~clStandardPaths()
@@ -78,7 +79,11 @@
     wxString pluginsDir = PLUGINS_DIR;
 #else
     wxFileName path(GetDataDir(), "");
+#   ifdef __MINGW64_VERSION_MAJOR
+    path.AppendDir(PLUGINS_DIR);
+#   else
     path.AppendDir("plugins");
+#   endif
     wxString pluginsDir = path.GetPath();
 #endif
     return pluginsDir;
diff -Naur codelite-orig/codelitephp/PHPParser/php_utils.cpp codelite-mingw/codelitephp/PHPParser/php_utils.cpp
--- codelite-orig/codelitephp/PHPParser/php_utils.cpp	2014-09-04 18:16:56.614800000 +0400
+++ codelite-mingw/codelitephp/PHPParser/php_utils.cpp	2014-09-04 19:03:12.911200000 +0400
@@ -98,9 +98,15 @@
     fn = wxFileName( PLUGINS_DIR, "");
     fn.AppendDir( "resources" );
 #else
+#   ifdef __MINGW64_VERSION_MAJOR
+    fn = wxFileName( wxStandardPaths::Get().GetExecutablePath());
+    fn.AppendDir( PLUGINS_DIR );
+    fn.AppendDir( "resources" );
+#   else
     fn = wxFileName( wxStandardPaths::Get().GetExecutablePath() );
     fn.AppendDir( "plugins" );
     fn.AppendDir( "resources" );
+#   endif
 #endif
     fn.AppendDir( "php" );
     return fn.GetPath();
diff -Naur codelite-orig/Interfaces/cl_defs.h codelite-mingw/Interfaces/cl_defs.h
--- codelite-orig/Interfaces/cl_defs.h	2014-09-04 18:16:54.259200000 +0400
+++ codelite-mingw/Interfaces/cl_defs.h	2014-09-04 18:27:21.972600000 +0400
@@ -57,6 +57,17 @@
 #    endif
 #    define CL_USE_NATIVEBOOK 0
 #else
+#   ifdef __MINGW64_VERSION_MAJOR
+#       ifndef PLUGINS_DIR
+#           define PLUGINS_DIR "/lib/codelite"
+#       endif
+#       ifndef INSTALL_DIR
+#           define INSTALL_DIR "/share/codelite"
+#       endif
+#       ifndef CFGDIR
+#           define CFGDIR "/share/codelite/config/cfg"
+#       endif
+#   endif
      // MSW / OSX and the rest of the world
 #    define CL_USE_NATIVEBOOK 0
 #endif
diff -Naur codelite-orig/Interfaces/plugin.h codelite-mingw/Interfaces/plugin.h
--- codelite-orig/Interfaces/plugin.h	2014-09-04 18:16:54.259200000 +0400
+++ codelite-mingw/Interfaces/plugin.h	2014-09-04 18:56:24.210000000 +0400
@@ -152,7 +152,11 @@
 #ifdef __WXGTK__
         wxString pluginsDir(PLUGINS_DIR, wxConvUTF8);
 #else
+#   ifdef __MINGW64_VERSION_MAJOR
+        wxString pluginsDir(wxStandardPaths::Get().GetDataDir() + wxT( PLUGINS_DIR ));
+#   else
         wxString pluginsDir(m_mgr->GetInstallDirectory() + wxT( "/plugins" ));
+#   endif
 #endif
         wxString basePath(pluginsDir + wxT("/resources/"));
 
diff -Naur codelite-orig/LiteEditor/app.cpp codelite-mingw/LiteEditor/app.cpp
--- codelite-orig/LiteEditor/app.cpp	2014-09-04 18:16:54.384000000 +0400
+++ codelite-mingw/LiteEditor/app.cpp	2014-09-04 19:29:41.446600000 +0400
@@ -314,6 +314,10 @@
 // Enable this process debugging priviliges
 // EnableDebugPriv();
 #endif
+
+#ifdef __MINGW64_VERSION_MAJOR
+    wxStandardPaths::Get().IgnoreAppSubDir("bin");
+#endif
 
     // Init resources and add the PNG handler
     wxSystemOptions::SetOption(_T("msw.remap"), 0);
@@ -463,7 +463,11 @@
 
 #else //__WXMSW__
     if(homeDir.IsEmpty()) { // did we got a basedir from user?
+#  ifdef __MINGW64_VERSION_MAJOR
+        homeDir = wxStandardPaths::Get().GetDataDir() + INSTALL_DIR;
+#  else
         homeDir = ::wxGetCwd();
+#  endif
     }
     wxFileName fnHomdDir(homeDir + wxT("/"));
 
@@ -598,7 +598,11 @@
         wxLocale::AddCatalogLookupPathPrefix(wxT("/usr/local/share/locale"));
 
 #elif defined(__WXMSW__)
+#   ifdef __MINGW64_VERSION_MAJOR
+        wxLocale::AddCatalogLookupPathPrefix(wxStandardPaths::Get().GetDataDir() + wxT("/share/locale"));
+#   else
         wxLocale::AddCatalogLookupPathPrefix(ManagerST::Get()->GetInstallDir() + wxT("\\locale"));
+#   endif
 #endif
 
         // This has to be done before the catalogues are added, as otherwise the wrong one (or none) will be found
@@ -883,7 +883,12 @@
 #ifdef __WXGTK__
     printersInstallDir = wxFileName(wxString(INSTALL_DIR, wxConvUTF8), "gdb_printers");
 #else
+#   ifdef __MINGW64_VERSION_MAJOR
+    wxString commdir(wxStandardPaths::Get().GetDataDir() + INSTALL_DIR);
+    printersInstallDir = wxFileName(commdir, "gdb_printers");
+#   else
     printersInstallDir = wxFileName(wxStandardPaths::Get().GetDataDir(), "gdb_printers");
+#   endif
 #endif
 
     // copy the files to ~/.codelite/gdb_printers
diff -Naur codelite-orig/LiteEditor/manager.cpp codelite-mingw/LiteEditor/manager.cpp
--- codelite-orig/LiteEditor/manager.cpp	2014-09-04 18:16:54.633600000 +0400
+++ codelite-mingw/LiteEditor/manager.cpp	2014-09-04 19:00:05.371200000 +0400
@@ -1913,7 +1913,11 @@
 #ifdef __WXGTK__
     wxString pluginsDir(PLUGINS_DIR, wxConvUTF8);
 #else
-    wxString pluginsDir(GetInstallDir() + wxT("/plugins"));
+#   ifdef __MINGW64_VERSION_MAJOR
+        wxString pluginsDir(GetInstallDir() + wxT( PLUGINS_DIR ));
+#   else
+        wxString pluginsDir(GetInstallDir() + wxT( "/plugins" ));
+#   endif
 #endif
 
     wxDir::GetAllFiles(pluginsDir + wxT("/resources/"), &files, wxT("*.accelerators"), wxDIR_FILES);
@@ -1941,7 +1945,11 @@
 #ifdef __WXGTK__
         wxString pluginsDir = wxString::From8BitData(PLUGINS_DIR);
 #else
-        wxString pluginsDir(GetInstallDir() + wxT("/plugins"));
+#   ifdef __MINGW64_VERSION_MAJOR
+        wxString pluginsDir(wxStandardPaths::Get().GetDataDir() + wxT( PLUGINS_DIR ));
+#   else
+        wxString pluginsDir(GetInstallDir() + wxT( "/plugins" ));
+#   endif
 #endif
         // append the content of all '*.accelerators' from the plugins
         // resources table
diff -Naur codelite-orig/Plugin/bitmap_loader.cpp codelite-mingw/Plugin/bitmap_loader.cpp
--- codelite-orig/Plugin/bitmap_loader.cpp	2014-09-04 18:16:54.992400000 +0400
+++ codelite-mingw/Plugin/bitmap_loader.cpp	2014-09-04 19:23:00.479800000 +0400
@@ -62,8 +62,13 @@
     wxString bitmapPath = wxString(INSTALL_DIR, wxConvUTF8);
     fn = wxFileName(bitmapPath, zipname);
 #else
+#   ifdef __MINGW64_VERSION_MAJOR
+    wxString bitmapPath(wxStandardPaths::Get().GetDataDir() + wxT( PLUGINS_DIR ));
+    fn = wxFileName(bitmapPath, zipname);
+#   else
     fn = wxFileName(wxStandardPaths::Get().GetDataDir(), zipname);
 #endif
+#endif
 
     if(m_manifest.empty() || m_toolbarsBitmaps.empty()) {
         m_zipPath = fn;
diff -Naur codelite-orig/Plugin/debuggermanager.cpp codelite-mingw/Plugin/debuggermanager.cpp
--- codelite-orig/Plugin/debuggermanager.cpp	2014-09-04 18:16:55.054800000 +0400
+++ codelite-mingw/Plugin/debuggermanager.cpp	2014-09-04 19:19:03.205800000 +0400
@@ -104,7 +104,12 @@
     wxString debuggersPath(PLUGINS_DIR, wxConvUTF8);
     debuggersPath += wxT("/debuggers");
 #else
+#   ifdef __MINGW64_VERSION_MAJOR
+    wxString debuggersPath(m_baseDir + wxT(PLUGINS_DIR));
+    debuggersPath += wxT("/debuggers");
+#   else
     wxString debuggersPath(m_baseDir + wxT("/debuggers"));
+#   endif
 #endif
 
     wxDir::GetAllFiles(debuggersPath, &files, fileSpec, wxDIR_FILES);
