From 107248e0dd32f612a8b6c87bdc63be2795091ae1 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Fri, 5 Dec 2014 23:10:22 +0300
Subject: [PATCH] Need Wx HTML module for linking.

---
 codelitephp/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/codelitephp/CMakeLists.txt b/codelitephp/CMakeLists.txt
index bbd7926..c170642 100644
--- a/codelitephp/CMakeLists.txt
+++ b/codelitephp/CMakeLists.txt
@@ -16,10 +16,10 @@ project( ${PLUGIN_NAME} )
 # It was noticed that when using MinGW gcc it is essential that 'core' is mentioned before 'base'.
 if (WITH_WEBVIEW)
     message("-- Internal Web Browser is enabled")
-    find_package(wxWidgets COMPONENTS adv aui base core xml xrc net stc webview REQUIRED)
+    find_package(wxWidgets COMPONENTS adv aui base core html xml xrc net stc webview REQUIRED)
 else()
     message("-- Internal Web Browser is disabled")
-    find_package(wxWidgets COMPONENTS adv aui base core xml xrc net stc REQUIRED)
+    find_package(wxWidgets COMPONENTS adv aui base core html xml xrc net stc REQUIRED)
 endif()
 
 # wxWidgets include (this will do all the magic to configure everything)
-- 
2.2.0

