--- codelite/CMakeLists.txt.orig	2014-09-06 01:12:26.700800000 +0400
+++ codelite/CMakeLists.txt	2014-09-06 01:23:39.548200000 +0400
@@ -98,6 +98,9 @@
              )
         endif()
         if (MINGW)
+          execute_process(COMMAND sh ${WX_TOOL} --debug=no --rescomp OUTPUT_VARIABLE WX_RC_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
+          string(REGEX REPLACE "windres" "" WX_RC_FLAGS ${WX_RC_FLAGS})
+          set (CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} ${WX_RC_FLAGS}")
           add_definitions(-D__WXMSW__)
         endif (MINGW)
     endif()
--- codelite/codelite_terminal/CMakeLists.txt.orig	2014-09-05 22:01:32.289600000 +0400
+++ codelite/codelite_terminal/CMakeLists.txt	2014-09-06 01:23:58.221400000 +0400
@@ -31,15 +31,11 @@
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
+set (RES_FILE "")
 if (MINGW)
-    add_definitions(-DWINDRES)
-    execute_process(COMMAND sh ${CL_WX_CONFIG} --debug=no --rescomp OUTPUT_VARIABLE WX_RC_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
-    string(REGEX REPLACE "windres" "" WX_RC_FLAGS ${WX_RC_FLAGS})
-    set (CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} ${WX_RC_FLAGS}")
-    add_executable(codelite-terminal ${SRCS} win_resources.rc)
-else()
-    add_executable(codelite-terminal ${SRCS})
+    set (RES_FILE "win_resources.rc")
 endif()
+add_executable(codelite-terminal ${SRCS} ${RES_FILE})
 
 set ( UTIL_LIB "" )
 if ( UNIX )
--- codelite/LiteEditor/CMakeLists.txt.orig	2014-09-05 22:01:29.466000000 +0400
+++ codelite/LiteEditor/CMakeLists.txt	2014-09-06 01:18:25.110800000 +0400
@@ -67,13 +67,12 @@
 
 FILE(GLOB SRCS "*.cpp")
 
+set (RES_FILE "")
 # Define the output
 if (MINGW)
-    add_definitions(-DWINDRES)
-    add_executable(codelite ${SRCS} code_parser.rc)
-else()
-    add_executable(codelite ${SRCS})
+    set(RES_FILES "code_parser.rc")
 endif()
+add_executable(codelite ${SRCS} ${RES_FILES})
 
 set( ADDITIONAL_LIBRARIES "" )
 
