--- pupnp-release-1.6.25/ixml/Makefile.am.orig	2018-05-29 14:34:04.960223600 +0300
+++ pupnp-release-1.6.25/ixml/Makefile.am	2018-05-29 14:34:14.507440400 +0300
@@ -22,7 +22,7 @@
 lib_LTLIBRARIES		= libixml.la
 
 libixml_la_LDFLAGS	= -version-info $(LT_VERSION_IXML) \
-			  -export-symbols-regex '^ixml.*'
+			  -export-symbols-regex '^ixml.*' -no-undefined
 
 libixml_la_SOURCES	= \
 			src/attr.c \
--- pupnp-release-1.6.25/threadutil/Makefile.am.orig	2018-05-29 14:32:55.625701500 +0300
+++ pupnp-release-1.6.25/threadutil/Makefile.am	2018-05-29 14:33:01.662912100 +0300
@@ -16,7 +16,7 @@
 
 lib_LTLIBRARIES		= libthreadutil.la
 
-libthreadutil_la_LDFLAGS = -version-info $(LT_VERSION_THREADUTIL)
+libthreadutil_la_LDFLAGS = -version-info $(LT_VERSION_THREADUTIL) -no-undefined
 
 libthreadutil_la_SOURCES = \
 			inc/FreeList.h \
--- pupnp-release-1.6.25/upnp/Makefile.am.orig	2018-05-29 14:33:10.825728400 +0300
+++ pupnp-release-1.6.25/upnp/Makefile.am	2018-05-29 14:33:52.056600800 +0300
@@ -42,7 +42,8 @@
 	-version-info $(LT_VERSION_UPNP) \
 	-export-symbols-regex '^Upnp.*' \
 	$(top_builddir)/threadutil/libthreadutil.la \
-	$(top_builddir)/ixml/libixml.la
+	$(top_builddir)/ixml/libixml.la \
+	-no-undefined -liphlpapi -lws2_32
 
 libupnp_la_SOURCES = \
 	src/inc/config.h \
--- pupnp-release-1.6.25/upnp/src/ssdp/ssdp_server.c.orig	2018-05-29 14:33:10.825728400 +0300
+++ pupnp-release-1.6.25/upnp/src/ssdp/ssdp_server.c	2018-05-29 14:33:52.056600800 +0300
@@ -41,6 +41,8 @@
 #ifndef WIN32
 	#include <sys/param.h>
 #else
+	#ifdef _MSC_VER
 	#define snprintf _snprintf
+	#endif
 #endif /* WIN32 */
 
 #include "config.h"