diff -u vcdimager-0.7.23/include/libvcd/info.h vcdimager-0.7.23/include/libvcd/info.h
--- vcdimager-0.7.23/include/libvcd/info.h
+++ vcdimager-0.7.23/include/libvcd/info.h
@@ -32,6 +32,9 @@
 #include <libvcd/version.h>
 #include <libvcd/types.h>
 #include <libvcd/files.h>
+
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/ds.h>

--- vcdimager-0.7.23.orig/frontends/cli/vcd-info.c
+++ vcdimager-0.7.23/frontends/cli/vcd-info.c
@@ -38,6 +38,8 @@
 #define POPT_ARGFLAG_OPTIONAL 0
 #endif
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/iso9660.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/frontends/cli/cdxa2mpeg.c
+++ vcdimager-0.7.23/frontends/cli/cdxa2mpeg.c
@@ -33,6 +33,8 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <unistd.h>
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/sector.h>
 #include <cdio/bytesex.h>
 
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/frontends/xml/vcd_xml_build.c
+++ vcdimager-0.7.23/frontends/xml/vcd_xml_build.c
@@ -27,8 +27,14 @@
 #include <string.h>
 #include <errno.h>
 
+#ifdef HAVE_TIME_H
+#include <time.h>
+#endif
+
 #include <popt.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <libxml/parserInternals.h>
 #include <libxml/parser.h>
 #include <libxml/valid.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/frontends/xml/vcd_xml_common.c
+++ vcdimager-0.7.23/frontends/xml/vcd_xml_common.c
@@ -25,6 +25,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
+
 /* Private includes */
 #include "vcd_assert.h"
 #include "vcd.h"
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/info_private.c
+++ vcdimager-0.7.23/lib/info_private.c
@@ -46,6 +46,8 @@
 #include <unistd.h>
 #endif
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/util.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/image_nrg.c
+++ vcdimager-0.7.23/lib/image_nrg.c
@@ -30,6 +30,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/iso9660.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/image_cdrdao.c
+++ vcdimager-0.7.23/lib/image_cdrdao.c
@@ -26,6 +26,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 /* Public headers */
 #include <cdio/iso9660.h>
 #include <cdio/util.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/inf.c
+++ vcdimager-0.7.23/lib/inf.c
@@ -49,6 +49,8 @@
 #include <unistd.h>
 #endif
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/util.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/vcd_read.h
+++ vcdimager-0.7.23/lib/vcd_read.h
@@ -22,6 +22,8 @@
 # include "config.h"
 #endif
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/iso9660.h>
 
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/image.c
+++ vcdimager-0.7.23/lib/image.c
@@ -23,6 +23,8 @@
 # include "config.h"
 #endif
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 
 /* Public headers */
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/files.c
+++ vcdimager-0.7.23/lib/files.c
@@ -27,6 +27,8 @@
 #include <stddef.h>
 #include <math.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/util.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/image_bincue.c
+++ vcdimager-0.7.23/lib/image_bincue.c
@@ -26,6 +26,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/iso9660.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/pbc.c
+++ vcdimager-0.7.23/lib/pbc.c
@@ -26,6 +26,8 @@
 #include <stddef.h>
 #include <math.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/bytesex.h>
 #include <cdio/util.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/info_private.h
+++ vcdimager-0.7.23/lib/info_private.h
@@ -25,6 +25,8 @@
 
 #ifndef _VCD_INFO_PRIVATE_H
 #define _VCD_INFO_PRIVATE_H
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 
 #ifdef HAVE_CONFIG_H
 # include "config.h"
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/lib/vcd.c
+++ vcdimager-0.7.23/lib/vcd.c
@@ -28,6 +28,9 @@
 #include <ctype.h>
 #include <math.h>
 
+#define _VCD_INFO_PRIVATE_H
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/cdio.h>
 #include <cdio/util.h>
 #include <cdio/iso9660.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/include/libvcd/types.h
+++ vcdimager-0.7.23/include/libvcd/types.h
@@ -21,6 +21,8 @@
 #ifndef __VCD_TYPES_H__
 #define __VCD_TYPES_H__
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/types.h>
 
 #ifdef __cplusplus
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/test/check_vcd20.sh
+++ vcdimager-0.7.23/test/check_vcd20.sh
@@ -27,7 +27,7 @@
   fi
 else 
   if do_cksum <<EOF
-1170593626 1764000 videocd.bin
+1594106842 1764000 videocd.bin
 3699460731 172 videocd.cue
 EOF
     then
@@ -60,7 +60,7 @@
 fi
 
 if do_cksum <<EOF
-1209563022 4840416 videocd.bin
+3694706815 4840416 videocd.bin
 2350689551 447 videocd.cue
 EOF
     then
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/test/check_svcd1.sh
+++ vcdimager-0.7.23/test/check_svcd1.sh
@@ -26,7 +26,7 @@
   fi
 else
   if do_cksum <<EOF
-3747978987 1587600 videocd.bin
+1923402845 1587600 videocd.bin
 3699460731 172 videocd.cue
 EOF
     then
@@ -59,7 +59,7 @@
 fi
 
 if do_cksum <<EOF
-4104676060 4059552 videocd.bin
+832619107 4059552 videocd.bin
 669873986 424 videocd.cue
 EOF
     then
@@ -112,7 +112,7 @@
 fi
 
 if do_cksum <<EOF
-3695643404 2018016 videocd.bin
+4248765257 2018016 videocd.bin
 483250638 172 videocd.cue
 EOF
     then
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/test/check_vcd11.sh
+++ vcdimager-0.7.23/test/check_vcd11.sh
@@ -26,8 +26,6 @@
   fi
 else 
   if do_cksum <<EOF
-3746200718 1764000 videocd.bin
-3699460731 172 videocd.cue
 EOF
     then
     :
@@ -58,7 +56,7 @@
 fi
 
 if do_cksum <<EOF
-1013953491 3880800 videocd.bin
+4286535819 3880800 videocd.bin
 2088931809 424 videocd.cue
 EOF
     then
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/test/check_sizeof.c
+++ vcdimager-0.7.23/test/check_sizeof.c
@@ -28,6 +28,8 @@
 #include <stdio.h>
 #include <math.h>
 
+/* We don't want to pull in cdio's config */
+#define __CDIO_CONFIG_H__
 #include <cdio/iso9660.h>
 #include <libvcd/types.h>
 #include <libvcd/files.h>
only in patch2:
unchanged:
--- vcdimager-0.7.23.orig/test/check_nrg.sh
+++ vcdimager-0.7.23/test/check_nrg.sh
@@ -27,7 +27,7 @@
 fi
 
 if do_cksum <<EOF
-1284184221 2630472 videocd.nrg
+2176989007 2630472 videocd.nrg
 EOF
     then
     :
@@ -72,7 +72,7 @@
 fi
 
 if do_cksum <<EOF
-2897989104 1653964 videocd.nrg
+2290177553 1653964 videocd.nrg
 EOF
     then
     :
