diff -Naur vcdimager-0.7.24-orig/example/Makefile.am vcdimager-0.7.24/example/Makefile.am
--- vcdimager-0.7.24-orig/example/Makefile.am	2005-07-16 21:50:08.000000000 +0500
+++ vcdimager-0.7.24/example/Makefile.am	2014-09-22 08:56:31.077400000 +0500
@@ -24,6 +24,6 @@
 
 INCLUDES      = -I$(top_srcdir) $(LIBVCD_CFLAGS) $(LIBCDIO_CFLAGS)
 
-info1_LDADD   = $(LIBVCDINFO_LIBS) $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
+info1_LDADD   = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
 info2_SOURCES = info2.cpp
-info2_LDADD   = $(LIBVCDINFO_LIBS) $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
+info2_LDADD   = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
diff -Naur vcdimager-0.7.24-orig/frontends/cli/Makefile.am vcdimager-0.7.24/frontends/cli/Makefile.am
--- vcdimager-0.7.24-orig/frontends/cli/Makefile.am	2004-02-12 05:57:23.000000000 +0400
+++ vcdimager-0.7.24/frontends/cli/Makefile.am	2014-09-22 08:57:20.685400000 +0500
@@ -22,13 +22,13 @@
 INCLUDES = -I$(top_srcdir) $(LIBPOPT_CFLAGS) $(LIBVCD_CFLAGS) $(LIBCDIO_CFLAGS) $(LIBISO9660_CFLAGS)
 
 vcdimager_SOURCES = vcdimager.c
-vcdimager_LDADD =  $(LIBISO9660_LIBS) $(LIBVCD_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS)
+vcdimager_LDADD =  $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS)
 
 cdxa2mpeg_SOURCES = cdxa2mpeg.c
-cdxa2mpeg_LDADD =  $(LIBISO9660_LIBS) $(LIBVCD_LIBS) $(LIBPOPT_LIBS)
+cdxa2mpeg_LDADD =  $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBPOPT_LIBS)
 
 vcd_info_SOURCES = vcd-info.c
-vcd_info_LDADD = $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBVCD_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
+vcd_info_LDADD = $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
 
 man_MANS = vcdimager.1 cdxa2mpeg.1 vcd-info.1
 
diff -Naur vcdimager-0.7.24-orig/frontends/xml/Makefile.am vcdimager-0.7.24/frontends/xml/Makefile.am
--- vcdimager-0.7.24-orig/frontends/xml/Makefile.am	2004-08-15 17:16:49.000000000 +0500
+++ vcdimager-0.7.24/frontends/xml/Makefile.am	2014-09-22 08:58:06.268600000 +0500
@@ -36,7 +36,7 @@
 
 Makefile: $(BUILT_SOURCES)
 
-vcdxbuild_LDADD = $(XML_LIBS) $(LIBVCD_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
+vcdxbuild_LDADD = $(XML_LIBS) $(LIBVCDINFO_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
 
 vcdxbuild_SOURCES = \
 	vcdxml.h \
@@ -50,7 +50,7 @@
 	vcd_xml_parse.c \
 	vcd_xml_parse.h 
 
-vcdxrip_LDADD = $(XML_LIBS) $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBVCD_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS)
+vcdxrip_LDADD = $(XML_LIBS) $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBPOPT_LIBS) $(LIBCDIO_LIBS)
 vcdxrip_SOURCES = \
 	vcd_xml_rip.c \
 	vcd_xml_common.c \
@@ -59,7 +59,7 @@
 	vcd_xml_dump.c \
 	vcd_xml_dtd.h 
 
-vcdxgen_LDADD = $(XML_LIBS) $(LIBPOPT_LIBS) $(LIBVCD_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
+vcdxgen_LDADD = $(XML_LIBS) $(LIBPOPT_LIBS) $(LIBVCDINFO_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
 vcdxgen_SOURCES = \
 	vcd_xml_gen.c \
 	vcd_xml_common.c \
@@ -68,7 +68,7 @@
 	vcd_xml_dump.c \
 	vcd_xml_dtd.h 
 
-vcdxminfo_LDADD = $(XML_LIBS) $(LIBPOPT_LIBS) $(LIBVCD_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
+vcdxminfo_LDADD = $(XML_LIBS) $(LIBPOPT_LIBS) $(LIBVCDINFO_LIBS) $(LIBCDIO_LIBS) $(LIBISO9660_LIBS)
 vcdxminfo_SOURCES = \
 	vcd_xml_common.c \
 	vcd_xml_common.h \
diff -Naur vcdimager-0.7.24-orig/test/Makefile.am vcdimager-0.7.24/test/Makefile.am
--- vcdimager-0.7.24-orig/test/Makefile.am	2005-05-16 08:09:39.000000000 +0500
+++ vcdimager-0.7.24/test/Makefile.am	2014-09-22 08:58:40.042600000 +0500
@@ -2,13 +2,13 @@
 
 INCLUDES = -I$(top_srcdir) $(POPT_CFLAGS) $(LIBVCD_CFLAGS) $(LIBCDIO_CFLAGS)
 
-mpegscan_LDADD = $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
-mpegscan2_LDADD = $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
-testimage_LDADD = $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
-check_sizeof_LDADD = $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
-check_bitfield_LDADD = $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
-testassert_LDADD = $(LIBVCD_LIBS) $(LIBISO9660_LIBS)
-testvcd_LDADD = $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS) $(LIBVCD_LIBS)
+mpegscan_LDADD = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
+mpegscan2_LDADD = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
+testimage_LDADD = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
+check_sizeof_LDADD = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
+check_bitfield_LDADD = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
+testassert_LDADD = $(LIBVCDINFO_LIBS) $(LIBISO9660_LIBS)
+testvcd_LDADD = $(LIBISO9660_LIBS) $(LIBVCDINFO_LIBS)
 
 # make check targets
 
