diff -urN gl2ps-1.3.8-source.orig/CMakeLists.txt gl2ps-1.3.8-source/CMakeLists.txt
--- gl2ps-1.3.8-source.orig/CMakeLists.txt	2012-11-27 19:41:06.000000000 +0000
+++ gl2ps-1.3.8-source/CMakeLists.txt	2015-01-22 13:58:33.264048300 +0000
@@ -125,7 +125,7 @@
   install(TARGETS lib shared DESTINATION lib${LIB_SUFFIX})
 endif(OPENGL_FOUND)
 
-if(WIN32)
+if(WIN32 AND NOT DEFINED ENV{MSYSTEM})
   set(GL2PS_DOC .)
 else(WIN32)
   set(GL2PS_DOC share/doc/gl2ps)
