--- openldap-2.4.39/build/openldap.m4.orig	2014-01-25 17:36:15.000000000 +0400
+++ openldap-2.4.39/build/openldap.m4	2014-08-17 23:33:19.906600000 +0400
@@ -250,8 +250,8 @@
 [ol_icu=no
 AC_CHECK_HEADERS( unicode/utypes.h )
 if test $ac_cv_header_unicode_utypes_h = yes ; then
-	dnl OL_ICULIBS="-licui18n -licuuc -licudata"
-	OL_ICULIBS="-licuuc -licudata"
+	dnl OL_ICULIBS="-licuin -licuuc -licudt"
+	OL_ICULIBS="-licuuc -licudt"
 
 	AC_CACHE_CHECK([for ICU libraries], [ol_cv_lib_icu], [
 		ol_LIBS="$LIBS"
--- openldap-2.4.39/configure.orig	2014-01-25 17:36:15.000000000 +0400
+++ openldap-2.4.39/configure	2014-08-17 23:55:39.130000000 +0400
@@ -21958,7 +21958,7 @@
 done
 
 if test $ac_cv_header_unicode_utypes_h = yes ; then
-		OL_ICULIBS="-licuuc -licudata"
+		OL_ICULIBS="-licuuc -licudt"
 
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ICU libraries" >&5
 $as_echo_n "checking for ICU libraries... " >&6; }
