--- a/src/portable.cpp
+++ b/src/portable.cpp
@@ -649,14 +649,14 @@
 {
   std::ios_base::openmode mode = std::ofstream::out | std::ofstream::binary;
   if (append) mode |= std::ofstream::app;
-  return std::ofstream(fs::path(fileName.str()), mode);
+  return std::ofstream(fs::path(fileName.str()).string(), mode);
 }
 
 std::ifstream Portable::openInputStream(const QCString &fileName,bool binary, bool openAtEnd)
 {
   std::ios_base::openmode mode = std::ifstream::in | std::ifstream::binary;
   if (binary)     mode |= std::ios::binary;
   if (openAtEnd)  mode |= std::ios::ate;
-  return std::ifstream(fs::path(fileName.str()), mode);
+  return std::ifstream(fs::path(fileName.str()).string(), mode);
 }
 
