--- doxygen-Release_1_12_0/CMakeLists.txt.orig	2024-08-07 09:32:57.000000000 +0200
+++ doxygen-Release_1_12_0/CMakeLists.txt	2024-11-16 23:52:07.619291200 +0100
@@ -82,7 +82,7 @@
 endif()
 
 # use C++17 standard for compiling
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS ON)
 
@@ -170,7 +170,7 @@
 endif()
 
 # needed for JavaCC
-set(JAVA_CC_EXTRA_FLAGS "-DJAVACC_CHAR_TYPE=\"unsigned char\"")
+set(JAVA_CC_EXTRA_FLAGS "-DJAVACC_CHAR_TYPE=\"char8_t\"")
 set(CMAKE_CXX_FLAGS       "${CMAKE_CXX_FLAGS}       ${JAVA_CC_EXTRA_FLAGS}")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${JAVA_CC_EXTRA_FLAGS}")
 
--- doxygen-Release_1_12_0/src/portable.cpp.orig	2024-11-17 00:13:04.143190500 +0100
+++ doxygen-Release_1_12_0/src/portable.cpp	2024-11-17 00:13:12.551015300 +0100
@@ -660,7 +660,7 @@
   std::ios_base::openmode mode = std::ofstream::out | std::ofstream::binary;
   if (append) mode |= std::ofstream::app;
 #if defined(__clang__) && defined(__MINGW32__)
-  return std::ofstream(fs::path(fileName.str()).wstring(), mode);
+  return std::ofstream(fs::path(fileName.str()).string(), mode);
 #else
   return std::ofstream(fs::path(fileName.str()), mode);
 #endif
@@ -672,7 +672,7 @@
   if (binary)     mode |= std::ios::binary;
   if (openAtEnd)  mode |= std::ios::ate;
 #if defined(__clang__) && defined(__MINGW32__)
-  return std::ifstream(fs::path(fileName.str()).wstring(), mode);
+  return std::ifstream(fs::path(fileName.str()).string(), mode);
 #else
   return std::ifstream(fs::path(fileName.str()), mode);
 #endif
