--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -285,7 +285,7 @@
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_MINSIZEREL ${EXECUTABLE_OUTPUT_PATH})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO ${EXECUTABLE_OUTPUT_PATH})
 
-if(win_static)
+if(win_static AND MSVC)
   set(CompilerFlags
       CMAKE_CXX_FLAGS
       CMAKE_CXX_FLAGS_DEBUG
--- a/addon/doxysearch/CMakeLists.txt
+++ b/addon/doxysearch/CMakeLists.tx
@@ -1,7 +1,7 @@
 find_package(XAPIAN REQUIRED)
 find_package(ZLIB REQUIRED)
 
-if(WIN32)
+if(MSVC)
   set(EXTRA_LIBS uuid rpcrt4 ws2_32)
 elseif(static_libxapian)
   set(EXTRA_LIBS uuid)
