--- doxygen-1.9.1/addon/doxysearch/CMakeLists.txt.orig	2016-11-04 00:16:18.538177200 -0400
+++ doxygen-1.9.1/addon/doxysearch/CMakeLists.txt	2016-11-04 00:16:58.765793800 -0400
@@ -1,7 +1,7 @@
 find_package(xapian REQUIRED)
 find_package(ZLIB REQUIRED)
 
-if (WIN32)
+if (WIN32 AND NOT(MINGW))
   set(WIN_EXTRA_LIBS uuid.lib rpcrt4.lib ws2_32.lib)
 endif()
 
--- doxygen-Release_1_8_14/CMakeLists.txt.orig	2017-12-25 11:26:10.000000000 -0500
+++ doxygen-Release_1_8_14/CMakeLists.txt	2018-04-12 06:47:49.952617600 -0400
@@ -121,7 +121,7 @@ else ()
 endif()
 set(LANG_CODES ${lcodes} CACHE STRING "List of language codes for which translations should be compiled in")
 
-if (win_static)
+if (win_static AND MSVC)
     set(CompilerFlags
         CMAKE_CXX_FLAGS
         CMAKE_CXX_FLAGS_DEBUG
