--- a/src/openssl/app.c	2018-06-05 18:36:57.000000000 +0200
+++ b/src/openssl/app.c	2018-09-27 10:04:37.239822900 +0200
@@ -21,6 +21,11 @@
 
 #include <libxml/tree.h>
 
+#if defined(_WIN32)
+    #define WIN32_LEAN_AND_MEAN 
+    #include <windows.h>
+#endif
+
 #include <openssl/evp.h>
 #include <openssl/rand.h>
 #include <openssl/pem.h>
--- a/src/openssl/crypto.c	2018-06-05 18:36:57.000000000 +0200
+++ b/src/openssl/crypto.c	2018-09-27 10:09:28.736701700 +0200
@@ -18,6 +18,11 @@
 
 #include <string.h>
 
+#if defined(_WIN32)
+    #define WIN32_LEAN_AND_MEAN 
+    #include <windows.h>
+#endif
+
 #include <xmlsec/xmlsec.h>
 #include <xmlsec/keys.h>
 
