diff --git a/build/cmake/aom_install.cmake b/build/cmake/aom_install.cmake
index cd40fe4245..71d625fb27 100644
--- a/build/cmake/aom_install.cmake
+++ b/build/cmake/aom_install.cmake
@@ -86,7 +86,8 @@ macro(setup_aom_install_targets)
       FILES "${AOM_PKG_CONFIG_FILE}"
       DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig")
     install(TARGETS ${AOM_INSTALL_LIBS} DESTINATION
-                    "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+                    "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+                    RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}")
 
     if(ENABLE_EXAMPLES)
       install(TARGETS ${AOM_INSTALL_BINS} DESTINATION
