diff -Naur Python-3.8.0-orig/Modules/Setup.config.in Python-3.8.0/Modules/Setup.config.in
--- Python-3.8.0-orig/Modules/Setup.config.in	2019-10-22 10:01:08.018156200 +0300
+++ Python-3.8.0/Modules/Setup.config.in	2019-10-22 10:02:53.303741100 +0300
@@ -11,4 +11,5 @@
 
 # build-in modules for windows platform:
 @USE_WIN32_MODULE@winreg ../PC/winreg.c
+@USE_WIN32_MODULE@msvcrt -DPy_BUILD_CORE ../PC/msvcrtmodule.c
 
