diff -Naur Python-3.8.0-orig/Modules/Setup Python-3.8.0/Modules/Setup
--- Python-3.8.0-orig/Modules/Setup	2019-10-22 10:01:07.612555400 +0300
+++ Python-3.8.0/Modules/Setup	2019-10-22 10:04:29.009909200 +0300
@@ -111,7 +111,7 @@
 _abc _abc.c				# Abstract base classes
 itertools itertoolsmodule.c		# Functions creating iterators for efficient looping
 atexit atexitmodule.c			# Register functions to be run at interpreter-shutdown
-_signal -DPy_BUILD_CORE_BUILTIN -I$(srcdir)/Include/internal signalmodule.c
+_signal -DPy_BUILD_CORE_BUILTIN -I$(srcdir)/Include/internal signalmodule.c -lws2_32
 _stat _stat.c				# stat.h interface
 time -DPy_BUILD_CORE_BUILTIN -I$(srcdir)/Include/internal timemodule.c	# -lm # time operations and variables
 _thread -DPy_BUILD_CORE_BUILTIN -I$(srcdir)/Include/internal _threadmodule.c	# low-level threading interface
