--- yelp-tools-3.12.1/tools/yelp.m4.orig	2014-02-02 23:28:10.000000000 +0400
+++ yelp-tools-3.12.1/tools/yelp.m4	2014-07-22 18:58:17.925000000 +0400
@@ -179,8 +179,8 @@
 	      $(INSTALL_DATA) "$$d$$lc/$$f" "$$helpdir$$f" || exit 1; \
 	    elif test "x$$lc" != "xC"; then \
 	      if test "x$(YELP_LC_MEDIA_LINKS)" != "x"; then \
-	        echo "$(LN_S) -f $(HELP_DIR)/C/$(HELP_ID)/$$f $$helpdir$$f"; \
-	        $(LN_S) -f "$(HELP_DIR)/C/$(HELP_ID)/$$f" "$$helpdir$$f" || exit 1; \
+	        echo "$(INSTALL_DATA) $$d/C/$$f $$helpdir$$f"; \
+	        $(INSTALL_DATA) "$$d/C/$$f" "$$helpdir$$f" || exit 1; \
 	      fi; \
 	    fi; \
 	  done; \
