diff --git a/install-sh b/install-sh
index 28d8bf7..6e3aa6c 100644
--- a/install-sh
+++ b/install-sh
@@ -41,6 +41,8 @@
 # This script is compatible with the BSD install script, but was written
 # from scratch.

+dir=`pwd`
+
 tab='	'
 nl='
 '
@@ -262,6 +264,7 @@
     # if $src (and thus $dsttmp) contains '*'.
     if test ! -f "$src" && test ! -d "$src"; then
       echo "$0: $src does not exist." >&2
+      echo "we on $PWD" >&2
       exit 1
     fi


