diff --git a/parse.c b/parse.c
index a996c49..851e4db 100644
--- a/parse.c
+++ b/parse.c
@@ -2279,6 +2279,18 @@ Parse_include_file(char *file, Boolean isSystem, Boolean depinc, int silent)
     /* Actually open the file... */
+#if (defined _WIN32 && ! defined __CYGWIN__)
+    char *error;
     fd = open(fullname, O_RDONLY);
     if (fd == -1) {
-	if (!silent)
+        fd = open(Cmd_Exec(getWindowsPathCmd(fullname), &error), O_RDONLY);
+    }
+#else
+    fd = open(fullname, O_RDONLY);
+#endif
+    if (fd == -1) {
+	if (!silent) {
 	    Parse_Error(PARSE_FATAL, "Cannot open %s", fullname);
+#if (defined _WIN32 && ! defined __CYGWIN__)
+        Parse_Error(PARSE_FATAL, "Cannot open %s", Cmd_Exec(getWindowsPathCmd(fullname), &error));
+#endif
+    }
 	free(fullname);
diff --git a/util.c b/util.c
index 25f4f5e..7fd2768 100644
--- a/util.c
+++ b/util.c
@@ -3032,2 +3032,11 @@ char * getUnixPathCmd(const char *path)
 
+char * getWindowsPathCmd(const char *path)
+{
+    return str_concat(
+            "cygpath -m", str_concat("\"",
+                            str_concat(path, "\"", 0),
+                            0),
+                            STR_ADDSPACE);
+}
+
 #endif
