--- a/suff.c
+++ b/suff.c
@@ -2623,7 +2623,7 @@ SuffPrintSuff(void *sp, void *dummy MAKE_ATTR_UNUSED)
     if (flags) {
 	fputs(" (", debug_file);
 	while (flags) {
-	    flag = 1 << (ffs(flags) - 1);
+	    flag = 1 << (__builtin_ffs(flags) - 1);
 	    flags &= ~flag;
 	    switch (flag) {
 		case SUFF_NULL:
--- a/targ.c
+++ b/targ.c
@@ -575,7 +575,7 @@ Targ_PrintType(int type)
     type &= ~OP_OPMASK;
 
     while (type) {
-	tbit = 1 << (ffs(type) - 1);
+	tbit = 1 << (__builtin_ffs(type) - 1);
 	type &= ~tbit;
 
 	switch(tbit) {
