diff -bur ZenLib-orig/Project/CMake/CMakeLists.txt ZenLib/Project/CMake/CMakeLists.txt
--- ZenLib-orig/Project/CMake/CMakeLists.txt	2024-03-02 17:24:27.110167300 -0700
+++ ZenLib/Project/CMake/CMakeLists.txt	2024-03-02 17:24:39.971893100 -0700
@@ -7,12 +7,6 @@
 set(ZenLib_PATCH_VERSION 41)
 set(ZenLib_VERSION ${ZenLib_MAJOR_VERSION}.${ZenLib_MINOR_VERSION}.${ZenLib_PATCH_VERSION})
 
-if(WIN32)
-  set(BUILD_SHARED_LIBS OFF)
-else()
-  option(BUILD_SHARED_LIBS "Build shared libs" ON)
-endif()
-
 option(ENABLE_UNICODE "Enable unicode support" ON)
 option(LARGE_FILES "Enable large files support" ON)
 