diff -bur ZenLib-orig/Project/CMake/libzen.pc.in ZenLib/Project/CMake/libzen.pc.in
--- ZenLib-orig/Project/CMake/libzen.pc.in	2024-03-02 20:23:51.431436800 -0700
+++ ZenLib/Project/CMake/libzen.pc.in	2024-03-02 20:24:50.708422800 -0700
@@ -1,9 +1,9 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=@LIB_INSTALL_DIR@
-includedir=@INCLUDE_INSTALL_DIR@
+libdir=${prefix}/@LIB_INSTALL_DIR@
+includedir=${prefix}/@INCLUDE_INSTALL_DIR@
 Unicode=@ZenLib_Unicode@
-Libs_Static=@LIB_INSTALL_DIR@/libzen.a -lpthread
+Libs_Static=${prefix}/@LIB_INSTALL_DIR@/libzen.a -lpthread
 
 Name: libzen
 Version: @ZenLib_VERSION@
