From af18060d521baf86c219d66a26631decb3b28e79 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Yannick=20P=C3=89ROUX?= <yannick.peroux@gmail.com>
Date: Tue, 15 Jul 2025 19:36:55 +0200
Subject: [PATCH] feat: add support for cachetools 6.0 (#1773)

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 3874354fd..20f79ce66 100644
--- a/setup.py
+++ b/setup.py
@@ -20,7 +20,7 @@
 
 
 DEPENDENCIES = (
-    "cachetools>=2.0.0,<6.0",
+    "cachetools>=2.0.0,<7.0",
     "pyasn1-modules>=0.2.1",
     # rsa==4.5 is the last version to support 2.7
     # https://github.com/sybrenstuvel/python-rsa/issues/152#issuecomment-643470233
