--- eigen-eigen-1306d75b4a21/eigen3.pc.in.orig	2014-08-04 13:51:54.000000000 +0300
+++ eigen-eigen-1306d75b4a21/eigen3.pc.in	2014-12-21 00:44:50.850000000 +0300
@@ -1,10 +1,12 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/@INCLUDE_INSTALL_DIR@
 
 Name: Eigen3
 Description: A C++ template library for linear algebra: vectors, matrices, and related algorithms
 Requires:
 Version: @EIGEN_VERSION_NUMBER@
 Libs:
-Cflags: -I${prefix}/@INCLUDE_INSTALL_DIR@
+Cflags: -I${includedir}
 
 