--- Vulkan-Headers-1.2.168/include/vulkan/vk_layer.h.orig	2021-01-25 10:04:08.000000000 +0100
+++ Vulkan-Headers-1.2.168/include/vulkan/vk_layer.h	2021-03-14 16:14:12.668255800 +0100
@@ -27,7 +27,13 @@
 #pragma once
 
 #include "vulkan.h"
-#if defined(__GNUC__) && __GNUC__ >= 4
+#if defined(_WIN32)
+#  if defined (VK_BUILDING_DLL)
+#    define VK_LAYER_EXPORT __declspec(dllexport)
+#  else
+#    define VK_LAYER_EXPORT
+#  endif
+#elif defined(__GNUC__) && __GNUC__ >= 4
 #define VK_LAYER_EXPORT __attribute__((visibility("default")))
 #elif defined(__SUNPRO_C) && (__SUNPRO_C >= 0x590)
 #define VK_LAYER_EXPORT __attribute__((visibility("default")))
