--- libksba-1.3.0/src/libksba.def.orig	2012-09-26 12:32:18.000000000 +0000
+++ libksba-1.3.0/src/libksba.def	2013-10-16 14:47:54.407552300 +0000
@@ -1,3 +1,4 @@
+EXPORTS
 ;; libksba.def - Exported symbols for W32
 ;;      Copyright (C) 2007 g10 Code GmbH
 ;;
@@ -28,7 +29,6 @@
 ;; if not, see <http://www.gnu.org/licenses/>.
 
 
-EXPORTS
       ksba_check_version  @1
 
       ksba_set_hash_buffer_function @2
