--- xpra-6.0.1/setup.py.orig	2024-06-23 09:08:23.770349400 +0200
+++ xpra-6.0.1/setup.py	2024-06-23 09:09:13.399817500 +0200
@@ -22,6 +22,8 @@
 if sys.version_info<(3, 10):
     raise RuntimeError("xpra no longer supports Python versions older than 3.10")
 
+sys.path.insert(0, os.path.dirname(os.path.realpath(__file__)))
+
 try:
     from distutils.core import setup
     from distutils.command.build import build
