--- postgresql-9.4.1/src/pl/plpython/Makefile.orig	2015-02-02 23:42:55.000000000 +0300
+++ postgresql-9.4.1/src/pl/plpython/Makefile	2015-02-10 21:33:26.353400000 +0300
@@ -27,8 +27,6 @@
 # and we have to remove -lpython from the link since we are building our own
 ifeq ($(PORTNAME), win32)
 shared_libpython = yes
-python_includespec := $(subst \,/,$(python_includespec))
-override python_libspec =
 endif
 
 # If we don't have a shared library, we have to skip it.
@@ -59,26 +57,6 @@
 DATA += plpythonu.control plpythonu--1.0.sql plpythonu--unpackaged--1.0.sql
 endif
 
-
-# Python on win32 ships with import libraries only for Microsoft Visual C++,
-# which are not compatible with mingw gcc. Therefore we need to build a
-# new import library to link with.
-ifeq ($(PORTNAME), win32)
-
-pytverstr=$(subst .,,${python_version})
-PYTHONDLL=$(subst \,/,$(WINDIR))/system32/python${pytverstr}.dll
-
-OBJS += libpython${pytverstr}.a
-
-libpython${pytverstr}.a: python${pytverstr}.def
-	dlltool --dllname python${pytverstr}.dll --def python${pytverstr}.def --output-lib libpython${pytverstr}.a
-
-python${pytverstr}.def:
-	pexports $(PYTHONDLL) > $@
-
-endif # win32
-
-
 SHLIB_LINK = $(python_libspec) $(python_additional_libs) $(filter -lintl,$(LIBS))
 
 REGRESS_OPTS = --dbname=$(PL_TESTDB)
@@ -187,9 +165,6 @@
 clean distclean: clean-lib
 	rm -f $(OBJS)
 	rm -rf $(pg_regress_clean_files)
-ifeq ($(PORTNAME), win32)
-	rm -f python${pytverstr}.def
-endif
 
 else # can't build
 
