diff --git a/include/lcms2.h.orig b/include/lcms2.h
index fd4ac7c97..1fea31d28 100644
--- a/include/lcms2.h.orig
+++ b/include/lcms2.h
@@ -227,7 +227,7 @@ typedef int                  cmsBool;
 
 
 // Calling convention -- this is hardly platform and compiler dependent
-#ifdef CMS_IS_WINDOWS_
+#if defined(CMS_IS_WINDOWS_) && !defined(__GNUC__)
 #  if defined(CMS_DLL) || defined(CMS_DLL_BUILD)
 #     ifdef __BORLANDC__
 #        define CMSEXPORT       __stdcall _export
