--- asciidoc-py3/a2x.py.orig	2019-06-07 22:23:57.665795800 +0300
+++ asciidoc-py3/a2x.py	2019-06-07 22:24:02.876093800 +0300
@@ -203,7 +203,7 @@
         if not mo:
             mo = re.match(r'^\s*(?P<arg0>[^ ]+)', cmd)
         if mo.group('arg0').endswith('.py'):
-            cmd = 'python ' + cmd
+            cmd = 'python3 ' + cmd
         # Remove redundant quoting -- this is not just cosmetic,
         # quoting seems to dramatically decrease the allowed command
         # length in Windows XP.
