diff -urN --ignore-all-space a/qtbase/configure b/qtbase/configure
--- a/qtbase/configure	2014-05-02 23:27:08.981352400 +0100
+++ b/qtbase/configure	2014-05-02 23:30:09.775693300 +0100
@@ -4685,7 +4685,7 @@
     fi
     if [ "$CFG_OPENGL" = "auto" ] || [ "$CFG_OPENGL" = "yes" ]; then
         if compileTest x11/opengl "OpenGL"; then
-            CFG_OPENGL=desktop
+            CFG_OPENGL=angle
         elif compileTest unix/opengles2 "OpenGL ES 2.x"; then
             CFG_OPENGL=es2
         else
@@ -5526,6 +5526,11 @@
     QT_CONFIG="$QT_CONFIG opengles2"
 fi
 
+if [ "$CFG_OPENGL" = "angle" ]; then
+    QCONFIG_FLAGS="$QCONFIG_FLAGS QT_OPENGL_ES QT_OPENGL_ES_2"
+    QT_CONFIG="$QT_CONFIG opengles2 egl angle"
+fi
+
 if [ "$CFG_SHARED" = "yes" ]; then
     QT_CONFIG="$QT_CONFIG shared"
     QTCONFIG_CONFIG="$QTCONFIG_CONFIG shared"
@@ -6480,7 +6485,7 @@
 report_support "  NIS ...................." "$CFG_NIS"
 report_support "  OpenGL / OpenVG:"
 report_support "    EGL .................." "$CFG_EGL"
-report_support "    OpenGL ..............." "$CFG_OPENGL" yes "Desktop OpenGL" es2 "OpenGL ES 2.x"
+report_support "    OpenGL ..............." "$CFG_OPENGL" yes "Desktop OpenGL" es2 "OpenGL ES 2.x" angle "ANGLE"
 report_support "    OpenVG ..............." "$CFG_OPENVG-$CFG_OPENVG_SHIVA" yes-yes "ShivaVG" yes-no "native"
 report_support "  PCRE ..................." "$CFG_PCRE" yes "system library" qt "bundled copy"
 if [ -n "$PKG_CONFIG" ]; then
Binary files a/qtbase/configure.exe and b/qtbase/configure.exe differ
diff -urN --ignore-all-space a/qtbase/src/plugins/platforms/windows/qwindowsglcontext.cpp b/qtbase/src/plugins/platforms/windows/qwindowsglcontext.cpp
--- a/qtbase/src/plugins/platforms/windows/qwindowsglcontext.cpp	2014-05-02 23:27:11.451493700 +0100
+++ b/qtbase/src/plugins/platforms/windows/qwindowsglcontext.cpp	2014-05-02 23:28:29.312947100 +0100
@@ -52,7 +52,9 @@
 #include <algorithm>
 
 #include <wingdi.h>
+#ifndef QT_OPENGL_ES_2
 #include <GL/gl.h>
+#endif
 
 // #define DEBUG_GL
 
