diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in qt-everywhere-src-5.12.4/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in
--- qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2019-06-15 15:41:27.053868400 +0300
+++ qt-everywhere-src-5.12.4/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2019-06-15 15:41:37.583886900 +0300
@@ -214,13 +214,13 @@
     list(APPEND _Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES \"$${CMAKE_INTERFACE_QT5_MODULE_DEPS}\")
 !!ENDIF
 
+!!IF equals(TEMPLATE, aux)
+    add_library(Qt5::$${CMAKE_MODULE_NAME} INTERFACE IMPORTED)
+!!ELSE
 !!IF !isEmpty(CMAKE_STATIC_TYPE)
     add_library(Qt5::$${CMAKE_MODULE_NAME} STATIC IMPORTED)
     set_property(TARGET Qt5::$${CMAKE_MODULE_NAME} PROPERTY IMPORTED_LINK_INTERFACE_LANGUAGES "CXX")
 !!ELSE
-!!IF equals(TEMPLATE, aux)
-    add_library(Qt5::$${CMAKE_MODULE_NAME} INTERFACE IMPORTED)
-!!ELSE
     add_library(Qt5::$${CMAKE_MODULE_NAME} SHARED IMPORTED)
 !!ENDIF
 !!ENDIF
