diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in qt-everywhere-src-5.12.4/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in
--- qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2019-06-15 15:46:26.574394400 +0300
+++ qt-everywhere-src-5.12.4/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2019-06-15 15:46:37.151213000 +0300
@@ -340,9 +340,8 @@
     # all the right information.
     macro(macro_process_prl_file prl_file_location Configuration)
         if (EXISTS \"${prl_file_location}\")
-            file(STRINGS \"${prl_file_location}\" prl_strings REGEX \"QMAKE_PRL_LIBS\")
-            string(REGEX REPLACE \"QMAKE_PRL_LIBS *= *([^\\n]*)\" \"\\\\1\" static_depends ${prl_strings} )
-            string(STRIP ${static_depends} static_depends)
+            file(STRINGS \"${prl_file_location}\" prl_strings REGEX \"QMAKE_PRL_LIBS_FOR_CMAKE\")
+            string(REGEX REPLACE \"QMAKE_PRL_LIBS_FOR_CMAKE *= *([^\\n]*)\" \"\\\\1\" static_depends ${prl_strings} )
             if (_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES)
                 set(_list_sep \";\")
             endif()
