--- b/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in.before
+++ a/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in
@@ -86,6 +86,10 @@ function(_qt5_$${CMAKE_MODULE_NAME}_process_prl_file prl_file_location Configura
                     find_package(Threads REQUIRED)
                     list(APPEND _lib_deps Threads::Threads)
                 else()
+                    if(_lib MATCHES "uuid|winspool|advapi32|comdlg32|crypt32|dnsapi|dwmapi|gdi32|imm32|IPHLPAPI|kernel32|mpr|netapi32|ole32|oleaut32|shell32|user32|userenv|uxtheme|version|winmm|ws2_32")
+                        list(APPEND _link_flags ${_flag})
+                        break()
+                    endif()
                     if(_search_paths)
                         find_library(_Qt5$${CMAKE_MODULE_NAME}_${Configuration}_${_lib}_PATH ${_lib} HINTS ${_search_paths} NO_DEFAULT_PATH)
