diff -urN qt-everywhere-opensource-src-5.5.0.orig/qtbase/tools/configure/configureapp.cpp qt-everywhere-opensource-src-5.5.0/qtbase/tools/configure/configureapp.cpp
--- qt-everywhere-opensource-src-5.5.0.orig/qtbase/tools/configure/configureapp.cpp	2015-08-26 14:29:09.999124600 +0100
+++ qt-everywhere-opensource-src-5.5.0/qtbase/tools/configure/configureapp.cpp	2015-08-26 15:41:02.749516900 +0100
@@ -3448,6 +3448,9 @@
             // FIXME: add detection
             configStream << "    QMAKE_DEFAULT_LIBDIRS = /lib /usr/lib" << endl;
             configStream << "    QMAKE_DEFAULT_INCDIRS = /usr/include /usr/local/include" << endl;
+        } else {
+            configStream << "    QMAKE_DEFAULT_LIBDIRS = " << Environment::detectCompilerDefaultLibDirs() << endl;
+            configStream << "    QMAKE_DEFAULT_INCDIRS = " << Environment::detectCompilerDefaultIncDirs() << endl;
         }
         configStream << "}" << endl;
         configStream << "QT_CONFIG += " << qtConfig.join(' ') << endl;
diff -urN qt-everywhere-opensource-src-5.5.0.orig/qtbase/tools/configure/environment.cpp qt-everywhere-opensource-src-5.5.0/qtbase/tools/configure/environment.cpp
--- qt-everywhere-opensource-src-5.5.0.orig/qtbase/tools/configure/environment.cpp	2015-08-26 14:29:10.004624600 +0100
+++ qt-everywhere-opensource-src-5.5.0/qtbase/tools/configure/environment.cpp	2015-08-26 15:41:02.742516900 +0100
@@ -40,6 +40,8 @@
 #include <qfile.h>
 #include <qfileinfo.h>
 #include <qstandardpaths.h>
+#include <qtemporaryfile.h>
+#include <qregularexpression.h>
 
 #include <process.h>
 #include <errno.h>
@@ -249,6 +251,112 @@
 #endif
 };
 
+/* Returns the exit-code. */
+int Environment::compileSomething(const QString &flags, const QString &code, QString &stdOut, QString &stdErr)
+{
+    CompilerInfo *info = compilerInfo(detectCompiler());
+    int returnCode = 1;
+    if (info) {
+        QTemporaryFile tmpIn("conftestXXXXXX.cpp");
+        QString tmpOutFileName;
+        {
+            QTemporaryFile tmpOut("conftestXXXXXX.o");
+            tmpOut.open();
+            tmpOut.close();
+            tmpOutFileName = tmpOut.fileName();
+        }
+        QString tmpStdErrFileName;
+        {
+            QTemporaryFile tmpStdErr("conftestXXXXXX.stderr");
+            tmpStdErr.open();
+            tmpStdErr.close();
+            tmpStdErrFileName = tmpStdErr.fileName();
+        }
+        if (tmpIn.open()) {
+            tmpIn.write(code.toUtf8());
+            tmpIn.close();
+            QString cmd = info->executable + QString(" ") + flags
+                          + QString(" -o ") + tmpOutFileName
+                          + QString(" ") + tmpIn.fileName() + " 2>" + tmpStdErrFileName;
+            stdOut = Environment::execute(cmd, &returnCode);
+            QFile tmpStdErr(tmpStdErrFileName);
+            if (tmpStdErr.open(QIODevice::ReadOnly | QIODevice::Text)) {
+                stdErr = tmpStdErr.readAll();
+                tmpStdErr.close();
+                tmpStdErr.remove();
+            }
+            QFile::remove(tmpOutFileName);
+        }
+    }
+    return returnCode;
+}
+
+QString Environment::detectCompilerDefaultIncDirs()
+{
+    QString stdOut;
+    QString stdErr;
+    int returnCode = Environment::compileSomething("-v ", "", stdOut, stdErr);
+    QStringList result;
+    int index = stdErr.indexOf("<...>");
+    if (index != -1) {
+        index = stdErr.indexOf("\n", index);
+        int lastIndex = index;
+        while (lastIndex != -1) {
+            if (stdErr[lastIndex + 1] != ' ')
+                break;
+            lastIndex += 2;
+            index = stdErr.indexOf("\n", lastIndex + 1);
+            if (index != -1) {
+                const QString newPath(QDir::cleanPath(stdErr.mid(lastIndex, index - lastIndex)));
+                if (result.indexOf(newPath) == -1)
+                    result.append(newPath);
+            }
+            lastIndex = index;
+        }
+    }
+    Q_UNUSED(returnCode);
+    QString resultString;
+    for (int i = 0; i < result.count(); ++i) {
+        resultString.append(result[i]);
+        if ( i != result.count() - 1)
+            resultString.append(" ");
+    }
+    return resultString;
+}
+
+QString Environment::detectCompilerDefaultLibDirs()
+{
+    QString stdOut;
+    QString stdErr;
+    int returnCode = Environment::compileSomething("-v ", "", stdOut, stdErr);
+    QStringList result;
+    int index = stdErr.indexOf("LIBRARY_PATH");
+    if (index != -1) {
+        index += strlen("LIBRARY_PATH");
+        int lastIndex = index;
+        while (lastIndex != -1) {
+            if (stdErr[lastIndex + 1] == '\n')
+                break;
+            index = stdErr.indexOf(";", lastIndex + 1);
+            if (index != -1) {
+                const QString newPath(QDir::cleanPath(stdErr.mid(lastIndex + 1, index - lastIndex - 1)));
+                if (result.indexOf(newPath) == -1)
+                    result.append(newPath);
+            }
+            lastIndex = index;
+        }
+    }
+    Q_UNUSED(returnCode);
+    QString resultString;
+    for (int i = 0; i < result.count(); ++i) {
+        resultString.append(result[i]);
+        if ( i != result.count() - 1)
+            resultString.append(" ");
+    }
+    return resultString;
+}
+
+
 /*!
     Creates a commandling from \a program and it \a arguments,
     escaping characters that needs it.
diff -urN qt-everywhere-opensource-src-5.5.0.orig/qtbase/tools/configure/environment.h qt-everywhere-opensource-src-5.5.0/qtbase/tools/configure/environment.h
--- qt-everywhere-opensource-src-5.5.0.orig/qtbase/tools/configure/environment.h	2015-08-26 14:29:10.018624600 +0100
+++ qt-everywhere-opensource-src-5.5.0/qtbase/tools/configure/environment.h	2015-08-26 15:41:02.735516900 +0100
@@ -58,6 +58,10 @@
     static Compiler compilerFromQMakeSpec(const QString &qmakeSpec);
     static QString gccVersion();
 
+    static int compileSomething(const QString &flags, const QString &code, QString &stdOut, QString &stdErr);
+    static QString detectCompilerDefaultIncDirs();
+    static QString detectCompilerDefaultLibDirs();
+
     static int execute(QStringList arguments, const QStringList &additionalEnv, const QStringList &removeEnv);
     static QString execute(const QString &command, int *returnCode = 0);
     static bool cpdir(const QString &srcDir, const QString &destDir);
