--- qt-everywhere-opensource-src-5.9.1.orig/qtlocation/src/plugins/geoservices/geoservices.pro	2017-06-07 19:05:58.022981900 +0200
+++ qt-everywhere-opensource-src-5.9.1/qtlocation/src/plugins/geoservices/geoservices.pro	2017-06-07 19:00:24.921062300 +0200
@@ -6,7 +6,7 @@
     SUBDIRS += osm
 }
 
-qtConfig(opengl):qtConfig(c++14):!win32|mingw:!qnx {
+qtConfig(opengl):qtConfig(c++14):!win32:!qnx {
     !exists(../../3rdparty/mapbox-gl-native/CMakeLists.txt) {
         warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
     } else {
