--- i686/qtserialbus/src/plugins/canbus/tinycan/tinycanbackend.cpp.orig	2018-10-19 09:41:28.137042000 +0300
+++ i686/qtserialbus/src/plugins/canbus/tinycan/tinycanbackend.cpp	2018-10-19 09:44:38.613376600 +0300
@@ -89,11 +89,11 @@
 
 Q_GLOBAL_STATIC(TinyCanGlobal, gTinyCan)
 
-class WriteNotifier : public QTimer
+class WriteNotifierTiny : public QTimer
 {
     // no Q_OBJECT macro!
 public:
-    WriteNotifier(TinyCanBackendPrivate *d, QObject *parent)
+    WriteNotifierTiny(TinyCanBackendPrivate *d, QObject *parent)
         : QTimer(parent)
         , dptr(d)
     {
@@ -211,7 +211,7 @@
         }
     }
 
-    writeNotifier = new WriteNotifier(this, q);
+    writeNotifier = new WriteNotifierTiny(this, q);
     writeNotifier->setInterval(0);
 
     isOpen = true;
