--- x86_64/qtbase/mkspecs/features/toolchain.prf.orig	2019-02-08 10:53:25.885805000 +0300
+++ x86_64/qtbase/mkspecs/features/toolchain.prf	2019-02-08 10:54:30.204718000 +0300
@@ -229,7 +229,11 @@
                 line ~= s/^[ \\t]*//  # remove leading spaces
                 contains(line, "LIBRARY_PATH=.*") {
                     line ~= s/^LIBRARY_PATH=//  # remove leading LIBRARY_PATH=
-                    paths = $$split(line, $$QMAKE_DIRLIST_SEP)
+                    equals(QMAKE_HOST.os, Windows) {
+                        paths = $$split(line, ";")
+                    } else {
+                        paths = $$split(line, $$QMAKE_DIRLIST_SEP)
+                    }
                     for (path, paths): \
                         QMAKE_DEFAULT_LIBDIRS += $$clean_path($$path)
                 } else: contains(line, "Library search paths:") {
