diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/qt_configure.prf qt-everywhere-src-5.12.4/qtbase/mkspecs/features/qt_configure.prf
--- qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/qt_configure.prf	2019-06-12 23:59:14.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtbase/mkspecs/features/qt_configure.prf	2019-06-15 16:16:14.496534900 +0300
@@ -573,12 +573,12 @@
                 # Under Windows, we look only for static libraries, as even for DLLs
                 # one actually links against a static import library.
                 mingw {
-                    lcan += \
-                        # MinGW supports UNIX-style library naming in addition to
-                        # the MSVC style.
-                        lib$${lib}.dll.a lib$${lib}.a \
-                        # Fun fact: prefix-less libraries are also supported.
-                        $${lib}.dll.a $${lib}.a
+                    # prefer (import) libraries for dynamic linking unless we're doing a
+                    # static build where we solely rely on presence of static libraries
+                    shared {
+                        lcan += $${QMAKE_PREFIX_STATICLIB}$${lib}.$${QMAKE_EXTENSION_IMPORTLIB} $${lib}.$${QMAKE_EXTENSION_IMPORTLIB}
+                    }
+                    lcan += $${QMAKE_PREFIX_STATICLIB}$${lib}.$${QMAKE_EXTENSION_STATICLIB} $${lib}.$${QMAKE_EXTENSION_STATICLIB}
                 }
                 lcan += $${lib}.lib
             }
