diff --git a/qtbase/mkspecs/common/g++-win32.conf b/qtbase/mkspecs/common/g++-win32.conf
index 6369436863c..c3a1f3a3732 100644
--- a/qtbase/mkspecs/common/g++-win32.conf
+++ b/qtbase/mkspecs/common/g++-win32.conf
@@ -17,7 +17,7 @@ include(g++-base.conf)
 
 MAKEFILE_GENERATOR      = MINGW
 QMAKE_PLATFORM          = win32 mingw
-CONFIG                 += precompile_header
+CONFIG                 += debug_and_release debug_and_release_target precompile_header
 DEFINES                += UNICODE _UNICODE WIN32 MINGW_HAS_SECURE_API=1
 QMAKE_COMPILER_DEFINES += __GNUC__ _WIN32
 # can't add 'DEFINES += WIN64' and 'QMAKE_COMPILER_DEFINES += _WIN64' defines for
diff --git a/qtbase/mkspecs/features/qt_helper_lib.prf b/qtbase/mkspecs/features/qt_helper_lib.prf
index 216c24c7aaa..8a9672e6036 100644
--- a/qtbase/mkspecs/features/qt_helper_lib.prf
+++ b/qtbase/mkspecs/features/qt_helper_lib.prf
@@ -60,7 +60,7 @@ win32|CONFIG(static, static|shared) {
         "QMAKE_DEFINES_$${ucmodule} = $$val_escape(MODULE_DEFINES)"
     android {
         MODULE_PRI_CONT += "QMAKE_LIBS_$${ucmodule} ="
-    } else: debug_and_release {
+    } else: if(msvc|qtConfig(debug_and_release)): {
         win32: \
             MODULE_DEBUG_LIBS = $$DESTDIR/$$prefix$${TARGET}d.$$suffix
         else: darwin: \