--- x86_64/qtmultimedia/src/plugins/common/evr/evrdefs.h.orig	2018-08-24 09:36:46.124193600 +0300
+++ x86_64/qtmultimedia/src/plugins/common/evr/evrdefs.h	2018-08-24 09:37:45.363581900 +0300
@@ -83,14 +83,6 @@
 #define MF_E_TRANSFORM_NEED_MORE_INPUT ((HRESULT)0xC00D6D72L)
 #endif
 
-#ifdef __GNUC__
-typedef struct MFVideoNormalizedRect {
-    float left;
-    float top;
-    float right;
-    float bottom;
-} MFVideoNormalizedRect;
-#endif
 
 #include <initguid.h>
 
--- x86_64/qtmultimedia/src/plugins/directshow/directshow.pro.orig	2018-08-24 20:27:17.612908800 +0300
+++ x86_64/qtmultimedia/src/plugins/directshow/directshow.pro	2018-08-24 20:27:55.017048200 +0300
@@ -14,6 +14,7 @@
 mingw {
     DEFINES -= WINVER=0x0601 _WIN32_WINNT=0x0601
     DEFINES += NO_DSHOW_STRSAFE
+    LIBS_PRIVATE += -lamstrmid
 }
 
 include(common/common.pri)
