--- qt5-static/lib/cmake/Qt5Qml/Qt5QmlConfig.cmake	2017-02-20 11:57:02.000000000 +0000
+++ qt-patch/Qt5QmlConfig.cmake	2017-07-19 20:32:57.570795900 +0000
@@ -361,6 +361,7 @@
     endif()


+    include("${CMAKE_CURRENT_LIST_DIR}/Qt5QmlConfigExtras.cmake")
     _populate_Qml_target_properties(RELEASE "libQt5Qml.a" "" TRUE)


@@ -371,7 +372,6 @@



-    include("${CMAKE_CURRENT_LIST_DIR}/Qt5QmlConfigExtras.cmake")


 _qt5_Qml_check_file_exists("${CMAKE_CURRENT_LIST_DIR}/Qt5QmlConfigVersion.cmake")
--- qt5-static/lib/cmake/Qt5Qml/Qt5QmlConfigExtras.cmake.before
+++ qt5-static/lib/cmake/Qt5Qml/Qt5QmlConfigExtras.cmake
@@ -1,5 +1,7 @@
 file(GLOB _qt5qml_other_plugins ${CMAKE_CURRENT_LIST_DIR}/Qt5Qml_*Factory.cmake)

-foreach(_other_plugin ${_qt5qml_other_plugins})
-    include(${_other_plugin} OPTIONAL)
-endforeach()
+if (NOT Qt5Qml_PLUGINS)
+    foreach(_other_plugin ${_qt5qml_other_plugins})
+        include(${_other_plugin} OPTIONAL)
+    endforeach()
+endif()
