--- qt5-static/lib/cmake/Qt5Qml/Qt5QmlConfig.cmake	2017-02-20 11:57:02.000000000 +0000
+++ qt-patch/Qt5QmlConfig.cmake	2017-07-19 20:32:57.570795900 +0000
@@ -209,6 +209,7 @@
         endforeach()
     endif()
 
+    include("${CMAKE_CURRENT_LIST_DIR}/Qt5QmlConfigExtras.cmake")
     _populate_Qml_target_properties(RELEASE "libQt5Qml.a" "" )
 
 
@@ -218,7 +219,6 @@
 
 
 
-    include("${CMAKE_CURRENT_LIST_DIR}/Qt5QmlConfigExtras.cmake")
 
 
 _qt5_Qml_check_file_exists("${CMAKE_CURRENT_LIST_DIR}/Qt5QmlConfigVersion.cmake")
