--- a/qtbase/configure.orig	2014-03-26 17:03:41.167200000 +0400
+++ b/qtbase/configure	2014-03-26 20:36:05.889000000 +0400
@@ -3544,7 +3544,7 @@
 
 # extract from one line like LIBRARY_PATH=/one/path:/another/path:...
 $1 == "LIBRARY_PATH" {
-    libs = split($2, library_paths, ":");
+    libs = split($2, library_paths, ";");
     print "DEFAULT_LIBDIRS=\"";
     for (lib in library_paths) {
         dir = normalize(library_paths[lib]);
