--- a/qtbase/configure.orig	2012-11-09 23:14:38.635414585 +0100
+++ b/qtbase/configure	2012-11-09 23:41:48.183518371 +0100
@@ -4606,7 +4606,7 @@
     fi
     if [ "$CFG_OPENGL" = "auto" ] || [ "$CFG_OPENGL" = "yes" ]; then
         if compileTest x11/opengl "OpenGL"; then
-            CFG_OPENGL=desktop
+            CFG_OPENGL=angle
         elif compileTest unix/opengles2 "OpenGL ES 2.x"; then
             CFG_OPENGL=es2
         else
@@ -5921,6 +5921,11 @@
     QT_CONFIG="$QT_CONFIG opengles2"
 fi
 
+if [ "$CFG_OPENGL" = "angle" ]; then
+    QCONFIG_FLAGS="$QCONFIG_FLAGS QT_OPENGL_ES QT_OPENGL_ES_2"
+    QT_CONFIG="$QT_CONFIG opengles2 egl angle"
+fi
+
 if [ "$CFG_SHARED" = "yes" ]; then
     QT_CONFIG="$QT_CONFIG shared"
     QTCONFIG_CONFIG="$QTCONFIG_CONFIG shared"
@@ -6776,7 +6787,7 @@
 report_support "    IPv6 ifname .........." "$CFG_IPV6IFNAME"
 report_support "    OpenSSL .............." "$CFG_OPENSSL" yes "loading libraries at run-time" linked "linked to the libraries"
 report_support "  NIS ...................." "$CFG_NIS"
-report_support "  OpenGL ................." "$CFG_OPENGL" yes "Desktop OpenGL" es2 "OpenGL ES 2.x"
+report_support "  OpenGL ................." "$CFG_OPENGL" yes "Desktop OpenGL" es2 "OpenGL ES 2.x" angle "ANGLE"
 report_support "  OpenVG ................." "$CFG_OPENVG-$CFG_OPENVG_SHIVA" yes-yes "ShivaVG" yes-no "native"
 report_support "  PCRE ..................." "$CFG_PCRE" yes "system library" qt "bundled copy"
 if [ -n "$PKG_CONFIG" ]; then
--- a/qtbase/src/plugins/platforms/windows/qwindowsglcontext.cpp.orig	2012-12-21 01:45:08.022424992 +0100
+++ b/qtbase/src/plugins/platforms/windows/qwindowsglcontext.cpp	2012-12-21 01:52:35.773119490 +0100
@@ -50,7 +50,9 @@
 #include <algorithm>
 
 #include <wingdi.h>
+#ifndef QT_OPENGL_ES_2
 #include <GL/gl.h>
+#endif
 
 // #define DEBUG_GL
 
