--- ossia-score-3.5.3/src/plugins/score-plugin-media/CMakeLists.txt.orig	2025-07-07 15:39:37.000000000 +0200
+++ ossia-score-3.5.3/src/plugins/score-plugin-media/CMakeLists.txt	2025-09-02 08:24:05.057361900 +0200
@@ -5,7 +5,7 @@
 
 # Packages
 find_package(${QT_VERSION} REQUIRED COMPONENTS Core Widgets)
-find_package(FFmpeg COMPONENTS AVCODEC AVFORMAT AVDEVICE AVUTIL SWRESAMPLE SWSCALE POSTPROC)
+find_package(FFmpeg COMPONENTS AVCODEC AVFORMAT AVDEVICE AVUTIL SWRESAMPLE SWSCALE)
 
 if(UNIX AND NOT APPLE)
   find_package(ZLIB QUIET)
@@ -171,7 +171,7 @@
 elseif(FFmpeg_FOUND)
   target_link_libraries(${PROJECT_NAME}
     PUBLIC
-      avcodec avformat avdevice swresample swscale postproc avutil
+      avcodec avformat avdevice swresample swscale avutil
   )
 
   if(APPLE)
--- ossia-score-3.5.3/cmake/ScoreExternalAddon.developer.cmake.orig	2025-07-07 15:39:37.000000000 +0200
+++ ossia-score-3.5.3/cmake/ScoreExternalAddon.developer.cmake	2025-09-02 08:08:38.933414700 +0200
@@ -79,7 +79,7 @@
 
 find_package(Qt6 6.2 OPTIONAL_COMPONENTS ShaderTools)
 
-find_package(FFmpeg COMPONENTS AVCODEC AVFORMAT AVDEVICE AVUTIL SWRESAMPLE SWSCALE POSTPROC)
+find_package(FFmpeg COMPONENTS AVCODEC AVFORMAT AVDEVICE AVUTIL SWRESAMPLE SWSCALE)
 
 # ossia-config.hpp
 file(CONFIGURE
