--- ossia-score-3.7.1/3rdparty/libossia/src/ossia/network/value/format_value.hpp.orig	2025-09-02 08:00:00.000000000 +0000
+++ ossia-score-3.7.1/3rdparty/libossia/src/ossia/network/value/format_value.hpp	2025-09-02 08:00:00.000000000 +0000
@@ -10,7 +10,12 @@
 
 #include <fmt/ranges.h>
 FMT_BEGIN_NAMESPACE
-#if FMT_VERSION >= 100000
+#if FMT_VERSION >= 120000
+template <typename T> struct is_container_adaptor<boost::container::flat_set<T>>
+    : std::false_type {
+
+};
+#elif FMT_VERSION >= 100000
 namespace detail {
 template <typename T> class is_container_adaptor_like<boost::container::flat_set<T>>
     : public std::false_type {
