--- ossia-score-3.8.2/src/plugins/score-plugin-gfx/CMakeLists.txt.orig	2026-08-09 00:00:00.000000000 +0000
+++ ossia-score-3.8.2/src/plugins/score-plugin-gfx/CMakeLists.txt	2026-08-09 00:00:00.000000000 +0000
@@ -501,3 +501,2 @@
   add_subdirectory("${3RDPARTY_FOLDER}/Spout" "${CMAKE_CURRENT_BINARY_DIR}/Spout")
-  target_include_directories(${PROJECT_NAME} PRIVATE "${3RDPARTY_FOLDER}/D3D12MemoryAllocator")
   target_link_libraries(${PROJECT_NAME} PRIVATE Spout d3d11)
@@ -508,2 +507,3 @@
 if(WIN32)
+  target_include_directories(${PROJECT_NAME} PRIVATE "${3RDPARTY_FOLDER}/D3D12MemoryAllocator")
   target_sources(${PROJECT_NAME} PRIVATE Gfx/CameraDevice.win32.cpp)
