--- a/src/conv/wks2csv/Makefile.am
+++ b/src/conv/wks2csv/Makefile.am
@@ -16,8 +16,8 @@
 else	
 
 wks2csv_LDADD = \
-	../../lib/libwps-@WPS_MAJOR_VERSION@.@WPS_MINOR_VERSION@.la \
-	../helper/libconvHelper.la $(REVENGE_LIBS) $(REVENGE_GENERATORS_LIBS) $(REVENGE_STREAM_LIBS)  @WKS2CSV_WIN32_RESOURCE@
+	../helper/libconvHelper.la ../../lib/libwps-@WPS_MAJOR_VERSION@.@WPS_MINOR_VERSION@.la \
+	$(REVENGE_LIBS) $(REVENGE_GENERATORS_LIBS) $(REVENGE_STREAM_LIBS)  @WKS2CSV_WIN32_RESOURCE@
 
 endif
 
--- a/src/conv/wks2raw/Makefile.am
+++ b/src/conv/wks2raw/Makefile.am
@@ -16,8 +16,8 @@
 else	
 
 wks2raw_LDADD = \
-	../../lib/libwps-@WPS_MAJOR_VERSION@.@WPS_MINOR_VERSION@.la \
-	../helper/libconvHelper.la $(REVENGE_LIBS) $(REVENGE_GENERATORS_LIBS) $(REVENGE_STREAM_LIBS)  @WKS2RAW_WIN32_RESOURCE@
+	../helper/libconvHelper.la ../../lib/libwps-@WPS_MAJOR_VERSION@.@WPS_MINOR_VERSION@.la \
+	$(REVENGE_LIBS) $(REVENGE_GENERATORS_LIBS) $(REVENGE_STREAM_LIBS)  @WKS2RAW_WIN32_RESOURCE@
 
 endif
 
--- a/src/conv/wks2text/Makefile.am
+++ b/src/conv/wks2text/Makefile.am
@@ -16,8 +16,8 @@
 else	
 
 wks2text_LDADD = \
-	../../lib/libwps-@WPS_MAJOR_VERSION@.@WPS_MINOR_VERSION@.la \
-	../helper/libconvHelper.la $(REVENGE_LIBS) $(REVENGE_GENERATORS_LIBS) $(REVENGE_STREAM_LIBS)  @WKS2TEXT_WIN32_RESOURCE@
+	../helper/libconvHelper.la ../../lib/libwps-@WPS_MAJOR_VERSION@.@WPS_MINOR_VERSION@.la \
+	$(REVENGE_LIBS) $(REVENGE_GENERATORS_LIBS) $(REVENGE_STREAM_LIBS)  @WKS2TEXT_WIN32_RESOURCE@
 
 endif
 
