diff -ur libepoxy-1.5.10-orig/src/dispatch_common.h libepoxy-1.5.10/src/dispatch_common.h
--- libepoxy-1.5.10-orig/src/dispatch_common.h
+++ libepoxy-1.5.10/src/dispatch_common.h
@@ -190,6 +190,8 @@
 #if USING_DISPATCH_TABLE
 void gl_init_dispatch_table(void);
 void gl_switch_to_dispatch_table(void);
+void egl_init_dispatch_table(void);
+void egl_switch_to_dispatch_table(void);
 void wgl_init_dispatch_table(void);
 void wgl_switch_to_dispatch_table(void);
 extern uint32_t gl_tls_index, gl_tls_size;
