--- cairo-1.14.0/build/aclocal.float.m4.orig	2014-10-14 01:46:33.000000000 +0000
+++ cairo-1.14.0/build/aclocal.float.m4	2014-10-14 06:16:08.348180700 +0000
@@ -31,10 +31,10 @@
 
 ]])], [
 
-if strings - conftest | grep noonsees >/dev/null ; then
+if strings - conftest$ac_exeext | grep noonsees >/dev/null ; then
   ax_cv_c_float_words_bigendian=yes
 fi
-if strings - conftest | grep seesnoon >/dev/null ; then
+if strings - conftest$ac_exeext | grep seesnoon >/dev/null ; then
   if test "$ax_cv_c_float_words_bigendian" = unknown; then
     ax_cv_c_float_words_bigendian=no
   else
