--- cairo-1.12.14/test/any2ppm.c.orig	2013-04-03 18:21:16 +0400
+++ cairo-1.12.14/test/any2ppm.c	2013-04-03 19:02:05 +0400
@@ -79,8 +79,11 @@
 
 #include <errno.h>
 
-#if HAVE_UNISTD_H && HAVE_FCNTL_H && HAVE_SIGNAL_H && HAVE_SYS_STAT_H && HAVE_SYS_SOCKET_H && HAVE_SYS_POLL_H && HAVE_SYS_UN_H
+#if HAVE_FCNTL_H
 #include <fcntl.h>
+#endif
+
+#if HAVE_UNISTD_H && HAVE_SIGNAL_H && HAVE_SYS_STAT_H && HAVE_SYS_SOCKET_H && HAVE_SYS_POLL_H && HAVE_SYS_UN_H
 #include <signal.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
@@ -869,6 +872,10 @@
     rsvg_set_default_dpi (72.0);
 #endif
 
+#if defined(_WIN32) && !defined (__CYGWIN__)
+    _setmode (1, _O_BINARY);
+#endif
+
     if (argc == 1)
 	err = any2ppm_daemon ();
     else
