From 90d50cd92315d6760069ad8062aba5e297370b20 Mon Sep 17 00:00:00 2001
From: Adrian Johnson <ajohnson@redneon.com>
Date: Sun, 19 Jun 2016 19:52:32 +0930
Subject: win32-print: fix unbounded surface assertion

https://lists.cairographics.org/archives/cairo/2016-June/027445.html

diff --git a/src/win32/cairo-win32-printing-surface.c b/src/win32/cairo-win32-printing-surface.c
index afc0b11..983ef59 100644
--- a/src/win32/cairo-win32-printing-surface.c
+++ b/src/win32/cairo-win32-printing-surface.c
@@ -389,7 +389,7 @@ _cairo_win32_printing_surface_analyze_operation (cairo_win32_printing_surface_t
     if (pattern->type == CAIRO_PATTERN_TYPE_SURFACE) {
 	cairo_surface_pattern_t *surface_pattern = (cairo_surface_pattern_t *) pattern;
 
-	if ( _cairo_surface_is_recording (surface_pattern->surface))
+	if (surface_pattern->surface->type == CAIRO_SURFACE_TYPE_RECORDING)
 	    return CAIRO_INT_STATUS_ANALYZE_RECORDING_SURFACE_PATTERN;
     }
 
-- 
cgit v0.10.2

