--- cairo-1.17.6/meson.build.orig	2022-03-19 17:47:32.644668100 +0100
+++ cairo-1.17.6/meson.build	2022-03-19 17:47:51.992280600 +0100
@@ -512,7 +512,7 @@
   wincodec_dep = cpp_compiler.find_library('windowscodecs', required: false)
   wincodec_header = cpp_compiler.has_header('wincodec.h')
 
-  if d2d_dep.found() and dwrite_dep.found() and d2d_header and dwrite_header and wincodec_dep.found() and wincodec_header
+  if d2d_dep.found() and dwrite_dep.found() and d2d_header and dwrite_header and wincodec_dep.found() and wincodec_header and false
     feature_conf.set('CAIRO_HAS_DWRITE_FONT', 1)
     built_features += [{
       'name': 'cairo-win32-dwrite-font',
