diff --git a/src/win32/cairo-dwrite-font.cpp b/src/win32/cairo-dwrite-font.cpp
index 994889e21..8f7b6d703 100644
--- a/src/win32/cairo-dwrite-font.cpp
+++ b/src/win32/cairo-dwrite-font.cpp
@@ -1453,7 +1453,7 @@ cairo_dwrite_font_face_create_for_dwrite_fontface (IDWriteFontFace *dwrite_font_
     face->dwriteface = dwriteface;
     face->have_color = false;
     face->rendering_params = NULL;
-    face->measuring_mode = DWRITE_MEASURING_MODE_NATURAL;
+    face->measuring_mode = DWRITE_MEASURING_MODE_GDI_CLASSIC;
 
     /* Ensure IDWriteFactory4 is available before enabling color fonts */
     if (DWriteFactory::Instance4()) {
