From 862fa31ae745b9fecf8a5663a1ff1c757ca18a9d Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Sat, 4 Apr 2015 18:49:50 +0800
Subject: [PATCH] Add path to libcrt2.o into the default library search
 path

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 reloc.ml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/reloc.ml b/reloc.ml
index 5f92e10..b4b96bd 100644
--- a/reloc.ml
+++ b/reloc.ml
@@ -1059,6 +1059,7 @@ let setup_toolchain () =
       [
        Filename.dirname (get_output1 (!gcc ^ " -print-libgcc-file-name"));
        get_output1 (!gcc ^ " -print-sysroot") ^ "/mingw/lib";
+       (match !toolchain with | `MINGW -> "/mingw32" | `MINGW64 -> "/mingw64" | _ -> "") ^ "/" ^ pre ^ "/lib";
       ];
     default_libs :=
       ["-lmingw32"; "-lgcc"; "-lmoldname"; "-lmingwex"; "-lmsvcrt";
--
2.3.4
