--- qttools-everywhere-src-5.15.18/src/shared/winutils/utils.cpp.orig	2025-03-30 13:49:56.000000000 +0200
+++ qttools-everywhere-src-5.15.18/src/shared/winutils/utils.cpp	2025-11-30 10:23:07.024058000 +0100
@@ -121,12 +121,12 @@
     const QFileInfoList &dlls = directory.entryInfoList(QStringList(nameFilter), QDir::Files);
     for (const QFileInfo &dllFi : dlls) {
         const QString dllPath = dllFi.absoluteFilePath();
+        bool isMinGW = platform == WindowsDesktopMinGW || platform == WindowsDesktopClangMinGW;
         bool matches = true;
         if (debugMatchMode != MatchDebugOrRelease && (platform & WindowsBased)) {
             bool debugDll;
-            if (readPeExecutable(dllPath, &errorMessage, 0, 0, &debugDll,
-                                 (platform == WindowsDesktopMinGW))) {
-                matches = debugDll == (debugMatchMode == MatchDebug);
+            if (readPeExecutable(dllPath, &errorMessage, 0, 0, &debugDll, isMinGW)) {
+                matches = debugDll == (debugMatchMode == MatchDebug) || isMinGW;
             } else {
                 std::wcerr << "Warning: Unable to read " << QDir::toNativeSeparators(dllPath)
                            << ": " << errorMessage;
