diff -bur scribus-1.6.5-orig/scribus/scconfig.h scribus-1.6.5/scribus/scconfig.h
--- scribus-1.6.5-orig/scribus/scconfig.h	2025-12-27 23:20:13.902474500 -0700
+++ scribus-1.6.5/scribus/scconfig.h	2025-12-27 23:20:59.839239700 -0700
@@ -17,12 +17,6 @@
  */
 
 /* Include win32 specific config instead of portable one if appropriate */
-#ifdef _MSC_VER
- #if (_MSC_VER >= 1200)
-  #include "win-config.h"
- #endif
-#else
- #include "config.h"
-#endif
+#include "win-config.h"
 
 #endif
