diff -bur scribus-1.6.5-orig/scribus/main_win32.cpp scribus-1.6.5/scribus/main_win32.cpp
--- scribus-1.6.5-orig/scribus/main_win32.cpp	2025-12-27 23:33:23.755262900 -0700
+++ scribus-1.6.5/scribus/main_win32.cpp	2025-12-27 23:34:06.759019600 -0700
@@ -122,7 +122,7 @@
 int mainApp(ScribusQApp& app)
 {
 	int appRetVal = 0;
-#ifndef _DEBUG
+#if !defined(_DEBUG) && defined(_MSC_VER)
 	__try
 	{
 #endif
@@ -133,7 +133,7 @@
 			if (ScribusQApp::useGUI)
 				appRetVal = ScribusQApp::exec();
 		}
-#ifndef _DEBUG
+#if !defined(_DEBUG) && defined(_MSC_VER)
 	}
 	__except( exceptionFilter(GetExceptionCode()) )
 	{
