diff -bur scribus-1.6.5-orig/scribus/scpainter.cpp scribus-1.6.5/scribus/scpainter.cpp
--- scribus-1.6.5-orig/scribus/scpainter.cpp	2025-12-27 23:56:22.898843800 -0700
+++ scribus-1.6.5/scribus/scpainter.cpp	2025-12-27 23:57:00.111854300 -0700
@@ -12,7 +12,7 @@
 #include "util_math.h"
 
 #include <cairo.h>
-#if CAIRO_HAS_FC_FONT
+#if 0
 #include <cairo-ft.h>
 #endif
 #include "text/storytext.h"
@@ -1592,7 +1592,7 @@
 
 	assert(!m_font.isNone());
 
-#if CAIRO_HAS_FC_FONT
+#if 0
 	FcPattern *pattern = FcPatternBuild(nullptr,
 										FC_FILE, FcTypeString, QFile::encodeName(m_font.fontFilePath()).data(),
 										FC_INDEX, FcTypeInteger, m_font.faceIndex(),
diff -bur scribus-1.6.5-orig/scribus/text/screenpainter.cpp scribus-1.6.5/scribus/text/screenpainter.cpp
--- scribus-1.6.5-orig/scribus/text/screenpainter.cpp	2025-12-27 23:54:06.718819500 -0700
+++ scribus-1.6.5/scribus/text/screenpainter.cpp	2025-12-27 23:54:39.351805300 -0700
@@ -6,7 +6,7 @@
  */
 
 #include <cairo.h>
-#if CAIRO_HAS_FC_FONT
+#if 0
 #include <cairo-ft.h>
 #endif
 
@@ -46,7 +46,7 @@
 {
 	bool showControls = gc.isEmpty() || (m_item->doc()->guidesPrefs().showControls &&
 										 ((gc.isSpace() && !gc.hasFlag(ScLayout_ImplicitSpace)) || gc.isControlGlyphs()));
-#if CAIRO_HAS_FC_FONT
+#if 0
 	if (m_painter->fillMode() == 1 && m_painter->maskMode() <= 0 && !showControls)
 	{
 		m_painter->save();
