diff -bur scribus-1.6.5-orig/scribus/plugins/fileloader/scribus150format/CMakeLists.txt scribus-1.6.5/scribus/plugins/fileloader/scribus150format/CMakeLists.txt
--- scribus-1.6.5-orig/scribus/plugins/fileloader/scribus150format/CMakeLists.txt	2025-12-28 02:45:06.696313500 -0700
+++ scribus-1.6.5/scribus/plugins/fileloader/scribus150format/CMakeLists.txt	2025-12-28 02:45:26.725137000 -0700
@@ -15,7 +15,7 @@
 
 add_library(${SCRIBUS_SCR150FORMAT_FL_PLUGIN} MODULE ${SCR150FORMAT_FL_PLUGIN_SOURCES})
 
-target_link_libraries(${SCRIBUS_SCR150FORMAT_FL_PLUGIN} ${EXE_NAME})
+target_link_libraries(${SCRIBUS_SCR150FORMAT_FL_PLUGIN} ${EXE_NAME} ${OSG_LIBRARY})
 
 if(WANT_PCH)
 	target_precompile_headers(${SCRIBUS_SCR150FORMAT_FL_PLUGIN} PRIVATE "../../plugins_pch.h")
diff -bur scribus-1.6.5-orig/scribus/plugins/import/pdf/CMakeLists.txt scribus-1.6.5/scribus/plugins/import/pdf/CMakeLists.txt
--- scribus-1.6.5-orig/scribus/plugins/import/pdf/CMakeLists.txt	2025-12-28 02:45:06.581989200 -0700
+++ scribus-1.6.5/scribus/plugins/import/pdf/CMakeLists.txt	2025-12-28 02:45:44.477821000 -0700
@@ -26,7 +26,7 @@
 add_library(${SCRIBUS_IMPORTPDF_PLUGIN} MODULE ${IMPORTPDF_PLUGIN_SOURCES})
 
 link_directories( /usr/local/lib )
-target_link_libraries(${SCRIBUS_IMPORTPDF_PLUGIN} ${EXE_NAME} ${poppler_LIBRARY})
+target_link_libraries(${SCRIBUS_IMPORTPDF_PLUGIN} ${EXE_NAME} ${poppler_LIBRARY} ${TIFF_LIBRARIES})
 
 if(WANT_PCH)
 	target_precompile_headers(${SCRIBUS_IMPORTPDF_PLUGIN} PRIVATE "../../plugins_pch.h")
