diff -bur scribus-1.6.5-orig/scribus/CMakeLists.txt scribus-1.6.5/scribus/CMakeLists.txt
--- scribus-1.6.5-orig/scribus/CMakeLists.txt	2025-12-28 06:49:16.947538900 -0700
+++ scribus-1.6.5/scribus/CMakeLists.txt	2025-12-28 06:49:38.823171900 -0700
@@ -752,6 +752,10 @@
 		ENABLE_EXPORTS TRUE
 	)
 	set(WIN32_ONLY_LIBS mscms.lib)
+
+	set(APP_ICON_RC "scribus.rc")
+
+	target_sources(${EXE_NAME} PRIVATE ${APP_ICON_RC})
 else()
 	set(WIN32_ONLY_LIBS)
 endif()
