diff -bur notepad-plus-plus-8.7.9-o/PowerEditor/gcc/makefile notepad-plus-plus-8.7.9/PowerEditor/gcc/makefile
--- notepad-plus-plus-8.7.9-o/PowerEditor/gcc/makefile	2025-04-09 16:31:12.313789400 -0600
+++ notepad-plus-plus-8.7.9/PowerEditor/gcc/makefile	2025-04-09 16:31:36.979775200 -0600
@@ -44,7 +44,7 @@
 
 CXXFLAGS := -include $(GCC_DIRECTORY)/gcc-fixes.h -std=c++20
 RC := $(CROSS_COMPILE)windres
-RCFLAGS :=
+RCFLAGS := --codepage=65001
 CPP_PATH := $(SCINTILLA_DIRECTORY)/include $(LEXILLA_DIRECTORY)/include
 CPP_DEFINE := UNICODE _UNICODE OEMRESOURCE NOMINMAX _WIN32_WINNT=_WIN32_WINNT_WIN7 NTDDI_VERSION=NTDDI_WIN7 TIXML_USE_STL TIXMLA_USE_STL
 LD := $(CXX)
