diff -Naur a/PowerEditor/gcc/makefile b/PowerEditor/gcc/makefile
--- a/PowerEditor/gcc/makefile	2025-04-02 18:57:02.000000000 +0200
+++ b/PowerEditor/gcc/makefile	2025-04-04 13:53:52.274389100 +0200
@@ -196,7 +196,7 @@
 RMDIR := $(SHELL_DIRECTORY)rm.exe -rf
 CP := $(SHELL_DIRECTORY)cp.exe
 RM := $(SHELL_DIRECTORY)rm.exe -f
-PREBUILD_EVENT_CMD := cmd //C "cd PowerEditor/src && NppLibsVersionH-generator.bat"
+PREBUILD_EVENT_CMD := cmd //C "cd $(POWEREDITOR_SRC_DIR) && NppLibsVersionH-generator.bat"
 normalize-path = $1
 else
 # a standard Windows system
