--- mpg123-1.15.3/configure.ac.orig	2013-04-10 19:40:22 +0400
+++ mpg123-1.15.3/configure.ac	2013-04-10 19:42:37 +0400
@@ -24,8 +24,8 @@
 AC_CANONICAL_HOST
 
 dnl Version 1.7 of automake is recommended
-AM_INIT_AUTOMAKE(1.7)
+AM_INIT_AUTOMAKE([silent-rules 1.7])
 AC_CONFIG_HEADERS([src/config.h])
 
 
 # You get strange symptoms like jack module build failing because the AC_C_CONST failed to detect the working const support.
