--- mpg123-1.15.3/configure.ac.orig	2013-04-10 19:40:22 +0400
+++ mpg123-1.15.3/configure.ac	2013-04-10 19:42:37 +0400
@@ -1747,14 +1747,14 @@
 				 -a "x${ac_cv_header_AL_alc_h}" = "xyes";
 			then #Linux
 				output_modules="$output_modules openal"
-				OPENAL_LIBS="-lopenal"
+				OPENAL_LIBS="-lOpenAL32"
 				OPENAL_CFLAGS="-DOPENAL_SUBDIR_AL"
 				HAVE_OPENAL="yes"
 			elif test "x${ac_cv_header_al_h}" = "xyes" \
 				 -a "x${ac_cv_header_alc_h}" = "xyes";
 			then #Windows?
 				output_modules="$output_modules openal"
-				OPENAL_LIBS="-lopenal"
+				OPENAL_LIBS="-lOpenAL32"
 				OPENAL_CFLAGS=""
 				HAVE_OPENAL="yes"
 			else
