From 7747019ea83ce3c12b57e6dd88aa18eaeb79770a Mon Sep 17 00:00:00 2001
From: Melven Roehrig-Zoellner <Melven.Roehrig-Zoellner@DLR.de>
Date: Thu, 4 Feb 2021 22:02:12 +0100
Subject: [PATCH 1/2] Dont build and install the examples

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 040ac71..b518178 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,10 +109,10 @@ if(TIXI_BUILD_TESTS)
 endif(TIXI_BUILD_TESTS)
 
 #demos
-add_subdirectory(examples/Demo)
-if (TIXI_ENABLE_FORTRAN)
-  add_subdirectory(examples/fortran77)
-endif(TIXI_ENABLE_FORTRAN)
+#add_subdirectory(examples/Demo)
+#if (TIXI_ENABLE_FORTRAN)
+#  add_subdirectory(examples/fortran77)
+#endif(TIXI_ENABLE_FORTRAN)
 
 # create the doc
 include(createDoc)
-- 
2.30.0.windows.2

