unchanged:
--- osl-0.9.2/Makefile.am.orig	2018-04-26 17:42:27.304069700 -0400
+++ osl-0.9.2/Makefile.am	2018-04-26 17:43:07.595866500 -0400
@@ -125,6 +125,7 @@
 #############################################################################
 
 libosl_la_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include -g
+libosl_la_LDFLAGS = -no-undefined
 libosl_la_SOURCES = \
 	source/scop.c \
 	source/statement.c \
only in patch2:
unchanged:
--- osl-0.9.2/configure.ac.orig	2018-04-26 17:50:19.682245700 -0400
+++ osl-0.9.2/configure.ac	2018-04-26 17:50:22.103526300 -0400
@@ -95,6 +95,7 @@
 AC_PROG_MAKE_SET
 AC_CHECK_PROG(CD, cd)
 AC_PROG_INSTALL
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
 AC_CHECK_PROGS(DOXYGEN, doxygen, doxygen, [])
