--- a/cvise-cli.py
+++ b/cvise-cli.py
@@ -19,7 +19,7 @@
 # add the known install location to the path
 destdir = os.getenv('DESTDIR', '')
 if importlib.util.find_spec('cvise') is None:
-    sys.path.append('@CMAKE_INSTALL_FULL_DATADIR@')
+    sys.path.append(os.path.join(sys.prefix, 'share'))
     sys.path.append(destdir + '@CMAKE_INSTALL_FULL_DATADIR@')
 
 import chardet  # noqa: E402
@@ -50,7 +50,7 @@
 def get_share_dir():
     # Test all known locations for the cvise directory
     share_dirs = [
-        os.path.join('@CMAKE_INSTALL_FULL_DATADIR@', '@cvise_PACKAGE@'),
+        os.path.join(os.path.join(sys.prefix, 'share'), '@cvise_PACKAGE@'),
         destdir + os.path.join('@CMAKE_INSTALL_FULL_DATADIR@', '@cvise_PACKAGE@'),
         os.path.join(script_path, '@cvise_SHARE_DIR_SUFFIX@'),
     ]
--- a/cvise/utils/externalprograms.py
+++ b/cvise/utils/externalprograms.py
@@ -31,7 +31,8 @@
             path = shutil.which(prog, path=local_folder)
 
             if not path:
-                search = os.path.join('@CMAKE_INSTALL_FULL_LIBEXECDIR@', '@cvise_PACKAGE@')
+                import sys
+                search = os.path.join(sys.prefix, 'lib', '@cvise_PACKAGE@')
                 path = shutil.which(prog, path=search)
             if not path:
                 search = DESTDIR + os.path.join('@CMAKE_INSTALL_FULL_LIBEXECDIR@', '@cvise_PACKAGE@')
@@ -44,6 +45,7 @@
             programs[prog] = path
 
     # Special case for clang-format
-    programs['clang-format'] = '@CLANG_FORMAT_PATH@'
+    import shutil
+    programs['clang-format'] = shutil.which(os.path.basename('@CLANG_FORMAT_PATH@'))
 
     return programs
