--- cvise-2.11.0/cvise-delta.py.orig	2024-11-10 10:58:12.000000000 +0100
+++ cvise-2.11.0/cvise-delta.py	2025-05-12 08:32:03.068475600 +0200
@@ -10,4 +10,6 @@
     )
     exit(0)
 
-subprocess.run(['cvise', '--pass-group=delta'] + sys.argv[1:])
+import os
+script_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'cvise')
+subprocess.run([sys.executable, script_path, '--pass-group=delta'] + sys.argv[1:])
