--- a/treesitter_delta/Main.cpp
+++ b/treesitter_delta/Main.cpp
@@ -103,7 +103,7 @@
   std::string Contents;
   for (size_t InputIndex = 0; InputIndex < InputPaths.size(); ++InputIndex) {
     const auto &InputPath = InputPaths[InputIndex];
-    if (!readFile(InputPath, Contents))
+    if (!readFile(InputPath.string(), Contents))
       continue; // The error details are logged by the function.
     auto Tree = parseFile(Contents, Parser.get());
     if (!Tree) {
