--- enca-1.14/configure.ac	2012-09-11 11:11:11.000000000 +0200
+++ patched/configure.ac	2014-01-01 18:48:45.981130518 +0100
@@ -39,13 +39,14 @@
 AC_GNU_SOURCE
 AC_AIX
 AC_ISC_POSIX
-AC_PROG_LIBTOOL
 AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PATH_PROG(MKTEMP_PROG, mktemp)
 
+LT_INIT
+
 dnl With GCC, be paranoiac.  It should compile really cleanly.
 dnl Except -Wno-sign-compare, but older gcc's don't know it.
 if test "$GCC" = yes; then
@@ -96,6 +96,13 @@
 
 dnl Check for good random number sources
 AC_CHECK_FILES(/dev/random /dev/urandom /dev/srandom /dev/arandom)
+LIBENCA_LDFLAGS=""
+case "$host_os" in
+  mingw*)
+    LIBENCA_LDFLAGS="-no-undefined ${wl}-no-undefined ${wl}-export-all-symbols" ;;
+esac
+AC_MSG_RESULT([$LIBENCA_LDFLAGS])
+AC_SUBST([LIBENCA_LDFLAGS])
 
 dnl Dirty path hack.  Helps some people with badly set up search paths.
 if test "$prefix" = "NONE"; then
--- enca-1.14/lib/Makefile.am	2009-08-25 10:33:34.000000000 +0200
+++ patched/lib/Makefile.am	2014-01-01 18:49:31.115205944 +0100
@@ -7,7 +7,7 @@
 # Any iface change    C++: 0:   A
 # Adding ifaces       C:   R:   A++
 # Changing ifaces     C:   R:   0
-libenca_la_LDFLAGS = -version-info 5:1:5
+libenca_la_LDFLAGS = @LIBENCA_LDFLAGS@ -version-info 5:1:5
 libenca_la_SOURCES = \
   common.c \
   ctype.c \
