diff -Naur icu-orig/source/config/mh-mingw icu/source/config/mh-mingw
--- icu-orig/source/config/mh-mingw	2018-03-01 10:09:20.533028500 +0300
+++ icu/source/config/mh-mingw	2018-03-01 10:09:21.141429600 +0300
@@ -10,11 +10,6 @@
 # This file is similar to mh-mingw64
 # Any changes made here may also need to be made in mh-mingw64
 
-# We install sbin tools into the same bin directory because
-# pkgdata needs some of the tools in sbin, and we can't always depend on
-# icu-config working on Windows.
-sbindir=$(bindir)
-
 ## Commands to generate dependency files
 GEN_DEPS.c=	$(CC) -E -MM $(DEFS) $(CPPFLAGS)
 GEN_DEPS.cc=	$(CXX) -E -MM $(DEFS) $(CPPFLAGS)
diff -Naur icu-orig/source/config/mh-mingw64 icu/source/config/mh-mingw64
--- icu-orig/source/config/mh-mingw64	2018-03-01 10:09:20.548628500 +0300
+++ icu/source/config/mh-mingw64	2018-03-01 10:09:21.141429600 +0300
@@ -10,11 +10,6 @@
 # This file is similar to mh-mingw
 # Any changes made here may also need to be made in mh-mingw
 
-# We install sbin tools into the same bin directory because
-# pkgdata needs some of the tools in sbin, and we can't always depend on
-# icu-config working on Windows.
-sbindir=$(bindir)
-
 ## Commands to generate dependency files
 GEN_DEPS.c=	$(CC) -E -MM $(DEFS) $(CPPFLAGS)
 GEN_DEPS.cc=	$(CXX) -E -MM $(DEFS) $(CPPFLAGS)
