--- openlibm-0.8.6/include/openlibm_fenv_aarch64.h.orig	2025-02-10 12:03:13.000000000 +0100
+++ openlibm-0.8.6/include/openlibm_fenv_aarch64.h	2025-03-10 22:38:32.629771200 +0100
@@ -36,8 +36,8 @@
 #endif
 
 /* The high 32 bits contain fpcr, low 32 contain fpsr. */
-typedef	__uint64_t	fenv_t;
-typedef	__uint64_t	fexcept_t;
+typedef	uint64_t	fenv_t;
+typedef	uint64_t	fexcept_t;
 
 /* Exception flags */
 #define	FE_INVALID	0x00000001
@@ -158,8 +158,8 @@
 __fenv_static inline int
 fegetenv(fenv_t *__envp)
 {
-	__uint64_t fpcr;
-	__uint64_t fpsr;
+	uint64_t fpcr;
+	uint64_t fpsr;
 
 	__mrs_fpcr(fpcr);
 	__mrs_fpsr(fpsr);
@@ -179,7 +179,7 @@
 	__msr_fpcr(__r);
 
 	__mrs_fpsr(__r);
-	*__envp |= (__uint32_t)__r;
+	*__envp |= (uint32_t)__r;
 	__r &= ~(_ENABLE_MASK);
 	__msr_fpsr(__r);
 	return (0);
@@ -190,7 +190,7 @@
 {
 
 	__msr_fpcr((*__envp) >> 32);
-	__msr_fpsr((fenv_t)(__uint32_t)*__envp);
+	__msr_fpsr((fenv_t)(uint32_t)*__envp);
 	return (0);
 }
 
