--- a/src/tools/idc/CMakeLists.txt
+++ b/src/tools/idc/CMakeLists.txt
@@ -13,6 +13,9 @@
          main.cpp
 )
 
+set_target_properties(idc PROPERTIES
+    OUTPUT_NAME "idc-qt6"
+)
 #### Keys ignored in scope 1:.:.:idc.pro:<TRUE>:
 # QMAKE_TARGET_DESCRIPTION = "Active Qt Interface Description Compiler"
 # _OPTION = "host_build"
--- a/tools/dumpcpp/CMakeLists.txt
+++ b/tools/dumpcpp/CMakeLists.txt
@@ -21,5 +21,8 @@
         Qt::Widgets
 )
 
+set_target_properties(dumpcpp PROPERTIES
+    OUTPUT_NAME "dumpcpp-qt6"
+)
 #### Keys ignored in scope 1:.:.:dumpcpp.pro:<TRUE>:
 # QMAKE_TARGET_DESCRIPTION = "Active Qt DumpCpp"
--- a/tools/dumpdoc/CMakeLists.txt
+++ b/tools/dumpdoc/CMakeLists.txt
@@ -16,5 +16,8 @@
         Qt::Widgets
 )
 
+set_target_properties(dumpdoc PROPERTIES
+    OUTPUT_NAME "dumpdoc-qt6"
+)
 #### Keys ignored in scope 1:.:.:dumpdoc.pro:<TRUE>:
 # QMAKE_TARGET_DESCRIPTION = "Active Qt DumpDoc"
--- a/tools/testcon/CMakeLists.txt
+++ b/tools/testcon/CMakeLists.txt
@@ -41,6 +41,9 @@
         ${testcon_resource_files}
 )
 
+set_target_properties(testcon PROPERTIES
+    OUTPUT_NAME "testcon-qt6"
+)
 
 #### Keys ignored in scope 1:.:.:testcon.pro:<TRUE>:
 # RC_FILE = "testcon.rc"
