--- leveldb-1.22/CMakeLists.txt.orig	2020-04-23 12:24:32.416255900 +0300
+++ leveldb-1.22/CMakeLists.txt	2020-04-23 12:25:20.791974600 +0300
@@ -27,6 +27,7 @@
 option(LEVELDB_BUILD_TESTS "Build LevelDB's unit tests" ON)
 option(LEVELDB_BUILD_BENCHMARKS "Build LevelDB's benchmarks" ON)
 option(LEVELDB_INSTALL "Install LevelDB's header and library" ON)
+option(LEVELDB_USE_SNAPPY "Link with snappy library" ON)
 
 include(TestBigEndian)
 test_big_endian(LEVELDB_IS_BIG_ENDIAN)
@@ -36,7 +37,9 @@
 
 include(CheckLibraryExists)
 check_library_exists(crc32c crc32c_value "" HAVE_CRC32C)
+if(LEVELDB_USE_SNAPPY)
 check_library_exists(snappy snappy_compress "" HAVE_SNAPPY)
+endif()
 check_library_exists(tcmalloc malloc "" HAVE_TCMALLOC)
 
 include(CheckCXXSymbolExists)
