--- eigen-eigen-1306d75b4a21/eigen3.pc.in.orig	2014-08-04 13:51:54.000000000 +0300
+++ eigen-eigen-1306d75b4a21/eigen3.pc.in	2014-12-21 00:44:50.850000000 +0300
@@ -1,10 +1,12 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/@INCLUDE_INSTALL_DIR@
 
 Name: Eigen3
 Description: A C++ template library for linear algebra: vectors, matrices, and related algorithms
 Requires:
 Version: @EIGEN_VERSION_NUMBER@
 Libs:
-Cflags: -I${prefix}/@INCLUDE_INSTALL_DIR@
+Cflags: -I${includedir}
 
--- eigen-eigen-1306d75b4a21/CMakeLists.txt.orig	2014-12-21 00:45:25.606600000 +0300
+++ eigen-eigen-1306d75b4a21/CMakeLists.txt	2014-12-21 00:45:50.519800000 +0300
@@ -92,9 +92,9 @@
 endif()
 
 option(EIGEN_BUILD_BTL "Build benchmark suite" OFF)
-if(NOT WIN32)
+if(NOT MSVC)
   option(EIGEN_BUILD_PKGCONFIG "Build pkg-config .pc file for Eigen" ON)
-endif(NOT WIN32)
+endif(NOT MSVC)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
@@ -304,7 +304,7 @@
     CACHE PATH "The directory relative to CMAKE_PREFIX_PATH where Eigen3Config.cmake is installed"
     )
 set(PKGCONFIG_INSTALL_DIR
-    "${CMAKE_INSTALL_DATADIR}/pkgconfig"
+    "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
     CACHE PATH "The directory relative to CMAKE_PREFIX_PATH where eigen3.pc is installed"
     )
 
