--- eigen-eigen-bdd17ee3b1b3/cmake/EigenTesting.cmake.orig	2014-08-04 13:51:54.000000000 +0300
+++ eigen-eigen-bdd17ee3b1b3/cmake/EigenTesting.cmake	2014-12-21 00:44:50.850000000 +0300
@@ -324,7 +324,7 @@
     # on all other system we rely on ${CMAKE_CXX_COMPILER}
     # supporting a "--version" or "/version" flag
     
-    if(WIN32 AND NOT CYGWIN)
+    if(WIN32 AND NOT CYGWIN AND NOT MINGW)
       set(EIGEN_CXX_FLAG_VERSION "/version")
     else()
       set(EIGEN_CXX_FLAG_VERSION "--version")
