--- eigen-eigen-1306d75b4a21/eigen3.pc.in.orig	2014-08-04 13:51:54.000000000 +0300
+++ eigen-eigen-1306d75b4a21/eigen3.pc.in	2014-12-21 00:44:50.850000000 +0300
@@ -1,6 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=@INCLUDE_INSTALL_DIR@
+includedir=@INCLUDE_INSTALL_DIR@
+
 Name: Eigen3
 Description: A C++ template library for linear algebra: vectors, matrices, and related algorithms
 Requires:
-Version: ${EIGEN_VERSION_NUMBER}
+Version: @EIGEN_VERSION_NUMBER@
 Libs:
-Cflags: -I${INCLUDE_INSTALL_DIR}
+Cflags: -I${includedir}
--- eigen-eigen-1306d75b4a21/CMakeLists.txt.orig	2014-12-21 00:45:25.606600000 +0300
+++ eigen-eigen-1306d75b4a21/CMakeLists.txt	2014-12-21 00:45:50.519800000 +0300
@@ -92,9 +92,9 @@
 endif()
 
 option(EIGEN_BUILD_BTL "Build benchmark suite" OFF)
-if(NOT WIN32)
+if(NOT MSVC)
   option(EIGEN_BUILD_PKGCONFIG "Build pkg-config .pc file for Eigen" ON)
-endif(NOT WIN32)
+endif(NOT MSVC)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
@@ -336,11 +336,11 @@
         SET(pkg_config_install_dir ${pkg_config_libdir})
         message(STATUS "found ${pkg_config_libdir}/pkgconfig" )
     else(pkg_config_libdir)
-        SET(pkg_config_install_dir ${CMAKE_INSTALL_PREFIX}/share)
+        SET(pkg_config_install_dir ${CMAKE_INSTALL_PREFIX}/lib)
         message(STATUS "pkgconfig not found; installing in ${pkg_config_install_dir}" )
     endif(pkg_config_libdir)
 
-    configure_file(eigen3.pc.in eigen3.pc)
+    configure_file(eigen3.pc.in eigen3.pc @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/eigen3.pc
         DESTINATION ${pkg_config_install_dir}/pkgconfig
         )
