diff -rupN eigen-3.1.2/eigen-eigen-5097c01bcdc4/cmake/EigenDetermineOSVersion.cmake eigen-3.1.2-new/eigen-eigen-5097c01bcdc4/cmake/EigenDetermineOSVersion.cmake
--- eigen-3.1.2/eigen-eigen-5097c01bcdc4/cmake/EigenDetermineOSVersion.cmake	2012-11-05 22:22:49.000000000 +0100
+++ eigen-3.1.2-new/eigen-eigen-5097c01bcdc4/cmake/EigenDetermineOSVersion.cmake	2013-03-01 23:03:26.732533050 +0100
@@ -26,21 +26,5 @@ function(DetermineShortWindowsName WIN_V
 endfunction()
 
 function(DetermineOSVersion OS_VERSION)
-  if (WIN32 AND CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
-    file (TO_NATIVE_PATH "$ENV{COMSPEC}" SHELL)
-    exec_program( ${SHELL} ARGS "/c" "ver" OUTPUT_VARIABLE ver_output)
-				
-      string(REGEX MATCHALL "[0-9]+"
-           ver_list "${ver_output}")
-      list(GET ver_list 0 _major)		   
-      list(GET ver_list 1 _minor)
-				
-    set(win_num_version ${_major}.${_minor})
-    DetermineShortWindowsName(win_version "${win_num_version}")
-    if(win_version)
-      set(${OS_VERSION} ${win_version} PARENT_SCOPE)
-    endif()
-  else()
-    set(${OS_VERSION} ${CMAKE_SYSTEM} PARENT_SCOPE)
-  endif()
+  set(${OS_VERSION} ${CMAKE_SYSTEM} PARENT_SCOPE)
 endfunction()
