diff -rupN eigen-eigen-ffa86ffb5570/Eigen/Core eigen-eigen-ffa86ffb5570-new/Eigen/Core
--- eigen-eigen-ffa86ffb5570/Eigen/Core	2013-07-24 03:48:35.000000000 +0200
+++ eigen-eigen-ffa86ffb5570-new/Eigen/Core	2013-08-05 22:23:54.447255489 +0200
@@ -22,7 +22,7 @@
 // Disable the ipa-cp-clone optimization flag with MinGW 6.x or newer (enabled by default with -O3)
 // See http://eigen.tuxfamily.org/bz/show_bug.cgi?id=556 for details.
 #if defined(__MINGW32__) && EIGEN_GNUC_AT_LEAST(4,6)
-  #pragma GCC optimize ("-fno-ipa-cp-clone")
+//  #pragma GCC optimize ("-fno-ipa-cp-clone")
 #endif
 
 #include <complex>
