--- a/installerfw.pri
+++ b/installerfw.pri
@@ -106,8 +106,6 @@
     $$IFW_SOURCE_TREE/src/libs/ifwtools \
     $$IFW_SOURCE_TREE/src/libs/installer
 
-CONFIG(libarchive): INCLUDEPATH += $$IFW_SOURCE_TREE/src/libs/3rdparty/libarchive
-
 CONFIG(lzmasdk) {
     INCLUDEPATH += $$IFW_SOURCE_TREE/src/libs/3rdparty/7zip
     win32:INCLUDEPATH += $$IFW_SOURCE_TREE/src/libs/3rdparty/7zip/win/CPP \
@@ -177,36 +175,7 @@
 }
 
 CONFIG(libarchive):equals(TEMPLATE, app) {
-    LIBS += -llibarchive
-    !isEmpty(IFW_ZLIB_LIBRARY) {
-        LIBS += $$IFW_ZLIB_LIBRARY
-    } else:!contains(QT_MODULES, zlib) {
-        unix:LIBS += -lz
-        win32:LIBS += -lzlib
-    }
-    !isEmpty(IFW_BZIP2_LIBRARY) {
-        LIBS += $$IFW_BZIP2_LIBRARY
-    } else {
-        unix:LIBS += -lbz2
-        win32:LIBS += -llibbz2
-    }
-    !isEmpty(IFW_LZMA_LIBRARY) {
-        LIBS += $$IFW_LZMA_LIBRARY
-    } else {
-        unix:LIBS += -llzma
-        win32:LIBS += -lliblzma
-    }
-    macos {
-        !isEmpty(IFW_ICONV_LIBRARY) {
-            LIBS += $$IFW_ICONV_LIBRARY
-        } else {
-            LIBS += -liconv
-        }
-    }
-
-    msvc:POST_TARGETDEPS += $$IFW_LIB_PATH/libarchive.lib
-    win32-g++*:POST_TARGETDEPS += $$IFW_LIB_PATH/liblibarchive.a
-    unix:POST_TARGETDEPS += $$IFW_LIB_PATH/liblibarchive.a
+    LIBS += -larchive
 }
 
 CONFIG(lzmasdk):equals(TEMPLATE, app) {
--- a/src/libs/3rdparty/3rdparty.pro
+++ b/src/libs/3rdparty/3rdparty.pro
@@ -2,9 +2,6 @@
 
 include(../../../installerfw.pri)
 
-CONFIG(libarchive) {
-    SUBDIRS += libarchive
-}
 CONFIG(lzmasdk) {
     SUBDIRS += 7zip
 }
--- a/src/libs/ifwtools/ifwtools_global.h
+++ b/src/libs/ifwtools/ifwtools_global.h
@@ -31,7 +31,7 @@
 
 #include <QtCore/QtGlobal>
 
-#ifndef QT_STATIC
+#if !defined(QT_STATIC) && !defined(Q_OS_WIN)
 #  ifdef BUILD_SHARED_IFWTOOLS
 #    define IFWTOOLS_EXPORT Q_DECL_EXPORT
 #  else
--- a/src/libs/installer/installer_global.h
+++ b/src/libs/installer/installer_global.h
@@ -30,7 +30,7 @@
 
 #include <QtCore/QtGlobal>
 
-#ifndef QT_STATIC
+#if !defined(QT_STATIC) && !defined(Q_OS_WIN)
 #  ifdef BUILD_LIB_INSTALLER
 #    define INSTALLER_EXPORT Q_DECL_EXPORT
 #  else
--- a/src/libs/kdtools/kdtoolsglobal.h
+++ b/src/libs/kdtools/kdtoolsglobal.h
@@ -31,7 +31,7 @@
 
 #include <QtCore/QtGlobal>
 
-#ifndef QT_STATIC
+#if !defined(QT_STATIC) && !defined(Q_OS_WIN)
 #  ifdef BUILD_SHARED_KDTOOLS
 #    define KDTOOLS_EXPORT Q_DECL_EXPORT
 #  else
--- a/src/sdk/sdk.pro
+++ b/src/sdk/sdk.pro
@@ -10,6 +10,7 @@
 }
 
 QT += network qml xml widgets concurrent
+greaterThan(QT_MAJOR_VERSION, 5):QT += core5compat
 # add the minimal plugin in static build to be able to start the installer headless with:
 # installer-binary --platform minimal
 # using QT += qpa_minimal_plugin would result in a minimal only compiled version
@@ -131,6 +132,7 @@
 win32 {
     # Use our own manifest file
     CONFIG -= embed_manifest_exe
+    LIBS += -lole32 -lmpr -loleaut32 -luuid
     RC_FILE = installerbase.rc
 }
 
--- a/tools/archivegen/archivegen.pro
+++ b/tools/archivegen/archivegen.pro
@@ -6,6 +6,7 @@
 
 QT -= gui
 QT += qml xml concurrent
+greaterThan(QT_MAJOR_VERSION, 5):QT += core5compat
 
 CONFIG -= import_plugins
 CONFIG(lzmasdk) {
@@ -16,6 +17,10 @@
 
 SOURCES += archive.cpp
 
+win32 {
+    LIBS += -lole32 -lmpr -loleaut32 -luuid
+}
+
 macx:include(../../no_app_bundle.pri)
 
 target.path = $$PREFIX/bin
--- a/tools/binarycreator/binarycreator.pro
+++ b/tools/binarycreator/binarycreator.pro
@@ -6,11 +6,16 @@
 
 QT -= gui
 QT += qml xml concurrent
+greaterThan(QT_MAJOR_VERSION, 5):QT += core5compat
 
 CONFIG += console
 DESTDIR = $$IFW_APP_PATH
 SOURCES = main.cpp
 
+win32 {
+    LIBS += -lole32 -lmpr -loleaut32 -luuid
+}
+
 macx:include(../../no_app_bundle.pri)
 
 target.path = $$PREFIX/bin
--- a/tools/devtool/devtool.pro
+++ b/tools/devtool/devtool.pro
@@ -2,6 +2,7 @@
 TARGET = devtool
 
 QT = core network qml xml concurrent
+greaterThan(QT_MAJOR_VERSION, 5):QT += core5compat
 
 include(../../installerfw.pri)
 
@@ -22,6 +23,10 @@
     binaryreplace.cpp \
     binarydump.cpp
 
+win32 {
+    LIBS += -lole32 -lmpr -loleaut32 -luuid
+}
+
 osx:include(../../no_app_bundle.pri)
 
 target.path = $$PREFIX/bin
--- a/tools/repogen/repogen.pro
+++ b/tools/repogen/repogen.pro
@@ -6,6 +6,7 @@
 
 QT -= gui
 QT += qml xml concurrent
+greaterThan(QT_MAJOR_VERSION, 5):QT += core5compat
 
 CONFIG -= import_plugins
 
@@ -17,6 +18,10 @@
 
 SOURCES += repogen.cpp
 
+win32 {
+    LIBS += -lole32 -lmpr -loleaut32 -luuid
+}
+
 macx:include(../../no_app_bundle.pri)
 
 target.path = $$PREFIX/bin
