diff --git a/DefineHeader.h b/DefineHeader.h
index cea0f02..321265c 100644
--- a/DefineHeader.h
+++ b/DefineHeader.h
@@ -23,10 +23,10 @@
 using namespace std;
 typedef unsigned char BYTE, *PBYTE;
 typedef unsigned char UCHAR;
-typedef unsigned short WCHAR;
+//typedef unsigned short WCHAR;
 typedef unsigned short USHORT;
 typedef unsigned int	UINT;
-typedef unsigned int	DWORD;
+//typedef unsigned int	DWORD;
 #define ALIGN(x, a)		__ALIGN_MASK((x), (a) - 1)
 #define __ALIGN_MASK(x, mask)	(((x) + (mask)) & ~(mask))
 #define RK28_SEC2_RESERVED_LEN 473
diff --git a/Makefile.am b/Makefile.am
index 1b6385d..110cc01 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,7 @@
 SUBDIRS =
 DIST_SUBDIRS = cfg
 
-AM_CPPFLAGS = -Wall -Werror -Wextra -Wreturn-type -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_LARGE_FILE $(LIBUSB1_CFLAGS)
+AM_CPPFLAGS = -Wextra -Wreturn-type -fpermissive -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_LARGE_FILE $(LIBUSB1_CFLAGS)
 
 bin_PROGRAMS = rkdeveloptool
 rkdeveloptool_SOURCES = main.cpp \
diff --git a/RKBoot.cpp b/RKBoot.cpp
index 7e73883..f58f1f5 100644
--- a/RKBoot.cpp
+++ b/RKBoot.cpp
@@ -15,11 +15,11 @@ bool CRKBoot::GetSignFlag()
 {
 	return m_bSignFlag;
 }
-DWORD CRKBoot::GetVersion()
+UINT CRKBoot::GetVersion()
 {
 	return m_version;
 }
-DWORD CRKBoot::GetMergeVersion()
+UINT CRKBoot::GetMergeVersion()
 {
 	return m_mergeVersion;
 }
diff --git a/RKImage.cpp b/RKImage.cpp
index 7774354..5595979 100644
--- a/RKImage.cpp
+++ b/RKImage.cpp
@@ -6,11 +6,11 @@
  */
 #include "RKImage.h"
 
-DWORD CRKImage::GetVersion()
+UINT CRKImage::GetVersion()
 {
 	return m_version;
 }
-DWORD CRKImage::GetMergeVersion()
+UINT CRKImage::GetMergeVersion()
 {
 	return m_mergeVersion;
 }
@@ -35,15 +35,15 @@ USHORT CRKImage::GetBackupSize()
 	pBackupSize = (USHORT *)&m_reserved[12];
 	return *pBackupSize;
 }
-DWORD CRKImage::GetBootOffset()
+UINT CRKImage::GetBootOffset()
 {
 	return m_bootOffset;
 }
-DWORD CRKImage::GetBootSize()
+UINT CRKImage::GetBootSize()
 {
 	return m_bootSize;
 }
-DWORD CRKImage::GetFWOffset()
+UINT CRKImage::GetFWOffset()
 {
 	return m_fwOffset;
 }
@@ -295,4 +295,4 @@ int CRKImage::GetMd5Data(PBYTE &lpMd5, PBYTE &lpSignMd5)
 bool CRKImage::GetSignFlag()
 {
 	return m_bSignFlag;
-}
\ No newline at end of file
+}
diff --git a/RKLog.cpp b/RKLog.cpp
index 37dc62b..9c8a50c 100644
--- a/RKLog.cpp
+++ b/RKLog.cpp
@@ -75,7 +75,8 @@ bool CRKLog::Write(string text)
 	string  strName;
 	FILE *file=NULL;
 	time(&now);
-	localtime_r(&now, &timeNow);
+	//localtime_r(&now, &timeNow);
+	localtime_s(&timeNow, &now);
 	sprintf(szDateTime, "%04d-%02d-%02d.txt", timeNow.tm_year + 1900, timeNow.tm_mon + 1, timeNow.tm_mday);
 	strName = m_path + m_name+szDateTime;
 
diff --git a/main.cpp b/main.cpp
index 72bd94b..56b54dc 100644
--- a/main.cpp
+++ b/main.cpp
@@ -3304,20 +3304,21 @@ int main(int argc, char* argv[])
 
 	g_ConfigItemVec.clear();
 	sprintf(szProgramProcPath, "/proc/%d/exe", getpid());
-	if (readlink(szProgramProcPath, szProgramDir, 256) == -1)
-		strcpy(szProgramDir, ".");
-	else {
+//	if (readlink(szProgramProcPath, szProgramDir, 256) == -1)
+//		strcpy(szProgramDir, ".");
+//	else {
 		char *pSlash;
 		pSlash = strrchr(szProgramDir, '/');
 		if (pSlash)
 			*pSlash = '\0';
-	}
+//	}
 	strLogDir = szProgramDir;
 	strLogDir +=  "/log/";
 	strConfigFile = szProgramDir;
 	strConfigFile += "/config.ini";
 	if (opendir(strLogDir.c_str()) == NULL)
-		mkdir(strLogDir.c_str(), S_IRWXU | S_IRWXG | S_IROTH);
+//		mkdir(strLogDir.c_str(), S_IRWXU | S_IRWXG | S_IROTH);
+		mkdir(strLogDir.c_str());
 	g_pLogObject = new CRKLog(strLogDir.c_str(), "log",true);
 
 	if(stat(strConfigFile.c_str(), &statBuf) < 0) {
diff --git a/rkdeveloptool.exe b/rkdeveloptool.exe
new file mode 100644
index 0000000..b482038
Binary files /dev/null and b/rkdeveloptool.exe differ
