diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a9a60c2e..d9bb0aa2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,15 +102,17 @@ if ( HLSL_SUPPORT_QUERY_GIT_COMMIT_INFO )
 endif()
 
 # adjust link option to enable debugging from kernel mode; not compatible with incremental linking
-if(NOT CMAKE_VERSION VERSION_LESS "3.13" AND WIN32 AND NOT CMAKE_C_COMPILER_ARCHITECTURE_ID STREQUAL "ARM64EC")
+if(NOT CMAKE_VERSION VERSION_LESS "3.13" AND MSVC AND NOT CMAKE_C_COMPILER_ARCHITECTURE_ID STREQUAL "ARM64EC")
   add_link_options(/DEBUGTYPE:CV,FIXUP,PDATA /INCREMENTAL:NO)
 endif()
 
 # enable control flow guard
-if(WIN32)
+if(MSVC)
 add_compile_options(/guard:cf)
 add_link_options(/guard:cf)
-endif(WIN32)
+else(MSVC)
+add_compile_options(-fcf-protection)
+endif(MSVC)
 
 # HLSL Change Ends
 
@@ -631,9 +633,9 @@ endif( ${CMAKE_SYSTEM_NAME} MATCHES SunOS )
 set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "")
 
 # enable warnings as errors for debug build
-if (WIN32)
+if (MSVC)
   SET(CMAKE_CXX_FLAGS_DEBUG  "${CMAKE_CXX_FLAGS_DEBUG} /WX")
-endif (WIN32)
+endif (MSVC)
 
 include(AddLLVM)
 include(TableGen)
@@ -744,9 +746,9 @@ endif()
 
 if (LLVM_INCLUDE_DOCS)
   add_subdirectory(docs)
+  add_hlsl_hctgen(DxilDocs OUTPUT docs/DXIL.rst CODE_TAG) # HLSL Change
 endif()
 
-add_hlsl_hctgen(DxilDocs OUTPUT docs/DXIL.rst CODE_TAG) # HLSL Change
 
 add_subdirectory(cmake/modules)
 
diff --git a/cmake/caches/PredefinedParams.cmake b/cmake/caches/PredefinedParams.cmake
index 2a2c63a58..eb259370d 100644
--- a/cmake/caches/PredefinedParams.cmake
+++ b/cmake/caches/PredefinedParams.cmake
@@ -23,19 +23,22 @@ set(CLANG_INCLUDE_TESTS ${LLVM_TEST_VALUE} CACHE BOOL "")
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL "")
 set(LLVM_APPEND_VC_REV ON CACHE BOOL "")
+set(LLVM_BUILD_TOOLS OFF CACHE BOOL "")
 set(LLVM_DEFAULT_TARGET_TRIPLE "dxil-ms-dx" CACHE STRING "")
 set(LLVM_ENABLE_EH ON CACHE BOOL "")
 set(LLVM_ENABLE_RTTI ON CACHE BOOL "")
 set(LLVM_INCLUDE_DOCS OFF CACHE BOOL "")
 set(LLVM_INCLUDE_EXAMPLES OFF CACHE BOOL "")
+set(LLVM_INCLUDE_TESTS OFF CACHE BOOL "")
 set(LLVM_OPTIMIZED_TABLEGEN OFF CACHE BOOL "")
 set(LLVM_TARGETS_TO_BUILD "None" CACHE STRING "")
 set(LIBCLANG_BUILD_STATIC ON CACHE BOOL "")
 set(CLANG_BUILD_EXAMPLES OFF CACHE BOOL "")
+set(CLANG_BUILD_TESTS OFF CACHE BOOL "")
 set(CLANG_CL OFF CACHE BOOL "")
 set(CLANG_ENABLE_ARCMT OFF CACHE BOOL "")
 set(CLANG_ENABLE_STATIC_ANALYZER OFF CACHE BOOL "")
-set(HLSL_INCLUDE_TESTS ON CACHE BOOL "")
-set(ENABLE_SPIRV_CODEGEN ON CACHE BOOL "")
-set(SPIRV_BUILD_TESTS ON CACHE BOOL "")
+set(HLSL_INCLUDE_TESTS OFF CACHE BOOL "")
+set(ENABLE_SPIRV_CODEGEN OFF CACHE BOOL "")
+set(SPIRV_BUILD_TESTS OFF CACHE BOOL "")
 set(LLVM_ENABLE_TERMINFO OFF CACHE BOOL "")
diff --git a/cmake/modules/HandleLLVMOptions.cmake b/cmake/modules/HandleLLVMOptions.cmake
index 42ab6cc91..d77e3833a 100644
--- a/cmake/modules/HandleLLVMOptions.cmake
+++ b/cmake/modules/HandleLLVMOptions.cmake
@@ -379,12 +379,20 @@ if( MSVC )
 
 elseif( LLVM_COMPILER_IS_GCC_COMPATIBLE )
   if (LLVM_ENABLE_WARNINGS)
-    append("-Wall -W -Wno-unused-parameter -Wwrite-strings -Wimplicit-fallthrough" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+    append("-Wall -W -Wno-unused-parameter -Wwrite-strings" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
     append("-Wcast-qual" CMAKE_CXX_FLAGS)
 
     # Disable unknown pragma warnings because the output is just too long with them.
     append("-Wno-unknown-pragmas" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
 
+    if (MINGW)
+      append("-Wno-gnu-zero-variadic-macro-arguments" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+      append("-Wno-missing-exception-spec" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+      append("-Wno-reorder-ctor" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+      append("-Wno-unused-const-variable" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+      append("-Wno-unused-function" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+    endif()
+
     add_flag_if_supported("-Wno-unused-but-set-variable" UNUSED_BUT_SET_VARIABLE)
     add_flag_if_supported("-Wno-deprecated-copy" DEPRECATED_COPY)
 
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 224d77974..7cf60e956 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -27,9 +27,9 @@ add_subdirectory(DXIL) # HLSL Change
 add_subdirectory(DxilContainer) # HLSL Change
 add_subdirectory(DxilPdbInfo) # HLSL Change
 add_subdirectory(DxilPIXPasses) # HLSL Change
-if(WIN32) # HLSL Change
+if(MSVC) # HLSL Change
   add_subdirectory(DxilDia) # HLSL Change
-endif(WIN32) # HLSL Change
+endif(MSVC) # HLSL Change
 add_subdirectory(DxilRootSignature) # HLSL Change
 add_subdirectory(DxcBindingTable) # HLSL Change
 add_subdirectory(DxrFallback) # HLSL Change
diff --git a/lib/Support/CMakeLists.txt b/lib/Support/CMakeLists.txt
index 396878aa2..a1627a06c 100644
--- a/lib/Support/CMakeLists.txt
+++ b/lib/Support/CMakeLists.txt
@@ -145,8 +145,8 @@ add_llvm_library(LLVMSupport
   )
 
 set_property(TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS "${system_libs}")
-if(WIN32)
+if(MSVC)
 set_property(TARGET LLVMSupport PROPERTY COMPILE_FLAGS /EHsc )
-endif(WIN32)
+endif(MSVC)
 
 target_link_libraries(LLVMSupport PUBLIC LLVMMSSupport) # HLSL Change
diff --git a/projects/CMakeLists.txt b/projects/CMakeLists.txt
index d3093985a..b879db204 100644
--- a/projects/CMakeLists.txt
+++ b/projects/CMakeLists.txt
@@ -1,7 +1,7 @@
 set(DXC_PROJECTS_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(DXC_PROJECTS_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 
-if(WIN32 AND HLSL_BUILD_DXILCONV)
+if(MSVC AND HLSL_BUILD_DXILCONV)
   add_subdirectory(include/Tracing)
   add_subdirectory(dxilconv)
-endif (WIN32 AND HLSL_BUILD_DXILCONV)
+endif (MSVC AND HLSL_BUILD_DXILCONV)
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 2c96814c4..78f0f35f6 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -21,10 +21,10 @@ add_llvm_tool_subdirectory(llvm-dis) # HLSL Change
 # add_llvm_tool_subdirectory(llvm-mc) # HLSL Change
 
 # HLSL Change Begins
-if (WIN32)
+if (MSVC)
 # This target can currently only be built on Windows.
 add_llvm_tool_subdirectory(dxexp)
-endif (WIN32)
+endif (MSVC)
 # HLSL Change ends
 
 # add_llvm_tool_subdirectory(llc) # HLSL Change
diff --git a/tools/clang/lib/Sema/CMakeLists.txt b/tools/clang/lib/Sema/CMakeLists.txt
index 3fd53759e..f5b3367ca 100644
--- a/tools/clang/lib/Sema/CMakeLists.txt
+++ b/tools/clang/lib/Sema/CMakeLists.txt
@@ -70,7 +70,9 @@ add_clang_library(clangSema
 # HLSL Change Begin
 # Sema got too big for debug builds on arm64ec, which means it will hit
 # other targets too eventually.
-if(WIN32)
+if(MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
-endif(WIN32)
+else(MSVC)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")
+endif(MSVC)
 # HLSL Change End
diff --git a/tools/clang/tools/CMakeLists.txt b/tools/clang/tools/CMakeLists.txt
index 383a3a3fc..2b50c4506 100644
--- a/tools/clang/tools/CMakeLists.txt
+++ b/tools/clang/tools/CMakeLists.txt
@@ -28,7 +28,7 @@ add_subdirectory(dxclib)
 add_subdirectory(dxc)
 
 # These targets can currently only be built on Windows.
-if (WIN32)
+if (MSVC)
 add_subdirectory(d3dcomp)
 add_subdirectory(dxrfallbackcompiler)
 add_subdirectory(dxa)
@@ -39,5 +39,5 @@ add_subdirectory(dxv)
 add_subdirectory(dxlib-sample)
 # UI powered by .NET.
 add_subdirectory(dotnetc)
-endif (WIN32)
+endif (MSVC)
 # HLSL Change Ends
diff --git a/tools/clang/tools/dxc/CMakeLists.txt b/tools/clang/tools/dxc/CMakeLists.txt
index 376adf422..4a525b653 100644
--- a/tools/clang/tools/dxc/CMakeLists.txt
+++ b/tools/clang/tools/dxc/CMakeLists.txt
@@ -2,9 +2,9 @@
 # This file is distributed under the University of Illinois Open Source License. See LICENSE.TXT for details.
 # Builds dxc.exe
 
-if (WIN32)
+if (MSVC)
   find_package(DiaSDK REQUIRED) # Used for constants and declarations.
-endif (WIN32)
+endif (MSVC)
 
 set( LLVM_LINK_COMPONENTS
   ${LLVM_TARGETS_TO_BUILD}
@@ -31,11 +31,15 @@ if(ENABLE_SPIRV_CODEGEN)
   target_link_libraries(dxc SPIRV-Tools)
 endif()
 
+if(MINGW)
+  target_link_options(dxc PRIVATE -mconsole -municode)
+endif()
+
 set_target_properties(dxc PROPERTIES VERSION ${CLANG_EXECUTABLE_VERSION})
 
-if (WIN32)
+if (MSVC)
   include_directories(AFTER ${DIASDK_INCLUDE_DIRS})
-endif (WIN32)
+endif (MSVC)
 
 include_directories(${LLVM_SOURCE_DIR}/tools/clang/tools)
 
diff --git a/tools/clang/tools/dxclib/CMakeLists.txt b/tools/clang/tools/dxclib/CMakeLists.txt
index 830dedfcf..6d24ba8e6 100644
--- a/tools/clang/tools/dxclib/CMakeLists.txt
+++ b/tools/clang/tools/dxclib/CMakeLists.txt
@@ -2,9 +2,9 @@
 # This file is distributed under the University of Illinois Open Source License. See LICENSE.TXT for details.
 # Builds dxclib
 
-if (WIN32)
+if (MSVC)
   find_package(DiaSDK REQUIRED) # Used for constants and declarations.
-endif (WIN32)
+endif (MSVC)
 
 set( LLVM_LINK_COMPONENTS
   ${LLVM_TARGETS_TO_BUILD}
@@ -25,9 +25,13 @@ if(ENABLE_SPIRV_CODEGEN)
   target_link_libraries(dxclib PRIVATE clangSPIRV)
 endif()
 
-if (WIN32)
+if(MINGW)
+  target_link_libraries(dxclib PRIVATE version)
+endif()
+
+if (MSVC)
   include_directories(AFTER ${DIASDK_INCLUDE_DIRS})
-endif (WIN32)
+endif (MSVC)
 
 target_compile_definitions(dxclib
     PRIVATE VERSION_STRING_SUFFIX=" for ${CMAKE_SYSTEM_NAME}")
diff --git a/tools/clang/tools/dxcompiler/CMakeLists.txt b/tools/clang/tools/dxcompiler/CMakeLists.txt
index 57e4c5372..ce7d2b083 100644
--- a/tools/clang/tools/dxcompiler/CMakeLists.txt
+++ b/tools/clang/tools/dxcompiler/CMakeLists.txt
@@ -53,7 +53,7 @@ set(SOURCES
   dxcompilerobj.cpp
   dxcvalidator.cpp
   DXCompiler.cpp
-  DXCompiler.rc
+#  DXCompiler.rc
   DXCompiler.def
   dxcfilesystem.cpp
   dxillib.cpp
@@ -103,11 +103,11 @@ set(LIBRARIES
   libclang
   )
 
-if(WIN32)
+if(MSVC)
   set(LIBRARIES
     ${LIBRARIES} 
     LLVMDxilDia)
-endif(WIN32)
+endif(MSVC)
 
 set(GENERATED_HEADERS
   ClangAttrClasses
@@ -120,13 +120,13 @@ set(GENERATED_HEADERS
   ClangStmtNodes
   )
 
-if (WIN32)
+if (MSVC)
   find_package(DiaSDK REQUIRED) # Used for constants and declarations.
-endif (WIN32)
+endif (MSVC)
 
 add_clang_library(dxcompiler SHARED ${SOURCES})
 add_dependencies(dxcompiler TablegenHLSLOptions) 
-if (WIN32)
+if (MSVC)
   # No DxcEtw on non-Windows platforms.
   add_dependencies(dxcompiler DxcEtw)
 endif()
diff --git a/tools/opt/CMakeLists.txt b/tools/opt/CMakeLists.txt
index a821f70b7..d79ee53e9 100644
--- a/tools/opt/CMakeLists.txt
+++ b/tools/opt/CMakeLists.txt
@@ -38,11 +38,11 @@ add_llvm_tool(opt
   opt.cpp
 )
 
-if(WIN32 AND HLSL_BUILD_DXILCONV)
+if(MSVC AND HLSL_BUILD_DXILCONV)
   add_compile_definitions(HAS_DXILCONV)
   target_link_libraries(opt DxilConvPasses)
   add_dependencies(opt DxilConvPasses)
-endif(WIN32 AND HLSL_BUILD_DXILCONV)
+endif(MSVC AND HLSL_BUILD_DXILCONV)
 
 export_executable_symbols(opt)
 
