diff --git a/tools/clang/utils/TableGen/TableGen.cpp b/tools/clang/utils/TableGen/TableGen.cpp
index 0a02bfca2..ab6413234 100644
--- a/tools/clang/utils/TableGen/TableGen.cpp
+++ b/tools/clang/utils/TableGen/TableGen.cpp
@@ -21,7 +21,9 @@
 
 // HLSL Change Starts
 #ifdef _WIN32
+#ifndef NOMINMAX
 #define NOMINMAX
+#endif
 #include <windows.h>
 #endif
 
diff --git a/utils/TableGen/TableGen.cpp b/utils/TableGen/TableGen.cpp
index a470e8662..b5ab299ba 100644
--- a/utils/TableGen/TableGen.cpp
+++ b/utils/TableGen/TableGen.cpp
@@ -22,7 +22,9 @@
 
 // HLSL Change Starts
 #ifdef _WIN32
+#ifndef NOMINMAX
 #define NOMINMAX
+#endif
 #include <windows.h>
 #endif
 
