diff -bur Trilinos-trilinos-release-16-1-0-o/packages/kokkos/cmake/kokkos_arch.cmake Trilinos-trilinos-release-16-1-0/packages/kokkos/cmake/kokkos_arch.cmake
--- Trilinos-trilinos-release-16-1-0-o/packages/kokkos/cmake/kokkos_arch.cmake	2025-04-14 19:34:12.003632100 -0600
+++ Trilinos-trilinos-release-16-1-0/packages/kokkos/cmake/kokkos_arch.cmake	2025-04-14 19:45:45.121723900 -0600
@@ -691,7 +691,7 @@
 endif()
 
 # Clang needs mcx16 option enabled for Windows atomic functions
-if(CMAKE_CXX_COMPILER_ID STREQUAL Clang AND WIN32)
+if(CMAKE_CXX_COMPILER_ID STREQUAL Clang AND WIN32 AND (NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64")))
   compiler_specific_options(Clang -mcx16)
 endif()
 
