--- a/packages/teuchos/comm/src/Teuchos_EReductionType.cpp
+++ b/packages/teuchos/comm/src/Teuchos_EReductionType.cpp
@@ -33,7 +33,7 @@
 } // namespace Details
 #endif // HAVE_TEUCHOS_MPI
 
-const char*
+TEUCHOSCOMM_LIB_DLL_EXPORT const char*
 toString (const EReductionType reductType)
 {
   switch (reductType) {
--- a/packages/teuchos/comm/src/Teuchos_EReductionType.hpp
+++ b/packages/teuchos/comm/src/Teuchos_EReductionType.hpp
@@ -47,7 +47,7 @@
 
 /// \brief Convert EReductionType to string representation.
 /// \relates EReductionType
-const char* toString (const EReductionType reductType);
+TEUCHOSCOMM_LIB_DLL_EXPORT const char* toString (const EReductionType reductType);
 
 #ifdef HAVE_TEUCHOS_MPI
 namespace Details {
--- a/packages/teuchos/core/src/Teuchos_Behavior.cpp
+++ b/packages/teuchos/core/src/Teuchos_Behavior.cpp
@@ -27,7 +27,7 @@
 }
 
 
-bool Behavior::fenceTimers() {
+TEUCHOSCORE_LIB_DLL_EXPORT bool Behavior::fenceTimers() {
   constexpr bool defaultValue = Teuchos::fenceTimersDefault();
 
   static bool value_ = defaultValue;
--- a/packages/teuchos/core/src/Teuchos_Behavior.hpp
+++ b/packages/teuchos/core/src/Teuchos_Behavior.hpp
@@ -10,11 +10,13 @@
 #ifndef TEUCHOS_BEHAVIORDECL_HPP
 #define TEUCHOS_BEHAVIORDECL_HPP
 
+#include "Teuchos_DLLExportMacro.h"
+
 namespace Teuchos{
 
 class Behavior {
 public:
-  static bool fenceTimers();
+  static TEUCHOSCORE_LIB_DLL_EXPORT bool fenceTimers();
 };
 
 }
--- a/packages/teuchos/core/src/Teuchos_SystemInformation.cpp
+++ b/packages/teuchos/core/src/Teuchos_SystemInformation.cpp
@@ -265,7 +265,7 @@
     registerEnvironmentVariablePrefix(prefix);
 }
 
-std::map<std::string, std::string> collectSystemInformation() {
+TEUCHOSCORE_LIB_DLL_EXPORT std::map<std::string, std::string> collectSystemInformation() {
 
   std::map<std::string, std::string> data;
 
--- a/packages/teuchos/core/src/Teuchos_SystemInformation.hpp
+++ b/packages/teuchos/core/src/Teuchos_SystemInformation.hpp
@@ -27,6 +27,8 @@
 #include <map>
 #include <string>
 
+#include "Teuchos_DLLExportMacro.h"
+
 namespace Teuchos::SystemInformation {
 
 /// Check whether a command is available on the system.
@@ -57,7 +59,7 @@
 void initializeCollection();
 
 /// Collect information about the system.
-std::map<std::string, std::string> collectSystemInformation();
+TEUCHOSCORE_LIB_DLL_EXPORT std::map<std::string, std::string> collectSystemInformation();
 
 } // namespace Teuchos::SystemInformation
 
