--- libicsneo/include/icsneo/platform/windows/dynamiclib.h.orig	2019-05-17 08:35:32.284229600 +0300
+++ libicsneo/include/icsneo/platform/windows/dynamiclib.h	2019-05-17 08:35:44.344050800 +0300
@@ -12,7 +12,7 @@
 // MSVC does not have the ability to specify a destructor
 #define ICSNEO_DESTRUCTOR
 
-#define icsneo_dynamicLibraryLoad() LoadLibrary(L"C:\\Users\\Phollinsky\\Code\\icsneonext\\build\\icsneoc.dll")
+#define icsneo_dynamicLibraryLoad() LoadLibrary(L"libicsneoc.dll")
 #define icsneo_dynamicLibraryGetFunction(handle, func) GetProcAddress((HMODULE) handle, func)
 #define icsneo_dynamicLibraryClose(handle) FreeLibrary((HMODULE) handle)
 
