--- a/pagelayout_editor/tools/pl_edit_tool.cpp
+++ b/pagelayout_editor/tools/pl_edit_tool.cpp
@@ -140,7 +140,7 @@
        {
             m_frame->SaveCopyInUndoList();
         }
-        catch( const fmt::v10::format_error& exc )
+        catch( const fmt::v11::format_error& exc )
         {
             wxLogWarning( wxS( "Exception \"%s\" serializing string ocurred." ),
                           exc.what() );
--- a/pagelayout_editor/tools/pl_point_editor.cpp
+++ b/pagelayout_editor/tools/pl_point_editor.cpp
@@ -197,7 +197,7 @@
                {
                     m_frame->SaveCopyInUndoList();
                 }
-                catch( const fmt::v10::format_error& exc )
+                catch( const fmt::v11::format_error& exc )
                 {
                     wxLogWarning( wxS( "Exception \"%s\" serializing string ocurred." ),
                                   exc.what() );
