From a32e4ee101e840d3b8b2e0506aaad4064bfb7b88 Mon Sep 17 00:00:00 2001
From: kreijstal <rainb@tfwno.gf>
Date: Mon, 24 Feb 2025 23:35:51 +0100
Subject: [PATCH] MSVC!=WIN32

---
 libs/kiplatform/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libs/kiplatform/CMakeLists.txt b/libs/kiplatform/CMakeLists.txt
index 04aa3bd8..29b4ce76 100644
--- a/libs/kiplatform/CMakeLists.txt
+++ b/libs/kiplatform/CMakeLists.txt
@@ -132,8 +132,8 @@ if( APPLE )
     )
 endif()
 
-if( MSVC )
+if( WIN32 )
 add_dependencies( kiplatform
     version_header
     )
-endif()
\ No newline at end of file
+endif()
-- 
2.48.1

