From 9d38bedb042c17e8c3141142611e2c2f35dc0bd2 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Fri, 3 Jun 2022 04:41:45 -0400
Subject: Add wxString header and remove wxT macro

---
 include/bitmap_store.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/bitmap_store.h b/include/bitmap_store.h
index 0d11546785..e176792a98 100644
--- a/include/bitmap_store.h
+++ b/include/bitmap_store.h
@@ -23,6 +23,8 @@
 #include <memory>
 #include <unordered_map>
 
+#include <wx/string.h>
+
 #include <bitmaps/bitmap_info.h>
 
 class ASSET_ARCHIVE;
@@ -76,7 +78,7 @@ public:
      */
     void ThemeChanged();
 
-    bool IsDarkTheme() const { return m_theme == wxT( "dark" ); }
+    bool IsDarkTheme() const { return m_theme == "dark" ; }
 
 private:
 
-- 
