--- src/plugins/azoth/plugins/xoox/CMakeLists.txt.orig	2014-07-18 23:55:42.420365000 +0300
+++ src/plugins/azoth/plugins/xoox/CMakeLists.txt	2014-07-18 23:56:02.860534100 +0300
@@ -167,6 +167,7 @@
 	${QXMPP_LIBRARIES}
 	${SPEEX_LIBRARIES}
 	${QCA2_LIBRARIES}
+    ${QTMULTIMEDIAKIT_LIBRARIES}
 	)
 install (TARGETS leechcraft_azoth_xoox DESTINATION ${LC_PLUGINS_DEST})
 install (FILES azothxooxsettings.xml DESTINATION ${LC_SETTINGS_DEST})
