--- src/plugins/azoth/plugins/otroid/cmake/FindLibOTR.cmake.orig	2014-07-12 23:40:52.923459600 +0300
+++ src/plugins/azoth/plugins/otroid/cmake/FindLibOTR.cmake	2014-07-13 00:57:33.387591100 +0300
@@ -8,29 +8,18 @@
 include(MacroEnsureVersion)
 include(FindPackageHandleStandardArgs)
 
-if (LIBOTR_INCLUDE_DIR AND LIBOTR_LIBRARY)
-    # Already in cache, be silent
-    set(LIBOTR_FIND_QUIETLY TRUE)
-endif ()
+find_package (PkgConfig)
+pkg_check_modules (PC_LIBOTR REQUIRED libotr>=3.2)
 
-find_path(LIBOTR_INCLUDE_DIR libotr/version.h)
-
-find_library(LIBOTR_LIBRARY NAMES otr libotr)
-
-# Determine version information from libotr/version.h
-if( LIBOTR_INCLUDE_DIR AND LIBOTR_LIBRARY )
-  execute_process(COMMAND grep "OTRL_VERSION" "${LIBOTR_INCLUDE_DIR}/libotr/version.h" OUTPUT_VARIABLE output)
-  string(REGEX MATCH "OTRL_VERSION \"[0-9]+\\.[0-9]+\\.[0-9]+" LIBOTR_VERSION "${output}")
-  string(REGEX REPLACE "^OTRL_VERSION \"" "" LIBOTR_VERSION "${LIBOTR_VERSION}")
-  # Check if version is at least 3.2.0
-  MACRO_ENSURE_VERSION("3.2.0" ${LIBOTR_VERSION} LIBOTR_FOUND)
-
-  if( LIBOTR_FOUND )
-    if( NOT LIBOTR_FIND_QUIETLY )
-      message( STATUS "Found libotr: ${LIBOTR_LIBRARY}")
-    endif()
-  else()
-    message(STATUS "libotr version 3.2.0 or greater required but found ${LIBOTR_VERSION}.")
-  endif()
-
-endif()
+find_path (LIBOTR_INCLUDE_DIR libotr/version.h
+          HINTS ${PC_LIBOTR_INCLUDEDIR} ${PC_LIBOTR_INCLUDE_DIRS}
+          PATH_SUFFIXES libotr)
+
+find_library(LIBOTR_LIBRARY NAMES libotr
+             HINTS ${PC_LIBOTR_LIBDIR} ${PC_LIBOTR_LIBRARY_DIRS} )
+find_library(LIBGCRYPT_LIBRARY NAMES  libgcrypt
+             HINTS ${PC_LIBOTR_LIBDIR} ${PC_LIBOTR_LIBRARY_DIRS} )
+find_library(LIBGPG_ERROR NAMES libgpg-error
+             HINTS ${PC_LIBOTR_LIBDIR} ${PC_LIBOTR_LIBRARY_DIRS} )
+set (LIBOTR_LIBRARY ${LIBOTR_LIBRARY} ${LIBGCRYPT_LIBRARY} ${LIBGPG_ERROR})
+message (STATUS "libor found: ${LIBOTR_LIBRARY}, includedir: ${LIBOTR_INCLUDE_DIR}")
