--- x86_64/src/plugins/lhtr/cmake/FindLibTidy.cmake.orig	2014-12-13 17:43:06.245824800 +0300
+++ x86_64/src/plugins/lhtr/cmake/FindLibTidy.cmake	2014-12-13 17:45:55.087482000 +0300
@@ -12,11 +12,17 @@
 endif (LIBTIDY_INCLUDE_DIR)
 
 # Look for the header file.
-find_path(LIBTIDY_INCLUDE_DIR tidy.h PATH_SUFFIXES tidy PATHS ${LIBTIDY_DIR}/include)
+find_path(LIBTIDY_INCLUDE_DIR tidy.h PATH_SUFFIXES tidy
+  PATHS
+    ${MINGW_PREFIX}/include
+    ${LIBTIDY_DIR}/include)
 mark_as_advanced(LIBTIDY_INCLUDE_DIR)
 
 # Look for the library.
-find_library(LIBTIDY_LIBRARY NAMES tidy PATHS ${LIBTIDY_DIR}/lib)
+find_library(LIBTIDY_LIBRARY NAMES tidy
+  PATHS
+    ${MINGW_PREFIX}/lib
+    ${LIBTIDY_DIR}/lib)
 mark_as_advanced(LIBTIDY_LIBRARY)
 
 if (LIBTIDY_INCLUDE_DIR AND LIBTIDY_LIBRARY)
