--- x86_64/src/cmake/FindQwt.cmake.orig	2014-12-13 16:52:50.274321100 +0300
+++ x86_64/src/cmake/FindQwt.cmake	2014-12-13 17:08:22.775657100 +0300
@@ -13,6 +13,7 @@
 	/usr/include/qwt6
 	/usr/include/qwt
 	/usr/local/lib/qwt.framework/Headers
+	${MINGW_PREFIX}/include/qwt
 )
 
 if (NOT USE_QT5)
@@ -22,6 +23,7 @@
 		/usr/local/lib
 		/usr/lib
 		/usr/local/lib/qwt.framework
+		${MINGW_PREFIX}/lib
 	)
 else ()
 	find_library (QWT_LIBRARIES
@@ -30,6 +32,7 @@
 		/usr/local/lib
 		/usr/lib
 		/usr/local/lib/qwt.framework
+		${MINGW_PREFIX}/lib
 	)
 endif ()
 
