--- x86_64/src/cmake/FindQJSON.cmake.orig	2014-12-14 00:40:10.259116500 +0300
+++ x86_64/src/cmake/FindQJSON.cmake	2014-12-14 00:42:52.264382600 +0300
@@ -11,7 +11,7 @@
 	# Already in cache
 	set (QJSON_FOUND TRUE)
 else ()
-	if (NOT WIN32)
+	if (NOT WIN32 OR MINGW)
 		find_library (QJSON_LIBRARIES
 			NAMES
 				qjson
@@ -19,6 +19,7 @@
 				${QJSON_DIR}
 				${LIB_INSTALL_DIR}
 				${KDE4_LIB_DIR}
+                ${MINGW_PREFIX}
 		)
 	else ()
 		if (NOT DEFINED QJSON_DIR)
