--- src/plugins/htthare/CMakeLists.txt.orig	2014-07-20 13:29:40.973786900 +0300
+++ src/plugins/htthare/CMakeLists.txt	2014-07-20 13:30:28.943530600 +0300
@@ -32,6 +32,8 @@
 	${QT_LIBRARIES}
 	${Boost_SYSTEM_LIBRARY}
 	${LEECHCRAFT_LIBRARIES}
+    ws2_32
+    mswsock
 	)
 install (TARGETS leechcraft_htthare DESTINATION ${LC_PLUGINS_DEST})
 install (FILES httharesettings.xml DESTINATION ${LC_SETTINGS_DEST})
