--- src/plugins/lmp/lmp.cpp.orig	2014-07-12 17:41:32.553911000 +0300
+++ src/plugins/lmp/lmp.cpp	2014-07-12 17:43:32.342762500 +0300
@@ -83,6 +83,21 @@
 
 			qputenv ("GST_REGISTRY_FORK", "no");
 		}
+#elif defined (Q_OS_WIN)
+		auto updateEnv = [] (const char *name, const QByteArray& relpath)
+		{
+			if (qgetenv (name).isEmpty ())
+				qputenv (name,
+						QDir::toNativeSeparators (QCoreApplication::applicationDirPath () + relpath).toUtf8());
+		};
+		updateEnv ("GST_PLUGIN_SYSTEM_PATH", "/gstreamer-1.0");
+		updateEnv ("GST_PLUGIN_SCANNER", "gst-plugin-scanner");
+		updateEnv ("GTK_PATH", "/gstreamer-1.0");
+		updateEnv ("GIO_EXTRA_MODULES", "/gio-modules");
+		updateEnv ("GSETTINGS_SCHEMA_DIR", "/gstreamer-1.0/schemas");
+		//updateEnv ("GST_DEBUG", "7");
+
+		qputenv ("GST_REGISTRY_FORK", "no");
 #endif
 
 		gint argc = 1;
