diff -Naur libdeflate-1.7.orig/Makefile libdeflate-1.7/Makefile
--- libdeflate-1.7.orig/Makefile	2020-11-10 04:29:50.000000000 +0100
+++ libdeflate-1.7/Makefile	2020-11-12 15:54:02.152338000 +0100
@@ -85,11 +85,10 @@
 
 # Compiling for Windows with MinGW?
 ifneq ($(findstring -mingw,$(shell $(CC) -dumpmachine 2>/dev/null)),)
-    STATIC_LIB_SUFFIX  := static.lib
     SHARED_LIB         := libdeflate.dll
     SHARED_LIB_SYMLINK :=
     SHARED_LIB_CFLAGS  :=
-    SHARED_LIB_LDFLAGS := -Wl,--out-implib,libdeflate.lib \
+    SHARED_LIB_LDFLAGS := -Wl,--out-implib,libdeflate.dll.a \
                           -Wl,--output-def,libdeflate.def \
                           -Wl,--add-stdcall-alias
     PROG_SUFFIX        := .exe
@@ -305,7 +304,8 @@
 install:all
 	install -d $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INCDIR) $(DESTDIR)$(BINDIR)
 	install -m644 $(STATIC_LIB) $(DESTDIR)$(LIBDIR)
-	install -m755 $(SHARED_LIB) $(DESTDIR)$(LIBDIR)
+	install -m644 $(SHARED_LIB).a $(DESTDIR)$(LIBDIR)
+	install -m755 $(SHARED_LIB) $(DESTDIR)$(BINDIR)
 	install -m644 libdeflate.h $(DESTDIR)$(INCDIR)
 	install -m755 gzip$(PROG_SUFFIX) \
 		$(DESTDIR)$(BINDIR)/libdeflate-gzip$(PROG_SUFFIX)
@@ -318,7 +318,8 @@
 
 uninstall:
 	rm -f $(DESTDIR)$(LIBDIR)/$(STATIC_LIB)				\
-	      $(DESTDIR)$(LIBDIR)/$(SHARED_LIB)				\
+	      $(DESTDIR)$(LIBDIR)/$(SHARED_LIB).a			\
+	      $(DESTDIR)$(BINDIR)/$(SHARED_LIB)				\
 	      $(DESTDIR)$(INCDIR)/libdeflate.h				\
 	      $(DESTDIR)$(BINDIR)/libdeflate-gzip$(PROG_SUFFIX)		\
 	      $(DESTDIR)$(BINDIR)/libdeflate-gunzip$(PROG_SUFFIX)
