diff -Naur nana-1.7.4.orig/build/cmake/compilers.cmake nana-1.7.4/build/cmake/compilers.cmake
--- nana-1.7.4.orig/build/cmake/compilers.cmake	2020-05-17 00:08:50.000000000 +0200
+++ nana-1.7.4/build/cmake/compilers.cmake	2022-03-11 09:33:12.048159900 +0100
@@ -16,9 +16,6 @@
 
     target_compile_options(nana PRIVATE  -Wall)
 
-        # todo: set in target property of nana
-    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -march=native -mtune=native -DNDEBUG")
-
     set(THREADS_PREFER_PTHREAD_FLAG ON)               #  todo - test this
     find_package(Threads REQUIRED)
     target_link_libraries(nana PRIVATE Threads::Threads)
