--- podman-5.0.1/Makefile.orig	2024-04-01 14:28:31.000000000 +0200
+++ podman-5.0.1/Makefile	2024-04-16 08:31:07.202655400 +0200
@@ -39,7 +39,7 @@
 LIBEXECPODMAN ?= ${LIBEXECDIR}/podman
 MANDIR ?= ${PREFIX}/share/man
 SHAREDIR_CONTAINERS ?= ${PREFIX}/share/containers
-ETCDIR ?= /etc
+ETCDIR ?= ${PREFIX}/etc
 LIBDIR ?= ${PREFIX}/lib
 TMPFILESDIR ?= ${LIBDIR}/tmpfiles.d
 USERTMPFILESDIR ?= ${PREFIX}/share/user-tmpfiles.d
