--- gammaray-2.11.3/3rdparty/backward-cpp/BackwardConfig.cmake.orig	2021-10-17 23:35:36.000000000 +0300
+++ gammaray-2.11.3/3rdparty/backward-cpp/BackwardConfig.cmake	2021-11-18 09:46:23.440523200 +0300
@@ -63,9 +63,9 @@
 	mark_as_advanced(LIBDW_INCLUDE_DIR LIBDW_LIBRARY)
 
 	# find libbfd
-	find_path(LIBBFD_INCLUDE_DIR NAMES "bfd.h")
+	find_path(LIBBFD_INCLUDE_DIR NAMES "bfd.h" PATH_SUFFIXES binutils)
 	find_path(LIBDL_INCLUDE_DIR NAMES "dlfcn.h")
-	find_library(LIBBFD_LIBRARY bfd)
+	find_library(LIBBFD_LIBRARY bfd PATH_SUFFIXES binutils)
 	find_library(LIBDL_LIBRARY dl)
 	set(LIBBFD_INCLUDE_DIRS ${LIBBFD_INCLUDE_DIR} ${LIBDL_INCLUDE_DIR})
 	set(LIBBFD_LIBRARIES ${LIBBFD_LIBRARY} ${LIBDL_LIBRARY})
@@ -77,7 +77,7 @@
 
 	# find libdwarf
 	find_path(LIBDWARF_INCLUDE_DIR NAMES "libdwarf.h" PATH_SUFFIXES libdwarf)
-	find_path(LIBELF_INCLUDE_DIR NAMES "libelf.h")
+	find_path(LIBELF_INCLUDE_DIR NAMES "libelf.h" PATH_SUFFIXES libelf)
 	find_path(LIBDL_INCLUDE_DIR NAMES "dlfcn.h")
 	find_library(LIBDWARF_LIBRARY dwarf)
 	find_library(LIBELF_LIBRARY elf)
