--- gammaray-2.11.3/CMakeLists.txt.orig	2021-10-17 23:35:36.000000000 +0300
+++ gammaray-2.11.3/CMakeLists.txt	2021-11-18 14:17:59.967446800 +0300
@@ -21,7 +21,7 @@
 #
 # -DGAMMARAY_INSTALL_QT_LAYOUT=[true|false]
 #  Install GammaRay in Qt folder using Qt layout folders:
-#    - plugins will go in <qt_folder>/plugins/gammaray
+#    - plugins will go in <qt_folder>/share/qt5/plugins/gammaray
 #    - all the other libs, including probe library will go in <qt_folder>/lib
 #    - qmake mkspecs file
 #  Default=false
@@ -658,10 +678,10 @@
   set(QCH_INSTALL_DIR "${CMAKE_INSTALL_DOCDIR}" CACHE PATH "Install location of Qt Assistant help files.")
   gammaray_convert_to_relative_path(QCH_INSTALL_DIR)
   if(WIN32)
-    set(PLUGIN_INSTALL_DIR "plugins/gammaray")
+    set(PLUGIN_INSTALL_DIR "share/qt5/plugins/gammaray")
     set(LIBEXEC_INSTALL_DIR "${BIN_INSTALL_DIR}")
-    set(DOC_INSTALL_DIR .)
-    set(TRANSLATION_INSTALL_DIR "translations")
+    set(DOC_INSTALL_DIR "share/doc/gammaray/")
+    set(TRANSLATION_INSTALL_DIR "share/qt5/translations")
   else()
     set(PLUGIN_INSTALL_DIR "${LIB_INSTALL_DIR}/gammaray")
     set(LIBEXEC_INSTALL_DIR "${LIB_INSTALL_DIR}/gammaray/libexec")
@@ -682,9 +702,9 @@
   else()
     set(PROBE_INSTALL_DIR ${LIB_INSTALL_DIR})
   endif()
-  set(PROBE_PLUGIN_INSTALL_DIR "plugins/gammaray")
+  set(PROBE_PLUGIN_INSTALL_DIR "share/qt5/plugins/gammaray")
   set(PLUGIN_INSTALL_DIR ${PROBE_PLUGIN_INSTALL_DIR})
-  set(TARGET_PLUGIN_INSTALL_DIR "plugins/gammaray-target")
+  set(TARGET_PLUGIN_INSTALL_DIR "share/qt5/plugins/gammaray-target")
 else()
   set(PROBE_INSTALL_DIR "${PLUGIN_INSTALL_DIR}/${GAMMARAY_PLUGIN_VERSION}/${GAMMARAY_PROBE_ABI}${GAMMARAY_PROBE_ABI_POSTFIX}")
   set(PROBE_PLUGIN_INSTALL_DIR ${PROBE_INSTALL_DIR})
