--- geocode-glib-3.13.3-orig/geocode-glib/geocode-glib.c	2014-06-25 00:37:44.000000000 +0200
+++ geocode-glib-3.13.3/geocode-glib/geocode-glib.c	2014-08-15 16:22:59.265705500 +0200
@@ -25,7 +25,9 @@
 #include <locale.h>
 #include <gio/gio.h>
 #include <libsoup/soup.h>
+#ifndef __MINGW32__
 #include <langinfo.h>
+#endif
 #include <geocode-glib/geocode-glib-private.h>
 
 /**
@@ -186,5 +186,9 @@
 char *
 _geocode_object_get_lang (void)
 {
+#ifdef G_OS_WIN32
+	return geocode_object_get_lang_for_locale (setlocale (LC_ALL, NULL));
+#else
 	return geocode_object_get_lang_for_locale (setlocale (LC_MESSAGES, NULL));
+#endif
 }
