--- meld-3.11.4/setup.py.orig	2014-09-24 12:59:19.388200000 +0500
+++ meld-3.11.4/setup.py	2014-09-24 13:01:37.448200000 +0500
@@ -6,6 +6,14 @@
 import meld.build_helpers
 import meld.conf
 
+from cx_Freeze import setup, Executable
+
+build_exe_options = {
+    "compressed": False,
+    "icon": "data/icons/meld.ico",
+    "includes": ["gi"],
+    "packages": ["gi", "weakref"],
+}
 
 setup(
     name=meld.conf.__package__,
@@ -25,6 +32,18 @@
         'Topic :: Software Development',
         'Topic :: Software Development :: Version Control',
     ],
+    options = {
+        "build_exe": build_exe_options,
+    },
+    executables = [
+        Executable(
+            "bin/meld",
+            base="Win32GUI",
+            targetName="Meld.exe",
+            shortcutName="Meld",
+            shortcutDir="ProgramMenuFolder",
+        ),
+    ],
     packages=[
         'meld',
         'meld.ui',
