--- meld-3.19.0/bin/meld.orig	2018-03-23 00:07:52.000000000 +0100
+++ meld-3.19.0/bin/meld	2018-04-17 10:50:43.104917400 +0200
@@ -347,6 +347,12 @@
 
 
 if __name__ == '__main__':
+    if sys.platform == 'win32':
+        # FontConfig on win32 at least with version <= 2.12.6 can cause several
+        # minutes 'hang' during first startup. So use native fonts backend.
+        assert "gi.repository.PangoCairo" not in sys.modules
+        os.environ['PANGOCAIRO_BACKEND'] = 'win32'
+
     setup_logging()
     disable_stdout_buffering()
     check_requirements()
@@ -360,10 +366,5 @@
         from gi.repository import GLib
         GLib.unix_signal_add(GLib.PRIORITY_DEFAULT, signal.SIGINT,
                              lambda *args: meld.meldapp.app.quit(), None)
-    if sys.platform == 'win32':
-        # FontConfig on win32 at least with version <= 2.12.6 can cause several
-        # minutes 'hang' during first startup. So use native fonts backend.
-        from gi.repository import PangoCairo
-        PangoCairo.FontMap.set_default(PangoCairo.Win32FontMap())
     status = meld.meldapp.app.run(sys.argv)
     sys.exit(status)
