--- manifold-3.2.1/bindings/python/CMakeLists.txt.orig	2025-08-06 09:14:41.000000000 +0300
+++ manifold-3.2.1/bindings/python/CMakeLists.txt	2025-08-26 14:48:49.652043700 +0300
@@ -28,6 +28,13 @@
   )
 
   add_custom_command(
+    TARGET manifold3d
+    POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/src/libmanifold.dll ${CMAKE_CURRENT_BINARY_DIR}/.
+    VERBATIM
+  )
+  
+  add_custom_command(
     TARGET manifold3d_stub
     POST_BUILD
     COMMAND
@@ -66,9 +73,9 @@
 target_include_directories(manifold3d PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 
 if(SKBUILD)
-  set(MANIFOLD_PYBIND_LIBDIR ${SKBUILD_PLATLIB_DIR})
+  set(MANIFOLD_PYBIND_LIBDIR ${PY_SITEARCH})
 else()
-  set(MANIFOLD_PYBIND_LIBDIR ${Python_SITEARCH})
+  set(MANIFOLD_PYBIND_LIBDIR ${PY_SITEARCH})
 endif()
 
 install(
