diff -Naur VulkanSceneGraph-1.1.8.orig/src/vsg/platform/win32/Win32_Window.cpp VulkanSceneGraph-1.1.8/src/vsg/platform/win32/Win32_Window.cpp
--- VulkanSceneGraph-1.1.8.orig/src/vsg/platform/win32/Win32_Window.cpp	2024-11-30 16:39:27.446661400 +0300
+++ VulkanSceneGraph-1.1.8/src/vsg/platform/win32/Win32_Window.cpp	2024-11-30 16:59:23.235665500 +0300
@@ -222,14 +222,14 @@
             {VK_F22                               ,              KEY_F22},
             {VK_F23                               ,              KEY_F23},
             {VK_F24                               ,              KEY_F24},
-            {VK_NAVIGATION_VIEW                   ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_MENU                   ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_UP                     ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_DOWN                   ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_LEFT                   ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_RIGHT                  ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_ACCEPT                 ,              KEY_Undefined},   // reserved
-            {VK_NAVIGATION_CANCEL                 ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_VIEW                   ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_MENU                   ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_UP                     ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_DOWN                   ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_LEFT                   ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_RIGHT                  ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_ACCEPT                 ,              KEY_Undefined},   // reserved
+//            {VK_NAVIGATION_CANCEL                 ,              KEY_Undefined},   // reserved
             {VK_NUMLOCK                           ,              KEY_Undefined},
             {VK_SCROLL                            ,              KEY_Scroll_Lock},
             {VK_LSHIFT                            ,              KEY_Shift_L},
@@ -263,30 +263,30 @@
             {VK_OEM_PERIOD                        ,              KEY_Period   },    // '.' any country
             {VK_OEM_2                             ,              KEY_Slash    },    // '/?' for US
             {VK_OEM_3                             ,              KEY_Tilde    },    // '`~' for US
-            {VK_GAMEPAD_A                         ,              KEY_Undefined},
-            {VK_GAMEPAD_B                         ,              KEY_Undefined},
-            {VK_GAMEPAD_X                         ,              KEY_Undefined},
-            {VK_GAMEPAD_Y                         ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_SHOULDER            ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_SHOULDER             ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_TRIGGER              ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_TRIGGER             ,              KEY_Undefined},
-            {VK_GAMEPAD_DPAD_UP                   ,              KEY_Undefined},
-            {VK_GAMEPAD_DPAD_DOWN                 ,              KEY_Undefined},
-            {VK_GAMEPAD_DPAD_LEFT                 ,              KEY_Undefined},
-            {VK_GAMEPAD_DPAD_RIGHT                ,              KEY_Undefined},
-            {VK_GAMEPAD_MENU                      ,              KEY_Undefined},
-            {VK_GAMEPAD_VIEW                      ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_THUMBSTICK_BUTTON    ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_THUMBSTICK_BUTTON   ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_THUMBSTICK_UP        ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_THUMBSTICK_DOWN      ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_THUMBSTICK_RIGHT     ,              KEY_Undefined},
-            {VK_GAMEPAD_LEFT_THUMBSTICK_LEFT      ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_THUMBSTICK_UP       ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_THUMBSTICK_DOWN     ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_THUMBSTICK_RIGHT    ,              KEY_Undefined},
-            {VK_GAMEPAD_RIGHT_THUMBSTICK_LEFT     ,              KEY_Undefined},
+//            {VK_GAMEPAD_A                         ,              KEY_Undefined},
+//            {VK_GAMEPAD_B                         ,              KEY_Undefined},
+//            {VK_GAMEPAD_X                         ,              KEY_Undefined},
+//            {VK_GAMEPAD_Y                         ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_SHOULDER            ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_SHOULDER             ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_TRIGGER              ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_TRIGGER             ,              KEY_Undefined},
+//            {VK_GAMEPAD_DPAD_UP                   ,              KEY_Undefined},
+//            {VK_GAMEPAD_DPAD_DOWN                 ,              KEY_Undefined},
+//            {VK_GAMEPAD_DPAD_LEFT                 ,              KEY_Undefined},
+//            {VK_GAMEPAD_DPAD_RIGHT                ,              KEY_Undefined},
+//            {VK_GAMEPAD_MENU                      ,              KEY_Undefined},
+//            {VK_GAMEPAD_VIEW                      ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_THUMBSTICK_BUTTON    ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_THUMBSTICK_BUTTON   ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_THUMBSTICK_UP        ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_THUMBSTICK_DOWN      ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_THUMBSTICK_RIGHT     ,              KEY_Undefined},
+//            {VK_GAMEPAD_LEFT_THUMBSTICK_LEFT      ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_THUMBSTICK_UP       ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_THUMBSTICK_DOWN     ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_THUMBSTICK_RIGHT    ,              KEY_Undefined},
+//            {VK_GAMEPAD_RIGHT_THUMBSTICK_LEFT     ,              KEY_Undefined},
             {VK_OEM_4                             ,              KEY_Leftbracket},  //  '[{' for US
             {VK_OEM_5                             ,              KEY_Backslash},    //  '\|' for US
             {VK_OEM_6                             ,              KEY_Rightbracket}, //  ']}' for US
