diff -bur VulkanSceneGraph-1.1.10-orig/include/vsg/commands/PipelineBarrier.h VulkanSceneGraph-1.1.10/include/vsg/commands/PipelineBarrier.h
--- VulkanSceneGraph-1.1.10-orig/include/vsg/commands/PipelineBarrier.h	2025-03-15 02:15:16 -0600
+++ VulkanSceneGraph-1.1.10/include/vsg/commands/PipelineBarrier.h	2025-03-15 02:17:37 -0600
@@ -17,6 +17,11 @@
 #include <vsg/state/Buffer.h>
 #include <vsg/state/Image.h>
 
+#ifdef _M_ARM64
+#pragma push_macro("MemoryBarrier")
+#undef MemoryBarrier
+#endif
+
 namespace vsg
 {
 
@@ -157,3 +162,7 @@
     VSG_type_name(vsg::PipelineBarrier);
 
 } // namespace vsg
+
+#ifdef _M_ARM64
+#pragma pop_macro("MemoryBarrier")
+#endif
\ No newline at end of file
