diff -Naur mosquitto-2.0.18.orig/libmosquitto.pc.in mosquitto-2.0.18/libmosquitto.pc.in
--- mosquitto-2.0.18.orig/libmosquitto.pc.in	2024-04-10 00:32:55.137531100 +0300
+++ mosquitto-2.0.18/libmosquitto.pc.in	2024-04-10 21:47:48.600521000 +0300
@@ -7,4 +7,5 @@
 Description: mosquitto MQTT library (C bindings)
 Version: @VERSION@
 Cflags: -I${includedir}
+Cflags.private: -DLIBMOSQUITTO_STATIC
 Libs: -L${libdir} -lmosquitto
diff -Naur mosquitto-2.0.18.orig/libmosquittopp.pc.in mosquitto-2.0.18/libmosquittopp.pc.in
--- mosquitto-2.0.18.orig/libmosquittopp.pc.in	2024-04-10 00:32:54.809404400 +0300
+++ mosquitto-2.0.18/libmosquittopp.pc.in	2024-04-10 21:47:48.616141900 +0300
@@ -7,4 +7,5 @@
 Description: mosquitto MQTT library (C++ bindings)
 Version: @VERSION@
 Cflags: -I${includedir}
+Cflags.private: -DLIBMOSQUITTO_STATIC
 Libs: -L${libdir} -lmosquittopp
