--- minizip-2.7.0/CMakeLists.txt.orig	2018-07-19 13:13:04.604557000 +0300
+++ minizip-2.7.0/CMakeLists.txt	2018-07-19 13:16:08.627480400 +0300
@@ -476,8 +476,17 @@
                 ${BZIP2_SRC} ${BZIP2_PUBLIC_HEADERS}
                 ${LZMA_SRC} ${LZMA_PUBLIC_HEADERS})
 
-if (MSVC AND BUILD_SHARED_LIBS)
-    set_target_properties(${PROJECT_NAME} PROPERTIES ARCHIVE_OUTPUT_NAME "minizip")
+if (WIN32)
+if (BUILD_SHARED_LIBS)
+  set_target_properties(${PROJECT_NAME} PROPERTIES 
+                        ARCHIVE_OUTPUT_NAME "minizip2"
+                        OUTPUT_NAME "minizip2"
+                        RUNTIME_OUTPUT_NAME "libminizip2")
+else ()
+  set_target_properties(${PROJECT_NAME} PROPERTIES 
+                        ARCHIVE_OUTPUT_NAME "libminizip2")
+endif ()
+  target_link_libraries(${PROJECT_NAME} -lcrypt32)
 endif ()
 
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${VERSION} SOVERSION ${SOVERSION})
