--- CGAL-4.6.1/CMakeLists.txt.orig	2015-07-30 23:35:01.281605000 +0300
+++ CGAL-4.6.1/CMakeLists.txt	2015-07-30 23:35:24.532518300 +0300
@@ -826,7 +826,7 @@
 foreach(lib ${CGAL_CONFIGURED_LIBRARIES} CGAL) # note the added fake 'CGAL'
  if(${lib}_LIBRARY)
     get_filename_component(lib_name "${${lib}_LIBRARY}" NAME)
-    set(${lib}_LIBRARY_INSTALLED "${CMAKE_INSTALL_PREFIX}/${CGAL_INSTALL_LIB_DIR}/${lib_name}")
+    set(${lib}_LIBRARY_INSTALLED "${CMAKE_INSTALL_PREFIX}/${CGAL_INSTALL_BIN_DIR}/${lib_name}")
  else()
     set(${lib}_LIBRARY_INSTALLED "")
   endif()
