--- libnice-0.1.23/tests/meson.build.orig	2025-11-28 15:30:46.843559500 +0100
+++ libnice-0.1.23/tests/meson.build	2025-11-28 15:30:15.724695500 +0100
@@ -87,7 +87,7 @@
 # In this case, in static mode (gstreamer-full), the test should be built after gstreamer-full to initialize
 # properly the plugins (gstreamer and libnice ones) with gst_init_static_plugins.
 # That's the reason the test is now disabled in static mode.
-if gst_dep.found() and not static_build
+if gst_dep.found() and not static_build and host_machine.system() != 'windows'
   gst_check = dependency('gstreamer-check-1.0', required: get_option('gstreamer'))
   if gst_check.found()
     test_defines = [
