--- libnice-0.1.8/agent/Makefile.am.orig	2015-01-01 19:43:03.942400000 +0300
+++ libnice-0.1.8/agent/Makefile.am	2015-01-01 19:44:15.187600000 +0300
@@ -86,6 +86,7 @@
 #
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir) --warn-all --accept-unprefixed
 INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
 
