--- xavs-0.1.55/configure	2017-11-01 07:06:44.216163200 -0400
+++ xavs-0.1.55/configure	2017-11-01 07:08:12.358962400 -0400
@@ -155,6 +155,7 @@
 host="${host#*-}"
 host_vendor="${host%%-*}"
 host_os="${host#*-}"
+host_ver="${host_os#*-}"
 
 case $host_os in
   beos*)
@@ -404,8 +405,7 @@
 fi
 
 if [ "$debug" != "yes" -a "$gprof" != "yes" ]; then
-    CFLAGS="$CFLAGS -s -fomit-frame-pointer"
-    LDFLAGS="$LDFLAGS -s"
+    CFLAGS="$CFLAGS -fomit-frame-pointer"
 fi
 
 if [ "$debug" = "yes" ]; then
--- xavs-0.1.55/Makefile	2017-11-01 07:09:46.732294400 -0400
+++ xavs-0.1.55/Makefile	2017-11-01 07:09:50.859793100 -0400
@@ -69,8 +69,8 @@
 $(SONAME): .depend $(OBJS) $(OBJASM)
 	$(CC) -shared -o $@ $(OBJS) $(OBJASM) -Wl,-soname,$(SOFLAGS) $(LDFLAGS)
 
-xavs$(EXE): $(OBJCLI) libxavs.a 
-	$(CC) -o $@ $+ $(LDFLAGS)
+xavs$(EXE): $(OBJCLI) $(SONAME)
+	$(CC) -o $@ $(OBJCLI) $(LDFLAGS) -L. -lxavs
 
 xavsvfw.dll: libxavs.a $(wildcard vfw/*.c vfw/*.h)
 	make -C vfw/build/cygwin
