--- xavs-0.1.55/common/i386/cpu-a.asm	2017-11-01 07:10:57.363170400 -0400
+++ xavs-0.1.55/common/i386/cpu-a.asm	2017-11-01 07:11:00.016877500 -0400
@@ -37,7 +37,6 @@
 
 cglobal xavs_cpu_cpuid_test
 cglobal xavs_cpu_cpuid
-cglobal xavs_emms
 
 ALIGN 16
 ;-----------------------------------------------------------------------------
--- xavs-0.1.55/common/i386/deblock_inter.asm	2017-11-01 21:31:38.326451200 -0400
+++ xavs-0.1.55/common/i386/deblock_inter.asm	2017-11-01 21:31:49.060857100 -0400
@@ -9,8 +9,6 @@
 
 
 SECTION .text
-cglobal xavs_deblock_v_chroma_mmxext
-cglobal xavs_deblock_h_chroma_mmxext
 
 
 ; out: %4 = |%1-%2|>%3
