diff -Naur libtre-orig/configure.ac libtre/configure.ac
--- libtre-orig/configure.ac	2015-01-13 21:37:45.236000000 +0300
+++ libtre/configure.ac	2015-01-13 22:03:05.152200000 +0300
@@ -2,10 +2,11 @@
 AC_INIT(TRE, 0.8.0, [tre-general@lists.laurikari.net])
 AC_CONFIG_SRCDIR([lib/regcomp.c])
 AC_CONFIG_AUX_DIR(utils)
+AC_CONFIG_MACRO_DIRS([m4])
 AC_CANONICAL_TARGET
-AM_INIT_AUTOMAKE(1.9.0)
-AC_PREREQ(2.59)
-AM_GNU_GETTEXT_VERSION(0.17)
+AM_INIT_AUTOMAKE([subdir-objects])
+AC_PREREQ(2.69)
+AM_GNU_GETTEXT_VERSION(0.18)
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -433,16 +433,6 @@
       [ $tre_includes ])
   fi
 fi
-
-case $host in
-  *-mingw*)
-    dnl wcsrtombs and wcstombs don't seem to work at all on MinGW.
-    if test "$tre_libutf8" != "yes"; then
-      tre_wchar_reason="Not supported on MinGW"
-      tre_wchar="no ($tre_wchar_reason)"
-    fi
-    ;;
-esac
 
 # Fail if wide character support was specifically requested but is
 # not supported on this system.
diff -Naur libtre-orig/src/Makefile.am libtre/src/Makefile.am
--- libtre-orig/src/Makefile.am	2015-01-13 21:37:45.267200000 +0300
+++ libtre/src/Makefile.am	2015-01-13 21:54:03.773400000 +0300
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 localedir = $(datadir)/locale
-INCLUDES = -I$(top_srcdir)/lib
+AM_CPPFLAGS = -I$(top_srcdir)/lib
 
 if TRE_AGREP
 bin_PROGRAMS = agrep
diff -Naur libtre-orig/tests/Makefile.am libtre/tests/Makefile.am
--- libtre-orig/tests/Makefile.am	2015-01-13 21:37:45.267200000 +0300
+++ libtre/tests/Makefile.am	2015-01-13 21:52:35.508600000 +0300
@@ -59,7 +59,7 @@
   randtest_LDADD = libxtre.la $(LDADD)
 endif !TRE_DEBUG
 
-INCLUDES = -I$(top_srcdir)/lib
+AM_CPPFLAGS = -I$(top_srcdir)/lib
 
 EXTRA_DIST = build-tests.sh
 
