--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -22,12 +22,6 @@
     OUTPUT_NAME
     "leptonica$<$<BOOL:${MSVC}>:-${PROJECT_VERSION}$<$<CONFIG:DEBUG>:d>>")
 
-if(MINGW)
-  set_target_properties(
-    leptonica PROPERTIES SUFFIX
-                         "-${PROJECT_VERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-endif(MINGW)
-
 if(BUILD_SHARED_LIBS)
   target_compile_definitions(leptonica PRIVATE -DLIBLEPT_EXPORTS)
 endif()
