--- a/src/sequencehelpers_p.h
+++ b/src/sequencehelpers_p.h
@@ -9,15 +9,15 @@
 
 namespace Utils
 {
-KGLOBALACCEL_EXPORT QKeySequence reverseKey(const QKeySequence &key);
+QKeySequence reverseKey(const QKeySequence &key);
 
-KGLOBALACCEL_EXPORT QKeySequence cropKey(const QKeySequence &key, int count);
+QKeySequence cropKey(const QKeySequence &key, int count);
 
 bool contains(const QKeySequence &key, const QKeySequence &other);
 
-KGLOBALACCEL_EXPORT bool matchSequences(const QKeySequence &key, const QList<QKeySequence> &keys);
+bool matchSequences(const QKeySequence &key, const QList<QKeySequence> &keys);
 
-KGLOBALACCEL_EXPORT QKeySequence mangleKey(const QKeySequence &key);
+QKeySequence mangleKey(const QKeySequence &key);
 }
 
 #endif // SEQUENCEHELPERS_H
