--- openal-soft-1.25.0/CMakeLists.txt.orig	2025-12-23 18:23:46.989333600 +0100
+++ openal-soft-1.25.0/CMakeLists.txt	2025-12-23 18:26:03.616070900 +0100
@@ -1679,7 +1681,7 @@
 
     add_library(${IMPL_TARGET} SHARED ${OPENAL_OBJS} ${ALC_OBJS} ${CORE_OBJS} ${RC_CONFIG}
         ${TARGET_PUBLIC_HEADERS})
-    if(WIN32)
+    if(WIN32 AND NOT MINGW)
         set_target_properties(${IMPL_TARGET} PROPERTIES PREFIX "")
     endif()
     target_link_libraries(${IMPL_TARGET} PRIVATE alsoft.common ${LINKER_FLAGS} ${EXTRA_LIBS}
