--- openal-soft-1.16.0/Alc/midi/fluidsynth.c.orig	2014-08-15 07:06:59.000000000 +0000
+++ openal-soft-1.16.0/Alc/midi/fluidsynth.c	2015-01-18 13:14:18.962605800 +0000
@@ -96,7 +96,7 @@
     ALboolean ret = AL_TRUE;
     if(!fsynth_handle)
     {
-        fsynth_handle = LoadLib("libfluidsynth.so.1");
+        fsynth_handle = LoadLib("libfluidsynth-1.dll");
         if(!fsynth_handle) return AL_FALSE;
 
 #define LOAD_FUNC(x) do {                                                     \
