--- openal-soft-1.21.0/CMakeLists.txt.orig	2015-11-03 09:47:33.546783000 +0300
+++ openal-soft-1.21.0/CMakeLists.txt	2015-11-03 09:48:28.685268600 +0300
@@ -825,6 +825,7 @@
 
 # Check OSS backend
 option(ALSOFT_REQUIRE_OSS "Require OSS backend" OFF)
+if(NOT WIN32)
 find_package(OSS)
 if(OSS_FOUND)
     option(ALSOFT_BACKEND_OSS "Enable OSS backend" ON)
@@ -841,6 +842,7 @@
 if(ALSOFT_REQUIRE_OSS AND NOT HAVE_OSS)
     message(FATAL_ERROR "Failed to enabled required OSS backend")
 endif()
+endif()
 
 # Check Solaris backend
 option(ALSOFT_REQUIRE_SOLARIS "Require Solaris backend" OFF)
