--- openal-1.25.2/CMakeLists.txt.orig	2026-06-23 20:08:41.022110400 +0800
+++ openal-1.25.2/CMakeLists.txt	2026-06-23 20:10:37.118037000 +0800
@@ -648,7 +648,7 @@
 # Some systems need libm for some math functions to work
 set(MATH_LIB )
 check_library_exists(m pow "" HAVE_LIBM)
-if(HAVE_LIBM)
+if(HAVE_LIBM AND NOT MINGW)
     list(APPEND MATH_LIB m)
     list(APPEND CMAKE_REQUIRED_LIBRARIES m)
 endif()
--- openal-soft-openal-soft-1.21.0/cmake/FindMySOFA.cmake.orig	2020-03-04 13:48:39.214749300 +0300
+++ openal-soft-openal-soft-1.21.0/cmake/FindMySOFA.cmake	2020-03-04 13:49:02.945683400 +0300
@@ -49,9 +49,11 @@
              DOC "The MySOFA library"
 )
 
+if(NOT MINGW)
 find_library(MYSOFA_M_LIBRARY NAMES m
              DOC "The math library for MySOFA"
 )
+endif()
 
 # handle the QUIETLY and REQUIRED arguments and set MYSOFA_FOUND to TRUE if
 # all listed variables are TRUE
