--- openal-soft-openal-soft-1.21.0/CMakeLists.txt.orig	2020-03-05 11:45:03.064395200 +0300
+++ openal-soft-openal-soft-1.21.0/CMakeLists.txt	2020-03-05 11:49:21.032159600 +0300
@@ -884,11 +884,13 @@
 if(WIN32)
     set(WINSDK_LIB_DIRS )
     set(WINSDK_INCLUDE_DIRS )
+    if(MSVC)
     find_package(WindowsSDK)
     if(WINDOWSSDK_FOUND)
         get_windowssdk_library_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_LIB_DIRS)
         get_windowssdk_include_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_INCLUDE_DIRS)
     endif()
+    endif()
 
     # Check MMSystem backend
     check_include_files("windows.h;mmsystem.h" HAVE_MMSYSTEM_H)
--- openal-soft-openal-soft-1.21.0/cmake/FindDSound.cmake.orig	2020-03-05 11:45:03.064395200 +0300
+++ openal-soft-openal-soft-1.21.0/cmake/FindDSound.cmake	2020-03-05 11:49:21.032159600 +0300
@@ -8,7 +8,7 @@
 #   DSOUND_LIBRARY     - the dsound library
 #
 
-if (WIN32)
+if (MSVC)
   FIND_PACKAGE(WindowsSDK)
   if (WINDOWSSDK_FOUND)
     get_windowssdk_library_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_LIB_DIRS)
