--- openal-soft-openal-soft-1.20.1/CMakeLists.txt.orig	2020-03-05 11:45:03.064395200 +0300
+++ openal-soft-openal-soft-1.20.1/CMakeLists.txt	2020-03-05 11:49:21.032159600 +0300
@@ -860,20 +860,26 @@
 IF(WIN32)
     SET(WINSDK_LIB_DIRS )
     SET(WINSDK_INCLUDE_DIRS )
+    IF(MSVC)
     FIND_PACKAGE(WindowsSDK)
     IF(WINDOWSSDK_FOUND)
         get_windowssdk_library_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_LIB_DIRS)
         get_windowssdk_include_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_INCLUDE_DIRS)
     ENDIF()
+    ENDIF()
 
     SET(OLD_REQUIRED_DEFINITIONS ${CMAKE_REQUIRED_DEFINITIONS})
     SET(CMAKE_REQUIRED_DEFINITIONS ${CMAKE_REQUIRED_DEFINITIONS} -D_WIN32_WINNT=0x0502)
 
     # Check MMSystem backend
     CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H)
+    IF(MSVC)
     FIND_LIBRARY(WINMM_LIBRARY NAMES winmm
         PATHS ${WINSDK_LIB_DIRS}
         PATH_SUFFIXES lib lib/x86 lib/x64)
+    ELSE()
+        SET(WINMM_LIBRARY winmm)
+    ENDIF()
     IF(HAVE_MMSYSTEM_H AND WINMM_LIBRARY)
         OPTION(ALSOFT_BACKEND_WINMM "Enable Windows Multimedia backend" ON)
         IF(ALSOFT_BACKEND_WINMM)
