--- a/hclient.c
+++ b/hclient.c
@@ -3692,10 +3692,10 @@ DestroyDeviceListCallback(
 }
 
 DWORD WINAPI
-AsynchReadThreadProc( 
-    PREAD_THREAD_CONTEXT    Context
-)
+AsynchReadThreadProc(
+    LPVOID c)
 {
+    PREAD_THREAD_CONTEXT Context = c;
     HANDLE  completionEvent;
     BOOL    readStatus;
     DWORD   waitStatus;
@@ -3818,9 +3818,9 @@ AsyncRead_End:
 
 DWORD WINAPI
 SynchReadThreadProc(
-    PREAD_THREAD_CONTEXT    Context
-)
+    LPVOID c)
 {
+    PREAD_THREAD_CONTEXT Context = c;
     ULONG   numReadsDone = 0;
     do 
     {
--- a/hclient.h
+++ b/hclient.h
@@ -140,12 +140,12 @@ vDisplayValueAttributes(
 
 DWORD WINAPI
 AsynchReadThreadProc(
-    PREAD_THREAD_CONTEXT    Context
+    LPVOID
 );
 
 DWORD WINAPI
 SynchReadThreadProc(
-    PREAD_THREAD_CONTEXT    Context
+    LPVOID
 );
 
 #endif
