--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -44,7 +44,7 @@
 
 # define the install directory for miscellaneous files
 if(SFML_OS_WINDOWS)
-    set(INSTALL_MISC_DIR .)
+    set(INSTALL_MISC_DIR share/CSFML)
 elseif(SFML_OS_LINUX OR SFML_OS_FREEBSD OR SFML_OS_MACOSX)
     set(INSTALL_MISC_DIR share/CSFML)
 endif()
--- a/cmake/Macros.cmake
+++ b/cmake/Macros.cmake
@@ -35,7 +35,7 @@
 
             # on Windows/gcc get rid of "lib" prefix for shared libraries,
             # and transform the ".dll.a" suffix into ".a" for import libraries
-            if (SFML_COMPILER_GCC OR SFML_COMPILER_CLANG)
+            if (FALSE)
                 set_target_properties(${target} PROPERTIES PREFIX "")
                 set_target_properties(${target} PROPERTIES IMPORT_SUFFIX ".a")
             endif()
