diff --git a/src/3rdparty/chromium/media/audio/audio_manager.cc b/src/3rdparty/chromium/media/audio/audio_manager.cc
index 4febfda..40dc9e9 100644
--- a/src/3rdparty/chromium/media/audio/audio_manager.cc
+++ b/src/3rdparty/chromium/media/audio/audio_manager.cc
@@ -108,12 +108,16 @@ AudioManager::~AudioManager() {
 std::unique_ptr<AudioManager> AudioManager::Create(
     std::unique_ptr<AudioThread> audio_thread,
     AudioLogFactory* audio_log_factory) {
+#if BUILDFLAG(IS_MINGW) // missing CreateAudioManager
+  return nullptr;
+#else
   std::unique_ptr<AudioManager> manager =
       CreateAudioManager(std::move(audio_thread), audio_log_factory);
 #if BUILDFLAG(ENABLE_WEBRTC)
   manager->InitializeDebugRecording();
 #endif
   return manager;
+#endif
 }

 // static
diff --git a/src/3rdparty/chromium/media/audio/simple_sources.cc b/src/3rdparty/chromium/media/audio/simple_sources.cc
index 6fd3b7e..db56ec9 100644
--- a/src/3rdparty/chromium/media/audio/simple_sources.cc
+++ b/src/3rdparty/chromium/media/audio/simple_sources.cc
@@ -12,6 +12,7 @@
 #include <stddef.h>

 #include <algorithm>
+#include <cstring>
 #include <memory>
 #include <numbers>
 #include <string_view>
diff --git a/src/3rdparty/chromium/media/audio/win/core_audio_util_win.cc b/src/3rdparty/chromium/media/audio/win/core_audio_util_win.cc
index bd32ecb..d7f369c 100644
--- a/src/3rdparty/chromium/media/audio/win/core_audio_util_win.cc
+++ b/src/3rdparty/chromium/media/audio/win/core_audio_util_win.cc
@@ -51,5 +51,27 @@ namespace {
 constexpr uint32_t KSAUDIO_SPEAKER_UNSUPPORTED_COUW = 0xFFFFFFFF;
-
+#if defined(__MINGW32__)
+#ifndef KSAUDIO_SPEAKER_2POINT1
+#define KSAUDIO_SPEAKER_2POINT1 \
+  (KSAUDIO_SPEAKER_STEREO | SPEAKER_LOW_FREQUENCY)
+#endif
+#ifndef KSAUDIO_SPEAKER_3POINT0
+#define KSAUDIO_SPEAKER_3POINT0 \
+  (KSAUDIO_SPEAKER_STEREO | SPEAKER_FRONT_CENTER)
+#endif
+#ifndef KSAUDIO_SPEAKER_3POINT1
+#define KSAUDIO_SPEAKER_3POINT1 \
+  (KSAUDIO_SPEAKER_3POINT0 | SPEAKER_LOW_FREQUENCY)
+#endif
+#ifndef KSAUDIO_SPEAKER_5POINT0
+#define KSAUDIO_SPEAKER_5POINT0 \
+  (KSAUDIO_SPEAKER_3POINT0 | SPEAKER_SIDE_LEFT | SPEAKER_SIDE_RIGHT)
+#endif
+#ifndef KSAUDIO_SPEAKER_7POINT0
+#define KSAUDIO_SPEAKER_7POINT0 \
+  (KSAUDIO_SPEAKER_5POINT0 | SPEAKER_BACK_LEFT | SPEAKER_BACK_RIGHT)
+#endif
+#endif
+
 REFERENCE_TIME BufferSizeInFramesToTimeDelta(uint32_t frames,
                                              DWORD bytes_per_sec,
                                              WORD bytes_per_frame) {
@@ -221,21 +243,27 @@ ChannelConfig GuessChannelConfig(WORD channels) {
     case 2:
       DVLOG(2) << "KSAUDIO_SPEAKER_STEREO";
       return KSAUDIO_SPEAKER_STEREO;
+#if !defined(COMPILER_GCC) // FIXME: missing KSAUDIO_SPEAKER_2POINT1
     case 3:
       DVLOG(2) << "KSAUDIO_SPEAKER_2POINT1";
       return KSAUDIO_SPEAKER_2POINT1;
+#endif
     case 4:
       DVLOG(2) << "KSAUDIO_SPEAKER_QUAD";
       return KSAUDIO_SPEAKER_QUAD;
+#if !defined(COMPILER_GCC) // FIXME: missing KSAUDIO_SPEAKER_5POINT0
     case 5:
       DVLOG(2) << "KSAUDIO_SPEAKER_5POINT0";
       return KSAUDIO_SPEAKER_5POINT0;
+#endif
     case 6:
       DVLOG(2) << "KSAUDIO_SPEAKER_5POINT1";
       return KSAUDIO_SPEAKER_5POINT1;
+#if !defined(COMPILER_GCC) // FIXME: missing KSAUDIO_SPEAKER_7POINT0
     case 7:
       DVLOG(2) << "KSAUDIO_SPEAKER_7POINT0";
       return KSAUDIO_SPEAKER_7POINT0;
+#endif
     case 8:
       DVLOG(2) << "KSAUDIO_SPEAKER_7POINT1";
       return KSAUDIO_SPEAKER_7POINT1;
@@ -1434,6 +1462,10 @@ std::pair<int, bool> CoreAudioUtil::GetVoiceProcessingEffectsAndCheckForAEC(
     return {{}, false};
   };
 
+#if defined(__MINGW32__)
+  return false_and_log_no_effect();
+#else
+
   // This check guarantees that the client has been initialized since that is
   // required by the IAudioEffectsManager to function. The effects manager will
   // only be able to report supported effects if the client is in communications
@@ -1499,6 +1531,7 @@ std::pair<int, bool> CoreAudioUtil::GetVoiceProcessingEffectsAndCheckForAEC(
   }
 
   return std::make_pair(effects, echo_cancellation_is_available);
+#endif
 }
 
 }  // namespace media
diff --git a/src/3rdparty/chromium/media/audio/win/device_enumeration_win.cc b/src/3rdparty/chromium/media/audio/win/device_enumeration_win.cc
index e2f35a5..84d05fa 100644
--- a/src/3rdparty/chromium/media/audio/win/device_enumeration_win.cc
+++ b/src/3rdparty/chromium/media/audio/win/device_enumeration_win.cc
@@ -76,6 +76,7 @@ static bool GetDeviceNamesWinImpl(EDataFlow data_flow,
     device.unique_id =
         base::WideToUTF8(static_cast<WCHAR*>(endpoint_device_id));

+#if !defined(COMPILER_GCC) // FIXME: missing Microsoft::WRL::ComPtr<IPropertyStore>
     // Retrieve user-friendly name of endpoint device.
     // Example: "Microphone (Realtek High Definition Audio)".
     Microsoft::WRL::ComPtr<IPropertyStore> properties;
@@ -98,6 +99,7 @@ static bool GetDeviceNamesWinImpl(EDataFlow data_flow,
       if (!suffix.empty())
         device.device_name += suffix;
     }
+#endif

     // Add combination of user-friendly and unique name to the output list.
     device_names->push_back(device);
diff --git a/src/3rdparty/chromium/media/base/audio_bus.cc b/src/3rdparty/chromium/media/base/audio_bus.cc
index 410e941..914a512 100644
--- a/src/3rdparty/chromium/media/base/audio_bus.cc
+++ b/src/3rdparty/chromium/media/base/audio_bus.cc
@@ -13,6 +13,7 @@
 #include <stdint.h>

 #include <algorithm>
+#include <cstring>
 #include <limits>
 #include <utility>

diff --git a/src/3rdparty/chromium/media/base/audio_parameters.h b/src/3rdparty/chromium/media/base/audio_parameters.h
index 4219618..9ee8129 100644
--- a/src/3rdparty/chromium/media/base/audio_parameters.h
+++ b/src/3rdparty/chromium/media/base/audio_parameters.h
@@ -30,7 +30,7 @@ namespace media {
 // instead of in Audio{Input,Output}Buffer to be able to calculate size like so.
 // Use a constexpr for the alignment value that's the same as
 // AudioBus::kChannelAlignment, since MSVC doesn't accept the latter to be used.
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #pragma warning(push)
 #pragma warning(disable : 4324)  // Disable warning for added padding.
 #endif
@@ -61,7 +61,7 @@ struct MEDIA_SHMEM_EXPORT alignas(kParametersAlignment)
   uint32_t bitstream_data_size;
   uint32_t bitstream_frames;
 };
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #pragma warning(pop)
 #endif

diff --git a/src/3rdparty/chromium/media/base/win/BUILD.gn b/src/3rdparty/chromium/media/base/win/BUILD.gn
index 5571e2d..3f5cd3c 100644
--- a/src/3rdparty/chromium/media/base/win/BUILD.gn
+++ b/src/3rdparty/chromium/media/base/win/BUILD.gn
@@ -7,12 +7,14 @@ import("//build/config/jumbo.gni")
 assert(is_win)

 config("delay_load_mf") {
-  ldflags = [
-    "/DELAYLOAD:d3d11.dll",
-    "/DELAYLOAD:mf.dll",
-    "/DELAYLOAD:mfplat.dll",
-    "/DELAYLOAD:mfreadwrite.dll",
-  ]
+  if (is_msvc) {
+    ldflags = [
+      "/DELAYLOAD:d3d11.dll",
+      "/DELAYLOAD:mf.dll",
+      "/DELAYLOAD:mfplat.dll",
+      "/DELAYLOAD:mfreadwrite.dll",
+    ]
+  }
 }

 jumbo_component("media_foundation_util") {
@@ -34,14 +36,25 @@ jumbo_component("media_foundation_util") {
     "//media/gpu:command_buffer_helper",
     "//third_party/libyuv",
   ]
-  libs = [
-    "d3d11.lib",
-    "mf.lib",
-    "mfplat.lib",
-    "mfreadwrite.lib",
-    "dxguid.lib",
-    "dxgi.lib",
-  ]
+  if (is_msvc) {
+    libs = [
+      "d3d11.lib",
+      "mf.lib",
+      "mfplat.lib",
+      "mfreadwrite.lib",
+      "dxguid.lib",
+      "dxgi.lib",
+    ]
+  } else {
+    libs = [
+      "d3d11",
+      "mf",
+      "mfplat",
+      "mfreadwrite",
+      "dxguid",
+      "dxgi",
+    ]
+  }

   # MediaFoundation is not available on Windows N, so must be delay loaded.
   all_dependent_configs = [ ":delay_load_mf" ]
@@ -59,11 +72,19 @@ component("color_space_util_win") {
     "//media/base",
     "//ui/gfx:color_space",
   ]
-  libs = [
-    "mf.lib",
-    "mfplat.lib",
-    "mfreadwrite.lib",
-  ]
+  if (is_msvc) {
+    libs = [
+      "mf.lib",
+      "mfplat.lib",
+      "mfreadwrite.lib",
+    ]
+  } else {
+    libs = [
+      "mf",
+      "mfplat",
+      "mfreadwrite",
+    ]
+  }

   # MediaFoundation is not available on Windows N, so must be delay loaded.
   all_dependent_configs = [ ":delay_load_mf" ]
diff --git a/src/3rdparty/chromium/media/base/win/color_space_util_win.cc b/src/3rdparty/chromium/media/base/win/color_space_util_win.cc
index d6bad9a..3d603f5 100644
--- a/src/3rdparty/chromium/media/base/win/color_space_util_win.cc
+++ b/src/3rdparty/chromium/media/base/win/color_space_util_win.cc
@@ -9,6 +9,33 @@
 
 #include <mfapi.h>
 
+#if defined(__MINGW32__)
+#if !defined(MFVideoPrimaries_Display_P3)
+#define MFVideoPrimaries_Display_P3 static_cast<MFVideoPrimaries>(13)
+#endif
+
+#if !defined(MFVideoTransFunc_BT1361_ECG)
+#define MFVideoTransFunc_BT1361_ECG \
+  static_cast<MFVideoTransferFunction>(18)
+#endif
+#if !defined(MFVideoTransFunc_SMPTE428)
+#define MFVideoTransFunc_SMPTE428 static_cast<MFVideoTransferFunction>(19)
+#endif
+
+#if !defined(MFVideoTransferMatrix_Identity)
+#define MFVideoTransferMatrix_Identity \
+  static_cast<MFVideoTransferMatrix>(6)
+#endif
+#if !defined(MFVideoTransferMatrix_FCC47)
+#define MFVideoTransferMatrix_FCC47 static_cast<MFVideoTransferMatrix>(7)
+#endif
+#if !defined(MFVideoTransferMatrix_YCgCo)
+#define MFVideoTransferMatrix_YCgCo static_cast<MFVideoTransferMatrix>(8)
+#endif
+#if !defined(MFVideoTransferMatrix_SMPTE2085)
+#define MFVideoTransferMatrix_SMPTE2085 \
+  static_cast<MFVideoTransferMatrix>(9)
+#endif
+#endif
+
 #include "base/logging.h"
 #include "media/base/video_color_space.h"
 
diff --git a/src/3rdparty/chromium/media/base/win/mf_helpers.cc b/src/3rdparty/chromium/media/base/win/mf_helpers.cc
index a20de3d..b979fb1 100644
--- a/src/3rdparty/chromium/media/base/win/mf_helpers.cc
+++ b/src/3rdparty/chromium/media/base/win/mf_helpers.cc
@@ -41,7 +41,9 @@
 namespace media {

 using Microsoft::WRL::ComPtr;
+#if !defined(COMPILER_GCC) // FIXME: missing Microsoft::WRL::MakeAndInitialize
 using Microsoft::WRL::MakeAndInitialize;
+#endif

 namespace {

@@ -147,6 +149,7 @@ HRESULT AddEncryptAttributes(const DecryptConfig& decrypt_config,
                              GUID* last_key_id) {
   DVLOG(3) << __func__;

+#if !defined(COMPILER_GCC) // FIXME: missing MFSampleEncryptionProtectionScheme
   MFSampleEncryptionProtectionScheme mf_protection_scheme;
   if (decrypt_config.encryption_scheme() == EncryptionScheme::kCenc) {
     mf_protection_scheme = MFSampleEncryptionProtectionScheme::
@@ -202,6 +205,7 @@ HRESULT AddEncryptAttributes(const DecryptConfig& decrypt_config,
   GUID key_id = GetGUIDFromString(decrypt_config.key_id());
   RETURN_IF_FAILED(mf_sample->SetGUID(MFSampleExtension_Content_KeyID, key_id));
   *last_key_id = key_id;
+#endif

   // IV
   size_t iv_length = decrypt_config.iv().length();
@@ -217,6 +221,7 @@ HRESULT AddEncryptAttributes(const DecryptConfig& decrypt_config,
       reinterpret_cast<const uint8_t*>(decrypt_config.iv().c_str()),
       iv_length));

+#if !defined(COMPILER_GCC) // FIXME: missing MFSampleExtension_Encryption_SubSample_Mapping
   // Handle subsample entries.
   const auto& subsample_entries = decrypt_config.subsamples();
   if (subsample_entries.empty()) {
@@ -235,6 +240,7 @@ HRESULT AddEncryptAttributes(const DecryptConfig& decrypt_config,
       MFSampleExtension_Encryption_SubSample_Mapping,
       reinterpret_cast<const uint8_t*>(mf_subsample_entries.data()),
       mf_sample_entries_size));
+#endif

   return S_OK;
 }
@@ -324,24 +330,34 @@ ChannelLayout ChannelConfigToChannelLayout(ChannelConfig config) {
       return CHANNEL_LAYOUT_MONO;
     case KSAUDIO_SPEAKER_STEREO:
       return CHANNEL_LAYOUT_STEREO;
+#if defined(KSAUDIO_SPEAKER_2POINT1)
     case KSAUDIO_SPEAKER_2POINT1:
       return CHANNEL_LAYOUT_2POINT1;
+#endif
+#if defined(KSAUDIO_SPEAKER_3POINT0)
     case KSAUDIO_SPEAKER_3POINT0:
       return CHANNEL_LAYOUT_SURROUND;
+#endif
+#if defined(KSAUDIO_SPEAKER_3POINT1)
     case KSAUDIO_SPEAKER_3POINT1:
       return CHANNEL_LAYOUT_3_1;
+#endif
     case KSAUDIO_SPEAKER_QUAD:
       return CHANNEL_LAYOUT_QUAD;
     case KSAUDIO_SPEAKER_SURROUND:
       return CHANNEL_LAYOUT_4_0;
+#if defined(KSAUDIO_SPEAKER_5POINT0)
     case KSAUDIO_SPEAKER_5POINT0:
       return CHANNEL_LAYOUT_5_0;
+#endif
     case KSAUDIO_SPEAKER_5POINT1:
       return CHANNEL_LAYOUT_5_1_BACK;
     case KSAUDIO_SPEAKER_5POINT1_SURROUND:
       return CHANNEL_LAYOUT_5_1;
+#if defined(KSAUDIO_SPEAKER_7POINT0)
     case KSAUDIO_SPEAKER_7POINT0:
       return CHANNEL_LAYOUT_7_0;
+#endif
     case KSAUDIO_SPEAKER_7POINT1:
       return CHANNEL_LAYOUT_7_1_WIDE_BACK;
     case KSAUDIO_SPEAKER_7POINT1_SURROUND:
@@ -710,6 +726,7 @@ HRESULT CreateDecryptConfigFromSample(

   EncryptionScheme encryption_scheme = EncryptionScheme::kUnencrypted;
   UINT32 mf_protection_scheme = 0;
+#if !defined(COMPILER_GCC) // FIXME: missing MFSampleExtension_Encryption_ProtectionScheme
   RETURN_IF_FAILED(mf_sample->GetUINT32(
       MFSampleExtension_Encryption_ProtectionScheme, &mf_protection_scheme));
   switch (mf_protection_scheme) {
@@ -728,6 +745,7 @@ HRESULT CreateDecryptConfigFromSample(
                   << mf_protection_scheme;
       return MF_E_UNEXPECTED;
   }
+#endif

   // IV
   UINT32 iv_length = 0;
@@ -754,6 +772,7 @@ HRESULT CreateDecryptConfigFromSample(
   base::win::ScopedCoMem<MediaFoundationSubsampleEntry> subsample_mappings;
   uint32_t subsample_mappings_size = 0;

+#if !defined(COMPILER_GCC) // FIXME: missing MFSampleExtension_Encryption_SubSample_Mapping
   // If `MFSampleExtension_Encryption_SubSample_Mapping` attribute doesn't
   // exist, we should not fail the call. i.e., Encrypted audio content.
   if (SUCCEEDED(mf_sample->GetAllocatedBlob(
@@ -772,6 +791,7 @@ HRESULT CreateDecryptConfigFromSample(
       }
     }
   }
+#endif

   // Key ID
   const auto key_id_string = GetStringFromGUID(key_id);
@@ -783,6 +803,7 @@ HRESULT CreateDecryptConfigFromSample(
   if (encryption_scheme == EncryptionScheme::kCenc) {
     *decrypt_config =
         DecryptConfig::CreateCencConfig(key_id_string, iv_string, subsamples);
+#if !defined(COMPILER_GCC) // FIXME: missing MFSampleExtension_Encryption_*
   } else {
     EncryptionPattern encryption_pattern;

@@ -802,6 +823,7 @@ HRESULT CreateDecryptConfigFromSample(
     DVLOG(3) << __func__ << ": encryption_pattern=" << encryption_pattern;
     *decrypt_config = DecryptConfig::CreateCbcsConfig(
         key_id_string, iv_string, subsamples, encryption_pattern);
+#endif
   }

   return S_OK;
diff --git a/src/3rdparty/chromium/media/capture/video/create_video_capture_device_factory.cc b/src/3rdparty/chromium/media/capture/video/create_video_capture_device_factory.cc
index f00517a..ae07584 100644
--- a/src/3rdparty/chromium/media/capture/video/create_video_capture_device_factory.cc
+++ b/src/3rdparty/chromium/media/capture/video/create_video_capture_device_factory.cc
@@ -19,7 +19,7 @@
 #include "media/capture/video/chromeos/public/cros_features.h"
 #include "media/capture/video/chromeos/video_capture_device_factory_chromeos.h"
 #include "media/capture/video/linux/video_capture_device_factory_linux.h"
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
 #include "media/capture/video/win/video_capture_device_factory_win.h"
 #elif BUILDFLAG(IS_APPLE)
 #include "media/capture/video/apple/video_capture_device_factory_apple.h"
@@ -62,7 +62,7 @@ CreatePlatformSpecificVideoCaptureDeviceFactory(
   if (base::SysInfo::IsRunningOnChromeOS())
     return std::make_unique<VideoCaptureDeviceFactoryChromeOS>(ui_task_runner);
   return std::make_unique<VideoCaptureDeviceFactoryLinux>(ui_task_runner);
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
   return std::make_unique<VideoCaptureDeviceFactoryWin>();
 #elif BUILDFLAG(IS_APPLE)
   return std::make_unique<VideoCaptureDeviceFactoryApple>();
diff --git a/src/3rdparty/chromium/media/capture/video/video_capture_buffer_tracker_factory_impl.cc b/src/3rdparty/chromium/media/capture/video/video_capture_buffer_tracker_factory_impl.cc
index a6c9565..9842145 100644
--- a/src/3rdparty/chromium/media/capture/video/video_capture_buffer_tracker_factory_impl.cc
+++ b/src/3rdparty/chromium/media/capture/video/video_capture_buffer_tracker_factory_impl.cc
@@ -24,7 +24,7 @@ namespace media {

 VideoCaptureBufferTrackerFactoryImpl::VideoCaptureBufferTrackerFactoryImpl() {}

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 VideoCaptureBufferTrackerFactoryImpl::VideoCaptureBufferTrackerFactoryImpl(
     scoped_refptr<DXGIDeviceManager> dxgi_device_manager)
     : dxgi_device_manager_(std::move(dxgi_device_manager)) {}
@@ -44,7 +44,7 @@ VideoCaptureBufferTrackerFactoryImpl::CreateTracker(
       return std::make_unique<GpuMemoryBufferTrackerApple>();
 #elif BUILDFLAG(IS_LINUX)
       return std::make_unique<V4L2GpuMemoryBufferTracker>();
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
       if (!dxgi_device_manager_) {
         return nullptr;
       }
@@ -53,7 +53,7 @@ VideoCaptureBufferTrackerFactoryImpl::CreateTracker(
       return nullptr;
 #endif
     default:
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
+#if BUILDFLAG(IS_MSVC) || BUILDFLAG(IS_MAC)
       // Since Windows and macOS capturer outputs NV12 only for GMBs and I420
       // for software frames, the pixel format is used to choose between shmem
       // and gmb trackers. Therefore I420 shmem trackers must not be reusable
diff --git a/src/3rdparty/chromium/media/capture/video/win/BUILD.gn b/src/3rdparty/chromium/media/capture/video/win/BUILD.gn
index a06417b..ae4a334 100644
--- a/src/3rdparty/chromium/media/capture/video/win/BUILD.gn
+++ b/src/3rdparty/chromium/media/capture/video/win/BUILD.gn
@@ -34,6 +34,22 @@ source_set("win") {
     "video_capture_device_win.cc",
     "video_capture_device_win.h",
   ]
+  if (is_mingw) {
+    sources -= [
+      "gpu_memory_buffer_tracker_win.cc",
+      "gpu_memory_buffer_tracker_win.h",
+      "sink_filter_win.cc",
+      "sink_filter_win.h",
+      "sink_input_pin_win.cc",
+      "sink_input_pin_win.h",
+      "video_capture_device_factory_win.cc",
+      "video_capture_device_factory_win.h",
+      "video_capture_device_mf_win.cc",
+      "video_capture_device_mf_win.h",
+      "video_capture_device_win.cc",
+      "video_capture_device_win.h",
+    ]
+  }

   public_deps = [
     "//media/capture:capture_base",
@@ -59,17 +75,19 @@ source_set("win") {
   ]

   libs = [
-    "mf.lib",
-    "mfplat.lib",
-    "mfreadwrite.lib",
-    "mfuuid.lib",
+    "mf",
+    "mfplat",
+    "mfreadwrite",
+    "mfuuid",
   ]

-  ldflags = [
-    "/DELAYLOAD:mf.dll",
-    "/DELAYLOAD:mfplat.dll",
-    "/DELAYLOAD:mfreadwrite.dll",
-  ]
+  if (is_msvc) {
+    ldflags = [
+      "/DELAYLOAD:mf.dll",
+      "/DELAYLOAD:mfplat.dll",
+      "/DELAYLOAD:mfreadwrite.dll",
+    ]
+  }
 }

 source_set("win_unittests") {
@@ -100,17 +118,19 @@ source_set("win_unittests") {
   ]

   libs = [
-    "mf.lib",
-    "mfplat.lib",
-    "mfreadwrite.lib",
-    "mfuuid.lib",
-    "dxguid.lib",
-  ]
-  ldflags = [
-    "/DELAYLOAD:mf.dll",
-    "/DELAYLOAD:mfplat.dll",
-    "/DELAYLOAD:mfreadwrite.dll",
+    "mf",
+    "mfplat",
+    "mfreadwrite",
+    "mfuuid",
+    "dxguid",
   ]
+  if (is_msvc) {
+    ldflags = [
+      "/DELAYLOAD:mf.dll",
+      "/DELAYLOAD:mfplat.dll",
+      "/DELAYLOAD:mfreadwrite.dll",
+    ]
+  }

   testonly = true
 }
diff --git a/src/3rdparty/chromium/media/capture/video/win/video_capture_device_factory_win.h b/src/3rdparty/chromium/media/capture/video/win/video_capture_device_factory_win.h
index b637a3b..71b1a38 100644
--- a/src/3rdparty/chromium/media/capture/video/win/video_capture_device_factory_win.h
+++ b/src/3rdparty/chromium/media/capture/video/win/video_capture_device_factory_win.h
@@ -132,7 +132,9 @@ class CAPTURE_EXPORT VideoCaptureDeviceFactoryWin

   class UsageReportHandler;
   scoped_refptr<UsageReportHandler> report_handler_;
+#if !defined(COMPILER_GCC) // FIXME: missing Microsoft::WRL::ComPtr<IMFSensorActivityMonitor>
   Microsoft::WRL::ComPtr<IMFSensorActivityMonitor> monitor_;
+#endif

   base::WeakPtrFactory<VideoCaptureDeviceFactoryWin> weak_ptr_factory_{this};
 };
diff --git a/src/3rdparty/chromium/media/cdm/BUILD.gn b/src/3rdparty/chromium/media/cdm/BUILD.gn
index 06cafd0..6753eff 100644
--- a/src/3rdparty/chromium/media/cdm/BUILD.gn
+++ b/src/3rdparty/chromium/media/cdm/BUILD.gn
@@ -128,7 +128,7 @@ source_set("cdm") {
     public_deps += [ "//components/crash/core/common:crash_key" ]
   }

-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "media_foundation_cdm_data.cc",
       "media_foundation_cdm_data.h",
diff --git a/src/3rdparty/chromium/media/cdm/api/content_decryption_module_export.h b/src/3rdparty/chromium/media/cdm/api/content_decryption_module_export.h
index 932708d..daea1c8 100644
--- a/src/3rdparty/chromium/media/cdm/api/content_decryption_module_export.h
+++ b/src/3rdparty/chromium/media/cdm/api/content_decryption_module_export.h
@@ -7,7 +7,7 @@

 // Define CDM_API so that functionality implemented by the CDM module
 // can be exported to consumers.
-#if defined(_WIN32)
+#if defined(_MSC_VER)

 #if defined(CDM_IMPLEMENTATION)
 #define CDM_API __declspec(dllexport)
@@ -15,9 +15,9 @@
 #define CDM_API __declspec(dllimport)
 #endif  // defined(CDM_IMPLEMENTATION)

-#else  // defined(_WIN32)
+#else  // defined(_MSC_VER)
 #define CDM_API __attribute__((visibility("default")))
-#endif  // defined(_WIN32)
+#endif  // defined(_MSC_VER)

 // Define CDM_CLASS_API to export class types. We have to add visibility
 // attributes to make sure virtual tables in CDM consumer and CDM implementation
@@ -25,14 +25,14 @@
 // about that for the internal symbols, but it has only become a practical issue
 // after introduction of LTO devirtualization. See more details on
 // https://crbug.com/609564#c35
-#if defined(_WIN32)
+#if defined(_MSC_VER)
 #if defined(__clang__)
 #define CDM_CLASS_API [[clang::lto_visibility_public]]
 #else
 #define CDM_CLASS_API
 #endif
-#else  // defined(_WIN32)
+#else  // defined(_MSC_VER)
 #define CDM_CLASS_API __attribute__((visibility("default")))
-#endif  // defined(_WIN32)
+#endif  // defined(_MSC_VER)

 #endif  // CDM_CONTENT_DECRYPTION_MODULE_EXPORT_H_
diff --git a/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.cc b/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.cc
index a2a9125..3ad5538 100644
--- a/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.cc
+++ b/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.cc
@@ -8,11 +8,11 @@
 #include "media/base/cdm_context.h"
 #include "media/cdm/cdm_helpers.h"

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include <optional>

 #include "media/cdm/media_foundation_cdm_data.h"
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)

 namespace media {

@@ -58,7 +58,7 @@ void CdmAuxiliaryHelper::GetStorageId(uint32_t version, StorageIdCB callback) {
   std::move(callback).Run(version, std::vector<uint8_t>());
 }

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void CdmAuxiliaryHelper::GetMediaFoundationCdmData(
     GetMediaFoundationCdmDataCB callback) {
   std::move(callback).Run(std::make_unique<MediaFoundationCdmData>(
@@ -69,6 +69,6 @@ void CdmAuxiliaryHelper::SetCdmClientToken(
     const std::vector<uint8_t>& client_token) {}

 void CdmAuxiliaryHelper::OnCdmEvent(CdmEvent event, HRESULT hresult) {}
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)

 }  // namespace media
diff --git a/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.h b/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.h
index d3dc1a8..b3cc53e 100644
--- a/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.h
+++ b/src/3rdparty/chromium/media/cdm/cdm_auxiliary_helper.h
@@ -91,11 +91,11 @@ class MEDIA_EXPORT CdmAuxiliaryHelper : public CdmAllocator,
                          ChallengePlatformCB callback) override;
   void GetStorageId(uint32_t version, StorageIdCB callback) override;

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   void GetMediaFoundationCdmData(GetMediaFoundationCdmDataCB callback) override;
   void SetCdmClientToken(const std::vector<uint8_t>& client_token) override;
   void OnCdmEvent(CdmEvent event, HRESULT hresult) override;
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 };

 }  // namespace media
diff --git a/src/3rdparty/chromium/media/cdm/cdm_document_service.h b/src/3rdparty/chromium/media/cdm/cdm_document_service.h
index 4daf875..e8d8119 100644
--- a/src/3rdparty/chromium/media/cdm/cdm_document_service.h
+++ b/src/3rdparty/chromium/media/cdm/cdm_document_service.h
@@ -52,10 +52,10 @@ class MEDIA_EXPORT CdmDocumentService {
       base::OnceCallback<void(uint32_t version,
                               const std::vector<uint8_t>& storage_id)>;

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   using GetMediaFoundationCdmDataCB =
       base::OnceCallback<void(std::unique_ptr<MediaFoundationCdmData>)>;
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)

   // Allows authorized services to verify that the underlying platform is
   // trusted. An example of a trusted platform is a Chrome OS device in
@@ -82,7 +82,7 @@ class MEDIA_EXPORT CdmDocumentService {
   //                 version does not exist.
   virtual void GetStorageId(uint32_t version, StorageIdCB callback) = 0;

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   // Gets the Media Foundation cdm data for the origin associated with the CDM.
   virtual void GetMediaFoundationCdmData(
       GetMediaFoundationCdmDataCB callback) = 0;
@@ -97,7 +97,7 @@ class MEDIA_EXPORT CdmDocumentService {
   // logic, e.g. disable the CDM in the current robustness level. For error
   // events, the `hresult` provides more details about the error.
   virtual void OnCdmEvent(CdmEvent event, HRESULT hresult) = 0;
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 };

 }  // namespace media
diff --git a/src/3rdparty/chromium/media/cdm/mock_helpers.h b/src/3rdparty/chromium/media/cdm/mock_helpers.h
index a8b290b..860636c 100644
--- a/src/3rdparty/chromium/media/cdm/mock_helpers.h
+++ b/src/3rdparty/chromium/media/cdm/mock_helpers.h
@@ -57,12 +57,12 @@ class MockCdmAuxiliaryHelper : public CdmAuxiliaryHelper {

   MOCK_METHOD(void, RecordUkm, (const CdmMetricsData&), (override));

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   MOCK_METHOD(void,
               GetMediaFoundationCdmData,
               (GetMediaFoundationCdmDataCB callback),
               (override));
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)

  private:
   std::unique_ptr<CdmAllocator> allocator_;
diff --git a/src/3rdparty/chromium/media/cdm/win/media_foundation_cdm.h b/src/3rdparty/chromium/media/cdm/win/media_foundation_cdm.h
index 0d17455..71b10f3 100644
--- a/src/3rdparty/chromium/media/cdm/win/media_foundation_cdm.h
+++ b/src/3rdparty/chromium/media/cdm/win/media_foundation_cdm.h
@@ -5,6 +5,8 @@
 #ifndef MEDIA_CDM_WIN_MEDIA_FOUNDATION_CDM_H_
 #define MEDIA_CDM_WIN_MEDIA_FOUNDATION_CDM_H_

+#if defined(COMPILER_MSVC)
+
 #include <mfcontentdecryptionmodule.h>
 #include <wrl.h>

@@ -187,4 +189,6 @@ class MEDIA_EXPORT MediaFoundationCdm final : public ContentDecryptionModule,

 }  // namespace media

+#endif
+
 #endif  // MEDIA_CDM_WIN_MEDIA_FOUNDATION_CDM_H_
diff --git a/src/3rdparty/chromium/media/device_monitors/BUILD.gn b/src/3rdparty/chromium/media/device_monitors/BUILD.gn
index 8be9855..b637491 100644
--- a/src/3rdparty/chromium/media/device_monitors/BUILD.gn
+++ b/src/3rdparty/chromium/media/device_monitors/BUILD.gn
@@ -37,7 +37,7 @@ source_set("device_monitors") {
     frameworks = [ "Foundation.framework" ]
   }

-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "system_message_window_win.cc",
       "system_message_window_win.h",
diff --git a/src/3rdparty/chromium/media/filters/BUILD.gn b/src/3rdparty/chromium/media/filters/BUILD.gn
index 60aaa65..f0fd5ef 100644
--- a/src/3rdparty/chromium/media/filters/BUILD.gn
+++ b/src/3rdparty/chromium/media/filters/BUILD.gn
@@ -206,7 +206,7 @@ jumbo_source_set("filters") {
       "passthrough_dts_audio_decoder.h",
     ]
   }
-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "win/media_foundation_audio_decoder.cc",
       "win/media_foundation_audio_decoder.h",
diff --git a/src/3rdparty/chromium/media/filters/ffmpeg_demuxer.cc b/src/3rdparty/chromium/media/filters/ffmpeg_demuxer.cc
index 90fb46e..65f301d 100644
--- a/src/3rdparty/chromium/media/filters/ffmpeg_demuxer.cc
+++ b/src/3rdparty/chromium/media/filters/ffmpeg_demuxer.cc
@@ -125,6 +125,7 @@ static base::TimeDelta ExtractStartTime(AVStream* stream) {
   if (stream->start_time != kNoFFmpegTimestamp)
     start_time = ConvertFromTimeBase(stream->time_base, stream->start_time);

+#if LIBAVFORMAT_VERSION_MAJOR < 61
   // Next try to use the first DTS value, for codecs where we know PTS == DTS
   // (excludes all H26x codecs). The start time must be returned in PTS.
   if (av_stream_get_first_dts(stream) != kInvalidPTSMarker &&
@@ -136,6 +137,7 @@ static base::TimeDelta ExtractStartTime(AVStream* stream) {
     if (first_pts < start_time)
       start_time = first_pts;
   }
+#endif

   return start_time;
 }
diff --git a/src/3rdparty/chromium/media/gpu/BUILD.gn b/src/3rdparty/chromium/media/gpu/BUILD.gn
index 0e64edc..53cf6d0 100644
--- a/src/3rdparty/chromium/media/gpu/BUILD.gn
+++ b/src/3rdparty/chromium/media/gpu/BUILD.gn
@@ -284,6 +284,30 @@ component("gpu") {
       "windows/vp9_video_rate_control_wrapper.cc",
       "windows/vp9_video_rate_control_wrapper.h",
     ]
+    if (is_mingw) {
+      sources -= [
+        "windows/av1_video_rate_control_wrapper.cc",
+        "windows/av1_video_rate_control_wrapper.h",
+        "windows/d3d12_video_helpers.cc",
+        "windows/d3d12_video_helpers.h",
+        "windows/h264_video_rate_control_wrapper.cc",
+        "windows/h264_video_rate_control_wrapper.h",
+        "windows/media_foundation_video_encode_accelerator_win.cc",
+        "windows/media_foundation_video_encode_accelerator_win.h",
+        "windows/mf_video_encoder_shared_state.cc",
+        "windows/mf_video_encoder_shared_state.h",
+        "windows/mf_video_encoder_switches.cc",
+        "windows/mf_video_encoder_switches.h",
+        "windows/mf_video_encoder_util.cc",
+        "windows/mf_video_encoder_util.h",
+        "windows/mf_video_processor_accelerator.cc",
+        "windows/mf_video_processor_accelerator.h",
+        "windows/video_rate_control_wrapper.cc",
+        "windows/video_rate_control_wrapper.h",
+        "windows/vp9_video_rate_control_wrapper.cc",
+        "windows/vp9_video_rate_control_wrapper.h",
+      ]
+    }
     if (!is_qtwebengine) {
       sources += [
         "windows/d3d12_video_encode_accelerator.cc",
@@ -308,27 +332,40 @@ component("gpu") {
       "//third_party/libvpx:libvpxrc",
       "//ui/display",
     ]
-    libs += [
-      "d3d9.lib",
-      "d3d11.lib",
-      "d3d12.lib",
-      "dxgi.lib",
-      "dxva2.lib",
-      "dxgi.lib",
-      "strmiids.lib",
-      "mf.lib",
-      "mfplat.lib",
-      "mfuuid.lib",
-    ]
-    ldflags += [
-      "/DELAYLOAD:d3d9.dll",
-      "/DELAYLOAD:d3d11.dll",
-      "/DELAYLOAD:d3d12.dll",
-      "/DELAYLOAD:dxva2.dll",
-      "/DELAYLOAD:dxgi.dll",
-      "/DELAYLOAD:mf.dll",
-      "/DELAYLOAD:mfplat.dll",
-    ]
+    if (is_msvc) {
+      libs += [
+        "d3d9.lib",
+        "d3d11.lib",
+        "d3d12.lib",
+        "dxva2.lib",
+        "dxgi.lib",
+        "strmiids.lib",
+        "mf.lib",
+        "mfplat.lib",
+        "mfuuid.lib",
+      ]
+      ldflags += [
+        "/DELAYLOAD:d3d9.dll",
+        "/DELAYLOAD:d3d11.dll",
+        "/DELAYLOAD:d3d12.dll",
+        "/DELAYLOAD:dxva2.dll",
+        "/DELAYLOAD:dxgi.dll",
+        "/DELAYLOAD:mf.dll",
+        "/DELAYLOAD:mfplat.dll",
+      ]
+    } else {
+      libs += [
+        "d3d9",
+        "d3d11",
+        "d3d12",
+        "dxgi",
+        "dxva2",
+        "strmiids",
+        "mf",
+        "mfplat",
+        "mfuuid",
+      ]
+    }
   }

   if (use_ozone) {
@@ -689,7 +726,7 @@ source_set("unit_tests") {
     deps += [ "//media/gpu/mac:unit_tests" ]
   }

-  if (is_win || use_vaapi) {
+  if (is_msvc || use_vaapi) {
     sources += [
       "av1_builder_unittest.cc",
       "h264_builder_unittest.cc",
diff --git a/src/3rdparty/chromium/media/gpu/gpu_video_encode_accelerator_factory.cc b/src/3rdparty/chromium/media/gpu/gpu_video_encode_accelerator_factory.cc
index d36cd42..61101c7 100644
--- a/src/3rdparty/chromium/media/gpu/gpu_video_encode_accelerator_factory.cc
+++ b/src/3rdparty/chromium/media/gpu/gpu_video_encode_accelerator_factory.cc
diff --git a/src/3rdparty/chromium/media/gpu/windows/supported_profile_helpers.cc b/src/3rdparty/chromium/media/gpu/windows/supported_profile_helpers.cc
index ece8b87..fa9c3c7 100644
--- a/src/3rdparty/chromium/media/gpu/windows/supported_profile_helpers.cc
+++ b/src/3rdparty/chromium/media/gpu/windows/supported_profile_helpers.cc
@@ -10,0 +11 @@
+#include "build/build_config.h"
@@ -68,0 +70 @@ class D3D11VideoDeviceWrapper : public D3DVideoDeviceWrapper {
+#if !BUILDFLAG(IS_MINGW)
@@ -113,0 +116 @@ class D3D12VideoDeviceWrapper : public D3DVideoDeviceWrapper {
+#endif
@@ -394,0 +398,3 @@ SupportedResolutionRangeMap GetSupportedD3D12VideoDecoderResolutions(
+#if BUILDFLAG(IS_MINGW)
+  return GetSupportedD3DVideoDecoderResolutions(nullptr, workarounds);
+#else
@@ -402,0 +409 @@ SupportedResolutionRangeMap GetSupportedD3D12VideoDecoderResolutions(
+#endif
diff --git a/src/3rdparty/chromium/media/gpu/windows/d3d11_decoder_configurator.cc b/src/3rdparty/chromium/media/gpu/windows/d3d11_decoder_configurator.cc
index 72ab1b4..c277986 100644
--- a/src/3rdparty/chromium/media/gpu/windows/d3d11_decoder_configurator.cc
+++ b/src/3rdparty/chromium/media/gpu/windows/d3d11_decoder_configurator.cc
diff --git a/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_selector.cc b/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_selector.cc
index deb6529..aca4ae4 100644
--- a/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_selector.cc
+++ b/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_selector.cc
@@ -254,9 +254,11 @@ std::unique_ptr<Texture2DWrapper> CopyTextureSelector::CreateTextureWrapper(
   if (FAILED(device->CreateTexture2D(&texture_desc, nullptr, &out_texture)))
     return nullptr;

+#if defined(COMPILER_MSVC)
   if (FAILED(
           SetDebugName(out_texture.Get(), "D3D11Decoder_CopyTextureSelector")))
     return nullptr;
+#endif

   return std::make_unique<CopyingTexture2DWrapper>(
       size,
diff --git a/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_wrapper.cc b/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_wrapper.cc
index 5a0ec8e..2185868 100644
--- a/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_wrapper.cc
+++ b/src/3rdparty/chromium/media/gpu/windows/d3d11_texture_wrapper.cc
@@ -134,6 +134,7 @@ D3D11Status DefaultTexture2DWrapper::Init(
   picture_buffer_gpu_resource_init_done_cb_ =
       std::move(picture_buffer_gpu_resource_init_done_cb);

+#if defined(COMPILER_MSVC)
   // Start construction of the GpuResources.
   // We send the texture itself, since we assume that we're using the angle
   // device for decoding.  Sharing seems not to work very well.  Otherwise, we
@@ -151,6 +152,7 @@ D3D11Status DefaultTexture2DWrapper::Init(
       video_device_, texture, array_slice, std::move(picture_buffer),
       std::move(gpu_resource_init_cb));
   return D3D11Status::Codes::kOk;
+#endif
 }

 void DefaultTexture2DWrapper::OnError(D3D11Status status) {
@@ -193,6 +195,7 @@ DefaultTexture2DWrapper::GpuResources::GpuResources(

   auto* shared_image_manager = helper_->GetSharedImageManager();

+#if defined(COMPILER_MSVC)
   // Usage flags to allow the display compositor to draw from it, video to
   // decode from it, and webgl/canvas to read from it.
   gpu::SharedImageUsageSet usage =
@@ -280,6 +283,7 @@ DefaultTexture2DWrapper::GpuResources::GpuResources(
   std::move(gpu_resource_init_cb)
       .Run(std::move(picture_buffer), std::move(shared_image_rep),
            std::move(shared_image));
+#endif
 }

 DefaultTexture2DWrapper::GpuResources::~GpuResources() = default;
diff --git a/src/3rdparty/chromium/media/gpu/windows/mf_audio_encoder.cc b/src/3rdparty/chromium/media/gpu/windows/mf_audio_encoder.cc
index 0d2ec9c..f4e14fe 100644
--- a/src/3rdparty/chromium/media/gpu/windows/mf_audio_encoder.cc
+++ b/src/3rdparty/chromium/media/gpu/windows/mf_audio_encoder.cc
@@ -403,6 +403,7 @@ void MFAudioEncoder::Initialize(const Options& options,
     return;
   }

+#if !defined(COMPILER_GCC) // FIXME: missing CODECAPI_AVEncAACEnableVBR
   if (options_.bitrate_mode.has_value() &&
       options_.bitrate_mode.value() == AudioEncoder::BitrateMode::kVariable &&
       options.codec == AudioCodec::kAAC) {
@@ -421,6 +422,7 @@ void MFAudioEncoder::Initialize(const Options& options,
       }
     }
   }
+#endif

   // We skip getting the stream counts and IDs because encoders only have one
   // input and output stream, and the ID of each is always 0.
diff --git a/src/3rdparty/chromium/media/midi/BUILD.gn b/src/3rdparty/chromium/media/midi/BUILD.gn
index 0ba1991..ac0e10e 100644
--- a/src/3rdparty/chromium/media/midi/BUILD.gn
+++ b/src/3rdparty/chromium/media/midi/BUILD.gn
@@ -148,12 +148,23 @@ component("midi") {
       "midi_manager_winrt.cc",
       "midi_manager_winrt.h",
     ]
-
-    libs += [ "setupapi.lib" ]
+    if (is_mingw) {
+      sources -= [
+        "midi_manager_winrt.cc",
+        "midi_manager_winrt.h",
+      ]
+      libs += [ "setupapi" ]
+    } else {
+      libs += [ "setupapi.lib" ]
+    }

     # This library is included in base in static builds.
     if (is_component_build) {
-      libs += [ "cfgmgr32.lib" ]
+      if (is_mingw) {
+        libs += [ "cfgmgr32" ]
+      } else {
+        libs += [ "cfgmgr32.lib" ]
+      }
     }
   }

diff --git a/src/3rdparty/chromium/media/midi/midi_manager_win.cc b/src/3rdparty/chromium/media/midi/midi_manager_win.cc
index 23c31dd..8853534 100644
--- a/src/3rdparty/chromium/media/midi/midi_manager_win.cc
+++ b/src/3rdparty/chromium/media/midi/midi_manager_win.cc
@@ -899,9 +899,11 @@ void MidiManagerWin::SendOnTaskRunner(MidiManagerClient* client,
 }

 MidiManager* MidiManager::Create(MidiService* service) {
+#if defined(COMPILER_MSVC)
   if (base::FeatureList::IsEnabled(features::kMidiManagerWinrt)) {
     return new MidiManagerWinrt(service);
   }
+#endif
   return new MidiManagerWin(service);
 }

diff --git a/src/3rdparty/chromium/media/mojo/clients/BUILD.gn b/src/3rdparty/chromium/media/mojo/clients/BUILD.gn
index e5086da..e0485a6 100644
--- a/src/3rdparty/chromium/media/mojo/clients/BUILD.gn
+++ b/src/3rdparty/chromium/media/mojo/clients/BUILD.gn
@@ -107,7 +107,7 @@ jumbo_source_set("clients") {
       "//ui/gl:gl",
     ]
   }
-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "win/media_foundation_renderer_client.cc",
       "win/media_foundation_renderer_client.h",
diff --git a/src/3rdparty/chromium/media/mojo/mojom/BUILD.gn b/src/3rdparty/chromium/media/mojo/mojom/BUILD.gn
index 50c2f63..950c7bf 100644
--- a/src/3rdparty/chromium/media/mojo/mojom/BUILD.gn
+++ b/src/3rdparty/chromium/media/mojo/mojom/BUILD.gn
@@ -708,7 +708,7 @@ mojom("mojom") {
     },
   ]

-  if (is_win) {
+  if (is_msvc) {
     cpp_typemaps += [
       {
         types = [
diff --git a/src/3rdparty/chromium/media/mojo/mojom/cdm_document_service.mojom b/src/3rdparty/chromium/media/mojo/mojom/cdm_document_service.mojom
index 4775950..08680f6 100644
--- a/src/3rdparty/chromium/media/mojo/mojom/cdm_document_service.mojom
+++ b/src/3rdparty/chromium/media/mojo/mojom/cdm_document_service.mojom
@@ -63,18 +63,18 @@ interface CdmDocumentService {
   // Gets the Media FoundationCDM data for the origin associated with the CDM.
   // - `media_foundation_cdm_data`: The CDM data for the origin associated with
   //                                the CDM.
-  [EnableIf=is_win]
+  [EnableIf=is_msvc]
   GetMediaFoundationCdmData() => (MediaFoundationCdmData cdm_data);

   // Sets the client token for the origin associated with the CDM. The token is
   // set by the CDM. The token is then saved in the Pref Service so that it can
   // be reused by the CDM for that same origin in the future.
-  [EnableIf=is_win]
+  [EnableIf=is_msvc]
   SetCdmClientToken(array<uint8> client_token);

   // Reports a CDM event, which can facilitate metrics reporting or fallback
   // logic.  For error events, the `hresult` provides more details about the
   // error.
-  [EnableIf=is_win]
+  [EnableIf=is_msvc]
   OnCdmEvent(CdmEvent event, uint32 hresult);
 };
diff --git a/src/3rdparty/chromium/media/mojo/services/BUILD.gn b/src/3rdparty/chromium/media/mojo/services/BUILD.gn
index 563c903..39696a9 100644
--- a/src/3rdparty/chromium/media/mojo/services/BUILD.gn
+++ b/src/3rdparty/chromium/media/mojo/services/BUILD.gn
@@ -188,7 +188,7 @@ jumbo_component("services") {
     deps += [ "//media/cdm:cdm_api" ]
   }

-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "media_foundation_gpu_info_monitor.cc",
       "media_foundation_gpu_info_monitor.h",
diff --git a/src/3rdparty/chromium/media/mojo/services/gpu_mojo_media_client_win.cc b/src/3rdparty/chromium/media/mojo/services/gpu_mojo_media_client_win.cc
index bd33420..b394675 100644
--- a/src/3rdparty/chromium/media/mojo/services/gpu_mojo_media_client_win.cc
+++ b/src/3rdparty/chromium/media/mojo/services/gpu_mojo_media_client_win.cc
@@ -126,7 +126,11 @@ class GpuMojoMediaClientWin final : public GpuMojoMediaClient {
   std::unique_ptr<AudioDecoder> CreatePlatformAudioDecoder(
       scoped_refptr<base::SequencedTaskRunner> task_runner,
       std::unique_ptr<MediaLog> media_log) final {
+#if defined(COMPILER_MSVC)
     return MediaFoundationAudioDecoder::Create();
+#else
+    return nullptr;
+#endif
   }

   VideoDecoderType GetPlatformDecoderImplementationType() final {
diff --git a/src/3rdparty/chromium/media/mojo/services/media_service_factory.cc b/src/3rdparty/chromium/media/mojo/services/media_service_factory.cc
index 9e1ffc8..b0f1720 100644
--- a/src/3rdparty/chromium/media/mojo/services/media_service_factory.cc
+++ b/src/3rdparty/chromium/media/mojo/services/media_service_factory.cc
@@ -17,7 +17,7 @@
 #include "media/mojo/services/android_mojo_media_client.h"  // nogncheck
 #endif

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "media/mojo/services/media_foundation_mojo_media_client.h"
 #endif

@@ -28,7 +28,7 @@ std::unique_ptr<MediaService> CreateMediaService(
 #if BUILDFLAG(IS_ANDROID)
   return std::make_unique<MediaService>(
       std::make_unique<AndroidMojoMediaClient>(), std::move(receiver));
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
   return std::make_unique<MediaService>(
       std::make_unique<MediaFoundationMojoMediaClient>(), std::move(receiver));
 #else
diff --git a/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.cc b/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.cc
index 9becbb9..e5a3346 100644
--- a/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.cc
+++ b/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.cc
@@ -60,7 +60,7 @@ url::Origin MojoCdmHelper::GetCdmOrigin() {
   return cdm_origin;
 }

-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 void MojoCdmHelper::GetMediaFoundationCdmData(
     GetMediaFoundationCdmDataCB callback) {
   ConnectToCdmDocumentService();
@@ -77,7 +77,7 @@ void MojoCdmHelper::OnCdmEvent(CdmEvent event, HRESULT hresult) {
   ConnectToCdmDocumentService();
   cdm_document_service_->OnCdmEvent(event, hresult);
 }
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)

 cdm::Buffer* MojoCdmHelper::CreateCdmBuffer(size_t capacity) {
   return GetAllocator()->CreateCdmBuffer(capacity);
diff --git a/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.h b/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.h
index 9bed939..745df11 100644
--- a/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.h
+++ b/src/3rdparty/chromium/media/mojo/services/mojo_cdm_helper.h
@@ -51,11 +51,11 @@ class MEDIA_MOJO_EXPORT MojoCdmHelper final : public CdmAuxiliaryHelper,
                          const std::string& challenge,
                          ChallengePlatformCB callback) final;
   void GetStorageId(uint32_t version, StorageIdCB callback) final;
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   void GetMediaFoundationCdmData(GetMediaFoundationCdmDataCB callback) final;
   void SetCdmClientToken(const std::vector<uint8_t>& client_token) final;
   void OnCdmEvent(CdmEvent event, HRESULT hresult) final;
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)

   // MojoCdmFileIO::Delegate implementation.
   void CloseCdmFileIO(MojoCdmFileIO* cdm_file_io) final;
diff --git a/src/3rdparty/chromium/media/renderers/BUILD.gn b/src/3rdparty/chromium/media/renderers/BUILD.gn
index 3f27372..05c5333 100644
--- a/src/3rdparty/chromium/media/renderers/BUILD.gn
+++ b/src/3rdparty/chromium/media/renderers/BUILD.gn
@@ -63,7 +63,7 @@ source_set("renderers") {
     "//ui/gl",
   ]

-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "win/media_engine_extension.cc",
       "win/media_engine_extension.h",
