diff --git a/src/3rdparty/chromium/net/BUILD.gn b/src/3rdparty/chromium/net/BUILD.gn
index 6e91e8c..04b56c1 100644
--- a/src/3rdparty/chromium/net/BUILD.gn
+++ b/src/3rdparty/chromium/net/BUILD.gn
@@ -1271,6 +1271,12 @@ component("net") {
       "ssl/client_cert_store_win.h",
       "ssl/ssl_platform_key_win.cc",
     ]
+    if (is_mingw) {
+      sources -= [
+        "base/network_change_notifier_win.cc",
+        "base/network_change_notifier_win.h",
+      ]
+    }
   }
 
   if (use_kerberos) {
@@ -1466,17 +1472,19 @@ component("net") {
 
   if (is_win) {
     libs = [
-      "crypt32.lib",
-      "dhcpcsvc.lib",
-      "iphlpapi.lib",
-      "ncrypt.lib",
-      "rpcrt4.lib",
-      "secur32.lib",
-      "urlmon.lib",
-      "winhttp.lib",
-    ]
-
-    ldflags = [ "/DELAYLOAD:urlmon.dll" ]
+      "crypt32",
+      "dhcpcsvc",
+      "iphlpapi",
+      "ncrypt",
+      "rpcrt4",
+      "secur32",
+      "urlmon",
+      "winhttp",
+    ]
+
+    if (is_msvc) {
+      ldflags = [ "/DELAYLOAD:urlmon.dll" ]
+    }
   }
 
   if (!disable_file_support) {
@@ -3174,8 +3182,8 @@ test("net_unittests") {
 
   if (is_win) {
     libs = [
-      "iphlpapi.lib",
-      "ncrypt.lib",
+      "iphlpapi",
+      "ncrypt",
     ]
   }
 
@@ -3186,7 +3194,7 @@ test("net_unittests") {
     if (is_win) {
       sources += [ "cert/internal/trust_store_win_unittest.cc" ]
       sources += [ "cert/internal/path_builder_trust_store_win_unittest.cc" ]
-      libs += [ "crypt32.lib" ]
+      libs += [ "crypt32" ]
     }
   }
 
diff --git a/src/3rdparty/chromium/net/base/network_change_notifier.cc b/src/3rdparty/chromium/net/base/network_change_notifier.cc
index 80e55b1..ff23698 100644
--- a/src/3rdparty/chromium/net/base/network_change_notifier.cc
+++ b/src/3rdparty/chromium/net/base/network_change_notifier.cc
@@ -307,7 +307,7 @@ std::unique_ptr<NetworkChangeNotifier> NetworkChangeNotifier::CreateIfNeeded(
         initial_type, initial_subtype);
   }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   std::unique_ptr<NetworkChangeNotifierWin> network_change_notifier =
       std::make_unique<NetworkChangeNotifierWin>();
   network_change_notifier->WatchForAddressChange();
diff --git a/src/3rdparty/chromium/net/base/network_interfaces_win.cc b/src/3rdparty/chromium/net/base/network_interfaces_win.cc
index c6e1f3c..4d28530 100644
--- a/src/3rdparty/chromium/net/base/network_interfaces_win.cc
+++ b/src/3rdparty/chromium/net/base/network_interfaces_win.cc
@@ -277,12 +277,14 @@ WifiPHYLayerProtocol GetWifiPHYLayerProtocol() {
       return WIFI_PHY_LAYER_PROTOCOL_G;
     case dot11_phy_type_ht:
       return WIFI_PHY_LAYER_PROTOCOL_N;
+#if !defined(COMPILER_GCC) // missing WIFI_PHY_LAYER_PROTOCOL_A*
     case dot11_phy_type_vht:
       return WIFI_PHY_LAYER_PROTOCOL_AC;
     case dot11_phy_type_dmg:
       return WIFI_PHY_LAYER_PROTOCOL_AD;
     case dot11_phy_type_he:
       return WIFI_PHY_LAYER_PROTOCOL_AX;
+#endif
     default:
       return WIFI_PHY_LAYER_PROTOCOL_UNKNOWN;
   }
diff --git a/src/3rdparty/chromium/net/dns/host_resolver_system_task.cc b/src/3rdparty/chromium/net/dns/host_resolver_system_task.cc
index a0820f0..1606c99 100644
--- a/src/3rdparty/chromium/net/dns/host_resolver_system_task.cc
+++ b/src/3rdparty/chromium/net/dns/host_resolver_system_task.cc
@@ -418,13 +418,13 @@ int SystemHostResolverCall(const std::string& host,
   if (host_resolver_flags & HOST_RESOLVER_CANONNAME)
     hints.ai_flags |= AI_CANONNAME;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   // See crbug.com/1176970. Flag not documented (other than the declaration
   // comment in ws2def.h) but confirmed by Microsoft to work for this purpose
   // and be safe.
   if (host_resolver_flags & HOST_RESOLVER_AVOID_MULTICAST)
     hints.ai_flags |= AI_DNS_ONLY;
-#endif  // BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_MSVC)
 
   // Restrict result set to only this socket type to avoid duplicates.
   hints.ai_socktype = SOCK_STREAM;
diff --git a/src/3rdparty/chromium/net/socket/udp_socket_win.cc b/src/3rdparty/chromium/net/socket/udp_socket_win.cc
index deb0561..0e60748 100644
--- a/src/3rdparty/chromium/net/socket/udp_socket_win.cc
+++ b/src/3rdparty/chromium/net/socket/udp_socket_win.cc
@@ -487,12 +487,14 @@ int UDPSocketWin::InternalConnect(const IPEndPoint& address) {
   DCHECK(!is_connected());
   DCHECK(!remote_address_.get());
 
+#if defined(COMPILER_MSVC)
   // Always do a random bind.
   // Ignore failures, which may happen if the socket was already bound.
   DWORD randomize_port_value = 1;
   setsockopt(socket_, SOL_SOCKET, SO_RANDOMIZE_PORT,
              reinterpret_cast<const char*>(&randomize_port_value),
              sizeof(randomize_port_value));
+#endif
 
   SockaddrStorage storage;
   if (!address.ToSockAddr(storage.addr, &storage.addr_len))
@@ -1287,6 +1289,7 @@ int DscpManager::PrepareForSend(const IPEndPoint& remote_address) {
   // We won't try this address again if we get an error.
   configured_.emplace(remote_address);
 
+#if defined(COMPILER_MSVC)
   // We don't need to call SetFlow if we already have a qos flow.
   bool new_flow = flow_id_ == 0;
 
@@ -1315,6 +1318,7 @@ int DscpManager::PrepareForSend(const IPEndPoint& remote_address) {
     api_->SetFlow(qos_handle_, flow_id_, QOSSetOutgoingDSCPValue, sizeof(buf),
                   &buf, 0, nullptr);
   }
+#endif
 
   return OK;
 }
diff --git a/src/3rdparty/chromium/net/ssl/ssl_platform_key_win.cc b/src/3rdparty/chromium/net/ssl/ssl_platform_key_win.cc
index 99a3939..b37f445 100644
--- a/src/3rdparty/chromium/net/ssl/ssl_platform_key_win.cc
+++ b/src/3rdparty/chromium/net/ssl/ssl_platform_key_win.cc
@@ -259,6 +259,7 @@ class SSLPlatformKeyCNG : public ThreadedSSLPrivateKey::Delegate {
     // result, if this is a TPM-based key, we only report PSS as supported if
     // the salt length will match the digest length.
     bool supports_pss = true;
+#if defined(COMPILER_MSVC)
     if (provider_name_ == MS_PLATFORM_KEY_STORAGE_PROVIDER) {
       DWORD salt_size = 0;
       DWORD size_of_salt_size = sizeof(salt_size);
@@ -270,6 +271,7 @@ class SSLPlatformKeyCNG : public ThreadedSSLPrivateKey::Delegate {
         supports_pss = false;
       }
     }
+#endif
     if (prefer_sha1_) {
       std::vector<uint16_t> ret = {
           SSL_SIGN_RSA_PKCS1_SHA1,
diff --git a/src/3rdparty/chromium/net/third_party/quiche/src/quiche/http2/adapter/window_manager.h b/src/3rdparty/chromium/net/third_party/quiche/src/quiche/http2/adapter/window_manager.h
index ffe1a7f..4666f32 100644
--- a/src/3rdparty/chromium/net/third_party/quiche/src/quiche/http2/adapter/window_manager.h
+++ b/src/3rdparty/chromium/net/third_party/quiche/src/quiche/http2/adapter/window_manager.h
@@ -4,6 +4,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <cstdint>
 #include <functional>
 
 #include "quiche/common/platform/api/quiche_export.h"
