diff --git a/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_provider.cc b/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_provider.cc
index de925c9..82f5808 100644
--- a/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_provider.cc
+++ b/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_provider.cc
@@ -68,7 +68,9 @@ std::unique_ptr<PlatformSensorProvider> PlatformSensorProvider::Create() {
   return std::make_unique<PlatformSensorProviderMac>();
 #elif BUILDFLAG(IS_ANDROID)
   return std::make_unique<PlatformSensorProviderAndroid>();
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MINGW)
+  return std::make_unique<PlatformSensorProviderWin>();
+#elif BUILDFLAG(IS_MSVC)
   if (PlatformSensorProvider::UseWindowsWinrt()) {
     return std::make_unique<PlatformSensorProviderWinrt>();
   } else {
diff --git a/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_reader_win.cc b/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_reader_win.cc
index 1343660..616f10f 100644
--- a/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_reader_win.cc
+++ b/src/3rdparty/chromium/services/device/generic_sensor/platform_sensor_reader_win.cc
@@ -13,7 +13,9 @@
 
 #include <Sensors.h>
 #include <comdef.h>
+#if defined(COMPILER_MSVC)
 #include <wrl/implements.h>
+#endif
 
 #include <iomanip>
 
@@ -255,6 +257,7 @@ std::unique_ptr<ReaderInitParams> CreateReaderInitParamsForSensor(
 
 }  // namespace
 
+#if defined(COMPILER_MSVC)
 // Class that implements ISensorEvents used by the ISensor interface to dispatch
 // state and data change events.
 class EventListener
@@ -358,6 +361,7 @@ class EventListener
   const raw_ptr<PlatformSensorReaderWin32> platform_sensor_reader_;
   SensorReading last_sensor_reading_;
 };
+#endif
 
 // static
 std::unique_ptr<PlatformSensorReaderWinBase> PlatformSensorReaderWin32::Create(
@@ -386,8 +390,12 @@ std::unique_ptr<PlatformSensorReaderWinBase> PlatformSensorReaderWin32::Create(
   if (FAILED(hr))
     return nullptr;
 
+#if defined(COMPILER_MSVC)
   return base::WrapUnique(
       new PlatformSensorReaderWin32(sensor, std::move(params)));
+#else
+  return nullptr;
+#endif
 }
 
 // static
@@ -408,6 +416,7 @@ Microsoft::WRL::ComPtr<ISensor> PlatformSensorReaderWin32::GetSensorForType(
   return sensor;
 }
 
+#if defined(COMPILER_MSVC)
 PlatformSensorReaderWin32::PlatformSensorReaderWin32(
     Microsoft::WRL::ComPtr<ISensor> sensor,
     std::unique_ptr<ReaderInitParams> params)
@@ -421,6 +430,7 @@ PlatformSensorReaderWin32::PlatformSensorReaderWin32(
   DCHECK(init_params_->reader_func);
   DCHECK(sensor_);
 }
+#endif
 
 void PlatformSensorReaderWin32::SetClient(Client* client) {
   base::AutoLock autolock(lock_);
diff --git a/src/3rdparty/chromium/services/device/geolocation/BUILD.gn b/src/3rdparty/chromium/services/device/geolocation/BUILD.gn
index a2f8504..0717e89 100644
--- a/src/3rdparty/chromium/services/device/geolocation/BUILD.gn
+++ b/src/3rdparty/chromium/services/device/geolocation/BUILD.gn
@@ -92,7 +92,7 @@ source_set("geolocation") {
     ]
   }
 
-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "win/location_provider_winrt.cc",
       "win/location_provider_winrt.h",
diff --git a/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.cc b/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.cc
index c4f4b0f..b9220ff 100644
--- a/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.cc
+++ b/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.cc
@@ -23,7 +23,9 @@ namespace device {

 namespace {

+#if !defined(__MINGW32__)
 using ::ABI::Windows::Devices::Geolocation::GeolocationAccessStatus;
+#endif
 using ::ABI::Windows::Security::Authorization::AppCapabilityAccess::
     IAppCapability;
 using ::Microsoft::WRL::ComPtr;
@@ -246,6 +248,9 @@ void SystemGeolocationSourceWin::OnLaunchUriFailure(HRESULT result) {
 }

 void SystemGeolocationSourceWin::OpenSystemPermissionSetting() {
+#if defined(__MINGW32__)
+  RecordUmaLaunchSettingsResult(E_NOTIMPL);
+#else
   using ::ABI::Windows::Foundation::IUriRuntimeClass;
   using ::ABI::Windows::Foundation::IUriRuntimeClassFactory;
   using ::ABI::Windows::System::ILauncherStatics;
@@ -303,9 +308,13 @@ void SystemGeolocationSourceWin::OpenSystemPermissionSetting() {
                << logging::SystemErrorCodeToString(hr);
     RecordUmaLaunchSettingsResult(hr);
   }
+#endif
 }

 void SystemGeolocationSourceWin::RequestPermission() {
+#if defined(__MINGW32__)
+  RecordUmaRequestAccessResult(E_NOTIMPL);
+#else
   using ::ABI::Windows::Devices::Geolocation::IGeolocatorStatics;
   if (request_location_access_op_) {
     return;
@@ -343,8 +352,10 @@ void SystemGeolocationSourceWin::RequestPermission() {
     return;
   }
   has_pending_system_prompt_ = true;
+#endif
 }

+#if !defined(__MINGW32__)
 void SystemGeolocationSourceWin::OnRequestLocationAccessSuccess(
     GeolocationAccessStatus status) {
   RecordUmaRequestAccessResult(S_OK);
@@ -358,5 +369,6 @@ void SystemGeolocationSourceWin::OnRequestLocationAccessFailure(
   RecordUmaRequestAccessResult(result);
   request_location_access_op_.Reset();
 }
+#endif

 }  // namespace device
diff --git a/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.h b/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.h
index 453e88c..ff083b2 100644
--- a/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.h
+++ b/src/3rdparty/chromium/services/device/public/cpp/geolocation/system_geolocation_source_win.h
@@ -48,16 +48,19 @@ class COMPONENT_EXPORT(GEOLOCATION) SystemGeolocationSourceWin
   void OnLaunchUriSuccess(uint8_t launched);
   void OnLaunchUriFailure(HRESULT result);

+#if !defined(__MINGW32__)
   void OnRequestLocationAccessSuccess(
       ABI::Windows::Devices::Geolocation::GeolocationAccessStatus
           geolocation_access_status);
   void OnRequestLocationAccessFailure(HRESULT result);
+#endif

   // The pending operation for launching the settings page, or nullptr if not
   // launching the settings page.
   Microsoft::WRL::ComPtr<ABI::Windows::Foundation::IAsyncOperation<bool>>
       launch_uri_op_;

+#if !defined(__MINGW32__)
   // The pending operation for requesting location access, or nullptr if not
   // requesting access.
   Microsoft::WRL::ComPtr<ABI::Windows::Foundation::IAsyncOperation<
@@ -66,5 +69,6 @@ class COMPONENT_EXPORT(GEOLOCATION) SystemGeolocationSourceWin
       ABI::Windows::Devices::Geolocation::GeolocationAccessStatus>>
       request_location_access_op_;
+#endif

   // True if the user was prompted to grant location permissions and the
   // permission status has not yet changed. Used for metrics logging.
diff --git a/src/3rdparty/chromium/services/device/hid/BUILD.gn b/src/3rdparty/chromium/services/device/hid/BUILD.gn
index 89a178a..7f266d4 100644
--- a/src/3rdparty/chromium/services/device/hid/BUILD.gn
+++ b/src/3rdparty/chromium/services/device/hid/BUILD.gn
@@ -59,10 +59,17 @@ source_set("hid") {
       "hid_service_win.cc",
       "hid_service_win.h",
     ]
-    libs = [
-      "hid.lib",
-      "setupapi.lib",
-    ]
+    if (is_msvc) {
+      libs = [
+        "hid.lib",
+        "setupapi.lib",
+      ]
+    } else {
+      libs = [
+        "hid",
+        "setupapi",
+      ]
+    }
   }
 
   if (is_mac) {
diff --git a/src/3rdparty/chromium/services/device/hid/hid_service_win.cc b/src/3rdparty/chromium/services/device/hid/hid_service_win.cc
index c96b6f1..5d36b9d 100644
--- a/src/3rdparty/chromium/services/device/hid/hid_service_win.cc
+++ b/src/3rdparty/chromium/services/device/hid/hid_service_win.cc
@@ -12,7 +12,9 @@
 #include <devpkey.h>
 #include <setupapi.h>
 #include <stddef.h>
+#if defined(COMPILER_MSVC)
 #include <wdmguid.h>
+#endif
 #include <winioctl.h>
 
 #include <algorithm>
@@ -80,7 +82,7 @@ std::optional<std::wstring> GetDeviceStringProperty(
     const DEVPROPKEY& property_key) {
   DEVPROPTYPE property_type;
   DWORD required_size;
-  if (SetupDiGetDeviceProperty(device_info_set, &device_info_data,
+  if (SetupDiGetDevicePropertyW(device_info_set, &device_info_data,
                                &property_key, &property_type,
                                /*PropertyBuffer=*/nullptr,
                                /*PropertyBufferSize=*/0, &required_size,
@@ -102,7 +104,7 @@ std::optional<std::wstring> GetDeviceStringProperty(
     return std::nullopt;
 
   std::wstring property_buffer;
-  if (!SetupDiGetDeviceProperty(
+  if (!SetupDiGetDevicePropertyW(
           device_info_set, &device_info_data, &property_key, &property_type,
           reinterpret_cast<PBYTE>(
               base::WriteInto(&property_buffer, required_size)),
@@ -124,7 +126,7 @@ std::optional<std::string> GetDeviceGuidProperty(
     const DEVPROPKEY& property_key) {
   DEVPROPTYPE property_type;
   GUID property_buffer;
-  if (!SetupDiGetDeviceProperty(
+  if (!SetupDiGetDevicePropertyW(
           device_info_set, &device_info_data, &property_key, &property_type,
           reinterpret_cast<PBYTE>(&property_buffer), sizeof(property_buffer),
           /*RequiredSize=*/nullptr, /*Flags=*/0)) {
diff --git a/src/3rdparty/chromium/services/device/serial/BUILD.gn b/src/3rdparty/chromium/services/device/serial/BUILD.gn
index 70e64a8..dbaa1fd 100644
--- a/src/3rdparty/chromium/services/device/serial/BUILD.gn
+++ b/src/3rdparty/chromium/services/device/serial/BUILD.gn
@@ -8,8 +8,12 @@ if (is_win || ((is_linux || is_chromeos) && use_udev) || is_mac) {
   config("platform_support") {
     visibility = [ ":serial" ]
     if (is_win) {
-      libs = [ "setupapi.lib" ]
-      ldflags = [ "/DELAYLOAD:setupapi.dll" ]
+      if (is_msvc) {
+        libs = [ "setupapi.lib" ]
+        ldflags = [ "/DELAYLOAD:setupapi.dll" ]
+      } else {
+        libs = [ "setupapi" ]
+      }
     }
   }
 
diff --git a/src/3rdparty/chromium/services/device/serial/serial_device_enumerator_win.cc b/src/3rdparty/chromium/services/device/serial/serial_device_enumerator_win.cc
index a5c7dc3..d357ce7 100644
--- a/src/3rdparty/chromium/services/device/serial/serial_device_enumerator_win.cc
+++ b/src/3rdparty/chromium/services/device/serial/serial_device_enumerator_win.cc
@@ -9,6 +9,7 @@
 #define INITGUID
 #include <devguid.h>
 #include <devpkey.h>
+#include <ntdef.h>
 #include <ntddser.h>
 #include <setupapi.h>
 #include <stdint.h>
@@ -50,7 +51,7 @@ std::optional<std::string> GetProperty(HDEVINFO dev_info,
 
   DEVPROPTYPE property_type;
   DWORD required_size;
-  if (SetupDiGetDeviceProperty(dev_info, dev_info_data, &property,
+  if (SetupDiGetDevicePropertyW(dev_info, dev_info_data, &property,
                                &property_type, /*PropertyBuffer=*/nullptr,
                                /*PropertyBufferSize=*/0, &required_size,
                                /*Flags=*/0) ||
@@ -60,7 +61,7 @@ std::optional<std::string> GetProperty(HDEVINFO dev_info,
   }
 
   std::wstring buffer;
-  if (!SetupDiGetDeviceProperty(
+  if (!SetupDiGetDevicePropertyW(
           dev_info, dev_info_data, &property, &property_type,
           reinterpret_cast<PBYTE>(base::WriteInto(&buffer, required_size)),
           required_size, /*RequiredSize=*/nullptr, /*Flags=*/0)) {
diff --git a/src/3rdparty/chromium/services/device/usb/BUILD.gn b/src/3rdparty/chromium/services/device/usb/BUILD.gn
index 77ae219..ffa4e15 100644
--- a/src/3rdparty/chromium/services/device/usb/BUILD.gn
+++ b/src/3rdparty/chromium/services/device/usb/BUILD.gn
@@ -80,12 +80,21 @@ static_library("usb") {
       "usb_service_win.h",
     ]
 
-    libs = [
-      "advapi32.lib",
-      "ole32.lib",
-      "setupapi.lib",
-      "winusb.lib",
-    ]
+    if (is_msvc) {
+      libs = [
+        "advapi32.lib",
+        "ole32.lib",
+        "setupapi.lib",
+        "winusb.lib",
+      ]
+    } else {
+      libs = [
+        "advapi32",
+        "ole32",
+        "setupapi",
+        "winusb",
+      ]
+    }
 
     deps += [ "//third_party/re2" ]
   }
diff --git a/src/3rdparty/chromium/services/device/usb/usb_service_win.cc b/src/3rdparty/chromium/services/device/usb/usb_service_win.cc
index 0888473..310943b 100644
--- a/src/3rdparty/chromium/services/device/usb/usb_service_win.cc
+++ b/src/3rdparty/chromium/services/device/usb/usb_service_win.cc
@@ -67,7 +67,7 @@ std::optional<uint32_t> GetDeviceUint32Property(HDEVINFO dev_info,
 
   DEVPROPTYPE property_type;
   uint32_t buffer;
-  if (!SetupDiGetDeviceProperty(
+  if (!SetupDiGetDevicePropertyW(
           dev_info, dev_info_data, &property, &property_type,
           reinterpret_cast<PBYTE>(&buffer), sizeof(buffer), nullptr, 0)) {
     USB_PLOG(ERROR) << "SetupDiGetDeviceProperty(" << property << ") failed";
@@ -94,7 +94,7 @@ std::optional<std::wstring> GetDeviceStringProperty(
 
   DEVPROPTYPE property_type;
   DWORD required_size;
-  if (SetupDiGetDeviceProperty(dev_info, dev_info_data, &property,
+  if (SetupDiGetDevicePropertyW(dev_info, dev_info_data, &property,
                                &property_type, nullptr, 0, &required_size, 0)) {
     USB_LOG(ERROR) << "SetupDiGetDeviceProperty(" << property
                    << ") unexpectedly succeeded";
@@ -114,7 +114,7 @@ std::optional<std::wstring> GetDeviceStringProperty(
   }
 
   std::wstring buffer;
-  if (!SetupDiGetDeviceProperty(
+  if (!SetupDiGetDevicePropertyW(
           dev_info, dev_info_data, &property, &property_type,
           reinterpret_cast<PBYTE>(base::WriteInto(&buffer, required_size)),
           required_size, nullptr, 0)) {
@@ -135,7 +135,7 @@ std::optional<std::vector<std::wstring>> GetDeviceStringListProperty(
 
   DEVPROPTYPE property_type;
   DWORD required_size;
-  if (SetupDiGetDeviceProperty(dev_info, dev_info_data, &property,
+  if (SetupDiGetDevicePropertyW(dev_info, dev_info_data, &property,
                                &property_type, nullptr, 0, &required_size, 0)) {
     USB_LOG(ERROR) << "SetupDiGetDeviceProperty(" << property
                    << ") unexpectedly succeeded";
@@ -160,7 +160,7 @@ std::optional<std::vector<std::wstring>> GetDeviceStringListProperty(
   }
 
   std::wstring buffer;
-  if (!SetupDiGetDeviceProperty(
+  if (!SetupDiGetDevicePropertyW(
           dev_info, dev_info_data, &property, &property_type,
           reinterpret_cast<PBYTE>(base::WriteInto(&buffer, required_size)),
           required_size, nullptr, 0)) {
diff --git a/src/3rdparty/chromium/services/proxy_resolver_win/BUILD.gn b/src/3rdparty/chromium/services/proxy_resolver_win/BUILD.gn
index ed4b004..fb72ec5 100644
--- a/src/3rdparty/chromium/services/proxy_resolver_win/BUILD.gn
+++ b/src/3rdparty/chromium/services/proxy_resolver_win/BUILD.gn
@@ -13,7 +13,11 @@ component("proxy_resolver_win") {
     "winhttp_api_wrapper_impl.h",
   ]
 
-  libs = [ "winhttp.lib" ]
+  if (is_msvc) {
+    libs = [ "winhttp.lib" ]
+  } else {
+    libs = [ "winhttp" ]
+  }
 
   deps = [
     "//base",
diff --git a/src/3rdparty/chromium/services/shape_detection/BUILD.gn b/src/3rdparty/chromium/services/shape_detection/BUILD.gn
index 8480cdf..2ba4c25 100644
--- a/src/3rdparty/chromium/services/shape_detection/BUILD.gn
+++ b/src/3rdparty/chromium/services/shape_detection/BUILD.gn
@@ -38,7 +38,7 @@ source_set("lib") {
       "QuartzCore.framework",
       "Vision.framework",
     ]
-  } else if (is_win) {
+  } else if (is_msvc) {
     sources += [
       "detection_utils_win.cc",
       "detection_utils_win.h",
diff --git a/src/3rdparty/chromium/services/shape_detection/face_detection_provider_win.h b/src/3rdparty/chromium/services/shape_detection/face_detection_provider_win.h
index f6b3823..53e5206 100644
--- a/src/3rdparty/chromium/services/shape_detection/face_detection_provider_win.h
+++ b/src/3rdparty/chromium/services/shape_detection/face_detection_provider_win.h
@@ -14,7 +14,9 @@
 #include "base/memory/weak_ptr.h"
 #include "mojo/public/cpp/bindings/pending_receiver.h"
 #include "mojo/public/cpp/bindings/self_owned_receiver.h"
+#if defined(COMPILER_MSVC)
 #include "services/shape_detection/face_detection_impl_win.h"
+#endif
 #include "services/shape_detection/public/mojom/facedetection_provider.mojom.h"
 
 namespace shape_detection {
@@ -43,11 +45,13 @@ class FaceDetectionProviderWin
       shape_detection::mojom::FaceDetectorOptionsPtr options) override;
 
  private:
+#if defined(COMPILER_MSVC)
   void OnFaceDetectorCreated(
       mojo::PendingReceiver<shape_detection::mojom::FaceDetection> receiver,
       ABI::Windows::Graphics::Imaging::BitmapPixelFormat pixel_format,
       Microsoft::WRL::ComPtr<ABI::Windows::Media::FaceAnalysis::IFaceDetector>
           face_detector);
+#endif
 
   mojo::SelfOwnedReceiverRef<mojom::FaceDetectionProvider> receiver_;
   base::WeakPtrFactory<FaceDetectionProviderWin> weak_factory_{this};
diff --git a/src/3rdparty/chromium/services/shape_detection/shape_detection_service.cc b/src/3rdparty/chromium/services/shape_detection/shape_detection_service.cc
index ad24ba9..1a75a04 100644
--- a/src/3rdparty/chromium/services/shape_detection/shape_detection_service.cc
+++ b/src/3rdparty/chromium/services/shape_detection/shape_detection_service.cc
@@ -27,7 +27,7 @@
 #include "services/shape_detection/barcode_detection_provider_impl.h"
 #endif
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "services/shape_detection/face_detection_provider_win.h"
 #elif BUILDFLAG(IS_MAC)
 #include "services/shape_detection/face_detection_provider_mac.h"
@@ -69,7 +69,7 @@ void ShapeDetectionService::BindFaceDetectionProvider(
       receiver.PassPipe().release().value());
 #elif BUILDFLAG(IS_MAC)
   FaceDetectionProviderMac::Create(std::move(receiver));
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
   FaceDetectionProviderWin::Create(std::move(receiver));
 #else
   FaceDetectionProviderImpl::Create(std::move(receiver));
diff --git a/src/3rdparty/chromium/services/video_capture/device_media_to_mojo_adapter.cc b/src/3rdparty/chromium/services/video_capture/device_media_to_mojo_adapter.cc
index eac863c..a9db714 100644
--- a/src/3rdparty/chromium/services/video_capture/device_media_to_mojo_adapter.cc
+++ b/src/3rdparty/chromium/services/video_capture/device_media_to_mojo_adapter.cc
@@ -162,15 +162,15 @@ void DeviceMediaToMojoAdapter::StartInternal(
     TRACE_EVENT0(TRACE_DISABLED_BY_DEFAULT("video_and_image_capture"),
                  "CreateVideoCaptureBufferPoolImpl");
     // Create a dedicated buffer pool for the device usage session.
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
     buffer_pool = base::MakeRefCounted<media::VideoCaptureBufferPoolImpl>(
         requested_settings.buffer_type, max_buffer_pool_buffer_count(),
         std::make_unique<media::VideoCaptureBufferTrackerFactoryImpl>(
             dxgi_device_manager_));
-#else   // BUILDFLAG(IS_WIN)
+#else   // BUILDFLAG(IS_MSVC)
     buffer_pool = base::MakeRefCounted<media::VideoCaptureBufferPoolImpl>(
         requested_settings.buffer_type, max_buffer_pool_buffer_count());
-#endif  // !BUILDFLAG(IS_WIN)
+#endif  // !BUILDFLAG(IS_MSVC)
   }
 
 #if BUILDFLAG(IS_CHROMEOS)
diff --git a/src/3rdparty/chromium/services/video_capture/video_capture_service_impl.cc b/src/3rdparty/chromium/services/video_capture/video_capture_service_impl.cc
index 76ce891..1717f84 100644
--- a/src/3rdparty/chromium/services/video_capture/video_capture_service_impl.cc
+++ b/src/3rdparty/chromium/services/video_capture/video_capture_service_impl.cc
@@ -319,6 +319,7 @@ void VideoCaptureServiceImpl::LazyInitializeDeviceFactory() {
 
   LazyInitializeGpuDependenciesContext();
 
+#if !BUILDFLAG(IS_MINGW)
   // Create the platform-specific device factory.
   // The task runner passed to CreateFactory is used for things that need to
   // happen on a "UI thread equivalent", e.g. obtaining screen rotation on
@@ -328,6 +329,7 @@ void VideoCaptureServiceImpl::LazyInitializeDeviceFactory() {
 
   auto video_capture_system = std::make_unique<media::VideoCaptureSystemImpl>(
       std::move(media_device_factory));
+#endif
 
 #if BUILDFLAG(IS_CHROMEOS)
   device_factory_ = std::make_unique<VirtualDeviceEnabledDeviceFactory>(
@@ -340,7 +342,7 @@ void VideoCaptureServiceImpl::LazyInitializeDeviceFactory() {
   device_factory_ash_adapter_ =
       std::make_unique<crosapi::VideoCaptureDeviceFactoryAsh>(
           device_factory_.get());
-#else
+#elif !BUILDFLAG(IS_MINGW)
   device_factory_ = std::make_unique<VirtualDeviceEnabledDeviceFactory>(
       std::make_unique<DeviceFactoryImpl>(std::move(video_capture_system)));
 #endif  // BUILDFLAG(IS_CHROMEOS)
diff --git a/src/3rdparty/chromium/services/webnn/BUILD.gn b/src/3rdparty/chromium/services/webnn/BUILD.gn
index 9d662a5..51893ed 100644
--- a/src/3rdparty/chromium/services/webnn/BUILD.gn
+++ b/src/3rdparty/chromium/services/webnn/BUILD.gn
@@ -47,7 +47,7 @@ component("webnn_service") {
     "//services/webnn/public/mojom",
   ]
 
-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "dml/adapter.cc",
       "dml/adapter.h",
@@ -179,7 +179,7 @@ source_set("tests") {
     "webnn_test_utils.h",
   ]
 
-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "dml/adapter_test.cc",
       "dml/command_queue_test.cc",
