diff --git a/src/3rdparty/chromium/ui/accessibility/ax_text_attributes.h b/src/3rdparty/chromium/ui/accessibility/ax_text_attributes.h
index 001c839..f4fdd11 100644
--- a/src/3rdparty/chromium/ui/accessibility/ax_text_attributes.h
+++ b/src/3rdparty/chromium/ui/accessibility/ax_text_attributes.h
@@ -5,6 +5,7 @@
 #ifndef UI_ACCESSIBILITY_AX_TEXT_ATTRIBUTES_H_
 #define UI_ACCESSIBILITY_AX_TEXT_ATTRIBUTES_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/3rdparty/chromium/ui/aura/BUILD.gn b/src/3rdparty/chromium/ui/aura/BUILD.gn
index 65f84ba..083fd86 100644
--- a/src/3rdparty/chromium/ui/aura/BUILD.gn
+++ b/src/3rdparty/chromium/ui/aura/BUILD.gn
@@ -166,7 +166,12 @@ jumbo_component("aura") {
     ]
   }
 
-  if (is_win) {
+  if (is_mingw) {
+    libs = [
+      "wtsapi32",
+      "dwmapi",
+    ]
+  } else {
     libs = [
       "wtsapi32.lib",
       "dwmapi.lib",
diff --git a/src/3rdparty/chromium/ui/base/BUILD.gn b/src/3rdparty/chromium/ui/base/BUILD.gn
index 903993c..55fae4c 100644
--- a/src/3rdparty/chromium/ui/base/BUILD.gn
+++ b/src/3rdparty/chromium/ui/base/BUILD.gn
@@ -293,11 +293,19 @@ jumbo_component("base") {
       "win/window_event_target.cc",
       "win/window_event_target.h",
     ]
-    ldflags = [ "/DELAYLOAD:dwmapi.dll" ]
-    libs += [
-      "dwmapi.lib",
-      "wtsapi32.lib",
-    ]
+    if (is_mingw) {
+      cflags_cc = [ "-Wno-sign-conversion" ]
+      libs += [
+        "dwmapi",
+        "wtsapi32",
+      ]
+    } else {
+      ldflags = [ "/DELAYLOAD:dwmapi.dll" ]
+      libs += [
+        "dwmapi.lib",
+        "wtsapi32.lib",
+      ]
+    }
   }
 
   if (is_mac) {
diff --git a/src/3rdparty/chromium/ui/base/ime/init/input_method_factory.cc b/src/3rdparty/chromium/ui/base/ime/init/input_method_factory.cc
index c99fd18..a30da75 100644
--- a/src/3rdparty/chromium/ui/base/ime/init/input_method_factory.cc
+++ b/src/3rdparty/chromium/ui/base/ime/init/input_method_factory.cc
@@ -12,7 +12,7 @@
 #include "ui/base/ui_base_features.h"
 #include "ui/gfx/switches.h"
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 #include "ui/base/ime/win/input_method_win_tsf.h"
 #elif BUILDFLAG(IS_APPLE)
 #include "ui/base/ime/mac/input_method_mac.h"
@@ -52,7 +52,7 @@ std::unique_ptr<InputMethod> CreateInputMethod(
   if (base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kHeadless))
     return base::WrapUnique(new MockInputMethod(ime_key_event_dispatcher));
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   return std::make_unique<InputMethodWinTSF>(ime_key_event_dispatcher, widget);
 #elif BUILDFLAG(IS_APPLE)
   return std::make_unique<InputMethodMac>(ime_key_event_dispatcher);
diff --git a/src/3rdparty/chromium/ui/base/ime/init/input_method_initializer.cc b/src/3rdparty/chromium/ui/base/ime/init/input_method_initializer.cc
index ce9d6fe..2623229 100644
--- a/src/3rdparty/chromium/ui/base/ime/init/input_method_initializer.cc
+++ b/src/3rdparty/chromium/ui/base/ime/init/input_method_initializer.cc
@@ -12,7 +12,7 @@
 #if defined(USE_AURA) && BUILDFLAG(IS_LINUX)
 #include "ui/base/ime/linux/fake_input_method_context.h"
 #include "ui/base/ime/linux/linux_input_method_context_factory.h"
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
 #include "ui/base/ime/init/input_method_factory.h"
 #include "ui/base/ime/win/tsf_bridge.h"
 #endif
@@ -21,13 +21,13 @@ namespace ui {
 
 void InitializeInputMethod() {
   TRACE_EVENT0("startup", "ui::InitializeInputMethod");
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   TSFBridge::Initialize();
 #endif
 }
 
 void ShutdownInputMethod() {
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   TSFBridge::Shutdown();
 #endif
 }
@@ -39,7 +39,7 @@ void InitializeInputMethodForTesting() {
                               -> std::unique_ptr<LinuxInputMethodContext> {
         return std::make_unique<FakeInputMethodContext>();
       });
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
   TSFBridge::InitializeForTesting();
 #endif
 }
@@ -49,7 +49,7 @@ void ShutdownInputMethodForTesting() {
   // The function owns the factory (as a static variable that's returned by
   // reference), so setting this to an empty factory will free the old one.
   GetInputMethodContextFactoryForTest() = LinuxInputMethodContextFactory();
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_MSVC)
   TSFBridge::Shutdown();
 #endif
 }
diff --git a/src/3rdparty/chromium/ui/base/ime/input_method.h b/src/3rdparty/chromium/ui/base/ime/input_method.h
index 03d9c95..74c0d32 100644
--- a/src/3rdparty/chromium/ui/base/ime/input_method.h
+++ b/src/3rdparty/chromium/ui/base/ime/input_method.h
@@ -47,7 +47,7 @@ class TextInputClient;
 // ui::InputMethod and owns it.
 class InputMethod {
  public:
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   typedef LRESULT NativeEventResult;
 #else
   typedef int32_t NativeEventResult;
@@ -66,7 +66,7 @@ class InputMethod {
   // Called when the top-level system window loses keyboard focus.
   virtual void OnBlur() = 0;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   // Called when the focused window receives native IME messages that are not
   // translated into other predefined event callbacks. Currently this method is
   // used only for IME functionalities specific to Windows.
diff --git a/src/3rdparty/chromium/ui/base/ime/mock_input_method.cc b/src/3rdparty/chromium/ui/base/ime/mock_input_method.cc
index 2218718..d9da6b7 100644
--- a/src/3rdparty/chromium/ui/base/ime/mock_input_method.cc
+++ b/src/3rdparty/chromium/ui/base/ime/mock_input_method.cc
@@ -67,7 +67,7 @@ void MockInputMethod::OnBlur() {
   observer_list_.Notify(&InputMethodObserver::OnBlur);
 }
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
 bool MockInputMethod::OnUntranslatedIMEMessage(const CHROME_MSG event,
                                                NativeEventResult* result) {
   if (result)
diff --git a/src/3rdparty/chromium/ui/base/ime/mock_input_method.h b/src/3rdparty/chromium/ui/base/ime/mock_input_method.h
index aa57ed6..c88e910 100644
--- a/src/3rdparty/chromium/ui/base/ime/mock_input_method.h
+++ b/src/3rdparty/chromium/ui/base/ime/mock_input_method.h
@@ -38,7 +38,7 @@ class COMPONENT_EXPORT(UI_BASE_IME) MockInputMethod : public InputMethod {
   void OnFocus() override;
   void OnBlur() override;
 
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_MSVC)
   bool OnUntranslatedIMEMessage(const CHROME_MSG event,
                                 NativeEventResult* result) override;
   void OnInputLocaleChanged() override;
diff --git a/src/3rdparty/chromium/ui/base/ime/win/BUILD.gn b/src/3rdparty/chromium/ui/base/ime/win/BUILD.gn
index 9db7511..ec8bcdd 100644
--- a/src/3rdparty/chromium/ui/base/ime/win/BUILD.gn
+++ b/src/3rdparty/chromium/ui/base/ime/win/BUILD.gn
@@ -30,7 +30,24 @@ jumbo_component("win") {
     "virtual_keyboard_debounce_timer.cc",
     "virtual_keyboard_debounce_timer.h",
   ]
-  jumbo_excluded_sources = [ "tsf_text_store.cc" ]
+  if (is_mingw) {
+    sources -= [
+      "input_method_win_tsf.cc",
+      "input_method_win_tsf.h",
+      "on_screen_keyboard_display_manager_input_pane.cc", # requires <wrl/event.h>
+      "on_screen_keyboard_display_manager_input_pane.h",
+      "tsf_bridge.cc",
+      "tsf_bridge.h",
+      "tsf_event_router.cc",
+      "tsf_event_router.h",
+      "tsf_input_scope.cc",
+      "tsf_input_scope.h",
+      "tsf_text_store.cc",
+      "tsf_text_store.h",
+    ]
+  } else {
+    jumbo_excluded_sources = [ "tsf_text_store.cc" ]
+  }
 
   defines = [ "IS_UI_BASE_IME_WIN_IMPL" ]
 
diff --git a/src/3rdparty/chromium/ui/base/ime/win/input_method_win_base.cc b/src/3rdparty/chromium/ui/base/ime/win/input_method_win_base.cc
index a21238e..aaf5886 100644
--- a/src/3rdparty/chromium/ui/base/ime/win/input_method_win_base.cc
+++ b/src/3rdparty/chromium/ui/base/ime/win/input_method_win_base.cc
@@ -42,10 +42,14 @@ constexpr size_t kExtraNumberOfChars = 20;
 
 std::unique_ptr<VirtualKeyboardController> CreateKeyboardController(
     HWND attached_window_handle) {
+#if defined(COMPILER_MSVC)
   if (base::win::GetVersion() >= base::win::Version::WIN10_RS4) {
     return std::make_unique<OnScreenKeyboardDisplayManagerInputPane>(
         attached_window_handle);
   } else {
+#else
+  if (base::win::GetVersion() >= base::win::Version::WIN8) {
+#endif
     return std::make_unique<OnScreenKeyboardDisplayManagerTabTip>(
         attached_window_handle);
   }
diff --git a/src/3rdparty/chromium/ui/base/ime/win/on_screen_keyboard_display_manager_input_pane.h b/src/3rdparty/chromium/ui/base/ime/win/on_screen_keyboard_display_manager_input_pane.h
index f89b23d..e43e369 100644
--- a/src/3rdparty/chromium/ui/base/ime/win/on_screen_keyboard_display_manager_input_pane.h
+++ b/src/3rdparty/chromium/ui/base/ime/win/on_screen_keyboard_display_manager_input_pane.h
@@ -5,6 +5,8 @@
 #ifndef UI_BASE_IME_WIN_ON_SCREEN_KEYBOARD_DISPLAY_MANAGER_INPUT_PANE_H_
 #define UI_BASE_IME_WIN_ON_SCREEN_KEYBOARD_DISPLAY_MANAGER_INPUT_PANE_H_
 
+#if defined(COMPILER_MSVC)
+
 #include <inputpaneinterop.h>
 #include <windows.ui.viewmanagement.h>
 #include <wrl/client.h>
@@ -83,4 +85,6 @@ class COMPONENT_EXPORT(UI_BASE_IME_WIN)
 
 }  // namespace ui
 
+#endif // COMPILER_MSVC
+
 #endif  // UI_BASE_IME_WIN_ON_SCREEN_KEYBOARD_DISPLAY_MANAGER_INPUT_PANE_H_
diff --git a/src/3rdparty/chromium/ui/display/BUILD.gn b/src/3rdparty/chromium/ui/display/BUILD.gn
index 180c1a9..1780e8b 100644
--- a/src/3rdparty/chromium/ui/display/BUILD.gn
+++ b/src/3rdparty/chromium/ui/display/BUILD.gn
@@ -98,6 +98,12 @@ jumbo_component("display") {
       "win/uwp_text_scale_factor.cc",
       "win/uwp_text_scale_factor.h",
     ]
+    if (is_mingw) {
+      sources -= [
+        "win/uwp_text_scale_factor.cc",
+        "win/uwp_text_scale_factor.h",
+      ]
+    }
   }
 
   if (is_ios) {
diff --git a/src/3rdparty/chromium/ui/display/win/dpi.cc b/src/3rdparty/chromium/ui/display/win/dpi.cc
index cc3c603..2e91de8 100644
--- a/src/3rdparty/chromium/ui/display/win/dpi.cc
+++ b/src/3rdparty/chromium/ui/display/win/dpi.cc
@@ -8,7 +8,9 @@
 
 #include "base/win/scoped_hdc.h"
 #include "ui/display/display.h"
+#if defined(COMPILER_MSVC)
 #include "ui/display/win/uwp_text_scale_factor.h"
+#endif
 
 namespace display {
 namespace win {
@@ -37,7 +39,9 @@ int GetDPIFromScalingFactor(float device_scaling_factor) {
 }
 
 double GetAccessibilityFontScale() {
+#if defined(COMPILER_MSVC)
   return 1.0 / UwpTextScaleFactor::Instance()->GetTextScaleFactor();
+#endif
 }
 
 namespace internal {
diff --git a/src/3rdparty/chromium/ui/display/win/screen_win.cc b/src/3rdparty/chromium/ui/display/win/screen_win.cc
index 0d27762..70fcc1d 100644
--- a/src/3rdparty/chromium/ui/display/win/screen_win.cc
+++ b/src/3rdparty/chromium/ui/display/win/screen_win.cc
@@ -65,9 +65,13 @@ std::optional<int> GetPerMonitorDPI(HMONITOR monitor) {
 
 float GetScaleFactorForDPI(int dpi, bool include_accessibility) {
   const float scale = display::win::internal::GetScalingFactorFromDPI(dpi);
+#if defined(COMPILER_MSVC)
   return include_accessibility
              ? (scale * UwpTextScaleFactor::Instance()->GetTextScaleFactor())
              : scale;
+#else
+  return scale;
+#endif
 }
 
 // Gets the raw monitor scale factor.
@@ -969,10 +973,12 @@ void ScreenWin::Initialize() {
       base::BindRepeating(&ScreenWin::OnWndProc, base::Unretained(this)));
   UpdateFromDisplayInfos(GetDisplayInfosFromSystem());
 
+#if defined(COMPILER_MSVC)
   // We want to remember that we've observed a screen metrics object so that we
   // can remove ourselves as an observer at some later point (either when the
   // metrics object notifies us it's going away or when we are destructed).
   scale_factor_observation_.Observe(UwpTextScaleFactor::Instance());
+#endif
 }
 
 std::optional<MONITORINFOEX> ScreenWin::MonitorInfoFromScreenPoint(
@@ -1159,6 +1165,7 @@ int ScreenWin::GetSystemMetricsForScaleFactor(float scale_factor,
                           GetPrimaryDisplay().device_scale_factor());
 }
 
+#if defined(COMPILER_MSVC)
 void ScreenWin::OnUwpTextScaleFactorChanged() {
   UpdateAllDisplaysAndNotify();
 }
@@ -1167,6 +1174,7 @@ void ScreenWin::OnUwpTextScaleFactorCleanup(UwpTextScaleFactor* source) {
   scale_factor_observation_.Reset();
   UwpTextScaleFactor::Observer::OnUwpTextScaleFactorCleanup(source);
 }
+#endif
 
 bool ScreenWin::PerProcessDPIAwarenessDisabledForTesting() const {
   return per_process_dpi_awareness_disabled_for_testing_;
diff --git a/src/3rdparty/chromium/ui/display/win/screen_win.h b/src/3rdparty/chromium/ui/display/win/screen_win.h
index 19aefcf..6efba30 100644
--- a/src/3rdparty/chromium/ui/display/win/screen_win.h
+++ b/src/3rdparty/chromium/ui/display/win/screen_win.h
@@ -40,8 +40,12 @@ class DisplayInfo;
 }  // namespace internal
 
 class DISPLAY_EXPORT ScreenWin : public Screen,
+#if defined(COMPILER_MSVC)
                                  public ColorProfileReader::Client,
                                  public UwpTextScaleFactor::Observer {
+#else
+                                 public ColorProfileReader::Client {
+#endif
  public:
   ScreenWin();
 
@@ -276,8 +280,10 @@ class DISPLAY_EXPORT ScreenWin : public Screen,
   //-----------------------------------------------------------------
   // UwpTextScaleFactor::Observer:
 
+#if defined(COMPILER_MSVC)
   void OnUwpTextScaleFactorChanged() override;
   void OnUwpTextScaleFactorCleanup(UwpTextScaleFactor* source) override;
+#endif
 
   // Tests don't want to use the actual DPI settings of the monitor(s) on
   // the machine running the test.
@@ -307,8 +313,10 @@ class DISPLAY_EXPORT ScreenWin : public Screen,
   // Information gathered from DXGI adapters and outputs.
   gfx::mojom::DXGIInfoPtr dxgi_info_;
 
+#if defined(COMPILER_MSVC)
   base::ScopedObservation<UwpTextScaleFactor, UwpTextScaleFactor::Observer>
       scale_factor_observation_{this};
+#endif
 
   // Used to avoid calling GetSystemMetricsForDpi in unit tests.
   bool per_process_dpi_awareness_disabled_for_testing_ = false;
diff --git a/src/3rdparty/chromium/ui/events/BUILD.gn b/src/3rdparty/chromium/ui/events/BUILD.gn
index 8e3b944..1aa519d 100644
--- a/src/3rdparty/chromium/ui/events/BUILD.gn
+++ b/src/3rdparty/chromium/ui/events/BUILD.gn
@@ -250,6 +250,10 @@ jumbo_component("events") {
       "win/stylus_handwriting_properties_win.cc",
       "win/system_event_state_lookup.cc",
     ]
+
+    if (is_mingw) {
+      sources -= [ "win/stylus_handwriting_properties_win.cc" ] # requires ShellHandwriting.h
+    }
   }
 
   if (is_mac) {
@@ -499,6 +503,10 @@ jumbo_component("gesture_detection") {
 
   defines = [ "GESTURE_DETECTION_IMPLEMENTATION" ]
 
+  if (is_mingw && is_clang) {
+    cflags_cc = [ "-Wno-vla-cxx-extension" ]
+  }
+
   if (is_android) {
     sources += [ "gesture_detection/gesture_configuration_android.cc" ]
   } else if (is_ios) {
diff --git a/src/3rdparty/chromium/ui/gfx/BUILD.gn b/src/3rdparty/chromium/ui/gfx/BUILD.gn
index 9e55ad4..398f962 100644
--- a/src/3rdparty/chromium/ui/gfx/BUILD.gn
+++ b/src/3rdparty/chromium/ui/gfx/BUILD.gn
@@ -56,10 +56,12 @@ if (is_win) {
   config("gfx_win_linker_flags") {
     # This is in a separate config so the flags can be applied to dependents.
     # ldflags in GN aren't automatically inherited.
-    ldflags = [
-      # Linking with OneCore.lib causes the following dlls to load.
-      "/DELAYLOAD:api-ms-win-core-handle-l1-1-0.dll",
-    ]
+    if (is_msvc) {
+      ldflags = [
+        # Linking with OneCore.lib causes the following dlls to load.
+        "/DELAYLOAD:api-ms-win-core-handle-l1-1-0.dll",
+      ]
+    }
   }
 }
 
@@ -245,6 +247,14 @@ jumbo_component("gfx") {
       "win/window_impl.cc",
       "win/window_impl.h",
     ]
+    if (is_mingw) {
+      sources -= [
+        "win/d3d_shared_fence.cc",
+        "win/d3d_shared_fence.h",
+        "win/text_analysis_source.cc",
+        "win/text_analysis_source.h"
+      ]
+    }
   }
   if (is_apple) {
     sources += [ "font_render_params_mac.cc" ]
diff --git a/src/3rdparty/chromium/ui/gfx/codec/BUILD.gn b/src/3rdparty/chromium/ui/gfx/codec/BUILD.gn
index def3b78..6ad85d8 100644
--- a/src/3rdparty/chromium/ui/gfx/codec/BUILD.gn
+++ b/src/3rdparty/chromium/ui/gfx/codec/BUILD.gn
@@ -36,7 +36,7 @@ jumbo_component("codec") {
     ]
   }
 
-  if (is_win) {
+  if (is_msvc) {
     cflags = [ "/wd4324" ]  # Structure was padded due to __declspec(align()),
                             # which is uninteresting.
   }
diff --git a/src/3rdparty/chromium/ui/gl/BUILD.gn b/src/3rdparty/chromium/ui/gl/BUILD.gn
index 8c44b2d..ae2ac89 100644
--- a/src/3rdparty/chromium/ui/gl/BUILD.gn
+++ b/src/3rdparty/chromium/ui/gl/BUILD.gn
@@ -276,15 +276,24 @@ jumbo_component("gl") {
       "vsync_thread_win_dxgi.h",
     ]
 
-    libs = [
-      "dxgi.lib",
-      "dxguid.lib",
-      "dwmapi.lib",
-    ]
-    ldflags = [
-      "/DELAYLOAD:dwmapi.dll",
-      "/DELAYLOAD:dxgi.dll",
-    ]
+    if (is_mingw) {
+      configs -= [ "//build/config/compiler:no_rtti" ]
+      libs = [
+        "dxgi",
+        "dxguid",
+        "dwmapi",
+      ]
+    } else {
+      libs = [
+        "dxgi.lib",
+        "dxguid.lib",
+        "dwmapi.lib",
+      ]
+      ldflags = [
+        "/DELAYLOAD:dwmapi.dll",
+        "/DELAYLOAD:dxgi.dll",
+      ]
+    }
     deps += [
       "//ui/base",
       "//ui/display:display",
diff --git a/src/3rdparty/chromium/ui/gl/dc_layer_tree.cc b/src/3rdparty/chromium/ui/gl/dc_layer_tree.cc
index f44333e..2a077eb 100644
--- a/src/3rdparty/chromium/ui/gl/dc_layer_tree.cc
+++ b/src/3rdparty/chromium/ui/gl/dc_layer_tree.cc
@@ -32,6 +32,7 @@ bool NeedSwapChainPresenter(const DCLayerOverlayParams* overlay) {
                                        DCLayerOverlayType::kDCompVisualContent;
 }
 
+#if !defined(COMPILER_GCC) // FIXME: missing Microsoft::WRL::ComPtr<IDCompositionVisual3>
 // Unconditionally get a IDCompositionVisual2 as a IDCompositionVisual3.
 //
 // |IDCompositionVisual3| should be available since Windows 8.1, but we noticed
@@ -68,6 +69,7 @@ D2D_MATRIX_4X4_F TransformToD2D_MATRIX_4X4_F(const gfx::Transform& transform) {
                           t.rc(0, 2), t.rc(1, 2), t.rc(2, 2), t.rc(3, 2),
                           t.rc(0, 3), t.rc(1, 3), t.rc(2, 3), t.rc(3, 3));
 }
+#endif
 
 // The size the surfaces in the pool. Used in |VisualSubtree::Update| to
 // determine how to scale the background color visual. This can be any size
@@ -610,6 +612,7 @@ bool DCLayerTree::VisualTree::VisualSubtree::Update(
     }
   }
 
+#if !defined(COMPILER_GCC)
   if (opacity_changed) {
     if (opacity_ != 1) {
       hr = CheckedCastToVisual3(clip_visual_)->SetOpacity(opacity_);
@@ -626,6 +629,7 @@ bool DCLayerTree::VisualTree::VisualSubtree::Update(
       CHECK_EQ(hr, S_OK);
     }
   }
+#endif
 
   if (rounded_corner_bounds_changed) {
     if (!rounded_corner_bounds_.IsEmpty()) {
@@ -688,6 +692,7 @@ bool DCLayerTree::VisualTree::VisualSubtree::Update(
     }
   }
 
+#if !defined(COMPILER_GCC)
   if (quad_to_root_transform_changed) {
     if (quad_to_root_transform_.Is2dTransform()) {
       const D2D_MATRIX_3X2_F matrix =
@@ -702,6 +707,7 @@ bool DCLayerTree::VisualTree::VisualSubtree::Update(
       CHECK_EQ(hr, S_OK);
     }
   }
+#endif
 
   if (nearest_neighbor_filter_changed) {
     hr = transform_visual_->SetBitmapInterpolationMode(
@@ -768,6 +774,7 @@ bool DCLayerTree::VisualTree::VisualSubtree::Update(
     // update. No visual changes are needed in this case.
   }
 
+#if !defined(COMPILER_GCC)
   if (quad_rect_changed || background_color_surface_changed ||
       background_color_changed) {
     if (!background_color_surface_ || background_color.fA == 0.0) {
@@ -791,6 +798,7 @@ bool DCLayerTree::VisualTree::VisualSubtree::Update(
       CHECK_EQ(hr, S_OK);
     }
   }
+#endif
 
   if (quad_to_root_transform_changed || quad_rect_changed ||
       allow_antialiasing_changed) {
@@ -1274,6 +1283,7 @@ base::expected<void, CommitError> DCLayerTree::CommitAndClearPendingOverlays(
   solid_color_surface_pool_->TrimAfterCommit();
 
   return status;
 }
 
 size_t DCLayerTree::GetNumSurfacesInPoolForTesting() const {
@@ -1396,8 +1405,10 @@ base::expected<void, CommitError> DCLayerTree::CommitAndClearPendingOverlays(
           CHECK_EQ(hr, S_OK);
 
+          Microsoft::WRL::ComPtr<IUnknown> primary_plane_surface_unknown;
+          primary_plane_surface_.As(&primary_plane_surface_unknown);
           overlay.overlay_image = DCLayerOverlayImage(
-              overlay.overlay_image->size(), primary_plane_surface_,
+              overlay.overlay_image->size(), primary_plane_surface_unknown,
               primary_plane_surface_serial_++);
           did_update_primary_plane_damage = true;
         } else {
diff --git a/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc b/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc
index 7f71dc1..20d6d21 100644
--- a/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc
+++ b/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.cc
@@ -63,6 +63,7 @@ void DelegatedInkPointRendererGpu::InitMessagePipeline(
 bool DelegatedInkPointRendererGpu::Initialize(
     IDCompositionDevice2* dcomp_device2,
     IDXGISwapChain1* root_swap_chain) {
+#if !defined(COMPILER_GCC)
   if (swap_chain_ == root_swap_chain && delegated_ink_trail_) {
     return true;
   }
@@ -89,6 +90,9 @@ bool DelegatedInkPointRendererGpu::Initialize(
   force_new_ink_trail_ = true;
 
   return true;
+#else
+  return false;
+#endif
 }
 
 bool DelegatedInkPointRendererGpu::DelegatedInkIsSupported(
@@ -99,9 +103,11 @@ bool DelegatedInkPointRendererGpu::DelegatedInkIsSupported(
     return false;
   }
 
+#if !defined(COMPILER_GCC)
   Microsoft::WRL::ComPtr<IDCompositionInkTrailDevice> ink_trail_device;
   HRESULT hr = dcomp_device.As(&ink_trail_device);
   return hr == S_OK;
+#endif
 }
 
 uint64_t DelegatedInkPointRendererGpu::GetMaximumNumberOfPointerIdsForTesting()
@@ -161,6 +167,7 @@ void DelegatedInkPointRendererGpu::ReportPointsDrawn() {
 
 void DelegatedInkPointRendererGpu::SetDelegatedInkTrailStartPoint(
     std::unique_ptr<gfx::DelegatedInkMetadata> metadata) {
+#if !defined(COMPILER_GCC)
   TRACE_EVENT_WITH_FLOW1(
       "delegated_ink_trails",
       "DelegatedInkPointRendererGpu::SetDelegatedInkTrailStartPoint",
@@ -253,6 +260,7 @@ void DelegatedInkPointRendererGpu::SetDelegatedInkTrailStartPoint(
   metadata_ = std::move(metadata);
   DrawSavedTrailPoints();
   force_new_ink_trail_ = false;
+#endif
 }
 
 void DelegatedInkPointRendererGpu::StoreDelegatedInkPoint(
@@ -481,8 +489,10 @@ DelegatedInkPointRendererGpu::MakeDelegatedInkOverlay(
   ink_layer.clip_rect = std::make_optional<gfx::Rect>(
       0, 0, presentation_rect.right(), presentation_rect.bottom());
   ink_layer.transform = gfx::Transform::MakeTranslation(
       metadata->presentation_area().OffsetFromOrigin());
+#if !defined(COMPILER_GCC)
   ink_layer.overlay_image = DCLayerOverlayImage(presentation_area_enclosed_size,
                                                 delegated_ink_trail_);
+#endif
   ink_layer.layer_id = gfx::OverlayLayerId::MakeVizInternal(
       gfx::OverlayLayerId::VizInternalId::kDelegatedInkTrail);
   SetDelegatedInkTrailStartPoint(std::move(metadata));
@@ -498,6 +508,7 @@ bool DelegatedInkPointRendererGpu::DrawDelegatedInkPoint(
     return false;
   }
 
+#if !defined(COMPILER_GCC)
   DCHECK(delegated_ink_trail_);
 
   DCompositionInkTrailPoint ink_point;
@@ -535,6 +546,9 @@ bool DelegatedInkPointRendererGpu::DrawDelegatedInkPoint(
   }
   delegated_ink_points_[point.pointer_id()][point] = token;
   return true;
+#else
+  return false;
+#endif
 }
 
 void DelegatedInkPointRendererGpu::InitializeForTesting(
diff --git a/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.h b/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.h
index 7864d26..f1763ed 100644
--- a/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.h
+++ b/src/3rdparty/chromium/ui/gl/delegated_ink_point_renderer_gpu.h
@@ -128,15 +128,19 @@ class GL_EXPORT DelegatedInkPointRendererGpu
 
   bool DrawDelegatedInkPoint(const gfx::DelegatedInkPoint& point);
 
+#if !defined(COMPILER_GCC) /* FIXME: missing IDCompositionDelegatedInkTrail from wrl/client.h */
   // The delegated ink trail object that the ink trail is drawn on. This is the
   // content of the ink visual.
   Microsoft::WRL::ComPtr<IDCompositionDelegatedInkTrail> delegated_ink_trail_;
+#endif
 
   // The swap chain whose updates |delegated_ink_trail_| is synchronized with.
   // If null, |delegated_ink_trail_| is synchronized with DComp commit.
   raw_ptr<IDXGISwapChain1> swap_chain_ = nullptr;
 
+#if !defined(COMPILER_GCC) /* FIXME: missing IDCompositionInkTrailDevice from wrl/client.h */
   Microsoft::WRL::ComPtr<IDCompositionInkTrailDevice> ink_trail_device_;
+#endif
 
   // The most recent metadata received. The metadata marks the last point of
   // the app rendered stroke, which corresponds to the first point of the
diff --git a/src/3rdparty/chromium/ui/gl/direct_composition_support.cc b/src/3rdparty/chromium/ui/gl/direct_composition_support.cc
index e8e41ff..bf76367 100644
--- a/src/3rdparty/chromium/ui/gl/direct_composition_support.cc
+++ b/src/3rdparty/chromium/ui/gl/direct_composition_support.cc
@@ -624,6 +624,7 @@ using PFN_DCOMPOSITION_WAIT = HRESULT(WINAPI*)(UINT count,
                                                DWORD timeoutInMs);
 PFN_DCOMPOSITION_WAIT g_wait_for_compositor_clock_function = nullptr;
 
+#if !defined(COMPILER_GCC)
 // DCompositionGetFrameId function pointer
 using PFN_DCOMPOSITION_GET_FRAME_ID =
     HRESULT(WINAPI*)(COMPOSITION_FRAME_ID_TYPE frameIdType,
@@ -638,6 +639,7 @@ using PFN_DCOMPOSITION_GET_STATISTICS =
                      COMPOSITION_TARGET_ID* targetIds,
                      UINT* actualTargetIdCount);
 PFN_DCOMPOSITION_GET_STATISTICS g_get_statistics_function = nullptr;
+#endif
 
 HRESULT DCompositionWaitForCompositorClock(UINT count,
                                            const HANDLE* handles,
@@ -646,6 +648,7 @@ HRESULT DCompositionWaitForCompositorClock(UINT count,
   return g_wait_for_compositor_clock_function(count, handles, timeoutInMs);
 }
 
+#if !defined(COMPILER_GCC)
 HRESULT DCompositionGetFrameId(COMPOSITION_FRAME_ID_TYPE frameIdType,
                                COMPOSITION_FRAME_ID* frameId) {
   DCHECK(g_get_frame_id_function);
@@ -661,6 +664,7 @@ HRESULT DCompositionGetStatistics(COMPOSITION_FRAME_ID frameId,
   return g_get_statistics_function(frameId, frameStats, targetIdCount,
                                    targetIds, actualTargetIdCount);
 }
+#endif
 
 void InitializeDirectComposition(
     Microsoft::WRL::ComPtr<ID3D11Device> d3d11_device) {
@@ -720,6 +724,7 @@ void InitializeDirectComposition(
   g_d3d11_device = d3d11_device.Detach();
 
   if (features::UseCompositorClockVSyncInterval()) {
+#if !defined(COMPILER_GCC)
     g_get_frame_id_function = reinterpret_cast<PFN_DCOMPOSITION_GET_FRAME_ID>(
         ::GetProcAddress(dcomp_module, "DCompositionGetFrameId"));
     CHECK(g_get_frame_id_function);
@@ -728,6 +733,7 @@ void InitializeDirectComposition(
         reinterpret_cast<PFN_DCOMPOSITION_GET_STATISTICS>(
             ::GetProcAddress(dcomp_module, "DCompositionGetStatistics"));
     CHECK(g_get_statistics_function);
+#endif
 
     g_wait_for_compositor_clock_function =
         reinterpret_cast<PFN_DCOMPOSITION_WAIT>(::GetProcAddress(
diff --git a/src/3rdparty/chromium/ui/gl/direct_composition_support.h b/src/3rdparty/chromium/ui/gl/direct_composition_support.h
index 4b376cd..d88798c 100644
--- a/src/3rdparty/chromium/ui/gl/direct_composition_support.h
+++ b/src/3rdparty/chromium/ui/gl/direct_composition_support.h
@@ -25,6 +25,7 @@ HRESULT DCompositionWaitForCompositorClock(UINT count,
                                            const HANDLE* handles,
                                            DWORD timeoutInMs);
 
+#if !defined(COMPILER_GCC) /* FIXME: missing COMPOSITION_FRAME_ID, COMPOSITION_FRAME_ID_TYPE from dcomp.h */
 // Wrapper for DcompositionGetFrameId Win32 dcomp.h function
 HRESULT DCompositionGetFrameId(COMPOSITION_FRAME_ID_TYPE frameIdType,
                                COMPOSITION_FRAME_ID* frameId);
@@ -35,6 +36,7 @@ HRESULT DCompositionGetStatistics(COMPOSITION_FRAME_ID frameId,
                                   UINT targetIdCount,
                                   COMPOSITION_TARGET_ID* targetIds,
                                   UINT* actualTargetIdCount);
+#endif
 
 // Initialize direct composition with the given d3d11 device.
 GL_EXPORT void InitializeDirectComposition(
diff --git a/src/3rdparty/chromium/ui/gl/init/BUILD.gn b/src/3rdparty/chromium/ui/gl/init/BUILD.gn
index f8fd6c7..ed77b43 100644
--- a/src/3rdparty/chromium/ui/gl/init/BUILD.gn
+++ b/src/3rdparty/chromium/ui/gl/init/BUILD.gn
@@ -52,8 +52,12 @@ jumbo_component("init") {
       "gl_initializer_win.cc",
     ]
 
-    libs = [ "dwmapi.lib" ]
-    ldflags = [ "/DELAYLOAD:dwmapi.dll" ]
+    if (is_mingw) {
+      libs = [ "dwmapi" ]
+    } else {
+      libs = [ "dwmapi.lib" ]
+      ldflags = [ "/DELAYLOAD:dwmapi.dll" ]
+    }
   } else if (is_mac) {
     sources += [
       "gl_factory_mac.cc",
diff --git a/src/3rdparty/chromium/ui/gl/swap_chain_presenter.cc b/src/3rdparty/chromium/ui/gl/swap_chain_presenter.cc
index d15792c..99658d2 100644
--- a/src/3rdparty/chromium/ui/gl/swap_chain_presenter.cc
+++ b/src/3rdparty/chromium/ui/gl/swap_chain_presenter.cc
@@ -413,6 +413,7 @@ bool TryDisableDesktopPlane(IDXGIDecodeSwapChain* decode_swap_chain,
     return false;
   }
 
+#if !defined(COMPILER_GCC)
   // Set the target region to the specified rectangle if necessary.
   RECT target_region = target_rect.ToRECT();
   if (target_region != original_target_rect) {
@@ -424,6 +425,7 @@ bool TryDisableDesktopPlane(IDXGIDecodeSwapChain* decode_swap_chain,
       return false;
     }
   }
+#endif
 
   return true;
 }
diff --git a/src/3rdparty/chromium/ui/native_theme/BUILD.gn b/src/3rdparty/chromium/ui/native_theme/BUILD.gn
index f7ac0fc..1bf278b 100644
--- a/src/3rdparty/chromium/ui/native_theme/BUILD.gn
+++ b/src/3rdparty/chromium/ui/native_theme/BUILD.gn
@@ -128,7 +128,11 @@ if (is_win) {
       "//ui/gfx",
     ]
 
-    libs = [ "uxtheme.lib" ]
+    if (is_mingw) {
+      libs = [ "uxtheme" ]
+    } else {
+      libs = [ "uxtheme.lib" ]
+    }
   }
 } else {
   source_set("native_theme_browser") {
diff --git a/src/3rdparty/chromium/ui/shell_dialogs/BUILD.gn b/src/3rdparty/chromium/ui/shell_dialogs/BUILD.gn
index 2c003ec..9e59882 100644
--- a/src/3rdparty/chromium/ui/shell_dialogs/BUILD.gn
+++ b/src/3rdparty/chromium/ui/shell_dialogs/BUILD.gn
@@ -41,7 +41,7 @@ jumbo_component("shell_dialogs") {
     "//url",
   ]
 
-  if (is_chromeos || is_castos) {
+  if (is_chromeos || is_castos || is_mingw) {
     sources += [ "shell_dialog_stub.cc" ]
   }
 
@@ -84,7 +84,7 @@ jumbo_component("shell_dialogs") {
     ]
   }
 
-  if (is_win) {
+  if (is_msvc) {
     sources += [
       "auto_close_dialog_event_handler_win.cc",
       "auto_close_dialog_event_handler_win.h",
diff --git a/src/3rdparty/chromium/ui/gfx/win/direct_write.cc b/src/3rdparty/chromium/ui/gfx/win/direct_write.cc
index 7641f6b..9d272de 100644
--- a/src/3rdparty/chromium/ui/gfx/win/direct_write.cc
+++ b/src/3rdparty/chromium/ui/gfx/win/direct_write.cc
@@ -74,9 +74,15 @@ bool GetFontCollection(Microsoft::WRL::ComPtr<IDWriteFactory>& factory,
 
     for (UINT32 font_index = 0; font_index < n_fonts; ++font_index) {
       Microsoft::WRL::ComPtr<IDWriteFontFaceReference> font_face;
+#if defined(__MINGW32__)
+      hr = factory3->CreateFontFaceReference_(font_file.Get(), font_index,
+                                              DWRITE_FONT_SIMULATIONS_NONE,
+                                              &font_face);
+#else
       hr = factory3->CreateFontFaceReference(font_file.Get(), font_index,
                                              DWRITE_FONT_SIMULATIONS_NONE,
                                              &font_face);
+#endif
       DCHECK(SUCCEEDED(hr));
 
       hr = font_set_builder->AddFontFaceReference(font_face.Get());
diff --git a/src/3rdparty/chromium/ui/gl/vsync_provider_win_dcomp.h b/src/3rdparty/chromium/ui/gl/vsync_provider_win_dcomp.h
index 3917a54..4156127 100644
--- a/src/3rdparty/chromium/ui/gl/vsync_provider_win_dcomp.h
+++ b/src/3rdparty/chromium/ui/gl/vsync_provider_win_dcomp.h
@@ -29,7 +29,9 @@ class GL_EXPORT VSyncProviderWinDComp : public gfx::VSyncProvider {
   bool IsHWClock() const override;
 
  private:
+#if !defined(COMPILER_GCC)
   std::optional<COMPOSITION_FRAME_STATS> prev_frame_stats_;
+#endif
 };
 
 }  // namespace gl
