diff --git a/src/3rdparty/chromium/v8/BUILD.gn b/src/3rdparty/chromium/v8/BUILD.gn
index 49da8fc..d0fca8c 100644
--- a/src/3rdparty/chromium/v8/BUILD.gn
+++ b/src/3rdparty/chromium/v8/BUILD.gn
@@ -40,10 +40,10 @@ declare_args() {
   v8_enable_future = false
 
   # Sets -DENABLE_SYSTEM_INSTRUMENTATION. Enables OS-dependent event tracing
-  v8_enable_system_instrumentation = (is_win || is_mac) && !v8_use_perfetto
+  v8_enable_system_instrumentation = (is_msvc || is_mac) && !v8_use_perfetto
 
   # Sets -DV8_ENABLE_ETW_STACK_WALKING. Enables ETW Stack Walking
-  v8_enable_etw_stack_walking = is_win
+  v8_enable_etw_stack_walking = is_msvc
 
   # Sets the GUID for the ETW provider
   v8_etw_guid = ""
@@ -1405,18 +1405,22 @@ config("toolchain") {
 
   if (v8_current_cpu == "x86") {
     defines += [ "V8_TARGET_ARCH_IA32" ]
-    if (is_win) {
+    if (is_msvc) {
       # Ensure no surprising artifacts from 80bit double math with x86.
       cflags += [ "/arch:SSE2" ]
+    } else {
+      cflags += [ "-msse2" ]
     }
   }
   if (v8_current_cpu == "x64") {
     defines += [ "V8_TARGET_ARCH_X64" ]
-    if (is_win) {
+    if (is_msvc) {
       # Increase the initial stack size. The default is 1MB, this is 2MB. This
       # applies only to executables and shared libraries produced by V8 since
       # ldflags are not pushed to dependants.
       ldflags += [ "/STACK:2097152" ]
+    } else {
+      ldflags += [ "-Wl,--stack,2097152" ]
     }
   }
   if (is_android && v8_android_log_stdout) {
@@ -1471,7 +1475,7 @@ config("toolchain") {
   }
 
   if (v8_no_inline) {
-    if (is_win) {
+    if (is_msvc) {
       cflags += [ "/Ob0" ]
     } else {
       cflags += [
@@ -1515,7 +1519,7 @@ config("toolchain") {
     ]
   }
 
-  if (is_win) {
+  if (is_msvc) {
     cflags += [
       "/wd4245",  # Conversion with signed/unsigned mismatch.
       "/wd4267",  # Conversion with possible loss of data.
@@ -1539,7 +1543,7 @@ config("toolchain") {
     ]
   }
 
-  if (!is_clang && is_win) {
+  if (!is_clang && is_msvc) {
     cflags += [
       "/wd4506",  # Benign "no definition for inline function"
 
@@ -1680,7 +1684,7 @@ config("toolchain") {
     ]
   }
 
-  if (!is_clang && !is_win) {
+  if (!is_clang && !is_msvc) {
     cflags += [
       # Disable gcc warnings for optimizations based on the assumption that
       # signed overflow does not occur. Generates false positives (see
@@ -6077,7 +6081,7 @@ v8_source_set("v8_base_without_compiler") {
   if (v8_enable_i18n_support) {
     deps += [ ":run_gen-regexp-special-case" ]
     sources += [ "$target_gen_dir/src/regexp/special-case.cc" ]
-    if (is_win) {
+    if (is_msvc) {
       deps += [ "$v8_icu_path:icudata" ]
     }
   } else {
@@ -6497,13 +6501,13 @@ v8_component("v8_libbase") {
     defines += [ "_CRT_RAND_S" ]  # for rand_s()
 
     libs = [
-      "dbghelp.lib",
-      "winmm.lib",
-      "ws2_32.lib",
+      "dbghelp",
+      "winmm",
+      "ws2_32",
     ]
 
     if (v8_enable_etw_stack_walking) {
-      libs += [ "advapi32.lib" ]  # Needed for TraceLoggingProvider.h
+      libs += [ "advapi32" ]  # Needed for TraceLoggingProvider.h
     }
 
     data_deps += [ "//build/win:runtime_libs" ]
@@ -6715,7 +6719,7 @@ v8_source_set("v8_heap_base") {
     "src/heap/base/worklist.cc",
   ]
 
-  if (is_clang || !is_win) {
+  if (is_clang || !is_msvc) {
     if (current_cpu == "x64") {
       sources += [ "src/heap/base/asm/x64/push_registers_asm.cc" ]
     } else if (current_cpu == "x86") {
@@ -6735,7 +6739,7 @@ v8_source_set("v8_heap_base") {
     } else if (current_cpu == "riscv64" || current_cpu == "riscv32") {
       sources += [ "src/heap/base/asm/riscv/push_registers_asm.cc" ]
     }
-  } else if (is_win) {
+  } else if (is_msvc) {
     if (current_cpu == "x64") {
       sources += [ "src/heap/base/asm/x64/push_registers_masm.asm" ]
     } else if (current_cpu == "x86") {
diff --git a/src/3rdparty/chromium/v8/include/v8-fast-api-calls.h b/src/3rdparty/chromium/v8/include/v8-fast-api-calls.h
index 74d4cb1..cca7748 100644
--- a/src/3rdparty/chromium/v8/include/v8-fast-api-calls.h
+++ b/src/3rdparty/chromium/v8/include/v8-fast-api-calls.h
@@ -328,7 +328,7 @@ class CTypeInfo {
 struct FastApiTypedArrayBase {
  public:
   // Returns the length in number of elements.
-  size_t V8_EXPORT length() const { return length_; }
+  size_t length() const { return length_; }
   // Checks whether the given index is within the bounds of the collection.
   void V8_EXPORT ValidateIndex(size_t index) const;
 
diff --git a/src/3rdparty/chromium/v8/include/v8config.h b/src/3rdparty/chromium/v8/include/v8config.h
index e649d8c..7a09850 100644
--- a/src/3rdparty/chromium/v8/include/v8config.h
+++ b/src/3rdparty/chromium/v8/include/v8config.h
@@ -344,6 +344,13 @@ path. Add it with -I<path> to the command line
 
 #if defined(__GNUC__)  // Clang in gcc mode.
 # define V8_CC_GNU 1
+# if defined(__MINGW32__)
+#  define V8_CC_MINGW32 1
+# endif
+# if defined(__MINGW64__)
+#  define V8_CC_MINGW64 1
+# endif
+# define V8_CC_MINGW (V8_CC_MINGW32 || V8_CC_MINGW64)
 #endif
 
 # define V8_HAS_ATTRIBUTE_ALWAYS_INLINE (__has_attribute(always_inline))
@@ -774,7 +781,7 @@ V8 shared library set USING_V8_SHARED.
 #elif defined(_M_IX86) || defined(__i386__)
 #define V8_HOST_ARCH_IA32 1
 #define V8_HOST_ARCH_32_BIT 1
-#elif defined(__AARCH64EL__) || defined(_M_ARM64)
+#elif defined(__AARCH64EL__) || defined(_M_ARM64) || defined(__aarch64__)
 #define V8_HOST_ARCH_ARM64 1
 #define V8_HOST_ARCH_64_BIT 1
 #elif defined(__ARMEL__)
@@ -827,7 +834,7 @@ V8 shared library set USING_V8_SHARED.
 #define V8_TARGET_ARCH_X64 1
 #elif defined(_M_IX86) || defined(__i386__)
 #define V8_TARGET_ARCH_IA32 1
-#elif defined(__AARCH64EL__) || defined(_M_ARM64)
+#elif defined(__AARCH64EL__) || defined(_M_ARM64) || defined(__aarch64__)
 #define V8_TARGET_ARCH_ARM64 1
 #elif defined(__ARMEL__)
 #define V8_TARGET_ARCH_ARM 1
diff --git a/src/3rdparty/chromium/v8/src/base/bits.h b/src/3rdparty/chromium/v8/src/base/bits.h
index 78f2c8b..eba170b 100644
--- a/src/3rdparty/chromium/v8/src/base/bits.h
+++ b/src/3rdparty/chromium/v8/src/base/bits.h
@@ -17,6 +17,13 @@
 #include "src/base/win32-headers.h"
 #endif
 
+#if V8_CC_MINGW
+#undef RotateLeft32
+#undef RotateLeft64
+#undef RotateRight32
+#undef RotateRight64
+#endif
+
 namespace v8 {
 namespace base {
 namespace bits {
diff --git a/src/3rdparty/chromium/v8/src/base/debug/stack_trace_win.cc b/src/3rdparty/chromium/v8/src/base/debug/stack_trace_win.cc
index f981bec..16655d5 100644
--- a/src/3rdparty/chromium/v8/src/base/debug/stack_trace_win.cc
+++ b/src/3rdparty/chromium/v8/src/base/debug/stack_trace_win.cc
@@ -81,7 +81,7 @@ bool InitializeSymbols() {
   }
 
   wchar_t exe_path[MAX_PATH];
-  GetModuleFileName(nullptr, exe_path, MAX_PATH);
+  GetModuleFileNameW(nullptr, exe_path, MAX_PATH);
   std::wstring exe_path_wstring(exe_path);
   // To get the path without the filename, we just need to remove the final
   // slash and everything after it.
diff --git a/src/3rdparty/chromium/v8/src/base/export-template.h b/src/3rdparty/chromium/v8/src/base/export-template.h
index 861cfe4..8107672 100644
--- a/src/3rdparty/chromium/v8/src/base/export-template.h
+++ b/src/3rdparty/chromium/v8/src/base/export-template.h
@@ -153,8 +153,10 @@
 
 EXPORT_TEMPLATE_TEST(DEFAULT, );
 EXPORT_TEMPLATE_TEST(DEFAULT, __attribute__((visibility("default"))));
+#ifdef _MSC_VER
 EXPORT_TEMPLATE_TEST(MSVC_HACK, __declspec(dllexport));
 EXPORT_TEMPLATE_TEST(DEFAULT, __declspec(dllimport));
+#endif
 
 #undef EXPORT_TEMPLATE_TEST
 #undef EXPORT_TEMPLATE_TEST_DEFAULT_DEFAULT
diff --git a/src/3rdparty/chromium/v8/src/base/macros.h b/src/3rdparty/chromium/v8/src/base/macros.h
index 93a99ec..38a4e7d 100644
--- a/src/3rdparty/chromium/v8/src/base/macros.h
+++ b/src/3rdparty/chromium/v8/src/base/macros.h
@@ -152,7 +152,7 @@ V8_INLINE Dest bit_cast(Source const& source) {
 // DISABLE_CFI_ICALL -- Disable Control Flow Integrity indirect call checks,
 // useful because calls into JITed code can not be CFI verified. Same for
 // UBSan's function pointer type checks.
-#ifdef V8_OS_WIN
+#ifdef V8_CC_MSVC
 // On Windows, also needs __declspec(guard(nocf)) for CFG.
 #define DISABLE_CFI_ICALL           \
   V8_CLANG_NO_SANITIZE("cfi-icall") \
@@ -242,8 +242,12 @@ V8_INLINE A implicit_cast(A x) {
 # else
 #  define V8_PTR_PREFIX   ""
 # endif  // V8_HOST_ARCH_64_BIT
-#elif V8_CC_MINGW64
-# define V8_PTR_PREFIX    "I64"
+#elif V8_CC_MINGW
+# if V8_TARGET_ARCH_X64
+#  define V8_PTR_PREFIX   "I64"
+# else
+#  define V8_PTR_PREFIX   ""
+# endif
 #elif V8_HOST_ARCH_64_BIT
 # define V8_PTR_PREFIX    "l"
 #else
diff --git a/src/3rdparty/chromium/v8/src/base/platform/platform-win32.cc b/src/3rdparty/chromium/v8/src/base/platform/platform-win32.cc
index a5558c7..459ac60 100644
--- a/src/3rdparty/chromium/v8/src/base/platform/platform-win32.cc
+++ b/src/3rdparty/chromium/v8/src/base/platform/platform-win32.cc
@@ -78,7 +78,6 @@ inline void MemoryFence() {
   __asm__ __volatile__("xchgl %%eax,%0 ":"=r" (barrier));
 }
 
-#endif  // __MINGW64_VERSION_MAJOR
 
 
 int localtime_s(tm* out_tm, const time_t* time) {
@@ -131,6 +130,7 @@ int strncpy_s(char* dest, size_t dest_size, const char* source, size_t count) {
   return 0;
 }
 
+#endif  // __MINGW64_VERSION_MAJOR
 #endif  // __MINGW32__
 
 namespace v8 {
@@ -771,13 +771,13 @@ void OS::EnsureWin32MemoryAPILoaded() {
   static bool loaded = false;
   if (!loaded) {
     VirtualAlloc2 = (VirtualAlloc2_t)GetProcAddress(
-        GetModuleHandle(L"kernelbase.dll"), "VirtualAlloc2");
+        GetModuleHandleW(L"kernelbase.dll"), "VirtualAlloc2");
 
     MapViewOfFile3 = (MapViewOfFile3_t)GetProcAddress(
-        GetModuleHandle(L"kernelbase.dll"), "MapViewOfFile3");
+        GetModuleHandleW(L"kernelbase.dll"), "MapViewOfFile3");
 
     UnmapViewOfFile2 = (UnmapViewOfFile2_t)GetProcAddress(
-        GetModuleHandle(L"kernelbase.dll"), "UnmapViewOfFile2");
+        GetModuleHandleW(L"kernelbase.dll"), "UnmapViewOfFile2");
 
     loaded = true;
   }
@@ -1054,7 +1054,7 @@ bool OS::DiscardSystemPages(void* address, size_t size) {
       reinterpret_cast<DiscardVirtualMemoryFunction>(-1))
     discard_virtual_memory =
         reinterpret_cast<DiscardVirtualMemoryFunction>(GetProcAddress(
-            GetModuleHandle(L"Kernel32.dll"), "DiscardVirtualMemory"));
+            GetModuleHandleW(L"Kernel32.dll"), "DiscardVirtualMemory"));
   // Use DiscardVirtualMemory when available because it releases faster than
   // MEM_RESET.
   DiscardVirtualMemoryFunction discard_function = discard_virtual_memory.load();
diff --git a/src/3rdparty/chromium/v8/src/base/platform/time.cc b/src/3rdparty/chromium/v8/src/base/platform/time.cc
index b6da0a6..e393197 100644
--- a/src/3rdparty/chromium/v8/src/base/platform/time.cc
+++ b/src/3rdparty/chromium/v8/src/base/platform/time.cc
@@ -816,13 +816,13 @@ ThreadTicks ThreadTicks::Now() {
   return ThreadTicks(ComputeThreadTicks());
 #elif V8_OS_FUCHSIA
   return ThreadTicks(GetFuchsiaThreadTicks());
+#elif V8_OS_WIN
+  return ThreadTicks::GetForThread(::GetCurrentThread());
 #elif(defined(_POSIX_THREAD_CPUTIME) && (_POSIX_THREAD_CPUTIME >= 0)) || \
   defined(V8_OS_ANDROID)
   return ThreadTicks(ClockNow(CLOCK_THREAD_CPUTIME_ID));
 #elif V8_OS_SOLARIS
   return ThreadTicks(gethrvtime() / Time::kNanosecondsPerMicrosecond);
-#elif V8_OS_WIN
-  return ThreadTicks::GetForThread(::GetCurrentThread());
 #else
   UNREACHABLE();
 #endif
@@ -876,6 +876,12 @@ void ThreadTicks::WaitUntilInitializedWin() {
 #endif
 }
 
+#ifdef __MINGW64_VERSION_MAJOR
+extern "C" {
+  extern uint64_t __rdtsc(void);
+}
+#endif
+
 #ifndef V8_HOST_ARCH_ARM64
 double ThreadTicks::TSCTicksPerSecond() {
   DCHECK(IsSupported());
diff --git a/src/3rdparty/chromium/v8/src/common/assert-scope.h b/src/3rdparty/chromium/v8/src/common/assert-scope.h
index 42d9649..064c33d 100644
--- a/src/3rdparty/chromium/v8/src/common/assert-scope.h
+++ b/src/3rdparty/chromium/v8/src/common/assert-scope.h
@@ -147,7 +147,7 @@ class CombinationAssertScope;
 template <typename Scope>
 class V8_NODISCARD CombinationAssertScope<Scope> : public Scope {
  public:
-  V8_EXPORT_PRIVATE static bool IsAllowed() {
+  static bool IsAllowed() {
     // Define IsAllowed() explicitly rather than with using Scope::IsAllowed, to
     // allow SFINAE removal of IsAllowed() when it's not defined (under debug).
     return Scope::IsAllowed();
@@ -164,12 +164,12 @@ class CombinationAssertScope<Scope, Scopes...>
 
  public:
   // Constructor for per-thread scopes.
-  V8_EXPORT_PRIVATE CombinationAssertScope() : Scope(), NextScopes() {}
+  CombinationAssertScope() : Scope(), NextScopes() {}
   // Constructor for per-isolate scopes.
-  V8_EXPORT_PRIVATE explicit CombinationAssertScope(Isolate* isolate)
+  explicit CombinationAssertScope(Isolate* isolate)
       : Scope(isolate), NextScopes(isolate) {}
 
-  V8_EXPORT_PRIVATE static bool IsAllowed() {
+  static bool IsAllowed() {
     return Scope::IsAllowed() && NextScopes::IsAllowed();
   }
 
diff --git a/src/3rdparty/chromium/v8/src/diagnostics/unwinding-info-win64.cc b/src/3rdparty/chromium/v8/src/diagnostics/unwinding-info-win64.cc
index a71b866..5c8c601 100644
--- a/src/3rdparty/chromium/v8/src/diagnostics/unwinding-info-win64.cc
+++ b/src/3rdparty/chromium/v8/src/diagnostics/unwinding-info-win64.cc
@@ -471,7 +471,7 @@ static decltype(
 void LoadNtdllUnwindingFunctionsOnce() {
   // Load functions from the ntdll.dll module.
   HMODULE ntdll_module =
-      LoadLibraryEx(L"ntdll.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
+      LoadLibraryExW(L"ntdll.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
   DCHECK_NOT_NULL(ntdll_module);
 
   // This fails on Windows 7.
diff --git a/src/3rdparty/chromium/v8/src/profiler/heap-snapshot-generator.cc b/src/3rdparty/chromium/v8/src/profiler/heap-snapshot-generator.cc
index 5a6e0c5..3b2adea 100644
--- a/src/3rdparty/chromium/v8/src/profiler/heap-snapshot-generator.cc
+++ b/src/3rdparty/chromium/v8/src/profiler/heap-snapshot-generator.cc
@@ -423,11 +423,11 @@ HeapSnapshot::HeapSnapshot(HeapProfiler* profiler,
   static_assert(kSystemPointerSize != 4 || sizeof(HeapGraphEdge) == 12);
   static_assert(kSystemPointerSize != 8 || sizeof(HeapGraphEdge) == 24);
   static_assert(kSystemPointerSize != 4 || sizeof(HeapEntry) == 32);
-#if V8_CC_MSVC
+#if V8_OS_WIN
   static_assert(kSystemPointerSize != 8 || sizeof(HeapEntry) == 48);
-#else   // !V8_CC_MSVC
+#else   // !V8_OS_WIN
   static_assert(kSystemPointerSize != 8 || sizeof(HeapEntry) == 40);
-#endif  // !V8_CC_MSVC
+#endif  // !V8_OS_WIN
   memset(&gc_subroot_entries_, 0, sizeof(gc_subroot_entries_));
 }
 
diff --git a/src/3rdparty/chromium/v8/src/snapshot/snapshot-utils.cc b/src/3rdparty/chromium/v8/src/snapshot/snapshot-utils.cc
index ea04fce..e67eb47 100644
--- a/src/3rdparty/chromium/v8/src/snapshot/snapshot-utils.cc
+++ b/src/3rdparty/chromium/v8/src/snapshot/snapshot-utils.cc
@@ -7,7 +7,7 @@
 #include "src/base/sanitizer/msan.h"
 
 #ifdef V8_USE_ZLIB
-#include "third_party/zlib/zlib.h"
+#include <zlib.h>
 #endif
 
 namespace v8 {
