diff --git a/src/3rdparty/chromium/third_party/breakpad/BUILD.gn b/src/3rdparty/chromium/third_party/breakpad/BUILD.gn
index abec806..cf4067d 100644
--- a/src/3rdparty/chromium/third_party/breakpad/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/breakpad/BUILD.gn
@@ -842,18 +842,25 @@ if (is_win) {
   # Adding the second condition enables this block for the "true" host
   # toolchain as well, which allows creating 64-bit binaries even when the host
   # toolchain is redefined to a 32-bit target.
-  if (is_clang) {
-    symupload_toolchain = "//build/toolchain/win:win_clang_$host_cpu"
+  if (is_msvc) {
+    if (is_clang) {
+      symupload_toolchain = "//build/toolchain/win:win_clang_$host_cpu"
+    } else {
+      symupload_toolchain = "//build/toolchain/win:$host_cpu"
+    }
   } else {
-    symupload_toolchain = "//build/toolchain/win:$host_cpu"
+      symupload_toolchain = "//build/toolchain/win:mingw_$host_cpu"
   }
   if (current_toolchain == host_toolchain ||
       current_toolchain == symupload_toolchain) {
     static_library("common_windows") {
       include_dirs = [
-        "$visual_studio_path/DIA SDK/include",
         "breakpad/src",
       ]
+      if (is_msvc) {
+        include_dirs += [ "$visual_studio_path/DIA SDK/include", ]
+      }
+
       sources = [
         "breakpad/src/common/windows/dia_util.cc",
         "breakpad/src/common/windows/dia_util.h",
@@ -882,9 +889,11 @@ if (is_win) {
 
     static_library("ms_symbol_server_converter") {
       include_dirs = [
-        "$visual_studio_path/DIA SDK/include",
         "breakpad/src",
       ]
+      if (is_msvc) {
+        include_dirs += [ "$visual_studio_path/DIA SDK/include", ]
+      }
 
       sources = [
         "breakpad/src/tools/windows/converter/ms_symbol_server_converter.cc",
@@ -892,14 +901,13 @@ if (is_win) {
 
       deps = [ "//third_party/breakpad:common_windows" ]
 
-      libs = [ "Pathcch.lib" ]
+      libs = [ "pathcch" ]
     }
 
     executable("dump_syms") {
       # TODO(scottmg) using this with VS2015 may break the crash server.
       # https://crbug.com/696671
       include_dirs = [
-        "$visual_studio_path/DIA SDK/include",
         "breakpad/src",
       ]
 
@@ -908,15 +916,18 @@ if (is_win) {
       deps = [ "//third_party/breakpad:common_windows" ]
 
       lib_dirs = []
-      if (current_cpu == "x64") {
-        lib_dirs += [ "$visual_studio_path/DIA SDK/lib/amd64" ]
-      } else {
-        lib_dirs += [ "$visual_studio_path/DIA SDK/lib" ]
+      if (is_msvc) {
+        include_dirs += [ "$visual_studio_path/DIA SDK/include", ]
+        if (current_cpu == "x64") {
+          lib_dirs += [ "$visual_studio_path/DIA SDK/lib/amd64" ]
+        } else {
+          lib_dirs += [ "$visual_studio_path/DIA SDK/lib" ]
+        }
       }
 
       libs = [
-        "diaguids.lib",
-        "imagehlp.lib",
+        "diaguids",
+        "imagehlp",
       ]
 
       if (is_clang) {
@@ -934,7 +945,6 @@ if (is_win) {
     }
     executable("symupload_win") {
       include_dirs = [
-        "$visual_studio_path/DIA SDK/include",
         "breakpad/src",
       ]
       output_name = "symupload"
@@ -959,18 +969,21 @@ if (is_win) {
 
       deps = [ "//third_party/breakpad:common_windows" ]
 
-      ldflags = [ "/LARGEADDRESSAWARE" ]
-
       lib_dirs = []
-      if (current_cpu == "x64") {
-        lib_dirs += [ "$visual_studio_path/DIA SDK/lib/amd64" ]
-      } else {
-        lib_dirs += [ "$visual_studio_path/DIA SDK/lib" ]
+      if (is_msvc) {
+        include_dirs += [ "$visual_studio_path/DIA SDK/include", ]
+        if (current_cpu == "x64") {
+          lib_dirs += [ "$visual_studio_path/DIA SDK/lib/amd64" ]
+        } else {
+          lib_dirs += [ "$visual_studio_path/DIA SDK/lib" ]
+        }
+
+        ldflags = [ "/LARGEADDRESSAWARE" ]
       }
 
       libs = [
-        "diaguids.lib",
-        "imagehlp.lib",
+        "diaguids",
+        "imagehlp",
       ]
 
       if (is_clang) {
@@ -982,7 +995,6 @@ if (is_win) {
 
     executable("google_converter") {
       include_dirs = [
-        "$visual_studio_path/DIA SDK/include",
         "breakpad/src",
       ]
 
@@ -1001,22 +1013,25 @@ if (is_win) {
         "//third_party/breakpad:ms_symbol_server_converter",
       ]
 
-      ldflags = [
-        "/DELAYLOAD:dbghelp.dll",
-        "/DELAYLOAD:dwmapi.dll",
-        "/DELAYLOAD:uxtheme.dll",
-      ]
-
       lib_dirs = []
-      if (current_cpu == "x64") {
-        lib_dirs += [ "$visual_studio_path/DIA SDK/lib/amd64" ]
-      } else {
-        lib_dirs += [ "$visual_studio_path/DIA SDK/lib" ]
+      if (is_msvc) {
+        include_dirs += [ "$visual_studio_path/DIA SDK/include", ]
+        if (current_cpu == "x64") {
+          lib_dirs += [ "$visual_studio_path/DIA SDK/lib/amd64" ]
+        } else {
+          lib_dirs += [ "$visual_studio_path/DIA SDK/lib" ]
+        }
+
+        ldflags = [
+          "/DELAYLOAD:dbghelp.dll",
+          "/DELAYLOAD:dwmapi.dll",
+          "/DELAYLOAD:uxtheme.dll",
+        ]
       }
 
       libs = [
-        "diaguids.lib",
-        "imagehlp.lib",
+        "diaguids",
+        "imagehlp",
       ]
 
       if (is_clang) {
