diff --git a/src/3rdparty/chromium/third_party/dawn/scripts/dawn_features.gni b/src/3rdparty/chromium/third_party/dawn/scripts/dawn_features.gni
index ba91e46..3051407 100644
--- a/src/3rdparty/chromium/third_party/dawn/scripts/dawn_features.gni
+++ b/src/3rdparty/chromium/third_party/dawn/scripts/dawn_features.gni
@@ -74,7 +74,7 @@ declare_args() {
   dawn_enable_d3d11 = is_win
 
   # Enables the compilation of Dawn's D3D12 backend
-  dawn_enable_d3d12 = is_win
+  dawn_enable_d3d12 = is_msvc
 
   # Enables the compilation of Dawn's Metal backend
   dawn_enable_metal = is_mac || is_ios
diff --git a/src/3rdparty/chromium/third_party/dawn/src/dawn/common/BUILD.gn b/src/3rdparty/chromium/third_party/dawn/src/dawn/common/BUILD.gn
index 48f594b..eb02692 100644
--- a/src/3rdparty/chromium/third_party/dawn/src/dawn/common/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/dawn/src/dawn/common/BUILD.gn
@@ -160,7 +160,7 @@ config("internal_config") {
       "-Wunused-macros",
     ]
 
-    if (is_win) {
+    if (is_msvc) {
       cflags += [
         # clang-cl doesn't know -pedantic, pass it explicitly to the clang driver
         "/clang:-pedantic",
@@ -173,7 +173,7 @@ config("internal_config") {
     }
   }
 
-  if (!is_clang && is_win) {
+  if (!is_clang && is_msvc) {
     # Dawn extends wgpu enums with internal enums.
     # MSVC considers these invalid switch values. crbug.com/dawn/397.
     cflags += [ "/wd4063" ]
