diff -ur a/src/3rdparty/chromium/third_party/pdfium/BUILD.gn b/src/3rdparty/chromium/third_party/pdfium/BUILD.gn
--- a/src/3rdparty/chromium/third_party/pdfium/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/pdfium/BUILD.gn	2023-12-14 21:26:38.466791500 +0800
@@ -210,6 +210,13 @@
         }
       }
     }
+  } else if (is_mingw) {
+    # Suppress myriad of incomplete enumeration warnings
+    cflags += [ "-Wno-switch" ]
+
+    if (current_cpu == "x86" && !is_clang) {
+      cflags += [ "-msse2" ]
+    }
   }
 
   if (is_ios) {
@@ -237,7 +244,7 @@
     }
   }
 
-  if (!is_win && !is_clang) {
+  if (!is_msvc && !is_clang) {
     cflags += [
       # Override -Wno-narrowing for GCC.
       "-Wnarrowing",
@@ -381,9 +388,9 @@
 
   if (is_win && !is_mingw) {
     libs += [
-      "advapi32.lib",
-      "gdi32.lib",
-      "user32.lib",
+      "advapi32",
+      "gdi32",
+      "user32",
     ]
   }
   if (is_mingw) {
diff -ur a/src/3rdparty/chromium/third_party/pdfium/skia/BUILD.gn b/src/3rdparty/chromium/third_party/pdfium/skia/BUILD.gn
--- a/src/3rdparty/chromium/third_party/pdfium/skia/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/pdfium/skia/BUILD.gn	2023-12-14 21:26:38.529204600 +0800
@@ -124,7 +124,9 @@
       # fairly small size. http://crbug.com/314387
       "SK_DEFAULT_FONT_CACHE_COUNT_LIMIT=256",
     ]
+  }
 
+  if (is_msvc) {
     cflags = [
       "/wd4244",  # conversion from 'type1( __int64)' to 'type2 (unsigned int)'
       "/wd4341",  # signed value is out of range for enum constant.
@@ -143,7 +145,7 @@
 
 source_set("skcms") {
   cflags = []
-  if (!is_win || is_clang) {
+  if (!is_msvc || is_clang) {
     cflags += [
       "-w",
       "-std=c11",
@@ -378,7 +380,7 @@
     ]
     public_configs = [ ":skia_config" ]
 
-    if (is_win) {
+    if (is_msvc) {
       cflags_cc = [
         "/wd5041",  # out-of-line definition for constexpr static data member is
                     # not needed and is deprecated in C++17
@@ -407,7 +409,7 @@
 if (current_cpu == "x86" || current_cpu == "x64") {
   skia_source_set("skia_opts_sse3") {
     sources = skia_opts.ssse3_sources
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-mssse3" ]
     }
     if (is_win) {
@@ -417,7 +419,7 @@
   }
   skia_source_set("skia_opts_sse42") {
     sources = skia_opts.sse42_sources
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-msse4.2" ]
     }
     if (is_win) {
@@ -427,17 +429,17 @@
   }
   skia_source_set("skia_opts_avx") {
     sources = skia_opts.avx_sources
-    if (!is_win) {
+    if (!is_msvc) {
       cflags = [ "-mavx" ]
     }
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX" ]
     }
     visibility = [ ":skia_opts" ]
   }
   skia_source_set("skia_opts_hsw") {
     sources = skia_opts.hsw_sources
-    if (!is_win) {
+    if (!is_msvc) {
       cflags = [
         "-mavx2",
         "-mbmi",
@@ -446,17 +448,17 @@
         "-mfma",
       ]
     }
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX2" ]
     }
     visibility = [ ":skia_opts" ]
   }
   skia_source_set("skia_opts_skx") {
     sources = skia_opts.skx_sources
-    if (!is_win) {
+    if (!is_msvc) {
       cflags = [ "-march=skylake-avx512" ]
     }
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX512" ]
     }
     visibility = [ ":skia_opts" ]
diff -ur a/src/3rdparty/chromium/third_party/pdfium/third_party/BUILD.gn b/src/3rdparty/chromium/third_party/pdfium/third_party/BUILD.gn
--- a/src/3rdparty/chromium/third_party/pdfium/third_party/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/pdfium/third_party/BUILD.gn	2023-12-14 21:26:38.576117000 +0800
@@ -389,7 +389,7 @@
       # Various files convert MAX_INT to double and similar without casting.
       "-Wno-implicit-int-float-conversion",
     ]
-  } else if (is_win && target_cpu == "x86") {
+  } else if (is_msvc && target_cpu == "x86") {
     cflags = [
       # Signed/unsigned comparisons.
       "/wd4018",
@@ -469,6 +469,13 @@
       if (is_win) {
         # Need to undefine the macro since it is redefined in tif_jpeg.c.
         configs -= [ "//build/config/win:lean_and_mean" ]
+        if (is_mingw) {
+          cflags = [
+            "-Wno-integer-overflow",
+            "-Wno-overflow",
+            "-Wno-shift-count-overflow"
+          ]
+        }
       }
       deps = [
         ":zlib",
