diff -ur a/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn b/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn
--- a/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/BUILD.gn	2023-12-14 21:31:42.376819300 +0800
@@ -275,7 +275,7 @@
     "GOOGLE_PROTOBUF_NO_STATIC_INITIALIZER",
   ]
   cflags = []
-  if (!is_clang || !is_win) {
+  if (!is_clang || !is_msvc) {
     cflags += [ "-Wno-deprecated-declarations" ]
   }
   if (is_clang && is_win) {
diff -ur a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn
--- a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/BUILD.gn	2023-12-14 21:29:42.048691200 +0800
@@ -40,7 +40,7 @@
 # is_clang -> could be clang-on-linux, clang-on-mac or clang-cl.exe.
 
 config("extra_warnings") {
-  if (is_win) {
+  if (is_msvc) {
     cflags = [
       "/W2",
       "/wd4244",  # conversion from 'float' to 'int', possible loss of data
@@ -86,7 +86,7 @@
       "-Wno-unknown-warning-option",
       "-Wno-unsafe-buffer-usage",
     ]
-  } else if (!is_clang && !is_win) {
+  } else if (!is_clang && !is_msvc) {
     # Use return std::move(...) for compatibility with old GCC compilers.
     cflags_cc = [ "-Wno-redundant-move" ]
 
@@ -104,7 +104,7 @@
 }
 
 config("no_rtti") {
-  if (is_win) {
+  if (is_msvc) {
     cflags_cc = [ "/GR-" ]
   } else {
     cflags_cc = [ "-fno-rtti" ]
@@ -113,7 +113,7 @@
 
 # Used in buildtools dependencies for standalone builds.
 config("c++17") {
-  if (is_win) {
+  if (is_msvc) {
     cflags_cc = [ "/std:c++17" ]
   } else {
     cflags_cc = [ "-std=c++17" ]
@@ -131,7 +131,7 @@
 }
 
 config("visibility_hidden") {
-  if (!is_win) {
+  if (!is_msvc) {
     cflags = [ "-fvisibility=hidden" ]
   }
 }
@@ -156,7 +156,7 @@
     ldflags += [ "-Wl,--build-id" ]
   }
 
-  if (is_clang || !is_win) {  # Clang or GCC, but not MSVC.
+  if (is_clang || !is_msvc) {  # Clang or GCC, but not MSVC.
     cflags += [
       "-fstrict-aliasing",
       "-Wformat",
@@ -170,7 +170,7 @@
     cflags += [ "-Wno-deprecated" ]
   }
 
-  if (is_win) {
+  if (is_msvc) {
     cflags += [
       "/bigobj",  # Some of our files are bigger than the regular limits.
       "/Gy",  # Enable function-level linking.
@@ -198,7 +198,7 @@
 
   # Treat warnings as errors, but give up on fuzzer builds.
   if (!is_fuzzer) {
-    if (is_win) {
+    if (is_msvc) {
       cflags += [ "/WX" ]
     } else {
       cflags += [ "-Werror" ]
@@ -207,7 +207,7 @@
 
   if (is_clang) {
     cflags += [ "-fcolor-diagnostics" ]
-    if (!is_win) {
+    if (!is_msvc) {
       cflags += [ "-fdiagnostics-show-template-tree" ]
     }
   }
@@ -280,7 +280,7 @@
     ]
   }
 
-  if (is_win && !is_clang) {
+  if (is_msvc && !is_clang) {
     # When using MSVC we need to manually pass the include dirs. clang-cl.exe
     # doesn't need them because it's smart enough to figure out the right path
     # by querying the registry on its own.
@@ -288,13 +288,15 @@
   }
 
   if (is_debug) {
-    if (is_win) {
+    if (is_msvc) {
       cflags += [ "/Z7" ]
       if (is_clang) {
         # Required to see symbols in windbg when building with clang-cl.exe.
         cflags += [ "-gcodeview-ghash" ]
         ldflags = [ "/DEBUG:GHASH" ]
       }
+    } else if (is_mingw) {
+      cflags += [ "-g" ]
     } else {
       libs += [ "dl" ]
     }
@@ -364,7 +366,7 @@
 
 config("debug_symbols") {
   cflags = []
-  if (is_win) {
+  if (is_msvc) {
     cflags = [ "/Od" ]
   } else {
     cflags = [ "-O0" ]
@@ -376,7 +378,7 @@
 
 config("release") {
   # Compiler flags for release builds.
-  if (is_win) {
+  if (is_msvc) {
     cflags = [
       "/O2",
       "/Zc:inline",
@@ -388,7 +390,7 @@
   } else {
     cflags = [ "-O3" ]
   }
-  if (!is_win) {
+  if (!is_msvc) {
     cflags += [
       "-fdata-sections",
       "-ffunction-sections",
@@ -396,7 +398,7 @@
   }
 
   # Linker flags for release builds.
-  if (is_win) {
+  if (is_msvc) {
     ldflags = [
       "/OPT:REF",
       "/OPT:ICF",
@@ -405,7 +407,7 @@
     ]
   } else if (is_mac) {
     ldflags = [ "-dead_strip" ]
-  } else if (!is_win && !is_wasm) {
+  } else if (!is_msvc && !is_wasm) {
     ldflags = [
       "-Wl,--gc-sections",
       "-Wl,--icf=all",
diff -ur a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/llvm.gni b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/llvm.gni
--- a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/llvm.gni	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/llvm.gni	2023-12-14 21:29:42.079952200 +0800
@@ -45,7 +45,7 @@
 
   # _mac_toolchain_dirs[0] contains the mac toolchain dir.
   mac_clangrt_dir = _mac_toolchain_dirs[1]
-} else if (is_win_host) {
+} else if (is_msvc) {
   if (is_hermetic_clang) {
     # Use the toolchain pulled by //tools/install-build-deps. This tracks
     # chromium's llvm dist.
diff -ur a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/msvc.gni b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/msvc.gni
--- a/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/msvc.gni	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/perfetto/gn/standalone/toolchain/msvc.gni	2023-12-14 21:29:42.079952200 +0800
@@ -15,7 +15,7 @@
 # We should never get in here by accident from the Chromium tree.
 assert(!defined(build_with_chromium) || !build_with_chromium)
 
-if (is_win_host) {
+if (is_msvc) {
   _find_msvc_out = exec_script("win_find_msvc.py", [], "list lines")
 
   # The output looks like this (without the line number "N:" part):
diff -ur a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/sys_types.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/sys_types.h
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/sys_types.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/sys_types.h	2023-12-14 21:29:42.095563900 +0800
@@ -26,10 +26,12 @@
 
 #if PERFETTO_BUILDFLAG(PERFETTO_OS_WIN)
 
-#if !PERFETTO_BUILDFLAG(PERFETTO_COMPILER_GCC)
+#if defined(_MSC_VER)
 // MinGW has these. clang-cl and MSVC, which use just the Windows SDK, don't.
 using uid_t = unsigned int;
 using pid_t = int;
+#else
+using uid_t = unsigned int;
 #endif  // !GCC
 
 #if defined(_WIN64)
