diff -ur a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn
--- a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn	2023-12-14 21:42:21.358863700 +0800
@@ -39,7 +39,7 @@
   defines = [
     "VK_ENABLE_BETA_EXTENSIONS",
   ]
-  if (is_clang || !is_win) {
+  if (is_clang || !is_msvc) {
     cflags = [
       "-Wno-conversion",
       "-Wno-extra-semi",
@@ -62,6 +62,9 @@
       "FALLBACK_DATA_DIRS=\"/usr/local/share:/usr/share\"",
     ]
   }
+  if (!vulkan_loader_shared) {
+    defines += [ "BUILD_STATIC_LOADER" ]
+  }
 }
 
 # Vulkan loader
@@ -77,7 +80,7 @@
     "USE_UNSAFE_FILE_SEARCH=1",
   ]
 
-  if (is_win) {
+  if (is_msvc) {
     cflags = [ "/wd4201" ]
   }
   if (is_linux || is_chromeos) {
@@ -141,7 +144,7 @@
     if (custom_vulkan_loader_library_name != "") {
       output_name = custom_vulkan_loader_library_name
     } else {
-      if (is_win) {
+      if (is_msvc) {
         output_name = "vulkan-1"
       } else {
         output_name = "vulkan"
@@ -162,7 +165,13 @@
         "loader/loader.rc",
         "loader/vulkan-1.def",
       ]
-      if (!is_clang) {
+      if (!vulkan_loader_shared) {
+        sources -= [
+          "loader/loader.rc",
+          "loader/vulkan-1.def",
+        ]
+      }
+      if (is_msvc && !is_clang) {
         cflags = [
           "/wd4054",  # Type cast from function pointer
           "/wd4055",  # Type cast from data pointer
@@ -181,7 +190,7 @@
       if (is_clang) {
         cflags = [ "-Wno-incompatible-pointer-types" ]
       }
-      libs = [ "Cfgmgr32.lib" ]
+      libs = [ "cfgmgr32" ]
     }
     if (is_linux || is_chromeos) {
       sources += [
diff -ur a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c
--- a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c	2023-12-14 21:42:21.421412000 +0800
@@ -91,6 +91,7 @@
 #endif
 }
 
+#ifndef BUILD_STATIC_LOADER
 BOOL WINAPI DllMain(HINSTANCE hinst, DWORD reason, LPVOID reserved) {
     switch (reason) {
         case DLL_PROCESS_ATTACH:
@@ -107,6 +108,7 @@
     }
     return TRUE;
 }
+#endif
 
 bool windows_add_json_entry(const struct loader_instance *inst,
                             char **reg_data,    // list of JSON files
