diff --git a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn
index 7bbd393..6d2cfef 100644
--- a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/BUILD.gn
@@ -39,7 +39,7 @@ config("vulkan_internal_config") {
   defines = [
     "VK_ENABLE_BETA_EXTENSIONS",
   ]
-  if (is_clang || !is_win) {
+  if (is_clang || !is_msvc) {
     cflags = [
       "-Wno-conversion",
       "-Wno-extra-semi",
@@ -80,7 +80,7 @@ config("vulkan_loader_config") {
     "LOADER_USE_UNSAFE_FILE_SEARCH=1",
   ]
 
-  if (is_win) {
+  if (is_msvc) {
     cflags = [ "/wd4201" ]
   }
   if (is_linux || is_chromeos) {
@@ -90,6 +90,9 @@ config("vulkan_loader_config") {
       "LOADER_ENABLE_LINUX_SORT",
     ]
   }
+  if (!vulkan_loader_shared) {
+    defines += [ "BUILD_STATIC_LOADER" ]
+  }
 }
 
 if (!is_android) {
@@ -146,7 +149,7 @@ if (!is_android) {
     if (custom_vulkan_loader_library_name != "") {
       output_name = custom_vulkan_loader_library_name
     } else {
-      if (is_win) {
+      if (is_msvc) {
         output_name = "vulkan-1"
       } else {
         output_name = "vulkan"
@@ -167,7 +170,13 @@ if (!is_android) {
         "loader/loader.rc",
         "loader/vulkan-1.def",
       ]
-      if (!is_clang) {
+      if (!vulkan_loader_shared) {
+        sources -= [
+          "loader/loader.rc",
+          "loader/vulkan-1.def",
+        ]
+      }
+      if (is_msvc && !is_clang) {
         cflags = [
           "/wd4054",  # Type cast from function pointer
           "/wd4055",  # Type cast from data pointer
@@ -186,7 +195,7 @@ if (!is_android) {
       if (is_clang) {
         cflags = [ "-Wno-incompatible-pointer-types" ]
       }
-      libs = [ "Cfgmgr32.lib" ]
+      libs = [ "cfgmgr32" ]
     }
     if (is_linux || is_chromeos) {
       sources += [
diff --git a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c
index 0f6c2c8..eeb281a 100644
--- a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c
+++ b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-loader/src/loader/loader_windows.c
@@ -95,6 +95,7 @@ void windows_initialization(void) {
 #endif
 }
 
+#ifndef BUILD_STATIC_LOADER
 BOOL WINAPI DllMain(HINSTANCE hinst, DWORD reason, LPVOID reserved) {
     (void)hinst;
     switch (reason) {
@@ -112,6 +113,7 @@ BOOL WINAPI DllMain(HINSTANCE hinst, DWORD reason, LPVOID reserved) {
     }
     return TRUE;
 }
+#endif
 
 bool windows_add_json_entry(const struct loader_instance *inst,
                             char **reg_data,    // list of JSON files
diff --git a/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn b/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn
index 97aab3a..fad0e7a 100644
--- a/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/vulkan_memory_allocator/BUILD.gn
@@ -10,7 +10,7 @@ declare_args() {
 
 config("vulkan_memory_allocator_config") {
   include_dirs = [ "include" ]
-  if (is_clang) {
+  if (is_clang || !is_msvc) {
     cflags_cc = [
       "-Wno-c++98-compat-extra-semi",
       "-Wno-deprecated-copy",
@@ -22,7 +22,7 @@ config("vulkan_memory_allocator_config") {
       "-Wno-unused-variable",
     ]
   }
-  if (is_win && !is_clang) {
+  if (is_msvc && !is_clang) {
     cflags_cc = [
       "/wd4189",  # local variable is initialized but not referenced
     ]
