diff -ur a/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/BUILD.gn b/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/BUILD.gn
--- a/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/BUILD.gn	2023-12-14 21:58:56.421204100 +0800
@@ -169,6 +169,9 @@
     ":spinlock_wait",
     "//third_party/abseil-cpp/absl/meta:type_traits",
   ]
+  if (is_mingw) {
+    libs = [ "pthread" ]
+  }
 }
 
 absl_source_set("throw_delegate") {
diff -ur a/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/internal/thread_identity.h b/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/internal/thread_identity.h
--- a/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/internal/thread_identity.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/abseil-cpp/absl/base/internal/thread_identity.h	2023-12-14 21:58:56.421204100 +0800
@@ -20,7 +20,7 @@
 #ifndef ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 #define ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 #include <pthread.h>
 // Defines __GOOGLE_GRTE_VERSION__ (via glibc-specific features.h) when
 // supported.
diff -ur a/src/3rdparty/chromium/third_party/boringssl/src/include/openssl/base.h b/src/3rdparty/chromium/third_party/boringssl/src/include/openssl/base.h
--- a/src/3rdparty/chromium/third_party/boringssl/src/include/openssl/base.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/boringssl/src/include/openssl/base.h	2023-12-14 21:58:56.483797300 +0800
@@ -63,6 +63,7 @@
 #if defined(__MINGW32__)
 // stdio.h is needed on MinGW for __MINGW_PRINTF_FORMAT.
 #include <stdio.h>
+#undef X509_NAME
 #endif
 
 #if defined(__APPLE__)
diff -ur a/src/3rdparty/chromium/third_party/brotli/BUILD.gn b/src/3rdparty/chromium/third_party/brotli/BUILD.gn
--- a/src/3rdparty/chromium/third_party/brotli/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/brotli/BUILD.gn	2023-12-14 21:58:56.530589100 +0800
@@ -197,7 +197,7 @@
       "//build/win:default_exe_manifest",
     ]
 
-    if (is_win && visual_studio_version == "2015") {
+    if (is_msvc && visual_studio_version == "2015") {
       # Disabling "result of 32-bit shift implicitly converted to 64 bits",
       # caused by code like: foo |= (1 << i);   // warning 4334
       cflags = [ "/wd4334" ]
diff -ur a/src/3rdparty/chromium/third_party/ced/BUILD.gn b/src/3rdparty/chromium/third_party/ced/BUILD.gn
--- a/src/3rdparty/chromium/third_party/ced/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/ced/BUILD.gn	2023-12-14 21:58:56.577531900 +0800
@@ -51,7 +51,7 @@
 
   public_configs = [ ":ced_config" ]
 
-  if (is_win) {
+  if (is_msvc) {
     defines = [ "COMPILER_MSVC" ]
     cflags = [
       "/wd4005",  # Macro defined twice.
@@ -76,7 +76,7 @@
 
   configs += [ ":ced_config_private" ]
 
-  if (is_win) {
+  if (is_msvc) {
     defines = [ "COMPILER_MSVC" ]
     cflags = [
       "/wd4310",  # Truncation of constant value.
diff -ur a/src/3rdparty/chromium/third_party/cpuinfo/BUILD.gn b/src/3rdparty/chromium/third_party/cpuinfo/BUILD.gn
--- a/src/3rdparty/chromium/third_party/cpuinfo/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/cpuinfo/BUILD.gn	2023-12-14 21:58:56.671291200 +0800
@@ -11,7 +11,7 @@
 }
 
 config("cpuinfo_config") {
-  if (is_clang || !is_win) {
+  if (is_clang || !is_msvc) {
   cflags = [
     # For chipset.c and utils.h.
     "-Wno-implicit-function-declaration",
diff -ur a/src/3rdparty/chromium/third_party/cpuinfo/src/src/x86/windows/init.c b/src/3rdparty/chromium/third_party/cpuinfo/src/src/x86/windows/init.c
--- a/src/3rdparty/chromium/third_party/cpuinfo/src/src/x86/windows/init.c	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/cpuinfo/src/src/x86/windows/init.c	2023-12-25 20:15:24.002337100 +0800
@@ -16,13 +16,22 @@
   #define CPUINFO_ALLOCA _alloca
 #endif
 
+#ifndef max
+#define max(a,b)             \
+({                           \
+    __typeof__ (a) _a = (a); \
+    __typeof__ (b) _b = (b); \
+    _a > _b ? _a : _b;       \
+})
+#endif
+
 
 static inline uint32_t bit_mask(uint32_t bits) {
 	return (UINT32_C(1) << bits) - UINT32_C(1);
 }
 
 static inline uint32_t low_index_from_kaffinity(KAFFINITY kaffinity) {
-	#if defined(_M_X64) || defined(_M_AMD64)
+	#if defined(_M_X64) || defined(_M_AMD64) || defined(__x86_64__)
 		unsigned long index;
 		_BitScanForward64(&index, (unsigned __int64) kaffinity);
 		return (uint32_t) index;
diff -ur a/src/3rdparty/chromium/third_party/crc32c/BUILD.gn b/src/3rdparty/chromium/third_party/crc32c/BUILD.gn
--- a/src/3rdparty/chromium/third_party/crc32c/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/crc32c/BUILD.gn	2023-12-14 21:58:56.718165800 +0800
@@ -34,7 +34,7 @@
       "HAVE_SSE42=0",
     ]
   }
-  if (is_clang || !is_win) {
+  if (is_clang || !is_msvc) {
     defines += [ "HAVE_BUILTIN_PREFETCH=1" ]
   } else {
     defines += [ "HAVE_BUILTIN_PREFETCH=0" ]
@@ -86,7 +86,7 @@
 
   configs += [ ":crc32c_config" ]
   if (current_cpu == "x86" || current_cpu == "x64") {
-    if (is_win && !is_clang) {
+    if (is_msvc && !is_clang) {
       cflags = [ "/arch:AVX" ]
     } else {
       cflags = [ "-msse4.2" ]
diff -ur a/src/3rdparty/chromium/third_party/dav1d/BUILD.gn b/src/3rdparty/chromium/third_party/dav1d/BUILD.gn
--- a/src/3rdparty/chromium/third_party/dav1d/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/dav1d/BUILD.gn	2023-12-14 21:58:56.765025600 +0800
@@ -100,7 +100,7 @@
     "libdav1d/include/dav1d",
     platform_config_root,
   ]
-  if (is_win && !is_clang) {
+  if (is_msvc && !is_clang) {
     include_dirs += [ "libdav1d/include/compat/msvc" ]
   }
 }
@@ -110,7 +110,7 @@
   "-D_POSIX_C_SOURCE=200112L",
 ]
 
-if (is_win) {
+if (is_msvc) {
   if (!is_clang) {
     dav1d_copts += [ "/wd4028" ]
   }
@@ -122,7 +122,7 @@
   if (is_mac || is_ios) {
     dav1d_copts += [ "-D_DARWIN_C_SOURCE" ]
   }
-  if (is_linux || is_chromeos || is_android || current_os == "aix") {
+  if (is_linux || is_chromeos || is_android || is_mingw || current_os == "aix") {
     if (!is_clang) {
       dav1d_copts += [ "-D_GNU_SOURCE" ]
     }
diff -ur a/src/3rdparty/chromium/third_party/dav1d/config/win/x64/config.h b/src/3rdparty/chromium/third_party/dav1d/config/win/x64/config.h
--- a/src/3rdparty/chromium/third_party/dav1d/config/win/x64/config.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/dav1d/config/win/x64/config.h	2023-12-14 21:58:56.780579300 +0800
@@ -47,6 +47,6 @@
 
 #define __USE_MINGW_ANSI_STDIO 1
 
-#define fseeko _fseeki64
+// #define fseeko _fseeki64
 
-#define ftello _ftelli64
+// #define ftello _ftelli64
diff -ur a/src/3rdparty/chromium/third_party/dav1d/config/win/x86/config.h b/src/3rdparty/chromium/third_party/dav1d/config/win/x86/config.h
--- a/src/3rdparty/chromium/third_party/dav1d/config/win/x86/config.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/dav1d/config/win/x86/config.h	2023-12-14 21:58:56.796205200 +0800
@@ -49,6 +49,6 @@
 
 #define __USE_MINGW_ANSI_STDIO 1
 
-#define fseeko _fseeki64
+// #define fseeko _fseeki64
 
-#define ftello _ftelli64
+// #define ftello _ftelli64
diff -ur a/src/3rdparty/chromium/third_party/flac/BUILD.gn b/src/3rdparty/chromium/third_party/flac/BUILD.gn
--- a/src/3rdparty/chromium/third_party/flac/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/flac/BUILD.gn	2023-12-17 01:02:49.889982300 +0800
@@ -137,10 +137,14 @@
 
     # win_utf8_io.c defines this itself.
     configs -= [ "//build/config/win:lean_and_mean" ]
-    cflags = [
-      "/wd4334",  # 32-bit shift converted to 64 bits.
-      "/wd4267",  # Converting from size_t to unsigned on 64-bit.
-    ]
+    if (is_mingw) {
+      defines += [ "HAVE_FSEEKO" ]
+    } else {
+      cflags = [
+        "/wd4334",  # 32-bit shift converted to 64 bits.
+        "/wd4267",  # Converting from size_t to unsigned on 64-bit.
+      ]
+    }
   } else {
     defines += [ "HAVE_INTTYPES_H" ]
   }
diff -ur a/src/3rdparty/chromium/third_party/flac/include/share/compat.h b/src/3rdparty/chromium/third_party/flac/include/share/compat.h
--- a/src/3rdparty/chromium/third_party/flac/include/share/compat.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/flac/include/share/compat.h	2023-12-17 01:10:05.374450500 +0800
@@ -168,6 +168,8 @@
  *************************************************************************************
  */
 #endif
+#else
+#include <inttypes.h>
 #endif /* defined _MSC_VER */
 
 #ifdef _WIN32
diff -ur a/src/3rdparty/chromium/third_party/hunspell/BUILD.gn b/src/3rdparty/chromium/third_party/hunspell/BUILD.gn
--- a/src/3rdparty/chromium/third_party/hunspell/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/hunspell/BUILD.gn	2023-12-14 21:58:56.858714900 +0800
@@ -74,7 +74,7 @@
 
   cflags = []
 
-  if (is_win) {
+  if (is_msvc) {
     cflags += [
       # TODO(jschuh): http://crbug.com/167187 size_t -> int
       "/wd4267",
diff -ur a/src/3rdparty/chromium/third_party/jsoncpp/BUILD.gn b/src/3rdparty/chromium/third_party/jsoncpp/BUILD.gn
--- a/src/3rdparty/chromium/third_party/jsoncpp/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/jsoncpp/BUILD.gn	2023-12-14 21:58:56.905590900 +0800
@@ -46,7 +46,7 @@
 
   include_dirs = [ "source/src/lib_json" ]
 
-  if (!is_win || is_clang) {
+  if (!is_msvc || is_clang) {
     cflags_cc = [ "-Wno-implicit-fallthrough" ]
   }
 }
diff -ur a/src/3rdparty/chromium/third_party/leveldatabase/BUILD.gn b/src/3rdparty/chromium/third_party/leveldatabase/BUILD.gn
--- a/src/3rdparty/chromium/third_party/leveldatabase/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/leveldatabase/BUILD.gn	2023-12-14 21:58:56.983720100 +0800
@@ -133,7 +133,7 @@
     "//third_party/snappy",
   ]
 
-  if (is_win) {
+  if (is_msvc) {
     # leveldb compiles with a limited set of warnings enabled, and is not
     # likely to accept fixes for other warnings. See crbug.com/698013
     cflags = [ "/wd4018" ]  # Signed/unsigned mismatch in comparison.
@@ -170,7 +170,7 @@
     "//third_party/snappy",
   ]
 
-  if (is_win) {
+  if (is_msvc) {
     # leveldb compiles with a limited set of warnings enabled, and is not
     # likely to accept fixes for other warnings. See crbug.com/698013
     cflags = [ "/wd4018" ]  # Signed/unsigned mismatch in comparison.
@@ -250,7 +250,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
 
-    if (is_win) {
+    if (is_msvc) {
       # util\cache_test.cc(167): warning C4018: '<': signed/unsigned mismatch
       # util\cache_test.cc(171): warning C4018: '<': signed/unsigned mismatch
       cflags = [ "/wd4018" ]  # Signed/unsigned mismatch in comparison.
diff -ur a/src/3rdparty/chromium/third_party/libaom/BUILD.gn b/src/3rdparty/chromium/third_party/libaom/BUILD.gn
--- a/src/3rdparty/chromium/third_party/libaom/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libaom/BUILD.gn	2023-12-23 18:45:23.974603300 +0800
@@ -100,7 +100,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-msse2" ]
     }
     sources = aom_av1_common_intrin_sse2
@@ -114,7 +114,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-msse3" ]
     }
     sources = aom_av1_encoder_intrin_sse3
@@ -125,7 +125,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-mssse3" ]
     }
     sources = aom_av1_common_intrin_ssse3
@@ -139,7 +139,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-msse4.1" ]
     }
     sources = aom_av1_common_intrin_sse4_1
@@ -153,7 +153,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (!is_win || is_clang) {
+    if (!is_msvc || is_clang) {
       cflags = [ "-msse4.2" ]
     }
     sources = aom_av1_encoder_intrin_sse4_2
@@ -164,7 +164,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX" ]
     } else {
       cflags = [ "-mavx" ]
@@ -177,7 +177,7 @@
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
-    if (is_win) {
+    if (is_msvc) {
       cflags = [ "/arch:AVX2" ]
     } else {
       cflags = [ "-mavx2" ]
@@ -241,6 +241,13 @@
     # overflows in the transform functions.
     defines += [ "DO_RANGE_CHECK_CLAMP=1" ]
   }
+  if (is_win) {
+    if (is_mingw) {
+      defines = [ "HAVE_PTHREAD_H=1" ]
+    } else {
+      defines = [ "HAVE_PTHREAD_H=0" ]
+    }
+  }
   deps = []
   if (current_cpu == "x86" || (current_cpu == "x64" && !is_msan)) {
     deps += [
@@ -283,6 +290,13 @@
   configs -= [ "//build/config/compiler:chromium_code" ]
   configs += [ "//build/config/compiler:no_chromium_code" ]
   configs += [ ":libaom_config" ]
+  if (is_win) {
+    if (is_mingw) {
+      defines = [ "HAVE_PTHREAD_H=1" ]
+    } else {
+      defines = [ "HAVE_PTHREAD_H=0" ]
+    }
+  }
 
   public_configs = [ ":libaom_external_config" ]
   public_deps = [ ":libaom" ]
diff -ur a/src/3rdparty/chromium/third_party/libaom/source/config/win/arm64/config/aom_config.h b/src/3rdparty/chromium/third_party/libaom/source/config/win/arm64/config/aom_config.h
--- a/src/3rdparty/chromium/third_party/libaom/source/config/win/arm64/config/aom_config.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libaom/source/config/win/arm64/config/aom_config.h	2023-12-23 00:52:35.482401900 +0800
@@ -75,7 +75,6 @@
 #define HAVE_FEXCEPT 1
 #define HAVE_MMX 0
 #define HAVE_NEON 1
-#define HAVE_PTHREAD_H 0
 #define HAVE_SSE 0
 #define HAVE_SSE2 0
 #define HAVE_SSE3 0
diff -ur a/src/3rdparty/chromium/third_party/libaom/source/config/win/ia32/config/aom_config.h b/src/3rdparty/chromium/third_party/libaom/source/config/win/ia32/config/aom_config.h
--- a/src/3rdparty/chromium/third_party/libaom/source/config/win/ia32/config/aom_config.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libaom/source/config/win/ia32/config/aom_config.h	2023-12-23 00:52:44.591772100 +0800
@@ -75,7 +75,6 @@
 #define HAVE_FEXCEPT 1
 #define HAVE_MMX 1
 #define HAVE_NEON 0
-#define HAVE_PTHREAD_H 0
 #define HAVE_SSE 1
 #define HAVE_SSE2 1
 #define HAVE_SSE3 1
diff -ur a/src/3rdparty/chromium/third_party/libaom/source/config/win/x64/config/aom_config.h b/src/3rdparty/chromium/third_party/libaom/source/config/win/x64/config/aom_config.h
--- a/src/3rdparty/chromium/third_party/libaom/source/config/win/x64/config/aom_config.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libaom/source/config/win/x64/config/aom_config.h	2023-12-23 00:52:53.779267600 +0800
@@ -75,7 +75,6 @@
 #define HAVE_FEXCEPT 1
 #define HAVE_MMX 1
 #define HAVE_NEON 0
-#define HAVE_PTHREAD_H 0
 #define HAVE_SSE 1
 #define HAVE_SSE2 1
 #define HAVE_SSE3 1
diff -ur a/src/3rdparty/chromium/third_party/libsrtp/BUILD.gn b/src/3rdparty/chromium/third_party/libsrtp/BUILD.gn
--- a/src/3rdparty/chromium/third_party/libsrtp/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libsrtp/BUILD.gn	2023-12-14 21:58:57.077533800 +0800
@@ -59,7 +59,7 @@
 
 config("libsrtp_warnings") {
   cflags = []
-  if (is_win) {
+  if (is_msvc) {
     cflags += [ "/wd4018" ]  # Signed/unsigned mismatch in comparison
   }
 }
diff -ur a/src/3rdparty/chromium/third_party/libvpx/source/libvpx/vpx_util/vpx_thread.h b/src/3rdparty/chromium/third_party/libvpx/source/libvpx/vpx_util/vpx_thread.h
--- a/src/3rdparty/chromium/third_party/libvpx/source/libvpx/vpx_util/vpx_thread.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libvpx/source/libvpx/vpx_util/vpx_thread.h	2023-12-14 21:58:57.155585800 +0800
@@ -27,7 +27,7 @@
 
 #if CONFIG_MULTITHREAD
 
-#if defined(_WIN32) && !HAVE_PTHREAD_H
+#if defined(_MSC_VER) && !HAVE_PTHREAD_H
 #include <errno.h>    // NOLINT
 #include <process.h>  // NOLINT
 #include <windows.h>  // NOLINT
diff -ur a/src/3rdparty/chromium/third_party/libxml/BUILD.gn b/src/3rdparty/chromium/third_party/libxml/BUILD.gn
--- a/src/3rdparty/chromium/third_party/libxml/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libxml/BUILD.gn	2023-12-14 21:58:57.233723400 +0800
@@ -31,7 +31,7 @@
 }
 
 config("libxml_warnings") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     cflags_c = [
       "/wd4018",  # Signed/unsigned mismatch in comparison.
       "/wd4267",  # TODO(brucedawson): http://crbug.com/554200 fix C4267
@@ -70,7 +70,7 @@
       # TODO(hans): See if we can fix upstream (http://crbug.com/763944).
       "-Wno-enum-compare",
     ]
-  } else if (is_linux || is_chromeos) {
+  } else if (is_linux || is_chromeos || is_mingw) {
     cflags = [
       # gcc spits out a bunch of warnings about passing too many arguments to
       # __xmlSimpleError.
diff -ur a/src/3rdparty/chromium/third_party/libxml/src/include/private/threads.h b/src/3rdparty/chromium/third_party/libxml/src/include/private/threads.h
--- a/src/3rdparty/chromium/third_party/libxml/src/include/private/threads.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libxml/src/include/private/threads.h	2023-12-14 21:58:57.280577400 +0800
@@ -8,7 +8,9 @@
     #include <pthread.h>
     #define HAVE_POSIX_THREADS
   #elif defined(_WIN32)
+    #ifndef WIN32_LEAN_AND_MEAN
     #define WIN32_LEAN_AND_MEAN
+    #endif
     #include <windows.h>
     #ifndef HAVE_COMPILER_TLS
       #include <process.h>
diff -ur a/src/3rdparty/chromium/third_party/libxslt/BUILD.gn b/src/3rdparty/chromium/third_party/libxslt/BUILD.gn
--- a/src/3rdparty/chromium/third_party/libxslt/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libxslt/BUILD.gn	2023-12-15 00:45:36.017082900 +0800
@@ -94,6 +94,9 @@
     include_dirs = [ "linux" ]
   } else if (is_win) {
     include_dirs = [ "win32" ]
+    if (is_mingw) {
+      cflags += [ "-Wno-incompatible-pointer-types" ]
+    } else {
     cflags += [
       "/wd4267",  # size_t to int.
 
@@ -101,6 +104,7 @@
       # 2015 64-bit warning for pointer truncation
       "/wd4311",
     ]
+    }
   } else if (is_apple) {
     include_dirs = [ "mac" ]
   }
diff -ur a/src/3rdparty/chromium/third_party/libyuv/BUILD.gn b/src/3rdparty/chromium/third_party/libyuv/BUILD.gn
--- a/src/3rdparty/chromium/third_party/libyuv/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/libyuv/BUILD.gn	2023-12-14 21:58:57.390041400 +0800
@@ -55,7 +55,7 @@
   all_dependent_configs = [ ":libyuv_config" ]
   deps = []
 
-  if (is_win && target_cpu == "x64" && !is_qtwebengine) {
+  if (is_msvc && target_cpu == "x64" && !is_qtwebengine) {
     # Compile with clang in order to get inline assembly
     public_deps = [ ":libyuv_internal(//build/toolchain/win:win_clang_x64)" ]
   } else {
@@ -169,7 +169,7 @@
   }
 
   # To enable AVX2 or other cpu optimization, pass flag here
-  if (!is_win) {
+  if (!is_msvc) {
     cflags = [
       # "-mpopcnt",
       # "-mavx2",
@@ -238,14 +238,14 @@
 
 if (libyuv_include_tests) {
   config("libyuv_unittest_warnings_config") {
-    if (!is_win) {
+    if (!is_msvc) {
       cflags = [
         # TODO(fbarchard): Fix sign and unused variable warnings.
         "-Wno-sign-compare",
         "-Wno-unused-variable",
       ]
     }
-    if (is_win) {
+    if (is_msvc) {
       cflags = [
         "/wd4245",  # signed/unsigned mismatch
         "/wd4189",  # local variable is initialized but not referenced
diff -ur a/src/3rdparty/chromium/third_party/nasm/BUILD.gn b/src/3rdparty/chromium/third_party/nasm/BUILD.gn
--- a/src/3rdparty/chromium/third_party/nasm/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/nasm/BUILD.gn	2023-12-14 21:58:57.436872900 +0800
@@ -62,6 +62,12 @@
   ]
 
   defines = [ "HAVE_CONFIG_H" ]
+  if (is_mingw) {
+    defines += [
+      "HAVE_FSEEKO",
+      "HAVE_MEMPCPY"
+    ]
+  }
 
   if (is_clang) {
     cflags = [
diff -ur a/src/3rdparty/chromium/third_party/nasm/include/nasmint.h b/src/3rdparty/chromium/third_party/nasm/include/nasmint.h
--- a/src/3rdparty/chromium/third_party/nasm/include/nasmint.h	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/nasm/include/nasmint.h	2023-12-14 21:58:57.483763400 +0800
@@ -35,6 +35,7 @@
 
 #if _I64_MAX == 9223372036854775807
 
+#ifdef _MSC_VER
 /* Windows-based compiler: use __int64 */
 typedef signed __int64		int64_t;
 typedef unsigned __int64	uint64_t;
@@ -42,6 +43,14 @@
 #define _pri64			"I64"
 #define INT64_C(x)		x ## i64
 #define UINT64_C(x)		x ## ui64
+#else
+typedef signed long long	int64_t;
+typedef unsigned long long	uint64_t;
+#define _scn64			"ll"
+#define _pri64			"ll"
+#define INT64_C(x)		x ## LL
+#define UINT64_C(x)		x ## ULL
+#endif
 
 #elif LONG_MAX == 9223372036854775807L
 
diff -ur a/src/3rdparty/chromium/third_party/node/node.py b/src/3rdparty/chromium/third_party/node/node.py
--- a/src/3rdparty/chromium/third_party/node/node.py	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/node/node.py	2023-12-14 21:58:57.515046500 +0800
@@ -40,7 +40,11 @@
 
 
 def RunNode(cmd_parts, stdout=None):
-  cmd = [GetBinaryPath()] + cmd_parts
+  if sys.platform == 'win32' and 'GCC' in sys.version:
+    # We have to invoke node.exe through winpty
+    cmd = ['node'] + cmd_parts
+  else:
+    cmd = [GetBinaryPath()] + cmd_parts
   process = subprocess.Popen(
       cmd, cwd=os.getcwd(), stdout=subprocess.PIPE, stderr=subprocess.PIPE,
       universal_newlines=True)
diff -ur a/src/3rdparty/chromium/third_party/nearby/src/internal/crypto/BUILD.gn b/src/3rdparty/chromium/third_party/nearby/src/internal/crypto/BUILD.gn
--- a/src/3rdparty/chromium/third_party/nearby/src/internal/crypto/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/nearby/src/internal/crypto/BUILD.gn	2023-12-14 21:58:57.577531500 +0800
@@ -105,7 +105,7 @@
       "capi_util.h",
       "unexportable_key_win.cc",
     ]
-    libs = [ "ncrypt.lib" ]
+    libs = [ "ncrypt" ]
   }
 
   # Some files are built when NSS is used for the platform certificate library.
diff -ur a/src/3rdparty/chromium/third_party/protobuf/BUILD.gn b/src/3rdparty/chromium/third_party/protobuf/BUILD.gn
--- a/src/3rdparty/chromium/third_party/protobuf/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/protobuf/BUILD.gn	2023-12-14 21:58:57.655594100 +0800
@@ -61,7 +61,7 @@
 }
 
 protobuf_lite_cflags = []
-if (is_win) {
+if (is_msvc) {
   protobuf_lite_cflags = [
     "/wd4018",  # signed/unsigned mismatch in comparison
     "/wd4065",  # switch statement contains 'default' but no 'case' labels
diff -ur a/src/3rdparty/chromium/third_party/protobuf/proto_library.gni b/src/3rdparty/chromium/third_party/protobuf/proto_library.gni
--- a/src/3rdparty/chromium/third_party/protobuf/proto_library.gni	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/protobuf/proto_library.gni	2023-12-14 21:58:57.733792900 +0800
@@ -576,7 +576,7 @@
           public_deps += [ "//third_party/protobuf:protobuf_lite" ]
         }
 
-        if (is_win) {
+        if (is_msvc) {
           cflags = [
             # disable: C4125 decimal digit terminates octal escape sequence
             # Protoc generates such sequences frequently, there's no obvious
diff -ur a/src/3rdparty/chromium/third_party/sqlite/BUILD.gn b/src/3rdparty/chromium/third_party/sqlite/BUILD.gn
--- a/src/3rdparty/chromium/third_party/sqlite/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/sqlite/BUILD.gn	2023-12-14 21:58:57.827502900 +0800
@@ -160,7 +160,7 @@
       "-Wno-#warnings",
     ]
   }
-  if (is_win && !is_clang) {
+  if (is_msvc && !is_clang) {
     cflags += [ "/wd4101" ]  # 'zTrace' unreferenced variable in src/src/vdbe.c
   }
 }
diff -ur a/src/3rdparty/chromium/third_party/woff2/BUILD.gn b/src/3rdparty/chromium/third_party/woff2/BUILD.gn
--- a/src/3rdparty/chromium/third_party/woff2/BUILD.gn	2023-10-04 03:49:54.000000000 +0800
+++ b/src/3rdparty/chromium/third_party/woff2/BUILD.gn	2023-12-14 21:58:57.874333800 +0800
@@ -33,7 +33,7 @@
   public_configs = [ ":woff2_includes" ]
 
   # TODO(ksakamoto): http://crbug.com/167187
-  if (is_win) {
+  if (is_msvc) {
     cflags = [ "/wd4267" ]  # Conversion from size_t to 'type'.
   }
 }
