diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index eb83a82..4b3103e 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -657,6 +657,15 @@ add_gn_build_artifacts_to_target(
     NINJA_STAMP QtWebEngineCore.stamp
     ${dataStampOption}
 )
+if (DEFINED QTWEBENGINE_THIRDPARTY_ARCHIVE_LIST_FILE AND EXISTS "${QTWEBENGINE_THIRDPARTY_ARCHIVE_LIST_FILE}")
+    get_filename_component(_qtwebengineThirdPartyArchiveList
+        "${QTWEBENGINE_THIRDPARTY_ARCHIVE_LIST_FILE}" ABSOLUTE)
+    target_link_options(WebEngineCore PRIVATE "@${_qtwebengineThirdPartyArchiveList}")
+endif()
+
+if (DEFINED QTWEBENGINE_THIRDPARTY_OBJECT_LIBRARY AND EXISTS "${QTWEBENGINE_THIRDPARTY_OBJECT_LIBRARY}")
+    target_link_libraries(WebEngineCore PRIVATE "${QTWEBENGINE_THIRDPARTY_OBJECT_LIBRARY}")
+endif()
 add_dependencies(WebEngineCore run_core_NinjaDone)
 if(COIN_BUG_699)
    set_property(TARGET WebEngineCore PROPERTY CXX_LINKER_LAUNCHER ${PROJECT_BINARY_DIR}/linker_ulimit.sh)
endif()
