--- a/src/3rdparty/chromium/third_party/protobuf/proto_library.gni
+++ b/src/3rdparty/chromium/third_party/protobuf/proto_library.gni
@@ -178,9 +178,15 @@
 # case crbug.com/338008085 is resolved.
 _this_dir = get_path_info(".", "abspath")
 
-_protoc_label = "$_this_dir:protoc($host_toolchain)"
-_protoc_path = get_label_info(_protoc_label, "root_out_dir") + "/protoc" +
-               _host_executable_suffix
+if (qtwebengine_private_tool_dir != "") {
+  _protoc_label = ""
+  _protoc_path =
+      qtwebengine_private_tool_dir + "/protoc" + _host_executable_suffix
+} else {
+  _protoc_label = "$_this_dir:protoc($host_toolchain)"
+  _protoc_path = get_label_info(_protoc_label, "root_out_dir") + "/protoc" +
+                 _host_executable_suffix
+}
 _protoc_gen_js_label =
     "//third_party/protobuf-javascript:protoc-gen-js($host_toolchain)"
 _protoc_gen_js_path = get_label_info(_protoc_gen_js_label, "root_out_dir") +
@@ -489,7 +495,10 @@
       outputs += _ts_outputs
     }
 
-    deps = [ _protoc_label ]
+    deps = []
+    if (_protoc_label != "") {
+      deps += [ _protoc_label ]
+    }
     if (host_toolchain_is_msan) {
       deps += [ "//third_party/instrumented_libs:ld-linux($host_toolchain)" ]
       configs = [
@@ -576,10 +585,16 @@
 
         _plugin_host_label =
             "${invoker.generator_plugin_label}($host_toolchain)"
-        _plugin_path =
-            get_label_info(_plugin_host_label, "root_out_dir") + "/" +
+        _plugin_name =
             get_label_info(_plugin_host_label, "name") + _host_executable_suffix
-        deps += [ _plugin_host_label ]
+        if (qtwebengine_private_tool_dir != "") {
+          _plugin_path = qtwebengine_private_tool_dir + "/" + _plugin_name
+        } else {
+          _plugin_path =
+              get_label_info(_plugin_host_label, "root_out_dir") + "/" +
+              _plugin_name
+          deps += [ _plugin_host_label ]
+        }
       } else {
         _plugin_path = invoker.generator_plugin_script
       }
@@ -798,7 +813,10 @@
 
     inputs = invoker.inputs
 
-    deps = [ _protoc_label ]
+    deps = []
+    if (_protoc_label != "") {
+      deps += [ _protoc_label ]
+    }
     if (defined(invoker.deps)) {
       deps += invoker.deps
     }
