--- a/src/3rdparty/chromium/third_party/brotli/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/brotli/BUILD.gn
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//testing/libfuzzer/fuzzer_test.gni")
+import("//build/config/qtwebengine_private_paths.gni")
 
 if (is_win && !is_mingw && !is_robolectric) {
   import("//build/config/win/visual_studio_version.gni")
@@ -184,9 +185,15 @@
 }
 
 if (current_toolchain == host_toolchain) {
-  executable("brotli") {
-    sources = [ "tools/brotli.c" ]
-    public_configs = [ ":includes" ]
+  if (qtwebengine_private_tool_dir != "") {
+    copy("brotli") {
+      sources = [ qtwebengine_private_tool_dir + "/brotli.exe" ]
+      outputs = [ "$root_out_dir/brotli.exe" ]
+    }
+  } else {
+    executable("brotli") {
+      sources = [ "tools/brotli.c" ]
+      public_configs = [ ":includes" ]
     configs += [ ":brotli_defines" ]
 
     deps = [
@@ -208,6 +215,7 @@
       configs -= [ "//build/config:debug" ]
       configs += [ "//build/config:release" ]
     }
+    }
   }
 }
 
