diff --git a/src/3rdparty/chromium/third_party/abseil-cpp/absl/container/internal/raw_hash_set.h b/src/3rdparty/chromium/third_party/abseil-cpp/absl/container/internal/raw_hash_set.h
index 70632095526..6de0dcc7e6d 100644
--- a/src/3rdparty/chromium/third_party/abseil-cpp/absl/container/internal/raw_hash_set.h
+++ b/src/3rdparty/chromium/third_party/abseil-cpp/absl/container/internal/raw_hash_set.h
@@ -3570,7 +3570,7 @@ class raw_hash_set {
     }
   }
 
-  static const PolicyFunctions& GetPolicyFunctions() {
+  static PolicyFunctions GetPolicyFunctions() {
     static_assert(sizeof(slot_type) <= (std::numeric_limits<uint32_t>::max)(),
                   "Slot size is too large. Use std::unique_ptr for value type "
                   "or use absl::node_hash_{map,set}.");
@@ -3581,7 +3581,7 @@ class raw_hash_set {
                   size_t{(std::numeric_limits<uint32_t>::max)()});
     static constexpr size_t kBackingArrayAlignment =
         BackingArrayAlignment(alignof(slot_type));
-    static const PolicyFunctions value = {
+    return {
         static_cast<uint32_t>(sizeof(key_type)),
         static_cast<uint32_t>(sizeof(value_type)),
         static_cast<uint32_t>(sizeof(slot_type)),
@@ -3600,7 +3600,6 @@ class raw_hash_set {
         &AllocateBackingArray<kBackingArrayAlignment, CharAlloc>,
         &DeallocateBackingArray<kBackingArrayAlignment, CharAlloc>,
         &raw_hash_set::transfer_unprobed_elements_to_next_capacity_fn};
-    return value;
   }
 
   // Bundle together CommonFields plus other objects which might be empty.
