diff --git a/dlib/CMakeLists.txt b/dlib/CMakeLists.txt
--- a/dlib/CMakeLists.txt	2018-01-23 01:27:34.000000000 +0100
+++ b/dlib/CMakeLists.txt	2018-02-17 14:32:43.388883900 +0100
@@ -196,7 +196,7 @@
 
    if (DLIB_ISO_CPP_ONLY)
        add_library(dlib STATIC ${source_files} )
-       if (UNIX AND NOT DLIB_IN_PROJECT_BUILD)
+       if ((UNIX OR MINGW) AND NOT DLIB_IN_PROJECT_BUILD)
            add_library(dlib_shared SHARED ${source_files} )
            add_dependencies(dlib_shared dlib)
        endif()
@@ -234,7 +234,7 @@
          dnn/tensor_tools.cpp
          )
 
-      if(UNIX)
+      if(UNIX OR MINGW)
          set(CMAKE_THREAD_PREFER_PTHREAD ON)
          find_package(Threads REQUIRED)
          set(dlib_needed_libraries ${dlib_needed_libraries} ${CMAKE_THREAD_LIBS_INIT})
@@ -701,7 +701,7 @@
       else()
          add_library(dlib STATIC ${source_files} )
       endif()
-      if (UNIX AND NOT DLIB_IN_PROJECT_BUILD)
+      if ((UNIX OR MINGW) AND NOT DLIB_IN_PROJECT_BUILD)
          if (DLIB_USE_CUDA)
             cuda_add_library(dlib_shared SHARED ${source_files} )
             add_dependencies(dlib_shared dlib)
@@ -725,7 +725,7 @@
    else()
       target_compile_options(dlib PRIVATE ${active_preprocessor_switches})
    endif()
-   if (UNIX AND NOT DLIB_IN_PROJECT_BUILD)
+   if ((UNIX OR MINGW) AND NOT DLIB_IN_PROJECT_BUILD)
       target_include_directories(dlib_shared
                                  INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/..>
                                  INTERFACE $<INSTALL_INTERFACE:include>
@@ -755,7 +755,7 @@
 
    # Install the library
    if (NOT DLIB_IN_PROJECT_BUILD)
-       if(UNIX)
+       if(UNIX OR MINGW)
            set_target_properties(dlib_shared PROPERTIES
                                         OUTPUT_NAME dlib 
                                         VERSION ${VERSION})
