--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -5,11 +5,11 @@
 libvo = $(top_builddir)/libvo/libvo.a $(LIBVO_LIBS)
 
 bin_PROGRAMS = mpeg2dec extract_mpeg2 corrupt_mpeg2
-mpeg2dec_SOURCES = mpeg2dec.c dump_state.c getopt.c gettimeofday.c
+mpeg2dec_SOURCES = mpeg2dec.c dump_state.c gettimeofday.c
 mpeg2dec_LDADD = $(libvo) $(libmpeg2) $(libmpeg2convert)
-extract_mpeg2_SOURCES = extract_mpeg2.c getopt.c
-corrupt_mpeg2_SOURCES = corrupt_mpeg2.c getopt.c
+extract_mpeg2_SOURCES = extract_mpeg2.c
+corrupt_mpeg2_SOURCES = corrupt_mpeg2.c
 
 man_MANS = mpeg2dec.1 extract_mpeg2.1
 
-EXTRA_DIST = getopt.h gettimeofday.h $(man_MANS)
+EXTRA_DIST = gettimeofday.h $(man_MANS)
