--- a/libvo/video_out_dx.c
+++ b/libvo/video_out_dx.c
@@ -34,6 +34,7 @@
 #include "mpeg2.h"
 #include "video_out.h"
 #include "mpeg2convert.h"
+#include "vo_internal.h"
 
 #include <ddraw.h>
 #include <initguid.h>
@@ -92,7 +93,7 @@
     switch (message) {
 
     case WM_WINDOWPOSCHANGED:
-	instance = (dx_instance_t *) GetWindowLong (hwnd, GWL_USERDATA);
+	instance = (dx_instance_t *) GetWindowLongPtr (hwnd, GWLP_USERDATA);
 
 	/* update the window position and size */
 	point_window.x = 0;
@@ -173,7 +174,7 @@
     /* store a directx_instance pointer into the window local storage
      * (for later use in event_handler).
      * We need to use SetWindowLongPtr when it is available in mingw */
-    SetWindowLong (instance->window, GWL_USERDATA, (LONG) instance);
+    SetWindowLongPtr (instance->window, GWLP_USERDATA, (LONG_PTR) instance);
 
     ShowWindow (instance->window, SW_SHOW);
 
