--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,13 +40,6 @@ if(HAVE_STRDUP)
 endif(HAVE_STRDUP)
 
 if (MINGW)
-    # Apply workaround for MinGW timespec redefinition (pthread.h / time.h)
-    include(CheckStructHasMember)
-    check_struct_has_member("struct timespec" tv_sec time.h HAVE_STRUCT_TIMESPEC)
-    if (HAVE_STRUCT_TIMESPEC)
-        add_definitions(-D_TIMESPEC_DEFINED=1)
-    endif()
-
     if (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         # Apply workaround for static/shared libraries on MinGW C/C++ compiler
         # Issue occurs with CMake >= 3.9.0, it doesn't filter out gcc,gcc_s,gcc_eh from
