--- a/source/common/charstr.h
+++ b/source/common/charstr.h
@@ -34,20 +34,20 @@
  */
 class U_COMMON_API_CLASS CharString : public UMemory {
 public:
-    U_COMMON_API CharString() : len(0) { buffer[0]=0; }
-    U_COMMON_API CharString(StringPiece s, UErrorCode &errorCode) : len(0) {
+    CharString() : len(0) { buffer[0]=0; }
+    CharString(StringPiece s, UErrorCode &errorCode) : len(0) {
         buffer[0]=0;
         append(s, errorCode);
     }
-    U_COMMON_API CharString(const CharString &s, UErrorCode &errorCode) : len(0) {
+    CharString(const CharString &s, UErrorCode &errorCode) : len(0) {
         buffer[0]=0;
         append(s, errorCode);
     }
-    U_COMMON_API CharString(const char *s, int32_t sLength, UErrorCode &errorCode) : len(0) {
+    CharString(const char *s, int32_t sLength, UErrorCode &errorCode) : len(0) {
         buffer[0]=0;
         append(s, sLength, errorCode);
     }
-    U_COMMON_API ~CharString() {}
+    ~CharString() {}
 
     /**
      * Move constructor; might leave src in an undefined state.
@@ -70,13 +70,13 @@
     U_COMMON_API CharString &copyFrom(const CharString &other, UErrorCode &errorCode);
     U_COMMON_API CharString &copyFrom(StringPiece s, UErrorCode &errorCode);
 
-    U_COMMON_API UBool isEmpty() const { return len==0; }
-    U_COMMON_API int32_t length() const { return len; }
-    U_COMMON_API char operator[](int32_t index) const { return buffer[index]; }
-    U_COMMON_API StringPiece toStringPiece() const { return StringPiece(buffer.getAlias(), len); }
+    UBool isEmpty() const { return len==0; }
+    int32_t length() const { return len; }
+    char operator[](int32_t index) const { return buffer[index]; }
+    StringPiece toStringPiece() const { return StringPiece(buffer.getAlias(), len); }
 
-    U_COMMON_API const char *data() const { return buffer.getAlias(); }
-    U_COMMON_API char *data() { return buffer.getAlias(); }
+    const char *data() const { return buffer.getAlias(); }
+    char *data() { return buffer.getAlias(); }
     /**
      * Allocates length()+1 chars and copies the NUL-terminated data().
      * The caller must uprv_free() the result.
@@ -99,17 +99,17 @@
      */
     U_COMMON_API int32_t extract(char *dest, int32_t capacity, UErrorCode &errorCode) const;
 
-    U_COMMON_API bool operator==(const CharString& other) const {
+    bool operator==(const CharString& other) const {
         return len == other.length() && (len == 0 || uprv_memcmp(data(), other.data(), len) == 0);
     }
-    U_COMMON_API bool operator!=(const CharString& other) const {
+    bool operator!=(const CharString& other) const {
         return !operator==(other);
     }
 
-    U_COMMON_API bool operator==(StringPiece other) const {
+    bool operator==(StringPiece other) const {
         return len == other.length() && (len == 0 || uprv_memcmp(data(), other.data(), len) == 0);
     }
-    U_COMMON_API bool operator!=(StringPiece other) const {
+    bool operator!=(StringPiece other) const {
         return !operator==(other);
     }
 
@@ -118,14 +118,14 @@
 
     U_COMMON_API bool contains(StringPiece s) const;
 
-    U_COMMON_API CharString &clear() { len=0; buffer[0]=0; return *this; }
+    CharString &clear() { len=0; buffer[0]=0; return *this; }
     U_COMMON_API CharString &truncate(int32_t newLength);
 
     U_COMMON_API CharString &append(char c, UErrorCode &errorCode);
-    U_COMMON_API CharString &append(StringPiece s, UErrorCode &errorCode) {
+    CharString &append(StringPiece s, UErrorCode &errorCode) {
         return append(s.data(), s.length(), errorCode);
     }
-    U_COMMON_API CharString &append(const CharString &s, UErrorCode &errorCode) {
+    CharString &append(const CharString &s, UErrorCode &errorCode) {
         return append(s.data(), s.length(), errorCode);
     }
     U_COMMON_API CharString &append(const char *s, int32_t sLength, UErrorCode &status);
--- a/source/common/normalizer2impl.h
+++ b/source/common/normalizer2impl.h
@@ -245,7 +245,7 @@
  */
 class U_COMMON_API_CLASS Normalizer2Impl : public UObject {
 public:
-    U_COMMON_API Normalizer2Impl() : normTrie(nullptr), fCanonIterData(nullptr) {}
+    Normalizer2Impl() : normTrie(nullptr), fCanonIterData(nullptr) {}
     U_COMMON_API virtual ~Normalizer2Impl();
 
     U_COMMON_API void init(const int32_t* inIndexes,
@@ -263,16 +263,16 @@
 
     // The trie stores values for lead surrogate code *units*.
     // Surrogate code *points* are inert.
-    U_COMMON_API uint16_t getNorm16(UChar32 c) const {
+    uint16_t getNorm16(UChar32 c) const {
         return U_IS_LEAD(c) ?
             static_cast<uint16_t>(INERT) :
             UCPTRIE_FAST_GET(normTrie, UCPTRIE_16, c);
     }
-    U_COMMON_API uint16_t getRawNorm16(UChar32 c) const {
+    uint16_t getRawNorm16(UChar32 c) const {
         return UCPTRIE_FAST_GET(normTrie, UCPTRIE_16, c);
     }
 
-    U_COMMON_API UNormalizationCheckResult getCompQuickCheck(uint16_t norm16) const {
+    UNormalizationCheckResult getCompQuickCheck(uint16_t norm16) const {
         if(norm16<minNoNo || MIN_YES_YES_WITH_CC<=norm16) {
             return UNORM_YES;
         } else if(minMaybeNo<=norm16) {
@@ -281,17 +281,17 @@
             return UNORM_NO;
         }
     }
-    U_COMMON_API UBool isAlgorithmicNoNo(uint16_t norm16) const {
+    UBool isAlgorithmicNoNo(uint16_t norm16) const {
         return limitNoNo <= norm16 && norm16 < minMaybeNo;
     }
-    U_COMMON_API UBool isCompNo(uint16_t norm16) const {
+    UBool isCompNo(uint16_t norm16) const {
         return minNoNo <= norm16 && norm16 < minMaybeNo;
     }
-    U_COMMON_API UBool isDecompYes(uint16_t norm16) const {
+    UBool isDecompYes(uint16_t norm16) const {
         return norm16 < minYesNo || minMaybeYes <= norm16;
     }
 
-    U_COMMON_API uint8_t getCC(uint16_t norm16) const {
+    uint8_t getCC(uint16_t norm16) const {
         if(norm16>=MIN_NORMAL_MAYBE_YES) {
             return getCCFromNormalYesOrMaybe(norm16);
         }
@@ -300,13 +300,13 @@
         }
         return getCCFromNoNo(norm16);
     }
-    U_COMMON_API static uint8_t getCCFromNormalYesOrMaybe(uint16_t norm16) {
+    static uint8_t getCCFromNormalYesOrMaybe(uint16_t norm16) {
         return static_cast<uint8_t>(norm16 >> OFFSET_SHIFT);
     }
-    U_COMMON_API static uint8_t getCCFromYesOrMaybeYes(uint16_t norm16) {
+    static uint8_t getCCFromYesOrMaybeYes(uint16_t norm16) {
         return norm16>=MIN_NORMAL_MAYBE_YES ? getCCFromNormalYesOrMaybe(norm16) : 0;
     }
-    U_COMMON_API uint8_t getCCFromYesOrMaybeYesCP(UChar32 c) const {
+    uint8_t getCCFromYesOrMaybeYesCP(UChar32 c) const {
         if (c < minCompNoMaybeCP) { return 0; }
         return getCCFromYesOrMaybeYes(getNorm16(c));
     }
@@ -316,7 +316,7 @@
      * @param c A Unicode code point.
      * @return The lccc(c) in bits 15..8 and tccc(c) in bits 7..0.
      */
-    U_COMMON_API uint16_t getFCD16(UChar32 c) const {
+    uint16_t getFCD16(UChar32 c) const {
         if(c<minDecompNoCP) {
             return 0;
         } else if(c<=0xffff) {
@@ -332,7 +332,7 @@
      * @param limit The end of the string, or NULL.
      * @return The lccc(c) in bits 15..8 and tccc(c) in bits 7..0.
      */
-    U_COMMON_API uint16_t nextFCD16(const char16_t*& s, const char16_t* limit) const {
+    uint16_t nextFCD16(const char16_t*& s, const char16_t* limit) const {
         UChar32 c=*s++;
         if(c<minDecompNoCP || !singleLeadMightHaveNonZeroFCD16(c)) {
             return 0;
@@ -350,7 +350,7 @@
      * @param s A valid pointer into a string. Requires start<s.
      * @return The lccc(c) in bits 15..8 and tccc(c) in bits 7..0.
      */
-    U_COMMON_API uint16_t previousFCD16(const char16_t* start, const char16_t*& s) const {
+    uint16_t previousFCD16(const char16_t* start, const char16_t*& s) const {
         UChar32 c=*--s;
         if(c<minDecompNoCP) {
             return 0;
@@ -370,7 +370,7 @@
     }
 
     /** Returns true if the single-or-lead code unit c might have non-zero FCD data. */
-    U_COMMON_API UBool singleLeadMightHaveNonZeroFCD16(UChar32 lead) const {
+    UBool singleLeadMightHaveNonZeroFCD16(UChar32 lead) const {
         // 0<=lead<=0xffff
         uint8_t bits=smallFCD[lead>>8];
         if(bits==0) { return false; }
@@ -568,15 +568,15 @@
     U_COMMON_API UBool norm16HasDecompBoundaryBefore(uint16_t norm16) const;
     U_COMMON_API UBool hasDecompBoundaryAfter(UChar32 c) const;
     U_COMMON_API UBool norm16HasDecompBoundaryAfter(uint16_t norm16) const;
-    U_COMMON_API UBool isDecompInert(UChar32 c) const { return isDecompYesAndZeroCC(getNorm16(c)); }
+    UBool isDecompInert(UChar32 c) const { return isDecompYesAndZeroCC(getNorm16(c)); }
 
-    U_COMMON_API UBool hasCompBoundaryBefore(UChar32 c) const {
+    UBool hasCompBoundaryBefore(UChar32 c) const {
         return c<minCompNoMaybeCP || norm16HasCompBoundaryBefore(getNorm16(c));
     }
-    U_COMMON_API UBool hasCompBoundaryAfter(UChar32 c, UBool onlyContiguous) const {
+    UBool hasCompBoundaryAfter(UChar32 c, UBool onlyContiguous) const {
         return norm16HasCompBoundaryAfter(getNorm16(c), onlyContiguous);
     }
-    U_COMMON_API UBool isCompInert(UChar32 c, UBool onlyContiguous) const {
+    UBool isCompInert(UChar32 c, UBool onlyContiguous) const {
         uint16_t norm16=getNorm16(c);
         return isCompYesAndZeroCC(norm16) &&
             (norm16 & HAS_COMP_BOUNDARY_AFTER) != 0 &&
@@ -584,9 +584,9 @@
             // The last check fetches the mapping's first unit and checks tccc<=1.
     }
 
-    U_COMMON_API UBool hasFCDBoundaryBefore(UChar32 c) const { return hasDecompBoundaryBefore(c); }
-    U_COMMON_API UBool hasFCDBoundaryAfter(UChar32 c) const { return hasDecompBoundaryAfter(c); }
-    U_COMMON_API UBool isFCDInert(UChar32 c) const { return getFCD16(c) <= 1; }
+    UBool hasFCDBoundaryBefore(UChar32 c) const { return hasDecompBoundaryBefore(c); }
+    UBool hasFCDBoundaryAfter(UChar32 c) const { return hasDecompBoundaryAfter(c); }
+    UBool isFCDInert(UChar32 c) const { return getFCD16(c) <= 1; }
 
   private:
     friend class InitCanonIterData;
--- a/source/common/sharedobject.h
+++ b/source/common/sharedobject.h
@@ -54,13 +54,13 @@
 class U_COMMON_API_CLASS SharedObject : public UObject {
 public:
     /** Initializes totalRefCount, softRefCount to 0. */
-    U_COMMON_API SharedObject() :
+    SharedObject() :
             softRefCount(0),
             hardRefCount(0),
             cachePtr(nullptr) {}
 
     /** Initializes totalRefCount, softRefCount to 0. */
-    U_COMMON_API SharedObject(const SharedObject &other) :
+    SharedObject(const SharedObject &other) :
             UObject(other),
             softRefCount(0),
             hardRefCount(0),
--- a/source/common/umutex.h
+++ b/source/common/umutex.h
@@ -84,8 +84,8 @@
     UErrorCode fErrCode{U_ZERO_ERROR};
 
 public:
-    U_COMMON_API void reset() { fState = 0; }
-    U_COMMON_API UBool isReset() { return umtx_loadAcquire(fState) == 0; }
+    void reset() { fState = 0; }
+    UBool isReset() { return umtx_loadAcquire(fState) == 0; }
 // Note: isReset() is used by service registration code.
 //                 Thread safety of this usage needs review.
 };
@@ -205,19 +205,19 @@
 class U_COMMON_API_CLASS UMutex {
 public:
     U_COMMON_API UMUTEX_CONSTEXPR UMutex() {}
-    U_COMMON_API ~UMutex() = default;
+    ~UMutex() = default;
 
     UMutex(const UMutex& other) = delete;
     UMutex& operator=(const UMutex& other) = delete;
     void* operator new(size_t) = delete;
 
     // requirements for C++ BasicLockable, allows UMutex to work with std::lock_guard
-    U_COMMON_API void lock() {
+    void lock() {
         std::mutex *m = fMutex.load(std::memory_order_acquire);
         if (m == nullptr) { m = getMutex(); }
         m->lock();
     }
-    U_COMMON_API void unlock() { fMutex.load(std::memory_order_relaxed)->unlock(); }
+    void unlock() { fMutex.load(std::memory_order_relaxed)->unlock(); }
 
     U_COMMON_API static void cleanup();
 
--- a/source/common/unicode/brkiter.h
+++ b/source/common/unicode/brkiter.h
@@ -132,7 +132,7 @@
      * @return the complement of the result of operator==
      * @stable ICU 2.0
      */
-    U_COMMON_API bool operator!=(const BreakIterator& rhs) const { return !operator==(rhs); }
+    bool operator!=(const BreakIterator& rhs) const { return !operator==(rhs); }
 
     /**
      * Return a polymorphic copy of this object.  This is an abstract
--- a/source/common/unicode/locid.h
+++ b/source/common/unicode/locid.h
@@ -739,7 +739,7 @@
      *
      * @stable ICU 49
      */
-    U_COMMON_API void setKeywordValue(const char* keywordName,
+    void setKeywordValue(const char* keywordName,
                                       const char* keywordValue,
                                       UErrorCode& status) {
         setKeywordValue(StringPiece{keywordName}, StringPiece{keywordValue}, status);
--- a/source/i18n/collationtailoring.h
+++ b/source/i18n/collationtailoring.h
@@ -56,7 +56,7 @@
     /**
      * Returns true if the constructor could not initialize properly.
      */
-    U_I18N_API UBool isBogus() { return settings == nullptr; }
+    UBool isBogus() { return settings == nullptr; }
 
     U_I18N_API UBool ensureOwnedData(UErrorCode &errorCode);
 
@@ -98,7 +98,7 @@
 
 // U_I18N_API because gencolusb uses it.
 struct U_I18N_API_CLASS CollationCacheEntry : public SharedObject {
-    U_I18N_API CollationCacheEntry(const Locale &loc, const CollationTailoring *t)
+    CollationCacheEntry(const Locale &loc, const CollationTailoring *t)
             : validLocale(loc), tailoring(t) {
         if(t != nullptr) {
             t->addRef();
--- a/source/i18n/unicode/calendar.h
+++ b/source/i18n/unicode/calendar.h
@@ -468,7 +468,7 @@
      *              otherwise.
      * @stable ICU 2.0
      */
-    U_I18N_API bool operator!=(const Calendar& that) const { return !operator==(that); }
+    bool operator!=(const Calendar& that) const { return !operator==(that); }
 
     /**
      * Returns true if the given Calendar object is equivalent to this
@@ -1565,7 +1565,7 @@
      * @return The year in era 0 of this calendar is counting backward from 1.
      * @internal
      */
-    U_I18N_API virtual bool isEra0CountingBackward() const { return false; }
+    virtual bool isEra0CountingBackward() const { return false; }
 
     /**
      * Related year difference.
@@ -1916,7 +1916,7 @@
      * @return     the current time without recomputing.
      * @stable ICU 2.0
      */
-    U_I18N_API UDate internalGetTime() const { return fTime; }
+    UDate internalGetTime() const { return fTime; }
 
     /**
      * Set the current time without affecting flags or fields.
@@ -1925,7 +1925,7 @@
      * @return        the current time without recomputing.
      * @stable ICU 2.0
      */
-    U_I18N_API void internalSetTime(UDate time) { fTime = time; }
+    void internalSetTime(UDate time) { fTime = time; }
 
     /**
      * The time fields containing values into which the millis is computed.
@@ -1984,7 +1984,7 @@
      * <code>computeGregorianFields()</code>.
      * @internal
      */
-    U_I18N_API int32_t getGregorianYear() const {
+    int32_t getGregorianYear() const {
         return fGregorianYear;
     }
 
@@ -1993,7 +1993,7 @@
      * <code>computeGregorianFields()</code>.
      * @internal
      */
-    U_I18N_API int32_t getGregorianMonth() const {
+    int32_t getGregorianMonth() const {
         return fGregorianMonth;
     }
 
@@ -2002,7 +2002,7 @@
      * computed by <code>computeGregorianFields()</code>.
      * @internal
      */
-    U_I18N_API int32_t getGregorianDayOfYear() const {
+    int32_t getGregorianDayOfYear() const {
         return fGregorianDayOfYear;
     }
 
@@ -2011,7 +2011,7 @@
      * computed by <code>computeGregorianFields()</code>.
      * @internal
      */
-    U_I18N_API int32_t getGregorianDayOfMonth() const {
+    int32_t getGregorianDayOfMonth() const {
       return fGregorianDayOfMonth;
     }
 #endif  /* U_HIDE_INTERNAL_API */
--- a/source/i18n/unicode/dcfmtsym.h
+++ b/source/i18n/unicode/dcfmtsym.h
@@ -270,7 +270,7 @@
      * @return         true if another object is semantically unequal to this one.
      * @stable ICU 2.0
      */
-    U_I18N_API bool operator!=(const DecimalFormatSymbols& other) const { return !operator==(other); }
+    bool operator!=(const DecimalFormatSymbols& other) const { return !operator==(other); }
 
     /**
      * Get one of the format symbols by its enum constant.
--- a/source/i18n/unicode/dtfmtsym.h
+++ b/source/i18n/unicode/dtfmtsym.h
@@ -180,7 +180,7 @@
      * @return         true if other is semantically unequal to this.
      * @stable ICU 2.0
      */
-    U_I18N_API bool operator!=(const DateFormatSymbols& other) const { return !operator==(other); }
+    bool operator!=(const DateFormatSymbols& other) const { return !operator==(other); }
 
     /**
      * Gets abbreviated era strings. For example: "AD" and "BC".
--- a/source/i18n/unicode/format.h
+++ b/source/i18n/unicode/format.h
@@ -120,7 +120,7 @@
      * @return         Return true if the given Format objects are not semantically.
      * @stable ICU 2.0
      */
-    U_I18N_API bool operator!=(const Format& other) const { return !operator==(other); }
+    bool operator!=(const Format& other) const { return !operator==(other); }
 
     /**
      * Clone this object polymorphically.  The caller is responsible
--- a/source/i18n/unicode/messageformat2.h
+++ b/source/i18n/unicode/messageformat2.h
@@ -102,7 +102,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API FormattedMessage format(const MessageArguments& arguments, UErrorCode& status) const {
+        FormattedMessage format(const MessageArguments& arguments, UErrorCode& status) const {
             (void) arguments;
             if (U_SUCCESS(status)) {
                 status = U_UNSUPPORTED_ERROR;
@@ -118,7 +118,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const Locale& getLocale() const { return locale; }
+        const Locale& getLocale() const { return locale; }
 
         /**
          * Serializes the data model as a string in MessageFormat 2.0 syntax.
@@ -326,7 +326,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const UnicodeString& getNormalizedPattern() const { return normalizedInput; }
+        const UnicodeString& getNormalizedPattern() const { return normalizedInput; }
 
     private:
         friend class Builder;
--- a/source/i18n/unicode/messageformat2_arguments.h
+++ b/source/i18n/unicode/messageformat2_arguments.h
@@ -57,7 +57,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API MessageArguments(const std::map<UnicodeString, Formattable>& args, UErrorCode& status) {
+        MessageArguments(const std::map<UnicodeString, Formattable>& args, UErrorCode& status) {
             if (U_FAILURE(status)) {
                 return;
             }
@@ -89,7 +89,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API MessageArguments() = default;
+        MessageArguments() = default;
         /**
          * Destructor.
          *
--- a/source/i18n/unicode/messageformat2_data_model.h
+++ b/source/i18n/unicode/messageformat2_data_model.h
@@ -268,7 +268,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API Operand() : contents(std::nullopt) {}
+            Operand() : contents(std::nullopt) {}
             /**
              * Variable operand constructor.
              *
@@ -278,7 +278,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API explicit Operand(const UnicodeString& v) : contents(VariableName(v)) {}
+            explicit Operand(const UnicodeString& v) : contents(VariableName(v)) {}
             /**
              * Literal operand constructor.
              *
@@ -288,7 +288,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API explicit Operand(const Literal& l) : contents(l) {}
+            explicit Operand(const Literal& l) : contents(l) {}
             /**
              * Non-member swap function.
              * @param o1 will get o2's contents
@@ -353,7 +353,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API UBool isWildcard() const { return !contents.has_value(); }
+            UBool isWildcard() const { return !contents.has_value(); }
             /**
              * Returns the contents of this key as a literal.
              * Precondition: !isWildcard()
@@ -370,7 +370,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API Key(const Key& other) : contents(other.contents) {}
+            Key(const Key& other) : contents(other.contents) {}
             /**
              * Wildcard constructor; constructs a Key representing the
              * catchall or wildcard key, '*'.
@@ -378,7 +378,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API Key() : contents(std::nullopt) {}
+            Key() : contents(std::nullopt) {}
             /**
              * Literal key constructor.
              *
@@ -388,7 +388,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API explicit Key(const Literal& lit) : contents(lit) {}
+            explicit Key(const Literal& lit) : contents(lit) {}
             /**
              * Non-member swap function.
              * @param k1 will get k2's contents
@@ -470,7 +470,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API std::vector<Key> getKeys() const {
+            std::vector<Key> getKeys() const {
                 return toStdVector<Key>(keys.getAlias(), len);
             }
             /**
@@ -557,7 +557,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API SelectorKeys() : len(0) {}
+            SelectorKeys() : len(0) {}
             /**
              * Non-member swap function.
              * @param s1 will get s2's contents
@@ -711,10 +711,10 @@
                 swap(m1.options, m2.options);
                 swap(m1.len, m2.len);
             }
-            U_I18N_API OptionMap() : len(0) {}
+            OptionMap() : len(0) {}
             U_I18N_API OptionMap(const OptionMap&);
             U_I18N_API OptionMap& operator=(OptionMap);
-            U_I18N_API std::vector<Option> getOptions() const {
+            std::vector<Option> getOptions() const {
                 return toStdVector<Option>(options.getAlias(), len);
             }
             U_I18N_API OptionMap(const UVector&, UErrorCode&);
@@ -1214,7 +1214,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API std::vector<Option> getAttributes() const { return attributes.getOptions(); }
+            std::vector<Option> getAttributes() const { return attributes.getOptions(); }
             /**
              * The mutable `Expression::Builder` class allows the operator to be constructed
              * incrementally.
@@ -1399,7 +1399,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API UBool isText() const { return std::holds_alternative<UnicodeString>(piece); }
+            UBool isText() const { return std::holds_alternative<UnicodeString>(piece); }
             /**
              * Checks if the part is a markup part.
              *
@@ -1408,7 +1408,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API UBool isMarkup() const { return std::holds_alternative<Markup>(piece); }
+            UBool isMarkup() const { return std::holds_alternative<Markup>(piece); }
             /**
              * Checks if the part is an expression part.
              *
@@ -1417,7 +1417,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API UBool isExpression() const { return std::holds_alternative<Expression>(piece); }
+            UBool isExpression() const { return std::holds_alternative<Expression>(piece); }
             /**
              * Accesses the expression of the part.
              * Precondition: isExpression()
@@ -1491,7 +1491,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API explicit PatternPart(const UnicodeString& t) : piece(t) {}
+            explicit PatternPart(const UnicodeString& t) : piece(t) {}
             /**
              * Expression part constructor. Returns an Expression pattern
              * part with expression `e`.
@@ -1501,7 +1501,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API explicit PatternPart(Expression&& e) : piece(e) {}
+            explicit PatternPart(Expression&& e) : piece(e) {}
             /**
              * Markup part constructor. Returns a Markup pattern
              * part with markup `m`
@@ -1511,7 +1511,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API explicit PatternPart(Markup&& m) : piece(m) {}
+            explicit PatternPart(Markup&& m) : piece(m) {}
             /**
              * Default constructor.
              * Puts the PatternPart into a valid but undefined state.
@@ -1519,7 +1519,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API PatternPart() = default;
+            PatternPart() = default;
         private:
             friend class Pattern;
 
@@ -1553,7 +1553,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API Iterator begin() const {
+            Iterator begin() const {
                 return Iterator(this, 0);
             }
             /**
@@ -1565,7 +1565,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API Iterator end() const {
+            Iterator end() const {
                 return Iterator(this, len);
             }
             /**
@@ -1660,7 +1660,7 @@
              * @internal ICU 75 technology preview
              * @deprecated This API is for technology preview only.
              */
-            U_I18N_API Pattern() : parts(LocalArray<PatternPart>()) {}
+            Pattern() : parts(LocalArray<PatternPart>()) {}
             /**
              * Non-member swap function.
              * @param p1 will get p2's contents
@@ -2136,7 +2136,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API std::vector<Binding> getLocalVariables() const {
+        std::vector<Binding> getLocalVariables() const {
             std::vector<Binding> result;
             if (!bogus) {
                 return toStdVector<Binding>(bindings.getAlias(), bindingsLen);
@@ -2151,7 +2151,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API std::vector<VariableName> getSelectors() const {
+        std::vector<VariableName> getSelectors() const {
             if (std::holds_alternative<Pattern>(body)) {
                 return {};
             }
@@ -2167,7 +2167,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API std::vector<Variant> getVariants() const {
+        std::vector<Variant> getVariants() const {
             // Return empty vector if no variants
             if (std::holds_alternative<Pattern>(body)) {
                 return {};
--- a/source/i18n/unicode/messageformat2_formattable.h
+++ b/source/i18n/unicode/messageformat2_formattable.h
@@ -130,7 +130,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API double getDouble(UErrorCode& status) const {
+        double getDouble(UErrorCode& status) const {
             if (U_SUCCESS(status)) {
                 if (isDecimal() && getType() == UFMT_DOUBLE) {
                     return (std::get_if<icu::Formattable>(&contents))->getDouble();
@@ -152,7 +152,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API int32_t getLong(UErrorCode& status) const {
+        int32_t getLong(UErrorCode& status) const {
             if (U_SUCCESS(status)) {
                 if (isDecimal() && getType() == UFMT_LONG) {
                     return std::get_if<icu::Formattable>(&contents)->getLong();
@@ -175,7 +175,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API int64_t getInt64Value(UErrorCode& status) const {
+        int64_t getInt64Value(UErrorCode& status) const {
             if (U_SUCCESS(status)) {
                 if (isDecimal() && getType() == UFMT_INT64) {
                     return std::get_if<icu::Formattable>(&contents)->getInt64();
@@ -212,7 +212,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const UnicodeString& getString(UErrorCode& status) const {
+        const UnicodeString& getString(UErrorCode& status) const {
             if (U_SUCCESS(status)) {
                 if (std::holds_alternative<UnicodeString>(contents)) {
                     return *std::get_if<UnicodeString>(&contents);
@@ -233,7 +233,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const DateInfo* getDate(UErrorCode& status) const {
+        const DateInfo* getDate(UErrorCode& status) const {
             if (U_SUCCESS(status)) {
                 if (isDate()) {
                     return std::get_if<DateInfo>(&contents);
@@ -250,7 +250,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API UBool isNumeric() const { return (getType() == UFMT_DOUBLE || getType() == UFMT_LONG || getType() == UFMT_INT64); }
+        UBool isNumeric() const { return (getType() == UFMT_DOUBLE || getType() == UFMT_LONG || getType() == UFMT_INT64); }
 
         /**
          * Gets the array value and count of this object. If this object
@@ -274,7 +274,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const FormattableObject* getObject(UErrorCode& status) const {
+        const FormattableObject* getObject(UErrorCode& status) const {
             if (U_SUCCESS(status)) {
                 // Can't return a reference since FormattableObject
                 // is an abstract class
@@ -320,7 +320,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable() : contents(0.0) {}
+        Formattable() : contents(0.0) {}
         /**
          * String constructor.
          *
@@ -329,7 +329,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable(const UnicodeString& s) : contents(s) {}
+        Formattable(const UnicodeString& s) : contents(s) {}
         /**
          * Double constructor.
          *
@@ -338,7 +338,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable(double d) : contents(d) {}
+        Formattable(double d) : contents(d) {}
         /**
          * Int64 constructor.
          *
@@ -347,7 +347,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable(int64_t i) : contents(i) {}
+        Formattable(int64_t i) : contents(i) {}
         /**
          * Date constructor.
          *
@@ -357,7 +357,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable(DateInfo&& d) : contents(std::move(d)) {}
+        Formattable(DateInfo&& d) : contents(std::move(d)) {}
         /**
          * Creates a Formattable object of an appropriate numeric type from a
          * a decimal number in string form.  The Formattable will retain the
@@ -382,7 +382,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable(const Formattable* arr, int32_t len) : contents(std::pair(arr, len)) {}
+        Formattable(const Formattable* arr, int32_t len) : contents(std::pair(arr, len)) {}
         /**
          * Object constructor.
          *
@@ -391,7 +391,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API Formattable(const FormattableObject* obj) : contents(obj) {}
+        Formattable(const FormattableObject* obj) : contents(obj) {}
         /**
          * Destructor.
          *
@@ -455,14 +455,14 @@
     /* const */ bool sourceIsLiteral;
 
   public:
-      U_I18N_API const UnicodeString& getName() const { return name; }
-      U_I18N_API const Formattable& getValue() const { return value; }
-      U_I18N_API bool isLiteral() const { return sourceIsLiteral; }
-      U_I18N_API ResolvedFunctionOption(const UnicodeString& n, const Formattable& f, bool s)
+      const UnicodeString& getName() const { return name; }
+      const Formattable& getValue() const { return value; }
+      bool isLiteral() const { return sourceIsLiteral; }
+      ResolvedFunctionOption(const UnicodeString& n, const Formattable& f, bool s)
           : name(n), value(f), sourceIsLiteral(s) {}
-      U_I18N_API ResolvedFunctionOption() {}
+      ResolvedFunctionOption() {}
       U_I18N_API ResolvedFunctionOption(ResolvedFunctionOption&&);
-      U_I18N_API ResolvedFunctionOption& operator=(ResolvedFunctionOption&& other) noexcept {
+      ResolvedFunctionOption& operator=(ResolvedFunctionOption&& other) noexcept {
           name = std::move(other.name);
           value = std::move(other.value);
           sourceIsLiteral = other.sourceIsLiteral;
@@ -700,7 +700,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API explicit FormattedPlaceholder(const UnicodeString& s) : fallback(s), type(kFallback) {}
+        explicit FormattedPlaceholder(const UnicodeString& s) : fallback(s), type(kFallback) {}
         /**
          * Constructor for fully formatted placeholders.
          *
@@ -712,7 +712,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API FormattedPlaceholder(const FormattedPlaceholder& input, FormattedValue&& output)
+        FormattedPlaceholder(const FormattedPlaceholder& input, FormattedValue&& output)
             : fallback(input.fallback), source(input.source),
             formatted(std::move(output)), previousOptions(FunctionOptions()), type(kEvaluated) {}
         /**
@@ -727,7 +727,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API FormattedPlaceholder(const FormattedPlaceholder& input, FunctionOptions&& opts, FormattedValue&& output)
+        FormattedPlaceholder(const FormattedPlaceholder& input, FunctionOptions&& opts, FormattedValue&& output)
             : fallback(input.fallback), source(input.source),
             formatted(std::move(output)), previousOptions(std::move(opts)), type(kEvaluated) {}
         /**
@@ -739,7 +739,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API FormattedPlaceholder(const Formattable& input, const UnicodeString& fb)
+        FormattedPlaceholder(const Formattable& input, const UnicodeString& fb)
             : fallback(fb), source(input), type(kUnevaluated) {}
         /**
          * Default constructor. Leaves the FormattedPlaceholder in a
@@ -748,7 +748,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API FormattedPlaceholder() : type(kNull) {}
+        FormattedPlaceholder() : type(kNull) {}
         /**
          * Returns the source `Formattable` value for this placeholder.
          * The result is undefined if this is a null operand.
@@ -768,7 +768,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API bool isFallback() const { return type == kFallback; }
+        bool isFallback() const { return type == kFallback; }
         /**
          * Returns true iff this is a null placeholder.
          *
@@ -778,7 +778,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API bool isNullOperand() const { return type == kNull; }
+        bool isNullOperand() const { return type == kNull; }
         /**
          * Returns true iff this has formatting output.
          *
@@ -788,7 +788,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API bool isEvaluated() const { return (type == kEvaluated); }
+        bool isEvaluated() const { return (type == kEvaluated); }
         /**
          * Returns true iff this represents a valid argument to the formatter.
          *
@@ -797,7 +797,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API bool canFormat() const { return !(isFallback() || isNullOperand()); }
+        bool canFormat() const { return !(isFallback() || isNullOperand()); }
         /**
          * Gets the fallback value of this placeholder, to be used in its place if an error occurs while
          * formatting it.
@@ -805,7 +805,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const UnicodeString& getFallback() const { return fallback; }
+        const UnicodeString& getFallback() const { return fallback; }
         /**
          * Returns the options of this placeholder. The result is the empty map if !isEvaluated().
          * @return A reference to an option map, capturing the options that were used
@@ -814,14 +814,14 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const FunctionOptions& options() const { return previousOptions; }
+        const FunctionOptions& options() const { return previousOptions; }
         /**
          * Returns the formatted output of this placeholder. The result is undefined if !isEvaluated().
          * @return          A fully formatted `FormattedPlaceholder`.
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API const FormattedValue& output() const { return formatted; }
+        const FormattedValue& output() const { return formatted; }
         /**
          * Move assignment operator:
          * The source FormattedPlaceholder will be left in a valid but undefined state.
@@ -837,7 +837,7 @@
          * @internal ICU 75 technology preview
          * @deprecated This API is for technology preview only.
          */
-        U_I18N_API FormattedPlaceholder(FormattedPlaceholder&& other) { *this = std::move(other); }
+        FormattedPlaceholder(FormattedPlaceholder&& other) { *this = std::move(other); }
         /**
          * Formats this as a string, using defaults.  If this is
          * either the null operand or is a fallback value, the return value is the result of formatting the
--- a/source/i18n/unicode/msgfmt.h
+++ b/source/i18n/unicode/msgfmt.h
@@ -493,7 +493,7 @@
      * @return this instance's UMessagePatternApostropheMode.
      * @stable ICU 4.8
      */
-    U_I18N_API UMessagePatternApostropheMode getApostropheMode() const {
+    UMessagePatternApostropheMode getApostropheMode() const {
         return msgPattern.getApostropheMode();
     }
 
--- a/source/i18n/unicode/numberformatter.h
+++ b/source/i18n/unicode/numberformatter.h
@@ -1588,7 +1588,7 @@
      * Check all members for errors.
      * @internal
      */
-    U_I18N_API bool copyErrorTo(UErrorCode &status) const {
+    bool copyErrorTo(UErrorCode &status) const {
         return notation.copyErrorTo(status) || precision.copyErrorTo(status) ||
                padder.copyErrorTo(status) || integerWidth.copyErrorTo(status) ||
                symbols.copyErrorTo(status) || scale.copyErrorTo(status) || usage.copyErrorTo(status) ||
--- a/source/i18n/unicode/numberrangeformatter.h
+++ b/source/i18n/unicode/numberrangeformatter.h
@@ -509,7 +509,7 @@
      *
      * @stable ICU 63
      */
-    U_I18N_API LocalizedNumberRangeFormatter() = default;
+    LocalizedNumberRangeFormatter() = default;
 
     /**
      * Returns a copy of this LocalizedNumberRangeFormatter.
