--- icu/source/tools/genren/Makefile.orig	2017-04-24 07:28:07.325453100 -0400
+++ icu/source/tools/genren/Makefile	2017-04-24 07:39:27.882902800 -0400
@@ -32,11 +32,11 @@ ifneq ($(patsubst %clean,,$(MAKECMDGOALS
 include $(BUILDDIR)/icudefs.mk
 endif
 
-COM=$(ICUDIR)/lib/libicuuc.$(SO)
-I18=$(ICUDIR)/lib/libicui18n.$(SO)
-LEX=$(ICUDIR)/lib/libiculx.$(SO)
+COM=$(ICUDIR)/bin/libicuuc.$(SO)
+I18=$(ICUDIR)/bin/libicui18n.$(SO)
+LEX=$(ICUDIR)/bin/libiculx.$(SO)
 DAT=$(ICUDIR)/stubdata/libicudata.$(SO)
-UIO=$(ICUDIR)/lib/libicuio.$(SO)
+UIO=$(ICUDIR)/bin/libicuio.$(SO)
 
 LIBS=$(COM) $(I18) $(LEX) $(UIO)
 
