diff -Naur icu-orig/source/tools/genren/Makefile icu/source/tools/genren/Makefile
--- icu-orig/source/tools/genren/Makefile	2016-09-21 09:07:26.000000000 +0300
+++ icu/source/tools/genren/Makefile	2018-03-01 10:09:05.338601800 +0300
@@ -32,11 +32,11 @@
 include $(BUILDDIR)/icudefs.mk
 endif
 
-COM=$(ICUDIR)/lib/libicuuc.$(SO)
-I18=$(ICUDIR)/lib/libicui18n.$(SO)
-LEX=$(ICUDIR)/lib/libiculx.$(SO)
+COM=$(ICUDIR)/bin/libicuuc.$(SO)
+I18=$(ICUDIR)/bin/libicui18n.$(SO)
+LEX=$(ICUDIR)/bin/libiculx.$(SO)
 DAT=$(ICUDIR)/stubdata/libicudata.$(SO)
-UIO=$(ICUDIR)/lib/libicuio.$(SO)
+UIO=$(ICUDIR)/bin/libicuio.$(SO)
 
 LIBS=$(COM) $(I18) $(LEX) $(UIO)
 
