--- a/source/i18n/unicode/tblcoll.h
+++ a/source/i18n/unicode/tblcoll.h
@@ -817,7 +817,7 @@
     * @param actualLocale the actual locale
     * @internal
     */
-    virtual void setLocales(const Locale& requestedLocale, const Locale& validLocale, const Locale& actualLocale) override;
+    U_I18N_API virtual void setLocales(const Locale& requestedLocale, const Locale& validLocale, const Locale& actualLocale) override;
 
 private:
     friend class CollationElementIterator;
