--- icu/source/config/mh-mingw.orig	2013-10-04 20:50:00.000000000 +0000
+++ icu/source/config/mh-mingw	2014-02-06 17:53:24.827138600 +0000
@@ -82,11 +82,11 @@
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
-#CURR_FULL_DIR=$(shell pwd -W)#M# for MSYS
-CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_FULL_DIR=$(shell $(SHELL) -c "pwd -W")#M# for MSYS
+#CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 # Current full path directory for use in source code in a -D compiler option.
-#CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell pwd -W))#M# for MSYS
-CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell $(SHELL) -c "pwd -W"))#M# for MSYS
+#CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 
 ## Compilation rules
 %.$(STATIC_O): $(srcdir)/%.c
--- icu/source/config/mh-mingw64.orig	2013-10-04 20:50:00.000000000 +0000
+++ icu/source/config/mh-mingw64	2014-02-06 17:53:10.783855300 +0000
@@ -82,11 +82,11 @@
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
-#CURR_FULL_DIR=$(shell pwd -W)#M# for MSYS
-CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_FULL_DIR=$(shell $(SHELL) -c "pwd -W")#M# for MSYS
+#CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 # Current full path directory for use in source code in a -D compiler option.
-#CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell pwd -W))#M# for MSYS
-CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell $(SHELL) -c "pwd -W"))#M# for MSYS
+#CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 
 ## Compilation rules
 %.$(STATIC_O): $(srcdir)/%.c
