diff -Naur icu-orig/source/config/mh-mingw icu/source/config/mh-mingw
--- icu-orig/source/config/mh-mingw	2016-09-13 23:29:47.000000000 +0300
+++ icu/source/config/mh-mingw	2018-03-01 10:09:19.300626300 +0300
@@ -96,11 +96,11 @@
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
-#CURR_FULL_DIR=$(shell pwd -W)#M# for MSYS
-CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_FULL_DIR=$(shell $(SHELL) -c "pwd -W")#M# for MSYS
+#CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 # Current full path directory for use in source code in a -D compiler option.
-#CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell pwd -W))#M# for MSYS
-CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell $(SHELL) -c "pwd -W"))#M# for MSYS
+#CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 
 ## Compilation rules
 %.$(STATIC_O): $(srcdir)/%.c
diff -Naur icu-orig/source/config/mh-mingw64 icu/source/config/mh-mingw64
--- icu-orig/source/config/mh-mingw64	2016-06-15 21:58:17.000000000 +0300
+++ icu/source/config/mh-mingw64	2018-03-01 10:09:19.300626300 +0300
@@ -96,11 +96,11 @@
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
-#CURR_FULL_DIR=$(shell pwd -W)#M# for MSYS
-CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_FULL_DIR=$(shell $(SHELL) -c "pwd -W")#M# for MSYS
+#CURR_FULL_DIR?=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 # Current full path directory for use in source code in a -D compiler option.
-#CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell pwd -W))#M# for MSYS
-CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell $(SHELL) -c "pwd -W"))#M# for MSYS
+#CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 
 ## Compilation rules
 %.$(STATIC_O): $(srcdir)/%.c
