diff -Naur icu-orig/source/configure.ac icu/source/configure.ac
--- icu-orig/source/configure.ac	2016-10-04 22:30:20.000000000 +0300
+++ icu/source/configure.ac	2018-03-01 10:09:26.414238800 +0300
@@ -1223,7 +1223,7 @@
 if test "$ICULIBSUFFIX" != ""
 then
     U_HAVE_LIB_SUFFIX=1
-    ICULIBSUFFIXCNAME=`echo _$ICULIBSUFFIX | sed 's/[^A-Za-z0-9_]/_/g'`
+    ICULIBSUFFIXCNAME=`echo $ICULIBSUFFIX | sed 's/[^A-Za-z0-9_]/_/g'`
     UCONFIG_CPPFLAGS="${UCONFIG_CPPFLAGS} -DU_HAVE_LIB_SUFFIX=1 -DU_LIB_SUFFIX_C_NAME=${ICULIBSUFFIXCNAME} "
 else
     U_HAVE_LIB_SUFFIX=0
diff -Naur icu-orig/source/tools/pkgdata/pkgdata.cpp icu/source/tools/pkgdata/pkgdata.cpp
--- icu-orig/source/tools/pkgdata/pkgdata.cpp	2018-03-01 10:09:26.024238100 +0300
+++ icu/source/tools/pkgdata/pkgdata.cpp	2018-03-01 10:09:26.414238800 +0300
@@ -1375,6 +1375,18 @@
             freeCmd = TRUE;
         }
 #if U_PLATFORM == U_PF_MINGW
+        fprintf(stderr, "genlib=%s targetdir=%s libfilenames=%s -o pkgdataflags=%s targetdir=%s libfilenames=%s objectfileandeverythingelse=%s %s%s %s %s",
+                pkgDataFlags[GENLIB],
+                targetDir,
+                libFileNames[LIB_FILE_MINGW],
+                pkgDataFlags[LDICUDTFLAGS],
+                targetDir,
+                libFileNames[LIB_FILE_VERSION_TMP],
+                objectFile,
+                pkgDataFlags[LD_SONAME],
+                pkgDataFlags[LD_SONAME][0] == 0 ? "" : libFileNames[LIB_FILE_VERSION_MAJOR],
+                pkgDataFlags[RPATH_FLAGS],
+                pkgDataFlags[BIR_FLAGS]);
         sprintf(cmd, "%s%s%s %s -o %s%s %s %s%s %s %s",
                 pkgDataFlags[GENLIB],
                 targetDir,
