--- icu/source/extra/scrptrun/Makefile.in.orig	2019-11-04 17:44:50.421177000 +0300
+++ icu/source/extra/scrptrun/Makefile.in	2019-11-04 17:44:58.858659600 +0300
@@ -12,9 +12,6 @@
 
 include $(top_builddir)/icudefs.mk
 
-## Platform-specific setup
-include @platform_make_fragment@
-
 ## Build directory information
 subdir = extra/scrptrun
 
--- icu/source/samples/layout/Makefile.in.orig	2019-11-04 17:43:46.697532200 +0300
+++ icu/source/samples/layout/Makefile.in	2019-11-04 17:43:53.651930000 +0300
@@ -13,9 +13,6 @@
 
 include $(top_builddir)/icudefs.mk
 
-## Platform-specific setup
-include @platform_make_fragment@
-
 ## Build directory information
 subdir = samples/layout
 
--- icu/source/samples/Makefile.in.orig	2019-11-04 17:43:20.443030600 +0300
+++ icu/source/samples/Makefile.in	2019-11-04 17:43:26.778392900 +0300
@@ -16,9 +16,6 @@
 ## Build directory information
 subdir = samples
 
-## Platform-specific setup
-include @platform_make_fragment@
-
 ## Files to remove for 'make clean'
 CLEANFILES = *~
 
