--- icu/source/config/mh-mingw64.orig	2021-01-31 17:25:18.521418000 -0800
+++ icu/source/config/mh-mingw64	2021-01-31 17:25:53.240156000 -0800
@@ -52,7 +52,7 @@
 #LINK.cc=      $(CXX) $(CXXFLAGS) $(LDFLAGS)
 
 ## Shared library options
-LD_SOOPTIONS= -Wl,-Bsymbolic
+LD_SOOPTIONS=
 
 ## Commands to make a shared library
 SHLIB.c=	$(CC) $(CFLAGS) $(LDFLAGS) -shared $(LD_SOOPTIONS) -Wl,--enable-auto-import -Wl,--out-implib=$(dir $@)$(notdir $(@:$(SO_TARGET_VERSION_MAJOR).$(SO)=))$(IMPORT_LIB_EXT)#M#
--- icu/source/config/mh-mingw.orig	2021-01-31 17:26:53.771392200 -0800
+++ icu/source/config/mh-mingw	2021-01-31 17:27:07.349690700 -0800
@@ -52,7 +52,7 @@
 #LINK.cc=      $(CXX) $(CXXFLAGS) $(LDFLAGS)
 
 ## Shared library options
-LD_SOOPTIONS= -Wl,-Bsymbolic
+LD_SOOPTIONS=
 
 ## Commands to make a shared library
 SHLIB.c=	$(CC) $(CFLAGS) $(LDFLAGS) -shared $(LD_SOOPTIONS) -Wl,--enable-auto-import -Wl,--out-implib=$(dir $@)$(notdir $(@:$(SO_TARGET_VERSION_MAJOR).$(SO)=))$(IMPORT_LIB_EXT)#M#
