--- a/source/config/mh-mingw.orig	2012-11-26 08:45:31 +0400
+++ b/source/config/mh-mingw	2012-11-26 08:45:35 +0400
@@ -63,11 +63,11 @@
 endif
 
 # Static library prefix and file extension
-LIBSICU = $(LIBPREFIX)$(STATIC_PREFIX)$(ICUPREFIX)
+LIBSICU = lib$(LIBPREFIX)$(STATIC_PREFIX)$(ICUPREFIX)
 A = a
 
 ## An import library is needed for z/OS and MSVC
-IMPORT_LIB_EXT = .lib
+IMPORT_LIB_EXT = .dll.a
 
 LIBPREFIX=
 
