From: Michele Locati <michele@locati.it>
Date: Fri, 10 May 2019 10:35:49 +0200
Subject: [PATCH] libtool: hide "library was moved"

--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -7956,7 +7956,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    #func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir

--- a/gettext-tools/examples/hello-c++-kde/admin/ltmain.sh
+++ b/gettext-tools/examples/hello-c++-kde/admin/ltmain.sh
@@ -1958,7 +1958,7 @@ EOF
 	# Find the relevant object directory and library name.
 	if test "X$installed" = Xyes; then
 	  if test ! -f "$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    $echo "$modename: warning: library \`$lib' was moved." 1>&2
+	    #$echo "$modename: warning: library \`$lib' was moved." 1>&2
 	    dir="$ladir"
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"

--- a/libtextstyle/build-aux/ltmain.sh
+++ b/libtextstyle/build-aux/ltmain.sh
@@ -7956,7 +7956,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    #func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir
