--- a/autogen.sh
+++ b/autogen.sh
@@ -145,7 +145,6 @@ if ! $skip_gnulib; then
     localename
     lock
     manywarnings
-    relocatable-lib-lgpl
     tsearch
     vasnprintf-posix
     vasnwprintf-posix
--- a/gettext-runtime/intl/Makefile.am
+++ b/gettext-runtime/intl/Makefile.am
@@ -33,7 +33,6 @@ MAINTAINERCLEANFILES =
 # Note that Automake's $(DEFAULT_INCLUDES) already contains
 # -I. -I$(srcdir) -I$(top_builddir).
 # -DBUILDING_LIBINTL: Change expansion of LIBINTL_SHLIB_EXPORTED macro.
-# -DBUILDING_DLL: Change expansion of RELOCATABLE_DLL_EXPORTED macro.
 # -DBUILDING_LIBRARY: Change expansion of SHLIB_EXPORTED macro. We're compiling
 #                     a library. Exporting the relevant symbols is desired.
 AM_CPPFLAGS = \
@@ -114,7 +113,7 @@ EXTRA_DIST += \
 #                       libintl_freelocale
 #   setlocale.c:        libintl_newlocale
 #                       libintl_setlocale
-#   relocatable.c:      libintl_set_relocation_prefix
+#   compat.c:           libintl_set_relocation_prefix
 #
 # The following files implement private API.
 #   gettextP.h:     dcigettext.c:       libintl_dcigettext
@@ -203,7 +202,8 @@ LIBINTLSOURCES = \
   setlocale.c \
   version.c \
   osdep.c \
-  intl-compat.c
+  intl-compat.c \
+  compat.c
 
 # We must not install the libintl.h/libintl.la files if we are on a
 # system which has the GNU gettext() function in its C library or in a
@@ -359,6 +359,8 @@ osdep.lo: $(srcdir)/osdep.c
 	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/osdep.c
 intl-compat.lo: $(srcdir)/intl-compat.c
 	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/intl-compat.c
+compat.lo: $(srcdir)/compat.c
+	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/compat.c
 
 # Dependencies.
 
@@ -392,6 +394,7 @@ setlocale.lo:        ../config.h $(srcdir)/gettextP.h libgnuintl.h $(srcdir)/gmo
 version.lo:          ../config.h libgnuintl.h
 osdep.lo:            ../config.h $(srcdir)/intl-exports.c $(srcdir)/os2compat.c
 intl-compat.lo:      ../config.h $(srcdir)/gettextP.h libgnuintl.h $(srcdir)/gmo.h $(srcdir)/loadinfo.h
+compat.lo:           ../config.h libgnuintl.h
 
 # Version information according to Woe32 conventions.
 EXTRA_DIST += libintl.rc
--- /dev/null
+++ b/gettext-runtime/intl/compat.c
@@ -0,0 +1,25 @@
+/* Symbols for backward compatibility.
+   Copyright (C) 2023 Free Software Foundation, Inc.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published by
+   the Free Software Foundation; either version 2.1 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.  */
+
+#include <config.h>
+
+/* Specification.  */
+#include "libgnuintl.h"
+
+void
+libintl_set_relocation_prefix (const char *orig_prefix, const char *curr_prefix)
+{
+}
diff --git a/gettext-runtime/intl/configure.ac b/gettext-runtime/intl/configure.ac
index 30a86065e..0c46ccb5e 100644
--- a/gettext-runtime/intl/configure.ac
+++ b/gettext-runtime/intl/configure.ac
@@ -42,10 +42,6 @@ gl_PROG_AR_RANLIB
 dnl Set preferences for the gnulib module 'threadlib'.
 gl_AVOID_WINPTHREAD
 
-dnl Set preferences for the gnulib module 'relocatable-lib-lgpl':
-dnl Enable relocatability unconditionally.
-enable_relocatable=yes
-
 dnl Early checks, assembled by gnulib.
 gl_EARLY
 
@@ -346,9 +342,6 @@ AH_VERBATIM([znoexport], [
 #define printf_frexp               _libintl_printf_frexp
 /* Symbols defined by gnulib module 'printf-frexpl'.  */
 #define printf_frexpl              _libintl_printf_frexpl
-/* Symbols defined by gnulib module 'relocatable-lib-lgpl'.  */
-#define relocate                   _libintl_relocate
-#define relocate2                  _libintl_relocate2
 /* Symbols defined by gnulib module 'setlocale-null'.  */
 #define setlocale_null             _libintl_setlocale_null
 #define setlocale_null_r           _libintl_setlocale_null_r
--- a/gnulib-local/modules/gettext-runtime-intl-misc
+++ b/gnulib-local/modules/gettext-runtime-intl-misc
@@ -11,12 +11,6 @@ AC_REQUIRE([gl_LOCALE_H_DEFAULTS])
 AC_REQUIRE([gt_INTL_THREAD_LOCALE_NAME])
 
 Makefile.am:
-# Parameterization of the 'relocatable-lib-lgpl' module.
-AM_CPPFLAGS += \
-  -DIN_LIBRARY -DINSTALLDIR=\"$(libdir)\" -DNO_XMALLOC -DBUILDING_DLL \
-  -Dset_relocation_prefix=libintl_set_relocation_prefix \
-  -DDEPENDS_ON_LIBICONV=1
-
 # Enable more warning options in this directory.
 AM_CFLAGS += @WARN_CFLAGS@
 
