From: Michele Locati <michele@locati.it>
Date: Fri, 10 May 2019 11:16:55 +0200
Subject: [PATCH] libtool: hide "remember to run"

--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -4543,7 +4543,7 @@ func_mode_install ()
       func_execute_cmds "$old_postinstall_cmds" 'exit $?'
     done
 
-    test -n "$future_libdirs" && \
+    false && test -n "$future_libdirs" && \
       func_warning "remember to run '$progname --finish$future_libdirs'"
 
     if test -n "$current_libdirs"; then

--- a/libtextstyle/build-aux/ltmain.sh
+++ b/libtextstyle/build-aux/ltmain.sh
@@ -4543,7 +4543,7 @@ func_mode_install ()
       func_execute_cmds "$old_postinstall_cmds" 'exit $?'
     done
 
-    test -n "$future_libdirs" && \
+    false && test -n "$future_libdirs" && \
       func_warning "remember to run '$progname --finish$future_libdirs'"
 
     if test -n "$current_libdirs"; then
