diff -urN gettext-0.21.1/gettext-tools/libgettextpo/textstyle.in.h.orig2 gettext-0.21.1/gettext-tools/libgettextpo/textstyle.in.h
--- gettext-0.21.1/gettext-tools/libgettextpo/textstyle.in.h.orig2	2023-01-20 18:53:23.229044300 +0100
+++ gettext-0.21.1/gettext-tools/libgettextpo/textstyle.in.h	2023-01-20 19:12:20.677728700 +0100
@@ -68,6 +68,10 @@
 # endif
 #endif
 
+#ifdef _WIN32
+# define fsync _commit
+#endif
+
 /* ----------------------------- From ostream.h ----------------------------- */
 
 /* Describes the scope of a flush operation.  */
