--- gettext-tools/gnulib-lib/libxml/parser.c.orig	2013-01-18 10:24:37 +0400
+++ gettext-tools/gnulib-lib/libxml/parser.c	2013-01-18 10:36:35 +0400
@@ -42,6 +42,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
+#ifdef HAVE_CTYPE_H
+#include <ctype.h>
+#endif
 #include <libxml/xmlmemory.h>
 #include <libxml/threads.h>
 #include <libxml/globals.h>
@@ -61,9 +64,6 @@
 #include <libxml/xmlschemastypes.h>
 #include <libxml/relaxng.h>
 #endif
-#ifdef HAVE_CTYPE_H
-#include <ctype.h>
-#endif
 #ifdef HAVE_STDLIB_H
 #include <stdlib.h>
 #endif
--- gettext-tools/gnulib-lib/libxml/threads.c.orig	2013-01-18 10:38:24 +0400
+++ gettext-tools/gnulib-lib/libxml/threads.c	2013-01-18 10:41:11 +0400
@@ -12,15 +12,16 @@
 
 #include <string.h>
 
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
 #include <libxml/threads.h>
 #include <libxml/globals.h>
 
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
-#ifdef HAVE_UNISTD_H
-#include <unistd.h>
-#endif
 #ifdef HAVE_STDLIB_H
 #include <stdlib.h>
 #endif
