--- gettext-0.19.8.1/gettext-runtime/gnulib-m4/asm-underscore.m4.orig	2017-01-17 11:42:29.417883700 +0300
+++ gettext-0.19.8.1/gettext-runtime/gnulib-m4/asm-underscore.m4	2017-01-17 11:42:37.039697600 +0300
@@ -29,7 +29,7 @@
 EOF
      # Look for the assembly language name in the .s file.
      AC_TRY_COMMAND(${CC-cc} $CFLAGS $CPPFLAGS $gl_c_asm_opt conftest.c) >/dev/null 2>&1
-     if LC_ALL=C grep -E '(^|[^a-zA-Z0-9_])_foo([^a-zA-Z0-9_]|$)' conftest.$gl_asmext >/dev/null; then
+     if LC_ALL=C grep -E '(^|[[^a-zA-Z0-9_]])_foo([[^a-zA-Z0-9_]]|$)' conftest.$gl_asmext >/dev/null; then
        gl_cv_prog_as_underscore=yes
      else
        gl_cv_prog_as_underscore=no
--- gettext-0.19.8.1/gettext-tools/gnulib-m4/asm-underscore.m4.orig	2017-01-17 11:41:49.073406600 +0300
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/asm-underscore.m4	2017-01-17 11:42:07.503241900 +0300
@@ -29,7 +29,7 @@
 EOF
      # Look for the assembly language name in the .s file.
      AC_TRY_COMMAND(${CC-cc} $CFLAGS $CPPFLAGS $gl_c_asm_opt conftest.c) >/dev/null 2>&1
-     if LC_ALL=C grep -E '(^|[^a-zA-Z0-9_])_foo([^a-zA-Z0-9_]|$)' conftest.$gl_asmext >/dev/null; then
+     if LC_ALL=C grep -E '(^|[[^a-zA-Z0-9_]])_foo([[^a-zA-Z0-9_]]|$)' conftest.$gl_asmext >/dev/null; then
        gl_cv_prog_as_underscore=yes
      else
        gl_cv_prog_as_underscore=no
--- gettext-0.19.8.1/gettext-runtime/configure.orig	2017-01-17 11:42:29.417883700 +0300
+++ gettext-0.19.8.1/gettext-runtime/configure	2017-01-17 11:42:37.039697600 +0300
@@ -24606,7 +24606,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; } >/dev/null 2>&1
-     if LC_ALL=C grep -E '(^|^a-zA-Z0-9_)_foo(^a-zA-Z0-9_|$)' conftest.$gl_asmext >/dev/null; then
+     if LC_ALL=C grep -E '(^|[^a-zA-Z0-9_])_foo(^[a-zA-Z0-9_]|$)' conftest.$gl_asmext >/dev/null; then
        gl_cv_prog_as_underscore=yes
      else
        gl_cv_prog_as_underscore=no
--- gettext-0.19.8.1/gettext-tools/configure.orig	2017-01-17 11:42:29.417883700 +0300
+++ gettext-0.19.8.1/gettext-tools/configure	2017-01-17 11:42:37.039697600 +0300
@@ -32286,7 +32286,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; } >/dev/null 2>&1
-     if LC_ALL=C grep -E '(^|^a-zA-Z0-9_)_foo(^a-zA-Z0-9_|$)' conftest.$gl_asmext >/dev/null; then
+     if LC_ALL=C grep -E '(^|[^a-zA-Z0-9_])_foo(^[a-zA-Z0-9_]|$)' conftest.$gl_asmext >/dev/null; then
        gl_cv_prog_as_underscore=yes
      else
        gl_cv_prog_as_underscore=no
