From 60d853ccaa4c926430e536b36d8b0f1be19360c1 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Sat, 26 May 2018 19:28:17 -0400
Subject: [PATCH] Use LF as newline in envsubst

---
 gettext-runtime/src/envsubst.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gettext-runtime/src/envsubst.c b/gettext-runtime/src/envsubst.c
index 941452b03..5bc482a75 100644
--- a/gettext-runtime/src/envsubst.c
+++ b/gettext-runtime/src/envsubst.c
@@ -27,6 +27,8 @@
 #include <string.h>
 #include <locale.h>
 
+#include <fcntl.h>
+
 #include "closeout.h"
 #include "error.h"
 #include "progname.h"
@@ -282,7 +282,10 @@ static void
 print_variable (const char *var_ptr, size_t var_len)
 {
   fwrite (var_ptr, var_len, 1, stdout);
+  setmode(fileno(stdout),O_BINARY);
   putchar ('\n');
+  fflush(stdout);
+  setmode(fileno(stdout),O_TEXT);
 }
 
 /* Print the variables contained in STRING to stdout, each one followed by a
-- 
2.16.2.windows.1

