for relocate()
--- gettext-0.19.8.1/gettext-tools/gnulib-lib/Makefile.am.orig	2019-05-11 22:20:53.069987300 +0200
+++ gettext-0.19.8.1/gettext-tools/gnulib-lib/Makefile.am	2019-05-11 22:21:54.510209600 +0200
@@ -48,6 +48,8 @@
 # Parametrization of the 'relocatable' module.
 AM_CPPFLAGS += -DDEPENDS_ON_LIBICONV=1 -DDEPENDS_ON_LIBINTL=1
 
+AM_CPPFLAGS += -DINSTALLDIR=\"$(bindir)\"
+
 # Parametrization of the 'libxml' module:
 # When building a shared library, don't export the variables
 # xmlMalloc, xmlMallocAtomic, xmlRealloc, xmlFree, xmlMemStrdup.
