diff --git a/lib/supersede.c b/lib/supersede.c
index a03cc6d..a3dfa4f 100644
--- a/lib/supersede.c
+++ b/lib/supersede.c
@@ -80,10 +80,12 @@ open_supersede (const char *filename, int flags, mode_t mode,
   int fd;
   /* Extra flags for existing devices.  */
   int extra_flags =
-    #if defined __sun
+    #if defined __sun || (defined _WIN32 && !defined __CYGWIN__)
     /* open ("/dev/null", O_TRUNC | O_WRONLY) fails with error EINVAL on Solaris
-       zones.  See <https://www.illumos.org/issues/13035>.  As a workaround, add
-       the O_CREAT flag, although it ought not to be necessary.  */
+       zones.  See <https://www.illumos.org/issues/13035>.
+       Likewise for open ("NUL", O_TRUNC | O_RDWR) on native Windows.
+       As a workaround, add the O_CREAT flag, although it ought not to be
+       necessary.  */
     O_CREAT;
     #else
     0;
@@ -207,8 +209,8 @@ open_supersede (const char *filename, int flags, mode_t mode,
               action->final_rename_temp = NULL;
               action->final_rename_dest = NULL;
             }
-          #if defined __sun
-          /* Work around <https://www.illumos.org/issues/13035>.  */
+          #if defined __sun || (defined _WIN32 && !defined __CYGWIN__)
+          /* See the comment regarding extra_flags, above.  */
           else if (errno == EINVAL)
             {
               struct stat statbuf;
-- 
cgit v1.1

