--- openshadinglanguage/src/liboslexec/llvm_ops.cpp.orig	2013-11-25 18:41:29.000000000 +0000
+++ openshadinglanguage/src/liboslexec/llvm_ops.cpp	2014-01-05 22:30:40.121825000 +0000
@@ -109,7 +109,7 @@
 using OIIO::safe_acosf;
 using OIIO::isinf;
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__GNUC__)
 using OIIO::roundf;
 using OIIO::truncf;
 using OIIO::expm1f;
@@ -846,7 +846,6 @@
                                 int vectype)
 {
     ShaderGlobals *sg = (ShaderGlobals *)sg_;
-    ShadingContext *ctx = (ShadingContext *)sg->context;
     RendererServices *rend = sg->renderer;
     if (rend->transform_points (sg, USTR(from), USTR(to), sg->time,
                                 (const Vec3 *)Pin, (Vec3 *)Pout, 1,
