--- OpenShadingLanguage-Release-1.4.0/src/liboslexec/CMakeLists.txt.orig	2014-01-06 01:46:24.519674100 +0000
+++ OpenShadingLanguage-Release-1.4.0/src/liboslexec/CMakeLists.txt	2014-01-06 01:46:49.636363500 +0000
@@ -68,6 +68,20 @@
       LIST (APPEND LLVM_COMPILE_FLAGS "-D${def}")
     endforeach()
 
+    # From http://llvm.org/docs/CMake.html#embedding-llvm-in-your-project
+    find_package(LLVM)
+
+    if( NOT LLVM_FOUND )
+        message(FATAL_ERROR "LLVM package can't be found. Set CMAKE_PREFIX_PATH variable to LLVM's installation prefix.")
+    endif()
+
+    include_directories( ${LLVM_INCLUDE_DIRS} )
+    link_directories( ${LLVM_LIBRARY_DIRS} )
+
+    llvm_map_components_to_libraries(REQ_LLVM_LIBRARIES jit native)
+
+    target_link_libraries(${REQ_LLVM_LIBRARIES})
+
     # First try looking in their build (clang++ first, then llvm-g++)
     FIND_PROGRAM(LLVM_BC_GENERATOR NAMES "clang++" "llvm-g++" PATHS "${LLVM_DIRECTORY}/bin" NO_DEFAULT_PATH NO_CMAKE_SYSTEM_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH)
 
