--- openshadinglanguage/src/liboslcomp/oslcomp_pvt.h.orig	2014-04-16 17:05:16.783394800 +0000
+++ openshadinglanguage/src/liboslcomp/oslcomp_pvt.h	2014-04-16 17:05:38.243394800 +0000
@@ -107,7 +107,7 @@
     bool error_encountered () const { return m_err; }
 
     /// Has a shader already been defined?
-    bool shader_is_defined () const { return m_shader; }
+    bool shader_is_defined () const { return m_shader ? true : false; }
 
     /// Define the shader we're compiling with the given AST root.
     ///
