diff -Naur openshadinglanguage-orig/src/liboslcomp/symtab.cpp openshadinglanguage/src/liboslcomp/symtab.cpp
--- openshadinglanguage-orig/src/liboslcomp/symtab.cpp	2014-08-31 21:12:36.509200000 +0400
+++ openshadinglanguage/src/liboslcomp/symtab.cpp	2014-08-31 21:28:41.286000000 +0400
@@ -343,7 +343,7 @@
                 std::cout << " (" << s->name()
                           << " in scope " << s->scope() << ")";
             std::cout << " :\n";
-            for (size_t i = 0;  i < s->numfields();  ++i) {
+            for (int i = 0;  i < s->numfields();  ++i) {
                 const StructSpec::FieldSpec & f (s->field(i));
                 std::cout << "\t" << f.name << " : "
                           << f.type.string() << "\n";
diff -Naur openshadinglanguage-orig/src/liboslexec/shadingsys.cpp openshadinglanguage/src/liboslexec/shadingsys.cpp
--- openshadinglanguage-orig/src/liboslexec/shadingsys.cpp	2014-08-31 21:12:36.587200000 +0400
+++ openshadinglanguage/src/liboslexec/shadingsys.cpp	2014-08-31 21:31:52.245600000 +0400
@@ -1647,7 +1647,7 @@
         // as connections between their respective fields.
         StructSpec *srcstruct = srccon.type.structspec();
         StructSpec *dststruct = dstcon.type.structspec();
-        for (size_t i = 0;  i < srcstruct->numfields();  ++i) {
+        for (int i = 0;  i < srcstruct->numfields();  ++i) {
             std::string s = Strutil::format("%s.%s", srcparam, srcstruct->field(i).name.c_str());
             std::string d = Strutil::format("%s.%s", dstparam, dststruct->field(i).name.c_str());
             ConnectShaders (srclayer, s.c_str(), dstlayer, d.c_str());
diff -Naur openshadinglanguage-orig/src/liboslexec/typespec.cpp openshadinglanguage/src/liboslexec/typespec.cpp
--- openshadinglanguage-orig/src/liboslexec/typespec.cpp	2014-08-31 21:12:36.587200000 +0400
+++ openshadinglanguage/src/liboslexec/typespec.cpp	2014-08-31 21:27:42.218600000 +0400
@@ -101,7 +101,7 @@
     ASSERT (a && b);
     if (a->numfields() != b->numfields())
         return false;
-    for (size_t i = 0;  i < a->numfields();  ++i)
+    for (int i = 0;  i < a->numfields();  ++i)
         if (! equivalent (a->field(i).type, b->field(i).type))
             return false;
     return true;
diff -Naur openshadinglanguage-orig/src/liboslexec/lpeparse.cpp openshadinglanguage/src/liboslexec/lpeparse.cpp
--- openshadinglanguage-orig/src/liboslexec/lpeparse.cpp	2014-08-31 21:44:27.475800000 +0400
+++ openshadinglanguage/src/liboslexec/lpeparse.cpp	2014-08-31 22:42:14.737400000 +0400
@@ -72,13 +72,13 @@
     m_minus_stop.insert(Labels::STOP);
 
     if (user_events)
-      for (int i = 0; i < user_events->size(); ++i)
+      for (size_t i = 0; i < user_events->size(); ++i)
       {
           m_label_position[(*user_events)[i]] = 0;
           m_basic_labels.insert((*user_events)[i]);
       }
    if (user_scatterings)
-      for (int i = 0; i < user_scatterings->size(); ++i)
+      for (size_t i = 0; i < user_scatterings->size(); ++i)
       {
           m_label_position[(*user_scatterings)[i]] = 1;
           m_basic_labels.insert((*user_scatterings)[i]);
