--- postgresql-9.4.1/configure.in.orig	2015-02-10 21:59:27.222800000 +0300
+++ postgresql-9.4.1/configure.in	2015-02-10 22:00:30.637800000 +0300
@@ -691,16 +691,6 @@
 #
 PGAC_ARG_BOOL(with, readline, yes,
               [do not use GNU Readline nor BSD Libedit for editing])
-# readline on MinGW has problems with backslashes in psql and other bugs.
-# This is particularly a problem with non-US code pages.
-# Therefore disable its use until we understand the cause. 2004-07-20
-if test "$PORTNAME" = "win32"; then
-  if test "$with_readline" = yes; then
-    AC_MSG_WARN([*** Readline does not work on MinGW --- disabling])
-    with_readline=no
-  fi
-fi
-
 
 #
 # Prefer libedit
