--- ../LIEF-0.16.5/./api/python/backend/setup.py	2025-04-19 05:12:25.000000000 -0300
+++ ./api/python/backend/setup.py	2025-04-30 13:38:39.587801139 -0300
@@ -46,7 +46,7 @@
 
 def _fix_env():
     config = get_config()
-    if sys.platform.startswith("win"):
+    if sysconfig.get_platform().startswith("win"):
         if config.build.ninja:
             from setuptools import msvc
             is64 = sys.maxsize > 2**32
