diff -Naur qtwebkit-5.212.0-alpha4-orig/Source/cmake/OptionsQt.cmake qtwebkit-5.212.0-alpha4/Source/cmake/OptionsQt.cmake
--- qtwebkit-5.212.0-alpha4-orig/Source/cmake/OptionsQt.cmake	2019-06-26 19:25:02.000000000 +0300
+++ qtwebkit-5.212.0-alpha4/Source/cmake/OptionsQt.cmake	2020-03-04 20:16:37.000000000 +0300
@@ -184,8 +184,10 @@
 
     set(CMAKE_SHARED_LIBRARY_PREFIX "")
     set(CMAKE_SHARED_MODULE_PREFIX "")
+    if(NOT MINGW)
     # QMake doesn't treat import libraries as a separate product kind
     set(CMAKE_IMPORT_LIBRARY_SUFFIX "${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    endif()
 endif ()
 
 WEBKIT_OPTION_BEGIN()