--- lp_presolve.c.orig	2018-08-11 10:51:31.973568400 +0900
+++ lp_presolve.c	2018-08-11 10:51:37.302560900 +0900
@@ -341,7 +341,7 @@
 /* Presolve routines for tightening the model                                    */
 /* ----------------------------------------------------------------------------- */
 
-INLINE REAL presolve_roundrhs(lprec *lp, REAL value, MYBOOL isGE)
+/*INLINE*/ REAL presolve_roundrhs(lprec *lp, REAL value, MYBOOL isGE)
 {
 #ifdef DoPresolveRounding
   REAL eps = PRESOLVE_EPSVALUE*1000,
