diff -bur MediaInfo-orig/Project/GNU/CLI/Makefile.am MediaInfo/Project/GNU/CLI/Makefile.am
--- MediaInfo-orig/Project/GNU/CLI/Makefile.am	2024-03-02 23:45:05.902154500 -0700
+++ MediaInfo/Project/GNU/CLI/Makefile.am	2024-03-02 23:46:28.384523800 -0700
@@ -7,6 +7,19 @@
     ../../../Source/CLI/Help.cpp \
     ../../../Source/Common/Core.cpp
 
+# Define the .rc file
+RC_FILE = ../../../Project/MSVC2019/CLI/MediaInfo_CLI.rc
+
+# Define the .o file that will be generated from the .rc file
+RC_OBJ = CLI_Main_RC.o
+
+# Add the .o file to the list of objects that the program depends on
+mediainfo_LDADD = $(RC_OBJ)
+
+# Rule to generate the .o file from the .rc file
+$(RC_OBJ): $(RC_FILE)
+	windres $< -o $@
+
 #mediainfo_LDFLAGS     = -no-undefined -version-info 0:0:0
 AM_CXXFLAGS = -std=c++11
 
diff -bur MediaInfo-orig/Project/GNU/GUI/Makefile.am MediaInfo/Project/GNU/GUI/Makefile.am
--- MediaInfo-orig/Project/GNU/GUI/Makefile.am	2024-03-02 23:45:05.901154800 -0700
+++ MediaInfo/Project/GNU/GUI/Makefile.am	2024-03-02 23:45:21.398862600 -0700
@@ -20,6 +20,19 @@
     ../../../Source/CLI/Help.cpp \
     ../../../Source/Common/Core.cpp
 
+# Define the .rc file
+GUI_RC_FILE = ../../../Project/MSVC2019/GUI/MediaInfo_GUI.rc
+
+# Define the .o file that will be generated from the .rc file
+GUI_RC_OBJ = GUI_Main_RC.o
+
+# Add the .o file to the list of objects that the program depends on
+mediainfo_gui_LDADD = $(GUI_RC_OBJ)
+
+# Rule to generate the .o file from the .rc file
+$(GUI_RC_OBJ): $(GUI_RC_FILE)
+	windres $< -o $@
+
 #mediainfo_gui_LDFLAGS     = -no-undefined -version-info 0:0:0
 
 AM_CXXFLAGS = -std=c++11
