--- muparser-2.2.6/include/muParserFixes.h.orig	2018-11-09 05:56:09.885150900 +0300
+++ muparser-2.2.6/include/muParserFixes.h	2018-11-09 05:56:33.250487300 +0300
@@ -81,7 +81,7 @@
  * MUPARSER_LOCAL is used for non-api symbols. */
 
 #ifndef MUPARSER_STATIC /* defined if muParser is compiled as a DLL */
-#ifdef MUPARSERLIB_EXPORTS /* defined if we are building the muParser DLL (instead of using it) */
+#if defined(MUPARSERLIB_EXPORTS) || defined(muparser_EXPORTS) /* defined if we are building the muParser DLL (instead of using it) */
 #define API_EXPORT_CXX MUPARSER_HELPER_DLL_EXPORT
 #else
 #define API_EXPORT_CXX MUPARSER_HELPER_DLL_IMPORT
