diff --unified --recursive --text --color eog-3.36.2_ori/src/meson.build eog-3.36.2/src/meson.build
--- eog-3.36.2_ori/src/meson.build	2020-05-03 11:52:05.883786800 +0800
+++ eog-3.36.2/src/meson.build	2020-05-03 11:52:40.257028200 +0800
@@ -203,6 +203,7 @@
   include_directories: top_inc,
   dependencies: deps,
   c_args: cflags,
+  gui_app: true,
   install: true,
   install_rpath: eog_prefix / eog_pkglibdir,
 )
