diff -urN a/setup.py b/setup.py
--- a/setup.py	2014-01-22 20:57:26.562219619 +0000
+++ b/setup.py	2014-01-22 20:57:27.105561368 +0000
@@ -961,7 +961,8 @@
                 for p in ['msvcrtmodule.c']]) )
 
             exts.append( Extension('_msi', [os.path.join(pc_srcdir, p)
-                for p in ['_msi.c']]) )
+                for p in ['_msi.c']],
+                libraries=['msi','cabinet','rpcrt4']) ) # To link with lib(msi|cabinet|rpcrt4).a
 
             exts.append( Extension('_subprocess', [os.path.join(pc_srcdir, p)
                 for p in ['_subprocess.c']]) )
