diff -urN a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2014-01-22 20:57:27.465566949 +0000
+++ b/Makefile.pre.in	2014-01-22 20:57:27.892240209 +0000
@@ -27,6 +27,7 @@
 VERSION=	@VERSION@
 srcdir=		@srcdir@
 VPATH=		@srcdir@
+MSYSVPATH=	@MSYSVPATH@
 abs_srcdir=	@abs_srcdir@
 abs_builddir=	@abs_builddir@
 build=		@build@
@@ -585,7 +586,7 @@
 		-DPREFIX='"$(prefix)"' \
 		-DEXEC_PREFIX='"$(exec_prefix)"' \
 		-DVERSION='"$(VERSION)"' \
-		-DVPATH='"$(VPATH)"' \
+		-DVPATH='"$(MSYSVPATH)"' \
 		-o $@ $(srcdir)/Modules/getpath.c
 
 # default sys.path calculations for windows platforms
diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2014-01-22 20:57:27.405566020 +0000
+++ b/configure.ac	2014-01-22 20:57:27.892240209 +0000
@@ -504,6 +504,17 @@
 AC_MSG_RESULT([$DELIM])
 AC_SUBST(DELIM)
 
+# MSYS make uses a 'virtual' VPATH, but getpath.c uses
+# GetModuleFileNameW (replacing \ with /). This allows the user to
+# define the 'actual 'real' value. Note, it should contain / not \,
+# (/ is returned by "pwd -W")
+AC_ARG_VAR(MSYSVPATH,
+           For MSYS, allows specifying the real VPATH. Use / not \)
+if test -z "$MSYSVPATH"; then
+    MSYSVPATH=$srcdir
+fi
+AC_SUBST(MSYSVPATH)
+
 AC_MSG_CHECKING([for init system calls])
 AC_SUBST(INITSYS)
 case $host in
