diff -urN a/Lib/distutils/cygwinccompiler.py b/Lib/distutils/cygwinccompiler.py
--- a/Lib/distutils/cygwinccompiler.py	2013-05-17 23:57:33.208405982 +0100
+++ b/Lib/distutils/cygwinccompiler.py	2013-05-17 23:57:33.479409122 +0100
@@ -442,7 +442,8 @@
     from distutils.spawn import find_executable
     import re
 
-    gcc_exe = find_executable('gcc')
+    gcc_exe = os.environ.get('CC') or find_executable('gcc')
+    ld_exe = os.environ.get('LD') or find_executable('ld')
     if gcc_exe:
         out = os.popen(gcc_exe + ' -dumpversion','r')
         out_string = out.read()
@@ -452,9 +453,11 @@
             gcc_version = LooseVersion(result.group(1))
         else:
             gcc_version = None
+        out = os.popen(gcc_exe + ' --print-prog-name ld','r')
+        ld_exe = out.read().decode('ascii').split()[0]
+        out.close()
     else:
         gcc_version = None
-    ld_exe = find_executable('ld')
     if ld_exe:
         out = os.popen(ld_exe + ' -v','r')
         out_string = out.read()
@@ -466,7 +469,7 @@
             ld_version = None
     else:
         ld_version = None
-    dllwrap_exe = find_executable('dllwrap')
+    dllwrap_exe = os.environ.get('DLLWRAP') or find_executable('dllwrap')
     if dllwrap_exe:
         out = os.popen(dllwrap_exe + ' --version','r')
         out_string = out.read()
diff -urN a/Modules/Setup.config.in b/Modules/Setup.config.in
--- a/Modules/Setup.config.in	2013-05-17 23:57:33.073404419 +0100
+++ b/Modules/Setup.config.in	2013-05-17 23:57:33.480409133 +0100
@@ -18,6 +18,8 @@
 @BUILDIN_WIN32_MODULE@operator operator.c	# operator.add() and similar goodies
 @BUILDIN_WIN32_MODULE@_locale _localemodule.c	# -lintl
 @BUILDIN_WIN32_MODULE@_winreg ../PC/_winreg.c
+@BUILDIN_WIN32_MODULE@time timemodule.c
+@BUILDIN_WIN32_MODULE@msvcrt ../PC/msvcrtmodule.c
 
 
 # The rest of the modules previously listed in this file are built
