diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-05-17 23:57:37.265452766 +0100
+++ b/setup.py	2013-05-17 23:57:37.638457042 +0100
@@ -1324,7 +1324,7 @@
 
         dbm_order = ['gdbm']
         # The standard Unix dbm module:
-        if host_platform not in ['cygwin']:
+        if host_platform not in ['cygwin', 'mingw', 'win32']:
             config_args = [arg.strip("'")
                            for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
             dbm_args = [arg for arg in config_args
