diff -urN a/setup.py b/setup.py
--- a/setup.py	2014-01-22 20:57:32.952318468 +0000
+++ b/setup.py	2014-01-22 20:57:33.438992662 +0000
@@ -1326,7 +1326,7 @@
 
         dbm_order = ['gdbm']
         # The standard Unix dbm module:
-        if host_platform not in ['cygwin']:
+        if host_platform not in ['cygwin', 'mingw', 'win32']:
             config_args = [arg.strip("'")
                            for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
             dbm_args = [arg for arg in config_args
