diff -urN a/Lib/compileall.py b/Lib/compileall.py
--- a/Lib/compileall.py	2014-01-22 20:57:37.069048823 +0000
+++ b/Lib/compileall.py	2014-01-22 20:57:37.429054390 +0000
@@ -42,6 +42,8 @@
     success = 1
     for name in names:
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
