diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-05-17 23:57:42.156508520 +0100
+++ b/setup.py	2013-05-17 23:57:42.538512843 +0100
@@ -533,15 +533,16 @@
         inc_dirs = self.compiler.include_dirs[:]
         lib_dirs = self.compiler.library_dirs[:]
         if not cross_compiling:
-            for d in (
-                '/usr/include',
-                ):
-                add_dir_to_list(inc_dirs, d)
-            for d in (
-                '/lib64', '/usr/lib64',
-                '/lib', '/usr/lib',
-                ):
-                add_dir_to_list(lib_dirs, d)
+            if not (sys.version.find('GCC') and host_platform in ['mingw', 'win32']):
+                for d in (
+                    '/usr/include',
+                    ):
+                    add_dir_to_list(inc_dirs, d)
+                for d in (
+                    '/lib64', '/usr/lib64',
+                    '/lib', '/usr/lib',
+                    ):
+                    add_dir_to_list(lib_dirs, d)
         exts = []
         missing = []
 
