diff -urN a/setup.py b/setup.py
--- a/setup.py	2014-10-11 14:52:21.109759000 +0100
+++ b/setup.py	2014-10-11 14:52:23.074871400 +0100
@@ -1318,7 +1318,7 @@
             missing.append('gdbm')
 
         # Unix-only modules
-        if host_platform not in ['win32']:
+        if not host_platform.startswith(('mingw', 'win')):
             # Steen Lumholt's termios module
             exts.append( Extension('termios', ['termios.c']) )
             # Jeremy Hylton's rlimit interface
