diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-06-03 08:13:06.102400000 +0400
+++ b/setup.py	2014-06-03 08:13:13.886800000 +0400
@@ -1316,7 +1316,7 @@
             missing.append('gdbm')
 
         # Unix-only modules
-        if host_platform not in ['win32']:
+        if not host_platform.startswith(('mingw', 'win')):
             # Steen Lumholt's termios module
             exts.append( Extension('termios', ['termios.c']) )
             # Jeremy Hylton's rlimit interface
