diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-02-18 09:32:07.191800000 +0400
+++ b/setup.py	2014-02-18 09:32:13.977800000 +0400
@@ -1317,7 +1317,7 @@
             missing.append('gdbm')
 
         # Unix-only modules
-        if host_platform not in ['win32']:
+        if not host_platform.startswith(('mingw', 'win')):
             # Steen Lumholt's termios module
             exts.append( Extension('termios', ['termios.c']) )
             # Jeremy Hylton's rlimit interface
