diff -urN a/Lib/distutils/cygwinccompiler.py b/Lib/distutils/cygwinccompiler.py
--- a/Lib/distutils/cygwinccompiler.py	2014-10-11 14:52:58.827916400 +0100
+++ b/Lib/distutils/cygwinccompiler.py	2014-10-11 14:53:00.844031700 +0100
@@ -85,6 +85,7 @@
     obj_extension = ".o"
     static_lib_extension = ".a"
     shared_lib_extension = ".dll"
+    dylib_lib_extension = ".dll.a"
     static_lib_format = "lib%s%s"
     shared_lib_format = "%s%s"
     exe_extension = ".exe"
