diff -urN a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2013-05-17 23:57:44.206531683 +0100
+++ b/Makefile.pre.in	2013-05-17 23:57:44.578535872 +0100
@@ -885,6 +885,7 @@
 	if test -f $(LDLIBRARY); then \
 		if test -n "$(DLLLIBRARY)" ; then \
 			$(INSTALL_SHARED) $(DLLLIBRARY) $(DESTDIR)$(BINDIR); \
+			mkdir -p $(DESTDIR)$(LIBPL); $(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBPL); \
 		else \
 			$(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBDIR)/$(INSTSONAME); \
 			if test $(LDLIBRARY) != $(INSTSONAME); then \
