diff -urN a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2014-01-22 20:57:40.942442030 +0000
+++ b/Makefile.pre.in	2014-01-22 20:57:41.375782060 +0000
@@ -890,6 +890,7 @@
 	if test -f $(LDLIBRARY); then \
 		if test -n "$(DLLLIBRARY)" ; then \
 			$(INSTALL_SHARED) $(DLLLIBRARY) $(DESTDIR)$(BINDIR); \
+			mkdir -p $(DESTDIR)$(LIBPL); $(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBPL); \
 		else \
 			$(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBDIR)/$(INSTSONAME); \
 			if test $(LDLIBRARY) != $(INSTSONAME); then \
