diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2014-02-18 09:32:43.259000000 +0400
+++ b/configure.ac	2014-02-18 09:32:43.383800000 +0400
@@ -1658,8 +1658,8 @@
 AC_HEADER_STDC
 AC_CHECK_HEADERS(asm/types.h conio.h curses.h direct.h dlfcn.h errno.h \
 fcntl.h grp.h \
-ieeefp.h io.h langinfo.h libintl.h ncurses.h poll.h process.h pthread.h \
-shadow.h signal.h stdint.h stropts.h termios.h thread.h \
+ieeefp.h io.h langinfo.h libintl.h ncurses.h poll.h process.h \
+shadow.h signal.h stdint.h stropts.h termios.h \
 unistd.h utime.h \
 sys/audioio.h sys/bsdtty.h sys/epoll.h sys/event.h sys/file.h sys/loadavg.h \
 sys/lock.h sys/mkdev.h sys/modem.h \
diff -Naur a/Python/thread.c b/Python/thread.c
--- a/Python/thread.c	2013-11-10 11:36:41.000000000 +0400
+++ b/Python/thread.c	2014-02-18 09:32:43.383800000 +0400
@@ -8,6 +8,12 @@
 #include "Python.h"
 
 
+#ifdef __MINGW32__
+# if !defined(HAVE_PTHREAD_H) || defined(NT_THREADS)
+#  undef _POSIX_THREADS
+# endif
+#endif
+
 #ifndef _POSIX_THREADS
 /* This means pthreads are not implemented in libc headers, hence the macro
    not present in unistd.h. But they still can be implemented as an external
