diff -Naur a/Lib/compileall.py b/Lib/compileall.py
--- a/Lib/compileall.py	2013-11-10 11:36:40.000000000 +0400
+++ b/Lib/compileall.py	2014-02-18 09:33:09.467000000 +0400
@@ -42,6 +42,8 @@
     success = 1
     for name in names:
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
