diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-02-18 09:33:31.587800000 +0400
+++ b/setup.py	2014-02-18 09:33:31.712600000 +0400
@@ -272,6 +272,9 @@
             args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
         self.compiler.set_executables(**args)
 
+        if host_platform not in ['mingw', 'win32']:
+            self.compiler.library_dirs.insert(0, '.')
+
         build_ext.build_extensions(self)
 
         longest = max([len(e.name) for e in self.extensions])
