diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2014-02-18 09:33:34.801400000 +0400
+++ b/configure.ac	2014-02-18 09:33:34.910600000 +0400
@@ -2493,9 +2493,18 @@
 fi
 
 # check if we need libintl for locale functions
+case $host in
+  *-*-mingw*)
+  dnl Native windows build don't use libintl (see _localemodule.c).
+  dnl Also we don't like setup.py to add "intl" library to the list
+  dnl when build _locale module.
+  ;;
+  *)
 AC_CHECK_LIB(intl, textdomain,
 	AC_DEFINE(WITH_LIBINTL, 1,
 	[Define to 1 if libintl is needed for locale functions.]))
+  ;;
+esac
 
 # checks for system dependent C++ extensions support
 case "$ac_sys_system" in
diff -Naur a/Modules/_localemodule.c b/Modules/_localemodule.c
--- a/Modules/_localemodule.c	2013-11-10 11:36:41.000000000 +0400
+++ b/Modules/_localemodule.c	2014-02-18 09:33:34.926200000 +0400
@@ -10,6 +10,13 @@
 ******************************************************************/
 
 #include "Python.h"
+#ifdef __MINGW32__
+/* The header libintl.h and library libintl may exist on mingw host.
+ * To be compatible with MSVC build we has to undef some defines.
+ */
+#undef HAVE_LIBINTL_H
+#undef HAVE_BIND_TEXTDOMAIN_CODESET
+#endif
 
 #include <stdio.h>
 #include <locale.h>
