diff -r 1c42f1ff2439 Makefile.pre.in
--- a/Makefile.pre.in	Mon Jul 14 12:20:01 2014 +0300
+++ b/Makefile.pre.in	Tue Jul 15 10:59:52 2014 +0200
@@ -947,6 +947,7 @@
 		distutils distutils/command distutils/tests $(XMLLIBSUBDIRS) \
 		multiprocessing multiprocessing/dummy \
 		unittest unittest/test \
+		msilib \
 		lib-old \
 		curses pydoc_data $(MACHDEPS)
 libinstall:	build_all $(srcdir)/Lib/$(PLATDIR) $(srcdir)/Modules/xxmodule.c
