--- Python-2.7.10/Include/object.h.orig	2015-06-26 10:28:53.932400000 +0300
+++ Python-2.7.10/Include/object.h	2015-06-26 10:29:01.950800000 +0300
@@ -311,7 +311,7 @@
 typedef int (*setattrofunc)(PyObject *, PyObject *, PyObject *);
 typedef int (*cmpfunc)(PyObject *, PyObject *);
 typedef PyObject *(*reprfunc)(PyObject *);
-typedef long (*hashfunc)(PyObject *);
+typedef Py_ssize_t (*hashfunc)(PyObject *);
 typedef PyObject *(*richcmpfunc) (PyObject *, PyObject *, int);
 typedef PyObject *(*getiterfunc) (PyObject *);
 typedef PyObject *(*iternextfunc) (PyObject *);
