diff -Naur Python-3.7.0-orig/setup.py Python-3.7.0/setup.py
--- Python-3.7.0-orig/setup.py	2018-07-12 10:22:55.752119400 +0300
+++ Python-3.7.0/setup.py	2018-07-12 10:22:59.246525500 +0300
@@ -807,7 +807,10 @@
                                libraries=['m']) )
 
         # readline
-        do_readline = self.compiler.find_library_file(lib_dirs, 'readline')
+        if not host_platform.startswith(('mingw', 'win')):
+            do_readline = self.compiler.find_library_file(lib_dirs, 'readline')
+        else:
+            do_readline = False
         readline_termcap_library = ""
         curses_library = ""
         # Cannot use os.popen here in py3k.
