--- setuptools-27.3.0/setuptools/monkey.py.orig	2016-09-23 11:38:23.517707200 +0300
+++ setuptools-27.3.0/setuptools/monkey.py	2016-09-23 11:39:40.362333800 +0300
@@ -90,7 +90,8 @@
             setuptools.extension.Extension
         )
 
-    patch_for_msvc_specialized_compiler()
+    if not 'GCC' in sys.version:
+        patch_for_msvc_specialized_compiler()
 
 
 def _patch_distribution_metadata_write_pkg_file():
