diff -urN pyside-setup-opensource-src-5.15.15/build_scripts/config.py.orig pyside-setup-opensource-src-5.15.15/build_scripts/config.py
--- pyside-setup-opensource-src-5.15.15/build_scripts/config.py.orig	2024-08-28 11:49:26.000000000 +0200
+++ pyside-setup-opensource-src-5.15.15/build_scripts/config.py	2024-11-05 20:35:01.919678700 +0100
@@ -94,7 +94,8 @@
             'Programming Language :: Python :: 3.8',
             'Programming Language :: Python :: 3.9',
             'Programming Language :: Python :: 3.10',
-            'Programming Language :: Python :: 3.11'
+            'Programming Language :: Python :: 3.11',
+            'Programming Language :: Python :: 3.12'
         ]
 
         self.setup_script_dir = None
