--- pyside-setup-opensource-src-5.15.18/sources/pyside2/libpyside/CMakeLists.txt.orig	2026-05-14 11:54:45.617284600 +0300
+++ pyside-setup-opensource-src-5.15.18/sources/pyside2/libpyside/CMakeLists.txt	2026-05-14 11:55:33.102033400 +0300
@@ -176,7 +176,7 @@
 configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/PySide2Config-spec.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/install/PySide2Config${SHIBOKEN_PYTHON_CONFIG_SUFFIX}.cmake"
-     INSTALL_DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2-${BINDING_API_VERSION}"
+     INSTALL_DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2"
      PATH_VARS PYSIDE_PYTHONPATH PYSIDE_TYPESYSTEMS PYSIDE_GLUE
 )
 
@@ -193,16 +193,16 @@
                         ARCHIVE DESTINATION "${LIB_INSTALL_DIR}"
                         RUNTIME DESTINATION bin)
 install(EXPORT PySide2Targets NAMESPACE PySide2::
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2-${BINDING_API_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pyside2${pyside2_SUFFIX}.pc"
         DESTINATION "${LIB_INSTALL_DIR}/pkgconfig")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/PySide2Config.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2-${BINDING_API_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/install/PySide2Config${SHIBOKEN_PYTHON_CONFIG_SUFFIX}.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2-${BINDING_API_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/PySide2ConfigVersion.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2-${BINDING_API_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide2")
--- pyside-setup-opensource-src-5.15.18/sources/shiboken2/data/CMakeLists.txt.orig	2026-05-14 11:53:19.817772100 +0300
+++ pyside-setup-opensource-src-5.15.18/sources/shiboken2/data/CMakeLists.txt	2026-05-14 11:53:55.296833300 +0300
@@ -33,7 +33,7 @@
 configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/Shiboken2Config-spec.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/install/Shiboken2Config${PYTHON_CONFIG_SUFFIX}.cmake"
-     INSTALL_DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2-${shiboken2_VERSION}"
+     INSTALL_DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2"
      PATH_VARS SHIBOKEN_PYTHON_MODULE_DIR SHIBOKEN_SHARED_LIBRARY_DIR
 )
 
@@ -45,16 +45,16 @@
                "${CMAKE_CURRENT_BINARY_DIR}/shiboken2${shiboken2_SUFFIX}.pc" @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/Shiboken2Config.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2-${shiboken2_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/install/Shiboken2Config${PYTHON_CONFIG_SUFFIX}.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2-${shiboken2_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/Shiboken2ConfigVersion.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2-${shiboken2_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2")
 
 install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/shiboken_helpers.cmake"
-        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2-${shiboken2_VERSION}")
+        DESTINATION "${LIB_INSTALL_DIR}/cmake/Shiboken2")
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/shiboken2${shiboken2_SUFFIX}.pc"
         DESTINATION "${LIB_INSTALL_DIR}/pkgconfig")
--- pyside-setup-opensource-src-5.15.18/sources/shiboken2/libshiboken/CMakeLists.txt.orig	2026-05-14 12:21:26.615452800 +0300
+++ pyside-setup-opensource-src-5.15.18/sources/shiboken2/libshiboken/CMakeLists.txt	2026-05-14 12:21:48.169352900 +0300
@@ -181,4 +181,4 @@
                             ARCHIVE DESTINATION "${LIB_INSTALL_DIR}"
                             RUNTIME DESTINATION bin)
 install(EXPORT Shiboken2Targets NAMESPACE Shiboken2::
-        DESTINATION ${LIB_INSTALL_DIR}/cmake/Shiboken2-${shiboken2_VERSION})
+        DESTINATION ${LIB_INSTALL_DIR}/cmake/Shiboken2)
