--- python-jsonrpc-server-0.4.0.orig/test/test_streams.py	2020-09-08 00:24:04.000000000 +0300
+++ python-jsonrpc-server-0.4.0/test/test_streams.py	2021-02-20 19:49:18.937960400 +0300
@@ -81,20 +81,12 @@
         'params': {}
     })
 
-    if os.name == 'nt':
-        assert wfile.getvalue() == (
-            b'Content-Length: 49\r\n'
-            b'Content-Type: application/vscode-jsonrpc; charset=utf8\r\n'
-            b'\r\n'
-            b'{"id": "hello", "method": "method", "params": {}}'
-        )
-    else:
-        assert wfile.getvalue() == (
-            b'Content-Length: 44\r\n'
-            b'Content-Type: application/vscode-jsonrpc; charset=utf8\r\n'
-            b'\r\n'
-            b'{"id":"hello","method":"method","params":{}}'
-        )
+    assert wfile.getvalue() == (
+        b'Content-Length: 44\r\n'
+        b'Content-Type: application/vscode-jsonrpc; charset=utf8\r\n'
+        b'\r\n'
+        b'{"id":"hello","method":"method","params":{}}'
+    )
 
 
 class JsonDatetime(datetime.datetime):
@@ -119,6 +111,7 @@
         hour=1,
         minute=1,
         second=1,
+        tzinfo=datetime.timezone.utc
     ))
 
     assert wfile.getvalue() in [
