--- wincertstore-0.2/tests.py.orig	2018-11-13 11:27:59.681041200 -0500
+++ wincertstore-0.2/tests.py	2018-11-13 11:45:13.853010500 -0500
@@ -73,7 +73,7 @@ class TestWinCertStore(unittest.TestCase
             sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             sock.connect(("pypi.python.org", 443))
             ssl_sock = ssl.wrap_socket(sock,
-                                       ssl_version=ssl.PROTOCOL_SSLv3,
+                                       ssl_version=ssl.PROTOCOL_TLS,
                                        ca_certs=certfile.name,
                                        cert_reqs=ssl.CERT_REQUIRED)
             if 0:
