diff -ur mupdf-1.19.1-source.orig/Makerules mupdf-1.19.1-source/Makerules
--- mupdf-1.19.1-source.orig/Makerules	2022-04-06 15:28:02.000000000 +0200
+++ mupdf-1.19.1-source/Makerules	2022-06-20 12:33:43.514981800 +0200
@@ -145,7 +145,7 @@
   CFLAGS += -DCLUSTER
 endif
 
-ifeq ($(OS),MINGW)
+ifeq ($(OS),xMINGW)
   WINDRES := windres
   HAVE_WIN32 := yes
 
@@ -168,8 +168,11 @@
     endif
   endif
 
-else ifeq ($(OS),Linux)
-  HAVE_OBJCOPY := yes
+else ifeq ($(OS),$(filter $(OS),Linux MINGW))
+  # MinGW objcopy does not handle "-z noexecstack"
+  ifeq ($(OS),Linux)
+     HAVE_OBJCOPY := yes
+  endif
 
   ifeq ($(shell pkg-config --exists freetype2 && echo yes),yes)
 	SYS_FREETYPE_CFLAGS := $(shell pkg-config --cflags freetype2)
