--- a/platform/gl/gl-main.c
+++ b/platform/gl/gl-main.c
@@ -3079,11 +3079,7 @@
 }
 #endif
 
-#ifdef _MSC_VER
 int main_utf8(int argc, char **argv)
-#else
-int main(int argc, char **argv)
-#endif
 {
 	const char *trace_file_name = NULL;
 	const char *profile_name = NULL;
--- a/platform/gl/gl-ui.c
+++ b/platform/gl/gl-ui.c
@@ -449,8 +449,6 @@
 #if defined(FREEGLUT) && (GLUT_API_VERSION >= 6)
 	glutKeyboardExtFunc(on_keyboard);
 #else
-	fz_warn(ctx, "This version of MuPDF has been built WITHOUT clipboard or unicode input support!");
-	fz_warn(ctx, "Please file a complaint with your friendly local distribution manager.");
 	glutKeyboardFunc(on_keyboard);
 #endif
 	glutSpecialFunc(on_special);
