--- a/scripts/wrap/state.py
+++ b/scripts/wrap/state.py
@@ -66,8 +66,8 @@
 class State:
     def __init__( self):
         self.os_name = platform.system()
-        self.windows = (self.os_name == 'Windows' or self.os_name.startswith('CYGWIN'))
-        self.cygwin = self.os_name.startswith('CYGWIN')
+        self.windows = False
+        self.cygwin = False
         self.openbsd = self.os_name == 'OpenBSD'
         self.linux = self.os_name == 'Linux'
         self.macos = self.os_name == 'Darwin'
