--- a/scripts/wrap/parse.py
+++ b/scripts/wrap/parse.py
@@ -338,12 +338,6 @@
                                     ' and we have no hard-coded info about size and offset of .regs.'
                                     ' {type0.spelling=} {type_.spelling=} {base_type_cursor.spelling}'
                                     )
-                    assert ret, (
-                            f'{key} has {keep_name}() but have not found size/location of .refs member.'
-                            f' {type_.spelling=}'
-                            f' {base_type_cursor.spelling=}'
-                            f': {fileline(base_type_cursor)}'
-                            )
 
         if type_.spelling in (
                 'struct fz_document',
