--- a/scripts/wrap/parse.py
+++ b/scripts/wrap/parse.py
@@ -252,18 +252,22 @@
                     if verbose:
                         jlib.log( 'There is a keep() fn for this type so it uses reference counting: {keep_name=}')
                     base_type_cursor = get_base_type( type_).get_declaration()
+                    if not base_type_cursor.is_definition():
+                        base_type_cursor2 = find_struct( tu, key, require_definition=False)
+                        if base_type_cursor2 and base_type_cursor2.is_definition():
+                            base_type_cursor = base_type_cursor2
                     if base_type_cursor.is_definition():
                         if verbose:
                             jlib.log( 'Type definition is available so we look for .refs member: {key=} {type_.spelling=} {fileline(base_type_cursor)=}')
                         if verbose:
                             jlib.log('type_.get_fields()')
-                            for cursor in get_members(type_):
+                            for cursor in get_members(base_type_cursor):
                                 jlib.log('    {cursor.spelling=}')
                             jlib.log('base_type_cursor.get_children()')
                             for cursor in base_type_cursor.get_children():
                                 jlib.log('    {cursor.spelling=}')
                             jlib.log('.')
-                        for cursor in get_members(type_):
+                        for cursor in get_members(base_type_cursor):
                             name = cursor.spelling
                             type2 = state.get_name_canonical( cursor.type)
                             if verbose:
@@ -289,47 +293,44 @@
                                     ' declaration, so have to hard-code the size and offset'
                                     ' of the refs member.'
                                     )
-                        if base_type_cursor.is_definition():
-                            if key == 'pdf_document':
-                                ret = 'super.refs', 32
-                            elif key == 'pdf_page':
-                                ret = 'super.refs', 32
-                            elif key == 'fz_pixmap':
-                                ret = 'storable.refs', 32
-                            elif key in (
-                                    'fz_colorspace',
-                                    'fz_image',
-                                    ):
-                                return 'key_storable.storable.refs', 32
-                            elif key == 'pdf_cmap':
-                                return 'storable.refs', 32
-                        else:
-                            #jlib.log( 'No definition available, i.e. forward decl only.')
-                            if key == 'pdf_obj':
-                                ret = 0, 16
-                            elif key == 'fz_path':
-                                ret = 0, 8
-                            elif key in (
-                                    'fz_separations',
-                                    'fz_halftone',
-                                    'pdf_annot',
-                                    'pdf_graft_map',
-                                    ):
-                                # Forward decl, first member is 'int regs;'.
-                                return 0, 32
-                            elif key in (
-                                    'fz_display_list',
-                                    'fz_glyph',
-                                    'fz_jbig2_globals',
-                                    'pdf_function',
-                                    ):
-                                # Forward decl, first member is 'fz_storable storable;'.
-                                return 0, 32
-                            elif key == 'fz_xml':
-                                # This only has a simple .refs member if the
-                                # .up member is null, so we don't attempt to
-                                # use it, by returning size=-1.
-                                ret = 0, -1
+                        if key == 'pdf_document':
+                            ret = 'super.refs', 32
+                        elif key == 'pdf_page':
+                            ret = 'super.refs', 32
+                        elif key == 'fz_pixmap':
+                            ret = 'storable.refs', 32
+                        elif key in (
+                                'fz_colorspace',
+                                'fz_image',
+                                ):
+                            return 'key_storable.storable.refs', 32
+                        elif key == 'pdf_cmap':
+                            return 'storable.refs', 32
+                        elif key == 'pdf_obj':
+                            ret = 0, 16
+                        elif key == 'fz_path':
+                            ret = 0, 8
+                        elif key in (
+                                'fz_separations',
+                                'fz_halftone',
+                                'pdf_annot',
+                                'pdf_graft_map',
+                                ):
+                            # First member is 'int refs;'.
+                            return 0, 32
+                        elif key in (
+                                'fz_display_list',
+                                'fz_glyph',
+                                'fz_jbig2_globals',
+                                'pdf_function',
+                                ):
+                            # First member is 'fz_storable storable;'.
+                            return 0, 32
+                        elif key == 'fz_xml':
+                            # This only has a simple .refs member if the
+                            # .up member is null, so we don't attempt to
+                            # use it, by returning size=-1.
+                            ret = 0, -1
 
                         if ret is None:
                             # Need to hard-code info for this type.
@@ -990,3 +991,10 @@
                 return ret
         if c.spelling == name:
             return c
+        if c.kind in (
+                clang.cindex.CursorKind.STRUCT_DECL,
+                clang.cindex.CursorKind.UNION_DECL,
+                ):
+            ret = find_name( c, name, nest+1)
+            if ret:
+                return ret
