--- a/scripts/wrap/__main__.py
+++ b/scripts/wrap/__main__.py
@@ -924,6 +924,7 @@
     del sys.path[0]
 
-autovenv.enter()
+if not os.environ.get('MSYSTEM'):
+    autovenv.enter()
 
 # Install/import pipcl. We don't pass 'pipcl' to above call to autovenv.enter()
 # because this would do `pip install pipcl`, which seems to fail if we are
--- a/scripts/wrap/swig.py
+++ b/scripts/wrap/swig.py
@@ -743,1 +743,1 @@
-                    PyObject* format_tb = PyDict_GetItem(traceback_dict, PyString_FromString("format_tb"));
+                    PyObject* format_tb = PyDict_GetItemString(traceback_dict, "format_tb");
