--- libLAS-1.8.1/CMakeLists.txt.orig	2026-08-28 15:32:03.939687100 +0800
+++ libLAS-1.8.1/CMakeLists.txt	2026-08-28 15:36:12.284197000 +0800
@@ -79,9 +79,6 @@
 
 # Path to additional CMake modules
 set(CMAKE_MODULE_PATH "${libLAS_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
-if (CMAKE_MAJOR_VERSION GREATER 2)
-    cmake_policy(SET CMP0042 OLD)
-endif()
 
 if (CMAKE_SYSTEM_NAME MATCHES "Linux")
 set(CMAKE_SKIP_RPATH ON)
