--- liblas-1.8.1/CMakeLists.txt.orig	2016-08-23 19:41:33.000000000 +0300
+++ liblas-1.8.1/CMakeLists.txt	2017-11-15 08:49:14.048434400 +0300
@@ -295,7 +295,7 @@
 ###############################################################################
 # Installation settings
 
-if(WIN32)
+if(WIN32 AND NOT MINGW)
   set(DEFAULT_LIB_SUBDIR lib)
   set(DEFAULT_DATA_SUBDIR .)
   set(DEFAULT_INCLUDE_SUBDIR include)
--- liblas-1.8.1/cmake/modules/FindLASzip.cmake.orig	2016-08-23 19:41:33.000000000 +0300
+++ liblas-1.8.1/cmake/modules/FindLASzip.cmake	2017-11-15 08:54:41.164954600 +0300
@@ -25,7 +25,7 @@
   SET(LASZIP_FIND_QUIETLY TRUE)
 ENDIF()
 
-IF(WIN32)
+IF(WIN32 AND NOT MINGW)
   SET(OSGEO4W_IMPORT_LIBRARY laszip)
   IF(DEFINED ENV{OSGEO4W_ROOT})
     SET(OSGEO4W_ROOT_DIR $ENV{OSGEO4W_ROOT})
@@ -36,24 +36,32 @@
   ENDIF()
 ENDIF()
 
+SET(_laszip_SEARCH_DIRS
+  ${LASZIP_ROOT_DIR}
+  /usr/local
+  /sw # Fink
+  /opt/local # DarwinPorts
+  /opt/csw # Blastwave
+  /opt/lib/ceres
+)
 
 FIND_PATH(LASZIP_INCLUDE_DIR
-  laszip.hpp
-  PATH_PREFIXES laszip
-  PATHS
-  /usr/include
-  /usr/local/include
-  ${OSGEO4W_ROOT_DIR}/include
-  NO_DEFAULT_PATH)
+  NAMES
+    laszip/laszip.hpp
+  HINTS
+    ${_laszip_SEARCH_DIRS}
+  PATH_SUFFIXES
+    include)
 
 SET(LASZIP_NAMES ${OSGEO4W_IMPORT_LIBRARY} laszip)
 
 FIND_LIBRARY(LASZIP_LIBRARY
-  NAMES ${LASZIP_NAMES}
-  PATHS
-  /usr/lib
-  /usr/local/lib
-  ${OSGEO4W_ROOT_DIR}/lib)
+  NAMES
+    ${LASZIP_NAMES}
+  HINTS
+    ${_laszip_SEARCH_DIRS}
+  PATH_SUFFIXES
+    lib64 lib)
 
 IF(LASZIP_FOUND)
   SET(LASZIP_LIBRARIES ${LASZIP_LIBRARY})
--- liblas-1.8.1/cmake/CMakeLists.txt.orig	2016-08-23 19:41:33.000000000 +0300
+++ liblas-1.8.1/cmake/CMakeLists.txt	2017-11-15 08:49:14.017234300 +0300
@@ -8,6 +8,9 @@
 if (NOT WIN32)
   set(INSTALL_CMAKE_DIR "share/cmake/${PROJECT_NAME}")
   set (PROJECT_ROOT_DIR "../../..")
+elseif (MINGW)
+  set(INSTALL_CMAKE_DIR "lib/cmake/${PROJECT_NAME}")
+  set (PROJECT_ROOT_DIR "../../..")
 else ()
   set(INSTALL_CMAKE_DIR "cmake")
   set (PROJECT_ROOT_DIR "..")
--- liblas-1.8.1/test/unit/spatialreference_test.cpp.orig	2016-08-23 19:41:33.000000000 +0300
+++ liblas-1.8.1/test/unit/spatialreference_test.cpp	2017-11-15 08:49:14.095234500 +0300
@@ -110,7 +110,11 @@
         liblas::Header const& header = reader.GetHeader();
         liblas::SpatialReference const& in_ref = header.GetSRS();
         
+#ifdef _WIN64
         const char* utm15_wkt = "PROJCS[\"NAD83 / UTM zone 15N\",GEOGCS[\"NAD83\",DATUM[\"North_American_Datum_1983\",SPHEROID[\"GRS 1980\",6378137,298.2572221010002,AUTHORITY[\"EPSG\",\"7019\"]],AUTHORITY[\"EPSG\",\"6269\"]],PRIMEM[\"Greenwich\",0],UNIT[\"degree\",0.0174532925199433],AUTHORITY[\"EPSG\",\"4269\"]],PROJECTION[\"Transverse_Mercator\"],PARAMETER[\"latitude_of_origin\",0],PARAMETER[\"central_meridian\",-93],PARAMETER[\"scale_factor\",0.9996],PARAMETER[\"false_easting\",500000],PARAMETER[\"false_northing\",0],UNIT[\"metre\",1,AUTHORITY[\"EPSG\",\"9001\"]],AUTHORITY[\"EPSG\",\"26915\"]]";
+#else
+        const char* utm15_wkt = "PROJCS[\"NAD83 / UTM zone 15N\",GEOGCS[\"NAD83\",DATUM[\"North_American_Datum_1983\",SPHEROID[\"GRS 1980\",6378137,298.2572221010042,AUTHORITY[\"EPSG\",\"7019\"]],AUTHORITY[\"EPSG\",\"6269\"]],PRIMEM[\"Greenwich\",0],UNIT[\"degree\",0.0174532925199433],AUTHORITY[\"EPSG\",\"4269\"]],PROJECTION[\"Transverse_Mercator\"],PARAMETER[\"latitude_of_origin\",0],PARAMETER[\"central_meridian\",-93],PARAMETER[\"scale_factor\",0.9996],PARAMETER[\"false_easting\",500000],PARAMETER[\"false_northing\",0],UNIT[\"metre\",1,AUTHORITY[\"EPSG\",\"9001\"]],AUTHORITY[\"EPSG\",\"26915\"]]";
+#endif
         ensure_equals("Input WKT comparison", in_ref.GetWKT(), utm15_wkt);
 
         const char* epsg4326_wkt = "GEOGCS[\"WGS 84\",DATUM[\"WGS_1984\",SPHEROID[\"WGS 84\",6378137,298.257223563,AUTHORITY[\"EPSG\",\"7030\"]],AUTHORITY[\"EPSG\",\"6326\"]],PRIMEM[\"Greenwich\",0],UNIT[\"degree\",0.0174532925199433],AUTHORITY[\"EPSG\",\"4326\"]]";
--- liblas-1.8.1/cmake/modules/BuildOSGeo4W.cmake.orig	2017-11-15 09:53:37.962878200 +0300
+++ liblas-1.8.1/cmake/modules/BuildOSGeo4W.cmake	2017-11-15 09:56:32.932509600 +0300
@@ -75,7 +75,6 @@
 macro(copy_files GLOBPAT DESTINATION  )
 
     file(GLOB_RECURSE COPY_FILES
-         RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
         ${GLOBPAT})
     foreach(FILENAME ${COPY_FILES})
         set(SRC "${FILENAME}")
