--- a/src/tools/idc/CMakeLists.txt
+++ b/src/tools/idc/CMakeLists.txt
@@ -13,3 +13,6 @@
         main.cpp
 )
 qt_internal_return_unless_building_tools()
+set_target_properties(idc PROPERTIES
+    OUTPUT_NAME "idc-qt6"
+)
--- a/tools/dumpcpp/CMakeLists.txt
+++ b/tools/dumpcpp/CMakeLists.txt
@@ -22,3 +22,6 @@
         Qt::Widgets
 )
 qt_internal_return_unless_building_tools()
+set_target_properties(dumpcpp PROPERTIES
+    OUTPUT_NAME "dumpcpp-qt6"
+)
--- a/tools/dumpdoc/CMakeLists.txt
+++ b/tools/dumpdoc/CMakeLists.txt
@@ -17,3 +17,6 @@
         Qt::Widgets
 )
 qt_internal_return_unless_building_tools()
+set_target_properties(dumpdoc PROPERTIES
+    OUTPUT_NAME "dumpdoc-qt6"
+)
 #### Keys ignored in scope 1:.:.:dumpdoc.pro:<TRUE>:
 # QMAKE_TARGET_DESCRIPTION = "Active Qt DumpDoc"
--- a/tools/testcon/CMakeLists.txt
+++ b/tools/testcon/CMakeLists.txt
@@ -39,3 +39,6 @@
     FILES
         ${testcon_resource_files}
 )
+set_target_properties(testcon PROPERTIES
+    OUTPUT_NAME "testcon-qt6"
+)
