diff -urN eccodes-2.26.0-Source/cmake/ecbuild_pkgconfig.cmake.orig eccodes-2.26.0-Source/cmake/ecbuild_pkgconfig.cmake
--- eccodes-2.26.0-Source/cmake/ecbuild_pkgconfig.cmake.orig	2022-03-03 15:38:48.000000000 +0100
+++ eccodes-2.26.0-Source/cmake/ecbuild_pkgconfig.cmake	2022-06-11 10:42:44.925949900 +0200
@@ -193,11 +193,7 @@
 
           if( _set_append )
 
-            if( _dir )
-              list( APPEND _pkgconfig_libs "${RPATH_FLAG}${_dir}" "-L${_dir}" "-l${_name}" )
-            else()
-              list( APPEND _pkgconfig_libs "-l${_name}" )
-            endif()
+            list( APPEND _pkgconfig_libs "-l${_name}" )
 
           endif()
 
