--- tcl8.6.13/win/tclWinInit.c.orig	2022-09-14 13:22:15.000000000 +0200
+++ tcl8.6.13/win/tclWinInit.c	2024-01-17 07:43:21.572824600 +0100
@@ -326,7 +326,7 @@
     end = strrchr(name, '\\');
     *end = '\0';
     p = strrchr(name, '\\');
-    if (p != NULL) {
+    if (p != NULL && !strcmp(p+1, "bin")) {
 	end = p;
     }
     *end = '\\';
@@ -377,7 +377,7 @@
     end = strrchr(name, '\\');
     *end = '\0';
     p = strrchr(name, '\\');
-    if (p != NULL) {
+    if (p != NULL && !strcmp(p+1, "bin")) {
 	end = p;
     }
     *end = '\\';
