diff -Naur tcl8.6.5-orig/win/tclWinInt.h tcl8.6.5/win/tclWinInt.h
--- tcl8.6.11-orig/win/tclWinInt.h	2016-02-25 23:12:38.000000000 +0300
+++ tcl8.6.11/win/tclWinInt.h	2016-03-03 08:47:51.129171100 +0300
@@ -57,7 +57,7 @@
 #ifndef TCL_Z_MODIFIER
 #   ifdef _WIN64
 #	if defined(__USE_MINGW_ANSI_STDIO) && __USE_MINGW_ANSI_STDIO
-#         define TCL_Z_MODIFIER        "ll"
+#         define TCL_Z_MODIFIER        "I64"
 #	else
 #         define TCL_Z_MODIFIER        "I"
 #	endif
diff -Naur tcl8.6.5-orig/win/Makefile.in tcl8.6.5/win/Makefile.in
--- tcl8.6.11-orig/win/Makefile.in
+++ tcl8.6.11/win/Makefile.in
+++ #<buffer Makefile.in>
@@ -85,7 +85,7 @@
 #CFLAGS = 		$(CFLAGS_DEBUG)
 #CFLAGS = 		$(CFLAGS_OPTIMIZE)
 #CFLAGS = 		$(CFLAGS_DEBUG) $(CFLAGS_OPTIMIZE)
-CFLAGS = 		@CFLAGS@ @CFLAGS_DEFAULT@ -D_ATL_XP_TARGETING=1 -D__USE_MINGW_ANSI_STDIO=0 -DMP_FIXED_CUTOFFS -DMP_NO_STDINT
+CFLAGS = 		@CFLAGS@ @CFLAGS_DEFAULT@ -D_ATL_XP_TARGETING=1 -DMP_FIXED_CUTOFFS -DMP_NO_STDINT
 
 # To compile without backward compatibility and deprecated code uncomment the
 # following
