--- tcl8.6.11/win/tcl.m4.orig	2021-01-20 17:25:46.749753100 -0800
+++ tcl8.6.11/win/tcl.m4	2021-01-20 17:28:04.328075100 -0800
@@ -758,6 +758,10 @@
 		MACHINE="AMD64" ; # assume AMD64 as default 64-bit build
 		AC_MSG_RESULT([   Using 64-bit $MACHINE mode])
 		;;
+	    arm64)
+		MACHINE="ARM64"
+		AC_MSG_RESULT([   Using 64-bit $MACHINE mode])
+		;;
 	    ia64)
 		MACHINE="IA64"
 		AC_MSG_RESULT([   Using 64-bit $MACHINE mode])
