--- a/win/configure.in
+++ b/win/configure.in
@@ -285,6 +285,7 @@
 
 eval "TCL_DLL_FILE=tcl${VER}${DLLSUFFIX}"
 
+eval "TCL_LIB_FLAG=\"-ltcl${VER}${LIBFLAGSUFFIX}\""
 eval "TCL_STUB_LIB_FILE=\"${LIBPREFIX}tclstub${VER}${LIBSUFFIX}\""
 eval "TCL_STUB_LIB_FLAG=\"-ltclstub${VER}${LIBFLAGSUFFIX}\""
 eval "TCL_BUILD_STUB_LIB_SPEC=\"-L`$CYGPATH $(pwd)` ${TCL_STUB_LIB_FLAG}\""
@@ -375,6 +376,7 @@
 AC_SUBST(TCL_IMPORT_LIB_FILE)
 AC_SUBST(TCL_IMPORT_LIB_FLAG)
 # empty on win
+AC_SUBST(TCL_LIBS)
 AC_SUBST(TCL_LIB_SPEC)
 AC_SUBST(TCL_STUB_LIB_FILE)
 AC_SUBST(TCL_STUB_LIB_FLAG)
@@ -458,7 +458,7 @@
 AC_SUBST(RC_DEFINES)
 AC_SUBST(RES)
 
-AC_CONFIG_FILES([Makefile tclConfig.sh tcl.hpj tclsh.exe.manifest])
+AC_CONFIG_FILES([Makefile tclConfig.sh tcl.hpj tclsh.exe.manifest tcl.pc])
 AC_OUTPUT
 
 dnl Local Variables:
--- a/win/Makefile.in
+++ b/win/Makefile.in
@@ -672,6 +672,9 @@
 	    echo Installing $(REG_LIB_FILE); \
 	    $(COPY) $(REG_LIB_FILE) "$(LIB_INSTALL_DIR)/reg${REGDOTVER}"; \
 	    fi
+	@echo "Installing pkg-config file to $(LIB_INSTALL_DIR)/pkgconfig/"
+	@$(INSTALL_DATA_DIR) "$(LIB_INSTALL_DIR)/pkgconfig"
+	@$(INSTALL_DATA) tcl.pc "$(LIB_INSTALL_DIR)/pkgconfig/tcl.pc"
 
 install-libraries: libraries install-tzdata install-msgs
 	@for i in "$(prefix)/lib" "$(INCLUDE_INSTALL_DIR)" \
