diff --git a/CMakeLists.txt b/CMakeLists.txt
index e324a0261..e5c174e58 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 # setup version numbers
 # TODO: Update these for each release!
 set(VERSION_MAJOR 3)
-set(VERSION_MINOR 0)
+set(VERSION_MINOR 2)
 set(VERSION_PATCH 0)
 set(VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
 
@@ -65,18 +65,13 @@ IF(NOT WIN32)
     OPTION(FREEGLUT_WAYLAND "Use Wayland (no X11)" OFF)
 ENDIF()
 
-
 SET(FREEGLUT_HEADERS
+    include/GL/glut.h
     include/GL/freeglut.h
     include/GL/freeglut_ucall.h
     include/GL/freeglut_ext.h
     include/GL/freeglut_std.h
 )
-IF(FREEGLUT_REPLACE_GLUT)
-    LIST(APPEND FREEGLUT_SRCS
-        include/GL/glut.h
-    )
-ENDIF()
 SET(FREEGLUT_SRCS
     ${FREEGLUT_HEADERS}
     src/fg_callbacks.c
