--- cloog-0.18.1/Makefile.am.orig	2013-10-16 21:08:32.616800000 +0400
+++ cloog-0.18.1/Makefile.am	2013-10-11 11:27:03.000000000 +0400
@@ -90,7 +90,7 @@
 AM_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include
 AM_CFLAGS = $(CFLAGS_WARN)
 libcloog_isl_la_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include @ISL_CPPFLAGS@ @OSL_CPPFLAGS@
-libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ \
+libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	-rpath $(libdir) @ISL_LDFLAGS@ @OSL_LDFLAGS@
 libcloog_isl_la_LIBADD = @ISL_LIBS@ @OSL_LIBS@ $(ISL_LA) $(OSL_LA)
 libcloog_isl_la_SOURCES = \
--- cloog-0.18.1/Makefile.in.orig	2013-10-16 21:09:19.058000000 +0400
+++ cloog-0.18.1/Makefile.in	2013-10-11 11:27:50.000000000 +0400
@@ -444,7 +444,7 @@
 AM_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include
 AM_CFLAGS = $(CFLAGS_WARN)
 libcloog_isl_la_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include @ISL_CPPFLAGS@ @OSL_CPPFLAGS@
-libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ \
+libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	-rpath $(libdir) @ISL_LDFLAGS@ @OSL_LDFLAGS@
 
 libcloog_isl_la_LIBADD = @ISL_LIBS@ @OSL_LIBS@ $(ISL_LA) $(OSL_LA)
--- cloog-0.18.1/isl/Makefile.am.orig	2013-10-16 21:09:33.160400000 +0400
+++ cloog-0.18.1/isl/Makefile.am	2013-10-11 11:27:21.000000000 +0400
@@ -154,7 +154,7 @@
 	isl_sample_piplib.h \
 	isl_piplib.c
 libisl_la_LIBADD = @PIPLIB_LIBS@ @GMP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@
 libisl_la_CPPFLAGS = $(INCLUDES) @PIPLIB_CPPFLAGS@ @GMP_CPPFLAGS@
 
--- cloog-0.18.1/isl/Makefile.in.orig	2013-10-16 21:09:54.282800000 +0400
+++ cloog-0.18.1/isl/Makefile.in	2013-10-11 11:27:40.000000000 +0400
@@ -645,7 +645,7 @@
 	isl_piplib.c
 
 libisl_la_LIBADD = @PIPLIB_LIBS@ @GMP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@
 
 libisl_la_CPPFLAGS = $(INCLUDES) @PIPLIB_CPPFLAGS@ @GMP_CPPFLAGS@
