--- src/BSDFProcessor-1.1.2/CMakeLists.txt	2016-03-05 15:48:02.000000000 +0100
+++ CMakeLists.txt	2016-03-12 01:35:17.460931600 +0100
@@ -8,8 +8,10 @@
 
 cmake_minimum_required(VERSION 2.8.11)
 
-set(LIBBSDF_DIR ../libbsdf)
-add_subdirectory(${LIBBSDF_DIR} LibbsdfBin)
+if(NOT LIBBSDF_DIR)
+    set(LIBBSDF_DIR ../libbsdf)
+    add_subdirectory(${LIBBSDF_DIR} LibbsdfBin)
+endif()
 include_directories(${LIBBSDF_DIR}/include)
 
 project(BSDFProcessor)
