--- CMakeLists.txt.old	2016-09-17 13:31:29.407837000 +0200
+++ CMakeLists.txt	2016-09-17 13:32:02.220837000 +0200
@@ -62,8 +62,8 @@
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /LARGEADDRESSAWARE")
     endif(CMAKE_SIZEOF_VOID_P MATCHES 4)
 else()
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse3")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse3")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse3 -O2")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse3 -O2")
 endif()
 
 find_package(OpenMP QUIET)
