--- dieterv-gobject-introspection-c9e2d88/giscanner/utils.py.orig	2013-05-08 10:59:38 +0400
+++ dieterv-gobject-introspection-c9e2d88/giscanner/utils.py	2013-05-08 11:13:12 +0400
@@ -109,7 +109,7 @@
     # FIXME: This hackish, but I'm not sure how to do this
     #        in a way which is compatible with both libtool 2.2
     #        and pre-2.2. Johan 2008-10-21
-    libname = libname.replace('.libs/.libs', '.libs')
+    libname = libname.replace('.libs/.libs', '.libs').replace('.libs\\.libs', '.libs')
     return libname
 
 
