--- dieterv-gobject-introspection-c9e2d88/Makefile-gir.am.orig	2013-05-08 12:56:52 +0400
+++ dieterv-gobject-introspection-c9e2d88/Makefile-gir.am	2013-05-08 13:06:55 +0400
@@ -56,11 +56,7 @@
 GLIB_INCLUDEDIR=$(shell pkg-config --variable=includedir glib-2.0)/glib-2.0
 GLIB_LIBDIR=$(shell pkg-config --variable=libdir glib-2.0)
 
-if OS_WIN32
-GLIB_LIBRARY=libglib-2.0-0
-else
 GLIB_LIBRARY=glib-2.0
-endif
 
 GLib_2_0_gir_LIBS = $(GLIB_LIBRARY) $(GOBJECT_LIBRARY)
 GLib_2_0_gir_SCANNERFLAGS = \
@@ -97,11 +93,7 @@
 GOBJECT_INCLUDEDIR=$(shell pkg-config --variable=includedir gobject-2.0)/glib-2.0
 GOBJECT_LIBDIR=$(shell pkg-config --variable=libdir gobject-2.0)
 
-if OS_WIN32
-GOBJECT_LIBRARY=libgobject-2.0-0
-else
 GOBJECT_LIBRARY=gobject-2.0
-endif
 
 GObject-2.0.gir: GLib-2.0.gir
 GObject_2_0_gir_LIBS = $(GOBJECT_LIBRARY)
@@ -129,11 +121,7 @@
 GMODULE_INCLUDEDIR=$(shell pkg-config --variable=includedir gmodule-2.0)/glib-2.0
 GMODULE_LIBDIR=$(shell pkg-config --variable=libdir gmodule-2.0)
 
-if OS_WIN32
-GMODULE_LIBRARY=libgmodule-2.0-0
-else
 GMODULE_LIBRARY=gmodule-2.0
-endif
 
 GModule-2.0.gir: GLib-2.0.gir
 
@@ -158,11 +146,7 @@
 GIO_INCLUDEDIR=$(shell pkg-config --variable=includedir gio-2.0)/glib-2.0
 GIO_LIBDIR=$(shell pkg-config --variable=libdir gio-2.0)
 
-if OS_WIN32
-GIO_LIBRARY=libgio-2.0-0
-else
 GIO_LIBRARY=gio-2.0
-endif
 
 if HAVE_GIO_UNIX
 GIO_UNIX_HDRS=$(shell pkg-config --variable=includedir gio-unix-2.0)/gio-unix-2.0/gio/*.h
