--- gobject-introspection-1.37.4/giscanner/scannerlexer.l.orig	2013-07-10 16:13:23.000000000 +0000
+++ gobject-introspection-1.37.4/giscanner/scannerlexer.l	2013-07-27 01:52:39.921187200 +0000
@@ -139,6 +139,7 @@
 
 "__asm"[\t\f\v\r ]+"volatile"  	        { if (!parse_ignored_macro()) REJECT; }
 "__asm__"[\t\f\v\r ]+"volatile"	        { if (!parse_ignored_macro()) REJECT; }
+"__asm__"[\t\f\v\r ]+"__volatile__"	{ if (!parse_ignored_macro()) REJECT; }
 "__asm" 	        	        { if (!parse_ignored_macro()) REJECT; }
 "__asm__" 	        	        { if (!parse_ignored_macro()) REJECT; }
 "__attribute__" 		        { if (!parse_ignored_macro()) REJECT; }
