--- collada-dom/collada-dom.pc.in.orig	2015-01-22 15:35:43.690800000 +0300
+++ collada-dom/collada-dom.pc.in	2015-01-22 15:36:23.689200000 +0300
@@ -2,11 +2,11 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@/bin
-libdir=@CMAKE_INSTALL_PREFIX@/lib@LIB_SUFFIX%
+libdir=@CMAKE_INSTALL_PREFIX@/lib@LIB_SUFFIX@
 includedir=@CMAKE_INSTALL_PREFIX@/@COLLADA_DOM_INCLUDE_INSTALL_DIR@
 
 Name: collada-dom
-Description: COLLADA Document Object Model (DOM), 1.4 support=%OPT_COLLADA14%, 1.5 support=%OPT_COLLADA15%
+Description: COLLADA Document Object Model (DOM), 1.4 support=@OPT_COLLADA14@, 1.5 support=@OPT_COLLADA15@
 Version: @COLLADA_DOM_VERSION@
 URL:  http://sourceforge.net/projects/collada-dom
 Libs: -L${libdir} -lcollada-dom@COLLADA_DOM_LIBRARY_SUFFIX@
