--- collada-dom/dom/include/dae.h.orig	2017-02-01 09:28:36.234499500 +0300
+++ collada-dom/dom/include/dae.h	2017-02-01 09:29:14.368583400 +0300
@@ -287,7 +287,11 @@
     daeSidRefCache sidRefCache;
     daeString COLLADA_VERSION, COLLADA_NAMESPACE; // dynamic
 
+#if _GLIBCXX_USE_CXX11_ABI > 0
+    std::unique_ptr<charEncoding> localCharEncoding;
+#else
     std::auto_ptr<charEncoding> localCharEncoding;
+#endif
     static charEncoding globalCharEncoding;
 };
 
--- collada-dom/dom/include/dae/daeErrorHandler.h.orig	2017-02-01 09:41:35.148150000 +0300
+++ collada-dom/dom/include/dae/daeErrorHandler.h	2017-02-01 09:43:31.936166700 +0300
@@ -55,7 +55,11 @@
 
 private:
 	static daeErrorHandler *_instance;
+#if _GLIBCXX_USE_CXX11_ABI > 0
+	static std::unique_ptr<daeErrorHandler> _defaultInstance;
+#else
 	static std::auto_ptr<daeErrorHandler> _defaultInstance;
+#endif
 };
 
 #endif
--- collada-dom/dom/src/dae/daeErrorHandler.cpp.orig	2017-02-01 09:57:42.158968700 +0300
+++ collada-dom/dom/src/dae/daeErrorHandler.cpp	2017-02-01 09:59:28.635164700 +0300
@@ -10,7 +10,11 @@
 #include <modules/stdErrPlugin.h>
 
 daeErrorHandler *daeErrorHandler::_instance = NULL;
+#if _GLIBCXX_USE_CXX11_ABI > 0
+std::unique_ptr<daeErrorHandler> daeErrorHandler::_defaultInstance(new stdErrPlugin);
+#else
 std::auto_ptr<daeErrorHandler> daeErrorHandler::_defaultInstance(new stdErrPlugin);
+#endif
 
 daeErrorHandler::daeErrorHandler() {
 }
