diff -aur old/configure 001/configure
--- old/configure	2014-05-13 03:20:24.000000000 -0300
+++ 001/configure	2014-12-17 16:52:24.986328100 -0200
@@ -1505,7 +1505,7 @@
   --with-gmp[=DIR]        use GMP instead of SILC Math [search in DIR/lib and DIR/include]
   --with-iconv[=DIR]      use libiconv [search in DIR/include and DIR/lib]
   --with-pthreads[=DIR]   use POSIX threads [search in DIR/include and DIR/lib]
-  --with-win32            compile native WIN32 (MinGW) code (-mno-cygwin)
+  --with-win32            compile native WIN32 (MinGW) code
   --with-simdir=DIR       directory for SIM modules [PREFIX/modules]
 
 Some influential environment variables:
@@ -15310,7 +15310,6 @@
 $as_echo "#define SILC_WIN32 /**/" >>confdefs.h
 
     win32_support=true
-    CFLAGS="-mno-cygwin $CFLAGS"
     LIBS="$LIBS -lwsock32"
 
 else
diff -aur old/configure.ac 001/configure.ac
--- old/configure.ac	2014-05-13 03:20:24.000000000 -0300
+++ 001/configure.ac	2014-12-17 16:52:11.393554600 -0200
@@ -1228,12 +1228,11 @@
 #
 AC_MSG_CHECKING(whether to compile native WIN32 code)
 AC_ARG_WITH(win32,
-  [  --with-win32            compile native WIN32 (MinGW) code (-mno-cygwin)],
+  [  --with-win32            compile native WIN32 (MinGW) code],
   [
     AC_MSG_RESULT(yes)
     AC_DEFINE([SILC_WIN32], [], [SILC_WIN32])
     win32_support=true
-    CFLAGS="-mno-cygwin $CFLAGS"
     LIBS="$LIBS -lwsock32"
   ],
   [
diff -aur old/win32/silcdefs.h 001/win32/silcdefs.h
--- old/win32/silcdefs.h	2014-05-05 02:29:05.000000000 -0300
+++ 001/win32/silcdefs.h	2014-12-17 16:51:50.421875000 -0200
@@ -103,7 +103,7 @@
 /* #undef Rfclose */
 /* #undef Rgethostbyname */
 
-/* Native WIN32 compilation (-mno-cygwin GCC option) under cygwin, though
+/* Native WIN32 compilation under cygwin, though
    the code compiles with any native WIN32 compiler. */
 #ifndef SILC_WIN32
 #define SILC_WIN32 1
