From 7aa3b231cd1a6359eb3e2cff15047f189f0bafa6 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 11 Dec 2013 21:58:11 +0000
Subject: [PATCH 1/2] cmake: Link OpenGL with glsl_test

---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1215af7..fcd364c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,9 @@ else()
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -s")
 endif()
 
+find_package(OpenGL REQUIRED)
+link_directories(${OpenGL_LIBRARY_DIRS})
+
 file(GLOB glcpp-library_sources src/glsl/glcpp/*.c)
 #file(GLOB glcpp-library_sources_remove src/glsl/glcpp/glcpp.c)
 #list(REMOVE_ITEM glcpp-library_sources ${glcpp-library_sources_remove})
@@ -35,7 +38,7 @@ target_link_libraries(glsl_compiler glsl_optimizer)
 
 file(GLOB glsl_test_sources tests/*.cpp)
 add_executable(glsl_test ${glsl_test_sources})
-target_link_libraries(glsl_test glsl_optimizer)
+target_link_libraries(glsl_test ${OPENGL_LIBRARIES} glsl_optimizer)
 
 file(GLOB glslopt_sources contrib/glslopt/*.cpp)
 add_executable(glslopt ${glslopt_sources})
-- 
1.8.5.1

