From 7c1500ad4ce2e2d279b60adc25db72956b08c85b Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 11 Dec 2013 21:58:28 +0000
Subject: [PATCH 2/2] Fix glslopt compile failure

---
 contrib/glslopt/Main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/glslopt/Main.cpp b/contrib/glslopt/Main.cpp
index c9e2546..bf2bbf4 100644
--- a/contrib/glslopt/Main.cpp
+++ b/contrib/glslopt/Main.cpp
@@ -17,7 +17,7 @@ static int printhelp(const char* msg)
 
 static bool init()
 {
-	gContext = glslopt_initialize(false);
+	gContext = glslopt_initialize(kGlslTargetOpenGL);
 	if( !gContext )
 		return false;
 	return true;
-- 
1.8.5.1

