diff -Naur pastix-6.2.2.orig/kernels/CMakeLists.txt pastix-6.2.2/kernels/CMakeLists.txt
--- pastix-6.2.2.orig/kernels/CMakeLists.txt	2022-08-24 10:58:00.030693000 +0200
+++ pastix-6.2.2/kernels/CMakeLists.txt	2023-05-23 14:16:55.892064700 +0200
@@ -185,9 +185,10 @@
 # Installation
 # ------------
 install(TARGETS pastix_kernels
-        EXPORT pastix_kernelsTargets
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib)
+        EXPORT pastix_kernelsTargets        
+        ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
+        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR})
 
 ### Add generated files to the global property
 add_documented_files(
diff -Naur pastix-6.2.2.orig/sopalin/parsec/CMakeLists.txt pastix-6.2.2/sopalin/parsec/CMakeLists.txt
--- pastix-6.2.2.orig/sopalin/parsec/CMakeLists.txt	2022-08-24 10:58:00.050694200 +0200
+++ pastix-6.2.2/sopalin/parsec/CMakeLists.txt	2023-05-23 14:20:00.889964800 +0200
@@ -121,9 +121,10 @@
 # Installation
 # ------------
 install(TARGETS pastix_parsec
-        EXPORT pastix_parsecTargets
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib)
+        EXPORT pastix_parsecTargets     
+        ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
+        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR})
 
 ### Add non-generated documented files to the global property
 add_documented_files(
diff -Naur pastix-6.2.2.orig/sopalin/starpu/CMakeLists.txt pastix-6.2.2/sopalin/starpu/CMakeLists.txt
--- pastix-6.2.2.orig/sopalin/starpu/CMakeLists.txt	2022-08-24 10:58:00.054694400 +0200
+++ pastix-6.2.2/sopalin/starpu/CMakeLists.txt	2023-05-23 14:20:03.870173900 +0200
@@ -113,9 +113,10 @@
 # Installation
 # ------------
 install(TARGETS pastix_starpu
-        EXPORT pastix_starpuTargets
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib)
+        EXPORT pastix_starpuTargets       
+        ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
+        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR})
 
 ### Add non-generated documented files to the global property
 add_documented_files(
diff -Naur pastix-6.2.2.orig/spm/src/CMakeLists.txt pastix-6.2.2/spm/src/CMakeLists.txt
--- pastix-6.2.2.orig/spm/src/CMakeLists.txt	2022-08-24 10:58:02.634839300 +0200
+++ pastix-6.2.2/spm/src/CMakeLists.txt	2023-05-23 14:20:07.038687900 +0200
@@ -175,9 +175,10 @@
 # Installation
 # ------------
 install(TARGETS spm
-        EXPORT spmTargets
+        EXPORT spmTargets       
         ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
-        LIBRARY DESTINATION ${LIB_INSTALL_DIR})
+        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR})
 
 # Add files to the documentation
 # ------------------------------
