--- a/.gitmodules
+++ b/.gitmodules
@@ -1,12 +1,12 @@
 [submodule "cmake_modules/morse_cmake"]
 	path = cmake_modules/morse_cmake
-	url = ../../solverstack/morse_cmake.git
+	url = https://gitlab.inria.fr/solverstack/morse_cmake.git
 	fetchRecurseSubmodules = true
 	ignored = untracked
 	update = checkout
 [submodule "spm"]
 	path = spm
-	url = ../../solverstack/spm.git
+	url = https://gitlab.inria.fr/solverstack/spm.git
 	fetchRecurseSubmodules = true
 	ignored = untracked
 	update = checkout
