From 1f3bb8df568894d0f9c14785ffb4b41de484b641 Mon Sep 17 00:00:00 2001
From: Evan Klitzke <evan@eklitzke.org>
Date: Wed, 23 Jun 2021 11:23:58 -0700
Subject: [PATCH] Make spdlog_ex::spdlog_ex work with fmt 8.0.0

---
 include/spdlog/common-inl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/spdlog/common-inl.h b/include/spdlog/common-inl.h
index c3e992224..5fa4e5d95 100644
--- a/include/spdlog/common-inl.h
+++ b/include/spdlog/common-inl.h
@@ -56,7 +56,7 @@ SPDLOG_INLINE spdlog_ex::spdlog_ex(std::string msg)
 SPDLOG_INLINE spdlog_ex::spdlog_ex(const std::string &msg, int last_errno)
 {
     memory_buf_t outbuf;
-    fmt::format_system_error(outbuf, last_errno, msg);
+    fmt::format_system_error(outbuf, last_errno, msg.c_str());
     msg_ = fmt::to_string(outbuf);
 }
