--- libdca-svn-r91/libdca/Makefile.am.orig	2014-05-04 00:58:18.652059800 +0000
+++ libdca-svn-r91/libdca/Makefile.am	2014-05-04 01:00:00.568501600 +0000
@@ -5,7 +5,7 @@
 libdca_la_SOURCES = dca_internal.h tables.h tables_quantization.h \
 		    tables_huffman.h tables_fir.h tables_adpcm.h tables_vq.h \
 		    bitstream.h bitstream.c parse.c downmix.c
-libdca_la_LDFLAGS = -no-undefined -version 0:0:0 -export-symbol-regex dca_.*
+libdca_la_LDFLAGS = -no-undefined -version 0:0:0 -export-symbol-regex '^dca_.*'
 libdca_la_LIBADD = $(LIBDCA_LIBS)
 
 pkgconfigdir = $(libdir)/pkgconfig
