--- gtkspell-2.0.16/gtkspell/gtkspell.c	2009-10-09 21:01:47.000000000 +0200
+++ gtkspell-2.0.16/gtkspell/gtkspell.c	2017-04-08 15:24:36.476426500 +0200
@@ -740,7 +740,13 @@
 	GtkSpell *spell;
 
 #ifdef ENABLE_NLS
+#ifdef G_OS_WIN32
+	gchar *dir = g_win32_get_package_installation_directory_of_module(NULL);
+	bindtextdomain(PACKAGE, g_build_filename (dir, "share", "locale", NULL));
+	g_free(dir);
+#else
 	bindtextdomain(PACKAGE, LOCALEDIR);
+#endif
 	bind_textdomain_codeset(PACKAGE, "UTF-8");
 #endif
 
