--- openjpeg-version.1.5.2/libopenjpeg/openjpeg.h.orig	2016-02-25 16:30:48.099002000 +0300
+++ openjpeg-version.1.5.2/libopenjpeg/openjpeg.h	2016-02-25 16:32:10.464040800 +0300
@@ -40,11 +40,15 @@
 ==========================================================
 */
 
+#if defined(_WIN32)
+#define OPJ_CALLCONV __stdcall
+#else
+#define OPJ_CALLCONV
+#endif
+
 #if defined(OPJ_STATIC) || !defined(_WIN32)
 #define OPJ_API
-#define OPJ_CALLCONV
 #else
-#define OPJ_CALLCONV __stdcall
 /*
 The following ifdef block is the standard way of creating macros which make exporting 
 from a DLL simpler. All files within this DLL are compiled with the OPJ_EXPORTS
@@ -58,7 +62,7 @@
 #else
 #define OPJ_API __declspec(dllimport)
 #endif /* OPJ_EXPORTS */
-#endif /* !OPJ_STATIC || !_WIN32 */
+#endif /* OPJ_STATIC || !_WIN32 */
 
 typedef int opj_bool;
 #define OPJ_TRUE 1
