--- oiio-Release-2.2.7.0.orig/src/field3d.imageio/CMakeLists.txt
+++ oiio-Release-2.2.7.0/src/field3d.imageio/CMakeLists.txt
@@ -10,6 +10,6 @@
     add_oiio_plugin (field3dinput.cpp field3doutput.cpp
                      INCLUDE_DIRS ${FIELD3D_INCLUDES}
                      LINK_LIBRARIES Field3D::Field3D
-                                    # ${HDF5_LIBRARIES}
+                                    ${HDF5_LIBRARIES}
                                     ${SZIP_LIBRARY})
 endif()
