--- ./configure.in.orig	2013-11-03 17:56:29.000000000 +0100
+++ ./configure.in	2015-02-22 19:47:45.372532600 +0100
@@ -94,8 +94,13 @@
 if test "x$enable_python" = "xauto"; then
 	PKG_CHECK_MODULES(PYTHON, [pygobject-3.0 >= $PYGOBJECT_REQUIRED])
 
-	pyoverridesdir=`$PYTHON -c "import gi;print (gi._overridesdir)"`
-	AC_SUBST(pyoverridesdir)
+	pyoverridesdir=`$PYTHON -c "import gi; from os.path import abspath; print(abspath(gi._overridesdir))"`
+	case "$host" in
+          *-*-mingw*)
+            pyoverridesdir=`cygpath -u $pyoverridesdir`
+            ;;
+        esac
+        AC_SUBST(pyoverridesdir)
 fi
 
 AM_CONDITIONAL(ENABLE_PYTHON, test x"$enable_python" = "xyes")
