--- goocanvas-2.0.2/src/Makefile.am.orig	2013-10-25 17:45:33.000000000 +0200
+++ goocanvas-2.0.2/src/Makefile.am	2015-01-13 00:13:55.110240700 +0100
@@ -1,9 +1,9 @@
 ## Process this file with automake to produce Makefile.in
 
 INCLUDES = \
-	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
-	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
-	-DG_LOG_DOMAIN=\"GooCanvas\" \
+	-DPACKAGE_DATA_DIR="$(datadir)" \
+	-DPACKAGE_LOCALE_DIR="$(prefix)/$(DATADIRNAME)/locale" \
+	-DG_LOG_DOMAIN=\'GooCanvas\' \
 	@PACKAGE_CFLAGS@
 
 #	-DG_DISABLE_DEPRECATED -DPANGO_DISABLE_DEPRECATED \
@@ -158,7 +158,7 @@
 GooCanvas_2_0_gir_SCANNERFLAGS = --identifier-prefix=Goo --symbol-prefix=goo --warn-all
 GooCanvas_2_0_gir_INCLUDES = Gtk-3.0
 GooCanvas_2_0_gir_CFLAGS = $(INCLUDES) -I$(top_srcdir)/src
-GooCanvas_2_0_gir_LIBS = goocanvas-2.0
+GooCanvas_2_0_gir_LIBS = $(srcdir)/$(lib_LTLIBRARIES)
 GooCanvas_2_0_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += GooCanvas-2.0.gir
 
