diff --git a/Texconv/texconv.cpp b/Texconv/texconv.cpp
index 282fb69..9d12cb6 100644
--- a/Texconv/texconv.cpp
+++ b/Texconv/texconv.cpp
@@ -3856,11 +3856,7 @@ int __cdecl wmain(_In_ int argc, _In_z_count_(argc) wchar_t* argv[])
                                     {
                                         options.pstrName = const_cast<wchar_t*>(L"HeifCompressionMethod");
                                         varValues.vt = VT_UI1;
-                                    #if defined(NTDDI_WIN10_CU)
-                                        varValues.bVal = WICHeifCompressionNone;
-                                    #else
                                         varValues.bVal = 0x1 /* WICHeifCompressionNone */;
-                                    #endif
                                     }
                                     else if (wicQuality >= 0.f)
                                     {
