From 2a308d0791f653220a22b824abe436860ce3519c Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 4 Feb 2015 23:11:31 +0000
Subject: [PATCH] tempname.h: Rename try to try_

---
 lib/tempname.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tempname.h b/lib/tempname.h
index f7c98d0..158c22e 100644
--- a/lib/tempname.h
+++ b/lib/tempname.h
@@ -52,6 +52,6 @@ extern int gen_tempname (char *tmpl, int suffixlen, int flags, int kind);
    returns with this value.  Otherwise, if errno is set to EEXIST, another
    name is tried, or else TRY_GEN_TEMPNAME returns -1. */
 extern int try_tempname(char *tmpl, int suffixlen, void *args,
-			int (*try) (char *, void *));
+			int (*try_) (char *, void *));
 
 #endif /* GL_TEMPNAME_H */
-- 
2.2.2

