diff -urN octave.orig/configure.ac octave/configure.ac
--- octave.orig/configure.ac	2015-02-05 12:35:51.102143700 +0000
+++ octave/configure.ac	2015-02-05 13:07:58.632643700 +0000
@@ -2800,7 +2800,7 @@
 
   warn_gui=""
   ## Check for Qt libraries
-  QT_MODULES="QtCore QtGui QtNetwork QtOpenGL"
+  QT_MODULES="Qt5Core Qt5Gui Qt5Network Qt5OpenGL Qt5PrintSupport"
   PKG_CHECK_MODULES(QT, [$QT_MODULES],
     [],
     [warn_gui="Qt libraries not found -- disabling GUI"
@@ -2826,18 +2826,18 @@
       ;;
     esac
 
-    ## Check for Qt4
-    if ! `$PKG_CONFIG --atleast-version=4.0.0 QtCore`; then
-      warn_gui="Qt >= 4.0.0 not found -- disabling GUI"
+    ## Check for Qt5
+    if ! `$PKG_CONFIG --atleast-version=5.0.0 Qt5Core`; then
+      warn_gui="Qt >= 5.0.0 not found -- disabling GUI"
       build_gui=no
     fi
   fi
 
   if test $build_gui = yes; then
-    AC_CHECK_TOOLS(MOC, [moc-qt5 moc-qt4 moc])
-    AC_CHECK_TOOLS(UIC, [uic-qt5 uic-qt4 uic])
-    AC_CHECK_TOOLS(RCC, [rcc-qt5 rcc-qt4 rcc])
-    AC_CHECK_TOOLS(LRELEASE, [lrelease-qt5 lrelease-qt4 lrelease])
+    AC_CHECK_TOOLS(MOC, [moc-qt5 moc moc-qt4])
+    AC_CHECK_TOOLS(UIC, [uic-qt5 uic uic-qt4])
+    AC_CHECK_TOOLS(RCC, [rcc-qt5 rcc rcc-qt4])
+    AC_CHECK_TOOLS(LRELEASE, [lrelease-qt5 lrelease lrelease-qt4])
     if test -n "$MOC" && test -n "$UIC" && test -n "$RCC" && test -n "$LRELEASE"; then
       AC_DEFINE(HAVE_QT, 1,
         [Define to 1 if Qt is available (libraries, developer header files, utility programs (moc, uic, rcc, and lrelease))])
diff -urN octave.orig/libgui/graphics/Canvas.cc octave/libgui/graphics/Canvas.cc
--- octave.orig/libgui/graphics/Canvas.cc	2015-02-05 12:35:51.107143700 +0000
+++ octave/libgui/graphics/Canvas.cc	2015-02-05 13:07:58.636143700 +0000
@@ -210,7 +210,7 @@
               QRectF r (bb(0), bb(1), bb(2), bb(3));
 
               r.adjust (-5, -5, 5, 5);
-              if (r.contains (event->posF ()))
+              if (r.contains (event->localPos ()))
                 {
                   currentObj = childObj;
                   break;
@@ -237,7 +237,7 @@
                   Matrix bb = it->get_properties ().get_boundingbox (true);
                   QRectF r (bb(0), bb(1), bb(2), bb(3));
 
-                  if (r.contains (event->posF ()))
+                  if (r.contains (event->localPos ()))
                     axesObj = *it;
                 }
 
diff -urN octave.orig/libgui/graphics/Object.cc octave/libgui/graphics/Object.cc
--- octave.orig/libgui/graphics/Object.cc	2015-02-05 12:35:51.111143700 +0000
+++ octave/libgui/graphics/Object.cc	2015-02-05 13:07:58.640143700 +0000
@@ -154,7 +154,7 @@
   QVariant v = obj->property ("QtHandles::Object");
 
   if (v.isValid ())
-    return reinterpret_cast<Object*> (qVariantValue<void*> (v));
+    return reinterpret_cast<Object*> (v.value<void*> ());
 
   return 0;
 }
diff -urN octave.orig/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp octave/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- octave.orig/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp	2015-02-05 12:34:50.434143700 +0000
+++ octave/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp	2015-02-05 13:07:58.686143700 +0000
@@ -39,6 +39,7 @@
 #include <QDragEnterEvent>
 #include <QDropEvent>
 #include <QUrl>
+#include <QMimeData>
 
 #include <fcntl.h>
 #include <io.h>
diff -urN octave.orig/libgui/src/find-files-dialog.cc octave/libgui/src/find-files-dialog.cc
--- octave.orig/libgui/src/find-files-dialog.cc	2015-02-05 12:35:51.116143700 +0000
+++ octave/libgui/src/find-files-dialog.cc	2015-02-05 13:07:58.643143700 +0000
@@ -125,7 +125,7 @@
   _file_list->horizontalHeader ()->restoreState (
     settings->value ("findfiles/column_state").toByteArray ());
   _file_list->horizontalHeader ()->setSortIndicatorShown (true);
-  _file_list->horizontalHeader ()->setClickable (true);
+  _file_list->horizontalHeader ()->setSectionsClickable (true);
   _file_list->horizontalHeader ()->setStretchLastSection (true);
   _file_list->sortByColumn (
                 settings->value ("findfiles/sort_files_by_column",0).toInt (),
diff -urN octave.orig/libgui/src/m-editor/file-editor.cc octave/libgui/src/m-editor/file-editor.cc
--- octave.orig/libgui/src/m-editor/file-editor.cc	2015-02-05 12:35:51.121143700 +0000
+++ octave/libgui/src/m-editor/file-editor.cc	2015-02-05 13:07:58.646643700 +0000
@@ -41,6 +41,7 @@
 #include <QTabBar>
 #include <QProcess>
 #include <QInputDialog>
+#include <QMimeData>
 #include <Qsci/qscicommandset.h>
 
 #include "octave-link.h"
diff -urN octave.orig/libgui/src/m-editor/file-editor-tab.cc octave/libgui/src/m-editor/file-editor-tab.cc
--- octave.orig/libgui/src/m-editor/file-editor-tab.cc	2015-02-05 12:35:51.126143700 +0000
+++ octave/libgui/src/m-editor/file-editor-tab.cc	2015-02-05 13:07:58.651143700 +0000
@@ -445,9 +445,8 @@
       bool update_apis_file = false;  // flag, whether update of apis files
 
       // get path to prepared api info
-      QDesktopServices desktopServices;
       QString prep_apis_path
-        = desktopServices.storageLocation (QDesktopServices::HomeLocation)
+        = QStandardPaths::writableLocation (QStandardPaths::HomeLocation)
           + "/.config/octave/"  + QString(OCTAVE_VERSION) + "/qsci/";
 
       // get settings which infos are used for octave
@@ -482,7 +481,7 @@
               // compare to local package list
               // FIXME: How to get user chosen location?
               QFileInfo local_pkg_list = QFileInfo (
-                desktopServices.storageLocation (QDesktopServices::HomeLocation)
+                QStandardPaths::writableLocation (QStandardPaths::HomeLocation)
                 + "/.octave_packages");
               if (local_pkg_list.exists ()
                   & (apis_date < local_pkg_list.lastModified ()) )
@@ -1351,7 +1350,7 @@
 QsciScintilla::EolMode
 file_editor_tab::detect_eol_mode ()
 {
-  char *text = _edit_area->text ().toAscii ().data ();
+  char *text = _edit_area->text ().toLatin1 ().data ();
   int text_size = _edit_area->text ().length ();
 
   char eol_lf = 0x0a;
diff -urN octave.orig/libgui/src/m-editor/find-dialog.cc octave/libgui/src/m-editor/find-dialog.cc
--- octave.orig/libgui/src/m-editor/find-dialog.cc	2015-02-05 12:35:51.131143700 +0000
+++ octave/libgui/src/m-editor/find-dialog.cc	2015-02-05 13:07:58.659643700 +0000
@@ -66,6 +66,13 @@
 
 #include <QtGui>
 #include <QIcon>
+#include <QLabel>
+#include <QLineEdit>
+#include <QCheckBox>
+#include <QPushButton>
+#include <QDialogButtonBox>
+#include <QVBoxLayout>
+#include <QMessageBox>
 #include "find-dialog.h"
 
 find_dialog::find_dialog (QsciScintilla* edit_area, QWidget *p)
diff -urN octave.orig/libgui/src/octave-gui.cc octave/libgui/src/octave-gui.cc
--- octave.orig/libgui/src/octave-gui.cc	2015-02-05 12:35:51.135143700 +0000
+++ octave/libgui/src/octave-gui.cc	2015-02-05 13:07:58.662643700 +0000
@@ -89,7 +89,7 @@
 // Disable all Qt messages by default.
 
 static void
-message_handler (QtMsgType, const char *)
+message_handler (QtMsgType, const QMessageLogContext&, const QString&)
 {
 }
 
@@ -105,7 +105,7 @@
 
   // Installing our handler suppresses the messages.
   if (show_gui_msgs.empty ())
-    qInstallMsgHandler (message_handler);
+    qInstallMessageHandler (message_handler);
 
   if (start_gui)
     {
@@ -113,7 +113,7 @@
       QTranslator gui_tr, qt_tr, qsci_tr;
 
       // Set the codec for all strings (before wizard)
-      QTextCodec::setCodecForCStrings (QTextCodec::codecForName ("UTF-8"));
+      QTextCodec::setCodecForLocale (QTextCodec::codecForName ("UTF-8"));
 
       // show wizard if this is the first run
       if (resource_manager::is_first_run ())
diff -urN octave.orig/libgui/src/resource-manager.cc octave/libgui/src/resource-manager.cc
--- octave.orig/libgui/src/resource-manager.cc	2015-02-05 12:35:51.138143700 +0000
+++ octave/libgui/src/resource-manager.cc	2015-02-05 13:07:58.666143700 +0000
@@ -61,10 +61,8 @@
   : settings_directory (), settings_file (), settings (0),
     default_settings (0)
 {
-  QDesktopServices desktopServices;
-
   QString home_path
-    = desktopServices.storageLocation (QDesktopServices::HomeLocation);
+    = QStandardPaths::writableLocation (QStandardPaths::HomeLocation);
 
   settings_directory = home_path + "/.config/octave";
 
diff -urN octave.orig/libgui/src/resource-manager.h octave/libgui/src/resource-manager.h
--- octave.orig/libgui/src/resource-manager.h	2015-02-05 12:35:51.141143700 +0000
+++ octave/libgui/src/resource-manager.h	2015-02-05 13:07:58.669643700 +0000
@@ -24,6 +24,7 @@
 #define RESOURCEMANAGER_H
 
 #include <QDesktopServices>
+#include <QStandardPaths>
 #include <QIcon>
 #include <QMap>
 #include <QSettings>
diff -urN octave.orig/libgui/src/shortcut-manager.cc octave/libgui/src/shortcut-manager.cc
--- octave.orig/libgui/src/shortcut-manager.cc	2015-02-05 12:35:51.145143700 +0000
+++ octave/libgui/src/shortcut-manager.cc	2015-02-05 13:07:58.674143700 +0000
@@ -86,7 +86,7 @@
 void
 shortcut_manager::do_init_data ()
 {
-  QKeySequence ctrl;
+  Qt::KeyboardModifier ctrl;
   int prefix;
 #if defined (Q_OS_MAC)
   // Use CMD key as an equivalent of Ctrl key on other platforms
@@ -100,15 +100,14 @@
   prefix = Qt::NoModifier;
 #endif
 
-  QKeySequence ctrl_shift = ctrl + Qt::ShiftModifier;
-  QKeySequence ctrl_alt = ctrl + Qt::AltModifier;
-
+  Qt::KeyboardModifier shift = Qt::ShiftModifier;
+  
   // actions of the main window
 
   // file
   init (tr ("New File"), "main_file:new_file", QKeySequence::New);
   init (tr ("New Function"), "main_file:new_function",
-        QKeySequence (ctrl_shift + Qt::Key_N));
+        QKeySequence (ctrl + shift + Qt::Key_N));
   init (tr ("New Figure"), "main_file:new_figure", QKeySequence ());
   init (tr ("Open File"), "main_file:open_file", QKeySequence::Open);
   init (tr ("Load Workspace"), "main_file:load_workspace", QKeySequence ());
@@ -123,7 +122,7 @@
   init (tr ("Select All"), "main_edit:select_all", QKeySequence ());
   init (tr ("Clear Clipboard"), "main_edit:clear_clipboard", QKeySequence ());
   init (tr ("Find in Files"), "main_edit:find_in_files",
-        QKeySequence (ctrl_shift + Qt::Key_F));
+        QKeySequence (ctrl + shift + Qt::Key_F));
   init (tr ("Clear Command Window"), "main_edit:clear_command_window",
         QKeySequence ());
   init (tr ("Clear Command History"), "main_edit:clear_history",
@@ -144,17 +143,17 @@
 
   // window
   init (tr ("Show Command Window"), "main_window:show_command",
-        prefix + ctrl_shift + Qt::Key_0);
+        prefix + ctrl + shift + Qt::Key_0);
   init (tr ("Show Command History"), "main_window:show_history",
-        prefix + ctrl_shift + Qt::Key_1);
+        prefix + ctrl + shift + Qt::Key_1);
   init (tr ("Show File Browser"), "main_window:show_file_browser",
-        prefix + ctrl_shift + Qt::Key_2);
+        prefix + ctrl + shift + Qt::Key_2);
   init (tr ("Show Workspace"), "main_window:show_workspace",
-        prefix + ctrl_shift + Qt::Key_3);
+        prefix + ctrl + shift + Qt::Key_3);
   init (tr ("Show Editor"), "main_window:show_editor",
-        prefix + ctrl_shift + Qt::Key_4);
+        prefix + ctrl + shift + Qt::Key_4);
   init (tr ("Show Documentation"), "main_window:show_doc",
-        prefix + ctrl_shift + Qt::Key_5);
+        prefix + ctrl + shift + Qt::Key_5);
   init (tr ("Command Window"), "main_window:command",
         prefix + ctrl + Qt::Key_0);
   init (tr ("Command History"), "main_window:history",
@@ -209,15 +208,15 @@
   init (tr ("Delete to End of Word"), "editor_edit:delete_end_word",
         QKeySequence::DeleteEndOfWord);
   init (tr ("Delete to Start of Line"), "editor_edit:delete_start_line",
-        QKeySequence (ctrl_shift + Qt::Key_Backspace));
+        QKeySequence (ctrl + shift + Qt::Key_Backspace));
   init (tr ("Delete to End of Line"), "editor_edit:delete_end_line",
-        QKeySequence (ctrl_shift + Qt::Key_Delete));
+        QKeySequence (ctrl + shift + Qt::Key_Delete));
   init (tr ("Delete Line"), "editor_edit:delete_line",
-        QKeySequence (ctrl_shift + Qt::Key_L));
+        QKeySequence (ctrl + shift + Qt::Key_L));
   init (tr ("Copy Line"), "editor_edit:copy_line",
-        QKeySequence (ctrl_shift + Qt::Key_C));
+        QKeySequence (ctrl + shift + Qt::Key_C));
   init (tr ("Cut Line"), "editor_edit:cut_line",
-        QKeySequence (ctrl_shift + Qt::Key_X));
+        QKeySequence (ctrl + shift + Qt::Key_X));
   init (tr ("Duplicate Selection/Line"), "editor_edit:duplicate_selection",
         QKeySequence (ctrl + Qt::Key_D));
   init (tr ("Transpose Line"), "editor_edit:transpose_line",
@@ -228,22 +227,22 @@
   init (tr ("Comment Selection"), "editor_edit:comment_selection",
         QKeySequence (ctrl + Qt::Key_R));
   init (tr ("Uncomment Selection"), "editor_edit:uncomment_selection",
-        QKeySequence (ctrl_shift + Qt::Key_R));
+        QKeySequence (ctrl + shift + Qt::Key_R));
   init (tr ("Uppercase Selection"), "editor_edit:upper_case",
         QKeySequence (ctrl + Qt::Key_U));
   init (tr ("Lowercase Selection"), "editor_edit:lower_case",
-        QKeySequence (ctrl_alt + Qt::Key_U));
+        QKeySequence (ctrl + Qt::AltModifier + Qt::Key_U));
 
 #if defined (Q_OS_MAC)
   init (tr ("Indent Selection"), "editor_edit:indent_selection",
         QKeySequence (prefix + Qt::Key_Tab));
   init (tr ("Unindent Selection"), "editor_edit:unindent_selection",
-        QKeySequence (prefix + Qt::ShiftModifier + Qt::Key_Tab));
+        QKeySequence (prefix + shift + Qt::Key_Tab));
 #else
   init (tr ("Indent Selection"), "editor_edit:indent_selection",
         QKeySequence (ctrl + Qt::Key_Tab));
   init (tr ("Unindent Selection"), "editor_edit:unindent_selection",
-        QKeySequence (ctrl_shift + Qt::Key_Tab));
+        QKeySequence (ctrl + shift + Qt::Key_Tab));
 #endif
 
   init (tr ("Convert Line Ednings to Windows"), "editor_edit:conv_eol_winows",
@@ -258,7 +257,7 @@
   init (tr ("Move to Matching Brace"), "editor_edit:move_to_brace",
         QKeySequence (ctrl + Qt::Key_M));
   init (tr ("Select to Matching Brace"), "editor_edit:select_to_brace",
-        QKeySequence (ctrl_shift + Qt::Key_M));
+        QKeySequence (ctrl + shift + Qt::Key_M));
   init (tr ("Toggle Bookmark"), "editor_edit:toggle_bookmark",
         QKeySequence (prefix + Qt::Key_F7));
   init (tr ("Next Bookmark"), "editor_edit:next_bookmark",
@@ -285,7 +284,7 @@
         QKeySequence (ctrl + Qt::Key_Underscore));
 #else
   init (tr ("Zoom Normal"), "editor_view:zoom_normal",
-        QKeySequence (ctrl_alt + Qt::Key_0));
+        QKeySequence (ctrl + Qt::AltModifier + Qt::Key_0));
 #endif
 
   // debug
@@ -340,7 +339,7 @@
   _dialog = 0;
   _level_hash.clear ();
 
-  tree_view->header ()->setResizeMode (QHeaderView::ResizeToContents);
+  tree_view->header ()->setSectionResizeMode (QHeaderView::ResizeToContents);
 
   QTreeWidgetItem *main = new QTreeWidgetItem (tree_view);
   main->setText (0, tr ("Main"));
@@ -406,10 +405,10 @@
 
       // write the shortcuts
       tree_item->setText (0, sc.description);
-      tree_item->setText (1, sc.default_sc [0]);
-      tree_item->setText (2, sc.actual_sc [0]);
-      tree_item->setText (3, sc.default_sc [1]);
-      tree_item->setText (4, sc.actual_sc [1]);
+      tree_item->setText (1, sc.default_sc [0].toString ());
+      tree_item->setText (2, sc.actual_sc [0].toString ());
+      tree_item->setText (3, sc.default_sc [1].toString ());
+      tree_item->setText (4, sc.actual_sc [1].toString ());
 
       _item_index_hash[tree_item] = i + 1; // index+1 to avoid 0
       _index_item_hash[i] = tree_item;
@@ -549,8 +548,8 @@
 
     }
 
-  _edit_actual->setText (_sc.at (index).actual_sc[_selected_set]);
-  _label_default->setText (_sc.at (index).default_sc[_selected_set]);
+  _edit_actual->setText (_sc.at (index).actual_sc[_selected_set].toString ());
+  _label_default->setText (_sc.at (index).default_sc[_selected_set].toString ());
   _handled_index = index;
 
   _edit_actual->setFocus ();
@@ -591,7 +590,7 @@
           shortcut_t double_shortcut = _sc.at (double_index);
           double_shortcut.actual_sc[_selected_set] = QKeySequence ();
           _sc.replace (double_index, double_shortcut);
-          _index_item_hash[double_index]->setText ((_selected_set + 1)*2, QKeySequence ());
+          _index_item_hash[double_index]->setText ((_selected_set + 1)*2, QKeySequence ().toString ());
         }
       else
         return;
@@ -604,7 +603,7 @@
   _sc.replace (_handled_index, shortcut);
 
   _index_item_hash[_handled_index]->setText ((_selected_set + 1)*2,
-                                             shortcut.actual_sc[_selected_set]);
+                                             shortcut.actual_sc[_selected_set].toString ());
 
   if (! shortcut.actual_sc[_selected_set].isEmpty ())
     _shortcut_hash[widget + sep + shortcut.actual_sc[_selected_set].toString ()] =
@@ -632,7 +631,7 @@
 
       // update the tree view
       QTreeWidgetItem* tree_item = _index_item_hash[i]; // get related tree item
-      tree_item->setText (2*set, sc.actual_sc [set-1]); // display new shortcut
+      tree_item->setText (2*set, sc.actual_sc [set-1].toString ()); // display new shortcut
     }
 }
 
@@ -723,6 +722,6 @@
       if(modifiers & Qt::MetaModifier)
         key += Qt::META;
 
-      setText (QKeySequence(key));
+      setText (QKeySequence(key).toString ());
     }
 }
diff -urN octave.orig/libgui/src/workspace-model.cc octave/libgui/src/workspace-model.cc
--- octave.orig/libgui/src/workspace-model.cc	2015-02-05 12:35:51.149143700 +0000
+++ octave/libgui/src/workspace-model.cc	2015-02-05 13:07:58.682143700 +0000
@@ -132,7 +132,7 @@
         {
           QString class_chars = resource_manager::storage_class_chars ();
           int actual_class
-            = class_chars.indexOf (_scopes[idx.row ()].toAscii ());
+            = class_chars.indexOf (_scopes[idx.row ()].toLatin1 ());
           if (actual_class >= 0)
             return QVariant (_storage_class_colors.at (actual_class));
           else
@@ -172,7 +172,7 @@
                 QString class_chars = resource_manager::storage_class_chars ();
 
                 int actual_class
-                  = class_chars.indexOf (_scopes[idx.row ()].toAscii ());
+                  = class_chars.indexOf (_scopes[idx.row ()].toLatin1 ());
 
                 if (actual_class >= 0)
                   {
