diff -urN libimobiledevice-1.2.0.orig/cython/debugserver.pxi libimobiledevice-1.2.0/cython/debugserver.pxi
--- libimobiledevice-1.2.0.orig/cython/debugserver.pxi	2015-07-22 22:43:52.357279900 +0100
+++ libimobiledevice-1.2.0/cython/debugserver.pxi	2015-07-22 22:44:40.303768000 +0100
@@ -44,13 +44,13 @@
 
 
 # from http://stackoverflow.com/a/17511714
-from cpython.string cimport PyString_AsString
+from cpython.bytes cimport PyBytes_AsString
 cdef char ** to_cstring_array(list_str):
     if not list_str:
         return NULL
     cdef char **ret = <char **>malloc(len(list_str) * sizeof(char *))
     for i in xrange(len(list_str)):
-        ret[i] = PyString_AsString(list_str[i])
+        ret[i] = PyBytes_AsString(list_str[i])
     return ret
 
 
