--- libimobiledevice-1.2.0/configure.ac.orig	2020-04-27 09:34:04.717441300 +0300
+++ libimobiledevice-1.2.0/configure.ac	2020-04-27 09:36:01.274032100 +0300
@@ -101,6 +101,7 @@
             if [test ! -d "$CYTHON_PLIST_INCLUDE_DIR"]; then
                 CYTHON=false
                 CYTHON_SUB=
+                PYTHON_SO=
                 cython_python_bindings=no
                 AC_MSG_RESULT([no])
                 AC_MSG_WARN([cannot find libplist Cython bindings. You should install your distribution specific libplist Cython bindings package.])
@@ -109,13 +110,16 @@
                 AC_MSG_RESULT([$CYTHON_PLIST_INCLUDE_DIR])
                 CYTHON_SUB=cython
                 cython_python_bindings=yes
+                PYTHON_SO=`$PYTHON -c "import distutils.sysconfig, sys; get = distutils.sysconfig.get_config_var; sys.stdout.write(get('EXT_SUFFIX') or get('SO') or '.so');"`
             fi
 else
             CYTHON_SUB=
+            PYTHON_SO=
             cython_python_bindings=no
 fi
 AM_CONDITIONAL([HAVE_CYTHON],[test "x$CYTHON_SUB" = "xcython"])
 AC_SUBST([CYTHON_SUB])
+AC_SUBST(PYTHON_SO)
 
 AC_ARG_ENABLE([openssl],
             [AS_HELP_STRING([--disable-openssl],
--- libimobiledevice-1.2.0/cython/Makefile.am.orig	2020-04-27 09:36:22.294151400 +0300
+++ libimobiledevice-1.2.0/cython/Makefile.am	2020-04-27 09:36:48.336869800 +0300
@@ -38,7 +38,7 @@
 imobiledevice_LTLIBRARIES = imobiledevice.la
 imobiledevice_la_SOURCES = imobiledevice.pyx
 imobiledevice_la_CFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src $(PYTHON_CPPFLAGS) $(AM_CFLAGS) -Wno-shadow -Wno-redundant-decls -Wno-switch-default -Wno-strict-aliasing -Wno-implicit-function-declaration -fvisibility=default
-imobiledevice_la_LDFLAGS = -module -avoid-version -L$(libdir) $(PYTHON_LDFLAGS) $(AM_LDFLAGS) -no-undefined
+imobiledevice_la_LDFLAGS = -module -avoid-version -L$(libdir) $(PYTHON_LDFLAGS) $(AM_LDFLAGS) -no-undefined -shrext $(PYTHON_SO)
 imobiledevice_la_LIBADD = $(top_builddir)/src/libimobiledevice.la
 otherincludedir = $(includedir)/libimobiledevice/cython
 otherinclude_HEADERS = imobiledevice.pxd
