--- libimobiledevice-1.3.0/cython/debugserver.pxi.orig	2025-05-20 08:37:15.525874600 +0200
+++ libimobiledevice-1.3.0/cython/debugserver.pxi	2025-05-20 09:50:35.309731500 +0200
@@ -43,19 +43,13 @@
         BaseError.__init__(self, *args, **kwargs)
 
 
-# from http://stackoverflow.com/a/17511714
-# https://github.com/libimobiledevice/libimobiledevice/pull/198
-from cpython cimport PY_MAJOR_VERSION
-if PY_MAJOR_VERSION <= 2:
-    from cpython.string cimport PyString_AsString
-else:
-    from cpython.bytes cimport PyBytes_AsString as PyString_AsString
+from cpython.bytes cimport PyBytes_AsString
 cdef char ** to_cstring_array(list_str):
     if not list_str:
         return NULL
     cdef char **ret = <char **>malloc(len(list_str) * sizeof(char *))
     for i in xrange(len(list_str)):
-        ret[i] = PyString_AsString(list_str[i])
+        ret[i] = PyBytes_AsString(list_str[i])
     return ret
 
 
