--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -691,7 +691,7 @@
 add_subdirectory(test/functional/fixtures)  # compile test programs
 add_subdirectory(runtime)
 get_directory_property(GENERATED_HELP_TAGS DIRECTORY runtime DEFINITION GENERATED_HELP_TAGS)
-if(WIN32)
+if(FALSE)
   install_helper(
     FILES ${DEPS_PREFIX}/share/nvim-qt/runtime/plugin/nvim_gui_shim.vim
     DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/nvim-qt/runtime/plugin)
--- a/src/nvim/CMakeLists.txt
+++ b/src/nvim/CMakeLists.txt
@@ -463,7 +463,7 @@
   endif()
 endif()
 
-if(WIN32)
+if(FALSE)
   # Copy DLLs and third-party tools to bin/ and install them along with nvim
   add_custom_target(nvim_runtime_deps ALL
     COMMAND ${CMAKE_COMMAND} -E copy_directory ${PROJECT_BINARY_DIR}/windows_runtime_deps/
