--- meson-0.47.1/mesonbuild/modules/python.py.orig	2018-07-02 19:41:43.000000000 +0200
+++ meson-0.47.1/mesonbuild/modules/python.py	2018-07-11 20:20:23.498385600 +0200
@@ -437,18 +437,23 @@
         super().__init__(*args, **kwargs)
         self.snippets.add('find_installation')
 
-    # https://www.python.org/dev/peps/pep-0397/
     def _get_win_pythonpath(self, name_or_path):
         if name_or_path not in ['python2', 'python3']:
             return None
+
+        # https://www.python.org/dev/peps/pep-0397/
         ver = {'python2': '-2', 'python3': '-3'}[name_or_path]
-        cmd = ['py', ver, '-c', "import sysconfig; print(sysconfig.get_config_var('BINDIR'))"]
-        _, stdout, _ = mesonlib.Popen_safe(cmd)
-        dir = stdout.strip()
-        if os.path.exists(dir):
-            return os.path.join(dir, 'python')
-        else:
-            return None
+        for launcher_args in [[name_or_path], ['py', ver]]:
+            try:
+                p, stdout = mesonlib.Popen_safe(
+                    launcher_args + [
+                        'import sys; sys.stdout.write(sys.executable)'])[:2]
+            except OSError:
+                continue
+            else:
+                if p.returncode == 0:
+                    return stdout.strip()
+        return None
 
     @permittedKwargs(['required'])
     def find_installation(self, interpreter, state, args, kwargs):
