--- meson-1.10.2/mesonbuild/interpreter/interpreter.py.orig	2026-03-15 14:05:37.000000000 +0100
+++ meson-1.10.2/mesonbuild/interpreter/interpreter.py	2026-03-17 09:05:37.665776300 +0100
@@ -701,7 +701,7 @@
         for k, v in variables.items():
             if not v:
                 FeatureNew.single_use('empty variable value in declare_dependency', '1.4.0', self.subproject, location=node)
-            if path_has_root(v) \
+            if os.path.isabs(v) \
                     and (self.is_subproject() or not is_parent_path(subproject_dir, v)) \
                     and is_parent_path(project_root, v) \
                     and os.path.isdir(v):
