--- meson-1.10.0/mesonbuild/options.py.orig	2025-12-08 17:48:52.000000000 +0100
+++ meson-1.10.0/mesonbuild/options.py	2026-01-08 08:26:59.395895700 +0100
@@ -973,7 +973,8 @@
 
     def sanitize_prefix(self, prefix: str) -> str:
         prefix = os.path.expanduser(prefix)
-        if not os.path.isabs(prefix):
+        seps = (os.sep, os.altsep) if os.altsep else (os.sep,)
+        if not os.path.isabs(prefix) and not prefix.startswith(seps):
             raise MesonException(f'prefix value {prefix!r} must be an absolute path')
         if prefix.endswith('/') or prefix.endswith('\\'):
             # On Windows we need to preserve the trailing slash if the
--- meson-1.10.0/mesonbuild/minstall.py.orig	2026-01-08 08:43:21.732769700 +0100
+++ meson-1.10.0/mesonbuild/minstall.py	2026-01-08 08:43:45.643037900 +0100
@@ -267,7 +267,8 @@
               'Standard error:', err, sep='\n')
 
 def get_destdir_path(destdir: str, fullprefix: str, path: str) -> str:
-    if os.path.isabs(path):
+    seps = (os.sep, os.altsep) if os.altsep else (os.sep,)
+    if os.path.isabs(path) or path.startswith(seps):
         output = destdir_join(destdir, path)
     else:
         output = os.path.join(fullprefix, path)
