--- meson-0.44.0/mesonbuild/mlog.py.orig	2017-12-11 11:27:42.817091100 +0100
+++ meson-0.44.0/mesonbuild/mlog.py	2017-12-11 11:31:42.639808200 +0100
@@ -19,7 +19,7 @@
 some to logging dir and some goes to both."""
 
 if platform.system().lower() == 'windows':
-    colorize_console = os.isatty(sys.stdout.fileno()) and os.environ.get('ANSICON')
+    colorize_console = os.isatty(sys.stdout.fileno())
 else:
     colorize_console = os.isatty(sys.stdout.fileno()) and os.environ.get('TERM') != 'dumb'
 log_dir = None
