--- meson-0.42.0/mesonbuild/compilers/c.py.orig	2017-08-13 20:43:17.000000000 +0200
+++ meson-0.42.0/mesonbuild/compilers/c.py	2017-08-15 17:37:46.853944200 +0200
@@ -593,6 +593,9 @@
         if self.get_id() == 'msvc':
             return False
 
+        if funcname in ['stpcpy', 'posix_memalign']:
+            return False
+
         # Detect function as a built-in
         #
         # Some functions like alloca() are defined as compiler built-ins which
