--- meson-0.41.1/mesonbuild/scripts/gtkdochelper.py.orig	2017-06-08 19:28:37.000000000 +0200
+++ meson-0.41.1/mesonbuild/scripts/gtkdochelper.py	2017-07-16 14:21:37.612414900 +0200
@@ -45,10 +45,25 @@
 parser.add_argument('--mode', dest='mode', default='')
 parser.add_argument('--installdir', dest='install_dir')
 
+
+def cyg2winpath(path):
+    assert os.name == "nt"
+    win_path = subprocess.check_output(
+        ["cygpath.exe", "-m" if os.sep == "/" else "-w", "--", path])
+    return win_path.decode("utf-8").strip()
+
+
+def cygcmd(args):
+    assert os.name == "nt"
+    return [cyg2winpath("/usr/bin/perl"),
+            cyg2winpath("/usr/bin/%s" % args[0])] + list(args[1:])
+
+
 def gtkdoc_run_check(cmd, cwd):
     # Put stderr into stdout since we want to print it out anyway.
     # This preserves the order of messages.
-    p, out = Popen_safe(cmd, cwd=cwd, stderr=subprocess.STDOUT)[0:2]
+    cmd = cygcmd(cmd)
+    p, out = Popen_safe(cmd, cwd=cwd, stderr=subprocess.STDOUT, encoding="utf-8", errors="replace")[0:2]
     if p.returncode != 0:
         err_msg = ["{!r} failed with status {:d}".format(cmd[0], p.returncode)]
         if out:
