diff --git i/run_unittests.py w/run_unittests.py
index 28264606..7c267e17 100755
--- i/run_unittests.py
+++ w/run_unittests.py
@@ -3441,7 +3441,7 @@ class LinuxlikeTests(BasePlatformTests):
                 break
         self.assertIsInstance(docbook_target, dict)
         ifile = self.introspect(['--target-files', 'generated-gdbus-docbook@cus'])[0]
-        self.assertEqual(t['filename'], 'gdbus/generated-gdbus-doc-' + ifile)
+        self.assertEqual(t['filename'], 'gdbus/generated-gdbus-doc-' + os.path.basename(ifile))
 
     def test_build_rpath(self):
         if is_cygwin():
