diff -bur log4cpp-orig/config/config.sub log4cpp/config/config.sub
--- log4cpp-orig/config/config.sub	2025-03-26 02:37:59.700200200 -0600
+++ log4cpp/config/config.sub	2025-03-26 02:38:44.153362800 -0600
@@ -248,7 +248,7 @@
 	# We use `pc' rather than `unknown'
 	# because (1) that's what they normally are, and
 	# (2) the word "unknown" tends to confuse beginning users.
-	i*86 | x86_64)
+	i*86 | x86_64 | aarch64)
 	  basic_machine=$basic_machine-pc
 	  ;;
 	# Object if more than one company name word.
@@ -260,7 +260,7 @@
 	# FIXME: clean up the formatting here.
 	vax-* | tahoe-* | i*86-* | i860-* | ia64-* | m32r-* | m68k-* | m68000-* \
 	      | m88k-* | sparc-* | ns32k-* | fx80-* | arc-* | c[123]* \
-	      | arm-*  | armbe-* | armle-* | armv*-* | strongarm-* | xscale-* \
+	      | arm-*  | armbe-* | armle-* | armv*-* | aarch64*-* | strongarm-* | xscale-* \
 	      | mips-* | pyramid-* | tron-* | a29k-* | romp-* | rs6000-* \
 	      | power-* | none-* | 580-* | cray2-* | h8300-* | h8500-* | i960-* \
 	      | xmp-* | ymp-* \
