--- src/src/libmongoc/CMakeLists.txt.orig	2026-03-03 09:18:30.000000000 -0600
+++ src/src/libmongoc/CMakeLists.txt	2026-03-11 14:13:30.309850000 -0500
@@ -981,7 +981,7 @@
    set_target_properties (mongoc_static PROPERTIES
       VERSION "${PROJECT_VERSION}"
       OUTPUT_NAME "${MONGOC_OUTPUT_BASENAME}${PROJECT_VERSION_MAJOR}"
-      pkg_config_REQUIRES "bson${PROJECT_VERSION_MAJOR}-static"
+      pkg_config_REQUIRES "bson${PROJECT_VERSION_MAJOR}-static zlib libzstd snappy libutf8proc"
       )
    if(MONGOC_ENABLE_STATIC_INSTALL)
       mongo_generate_pkg_config (mongoc_static FILENAME mongoc${PROJECT_VERSION_MAJOR}-static.pc INSTALL)
@@ -1360,8 +1360,7 @@

 # Collect link items for the static library to be inserted into the pkg-config
 if(TARGET mongoc_static)
-   set(link_options ${ZLIB_LIBRARIES}
-      ${SNAPPY_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY})
+   set(link_options ${LIBMONGOCRYPT_LIBRARY})
    # Replace all absolute paths with search-dir link-file options:
    list(TRANSFORM link_options
         REPLACE "^(.+)/lib([^/\\.]+)\\..+$"

