--- src/build/cmake/GeneratePkgConfig.cmake.orig	2026-03-03 09:18:30.000000000 -0600
+++ src/build/cmake/GeneratePkgConfig.cmake	2026-03-11 15:09:52.680977200 -0500
@@ -304,18 +304,8 @@
     set(has_defs $<NOT:$<STREQUAL:,${gx_defs}>>)
     set(def_joiner " -D")
     string(APPEND cflags $<${has_defs}:${def_joiner}$<JOIN:${gx_defs},${def_joiner}>>)
-    # Includes:
-    set(gx_inc $<GENEX_EVAL:$<TARGET_PROPERTY:pkg_config_INCLUDE_DIRECTORIES>>)
-    set(gx_inc "$<REMOVE_DUPLICATES:${gx_inc}>")
-    set(gx_abs_inc "$<FILTER:${gx_inc},INCLUDE,^/>")
-    set(gx_rel_inc "$<FILTER:${gx_inc},EXCLUDE,^/>")
-    set(has_abs_inc $<NOT:$<STREQUAL:,${gx_abs_inc}>>)
-    set(has_rel_inc $<NOT:$<STREQUAL:,${gx_rel_inc}>>)
-    string(APPEND cflags $<${has_rel_inc}: " -I\${prefix}/"
-                            $<JOIN:${gx_rel_inc}, " -I\${prefix}/" >>
-                         $<${has_abs_inc}: " -I"
-                            $<JOIN:${gx_abs_inc}, " -I" >>)
     string(APPEND content "Libs: ${libs}\n")
     string(APPEND content "Cflags: ${cflags}\n")
+    string(APPEND content "Libs.private: -lws2_32\n")
     set("${out}" "${content}" PARENT_SCOPE)
 endfunction()

