--- popt-1.16/popthelp.c.orig	2009-08-28 04:06:33 +0400
+++ popt-1.16/popthelp.c	2012-03-31 08:39:03 +0400
@@ -312,7 +312,7 @@
 {
     size_t maxLeftCol = columns->cur;
     size_t indentLength = maxLeftCol + 5;
-    size_t lineLength = columns->max - indentLength;
+    size_t lineLength = (indentLength >= columns->max) ? 1 : columns->max - indentLength;
     const char * help = D_(translation_domain, opt->descrip);
     const char * argDescrip = getArgDescrip(opt, translation_domain);
     /* Display shortName iff printable non-space. */
