--- popt-1.18/src/popt.c.orig	2021-05-28 20:43:50.210830500 -0700
+++ popt-1.18/src/popt.c	2021-05-28 20:44:00.459108900 -0700
@@ -510,7 +510,7 @@
     if (rc) goto exit;
     rc = setreuid(getuid(), getuid());
     if (rc) goto exit;
-#else
+#elif !defined(_WIN32)
     /* refuse to exec if we cannot drop suid/sgid privileges */
     if (getuid() != geteuid() || getgid() != getegid()) {
 	errno = ENOTSUP;
