--- libgxps-0.2.3.2/libgxps/Makefile.am.orig	2015-09-24 21:02:52.477919700 +0200
+++ libgxps-0.2.3.2/libgxps/Makefile.am	2015-09-24 21:06:22.835977000 +0200
@@ -63,7 +63,11 @@
 	$(LCMS2_CFLAGS)		\
 	$(LIBPNG_CFLAGS)	\
 	$(WARN_CFLAGS)		\
-	$(AM_CFLAGS)
+	$(AM_CFLAGS) \
+    $(LIBPNG_CFLAGS) \
+    $(CAIRO_PDF_CFLAGS) \
+    $(CAIRO_PS_CFLAGS) \
+    $(CAIRO_SVG_CFLAGS)
 
 libgxps_la_LDFLAGS = \
 	-version-info $(GXPS_LT_VERSION_INFO)   \
@@ -77,7 +81,11 @@
 	$(LIBPNG_LIBS)	\
 	$(LIBJPEG)	\
 	$(LIBM)         \
-	$(LIBTIFF)
+	$(LIBTIFF) \
+	$(LIBPNG_LIBS) \
+	$(CAIRO_PDF_LIBS) \
+	$(CAIRO_PS_LIBS) \
+	$(CAIRO_SVG_LIBS)
 
 EXTRA_DIST = \
 	gxps-version.h.in
@@ -111,4 +119,4 @@
 typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 
 CLEANFILES += $(gir_DATA) $(typelib_DATA)
-endif
\ Manca newline alla fine del file
+endif
