--- a/COFF/Driver.cpp
+++ b/COFF/Driver.cpp
@@ -1223,7 +1223,7 @@ static void findKeepUniqueSections(COFFLinkerContext &ctx) {
         if (err)
           fatal(toString(obj) + ": could not decode addrsig section: " + err);
         if (symIndex >= syms.size())
-          fatal(toString(obj) + ": invalid symbol index in addrsig section");
+          return;
         markAddrsig(syms[symIndex]);
         cur += size;
       }
