diff -u -r gtkspell-2.0.16/configure.ac gtkspell-2.0.16-enchant2/configure.ac
--- gtkspell-2.0.16/configure.ac	2009-10-23 04:52:31.000000000 +0200
+++ gtkspell-2.0.16-enchant2/configure.ac	2018-01-18 12:42:06.366410232 +0100
@@ -12,12 +12,12 @@
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
 AC_CONFIG_HEADERS([config.h])
 
-SPELLER_LIB=-lenchant
+SPELLER_LIB=-lenchant-2
       
 AC_SUBST(SPELLER_LIB)
 GTKSPELL_PACKAGES=gtk+-2.0
 AC_SUBST(GTKSPELL_PACKAGES)
-PKG_CHECK_MODULES(GTKSPELL, $GTKSPELL_PACKAGES enchant >= 0.4.0 )
+PKG_CHECK_MODULES(GTKSPELL, $GTKSPELL_PACKAGES enchant-2 >= 2.2.0 )
 AC_SUBST(GTKSPELL_CFLAGS)
 AC_SUBST(GTKSPELL_LIBS)
 
diff -u -r gtkspell-2.0.16/gtkspell/gtkspell.c gtkspell-2.0.16-enchant2/gtkspell/gtkspell.c
--- gtkspell-2.0.16/gtkspell/gtkspell.c	2009-10-09 21:01:47.000000000 +0200
+++ gtkspell-2.0.16-enchant2/gtkspell/gtkspell.c	2018-01-18 12:41:37.146338802 +0100
@@ -277,7 +277,7 @@
 	get_word_extents_from_mark(spell->buffer, &start, &end, spell->mark_click);
 	word = gtk_text_buffer_get_text(spell->buffer, &start, &end, FALSE);
 	
-	enchant_dict_add_to_pwl( spell->speller, word, strlen(word));
+	enchant_dict_add( spell->speller, word, strlen(word));
 
 	gtkspell_recheck_all(spell);
 
