diff -u -r ../src/ctags-5.8/general.h ./general.h
--- ../src/ctags-5.8/general.h	2007-05-03 05:21:08.000000000 +0200
+++ ./general.h	2014-10-14 23:57:39.251620300 +0200
@@ -56,7 +56,7 @@
 /*  This is a helpful internal feature of later versions (> 2.7) of GCC
  *  to prevent warnings about unused variables.
  */
-#if (__GNUC__ > 2  ||  (__GNUC__ == 2  &&  __GNUC_MINOR__ >= 7)) && !defined (__GNUG__)
+#if 0 && (__GNUC__ > 2  ||  (__GNUC__ == 2  &&  __GNUC_MINOR__ >= 7)) && !defined (__GNUG__)
 # define __unused__  __attribute__((unused))
 # define __printf__(s,f)  __attribute__((format (printf, s, f)))
 #else
