--- pdcurses-3.8/wincon/Makefile.orig	2019-02-19 08:38:11.617435700 +0300
+++ pdcurses-3.8/wincon/Makefile	2019-02-19 08:42:11.280656700 +0300
@@ -52,14 +52,14 @@
 ifeq ($(DLL),Y)
 	CFLAGS += -DPDC_DLL_BUILD
 	LIBEXE = $(CC)
-	LIBFLAGS = -Wl,--out-implib,pdcurses.a -shared -o
+	LIBFLAGS = -Wl,--out-implib,libpdcurses.dll.a -shared -o
 	LIBCURSES = pdcurses.dll
 	CLEAN = $(LIBCURSES) *.a
 	RESOURCE = pdcurses.o
 else
 	LIBEXE = $(AR)
 	LIBFLAGS = rcv
-	LIBCURSES = pdcurses.a
+	LIBCURSES = libpdcurses.a
 	CLEAN = *.a
 endif
 
@@ -69,7 +69,7 @@
 
 .PHONY: all libs clean demos dist
 
-all:	libs
+all:	libs demos
 
 libs:	$(LIBCURSES)
 
