From a2ff4e8c55199ec622d30dc2e5d49f01e37cd40e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20M=C3=BCller?= <nexyon@gmail.com>
Date: Thu, 17 Jan 2019 19:35:42 +0100
Subject: [PATCH 09/16] Bugfix: memory leak in python API.

---
 bindings/python/PySound.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/bindings/python/PySound.cpp b/bindings/python/PySound.cpp
index 2ab1974..82c6036 100644
--- a/bindings/python/PySound.cpp
+++ b/bindings/python/PySound.cpp
@@ -140,8 +140,6 @@ Sound_data(Sound* self)
 
 	std::memcpy(data, buffer->getBuffer(), buffer->getSize());
 
-	Py_INCREF(array);
-
 	return reinterpret_cast<PyObject*>(array);
 }
 
-- 
2.22.0

