From 2fb9862bfd7dd953a51f9bf4763bf0b308ddee13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20M=C3=BCller?= <nexyon@gmail.com>
Date: Fri, 10 May 2019 23:04:14 +0200
Subject: [PATCH 16/16] Fix: Missing include in FileManager.h.

---
 include/file/FileManager.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/file/FileManager.h b/include/file/FileManager.h
index 03943ea..5670860 100644
--- a/include/file/FileManager.h
+++ b/include/file/FileManager.h
@@ -27,6 +27,7 @@
 
 #include <list>
 #include <memory>
+#include <string>
 
 AUD_NAMESPACE_BEGIN
 
-- 
2.22.0

