From ece0842776e9a1b3689e6063a08618f6a67d836e Mon Sep 17 00:00:00 2001
From: nutti <nutti.metro@gmail.com>
Date: Wed, 29 Jul 2020 08:01:13 +0900
Subject: [PATCH 29/35] Fix corrupted document in python bindings

---
 bindings/python/PySound.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bindings/python/PySound.cpp b/bindings/python/PySound.cpp
index 62ee343..f2debcc 100644
--- a/bindings/python/PySound.cpp
+++ b/bindings/python/PySound.cpp
@@ -1395,9 +1395,9 @@ PyDoc_STRVAR(M_aud_Sound_threshold_doc,
 			 "   all between to 0.\n\n"
 			 "   :arg threshold: Threshold value over which an amplitude counts\n"
 			 "      non-zero.\n\n"
-			 ":type threshold: float\n"
-			 ":return: The created :class:`Sound` object.\n"
-			 ":rtype: :class:`Sound`");
+			 "   :type threshold: float\n"
+			 "   :return: The created :class:`Sound` object.\n"
+			 "   :rtype: :class:`Sound`");
 
 static PyObject *
 Sound_threshold(Sound* self, PyObject* args)
-- 
2.29.2

