From 9a6a802a3346c2cd72b4c6f1f0160c20ffaa4e0d Mon Sep 17 00:00:00 2001
From: Joerg Mueller <nexyon@gmail.com>
Date: Mon, 3 Aug 2020 16:48:27 +0200
Subject: [PATCH 30/35] Bugfix: JOS resampling type bugfix caused integer
 underflow.

---
 src/respec/JOSResampleReader.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/respec/JOSResampleReader.cpp b/src/respec/JOSResampleReader.cpp
index 6753a2e..378986f 100644
--- a/src/respec/JOSResampleReader.cpp
+++ b/src/respec/JOSResampleReader.cpp
@@ -119,8 +119,8 @@ void JOSResampleReader::updateBuffer(int size, double factor, int samplesize)
 			P = int_to_fp(m_L) - P;\
 \
 			end = std::floor((m_len - 1) / double(m_L) + m_P) - 1;\
-			if(m_cache_valid - m_n - 2 < end)\
-				end = m_cache_valid - m_n - 2;\
+			if(m_cache_valid - int(m_n) - 2 < end)\
+				end = m_cache_valid - int(m_n) - 2;\
 \
 			data = buf + (m_n + 2 + end) * m_channels - 1;\
 			l = fp_to_int(P);\
@@ -166,8 +166,8 @@ void JOSResampleReader::updateBuffer(int size, double factor, int samplesize)
 			P = 0 - P;\
 \
 			end = (int_to_fp(m_len) - P) / P_increment - 1;\
-			if(m_cache_valid - m_n - 2 < end)\
-				end = m_cache_valid - m_n - 2;\
+			if(m_cache_valid - int(m_n) - 2 < end)\
+				end = m_cache_valid - int(m_n) - 2;\
 \
 			P += P_increment * end;\
 			data = buf + (m_n + 2 + end) * m_channels - 1;\
-- 
2.29.2

