From 789832ec009b6e4fdb5e1167ba7e8467757b2367 Mon Sep 17 00:00:00 2001
From: Joerg Mueller <nexyon@gmail.com>
Date: Tue, 1 Sep 2020 16:30:08 +0200
Subject: [PATCH 31/35] Fix numpy import.

---
 bindings/python/PySound.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/python/PySound.cpp b/bindings/python/PySound.cpp
index f2debcc..3362830 100644
--- a/bindings/python/PySound.cpp
+++ b/bindings/python/PySound.cpp
@@ -2016,7 +2016,7 @@ AUD_API Sound* checkSound(PyObject* sound)
 
 bool initializeSound()
 {
-	import_array();
+	import_array1(false);
 
 	return PyType_Ready(&SoundType) >= 0;
 }
-- 
2.29.2

