diff --git a/BUILD.gn b/BUILD.gn
index f227c834..9f3b4a60 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -16,39 +16,8 @@ config("absl_component_build") {
   defines = [ "ABSL_CONSUME_DLL" ]
 }
 
-component("absl") {
+static_library("absl") {
   public_deps = [ ":absl_component_deps" ]
-  if (is_component_build) {
-    public_configs = [ ":absl_component_build" ]
-
-    if (is_win && is_clang) {
-      if (current_cpu == "x64") {
-        if (is_debug) {
-          sources = [ "symbols_x64_dbg.def" ]
-        } else {
-          if (is_asan) {
-            sources = [ "symbols_x64_rel_asan.def" ]
-          } else {
-            sources = [ "symbols_x64_rel.def" ]
-          }
-        }
-      }
-      if (current_cpu == "x86") {
-        if (is_debug) {
-          sources = [ "symbols_x86_dbg.def" ]
-        } else {
-          sources = [ "symbols_x86_rel.def" ]
-        }
-      }
-      if (current_cpu == "arm64") {
-        if (is_debug) {
-          sources = [ "symbols_arm64_dbg.def" ]
-        } else {
-          sources = [ "symbols_arm64_rel.def" ]
-        }
-      }
-    }
-  }
 }
 
 group("absl_component_deps") {
diff --git a/absl.gni b/absl.gni
index 48e1ce78..a4721461 100644
--- a/absl.gni
+++ b/absl.gni
@@ -40,7 +40,7 @@ template("absl_source_set") {
       if (!defined(defines)) {
         defines = []
       }
-      if (is_component_build) {
+      if (!is_mingw && is_component_build) {
         defines += [ "ABSL_BUILD_DLL" ]
         if (!is_win && current_os != "aix") {
           configs -= [ "//build/config/gcc:symbol_visibility_hidden" ]
diff --git a/absl/base/BUILD.gn b/absl/base/BUILD.gn
index 98e75941..c4aa2f85 100644
--- a/absl/base/BUILD.gn
+++ b/absl/base/BUILD.gn
@@ -190,6 +190,9 @@ absl_source_set("base") {
     ":spinlock_wait",
     "//third_party/abseil-cpp/absl/meta:type_traits",
   ]
+  if (is_mingw) {
+    libs = [ ":libpthread.a" ]
+  }
 }
 
 absl_test("attributes_test") {
diff --git a/absl/base/internal/thread_identity.h b/absl/base/internal/thread_identity.h
index acfc15a8..a3f3df78 100644
--- a/absl/base/internal/thread_identity.h
+++ b/absl/base/internal/thread_identity.h
@@ -20,7 +20,7 @@
 #ifndef ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 #define ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 #include <pthread.h>
 // Defines __GOOGLE_GRTE_VERSION__ (via glibc-specific features.h) when
 // supported.
diff --git a/absl/random/internal/BUILD.gn b/absl/random/internal/BUILD.gn
index 46dcff5a..4182eed6 100644
--- a/absl/random/internal/BUILD.gn
+++ b/absl/random/internal/BUILD.gn
@@ -40,7 +40,7 @@ absl_source_set("seed_material") {
   sources = [ "seed_material.cc" ]
   if (is_win) {
     # TODO(mbonadei): In the bazel file this is -DEFAULTLIB:bcrypt.lib.
-    libs = [ "bcrypt.lib" ]
+    libs = [ "bcrypt" ]
   }
   deps = [
     ":fast_uniform_bits",
diff --git a/absl/time/internal/cctz/src/time_zone_lookup.cc b/absl/time/internal/cctz/src/time_zone_lookup.cc
index e8f1d930..9f238db6 100644
--- a/absl/time/internal/cctz/src/time_zone_lookup.cc
+++ b/absl/time/internal/cctz/src/time_zone_lookup.cc
@@ -32,7 +32,7 @@
 #include <zircon/types.h>
 #endif
 
-#if defined(_WIN32)
+#if defined(_MSC_VER)
 // Include only when <icu.h> is available.
 // https://learn.microsoft.com/en-us/windows/win32/intl/international-components-for-unicode--icu-
 // https://devblogs.microsoft.com/oldnewthing/20210527-00/?p=105255
