diff -bur qtbase-orig/src/network/ssl/qsslsocket_openssl_symbols.cpp qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp
--- qtbase-orig/src/network/ssl/qsslsocket_openssl_symbols.cpp	2026-01-24 18:34:56.233892500 -0700
+++ qtbase/src/network/ssl/qsslsocket_openssl_symbols.cpp	2026-01-24 18:35:28.597004900 -0700
@@ -692,7 +692,7 @@
 #if defined(Q_PROCESSOR_X86_64)
 #define QT_SSL_SUFFIX "-x64"
 #elif defined(Q_PROCESSOR_ARM_64)
-#define QT_SSL_SUFFIX "-arm64"
+#define QT_SSL_SUFFIX
 #elif defined(Q_PROCESSOR_ARM_32)
 #define QT_SSL_SUFFIX "-arm"
 #else
