--- qtbase-everywhere-src-5.14.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in.orig       2019-12-07 06:27:07.000000000 +0000
+++ qtbase-everywhere-src-5.14.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in    2020-01-23 19:45:11.239484100 +0000
@@ -71,7 +71,7 @@

         string(REGEX REPLACE \"QMAKE_PRL_LIBS_FOR_CMAKE[ \\t]*=[ \\t]*([^\\n]*)\" \"\\\\1\" _static_depends \"${_prl_strings}\")
         string(REGEX REPLACE \"[ \\t]+\" \";\" _standard_libraries \"${CMAKE_CXX_STANDARD_LIBRARIES}\")
-        set(_search_paths)
+        set(_search_paths ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES})
         string(REPLACE \"\\$\\$[QT_INSTALL_LIBS]\" \"${_qt5_install_libs}\" _static_depends \"${_static_depends}\")
         foreach(_flag ${_static_depends})
             string(REPLACE \"\\\"\" \"\" _flag ${_flag})
