diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp qt-everywhere-src-5.12.4/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp
--- qt-everywhere-src-5.12.4-orig/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp	2019-06-12 23:59:14.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtbase/src/plugins/platforms/minimal/qminimalintegration.cpp	2019-06-15 16:01:25.262773000 +0300
@@ -161,7 +161,7 @@
 #endif
 
         if (!m_fontDatabase) {
-#if QT_CONFIG(fontconfig)
+#if QT_CONFIG(fontconfig) && !defined(Q_OS_WIN)
             m_fontDatabase = new QGenericUnixFontDatabase;
 #else
             m_fontDatabase = QPlatformIntegration::fontDatabase();
