diff -urN scalapack-2.0.2.orig/CMakeLists.txt scalapack-2.0.2/CMakeLists.txt
--- scalapack-2.0.2.orig/CMakeLists.txt	2012-05-01 19:58:24.000000000 -0700
+++ scalapack-2.0.2/CMakeLists.txt	2019-11-21 03:03:12.379652300 -0800
@@ -239,6 +239,7 @@
    add_library(scalapack ${blacs} ${tools-C} ${pblas} ${ptools} ${redist} ${src-C})
    target_link_libraries( scalapack ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
    add_library(scalapack-F ${pblas-F} ${pbblas} ${ptzblas} ${tools} ${src} ${extra_lapack} )
+   set_target_properties(scalapack-F PROPERTIES LINKER_LANGUAGE Fortran)
    target_link_libraries( scalapack-F ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
    scalapack_install_library(scalapack)
    scalapack_install_library(scalapack-F)
