--- a/bindings/python/setup.py
+++ b/bindings/python/setup.py
@@ -25,8 +25,8 @@
     LIBRARY_FILE = "libunicorn.2.dylib"
     STATIC_LIBRARY_FILE = "libunicorn.a"
 elif sys.platform in ('win32', 'cygwin'):
-    LIBRARY_FILE = "unicorn.dll"
-    STATIC_LIBRARY_FILE = "unicorn.lib"
+    LIBRARY_FILE = "libunicorn.dll"
+    STATIC_LIBRARY_FILE = "libunicorn.a"
 else:
     LIBRARY_FILE = "libunicorn.so.2"
     STATIC_LIBRARY_FILE = "libunicorn.a"
--- a/bindings/python/unicorn/unicorn_py3/unicorn.py
+++ b/bindings/python/unicorn/unicorn_py3/unicorn.py
@@ -124,7 +124,7 @@
         'darwin': 'libunicorn.2.dylib',
         'linux':  'libunicorn.so.2',
         'linux2': 'libunicorn.so.2',
-        'win32':  'unicorn.dll'
+        'win32':  'libunicorn.dll'
     }.get(platform, "libunicorn.so")
 
     def __attempt_load(libname: str):
