--- unicorn-2.0.0-rc5/bindings/python/setup.py.orig	2022-01-14 00:15:15.621000000 +0100
+++ unicorn-2.0.0-rc5/bindings/python/setup.py	2022-01-14 09:15:19.615579000 +0100
@@ -35,8 +35,8 @@
     LIBRARY_FILE = "libunicorn.dylib"
     STATIC_LIBRARY_FILE = None
 elif SYSTEM in ('win32', 'cygwin'):
-    LIBRARY_FILE = "unicorn.dll"
-    STATIC_LIBRARY_FILE = "unicorn.lib"
+    LIBRARY_FILE = "libunicorn.dll"
+    STATIC_LIBRARY_FILE = "libunicorn.a"
 else:
     LIBRARY_FILE = "libunicorn.so"
     STATIC_LIBRARY_FILE = None
