diff --git a/bindings/python/setup.py b/bindings/python/setup.py
index 8c579ad3..28622d30 100755
--- a/bindings/python/setup.py
+++ b/bindings/python/setup.py
@@ -35,8 +35,8 @@ if SYSTEM == 'darwin':
     LIBRARY_FILE = "libunicorn.2.dylib"
     STATIC_LIBRARY_FILE = "libunicorn.a"
 elif SYSTEM in ('win32', 'cygwin'):
-    LIBRARY_FILE = "unicorn.dll"
-    STATIC_LIBRARY_FILE = "unicorn.lib"
+    LIBRARY_FILE = "libunicorn.dll"
+    STATIC_LIBRARY_FILE = "libunicorn.a"
 else:
     LIBRARY_FILE = "libunicorn.so.2"
     STATIC_LIBRARY_FILE = "libunicorn.a"
diff --git a/bindings/python/unicorn/unicorn.py b/bindings/python/unicorn/unicorn.py
index 2e6a938f..86d3e306 100644
--- a/bindings/python/unicorn/unicorn.py
+++ b/bindings/python/unicorn/unicorn.py
@@ -23,7 +23,7 @@ if _python2:
     range = xrange
 
 _lib = { 'darwin': 'libunicorn.2.dylib',
-         'win32': 'unicorn.dll',
+         'win32': 'libunicorn.dll',
          'cygwin': 'cygunicorn.dll',
          'linux': 'libunicorn.so.2',
          'linux2': 'libunicorn.so.2' }
