diff -bur unicorn-2.1.4-orig/qemu/include/sysemu/os-win32.h unicorn-2.1.4/qemu/include/sysemu/os-win32.h
--- unicorn-2.1.4-orig/qemu/include/sysemu/os-win32.h	2026-01-10 23:42:58.122194700 -0800
+++ unicorn-2.1.4/qemu/include/sysemu/os-win32.h	2026-01-10 23:43:21.202293700 -0800
@@ -30,6 +30,7 @@
 #include <windows.h>
 #include <winnt.h> // For vectorized handler
 #include <ws2tcpip.h>
+#include <setjmp.h>
 
 #if defined(_WIN64)
 /* On w64, setjmp is implemented by _setjmp which needs a second parameter.
@@ -45,7 +46,11 @@
 #else // MingW
 
 #undef setjmp
-#define setjmp(env) _setjmp(env, NULL)
+#ifdef __aarch64__
+  #define setjmp(env) __mingw_setjmp(env)
+#else
+  #define setjmp(env) _setjmp(env, NULL)
+#endif
 #endif
 
 #endif
