--- ImageMagick-6.9.0-4/magick/nt-base.h.orig	2015-01-18 16:39:35.182400000 +0300
+++ ImageMagick-6.9.0-4/magick/nt-base.h	2015-01-18 16:39:37.740800000 +0300
@@ -38,6 +38,8 @@
 #include <process.h>
 #include <errno.h>
 #include <malloc.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #include <sys/utime.h>
 #if defined(_DEBUG) && !defined(__MINGW32__) && !defined(__MINGW64__)
 #include <crtdbg.h>
 #endif
