--- ImageMagick-6.9.0-4/magick/distribute-cache.c.orig	2015-01-18 16:07:04.357600000 +0300
+++ ImageMagick-6.9.0-4/magick/distribute-cache.c	2015-01-18 16:11:40.668800000 +0300
@@ -86,6 +86,8 @@
 #define SOCKET_TYPE int
 #define MAGICKCORE_HAVE_DISTRIBUTE_CACHE
 #elif defined(MAGICKCORE_WINDOWS_SUPPORT)
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #define CHAR_TYPE_CAST (char *)
 #define CLOSE_SOCKET(socket) (void) closesocket(socket)
 #define HANDLER_RETURN_TYPE DWORD WINAPI
--- ImageMagick-6.9.0-4/magick/nt-base.h.orig	2015-01-18 16:39:35.182400000 +0300
+++ ImageMagick-6.9.0-4/magick/nt-base.h	2015-01-18 16:39:37.740800000 +0300
@@ -38,6 +38,8 @@
 #include <process.h>
 #include <errno.h>
 #include <malloc.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #include <sys/utime.h>
 #if defined(_DEBUG) && !defined(__MINGW32__) && !defined(__MINGW64__)
 #include <crtdbg.h>
 #endif
--- ImageMagick-6.9.9-15/coders/xtrn.ori.c	2017-09-30 13:37:32.071883600 +0200
+++ ImageMagick-6.9.9-15/coders/xtrn.c	2017-09-30 13:38:26.238127000 +0200
@@ -59,7 +59,7 @@
 #include "magick/memory_.h"
 #include "magick/module.h"
 #include "magick/string_.h"
-#if defined(MAGICKCORE_WINDOWS_SUPPORT)
+#if defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__MINGW32__) && !defined(__MINGW64__)
 #define WIN32_LEAN_AND_MEAN
 #define VC_EXTRALEAN
 #include <windows.h>
@@ -99,7 +99,7 @@
 %    o exception: return any errors or warnings in this structure.
 %
 */
-#if defined(MAGICKCORE_WINDOWS_SUPPORT)
+#if defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__MINGW32__) && !defined(__MINGW64__)
 static Image *ReadXTRNImage(const ImageInfo *image_info,
   ExceptionInfo *exception)
 {
@@ -200,7 +200,7 @@
     *entry;
 
   entry=SetMagickInfo("XTRNARRAY");
-#if defined(MAGICKCORE_WINDOWS_SUPPORT)
+#if defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__MINGW32__) && !defined(__MINGW64__)
   entry->decoder=ReadXTRNImage;
   entry->encoder=WriteXTRNImage;
 #endif
@@ -265,7 +265,7 @@
 %
 */
 
-#if defined(MAGICKCORE_WINDOWS_SUPPORT)
+#if defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__MINGW32__) && !defined(__MINGW64__)
 static size_t SafeArrayFifo(const Image *image,const void *data,
   const size_t length)
 {
