--- quassel-0.12.4/src/qtui/CMakeLists.txt.orig	2017-04-07 10:34:34.947277500 +0300
+++ quassel-0.12.4/src/qtui/CMakeLists.txt	2017-04-07 10:40:40.835902900 +0300
@@ -132,7 +132,13 @@
 if (HAVE_WEBKIT)
     add_definitions(-DHAVE_WEBKIT)
     list(APPEND QT_MODULES WebKit)
+    if (Qt5WebKit_FOUND OR QT_QTWEBKIT_FOUND)
+        include_directories(${Qt5WebKit_INCLUDE_DIRS} ${QT_QTWEBKIT_INCLUDE_DIRS})
+    endif ()
     if (USE_QT5)
+        if (Qt5WebKitWidgets_FOUND)
+            include_directories(${Qt5WebKitWidgets_INCLUDE_DIRS})
+        endif ()
         list(APPEND QT_MODULES WebKitWidgets)
     endif()
 endif()
