diff -bur stellarium-23.3-orig/src/CMakeLists.txt stellarium-23.3/src/CMakeLists.txt
--- stellarium-23.3-orig/src/CMakeLists.txt	2023-09-29 00:47:25.195434700 -0600
+++ stellarium-23.3/src/CMakeLists.txt	2023-09-29 00:49:13.758547000 -0600
@@ -645,11 +645,6 @@
 	  "
      )
      ENDIF()
-     # workaround for https://bugreports.qt.io/browse/QTBUG-105984
-     IF(ENABLE_MEDIA AND (${QT_VERSION} VERSION_GREATER_EQUAL "6.4.0"))
-	  GET_FILENAME_COMPONENT(_qt_bin_dir "${WINDEPLOYQT_COMMAND}" DIRECTORY)
-	  INSTALL(FILES ${_qt_bin_dir}/../plugins/multimedia/windowsmediaplugin.dll DESTINATION ${CMAKE_INSTALL_PREFIX}/qtstuff/multimedia)
-     ENDIF()
 ENDIF()
 
 SET_TARGET_PROPERTIES(stelMain PROPERTIES FOLDER "src/core")
