--- flask-babel-4.0.0.orig/docs/conf.py	2024-03-05 00:43:10.221129500 +0300
+++ flask-babel-4.0.0/docs/conf.py	2024-03-05 00:43:27.772419000 +0300
@@ -49,7 +49,7 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = 'furo'
+#html_theme = 'furo'
 
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
