--- src/errors.h.orig	  2014-04-25 10:24:16.396471000 +0200
+++ src/errors.h    2014-04-25 10:51:47.402099600 +0200
@@ -1,18 +1,26 @@
 /* error.h
  * Error handling for pngnq 
  */
-#include "syslog.h"
+
+#ifndef _WIN32
+    #include "syslog.h"
+#endif
 
 /* Error codes */
 #define PNGNQ_ERR_NONE 0
 #define PNGNQ_ERR_ 0
 
-#define PNGNQ_LOG_ERR(...)(syslog(LOG_ERR,\
-    "pngnq - Error in %s near line %d:",__FILE__,__LINE__));\
-    syslog(LOG_ERR, __VA_ARGS__); 
-    
-#define PNGNQ_LOG_WARNING(...)(syslog(LOG_WARNING,"pngnq - warning: "));\
-    syslog(LOG_WARNING, __VA_ARGS__);
+#ifdef _WIN32
+    #define PNGNQ_LOG_ERR(...)
+    #define PNGNQ_LOG_WARNING(...)
+#else
+    #define PNGNQ_LOG_ERR(...)(syslog(LOG_ERR,\
+        "pngnq - Error in %s near line %d:",__FILE__,__LINE__));\
+        syslog(LOG_ERR, __VA_ARGS__); 
+        
+    #define PNGNQ_LOG_WARNING(...)(syslog(LOG_WARNING,"pngnq - warning: "));\
+        syslog(LOG_WARNING, __VA_ARGS__);
+#endif
 
 #define PNGNQ_ERROR(...) (fprintf(stderr,\
     "pngnq - Error in %s near line %d :\n",__FILE__,__LINE__));\
