From 7593fd7ea0985e273c82b6e80e66a801772024de Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Orhun=20Parmaks=C4=B1z?= <orhunparmaksiz@gmail.com>
Date: Mon, 25 Mar 2024 21:21:11 +0300
Subject: [PATCH] chore: update Cargo.lock for latest version

---
 Cargo.lock | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.lock b/Cargo.lock
index 059f83ec9..1a749ed7a 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -735,7 +735,7 @@ dependencies = [
 
 [[package]]
 name = "lsd"
-version = "1.1.1"
+version = "1.1.2"
 dependencies = [
  "assert_cmd",
  "assert_fs",
