diff -r -c googletest.orig/googlemock/CMakeLists.txt googletest/googlemock/CMakeLists.txt
*** googletest.orig/googlemock/CMakeLists.txt	2016-05-27 21:52:30.900502400 -0300
--- googletest/googlemock/CMakeLists.txt	2016-05-27 21:41:13.527008600 -0300
***************
*** 104,110 ****
  #
  # Install rules
  install(TARGETS gmock gmock_main
!   DESTINATION lib)
  install(DIRECTORY ${gmock_SOURCE_DIR}/include/gmock
    DESTINATION include)
  
--- 104,112 ----
  #
  # Install rules
  install(TARGETS gmock gmock_main
!         RUNTIME DESTINATION bin
!         LIBRARY DESTINATION lib
!         ARCHIVE DESTINATION lib)
  install(DIRECTORY ${gmock_SOURCE_DIR}/include/gmock
    DESTINATION include)
  

diff -r -c googletest.orig/googletest/CMakeLists.txt googletest/googletest/CMakeLists.txt
*** googletest.orig/googletest/CMakeLists.txt	2016-05-27 21:52:35.619587300 -0300
--- googletest/googletest/CMakeLists.txt	2016-05-27 21:41:27.463913700 -0300
***************
*** 103,109 ****
  #
  # Install rules
  install(TARGETS gtest gtest_main
!   DESTINATION lib)
  install(DIRECTORY ${gtest_SOURCE_DIR}/include/gtest
    DESTINATION include)
  
--- 103,111 ----
  #
  # Install rules
  install(TARGETS gtest gtest_main
!         RUNTIME DESTINATION bin
!         LIBRARY DESTINATION lib
!         ARCHIVE DESTINATION lib)
  install(DIRECTORY ${gtest_SOURCE_DIR}/include/gtest
    DESTINATION include)
  
