--- ./libpurple/Makefile.am.orig	2015-05-17 21:54:49.202429500 +0200
+++ ./libpurple/Makefile.am	2015-05-17 23:06:31.848670500 +0200
@@ -463,6 +463,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = --add-include-path=$(prefix)/share/gir-1.0 --warn-all
 INTROSPECTION_COMPILER_ARGS = --includedir=$(prefix)/share/gir-1.0
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 if HAVE_INTROSPECTION
 introspection_sources = \
@@ -480,7 +481,7 @@
 endif
 
 Purple_3_0_gir_CFLAGS = \
-	-I$(srcdir) \
+	-I$(top_srcdir) \
 	$(INCLUDES) \
 	-DDATADIR=\"$(datadir)\" \
 	-DLIBDIR=\"$(libdir)/purple-$(PURPLE_MAJOR_VERSION)/\" \
@@ -506,10 +507,10 @@
 Purple_3_0_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += Purple-$(PURPLE_MAJOR_VERSION).$(PURPLE_MINOR_VERSION).gir
 
-girdir = $(prefix)/share/gir-1.0 $(INTROSPECTION_GIRDIR)
+girdir = $(datadir)/gir-1.0
 gir_DATA = $(INTROSPECTION_GIRS)
 
-typelibdir = $(prefix)/share/gir-1.0 $(INTROSPECTION_TYPELIBDIR)
+typelibdir = $(libdir)/girepository-1.0
 typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 
 CLEANFILES += $(gir_DATA) $(typelib_DATA)
--- ./pidgin/Makefile.am.orig	2015-05-17 21:54:49.224430000 +0200
+++ ./pidgin/Makefile.am	2015-05-17 23:07:16.130585900 +0200
@@ -266,6 +266,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = --warn-all --add-include-path=$(top_builddir)/libpurple --add-include-path=$(prefix)/share/gir-1.0
 INTROSPECTION_COMPILER_ARGS = --includedir=$(top_builddir)/libpurple --includedir=$(prefix)/share/gir-1.0
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 if HAVE_INTROSPECTION
 introspection_sources = $(libpidgininclude_HEADERS)
@@ -300,10 +301,10 @@
 Pidgin_3_0_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += Pidgin-$(PURPLE_MAJOR_VERSION).$(PURPLE_MINOR_VERSION).gir
 
-girdir = $(prefix)/share/gir-1.0 $(INTROSPECTION_GIRDIR)
+girdir = $(datadir)/gir-1.0
 gir_DATA = $(INTROSPECTION_GIRS)
 
-typelibdir = $(prefix)/share/gir-1.0 $(INTROSPECTION_TYPELIBDIR)
+typelibdir = $(libdir)/girepository-1.0
 typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 
 CLEANFILES += $(gir_DATA) $(typelib_DATA)
