diff -aurN 001/configure.ac 002/configure.ac
--- 001/configure.ac
+++ 002/configure.ac
@@ -828,7 +828,7 @@
 	    [enable_consoleui=no], [$GNT_LIBS])
 
 	if test "x$is_win32" = "xyes" ; then
-		ncurses_sys_prefix="/usr/$host/sys-root/mingw"
+		ncurses_sys_prefix="$(cygpath --mixed $prefix)"
 	else
 		ncurses_sys_prefix="/usr"
 	fi
diff -aurN 001/finch/libgnt/Makefile.am 002/finch/libgnt/Makefile.am
--- 001/finch/libgnt/Makefile.am
+++ 002/finch/libgnt/Makefile.am
@@ -90,7 +90,8 @@
 	$(GNT_LIBS) \
 	$(LIBXML_LIBS) \
 	$(INTROSPECTION_LIBS) \
-	$(PY_LIBS)
+	$(PY_LIBS) \
+	-lncursesw
 
 if IS_WIN32
 
