--- petsc-3.17.0/config/BuildSystem/config/packages/openblas.py.orig	2022-03-31 06:40:23.000000000 +0200
+++ petsc-3.17.0/config/BuildSystem/config/packages/openblas.py	2022-05-30 09:24:12.674202500 +0200
@@ -19,7 +19,7 @@
     self.functions              = ['openblas_get_config']
     self.liblist                = [['libopenblas.a']]
     self.precisions             = ['single','double']
-    self.buildLanguages         = ['C','FC']
+    self.buildLanguages         = ['C']
     self.installwithbatch       = 1
     self.usespthreads           = 0
 
