diff -urN petsc-3.8.3.orig/config/BuildSystem/config/packages/MPI.py petsc-3.8.3/config/BuildSystem/config/packages/MPI.py
--- petsc-3.8.3.orig/config/BuildSystem/config/packages/MPI.py	2017-09-27 03:55:36.999998400 +0500
+++ petsc-3.8.3/config/BuildSystem/config/packages/MPI.py	2018-02-26 17:37:41.888726500 +0500
@@ -11,7 +11,7 @@
 class Configure(config.package.Package):
   def __init__(self, framework):
     config.package.Package.__init__(self, framework)
-    self.functions          = ['MPI_Init', 'MPI_Comm_create']
+    self.functions          = [] # ['MPI_Init', 'MPI_Comm_create']
     self.includes           = ['mpi.h']
     liblist_mpich         = [['fmpich2.lib','fmpich2g.lib','fmpich2s.lib','mpi.lib'],
                              ['fmpich2.lib','fmpich2g.lib','mpi.lib'],['fmpich2.lib','mpich2.lib'],
@@ -35,7 +35,7 @@
     liblist_other         = [['libmpich.a','libpthread.a'],['libmpi++.a','libmpi.a']]
     liblist_single        = [['libmpi.a'],['libmpich.a'],['mpi.lib'],['mpich2.lib'],['mpich.lib'],
                              [os.path.join('amd64','msmpi.lib')],[os.path.join('i386','msmpi.lib')]]
-    self.liblist          = liblist_mpich + liblist_lam + liblist_msmpi + liblist_other + liblist_single
+    self.liblist          = [['libmsmpi.a']] # liblist_mpich + liblist_lam + liblist_msmpi + liblist_other + liblist_single
     # defaults to --with-mpi=yes
     self.required         = 1
     self.complex          = 1
@@ -532,3 +532,6 @@
     if self.libraries.check(self.dlib, "MPIDI_CH3I_sock_fixed_nbc_progress"):
       # Indicates that this bug was fixed: http://trac.mpich.org/projects/mpich/ticket/1785
       self.addDefine('HAVE_MPICH_CH3_SOCK_FIXED_NBC_PROGRESS', 1)
+    # Override available functions for MS-MPI 2012+ as they are not detected in 32-bit mode
+    for f in '''MPI_Comm_spawn MPI_Type_get_envelope MPI_Type_get_extent MPI_Type_dup MPI_Init_thread MPI_Finalized MPI_Exscan'''.split():
+      self.addDefine('HAVE_' + f.upper(),1)
