diff -Naur glpk-4.65.orig/src/env/time.c glpk-4.65/src/env/time.c
--- glpk-4.65.orig/src/env/time.c	2018-04-29 17:19:55.861485500 -0400
+++ glpk-4.65/src/env/time.c	2018-04-29 17:22:26.023473000 -0400
@@ -55,13 +55,15 @@
 double glp_time(void)
 {     struct timeval tv;
       struct tm *tm;
+      time_t sec;
       int j;
       double t;
+      sec = tv.tv_sec;
       gettimeofday(&tv, NULL);
 #if 0 /* 29/I-2017 */
-      tm = gmtime(&tv.tv_sec);
+      tm = gmtime(&sec);
 #else
-      tm = xgmtime(&tv.tv_sec);
+      tm = xgmtime(&sec);
 #endif
       j = jday(tm->tm_mday, tm->tm_mon + 1, 1900 + tm->tm_year);
       xassert(j >= 0);
