--- a/setupbase.py	2018-12-17 11:01:51.000000000 +0100
+++ b/setupbase.py	2019-01-13 00:06:03.916440500 +0100
@@ -553,7 +553,8 @@
     def build_jstranslation(self, trd):
         lang = trd[-5:]
         run([
-            pjoin('node_modules', '.bin', 'po2json'),
+            'node',
+            pjoin('node_modules', 'po2json', 'bin', 'po2json'),
             '-p', '-F',
             '-f', 'jed1.x',
             '-d', 'nbjs',
