--- libmpeg2/libvo/video_out_dx.c	2014-07-11 00:22:33.440400000 +0400
+++ libmpeg2/libvo/video_out_dx.c.new	2014-07-11 00:22:27.824400000 +0400
@@ -92,7 +92,7 @@
     switch (message) {
 
     case WM_WINDOWPOSCHANGED:
-	instance = (dx_instance_t *) GetWindowLong (hwnd, GWL_USERDATA);
+	instance = (dx_instance_t *) GetWindowLongPtr (hwnd, GWLP_USERDATA);
 
 	/* update the window position and size */
 	point_window.x = 0;
@@ -173,7 +173,7 @@
     /* store a directx_instance pointer into the window local storage
      * (for later use in event_handler).
      * We need to use SetWindowLongPtr when it is available in mingw */
-    SetWindowLong (instance->window, GWL_USERDATA, (LONG) instance);
+    SetWindowLongPtr (instance->window, GWLP_USERDATA, (LONG) instance);
 
     ShowWindow (instance->window, SW_SHOW);
 
