--- a/nt/cmdproxy.c
+++ b/nt/cmdproxy.c
@@ -37,7 +37,25 @@
 
 /* We don't want to include stdio.h because we are already duplicating
    lots of it here */
+/* Copied from mingw-w64-headers/crt/stdio.h because _snprint is not
+   exported in UCRT environment. But exported from MSVCRT. */
+#ifdef _UCRT
+  __attribute__((__format__ (__MINGW_PRINTF_FORMAT, 3, 0))) __MINGW_ATTRIB_NONNULL(3)
+  int __cdecl _vsnprintf(char * __restrict__ _Dest,size_t _Count,const char * __restrict__ _Format,va_list _Args) __MINGW_ATTRIB_DEPRECATED_SEC_WARN;
+  __mingw_ovr __MINGW_ATTRIB_DEPRECATED_SEC_WARN
+  __attribute__((__format__ (__MINGW_PRINTF_FORMAT, 3, 4))) __MINGW_ATTRIB_NONNULL(3)
+  int __cdecl _snprintf(char * __restrict__ _Dest,size_t _Count,const char * __restrict__ _Format,...)
+  {
+    __builtin_va_list __ap;
+    int __ret;
+    __builtin_va_start(__ap, _Format);
+    __ret = _vsnprintf(_Dest, _Count, _Format, __ap);
+    __builtin_va_end(__ap);
+    return __ret;
+  }
+#else
 extern int _snprintf (char *buffer, size_t count, const char *format, ...);
+#endif
 
 /*******  Mock C library routines  *********************************/
 
