diff --git a/lisp/term/w32-win.el b/lisp/term/w32-win.el
index 6db7e20..d4fc0cd 100644
--- a/lisp/term/w32-win.el
+++ b/lisp/term/w32-win.el
@@ -265,7 +265,7 @@ This returns an error if any Emacs frames are X frames, or always under W32."
        '(gdk-pixbuf "libgdk_pixbuf-2.0-0.dll")
        '(glib "libglib-2.0-0.dll")
        '(gobject "libgobject-2.0-0.dll")
-       '(gnutls "libgnutls-28.dll" "libgnutls-26.dll")
+       '(gnutls "libgnutls-30.dll" "libgnutls-28.dll" "libgnutls-26.dll")
        '(libxml2 "libxml2-2.dll" "libxml2.dll")
        '(zlib "zlib1.dll" "libz-1.dll")))
 
diff --git a/src/gnutls.c b/src/gnutls.c
index d363fea..c8a2b72 100644
--- a/src/gnutls.c
+++ b/src/gnutls.c
@@ -1020,8 +1020,10 @@ one trustfile (usually a CA bundle).  */)
     return gnutls_make_error (ret);
 
   GNUTLS_INITSTAGE (proc) = GNUTLS_STAGE_CRED_SET;
-  ret = emacs_gnutls_handshake (XPROCESS (proc));
-  if (ret < GNUTLS_E_SUCCESS)
+  do {
+    ret = emacs_gnutls_handshake (XPROCESS (proc));
+  } while (ret == GNUTLS_E_INTERRUPTED || ret == GNUTLS_E_AGAIN);
+  if (ret < GNUTLS_E_SUCCESS && ret != GNUTLS_E_WARNING_ALERT_RECEIVED)
     return gnutls_make_error (ret);
 
   /* Now verify the peer, following
