diff -urN emacs-emacs-28.0.91-orig/configure.ac emacs-emacs-28.0.91/configure.ac
--- emacs-emacs-28.0.91-orig/configure.ac	2022-01-08 20:09:29.000000000 +0800
+++ emacs-emacs-28.0.91/configure.ac	2022-01-21 07:46:26.195050900 +0800
@@ -147,7 +147,7 @@
       fi
       cc_target=`$cc -v 2>&1 | sed -n 's/Target: //p'`
       case "$cc_target" in
-          *-*) host=$cc_target
+          *-*) host=$build_alias
 	      ;;
           "") AC_MSG_ERROR([Impossible to obtain $cc compiler target.
 Please explicitly provide --host.])
@@ -5711,7 +5711,7 @@
 LIBS=$OLD_LIBS
 
 if test "${opsys}" = "mingw32"; then
-  CPPFLAGS="$CPPFLAGS -DUSE_CRT_DLL=1 -I \${abs_top_srcdir}/nt/inc"
+  CPPFLAGS="$CPPFLAGS -I \${abs_top_srcdir}/nt/inc"
   # Remove unneeded switches from the value of CC that goes to Makefiles
   CC=`AS_ECHO(["$CC"]) | sed -e "s,$GCC_TEST_OPTIONS,,"`
 fi
