diff -aur emacs-28.2-orig/configure.ac emacs-28.2/configure.ac
--- emacs-28.2-orig/configure.ac	2022-09-07 06:24:51.000000000 +0800
+++ emacs-28.2/configure.ac	2022-10-09 20:40:07.436552500 +0800
@@ -147,7 +147,7 @@
       fi
       cc_target=`$cc -v 2>&1 | sed -n 's/Target: //p'`
       case "$cc_target" in
-          *-*) host=$cc_target
+          *-*) host=$target_alias
 	      ;;
           "") AC_MSG_ERROR([Impossible to obtain $cc compiler target.
 Please explicitly provide --host.])
@@ -5765,8 +5765,8 @@
   mingw32)
    ## Is it any better under MinGW64 to relocate emacs into higher addresses?
    case "$canonical" in
-     x86_64-*-*) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-heap,0x00100000 -Wl,-image-base,0x400000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
-     *) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-heap,0x00100000 -Wl,-image-base,0x01000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
+     x86_64-*-*) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-image-base,0x400000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
+     *) LD_SWITCH_SYSTEM_TEMACS="-Wl,-stack,0x00800000 -Wl,-image-base,0x01000000 -Wl,-entry,__start -Wl,-Map,./temacs.map" ;;
    esac
    ## If they want unexec, disable Windows ASLR for the Emacs binary
    if test "$with_dumping" = "unexec"; then
--- a/nt/mingw-cfg.site
+++ b/nt/mingw-cfg.site
@@ -29,6 +29,10 @@
 # are necessary to steer the test in the direction you need, by
 # judiciously setting variables that control the test results.
 
+# We want to use sys/wait.h from nt/inc
+# https://lists.gnu.org/archive/html/help-gnu-emacs/2023-05/msg00107.html
+ac_cv_header_sys_wait_h=yes
+
 # We want to use getopt.h from gnulib
 ac_cv_header_getopt_h=no
 
