--- emacs-30.2/lisp/term/w32-win.el.orig	2025-03-01 15:42:10.000000000 +0100
+++ emacs-30.2/lisp/term/w32-win.el	2025-09-23 17:33:55.923110200 +0200
@@ -287,7 +287,7 @@
        (if (>= libgnutls-version 30400)
 	   '(gnutls "libgnutls-30.dll")
 	 '(gnutls "libgnutls-28.dll" "libgnutls-26.dll"))
-       '(libxml2 "libxml2-2.dll" "libxml2.dll")
+       '(libxml2 "libxml2-16.dll" "libxml2-2.dll" "libxml2.dll")
        '(zlib "zlib1.dll" "libz-1.dll")
        '(lcms2 "liblcms2-2.dll")
        '(gccjit "libgccjit-0.dll")
@@ -303,7 +303,7 @@
            '(tree-sitter "libtree-sitter-0.24.dll"
                          "libtree-sitter.dll"
                          "libtree-sitter-0.dll")
-         '(tree-sitter "libtree-sitter-0.25.dll"))))
+         '(tree-sitter "libtree-sitter-0.26.dll" "libtree-sitter-0.25.dll"))))
 
 ;;; multi-tty support
 (defvar w32-initialized nil
