--- a/lisp/term/w32-win.el
+++ b/lisp/term/w32-win.el
@@ -314,7 +314,7 @@
 '(gccjit "libgccjit-0.dll")
        ;; MSYS2 distributes libtree-sitter.dll, without API version
        ;; number...
-       '(tree-sitter "libtree-sitter.dll" "libtree-sitter-0.dll")))
+       '(tree-sitter "libtree-sitter.dll" "libtree-sitter-0.dll" "libtree-sitter-0.25.dll")))

 ;;; multi-tty support
 (defvar w32-initialized nil
