diff -bur expresscpp-0.20.0-orig/include/expresscpp/url.hpp expresscpp-0.20.0/include/expresscpp/url.hpp
--- expresscpp-0.20.0-orig/include/expresscpp/url.hpp	2025-04-08 08:43:35.565101800 -0600
+++ expresscpp-0.20.0/include/expresscpp/url.hpp	2025-04-08 08:43:43.053145300 -0600
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
 
 #include "expresscpp/exports.hpp"
 
diff -bur expresscpp-0.20.0-orig/src/middleware/serve_static_provider.cpp expresscpp-0.20.0/src/middleware/serve_static_provider.cpp
--- expresscpp-0.20.0-orig/src/middleware/serve_static_provider.cpp	2025-04-08 08:43:35.549577000 -0600
+++ expresscpp-0.20.0/src/middleware/serve_static_provider.cpp	2025-04-08 08:44:40.469674400 -0600
@@ -1,6 +1,7 @@
 #include "expresscpp/middleware/serve_static_provider.hpp"
 
 #include <filesystem>
+#include <fstream>
 
 #include "expresscpp/console.hpp"
 #include "expresscpp/impl/session.hpp"
