diff -Naur xalan_c-1.12-orig/src/xalanc/Harness/XalanFileUtility.cpp xalan_c-1.12/src/xalanc/Harness/XalanFileUtility.cpp
--- xalan_c-1.12-orig/src/xalanc/Harness/XalanFileUtility.cpp	2020-06-07 12:17:25.000000000 +0300
+++ xalan_c-1.12/src/xalanc/Harness/XalanFileUtility.cpp	2020-10-21 12:18:56.162656900 +0300
@@ -2119,7 +2119,7 @@
 
     if (result == 0)
     {
-#if defined(_MSC_VER)
+#if defined(_WIN32)
         CharVectorType theBuffer(getMemoryManager());
         TranscodeToLocalCodePage(theHTMLFile, theBuffer, true);
 
diff -Naur xalan_c-1.12-orig/src/xalanc/Include/PlatformDefinitions.hpp.in xalan_c-1.12/src/xalanc/Include/PlatformDefinitions.hpp.in
--- xalan_c-1.12-orig/src/xalanc/Include/PlatformDefinitions.hpp.in	2020-06-07 12:17:25.000000000 +0300
+++ xalan_c-1.12/src/xalanc/Include/PlatformDefinitions.hpp.in	2020-10-21 12:19:41.732155100 +0300
@@ -59,7 +59,7 @@
 //  These defines provide the platform specific keywords that they need
 //  to do this.
 // ---------------------------------------------------------------------------
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define XALAN_PLATFORM_EXPORT     __declspec(dllexport)
 #define XALAN_PLATFORM_IMPORT     __declspec(dllimport)
 #define XALAN_PLATFORM_EXPORT_FUNCTION(T) XALAN_PLATFORM_EXPORT T
diff -Naur xalan_c-1.12-orig/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp xalan_c-1.12/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp
--- xalan_c-1.12-orig/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2020-06-07 12:17:25.000000000 +0300
+++ xalan_c-1.12/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2020-10-21 12:21:02.761669500 +0300
@@ -26,7 +26,7 @@
 
 
 #include <cstring>
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #include <io.h>
 #else
 #include <dirent.h>
@@ -59,7 +59,7 @@
 
 
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 
 struct FindFileStruct : public _wfinddata_t
 {
@@ -240,7 +240,7 @@
             bool                        fIncludeSelfAndParent = false)
 #endif
 {
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     FindFileStruct      theFindData;
     
     #ifdef _WIN64
diff -Naur xalan_c-1.12-orig/src/xalanc/XalanExe/XalanExe.cpp xalan_c-1.12/src/xalanc/XalanExe/XalanExe.cpp
--- xalan_c-1.12-orig/src/xalanc/XalanExe/XalanExe.cpp	2020-06-07 12:17:25.000000000 +0300
+++ xalan_c-1.12/src/xalanc/XalanExe/XalanExe.cpp	2020-10-21 12:17:22.024146300 +0300
@@ -50,7 +50,7 @@
 
 
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define XALAN_USE_WINDOWS_TIMING
 #endif
 
diff -Naur xalan_c-1.12-orig/src/xalanc/XPathCAPI/XPathCAPI.h xalan_c-1.12/src/xalanc/XPathCAPI/XPathCAPI.h
--- xalan_c-1.12-orig/src/xalanc/XPathCAPI/XPathCAPI.h	2020-06-07 12:17:25.000000000 +0300
+++ xalan_c-1.12/src/xalanc/XPathCAPI/XPathCAPI.h	2020-10-21 12:18:14.132904600 +0300
@@ -20,7 +20,7 @@
 
 
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 
 #if defined(XALAN_BUILD_DLL)
 
