--- a/superlu.pc.in
+++ b/superlu.pc.in
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@
+includedir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/superlu
 
 Name: @CMAKE_PROJECT_NAME@
 Description: Direct solution of large, sparse systems of linear equations
@@ -8,5 +8,5 @@
 URL: https://portal.nersc.gov/project/sparse/superlu/
 
 Libs: -L${libdir} -lsuperlu
-Libs.private: @BLAS_LIB@ -lm
+Requires.private: openblas
 Cflags: -I${includedir}
