diff -urN superlu-5.2.2.orig/superlu.pc.in superlu-5.2.2/superlu.pc.in
--- superlu-5.2.2.orig/superlu.pc.in	2020-10-18 03:23:21.000000000 +0500
+++ superlu-5.2.2/superlu.pc.in	2020-12-28 16:02:56.764671800 +0500
@@ -1,12 +1,12 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@
-includedir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@
+includedir=@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/superlu
 
 Name: @CMAKE_PROJECT_NAME@
 Description: Direct solution of large, sparse systems of linear equations
 Version: @PROJECT_VERSION@
-URL: http://crd-legacy.lbl.gov/~xiaoye/SuperLU/
+URL: https://portal.nersc.gov/project/sparse/superlu/
 
 Libs: -L${libdir} -lsuperlu
-Libs.private: @BLAS_LIB@ -lm
+Requires.private: openblas
 Cflags: -I${includedir}
