diff --git a/CMakeLists.txt b/CMakeLists.txt
index 719673fb..8b75385f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1168,9 +1168,9 @@ endif()
 # In case ZLIB_COMPAT=OFF, the CMake target and file follows zlib-ng naming convention
 if (ZLIB_COMPAT)
     if (TARGET zlib)
-        set_target_properties(zlib PROPERTIES EXPORT_NAME ZLIB)
+        set_target_properties(zlib PROPERTIES EXPORT_NAME ZLIB PREFIX "")
     else()
-        set_target_properties(zlibstatic PROPERTIES EXPORT_NAME ZLIB)
+        set_target_properties(zlibstatic PROPERTIES EXPORT_NAME ZLIB PREFIX "")
     endif()
 endif()

@@ -1190,20 +1190,20 @@ endforeach()
 if(WIN32)
     # Shared library
     if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
-        set_target_properties(zlib PROPERTIES OUTPUT_NAME zlib${SUFFIX})
+        set_target_properties(zlib PROPERTIES OUTPUT_NAME zlib${SUFFIX} PREFIX "")
     endif()
     # Static library
     if(NOT DEFINED BUILD_SHARED_LIBS)
         if(MSVC)
             set_target_properties(zlibstatic PROPERTIES OUTPUT_NAME zlibstatic${SUFFIX})
         else()
-            set_target_properties(zlibstatic PROPERTIES OUTPUT_NAME z${SUFFIX})
+            set_target_properties(zlibstatic PROPERTIES OUTPUT_NAME zlib${SUFFIX} PREFIX "")
         endif()
     elseif(NOT BUILD_SHARED_LIBS)
         if(MSVC)
             set_target_properties(zlib PROPERTIES OUTPUT_NAME zlibstatic${SUFFIX})
         else()
-            set_target_properties(zlib PROPERTIES OUTPUT_NAME z${SUFFIX})
+            set_target_properties(zlib PROPERTIES OUTPUT_NAME zlib${SUFFIX} PREFIX "")
         endif()
     endif()
 else()
@@ -1212,12 +1212,12 @@ else()
 endif()

 if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
-    set_target_properties(zlib PROPERTIES DEFINE_SYMBOL ZLIB_DLL)
+    set_target_properties(zlib PROPERTIES DEFINE_SYMBOL ZLIB_DLL PREFIX "")

     if(ZLIB_COMPAT)
-        set_target_properties(zlib PROPERTIES SOVERSION 1)
+        set_target_properties(zlib PROPERTIES SOVERSION 1 PREFIX "")
     else()
-        set_target_properties(zlib PROPERTIES SOVERSION 2)
+        set_target_properties(zlib PROPERTIES SOVERSION 2 PREFIX "")
     endif()

     if(NOT CYGWIN)
@@ -1228,7 +1228,7 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
         #
         # This has no effect with MSVC, on that platform the version info for
         # the DLL comes from the resource file win32/zlib1.rc
-        set_target_properties(zlib PROPERTIES VERSION ${ZLIB_FULL_VERSION})
+        set_target_properties(zlib PROPERTIES VERSION ${ZLIB_FULL_VERSION} PREFIX "")
     endif()

     if(UNIX)
@@ -1249,9 +1249,9 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
     elseif(WIN32)
         # Creates zlib1.dll when building shared library version
         if(ZLIB_COMPAT)
-            set_target_properties(zlib PROPERTIES SUFFIX "1.dll")
+            set_target_properties(zlib PROPERTIES SUFFIX "1.dll" PREFIX "")
         else()
-            set_target_properties(zlib PROPERTIES SUFFIX "2.dll")
+            set_target_properties(zlib PROPERTIES SUFFIX "2.dll" PREFIX "")
         endif()
     endif()
 endif()
