Index: mscgen/src/main.c
===================================================================
--- mscgen/src/main.c	(revision 195)
+++ mscgen/src/main.c	(revision 196)
@@ -851,6 +851,7 @@
 
     nextYmin = ymin = gOpts.entityHeadGap;
     yskipmax = 0;
+    ymax = 0;
 
     MscResetArcIterator(m);
     do
Index: mscgen/ChangeLog
===================================================================
--- mscgen/ChangeLog	(revision 195)
+++ mscgen/ChangeLog	(revision 196)
@@ -1,3 +1,7 @@
+0.21:
+      Apply patch from damian.sparrow to fix warning in usage.c (issue #72).
+      Fix lack of initialisation for ymax producing a compiler warning (issue
+      #73).
 0.20: 05/03/2011
       Fix spelling errors (issue #58)
       Add a MSC validation phase to check that all the entity names are known
