Index: mscgen-0.20/src/usage.c
===================================================================
--- mscgen-0.20.orig/src/usage.c	2011-03-01 22:16:04.000000000 +0100
+++ mscgen-0.20/src/usage.c	2012-07-20 23:53:25.415071528 +0200
@@ -59,7 +59,7 @@ void Usage(void)
 "              command line.\n"
 " -o <file>   Write output to the named file.  This option must be specified if \n"
 "              input is taken from stdin, otherwise the output filename\n"
-"              defaults to <infile>.<type>.  This may also be specified as '\-'\n"
+"              defaults to <infile>.<type>.  This may also be specified as '-'\n"
 "              to write output directly to stdout.\n"
 #ifdef USE_FREETYPE
 " -F <font>   Use specified font for PNG output.  This must be a font specifier\n"
