diff -urN mscgen-0.20-orig/src/gd_out.c mscgen-0.20/src/gd_out.c
--- mscgen-0.20-orig/src/gd_out.c	2011-03-05 18:51:50 +0800
+++ mscgen-0.20/src/gd_out.c	2021-10-30 02:09:50 +0800
@@ -204,7 +204,7 @@
                         (char *)string);
     if(r)
     {
-        fprintf(stderr, "Error: gdoTextWidth: %s (GDFONTPATH=%s)\n", r, getenv_s("GDFONTPATH"));
+        fprintf(stderr, "Error: gdoTextWidth: %s (GDFONTPATH=%s)\n", r, getenv_sf("GDFONTPATH"));
         exit(EXIT_FAILURE);
     }
 
@@ -236,7 +236,7 @@
                         "gHELLOWt");
     if(r)
     {
-        fprintf(stderr, "Error: gdoTextHeight: %s (GDFONTPATH=%s)\n", r, getenv_s("GDFONTPATH"));
+        fprintf(stderr, "Error: gdoTextHeight: %s (GDFONTPATH=%s)\n", r, getenv_sf("GDFONTPATH"));
         exit(EXIT_FAILURE);
     }
 
@@ -320,7 +320,7 @@
 
         if(r)
         {
-            fprintf(stderr, "Error: gdoTextR: %s (GDFONTPATH=%s)\n", r, getenv_s("GDFONTPATH"));
+            fprintf(stderr, "Error: gdoTextR: %s (GDFONTPATH=%s)\n", r, getenv_sf("GDFONTPATH"));
             exit(EXIT_FAILURE);
         }
 #else
diff -urN mscgen-0.20-orig/src/safe.c mscgen-0.20/src/safe.c
--- mscgen-0.20-orig/src/safe.c	2011-03-05 18:51:50 +0800
+++ mscgen-0.20/src/safe.c	2021-10-30 02:09:11 +0800
@@ -100,7 +100,7 @@
     return r;
 }
 
-const char *getenv_s(const char *name)
+const char *getenv_sf(const char *name)
 {
     char *r = getenv(name);
 
diff -urN mscgen-0.20-orig/src/safe.h mscgen-0.20/src/safe.h
--- mscgen-0.20-orig/src/safe.h	2011-03-05 18:51:50 +0800
+++ mscgen-0.20/src/safe.h	2021-10-30 02:09:21 +0800
@@ -47,7 +47,7 @@
 void *malloc_s(size_t size);
 void *zalloc_s(size_t size);
 char *strdup_s(const char *s);
-const char *getenv_s(const char *name);
+const char *getenv_sf(const char *name);
 
 /*#pragma GCC poison malloc strdup calloc*/
 
