--- udunits-2.2.28/CMakeLists.txt.orig	2020-12-29 05:57:41.000000000 +0800
+++ udunits-2.2.28/CMakeLists.txt	2026-08-18 10:47:35.286029500 +0800
@@ -83,9 +83,9 @@
 #STRING(REGEX REPLACE "\\\\" "\\\\\\\\" DEFAULT_UDUNITS2_XML_PATH "${tmp}")
 
 # Replace '\' with '\\' on Windows.
-IF(MSVC)
+IF(WIN32)
     STRING(REPLACE "\\" "\\\\" tmp "${tmp}")
-ENDIF(MSVC)
+ENDIF(WIN32)
 
 SET(DEFAULT_UDUNITS2_XML_PATH "${tmp}")
 MESSAGE("DEFAULT_UDUNITS2_XML_PATH = \"${DEFAULT_UDUNITS2_XML_PATH}\"")
