--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,11 @@
 
 # Set static/dynamic build options
 SET(LIB_TYPE STATIC)
-SET(RUNTIME_INSTALL_DIR lib)
+IF (WIN32)
+    SET(RUNTIME_INSTALL_DIR bin)
+ELSE()
+    SET(RUNTIME_INSTALL_DIR lib)
+ENDIF()
 SET(LIBRARY_INSTALL_DIR lib)
 SET(ARCHIVE_INSTALL_DIR lib)
 IF (ALEMBIC_SHARED_LIBS)
@@ -131,7 +135,7 @@
     endif()
 endif()
 
-IF (NOT ${WINDOWS})
+IF (NOT MSVC)
     SET(EXTERNAL_MATH_LIBS "-lm")
 ELSE()
     SET(EXTERNAL_MATH_LIBS "")
--- a/lib/Alembic/CMakeLists.txt	2018-10-12 16:57:05.824799900 -0400
+++ b/lib/Alembic/CMakeLists.txt	2018-10-12 17:02:05.467135400 -0400
@@ -85,11 +85,12 @@
   )
 
 SET( ALEMBIC_LIB_INSTALL_DIR lib CACHE STRING "Where to install the Alembic libs")
+SET( ALEMBIC_BIN_INSTALL_DIR bin CACHE STRING "Where to install the Alembic runtime")
 INSTALL(TARGETS Alembic
         EXPORT AlembicTargets
         LIBRARY DESTINATION ${ALEMBIC_LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${ALEMBIC_LIB_INSTALL_DIR}
-        RUNTIME DESTINATION ${ALEMBIC_LIB_INSTALL_DIR})
+        RUNTIME DESTINATION ${ALEMBIC_BIN_INSTALL_DIR})
 
 #-******************************************************************************
 # PACKAGE EXPORTS
