--- alembic-1.8.12/lib/Alembic/Util/Foundation.h.orig	2026-07-02 09:34:38.000000000 +0800
+++ alembic-1.8.12/lib/Alembic/Util/Foundation.h	2026-07-15 18:07:31.741548300 +0800
@@ -62,7 +62,7 @@
 
 #include <Alembic/Util/Export.h>
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
@@ -140,7 +140,7 @@
 };
 
 // inspired by boost::mutex
-#ifdef _MSC_VER
+#ifdef _WIN32
 
 class mutex : noncopyable
 {
