--- pkgconf-1.7.3/libpkgconf/stdinc.h.orig	2019-07-12 13:10:54.000000000 +0200
+++ pkgconf-1.7.3/libpkgconf/stdinc.h	2020-10-18 17:39:44.838906800 +0200
@@ -31,11 +31,7 @@
 # include <windows.h>
 # include <malloc.h>
 # define PATH_DEV_NULL	"nul"
-# ifdef _WIN64
-#  define SIZE_FMT_SPECIFIER	"%I64u"
-# else
-#  define SIZE_FMT_SPECIFIER	"%u"
-# endif
+# define SIZE_FMT_SPECIFIER	"%zu"
 # ifndef ssize_t
 # ifndef __MINGW32__
 #  include <BaseTsd.h>
