--- pkgconf-1.7.3/libpkgconf/path.c.orig	2019-07-12 13:10:54.000000000 +0200
+++ pkgconf-1.7.3/libpkgconf/path.c	2020-10-18 16:14:04.551602400 +0200
@@ -322,7 +322,7 @@
 {
 #if defined(HAVE_CYGWIN_CONV_PATH) && defined(__MSYS__)
 	ssize_t size;
-	char *tmpbuf, *ti;
+	char *tmpbuf;
 
 	size = cygwin_conv_path(CCP_POSIX_TO_WIN_A, buf, NULL, 0);
 	if (size < 0 || (size_t) size > buflen)
@@ -334,13 +334,6 @@
 
 	pkgconf_strlcpy(buf, tmpbuf, buflen);
 	free(tmpbuf);
-
-	/* rewrite any backslash arguments for best compatibility */
-	for (ti = buf; *ti != '\0'; ti++)
-	{
-		if (*ti == '\\')
-			*ti = '/';
-	}
 #else
 	char *tmpbuf;
 
@@ -358,5 +351,15 @@
 	}
 #endif
 
+#if defined(__MSYS__) || defined(__MINGW32__)
+	char *ti;
+	/* rewrite any backslash arguments for best compatibility */
+	for (ti = buf; *ti != '\0'; ti++)
+	{
+		if (*ti == '\\')
+			*ti = '/';
+	}
+#endif
+
 	return true;
 }
