From 384ade5f316367f15da9dbf09853e06fe29bf2bf Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Fri, 4 Aug 2023 08:24:32 +0000
Subject: [PATCH] path: prepend paths rather than append paths when processing
 --with-path arguments

---
 libpkgconf/path.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libpkgconf/path.c b/libpkgconf/path.c
index e591298d..a48a226e 100644
--- a/libpkgconf/path.c
+++ b/libpkgconf/path.c
@@ -112,7 +112,7 @@ pkgconf_path_add(const char *text, pkgconf_list_t *dirlist, bool filter)
 	}
 #endif
 
-	pkgconf_node_insert_tail(&node->lnode, node, dirlist);
+	pkgconf_node_insert(&node->lnode, node, dirlist);
 }
 
 /*
