diff --git a/libpkgconf/pkg.c b/libpkgconf/pkg.c
index 491c0de..c80269b 100644
--- a/libpkgconf/pkg.c
+++ b/libpkgconf/pkg.c
@@ -65,6 +65,20 @@ str_has_suffix(const char *str, const char *suffix)
	return !strncasecmp(str + str_len - suf_len, suffix, suf_len);
 }
 
+static inline char *
+path_sep_rchr(char *buf)
+{
+	char *pathbuf = strrchr(buf, PKG_DIR_SEP_S);
+#ifdef _WIN32
+	char *altpathbuf = strrchr(buf, '/');
+
+	if (altpathbuf != NULL && (pathbuf == NULL || altpathbuf > pathbuf))
+		pathbuf = altpathbuf;
+#endif
+
+	return pathbuf;
+}
+
 static char *
 pkg_get_parent_dir(pkgconf_pkg_t *pkg)
 {
@@ -130,9 +144,7 @@ pkg_get_parent_dir(pkgconf_pkg_t *pkg)
	}
 #endif
 
-	pathbuf = strrchr(buf, PKG_DIR_SEP_S);
-	if (pathbuf == NULL)
-		pathbuf = strrchr(buf, '/');
+	pathbuf = path_sep_rchr(buf);
	if (pathbuf != NULL)
		pathbuf[0] = '\0';
 
@@ -283,15 +295,11 @@ determine_prefix(const pkgconf_pkg_t *pkg, char *buf, size_t buflen)
	pkgconf_strlcpy(buf, pkg->filename, buflen);
	pkgconf_path_relocate(buf, buflen);
 
-	pathiter = strrchr(buf, PKG_DIR_SEP_S);
-	if (pathiter == NULL)
-		pathiter = strrchr(buf, '/');
+	pathiter = path_sep_rchr(buf);
	if (pathiter != NULL)
		pathiter[0] = '\0';
 
-	pathiter = strrchr(buf, PKG_DIR_SEP_S);
-	if (pathiter == NULL)
-		pathiter = strrchr(buf, '/');
+	pathiter = path_sep_rchr(buf);
	if (pathiter == NULL)
		return NULL;
 
@@ -301,9 +309,7 @@ determine_prefix(const pkgconf_pkg_t *pkg, char *buf, size_t buflen)
 
	/* okay, work backwards and do it again. */
	pathiter[0] = '\0';
-	pathiter = strrchr(buf, PKG_DIR_SEP_S);
-	if (pathiter == NULL)
-		pathiter = strrchr(buf, '/');
+	pathiter = path_sep_rchr(buf);
	if (pathiter == NULL)
		return NULL;
 
