diff -ru a/files/Source/Metadata/XTIFF.cpp b/files/Source/Metadata/XTIFF.cpp
--- a/files/Source/Metadata/XTIFF.cpp 2022-05-18 06:56:41.578314458 -0700
+++ b/files/Source/Metadata/XTIFF.cpp 2022-05-18 12:51:56.674454816 -0700
@@ -746,7 +746,7 @@
 				continue;
 			}
 			// type of storage may differ (e.g. rationnal array vs float array type)
-			if((unsigned)_TIFFDataSize(tif_tag_type) != FreeImage_TagDataWidth(tag_type)) {
+			if((unsigned)TIFFFieldSetGetSize(fld) != FreeImage_TagDataWidth(tag_type)) {
 				// skip tag or _TIFFmemcpy will fail
 				continue;
 			}
