diff -Naur libvpx-1.4.0-orig/build/make/configure.sh libvpx-1.4.0/build/make/configure.sh
--- libvpx-1.4.0-orig/build/make/configure.sh	2015-04-03 21:49:19.000000000 +0300
+++ libvpx-1.4.0/build/make/configure.sh	2015-04-07 09:41:34.049400000 +0300
@@ -556,11 +556,14 @@
       --libdir=*)
         libdir="${optval}"
         ;;
+      --bindir=*)
+        bindir="${optval}"
+        ;;
       --sdk-path=*)
         [ -d "${optval}" ] || die "Not a directory: ${optval}"
         sdk_path="${optval}"
         ;;
-      --libc|--as|--prefix|--libdir|--sdk-path)
+      --libc|--as|--prefix|--libdir|--bindir|--sdk-path)
         die "Option ${opt} requires argument"
         ;;
       --help|-h)
@@ -589,9 +592,14 @@
   prefix="${prefix%/}"
   libdir="${libdir:-${prefix}/lib}"
   libdir="${libdir%/}"
+  bindir="${bindir:-${prefix}/bin}"
+  bindir="${bindir%/}"
   if [ "${libdir#${prefix}}" = "${libdir}" ]; then
     die "Libdir ${libdir} must be a subdirectory of ${prefix}"
   fi
+  if [ "${bindir#${prefix}}" = "${bindir}" ]; then
+    die "Bindir ${bindir} must be a subdirectory of ${prefix}"
+  fi
 }
 
 post_process_cmdline() {
diff -Naur libvpx-1.4.0-orig/build/make/Makefile libvpx-1.4.0/build/make/Makefile
--- libvpx-1.4.0-orig/build/make/Makefile	2015-04-03 21:49:19.000000000 +0300
+++ libvpx-1.4.0/build/make/Makefile	2015-04-07 09:41:34.049400000 +0300
@@ -297,7 +297,7 @@
 	$(qexec)$$(LD) -shared $$(LDFLAGS) \
             -Wl,--no-undefined -Wl,-soname,$$(SONAME) \
             -Wl,--version-script,$$(EXPORTS_FILE) -o $$@ \
-            $$(filter %.o,$$^) $$(extralibs)
+            -Wl,--out-implib=$$(subst $(2),.dll.a,$(1)) $$(filter %.o,$$^) $$(extralibs)
 endef
 
 define dl_template
@@ -383,7 +383,7 @@
 .libs: $(LIBS)
 	@touch $@
 $(foreach lib,$(filter %_g.a,$(LIBS)),$(eval $(call archive_template,$(lib))))
-$(foreach lib,$(filter %so.$(SO_VERSION_MAJOR).$(SO_VERSION_MINOR).$(SO_VERSION_PATCH),$(LIBS)),$(eval $(call so_template,$(lib))))
+$(foreach lib,$(filter %-$(VERSION_MAJOR).dll,$(LIBS)),$(eval $(call so_template,$(lib),-$(VERSION_MAJOR).dll)))
 $(foreach lib,$(filter %$(SO_VERSION_MAJOR).dylib,$(LIBS)),$(eval $(call dl_template,$(lib))))
 
 INSTALL-LIBS=$(call cond_enabled,CONFIG_INSTALL_LIBS,INSTALL-LIBS)
diff -Naur libvpx-1.4.0-orig/configure libvpx-1.4.0/configure
--- libvpx-1.4.0-orig/configure	2015-04-03 21:49:19.000000000 +0300
+++ libvpx-1.4.0/configure	2015-04-07 09:41:34.049400000 +0300
@@ -516,6 +516,7 @@
 DIST_DIR?=\$(DESTDIR)${prefix}
 endif
 LIBSUBDIR=${libdir##${prefix}/}
+BINSUBDIR=${bindir##${prefix}/}
 
 VERSION_STRING=${VERSION_STRING}
 
@@ -549,9 +550,13 @@
         # magic happens after the command line has been parsed.
         if ! enabled linux; then
             if enabled gnu; then
-                echo "--enable-shared is only supported on ELF; assuming this is OK"
+                echo "--enable-shared is only supported on ELF and PE; assuming this is OK"
+            elif enabled win32; then
+                echo "--enable-shared is only supported on ELF and PE; assuming this is OK"
+            elif enabled win64; then
+                echo "--enable-shared is only supported on ELF and PE; assuming this is OK"
             else
-                die "--enable-shared only supported on ELF for now"
+                die "--enable-shared only supported on ELF and PE for now"
             fi
         fi
     fi
diff -Naur libvpx-1.4.0-orig/examples.mk libvpx-1.4.0/examples.mk
--- libvpx-1.4.0-orig/examples.mk	2015-04-03 21:49:19.000000000 +0300
+++ libvpx-1.4.0/examples.mk	2015-04-07 09:41:34.065000000 +0300
@@ -275,7 +275,7 @@
 
 # Instantiate linker template for all examples.
 CODEC_LIB=$(if $(CONFIG_DEBUG_LIBS),vpx_g,vpx)
-SHARED_LIB_SUF=$(if $(filter darwin%,$(TGT_OS)),.dylib,.so)
+SHARED_LIB_SUF=$(if $(filter darwin%,$(TGT_OS)),.dylib,.dll)
 CODEC_LIB_SUF=$(if $(CONFIG_SHARED),$(SHARED_LIB_SUF),.a)
 $(foreach bin,$(BINS-yes),\
     $(if $(BUILD_OBJS),$(eval $(bin):\
diff -Naur libvpx-1.4.0-orig/libs.mk libvpx-1.4.0/libs.mk
--- libvpx-1.4.0-orig/libs.mk	2015-04-03 21:49:19.000000000 +0300
+++ libvpx-1.4.0/libs.mk	2015-04-07 09:41:34.065000000 +0300
@@ -131,6 +131,7 @@
 INSTALL_MAPS += include/vpx/% $(SRC_PATH_BARE)/vpx/%
 INSTALL_MAPS += include/vpx/% $(SRC_PATH_BARE)/vpx_ports/%
 INSTALL_MAPS += $(LIBSUBDIR)/%     %
+INSTALL_MAPS += $(BINSUBDIR)/%     %
 INSTALL_MAPS += src/%     $(SRC_PATH_BARE)/%
 ifeq ($(CONFIG_MSVS),yes)
 INSTALL_MAPS += $(foreach p,$(VS_PLATFORMS),$(LIBSUBDIR)/$(p)/%  $(p)/Release/%)
@@ -239,18 +240,18 @@
 LIBVPX_SO_SYMLINKS      := $(addprefix $(LIBSUBDIR)/, \
                              libvpx.dylib  )
 else
-LIBVPX_SO               := libvpx.so.$(SO_VERSION_MAJOR).$(SO_VERSION_MINOR).$(SO_VERSION_PATCH)
+LIBVPX_SO               := libvpx-$(VERSION_MAJOR).dll
 EXPORT_FILE             := libvpx.ver
-LIBVPX_SO_SYMLINKS      := $(addprefix $(LIBSUBDIR)/, \
-                             libvpx.so libvpx.so.$(SO_VERSION_MAJOR) \
-                             libvpx.so.$(SO_VERSION_MAJOR).$(SO_VERSION_MINOR))
+LIBVPX_SO_SYMLINKS      := $(addprefix $(BINSUBDIR)/, \
+                           libvpx.dll libvpx-$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH).dll \
+                           libvpx-$(VERSION_MAJOR).$(VERSION_MINOR).dll)
 endif
 
 LIBS-$(BUILD_LIBVPX_SO) += $(BUILD_PFX)$(LIBVPX_SO)\
                            $(notdir $(LIBVPX_SO_SYMLINKS))
 $(BUILD_PFX)$(LIBVPX_SO): $(LIBVPX_OBJS) $(EXPORT_FILE)
 $(BUILD_PFX)$(LIBVPX_SO): extralibs += -lm
-$(BUILD_PFX)$(LIBVPX_SO): SONAME = libvpx.so.$(SO_VERSION_MAJOR)
+$(BUILD_PFX)$(LIBVPX_SO): SONAME = libvpx-$(VERSION_MAJOR).dll
 $(BUILD_PFX)$(LIBVPX_SO): EXPORTS_FILE = $(EXPORT_FILE)
 
 libvpx.ver: $(call enabled,CODEC_EXPORTS)
@@ -281,7 +282,7 @@
 
 
 INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(LIBVPX_SO_SYMLINKS)
-INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(LIBSUBDIR)/$(LIBVPX_SO)
+INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(BINSUBDIR)/$(LIBVPX_SO)
 
 
 LIBS-$(BUILD_LIBVPX) += vpx.pc
@@ -443,7 +444,7 @@
 BINS-$(BUILD_LIBVPX) += $(LIBVPX_TEST_BINS)
 
 CODEC_LIB=$(if $(CONFIG_DEBUG_LIBS),vpx_g,vpx)
-CODEC_LIB_SUF=$(if $(CONFIG_SHARED),.so,.a)
+CODEC_LIB_SUF=$(if $(CONFIG_SHARED),.dll,.a)
 $(foreach bin,$(LIBVPX_TEST_BINS),\
     $(if $(BUILD_LIBVPX),$(eval $(bin): \
         lib$(CODEC_LIB)$(CODEC_LIB_SUF) libgtest.a ))\
