--- libvpx-v1.2.0/configure.orig	2013-09-04 18:49:22.000000000 +0000
+++ libvpx-v1.2.0/configure	2013-10-02 08:57:41.927885400 +0000
@@ -458,9 +459,13 @@
         # magic happens after the command line has been parsed.
         if ! enabled linux; then
             if enabled gnu; then
-                echo "--enable-shared is only supported on ELF; assuming this is OK"
+                echo "--enable-shared is only supported on ELF and PE; assuming this is OK"
+            elif enabled win32; then
+                echo "--enable-shared is only supported on ELF and PE; assuming this is OK"
+            elif enabled win64; then
+                echo "--enable-shared is only supported on ELF and PE; assuming this is OK"
             else
-                die "--enable-shared only supported on ELF for now"
+                die "--enable-shared only supported on ELF and PE for now"
             fi
         fi
     fi
--- libvpx-v1.0.0/build/make/Makefile.orig	2012-01-27 22:36:39 +0400
+++ libvpx-v1.0.0/build/make/Makefile	2012-04-16 01:33:19 +0400
@@ -316,7 +316,7 @@
 .libs: $(LIBS)
 	@touch $@
 $(foreach lib,$(filter %_g.a,$(LIBS)),$(eval $(call archive_template,$(lib))))
-$(foreach lib,$(filter %so.$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH),$(LIBS)),$(eval $(call so_template,$(lib))))
+$(foreach lib,$(filter %-$(VERSION_MAJOR).dll,$(LIBS)),$(eval $(call so_template,$(lib))))
 $(foreach lib,$(filter %$(VERSION_MAJOR).dylib,$(LIBS)),$(eval $(call dl_template,$(lib))))
 
 INSTALL-LIBS=$(call cond_enabled,CONFIG_INSTALL_LIBS,INSTALL-LIBS)
--- libvpx-v1.0.0/configure.orig	2012-04-16 01:18:31 +0400
+++ libvpx-v1.0.0/configure	2012-04-16 02:01:35 +0400
@@ -406,6 +406,7 @@
 DIST_DIR?=\$(DESTDIR)${prefix}
 endif
 LIBSUBDIR=${libdir##${prefix}/}
+BINSUBDIR=${bindir##${prefix}/}
 
 VERSION_STRING=${VERSION_STRING}
 
--- libvpx-v1.0.0/build/make/configure.sh.orig	2012-01-27 22:36:39 +0400
+++ libvpx-v1.0.0/build/make/configure.sh	2012-04-16 02:15:12 +0400
@@ -477,11 +477,14 @@
         --libdir=*)
         libdir="${optval}"
         ;;
+        --bindir=*)
+        bindir="${optval}"
+        ;;
         --sdk-path=*)
         [ -d "${optval}" ] || die "Not a directory: ${optval}"
         sdk_path="${optval}"
         ;;
-        --libc|--as|--prefix|--libdir|--sdk-path)
+        --libc|--as|--prefix|--libdir|--bindir|--sdk-path)
         die "Option ${opt} requires argument"
         ;;
         --help|-h) show_help
@@ -508,9 +511,14 @@
     prefix="${prefix%/}"
     libdir="${libdir:-${prefix}/lib}"
     libdir="${libdir%/}"
+    bindir="${bindir:-${prefix}/bin}"
+    bindir="${bindir%/}"
     if [ "${libdir#${prefix}}" = "${libdir}" ]; then
         die "Libdir ${libdir} must be a subdirectory of ${prefix}"
     fi
+    if [ "${bindir#${prefix}}" = "${bindir}" ]; then
+        die "Bindir ${bindir} must be a subdirectory of ${prefix}"
+    fi
 }
 
 
--- libvpx-v1.3.0/libs.mk.orig	2014-01-10 20:12:42.000000000 +0000
+++ libvpx-v1.3.0/libs.mk	2014-02-09 17:56:21.946849800 +0000
@@ -162,6 +162,7 @@
 INSTALL_MAPS += include/vpx/% $(SRC_PATH_BARE)/vpx/%
 INSTALL_MAPS += include/vpx/% $(SRC_PATH_BARE)/vpx_ports/%
 INSTALL_MAPS += $(LIBSUBDIR)/%     %
+INSTALL_MAPS += $(BINSUBDIR)/%     %
 INSTALL_MAPS += src/%     $(SRC_PATH_BARE)/%
 ifeq ($(CONFIG_MSVS),yes)
 INSTALL_MAPS += $(foreach p,$(VS_PLATFORMS),$(LIBSUBDIR)/$(p)/%  $(p)/Release/%)
@@ -271,19 +272,19 @@
 LIBVPX_SO_SYMLINKS      := $(addprefix $(LIBSUBDIR)/, \
                              libvpx.dylib  )
 else
-LIBVPX_SO               := libvpx.so.$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH)
+LIBVPX_SO               := libvpx-$(VERSION_MAJOR).dll
 EXPORT_FILE             := libvpx.ver
 SYM_LINK                := libvpx.so
-LIBVPX_SO_SYMLINKS      := $(addprefix $(LIBSUBDIR)/, \
-                             libvpx.so libvpx.so.$(VERSION_MAJOR) \
-                             libvpx.so.$(VERSION_MAJOR).$(VERSION_MINOR))
+LIBVPX_SO_SYMLINKS      := $(addprefix $(BINSUBDIR)/, \
+                           libvpx.dll libvpx-$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH).dll \
+                           libvpx-$(VERSION_MAJOR).$(VERSION_MINOR).dll)
 endif
 
 LIBS-$(BUILD_LIBVPX_SO) += $(BUILD_PFX)$(LIBVPX_SO)\
                            $(notdir $(LIBVPX_SO_SYMLINKS))
 $(BUILD_PFX)$(LIBVPX_SO): $(LIBVPX_OBJS) $(EXPORT_FILE)
 $(BUILD_PFX)$(LIBVPX_SO): extralibs += -lm
-$(BUILD_PFX)$(LIBVPX_SO): SONAME = libvpx.so.$(VERSION_MAJOR)
+$(BUILD_PFX)$(LIBVPX_SO): SONAME = libvpx-$(VERSION_MAJOR).dll
 $(BUILD_PFX)$(LIBVPX_SO): EXPORTS_FILE = $(EXPORT_FILE)
 
 libvpx.ver: $(call enabled,CODEC_EXPORTS)
@@ -314,7 +315,7 @@
 
 
 INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(LIBVPX_SO_SYMLINKS)
-INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(LIBSUBDIR)/$(LIBVPX_SO)
+INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(BINSUBDIR)/$(LIBVPX_SO)
 
 
 LIBS-$(BUILD_LIBVPX) += vpx.pc
@@ -476,7 +477,7 @@
 INSTALL-SRCS-$(CONFIG_CODEC_SRCS) += $(LIBVPX_TEST_SRCS)
 
 CODEC_LIB=$(if $(CONFIG_DEBUG_LIBS),vpx_g,vpx)
-CODEC_LIB_SUF=$(if $(CONFIG_SHARED),.so,.a)
+CODEC_LIB_SUF=$(if $(CONFIG_SHARED),.dll,.a)
 $(foreach bin,$(LIBVPX_TEST_BINS),\
     $(if $(BUILD_LIBVPX),$(eval $(bin): \
         lib$(CODEC_LIB)$(CODEC_LIB_SUF) libgtest.a ))\
--- libvpx-v1.3.0/examples.mk.orig	2014-02-09 17:35:42.615974800 +0000
+++ libvpx-v1.3.0/examples.mk	2014-02-09 17:58:15.301744100 +0000
@@ -191,7 +191,7 @@
 
 # Instantiate linker template for all examples.
 CODEC_LIB=$(if $(CONFIG_DEBUG_LIBS),vpx_g,vpx)
-SHARED_LIB_SUF=$(if $(filter darwin%,$(TGT_OS)),.dylib,.so)
+SHARED_LIB_SUF=$(if $(filter darwin%,$(TGT_OS)),.dylib,.dll)
 CODEC_LIB_SUF=$(if $(CONFIG_SHARED),$(SHARED_LIB_SUF),.a)
 $(foreach bin,$(BINS-yes),\
     $(if $(BUILD_OBJS),$(eval $(bin):\
