diff -Naur libvpx-1.4.0-orig/configure libvpx-1.4.0/configure
--- libvpx-1.4.0-orig/configure	2015-04-07 09:41:34.642200000 +0300
+++ libvpx-1.4.0/configure	2015-04-07 09:41:34.720200000 +0300
@@ -256,7 +256,6 @@
     ${ARCH_EXT_LIST}
     vpx_ports
     stdint_h
-    pthread_h
     sys_mman_h
     unistd_h
 "
@@ -611,7 +610,6 @@
 EOF
     # check system headers
     check_header stdint.h
-    check_header pthread.h
     check_header sys/mman.h
     check_header unistd.h # for sysconf(3) and friends.
 
diff -Naur libvpx-1.4.0-orig/vp8/common/generic/systemdependent.c libvpx-1.4.0/vp8/common/generic/systemdependent.c
--- libvpx-1.4.0-orig/vp8/common/generic/systemdependent.c	2015-04-03 21:49:19.000000000 +0300
+++ libvpx-1.4.0/vp8/common/generic/systemdependent.c	2015-04-07 09:41:34.720200000 +0300
@@ -21,7 +21,8 @@
 #if CONFIG_MULTITHREAD
 #if HAVE_UNISTD_H && !defined(__OS2__)
 #include <unistd.h>
-#elif defined(_WIN32)
+#endif
+#if defined(_WIN32)
 #include <windows.h>
 typedef void (WINAPI *PGNSI)(LPSYSTEM_INFO);
 #elif defined(__OS2__)
