--- libvpx-v1.2.0/build/make/configure.sh.orig	2013-10-02 08:59:30.678194900 +0000
+++ libvpx-v1.2.0/build/make/configure.sh	2013-10-02 09:00:28.811576900 +0000
@@ -584,13 +584,13 @@
 }
 
 setup_gnu_toolchain() {
-        CC=${CC:-${CROSS}gcc}
-        CXX=${CXX:-${CROSS}g++}
-        AR=${AR:-${CROSS}ar}
-        LD=${LD:-${CROSS}${link_with_cc:-ld}}
-        AS=${AS:-${CROSS}as}
-    STRIP=${STRIP:-${CROSS}strip}
-    NM=${NM:-${CROSS}nm}
+        CC=${CC:-gcc}
+        CXX=${CXX:-g++}
+        AR=${AR:-ar}
+        LD=${LD:-${link_with_cc:-ld}}
+        AS=${AS:-as}
+    STRIP=${STRIP:-strip}
+    NM=${NM:-nm}
         AS_SFX=.s
         EXE_SFX=
 }
