--- libvpx-v1.3.0/vpx/src/svc_encodeframe.c.orig	2014-01-10 20:12:42.000000000 +0000
+++ libvpx-v1.3.0/vpx/src/svc_encodeframe.c	2014-02-09 18:20:00.603996400 +0000
@@ -23,10 +23,12 @@
 #include "vpx/vp8cx.h"
 #include "vpx/vpx_encoder.h"
 
-#if defined(__MINGW32__) && !defined(MINGW_HAS_SECURE_API)
+#if defined(__MINGW32__)
 #define strtok_r strtok_s
+#  if !defined(MINGW_HAS_SECURE_API)
 // proto from /usr/x86_64-w64-mingw32/include/sec_api/string_s.h
 _CRTIMP char *__cdecl strtok_s(char *str, const char *delim, char **context);
+#  endif
 #endif
 
 #ifdef _MSC_VER
