diff -Naur libvpx-1.4.0-orig/libs.mk libvpx-1.4.0/libs.mk
--- libvpx-1.4.0-orig/libs.mk	2015-04-07 09:41:37.746600000 +0300
+++ libvpx-1.4.0/libs.mk	2015-04-07 09:41:37.809000000 +0300
@@ -268,7 +268,7 @@
 CLEAN-OBJS += libvpx.syms
 
 define libvpx_symlink_template
-$(1): $(2)
+$(1): $(3)
 	@echo "    [LN]     $(2) $$@"
 	$(qexec)mkdir -p $$(dir $$@)
 	$(qexec)ln -sf $(2) $$@
@@ -276,10 +276,12 @@
 
 $(eval $(call libvpx_symlink_template,\
     $(addprefix $(BUILD_PFX),$(notdir $(LIBVPX_SO_SYMLINKS))),\
+    $(BUILD_PFX)$(LIBVPX_SO),\
     $(BUILD_PFX)$(LIBVPX_SO)))
 $(eval $(call libvpx_symlink_template,\
     $(addprefix $(DIST_DIR)/,$(LIBVPX_SO_SYMLINKS)),\
-    $(LIBVPX_SO)))
+    $(LIBVPX_SO),\
+    $(DIST_DIR)/bin/$(LIBVPX_SO)))
 
 
 INSTALL-LIBS-$(BUILD_LIBVPX_SO) += $(LIBVPX_SO_SYMLINKS)
