diff --git a/vp8/common/threading.h b/vp8/common/threading.h
index 183b49b..a76f636 100644
--- a/vp8/common/threading.h
+++ b/vp8/common/threading.h
@@ -25,7 +25,12 @@ extern "C" {
 /* Win32 */
 #include <process.h>
 #include <windows.h>
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2)
+#define THREAD_FUNCTION __attribute__((force_align_arg_pointer)) \
+  unsigned int __stdcall
+#else
 #define THREAD_FUNCTION unsigned int __stdcall
+#endif
 #define THREAD_FUNCTION_RETURN DWORD
 #define THREAD_SPECIFIC_INDEX DWORD
 #define pthread_t HANDLE
diff --git a/vpx_util/vpx_thread.h b/vpx_util/vpx_thread.h
index 2062abd..a975fcf 100644
--- a/vpx_util/vpx_thread.h
+++ b/vpx_util/vpx_thread.h
@@ -44,7 +44,11 @@ typedef struct {
 // simplistic pthread emulation layer
 
 // _beginthreadex requires __stdcall
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2)
+#define THREADFN __attribute__((force_align_arg_pointer)) unsigned int __stdcall
+#else
 #define THREADFN unsigned int __stdcall
+#endif
 #define THREAD_RETURN(val) (unsigned int)((DWORD_PTR)val)
 
 static INLINE int pthread_create(pthread_t* const thread, const void* attr,
