--- bullet3-2.85/examples/ExampleBrowser/GwenGUISupport/GwenParameterInterface.cpp.orig	2016-10-14 11:16:44.014102500 +0300
+++ bullet3-2.85/examples/ExampleBrowser/GwenGUISupport/GwenParameterInterface.cpp	2016-10-14 11:16:54.234323200 +0300
@@ -1,5 +1,6 @@
 #include "GwenParameterInterface.h"
 #include "gwenInternalData.h"
+#include <string.h>
 
 struct MyButtonEventHandler : public Gwen::Event::Handler
 {
