--- a/configure.cmake
+++ b/configure.cmake
@@ -35,9 +35,7 @@
 else()
     find_package(Ninja ${QT_CONFIGURE_CHECK_ninja_version})
     find_package(Gn ${QT_REPO_MODULE_VERSION} EXACT)
-    set(Python3_ROOT_DIR $ENV{PYTHON3_PATH})
-    find_package(Python3 ${QT_CONFIGURE_CHECK_python3_version})
-    unset(Python3_ROOT_DIR)
+    find_package(Python3 3.8)
     find_package(GPerf)
     find_package(BISON)
     find_package(FLEX)
