diff -Naur cmocka-1.1.3.orig/src/CMakeLists.txt cmocka-1.1.3/src/CMakeLists.txt
--- cmocka-1.1.3.orig/src/CMakeLists.txt	2018-10-05 17:16:59.053741500 -0400
+++ cmocka-1.1.3/src/CMakeLists.txt	2018-10-05 17:28:05.256940600 -0400
@@ -78,6 +78,8 @@
             ${CMAKE_INSTALL_LIBDIR}
         LIBRARY DESTINATION
             ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION
+            ${CMAKE_INSTALL_BINDIR}
         COMPONENT
             ${PROJECT_NAME})
 
