--- bsddb3-6.2.9/setup3.py.orig	2022-06-12 09:37:28.978102500 +0200
+++ bsddb3-6.2.9/setup3.py	2022-06-12 10:00:04.408752300 +0200
@@ -539,7 +539,6 @@
                                depends = ['Modules/bsddb.h'],
                                include_dirs = [ incdir ],
                                library_dirs = [ libdir ],
-                               runtime_library_dirs = [ libdir ],
                                libraries = libname,
                                extra_link_args = lflags_arg,
                                )],
