--- a/m4/mpc.m4
+++ b/m4/mpc.m4
@@ -174,36 +174,6 @@
 AC_DEFUN([MPC_WINDOWS], [
    if test "$enable_shared" = yes; then
      MPC_LDFLAGS="$MPC_LDFLAGS -no-undefined"
-     AC_MSG_CHECKING(for DLL/static gmp)
-     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#include "gmp.h"
-#if !__GMP_LIBGMP_DLL
-#error
-error
-#endif
-     ]], [[]])],[AC_MSG_RESULT(DLL)],[
-  AC_MSG_RESULT(static)
-  AC_MSG_ERROR([gmp is not available as a DLL: use --enable-static --disable-shared]) ])
-     AC_MSG_CHECKING(for DLL/static mpfr)
-     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#include "mpfr.h"
-#if !__GMP_LIBGMP_DLL
-#error
-error
-#endif
-     ]], [[]])],[AC_MSG_RESULT(DLL)],[
-  AC_MSG_RESULT(static)
-  AC_MSG_ERROR([mpfr is not available as a DLL: use --enable-static --disable-shared]) ])
-   else
-     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#include "gmp.h"
-#if __GMP_LIBGMP_DLL
-#error
-error
-#endif
-     ]], [[]])],[AC_MSG_RESULT(static)],[
-  AC_MSG_RESULT(DLL)
-  AC_MSG_ERROR([gmp is only available as a DLL: use --disable-static --enable-shared]) ])
   fi
   ;;
 ])
