--- phonon-4.11.1/CMakeLists.txt.orig	2020-03-16 12:26:31.255115200 +0300
+++ phonon-4.11.1/CMakeLists.txt	2020-03-16 12:26:45.197960800 +0300
@@ -23,6 +23,7 @@
 feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
+include(GNUInstallDirs)
 include(KDEInstallDirs)
 include(KDECMakeSettings)
 include(KDECompilerSettings)
