diff -urN phonon-4.9.0.orig/CMakeLists.txt phonon-4.9.0/CMakeLists.txt
--- phonon-4.9.0.orig/CMakeLists.txt	2016-04-18 06:25:17.000000000 +0200
+++ phonon-4.9.0/CMakeLists.txt	2016-08-18 16:47:30.008018300 +0200
@@ -125,10 +125,21 @@
        FALSE)
 
 if(PHONON_INSTALL_QT_EXTENSIONS_INTO_SYSTEM_QT)
-    # in this case, the values have to be FORCED into the cache
-    set(PHONON_QT_IMPORTS_INSTALL_DIR ${QT_IMPORTS_DIR} CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to." FORCE)
-    set(PHONON_QT_MKSPECS_INSTALL_DIR  ${QT_MKSPECS_DIR}/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to." FORCE)
-    set(PHONON_QT_PLUGIN_INSTALL_DIR   ${QT_PLUGINS_DIR}/designer CACHE PATH "The directory where Phonon Qt plugins will be installed to." FORCE)
+    if(NOT MSYS)
+        # in this case, the values have to be FORCED into the cache
+        set(PHONON_QT_IMPORTS_INSTALL_DIR ${QT_IMPORTS_DIR} CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to." FORCE)
+        set(PHONON_QT_MKSPECS_INSTALL_DIR  ${QT_MKSPECS_DIR}/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to." FORCE)
+        set(PHONON_QT_PLUGIN_INSTALL_DIR   ${QT_PLUGINS_DIR}/designer CACHE PATH "The directory where Phonon Qt plugins will be installed to." FORCE)
+    else()
+        STRING(REGEX MATCH "/mingw64.*" QT_IMPORTS_DIR ${QT_IMPORTS_DIR})
+        STRING(REGEX MATCH "/mingw64.*" QT_MKSPECS_DIR ${QT_MKSPECS_DIR})
+        STRING(REGEX MATCH "/mingw64.*" QT_PLUGINS_DIR ${QT_PLUGINS_DIR})
+
+        # in this case, the values have to be FORCED into the cache
+        set(PHONON_QT_IMPORTS_INSTALL_DIR ${QT_IMPORTS_DIR} CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to." FORCE)
+        set(PHONON_QT_MKSPECS_INSTALL_DIR  ${QT_MKSPECS_DIR}/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to." FORCE)
+        set(PHONON_QT_PLUGIN_INSTALL_DIR   ${QT_PLUGINS_DIR}/designer CACHE PATH "The directory where Phonon Qt plugins will be installed to." FORCE)
+    endif()
 else(PHONON_INSTALL_QT_EXTENSIONS_INTO_SYSTEM_QT)
     # in this case, the values can be adjusted by the user e.g. via cmake-gui, so no FORCE
 
