diff -urN phonon-4.9.0.org/CMakeLists.txt phonon-4.9.0/CMakeLists.txt
--- phonon-4.9.0.org/CMakeLists.txt	2016-04-18 06:25:17.000000000 +0200
+++ phonon-4.9.0/CMakeLists.txt	2016-08-05 11:38:55.997080300 +0200
@@ -135,9 +135,16 @@
     # Please do note that these paths are most likely random nonensense depending on what OS and distribution is used, they most likely
     # need manual adjustment for the actual envrionment.
     if(PHONON_BUILD_PHONON4QT5)
-        set(PHONON_QT_IMPORTS_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/qt5/imports CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to.")
-        set(PHONON_QT_MKSPECS_INSTALL_DIR  share/qt5/mkspecs/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to.")
-        set(PHONON_QT_PLUGIN_INSTALL_DIR   ${CMAKE_INSTALL_LIBDIR}/qt5/plugins/designer   CACHE PATH "The directory where Phonon Qt plugins will be installed to." )
+        # MSYS2 use mingw{32,64}/share path rather than /share so fix paths and warnings
+        if(MSYS)
+            set(PHONON_QT_IMPORTS_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/qt5/imports CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to.")
+            set(PHONON_QT_MKSPECS_INSTALL_DIR  ${CMAKE_INSTALL_PREFIX}/share/qt5/mkspecs/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to.")
+            set(PHONON_QT_PLUGIN_INSTALL_DIR   ${CMAKE_INSTALL_PREFIX}/share/qt5/plugins/designer   CACHE PATH "The directory where Phonon Qt plugins will be installed to." )
+        else()
+            set(PHONON_QT_IMPORTS_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/qt5/imports CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to.")
+            set(PHONON_QT_MKSPECS_INSTALL_DIR  share/qt5/mkspecs/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to.")
+            set(PHONON_QT_PLUGIN_INSTALL_DIR   ${CMAKE_INSTALL_LIBDIR}/qt5/plugins/designer   CACHE PATH "The directory where Phonon Qt plugins will be installed to." )
+        endif()
     else()
         set(PHONON_QT_IMPORTS_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/qt4/imports CACHE PATH "The directory where Phonon Qt imports (QML) will be installed to.")
         set(PHONON_QT_MKSPECS_INSTALL_DIR  share/qt4/mkspecs/modules  CACHE PATH "The directory where Phonon mkspecs will be installed to.")
@@ -149,7 +156,7 @@
 # Check whether the QML plugins will be installed into the system Qt dir, and warn if not:
 make_abs_install_path(absImportsDir "${PHONON_QT_IMPORTS_INSTALL_DIR}")
 
-if(NOT "${absImportsDir}" STREQUAL "${QT_IMPORTS_DIR}")
+if(NOT "${absImportsDir}" STREQUAL "${QT_IMPORTS_DIR}" AND NOT MSYS)
     message(STATUS "PHONON_QT_IMPORTS_DIR is set to ${absImportsDir}.
       The QML plugins for Phonon will not be installed into the Qt system installation directory,
       which is ${QT_IMPORTS_DIR} .
@@ -164,7 +171,7 @@
 # Check whether the mkspecs file will be installed into the system Qt dir, and warn if not:
 make_abs_install_path(absMkspecsDir "${PHONON_QT_MKSPECS_INSTALL_DIR}")
 
-if(NOT "${absMkspecsDir}" STREQUAL "${QT_MKSPECS_DIR}/modules")
+if(NOT "${absMkspecsDir}" STREQUAL "${QT_MKSPECS_DIR}/modules" AND NOT MSYS)
     message(STATUS "PHONON_QT_MKSPECS_INSTALL_DIR is set to ${absMkspecsDir}.
       The Qt mkspecs file for Phonon will not be installed into the Qt system installation directory,
       which is ${QT_MKSPECS_DIR}/modules .
@@ -179,7 +186,7 @@
 # Check whether the designer plugins will be installed into the system Qt dir, and warn if not:
 make_abs_install_path(absPluginDir "${PHONON_QT_PLUGIN_INSTALL_DIR}")
 
-if(NOT "${absPluginDir}" STREQUAL "${QT_PLUGINS_DIR}/designer")
+if(NOT "${absPluginDir}" STREQUAL "${QT_PLUGINS_DIR}/designer" AND NOT MSYS)
     message(STATUS "PHONON_QT_PLUGIN_INSTALL_DIR is set to ${absPluginDir}.
       The Qt designer plugins for Phonon will not be installed into the Qt system installation directory,
       which is ${QT_PLUGINS_DIR}/designer .
