--- ngtcp2-1.6.0/lib/libngtcp2.pc.in.orig	2024-06-13 11:54:29.073135500 +0200
+++ ngtcp2-1.6.0/lib/libngtcp2.pc.in	2024-07-31 20:53:37.924575600 +0200
@@ -31,3 +31,4 @@
 Version: @VERSION@
 Libs: -L${libdir} -lngtcp2
 Cflags: -I${includedir}
+Cflags.private: -DNGTCP2_STATICLIB
--- ngtcp2-1.6.0/crypto/gnutls/libngtcp2_crypto_gnutls.pc.in.orig	2024-06-13 11:54:29.057135500 +0200
+++ ngtcp2-1.6.0/crypto/gnutls/libngtcp2_crypto_gnutls.pc.in	2024-07-31 20:57:37.269221700 +0200
@@ -31,3 +31,4 @@
 Version: @VERSION@
 Libs: -L${libdir} -lngtcp2_crypto_gnutls
 Cflags: -I${includedir}
+Cflags.private: -DNGTCP2_STATICLIB
