--- a/src/callbacks.c
+++ b/src/callbacks.c
@@ -1212,7 +1212,7 @@
 	}
 }
 
-G_MODULE_EXPORT void
+void
 on_button_confirm_yes_clicked (GtkButton * button, gpointer user_data)
 {
 	gchar *file;
@@ -1372,7 +1372,7 @@
  * 5 - Charts window
  **********************************************************************/
 
-G_MODULE_EXPORT void
+void
 on_combobox_stat_module_changed (GtkComboBox *cmb, gpointer user_data)
 {
 	static gchar *stat_title = NULL;
@@ -1445,7 +1445,7 @@
 	plot_draw_chart (1);
 }
 
-G_MODULE_EXPORT void
+void
 on_combobox_stat_type_changed (GtkComboBox *cmb, gpointer user_data)
 {
 	static gboolean init = TRUE;
