--- gtk-doc-1.18/acinclude.m4.orig	2011-06-17 12:39:37 +0400
+++ gtk-doc-1.18/acinclude.m4	2012-09-07 12:18:56 +0400
@@ -9,7 +9,11 @@
   AC_ARG_WITH([xml-catalog],
               AC_HELP_STRING([--with-xml-catalog=CATALOG],
                              [path to xml catalog to use]),,
-              [with_xml_catalog=/etc/xml/catalog])
+              [for c in $XML_CATALOG_FILES /etc/xml/catalog
+               do
+                 with_xml_catalog=$c
+                 break
+               done])
   jh_found_xmlcatalog=true
   XML_CATALOG_FILE="$with_xml_catalog"
   AC_SUBST([XML_CATALOG_FILE])
