--- gtk-doc-1.34.0/gtkdoc/meson.build.orig	2024-09-17 12:21:35.078577400 +0200
+++ gtk-doc-1.34.0/gtkdoc/meson.build	2024-09-17 12:22:11.010584300 +0200
@@ -9,7 +9,11 @@
 config_py.set('XSLTPROC', xsltproc_prg_path)
 
 config_py.set('VERSION', meson.project_version())
-config_py.set('EXEEXT', '')
+if host_machine.system() == 'windows'
+  config_py.set('EXEEXT', '.exe')
+else
+  config_py.set('EXEEXT', '')
+endif
 
 configure_file(
   input: 'config_data.py.in',
