--- gtk-doc-1.18/gtkdoc-scangobj.in.orig	2013-04-06 18:29:34 +0400
+++ gtk-doc-1.18/gtkdoc-scangobj.in	2013-04-06 21:00:15 +0400
@@ -1380,7 +1380,7 @@
 # Linking scanner
 # FIXME: Can we turn off as-needed for the docs (or better fix it?)
 #$command = "$LD -Wl,--no-as-needed $o_file $LDFLAGS -o $MODULE-scan";
-$command = "$LD $stdout $o_file $LDFLAGS -o $MODULE-scan";
+$command = "$LD $stdout $o_file $LDFLAGS -o $MODULE-scan.exe";
 system("($command)") == 0 or die "Linking of scanner failed: $!\n";
 
 # Running scanner $MODULE-scan ";
--- gtk-doc-1.18/gtkdoc-scanobj.in.orig	2013-04-06 18:29:34 +0400
+++ gtk-doc-1.18/gtkdoc-scanobj.in	2013-04-06 21:00:41 +0400
@@ -538,7 +538,7 @@
 # Linking scanner
 # FIXME: Can we turn off as-needed for the docs (or better fix it?)
 #$command = "$LD -Wl,--no-as-needed $o_file $LDFLAGS -o $MODULE-scan";
-$command = "$LD $o_file $LDFLAGS -o $MODULE-scan";
+$command = "$LD $o_file $LDFLAGS -o $MODULE-scan.exe";
 system("($command)") == 0 or die "Linking of scanner failed: $!\n";
 
 # Running scanner $MODULE-scan ";
