diff -ur gtk-doc-1.18.orig/gtk-doc.flat.make gtk-doc-1.18/gtk-doc.flat.make
--- gtk-doc-1.18.orig/gtk-doc.flat.make	2013-04-12 14:54:01 +0400
+++ gtk-doc-1.18/gtk-doc.flat.make	2013-04-16 19:42:12 +0400
@@ -92,7 +92,7 @@
 
 #### scan ####
 
-scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	@echo '  DOC   Scanning header files'
 	@_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
diff -ur gtk-doc-1.18.orig/gtk-doc.make gtk-doc-1.18/gtk-doc.make
--- gtk-doc-1.18.orig/gtk-doc.make	2013-04-12 14:54:01 +0400
+++ gtk-doc-1.18/gtk-doc.make	2013-04-16 19:42:16 +0400
@@ -92,7 +92,7 @@
 
 #### scan ####
 
-scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	@echo '  DOC   Scanning header files'
 	@_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
diff -ur gtk-doc-1.18.orig/gtk-doc.notmpl-flat.make gtk-doc-1.18/gtk-doc.notmpl-flat.make
--- gtk-doc-1.18.orig/gtk-doc.notmpl-flat.make	2013-04-12 14:54:01 +0400
+++ gtk-doc-1.18/gtk-doc.notmpl-flat.make	2013-04-16 19:42:20 +0400
@@ -90,7 +90,7 @@
 
 #### scan ####
 
-scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	@echo '  DOC   Scanning header files'
 	@_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
diff -ur gtk-doc-1.18.orig/gtk-doc.notmpl.make gtk-doc-1.18/gtk-doc.notmpl.make
--- gtk-doc-1.18.orig/gtk-doc.notmpl.make	2013-04-12 14:54:01 +0400
+++ gtk-doc-1.18/gtk-doc.notmpl.make	2013-04-16 19:42:23 +0400
@@ -90,7 +90,7 @@
 
 #### scan ####
 
-scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	@echo '  DOC   Scanning header files'
 	@_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
diff -ur gtk-doc-1.18.orig/tests/gtk-doc.make gtk-doc-1.18/tests/gtk-doc.make
--- gtk-doc-1.18.orig/tests/gtk-doc.make	2013-04-12 14:54:01 +0400
+++ gtk-doc-1.18/tests/gtk-doc.make	2013-04-16 19:42:32 +0400
@@ -85,7 +85,7 @@
 
 #### scan ####
 
-scan-build.stamp: ts $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: ts $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	@ts1=`cat ts`;ts2=`date +%s.%N`;tsd=`echo $$ts2-$$ts1 | bc`; \
 	echo "  DOC   `date --utc --date @0$$tsd +%H:%M:%S.%N`: Scanning header files"
 	@_source_dir='' ; \
diff -ur gtk-doc-1.18.orig/tests/gtk-doc.notmpl.make gtk-doc-1.18/tests/gtk-doc.notmpl.make
--- gtk-doc-1.18.orig/tests/gtk-doc.notmpl.make	2013-04-12 14:54:01 +0400
+++ gtk-doc-1.18/tests/gtk-doc.notmpl.make	2013-04-16 19:42:35 +0400
@@ -82,7 +82,7 @@
 
 #### scan ####
 
-scan-build.stamp: ts $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: ts $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	@ts1=`cat ts`;ts2=`date +%s.%N`;tsd=`echo $$ts2-$$ts1 | bc`; \
 	echo "  DOC   `date --utc --date @0$$tsd +%H:%M:%S.%N`: Scanning header files"
 	@_source_dir='' ; \
