diff -ur gtk-doc-1.18.orig/build-aux/gnome-doc-utils.make gtk-doc-1.18/build-aux/gnome-doc-utils.make
--- gtk-doc-1.18.orig/build-aux/gnome-doc-utils.make	2011-06-23 13:09:14.000000000 +0000
+++ gtk-doc-1.18/build-aux/gnome-doc-utils.make	2013-08-03 19:40:45.086617800 +0000
@@ -133,12 +133,12 @@
 _xml2po ?= `which xml2po`
 _xml2po_mode = $(if $(DOC_ID),mallard,docbook)
 
-_db2html ?= `$(PKG_CONFIG) --variable db2html gnome-doc-utils`
-_db2omf  ?= `$(PKG_CONFIG) --variable db2omf gnome-doc-utils`
-_malrng  ?= `$(PKG_CONFIG) --variable malrng gnome-doc-utils`
-_chunks  ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
-_credits ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
-_ids ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
+_db2html ?= `$(PKG_CONFIG) --dont-define-prefix --variable db2html gnome-doc-utils`
+_db2omf  ?= `$(PKG_CONFIG) --dont-define-prefix --variable db2omf gnome-doc-utils`
+_malrng  ?= `$(PKG_CONFIG) --dont-define-prefix --variable malrng gnome-doc-utils`
+_chunks  ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
+_credits ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
+_ids ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
 
 if ENABLE_SK
 _ENABLE_SK = true
diff -ur gtk-doc-1.18.orig/configure gtk-doc-1.18/configure
--- gtk-doc-1.18.orig/configure	2013-08-03 18:46:55.133965900 +0000
+++ gtk-doc-1.18/configure	2013-08-03 19:40:14.885282700 +0000
@@ -12721,7 +12721,7 @@
 	TEST_DEPS_LIBS=$pkg_cv_TEST_DEPS_LIBS
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-		glib_prefix="`$PKG_CONFIG --variable=prefix glib-2.0`"
+		glib_prefix="`$PKG_CONFIG --dont-define-prefix --variable=prefix glib-2.0`"
 		gtk_doc_use_libtool="yes"
 		build_tests="yes"
 
diff -ur gtk-doc-1.18.orig/configure.ac gtk-doc-1.18/configure.ac
--- gtk-doc-1.18.orig/configure.ac	2013-08-03 18:46:45.504243100 +0000
+++ gtk-doc-1.18/configure.ac	2013-08-03 19:40:09.265569100 +0000
@@ -193,7 +193,7 @@
 
 dnl if glib is available we can enable the tests
 PKG_CHECK_MODULES(TEST_DEPS, [glib-2.0 >= 2.6.0 gobject-2.0 >= 2.6.0],
-	[	glib_prefix="`$PKG_CONFIG --variable=prefix glib-2.0`"
+	[	glib_prefix="`$PKG_CONFIG --dont-define-prefix --variable=prefix glib-2.0`"
 		gtk_doc_use_libtool="yes"
 		build_tests="yes"
 	],
diff -ur gtk-doc-1.18.orig/gnome-doc-utils.make gtk-doc-1.18/gnome-doc-utils.make
--- gtk-doc-1.18.orig/gnome-doc-utils.make	2011-06-17 08:39:55.000000000 +0000
+++ gtk-doc-1.18/gnome-doc-utils.make	2013-08-03 19:41:45.952846800 +0000
@@ -133,12 +133,12 @@
 _xml2po ?= `which xml2po`
 _xml2po_mode = $(if $(DOC_ID),mallard,docbook)
 
-_db2html ?= `$(PKG_CONFIG) --variable db2html gnome-doc-utils`
-_db2omf  ?= `$(PKG_CONFIG) --variable db2omf gnome-doc-utils`
-_malrng  ?= `$(PKG_CONFIG) --variable malrng gnome-doc-utils`
-_chunks  ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
-_credits ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
-_ids ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
+_db2html ?= `$(PKG_CONFIG) --dont-define-prefix --variable db2html gnome-doc-utils`
+_db2omf  ?= `$(PKG_CONFIG) --dont-define-prefix --variable db2omf gnome-doc-utils`
+_malrng  ?= `$(PKG_CONFIG) --dont-define-prefix --variable malrng gnome-doc-utils`
+_chunks  ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
+_credits ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
+_ids ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
 
 if ENABLE_SK
 _ENABLE_SK = true
diff -ur gtk-doc-1.18.orig/gtkdoc-depscan.in gtk-doc-1.18/gtkdoc-depscan.in
--- gtk-doc-1.18.orig/gtkdoc-depscan.in	2011-08-03 12:25:53.000000000 +0000
+++ gtk-doc-1.18/gtkdoc-depscan.in	2013-08-03 19:40:20.330474200 +0000
@@ -262,7 +262,7 @@
     path = path and path.split(':') or []
 
     prefix = popen(
-        'pkg-config --variable=prefix glib-2.0'
+        'pkg-config --dont-define-prefix --variable=prefix glib-2.0'
         ).readline().rstrip()
 
     path.insert(0, prefix)
diff -ur gtk-doc-1.18.orig/gtkdoc-fixxref.in gtk-doc-1.18/gtkdoc-fixxref.in
--- gtk-doc-1.18.orig/gtkdoc-fixxref.in	2011-08-19 13:10:14.000000000 +0000
+++ gtk-doc-1.18/gtkdoc-fixxref.in	2013-08-03 19:40:26.043199600 +0000
@@ -109,7 +109,7 @@
 
 # We scan the directory containing GLib and any directories in GNOME2_PATH
 # first, but these will be overriden by any later scans.
-$dir = `pkg-config --variable=prefix glib-2.0`;
+$dir = `pkg-config --dont-define-prefix --variable=prefix glib-2.0`;
 $dir =~ s/\s+$//;
 $dir = $dir . "/share/gtk-doc/html";
 if (-d $dir && $dir ne $HTML_DIR) {
diff -ur gtk-doc-1.18.orig/gtkdoc-rebase.in gtk-doc-1.18/gtkdoc-rebase.in
--- gtk-doc-1.18.orig/gtkdoc-rebase.in	2011-06-17 08:39:37.000000000 +0000
+++ gtk-doc-1.18/gtkdoc-rebase.in	2013-08-03 19:40:30.496265100 +0000
@@ -108,7 +108,7 @@
     }
 }
 
-$dir = `pkg-config --variable=prefix glib-2.0`;
+$dir = `pkg-config --dont-define-prefix --variable=prefix glib-2.0`;
 $dir =~ s/^\s*(\S*)\s*$/$1/;
 $dir = $dir . "/share/gtk-doc/html";
 print "Prepending GLib directory $dir\n" if $VERBOSE;
diff -ur gtk-doc-1.18.orig/help/manual/Makefile.in gtk-doc-1.18/help/manual/Makefile.in
--- gtk-doc-1.18.orig/help/manual/Makefile.in	2013-08-03 18:46:57.457260900 +0000
+++ gtk-doc-1.18/help/manual/Makefile.in	2013-08-03 19:40:58.302296000 +0000
@@ -644,12 +644,12 @@
 
 @HAVE_GNOME_DOC_UTILS_TRUE@_xml2po ?= `which xml2po`
 
-@HAVE_GNOME_DOC_UTILS_TRUE@_db2html ?= `$(PKG_CONFIG) --variable db2html gnome-doc-utils`
-@HAVE_GNOME_DOC_UTILS_TRUE@_db2omf  ?= `$(PKG_CONFIG) --variable db2omf gnome-doc-utils`
-@HAVE_GNOME_DOC_UTILS_TRUE@_malrng  ?= `$(PKG_CONFIG) --variable malrng gnome-doc-utils`
-@HAVE_GNOME_DOC_UTILS_TRUE@_chunks  ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
-@HAVE_GNOME_DOC_UTILS_TRUE@_credits ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
-@HAVE_GNOME_DOC_UTILS_TRUE@_ids ?= `$(PKG_CONFIG) --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
+@HAVE_GNOME_DOC_UTILS_TRUE@_db2html ?= `$(PKG_CONFIG) --dont-define-prefix --variable db2html gnome-doc-utils`
+@HAVE_GNOME_DOC_UTILS_TRUE@_db2omf  ?= `$(PKG_CONFIG) --dont-define-prefix --variable db2omf gnome-doc-utils`
+@HAVE_GNOME_DOC_UTILS_TRUE@_malrng  ?= `$(PKG_CONFIG) --dont-define-prefix --variable malrng gnome-doc-utils`
+@HAVE_GNOME_DOC_UTILS_TRUE@_chunks  ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/chunks.xsl
+@HAVE_GNOME_DOC_UTILS_TRUE@_credits ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/credits.xsl
+@HAVE_GNOME_DOC_UTILS_TRUE@_ids ?= `$(PKG_CONFIG) --dont-define-prefix --variable xmldir gnome-doc-utils`/gnome/xslt/docbook/utils/ids.xsl
 @ENABLE_SK_TRUE@@HAVE_GNOME_DOC_UTILS_TRUE@_skpkgdatadir ?= `scrollkeeper-config --pkgdatadir`
 @ENABLE_SK_TRUE@@HAVE_GNOME_DOC_UTILS_TRUE@_sklocalstatedir ?= `scrollkeeper-config --pkglocalstatedir`
 @ENABLE_SK_TRUE@@HAVE_GNOME_DOC_UTILS_TRUE@_skcontentslist ?= $(_skpkgdatadir)/Templates/C/scrollkeeper_cl.xml
