--- a/configure.ac	Thu Apr 07 14:13:07 2016 +0200
+++ b/configure.ac	Wed Apr 13 13:25:32 2016 +0200
@@ -65,6 +65,13 @@
     AC_DEFINE([HAVE___PROGNAME], [1], [Define if libc defines __progname])
 fi

+AC_CACHE_CHECK([whether we are building for a Win32 host], [ac_cv_win32_host],
+	       AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#ifdef _WIN32
+ choke me
+#endif
+ ]])],
+ [ac_cv_win32_host=no], [ac_cv_win32_host=yes]))
+AM_CONDITIONAL([WIN32_HOST], [test "x$ac_cv_win32_host" = "xyes"])

 AC_CHECK_HEADERS([fts.h stdbool.h strings.h unistd.h])

--- a/lib/Makefile.am	Thu Apr 07 14:13:07 2016 +0200
+++ b/lib/Makefile.am	Wed Apr 13 13:25:32 2016 +0200
@@ -10,7 +10,19 @@
 libzip_la_LDFLAGS=-no-undefined -version-info 4:0:0
 libzip_la_LIBADD=@LTLIBOBJS@

+if WIN32_HOST
+IO_SOURCES=\
+	zip_source_win32a.c \
+	zip_source_win32handle.c \
+	zip_source_win32utf8.c \
+	zip_source_win32w.c
+else
+IO_SOURCES=\
+	zip_source_file.c
+endif
+
 libzip_la_SOURCES=\
+	${IO_SOURCES} \
	zip_add.c \
	zip_add_dir.c \
	zip_add_entry.c \
@@ -80,7 +92,6 @@
	zip_source_crc.c \
	zip_source_deflate.c \
	zip_source_error.c \
-	zip_source_file.c \
	zip_source_filep.c \
	zip_source_free.c \
	zip_source_function.c \
