diff -bur poppler-22.07.0/glib/CMakeLists.txt poppler-22.07.0-c/glib/CMakeLists.txt
--- poppler-22.07.0/glib/CMakeLists.txt	2022-07-04 12:09:04.000000000 -0600
+++ poppler-22.07.0-c/glib/CMakeLists.txt	2022-07-31 12:34:35.871905700 -0600
@@ -43,9 +43,8 @@
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.h
-  COMMAND ${GLIB2_MKENUMS_PYTHON}
+  COMMAND ${GLIB2_MKENUMS}
   ARGS
-    ${GLIB2_MKENUMS}
     --template poppler-enums.h.template
     ${poppler_glib_public_headers} > ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
@@ -55,9 +54,8 @@
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.c
-  COMMAND ${GLIB2_MKENUMS_PYTHON}
+  COMMAND ${GLIB2_MKENUMS}
   ARGS
-    ${GLIB2_MKENUMS}
     --template poppler-enums.c.template
     ${poppler_glib_public_headers} > ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.c
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
