--- CMakeLists.txt	2024-11-15 17:35:15.354633922 +0100
+++ CMakeLists.txt	2024-11-15 17:38:16.349196965 +0100
@@ -713,6 +713,7 @@
     poppler/Stream-CCITT.h
     poppler/Stream.h
     poppler/StructElement.h
+    poppler/StructTreeRoot.h
     poppler/UnicodeMap.h
     poppler/UnicodeMapFuncs.h
     poppler/UnicodeMapTables.h
@@ -755,6 +756,8 @@
     splash/SplashBitmap.h
     splash/SplashClip.h
     splash/SplashErrorCodes.h
+    splash/SplashFTFont.h
+    splash/SplashFont.h
     splash/SplashFontEngine.h
     splash/SplashFontFile.h
     splash/SplashFontFileID.h
