From 5b6861516896d6f07ce50b1cdc2146c63b0f7034 Mon Sep 17 00:00:00 2001
From: Jan Sundermeyer <sunderme@gmx.de>
Date: Sun, 28 Dec 2025 18:52:30 +0100
Subject: [PATCH] fix crash

---
 poppler/SplashOutputDev.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/poppler/SplashOutputDev.cc b/poppler/SplashOutputDev.cc
index 356c4dd1a..21a6d3a65 100644
--- a/poppler/SplashOutputDev.cc
+++ b/poppler/SplashOutputDev.cc
@@ -1880,6 +1880,9 @@ reload:
         if (!fileName.empty()) {
             fontsrc->setFile(fileName);
         } else {
+            if (!tmpBuf) {
+                goto err;
+            }
             fontsrc->setBuf(std::move(tmpBuf.value()));
         }
 
-- 
GitLab

