From 77d32d6fc0ccc66efe2af229cb13bcd41d189c00 Mon Sep 17 00:00:00 2001
From: Jan Sundermeyer <sunderme@gmx.de>
Date: Sun, 28 Dec 2025 18:52:30 +0100
Subject: [PATCH 1/2] fix crash

---
 poppler/SplashOutputDev.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/poppler/SplashOutputDev.cc b/poppler/SplashOutputDev.cc
index 356c4dd1a3..21a6d3a65d 100644
--- a/poppler/SplashOutputDev.cc
+++ b/poppler/SplashOutputDev.cc
@@ -1880,6 +1880,9 @@ reload:
         if (!fileName.empty()) {
             fontsrc->setFile(fileName);
         } else {
+            if (!tmpBuf) {
+                goto err;
+            }
             fontsrc->setBuf(std::move(tmpBuf.value()));
         }
 
-- 
GitLab


From 06c83ea59ac77bae1fdc13fd9deefa6dd677145b Mon Sep 17 00:00:00 2001
From: Jan Sundermeyer <sunderme@gmx.de>
Date: Sun, 4 Jan 2026 10:06:15 +0100
Subject: [PATCH 2/2] fix locking issue

---
 poppler/GlobalParams.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/poppler/GlobalParams.cc b/poppler/GlobalParams.cc
index c2b572a024..fd69248f31 100644
--- a/poppler/GlobalParams.cc
+++ b/poppler/GlobalParams.cc
@@ -858,8 +858,8 @@ static FcPattern *buildFcPattern(const GfxFont &font, const GooString *base14Nam
 
 std::optional<std::string> GlobalParams::findFontFile(const std::string &fontName)
 {
-    setupBaseFonts(POPPLER_FONTSDIR);
     globalParamsLocker();
+    setupBaseFonts(POPPLER_FONTSDIR);
     const auto fontFile = fontFiles.find(fontName);
     if (fontFile != fontFiles.end()) {
         return fontFile->second;
-- 
GitLab

