--- a/glib/poppler-private.h
+++ b/glib/poppler-private.h
@@ -150,7 +150,7 @@ GooString *_poppler_convert_date_time_to_pdf_date(GDateTime *datetime);
 #define POPPLER_DEFINE_BOXED_TYPE(TypeName, type_name, copy_func, free_func)                                                                                                                                                                   \
     GType type_name##_get_type(void)                                                                                                                                                                                                           \
     {                                                                                                                                                                                                                                          \
-        static volatile gsize g_define_type_id__volatile = 0;                                                                                                                                                                                  \
+        static gsize g_define_type_id__volatile = 0;                                                                                                                                                                                  \
         if (g_once_init_enter(&g_define_type_id__volatile)) {                                                                                                                                                                                  \
             GType g_define_type_id = g_boxed_type_register_static(g_intern_static_string(#TypeName), (GBoxedCopyFunc)copy_func, (GBoxedFreeFunc)free_func);                                                                                    \
             g_once_init_leave(&g_define_type_id__volatile, g_define_type_id);                                                                                                                                                                  \
