diff -urN flex-2.5.39.orig/tables.c flex-2.5.39/tables.c
--- flex-2.5.39.orig/tables.c	2014-08-06 13:19:47.243084400 +0100
+++ flex-2.5.39/tables.c	2014-08-06 13:32:49.126584400 +0100
@@ -36,6 +36,15 @@
 #include "flexdef.h"
 #include "tables.h"
 
+#if defined(_WIN32)
+#include <stdint.h>
+#define htonl(_val) ( ((uint16_t)(_val) & 0xff00) >> 8 | ((uint16_t)(_val) & 0xff) << 8 )
+#define htons(_val) ( ((uint32_t)(_val) & 0xff000000) >> 24 | \
+                      ((uint32_t)(_val) & 0x00ff0000) >>  8 | \
+                      ((uint32_t)(_val) & 0x0000ff00) <<  8 | \
+                      ((uint32_t)(_val) & 0x000000ff) << 24 )
+#endif
+
 /** Convert size_t to t_flag.
  *  @param n in {1,2,4}
  *  @return YYTD_DATA*. 
Binary files flex-2.5.39.orig/tables.o and flex-2.5.39/tables.o differ
