diff -urN flex-2.5.39.orig/configure.ac flex-2.5.39/configure.ac
--- flex-2.5.39.orig/configure.ac	2014-08-06 13:37:52.128853300 +0100
+++ flex-2.5.39/configure.ac	2014-08-06 13:41:09.972914200 +0100
@@ -96,6 +96,9 @@
 
 AC_CHECK_LIB(m, log10)
 
+# Needed for MSYS2's MinGW-w64.
+AC_SEARCH_LIBS(regcomp, [regex gnurx])
+
 # Checks for typedefs, structures, and compiler characteristics.
 
 AC_HEADER_STDBOOL
