diff -urN flex-2.5.39.orig/filter.c flex-2.5.39/filter.c
--- flex-2.5.39.orig/filter.c	2014-03-26 12:46:44.000000000 +0000
+++ flex-2.5.39/filter.c	2014-08-06 17:54:18.843588400 +0100
@@ -152,6 +152,7 @@
 	/* Now we are the right-most unprocessed link in the chain.
 	 */
 
+#ifdef HAVE_FORK
 	fflush (stdout);
 	fflush (stderr);
 
@@ -177,6 +178,7 @@
 			flexfatal (_("dup2(pipes[0],0)"));
 		close (pipes[0]);
         fseek (stdin, 0, SEEK_CUR);
+#endif
 
 		/* run as a filter, either internally or by exec */
 		if (chain->filter_func) {
@@ -193,6 +195,7 @@
                     chain->argv[0]);
 		}
 
+#ifdef HAVE_PIPE
 		exit (1);
 	}
 
@@ -203,6 +206,7 @@
 	close (pipes[1]);
     fseek (stdout, 0, SEEK_CUR);
 
+#endif
 	return true;
 }
 
@@ -329,9 +333,11 @@
 		lerrsf (_("error closing output file %s"),
 			outfilename ? outfilename : "<stdout>");
 
+#ifdef HAVE_FORK
 	while (wait (0) > 0) ;
 
 	exit (0);
+#endif
 	return 0;
 }
 
