diff -bur SDL_mixer-1.2.12-o/mixer.c SDL_mixer-1.2.12/mixer.c
--- SDL_mixer-1.2.12-o/mixer.c	2025-04-13 21:48:38.415353500 -0600
+++ SDL_mixer-1.2.12/mixer.c	2025-04-13 21:49:14.857172700 -0600
@@ -39,6 +39,10 @@
 #include "dynamic_mp3.h"
 #include "dynamic_ogg.h"
 
+#ifdef USE_FLUIDSYNTH_MIDI
+  #include "dynamic_fluidsynth.h"
+#endif
+
 #define __MIX_INTERNAL_EFFECT__
 #include "effects_internal.h"
 
