diff -urN kpty-5.25.0.orig/CMakeLists.txt kpty-5.25.0/CMakeLists.txt
--- kpty-5.25.0.orig/CMakeLists.txt	2016-08-07 15:56:27.000000000 +0200
+++ kpty-5.25.0/CMakeLists.txt	2016-08-23 12:46:13.598686100 +0200
@@ -42,6 +42,7 @@
 include(ECMSetupVersion)
 include(ECMGenerateHeaders)
 include(ECMPackageConfigHelpers)
+include(ECMMarkNonGuiExecutable)
 
 ecm_setup_version(
   ${KF5_VERSION}
diff -urN kpty-5.25.0.orig/src/kgrantpty.c kpty-5.25.0/src/kgrantpty.c
--- kpty-5.25.0.orig/src/kgrantpty.c	2016-08-07 15:56:27.000000000 +0200
+++ kpty-5.25.0/src/kgrantpty.c	2016-08-23 12:59:15.870556500 +0200
@@ -40,15 +40,24 @@
 
 #include <config-pty.h>
 
-#include <sys/types.h>
 #include <errno.h>
-#include <grp.h>
+
+#ifdef _WIN32
+# include <mingw/sys/types.h>
+# include <mingw/grp.h>
+# include <mingw/sys/ioctl.h>
+# include <mingw/unistd.h>
+#else
+# include <sys/types.h>
+# include <grp.h>
+# include <sys/ioctl.h>
+# include <unistd.h>
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
-#include <sys/ioctl.h>
-#include <unistd.h>
 #if HAVE_PTY_H
 #  include <pty.h>
 #endif
diff -urN kpty-5.25.0.orig/src/kpty.cpp kpty-5.25.0/src/kpty.cpp
--- kpty-5.25.0.orig/src/kpty.cpp	2016-08-23 12:45:04.545236300 +0200
+++ kpty-5.25.0/src/kpty.cpp	2016-08-23 12:58:56.559044100 +0200
@@ -46,9 +46,20 @@
 #endif
 
 #include <sys/types.h>
-#include <sys/ioctl.h>
+
+#ifdef _WIN32
+# include <mingw/sys/ioctl.h>
+# include <mingw/sys/resource.h>
+# include <mingw/grp.h>
+# include <mingw/unistd.h>
+#else
+# include <sys/ioctl.h>
+# include <sys/resource.h>
+# include <grp.h>
+# include <unistd.h>
+#endif
+
 #include <sys/time.h>
-#include <sys/resource.h>
 #include <sys/stat.h>
 #include <sys/param.h>
 
@@ -58,8 +69,6 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <unistd.h>
-#include <grp.h>
 
 #if HAVE_PTY_H
 # include <pty.h>
