diff --git a/Cargo.lock b/Cargo.lock
index 7135bf1a..21f93e79 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1706,7 +1706,6 @@ dependencies = [
  "indoc",
  "insta",
  "itertools 0.12.1",
- "jj-cli",
  "jj-lib",
  "libc",
  "maplit",
diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index fd1085b8..d16a5acf 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -22,28 +22,13 @@ include = [
     "/tests/",
     "!*.pending-snap",
     "!*.snap*",
-    "/tests/cli-reference@.md.snap"
+    "/tests/cli-reference@.md.snap",
 ]
 
 [[bin]]
 name = "jj"
 path = "src/main.rs"
 
-[[bin]]
-name = "fake-editor"
-path = "testing/fake-editor.rs"
-required-features = ["test-fakes"]
-
-[[bin]]
-name = "fake-diff-editor"
-path = "testing/fake-diff-editor.rs"
-required-features = ["test-fakes"]
-
-[[bin]]
-name = "fake-formatter"
-path = "testing/fake-formatter.rs"
-required-features = ["test-fakes"]
-
 [[test]]
 name = "runner"
 
@@ -105,14 +90,11 @@ insta = { workspace = true }
 test-case = { workspace = true }
 testutils = { workspace = true }
 # https://github.com/rust-lang/cargo/issues/2911#issuecomment-1483256987
-jj-cli = { path = ".", features = ["test-fakes"], default-features = false }
 
 [features]
 default = ["watchman"]
 bench = ["dep:criterion"]
 packaging = []
-test-fakes = ["jj-lib/testing"]
-vendored-openssl = ["git2/vendored-openssl", "jj-lib/vendored-openssl"]
 watchman = ["jj-lib/watchman"]
 
 [package.metadata.binstall]
