diff --git a/Cargo.lock b/Cargo.lock
index 6df57da61..466996a69 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1879,7 +1879,6 @@ dependencies = [
  "indoc",
  "insta",
  "itertools 0.13.0",
- "jj-cli",
  "jj-lib",
  "libc",
  "maplit",
diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index a1d0523b9..28d041337 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -29,21 +29,6 @@ include = [
 name = "jj"
 path = "src/main.rs"
 
-[[bin]]
-name = "fake-editor"
-path = "testing/fake-editor.rs"
-required-features = ["test-fakes"]
-
-[[bin]]
-name = "fake-diff-editor"
-path = "testing/fake-diff-editor.rs"
-required-features = ["test-fakes"]
-
-[[bin]]
-name = "fake-formatter"
-path = "testing/fake-formatter.rs"
-required-features = ["test-fakes"]
-
 [[test]]
 name = "runner"
 
@@ -107,14 +92,11 @@ insta = { workspace = true }
 test-case = { workspace = true }
 testutils = { workspace = true }
 # https://github.com/rust-lang/cargo/issues/2911#issuecomment-1483256987
-jj-cli = { path = ".", features = ["test-fakes"], default-features = false }
 
 [features]
 default = ["watchman"]
 bench = ["dep:criterion"]
 packaging = []
-test-fakes = ["jj-lib/testing"]
-vendored-openssl = ["git2/vendored-openssl", "jj-lib/vendored-openssl"]
 watchman = ["jj-lib/watchman"]
 
 [package.metadata.binstall]
