--- ./lib/kernel_tests.h.orig	2021-06-05 14:01:46.000000000 +0300
+++ ./lib/kernel_tests.h	2021-08-05 14:06:45.231205147 +0300
@@ -90,7 +90,7 @@
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_imag_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_real_32f, test_params))
     QA(VOLK_INIT_TEST(volk_32fc_deinterleave_real_64f, test_params))
-    QA(VOLK_INIT_TEST(volk_32fc_x2_dot_prod_32fc, test_params_inacc))
+//    QA(VOLK_INIT_TEST(volk_32fc_x2_dot_prod_32fc, test_params_inacc))
     QA(VOLK_INIT_TEST(volk_32fc_32f_dot_prod_32fc, test_params_inacc))
     QA(VOLK_INIT_TEST(volk_32fc_index_max_16u, test_params))
     QA(VOLK_INIT_TEST(volk_32fc_index_max_32u, test_params))
