diff --git a/PKG-INFO b/PKG-INFO
index da2af3e..132b16b 100644
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -31,7 +31,6 @@ Requires-Python: <3.15,>=3.10
 Description-Content-Type: text/x-rst
 Requires-Dist: passagemath-objects~=10.6.37.0
 Requires-Dist: memory_allocator
-Requires-Dist: cysignals<1.12.4; sys_platform == "win32"
 Requires-Dist: cysignals!=1.12.0,>=1.11.2
 Provides-Extra: test
 
diff --git a/pyproject.toml b/pyproject.toml
index a60bce7..7810941 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,7 +10,7 @@ requires = [
     'passagemath-objects ~= 10.6.37.0',
     'cython >=3.0.8, <3.2.0', 'cython >=3.0.8,<3.2.0',
     'memory_allocator',
-    'cysignals <1.12.4; sys_platform == "win32"', 'cysignals >=1.11.2, != 1.12.0',
+    'cysignals >=1.11.2, != 1.12.0',
 ]
 build-backend = "setuptools.build_meta"

@@ -20,7 +20,7 @@ description = "passagemath: Linear and mixed integer linear optimization backend
 dependencies = [
     'passagemath-objects ~= 10.6.37.0',
     'memory_allocator',
-    'cysignals <1.12.4; sys_platform == "win32"', 'cysignals >=1.11.2, != 1.12.0',
+    'cysignals >=1.11.2, != 1.12.0',
 ]
 dynamic = ["version"]
 license = "GPL-2.0-or-later"
diff --git a/passagemath_glpk.egg-info/requires.txt b/passagemath_glpk.egg-info/requires.txt
index 38f48ce..c252984 100644
--- a/passagemath_glpk.egg-info/requires.txt
+++ b/passagemath_glpk.egg-info/requires.txt
@@ -2,7 +2,4 @@ passagemath-objects~=10.6.37.0
 memory_allocator
 cysignals!=1.12.0,>=1.11.2

-[:sys_platform == "win32"]
-cysignals<1.12.4
-
 [test]
diff --git a/passagemath_glpk.egg-info/PKG-INFO b/passagemath_glpk.egg-info/PKG-INFO
index da2af3e..132b16b 100644
--- a/passagemath_glpk.egg-info/PKG-INFO
+++ b/passagemath_glpk.egg-info/PKG-INFO
@@ -31,7 +31,6 @@ Requires-Python: <3.15,>=3.10
 Description-Content-Type: text/x-rst
 Requires-Dist: passagemath-objects~=10.6.37.0
 Requires-Dist: memory_allocator
-Requires-Dist: cysignals<1.12.4; sys_platform == "win32"
 Requires-Dist: cysignals!=1.12.0,>=1.11.2
 Provides-Extra: test
 
