--- gmp-6.2.1-orig/configure.ac	2020-11-15 02:45:09.000000000 +0800
+++ gmp-6.2.1/configure.ac	2023-01-19 21:14:43.633665000 +0800
@@ -2666,16 +2666,6 @@
 AC_SUBST(LIBGMP_DLL,0)
 case $host in
   *-*-cygwin* | *-*-mingw* | *-*-msys | *-*-pw32* | *-*-os2*)
-    # By default, build only static.
-    if test -z "$enable_shared"; then
-      enable_shared=no
-    fi
-    # Don't allow both static and DLL.
-    if test "$enable_shared" != no && test "$enable_static" != no; then
-      AC_MSG_ERROR([cannot build both static and DLL, since gmp.h is different for each.
-Use "--disable-static --enable-shared" to build just a DLL.])
-    fi
-
     # "-no-undefined" is required when building a DLL, see documentation on
     # AC_LIBTOOL_WIN32_DLL.
     #
