diff --git a/gmp-6.2.1.orig/gmp-h.in b/gmp-6.2.1/gmp-h.in
index 3d449d427..942fab4b1 100644
--- a/gmp-6.2.1.orig/gmp-h.in
+++ b/gmp-6.2.1/gmp-h.in
@@ -101,7 +101,7 @@ see https://www.gnu.org/licenses/.  */
 
 #if defined (__GNUC__)
 #define __GMP_DECLSPEC_EXPORT  __declspec(__dllexport__)
-#define __GMP_DECLSPEC_IMPORT  __declspec(__dllimport__)
+#define __GMP_DECLSPEC_IMPORT
 #endif
 #if defined (_MSC_VER) || defined (__BORLANDC__)
 #define __GMP_DECLSPEC_EXPORT  __declspec(dllexport)
