--- OpenXR-SDK-release-1.0.9/src/loader/CMakeLists.txt.orig	2020-06-18 12:38:49.709702500 +0300
+++ OpenXR-SDK-release-1.0.9/src/loader/CMakeLists.txt	2020-06-18 12:39:37.175363800 +0300
@@ -153,10 +153,6 @@
                 libopenxr_loader.so.${MAJOR}.${MINOR}
     )
 
-    set(XR_API_VERSION "${MAJOR}.${MINOR}")
-    set(EXTRA_LIBS ${CMAKE_THREAD_LIBS_INIT})
-    configure_file(openxr.pc.in openxr.pc @ONLY)
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/openxr.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 elseif(WIN32)
     if(MSVC)
         # WindowsStore (UWP) apps must be compiled with dynamic CRT linkage (default)
@@ -210,6 +206,13 @@
     endif()
 endif()
 
+if(NOT MSVC)
+    set(XR_API_VERSION "${MAJOR}.${MINOR}")
+    set(EXTRA_LIBS ${CMAKE_THREAD_LIBS_INIT})
+    configure_file(openxr.pc.in openxr.pc @ONLY)
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/openxr.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+endif()
+
 # Copy loader to conformance_cli binary folder if built as dynamic
 if(DYNAMIC_LOADER AND BUILD_CONFORMANCE_TESTS)
     add_custom_command(TARGET openxr_loader POST_BUILD
