--- a/src/loader/CMakeLists.txt
+++ b/src/loader/CMakeLists.txt
@@ -263,10 +270,12 @@
                 -ffunction-sections
                 -fdata-sections
     )
+    if(NOT MINGW)
     # Make build depend on the version script/export map
     target_sources(openxr_loader PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/openxr-loader.map)
     # Add the linker flag.
     set_target_properties(openxr_loader PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/openxr-loader.map")
+    endif()
     # For GCC version 7.1 or greater, we need to disable the implicit fallthrough warning since
     # there's no consistent way to satisfy all compilers until they all accept the C++17 standard
     if(CMAKE_COMPILER_IS_GNUCC AND NOT (CMAKE_CXX_COMPILER_VERSION LESS 7.1))
