--- installer-0.4.0/src/installer/__main__.py.orig	2022-02-06 15:44:54.255000200 +0100
+++ installer-0.4.0/src/installer/__main__.py	2022-02-06 15:44:57.255156900 +0100
@@ -90,7 +90,7 @@
     with installer.sources.WheelFile.open(args.wheel) as source:
         destination = installer.destinations.SchemeDictionaryDestination(
             get_scheme_dict(source.distribution, prefix=args.prefix),
-            sys.executable,
+            r'<launcher_dir>\python.exe',
             installer.utils.get_launcher_kind(),
             bytecode_optimization_levels=bytecode_levels,
             destdir=args.destdir,
