diff --git a/src/jrd/version.h b/src/jrd/version.h
index 9f4373c..356137c 100644
--- a/src/jrd/version.h
+++ b/src/jrd/version.h
@@ -143,6 +143,10 @@
 
 #endif
 
+#ifndef _MSC_VER
+#define RC_TARGET_FILENAME ""
+#define RC_TARGET_NAME "firebird"
+#endif
 
 #ifdef NDEBUG
 #define VER_DBG
@@ -151,7 +155,7 @@
 #endif
 
 
-#ifdef RC_ARH_x64
+#if defined(RC_ARH_x64) || defined(__x86_64__)
 #define VER_ARCH "64-bit"
 #else
 #define VER_ARCH "32-bit"
diff --git a/src/jrd/version.rc b/src/jrd/version.rc
index 75ff347..281339c 100644
--- a/src/jrd/version.rc
+++ b/src/jrd/version.rc
@@ -38,8 +38,8 @@ BEGIN
       VALUE "CompanyName", "Firebird Project\0"
       VALUE "FileDescription", "Firebird " VER_FILEDESC " (" VER_ARCH VER_DBG ")"
       VALUE "FileVersion", FILE_VER_STRING
-      VALUE "InternalName", STRINGIZE(RC_TARGET_NAME)
-      VALUE "OriginalFilename", STRINGIZE(RC_TARGET_FILENAME)
+      VALUE "InternalName", RC_TARGET_NAME
+      VALUE "OriginalFilename", RC_TARGET_FILENAME
       VALUE "LegalCopyright", "All Copyright (c) retained by individual contributors - original code Copyright (c) 2000 Inprise Corporation\0"
       VALUE "ProductName", "Firebird SQL Server\0"
       VALUE "ProductVersion", PRODUCT_VER_STRING
