diff -bur firebird-5.0.4-orig/src/common/ThreadStart.h firebird-5.0.4/src/common/ThreadStart.h
--- firebird-5.0.4-orig/src/common/ThreadStart.h
+++ firebird-5.0.4/src/common/ThreadStart.h
@@ -76,7 +76,7 @@ class Thread
 public:
 #ifdef WIN_NT
 	typedef HANDLE Handle;
-	constexpr static Handle INVALID_HANDLE = INVALID_HANDLE_VALUE;
+	inline static const Handle INVALID_HANDLE = INVALID_HANDLE_VALUE;
 #endif
 #ifdef USE_POSIX_THREADS
 	typedef pthread_t Handle;
