--- a/include/libimobiledevice-glue/socket.h
+++ b/include/libimobiledevice-glue/socket.h
@@ -25,6 +25,16 @@
 #include <stdlib.h>
 #include <stdint.h>
 
+#ifdef WIN32
+#define LIBIMOBILEDEVICE_GLUE_API __declspec( dllexport )
+#else
+#ifdef HAVE_FVISIBILITY
+#define LIBIMOBILEDEVICE_GLUE_API __attribute__((visibility("default")))
+#else
+#define LIBIMOBILEDEVICE_GLUE_API
+#endif
+#endif
+
 enum fd_mode {
 	FDM_READ,
 	FDM_WRITE,
@@ -52,11 +62,11 @@
 int socket_accept(int fd, uint16_t port);
 
 int socket_shutdown(int fd, int how);
-int socket_close(int fd);
+LIBIMOBILEDEVICE_GLUE_API int socket_close(int fd);
 
 int socket_receive(int fd, void *data, size_t length);
 int socket_peek(int fd, void *data, size_t length);
-int socket_receive_timeout(int fd, void *data, size_t length, int flags, unsigned int timeout);
+LIBIMOBILEDEVICE_GLUE_API int socket_receive_timeout(int fd, void *data, size_t length, int flags, unsigned int timeout);
 int socket_send(int fd, void *data, size_t length);
 
 int socket_get_socket_port(int fd, uint16_t *port);
--- a/src/socket.c
+++ b/src/socket.c
@@ -1185,7 +1185,7 @@
 	return shutdown(fd, how);
 }
 
-LIBIMOBILEDEVICE_GLUE_API int socket_close(int fd) {
+int socket_close(int fd) {
 #ifdef WIN32
 	return closesocket(fd);
 #else
@@ -1203,7 +1203,7 @@
 	return socket_receive_timeout(fd, data, length, MSG_PEEK, RECV_TIMEOUT);
 }
 
-LIBIMOBILEDEVICE_GLUE_API int socket_receive_timeout(int fd, void *data, size_t length, int flags, unsigned int timeout)
+int socket_receive_timeout(int fd, void *data, size_t length, int flags, unsigned int timeout)
 {
 	int res;
 	int result;
