From 822119d474927a17c776e7ea2956b2ebf7a96947 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 22 Nov 2014 22:15:35 +0000
Subject: [PATCH 3/7] MinGW-w64: don't pass --static to pkg-config if SYS =
 mingw32

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index d136fffa22..651c3e20d3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -868,7 +868,7 @@ PKG_PROG_PKG_CONFIG()
 
 dnl On some OS we need static linking
 AS_IF([test -n "${PKG_CONFIG}" ],[
-    AS_IF([test "${SYS}" = "mingw32" -o "${SYS}" = "darwin" -o "${SYS}" = "os2" ],[
+    AS_IF([test "${SYS}" = "darwin" -o "${SYS}" = "os2" ],[
         PKG_CONFIG="${PKG_CONFIG} --static"
     ])
 ])
-- 
2.15.0

