From e4b28e32b706666286d861dbc0a93f796ad968e0 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Tue, 10 Mar 2015 19:36:25 +0000
Subject: [PATCH] remove AllocConsole reference for WINSTORE


diff --git a/modules/control/intromsg.h b/modules/control/intromsg.h
index 7af9520..a946a0a 100644
--- a/modules/control/intromsg.h
+++ b/modules/control/intromsg.h
@@ -23,7 +23,9 @@ static inline void intf_consoleIntroMsg(intf_thread_t *intf)
 {
     if (getenv( "PWD" ) == NULL) /* detect Cygwin shell or Wine */
     {
+#if !VLC_WINSTORE_APP
         AllocConsole();
+#endif
         freopen("CONOUT$", "w", stdout);
         freopen("CONOUT$", "w", stderr);
         freopen("CONIN$", "r", stdin);
-- 
2.3.1

