--- vlc-3.0.4/configure.ac.orig	2018-11-19 11:31:00.675642000 +0300
+++ vlc-3.0.4/configure.ac	2018-11-19 11:31:46.071721700 +0300
@@ -1148,14 +1148,6 @@
     ])
     AS_IF([test "${ac_cv_c_omit_frame_pointer}" = "no"], [VLC_RESTORE_FLAGS])
   ])
-
-  AX_APPEND_COMPILE_FLAGS([-fstack-protector-strong])
-  AX_APPEND_COMPILE_FLAGS([-fstack-protector-strong], [CXXFLAGS])
-  dnl Win32 requires linking to ssp for stack-protection
-  AS_IF([test "${SYS}" = "mingw32"], [
-      LDFLAGS="${LDFLAGS} -lssp"
-      AS_IF([test "${vlc_winstore_app}" != 1], [LDFLAGS="${LDFLAGS} -ladvapi32"])
-  ])
 ])
 
 VLC_SAVE_FLAGS
