diff -Naur libogg-1.3.4.orig/CMakeLists.txt libogg-1.3.4/CMakeLists.txt
--- libogg-1.3.4.orig/CMakeLists.txt	2019-09-02 15:05:06.093041600 -0400
+++ libogg-1.3.4/CMakeLists.txt	2019-09-02 15:08:44.811923200 -0400
@@ -103,6 +103,14 @@
     VERSION ${LIB_VERSION}
     PUBLIC_HEADER "${OGG_HEADERS}"
 )
+if(WIN32)
+    set_target_properties(ogg
+        PROPERTIES
+        OUTPUT_NAME ogg
+        RUNTIME_OUTPUT_NAME ogg-${LIB_SOVERSION}
+        ARCHIVE_OUTPUT_NAME ogg
+    )
+endif()
 
 if(BUILD_FRAMEWORK)
     set_target_properties(ogg PROPERTIES
diff -Naur libogg-1.3.4.orig/win32/ogg.def libogg-1.3.4/win32/ogg.def
--- libogg-1.3.4.orig/win32/ogg.def	2019-09-02 15:05:06.093041600 -0400
+++ libogg-1.3.4/win32/ogg.def	2019-09-02 15:08:44.811923200 -0400
@@ -2,10 +2,11 @@
 ; ogg.def
 ; List of exported functions for Windows builds.
 ;
-LIBRARY ogg
+LIBRARY "libogg-0"
 EXPORTS
 ;
 oggpack_writeinit
+oggpack_writecheck
 oggpack_writetrunc
 oggpack_writealign
 oggpack_writecopy
@@ -24,6 +25,7 @@
 oggpack_get_buffer
 ;
 oggpackB_writeinit
+oggpackB_writecheck
 oggpackB_writetrunc
 oggpackB_writealign
 oggpackB_writecopy
@@ -49,6 +51,7 @@
 ogg_sync_clear
 ogg_sync_reset
 ogg_sync_destroy
+ogg_sync_check
 ogg_sync_buffer
 ogg_sync_wrote
 ogg_sync_pageseek
@@ -62,6 +65,7 @@
 ogg_stream_reset
 ogg_stream_reset_serialno
 ogg_stream_destroy
+ogg_stream_check
 ogg_stream_eos
 ogg_stream_pageout_fill
 ogg_stream_flush_fill
