--- ffmpeg-python-0.2.0/setup.py.orig	2019-07-06 02:17:30.000000000 +0200
+++ ffmpeg-python-0.2.0/setup.py	2024-10-19 12:35:44.851224900 +0200
@@ -60,7 +60,6 @@
 setup(
     name='ffmpeg-python',
     packages=['ffmpeg'],
-    setup_requires=['pytest-runner'],
     tests_require=['pytest', 'pytest-mock'],
     version=version,
     description='Python bindings for FFmpeg - with complex filtering support',
