diff -urN metis-5.1.0.orig/GKlib/gk_arch.h metis-5.1.0/GKlib/gk_arch.h
--- metis-5.1.0.orig/GKlib/gk_arch.h	2017-04-30 20:26:49.002753400 +0100
+++ metis-5.1.0/GKlib/gk_arch.h	2017-04-30 20:27:34.087026100 +0100
@@ -41,7 +41,9 @@
 #endif
   #include <inttypes.h>
   #include <sys/types.h>
-  #include <sys/resource.h>
+  #ifndef __MINGW32__
+    #include <sys/resource.h>
+  #endif
   #include <sys/time.h>
 #endif
 
