diff -bur oneDNN-3.3.2-orig/src/cpu/aarch64/cpu_isa_traits.hpp oneDNN-3.3.2/src/cpu/aarch64/cpu_isa_traits.hpp
--- oneDNN-3.3.2-orig/src/cpu/aarch64/cpu_isa_traits.hpp	2023-12-03 02:20:17.434762400 -0700
+++ oneDNN-3.3.2/src/cpu/aarch64/cpu_isa_traits.hpp	2023-12-03 02:24:46.795841200 -0700
@@ -28,7 +28,7 @@
 
 /* in order to make selinux happy memory that would be marked with X-bit should
  * be obtained with mmap */
-#if !defined(_WIN32)
+#if !defined(_WIN32) && !defined(__MINGW64__)
 #define XBYAK_USE_MMAP_ALLOCATOR
 #endif
 
diff -bur oneDNN-3.3.2-orig/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h oneDNN-3.3.2/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h
--- oneDNN-3.3.2-orig/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h	2023-12-03 02:20:17.514361300 -0700
+++ oneDNN-3.3.2/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h	2023-12-03 02:26:03.196606300 -0700
@@ -37,7 +37,7 @@
 #include <unordered_set>
 #include <vector>
 
-#if defined(__GNUC__) || defined(__APPLE__)
+#if (defined(__GNUC__) || defined(__APPLE__)) && !defined(__MINGW64__)
 #ifndef XBYAK_USE_MMAP_ALLOCATOR
 #define XBYAK_USE_MMAP_ALLOCATOR
 #endif
@@ -46,7 +46,7 @@
 #include <cmath>
 #include <functional>
 
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__MINGW64__)
 #include <cassert>
 #include <stdlib.h>
 #include <sys/mman.h>
diff -bur oneDNN-3.3.2-orig/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_code_array.h oneDNN-3.3.2/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_code_array.h
--- oneDNN-3.3.2-orig/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_code_array.h	2023-12-03 02:20:17.517356400 -0700
+++ oneDNN-3.3.2/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_code_array.h	2023-12-03 02:29:02.393195200 -0700
@@ -20,7 +20,7 @@
 static const size_t CSIZE = sizeof(uint32_t);
 
 inline void *AlignedMalloc(size_t size, size_t alignment) {
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW64__)
   return _aligned_malloc(size, alignment);
 #else
   void *p;
@@ -30,7 +30,7 @@
 }
 
 inline void AlignedFree(void *p) {
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW64__)
   _aligned_free(p);
 #else
   free(p);
@@ -277,7 +277,7 @@
     default:
       return false;
     }
-#if defined(__GNUC__) || defined(__APPLE__)
+#if (defined(__GNUC__) || defined(__APPLE__)) && !defined(__MINGW64__)
     size_t pageSize = inner::getPageSize();
     size_t iaddr = reinterpret_cast<size_t>(addr);
     size_t roundAddr = iaddr & ~(pageSize - static_cast<size_t>(1));
diff -bur oneDNN-3.3.2-orig/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_inner.h oneDNN-3.3.2/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_inner.h
--- oneDNN-3.3.2-orig/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_inner.h	2023-12-03 02:20:17.519359500 -0700
+++ oneDNN-3.3.2/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_inner.h	2023-12-03 02:23:15.371623000 -0700
@@ -20,7 +20,7 @@
 namespace inner {
 
 inline size_t getPageSize() {
-#ifdef __GNUC__
+#if defined (__GNUC__) && !defined(__MINGW64__)
   static const size_t pageSize = sysconf(_SC_PAGESIZE);
 #else
   static const size_t pageSize = 4096;
