--- a/setup.py
+++ b/setup.py
@@ -96,6 +96,8 @@
         os.path.join(connector, "lib"),
     ]
     libraries = [
+        client,
+        "z",
         "kernel32",
         "advapi32",
         "wsock32",
@@ -104,14 +106,13 @@
         "crypt32",
         "secur32",
         "bcrypt",
-        client,
     ]
     include_dirs = [
         os.path.join(connector, "include", "mariadb"),
         os.path.join(connector, "include"),
     ]
 
-    extra_link_args = ["/MANIFEST"]
+    extra_link_args = []
 
     define_macros = [
         ("version_info", release_info["version_info"]),
