--- a/CMakeLists.txt.orig	2026-05-10 05:02:09.000000000 +0800
+++ b/CMakeLists.txt	2026-05-11 14:35:32.977238700 +0800
@@ -217,7 +217,7 @@
 
 set(CMAKE_CXX_STANDARD 17) # default C++ version for new target afterward
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-  add_definitions(
+  add_compile_options(
     -Wall
   )
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pthread")
@@ -225,7 +225,7 @@
     add_definitions(-O0 -g3)
   endif ()
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-  add_definitions(
+  add_compile_options(
     -Wall
   )
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pthread")
