diff -ru xvidcore/build/generic/configure patched/build/generic/configure
--- xvidcore/build/generic/configure	2011-05-23 12:22:01.000000000 +0200
+++ patched/build/generic/configure	2013-10-12 09:40:38.079753682 +0200
@@ -4341,8 +4341,8 @@
 $as_echo "ok" >&6; }
 	STATIC_LIB="xvidcore.\$(STATIC_EXTENSION)"
 	SHARED_LIB="xvidcore.\$(SHARED_EXTENSION)"
-	SPECIFIC_LDFLAGS="-mno-cygwin -shared -Wl,--dll,--out-implib,\$@.a libxvidcore.def"
-	SPECIFIC_CFLAGS="-mno-cygwin"
+	SPECIFIC_LDFLAGS="-shared -Wl,--dll,--out-implib,\$@.a libxvidcore.def"
+	SPECIFIC_CFLAGS=""
 	;;
      darwin*|raphsody*)
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
@@ -4577,7 +4577,12 @@
 $as_echo "win32" >&6; }
                    NASM_FORMAT="win32"
                fi
+#FIXME MINGW64
+       if test "$ARCHITECTURE" = "X86_64" ; then
+	       PREFIX=""
+	   else
 	       PREFIX="-DWINDOWS"
+	   fi
 	       MARK_FUNCS=""
                ;;
            *darwin*)
@@ -4599,9 +4604,10 @@
        ASSEMBLY_EXTENSION="asm"
        AFLAGS="-I\$(<D)/ -I../../src/ -f $NASM_FORMAT $PREFIX $MARK_FUNCS"
        ASSEMBLY_SOURCES="SRC_${ARCHITECTURE}"
-       if test "$ARCHITECTURE" = "X86_64" ; then
-               AFLAGS=${AFLAGS}" -DARCH_IS_X86_64"
-       fi
+#FIXME MINGW64
+#       if test "$ARCHITECTURE" = "X86_64" ; then
+#               AFLAGS=${AFLAGS}" -DARCH_IS_X86_64"
+#       fi
    else
        { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: no correct assembler was found - Compiling generic sources only" >&5
 $as_echo "$as_me: WARNING: no correct assembler was found - Compiling generic sources only" >&2;}
@@ -4802,7 +4808,7 @@
 fi
 
 
-if test "$GCC" = "yes" ; then
+if test " " = "yes" ; then
 cat << EOF > test.c
 #include <stdio.h>
 int main(int argc, char **argv)
diff -ru xvidcore/build/generic/configure.in patched/build/generic/configure.in
--- xvidcore/build/generic/configure.in	2011-05-18 09:59:28.000000000 +0200
+++ patched/build/generic/configure.in	2013-10-12 09:38:34.711413395 +0200
@@ -438,7 +438,12 @@
                    AC_MSG_RESULT([win32])
                    NASM_FORMAT="win32"
                fi
+#FIXME MINGW64
+       if test "$ARCHITECTURE" = "X86_64" ; then
+	       PREFIX=""
+	   else
 	       PREFIX="-DWINDOWS"
+	   fi
 	       MARK_FUNCS=""
                ;;
            *darwin*)
@@ -458,9 +463,10 @@
        ASSEMBLY_EXTENSION="asm"
        AFLAGS="-I\$(<D)/ -I../../src/ -f $NASM_FORMAT $PREFIX $MARK_FUNCS"
        ASSEMBLY_SOURCES="SRC_${ARCHITECTURE}"
-       if test "$ARCHITECTURE" = "X86_64" ; then
-               AFLAGS=${AFLAGS}" -DARCH_IS_X86_64"
-       fi
+#FIXME MINGW64
+#       if test "$ARCHITECTURE" = "X86_64" ; then
+#               AFLAGS=${AFLAGS}" -DARCH_IS_X86_64"
+#       fi
    else 
        AC_MSG_WARN([no correct assembler was found - Compiling generic sources only])
        ARCHITECTURE="GENERIC"
