--- gobject-introspection-1.60.1/tools/g-ir-tool-template.in.orig	2019-01-06 11:32:10.000000000 +0100
+++ gobject-introspection-1.60.1/tools/g-ir-tool-template.in	2019-04-20 20:37:12.446804000 +0200
@@ -49,18 +49,12 @@
 datadir = os.path.abspath(os.path.join(filedir, '..', 'share'))
 # Fallback to hard-coded paths if the relocatable paths are wrong
 if not os.path.isdir(os.path.join(datadir, 'gir-1.0')):
-    datadir = "@datarootdir@"
+    datadir = os.path.join(sys.prefix, "share")
 
 builtins.__dict__['DATADIR'] = datadir
 
-# Respect gir_dir_prefix for meson and autotools
-girdir = ''
-# for meson
-if '@gir_dir_prefix@' and not '@gir_dir_prefix@'.startswith('@'):
-    girdir = os.path.abspath(os.path.join(filedir, '..', '@gir_dir_prefix@'))
-# for autotools
-elif '@GIR_DIR@' and not '@GIR_DIR@'.startswith('@'):
-    girdir = os.path.dirname(os.path.abspath('@GIR_DIR@'))
+girdir = os.path.join(sys.prefix, "share")
+
 builtins.__dict__['GIRDIR'] = girdir
 
 # Again, relative paths first so that the installation prefix is relocatable
@@ -91,7 +85,7 @@
     else:
         # Okay, we're not running uninstalled and the prefix is not
         # relocatable. Use hard-coded libdir.
-        pylibdir = os.path.join('@libdir@', 'gobject-introspection')
+        pylibdir = os.path.join(sys.prefix, 'lib', 'gobject-introspection')
 
 sys.path.insert(0, pylibdir)
 
