--- dieterv-gobject-introspection-c9e2d88/giscanner/dumper.py.orig	2013-05-08 13:07:54 +0400
+++ dieterv-gobject-introspection-c9e2d88/giscanner/dumper.py	2013-05-08 13:08:00 +0400
@@ -103,6 +103,7 @@
         # We have to use the current directory to work around Unix
         # sysadmins who mount /tmp noexec
         tmpdir = tempfile.mkdtemp('', 'tmp-introspect', dir=os.getcwd())
+        os.mkdir (os.path.join (tmpdir, '.libs'))
 
         tpl_args = {}
         if self._uninst_srcdir is not None:
