--- dieterv-gobject-introspection-c9e2d88/giscanner/dumper.py.orig	2013-05-08 13:08:26 +0400
+++ dieterv-gobject-introspection-c9e2d88/giscanner/dumper.py	2013-05-08 13:11:35 +0400
@@ -260,7 +260,7 @@
             args.extend(['-o', output])
         if libtool:
             if os.name == 'nt':
-                args.append('-export-all-symbols')
+                args.append('-Wl,--export-all-symbols')
             else:
                 args.append('-export-dynamic')
 
