--- gobject-introspection-1.39.3/giscanner/gdumpparser.py.orig	2014-01-11 16:55:55.000000000 +0000
+++ gobject-introspection-1.39.3/giscanner/gdumpparser.py	2014-02-16 06:37:12.574424000 +0000
@@ -23,6 +23,7 @@
 import tempfile
 import shutil
 import subprocess
+import time
 from xml.etree.cElementTree import parse
 
 from . import ast
@@ -171,7 +172,14 @@
             return parse(out_path)
         finally:
             if not utils.have_debug_flag('save-temps'):
-                shutil.rmtree(self._binary.tmpdir)
+                def rmtree_onerror (func, path, exinfo):
+                    time.sleep (1)
+                    try:
+                      func (path)
+                    except:
+                      raise
+
+                shutil.rmtree(self._binary.tmpdir, False, rmtree_onerror)
 
     # Parser
 
