--- gobject-introspection-1.39.3/tests/offsets/Makefile.am.orig	2014-01-23 18:49:50.000000000 +0000
+++ gobject-introspection-1.39.3/tests/offsets/Makefile.am	2014-02-21 08:19:55.426765500 +0000
@@ -18,7 +18,7 @@
 	offsets.c
 liboffsets_la_CPPFLAGS = $(GIREPO_CFLAGS)
 # dummy rpath to get built dynamically (huh?)
-liboffsets_la_LDFLAGS = -avoid-version -rpath $(libdir)
+liboffsets_la_LDFLAGS = -avoid-version -rpath $(libdir) -no-undefined
 
 Offsets-1.0.gir: liboffsets.la offsets.h
 Offsets_1_0_gir_INCLUDES = GObject-2.0
