--- gobject-introspection-git-39fc1013ba84d885272123e954b2aea27d5117cd/giscanner/dumper.py.orig	2012-05-13 01:54:23 +0400
+++ gobject-introspection-git-39fc1013ba84d885272123e954b2aea27d5117cd/giscanner/dumper.py	2012-05-13 02:34:37 +0400
@@ -111,6 +111,10 @@
         tpl_args['init_sections'] = "\n".join(self._options.init_sections)
 
         c_path = self._generate_tempfile(tmpdir, '.c')
+
+        if os.name == 'nt':
+            c_path = c_path.replace('\\','/')
+
         f = open(c_path, 'w')
         f.write(_PROGRAM_TEMPLATE % tpl_args)
 
@@ -145,11 +150,16 @@
 
         if os.name == 'nt':
             ext = '.exe'
+            o_path = o_path.replace('\\','/')
         else:
             ext = ''
 
         bin_path = self._generate_tempfile(tmpdir, ext)
 
+        if os.name == 'nt':
+            bin_path = bin_path.replace('\\','/')
+            os.mkdir (os.path.join (tmpdir, '.libs'))
+
         try:
             self._compile(o_path, c_path)
         except CompilerError, e:
