--- gobject-introspection-1.39.0/giscanner/giscannermodule.c.orig	2014-01-29 11:52:46.121400000 +0400
+++ gobject-introspection-1.39.0/giscanner/giscannermodule.c	2014-01-29 11:53:59.020200000 +0400
@@ -439,7 +439,7 @@
 #endif
 #endif
 
-#ifndef MSVC_USE_FD_DIRECTLY
+#if !defined(MSVC_USE_FD_DIRECTLY) && !defined(__MINGW64_VERSION_MAJOR)
   {
 #if defined(PY_MAJOR_VERSION) && PY_MAJOR_VERSION==2 && PY_MINOR_VERSION==5
 #define PYTHON_MSVCRXX_DLL "msvcr71.dll"
