diff --git a/giscanner/ccompiler.py b/giscanner/ccompiler.py
index 63110a2..1c66aac 100644
--- a/giscanner/ccompiler.py
+++ b/giscanner/ccompiler.py
@@ -135,9 +135,9 @@ class CCompiler(object):
         # When we are not using Visual C++ (i.e. we are using GCC)...
         else:
             is_msvc = False
-            libtool = get_libtool_command(options)
+            libtool = utils.get_libtool_command(options)
             if libtool:
-                args.append(which(os.environ.get('SHELL', 'sh.exe')))
+                args.append(utils.which(os.environ.get('SHELL', 'sh.exe')))
                 args.extend(libtool)
                 args.append('--mode=execute')
             # FIXME: it could have prefix (i686-w64-mingw32-dlltool.exe)
