diff -urN tfel-TFEL-3.3.0.orig/mfront/src/BehaviourMaterialProperty.cxx tfel-TFEL-3.3.0/mfront/src/BehaviourMaterialProperty.cxx
--- tfel-TFEL-3.3.0.orig/mfront/src/BehaviourMaterialProperty.cxx	2020-10-26 09:54:54.493735900 +0100
+++ tfel-TFEL-3.3.0/mfront/src/BehaviourMaterialProperty.cxx	2020-10-26 09:58:47.367055500 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include "TFEL/Raise.hxx"
 #include "MFront/MFrontLogStream.hxx"
 #include "MFront/BehaviourDescription.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mfront/src/BroydenSolvers.cxx tfel-TFEL-3.3.0/mfront/src/BroydenSolvers.cxx
--- tfel-TFEL-3.3.0.orig/mfront/src/BroydenSolvers.cxx	2020-10-26 09:54:54.564739900 +0100
+++ tfel-TFEL-3.3.0/mfront/src/BroydenSolvers.cxx	2020-10-26 09:59:01.162844500 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include "MFront/MFrontDebugMode.hxx"
 #include "MFront/BehaviourDescription.hxx"
 #include "MFront/PerformanceProfiling.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mfront/src/LevenbergMarquardtSolvers.cxx tfel-TFEL-3.3.0/mfront/src/LevenbergMarquardtSolvers.cxx
--- tfel-TFEL-3.3.0.orig/mfront/src/LevenbergMarquardtSolvers.cxx	2020-10-26 09:54:58.212948600 +0100
+++ tfel-TFEL-3.3.0/mfront/src/LevenbergMarquardtSolvers.cxx	2020-10-26 09:59:13.808567800 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions. 
  */
 
+#include <ostream>
 #include "MFront/MFrontDebugMode.hxx"
 #include "MFront/BehaviourDescription.hxx"
 #include "MFront/PerformanceProfiling.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mfront/src/MohrCoulombStressCriterion.cxx tfel-TFEL-3.3.0/mfront/src/MohrCoulombStressCriterion.cxx
--- tfel-TFEL-3.3.0.orig/mfront/src/MohrCoulombStressCriterion.cxx	2020-10-26 09:54:58.864985900 +0100
+++ tfel-TFEL-3.3.0/mfront/src/MohrCoulombStressCriterion.cxx	2020-10-26 11:24:59.559888000 +0100
@@ -159,7 +159,7 @@
       if (r == FLOWCRITERION) {
 #if __cplusplus >= 201703L
         code += "const auto [seqf" + id + ",dseq" + id + "_ds" + id + "] = ";
-        code += "computeMohrCoulombStressCriterionNormal(this->" + cn + ", s" +
+        code += "computeMohrCoulombStressCriterionNormal(this->" + pn + ", s" +
                 id + ");\n";
 #else  /* __cplusplus >= 201703L */
         code += "stress seqf" + id + ";\n";
diff -urN tfel-TFEL-3.3.0.orig/mfront/src/NewtonRaphsonSolvers.cxx tfel-TFEL-3.3.0/mfront/src/NewtonRaphsonSolvers.cxx
--- tfel-TFEL-3.3.0.orig/mfront/src/NewtonRaphsonSolvers.cxx	2020-10-26 09:54:58.916988900 +0100
+++ tfel-TFEL-3.3.0/mfront/src/NewtonRaphsonSolvers.cxx	2020-10-26 09:59:24.325169300 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include "TFEL/Raise.hxx"
 #include "MFront/MFrontDebugMode.hxx"
 #include "MFront/BehaviourDescription.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mfront/src/SecondBroydenSolver.cxx tfel-TFEL-3.3.0/mfront/src/SecondBroydenSolver.cxx
--- tfel-TFEL-3.3.0.orig/mfront/src/SecondBroydenSolver.cxx	2020-10-26 09:54:59.366014500 +0100
+++ tfel-TFEL-3.3.0/mfront/src/SecondBroydenSolver.cxx	2020-10-26 09:59:33.123672600 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions. 
  */
 
+#include <ostream>
 #include "MFront/MFrontDebugMode.hxx"
 #include "MFront/BehaviourDescription.hxx"
 #include "MFront/PerformanceProfiling.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/AsterCohesiveZoneModel.cxx tfel-TFEL-3.3.0/mtest/src/AsterCohesiveZoneModel.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/AsterCohesiveZoneModel.cxx	2020-10-26 09:56:31.357276100 +0100
+++ tfel-TFEL-3.3.0/mtest/src/AsterCohesiveZoneModel.cxx	2020-10-26 10:00:03.188392200 +0100
@@ -12,8 +12,8 @@
  */
 
 #include <cmath>
+#include <ostream>
 #include <algorithm>
-
 #include "TFEL/Raise.hxx"
 #include "TFEL/Math/tmatrix.hxx"
 #include "TFEL/System/ExternalLibraryManager.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/AsterSmallStrainBehaviour.cxx tfel-TFEL-3.3.0/mtest/src/AsterSmallStrainBehaviour.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/AsterSmallStrainBehaviour.cxx	2020-10-26 09:56:31.437280700 +0100
+++ tfel-TFEL-3.3.0/mtest/src/AsterSmallStrainBehaviour.cxx	2020-10-26 10:00:17.803228100 +0100
@@ -12,8 +12,8 @@
  */
 
 #include<cmath>
+#include <ostream>
 #include<algorithm>
-
 #include"TFEL/Raise.hxx"
 #include"TFEL/Math/tmatrix.hxx"
 #include"TFEL/Math/stensor.hxx"
@@ -21,7 +21,6 @@
 #include"MFront/MFrontLogStream.hxx"
 #include"MFront/Aster/Aster.hxx"
 #include"MFront/Aster/AsterComputeStiffnessTensor.hxx"
-
 #include"MTest/CurrentState.hxx"
 #include"MTest/BehaviourWorkSpace.hxx"
 #include"MTest/UmatNormaliseTangentOperator.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/Behaviour.cxx tfel-TFEL-3.3.0/mtest/src/Behaviour.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/Behaviour.cxx	2020-10-26 09:56:31.484283400 +0100
+++ tfel-TFEL-3.3.0/mtest/src/Behaviour.cxx	2020-10-26 10:00:28.896862600 +0100
@@ -11,8 +11,8 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include <stdexcept>
-
 #include "TFEL/Raise.hxx"
 #include "TFEL/System/ExternalLibraryManager.hxx"
 
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/CastemStandardBehaviour.cxx tfel-TFEL-3.3.0/mtest/src/CastemStandardBehaviour.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/CastemStandardBehaviour.cxx	2020-10-26 09:56:33.795415600 +0100
+++ tfel-TFEL-3.3.0/mtest/src/CastemStandardBehaviour.cxx	2020-10-26 10:00:48.053958400 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include "TFEL/Raise.hxx"
 #include "TFEL/System/ExternalLibraryManager.hxx"
 #include "MFront/Castem/Castem.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/CastemUmatSmallStrainBehaviour.cxx tfel-TFEL-3.3.0/mtest/src/CastemUmatSmallStrainBehaviour.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/CastemUmatSmallStrainBehaviour.cxx	2020-10-26 09:56:33.860419300 +0100
+++ tfel-TFEL-3.3.0/mtest/src/CastemUmatSmallStrainBehaviour.cxx	2020-10-26 10:01:01.271714400 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include <algorithm>
 #include "TFEL/Raise.hxx"
 #include "MFront/MFrontLogStream.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/MistralBehaviour.cxx tfel-TFEL-3.3.0/mtest/src/MistralBehaviour.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/MistralBehaviour.cxx	2020-10-26 09:56:34.598461500 +0100
+++ tfel-TFEL-3.3.0/mtest/src/MistralBehaviour.cxx	2020-10-26 10:01:11.801316600 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include "TFEL/Raise.hxx"
 #include "TFEL/Utilities/CxxTokenizer.hxx"
 #include "MFront/MFrontLogStream.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/NonLinearConstraint.cxx tfel-TFEL-3.3.0/mtest/src/NonLinearConstraint.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/NonLinearConstraint.cxx	2020-10-26 09:56:34.628463200 +0100
+++ tfel-TFEL-3.3.0/mtest/src/NonLinearConstraint.cxx	2020-10-26 10:01:21.017843800 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions.
  */
 
+#include <ostream>
 #include <cmath>
 #include <string>
 #include "TFEL/Raise.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/PipeCubicElement.cxx tfel-TFEL-3.3.0/mtest/src/PipeCubicElement.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/PipeCubicElement.cxx	2020-10-26 09:56:34.655464800 +0100
+++ tfel-TFEL-3.3.0/mtest/src/PipeCubicElement.cxx	2020-10-26 10:01:33.386551200 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions. 
  */
 
+#include <ostream>
 #include"TFEL/Material/ModellingHypothesis.hxx"
 #include"MFront/MFrontLogStream.hxx"
 #include"MTest/RoundingMode.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/PipeLinearElement.cxx tfel-TFEL-3.3.0/mtest/src/PipeLinearElement.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/PipeLinearElement.cxx	2020-10-26 09:56:34.680466200 +0100
+++ tfel-TFEL-3.3.0/mtest/src/PipeLinearElement.cxx	2020-10-26 10:01:41.170996500 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions. 
  */
 
+#include <ostream>
 #include"TFEL/Material/ModellingHypothesis.hxx"
 #include"MFront/MFrontLogStream.hxx"
 #include"MTest/RoundingMode.hxx"
diff -urN tfel-TFEL-3.3.0.orig/mtest/src/PipeQuadraticElement.cxx tfel-TFEL-3.3.0/mtest/src/PipeQuadraticElement.cxx
--- tfel-TFEL-3.3.0.orig/mtest/src/PipeQuadraticElement.cxx	2020-10-26 09:56:34.787472300 +0100
+++ tfel-TFEL-3.3.0/mtest/src/PipeQuadraticElement.cxx	2020-10-26 10:01:49.781489000 +0100
@@ -11,6 +11,7 @@
  * project under specific licensing conditions. 
  */
 
+#include <ostream>
 #include"TFEL/Material/ModellingHypothesis.hxx"
 #include"MFront/MFrontLogStream.hxx"
 #include"MTest/RoundingMode.hxx"
