--- libmatroska-1.5.1.orig/CMakeLists.txt	2019-05-26 12:33:37.374638000 -0400
+++ libmatroska-1.5.1/CMakeLists.txt	2019-05-26 12:35:29.399371900 -0400
@@ -72,6 +72,11 @@
 endif()
 if(BUILD_SHARED_LIBS)
   target_compile_definitions(matroska PUBLIC MATROSKA_DLL)
+  if(MINGW)
+    set_target_properties(matroska
+      PROPERTIES
+      RUNTIME_OUTPUT_NAME matroska-6)
+  endif()
   set_target_properties(matroska PROPERTIES
     DEFINE_SYMBOL "MATROSKA_DLL_EXPORT"
     CXX_VISIBILITY_PRESET hidden
