--- libmatroska-1.4.9/matroska/KaxBlock.h.orig	2018-05-17 11:04:31.350209800 +0300
+++ libmatroska-1.4.9/matroska/KaxBlock.h	2018-05-17 11:04:34.017814500 +0300
@@ -205,7 +205,7 @@
     const KaxTrackEntry * ParentTrack;
 };
 
-class KaxInternalBlock : public EbmlBinary {
+class MATROSKA_DLL_API KaxInternalBlock : public EbmlBinary {
   public:
     KaxInternalBlock(EBML_DEF_CONS EBML_DEF_SEP bool bSimple EBML_DEF_SEP EBML_EXTRA_PARAM) :EBML_DEF_BINARY_INIT EBML_DEF_SEP bLocalTimecodeUsed(false), mLacing(LACING_AUTO), mInvisible(false)
       ,ParentCluster(NULL), bIsSimple(bSimple), bIsKeyframe(true), bIsDiscardable(false)
