--- nghttp2-1.50.0/lib/CMakeLists.txt.orig	2022-10-31 19:10:27 +0000
+++ nghttp2-1.50.0/lib/CMakeLists.txt	2022-10-31 19:08:45 +0000
@@ -44,6 +44,7 @@
   set_target_properties(nghttp2 PROPERTIES
     COMPILE_FLAGS "${WARNCFLAGS}"
     VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
+    RUNTIME_OUTPUT_NAME "nghttp2-${LT_SOVERSION}"
     C_VISIBILITY_PRESET hidden
   )
   target_include_directories(nghttp2 INTERFACE
