--- DevIL/DevIL/src-IL/src/il_jp2.cpp	2021-04-07 17:54:20.123823600 +0200
+++ DevIL/DevIL/src-IL/src/il_jp2.cpp.new	2021-04-07 17:52:09.751854600 +0200
@@ -314,13 +314,13 @@
 
 
 
-static int iJp2_file_read(jas_stream_obj_t *obj, char *buf, int cnt)
+static int iJp2_file_read(jas_stream_obj_t *obj, char *buf, unsigned  int cnt)
 {
 	obj;
 	return iread(buf, 1, cnt);
 }
 
-static int iJp2_file_write(jas_stream_obj_t *obj, char *buf, int cnt)
+static int iJp2_file_write(jas_stream_obj_t *obj, const char *buf, unsigned int cnt)
 {
 	obj;
 	return iwrite(buf, 1, cnt);
