--- dia/lib/dia_dirs.c.orig	2023-09-24 22:34:17.435312000 +0200
+++ dia/lib/dia_dirs.c	2023-09-24 22:34:27.268390400 +0200
@@ -88,7 +88,7 @@
 {
 #ifdef G_OS_WIN32
   char *sLoc = g_win32_get_package_installation_directory_of_module (NULL);
-  char *returnPath = g_build_filename (sLoc, "dia", NULL);
+  char *returnPath = g_build_filename (sLoc, "lib", "dia", NULL);
 
   g_clear_pointer (&sLoc, g_free);
 
