--- dia/plug-ins/pdf/pdf-import.cpp.orig	2025-09-07 19:54:56.769136400 +0200
+++ dia/plug-ins/pdf/pdf-import.cpp	2025-09-07 19:55:06.490881600 +0200
@@ -365,7 +365,7 @@
     if ((pf = strstr (family, " Oblique")) != NULL)
       *pf = 0;
 
-    const double *fm = f->getFontMatrix();
+    const auto& fm = f->getFontMatrix();
     double fsize = state->getTransformedFontSize();
     if (fm[0] != 0)
       fsize *= fabs(fm[3] / fm[0]);
--- dia/plug-ins/pdf/pdf-import.cpp.orig	2025-10-10 20:37:15.249754400 +0200
+++ dia/plug-ins/pdf/pdf-import.cpp	2025-10-10 20:41:58.322149300 +0200
@@ -776,7 +776,12 @@
 DiaOutputDev::drawString(GfxState *state, GooString *s)
 {
   Color text_color = this->fill_color;
+#if (POPPLER_VERSION_MAJOR > 25) || \
+    (POPPLER_VERSION_MAJOR == 25 && POPPLER_VERSION_MINOR >= 10)
+  int len = s->size();
+#else
   int len = s->getLength();
+#endif
   DiaObject *obj;
   gchar *utf8 = NULL;
   DiaFont *font;
--- dia/plug-ins/pdf/pdf-import.cpp.orig	2026-01-20 19:12:47.612762200 +0100
+++ dia/plug-ins/pdf/pdf-import.cpp	2026-01-20 19:14:40.751100100 +0100
@@ -846,8 +846,12 @@
       int rowstride = gdk_pixbuf_get_rowstride (pixbuf);
       unsigned char *pixels = gdk_pixbuf_get_pixels (pixbuf);
       int y;
-
+#if (POPPLER_VERSION_MAJOR > 26) || \
+    (POPPLER_VERSION_MAJOR == 26 && POPPLER_VERSION_MINOR >= 1)
+      imgStr.rewind();
+#else
       imgStr.reset(); // otherwise getLine() is crashing right away
+#endif
       line = imgStr.getLine ();
       for (y = 0; y < height && line; ++y) {
 	unsigned char *dest = pixels + y * rowstride;
--- dia/plug-ins/pdf/pdf-import.cpp.orig	2026-02-24 20:41:33.179842200 +0100
+++ dia/plug-ins/pdf/pdf-import.cpp	2026-02-24 20:43:38.796177000 +0100
@@ -109,7 +109,7 @@
     return FALSE;
   }
   //! Apparently no effect at all - so we translate everything to Dia space ouself
-  void setDefaultCTM(double *ctm)
+  void setDefaultCTM(auto ctm)
   {
     DiaMatrix mat;
 
@@ -822,7 +822,7 @@
   GdkPixbuf *pixbuf;
   Point pos;
   DiaObjectChange *change;
-  const double *ctm = state->getCTM();
+  const auto ctm = state->getCTM();
 
   pos.x = ctm[4] * scale;
   // there is some undocumented magic done with the ctm for drawImage
--- dia/plug-ins/pdf/pdf-import.cpp.orig	2026-07-09 03:31:14.616721000 +0800
+++ dia/plug-ins/pdf/pdf-import.cpp	2026-07-09 03:34:17.630154600 +0800
@@ -82,14 +82,14 @@
 			bool (*annotDisplayDecideCbk)(Annot *annot, void *user_data),
 			void *annotDisplayDecideCbkData)
   {
-    const PDFRectangle *mediaBox = page->getMediaBox();
-    const PDFRectangle *clipBox = page->getCropBox ();
+    const PDFRectangle &mediaBox = page->getMediaBox();
+    const PDFRectangle &clipBox = page->getCropBox ();
 
     if (page->isOk()) {
-      real w1 = (clipBox->x2 - clipBox->x1);
-      real h1 = (clipBox->y2 - clipBox->y1);
-      real w2 = (mediaBox->x2 - mediaBox->x1);
-      real h2 = (mediaBox->y2 - mediaBox->y1);
+      real w1 = (clipBox.x2 - clipBox.x1);
+      real h1 = (clipBox.y2 - clipBox.y1);
+      real w2 = (mediaBox.x2 - mediaBox.x1);
+      real h2 = (mediaBox.y2 - mediaBox.y1);
 
       if (w2 > w1 || h2 > h1) {
         page_width = w2 * scale;
