--- dia/plug-ins/pdf/pdf-import.cpp.orig	2025-09-07 19:54:56.769136400 +0200
+++ dia/plug-ins/pdf/pdf-import.cpp	2025-09-07 19:55:06.490881600 +0200
@@ -365,7 +365,7 @@
     if ((pf = strstr (family, " Oblique")) != NULL)
       *pf = 0;
 
-    const double *fm = f->getFontMatrix();
+    const auto& fm = f->getFontMatrix();
     double fsize = state->getTransformedFontSize();
     if (fm[0] != 0)
       fsize *= fabs(fm[3] / fm[0]);
--- dia/plug-ins/pdf/pdf-import.cpp.orig	2025-10-10 20:37:15.249754400 +0200
+++ dia/plug-ins/pdf/pdf-import.cpp	2025-10-10 20:41:58.322149300 +0200
@@ -776,7 +776,12 @@
 DiaOutputDev::drawString(GfxState *state, GooString *s)
 {
   Color text_color = this->fill_color;
+#if (POPPLER_VERSION_MAJOR > 25) || \
+    (POPPLER_VERSION_MAJOR == 25 && POPPLER_VERSION_MINOR >= 10)
+  int len = s->size();
+#else
   int len = s->getLength();
+#endif
   DiaObject *obj;
   gchar *utf8 = NULL;
   DiaFont *font;
--- dia/plug-ins/pdf/pdf-import.cpp.orig	2026-01-20 19:12:47.612762200 +0100
+++ dia/plug-ins/pdf/pdf-import.cpp	2026-01-20 19:14:40.751100100 +0100
@@ -846,8 +846,12 @@
       int rowstride = gdk_pixbuf_get_rowstride (pixbuf);
       unsigned char *pixels = gdk_pixbuf_get_pixels (pixbuf);
       int y;
-
+#if (POPPLER_VERSION_MAJOR > 26) || \
+    (POPPLER_VERSION_MAJOR == 26 && POPPLER_VERSION_MINOR >= 1)
+      imgStr.rewind();
+#else
       imgStr.reset(); // otherwise getLine() is crashing right away
+#endif
       line = imgStr.getLine ();
       for (y = 0; y < height && line; ++y) {
 	unsigned char *dest = pixels + y * rowstride;
