diff -aurN 000/Source/nsisunz.cpp 001/Source/nsisunz.cpp
--- 000/Source/nsisunz.cpp	2010-02-06 20:06:02.000000000 -0200
+++ 001/Source/nsisunz.cpp	2015-04-30 14:45:04.808967800 -0300
@@ -17,13 +17,11 @@
 #include <windows.h>
 #include <stdio.h>
 #include <commctrl.h>
-#include "..\ExDLL\exdll.h"
+#include <nsis/pluginapi.h>
 
 //#include "exdll.h"
 
-extern "C" {
-#include "zlib/unzip.h"
-};
+#include <minizip/unzip.h>
 
 //Strings used all over the place
 TCHAR szSuccess[]        = {_T('s'),_T('u'),_T('c'),_T('c'),_T('e'),_T('s'),_T('s'),_T('\0')};
