diff --git a/src/tools/qtgrpcgen/clientdeclarationprinter.cpp b/src/tools/qtgrpcgen/clientdeclarationprinter.cpp
index 9bd57a1..eedb91f 100644
--- a/src/tools/qtgrpcgen/clientdeclarationprinter.cpp
+++ b/src/tools/qtgrpcgen/clientdeclarationprinter.cpp
@@ -8,7 +8,6 @@
 #include <google/protobuf/io/printer.h>
 #include <google/protobuf/io/zero_copy_stream.h>
 #include <google/protobuf/stubs/common.h>
-#include <google/protobuf/stubs/logging.h>
 
 #include <string>
 #include <unordered_set>
diff --git a/src/tools/qtgrpcgen/qgrpcgenerator.cpp b/src/tools/qtgrpcgen/qgrpcgenerator.cpp
index 4cab74d..1d5e04d 100644
--- a/src/tools/qtgrpcgen/qgrpcgenerator.cpp
+++ b/src/tools/qtgrpcgen/qgrpcgenerator.cpp
@@ -12,7 +12,7 @@
 
 #include <set>
 #include <google/protobuf/compiler/code_generator.h>
+#include <google/protobuf/descriptor_legacy.h>
-#include <google/protobuf/stubs/logging.h>
 #include <google/protobuf/stubs/common.h>
 #include <google/protobuf/io/printer.h>
 #include <google/protobuf/io/zero_copy_stream.h>
@@ -35,7 +34,7 @@ bool QGrpcGenerator::Generate(const FileDescriptor *file,
                               std::string *error) const
 {
     assert(file != nullptr && generatorContext != nullptr);
-    if (file->syntax() != FileDescriptor::SYNTAX_PROTO3) {
+    if (FileDescriptorLegacy(file).syntax() != FileDescriptorLegacy::SYNTAX_PROTO3) {
         *error = "Invalid proto used. qtgrpcgen only supports 'proto3' syntax";
         return false;
     }
diff --git a/src/tools/qtprotobufgen/qprotobufgenerator.cpp b/src/tools/qtprotobufgen/qprotobufgenerator.cpp
index 5a1ccc2..e0d8981 100644
--- a/src/tools/qtprotobufgen/qprotobufgenerator.cpp
+++ b/src/tools/qtprotobufgen/qprotobufgenerator.cpp
@@ -17,7 +17,7 @@
 #include <array>
 #include <numeric>
 #include <set>
+#include <google/protobuf/descriptor_legacy.h>
-#include <google/protobuf/stubs/logging.h>
 #include <google/protobuf/stubs/common.h>
 #include <google/protobuf/io/printer.h>
 #include <google/protobuf/io/zero_copy_stream.h>
@@ -41,7 +40,7 @@ bool QProtobufGenerator::Generate(const FileDescriptor *file,
 {
     assert(file != nullptr && generatorContext != nullptr);
 
-    if (file->syntax() != FileDescriptor::SYNTAX_PROTO3) {
+    if (FileDescriptorLegacy(file).syntax() != FileDescriptorLegacy::SYNTAX_PROTO3) {
         *error = "Invalid proto used. qtprotobufgen only supports 'proto3' syntax";
         return false;
     }
diff --git a/src/tools/qtprotoccommon/generatorbase.cpp b/src/tools/qtprotoccommon/generatorbase.cpp
index bc51206..1004c61 100644
--- a/src/tools/qtprotoccommon/generatorbase.cpp
+++ b/src/tools/qtprotoccommon/generatorbase.cpp
@@ -5,7 +5,6 @@
 #include "generatorbase.h"
 
 #include <google/protobuf/descriptor.h>
-#include <google/protobuf/stubs/logging.h>
 #include <google/protobuf/stubs/common.h>
 #include <google/protobuf/io/printer.h>
 #include <google/protobuf/io/zero_copy_stream.h>
