From e470ec83cf1ce72b7095ea737c357e4be5312236 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Thu, 2 Feb 2017 09:56:22 +0300
Subject: [PATCH 5/6] Guard compile flags for MSVC

---
 DevIL/src-ILUT/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/DevIL/src-ILUT/CMakeLists.txt b/DevIL/src-ILUT/CMakeLists.txt
index 296f9955..a2926bec 100644
--- a/DevIL/src-ILUT/CMakeLists.txt
+++ b/DevIL/src-ILUT/CMakeLists.txt
@@ -32,7 +32,9 @@ if(WIN32)
 		else(UNICODE)
 			set(ILUT_RSRC ${ILUT_RSRC} "msvc/resources/IL Logo.ico" "msvc/ILUT.rc")
 		endif(UNICODE)
-		set(CMAKE_C_FLAGS_RELEASE "/MT /O2")
+		if(MSVC)
+			set(CMAKE_C_FLAGS_RELEASE "/MT /O2")
+		endif(MSVC)
 	else(BUILD_SHARED_LIBS)
 		add_definitions(-DIL_STATIC_LIB)
 	endif(BUILD_SHARED_LIBS)
-- 
2.11.0

