--- build-x86_64/build/gcc/Makefile.orig	2017-08-25 02:09:16.910819400 +0900
+++ build-x86_64/build/gcc/Makefile	2017-08-25 02:09:51.862555800 +0900
@@ -38,7 +38,7 @@
 CBASEFLAGS = -Wall -Wextra -fno-rtti -fno-exceptions -std=c++11
 JAVAINCS   = -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
 CXX = g++
-INSTALL=install -o $(USER) -g $(USER)
+INSTALL=install
 
 # Library's major version number -- Increment in case of incompatible API
 # change.
