--- gnupg-2.1.3/build-aux/potomo	2015-04-30 18:27:58 +0400
+++ gnupg-2.1.3/build-aux/potomo	2015-04-30 15:05:06 +0400
@@ -57,7 +57,7 @@
         ;;   
     *)
         echo "potomo: '$infile' converting from $fromset to utf-8" >&2
-        iconv --silent --from-code=$fromset --to-code=utf-8 < "$infile" |\
+        iconv -s --from-code=$fromset --to-code=utf-8 < "$infile" |\
             sed "/^\"Content-Type:/ s/charset=[a-zA-Z0-9_-]*/charset=utf-8/"|\
             msgfmt --output-file="$outfile" -
         ;;
