--- a/common/homedir.c
+++ b/common/homedir.c
@@ -760,6 +760,9 @@
 check_portable_app (const char *dir)
 {
   char *fname;
+#ifdef __MINGW32__
+  unsigned int flags;
+#endif
 
   fname = xstrconcat (dir, DIRSEP_S "gpgconf.exe", NULL);
   if (!gnupg_access (fname, F_OK))
@@ -766,6 +769,13 @@
   fname = xstrconcat (dir, DIRSEP_S "gpgconf.exe", NULL);
   if (!gnupg_access (fname, F_OK))
     {
+#ifdef __MINGW32__
+      w32_portable_app = 1;
+      log_get_prefix (&flags);
+      log_set_prefix (NULL, (flags | GPGRT_LOG_NO_REGISTRY));
+      xfree (fname);
+      return;
+#endif
       strcpy (fname + strlen (fname) - 3, "ctl");
       parse_gpgconf_ctl (fname);
       if ((gpgconf_ctl.found && gpgconf_ctl.empty)
@@ -1548,7 +1558,7 @@
   if (!name)
     {
-      name = xstrconcat (w32_commondir (), DIRSEP_S, "etc", DIRSEP_S,
+      name = xstrconcat (w32_rootdir (), DIRSEP_S, "etc", DIRSEP_S,
                          my_gnupg_dirname (), NULL);
       gpgrt_annotate_leaked_object (name);
