--- a/common/t-stringhelp.c
+++ b/common/t-stringhelp.c
@@ -410,40 +410,40 @@
 
   out = make_filename_try ("~/foo", "bar", NULL);
   if (!out)
     fail (2);
   else if (home)
     {
       if (strlen (out) < homelen + 7)
         fail (2);
       else if (strncmp (out, home, homelen))
         fail (2);
-      else if (strcmp (out+homelen, "/foo/bar"))
+      else if (compare_filenames (out+homelen, "/foo/bar"))
         fail (2);
     }
   else
     {
       if (strcmp (out, "~/foo/bar"))
         fail (2);
     }
   xfree (out);
 
   out = make_filename_try ("~", "bar", NULL);
   if (!out)
     fail (2);
   else if (home)
     {
       if (strlen (out) < homelen + 3)
         fail (2);
       else if (strncmp (out, home, homelen))
         fail (2);
-      else if (strcmp (out+homelen, "/bar"))
+      else if (compare_filenames (out+homelen, "/bar"))
         fail (2);
     }
   else
     {
       if (strcmp (out, "~/bar"))
         fail (2);
     }
   xfree (out);
