--- Vulkan-ValidationLayers-1.2.179.orig/layers/CMakeLists.txt	2021-05-25 04:31:03.000000000 +0530
+++ Vulkan-ValidationLayers-1.2.179/layers/CMakeLists.txt	2021-05-31 02:07:58.673155900 +0530
@@ -81,6 +81,7 @@
 # System-specific macro to create a library target.
 macro(AddVkLayer target LAYER_COMPILE_DEFINITIONS)
     add_library(VkLayer_${target} SHARED ${ARGN})
+    set_target_properties(VkLayer_${target} PROPERTIES DEFINE_SYMBOL "VK_BUILDING_DLL")
     set_target_properties(VkLayer_${target} PROPERTIES CXX_STANDARD ${VVL_CPP_STANDARD})
     target_compile_definitions(VkLayer_${target} PUBLIC ${LAYER_COMPILE_DEFINITIONS})
     target_link_libraries(VkLayer_${target} PRIVATE VkLayer_utils)
--- Vulkan-ValidationLayers/layers/vk_format_utils.h.orig	2020-02-28 11:50:38.167672800 +0300
+++ Vulkan-ValidationLayers/layers/vk_format_utils.h	2020-02-28 11:51:10.078172600 +0300
@@ -25,7 +25,13 @@
 #include "vulkan/vulkan.h"
 
 #if !defined(VK_LAYER_EXPORT)
-#if defined(__GNUC__) && __GNUC__ >= 4
+#if defined(_WIN32)
+#  if defined (VK_BUILDING_DLL)
+#    define VK_LAYER_EXPORT __declspec(dllexport)
+#  else
+#    define VK_LAYER_EXPORT
+#  endif
+#elif defined(__GNUC__) && __GNUC__ >= 4
 #define VK_LAYER_EXPORT __attribute__((visibility("default")))
 #elif defined(__SUNPRO_C) && (__SUNPRO_C >= 0x590)
 #define VK_LAYER_EXPORT __attribute__((visibility("default")))
