diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01aa01f..d18ee33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,7 +77,11 @@ if(BUILD_TESTING AND INJA_BUILD_TESTS)
 
   add_executable(inja_test test/test.cpp)
   target_link_libraries(inja_test PRIVATE inja)
-  add_test(inja_test ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/inja_test)
+  target_include_directories(inja_test PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/third_party/include>
+  )
+  add_test(NAME inja_test COMMAND inja_test
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test)
 
   if(MSVC)
     target_compile_options(inja_test PRIVATE /W4 /permissive-)
@@ -88,12 +92,20 @@ if(BUILD_TESTING AND INJA_BUILD_TESTS)
 
   add_library(single_inja INTERFACE)
   target_compile_features(single_inja INTERFACE cxx_std_11)
-  target_include_directories(single_inja INTERFACE single_include include third_party/include)
+  target_link_libraries(single_inja INTERFACE ${INJA_SELECTED_JSON_LIBRARY})
+  target_include_directories(single_inja INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  )
+
 
   add_executable(single_inja_test test/test.cpp)
   target_link_libraries(single_inja_test PRIVATE single_inja)
+  target_include_directories(single_inja_test PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/third_party/include>
+  )
 
-  add_test(single_inja_test ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/single_inja_test)
+  add_test(NAME single_inja_test COMMAND single_inja_test
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test)
 endif()
 
 
