diff -Naur leocad-23.03.origin/leocad.pro leocad-23.03/leocad.pro
--- leocad-23.03.origin/leocad.pro	2023-03-20 02:09:47.000000000 +0300
+++ leocad-23.03/leocad.pro	2023-12-25 23:41:57.637237400 +0300
@@ -53,7 +53,7 @@
 }
 
 isEmpty(QMAKE_LRELEASE) {
-	win32:QMAKE_LRELEASE = $$[QT_INSTALL_BINS]\\lrelease.exe
+	!win32:QMAKE_LRELEASE = $$[QT_INSTALL_BINS]\\lrelease.exe
 	else:QMAKE_LRELEASE = $$[QT_INSTALL_BINS]/lrelease
 	unix {
 		!exists($$QMAKE_LRELEASE) { QMAKE_LRELEASE = lrelease-qt4 }
@@ -92,7 +92,7 @@
 unix:!macx {
 	TARGET = leocad
 } else {
-	TARGET = LeoCAD
+	TARGET = leocad
 }
 
 CONFIG(debug, debug|release) {
@@ -106,7 +106,7 @@
 RCC_DIR = $$DESTDIR/.qrc
 UI_DIR = $$DESTDIR/.ui
 
-unix:!macx {
+win32 {
 	isEmpty(INSTALL_PREFIX):INSTALL_PREFIX = /usr
 	isEmpty(BIN_DIR):BIN_DIR = $$INSTALL_PREFIX/bin
 	isEmpty(DOCS_DIR):DOCS_DIR = $$INSTALL_PREFIX/share/doc/leocad
@@ -116,14 +116,14 @@
 	isEmpty(MIME_DIR):MIME_DIR = $$INSTALL_PREFIX/share/mime/packages
 	isEmpty(APPDATA_DIR):APPDATA_DIR = $$INSTALL_PREFIX/share/metainfo
 
-	target.path = $$BIN_DIR
-	docs.path = $$DOCS_DIR
+	target.path = $$PREFIX/bin
+	docs.path = $$PREFIX/share/doc/leocad
 	docs.files = docs/README.md docs/CREDITS.txt docs/COPYING.txt
-	man.path = $$MAN_DIR
+	man.path = $$PREFIX/share/man/man1
 	man.files = docs/leocad.1
-	desktop.path = $$DESKTOP_DIR
+	desktop.path = $$PREFIX/share/applications
 	desktop.files = qt/leocad.desktop
-	icons.path = $$ICON_DIR/hicolor
+	icons.path = $$PREFIX/share/icons/hicolor
 	icons.files += tools/icon/128x128
 	icons.files += tools/icon/16x16
 	icons.files += tools/icon/24x24
@@ -133,9 +133,9 @@
 	icons.files += tools/icon/512x512
 	icons.files += tools/icon/64x64
 	icons.files += tools/icon/scalable
-	mime.path = $$MIME_DIR
+	mime.path = $$PREFIX/share/mime/packages
 	mime.files = qt/leocad.xml
-	appdata.path = $$APPDATA_DIR
+	appdata.path = $$PREFIX/share/metainfo
 	appdata.files = tools/setup/leocad.appdata.xml
 
 	INSTALLS += target docs man desktop icons mime appdata
diff -Naur leocad-23.03.origin/qt/qtmain.cpp leocad-23.03/qt/qtmain.cpp
--- leocad-23.03.origin/qt/qtmain.cpp	2023-03-20 02:09:47.000000000 +0300
+++ leocad-23.03/qt/qtmain.cpp	2023-12-25 23:42:07.382339500 +0300
@@ -189,7 +189,7 @@
 	LibraryPaths += qMakePair(QDir::cleanPath(QCoreApplication::applicationDirPath() + "/library.bin"), true);
 #endif
 
-#ifdef Q_OS_LINUX
+#ifdef Q_OS_WIN
 	LibraryPaths += qMakePair(QDir::cleanPath(QCoreApplication::applicationDirPath() + "/../share/leocad/library.bin"), true);
 #endif
 
