diff -Naur leocad-25.09-orig/leocad.pro leocad-25.09/leocad.pro
--- leocad-25.09-orig/leocad.pro	2025-09-04 21:10:51.502607600 +0300
+++ leocad-25.09/leocad.pro	2025-09-04 21:13:46.628653700 +0300
@@ -48,7 +48,7 @@
 	LIBS += -ladvapi32 -lshell32 -lopengl32 -luser32
 } else {
 	PRECOMPILED_HEADER = common/lc_global.h
-	LIBS += -lz
+	LIBS += -lz -lopengl32
 	QMAKE_CXXFLAGS_WARN_ON += -Wno-unused-parameter
 }
 
diff -Naur leocad-25.09-orig/qt/lc_qutils.h leocad-25.09/qt/lc_qutils.h
--- leocad-25.09-orig/qt/lc_qutils.h	2025-09-04 21:10:51.643816100 +0300
+++ leocad-25.09/qt/lc_qutils.h	2025-09-04 21:11:23.959196200 +0300
@@ -12,6 +12,8 @@
 
 #ifdef Q_OS_WIN
 
+#include <windows.h>
+
 int lcTerminateChildProcess(QWidget* Parent, const qint64 Pid, const qint64 Ppid);
 bool lcRunElevatedProcess(const LPCWSTR ExeName, const LPCWSTR Arguments, const LPCWSTR WorkingFolder);
 
