diff -Naur leocad-25.09-orig/leocad.pro leocad-25.09/leocad.pro
--- leocad-25.09-orig/leocad.pro	2025-09-04 21:10:51.502607600 +0300
+++ leocad-25.09/leocad.pro	2025-09-04 21:13:46.628653700 +0300
@@ -48,7 +48,7 @@
 	LIBS += -ladvapi32 -lshell32 -lopengl32 -luser32
 } else {
 	PRECOMPILED_HEADER = common/lc_global.h
-	LIBS += -lz
+	LIBS += -lz -lopengl32
 	QMAKE_CXXFLAGS_WARN_ON += -Wno-unused-parameter
 }
 
