--- coin3d/CMakeLists.txt.orig	2018-07-21 23:04:17.716238200 +0300
+++ coin3d/CMakeLists.txt	2018-07-21 23:04:55.259385600 +0300
@@ -150,9 +150,6 @@
 	if(COIN_BUILD_MSVC_MP)
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
 	endif ()
-	if(COIN_BUILD_SHARED_LIBS)
-		add_definitions(-DCOIN_MAKE_DLL)
-	endif()
 
 	# enable C++ exception handling (for code that uses boost in SbByteBuffer)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
@@ -172,6 +169,9 @@
 	set(CMAKE_MINSIZEREL_POSTFIX "${COIN_MAJOR_VERSION}${CMAKE_MINSIZEREL_POSTFIX}")
 	set(CMAKE_RELWITHDEBINFO_POSTFIX "${COIN_MAJOR_VERSION}${CMAKE_RELWITHDEBINFO_POSTFIX}")
 	set(CMAKE_DEBUG_POSTFIX "${COIN_MAJOR_VERSION}${CMAKE_DEBUG_POSTFIX}")
+	if(COIN_BUILD_SHARED_LIBS)
+		add_definitions(-DCOIN_MAKE_DLL)
+	endif()
 elseif(APPLE)
 	cmake_policy(SET CMP0042 NEW)
 	set(CMAKE_SHARED_LINKER_FLAGS "-framework CoreFoundation -framework CoreGraphics")
--- coin3d/examples/dynloadextensions/CMakeLists.txt.orig	2018-07-22 00:22:42.902462700 +0300
+++ coin3d/examples/dynloadextensions/CMakeLists.txt	2018-07-22 00:26:27.204292000 +0300
@@ -8,6 +8,8 @@
 
 add_library(Switchboard SHARED Switchboard.cpp)
 target_compile_definitions(Switchboard PRIVATE "-DSWITCHBOARD_INTERNAL")
+add_dependencies(Switchboard Coin)
+target_link_libraries(Switchboard Coin)
 
 add_library(SwitchboardOperator SHARED SwitchboardOperator.cpp)
 target_compile_definitions(SwitchboardOperator PRIVATE "-DSWITCHBOARDOPERATOR_INTERNAL")
