--- a/src/framework/CMakeLists.txt
+++ b/src/framework/CMakeLists.txt
@@ -86,12 +86,7 @@
 
 if(WIN32)
   if(MINGW)
-    install(FILES "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/libmlt-${MLT_VERSION_MAJOR}.dll"
-      DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      RENAME libmlt.dll
-    )
     target_link_options(mlt PRIVATE -Wl,--output-def,libmlt.def)
-    install(FILES "${CMAKE_BINARY_DIR}/libmlt.def" DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
   target_sources(mlt PRIVATE ../win32/win32.c ../win32/strptime.c)
   target_link_libraries(mlt PRIVATE Iconv::Iconv)
--- a/src/mlt++/CMakeLists.txt
+++ b/src/mlt++/CMakeLists.txt
@@ -78,12 +78,7 @@
 
 if(WIN32)
   if(MINGW)
-    install(FILES "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/libmlt++-${MLT_VERSION_MAJOR}.dll"
-      DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      RENAME libmlt++.dll
-    )
     target_link_options(mlt++ PRIVATE -Wl,--output-def,libmlt++.def)
-    install(FILES "${CMAKE_BINARY_DIR}/libmlt++.def" DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
   target_compile_definitions(mlt++ PRIVATE MLTPP_EXPORTS)
 endif()
--- a/src/swig/python/CMakeLists.txt
+++ b/src/swig/python/CMakeLists.txt
@@ -5,7 +5,7 @@
 target_link_libraries(mltpython PRIVATE mlt mlt++ Python3::Module)
 set_target_properties(mltpython PROPERTIES PREFIX "_" OUTPUT_NAME "mlt${MLT_VERSION_MAJOR}")
 
-string(REGEX MATCH "[lL]ib.*" PYTHON_MODULE_INSTALL_DIR ${Python3_SITEARCH})
+set(PYTHON_MODULE_INSTALL_DIR "lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}/site-packages")
 
 install(TARGETS mltpython DESTINATION ${PYTHON_MODULE_INSTALL_DIR})
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mlt.py
