diff -bur mlt-7.24.0-orig/src/framework/CMakeLists.txt mlt-7.24.0/src/framework/CMakeLists.txt
--- mlt-7.24.0-orig/src/framework/CMakeLists.txt	2024-06-09 03:07:57.858510100 -0600
+++ mlt-7.24.0/src/framework/CMakeLists.txt	2024-06-09 03:08:40.775369700 -0600
@@ -97,7 +97,6 @@
 if(WIN32)
   if(MINGW)
     target_link_options(mlt PRIVATE -Wl,--output-def,${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/libmlt-${MLT_VERSION_MAJOR}.def)
-    install(FILES "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/libmlt-${MLT_VERSION_MAJOR}.def" DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
   target_sources(mlt PRIVATE ../win32/win32.c ../win32/strptime.c)
   target_link_libraries(mlt PRIVATE Iconv::Iconv)
diff -bur mlt-7.24.0-orig/src/mlt++/CMakeLists.txt mlt-7.24.0/src/mlt++/CMakeLists.txt
--- mlt-7.24.0-orig/src/mlt++/CMakeLists.txt	2024-06-09 03:07:57.871510200 -0600
+++ mlt-7.24.0/src/mlt++/CMakeLists.txt	2024-06-09 03:09:11.414556000 -0600
@@ -86,7 +86,6 @@
 if(WIN32)
   if(MINGW)
     target_link_options(mlt++ PRIVATE -Wl,--output-def,${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/libmlt++-${MLT_VERSION_MAJOR}.def)
-    install(FILES "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/libmlt++-${MLT_VERSION_MAJOR}.def" DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
   target_compile_definitions(mlt++ PRIVATE MLTPP_EXPORTS)
 endif()
diff -bur mlt-7.24.0-orig/src/swig/python/CMakeLists.txt mlt-7.24.0/src/swig/python/CMakeLists.txt
--- mlt-7.24.0-orig/src/swig/python/CMakeLists.txt	2024-06-09 03:07:58.045530400 -0600
+++ mlt-7.24.0/src/swig/python/CMakeLists.txt	2024-06-09 03:09:52.453802400 -0600
@@ -6,7 +6,7 @@
 set_target_properties(mltpython PROPERTIES PREFIX "_" OUTPUT_NAME "mlt${MLT_VERSION_MAJOR}")
 set_target_properties(mltpython PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/python")
 
-string(REGEX MATCH "[lL]ib.*" PYTHON_MODULE_INSTALL_DIR ${Python3_SITEARCH})
+set(PYTHON_MODULE_INSTALL_DIR "lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}/site-packages")
 
 install(TARGETS mltpython DESTINATION ${PYTHON_MODULE_INSTALL_DIR})
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mlt.py
