diff -bur mlt-7.22.0-orig/src/framework/mlt_repository.c mlt-7.22.0/src/framework/mlt_repository.c
--- mlt-7.22.0-orig/src/framework/mlt_repository.c	2024-03-09 19:36:20.988094400 -0700
+++ mlt-7.22.0/src/framework/mlt_repository.c	2024-03-09 19:37:46.059073700 -0700
@@ -580,7 +580,7 @@
                     // load the preset
                     mlt_properties preset = mlt_properties_load(fullname);
                     if (preset && mlt_properties_count(preset)) {
-                        snprintf(fullname, 1024, "%s/%s", path, de->d_name);
+                        snprintf(fullname, sizeof(fullname), "%s/%s", path, de->d_name);
                         mlt_properties_set_data(properties,
                                                 fullname,
                                                 preset,
