--- a/src/modules/gdk/CMakeLists.txt
+++ b/src/modules/gdk/CMakeLists.txt
@@ -19,7 +19,7 @@
 if(TARGET PkgConfig::pango AND TARGET PkgConfig::fontconfig)
   target_sources(mltgdk PRIVATE producer_pango.c)
   target_link_libraries(mltgdk PRIVATE PkgConfig::pango PkgConfig::fontconfig PkgConfig::pangoft2)
-  if(APPLE)
+  if(APPLE OR MINGW)
     target_link_libraries(mltgdk PRIVATE iconv)
   endif()
   target_compile_definitions(mltgdk PRIVATE USE_PANGO)
