diff -Nur mlt-6.24.0-orig/CMakeLists.txt mlt-6.24.0/CMakeLists.txt
--- mlt-6.24.0-orig/CMakeLists.txt	2020-12-07 06:37:27.000000000 +0100
+++ mlt-6.24.0/CMakeLists.txt	2021-03-14 09:33:02.232366800 +0100
@@ -49,7 +49,7 @@
     src/modules/motion_est
     src/modules/normalize
     src/modules/oldfilm
-    src/modules/opencv
+    #src/modules/opencv
     src/modules/opengl
     src/modules/plus
     src/modules/plusgpl
@@ -57,7 +57,7 @@
     src/modules/resample
     src/modules/rubberband
     src/modules/rtaudio
-    src/modules/sdl
+    #src/modules/sdl
     src/modules/sdl2
     src/modules/sox
     src/modules/vid.stab
