From 8441d5863cb90ed48199d6976bc6e2fcfbe3365a Mon Sep 17 00:00:00 2001
From: Morten Welinder <terra@gnome.org>
Date: Sat, 2 May 2020 21:24:12 -0400
Subject: [PATCH] Compilation: fix win32 build

---
 NEWS            | 3 +++
 gsf/Makefile.am | 3 +--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index bbfeefd..2c3c986 100644
--- a/NEWS
+++ b/NEWS
@@ -1,5 +1,8 @@
 libgsf 1.14.48
 
+Greg Hellings:
+	* Fix win32 build.
+
 --------------------------------------------------------------------------
 libgsf 1.14.47
 
diff --git a/gsf/Makefile.am b/gsf/Makefile.am
index 20d92bc..121015e 100644
--- a/gsf/Makefile.am
+++ b/gsf/Makefile.am
@@ -18,8 +18,7 @@ EXTRA_DIST = libgsf.syms
 libgsf_1_la_LIBADD  = $(LIBGSF_LIBS) $(Z_LIBS) $(BZ2_LIBS)
 libgsf_1_la_LDFLAGS = -version-info $(VERSION_INFO) -export-symbols $(srcdir)/libgsf.syms
 if WITH_WIN32
-  libgsf_1_la_DEPENDENCIES = lib.def
-  libgsf_1_la_LDFLAGS += -no-undefined -export-symbols lib.def
+  libgsf_1_la_LDFLAGS += -no-undefined
 endif
 
 libgsf_1_la_SOURCES =		\
-- 
GitLab

