diff -c -r glog-4d391fe692ae6b9e0105f473945c415a3ce5a401.orig/CMakeLists.txt glog-4d391fe692ae6b9e0105f473945c415a3ce5a401/CMakeLists.txt
*** glog-4d391fe692ae6b9e0105f473945c415a3ce5a401.orig/CMakeLists.txt	2015-12-17 05:31:55.000000000 -0200
--- glog-4d391fe692ae6b9e0105f473945c415a3ce5a401/CMakeLists.txt	2016-03-16 18:27:21.309311200 -0300
***************
*** 333,339 ****
  set (CMAKE_VISIBILITY_INLINES_HIDDEN 1)
  
  set (GLOG_PUBLIC_H
-   ${CMAKE_CURRENT_BINARY_DIR}/config.h
    ${CMAKE_CURRENT_BINARY_DIR}/glog/logging.h
    ${CMAKE_CURRENT_BINARY_DIR}/glog/raw_logging.h
    ${CMAKE_CURRENT_BINARY_DIR}/glog/stl_logging.h
--- 333,338 ----
***************
*** 343,348 ****
--- 342,348 ----
  
  set (GLOG_SRCS
    ${GLOG_PUBLIC_H}
+   ${CMAKE_CURRENT_BINARY_DIR}/config.h
    src/base/commandlineflags.h
    src/base/googleinit.h
    src/base/mutex.h
***************
*** 418,424 ****
  target_include_directories (glog BEFORE PUBLIC
    "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>"
    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>"
!   "$<INSTALL_INTERFACE:include>"
    PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)

--- 418,424 ----
  target_include_directories (glog BEFORE PUBLIC
    "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>"
    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>"
!   "$<INSTALL_INTERFACE:include/glog>"
    PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
    
diff -c -r glog-4d391fe692ae6b9e0105f473945c415a3ce5a401.orig/src/logging_unittest.cc glog-4d391fe692ae6b9e0105f473945c415a3ce5a401/src/logging_unittest.cc
*** glog-4d391fe692ae6b9e0105f473945c415a3ce5a401.orig/src/logging_unittest.cc	2015-12-17 05:31:55.000000000 -0200
--- glog-4d391fe692ae6b9e0105f473945c415a3ce5a401/src/logging_unittest.cc	2016-03-16 18:27:21.309311200 -0300
***************
*** 317,323 ****
--- 317,327 ----
    const char const_s[] = "const array";
    RAW_LOG(INFO, "%s", const_s);
    void* p = reinterpret_cast<void*>(0x12345678);
+ #if _WIN64
+   RAW_LOG(INFO, "ptr %x", p);
+ #else
    RAW_LOG(INFO, "ptr %p", p);
+ #endif
    p = NULL;
    RAW_LOG(INFO, "ptr %p", p);
    int j = 1000;
