diff -c -r glog-0.3.4.orig/src/utilities.h glog-0.3.4/src/utilities.h
*** glog-0.3.4.orig/src/utilities.h	2015-03-11 01:02:27.000000000 -0300
--- glog-0.3.4/src/utilities.h	2016-03-10 14:56:29.059644600 -0300
***************
*** 75,81 ****
  #include <string>
  
  #if defined(OS_WINDOWS)
! # include "port.h"
  #endif
  
  #include "config.h"
--- 75,81 ----
  #include <string>
  
  #if defined(OS_WINDOWS)
! # include "windows/port.h"
  #endif
  
  #include "config.h"
diff -c -r glog-0.3.4.orig/src/windows/port.h glog-0.3.4/src/windows/port.h
*** glog-0.3.4.orig/src/windows/port.h	2015-03-11 01:02:27.000000000 -0300
--- glog-0.3.4/src/windows/port.h	2016-03-10 14:44:58.427325100 -0300
***************
*** 111,122 ****
   * because they don't always NUL-terminate. :-(  We also can't use the
   * name vsnprintf, since windows defines that (but not snprintf (!)).
   */
! extern int snprintf(char *str, size_t size,
                                         const char *format, ...);
! extern int safe_vsnprintf(char *str, size_t size,
                            const char *format, va_list ap);
  #define vsnprintf(str, size, format, ap)  safe_vsnprintf(str, size, format, ap)
  #define va_copy(dst, src)  (dst) = (src)
  
  /* Windows doesn't support specifying the number of buckets as a
   * hash_map constructor arg, so we leave this blank.
--- 111,126 ----
   * because they don't always NUL-terminate. :-(  We also can't use the
   * name vsnprintf, since windows defines that (but not snprintf (!)).
   */
! #ifndef HAVE_SNPRINTF
! extern int GOOGLE_GLOG_DLL_DECL snprintf(char *str, size_t size,
                                         const char *format, ...);
! #endif
! extern int GOOGLE_GLOG_DLL_DECL safe_vsnprintf(char *str, size_t size,
                            const char *format, va_list ap);
  #define vsnprintf(str, size, format, ap)  safe_vsnprintf(str, size, format, ap)
+ #ifndef va_copy
  #define va_copy(dst, src)  (dst) = (src)
+ #endif
  
  /* Windows doesn't support specifying the number of buckets as a
   * hash_map constructor arg, so we leave this blank.
***************
*** 132,137 ****
--- 136,142 ----
  #endif  // _MSC_VER
  
  // ----------------------------------- THREADS
+ #ifndef __MINGW32__
  typedef DWORD pthread_t;
  typedef DWORD pthread_key_t;
  typedef LONG pthread_once_t;
***************
*** 143,148 ****
--- 148,154 ----
    localtime_s(result, timep);
    return result;
  }
+ #endif  // __MINGW32__
  
  inline char* strerror_r(int errnum, char* buf, size_t buflen) {
    strerror_s(buf, buflen, errnum);
