--- libnumbertext-1.0.2/configure.ac	2018-06-12 14:48:53.518952998 +0100
+++ libnumbertext-1.0.2/configure.ac	2018-06-12 14:50:48.000000000 +0100
@@ -3,7 +3,7 @@
 m4_define([numbertext_version_micro],[2])
 m4_define([numbertext_version],[numbertext_version_major.numbertext_version_minor.numbertext_version_micro])
 
-AC_PREREQ([2.68])
+AC_PREREQ([2.69])
 
 AC_INIT([libnumbertext],[numbertext_version])
 AC_CONFIG_MACRO_DIR([m4])
@@ -18,7 +18,7 @@
 AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
-AC_PROG_LIBTOOL
+LT_INIT
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_LANG([C++])

