diff -bur suil-0.10.18-orig/src/win_in_gtk2.cpp suil-0.10.18/src/win_in_gtk2.cpp
--- suil-0.10.18-orig/src/win_in_gtk2.cpp	2023-02-01 23:29:52.969298600 -0700
+++ suil-0.10.18/src/win_in_gtk2.cpp	2023-02-01 23:30:11.184529800 -0700
@@ -27,9 +27,6 @@
 #define SUIL_WIN_WRAPPER(obj) \
   (G_TYPE_CHECK_INSTANCE_CAST((obj), SUIL_TYPE_WIN_WRAPPER, SuilWinWrapper))
 
-using SuilWinWrapper      = _SuilWinWrapper;
-using SuilWinWrapperClass = _SuilWinWrapperClass;
-
 struct _SuilWinWrapper {
   GtkDrawingArea              area;
   SuilWrapper*                wrapper;
@@ -44,6 +41,9 @@
   GtkDrawingAreaClass parent_class;
 };
 
+using SuilWinWrapper      = _SuilWinWrapper;
+using SuilWinWrapperClass = _SuilWinWrapperClass;
+
 GType
 suil_win_wrapper_get_type(void); // Accessor for SUIL_TYPE_WIN_WRAPPER
 
