diff --git a/src/lib/CMakeLists.txt b/src/lib/CMakeLists.txt
index ce7828b..15c8929 100644
--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -42,6 +42,7 @@ target_include_directories(partio
         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
 
+target_link_libraries(partio PUBLIC Threads::Threads)
 if (ZLIB_FOUND)
     target_include_directories(partio
         PUBLIC
