--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -327,6 +327,7 @@
     else ()
             set(arch x86)
     endif ()
+    if (FALSE)
     #
     #  Now find signtool
     if (NOT SIGNTOOL)
@@ -407,6 +408,7 @@
             "${CMAKE_BINARY_DIR}/deploy/imageformats/qtiff.dll"
             "${CMAKE_BINARY_DIR}/deploy/imageformats/qwebp.dll"
             "${CMAKE_BINARY_DIR}/deploy/imageformats/qgif.dll")
+    endif()
 
 elseif(APPLE)
     find_library(Cocoa Cocoa)
