diff --git a/html/man/Makefile.am b/html/man/Makefile.am
index ca222de..c7421ac 100644
--- a/html/man/Makefile.am
+++ b/html/man/Makefile.am
@@ -85,7 +85,6 @@ docfiles = \
 	ras2tiff.1.html \
 	raw2tiff.1.html \
 	rgb2ycbcr.1.html \
-	sgi2tiff.1.html \
 	thumbnail.1.html \
 	tiff2bw.1.html \
 	tiff2pdf.1.html \
@@ -96,12 +95,10 @@ docfiles = \
 	tiffcrop.1.html \
 	tiffdither.1.html \
 	tiffdump.1.html \
-	tiffgt.1.html \
 	tiffinfo.1.html \
 	tiffmedian.1.html \
 	tiffset.1.html \
-	tiffsplit.1.html \
-	tiffsv.1.html
+	tiffsplit.1.html 
 
 dist_doc_DATA = $(indexfile) $(docfiles)
 
