--- llama.cpp-b7132/vendor/cpp-httplib/CMakeLists.txt.orig	2025-11-23 11:16:28.842405300 +0100
+++ llama.cpp-b7132/vendor/cpp-httplib/CMakeLists.txt	2025-11-23 11:16:31.981633700 +0100
@@ -77,6 +77,9 @@
             message(STATUS "OpenSSL found: ${OPENSSL_VERSION}")
             set(CPPHTTPLIB_OPENSSL_SUPPORT TRUE)
             target_link_libraries(${TARGET} PUBLIC OpenSSL::SSL OpenSSL::Crypto)
+            if(WIN32)
+              target_link_libraries(${TARGET} PUBLIC crypt32)
+            endif()
         endif()
     else()
         message(STATUS "OpenSSL not found, SSL support disabled")
