diff --git a/CMakeLists.txt b/CMakeLists.txt
index 432b8bf..6a71ccf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,3 +14,12 @@ if(BUILD_TESTS AND (PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR))
     endif()
     add_subdirectory(tests)
 endif()
+
+include(GNUInstallDirs)
+
+set(CMAKE_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/cmakerc"
+    CACHE STRING "Package destination dir")
+
+install(FILES "CMakeRC.cmake"
+    DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
+    RENAME "cmakerc-config.cmake")
