--- a/libusb/core.c
+++ b/libusb/core.c
@@ -2181,7 +2181,7 @@ void API_EXPORTED libusb_set_log_cb(libusb_context *ctx, libusb_log_cb cb,
  * on this platform
  * \returns LIBUSB_ERROR_NOT_FOUND if LIBUSB_OPTION_USE_USBDK is valid on this platform but UsbDk is not available
  */
-int API_EXPORTED libusb_set_option(libusb_context *ctx,
+int _cdecl libusb_set_option(libusb_context *ctx,
 	enum libusb_option option, ...)
 {
 	int arg = 0, r = LIBUSB_SUCCESS;
--- a/libusb/libusb.h
+++ b/libusb/libusb.h
@@ -2126,7 +2126,7 @@ enum libusb_option {
 	LIBUSB_OPTION_MAX = 3
 };
 
-int LIBUSB_CALL libusb_set_option(libusb_context *ctx, enum libusb_option option, ...);
+int __cdecl libusb_set_option(libusb_context *ctx, enum libusb_option option, ...);
 
 #if defined(__cplusplus)
 }
