--- libmowgli-2.0.0/src/libmowgli/platform/win32/win32_stdinc.h.orig	2014-07-10 09:10:36.115000000 +0400
+++ libmowgli-2.0.0/src/libmowgli/platform/win32/win32_stdinc.h	2014-07-10 09:17:55.234600000 +0400
@@ -37,11 +37,12 @@
 # define snprintf			_snprintf
 #endif
 
+#if !defined(__MINGW32__)
 struct timezone {
 	int tz_minuteswest;
 	int tz_dsttime;
 };
-
+#endif
 extern int gettimeofday(struct timeval *tv, struct timezone *tz);
 extern int setenv(const char *name, const char *value, int overwrite);
 extern int pipe(int pipefd[2]);
@@ -50,11 +51,6 @@
 extern int inet_pton(int af, const char *src, void *dst);
 extern const char *inet_ntop(int af, const void *addr, char *host, size_t hostlen);
 
-/* MSYS autoconf is fucko. */
-#ifndef HAVE_WINSOCK2_H
-#define HAVE_WINSOCK2_H
-#endif
-
 #define HAVE_SELECT
 
 #endif
