From 2a073ff20e2232a54adf1ea1afd6bd78af83bd50 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 31 Oct 2015 21:39:48 +0000
Subject: [PATCH 2/5] MinGW-w64: Fix libiberty makefile

---
 libiberty/Makefile.in | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libiberty/Makefile.in b/libiberty/Makefile.in
index 9b87720..eddde49 100644
--- a/libiberty/Makefile.in
+++ b/libiberty/Makefile.in
@@ -130,7 +130,7 @@ CFILES = alloca.c argv.c asprintf.c atexit.c				\
 	d-demangle.c dwarfnames.c dyn-string.c				\
 	fdmatch.c ffs.c fibheap.c filename_cmp.c floatformat.c		\
 	fnmatch.c fopen_unlocked.c					\
-	getcwd.c getopt.c getopt1.c getpagesize.c getpwd.c getruntime.c	\
+	getcwd.c getpagesize.c getpwd.c getruntime.c	\
          gettimeofday.c                                                 \
 	hashtab.c hex.c							\
 	index.c insque.c						\
@@ -155,8 +155,8 @@ CFILES = alloca.c argv.c asprintf.c atexit.c				\
 	 strtoul.c strndup.c strnlen.c strverscmp.c			\
 	timeval-utils.c tmpnam.c					\
 	unlink-if-ordinary.c						\
-	vasprintf.c vfork.c vfprintf.c vprintf.c vsnprintf.c vsprintf.c	\
-	waitpid.c							\
+	vasprintf.c vfprintf.c vprintf.c vsnprintf.c vsprintf.c	\
+	msdos.c							\
 	xatexit.c xexit.c xmalloc.c xmemdup.c xstrdup.c xstrerror.c	\
 	 xstrndup.c
 
@@ -172,7 +172,7 @@ REQUIRED_OFILES =							\
 	./fdmatch.$(objext) ./fibheap.$(objext)				\
 	./filename_cmp.$(objext) ./floatformat.$(objext)		\
 	./fnmatch.$(objext) ./fopen_unlocked.$(objext)			\
-	./getopt.$(objext) ./getopt1.$(objext) ./getpwd.$(objext)	\
+	./getpwd.$(objext)	\
 	./getruntime.$(objext) ./hashtab.$(objext) ./hex.$(objext)	\
 	./lbasename.$(objext) ./lrealpath.$(objext)			\
 	./make-relative-prefix.$(objext) ./make-temp-file.$(objext)	\
@@ -221,9 +221,9 @@ CONFIGURED_OFILES = ./asprintf.$(objext) ./atexit.$(objext)		\
 	 ./strrchr.$(objext) ./strstr.$(objext) ./strtod.$(objext)	\
 	 ./strtol.$(objext) ./strtoul.$(objext) ./strverscmp.$(objext)	\
 	./tmpnam.$(objext)						\
-	./vasprintf.$(objext) ./vfork.$(objext) ./vfprintf.$(objext)	\
+	./vasprintf.$(objext) ./vfprintf.$(objext)	\
 	 ./vprintf.$(objext) ./vsnprintf.$(objext) ./vsprintf.$(objext)	\
-	./waitpid.$(objext)
+	./msdos.$(objext)
 
 # These files are installed if the library has been configured to do so.
 INSTALLED_HEADERS =                                                     \
-- 
2.6.3

