From 592bdfbc3eb61cf5dec3b040784ab392e06d226e Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 31 Oct 2015 21:40:03 +0000
Subject: [PATCH 3/5] MinGW-w64: Fix libibery configure

---
 libiberty/configure    | 16 +---------------
 libiberty/configure.ac |  6 +-----
 2 files changed, 2 insertions(+), 20 deletions(-)

diff --git a/libiberty/configure b/libiberty/configure
index 96feaed..fc78353 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -5394,7 +5394,6 @@ funcs="$funcs vfprintf"
 funcs="$funcs vprintf"
 funcs="$funcs vsnprintf"
 funcs="$funcs vsprintf"
-funcs="$funcs waitpid"
 funcs="$funcs setproctitle"
 
 # Also in the old function.def file: alloca, vfork, getopt.
@@ -5721,13 +5720,6 @@ esac
  ;;
 esac
 
-    case " $LIBOBJS " in
-  *" waitpid.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS waitpid.$ac_objext"
- ;;
-esac
-
-
     for f in $funcs; do
       case "$f" in
 	asprintf | basename | bcmp | bcopy | bzero | clock | ffs | getpagesize | index | insque | mempcpy | mkstemps | random | rindex | sigsetmask | stpcpy | stpncpy | strdup | strndup | strnlen | strverscmp | vasprintf | waitpid)
@@ -5868,12 +5860,6 @@ esac
 esac
 
     case " $LIBOBJS " in
-  *" vfork.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS vfork.$ac_objext"
- ;;
-esac
-
-    case " $LIBOBJS " in
   *" waitpid.$ac_objext "* ) ;;
   *) LIBOBJS="$LIBOBJS waitpid.$ac_objext"
  ;;
@@ -6335,7 +6321,7 @@ fi
   if test $ac_cv_func_vfork_works = no; then
     case " $LIBOBJS " in
   *" vfork.$ac_objext "* ) ;;
-  *) LIBOBJS="$LIBOBJS vfork.$ac_objext"
+  *) 
  ;;
 esac
 
diff --git a/libiberty/configure.ac b/libiberty/configure.ac
index 3380819..fc57948 100644
--- a/libiberty/configure.ac
+++ b/libiberty/configure.ac
@@ -372,7 +372,6 @@ funcs="$funcs vfprintf"
 funcs="$funcs vprintf"
 funcs="$funcs vsnprintf"
 funcs="$funcs vsprintf"
-funcs="$funcs waitpid"
 funcs="$funcs setproctitle"
 
 # Also in the old function.def file: alloca, vfork, getopt.
@@ -473,6 +472,7 @@ if test -n "${with_target_subdir}"; then
     AC_LIBOBJ([insque])
     AC_LIBOBJ([mempcpy])
     AC_LIBOBJ([mkstemps])
+    AC_LIBOBJ([msdos])
     AC_LIBOBJ([random])
     AC_LIBOBJ([rindex])
     AC_LIBOBJ([sigsetmask])
@@ -482,7 +482,6 @@ if test -n "${with_target_subdir}"; then
     AC_LIBOBJ([strnlen])
     AC_LIBOBJ([strverscmp])
     AC_LIBOBJ([vasprintf])
-    AC_LIBOBJ([waitpid])
 
     for f in $funcs; do
       case "$f" in
@@ -630,9 +629,6 @@ if test -z "${setobjs}"; then
   AC_REPLACE_FUNCS($funcs)
   libiberty_AC_FUNC_C_ALLOCA
   AC_FUNC_FORK
-  if test $ac_cv_func_vfork_works = no; then
-    AC_LIBOBJ([vfork])
-  fi
   # We only need _doprnt if we might use it to implement v*printf.
   if test $ac_cv_func_vprintf != yes \
      || test $ac_cv_func_vfprintf != yes \
-- 
2.6.3

