--- webui-2.5.0-beta.3.orig/CMakeLists.txt	2026-04-02 21:47:54.938734800 +0300
+++ webui-2.5.0-beta.3/CMakeLists.txt	2026-04-02 21:48:32.685081300 +0300
@@ -31,6 +31,10 @@
 if(BUILD_SHARED_LIBS AND WIN32)
     target_compile_definitions(webui PRIVATE CIVETWEB_DLL_EXPORTS PUBLIC CIVETWEB_DLL_IMPORTS)
     target_link_libraries(webui PRIVATE ws2_32)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        find_package(Threads REQUIRED)
+        target_link_libraries(webui PRIVATE Threads::Threads)
+    endif()
 endif()
 
 if(WEBUI_USE_TLS)
