--- a/control-center/meson.build
+++ b/control-center/meson.build
@@ -67,7 +67,7 @@
 
 if windows
 	gda_control_center_cflags += ['-mwindows']
-	gda_control_center_res = windows_module.compile_resources(
+	gda_control_center_res = windows_module.compile_resources( 'control-center-res.rc',
 		depend_files: 'control-center-res.rc'
 		)
 	gda_control_center_sources += [ gda_control_center_res ]
--- a/tools/browser/meson.build
+++ b/tools/browser/meson.build
@@ -84,7 +84,7 @@
 
 subdir('data')
 
-i18n.merge_file('browser_desktop',
+i18n.merge_file(
 	po_dir: join_paths(meson.source_root(),'po'),
 	type: 'desktop',
 	input: 'org.gnome.gda.Browser.desktop.in',
