--- a/meson.build
+++ b/meson.build
@@ -246,7 +246,8 @@
 	'-DISO_CODES_PREFIX="'+iso_codes_prefix+'"'
 	]
 
-libcryp_config = find_program('libcryp-config', required : false)
+gcrypt_dep = dependency('libgcrypt')
+libcryp_config = find_program('libgcrypt-config', required : false)
 if libcryp_config.found()
   conf.set('HAVE_LIBCRYPT_CONFIG', 1)
 endif
--- a/libgda-ui/meson.build
+++ b/libgda-ui/meson.build
@@ -1,5 +1,6 @@
 libgda_ui_deps = [
 	libgda_dep,
+	gcrypt_dep,
 	gtk_dep,
 	gio_dep
 	]
