--- wxWidgets-3.2.9/build/cmake/toolkit.cmake.orig	2025-12-07 23:59:42.000000000 +0100
+++ wxWidgets-3.2.9/build/cmake/toolkit.cmake	2026-01-01 10:58:09.097069000 +0100
@@ -129,7 +129,7 @@
 endif()
 
 if(WXQT)
-    find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
+    find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core)
 
     if(QT_VERSION_MAJOR EQUAL 5)
         set(QT_COMPONENTS Core Widgets Gui OpenGL OpenGL Test)
