--- pugixml-1.8/CMakeLists.txt.orig	2018-01-26 07:45:48.648304000 +0300
+++ pugixml-1.8/CMakeLists.txt	2018-01-26 07:46:22.789039700 +0300
@@ -56,8 +56,8 @@
 endif()
 
 install(TARGETS pugixml EXPORT pugixml-config
-	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}${INSTALL_SUFFIX}
-	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}${INSTALL_SUFFIX}
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}${INSTALL_SUFFIX})
 install(EXPORT pugixml-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pugixml)
--- pugixml-1.8/scripts/pugixml.pc.in.orig	2018-01-26 07:52:21.040361600 +0300
+++ pugixml-1.8/scripts/pugixml.pc.in	2018-01-26 07:52:28.807192500 +0300
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 includedir=${prefix}/include/pugixml-@PUGIXML_VERSION_STRING@
-libdir=${exec_prefix}/lib/pugixml-@PUGIXML_VERSION_STRING@
+libdir=${exec_prefix}/lib
 
 Name: pugixml
 Description: Light-weight, simple and fast XML parser for C++ with XPath support.
