--- pugixml-1.7-orig/scripts/CMakeLists.txt	2016-08-18 06:30:37.713623400 +0200
+++ pugixml-1.7/scripts/CMakeLists.txt	2016-08-21 21:02:03.684491700 +0200
@@ -33,14 +33,15 @@
 endif()
 
 set_target_properties(pugixml PROPERTIES VERSION 1.7 SOVERSION 1)
+get_target_property(vernum pugixml VERSION)
 
 install(TARGETS pugixml EXPORT pugixml-config 
-	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/pugixml-${vernum}
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/pugixml-${vernum}
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
 
-install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/pugixml-${vernum})
 install(EXPORT pugixml-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pugixml)
 
 if(BUILD_TESTS)
