--- gavl-1.4.0/gavl/cputest.c	2017-10-29 22:28:19.871203800 -0400
+++ gavl-1.4.0/gavl/cputest.c	2017-10-29 22:28:31.921218800 -0400
@@ -69,6 +69,8 @@
      int rval = 0;
     int eax, ebx, ecx, edx;
     int max_std_level, max_ext_level, std_caps=0, ext_caps=0;
+    
+#ifndef ARCH_X86_64
     long a, c;
 
     __asm__ __volatile__ (
@@ -94,6 +96,7 @@
 
     if (a == c)
         return 0; /* CPUID not supported */
+#endif // !ARCH_X86_64
 
     cpuid(0, max_std_level, ebx, ecx, edx);
 
