--- cmake-4.3.0.orig/Source/QtDialog/CMakeLists.txt	2026-03-17 16:29:16.000000000 +0300
+++ cmake-4.3.0/Source/QtDialog/CMakeLists.txt	2026-03-22 11:58:12.738304400 +0300
@@ -67,7 +67,7 @@
 endif()
 
 # We need to install platform plugin and add qt.conf for Qt5 on Mac and Windows.
-if(CMake_INSTALL_DEPENDENCIES AND (APPLE OR WIN32) AND NOT COMMAND qt_generate_deploy_app_script)
+if(CMake_INSTALL_DEPENDENCIES AND (APPLE OR WIN32) AND NOT MINGW AND NOT COMMAND qt_generate_deploy_app_script)
   function(_qt_get_plugin_name_with_version target out_var)
       string(REGEX REPLACE "^Qt::(.+)" "Qt${CMake_QT_MAJOR_VERSION}::\\1"
              qt_plugin_with_version "${target}")
@@ -329,7 +329,7 @@
   " ${COMPONENT})
 endif()
 
-if(CMake_INSTALL_DEPENDENCIES AND (APPLE OR WIN32))
+if(CMake_INSTALL_DEPENDENCIES AND (APPLE OR WIN32) AND NOT MINGW)
   if(COMMAND qt_generate_deploy_app_script)
     # Qt libraries are found via @rpath, so point the runtime path at them.
     set_property(TARGET cmake-gui PROPERTY INSTALL_RPATH "@executable_path/../Frameworks")
