diff --git a/Modules/FindPython/Support.cmake b/Modules/FindPython/Support.cmake
index 6029be0e2a..1f0496103e 100644
--- a/Modules/FindPython/Support.cmake
+++ b/Modules/FindPython/Support.cmake
@@ -1155,7 +1155,7 @@ function (_PYTHON_VALIDATE_INTERPRETER)
 
     if (WIN32)
       # In this case, check if the interpreter is compatible with the target processor architecture
-      if (_$_PYTHON_PREFIX}_ARCHITECTURE_ID MATCHES "ARM")
+      if (_${_PYTHON_PREFIX}_ARCHITECTURE_ID MATCHES "ARM|arm|AARCH64|aarch64")
         set(target_arm TRUE)
       else()
         set(target_arm FALSE)
@@ -1167,8 +1167,8 @@ function (_PYTHON_VALIDATE_INTERPRETER)
         ERROR_QUIET
         OUTPUT_STRIP_TRAILING_WHITESPACE)
       string(TOUPPER "${platform}" platform)
-      if (result OR ((target_arm AND NOT platform MATCHES "ARM") OR
-                     (NOT target_arm AND platform MATCHES "ARM")))
+      if (result OR ((target_arm AND NOT platform MATCHES "ARM|AARCH64") OR
+                     (NOT target_arm AND platform MATCHES "ARM|AARCH64")))
         # interpreter not usable or has wrong architecture
         if (result)
           set_property (CACHE _${_PYTHON_PREFIX}_Interpreter_REASON_FAILURE PROPERTY VALUE "Cannot use the interpreter \"${_${_PYTHON_PREFIX}_EXECUTABLE}\"")
