--- cmake-3.22.0/Source/cmTimestamp.cxx.orig	2021-11-21 19:49:45.959000000 +0100
+++ cmake-3.22.0/Source/cmTimestamp.cxx	2021-11-21 19:49:59.378057000 +0100
@@ -194,7 +194,7 @@
 
   char buffer[16];
 
-#ifdef __MINGW32__
+#if defined(__MINGW32__) && !defined(_UCRT)
   /* See a bug in MinGW: https://sourceforge.net/p/mingw-w64/bugs/793/. A work
    * around is to try to use strftime() from ucrtbase.dll. */
   using T = size_t(WINAPI*)(char*, size_t, const char*, const struct tm*);
