--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,13 +77,7 @@
     include_directories(${Gflags_INCLUDE_PATH})
   endif()
 
-  find_package(Glog REQUIRED)
-  if(Glog_FOUND)
-    include_directories(${Glog_INCLUDE_PATH})
-  endif()
-  if(Glog_STATIC)
-    add_definitions(-DGOOGLE_GLOG_DLL_DECL=)
-  endif()
+  find_package(glog REQUIRED CONFIG)
 
   set(RIME_ENABLE_LOGGING 1)
 
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -41,7 +41,7 @@
 
 set(rime_core_deps
     ${Boost_LIBRARIES}
-    ${Glog_LIBRARY}
+    glog::glog
     ${YamlCpp_LIBRARY}
     ${CMAKE_THREAD_LIBS_INIT}
     ${rime_optional_deps})
--- a/src/rime/common.h
+++ b/src/rime/common.h
@@ -26,7 +26,6 @@
 #include <boost/signals2/signal.hpp>
 
 #ifdef RIME_ENABLE_LOGGING
-#define GLOG_NO_ABBREVIATED_SEVERITIES
 #include <glog/logging.h>
 #else
 #include "no_logging.h"
--- a/src/rime/lever/deployment_tasks.cc
+++ b/src/rime/lever/deployment_tasks.cc
@@ -633,7 +633,7 @@
   // which causes permission issue on Android
   // https://github.com/google/glog/blob/b58718f37cf58fa17f48bf1d576974d133d89839/src/logging.cc#L2410
   if (FLAGS_log_dir.empty()) {
-    google::GetExistingTempDirectories(&dirs);
+    dirs = google::GetLoggingDirectories();
   } else {
     dirs.push_back(FLAGS_log_dir);
   }
