--- ./Makefile.am.orig	2015-02-17 21:26:00.851063100 +0100
+++ ./Makefile.am	2015-02-17 22:01:53.498641500 +0100
@@ -130,7 +130,7 @@
 	g-ir-compiler --includedir=$(BUILD_DIR) -o $@ $<
 
 GExiv2-$(GIR_VERSION).gir: lib@PACKAGE_NAME@.la
-	g-ir-scanner --verbose                          \
+	CC="$(CC)" g-ir-scanner --verbose                          \
 		--namespace=GExiv2                      \
 		--nsversion=$(GIR_VERSION)              \
 		--symbol-prefix=@PACKAGE_NAME@              \
@@ -141,6 +141,6 @@
 		-I @PACKAGE_NAME@                           \
 		$(GEXIV2_public_HEADERS)      \
 		--library lib@PACKAGE_NAME@.la                \
-		$(LIBTOOL)                              \
+		--libtool `which libtool`                              \
 		--pkg=gobject-2.0
 
