--- ./configure.ac.orig	2014-09-17 00:09:05.000000000 +0200
+++ ./configure.ac	2015-02-21 21:20:44.406810300 +0100
@@ -122,7 +122,7 @@
 if test "x$enable_introspection" = "xyes"; then
     # Used to determine where Python wants the .gi file to be installed ...
     # if Python is not installed, PYTHON2 and PYTHON3 will be empty
-    PYTHON="import gi; print(gi._overridesdir)"
+    PYTHON="import gi; from os.path import abspath; print(abspath(gi._overridesdir))"
     PYTHON2_GIDIR=$(python2 -c "$PYTHON")
     AC_MSG_RESULT([Python2 GObject Introspection directory: $PYTHON2_GIDIR])
     PYTHON3_GIDIR=$(python3 -c "$PYTHON")
--- ./Makefile.am.orig	2014-09-17 00:07:53.000000000 +0200
+++ ./Makefile.am	2015-02-21 21:21:24.598425100 +0100
@@ -4,9 +4,9 @@
 pkgconfigdir     = $(libdir)/pkgconfig
 GEXIV2_publicdir = $(includedir)/gexiv2
 girdir           = $(datadir)/gir-1.0
-typelibdir       = `pkg-config gobject-introspection-no-export-1.0 --variable typelibdir`
-py2gidir         = $(PYTHON2_GIDIR)
-py3gidir         = $(PYTHON3_GIDIR)
+typelibdir       = `pkg-config gobject-introspection-no-export-1.0 --variable typelibdir  --dont-define-prefix`
+py2gidir         = `cygpath -u $(PYTHON2_GIDIR)`
+py3gidir         = `cygpath -u $(PYTHON3_GIDIR)`
 vapidir          = $(datadir)/vala/vapi
 
 # Extra package files. #
@@ -130,7 +130,7 @@
 	g-ir-compiler --includedir=$(BUILD_DIR) -o $@ $<
 
 GExiv2-$(GIR_VERSION).gir: lib@PACKAGE_NAME@.la
-	g-ir-scanner --verbose                          \
+	CC="$(CC)" g-ir-scanner --verbose                          \
 		--namespace=GExiv2                      \
 		--nsversion=$(GIR_VERSION)              \
 		--symbol-prefix=@PACKAGE_NAME@              \
@@ -141,6 +141,6 @@
 		-I @PACKAGE_NAME@                           \
 		$(GEXIV2_public_HEADERS)      \
 		--library lib@PACKAGE_NAME@.la                \
-		$(LIBTOOL)                              \
+		--libtool `which libtool`                              \
 		--pkg=gobject-2.0
 
