--- a/src/platform/platformtheme.h
+++ b/src/platform/platformtheme.h
@@ -374,7 +374,7 @@
 // of these events.
 
 template<typename T>
-class KIRIGAMIPLATFORM_EXPORT PropertyChangedEvent : public QEvent
+class PropertyChangedEvent : public QEvent
 {
 public:
     PropertyChangedEvent(PlatformTheme *theme, const T &previous, const T &current)
@@ -389,7 +389,7 @@
     T oldValue;
     T newValue;
 
-    static QEvent::Type type;
+    static KIRIGAMIPLATFORM_EXPORT QEvent::Type type;
 };
 
 using DataChangedEvent = PropertyChangedEvent<std::shared_ptr<PlatformThemeData>>;
