--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -192,7 +192,7 @@
   set(QCA_PLUGINS_INSTALL_DIR "${LIB_INSTALL_DIR}/${QCA_LIB_NAME}" CACHE PATH "Directory where qca plugins will install")
   set(QCA_BINARY_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Directory where qca plugins will install")
   set(QCA_LIBRARY_INSTALL_DIR "${LIB_INSTALL_DIR}" CACHE PATH "Directory where qca library will install")
-  set(QCA_FEATURE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/mkspecs/features" CACHE PATH "Directory where qca feature file will install")
+  set(QCA_FEATURE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/qt5/mkspecs/features" CACHE PATH "Directory where qca feature file will install")
   if(NOT QCA_SUFFIX)
     set(QCA_INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "Directory where qca public headers will install")
     set(QCA_PRIVATE_INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "Directory where qca headers will install")
--- a/cmake/modules/QcaMacro.cmake
+++ b/cmake/modules/QcaMacro.cmake
@@ -13,6 +13,18 @@
   EXEC_PROGRAM( ${QMAKE_EXECUTABLE} ARGS "-query QT_INSTALL_DOCS" OUTPUT_VARIABLE QT_DOC_DIR )
   EXEC_PROGRAM( ${QMAKE_EXECUTABLE} ARGS "-query QT_INSTALL_DATA" OUTPUT_VARIABLE QT_DATA_DIR )
   EXEC_PROGRAM( ${QMAKE_EXECUTABLE} ARGS "-query QT_HOST_DATA" OUTPUT_VARIABLE QT_ARCHDATA_DIR )
+  
+  if(MINGW)
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_LIBRARY_DIR}" OUTPUT_VARIABLE QT_LIBRARY_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_PREFIX_DIR}" OUTPUT_VARIABLE QT_PREFIX_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_PLUGINS_DIR}" OUTPUT_VARIABLE QT_PLUGINS_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_BINARY_DIR}" OUTPUT_VARIABLE QT_BINARY_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_HEADERS_DIR}" OUTPUT_VARIABLE QT_HEADERS_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_DOC_DIR}" OUTPUT_VARIABLE QT_DOC_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_DATA_DIR}" OUTPUT_VARIABLE QT_DATA_DIR )
+    EXEC_PROGRAM( "cygpath" ARGS "-u ${QT_ARCHDATA_DIR}" OUTPUT_VARIABLE QT_ARCHDATA_DIR )
+  endif()
+  
   SET( QT_MKSPECS_DIR "${QT_ARCHDATA_DIR}/mkspecs" )
 ENDMACRO(SETUP_QT5_DIRS)
 
