--- mpfr-4.2.0-orig/configure.ac	2023-01-06 18:55:57.000000000 +0800
+++ mpfr-4.2.0/configure.ac	2023-01-20 01:30:14.536869200 +0800
@@ -630,9 +630,7 @@
 # error "Dead man"
 error
 #endif
-     ]], [[]])],[AC_MSG_RESULT(DLL)],[
-  AC_MSG_RESULT(static)
-  AC_MSG_ERROR([libgmp isn't provided as a DLL: use --enable-static --disable-shared]) ])
+     ]], [[]])],[AC_MSG_RESULT(DLL)])
    else
      AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "gmp.h"
@@ -640,9 +638,7 @@
 # error "Dead man"
 error
 #endif
-     ]], [[]])],[AC_MSG_RESULT(static)],[
-  AC_MSG_RESULT(DLL)
-  AC_MSG_ERROR([libgmp is provided as a DLL: use --disable-static --enable-shared]) ])
+     ]], [[]])],[AC_MSG_RESULT(static)])
   fi
   ;;
 esac
