--- a/libtransmission/platform.cc
+++ b/libtransmission/platform.cc
@@ -280,7 +280,8 @@
     auto const module_path = tr_win32_native_to_utf8({ std::data(wide_module_path) });
     if (auto const dir = tr_sys_path_dirname(module_path); !std::empty(dir))
     {
-        if (auto const path = tr_pathbuf{ dir, "/public_html"sv }; isWebClientDir(path))
+        auto const prefix_dir = dir.substr(0, dir.rfind("\\bin"));
+        if (auto const path = tr_pathbuf{ prefix_dir, "/share/transmission/public_html"sv }; isWebClientDir(path))
         {
             return std::string{ path };
         }
