--- a/configure.ac
+++ b/configure.ac
@@ -583,7 +583,7 @@
      CXXFLAGS="$CXXFLAGS -mms-bitfields -mwin32 -mwindows"
      CPPFLAGS="$CPPFLAGS -DWINVER=0x0600 -D_WIN32_WINNT=0x0600 -DWIN32_LEAN_AND_MEAN"
      # CPPFLAGS="$CPPFLAGS -D__USE_MINGW_ANSI_STDIO=1 -D__STDC_FORMAT_MACROS=1"
-     LIBS="$LIBS -liphlpapi -lshell32 -lws2_32"
+     LIBS="$LIBS -liphlpapi -lshell32 -lws2_32 -lole32 -lshlwapi -lcrypt32 -luuid"
      transmissionlocaledir="locale"
      if test -z "$host_alias"; then
        hostaliaswindres=
--- a/gtk/tr-prefs.c
+++ b/gtk/tr-prefs.c
@@ -8,6 +8,7 @@
 
 #include <ctype.h> /* isspace */
 #include <limits.h> /* USHRT_MAX, INT_MAX */
+#include <winsock2.h>
 #include <unistd.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
--- a/qt/qtr.pro
+++ b/qt/qtr.pro
@@ -7,7 +7,7 @@
 target.path = /bin
 INSTALLS += target
 
-unix: INSTALLS += man
+INSTALLS += man
 man.path = /share/man/man1/
 man.files = transmission-qt.1
 
@@ -36,7 +36,7 @@
 LIBS += $${LIBUPNP_LIBS}
 LIBS += $${LIBNATPMP_LIBS}
 unix: LIBS += -L$${EVENT_TOP}/lib -lz -lrt
-win32:LIBS += -levent-2.0 -lws2_32 -lintl
+win32:LIBS += -levent -lws2_32 -lintl -lole32 -lshlwapi -lcrypt32 -luuid -lz
 win32:LIBS += -lidn -liconv -lwldap32 -liphlpapi
 
 TRANSLATIONS += translations/transmission_de.ts \
