--- transmission-4.1.0/qt/Application.cc.orig	2026-01-30 10:57:05.557996000 +0100
+++ transmission-4.1.0/qt/Application.cc	2026-01-30 10:57:01.642891100 +0100
@@ -16,6 +16,10 @@
 #include <QSystemTrayIcon>
 
 #ifdef QT_DBUS_LIB
+#undef QT_DBUS_LIB
+#endif
+
+#ifdef QT_DBUS_LIB
 #include <QDBusConnection>
 #include <QDBusMessage>
 #include <QDBusReply>
--- transmission-4.1.0/CMakeLists.txt.orig	2026-01-27 00:59:25.000000000 +0100
+++ transmission-4.1.0/CMakeLists.txt	2026-01-30 10:52:00.298256200 +0100
@@ -440,7 +440,7 @@
         endforeach()
     endif()
 
-    if(QT_TARGETS)
+    if(QT_TARGETS AND NOT MINGW)
         foreach(M ${QT_OPTIONAL_MODULES})
             find_package(Qt${Qt_VERSION_MAJOR}${M} ${QT_MINIMUM} QUIET)
             if(Qt${Qt_VERSION_MAJOR}${M}_FOUND)
@@ -461,7 +461,7 @@
     endif()
 
     set(QT_FOUND ON)
-    if(NOT QT_TARGETS OR NOT (ENABLE_QT_COM_INTEROP OR ENABLE_QT_DBUS_INTEROP))
+    if(NOT QT_TARGETS OR NOT (ENABLE_QT_COM_INTEROP OR ENABLE_QT_DBUS_INTEROP OR MINGW))
         if(QT_IS_REQUIRED)
             message(FATAL_ERROR "Unable to find required Qt libraries (Qt${Qt_VERSION_MAJOR}${MISSING_QT_MODULE})")
         endif()
