From 93827a00fc926d951c75f08fdb1d491912ff7065 Mon Sep 17 00:00:00 2001
From: Denis Demidov <dennis.demidov@gmail.com>
Date: Wed, 27 Aug 2025 16:52:10 +0300
Subject: [PATCH] close(#289): update minimum cmake version

---
 CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b9999932..5917c0c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Build type")
     message(STATUS "No build type selected, default to ${CMAKE_BUILD_TYPE}")
@@ -6,10 +6,6 @@ endif()
 
 project(AMGCL)
 
-if (NOT (CMAKE_VERSION LESS 3.3))
-    cmake_policy(SET CMP0058 OLD)
-endif()
-
 set(AMGCL_MASTER_PROJECT OFF)
 if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
     set(AMGCL_MASTER_PROJECT ON)
