diff -Naur Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0-orig/CMakeLists.txt Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0/CMakeLists.txt
--- Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0-orig/CMakeLists.txt	2017-06-29 09:45:41.898236900 +0300
+++ Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0/CMakeLists.txt	2017-06-29 09:51:53.242895700 +0300
@@ -50,6 +50,10 @@
     # overwriting of LVL artifacts when installing these downstream repos.
     option(INSTALL_LVL_FILES "Install content from LoaderAndValidationLayers repo" ON)
 endif()
+
+if(MINGW)
+    option(INSTALL_LVL_FILES "Install content from LoaderAndValidationLayers repo" ON)
+endif()
 
 set(SCRIPTS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/scripts")
 
@@ -325,7 +325,11 @@
     endif()
 endif()
 
-if(UNIX)
+if(MINGW)
+    include(GNUInstallDirs)
+endif()
+
+if(NOT MSVC)
     if(INSTALL_LVL_FILES)
         install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/vulkan" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
     endif()
diff -Naur Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0-orig/layers/CMakeLists.txt Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0/layers/CMakeLists.txt
--- Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0-orig/layers/CMakeLists.txt	2017-06-29 09:45:41.898236900 +0300
+++ Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0/layers/CMakeLists.txt	2017-06-29 09:57:48.711421600 +0300
@@ -104,6 +104,9 @@
     add_dependencies(VkLayer_${target} generate_helper_files)
     target_link_Libraries(VkLayer_${target} VkLayer_utils)
     add_dependencies(VkLayer_${target} generate_helper_files VkLayer_utils)
+    if(MINGW)
+      install(TARGETS VkLayer_${target} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    endif()
     endmacro()
 else()
     macro(add_vk_layer target)
@@ -157,6 +160,7 @@
     add_library(VkLayer_utils STATIC vk_layer_config.cpp vk_layer_extension_utils.cpp vk_layer_utils.cpp vk_format_utils.cpp)
     if(MINGW)
         target_link_libraries(VkLayer_utils shlwapi)
+        install(TARGETS VkLayer_utils RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif()
 else()
     add_library(VkLayer_utils SHARED vk_layer_config.cpp vk_layer_extension_utils.cpp vk_layer_utils.cpp vk_format_utils.cpp)
diff -Naur Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0-orig/loader/CMakeLists.txt Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0/loader/CMakeLists.txt
--- Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0-orig/loader/CMakeLists.txt	2017-06-29 10:00:31.117255000 +0300
+++ Vulkan-LoaderAndValidationLayers-sdk-1.0.46.0/loader/CMakeLists.txt	2017-06-29 10:08:16.935379100 +0300
@@ -117,7 +117,15 @@
     add_library(${API_LOWERCASE}-${MAJOR} SHARED $<TARGET_OBJECTS:loader-opt> $<TARGET_OBJECTS:loader-norm> ${CMAKE_CURRENT_BINARY_DIR}/${API_LOWERCASE}-${MAJOR}${DEFFILE_SUFFIX}.def ${CMAKE_CURRENT_SOURCE_DIR}/loader.rc)
     add_library(VKstatic.${MAJOR} STATIC $<TARGET_OBJECTS:loader-opt> $<TARGET_OBJECTS:loader-norm>)
     # Suppress conflicting libs warning for debug builds.
-    set_target_properties(${API_LOWERCASE}-${MAJOR} PROPERTIES LINK_FLAGS_DEBUG /ignore:4098)
+    if(MINGW)
+      set_target_properties(${API_LOWERCASE}-${MAJOR}
+                            PROPERTIES
+                            LIBRARY_OUTPUT_NAME ${API_LOWERCASE}
+                            RUNTIME_OUTPUT_NAME ${API_LOWERCASE}-${MAJOR}
+                            ARCHIVE_OUTPUT_NAME ${API_LOWERCASE})
+    else()
+      set_target_properties(${API_LOWERCASE}-${MAJOR} PROPERTIES LINK_FLAGS_DEBUG /ignore:4098)
+    endif()
     set_target_properties(VKstatic.${MAJOR} PROPERTIES OUTPUT_NAME VKstatic.${MAJOR})
     target_link_libraries(${API_LOWERCASE}-${MAJOR} shlwapi)
     add_dependencies(${API_LOWERCASE}-${MAJOR} generate_helper_files loader_gen_files)
@@ -132,10 +139,14 @@
         file(TO_NATIVE_PATH ${CMAKE_CURRENT_BINARY_DIR}/../demos/ COPY_DST_PATH)
         file(TO_NATIVE_PATH ${CMAKE_CURRENT_BINARY_DIR}/../tests/ COPY_DST_TEST_PATH)
     endif()
-    add_custom_command(TARGET ${API_LOWERCASE}-${MAJOR} POST_BUILD
-      COMMAND xcopy /Y /I ${COPY_SRC_PATH} ${COPY_DST_PATH})
-    add_custom_command(TARGET ${API_LOWERCASE}-${MAJOR} POST_BUILD
-      COMMAND xcopy /Y /I ${COPY_SRC_PATH} ${COPY_DST_TEST_PATH})
+    if(MINGW)
+      install(TARGETS ${API_LOWERCASE}-${MAJOR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    else()
+      add_custom_command(TARGET ${API_LOWERCASE}-${MAJOR} POST_BUILD
+        COMMAND xcopy /Y /I ${COPY_SRC_PATH} ${COPY_DST_PATH})
+      add_custom_command(TARGET ${API_LOWERCASE}-${MAJOR} POST_BUILD
+        COMMAND xcopy /Y /I ${COPY_SRC_PATH} ${COPY_DST_TEST_PATH})
+    endif()
 
 else()
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wpointer-arith")
