--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@
 find_package(SoQt CONFIG)
 
 if (SoQt_FOUND)
-    find_package(Qt5 COMPONENTS Core Widgets Gui REQUIRED)
+    find_package(Qt6 COMPONENTS Core Widgets Gui REQUIRED)
 endif()
 
 find_package(Python REQUIRED COMPONENTS Interpreter Development)
--- a/distutils_cmake/CMakeLists.txt
+++ b/distutils_cmake/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.5)
-project(pivy_cmake_setup NONE)
+project(pivy_cmake_setup CXX)
 
 
 find_package(Coin CONFIG REQUIRED)
--- a/qtinfo.py
+++ b/qtinfo.py
@@ -7,7 +7,7 @@
         if qmake_command:
             self._qmake_command = qmake_command
         else:
-            self._qmake_command = [find_executable("qmake"),]
+            self._qmake_command = [find_executable("qmake6"),]
         self._dict = {}
         # bind all variables early at __init__ time.
         for thing in self.__class__.__dict__:
--- a/setup.py
+++ b/setup.py
@@ -573,7 +573,7 @@
                     else:
                         if "MSYSTEM" in os.environ:
                             CPP_FLAGS += ' -DSOQT_DLL'
-                        LDFLAGS_LIBS += ' -lSoQt'
+                        LDFLAGS_LIBS += ' -lSoQt -lQt6Core'
                 
                 if module == "coin":
                     if sys.platform == 'win32' and "MSYSTEM" not in os.environ:
