--- pivy-0.6.11.orig/interfaces/CMakeLists.txt	2026-03-27 08:54:33.000000000 +0300
+++ pivy-0.6.11/interfaces/CMakeLists.txt	2026-03-28 18:18:24.273011500 +0300
@@ -32,6 +32,8 @@
 
 if (MSVC)
     set_target_properties(coin PROPERTIES COMPILE_FLAGS "/bigobj")
+elseif (MINGW)
+    set_target_properties(coin PROPERTIES COMPILE_FLAGS "-Wa,-mbig-obj")
 endif ()
 
 target_include_directories(coin
@@ -71,6 +73,8 @@
 
     if (MSVC)
         set_target_properties(coin PROPERTIES COMPILE_FLAGS "/bigobj")
+    elseif (MINGW)
+        set_target_properties(coin PROPERTIES COMPILE_FLAGS "-Wa,-mbig-obj")
     endif ()
 
     if (PIVY_USE_QT6)
