diff -Naur openshadinglanguage-orig/src/liboslexec/llvm_ops.cpp openshadinglanguage/src/liboslexec/llvm_ops.cpp
--- openshadinglanguage-orig/src/liboslexec/llvm_ops.cpp	2014-08-31 22:43:33.401600000 +0400
+++ openshadinglanguage/src/liboslexec/llvm_ops.cpp	2014-08-31 23:11:06.558600000 +0400
@@ -119,6 +119,12 @@
 using OIIO::logbf;
 using OIIO::exp2f;
 #endif
+
+#ifdef __MINGW64_VERSION_MAJOR
+using OIIO::isnan;
+using OIIO::isinf;
+using OIIO::isfinite;
+#endif
 
 #if defined(__FreeBSD__)
 #include <sys/param.h>
diff -Naur openshadinglanguage-orig/src/liboslexec/shadingsys.cpp openshadinglanguage/src/liboslexec/shadingsys.cpp
--- openshadinglanguage-orig/src/liboslexec/shadingsys.cpp	2014-08-31 21:12:36.587200000 +0400
+++ openshadinglanguage/src/liboslexec/shadingsys.cpp	2014-08-31 21:31:52.245600000 +0400
@@ -47,6 +47,10 @@
 #include <OpenImageIO/filesystem.h>
 #include <OpenImageIO/optparser.h>
 #include <OpenImageIO/fmath.h>
+
+#ifdef __MINGW64_VERSION_MAJOR
+using OIIO::isfinite;
+#endif
 
 using namespace OSL;
 using namespace OSL::pvt;
