--- OpenShadingLanguage-Release-1.7.3/src/doc/CMakeLists.txt.orig	2016-07-12 10:29:16.503595500 +0300
+++ OpenShadingLanguage-Release-1.7.3/src/doc/CMakeLists.txt	2016-07-12 10:30:10.043595500 +0300
@@ -1,11 +1,11 @@
 set (public_docs osl-languagespec.pdf)
 # Some day: CLA-INDIVIDUAL CLA-CORPORATE
 
-install (FILES ${public_docs} DESTINATION doc COMPONENT documentation)
+install (FILES ${public_docs} DESTINATION share/OSL/doc COMPONENT documentation)
 
 install ( FILES "${PROJECT_SOURCE_DIR}/LICENSE"
                 "${PROJECT_SOURCE_DIR}/INSTALL"
                 "${PROJECT_SOURCE_DIR}/CHANGES"
                 "${PROJECT_SOURCE_DIR}/README.md"
-          DESTINATION . )
+          DESTINATION share/OSL )
 
