--- OpenShadingLanguage-Release-1.8.9/src/doc/CMakeLists.txt.orig	2017-06-19 08:47:39.155244200 +0300
+++ OpenShadingLanguage-Release-1.8.9/src/doc/CMakeLists.txt	2017-06-19 08:49:15.806424100 +0300
@@ -1,10 +1,10 @@
 set (public_docs osl-languagespec.pdf)
 # Some day: CLA-INDIVIDUAL CLA-CORPORATE
 
-install (FILES ${public_docs} DESTINATION doc COMPONENT documentation)
+install (FILES ${public_docs} DESTINATION share/OSL/doc COMPONENT documentation)
 
 install ( FILES "${PROJECT_SOURCE_DIR}/LICENSE"
                 "${PROJECT_SOURCE_DIR}/INSTALL.md"
                 "${PROJECT_SOURCE_DIR}/CHANGES.md"
                 "${PROJECT_SOURCE_DIR}/README.md"
-          DESTINATION . )
+          DESTINATION share/OSL )
