--- OpenShadingLanguage-Release-1.10.6/src/liboslcomp/osllex.l.orig	2019-07-31 13:40:13.522982700 +0300
+++ OpenShadingLanguage-Release-1.10.6/src/liboslcomp/osllex.l	2019-07-31 13:40:31.481640300 +0300
@@ -112,7 +112,7 @@
 #ifdef _WIN32
 #define YY_NO_UNISTD_H
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define strtoll _strtoi64
 #include <io.h>
 #define isatty _isatty
--- OpenShadingLanguage-Release-1.10.6/src/liboslexec/osolex.l.orig	2019-07-31 13:40:13.522982700 +0300
+++ OpenShadingLanguage-Release-1.10.6/src/liboslexec/osolex.l	2019-07-31 13:40:31.481640300 +0300
@@ -112,7 +112,7 @@
 #define YY_NO_UNISTD_H
 #endif
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 #include <io.h>
 #define isatty _isatty
 #endif