--- a/src/liboslexec/pointcloud.cpp
+++ b/src/liboslexec/pointcloud.cpp
@@ -114,8 +114,6 @@
             return 0;  // No "position" attribute -- fail
     }
 
-    static_assert(sizeof(size_t) == sizeof(Partio::ParticleIndex),
-                  "Partio ParticleIndex should be the size of a size_t");
     // FIXME -- if anybody cares about an architecture in which that is not
     // the case, we can easily allocate local space to retrieve the indices,
     // then copy them back to the caller's indices.
@@ -246,8 +244,6 @@
         count = maxn;
     }
 
-    static_assert(sizeof(size_t) == sizeof(Partio::ParticleIndex),
-                  "Partio ParticleIndex should be the size of a size_t");
     // FIXME -- if anybody cares about an architecture in which that is not
     // the case, we can easily allocate local space to retrieve the indices,
     // then copy them back to the caller's indices.
