--- OpenShadingLanguage-Release-1.11.8.0/src/testshade/CMakeLists.txt.orig	2020-06-10 10:46:40.209699400 +0300
+++ OpenShadingLanguage-Release-1.11.8.0/src/testshade/CMakeLists.txt	2020-06-10 10:48:16.731089800 +0300
@@ -56,7 +56,9 @@
                        PROPERTIES
                        VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
                        SOVERSION ${SOVERSION}
-                       OUTPUT_NAME libtestshade${OSL_LIBNAME_SUFFIX}
+                       OUTPUT_NAME testshade
+                       RUNTIME_OUTPUT_NAME libtestshade${OSL_LIBNAME_SUFFIX}
+                       ARCHIVE_OUTPUT_NAME testshade
                        )
 
 target_link_libraries (libtestshade
