diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 8016d8e5..03e38977 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -58,7 +58,7 @@ checked_find_package (pugixml REQUIRED
 # LLVM library setup
 checked_find_package (LLVM REQUIRED
                       VERSION_MIN 11.0
-                      VERSION_MAX 19.9
+                      VERSION_MAX 21.9
                       PRINT LLVM_SYSTEM_LIBRARIES CLANG_LIBRARIES
                             LLVM_SHARED_MODE)
 # ensure include directory is added (in case of non-standard locations
diff --git a/src/include/OSL/platform.h b/src/include/OSL/platform.h
index 6a94bdc4..e4f00a43 100644
--- a/src/include/OSL/platform.h
+++ b/src/include/OSL/platform.h
@@ -20,6 +20,7 @@
 
 #include <cstring>
 #include <memory>
+#include <cstdint>
 
 #include <OSL/oslversion.h>
 
diff --git a/src/liboslexec/llvm_util.cpp b/src/liboslexec/llvm_util.cpp
index df23c38d..0490eb2d 100644
--- a/src/liboslexec/llvm_util.cpp
+++ b/src/liboslexec/llvm_util.cpp
@@ -3212,7 +3212,11 @@ LLVM_Util::loop_after_block() const
 llvm::Type*
 LLVM_Util::type_union(cspan<llvm::Type*> types)
 {
+#if OSL_LLVM_VERSION >= 200
+    llvm::DataLayout target(module()->getDataLayout());
+#else
     llvm::DataLayout target(module());
+#endif
     size_t max_size  = 0;
     size_t max_align = 1;
     for (auto t : types) {
