--- OpenShadingLanguage-Release-1.11.8.0/src/liboslexec/shadingsys.cpp.orig	2020-10-08 11:06:00.929372700 +0300
+++ OpenShadingLanguage-Release-1.11.8.0/src/liboslexec/shadingsys.cpp	2020-10-08 11:23:40.846943100 +0300
@@ -3400,8 +3402,8 @@
 
     bool ok = true;
     std::string groupfilename = tmpdir + "/shadergroup";
-    std::ofstream groupfile;
-    OIIO::Filesystem::open(groupfile, groupfilename);
+    OIIO::ofstream groupfile;
+    OIIO::Filesystem::open(groupfile,  OIIO::string_view(groupfilename).c_str());
     if (groupfile.good()) {
         groupfile << group.serialize();
         groupfile.close ();
