--- json-c-json-c-0.14/CMakeLists.txt.orig	2020-04-27 16:11:51.071245600 +0300
+++ json-c-json-c-0.14/CMakeLists.txt	2020-04-27 16:12:26.757263100 +0300
@@ -309,6 +309,7 @@
     ${PROJECT_SOURCE_DIR}/json_inttypes.h
     ${PROJECT_SOURCE_DIR}/json_object.h
     ${PROJECT_SOURCE_DIR}/json_object_iterator.h
+    ${PROJECT_SOURCE_DIR}/json_object_private.h
     ${PROJECT_SOURCE_DIR}/json_pointer.h
     ${PROJECT_SOURCE_DIR}/json_tokener.h
     ${PROJECT_SOURCE_DIR}/json_types.h
@@ -320,7 +321,6 @@
 
 set(JSON_C_HEADERS
     ${JSON_C_PUBLIC_HEADERS}
-    ${PROJECT_SOURCE_DIR}/json_object_private.h
     ${PROJECT_SOURCE_DIR}/random_seed.h
     ${PROJECT_SOURCE_DIR}/strerror_override.h
     ${PROJECT_SOURCE_DIR}/strerror_override_private.h
