--- poetry-1.1.13/poetry/console/__init__.py.orig	2022-02-10 22:18:00.000000000 +0100
+++ poetry-1.1.13/poetry/console/__init__.py	2022-06-21 09:15:44.687097500 +0200
@@ -1,3 +1,8 @@
+import sys
+import os
+vendor = os.path.normpath(os.path.join(os.path.realpath(__file__), "../..", "_vendor"))
+sys.path.insert(0, vendor)
+
 from .application import Application
 
 
--- poetry-1.1.13/pyproject.toml.orig	2022-02-10 22:18:00.000000000 +0100
+++ poetry-1.1.13/pyproject.toml	2022-06-21 09:49:09.948110100 +0200
@@ -25,43 +25,7 @@
 python = "~2.7 || ^3.5"
 
 poetry-core = "~1.0.7"
-cleo = "^0.8.1"
-clikit = "^0.6.2"
-crashtest = { version = "^0.3.0", python = "^3.6" }
-requests = "^2.18"
-cachy = "^0.3.0"
-requests-toolbelt = "^0.9.1"
-cachecontrol = [
-    { version = "^0.12.4", extras = ["filecache"], python = "<3.6" },
-    { version = "^0.12.9", extras = ["filecache"], python = "^3.6" }
-]
-pkginfo = "^1.4"
-html5lib = "^1.0"
-shellingham = "^1.1"
-tomlkit = ">=0.7.0,<1.0.0"
-pexpect = "^4.7.0"
-packaging = "^20.4"
-virtualenv = { version = "^20.0.26" }
 
-# The typing module is not in the stdlib in Python 2.7
-typing = { version = "^3.6", python = "~2.7" }
-
-# Use pathlib2 for Python 2.7
-pathlib2 = { version = "^2.3", python = "~2.7" }
-# Use futures on Python 2.7
-futures = { version = "^3.3.0", python = "~2.7" }
-# Use glob2 for Python 2.7 and 3.4
-glob2 = { version = "^0.6", python = "~2.7" }
-# functools32 is needed for Python 2.7
-functools32 = { version = "^3.2.3", python = "~2.7" }
-keyring = [
-    { version = "^18.0.1", python = "~2.7" },
-    { version = "^20.0.1", python = "~3.5" },
-    { version = ">=21.2.0", python = "^3.6" }
-]
-# Use subprocess32 for Python 2.7
-subprocess32 = { version = "^3.5", python = "~2.7" }
-importlib-metadata = {version = "^1.6.0", python = "<3.8"}
 
 [tool.poetry.dev-dependencies]
 pytest = [
