diff -rup gdl-2.31.2/gdl/Makefile.am gdl-2.31.2-ricky/gdl/Makefile.am
--- gdl-2.31.2/gdl/Makefile.am	2010-05-02 20:03:06.000000000 +0800
+++ gdl-2.31.2-ricky/gdl/Makefile.am	2016-10-06 16:21:27.766587200 +0800
@@ -30,7 +30,9 @@ libgdl_headers = \
 	gdl-dock-item-grip.h \
 	gdl-dock-layout.h \
 	gdl-dock-bar.h \
-	gdl-dock-placeholder.h
+	gdl-dock-placeholder.h \
+	gdl-dock-notebook.h \
+	gdl-switcher.h
 	
 libgdl_1_la_SOURCES = \
 	$(libgdl_headers) \
