--- aas-sign-0.4.0/CMakeLists.txt.orig	2026-04-19 18:27:17.000000000 +0200
+++ aas-sign-0.4.0/CMakeLists.txt	2026-04-24 20:19:18.165499700 +0200
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.20)
 project(aas-sign VERSION 1.0.0 LANGUAGES CXX C)
+include(GNUInstallDirs)
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
@@ -89,3 +90,5 @@
   list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
   include(Fuzzing)
 endif()
+
+install(TARGETS aas-sign RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
