--- aas-sign-0.4.0/CMakeLists.txt.orig	2026-04-24 21:15:56.096087300 +0200
+++ aas-sign-0.4.0/CMakeLists.txt	2026-04-24 21:17:34.977556800 +0200
@@ -13,11 +13,7 @@
 
 include(FetchContent)
 
-FetchContent_Declare(json
-  URL                     https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz
-  URL_HASH                SHA256=d6c65aca6b1ed68e7a182f4757257b107ae403032760ed6ef121c9d55e81757d
-  DOWNLOAD_EXTRACT_TIMESTAMP FALSE)
-FetchContent_MakeAvailable(json)
+find_package(nlohmann_json CONFIG REQUIRED)
 
 if(NOT WIN32)
   set(ENABLE_PROGRAMS OFF CACHE BOOL "" FORCE)
