--- triton/src/libtriton/bindings/python/namespaces/initStubsNamespace.cpp.orig	2023-12-19 12:12:09.571756100 +0300
+++ triton/src/libtriton/bindings/python/namespaces/initStubsNamespace.cpp	2023-12-19 12:12:34.038359100 +0300
@@ -127,7 +127,7 @@
 
       static PyObject* initI386(void) {
         PyObject* dict = xPyDict_New();
-        xPyDict_SetItemString(dict, "SYSTEMV", initSystemV(triton::stubs::i386::systemv::libc::code, triton::stubs::i386::systemv::libc::symbols));
+        xPyDict_SetItemString(dict, "SYSTEMV", initSystemV(triton::stubs::iX86::systemv::libc::code, triton::stubs::iX86::systemv::libc::symbols));
         PyObject* dictClass = xPyClass_New(nullptr, dict, xPyString_FromString("I386"));
         return dictClass;
       }
--- triton/src/libtriton/includes/triton/stubs.hpp.orig	2023-12-19 12:09:12.625778700 +0300
+++ triton/src/libtriton/includes/triton/stubs.hpp	2023-12-19 12:09:17.470779500 +0300
@@ -35,7 +35,7 @@
         };
       };
     };
-    namespace i386 {
+    namespace iX86 {
       namespace systemv {
         namespace libc {
           /*! Symbols mapping. Each entry points on its position in `systemv::libc::code`. */
--- triton/src/libtriton/stubs/i386-systemv-libc.cpp.orig	2023-12-19 12:04:25.108889200 +0300
+++ triton/src/libtriton/stubs/i386-systemv-libc.cpp	2023-12-19 12:05:18.486890300 +0300
@@ -10,11 +10,11 @@
 
 namespace triton {
   namespace stubs {
-    namespace i386 {
+    namespace iX86 {
       namespace systemv {
         namespace libc {
 
-          /* This map contains the mapping of function names and their offset in `i386::systemv::libc::code` */
+          /* This map contains the mapping of function names and their offset in `iX86::systemv::libc::code` */
           std::map<std::string, triton::uint64> symbols = {
             {"memccpy", 0x0},
             {"memchr", 0x4f},
@@ -7720,6 +7720,6 @@
           };
         }; /* libc namespace */
       }; /* systemv namespace */
-    }; /* i386 namespace */
+    }; /* iX86 namespace */
   }; /* stubs namespace */
 }; /* triton namespace */
