--- avrdude-6.2-orig/configure.ac
+++ avrdude-6.2/configure.ac
@@ -34,7 +34,7 @@
 AC_CONFIG_HEADERS(ac_cfg.h)
 AC_CONFIG_MACRO_DIR([m4])
 
-LT_INIT()
+LT_INIT([win32-dll])
 
 # Checks for programs.
 AC_PROG_CC
--- avrdude-6.2-orig/Makefile.am
+++ avrdude-6.2/Makefile.am
@@ -178,5 +178,5 @@
 libavrdude_la_SOURCES = $(libavrdude_a_SOURCES)
-libavrdude_la_LDFLAGS = -version-info 1:0
+libavrdude_la_LDFLAGS = -no-undefined -version-info 1:0
 
 include_HEADERS = libavrdude.h
 
