--- a/sources/pyside6/PySide6/__init__.py.in
+++ b/sources/pyside6/PySide6/__init__.py.in
@@ -23,7 +23,7 @@
     # is executed in the build directory. We need libpyside and Qt in addition.
     shiboken6 = Path(root).parent / 'shiboken6' / 'libshiboken'
     if not shiboken6.is_dir():
-        raise ImportError(shiboken6 + ' does not exist')
+        raise ImportError(str(shiboken6) + ' does not exist')
     result = [shiboken6, root / 'libpyside']
     for path in os.environ.get('PATH').split(';'):
         if path:
@@ -51,7 +51,7 @@
 
     pyside_package_dir = Path(__file__).parent.resolve()
 
-    if sys.platform == 'win32' and sys.version_info[0] == 3 and sys.version_info[1] >= 8:
+    if sys.platform == 'win32' and 'MSC' in sys.version and sys.version_info[0] == 3 and sys.version_info[1] >= 8:
         for dir in _additional_dll_directories(pyside_package_dir):
             os.add_dll_directory(dir)
 
--- a/build_scripts/utils.py
+++ b/build_scripts/utils.py
@@ -246,7 +246,7 @@
 
 def platform_cmake_options():
     result = []
-    if sys.platform == 'win32':
+    if sys.platform == 'win32' and 'MSC' in sys.version:
         # Prevent cmake from auto-detecting clang if it is in path.
         result.append("-DCMAKE_C_COMPILER=cl.exe")
         result.append("-DCMAKE_CXX_COMPILER=cl.exe")
