--- a/sources/shiboken6/generator/CMakeLists.txt
+++ b/sources/shiboken6/generator/CMakeLists.txt
@@ -47,6 +47,10 @@
                    qtdoc/rstformat.h)
     target_compile_definitions(shiboken6 PUBLIC DOCSTRINGS_ENABLED QT_LEAN_HEADERS=1)
 endif()
+
+if(MINGW)
+  target_link_options(shiboken6 PRIVATE "-municode")
+endif()
 
 configure_file(shibokenconfig.h.in "${CMAKE_CURRENT_BINARY_DIR}/shibokenconfig.h" @ONLY)
 
--- a/build_scripts/utils.py
+++ b/build_scripts/utils.py
@@ -86,7 +86,7 @@
 
 def platform_cmake_options(as_tuple_list=False):
     result = []
-    if sys.platform == 'win32':
+    if sys.platform == 'win32' and 'MSC' in sys.version:
         # Prevent cmake from auto-detecting clang if it is in path.
         if as_tuple_list:
             result.append(("CMAKE_C_COMPILER", "cl.exe"))
