--- a/sources/shiboken6_generator/ApiExtractor/typesystemparser.cpp
+++ b/sources/shiboken6_generator/ApiExtractor/typesystemparser.cpp
@@ -1384,7 +1384,7 @@ FlagsTypeEntryPtr
 {
     if (!checkRootElement())
         return nullptr;
-    auto ftype = std::make_shared<FlagsTypeEntry>(u"QFlags<"_s + enumEntry->name() + u'>',
+    auto ftype = std::make_shared<FlagsTypeEntry>(u"QFlags<"_s + enumEntry->qualifiedCppName() + u'>',
                                                   since,
                                                   typeSystemTypeEntry(currentParentTypeEntry()));
     ftype->setOriginator(enumEntry);
