--- pyside-setup-everywhere-src-6.11.0/sources/shiboken6_generator/ApiExtractor/clangparser/compilersupport.cpp.orig	2026-05-13 14:46:56.568835800 +0300
+++ pyside-setup-everywhere-src-6.11.0/sources/shiboken6_generator/ApiExtractor/clangparser/compilersupport.cpp	2026-05-13 14:47:33.285719500 +0300
@@ -338,7 +338,12 @@
 QString compilerFromCMake()
 {
 #ifdef CMAKE_CXX_COMPILER
-    return QString::fromLocal8Bit(CMAKE_CXX_COMPILER);
+    QString buildCompiler = QString::fromLocal8Bit(CMAKE_CXX_COMPILER);
+    QFileInfo comp_info(buildCompiler);
+    if (comp_info.exists()) {
+        return buildCompiler;
+    }
+    return comp_info.fileName();
 #else
     return {};
 #endif
