diff -Naur libxlsxwriter-RELEASE_0.7.5.org/CMakeLists.txt libxlsxwriter-RELEASE_0.7.5/CMakeLists.txt
--- libxlsxwriter-RELEASE_0.7.5.org/CMakeLists.txt	2017-09-25 07:49:53.000000000 +0800
+++ libxlsxwriter-RELEASE_0.7.5/CMakeLists.txt	2017-10-30 17:42:56.913593700 +0800
@@ -143,12 +143,15 @@
     add_library(${PROJECT_NAME} SHARED ${LXW_SOURCES})
 endif()
 
+target_link_libraries(${PROJECT_NAME} ${ZLIB_LIBRARIES})
+
 if(MSVC)
     add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD
         COMMAND ${CMAKE_COMMAND} -E rename
         ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pdb
         $<TARGET_FILE_DIR:${PROJECT_NAME}>/${PROJECT_NAME}.pdb
     )
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif()
 
 # TESTS
@@ -241,26 +244,26 @@
 
 if(MSVC)
     if(CMAKE_CL_64)
-        install(TARGETS ${PROJECT_NAME}
-            LIBRARY DESTINATION "lib/x64/\${CMAKE_INSTALL_CONFIG_NAME}"
-            ARCHIVE DESTINATION "lib/x64/\${CMAKE_INSTALL_CONFIG_NAME}"
-        )
-        install(FILES $<TARGET_FILE_DIR:${PROJECT_NAME}>/${PROJECT_NAME}.pdb
-            DESTINATION "lib/x64/\${CMAKE_INSTALL_CONFIG_NAME}"
-        )
+        set(MSVC_FOLDER_PREFIX x64)
     else()
-        install(TARGETS ${PROJECT_NAME}
-            LIBRARY DESTINATION "lib/Win32/\${CMAKE_INSTALL_CONFIG_NAME}"
-            ARCHIVE DESTINATION "lib/Win32/\${CMAKE_INSTALL_CONFIG_NAME}"
-        )
+        set(MSVC_FOLDER_PREFIX Win32)
+    endif()
+
+    install(TARGETS ${PROJECT_NAME}
+        LIBRARY DESTINATION "lib/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
+        ARCHIVE DESTINATION "lib/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
+        RUNTIME DESTINATION "bin/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
+    )
+    if (BUILD_STATIC)
         install(FILES $<TARGET_FILE_DIR:${PROJECT_NAME}>/${PROJECT_NAME}.pdb
-            DESTINATION "lib/Win32/\${CMAKE_INSTALL_CONFIG_NAME}"
+            DESTINATION "lib/{MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
         )
     endif()
 else(MSVC)
     install(TARGETS ${PROJECT_NAME}
         LIBRARY DESTINATION lib
         ARCHIVE DESTINATION lib
+        RUNTIME DESTINATION bin
     )
 endif(MSVC)
 install(FILES include/xlsxwriter.h DESTINATION include)
