Escape backslash characters in strings.

diff -urN boto-2.49.0/boto/__init__.py.orig boto-2.49.0/boto/__init__.py
--- boto-2.49.0/boto/__init__.py.orig	2018-07-11 22:42:42.000000000 +0200
+++ boto-2.49.0/boto/__init__.py	2024-11-06 10:37:26.382606400 +0100
@@ -1136,7 +1136,7 @@
     * gs://bucket
     * s3://bucket
     * filename (which could be a Unix path like /a/b/c or a Windows path like
-      C:\a\b\c)
+      C:\\a\\b\\c)
 
     The last example uses the default scheme ('file', unless overridden).
     """
diff -urN boto-2.49.0/boto/pyami/config.py.orig boto-2.49.0/boto/pyami/config.py
--- boto-2.49.0/boto/pyami/config.py.orig	2018-07-11 19:35:36.000000000 +0200
+++ boto-2.49.0/boto/pyami/config.py	2024-11-06 10:27:59.943008800 +0100
@@ -95,7 +95,7 @@
     def load_from_path(self, path):
         file = open(path)
         for line in file.readlines():
-            match = re.match("^#import[\s\t]*([^\s^\t]*)[\s\t]*$", line)
+            match = re.match("^#import[\\s\t]*([^\\s^\t]*)[\\s\t]*$", line)
             if match:
                 extended_file = match.group(1)
                 (dir, file) = os.path.split(path)
