--- memphis-0.2.3/memphis/Makefile.am.orig	2014-10-19 13:32:28.984200000 +0400
+++ memphis-0.2.3/memphis/Makefile.am	2014-10-19 13:32:39.561000000 +0400
@@ -95,6 +95,7 @@
 Memphis_0_2_gir_LIBS = libmemphis-0.2.la
 Memphis_0_2_gir_FILES = $(addprefix $(srcdir)/,$(introspection_sources))
 INTROSPECTION_GIRS += Memphis-0.2.gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 girdir = $(datadir)/gir-1.0
 dist_gir_DATA = $(INTROSPECTION_GIRS)
