diff -Naur tiled-1.11.1.origin/src/plugins/python/python.qbs tiled-1.11.1/src/plugins/python/python.qbs
--- tiled-1.11.1.origin/src/plugins/python/python.qbs	2025-01-13 19:24:11.082758300 +0300
+++ tiled-1.11.1/src/plugins/python/python.qbs	2025-01-13 19:24:40.052299600 +0300
@@ -42,17 +42,6 @@
         cpp.linkerFlags: pkgConfigPython3.linkerFlags
     }
 
-    Properties {
-        condition: qbs.targetOS.contains("windows")
-        cpp.includePaths: [Environment.getEnv("PYTHONHOME") + "/include"]
-        cpp.libraryPaths: [Environment.getEnv("PYTHONHOME") + "/libs"]
-        cpp.dynamicLibraries: {
-            if (qbs.toolchain.contains("mingw"))
-                return [FileInfo.joinPaths(Environment.getEnv("PYTHONHOME"), pythonDllProbe.fileNamePrefix + ".dll")];
-            return ["python3"];
-        }
-    }
-
     files: [
         "plugin.json",
         "pythonplugin.cpp",
