diff --git a/qbs/imports/TiledPlugin.qbs b/qbs/imports/TiledPlugin.qbs
index 6b835638b..a581b9b2c 100644
--- a/qbs/imports/TiledPlugin.qbs
+++ b/qbs/imports/TiledPlugin.qbs
@@ -37,7 +37,7 @@ DynamicLibrary {
     Group {
         qbs.install: true
         qbs.installDir: {
-            if (qbs.targetOS.contains("windows"))
+            if (project.windowsLayout)
                 return "plugins/tiled"
             else if (qbs.targetOS.contains("macos"))
                 return "Tiled.app/Contents/PlugIns"
diff --git a/qbs/imports/TiledQtGuiApplication.qbs b/qbs/imports/TiledQtGuiApplication.qbs
index 9301573b4..b63c06cbb 100644
--- a/qbs/imports/TiledQtGuiApplication.qbs
+++ b/qbs/imports/TiledQtGuiApplication.qbs
@@ -29,7 +29,7 @@ QtGuiApplication {
     Group {
         qbs.install: true
         qbs.installDir: {
-            if (qbs.targetOS.contains("windows")) {
+            if (project.windowsLayout) {
                 return "";
             } else if (qbs.targetOS.contains("darwin")) {
                 // Non-bundle applications are installed into the main Tiled.app bundle
diff --git a/src/libtiled/libtiled.qbs b/src/libtiled/libtiled.qbs
index c1d72ae74..104743512 100644
--- a/src/libtiled/libtiled.qbs
+++ b/src/libtiled/libtiled.qbs
@@ -37,6 +37,9 @@ DynamicLibrary {
         if (project.staticZstd || pkgConfigZstd.found)
             defs.push("TILED_ZSTD_SUPPORT");
 
+        if (project.windowsLayout)
+            defs.push("TILED_WINDOWS_LAYOUT");
+
         return defs;
     }
     cpp.dynamicLibraries: {
@@ -201,7 +204,10 @@ DynamicLibrary {
         qbs.install: true
         qbs.installDir: {
             if (qbs.targetOS.contains("windows"))
-                return ""
+                if (project.windowsLayout)
+                    return ""
+                else
+                    return "bin"
             else
                 return "lib"
         }
diff --git a/src/libtiled/pluginmanager.cpp b/src/libtiled/pluginmanager.cpp
index 86a6bbd63..bfb2abd85 100644
--- a/src/libtiled/pluginmanager.cpp
+++ b/src/libtiled/pluginmanager.cpp
@@ -189,7 +189,7 @@ void PluginManager::loadPlugins()
     QString pluginPath = QCoreApplication::applicationDirPath();
 #endif
 
-#if defined(Q_OS_WIN32)
+#if defined(TILED_WINDOWS_LAYOUT)
     pluginPath += QStringLiteral("/plugins/tiled");
 #elif defined(Q_OS_MAC)
     pluginPath += QStringLiteral("/../PlugIns");
diff --git a/src/libtiledquick/libtiledquick.qbs b/src/libtiledquick/libtiledquick.qbs
index 656f146d1..f9cbb02d2 100644
--- a/src/libtiledquick/libtiledquick.qbs
+++ b/src/libtiledquick/libtiledquick.qbs
@@ -69,7 +69,10 @@ DynamicLibrary {
         qbs.install: true
         qbs.installDir: {
             if (qbs.targetOS.contains("windows"))
-                return ""
+                if (project.windowsLayout)
+                    return ""
+                else
+                    return "bin"
             else
                 return "lib"
         }
diff --git a/src/tiled/languagemanager.cpp b/src/tiled/languagemanager.cpp
index 72979831d..2d7295da7 100644
--- a/src/tiled/languagemanager.cpp
+++ b/src/tiled/languagemanager.cpp
@@ -42,7 +42,7 @@ LanguageManager::LanguageManager()
     , mAppTranslator(nullptr)
 {
     mTranslationsDir = QCoreApplication::applicationDirPath();
-#if defined(Q_OS_WIN32)
+#if defined(TILED_WINDOWS_LAYOUT)
     mTranslationsDir += QStringLiteral("/translations");
 #elif defined(Q_OS_MAC)
     mTranslationsDir += QStringLiteral("/../Translations");
diff --git a/src/tiled/libtilededitor.qbs b/src/tiled/libtilededitor.qbs
index 422df509d..1d707d7b5 100644
--- a/src/tiled/libtilededitor.qbs
+++ b/src/tiled/libtilededitor.qbs
@@ -52,6 +52,9 @@ DynamicLibrary {
         if (project.snapshot)
             defs.push("TILED_SNAPSHOT");
 
+        if (project.windowsLayout)
+            defs.push("TILED_WINDOWS_LAYOUT");
+
         if (qbs.targetOS.contains("linux") && project.dbus && Qt.dbus.present)
             defs.push("TILED_ENABLE_DBUS");
 
@@ -596,7 +599,10 @@ DynamicLibrary {
         qbs.install: true
         qbs.installDir: {
             if (qbs.targetOS.contains("windows"))
-                return ""
+                if (project.windowsLayout)
+                    return ""
+                else
+                    return "bin"
             else
                 return "lib"
         }
diff --git a/src/tiledquick/tiledquick.qbs b/src/tiledquick/tiledquick.qbs
index 832435244..5c72a3cc2 100644
--- a/src/tiledquick/tiledquick.qbs
+++ b/src/tiledquick/tiledquick.qbs
@@ -47,7 +47,7 @@ QtGuiApplication {
         condition: !qbs.targetOS.contains("darwin")
         qbs.install: true
         qbs.installDir: {
-            if (qbs.targetOS.contains("windows"))
+            if (project.windowsLayout)
                 return ""
             else
                 return "bin"
diff --git a/tiled.qbs b/tiled.qbs
index 2ffb30fbb..059b06070 100644
--- a/tiled.qbs
+++ b/tiled.qbs
@@ -13,6 +13,7 @@ Project {
     property bool installHeaders: false
     property bool useRPaths: true
     property bool windowsInstaller: false
+    property bool windowsLayout: qbs.targetOS.contains("windows")
     property bool staticZstd: false
     property bool sentry: false
     property bool dbus: true
diff --git a/translations/translations.qbs b/translations/translations.qbs
index 52af2d080..81b49af71 100644
--- a/translations/translations.qbs
+++ b/translations/translations.qbs
@@ -18,7 +18,7 @@ Product {
         fileTagsFilter: product.type
         qbs.install: true
         qbs.installDir: {
-            if (qbs.targetOS.contains("windows"))
+            if (project.windowsLayout)
                 return "translations"
             else if (qbs.targetOS.contains("macos"))
                 return "Tiled.app/Contents/Translations"
