diff -Naur old/src/message.c new/src/message.c
--- old/src/message.c	2012-08-31 02:15:35.000000000 -1000
+++ new/src/message.c	2014-01-06 01:22:43.316831758 -1000
@@ -441,7 +441,7 @@
 	/* Fragment and send according to policy */
 	if (!err && messagep && *messagep) {
 	    if (context) {
-		err = fragment_and_send(ops, NULL, context, *messagep,
+		err = fragment_and_send(ops, opdata, context, *messagep,
 			fragPolicy, messagep);
 	    }
 	}
