--- src/meson.build.orig	2026-04-25 22:52:40.981768100 -0500
+++ src/meson.build	2026-04-25 22:52:47.156275100 -0500
@@ -53,7 +53,7 @@
 config_h.set('HAVE_STRINGS_H', cc.has_header('strings.h'))
 
 config_h.set_quoted('ZOITECHATLIBDIR',
-  join_paths(get_option('prefix'), get_option('libdir'), 'zoitechat/plugins')
+  join_paths('./../', get_option('libdir'), 'zoitechat/plugins')
 )
 
 if libssl_dep.found()
