--- src/src/fe-gtk/meson.build.orig	2026-04-20 12:56:03.000000000 -0500
+++ src/src/fe-gtk/meson.build	2026-04-25 23:22:12.189844300 -0500
@@ -128,8 +128,11 @@
   zoitechat_gtk_ldflags += '-Wl,-e,mainCRTStartup'
 endif
 
+win = import('windows')
+exe_icon = win.compile_resources('../../zoitechat.rc')
+
 executable('zoitechat',
-  sources:  resources + zoitechat_gtk_sources,
+  sources:  [resources, zoitechat_gtk_sources, exe_icon],
   dependencies: zoitechat_gtk_deps,
   link_with: zoitechat_theme_lib,
   c_args: zoitechat_gtk_cflags,
