--- libsndfile-1.0.25/configure.ac.orig	2012-09-15 09:59:44 +0400
+++ libsndfile-1.0.25/configure.ac	2012-09-15 10:07:57 +0400
@@ -468,10 +468,11 @@
 	AC_MSG_WARN([[******************************************************************]])
 	fi
 
+AUTOGEN="autogen --writable"
 if test x"$ac_cv_prog_HAVE_AUTOGEN" = "xno" ; then
-	AC_MSG_WARN([[Touching files in directory tests/.]])
-	touch tests/*.c tests/*.h
+	AUTOGEN="touch $@"
 	fi
+AC_SUBST(AUTOGEN)
 
 #====================================================================================
 # Settings for the HTML documentation.
--- libsndfile-1.0.25/src/Makefile.am.orig	2012-09-15 09:59:44 +0400
+++ libsndfile-1.0.25/src/Makefile.am	2012-09-15 10:08:56 +0400
@@ -78,7 +78,7 @@
 G72x_g72x_test_LDADD = G72x/libg72x.la
 
 test_endswap.c: test_endswap.def test_endswap.tpl
-	autogen --writable test_endswap.def
+	@AUTOGEN@ test_endswap.def
 
 genfiles : test_endswap.c $(SYMBOL_FILES)
 
--- libsndfile-1.0.25/tests/Makefile.am.orig	2012-09-15 09:59:41 +0400
+++ libsndfile-1.0.25/tests/Makefile.am	2012-09-15 10:09:37 +0400
@@ -179,31 +179,31 @@
 #===============================================================================
 
 write_read_test.c: write_read_test.def write_read_test.tpl
-	autogen --writable write_read_test.def
+	@AUTOGEN@ write_read_test.def
 
 pcm_test.c: pcm_test.def pcm_test.tpl
-	autogen --writable pcm_test.def
+	@AUTOGEN@ pcm_test.def
 
 header_test.c: header_test.def header_test.tpl
-	autogen --writable header_test.def
+	@AUTOGEN@ header_test.def
 
 utils.c utils.h : utils.def utils.tpl
-	autogen --writable utils.def
+	@AUTOGEN@ utils.def
 
 scale_clip_test.c: scale_clip_test.def scale_clip_test.tpl
-	autogen --writable scale_clip_test.def
+	@AUTOGEN@ scale_clip_test.def
 
 pipe_test.c: pipe_test.def pipe_test.tpl
-	autogen --writable pipe_test.def
+	@AUTOGEN@ pipe_test.def
 
 rdwr_test.c: rdwr_test.def rdwr_test.tpl
-	autogen --writable rdwr_test.def
+	@AUTOGEN@ rdwr_test.def
 
 floating_point_test.c: floating_point_test.def floating_point_test.tpl
-	autogen --writable floating_point_test.def
+	@AUTOGEN@ floating_point_test.def
 
 benchmark.c: benchmark.def benchmark.tpl
-	autogen --writable benchmark.def
+	@AUTOGEN@ benchmark.def
 
 genfiles : write_read_test.c pcm_test.c header_test.c utils.c \
 		scale_clip_test.c pipe_test.c floating_point_test.c rdwr_test.c \
