--- libsndfile-1.0.25/configure.ac.orig	2014-04-21 07:35:07.656800000 +0400
+++ libsndfile-1.0.25/configure.ac	2014-04-21 07:35:43.256000000 +0400
@@ -335,7 +335,7 @@
 		enable_external_libs=yes
 
 		EXTERNAL_CFLAGS="$FLAC_CFLAGS $OGG_CFLAGS $VORBISENC_CFLAGS $SPEEX_CFLAGS"
-		EXTERNAL_LIBS="$FLAC_LIBS $VORBISENC_LIBS $SPEEX_LIBS"
+		EXTERNAL_LIBS="$FLAC_LIBS $OGG_LIBS $VORBIS_LIBS $VORBISENC_LIBS $SPEEX_LIBS"
 	else
 		echo
 		AC_MSG_WARN([[*** One or more of the external libraries (ie libflac, libogg and]])
