--- matplotlib-2.1.2/setup.py.orig	2018-01-18 07:21:34.000000000 +0300
+++ matplotlib-2.1.2/setup.py	2018-01-25 08:35:39.807755700 +0300
@@ -102,7 +102,7 @@
     setupext.BackendGtk3Agg(),
     setupext.BackendGtk3Cairo(),
     setupext.BackendGtkAgg(),
-    setupext.BackendTkAgg(),
+    #setupext.BackendTkAgg(),
     setupext.BackendWxAgg(),
     setupext.BackendGtk(),
     setupext.BackendAgg(),
