--- med-4.1.0/tools/medimport/CMakeLists.txt.orig	2020-03-12 16:45:46.000000000 +0100
+++ med-4.1.0/tools/medimport/CMakeLists.txt	2020-10-31 09:02:06.933956100 +0100
@@ -59,7 +59,10 @@
     VERSION 0.4.0)
   TARGET_LINK_LIBRARIES(medimportengine medC)
   SET_TARGET_PROPERTIES(medimportengine PROPERTIES OUTPUT_NAME medimport)
-  INSTALL(TARGETS medimportengine DESTINATION lib${LIB_SUFFIX})
+  INSTALL(TARGETS medimportengine
+          RUNTIME DESTINATION bin
+          LIBRARY DESTINATION lib${LIB_SUFFIX}
+          ARCHIVE DESTINATION lib${LIB_SUFFIX})
   SET(_lib_to_link "medimportengine")
 ENDIF()
 
@@ -69,7 +72,10 @@
   TARGET_LINK_LIBRARIES(medimportengine_static medC_static)
   MED_SET_DEFINITIONS(medimportengine_static MED3_USESTATIC)
   SET_TARGET_PROPERTIES(medimportengine_static PROPERTIES OUTPUT_NAME medimport)
-  INSTALL(TARGETS medimportengine_static DESTINATION lib${LIB_SUFFIX})
+  INSTALL(TARGETS medimportengine_static
+          RUNTIME DESTINATION bin
+          LIBRARY DESTINATION lib${LIB_SUFFIX}
+          ARCHIVE DESTINATION lib${LIB_SUFFIX})
   # Give precendence to the shared object for the linking of 
   # the executable medimport:
   IF(NOT _lib_to_link)
--- med-4.1.0/src/CMakeLists.txt.orig	2020-03-12 16:45:46.000000000 +0100
+++ med-4.1.0/src/CMakeLists.txt	2020-10-31 09:10:55.720325200 +0100
@@ -113,7 +113,10 @@
       SOVERSION 11
       VERSION   11.0.1)
     TARGET_LINK_LIBRARIES(medfwrap medC)
-    INSTALL(TARGETS medfwrap EXPORT medfileTargetsF DESTINATION lib${LIB_SUFFIX})
+    INSTALL(TARGETS medfwrap EXPORT medfileTargetsF
+            RUNTIME DESTINATION bin
+            LIBRARY DESTINATION lib${LIB_SUFFIX}
+            ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
     # Add Shared MED library
     ADD_LIBRARY(med SHARED MEDiterators.c)
@@ -124,7 +127,10 @@
     TARGET_LINK_LIBRARIES(med medfwrap)    
    
     # Install only the resulting library:
-    INSTALL(TARGETS med EXPORT medTargetsF DESTINATION lib${LIB_SUFFIX})
+    INSTALL(TARGETS med EXPORT medfileTargetsF
+            RUNTIME DESTINATION bin
+            LIBRARY DESTINATION lib${LIB_SUFFIX}
+            ARCHIVE DESTINATION lib${LIB_SUFFIX})
   ENDIF()
 
   ######### Static Libraries ##########
@@ -137,7 +143,10 @@
     ADD_LIBRARY(medfwrap_static STATIC ${medfort_wrap_SOURCES})
     SET_TARGET_PROPERTIES(medfwrap_static PROPERTIES OUTPUT_NAME medfwrap)
     TARGET_LINK_LIBRARIES(medfwrap_static medC_static)
-    INSTALL(TARGETS medfwrap_static EXPORT medfileTargetsF DESTINATION lib${LIB_SUFFIX})
+    INSTALL(TARGETS medfwrap_static EXPORT medfileTargetsF
+            RUNTIME DESTINATION bin
+            LIBRARY DESTINATION lib${LIB_SUFFIX}
+            ARCHIVE DESTINATION lib${LIB_SUFFIX})
     
     # Add Static MED library
     ADD_LIBRARY(med_static STATIC MEDiterators.c)
@@ -145,7 +154,10 @@
     TARGET_LINK_LIBRARIES(med_static medfwrap_static)
 
     # Install only the resulting library:  
-    INSTALL(TARGETS med_static EXPORT medfileTargetsF DESTINATION lib${LIB_SUFFIX})
+    INSTALL(TARGETS med_static EXPORT medfileTargetsF
+            RUNTIME DESTINATION bin
+            LIBRARY DESTINATION lib${LIB_SUFFIX}
+            ARCHIVE DESTINATION lib${LIB_SUFFIX})
   ENDIF()
   
 ENDIF(CMAKE_Fortran_COMPILER_WORKS)
--- med-4.1.0/src/CMakeLists.txt.orig	2020-10-31 09:11:57.594041300 +0100
+++ med-4.1.0/src/CMakeLists.txt	2020-10-31 09:15:01.643420300 +0100
@@ -83,7 +83,10 @@
   TARGET_LINK_LIBRARIES(medC ${HDF5_LIBS} ${MPI_LIBS})
   MED_SET_DEFINITIONS(medC NOGDI)
   
-  INSTALL(TARGETS medC EXPORT ${_export_group} DESTINATION lib${LIB_SUFFIX})  
+  INSTALL(TARGETS medC EXPORT ${_export_group}
+          RUNTIME DESTINATION bin
+          LIBRARY DESTINATION lib${LIB_SUFFIX}
+          ARCHIVE DESTINATION lib${LIB_SUFFIX})
 ENDIF()
 
 ######### Static Libraries ##########
@@ -97,7 +100,10 @@
   TARGET_LINK_LIBRARIES(medC_static ${HDF5_LIBS} ${MPI_LIBS})
   MED_SET_DEFINITIONS(medC_static NOGDI)
   
-  INSTALL(TARGETS medC_static EXPORT ${_export_group} DESTINATION lib${LIB_SUFFIX})
+  INSTALL(TARGETS medC_static EXPORT ${_export_group}
+          RUNTIME DESTINATION bin
+          LIBRARY DESTINATION lib${LIB_SUFFIX}
+          ARCHIVE DESTINATION lib${LIB_SUFFIX})
 ENDIF()
 
 ########################### Fortran stuff ###################################
