--- a/include/med_hdfi.h
+++ b/include/med_hdfi.h
@@ -411,7 +411,7 @@ extern MEDC_EXPORT
 med_err _MEDvisit(const med_idt fid, const char * const srcpath, const char * const dstpath,
 		  medvisitorfunc);
 extern MEDC_EXPORT
-med_err _MEDiterate(const med_idt fid, herr_t (*func)(), void * itdatas );
+med_err _MEDiterate(const med_idt fid, medvisitorfunc, void * itdatas );
 
 extern MEDC_EXPORT
 med_err
--- a/src/ci/_MEDfieldComputingStepCheck236.c
+++ b/src/ci/_MEDfieldComputingStepCheck236.c
@@ -146,7 +146,7 @@ med_err _MEDfieldComputingStepCheck236(m
        * On vérifie que le nom du maillage par défaut est le même pour toutes
        * les étapes de calcul de tous les couples (entitytype,geotype)
        */
-      if (_MEDiterate(_cstpid, _MEDcheckAttributeStringFunc, &_itdatas ) < 0) {
+      if (_MEDiterate(_cstpid, (medvisitorfunc)_MEDcheckAttributeStringFunc, &_itdatas ) < 0) {
 	MED_ERR_(_ret,MED_ERR_INVALID,MED_ERR_ATTRIBUTE,MED_ERR_VALUE_MSG);
 	SSCRUTE(_itdatas.attvalprec); SSCRUTE(_itdatas.attval); goto ERROR;
       }
@@ -157,7 +157,7 @@ med_err _MEDfieldComputingStepCheck236(m
      * les étapes de calcul de tous les couples (entitytype,geotype)
      */
     if (checkmultiplemesh) {
-      if (_MEDiterate(_cstpid, _MEDchecknSublinkFunc, multiplemesh ) < 0) {
+      if (_MEDiterate(_cstpid, (medvisitorfunc)_MEDchecknSublinkFunc, multiplemesh ) < 0) {
 	MED_ERR_(_ret,MED_ERR_COUNT,MED_ERR_DATAGROUP,MED_ERR_NBR_MSG MED_ERR_MESH_MSG);
 	SSCRUTE(_path);goto ERROR;
       }
--- a/src/hdfi/_MEDiterate.c
+++ b/src/hdfi/_MEDiterate.c
@@ -25,7 +25,7 @@
 /* 	  herr_t (*func)( hid_t g_id, const char *name, const H5L_info_t *info, void *op_data) ) */
 
 med_err
-_MEDiterate(const med_idt fid,  herr_t (*func)(), void * itdatas )
+_MEDiterate(const med_idt fid,  herr_t (*func)(hid_t g_id, const char *name, const H5L_info_t *info, void *op_data), void * itdatas )
 {
 
   med_err    _ret=-1;
--- a/src/hdfi/_MEDvisit.c
+++ b/src/hdfi/_MEDvisit.c
@@ -26,7 +26,7 @@
 
 med_err
 _MEDvisit(const med_idt fid, const char * const srcpath, const char * const dstpath,
-	  herr_t (*func)() )
+	  herr_t (*func)(hid_t g_id, const char *name, const H5L_info_t *info, void *op_data) )
 {
 
   med_err    _ret=-1;
