--- a/configure.ac
+++ b/configure.ac
@@ -64,8 +64,8 @@
 AC_SUBST([CODEBLOCKS_LIBS])
 
 #cb_plugin_datadir="`$PKG_CONFIG codeblocks --variable plugindir`"
-cb_plugin_datadir="`$PKG_CONFIG codeblocks --variable sharedir`/codeblocks/"
-cb_plugin_libdir="`$PKG_CONFIG codeblocks --variable libdir`/codeblocks/plugins/"
+cb_plugin_datadir="`$PKG_CONFIG codeblocks --dont-define-prefix --variable sharedir`/codeblocks/"
+cb_plugin_libdir="`$PKG_CONFIG codeblocks --dont-define-prefix --variable libdir`/codeblocks/plugins/"
 
 AC_SUBST([cb_plugin_datadir])
 AC_SUBST([cb_plugin_libdir])
