diff --git a/keras/backend/load_backend.py b/keras/backend/load_backend.py
index 4942f613d..f7256b191 100644
--- a/keras/backend/load_backend.py
+++ b/keras/backend/load_backend.py
@@ -23,8 +23,8 @@ else:
         _keras_base_dir = '/tmp'
     _keras_dir = os.path.join(_keras_base_dir, '.keras')
 
-# Default backend: TensorFlow.
-_BACKEND = 'tensorflow'
+# Default backend: Theano.
+_BACKEND = 'theano'
 
 # Attempt to read Keras config file.
 _config_path = os.path.expanduser(os.path.join(_keras_dir, 'keras.json'))
