diff -aurN 0001/src/tools/munch.cxx 0002/src/tools/munch.cxx
--- 0001/src/tools/munch.cxx
+++ 0002/src/tools/munch.cxx
@@ -361,11 +361,21 @@
         nptr++;
       }
       if (ft == 'P') {
+              if (numpfx == MAX_PREFIXES) {
+                  fprintf(stderr,"MAX_PREFIXES (%d) reached.",MAX_PREFIXES);
+                  fprintf(stderr,"please edit value in src/tools/munch.h and rebuild hunspell\n");
+                  exit(1);
+              }
         ptable[numpfx].aep = ptr;
         ptable[numpfx].num = numents;
         fprintf(stderr, "ptable %d num is %d\n", numpfx, ptable[numpfx].num);
         numpfx++;
       } else {
+             if (numsfx == MAX_SUFFIXES) {
+                 fprintf(stderr,"MAX_SUFFIXES (%d) reached.",MAX_SUFFIXES);
+                 fprintf(stderr,"please edit value in src/tools/munch.h and rebuild hunspell\n");
+                 exit(1);
+             }
         stable[numsfx].aep = ptr;
         stable[numsfx].num = numents;
         fprintf(stderr, "stable %d num is %d\n", numsfx, stable[numsfx].num);
