diff -aurN 0006/src/tools/munch.cxx 0007/src/tools/munch.cxx
--- 0006/src/tools/munch.cxx
+++ 0007/src/tools/munch.cxx
@@ -47,6 +47,7 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <stddef.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -aurN 0006/src/tools/unmunch.cxx 0007/src/tools/unmunch.cxx
--- 0006/src/tools/unmunch.cxx
+++ 0007/src/tools/unmunch.cxx
@@ -49,6 +49,7 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <stddef.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
