--- hunspell-1.7.0/configure.ac.orig	2019-05-28 10:02:21.550919600 +0300
+++ hunspell-1.7.0/configure.ac	2019-05-28 10:04:50.211127700 +0300
@@ -80,11 +80,13 @@
 AS_IF([test "x$with_readline" != xno],
 	[AC_CHECK_LIB([tinfo],[tgetent], [
 		TERMLIB=-ltinfo
-        ],[AC_CHECK_LIB([curses],[tparm],[
+	],[AC_CHECK_LIB([ncurses],[tparm],[
 		TERMLIB=-lncurses
+	],[AC_CHECK_LIB([curses],[tparm],[
+		TERMLIB=-lcurses
 	],[AC_CHECK_LIB([termcap],[tgetent],[
 		TERMLIB=-ltermcap
-	])])])
+	])])])])
 	LDSAVE=$LDFLAGS
 	LDFLAGS="$LDFLAGS $TERMLIB"
 	AC_CHECK_LIB([readline],[readline],[
