diff -aurN hunspell-1.3.4/src/tools/munch.c.orig hunspell-1.3.4/src/tools/munch.c > hunspell-1.3.4-exit-on-MAX_PREFIXES_SUFFIXES.patch
--- hunspell-1.3.4/src/tools/munch.c.orig	2016-04-16 06:47:17.896633300 -0400
+++ hunspell-1.3.4/src/tools/munch.c	2016-04-16 06:51:17.309889900 -0400
@@ -360,11 +360,21 @@
         nptr++;
       }
       if (ft == 'P') {
+              if (numpfx == MAX_PREFIXES) {
+                  fprintf(stderr,"MAX_PREFIXES (%d) reached.",MAX_PREFIXES);
+                  fprintf(stderr,"please edit value in src/tools/munch.h and rebuild hunspell\n");
+                  exit(1);
+              }
         ptable[numpfx].aep = ptr;
         ptable[numpfx].num = numents;
         fprintf(stderr, "ptable %d num is %d\n", numpfx, ptable[numpfx].num);
         numpfx++;
       } else {
+             if (numsfx == MAX_SUFFIXES) {
+                 fprintf(stderr,"MAX_SUFFIXES (%d) reached.",MAX_SUFFIXES);
+                 fprintf(stderr,"please edit value in src/tools/munch.h and rebuild hunspell\n");
+                 exit(1);
+             }
         stable[numsfx].aep = ptr;
         stable[numsfx].num = numents;
         fprintf(stderr, "stable %d num is %d\n", numsfx, stable[numsfx].num);
