diff -aur hunspell-1.4.1-orig/src/tools/unmunch.cxx hunspell-1.4.1/src/tools/unmunch.cxx
--- hunspell-1.4.1-orig/src/tools/unmunch.cxx	2016-05-03 03:55:52 -0400
+++ hunspell-1.4.1/src/tools/unmunch.cxx	2016-05-18 02:59:25 -0400
@@ -516,9 +516,17 @@
   if (n > 0) {
     char* dp = (char*)memchr(mp, (int)((unsigned char)delim), n);
     if (dp) {
+#if defined(_WIN32)
+      size_t nc;
+#else
       int nc;
+#endif
       *stringp = dp + 1;
+#if defined(_WIN32)
+      nc = (size_t)((size_t)dp - (size_t)mp);
+#else
       nc = (int)((unsigned long)dp - (unsigned long)mp);
+#endif
       rv = (char*)malloc(nc + 1);
       if (rv) {
         memcpy(rv, mp, nc);
diff -aur hunspell-1.4.1-orig/src/tools/munch.cxx hunspell-1.4.1/src/tools/munch.cxx
--- hunspell-1.4.1-orig/src/tools/munch.cxx	2016-05-18 02:47:17 -0400
+++ hunspell-1.4.1/src/tools/munch.cxx	2016-05-18 02:59:25 -0400
@@ -839,9 +839,17 @@
   if (n > 0) {
     char* dp = (char*)memchr(mp, (int)((unsigned char)delim), n);
     if (dp) {
+#if defined(_WIN32)
+      size_t nc;
+#else
       int nc;
+#endif
       *stringp = dp + 1;
+#if defined(_WIN32)
+      nc = (size_t)((size_t)dp - (size_t)mp);
+#else
       nc = (int)((unsigned long)dp - (unsigned long)mp);
+#endif
       rv = (char*)malloc(nc + 1);
       if (rv) {
         memcpy(rv, mp, nc);
