--- libsbml-5.18.0/src/sbml/math/FormulaParser.cpp.orig	2019-11-11 21:41:57.046424400 +0300
+++ libsbml-5.18.0/src/sbml/math/FormulaParser.cpp	2019-11-11 21:42:30.568341700 +0300
@@ -417,7 +417,7 @@
 
   while (1)
   {
-    state  = (long) Stack_peek(stack);
+    state  = (long)(intptr_t) Stack_peek(stack);
     action = FormulaParser_getAction(state, token);
 
     if (action == ACCEPT_STATE)
@@ -460,7 +460,7 @@
     {
       rule  = -action;
       node  = FormulaParser_reduceStackByRule(stack, rule);
-      state = (long) Stack_peek(stack);
+      state = (long)(intptr_t) Stack_peek(stack);
 
       Stack_push( stack, node );
       Stack_push( stack, (void *) FormulaParser_getGoto(state, rule) );
