--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -644,8 +644,8 @@
  	install(DIRECTORY "${CMAKE_BINARY_DIR}/windeployqt/" DESTINATION bin)
 endif()
 
-if(UNIX)
-  set(config_install_dir "${CMAKE_INSTALL_PREFIX}/lib/cmake/${PROJECT_NAME}")
+if(UNIX OR MINGW)
+  set(config_install_dir "lib/cmake/${PROJECT_NAME}")
 else()
   set(config_install_dir CMake)
 endif()
--- a/libosmscout-map-opengl/CMakeLists.txt
+++ b/libosmscout-map-opengl/CMakeLists.txt
@@ -48,7 +48,7 @@
 if(APPLE)
 	set(SHADER_INSTALL_DIR "/Library/Application Support/osmscout/shaders")
 	set(FONTS_INSTALL_DIR "/Library/Application Support/osmscout/fonts")
-elseif(WIN32)
+elseif(MSVC)
 	set(SHADER_INSTALL_DIR "C:/ProgramData/osmscout/shaders")
 	set(FONTS_INSTALL_DIR "C:/ProgramData/osmscout/fonts")
 else() # Linux
--- a/OSMScout2/CMakeLists.txt
+++ b/OSMScout2/CMakeLists.txt
@@ -122,7 +122,7 @@
 if(APPLE)
 	install(FILES ${TRANSLATION_QM_FILES}
 			DESTINATION "/Library/Application Support/osmscout/OSMScout2/translations")
-elseif(WIN32)
+elseif(MSVC)
 #	install(FILES ${TRANSLATION_QM_FILES}
 #			DESTINATION C:/ProgramData/osmscout/OSMScout2/translations)
 else() # Linux
