diff --git a/lib/WrFFrI.c b/lib/WrFFrI.c
index 0769f809e8d3c07d3f050a25469a828b73c37e03..1e54fe92ed82c70867367a214e0177d0f53001ea 100644
--- a/lib/WrFFrI.c
+++ b/lib/WrFFrI.c
@@ -111,13 +111,19 @@ XpmWriteFileFromXpmImage(filename, image, info)
 	name = filename;
 #else
 	if (!(name = rindex(filename, '/'))
-#ifdef AMIGA
+#if defined(AMIGA) || defined(WIN32)
 	    && !(name = rindex(filename, ':'))
 #endif
      )
 	    name = filename;
 	else
 	    name++;
+#ifdef WIN32
+	if (s = rindex(name, '\\')) {
+        s++;
+        name = s;
+    }
+#endif
 #endif
 	/* let's try to make a valid C syntax name */
 	if (index(name, '.')) {
