diff -rup gtksourceviewmm-2.10.3/gtksourceview/gtksourceviewmm/Makefile.am gtksourceviewmm-2.10.3-ricky/gtksourceview/gtksourceviewmm/Makefile.am
--- gtksourceviewmm-2.10.3/gtksourceview/gtksourceviewmm/Makefile.am	2012-05-16 00:25:22.835335900 +0800
+++ gtksourceviewmm-2.10.3-ricky/gtksourceview/gtksourceviewmm/Makefile.am	2015-10-23 14:59:11.329021800 +0800
@@ -5,7 +5,7 @@ include $(srcdir)/filelist.am
 include $(top_srcdir)/build/compile-binding.am
 
 AM_CPPFLAGS = $(binding_includes) $(binding_cppflags) $(GTHREAD_CFLAGS) $(GTKSOURCEVIEWMM_CFLAGS)
-AM_CXXFLAGS = $(GTKSOURCEVIEWMM_WXXFLAGS)
+AM_CXXFLAGS = $(GTKSOURCEVIEWMM_WXXFLAGS) -std=c++11
 
 libgtksourceviewmm_@GTKSOURCEVIEWMM_API_VERSION@_la_SOURCES = $(binding_sources)
 libgtksourceviewmm_@GTKSOURCEVIEWMM_API_VERSION@_la_LDFLAGS = -no-undefined -version-info $(GTKSOURCEVIEWMM_SO_VERSION)
diff -rup gtksourceviewmm-2.10.3/gtksourceview/gtksourceviewmm/Makefile.in gtksourceviewmm-2.10.3-ricky/gtksourceview/gtksourceviewmm/Makefile.in
--- gtksourceviewmm-2.10.3/gtksourceview/gtksourceviewmm/Makefile.in	2012-05-16 03:35:31.192869700 +0800
+++ gtksourceviewmm-2.10.3-ricky/gtksourceview/gtksourceviewmm/Makefile.in	2015-10-23 14:59:54.278595300 +0800
@@ -344,7 +344,7 @@ binding_private_HEADERS = $(files_built_
 binding_built_sources = $(files_built_cc) $(files_built_h) $(files_built_ph)
 MAINTAINERCLEANFILES = $(binding_built_sources) $(addprefix $(srcdir)/,$(binding_built_sources))
 AM_CPPFLAGS = $(binding_includes) $(binding_cppflags) $(GTHREAD_CFLAGS) $(GTKSOURCEVIEWMM_CFLAGS)
-AM_CXXFLAGS = $(GTKSOURCEVIEWMM_WXXFLAGS)
+AM_CXXFLAGS = $(GTKSOURCEVIEWMM_WXXFLAGS) -std=c++11
 libgtksourceviewmm_@GTKSOURCEVIEWMM_API_VERSION@_la_SOURCES = $(binding_sources)
 libgtksourceviewmm_@GTKSOURCEVIEWMM_API_VERSION@_la_LDFLAGS = -no-undefined -version-info $(GTKSOURCEVIEWMM_SO_VERSION)
 libgtksourceviewmm_@GTKSOURCEVIEWMM_API_VERSION@_la_LIBADD = $(GTKSOURCEVIEWMM_LIBS)
