From 3d39e3ee8bcb722c22ead6256ef88e11067fd37f Mon Sep 17 00:00:00 2001
From: GH Cao <driver1998.ms@outlook.com>
Date: Sun, 20 Apr 2025 01:26:57 +0800
Subject: Append manifest for mingw

---
 src/res/res.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/res/res.rc b/src/res/res.rc
index 26f268165..11395dac9 100644
--- a/src/res/res.rc
+++ b/src/res/res.rc
@@ -61,3 +61,5 @@ BEGIN
         END
     END
 END
+
+1 RT_MANIFEST "aegisub.exe.manifest"
\ No newline at end of file
-- 
2.49.0

