From 217b3f883569056712713c703f321cc71b3f47c8 Mon Sep 17 00:00:00 2001
From: GH Cao <driver1998.ms@outlook.com>
Date: Sun, 20 Apr 2025 00:10:35 +0800
Subject: [PATCH 3/3] subtitles_provider_csri: fix type mismatch error

---
 src/subtitles_provider_csri.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/subtitles_provider_csri.cpp b/src/subtitles_provider_csri.cpp
index e616b5868..21d2170ba 100644
--- a/src/subtitles_provider_csri.cpp
+++ b/src/subtitles_provider_csri.cpp
@@ -98,7 +98,7 @@ void CSRISubtitlesProvider::DrawSubtitles(VideoFrame &dst, double time) {
 	}
 	frame.pixfmt = CSRI_F_BGR_;
 
-	csri_fmt format = { frame.pixfmt, dst.width, dst.height };
+	csri_fmt format = { frame.pixfmt, (unsigned)dst.width, (unsigned)dst.height };
 
 	std::lock_guard<std::mutex> lock(csri_mutex);
 	if (!csri_request_fmt(instance.get(), &format))
-- 
2.47.1.windows.2

