--- scdoc-1.11.2/src/main.c.orig	2022-02-06 02:16:47.709677200 -0500
+++ scdoc-1.11.2/src/main.c	2022-02-06 02:22:05.844377900 -0500
@@ -110,7 +110,11 @@ static void parse_preamble(struct parser
 		date_time = time(NULL);
 	}
 	struct tm *date_tm = gmtime(&date_time);
+#if defined(__MINGW32__)
+	strftime(date, sizeof(date), "%Y-%m-%d", date_tm);
+#else
 	strftime(date, sizeof(date), "%F", date_tm);
+#endif
 	while ((ch = parser_getch(p)) != UTF8_INVALID) {
 		if ((ch < 0x80 && isalnum((unsigned char)ch))
 				|| ch == '_' || ch == '-' || ch == '.') {
