--- ./Makefile.am.orig	2012-08-27 16:09:02.000000000 +0200
+++ ./Makefile.am	2015-09-28 12:33:14.881629000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = src man misc
+SUBDIRS = src misc
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
--- ./src/update-desktop-database.c.orig	2013-01-07 08:59:20.000000000 +0100
+++ ./src/update-desktop-database.c	2015-09-28 12:33:56.081480800 +0200
@@ -278,7 +278,7 @@
   mask = umask(0);
   (void) umask (mask);
 
-  fchmod (fd, 0666 & ~mask);
+  //fchmod (fd, 0666 & ~mask);
 
   fp = fdopen (fd, "w+");
   if (fp == NULL)
