--- a/src/podofo/main/PdfXRef.h
+++ b/src/podofo/main/PdfXRef.h
@@ -22,7 +22,7 @@ class OutputStreamDevice;
  *
  * This is an internal class of PoDoFo used by PdfWriter.
  */
-class PdfXRef
+class PODOFO_API PdfXRef
 {
 protected:
     struct XRefItem
--- a/src/podofo/main/PdfXRefStream.h
+++ b/src/podofo/main/PdfXRefStream.h
@@ -23,7 +23,7 @@ class PdfIndirectObjectList;
  *
  * This is an internal class of PoDoFo used by PdfWriter.
  */
-class PdfXRefStream : public PdfXRef
+class PODOFO_API PdfXRefStream : public PdfXRef
 {
 public:
     /** Create a new XRef table
