--- podofo-0.9.3/test/unit/EncryptTest.h.orig	2015-10-29 09:23:29.880214600 +0300
+++ podofo-0.9.3/test/unit/EncryptTest.h	2015-10-29 10:06:57.095159600 +0300
@@ -83,7 +83,7 @@
 
  private:
   char* m_pEncBuffer;
-  long  m_lLen;
+  intptr_t  m_lLen;
   int   m_protection;
   
 };
--- podofo-0.9.3/test/unit/EncryptTest.cpp.orig	2015-10-29 09:24:54.483214600 +0300
+++ podofo-0.9.3/test/unit/EncryptTest.cpp	2015-10-29 10:08:09.945873800 +0300
@@ -208,7 +208,8 @@
     
     // Decrypt buffer
     try {
-        pEncrypt->Decrypt( pEncryptedBuffer, nOutputLen, pDecryptedBuffer, m_lLen );
+        intptr_t mlen = m_lLen;
+        pEncrypt->Decrypt( pEncryptedBuffer, nOutputLen, pDecryptedBuffer, mlen );
     } catch (PdfError &e) {
         CPPUNIT_FAIL(e.ErrorMessage(e.GetError()));
     }
