--- podofo-0.9.3/src/CMakeLists.txt.orig	2014-06-15 13:27:21.000000000 +0200
+++ podofo-0.9.3/src/CMakeLists.txt	2015-10-10 19:24:45.115148986 +0200
@@ -107,11 +107,6 @@
   doc/PdfXObject.cpp
   )
 
-IF(WIN32)
-    # If we build for windows systems, we also include the resource file
-    SET(PODOFO_DOC_SOURCES ${PODOFO_DOC_SOURCES} doc/podofo-doc.rc)
-ENDIF(WIN32)
-
 SET(PODOFO_HEADERS
   podofo-base.h
   podofo.h)
