--- podofo-0.9.5/CMakeLists.txt.orig	2017-02-01 22:23:03.000000000 +0100
+++ podofo-0.9.5/CMakeLists.txt	2017-09-26 21:20:48.123619800 +0200
@@ -379,15 +379,15 @@
 
 IF(NOT PODOFO_BUILD_LIB_ONLY)
 
-FIND_PACKAGE(CppUnit)
+# FIND_PACKAGE(CppUnit)
 
-IF(CppUnit_FOUND)
-  MESSAGE("Found cppunit. Unit tests will be built.")
-  SET(PODOFO_HAVE_CPPUNIT CppUnit_FOUND)
-  INCLUDE_DIRECTORIES(${CPPUNIT_INCLUDE_DIR})
-ELSE(CppUnit_FOUND)
-  MESSAGE("Cppunit not found. No unit tests will be built.")
-ENDIF(CppUnit_FOUND)
+# IF(CppUnit_FOUND)
+  # MESSAGE("Found cppunit. Unit tests will be built.")
+  # SET(PODOFO_HAVE_CPPUNIT CppUnit_FOUND)
+  # INCLUDE_DIRECTORIES(${CPPUNIT_INCLUDE_DIR})
+# ELSE(CppUnit_FOUND)
+  # MESSAGE("Cppunit not found. No unit tests will be built.")
+# ENDIF(CppUnit_FOUND)
 
 ENDIF(NOT PODOFO_BUILD_LIB_ONLY)
 
