diff -Naur minizip-2.7.0-orig/CMakeLists.txt minizip-2.7.0/CMakeLists.txt
--- minizip-2.7.0-orig/CMakeLists.txt	2018-10-29 02:47:53.000000000 +0300
+++ minizip-2.7.0/CMakeLists.txt	2018-11-01 10:46:19.723738900 +0300
@@ -43,8 +43,8 @@
 # Set cmake debug postfix to d
 set(CMAKE_DEBUG_POSTFIX "d")
 
-set(MINIZIP_PC ${CMAKE_CURRENT_BINARY_DIR}/minizip.pc)
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/minizip.pc.cmakein ${MINIZIP_PC} @ONLY)
+set(MINIZIP_PC ${CMAKE_CURRENT_BINARY_DIR}/minizip2.pc)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/minizip2.pc.cmakein ${MINIZIP_PC} @ONLY)
 
 set(PROJECT_NAME libminizip)
 
diff -Naur minizip-2.7.0-orig/minizip.pc.cmakein minizip-2.7.0/minizip.pc.cmakein
--- minizip-2.7.0-orig/minizip.pc.cmakein	2018-10-29 02:47:53.000000000 +0300
+++ minizip-2.7.0/minizip.pc.cmakein	1970-01-01 03:00:00.000000000 +0300
@@ -1,13 +0,0 @@
-prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@INSTALL_LIB_DIR@
-sharedlibdir=@INSTALL_LIB_DIR@
-includedir=@INSTALL_INC_DIR@
-
-Name: minizip
-Description: Minizip zip file manipulation library
-Version: @VERSION@
-
-Requires: zlib
-Libs: -L${libdir} -L${sharedlibdir} -lminizip
-Cflags: -I${includedir}
diff -Naur minizip-2.7.0-orig/minizip2.pc.cmakein minizip-2.7.0/minizip2.pc.cmakein
--- minizip-2.7.0-orig/minizip2.pc.cmakein	1970-01-01 03:00:00.000000000 +0300
+++ minizip-2.7.0/minizip2.pc.cmakein	2018-11-01 10:45:52.782491500 +0300
@@ -0,0 +1,13 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: minizip2
+Description: Minizip2 zip file manipulation library
+Version: @VERSION@
+
+Requires: zlib
+Libs: -L${libdir} -lminizip2
+Libs.private: -lz
+Cflags: -I${includedir}
