diff -Naur a/include/armadillo_bits/config.hpp.cmake b/include/armadillo_bits/config.hpp.cmake
--- a/include/armadillo_bits/config.hpp.cmake	2015-04-10 11:54:47.000000000 +0600
+++ b/include/armadillo_bits/config.hpp.cmake	2015-05-17 14:14:07.119929900 +0600
@@ -6,7 +6,18 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-
+#if defined(_WIN64)
+  #if !defined(ARMA_BLAS_LONG) && !defined(ARMA_BLAS_LONG_LONG)
+    #define ARMA_BLAS_LONG_LONG
+  #endif
+  #if !defined(ARMA_64BIT_WORD)
+    #define ARMA_64BIT_WORD
+  #endif
+#elif defined(_WIN32)
+  #if !defined(ARMA_32BIT_WORD)
+    #define ARMA_32BIT_WORD
+  #endif
+#endif
 
 #if !defined(ARMA_USE_LAPACK)
 #cmakedefine ARMA_USE_LAPACK
