--- a/cmake_aux/Modules/ARMA_FindSuperLU5.cmake
+++ b/cmake_aux/Modules/ARMA_FindSuperLU5.cmake
@@ -32,7 +32,7 @@
     
     string(REGEX REPLACE ".*version ([0-9]).*" "\\1" SLU_VERSION_MAJOR "${_slu_HEADER_CONTENTS}")
     
-    if("${SLU_VERSION_MAJOR}" EQUAL "5")
+    if("${SLU_VERSION_MAJOR}" GREATER_EQUAL "5")
       set(SuperLU_FOUND YES)
     else()
       if("${SLU_VERSION_MAJOR}" EQUAL "4")
