diff -urN armadillo-11.0.1/cmake_aux/Modules/ARMA_FindSuperLU5.cmake.orig armadillo-11.0.1/cmake_aux/Modules/ARMA_FindSuperLU5.cmake
--- armadillo-11.0.1/cmake_aux/Modules/ARMA_FindSuperLU5.cmake.orig	2016-06-16 18:24:41.000000000 +0200
+++ armadillo-11.0.1/cmake_aux/Modules/ARMA_FindSuperLU5.cmake	2022-04-20 19:02:20.871178800 +0200
@@ -6,15 +6,8 @@
 #  SuperLU_INCLUDE_DIR  - directory of SuperLU headers
 
 find_path(SuperLU_INCLUDE_DIR slu_ddefs.h
-  /usr/include/superlu/
-  /usr/include/SuperLU/
-  /usr/include/
-  /usr/local/include/superlu/
-  /usr/local/include/SuperLU/
-  /usr/local/include/
-  /opt/local/include/superlu/
-  /opt/local/include/SuperLU/
-  /opt/local/include/
+  PATHS ${CMAKE_SYSTEM_INCLUDE_PATH} /usr/include /usr/local/include /opt/local/include
+  PATH_SUFFIXES superlu SuperLU ""
 )
 
 find_library(SuperLU_LIBRARY
