--- uhttpmock-0.3.0-orig/libuhttpmock/uhm-server.c	2014-06-18 01:03:41.000000000 +0200
+++ uhttpmock-0.3.0/libuhttpmock/uhm-server.c	2014-08-10 15:29:26.737015000 +0200
@@ -49,10 +49,22 @@
 #include <glib/gi18n.h>
 #include <libsoup/soup.h>
 #include <string.h>
+#ifdef G_OS_WIN32
+#include <winsock2.h>
+#include <stdint.h>
+#define htonl(_val) ( ((uint16_t)(_val) & 0xff00) >> 8 | ((uint16_t)(_val) & 0xff) << 8 )
+#define htons(_val) ( ((uint32_t)(_val) & 0xff000000) >> 24 | \
+                      ((uint32_t)(_val) & 0x00ff0000) >>  8 | \
+                      ((uint32_t)(_val) & 0x0000ff00) <<  8 | \
+                      ((uint32_t)(_val) & 0x000000ff) << 24 )
+#else
 #include <arpa/inet.h>
+#endif
 #include <sys/types.h>
+#ifndef G_OS_WIN32
 #include <netinet/in.h>
 #include <sys/socket.h>
+#endif
 
 #include "uhm-default-tls-certificate.h"
 #include "uhm-resolver.h"
