From 2b9cff41aa6dbd8105da7d2458c8b58e9ec25e9c Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 27 Feb 2014 17:06:21 +0000
Subject: [PATCH 2/4] Define QT_WA() as unicode

---
 src/QtSolutions/qtlockedfile_win.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/QtSolutions/qtlockedfile_win.cpp b/src/QtSolutions/qtlockedfile_win.cpp
index 4cd2003..6aa3921 100755
--- a/src/QtSolutions/qtlockedfile_win.cpp
+++ b/src/QtSolutions/qtlockedfile_win.cpp
@@ -45,6 +45,10 @@
 // Maximum number of concurrent read locks. Must not be greater than MAXIMUM_WAIT_OBJECTS
 #define MAX_READERS MAXIMUM_WAIT_OBJECTS
 
+#if QT_VERSION >= 0x050000
+#define QT_WA(unicode, ansi) unicode
+#endif
+
 Qt::HANDLE QtLockedFile::getMutexHandle(int idx, bool doCreate)
 {
     if (mutexname.isEmpty()) {
-- 
1.9.0

