From 01631c14028dd83ab25e8c1b68488ee59624a94a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 27 Dec 2014 14:58:27 +0000
Subject: [PATCH 03/14] Win32: Link to Windows Terminal Services (wtsapi32)

---
 octopi.pro | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/octopi.pro b/octopi.pro
index 951a481..7bbc964 100644
--- a/octopi.pro
+++ b/octopi.pro
@@ -14,6 +14,8 @@ OBJECTS_DIR += build
 MOC_DIR += build
 UI_DIR += build
 
+win32: LIBS += -lwtsapi32
+
 HEADERS += src/QtSolutions/qtsingleapplication.h \
         src/QtSolutions/qtlocalpeer.h \
         repoeditor/repoentry.h \
-- 
2.8.2

