From 1a9c8fc2e4d7547a4f0f10b01dfd57b4ffcda31f Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 27 Feb 2014 17:15:34 +0000
Subject: [PATCH 4/5] Hack out ProcessWrapper::onSingleShot() to do nothing on
 _WIN32

---
 src/utils/processwrapper.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/utils/processwrapper.cpp b/src/utils/processwrapper.cpp
index 57f157f..dc59800 100644
--- a/src/utils/processwrapper.cpp
+++ b/src/utils/processwrapper.cpp
@@ -76,6 +76,8 @@ void ProcessWrapper::onProcessStarted()
  */
 void ProcessWrapper::onSingleShot()
 {
+#if defined(_WIN32)
+#else
   QProcess proc;
   QProcess pAux;
   QString saux;
@@ -121,6 +123,7 @@ void ProcessWrapper::onSingleShot()
       }
     }
   }
+#endif
 
   emit finishedTerminal(0, QProcess::NormalExit);
 }
-- 
1.9.0

