From c8efce3b7bc5f35e32dbfb41265102fd80e8f2fc Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 21 Sep 2014 13:49:06 +0100
Subject: [PATCH 5/5] Hack around crash on files tab due to null pointer

---
 src/mainwindow_refresh.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/mainwindow_refresh.cpp b/src/mainwindow_refresh.cpp
index a94bfe7..ad1b92f 100644
--- a/src/mainwindow_refresh.cpp
+++ b/src/mainwindow_refresh.cpp
@@ -1086,7 +1086,10 @@ void MainWindow::refreshTabFiles(bool clearContents, bool neverQuit)
         }
         while ( parent != fakeRoot );
 
-        parent->appendRow ( item );
+        /* This isn't the correct fix for the files tab crash
+         * but it allows us to continue. */
+        if (parent != 0)
+          parent->appendRow ( item );
       }
 
       lastItem = item;
-- 
2.1.0

