From 43e2e0cb559605e3fab7c708b54e360b688dd752 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 28 Sep 2014 23:15:23 +0100
Subject: [PATCH 05/13] Append to instead of overwrite m_readAllStandardOutput

---
 src/unixcommand.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/unixcommand.cpp b/src/unixcommand.cpp
index dff3ffb..7badd37 100644
--- a/src/unixcommand.cpp
+++ b/src/unixcommand.cpp
@@ -660,7 +660,7 @@ void UnixCommand::executeCommandAsNormalUser(const QString &pCommand)
 void UnixCommand::processReadyReadStandardOutput()
 {
   if (m_process->isOpen())
-    m_readAllStandardOutput = m_process->readAllStandardOutput();
+    m_readAllStandardOutput += m_process->readAllStandardOutput();
 }
 
 /*
-- 
2.1.1

