From 663a6b5a4445a7ad2aca02b05d127d8890d8e033 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 27 Feb 2014 23:21:00 +0000
Subject: [PATCH 5/5] Hardcode relative path to sh.exe as getSUCommand() for
 _WIN32

---
 src/wmhelper.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/wmhelper.cpp b/src/wmhelper.cpp
index 733c86d..fb1c2fe 100644
--- a/src/wmhelper.cpp
+++ b/src/wmhelper.cpp
@@ -269,6 +269,9 @@ QString WMHelper::getGKSUCommand(){
 QString WMHelper::getSUCommand(){
   QString result(ctn_NO_SU_COMMAND);
 
+#if defined(_WIN32)
+  return QCoreApplication::applicationDirPath()+QString("/../../bin/sh.exe -c ");
+#endif
   if (isXFCERunning() && (UnixCommand::hasTheExecutable(ctn_GKSU_2))){
     result = getGKSUCommand();
   }
-- 
1.9.0

