From 451ffd86a8f7dae3675db0863f599f399447c39a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 28 Sep 2014 23:15:23 +0100
Subject: [PATCH 06/14] Append to instead of overwrite m_readAllStandardOutput

---
 src/unixcommand.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/unixcommand.cpp b/src/unixcommand.cpp
index 438333d..70b739b 100644
--- a/src/unixcommand.cpp
+++ b/src/unixcommand.cpp
@@ -697,7 +697,7 @@ void UnixCommand::executeCommandAsNormalUser(const QString &pCommand)
 void UnixCommand::processReadyReadStandardOutput()
 {
   if (m_process->isOpen())
-    m_readAllStandardOutput = m_process->readAllStandardOutput();
+    m_readAllStandardOutput += m_process->readAllStandardOutput();
 }
 
 /*
-- 
2.4.2

