From 0669400a5a6d03c1952058a243eda07e421d0af3 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 27 Sep 2014 15:44:55 +0100
Subject: [PATCH 09/10] MSYS2: Hack around crash in refreshTabFiles

---
 src/mainwindow_refresh.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/mainwindow_refresh.cpp b/src/mainwindow_refresh.cpp
index a94bfe7..ad1b92f 100644
--- a/src/mainwindow_refresh.cpp
+++ b/src/mainwindow_refresh.cpp
@@ -1086,7 +1086,10 @@ void MainWindow::refreshTabFiles(bool clearContents, bool neverQuit)
         }
         while ( parent != fakeRoot );
 
-        parent->appendRow ( item );
+        /* This isn't the correct fix for the files tab crash
+         * but it allows us to continue. */
+        if (parent != 0)
+          parent->appendRow ( item );
       }
 
       lastItem = item;
-- 
2.1.1

