From 6b787380ce8f98bd27b4352d9435df9436be0a83 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 12 Nov 2021 13:17:41 +0530
Subject: [PATCH 4/5] fix python

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e06a3bd0..8501a448 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -234,8 +234,8 @@ if(MINGW)
     if(ENABLE_PYTHON_2)
         message(FATAL_ERROR "Python 2.7 is not supported on Windows at the moment.")
     else(ENABLE_PYTHON_2)
-        find_package(PythonInterp 3.8 EXACT)
-        find_package(PythonLibs 3.8 EXACT)
+        find_package(PythonInterp 3.8)
+        find_package(PythonLibs 3.8)
     endif(ENABLE_PYTHON_2)
     if (PYTHONLIBS_FOUND AND PYTHONINTERP_FOUND)
         if(ENABLE_PYTHON_2)
-- 
2.33.1.windows.1

