From d4c24332d5ba3df3a7d7505525b18e9535ccef05 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 12 Nov 2021 13:30:52 +0530
Subject: [PATCH 5/5] don't install installer

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8501a448..b2e196d9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -917,10 +917,6 @@ configure_file(config-ocio.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-ocio.h )
 check_function_exists(powf HAVE_POWF)
 configure_file(config-powf.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-powf.h)
 
-if(WIN32)
-    include(${CMAKE_CURRENT_LIST_DIR}/packaging/windows/installer/ConfigureInstallerNsis.cmake)
-endif()
-
 message("\nBroken tests:")
 foreach(tst ${KRITA_BROKEN_TESTS})
     message("    * ${tst}")
-- 
2.33.1.windows.1

