--- a/libs/global/KisWindowsPackageUtils.cpp
+++ b/libs/global/KisWindowsPackageUtils.cpp
@@ -67,8 +67,10 @@
 // ---
 // KF_FLAG_RETURN_FILTER_REDIRECTION_TARGET
 // shlobj_core.h / Windows 10 v1703
+// mingw-w64 shlobj.h
 // ---
-#ifndef KF_FLAG_RETURN_FILTER_REDIRECTION_TARGET
+#if !defined(KF_FLAG_RETURN_FILTER_REDIRECTION_TARGET) && !defined(__MINGW64_VERSION_MAJOR)
+// mingw-w64 already defines this in shlobj.h as part of KNOWN_FOLDER_FLAG enum
 constexpr int KF_FLAG_RETURN_FILTER_REDIRECTION_TARGET = 0x00040000;
 #endif
