--- a/libs/flake/KoShapeBulkActionLock.h
+++ b/libs/flake/KoShapeBulkActionLock.h
@@ -12,6 +12,11 @@
 #include <KisAdaptedLock.h>
 #include <QList>
 #include <QRect>
 
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
 class KoShape;
 struct KoShapeBulkActionInterface;
--- a/plugins/paintops/libpaintop/KisSensorPackInterface.h
+++ b/plugins/paintops/libpaintop/KisSensorPackInterface.h
@@ -9,6 +9,7 @@
 
 #include "kritapaintop_export.h"
 #include <QSharedData>
+#include <vector>
 
 struct KisSensorData;
 struct KisCurveOptionDataCommon;
