diff -ru enchant-2.2.3_ori/src/lib.c enchant-2.2.3/src/lib.c
--- enchant-2.2.3_ori/src/lib.c	2018-02-04 18:00:53.000000000 +0100
+++ enchant-2.2.3/src/lib.c	2018-08-14 16:26:44.587391300 +0200
@@ -1006,6 +1006,13 @@
 {
 	g_return_val_if_fail (g_module_supported (), NULL);

+#ifdef G_OS_WIN32
+    // gnulib's relocation does not work properly on win32 without setting the prefix explicitly
+    gchar *module_dir = g_win32_get_package_installation_directory_of_module(NULL);
+    enchant_set_prefix_dir(module_dir);
+    g_free(module_dir);
+#endif
+
 	EnchantBroker *broker = g_new0 (EnchantBroker, 1);
 	broker->dict_map = g_hash_table_new_full (g_str_hash, g_str_equal,
 						  g_free, enchant_dict_destroyed);
