--- a/providers/enchant_voikko.c
+++ b/providers/enchant_voikko.c
@@ -50,7 +50,7 @@
 static int
 voikko_dict_check (EnchantDict * me, const char *const word, size_t len)
 {
-	char *word_nul = strndup(word, len);
+	char *word_nul = g_strndup(word, len);
 	int result = voikkoSpellCstr((struct VoikkoHandle *)me->user_data, word_nul);
 	free(word_nul);
 	if (result == VOIKKO_SPELL_FAILED)
@@ -65,7 +65,7 @@
 voikko_dict_suggest (EnchantDict * me, const char *const word,
 		     size_t len, size_t * out_n_suggs)
 {
-	char *word_nul = strndup(word, len);
+	char *word_nul = g_strndup(word, len);
 	char **voikko_sugg_arr = voikkoSuggestCstr((struct VoikkoHandle *)me->user_data, word_nul);
 	free(word_nul);
 	if (voikko_sugg_arr == NULL)
