--- qca/cmake/modules/FindNss.cmake.orig	2014-12-13 15:22:53.153623400 +0300
+++ qca/cmake/modules/FindNss.cmake	2014-12-13 15:24:46.642114600 +0300
@@ -17,10 +17,8 @@
   SET(NSS_FOUND TRUE)
 
 else()
-  if(NOT WIN32)
-    find_package(PkgConfig)
-    pkg_search_module(NSS nss)
-  endif(NOT WIN32)
+  find_package(PkgConfig)
+  pkg_search_module(NSS nss)
 
   if (NSS_FOUND)
     if (NOT Nss_FIND_QUIETLY)
