--- qca/CMakeLists.txt.orig	2015-07-16 11:01:38.665447700 +0300
+++ qca/CMakeLists.txt	2015-07-16 11:19:17.252578400 +0300
@@ -277,10 +277,11 @@
 endif (APPLE)
 
 message(STATUS "Checking for certstore..")
+if(NOT QCA_SYSTEMSTORE_PATH)
 # fixme add OR mac
-if( WIN32 )
+if( MSVC )
 # USE BUILTIN
-else ( WIN32 )
+else ( MSVC )
   if ( DEFINED ENV{QC_CERTSTORE_PATH} )
     if(EXISTS $ENV{QC_CERTSTORE_PATH})
       set( qca_CERTSTORE $ENV{QC_CERTSTORE_PATH})
@@ -302,7 +303,9 @@
       endif(EXISTS ${_current_try})
     endforeach (_current_try)
   endif( DEFINED ENV{QC_CERTSTORE_PATH} )
-endif(WIN32)
+else()
+ set( qca_CERTSTORE ${QCA_SYSTEMSTORE_PATH})
+endif(MSVC)
 
 if (qca_CERTSTORE)
   message(STATUS "Found system certstore")
@@ -316,6 +317,10 @@
 endif (qca_CERTSTORE)
 message(STATUS "certstore path: " ${qca_CERTSTORE})
 add_definitions( -DQCA_SYSTEMSTORE_PATH="${qca_CERTSTORE}" )
+else()
+message(STATUS "certstore path: " ${QCA_SYSTEMSTORE_PATH})
+add_definitions( -DQCA_SYSTEMSTORE_PATH="${qca_CERTSTORE}" )
+endif()
 
 
 set( private_HEADERS ${qca_INCLUDEDIR}/QtCrypto/qca_plugin.h ${qca_INCLUDEDIR}/QtCrypto/qca_systemstore.h )
--- qca/src/CMakeLists.txt.orig	2015-07-16 11:09:16.212688400 +0300
+++ qca/src/CMakeLists.txt	2015-07-16 11:16:34.878817400 +0300
@@ -52,7 +52,7 @@
 	support/syncthread.cpp
 )
 
-IF (WIN32)
+IF (WIN32 AND NOT MINGW)
   SET( nonmoc_SOURCES ${nonmoc_SOURCES} qca_systemstore_win.cpp )
 elseif(APPLE)
   set( nonmoc_SOURCES ${nonmoc_SOURCES} qca_systemstore_mac.cpp)
--- qca/src/qca_systemstore_flatfile.cpp.orig	2015-07-16 11:16:51.678137300 +0300
+++ qca/src/qca_systemstore_flatfile.cpp	2015-07-16 11:35:28.073486600 +0300
@@ -21,6 +21,7 @@
 #include "qca_systemstore.h"
 
 #include <QFile>
+#include <QCoreApplication>
 
 namespace QCA {
 
@@ -32,7 +32,12 @@
 
 CertificateCollection qca_get_systemstore(const QString &provider)
 {
+#ifdef Q_OS_WIN
+	QString certstore = QString("%1%2").arg(QCoreApplication::applicationDirPath()).arg(QCA_SYSTEMSTORE_PATH);
+	return CertificateCollection::fromFlatTextFile(certstore, 0, provider);
+#else
 	return CertificateCollection::fromFlatTextFile(QCA_SYSTEMSTORE_PATH, 0, provider);
+#endif
 }
 
 }
