diff -urN bullet3-3.21-orig/examples/RobotSimulator/CMakeLists.txt bullet3-3.21/examples/RobotSimulator/CMakeLists.txt
--- bullet3-3.21-orig/examples/RobotSimulator/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/examples/RobotSimulator/CMakeLists.txt	2022-01-25 14:20:30.553799700 +0800
@@ -75,7 +75,7 @@
 
 IF(WIN32)
 	IF(BUILD_ENET OR BUILD_CLSOCKET)
-		TARGET_LINK_LIBRARIES(App_RobotSimulator ws2_32 Winmm)
+		TARGET_LINK_LIBRARIES(App_RobotSimulator clsocket ws2_32 winmm)
 	ENDIF(BUILD_ENET OR BUILD_CLSOCKET)
 ENDIF(WIN32)
 
@@ -96,7 +96,7 @@
 	MinitaurSetup.h
 )
 
-ADD_EXECUTABLE(App_RobotSimulator_NoGUI  ${RobotSimulator_NoGUI_SRCS})
+ADD_EXECUTABLE(App_RobotSimulator_NoGUI  ${RobotSimulator_NoGUI_SRCS} ${RobotSimulator_Shared_SRCS})
 
 SET_TARGET_PROPERTIES(App_RobotSimulator_NoGUI PROPERTIES VERSION ${BULLET_VERSION})
 SET_TARGET_PROPERTIES(App_RobotSimulator_NoGUI PROPERTIES DEBUG_POSTFIX "_d")
@@ -106,7 +106,7 @@
 
 IF(WIN32)
 	IF(BUILD_ENET OR BUILD_CLSOCKET)
-		TARGET_LINK_LIBRARIES(App_RobotSimulator_NoGUI ws2_32 Winmm)
+		TARGET_LINK_LIBRARIES(App_RobotSimulator_NoGUI clsocket ws2_32 winmm)
 	ENDIF(BUILD_ENET OR BUILD_CLSOCKET)
 ELSE()
 	IF(APPLE)
@@ -133,7 +133,7 @@
         HelloBulletRobotics.cpp
 )
 
-ADD_EXECUTABLE(App_HelloBulletRobotics  ${App_HelloBulletRobotics_SRCS})
+ADD_EXECUTABLE(App_HelloBulletRobotics  ${App_HelloBulletRobotics_SRCS} ${RobotSimulator_Shared_SRCS})
 
 SET_TARGET_PROPERTIES(App_HelloBulletRobotics PROPERTIES VERSION ${BULLET_VERSION})
 SET_TARGET_PROPERTIES(App_HelloBulletRobotics PROPERTIES DEBUG_POSTFIX "_d")
@@ -143,7 +143,7 @@
 
 IF(WIN32)
         IF(BUILD_ENET OR BUILD_CLSOCKET)
-                TARGET_LINK_LIBRARIES(App_HelloBulletRobotics ws2_32 Winmm)
+                TARGET_LINK_LIBRARIES(App_HelloBulletRobotics clsocket ws2_32 winmm)
         ENDIF(BUILD_ENET OR BUILD_CLSOCKET)
 ELSE()
 	IF(NOT APPLE)
diff -urN bullet3-3.21-orig/examples/ThirdPartyLibs/clsocket/CMakeLists.txt bullet3-3.21/examples/ThirdPartyLibs/clsocket/CMakeLists.txt
--- bullet3-3.21-orig/examples/ThirdPartyLibs/clsocket/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/examples/ThirdPartyLibs/clsocket/CMakeLists.txt	2022-01-24 00:46:18.827426400 +0800
@@ -33,12 +33,13 @@
     add_definitions(${OSDEF})
 elseif(WIN32)
     add_definitions(-DWIN32)
-    SET(PROJECT_LIBS Ws2_32.lib)
     if(MINGW)
         # Special MINGW stuff here
+        SET(PROJECT_LIBS ws2_32 winmm)
     elseif(MSVC)
         # Special MSVC stuff here
         add_definitions(-D_WINSOCK_DEPRECATED_NO_WARNINGS)
+        SET(PROJECT_LIBS Ws2_32.lib)
     else()
         # No idea what it is, but there's a good chance it's too weird.
         MESSAGE( FATAL_ERROR "Using unknown WIN32 compiler... NOT. Please add to build system." )
diff -urN bullet3-3.21-orig/examples/TwoJoint/CMakeLists.txt bullet3-3.21/examples/TwoJoint/CMakeLists.txt
--- bullet3-3.21-orig/examples/TwoJoint/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/examples/TwoJoint/CMakeLists.txt	2022-01-25 14:23:57.763651400 +0800
@@ -177,7 +177,7 @@
 
 IF(WIN32)
 	IF(BUILD_ENET OR BUILD_CLSOCKET)
-		TARGET_LINK_LIBRARIES(App_TwoJoint ws2_32 )
+		TARGET_LINK_LIBRARIES(App_TwoJoint clsocket ws2_32 winmm)
 	ENDIF(BUILD_ENET OR BUILD_CLSOCKET)
 ENDIF(WIN32)
 
diff -urN bullet3-3.21-orig/examples/ThirdPartyLibs/Gwen/CMakeLists.txt bullet3-3.21/examples/ThirdPartyLibs/Gwen/CMakeLists.txt
--- bullet3-3.21-orig/examples/ThirdPartyLibs/Gwen/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/examples/ThirdPartyLibs/Gwen/CMakeLists.txt	2022-01-24 21:17:20.852907000 +0800
@@ -15,7 +15,11 @@
         ADD_DEFINITIONS("-DDYNAMIC_LOAD_X11_FUNCTIONS=1")
 ENDIF()
 
+IF(MSVC)
 ADD_DEFINITIONS( -DGLEW_STATIC -DGWEN_COMPILE_STATIC -D_HAS_EXCEPTIONS=0 )
+ELSE()
+ADD_DEFINITIONS( -DGLEW_STATIC )
+ENDIF()
 
 FILE(GLOB gwen_SRCS "*.cpp" "Controls/*.cpp" "Controls/Dialog/*.cpp" "Controls/Dialogs/*.cpp" "Controls/Layout/*.cpp" "Controls/Property/*.cpp" "Input/*.cpp" "Platforms/*.cpp" "Renderers/*.cpp" "Skins/*.cpp")
 FILE(GLOB gwen_HDRS "*.h" "Controls/*.h" "Controls/Dialog/*.h" "Controls/Dialogs/*.h" "Controls/Layout/*.h" "Controls/Property/*.h" "Input/*.h" "Platforms/*.h" "Renderers/*.h" "Skins/*.h")
@@ -23,7 +27,7 @@
 ADD_LIBRARY(gwen ${gwen_SRCS} ${gwen_HDRS})
 IF (BUILD_SHARED_LIBS)
 IF(WIN32 OR APPLE)
-  target_link_libraries(gwen ${OPENGL_gl_LIBRARY})
+	TARGET_LINK_LIBRARIES(gwen ${OPENGL_gl_LIBRARY} ${OPENGL_glu_LIBRARY})
 ENDIF()
 ENDIF()
 
diff -urN bullet3-3.21-orig/Extras/BulletRobotics/CMakeLists.txt bullet3-3.21/Extras/BulletRobotics/CMakeLists.txt
--- bullet3-3.21-orig/Extras/BulletRobotics/CMakeLists.txt	2022-01-17 18:47:52.064525200 +0800
+++ bullet3-3.21/Extras/BulletRobotics/CMakeLists.txt	2022-01-24 21:21:09.276972100 +0800
@@ -285,6 +285,11 @@
 	TARGET_LINK_LIBRARIES(BulletRobotics BulletInverseDynamicsUtils BulletWorldImporter BulletFileLoader BulletSoftBody BulletDynamics BulletCollision BulletInverseDynamics LinearMath Bullet3Common)
 ENDIF (BUILD_SHARED_LIBS)
 
+IF(WIN32)
+	IF(BUILD_ENET OR BUILD_CLSOCKET)
+		TARGET_LINK_LIBRARIES(BulletRobotics clsocket ws2_32 winmm)
+	ENDIF(BUILD_ENET OR BUILD_CLSOCKET)
+ENDIF(WIN32)
   
 
 INSTALL (
