diff -urN bullet3-3.24/examples/ThirdPartyLibs/Gwen/Renderers/OpenGL_DebugFont.cpp.orig bullet3-3.24/examples/ThirdPartyLibs/Gwen/Renderers/OpenGL_DebugFont.cpp
--- bullet3-3.24/examples/ThirdPartyLibs/Gwen/Renderers/OpenGL_DebugFont.cpp.orig	2022-06-17 13:25:36.887284400 +0200
+++ bullet3-3.24/examples/ThirdPartyLibs/Gwen/Renderers/OpenGL_DebugFont.cpp	2022-06-17 13:25:48.207106500 +0200
@@ -13,6 +13,8 @@
 #if defined(__APPLE__) && !defined(VMDMESA)
 #include <OpenGL/OpenGL.h>
 #include <OpenGL/gl.h>
+#elif defined(__MINGW32__)
+#include <GL/gl.h>
 #else
 #include "glad/gl.h"
 #endif  //__APPLE__
