diff -urN bullet3-3.21-orig/Extras/BulletRobotics/CMakeLists.txt bullet3-3.21/Extras/BulletRobotics/CMakeLists.txt
--- bullet3-3.21-orig/Extras/BulletRobotics/CMakeLists.txt	2022-01-17 18:47:52.064525200 +0800
+++ bullet3-3.21/Extras/BulletRobotics/CMakeLists.txt	2022-01-24 21:21:09.276972100 +0800
@@ -299,6 +299,7 @@
 
 	INSTALL(TARGETS
 		BulletRobotics
+		RUNTIME DESTINATION bin
 		LIBRARY DESTINATION lib${LIB_SUFFIX}
 		ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	)
diff -urN bullet3-3.21-orig/Extras/BulletRoboticsGUI/CMakeLists.txt bullet3-3.21/Extras/BulletRoboticsGUI/CMakeLists.txt
--- bullet3-3.21-orig/Extras/BulletRoboticsGUI/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/Extras/BulletRoboticsGUI/CMakeLists.txt	2022-01-26 01:18:47.040707600 +0800
@@ -173,6 +173,7 @@
 
 	INSTALL(TARGETS
 		BulletRoboticsGUI
+		RUNTIME DESTINATION bin
 		LIBRARY DESTINATION lib${LIB_SUFFIX}
 		ARCHIVE DESTINATION lib${LIB_SUFFIX}
 	)
