diff -urN bullet3-3.21-orig/examples/SharedMemory/CMakeLists.txt bullet3-3.21/examples/SharedMemory/CMakeLists.txt
--- bullet3-3.21-orig/examples/SharedMemory/CMakeLists.txt	2021-11-25 05:59:22.317000000 +0800
+++ bullet3-3.21/examples/SharedMemory/CMakeLists.txt	2022-01-24 21:14:36.718519100 +0800
@@ -254,11 +254,13 @@
 					../MultiThreading/b3Win32ThreadSupport.h
 					${BULLET_PHYSICS_SOURCE_DIR}/build3/bullet.rc
 				)
+				IF(MSVC)
 				IF (CMAKE_CL_64)
 					LINK_DIRECTORIES(${BULLET_PHYSICS_SOURCE_DIR}/examples/ThirdPartyLibs/openvr/lib/win64)
 				ELSE(CMAKE_CL_64)
 					LINK_DIRECTORIES(${BULLET_PHYSICS_SOURCE_DIR}/examples/ThirdPartyLibs/openvr/lib/win32)
 				ENDIF(CMAKE_CL_64)
+				ENDIF()
 				ELSE(WIN32)
 
 					set_source_files_properties(../ThirdPartyLibs/openvr/samples/shared/pathtools.cpp ../StandaloneMain/hellovr_opengl_main.cpp  PROPERTIES COMPILE_FLAGS "-x objective-c++")
@@ -306,7 +308,7 @@
         )
 
 
-IF (NOT INTERNAL_CREATE_DISTRIBUTABLE_MSVC_PROJECTFILES)
+IF (MSVC AND NOT INTERNAL_CREATE_DISTRIBUTABLE_MSVC_PROJECTFILES)
 	IF (CMAKE_CL_64)
 		ADD_CUSTOM_COMMAND(
 			TARGET App_PhysicsServer_SharedMemory_VR
@@ -326,7 +328,7 @@
                 POST_BUILD
                 COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory ${BULLET_PHYSICS_SOURCE_DIR}/data ${PROJECT_BINARY_DIR}/data
         )
-ENDIF (NOT INTERNAL_CREATE_DISTRIBUTABLE_MSVC_PROJECTFILES)
+ENDIF (MSVC AND NOT INTERNAL_CREATE_DISTRIBUTABLE_MSVC_PROJECTFILES)
 
 
 SET_TARGET_PROPERTIES(App_PhysicsServer_SharedMemory_VR PROPERTIES  COMPILE_DEFINITIONS BT_ENABLE_VR )
