diff -urN bullet3-3.24/examples/TwoJoint/CMakeLists.txt.orig bullet3-3.24/examples/TwoJoint/CMakeLists.txt
--- bullet3-3.24/examples/TwoJoint/CMakeLists.txt.orig	2022-04-26 04:39:24.000000000 +0200
+++ bullet3-3.24/examples/TwoJoint/CMakeLists.txt	2022-06-19 13:56:43.406584400 +0200
@@ -149,7 +149,7 @@
 			${BULLET_PHYSICS_SOURCE_DIR}/btgui/OpenGLWindow/GlewWindows
 	)
 	ADD_DEFINITIONS(-DGLEW_STATIC)
-	LINK_LIBRARIES( ${OPENGL_gl_LIBRARY} ${OPENGL_glu_LIBRARY} )
+	LINK_LIBRARIES( ${OPENGL_gl_LIBRARY} ${OPENGL_glu_LIBRARY} winmm)
 ELSE(WIN32)
 	IF(APPLE)
 		find_library(COCOA NAMES Cocoa)
