diff -urN bullet3-3.21-orig/examples/pybullet/CMakeLists.txt bullet3-3.21/examples/pybullet/CMakeLists.txt
--- bullet3-3.21-orig/examples/pybullet/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/examples/pybullet/CMakeLists.txt	2022-01-24 19:43:20.996547300 +0800
@@ -26,6 +26,7 @@
 	
 SET(pybullet_SRCS
 	pybullet.c
+	../Importers/ImportURDFDemo/urdfStringSplit.cpp
 )
 
 IF(BUILD_CLSOCKET)
@@ -67,7 +68,7 @@
 
 IF(WIN32)
 	IF(BUILD_ENET OR BUILD_CLSOCKET)
-		TARGET_LINK_LIBRARIES(pybullet ws2_32 )
+		TARGET_LINK_LIBRARIES(pybullet clsocket ws2_32 winmm )
 	ENDIF()
 
 	SET_TARGET_PROPERTIES(pybullet PROPERTIES SUFFIX ".pyd" )
@@ -88,11 +89,18 @@
 ENDIF ()
 # else Linux: dont link
 
-IF(WIN32)
+IF(MSVC)
   SET(PYTHON_SITE_PACKAGES Lib/site-packages CACHE PATH "Python install path")
 ELSE()
   SET(PYTHON_SITE_PACKAGES lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages CACHE PATH "Python install path")
 ENDIF()
 
-INSTALL(TARGETS pybullet DESTINATION ${PYTHON_SITE_PACKAGES})
+INSTALL(TARGETS pybullet
+	RUNTIME DESTINATION ${PYTHON_SITE_PACKAGES}
+	COMPONENT pybullet EXCLUDE_FROM_ALL
+	)
 
+INSTALL(DIRECTORY gym/
+	DESTINATION ${PYTHON_SITE_PACKAGES}
+	COMPONENT pybullet EXCLUDE_FROM_ALL
+	)
