diff -urN bullet3-3.21-orig/examples/RobotSimulator/CMakeLists.txt bullet3-3.21/examples/RobotSimulator/CMakeLists.txt
--- bullet3-3.21-orig/examples/RobotSimulator/CMakeLists.txt	2021-11-25 05:59:22.827000000 +0800
+++ bullet3-3.21/examples/RobotSimulator/CMakeLists.txt	2022-01-26 01:14:36.932667500 +0800
@@ -15,6 +15,35 @@
 	MinitaurSetup.cpp
 	MinitaurSetup.h
 	../../examples/ExampleBrowser/InProcessExampleBrowser.cpp
+	../Importers/ImportColladaDemo/LoadMeshFromCollada.cpp
+	../Importers/ImportMeshUtility/b3ImportMeshUtility.cpp
+	../Importers/ImportMJCFDemo/BulletMJCFImporter.cpp
+	../Importers/ImportObjDemo/LoadMeshFromObj.cpp
+	../Importers/ImportObjDemo/Wavefront2GLInstanceGraphicsShape.cpp
+	../Importers/ImportURDFDemo/BulletURDFImporter.cpp
+	../Importers/ImportURDFDemo/MyMultiBodyCreator.cpp
+	../Importers/ImportURDFDemo/URDF2Bullet.cpp
+	../Importers/ImportURDFDemo/UrdfParser.cpp
+	../Importers/ImportURDFDemo/urdfStringSplit.cpp
+        ../MultiThreading/b3ThreadSupportInterface.cpp
+        ../MultiThreading/b3Win32ThreadSupport.cpp
+	../SharedMemory/IKTrajectoryHelper
+	../SharedMemory/InProcessMemory.cpp
+	../SharedMemory/b3PluginManager.cpp
+	../SharedMemory/b3RobotSimulatorClientAPI_NoDirect.cpp
+	../SharedMemory/b3RobotSimulatorClientAPI_NoDirect.h
+	../SharedMemory/b3RobotSimulatorClientAPI_NoGUI.cpp
+	../SharedMemory/PhysicsClient.cpp
+	../SharedMemory/PhysicsClientSharedMemory.cpp
+	../SharedMemory/PhysicsDirect.cpp
+	../SharedMemory/PhysicsServer.cpp
+	../SharedMemory/PhysicsServerCommandProcessor.cpp
+	../SharedMemory/PhysicsServerSharedMemory.cpp
+	../SharedMemory/Win32SharedMemory.cpp
+	../ThirdPartyLibs/stb_image/stb_image.cpp
+	../ThirdPartyLibs/tinyxml2/tinyxml2.cpp
+	../ThirdPartyLibs/Wavefront/tiny_obj_loader.cpp
+	../Utils/RobotLoggingUtil.cpp
 	../SharedMemory/GraphicsServerExample.cpp
 	../SharedMemory/GraphicsClientExample.cpp
 	../SharedMemory/RemoteGUIHelper.cpp
@@ -30,6 +59,16 @@
 )	
 
 
+SET(RobotSimulator_Shared_SRCS
+        ../MultiThreading/b3ThreadSupportInterface.cpp
+        ../MultiThreading/b3Win32ThreadSupport.cpp
+	../SharedMemory/InProcessMemory.cpp
+	../SharedMemory/Win32SharedMemory.cpp
+	../SharedMemory/b3RobotSimulatorClientAPI_NoDirect.cpp
+	../SharedMemory/b3RobotSimulatorClientAPI_NoDirect.h
+	../SharedMemory/b3RobotSimulatorClientAPI_NoGUI.cpp
+	../Utils/b3Clock.cpp
+)
 
 IF(BUILD_CLSOCKET)
  ADD_DEFINITIONS(-DBT_ENABLE_CLSOCKET)
diff -urN bullet3-3.21-orig/Extras/BulletRoboticsGUI/CMakeLists.txt bullet3-3.21/Extras/BulletRoboticsGUI/CMakeLists.txt
--- bullet3-3.21-orig/Extras/BulletRoboticsGUI/CMakeLists.txt	2021-11-25 05:59:22.000000000 +0800
+++ bullet3-3.21/Extras/BulletRoboticsGUI/CMakeLists.txt	2022-01-26 01:18:47.040707600 +0800
@@ -107,6 +107,34 @@
 	../../examples/SharedMemory/PhysicsServerExample.cpp
 	../../examples/SharedMemory/PhysicsServerExampleBullet2.cpp
 	../../examples/SharedMemory/SharedMemoryInProcessPhysicsC_API.cpp		
+
+	../../examples/MultiThreading/b3ThreadSupportInterface.cpp
+	../../examples/MultiThreading/b3Win32ThreadSupport.cpp
+	../../examples/Importers/ImportMJCFDemo/BulletMJCFImporter.cpp
+	../../examples/Importers/ImportMeshUtility/b3ImportMeshUtility.cpp
+	../../examples/Importers/ImportURDFDemo/UrdfParser.cpp
+	../../examples/Importers/ImportURDFDemo/urdfStringSplit.cpp
+	../../examples/Importers/ImportURDFDemo/MyMultiBodyCreator.cpp
+	../../examples/Importers/ImportURDFDemo/BulletURDFImporter.cpp
+	../../examples/Importers/ImportObjDemo/LoadMeshFromObj.cpp
+	../../examples/Importers/ImportObjDemo/Wavefront2GLInstanceGraphicsShape.cpp
+	../../examples/Importers/ImportURDFDemo/URDF2Bullet.cpp
+	../../examples/Importers/ImportColladaDemo/LoadMeshFromCollada.cpp
+	../../examples/SharedMemory/PhysicsDirect.cpp
+	../../examples/SharedMemory/PhysicsServer.cpp
+	../../examples/SharedMemory/PhysicsClient.cpp
+	../../examples/SharedMemory/PhysicsClientSharedMemory.cpp
+	../../examples/SharedMemory/PhysicsServerCommandProcessor.cpp
+	../../examples/SharedMemory/PhysicsServerSharedMemory.cpp
+	../../examples/SharedMemory/PhysicsServerSharedMemory.h
+	../../examples/SharedMemory/InProcessMemory.cpp
+	../../examples/SharedMemory/IKTrajectoryHelper.cpp
+	../../examples/SharedMemory/Win32SharedMemory.cpp
+	../../examples/SharedMemory/b3PluginManager.cpp
+	../../examples/ThirdPartyLibs/stb_image/stb_image.cpp
+	../../examples/ThirdPartyLibs/Wavefront/tiny_obj_loader.cpp
+	../../examples/ThirdPartyLibs/tinyxml2/tinyxml2.cpp
+	../../examples/Utils/RobotLoggingUtil.cpp
 )
 
 IF(BUILD_CLSOCKET)
