diff -Naur bullet3-2.87-patched/examples/ExampleBrowser/CMakeLists.txt bullet3-2.87/examples/ExampleBrowser/CMakeLists.txt
--- bullet3-2.87-patched/examples/ExampleBrowser/CMakeLists.txt	2021-10-02 20:12:36.053368900 +0300
+++ bullet3-2.87/examples/ExampleBrowser/CMakeLists.txt	2021-10-02 20:32:28.481945200 +0300
@@ -388,3 +388,7 @@
 			SET_TARGET_PROPERTIES(App_ExampleBrowser PROPERTIES  RELWITHDEBINFO_POSTFIX "_RelWithDebugInfo")
 ENDIF(INTERNAL_ADD_POSTFIX_EXECUTABLE_NAMES)
 
+INSTALL(TARGETS BulletExampleBrowserLib
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib
+	)
diff -Naur bullet3-2.87-patched/examples/OpenGLWindow/CMakeLists.txt bullet3-2.87/examples/OpenGLWindow/CMakeLists.txt
--- bullet3-2.87-patched/examples/OpenGLWindow/CMakeLists.txt	2021-10-02 20:12:39.584501400 +0300
+++ bullet3-2.87/examples/OpenGLWindow/CMakeLists.txt	2021-10-02 21:24:30.199734100 +0300
@@ -67,3 +67,8 @@
 endif()
 
 #target_link_libraries(OpenGLWindow ${OPENGL_gl_LIBRARY})
+
+INSTALL(TARGETS OpenGLWindow
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib
+	)
diff -Naur bullet3-2.87-patched/examples/ThirdPartyLibs/BussIK/CMakeLists.txt bullet3-2.87/examples/ThirdPartyLibs/BussIK/CMakeLists.txt
--- bullet3-2.87-patched/examples/ThirdPartyLibs/BussIK/CMakeLists.txt	2021-10-02 20:12:43.256268200 +0300
+++ bullet3-2.87/examples/ThirdPartyLibs/BussIK/CMakeLists.txt	2021-10-02 20:30:55.844260400 +0300
@@ -8,3 +8,8 @@
 FILE(GLOB BussIK_HDRS "*.h" )
 
 ADD_LIBRARY(BussIK ${BussIK_SRCS} ${BussIK_HDRS})
+
+INSTALL(TARGETS BussIK
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib
+	)
diff -Naur bullet3-2.87-patched/examples/ThirdPartyLibs/Gwen/CMakeLists.txt bullet3-2.87/examples/ThirdPartyLibs/Gwen/CMakeLists.txt
--- bullet3-2.87-patched/examples/ThirdPartyLibs/Gwen/CMakeLists.txt	2021-10-02 20:12:43.803130000 +0300
+++ bullet3-2.87/examples/ThirdPartyLibs/Gwen/CMakeLists.txt	2021-10-02 20:30:08.955128000 +0300
@@ -26,3 +26,8 @@
   target_link_libraries(gwen ${OPENGL_gl_LIBRARY})
 ENDIF()
 ENDIF()
+
+INSTALL(TARGETS gwen
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib
+	)
diff -Naur bullet3-2.87-patched/examples/pybullet/CMakeLists.txt bullet3-2.87/examples/pybullet/CMakeLists.txt
--- bullet3-2.87-patched/examples/pybullet/CMakeLists.txt	2021-10-02 20:12:58.380779200 +0300
+++ bullet3-2.87/examples/pybullet/CMakeLists.txt	2021-10-02 20:18:09.844275400 +0300
@@ -159,3 +159,14 @@
 	SET_TARGET_PROPERTIES(pybullet PROPERTIES SUFFIX ".pyd" )
 ENDIF(WIN32)
 
+SET(PYTHON_SITE_PACKAGES lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages CACHE PATH "Python install path")
+
+INSTALL(TARGETS pybullet
+	RUNTIME DESTINATION ${PYTHON_SITE_PACKAGES}
+	COMPONENT pybullet EXCLUDE_FROM_ALL
+	)
+
+INSTALL(DIRECTORY gym/
+	DESTINATION ${PYTHON_SITE_PACKAGES}
+	COMPONENT pybullet EXCLUDE_FROM_ALL
+	)
