From 6d7c1a8ba616351711829f26b36ff18f5e18ee10 Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Fri, 3 Apr 2015 20:45:14 +0800
Subject: [PATCH] Fix an installation path bug

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index f60343d..4501dd8 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -919,7 +919,7 @@ art_config.h:	gen_art_config$(EXEEXT)
 	./gen_art_config > art_config.h
 
 @OS_WIN32_TRUE@install-libtool-import-lib:
-@OS_WIN32_TRUE@	$(INSTALL) .libs/libart_lgpl_2.dll.a $(DESTDIR)$(libdir)
+@OS_WIN32_TRUE@	$(INSTALL) -D .libs/libart_lgpl_2.dll.a $(DESTDIR)$(libdir)/libart_lgpl_2.dll.a
 @OS_WIN32_TRUE@uninstall-libtool-import-lib:
 @OS_WIN32_TRUE@	-rm $(DESTDIR)$(libdir)/libart_lgpl_2.dll.a
 @OS_WIN32_FALSE@install-libtool-import-lib:
-- 
2.3.4

