diff -urN parmetis-4.0.3.orig/metis/GKlib/gk_getopt.h parmetis-4.0.3/metis/GKlib/gk_getopt.h
--- parmetis-4.0.3.orig/metis/GKlib/gk_getopt.h	2013-03-30 08:24:50.000000000 -0700
+++ parmetis-4.0.3/metis/GKlib/gk_getopt.h	2019-11-11 22:48:38.248965700 -0800
@@ -52,12 +52,11 @@
 
 
 /* Function prototypes */
-extern int gk_getopt(int __argc, char **__argv, char *__shortopts);
-extern int gk_getopt_long(int __argc, char **__argv, char *__shortopts,
-              struct gk_option *__longopts, int *__longind);
-extern int gk_getopt_long_only (int __argc, char **__argv,
-              char *__shortopts, struct gk_option *__longopts, int *__longind);
-
+extern int gk_getopt(int gk_argc, char **gk_argv, char *__shortopts);
+extern int gk_getopt_long(int gk_argc, char **gk_argv, char *__shortopts,
+               struct gk_option *__longopts, int *__longind);
+extern int gk_getopt_long_only (int gk_argc, char **gk_argv,
+               char *__shortopts, struct gk_option *__longopts, int *__longind);
 
 
 #endif
