diff -ura sdl2-compat-2.32.70/CMakeLists.txt sdl2-compat-2.32.70.new/CMakeLists.txt
--- sdl2-compat-2.32.70/CMakeLists.txt	2026-06-08 16:01:06 +0100
+++ sdl2-compat-2.32.70.new/CMakeLists.txt	2026-08-23 10:21:26 +0100
@@ -381,7 +381,7 @@
   )
 endif()
 
-if(MINGW)
+if(0)
   set_property(TARGET SDL2 APPEND_STRING PROPERTY LINK_FLAGS " -nostdlib")
   if(USE_CLANG)
     if(SDL_CPU_X86)
diff -ura sdl2-compat-2.32.70/src/sdl2_compat.c sdl2-compat-2.32.70.new/src/sdl2_compat.c
--- sdl2-compat-2.32.70/src/sdl2_compat.c	2026-06-08 16:01:06 +0100
+++ sdl2-compat-2.32.70.new/src/sdl2_compat.c	2026-08-23 10:31:54 +0100
@@ -1197,7 +1197,7 @@
 __declspec(selectany) int _fltused = 1;
 #endif
 #if defined(__MINGW32__)
-#define _DllMainCRTStartup DllMainCRTStartup
+#define _DllMainCRTStartup DllMain
 #endif
 #if defined(__WATCOMC__)
 #define _DllMainCRTStartup LibMain
