From 1c27627e75bbf1c028d3e5fec504744070ee0621 Mon Sep 17 00:00:00 2001
From: Alberto Fanjul <albertofanjul@gmail.com>
Date: Fri, 19 Oct 2018 11:20:13 +0200
Subject: [PATCH] Fix vapi generation on msys2

---
 meson_vapi_link.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson_vapi_link.py b/meson_vapi_link.py
index 435cd95..a419fc0 100644
--- a/meson_vapi_link.py
+++ b/meson_vapi_link.py
@@ -4,7 +4,7 @@
 import subprocess
 import sys
 
-vapidir = os.path.join(os.environ['MESON_INSTALL_DESTDIR_PREFIX'], sys.argv[1], 'vala', 'vapi')
+vapidir = os.path.join(os.environ['MESON_INSTALL_DESTDIR_PREFIX'], 'share', 'vala', 'vapi')
 
 # FIXME: meson will not track the creation of these files
 #        https://github.com/mesonbuild/meson/blob/master/mesonbuild/scripts/uninstall.py#L39
