--- a/CMakeLists.txt.orig	2015-05-25 15:36:19.241200000 +0300
+++ b/CMakeLists.txt	2015-05-25 15:54:10.095600000 +0300
@@ -98,7 +98,7 @@
         string(TOUPPER ${conf} conf2)
         set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${conf2} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${conf}/bin)
     endforeach()
-elseif (UNIX)
+elseif (UNIX OR MINGW)
     set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${output_dir}/bin)
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${output_dir}/bin)
 endif()
@@ -113,7 +113,7 @@
 # icu
 ########################################
 
-if (UNIX)
+if (UNIX OR MINGW)
     if (NOT ICU_INCLUDE_DIR)
         find_path(ICU_INCLUDE_DIR unicode/ucnv.h
             PATHS
@@ -159,7 +159,9 @@
 set(FB_SERVICE_PORT 3050)
 
 if (WIN32)
+  if (NOT MINGW)
     set(FB_PREFIX "c:\\\\Program Files\\\\Firebird\\\\")
+  endif()
     set(FB_IPC_NAME "FIREBIRD")
 endif()
 
@@ -192,7 +194,7 @@
         endif()
     endif(MSVC)
 
-    set(LIB_Ws2_32 Ws2_32)
+    set(LIB_Ws2_32 ws2_32)
     set(LIB_comctl32 comctl32)
     set(LIB_mpr mpr)
     set(LIB_version version)
@@ -244,7 +246,7 @@
 #
 ################################################################################
 
-if (WIN32)
+if (MSVC)
     # icu
     if (NOT ICU_EXTRACT)
         message(STATUS "Extracting pre-built ICU")
--- a/src/CMakeLists.txt.orig	2015-05-25 15:56:46.615200000 +0300
+++ b/src/CMakeLists.txt	2015-05-25 16:00:00.460800000 +0300
@@ -852,7 +852,7 @@
 
 endif() # if (NOT CMAKE_CROSSCOMPILING)
 
-if (WIN32)
+if (MSVC)
     add_custom_command(
         TARGET boot_isql
         POST_BUILD