From c9e87fdeea5332f1e70e80f4a4bb62bd37259a2d Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Tue, 25 Nov 2014 12:59:43 +0300
Subject: [PATCH] our mingw-w64 have different icu libs naming.

---
 src/common/unicode_util.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/common/unicode_util.cpp b/src/common/unicode_util.cpp
index 3a5b073..65a2012 100644
--- a/src/common/unicode_util.cpp
+++ b/src/common/unicode_util.cpp
@@ -58,8 +58,13 @@ using namespace Firebird;
 
 namespace {
 #if defined(WIN_NT)
+#ifdef __MINGW32__
+const char* const inTemplate = "libicuin%s.dll";
+const char* const ucTemplate = "libicuuc%s.dll";
+#else
 const char* const inTemplate = "icuin%s.dll";
 const char* const ucTemplate = "icuuc%s.dll";
+#endif
 #elif defined(DARWIN)
 //const char* const inTemplate = "/Library/Frameworks/Firebird.framework/Versions/A/Libraries/libicui18n.dylib";
 //const char* const ucTemplate = "/Library/Frameworks/Firebird.framework/versions/A/Libraries/libicuuc.dylib";
-- 
2.2.0

From 57ad0dada3cca5dfb452bd3c39ac45ff3b1d554b Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 23:15:45 +0300
Subject: [PATCH] Fix UNIX path for mingw compiler.

---
 src/misc/writeBuildNum.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/misc/writeBuildNum.sh b/src/misc/writeBuildNum.sh
index 9f23244..d4e58f7 100755
--- a/src/misc/writeBuildNum.sh
+++ b/src/misc/writeBuildNum.sh
@@ -96,6 +96,10 @@ createMakeVersion() {
 
 OdsH="${Root}/src/jrd/ods.h"
 Mini="/tmp/miniods.h"
+if [ -n $MSYSTEM ]
+  then
+    Mini=$(cygpath -wm $Mini)
+fi
 TestCpp="/tmp/test.cpp"
 AOut="/tmp/a.out"
 
-- 
2.2.0
