From c9e87fdeea5332f1e70e80f4a4bb62bd37259a2d Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Tue, 25 Nov 2014 12:59:43 +0300
Subject: [PATCH] our mingw-w64 have different icu libs naming.

---
 src/common/unicode_util.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/common/unicode_util.cpp b/src/common/unicode_util.cpp
index 3a5b073..65a2012 100644
--- a/src/common/unicode_util.cpp
+++ b/src/common/unicode_util.cpp
@@ -58,8 +58,13 @@ using namespace Firebird;
 
 namespace {
 #if defined(WIN_NT)
+#ifdef __MINGW32__
+const char* const inTemplate = "libicuin%s.dll";
+const char* const ucTemplate = "libicuuc%s.dll";
+#else
 const char* const inTemplate = "icuin%s.dll";
 const char* const ucTemplate = "icuuc%s.dll";
+#endif
 #elif defined(DARWIN)
 //const char* const inTemplate = "/Library/Frameworks/Firebird.framework/Versions/A/Libraries/libicui18n.dylib";
 //const char* const ucTemplate = "/Library/Frameworks/Firebird.framework/versions/A/Libraries/libicuuc.dylib";
-- 
2.2.0

From 57ad0dada3cca5dfb452bd3c39ac45ff3b1d554b Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 23:15:45 +0300
Subject: [PATCH] Fix UNIX path for mingw compiler.

---
 src/misc/writeBuildNum.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/misc/writeBuildNum.sh b/src/misc/writeBuildNum.sh
index 9f23244..d4e58f7 100755
--- a/src/misc/writeBuildNum.sh
+++ b/src/misc/writeBuildNum.sh
@@ -96,6 +96,10 @@ createMakeVersion() {
 
 OdsH="${Root}/src/jrd/ods.h"
 Mini="/tmp/miniods.h"
+if [ -n $MSYSTEM ]
+  then
+    Mini=$(cygpath -wm $Mini)
+fi
 TestCpp="/tmp/test.cpp"
 AOut="/tmp/a.out"
 
-- 
2.2.0

From c89f2fdd21146ddfd95e57232c6ea47bda44a016 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Mon, 25 May 2015 15:14:47 +0300
Subject: [PATCH] cloop: Fix building with mingw.

---
 extern/cloop/Makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/extern/cloop/Makefile b/extern/cloop/Makefile
index 2056e4a..11e0901 100644
--- a/extern/cloop/Makefile
+++ b/extern/cloop/Makefile
@@ -11,7 +11,7 @@ LD	:= $(CXX)
 SRC_DIR		:= src
 BUILD_DIR	:= build
 OUT_DIR		:= output
-SHRLIB_EXT	:= .so
+SHRLIB_EXT	:= .dll
 EXE_EXT		:=
 
 OBJ_DIR := $(BUILD_DIR)/$(TARGET)
@@ -27,7 +27,7 @@ SRCS_CPP := $(foreach sdir,$(SRC_DIRS),$(wildcard $(sdir)/*.cpp))
 OBJS_C := $(patsubst $(SRC_DIR)/%.c,$(OBJ_DIR)/%.o,$(SRCS_C))
 OBJS_CPP := $(patsubst $(SRC_DIR)/%.cpp,$(OBJ_DIR)/%.o,$(SRCS_CPP))
 
-C_FLAGS := -ggdb -fPIC -MMD -MP -W -Wall -Wno-unused-parameter
+C_FLAGS := -ggdb -MMD -MP -W -Wall -Wno-unused-parameter
 CXX_FLAGS := $(C_FLAGS)
 FPC_FLAGS := -Mdelphi
 
@@ -106,23 +106,23 @@ $(BIN_DIR)/test1-c$(SHRLIB_EXT): \
 	$(OBJ_DIR)/tests/test1/CalcCApi.o \
 	$(OBJ_DIR)/tests/test1/CTest.o \
 
-	$(LD) $^ -shared -ldl -o $@
+	$(LD) $^ -shared -o $@
 
 $(BIN_DIR)/test1-c$(EXE_EXT): \
 	$(OBJ_DIR)/tests/test1/CalcCApi.o \
 	$(OBJ_DIR)/tests/test1/CTest.o \
 
-	$(LD) $^ -ldl -o $@
+	$(LD) $^ -o $@
 
 $(BIN_DIR)/test1-cpp$(SHRLIB_EXT): \
 	$(OBJ_DIR)/tests/test1/CppTest.o \
 
-	$(LD) $^ -shared -ldl -o $@
+	$(LD) $^ -shared -o $@
 
 $(BIN_DIR)/test1-cpp$(EXE_EXT): \
 	$(OBJ_DIR)/tests/test1/CppTest.o \
 
-	$(LD) $^ -ldl -o $@
+	$(LD) $^ -o $@
 
 $(BIN_DIR)/test1-pascal$(SHRLIB_EXT): \
 	$(SRC_DIR)/tests/test1/PascalClasses.pas \
@@ -130,7 +130,7 @@ $(BIN_DIR)/test1-pascal$(SHRLIB_EXT): \
 	$(SRC_DIR)/tests/test1/CalcPascalApi.pas \
 
 ifeq ($(WITH_FPC),1)
-	fpc $(FPC_FLAGS) -fPIC -FU$(OBJ_DIR)/tests/test1 -o$(BIN_DIR)/test1-pascal$(SHRLIB_EXT) $(SRC_DIR)/tests/test1/PascalLibrary.dpr
+	fpc $(FPC_FLAGS) -FU$(OBJ_DIR)/tests/test1 -o$(BIN_DIR)/test1-pascal$(SHRLIB_EXT) $(SRC_DIR)/tests/test1/PascalLibrary.dpr
 endif
 
 $(BIN_DIR)/test1-pascal$(EXE_EXT): \
-- 
2.4.1
