diff -up xpdf-4.00/xpdf/CMakeLists.txt.manfix xpdf-4.00/xpdf/CMakeLists.txt
--- xpdf-4.00/xpdf/CMakeLists.txt.manfix	2017-08-16 19:43:23.612344976 -0400
+++ xpdf-4.00/xpdf/CMakeLists.txt	2017-08-16 19:43:46.189746281 -0400
@@ -119,7 +119,7 @@ else ()
   target_link_libraries(pdftops goo fofi ${LCMS_LIBRARY} ${HAVE_PAPER_H})
 endif ()
 install(TARGETS pdftops RUNTIME DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftops.1 DESTINATION man/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftops.1 DESTINATION share/man/man1)
 
 #--- pdftotext
 
@@ -130,7 +130,7 @@ add_executable(pdftotext
 )
 target_link_libraries(pdftotext goo fofi ${LCMS_LIBRARY} ${HAVE_PAPER_H})
 install(TARGETS pdftotext RUNTIME DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftotext.1 DESTINATION man/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftotext.1 DESTINATION share/man/man1)
 
 #--- pdftohtml
 
@@ -147,7 +147,7 @@ if (HAVE_SPLASH AND PNG_FOUND)
                         ${DTYPE_LIBRARY}
                         ${LCMS_LIBRARY} ${PNG_LIBRARIES} ${HAVE_PAPER_H})
   install(TARGETS pdftohtml RUNTIME DESTINATION bin)
-  install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftohtml.1 DESTINATION man/man1)
+  install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftohtml.1 DESTINATION share/man/man1)
 endif ()
 
 #--- pdfinfo
@@ -158,7 +158,7 @@ add_executable(pdfinfo
 )
 target_link_libraries(pdfinfo goo fofi ${LCMS_LIBRARY} ${HAVE_PAPER_H})
 install(TARGETS pdfinfo RUNTIME DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/pdfinfo.1 DESTINATION man/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/pdfinfo.1 DESTINATION share/man/man1)
 
 #--- pdffonts
 
@@ -168,7 +168,7 @@ add_executable(pdffonts
 )
 target_link_libraries(pdffonts goo fofi ${LCMS_LIBRARY} ${HAVE_PAPER_H})
 install(TARGETS pdffonts RUNTIME DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/pdffonts.1 DESTINATION man/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/pdffonts.1 DESTINATION share/man/man1)
 
 #--- pdfdetach
 
@@ -178,7 +178,7 @@ add_executable(pdfdetach
 )
 target_link_libraries(pdfdetach goo fofi ${LCMS_LIBRARY} ${HAVE_PAPER_H})
 install(TARGETS pdfdetach RUNTIME DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/pdfdetach.1 DESTINATION man/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/pdfdetach.1 DESTINATION share/man/man1)
 
 #--- pdftoppm
 
@@ -193,7 +193,7 @@ if (HAVE_SPLASH)
                         ${DTYPE_LIBRARY}
                         ${LCMS_LIBRARY} ${HAVE_PAPER_H})
   install(TARGETS pdftoppm RUNTIME DESTINATION bin)
-  install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftoppm.1 DESTINATION man/man1)
+  install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftoppm.1 DESTINATION share/man/man1)
 endif ()
 
 #--- pdftopng
@@ -209,7 +209,7 @@ if (HAVE_SPLASH AND PNG_FOUND)
                         ${DTYPE_LIBRARY}
                         ${LCMS_LIBRARY} ${PNG_LIBRARIES} ${HAVE_PAPER_H})
   install(TARGETS pdftopng RUNTIME DESTINATION bin)
-  install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftopng.1 DESTINATION man/man1)
+  install(FILES ${PROJECT_SOURCE_DIR}/doc/pdftopng.1 DESTINATION share/man/man1)
 endif ()
 
 #--- pdfimages
@@ -221,8 +221,8 @@ add_executable(pdfimages
 )
 target_link_libraries(pdfimages goo fofi ${LCMS_LIBRARY} ${HAVE_PAPER_H})
 install(TARGETS pdfimages RUNTIME DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/pdfimages.1 DESTINATION man/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/pdfimages.1 DESTINATION share/man/man1)
 
 #--- xpdfrc man page
 
-install(FILES ${PROJECT_SOURCE_DIR}/doc/xpdfrc.5 DESTINATION man/man5)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/xpdfrc.5 DESTINATION share/man/man5)
diff -up xpdf-4.00/xpdf-qt/CMakeLists.txt.manfix xpdf-4.00/xpdf-qt/CMakeLists.txt
--- xpdf-4.00/xpdf-qt/CMakeLists.txt.manfix	2017-08-16 19:44:45.218180999 -0400
+++ xpdf-4.00/xpdf-qt/CMakeLists.txt	2017-08-16 19:44:50.394043748 -0400
@@ -87,5 +87,5 @@ if ((QT4_FOUND OR Qt5Widgets_FOUND)
   set_property(TARGET xpdf PROPERTY WIN32_EXECUTABLE 1)
 
   install(TARGETS xpdf RUNTIME DESTINATION bin)
-  install(FILES ${PROJECT_SOURCE_DIR}/doc/xpdf.1 DESTINATION man/man1)
+  install(FILES ${PROJECT_SOURCE_DIR}/doc/xpdf.1 DESTINATION share/man/man1)
 endif ()
