--- a/build.nims
+++ b/build.nims
@@ -188,7 +188,7 @@
         if config.isDeveloper and not flags.contains("NOWEBVIEW"):
             discard gorgeEx "src\\extras\\webview\\deps\\build.bat"
             #discard gorgeEx "src\\extras\\webview\\deps\\build-new.bat"
-        --passL:"\"-static-libstdc++ -static-libgcc -Wl,-Bstatic -lstdc++ -Wl,-Bdynamic\""
+        --passL:""
         --gcc.linkerexe:"g++"
 
     proc unixHostSpecific() =
--- a/src/extras/pfd.nim
+++ b/src/extras/pfd.nim
@@ -20,14 +20,14 @@
 #=======================================
 
 when defined(windows):
-    {.compile("pfd/pfd.cc","-std=c++11 -static-libstdc++ -DSTRSAFE_NO_DEPRECATE").}
+    {.compile("pfd/pfd.cc","-std=c++11 -DSTRSAFE_NO_DEPRECATE").}
 else:
     {.compile("pfd/pfd.cc","-std=c++11").}
 
 {.passC: "-I" & parentDir(currentSourcePath()) .}
 
 when defined(windows):
-    {.passL:"-static-libstdc++ -L -lversion -lole32 -luuid -lshell32 -luser32 -lkernel32 -lgdi32 -lcomctl32 -loleaut32".}
+    {.passL:"-L -lversion -lole32 -luuid -lshell32 -luser32 -lkernel32 -lgdi32 -lcomctl32 -loleaut32".}
 else:
     {.passL:"-lstdc++".}
 
--- a/src/extras/webview.nim
+++ b/src/extras/webview.nim
@@ -37,7 +37,7 @@
     {.passL: "-lstdc++ -framework WebKit".}
 elif defined(windows):
     {.passC: "-DWEBVIEW_EDGE=1 -mwindows".}
-    {.passL: """-static-libstdc++ -std=c++17 -L""" & currentSourcePath().splitPath.head & """/webview/deps/dlls/x64 -lwebview -lWebView2Loader""".}
+    {.passL: """-std=c++17 -lwebview""".}
 
 {.push header: "webview/webview.h", cdecl.}
 
--- a/src/library/Net.nim
+++ b/src/library/Net.nim
@@ -39,7 +39,7 @@
         else:
             {.passL: "src/deps/openssl/macos/amd64/libssl.a src/deps/openssl/macos/amd64/libcrypto.a".}
     elif defined(windows): 
-        {.passL: "-Bstatic -Lsrc/deps/openssl/windows/amd64 -lssl -lcrypto -Bdynamic -lws2_32 -lcrypt32".}
+        {.passL: "-lssl -lcrypto -lws2_32 -lcrypt32".}
 
 #=======================================
 # Libraries
--- a/src/vm/values/custom/vregex.nim
+++ b/src/vm/values/custom/vregex.nim
@@ -25,7 +25,7 @@
     else:
         {.passL: "src/deps/pcre/macos/amd64/libpcre.a".}
 elif defined(windows): 
-    {.passL: "-Bstatic -Lsrc/deps/pcre/windows/amd64 -lpcre -Bdynamic".}
+    {.passL: "-lpcre".}
 
 #=======================================
 # Libraries
