--- a/makefile.in
+++ b/makefile.in
@@ -17,7 +17,7 @@
 bindir=@bindir@
 libdir=@libdir@
 includedir=@includedir@
-pkgconfigexecdir=${prefix}/libdata/pkgconfig
+pkgconfigexecdir=${prefix}/lib/pkgconfig
 
 INSTALL=@INSTALL@
 INSTALL_DATA=@INSTALL_DATA@
@@ -789,9 +789,9 @@
 	done
 
 install:
-	${MKDIR_P} ${DESTDIR}${bindir} ${DESTDIR}${includedir} ${DESTDIR}${libdir} ${DESTDIR}${pkgconfigexecdir}
+	${MKDIR_P} ${DESTDIR}${bindir} ${DESTDIR}${includedir}/nauty ${DESTDIR}${libdir} ${DESTDIR}${pkgconfigexecdir}
 	${INSTALL} ${GTOOLS} ${DESTDIR}${bindir}
-	${INSTALL_DATA} ${GTOOLSH} ${DESTDIR}${includedir}
+	${INSTALL_DATA} ${GTOOLSH} ${DESTDIR}${includedir}/nauty
 	${INSTALL} ${GLIBS} ${DESTDIR}${libdir}
 	for lib in ${GLIBS} ; do \
 	   mv -f ${DESTDIR}${libdir}/$${lib} ${DESTDIR}${libdir}/lib$${lib} ; \
--- a/nauty-pc.in
+++ b/nauty-pc.in
@@ -8,4 +8,4 @@
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} LIBS
 Libs.private:
-Cflags: -I${includedir} CFLGS
+Cflags: -I${includedir}/nauty CFLGS
