--- Vulkan-Loader-1.4.317/loader/CMakeLists.txt.orig	2025-06-10 09:49:11.820099500 +0300
+++ Vulkan-Loader-1.4.317/loader/CMakeLists.txt	2025-06-10 09:50:27.382608300 +0300
@@ -542,7 +543,7 @@
 if (PKG_CONFIG_FOUND)
     if(WIN32)
         if(MINGW)
-            set(VULKAN_LIB_SUFFIX "-1.dll")
+            set(VULKAN_LIB_SUFFIX "-1")
         else()
             set(VULKAN_LIB_SUFFIX "-1")
         endif()
