--- aspell-fr-0.50-3/configure.orig	2015-08-24 15:25:37.040164200 +0200
+++ aspell-fr-0.50-3/configure	2015-08-24 15:26:51.598072800 +0200
@@ -69,10 +69,12 @@ if test x = "x$WORD_LIST_COMPRESS"
 
 echo $ECHO_N "Finding Dictionary file location ... $ECHO_C"
 dictdir=`$ASPELL dump config dict-dir`
+dictdir=`cygpath -u $dictdir`
 echo $dictdir
 
 echo $ECHO_N "Finding Data file location ... $ECHO_C"
 datadir=`$ASPELL dump config data-dir`
+datadir=`cygpath -u $datadir`
 echo $datadir
 
 echo "ASPELL = $ASPELL" > Makefile
