diff -Naur setuptools_scm-3.1.0.orig/src/setuptools_scm/file_finder_git.py setuptools_scm-3.1.0/src/setuptools_scm/file_finder_git.py
--- setuptools_scm-3.1.0.orig/src/setuptools_scm/file_finder_git.py	2018-08-14 07:14:03.731864500 -0400
+++ setuptools_scm-3.1.0/src/setuptools_scm/file_finder_git.py	2018-08-14 07:14:58.107190100 -0400
@@ -13,7 +13,7 @@
         with open(os.devnull, "wb") as devnull:
             out = subprocess.check_output(
                 ["git", "rev-parse", "--show-toplevel"],
-                cwd=(path or "."),
+                cwd=(path),
                 universal_newlines=True,
                 stderr=devnull,
             )
diff -Naur setuptools_scm-3.1.0.orig/src/setuptools_scm/file_finder_hg.py setuptools_scm-3.1.0/src/setuptools_scm/file_finder_hg.py
--- setuptools_scm-3.1.0.orig/src/setuptools_scm/file_finder_hg.py	2018-08-14 07:14:03.731864500 -0400
+++ setuptools_scm-3.1.0/src/setuptools_scm/file_finder_hg.py	2018-08-14 07:14:52.274812200 -0400
@@ -9,7 +9,7 @@
         with open(os.devnull, "wb") as devnull:
             out = subprocess.check_output(
                 ["hg", "root"],
-                cwd=(path or "."),
+                cwd=(path),
                 universal_newlines=True,
                 stderr=devnull,
             )
diff -Naur setuptools_scm-3.1.0.orig/src/setuptools_scm/integration.py setuptools_scm-3.1.0/src/setuptools_scm/integration.py
--- setuptools_scm-3.1.0.orig/src/setuptools_scm/integration.py	2018-08-14 07:14:03.731864500 -0400
+++ setuptools_scm-3.1.0/src/setuptools_scm/integration.py	2018-08-14 07:16:09.639007700 -0400
@@ -28,7 +28,7 @@
         command = ep.load()
         if isinstance(command, str):
             # this technique is deprecated
-            res = do(ep.load(), path or ".").splitlines()
+            res = do(ep.load(), path).splitlines()
         else:
             res = command(path)
         if res:
