From a50a43709532293ade135ef1cf3356ed85f65d72 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen@syrusdark.website>
Date: Thu, 28 Jan 2021 20:30:44 +0530
Subject: [PATCH] posix path to win32 path

---
 src/setuptools_scm/utils.py | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/setuptools_scm/utils.py b/src/setuptools_scm/utils.py
index 413e98a..d80107b 100644
--- a/src/setuptools_scm/utils.py
+++ b/src/setuptools_scm/utils.py
@@ -86,6 +86,22 @@ def _popen_pipes(cmd, cwd):
         ),
     )
 
+def cyg2win(path):
+    # from https://stackoverflow.com/a/50137718/12858827
+    import os
+    import re
+    match = re.match('(/(cygdrive/)?)(.*)', path)
+    if not match:
+        return path.replace('/', '\\')
+    dirs = match.group(3).split('/')
+    dirs[0] = f'{dirs[0].upper()}:'
+    return '\\'.join(dirs)
+
+def iscygpath(path):
+    if path.startswith('/'):
+        return True
+    else:
+        return False
 
 def do_ex(cmd, cwd="."):
     trace("cmd", repr(cmd))
@@ -94,12 +110,16 @@ def do_ex(cmd, cwd="."):
 
     p = _popen_pipes(cmd, cwd)
     out, err = p.communicate()
+    out = out.decode('utf-8')
+    if iscygpath(out):
+        out = cyg2win(out)
     if out:
         trace("out", repr(out))
     if err:
         trace("err", repr(err))
     if p.returncode:
         trace("ret", p.returncode)
+    out = out.encode('utf-8')
     return ensure_stripped_str(out), ensure_stripped_str(err), p.returncode
 
 
-- 
2.29.2.windows.2

