--- origsrc/sofia-sip-1.12.11/libsofia-sip-ua/su/sofia-sip/su.h	2011-03-11 15:49:19.000000000 +0100
+++ src/sofia-sip-1.12.11/libsofia-sip-ua/su/sofia-sip/su.h	2016-08-07 22:52:58.333432900 +0200
@@ -320,9 +320,14 @@
 #endif
 
 #if SU_HAVE_WINSOCK
+#if HAVE_MINGW64
+#define su_inet_pton inet_pton
+#define su_inet_ntop inet_ntop
+#else  
 SOFIAPUBFUN int su_inet_pton(int af, char const *src, void *dst);
 SOFIAPUBFUN const char *su_inet_ntop(int af, void const *src,
 				  char *dst, size_t size);
+#endif
 SOFIAPUBFUN ssize_t
   su_send(su_socket_t s, void *buffer, size_t length, int flags),
   su_sendto(su_socket_t s, void *buffer, size_t length, int flags,
