diff --git a/src/main.cpp b/src/main.cpp
index 0c4b8cf..e835ad3 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -270,8 +270,15 @@ public:
         dir.cd("shotcut");
         dir.cd("translations");
 #elif defined(Q_OS_WIN)
-        dir.cd("share");
-        dir.cd("translations");
+        if (dir.cd("share")) {
+            dir.cd("translations");
+        } else {
+            // MSYS2 uses a filesystem layout similar to *nix
+            dir.cdUp();
+            dir.cd("share");
+            dir.cd("shotcut");
+            dir.cd("translations");
+        }
 #else
         dir.cdUp();
         dir.cd("share");
diff --git a/translations/CMakeLists.txt b/translations/CMakeLists.txt
index 952ce95..e56fda2 100644
--- a/translations/CMakeLists.txt
+++ b/translations/CMakeLists.txt
@@ -58,8 +58,8 @@ add_custom_target(translations COMMAND
 qt5_add_translation(QM_FILES ${TS_FILES})
 add_custom_target(qm ALL DEPENDS ${QM_FILES})
 
-if(UNIX AND NOT APPLE)
+# if(UNIX AND NOT APPLE)
   include(GNUInstallDirs)
   install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ DESTINATION ${CMAKE_INSTALL_DATADIR}/shotcut/translations
     FILES_MATCHING PATTERN "*.qm" PATTERN CMakeFiles EXCLUDE)
-endif()
+# endif()
