diff -Naur a/Makefile.mingw b/Makefile.mingw
--- a/Makefile.mingw	1969-12-31 21:00:00.000000000 -0300
+++ b/Makefile.mingw	2024-06-17 12:37:55.857370395 -0300
@@ -0,0 +1,38 @@
+LUA_VERSION=5.1
+LUA_VERSION_SHORT=51
+
+PREFIX=/usr/local
+LUA_BIN=$(PREFIX)/bin
+LUA_INC=$(PREFIX)/include/lua$(LUA_VERSION)
+LUA_CMODULE_DIR=$(PREFIX)/lib/lua/$(LUA_VERSION)
+LUA_EXE_NAME=lua$(LUA_VERSION)
+DLL_EXT=.dll
+LUA_DLL=$(LUA_BIN)/lua$(LUA_VERSION_SHORT)$(DLL_EXT)
+
+SRC_FILES= src/lfs.c
+DEF_FILE= src/lfs.def
+OBJ_FILES= src/lfs.o
+TEST_FILE= tests/test.lua
+OUT_FILE= lfs.dll
+
+MKDIR_P= mkdir -p
+RM= rm -f
+INSTALL= install -p
+INSTALL_EXEC= $(INSTALL) -m 0755
+INSTALL_DATA= $(INSTALL) -m 0644
+
+all: $(OUT_FILE)
+
+$(OBJ_FILES): $(SRC_FILES)
+	$(CC) -c "-I$(LUA_INC)" $(CFLAGS) -o $@ $<
+
+$(OUT_FILE): $(OBJ_FILES) $(DEF_FILE)
+	$(CC) -shared $(LDFLAGS) -o $@ "$(LUA_DLL)" $<
+
+install: $(OUT_FILE)
+	$(MKDIR_P) $(DESTDIR)$(LUA_CMODULE_DIR)
+	$(RM) $(DESTDIR)$(LUA_CMODULE_DIR)/$(OUT_FILE)
+	$(INSTALL_EXEC) $(OUT_FILE) $(DESTDIR)$(LUA_CMODULE_DIR)/$(OUT_FILE)
+
+check:
+	"$(LUA_BIN)/$(LUA_EXE_NAME)" $(TEST_FILE)
\ No newline at end of file
