diff --git a/glycin-thumbnailer/meson.build b/glycin-thumbnailer/meson.build
--- a/glycin-thumbnailer/meson.build
+++ b/glycin-thumbnailer/meson.build
@@ -7,6 +7,8 @@ if not get_option('libglycin')
     dependency('glycin-2')
 endif
 
+exe_suffix = host_machine.system() == 'windows' ? '.exe' : ''
+
 cargo_build = custom_target(
     'glycin-thumbnailer',
     build_by_default: true,
@@ -28,14 +30,14 @@
 custom_target(
     'glycin-thumbnailer-cp-binary',
     depends: cargo_build,
     build_by_default: true,
     build_always_stale: true,
     install: true,
     install_dir: bindir,
-    output: 'glycin-thumbnailer',
+    output: 'glycin-thumbnailer' + exe_suffix,
     command: [
         'cp',
-        cargo_artifact_dir / rust_target / 'glycin-thumbnailer',
+        cargo_artifact_dir / rust_target / ('glycin-thumbnailer' + exe_suffix),
         '@OUTPUT@',
     ],
 )
