From 7a93d66616f8aa20136b40822aeeb10438854584 Mon Sep 17 00:00:00 2001
From: David Macek <david.macek.0@gmail.com>
Date: Thu, 19 Feb 2015 13:42:35 +0100
Subject: [PATCH 1/2] Fix windres invocation

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 103976c..32963e6 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -627,7 +627,7 @@ libgeany_la_LIBADD = $(top_builddir)/scintilla/libscintilla.la \
 # tell automake we have a C++ file so it uses the C++ linker we need for Scintilla
 nodist_EXTRA_libgeany_la_SOURCES = dummy1.cxx
 CLEANFILES = $(am__append_10) signallist.i
-@MINGW_TRUE@WINDRES = $(host_alias)-windres
+@MINGW_TRUE@WINDRES = windres
 glade_file = $(top_srcdir)/data/geany.glade
 
 # install the run script
--- a/src/Makefile.am	2016-11-13 16:44:30.000000000 +0100
+++ b/src/Makefile.am	2017-02-15 22:15:10.265165500 +0100
@@ -144,7 +144,7 @@
 
 geany_LDADD += geany_private.res
 
-WINDRES = $(host_alias)-windres
+WINDRES = windres
 
 geany_private.res: $(top_srcdir)/geany_private.rc
 	$(WINDRES) -i $(top_srcdir)/geany_private.rc --input-format=rc -o $@ -O coff
-- 
2.11.1.windows.1

