diff -Naur osgocean/include/osgOcean/Cylinder osgocean-patched/include/osgOcean/Cylinder
--- a/include/osgOcean/Cylinder	2014-05-25 20:39:33.715800000 +0400
+++ b/include/osgOcean/Cylinder	2014-06-07 20:49:58.658400000 +0400
@@ -16,13 +16,14 @@
 */
 
 #pragma once
+#include <osgOcean/Export>
 #include <osg/Geometry>
 #include <osg/Math>
 
 namespace osgOcean
 {
 
-class Cylinder : public osg::Geometry
+class OSGOCEAN_EXPORT Cylinder : public osg::Geometry
 {
 protected:
 	osg::ref_ptr<osg::Vec4Array> _colorArray;
