diff -r -c wkhtmltopdf.orig/src/image/image.pro wkhtmltopdf/src/image/image.pro
*** wkhtmltopdf.orig/src/image/image.pro	2017-02-26 14:49:06.506390800 -0300
--- wkhtmltopdf/src/image/image.pro	2017-02-26 15:00:22.991627900 -0300
***************
*** 41,47 ****
  include(../shared/shared.pri)
  
  CONFIG(shared, shared|static) {
!   LIBS += -L../../bin -lwkhtmltox
  } else {
    include(../lib/lib.pri)
  }
--- 41,51 ----
  include(../shared/shared.pri)
  
  CONFIG(shared, shared|static) {
!   win32-g++* {
!     LIBS += ../../bin/libwkhtmltox.dll.a
!   } else {
!     LIBS += -L../../bin -lwkhtmltox
!   }
  } else {
    include(../lib/lib.pri)
  }
diff -r -c wkhtmltopdf.orig/src/lib/lib.pro wkhtmltopdf/src/lib/lib.pro
*** wkhtmltopdf.orig/src/lib/lib.pro	2017-02-26 14:49:06.584516300 -0300
--- wkhtmltopdf/src/lib/lib.pro	2017-02-26 15:00:22.991627900 -0300
***************
*** 43,45 ****
--- 43,52 ----
  
  DESTDIR = ../../bin
  target.path=$$INSTALLBASE/lib
+ 
+ win32-g++* {
+    TARGET=libwkhtmltox
+    TARGET_VERSION_EXT=.dll
+    QMAKE_POST_LINK += $$QMAKE_MOVE $$quote($$DESTDIR/liblibwkhtmltox.dll.a) \
+                                    $$quote($$DESTDIR/libwkhtmltox.dll.a)
+ }
diff -r -c wkhtmltopdf.orig/src/pdf/pdf.pro wkhtmltopdf/src/pdf/pdf.pro
*** wkhtmltopdf.orig/src/pdf/pdf.pro	2017-02-26 14:49:06.678268300 -0300
--- wkhtmltopdf/src/pdf/pdf.pro	2017-02-26 15:00:22.991627900 -0300
***************
*** 41,47 ****
  include(../shared/shared.pri)
  
  CONFIG(shared, shared|static) {
!   LIBS += -L../../bin -lwkhtmltox
  } else {
    include(../lib/lib.pri)
  }
--- 41,51 ----
  include(../shared/shared.pri)
  
  CONFIG(shared, shared|static) {
!   win32-g++* {
!     LIBS += ../../bin/libwkhtmltox.dll.a
!   } else {
!     LIBS += -L../../bin -lwkhtmltox
!   }
  } else {
    include(../lib/lib.pri)
  }
