diff -c -r wkhtmltopdf-0.13.orig/scripts/build.py wkhtmltopdf-0.13/scripts/build.py
*** wkhtmltopdf-0.13.orig/scripts/build.py	2015-07-25 05:14:00.000000000 -0300
--- wkhtmltopdf-0.13/scripts/build.py	2016-04-01 15:20:57.365883600 -0300
***************
*** 983,1011 ****
  
  def build_mingw64_cross(config, basedir):
      version, simple_version = get_version(basedir)
!     build_deplibs(config, basedir, mingw_w64=MINGW_W64_PREFIX.get(rchop(config, '-dbg')))
! 
!     libdir = os.path.join(basedir, config, 'deplibs')
!     qtdir  = os.path.join(basedir, config, 'qt')
  
      configure_args = qt_config('mingw-w64-cross',
!         '--prefix=%s/qtbase' % qtdir,
!         '-I%s/include'       % libdir,
!         '-L%s/lib'           % libdir,
          '-device-option CROSS_COMPILE=%s-' % MINGW_W64_PREFIX[rchop(config, '-dbg')])
  
!     os.environ['OPENSSL_LIBS']  = '-lssl -lcrypto -L%s/lib -lws2_32 -lgdi32 -lcrypt32' % libdir
!     os.environ['SQLITE3SRCDIR'] = '%s/../qt/qtbase/src/3rdparty/sqlite' % basedir
! 
!     mkdir_p(qtdir)
!     build_qtmodule(qtdir, 'qtbase', 'make -j%d' % CPU_COUNT,
!         '%s/../qt/qtbase/configure %s' % (basedir, configure_args))
!     build_qtmodule(qtdir, 'qtsvg',  'make -j%d' % CPU_COUNT,
!         '%s/qtbase/bin/qmake %s/../qt/qtsvg/qtsvg.pro' % (qtdir, basedir))
!     build_qtmodule(qtdir, 'qtxmlpatterns', 'make -j%d' % CPU_COUNT,
!         '%s/qtbase/bin/qmake %s/../qt/qtxmlpatterns/qtxmlpatterns.pro' % (qtdir, basedir))
!     build_qtmodule(qtdir, 'qtwebkit', 'make -j%d' % CPU_COUNT,
!         '%s/qtbase/bin/qmake %s/../qt/qtwebkit/WebKit.pro WEBKIT_CONFIG-=build_webkit2' % (qtdir, basedir))
  
      appdir = os.path.join(basedir, config, 'app')
      mkdir_p(appdir)
--- 983,997 ----
  
  def build_mingw64_cross(config, basedir):
      version, simple_version = get_version(basedir)
!     prefix = os.environ['PREFIX']
  
      configure_args = qt_config('mingw-w64-cross',
!         '--prefix=%s'  % prefix,
!         '-I%s/include' % prefix,
!         '-L%s/lib'     % prefix,
          '-device-option CROSS_COMPILE=%s-' % MINGW_W64_PREFIX[rchop(config, '-dbg')])
  
!     os.environ['OPENSSL_LIBS']  = '-lssl -lcrypto -L%s/lib -lws2_32 -lgdi32 -lcrypt32' % prefix
  
      appdir = os.path.join(basedir, config, 'app')
      mkdir_p(appdir)
***************
*** 1014,1023 ****
  
      # set up cross compiling prefix correctly
      os.environ['WKHTMLTOX_VERSION'] = version
!     shell('%s/qtbase/bin/qmake -set CROSS_COMPILE %s-' % (qtdir, MINGW_W64_PREFIX[rchop(config, '-dbg')]))
!     shell('%s/qtbase/bin/qmake -spec win32-g++ %s/../wkhtmltopdf.pro' % (qtdir, basedir))
      shell('make')
-     shutil.copy('bin/libwkhtmltox0.a', 'bin/wkhtmltox.lib')
      shell('rm -f bin/lib*.dll')
      for dll in ['libgcc_s_sjlj-1.dll', 'libgcc_s_seh-1.dll', 'libstdc++-6.dll']:
          dll_path = get_output('dpkg', '-S', dll)
--- 1000,1008 ----
  
      # set up cross compiling prefix correctly
      os.environ['WKHTMLTOX_VERSION'] = version
!     shell('%s/bin/qmake -set CROSS_COMPILE %s-' % (prefix, MINGW_W64_PREFIX[rchop(config, '-dbg')]))
!     shell('%s/bin/qmake -spec win32-g++ %s/../wkhtmltopdf.pro' % (prefix, basedir))
      shell('make')
      shell('rm -f bin/lib*.dll')
      for dll in ['libgcc_s_sjlj-1.dll', 'libgcc_s_seh-1.dll', 'libstdc++-6.dll']:
          dll_path = get_output('dpkg', '-S', dll)
***************
*** 1027,1036 ****
                  if exists(loc) and MINGW_W64_PREFIX[rchop(config, '-dbg')] in loc and '-posix' not in loc:
                      shell('cp %s bin/' % loc)
  
-     os.chdir(os.path.join(basedir, '..'))
-     shell('makensis -DVERSION=%s -DSIMPLE_VERSION=%s -DTARGET=%s -DMINGW -DARCH=%s wkhtmltox.nsi' % \
-             (version, nsis_version(simple_version), config, rchop(config, '-dbg').split('-')[-1]))
- 
  # -------------------------------------------------- Linux schroot environment
  
  def check_linux_schroot(config):
--- 1012,1017 ----
