From d9d43d26305f28e40b4b292edbe53c678f38c149 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 26 Jan 2015 10:24:48 +0000
Subject: [PATCH 1/2] MSYS2: Adopt FHS layout in cmake

---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98dbc70..7ec7e5b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,7 +58,7 @@ set(BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin)
 endif()
 if(MAN_INSTALL_DIR)
 else()
-    if(WIN32)
+    if(WIN32 AND NOT MINGW)
         set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/man/man1)
     else()
         set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/man/man1)
@@ -66,7 +66,7 @@ else()
 endif()
 if(DOC_INSTALL_DIR)
 else()
-    if(WIN32)
+    if(WIN32 AND NOT MINGW)
         set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/doc)
     else()
         set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/qhull)
@@ -423,9 +423,9 @@ install(TARGETS ${qhull_TARGETS_INSTALL}
         LIBRARY DESTINATION ${LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
-install(FILES ${libqhull_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
-install(FILES ${libqhull_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
-install(FILES ${libqhullcpp_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/libqhullcpp)
+install(FILES ${libqhull_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/qhull)
+install(FILES ${libqhull_DOC}        DESTINATION ${DOC_INSTALL_DIR}/qhull)
+install(FILES ${libqhullcpp_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/qhullcpp)
 install(FILES html/qhull.man         DESTINATION ${MAN_INSTALL_DIR} RENAME qhull.1)
 install(FILES html/rbox.man          DESTINATION ${MAN_INSTALL_DIR} RENAME rbox.1)
 install(FILES ${doc_FILES}           DESTINATION ${DOC_INSTALL_DIR})
-- 
2.2.2

