diff -c -r qhull-5bbc75608c817b50383a0c24c3977cc09d0bbfde.orig/CMakeLists.txt qhull-5bbc75608c817b50383a0c24c3977cc09d0bbfde/CMakeLists.txt
*** qhull-5bbc75608c817b50383a0c24c3977cc09d0bbfde.orig/CMakeLists.txt	2016-02-07 20:45:58.000000000 -0200
--- qhull-5bbc75608c817b50383a0c24c3977cc09d0bbfde/CMakeLists.txt	2016-03-14 02:28:21.593358000 -0300
***************
*** 88,94 ****
  endif()
  if(MAN_INSTALL_DIR)
  else()
!     if(WIN32)
          set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/man/man1)
      else()
          set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/man/man1)
--- 88,94 ----
  endif()
  if(MAN_INSTALL_DIR)
  else()
!     if(WIN32 AND NOT MINGW)
          set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/man/man1)
      else()
          set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/man/man1)
***************
*** 96,102 ****
  endif()
  if(DOC_INSTALL_DIR)
  else()
!     if(WIN32)
          set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/doc)
      else()
          set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/qhull)
--- 96,102 ----
  endif()
  if(DOC_INSTALL_DIR)
  else()
!     if(WIN32 AND NOT MINGW)
          set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/doc)
      else()
          set(DOC_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/qhull)
***************
*** 615,625 ****
          LIBRARY DESTINATION ${LIB_INSTALL_DIR}
          ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
  
! install(FILES ${libqhull_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
! install(FILES ${libqhull_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
! install(FILES ${libqhullr_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull_r)
! install(FILES ${libqhullr_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull_r)
! install(FILES ${libqhullcpp_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/libqhullcpp)
  install(FILES html/qhull.man         DESTINATION ${MAN_INSTALL_DIR} RENAME qhull.1)
  install(FILES html/rbox.man          DESTINATION ${MAN_INSTALL_DIR} RENAME rbox.1)
  install(FILES ${doc_FILES}           DESTINATION ${DOC_INSTALL_DIR})
--- 615,625 ----
          LIBRARY DESTINATION ${LIB_INSTALL_DIR}
          ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
  
! install(FILES ${libqhull_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/qhull)
! install(FILES ${libqhull_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/qhull)
! install(FILES ${libqhullr_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/qhull_r)
! install(FILES ${libqhullr_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/qhull_r)
! install(FILES ${libqhullcpp_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/qhullcpp)
  install(FILES html/qhull.man         DESTINATION ${MAN_INSTALL_DIR} RENAME qhull.1)
  install(FILES html/rbox.man          DESTINATION ${MAN_INSTALL_DIR} RENAME rbox.1)
  install(FILES ${doc_FILES}           DESTINATION ${DOC_INSTALL_DIR})
