--- a/skbuild/platform_specifics/platform_factory.py
+++ b/skbuild/platform_specifics/platform_factory.py
@@ -13,7 +13,7 @@
     to the current platform."""
     this_platform = platform.system().lower()
 
-    if this_platform == "windows":
+    if this_platform == "windows" and "MSC" in platform.python_compiler():
         from . import windows
 
         return windows.WindowsPlatform()
@@ -23,7 +23,7 @@
 
         return linux.LinuxPlatform()
 
-    if this_platform.startswith("cygwin"):
+    if this_platform.startswith("cygwin") or this_platform == "windows":
         from . import cygwin
 
         return cygwin.CygwinPlatform()
