--- a/skbuild/platform_specifics/platform_factory.py
+++ b/skbuild/platform_specifics/platform_factory.py
@@ -14,7 +14,7 @@
     to the current platform."""
     this_platform = platform.system().lower()
 
-    if this_platform == "windows":
+    if this_platform == "windows" and "MSC" in platform.python_compiler():
         from . import windows  # noqa: PLC0415
 
         return windows.WindowsPlatform()
@@ -25,7 +25,7 @@
 
         return linux.LinuxPlatform()
 
-    if this_platform.startswith("cygwin"):
+    if this_platform.startswith("cygwin") or this_platform == "windows":
         from . import cygwin  # noqa: PLC0415
 
         return cygwin.CygwinPlatform()
