--- QxOrm-1.4.9-orig/QxOrm.pro	2023-06-05 22:10:43.000000000 +0300
+++ QxOrm-1.4.9/QxOrm.pro	2024-07-23 21:02:10.165408900 +0300
@@ -66,7 +66,6 @@
 TARGET = QxOrm
 } # CONFIG(debug, debug|release)
 
-unix {
 isEmpty(PREFIX) { PREFIX = /usr/local }
 target.path = $$PREFIX/lib/
 headers.path = $$PREFIX/include/
@@ -74,7 +73,6 @@
 headers_inl.path = $$PREFIX/inl/
 headers_inl.files = ./inl/*
 INSTALLS = target headers headers_inl
-} # unix
 
 ####################################
 # Generate pkg-config files (*.pc) #
