diff --git a/examples/app/example-window.ui b/examples/app/example-window.ui
index 74a33dd..58e9b83 100644
--- a/examples/app/example-window.ui
+++ b/examples/app/example-window.ui
@@ -83,6 +83,8 @@
     <child>
       <object class="DzlDockBin" id="dockbin">
         <property name="visible">true</property>
+        <property name="top-visible">true</property>
+        <property name="bottom-visible">true</property>
         <child>
           <object class="GtkStack" id="stack">
             <property name="expand">true</property>
@@ -117,24 +119,105 @@
             <property name="visible">true</property>
             <child>
               <object class="GtkLabel">
-                <property name="label">Left</property>
+                <property name="label">Left - GtkBox</property>
                 <property name="visible">true</property>
               </object>
             </child>
           </object>
         </child>
         <child type="right">
-          <object class="GtkBox">
-            <property name="orientation">vertical</property>
+          <object class="DzlDockStack">
+            <property name="show-pinned-button">true</property>
+            <property name="expand">true</property>
+            <property name="visible">false</property>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="title">_Font chooser</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkFontChooserWidget">
+                    <property name="expand">true</property>
+                    <property name="visible">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="title">_Textview</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkTextView">
+                    <property name="monospace">true</property>
+                    <property name="visible">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+          </object>
+        </child>          
+        <child type="top">
+          <object class="DzlDockWidget">
+            <property name="can-close">true</property>
+            <property name="expand">true</property>
+            <property name="icon-name">color-select-symbolic</property>
+            <property name="title">Color</property>
             <property name="visible">true</property>
             <child>
               <object class="GtkLabel">
-                <property name="label">Right</property>
+                <property name="label">Top - DzlDockWidget</property>
                 <property name="visible">true</property>
               </object>
             </child>
           </object>
         </child>
+        <child type="bottom">
+          <object class="DzlDockStack">
+            <property name="show-pinned-button">true</property>
+            <property name="style">icons</property>
+            <property name="expand">true</property>
+            <property name="visible">true</property>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="icon-name">color-select-symbolic</property>
+                <property name="title">Color</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkColorChooserWidget">
+                    <property name="border-width">12</property>
+                    <property name="halign">center</property>
+                    <property name="visible">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="title">Swatches</property>
+                <property name="icon-name">preferences-color-symbolic</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkAppChooserWidget">
+                    <property name="border-width">12</property>
+                    <property name="halign">center</property>
+                    <property name="visible">true</property>
+                    <property name="content_type">x-scheme-handler/mailto</property>
+                    <property name="show_all">true</property>
+                    <property name="show_other">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+          </object>
+        </child>      
       </object>
     </child>
   </template>
diff --git a/src/files/dzl-recursive-file-monitor.c b/src/files/dzl-recursive-file-monitor.c
index 824fd83..f1bd11b 100644
--- a/src/files/dzl-recursive-file-monitor.c
+++ b/src/files/dzl-recursive-file-monitor.c
@@ -174,9 +174,13 @@ resolve_file (GFile *file)
     return g_object_ref (file);
 
   orig_path = g_file_get_path (file);
+  #ifdef G_OS_WIN32 
+  real_path = _fullpath(NULL, orig_path, _MAX_PATH);
+  #else
   real_path = realpath (orig_path, NULL);
+  #endif
 
-  /* unlikely, but PATH_MAX exceeded */
+  /* unlikely, but _MAX_PATH exceeded */
   if (real_path == NULL)
     return g_object_ref (file);
 
diff --git a/src/search/dzl-fuzzy-mutable-index.c b/src/search/dzl-fuzzy-mutable-index.c
index 49a6c8a..1230c64 100644
--- a/src/search/dzl-fuzzy-mutable-index.c
+++ b/src/search/dzl-fuzzy-mutable-index.c
@@ -63,11 +63,11 @@ struct _DzlFuzzyMutableIndex
 typedef struct
 {
   guint64 id : 32;
-  guint64 pos : 16;
+  guint64 pos : 32;
 } DzlFuzzyMutableIndexItem;
 #pragma pack(pop)
 
-G_STATIC_ASSERT (sizeof(DzlFuzzyMutableIndexItem) == 6);
+G_STATIC_ASSERT (sizeof(DzlFuzzyMutableIndexItem) == 8);
 
 typedef struct
 {
diff --git a/src/search/dzl-trie.c b/src/search/dzl-trie.c
index b4ac47c..299d0f1 100644
--- a/src/search/dzl-trie.c
+++ b/src/search/dzl-trie.c
@@ -110,6 +110,7 @@ struct _DzlTrie
    DzlTrieNode    *root;
 };
 
+#ifndef G_OS_WIN32
 #if GLIB_SIZEOF_VOID_P == 8
   G_STATIC_ASSERT (sizeof(gpointer) == 8);
   G_STATIC_ASSERT ((FIRST_CHUNK_KEYS-1) == 3);
@@ -123,6 +124,7 @@ struct _DzlTrie
   G_STATIC_ASSERT(sizeof(DzlTrieNode) == 20);
   G_STATIC_ASSERT(sizeof(DzlTrieNodeChunk) == 12);
 #endif
+#endif
 
 /**
  * dzl_trie_malloc0:
diff --git a/tests/test-directory-reaper.c b/tests/test-directory-reaper.c
index 99bfcd7..fd69b29 100644
--- a/tests/test-directory-reaper.c
+++ b/tests/test-directory-reaper.c
@@ -80,19 +80,25 @@ test_reaper_basic (void)
     g_autofree gchar *cwd = g_get_current_dir ();
     g_autofree gchar *name = g_build_filename ("reaper", "parent-link", NULL);
 
+#ifndef G_OS_WIN32
     if (symlink (cwd, name) != 0)
       g_error ("Failed to create symlink");
+#endif
   }
 
   /* also create link to out-of-tree to ensure we don't follow
    * symlinks and delete files there.
    */
+#ifndef G_OS_WIN32
   g_assert_cmpint (0, ==, symlink ("../out-of-tree", "reaper/symlink-check"));
+#endif
   g_assert_true (g_file_test ("reaper/symlink-check", G_FILE_TEST_IS_SYMLINK));
   g_assert_true (g_file_test ("reaper/symlink-check", G_FILE_TEST_EXISTS));
   g_assert_true (g_file_test ("out-of-tree/delete-check", G_FILE_TEST_IS_REGULAR));
 
+#ifndef G_OS_WIN32
   g_assert_cmpint (0, ==, symlink ("../../../out-of-tree", "reaper/a/b/d"));
+#endif
   g_assert_true (g_file_test ("reaper/a/b/d", G_FILE_TEST_IS_SYMLINK));
   g_assert_true (g_file_test ("reaper/a/b/d", G_FILE_TEST_EXISTS));
 
