diff --git a/examples/app/example-window.ui b/examples/app/example-window.ui
index 74a33dd..58e9b83 100644
--- a/examples/app/example-window.ui
+++ b/examples/app/example-window.ui
@@ -83,6 +83,8 @@
     <child>
       <object class="DzlDockBin" id="dockbin">
         <property name="visible">true</property>
+        <property name="top-visible">true</property>
+        <property name="bottom-visible">true</property>
         <child>
           <object class="GtkStack" id="stack">
             <property name="expand">true</property>
@@ -117,24 +119,105 @@
             <property name="visible">true</property>
             <child>
               <object class="GtkLabel">
-                <property name="label">Left</property>
+                <property name="label">Left - GtkBox</property>
                 <property name="visible">true</property>
               </object>
             </child>
           </object>
         </child>
         <child type="right">
-          <object class="GtkBox">
-            <property name="orientation">vertical</property>
+          <object class="DzlDockStack">
+            <property name="show-pinned-button">true</property>
+            <property name="expand">true</property>
+            <property name="visible">false</property>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="title">_Font chooser</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkFontChooserWidget">
+                    <property name="expand">true</property>
+                    <property name="visible">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="title">_Textview</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkTextView">
+                    <property name="monospace">true</property>
+                    <property name="visible">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+          </object>
+        </child>          
+        <child type="top">
+          <object class="DzlDockWidget">
+            <property name="can-close">true</property>
+            <property name="expand">true</property>
+            <property name="icon-name">color-select-symbolic</property>
+            <property name="title">Color</property>
             <property name="visible">true</property>
             <child>
               <object class="GtkLabel">
-                <property name="label">Right</property>
+                <property name="label">Top - DzlDockWidget</property>
                 <property name="visible">true</property>
               </object>
             </child>
           </object>
         </child>
+        <child type="bottom">
+          <object class="DzlDockStack">
+            <property name="show-pinned-button">true</property>
+            <property name="style">icons</property>
+            <property name="expand">true</property>
+            <property name="visible">true</property>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="icon-name">color-select-symbolic</property>
+                <property name="title">Color</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkColorChooserWidget">
+                    <property name="border-width">12</property>
+                    <property name="halign">center</property>
+                    <property name="visible">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+            <child>
+              <object class="DzlDockWidget">
+                <property name="can-close">true</property>
+                <property name="expand">true</property>
+                <property name="title">Swatches</property>
+                <property name="icon-name">preferences-color-symbolic</property>
+                <property name="visible">true</property>
+                <child>
+                  <object class="GtkAppChooserWidget">
+                    <property name="border-width">12</property>
+                    <property name="halign">center</property>
+                    <property name="visible">true</property>
+                    <property name="content_type">x-scheme-handler/mailto</property>
+                    <property name="show_all">true</property>
+                    <property name="show_other">true</property>
+                  </object>
+                </child>
+              </object>
+            </child>
+          </object>
+        </child>      
       </object>
     </child>
   </template>
