From da865d1559dc384075d7cc2b2745655e693a63f0 Mon Sep 17 00:00:00 2001
From: albfan <albertofanjul@gmail.com>
Date: Mon, 5 Aug 2019 20:54:39 +0200
Subject: [PATCH] Install to bin dir

---
 src/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index 7a47cd9..b908876 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -86,7 +86,6 @@ libdazzle = shared_library(
          dependencies: libdazzle_deps,
   include_directories: [ root_inc, src_inc ],
               install: true,
-          install_dir: libdir
 )
 
 libdazzle_dep = declare_dependency(
-- 
2.22.0

