--- json-c-json-c-0.13-20171207/Makefile.am.orig	2018-01-18 14:53:42.975721500 +0300
+++ json-c-json-c-0.13-20171207/Makefile.am	2018-01-18 14:54:09.689408500 +0300
@@ -26,6 +26,7 @@
 	json_inttypes.h \
 	json_object.h \
 	json_object_iterator.h \
+	json_object_private.h \
 	json_pointer.h \
 	json_tokener.h \
 	json_util.h \
@@ -34,7 +35,6 @@
 	printbuf.h
 
 noinst_HEADERS=\
-	json_object_private.h \
 	math_compat.h \
 	strdup_compat.h \
 	snprintf_compat.h \
