--- json-c-json-c-0.14/CMakeLists.txt.orig	2020-04-27 16:22:14.788180700 +0300
+++ json-c-json-c-0.14/CMakeLists.txt	2020-04-27 16:25:33.974302700 +0300
@@ -380,10 +380,16 @@
     ${JSON_C_SOURCES}
     ${JSON_C_HEADERS}
 )
+
+set(JSON_C_SOVERSION 5)
 set_target_properties(${PROJECT_NAME} PROPERTIES
     VERSION 5.0.0
-    SOVERSION 5)
+    SOVERSION ${JSON_C_SOVERSION})
 
+set_target_properties(${PROJECT_NAME} PROPERTIES
+                      OUTPUT_NAME ${PROJECT_NAME}
+                      RUNTIME_OUTPUT_NAME ${PROJECT_NAME}-${JSON_C_SOVERSION}
+                      ARCHIVE_OUTPUT_NAME ${PROJECT_NAME})
 # If json-c is used as subroject it set to target correct interface -I flags and allow
 # to build external target without extra include_directories(...)
 target_include_directories(${PROJECT_NAME}
