--- a/CMakeLists.txt	2020-08-03 08:55:10.090125184 -0400
+++ b/CMakeLists.txt	2020-08-03 08:57:21.002463125 -0400
@@ -416,9 +416,17 @@
     ${JSON_C_SOURCES}
     ${JSON_C_HEADERS}
 )
+
+set(JSON_C_SOVERSION 5)
 set_target_properties(${PROJECT_NAME} PROPERTIES
     VERSION 5.1.0
-    SOVERSION 5)
+    SOVERSION ${JSON_C_SOVERSION})
+
+set_target_properties(${PROJECT_NAME} PROPERTIES
+                      OUTPUT_NAME ${PROJECT_NAME}
+                      RUNTIME_OUTPUT_NAME ${PROJECT_NAME}-${JSON_C_SOVERSION}
+                      ARCHIVE_OUTPUT_NAME ${PROJECT_NAME})
+
 list(APPEND CMAKE_TARGETS ${PROJECT_NAME})
 # If json-c is used as subroject it set to target correct interface -I flags and allow
 # to build external target without extra include_directories(...)
