diff -bur libdbus-c++-0.9.0-orig/config.sub libdbus-c++-0.9.0/config.sub
--- libdbus-c++-0.9.0-orig/config.sub	2023-10-02 00:32:59.784577800 -0600
+++ libdbus-c++-0.9.0/config.sub	2023-10-02 00:33:15.442086300 -0600
@@ -311,7 +311,7 @@
 	# We use `pc' rather than `unknown'
 	# because (1) that's what they normally are, and
 	# (2) the word "unknown" tends to confuse beginning users.
-	i*86 | x86_64)
+	i*86 | x86_64 | aarch64)
 	  basic_machine=$basic_machine-pc
 	  ;;
 	# Object if more than one company name word.
@@ -379,7 +379,7 @@
 	| tron-* \
 	| v850-* | v850e-* | vax-* \
 	| we32k-* \
-	| x86-* | x86_64-* | xc16x-* | xps100-* | xscale-* | xscalee[bl]-* \
+	| x86-* | x86_64-* | aarch64-* | xc16x-* | xps100-* | xscale-* | xscalee[bl]-* \
 	| xstormy16-* | xtensa*-* \
 	| ymp-* \
 	| z8k-* | z80-*)
