--- wxWidgets-3.0.2/wx-config.in.orig	2015-12-12 14:49:36.241974800 +0000
+++ wxWidgets-3.0.2/wx-config.in	2015-12-12 14:53:06.242503400 +0000
@@ -401,6 +401,9 @@
 
 # Determine the base directories we require.
 prefix=${input_option_prefix-${this_prefix:-@prefix@}}
+if [ "x${MSYSTEM}" = "xMINGW32" ] || [ "x${MSYSTEM}" = "xMINGW64" ]; then
+    prefix=$(cygpath -m ${prefix})
+fi
 exec_prefix=${input_option_exec_prefix-${input_option_prefix-${this_exec_prefix:-@exec_prefix@}}}
 wxconfdir="@libdir@/wx/config"
 
