diff --git a/src/net.c b/src/net.c
index 8793180..706bb51 100644
--- a/src/net.c
+++ b/src/net.c
@@ -44,7 +44,9 @@
 #endif
 #ifdef W32_NATIVE
 # define WIN32_LEAN_AND_MEAN    /* do not include more than necessary */
-# define _WIN32_WINNT 0x0502    /* Windows XP SP2 or later */
+# ifndef _WIN32_WINNT
+#  define _WIN32_WINNT 0x0502    /* Windows XP SP2 or later */
+# endif
 # include <winsock2.h>
 # include <ws2tcpip.h>
 #endif
diff --git a/src/tools.c b/src/tools.c
index 8c9efe1..6a8be58 100644
--- a/src/tools.c
+++ b/src/tools.c
@@ -35,7 +35,9 @@
 #include <unistd.h>
 #ifdef W32_NATIVE
 # define WIN32_LEAN_AND_MEAN    /* do not include more than necessary */
-# define _WIN32_WINNT 0x0502    /* Windows XP SP2 or later */
+# ifndef _WIN32_WINNT
+#  define _WIN32_WINNT 0x0502    /* Windows XP SP2 or later */
+# endif
 # include <windows.h>
 # include <io.h>
 # include <conio.h>
