diff --git a/src/msmtp.c b/src/msmtp.c
index 6e18e1c..845dd18 100644
--- a/src/msmtp.c
+++ b/src/msmtp.c
@@ -27,6 +27,12 @@
 # include "config.h"
 #endif
 
+#ifdef W32_NATIVE
+# define WIN32_LEAN_AND_MEAN
+# include <windows.h>
+# include <shlwapi.h>
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdarg.h>
@@ -67,7 +73,7 @@ extern int optind;
 #endif /* HAVE_TLS */
 
 /* Default file names. */
-#ifdef W32_NATIVE
+#ifdef _MSC_VER
 #define SYSCONFFILE     "msmtprc.txt"
 #define USERCONFFILE    "msmtprc.txt"
 #else /* UNIX */
@@ -2234,7 +2240,21 @@ void msmtp_print_version(void)
     printf(_("NLS: "));
 #ifdef ENABLE_NLS
     printf(_("enabled"));
+#ifdef W32_NATIVE
+    char exepath[MAX_PATH + 1];
+    GetModuleFileNameA(NULL, exepath, MAX_PATH);
+    PathRemoveFileSpecA(exepath);
+    char *localedir = get_filename(exepath, "..\\share\\locale");
+    if (localedir)
+    {
+        printf(_(", LOCALEDIR is %s"), localedir);
+        free(localedir);
+    } else {
+        printf(_(", LOCALEDIR is %s"), "");
+    }
+#else
     printf(_(", LOCALEDIR is %s"), LOCALEDIR);
+#endif
 #else
     printf(_("disabled"));
 #endif
@@ -3814,7 +3834,20 @@ int main(int argc, char *argv[])
     /* internationalization with gettext */
 #ifdef ENABLE_NLS
     setlocale(LC_ALL, "");
+#ifdef W32_NATIVE
+    char exepath[MAX_PATH + 1];
+    GetModuleFileNameA(NULL, exepath, MAX_PATH);
+    PathRemoveFileSpecA(exepath);
+    char *localedir = get_filename(exepath, "..\\share\\locale");
+    if (localedir)
+    {
+        bindtextdomain(PACKAGE, localedir);
+    } else {
+        bindtextdomain(PACKAGE, NULL);
+    }
+#else
     bindtextdomain(PACKAGE, LOCALEDIR);
+#endif
     textdomain(PACKAGE);
 #endif
 
diff --git a/src/password.c b/src/password.c
index b4a27ae..fa0ef3a 100644
--- a/src/password.c
+++ b/src/password.c
@@ -43,7 +43,7 @@
 #include "xalloc.h"
 #include "password.h"
 
-#ifdef W32_NATIVE
+#ifdef _MSC_VER
 #define SYSNETRCFILE    "netrc.txt"
 #define USERNETRCFILE   "netrc.txt"
 #else /* UNIX */
diff --git a/src/tools.c b/src/tools.c
index f9db3db..50c0385 100644
--- a/src/tools.c
+++ b/src/tools.c
@@ -33,6 +33,7 @@
 # include <io.h>
 # include <conio.h>
 # include <lmcons.h>
+# include <shlwapi.h>
 # include <sys/locking.h>
 # include <limits.h>
 #else
@@ -318,7 +319,7 @@ const char *get_prgname(const char *argv0)
 
 char *get_sysconfdir(void)
 {
-#ifdef W32_NATIVE
+#if defined(_MSC_VER)
 
     BYTE sysconfdir[MAX_PATH + 1];
     HKEY hkey;
@@ -345,6 +346,17 @@ char *get_sysconfdir(void)
     RegCloseKey(hkey);
     return xstrdup((char *)sysconfdir);
 
+#elif defined(__MINGW32__)
+
+    char exepath[MAX_PATH + 1];
+    GetModuleFileNameA(NULL, exepath, MAX_PATH);
+    PathRemoveFileSpecA(exepath);
+
+    char *sysconfdir = get_filename(exepath, "..\\etc");
+    if (sysconfdir)
+        return xstrdup(sysconfdir);
+    return xstrdup("");
+
 #else /* UNIX */
 
 #ifdef SYSCONFDIR
