--- a/Src/m_ts_fullg.F90
+++ b/Src/m_ts_fullg.F90
@@ -685,7 +685,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,iu,ind,ju,sp,sp_col,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do io = 1 , nr
           ! Quickly go past the buffer atoms...
           if ( l_ncol(io) /= 0 ) then
@@ -735,7 +735,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,iu,sp,sp_col,ind,ju,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do io = 1 , nr
           if ( l_ncol(io) /= 0 ) then
             sp = s_ptr(io)
--- a/Src/m_ts_fullk.F90
+++ b/Src/m_ts_fullk.F90
@@ -741,7 +741,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,sp,sp_col,iu,iuT,ind,ju,juT,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do io = 1 , nr
           ! Quickly go past the buffer atoms...
           if ( l_ncol(io) /= 0 ) then
@@ -796,7 +796,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&   private(io,sp,sp_col,iu,iuT,ind,ju,juT,sind), &
-!$OMP&   reduction(-:q)
+!$OMP&   reduction(+:q)
         do io = 1 , nr
           if ( l_ncol(io) /= 0 ) then
             sp = s_ptr(io)
--- a/Src/m_ts_mumpsg.F90
+++ b/Src/m_ts_mumpsg.F90
@@ -712,7 +712,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(ir,jo,ind,io,Hn,ind_H,s_ptr_begin,s_ptr_end,sin), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do ir = 1 , mum%NRHS
           
           ! this is column index
--- a/Src/m_ts_mumpsk.F90
+++ b/Src/m_ts_mumpsk.F90
@@ -749,7 +749,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(jr,jo,sp,sp_col,lp_col,ind,io,ind_H,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do jr = 1 , mum%NRHS
 
           ! this is row index
@@ -818,7 +818,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(jr,jo,sp,sp_col,lp_col,ind,io,ind_H,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do jr = 1 , mum%NRHS
           jo = ts2s_orb(jr)
           sp = s_ptr(jo)
--- a/Src/m_ts_trig.F90
+++ b/Src/m_ts_trig.F90
@@ -742,7 +742,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,iu,sp,sp_col,ind,idx,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do iu = 1 , r%n
           io = r%r(iu)
           if ( l_ncol(io) /= 0 ) then
@@ -797,7 +797,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,iu,sp,sp_col,ind,idx,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
         do iu = 1 , r%n
           io = r%r(iu)
           if ( l_ncol(io) /= 0 ) then
--- a/Src/m_ts_trik.F90
+++ b/Src/m_ts_trik.F90
@@ -779,7 +779,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,iu,sp,sp_col,ind,idx,idxT,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
        do iu = 1 , r%n
           io = r%r(iu)
           if ( l_ncol(io) /= 0 ) then
@@ -827,7 +827,7 @@
 
 !$OMP parallel do default(shared), &
 !$OMP&  private(io,iu,sp,sp_col,ind,idx,idxT,sind), &
-!$OMP&  reduction(-:q)
+!$OMP&  reduction(+:q)
        do iu = 1 , r%n
           io = r%r(iu)
           if ( l_ncol(io) /= 0 ) then
