diff --git a/tools/generate-symbolic-dark.cpp b/tools/generate-symbolic-dark.cpp
index 7e94c93..0907e97 100644
--- a/tools/generate-symbolic-dark.cpp
+++ b/tools/generate-symbolic-dark.cpp
@@ -131,7 +131,30 @@ int main(int argc, char **argv)
 
             // keep symlinks
             if (inputFileInfo.isSymLink()) {
-                QFile::link(inputFileInfo.absoluteDir().relativeFilePath(inputFileInfo.symLinkTarget()), outputFilePath);
+                if (outputFileInfo.exists()) {
+                    if (!QFile::remove(outputFileInfo.filePath())) {
+                        qWarning() << u"Failed to remove old symlink:" << outputFilePath;
+                        return 1;
+                    }
+                }
+                std::error_code ec;
+                const auto targetAbsolute = QFileInfo(inputFileInfo.symLinkTarget()).filesystemFilePath();
+                const auto targetRelative = std::filesystem::relative(targetAbsolute, inputFileInfo.filesystemAbsolutePath(), ec);
+                if (ec) {
+                    qWarning() << u"Failed to resolve symlink:" << targetAbsolute.native() << u"Error:" << ec.message();
+                    return 1;
+                }
+                const auto symlink = outputFileInfo.filesystemFilePath();
+                // qInfo() << u"Create symlink:" << symlink.native() << u"with target" << targetRelative.native();
+                std::filesystem::create_symlink(targetRelative, symlink, ec);
+                if (ec) {
+                    // qWarning() << u"Failed to create symlink:" << symlink.native() << u"with target" << targetRelative.native() << u"Error:" << ec.message()
+                               // << u"falling back to copy";
+                    if (!QFile::copy(inputFilePath, outputFilePath)) {
+                        qWarning() << u"Failed to create copy:" << outputFilePath << u"of" << inputFilePath << qt_error_string();
+                        return 1;
+                    }
+                }
                 continue;
             }
 
