diff -urN SuiteSparse-5.8.1.orig/SLIP_LU/Lib/Makefile SuiteSparse-5.8.1/SLIP_LU/Lib/Makefile
--- SuiteSparse-5.8.1.orig/SLIP_LU/Lib/Makefile	2020-07-15 04:04:44.000000000 +0500
+++ SuiteSparse-5.8.1/SLIP_LU/Lib/Makefile	2020-12-22 19:10:14.936075900 +0500
@@ -20,7 +20,7 @@
 # CFLAGS += -Wall -Wextra -Wpedantic -Werror
 
 # SLIP_LU depends on SuiteSparse_config, AMD, COLAMD, M, GMP, and MPFR
-LDLIBS += -lsuitesparseconfig -lamd -lcolamd -lm -lgmp -lmpfr
+LDLIBS += -lsuitesparseconfig -lamd -lcolamd -lm -lmpfr -lgmp
 
 C = $(CC) $(CF) -I../Include -I../../COLAMD/Include -I../../AMD/Include -I../../SuiteSparse_config
 
