--- potrace-1.15/configure.ac.orig	2017-12-01 07:10:43.783207400 +0300
+++ potrace-1.15/configure.ac	2017-12-01 07:10:59.491835900 +0300
@@ -28,7 +28,7 @@
 dnl figure out compiler options
 
 dnl Check for compiler
-AC_PROG_CC([clang gcc cc c99 mgcc c89 pcc opencc sunc99 suncc])
+AC_PROG_CC([gcc cc c99 mgcc c89 pcc opencc sunc99 suncc clang])
 
 dnl Also add CADD to the CFLAGS at configure time or compile time
 AC_SUBST(CADD)
