From 30871d3e01bf961c74802cd495cc902c4cd034f4 Mon Sep 17 00:00:00 2001
From: Yuanmiao Gui <gui_yuan_miao@163.com>
Date: Thu, 3 Jan 2019 20:57:36 +0800
Subject: [PATCH 2/3] include intrin.h

---
 src/util/simd.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util/simd.cpp b/src/util/simd.cpp
index 7ce5b5e..02bc73d 100644
--- a/src/util/simd.cpp
+++ b/src/util/simd.cpp
@@ -20,6 +20,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "log_stream.h"
 
 #ifdef _WIN32
+#include <intrin.h>
 #define cpuid(info,x)    __cpuidex(info,x,0)
 #else
 inline void cpuid(int CPUInfo[4], int InfoType) {
-- 
2.18.0.windows.1

