From cf8a6764b6da6b91f04df52498c7dcfaa18cb3ca Mon Sep 17 00:00:00 2001
From: Yuanmiao Gui <gui_yuan_miao@163.com>
Date: Thu, 3 Jan 2019 20:58:15 +0800
Subject: [PATCH 3/3] add explicit template instantiation

---
 src/dp/needleman_wunsch.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/dp/needleman_wunsch.cpp b/src/dp/needleman_wunsch.cpp
index af3da04..f66fd0e 100644
--- a/src/dp/needleman_wunsch.cpp
+++ b/src/dp/needleman_wunsch.cpp
@@ -171,6 +171,8 @@ const Fixed_score_buffer<_score>& needleman_wunsch(sequence query, sequence subj
 	return mtx.score_buffer();
 }
 
+template Fixed_score_buffer<int> const& needleman_wunsch<int, Local>(sequence, sequence, int&, Local const&, int const&);
+
 int needleman_wunsch(sequence query, sequence subject, int qbegin, int qend, int sbegin, int send, unsigned node, unsigned edge, Diag_graph &diags, bool log)
 {
 	const sequence q = query.subseq(qbegin, qend), s = subject.subseq(sbegin, send);
-- 
2.18.0.windows.1

