From 9d27727bf2f61b387d9250acd925647fc22357d8 Mon Sep 17 00:00:00 2001
From: Yuanmiao Gui <gui_yuan_miao@163.com>
Date: Sun, 23 Dec 2018 17:40:02 +0800
Subject: [PATCH] add explicit template instantiation

---
 src/dp/needleman_wunsch.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/dp/needleman_wunsch.cpp b/src/dp/needleman_wunsch.cpp
index af3da04..f66fd0e 100644
--- a/src/dp/needleman_wunsch.cpp
+++ b/src/dp/needleman_wunsch.cpp
@@ -171,6 +171,8 @@ const Fixed_score_buffer<_score>& needleman_wunsch(sequence query, sequence subj
 	return mtx.score_buffer();
 }
 
+template Fixed_score_buffer<int> const& needleman_wunsch<int, Local>(sequence, sequence, int&, Local const&, int const&);
+
 int needleman_wunsch(sequence query, sequence subject, int qbegin, int qend, int sbegin, int send, unsigned node, unsigned edge, Diag_graph &diags, bool log)
 {
 	const sequence q = query.subseq(qbegin, qend), s = subject.subseq(sbegin, send);
-- 
2.19.1.windows.1

