--- ceres-solver-2.2.0/include/ceres/jet.h.orig	2023-10-13 04:15:01.000000000 +0300
+++ ceres-solver-2.2.0/include/ceres/jet.h	2026-04-28 07:40:12.520075600 +0300
@@ -875,17 +875,29 @@
 // Ceres itself is built).
 inline double BesselJ0(double x) {
   CERES_DISABLE_DEPRECATED_WARNING
+#ifndef __STRICT_ANSI__
   return j0(x);
+#else
+  return _j0(x);
+#endif
   CERES_RESTORE_DEPRECATED_WARNING
 }
 inline double BesselJ1(double x) {
   CERES_DISABLE_DEPRECATED_WARNING
+#ifndef __STRICT_ANSI__
   return j1(x);
+#else
+  return _j1(x);
+#endif
   CERES_RESTORE_DEPRECATED_WARNING
 }
 inline double BesselJn(int n, double x) {
   CERES_DISABLE_DEPRECATED_WARNING
+#ifndef __STRICT_ANSI__
   return jn(n, x);
+#else
+  return _jn(n, x);
+#endif
   CERES_RESTORE_DEPRECATED_WARNING
 }
 
