diff -Naur ceres-solver-1.13.0.orig/internal/ceres/levenberg_marquardt_strategy_test.cc ceres-solver-1.13.0/internal/ceres/levenberg_marquardt_strategy_test.cc
--- ceres-solver-1.13.0.orig/internal/ceres/levenberg_marquardt_strategy_test.cc	2017-08-03 15:10:55.000000000 +0800
+++ ceres-solver-1.13.0/internal/ceres/levenberg_marquardt_strategy_test.cc	2017-10-16 11:34:48.510059700 +0800
@@ -149,9 +149,7 @@
     // are versions of glog which are not in the google namespace.
     using namespace google;
 
-#if defined(_MSC_VER)
-    // Use GLOG_WARNING to support MSVC if GLOG_NO_ABBREVIATED_SEVERITIES
-    // is defined.
+#if defined(_WIN32)
     EXPECT_CALL(log,
                 Log(GLOG_WARNING, _, HasSubstr("Failed to compute a step")));
 #else
