unchanged:
--- mono/mcs/build/rules-orig.make	2017-11-25 17:57:58.011752400 -0500
+++ mono/mcs/build/rules.make	2017-11-25 17:58:11.252185300 -0500
@@ -85,7 +85,7 @@
 include $(topdir)/build/platforms/$(BUILD_PLATFORM).make
 
 PROFILE_PLATFORM = $(if $(PLATFORMS),$(if $(filter $(PLATFORMS),$(HOST_PLATFORM)),$(HOST_PLATFORM),$(error Unknown platform "$(HOST_PLATFORM)" for profile "$(PROFILE)")))
-PROFILE_DIRECTORY = $(PROFILE)$(if $(PROFILE_PLATFORM),-$(PROFILE_PLATFORM))
+PROFILE_DIRECTORY = $(PROFILE)
 
 ifdef PLATFORM_CORLIB
 corlib = $(PLATFORM_CORLIB)
only in patch2:
unchanged:
--- mono/mcs/Makefile-orig	2017-11-25 17:59:28.416265000 -0500
+++ mono/mcs/Makefile	2017-11-25 17:59:47.960113800 -0500
@@ -1,3 +1,5 @@
+export MSYS2_ARG_CONV_EXCL=*
+
 thisdir := .
 
 SUBDIRS := build jay mcs class nunit24 ilasm tools tests errors docs packages
