--- curl-8.10.0/lib/CMakeLists.txt.orig	2024-09-11 07:37:35.000000000 +0200
+++ curl-8.10.0/lib/CMakeLists.txt	2024-09-11 12:37:27.324344400 +0200
@@ -24,6 +24,7 @@
 set(LIB_NAME libcurl)
 set(LIBCURL_OUTPUT_NAME libcurl CACHE STRING "Basename of the curl library")
 add_definitions("-DBUILDING_LIBCURL")
+set(CURL_BINDIR "${CMAKE_INSTALL_PREFIX}/bin")
 
 configure_file("curl_config.h.cmake" "${CMAKE_CURRENT_BINARY_DIR}/curl_config.h")
 
@@ -57,6 +58,8 @@
   target_link_libraries(curlu PRIVATE ${CURL_LIBS})
 endif()
 
+set_source_files_properties(pathtools.c PROPERTIES SKIP_UNITY_BUILD_INCLUSION ON)
+
 if(ENABLE_CURLDEBUG)
   # We must compile these sources separately to avoid memdebug.h redefinitions
   # applying to them.

--- a/lib/curl_config.h.cmake
+++ b/lib/curl_config.h.cmake
@@ -35,6 +35,9 @@
 /* Default SSL backend */
 #cmakedefine CURL_DEFAULT_SSL_BACKEND "${CURL_DEFAULT_SSL_BACKEND}"
 
+/* Location of executable */
+#cmakedefine CURL_BINDIR "${CURL_BINDIR}"
+
 /* disables alt-svc */
 #cmakedefine CURL_DISABLE_ALTSVC 1
 
--- a/lib/Makefile.inc
+++ b/lib/Makefile.inc
@@ -197,6 +197,7 @@
   noproxy.c          \
   openldap.c         \
   parsedate.c        \
+  pathtools.c        \
   pingpong.c         \
   pop3.c             \
   progress.c         \
@@ -333,6 +334,7 @@
   nonblock.h         \
   noproxy.h          \
   parsedate.h        \
+  pathtools.h        \
   pingpong.h         \
   pop3.h             \
   progress.h         \
--- a/lib/url.c
+++ b/lib/url.c
@@ -119,6 +119,9 @@
 #include "altsvc.h"
 #include "dynbuf.h"
 #include "headers.h"
+#if defined(__MINGW32__)
+#include "pathtools.h"
+#endif
 
 /* The last 3 #include files should be in this order */
 #include "curl_printf.h"
@@ -436,22 +439,47 @@
   if(Curl_ssl_backend() != CURLSSLBACKEND_SCHANNEL &&
      Curl_ssl_backend() != CURLSSLBACKEND_SECURETRANSPORT) {
 #if defined(CURL_CA_BUNDLE)
+#if defined(__MINGW32__)
+    char *relocated_bundle = single_path_relocation_lib(CURL_BINDIR, CURL_CA_BUNDLE);
+    result = Curl_setstropt(&set->str[STRING_SSL_CAFILE], relocated_bundle);
+    free(relocated_bundle);
+#else
     result = Curl_setstropt(&set->str[STRING_SSL_CAFILE], CURL_CA_BUNDLE);
+#endif /* defined(__MINGW32__) */
     if(result)
       return result;
 #ifndef CURL_DISABLE_PROXY
+#if defined(__MINGW32__)
+    relocated_bundle = single_path_relocation_lib(CURL_BINDIR, CURL_CA_BUNDLE);
+    result = Curl_setstropt(&set->str[STRING_SSL_CAFILE_PROXY],
+                            relocated_bundle);
+    free(relocated_bundle);
+#else
     result = Curl_setstropt(&set->str[STRING_SSL_CAFILE_PROXY],
                             CURL_CA_BUNDLE);
+#endif
     if(result)
       return result;
 #endif
 #endif
 #if defined(CURL_CA_PATH)
+#if defined(__MINGW32__)
+    char *relocated_ca_path = single_path_relocation_lib(CURL_BINDIR, CURL_CA_PATH);
+    result = Curl_setstropt(&set->str[STRING_SSL_CAPATH], relocated_ca_path);
+    free(relocated_ca_path);
+#else
     result = Curl_setstropt(&set->str[STRING_SSL_CAPATH], CURL_CA_PATH);
+#endif /* defined(__MINGW32__) */
     if(result)
       return result;
 #ifndef CURL_DISABLE_PROXY
+#if defined(__MINGW32__)
+    relocated_ca_path = single_path_relocation_lib(CURL_BINDIR, CURL_CA_PATH);
+    result = Curl_setstropt(&set->str[STRING_SSL_CAPATH_PROXY], relocated_ca_path);
+    free(relocated_ca_path);
+#else
     result = Curl_setstropt(&set->str[STRING_SSL_CAPATH_PROXY], CURL_CA_PATH);
+#endif
     if(result)
       return result;
 #endif
