--- curl-8.14.0/CMake/FindNGTCP2.cmake.orig	2025-05-28 07:41:50.000000000 +0200
+++ curl-8.14.0/CMake/FindNGTCP2.cmake	2025-05-30 18:51:23.728168400 +0200
@@ -73,11 +73,17 @@
    NOT DEFINED NGTCP2_INCLUDE_DIR AND
    NOT DEFINED NGTCP2_LIBRARY)
   find_package(PkgConfig QUIET)
-  pkg_check_modules(NGTCP2 ${NGTCP2_PC_REQUIRES})
+  pkg_check_modules(NGTCP2 "libngtcp2")
+  if (_ngtcp2_crypto_backend)
+    pkg_check_modules(NGTCP2_${_ngtcp2_crypto_backend} "lib${_crypto_library_lower}")
+  endif()
 endif()
 
 if(NGTCP2_FOUND)
   string(REPLACE ";" " " NGTCP2_CFLAGS "${NGTCP2_CFLAGS}")
+  if (NGTCP2_${_ngtcp2_crypto_backend}_FOUND)
+    set(NGTCP2_LIBRARIES ${NGTCP2_LIBRARIES} ${NGTCP2_${_ngtcp2_crypto_backend}_LIBRARIES})
+  endif()
   message(STATUS "Found NGTCP2 (via pkg-config): ${NGTCP2_INCLUDE_DIRS} (found version \"${NGTCP2_VERSION}\")")
 else()
   find_path(NGTCP2_INCLUDE_DIR NAMES "ngtcp2/ngtcp2.h")
