diff -urN curl-7.40.0.orig/configure.ac curl-7.40.0/configure.ac
--- curl-7.40.0.orig/m4/xc-lt-iface.m4	2014-07-14 19:49:18.000000000 +0100
+++ curl-7.40.0/m4/xc-lt-iface.m4	2014-08-28 21:39:53.352497100 +0100
@@ -72,7 +72,7 @@
 if test "x$xc_lt_want_enable_shared" = 'xyes' &&
   test "x$xc_lt_want_enable_static" = 'xyes'; then
   case $host_os in @%:@ (
-    mingw* | pw32* | cegcc* | os2* | aix*)
+    pw32* | cegcc* | os2* | aix*)
       xc_lt_want_enable_static='no'
       ;;
   esac
