diff -urN curl-7.84.0/libcurl.pc.in.orig curl-7.84.0/libcurl.pc.in
--- curl-7.84.0/libcurl.pc.in.orig	2022-06-22 11:49:29.000000000 +0200
+++ curl-7.84.0/libcurl.pc.in	2022-06-30 18:10:26.835823300 +0200
@@ -36,6 +36,8 @@
 URL: https://curl.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @CURLVERSION@
+Requires.private: libidn2 libbrotlidec
 Libs: -L${libdir} -lcurl @LIBCURL_NO_SHARED@
 Libs.private: @LIBCURL_LIBS@
 Cflags: -I${includedir} @CPPFLAG_CURL_STATICLIB@
+Cflags.private: -DCURL_STATICLIB
diff -urN curl-7.84.0/configure.ac.orig curl-7.84.0/configure.ac
--- curl-7.84.0/configure.ac.orig	2022-06-30 18:13:08.954675200 +0200
+++ curl-7.84.0/configure.ac	2022-06-30 18:18:29.024823500 +0200
@@ -1914,7 +1914,7 @@
 
 dnl link required libraries for USE_WIN32_CRYPTO or USE_SCHANNEL
 if test "x$USE_WIN32_CRYPTO" = "x1" -o "x$USE_SCHANNEL" = "x1"; then
-  LIBS="-ladvapi32 -lcrypt32 $LIBS"
+  LIBS="-ladvapi32 -lcrypt32 -lbcrypt $LIBS"
 fi
 
 case "x$OPENSSL_ENABLED$GNUTLS_ENABLED$NSS_ENABLED$MBEDTLS_ENABLED$WOLFSSL_ENABLED$SCHANNEL_ENABLED$SECURETRANSPORT_ENABLED$BEARSSL_ENABLED$AMISSL_ENABLED$RUSTLS_ENABLED"
