--- PKGBUILD	2017-02-28 10:05:21.330804200 +0300
+++ PKGBUILD-winssl-bin	2017-02-28 10:13:15.667964800 +0300
@@ -1,6 +1,6 @@
 # Maintainer: Alexey Pavlov <alexpux@gmail.com>
-_variant=-openssl
-#_variant=-winssl
+#_variant=-openssl
+_variant=-winssl
 #_variant=-gnutls
 
 if [ "${_variant}" = "-openssl" ]; then
@@ -11,7 +11,7 @@
 
 _realname=curl
-pkgbase=mingw-w64-${_realname}
-pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}${_namesuff}"
+pkgbase=mingw-w64-${_realname}${_namesuff}-bin
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}${_namesuff}-bin"
 pkgver=7.54.0
 pkgrel=1
 pkgdesc="An URL retrival utility and library. (mingw-w64)"
@@ -33,11 +33,8 @@
          $([[ "$_variant" == "-gnutls" ]] && echo \
             "${MINGW_PACKAGE_PREFIX}-ca-certificates" \
             "${MINGW_PACKAGE_PREFIX}-gnutls")
+         "${MINGW_PACKAGE_PREFIX}-${_realname}"
          )
-if [ -n "${_namesuff}" ]; then
-  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-fi
 options=('staticlibs')
 source=("${url}/download/${_realname}-${pkgver}.tar.bz2"{,.asc}
         "0001-Make-cURL-relocatable.patch"
@@ -106,8 +103,11 @@
 
 package() {
   cd "${srcdir}/build-${CARCH}"
-  make DESTDIR="${pkgdir}" install
-  
-  local PREFIX_DEPS=$(cygpath -am ${MINGW_PREFIX})
-  sed -s "s|${PREFIX_DEPS}|${MINGW_PREFIX}|g" -i ${pkgdir}${MINGW_PREFIX}/bin/curl-config
+  make DESTDIR="${pkgdir}-tmp" install
+
+  install -d -m755 "${pkgdir}/${MINGW_PREFIX}/bin/curl-winssl"
+  install -m755 "${pkgdir}-tmp/${MINGW_PREFIX}/bin/curl.exe" "${pkgdir}/${MINGW_PREFIX}/bin/curl-winssl"
+  install -m755 "${pkgdir}-tmp/${MINGW_PREFIX}/bin/libcurl-4.dll" "${pkgdir}/${MINGW_PREFIX}/bin/curl-winssl"
+
+  rm -rf "${pkgdir}-tmp"
 }
