--- origsrc/mscgen-0.20/src/main.c	2015-09-18 16:13:31.213511400 +0100
+++ src/mscgen-0.20/src/main.c	2015-09-18 16:14:04.200167200 +0100
@@ -1620,7 +1620,7 @@ int main(const int argc, const char *arg
         if(!envFont)
         {
             /* Pick a default font */
-            snprintf(gOutputFont, bufLen, "helvetica");
+            snprintf(gOutputFont, bufLen, "DejaVu Sans");
         }
         else if(snprintf(gOutputFont, bufLen, "%s", envFont) >= bufLen)
         {
