--- origsrc/mscgen-0.20/CYGWIN-PATCHES/README	1970-01-01 01:00:00.000000000 +0100
+++ src/mscgen-0.20/CYGWIN-PATCHES/README	2015-09-18 17:13:31.394585400 +0100
@@ -0,0 +1,82 @@
+mscgen-0.20-4
+-------------------------------------------
+Message Sequence Chart rendering programme.
+
+Runtime requirements:
+  cygwin
+  dejavu-fonts
+  libgd3
+
+Build requirements:
+  binutils
+  bison
+  cygport
+  flex
+  gawk
+  gcc-core
+  grep
+  libgd-devel
+  make
+  sed
+
+Canonical homepage:
+  http://www.mcternan.me.uk/mscgen/
+
+Canonical download:
+  http://www.mcternan.me.uk/mscgen/
+
+-------------------------------------------
+
+Build instructions:
+  unpack mscgen-0.20-4-src.tar.xz
+    if you use setup to install this src package, it will be
+        unpacked under /usr/src automatically
+  cd /usr/src
+  cygport ./mscgen.cygport all
+
+This will create mscgen-0.20-4.x86_64/dist,
+  which will contain the files to upload.
+
+-------------------------------------------
+
+Files included in the binary packages:
+
+(mscgen-0.20-4.tar.xz)
+  /usr/bin/mscgen.exe
+  /usr/share/doc/Cygwin/mscgen.README
+  /usr/share/doc/mscgen/ChangeLog
+  /usr/share/doc/mscgen/COPYING
+  /usr/share/doc/mscgen/examples/boxes_example.msc
+  /usr/share/doc/mscgen/examples/client_server.msc
+  /usr/share/doc/mscgen/examples/colour_sample.msc
+  /usr/share/doc/mscgen/examples/msg_types.msc
+  /usr/share/doc/mscgen/examples/simple_prog_desc.msc
+  /usr/share/doc/mscgen/mscgen.lang
+  /usr/share/doc/mscgen/README
+  /usr/share/doc/mscgen/TODO
+  /usr/share/man/man1/mscgen.1.gz
+
+
+-------------------------------------------
+
+Port Notes:
+
+---- version 0.20-4 ----
+Use 'DejaVu Sans' as the default font, as 'Luxi Sans' is no longer provided by
+Cygwin.
+
+---- version 0.20-3 ----
+Rebuilt with FreeType font rendering enabled.
+
+---- version 0.20-2 ----
+First version maintained by Dave Stacey. Rebuilt wiith libgd3 and three
+patches from Fedora.
+
+---- version 0.20-1 ----
+New version following upstream release
+
+---- version 0.19-1 ----
+New version following upstream release
+
+
+Cygwin port maintained by Dave Stacey.
