diff -u -r djvulibre-3.5.22/configure djvulibre-3.5.22/configure
--- djvulibre-3.5.22/configure	2009-05-21 17:15:50.000000000 +0200
+++ djvulibre-3.5.22/configure	2009-11-13 21:42:30.000000000 +0100
@@ -2166,7 +2166,6 @@
 case "$host" in
   *-mingw32)
      DLLFLAGS="$DLLFLAGS -Wl,--export-all-symbols"
-     LIBS=-lmsvcp60
   ;;
   *-darwin* | *-macos10*)
      if test -d /opt/local ; then
diff -u -r djvulibre-3.5.22/configure.ac djvulibre-3.5.22/configure.ac
--- djvulibre-3.5.22/configure.ac	2009-05-21 17:15:50.000000000 +0200
+++ djvulibre-3.5.22/configure.ac	2009-11-13 21:42:16.000000000 +0100
@@ -75,7 +75,6 @@
 case "$host" in
   *-mingw32)
      DLLFLAGS="$DLLFLAGS -Wl,--export-all-symbols"
-     LIBS=-lmsvcp60
   ;;
   *-darwin* | *-macos10*)
      if test -d /opt/local ; then
