--- djvulibre-3.5.22/tools/Makefile.in	2008-09-06 04:52:46.000000000 +0200
+++ djvulibre-3.5.22/tools/Makefile.in	2009-06-23 10:53:57.000000000 +0200
@@ -38,6 +38,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 LIBTOOL = @LIBTOOL@
+EXEEXT = @EXEEXT@
 
 DEFS = @DEFS@
 OPTS = @OPTS@
@@ -60,10 +61,10 @@
 
 SUBDIRS = jb2cmp
 
-PROGRAMS = bzz c44 cjb2 cpaldjvu csepdjvu \
-	   ddjvu djvm djvmcvt djvudump \
-	   djvups djvuextract djvumake \
-	   djvused djvutxt djvuserve
+PROGRAMS = bzz${EXEEXT} c44${EXEEXT} cjb2${EXEEXT} cpaldjvu${EXEEXT} csepdjvu${EXEEXT} \
+	   ddjvu${EXEEXT} djvm${EXEEXT} djvmcvt${EXEEXT} djvudump${EXEEXT} \
+	   djvups${EXEEXT} djvuextract${EXEEXT} djvumake${EXEEXT} \
+	   djvused${EXEEXT} djvutxt${EXEEXT} djvuserve${EXEEXT}
 
 SCRIPTS = djvudigital any2djvu
 
@@ -113,67 +114,67 @@
 jb2cmp/libjb2cmp.a: FORCE
 	cd jb2cmp && ${MAKE}
 
-annotate: annotate.o
+annotate${EXEEXT}: annotate.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} annotate.o ${LIBDJVU} ${LIBS}
 
-bzz: bzz.o
+bzz${EXEEXT}: bzz.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} bzz.o ${LIBDJVU} ${LIBS}
 
-c44: c44.o
+c44${EXEEXT}: c44.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} c44.o ${LIBDJVU} ${LIBS}
 
-cjb2: cjb2.o ${JB2OBJS}
+cjb2${EXEEXT}: cjb2.o ${JB2OBJS}
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} cjb2.o ${JB2OBJS} ${LIBDJVU} ${MORELIBS}
 
-cpaldjvu: cpaldjvu.o ${JB2OBJS}
+cpaldjvu${EXEEXT}: cpaldjvu.o ${JB2OBJS}
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} cpaldjvu.o ${JB2OBJS} ${LIBDJVU} ${LIBS}
 
-csepdjvu: csepdjvu.o ${JB2OBJS}
+csepdjvu${EXEEXT}: csepdjvu.o ${JB2OBJS}
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} csepdjvu.o ${JB2OBJS} ${LIBDJVU} ${LIBS}
 
-ddjvu: ddjvu.o tiff2pdf.o
+ddjvu${EXEEXT}: ddjvu.o tiff2pdf.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} ddjvu.o tiff2pdf.o ${LIBDJVU} ${MORELIBS}
 
-djvm: djvm.o
+djvm${EXEEXT}: djvm.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvm.o ${LIBDJVU} ${LIBS}
 
-djvmcvt: djvmcvt.o
+djvmcvt${EXEEXT}: djvmcvt.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvmcvt.o ${LIBDJVU} ${LIBS}
 
-djvudump: djvudump.o
+djvudump${EXEEXT}: djvudump.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvudump.o ${LIBDJVU} ${LIBS}
 
-djvups: djvups.o
+djvups${EXEEXT}: djvups.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvups.o ${LIBDJVU} ${LIBS}
 
-djvuextract: djvuextract.o
+djvuextract${EXEEXT}: djvuextract.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvuextract.o ${LIBDJVU} ${LIBS}
 
-djvumake: djvumake.o
+djvumake${EXEEXT}: djvumake.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvumake.o ${LIBDJVU} ${LIBS}
 
-djvused: djvused.o
+djvused${EXEEXT}: djvused.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvused.o ${LIBDJVU} ${LIBS}
 
-djvutxt: djvutxt.o
+djvutxt${EXEEXT}: djvutxt.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvutxt.o ${LIBDJVU} ${LIBS}
 
-djvuserve: djvuserve.o
+djvuserve${EXEEXT}: djvuserve.o
 	${LIBTOOL} --mode=link \
 	  ${CXX} -o $@ ${CXXFLAGS} djvuserve.o ${LIBDJVU} ${LIBS}
 
