--- a/examples/ida/F2003_openmp/CMakeLists.txt
+++ b/examples/ida/F2003_openmp/CMakeLists.txt
@@ -48,7 +48,7 @@
     # libraries to link against
     target_link_libraries(
       ${example} sundials_ida sundials_fida_mod sundials_nvecopenmp
-      sundials_fnvecopenmp_mod ${SUNDIALS_LIBS})
+      sundials_fnvecopenmp_mod ${SUNDIALS_LIBS} OpenMP::OpenMP_Fortran)
 
     # Install fortran modules to a unique directory to avoid naming collisions
     set(CMAKE_Fortran_MODULE_DIRECTORY
--- a/examples/ida/F2003_openmp/idaHeat2D_kry_omp_f2003.f90
+++ b/examples/ida/F2003_openmp/idaHeat2D_kry_omp_f2003.f90
@@ -40,7 +40,7 @@
 
   !======= Inclusions ===========
   use, intrinsic :: iso_c_binding
-  use, intrinsic :: omp_lib
+  use omp_lib
   use fsundials_core_mod
 
   !======= Declarations =========
