diff -Naur sundials-5.7.0.orig/cmake/SundialsBuildOptionsPost.cmake sundials-5.7.0/cmake/SundialsBuildOptionsPost.cmake
--- sundials-5.7.0.orig/cmake/SundialsBuildOptionsPost.cmake	2021-01-29 15:14:52.000000000 -0600
+++ sundials-5.7.0/cmake/SundialsBuildOptionsPost.cmake	2021-07-12 14:56:16.412442500 -0500
@@ -204,3 +204,24 @@
                 DEPENDS_ON ENABLE_PETSC PETSC_FOUND
                 ADVANCED)
 list(APPEND SUNDIALS_BUILD_LIST "BUILD_SUNNONLINSOL_PETSCSNES")
+
+set (SUNDIALS_NVECTOR_F2C_TYPES)
+if (BUILD_NVECTOR_SERIAL)
+  list (APPEND SUNDIALS_NVECTOR_F2C_TYPES "serial")
+  set (SUNDIALS_NVECTOR_F2C_serial_SUFFIX "_serial")
+endif ()
+
+if (BUILD_NVECTOR_PARALLEL)
+  list (APPEND SUNDIALS_NVECTOR_F2C_TYPES "parallel")
+  set (SUNDIALS_NVECTOR_F2C_parallel_SUFFIX "_parallel")
+endif ()
+
+if (BUILD_NVECTOR_OPENMP)
+  list (APPEND SUNDIALS_NVECTOR_F2C_TYPES "openmp")
+  set (SUNDIALS_NVECTOR_F2C_openmp_SUFFIX "_openmp")
+endif ()
+
+if (BUILD_NVECTOR_PARALLEL)
+  list (APPEND SUNDIALS_NVECTOR_F2C_TYPES "pthreads")
+  set (SUNDIALS_NVECTOR_F2C_pthreads_SUFFIX "_pthreads")
+endif ()
diff -Naur sundials-5.7.0.orig/src/sunlinsol/pcg/CMakeLists.txt sundials-5.7.0/src/sunlinsol/pcg/CMakeLists.txt
--- sundials-5.7.0.orig/src/sunlinsol/pcg/CMakeLists.txt	2021-01-29 15:14:53.000000000 -0600
+++ sundials-5.7.0/src/sunlinsol/pcg/CMakeLists.txt	2021-07-12 14:56:53.128858200 -0500
@@ -55,8 +55,25 @@
       ${sunlinsollib_VERSION}
     SOVERSION
       ${sunlinsollib_SOVERSION}
+    STATIC_ONLY
   )
 
+  foreach (type ${SUNDIALS_NVECTOR_F2C_TYPES})
+    sundials_add_library(sundials_fsunlinsolpcg${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      SOURCES
+        fsunlinsol_pcg.c
+      LINK_LIBRARIES
+        PRIVATE sundials_sunlinsolpcg sundials_fnvec${type}
+      OUTPUT_NAME
+        sundials_fsunlinsolpcg${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      VERSION
+        ${sunlinsollib_VERSION}
+      SOVERSION
+        ${sunlinsollib_SOVERSION}
+      SHARED_ONLY
+    )
+  endforeach ()
+
   #
   message(STATUS "Added SUNLINSOL_PCG F77 interface")
 endif()
diff -Naur sundials-5.7.0.orig/src/sunlinsol/spbcgs/CMakeLists.txt sundials-5.7.0/src/sunlinsol/spbcgs/CMakeLists.txt
--- sundials-5.7.0.orig/src/sunlinsol/spbcgs/CMakeLists.txt	2021-01-29 15:14:53.000000000 -0600
+++ sundials-5.7.0/src/sunlinsol/spbcgs/CMakeLists.txt	2021-07-12 14:57:17.252858600 -0500
@@ -55,6 +55,23 @@
       ${sunlinsollib_VERSION}
     SOVERSION
       ${sunlinsollib_SOVERSION}
+    STATIC_ONLY
   )
+
+  foreach (type ${SUNDIALS_NVECTOR_F2C_TYPES})
+    sundials_add_library(sundials_fsunlinsolspbcgs${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      SOURCES
+        fsunlinsol_spbcgs.c
+      LINK_LIBRARIES
+        PRIVATE sundials_sunlinsolspbcgs sundials_fnvec${type}
+      OUTPUT_NAME
+        sundials_fsunlinsolspbcgs${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      VERSION
+        ${sunlinsollib_VERSION}
+      SOVERSION
+        ${sunlinsollib_SOVERSION}
+      SHARED_ONLY
+    )
+  endforeach ()
   message(STATUS "Added SUNLINSOL_SPBCGS F77 interface")
 endif()
diff -Naur sundials-5.7.0.orig/src/sunlinsol/spfgmr/CMakeLists.txt sundials-5.7.0/src/sunlinsol/spfgmr/CMakeLists.txt
--- sundials-5.7.0.orig/src/sunlinsol/spfgmr/CMakeLists.txt	2021-01-29 15:14:53.000000000 -0600
+++ sundials-5.7.0/src/sunlinsol/spfgmr/CMakeLists.txt	2021-07-12 14:57:31.730571900 -0500
@@ -54,6 +54,23 @@
       ${sunlinsollib_VERSION}
     SOVERSION
       ${sunlinsollib_SOVERSION}
+    STATIC_ONLY
   )
+
+  foreach (type ${SUNDIALS_NVECTOR_F2C_TYPES})
+    sundials_add_library(sundials_fsunlinsolspfgmr${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      SOURCES
+        fsunlinsol_spfgmr.c
+      LINK_LIBRARIES
+        PRIVATE sundials_sunlinsolspfgmr sundials_fnvec${type}
+      OUTPUT_NAME
+        sundials_fsunlinsolspfgmr${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      VERSION
+        ${sunlinsollib_VERSION}
+      SOVERSION
+        ${sunlinsollib_SOVERSION}
+      SHARED_ONLY
+    )
+  endforeach ()
   message(STATUS "Added SUNLINSOL_SPFGMR F77 interface")
 endif()
diff -Naur sundials-5.7.0.orig/src/sunlinsol/spgmr/CMakeLists.txt sundials-5.7.0/src/sunlinsol/spgmr/CMakeLists.txt
--- sundials-5.7.0.orig/src/sunlinsol/spgmr/CMakeLists.txt	2021-01-29 15:14:53.000000000 -0600
+++ sundials-5.7.0/src/sunlinsol/spgmr/CMakeLists.txt	2021-07-12 14:57:43.994334100 -0500
@@ -54,6 +54,23 @@
       ${sunlinsollib_VERSION}
     SOVERSION
       ${sunlinsollib_SOVERSION}
+    STATIC_ONLY
   )
+
+  foreach (type ${SUNDIALS_NVECTOR_F2C_TYPES})
+    sundials_add_library(sundials_fsunlinsolspgmr${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      SOURCES
+        fsunlinsol_spgmr.c
+      LINK_LIBRARIES
+        PRIVATE sundials_sunlinsolspgmr sundials_fnvec${type}
+      OUTPUT_NAME
+        sundials_fsunlinsolspgmr${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      VERSION
+        ${sunlinsollib_VERSION}
+      SOVERSION
+        ${sunlinsollib_SOVERSION}
+      SHARED_ONLY
+    )
+  endforeach ()
   message(STATUS "Added SUNLINSOL_SPGMR F77 interface")
 endif()
diff -Naur sundials-5.7.0.orig/src/sunlinsol/sptfqmr/CMakeLists.txt sundials-5.7.0/src/sunlinsol/sptfqmr/CMakeLists.txt
--- sundials-5.7.0.orig/src/sunlinsol/sptfqmr/CMakeLists.txt	2021-01-29 15:14:53.000000000 -0600
+++ sundials-5.7.0/src/sunlinsol/sptfqmr/CMakeLists.txt	2021-07-12 14:57:56.830062600 -0500
@@ -54,6 +54,23 @@
       ${sunlinsollib_VERSION}
     SOVERSION
       ${sunlinsollib_SOVERSION}
+    STATIC_ONLY
   )
+
+  foreach (type ${SUNDIALS_NVECTOR_F2C_TYPES})
+    sundials_add_library(sundials_fsunlinsolsptfqmr${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      SOURCES
+        fsunlinsol_sptfqmr.c
+      LINK_LIBRARIES
+        PRIVATE sundials_sunlinsolsptfqmr sundials_fnvec${type}
+      OUTPUT_NAME
+        sundials_fsunlinsolsptfqmr${SUNDIALS_NVECTOR_F2C_${type}_SUFFIX}
+      VERSION
+        ${sunlinsollib_VERSION}
+      SOVERSION
+        ${sunlinsollib_SOVERSION}
+      SHARED_ONLY
+    )
+  endforeach ()
   message(STATUS "Added SUNLINSOL_SPTFQMR F77 interface")
 endif()
