Add attribute to dllexport function

diff -urN sundials-7.0.0/include/cvodes/cvodes.h.orig sundials-7.0.0/include/cvodes/cvodes.h
--- sundials-7.0.0/include/cvodes/cvodes.h.orig	2024-03-01 18:41:17.321314600 +0100
+++ sundials-7.0.0/include/cvodes/cvodes.h	2024-03-01 18:42:56.424206800 +0100
@@ -579,7 +579,8 @@
                                                CVadjCheckPointRec* ckpnt);
 
 /* CVLS interface function that depends on CVRhsFn */
-int CVodeSetJacTimesRhsFnB(void* cvode_mem, int which, CVRhsFn jtimesRhsFn);
+SUNDIALS_EXPORT int CVodeSetJacTimesRhsFnB(void* cvode_mem, int which,
+                                           CVRhsFn jtimesRhsFn);
 
 /* Undocumented Optional Output Functions For Backward Problems */
 
