--- sundials-5.7.0.orig/cmake/macros/SundialsAddLibrary.cmake	2021-01-29 15:14:52.000000000 -0600
+++ sundials-5.7.0/cmake/macros/SundialsAddLibrary.cmake	2021-07-09 13:43:07.076625400 -0500
@@ -324,7 +324,7 @@
       endif()
 
       # install phase
-      install(TARGETS ${_actual_target_name} DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT sundials-targets)
+      install(TARGETS ${_actual_target_name} EXPORT sundials-targets)
 
     endif()
 
