--- sundials-5.7.0.orig/cmake/tpl/FindSUPERLUDIST.cmake	2021-01-29 15:14:51.000000000 -0600
+++ sundials-5.7.0/cmake/tpl/FindSUPERLUDIST.cmake	2021-07-10 15:12:16.512453400 -0500
@@ -68,7 +68,7 @@
 endif()
 
 # find the library configuration file
-if(SUPERLUDIST_LIBRARY AND SUPERLUDIST_INCLUDE_DIR)
+if(SUPERLUDIST_LIBRARY AND SUPERLUDIST_INCLUDE_DIR AND NOT SUPERLUDIST_INDEX_SIZE)
   find_file(SUPERLUDIST_CONFIG_PATH superlu_dist_config.h PATHS ${SUPERLUDIST_INCLUDE_DIR})
   file(STRINGS ${SUPERLUDIST_CONFIG_PATH} _strings_with_index_size REGEX "XSDK_INDEX_SIZE")
   list(GET _strings_with_index_size 0 _index_size_string)
