Use CMake target installed by SuiteSparse if available.

diff -urN sundials-6.7.0/cmake/tpl/FindKLU.cmake.orig sundials-6.7.0/cmake/tpl/FindKLU.cmake
--- sundials-6.7.0/cmake/tpl/FindKLU.cmake.orig	2023-12-18 18:26:43.000000000 +0100
+++ sundials-6.7.0/cmake/tpl/FindKLU.cmake	2023-12-30 20:26:08.649091700 +0100
@@ -30,6 +30,14 @@
 #   KLU_LIBRARIES   - all of the libraries needed for KLU
 # ---------------------------------------------------------------
 
+find_package(KLU CONFIG)
+
+if(TARGET SuiteSparse::KLU)
+  set(KLU_FOUND ON)
+  add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)
+  return()
+endif()
+
 # Set library prefixes for Windows
 if(WIN32)
   set(CMAKE_FIND_LIBRARY_PREFIXES lib ${CMAKE_FIND_LIBRARY_PREFIXES})
