diff -Naur sundials-5.7.0.orig/cmake/tpl/FindPETSC.cmake sundials-5.7.0/cmake/tpl/FindPETSC.cmake
--- sundials-5.7.0.orig/cmake/tpl/FindPETSC.cmake	2021-01-29 15:14:51.000000000 -0600
+++ sundials-5.7.0/cmake/tpl/FindPETSC.cmake	2021-07-12 16:44:17.643057200 -0500
@@ -423,8 +423,8 @@
   # The location of these files has changed with different PETSc versions,
   # so look in a few different locations for them.
   if (EXISTS "${PETSC_LIBRARY_DIR}/petsc/conf/petscvariables") # > 3.5
-    set (petsc_conf_rules "${PETSC_LIBRARY_DIR}/petsc/conf/rules")
-    set (petsc_conf_variables "${PETSC_LIBRARY_DIR}/petsc/conf/variables")
+    set (petsc_conf_rules "${PETSC_LIBRARY_DIR}/petsc/conf/petscrules")
+    set (petsc_conf_variables "${PETSC_LIBRARY_DIR}/petsc/conf/petscvariables")
   elseif (EXISTS "${PETSC_INCLUDE_DIR}/petscconf.h")   # > 2.3.3
     set (petsc_conf_rules "${PETSC_DIR_}/conf/rules")
     set (petsc_conf_variables "${PETSC_DIR_}/conf/variables")
@@ -518,7 +518,10 @@
       # There is no libpetscvec
       set (PETSC_LIBRARY_VEC "NOTFOUND" CACHE INTERNAL "Cleared" FORCE)
 
-      petsc_find_library_in_list (SINGLE petsc PETSC_LIBRARIES_REMAINING)
+      if (NOT PETSC_LIBRARY_SINGLE)
+        petsc_find_library_in_list (SINGLE petsc PETSC_LIBRARIES_REMAINING)
+      endif ()
+
       # Debian 9/Ubuntu 16.04 uses _real and _complex extensions when using libraries in /usr/lib/petsc.
       if (NOT PETSC_LIBRARY_SINGLE)
         petsc_find_library_in_list (SINGLE petsc_real PETSC_LIBRARIES_REMAINING)
