--- a/cmake/tpl/FindPETSC.cmake
+++ b/cmake/tpl/FindPETSC.cmake
@@ -34,7 +34,8 @@
     set(_pkg_version_spec ">=${PETSC_FIND_VERSION}")
   endif()
 endif()
-pkg_check_modules(PKG_PETSC "PETSc${_pkg_version_spec}")
+set(PETSC_PKG_CONFIG_MODULE "PETSc" CACHE STRING "The name of the PETSc pkg-config module")
+pkg_check_modules(PKG_PETSC "${PETSC_PKG_CONFIG_MODULE}${_pkg_version_spec}")
 unset(_pkg_version_spec)
 
 # Find the PETSC libraries
@@ -121,5 +121,5 @@
                INTERFACE_LINK_LIBRARIES "${PETSC_LIBRARIES}")
 endif()
 
-mark_as_advanced(PETSC_INCLUDE_DIRS PETSC_LIBRARIES PETSC_VERSION_MAJOR
+mark_as_advanced(PETSC_PKG_CONFIG_MODULE PETSC_INCLUDE_DIRS PETSC_LIBRARIES PETSC_VERSION_MAJOR
                  PETSC_VERSION_MINOR PETSC_VERSION_PATCH PETSC_VERSION)
