--- libassuan-2.1.1/configure.ac.orig	2013-10-16 13:39:08.255597600 +0000
+++ libassuan-2.1.1/configure.ac	2013-10-16 13:45:42.764693900 +0000
@@ -234,6 +234,11 @@
 AM_CONDITIONAL(HAVE_W32CE_SYSTEM, test "$have_w32ce_system" = yes)
 AM_CONDITIONAL(HAVE_W64_SYSTEM, test "$have_w64_system" = yes)
 
+case "${host}" in
+    *-w64-mingw32*)
+	AC_DEFINE(_GNU_SOURCE, 1, Activate GNU extensions for mingw-w64)
+    ;;
+esac
 
 #
 # Provide information about the build.
