--- tesseract-ocr-298e31465a44/cube/tess_lang_mod_edge.h.orig	2010-11-23 18:34:14.000000000 +0000
+++ tesseract-ocr-298e31465a44/cube/tess_lang_mod_edge.h	2014-08-25 21:42:01.540317700 +0100
@@ -177,7 +177,7 @@
   // to quickly lookup exisiting edges to converge during search
   inline unsigned int Hash() const {
     return static_cast<unsigned int>(((start_edge_ | end_edge_) ^
-      ((reinterpret_cast<unsigned long int>(dawg_)))) ^
+      ((reinterpret_cast<uintptr_t>(dawg_)))) ^
       ((unsigned int)edge_mask_) ^
       class_id_);
   }
