--- a/src/training/CMakeLists.txt
+++ b/src/training/CMakeLists.txt
@@ -395,6 +395,11 @@
     generate_export_header(pango_training EXPORT_MACRO_NAME
                            TESS_PANGO_TRAINING_API)
     project_group(pango_training "Training Tools")
+    install(
+      TARGETS pango_training
+      RUNTIME DESTINATION bin
+      LIBRARY DESTINATION lib
+      ARCHIVE DESTINATION lib)
 
     # ##########################################################################
     # EXECUTABLE text2image
