--- tesseract-4.0.0/src/ccstruct/imagedata.cpp.orig     2018-10-29 13:26:44.598482600 +0000
+++ tesseract-4.0.0/src/ccstruct/imagedata.cpp  2018-10-29 13:27:01.678763600 +0000
@@ -26,9 +26,9 @@

 #if defined(__MINGW32__)
 #include <unistd.h>
-#else
-#include <thread>
 #endif
+#include <thread>
+

 #include "allheaders.h"  // for pixDestroy, pixGetHeight, pixGetWidth, lept_...
 #include "boxread.h"     // for ReadMemBoxes
