--- tesseract-3.04.00/ccutil/Makefile.am.orig	2015-07-14 22:28:59.961600000 +0300
+++ tesseract-3.04.00/ccutil/Makefile.am	2015-07-14 22:29:49.101600000 +0300
@@ -42,7 +42,7 @@
     params.cpp universalambigs.cpp
 
 if T_WIN
-AM_CPPFLAGS += -I$(top_srcdir)/vs2008/port -DWINDLLNAME=\"lib@GENERIC_LIBRARY_NAME@\"
+AM_CPPFLAGS += -I$(top_srcdir)/vs2010/port -DWINDLLNAME=\"lib@GENERIC_LIBRARY_NAME@\"
 noinst_HEADERS += ../vs2010/port/strtok_r.h
 libtesseract_ccutil_la_SOURCES += ../vs2010/port/strtok_r.cpp
 endif
