--- tesseract-4.1.1-orig/configure.ac	2019-12-26 22:21:51 +0800
+++ tesseract-4.1.1/configure.ac	2021-11-14 00:20:31 +0800
@@ -88,7 +88,7 @@
     mingw*)
         AM_CONDITIONAL([T_WIN], true)
         AM_CONDITIONAL([ADD_RT], false)
-        AC_SUBST([AM_LDFLAGS], ['-Wl,-no-undefined -Wl,--as-needed'])
+        AC_SUBST([AM_LDFLAGS], ['-no-undefined'])
         ;;
     cygwin*)
         AM_CONDITIONAL([ADD_RT], false)
--- tesseract-4.1.1/src/api/Makefile.am.orig	2021-05-05 11:01:16.091542600 -0700
+++ tesseract-4.1.1/src/api/Makefile.am	2021-05-05 11:04:58.638484300 -0700
@@ -103,7 +103,7 @@
 if T_WIN
 tesseract_LDADD += -ltiff
 tesseract_LDADD += -lws2_32
-libtesseract_la_LDFLAGS += -no-undefined -Wl,--as-needed -lws2_32
+libtesseract_la_LDFLAGS += -no-undefined -lws2_32
 endif
 if ADD_RT
 tesseract_LDADD += -lrt
