--- a/src/io/fs.cpp
+++ b/src/io/fs.cpp
@@ -3,6 +3,7 @@
 #include <iostream>     // cout
 #include <fstream>      // File
 #include <iterator>     // std::back_inserter
+#include <cstdint>      // uint32_t
 #include <algorithm>    // std::unique
 #include <sys/stat.h>
 
--- a/src/tools/text.cpp
+++ b/src/tools/text.cpp
@@ -2,6 +2,7 @@
 
 #include <algorithm>
 #include <regex>
+#include <cstring>
 
 std::string toLower(const std::string& _string) {
     std::string std = _string;
