diff -r -c libkml-1.3.1-b1.orig/src/swig/CMakeLists.txt libkml-1.3.1-b1/src/swig/CMakeLists.txt
*** libkml-1.3.1-b1.orig/src/swig/CMakeLists.txt	2016-07-04 00:06:19.000000000 -0300
--- libkml-1.3.1-b1/src/swig/CMakeLists.txt	2016-07-04 01:09:33.452965300 -0300
***************
*** 15,26 ****
  endif()
  
  if(WITH_PYTHON)
!   find_package(PythonLibs REQUIRED)
    if(NOT PYTHONLIBS_FOUND)
      message(FATAL_ERROR "Python Libraries not found.")
    endif()
    
!   find_package(PythonInterp REQUIRED)
  
    include_directories(${PYTHON_INCLUDE_PATH})
  
--- 15,26 ----
  endif()
  
  if(WITH_PYTHON)
!   find_package(PythonLibs 2.7 REQUIRED)
    if(NOT PYTHONLIBS_FOUND)
      message(FATAL_ERROR "Python Libraries not found.")
    endif()
    
!   find_package(PythonInterp 2.7 REQUIRED)
  
    include_directories(${PYTHON_INCLUDE_PATH})
  
***************
*** 28,39 ****
      add_definitions("-DMS_WIN64")
    endif()
  
!   if(NOT DEFINED PYTHON_INSTALL_DIR)
      execute_process(COMMAND
        ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))"
        OUTPUT_VARIABLE PYTHON_INSTALL_DIR
        OUTPUT_STRIP_TRAILING_WHITESPACE )
!   endif(NOT DEFINED PYTHON_INSTALL_DIR)
  
    file(GLOB IFILES "kml*.i")
    foreach(INTERFACE_FILE ${IFILES})
--- 28,41 ----
      add_definitions("-DMS_WIN64")
    endif()
  
!   if(MINGW)
!     set(PYTHON_INSTALL_DIR "${LIB_INSTALL_DIR}/python2.7/site-packages")
!   elseif(NOT DEFINED PYTHON_INSTALL_DIR)
      execute_process(COMMAND
        ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))"
        OUTPUT_VARIABLE PYTHON_INSTALL_DIR
        OUTPUT_STRIP_TRAILING_WHITESPACE )
!   endif()
  
    file(GLOB IFILES "kml*.i")
    foreach(INTERFACE_FILE ${IFILES})
