--- a/cmake/ProjUtilities.cmake
+++ b/cmake/ProjUtilities.cmake
@@ -52,7 +52,7 @@
 
   # On Windows, ABI version is specified using binary file name suffix.
   # On Unix, suffix is empty and SOVERSION is used instead.
-  if(WIN32)
+  if(MSVC)
     string(LENGTH "${${PROJECT_NAME}_ABI_VERSION}" abilen)
     if(abilen GREATER 0)
       set(SUFFIX "_${${PROJECT_NAME}_ABI_VERSION}")
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -393,7 +393,7 @@
   $<BUILD_INTERFACE:${PROJ_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
-if(WIN32)
+if(MSVC)
   set_target_properties(proj
     PROPERTIES
     VERSION "${PROJ_VERSION}"
