--- libfyaml-0.9.4/src/internal/libfyaml-parser.c.orig	2026-02-13 10:40:21.772895100 +0100
+++ libfyaml-0.9.4/src/internal/libfyaml-parser.c	2026-02-13 10:40:31.199991000 +0100
@@ -1623,6 +1623,7 @@
 	return kv ? kv->key : NULL;
 }
 
+#if 0
 static void do_accel_kv(const struct fy_parse_cfg *cfg, int argc, char *argv[])
 {
 	struct fy_kv_store kvs;
@@ -1687,6 +1688,7 @@
 
 	return 0;
 }
+#endif
 
 #if 0
 static void test_diag_output(struct fy_diag *diag, void *user, const char *buf, size_t len)
