--- libfyaml-0.9.4/cmake/generate-def.cmake.orig	2026-02-02 19:28:08.000000000 +0100
+++ libfyaml-0.9.4/cmake/generate-def.cmake	2026-02-13 10:46:12.265075300 +0100
@@ -49,7 +49,6 @@
 ; This file exports the public API symbols marked with FY_EXPORT.
 ; Regenerate with: cmake -DINPUT_HEADER=include/libfyaml.h -DOUTPUT_DEF=src/lib/fyaml.def -P cmake/generate-def.cmake
 
-LIBRARY fyaml
 EXPORTS
 ")
 
