--- a/Makefile.am
+++ b/Makefile.am
@@ -49,7 +49,7 @@
 
 libzimg_la_SOURCES = dummy.cpp
 libzimg_la_LIBADD = libzimg_internal.la
-libzimg_la_LDFLAGS = -no-undefined -version-info 2
+libzimg_la_LDFLAGS = -no-undefined -version-info 2 -export-symbols $(srcdir)/_msvc/zimg.def
 
 libzimg_internal_la_SOURCES = \
 	src/zimg/api/zimg.cpp \
--- a/_msvc/zimg.def
+++ b/_msvc/zimg.def
@@ -1,4 +1,3 @@
-LIBRARY z
 EXPORTS
 	zimg_get_version_info
 	zimg_get_api_version
