diff -r -c flatbuffers-1.6.0.orig/CMakeLists.txt flatbuffers-1.6.0/CMakeLists.txt
*** flatbuffers-1.6.0.orig/CMakeLists.txt	2017-02-15 20:36:51.000000000 -0200
--- flatbuffers-1.6.0/CMakeLists.txt	2017-02-26 17:00:24.447595100 -0300
***************
*** 220,232 ****
  if(FLATBUFFERS_INSTALL)
    install(DIRECTORY include/flatbuffers DESTINATION include)
    if(FLATBUFFERS_BUILD_FLATLIB)
!     install(TARGETS flatbuffers DESTINATION lib)
    endif()
    if(FLATBUFFERS_BUILD_FLATC)
      install(TARGETS flatc DESTINATION bin)
    endif()
    if(FLATBUFFERS_BUILD_SHAREDLIB)
!     install(TARGETS flatbuffers_shared DESTINATION lib)
    endif()
  endif()
  
--- 220,234 ----
  if(FLATBUFFERS_INSTALL)
    install(DIRECTORY include/flatbuffers DESTINATION include)
    if(FLATBUFFERS_BUILD_FLATLIB)
!     install(TARGETS flatbuffers
!       LIBRARY DESTINATION lib ARCHIVE DESTINATION lib RUNTIME DESTINATION bin)
    endif()
    if(FLATBUFFERS_BUILD_FLATC)
      install(TARGETS flatc DESTINATION bin)
    endif()
    if(FLATBUFFERS_BUILD_SHAREDLIB)
!     install(TARGETS flatbuffers_shared
!       LIBRARY DESTINATION lib ARCHIVE DESTINATION lib RUNTIME DESTINATION bin)
    endif()
  endif()
  
Only in flatbuffers-1.6.0.orig/: CONTRIBUTING.md
Only in flatbuffers-1.6.0.orig/docs/source: CONTRIBUTING.md
