--- node-v0.10.29/src/node_counters.h.orig	2014-06-09 18:04:36.000000000 +0100
+++ node-v0.10.29/src/node_counters.h	2014-07-25 23:58:28.185650400 +0100
@@ -45,7 +45,7 @@
 #define NODE_COUNT_NET_BYTES_SENT(bytes)
 #define NODE_COUNT_NET_BYTES_RECV(bytes)
 #define NODE_COUNT_GET_GC_RAWTIME()
-#define NODE_COUNT_GC_PERCENTTIME()
+#define NODE_COUNT_GC_PERCENTTIME(percent)
 #define NODE_COUNT_PIPE_BYTES_SENT(bytes)
 #define NODE_COUNT_PIPE_BYTES_RECV(bytes)
 #endif
