From 9c37b8e1d25c341504e87b84cf926133b9eff8fb Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Tue, 14 Feb 2017 21:43:31 -0500
Subject: [PATCH 02/N] Fix system-icu build

---
 deps/v8/src/v8.gyp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/deps/v8/src/v8.gyp b/deps/v8/src/v8.gyp
index 93926024..6637fb50 100644
--- a/deps/v8/src/v8.gyp
+++ b/deps/v8/src/v8.gyp
@@ -1818,11 +1818,6 @@
             'runtime/runtime-intl.cc',
           ],
         }],
-        ['OS=="win" and v8_enable_i18n_support==1', {
-          'dependencies': [
-            '<(icu_gyp_path):icudata',
-          ],
-        }],
       ],
     },
     {
-- 
2.19.0.rc1.windows.1

