From e05442d383e30a2387f659d579d7d19fe923a621 Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Tue, 14 Feb 2017 22:26:28 -0500
Subject: [PATCH 03/11] Include win32-headers.h in v8/atomicops.h

---
 deps/v8/src/base/atomicops.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/deps/v8/src/base/atomicops.h b/deps/v8/src/base/atomicops.h
index ea33e48..ac759ad 100644
--- a/deps/v8/src/base/atomicops.h
+++ b/deps/v8/src/base/atomicops.h
@@ -28,6 +28,10 @@
 #include <stdint.h>
 #include "src/base/build_config.h"
 
+#if defined(_WIN32)
+#include "win32-headers.h"
+#endif
+
 #if defined(_WIN32) && defined(V8_HOST_ARCH_64_BIT)
 // windows.h #defines this (only on x64). This causes problems because the
 // public API also uses MemoryBarrier at the public name for this fence. So, on
-- 
2.13.0

