diff -urN v8.orig/tools/gyp/v8.gyp v8/tools/gyp/v8.gyp
--- v8.orig/tools/gyp/v8.gyp	2014-07-25 00:41:24.863487500 +0100
+++ v8/tools/gyp/v8.gyp	2014-07-25 00:46:42.823863300 +0100
@@ -75,7 +75,14 @@
                   },
                 }],
                 ['soname_version!=""', {
-                  'product_extension': 'so.<(soname_version)',
+                  'conditions': [
+                    ['OS=="win"', {
+                      'product_extension': '<(soname_version).dll',
+                      'ldflags': ['-Wl,--out-implib,<(PRODUCT_DIR)/libv8.<(soname_version).dll.a']
+                    }, {
+                      'product_extension': 'so.<(soname_version)',
+                    }],
+                  ],
                 }],
               ],
             },
