From 6023179bd2dd24ac4c5228bf1fad873b4008559a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 30 Aug 2014 23:59:05 +0100
Subject: [PATCH 4/5] Win32: Determine and use EXEEXT

---
 Makefile         | 63 +++++++++++++++++++++++++++++---------------------------
 tools/install.py |  4 +++-
 2 files changed, 36 insertions(+), 31 deletions(-)

diff -urN node-v0.10.35.orig/Makefile node-v0.10.35/Makefile
--- node-v0.10.35/Makefile.orig	2014-12-22 21:22:32.000000000 +0000
+++ node-v0.10.35/Makefile	2015-01-13 22:11:43.244268800 +0000
@@ -6,7 +6,10 @@
 DESTDIR ?=
 SIGN ?=
 
-NODE ?= ./node
+# Determine EXEEXT
+EXEEXT=$(shell $(PYTHON) -c "import sysconfig; print(sysconfig.get_config_var('EXE'))")
+
+NODE ?= ./node$(EXEEXT)
 
 # Default to verbose builds.
 # To do quiet/pretty builds, run `make V=` to set V to an empty string,
@@ -16,31 +19,31 @@
 # BUILDTYPE=Debug builds both release and debug builds. If you want to compile
 # just the debug build, run `make -C out BUILDTYPE=Debug` instead.
 ifeq ($(BUILDTYPE),Release)
-all: out/Makefile node
+all: out/Makefile node$(EXEEXT)
 else
-all: out/Makefile node node_g
+all: out/Makefile node$(EXEEXT) node_g$(EXEEXT)
 endif
 
 # The .PHONY is needed to ensure that we recursively use the out/Makefile
 # to check for changes.
-.PHONY: node node_g
+.PHONY: node$(EXEEXT) node_g$(EXEEXT)
 
 ifeq ($(USE_NINJA),1)
-node: config.gypi
+node$(EXEEXT): config.gypi
 	$(NINJA) -C out/Release/
-	ln -fs out/Release/node node
+	ln -fs out/Release/node$(EXEEXT) $@
 
-node_g: config.gypi
+node_g$(EXEEXT): config.gypi
 	$(NINJA) -C out/Debug/
-	ln -fs out/Debug/node $@
+	ln -fs out/Debug/node$(EXEEXT) $@
 else
-node: config.gypi out/Makefile
+node$(EXEEXT): config.gypi out/Makefile
 	$(MAKE) -C out BUILDTYPE=Release V=$(V)
-	ln -fs out/Release/node node
+	ln -fs out/Release/node$(EXEEXT) $@
 
-node_g: config.gypi out/Makefile
+node_g$(EXEEXT): config.gypi out/Makefile
 	$(MAKE) -C out BUILDTYPE=Debug V=$(V)
-	ln -fs out/Debug/node $@
+	ln -fs out/Debug/node$(EXEEXT) $@
 endif
 
 out/Makefile: common.gypi deps/uv/uv.gyp deps/http_parser/http_parser.gyp deps/zlib/zlib.gyp deps/v8/build/common.gypi deps/v8/tools/gyp/v8.gyp node.gyp config.gypi
@@ -61,7 +64,7 @@
 	$(PYTHON) tools/install.py $@ $(DESTDIR)
 
 clean:
-	-rm -rf out/Makefile node node_g out/$(BUILDTYPE)/node blog.html email.md
+	-rm -rf out/Makefile node$(EXEEXT) node_g$(EXEEXT) out/$(BUILDTYPE)/node$(EXEEXT) blog.html email.md
 	-find out/ -name '*.o' -o -name '*.a' | xargs rm -rf
 	-rm -rf node_modules
 
@@ -69,7 +72,7 @@
 	-rm -rf out
 	-rm -f config.gypi
 	-rm -f config.mk
-	-rm -rf node node_g blog.html email.md
+	-rm -rf node$(EXEEXT) node_g$(EXEEXT) blog.html email.md
 	-rm -rf node_modules
 
 test: all
@@ -83,8 +86,8 @@
 	$(PYTHON) tools/test.py --mode=release --valgrind simple message
 
 test/gc/node_modules/weak/build/Release/weakref.node:
-	@if [ ! -f node ]; then make all; fi
-	./node deps/npm/node_modules/node-gyp/bin/node-gyp rebuild \
+	@if [ ! -f node$(EXEEXT) ]; then make all; fi
+	./node$(EXEEXT) deps/npm/node_modules/node-gyp/bin/node-gyp rebuild \
 		--directory="$(shell pwd)/test/gc/node_modules/weak" \
 		--nodedir="$(shell pwd)"
 
@@ -119,11 +122,11 @@
 test-internet: all
 	$(PYTHON) tools/test.py internet
 
-test-npm: node
-	./node deps/npm/test/run.js
+test-npm: node$(EXEEXT)
+	./node$(EXEEXT) deps/npm/test/run.js
 
-test-npm-publish: node
-	npm_package_config_publishtest=true ./node deps/npm/test/run.js
+test-npm-publish: node$(EXEEXT)
+	npm_package_config_publishtest=true ./node$(EXEEXT) deps/npm/test/run.js
 
 test-timers:
 	$(MAKE) --directory=tools faketime
@@ -144,7 +147,7 @@
 	out/doc/sh_main.js    \
 	out/doc/sh_javascript.min.js
 
-doc: $(apidoc_dirs) $(website_files) $(apiassets) $(apidocs) tools/doc/ out/doc/changelog.html node
+doc: $(apidoc_dirs) $(website_files) $(apiassets) $(apidocs) tools/doc/ out/doc/changelog.html node$(EXEEXT)
 
 doc-branch: NODE_DOC_VERSION = v$(shell $(PYTHON) tools/getnodeversion.py | cut -f1,2 -d.)
 doc-branch: doc
@@ -155,24 +158,24 @@
 out/doc/api/assets/%: doc/api_assets/% out/doc/api/assets/
 	cp $< $@
 
-out/doc/changelog.html: ChangeLog doc/changelog-head.html doc/changelog-foot.html tools/build-changelog.sh node
+out/doc/changelog.html: ChangeLog doc/changelog-head.html doc/changelog-foot.html tools/build-changelog.sh node$(EXEEXT)
 	bash tools/build-changelog.sh
 
 out/doc/%: doc/%
 	cp -r $< $@
 
-out/doc/api/%.json: doc/api/%.markdown node
-	NODE_DOC_VERSION=$(NODE_DOC_VERSION) out/Release/node tools/doc/generate.js --format=json $< > $@
+out/doc/api/%.json: doc/api/%.markdown node$(EXEEXT)
+	NODE_DOC_VERSION=$(NODE_DOC_VERSION) out/Release/node$(EXEEXT) tools/doc/generate.js --format=json $< > $@
 
-out/doc/api/%.html: doc/api/%.markdown node
-	NODE_DOC_VERSION=$(NODE_DOC_VERSION) out/Release/node tools/doc/generate.js --format=html --template=doc/template.html $< > $@
+out/doc/api/%.html: doc/api/%.markdown node$(EXEEXT)
+	NODE_DOC_VERSION=$(NODE_DOC_VERSION) out/Release/node$(EXEEXT) tools/doc/generate.js --format=html --template=doc/template.html $< > $@
 
 email.md: ChangeLog tools/email-footer.md
 	bash tools/changelog-head.sh | sed 's|^\* #|* \\#|g' > $@
 	cat tools/email-footer.md | sed -e 's|__VERSION__|'$(VERSION)'|g' >> $@
 
 blog.html: email.md
-	cat $< | ./node tools/doc/node_modules/.bin/marked > $@
+	cat $< | ./node$(EXEEXT) tools/doc/node_modules/.bin/marked > $@
 
 website-upload: doc
 	rsync -r out/doc/ node@nodejs.org:~/web/nodejs.org/
@@ -279,7 +282,7 @@
 		--out $(PKG)
 	SIGN="$(INT_SIGN)" PKG="$(PKG)" bash tools/osx-productsign.sh
 
-$(TARBALL): release-only node doc
+$(TARBALL): release-only node$(EXEEXT) doc
 	git archive --format=tar --prefix=$(TARNAME)/ HEAD | tar xf -
 	mkdir -p $(TARNAME)/doc/api
 	cp doc/node.1 $(TARNAME)/doc/node.1
@@ -366,9 +369,9 @@
 	 benchmark/http_simple_bench.sh
 
 bench-idle:
-	./node benchmark/idle_server.js &
+	./node$(EXEEXT) benchmark/idle_server.js &
 	sleep 1
-	./node benchmark/idle_clients.js &
+	./node$(EXEEXT) benchmark/idle_clients.js &
 
 jslintfix:
 	PYTHONPATH=tools/closure_linter/ $(PYTHON) tools/closure_linter/closure_linter/fixjsstyle.py --strict --nojsdoc -r lib/ -r src/ --exclude_files lib/punycode.js
diff -urN node-v0.10.32.orig/tools/install.py node-v0.10.32/tools/install.py
--- node-v0.10.32.orig/tools/install.py	2014-09-22 15:08:38.922406400 +0100
+++ node-v0.10.32/tools/install.py	2014-09-22 15:08:54.221349100 +0100
@@ -11,6 +11,7 @@
 import re
 import shutil
 import sys
+import sysconfig
 
 # set at init time
 dst_dir = None
@@ -127,7 +128,8 @@
     action(files, subdir + '/')
 
 def files(action):
-  action(['out/Release/node'], 'bin/node')
+  exeext=sysconfig.get_config_var('EXE')
+  action(['out/Release/node' + exeext], 'bin/node' + exeext)
 
   if 'true' == variables.get('node_use_dtrace'):
     action(['out/Release/node.d'], 'lib/dtrace/node.d')
--
2.1.0
