--- node-v0.10.29.orig/configure	2014-07-25 16:15:00.638405900 +0100
+++ node-v0.10.29/configure	2014-07-25 16:15:03.632577100 +0100
@@ -706,7 +706,7 @@
   gyp_args += ['-f', 'ninja-' + flavor]
 elif options.use_xcode:
   gyp_args += ['-f', 'xcode']
-elif flavor == 'win':
+elif flavor == 'win' and sys.platform != 'msys':
   gyp_args += ['-f', 'msvs', '-G', 'msvs_version=auto']
 else:
   gyp_args += ['-f', 'make-' + flavor]
