From 6b45f3c2635b1fea7217e40e123567f113911344 Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Wed, 15 Feb 2017 02:03:24 -0500
Subject: [PATCH 06/15] Define _WIN32_WINNT in node.gyp

---
 node.gypi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/node.gypi b/node.gypi
index fe85f1f3..4774fff3 100644
--- a/node.gypi
+++ b/node.gypi
@@ -1,5 +1,8 @@
 {
   'conditions': [
+    [ 'OS=="win"', {
+      'defines': [ '_WIN32_WINNT=0x0600' ],
+    }],
     [ 'node_shared=="false"', {
       'msvs_settings': {
         'VCManifestTool': {
-- 
2.17.0.windows.1

