From b2a82cd1865cbf1dc21db290b1be33439646dd04 Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Wed, 15 Feb 2017 02:03:24 -0500
Subject: [PATCH 06/23] Define _WIN32_WINNT in node.gyp

---
 node.gypi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/node.gypi b/node.gypi
index 3990c59e..2705cefe 100644
--- a/node.gypi
+++ b/node.gypi
@@ -55,5 +55,6 @@
       'defines!': [
         'NODE_PLATFORM="win"',
       ],
+      'defines': [ '_WIN32_WINNT=0x0600' ],
       'defines': [
         'FD_SETSIZE=1024',
         # we need to use node's preferred "win32" rather than gyp's preferred "win"
-- 
2.17.0.windows.1
