From 06c18d111b8dbd75326b2f67cec5aae6fbbdee71 Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Wed, 15 Feb 2017 02:45:09 -0500
Subject: [PATCH 07/15] Skip sys/resource and unneeded includes in node.cc

---
 src/node.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/node.cc b/src/node.cc
index 8ab9e4d1..d76c1ca6 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -76,7 +76,7 @@
 #define getpid GetCurrentProcessId
 #define umask _umask
 typedef int mode_t;
-#else
+#elif !defined(__MINGW64_VERSION_MAJOR)
 #include <pthread.h>
 #include <sys/resource.h>  // getrlimit, setrlimit
 #include <unistd.h>  // setuid, getuid
-- 
2.17.0.windows.1

