--- node-v0.10.29/node.gyp.orig	2014-07-26 00:04:52.435628300 +0100
+++ node-v0.10.29/node.gyp	2014-07-26 00:03:45.343790800 +0100
@@ -213,7 +213,10 @@
           ]
         } ],
         [ 'node_use_perfctr=="true"', {
-          'defines': [ 'HAVE_PERFCTR=1' ],
+          'defines': [
+            'HAVE_PERFCTR=1',
+            '_WIN32_WINNT=0x0600',
+          ],
           'dependencies': [ 'node_perfctr' ],
           'sources': [
             'src/node_win32_perfctr_provider.h',
