From 70588e70284f6dd3e238d1cf2ff6b5c1f9d7e72b Mon Sep 17 00:00:00 2001
From: mati865 <mati865@gmail.com>
Date: Sun, 4 Sep 2016 18:30:30 +0200
Subject: [PATCH 08/12] Use shell wrapper script for npm on MSYS2 MinGW w64

---
 tools/install.py | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/tools/install.py b/tools/install.py
index e61d71a..e6dfdf3 100644
--- a/tools/install.py
+++ b/tools/install.py
@@ -2,6 +2,7 @@
 
 import errno
 import json
+import platform
 import os
 import re
 import shutil
@@ -94,7 +95,17 @@ def npm_files(action):
   if action == uninstall:
     action([link_path], 'bin/npm')
   elif action == install:
-    try_symlink('../lib/node_modules/npm/bin/npm-cli.js', link_path)
+    # Use MSYS2 shell script wrapper.
+    if ((os.name == 'nt' and 'GCC' in sys.version) or
+        (os.name == 'posix' and 'NT' in platform.system())):
+      # Copy the wrapper shell script
+      try_copy('deps/npm/bin/npm', link_path)
+      # deps/npm/bin/npm is a shell script so the shebang must run sh, not node
+      s = open(link_path, 'r').read()
+      s = re.sub(r'#!.*\n', '#!' + '/usr/bin/env sh' + '\n', s)
+      open(link_path, 'w').write(s)
+    else:
+      try_symlink('../lib/node_modules/npm/bin/npm-cli.js', link_path)
   else:
     assert(0) # unhandled action type
 
-- 
2.9.1

