From 3b353a4dbc0e96a35d18dde94a787945f1c5c58a Mon Sep 17 00:00:00 2001
From: Alethea Rose <alethea@alethearose.com>
Date: Wed, 15 Feb 2017 19:11:53 -0500
Subject: [PATCH 08/23] Link with -municode and extern wmain for Unicode

---
 node.gyp  | 9 +++++++++
 node.gypi | 1 +
 2 files changed, 10 insertions(+)

diff --git a/node.gyp b/node.gyp
index fc14d6e5..f08fcdf3 100644
--- a/node.gyp
+++ b/node.gyp
@@ -592,6 +592,9 @@
       'msvs_disabled_warnings!': [4244],
 
       'conditions': [
+        [ 'OS=="win"', {
+          'ldflags': [ '-municode' ],
+        }],
         [ 'node_shared_hdr_histogram=="false"', {
           'dependencies': [
             'deps/histogram/histogram.gyp:histogram',
@@ -1454,6 +1457,9 @@
         [ 'OS in "linux mac openharmony"', {
           'defines': ['NODE_JS2C_USE_STRING_LITERALS'],
         }],
+        [ 'OS=="win"', {
+          'ldflags': [ '-municode' ],
+        }],
         [ 'debug_node=="true"', {
           'cflags!': [ '-O3' ],
           'cflags': [ '-g', '-O0' ],
@@ -1513,6 +1519,9 @@
             'deps/nbytes/nbytes.gyp:nbytes',
           ],
         }],
+        [ 'OS=="win"', {
+          'ldflags': [ '-municode' ],
+        }],
         [ 'node_use_openssl=="true"', {
           'dependencies': [
             'deps/ncrypto/ncrypto.gyp:ncrypto',
diff --git a/node.gypi b/node.gypi
index a851fbbc..08d0f492 100644
--- a/node.gypi
+++ b/node.gypi
@@ -67,6 +67,7 @@
         'FD_SETSIZE=1024',
         # we need to use node's preferred "win32" rather than gyp's preferred "win"
         'NODE_PLATFORM="win32"',
+        'UNICODE',
         '_UNICODE=1',
       ],
       'conditions': [
-- 
2.17.0.windows.1
