From 06665ce1cb24ada9489b21ef81d9fe360af17de8 Mon Sep 17 00:00:00 2001
From: mati865 <mati865@gmail.com>
Date: Wed, 14 Sep 2016 19:40:47 +0200
Subject: [PATCH 13/13] use windmc instead of mc

---
 node.gyp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/node.gyp b/node.gyp
index 39acb6c..2ee0d7a 100644
--- a/node.gyp
+++ b/node.gyp
@@ -623,7 +623,7 @@
                 'tools/msvs/genfiles/node_etw_provider.h',
                 'tools/msvs/genfiles/node_etw_providerTEMP.BIN',
               ],
-              'action': [ 'mc <@(_inputs) -h tools/msvs/genfiles -r tools/msvs/genfiles' ]
+              'action': [ 'windmc <@(_inputs) -h tools/msvs/genfiles -r tools/msvs/genfiles' ]
             }
           ]
         } ]
-- 
2.9.1

