diff -urN node-v6.11.0.a/deps/v8/src/objects.h node-v6.11.0.b/deps/v8/src/objects.h
--- node-v6.11.0.a/deps/v8/src/objects.h	2017-06-06 21:24:33.000000000 +0200
+++ node-v6.11.0.b/deps/v8/src/objects.h	2018-04-01 11:00:08.648082800 +0200
@@ -3261,22 +3261,10 @@
 template <typename Derived, typename Shape, typename Key>
 class HashTable : public HashTableBase {
  public:
-  // Wrapper methods
-  inline uint32_t Hash(Key key) {
-    if (Shape::UsesSeed) {
-      return Shape::SeededHash(key, GetHeap()->HashSeed());
-    } else {
-      return Shape::Hash(key);
-    }
-  }
-
-  inline uint32_t HashForObject(Key key, Object* object) {
-    if (Shape::UsesSeed) {
-      return Shape::SeededHashForObject(key, GetHeap()->HashSeed(), object);
-    } else {
-      return Shape::HashForObject(key, object);
-    }
-  }
+  // Wrapper methods.  Defined in src/objects-inl.h
+  // to break a cycle with src/heap/heap.h.
+  inline uint32_t Hash(Key key);
+  inline uint32_t HashForObject(Key key, Object* object);
 
   // Returns a new HashTable object.
   MUST_USE_RESULT static Handle<Derived> New(
diff -urN node-v6.11.0.a/deps/v8/src/objects-body-descriptors.h node-v6.11.0.b/deps/v8/src/objects-body-descriptors.h
--- node-v6.11.0.a/deps/v8/src/objects-body-descriptors.h	2017-06-06 21:24:33.000000000 +0200
+++ node-v6.11.0.b/deps/v8/src/objects-body-descriptors.h	2018-04-01 11:00:08.645082600 +0200
@@ -99,7 +99,7 @@
 
   template <typename StaticVisitor>
   static inline void IterateBody(HeapObject* obj, int object_size) {
-    IterateBody(obj);
+    IterateBody<StaticVisitor>(obj);
   }
 };
 
diff -urN node-v6.11.0.a/deps/v8/src/objects-inl.h node-v6.11.0.b/deps/v8/src/objects-inl.h
--- node-v6.11.0.a/deps/v8/src/objects-inl.h	2017-06-06 21:24:33.000000000 +0200
+++ node-v6.11.0.b/deps/v8/src/objects-inl.h	2018-04-01 11:00:08.647082700 +0200
@@ -36,6 +36,27 @@
 namespace v8 {
 namespace internal {
 
+template <typename Derived, typename Shape, typename Key>
+uint32_t HashTable<Derived, Shape, Key>::Hash(Key key) {
+  if (Shape::UsesSeed) {
+    return Shape::SeededHash(key, GetHeap()->HashSeed());
+  } else {
+    return Shape::Hash(key);
+  }
+}
+
+
+template <typename Derived, typename Shape, typename Key>
+uint32_t HashTable<Derived, Shape, Key>::HashForObject(Key key,
+                                                       Object* object) {
+  if (Shape::UsesSeed) {
+    return Shape::SeededHashForObject(key, GetHeap()->HashSeed(), object);
+  } else {
+    return Shape::HashForObject(key, object);
+  }
+}
+
+
 PropertyDetails::PropertyDetails(Smi* smi) {
   value_ = smi->value();
 }
