 src/node.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/node.h b/src/node.h
index f2740116..de28015e 100644
--- a/src/node.h
+++ b/src/node.h
@@ -45,13 +45,18 @@
 #ifdef BUILDING_NODE_EXTENSION
 # undef BUILDING_V8_SHARED
 # undef BUILDING_UV_SHARED
+#if defined(__MINGW32__)
+# undef USING_V8_SHARED
+# undef USING_UV_SHARED
+#else
 # define USING_V8_SHARED 1
 # define USING_UV_SHARED 1
 #endif
+#endif
 
 // This should be defined in make system.
 // See issue https://github.com/nodejs/node-v0.x-archive/issues/1236
-#if defined(__MINGW32__) || defined(_MSC_VER)
+#if defined(_MSC_VER)
 #ifndef _WIN32_WINNT
 # define _WIN32_WINNT 0x0600  // Windows Server 2008
 #endif
