--- node-v11.15.0-orig/deps/v8/src/base/platform/platform-win32.cc.orig	2019-06-28 08:53:22.904659200 +0300
+++ node-v11.15.0/deps/v8/src/base/platform/platform-win32.cc	2019-06-28 08:53:44.904690000 +0300
@@ -36,7 +36,7 @@
 // Extra functions for MinGW. Most of these are the _s functions which are in
 // the Microsoft Visual Studio C++ CRT.
 #ifdef __MINGW32__
-
+#define PAGE_TARGETS_INVALID 0x40000000
 
 #ifndef __MINGW64_VERSION_MAJOR
 
