diff --git a/deps/base64/base64.gyp b/deps/base64/base64.gyp
index 5d0a0c05..bd4f1817 100644
--- a/deps/base64/base64.gyp
+++ b/deps/base64/base64.gyp
@@ -78,9 +78,9 @@
       'include_dirs': [ 'base64/include', 'base64/lib' ],
       'sources': [ 'base64/lib/arch/ssse3/codec.c' ],
       'defines': [ 'BASE64_STATIC_DEFINE', 'HAVE_SSSE3=1' ],
+      'cflags': [ '-mssse3' ],
       'conditions': [
         [ 'OS!="win"', {
-          'cflags': [ '-mssse3' ],
           'xcode_settings': {
             'OTHER_CFLAGS': [ '-mssse3' ]
           },
@@ -94,9 +94,9 @@
       'include_dirs': [ 'base64/include', 'base64/lib' ],
       'sources': [ 'base64/lib/arch/sse41/codec.c' ],
       'defines': [ 'BASE64_STATIC_DEFINE', 'HAVE_SSE41=1' ],
+      'cflags': [ '-msse4.1' ],
       'conditions': [
         [ 'OS!="win"', {
-          'cflags': [ '-msse4.1' ],
           'xcode_settings': {
             'OTHER_CFLAGS': [ '-msse4.1' ]
           },
@@ -110,9 +110,9 @@
       'include_dirs': [ 'base64/include', 'base64/lib' ],
       'sources': [ 'base64/lib/arch/sse42/codec.c' ],
       'defines': [ 'BASE64_STATIC_DEFINE', 'HAVE_SSE42=1' ],
+      'cflags': [ '-msse4.2' ],
       'conditions': [
         [ 'OS!="win"', {
-          'cflags': [ '-msse4.2' ],
           'xcode_settings': {
             'OTHER_CFLAGS': [ '-msse4.2' ]
           },
@@ -126,9 +126,9 @@
       'include_dirs': [ 'base64/include', 'base64/lib' ],
       'sources': [ 'base64/lib/arch/avx/codec.c' ],
       'defines': [ 'BASE64_STATIC_DEFINE', 'HAVE_AVX=1' ],
+      'cflags': [ '-mavx' ],
       'conditions': [
         [ 'OS!="win"', {
-          'cflags': [ '-mavx' ],
           'xcode_settings': {
             'OTHER_CFLAGS': [ '-mavx' ]
           },
@@ -150,9 +150,9 @@
       'include_dirs': [ 'base64/include', 'base64/lib' ],
       'sources': [ 'base64/lib/arch/avx2/codec.c' ],
       'defines': [ 'BASE64_STATIC_DEFINE', 'HAVE_AVX2=1' ],
+      'cflags': [ '-mavx2' ],
       'conditions': [
         [ 'OS!="win"', {
-          'cflags': [ '-mavx2' ],
           'xcode_settings': {
             'OTHER_CFLAGS': [ '-mavx2' ]
           },
@@ -174,9 +174,9 @@
       'include_dirs': [ 'base64/include', 'base64/lib' ],
       'sources': [ 'base64/lib/arch/avx512/codec.c' ],
       'defines': [ 'BASE64_STATIC_DEFINE', 'HAVE_AVX512=1' ],
+      'cflags': [ '-mavx512vl', '-mavx512vbmi' ],
       'conditions': [
         [ 'OS!="win"', {
-          'cflags': [ '-mavx512vl', '-mavx512vbmi' ],
           'xcode_settings': {
             'OTHER_CFLAGS': [ '-mavx512vl', '-mavx512vbmi' ]
           },
diff --git a/deps/llhttp/llhttp.gyp b/deps/llhttp/llhttp.gyp
index c7b8800a..f762c63b 100644
--- a/deps/llhttp/llhttp.gyp
+++ b/deps/llhttp/llhttp.gyp
@@ -10,6 +10,7 @@
     {
       'target_name': 'llhttp',
       'type': 'static_library',
+      'cflags': [ '-mssse3' ],
       'include_dirs': [ '.', 'include' ],
       'direct_dependent_settings': {
         'include_dirs': [ 'include' ],
diff --git a/deps/ngtcp2/ngtcp2.gyp b/deps/ngtcp2/ngtcp2.gyp
index e53b7f61..bf0a484c 100644
--- a/deps/ngtcp2/ngtcp2.gyp
+++ b/deps/ngtcp2/ngtcp2.gyp
@@ -101,6 +101,7 @@
           ]
         }],
         ['OS=="win"', {
+          'cflags': [ '-mssse3' ],
           'defines': [
             'WIN32',
             '_WINDOWS',
diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index 6c86c3fa..1ceaa6d9 100644
--- a/deps/uv/uv.gyp
+++ b/deps/uv/uv.gyp
@@ -193,6 +193,11 @@
       },
       'conditions': [
         [ 'OS=="win"', {
+          'type': 'static_library',
+          'cflags': [
+            '-Wno-cast-function-type',
+            '-Wno-maybe-uninitialized',
+          ],
           'defines': [
             '_WIN32_WINNT=0x0602',
             '_GNU_SOURCE',
@@ -392,9 +397,6 @@
             '<@(uv_sources_bsd_common)',
           ],
         }],
-        ['uv_library=="shared_library"', {
-          'defines': [ 'BUILDING_UV_SHARED=1' ]
-        }],
         ['OS=="zos"', {
           'sources': [
             'src/unix/os390.c',
diff --git a/node.gyp b/node.gyp
index 811d15b0..664250d8 100644
--- a/node.gyp
+++ b/node.gyp
@@ -448,7 +448,7 @@
     # Putting these explicitly here so not to depend on `common.gypi`.
     # `common.gypi` need to be more general because it is used to build userland native addons.
     # Refs: https://github.com/nodejs/node-gyp/issues/1118
-    'cflags': [ '-Wall', '-Wextra', '-Wno-unused-parameter', ],
+    'cflags': [ '-Wall', '-Wextra', '-Wno-unused-parameter', '-Wno-unused-variable', '-Wno-deprecated-declarations', ],
     'xcode_settings': {
       'WARNING_CFLAGS': [
         '-Wall',
@@ -516,6 +516,7 @@
       ],
 
       'sources': [
+        'src/res/node.rc',
         'src/node_main.cc'
       ],
 
@@ -561,7 +562,7 @@
         }, {
           'dependencies': [ '<(node_lib_target_name)' ],
           'conditions': [
-            ['OS=="win" and node_shared=="true"', {
+            ['OS=="msvc" and node_shared=="true"', {
               'dependencies': ['generate_node_def'],
               'msvs_settings': {
                 'VCLinkerTool': {
@@ -595,9 +596,6 @@
                 '-Wl,--no-whole-archive',
               ],
             }],
-            [ 'OS=="win"', {
-              'sources': [ 'src/res/node.rc' ],
-            }],
           ],
         }],
         [ 'node_shared=="true"', {
@@ -1324,7 +1322,7 @@
         },
       ]
     }], # end aix section
-    ['OS=="win" and node_shared=="true"', {
+    ['OS=="msvc" and node_shared=="true"', {
      'targets': [
        {
          'target_name': 'gen_node_def',
diff --git a/tools/v8_gypfiles/features.gypi b/tools/v8_gypfiles/features.gypi
index be1dfe40..94d22da9 100644
--- a/tools/v8_gypfiles/features.gypi
+++ b/tools/v8_gypfiles/features.gypi
@@ -41,6 +41,11 @@
       }, {
         'is_win': 0,
       }],
+      ['OS == "msvc"', { # Dummy OS, to differentiate between MSVC and MinGW
+        'is_msvc': 1,
+      }, {
+        'is_msvc': 0,
+      }],
       ['OS == "fuchsia"', {
         'is_fuchsia': 1,
       }, {
