diff --git a/common.gypi b/common.gypi
index c6b9fa05..229814b0 100644
--- a/common.gypi
+++ b/common.gypi
@@ -419,11 +419,11 @@
         'defines': ['V8_ENABLE_DIRECT_LOCAL',],
       }],
       ['OS == "win"', {
+        'cflags_cc': [ '-std=gnu++20' ],
         'defines': [
           'WIN32',
           # we don't really want VC++ warning us about
           # how dangerous C functions are...
-          '_CRT_SECURE_NO_DEPRECATE',
           # ... or that C implementations shouldn't use
           # POSIX names
           '_CRT_NONSTDC_NO_DEPRECATE',
