--- node-v18.12.1-orig/src/node_http2.cc	2022-11-05 00:13:14.000000000 +0800
+++ node-v18.12.1/src/node_http2.cc	2022-11-15 21:03:20.571885800 +0800
@@ -644,7 +644,7 @@
           duration,
           statistics_);
 
-  env()->SetImmediate([entry = move(entry)](Environment* env) {
+  env()->SetImmediate([entry = std::move(entry)](Environment* env) {
     if (HasHttp2Observer(env))
       entry->Notify(env);
   });
