diff --git a/deps/v8/BUILD.gn b/deps/v8/BUILD.gn
index e2e0c16d..0908d233 100644
--- a/deps/v8/BUILD.gn
+++ b/deps/v8/BUILD.gn
@@ -42,7 +42,7 @@ declare_args() {
   v8_enable_system_instrumentation = (is_win || is_mac) && !v8_use_perfetto
 
   # Sets -DV8_ENABLE_ETW_STACK_WALKING. Enables ETW Stack Walking
-  v8_enable_etw_stack_walking = is_win
+  v8_enable_etw_stack_walking = is_msvc
 
   # Sets the GUID for the ETW provider
   v8_etw_guid = ""
diff --git a/tools/v8_gypfiles/features.gypi b/tools/v8_gypfiles/features.gypi
index db9bda60..e1c72e4f 100644
--- a/tools/v8_gypfiles/features.gypi
+++ b/tools/v8_gypfiles/features.gypi
@@ -68,7 +68,7 @@
       }, {
         'v8_enable_system_instrumentation': 0,
       }],
-      ['OS == "win"', {
+      ['OS == "msvc"', {
         'v8_enable_etw_stack_walking': 1,
       }, {
         'v8_enable_etw_stack_walking': 0,
