diff --git a/deps/v8/gni/v8.gni b/deps/v8/gni/v8.gni
index bf265b29..74e5e8e1 100644
--- a/deps/v8/gni/v8.gni
+++ b/deps/v8/gni/v8.gni
@@ -192,7 +192,7 @@ declare_args() {
   v8_enable_memory_sealing = false
 
   # Sets -DV8_ENABLE_ETW_STACK_WALKING. Enables ETW Stack Walking
-  v8_enable_etw_stack_walking = is_win
+  v8_enable_etw_stack_walking = is_msvc
 }
 
 if (v8_use_external_startup_data == "") {
diff --git a/tools/v8_gypfiles/features.gypi b/tools/v8_gypfiles/features.gypi
index 9b2917e4..1c71c2e4 100644
--- a/tools/v8_gypfiles/features.gypi
+++ b/tools/v8_gypfiles/features.gypi
@@ -68,7 +68,7 @@
       }, {
         'v8_enable_system_instrumentation': 0,
       }],
-      ['OS == "win"', {
+      ['OS == "msvc"', {
         'v8_enable_etw_stack_walking': 1,
       }, {
         'v8_enable_etw_stack_walking': 0,
