diff --git a/deps/v8/BUILD.gn b/deps/v8/BUILD.gn
index 7e1e4b83..80ca3ad7 100644
--- a/deps/v8/BUILD.gn
+++ b/deps/v8/BUILD.gn
@@ -42,7 +42,7 @@ declare_args() {
   v8_enable_system_instrumentation = (is_win || is_mac) && !v8_use_perfetto
 
   # Sets -DV8_ENABLE_ETW_STACK_WALKING. Enables ETW Stack Walking
-  v8_enable_etw_stack_walking = is_win
+  v8_enable_etw_stack_walking = is_msvc
 
   # Sets the GUID for the ETW provider
   v8_etw_guid = ""
diff --git a/deps/v8/src/api/api.cc b/deps/v8/src/api/api.cc
index 7807545d..859b396c 100644
--- a/deps/v8/src/api/api.cc
+++ b/deps/v8/src/api/api.cc
@@ -10234,7 +10234,7 @@ void v8::Isolate::LocaleConfigurationChangeNotification() {
 #endif  // V8_INTL_SUPPORT
 }
 
-#if defined(V8_OS_WIN)
+#if defined(V8_OS_WIN) && defined(V8_ENABLE_ETW_STACK_WALKING)
 void Isolate::SetFilterETWSessionByURLCallback(
     FilterETWSessionByURLCallback callback) {
   i::Isolate* i_isolate = reinterpret_cast<i::Isolate*>(this);
diff --git a/tools/v8_gypfiles/features.gypi b/tools/v8_gypfiles/features.gypi
index be1dfe40..7436b7a0 100644
--- a/tools/v8_gypfiles/features.gypi
+++ b/tools/v8_gypfiles/features.gypi
@@ -68,7 +68,7 @@
       }, {
         'v8_enable_system_instrumentation': 0,
       }],
-      ['OS == "win"', {
+      ['OS == "msvc"', {
         'v8_enable_etw_stack_walking': 1,
       }, {
         'v8_enable_etw_stack_walking': 0,
