--- node-v18.12.1-orig/deps/v8/third_party/googletest/BUILD.gn	2022-11-05 00:13:14.000000000 +0800
+++ node-v18.12.1/deps/v8/third_party/googletest/BUILD.gn	2022-11-25 01:35:14.998258500 +0800
@@ -32,7 +32,7 @@
     "src/googletest/include",
   ]
 
-  if (is_win) {
+  if (is_msvs) {
     cflags = [ "/wd4800" ]  # Unused variable warning.
   }
 }
--- node-v18.12.1-orig/deps/v8/third_party/jsoncpp/BUILD.gn	2022-11-05 00:13:14.000000000 +0800
+++ node-v18.12.1/deps/v8/third_party/jsoncpp/BUILD.gn	2022-11-25 01:35:40.981744700 +0800
@@ -12,7 +12,7 @@
   # declaration flag.
   # This temporary flag allowing clients to update to the new version, and then
   # update to the new StreamWriter and CharReader classes.
-  if (!is_win || is_clang) {
+  if (!is_msvs|| is_clang) {
     cflags_cc = [ "-Wno-deprecated-declarations" ]
   }
 }
@@ -42,7 +42,7 @@
 
   include_dirs = [ "source/src/lib_json" ]
 
-  if (!is_win || is_clang) {
+  if (!is_msvs || is_clang) {
     cflags_cc = [ "-Wno-implicit-fallthrough" ]
   }
 }
--- node-v18.12.1-orig/deps/v8/third_party/zlib/BUILD.gn	2022-11-05 00:13:14.000000000 +0800
+++ node-v18.12.1/deps/v8/third_party/zlib/BUILD.gn	2022-11-25 01:38:40.292000700 +0800
@@ -28,7 +28,7 @@
     defines += [ "ZLIB_DEBUG" ]
   }
 
-  if (is_win && !is_clang) {
+  if (is_msvs && !is_clang) {
     # V8 supports building with msvc, these silence some warnings that
     # causes compilation to fail (https://crbug.com/1255096).
     cflags = [
@@ -91,7 +91,7 @@
       "adler32_simd.h",
     ]
 
-    if (!is_win || is_clang) {
+    if (!is_msvs || is_clang) {
       cflags = [ "-mssse3" ]
     }
   }
@@ -218,7 +218,7 @@
       "crc32_simd.h",
     ]
 
-    if (!is_win || is_clang) {
+    if (!is_msvs || is_clang) {
       cflags = [
         "-msse4.2",
         "-mpclmul",
@@ -250,7 +250,7 @@
       "fill_window_sse.c",
     ]
 
-    if (!is_win || is_clang) {
+    if (!is_msvs || is_clang) {
       cflags = [
         "-msse4.2",
         "-mpclmul",
--- node-v18.12.1-orig/deps/v8/third_party/zlib/cpu_features.c	2022-11-05 00:13:14.000000000 +0800
+++ node-v18.12.1/deps/v8/third_party/zlib/cpu_features.c	2022-11-26 00:32:57.486807600 +0800
@@ -9,7 +9,7 @@
 #include "zutil.h"
 
 #include <stdint.h>
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #include <intrin.h>
 #elif defined(ADLER32_SIMD_SSSE3)
 #include <cpuid.h>
@@ -46,7 +46,7 @@
 #include <zircon/types.h>
 #elif defined(ARMV8_OS_WINDOWS) || defined(X86_WINDOWS)
 #include <windows.h>
-#elif !defined(_MSC_VER)
+#elif !defined(_WIN32)
 #include <pthread.h>
 #else
 #error cpu_features.c CPU feature detection in not defined for your platform
@@ -143,7 +143,7 @@
     int x86_cpu_has_pclmulqdq;
     int abcd[4];
 
-#ifdef _MSC_VER
+#ifdef _WIN32
     __cpuid(abcd, 1);
 #else
     __cpuid(1, abcd[0], abcd[1], abcd[2], abcd[3]);
