diff --git a/deps/v8/src/base/platform/platform-win32.cc b/deps/v8/src/base/platform/platform-win32.cc
index fa03a80e..38fa8f10 100644
--- a/deps/v8/src/base/platform/platform-win32.cc
+++ b/deps/v8/src/base/platform/platform-win32.cc
@@ -69,6 +69,12 @@ static_assert(offsetof(V8_CRITICAL_SECTION, SpinCount) ==
 // the Microsoft Visual Studio C++ CRT.
 #ifdef __MINGW32__
 
+// FIXME: Missing from <sysinfoapi.h>
+BOOL IsUserCetAvailableInEnvironment(
+  DWORD UserCetEnvironment
+);
+
+#define USER_CET_ENVIRONMENT_WIN32_PROCESS 0x00000000U
 
 #ifndef __MINGW64_VERSION_MAJOR
 
diff --git a/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h b/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h
index acfc15a8..a3f3df78 100644
--- a/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h
+++ b/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h
@@ -20,7 +20,7 @@
 #ifndef ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 #define ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 #include <pthread.h>
 // Defines __GOOGLE_GRTE_VERSION__ (via glibc-specific features.h) when
 // supported.
diff --git a/deps/v8/third_party/abseil-cpp/absl/time/internal/cctz/src/time_zone_lookup.cc b/deps/v8/third_party/abseil-cpp/absl/time/internal/cctz/src/time_zone_lookup.cc
index 89791745..eb1fe66f 100644
--- a/deps/v8/third_party/abseil-cpp/absl/time/internal/cctz/src/time_zone_lookup.cc
+++ b/deps/v8/third_party/abseil-cpp/absl/time/internal/cctz/src/time_zone_lookup.cc
@@ -47,6 +47,7 @@
 #include <wchar.h>
 #include <windows.globalization.h>
 #include <windows.h>
+#include <winstring.h> // for WindowsCreateStringReference
 #endif
 #endif
 
diff --git a/deps/v8/third_party/zlib/cpu_features.c b/deps/v8/third_party/zlib/cpu_features.c
index 34ae7b91..746654e6 100644
--- a/deps/v8/third_party/zlib/cpu_features.c
+++ b/deps/v8/third_party/zlib/cpu_features.c
@@ -9,7 +9,7 @@
 #include "zutil.h"
 
 #include <stdint.h>
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #include <intrin.h>
 #elif defined(ADLER32_SIMD_SSSE3)
 #include <cpuid.h>
@@ -56,7 +56,7 @@ int ZLIB_INTERNAL riscv_cpu_enable_vclmul = 0;
 #include <windows.h>
 #elif defined(ARMV8_OS_IOS)
 #include <sys/sysctl.h>
-#elif !defined(_MSC_VER)
+#elif !defined(_WIN32)
 #include <pthread.h>
 #else
 #error cpu_features.c CPU feature detection in not defined for your platform
@@ -168,7 +168,7 @@ static void _cpu_check_features(void)
     int x86_cpu_has_pclmulqdq;
     int abcd[4];
 
-#ifdef _MSC_VER
+#ifdef _WIN32
     __cpuid(abcd, 1);
 #else
     __cpuid(1, abcd[0], abcd[1], abcd[2], abcd[3]);
