diff --git a/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h b/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h
index b6e917ce..8b298c61 100644
--- a/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h
+++ b/deps/v8/third_party/abseil-cpp/absl/base/internal/thread_identity.h
@@ -20,7 +20,7 @@
 #ifndef ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 #define ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 #include <pthread.h>
 // Defines __GOOGLE_GRTE_VERSION__ (via glibc-specific features.h) when
 // supported.
diff --git a/deps/v8/third_party/zlib/cpu_features.c b/deps/v8/third_party/zlib/cpu_features.c
index 64e0428c..8f69b299 100644
--- a/deps/v8/third_party/zlib/cpu_features.c
+++ b/deps/v8/third_party/zlib/cpu_features.c
@@ -9,7 +9,7 @@
 #include "zutil.h"
 
 #include <stdint.h>
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #include <intrin.h>
 #elif defined(ADLER32_SIMD_SSSE3)
 #include <cpuid.h>
@@ -52,7 +52,7 @@ int ZLIB_INTERNAL x86_cpu_enable_avx512 = 0;
 #include <windows.h>
 #elif defined(ARMV8_OS_IOS)
 #include <sys/sysctl.h>
-#elif !defined(_MSC_VER)
+#elif !defined(_WIN32)
 #include <pthread.h>
 #else
 #error cpu_features.c CPU feature detection in not defined for your platform
@@ -161,7 +161,7 @@ static void _cpu_check_features(void)
     int x86_cpu_has_pclmulqdq;
     int abcd[4];
 
-#ifdef _MSC_VER
+#ifdef _WIN32
     __cpuid(abcd, 1);
 #else
     __cpuid(1, abcd[0], abcd[1], abcd[2], abcd[3]);
