diff --git a/common.gypi b/common.gypi
index c6b9fa05..d6eb8655 100644
--- a/common.gypi
+++ b/common.gypi
@@ -429,8 +429,6 @@
           '_CRT_NONSTDC_NO_DEPRECATE',
           # Make sure the STL doesn't try to use exceptions
           '_HAS_EXCEPTIONS=0',
-          'BUILDING_V8_SHARED=1',
-          'BUILDING_UV_SHARED=1',
         ],
       }],
       [ 'OS in "linux freebsd openbsd solaris aix os400"', {
diff --git a/node.gypi b/node.gypi
index 95133818..02d0878a 100644
--- a/node.gypi
+++ b/node.gypi
@@ -44,8 +44,6 @@
       'conditions': [
         ['"<(_type)"=="executable"', {
           'defines': [
-            'USING_UV_SHARED',
-            'USING_V8_SHARED',
             'BUILDING_NODE_EXTENSION'
           ],
           'defines!': [
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 4d69a59f..920d5d15 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -1841,6 +1841,17 @@
                 'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']
               },
             }],
+            ['OS=="win"', {
+              # We are building / using the static library
+              'defines!': [
+                'BUILDING_V8_SHARED',
+              ],
+              'direct_dependent_settings': {
+                'defines!': [
+                  'USING_V8_SHARED',
+                ],
+              },
+            }],
             ['soname_version!=""', {
               'product_extension': 'so.<(soname_version)',
             }],
