--- uv/config-mingw.mk.orig	2015-01-12 13:49:48.669668300 +0000
+++ uv/config-mingw.mk	2015-01-12 13:53:37.714253300 +0000
@@ -24,7 +24,7 @@
 AR = $(PREFIX)ar
 E=.exe
 
-CFLAGS=$(CPPFLAGS) -g --std=gnu89 -D_WIN32_WINNT=0x0600
+CFLAGS=$(CPPFLAGS) -g --std=gnu89 -D_WIN32_WINNT=0x0501
 LDFLAGS=-lm
 
 WIN_SRCS=$(wildcard $(SRCDIR)/src/win/*.c)
--- uv/uv.gyp.orig	2015-01-12 14:08:34.044072700 +0000
+++ uv/uv.gyp	2015-01-12 14:08:38.683661900 +0000
@@ -70,7 +70,7 @@
       'conditions': [
         [ 'OS=="win"', {
           'defines': [
-            '_WIN32_WINNT=0x0600',
+            '_WIN32_WINNT=0x0501',
             '_GNU_SOURCE',
           ],
           'sources': [
