--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -3,7 +3,7 @@
 # Use of this source code is governed by a BSD-style
 # license that can be found in the LICENSE file.
 
-if(NOT WIN32)
+if(NOT MSVC)
   add_subdirectory(gdk-pixbuf)
 endif()
 
--- a/plugins/gdk-pixbuf/CMakeLists.txt
+++ b/plugins/gdk-pixbuf/CMakeLists.txt
@@ -25,8 +25,8 @@
 # shared library.
 target_link_libraries(pixbufloader-jxl jxl jxl_threads skcms-interface PkgConfig::Gdk-Pixbuf)
 
-pkg_get_variable(GDK_PIXBUF_MODULEDIR gdk-pixbuf-2.0 gdk_pixbuf_moduledir)
-install(TARGETS pixbufloader-jxl LIBRARY DESTINATION "${GDK_PIXBUF_MODULEDIR}")
+execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} gdk-pixbuf-2.0 --variable gdk_pixbuf_moduledir --define-variable=prefix=${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE GDK_PIXBUF_MODULEDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+install(TARGETS pixbufloader-jxl DESTINATION "${GDK_PIXBUF_MODULEDIR}")
 
 # Instead of the following, we might instead add the
 # mime type image/jxl to
--- a/plugins/gimp/CMakeLists.txt
+++ b/plugins/gimp/CMakeLists.txt
@@ -24,5 +24,5 @@
 target_include_directories(file-jxl PUBLIC
     ${PROJECT_SOURCE_DIR})  # for plugins/gimp absolute paths.
 
-pkg_get_variable(GIMP_LIB_DIR gimp-2.0 gimplibdir)
-install(TARGETS file-jxl RUNTIME DESTINATION "${GIMP_LIB_DIR}/plug-ins/file-jxl/")
+execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} gimp-2.0 --variable gimplibdir --define-variable=prefix=${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE GIMP_LIB_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+install(TARGETS file-jxl DESTINATION "${GIMP_LIB_DIR}/plug-ins/file-jxl/")
