diff -bur libjxl-c/tools/cpu/cpu.cc libjxl/tools/cpu/cpu.cc
--- libjxl-c/tools/cpu/cpu.cc	2022-08-20 01:57:22.755144900 -0600
+++ libjxl/tools/cpu/cpu.cc	2022-08-20 02:09:28.859810900 -0600
@@ -32,6 +32,10 @@
 #include <string>
 #include <vector>
 
+#ifdef __MINGW32__
+#include <Windows.h>
+#endif
+
 #include "lib/jxl/base/bits.h"
 #include "lib/jxl/base/compiler_specific.h"
 #include "lib/jxl/base/status.h"
@@ -408,7 +412,13 @@
   // TODO(deymo): Actually look up the CPU topology and model the big/small
   // core split.
   pt->logical_per_core = 1;
+#ifdef __MINGW32__
+  SYSTEM_INFO sysInfo;
+  GetSystemInfo(&sysInfo);
+  pt->cores_per_package = sysInfo.dwNumberOfProcessors;
+#else  
   pt->cores_per_package = sysconf(_SC_NPROCESSORS_ONLN);
+#endif  
   pt->packages = 1;
   return true;
 #endif
