diff -bur breeze-6.0.2-orig/CMakeLists.txt breeze-6.0.2/CMakeLists.txt
--- breeze-6.0.2-orig/CMakeLists.txt	2024-03-15 11:37:01.214372900 -0600
+++ breeze-6.0.2/CMakeLists.txt	2024-03-15 11:37:03.527418600 -0600
@@ -118,7 +118,7 @@
         I18n
         IconThemes
         WindowSystem)
-    if(NOT WIN32 AND NOT APPLE)
+    if(NOT APPLE)
         find_package(KF6KCMUtils ${KF6_MIN_VERSION})
         set_package_properties(KF6KCMUtils PROPERTIES
             TYPE REQUIRED
diff -bur breeze-6.0.2-orig/kstyle/config/main.cpp breeze-6.0.2/kstyle/config/main.cpp
--- breeze-6.0.2-orig/kstyle/config/main.cpp	2024-03-15 11:37:01.189488300 -0600
+++ breeze-6.0.2/kstyle/config/main.cpp	2024-03-15 11:37:09.884939500 -0600
@@ -28,7 +28,8 @@
     KCMultiDialog dialog;
     dialog.setWindowTitle(i18n("Breeze Settings"));
     dialog.addModule(KPluginMetaData(QStringLiteral("kstyle_config/breezestyleconfig")));
-    dialog.addModule(KPluginMetaData(QStringLiteral("org.kde.kdecoration2.kcm/kcm_breezedecoration.so")));
+    // No KDecoration2 in Windows
+    //dialog.addModule(KPluginMetaData(QStringLiteral("org.kde.kdecoration2.kcm/kcm_breezedecoration.so")));
     dialog.show();
 
     const auto children = dialog.findChildren<QAbstractScrollArea *>();
