--- gw-1.2.6/src/drawing.cpp.orig	2025-10-30 14:42:35.000000000 +0100
+++ gw-1.2.6/src/drawing.cpp	2026-01-17 15:39:43.827489100 +0100
@@ -408,7 +408,7 @@
         points[3] = SkPoint::Make(endX, endY);
         points[4] = SkPoint::Make(endX, yScaledOffset);
         path.reset();
-        path.addPoly(points, 5, true);
+        path.addPoly(points, true);
         canvas->drawPath(path, faceColor);
     }
 
@@ -426,7 +426,7 @@
         points[3] = SkPoint::Make(slopedEndX, midY);
         points[4] = SkPoint::Make(endX, yScaledOffset);
         path.reset();
-        path.addPoly(points, 5, true);
+        path.addPoly(points, true);
         canvas->drawPath(path, faceColor);
     }
 
@@ -444,13 +444,13 @@
         points[3] = SkPoint::Make(endX, endY);
         points[4] = SkPoint::Make(endX, yScaledOffset);
         path.reset();
-        path.addPoly(points, 5, true);
+        path.addPoly(points, true);
         canvas->drawPath(path, faceColor);
         if (edged) {
             points[0].fY += 0.5;
             points[4].fY += 0.5;
             path.reset();
-            path.addPoly(points, 5, true);
+            path.addPoly(points, true);
             canvas->drawPath(path, edgeColor);
         }
     }
@@ -470,13 +470,13 @@
         points[3] = SkPoint::Make(slopedEndX, midY);
         points[4] = SkPoint::Make(endX, yScaledOffset);
         path.reset();
-        path.addPoly(points, 5, true);
+        path.addPoly(points, true);
         canvas->drawPath(path, faceColor);
         if (edged) {
             points[0].fY += 0.5;
             points[4].fY += 0.5;
             path.reset();
-            path.addPoly(points, 5, true);
+            path.addPoly(points, true);
             canvas->drawPath(path, edgeColor);
         }
     }
--- gw-1.2.6/src/themes.cpp.orig	2025-10-30 14:42:35.000000000 +0100
+++ gw-1.2.6/src/themes.cpp	2026-01-17 15:40:20.403684900 +0100
@@ -156,7 +156,7 @@
 
         lcGTFJoins.setStyle(SkPaint::kStroke_Style);
         lcGTFJoins.setStrokeWidth(2);
-        lcGTFJoins.setPathEffect(SkDashPathEffect::Make(intervals, 2, 0.0f));
+        lcGTFJoins.setPathEffect(SkDashPathEffect::Make(intervals, 0.0f));
 
         lcLabel.setStyle(SkPaint::kStroke_Style);
         lcLabel.setStrokeWidth(1);
@@ -1306,7 +1306,7 @@
         SkPaint paint1;
         const SkPaint* pnt = &paint1;
         overlay.setSize(fontTypefaceSize * yScale);
-        overlay.getBounds(glyphs, 1, bounds, pnt);
+        overlay.getBounds({glyphs, 1}, {bounds, 1}, pnt);
         overlayCharWidth = bounds[0].width();
         overlayHeight = bounds[0].height();
         overlayWidth = overlay.measureText("Gy", 1, SkTextEncoding::kUTF8);
