From 5368ef34b24109f78c5498a42a263152db6e6d41 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 22 Nov 2014 22:15:32 +0000
Subject: [PATCH 1/6] MinGW-w64: fix redeclaration of MFAudioFormat_Dolby_AC3

---
 modules/codec/mft.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/codec/mft.c b/modules/codec/mft.c
index 67a3dd6..eb49a96 100644
--- a/modules/codec/mft.c
+++ b/modules/codec/mft.c
@@ -172,7 +172,9 @@ static const pair_format_guid video_format_table[] =
     { 0, NULL }
 };
 
+#ifndef __MINGW64_VERSION_MAJOR
 DEFINE_GUID(MFAudioFormat_Dolby_AC3, 0xe06d802c, 0xdb46, 0x11cf, 0xb4, 0xd1, 0x00, 0x80, 0x5f, 0x6c, 0xbb, 0xea);
+#endif
 /*
  * We cannot use the FOURCC code for audio either since the
  * WAVE_FORMAT value is used to create the GUID.
-- 
2.1.3

