--- vlc/modules/codec/mft.c.orig	2014-09-25 16:26:05.454200000 +0500
+++ vlc/modules/codec/mft.c	2014-09-25 16:43:53.934800000 +0500
@@ -172,7 +172,9 @@
     { 0, NULL }
 };
 
+#ifndef __MINGW64_VERSION_MAJOR
 DEFINE_GUID(MFAudioFormat_Dolby_AC3, 0xe06d802c, 0xdb46, 0x11cf, 0xb4, 0xd1, 0x00, 0x80, 0x5f, 0x6c, 0xbb, 0xea);
+#endif
 /*
  * We cannot use the FOURCC code for audio either since the
  * WAVE_FORMAT value is used to create the GUID.
