--- vlc/modules/access/dtv/access.c	2014-10-15 18:15:34.494531200 +0100
+++ vlc/modules/access/dtv/access.c	2014-10-15 18:24:55.339493900 +0100
@@ -727,8 +727,13 @@
     char *mod = var_InheritString (obj, var);
     if (mod == NULL)
         return "";
-
-    size_t n = sizeof (modulation_vlc) / sizeof (modulation_vlc[0]);
+    
+#ifdef __MINGW32__
+    unsigned int n;
+#else
+    size_t_n;
+#endif
+    n = sizeof (modulation_vlc) / sizeof (modulation_vlc[0]);
     const char *const *p = lfind (mod, modulation_vlc, &n, sizeof (mod), modcmp);
     if (p != NULL)
     {
