From 4c2152fe4107ac37b2ada91af87ca8189b641eac Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 22 Nov 2014 22:15:33 +0000
Subject: [PATCH 2/7] MinGW-w64: lfind's _NumOfElements is an unsigned int

---
 modules/access/dtv/access.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/modules/access/dtv/access.c b/modules/access/dtv/access.c
index 95b1dd7..433e65d 100644
--- a/modules/access/dtv/access.c
+++ b/modules/access/dtv/access.c
@@ -727,8 +727,13 @@ static const char *var_InheritModulation (vlc_object_t *obj, const char *var)
     char *mod = var_InheritString (obj, var);
     if (mod == NULL)
         return "";
-
-    size_t n = sizeof (modulation_vlc) / sizeof (modulation_vlc[0]);
+
+#ifdef __MINGW32__
+    unsigned int n;
+#else
+    size_t_n;
+#endif
+    n = sizeof (modulation_vlc) / sizeof (modulation_vlc[0]);
     const char *const *p = lfind (mod, modulation_vlc, &n, sizeof (mod), modcmp);
     if (p != NULL)
     {
-- 
2.1.3

