--- vlc/modules/gui/qt4/Makefile.am	2014-10-16 20:03:35.772278000 +0100
+++ vlc/modules/gui/qt4/Makefile.am	2014-10-16 20:04:00.121609700 +0100
@@ -13,14 +13,14 @@
 #   - Add it to DEPS_res
 
 SUFFIXES += .ui .h .hpp .moc.cpp
-
+QT_CFLAGS += -DNO_TIME_PLZ
 libqt4_plugin_la_CXXFLAGS = $(AM_CXXFLAGS) $(QT_CFLAGS) $(CXXFLAGS_qt4)
 libqt4_plugin_la_LIBADD = $(QT_LIBS) $(LIBS_qt4)
 libqt4_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(guidir)'
 if HAVE_OS2
 else
 if HAVE_WIN32
-libqt4_plugin_la_LIBADD += -lole32
+libqt4_plugin_la_LIBADD += -lole32 -lcomctl32
 else
 libqt4_plugin_la_LIBADD += $(X_LIBS) $(X_PRE_LIB) -lX11
 endif
--- vlc/modules/gui/qt4/qt4.cpp	2014-10-17 03:50:18.069299400 +0100
+++ vlc/modules/gui/qt4/qt4.cpp	2014-10-17 03:48:50.365549900 +0100
@@ -52,8 +52,10 @@
 #ifdef _WIN32 /* For static builds */
  #include <QtPlugin>
  #if HAS_QT5
-  Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin)
-  Q_IMPORT_PLUGIN(AccessibleFactory)
+  #ifdef QT_STATICPLUGIN
+   Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin)
+   Q_IMPORT_PLUGIN(AccessibleFactory)
+  #endif
  #else
   Q_IMPORT_PLUGIN(qjpeg)
   Q_IMPORT_PLUGIN(qtaccessiblewidgets)

#@@ -13,7 +13,7 @@
# #   - Add it to DEPS_res
# 
# SUFFIXES += .ui .h .hpp .moc.cpp
#-
#+QT_CFLAGS += -DNO_TIME_PLZ -DQT_STATICPLUGIN
# libqt4_plugin_la_CXXFLAGS = $(AM_CXXFLAGS) $(QT_CFLAGS) $(CXXFLAGS_qt4)
# libqt4_plugin_la_LIBADD = $(QT_LIBS) $(LIBS_qt4)
# libqt4_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(guidir)'
#
