From a2a48f6d68908351363c1320e78bf233eed122d1 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Thu, 11 Dec 2014 19:04:53 +0000
Subject: [PATCH] pollfd is defined from Vista onwards

---
 include/vlc_fixups.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/vlc_fixups.h b/include/vlc_fixups.h
index cee1254..86c58b8 100644
--- a/include/vlc_fixups.h
+++ b/include/vlc_fixups.h
@@ -270,6 +270,11 @@ int inet_pton(int, const char *, void *);
 const char *inet_ntop(int, const void *, char *, int);
 #endif
 
+/* Windows has pollfd from Vista+ */
+#if (_WIN32_WINNT >= 0x0600)
+#define HAVE_STRUCT_POLLFD
+#endif
+
 #ifndef HAVE_STRUCT_POLLFD
 enum
 {
-- 
2.2.0

