--- vlc/modules/codec/Makefile.am	2014-10-16 06:13:45.657952900 +0100
+++ vlc/modules/codec/Makefile.am	2014-10-16 06:40:02.767130700 +0100
@@ -40,15 +40,15 @@
 libg711_plugin_la_SOURCES = codec/g711.c
 codec_LTLIBRARIES += libg711_plugin.la
 
-libfluidsynth_plugin_la_SOURCES = codec/fluidsynth.c
-libfluidsynth_plugin_la_CFLAGS = $(AM_CFLAGS) $(FLUIDSYNTH_CFLAGS)
-libfluidsynth_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(codecdir)'
-libfluidsynth_plugin_la_LIBADD = $(FLUIDSYNTH_LIBS)
-EXTRA_LTLIBRARIES += libfluidsynth_plugin.la
-codec_LTLIBRARIES += $(LTLIBfluidsynth)
-if HAVE_DARWIN
-libfluidsynth_plugin_la_LDFLAGS += -Wl,-framework,CoreFoundation,-framework,CoreServices
-endif
+#libfluidsynth_plugin_la_SOURCES = codec/fluidsynth.c
+#libfluidsynth_plugin_la_CFLAGS = $(AM_CFLAGS) $(FLUIDSYNTH_CFLAGS)
+#libfluidsynth_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(codecdir)'
+#libfluidsynth_plugin_la_LIBADD = $(FLUIDSYNTH_LIBS)
+#EXTRA_LTLIBRARIES += libfluidsynth_plugin.la
+#codec_LTLIBRARIES += $(LTLIBfluidsynth)
+#if HAVE_DARWIN
+#libfluidsynth_plugin_la_LDFLAGS += -Wl,-framework,CoreFoundation,-framework,CoreServices
+#endif
 
 liblpcm_plugin_la_SOURCES = codec/lpcm.c
 codec_LTLIBRARIES += liblpcm_plugin.la
