--- vlc/modules/gui/qt4/Makefile.am	2014-10-16 20:03:35.772278000 +0100
+++ vlc/modules/gui/qt4/Makefile.am	2014-10-16 20:04:00.121609700 +0100
@@ -13,7 +13,7 @@
 #   - Add it to DEPS_res
 
 SUFFIXES += .ui .h .hpp .moc.cpp
-
+QT_CFLAGS += -DNO_TIME_PLZ
 libqt4_plugin_la_CXXFLAGS = $(AM_CXXFLAGS) $(QT_CFLAGS) $(CXXFLAGS_qt4)
 libqt4_plugin_la_LIBADD = $(QT_LIBS) $(LIBS_qt4)
 libqt4_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(guidir)'
