From 9814cf1ba3fc83416cf117536fbe253ab488a4ad Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Wed, 18 Feb 2015 15:28:58 +0000
Subject: [PATCH 10/11] MEDIASUBTYPE_I420 is in add wmcodecdsp.h

---
 modules/access/dshow/vlc_dshow.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/access/dshow/vlc_dshow.h b/modules/access/dshow/vlc_dshow.h
index 816884e..be98882 100644
--- a/modules/access/dshow/vlc_dshow.h
+++ b/modules/access/dshow/vlc_dshow.h
@@ -37,6 +37,7 @@
 #include <limits.h>
 #include <strmif.h>
 #include <ksmedia.h>
+#include <wmcodecdsp.h>
 #include <ddraw.h>
 
 #ifndef __MINGW64_VERSION_MAJOR
-- 
2.3.0

