--- vlc/modules/codec/Makefile.am	2014-10-16 21:12:41.677077100 +0100
+++ vlc/modules/codec/Makefile.am	2014-10-17 00:33:52.392962300 +0100
@@ -110,12 +110,12 @@
 EXTRA_LTLIBRARIES += libjpeg_plugin.la
 codec_LTLIBRARIES += $(LTLIBjpeg)
 
-libsvgdec_plugin_la_SOURCES = codec/svg.c
-libsvgdec_plugin_la_CFLAGS = $(AM_CLAGS) $(CFLAGS_svgdec)
-libsvgdec_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(codecdir)'  $(LDFLAGS_svg)
-libsvgdec_plugin_la_LIBADD = $(LIBS_svgdec)
-EXTRA_LTLIBRARIES += libsvgdec_plugin.la
-codec_LTLIBRARIES += $(LTLIBsvgdec)
+#libsvgdec_plugin_la_SOURCES = codec/svg.c
+#libsvgdec_plugin_la_CFLAGS = $(AM_CLAGS) $(CFLAGS_svgdec)
+#libsvgdec_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(codecdir)'  $(LDFLAGS_svg)
+#libsvgdec_plugin_la_LIBADD = $(LIBS_svgdec)
+#EXTRA_LTLIBRARIES += libsvgdec_plugin.la
+#codec_LTLIBRARIES += $(LTLIBsvgdec)
 
 libsdl_image_plugin_la_SOURCES = codec/sdl_image.c
 libsdl_image_plugin_la_CFLAGS = $(AM_CFLAGS) $(SDL_IMAGE_CFLAGS)
