--- vlc/modules/text_renderer/Makefile.am	2014-10-17 05:02:41.033285100 +0100
+++ vlc/modules/text_renderer/Makefile.am	2014-10-17 05:03:47.089513900 +0100
@@ -35,15 +35,15 @@
 EXTRA_LTLIBRARIES += libquartztext_plugin.la
 text_renderer_LTLIBRARIES += $(LTLIBquartztext)
 
-libsvg_plugin_la_SOURCES = $(SOURCES_svg)
-libsvg_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(CPPFLAGS_svg) 	-DMODULE_NAME_IS_svg
-libsvg_plugin_la_CFLAGS = $(AM_CFLAGS) $(CFLAGS_svg)
-libsvg_plugin_la_CXXFLAGS = $(AM_CXXFLAGS) $(CXXFLAGS_svg)
-libsvg_plugin_la_OBJCFLAGS = $(AM_OBJCFLAGS) $(OBJCFLAGS_svg)
-libsvg_plugin_la_LIBADD = $(LIBS_svg)
-libsvg_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(text_rendererdir)' $(LDFLAGS_svg)
-EXTRA_LTLIBRARIES += libsvg_plugin.la
-text_renderer_LTLIBRARIES += $(LTLIBsvg)
+#libsvg_plugin_la_SOURCES = $(SOURCES_svg)
+#libsvg_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(CPPFLAGS_svg) 	-DMODULE_NAME_IS_svg
+#libsvg_plugin_la_CFLAGS = $(AM_CFLAGS) $(CFLAGS_svg)
+#libsvg_plugin_la_CXXFLAGS = $(AM_CXXFLAGS) $(CXXFLAGS_svg)
+#libsvg_plugin_la_OBJCFLAGS = $(AM_OBJCFLAGS) $(OBJCFLAGS_svg)
+#libsvg_plugin_la_LIBADD = $(LIBS_svg)
+#libsvg_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(text_rendererdir)' $(LDFLAGS_svg)
+#EXTRA_LTLIBRARIES += libsvg_plugin.la
+#text_renderer_LTLIBRARIES += $(LTLIBsvg)
 
 libtdummy_plugin_la_SOURCES = $(SOURCES_tdummy)
 libtdummy_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(CPPFLAGS_tdummy) 	-DMODULE_NAME_IS_tdummy
