From e956727cf7fbe8fdd164a92cdc582cc839a2ec7d Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Thu, 5 Mar 2015 01:01:53 +0000
Subject: [PATCH] Add include on WINSTORE for QueueTimer functions


diff --git a/src/win32/thread.c b/src/win32/thread.c
index 2f94fab..0f62610 100644
--- a/src/win32/thread.c
+++ b/src/win32/thread.c
@@ -46,6 +46,10 @@ static vlc_cond_t  super_variable;
 
 /*** Common helpers ***/
 #if VLC_WINSTORE_APP
+#include <threadpoollegacyapiset.h>
+#ifndef TLS_OUT_OF_INDEXES
+#define TLS_OUT_OF_INDEXES (~0u)
+#endif
 static bool isCancelled(void);
 #endif
 
-- 
2.3.1

