From 5d2e3cac97b6c119e6a55f5a6c715a7932556634 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Mon, 18 May 2015 17:26:53 +0100
Subject: [PATCH 1/2] fixups: time.h is assumed in many places

---
 include/vlc_fixups.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/vlc_fixups.h b/include/vlc_fixups.h
index f844641..0c72af7 100644
--- a/include/vlc_fixups.h
+++ b/include/vlc_fixups.h
@@ -26,9 +26,7 @@
 #ifndef LIBVLC_FIXUPS_H
 # define LIBVLC_FIXUPS_H 1
 
-#if !defined (HAVE_GMTIME_R) || !defined (HAVE_LOCALTIME_R)
 # include <time.h> /* time_t */
-#endif
 
 #ifndef HAVE_LLDIV
 typedef struct
-- 
2.4.1

