--- vlc/modules/access/Makefile.am	2014-10-15 20:51:11.380079600 +0100
+++ vlc/modules/access/Makefile.am	2014-10-15 20:52:38.563742000 +0100
@@ -226,6 +226,9 @@
 libcdda_plugin_la_CFLAGS = $(AM_CFLAGS) $(LIBCDDB_CFLAGS)
 libcdda_plugin_la_LIBADD = $(LIBCDDB_LIBS)
 libcdda_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(accessdir)'
+if HAVE_WIN32
+libcdda_plugin_la_LIBADD += $(SOCKET_LIBS)
+endif
 if HAVE_DARWIN
 libcdda_plugin_la_LIBADD += -liconv
 libcdda_plugin_la_LDFLAGS += -Wl,-framework,IOKit,-framework,CoreFoundation
