--- vlc/modules/access/Makefile.am	2014-10-15 21:40:34.609377400 +0100
+++ vlc/modules/access/Makefile.am	2014-10-15 21:41:18.183410600 +0100
@@ -286,6 +286,9 @@
 liblibbluray_plugin_la_CFLAGS = $(AM_CFLAGS) $(BLURAY_CFLAGS)
 liblibbluray_plugin_la_LIBADD = $(BLURAY_LIBS)
 liblibbluray_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(accessdir)'
+if HAVE_WIN32
+liblibbluray_plugin_la_LDFLAGS += $(LIBXML2_LIBS)
+endif
 access_LTLIBRARIES += $(LTLIBlibbluray)
 EXTRA_LTLIBRARIES += liblibbluray_plugin.la
 
