--- scalapack-2.2.3/CMakeLists.txt.orig	2026-03-24 20:40:03.000000000 +0100
diff -urN scalapack-2.2.3/CMakeLists.txt.orig scalapack-2.2.3/CMakeLists.txt
+++ scalapack-2.2.3/CMakeLists.txt	2026-08-28 17:25:50.134179500 +0200
@@ -77,7 +77,9 @@
 
 macro(SCALAPACK_install_library lib)
   install(TARGETS ${lib} EXPORT scalapack-targets
-    RUNTIME DESTINATION Testing
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    RUNTIME DESTINATION bin
   )
 endmacro()
 
@@ -90,7 +92,7 @@
 
 # Organize output files.  On Windows this also keeps .dll files next
 # to the .exe files that need them, making tests easy to run.
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${SCALAPACK_BINARY_DIR}/TESTING)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${SCALAPACK_BINARY_DIR}/bin)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${SCALAPACK_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${SCALAPACK_BINARY_DIR}/lib)
 
