diff -urN scalapack-2.0.2.orig/CMakeLists.txt scalapack-2.0.2/CMakeLists.txt
--- scalapack-2.0.2.orig/CMakeLists.txt	2012-05-01 19:58:24.000000000 -0700
+++ scalapack-2.0.2/CMakeLists.txt	2019-11-22 03:00:33.652954700 -0800
@@ -243,7 +243,7 @@
    scalapack_install_library(scalapack)
    scalapack_install_library(scalapack-F)
 endif (UNIX)
-add_subdirectory(TESTING)
+#add_subdirectory(TESTING)
 
 # --------------------------------------------------
 # CPACK Packaging 
