--- a/cmake/FindFFmpeg.cmake
+++ b/cmake/FindFFmpeg.cmake
@@ -73,14 +73,14 @@
 #
 macro(find_component _component _pkgconfig _library _header)
 
-  if (NOT WIN32)
+  if (NOT MSVC)
      # use pkg-config to get the directories and then use these values
      # in the FIND_PATH() and FIND_LIBRARY() calls
      find_package(PkgConfig)
      if (PKG_CONFIG_FOUND)
        pkg_check_modules(PC_${_component} QUIET ${_pkgconfig})
      endif ()
-  endif (NOT WIN32)
+  endif (NOT MSVC)
 
   find_path(${_component}_INCLUDE_DIRS ${_header}
     HINTS
