--- ./src/libical/icallangbind.h.orig	2016-03-26 13:56:32.737612000 +0100
+++ ./src/libical/icallangbind.h	2016-03-26 13:56:34.581488900 +0100
@@ -63,6 +63,6 @@
 
 LIBICAL_ICAL_EXPORT const char *icallangbind_quote_as_ical(const char *str);
 
-LIBICAL_ICAL_EXPORT char *callangbind_quote_as_ical_r(const char *str);
+LIBICAL_ICAL_EXPORT char *icallangbind_quote_as_ical_r(const char *str);
 
 #endif
