--- libical-3.0.1/src/libical/CMakeLists.txt.orig	2018-01-29 15:15:01.298593900 +0300
+++ libical-3.0.1/src/libical/CMakeLists.txt	2018-01-29 15:15:04.740402200 +0300
@@ -17,7 +17,7 @@
   include_directories(${ICU_INCLUDE_DIR})
 endif()
 
-set(PACKAGE_DATA_DIR "\\\"${CMAKE_INSTALL_PREFIX}/share/libical\\\"")
+set(PACKAGE_DATA_DIR "${CMAKE_INSTALL_PREFIX}/share/libical")
 add_definitions(-DPACKAGE_DATA_DIR="${PACKAGE_DATA_DIR}")
 
 set(ICALSCRIPTS ${CMAKE_SOURCE_DIR}/scripts/)
