diff --git a/version.sh b/version.sh
index c8dda2f5..bd6f1a35 100755
--- a/version.sh
+++ b/version.sh
@@ -13,6 +13,9 @@ if [ -d ${GIT_DIR:-.git} ] && tag=`git describe --tags`; then
 		v="$v"-dirty
 elif [ -n "$RPM_PACKAGE_VERSION" ] && [ -n "$RPM_PACKAGE_RELEASE" ]; then
 	v="v$RPM_PACKAGE_VERSION-$RPM_PACKAGE_RELEASE"
+elif [ -n "$MINGW_PACKAGE_VERSION" ] && [ -n "$MINGW_PACKAGE_RELEASE" ]; then
+	#building with makepkg under MSYS
+	v="v$MINGW_PACKAGE_VERSION-$MINGW_PACKAGE_RELEASE"
 else # XXX: Equivalent for .deb packages?
 	v="$v"-unknown
 fi
