diff --git a/Makefile.am b/Makefile.am
index 4ce076e0..ce08613f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -18,6 +18,8 @@ endif
 lib_LTLIBRARIES = libopenconnect.la
 sbin_PROGRAMS = openconnect
 man8_MANS = openconnect.8
+bin_PROGRAMS :=
+sbin_SCRIPTS :=
 noinst_PROGRAMS :=
 
 AM_CFLAGS = @WFLAGS@
@@ -34,6 +36,7 @@ openconnect_LDADD = libopenconnect.la $(SSL_LIBS) $(LIBXML2_LIBS) \
 
 if OPENCONNECT_WIN32
 openconnect_SOURCES += openconnect.rc
+sbin_SCRIPTS += vpnc-script-win.js
 endif
 library_srcs = ssl.c http.c textbuf.c http-auth.c auth-common.c \
 	auth-html.c library.c compat.c lzs.c mainloop.c script.c \
@@ -167,7 +170,16 @@ endif
 endif
 
 if OPENCONNECT_SYSTEM_KEYS
+if OPENCONNECT_WIN32
+#install list-system-keys on Windows
+bin_PROGRAMS += list-system-keys
+else
+#Don't install list-system-keys
+#It breaks the Linux COPR builds since the native RPM doesn't list it as
+#one of the files that are expected to be installed.
+#see commit 17887b53e171ec7223340092dc042298a29defb0
 noinst_PROGRAMS += list-system-keys
+endif #OPENCONNECT_WIN32
 list_system_keys_SOURCES = list-system-keys.c
 list_system_keys_CFLAGS = $(GNUTLS_CFLAGS)
 list_system_keys_LDADD = $(GNUTLS_LIBS)
@@ -388,13 +400,18 @@ DISTCLEANFILES += .*.dll.d .*.exe.d file-list*.txt instfiles.nsh uninstfiles.nsh
 #EXTRA_DLLDIRS := $(OPENCONNECT_GUI_DIR)/bin $(libdir)/qt5/plugins/platforms $(libdir)/qt5/plugins/styles
 
 DLL_EXECUTABLES := openconnect$(EXEEXT) $(EXTRA_EXECUTABLES)
+endif
 
 if OPENCONNECT_WINTUN
 WINTUN_DLL = .libs/wintun.dll
 DISTCLEANFILES += $(WINTUN_DLL)
+sbin_SCRIPTS += .libs/wintun.dll
+if BUILD_NSIS
 DLL_EXECUTABLES += wintun.dll
 endif
+endif
 
+if BUILD_NSIS
 if OPENCONNECT_SYSTEM_KEYS
 DLL_EXECUTABLES += list-system-keys$(EXEEXT)
 endif
@@ -417,6 +434,7 @@ export V AM_DEFAULT_VERBOSITY bindir libdir OBJDUMP DLL_EXECUTABLES EXTRA_DLLDIR
 
 file-list.txt: Makefile.dlldeps openconnect$(EXEEXT) libopenconnect.la $(WINTUN_DLL)
 	@$(MAKE) --no-print-directory -f $< $@
+endif #BUILD_NSIS
 
 # Wintun Layer 3 TUN driver for Windows 7 and newer
 # (see https://wintun.net)
@@ -437,6 +455,7 @@ vpnc-script-win.js:
 	    jq -r '.[0] | "// This script matches the version found at " + (.web_url | sub("/commit/"; "/blob/")) + "/vpnc-script-win.js\n// Updated on " + .authored_date[:10] + " by " + .author_name + " <" + .author_email + "> (\"" + .title + "\")\n//"' > $@
 	curl https://gitlab.com/openconnect/vpnc-scripts/raw/master/vpnc-script-win.js >> $@
 
+if BUILD_NSIS
 # Let make find the file in VPATH
 file-list-%.txt: %
 	echo $< > $@
@@ -484,4 +503,4 @@ openconnect-installer-$(INSTALLER_SUFFIX).exe: openconnect.nsi instfiles.nsh uni
 	$(AM_V_MAKENSIS) $(MAKENSIS) $<
 	ln -f "$$(grep -E '^OutFile' openconnect.nsi | cut -f2 -d\")" $@
 
-endif
+endif #BUILD_NSIS
