--- openconnect-v9.12/Makefile.am.orig	2024-12-28 13:11:53.382917200 +0100
+++ openconnect-v9.12/Makefile.am	2024-12-28 13:12:46.208521800 +0100
@@ -403,9 +403,6 @@
 endif
 
 if OPENCONNECT_WINTUN
-WINTUN_DLL = .libs/wintun.dll
-DISTCLEANFILES += $(WINTUN_DLL)
-sbin_SCRIPTS += .libs/wintun.dll
 if BUILD_NSIS
 DLL_EXECUTABLES += wintun.dll
 endif
@@ -432,7 +429,7 @@
 
 export V AM_DEFAULT_VERBOSITY bindir libdir OBJDUMP DLL_EXECUTABLES EXTRA_DLLDIRS
 
-file-list.txt: Makefile.dlldeps openconnect$(EXEEXT) libopenconnect.la $(WINTUN_DLL)
+file-list.txt: Makefile.dlldeps openconnect$(EXEEXT) libopenconnect.la
 	@$(MAKE) --no-print-directory -f $< $@
 endif #BUILD_NSIS
 
@@ -441,20 +438,6 @@
 WINTUNDRIVER = wintun-0.14.1.zip
 WINTUNSHA256 = 07c256185d6ee3652e09fa55c0b673e2624b565e02c4b9091c79ca7d2f24ef51
 
-$(WINTUNDRIVER):
-	curl https://www.wintun.net/builds/$(WINTUNDRIVER) -o $@
-
-.libs/wintun.dll: $(WINTUNDRIVER)
-	echo $(WINTUNSHA256) $< | sha256sum -c
-	unzip -DD -o -j -d .libs $< wintun/bin/$(WINTUN_ARCH)/wintun.dll
-
-# Latest vpnc-script-win.js, annotated with a header documenting the
-# exact source revision.
-vpnc-script-win.js:
-	curl 'https://gitlab.com/api/v4/projects/openconnect%2Fvpnc-scripts/repository/commits?path=vpnc-script-win.js&branch=master' | \
-	    jq -r '.[0] | "// This script matches the version found at " + (.web_url | sub("/commit/"; "/blob/")) + "/vpnc-script-win.js\n// Updated on " + .authored_date[:10] + " by " + .author_name + " <" + .author_email + "> (\"" + .title + "\")\n//"' > $@
-	curl https://gitlab.com/openconnect/vpnc-scripts/raw/master/vpnc-script-win.js >> $@
-
 if BUILD_NSIS
 # Let make find the file in VPATH
 file-list-%.txt: %
