diff --git a/src/mesa/compat/c11/threads_win32.h b/src/mesa/compat/c11/threads_win32.h
index 9f00d444..be25122b 100644
--- a/src/mesa/compat/c11/threads_win32.h
+++ b/src/mesa/compat/c11/threads_win32.h
@@ -175,7 +175,7 @@ static int impl_tss_dtor_register(tss_t key, tss_dtor_t dtor)
     return 0;
 }
 
-static void impl_tss_dtor_invoke()
+static void impl_tss_dtor_invoke(void)
 {
     int i;
     for (i = 0; i < EMULATED_THREADS_TSS_DTOR_SLOTNUM; i++) {
