diff --git a/jaraco/envs.py b/jaraco/envs.py
index 50eae62..0058f11 100644
--- a/jaraco/envs.py
+++ b/jaraco/envs.py
@@ -34,7 +34,7 @@ class VirtualEnv:
         subprocess.check_call(cmd, env=env)
 
     def exe(self, cmd='python'):
-        bin_or_scripts = 'Scripts' if platform.system() == 'Windows' else 'bin'
+        bin_or_scripts = 'Scripts' if 'MINGW' not in sys.version else 'bin'
         return os.path.join(self.dir, bin_or_scripts, cmd)
 
     def env_vars(self):
