diff --git a/libs/libutils/src/globalattributes.cpp b/libs/libutils/src/globalattributes.cpp
index f9a55529d..a791d5b42 100644
--- a/libs/libutils/src/globalattributes.cpp
+++ b/libs/libutils/src/globalattributes.cpp
@@ -50,7 +50,7 @@ const QString GlobalAttributes::DbModelBkpExt(".dbk");
 const QString GlobalAttributes::ObjMetadataExt(".omf");
 const QString GlobalAttributes::DirSeparator("/");
 const QString GlobalAttributes::ResourcesDir("res");
-const QString GlobalAttributes::ConfigurationsDir("conf");
+const QString GlobalAttributes::ConfigurationsDir("../etc/pgmodeler");
 const QString GlobalAttributes::DefaultConfsDir("defaults");
 const QString GlobalAttributes::ConfsBackupsDir("backups");
 const QString GlobalAttributes::SchemasDir("schemas");
@@ -59,9 +59,9 @@ const QString GlobalAttributes::XMLSchemaDir("xml");
 const QString GlobalAttributes::CatalogSchemasDir("catalog");
 const QString GlobalAttributes::DataDictSchemaDir("datadict");
 const QString GlobalAttributes::AlterSchemaDir("alter");
-const QString GlobalAttributes::LanguagesDir("lang");
-const QString GlobalAttributes::SamplesDir("samples");
-const QString GlobalAttributes::PluginsDir("plugins");
+const QString GlobalAttributes::LanguagesDir("../share/pgmodeler/lang");
+const QString GlobalAttributes::SamplesDir("../share/pgmodeler/samples");
+const QString GlobalAttributes::PluginsDir("../lib/pgmodeler/plugins");
 const QString GlobalAttributes::SchemaExt(".sch");
 const QString GlobalAttributes::ObjectDTDDir("dtd");
 const QString GlobalAttributes::ObjectDTDExt(".dtd");
@@ -276,7 +276,7 @@ void GlobalAttributes::setSearchPath(const QString &search_path)
 {
 	setConfigFilesPaths();
 
-	SchemasRootPath=GlobalAttributes::getPathFromEnv("PGMODELER_SCHEMAS_PATH", SCHEMASDIR, QString("%1/%2").arg(search_path, SchemasDir));
+	SchemasRootPath=GlobalAttributes::getPathFromEnv("PGMODELER_SCHEMAS_PATH", SCHEMASDIR, QString("%1/../share/pgmodeler/%2").arg(search_path, SchemasDir));
 	LanguagesPath=GlobalAttributes::getPathFromEnv("PGMODELER_LANG_PATH", LANGDIR, QString("%1/%2").arg(search_path, LanguagesDir));
 	SamplesPath=getPathFromEnv("PGMODELER_SAMPLES_PATH", SAMPLESDIR, QString("%1/%2").arg(search_path, SamplesDir));
 	TmplConfigurationPath=getPathFromEnv("PGMODELER_TMPL_CONF_PATH", CONFDIR, QString("%1/%2").arg(search_path, ConfigurationsDir));
