--- pgmodeler-1.1.0.orig/libs/libutils/src/globalattributes.cpp	2024-02-23 17:25:50.495975900 +0300
+++ pgmodeler-1.1.0/libs/libutils/src/globalattributes.cpp	2024-02-23 21:01:40.699745400 +0300
@@ -66,7 +66,7 @@
 const QString GlobalAttributes::ObjMetadataExt(".omf");
 const QString GlobalAttributes::DirSeparator("/");
 const QString GlobalAttributes::ResourcesDir("res");
-const QString GlobalAttributes::ConfigurationsDir("conf");
+const QString GlobalAttributes::ConfigurationsDir("../etc/pgmodeler");
 const QString GlobalAttributes::DefaultConfsDir("defaults");
 const QString GlobalAttributes::ConfsBackupsDir("backups");
 const QString GlobalAttributes::SchemasDir("schemas");
@@ -75,9 +75,9 @@
 const QString GlobalAttributes::CatalogSchemasDir("catalog");
 const QString GlobalAttributes::DataDictSchemaDir("datadict");
 const QString GlobalAttributes::AlterSchemaDir("alter");
-const QString GlobalAttributes::LanguagesDir("lang");
-const QString GlobalAttributes::SamplesDir("samples");
-const QString GlobalAttributes::PluginsDir("plugins");
+const QString GlobalAttributes::LanguagesDir("../share/pgmodeler/lang");
+const QString GlobalAttributes::SamplesDir("../share/pgmodeler/samples");
+const QString GlobalAttributes::PluginsDir("../lib/pgmodeler/plugins");
 const QString GlobalAttributes::SchemaExt(".sch");
 const QString GlobalAttributes::ObjectDTDDir("dtd");
 const QString GlobalAttributes::ObjectDTDExt(".dtd");
@@ -322,7 +322,7 @@
 	setCustomPaths(search_path);
 	setConfigFilesPaths();
 
-	SchemasRootPath=getPathFromEnv(EnvSchemasPath, SCHEMASDIR, QString("%1/%2").arg(search_path, SchemasDir));
+	SchemasRootPath=getPathFromEnv(EnvSchemasPath, SCHEMASDIR, QString("%1/../share/pgmodeler/%2").arg(search_path, SchemasDir));
 	LanguagesPath=getPathFromEnv(EnvLangPath, LANGDIR, QString("%1/%2").arg(search_path, LanguagesDir));
 	SamplesPath=getPathFromEnv(EnvSamplesPath, SAMPLESDIR, QString("%1/%2").arg(search_path, SamplesDir));
 	TmplConfigurationPath=getPathFromEnv(EnvTmplConfPath, CONFDIR, QString("%1/%2").arg(search_path, ConfigurationsDir));
