From 950f752f6dbce900bb303907ebd1a6f5e774bb20 Mon Sep 17 00:00:00 2001
From: Alberto Fanjul <albertofanjul@gmail.com>
Date: Thu, 29 Apr 2021 03:27:07 +0200
Subject: [PATCH 1/2] Avoid xcursorgen

---
 cursors/meson.build | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/cursors/meson.build b/cursors/meson.build
index dfe462c3..c5bdc858 100644
--- a/cursors/meson.build
+++ b/cursors/meson.build
@@ -1,6 +1,5 @@
 cursor_dir = join_paths(icon_dir, 'cursors')
 rsvg = find_program('rsvg-convert')
-xcursorgen = find_program('xcursorgen')
 
 cursors = [
     'all-scroll',
@@ -124,21 +123,6 @@ cursors += [
     'watch'
 ]
 
-foreach cursor : cursors
-    message('generating: ' + cursor)
-
-    run_command(
-        xcursorgen,
-        cursor + '.in',
-        '-p', join_paths (meson.build_root(), 'cursors'),
-        join_paths (meson.build_root(), 'cursors', cursor)
-    )
-
-    install_data(
-        join_paths (meson.build_root(), 'cursors', cursor),
-        install_dir: cursor_dir
-    )
-endforeach
 
 links = []
 
-- 
2.30.1

