diff -urN libf2c.orig/makefile.u libf2c/makefile.u
--- libf2c.orig/makefile.u	2006-05-13 15:41:29.000000000 -0700
+++ libf2c/makefile.u	2006-05-13 15:42:04.000000000 -0700
@@ -69,7 +69,9 @@
 OFILES = $(MISC) $(POW) $(CX) $(DCX) $(REAL) $(DBL) $(INT) \
 	$(HALF) $(CMP) $(EFL) $(CHAR) $(I77) $(TIME)
 
-all: f2c.h signal1.h sysdep1.h libf2c.a
+all: libf2c.a
+
+$(OFILES): f2c.h signal1.h sysdep1.h
 
 libf2c.a: $(OFILES)
 	ar r libf2c.a $?
