--- origsrc/makefile.u	2011-01-13 18:58:35.313822600 -0600
+++ src/makefile.u	2011-01-13 19:02:29.539030100 -0600
@@ -90,6 +90,9 @@ libf2c.a: $(OFILES)
 libf2c.so: $(OFILES)
 	$(CC) -shared -o libf2c.so $(OFILES)
 
+libf2c.dll.a: $(OFILES)
+	$(CC) -shared -Wl,--out-implib,libf2c.dll.a -o cygf2c-2.dll $(OFILES)
+
 ### If your system lacks ranlib, you don't need it; see README.
 
 f77vers.o: f77vers.c
@@ -187,8 +190,8 @@ xwsne.o:	fmt.h
 arith.h: arithchk.c
 	$(CC) $(CFLAGS) -DNO_FPINIT arithchk.c -lm ||\
 	 $(CC) -DNO_LONG_LONG $(CFLAGS) -DNO_FPINIT arithchk.c -lm
-	./a.out >arith.h
-	rm -f a.out arithchk.o
+	./a.exe >arith.h
+	rm -f a.exe arithchk.o
 
 check:
 	xsum Notice README abort_.c arithchk.c backspac.c c_abs.c c_cos.c \
