--- at-spi2-core-2.48.0/meson.build.orig	2023-03-25 15:44:40.638657500 +0100
+++ at-spi2-core-2.48.0/meson.build	2023-03-19 15:23:15.000000000 +0100
@@ -62,7 +62,7 @@
 # Symbol visibility
 if get_option('default_library') != 'static'
   if host_system == 'windows'
-    atspi_conf.set('DLL_EXPORT', true)
+    at_spi_conf.set('DLL_EXPORT', true)
     at_spi_conf.set('_ATK_EXTERN', '__declspec(dllexport) extern')
     if cc.get_id() != 'msvc'
       test_cflags += ['-fvisibility=hidden']
