--- groonga-9.0.3/vendor/mruby/Makefile.am.orig	2019-06-21 10:56:35.794483100 +0300
+++ groonga-9.0.3/vendor/mruby/Makefile.am	2019-06-21 10:56:55.374510500 +0300
@@ -8,6 +8,7 @@
 	mruby-dir/src/Win/dirent.c
 
 DEFAULT_INCLUDES =						\
+	-I$(top_builddir)			\
 	-I$(srcdir)/../mruby-source/include			\
 	-I$(srcdir)/../mruby-source/src				\
 	-I$(srcdir)/../mruby-source/mrbgems/mruby-compiler/core	\
