diff -bur groonga-15.0.4-o/vendor/mruby-include/mruby/throw.h groonga-15.0.4/vendor/mruby-include/mruby/throw.h
--- groonga-15.0.4-o/vendor/mruby-include/mruby/throw.h	2025-04-18 02:46:03.161275800 -0600
+++ groonga-15.0.4/vendor/mruby-include/mruby/throw.h	2025-04-18 02:46:12.701092800 -0600
@@ -33,7 +33,7 @@
 #if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #define MRB_SETJMP _setjmp
 #define MRB_LONGJMP _longjmp
-#elif defined(__MINGW64__) && defined(__GNUC__) && __GNUC__ >= 4
+#elif defined(__MINGW64__) && defined(__GNUC__) && __GNUC__ >= 4 && !defined(_M_ARM64)
 #define MRB_SETJMP __builtin_setjmp
 #define MRB_LONGJMP __builtin_longjmp
 #else
diff -bur groonga-15.0.4-o/vendor/mruby-source/include/throw.h groonga-15.0.4/vendor/mruby-source/include/mruby/throw.h
--- groonga-15.0.4-o/vendor/mruby-source/include/mruby/throw.h	2025-04-18 02:46:03.161275800 -0600
+++ groonga-15.0.4/vendor/mruby-source/include/mruby/throw.h	2025-04-18 02:46:12.701092800 -0600
@@ -33,7 +33,7 @@
 #if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #define MRB_SETJMP _setjmp
 #define MRB_LONGJMP _longjmp
-#elif defined(__MINGW64__) && defined(__GNUC__) && __GNUC__ >= 4
+#elif defined(__MINGW64__) && defined(__GNUC__) && __GNUC__ >= 4 && !defined(_M_ARM64)
 #define MRB_SETJMP __builtin_setjmp
 #define MRB_LONGJMP __builtin_longjmp
 #elsediff -bur groonga-15.0.4-o/lib/windows.c groonga-15.0.4/lib/windows.c
--- a/lib/windows.c	2025-04-18 02:59:44.604000500 -0600
+++ b/lib/windows.c	2025-04-18 03:03:42.141936100 -0600
@@ -403,6 +401,11 @@
   frame.AddrPC.Offset = context->Rip;
   frame.AddrFrame.Offset = context->Rbp;
   frame.AddrStack.Offset = context->Rsp;
+# elif defined(_M_ARM64)
+  machine_type = IMAGE_FILE_MACHINE_ARM64;
+  frame.AddrPC.Offset = context->Pc;
+  frame.AddrFrame.Offset = context->Fp;
+  frame.AddrStack.Offset = context->Sp;
 # else /* _M_IX86 */
 #  error "Intel x86, Intel Itanium and x64 are only supported architectures"
 # endif /* _M_IX86 */
