diff -Naur nodeeditor-2.1.3.orig/include/nodes/internal/QStringStdHash.hpp nodeeditor-2.1.3/include/nodes/internal/QStringStdHash.hpp
--- nodeeditor-2.1.3.orig/include/nodes/internal/QStringStdHash.hpp	2019-05-01 13:11:24.000000000 +0200
+++ nodeeditor-2.1.3/include/nodes/internal/QStringStdHash.hpp	2021-08-17 22:16:37.038928200 +0200
@@ -5,15 +5,3 @@
 #include <QtCore/QString>
 #include <QtCore/QVariant>
 
-namespace std
-{
-template<>
-struct hash<QString>
-{
-  inline std::size_t
-  operator()(QString const &s) const
-  {
-    return qHash(s);
-  }
-};
-}
