From 528dc19da11d3b35e61a81b142430252db1db30b Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Fri, 3 Apr 2015 10:53:09 +0800
Subject: [PATCH] Add DESTDIR in configure

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 Makefile         | 4 ++--
 build/install.sh | 4 ++--
 configure        | 8 +++++++-
 3 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index c8bd7a4..0ee0046 100644
--- a/Makefile
+++ b/Makefile
@@ -23,8 +23,8 @@ install:

 .PHONY: install-META
 install-META: camlp4/META
-	mkdir -p ${PKGDIR}/camlp4/
-	cp -f camlp4/META ${PKGDIR}/camlp4/
+	mkdir -p ${DESTDIR}${PKGDIR}/camlp4/
+	cp -f camlp4/META ${DESTDIR}${PKGDIR}/camlp4/

 camlp4/META: camlp4/META.in
 	sed -e s/@@VERSION@@/${version}/g $? > $@
diff --git a/build/install.sh b/build/install.sh
index 9d98d6c..220e878 100644
--- a/build/install.sh
+++ b/build/install.sh
@@ -26,8 +26,8 @@ SAVED_LIBDIR="${LIBDIR}"

 . ./config.sh

-BINDIR="${SAVED_BINDIR:-${BINDIR}}"
-LIBDIR="${SAVED_LIBDIR:-${LIBDIR}}"
+BINDIR="$DESTDIR${SAVED_BINDIR:-${BINDIR}}"
+LIBDIR="$DESTDIR${SAVED_LIBDIR:-${LIBDIR}}"

 not_installed=$PWD/_build/not_installed

diff --git a/configure b/configure
index e859501..551f46e 100644
--- a/configure
+++ b/configure
@@ -17,8 +17,11 @@ for i in "$@"; do
     --pkgdir=*)
       PKGDIR=${i##*=}
       ;;
+    --destdir=*)
+      DESTDIR=${i##*=}
+      ;;
     *)
-      echo "usage: ./configure [--bindir=<dir>] [--libdir=<dir>] [--pkgdir=<dir>]" 1>&2
+      echo "usage: ./configure [--bindir=<dir>] [--libdir=<dir>] [--pkgdir=<dir>] [--destdir=<dir>]" 1>&2
       exit 2
       ;;
   esac
@@ -44,6 +47,7 @@ OCAMLC="`which ocamlc`"
 LIBDIR="${LIBDIR:-$standard_library}"
 BINDIR="${BINDIR:-`dirname $OCAMLC`}"
 PKGDIR="${PKGDIR:-$standard_library}"
+DESTDIR="${DESTDIR}"

 if [ -x "`which ocamlopt`" ]; then
     OB_FLAGS=""
@@ -58,6 +62,7 @@ EXE="$ext_exe"
 LIBDIR="$LIBDIR"
 BINDIR="$BINDIR"
 PKGDIR="$PKGDIR"
+DESTDIR="$DESTDIR"
 OB_FLAGS=$OB_FLAGS
 EOF

@@ -65,4 +70,5 @@ cat >> myocamlbuild_config.ml <<EOF
 let libdir="$LIBDIR"
 let bindir="$BINDIR"
 let pkgdir="$PKGDIR"
+let destdir="$DESTDIR"
 EOF
--
2.3.4
