From 3c6a0990d9937551e0750ccb6b65adf00aeca091 Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Thu, 9 Apr 2015 07:52:23 +0800
Subject: [PATCH] Guess the library path on msys2

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 camlp4/config/Camlp4_config.ml | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/camlp4/config/Camlp4_config.ml b/camlp4/config/Camlp4_config.ml
index 88e14d6..4e7f245 100644
--- a/camlp4/config/Camlp4_config.ml
+++ b/camlp4/config/Camlp4_config.ml
@@ -17,12 +17,27 @@
  * - Nicolas Pouillard: refactoring
  *)
 
-let ocaml_standard_library = Camlp4_import.standard_library;;
+let canonical_path_name p =
+  let current = Sys.getcwd () in
+  try
+    Sys.chdir p;
+    let p' = Sys.getcwd () in
+    Sys.chdir current;
+    p'
+  with Sys_error _ ->
+    (* We give up to find a canonical name and just return it... *)
+    p
+
+let camlp4bin = Filename.dirname (canonical_path_name Sys.executable_name)
+
+let camlp4root = Filename.dirname camlp4bin
+
+let ocaml_standard_library = List.fold_left (fun base dir -> Filename.concat base dir) camlp4root ["lib";"ocaml"];;
 
 let camlp4_standard_library =
   try Sys.getenv "CAMLP4LIB"
   with Not_found ->
-    Camlp4_import.camlp4_standard_library;;
+    Filename.concat ocaml_standard_library "camlp4";;
 
 let version = Sys.ocaml_version;;
 let program_name = ref "camlp4";;
-- 
2.3.4

