--- libdca-svn-r91/libdca/Makefile.am.orig	2013-07-31 06:50:29.308739400 +0000
+++ libdca-svn-r91/libdca/Makefile.am	2013-07-31 06:55:38.104451500 +0000
@@ -14,7 +14,7 @@
 EXTRA_DIST = configure.incl
 
 install-exec-hook:
-	ln -sf "$(DESTDIR)$(libdir)/libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
+	ln -sf "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
 
 uninstall-hook:
 	unlink "$(DESTDIR)$(libdir)/libdts.a"
