--- wineditline-2.205/src/CMakeLists.txt.orig	2018-04-29 18:12:25.635939900 -0400
+++ wineditline-2.205/src/CMakeLists.txt	2018-04-29 18:13:45.431152000 -0400
@@ -14,11 +14,11 @@
 target_link_libraries(edit_test edit)
 add_executable(edit_test_dll libedit_test_dll.c)
 install (TARGETS edit edit_test edit_test_dll
-  DESTINATION "${CMAKE_SOURCE_DIR}/bin${LIB_SUFFIX}")
+  RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 install (TARGETS edit_static
-  DESTINATION "${CMAKE_SOURCE_DIR}/lib${LIB_SUFFIX}")
+  DESTINATION lib)
 install (FILES editline/readline.h
-  DESTINATION "${CMAKE_SOURCE_DIR}/include/editline")
+  DESTINATION "include/editline")
 string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
 if (MSVC AND uppercase_CMAKE_BUILD_TYPE MATCHES "DEBUG")
   install (FILES ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/edit_static.dir/edit_static.pdb
