--- wineditline-2.101/src/CMakeLists.txt.orig	2014-12-01 22:42:00.102800000 +0300
+++ wineditline-2.101/src/CMakeLists.txt	2014-12-01 22:42:09.494000000 +0300
@@ -2,7 +2,7 @@
 include_directories("${PROJECT_BINARY_DIR}")
 include(GenerateExportHeader)
 add_library(edit_static STATIC editline.c fn_complete.c history.c)
-add_library(edit SHARED editline.c fn_complete.c history.c libedit.def)
+add_library(edit SHARED editline.c fn_complete.c history.c)
 set_target_properties(edit PROPERTIES PREFIX "")
 GENERATE_EXPORT_HEADER(edit
   BASE_NAME edit
