--- wineditline-2.205/src/editline/readline.h	2019-07-03 09:09:00.000000000 +0200
+++ wineditline-2.205/src/editline/readline.h	2019-11-30 16:56:47.213858800 +0100
@@ -46,6 +46,10 @@
 */
 #define DEFAULT_HISTORY_SIZE    200  /* default number of history entries */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef char **rl_completion_func_t(const char *, int, int);
 typedef char *rl_compentry_func_t(const char *, int);
 typedef void rl_compentryfree_func_t(void *);
