--- a/winpr/include/winpr/spec.h
+++ b/winpr/include/winpr/spec.h
@@ -956,22 +956,4 @@
 
 #endif
 
-#if defined(_WIN32) || defined(__CYGWIN__)
-#ifdef __GNUC__
-#define DECLSPEC_EXPORT __attribute__((dllexport))
-#define DECLSPEC_IMPORT __attribute__((dllimport))
-#else
-#define DECLSPEC_EXPORT __declspec(dllexport)
-#define DECLSPEC_IMPORT __declspec(dllimport)
-#endif
-#else
-#if defined(__GNUC__) && __GNUC__ >= 4
-#define DECLSPEC_EXPORT __attribute__((visibility("default")))
-#define DECLSPEC_IMPORT
-#else
-#define DECLSPEC_EXPORT
-#define DECLSPEC_IMPORT
-#endif
-#endif
-
 #endif /* WINPR_SPEC_H */
