--- oiio-Release-2.1.18.1.orig/src/cmake/compiler.cmake
+++ oiio-Release-2.1.18.1/src/cmake/compiler.cmake
@@ -83,12 +83,6 @@
 option (STOP_ON_WARNING "Stop building if there are any compiler warnings" OFF)
 if (NOT MSVC)
     add_compile_options ("-Wall")
-    if (STOP_ON_WARNING OR DEFINED ENV{CI})
-        add_compile_options ("-Werror")
-        # N.B. Force CI builds (Travis defines $CI) to use -Werror, even if
-        # STOP_ON_WARNING has been switched off by default, which we may do
-        # in release branches.
-    endif ()
 endif ()
 
 
