--- oiio-Release-2.1.18.1.orig/src/cmake/compiler.cmake
+++ oiio-Release-2.1.18.1/src/cmake/compiler.cmake
@@ -83,12 +83,6 @@
     if (EXTRA_WARNINGS)
         add_compile_options ("-Wextra")
     endif ()
-    if (STOP_ON_WARNING OR DEFINED ENV{CI})
-        add_compile_options ("-Werror")
-        # N.B. Force CI builds (Travis defines $CI) to use -Werror, even if
-        # STOP_ON_WARNING has been switched off by default, which we may do
-        # in release branches.
-    endif ()
 endif ()
 
 
