--- a/src/field3d.imageio/CMakeLists.txt
+++ b/src/field3d.imageio/CMakeLists.txt
@@ -11,6 +11,6 @@
     add_oiio_plugin (field3dinput.cpp field3doutput.cpp
                      INCLUDE_DIRS ${FIELD3D_INCLUDES}
                      LINK_LIBRARIES ${FIELD3D_LIBRARIES}
-                                    # ${HDF5_LIBRARIES}
+                                    ${HDF5_LIBRARIES}
                                     ${SZIP_LIBRARY})
 endif()
