--- oiio-Release-2.2.16.0/src/cmake/Config.cmake.in.orig	2021-07-16 10:06:12.379000000 +0200
+++ oiio-Release-2.2.16.0/src/cmake/Config.cmake.in	2021-07-16 10:06:28.886090100 +0200
@@ -2,6 +2,15 @@
 
 include(CMakeFindDependencyMacro)
 
+# Compute the installation prefix relative to this file.
+get_filename_component(_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_INSTALL_PREFIX "${_INSTALL_PREFIX}" PATH)
+get_filename_component(_INSTALL_PREFIX "${_INSTALL_PREFIX}" PATH)
+get_filename_component(_INSTALL_PREFIX "${_INSTALL_PREFIX}" PATH)
+if(_INSTALL_PREFIX STREQUAL "/")
+  set(_INSTALL_PREFIX "")
+endif()
+
 # add here all the find_dependency() whenever switching to config based dependencies
 if (@OpenEXR_VERSION@ VERSION_GREATER_EQUAL 3.0)
     find_dependency(Imath @Imath_VERSION@
