--- oiio-Release-1.4.8/src/gif.imageio/gifinput.cpp.orig	2014-05-23 20:21:53.000000000 +0400
+++ oiio-Release-1.4.8/src/gif.imageio/gifinput.cpp	2014-05-29 09:37:42.109200000 +0400
@@ -437,7 +437,7 @@
 GIFInput::close (void)
 {
     if (m_gif_file) {
-        if (DGifCloseFile (m_gif_file) == GIF_ERROR) {
+        if (DGifCloseFile (m_gif_file, NULL) == GIF_ERROR) {
             error ("Error trying to close the file.");
             return false;
         }
