Only in C++fix: out.so
diff -u -r C++/uvcview.rc C++fix/uvcview.rc
--- C++/uvcview.rc	2015-04-23 08:52:46.193018500 -0700
+++ C++fix/uvcview.rc	2015-04-23 08:53:32.858881400 -0700
@@ -84,7 +84,7 @@
     BEGIN
         MENUITEM "&Refresh\tF5",        ID_REFRESH
         MENUITEM SEPARATOR
-        MENUITEM "Save Current &View ..."        ID_SAVE
+        MENUITEM "Save Current &View ...",  ID_SAVE
         MENUITEM "Save As (&txt) ...",      ID_SAVEALL
         MENUITEM "Save As (&xml) ...\tF2",  ID_SAVEXML
         MENUITEM SEPARATOR
@@ -130,7 +130,7 @@
 BEGIN
     IDS_STANDARD_FONT           "Courier"
     IDS_STANDARD_FONT_HEIGHT    "\13"
-    IDS_STANDARD_FONT_WIDTH     "\8"
+    IDS_STANDARD_FONT_WIDTH     "\08"
 END
 
 STRINGTABLE DISCARDABLE
