--- a/usbdesc.h
+++ b/usbdesc.h
@@ -81,7 +81,6 @@ Revision History:
 #define USB_OTHER_SPEED_CONFIGURATION_DESCRIPTOR_TYPE  0x07
 #define USB_INTERFACE_POWER_DESCRIPTOR_TYPE            0x08
 #define USB_OTG_DESCRIPTOR_TYPE                        0x09
-#define USB_DEBUG_DESCRIPTOR_TYPE                      0x0A
 #define USB_IAD_DESCRIPTOR_TYPE                        0x0B
 
 //
--- a/uvcdesc.h
+++ b/uvcdesc.h
@@ -14,8 +14,6 @@
 #define ___UVCDESC_H___
 
 
-// USB Video Device Class Code
-#define USB_DEVICE_CLASS_VIDEO     0x0E 
 
 // Video sub-classes
 #define SUBCLASS_UNDEFINED              0x00
