--- postgresql-13.1/src/backend/storage/ipc/latch.c.orig	2021-07-17 14:09:21.786258600 -0700
+++ postgresql-13.1/src/backend/storage/ipc/latch.c	2021-07-17 14:10:01.364376500 -0700
@@ -677,7 +677,7 @@
 	 * pending signals are serviced.
 	 */
 	set->handles[0] = pgwin32_signal_event;
-	StaticAssertStmt(WSA_INVALID_EVENT == NULL, "");
+	/*StaticAssertStmt(WSA_INVALID_EVENT == NULL, "");*/
 #endif
 
 	return set;
