--- xalan_c-1.12/cmake/XalanICU.cmake.orig	2020-10-20 14:49:06.663689100 +0300
+++ xalan_c-1.12/cmake/XalanICU.cmake	2020-10-20 14:49:56.182697400 +0300
@@ -19,4 +19,4 @@
 
 # Determine if ICU is available
 
-find_package(ICU COMPONENTS uc data i18n)
+find_package(ICU COMPONENTS uc data in)
--- xalan_c-1.12/src/xalanc/CMakeLists.txt.orig	2020-10-20 15:01:19.432528900 +0300
+++ xalan_c-1.12/src/xalanc/CMakeLists.txt	2020-10-20 15:00:48.446070900 +0300
@@ -999,7 +999,7 @@
   $<INSTALL_INTERFACE:include/xalanc>)
 if(transcoder STREQUAL "icu")
   target_compile_definitions(xalan-c PRIVATE XALAN_USE_ICU=1)
-  target_link_libraries(xalan-c ICU::uc ICU::i18n ICU::data)
+  target_link_libraries(xalan-c ICU::uc ICU::in ICU::data)
 endif()
 target_link_libraries(xalan-c XercesC::XercesC Threads::Threads)
 if(msgloader STREQUAL "inmemory")
--- xalan_c-1.12/src/xalanc/Utils/MsgCreator/CMakeLists.txt.orig	2020-10-20 15:15:20.021966400 +0300
+++ xalan_c-1.12/src/xalanc/Utils/MsgCreator/CMakeLists.txt	2020-10-20 15:14:31.635110200 +0300
@@ -43,7 +43,7 @@
   $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/src>)
 target_link_libraries(MsgCreator XercesC::XercesC)
 if(transcoder STREQUAL "icu")
-  target_link_libraries(MsgCreator ICU::uc ICU::i18n)
+  target_link_libraries(MsgCreator ICU::uc ICU::in)
 endif()
 
 set_target_properties(MsgCreator PROPERTIES FOLDER "Message Library")
