diff -bur CalcMySky-0.3.1-orig/ShowMySky/CMakeLists.txt CalcMySky-0.3.1/ShowMySky/CMakeLists.txt
--- CalcMySky-0.3.1-orig/ShowMySky/CMakeLists.txt	2023-10-03 12:55:58.401386500 -0600
+++ CalcMySky-0.3.1/ShowMySky/CMakeLists.txt	2023-10-03 12:56:49.974489900 -0600
@@ -58,7 +58,7 @@
 install(TARGETS ShowMySky
         EXPORT ShowMySky-Qt${QT_VERSION}Config
         LIBRARY DESTINATION "${installLibDir}"
-        RUNTIME DESTINATION "${installLibDir}"
+        RUNTIME DESTINATION "${installBinDir}"
         ARCHIVE DESTINATION "${installLibDir}"
         INCLUDES DESTINATION "${installIncDir}"
         )
