diff -bur CalcMySky-0.3.1-orig/ShowMySky/api/ShowMySky/AtmosphereRenderer.hpp CalcMySky-0.3.1/ShowMySky/api/ShowMySky/AtmosphereRenderer.hpp
--- CalcMySky-0.3.1-orig/ShowMySky/api/ShowMySky/AtmosphereRenderer.hpp	2023-10-03 16:27:38.436405700 -0600
+++ CalcMySky-0.3.1/ShowMySky/api/ShowMySky/AtmosphereRenderer.hpp	2023-10-03 16:27:49.348575900 -0600
@@ -306,7 +306,7 @@
  *
  * This is the name that should be passed to dlopen-like functions. Use this macro for portability instead of hard-coding a string.
  */
-#define SHOWMYSKY_LIB_NAME (QT_VERSION_MAJOR==5 ? "ShowMySky-Qt5" : "ShowMySky-Qt6")
+#define SHOWMYSKY_LIB_NAME (QT_VERSION_MAJOR==5 ? "libShowMySky-Qt5" : "libShowMySky-Qt6")
 }
 
 #endif
