--- uriparser-0.9.2.orig/tool/uriparse.c	2019-04-23 16:34:08.307296786 -0400
+++ uriparser-0.9.2/tool/uriparse.c	2019-04-23 16:41:30.277003453 -0400
@@ -43,7 +43,7 @@
 #ifdef _WIN32
 # include <winsock2.h>
 # include <ws2tcpip.h>
-# ifdef __MINGW32__
+# if (_WIN32_WINNT < 0x600)
 WINSOCK_API_LINKAGE const char WSAAPI inet_ntop(
 		int af, const void *src, char *dst, socklen_t size);
 # endif
--- uriparser-0.9.3/CMakeLists.txt.orig	2019-08-05 22:14:42.251048700 +0300
+++ uriparser-0.9.3/CMakeLists.txt	2019-08-05 22:20:08.513709900 +0300
@@ -152,7 +152,10 @@
     set_property(TARGET uriparser PROPERTY VERSION ${URIPARSER_SO_CURRENT_MINUS_AGE}.${URIPARSER_SO_AGE}.${URIPARSER_SO_REVISION})
     set_property(TARGET uriparser PROPERTY SOVERSION ${URIPARSER_SO_CURRENT_MINUS_AGE})
     if(WIN32)
-        set_property(TARGET uriparser PROPERTY SUFFIX "-${URIPARSER_SO_CURRENT_MINUS_AGE}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+        set_target_properties(uriparser PROPERTIES
+                              OUTPUT_NAME uriparser
+                              RUNTIME_OUTPUT_NAME uriparser-${URIPARSER_SO_CURRENT_MINUS_AGE}
+                              ARCHIVE_OUTPUT_NAME uriparser)
     endif()
 endif()
 
