--- libdca-r91/src/Makefile.am.orig	2012-04-15 00:06:21 +0400
+++ libdca-r91/src/Makefile.am	2012-04-15 00:09:50 +0400
@@ -24,7 +24,7 @@
 	for a in dcadec extract_dca; do \
 		oldname=`echo "$$a" | sed -e 's/ca/ts/' | sed '$(transform)'` ; \
 		name=`echo "$$a" | sed '$(transform)'` ; \
-		ln -sf "$(man1dir)/$${name}.1" "$(DESTDIR)$(man1dir)/$${oldname}.1" ; \
+		ln -sf "$(DESTDIR)$(man1dir)/$${name}.1" "$(DESTDIR)$(man1dir)/$${oldname}.1" ; \
 	done
 
 uninstall-hook:
--- libdca-r91/libdca/Makefile.am.orig	2012-04-15 00:06:21 +0400
+++ libdca-r91/libdca/Makefile.am	2012-04-15 00:10:19 +0400
@@ -14,7 +14,7 @@
 EXTRA_DIST = configure.incl
 
 install-exec-hook:
-	ln -sf "libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
+	ln -sf "$(DESTDIR)$(libdir)/libdca.a" "$(DESTDIR)$(libdir)/libdts.a"
 
 uninstall-hook:
 	unlink "$(DESTDIR)$(libdir)/libdts.a"
