diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bb01a0d..7738867 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -86,9 +86,9 @@ set_target_properties (${PROJECT_NAME} PROPERTIES
   PUBLIC_HEADER "${PROJECT_NAME}.h")
 if (BUILD_FRAMEWORK)
   set_target_properties (${PROJECT_NAME} PROPERTIES FRAMEWORK TRUE)
-elseif (NOT WIN32)
+else()
   set (TARGET_PCS ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc)
-  configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${TARGET_PCS})
+  configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${TARGET_PCS} @ONLY)
   install (FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 endif ()
 
@@ -110,9 +110,9 @@ if (WITH_LSR_BINDINGS)
     PUBLIC_HEADER "${LSR}.h")
   if (BUILD_FRAMEWORK)
     set_target_properties (${LSR} PROPERTIES FRAMEWORK TRUE)
-  elseif (NOT WIN32)
+  else()
     set (TARGET_PCS "${TARGET_PCS} ${CMAKE_CURRENT_BINARY_DIR}/${LSR}.pc")
-    configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${LSR}.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${LSR}.pc)
+    configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${LSR}.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${LSR}.pc @ONLY)
     install (FILES ${CMAKE_CURRENT_BINARY_DIR}/${LSR}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
   endif ()
 endif ()
diff --git a/src/soxr-lsr.pc.in b/src/soxr-lsr.pc.in
index 7b75757..cd13f2e 100644
--- a/src/soxr-lsr.pc.in
+++ b/src/soxr-lsr.pc.in
@@ -1,5 +1,10 @@
-Name: ${LSR}
-Description: ${DESCRIPTION_SUMMARY} (with libsamplerate-like bindings)
-Version: ${PROJECT_VERSION}
-Libs: -L${LIB_INSTALL_DIR} -l${LSR}
-Cflags: -I${INCLUDE_INSTALL_DIR}
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: @LSR@
+Description: @DESCRIPTION_SUMMARY@ (with libsamplerate-like bindings)
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -l@LSR@
+Cflags: -I${includedir}
diff --git a/src/soxr.pc.in b/src/soxr.pc.in
index 69d225b..579cde7 100644
--- a/src/soxr.pc.in
+++ b/src/soxr.pc.in
@@ -1,5 +1,10 @@
-Name: ${PROJECT_NAME}
-Description: ${DESCRIPTION_SUMMARY}
-Version: ${PROJECT_VERSION}
-Libs: -L${LIB_INSTALL_DIR} -l${PROJECT_NAME}
-Cflags: -I${INCLUDE_INSTALL_DIR}
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: @PROJECT_NAME@
+Description: @DESCRIPTION_SUMMARY@
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -l@PROJECT_NAME@
+Cflags: -I${includedir}
