--- code/CMakeLists.txt	2017-10-31 16:05:50.905208900 -0400
+++ code/CMakeLists.txt	2017-10-31 16:05:55.852473400 -0400
@@ -188,11 +188,7 @@
   set (INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include")
 endif ()
 if (NOT DEFINED DOC_INSTALL_DIR)
-  if (UNIX)
     set (DOC_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/doc/lib${PROJECT_NAME}")
-  else ()
-    set (DOC_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/doc")
-  endif ()
 endif ()
 
 if (APPLE)
