--- src/lread.c.orig	2019-09-08 15:22:32.286341300 +0000
+++ src/lread.c	2019-09-08 15:27:03.331768100 +0000
@@ -4659,7 +4659,12 @@
         } /* Vinstallation_directory != Vsource_directory */
 
     } /* if Vinstallation_directory */
-
+  else
+    {
+      Vsource_directory
+        = Fexpand_file_name (build_string ("../"),
+                             Fcar (decode_env_path (0, PATH_DATA, 0)));
+    }
   return lpath;
 }
 
