--- gimp-3.0.4/meson.build.orig	2025-05-18 00:51:03.000000000 +0200
+++ gimp-3.0.4/meson.build	2025-05-18 11:35:03.183040900 +0200
@@ -338,7 +338,7 @@
 
 # Generate native .pdb (CodeView) debug symbols (for DIA or DbgHelp debuggers and LLDB)
 pdb_support = cc.has_argument('-gcodeview') and cc.has_link_argument('-Wl,--pdb=')
-if platform_windows and pdb_support and cc.get_id() == 'clang'
+if platform_windows and pdb_support and cc.get_id() == 'clang' and false
   compiler_args += '-gcodeview'
   linker_args += '-Wl,--pdb='
 endif
