--- gimp-3.2.0/libgimp/meson.build.orig	2026-03-14 10:47:44.000000000 +0100
+++ gimp-3.2.0/libgimp/meson.build	2026-03-15 09:22:26.999601800 +0100
@@ -627,7 +627,7 @@
   endif
 endif
 
-if not platform_osx and host_cpu_family != 'x86'
+if not platform_osx and host_cpu_family != 'x86' and false
   # Verify .def files for Windows linking.
   # We check this on non-Windows platform (Debian) on CI, and on Windows itself.
   custom_target('check-def-files',
