--- ./app/core/gimpparamspecs-duplicate.c	2016-07-04 10:33:22.449542700 +0200
+++ ./app/core/gimpparamspecs-duplicate.c	2016-07-04 10:34:54.505304400 +0200
@@ -67,7 +67,7 @@
                                      spec->default_value,
                                      pspec->flags);
 
-          if (GEGL_IS_PARAM_SPEC_MULTILINE (pspec))
+          if (gegl_param_spec_get_property_key(pspec, "multiline"))
             {
               g_param_spec_set_qdata (new, multiline_quark,
                                       GINT_TO_POINTER (TRUE));
