--- qt-everywhere-opensource-src-4.8.7/configure.orig	2015-05-07 17:14:56.000000000 +0300
+++ qt-everywhere-opensource-src-4.8.7/configure	2015-06-03 15:18:52.103556600 +0300
@@ -48,9 +48,19 @@
 relconf=`basename $0`
 # the directory of this script is the "source tree"
 relpath=`dirname $0`
-relpath=`(cd "$relpath"; /bin/pwd)`
+if [ -n "$MSYSTEM" ]; then
+  relpath=$(cygpath -ma "$relpath")
+  relpath=${relpath%/}
+else
+  relpath=`(cd "$relpath"; /bin/pwd)`
+fi
 # the current directory is the "build tree" or "object tree"
-outpath=`/bin/pwd`
+if [ -n "$MSYSTEM" ]; then
+  outpath=$(cygpath -ma $(/bin/pwd))
+  outpath=${outpath%/}
+else
+  outpath=`/bin/pwd`
+fi
 
 #license file location
 LICENSE_FILE="$QT_LICENSE_FILE"
@@ -1376,7 +1386,11 @@
         CFG_PSQL_CONFIG="$VAL"
         ;;
     prefix)
-        QT_INSTALL_PREFIX="$VAL"
+        if [ -n "$MSYSTEM" ]; then
+          QT_INSTALL_PREFIX=$(cygpath -ma "$VAL")
+        else
+          QT_INSTALL_PREFIX="$VAL"
+        fi
         ;;
     hostprefix)
 	QT_HOST_PREFIX="$VAL"
@@ -2966,6 +2776,12 @@
             - Also available for Reliant UNIX: reliant-cds-64
         "
         ;;
+     MSYS*:* | MINGW*:*)
+        PLATFORM=win32-g++
+        if [ -z "$XPLATFORM" ]; then
+            XPLATFORM="win32-g++"
+        fi
+        ;;
      CYGWIN*:*)
         PLATFORM=cygwin-g++
         ;;
@@ -3754,6 +3774,8 @@
     if [ -z "$QT_INSTALL_SETTINGS" ]; then #default
         if [ "$PLATFORM_MAC" = "yes" ]; then
 	    QT_INSTALL_SETTINGS=/Library/Preferences/Qt
+        elif [ "$XPLATFORM_MINGW" = "yes" ]; then
+	    QT_INSTALL_SETTINGS="$QT_INSTALL_PREFIX/etc"
         else
 	    QT_INSTALL_SETTINGS=/etc/xdg
         fi
@@ -4943,7 +4965,15 @@
 
     #mkspecs/default is used as a (gasp!) default mkspec so QMAKESPEC needn't be set once configured
     rm -rf mkspecs/default
-    ln -s `echo $XQMAKESPEC | sed "s,^${relpath}/mkspecs/,,"` mkspecs/default
+    if [ "$XPLATFORM_MINGW" = "yes" ]; then
+      mkdir -p mkspecs/default
+      _spec=$(echo $XQMAKESPEC | sed "s,^${relpath}/mkspecs/,,")
+      echo "QMAKESPEC_ORIGINAL=$XQMAKESPEC" > mkspecs/default/qmake.conf
+      echo "include(../${_spec}/qmake.conf)" >> mkspecs/default/qmake.conf
+      echo "#include \"../${_spec}/qplatformdefs.conf\"" > mkspecs/default/qplatformdefs.h
+    else
+      ln -s `echo $XQMAKESPEC | sed "s,^${relpath}/mkspecs/,,"` mkspecs/default
+    fi
     # fix makefiles
     for mkfile in GNUmakefile Makefile; do
         EXTRA_LFLAGS=
@@ -4953,7 +4985,11 @@
 #           if which qmake >/dev/null 2>&1 && [ -f qmake/qmake.pro ]; then
 #               (cd qmake && qmake) >/dev/null 2>&1 && continue
 #           fi
-            in_mkfile="${mkfile}.unix"
+            if [ "$PLATFORM" = "win32-g++" ]; then
+              in_mkfile="${mkfile}.win32-g++-sh"
+            else
+              in_mkfile="${mkfile}.unix"
+            fi
         fi
         in_mkfile="$relpath/qmake/$in_mkfile"
         mkfile="$outpath/qmake/$mkfile"
