diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/CMakeLists.txt onnxruntime-1.24.4/cmake/CMakeLists.txt
--- onnxruntime-1.24.4-pristine/cmake/CMakeLists.txt	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/CMakeLists.txt	2026-04-28 20:53:08.128217200 +0200
@@ -561,7 +561,7 @@
 include(external/onnxruntime_external_deps.cmake)
 
 set(ORT_WARNING_FLAGS)
-if (WIN32)
+if (MSVC)
     # class needs to have dll-interface to be used by clients
     list(APPEND ORT_WARNING_FLAGS "/wd4251")
     # issued by thrust nonstandard extension used: nameless struct/union
@@ -1081,7 +1081,9 @@
       target_compile_definitions(${target_name} PRIVATE USE_KLEIDIAI)
     endif()
 
+    if(NOT MINGW)
     set_target_properties(${target_name} PROPERTIES COMPILE_WARNING_AS_ERROR ON)
+    endif()
     if (onnxruntime_USE_CUDA)
       # Suppress a "conversion_function_not_usable" warning in gsl/span
       target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xcudafe \"--diag_suppress=conversion_function_not_usable\">")
@@ -1214,6 +1216,9 @@
 
 function(onnxruntime_add_shared_library target_name)
   add_library(${target_name} SHARED ${ARGN})
+  if(MINGW)
+    set_target_properties(${target_name} PROPERTIES PREFIX "")
+  endif()
   onnxruntime_configure_target(${target_name})
   if(WIN32)
         target_compile_definitions(${target_name} PRIVATE VER_MAJOR=${VERSION_MAJOR_PART})
@@ -1236,6 +1241,9 @@
   else()
     #On Windows, this target shouldn't generate an import lib, but I don't know how to disable it.
     add_library(${target_name} MODULE ${ARGN})
+    if(MINGW)
+        set_target_properties(${target_name} PROPERTIES PREFIX "")
+    endif()
     if(WIN32)
         target_compile_definitions(${target_name} PRIVATE VER_MAJOR=${VERSION_MAJOR_PART})
         target_compile_definitions(${target_name} PRIVATE VER_MINOR=${VERSION_MINOR_PART})
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/adjust_global_compile_flags.cmake onnxruntime-1.24.4/cmake/adjust_global_compile_flags.cmake
--- onnxruntime-1.24.4-pristine/cmake/adjust_global_compile_flags.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/adjust_global_compile_flags.cmake	2026-04-28 20:53:05.723558400 +0200
@@ -217,6 +217,10 @@
 endmacro()
 
 #Set global compile flags for all the source code(including third_party code like protobuf)
+if(MINGW)
+  enable_language(ASM_MASM)
+  set(CMAKE_ASM_MASM_FLAGS "${CMAKE_ASM_MASM_FLAGS} -I${CMAKE_CURRENT_SOURCE_DIR}/../onnxruntime/core/mlas/lib/amd64")
+endif()
 #This section must be before any add_subdirectory, otherwise build may fail because /MD,/MT mismatch
 if (MSVC)
   if (onnxruntime_target_platform STREQUAL "ARM64" OR
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/detect_onnxruntime_target_platform.cmake onnxruntime-1.24.4/cmake/detect_onnxruntime_target_platform.cmake
--- onnxruntime-1.24.4-pristine/cmake/detect_onnxruntime_target_platform.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/detect_onnxruntime_target_platform.cmake	2026-04-28 20:53:06.266379700 +0200
@@ -4,7 +4,7 @@
 
 unset(onnxruntime_target_platform)
 
-if (MSVC)
+if (MSVC OR MINGW)
   if (CMAKE_VS_PLATFORM_NAME)
     # Multi-platform generator
     set(onnxruntime_target_platform ${CMAKE_VS_PLATFORM_NAME})
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/onnxruntime.cmake onnxruntime-1.24.4/cmake/onnxruntime.cmake
--- onnxruntime-1.24.4-pristine/cmake/onnxruntime.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/onnxruntime.cmake	2026-04-28 20:53:06.904214800 +0200
@@ -82,7 +82,6 @@
-      ${SYMBOL_FILE}
       "${ONNXRUNTIME_ROOT}/core/dll/dllmain.cc"
       "${ONNXRUNTIME_ROOT}/core/dll/delay_load_hook.cc"
-      "${ONNXRUNTIME_ROOT}/core/dll/onnxruntime.rc"
+      
     )
   elseif(onnxruntime_BUILD_APPLE_FRAMEWORK)
     # apple framework requires the header file be part of the library
@@ -153,7 +152,9 @@
       target_link_options(onnxruntime PRIVATE  "LINKER:--version-script=${SYMBOL_FILE}" "LINKER:--no-undefined" "LINKER:--gc-sections" "LINKER:-z,noexecstack")
     endif()
   else()
+    if(MSVC)
     target_link_options(onnxruntime PRIVATE  "-DEF:${SYMBOL_FILE}")
+    endif()
   endif()
 
 
@@ -319,7 +321,7 @@
         FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 
-if (WIN32 AND NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
+if (MSVC AND NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
   # Workaround STL bug https://github.com/microsoft/STL/issues/434#issuecomment-921321254
   # Note that the workaround makes std::system_error crash before Windows 10
 
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/onnxruntime_common.cmake onnxruntime-1.24.4/cmake/onnxruntime_common.cmake
--- onnxruntime-1.24.4-pristine/cmake/onnxruntime_common.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/onnxruntime_common.cmake	2026-04-28 20:53:06.599443400 +0200
@@ -79,7 +79,7 @@
 endif()
 
 # platform-specific device discovery files
-if (WIN32)
+if (MSVC)
     list(APPEND onnxruntime_common_src_patterns
          "${ONNXRUNTIME_ROOT}/core/platform/windows/device_discovery.cc")
 elseif (LINUX)
@@ -112,6 +112,10 @@
 file(GLOB onnxruntime_common_src CONFIGURE_DEPENDS
     ${onnxruntime_common_src_patterns}
     )
+if(WIN32 AND NOT MSVC)
+    list(FILTER onnxruntime_common_src EXCLUDE REGEX "windows/telemetry")
+    list(FILTER onnxruntime_common_src EXCLUDE REGEX "windows/logging/")
+endif()
 
 # Remove new/delete intercept. To deal with memory leaks
 # Use either non-mimalloc build OR use mimalloc built-in features.
@@ -124,7 +128,7 @@
 source_group(TREE ${REPO_ROOT} FILES ${onnxruntime_common_src})
 
 onnxruntime_add_static_library(onnxruntime_common ${onnxruntime_common_src})
-if(WIN32)
+if(MSVC)
   if("cxx_std_23" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
     set_property(TARGET onnxruntime_common PROPERTY CXX_STANDARD 23)
     target_compile_options(onnxruntime_common PRIVATE "/Zc:char8_t-")
@@ -174,6 +178,9 @@
 
 
 target_link_libraries(onnxruntime_common PUBLIC safeint_interface ${GSL_TARGET} ${ABSEIL_LIBS} date::date)
+if(MINGW)
+  target_link_libraries(onnxruntime_common PRIVATE pathcch)
+endif()
 
 add_dependencies(onnxruntime_common ${onnxruntime_EXTERNAL_DEPENDENCIES})
 
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/onnxruntime_graph.cmake onnxruntime-1.24.4/cmake/onnxruntime_graph.cmake
--- onnxruntime-1.24.4-pristine/cmake/onnxruntime_graph.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/onnxruntime_graph.cmake	2026-04-28 20:52:21.086393300 +0200
@@ -136,7 +136,7 @@
   target_compile_definitions(onnxruntime_graph PRIVATE BUILD_MS_EXPERIMENTAL_OPS=1)
 endif()
 
-if (WIN32)
+if (MSVC)
   set(onnxruntime_graph_static_library_flags
       -IGNORE:4221 # LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation that consumes this library
   )
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/onnxruntime_mlas.cmake onnxruntime-1.24.4/cmake/onnxruntime_mlas.cmake
--- onnxruntime-1.24.4-pristine/cmake/onnxruntime_mlas.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/onnxruntime_mlas.cmake	2026-04-28 20:53:06.145412900 +0200
@@ -75,6 +75,19 @@
 
 #TODO: set MASM flags properly
 function(setup_mlas_source_for_windows)
+  if(MINGW)
+    set_source_files_properties(${MLAS_SRC_DIR}/qgemm_kernel_sse41.cpp PROPERTIES COMPILE_FLAGS "-msse4.1")
+    set_source_files_properties(${MLAS_SRC_DIR}/qgemm_kernel_amx.cpp PROPERTIES COMPILE_FLAGS "-mavx512f -mavx512bw -mavx512dq -mavx512vl -mamx-int8 -mamx-tile")
+    set_source_files_properties(${MLAS_SRC_DIR}/intrinsics/avx512/quantize_avx512f.cpp PROPERTIES COMPILE_FLAGS "-mavx512f")
+    set_source_files_properties(${MLAS_SRC_DIR}/sqnbitgemm_kernel_avx512.cpp PROPERTIES COMPILE_FLAGS "-mavx512f -mavx512bw -mavx512dq -mavx512vl -mavx512vnni -mfma")
+    set_source_files_properties(${MLAS_SRC_DIR}/sqnbitgemm_kernel_avx512vnni.cpp PROPERTIES COMPILE_FLAGS "-mavx512f -mavx512bw -mavx512dq -mavx512vl -mavx512vnni -mfma")
+    set_source_files_properties(${MLAS_SRC_DIR}/sqnbitgemm_lut_kernel_avx2.cpp PROPERTIES COMPILE_FLAGS "-mavx2 -mfma")
+    set_source_files_properties(${MLAS_SRC_DIR}/sqnbitgemm_kernel_avx2.cpp PROPERTIES COMPILE_FLAGS "-mavx2 -mfma -mavxvnni -mf16c")
+    set_source_files_properties(${MLAS_SRC_DIR}/rotary_embedding_kernel_avx2.cpp PROPERTIES COMPILE_FLAGS "-mavx2 -mfma -mf16c")
+    set_source_files_properties(${MLAS_SRC_DIR}/qgemm_kernel_avx2.cpp PROPERTIES COMPILE_FLAGS "-mavx2 -mfma")
+    set_source_files_properties(${MLAS_SRC_DIR}/qgemm_kernel_sse.cpp PROPERTIES COMPILE_FLAGS "-msse4.1")
+    set_source_files_properties(${MLAS_SRC_DIR}/q4gemm_avx512.cpp PROPERTIES COMPILE_FLAGS "-mavx512f -mavx512bw -mavx512dq -mavx512vl -mavx512vnni -mfma")
+  endif()
 
   #
   # Sources common for all platforms.
@@ -193,12 +206,20 @@
     file(GLOB_RECURSE mlas_platform_srcs_avx CONFIGURE_DEPENDS
       "${MLAS_SRC_DIR}/intrinsics/avx/*.cpp"
     )
+    if(MSVC)
     set_source_files_properties(${mlas_platform_srcs_avx} PROPERTIES COMPILE_FLAGS "/arch:AVX")
+    elseif(MINGW)
+    set_source_files_properties(${mlas_platform_srcs_avx} PROPERTIES COMPILE_FLAGS "-mavx")
+    endif()
 
     file(GLOB_RECURSE mlas_platform_srcs_avx2 CONFIGURE_DEPENDS
       "${MLAS_SRC_DIR}/intrinsics/avx2/*.cpp"
     )
+    if(MSVC)
     set_source_files_properties(${mlas_platform_srcs_avx2} PROPERTIES COMPILE_FLAGS "/arch:AVX2")
+    elseif(MINGW)
+    set_source_files_properties(${mlas_platform_srcs_avx2} PROPERTIES COMPILE_FLAGS "-mavx2 -mfma")
+    endif()
 
     target_sources(onnxruntime_mlas PRIVATE
       ${MLAS_SRC_DIR}/dgemm.cpp
@@ -261,7 +282,6 @@
 
     if(MSVC_VERSION GREATER_EQUAL 1933)
       target_sources(onnxruntime_mlas PRIVATE
-        ${MLAS_SRC_DIR}/amd64/cvtfp16Avx.asm
       )
     endif()
 
@@ -353,7 +373,7 @@
     )
   endif()
   target_sources(onnxruntime_mlas PRIVATE ${mlas_platform_srcs})
-elseif(MSVC)
+elseif(MSVC OR MINGW)
   setup_mlas_source_for_windows()
 else()
     if(APPLE)
@@ -869,7 +889,7 @@
     set_target_properties(${mlas_target} PROPERTIES FOLDER "ONNXRuntime")
 endforeach()
 
-if (WIN32)
+if (MSVC)
   target_compile_options(onnxruntime_mlas PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:/wd6385>" "$<$<COMPILE_LANGUAGE:CXX>:/wd4127>")
   if (onnxruntime_ENABLE_STATIC_ANALYSIS)
     target_compile_options(onnxruntime_mlas PRIVATE  "$<$<COMPILE_LANGUAGE:CXX>:/analyze:stacksize 131072>")
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/onnxruntime_providers_cpu.cmake onnxruntime-1.24.4/cmake/onnxruntime_providers_cpu.cmake
--- onnxruntime-1.24.4-pristine/cmake/onnxruntime_providers_cpu.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/onnxruntime_providers_cpu.cmake	2026-04-28 20:52:22.202741200 +0200
@@ -227,7 +227,7 @@
   )
 
   source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_shared_cc_srcs})
-  onnxruntime_add_shared_library(onnxruntime_providers_shared ${onnxruntime_providers_shared_cc_srcs} "${ONNXRUNTIME_ROOT}/core/dll/onnxruntime.rc")
+  onnxruntime_add_shared_library(onnxruntime_providers_shared ${onnxruntime_providers_shared_cc_srcs} )
   set_target_properties(onnxruntime_providers_shared PROPERTIES FOLDER "ONNXRuntime")
   set_target_properties(onnxruntime_providers_shared PROPERTIES LINKER_LANGUAGE CXX)
   #ON AIX, to call dlopen on onnxruntime_providers_shared, we need to generate this library as shared object .so file.
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/onnxruntime_unittests.cmake onnxruntime-1.24.4/cmake/onnxruntime_unittests.cmake
--- onnxruntime-1.24.4-pristine/cmake/onnxruntime_unittests.cmake	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/onnxruntime_unittests.cmake	2026-04-28 20:53:05.045981900 +0200
@@ -1335,6 +1335,9 @@
       target_link_libraries(onnx_test_runner PRIVATE Python::Python)
     endif()
     set_target_properties(onnx_test_runner PROPERTIES FOLDER "ONNXRuntimeTest")
+if(MINGW)
+  target_link_options(onnx_test_runner PRIVATE -municode)
+endif()
 
     install(TARGETS onnx_test_runner
             ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -1497,6 +1500,9 @@
       target_link_libraries(onnxruntime_perf_test PRIVATE onnx_test_runner_common absl::flags absl::flags_parse ${onnx_test_libs})
     endif()
     set_target_properties(onnxruntime_perf_test PROPERTIES FOLDER "ONNXRuntimeTest")
+if(MINGW)
+  target_link_options(onnxruntime_perf_test PRIVATE -municode)
+endif()
 
 endif()
 
@@ -1863,7 +1869,7 @@
     endif()
   else()
     set(ONNXRUNTIME_CUSTOM_OP_LIB_LINK_FLAG "-DEF:${TEST_SRC_DIR}/testdata/custom_op_library/custom_op_library.def")
-    if (NOT onnxruntime_USE_CUDA)
+    if (MSVC AND NOT onnxruntime_USE_CUDA)
       target_compile_options(custom_op_library PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options /wd26409>"
                     "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/wd26409>")
     endif()
diff -ruN '--exclude=*.orig' '--exclude=build-*' onnxruntime-1.24.4-pristine/cmake/tensorboard/compat/proto/CMakeLists.txt onnxruntime-1.24.4/cmake/tensorboard/compat/proto/CMakeLists.txt
--- onnxruntime-1.24.4-pristine/cmake/tensorboard/compat/proto/CMakeLists.txt	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/cmake/tensorboard/compat/proto/CMakeLists.txt	2026-04-28 20:52:21.205471600 +0200
@@ -20,7 +20,7 @@
 onnxruntime_add_include_to_target(tensorboard ${PROTOBUF_LIB})
 target_include_directories(tensorboard PRIVATE ${PROJECT_BINARY_DIR})
 add_dependencies(tensorboard ${onnxruntime_EXTERNAL_DEPENDENCIES})
-if(WIN32)
+if(MSVC)
   target_compile_options(tensorboard PRIVATE "/wd4100" "/wd4125" "/wd4127" "/wd4267" "/wd4456" "/wd4800" "/wd6011" "/wd6387" "/wd28182")
 endif()
 
