diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/cerr_sink.h onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/cerr_sink.h
--- onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/cerr_sink.h	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/cerr_sink.h	2026-04-30 15:49:03.914954500 +0200
@@ -12,7 +12,7 @@
 /// A std::cerr based ISink
 /// </summary>
 /// <seealso cref="ISink" />
-#ifdef _WIN32
+#ifdef _MSC_VER
 class CErrSink : public WOStreamSink {
  public:
   CErrSink() : WOStreamSink(std::wcerr, /*flush*/ false) {  // std::cerr isn't buffered so no flush required
diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/clog_sink.h onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/clog_sink.h
--- onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/clog_sink.h	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/clog_sink.h	2026-04-30 15:49:03.870067500 +0200
@@ -12,7 +12,7 @@
 /// A std::clog based ISink
 /// </summary>
 /// <seealso cref="ISink" />
-#ifdef _WIN32
+#ifdef _MSC_VER
 class CLogSink : public WOStreamSink {
  public:
   CLogSink() : WOStreamSink(std::wclog, /*flush*/ true) {
diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/file_sink.h onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/file_sink.h
--- onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/file_sink.h	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/file_sink.h	2026-04-30 15:49:03.320952900 +0200
@@ -4,6 +4,7 @@
 #pragma once
 
 #include <fstream>
+#include <filesystem>
 #include "core/common/logging/sinks/ostream_sink.h"
 
 namespace onnxruntime {
@@ -75,7 +76,7 @@
   /// <param name="filter_user_data">If set to <c>true</c> [removes user data].</param>
   /// <remarks>Filtering of user data can alternatively be done at the <see cref="LoggingManager" /> level.</remarks>
   FileSink(const std::wstring& filename, bool append, bool filter_user_data)
-      : FileSink{std::make_unique<std::wofstream>(filename, std::ios::out | (append ? std::ios::app : std::ios::trunc)),
+      : FileSink{std::make_unique<std::wofstream>(std::filesystem::path{filename}, std::ios::out | (append ? std::ios::app : std::ios::trunc)),
                  filter_user_data} {
   }
 
diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/ostream_sink.cc onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/ostream_sink.cc
--- onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/ostream_sink.cc	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/ostream_sink.cc	2026-04-30 15:49:03.822635000 +0200
@@ -21,7 +21,7 @@
 };
 #endif
 
-#ifndef _WIN32
+#if 1  // OStreamSink always available
 void OStreamSink::SendImpl(const Timestamp& timestamp, const std::string& logger_id, const Capture& message) {
   // operator for formatting of timestamp in ISO8601 format including microseconds
   using timestamp_ns::operator<<;
@@ -64,7 +64,8 @@
     stream_->flush();
   }
 }
-#else
+#endif
+#if defined(_WIN32)
 void WOStreamSink::SendImpl(const Timestamp& timestamp, const std::string& logger_id, const Capture& message) {
   // operator for formatting of timestamp in ISO8601 format including microseconds
   using date::operator<<;
diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/ostream_sink.h onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/ostream_sink.h
--- onnxruntime-1.24.4-pristine/onnxruntime/core/common/logging/sinks/ostream_sink.h	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/common/logging/sinks/ostream_sink.h	2026-04-30 15:49:03.706043500 +0200
@@ -12,7 +12,7 @@
 
 namespace onnxruntime {
 namespace logging {
-#ifndef _WIN32
+#if 1  // OStreamSink always available
 /// <summary>
 /// A std::ostream based ISink
 /// </summary>
@@ -30,7 +30,8 @@
   std::ostream* stream_;
   const bool flush_;
 };
-#else
+#endif
+#if defined(_WIN32)
 /// <summary>
 /// A std::wostream based ISink
 /// </summary>
diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/common/profiler.cc onnxruntime-1.24.4/onnxruntime/core/common/profiler.cc
--- onnxruntime-1.24.4-pristine/onnxruntime/core/common/profiler.cc	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/common/profiler.cc	2026-04-30 15:49:03.198040400 +0200
@@ -2,6 +2,7 @@
 // Licensed under the MIT License.
 
 #include "profiler.h"
+#include <filesystem>
 
 namespace onnxruntime {
 namespace profiling {
@@ -57,7 +58,7 @@
 void Profiler::StartProfiling(const std::basic_string<T>& file_name) {
   enabled_ = true;
 #if !defined(__wasm__)
-  profile_stream_.open(file_name, std::ios::out | std::ios::trunc);
+  profile_stream_.open(std::filesystem::path{file_name}, std::ios::out | std::ios::trunc);
 #endif
   profile_stream_file_ = ToUTF8String(file_name);
   profiling_start_time_ = std::chrono::high_resolution_clock::now();
diff -ruN '--exclude=build-*' '--exclude=*.orig' '--exclude=*-pristine' onnxruntime-1.24.4-pristine/onnxruntime/core/framework/allocation_planner.cc onnxruntime-1.24.4/onnxruntime/core/framework/allocation_planner.cc
--- onnxruntime-1.24.4-pristine/onnxruntime/core/framework/allocation_planner.cc	2026-03-16 23:52:22.000000000 +0100
+++ onnxruntime-1.24.4/onnxruntime/core/framework/allocation_planner.cc	2026-04-30 15:49:03.572719800 +0200
@@ -6,6 +6,7 @@
 #include <algorithm>
 #include <deque>
 #include <sstream>
+#include <filesystem>
 #include <ctime>
 #include <iomanip>
 #include <iterator>
@@ -2572,7 +2573,7 @@
   if (config_file_.empty()) {
     return;
   }
-  std::ifstream if_stream(config_file_);
+  std::ifstream if_stream{std::filesystem::path{config_file_}};
   if (if_stream.is_open()) {
     try {
       json json_config = json::parse(if_stream);
@@ -2621,7 +2622,7 @@
         json_config["devices"].insert(json_config["devices"].end(), std::to_string(device_type));
       }
     }
-    std::ofstream of_stream(config_file_);
+    std::ofstream of_stream{std::filesystem::path{config_file_}};
     if (of_stream.is_open()) {
       of_stream << json_config.dump();
       of_stream.close();
@@ -2638,7 +2639,7 @@
   IGraphPartitioner::GraphPartitioningStrategy partitioner_type =
       IGraphPartitioner::GraphPartitioningStrategy::DeviceBasedPartition;
   if (!config_file.empty()) {
-    std::ifstream f(config_file);
+    std::ifstream f{std::filesystem::path{config_file}};
     if (f.is_open()) {
       try {
         json json_config = json::parse(f);
