--- a/cmake/external/abseil-cpp.cmake
+++ b/cmake/external/abseil-cpp.cmake
@@ -39,7 +39,7 @@
     URL_HASH SHA1=${DEP_SHA1_abseil_cpp}
     EXCLUDE_FROM_ALL
     PATCH_COMMAND ${ABSL_PATCH_COMMAND}
-    FIND_PACKAGE_ARGS 20250814 NAMES absl
+    FIND_PACKAGE_ARGS NAMES absl REQUIRED
 )
 
 onnxruntime_fetchcontent_makeavailable(abseil_cpp)
--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -44,7 +44,7 @@
     URL ${DEP_URL_re2}
     URL_HASH SHA1=${DEP_SHA1_re2}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS NAMES re2
+    FIND_PACKAGE_ARGS NAMES re2 REQUIRED
 )
 onnxruntime_fetchcontent_makeavailable(re2)
 
@@ -72,7 +72,7 @@
     URL ${DEP_URL_googletest}
     URL_HASH SHA1=${DEP_SHA1_googletest}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS 1.14.0...<2.0.0 NAMES GTest
+    FIND_PACKAGE_ARGS 1.14.0...<2.0.0 NAMES GTest REQUIRED
   )
   FetchContent_MakeAvailable(googletest)
 endif()
@@ -88,7 +88,7 @@
     URL ${DEP_URL_google_benchmark}
     URL_HASH SHA1=${DEP_SHA1_google_benchmark}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS NAMES benchmark
+    FIND_PACKAGE_ARGS NAMES benchmark REQUIRED
   )
   onnxruntime_fetchcontent_makeavailable(google_benchmark)
 endif()
@@ -200,7 +200,7 @@
   URL_HASH SHA1=${DEP_SHA1_protobuf}
   PATCH_COMMAND ${ONNXRUNTIME_PROTOBUF_PATCH_COMMAND}
   EXCLUDE_FROM_ALL
-  FIND_PACKAGE_ARGS NAMES Protobuf protobuf
+  FIND_PACKAGE_ARGS NAMES Protobuf protobuf REQUIRED
 )
 
 set(protobuf_BUILD_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
@@ -277,7 +277,7 @@
   EXCLUDE_FROM_ALL
   PATCH_COMMAND
     ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/date/date.patch
-  FIND_PACKAGE_ARGS 3...<4 NAMES date
+  FIND_PACKAGE_ARGS 3...<4 NAMES date REQUIRED
 )
 onnxruntime_fetchcontent_makeavailable(date)
 
@@ -291,12 +291,11 @@
      mp11
      URL ${DEP_URL_mp11}
      EXCLUDE_FROM_ALL
-     FIND_PACKAGE_ARGS NAMES Boost
+     FIND_PACKAGE_ARGS NAMES Boost REQUIRED
     )
-    FetchContent_Populate(mp11)
+    onnxruntime_fetchcontent_makeavailable(mp11)
     if(NOT TARGET Boost::mp11)
-      add_library(Boost::mp11 IMPORTED INTERFACE)
-      target_include_directories(Boost::mp11 INTERFACE $<BUILD_INTERFACE:${mp11_SOURCE_DIR}/include>)
+      add_library(Boost::mp11 ALIAS Boost::headers)
     endif()
   endif()
 endif()
@@ -309,7 +308,7 @@
     URL ${DEP_URL_json}
     URL_HASH SHA1=${DEP_SHA1_json}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS 3.10 NAMES nlohmann_json
+    FIND_PACKAGE_ARGS 3.10 NAMES nlohmann_json REQUIRED
 )
 onnxruntime_fetchcontent_makeavailable(nlohmann_json)
 
@@ -412,7 +411,7 @@
   # containing microsoft/GSL#1213 (commit 543d0dd).
   PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/gsl/1213.patch
   EXCLUDE_FROM_ALL
-  FIND_PACKAGE_ARGS 4.0 NAMES Microsoft.GSL
+  FIND_PACKAGE_ARGS 4.0 NAMES Microsoft.GSL REQUIRED
 )
 set(GSL_TARGET "Microsoft.GSL::GSL")
 set(GSL_INCLUDE_DIR "$<TARGET_PROPERTY:${GSL_TARGET},INTERFACE_INCLUDE_DIRECTORIES>")
@@ -467,10 +466,14 @@
     URL_HASH SHA1=${DEP_SHA1_flatbuffers}
     PATCH_COMMAND ${ONNXRUNTIME_FLATBUFFERS_PATCH_COMMAND}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS 23.5.9 NAMES Flatbuffers flatbuffers
+    FIND_PACKAGE_ARGS 23.5.9 NAMES Flatbuffers flatbuffers REQUIRED
 )
 
 onnxruntime_fetchcontent_makeavailable(flatbuffers)
+if(TARGET flatbuffers::flatbuffers_shared AND NOT TARGET flatbuffers::flatbuffers)
+  message(STATUS "Aliasing flatbuffers::flatbuffers to flatbuffers::flatbuffers_shared")
+  add_library(flatbuffers::flatbuffers ALIAS flatbuffers::flatbuffers_shared)
+endif()
 if(NOT flatbuffers_FOUND)
   if(NOT TARGET flatbuffers::flatbuffers)
     add_library(flatbuffers::flatbuffers ALIAS flatbuffers)
@@ -532,7 +535,7 @@
   URL_HASH SHA1=${DEP_SHA1_onnx}
   PATCH_COMMAND ${ONNXRUNTIME_ONNX_PATCH_COMMAND}
   EXCLUDE_FROM_ALL
-  FIND_PACKAGE_ARGS NAMES ONNX onnx
+  FIND_PACKAGE_ARGS NAMES ONNX onnx REQUIRED
 )
 
 onnxruntime_fetchcontent_makeavailable(onnx)
@@ -545,8 +548,9 @@
   message(STATUS "Aliasing ONNX::onnx_proto to onnx_proto")
   add_library(onnx_proto ALIAS ONNX::onnx_proto)
 endif()
-if(onnxruntime_USE_VCPKG)
-  find_package(Eigen3 CONFIG REQUIRED)
+find_package(Eigen3 CONFIG REQUIRED)
+if(Eigen3_FOUND)
+  get_target_property(eigen_INCLUDE_DIRS Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
 else()
   include(external/eigen.cmake)
 endif()
@@ -627,7 +631,7 @@
     URL_HASH SHA1=${DEP_SHA1_cxxopts}
     PATCH_COMMAND ${ONNXRUNTIME_CXXOPTS_PATCH_COMMAND}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS NAMES cxxopts
+    FIND_PACKAGE_ARGS NAMES cxxopts REQUIRED
   )
   set(CXXOPTS_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
   set(CXXOPTS_BUILD_TESTS OFF CACHE BOOL "" FORCE)
--- a/cmake/external/pybind11.cmake
+++ b/cmake/external/pybind11.cmake
@@ -6,6 +6,6 @@
     URL ${DEP_URL_pybind11}
     URL_HASH SHA1=${DEP_SHA1_pybind11}
     EXCLUDE_FROM_ALL
-    FIND_PACKAGE_ARGS 3.0 NAMES pybind11
+    FIND_PACKAGE_ARGS 3.0 NAMES pybind11 REQUIRED
 )
 onnxruntime_fetchcontent_makeavailable(pybind11_project)
