--- tremor-r18223/Makefile.am.orig	2012-04-01 23:22:33 +0400
+++ tremor-r18223/Makefile.am	2012-04-02 00:12:13 +0400
@@ -17,7 +17,7 @@
                         registry.h window.h window_lookup.h\
                         codec_internal.h backends.h \
 			asm_arm.h ivorbiscodec.h
-libvorbisidec_la_LDFLAGS = -version-info @V_LIB_CURRENT@:@V_LIB_REVISION@:@V_LIB_AGE@
+libvorbisidec_la_LDFLAGS = -no-undefined -version-info @V_LIB_CURRENT@:@V_LIB_REVISION@:@V_LIB_AGE@
 libvorbisidec_la_LIBADD = @OGG_LIBS@
 
 EXTRA_PROGRAMS = ivorbisfile_example iseeking_example
