--- p11-kit-0.17.5/p11-kit/Makefile.am.orig	2014-07-04 14:48:45.000000000 +0100
+++ p11-kit-0.17.5/p11-kit/Makefile.am	2014-08-28 19:38:29.226555600 +0100
@@ -100,7 +100,6 @@
 
 # Proxy module is actually same as library, so install a link
 install-exec-hook:
-	$(LN_S) -f `readlink $(DESTDIR)$(libdir)/libp11-kit.so` $(DESTDIR)$(libdir)/p11-kit-proxy.so
 	$(MKDIR_P) $(DESTDIR)$(p11_package_config_modules)
 
 endif
