--- p11-kit-0.21.3/common/test-path.c.orig	2014-02-13 07:36:36.000000000 +0000
+++ p11-kit-0.21.3/common/test-path.c	2014-07-28 06:29:16.182509100 +0000
@@ -131,7 +131,7 @@
 	                    p11_path_expand ("~/.config/path"));
 #else /* OS_WIN32 */
 	putenv ("HOME=C:\\Users\\blah");
-	assert_str_eq_free ("C:\\Users\\blah\\path",
+	assert_str_eq_free ("C:\\Users\\blah\\my/path",
 	                    p11_path_expand ("~/my/path"));
 	assert_str_eq_free ("C:\\Users\\blah\\path",
 	                    p11_path_expand ("~\\path"));
