--- p11-kit-0.20.3/common/compat.c.orig	2014-07-04 14:48:45.000000000 +0100
+++ p11-kit-0.20.3/common/compat.c	2014-08-30 20:55:34.508248200 +0100
@@ -323,7 +323,7 @@
 		return NULL;
 	}
 
-	map->file  = CreateFile (path, GENERIC_READ, 0, NULL, OPEN_EXISTING, FILE_FLAG_RANDOM_ACCESS, NULL);
+	map->file  = CreateFile (path, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_FLAG_RANDOM_ACCESS, NULL);
 	if (map->file == INVALID_HANDLE_VALUE) {
 		errn = GetLastError ();
 		free (map);
