--- p11-kit-0.20.3/common/path.c.orig	2014-08-28 15:38:41.193607200 +0100
+++ p11-kit-0.20.3/common/path.c	2014-08-28 16:18:10.646132100 +0100
@@ -336,8 +336,12 @@
      strrchr (exe_path, '/')[1] = '\0';
   }
   char * rel_to_datadir = get_relative_path (P11_BINDIR, path);
+  p11_message ("get_relative_path (P11_BINDIR = %s, path = %s) = %s", P11_BINDIR, path, rel_to_datadir);
+  p11_message ("exe_path(1) = %s", exe_path);
   strcat (exe_path, rel_to_datadir);
+  p11_message ("exe_path(2) = %s", exe_path);
   simplify_path (&exe_path[0]);
+  p11_message ("exe_path(3) = %s", exe_path);
   return malloc_copy_string(exe_path);
 #else
   return malloc_copy_string(path);
