--- p11-kit-0.20.1/common/compat.c	2014-07-22 16:37:06.456094868 +0200
+++ p11-kit-0.20.1/common/compat.c	2014-07-22 16:42:16.440081230 +0200
@@ -833,7 +833,12 @@
             size_t buflen)
 {
 #ifdef OS_WIN32
-	return strerror_s (buf, buflen, errnum);
+        /* non-recursive implementation of strerror_r */
+        char *retbuf;
+        retbuf = strerror(errnum);
+        strncpy(buf, retbuf, buflen);
+        buf[buflen - 1] = '\0';
+        return 0;
 #else
 	#error no strerror_r implementation
 #endif
