diff -Naur p11-kit-0.23.16.1.orig/p11-kit/conf.c p11-kit-0.23.16.1/p11-kit/conf.c
--- p11-kit-0.23.16.1.orig/p11-kit/conf.c	2019-05-23 15:09:01.266282300 -0400
+++ p11-kit-0.23.16.1/p11-kit/conf.c	2019-05-23 15:17:23.528706900 -0400
@@ -232,9 +232,11 @@
 		if (getauxval (AT_SECURE)) {
 			p11_debug ("skipping user config in setuid or setgid program");
 			mode = CONF_USER_NONE;
+#ifndef _WIN32
 		} else if (getuid () == 0) {
 			p11_debug ("skipping user config in program running as root");
 			mode = CONF_USER_NONE;
+#endif
 		} else if (secure_getenv ("P11_KIT_NO_USER_CONFIG")) {
 			p11_debug ("skipping user config due to P11_NO_USER_CONFIG");
 			mode = CONF_USER_NONE;
