diff -aur p11-kit-0.23.18/p11-kit/Makefile.am p11-kit-0.23.18.1/p11-kit/Makefile.am
--- p11-kit-0.23.18/p11-kit/Makefile.am	2019-09-30 20:57:35.000000000 +0800
+++ p11-kit-0.23.18.1/p11-kit/Makefile.am	2020-01-07 12:00:40.029248200 +0800
@@ -88,7 +88,7 @@
 	-no-undefined \
 	-version-info $(P11KIT_LT_RELEASE)
 
-if HAVE_LD_VERSION_SCRIPT
+if !OS_WIN32
 libp11_kit_la_LDFLAGS += -Wl,--version-script=$(srcdir)/p11-kit/libp11-kit.map
 else
 libp11_kit_la_LDFLAGS += -export-symbols-regex '^C_GetFunctionList|^p11_kit_'
