diff -Naur opencolorio-1.1.1-orig/export/OpenColorIO/OpenColorABI.h.in opencolorio-1.1.1/export/OpenColorIO/OpenColorABI.h.in
--- opencolorio-1.1.1-orig/export/OpenColorIO/OpenColorABI.h.in	2019-03-28 08:12:57.000000000 +0300
+++ opencolorio-1.1.1/export/OpenColorIO/OpenColorABI.h.in	2019-07-31 21:59:45.504868700 +0300
@@ -73,15 +73,7 @@
 
 // If supported, define OCIOEXPORT, OCIOHIDDEN
 // (used to choose which symbols to export from OpenColorIO)
-#if defined __GNUC__
-    #if __GNUC__ >= 4
-        #define OCIOEXPORT __attribute__ ((visibility("default")))
-        #define OCIOHIDDEN __attribute__ ((visibility("hidden")))
-    #else
-        #define OCIOEXPORT
-        #define OCIOHIDDEN
-    #endif
-#elif defined(_WIN32) || defined(_WIN64) || defined(_WINDOWS) || defined(_MSC_VER)
+#if defined(_WIN32) || defined(_WIN64) || defined(_WINDOWS) || defined(_MSC_VER)
     // Windows requires you to export from the main library and then import in any others
 	#ifndef OpenColorIO_STATIC
 		#if defined OpenColorIO_EXPORTS
@@ -93,6 +85,14 @@
 		#define OCIOEXPORT
 	#endif
     #define OCIOHIDDEN
+#elif defined __GNUC__
+    #if __GNUC__ >= 4
+        #define OCIOEXPORT __attribute__ ((visibility("default")))
+        #define OCIOHIDDEN __attribute__ ((visibility("hidden")))
+    #else
+        #define OCIOEXPORT
+        #define OCIOHIDDEN
+    #endif
 #else // Others platforms not supported atm
     #define OCIOEXPORT
     #define OCIOHIDDEN
