--- opencolorio-1.1.1/src/pyglue/CMakeLists.txt.orig	2020-06-10 09:11:19.921303600 +0300
+++ opencolorio-1.1.1/src/pyglue/CMakeLists.txt	2020-06-10 09:20:37.253652200 +0300
@@ -69,8 +69,9 @@
 target_link_OCIO(PyOpenColorIO)
 
 if(WIN32)
-    # Python extension on Windows is pyd
-    set_target_properties(PyOpenColorIO PROPERTIES SUFFIX ".pyd")
+    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import distutils.sysconfig, sys; get = distutils.sysconfig.get_config_var; sys.stdout.write(get('EXT_SUFFIX') or get('SO') or '.pyd');"
+                    OUTPUT_VARIABLE PYTHON_MOD_EXT)
+    set_target_properties(PyOpenColorIO PROPERTIES SUFFIX "${PYTHON_MOD_EXT}")
 endif()
 
 if(OCIO_PYGLUE_LINK OR WIN32)
