From 9ed4d8deae06621729366b4d654e3e2248691301 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 9 Apr 2022 08:01:46 -0400
Subject: wxWidgets doxygen 1.9 fixes

---
 docs/doxygen/Doxyfile | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/docs/doxygen/Doxyfile b/docs/doxygen/Doxyfile
index 0a812be2eb..1f35fd0009 100644
--- a/docs/doxygen/Doxyfile
+++ b/docs/doxygen/Doxyfile
@@ -29,7 +29,6 @@ MULTILINE_CPP_IS_BRIEF = NO
 INHERIT_DOCS           = YES
 SEPARATE_MEMBER_PAGES  = NO
 TAB_SIZE               = 4
-TCL_SUBST              =
 OPTIMIZE_OUTPUT_FOR_C  = NO
 OPTIMIZE_OUTPUT_JAVA   = NO
 OPTIMIZE_FOR_FORTRAN   = NO
@@ -327,7 +326,6 @@ VERBATIM_HEADERS       = NO # Default: YES
 #---------------------------------------------------------------------------
 
 ALPHABETICAL_INDEX     = YES
-COLS_IN_ALPHA_INDEX    = 5
 IGNORE_PREFIX          = wx
 
 
@@ -440,7 +438,6 @@ PDF_HYPERLINKS         = YES
 USE_PDFLATEX           = YES
 LATEX_BATCHMODE        = YES # Default: NO
 LATEX_HIDE_INDICES     = NO
-LATEX_SOURCE_CODE      = NO
 LATEX_BIB_STYLE        = plain
 
 
@@ -482,7 +479,6 @@ XML_PROGRAMLISTING     = NO # Default: YES
 
 GENERATE_DOCBOOK       = NO
 DOCBOOK_OUTPUT         = docbook
-DOCBOOK_PROGRAMLISTING = NO
 
 
 #---------------------------------------------------------------------------
@@ -527,15 +523,12 @@ GENERATE_TAGFILE       =
 ALLEXTERNALS           = NO
 EXTERNAL_GROUPS        = YES
 EXTERNAL_PAGES         = YES
-PERL_PATH              = /usr/bin/perl
 
 
 #---------------------------------------------------------------------------
 # dot Tool Options
 #---------------------------------------------------------------------------
 
-CLASS_DIAGRAMS         = YES
-MSCGEN_PATH            =
 DIA_PATH               =
 HIDE_UNDOC_RELATIONS   = YES
 HAVE_DOT               = YES # Default: NO
-- 
