--- fontconfig-2.15.0/src/meson.build.orig	2024-02-18 11:57:16.082497200 +0100
+++ fontconfig-2.15.0/src/meson.build	2024-02-18 11:56:40.872987500 +0100
@@ -57,8 +57,9 @@
 # Define FcPublic appropriately for exports on windows
 fc_extra_c_args = []
 
-if cc.get_argument_syntax() == 'msvc'
+if host_machine.system() == 'windows' and get_option('default_library') == 'shared'
   fc_extra_c_args += '-DFcPublic=__declspec(dllexport)'
+  fc_extra_c_args += '-DDLL_EXPORT'
 endif
 
 libfontconfig = library('fontconfig',
