--- fontconfig-2.15.0/conf.d/link_confs.py.orig	2024-02-18 12:02:06.238444000 +0100
+++ fontconfig-2.15.0/conf.d/link_confs.py	2024-02-18 11:20:14.335114100 +0100
@@ -26,6 +26,8 @@
     if not os.path.exists(confpath):
         os.makedirs(confpath)
 
+    args.availpath = str(PurePath(os.environ.get('DESTDIR'), *PurePath(args.availpath).parts[1:]))
+
     for link in args.links:
         src = os.path.join(args.availpath, link)
         dst = os.path.join(confpath, link)
@@ -33,6 +35,12 @@
             os.remove(dst)
         except FileNotFoundError:
             pass
+        
+        if os.name == "nt":
+            import shutil
+            shutil.copyfile(src, dst)
+            continue
+
         try:
             os.symlink(os.path.relpath(src, start=args.confpath), dst)
         except NotImplementedError:
