diff -bur sqlitebrowser-3.13.1-orig/CMakeLists.txt sqlitebrowser-3.13.1/CMakeLists.txt
--- sqlitebrowser-3.13.1-orig/CMakeLists.txt	2025-02-19 23:35:50 +0000
+++ sqlitebrowser-3.13.1/CMakeLists.txt	2025-02-19 23:36:29 +0000
@@ -417,7 +417,7 @@
         )
         target_sources(${PROJECT_NAME} PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/sqlbicon.o")
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-subsystem,windows")
-        set(WIN32_STATIC_LINK -Wl,-Bstatic -lssl -lcrypto -lws2_32)
+        set(WIN32_LINK -lssl -lcrypto -lws2_32)
         set(ADDITIONAL_LIBS lzma)
     else()
         target_sources(${PROJECT_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src/winapp.rc")
@@ -485,7 +485,7 @@
 target_link_libraries(${PROJECT_NAME}
     ${LPTHREAD}
     ${QT_LIBS}
-    ${WIN32_STATIC_LINK}
+    ${WIN32_LINK}
     ${ADDITIONAL_LIBS}
 )
 
