diff -Naur osgocean/include/osgOcean/Cylinder osgocean-patched/include/osgOcean/Cylinder
--- osgocean/include/osgOcean/Cylinder	2014-05-25 20:39:33.715800000 +0400
+++ osgocean-patched/include/osgOcean/Cylinder	2014-06-07 20:49:58.658400000 +0400
@@ -16,13 +16,14 @@
 */
 
 #pragma once
+#include <osgOcean/Export>
 #include <osg/Geometry>
 #include <osg/Math>
 
 namespace osgOcean
 {
 
-class Cylinder : public osg::Geometry
+class OSGOCEAN_EXPORT Cylinder : public osg::Geometry
 {
 protected:
 	osg::ref_ptr<osg::Vec4Array> _colorArray;
diff -Naur osgocean/include/osgOcean/MipmapGeometryVBO osgocean-patched/include/osgOcean/MipmapGeometryVBO
--- osgocean/include/osgOcean/MipmapGeometryVBO	2014-05-25 20:39:33.684600000 +0400
+++ osgocean-patched/include/osgOcean/MipmapGeometryVBO	2014-06-07 19:49:02.267400000 +0400
@@ -144,7 +144,7 @@
         * Rather than use standard computeBound() this computes the bounding box
         * using the offset and the worldSize.
         */
-        osg::BoundingBox computeBound( void ) const;
+        osg::BoundingBox computeBoundingBox( void ) const;
 
         /** 
         * Set the tile offset for shader positioning.
@@ -164,7 +164,7 @@
             }
             
             dirtyBound();
-            setBound( computeBound() );
+            setBound( computeBoundingBox() );
         }
 
         /** 
diff -Naur osgocean/src/osgOcean/GodRays.cpp osgocean-patched/src/osgOcean/GodRays.cpp
--- osgocean/src/osgOcean/GodRays.cpp	2014-05-25 20:39:34.199400000 +0400
+++ osgocean-patched/src/osgOcean/GodRays.cpp	2014-06-07 19:51:10.892400000 +0400
@@ -302,7 +302,7 @@
 
         // If the eye isn't contained withing the god ray volume, 
         // we need to recompute the bounds or they get clipped.
-        if(!getDrawable(0)->getBound().contains( eye )){
+        if(!getDrawable(0)->getBoundingBox().contains( eye )){
             getDrawable(0)->dirtyBound();
             getDrawable(1)->dirtyBound();
         }
diff -Naur osgocean/src/osgOcean/MipmapGeometryVBO.cpp osgocean-patched/src/osgOcean/MipmapGeometryVBO.cpp
--- osgocean/src/osgOcean/MipmapGeometryVBO.cpp	2014-05-25 20:39:34.199400000 +0400
+++ osgocean-patched/src/osgOcean/MipmapGeometryVBO.cpp	2014-06-07 19:52:04.940400000 +0400
@@ -83,7 +83,7 @@
     {
     }
 
-    osg::BoundingBox MipmapGeometryVBO::computeBound( void ) const 
+    osg::BoundingBox MipmapGeometryVBO::computeBoundingBox( void ) const
     {
         osg::BoundingBox bb;
         
--- osgocean/src/osgOcean/OceanScene.cpp.orig	2014-12-08 08:02:04.720600000 +0300
+++ osgocean/src/osgOcean/OceanScene.cpp	2014-12-08 08:02:29.867800000 +0300
@@ -881,7 +881,8 @@
 
                 bool surfaceVisible = _oceanSurface->isVisible(*cv, eyeAboveWater);
 
-                (*_oceanSurface->getCullCallback())(_oceanSurface.get(), &nv);
+                osg::Callback* callback = _oceanSurface->getCullCallback();
+                if (callback) callback->run(_oceanSurface.get(), &nv);
 
                 preRenderCull(*cv, eyeAboveWater, surfaceVisible);     // reflections/refractions
                 
