--- qbittorrent-4.2.5/cmake/Modules/winconf.cmake.orig	2020-04-27 11:00:06.269598100 +0300
+++ qbittorrent-4.2.5/cmake/Modules/winconf.cmake	2020-04-27 11:00:22.114465500 +0300
@@ -24,9 +24,9 @@
 #set(Boost_USE_STATIC_RUNTIME True)
 
 add_definitions(
-    -DNTDDI_VERSION=0x06010000
-    -D_WIN32_WINNT=0x0601
-    -D_WIN32_IE=0x0601
+    -DNTDDI_VERSION=0x06020000
+    -D_WIN32_WINNT=0x0602
+    -D_WIN32_IE=0x0602
     -DUNICODE
     -D_UNICODE
     -DWIN32
--- qbittorrent-4.2.5/winconf.pri.orig	2020-04-27 11:02:04.415004000 +0300
+++ qbittorrent-4.2.5/winconf.pri	2020-04-27 11:02:14.649455300 +0300
@@ -9,9 +9,9 @@
 # TODO: remove the following define as it is not used since libtorrent >= 1.2
 DEFINES += TORRENT_DISABLE_RESOLVE_COUNTRIES
 
-DEFINES += NTDDI_VERSION=0x06010000
-DEFINES += _WIN32_WINNT=0x0601
-DEFINES += _WIN32_IE=0x0601
+DEFINES += NTDDI_VERSION=0x06020000
+DEFINES += _WIN32_WINNT=0x0602
+DEFINES += _WIN32_IE=0x0602
 DEFINES += UNICODE
 DEFINES += _UNICODE
 DEFINES += WIN32
