--- qbittorrent-4.5.2/dist/CMakeLists.txt.orig	2023-04-19 21:35:39.672514900 +0200
+++ qbittorrent-4.5.2/dist/CMakeLists.txt	2023-04-19 21:35:49.917744500 +0200
@@ -1,6 +1,6 @@
 if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     add_subdirectory(mac)
-elseif (UNIX AND (NOT APPLE))
+elseif (UNIX AND (NOT APPLE) OR MINGW)
     add_subdirectory(unix)
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     add_subdirectory(windows)
