--- a/Makefile.PL
+++ b/Makefile.PL
@@ -288,11 +288,11 @@
         my $found = 0;
         my @pairs = ();
         # Library names depend on the compiler
-        @pairs = (['eay32','ssl32'],['crypto.dll','ssl.dll'],['crypto','ssl']) if $Config{cc} =~ /gcc/;
+        @pairs = (['eay32','ssl32'],['crypto.dll','ssl.dll'],['crypto','ssl']) if $Config{cc} =~ /(gcc|clang)/;
         @pairs = (['libeay32','ssleay32'],['libeay32MD','ssleay32MD'],['libeay32MT','ssleay32MT'],['libcrypto','libssl'],['crypto','ssl']) if $Config{cc} =~ /cl/;
         FOUND: for my $dir (@{$opts->{lib_paths}}) {
           for my $p (@pairs) {
-            $found = 1 if ($Config{cc} =~ /gcc/ && -f "$dir/lib$p->[0].a" && -f "$dir/lib$p->[1].a");
+            $found = 1 if ($Config{cc} =~ /(gcc|clang)/ && -f "$dir/lib$p->[0].a" && -f "$dir/lib$p->[1].a");
             $found = 1 if ($Config{cc} =~ /cl/ && -f "$dir/$p->[0].lib" && -f "$dir/$p->[1].lib");
             if ($found) {
               $opts->{lib_links} = [$p->[0], $p->[1], 'crypt32']; # Some systems need this system lib crypt32 too
