diff --git a/configure.ac b/configure.ac
index e4a94af..fac4ed7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,10 +2,11 @@ dnl Process this file with autoconf to produce a configure script.
 AC_INIT(TRE, 0.8.0)
 AC_CONFIG_SRCDIR([lib/regcomp.c])
 AC_CONFIG_AUX_DIR(utils)
+AC_CONFIG_MACRO_DIRS([m4])
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE([foreign])
-AC_PREREQ(2.59)
-AM_GNU_GETTEXT_VERSION(0.17)
+AC_PREREQ(2.69)
+AM_GNU_GETTEXT_VERSION(0.18)
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -433,16 +434,6 @@ if test "$tre_enable_wchar" != "no"; then
   fi
 fi
 
-case $host in
-  *-mingw*)
-    dnl wcsrtombs and wcstombs don't seem to work at all on MinGW.
-    if test "$tre_libutf8" != "yes"; then
-      tre_wchar_reason="Not supported on MinGW"
-      tre_wchar="no ($tre_wchar_reason)"
-    fi
-    ;;
-esac
-
 # Fail if wide character support was specifically requested but is
 # not supported on this system.
 if test "$tre_enable_wchar" = "yes"; then
