--- qt-everywhere-src-6.2.0/qtbase/mkspecs/common/g++-win32.conf.orig	2021-10-07 09:56:09.819539300 +0300
+++ qt-everywhere-src-6.2.0/qtbase/mkspecs/common/g++-win32.conf	2021-10-07 10:02:23.970867500 +0300
@@ -32,6 +32,7 @@
 QMAKE_CFLAGS_SSE2      += -mstackrealign
 
 QMAKE_CXXFLAGS_EXCEPTIONS_ON = -fexceptions
+QMAKE_CXXFLAGS_EXCEPTIONS_OFF = -fno-exceptions
 
 QMAKE_INCDIR            =
 
@@ -44,6 +45,7 @@
 QMAKE_LFLAGS_CONSOLE    = -Wl,-subsystem,console
 QMAKE_LFLAGS_WINDOWS    = -Wl,-subsystem,windows
 QMAKE_LFLAGS_DLL        = -shared
+QMAKE_LFLAGS_STATIC_LIB = -static
 QMAKE_LFLAGS_GCSECTIONS = -Wl,--gc-sections
 equals(QMAKE_HOST.os, Windows) {
     QMAKE_LINK_OBJECT_MAX = 10
@@ -56,6 +58,9 @@
 QMAKE_EXTENSION_STATICLIB = a
 QMAKE_LIB_EXTENSIONS    = a dll.a
 
+PKG_CONFIG              = $${CROSS_COMPILE}pkg-config
+QMAKE_PKG_CONFIG        = $${CROSS_COMPILE}pkg-config
+
 QMAKE_LIBS              =
 QMAKE_LIBS_GUI          = -lgdi32 -lcomdlg32 -loleaut32 -limm32 -lwinmm -lws2_32 -lole32 -luuid -luser32 -ladvapi32
 QMAKE_LIBS_NETWORK      = -lws2_32
@@ -64,9 +69,10 @@
 QMAKE_LIBS_OPENGL_ES2_DEBUG = -lgdi32 -luser32
 QMAKE_LIBS_COMPAT       = -ladvapi32 -lshell32 -lcomdlg32 -luser32 -lgdi32 -lws2_32
 
-QMAKE_IDL               = midl
+QMAKE_IDL               = $${CROSS_COMPILE}widl
 QMAKE_LIB               = $${CROSS_COMPILE}ar -rc
 QMAKE_RC                = $${CROSS_COMPILE}windres
+QMAKE_DLLTOOL           = $${CROSS_COMPILE}dlltool
 
 QMAKE_STRIP             = $${CROSS_COMPILE}strip
 QMAKE_STRIPFLAGS_LIB   += --strip-unneeded
--- qt-everywhere-src-6.2.0/qtbase/mkspecs/features/default_post.prf.orig	2021-10-07 10:01:04.313014900 +0300
+++ qt-everywhere-src-6.2.0/qtbase/mkspecs/features/default_post.prf	2021-10-07 10:01:38.737973300 +0300
@@ -87,8 +87,8 @@
     enable_gdb_index: QMAKE_LFLAGS += $$QMAKE_LFLAGS_GDB_INDEX
 }
 
+static: QMAKE_LFLAGS += $$QMAKE_LFLAGS_STATIC_LIB
 dll:win32: QMAKE_LFLAGS += $$QMAKE_LFLAGS_DLL
-static:mac: QMAKE_LFLAGS += $$QMAKE_LFLAGS_STATIC_LIB
 staticlib:unix {
     QMAKE_CFLAGS += $$QMAKE_CFLAGS_STATIC_LIB
     QMAKE_CXXFLAGS += $$QMAKE_CXXFLAGS_STATIC_LIB
