--- qt-everywhere-src-6.2.0/qtbase/src/corelib/Qt6CoreConfigExtras.cmake.in.orig	2021-10-07 10:34:57.233434800 +0300
+++ qt-everywhere-src-6.2.0/qtbase/src/corelib/Qt6CoreConfigExtras.cmake.in	2021-10-07 10:35:00.486881300 +0300
@@ -18,6 +18,15 @@
     endif()
     set_property(TARGET @QT_CMAKE_EXPORT_NAMESPACE@::Core APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG>)
     set_property(TARGET @QT_CMAKE_EXPORT_NAMESPACE@::Core PROPERTY INTERFACE_COMPILE_FEATURES cxx_decltype)
+    
+    get_target_property(_libType @QT_CMAKE_EXPORT_NAMESPACE@::Core TYPE)
+    if(_libType STREQUAL \"STATIC_LIBRARY\" AND WIN32)
+        set(_isExe $<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>)
+        # INTERFACE_LINK_LIBRARIES is used to pass a linker flag (-static)
+        #                                      and a library (ws2_32)
+        set_property(TARGET @QT_CMAKE_EXPORT_NAMESPACE@::Core APPEND PROPERTY INTERFACE_LINK_LIBRARIES \"$<${_isExe}:-Bstatic;ws2_32>\")
+        unset(_isExe)
+    endif()
 endif()
 
 set(CMAKE_AUTOMOC_MACRO_NAMES "Q_OBJECT" "Q_GADGET" "Q_NAMESPACE" "Q_NAMESPACE_EXPORT")
