--- a/qtbase/src/corelib/Qt6CoreConfigExtras.cmake.in
+++ b/qtbase/src/corelib/Qt6CoreConfigExtras.cmake.in
@@ -24,6 +24,15 @@
     endif()
     set_property(TARGET ${__qt_core_target} APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG>)
     set_property(TARGET ${__qt_core_target} PROPERTY INTERFACE_COMPILE_FEATURES cxx_decltype)
+    
+    get_target_property(_libType @QT_CMAKE_EXPORT_NAMESPACE@::Core TYPE)
+    if(_libType STREQUAL \"STATIC_LIBRARY\" AND WIN32)
+        set(_isExe $<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>)
+        # INTERFACE_LINK_LIBRARIES is used to pass a linker flag (-static)
+        #                                      and a library (ws2_32)
+        set_property(TARGET @QT_CMAKE_EXPORT_NAMESPACE@::Core APPEND PROPERTY INTERFACE_LINK_LIBRARIES \"$<${_isExe}:-Bstatic;ws2_32>\")
+        unset(_isExe)
+    endif()
 endif()
 
 set(CMAKE_AUTOMOC_MACRO_NAMES "Q_OBJECT" "Q_GADGET" "Q_GADGET_EXPORT" "Q_NAMESPACE" "Q_NAMESPACE_EXPORT")
