diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1111111..2222222 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,9 @@
 
 # set QT_SUPERBUILD early, so that qtbase/.cmake.conf can check it
 set(QT_SUPERBUILD TRUE)
+
+find_package(PkgConfig QUIET)
+list(APPEND PKG_CONFIG_EXECUTABLE "--static")
 
 # Include qtbase's .cmake.conf for access to QT_REPO_MODULE_VERSION
 set(__qt6_qtbase_src_path "${CMAKE_CURRENT_SOURCE_DIR}/qtbase")
diff --git a/qtimageformats/src/imageformats/configure.cmake b/qtimageformats/src/imageformats/configure.cmake
index 1111111..2222222 100644
--- a/qtimageformats/src/imageformats/configure.cmake
+++ b/qtimageformats/src/imageformats/configure.cmake
@@ -19,6 +19,8 @@ set_property(CACHE INPUT_webp PROPERTY STRINGS undefined no qt system)
 
 qt_find_package(WrapJasper PROVIDED_TARGETS WrapJasper::WrapJasper MODULE_NAME imageformats QMAKE_LIB jasper)
 qt_find_package(TIFF PROVIDED_TARGETS TIFF::TIFF MODULE_NAME imageformats QMAKE_LIB tiff)
+### <sigh>
+target_link_libraries(TIFF::TIFF INTERFACE -ltiff -lwebp -lsharpyuv -lzstd -llzma -lLerc -ljbig -ljpeg -ldeflate -lz)
 # Threads::Threads might be brought in via a top-level CMakeLists.txt find_package dependency
 # in which case if the system WebpConfig.cmake depends Threads, it shouldn't try to promote it to
 # global to avoid a 'global promotion of a target in a different subdirectory' error.
diff --git a/qtimageformats/cmake/FindLibMng.cmake b/qtimageformats/cmake/FindLibMng.cmake
index 1111111..2222222 100644
--- a/qtimageformats/cmake/FindLibMng.cmake
+++ b/qtimageformats/cmake/FindLibMng.cmake
@@ -7,6 +7,7 @@
 endif()
 
 find_package(PkgConfig QUIET)
+#list(APPEND PKG_CONFIG_EXECUTABLE "--static")
 
 if(PkgConfig_FOUND)
     pkg_check_modules(Libmng IMPORTED_TARGET libmng)
diff --git a/qtimageformats/cmake/FindWrapWebP.cmake b/qtimageformats/cmake/FindWrapWebP.cmake
index 1111111..2222222 100644
--- a/qtimageformats/cmake/FindWrapWebP.cmake
+++ b/qtimageformats/cmake/FindWrapWebP.cmake
@@ -51,7 +51,7 @@ mark_as_advanced(WebP_INCLUDE_DIR WebP_LIBRARY WebP_demux_INCLUDE_DIR WebP_demux
 if(WrapWebP_FOUND)
     set(WebP_FOUND ON)
     add_library(WrapWebP::WrapWebP INTERFACE IMPORTED)
-    target_link_libraries(WrapWebP::WrapWebP INTERFACE ${WebP_LIBRARY} ${WebP_demux_LIBRARY} ${WebP_mux_LIBRARY})
+    target_link_libraries(WrapWebP::WrapWebP INTERFACE ${WebP_LIBRARY} ${WebP_demux_LIBRARY} ${WebP_mux_LIBRARY} -lsharpyuv)
     target_include_directories(WrapWebP::WrapWebP
                                INTERFACE ${WebP_INCLUDE_DIR} ${WebP_demux_INCLUDE_DIR} ${WebP_mux_INCLUDE_DIR})
 endif()
