--- qt-everywhere-src-6.2.0/qtbase/src/corelib/global/qlibraryinfo.cpp.orig	2021-10-14 08:14:01.209986500 +0300
+++ qt-everywhere-src-6.2.0/qtbase/src/corelib/global/qlibraryinfo.cpp	2021-10-14 08:22:22.078864300 +0300
@@ -357,6 +357,13 @@
 #endif // Q_OS_DARWIN
         // We make the prefix path absolute to the executable's directory.
         appDir = QCoreApplication::applicationDirPath();
+        QDir pwd(appDir);
+        // Assume we have posix-like folder structure
+        if (pwd.dirName().toLower() == "bin") {
+            if (pwd.cdUp()) {
+                appDir = pwd.absolutePath();
+            }
+        }
     } else {
         appDir = QDir::currentPath();
     }
