--- a/qtbase/src/tools/windeployqt/main.cpp
+++ b/qtbase/src/tools/windeployqt/main.cpp
@@ -571,9 +571,6 @@
     options->quickImports = !parser->isSet(noQuickImportOption);
 
     // default to deployment of compiler runtime for windows desktop configurations
-    if (options->platform == WindowsDesktopMinGW || options->platform == WindowsDesktopClangMinGW
-        || options->platform.testFlags(WindowsDesktopMsvc) || parser->isSet(compilerRunTimeOption))
-        options->compilerRunTime = true;
     if (parser->isSet(noCompilerRunTimeOption))
         options->compilerRunTime = false;
 
@@ -1378,7 +1375,7 @@
     QStringList result;
     switch (platform) {
     case WindowsDesktopMinGW: {
-        const QStringList minGWRuntimes = { "*gcc_"_L1, "*stdc++"_L1, "*winpthread"_L1 };
+        const QStringList minGWRuntimes = { "*gcc_s_"_L1, "*stdc++"_L1, "*winpthread"_L1 };
         result.append(findMinGWRuntimePaths(qtBinDir, platform, minGWRuntimes));
         break;
     }
@@ -1642,7 +1639,7 @@
         disabled[QtQmlModuleId] = 1;
         disabled[QtQuickModuleId] = 1;
     }
-
+#if 0
     // Some Windows-specific checks: Qt5Core depends on ICU when configured with "-icu". Other than
     // that, Qt5WebKit has a hard dependency on ICU.
     if (options.platform.testFlag(WindowsBased))  {
@@ -1682,7 +1679,7 @@
             } // !icuLibs.isEmpty()
         } // Qt6Core/Qt6WebKit
     } // Windows
-
+#endif
     QStringList openSslLibs;
     if (!options.openSslRootDirectory.isEmpty()) {
         openSslLibs = findOpenSslLibraries(options.openSslRootDirectory, options.platform);
