--- a/qttools/src/qdoc/cmake/QDocConfiguration.cmake
+++ b/qttools/src/qdoc/cmake/QDocConfiguration.cmake
@@ -8,7 +8,7 @@
 
 # List of explicitly supported Clang versions for QDoc
 set(QDOC_SUPPORTED_CLANG_VERSIONS
-    "21.1" "20.1" "19.1" "18.1" "17.0.6"
+    "22.1" "22.0" "21.1" "20.1" "19.1" "18.1" "17.0.6"
 )
 
 # Check for QDoc coverage dependencies
--- a/qttools/src/qdoc/qdoc/CMakeLists.txt
+++ b/qttools/src/qdoc/qdoc/CMakeLists.txt
@@ -107,6 +107,7 @@
         QT_NO_KEYWORDS
         # To provide the ability to workaround version-specific Clang issues.
         # A re-export of (LLVM|CLANG)_VERSION_MAJOR done in WrapLibClang.cmake
+        QT_NO_EMIT
         LIBCLANG_VERSION_MAJOR=${QT_LIB_CLANG_VERSION_MAJOR}
 )
 qt_internal_return_unless_building_tools()
--- a/qttools/src/qdoc/qdoc/src/qdoc/clangcodeparser.cpp
+++ b/qttools/src/qdoc/qdoc/src/qdoc/clangcodeparser.cpp
@@ -42,7 +42,12 @@
 #include <clang/Lex/Lexer.h>
 #include <llvm/Support/Casting.h>
 
-#include "clang/AST/QualTypeNames.h"
+#if LIBCLANG_VERSION_MAJOR >= 22
+// LLVM 22 provides a compatible QualTypeNames implementation in libclang.
+#  include <clang/AST/QualTypeNames.h>
+#else
+#  include "clang/AST/QualTypeNames.h"
+#endif
 #include "template_declaration.h"
 
 #include <cstdio>
