--- snappy-1.1.1/Makefile.am.orig	2013-10-15 14:41:48.000000000 +0400
+++ snappy-1.1.1/Makefile.am	2014-09-11 09:48:08.146000000 +0400
@@ -3,7 +3,7 @@
 # Library.
 lib_LTLIBRARIES = libsnappy.la
 libsnappy_la_SOURCES = snappy.cc snappy-sinksource.cc snappy-stubs-internal.cc snappy-c.cc
-libsnappy_la_LDFLAGS = -version-info $(SNAPPY_LTVERSION)
+libsnappy_la_LDFLAGS = -version-info $(SNAPPY_LTVERSION) -no-undefined
 
 include_HEADERS = snappy.h snappy-sinksource.h snappy-stubs-public.h snappy-c.h
 noinst_HEADERS = snappy-internal.h snappy-stubs-internal.h snappy-test.h
