--- libnfs-libnfs-2.0.0/lib/Makefile.am.orig	2018-05-29 12:56:34.716417100 +0300
+++ libnfs-libnfs-2.0.0/lib/Makefile.am	2018-05-29 12:56:43.967233400 +0300
@@ -26,7 +26,7 @@
 SOCURRENT=11
 SOREVISION=0
 SOAGE=0
-libnfs_la_LDFLAGS = -version-info $(SOCURRENT):$(SOREVISION):$(SOAGE)
+libnfs_la_LDFLAGS = -no-undefined -version-info $(SOCURRENT):$(SOREVISION):$(SOAGE) -lws2_32 -export-symbols $(top_srcdir)/lib/libnfs-win32.def
 libnfs_la_LIBADD = \
 	../mount/libmount.la \
 	../nfs/libnfs.la \
--- libnfs-libnfs-2.0.0/lib/libnfs-win32.def.orig	2018-05-29 12:56:34.716417100 +0300
+++ libnfs-libnfs-2.0.0/lib/libnfs-win32.def	2018-05-29 12:56:43.967233400 +0300
@@ -1,4 +1,4 @@
-LIBRARY libnfs
+LIBRARY libnfs-11
 EXPORTS
 mount_free_export_list
 mount_getexports
