--- ogre-1.11.1/CMake/Utils/PrecompiledHeader.cmake.orig	2018-07-20 14:28:10.807064900 +0300
+++ ogre-1.11.1/CMake/Utils/PrecompiledHeader.cmake	2018-07-20 14:28:40.431516900 +0300
@@ -80,7 +80,7 @@
                item MATCHES "/usr/include$")
           # workaround for GCC 6.x bug
         else()
-          LIST(APPEND ${_out_compile_flags} "${_PCH_isystem_prefix}\"${item}\"")
+          LIST(APPEND ${_out_compile_flags} "${_PCH_isystem_prefix} \"${item}\"")
         endif()
     ENDFOREACH(item)
 
@@ -90,7 +90,7 @@
                item MATCHES "/usr/include$")
           # workaround for GCC 6.x bug
         else()
-          LIST(APPEND ${_out_compile_flags} "${_PCH_isystem_prefix}\"${item}\"")
+          LIST(APPEND ${_out_compile_flags} "${_PCH_isystem_prefix} \"${item}\"")
         endif()
     ENDFOREACH(item)
 
