diff -Naur ogre-1.12.10.a/PlugIns/Assimp/CMakeLists.txt ogre-1.12.10.b/PlugIns/Assimp/CMakeLists.txt
--- ogre-1.12.10.a/PlugIns/Assimp/CMakeLists.txt	2020-12-13 23:57:21.000000000 +0100
+++ ogre-1.12.10.b/PlugIns/Assimp/CMakeLists.txt	2020-12-26 09:01:20.648143200 +0100
@@ -8,7 +8,11 @@
 target_link_libraries(Codec_Assimp PUBLIC OgreMain PRIVATE fix::assimp)
 if(OGRE_BUILD_DEPENDENCIES)
     # hack to get our CI build going, despite horribly broken assimp cmake
-    target_link_libraries(Codec_Assimp PRIVATE ZLIB::ZLIB ${OGREDEPS_PATH}/lib/IrrXML.lib)
+    if(MSVC)
+      target_link_libraries(Codec_Assimp PRIVATE ZLIB::ZLIB ${OGREDEPS_PATH}/lib/IrrXML.lib)
+    else()
+      target_link_libraries(Codec_Assimp PRIVATE ZLIB::ZLIB ${OGREDEPS_PATH}/lib/libIrrXML.a)
+    endif()
 endif()
 
 ogre_config_framework(Codec_Assimp)
