--- a/CMake/ConfigureBuild.cmake
+++ b/CMake/ConfigureBuild.cmake
@@ -156,8 +156,12 @@
 endif ()
 
 # Create the pkg-config package files on Unix systems
-if (UNIX)
-  set(OGRE_PLUGIN_EXT ".so")
+if (UNIX OR MINGW)
+  if (MINGW)
+    set(OGRE_PLUGIN_EXT ".dll")
+  else()
+    set(OGRE_PLUGIN_EXT ".so")
+  endif()
   set(OGRE_PAGING_ADDITIONAL_PACKAGES "")
   if (OGRE_STATIC)
     set(OGRE_PLUGIN_EXT ".a")
