diff -bur ogre-13.4.4-c/Samples/Common/include/SamplePlugin.h ogre-13.4.4/Samples/Common/include/SamplePlugin.h
--- ogre-13.4.4-c/Samples/Common/include/SamplePlugin.h	2022-08-06 14:44:00.000000000 -0600
+++ ogre-13.4.4/Samples/Common/include/SamplePlugin.h	2022-08-16 22:43:44.863604600 -0600
@@ -35,7 +35,7 @@
 #if defined( OGRE_STATIC_LIB )
 #  define _OgreSampleExport
 #else
-#  if (OGRE_PLATFORM == OGRE_PLATFORM_WIN32 || OGRE_PLATFORM == OGRE_PLATFORM_WINRT) && !defined(__MINGW32__)
+#  if (OGRE_PLATFORM == OGRE_PLATFORM_WIN32 || OGRE_PLATFORM == OGRE_PLATFORM_WINRT)
 #    define _OgreSampleExport __declspec(dllexport)
 #  elif defined ( OGRE_GCC_VISIBILITY )
 #   define _OgreSampleExport  __attribute__ ((visibility("default")))
