diff -bur ogre-14.5.1-orig/Components/Bites/src/OgreWIN32ConfigDialog.cpp ogre-14.5.1/Components/Bites/src/OgreWIN32ConfigDialog.cpp
--- ogre-14.5.1-orig/Components/Bites/src/OgreWIN32ConfigDialog.cpp	2026-01-18 13:17:46.218635200 -0700
+++ ogre-14.5.1/Components/Bites/src/OgreWIN32ConfigDialog.cpp	2026-01-18 14:12:13.403028500 -0700
@@ -59,6 +59,10 @@
 #endif
     };
 
+// Stringify macros for version numbers
+#define OGRE_BITES_STRINGIFY(x) #x
+#define OGRE_BITES_TOSTRING(x) OGRE_BITES_STRINGIFY(x)
+
     ConfigDialog::ConfigDialog() : mImpl(new ConfigDialog::PrivateData())
     {
 		#ifdef __MINGW32__
@@ -66,9 +70,9 @@
                 mImpl->mHInstance = GetModuleHandle( NULL );
 			#else
 				#if OGRE_DEBUG_MODE == 1
-        		    mImpl->mHInstance = GetModuleHandle("OgreMain_d.dll");
+        		    mImpl->mHInstance = GetModuleHandle("OgreBites_d-" OGRE_BITES_TOSTRING(OGRE_VERSION_MAJOR) "." OGRE_BITES_TOSTRING(OGRE_VERSION_MINOR) ".dll");
 				#else
-					mImpl->mHInstance = GetModuleHandle("OgreMain.dll");
+					mImpl->mHInstance = GetModuleHandle("OgreBites-" OGRE_BITES_TOSTRING(OGRE_VERSION_MAJOR) "." OGRE_BITES_TOSTRING(OGRE_VERSION_MINOR) ".dll");
 				#endif
 			#endif
 		#else
