diff -bur ogre-1.12.6-c/OgreMain/src/OgrePlatformInformation.cpp ogre-1.12.6/OgreMain/src/OgrePlatformInformation.cpp
--- ogre-1.12.6-c/OgreMain/src/OgrePlatformInformation.cpp	2020-04-25 17:11:59.000000000 -0600
+++ ogre-1.12.6/OgreMain/src/OgrePlatformInformation.cpp	2022-08-09 21:44:43.608972000 -0600
@@ -36,7 +36,7 @@
 
     #if OGRE_PLATFORM == OGRE_PLATFORM_ANDROID
         #include <cpu-features.h>
-    #elif OGRE_CPU == OGRE_CPU_ARM 
+    #elif OGRE_CPU == OGRE_CPU_ARM && !defined(__MINGW32__)
         #include <sys/sysctl.h>
         #if __MACH__
             #include <mach/machine.h>
