--- a/cmake/AvogadroLibsConfig.cmake.in
+++ b/cmake/AvogadroLibsConfig.cmake.in
@@ -8,7 +8,7 @@
 set(AvogadroLibs_VERSION_PATCH   "@AvogadroLibs_VERSION_PATCH@")
 set(AvogadroLibs_VERSION         "${AvogadroLibs_VERSION_MAJOR}.${AvogadroLibs_VERSION_MINOR}.${AvogadroLibs_VERSION_PATCH}")
 
-set(AvogadroLibs_INSTALL_PREFIX  "@CMAKE_INSTALL_PREFIX@")
+set(AvogadroLibs_INSTALL_PREFIX  "${CMAKE_CURRENT_LIST_DIR}/../../..")
 set(AvogadroLibs_INCLUDE_DIRS    "${AvogadroLibs_INSTALL_PREFIX}/@INSTALL_INCLUDE_DIR@")
 set(AvogadroLibs_LIBRARY_DIR     "${AvogadroLibs_INSTALL_PREFIX}/@INSTALL_LIBRARY_DIR@")
 set(AvogadroLibs_RUNTIME_DIR     "${AvogadroLibs_INSTALL_PREFIX}/@INSTALL_RUNTIME_DIR@")
