--- harfbuzz-10.3.0/src/hb-directwrite.cc.orig	2025-02-21 08:53:58.020354200 +0100
+++ harfbuzz-10.3.0/src/hb-directwrite.cc	2025-02-21 08:54:10.274456300 +0100
@@ -277,7 +277,7 @@
 _hb_directwrite_shaper_font_data_destroy (hb_directwrite_font_data_t *data)
 {
   if (data != HB_SHAPER_DATA_SUCCEEDED)
-    ((IDWriteFont *) data)->Release();
+    ((IDWriteFont *) (const void *) data)->Release();
 }
 
 
