--- atk-2.11.90/atk/Makefile.am.orig	2014-03-07 01:56:34.868691600 +0000
+++ atk-2.11.90/atk/Makefile.am	2014-03-07 01:59:53.723943000 +0000
@@ -187,8 +187,8 @@
 # ---------- Win32 stuff ----------
 
 if OS_WIN32
-libatk_1_0_la_LDFLAGS += -export-symbols atk.def -no-undefined -Wl,atk-win32-res.o
-libatk_1_0_la_DEPENDENCIES = atk-win32-res.o atk.def
+libatk_1_0_la_LDFLAGS += -export-symbols-regex '^atk_' -no-undefined -Wl,atk-win32-res.o
+libatk_1_0_la_DEPENDENCIES = atk-win32-res.o atk.def
 
 install-def-file:
 	$(INSTALL) -d $(DESTDIR)$(libdir)
