--- a/setup.py
+++ b/setup.py
@@ -45,8 +45,6 @@
             else:
                 self.importLibraryName = os.path.join(self.build_implib,
                         "lib%s.a" % ext.name)
-                extraLinkArgs.append("-Wl,--add-stdcall-alias")
-                extraLinkArgs.append("-Wl,--enable-stdcall-fixup")
                 extraLinkArgs.append("-Wl,--out-implib=%s" % \
                         self.importLibraryName)
             ext.libraries = ["ole32"]
