--- a/configure.ac
+++ b/configure.ac
@@ -347,11 +347,9 @@
 AC_LANG_PUSH([C++])
 AC_CHECK_HEADERS([boost/pool/pool_alloc.hpp])
 AC_LANG_POP([C++])
-AX_BOOST_SYSTEM
-LDFLAGS="$LDFLAGS $BOOST_LDFLAGS $BOOST_SYSTEM_LIB"
 fi
 
-if test "x$want_boost" = xyes -a "x$boost_new" = xyes -a "x$ac_cv_header_boost_pool_pool_alloc_hpp" = xyes -a "x$link_system" = xyes ; then
+if test "x$want_boost" = xyes -a "x$boost_new" = xyes -a "x$ac_cv_header_boost_pool_pool_alloc_hpp" = xyes ; then
     using_boost_pool=yes
     USE_BOOST_POOL="#define USE_BOOST_POOL 1"
 else
