--- tulip-5.5.1/cmake/TulipPython.cmake.orig	2021-07-13 22:20:27.589248200 +0200
+++ tulip-5.5.1/cmake/TulipPython.cmake	2021-07-13 22:20:32.731368500 +0200
@@ -41,6 +41,10 @@
                   OUTPUT_VARIABLE TulipPythonModulesInstallDir)
   STRING(REPLACE "\n" "" TulipPythonModulesInstallDir "${TulipPythonModulesInstallDir}")
 
+IF(MINGW)
+  SET(TulipPythonModulesInstallDir ${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION}/site-packages)
+ENDIF(MINGW)
+
 ELSE(TULIP_PYTHON_SITE_INSTALL)
   SET(TulipPythonModulesInstallDir ${CMAKE_INSTALL_PREFIX}/${TulipLibInstallDir}/tulip/python)
 ENDIF(TULIP_PYTHON_SITE_INSTALL)
