diff -Naur tulip-5.1.0-orig/library/tulip-gui/src/CoordEditor.cpp tulip-5.1.0/library/tulip-gui/src/CoordEditor.cpp
--- tulip-5.1.0-orig/library/tulip-gui/src/CoordEditor.cpp	2017-11-09 01:54:34.000000000 +0300
+++ tulip-5.1.0/library/tulip-gui/src/CoordEditor.cpp	2018-07-10 10:46:10.828636100 +0300
@@ -22,6 +22,7 @@
 #include "ui_CoordEditor.h"
 
 #include <cfloat>
+#include <QDoubleValidator>
 
 using namespace tlp;
 
diff -Naur tulip-5.1.0-orig/library/tulip-gui/src/CSVImportWizard.cpp tulip-5.1.0/library/tulip-gui/src/CSVImportWizard.cpp
--- tulip-5.1.0-orig/library/tulip-gui/src/CSVImportWizard.cpp	2017-11-09 01:54:34.000000000 +0300
+++ tulip-5.1.0/library/tulip-gui/src/CSVImportWizard.cpp	2018-07-10 10:42:35.426047000 +0300
@@ -20,6 +20,7 @@
 
 #include <QVBoxLayout>
 #include <QLabel>
+#include <QHeaderView>
 
 #include <tulip/CSVGraphImport.h>
 #include <tulip/CSVParserConfigurationWidget.h>
diff -Naur tulip-5.1.0-orig/plugins/view/SOMView/SOMPropertiesWidget.cpp tulip-5.1.0/plugins/view/SOMView/SOMPropertiesWidget.cpp
--- tulip-5.1.0-orig/plugins/view/SOMView/SOMPropertiesWidget.cpp	2017-11-09 01:54:34.000000000 +0300
+++ tulip-5.1.0/plugins/view/SOMView/SOMPropertiesWidget.cpp	2018-07-10 10:52:09.728669800 +0300
@@ -27,6 +27,7 @@
 #include <QDoubleValidator>
 #include <QRadioButton>
 #include <QPushButton>
+#include <QButtonGroup>
 
 #include <tulip/GraphPropertiesSelectionWidget.h>
 #include <tulip/ColorScalesManager.h>
diff -Naur tulip-5.1.0-orig/software/tulip/src/TulipWelcomePage.cpp tulip-5.1.0/software/tulip/src/TulipWelcomePage.cpp
--- tulip-5.1.0-orig/software/tulip/src/TulipWelcomePage.cpp	2017-11-09 01:54:35.000000000 +0300
+++ tulip-5.1.0/software/tulip/src/TulipWelcomePage.cpp	2018-07-10 10:59:04.964402000 +0300
@@ -23,6 +23,7 @@
 #include <QFileInfo>
 #include <QDesktopServices>
 #include <QApplication>
+#include <QStyle>
 #include <QNetworkAccessManager>
 #include <QNetworkRequest>
 #include <QNetworkReply>
