diff -bur tulip-6.0.1-orig/library/tulip-python/bindings/tulip-core/CMakeLists.txt tulip-6.0.1/library/tulip-python/bindings/tulip-core/CMakeLists.txt
--- tulip-6.0.1-orig/library/tulip-python/bindings/tulip-core/CMakeLists.txt	2025-04-11 05:54:01.076325400 -0600
+++ tulip-6.0.1/library/tulip-python/bindings/tulip-core/CMakeLists.txt	2025-04-11 05:58:16.550556500 -0600
@@ -101,6 +101,7 @@
 
 IF(WIN32)
   TARGET_COMPILE_DEFINITIONS(${LibTulipCorePythonBindingsName} PUBLIC "DLL_TULIP_PYTHON")
+  TARGET_COMPILE_DEFINITIONS(${LibTulipCorePythonBindingsName} PUBLIC "DLL_TULIP")
 ENDIF(WIN32)
 TARGET_COMPILE_DEFINITIONS(${LibTulipCorePythonBindingsName} PUBLIC "protected=public")
 TARGET_COMPILE_DEFINITIONS(${LibTulipCorePythonBindingsName} PUBLIC "SIP_PROTECTED_IS_PUBLIC")
diff -bur tulip-6.0.1-orig/plugins/view/MatrixView/CMakeLists.txt tulip-6.0.1/plugins/view/MatrixView/CMakeLists.txt
--- tulip-6.0.1-orig/plugins/view/MatrixView/CMakeLists.txt	2025-04-11 06:00:02.282979700 -0600
+++ tulip-6.0.1/plugins/view/MatrixView/CMakeLists.txt	2025-04-11 06:01:15.986879800 -0600
@@ -21,6 +21,11 @@
 QTX_ADD_RESOURCES(RCC_SRCS MatrixView.qrc)
 
 ADD_LIBRARY(MatrixView-${Tulip_VERSION} SHARED ${LIB_SRCS} ${LIB_UI_HEADERS} ${LIB_MOC_SRCS} ${RCC_SRCS})
+
+IF(WIN32)
+  TARGET_COMPILE_DEFINITIONS(MatrixView-${Tulip_VERSION} PUBLIC "DLL_TULIP")
+ENDIF(WIN32)
+
 TARGET_LINK_LIBRARIES(MatrixView-${Tulip_VERSION}
     PUBLIC ${LibTulipCoreName} ${LibTulipOGLName} ${LibTulipGUIName}
     INTERFACE OpenGL::GL)
