diff -urN tulip-5.5.1-orig/library/tulip-python/bindings/tulip-core/CMakeLists.txt tulip-5.5.1/library/tulip-python/bindings/tulip-core/CMakeLists.txt
--- tulip-5.5.1-orig/library/tulip-python/bindings/tulip-core/CMakeLists.txt	2021-01-21 06:13:08.000000000 +0800
+++ tulip-5.5.1/library/tulip-python/bindings/tulip-core/CMakeLists.txt	2021-11-22 07:21:34.066972700 +0800
@@ -1,7 +1,7 @@
 ADD_CORE_FILES(.)
 
 IF(WIN32)
-  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDLL_TULIP_PYTHON")
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDLL_TULIP_PYTHON -DDLL_TULIP")
 ENDIF(WIN32)
 
 # Visual Studio has a limit for string constants length
diff -urN tulip-5.5.1-orig/plugins/view/MatrixView/CMakeLists.txt tulip-5.5.1/plugins/view/MatrixView/CMakeLists.txt
--- tulip-5.5.1-orig/plugins/view/MatrixView/CMakeLists.txt	2021-01-21 06:13:08.000000000 +0800
+++ tulip-5.5.1/plugins/view/MatrixView/CMakeLists.txt	2021-11-22 09:32:15.277494400 +0800
@@ -1,5 +1,9 @@
 INCLUDE_DIRECTORIES(${TulipCoreBuildInclude} ${TulipCoreInclude} ${TulipOGLInclude} ${TulipGUIInclude} ${OPENGL_INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 
+IF(WIN32)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDLL_TULIP")
+ENDIF(WIN32)
+
 SET(LIB_SRCS
     MatrixView.cpp
     PropertyValuesDispatcher.cpp
