--- a/library/tulip-gui/src/PythonVersionChecker.cpp
+++ b/library/tulip-gui/src/PythonVersionChecker.cpp
@@ -25,6 +25,11 @@
 
 #include <iostream>
 
+#ifdef WIN32
+#include <QFileInfo>
+#include <windows.h>
+#endif
+
 using namespace tlp;
 using namespace std;
 
@@ -34,10 +39,6 @@
 // Windows specific functions
 #ifdef WIN32
 
-#include <QFileInfo>
-
-#include <windows.h>
-
 #ifndef MSYS2_PYTHON
 // Check if a path is a valid Python Home, meaning it is not empty and contains the python
 // executable
--- a/library/tulip-gui/src/TlpQtTools.cpp
+++ b/library/tulip-gui/src/TlpQtTools.cpp
@@ -367,7 +367,11 @@
   // depending on Python.
   auto pythonHome = tlp::PythonVersionChecker::getPythonHome();
   if (!pythonHome.isEmpty())
-    SetDllDirectory(pythonHome.toUtf8().data());
+#ifdef UNICODE
+    SetDllDirectory(pythonHome.toStdWString().data());
+#else
+    SetDllDirectory(pythonHome.toStdString().data());
+#endif
 #endif
 #endif
 #endif
