diff -bur tulip-6.0.1-orig/software/crash_handling/CMakeLists.txt tulip-6.0.1/software/crash_handling/CMakeLists.txt
--- tulip-6.0.1-orig/software/crash_handling/CMakeLists.txt	2025-04-11 06:38:54.238745600 -0600
+++ tulip-6.0.1/software/crash_handling/CMakeLists.txt	2025-03-31 15:30:22.000000000 -0600
@@ -20,9 +20,9 @@
     # for libbfd.a which is needed by crashandling static library
     # Check if libsframe is present in the MinGW build environment
     # FIND_LIBRARY(SFRAME_LIBRARY sframe PATHS ${MINGW_LIB_PATH} ${MINGW_LIB64_PATH} REQUIRED)
-    FIND_LIBRARY(SFRAME_LIBRARY sframe REQUIRED)
+    #FIND_LIBRARY(SFRAME_LIBRARY sframe REQUIRED)
     FIND_PACKAGE(Zstd)
-    MARK_AS_ADVANCED(SFRAME_LIBRARY)
+    #MARK_AS_ADVANCED(SFRAME_LIBRARY)
   ENDIF(MINGW)
 
 ENDIF(LINUX OR MINGW)
