--- PyICU-2.5.orig/setup.py	2019-04-12 21:52:55.795377100 -0400
+++ PyICU-2.5/setup.py	2019-04-14 22:08:46.314364100 -0400
@@ -83,7 +83,7 @@
     'darwin': True,
     'linux': True,
     'freebsd': False, # not tested
-    'win32': False,   # no icu-config
+    'win32': True,
     'sunos5': False,  # not tested
     'cygwin': False,  # not tested
 }
@@ -92,7 +92,7 @@
     'darwin': False,  # no pkg-config ?
     'linux': True,
     'freebsd': False, # not tested
-    'win32': False,   # no pkg-config ?
+    'win32': True,
     'sunos5': False,  # not tested
     'cygwin': False,  # not tested
 }
@@ -101,7 +101,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['/usr/local/include'],
-    'win32': ['c:/icu/include'],
+    'win32': [],
     'sunos5': [],
     'cygwin': [],
 }
@@ -107,7 +107,7 @@
 }
 
 if sys.platform == 'win32' and sys.version_info < (3,9):
-    ver_flag = '/D%s=\\"%s\\"'
+    ver_flag = '-D%s=\"%s\"'
 else:
     ver_flag = '-D%s="%s"'
 
@@ -134,7 +134,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['-std=c++11'],
-    'win32': ['/Zc:wchar_t', '/EHsc'],
+    'win32': [],
     'sunos5': ['-std=c++11'],
     'cygwin': ['-D_GNU_SOURCE=1', '-std=c++11'],
 }
@@ -144,7 +144,7 @@
     'darwin': ['-O0', '-g', '-DDEBUG'],
     'linux': ['-O0', '-g', '-DDEBUG'],
     'freebsd': ['-O0', '-g', '-DDEBUG'],
-    'win32': ['/Od', '/DDEBUG'],
+    'win32': ['-O0', '-g', '-DDEBUG'],
     'sunos5': ['-DDEBUG'],
     'cygwin': ['-Og', '-g', '-DDEBUG'],
 }
@@ -153,7 +153,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['-L/usr/local/lib'],
-    'win32': ['/LIBPATH:c:/icu/lib'],
+    'win32': [],
     'sunos5': [],
     'cygwin': [],
 }
@@ -162,7 +162,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['icui18n', 'icuuc', 'icudata'],
-    'win32': ['icuin', 'icuuc', 'icudt'],
+    'win32': [],
     'sunos5': ['icui18n', 'icuuc', 'icudata'],
     'cygwin': ['icui18n', 'icuuc', 'icudata'],
 }
