--- ollama-0.12.9/ml/path.go.orig	2025-11-01 21:17:37.803193200 +0100
+++ ollama-0.12.9/ml/path.go	2025-11-01 21:17:40.764164400 +0100
@@ -26,7 +26,7 @@
 	var libPath string
 	switch runtime.GOOS {
 	case "windows":
-		libPath = filepath.Join(filepath.Dir(exe), "lib", "ollama")
+		libPath = filepath.Join(filepath.Dir(exe), "..", "lib", "ollama")
 	case "linux":
 		libPath = filepath.Join(filepath.Dir(exe), "..", "lib", "ollama")
 	case "darwin":
--- ollama-0.12.9/ml/backend/ggml/ggml/src/ggml.go.orig	2025-10-31 23:23:28.000000000 +0100
+++ ollama-0.12.9/ml/backend/ggml/ggml/src/ggml.go	2025-11-01 21:15:47.971647000 +0100
@@ -63,7 +62,7 @@
 	case "darwin":
 		value = filepath.Dir(exe)
 	case "windows":
-		value = filepath.Join(filepath.Dir(exe), "lib", "ollama")
+		value = filepath.Join(filepath.Dir(exe), "..", "lib", "ollama")
 	default:
 		value = filepath.Join(filepath.Dir(exe), "..", "lib", "ollama")
 	}
--- ollama-0.20.2/ml/backend/ggml/ggml/src/ggml-backend-reg.cpp.orig	2026-04-04 06:50:55.000000000 +0200
+++ ollama-0.20.2/ml/backend/ggml/ggml/src/ggml-backend-reg.cpp	2026-04-04 22:51:15.903952900 +0200
@@ -521,7 +521,7 @@
 }
 
 static fs::path backend_filename_prefix() {
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
     return fs::u8path("ggml-");
 #else
     return fs::u8path("libggml-");
