diff -bur ollama-orig/llm/ext_server/server.cpp ollama/llm/ext_server/server.cpp
--- ollama-orig/llm/ext_server/server.cpp	2024-05-12 05:17:42.191887500 -0600
+++ ollama/llm/ext_server/server.cpp	2024-05-12 05:17:55.851547200 -0600
@@ -2771,27 +2771,7 @@
     shutdown_handler(signal);
 }
 
-#if defined(_WIN32)
-char* wchar_to_char(const wchar_t* wstr) {
-    if (wstr == nullptr) return nullptr;
-
-    // Determine the number of bytes needed for the UTF-8 string
-    int bytes = WideCharToMultiByte(CP_UTF8, 0, wstr, -1, nullptr, 0, nullptr, nullptr);
-    char* str = new char[bytes];
-
-    // Convert the wide-character string to a UTF-8 string
-    WideCharToMultiByte(CP_UTF8, 0, wstr, -1, str, bytes, nullptr, nullptr);
-    return str;
-}
-
-int wmain(int argc, wchar_t **wargv) {
-    char** argv = new char*[argc];
-    for (int i = 0; i < argc; ++i) {
-        argv[i] = wchar_to_char(wargv[i]);
-    }
-#else
 int main(int argc, char **argv) {
-#endif
 
 #if SERVER_VERBOSE != 1
     log_disable();
