--- glade3-3.8.6/configure.ac.orig	2019-12-30 17:15:04.993510200 +0100
+++ glade3-3.8.6/configure.ac	2019-12-30 17:15:10.698110800 +0100
@@ -192,7 +192,7 @@
 # Glade User Manual (requires yelp-tools)
 # ==================================================================
 
-YELP_HELP_INIT
+# YELP_HELP_INIT
 
 dnl ================================================================
 dnl Check for windows
@@ -290,7 +290,6 @@
 po/Makefile.in
 doc/Makefile
 doc/version.xml
-help/Makefile
 build/mingw-w64/glade.nsi
 ])
 
--- glade3-3.8.6/Makefile.am.orig	2017-08-07 20:37:38.000000000 +0200
+++ glade3-3.8.6/Makefile.am	2019-12-30 17:18:30.575110900 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = po data gladeui src plugins doc help
+SUBDIRS = po data gladeui src plugins doc
 
 ACLOCAL_AMFLAGS = -I m4
 
