From e4aa0e45776290a8464d2e45488b6fbebd7fcaf4 Mon Sep 17 00:00:00 2001
From: Benedikt Bitterli <mail@noobody.org>
Date: Sun, 22 Nov 2015 12:26:15 +0100
Subject: [PATCH 4/4] CMake: Install compiled example executables

---
 example/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/example/CMakeLists.txt b/example/CMakeLists.txt
index 8063d89..eb3e318 100644
--- a/example/CMakeLists.txt
+++ b/example/CMakeLists.txt
@@ -27,6 +27,9 @@ endif()
 
 foreach (example ${EXAMPLES})
     add_executable(${example} ${example}/${example}.cpp)
+    install(TARGETS ${example}
+            COMPONENT examples
+            DESTINATION "${DOC_INSTALL_DIR}/examples/${example}")
 endforeach()
 
 add_custom_target(examples ALL DEPENDS ${EXAMPLES})
-- 
2.5.0

