--- rapidjson-1.1.0/CMakeLists.txt.orig	2018-01-22 08:11:50.772633600 +0300
+++ rapidjson-1.1.0/CMakeLists.txt	2018-01-22 08:12:18.790021800 +0300
@@ -58,6 +58,9 @@
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
         endif()
     endif()
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0)
+        add_compile_options(-Wno-implicit-fallthrough)
+    endif()
     if (RAPIDJSON_BUILD_ASAN)
         if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.0")
             message(FATAL_ERROR "GCC < 4.8 doesn't support the address sanitizer")
