--- lxml-3.3.1/setupinfo.py.orig	2014-03-05 01:32:55.698960400 +0000
+++ lxml-3.3.1/setupinfo.py	2014-03-05 01:47:39.983538500 +0000
@@ -1,6 +1,7 @@
 import sys, os, os.path
 from distutils.core import Extension
 from distutils.errors import DistutilsOptionError
+from distutils.util import get_platform
 from versioninfo import get_base_dir, split_version
 
 try:
@@ -180,7 +181,7 @@
     return result
 
 def libraries():
-    if sys.platform in ('win32',):
+    if get_platform().startswith('win'):
         libs = ['libxslt', 'libexslt', 'libxml2', 'iconv']
         if OPTION_STATIC:
             libs = ['%s_a' % lib for lib in libs]
