diff -bur nss-3.81-c/nss/coreconf/WIN32.mk nss-3.81/nss/coreconf/WIN32.mk
--- nss-3.81-c/nss/coreconf/WIN32.mk	2022-07-30 19:03:00.244011400 -0600
+++ nss-3.81/nss/coreconf/WIN32.mk	2022-07-30 19:27:38.439836200 -0600
@@ -204,14 +204,17 @@
 endif
 
 ifeq (,$(filter-out x386 x86_64,$(CPU_ARCH)))
-ifdef USE_64
+ifdef USE_ARM64
+	CPU_ARCH = aarch64
+else	
+	ifdef USE_64
 	DEFINES += -D_AMD64_
 	# Use subsystem 5.02 to allow running on Windows XP.
 	ifeq ($(_MSC_VER_GE_11),1)
 		LDFLAGS += -SUBSYSTEM:CONSOLE,5.02
 	endif
 	CPU_ARCH = x86_64
-else
+	else
 	DEFINES += -D_X86_
 	# VS2012 defaults to -arch:SSE2. Use -arch:IA32 to avoid requiring
 	# SSE2. Clang-cl gets confused by -arch:IA32, so don't add it.
@@ -224,6 +227,7 @@
 		LDFLAGS += -SUBSYSTEM:CONSOLE,5.01
 	endif
 	CPU_ARCH = x386
+	endif
 endif
 endif
 ifeq ($(CPU_ARCH), ALPHA)
