--- a/configure.ac
+++ b/configure.ac
@@ -57,9 +57,9 @@
 		  AC_CHECK_HEADER([GL/gl.h],
 		  		  [],
 		  		  AC_MSG_ERROR([GL not found]))
-		  AC_CHECK_LIB([GL],
+		  AC_CHECK_LIB([opengl32],
 			       [glBegin],
-			       [GL_LIBS=-lGL],
+			       [GL_LIBS=-lopengl32],
 			       AC_MSG_ERROR([GL required]))
 		  ])
 
@@ -72,15 +72,15 @@
 	[AS_HELP_STRING([--with-glut=DIR],
 			[glut install directory])],
 	[GLUT_CFLAGS="-I$withval/include"
-	 GLUT_LIBS="-L$withval/lib -lglut"],
+	 GLUT_LIBS="-L$withval/lib -lfreeglut"],
 	[GLUT_CFLAGS=""
-	 GLUT_LIBS="-lglut"]
+	 GLUT_LIBS="-lfreeglut"]
         )
 AS_IF([test "x$with_glut" != xno],
       [AC_CHECK_HEADER([GL/glut.h],
                        [],
                        [glut_enabled=no])
-       AC_CHECK_LIB([glut],
+       AC_CHECK_LIB([freeglut],
                     [glutInit],
                     [],
                     [glut_enabled=no])],
@@ -104,9 +104,9 @@
 		  [AC_CHECK_HEADER([GL/glu.h],
 				   [],
 		  		   AC_MSG_ERROR([GLU not found]))
-		   AC_CHECK_LIB([GLU],
+		   AC_CHECK_LIB([glu32],
 				[gluBeginCurve],
-				[GLU_LIBS=-lGLU],
+				[GLU_LIBS=-lglu32],
 				AC_MSG_ERROR([GLU required])) ])
 
 DEMO_CFLAGS="$DEMO_CFLAGS $GLU_CFLAGS"
@@ -277,6 +277,9 @@
 cygwin*)
     DEMO_LIBS="$DEMO_LIBS -lX11"
     ;;
+mingw*)
+    DEMO_LIBS="$DEMO_LIBS -lwinmm"
+    ;;
 *)
     DEMO_LIBS="$DEMO_LIBS -lm"
     ;;
