From 8c9a5afc094b73fe94b05bff6ed3e43813f576ee Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Sat, 28 Aug 2021 21:03:41 +0530
Subject: [PATCH 2/3] CMake: Install dll in bindir

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index abccfed..fa0eadb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -501,7 +501,7 @@ if(ENABLE_SHARED)
     VERSION ${FFTS_MAJOR}.${FFTS_MINOR}.${FFTS_MICRO}
   )
 
-  install( TARGETS ffts_shared DESTINATION ${LIB_INSTALL_DIR} )
+  install( TARGETS ffts_shared DESTINATION ${CMAKE_INSTALL_BINDIR} )
 endif(ENABLE_SHARED)
 
 if(ENABLE_STATIC)
-- 
2.33.0

