--- a/libvaladoc/moduleloader.vala
+++ b/libvaladoc/moduleloader.vala
@@ -64,7 +64,9 @@
 			if (is_doclet(local_path)) {
 				return local_path;
 			} else {
-				return Path.build_filename (Config.PACKAGE_VALADOC_LIBDIR, pluginsubdir, pluginpath);
+				string? prefix = Win32.get_package_installation_directory_of_module(null);
+				string? doclets = prefix + "/lib/valadoc" + Config.PACKAGE_SUFFIX;
+				return Path.build_filename (doclets, pluginsubdir, pluginpath);
 			}
 		}
 
@@ -73,7 +75,9 @@
 
 	public static string get_doclet_path (string? docletpath, ErrorReporter reporter) {
 		if (docletpath == null) {
-			return Path.build_filename (Config.PACKAGE_VALADOC_LIBDIR, "doclets", "html");
+			string? prefix = Win32.get_package_installation_directory_of_module(null);
+			string? doclets = prefix + "/lib/valadoc" + Config.PACKAGE_SUFFIX;
+			return Path.build_filename (doclets, "doclets", "html");
 		}
 
 		return get_plugin_path (docletpath, "doclets");
