--- a/cmake/Modules/Tools.cmake
+++ b/cmake/Modules/Tools.cmake
@@ -12,11 +12,11 @@
     check_language(Fortran)
     if(CMAKE_Fortran_COMPILER)
       enable_language(Fortran)
-      add_executable(chain.x ${LAMMPS_TOOLS_DIR}/chain.f90)
-      target_link_libraries(chain.x PRIVATE ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
-      add_executable(micelle2d.x ${LAMMPS_TOOLS_DIR}/micelle2d.f90)
-      target_link_libraries(micelle2d.x PRIVATE ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
-      install(TARGETS chain.x micelle2d.x DESTINATION ${CMAKE_INSTALL_BINDIR})
+      add_executable(chain ${LAMMPS_TOOLS_DIR}/chain.f90)
+      target_link_libraries(chain PRIVATE ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
+      add_executable(micelle2d ${LAMMPS_TOOLS_DIR}/micelle2d.f90)
+      target_link_libraries(micelle2d PRIVATE ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
+      install(TARGETS chain micelle2d DESTINATION ${CMAKE_INSTALL_BINDIR})
     else()
       message(WARNING "No suitable Fortran compiler found, skipping build of 'chain.x' and 'micelle2d.x'")
     endif()
