From 61e98c296f01a10bc2ce49f546b2f28d09e73572 Mon Sep 17 00:00:00 2001
From: Markus Rathgeb <maggu2810@gmail.com>
Date: Tue, 5 Oct 2021 10:29:13 +0200
Subject: [PATCH 2/2] fix dll import / export instructions

Signed-off-by: Markus Rathgeb <maggu2810@gmail.com>
---
 source/corvusoft/restbed/http.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/corvusoft/restbed/http.hpp b/source/corvusoft/restbed/http.hpp
index 934744f..46e2c95 100644
--- a/source/corvusoft/restbed/http.hpp
+++ b/source/corvusoft/restbed/http.hpp
@@ -19,9 +19,9 @@
 //Windows DLL Exports
 #if defined(_WIN32) || defined(__WIN32__) || defined(WIN32) || defined(_WIN64)
 	#ifdef WIN_DLL_EXPORT
-		#define HTTP_EXPORT __declspec(dllexport)
+		#define HTTP_EXPORT [[gnu::dllexport]]
 	#else
-		#define HTTP_EXPORT __declspec(dllimport)
+		#define HTTP_EXPORT [[gnu::dllimport]]
 	#endif
 #else
 	#define HTTP_EXPORT
-- 
2.33.0

