diff --git a/Makefile b/Makefile
index 1fca8ddc..c6097eda 100644
--- a/Makefile
+++ b/Makefile
@@ -54,7 +54,7 @@ ifneq (, $(findstring darwin, $(SYS)))
 	else
 		include Makefile.osx
 	endif
-else ifneq (, $(findstring linux, $(SYS))$(findstring gnu, $(SYS)))
+else ifneq (, $(findstring linux, $(SYS)))
 	DAEMON_SRC += $(DAEMON_SRC_DIR)/UnixDaemon.cpp
 	include Makefile.linux
 else ifneq (, $(findstring freebsd, $(SYS))$(findstring openbsd, $(SYS)))
diff --git a/Makefile.mingw b/Makefile.mingw
index a1a861e6..1e1ad35e 100644
--- a/Makefile.mingw
+++ b/Makefile.mingw
@@ -5,7 +5,6 @@ WINDRES = windres
 
 CXXFLAGS := $(CXX_DEBUG) -fPIC -msse
 INCFLAGS = -I$(DAEMON_SRC_DIR) -IWin32
-LDFLAGS := ${LD_DEBUG} -static
 
 NEEDED_CXXFLAGS += -std=c++17 -DWIN32_LEAN_AND_MEAN
 
@@ -14,7 +13,7 @@ BOOST_SUFFIX = -mt
 
 # UPNP Support
 ifeq ($(USE_UPNP),yes)
-	CXXFLAGS += -DUSE_UPNP -DMINIUPNP_STATICLIB
+	CXXFLAGS += -DUSE_UPNP
 	LDLIBS = -lminiupnpc
 endif
 
