--- a/Makefile.mingw
+++ b/Makefile.mingw
@@ -5,7 +5,6 @@ WINDRES = windres
 
 CXXFLAGS := $(CXX_DEBUG) -fPIC -msse
 INCFLAGS = -I$(DAEMON_SRC_DIR) -IWin32
-LDFLAGS := ${LD_DEBUG} -static
 
 NEEDED_CXXFLAGS += -std=c++17 -DWIN32_LEAN_AND_MEAN
 
@@ -14,7 +13,7 @@ BOOST_SUFFIX = -mt
 
 # UPNP Support
 ifeq ($(USE_UPNP),yes)
-	CXXFLAGS += -DUSE_UPNP -DMINIUPNP_STATICLIB
+	CXXFLAGS += -DUSE_UPNP
 	LDLIBS = -lminiupnpc
 endif
 
