--- a/Makefile.mingw
+++ b/Makefile.mingw
@@ -5,7 +5,7 @@ WINDRES = windres
 
 CXXFLAGS := $(CXX_DEBUG) -fPIC -msse
 INCFLAGS := -I$(DAEMON_SRC_DIR) -IWin32
-LDFLAGS := ${LD_DEBUG} -static
+LDFLAGS := ${LD_DEBUG}
 
 NEEDED_CXXFLAGS += -std=c++17
 DEFINES += -DWIN32_LEAN_AND_MEAN
@@ -15,7 +15,7 @@ BOOST_SUFFIX = -mt
 
 # UPNP Support
 ifeq ($(USE_UPNP),yes)
-	DEFINES += -DUSE_UPNP -DMINIUPNP_STATICLIB
+	DEFINES += -DUSE_UPNP
 	LDLIBS = -lminiupnpc
 endif
 
