diff -Naur hdf-4.2.13-orig/CMakeLists.txt hdf-4.2.13/CMakeLists.txt
--- hdf-4.2.13-orig/CMakeLists.txt	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/CMakeLists.txt	2018-02-14 14:17:41.431264600 +0300
@@ -226,7 +226,7 @@
   set (HDF4_INSTALL_INCLUDE_DIR include)
 endif ()
 if (NOT HDF4_INSTALL_DATA_DIR)
-  if (NOT WIN32)
+  if (NOT MSVC)
     if (APPLE)
       if (HDF4_BUILD_FRAMEWORKS)
         set (HDF4_INSTALL_EXTRA_DIR ../SharedSupport)
@@ -301,7 +301,7 @@
   set (CMAKE_Fortran_MODULE_DIRECTORY
       ${PROJECT_BINARY_DIR}/bin/fortran CACHE PATH "Single Directory for all fortran modules."
   )
-  if (WIN32)
+  if (MSVC)
     set (CMAKE_TEST_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${CTEST_CONFIGURATION_TYPE})
     set (CMAKE_PDB_OUTPUT_DIRECTORY
         ${PROJECT_BINARY_DIR}/bin CACHE PATH "Single Directory for all pdb files."
@@ -578,7 +578,7 @@
 # Option to build HDF4 xdr Library
 #-----------------------------------------------------------------------------
 option (HDF4_BUILD_XDR_LIB  "Build HDF4 XDR Library" OFF)
-if (WIN32 AND NOT MINGW)
+if (WIN32)
   set (HDF4_BUILD_XDR_LIB ON CACHE BOOL "Build HDF4 XDR Library" FORCE)
 else ()
   include (FindXDR)
diff -Naur hdf-4.2.13-orig/mfhdf/fortran/mfsdf.c hdf-4.2.13/mfhdf/fortran/mfsdf.c
--- hdf-4.2.13-orig/mfhdf/fortran/mfsdf.c	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/mfhdf/fortran/mfsdf.c	2018-02-14 14:18:43.158693200 +0300
@@ -47,7 +47,7 @@
 FRETVAL(intf) nsfsflmd();
 #endif /* PROTOTYPE */
 
-#if defined _WIN32 && !defined CMAKE_INTDIR
+#if defined _MSC_VER && !defined CMAKE_INTDIR
 
 /*-----------------------------------------------------------------------------
  * Name:    scend
diff -Naur hdf-4.2.13-orig/mfhdf/hdfimport/CMakeLists.txt hdf-4.2.13/mfhdf/hdfimport/CMakeLists.txt
--- hdf-4.2.13-orig/mfhdf/hdfimport/CMakeLists.txt	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/mfhdf/hdfimport/CMakeLists.txt	2018-02-14 14:15:22.137034200 +0300
@@ -4,7 +4,7 @@
 INCLUDE_DIRECTORIES (${HDF4_HDFSOURCE_DIR})
 INCLUDE_DIRECTORIES (${HDF4_MFHDFSOURCE_DIR})
 
-if (WIN32)
+if (MSVC)
   set (CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} /NODEFAULTLIB:LIBCMT")
 endif ()
 
diff -Naur hdf-4.2.13-orig/mfhdf/hdiff/CMakeLists.txt hdf-4.2.13/mfhdf/hdiff/CMakeLists.txt
--- hdf-4.2.13-orig/mfhdf/hdiff/CMakeLists.txt	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/mfhdf/hdiff/CMakeLists.txt	2018-02-14 14:13:01.839865000 +0300
@@ -4,7 +4,7 @@
 INCLUDE_DIRECTORIES (${HDF4_HDFSOURCE_DIR})
 INCLUDE_DIRECTORIES (${HDF4_MFHDFSOURCE_DIR})
 
-if (WIN32)
+if (MSVC)
   set (CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} /NODEFAULTLIB:LIBCMT")
 endif ()
 
diff -Naur hdf-4.2.13-orig/mfhdf/libsrc/local_nc.h hdf-4.2.13/mfhdf/libsrc/local_nc.h
--- hdf-4.2.13-orig/mfhdf/libsrc/local_nc.h	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/mfhdf/libsrc/local_nc.h	2018-02-14 14:20:38.310543800 +0300
@@ -52,8 +52,8 @@
 #endif
 #endif /* __CYGWIN__ */
 
-#include    <rpc/types.h>
-#include    <rpc/xdr.h>
+#include    <../xdr/types.h>
+#include    <../xdr/xdr.h>
 #else    /* NO_SYS_XDR_INC */
 #include      <types.h>  /* <types.h */
 #include      <xdr.h>    /* <xdr.h> */
@@ -400,7 +400,7 @@
 #define nncadel           H4_F77_FUNC(ncadel, NCADEL)
 #define nncsfil           H4_F77_FUNC(ncsfil, NCSFIL)
 
-#ifdef WIN32
+#ifdef _MSC_VER
 HDFFCLIBAPI void nncpopt
     PROTO((int* val));
 HDFFCLIBAPI void nncgopt
diff -Naur hdf-4.2.13-orig/mfhdf/xdr/CMakeLists.txt hdf-4.2.13/mfhdf/xdr/CMakeLists.txt
--- hdf-4.2.13-orig/mfhdf/xdr/CMakeLists.txt	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/mfhdf/xdr/CMakeLists.txt	2018-02-14 14:17:33.189816100 +0300
@@ -17,8 +17,8 @@
 INCLUDE_DIRECTORIES (${HDF4_MFHDF_XDR_DIR})
 
 add_library (${HDF4_MF_XDR_LIB_TARGET} STATIC ${HDF4_MFHDF_XDR_SRCS} ${HDF4_MFHDF_XDR_HDRS})
-if (MSVC OR MINGW)
-  target_link_libraries (${HDF4_MF_XDR_LIB_TARGET} PUBLIC "ws2_32.lib")
+if (WIN32)
+  target_link_libraries (${HDF4_MF_XDR_LIB_TARGET} PUBLIC "ws2_32")
 endif ()
 TARGET_C_PROPERTIES (${HDF4_MF_XDR_LIB_TARGET} STATIC " " " ")
 set_global_variable (HDF4_LIBRARIES_TO_EXPORT "${HDF4_LIBRARIES_TO_EXPORT};${HDF4_MF_XDR_LIB_TARGET}")
@@ -35,8 +35,8 @@
 
 if (BUILD_SHARED_LIBS)
   add_library (${HDF4_MF_XDR_LIBSH_TARGET} SHARED ${HDF4_MFHDF_XDR_SRCS} ${HDF4_MFHDF_XDR_HDRS})
-  if (MSVC OR MINGW)
-    target_link_libraries (${HDF4_MF_XDR_LIBSH_TARGET} PUBLIC "ws2_32.lib")
+  if (WIN32)
+    target_link_libraries (${HDF4_MF_XDR_LIBSH_TARGET} PUBLIC "ws2_32")
   endif ()
   TARGET_C_PROPERTIES (${HDF4_MF_XDR_LIBSH_TARGET} SHARED " " " ")
   set_global_variable (HDF4_LIBRARIES_TO_EXPORT "${HDF4_LIBRARIES_TO_EXPORT};${HDF4_MF_XDR_LIBSH_TARGET}")
diff -Naur hdf-4.2.13-orig/CMakeInstallation.cmake hdf-4.2.13/CMakeInstallation.cmake
--- hdf-4.2.13-orig/CMakeInstallation.cmake	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/CMakeInstallation.cmake	2018-02-14 14:36:33.513117500 +0300
@@ -3,7 +3,7 @@
 #-----------------------------------------------------------------------------
 # Check for Installation Utilities
 #-----------------------------------------------------------------------------
-if (WIN32)
+if (MSVC)
   set (PF_ENV_EXT "(x86)")
   find_program (NSIS_EXECUTABLE NSIS.exe PATHS "$ENV{ProgramFiles}\\NSIS" "$ENV{ProgramFiles${PF_ENV_EXT}}\\NSIS")
   if(NOT CPACK_WIX_ROOT)
@@ -184,7 +184,7 @@
 #-----------------------------------------------------------------------------
 # Configure the COPYING.txt file for the windows binary package
 #-----------------------------------------------------------------------------
-if (WIN32)
+if (MSVC)
   configure_file (${HDF4_SOURCE_DIR}/COPYING ${HDF4_BINARY_DIR}/COPYING.txt @ONLY)
 endif ()
 
diff -Naur hdf-4.2.13-orig/config/cmake_ext_mod/ConfigureChecks.cmake hdf-4.2.13/config/cmake_ext_mod/ConfigureChecks.cmake
--- hdf-4.2.13-orig/config/cmake_ext_mod/ConfigureChecks.cmake	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/config/cmake_ext_mod/ConfigureChecks.cmake	2018-02-14 14:44:48.177061500 +0300
@@ -58,19 +58,17 @@
 
 set (WINDOWS)
 if (WIN32)
+  set (WINDOWS 1)
   if (MINGW)
     set (${HDF_PREFIX}_HAVE_MINGW 1)
-    set (WINDOWS 1) # MinGW tries to imitate Windows
     set (CMAKE_REQUIRED_FLAGS "-DWIN32_LEAN_AND_MEAN=1 -DNOGDI=1")
+    set (CMAKE_REQUIRED_LIBRARIES "ws2_32;wsock32")
   endif ()
   set (${HDF_PREFIX}_HAVE_WIN32_API 1)
-  set (CMAKE_REQUIRED_LIBRARIES "ws2_32.lib;wsock32.lib")
-  if (NOT UNIX AND NOT MINGW)
-    set (WINDOWS 1)
+  if (MSVC)
+    set (CMAKE_REQUIRED_LIBRARIES "ws2_32.lib;wsock32.lib")
     set (CMAKE_REQUIRED_FLAGS "/DWIN32_LEAN_AND_MEAN=1 /DNOGDI=1")
-    if (MSVC)
-      set (${HDF_PREFIX}_HAVE_VISUAL_STUDIO 1)
-    endif ()
+    set (${HDF_PREFIX}_HAVE_VISUAL_STUDIO 1)
   endif ()
 endif ()
 
@@ -82,10 +80,8 @@
   set (${HDF_PREFIX}_HAVE_STRDUP 1)
   set (${HDF_PREFIX}_HAVE_SYSTEM 1)
   set (${HDF_PREFIX}_HAVE_LONGJMP 1)
-  if (NOT MINGW)
-    set (${HDF_PREFIX}_HAVE_GETHOSTNAME 1)
-  endif ()
-  if (NOT UNIX AND NOT CYGWIN AND NOT MINGW)
+  set (${HDF_PREFIX}_HAVE_GETHOSTNAME 1)
+  if (NOT UNIX AND NOT CYGWIN)
     set (${HDF_PREFIX}_HAVE_GETCONSOLESCREENBUFFERINFO 1)
   endif ()
   if (MINGW)
diff -Naur hdf-4.2.13-orig/config/cmake_ext_mod/HDFMacros.cmake hdf-4.2.13/config/cmake_ext_mod/HDFMacros.cmake
--- hdf-4.2.13-orig/config/cmake_ext_mod/HDFMacros.cmake	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/config/cmake_ext_mod/HDFMacros.cmake	2018-02-14 14:51:20.357611800 +0300
@@ -117,7 +117,7 @@
     set (LIB_RELEASE_NAME "${libname}")
     set (LIB_DEBUG_NAME "${libname}${LIB_DEBUG_SUFFIX}")
   else ()
-    if (WIN32)
+    if (MSVC)
       set (LIB_RELEASE_NAME "lib${libname}")
       set (LIB_DEBUG_NAME "lib${libname}${LIB_DEBUG_SUFFIX}")
     else ()
@@ -155,7 +155,7 @@
   endif ()
 
   #----- Use MSVC Naming conventions for Shared Libraries
-  if (MINGW AND ${libtype} MATCHES "SHARED")
+  if (MSVC AND ${libtype} MATCHES "SHARED")
     set_target_properties (${libtarget}
         PROPERTIES
         IMPORT_SUFFIX ".lib"
@@ -178,7 +178,7 @@
 
   if (${libtype} MATCHES "SHARED")
     if (WIN32)
-      if (MINGW)
+      if (MSVC)
         set_target_properties (${libtarget} PROPERTIES
             IMPORTED_IMPLIB "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${IMPORT_LIB_NAME}.lib"
             IMPORTED_LOCATION "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${IMPORT_LIB_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}"
@@ -207,7 +207,7 @@
       endif ()
     endif ()
   else ()
-    if (WIN32 AND NOT MINGW)
+    if (MSVC)
       set_target_properties (${libtarget} PROPERTIES
           IMPORTED_LOCATION "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_BUILD_TYPE}/${IMPORT_LIB_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}"
           IMPORTED_LINK_INTERFACE_LANGUAGES "C"
@@ -236,7 +236,7 @@
 macro (HDF_README_PROPERTIES target_fortran)
   set (BINARY_SYSTEM_NAME ${CMAKE_SYSTEM_NAME})
   set (BINARY_PLATFORM "${CMAKE_SYSTEM_NAME}")
-  if (WIN32)
+  if (MSVC)
     set (BINARY_EXAMPLE_ENDING "zip")
     set (BINARY_INSTALL_ENDING "msi")
     if (CMAKE_CL_64)
diff -Naur hdf-4.2.13-orig/hdf/src/H4api_adpt.h hdf-4.2.13/hdf/src/H4api_adpt.h
--- hdf-4.2.13-orig/hdf/src/H4api_adpt.h	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/hdf/src/H4api_adpt.h	2018-02-14 15:19:28.071305100 +0300
@@ -38,7 +38,7 @@
 
 #if defined(hdf_shared_EXPORTS)
   #if defined (_MSC_VER) || defined(__MINGW32__)  /* MSVC Compiler Case */
-    #define HDFERRPUBLIC __declspec(dllimport)
+    #define HDFERRPUBLIC __declspec(dllexport) extern
     #define HDFPUBLIC __declspec(dllexport)
     #define HDFLIBAPI extern __declspec(dllexport)
     #define HDFFCLIBAPI extern __declspec(dllimport)
diff -Naur hdf-4.2.13-orig/mfhdf/dumper/CMakeLists.txt hdf-4.2.13/mfhdf/dumper/CMakeLists.txt
--- hdf-4.2.13-orig/mfhdf/dumper/CMakeLists.txt	2017-06-25 00:00:04.000000000 +0300
+++ hdf-4.2.13/mfhdf/dumper/CMakeLists.txt	2018-02-14 15:19:28.071305100 +0300
@@ -30,7 +30,7 @@
 endif ()
 TARGET_C_PROPERTIES (hdp STATIC " " " ")
 target_link_libraries (hdp PRIVATE ${HDF4_MF_LIB_TARGET})
-if (WIN32)
+if (MSVC)
   set_target_properties (hdp PROPERTIES LINK_FLAGS "/STACK:10000000")
 endif ()
 set_global_variable (HDF4_UTILS_TO_EXPORT "${HDF4_UTILS_TO_EXPORT};hdp")
