--- hdf-4.2.15/mfhdf/libsrc/CMakeLists.txt.orig	2020-04-27 17:40:17.103987100 +0300
+++ hdf-4.2.15/mfhdf/libsrc/CMakeLists.txt	2020-04-27 17:41:12.828589100 +0300
@@ -160,11 +160,11 @@
 
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf4.pc.in
-    ${HDF4_BINARY_DIR}/CMakeFiles/${HDF4_SRC_LIB_CORENAME}-${HDF4_PACKAGE_VERSION}.pc
+    ${HDF4_BINARY_DIR}/CMakeFiles/${HDF4_SRC_LIB_CORENAME}.pc
     @ONLY
 )
 install (
-    FILES ${HDF4_BINARY_DIR}/CMakeFiles/${HDF4_SRC_LIB_CORENAME}-${HDF4_PACKAGE_VERSION}.pc
+    FILES ${HDF4_BINARY_DIR}/CMakeFiles/${HDF4_SRC_LIB_CORENAME}.pc
     DESTINATION ${HDF4_INSTALL_LIB_DIR}/pkgconfig
     COMPONENT libraries
 )
