--- a/sklearn/_build_utils/openmp_helpers.py
+++ b/sklearn/_build_utils/openmp_helpers.py
@@ -13,7 +13,7 @@
 
 
 def get_openmp_flag():
-    if sys.platform == "win32":
+    if sys.platform == "win32" and 'MSC' in sys.version:
         return ["/openmp"]
     elif sys.platform == "darwin" and "openmp" in os.getenv("CPPFLAGS", ""):
         # -fopenmp can't be passed as compile flag when using Apple-clang.
