--- a/sklearn/_build_utils/openmp_helpers.py
+++ b/sklearn/_build_utils/openmp_helpers.py
@@ -23,7 +23,7 @@
 
     if sys.platform == "win32" and ("icc" in compiler or "icl" in compiler):
         return ["/Qopenmp"]
-    elif sys.platform == "win32":
+    elif sys.platform == "win32" and not 'GCC' in sys.version:
         return ["/openmp"]
     elif sys.platform in ("darwin", "linux") and "icc" in compiler:
         return ["-qopenmp"]
