To prevent rmtree() erroring out because of external file scanners
--- scikit-learn-0.21.3/sklearn/_build_utils/pre_build_helpers.py.orig	2019-08-15 10:23:10.181641800 +0200
+++ scikit-learn-0.21.3/sklearn/_build_utils/pre_build_helpers.py	2019-08-15 10:30:40.014055300 +0200
@@ -7,6 +7,7 @@
 import os
 import sys
 import glob
+import time
 import tempfile
 import textwrap
 import subprocess
@@ -54,6 +55,7 @@
             raise
         finally:
             os.chdir(start_dir)
+            time.sleep(2)
 
     return output
 
