--- a/setup.py
+++ b/setup.py
@@ -575,7 +575,7 @@
             optimization_level = extension.get(
                 "optimization_level", default_optimization_level
             )
-            if os.name == "posix":
+            if not "MSC" in sys.version:
                 extra_compile_args.append(f"-{optimization_level}")
             else:
                 extra_compile_args.append(f"/{optimization_level}")
