diff -ruN GtkAda.orig/src/Makefile.common.in GtkAda/src/Makefile.common.in
--- GtkAda.orig/src/Makefile.common.in	2014-05-03 11:38:47.694069700 +0200
+++ GtkAda/src/Makefile.common.in	2014-06-15 12:30:09.375825900 +0200
@@ -28,11 +28,15 @@
 
 CP = cp -p -f
 ifeq ($(OS),Windows_NT)
-   LN=$(CP)
+   LN=ln -f
    EXEEXT=.exe
+   SOIMPLIB=$(shell basename $(LIBNAME)).a
+   LDSOIMPLIB=-Wl,--out-implib,$(SOIMPLIB)
 else
    LN=ln -f -s
    EXEEXT=
+   SOIMPLIB=
+   LDSOIMPLIB=
 endif
 
 CHMOD = chmod
@@ -114,8 +118,8 @@
 ${OBJ_DIR}/${SONAME}: ${wildcard ${SRC_FILES:%=${OBJ_DIR}/%}}
 	@echo Building ${SONAME}
 	@cd ${OBJ_DIR}; $(CC) -shared $(FPIC) $(OS_SPECIFIC_LINK_OPTIONS) \
-	  -o $(SONAME) $(SO_OPTS)$(SONAME) ${LDFLAGS} ${SRC_FILES} \
-          $(GTKADA_LIBS) $(GTK_LIBS) $(EXTERNAL_LIBS)
+	  -o $(SONAME) $(SO_OPTS)$(SONAME) $(LDSOIMPLIB) ${LDFLAGS} \
+	  ${SRC_FILES} $(GTKADA_LIBS) $(GTK_LIBS) $(EXTERNAL_LIBS)
 	@cd ${OBJ_DIR}; $(RM) $(LIBNAME)
 	@cd ${OBJ_DIR}; $(LN) $(SONAME) $(LIBNAME)
 	@cd ${OBJ_DIR}; ${CHMOD} +x ${LIBNAME}
diff -ruN GtkAda.orig/src/Makefile.in GtkAda/src/Makefile.in
--- GtkAda.orig/src/Makefile.in	2014-05-03 11:38:38.646129600 +0200
+++ GtkAda/src/Makefile.in	2014-06-15 12:29:06.916570100 +0200
@@ -58,6 +58,7 @@
 INSTALL_SCRIPT  = @INSTALL_SCRIPT@
 
 ## General and public targets
+prep: dirs ada ${c_objects_dir}
 
 libs: ada ${c_objects_dir} $(objc_objects_dir)
 ifeq ($(BUILD_SHARED),yes)
@@ -106,6 +107,7 @@
 		if [ "$(OS)" = "Windows_NT" ]; then \
 		     $(INSTALL_DATA) ${OBJ_DIR}/$(SONAME) "${DESTDIR}$(bindir)"; \
 		     $(CHMOD) +x "${DESTDIR}$(bindir)/$(SONAME)"; \
+		     $(INSTALL_DATA) ${OBJ_DIR}/$(SOIMPLIB) "${DESTDIR}$(libdir)"; \
 		else \
 	             $(INSTALL_PROGRAM) ${OBJ_DIR}/$(SONAME) "${DESTDIR}$(alidir)/relocatable"; \
 		     $(RM) "${DESTDIR}$(libdir)/$(LIBNAME)"; \
diff -ruN GtkAda.orig/src/opengl/Makefile.in GtkAda/src/opengl/Makefile.in
--- GtkAda.orig/src/opengl/Makefile.in	2014-05-03 11:38:12.905029300 +0200
+++ GtkAda/src/opengl/Makefile.in	2014-06-15 12:29:06.921560000 +0200
@@ -10,11 +10,8 @@
 OPENGL_OBJECTS   = gtk-glarea.o gdk-gl.o glu_h.o gl_h.o
 OPENGL_C_OBJECTS = gtkglarea.o gdkgl.o
 OPENGL_C_OBJECTS_DIR = ${OPENGL_C_OBJECTS:%=${OBJ_DIR}/%}
-ifeq ($(OS),Windows_NT)
-    GTKADA_LIBS  = ../../lib-obj/libgtkada.dll
-else
-    GTKADA_LIBS  = -L../../lib-obj -lgtkada  # relative to OBJ_DIR
-endif
+GTKADA_LIBS  = -L../../lib-obj -lgtkada  # relative to OBJ_DIR
+
 EXTERNAL_LIBS    = @GL_LIBS@
 OPENGL_FLAGS     = @GL_CFLAGS@
 GPRFILE          = opengl
diff -ruN GtkAda.orig/src/tools/gtkada-config.in GtkAda/src/tools/gtkada-config.in
--- GtkAda.orig/src/tools/gtkada-config.in	2014-05-03 11:38:11.999768500 +0200
+++ GtkAda/src/tools/gtkada-config.in	2014-06-15 12:29:06.925555600 +0200
@@ -9,7 +9,7 @@
 
 # OS Specific relocatable settings
 if [ "$OS" = "Windows_NT" ]; then
-    os_reloc="-L${prefix}/bin -lgtkada-@GTKADA_MAJOR_VERSION@.@GTKADA_MINOR_VERSION@"
+    os_reloc="-L${prefix}/lib -lgtkada"
 else
     os_reloc="-L${libdir}/relocatable -lgtkada"
 fi
