--- numba-0.61.2/numba/__init__.py.orig	2025-04-07 18:34:14.000000000 +0200
+++ numba-0.61.2/numba/__init__.py	2025-06-09 14:11:43.353245400 +0200
@@ -39,8 +39,8 @@
                f"{numpy_version[0]}.{numpy_version[1]}.")
         raise ImportError(msg)
 
-    if numpy_version > (2, 2):
-        msg = (f"Numba needs NumPy 2.2 or less. Got NumPy "
+    if numpy_version > (2, 3):
+        msg = (f"Numba needs NumPy 2.3 or less. Got NumPy "
                f"{numpy_version[0]}.{numpy_version[1]}.")
         raise ImportError(msg)
 
--- numba-0.61.2/setup.py.orig	2025-04-07 18:34:14.000000000 +0200
+++ numba-0.61.2/setup.py	2025-06-09 14:21:25.689433500 +0200
@@ -23,7 +23,7 @@
 max_python_version = "3.14"  # exclusive
 min_numpy_build_version = "2.0.0rc1"
 min_numpy_run_version = "1.24"
-max_numpy_run_version = "2.3"
+max_numpy_run_version = "2.4"
 min_llvmlite_version = "0.44.0dev0"
 max_llvmlite_version = "0.45"
 
