--- numba-0.63.1/numba/__init__.py.orig	2025-12-23 18:42:41.966985400 +0100
+++ numba-0.63.1/numba/__init__.py	2025-12-23 18:42:51.289454200 +0100
@@ -39,8 +39,8 @@
                f"{numpy_version[0]}.{numpy_version[1]}.")
         raise ImportError(msg)
 
-    if numpy_version > (2, 3):
-        msg = (f"Numba needs NumPy 2.3 or less. Got NumPy "
+    if numpy_version > (2, 4):
+        msg = (f"Numba needs NumPy 2.4 or less. Got NumPy "
                 f"{numpy_version[0]}.{numpy_version[1]}.")
         raise ImportError(msg)
 
--- numba-0.63.1/setup.py.orig	2025-12-23 18:42:54.368599100 +0100
+++ numba-0.63.1/setup.py	2025-12-23 18:43:03.060381200 +0100
@@ -23,7 +23,7 @@
 max_python_version = "3.15"  # exclusive
 min_numpy_build_version = "2.0.0rc1"
 min_numpy_run_version = "1.22"
-max_numpy_run_version = "2.4"
+max_numpy_run_version = "2.5"
 min_llvmlite_version = "0.46.0dev0"
 max_llvmlite_version = "0.47"
 
