--- a/numba/__init__.py
+++ b/numba/__init__.py
@@ -39,6 +39,11 @@ def extract_version(mod):
                f"{numpy_version[0]}.{numpy_version[1]}.")
         raise ImportError(msg)
 
+    if numpy_version > (2, 1):
+        msg = (f"Numba needs NumPy 2.1 or less. Got NumPy "
+               f"{numpy_version[0]}.{numpy_version[1]}.")
+        raise ImportError(msg)
+
     try:
         import scipy
     except ImportError:
--- a/setup.py
+++ b/setup.py
@@ -21,8 +21,9 @@
 
 min_python_version = "3.10"
 max_python_version = "3.14"  # exclusive
-min_numpy_build_version = "1.11"
+min_numpy_build_version = "2.0.0rc1"
 min_numpy_run_version = "1.24"
+max_numpy_run_version = "2.2"
 min_llvmlite_version = "0.44.0dev0"
 max_llvmlite_version = "0.45"
 
@@ -366,10 +367,10 @@ def check_file_at_path(path2file):
 
 packages = find_packages(include=["numba", "numba.*"])
 
-build_requires = ['numpy >={}'.format(min_numpy_build_version)]
+build_requires = ['numpy >={},<{}'.format(min_numpy_build_version, max_numpy_run_version)]
 install_requires = [
     'llvmlite >={},<{}'.format(min_llvmlite_version, max_llvmlite_version),
-    'numpy >={}'.format(min_numpy_run_version),
+    'numpy >={},<{}'.format(min_numpy_run_version, max_numpy_run_version),
 ]
 
 metadata = dict(
