--- a/numba/np/ufunc/omppool.cpp.old	2020-08-12 13:00:24.000000000 -0700
+++ a/numba/np/ufunc/omppool.cpp	2020-08-12 16:45:41.073791800 -0700
@@ -15,7 +15,7 @@
 #include "workqueue.h"
 #include "gufunc_scheduler.h"
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 #include <malloc.h>
 #else
 #include <pthread.h>
@@ -28,7 +28,7 @@
 #define _DEBUG_FORK 0
 
 // OpenMP vendor strings
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define _OMP_VENDOR "MS"
 #elif defined(__clang__)
 #define _OMP_VENDOR "Intel"
@@ -42,7 +42,7 @@
 #endif
 
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define THREAD_LOCAL(ty) __declspec(thread) ty
 #else
 /* Non-standard C99 extension that's understood by gcc and clang */
