diff --git a/src/common/mathutil.h b/src/common/mathutil.h
index 560929239..7efbd8ede 100644
--- a/src/common/mathutil.h
+++ b/src/common/mathutil.h
@@ -1027,8 +1027,8 @@ inline uint32_t BitfieldReverse(uint32_t value)
 }
 
 // Count the 1 bits.
-#if defined(_MSC_VER) && !defined(__clang__)
-#    if defined(_M_IX86) || defined(_M_X64)
+#if defined(_WIN32) && !defined(__clang__)
+#    if defined(_M_IX86) || defined(_M_X64) || defined(__X86__) || defined(__x86_64__)
 namespace priv
 {
 // Check POPCNT instruction support and cache the result.
@@ -1071,7 +1071,7 @@ inline int BitCount(uint64_t bits)
 {
     if (priv::kHasPopcnt)
     {
-#        if defined(_M_X64)
+#        if defined(_M_X64) || defined(__x86_64__)
         return static_cast<int>(__popcnt64(bits));
 #        else   // x86
         return static_cast<int>(__popcnt(static_cast<uint32_t>(bits >> 32)) +
diff --git a/src/common/serializer/JsonSerializer.cpp b/src/common/serializer/JsonSerializer.cpp
index 0028331d9..e675c406b 100644
--- a/src/common/serializer/JsonSerializer.cpp
+++ b/src/common/serializer/JsonSerializer.cpp
@@ -173,7 +173,7 @@ void JsonSerializer::addValue(const std::string &name, rapidjson::Value &&value)
     }
     else
     {
-        rapidjson::Value nameValue(name, mAllocator);
+        rapidjson::Value nameValue;
         mDoc.AddMember(nameValue, std::move(value), mAllocator);
     }
 }
diff --git a/util/capture/frame_capture_test_utils.cpp b/util/capture/frame_capture_test_utils.cpp
index 41d52c0d9..a315ecbf2 100644
--- a/util/capture/frame_capture_test_utils.cpp
+++ b/util/capture/frame_capture_test_utils.cpp
@@ -85,7 +85,7 @@ bool LoadTraceInfoFromJSON(const std::string &traceName,
         return false;
     }
 
-    const rapidjson::Document::Object &meta = doc["TraceMetadata"].GetObj();
+    const rapidjson::Document::Object &meta = doc["TraceMetadata"].GetObject();
 
     strncpy(traceInfoOut->name, traceName.c_str(), kTraceInfoMaxNameLen);
     traceInfoOut->contextClientMajorVersion = meta["ContextClientMajorVersion"].GetInt();
