diff --git a/BUILD.gn b/BUILD.gn
index 44954b2d..0e62c356 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -367,7 +367,7 @@ config("spvtools_internal_config") {
       "-Wno-unreachable-code-break",
       "-Wno-unreachable-code-return",
     ]
-  } else if (!is_win) {
+  } else if (!is_msvs) {
     # Work around a false-positive on a Skia GCC 10 builder.
     cflags += [ "-Wno-format-truncation" ]
   } else {
@@ -375,7 +375,7 @@ config("spvtools_internal_config") {
     cflags += [ "/Zc:__cplusplus" ]
   }
 
-  if (!is_win) {
+  if (!is_msvs) {
     cflags += [ "-std=c++17" ]
   } else {
     cflags += [ "/std:c++17" ]
