--- geany-plugins-1.29/geanypy/src/Makefile.am.orig	2017-02-19 00:20:00.175841500 +0300
+++ geany-plugins-1.29/geanypy/src/Makefile.am	2017-02-19 00:20:10.096408900 +0300
@@ -4,7 +4,7 @@
 geanyplugin_LTLIBRARIES		=	geanypy.la
 geanyplugindir				=	$(libdir)/geany
 
-geanypy_la_LDFLAGS			=	-module -avoid-version -Wl,--export-dynamic
+geanypy_la_LDFLAGS			=	-module -avoid-version -no-undefined -Wl,--export-dynamic
 geanypy_la_CPPFLAGS			=	@GEANY_CFLAGS@ @PYGTK_CFLAGS@ @PYTHON_CPPFLAGS@ \
 								-DGEANYPY_PYTHON_DIR="\"$(libdir)/geany/geanypy\"" \
 								-DGEANYPY_PLUGIN_DIR="\"$(libdir)/geany\"" \
