--- a/freetype/raw.py
+++ b/freetype/raw.py
@@ -36,7 +36,7 @@
     # on windows all ctypes does when checking for the library
     # is to append .dll to the end and look for an exact match
     # within any entry in PATH.
-    filename = ctypes.util.find_library('freetype')
+    filename = ctypes.util.find_library('libfreetype-6')
 
     if filename is None:
         if platform.system() == 'Windows':
