--- cfitsio/CMakeLists.txt.orig	2018-05-16 11:30:59.475341100 +0300
+++ cfitsio/CMakeLists.txt	2018-05-16 11:31:06.292553100 +0300
@@ -72,7 +72,7 @@
 ENDIF()
 
 # Support for remote file drivers is not implemented for native Windows:
-IF (NOT MSVC)
+IF (NOT WIN32)
     # Find library needed for gethostbyname:
     CHECK_FUNCTION_EXISTS("gethostbyname" CMAKE_HAVE_GETHOSTBYNAME)
     IF(NOT CMAKE_HAVE_GETHOSTBYNAME)
