From dcc5a269e16525b119c76a06bf3abbaa41af5080 Mon Sep 17 00:00:00 2001
From: kreijstal <kreijstal@hotmail.com>
Date: Wed, 8 May 2024 19:27:28 +0200
Subject: [PATCH] make it more friendly for mingw enviroments

---
 CMakeLists.txt  | 2 +-
 python/setup.py | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d3ecfc..fbd9140 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ if (SPM_CROSS_SYSTEM_PROCESSOR)
 endif()
 
 # Disable shared build on windows
-if(WIN32)
+if(MSVC)
   set(SPM_ENABLE_SHARED OFF)
 endif()
 
diff --git a/python/setup.py b/python/setup.py
index d600321..d0a1552 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -102,8 +102,7 @@ class build_ext(_build_ext):
     ext.extra_link_args = libs
     _build_ext.build_extension(self, ext)
 
-
-if os.name == 'nt':
+if os.name == 'nt' and not os.environ.get('SENTENCEPIECE_MINGW_SKIP'):
   # Must pre-install sentencepice into build directory.
   arch = 'win32'
   if sys.maxsize > 2**32:
-- 
2.45.0

