From dcc5a269e16525b119c76a06bf3abbaa41af5080 Mon Sep 17 00:00:00 2001
From: kreijstal <kreijstal@hotmail.com>
Date: Wed, 8 May 2024 19:27:28 +0200
Subject: [PATCH] make it more friendly for mingw enviroments

---
 python/setup.py | 3 +--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/python/setup.py b/python/setup.py
index d600321..d0a1552 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -102,13 +102,16 @@
     ext.extra_link_args = libs
     _build_ext.build_extension(self, ext)

-
 if os.name == 'nt':
+
   # Must pre-install sentencepice into build directory.
   arch = 'win32'
   if sys.maxsize > 2**32:
     arch = 'amd64'
-  if os.path.exists('..\\build\\root_{}\\lib'.format(arch)):
+  if True:
+    cflags = ['-I../build/root/include', '-L../build/root/bin']
+    libs =['-L../build/root/bin','-lsentencepiece','-lsentencepiece_train']
+  elif os.path.exists('..\\build\\root_{}\\lib'.format(arch)):
     cflags = ['/std:c++17', '/I..\\build\\root_{}\\include'.format(arch)]
     libs = [
         '..\\build\\root_{}\\lib\\sentencepiece.lib'.format(arch),
2.45.0

