diff --git a/config/BUILD.gn b/config/BUILD.gn
index d0517299b..9679a9fd0 100644
--- a/config/BUILD.gn
+++ b/config/BUILD.gn
@@ -127,13 +127,13 @@ config("debug") {
     defines += [ "DYNAMIC_ANNOTATIONS_PREFIX=NACL_" ]
   }

-  if (is_win) {
+  if (is_win && !is_mingw) {
     if (!enable_iterator_debugging && !use_custom_libcxx) {
       # Iterator debugging is enabled by default by the compiler on debug
       # builds, and we have to tell it to turn it off.
       defines += [ "_HAS_ITERATOR_DEBUGGING=0" ]
     }
-  } else if (is_linux && current_cpu == "x64" && enable_iterator_debugging) {
+  } else if ((is_linux || is_mingw) && current_cpu == "x64" && enable_iterator_debugging) {
     # Enable libstdc++ debugging facilities to help catch problems early, see
     # http://crbug.com/65151 .
     # TODO(phajdan.jr): Should we enable this for all of POSIX?
@@ -167,7 +167,7 @@ config("release") {

 # This config defines the default libraries applied to all targets.
 config("default_libs") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     # TODO(brettw) this list of defaults should probably be smaller, and
     # instead the targets that use the less common ones (e.g. wininet or
     # winspool) should include those explicitly.
@@ -297,8 +297,13 @@ group("shared_library_deps") {

 # Windows linker setup for EXEs and DLLs.
 if (is_win) {
-  _windows_linker_configs = [
-    "//build/config/win:sdk_link",
+  _windows_linker_configs = []
+  if(!is_mingw){
+    _windows_linker_configs += [
+      "//build/config/win:sdk_link",
+    ]
+  }
+  _windows_linker_configs += [
     "//build/config/win:common_linker_setup",
   ]
 }
diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 11e0a2182..1b8f1fe04 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -133,8 +133,12 @@ declare_args() {
   # Whether we're a traditional desktop unix.
   is_desktop_linux = current_os == "linux"

+  # Set to true when compiling with the MinGW GCC compiler on the MSYS2 environment.
+  is_mingw = getenv("MSYSTEM") == "MINGW64" || getenv("MSYSTEM") == "MINGW32"
+
   # Set to true when compiling with the Clang compiler.
-  is_clang = current_os != "linux" ||
+  is_clang = current_os != "linux" ||
+            !(getenv("MSYSTEM") == "MINGW64" || getenv("MSYSTEM") == "MINGW32") ||
              (current_cpu != "s390x" && current_cpu != "s390" &&
               current_cpu != "ppc64" && current_cpu != "ppc" &&
               current_cpu != "mips" && current_cpu != "mips64")
@@ -203,11 +207,15 @@ if (host_toolchain == "") {
     if (target_cpu == "x86" || target_cpu == "x64") {
       if (is_clang) {
         host_toolchain = "//build/toolchain/win:win_clang_$target_cpu"
+      } else if (is_mingw) {
+        host_toolchain = "//build/toolchain/win:mingw_$target_cpu"
       } else {
         host_toolchain = "//build/toolchain/win:$target_cpu"
       }
     } else if (is_clang) {
       host_toolchain = "//build/toolchain/win:win_clang_$host_cpu"
+    } else if (is_mingw) {
+      host_toolchain = "//build/toolchain/win:mingw_$host_cpu"
     } else {
       host_toolchain = "//build/toolchain/win:$host_cpu"
     }
@@ -243,6 +251,8 @@ if (target_os == "android") {
   # Beware, win cross builds have some caveats, see docs/win_cross.md
   if (is_clang) {
     _default_toolchain = "//build/toolchain/win:win_clang_$target_cpu"
+  } else if (is_mingw) {
+    _default_toolchain = "//build/toolchain/win:mingw_$target_cpu"
   } else {
     _default_toolchain = "//build/toolchain/win:$target_cpu"
   }
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 5fa34150a..e5f0cffa5 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -269,7 +269,7 @@ config("compiler") {

   # In general, Windows is totally different, but all the other builds share
   # some common GCC configuration.
-  if (!is_win) {
+  if (!is_win || is_mingw) {
     # Common POSIX compiler flags setup.
     # --------------------------------
     cflags += [ "-fno-strict-aliasing" ]  # See http://crbug.com/32204
@@ -283,7 +283,7 @@ config("compiler") {
       } else {
         cflags += [ "-fstack-protector" ]
       }
-    } else if ((is_posix && !is_chromeos && !is_nacl) || is_fuchsia) {
+    } else if ((is_posix && !is_chromeos && !is_nacl) || is_fuchsia || is_mingw) {
       # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
       # See also https://crbug.com/533294
       cflags += [ "--param=ssp-buffer-size=4" ]
@@ -294,6 +294,12 @@ config("compiler") {
       } else if (current_os != "aix") {
         # Not available on aix.
         cflags += [ "-fstack-protector" ]
+        if(is_mingw) {
+          # In MingW's GCC -fstack-protector
+          # needs to be passed to the linker as well
+          # for it to link against the ssp library
+          ldflags += [ "-fstack-protector" ]
+        }
       }
     }

@@ -320,7 +326,7 @@ config("compiler") {

   # Non-Mac Posix and Fuchsia compiler flags setup.
   # -----------------------------------
-  if ((is_posix && !(is_mac || is_ios)) || is_fuchsia) {
+  if ((is_posix && !(is_mac || is_ios)) || is_fuchsia || is_mingw) {
     if (enable_profiling) {
       if (!is_debug) {
         cflags += [ "-g" ]
@@ -476,7 +482,7 @@ config("compiler") {
     # TODO(thakis): Make the driver pass --color-diagnostics to the linker
     # if -fcolor-diagnostics is passed to it, and pass -fcolor-diagnostics
     # in ldflags instead.
-    if (is_win) {
+    if (is_win && !is_mingw) {
       # On Windows, we call the linker directly, instead of calling it through
       # the driver.
       ldflags += [ "--color-diagnostics" ]
@@ -501,7 +507,7 @@ config("compiler") {

   # C11/C++11 compiler flags setup.
   # ---------------------------
-  if (is_linux || is_android || (is_nacl && is_clang) || current_os == "aix") {
+  if (is_linux || is_android || is_mingw || (is_nacl && is_clang) || current_os == "aix") {
     if (target_os == "android") {
       cxx11_override = use_cxx11_on_android
     } else {
@@ -587,7 +593,7 @@ config("compiler") {
       lto_opt_level = 0
     }

-    if (is_win) {
+    if (is_win && !is_mingw) {
       # This is a straight translation of the non-Windows flags below,
       # except we do not use the ThinLTO cache, which leaks temporary
       # files on Windows (https://crbug.com/871962).
@@ -646,7 +652,7 @@ config("compiler") {
     # built with ThinLTO (see https://crbug.com/767901) as well as the GVR shim.
     if (!is_android) {
       cflags += [ "-fwhole-program-vtables" ]
-      if (!is_win) {
+      if (!is_win || is_mingw) {
         ldflags += [ "-fwhole-program-vtables" ]
       }
     }
@@ -668,12 +674,12 @@ config("compiler") {

   if (compiler_timing) {
     if (is_clang && !is_nacl) {
-      if (is_win) {
+      if (is_win && !is_mingw) {
         cflags += [ "/clang:-ftime-trace" ]
       } else {
         cflags += [ "-ftime-trace" ]
       }
-    } else if (is_win) {
+    } else if (is_win && !is_mingw) {
       cflags += [
         # "Documented" here:
         # http://aras-p.info/blog/2017/10/23/Best-unknown-MSVC-flag-d2cgsummary/
@@ -705,7 +711,7 @@ config("compiler") {
   # Assign any flags set for the C compiler to asmflags so that they are sent
   # to the assembler. The Windows assembler takes different types of flags
   # so only do so for posix platforms.
-  if (is_posix || is_fuchsia) {
+  if (is_posix || is_fuchsia || is_mingw) {
     asmflags += cflags
     asmflags += cflags_c
   }
@@ -721,7 +727,7 @@ config("compiler_cpu_abi") {
   ldflags = []
   defines = []

-  if ((is_posix && !(is_mac || is_ios)) || is_fuchsia) {
+  if ((is_posix && !(is_mac || is_ios)) || is_fuchsia || is_mingw) {
     # CPU architecture. We may or may not be doing a cross compile now, so for
     # simplicity we always explicitly set the architecture.
     if (current_cpu == "x64") {
@@ -1075,7 +1081,7 @@ config("compiler_deterministic") {
   # Eliminate build metadata (__DATE__, __TIME__ and __TIMESTAMP__) for
   # deterministic build.  See https://crbug.com/314403
   if (!is_official_build) {
-    if (is_win && !is_clang) {
+    if (is_win && !is_clang && !is_mingw) {
       cflags += [
         "/wd4117",  # Trying to define or undefine a predefined macro.
         "/D__DATE__=",
@@ -1236,7 +1242,7 @@ config("default_warnings") {
   cflags_cc = []
   ldflags = []

-  if (is_win) {
+  if (is_win && !is_mingw) {
     if (treat_warnings_as_errors) {
       cflags += [ "/WX" ]
     }
@@ -1446,15 +1452,31 @@ config("default_warnings") {
       # files.
       cflags += [ "-Wno-packed-not-aligned" ]
     }
+
+    if (is_mingw) {
+      cflags += [
+        "-Wno-attributes",           # "__decspec(dllimport) inline"
+        "-Wno-format",               # PRIu64 llu support on MinGW
+        "-Wno-unused-but-set-variable",
+        "-Wno-stringop-overflow",    # False positive overflow error on gcc 10
+
+      ]
+      cflags_cc += [
+        "-Wno-subobject-linkage",    # Tests having code in header files
+      ]
+    }
   }

   # Common Clang and GCC warning setup.
-  if (!is_win || is_clang) {
+  if (!is_win || is_clang || is_mingw) {
     cflags += [
       # Disables.
       "-Wno-missing-field-initializers",  # "struct foo f = {0};"
       "-Wno-unused-parameter",  # Unused function parameters.
     ]
+    if(is_clang){
+      cflags += ["-Wno-range-loop-construct"] # Use of "conditionally-supported" offsetof in c++17
+    }
   }

   if (is_clang) {
@@ -1537,7 +1559,7 @@ config("default_warnings") {
 # part of Chromium.

 config("chromium_code") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     cflags = [ "/W4" ]  # Warning level 4.

     if (is_clang) {
@@ -1632,7 +1654,7 @@ config("no_chromium_code") {
   cflags_cc = []
   defines = []

-  if (is_win) {
+  if (is_win && !is_mingw) {
     cflags += [
       "/W3",  # Warning level 3.
       "/wd4800",  # Disable warning when forcing value to bool.
@@ -1691,7 +1713,7 @@ config("noshadowing") {
 # Allows turning Run-Time Type Identification on or off.

 config("rtti") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     cflags_cc = [ "/GR" ]
   } else {
     cflags_cc = [ "-frtti" ]
@@ -1701,7 +1723,7 @@ config("rtti") {
 config("no_rtti") {
   # Some sanitizer configs may require RTTI to be left enabled globally
   if (!use_rtti) {
-    if (is_win) {
+    if (is_win && !is_mingw) {
       cflags_cc = [ "/GR-" ]
     } else {
       cflags_cc = [ "-fno-rtti" ]
@@ -1754,7 +1776,7 @@ config("thin_archive") {
 # Note: exceptions are disallowed in Google code.

 config("exceptions") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     # Enables exceptions in the STL.
     if (!use_custom_libcxx) {
       defines = [ "_HAS_EXCEPTIONS=1" ]
@@ -1767,7 +1789,7 @@ config("exceptions") {
 }

 config("no_exceptions") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     # Disables exceptions in the STL.
     # libc++ uses the __has_feature macro to control whether to use exceptions,
     # so defining this macro is unnecessary. Defining _HAS_EXCEPTIONS to 0 also
@@ -1828,7 +1850,7 @@ config("no_shorten_64_warnings") {
     if (is_clang) {
       cflags = [ "-Wno-shorten-64-to-32" ]
     } else {
-      if (is_win) {
+      if (is_win && !is_mingw) {
         # MSVC does not have an explicit warning equivalent to
         # -Wshorten-64-to-32 but 4267 warns for size_t -> int
         # on 64-bit builds, so is the closest.
@@ -1868,12 +1890,22 @@ config("no_incompatible_pointer_warnings") {

 # Shared settings for both "optimize" and "optimize_max" configs.
 # IMPORTANT: On Windows "/O1" and "/O2" must go before the common flags.
-if (is_win) {
-  common_optimize_on_cflags = [
+if (is_win && !is_mingw) {
+  common_optimize_on_cflags = []
+  if(is_clang) {
+    common_optimize_on_cflags += [
     "/Ob2",  # Both explicit and auto inlining.
+    ]
+  } else {
+  common_optimize_on_cflags += [
+    "/Ob3",  # Both explicit and auto inlining.
+    ]
+  }
+  common_optimize_on_cflags += [
     "/Oy-",  # Disable omitting frame pointers, must be after /O2.
     "/Zc:inline",  # Remove unreferenced COMDAT (faster links).
   ]
+
   if (!is_asan) {
     common_optimize_on_cflags += [
       # Put data in separate COMDATs. This allows the linker
@@ -1976,7 +2008,7 @@ config("default_stack_frames") {

 # Default "optimization on" config.
 config("optimize") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     # Favor size over speed, /O1 must be before the common flags.
     # /O1 implies /Os and /GF.
     cflags = [ "/O1" ] + common_optimize_on_cflags + [ "/Oi" ]
@@ -1997,7 +2029,7 @@ config("optimize") {

 # Turn off optimizations.
 config("no_optimize") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     cflags = [
       "/Od",  # Disable optimization.
       "/Ob0",  # Disable all inlining (on by default).
@@ -2043,7 +2075,7 @@ config("optimize_max") {
     configs = [ "//build/config/nacl:irt_optimize" ]
   } else {
     ldflags = common_optimize_on_ldflags
-    if (is_win) {
+    if (is_win && !is_mingw) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
       cflags = [ "/O2" ] + common_optimize_on_cflags
@@ -2075,7 +2107,7 @@ config("optimize_speed") {
     configs = [ "//build/config/nacl:irt_optimize" ]
   } else {
     ldflags = common_optimize_on_ldflags
-    if (is_win) {
+    if (is_win && !is_mingw) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
       cflags = [ "/O2" ] + common_optimize_on_cflags
@@ -2204,7 +2236,7 @@ config("win_pdbaltpath") {

 # Full symbols.
 config("symbols") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     if (is_clang) {
       cflags = [ "/Z7" ]  # Debug information in the .obj files.
     } else {
@@ -2310,7 +2342,7 @@ config("symbols") {
 # This config guarantees to hold symbol for stack trace which are shown to user
 # when crash happens in unittests running on buildbot.
 config("minimal_symbols") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     # Functions, files, and line tables only.
     cflags = []
     ldflags = [ "/DEBUG" ]
@@ -2367,7 +2399,7 @@ config("minimal_symbols") {
 # told to not generate debug information and the linker then just puts function
 # names in the final debug information.
 config("no_symbols") {
-  if (is_win) {
+  if (is_win && !is_mingw) {
     ldflags = [ "/DEBUG" ]

     # All configs using /DEBUG should include this:
diff --git a/config/linux/pkg-config.py b/config/linux/pkg-config.py
index 5adf70cc3..76899e01c 100644
--- a/config/linux/pkg-config.py
+++ b/config/linux/pkg-config.py
@@ -109,7 +109,8 @@ def main():
   # If this is run on non-Linux platforms, just return nothing and indicate
   # success. This allows us to "kind of emulate" a Linux build from other
   # platforms.
-  if "linux" not in sys.platform:
+  # Update: MinGW builds do support pkg-config, and MSVC vcpkg does as well.
+  if "linux" not in sys.platform and not ("GCC" in sys.version and sys.platform=='win32') :
     print("[[],[],[],[],[]]")
     return 0

@@ -193,7 +194,6 @@ def main():
   cmd = [options.pkg_config, "--cflags", "--libs"] + args
   if options.debug:
     sys.stderr.write('Running: %s\n' % ' '.join(cmd))
-
   try:
     flag_string = subprocess.check_output(cmd).decode('utf-8')
   except:
diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
index 7b44f0e43..95afdb5ce 100644
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -45,7 +45,7 @@ declare_args() {
 # is applied to all targets. It is here to separate out the logic that is
 # Windows-only.
 config("compiler") {
-  if (current_cpu == "x86") {
+  if (current_cpu == "x86" && !is_mingw) {
     asmflags = [
       # When /safeseh is specified, the linker will only produce an image if it
       # can also produce a table of the image's safe exception handlers. This
@@ -56,27 +56,36 @@ config("compiler") {
     ]
   }

-  cflags = [
-    "/Gy",  # Enable function-level linking.
-    "/FS",  # Preserve previous PDB behavior.
-    "/bigobj",  # Some of our files are bigger than the regular limits.
-    "/utf-8",  # Assume UTF-8 by default to avoid code page dependencies.
-  ]
-
+  if(!is_mingw) {
+    cflags = [
+      "/Gy",  # Enable function-level linking.
+      "/FS",  # Preserve previous PDB behavior.
+      "/bigobj",  # Some of our files are bigger than the regular limits.
+      "/utf-8",  # Assume UTF-8 by default to avoid code page dependencies.
+    ]
+  } else {
+    cflags = [
+      "-Wa,-mbig-obj",
+      "-fno-keep-inline-dllexport"
+    ]
+  }
+
   if (is_clang) {
     cflags += [ "/Zc:twoPhase" ]
   }

-  # Force C/C++ mode for the given GN detected file type. This is necessary
-  # for precompiled headers where the same source file is compiled in both
-  # modes.
-  cflags_c = [ "/TC" ]
-  cflags_cc = [ "/TP" ]
+  if(!is_mingw) {
+    # Force C/C++ mode for the given GN detected file type. This is necessary
+    # for precompiled headers where the same source file is compiled in both
+    # modes.
+    cflags_c = [ "/TC" ]
+    cflags_cc = [ "/TP" ]

-  cflags += [
-    # Work around crbug.com/526851, bug in VS 2015 RTM compiler.
-    "/Zc:sizedDealloc-",
-  ]
+    cflags += [
+      # Work around crbug.com/526851, bug in VS 2015 RTM compiler.
+      "/Zc:sizedDealloc-",
+    ]
+  }

   if (is_clang) {
     # Don't look for includes in %INCLUDE%.
@@ -150,7 +159,7 @@ config("compiler") {
     ldflags += [ "/lldignoreenv" ]
   }

-  if (!is_debug && !is_component_build) {
+  if (!is_debug && !is_component_build && !is_mingw) {
     # Enable standard linker optimizations like GC (/OPT:REF) and ICF in static
     # release builds.
     # Release builds always want these optimizations, so enable them explicitly.
@@ -175,17 +184,19 @@ config("compiler") {
     ldflags += [ "/PROFILE" ]
   }

-  # arflags apply only to static_libraries. The normal linker configs are only
-  # set for executable and shared library targets so arflags must be set
-  # elsewhere. Since this is relatively contained, we just apply them in this
-  # more general config and they will only have an effect on static libraries.
-  arflags = [
-    # "No public symbols found; archive member will be inaccessible." This
-    # means that one or more object files in the library can never be
-    # pulled in to targets that link to this library. It's just a warning that
-    # the source file is a no-op.
-    "/ignore:4221",
-  ]
+  if(!is_mingw){
+    # arflags apply only to static_libraries. The normal linker configs are only
+    # set for executable and shared library targets so arflags must be set
+    # elsewhere. Since this is relatively contained, we just apply them in this
+    # more general config and they will only have an effect on static libraries.
+    arflags = [
+      # "No public symbols found; archive member will be inaccessible." This
+      # means that one or more object files in the library can never be
+      # pulled in to targets that link to this library. It's just a warning that
+      # the source file is a no-op.
+      "/ignore:4221",
+    ]
+  }
 }

 # This is included by reference in the //build/config/compiler:runtime_library
@@ -324,19 +335,27 @@ config("sdk_link") {
 # targets who want different library configurations can remove this and specify
 # their own.
 config("common_linker_setup") {
-  ldflags = [
-    "/FIXED:NO",
-    "/ignore:4199",
-    "/ignore:4221",
-    "/NXCOMPAT",
-    "/DYNAMICBASE",
-  ]
-
-  if (win_linker_timing) {
-    ldflags += [
-      "/time",
-      "/verbose:incr",
+  if(is_mingw) {
+      # Enable DEP and ASLR
+    ldflags = [
+      "-Wl,-dynamicbase",
+      "-Wl,-nxcompat",
+     ]
+  } else {
+    ldflags = [
+      "/FIXED:NO",
+      "/ignore:4199",
+      "/ignore:4221",
+      "/NXCOMPAT",
+      "/DYNAMICBASE",
     ]
+
+    if (win_linker_timing) {
+      ldflags += [
+        "/time",
+        "/verbose:incr",
+      ]
+    }
   }
 }

@@ -442,20 +461,22 @@ config("delayloads_not_for_child_dll") {
 # See https://msdn.microsoft.com/en-us/library/2kzt1wy3.aspx for a reference of
 # what each value does.
 config("default_crt") {
-  if (is_component_build) {
-    # Component mode: dynamic CRT. Since the library is shared, it requires
-    # exceptions or will give errors about things not matching, so keep
-    # exceptions on.
-    configs = [ ":dynamic_crt" ]
-  } else {
-    if (current_os == "winuwp") {
-      # https://blogs.msdn.microsoft.com/vcblog/2014/06/10/the-great-c-runtime-crt-refactoring/
-      # contains a details explanation of what is happening with the Windows
-      # CRT in Visual Studio releases related to Windows store applications.
+  if(!is_mingw){
+    if (is_component_build) {
+      # Component mode: dynamic CRT. Since the library is shared, it requires
+      # exceptions or will give errors about things not matching, so keep
+      # exceptions on.
       configs = [ ":dynamic_crt" ]
     } else {
-      # Desktop Windows: static CRT.
-      configs = [ ":static_crt" ]
+      if (current_os == "winuwp") {
+        # https://blogs.msdn.microsoft.com/vcblog/2014/06/10/the-great-c-runtime-crt-refactoring/
+        # contains a details explanation of what is happening with the Windows
+        # CRT in Visual Studio releases related to Windows store applications.
+        configs = [ ":dynamic_crt" ]
+      } else {
+        # Desktop Windows: static CRT.
+        configs = [ ":static_crt" ]
+      }
     }
   }
 }
@@ -496,42 +517,54 @@ config("static_crt") {
 if (current_cpu == "x64") {
   # The number after the comma is the minimum required OS version.
   # 5.02 = Windows Server 2003.
-  subsystem_version_suffix = ",5.02"
+  subsystem_version_suffix = "5.02"
 } else if (current_cpu == "arm64") {
   # Windows ARM64 requires Windows 10.
-  subsystem_version_suffix = ",10.0"
+  subsystem_version_suffix = "10.0"
 } else {
   # 5.01 = Windows XP.
-  subsystem_version_suffix = ",5.01"
+  subsystem_version_suffix = "5.01"
 }

 config("console") {
-  ldflags = [ "/SUBSYSTEM:CONSOLE$subsystem_version_suffix" ]
+  if(is_mingw){
+    cflags = [ "-mconsole" ]
+    ldflags = [ "-Wl,--subsystem,console:$subsystem_version_suffix" ]
+  } else {
+    ldflags = [ "/SUBSYSTEM:CONSOLE,$subsystem_version_suffix" ]
+  }
 }
 config("windowed") {
-  ldflags = [ "/SUBSYSTEM:WINDOWS$subsystem_version_suffix" ]
+  if(is_mingw){
+    cflags = [ "-mwindows" ]
+    ldflags = [ "-Wl,--subsystem,windows:$subsystem_version_suffix" ]
+  } else {
+    ldflags = [ "/SUBSYSTEM:WINDOWS,$subsystem_version_suffix" ]
+  }
 }

 # Incremental linking ----------------------------------------------------------

 # Applies incremental linking or not depending on the current configuration.
 config("default_incremental_linking") {
-  # Enable incremental linking for debug builds and all component builds - any
-  # builds where performance is not job one.
-  # TODO(thakis): Always turn this on with lld, no reason not to.
-  if (is_debug || is_component_build) {
-    ldflags = [ "/INCREMENTAL" ]
-    if (use_lld) {
-      # lld doesn't use ilk files and doesn't really have an incremental link
-      # mode; the only effect of the flag is that the .lib file timestamp isn't
-      # updated if the .lib doesn't change.
-      # TODO(thakis): Why pass /OPT:NOREF for lld, but not otherwise?
-      # TODO(thakis): /INCREMENTAL is on by default in link.exe, but not in
-      # lld.
-      ldflags += [ "/OPT:NOREF" ]
+  if(!is_mingw){
+    # Enable incremental linking for debug builds and all component builds - any
+    # builds where performance is not job one.
+    # TODO(thakis): Always turn this on with lld, no reason not to.
+    if (is_debug || is_component_build) {
+      ldflags = [ "/INCREMENTAL" ]
+      if (use_lld) {
+        # lld doesn't use ilk files and doesn't really have an incremental link
+        # mode; the only effect of the flag is that the .lib file timestamp isn't
+        # updated if the .lib doesn't change.
+        # TODO(thakis): Why pass /OPT:NOREF for lld, but not otherwise?
+        # TODO(thakis): /INCREMENTAL is on by default in link.exe, but not in
+        # lld.
+        ldflags += [ "/OPT:NOREF" ]
+      }
+    } else {
+      ldflags = [ "/INCREMENTAL:NO" ]
     }
-  } else {
-    ldflags = [ "/INCREMENTAL:NO" ]
   }
 }

@@ -543,6 +576,9 @@ config("unicode") {
     "_UNICODE",
     "UNICODE",
   ]
+  if(is_mingw) {
+    cflags = [ "-municode" ]
+  }
 }

 # Lean and mean ----------------------------------------------------------------
@@ -581,3 +617,10 @@ config("cygprofile_instrumentation") {
     "-finstrument-functions-after-inlining",
   ]
 }
+
+# Let unit tests see all "for_testing" dll symbols.
+config("export_all_symbols") {
+  if(is_win && is_mingw){
+    ldflags = [ "-Wl,--export-all-symbols" ]
+  }
+}
diff --git a/config/win/manifest.gni b/config/win/manifest.gni
index b18a4a141..4eaf3510f 100644
--- a/config/win/manifest.gni
+++ b/config/win/manifest.gni
@@ -79,12 +79,14 @@ if (is_win) {
       foreach(i, rebase_path(invoker.sources, root_build_dir)) {
         manifests += [ "/manifestinput:" + i ]
       }
-      ldflags = [
-                  "/manifest:embed",
+      if(!is_mingw){
+        ldflags = [
+                    "/manifest:embed",

-                  # We handle UAC by adding explicit .manifest files instead.
-                  "/manifestuac:no",
-                ] + manifests
+                    # We handle UAC by adding explicit .manifest files instead.
+                    "/manifestuac:no",
+                  ] + manifests
+      }
     }

     # This source set only exists to add a dep on the invoker's deps and to
diff --git a/config/x64.gni b/config/x64.gni
index eead1e1b1..1315ce58b 100644
--- a/config/x64.gni
+++ b/config/x64.gni
@@ -16,7 +16,7 @@ if (current_cpu == "x64") {
     x64_arch = ""
   }

-  if ((is_posix && !(is_mac || is_ios)) || is_fuchsia) {
+  if ((is_posix && !(is_mac || is_ios)) || is_fuchsia || is_mingw) {
     if (x64_arch == "") {
       x64_arch = "x86-64"
     }
diff --git a/toolchain/gcc_link_wrapper.py b/toolchain/gcc_link_wrapper.py
index 8892f14bf..22e56dd6e 100644
--- a/toolchain/gcc_link_wrapper.py
+++ b/toolchain/gcc_link_wrapper.py
@@ -38,6 +38,9 @@ def main():
   parser.add_argument('--strip',
                       help='The strip binary to run',
                       metavar='PATH')
+  parser.add_argument('--objcopy',
+                      help='The objcopy binary to run',
+                      metavar='PATH')
   parser.add_argument('--unstripped-file',
                       help='Executable file produced by linking command',
                       metavar='FILE')
@@ -63,9 +66,19 @@ def main():

   # Finally, strip the linked executable (if desired).
   if args.strip:
+    if args.objcopy:
+      result = subprocess.call(wrapper_utils.CommandToRun(
+          [args.objcopy, '--only-keep-debug', args.unstripped_file, args.output + '.debug']))
+      if result != 0:
+        return result
     result = subprocess.call(CommandToRun([
         args.strip, '-o', args.output, args.unstripped_file
         ]))
+    if result != 0:
+      return result
+    if args.objcopy:
+      result = subprocess.call(wrapper_utils.CommandToRun(
+          [args.objcopy, '--add-gnu-debuglink', args.output + '.debug',args.output]))

   return result

diff --git a/toolchain/gcc_solink_wrapper.py b/toolchain/gcc_solink_wrapper.py
index 540b1eda3..587f942cb 100644
--- a/toolchain/gcc_solink_wrapper.py
+++ b/toolchain/gcc_solink_wrapper.py
@@ -17,10 +17,11 @@ import sys

 import wrapper_utils

-
 def CollectSONAME(args):
   """Replaces: readelf -d $sofile | grep SONAME"""
   toc = ''
+  if ("GCC" in sys.version and sys.platform=='win32'): # Mingw's readelf doesn't work on PE files
+    return 0, toc
   readelf = subprocess.Popen(wrapper_utils.CommandToRun(
       [args.readelf, '-d', args.sofile]), stdout=subprocess.PIPE, bufsize=-1)
   for line in readelf.stdout:
@@ -32,6 +33,8 @@ def CollectSONAME(args):
 def CollectDynSym(args):
   """Replaces: nm --format=posix -g -D $sofile | cut -f1-2 -d' '"""
   toc = ''
+  if ("GCC" in sys.version and sys.platform=='win32'): # Mingw's nm doesn't work on PE files
+    return 0, toc
   nm = subprocess.Popen(wrapper_utils.CommandToRun([
       args.nm, '--format=posix', '-g', '-D', args.sofile]),
                         stdout=subprocess.PIPE, bufsize=-1)
@@ -70,6 +73,9 @@ def main():
   parser.add_argument('--strip',
                       help='The strip binary to run',
                       metavar='PATH')
+  parser.add_argument('--objcopy',
+                      help='The objcopy binary to run',
+                      metavar='PATH')
   parser.add_argument('--sofile',
                       required=True,
                       help='Shared object file produced by linking command',
@@ -141,11 +147,20 @@ def main():

   # Finally, strip the linked shared object file (if desired).
   if args.strip:
+    if args.objcopy:
+      result = subprocess.call(wrapper_utils.CommandToRun(
+          [args.objcopy, '--only-keep-debug', args.sofile, args.output + '.debug']))
+      if result != 0:
+        return result
     result = subprocess.call(wrapper_utils.CommandToRun(
         [args.strip, '-o', args.output, args.sofile]))
+    if result != 0:
+      return result
+    if args.objcopy:
+      result = subprocess.call(wrapper_utils.CommandToRun(
+          [args.objcopy, '--add-gnu-debuglink', args.output + '.debug',args.output]))

   return result

-
 if __name__ == "__main__":
   sys.exit(main())
diff --git a/toolchain/gcc_toolchain.gni b/toolchain/gcc_toolchain.gni
index 744c78cbd..0d33e1912 100644
--- a/toolchain/gcc_toolchain.gni
+++ b/toolchain/gcc_toolchain.gni
@@ -44,6 +44,7 @@ if (is_linux && target_os == "android") {
 #  - cc
 #  - cxx
 #  - ld
+#  - rc (mingw only)
 #
 # Optional parameters that control the tools:
 #
@@ -96,7 +97,9 @@ template("gcc_toolchain") {
     assert(defined(invoker.cc), "gcc_toolchain() must specify a \"cc\" value")
     assert(defined(invoker.cxx), "gcc_toolchain() must specify a \"cxx\" value")
     assert(defined(invoker.ld), "gcc_toolchain() must specify a \"ld\" value")
-
+    if(is_mingw){
+      assert(defined(invoker.rc), "gcc_toolchain() must specify a \"rc\" value")
+    }
     # This define changes when the toolchain changes, forcing a rebuild.
     # Nothing should ever use this define.
     if (defined(invoker.rebuild_define)) {
@@ -227,17 +230,22 @@ template("gcc_toolchain") {
     asm = asm_prefix + invoker.cc
     ar = invoker.ar
     ld = "$goma_ld${invoker.ld}"
-    if (defined(invoker.readelf)) {
-      readelf = invoker.readelf
-    } else {
-      readelf = "readelf"
-    }
-    if (defined(invoker.nm)) {
-      nm = invoker.nm
+    if(!is_mingw){
+      if (defined(invoker.readelf)) {
+        readelf = invoker.readelf
+      } else {
+        readelf = "readelf"
+      }
+      if (defined(invoker.nm)) {
+        nm = invoker.nm
+      } else {
+        nm = "nm"
+      }
     } else {
-      nm = "nm"
+      rc = invoker.rc
     }

+
     if (defined(invoker.shlib_extension)) {
       default_shlib_extension = invoker.shlib_extension
     } else {
@@ -315,6 +323,16 @@ template("gcc_toolchain") {
       outputs = [ "$object_subdir/{{source_name_part}}.o" ]
     }

+    if(is_mingw){
+      tool("rc") {
+        depfile = "{{output}}.d"
+        command = "$rc -i {{source}} -o {{output}}"
+        depsformat = "gcc"
+        description = "RC {{output}}"
+        outputs = [ "$object_subdir/{{source_name_part}}.o" ]
+      }
+    }
+
     tool("asm") {
       # For GCC we can just use the C compiler to compile assembly.
       depfile = "{{output}}.d"
@@ -375,8 +393,6 @@ template("gcc_toolchain") {
       # .TOC file, overwrite it, otherwise, don't change it.
       tocfile = sofile + ".TOC"

-      link_command = "$ld -shared -Wl,-soname=\"$soname\" {{ldflags}}${extra_ldflags} -o \"$unstripped_sofile\" @\"$rspfile\""
-
       # Generate a map file to be used for binary size analysis.
       # Map file adds ~10% to the link time on a z620.
       # With target_os="android", libchrome.so.map.gz is ~20MB.
@@ -386,19 +402,41 @@ template("gcc_toolchain") {
         map_switch = " --map-file \"$map_file\""
       }

-      assert(defined(readelf), "to solink you must have a readelf")
-      assert(defined(nm), "to solink you must have an nm")
       strip_switch = ""
       if (defined(invoker.strip)) {
         strip_switch = "--strip=${invoker.strip} "
       }

+      objcopy_switch = ""
+      if (defined(invoker.objcopy)) {
+        objcopy_switch = "--objcopy=${invoker.objcopy} "
+      }
+
+      mingw_flags = ""
+      if(is_mingw){
+        # Have MinGW linker generate a .def file and a .a import library
+        mingw_flags = " -Wl,--dll -Wl,--output-def=\"$sofile\".def -Wl,--out-implib=\"$sofile\".a "
+        readelf = ""
+        if (defined(invoker.readelf)) {
+          readelf = "${invoker.readelf} "
+        }
+        nm = ""
+        if (defined(invoker.nm)) {
+          readelf = "${invoker.nm} "
+        }
+      } else {
+        assert(defined(readelf), "to solink you must have a readelf")
+        assert(defined(nm), "to solink you must have an nm")
+      }
+
+      link_command = "$ld -shared -Wl,-soname=\"$soname\" ${mingw_flags} {{ldflags}}${extra_ldflags} -o \"$unstripped_sofile\" @\"$rspfile\""
+
       # This needs a Python script to avoid using a complex shell command
       # requiring sh control structures, pipelines, and POSIX utilities.
       # The host might not have a POSIX shell and utilities (e.g. Windows).
       solink_wrapper =
           rebase_path("//build/toolchain/gcc_solink_wrapper.py", root_build_dir)
-      command = "$python_path \"$solink_wrapper\" --readelf=\"$readelf\" --nm=\"$nm\" $strip_switch--sofile=\"$unstripped_sofile\" --tocfile=\"$tocfile\"$map_switch --output=\"$sofile\" -- $link_command"
+      command = "$python_path \"$solink_wrapper\" --readelf=\"$readelf\" --nm=\"$nm\" $objcopy_switch$strip_switch--sofile=\"$unstripped_sofile\" --tocfile=\"$tocfile\"$map_switch --output=\"$sofile\" -- $link_command"

       if (target_cpu == "mipsel" && is_component_build && is_android) {
         rspfile_content = "-Wl,--start-group -Wl,--whole-archive {{inputs}} {{solibs}} -Wl,--no-whole-archive {{libs}} -Wl,--end-group"
@@ -527,11 +565,15 @@ template("gcc_toolchain") {
       if (defined(invoker.strip)) {
         strip_switch = " --strip=\"${invoker.strip}\" --unstripped-file=\"$unstripped_outfile\""
       }
+      objcopy_switch = ""
+      if (defined(invoker.objcopy)) {
+        objcopy_switch = " --objcopy=${invoker.objcopy} "
+      }

       if (strip_switch != "" || map_switch != "") {
         link_wrapper =
             rebase_path("//build/toolchain/gcc_link_wrapper.py", root_build_dir)
-        command = "$python_path \"$link_wrapper\" --output=\"$outfile\"$strip_switch$map_switch -- $link_command"
+        command = "$python_path \"$link_wrapper\" --output=\"$outfile\"$objcopy_switch$strip_switch$map_switch -- $link_command"
       } else {
         command = link_command
       }
diff --git a/toolchain/win/BUILD.gn b/toolchain/win/BUILD.gn
index 0c356e162..e1c7cdb22 100644
--- a/toolchain/win/BUILD.gn
+++ b/toolchain/win/BUILD.gn
@@ -9,6 +9,7 @@ import("//build/config/win/visual_studio_version.gni")
 import("//build/toolchain/cc_wrapper.gni")
 import("//build/toolchain/goma.gni")
 import("//build/toolchain/toolchain.gni")
+import("//build/toolchain/gcc_toolchain.gni")

 # Should only be running on Windows.
 assert(is_win)
@@ -46,13 +47,15 @@ if (current_toolchain == default_toolchain) {
   } else {
     configuration_name = "Release"
   }
-  exec_script("../../vs_toolchain.py",
-              [
-                "copy_dlls",
-                rebase_path(root_build_dir),
-                configuration_name,
-                target_cpu,
-              ])
+  if(!is_mingw){
+    exec_script("../../vs_toolchain.py",
+                [
+                  "copy_dlls",
+                  rebase_path(root_build_dir),
+                  configuration_name,
+                  target_cpu,
+                ])
+  }
 }

 if (host_os == "win") {
@@ -520,3 +523,30 @@ if (target_os == "winuwp") {
     }
   }
 }
+
+template("mingw_toolchain") {
+  gcc_toolchain("mingw_" + target_name) {
+    forward_variables_from(invoker, "*")
+    cc = "gcc"
+    cxx = "g++"
+    ar = "ar"
+    ld = cxx
+    rc = "windres"
+    strip = "strip"
+    if(is_debug){
+      objcopy = "objcopy"
+    }
+    executable_extension = ".exe"
+    toolchain_args = {
+      current_cpu = target_name
+      current_os = "win"
+      is_clang = false
+    }
+  }
+}
+
+mingw_toolchain("x64") {
+}
+
+mingw_toolchain("x86") {
+}
diff --git a/util/lastchange.py b/util/lastchange.py
index 874870ad5..a4fc0be8d 100644
--- a/util/lastchange.py
+++ b/util/lastchange.py
@@ -191,7 +191,10 @@ def GetGitTopDirectory(source_dir):
   Returns:
     The output of "git rev-parse --show-toplevel" as a string
   """
-  return _RunGitCommand(source_dir, ['rev-parse', '--show-toplevel'])
+  directory = _RunGitCommand(source_dir, ['rev-parse', '--show-toplevel'])
+  if "GCC" in sys.version and sys.platform=='win32':
+    return subprocess.check_output(["cygpath", "-w", directory]).strip(b"\n").decode()
+  return directory


 def WriteIfChanged(file_name, contents):
diff --git a/win/BUILD.gn b/win/BUILD.gn
index 9be769fd0..587281dd1 100644
--- a/win/BUILD.gn
+++ b/win/BUILD.gn
@@ -16,7 +16,7 @@ windows_manifest("default_exe_manifest") {
   ]
 }

-if (is_win) {
+if (is_win && !is_mingw) {
   assert(host_os != "mac" || target_cpu != "x86",
          "Windows cross-builds from Mac must be 64-bit.")

