--- v8-3.27.34.6/src/platform-win32.cc.orig	2014-07-20 00:07:18.152835800 +0100
+++ v8-3.27.34.6/src/platform-win32.cc	2014-07-20 00:07:48.888738700 +0100
@@ -1146,6 +1146,16 @@
 }
 
 
+#else  // __MINGW32__
+std::vector<OS::SharedLibraryAddress> OS::GetSharedLibraryAddresses() {
+  return std::vector<OS::SharedLibraryAddress>();
+}
+
+
+void OS::SignalCodeMovingGC() { }
+#endif  // __MINGW32__
+
+
 uint64_t OS::TotalPhysicalMemory() {
   MEMORYSTATUSEX memory_info;
   memory_info.dwLength = sizeof(memory_info);
@@ -1158,16 +1168,6 @@
 }
 
 
-#else  // __MINGW32__
-std::vector<OS::SharedLibraryAddress> OS::GetSharedLibraryAddresses() {
-  return std::vector<OS::SharedLibraryAddress>();
-}
-
-
-void OS::SignalCodeMovingGC() { }
-#endif  // __MINGW32__
-
-
 int OS::NumberOfProcessorsOnline() {
   SYSTEM_INFO info;
   GetSystemInfo(&info);
