--- v8-3.27.34.6/tools/gyp/v8.gyp	2014-07-20 16:51:42.314149400 +0100
+++ v8-3.27.34.6/tools/gyp/v8.gyp.new	2014-07-20 16:48:06.251212900 +0100
@@ -94,6 +94,7 @@
               'conditions': [
                 ['OS=="win"', {
                   'product_extension': '<(soname_version).dll',
+                  'ldflags': ['-Wl,--out-implib,<(PRODUCT_DIR)/libv8.<(soname_version).dll.a']
                 }, {
                   'product_extension': 'so.<(soname_version)',
                 }],
