--- v8-git-mirror-4.2.47/build/standalone.gypi.orig	2015-02-03 03:26:35.000000000 +0000
+++ v8-git-mirror-4.2.47/build/standalone.gypi	2015-02-13 11:40:44.605828900 +0000
@@ -364,6 +364,9 @@
               '_HAS_EXCEPTIONS=0',
             ],
           }],
+          ['gcc_version!="unknown"', {
+            'cflags_cc': [ '-Wnon-virtual-dtor', '-fno-rtti', '-std=gnu++0x' ],
+          }],
         ],
         'msvs_cygwin_dirs': ['<(DEPTH)/third_party/cygwin'],
         'msvs_disabled_warnings': [4355, 4800],
