--- v8-git-mirror-4.2.47/src/base/platform/platform-win32.cc.orig	2015-02-13 12:32:02.012827200 +0000
+++ v8-git-mirror-4.2.47/src/base/platform/platform-win32.cc	2015-02-13 12:33:09.326827200 +0000
@@ -42,9 +42,6 @@
   __asm__ __volatile__("xchgl %%eax,%0 ":"=r" (barrier));
 }
 
-#endif  // __MINGW64_VERSION_MAJOR
-
-
 int localtime_s(tm* out_tm, const time_t* time) {
   tm* posix_local_time_struct = localtime(time);
   if (posix_local_time_struct == NULL) return 1;
@@ -52,6 +49,7 @@
   return 0;
 }
 
+#endif  // __MINGW64_VERSION_MAJOR
 
 int fopen_s(FILE** pFile, const char* filename, const char* mode) {
   *pFile = fopen(filename, mode);
