diff --git a/include/v8config.h b/include/v8config.h
index 33bb3f9..546830d 100644
--- a/include/v8config.h
+++ b/include/v8config.h
@@ -340,6 +340,13 @@ path. Add it with -I<path> to the command line
 
 #if defined(__GNUC__)  // Clang in gcc mode.
 # define V8_CC_GNU 1
+# if defined(__MINGW32__)
+#  define V8_CC_MINGW32 1
+# endif
+# if defined(__MINGW64__)
+#  define V8_CC_MINGW64 1
+# endif
+# define V8_CC_MINGW (V8_CC_MINGW32 || V8_CC_MINGW64)
 #endif
 
 # define V8_HAS_ATTRIBUTE_ALWAYS_INLINE (__has_attribute(always_inline))
@@ -749,7 +756,7 @@ V8 shared library set USING_V8_SHARED.
 #elif defined(_M_IX86) || defined(__i386__)
 #define V8_HOST_ARCH_IA32 1
 #define V8_HOST_ARCH_32_BIT 1
-#elif defined(__AARCH64EL__) || defined(_M_ARM64)
+#elif defined(__AARCH64EL__) || defined(_M_ARM64) || defined(__aarch64__)
 #define V8_HOST_ARCH_ARM64 1
 #define V8_HOST_ARCH_64_BIT 1
 #elif defined(__ARMEL__)
@@ -802,7 +809,7 @@ V8 shared library set USING_V8_SHARED.
 #define V8_TARGET_ARCH_X64 1
 #elif defined(_M_IX86) || defined(__i386__)
 #define V8_TARGET_ARCH_IA32 1
-#elif defined(__AARCH64EL__) || defined(_M_ARM64)
+#elif defined(__AARCH64EL__) || defined(_M_ARM64) || defined(__aarch64__)
 #define V8_TARGET_ARCH_ARM64 1
 #elif defined(__ARMEL__)
 #define V8_TARGET_ARCH_ARM 1
 