diff --git a/src/base/platform/time.cc b/src/base/platform/time.cc
index c2d36cc2..0b724fcc 100644
--- a/src/base/platform/time.cc
+++ b/src/base/platform/time.cc
@@ -802,6 +802,8 @@ ThreadTicks ThreadTicks::Now() {
   UNREACHABLE();
 #elif V8_OS_DARWIN
   return ThreadTicks(ComputeThreadTicks());
+#elif V8_OS_WIN
+  return ThreadTicks::GetForThread(::GetCurrentThread());
 #elif V8_OS_FUCHSIA
   return ThreadTicks(GetFuchsiaThreadTicks());
 #elif (defined(_POSIX_THREAD_CPUTIME) && (_POSIX_THREAD_CPUTIME >= 0)) || \
@@ -809,8 +811,6 @@ ThreadTicks ThreadTicks::Now() {
   return ThreadTicks(ClockNow(CLOCK_THREAD_CPUTIME_ID));
 #elif V8_OS_SOLARIS
   return ThreadTicks(gethrvtime() / Time::kNanosecondsPerMicrosecond);
-#elif V8_OS_WIN
-  return ThreadTicks::GetForThread(::GetCurrentThread());
 #else
   UNREACHABLE();
 #endif
