diff -ruN --ex v8-14.6.202.26/src/base/numbers/diy-fp.h v8-work4/src/base/numbers/diy-fp.h
--- a/src/base/numbers/diy-fp.h	2026-03-20 22:50:53
+++ b/src/base/numbers/diy-fp.h	2026-04-10 20:26:59
@@ -45,7 +45,7 @@
   }
 
   // this = this * other.
-  V8_BASE_EXPORT void Multiply(const DiyFp& other) {
+  void Multiply(const DiyFp& other) {
     *this = Times(*this, other);
   }
 
diff -ruN --ex v8-14.6.202.26/src/codegen/external-reference.h v8-work4/src/codegen/external-reference.h
--- a/src/codegen/external-reference.h	2026-03-20 22:50:53
+++ b/src/codegen/external-reference.h	2026-04-10 20:27:25
@@ -637,7 +637,7 @@
   template <typename SubjectChar, typename PatternChar>
   static ExternalReference search_string_raw();
 
-  V8_EXPORT_PRIVATE static ExternalReference FromRawAddress(Address address) {
+  static ExternalReference FromRawAddress(Address address) {
     // Use unchecked constructor.
     return ExternalReference{address, nullptr};
   }
diff -ruN --ex v8-14.6.202.26/src/compiler/wasm-call-descriptors.h v8-work4/src/compiler/wasm-call-descriptors.h
--- a/src/compiler/wasm-call-descriptors.h	2026-03-20 22:50:53
+++ b/src/compiler/wasm-call-descriptors.h	2026-04-10 20:27:32
@@ -36,7 +36,7 @@
   V8_EXPORT_PRIVATE compiler::CallDescriptor* GetLoweredCallDescriptor(
       const compiler::CallDescriptor* original);
 #else
-  V8_EXPORT_PRIVATE compiler::CallDescriptor* GetLoweredCallDescriptor(
+  compiler::CallDescriptor* GetLoweredCallDescriptor(
       const compiler::CallDescriptor* original) {
     UNREACHABLE();
   }
diff -ruN --ex v8-14.6.202.26/src/execution/interrupts-scope.h v8-work4/src/execution/interrupts-scope.h
--- a/src/execution/interrupts-scope.h	2026-03-20 22:50:53
+++ b/src/execution/interrupts-scope.h	2026-04-10 20:26:59
@@ -19,7 +19,7 @@
  public:
   enum Mode : uint8_t { kPostponeInterrupts, kRunInterrupts, kNoop };
 
-  V8_EXPORT_PRIVATE InterruptsScope(Isolate* isolate, uint32_t intercept_mask,
+  InterruptsScope(Isolate* isolate, uint32_t intercept_mask,
                                     Mode mode)
       : stack_guard_(nullptr),
         intercept_mask_(intercept_mask),
diff -ruN --ex v8-14.6.202.26/src/heap/conservative-stack-visitor.h v8-work4/src/heap/conservative-stack-visitor.h
--- a/src/heap/conservative-stack-visitor.h	2026-03-20 22:50:53
+++ b/src/heap/conservative-stack-visitor.h	2026-04-10 20:26:59
@@ -33,7 +33,7 @@
 // 5) OnlyScanMainV8Heap() - returns true if the visitor does not handle the
 // external code and trusted spaces.
 template <typename ConcreteVisitor>
-class V8_EXPORT_PRIVATE ConservativeStackVisitorBase
+class ConservativeStackVisitorBase
     : public ::heap::base::StackVisitor {
  public:
   ConservativeStackVisitorBase(Isolate* isolate, RootVisitor* root_visitor);
diff -ruN --ex v8-14.6.202.26/src/logging/runtime-call-stats.h v8-work4/src/logging/runtime-call-stats.h
--- a/src/logging/runtime-call-stats.h	2026-03-20 22:50:53
+++ b/src/logging/runtime-call-stats.h	2026-04-10 20:27:15
@@ -831,7 +831,7 @@
 class RuntimeCallStats {
  public:
   enum ThreadType { kMainIsolateThread, kWorkerThread };
-  explicit V8_EXPORT_PRIVATE RuntimeCallStats(ThreadType thread_type) {}
+  explicit RuntimeCallStats(ThreadType thread_type) {}
 };
 
 class WorkerThreadRuntimeCallStatsScope {
diff -ruN --ex v8-14.6.202.26/src/objects/js-objects.h v8-work4/src/objects/js-objects.h
--- a/src/objects/js-objects.h	2026-03-20 22:50:53
+++ b/src/objects/js-objects.h	2026-04-10 20:26:59
@@ -751,7 +751,7 @@
       PropertyNormalizationMode mode, int expected_additional_properties,
       bool use_cache, const char* reason);
 
-  V8_EXPORT_PRIVATE static void NormalizeProperties(
+  static void NormalizeProperties(
       Isolate* isolate, DirectHandle<JSObject> object,
       PropertyNormalizationMode mode, int expected_additional_properties,
       const char* reason) {
diff -ruN --ex v8-14.6.202.26/src/objects/map.h v8-work4/src/objects/map.h
--- a/src/objects/map.h	2026-03-20 22:50:53
+++ b/src/objects/map.h	2026-04-10 20:26:59
@@ -607,18 +607,18 @@
       InternalIndex descriptor_number, PropertyConstness constness,
       DirectHandle<Object> value);
 
-  V8_EXPORT_PRIVATE static Handle<Map> Normalize(
+  static Handle<Map> Normalize(
       Isolate* isolate, DirectHandle<Map> map, InstanceType new_instance_type,
       ElementsKind new_elements_kind, DirectHandle<JSPrototype> new_prototype,
       PropertyNormalizationMode mode, bool use_cache, const char* reason);
-  V8_EXPORT_PRIVATE static Handle<Map> Normalize(
+  static Handle<Map> Normalize(
       Isolate* isolate, DirectHandle<Map> map, ElementsKind new_elements_kind,
       DirectHandle<JSPrototype> new_prototype, PropertyNormalizationMode mode,
       bool use_cache, const char* reason) {
     return Normalize(isolate, map, map->instance_type(), new_elements_kind,
                      new_prototype, mode, use_cache, reason);
   }
-  V8_EXPORT_PRIVATE static Handle<Map> Normalize(
+  static Handle<Map> Normalize(
       Isolate* isolate, DirectHandle<Map> map, InstanceType new_instance_type,
       ElementsKind new_elements_kind, DirectHandle<JSPrototype> new_prototype,
       PropertyNormalizationMode mode, const char* reason) {
@@ -626,7 +626,7 @@
     return Normalize(isolate, map, new_instance_type, new_elements_kind,
                      new_prototype, mode, kUseCache, reason);
   }
-  V8_EXPORT_PRIVATE static Handle<Map> Normalize(
+  static Handle<Map> Normalize(
       Isolate* isolate, DirectHandle<Map> map, ElementsKind new_elements_kind,
       DirectHandle<JSPrototype> new_prototype, PropertyNormalizationMode mode,
       const char* reason) {
diff -ruN --ex v8-14.6.202.26/src/objects/string-inl.h v8-work4/src/objects/string-inl.h
--- a/src/objects/string-inl.h	2026-03-20 22:50:53
+++ b/src/objects/string-inl.h	2026-04-10 20:26:59
@@ -848,7 +848,7 @@
 // static
 template <template <typename> typename HandleType>
   requires(std::is_convertible_v<HandleType<String>, DirectHandle<String>>)
-V8_EXPORT_PRIVATE HandleType<String> String::SlowFlatten(
+HandleType<String> String::SlowFlatten(
     Isolate* isolate, HandleType<ConsString> cons, AllocationType allocation) {
   DCHECK(!cons->IsFlat());
   DCHECK_NE(cons->second()->length(), 0);  // Equivalent to !IsFlat.
diff -ruN --ex v8-14.6.202.26/src/utils/address-map.h v8-work4/src/utils/address-map.h
--- a/src/utils/address-map.h	2026-03-20 22:50:53
+++ b/src/utils/address-map.h	2026-04-10 20:26:59
@@ -61,7 +61,7 @@
   RootIndexMap& operator=(const RootIndexMap&) = delete;
 
   // Returns true on successful lookup and sets *|out_root_list|.
-  V8_EXPORT_PRIVATE bool Lookup(Tagged<HeapObject> obj,
+  bool Lookup(Tagged<HeapObject> obj,
                                 RootIndex* out_root_list) const {
     Maybe<uint32_t> maybe_index = map_->Get(obj);
     if (maybe_index.IsJust()) {
