--- a/third_party/googletest/BUILD.gn
+++ b/third_party/googletest/BUILD.gn
@@ -32,7 +32,7 @@
     "src/googletest/include",
   ]
 
-  if (is_win) {
+  if (is_msvs) {
     cflags = [ "/wd4800" ]  # Unused variable warning.
   }
 }
--- a/third_party/jsoncpp/BUILD.gn
+++ b/third_party/jsoncpp/BUILD.gn
@@ -12,7 +12,7 @@
   # declaration flag.
   # This temporary flag allowing clients to update to the new version, and then
   # update to the new StreamWriter and CharReader classes.
-  if (!is_win || is_clang) {
+  if (!is_msvs || is_clang) {
     cflags_cc = [ "-Wno-deprecated-declarations" ]
   }
 }
@@ -42,7 +42,7 @@
 
   include_dirs = [ "source/src/lib_json" ]
 
-  if (!is_win || is_clang) {
+  if (!is_msvs || is_clang) {
     cflags_cc = [ "-Wno-implicit-fallthrough" ]
   }
 }
--- a/third_party/zlib/google/BUILD.gn
+++ b/third_party/zlib/google/BUILD.gn
@@ -44,5 +44,8 @@
     "compression_utils_portable.cc",
     "compression_utils_portable.h",
   ]
+  defines = [ 
+    "USE_SYSTEM_ZLIB=1"
+  ]
   public_deps = [ "//third_party/zlib" ]
 }
