diff --git a/src/base/macros.h b/src/base/macros.h
index 25a533c..5591276 100644
--- a/src/base/macros.h
+++ b/src/base/macros.h
@@ -7,6 +7,7 @@
 
 #include <limits>
 #include <type_traits>
+#include <cstdint>
 
 #include "src/base/compiler-specific.h"
 #include "src/base/logging.h"
diff --git a/src/base/logging.h b/src/base/logging.h
index 4c129ab..600c00b 100644
--- a/src/base/logging.h
+++ b/src/base/logging.h
@@ -9,6 +9,7 @@
 #include <cstring>
 #include <sstream>
 #include <string>
+#include <cstdint>
 
 #include "src/base/base-export.h"
 #include "src/base/build_config.h"
