diff --git a/BUILD.gn b/BUILD.gn
index 9d4cb6a..be3cf21 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -20,39 +20,8 @@ config("absl_component_build") {
 assert(!is_nacl || is_nacl_saigo,
        "base must not be built in most nacl toolchains")
 
-component("absl") {
+static_library("absl") {
   public_deps = [ ":absl_component_deps" ]
-  if (is_component_build) {
-    public_configs = [ ":absl_component_build" ]
-
-    if (is_win && is_clang) {
-      if (current_cpu == "x64") {
-        if (is_debug) {
-          sources = [ "symbols_x64_dbg.def" ]
-        } else {
-          if (is_asan) {
-            sources = [ "symbols_x64_rel_asan.def" ]
-          } else {
-            sources = [ "symbols_x64_rel.def" ]
-          }
-        }
-      }
-      if (current_cpu == "x86") {
-        if (is_debug) {
-          sources = [ "symbols_x86_dbg.def" ]
-        } else {
-          sources = [ "symbols_x86_rel.def" ]
-        }
-      }
-      if (current_cpu == "arm64") {
-        if (is_debug) {
-          sources = [ "symbols_arm64_dbg.def" ]
-        } else {
-          sources = [ "symbols_arm64_rel.def" ]
-        }
-      }
-    }
-  }
 }
 
 group("absl_component_deps") {
diff --git a/absl.gni b/absl.gni
index cf8a39e..5fc078d 100644
--- a/absl.gni
+++ b/absl.gni
@@ -30,14 +30,6 @@ template("absl_source_set") {
       "//third_party/abseil-cpp:absl_define_config",
     ]
 
-    if (is_component_build) {
-      defines = [ "ABSL_BUILD_DLL" ]
-      if (!is_win) {
-        configs -= [ "//build/config/gcc:symbol_visibility_hidden" ]
-        configs += [ "//build/config/gcc:symbol_visibility_default" ]
-      }
-    }
-
     if (!defined(public_configs)) {
       public_configs = []
     }
diff --git a/absl/base/BUILD.gn b/absl/base/BUILD.gn
index 800dfd6..a3f1dc9 100644
--- a/absl/base/BUILD.gn
+++ b/absl/base/BUILD.gn
@@ -178,6 +178,9 @@ absl_source_set("base") {
     ":spinlock_wait",
     "//third_party/abseil-cpp/absl/meta:type_traits",
   ]
+  if (is_mingw) {
+    libs = [ "pthread" ]
+  }
 }
 
 absl_source_set("throw_delegate") {
diff --git a/absl/base/internal/thread_identity.h b/absl/base/internal/thread_identity.h
index b6e917c..8b298c6 100644
--- a/absl/base/internal/thread_identity.h
+++ b/absl/base/internal/thread_identity.h
@@ -20,7 +20,7 @@
 #ifndef ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 #define ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 #include <pthread.h>
 // Defines __GOOGLE_GRTE_VERSION__ (via glibc-specific features.h) when
 // supported.
diff --git a/absl/random/internal/BUILD.gn b/absl/random/internal/BUILD.gn
index 54754f7..0e74fb5 100644
--- a/absl/random/internal/BUILD.gn
+++ b/absl/random/internal/BUILD.gn
@@ -39,7 +39,7 @@ absl_source_set("seed_material") {
   sources = [ "seed_material.cc" ]
   if (is_win) {
     # TODO(mbonadei): In the bazel file this is -DEFAULTLIB:bcrypt.lib.
-    libs = [ "bcrypt.lib" ]
+    libs = [ "bcrypt" ]
   }
   deps = [
     ":fast_uniform_bits",
diff --git a/absl/time/internal/cctz/src/time_zone_lookup.cc b/absl/time/internal/cctz/src/time_zone_lookup.cc
index d22691b..a7e4fa7 100644
--- a/absl/time/internal/cctz/src/time_zone_lookup.cc
+++ b/absl/time/internal/cctz/src/time_zone_lookup.cc
@@ -43,7 +43,7 @@
 // MinGW did not add it until NTDDI_WIN10_NI (SDK version 10.0.22621.0).
 #if ((defined(_WIN32_WINNT_WIN10) && !defined(__MINGW32__)) ||        \
      (defined(NTDDI_WIN10_NI) && NTDDI_VERSION >= NTDDI_WIN10_NI)) && \
-    (_WIN32_WINNT >= _WIN32_WINNT_WINXP)
+    (_WIN32_WINNT >= _WIN32_WINNT_WINXP) && !defined(__MINGW32__)
 #define USE_WIN32_LOCAL_TIME_ZONE
 #include <roapi.h>
 #include <tchar.h>

