diff -ruN --ex abseil-new/BUILD.gn abseil-work2/BUILD.gn
--- a/BUILD.gn	2026-04-09 17:23:39
+++ b/BUILD.gn	2026-04-09 18:39:24
@@ -17,37 +17,8 @@
   defines = [ "ABSL_CONSUME_DLL" ]
 }
 
-component("absl") {
+static_library("absl") {
   public_deps = [ ":absl_component_deps" ]
-  if (is_component_build) {
-    public_configs = [ ":absl_component_build" ]
-
-    if (is_win && is_clang) {
-      file = "symbols_${current_cpu}"
-      if (is_debug) {
-        file = "${file}_dbg"
-      } else {
-        file = "${file}_rel"
-      }
-      if (!is_debug && current_cpu == "x64" && is_asan) {
-        file = "${file}_asan"
-      }
-      if (use_cxx23) {
-        file = "${file}_cxx23"
-      }
-      file = "${file}.def"
-      sources = [ file ]
-    }
-  }
-  if (is_win) {
-    all_dependent_configs = [ "//tools/win/DebugVisualizers:absl" ]
-    inputs = [
-      # absl.natvis listed as an input here instead of in
-      # //tools/win/DebugVisualizers:absl to prevent unnecessary size increase
-      # in generated build files.
-      "//tools/win/DebugVisualizers/absl.natvis",
-    ]
-  }
 }
 
 group("absl_component_deps") {
diff -ruN --ex abseil-new/absl/base/BUILD.gn abseil-work2/absl/base/BUILD.gn
--- a/absl/base/BUILD.gn	2026-04-09 17:23:39
+++ b/absl/base/BUILD.gn	2026-04-09 18:39:24
@@ -194,6 +194,9 @@
     ":spinlock_wait",
     "//third_party/abseil-cpp/absl/meta:type_traits",
   ]
+  if (is_mingw) {
+    libs = [ ":libpthread.a" ]
+  }
 }
 
 absl_test("attributes_test") {
diff -ruN --ex abseil-new/absl/base/internal/thread_identity.h abseil-work2/absl/base/internal/thread_identity.h
--- a/absl/base/internal/thread_identity.h	2026-04-09 17:23:39
+++ b/absl/base/internal/thread_identity.h	2026-04-09 18:39:24
@@ -20,7 +20,7 @@
 #ifndef ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 #define ABSL_BASE_INTERNAL_THREAD_IDENTITY_H_
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 #include <pthread.h>
 // Defines __GOOGLE_GRTE_VERSION__ (via glibc-specific features.h) when
 // supported.
diff -ruN --ex abseil-new/absl/random/internal/BUILD.gn abseil-work2/absl/random/internal/BUILD.gn
--- a/absl/random/internal/BUILD.gn	2026-04-09 17:23:39
+++ b/absl/random/internal/BUILD.gn	2026-04-09 18:39:24
@@ -40,7 +40,7 @@
   sources = [ "seed_material.cc" ]
   if (is_win) {
     # TODO(mbonadei): In the bazel file this is -DEFAULTLIB:bcrypt.lib.
-    libs = [ "bcrypt.lib" ]
+    libs = [ "bcrypt" ]
   }
   deps = [
     ":fast_uniform_bits",
diff -ruN --ex abseil-new/absl/time/internal/cctz/src/time_zone_name_win.cc abseil-work2/absl/time/internal/cctz/src/time_zone_name_win.cc
--- a/absl/time/internal/cctz/src/time_zone_name_win.cc	2026-04-09 17:23:39
+++ b/absl/time/internal/cctz/src/time_zone_name_win.cc	2026-04-09 18:39:24
@@ -20,6 +20,7 @@
 #define NOMINMAX
 #endif  // !defined(NOMINMAX)
 #include <windows.h>
+#include <timezoneapi.h>
 
 #include <algorithm>
 #include <atomic>
diff -ruN --ex abseil-new/absl.gni abseil-work2/absl.gni
--- a/absl.gni	2026-04-09 17:23:37
+++ b/absl.gni	2026-04-09 18:39:24
@@ -40,7 +40,7 @@
       if (!defined(defines)) {
         defines = []
       }
-      if (is_component_build) {
+      if (!is_mingw && is_component_build) {
         defines += [ "ABSL_BUILD_DLL" ]
         if (!is_win && current_os != "aix") {
           configs -= [ "//build/config/gcc:symbol_visibility_hidden" ]
