diff -ruN --ex v8-14.6.202.26/third_party/highway/BUILD.gn v8-work3/third_party/highway/BUILD.gn
--- a/third_party/highway/BUILD.gn	2026-03-20 22:50:53
+++ b/third_party/highway/BUILD.gn	2026-04-10 17:15:35
@@ -33,6 +33,11 @@
     # enable emulation until highway aix support is ready.
     defines += [ "HWY_BROKEN_EMU128=0" ]
   }
+  if (is_mingw) {
+    # GCC on MinGW does not support the "avx10.2-512" target attribute string
+    # even in GCC 15.2. Disable AVX10.2 to avoid compilation failures.
+    defines += [ "HWY_BROKEN_AVX10_2=HWY_AVX10_2" ]
+  }
 }
 
 source_set("libhwy") {
