--- webrtc-audio-processing-v1.3/webrtc/api/task_queue/task_queue_base.h.orig	2026-05-05 08:09:31.427463400 +0200
+++ webrtc-audio-processing-v1.3/webrtc/api/task_queue/task_queue_base.h	2026-05-05 08:09:30.445450700 +0200
@@ -15,6 +15,7 @@
 #include "api/task_queue/queued_task.h"
 #include "rtc_base/system/rtc_export.h"
 #include "rtc_base/thread_annotations.h"
+#include <cstdint>
 
 namespace webrtc {
 
