--- a/Makefile.inc
+++ b/Makefile.inc
@@ -1045,7 +1045,7 @@
         ln -fs $(call qndir,$1.$5) $(call qndir,$1) 2>/dev/null)
     endif
   else # MinGW
-    solink = -o $(call quote,$1) $(call linkopt,-soname,$(strip $(call qndir,$1).$5))
+    solink = -o $(call quote,$1) $(call linkopt,--out-implib,$(OUTDIR)/$(strip $(call qndir,$1).a))
   endif
 else # macOS
   solink = -o $(call qbname,$1).$2$(call qsuffix,$1) \
--- a/Makefile
+++ b/Makefile
@@ -1544,15 +1544,20 @@
 	@mkdir -p $(PREFIX)/$(POUTDIR) $(PREFIX)/$(PBINDIR) $(PREFIX)/$(PINCDIR) $(PREFIX)/$(PSRCDIR)
 	@echo
 	@echo "LIBXSMM installing libraries..."
-	@$(CP) -va $(OUTDIR)/libxsmmnoblas.$(DLIBEXT)* $(PREFIX)/$(POUTDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmnoblas.$(DLIBEXT) $(PREFIX)/$(PBINDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmnoblas.$(DLIBEXT).a $(PREFIX)/$(POUTDIR) 2>/dev/null || true
	@$(CP) -v  $(OUTDIR)/libxsmmnoblas.$(SLIBEXT)  $(PREFIX)/$(POUTDIR) 2>/dev/null || true
-	@$(CP) -va $(OUTDIR)/libxsmmgen.$(DLIBEXT)* $(PREFIX)/$(POUTDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmgen.$(DLIBEXT) $(PREFIX)/$(PBINDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmgen.$(DLIBEXT).a $(PREFIX)/$(POUTDIR) 2>/dev/null || true
 	@$(CP) -v  $(OUTDIR)/libxsmmgen.$(SLIBEXT)  $(PREFIX)/$(POUTDIR) 2>/dev/null || true
-	@$(CP) -va $(OUTDIR)/libxsmmext.$(DLIBEXT)* $(PREFIX)/$(POUTDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmext.$(DLIBEXT) $(PREFIX)/$(PBINDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmext.$(DLIBEXT).a $(PREFIX)/$(POUTDIR) 2>/dev/null || true
 	@$(CP) -v  $(OUTDIR)/libxsmmext.$(SLIBEXT)  $(PREFIX)/$(POUTDIR) 2>/dev/null || true
-	@$(CP) -va $(OUTDIR)/libxsmmf.$(DLIBEXT)* $(PREFIX)/$(POUTDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmf.$(DLIBEXT) $(PREFIX)/$(PBINDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmmf.$(DLIBEXT).a $(PREFIX)/$(POUTDIR) 2>/dev/null || true
 	@$(CP) -v  $(OUTDIR)/libxsmmf.$(SLIBEXT)  $(PREFIX)/$(POUTDIR) 2>/dev/null || true
-	@$(CP) -va $(OUTDIR)/libxsmm.$(DLIBEXT)* $(PREFIX)/$(POUTDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmm.$(DLIBEXT) $(PREFIX)/$(PBINDIR) 2>/dev/null || true
+	@$(CP) -va $(OUTDIR)/libxsmm.$(DLIBEXT).a $(PREFIX)/$(POUTDIR) 2>/dev/null || true
 	@$(CP) -v  $(OUTDIR)/libxsmm.$(SLIBEXT)  $(PREFIX)/$(POUTDIR) 2>/dev/null || true
 	@if [ -e $(OUTDIR)/mic/libxsmmnoblas.$(DLIBEXT) ]; then \
 		mkdir -p $(PREFIX)/$(POUTDIR)/mic; \
