--- a/Makefile.inc
+++ b/Makefile.inc
@@ -1465,7 +1465,6 @@
       CPEDANTIC += -Wformat=2
     else # MinGW
       CXXFLAGS += -fno-asynchronous-unwind-tables
-      FCFLAGS += -fno-asynchronous-unwind-tables
       CFLAGS += -fno-asynchronous-unwind-tables
       ifneq (,$(filter 0 1,$(PEDANTIC)))
         CFLAGS += -Wno-format
@@ -1948,11 +1947,6 @@
     ifneq (0,$(OPT))
       CXXFLAGS += -fdata-sections -ffunction-sections
       CFLAGS += -fdata-sections -ffunction-sections
-      ifeq (,$(filter-out 0,$(INTEL) $(ICX)))
-        FCFLAGS += -fdata-sections -ffunction-sections
-      else ifneq (,$(strip $(GFC)))
-        FCFLAGS += -fdata-sections -ffunction-sections
-      endif
       # --gc-sections: relies on section-flags present at compile-stage
       LDFLAGS := $(call linkopt,--gc-sections) $(LDFLAGS)
       ifeq (0,$(VISIBILITY)) # -fvisibility=hidden may cause crashes
