--- a/src/declarativeimports/calendar/CMakeLists.txt
+++ b/src/declarativeimports/calendar/CMakeLists.txt
@@ -23,6 +23,10 @@
     KF5::CoreAddons
 )
 
+if(MINGW)
+    set_target_properties(calendarplugin PROPERTIES PREFIX "")
+endif()
+
 install(TARGETS calendarplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/calendar)
 install(DIRECTORY qml/ DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/calendar)
 
--- a/src/declarativeimports/core/CMakeLists.txt
+++ b/src/declarativeimports/core/CMakeLists.txt
@@ -75,6 +75,10 @@
     target_link_libraries(corebindingsplugin Qt${QT_MAJOR_VERSION}::Gui_GLESv2)
 endif()
 
+if(MINGW)
+    set_target_properties(corebindingsplugin PROPERTIES PREFIX "")
+endif()
+
 install(TARGETS corebindingsplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/core)
 install(FILES qmldir DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/core)
 
--- a/src/declarativeimports/plasmacomponents/CMakeLists.txt
+++ b/src/declarativeimports/plasmacomponents/CMakeLists.txt
@@ -24,6 +24,9 @@
         KF5::Service #for kplugininfo.h
 )
 
+if(MINGW)
+    set_target_properties(plasmacomponentsplugin PROPERTIES PREFIX "")
+endif()
 
 install(TARGETS plasmacomponentsplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/components)
 
--- a/src/declarativeimports/plasmaextracomponents/CMakeLists.txt
+++ b/src/declarativeimports/plasmaextracomponents/CMakeLists.txt
@@ -19,6 +19,9 @@
         KF5::Service
         KF5::Plasma)
 
+if(MINGW)
+    set_target_properties(plasmaextracomponentsplugin PROPERTIES PREFIX "")
+endif()
 
 install(TARGETS plasmaextracomponentsplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/extras)
 
--- a/src/declarativeimports/platformcomponents/CMakeLists.txt
+++ b/src/declarativeimports/platformcomponents/CMakeLists.txt
@@ -17,6 +17,10 @@
     KF5::IconThemes
 )
 
+if(MINGW)
+    set_target_properties(platformcomponentsplugin PROPERTIES PREFIX "")
+endif()
+
 install(TARGETS platformcomponentsplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/platformcomponents)
 install(FILES qmldir DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/platformcomponents)
 
