--- a/CMake/resources.cmake
+++ b/CMake/resources.cmake
@@ -22,7 +22,7 @@
 #######################################################################
 # headers
 find_file(HAVE_ALSA_ASOUNDLIB_H alsa/asoundlib.h)
-find_file(HAVE_DLFCN_H dlfcn.h)
+set(HAVE_DLFCN_H 0)
 find_file(HAVE_FREETYPE_H freetype.h PATH_SUFFIXES freetype2 freetype2/freetype)
 find_file(HAVE_GL_GL_H GL/gl.h)
 find_file(HAVE_GL_GLU_H GL/glu.h)
--- a/CMake/resources.cmake
+++ b/CMake/resources.cmake
@@ -86,7 +86,7 @@
 #######################################################################
 # libraries
 find_library(LIB_CAIRO cairo)
-find_library(LIB_dl dl)
+set(LIB_dl 0)
 find_library(LIB_fontconfig fontconfig)
 find_library(LIB_freetype freetype)
 find_library(LIB_GL GL)
