From ff0a2ccf4ee0685a8032c460495a0f6468b7fb7d Mon Sep 17 00:00:00 2001
From: albfan <albfan@gnome.org>
Date: Sun, 30 Dec 2018 12:49:11 +0100
Subject: Fix libgit2-glib cache

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 44a6e40..6277097 100644
--- a/meson.build
+++ b/meson.build
@@ -130,7 +130,7 @@ gsettings_desktop_schemas_dep = dependency('gsettings-desktop-schemas')
 gtk_dep = dependency('gtk+-3.0', version: '>= 3.20.0')
 gtkspell_dep = dependency('gtkspell3-3.0')
 gtksourceview_dep = dependency('gtksourceview-3.0', version: '>= 3.10')
-libgit2_glib_dep = dependency('libgit2-glib-1.0', version: ['>= 0.25.0', '< 0.27.0'])
+libgit2_glib_dep = dependency('libgit2-glib-1.0', version: '>= 0.27.7')
 libpeas_dep = dependency('libpeas-1.0')
 libsecret_dep = dependency('libsecret-1')
 libsoup_dep = dependency('libsoup-2.4')
-- 
2.17.1

