From 05e92cc59017d57ab7f4f5e33ca51371ebc8503a Mon Sep 17 00:00:00 2001
From: Luke Street <luke.street@encounterpc.com>
Date: Mon, 1 Jan 2018 17:54:13 -0500
Subject: [PATCH] Use DLL_EXPORT condition for DllMain

---
 src/plist.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plist.c b/src/plist.c
index f62e6be..d563fe5 100644
--- a/src/plist.c
+++ b/src/plist.c
@@ -29,7 +29,7 @@
 #include <stdio.h>
 #include <math.h>
 
-#ifdef WIN32
+#ifdef DLL_EXPORT
 #include <windows.h>
 #else
 #include <pthread.h>
@@ -56,7 +56,7 @@ static void internal_plist_deinit(void)
     plist_xml_deinit();
 }
 
-#ifdef WIN32
+#ifdef DLL_EXPORT
 
 typedef volatile struct {
     LONG lock;
