--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -92,7 +92,7 @@
         DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/)
 # Link the built library to the package in the binary directory
 add_custom_command(TARGET Spglib_python POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E create_symlink
+        COMMAND ${CMAKE_COMMAND} -E copy
         $<TARGET_FILE:Spglib_python>
         ${CMAKE_CURRENT_BINARY_DIR}/spglib/$<TARGET_FILE_NAME:Spglib_python>
 )
