--- libgda-5.2.2/configure.ac.orig	2014-08-21 15:51:22.890400000 +0400
+++ libgda-5.2.2/configure.ac	2014-08-21 15:51:55.603600000 +0400
@@ -288,6 +288,14 @@
 			GLADE_PIXMAP=`$PKG_CONFIG --variable=pixmapdir gladeui-2.0`], [have_glade=no])
 fi
 
+case "$host" in
+  *-*-mingw*)
+    GLADE_CATALOG=`cygpath -u $GLADE_CATALOG`
+    GLADE_PIXMAP=`cygpath -u $GLADE_PIXMAP`
+    ;;
+esac
+
+
 AM_CONDITIONAL(HAVE_GDKPIXBUF, test x"$have_gdkpixbuf" = "xyes")
 AC_SUBST(GDKPIXBUF_CFLAGS)
 AC_SUBST(GDKPIXBUF_LIBS)
