diff -urN PDAL-2.4.2-src/pdal/util/FileUtils.cpp.orig PDAL-2.4.2-src/pdal/util/FileUtils.cpp
--- PDAL-2.4.2-src/pdal/util/FileUtils.cpp.orig	2022-06-15 00:11:24.813443300 +0200
+++ PDAL-2.4.2-src/pdal/util/FileUtils.cpp	2022-06-15 00:24:34.739874100 +0200
@@ -234,7 +234,7 @@
         fs::directory_iterator end;
         while (it != end)
         {
-            files.push_back(fromNative(it->path()));
+            files.push_back(it->path().u8string());
             it++;
         }
     }
@@ -556,7 +556,7 @@
         }
     }
 
-#ifndef _WIN32
+#ifndef PDAL_WIN32_STL
     ctx.m_fd = ::open(filename.c_str(), readOnly ? O_RDONLY : O_RDWR);
 #else
     ctx.m_fd = ::_wopen(toNative(filename).data(), readOnly ? O_RDONLY : O_RDWR);
