diff -urN PDAL-2.5.0-src/pdal/util/pdal_util_export.hpp.orig PDAL-2.5.0-src/pdal/util/pdal_util_export.hpp
--- PDAL-2.5.0-src/pdal/util/pdal_util_export.hpp.orig	2023-01-13 18:18:05.000000000 +0100
+++ PDAL-2.5.0-src/pdal/util/pdal_util_export.hpp	2023-01-24 09:52:12.157579500 +0100
@@ -42,7 +42,11 @@
 #   define PDAL_DLL   __declspec(dllexport)
 #   define PDAL_LOCAL
 #   define PDAL_DLL_UNIX
+#ifdef __GNUC__
+#   define PDAL_DLL_DEPRECATED __attribute__((deprecated)) PDAL_DLL
+#else
 #   define PDAL_DLL_DEPRECATED   __declspec(deprecated, dllexport)
+#endif
 #else
 #   define PDAL_DLL     __attribute__ ((visibility("default")))
 #   define PDAL_LOCAL   __attribute__((visibility("hidden")))
