From b266b48b70a736927a8ee585fc888f270005973b Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 23 Feb 2015 22:16:54 +0000
Subject: [PATCH 4/7] lang: Add autotools (ac, am, in, m4, pc)

---
 src/lang.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lang.c b/src/lang.c
index 17783ae..df58f77 100644
--- a/src/lang.c
+++ b/src/lang.c
@@ -9,6 +9,7 @@ lang_spec_t langs[] = {
     { "ada", { "ada", "adb", "ads" } },
     { "asciidoc", { "adoc", "ad", "asc", "asciidoc" } },
     { "asm", { "asm", "s" } },
+    { "autotools", { "ac", "am", "in", "m4", "pc" } },
     { "batch", { "bat", "cmd" } },
     { "bitbake", { "bb", "bbappend", "bbclass", "inc" } },
     { "bro", { "bro", "bif" } },
-- 
2.12.1

