--- the_silver_searcher-0.27.0/configure.ac.orig	2015-01-19 15:22:41.877800000 +0300
+++ the_silver_searcher-0.27.0/configure.ac	2015-01-19 15:22:45.902600000 +0300
@@ -50,6 +50,14 @@
 
 AC_CHECK_FUNCS(fgetln getline realpath strlcpy strndup vasprintf madvise posix_fadvise)
 
+case "$host" in
+  *-*-mingw*)
+    LIBS="$LIBS -lshlwapi"
+  ;;
+  *)
+  ;;
+esac
+
 AC_CONFIG_FILES([Makefile the_silver_searcher.spec])
 AC_CONFIG_HEADERS([src/config.h])
 
--- the_silver_searcher-0.27.0/src/util.c.orig	2015-01-19 15:24:40.718600000 +0300
+++ the_silver_searcher-0.27.0/src/util.c	2015-01-19 15:24:43.058600000 +0300
@@ -9,6 +9,7 @@
 #include "config.h"
 
 #ifdef _WIN32
+#include <windows.h>
 #define flockfile(x)
 #define funlockfile(x)
 #define getc_unlocked(x) getc(x)
--- the_silver_searcher-0.27.0/src/print.c.orig	2015-01-19 15:46:04.745600000 +0300
+++ the_silver_searcher-0.27.0/src/print.c	2015-01-19 15:55:36.813200000 +0300
@@ -34,9 +34,9 @@
         print_path(path, ':');
     }
     if (opts.color) {
-        fprintf(out_fd, "%s%lu%s%c", opts.color_line_number, count, color_reset, sep);
+        fprintf(out_fd, "%s%zu%s%c", opts.color_line_number, count, color_reset, sep);
     } else {
-        fprintf(out_fd, "%lu%c", count, sep);
+        fprintf(out_fd, "%zu%c", count, sep);
     }
 }
 
@@ -149,8 +149,8 @@
                     print_line_number(line, ';');
                     for (; last_printed_match < cur_match; last_printed_match++) {
                         fprintf(out_fd, "%lu %lu",
-                                (matches[last_printed_match].start - prev_line_offset),
-                                (matches[last_printed_match].end - matches[last_printed_match].start));
+                                (unsigned long)(matches[last_printed_match].start - prev_line_offset),
+                                ((unsigned long)matches[last_printed_match].end - matches[last_printed_match].start));
                         last_printed_match == cur_match - 1 ? fputc(':', out_fd) : fputc(',', out_fd);
                     }
                     print_line(buf, i, prev_line_offset);
@@ -243,7 +243,7 @@
 void print_column_number(const match_t matches[], size_t last_printed_match,
                          size_t prev_line_offset, const char sep) {
     fprintf(out_fd, "%lu%c",
-            (matches[last_printed_match].start - prev_line_offset) + 1,
+            (unsigned long)((matches[last_printed_match].start - prev_line_offset) + 1),
             sep);
 }
 
