diff -urN the_silver_searcher-0.27.0.orig/src/lang.c the_silver_searcher-0.27.0/src/lang.c
--- the_silver_searcher-0.27.0.orig/src/lang.c	2015-02-22 13:29:01.378778800 +0000
+++ the_silver_searcher-0.27.0/src/lang.c	2015-02-22 13:35:00.015227300 +0000
@@ -40,6 +40,7 @@
     { "lua", { "lua" } },
     { "m4", { "m4" } },
     { "make", { "^Makefile", "^Makefile.Debug", "^Makefile.Release", "Makefiles", "mk", "mak" } },
+    { "makepkg", { "^PKGBUILD", "diff", "patch", "in", "install" } },
     { "markdown", { "markdown", "mdown", "mdwn", "mkdn", "mkd", "md" } },
     { "mason", { "mas", "mhtml", "mpl", "mtxt" } },
     { "matlab", { "m" } },
