--- libcaca-0.99.beta17/caca/Makefile.am.orig	2010-02-08 16:34:28 +0400
+++ libcaca-0.99.beta17/caca/Makefile.am	2012-04-03 19:05:11 +0400
@@ -73,7 +73,7 @@
 	driver/win32.c \
 	$(NULL)
 
-install-exec-local:
+install-exec-local: install-libLTLIBRARIES install-pkgconfigDATA install-includeHEADERS
 	$(mkinstalldirs) $(DESTDIR)$(libdir)
 	. ./libcaca.la || exit 1; \
 	for x in $$library_names; do \
--- libcaca-0.99.beta17/cxx/Makefile.am.orig	2010-02-08 16:34:28 +0400
+++ libcaca-0.99.beta17/cxx/Makefile.am	2012-04-03 19:07:39 +0400
@@ -22,7 +22,7 @@
 cxxtest_LDADD = libcaca++.la ../caca/libcaca.la
 
 if USE_CXX
-install-exec-local:
+install-exec-local: install-libLTLIBRARIES install-pkgconfigDATA install-includeHEADERS
 	$(mkinstalldirs) $(DESTDIR)$(libdir)
 	. ./libcaca++.la || exit 1; \
 	for x in $$library_names; do \
--- libcaca-0.99.beta17/doc/Makefile.am.orig	2010-02-08 16:34:28 +0400
+++ libcaca-0.99.beta17/doc/Makefile.am	2012-04-03 19:09:57 +0400
@@ -39,7 +39,7 @@
 clean-local:
 	-rm -Rf html latex man
 
-install-data-local:
+install-data-local: install-man1
 if BUILD_DOCUMENTATION
 	mkdir -p $(DESTDIR)$(datadir)/doc
 	ln -sf libcaca-dev $(DESTDIR)$(datadir)/doc/libcucul-dev
