--- onig-5.9.5/Makefile.am.orig	2013-10-21 11:53:41.000000000 +0400
+++ onig-5.9.5/Makefile.am	2014-10-20 20:01:37.139200000 +0400
@@ -31,7 +31,7 @@
 	$(encdir)/euc_tw.c $(encdir)/euc_kr.c $(encdir)/big5.c \
 	$(encdir)/gb18030.c $(encdir)/koi8_r.c $(encdir)/cp1251.c
 
-libonig_la_LDFLAGS = -version-info $(LTVERSION)
+libonig_la_LDFLAGS = -version-info $(LTVERSION) -no-undefined
 
 EXTRA_DIST = oniguruma.pc.in HISTORY README.ja index.html index_ja.html \
 	doc/API doc/API.ja doc/RE doc/RE.ja doc/FAQ doc/FAQ.ja \
