--- a/setup.py
+++ b/setup.py
@@ -518,7 +518,7 @@
                 + list(GRPCIO_CC_SRCS)
                 + core_c_files
                 + asm_files
-                + ["third_party/abseil-cpp/absl/log/initialize.cc"]
+                + ([] if BUILD_WITH_SYSTEM_ABSL else ["third_party/abseil-cpp/absl/log/initialize.cc"])
             ),
             include_dirs=list(EXTENSION_INCLUDE_DIRECTORIES),
             libraries=list(EXTENSION_LIBRARIES),
