--- mbedtls-2.2.1/CMakeLists.txt.orig	2016-01-06 16:52:30.979905100 +0200
+++ mbedtls-2.2.1/CMakeLists.txt	2016-01-06 16:52:56.855145500 +0200
@@ -86,7 +86,8 @@
 endif()
 
 # targets for doxygen only work on Unix
-if(UNIX)
+# and on MINGW
+if(UNIX OR MINGW)
     ADD_CUSTOM_TARGET(apidoc
         COMMAND mkdir -p apidoc
         COMMAND cp include/mbedtls/config.h include/mbedtls/config.h.bak
@@ -98,7 +99,7 @@
     ADD_CUSTOM_TARGET(apidoc_clean
         COMMAND rm -rf apidoc
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-endif(UNIX)
+endif(UNIX OR MINGW)
 
 if(ENABLE_TESTING)
     enable_testing()
