--- a/include/mbedtls/mbedtls_config.h
+++ b/include/mbedtls/mbedtls_config.h
@@ -109,1 +109,1 @@
-#define MBEDTLS_ERROR_STRERROR_DUMMY
+//#define MBEDTLS_ERROR_STRERROR_DUMMY
--- a/tf-psa-crypto/include/psa/crypto_config.h
+++ b/tf-psa-crypto/include/psa/crypto_config.h
@@ -447,1 +447,1 @@
-//#define MBEDTLS_THREADING_PTHREAD
+#define MBEDTLS_THREADING_PTHREAD
@@ -470,1 +470,1 @@
-//#define MBEDTLS_THREADING_C
+#define MBEDTLS_THREADING_C
@@ -608,1 +608,1 @@
-//#define MBEDTLS_DEPRECATED_WARNING
+#define MBEDTLS_DEPRECATED_WARNING
@@ -1737,1 +1737,1 @@
-//#define MBEDTLS_HAVE_SSE2
+#define MBEDTLS_HAVE_SSE2
