--- mbedtls-2.1.0/tests/CMakeLists.txt.orig	2015-09-04 15:38:26.000000000 +0300
+++ mbedtls-2.1.0/tests/CMakeLists.txt	2015-09-13 21:45:47.677244800 +0300
@@ -101,7 +101,7 @@
     file(TO_NATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/data_files" target)
 
     if (NOT EXISTS ${link})
-        if (CMAKE_HOST_UNIX)
+        if (CMAKE_HOST_UNIX OR MSYS)
             set(command ln -s ${target} ${link})
         else()
             set(command cmd.exe /c mklink /d ${link} ${target})
