diff -Naur gdbm-1.14.1.orig/src/falloc.c gdbm-1.14.1/src/falloc.c
--- gdbm-1.14.1.orig/src/falloc.c	2018-05-06 21:50:23.214440900 -0400
+++ gdbm-1.14.1/src/falloc.c	2018-05-07 14:15:48.186151000 -0400
@@ -278,7 +278,7 @@
 
   /* Split the header block. */
   temp = GDBM_DEBUG_ALLOC ("push_avail_block:malloc-failure",
-			      malloc (av_size));
+			      calloc (1, av_size));
   if (temp == NULL)
     {
       GDBM_SET_ERRNO (dbf, GDBM_MALLOC_ERROR, TRUE);
diff -Naur gdbm-1.14.1.orig/src/gdbmopen.c gdbm-1.14.1/src/gdbmopen.c
--- gdbm-1.14.1.orig/src/gdbmopen.c	2018-05-06 21:50:23.243439700 -0400
+++ gdbm-1.14.1/src/gdbmopen.c	2018-05-07 14:15:12.620590400 -0400
@@ -537,7 +537,7 @@
         {
 	  (dbf->bucket_cache[index]).ca_bucket = 
 	    GDBM_DEBUG_ALLOC ("_gdbm_init_cache:bucket-malloc-failure",
-	                      malloc (dbf->header->bucket_size));
+	                      calloc (1, dbf->header->bucket_size));
           if ((dbf->bucket_cache[index]).ca_bucket == NULL)
 	    {
               GDBM_SET_ERRNO (dbf, GDBM_MALLOC_ERROR, TRUE);
