From 0865ea25bf981008921c1231a98c907b16e84c20 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 22 Sep 2014 22:53:17 +0100
Subject: [PATCH 2/2] Add --ninja option

---
 script/build.py | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/script/build.py b/script/build.py
index 6442cdc..535e904 100755
--- a/script/build.py
+++ b/script/build.py
@@ -13,11 +13,14 @@ SOURCE_ROOT = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
 def main():
   os.chdir(SOURCE_ROOT)
 
-  ninja = os.path.join('vendor', 'depot_tools', 'ninja')
-  if sys.platform == 'win32':
-    ninja += '.exe'
-
   args = parse_args()
+
+  ninja = args.ninja
+  if ninja == None:
+    ninja = os.path.join('vendor', 'depot_tools', 'ninja')
+    if sys.platform == 'win32':
+      ninja += '.exe'
+
   for config in args.configuration:
     build_path = os.path.join('out', config)
     ret = subprocess.call([ninja, '-C', build_path, args.target])
@@ -27,6 +30,10 @@ def main():
 
 def parse_args():
   parser = argparse.ArgumentParser(description='Build atom-shell')
+  parser.add_argument('-n', '--ninja',
+                      help='Select ninja executable',
+                      default=None,
+                      required=False)
   parser.add_argument('-c', '--configuration',
                       help='Build with Release or Debug configuration',
                       nargs='+',
-- 
2.1.1

