--- a/C/Sha512.c	2024-11-09 11:00:00.000000000 +0100
+++ b/C/Sha512.c	2024-12-01 00:36:20.615532600 +0100
@@ -514,7 +514,9 @@
 #ifdef __clang_major__
   #pragma GCC diagnostic ignored "-Wlanguage-extension-token"
 #endif
+#ifdef _MSC_VER
     __try
+#endif
     {
 #if 0 // 1 : for debug (reduced version to detect sha512)
       const uint64x2_t a = vdupq_n_u64(1);
@@ -540,10 +542,12 @@
 #endif
 #endif
     }
+#ifdef _MSC_VER
     __except (EXCEPTION_EXECUTE_HANDLER)
     {
       // printf("\n==== CPU_IsSupported_SHA512 EXCEPTION_EXECUTE_HANDLER\n");
     }
+#endif
   }
   return False;
 #else
