diff --git a/libs/tk/ydk/win32/gdkgc-win32.c b/libs/tk/ydk/win32/gdkgc-win32.c
index e7e6fd8f2e..baf19afae7 100644
--- a/libs/tk/ydk/win32/gdkgc-win32.c
+++ b/libs/tk/ydk/win32/gdkgc-win32.c
@@ -744,7 +744,7 @@ static GdkDrawableImplWin32 *
 get_impl_drawable (GdkDrawable *drawable)
 {
   if (GDK_IS_OFFSCREEN_WINDOW (drawable))
-    return _gdk_offscreen_window_get_real_drawable (GDK_OFFSCREEN_WINDOW (drawable));
+    return (GdkDrawableImplWin32 *) _gdk_offscreen_window_get_real_drawable (GDK_OFFSCREEN_WINDOW (drawable));
   if (GDK_IS_DRAWABLE_IMPL_WIN32 (drawable))
     return GDK_DRAWABLE_IMPL_WIN32(drawable);
   else if (GDK_IS_WINDOW (drawable))
