diff --git a/libs/tk/ytk/gtksocket-win32.c b/libs/tk/ytk/gtksocket-win32.c
index 152d743c33..6ba62ac7dc 100644
--- a/libs/tk/ytk/gtksocket-win32.c
+++ b/libs/tk/ytk/gtksocket-win32.c
@@ -46,7 +46,7 @@ _gtk_socket_windowing_get_id (GtkSocket *socket)
   g_return_val_if_fail (GTK_IS_SOCKET (socket), 0);
   g_return_val_if_fail (GTK_WIDGET_ANCHORED (socket), 0);
 
-  if (!gtk_widget_get_realized (socket))
+  if (!gtk_widget_get_realized ((GtkWidget *) socket))
     gtk_widget_realize (GTK_WIDGET (socket));
 
   return (GdkNativeWindow) GDK_WINDOW_HWND (GTK_WIDGET (socket)->window);
