--- a/XClib/beefleg.h
+++ b/XClib/beefleg.h
@@ -626,10 +626,10 @@
 
 //random number generation macros
 #define unirandinc0() \
-    ( ((double) random()) / ((double) RAND_MAX) )
+    ( ((double) rand()) / ((double) RAND_MAX) )
 
 #define unirandex0() \
-    ( (1.+random()) / (1.+RAND_MAX) )
+    ( (1.+rand()) / (1.+RAND_MAX) )
 
 #define normrand() \
     ( sqrt(-2.*log(unirandex0())) * cos(M_PI*2.*unirandinc0()) )
--- a/XClib/beefun.c
+++ b/XClib/beefun.c
@@ -322,13 +322,13 @@
 // initialize pseudo random number generator
 void beefrandinit_(unsigned int *seed)
 {
-    srandom(*seed);
+    srand(*seed);
 }
 
 // initialize pseudo random number generator with default seed
 void beefrandinitdef_()
 {
-    srandom(defaultseed);
+    srand(defaultseed);
 }
 
 // calculate ensemble energies
