--- a/Modules/CMakeLists.txt
+++ b/Modules/CMakeLists.txt
@@ -215,6 +215,7 @@
     sockets.c
     stack.c)
 qe_add_library(qe_modules_c ${src_modules_c})
+set_target_properties(qe_modules_c PROPERTIES Fortran_MODULE_DIRECTORY "")
 
 if(TARGET gitrev)
    add_dependencies(qe_modules gitrev)
--- a/UtilXlib/CMakeLists.txt
+++ b/UtilXlib/CMakeLists.txt
@@ -34,6 +34,7 @@
     memusage.c
     ptrace.c)
 qe_add_library(qe_utilx_c ${src_util_c})
+set_target_properties(qe_utilx_c PROPERTIES Fortran_MODULE_DIRECTORY "")
 target_link_libraries(qe_utilx
     PUBLIC
         qe_utilx_c # cclock is exposed by the mytime module in clocks_handler
--- a/XClib/CMakeLists.txt
+++ b/XClib/CMakeLists.txt
@@ -43,6 +43,7 @@
     pbecor.c
     beefun.c)
 qe_add_library(qe_libbeef ${sources_libbeef})
+set_target_properties(qe_libbeef PROPERTIES Fortran_MODULE_DIRECTORY "")
 target_link_libraries(qe_libbeef 
     PUBLIC
         qe_openacc_c
--- a/external/pw2qmcpack/CMakeLists.txt
+++ b/external/pw2qmcpack/CMakeLists.txt
@@ -47,6 +47,7 @@
 qe_add_library(pw2qmcpack_esh5 src/esh5_interfaces.c)
 target_compile_definitions(pw2qmcpack_esh5 PUBLIC "H5_USE_16_API;__HDF5_C")
 target_link_libraries(pw2qmcpack_esh5 PRIVATE pw2qmcpack_hdf5_c pw2qmcpack_mpi_c)
+set_target_properties(pw2qmcpack_esh5 PROPERTIES Fortran_MODULE_DIRECTORY "")
 
 set(sources src/pw2qmcpack.f90)
 qe_add_executable(pw2qmcpack_exe ${sources})
