--- a/elmergrid/src/CMakeLists.txt
+++ b/elmergrid/src/CMakeLists.txt
@@ -12,19 +12,13 @@
   ENDIF()
 ENDIF()
 
-# Don't install metis
-SET(METIS_INSTALL FALSE CACHE BOOL "")
-ADD_SUBDIRECTORY(metis-5.1.0)
-
 SET(elmergrid_SRCS egdef.h egparallel.h egnative.h egconvert.h
   egexport.h egextra.h egmesh.h 
   egtypes.h egutils.h egparallel.c egnative.c egconvert.c egexport.c
   egextra.c egmesh.c fempre.c 
   egutils.c)
 
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/metis)
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
-INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/elmergrid/src/metis)
 INCLUDE_DIRECTORIES(${CMAKE_BINARY_DIR}/elmergrid/src)
 
 ADD_DEFINITIONS(-DDISABLE_MATC)
--- a/elmergrid/src/egparallel.c
+++ b/elmergrid/src/egparallel.c
@@ -50,7 +50,7 @@
 #include "../config.h"
 
 #if USE_METIS
-#include "metis-5.1.0/include/metis.h"
+#include <metis.h>
 #endif
 
 
