From 48e9430ccb858ca5bda28b967a0c84b51e2404b2 Mon Sep 17 00:00:00 2001
From: "E. Albiter" <ealbiter@gmail.com>
Date: Fri, 15 Oct 2021 13:12:48 -0500
Subject: [PATCH] Compatibility with Qwt 6.2. Fix for #293

---
 ElmerGUI/Application/src/convergenceview.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ElmerGUI/Application/src/convergenceview.h b/ElmerGUI/Application/src/convergenceview.h
index 377b644b99..4259eb730e 100755
--- a/ElmerGUI/Application/src/convergenceview.h
+++ b/ElmerGUI/Application/src/convergenceview.h
@@ -51,8 +51,8 @@
 #include <qwt_plot_curve.h>
 #include <qwt_plot_grid.h>
 #include <qwt_legend.h>
-/*#include <qwt_data.h> <-- deprecated in Qwt6, using qwt_compat.h instead*/
-#include <qwt_compat.h>
+/*#include <qwt_data.h> <-- deprecated in Qwt6, using qwt_compat.h instead
+#include <qwt_compat.h> <-- Removed in Qwt 6.2 */
 #include <qwt_text.h>
 #include <qwt_scale_engine.h>
 
@@ -76,8 +76,8 @@ class CurveData
   
 private:
   int d_count;
-  QwtArray<double> d_x;
-  QwtArray<double> d_y;
+  QVector<double> d_x;
+  QVector<double> d_y;
 };
 
 class Curve
