--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,21 +204,16 @@
   #     set(OpenMP_libiomp5_LIBRARY ${OpenMP_CXX_LIB_NAMES})
   #   endif()
   # else()
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-    SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_Fortran_FLAGS}")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
   # endif()
 
   FIND_PACKAGE(OpenMP REQUIRED)
-  
-  # Test compiler support for OpenMP 4.0 features used
-  INCLUDE(testOpenMP40)
-  IF(CMAKE_Fortran_COMPILER_SUPPORTS_OPENMP40)
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+  SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_Fortran_FLAGS}")
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+  IF(OpenMP_Fortran_VERSION VERSION_GREATER_EQUAL 4.0)
     SET(HAVE_OPENMP40 TRUE)
     MARK_AS_ADVANCED(HAVE_OPENMP40)
-    # Test compiler support for advanced OpenMP 4.5 features used
-    INCLUDE(testOpenMP45)
-    IF(CMAKE_Fortran_COMPILER_SUPPORTS_OPENMP45)
+    IF(OpenMP_Fortran_VERSION VERSION_GREATER_EQUAL 4.5)
       SET(HAVE_OPENMP45 TRUE)
       MARK_AS_ADVANCED(HAVE_OPENMP45)
     ENDIF()
