diff -bur hypre-2.33.0-o/src/parcsr_ls/CMakeLists.txt hypre-2.33.0/src/parcsr_ls/CMakeLists.txt
--- hypre-2.33.0-o/src/parcsr_ls/CMakeLists.txt	2025-04-20 02:33:32.494730700 -0600
+++ hypre-2.33.0/src/parcsr_ls/CMakeLists.txt	2025-04-20 02:37:05.188410400 -0600
@@ -150,6 +150,10 @@
   par_mgr_device.c
 )
 
+if (HYPRE_ENABLE_DSUPERLU)
+  list(APPEND SRCS dsuperlu.c)
+endif ()
+
 target_sources(${PROJECT_NAME}
   PRIVATE ${SRCS}
           ${HDRS}
