--- portaudio/src/hostapi/wasapi/pa_win_wasapi.c.orig	2021-07-23 16:25:22.408000000 +0200
+++ portaudio/src/hostapi/wasapi/pa_win_wasapi.c	2021-07-23 16:25:32.772678700 +0200
@@ -111,7 +111,7 @@
 #include "pa_ringbuffer.h"
 #include "pa_win_coinitialize.h"
 
-#if !defined(NTDDI_VERSION) || (defined(__GNUC__) && (__GNUC__ <= 6) && !defined(__MINGW64__))
+#if !defined(NTDDI_VERSION) || (defined(__GNUC__) && (__GNUC__ <= 6) && !defined(__MINGW64__) && !defined(__clang__))
 
     #undef WINVER
     #undef _WIN32_WINNT
