--- ./configure.in.orig	2021-04-01 02:35:42.000000000 +0200
+++ ./configure.in	2021-07-19 20:13:23.468337000 +0200
@@ -303,8 +303,8 @@
         if [[ "x$with_asio" = "xyes" ]]; then
             ASIODIR="$with_asiodir"
             add_objects src/hostapi/asio/pa_asio.o src/common/pa_ringbuffer.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o src/os/win/pa_win_coinitialize.o src/hostapi/asio/iasiothiscallresolver.o $ASIODIR/common/asio.o $ASIODIR/host/asiodrivers.o $ASIODIR/host/pc/asiolist.o
-            LIBS="${LIBS} -lwinmm -lm -lole32 -luuid"
-            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -lole32 -luuid"
+            LIBS="${LIBS} -lwinmm -lm -lole32"
+            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -lole32"
             CFLAGS="$CFLAGS -ffast-math -fomit-frame-pointer -I\$(top_srcdir)/src/hostapi/asio -I$ASIODIR/host/pc -I$ASIODIR/common -I$ASIODIR/host -UPA_USE_ASIO -DPA_USE_ASIO=1 -DWINDOWS"
             INCLUDES="$INCLUDES pa_asio.h"
 
@@ -320,8 +320,8 @@
         if [[ "x$with_wdmks" = "xyes" ]]; then
             DXDIR="$with_dxdir"
             add_objects src/hostapi/wdmks/pa_win_wdmks.o src/common/pa_ringbuffer.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o src/os/win/pa_win_wdmks_utils.o src/os/win/pa_win_waveformat.o
-            LIBS="${LIBS} -lwinmm -lm -luuid -lsetupapi -lole32"
-            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -L$DXDIR/lib -luuid -lsetupapi -lole32"
+            LIBS="${LIBS} -lwinmm -lm -lsetupapi -lole32"
+            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -L$DXDIR/lib -lsetupapi -lole32"
             #VC98="\"/c/Program Files/Microsoft Visual Studio/VC98/Include\""
             #CFLAGS="$CFLAGS -I$VC98 -DPA_NO_WMME -DPA_NO_ASIO"
             CFLAGS="$CFLAGS -I$DXDIR/include -UPA_USE_WDMKS -DPA_USE_WDMKS=1"
@@ -330,7 +330,7 @@
 
         if [[ "x$with_wmme" = "xyes" ]]; then
             add_objects src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o src/os/win/pa_win_waveformat.o
-            LIBS="${LIBS} -lwinmm -lm -lole32 -luuid"
+            LIBS="${LIBS} -lwinmm -lm -lole32"
             DLL_LIBS="${DLL_LIBS} -lwinmm"
             CFLAGS="$CFLAGS -UPA_USE_WMME -DPA_USE_WMME=1"
             INCLUDES="$INCLUDES pa_win_wmme.h pa_win_waveformat.h"
@@ -338,7 +338,7 @@
 
         if [[ "x$with_wasapi" = "xyes" ]]; then
             add_objects src/hostapi/wasapi/pa_win_wasapi.o src/common/pa_ringbuffer.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o src/os/win/pa_win_coinitialize.o src/os/win/pa_win_waveformat.o
-            LIBS="${LIBS} -lwinmm -lm -lole32 -luuid"
+            LIBS="${LIBS} -lwinmm -lm -lole32"
             DLL_LIBS="${DLL_LIBS} -lwinmm -lole32"
             CFLAGS="$CFLAGS -UPA_USE_WASAPI -DPA_USE_WASAPI=1"
             INCLUDES="$INCLUDES pa_win_wasapi.h pa_win_waveformat.h"
