From d2c834f3f37f5e473002def7fbf4cf0c567dcdb8 Mon Sep 17 00:00:00 2001
From: Peiyuan Song <squallatf@gmail.com>
Date: Mon, 24 Mar 2025 09:39:42 +0800
Subject: [PATCH] remove the `lib` prefix when enabling mimalloc-redirect for
 mingw

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0d780fa1..e5c5f46a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -592,6 +592,9 @@ if(MI_BUILD_SHARED)
     # install(FILES "$<TARGET_FILE_DIR:mimalloc>/${mi_libname}.dll.pdb" DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
   if(WIN32 AND MI_WIN_REDIRECT)
+    if(MINGW)
+      set_property(TARGET mimalloc PROPERTY PREFIX "")
+    endif()
     # On windows, link and copy the mimalloc redirection dll too.
     if(CMAKE_GENERATOR_PLATFORM STREQUAL "arm64ec")
       set(MIMALLOC_REDIRECT_SUFFIX "-arm64ec")
-- 
2.49.0.windows.1

