--- coin-Coin-4.0.0/include/Inventor/elements/SoGLLazyElement.h.orig	2021-08-08 22:36:30.241161500 -0700
+++ coin-Coin-4.0.0/include/Inventor/elements/SoGLLazyElement.h	2021-08-08 22:37:21.225549600 -0700
@@ -74,7 +74,7 @@
 
   void reset(SoState* state, uint32_t bitmask) const;
 
-  typedef struct COIN_DLL_API {
+  struct COIN_DLL_API GLState {
     uint32_t cachebitmask;
     uint32_t diffuse;
     SbColor ambient;
@@ -97,7 +97,7 @@
     SbUniqueId diffusenodeid;
     SbUniqueId transpnodeid;
     uint32_t reserved[4];
-  } GLState;
+  };
 
   virtual void setDiffuseElt(SoNode*,  int32_t numcolors,
                              const SbColor * colors, SoColorPacker * packer);
