--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -295,7 +295,7 @@
 ENDIF(WIN32 AND MSVC)
 
 # Unix, linux:
-if(UNIX)
+if(UNIX OR MINGW)
    SET_TARGET_PROPERTIES(
       ${GEOTIFF_LIBRARY_TARGET}
       PROPERTIES
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -6,7 +6,7 @@
 # ${INSTALL_CMAKE_DIR} and @PROJECT_ROOT_DIR@ is the relative
 # path to the root from there.  (Note that the whole install tree can
 # be relocated.)
-if (NOT WIN32)
+if (NOT MSVC)
   set (INSTALL_CMAKE_DIR "share/cmake/${PROJECT_NAME}")
   set (PROJECT_ROOT_DIR "../../..")
 else ()
