diff -Naur libgeotiff-1.5.1.orig/bin/Makefile.am libgeotiff-1.5.1/bin/Makefile.am
--- libgeotiff-1.5.1.orig/bin/Makefile.am	2019-04-02 16:00:18.771074061 -0400
+++ libgeotiff-1.5.1/bin/Makefile.am	2019-04-02 16:01:42.484867414 -0400
@@ -12,9 +12,9 @@
 TIFF_CFLAGS = @TIFF_INC@ -DHAVE_TIFF=1
 endif
 
-AM_CFLAGS = -I../ -I../libxtiff $(TIFF_CFLAGS) @PROJ_INCLUDE@
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/libxtiff $(TIFF_CFLAGS) @PROJ_INCLUDE@
 
-LDADD = ../libgeotiff.la
+LDADD = $(top_builddir)/libgeotiff.la
 
 EXTRA_DIST =    getopt.c
 
diff -Naur libgeotiff-1.5.1.orig/libxtiff/Makefile.am libgeotiff-1.5.1/libxtiff/Makefile.am
--- libgeotiff-1.5.1.orig/libxtiff/Makefile.am	2019-04-02 16:00:18.771074061 -0400
+++ libgeotiff-1.5.1/libxtiff/Makefile.am	2019-04-02 16:04:25.129084271 -0400
@@ -4,7 +4,7 @@
 TIFF_CFLAGS = @TIFF_INC@ -DHAVE_TIFF=1
 endif
 
-AM_CFLAGS = -I../ $(TIFF_CFLAGS) 
+AM_CFLAGS = -I$(top_srcdir) $(TIFF_CFLAGS) 
 
 libxtiff_la_SOURCES = xtiff.c
 
diff -Naur libgeotiff-1.5.1.orig/Makefile.am libgeotiff-1.5.1/Makefile.am
--- libgeotiff-1.5.1.orig/Makefile.am	2019-04-02 16:00:18.771074061 -0400
+++ libgeotiff-1.5.1/Makefile.am	2019-04-02 16:05:07.432645580 -0400
@@ -11,7 +11,7 @@
 TIFF_CFLAGS = @TIFF_INC@ -DHAVE_TIFF=1
 endif
 
-AM_CFLAGS = -I./libxtiff $(PROJ_CFLAGS) $(TIFF_CFLAGS)
+AM_CFLAGS = -I$(top_srcdir)/libxtiff $(PROJ_CFLAGS) $(TIFF_CFLAGS)
 
 include_HEADERS =   geotiff.h \
                     geotiffio.h \
@@ -52,7 +52,7 @@
                         geo_strtod.c \
                         geotiff_proj4.c
 
-libgeotiff_la_LDFLAGS = -version-info 5:1:0
+libgeotiff_la_LDFLAGS = -version-info 5:1:0 -no-undefined
 
 libgeotiff_la_LIBADD = libxtiff/libxtiff.la
 
