--- ../configure.ac	2022-04-20 03:47:13.000000000 +0800
+++ ./configure.ac	2022-10-02 12:31:13.726779200 +0800
@@ -171,7 +171,6 @@
 		 gen.tab/Makefile
 		 lib/Makefile
 		 bin/Makefile
-		 doc/Makefile
 		 test/Makefile
 		 test/unicode-conformance/Makefile])
 AC_OUTPUT
--- ../Makefile.am	2022-04-20 03:47:13.000000000 +0800
+++ ./Makefile.am	2022-10-02 12:50:40.140477800 +0800
@@ -4,7 +4,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 ## The order of subdirs is important, don't change without a reason.
-SUBDIRS = gen.tab lib bin doc test
+SUBDIRS = gen.tab lib bin test
 
 EXTRA_DIST = autogen.sh ChangeLog.old \
 	meson.build meson_options.txt bin/meson.build doc/meson.build \
