Append suffix to static library name only for MSVC.

diff -urN jsoncpp-1.9.6/src/lib_json/CMakeLists.txt.orig jsoncpp-1.9.6/src/lib_json/CMakeLists.txt
--- jsoncpp-1.9.6/src/lib_json/CMakeLists.txt.orig	2024-09-10 05:18:29.000000000 +0200
+++ jsoncpp-1.9.6/src/lib_json/CMakeLists.txt	2024-11-19 14:07:18.639721600 +0100
@@ -143,7 +143,7 @@
 
     # avoid name clashes on windows as the shared import lib is also named jsoncpp.lib
     if(NOT DEFINED STATIC_SUFFIX AND BUILD_SHARED_LIBS)
-        if (WIN32)
+        if (MSVC)
             set(STATIC_SUFFIX "_static")
         else()
             set(STATIC_SUFFIX "")
