--- jsoncpp-1.9.0.orig/src/lib_json/CMakeLists.txt	2019-07-09 20:27:14.587965000 -0400
+++ jsoncpp-1.9.0/src/lib_json/CMakeLists.txt	2019-07-09 20:30:37.073636200 -0400
@@ -75,6 +75,8 @@
 add_library(jsoncpp_lib ${PUBLIC_HEADERS} ${jsoncpp_sources})
 set_target_properties( jsoncpp_lib PROPERTIES VERSION ${JSONCPP_VERSION} SOVERSION ${JSONCPP_SOVERSION})
 set_target_properties( jsoncpp_lib PROPERTIES OUTPUT_NAME jsoncpp
+                        RUNTIME_OUTPUT_NAME jsoncpp-${JSONCPP_SOVERSION}
+                        ARCHIVE_OUTPUT_NAME jsoncpp
                         DEBUG_OUTPUT_NAME jsoncpp${DEBUG_LIBNAME_SUFFIX} )
 set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
