diff -Naur jsoncpp-1.9.4.orig/src/lib_json/CMakeLists.txt jsoncpp-1.9.4/src/lib_json/CMakeLists.txt
--- jsoncpp-1.9.4.orig/src/lib_json/CMakeLists.txt	2020-09-26 05:19:16.000000000 +0300
+++ jsoncpp-1.9.4/src/lib_json/CMakeLists.txt	2020-10-01 03:53:24.121074800 +0300
@@ -127,6 +127,8 @@
     add_library(${SHARED_LIB} SHARED ${PUBLIC_HEADERS} ${JSONCPP_SOURCES})
     set_target_properties(${SHARED_LIB} PROPERTIES
         OUTPUT_NAME jsoncpp
+        RUNTIME_OUTPUT_NAME jsoncpp-${PROJECT_SOVERSION}
+        ARCHIVE_OUTPUT_NAME jsoncpp
         VERSION ${PROJECT_VERSION}
         SOVERSION ${PROJECT_SOVERSION}
         POSITION_INDEPENDENT_CODE ON
