diff --git a/cmake/Findepoxy.cmake b/cmake/Findepoxy.cmake
index 6163859..3b2c300 100644
--- a/cmake/Findepoxy.cmake
+++ b/cmake/Findepoxy.cmake
@@ -32,7 +32,6 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
-if (NOT WIN32)
   find_package(PkgConfig)
   pkg_check_modules(PKG_epoxy QUIET epoxy)
 
@@ -45,4 +44,3 @@ if (NOT WIN32)
   find_package_handle_standard_args(epoxy DEFAULT_MSG epoxy_LIBRARY epoxy_INCLUDE_DIR)
 
   mark_as_advanced(epoxy_INCLUDE_DIR epoxy_LIBRARY)
-endif()
