diff -urN qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h
--- qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h	2015-08-02 00:11:35.019827900 +0100
+++ qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h	2015-08-02 00:13:59.523677600 +0100
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class ProjectPartDoNotExistException : public std::exception
diff -urN qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h
--- qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h	2015-08-02 00:11:35.019827900 +0100
+++ qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h	2015-08-02 00:13:55.365649500 +0100
@@ -33,6 +33,12 @@
 
 #include <filecontainer.h>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitDoesNotExistException : public std::exception
diff -urN qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h
--- qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h	2015-08-02 00:11:35.020328000 +0100
+++ qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h	2015-08-02 00:13:51.642676800 +0100
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitFileNotExitsException : public std::exception
diff -urN qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitisnullexception.h qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitisnullexception.h
--- qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitisnullexception.h	2015-08-02 00:11:35.021328100 +0100
+++ qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitisnullexception.h	2015-08-02 00:13:48.855322800 +0100
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitIsNullException : public std::exception
diff -urN qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h
--- qt-creator-opensource-src-3.5.0-beta1.orig/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h	2015-08-02 00:11:35.020828000 +0100
+++ qt-creator-opensource-src-3.5.0-beta1/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h	2015-08-02 00:13:45.302371700 +0100
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitParseErrorException : public std::exception
