--- qt-creator-opensource-src-3.2.0-rc1/src/plugins/clangcodemodel/clang_installation.pri.orig	2014-08-01 10:17:30.000000000 +0100
+++ qt-creator-opensource-src-3.2.0-rc1/src/plugins/clangcodemodel/clang_installation.pri	2014-08-17 20:14:44.611209200 +0100
@@ -50,7 +50,7 @@
     error("Cannot find clang shared library at $${LLVM_INSTALL_DIR}")
 }
 
-win32 {
+win32:!win32-g++ {
     LLVM_INCLUDEPATH = "$$LLVM_INSTALL_DIR/include"
     CLANG_LIB_PATH = $$findClangOnWindows()
     CLANG_LIB = clang
@@ -61,7 +61,7 @@
     LLVM_VERSION = 3.4
 }
 
-unix {
+unix|win32-g++ {
     LLVM_CONFIG = $$findLLVMConfig()
 
     LLVM_VERSION = $$system($$LLVM_CONFIG --version)
@@ -93,7 +93,33 @@
     }
 
     LLVM_LIBS = -L$${LLVM_LIBDIR}
-    LLVM_LIBS += -l$${CLANG_LIB}
+
+    LLVM_STATIC = $$(LLVM_STATIC)
+    isEqual($$(LLVM_STATIC), 0): {
+        LLVM_LIBS += -l$${CLANG_LIB}
+    } else {
+        # Since there is no clang-config ..
+        LLVM_LIBS += -Wl,--start-group
+        CLANG_LIBS = $$files($${LLVM_LIBDIR}/libclang*)
+        for (CLANG_LIB, CLANG_LIBS) {
+            CLANG_LIB = $$basename(CLANG_LIB)
+            CLANG_LIB ~= s,^$$QMAKE_PREFIX_STATICLIB,,
+            CLANG_LIB ~= s,\.$$QMAKE_EXTENSION_STATICLIB\$,,
+            LLVM_LIBS += -l$${CLANG_LIB}
+        }
+        # We might want to try "$$LLVM_CONFIG --libs analysis" instead.
+        LLVM_LIBS += $$system($$LLVM_CONFIG --libs)
+        LLVM_LIBS += $$system($$LLVM_CONFIG --system-libs)
+        LLVM_LIBS += -Wl,--end-group
+    }
+}
+
+# llvm-config --ldflags could be used here, but
+# that lists lots of unnecessary libraries:
+# llvm-config.exe --ldflags
+# -LC:\msys64\mingw64/lib  -lz -lffi -lncurses -lshell32 -lpsapi -limagehlp -lm
+win32 {
+    LLVM_LIBS += -limagehlp
 }
 
 DEFINES += CLANG_VERSION=\\\"$${LLVM_VERSION}\\\"
