--- qt-creator-opensource-src-3.1.0-rc1/src/plugins/qmakeprojectmanager/qmakeproject.cpp.orig	2014-04-14 13:11:22.596424400 +0000
+++ qt-creator-opensource-src-3.1.0-rc1/src/plugins/qmakeprojectmanager/qmakeproject.cpp	2014-04-14 13:12:14.041424400 +0000
@@ -1001,7 +1001,7 @@
         // It is important to override the spec file only for the creator evaluator,
         // and not the qmake buildstep used to build the app (as we use the makefiles).
         const char IOSQT[] = "Qt4ProjectManager.QtVersion.Ios"; // from Ios::Constants
-        if (qtVersion->type() == QLatin1String(IOSQT))
+        if (qtVersion && qtVersion->type() == QLatin1String(IOSQT))
             m_qmakeGlobals->xqmakespec = QLatin1String("macx-xcode");
     }
     ++m_qmakeGlobalsRefCnt;
