--- qt-creator-opensource-src-3.5.0-beta1/src/libs/3rdparty/sqlite/sqlite.pri.orig	2015-07-07 14:02:27.000000000 +0100
+++ qt-creator-opensource-src-3.5.0-beta1/src/libs/3rdparty/sqlite/sqlite.pri	2015-07-08 18:22:04.719461900 +0100
@@ -7,7 +7,7 @@
 
 SOURCES += $$PWD/sqlite3.c
 
-win32:DEFINES += SQLITE_API=__declspec(dllexport)
+win32:DEFINES += $$shell_quote(SQLITE_API=__declspec(dllexport))
 unix:DEFINES += SQLITE_API=\"__attribute__((visibility(\\\"default\\\")))\"
 
 gcc {
