diff -urN qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsbuildstep.cpp qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsbuildstep.cpp
--- qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsbuildstep.cpp	2014-08-19 12:50:15.621530300 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsbuildstep.cpp	2014-08-19 13:30:13.985530300 +0100
@@ -221,7 +221,7 @@
     // Report errors:
     foreach (const qbs::ErrorItem &item, m_job->error().items())
         createTaskAndOutput(ProjectExplorer::Task::Error, item.description(),
-                            item.codeLocation().fileName(), item.codeLocation().line());
+                            item.codeLocation().filePath(), item.codeLocation().line());
 
     QbsProject *pro = static_cast<QbsProject *>(project());
 
diff -urN qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbscleanstep.cpp qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbscleanstep.cpp
--- qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbscleanstep.cpp	2014-08-19 12:50:15.630530300 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbscleanstep.cpp	2014-08-19 13:30:13.989530300 +0100
@@ -181,7 +181,7 @@
     // Report errors:
     foreach (const qbs::ErrorItem &item, m_job->error().items()) {
         createTaskAndOutput(ProjectExplorer::Task::Error, item.description(),
-                            item.codeLocation().fileName(), item.codeLocation().line());
+                            item.codeLocation().filePath(), item.codeLocation().line());
     }
 
     QTC_ASSERT(m_fi, return);
diff -urN qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsinstallstep.cpp qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsinstallstep.cpp
--- qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsinstallstep.cpp	2014-08-19 12:50:15.608530300 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsinstallstep.cpp	2014-08-19 13:30:13.992530300 +0100
@@ -191,7 +191,7 @@
     // Report errors:
     foreach (const qbs::ErrorItem &item, m_job->error().items()) {
         createTaskAndOutput(ProjectExplorer::Task::Error, item.description(),
-                            item.codeLocation().fileName(), item.codeLocation().line());
+                            item.codeLocation().filePath(), item.codeLocation().line());
     }
 
     QTC_ASSERT(m_fi, return);
diff -urN qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbslogsink.cpp qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbslogsink.cpp
--- qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbslogsink.cpp	2014-08-19 12:50:15.666530300 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbslogsink.cpp	2014-08-19 13:30:13.995530300 +0100
@@ -72,7 +72,7 @@
     foreach (const qbs::ErrorItem &item, warning.items())
         emit newTask(ProjectExplorer::Task(ProjectExplorer::Task::Warning,
                                            item.description(),
-                                           Utils::FileName::fromString(item.codeLocation().fileName()),
+                                           Utils::FileName::fromString(item.codeLocation().filePath()),
                                            item.codeLocation().line(),
                                            ProjectExplorer::Constants::TASK_CATEGORY_BUILDSYSTEM));
 }
diff -urN qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsnodes.cpp qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsnodes.cpp
--- qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsnodes.cpp	2014-08-19 12:50:15.616530300 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsnodes.cpp	2014-08-19 13:30:13.999530300 +0100
@@ -348,7 +348,7 @@
 
     setIcon(m_groupIcon);
 
-    QbsFileNode *idx = new QbsFileNode(grp->location().fileName(),
+    QbsFileNode *idx = new QbsFileNode(grp->location().filePath(),
                                        ProjectExplorer::ProjectFileType, false,
                                        grp->location().line());
     addFileNodes(QList<ProjectExplorer::FileNode *>() << idx);
@@ -428,7 +428,7 @@
     m_productPath = productPath;
     m_qbsGroupData = grp;
 
-    setPath(grp->location().fileName());
+    setPath(grp->location().filePath());
     setDisplayName(grp->name());
 
     QbsFileNode *idx = 0;
@@ -438,7 +438,7 @@
             break;
     }
     QTC_ASSERT(idx, return);
-    idx->setPathAndLine(grp->location().fileName(), grp->location().line());
+    idx->setPathAndLine(grp->location().filePath(), grp->location().line());
 
     setupFiles(this, grp->allFilePaths(), productPath, updateExisting);
 
@@ -545,14 +545,14 @@
 // --------------------------------------------------------------------
 
 QbsProductNode::QbsProductNode(const qbs::ProductData &prd) :
-    QbsBaseProjectNode(prd.location().fileName())
+    QbsBaseProjectNode(prd.location().filePath())
 {
     if (m_productIcon.isNull())
         m_productIcon = generateIcon(QString::fromLatin1(Constants::QBS_PRODUCT_OVERLAY_ICON));
 
     setIcon(m_productIcon);
 
-    ProjectExplorer::FileNode *idx = new QbsFileNode(prd.location().fileName(),
+    ProjectExplorer::FileNode *idx = new QbsFileNode(prd.location().filePath(),
                                                      ProjectExplorer::ProjectFileType, false,
                                                      prd.location().line());
     addFileNodes(QList<ProjectExplorer::FileNode *>() << idx);
@@ -627,8 +627,8 @@
     bool updateExisting = productWasEnabled != productIsEnabled;
 
     setDisplayName(prd.name());
-    setPath(prd.location().fileName());
-    const QString &productPath = QFileInfo(prd.location().fileName()).absolutePath();
+    setPath(prd.location().filePath());
+    const QString &productPath = QFileInfo(prd.location().filePath()).absolutePath();
 
     // Find the QbsFileNode we added earlier:
     QbsFileNode *idx = 0;
@@ -638,7 +638,7 @@
             break;
     }
     QTC_ASSERT(idx, return);
-    idx->setPathAndLine(prd.location().fileName(), prd.location().line());
+    idx->setPathAndLine(prd.location().filePath(), prd.location().line());
 
     QList<ProjectExplorer::ProjectNode *> toAdd;
     QList<ProjectExplorer::ProjectNode *> toRemove = subProjectNodes();
@@ -726,7 +726,7 @@
     foreach (const qbs::ProjectData &subData, prjData.subProjects()) {
         QbsProjectNode *qn = findProjectNode(subData.name());
         if (!qn) {
-            QbsProjectNode *subProject = new QbsProjectNode(subData.location().fileName());
+            QbsProjectNode *subProject = new QbsProjectNode(subData.location().filePath());
             subProject->update(subData);
             toAdd << subProject;
         } else {
@@ -831,13 +831,13 @@
 static QSet<QString> referencedBuildSystemFiles(const qbs::ProjectData &data)
 {
     QSet<QString> result;
-    result.insert(data.location().fileName());
+    result.insert(data.location().filePath());
     foreach (const qbs::ProjectData &subProject, data.subProjects())
         result.unite(referencedBuildSystemFiles(subProject));
     foreach (const qbs::ProductData &product, data.products()) {
-        result.insert(product.location().fileName());
+        result.insert(product.location().filePath());
         foreach (const qbs::GroupData &group, product.groups())
-            result.insert(group.location().fileName());
+            result.insert(group.location().filePath());
     }
 
     return result;
diff -urN qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsproject.cpp qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsproject.cpp
--- qt-creator-opensource-src-3.2.0.orig/src/plugins/qbsprojectmanager/qbsproject.cpp	2014-08-19 12:50:15.650530300 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/qbsprojectmanager/qbsproject.cpp	2014-08-19 13:30:14.004530300 +0100
@@ -165,14 +165,14 @@
 
 static void collectFilesForProject(const qbs::ProjectData &project, QSet<QString> &result)
 {
-    result.insert(project.location().fileName());
+    result.insert(project.location().filePath());
     foreach (const qbs::ProductData &prd, project.products()) {
         foreach (const qbs::GroupData &grp, prd.groups()) {
             foreach (const QString &file, grp.allFilePaths())
                 result.insert(file);
-            result.insert(grp.location().fileName());
+            result.insert(grp.location().filePath());
         }
-        result.insert(prd.location().fileName());
+        result.insert(prd.location().filePath());
     }
     foreach (const qbs::ProjectData &subProject, project.subProjects())
         collectFilesForProject(subProject, result);
@@ -229,7 +229,7 @@
 {
     QTC_ASSERT(m_qbsProject.isValid(), return false);
     QStringList allPaths = groupData.allFilePaths();
-    const QString productFilePath = productData.location().fileName();
+    const QString productFilePath = productData.location().filePath();
     ChangeExpector expector(productFilePath, m_qbsDocuments);
     foreach (const QString &path, filePaths) {
         qbs::ErrorInfo err = m_qbsProject.addFiles(productData, groupData, QStringList() << path);
@@ -254,7 +254,7 @@
 {
     QTC_ASSERT(m_qbsProject.isValid(), return false);
     QStringList allPaths = groupData.allFilePaths();
-    const QString productFilePath = productData.location().fileName();
+    const QString productFilePath = productData.location().filePath();
     ChangeExpector expector(productFilePath, m_qbsDocuments);
     foreach (const QString &path, filePaths) {
         qbs::ErrorInfo err
@@ -546,7 +546,7 @@
     foreach (const qbs::ErrorItem &item, e.items())
         TaskHub::addTask(Task::Error, item.description(),
                          ProjectExplorer::Constants::TASK_CATEGORY_BUILDSYSTEM,
-                         FileName::fromString(item.codeLocation().fileName()),
+                         FileName::fromString(item.codeLocation().filePath()),
                          item.codeLocation().line());
 
 }
@@ -690,7 +690,7 @@
             part->project = this;
             part->displayName = grp.name();
             part->projectFile = QString::fromLatin1("%1:%2:%3")
-                    .arg(grp.location().fileName())
+                    .arg(grp.location().filePath())
                     .arg(grp.location().line())
                     .arg(grp.location().column());
             part->evaluateToolchain(ToolChainKitInformation::toolChain(k),
@@ -761,7 +761,7 @@
             if (!ta.isExecutable())
                 continue;
             applications.list << ProjectExplorer::BuildTargetInfo(Utils::FileName::fromString(ta.filePath()),
-                    Utils::FileName::fromString(productData.location().fileName()));
+                    Utils::FileName::fromString(productData.location().filePath()));
         }
     }
     activeTarget()->setApplicationTargets(applications);
