--- qt-creator-opensource-src-3.2.0/src/plugins/projectexplorer/abi.cpp.orig	2014-08-18 08:52:02.000000000 +0100
+++ qt-creator-opensource-src-3.2.0/src/plugins/projectexplorer/abi.cpp	2014-09-02 20:05:53.652181700 +0100
@@ -466,6 +466,7 @@
         } else if (p == QLatin1String("i386") || p == QLatin1String("i486") || p == QLatin1String("i586")
                    || p == QLatin1String("i686") || p == QLatin1String("x86")) {
             arch = Abi::X86Architecture;
+            width = 32;
         } else if (p.startsWith(QLatin1String("arm"))) {
             arch = Abi::ArmArchitecture;
             width = 32;
