--- qt-creator-opensource-src-3.2.0-rc1/qtcreator.pro.orig	2014-08-01 10:17:30.000000000 +0100
+++ qt-creator-opensource-src-3.2.0-rc1/qtcreator.pro	2014-08-18 13:59:45.446170300 +0100
@@ -67,6 +67,9 @@
     cache(QBS_RELATIVE_SEARCH_PATH)
     QBS_CONFIG_ADDITION = qbs_no_dev_install qbs_enable_project_file_updates
     cache(CONFIG, add, QBS_CONFIG_ADDITION)
+} else:minQtVersion(5, 0, 0) {
+    QBS_CONFIG_ADDITION = qbs_enable_project_file_updates
+    cache(CONFIG, add, QBS_CONFIG_ADDITION)
 }
 
 contains(QT_ARCH, i386): ARCHITECTURE = x86
