--- soundtouch/source/SoundTouch/Makefile.am.orig	2012-09-16 18:22:55 +0400
+++ soundtouch/source/SoundTouch/Makefile.am	2012-09-16 18:23:19 +0400
@@ -39,6 +39,7 @@
 noinst_LTLIBRARIES=libSoundTouchMMX.la libSoundTouchSSE.la
 libSoundTouchMMX_la_SOURCES=mmx_optimized.cpp
 libSoundTouchSSE_la_SOURCES=sse_optimized.cpp
+libSoundTouch_la_LDFLAGS=-no-undefined
 
 # We enable optimizations by default.
 # If MMX is supported compile with -mmmx.
