--- soundtouch/config/am_include.mk.orig	2012-12-28 20:56:45.000000000 +0000
+++ soundtouch/config/am_include.mk	2013-07-31 15:01:37.806263400 +0000
@@ -28,7 +28,3 @@
 ## in case a user decides to build in a separate directory from the base package
 ## directory. Using absolute, or relative paths is a bad idea.
 INCLUDES=-I$(top_srcdir)/include
-
-
-# doc directory
-pkgdocdir=$(prefix)/doc/@PACKAGE@
--- soundtouch/Makefile.am.orig	2012-12-28 20:56:45.000000000 +0000
+++ soundtouch/Makefile.am	2013-07-31 15:06:06.219347500 +0000
@@ -28,7 +28,7 @@
 
 
 # list files that are documentation to be packaged in a release tarball and installed
-pkgdoc_DATA=COPYING.TXT README.html
+dist_doc_DATA=COPYING.TXT README.html
 
 # extra data files that are to be pacakged in a release tarball and installed into the data directory
 #pkgdata_DATA=
