--- soundtouch/source/SoundStretch/Makefile.am.orig	2024-03-29 19:49:53.000000000 +0100
+++ soundtouch/source/SoundStretch/Makefile.am	2024-10-14 23:35:06.794528300 +0200
@@ -44,7 +44,7 @@
 #soundstretch_LDFLAGS=-s
 
 ## additional compiler flags
-soundstretch_CXXFLAGS=$(AM_CXXFLAGS)
+soundstretch_CXXFLAGS=$(AM_CXXFLAGS) -municode
 
 #clean-local:
 #	-rm -f additional-files-to-remove-on-make-clean
