--- libproxy-0.4.15/bindings/python/python2/CMakeLists.txt.orig	2015-07-23 12:46:27.038511800 +0300
+++ libproxy-0.4.15/bindings/python/python2/CMakeLists.txt	2015-07-23 12:47:10.174197800 +0300
@@ -14,7 +14,7 @@
 
     execute_process(COMMAND
 		    ${PYTHON2_EXECUTABLE}
-		    -c "import distutils.sysconfig ; print (distutils.sysconfig.get_python_lib(plat_specific=0))"
+		    -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib(prefix='')"
 		    OUTPUT_VARIABLE PYTHON2_SITEPKG_DIR
 		    OUTPUT_STRIP_TRAILING_WHITESPACE)
   endif()
--- libproxy-0.4.15/bindings/python/python3/CMakeLists.txt.orig	2015-07-23 12:46:27.038511800 +0300
+++ libproxy-0.4.15/bindings/python/python3/CMakeLists.txt	2015-07-23 12:47:10.174197800 +0300
@@ -14,7 +14,7 @@
 
     execute_process(COMMAND
 		    ${PYTHON3_EXECUTABLE}
-		    -c "import distutils.sysconfig ; print (distutils.sysconfig.get_python_lib(plat_specific=0))"
+		    -c "import distutils.sysconfig ; print (distutils.sysconfig.get_python_lib(prefix=''))"
 		    OUTPUT_VARIABLE PYTHON3_SITEPKG_DIR
 		    OUTPUT_STRIP_TRAILING_WHITESPACE)
   endif()
