diff -bur jack2-1.9.22-orig/windows/JackAtomic_os.h jack2-1.9.22/windows/JackAtomic_os.h
--- jack2-1.9.22-orig/windows/JackAtomic_os.h	2026-01-04 15:19:27.297266800 -0700
+++ jack2-1.9.22/windows/JackAtomic_os.h	2026-01-04 15:31:26.054431100 -0700
@@ -55,19 +55,12 @@
 
 #else
 
-#define LOCK "lock ; "
+#include <windows.h>
 
 static inline char CAS(volatile UInt32 value, UInt32 newvalue, volatile void* addr)
 {
-    register char ret;
-    __asm__ __volatile__ (
-        "# CAS \n\t"
-        LOCK "cmpxchg %2, (%1) \n\t"
-        "sete %0               \n\t"
-        : "=a" (ret)
-        : "c" (addr), "d" (newvalue), "a" (value)
-        );
-    return ret;
+    LONG old = InterlockedCompareExchange((LONG*)addr, newvalue, value);
+    return (old == value) ? 1 : 0;
 }
 
 #endif
