--- aria2-1.18.10/src/Makefile.am.orig	2015-04-02 22:02:29.635051500 +0200
+++ aria2-1.18.10/src/Makefile.am	2015-04-02 22:02:39.234635300 +0200
@@ -713,9 +713,9 @@
 
 libaria2_la_SOURCES = $(SRCS)
 
-libaria2_la_LIBADD = @WSLAY_LIBS@
+libaria2_la_LIBADD = @WSLAY_LIBS@ @LTLIBINTL@
 
-LDADD = libaria2.la @LIBINTL@ @ALLOCA@ #-lprofiler
+LDADD = libaria2.la @ALLOCA@ #-lprofiler
 #aria2c_LDFLAGS = -pg
 AM_CPPFLAGS =  -Wall\
 	-I$(top_srcdir)/lib -I$(top_srcdir)/intl\
