diff -bur exempi-orig/public/include/XMPCommon/XMPCommonDefines.h exempi/public/include/XMPCommon/XMPCommonDefines.h
--- exempi-orig/public/include/XMPCommon/XMPCommonDefines.h	2024-06-15 11:07:50.176832300 -0600
+++ exempi/public/include/XMPCommon/XMPCommonDefines.h	2024-06-15 11:08:41.376698000 -0600
@@ -87,7 +87,7 @@
 	// =========================
 	#if XMP_WinBuild
 		#define SUPPORT_SHARED_POINTERS_WITH_ALLOCATORS 1
-		#if _MSC_VER <= 1600
+		#if _MSC_VER <= 1600 && !defined(__MINGW64__)
 			#define SUPPORT_STD_ATOMIC_IMPLEMENTATION 0
 			#define SUPPORT_SHARED_POINTERS_IN_TR1 1
 			#define SUPPORT_SHARED_POINTERS_IN_STD 0
diff -bur exempi-orig/source/SuppressSAL.h exempi/source/SuppressSAL.h
--- exempi-orig/source/SuppressSAL.h	2024-06-15 11:07:50.211863200 -0600
+++ exempi/source/SuppressSAL.h	2024-06-15 11:09:25.144530500 -0600
@@ -17,7 +17,7 @@
 #ifndef _H_SuppressSAL
 #define _H_SuppressSAL
 
-#if !defined(_WIN32) && !defined(_WIN64) /* The following definition is applicable only for non-windows platform */
+#if defined(__MINGW64__) || (!defined(_WIN32) && !defined(_WIN64)) /* The following definition is applicable only for non-msvc platform */
 
 #define _In_
 #define _In_opt_
