--- libpqxx-8.0.2.orig/include/pqxx/internal/conversions.hxx	2026-07-18 19:47:15.000000000 +0300
+++ libpqxx-8.0.2/include/pqxx/internal/conversions.hxx	2026-09-22 22:32:56.877258500 +0300
@@ -46,12 +46,12 @@
 /** Don't worry about the exact parameter types: the sizes will be reasonably
  * small, and nonnegative.
  */
-PQXX_LIBEXPORT PQXX_COLD std::string
+PQXX_COLD PQXX_LIBEXPORT std::string
 state_buffer_overrun(int have_bytes, int need_bytes);
 
 
 template<typename HAVE, typename NEED>
-PQXX_INLINE_COV PQXX_COLD inline std::string
+PQXX_COLD PQXX_INLINE_COV inline std::string
 state_buffer_overrun(HAVE have_bytes, NEED need_bytes)
 {
   return state_buffer_overrun(
