--- libsquish-1.14/CMakeLists.txt.orig	2017-02-02 08:34:05.400385200 +0300
+++ libsquish-1.14/CMakeLists.txt	2017-02-02 08:36:07.932436600 +0300
@@ -98,7 +98,17 @@
 
 INSTALL(
     TARGETS squish
+    RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
     PUBLIC_HEADER DESTINATION include
     )
+
+IF (BUILD_SQUISH_EXTRA)
+    IF (PNG_FOUND)
+        INSTALL(
+            TARGETS squishpng
+            DESTINATION bin
+        )
+    ENDIF (PNG_FOUND)
+ENDIF (BUILD_SQUISH_EXTRA)
\ В конце файла нет новой строки
