--- a/src/wxterminal/wxt_gui.cpp
+++ b/src/wxterminal/wxt_gui.cpp
@@ -638,7 +638,7 @@
 
 		RECT rect;
 		rect.left = rect.top = 0;
-		unsigned dpi = GetDPI();
+		unsigned dpi = GetDPI().GetWidth();
 		rect.right  = MulDiv(panel->plot.device_xmax, dpi, 10);
 		rect.bottom = MulDiv(panel->plot.device_ymax, dpi, 10);
 		HDC hmf = CreateEnhMetaFileW(NULL, fullpathFilename.wc_str(), &rect, NULL);
@@ -708,7 +708,7 @@
 		wxSize ppi = wxdc->GetPPI();
 		unsigned dpi = 96;
 #ifdef _WIN32
-		dpi = GetDPI();
+		dpi = GetDPI().GetWidth();
 #endif
 		double scaleX = ppi.GetWidth() / (double) dpi;
 		double scaleY = ppi.GetHeight() / (double) dpi;
