--- a/docs/doc2web.c
+++ b/docs/doc2web.c
@@ -582,10 +582,12 @@
 			strncpy(knownfile, path, PATH_MAX-1);
 			strncat(knownfile, sectionname, PATH_MAX-strlen(knownfile)-6);
 			strcat(knownfile,".html");
+#ifndef _WIN32
 			if (symlink(locfile, knownfile)) {
 			    perror("doc2web: Can't create symlink ");
 			    exit(EXIT_FAILURE);
 			}
+#endif
 			fprintf(stderr,"Creating symlink %s for %s\n", knownfile, newfile);
 
 			/* Start the new file */
