diff -ru gnuplot-main.orig/term/lua/gnuplot-tikz.lua gnuplot-main/term/lua/gnuplot-tikz.lua
--- gnuplot-main.orig/term/lua/gnuplot-tikz.lua	2026-01-05 16:02:15.698157126 +0100
+++ gnuplot-main/term/lua/gnuplot-tikz.lua	2026-01-05 16:03:05.620939358 +0100
@@ -2543,9 +2543,9 @@
 term_help = function(helptext)
   local w
   for w in string.gmatch(helptext, "([^\n]*)\n") do
-    w = string.gsub(w, "\\", "\\\\")
-    w = string.gsub(w, "\"", "\\\"")
-    io.write('"'..w.."\",\n")
+    w0 = string.gsub(w, "\\", "\\\\")
+    w0 = string.gsub(w0, "\"", "\\\"")
+    io.write('"'..w0.."\",\n")
   end
 --[[  
   local out = string.gsub(helptext, "\n", "\",\n\"")
