diff -bur superlu_mt-4.0.2-orig/TESTING/MATGEN/CMakeLists.txt superlu_mt-4.0.2/TESTING/MATGEN/CMakeLists.txt
--- superlu_mt-4.0.2-orig/TESTING/MATGEN/CMakeLists.txt	2026-04-14 03:36:01.777320900 -0600
+++ superlu_mt-4.0.2/TESTING/MATGEN/CMakeLists.txt	2026-04-14 03:36:07.282674600 -0600
@@ -107,4 +107,4 @@
 endif()
 
 add_library(matgen ${sources})
-target_link_libraries(matgen superlu_mt${PLAT})
+target_link_libraries(matgen superlu_mt${PLAT} ${BLAS_LIB})
