--- ./configure.ac.orig	2014-10-17 21:04:58.784468100 +0200
+++ ./configure.ac	2014-10-17 21:05:12.577242800 +0200
@@ -8,24 +8,7 @@
 
 AM_PATH_PYTHON
 PKG_CHECK_MODULES(POSTR,
-		  pygtk-2.0
-		  nautilus-python >= 0.6.1)
-
-AC_ARG_WITH(nautilus-extension-dir,
-	    [AS_HELP_STRING([--with-nautilus-extension-dir],
-		    [specify the nautilus extension directory])])
-
-AC_MSG_CHECKING([for nautilus extension directory])
-if test -n "$with_nautilus_extension_dir"; then
-    NAUTILUS_EXTENSION_DIR=$with_nautilus_extension_dir
-else
-dnl nautilus-python provides the pythondir variable, however its
-dnl path is absolute and the prefix can not be overrided (aarrgh!).
-dnl So, we have to use a workaround.
-  test "x$prefix" != "xNONE" && NAUTILUS_EXTENSION_DIR=`$PKG_CONFIG --variable=pythondir nautilus-python | sed -n -e 's,/$,,' -e '/.*\/lib\/nautilus\/.*\/python$/{s,.*/\(lib/nautilus/.*/python\)$,\1,;p;q;}'`
-  NAUTILUS_EXTENSION_DIR=[$prefix/$NAUTILUS_EXTENSION_DIR]
-fi
-AC_SUBST([nautilusextensiondir], [$NAUTILUS_EXTENSION_DIR])
+		  pygtk-2.0)
 
 dnl i10n support / gettext
 IT_PROG_INTLTOOL([0.40.0])
