--- lzo-2.06/Makefile.am	2011-08-12 18:02:59.000000000 +0200
+++ lzo-2.06/Makefile.am	2011-11-16 23:22:59.312500000 +0100
@@ -72,7 +72,7 @@ pkginclude_HEADERS = \
 
 lib_LTLIBRARIES += src/liblzo2.la
 
-src_liblzo2_la_LDFLAGS = -version-info 2:0:0 $(AM_LDFLAGS)
+src_liblzo2_la_LDFLAGS = -version-info 2:0:0 -no-undefined $(AM_LDFLAGS)
 
 src_liblzo2_la_SOURCES = \
     src/lzo1.c src/lzo1_99.c src/lzo1a.c src/lzo1a_99.c src/lzo1b_1.c \
--- lzo-2.06/Makefile.in.orig	2011-08-12 20:02:59 +0400
+++ lzo-2.06/Makefile.in	2012-05-15 12:59:56 +0400
@@ -391,7 +391,7 @@
     include/lzo/lzo_asm.h include/lzo/lzoconf.h include/lzo/lzodefs.h \
     include/lzo/lzoutil.h
 
-src_liblzo2_la_LDFLAGS = -version-info 2:0:0 $(AM_LDFLAGS)
+src_liblzo2_la_LDFLAGS = -version-info 2:0:0 -no-undefined $(AM_LDFLAGS)
 src_liblzo2_la_SOURCES = \
     src/lzo1.c src/lzo1_99.c src/lzo1a.c src/lzo1a_99.c src/lzo1b_1.c \
     src/lzo1b_2.c src/lzo1b_3.c src/lzo1b_4.c src/lzo1b_5.c src/lzo1b_6.c \
