From d2df61dcde8079e1f7e5f8bcffefefa10277540f Mon Sep 17 00:00:00 2001
From: Paul Eipper <lkraider+git@gmail.com>
Date: Sun, 2 Nov 2014 21:17:53 -0200
Subject: [PATCH] Fix pid_t redefinition error

---
 mingw.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/mingw.h b/mingw.h
index f284583..23a0f09 100644
--- a/mingw.h
+++ b/mingw.h
@@ -153,8 +153,6 @@ struct pollfd {
 /* Winsock uses int instead of the usual socklen_t */
 typedef int socklen_t;
 
-typedef int pid_t;
-
 /* Function prototypes for functions in mingw.c */
 unsigned int win32_sleep(unsigned int);
 int     win32_inet_aton(const char *, struct in_addr *);
