From: Christoph Egger <christoph@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=690006
Description: Revert broken patch from release 1.1.6

diff --git a/src/drivers/fluid_oss.c b/src/drivers/fluid_oss.c
index 840dad1..40761db 100644
--- a/src/drivers/fluid_oss.c
+++ b/src/drivers/fluid_oss.c
@@ -45,7 +45,7 @@
 #define BUFFER_LENGTH 512
 
 // Build issue on some systems (OSS 4.0)?
-#ifdef SNDCTL_DSP_CHANNELS
+#ifndef SOUND_PCM_WRITE_CHANNELS
   #define SOUND_PCM_WRITE_CHANNELS        SNDCTL_DSP_CHANNELS
 #endif
