--- fluidsynth-1.1.6/src/Makefile.am.orig	2012-08-16 04:01:13.000000000 +0000
+++ fluidsynth-1.1.6/src/Makefile.am	2014-02-10 04:29:46.259284600 +0000
@@ -173,6 +173,7 @@
   -I$(top_srcdir)/src/utils \
   -I$(top_srcdir)/src/sfloader \
   -I$(top_srcdir)/src/bindings \
+  -I$(top_builddir)/include \
   $(LASH_CFLAGS) $(LADCCA_CFLAGS) \
   $(READLINE_CFLAGS) $(JACK_CFLAGS) $(ALSA_CFLAGS) $(PULSE_CFLAGS) \
   $(PORTAUDIO_CFLAGS) $(DART_CFLAGS) $(GLIB_CFLAGS) $(LIBSNDFILE_CFLAGS) \
