--- fluidsynth-1.1.6/src/bindings/fluid_rtkit.c.orig	2012-08-16 04:01:13.000000000 +0000
+++ fluidsynth-1.1.6/src/bindings/fluid_rtkit.c	2014-03-22 12:43:34.079336600 +0000
@@ -27,7 +27,8 @@
 
 #include "fluidsynth_priv.h"
 
-#ifdef DBUS_SUPPORT
+/* This code is never used on W32 */
+#if defined(DBUS_SUPPORT) && !defined(_WIN32)
 
 #include <errno.h>
 
@@ -292,8 +293,6 @@
         return ret;
 }
 
-#else
-
 int rtkit_make_realtime(DBusConnection *connection, pid_t thread, int priority) {
         return -ENOTSUP;
 }
@@ -347,7 +346,7 @@
 		return res;
 	}
         dbus_error_free(&error);
-	
+
 	/* Make sure we don't fail by wanting too much */
 	max_prio = rtkit_get_max_realtime_priority(conn);
 	if (max_prio < 0) 
@@ -372,7 +371,6 @@
 		setrlimit(RLIMIT_RTTIME, &old_limit);
 	}
 	MAKE_REALTIME_RETURN(res);
-	
 }
 
 #endif
