--- fluidsynth-2.4.8/CMakeLists.txt.orig	2025-09-01 06:34:55.853185300 +0200
+++ fluidsynth-2.4.8/CMakeLists.txt	2025-09-01 06:35:49.041888500 +0200
@@ -199,10 +199,6 @@
 if ( CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID STREQUAL "Intel" )
   # If we ever bump to CMake 3.29+, replace this with CMAKE_LANG_COMPILER_LINKER_ID
   if (NOT (CMAKE_SYSTEM_NAME MATCHES "Darwin|OS2|Emscripten|SunOS") )
-    set ( CMAKE_EXE_LINKER_FLAGS
-          "${CMAKE_EXE_LINKER_FLAGS} -Wl,--as-needed" )
-    set ( CMAKE_SHARED_LINKER_FLAGS
-          "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined" )
   endif ()
 
   # define some warning flags
