diff -urN numpy-1.10.2.orig/numpy/distutils/misc_util.py numpy-1.10.2/numpy/distutils/misc_util.py
--- numpy-1.10.2.orig/numpy/distutils/misc_util.py	2016-01-04 00:38:06.433855300 +0000
+++ numpy-1.10.2/numpy/distutils/misc_util.py	2016-01-04 00:40:27.356823400 +0000
@@ -396,6 +396,10 @@
             return True
         if os.environ.get('MSYSTEM', '')=='MINGW32':
             return True
+        if os.environ.get('MSYSTEM', '')=='MINGW64':
+            return True
+        if 'GCC' in sys.version:
+            return True
     return False
 
 def msvc_runtime_library():
