--- numpy-1.10.0/numpy/distutils/misc_util.py.orig	2015-10-08 15:20:14.760758300 +0300
+++ numpy-1.10.0/numpy/distutils/misc_util.py	2015-10-08 15:21:22.951938500 +0300
@@ -13,6 +13,7 @@
 
 import distutils
 from distutils.errors import DistutilsError
+from sysconfig import _POSIX_BUILD
 try:
     from threading import local as tlocal
 except ImportError:
@@ -385,6 +386,10 @@
             return True
         if os.environ.get('MSYSTEM', '')=='MINGW32':
             return True
+        if os.environ.get('MSYSTEM', '')=='MINGW64':
+            return True
+        if _POSIX_BUILD:
+            return True
     return False
 
 def msvc_runtime_library():
