--- numpy-1.10.2/numpy/distutils/fcompiler/gnu.py.orig	2014-08-28 15:49:10.742000000 +0400
+++ numpy-1.10.2/numpy/distutils/fcompiler/gnu.py	2014-08-28 15:50:44.903600000 +0400
@@ -63,7 +63,7 @@
                     return ('gfortran', m.group(1))
         else:
             # Output probably from --version, try harder:
-            m = re.search(r'GNU Fortran\s+95.*?([0-9-.]+)', version_string)
+            m = re.search(r'GNU Fortran.*?\-?(\d*(?:\.\d+)+)', version_string)
             if m:
                 return ('gfortran', m.group(1))
             m = re.search(
