--- numpy-1.9.0/numpy/core/tests/test_multiarray.py.orig	2014-09-07 10:57:48.000000000 +0200
+++ numpy-1.9.0/numpy/core/tests/test_multiarray.py	2014-09-22 20:42:55.177362800 +0200
@@ -4708,7 +4708,7 @@
         """Test workarounds for 32-bit limited fwrite, fseek, and ftell
         calls in windows. These normally would hang doing something like this.
         See http://projects.scipy.org/numpy/ticket/1660"""
-        if sys.platform != 'win32':
+        if sys.platform != 'win32' or "MSYSTEM" in os.environ:
             return
         try:
             # before workarounds, only up to 2**32-1 worked
