diff -urN numpy-1.10.2.orig/numpy/distutils/fcompiler/compaq.py numpy-1.10.2/numpy/distutils/fcompiler/compaq.py
--- numpy-1.10.2.orig/numpy/distutils/fcompiler/compaq.py	2016-01-04 00:44:11.333672900 +0000
+++ numpy-1.10.2/numpy/distutils/fcompiler/compaq.py	2016-01-04 00:46:47.573905500 +0000
@@ -10,7 +10,7 @@
 from distutils.errors import DistutilsPlatformError
 
 compilers = ['CompaqFCompiler']
-if os.name != 'posix' or sys.platform[:6] == 'cygwin' :
+if (os.name != 'posix' or sys.platform[:6] == 'cygwin') and not 'GCC' in sys.version:
     # Otherwise we'd get a false positive on posix systems with
     # case-insensitive filesystems (like darwin), because we'll pick
     # up /bin/df
diff -urN numpy-1.10.2.orig/numpy/distutils/msvccompiler.py numpy-1.10.2/numpy/distutils/msvccompiler.py
--- numpy-1.10.2.orig/numpy/distutils/msvccompiler.py	2016-01-04 00:44:11.283672800 +0000
+++ numpy-1.10.2/numpy/distutils/msvccompiler.py	2016-01-04 00:44:31.151202200 +0000
@@ -14,7 +14,7 @@
     def initialize(self, plat_name=None):
         environ_lib = os.getenv('lib')
         environ_include = os.getenv('include')
-        distutils.msvccompiler.MSVCCompiler.initialize(self, plat_name)
+        distutils.msvccompiler.MSVCCompiler.initialize(self)
         if environ_lib is not None:
             os.environ['lib'] = environ_lib + os.environ['lib']
         if environ_include is not None:
