diff -urN numpy-1.10.2.orig/numpy/distutils/fcompiler/compaq.py numpy-1.10.2/numpy/distutils/fcompiler/compaq.py
--- numpy-1.10.2.orig/numpy/distutils/fcompiler/compaq.py	2016-01-04 00:44:11.333672900 +0000
+++ numpy-1.10.2/numpy/distutils/fcompiler/compaq.py	2016-01-04 00:46:47.573905500 +0000
@@ -10,7 +10,7 @@
 from distutils.errors import DistutilsPlatformError
 
 compilers = ['CompaqFCompiler']
-if os.name != 'posix' or sys.platform[:6] == 'cygwin' :
+if (os.name != 'posix' or sys.platform[:6] == 'cygwin') and not 'GCC' in sys.version:
     # Otherwise we'd get a false positive on posix systems with
     # case-insensitive filesystems (like darwin), because we'll pick
     # up /bin/df
diff -aur numpy-1.11.2/numpy/distutils/msvccompiler.py.orig numpy-1.11.2/numpy/distutils/msvccompiler.py
--- numpy-1.11.2/numpy/distutils/msvccompiler.py.orig	2016-12-12 14:41:40.479160800 -0500
+++ numpy-1.11.2/numpy/distutils/msvccompiler.py	2016-12-12 14:45:23.036996800 -0500
@@ -47,7 +47,7 @@
         # by MSVCCompiler.initialize, so save them for later merge.
         environ_lib = os.getenv('lib', '')
         environ_include = os.getenv('include', '')
-        _MSVCCompiler.initialize(self, plat_name)
+        _MSVCCompiler.initialize(self)
 
         # Merge current and previous values of 'lib' and 'include'
         os.environ['lib'] = _merge(environ_lib, os.environ['lib'])
