--- numpy-orig/numpy/core/setup.py	2018-04-25 23:12:01.000000000 +0200
+++ numpy/numpy/core/setup.py	2018-05-25 21:58:05.675465800 +0200
@@ -688,7 +688,7 @@
                        ]
     
     # Must be true for CRT compilers but not MinGW/cygwin. See gh-9977.
-    is_msvc = platform.system() == 'Windows'
+    is_msvc = platform.system() == 'Windows' and not 'GCC' in sys.version
     config.add_installed_library('npymath',
             sources=npymath_sources + [get_mathlib_info],
             install_dir='lib',
