diff --git a/numpy/core/include/numpy/npy_common.h b/numpy/core/include/numpy/npy_common.h
index 78d6d31..cc7d060 100644
--- a/numpy/core/include/numpy/npy_common.h
+++ b/numpy/core/include/numpy/npy_common.h
@@ -179,7 +179,7 @@
     #include <io.h>
 
 /* mingw based on 3.4.5 has lseek but not ftell/fseek */
-#if defined(__MINGW32__) || defined(__MINGW64__)
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR)
 extern int __cdecl _fseeki64(FILE *, long long, int);
 extern long long __cdecl _ftelli64(FILE *);
 #endif
