--- numpy-1.21.0/numpy/distutils/fcompiler/__init__.py.orig	2021-07-22 19:42:37.638429800 -0700
+++ numpy-1.21.0/numpy/distutils/fcompiler/__init__.py	2021-07-22 19:42:43.012362300 -0700
@@ -797,21 +797,23 @@
         try:
             c = new_fcompiler(plat=platform, compiler=compiler_type,
                               c_compiler=c_compiler)
-            c.customize(dist)
-            v = c.get_version()
-            if requiref90 and c.compiler_f90 is None:
-                v = None
-                new_compiler = c.suggested_f90_compiler
-                if new_compiler:
-                    log.warn('Trying %r compiler as suggested by %r '
-                             'compiler for f90 support.' % (compiler_type,
-                                                            new_compiler))
-                    c = new_fcompiler(plat=platform, compiler=new_compiler,
-                                      c_compiler=c_compiler)
-                    c.customize(dist)
-                    v = c.get_version()
-                    if v is not None:
-                        compiler_type = new_compiler
+            if c is not None:
+                c.customize(dist)
+                v = c.get_version()
+                if requiref90 and c.compiler_f90 is None:
+                    v = None
+                    new_compiler = c.suggested_f90_compiler
+                    if new_compiler:
+                        log.warn('Trying %r compiler as suggested by %r '
+                                 'compiler for f90 support.' % (compiler_type,
+                                                                new_compiler))
+                        c = new_fcompiler(plat=platform, compiler=new_compiler,
+                                          c_compiler=c_compiler)
+                        if c is not None:
+                            c.customize(dist)
+                            v = c.get_version()
+                            if v is not None:
+                                compiler_type = new_compiler
             if requiref90 and c.compiler_f90 is None:
                 raise ValueError('%s does not support compiling f90 codes, '
                                  'skipping.' % (c.__class__.__name__))
