--- numpy-1.0.4/numpy/core/include/numpy/fenv/fenv.c.orig	2007-11-07 16:05:15.000000000 -0600
+++ numpy-1.0.4/numpy/core/include/numpy/fenv/fenv.c	2007-11-29 22:29:31.513750000 -0600
@@ -26,6 +26,9 @@
  * $FreeBSD$
  */
 
+#ifndef NUMPY_FENV_C
+#define NUMPY_FENV_C
+
 #include <sys/types.h>
 #include "fenv.h"
 
@@ -36,3 +39,5 @@
 	{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
 	  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff }
 };
+
+#endif /* NUMPY_FENV_C */
