--- numpy-1.10.0/numpy/distutils/fcompiler/compaq.py.orig	2015-10-08 14:58:05.495698100 +0300
+++ numpy-1.10.0/numpy/distutils/fcompiler/compaq.py	2015-10-08 14:59:02.712975800 +0300
@@ -4,13 +4,14 @@
 
 import os
 import sys
+from sysconfig import _POSIX_BUILD
 
 from numpy.distutils.fcompiler import FCompiler
 from numpy.distutils.compat import get_exception
 from distutils.errors import DistutilsPlatformError
 
 compilers = ['CompaqFCompiler']
-if os.name != 'posix' or sys.platform[:6] == 'cygwin' :
+if (os.name != 'posix' or sys.platform[:6] == 'cygwin') and not _POSIX_BUILD :
     # Otherwise we'd get a false positive on posix systems with
     # case-insensitive filesystems (like darwin), because we'll pick
     # up /bin/df
--- numpy-1.10.0/numpy/distutils/msvccompiler.py.orig	2015-10-08 15:14:33.374900300 +0300
+++ numpy-1.10.0/numpy/distutils/msvccompiler.py	2015-10-08 15:14:40.886149100 +0300
@@ -10,7 +10,7 @@
     def initialize(self, plat_name=None):
         environ_lib = os.getenv('lib')
         environ_include = os.getenv('include')
-        distutils.msvccompiler.MSVCCompiler.initialize(self, plat_name)
+        distutils.msvccompiler.MSVCCompiler.initialize(self)
         if environ_lib is not None:
             os.environ['lib'] = environ_lib + os.environ['lib']
         if environ_include is not None:
