--- a/configure.ac
+++ b/configure.ac
@@ -41,9 +41,9 @@
   dnl Fallback to the old libusb
   dnl libusb >= 0.1.8 is required, as we need usb_interrupt_read()
   AS_ECHO("using libusb");
-  AC_CHECK_HEADER(usb.h,
-                  AC_CHECK_LIB(usb, usb_interrupt_read,
-                               [LIBS="$LIBS -lusb"
+  AC_CHECK_HEADER(lusb0_usb.h,
+                  AC_CHECK_LIB(usb0, usb_interrupt_read,
+                               [LIBS="$LIBS -lusb0"
                                 HAVE_USB=yes]))
 fi
 

--- a/src/dfu-device.h
+++ b/src/dfu-device.h
@@ -8,7 +8,7 @@
 #ifdef HAVE_LIBUSB_1_0
 #include <libusb.h>
 #else
-#include <usb.h>
+#include <lusb0_usb.h>
 #endif
 
 // Atmel device classes are now defined with one bit per class.
--- a/src/dfu.c
+++ b/src/dfu.c
@@ -29,7 +29,7 @@
 #ifdef HAVE_LIBUSB_1_0
 #include <libusb.h>
 #else
-#include <usb.h>
+#include <lusb0_usb.h>
 #endif
 #include <errno.h>
 #include "dfu.h"
--- a/src/dfu.h
+++ b/src/dfu.h
@@ -27,7 +27,7 @@
 #ifdef HAVE_LIBUSB_1_0
 #include <libusb.h>
 #else
-#include <usb.h>
+#include <lusb0_usb.h>
 #endif
 #include <stdint.h>
 #include <stddef.h>
--- a/src/main.c
+++ b/src/main.c
@@ -26,7 +26,7 @@
 #ifdef HAVE_LIBUSB_1_0
 #include <libusb.h>
 #else
-#include <usb.h>
+#include <lusb0_usb.h>
 #endif
 
 #include "config.h"
