--- a/CMake/AbseilDll.cmake
+++ b/CMake/AbseilDll.cmake
@@ -747,7 +747,11 @@
   else()
     set(_dll "abseil_dll")
     set(_dll_files ${ABSL_INTERNAL_DLL_FILES})
-    set(_dll_libs "")
+    if(MINGW)
+      set(_dll_libs Threads::Threads bcrypt dbghelp)
+    else()
+      set(_dll_libs "")
+    endif()
     set(_dll_compile_definitions "")
     set(_dll_includes "")
     set(_dll_consume "ABSL_CONSUME_DLL")
--- a/absl/synchronization/internal/pthread_waiter.h
+++ b/absl/synchronization/internal/pthread_waiter.h
@@ -16,7 +16,7 @@
 #ifndef ABSL_SYNCHRONIZATION_INTERNAL_PTHREAD_WAITER_H_
 #define ABSL_SYNCHRONIZATION_INTERNAL_PTHREAD_WAITER_H_
 
-#if !defined(_WIN32) && !defined(__MINGW32__)
+#if !defined(_MSC_VER)
 #include <pthread.h>
 
 #include "absl/base/config.h"
