From 04305aecf17a16b1253c7bf95751a10b80cc1c53 Mon Sep 17 00:00:00 2001
From: udayrajsingh288 <udayrajsingh288@gmail.com>
Date: Fri, 10 Jul 2026 09:42:08 +0530
Subject: [PATCH 1/2] Bug Fix: changed macro to allow all windows compilers

---
 src/clinfo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/clinfo.c b/src/clinfo.c
index 24c12f2..2c12dd7 100644
--- a/src/clinfo.c
+++ b/src/clinfo.c
@@ -9,7 +9,7 @@
  * On Windows, we must get the module handle for it, on Unix-like
  * systems we can just use RTLD_DEFAULT
  */
-#ifdef _MSC_VER
+#if defined(_WIN32) || defined(_WIN64)
 # include <windows.h>
 # define dlsym GetProcAddress
 # define DL_MODULE GetModuleHandle("OpenCL")
