From 165a40221da841736fbfc956ef8510205d778e0b Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Mon, 10 Sep 2018 07:42:25 +0300
Subject: [PATCH] Fix compat header

---
 mz_compat.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mz_compat.h b/mz_compat.h
index b2bfa9d..d021acc 100644
--- a/mz_compat.h
+++ b/mz_compat.h
@@ -15,9 +15,10 @@
 #define MZ_COMPAT_H
 
 #include <stdint.h>
+#include <time.h>
 
 #ifdef __cplusplus
-"C" {
+extern "C" {
 #endif
 
 #ifdef HAVE_ZLIB
-- 
2.18.0

