--- wgpu-0.32.0.orig/wgpu/_coreutils.py	2020-02-02 03:00:00.000000000 +0300
+++ wgpu-0.32.0/wgpu/_coreutils.py	2026-09-04 16:03:27.651589500 +0300
@@ -65,6 +65,12 @@
     if path.exists():
         return str(path)
 
+    # Windows MSYS2 file layout like linux but store libraries
+    # in bin directory
+    path = Path(sys.exec_prefix) / "bin" / name
+    if path.exists():
+        return str(path)
+
     # Windows conda file layout is slightly different and includes a
     # Library before the include directory
     path = Path(sys.exec_prefix) / "Library" / "bin" / name
