diff -bur thrift-0.13.0-c/lib/cpp/test/Benchmark.cpp thrift-0.13.0/lib/cpp/test/Benchmark.cpp
--- thrift-0.13.0-c/lib/cpp/test/Benchmark.cpp	2019-10-10 16:34:27.000000000 -0600
+++ thrift-0.13.0/lib/cpp/test/Benchmark.cpp	2022-08-08 19:32:33.426430300 -0600
@@ -20,8 +20,8 @@
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
-#include <iostream>
 #define _USE_MATH_DEFINES
+#include <iostream>
 #include <math.h>
 #include <memory>
 #include "thrift/protocol/TBinaryProtocol.h"
