--- a/build/cmake/DefineInstallationPaths.cmake
+++ b/build/cmake/DefineInstallationPaths.cmake
@@ -30,7 +30,7 @@
 set(PKGCONFIG_INSTALL_DIR "lib/pkgconfig" CACHE PATH "The subdirectory to install pkgconfig config files (default: lib/pkgconfig)")
 set(DOC_INSTALL_DIR "share/doc" CACHE PATH "The subdirectory to install documentation files (default: share/doc)")
 set(prefix "${CMAKE_INSTALL_PREFIX}")
-set(exec_prefix "${CMAKE_INSTALL_PREFIX}/bin")
-set(libdir "${CMAKE_INSTALL_PREFIX}/lib")
-set(includedir "${CMAKE_INSTALL_PREFIX}/include")
+set(exec_prefix "\${prefix}/bin")
+set(libdir "\${prefix}/lib")
+set(includedir "\${prefix}/include")
 set(cmakedir "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DIR}")
--- a/build/cmake/GenerateConfigModule.cmake
+++ b/build/cmake/GenerateConfigModule.cmake
@@ -18,9 +18,9 @@
 #
 
 include(CMakePackageConfigHelpers)
-set(PACKAGE_INCLUDE_INSTALL_DIR "${includedir}/thrift")
-set(PACKAGE_CMAKE_INSTALL_DIR "${cmakedir}/thrift")
-set(PACKAGE_BIN_INSTALL_DIR "${exec_prefix}")
+set(PACKAGE_INCLUDE_INSTALL_DIR "\${PACKAGE_PREFIX_DIR}/${INCLUDE_INSTALL_DIR}/thrift")
+set(PACKAGE_CMAKE_INSTALL_DIR "\${PACKAGE_PREFIX_DIR}/${CMAKE_INSTALL_DIR}/thrift")
+set(PACKAGE_BIN_INSTALL_DIR "\${PACKAGE_PREFIX_DIR}/bin")
 
 # In CYGWIN enviroment below commands does not work properly
 if (NOT CYGWIN)
