diff --git a/man/CMakeLists.txt b/man/CMakeLists.txt
index 5ce2fc7..f87393a 100644
--- a/man/CMakeLists.txt
+++ b/man/CMakeLists.txt
@@ -309,7 +309,7 @@ macro(define_symlink_target NAME EXT)
 		list(GET MAN_ALIAS ${i} SRC)
 		list(GET MAN_ALIAS ${j} DST)
 		add_custom_command(OUTPUT ${DST}.${EXT}
-			COMMAND ln -sf ${SRC}.${EXT} ${DST}.${EXT})
+			COMMAND cp -f ${SRC}.${EXT} ${DST}.${EXT})
 		list(APPEND ${NAME}_LINK_FILES ${DST}.${EXT})
 	endforeach()
 	add_custom_target(${NAME} DEPENDS ${${NAME}_LINK_FILES})
