--- a/configure
+++ b/configure
@@ -325,12 +325,15 @@
 #localedir='${datarootdir}/locale'
 
 # For cross compiling
-_host=""
-_host_cpu=""
+_host="$host"
+_host_cpu="$host_cpu"
 _host_vendor=""
 _host_os=""
 _host_alias=""
 
+_build="$build"
+_build_cpu="$build_cpu"
+
 _port_mk="ports.mk"
 
 # Use temp files in the build directory
@@ -3480,7 +3483,7 @@
 		;;
 esac
 
-if test -n "$_host"; then
+if test "$_host" != "$_build" ; then
 	# Cross-compiling mode - add your target here if needed
 	echo "Cross-compiling to $_host"
 	case "$_host" in
