--- origsrc/usrsctp/programs/ekr_peer.c	2015-12-30 21:43:16.148770100 +0100
+++ src/usrsctp/programs/ekr_peer.c	2016-01-03 00:22:14.680242600 +0100
@@ -483,7 +483,7 @@
 		perror("usrsctp_connect");
 	}
 	for (;;) {
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
 		if (gets_s(line, LINE_LENGTH) == NULL) {
 #else
 		if (fgets(line, LINE_LENGTH, stdin) == NULL) {
--- origsrc/usrsctp/programs/rtcweb.c	2015-12-30 21:43:16.160092000 +0100
+++ src/usrsctp/programs/rtcweb.c	2016-01-03 00:21:32.151324200 +0100
@@ -1426,7 +1426,7 @@
 	unlock_peer_connection(&peer_connection);
 
 	for (;;) {
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
 		if (gets_s(line, LINE_LENGTH) == NULL) {
 #else
 		if (fgets(line, LINE_LENGTH, stdin) == NULL) {
--- origsrc/usrsctp-0.9.3.0/usrsctplib/Makefile.am	2016-04-08 16:14:31.000000000 +0200
+++ src/usrsctp-0.9.3.0/usrsctplib/Makefile.am	2016-08-05 22:38:12.280677700 +0200
@@ -76,6 +76,6 @@
                          netinet6/sctp6_var.h \
                          netinet6/sctp6_usrreq.c
 libusrsctp_la_CFLAGS = $(LIBCFLAGS)
-libusrsctp_la_LDFLAGS = -version-info 1:0:0
+libusrsctp_la_LDFLAGS = -version-info 1:0:0 $(EXTRA_LDFLAGS)
 include_HEADERS = usrsctp.h
 
--- origsrc/usrsctp/usrsctplib/user_malloc.h	2015-12-30 21:43:16.667559000 +0100
+++ src/usrsctp/usrsctplib/user_malloc.h	2016-01-01 15:18:22.959992600 +0100
@@ -41,7 +41,7 @@
 #include <strings.h>
 #include <stdint.h>
 #else
-#if defined(_MSC_VER) && _MSC_VER >= 1600
+#if (defined(_MSC_VER) && _MSC_VER >= 1600) || defined(__MINGW32__)
 #include <stdint.h>
 #elif defined(SCTP_STDINT_INCLUDE)
 #include SCTP_STDINT_INCLUDE
--- origsrc/usrsctp/usrsctplib/user_socketvar.h	2015-12-30 21:43:16.818640800 +0100
+++ src/usrsctp/usrsctplib/user_socketvar.h	2016-01-01 14:09:15.820425100 +0100
@@ -102,7 +102,9 @@
  */
 #if defined (__Userspace_os_Windows)
 #define AF_ROUTE  17
+#ifndef __MINGW32__
 typedef __int32 pid_t;
+#endif
 typedef unsigned __int32 uid_t;
 enum sigType {
 	SIGNAL = 0,
--- origsrc/usrsctp/usrsctplib/usrsctp.h	2015-12-30 21:43:16.830029300 +0100
+++ src/usrsctp/usrsctplib/usrsctp.h	2016-01-01 15:18:51.143073300 +0100
@@ -59,7 +59,7 @@
 #endif
 
 #ifdef _WIN32
-#if defined(_MSC_VER) && _MSC_VER >= 1600
+#if (defined(_MSC_VER) && _MSC_VER >= 1600) || defined(__MINGW32__)
 #include <stdint.h>
 #elif defined(SCTP_STDINT_INCLUDE)
 #include SCTP_STDINT_INCLUDE
--- origsrc/usrsctp-0.9.3.0/usrsctplib/netinet/sctp_os_userspace.h	2016-04-08 16:14:31.000000000 +0200
+++ src/usrsctp-0.9.3.0/usrsctplib/netinet/sctp_os_userspace.h	2016-08-06 00:04:52.853091000 +0200
@@ -78,7 +78,7 @@
 #define IPVERSION  4
 #define MAXTTL     255
 /* VS2010 comes with stdint.h */
-#if _MSC_VER >= 1600
+#if (_MSC_VER >= 1600) || defined(__MINGW32__)
 #include <stdint.h>
 #else
 #define uint64_t   unsigned __int64
@@ -218,7 +218,7 @@
 
 #define bzero(buf, len) memset(buf, 0, len)
 #define bcopy(srcKey, dstKey, len) memcpy(dstKey, srcKey, len)
-#if _MSC_VER < 1900
+#if _MSC_VER < 1900  && !defined(__MINGW32__)
 #define snprintf(data, size, format, ...) _snprintf_s(data, size, _TRUNCATE, format, __VA_ARGS__)
 #endif
 #define inline __inline
@@ -1159,5 +1159,4 @@
 	    ((tvp)->tv_usec cmp (uvp)->tv_usec) :			\
 	    ((tvp)->tv_sec cmp (uvp)->tv_sec))
 #endif
-
 #endif
--- origsrc/usrsctp-0.9.3.0/configure.ac	2016-04-08 16:14:31.000000000 +0200
+++ src/usrsctp-0.9.3.0/configure.ac	2016-08-05 22:38:41.057701000 +0200
@@ -34,6 +34,11 @@
 AC_PROG_CC
 AC_PROG_LIBTOOL
 AC_CANONICAL_HOST
+case "$host_os" in
+mingw*)
+    EXTRA_LDFLAGS="-no-undefined" ;;
+esac
+AC_SUBST([EXTRA_LDFLAGS])
 AC_CONFIG_MACRO_DIR([m4])
 
 LIBCFLAGS="-DSCTP_PROCESS_LEVEL_LOCKS -DSCTP_SIMPLE_ALLOCATOR -D__Userspace__"
