--- a/src/pipcl.py
+++ b/src/pipcl.py
@@ -2055,1 +2055,1 @@
-def base_compiler(vs=None, pythonflags=None, cpp=False, use_env=True):
+def base_compiler(vs=None, pythonflags=None, cpp=False, use_env=False):
@@ -2367,2 +2367,2 @@
          else:
-            rpath_flag = "-Wl,-rpath,'$ORIGIN',-z,origin"
+            rpath_flag = ""
@@ -3094,1 +3094,1 @@
-        sep = ' ' if windows() else ' \\\n'
+        sep = ' '
@@ -3253,1 +3253,1 @@
-    return platform.system() == 'Windows'
+    return False
@@ -3271,1 +3271,1 @@
-    return platform.system() == 'Linux'
+    return True
@@ -3414,6 +3414,6 @@
             log2(f'Using {python_config=}.')
             try:
-                self.includes = run( f'{python_config} --includes', capture=1, verbose=0).strip()
+                self.includes = run( f'sh -c "{python_config} --includes"', capture=1, verbose=0).strip()
             except Exception as e:
                 raise Exception('We require python development tools to be installed.') from e
-            self.ldflags = run( f'{python_config} --ldflags', capture=1, verbose=0).strip()
+            self.ldflags = run( f'sh -c "{python_config} --ldflags"', capture=1, verbose=0).strip()
