--- libchamplain-0.12.9/champlain/Makefile.am.orig	2014-10-19 14:00:19.413400000 +0400
+++ libchamplain-0.12.9/champlain/Makefile.am	2014-10-19 14:00:28.430200000 +0400
@@ -159,6 +159,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = --warn-all
 INTROSPECTION_COMPILER_ARGS =
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 introspection_sources = \
 	$(filter-out $(srcdir)/champlain-version.h,$(libchamplain_headers_public))	\
 	$(libchamplain_sources)
--- libchamplain-0.12.9/champlain-gtk/Makefile.am.orig	2014-10-19 14:00:59.708000000 +0400
+++ libchamplain-0.12.9/champlain-gtk/Makefile.am	2014-10-19 14:01:02.547200000 +0400
@@ -70,6 +70,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = --warn-all --add-include-path=$(top_builddir)/champlain
 INTROSPECTION_COMPILER_ARGS = --includedir=$(top_builddir)/champlain
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 introspection_sources = $(libchamplain_gtk_@CHAMPLAIN_API_VERSION@_la_SOURCES)
 
