From a0439b30dd176be9a588d88204665e77b58d415e Mon Sep 17 00:00:00 2001
From: Abderrahim Kitouni <akitouni@gnome.org>
Date: Sun, 7 Mar 2021 17:52:35 +0100
Subject: [PATCH] main: add the entrypoint that setup.py expects

---
 gidocgen/gidocmain.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gidocgen/gidocmain.py b/gidocgen/gidocmain.py
index fd41be1..3a8c132 100644
--- a/gidocgen/gidocmain.py
+++ b/gidocgen/gidocmain.py
@@ -125,5 +125,9 @@ def run_main(args):
     return GIDocGenApp().run(args)
 
 
+def main():
+    return run_main(sys.argv[1:])
+
+
 if __name__ == '__main__':
-    sys.exit(run_main(sys.argv[1:]))
+    sys.exit(main())
-- 
GitLab

