--- Botan-2.4.0/src/scripts/build_docs.py.orig	2018-02-07 09:49:41.644022500 +0300
+++ Botan-2.4.0/src/scripts/build_docs.py	2018-02-07 09:55:23.511136300 +0300
@@ -55,7 +55,7 @@
     logging.debug("Executing %s", ' '.join(cmd_line))
 
     proc = subprocess.Popen(cmd_line,
-                            close_fds=True,
+                            close_fds=not sys.platform == 'win32',
                             stdout=subprocess.PIPE,
                             stderr=subprocess.PIPE,
                             cwd=cwd)
