--- Botan-3.10.0/src/scripts/install.py.orig	2025-11-06 17:48:58.000000000 +0100
+++ Botan-3.10.0/src/scripts/install.py	2026-01-08 09:35:02.576289200 +0100
@@ -72,7 +72,7 @@
         if os.path.sep != "/":
             raise PrependDestdirError("Only UNIX-like file systems using forward slash " \
                                       "separator supported when DESTDIR is set.")
-        if not os.path.isabs(path):
+        if not os.path.isabs(path) and not path.startswith(os.path.sep):
             raise PrependDestdirError("--prefix must be an absolute path when DESTDIR is set.")
 
         path = os.path.normpath(path)
