--- Botan-1.11.21/src/lib/utils/calendar.cpp.orig	2015-10-11 14:31:25.000000000 +0300
+++ Botan-1.11.21/src/lib/utils/calendar.cpp	2015-10-12 19:59:06.522921000 +0300
@@ -39,8 +39,6 @@
    return tm;
    }
 
-#if !defined(BOTAN_TARGET_OS_HAS_TIMEGM) && !defined(BOTAN_TARGET_OS_HAS_MKGMTIME)
-
 #if defined(BOTAN_HAS_BOOST_DATETIME)
 
 std::time_t boost_timegm(std::tm *tm)
@@ -109,8 +107,6 @@
 }
 #endif // BOTAN_HAS_BOOST_DATETIME
 
-#endif
-
 }
 
 std::chrono::system_clock::time_point calendar_point::to_std_timepoint()
@@ -136,12 +132,7 @@
    tm.tm_year  = year - 1900;
 
    // Define a function alias `botan_timegm`
-   #if defined(BOTAN_TARGET_OS_HAS_TIMEGM)
-   std::time_t (&botan_timegm)(std::tm *tm) = timegm;
-   #elif defined(BOTAN_TARGET_OS_HAS_MKGMTIME)
-   // http://stackoverflow.com/questions/16647819/timegm-cross-platform
-   std::time_t (&botan_timegm)(std::tm *tm) = _mkgmtime;
-   #elif defined(BOTAN_HAS_BOOST_DATETIME)
+   #if defined(BOTAN_HAS_BOOST_DATETIME)
    std::time_t (&botan_timegm)(std::tm *tm) = boost_timegm;
    #else
    std::time_t (&botan_timegm)(std::tm *tm) = fallback_timegm;
