--- Botan-1.11.28/src/lib/entropy/win32_stats/es_win32.cpp.orig	2016-02-06 17:05:24.361838900 +0200
+++ Botan-1.11.28/src/lib/entropy/win32_stats/es_win32.cpp	2016-02-06 17:05:27.910173600 +0200
@@ -20,7 +20,6 @@
    First query a bunch of basic statistical stuff, though
    don't count it for much in terms of contributed entropy.
    */
-   accum.add(GetTickCount64(), BOTAN_ENTROPY_ESTIMATE_SYSTEM_DATA);
    accum.add(GetMessagePos(), BOTAN_ENTROPY_ESTIMATE_SYSTEM_DATA);
    accum.add(GetMessageTime(), BOTAN_ENTROPY_ESTIMATE_SYSTEM_DATA);
    accum.add(GetInputState(), BOTAN_ENTROPY_ESTIMATE_SYSTEM_DATA);
