diff -Naur luanti-5.11.0.orig/CMakeLists.txt luanti-5.11.0/CMakeLists.txt
--- luanti-5.11.0.orig/CMakeLists.txt	2025-03-10 01:29:22.793169700 +0300
+++ luanti-5.11.0/CMakeLists.txt	2025-03-10 01:29:40.614933400 +0300
@@ -139,7 +139,7 @@
 
 # Installation
 
-if(WIN32)
+if(WIN32 AND FALSE)
 	set(SHAREDIR ".")
 	set(BINDIR "bin")
 	set(DOCDIR "doc")
@@ -153,7 +153,7 @@
 	set(DOCDIR "${SHAREDIR}/${PROJECT_NAME}")
 	set(EXAMPLE_CONF_DIR ${DOCDIR})
 	set(LOCALEDIR "${SHAREDIR}/locale")
-elseif(UNIX) # Linux, BSD etc
+elseif(UNIX OR TRUE) # Linux, BSD etc
 	if(RUN_IN_PLACE)
 		set(SHAREDIR ".")
 		set(BINDIR "bin")
@@ -260,7 +260,7 @@
 install(FILES "doc/world_format.md" DESTINATION "${DOCDIR}" COMPONENT "Docs")
 install(FILES "minetest.conf.example" DESTINATION "${EXAMPLE_CONF_DIR}")
 
-if(UNIX AND NOT APPLE)
+if(UNIX OR TRUE)
 	install(FILES "doc/luanti.6" "doc/luantiserver.6" DESTINATION "${MANDIR}/man6")
 	install(FILES "misc/net.minetest.minetest.desktop" DESTINATION "${XDG_APPS_DIR}")
 	install(FILES "misc/net.minetest.minetest.metainfo.xml" DESTINATION "${METAINFODIR}")
@@ -314,7 +314,7 @@
 	DESCRIPTION "Documentation about ${PROJECT_NAME_CAPITALIZED} and ${PROJECT_NAME_CAPITALIZED} modding"
 )
 
-if(WIN32)
+if(WIN32 AND FALSE)
 	# Include all dynamically linked runtime libraries such as MSVCRxxx.dll
 	include(InstallRequiredSystemLibraries)
 
diff -Naur luanti-5.11.0.orig/src/porting.cpp luanti-5.11.0/src/porting.cpp
--- luanti-5.11.0.orig/src/porting.cpp	2025-03-10 01:29:20.011816900 +0300
+++ luanti-5.11.0/src/porting.cpp	2025-03-10 01:34:56.160046100 +0300
@@ -435,7 +435,7 @@
 	std::string exepath(buf);
 
 	// Use ".\bin\.."
-	path_share = exepath + "\\..";
+	path_share = exepath + DIR_DELIM ".." DIR_DELIM "share" DIR_DELIM + PROJECT_NAME;
 	if (detectMSVCBuildDir(exepath)) {
 		// The msvc build dir schould normaly not be present if properly installed,
 		// but its useful for debugging.
