diff --git a/Makefile b/Makefile
index 764f411a..2b74913d 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@ VERSION := 0.6.3
 # install directory layout
 PREFIX ?= /usr/local
 INCLUDEDIR ?= $(PREFIX)/include
+BINDIR ?= $(PREFIX)/bin
 LIBDIR ?= $(PREFIX)/lib
 PCLIBDIR ?= $(LIBDIR)/pkgconfig
 
@@ -32,31 +33,29 @@ ifeq ($(shell uname),Darwin)
 	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).dylib
 	LINKSHARED += -dynamiclib -Wl,-install_name,$(LIBDIR)/libtree-sitter.$(SONAME_MAJOR).dylib
 else
-	SOEXT = so
-	SOEXTVER_MAJOR = so.$(SONAME_MAJOR)
-	SOEXTVER = so.$(SONAME_MAJOR).$(SONAME_MINOR)
-	LINKSHARED += -shared -Wl,-soname,libtree-sitter.so.$(SONAME_MAJOR)
+	SOEXT = dll
+	LINKSHARED += -shared
 endif
 ifneq (,$(filter $(shell uname),FreeBSD NetBSD DragonFly))
 	PCLIBDIR := $(PREFIX)/libdata/pkgconfig
 endif
 
-all: libtree-sitter.a libtree-sitter.$(SOEXTVER)
+all: libtree-sitter.a libtree-sitter.$(SOEXT) libtree-sitter.dll.a
 
 libtree-sitter.a: $(OBJ)
 	$(AR) rcs $@ $^
 
-libtree-sitter.$(SOEXTVER): $(OBJ)
-	$(CC) $(LDFLAGS) $(LINKSHARED) $^ $(LDLIBS) -o $@
-	ln -sf $@ libtree-sitter.$(SOEXT)
-	ln -sf $@ libtree-sitter.$(SOEXTVER_MAJOR)
+libtree-sitter.$(SOEXT): $(OBJ)
+	$(CC) $(LDFLAGS) $(LINKSHARED) $^ $(LDLIBS) -o $@ -Wl,--out-implib,$@.a
+
+libtree-sitter.dll.a: libtree-sitter.$(SOEXT)
 
 install: all
 	install -d '$(DESTDIR)$(LIBDIR)'
 	install -m755 libtree-sitter.a '$(DESTDIR)$(LIBDIR)'/libtree-sitter.a
-	install -m755 libtree-sitter.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter.$(SOEXTVER)
-	ln -sf libtree-sitter.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter.$(SOEXTVER_MAJOR)
-	ln -sf libtree-sitter.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter.$(SOEXT)
+	install -m755 libtree-sitter.dll.a '$(DESTDIR)$(LIBDIR)'/libtree-sitter.dll.a
+	install -d '$(DESTDIR)$(BINDIR)'
+	install -m755 libtree-sitter.$(SOEXT) '$(DESTDIR)$(BINDIR)'/libtree-sitter.$(SOEXT)
 	install -d '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter
 	install -m644 lib/include/tree_sitter/*.h '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter/
 	install -d '$(DESTDIR)$(PCLIBDIR)'
@@ -66,6 +65,6 @@ install: all
 	    tree-sitter.pc.in > '$(DESTDIR)$(PCLIBDIR)'/tree-sitter.pc
 
 clean:
-	rm -f lib/src/*.o libtree-sitter.a libtree-sitter.$(SOEXT) libtree-sitter.$(SOEXTVER_MAJOR) libtree-sitter.$(SOEXTVER)
+	rm -f lib/src/*.o libtree-sitter.a libtree-sitter.dll.a libtree-sitter.$(SOEXT)
 
 .PHONY: all install clean
