--- a/src/loader/CMakeLists.txt
+++ a/src/loader/CMakeLists.txt
@@ -203,6 +203,12 @@
     target_sources(
         openxr_loader PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/loader.rc"
     )
+    if(DYNAMIC_LOADER AND NOT (CMAKE_SYSTEM_NAME STREQUAL "WindowsStore"))
+         target_sources(
+              openxr_loader
+              PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/openxr-loader.def"
+              )
+    endif()
 
     if(MSVC)
         if(DYNAMIC_LOADER AND NOT (CMAKE_SYSTEM_NAME STREQUAL "WindowsStore"))
@@ -212,10 +218,6 @@
                 PROPERTIES MSVC_RUNTIME_LIBRARY
                            "MultiThreaded$<$<CONFIG:Debug>:Debug>"
             )
-            target_sources(
-                openxr_loader
-                PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/openxr-loader.def"
-            )
         else()
             # WindowsStore (UWP) apps must be compiled with dynamic CRT linkage (default)
             # Otherwise for static libs, link the CRT dynamically
