--- a/src/api_layers/CMakeLists.txt
+++ b/src/api_layers/CMakeLists.txt
@@ -63,6 +63,7 @@
 
 add_library(XrApiLayer_api_dump SHARED
     api_dump.cpp
+    XrApiLayer_api_dump.def
     ${PROJECT_SOURCE_DIR}/src/common/hex_and_handles.h
     # target-specific generated files
     ${GENERATED_OUTPUT}
@@ -129,6 +130,7 @@
 
 add_library(XrApiLayer_core_validation SHARED
     core_validation.cpp
+    XrApiLayer_core_validation.def
     ${PROJECT_SOURCE_DIR}/src/common/hex_and_handles.h
     ${PROJECT_SOURCE_DIR}/src/common/object_info.cpp
     ${PROJECT_SOURCE_DIR}/src/common/object_info.h
--- a/src/loader/CMakeLists.txt
+++ a/src/loader/CMakeLists.txt
@@ -7,7 +7,7 @@
 if(DYNAMIC_LOADER)
     add_definitions(-DXRAPI_DLL_EXPORT)
     set(LIBRARY_TYPE SHARED)
-    if(MSVC)
+    if(WIN32)
         list(APPEND openxr_loader_RESOURCE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/openxr-loader.def)
     endif()
 else() # build static lib
