--- a/src/loader/CMakeLists.txt
+++ b/src/loader/CMakeLists.txt
@@ -398,7 +398,7 @@
 # Create alias so that it can be used the same whether vendored as source or found with CMake.
 add_library(OpenXR::openxr_loader ALIAS openxr_loader)
 
-if(WIN32 AND NOT INSTALL_TO_ARCHITECTURE_PREFIXES)
+if(MSVC AND NOT INSTALL_TO_ARCHITECTURE_PREFIXES)
     set(TARGET_DESTINATION cmake)
 else()
     set(TARGET_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/openxr/)
