--- webkitgtk-2.4.9/Source/WebCore/platform/graphics/GLContext.cpp.orig	2015-07-12 20:33:29.627600000 +0300
+++ webkitgtk-2.4.9/Source/WebCore/platform/graphics/GLContext.cpp	2015-07-12 20:33:36.054800000 +0300
@@ -159,7 +159,7 @@
         return glxContext.release();
 #endif
 #if USE(EGL)
-    if (OwnPtr<GLContext> eglContext = GLContextEGL::createContext(windowHandle, sharingContext))
+    if (OwnPtr<GLContext> eglContext = GLContextEGL::createContext((EGLNativeWindowType)windowHandle, sharingContext))
         return eglContext.release();
 #endif
     return nullptr;
