--- a/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp
+++ b/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp
@@ -487,7 +487,12 @@ JSValue CLoop::execute(CallFrame* callFrame, Opcode entryOpcode, bool isInitiali
         rBasePC.vp = codeBlock->instructions().begin();
 #endif // USE(JSVALUE64)
 
+#if ENABLE(COMPUTED_GOTO_OPCODES)
         goto llint_generic_return_point;
+#else
+        /* (most probably) causes null pointer dereference:
+         * NEXT_INSTRUCTION(); */
+#endif
 
     } // END doReturnHelper.
 
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -794,7 +794,7 @@
 #endif
 
 /* Configure the interpreter */
-#if COMPILER(GCC)
+#if COMPILER(GCC) && !OS(WINDOWS)
 #define HAVE_COMPUTED_GOTO 1
 #endif
 
