--- webkitgtk-2.4.9/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp.orig	2017-12-06 14:59:40.768262788 -0500
+++ webkitgtk-2.4.9/Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp	2017-12-06 15:03:10.000000000 -0500
@@ -128,7 +128,7 @@
         for (i = 0; i < numberOfSuggestions; i++)
             guesses.append(String::fromUTF8(suggestions[i]));
 
-        enchant_dict_free_suggestions(*iter, suggestions);
+        enchant_dict_free_string_list(*iter, suggestions);
     }
 
     return guesses;
--- webkitgtk-2.4.11/Source/autotools/FindDependencies.m4.orig	2018-07-04 12:52:47.582565800 +0200
+++ webkitgtk-2.4.11/Source/autotools/FindDependencies.m4	2018-07-04 12:53:18.535026100 +0200
@@ -154,7 +154,7 @@
 AC_SUBST(PANGO_LIBS)
 
 if test "$enable_spellcheck" = "yes"; then
-    PKG_CHECK_MODULES(ENCHANT, enchant >= enchant_required_version, [], [enable_spellcheck="no"])
+    PKG_CHECK_MODULES(ENCHANT, enchant-2 >= enchant_required_version, [], [enable_spellcheck="no"])
     AC_SUBST(ENCHANT_CFLAGS)
     AC_SUBST(ENCHANT_LIBS)
 fi
