diff --git a/src/config.h.W32 b/src/config.h.W32
index be2a33e..f43c2ad 100644
--- a/src/config.h.W32
+++ b/src/config.h.W32
@@ -294,7 +294,7 @@ this program.  If not, see <http://www.gnu.org/licenses/>.  */
 /* Define to 1 if `d_type' is a member of `struct dirent'. */
 /* SV 57152: MinGW64 version of dirent doesn't support d_type. */
 #ifndef __MINGW64__
-# define HAVE_STRUCT_DIRENT_D_TYPE 1
+/* # define HAVE_STRUCT_DIRENT_D_TYPE 1 */
 #endif
 
 /* Define to 1 if you have the `isatty' function. */
