--- a/lib/CL/pocl_llvm_build.cc
+++ b/lib/CL/pocl_llvm_build.cc
@@ -72,9 +72,7 @@ POP_COMPILER_DIAGS
 #include "pocl_runtime_config.h"
 #include "pocl_file_util.h"
 #include "pocl_cache.h"
-#ifdef HAVE_DLFCN_H
 #include "pocl_dynlib.h"
-#endif
 #include "LLVMUtils.h"
 #include "pocl_util.h"
 
@@ -1135,12 +1133,22 @@ int pocl_invoke_clang(const char* TTriple, const char** Args) {
 #else
   DiagnosticsEngine Diags(DiagID, DiagOpts, DiagClient);
 #endif
-  clang::driver::Driver TheDriver(pocl_get_path("CLANG", CLANGCC),
-                                  TTriple, Diags);
+  std::string ClangPath = CLANGCC;
+  if (!pocl_get_bool_option("POCL_BUILDING", 0)) {
+    const char *LibraryPath = pocl_dynlib_pathname((void *)pocl_invoke_clang);
+    if (LibraryPath) {
+      ClangPath = LibraryPath;
+      ClangPath.resize(ClangPath.find_last_of("/\\") + 1);
+      ClangPath += "clang.exe";
+    }
+  }
+  ClangPath = pocl_get_path("CLANG", ClangPath.c_str());
+  clang::driver::Driver TheDriver(ClangPath, TTriple, Diags);
 
   const char **ArgsEnd = Args;
   while (*ArgsEnd != nullptr) { ArgsEnd++; }
   llvm::SmallVector<const char*, 0> ArgsArray(Args, ArgsEnd);
+  ArgsArray[0] = ClangPath.c_str();
 
   int NumExtraArgs;
   const char *ExtraArgs = pocl_get_args("CLANG", &NumExtraArgs);
