diff -Naur QtBinPatcher-2.2.0-orig/QtBinPatcher.cpp QtBinPatcher-2.2.0/QtBinPatcher.cpp
--- QtBinPatcher-2.2.0-orig/QtBinPatcher.cpp	2015-07-05 19:17:34.000000000 +0300
+++ QtBinPatcher-2.2.0/QtBinPatcher.cpp	2021-10-13 09:26:08.664452400 +0300
@@ -254,11 +254,11 @@
     // Files for patching in Qt4.
     static const TElement Elements4[] = {
         { "/lib/",             "*.prl",              false },
-        { "/demos/shared/",    "libdemo_shared.prl", false },
+        { "/share/qt4/demos/shared/",    "libdemo_shared.prl", false },
         { "/lib/pkgconfig/",   "Qt*.pc",             false },
         { "/lib/pkgconfig/",   "phonon*.pc",         false },
 #if defined(OS_WINDOWS)
-        { "/mkspecs/default/", "qmake.conf",         false },
+        { "/share/qt4/mkspecs/default/", "qmake.conf",         false },
         { "/",                 ".qmake.cache",       false }
 #elif defined(OS_LINUX)
         { "/lib/pkgconfig/",   "qt*.pc",             false },
@@ -269,20 +269,33 @@
 
     // Files for patching in Qt5.
     static const TElement Elements5[] = {
-        { "/",                            "*.la",                         true  },
-        { "/",                            "*.prl",                        true  },
-        { "/lib/pkgconfig/",              "Qt5*.pc",                      true  },
-        { "/lib/pkgconfig/",              "Enginio*.pc",                  true  },
-        { "/",                            "*.pri",                        true  },
-        { "/lib/cmake/Qt5LinguistTools/", "Qt5LinguistToolsConfig.cmake", false },
-        { "/mkspecs/default-host/",       "qmake.conf",                   false },
+#if !defined(OS_WINDOWS)
+        { "/",                                    "*.la",                         true  },
+#endif
+        { "/lib/",                                "*.prl",                        true  },
+        { "/lib/pkgconfig/",                      "Qt5*.pc",                      true  },
+        { "/lib/pkgconfig/",                      "Enginio*.pc",                  true  },
+        { "/share/qt5/",                          "*.pri",                        true  },
+        { "/share/qt5/",                          "*.prl",                        true  },
+        { "/lib/cmake/Qt5Gui/",                   "Qt5GuiConfigExtras.cmake",     false },
+        { "/lib/cmake/Qt5LinguistTools/",         "Qt5LinguistToolsConfig.cmake", false },
+        { "/share/qt5/mkspecs/default-host/",     "qmake.conf",                   false },
 #ifdef OS_WINDOWS
-        { "/mkspecs/default/",            "qmake.conf",                   false },
-        { "/",                            ".qmake.cache",                 false },
-        { "/lib/",                        "prl.txt",                      false }
+        { "/share/qt5/mkspecs/default/",          "qmake.conf",                   false },
+        { "/",                                    ".qmake.cache",                 false },
+        { "/lib/",                                "prl.txt",                      false }
 #endif
     };
 
+    // Files for patching in Qt6.
+    static const TElement Elements6[] = {
+        { "/lib/",                                "*.prl",                        true  },
+        { "/lib/pkgconfig/",                      "Qt6*.pc",                      true  },
+        { "/share/qt6/",                          "*.pri",                        true  },
+        { "/share/qt6/",                          "*.prl",                        true  },
+        { "/lib/cmake/",                          "Qt6*.cmake",                   true  }
+    };
+
     // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
     m_TxtFilesForPatch.clear();
@@ -298,6 +311,10 @@
             Elements = Elements5;
             Count = sizeof(Elements5)/sizeof(Elements5[0]);
             break;
+        case '6' :
+            Elements = Elements6;
+            Count = sizeof(Elements6)/sizeof(Elements6[0]);
+            break;
         default :
             LOG_E("Unsupported Qt version (%c).", m_QMake.qtVersion());
             return false;
@@ -354,6 +371,23 @@
 #endif
     };
 
+    // Files for patching in Qt6.
+    static const TElement Elements6[] = {
+#if defined(OS_WINDOWS)
+        { "/bin/", "qmake.exe"    },
+        { "/bin/", "qmake6.exe"   },
+        { "/bin/", "lrelease.exe" },
+        { "/bin/", "qdoc.exe"     },
+        { "/bin/", "Qt6Core*.dll" },
+        { "/lib/", "Qt6Core*.dll" }
+#elif defined(OS_LINUX)
+        { "/bin/", "qmake"        },
+        { "/bin/", "lrelease"     },
+        { "/bin/", "qdoc"         },
+        { "/lib/", "libQt6Core.so" }
+#endif
+    };
+
     // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
     m_BinFilesForPatch.clear();
@@ -369,6 +403,10 @@
             Elements = Elements5;
             Count = sizeof(Elements5)/sizeof(Elements5[0]);
             break;
+        case '6' :
+            Elements = Elements6;
+            Count = sizeof(Elements6)/sizeof(Elements6[0]);
+            break;
         default :
             LOG_E("Unsupported Qt version (%c).", m_QMake.qtVersion());
             return false;
