--- qtbinpatcher-2.1.0/QtBinPatcher.cpp.orig	2014-10-19 21:58:34.945200000 +0400
+++ qtbinpatcher-2.1.0/QtBinPatcher.cpp	2014-10-19 22:33:35.582800000 +0400
@@ -256,7 +256,7 @@
     static const TElement Elements4[] = {
         { "/lib/",             "*.prl",              false },
         { "/demos/shared/",    "libdemo_shared.prl", false },
-        { "/lib/pkgconfig/",   "*.pc",               false },
+        { "/lib/pkgconfig/",   "Qt*.pc",             false },
 #if defined(OS_WINDOWS)
         { "/mkspecs/default/", "qmake.conf",         false },
         { "/",                 ".qmake.cache",       false }
@@ -268,9 +268,8 @@
 
     // Files for patching in Qt5.
     static const TElement Elements5[] = {
-        { "/",                            "*.la",                         true  },
         { "/",                            "*.prl",                        true  },
-        { "/",                            "*.pc",                         true  },
+        { "/lib/pkgconfig/",              "Qt*.pc",                       true  },
         { "/",                            "*.pri",                        true  },
         { "/lib/cmake/Qt5LinguistTools/", "Qt5LinguistToolsConfig.cmake", false },
         { "/mkspecs/default-host/",       "qmake.conf",                   false },
@@ -278,6 +277,8 @@
         { "/mkspecs/default/",            "qmake.conf",                   false },
         { "/",                            ".qmake.cache",                 false },
         { "/lib/",                        "prl.txt",                      false }
+#elif defined(OS_LINUX)
+        { "/lib/",                        "*.la",                         true  }
 #endif
     };
 
