diff -Naur qtbinpatcher-2.0.0-src/Functions.cpp qtbinpatcher-2.0.0-src-p/Functions.cpp
--- qtbinpatcher-2.0.0-src/Functions.cpp	2014-06-09 12:59:24.000000000 +0400
+++ qtbinpatcher-2.0.0-src-p/Functions.cpp	2014-06-10 07:56:40.516400000 +0400
@@ -104,7 +104,7 @@
 
 char Functions::nativeSeparator()
 {
-    #ifdef OS_WINDOWS
+    #if defined(OS_WINDOWS) && !defined(__MINGW32__)
         return '\\';
     #else
         return '/';
@@ -118,8 +118,13 @@
     #ifdef OS_WINDOWS
         string Result = path;
         for (string::iterator Iter = Result.begin(); Iter != Result.end(); ++Iter)
-            if (*Iter == '/')
-                *Iter = '\\';
+            #ifndef __MINGW32__
+                if (*Iter == '/')
+                    *Iter = '\\';
+            #else
+                if (*Iter == '\\')
+                    *Iter = '/';
+            #endif
         return Result;
     #else
         return path;
--- qtbinpatcher-2.0.0-src/QMake.cpp	2014-06-09 15:43:50 +0400
+++ qtbinpatcher-2.0.0-src-p/QMake.cpp	2014-06-11 09:10:46 +0400
@@ -75,7 +75,7 @@
             m_QMakePath.clear();
     }
     else {
-        string CurrentDir = currentDir();
+        string CurrentDir = toNativeSeparators(currentDir());
         m_QMakePath = CurrentDir + nativeSeparator() + m_QMakeName;
         if (isFileExists(m_QMakePath)) {
             // One level up.
