--- pyzmq-15.2.0.orig/setup.py	2016-01-12 14:31:19.000000000 +0000
+++ pyzmq-15.2.0/setup.py	2016-07-19 15:51:01.966344700 +0100
@@ -449,7 +449,10 @@
             if not zmq_prefix and not os.path.exists(local_dll):
                 fatal("ZMQ directory must be specified on Windows via setup.cfg or 'python setup.py configure --zmq=/path/to/zeromq2'")
             try:
-                shutil.copy(pjoin(zmq_prefix, 'lib', 'libzmq.dll'), local_dll)
+                if 'MSYSTEM' in os.environ:
+                    shutil.copy(pjoin(zmq_prefix, 'bin', 'libzmq.dll'), local_dll)
+                else:
+                    shutil.copy(pjoin(zmq_prefix, 'lib', 'libzmq.dll'), local_dll)
             except Exception:
                 if not os.path.exists(local_dll):
                     warn("Could not copy libzmq into zmq/, which is usually necessary on Windows."
