--- py2zmq-16.0.2/setup.py	2016-07-19 16:31:03.100214900 +0100
+++ pyzmq-16.0.2/setup.py	2016-07-19 16:38:12.475410600 +0100
@@ -120,7 +120,7 @@
     # is this necessary?
     if sys.platform.startswith('freebsd'):
         settings['libraries'].append('pthread')
-    elif sys.platform.startswith('win'):
+    elif sys.platform.startswith('win') and 'MSYSTEM' not in os.environ:
         # link against libzmq in build dir:
         plat = distutils.util.get_platform()
         temp = 'temp.%s-%i.%i' % (plat, sys.version_info[0], sys.version_info[1])
@@ -351,7 +351,7 @@
     def create_tempdir(self):
         self.erase_tempdir()
         os.makedirs(self.tempdir)
-        if sys.platform.startswith('win'):
+        if sys.platform.startswith('win') and 'MSYSTEM' not in os.environ:
             # fetch libzmq.dll into local dir
             local_dll = pjoin(self.tempdir, 'libzmq.dll')
             if not self.config['zmq_prefix'] and not os.path.exists(local_dll):
@@ -394,7 +394,7 @@
         """
         if 'bundle_libzmq_dylib' in self.config:
             return self.config['bundle_libzmq_dylib']
-        elif (sys.platform.startswith('win') or self.cross_compiling) \
+        elif ((sys.platform.startswith('win') and 'MSYSTEM' not in os.environ) or self.cross_compiling) \
                 and not self.config['libzmq_extension']:
             # always bundle libzmq on Windows and cross-compilation
             return True
@@ -443,9 +443,9 @@
             warn("Detected ZMQ version: %s. Some new features in libzmq may not be exposed by pyzmq." % vs)
             line()
 
-        if sys.platform.startswith('win'):
+        if sys.platform.startswith('win') and 'MSYSTEM' not in os.environ:
             # fetch libzmq.dll into local dir
             local_dll = localpath('zmq', libzmq_name + '.dll')
             if not zmq_prefix and not os.path.exists(local_dll):
                 fatal("ZMQ directory must be specified on Windows via setup.cfg or 'python setup.py configure --zmq=/path/to/zeromq2'")
             try:
@@ -565,7 +565,7 @@
             # this may not be sufficiently precise
             libzmq.define_macros.append(('ZMQ_USE_POLL', 1))
         
-        if sys.platform.startswith('win'):
+        if sys.platform.startswith('win') and 'MSYSTEM' not in os.environ:
             # include defines from zeromq msvc project:
             libzmq.define_macros.append(('FD_SETSIZE', 1024))
             libzmq.define_macros.append(('DLL_EXPORT', 1))
