diff -r -c winsparkle-0.5.orig/cmake/CMakeLists.txt winsparkle-0.5/cmake/CMakeLists.txt
*** winsparkle-0.5.orig/cmake/CMakeLists.txt	2016-03-13 13:46:32.000000000 -0300
--- winsparkle-0.5/cmake/CMakeLists.txt	2016-05-11 16:30:07.227016200 -0300
***************
*** 35,53 ****
  set(CONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}")
  set(CMAKE_DEBUG_POSTFIX "d")
  
- set(EXPAT_INCLUDE_DIRS "${ROOT_DIR}/3rdparty/expat/lib")
- 
- # bundled expat
- add_subdirectory(expat)
- 
- # bundled wxWidgets
- add_subdirectory(wxWidgets)
- 
  # WinSparkle
  set(SOURCE_DIR ${ROOT_DIR}/src)
  
  include_directories(${ROOT_DIR}/include)
! include_directories(${EXPAT_INCLUDE_DIRS})
  include_directories(${wxWidgets_INCLUDE_DIRS})
  
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
--- 35,45 ----
  set(CONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}")
  set(CMAKE_DEBUG_POSTFIX "d")
  
  # WinSparkle
  set(SOURCE_DIR ${ROOT_DIR}/src)
  
  include_directories(${ROOT_DIR}/include)
! find_package(wxWidgets REQUIRED)
  include_directories(${wxWidgets_INCLUDE_DIRS})
  
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
***************
*** 85,93 ****
                               -o ${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE}.o)
  endforeach()
  
! add_library(${PROJECT_NAME} SHARED ${SOURCES} $<TARGET_OBJECTS:wxWidgets> $<TARGET_OBJECTS:expat>)
  
! target_link_libraries(${PROJECT_NAME} wininet version rpcrt4 comctl32)
  
  set_target_properties(${PROJECT_NAME} PROPERTIES
                        VERSION ${LIB_MAJOR_VERSION}.${LIB_MINOR_VERSION}.${LIB_PATCH_VERSION}
--- 77,85 ----
                               -o ${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE}.o)
  endforeach()
  
! add_library(${PROJECT_NAME} SHARED ${SOURCES})
  
! target_link_libraries(${PROJECT_NAME} wininet version rpcrt4 comctl32 expat ${wxWidgets_LIBRARIES})
  
  set_target_properties(${PROJECT_NAME} PROPERTIES
                        VERSION ${LIB_MAJOR_VERSION}.${LIB_MINOR_VERSION}.${LIB_PATCH_VERSION}
diff -r -c winsparkle-0.5.orig/src/ui.cpp winsparkle-0.5/src/ui.cpp
*** winsparkle-0.5.orig/src/ui.cpp	2016-03-13 13:46:32.000000000 -0300
--- winsparkle-0.5/src/ui.cpp	2016-05-11 16:36:55.417792300 -0300
***************
*** 1464,1470 ****
      SignalReady();
  
      // Run the app:
!     wxMSWDisableSettingHighDPIAware();
      wxEntry(ms_hInstance);
  }
  
--- 1464,1470 ----
      SignalReady();
  
      // Run the app:
!     //wxMSWDisableSettingHighDPIAware();
      wxEntry(ms_hInstance);
  }
  
