--- a/cppwinrt/cmd_reader.h
+++ b/cppwinrt/cmd_reader.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <algorithm>
 #include <cassert>
 #include <array>
 #include <limits>
--- a/cppwinrt/helpers.h
+++ b/cppwinrt/helpers.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <chrono>
+
 namespace cppwinrt
 {
     static auto get_start_time()
--- a/cppwinrt/code_writers.h
+++ b/cppwinrt/code_writers.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <functional>
+
 namespace cppwinrt
 {
     struct finish_with
