--- osm-gps-map/Makefile.am.orig	2015-01-11 18:58:00.502800000 +0300
+++ osm-gps-map/Makefile.am	2015-01-11 18:58:16.420200000 +0300
@@ -7,7 +7,7 @@
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-osm_gps_mapdocdir = ${prefix}/doc/osm-gps-map
+osm_gps_mapdocdir = ${prefix}/share/doc/osm-gps-map
 osm_gps_mapdoc_DATA =                           \
 	README                                      \
 	COPYING                                     \
