--- Samples/EndlessWorld/include/EndlessWorld.h.orig	2013-12-11 19:33:26.968200000 +0400
+++ Samples/EndlessWorld/include/EndlessWorld.h	2013-12-11 19:37:59.153600000 +0400
@@ -132,7 +132,7 @@
 				Real x = (pt.x / 2) + 0.5f;
 				Real y = 1 - ((pt.y / 2) + 0.5f);
 
-				String lName = StringConverter::toString((unsigned long)(t))+"/"+"LodInfoLabel";
+				String lName = StringConverter::toString((size_t)(t))+"/"+"LodInfoLabel";
 
 				OverlayElement *l = OverlayManager::getSingleton().createOverlayElement("TextArea", lName);
 				l->setCaption("Target="+StringConverter::toString(t->getTargetLodLevel())+"\nHighest="+
