--- ogre3d-2.0.0.r7490.07ad8061ec8d/OgreMain/include/OgreStringConverter.h.orig	2015-02-18 00:20:40.288600000 +0300
+++ ogre3d-2.0.0.r7490.07ad8061ec8d/OgreMain/include/OgreStringConverter.h	2015-02-18 00:21:01.223800000 +0300
@@ -107,7 +107,7 @@
         static String toString(size_t val, 
             unsigned short width = 0, char fill = ' ', 
             std::ios::fmtflags flags = std::ios::fmtflags(0));
-        #if OGRE_COMPILER == OGRE_COMPILER_MSVC
+        #if OGRE_COMPILER == OGRE_COMPILER_MSVC || defined(__MINGW32__)
         /** Converts an unsigned long to a String. */
         static String toString(unsigned long val, 
             unsigned short width = 0, char fill = ' ', 
--- ogre3d-2.0.0.r7490.07ad8061ec8d/OgreMain/src/OgreStringConverter.cpp.orig	2015-02-18 00:21:25.996600000 +0300
+++ ogre3d-2.0.0.r7490.07ad8061ec8d/OgreMain/src/OgreStringConverter.cpp	2015-02-18 00:21:36.479800000 +0300
@@ -128,7 +128,7 @@
         stream << val;
         return stream.str();
     }
-#if OGRE_COMPILER == OGRE_COMPILER_MSVC
+#if OGRE_COMPILER == OGRE_COMPILER_MSVC || defined(__MINGW32__)
     //-----------------------------------------------------------------------
     String StringConverter::toString(unsigned long val, 
         unsigned short width, char fill, std::ios::fmtflags flags)
