--- ogre3d-1.9.0/OgreMain/CMakeLists.txt.orig	2013-12-10 14:50:36.000000000 +0400
+++ ogre3d-1.9.0/OgreMain/CMakeLists.txt	2013-12-15 21:25:07.186200000 +0400
@@ -220,7 +220,7 @@
 list(APPEND HEADER_FILES ${THREAD_HEADER_FILES})
 
 # Add needed definitions and nedmalloc include dir
-add_definitions(-DOGRE_NONCLIENT_BUILD -DFREEIMAGE_LIB -D_MT -D_USRDLL)
+add_definitions(-DOGRE_NONCLIENT_BUILD -D_MT -D_USRDLL)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src/nedmalloc)
 
 set(LIBRARIES ${PLATFORM_LIBS})
