--- qownnotes-25.4.2/libraries/fakevim/fakevim/fakevimhandler.cpp.orig	2025-04-11 21:37:00.000000000 +0200
+++ qownnotes-25.4.2/libraries/fakevim/fakevim/fakevimhandler.cpp	2025-04-12 09:31:27.710352700 +0200
@@ -1057,7 +1057,7 @@
         else if (cc == '\n')
             res += "<CR>";
         else
-            res += QStringLiteral("\\x%1").arg(c.unicode(), 2, 16, QLatin1Char('0'));
+            res += QStringLiteral("\\x%1").arg(int16_t(c.unicode()), 2, 16, QLatin1Char('0'));
     }
     return res;
 }
