--- openvr-1.0.15/src//CMakeLists.txt.orig	2018-05-15 15:10:34.425016800 +0300
+++ openvr-1.0.15/src//CMakeLists.txt	2018-05-15 15:11:41.721935100 +0300
@@ -104,11 +104,14 @@
 target_link_libraries(${LIBNAME} ${EXTRA_LIBS} ${CMAKE_DL_LIBS})
 target_include_directories(${LIBNAME} PUBLIC ${OPENVR_HEADER_DIR})
 
-install(TARGETS ${LIBNAME} DESTINATION lib)
+install(TARGETS ${LIBNAME}
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
 install(FILES ${PUBLIC_HEADER_FILES} DESTINATION include/openvr)
 
 # Generate a .pc file for linux environments
-if(PLATFORM_NAME MATCHES "linux")
+if (NOT MSVC)
     set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/share/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
     CONFIGURE_FILE("openvr.pc.in" "openvr.pc" @ONLY)
 
