--- a/build.rs
+++ b/build.rs
@@ -72,6 +72,13 @@ fn target_os_is(name: &str) -> bool {
     }
 }
 
+fn target_abi_is(name: &str) -> bool {
+    match env::var_os("CARGO_CFG_TARGET_ABI") {
+        Some(s) => s == name,
+        None => false,
+    }
+}
+
 /// Try to find a version of llvm-config that is compatible with this crate.
 ///
 /// If $LLVM_SYS_<VERSION>_PREFIX is set, look for llvm-config ONLY in there. The assumption is
@@ -461,6 +468,10 @@ fn get_system_libcpp() -> Option<&'static str> {
         Some("c++")
     } else if target_os_is("freebsd") || target_os_is("openbsd") {
         Some("c++")
+    } else if target_abi_is("llvm") {
+        // These are `*-pc-windows-gnullvm` targets, which are based on LLVM
+        // toolchain, so libc++ is the only cxx std library here
+        Some("c++")
     } else {
         // Otherwise assume GCC's libstdc++.
         // This assumption is probably wrong on some platforms, but it can be
