--- a/cmake/FindDXSDK.cmake
+++ b/cmake/FindDXSDK.cmake
@@ -38,6 +38,7 @@
         NAMES
             D3D11.h D3Dcompiler.h
         PATHS
+            "${DXSDK_ROOT}/include"
             "${DXSDK_LOCATION}/Include"
             "$ENV{DXSDK_LOCATION}/Include"
             "${DXSDK_ROOT}/Include"
@@ -56,8 +57,12 @@
     endif()
 
     find_path(LIBRARY_DIR
+        NAMES
             d3d11.lib
+            libd3d11.a
+            libd3d11.dll.a
         PATHS
+            "${DXSDK_ROOT}/lib"
             "${DXSDK_LOCATION}/Lib/${ARCH}"
             "$ENV{DXSDK_LOCATION}/Lib/${ARCH}"
             "${DXSDK_ROOT}/Lib/${ARCH}"
@@ -74,10 +79,9 @@
 
         find_library(DXSDK_${DX_LIB}_LIBRARY
             NAMES 
-                ${DX_LIB}.lib
+                ${DX_LIB}
             PATHS
                 ${DXSDK_LIBRARY_DIR}
-            NO_DEFAULT_PATH
         )
 
         if(NOT DXSDK_${DX_LIB}_LIBRARY)
--- a/cmake/FindOpenCL.cmake
+++ b/cmake/FindOpenCL.cmake
@@ -66,7 +66,7 @@
 
     find_library( OPENCL_LIBRARIES
         NAMES
-            OpenCL.lib
+            OpenCL
         PATHS
             ENV OpenCL_LIBPATH
             "$ENV{ATISTREAMSDKROOT}/lib/x86_64"
--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -43,9 +43,7 @@
 
     #---------------------------------------------------------------------------
     if( OPENMP_FOUND )
-        if (CMAKE_COMPILER_IS_GNUCXX)
-            list(APPEND PLATFORM_CPU_LIBRARIES gomp)
-        endif()
+        list(APPEND PLATFORM_CPU_LIBRARIES ${OpenMP_CXX_LIBRARIES})
     endif()
 
     if( TBB_FOUND )
@@ -172,7 +170,7 @@
 
 
     # Build dynamic libs  ----------------------------------
-    if (BUILD_SHARED_LIBS AND NOT WIN32 AND NOT IOS)
+    if (BUILD_SHARED_LIBS AND NOT MSVC AND NOT IOS)
 
         # generate dynamic-link targets
 
--- a/opensubdiv/osd/CMakeLists.txt
+++ b/opensubdiv/osd/CMakeLists.txt
@@ -74,9 +74,7 @@
 
     list(APPEND PUBLIC_HEADER_FILES ${OPENMP_PUBLIC_HEADERS})
 
-    if (CMAKE_COMPILER_IS_GNUCXX)
-        list(APPEND PLATFORM_CPU_LIBRARIES gomp)
-    endif()
+    list(APPEND PLATFORM_CPU_LIBRARIES ${OpenMP_CXX_LIBRARIES})
 endif()
 
 list(APPEND DOXY_HEADER_FILES ${OPENMP_PUBLIC_HEADERS})
@@ -97,9 +95,7 @@
 
     list(APPEND PUBLIC_HEADER_FILES ${TBB_PUBLIC_HEADERS})
 
-    if (CMAKE_COMPILER_IS_GNUCXX)
-        list(APPEND PLATFORM_CPU_LIBRARIES gomp)
-    endif()
+    list(APPEND PLATFORM_CPU_LIBRARIES ${OpenMP_CXX_LIBRARIES})
 
     list(APPEND PLATFORM_CPU_LIBRARIES
         ${TBB_LIBRARIES}
