diff -Naur glib-2.46.0-orig/gio/gsocket.c glib-2.46.0/gio/gsocket.c
--- glib-2.46.0-orig/gio/gsocket.c	2015-08-31 16:08:34.000000000 +0300
+++ glib-2.46.0/gio/gsocket.c	2015-09-22 09:09:04.614548700 +0300
@@ -1951,7 +1951,7 @@
 
 #if !defined(HAVE_IF_NAMETOINDEX) && defined(G_OS_WIN32)
 static guint
-if_nametoindex (const gchar *iface)
+w32_if_nametoindex (const gchar *iface)
 {
   PIP_ADAPTER_ADDRESSES addresses = NULL, p;
   gulong addresses_len = 0;
@@ -2004,6 +2004,7 @@
 }
 
 #define HAVE_IF_NAMETOINDEX 1
+#define if_nametoindex w32_if_nametoindex
 #endif
 
 static gboolean
