--- amqp-cpp-4.1.6/CMakeLists.txt.orig	2020-05-07 10:52:04.145372100 +0300
+++ amqp-cpp-4.1.6/CMakeLists.txt	2020-05-07 11:14:34.490639100 +0300
@@ -83,6 +83,9 @@
     add_library(${PROJECT_NAME} SHARED ${src_MAIN} ${src_LINUX_TCP})
     # set shared lib version
     set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${SO_VERSION})
+    if (WIN32)
+        target_link_libraries(${PROJECT_NAME} PUBLIC ws2_32)
+    endif()
 else()
     # create static lib
     #add_library(${PROJECT_NAME} STATIC ${SRCS})
