--- tclvfs/configure.orig	2015-08-01 02:38:12.000000000 +0200
+++ tclvfs/configure	2015-08-01 02:40:06.818379500 +0200
@@ -1351,7 +1351,7 @@
 echo "${ECHO_T}ok (TEA ${TEA_VERSION})" >&6
     fi
     case "`uname -s`" in
-	*win32*|*WIN32*|*CYGWIN_NT*|*CYGWIN_9*|*CYGWIN_ME*|*MINGW32_*)
+	*win32*|*WIN32*|*CYGWIN_NT*|*CYGWIN_9*|*CYGWIN_ME*|*MINGW*)
 	    # Extract the first word of "cygpath", so it can be a program name with args.
 set dummy cygpath; ac_word=$2
 echo "$as_me:$LINENO: checking for $ac_word" >&5
