diff -bur newton-4.00-o/sdk/dCore/ndUtils.h newton-4.00/sdk/dCore/ndUtils.h
--- newton-4.00-o/sdk/dCore/ndUtils.h	2025-04-20 14:14:38.360828800 -0600
+++ newton-4.00/sdk/dCore/ndUtils.h	2025-04-20 14:14:58.841847600 -0600
@@ -267,7 +267,7 @@
 	public:
 	//#define D_FLOAT_EXCEPTIONS_MASK	(EM_INVALID | EM_DENORMAL | EM_ZERODIVIDE)
 	#if defined (WIN32) || defined(_WIN32)
-		#define D_FLOAT_EXCEPTIONS_MASK	(EM_INVALID | EM_DENORMAL)
+		#define D_FLOAT_EXCEPTIONS_MASK	(_EM_INVALID | _EM_DENORMAL)
 	#else
 		#define D_FLOAT_EXCEPTIONS_MASK	(FE_INVALID | FE_INEXACT)
 	#endif
diff -bur newton-4.00-o/thirdParty/hacd/src/VHACD_Lib/inc/vhacdVector.h newton-4.00/thirdParty/hacd/src/VHACD_Lib/inc/vhacdVector.h
--- newton-4.00-o/thirdParty/hacd/src/VHACD_Lib/inc/vhacdVector.h	2025-04-20 14:14:40.222419500 -0600
+++ newton-4.00/thirdParty/hacd/src/VHACD_Lib/inc/vhacdVector.h	2025-04-20 14:14:47.961880000 -0600
@@ -17,6 +17,7 @@
 #define _ND_VHACD_VECTOR_H_
 #include <iostream>
 #include <math.h>
+#include <stdint.h>
 
 namespace nd_ 
 {
