diff -bur newton-4.00-o/applications/ndSandbox/CMakeLists.txt newton-4.00/applications/ndSandbox/CMakeLists.txt
--- newton-4.00-o/applications/ndSandbox/CMakeLists.txt	2025-04-20 14:18:53.386733600 -0600
+++ newton-4.00/applications/ndSandbox/CMakeLists.txt	2025-04-20 14:19:14.536978300 -0600
@@ -106,7 +106,9 @@
 
     target_link_libraries (${projectName} glu32 opengl32)
 
+	if(MSVC)
 	target_link_options(${projectName} PUBLIC "/DEBUG") 
+	endif()
 
     if(CMAKE_VS_MSBUILD_COMMAND OR CMAKE_VS_DEVENV_COMMAND)
 		set_target_properties(${projectName} PROPERTIES COMPILE_FLAGS "/YundSandboxStdafx.h")
diff -bur newton-4.00-o/sdk/CMakeLists.txt newton-4.00/sdk/CMakeLists.txt
--- newton-4.00-o/sdk/CMakeLists.txt	2025-04-20 14:18:51.416820600 -0600
+++ newton-4.00/sdk/CMakeLists.txt	2025-04-20 14:19:45.904871900 -0600
@@ -128,7 +128,7 @@
 			target_link_libraries (${projectName} ndSolverOpenCL)
 		endif()
 
-		if(MSVC OR MINGW)
+		if(MSVC)
 			target_link_options(${projectName} PUBLIC "/DEBUG") 
 		endif()
 
diff -bur newton-4.00-o/sdk/dCollision/CMakeLists.txt newton-4.00/sdk/dCollision/CMakeLists.txt
--- newton-4.00-o/sdk/dCollision/CMakeLists.txt	2025-04-20 14:18:51.433216900 -0600
+++ newton-4.00/sdk/dCollision/CMakeLists.txt	2025-04-20 14:19:44.680198600 -0600
@@ -32,7 +32,7 @@
 	add_definitions(-D_D_COLLISION_EXPORT_DLL)
 	add_library(${projectName} SHARED ${CPP_SOURCE})
 
-	if(MSVC OR MINGW)
+	if(MSVC)
 		target_link_options(${projectName} PUBLIC "/DEBUG") 
 	endif()
 else()
diff -bur newton-4.00-o/sdk/dCore/CMakeLists.txt newton-4.00/sdk/dCore/CMakeLists.txt
--- newton-4.00-o/sdk/dCore/CMakeLists.txt	2025-04-20 14:18:51.441726700 -0600
+++ newton-4.00/sdk/dCore/CMakeLists.txt	2025-04-20 14:19:43.920998500 -0600
@@ -26,7 +26,7 @@
 	add_definitions(-D_D_CORE_EXPORT_DLL)
 	add_library(${projectName} SHARED ${CPP_SOURCE})
 
-	if(MSVC OR MINGW)
+	if(MSVC)
 		target_link_options(${projectName} PUBLIC "/DEBUG") 
 	endif()
 
diff -bur newton-4.00-o/sdk/dNewton/CMakeLists.txt newton-4.00/sdk/dNewton/CMakeLists.txt
--- newton-4.00-o/sdk/dNewton/CMakeLists.txt	2025-04-20 14:18:51.465359900 -0600
+++ newton-4.00/sdk/dNewton/CMakeLists.txt	2025-04-20 14:19:42.993469400 -0600
@@ -93,7 +93,7 @@
 	add_definitions(-D_D_NEWTON_EXPORT_DLL)
 	add_library(${projectName} SHARED ${CPP_SOURCE})
 
-	if(MSVC OR MINGW)
+	if(MSVC)
 		target_link_options(${projectName} PUBLIC "/DEBUG") 
 	endif()
 else()
diff -bur newton-4.00-o/sdk/dNewton/dExtensions/dAvx2/CMakeLists.txt newton-4.00/sdk/dNewton/dExtensions/dAvx2/CMakeLists.txt
--- newton-4.00-o/sdk/dNewton/dExtensions/dAvx2/CMakeLists.txt	2025-04-20 14:18:51.466863000 -0600
+++ newton-4.00/sdk/dNewton/dExtensions/dAvx2/CMakeLists.txt	2025-04-20 14:19:41.490064300 -0600
@@ -51,7 +51,7 @@
 	add_library(${projectName} SHARED ${CPP_SOURCE})
 endif()
 
-if(MSVC OR MINGW)
+if(MSVC)
 	target_link_options(${projectName} PUBLIC "/DEBUG") 
 endif()
 
