diff -rupN google-breakpad-0.1.svn.969/configure.ac google-breakpad-0.1.svn.969_mod/configure.ac
--- google-breakpad-0.1.svn.969/configure.ac	2012-06-12 01:52:23.000000000 +0200
+++ google-breakpad-0.1.svn.969_mod/configure.ac	2012-06-12 13:20:41.264351886 +0200
@@ -44,7 +44,7 @@ AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_CPP
 AC_PROG_CXX
-AC_PROG_RANLIB
+LT_INIT([disable-shared])
 AC_CHECK_TOOL(WINDRES, [windres])
 AM_CONDITIONAL(GCC, test "$GCC" = yes)   # let the Makefile know if we're gcc
 
diff -rupN google-breakpad-0.1.svn.969/Makefile.am google-breakpad-0.1.svn.969_mod/Makefile.am
--- google-breakpad-0.1.svn.969/Makefile.am	2012-06-12 01:52:23.000000000 +0200
+++ google-breakpad-0.1.svn.969_mod/Makefile.am	2012-06-12 13:14:22.051174459 +0200
@@ -122,22 +122,22 @@
 pkgconfig_DATA =
 
 ## Libraries
-noinst_LIBRARIES =
-lib_LIBRARIES =
+noinst_LTLIBRARIES =
+lib_LTLIBRARIES =
 bin_PROGRAMS =
 check_PROGRAMS =
 
 if !DISABLE_PROCESSOR
-lib_LIBRARIES += src/libbreakpad.a
+lib_LTLIBRARIES += src/libbreakpad.la
 pkgconfig_DATA += breakpad.pc
-noinst_LIBRARIES += src/third_party/libdisasm/libdisasm.a
+noinst_LTLIBRARIES += src/third_party/libdisasm/libdisasm.la
 endif
 
 if LINUX_HOST
-lib_LIBRARIES += src/client/linux/libbreakpad_client.a
+lib_LTLIBRARIES += src/client/linux/libbreakpad_client.la
 pkgconfig_DATA += breakpad-client.pc
 
-src_client_linux_libbreakpad_client_a_SOURCES = \
+src_client_linux_libbreakpad_client_la_SOURCES = \
 	src/client/linux/crash_generation/crash_generation_client.cc \
 	src/client/linux/crash_generation/crash_generation_server.cc \
 	src/client/linux/dump_writer_common/seccomp_unwinder.cc \
@@ -161,13 +161,13 @@
 	src/common/linux/memory_mapped_file.cc \
 	src/common/linux/safe_readlink.cc
 if ANDROID_HOST
-src_client_linux_libbreakpad_client_a_SOURCES += \
+src_client_linux_libbreakpad_client_la_SOURCES += \
 	src/common/android/breakpad_getcontext.S
 endif
 endif LINUX_HOST
 
 if !DISABLE_PROCESSOR
-src_libbreakpad_a_SOURCES = \
+src_libbreakpad_la_SOURCES = \
 	src/google_breakpad/common/breakpad_types.h \
 	src/google_breakpad/common/minidump_format.h \
 	src/google_breakpad/common/minidump_size.h \
@@ -283,9 +283,9 @@
 	src/processor/tokenize.cc \
 	src/processor/tokenize.h
 
-src_libbreakpad_a_LIBADD = src/third_party/libdisasm/libdisasm.a
+src_libbreakpad_la_LIBADD = src/third_party/libdisasm/libdisasm.la
 
-src_third_party_libdisasm_libdisasm_a_SOURCES = \
+src_third_party_libdisasm_libdisasm_la_SOURCES = \
 	src/third_party/libdisasm/ia32_implicit.c \
 	src/third_party/libdisasm/ia32_implicit.h \
 	src/third_party/libdisasm/ia32_insn.c \
@@ -489,7 +489,8 @@
 	-I$(top_srcdir)/src/testing/include \
 	-I$(top_srcdir)/src/testing/gtest/include \
 	-I$(top_srcdir)/src/testing/gtest \
-	-I$(top_srcdir)/src/testing
+	-I$(top_srcdir)/src/testing \
+	$(AM_CPPFLAGS)
 src_client_linux_linux_client_unittest_shlib_LDFLAGS = \
 	-shared \
 	-Wl,-h,linux_client_unittest_shlib
@@ -525,8 +526,8 @@
 
 src_client_linux_linux_client_unittest_shlib_DEPENDENCIES = \
 	src/client/linux/linux_dumper_unittest_helper \
-	src/client/linux/libbreakpad_client.a \
-	src/libbreakpad.a
+	src/client/linux/libbreakpad_client.la \
+	src/libbreakpad.la
 
 src_client_linux_linux_client_unittest_SOURCES =
 src_client_linux_linux_client_unittest_LDFLAGS = \
@@ -549,8 +550,11 @@
 	src/common/linux/elf_core_dump.cc
 
 src_tools_linux_core2md_core2md_LDADD = \
-	src/client/linux/libbreakpad_client.a
+	src/client/linux/libbreakpad_client.la \
+	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
+src_tools_linux_dump_syms_dump_syms_CPPFLAGS = \
+	$(AM_CPPFLAGS)
 src_tools_linux_dump_syms_dump_syms_SOURCES = \
 	src/common/dwarf_cfi_to_module.cc \
 	src/common/dwarf_cu_to_module.cc \
@@ -572,6 +576,8 @@
 	src/common/linux/safe_readlink.cc \
 	src/tools/linux/dump_syms/dump_syms.cc
 
+src_tools_linux_md2core_minidump_2_core_CPPFLAGS = \
+	$(AM_CPPFLAGS)
 src_tools_linux_md2core_minidump_2_core_SOURCES = \
 	src/common/linux/memory_mapped_file.cc \
 	src/tools/linux/md2core/minidump-2-core.cc
@@ -629,7 +635,8 @@
 	-I$(top_srcdir)/src/testing/gtest/include \
 	-I$(top_srcdir)/src/testing/gtest \
 	-I$(top_srcdir)/src/testing \
-	$(PTHREAD_CFLAGS)
+	$(PTHREAD_CFLAGS) \
+	$(AM_CPPFLAGS)
 src_common_dumper_unittest_LDADD = $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 endif
 
@@ -775,7 +782,7 @@
 	src/processor/stackwalker_x86.o \
 	src/processor/symbolic_constants_win.o \
 	src/processor/tokenize.o \
-	src/third_party/libdisasm/libdisasm.a \
+	src/third_party/libdisasm/libdisasm.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
 src_processor_disassembler_x86_unittest_SOURCES = \
@@ -791,7 +798,7 @@
 	-I$(top_srcdir)/src/testing
 src_processor_disassembler_x86_unittest_LDADD = \
 	src/processor/disassembler_x86.o \
-	src/third_party/libdisasm/libdisasm.a \
+	src/third_party/libdisasm/libdisasm.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
 src_processor_fast_source_line_resolver_unittest_SOURCES = \
@@ -910,7 +917,7 @@
 	src/processor/stackwalker_x86.o \
 	src/processor/symbolic_constants_win.o \
 	src/processor/tokenize.o \
-	src/third_party/libdisasm/libdisasm.a \
+	src/third_party/libdisasm/libdisasm.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 
 src_processor_minidump_unittest_SOURCES = \
@@ -1051,7 +1058,7 @@
 	src/testing/gtest/src/gtest_main.cc \
 	src/testing/src/gmock-all.cc
 src_processor_stackwalker_amd64_unittest_LDADD = \
-	src/libbreakpad.a \
+	src/libbreakpad.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 src_processor_stackwalker_amd64_unittest_CPPFLAGS = \
 	-I$(top_srcdir)/src \
@@ -1067,7 +1074,7 @@
 	src/testing/gtest/src/gtest_main.cc \
 	src/testing/src/gmock-all.cc
 src_processor_stackwalker_arm_unittest_LDADD = \
-	src/libbreakpad.a \
+	src/libbreakpad.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 src_processor_stackwalker_arm_unittest_CPPFLAGS = \
 	-I$(top_srcdir)/src \
@@ -1083,7 +1090,7 @@
 	src/testing/gtest/src/gtest_main.cc \
 	src/testing/src/gmock-all.cc
 src_processor_stackwalker_arm64_unittest_LDADD = \
-	src/libbreakpad.a \
+	src/libbreakpad.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 src_processor_stackwalker_arm64_unittest_CPPFLAGS = \
 	-I$(top_srcdir)/src \
@@ -1099,7 +1106,7 @@
 	src/testing/gtest/src/gtest_main.cc \
 	src/testing/src/gmock-all.cc
 src_processor_stackwalker_address_list_unittest_LDADD = \
-	src/libbreakpad.a \
+	src/libbreakpad.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 src_processor_stackwalker_address_list_unittest_CPPFLAGS = \
 	-I$(top_srcdir)/src \
@@ -1115,7 +1122,7 @@
 	src/testing/gtest/src/gtest_main.cc \
 	src/testing/src/gmock-all.cc
 src_processor_stackwalker_mips_unittest_LDADD = \
-	src/libbreakpad.a \
+	src/libbreakpad.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 src_processor_stackwalker_mips_unittest_CPPFLAGS = \
 	-I$(top_srcdir)/src \
@@ -1131,7 +1138,7 @@
 	src/testing/gtest/src/gtest_main.cc \
 	src/testing/src/gmock-all.cc
 src_processor_stackwalker_x86_unittest_LDADD = \
-	src/libbreakpad.a \
+	src/libbreakpad.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 src_processor_stackwalker_x86_unittest_CPPFLAGS = \
 	-I$(top_srcdir)/src \
@@ -1219,7 +1226,7 @@
 	src/processor/stackwalker_sparc.o \
 	src/processor/stackwalker_x86.o \
 	src/processor/tokenize.o \
-	src/third_party/libdisasm/libdisasm.a \
+	src/third_party/libdisasm/libdisasm.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS) $(SOCKET_LIBS)
 
 src_processor_minidump_stackwalk_SOURCES = \
@@ -1258,7 +1265,7 @@
 	src/processor/stackwalker_x86.o \
 	src/processor/symbolic_constants_win.o \
 	src/processor/tokenize.o \
-	src/third_party/libdisasm/libdisasm.a \
+	src/third_party/libdisasm/libdisasm.la \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS) $(SOCKET_LIBS) @LIBOBJS@
 
 endif !DISABLE_PROCESSOR
