--- breakpad.build.x86_64/src/common/pecoff/pecoffutils.cc.orig	2015-07-18 20:50:02.583426500 +0300
+++ breakpad.build.x86_64/src/common/pecoff/pecoffutils.cc	2015-07-18 20:50:19.279381500 +0300
@@ -267,9 +267,8 @@
           (export_rva < (data_directory_export_entry->mVirtualAddress + data_directory_export_entry->mSize)))
         continue;
 
-      Module::Extern* ext = new Module::Extern;
+      Module::Extern* ext = new Module::Extern(export_rva + GetLoadingAddress(obj_base));
       ext->name = export_name;
-      ext->address = export_rva + GetLoadingAddress(obj_base);
       module->AddExtern(ext);
     }
 
