--- origsrc/svn/src/tools/gyp/pylib/gyp/__init__.py	2012-08-07 22:25:40.000000000 +0100
+++ src/svn/src/tools/gyp/pylib/gyp/__init__.py	2014-02-12 00:39:43.298921600 +0000
@@ -328,7 +328,7 @@ def gyp_main(args):
   # behavior to be used elsewhere.
   # TODO(mark): Remove this option when http://crbug.com/35878 is fixed.
   parser.add_option('--no-circular-check', dest='circular_check',
-                    action='store_false', default=True, regenerate=False,
+                    action='store_false', default=True, regenerate=True,
                     help="don't check for circular relationships between files")
 
   # We read a few things from ~/.gyp, so set up a var for that.
