From fa6518f7eaaa0896f5502f230f4c214110708d93 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E8=82=96=E5=AF=85=E4=B8=9C?= <xydarcher@uestc.edu.cn>
Date: Thu, 3 Jan 2019 11:39:11 +0800
Subject: [PATCH] fix for mingw build.

---
 configure.ac | 18 +++++++++++++++---
 makefile.in  |  6 ++++++
 2 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1445de97..9baeb0d2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,9 +85,17 @@ AC_FUNC_ERROR_AT_LINE
 AC_FUNC_MALLOC
 AC_FUNC_REALLOC
 
-# AC_CHECK_LIB (library, function, [action-if-found], [action-if-not-found], [other-libraries])
-AC_CHECK_LIB([c],[main], [],[AC_MSG_ERROR(Could not use standard C library)],   [])
-AC_CHECK_LIB([m],[main], [],[AC_MSG_ERROR(Could not use standard math library)],[])
+# AC_CHECK_LIB (library, function, [action-if-found], [action-if-not-found], [other-libraries])AC_CANONICAL_HOST
+AC_CANONICAL_HOST
+case $host_os in
+	mingw*)
+		AC_CHECK_LIB([m],[sqrt], [],[AC_MSG_ERROR(Could not use standard math library)],[])
+		;;
+	*)
+		AC_CHECK_LIB([c],[main], [],[AC_MSG_ERROR(Could not use standard C library)],   [])
+		AC_CHECK_LIB([m],[main], [],[AC_MSG_ERROR(Could not use standard math library)],[])
+		;;
+esac 
 
 # AC_CHECK_FUNC(function, [action-if-found], [action-if-not-found])
 AC_CHECK_FUNC([malloc],  [],[AC_MSG_ERROR(Could not use malloc())])
@@ -241,6 +249,10 @@ darwin*)
     SH_LIB=libliquid.dylib
     REBIND=""
     ;;
+mingw*)
+    SH_LIB=libliquid.dll
+    REBIND=""
+    ;;
 *)
     SH_LIB=libliquid.so
     REBIND=ldconfig
diff --git a/makefile.in b/makefile.in
index ac47bc2c..e37a49ea 100644
--- a/makefile.in
+++ b/makefile.in
@@ -1166,6 +1166,12 @@ libliquid.a: $(objects)
 libliquid.dylib: $(objects)
 	$(CC) -dynamiclib -install_name $@ -o $@ $^ $(LDFLAGS) $(LIBS)
 
+# windows mingw
+#
+# gcc -shared -install_name libliquid.dll -o libliquid.dll libmodem.a libutility.a 
+libliquid.dll: $(objects)
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive,--output-def,libliquid.def,--out-implib,imp_libliquid.lib $^ -Wl,-no-whole-archive $(LIBS)
+
 # linux, et al
 libliquid.so: libliquid.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Xlinker -soname=$@ -o $@ -Wl,-whole-archive $^ -Wl,-no-whole-archive $(LIBS)
-- 
2.20.1.windows.1

