--- swipl-9.2.1/cmake/port/Windows.cmake      2024-02-15 07:31:29.593572000 +0100
+++ swipl-9.2.1/cmake/port/Windows.cmake      2024-02-15 07:30:51.689756700 +0100
@@ -17,6 +17,8 @@
 if(MINGW_ROOT)
   include(port/MinGW)
 elseif(CONDA)
   include(port/CondaWindows)
+elseif(MSYS2)
+  include(port/msys2)
 endif()

 if(NOT PLHOME)

+++ swipl-9.2.1/cmake/port/msys2.cmake       2024-02-15 07:32:00.0 +0100
@@ -0,0 +1,4 @@
+message("Building for MSYS2")
+
+set(PLHOME no-home)
+set(PLRELHOME ../lib/${SWIPL_INSTALL_DIR})
