--- swipl-8.4.3/packages/cmake/PrologPackage.cmake	2022-03-17 15:47:46 +0000
+++ swipl-8.4.3/packages/cmake/PrologPackage.cmake	2022-03-17 15:47:15 +0000
@@ -221,6 +221,7 @@
 
 function(install_dll)
 if(WIN32)
+  if(NOT "${WIN32_DLLS}" STREQUAL "")
   set(dlls)
 
   foreach(lib ${ARGN})
@@ -264,5 +265,6 @@
        DESTINATION ${CMAKE_BINARY_DIR}/src)
   install(FILES ${dlls}
 	  DESTINATION ${SWIPL_INSTALL_ARCH_EXE})
+  endif()
 endif()
 endfunction()
