--- swipl-8.4.3/cmake/CStack.cmake.old  2022-06-28 13:46:34.320820200 +0200
+++ swipl-8.4.3/cmake/CStack.cmake      2022-06-28 14:56:54.511545700 +0200
@@ -1,11 +1,9 @@
 function(target_c_stack target limit)
 if(WIN32)
-  if(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
-    set_target_properties(${target} PROPERTIES
-                         LINK_FLAGS "-Wl,--stack,${limit}")
+  if(MSVC)
+    set_target_properties(${target} PROPERTIES LINK_FLAGS /STACK:${limit})
   else()
-    set_target_properties(${target} PROPERTIES
-                         LINK_FLAGS /STACK:${limit})
+    set_target_properties(${target} PROPERTIES LINK_FLAGS "-Wl,--stack,${limit}")
   endif()
 endif()
 endfunction()
