--- swipl-8.4.3/CMakeLists.txt.old	2022-06-17 10:48:26 +0000
+++ swipl-8.4.3/CMakeLists.txt	2022-06-17 10:49:01 +0000
@@ -120,7 +120,7 @@
 ################
 # Installation directories
 
-if(WIN32)
+if(WIN32 AND NOT MSYS2)
 
   set(SWIPL_INSTALL_DIR "."
       CACHE STRING "Directory below <prefix> for installation")
@@ -134,7 +134,7 @@
   set(SWIPL_TMP_DIR "c:/tmp" CACHE STRING
       "Directory to be used if the environment variable TEMP is not set")
 
-else(WIN32)
+else()
 
   if(SWIPL_VERSIONED_DIR)
     set(SWIPL_INSTALL_DIR_DEF  swipl-${SWIPL_VERSION_STRING})
@@ -169,7 +169,7 @@
 
   set(SWIPL_INSTALL_MODULES  ${SWIPL_INSTALL_PREFIX}/lib/${SWIPL_ARCH})
 
-endif(WIN32)
+endif()
 
 set(SWIPL_CMAKE_NAMESPACE swipl::)
 set(SWIPL_INSTALL_CMAKE_CONFIG_DIR lib/cmake/swipl)
