--- swipl-9.0.0/boot/build_home.pl      2022-02-10 11:09:04 +0000
+++ swipl-9.0.0/boot/build_home.pl      2022-06-27 17:27:29 +0000
@@ -199,5 +199,8 @@
 add_package_path(PkgBinDir) :-
+    (   current_prolog_flag(windows, true)
+    ->  win_add_dll_directory(PkgBinDir, _)
+    ;   true),
     assertz(user:file_search_path(foreign, PkgBinDir)).

 :- if(\+ current_prolog_flag(emscripten, true)).
 % disabled as we do not (yet) have packages and opendir() is broken
@@ -206,6 +206,13 @@
           add_package(Pkg, PkgBinDir)).
 :- endif.

+% Add %MINGW_PREFIX%/bin to dll search path under MSYS2
+:- if(current_prolog_flag(windows, true)).
+:- getenv('MINGW_PREFIX', Prefix),
+    format(atom(Bin), '~w/bin', [Prefix]),
+    win_add_dll_directory(Bin, _).
+:- endif.
+
 %!  set_version_info
 %
 %   Indicate we are running from the   build directory rather than using
