--- swipl-9.0.2/packages/semweb/ntriples.c	2022-12-08 10:13:03 +0000
+++ swipl-9.0.2/packages/semweb/ntriples.c.new	2022-12-08 10:12:40 +0000
@@ -38,6 +38,7 @@
 #include <string.h>
 #include <assert.h>
 #include "turtle_chars.c"
+#include "pl-utf8.h"
 
 static atom_t ATOM_end_of_file;
 
@@ -273,9 +275,8 @@
     free(b->buf);
 }
 
-
 static inline int
-addBuf(string_buffer *b, int c)
+addBuf_wchar(string_buffer *b, wchar_t c)
 { if ( b->in < b->end )
   { *b->in++ = c;
     return TRUE;
@@ -284,6 +285,15 @@
   return growBuffer(b, c);
 }
 
+static inline int
+addBuf(string_buffer *b, int c)
+{ if ( c <= 0xffff )
+    return addBuf_wchar(b, c) ;
+
+  int l, t ;
+  utf16_encode(c, &l, &t);
+  return addBuf_wchar(b, l) && addBuf_wchar(b, t) ;
+}
 
 static inline int
 bufSize(string_buffer *b)
