--- swipl-8.4.3/CMakeLists.txt.old	2022-06-17 10:48:26 +0000
+++ swipl-8.4.3/CMakeLists.txt	2022-06-17 10:49:01 +0000
@@ -104,7 +104,7 @@
 ################
 # Installation directories
 
-if(WIN32)
+if(WIN32 AND NOT MSYS2)
 
   set(SWIPL_INSTALL_DIR "."
       CACHE STRING "Directory below <prefix> for installation")
@@ -118,7 +118,7 @@
   set(SWIPL_TMP_DIR "c:/tmp" CACHE STRING
       "Directory to be used if the environment variable TEMP is not set")
 
-else(WIN32)
+else()
 
   if(SWIPL_VERSIONED_DIR)
     set(SWIPL_INSTALL_DIR_DEF  swipl-${SWIPL_VERSION_STRING})
@@ -153,7 +153,7 @@
 
   set(SWIPL_INSTALL_MODULES  ${SWIPL_INSTALL_PREFIX}/lib/${SWIPL_ARCH})
 
-endif(WIN32)
+endif()
 
 set(SWIPL_CMAKE_NAMESPACE swipl::)
 set(SWIPL_INSTALL_CMAKE_CONFIG_DIR lib/cmake/swipl)
--- swipl-8.4.3/src/CMakeLists.txt.old	2022-06-17 12:01:22 +0000
+++ swipl-8.4.3/src/CMakeLists.txt	2022-06-17 12:00:52 +0000
@@ -670,16 +670,17 @@
	    FILES ${CMAKE_CURRENT_SOURCE_DIR}/swipl.ico
	    DESTINATION ${SWIPL_INSTALL_PREFIX})

-else(WIN32)
+endif()

 if(NOT BUILD_MACOS_BUNDLE)
 # Create symbolic link from public installation dir to executables
 install(DIRECTORY DESTINATION bin)
 ilink(${CMAKE_INSTALL_PREFIX}/${SWIPL_INSTALL_ARCH_EXE}/swipl
       ${CMAKE_INSTALL_PREFIX}/bin/swipl)
 ilink(${CMAKE_INSTALL_PREFIX}/${SWIPL_INSTALL_ARCH_EXE}/swipl-ld
       ${CMAKE_INSTALL_PREFIX}/bin/swipl-ld)
+ilink(${CMAKE_INSTALL_PREFIX}/${SWIPL_INSTALL_ARCH_EXE}/libswipl.dll
+      ${CMAKE_INSTALL_PREFIX}/bin/libswipl.dll)
 endif()

-endif(WIN32)

