--- swipl-8.4.3/library/prolog_config.pl	2022-08-09 07:31:18 +0000
+++ swipl-8.4.3/library/prolog_config.pl.new	2022-08-09 07:31:11 +0000
@@ -92,7 +92,11 @@
     (   current_prolog_flag(c_libdir, Rel)
     ->  atomic_list_concat([Home, Rel], /, Value)
     ;   current_prolog_flag(windows, true)
-    ->  atomic_list_concat([Home, bin], /, Value)
+    ->  (   current_prolog_flag(msys2, true)
+        ->  current_prolog_flag(arch, Arch),
+            atomic_list_concat([Home, bin, Arch], /, Value)
+        ;   atomic_list_concat([Home, bin], /, Value)
+        )
     ;   apple_bundle_libdir(LibDir)
     ->  Value = LibDir
     ;   current_prolog_flag(arch, Arch)
