--- swipl-8.4.3/cmake/port/Windows.cmake	2022-06-21 13:08:37 +0000
+++ src/swipl-8.4.3/cmake/Port/Windows.cmake	2022-07-20 13:28:41 +0000
@@ -2,6 +2,9 @@
 
 add_compile_options(-D__WINDOWS__)
 add_compile_options(-D_WIN32_WINNT=0x0600)
+string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}-${CMAKE_HOST_SYSTEM_NAME}" MYPLARCH)
+add_compile_options(-DPLARCH="${MYPLARCH}")
+
 if(CMAKE_SIZEOF_VOID_P EQUAL 8)
   add_compile_options(-DWIN64)
   set(WIN64 1)
--- swipl-8.4.3/src/config/win64.h	2022-06-21 13:08:37 +0000
+++ src/swipl-8.4.3/src/config/win64.h	2022-07-20 13:32:23 +0000
@@ -36,7 +36,9 @@
 
 #define JMPBUF_ALIGNMENT 16		/* How to get this value? */
 #define BOOTFILE	"boot64.prc"
+#ifndef PLARCH
 #define PLARCH		"x64-win64"
+#endif
 #define SIZEOF_VOIDP	8
 #ifdef _MSC_VER
 typedef long long ssize_t;
--- swipl-8.4.3/src/config/win32.h	2022-06-21 13:08:37 +0000
+++ src/swipl-8.4.3/src/config/win32.h	2022-07-20 13:33:14 +0000
@@ -35,7 +35,9 @@
 #define WIN32 1
 
 #define BOOTFILE	"boot32.prc"
+#ifndef PLARCH
 #define PLARCH		"i386-win32"
+#endif
 #define SIZEOF_VOIDP	4
 #ifdef _MSC_VER
 typedef long ssize_t;
