--- swipl-9.0.0/library/build/tools.pl    2022-11-24 12:05:19 +0000
+++ swipl-9.0.0/library/build/tools.pl.new        2022-11-26 09:42:23 +0000
@@ -376,11 +376,12 @@
 prolog_library_dir(Dir) :-
     prolog:runtime_config(c_libdir, Dir),
     !.
 prolog_library_dir(Dir) :-
     current_prolog_flag(windows, true),
+    \+ current_prolog_flag(msys2, true),
     current_prolog_flag(home, Home),
     !,
     atomic_list_concat([Home, bin], /, Dir).
 prolog_library_dir(Dir) :-
     current_prolog_flag(home, Home),
     (   current_prolog_flag(c_libdir, Rel)
