--- libdvdread-4.9.9/Makefile.am.orig	2014-03-02 10:58:24.000000000 +0000
+++ libdvdread-4.9.9/Makefile.am	2014-04-26 11:30:28.618513400 +0000
@@ -33,7 +33,7 @@
 
 libdvdread_la_LIBADD = $(CSS_LIBS)
 
-libdvdread_la_LDFLAGS = -version-info $(DVDREAD_LTVERSION) \
+libdvdread_la_LDFLAGS = -no-undefined -version-info $(DVDREAD_LTVERSION) \
 	-export-symbols-regex "(^dvd.*|^nav.*|^ifo.*|^DVD.*|^UDF.*)"
 
 pkgincludedir = $(includedir)/dvdread
