diff --git a/meson/evas_loader_conf.sh b/meson/evas_loader_conf.sh
index bacc3660a1..9491dbf0f2 100755
--- a/meson/evas_loader_conf.sh
+++ b/meson/evas_loader_conf.sh
@@ -2,9 +2,11 @@
 
 ext="${ext##*.}"
 # skip trailing dot if any
+c1="$(cygpath -u $1)"
+
 case "$1" in 
-    *.) original_loader=${1%?};;
-    *) original_loader=$1;;
+    *.) original_loader=${c1%?};;
+    *) original_loader=$c1;;
 esac
 original_name="$(basename "$original_loader")"
 loader_dir="$(dirname "$original_loader")"
@@ -18,5 +20,6 @@ fi
 
 shift
 for x in "$@"; do
-  ln -sf "$original_name" "$DESTDIR""$loader_dir"/"$loader_name"."$x"
+  cx="$(cygpath -u $x)"
+  echo ln -sf "$original_name" "$DESTDIR""$MINGW_PREFIX""$loader_dir"/"$loader_name"."$cx"
 done
diff --git a/meson/meson_modules.sh b/meson/meson_modules.sh
index 328cc9948e..b5ce04e80b 100755
--- a/meson/meson_modules.sh
+++ b/meson/meson_modules.sh
@@ -1,8 +1,9 @@
 #!/bin/sh
 
 for x in "$@" ; do
-  dir="$(dirname $x)"
-  filename="$(basename $x)"
+  cx="$(cygpath -u $x)"
+  dir="$(dirname $cx)"
+  filename="$(basename $cx)"
   ext="${filename##*.}"
-  mv "$DESTDIR""$x" "$DESTDIR""$dir"/module."$ext"
+  echo mv "$DESTDIR""$MINGW_PREFIX""$cx" "$DESTDIR""$dir"/module."$ext"
 done
