--- setuptools-51.1.2/setuptools/command/easy_install.py.orig	2021-01-09 00:44:52.000000000 +0530
+++ setuptools-51.1.2/setuptools/command/easy_install.py	2021-01-12 11:40:02.894168900 +0530
@@ -1314,8 +1314,8 @@
     )
 
     DEFAULT_SCHEME = dict(
-        install_dir='$base/Lib/site-packages',
-        script_dir='$base/Scripts',
+        install_dir='$base/lib/python$py_version_short/site-packages',
+        script_dir='$base/bin',
     )
 
     def _expand(self, *attrs):
