--- setuptools-7.0/setuptools/extension.py.orig	2014-10-20 00:40:16.151400000 +0400
+++ setuptools-7.0/setuptools/extension.py	2014-10-20 00:40:45.978600000 +0400
@@ -6,12 +6,9 @@
 import distutils.extension
 
 from .dist import _get_unpatched
-from . import msvc9_support
 
 _Extension = _get_unpatched(distutils.core.Extension)
 
-msvc9_support.patch_for_specialized_compiler()
-
 def _have_cython():
     """
     Return True if Cython can be imported.
