diff -Naur libgit2-1.8.0.orig/src/libgit2/CMakeLists.txt libgit2-1.8.0/src/libgit2/CMakeLists.txt
--- libgit2-1.8.0.orig/src/libgit2/CMakeLists.txt	2024-03-21 00:50:41.444310600 +0300
+++ libgit2-1.8.0/src/libgit2/CMakeLists.txt	2024-03-21 00:53:19.841827200 +0300
@@ -77,7 +77,7 @@
 	set_target_properties(libgit2package PROPERTIES VERSION ${libgit2_VERSION})
 	set_target_properties(libgit2package PROPERTIES SOVERSION "${libgit2_VERSION_MAJOR}.${libgit2_VERSION_MINOR}")
 	if(LIBGIT2_FILENAME)
-		target_compile_definitions(libgit2package PRIVATE LIBGIT2_FILENAME=\"${LIBGIT2_FILENAME}\")
+		target_compile_definitions(libgit2package PRIVATE LIBGIT2_FILENAME=${LIBGIT2_FILENAME})
 		set_target_properties(libgit2package PROPERTIES OUTPUT_NAME ${LIBGIT2_FILENAME})
 	elseif(DEFINED LIBGIT2_PREFIX)
 		set_target_properties(libgit2package PROPERTIES PREFIX "${LIBGIT2_PREFIX}")
diff -Naur libgit2-1.8.0.orig/src/libgit2/git2.rc libgit2-1.8.0/src/libgit2/git2.rc
--- libgit2-1.8.0.orig/src/libgit2/git2.rc	2024-03-21 00:50:41.569313400 +0300
+++ libgit2-1.8.0/src/libgit2/git2.rc	2024-03-21 00:55:37.504984900 +0300
@@ -2,23 +2,15 @@
 #include "../../include/git2/version.h"
 
 #ifndef LIBGIT2_FILENAME
-# ifdef __GNUC__
-#  define LIBGIT2_FILENAME git2
-# else
-#  define LIBGIT2_FILENAME "git2"
-# endif
+# define LIBGIT2_FILENAME git2
 #endif
 
 #ifndef LIBGIT2_COMMENTS
 # define LIBGIT2_COMMENTS "For more information visit https://libgit2.org/"
 #endif
 
-#ifdef __GNUC__
-# define _STR(x) #x
-# define STR(x) _STR(x)
-#else
-# define STR(x) x
-#endif
+#define _STR(x) #x
+#define STR(x) _STR(x)
 
 #ifdef __GNUC__
 VS_VERSION_INFO		VERSIONINFO
