--- a/cmake/PkgBuildConfig.cmake
+++ b/cmake/PkgBuildConfig.cmake
@@ -26,9 +26,9 @@
 
 	# Write .pc "header"
 	file(WRITE "${PKGCONFIG_FILE}"
-		"prefix=\"${CMAKE_INSTALL_PREFIX}\"\n"
-		"libdir=\"${CMAKE_INSTALL_FULL_LIBDIR}\"\n"
-		"includedir=\"${CMAKE_INSTALL_FULL_INCLUDEDIR}\"\n"
+		"prefix=${CMAKE_INSTALL_PREFIX}\n"
+		"libdir=\${prefix}/${CMAKE_INSTALL_LIBDIR}\n"
+		"includedir=\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}\n"
 		"\n"
 		"Name: ${PKGCONFIG_NAME}\n"
 		"Description: ${PKGCONFIG_DESCRIPTION}\n"
