--- CMakeLists.txt	2016-03-05 15:12:41.000000000 +0100
+++ CMakeLists.txt	2016-03-20 00:30:26.784591500 +0100
@@ -39,8 +39,8 @@
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /arch:SSE2")
     endif()
 else()
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse3")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse3")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse3 -O2")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse3 -O2")
 endif()
 
 option(USE_OpenMP "Use OpenMP" ON)
