--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,6 +148,8 @@
 	elseif(WIRESHARK_TARGET_PLATFORM MATCHES "win32")
 		message(FATAL_ERROR "Deprecated target platform ${WIRESHARK_TARGET_PLATFORM}. See https://gitlab.com/wireshark/wireshark/-/issues/17779 for details.")
 		set(WIRESHARK_TARGET_PROCESSOR_ARCHITECTURE x86)
+	elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64")
+		set(WIRESHARK_TARGET_PROCESSOR_ARCHITECTURE arm64)
 	else()
 		message(FATAL_ERROR "Invalid target platform: ${WIRESHARK_TARGET_PLATFORM}")
 	endif()
--- a/wsutil/CMakeLists.txt
+++ b/wsutil/CMakeLists.txt
@@ -190,7 +190,7 @@
 if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
 	set(COMPILER_CAN_HANDLE_SSE4_2 TRUE)
 	set(SSE4_2_FLAG "")
-else()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i686|x86|x86_64|AMD64")
 	check_c_compiler_flag(-msse4.2 COMPILER_CAN_HANDLE_SSE4_2)
 	if(COMPILER_CAN_HANDLE_SSE4_2)
 		set(SSE4_2_FLAG "-msse4.2")
@@ -200,11 +200,15 @@
 			set(SSE4_2_FLAG "-xarch=sse4_2")
 		endif()
 	endif()
-	if(SSE4_2_FLAG)
-		message(STATUS "SSE4.2 compiler flag: ${SSE4_2_FLAG}")
-	else()
-		message(STATUS "No SSE4.2 compiler flag enabled")
-	endif()
+else()
+	set(COMPILE_CAN_HANDLE_SSE4_2 FALSE)
+	set(SSE4_2_FLAG "")
+endif()
+
+if(SSE4_2_FLAG)
+	message(STATUS "SSE4.2 compiler flag: ${SSE4_2_FLAG}")
+else()
+	message(STATUS "No SSE4.2 compiler flag enabled")
 endif()
 if(COMPILER_CAN_HANDLE_SSE4_2)
 	#
