diff --git a/configure.ac b/configure.ac
index 38be9f8..bbcf689 100644
--- a/configure.ac
+++ b/configure.ac
@@ -182,9 +182,9 @@ function buildSDL() {
 
      dnl do not depend on cygwin.dll under cygwin build
      PACKAGE_CFLAGS="${PACKAGE_CFLAGS} -mms-bitfields"
-     PACKAGE_LIBS="${PACKAGE_LIBS} -lwsock32 -lws2_32 -static-libgcc -static-libstdc++"
-     PACKAGE_LIBS="${PACKAGE_LIBS} -Wl,-Bstatic -Wl,-Map=output.map"
-     PACKAGE_LIBS="${PACKAGE_LIBS} `sdl2-config --static-libs` `pkg-config freetype2 --libs`"
+     PACKAGE_LIBS="${PACKAGE_LIBS} -lwsock32 -lws2_32"
+     PACKAGE_LIBS="${PACKAGE_LIBS} -Wl,-Map=output.map"
+     PACKAGE_LIBS="${PACKAGE_LIBS} `sdl2-config --libs` `pkg-config freetype2 --libs`"
      AC_DEFINE(_Win32, 1, [Windows build])
      ;;
 
@@ -315,7 +315,7 @@ function buildConsole() {
          TARGET="Building Cygwin MinGW console version."
          AC_DEFINE(__MINGW32__, 1, [as above])
          AC_DEFINE(_UnixOS, 1, [Building under Unix like systems.])
-         PACKAGE_LIBS="${PACKAGE_LIBS} -Wl,-Bstatic -mconsole -lwsock32 -lws2_32 -static-libgcc"
+         PACKAGE_LIBS="${PACKAGE_LIBS} -mconsole -lwsock32 -lws2_32"
          BUILD_SUBDIRS="src/common src/platform/console"
       fi
       AC_DEFINE(_Win32, 1, [Windows build])
@@ -339,8 +339,8 @@ function buildWeb() {
      AC_DEFINE(__MINGW32__, 1, [as above])
      AC_DEFINE(_Win32, 1, [Windows build])
      PACKAGE_CFLAGS="-I$prefix/include ${PACKAGE_CFLAGS} -mms-bitfields -D__USE_MINGW_ANSI_STDIO"
-     PACKAGE_LIBS="${PACKAGE_LIBS} -L$prefix/lib -Wl,-Bstatic -mconsole -lmingw32 -lwsock32 "
-     PACKAGE_LIBS="${PACKAGE_LIBS} -lmicrohttpd -lws2_32 -static-libgcc -static-libstdc++"
+     PACKAGE_LIBS="${PACKAGE_LIBS} -L$prefix/lib -mconsole -lmingw32 -lwsock32 "
+     PACKAGE_LIBS="${PACKAGE_LIBS} -lmicrohttpd -lws2_32"
    else
      TARGET="Building Web server version."
      PACKAGE_LIBS="${PACKAGE_LIBS} -lm -ldl -lpthread -lmicrohttpd"
@@ -398,7 +398,7 @@ function buildFLTK() {
    case "${host_os}" in
      *mingw* | cygwin*)
      FLTK_CXXFLAGS="${FLTK_CXXFLAGS} -mms-bitfields"
-     PACKAGE_LIBS="-Wl,-Bstatic ${PACKAGE_LIBS} -lwsock32 -lws2_32 -static-libgcc -static-libstdc++"
+     PACKAGE_LIBS="${PACKAGE_LIBS} -lwsock32 -lws2_32"
      AC_DEFINE(_Win32, 1, [Windows build])
      AC_DEFINE(_USE_MATH_DEFINES, 1, [for M_PI in math.h])
      AC_DEFINE(realpath(F, R), _fullpath(R, F, PATH_MAX), [add missing realpath for system.cpp])
