--- crypto++/config_cpu.h.orig	2023-10-12 20:50:28.004926800 +0300
+++ crypto++/config_cpu.h	2023-10-12 20:51:20.531290800 +0300
@@ -203,7 +203,7 @@
 	///  <tt>_MSC_VER</tt> nor <tt>__BORLANDC__</tt> are defined.
 	#define CRYPTOPP_GNU_STYLE_INLINE_ASSEMBLY ...
 #elif defined(CRYPTOPP_MSC_VERSION) || defined(__BORLANDC__) || \
-	(defined(CRYPTOPP_WIN32_AVAILABLE) && defined(CRYPTOPP_LLVM_CLANG_VERSION))
+	(defined(CRYPTOPP_WIN32_AVAILABLE) && defined(CRYPTOPP_LLVM_CLANG_VERSION) && !defined(__MINGW32__))
 	#define CRYPTOPP_MS_STYLE_INLINE_ASSEMBLY 1
 #else
 	#define CRYPTOPP_GNU_STYLE_INLINE_ASSEMBLY 1
