diff -bur filezilla-3.70.6-orig/locales/Makefile.am filezilla-3.70.6/locales/Makefile.am
--- filezilla-3.70.6-orig/locales/Makefile.am	2026-06-12 03:20:30.353709800 -0600
+++ filezilla-3.70.6/locales/Makefile.am	2026-06-12 03:21:16.388286200 -0600
@@ -54,7 +54,7 @@
 	  exit 1; \
 	fi; \
 	if [ "$${file%.po.new}" != "en" ]; then \
-	  $(MSGMERGE) $@ filezilla.pot -o $@~ || exit 1; \
+	  $(MSGMERGE) $@ $(srcdir)/filezilla.pot -o $@~ || exit 1; \
 	  mv $@~ $@; \
 	fi
 
