diff --git a/libviews/controlflowgraphview.cpp b/libviews/controlflowgraphview.cpp
index 87852f4..dd861bd 100644
--- a/libviews/controlflowgraphview.cpp
+++ b/libviews/controlflowgraphview.cpp
@@ -2978,7 +2978,7 @@ void ControlFlowGraphView::refresh(bool reset)
     connect(_renderProcess, &QProcess::errorOccurred,
             this, &ControlFlowGraphView::dotError);
 
-    static constexpr auto finishedPtr =
+    static const auto finishedPtr =
         static_cast<void (QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished);
     connect(_renderProcess, finishedPtr,
             this, &ControlFlowGraphView::dotExited);
