--- sip-4.19.12/configure.py.orig	2018-07-11 08:04:25.493353200 +0300
+++ sip-4.19.12/configure.py	2018-07-11 08:05:49.180700000 +0300
@@ -278,12 +278,16 @@
     """
     siputils.inform("Creating %s..." % module)
 
+    sip_inst_exe = os.path.join(sip_bin_dir, "sip")
+    if sys.platform == 'win32':
+        sip_inst_exe += '.exe'
+
     content = {
         "sip_config_args":  sys.argv[1:],
         "sip_version":      sip_version,
         "sip_version_str":  sip_version_str,
         "platform":         build_platform,
-        "sip_bin":          os.path.join(sip_bin_dir, "sip"),
+        "sip_bin":          sip_inst_exe,
         "sip_inc_dir":      sip_inc_dir,
         "sip_root_dir":     sip_root_dir,
         "sip_module_dir":   sip_module_dir,
