--- libconfig-1.7.2/lib/wincompat.h.orig	2021-05-15 13:46:48.586000000 +0200
+++ libconfig-1.7.2/lib/wincompat.h	2021-05-15 13:47:09.795634800 +0200
@@ -24,6 +24,7 @@
 #define __wincompat_h
 
 #include <limits.h>
+#include <stdio.h>
 
 #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) \
   || defined(WIN64) || defined(_WIN64) || defined(__WIN64__)
@@ -44,11 +45,11 @@
 
 #define fileno _fileno
 
-#if _MSC_VER < 1800
+#if defined(_MSC_VER) && _MSC_VER < 1800
 #define snprintf  _snprintf
 #endif
 
-#if !defined(__MINGW32__) && _MSC_VER < 1800
+#if !defined(__MINGW32__) && defined(_MSC_VER) && _MSC_VER < 1800
 #define atoll     _atoi64
 #define strtoull  _strtoui64
 #define strtoll   _strtoi64
