--- ois-v1-3/includes/win32/Win32Prereqs.h	2010-09-08 06:30:27.000000000 +0200
+++ patched/includes/win32/Win32Prereqs.h	2012-06-13 16:53:58.000000000 +0200
@@ -28,6 +28,15 @@
 #include <windows.h>
 #include <dinput.h>
 
+#ifdef __MINGW32__
+#ifndef IDirectInput8
+#define IDirectInput8 IDirectInput8A
+#endif
+#ifndef IDirectInputDevice8
+#define IDirectInputDevice8 IDirectInputDevice8A
+#endif
+#endif
+
 #ifdef OIS_WIN32_XINPUT_SUPPORT
 #	include <XInput.h>
 #endif
