--- ois-v1-3/src/win32/Win32ForceFeedback.cpp	2010-04-04 17:56:16.000000000 +0200
+++ patched/src/win32/Win32ForceFeedback.cpp	2012-06-17 20:56:27.362981000 +0200
@@ -195,7 +195,12 @@
 {
 	ConstantEffect *eff = static_cast<ConstantEffect*>(effect->getForceEffect());
 
+	//FIXME: Not implemented in mingw yet
+#ifdef __MINGW32__
+	DWORD           rgdwAxes[2]     = { 0, 4 };
+#else
 	DWORD           rgdwAxes[2]     = { DIJOFS_X, DIJOFS_Y };
+#endif
 	LONG            rglDirection[2] = { 0, 0 };
 	DIENVELOPE      diEnvelope;
 	DICONSTANTFORCE cf;
@@ -219,7 +224,12 @@
 {
 	RampEffect *eff = static_cast<RampEffect*>(effect->getForceEffect());
 
+	//FIXME: Not implemented in mingw yet
+#ifdef __MINGW32__
+	DWORD           rgdwAxes[2]     = { 0, 4 };
+#else
 	DWORD           rgdwAxes[2]     = { DIJOFS_X, DIJOFS_Y };
+#endif
 	LONG            rglDirection[2] = { 0, 0 };
 	DIENVELOPE      diEnvelope;
 	DIRAMPFORCE     rf;
@@ -238,7 +248,12 @@
 {
 	PeriodicEffect *eff = static_cast<PeriodicEffect*>(effect->getForceEffect());
 
+	//FIXME: Not implemented in mingw yet
+#ifdef __MINGW32__
+	DWORD           rgdwAxes[2]     = { 0, 4 };
+#else
 	DWORD           rgdwAxes[2]     = { DIJOFS_X, DIJOFS_Y };
+#endif
 	LONG            rglDirection[2] = { 0, 0 };
 	DIENVELOPE      diEnvelope;
 	DIPERIODIC      pf;
@@ -268,7 +283,12 @@
 {
 	ConditionalEffect *eff = static_cast<ConditionalEffect*>(effect->getForceEffect());
 
+	//FIXME: Not implemented in mingw yet
+#ifdef __MINGW32__
+	DWORD           rgdwAxes[2]     = { 0, 4 };
+#else
 	DWORD           rgdwAxes[2]     = { DIJOFS_X, DIJOFS_Y };
+#endif
 	LONG            rglDirection[2] = { 0, 0 };
 	DIENVELOPE      diEnvelope;
 	DICONDITION     cf;
