diff --git a/dtkwidget.cmake b/dtkwidget.cmake
index 2568898ef..1a062f99a 100644
--- a/dtkwidget.cmake
+++ b/dtkwidget.cmake
@@ -9,7 +9,7 @@ set(CMAKE_AUTORCC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 # Export all symbols on Windows
-if(WIN32 AND NOT MSVC)
+if(WIN32)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--export-all-symbols")
 endif()
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a77a19b71..288ef2060 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -113,7 +113,9 @@ set_target_properties(${LIB_NAME}
 install(
     TARGETS ${LIB_NAME}
     EXPORT ${DtkWidget}Targets
-    DESTINATION ${LIBRARY_INSTALL_DIR}
+    LIBRARY DESTINATION ${LIBRARY_INSTALL_DIR}
+    ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR}
 )
 
