--- a/Dependencies/CMakeLists.txt
+++ b/Dependencies/CMakeLists.txt
@@ -38,4 +38,4 @@
 #)
 #set_target_properties(Skyx PROPERTIES FOLDER Dependencies)
 
-add_subdirectory(TinyXML)
\ No newline at end of file
+#add_subdirectory(TinyXML)
\ No newline at end of file
--- a/Ogitor/CMakeLists.txt
+++ b/Ogitor/CMakeLists.txt
@@ -145,7 +145,7 @@
 
 message(STATUS ${OGRE_LIBRARY})
 
-target_link_libraries(Ogitor ${OGRE_LIBRARIES} OFS OgreTerrainConverter PagedGeometry TinyXML)
+target_link_libraries(Ogitor ${OGRE_LIBRARIES} OFS OgreTerrainConverter PagedGeometry tinyxml)
 
 # specify a precompiled header to use
 use_precompiled_header(Ogitor 
--- a/Plugins/DotSceneSerializer/CMakeLists.txt
+++ b/Plugins/DotSceneSerializer/CMakeLists.txt
@@ -27,7 +27,7 @@
 link_directories(${OGITOR_LIBPATH})
 link_directories(${DEPENDENCIES_LIBPATH})
 
-link_libraries(${OGRE_LIBRARIES} Ogitor TinyXML)
+link_libraries(${OGRE_LIBRARIES} Ogitor tinyxml)
 ogitor_add_library(DotSceneSerializer SHARED ${HEADER_FILES} ${SOURCE_FILES})
 #target_link_libraries(OgreMain Ogitor)
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)=
--- a/Plugins/OgModularZone/CMakeLists.txt
+++ b/Plugins/OgModularZone/CMakeLists.txt
@@ -75,7 +75,7 @@
 	link_directories(${OGITOR_LIBPATH})
 	link_directories(${DEPENDENCIES_LIBPATH})
 
-	link_libraries(${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor TinyXML)
+	link_libraries(${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor tinyxml)
 	ogitor_add_library(OgModularZone SHARED ${SRCS})
 	set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)
 	
--- a/Plugins/OgScreenEffectEditor/CMakeLists.txt
+++ b/Plugins/OgScreenEffectEditor/CMakeLists.txt
@@ -62,7 +62,7 @@
 	link_directories(${OGITOR_LIBPATH})
 	link_directories(${DEPENDENCIES_LIBPATH})
 
-	link_libraries(${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor TinyXML)
+	link_libraries(${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor tinyxml)
 	ogitor_add_library(OgScreenEffectEditor SHARED ${SRCS})
 	set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib)
 	
--- a/qtOgitor/CMakeLists.txt
+++ b/qtOgitor/CMakeLists.txt
@@ -259,9 +259,9 @@
     list(APPEND OGRE_LIBRARIES OgreBitesQt)
 
     if(OGITOR_MESHMAGICK_INTERFACE)
-        target_link_libraries(qtOgitor QtPropertyBrowser ${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor meshmagick_shared_lib ImageConverter GenericTextEditor GenericImageEditor TinyXML)
+        target_link_libraries(qtOgitor QtPropertyBrowser ${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor meshmagick_shared_lib ImageConverter GenericTextEditor GenericImageEditor tinyxml)
     else(OGITOR_MESHMAGICK_INTERFACE)
-        target_link_libraries(qtOgitor QtPropertyBrowser ${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor ImageConverter GenericTextEditor GenericImageEditor TinyXML)
+        target_link_libraries(qtOgitor QtPropertyBrowser ${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES} ${OGRE_LIBRARIES} Ogitor ImageConverter GenericTextEditor GenericImageEditor tinyxml)
     endif(OGITOR_MESHMAGICK_INTERFACE)
 
     if(APPLE)
