--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,7 +140,7 @@
 set(OGITOR_UNITY_FILES_PER_UNIT "50" CACHE STRING "Number of files per compilation unit in Unity build.")
 
 # configure global paths for OgitorsGlobals
-if(UNIX)
+if(UNIX OR MINGW)
     set(DBG_POSTFIX "")
 	set(OGITOR_BIN_PATH "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Ogitor bin path" FORCE)
 	set(OGITOR_RUN_PATH "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Ogitor run path" FORCE)
@@ -161,7 +161,7 @@
 	if (APPLE)
 	set(OGITOR_PLUGIN_ICON_PATH "../Plugins/Icons" CACHE PATH "Ogitor plugins icon path" FORCE)
 	endif(APPLE)
-else(UNIX)
+else(UNIX OR MINGW)
     # We are on Windows
     set(OGITOR_BIN_PATH "." CACHE PATH "Ogitor bin path" FORCE)
     set(OGITOR_RUN_PATH "." CACHE PATH "Ogitor run path" FORCE)
@@ -173,7 +173,7 @@
     set(OGITOR_PLUGIN_PATH "../Plugins" CACHE PATH "Ogitor plugins path" FORCE)
     set(OGITOR_PLUGIN_ICON_PATH "../Plugins/Icons" CACHE PATH "Ogitor plugins icon path" FORCE)
     set(OGITOR_LIBOGREOFSPLUGIN_PATH "." CACHE PATH "libOgreOfsPlugin path" FORCE)
-endif(UNIX)
+endif(UNIX OR MINGW)
 
 if(NOT IS_ABSOLUTE ${OGITOR_LANGUAGE_PATH})
     set(OGITOR_LANGUAGE_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}/bin/${OGITOR_LANGUAGE_PATH}")
@@ -279,7 +279,7 @@
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules/install.cmake)
 
-if(UNIX)
+if(UNIX OR MINGW)
     # apple and linux
     if(OGITOR_DIST)
         FILE(GLOB ogitor_cmake_scripts "${CMAKE_SOURCE_DIR}/CMakeModules/findscripts/*.cmake")
--- a/RunPath/CMakeLists.txt
+++ b/RunPath/CMakeLists.txt
@@ -3,7 +3,7 @@
 ###############################################################
 
 # Set media target directory
-if(WIN32)
+if(MSVC)
 	install(DIRECTORY ${CMAKE_SOURCE_DIR}/RunPath/
 		DESTINATION ${CMAKE_INSTALL_PREFIX}
 		CONFIGURATIONS Release RelWithDebInfo MinSizeRel
@@ -26,9 +26,9 @@
 		DESTINATION bin
 		CONFIGURATIONS Debug
 	)
-endif(WIN32)
+endif(MSVC)
 
-if(UNIX)
+if(UNIX OR MINGW)
 	install(DIRECTORY ${CMAKE_SOURCE_DIR}/RunPath/
 		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/ogitor
 		PATTERN "CMakeLists.txt" EXCLUDE
@@ -51,7 +51,7 @@
 		#${CMAKE_CURRENT_SOURCE_DIR}/bin/SampleAppConfig_dist.xml
 		DESTINATION bin
 	)
-endif(UNIX)
+endif(UNIX OR MINGW)
 
 if (APPLE)
 	file(COPY Plugins/Icons DESTINATION ${CMAKE_BINARY_DIR}/Plugins)
--- a/Ogitor/src/OgitorsRoot.cpp
+++ b/Ogitor/src/OgitorsRoot.cpp
@@ -756,9 +756,9 @@
 
 #if OGRE_PLATFORM == OGRE_PLATFORM_WIN32
 #ifdef DEBUG
-        mSystem->GetFileList("../Plugins/*Script_d.dll", scriptPluginList);
+        mSystem->GetFileList("../lib/ogitor/*Script_d.dll", scriptPluginList);
 #else
-        mSystem->GetFileList("../Plugins/*Script.dll", scriptPluginList);
+        mSystem->GetFileList("../lib/ogitor/*Script.dll", scriptPluginList);
 #endif
 #elif OGRE_PLATFORM == OGRE_PLATFORM_LINUX
 #ifdef DEBUG
@@ -797,9 +797,9 @@
 
 #if OGRE_PLATFORM == OGRE_PLATFORM_WIN32
 #ifdef DEBUG
-        mSystem->GetFileList("../Plugins/*_d.dll", pluginList);
+        mSystem->GetFileList("../lib/ogitor/*_d.dll", pluginList);
 #else
-        mSystem->GetFileList("../Plugins/*.dll", pluginList);
+        mSystem->GetFileList("../lib/ogitor/*.dll", pluginList);
 #endif
 #elif OGRE_PLATFORM == OGRE_PLATFORM_LINUX
 #ifdef DEBUG
--- a/OgreOfsPlugin/CMakeLists.txt
+++ b/OgreOfsPlugin/CMakeLists.txt
@@ -20,7 +20,7 @@
 target_link_libraries(OgreOfsPlugin ${OGRE_LIBRARIES} OFS)
 
 install(TARGETS OgreOfsPlugin
-        LIBRARY DESTINATION ${OGITOR_LIBOGREOFSPLUGIN_INSTALL_PATH}
-        ARCHIVE DESTINATION ${OGITOR_LIBOGREOFSPLUGIN_INSTALL_PATH}
-        RUNTIME DESTINATION ${OGITOR_LIBOGREOFSPLUGIN_INSTALL_PATH})
+        LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 # vim: set sw=2 ts=2 noet:
