From 027aac26cd33573e20f843a5ec3217c061359921 Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Fri, 3 Apr 2015 08:18:50 +0800
Subject: [PATCH] camlheader has the extension .exe in msys2

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 bytecomp/bytelink.ml | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/bytecomp/bytelink.ml b/bytecomp/bytelink.ml
index c0f8f6a..f3216e0 100644
--- a/bytecomp/bytelink.ml
+++ b/bytecomp/bytelink.ml
@@ -304,7 +304,15 @@ let link_bytecode ppf tolink exec_name standalone =
         let inchan = open_in_bin (find_in_path !load_path header) in
         copy_file inchan outchan;
         close_in inchan
-      with Not_found | Sys_error _ -> ()
+      with Not_found | Sys_error _ ->
+			try
+			  let header =
+			    if String.length !Clflags.use_runtime > 0
+			    then "camlheader_ur.exe" else "camlheader" ^ !Clflags.runtime_variant ^ ".exe" in
+			  let inchan = open_in_bin (find_in_path !load_path header) in
+			  copy_file inchan outchan;
+			  close_in inchan
+			with Not_found | Sys_error _ -> ()
     end;
     Bytesections.init_record outchan;
     (* The path to the bytecode interpreter (in use_runtime mode) *)
--
2.3.4
