From 4b50b7e0f754239a40e3070cbf28d27d5bddffb5 Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Sun, 12 Apr 2015 09:39:22 +0800
Subject: [PATCH] Use the standard paths as default in ocamlbuild

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 ocamlbuild/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ocamlbuild/Makefile b/ocamlbuild/Makefile
index b40d0ea..1d65ada 100644
--- a/ocamlbuild/Makefile
+++ b/ocamlbuild/Makefile
@@ -134,8 +134,8 @@ ocamlbuild_pack.cmx: $(PACK_CMX)
 # The config file
 
 ocamlbuild_config.ml: ../config/Makefile
-	(echo 'let bindir = "$(BINDIR)"'; \
-	 echo 'let libdir = "$(LIBDIR)"'; \
+	(echo 'let bindir = Filename.dirname Sys.executable_name'; \
+	 echo 'let libdir = List.fold_left (fun base dir -> Filename.concat base dir) bindir ["..";"lib";"ocaml"]'; \
 	 echo 'let supports_shared_libraries = $(SUPPORTS_SHARED_LIBRARIES)';\
 	 echo 'let a = "$(A)"'; \
 	 echo 'let o = "$(O)"'; \
-- 
2.3.4

