--- miniupnpc-1.9.20141128/CMakeLists.txt.orig	2015-01-16 12:07:41.100000000 +0300
+++ miniupnpc-1.9.20141128/CMakeLists.txt	2015-01-16 12:08:11.644800000 +0300
@@ -92,9 +92,7 @@
 endif (WIN32)
 
 if (WIN32)
-  find_library (WINSOCK2_LIBRARY NAMES ws2_32 WS2_32 Ws2_32)
-  find_library (IPHLPAPI_LIBRARY NAMES iphlpapi)
-  set (LDLIBS ${WINSOCK2_LIBRARY} ${IPHLPAPI_LIBRARY} ${LDLIBS})
+  set (LDLIBS -lws2_32 -liphlpapi ${LDLIBS})
 #elseif (CMAKE_SYSTEM_NAME STREQUAL "Solaris")
 #  find_library (SOCKET_LIBRARY NAMES socket)
 #  find_library (NSL_LIBRARY NAMES nsl)
