--- miniupnpc-2.1.20190408/CMakeLists.txt.orig	2019-06-07 11:56:47.649215800 +0300
+++ miniupnpc-2.1.20190408/CMakeLists.txt	2019-06-07 11:56:51.919221800 +0300
@@ -99,7 +99,7 @@
 
 if (UPNPC_BUILD_STATIC)
   add_library (libminiupnpc-static STATIC ${MINIUPNPC_SOURCES})
-  if (WIN32)
+  if (MSVC)
     set_target_properties (libminiupnpc-static PROPERTIES OUTPUT_NAME "libminiupnpc")
   else()
     set_target_properties (libminiupnpc-static PROPERTIES OUTPUT_NAME "miniupnpc")
