--- miniupnpc-2.0.20170509/CMakeLists.txt.orig	2016-12-16 11:04:32 +0100
+++ miniupnpc-2.0.20170509/CMakeLists.txt	2017-09-25 22:41:23 +0200
@@ -76,6 +76,7 @@
   minisoap.c
   minissdpc.c
   miniwget.c
+  upnpc.c
   upnpcommands.c
   upnpdev.c
   upnpreplyparse.c
@@ -96,9 +97,7 @@
 endif (WIN32)
 
 if (WIN32)
-  find_library (WINSOCK2_LIBRARY NAMES ws2_32 WS2_32 Ws2_32)
-  find_library (IPHLPAPI_LIBRARY NAMES iphlpapi)
-  set (LDLIBS ${WINSOCK2_LIBRARY} ${IPHLPAPI_LIBRARY} ${LDLIBS})
+  set (LDLIBS -lws2_32 -liphlpapi ${LDLIBS})
 #elseif (CMAKE_SYSTEM_NAME STREQUAL "Solaris")
 #  find_library (SOCKET_LIBRARY NAMES socket)
 #  find_library (NSL_LIBRARY NAMES nsl)
