--- a/Qt4Qt5/qscintilla.pro	2018-04-10 14:00:13.000000000 +0300
+++ b/Qt4Qt5/qscintilla.pro	2018-05-21 08:01:02.387948900 +0300
@@ -25,6 +25,7 @@
 TEMPLATE = lib
 CONFIG += qt warn_off thread exceptions hide_symbols
 
+PREFIX = %INSTALL_PREFIX%
 CONFIG(debug, debug|release) {
     mac: {
         TARGET = qscintilla2_qt$${QT_MAJOR_VERSION}_debug
@@ -68,25 +69,31 @@
 # Scintilla namespace rather than pollute the global namespace.
 #DEFINES += SCI_NAMESPACE
 
-target.path = $$[QT_INSTALL_LIBS]
+winpath_target = $$system(cygpath -u $$[QT_INSTALL_LIBS])
+target.path = $$PREFIX$$winpath_target
 INSTALLS += target
 
-header.path = $$[QT_INSTALL_HEADERS]
+winpath_header = $$system(cygpath -u $$[QT_INSTALL_HEADERS])
+header.path = $$PREFIX$$winpath_header
 header.files = Qsci
 INSTALLS += header
 
-trans.path = $$[QT_INSTALL_TRANSLATIONS]
+winpath_trans = $$system(cygpath -u $$[QT_INSTALL_TRANSLATIONS])
+trans.path = $$PREFIX$$winpath_trans
 trans.files = qscintilla_*.qm
 INSTALLS += trans
 
-qsci.path = $$[QT_INSTALL_DATA]
+winpath_qsci = $$system(cygpath -u $$[QT_INSTALL_DATA])
+qsci.path = $$PREFIX$$winpath_qsci
 qsci.files = ../qsci
 INSTALLS += qsci
 
 greaterThan(QT_MAJOR_VERSION, 4) {
-    features.path = $$[QT_HOST_DATA]/mkspecs/features
+    winpath_feature = $$system(cygpath -u $$[QT_HOST_DATA]/mkspecs/features)
+    features.path = $$PREFIX$$winpath_feature
 } else {
-    features.path = $$[QT_INSTALL_DATA]/mkspecs/features
+    winpath_feature = $$system(cygpath -u $$[QT_INSTALL_DATA]/mkspecs/features)
+    features.path = $$PREFIX$$winpath_feature
 }
 CONFIG(staticlib) {
     features.files = $$PWD/features_staticlib/qscintilla2.prf
--- a/designer-Qt4Qt5/designer.pro	2014-07-11 00:03:19.000000000 +0300
+++ b/designer-Qt4Qt5/designer.pro	2014-12-16 16:53:40.823600000 +0300
@@ -22,5 +22,7 @@
 HEADERS = qscintillaplugin.h
 SOURCES = qscintillaplugin.cpp
 
-target.path = $$[QT_INSTALL_PLUGINS]/designer
+PREFIX = %INSTALL_PREFIX%
+winpath = $$system(cygpath -u $$[QT_INSTALL_PLUGINS]/designer)
+target.path = $$PREFIX$$winpath
 INSTALLS += target
