--- a/configure.ac
+++ b/configure.ac
@@ -853,7 +853,7 @@
 AM_CONDITIONAL([NO_HELP], [test "x$has_no_help" = xtrue])
 
 # Additional libs of hicum group
-AC_CHECK_LIB(stdc++, main, XTRALIBS="$XTRALIBS -lstdc++",,)
+dnl AC_CHECK_LIB(stdc++, main, XTRALIBS="$XTRALIBS -lstdc++",,)
 AC_SUBST(XTRALIBS, $XTRALIBS)
 
 LIBS="$LIBS $XTRALIBS"
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -117,6 +117,8 @@
 	conf.h \
 	ngspice.c
 
+nodist_EXTRA_ngspice_SOURCES = dummy.cpp
+
 ngspice_CPPFLAGS = $(AM_CPPFLAGS) -DSIMULATOR
 
 if WINGUI
@@ -131,7 +133,8 @@
 
 if WINGUI
 ngspice_LDADD += \
-	frontend/wdisp/libwindisp.la
+	frontend/wdisp/libwindisp.la \
+	-lgdi32 -lcomdlg32
 endif
 
 ngspice_LDADD += \
@@ -617,7 +620,8 @@
 libngspice_la_CFLAGS = -shared
 
 libngspice_la_LDFLAGS =  -shared
-libngspice_la_LDFLAGS +=  -lstdc++
+#libngspice_la_LDFLAGS +=  -lstdc++
+nodist_EXTRA_libngspice_la_SOURCES = dummy.cpp
 
 if SHWIN
 libngspice_la_LDFLAGS += -Wl,--output-def=ngspice.def  -Wl,--out-implib=ngspice.dll.a
