--- ngspice-34/configure.ac.orig	2021-06-01 14:13:02.841416700 -0700
+++ ngspice-34/configure.ac	2021-06-01 14:13:32.380022600 -0700
@@ -772,7 +772,7 @@
 AM_CONDITIONAL([NO_HELP], [test "x$has_no_help" = xtrue])
 
 # Additional libs of hicum group
-AC_CHECK_LIB(stdc++, main, XTRALIBS="$XTRALIBS -lstdc++",,)
+dnl AC_CHECK_LIB(stdc++, main, XTRALIBS="$XTRALIBS -lstdc++",,)
 AC_SUBST(XTRALIBS, $XTRALIBS)
 
 LIBS="$LIBS $XTRALIBS"
--- ngspice-34/src/Makefile.am.orig	2021-06-01 14:42:50.696457300 -0700
+++ ngspice-34/src/Makefile.am	2021-06-01 14:45:07.914478200 -0700
@@ -111,6 +111,8 @@
 	conf.h \
 	ngspice.c
 
+nodist_EXTRA_ngspice_SOURCES = dummy.cpp
+
 ngspice_CPPFLAGS = $(AM_CPPFLAGS) -DSIMULATOR
 
 if WINGUI
@@ -125,7 +125,8 @@
 
 if WINGUI
 ngspice_LDADD += \
-	frontend/wdisp/libwindisp.la
+	frontend/wdisp/libwindisp.la \
+	-lgdi32 -lcomdlg32
 endif
 
 ngspice_LDADD += \
@@ -573,7 +573,8 @@
 libngspice_la_CFLAGS = -shared
 
 libngspice_la_LDFLAGS =  -shared
-libngspice_la_LDFLAGS +=  -lstdc++
+#libngspice_la_LDFLAGS +=  -lstdc++
+nodist_EXTRA_libngspice_la_SOURCES = dummy.cpp
 
 if SHWIN
 libngspice_la_LDFLAGS += -Wl,--output-def=ngspice.def  -Wl,--out-implib=ngspice.dll.a
