diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c0f64ba82..f1b27b861 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -284,7 +284,7 @@ include(CMakePackageConfigHelpers)
 configure_package_config_file(
   ../cmake/exiv2Config.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/exiv2Config.cmake
-  INSTALL_DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/exiv2"
+  INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2"
 )
 
 install(FILES
@@ -294,7 +294,7 @@ install(FILES
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/exiv2)
 
 install(EXPORT exiv2Export
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/exiv2"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2"
     NAMESPACE Exiv2::
 )
 
@@ -302,5 +302,5 @@ install(
     FILES
         ${CMAKE_CURRENT_BINARY_DIR}/exiv2ConfigVersion.cmake
         ${CMAKE_CURRENT_BINARY_DIR}/exiv2Config.cmake
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/exiv2")
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2")
 
