diff --git a/src/makernote_int.cpp b/src/makernote_int.cpp
index 467ce205..8b6e9adc 100644
--- a/src/makernote_int.cpp
+++ b/src/makernote_int.cpp
@@ -930,6 +930,13 @@ namespace Exiv2 {
                                 uint32_t    size,
                                 ByteOrder   /*byteOrder*/)
     {
+        // FIXME: workaround for overwritten OM System header in Olympus files (https://github.com/Exiv2/exiv2/issues/2542)
+        if (size >= 14 &&   std::string(reinterpret_cast<const char*>(pData), 14)
+                         == std::string("OM SYSTEM\0\0\0II", 14)) {
+          // Require at least the header and an IFD with 1 entry
+          if (size < OMSystemMnHeader::sizeOfSignature() + 18) return 0;
+          return newOMSystemMn2(tag, group, olympus2Id);
+        }
         if (size < 10 ||   std::string(reinterpret_cast<const char*>(pData), 10)
                         != std::string("OLYMPUS\0II", 10)) {
             // Require at least the header and an IFD with 1 entry
