--- exiv2-0.27.1/include/exiv2/error.hpp.orig	2019-05-30 15:17:39.230103600 +0300
+++ exiv2-0.27.1/include/exiv2/error.hpp	2019-05-30 15:17:41.930107400 +0300
@@ -304,7 +304,7 @@
         //! @name Manipulators
         //@{
         //! Assemble the error message from the arguments
-        EXIV2API void setMsg();
+        void setMsg();
         //@}
 
         // DATA
--- exiv2-0.27.1/src/error.cpp.orig	2019-05-30 15:17:25.840084900 +0300
+++ exiv2-0.27.1/src/error.cpp	2019-05-30 15:17:29.910090600 +0300
@@ -225,7 +225,7 @@
     }
 
     template<>
-    void BasicError<char>::setMsg()
+    EXIV2API void BasicError<char>::setMsg()
     {
         std::string msg = _(errMsg(code_));
         std::string::size_type pos;
@@ -262,7 +262,7 @@
 
 #ifdef EXV_UNICODE_PATH
     template<>
-    void BasicError<wchar_t>::setMsg()
+    EXIV2API void BasicError<wchar_t>::setMsg()
     {
         std::string s = _(errMsg(code_));
         std::wstring wmsg(s.begin(), s.end());
