--- gtk+-2.24.10/docs/reference/gtk/gtk2.types.orig	2012-02-06 20:26:50 +0400
+++ gtk+-2.24.10/docs/reference/gtk/gtk2.types	2012-04-27 00:56:27 +0400
@@ -110,19 +110,15 @@
 gtk_option_menu_get_type
 gtk_orientable_get_type
 gtk_page_setup_get_type
-gtk_page_setup_unix_dialog_get_type
 gtk_paned_get_type
 gtk_paper_size_get_type
 gtk_pixmap_get_type
 gtk_plug_get_type
 gtk_preview_get_type
-gtk_printer_get_type
 gtk_print_context_get_type
-gtk_print_job_get_type
 gtk_print_operation_get_type
 gtk_print_operation_preview_get_type
 gtk_print_settings_get_type
-gtk_print_unix_dialog_get_type
 gtk_progress_bar_get_type
 gtk_progress_get_type
 gtk_radio_action_get_type
