--- gtk+-2.24.20/gtk-doc.make.orig	2013-07-28 21:54:36.432029300 +0000
+++ gtk+-2.24.20/gtk-doc.make	2013-07-28 21:55:23.216470200 +0000
@@ -103,7 +103,7 @@
 GTK_DOC_V_INTROSPECT_=$(GTK_DOC_V_INTROSPECT_$(AM_DEFAULT_VERBOSITY))
 GTK_DOC_V_INTROSPECT_0=@echo "  DOC   Introspecting gobjects";
 
-scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB)
+scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB) setup-build.stamp
 	$(GTK_DOC_V_SCAN)_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
