--- gtk+-2.24.20/gtk/Makefile.am.orig	2013-07-28 22:06:06.408145200 +0000
+++ gtk+-2.24.20/gtk/Makefile.am	2013-07-28 23:02:09.899754400 +0000
@@ -3,7 +3,8 @@
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ENV = \
-	CC="$(CC)"
+	CC="$(CC)" \
+	PATH="$(builddir):$(builddir)/.libs:$${PATH}:"
 INTROSPECTION_SCANNER_ARGS = \
 	--add-include-path=../gdk
 INTROSPECTION_COMPILER_ARGS = \
