--- a/macros/neon.m4
+++ b/macros/neon.m4
@@ -363,8 +363,6 @@
 
 AC_CACHE_CHECK([for library containing $1], [ne_cv_libsfor_$1], [
   case $ne_cv_os_uname in
-  MINGW*|MSYS_NT*)
-    ;;
   *)
     case $1 in
     getaddrinfo)
@@ -379,10 +377,6 @@
       ne__prologue="#include <libintl.h>"
       ne__code="bindtextdomain(\"\",\"\");"
       ;;
-    *)
-      ne__prologue=""
-      ne__code="$1();"
-      ;;
     esac
     ;;
   esac
