--- a/Makefile.in
+++ b/Makefile.in
@@ -129,16 +129,16 @@
 
 install-html:
 	$(INSTALL) -d $(DESTDIR)$(docdir)/html
-	for d in doc/html/*.html; do \
+	for d in $(srcdir)/doc/html/*.html; do \
 		$(INSTALL_DATA) $$d $(DESTDIR)$(docdir)/html; \
 	done
 
 install-man:
 	$(INSTALL) -d $(DESTDIR)$(man3dir)
 	$(INSTALL) -d $(DESTDIR)$(man1dir)
-	for m in doc/man/*.3; do \
+	for m in $(srcdir)/doc/man/*.3; do \
 	 $(INSTALL_DATA) $$m $(DESTDIR)$(man3dir); done
-	for m in doc/man/*.1; do \
+	for m in $(srcdir)/doc/man/*.1; do \
 	 $(INSTALL_DATA) $$m $(DESTDIR)$(man1dir); done
 
 install: install-@ALLOW_INSTALL@
