--- jemalloc-3.6.0/configure.ac.orig	2014-12-01 06:22:10.508800000 +0300
+++ jemalloc-3.6.0/configure.ac	2014-12-01 06:23:07.687000000 +0300
@@ -341,8 +341,8 @@
 	  importlib="${so}"
 	  DSO_LDFLAGS="-shared"
 	fi
-	a="lib"
-	libprefix=""
+	a="a"
+	libprefix="lib"
 	SOREV="${so}"
 	PIC_CFLAGS=""
 	;;
--- jemalloc-3.6.0/configure.orig	2014-12-01 06:29:33.067600000 +0300
+++ jemalloc-3.6.0/configure	2014-12-01 06:29:07.811200000 +0300
@@ -4924,8 +4924,8 @@
 	  importlib="${so}"
 	  DSO_LDFLAGS="-shared"
 	fi
-	a="lib"
-	libprefix=""
+	a="a"
+	libprefix="lib"
 	SOREV="${so}"
 	PIC_CFLAGS=""
 	;;
