--- jemalloc-5.0.1/configure.ac.orig	2018-01-13 03:45:41.340437700 -0500
+++ jemalloc-5.0.1/configure.ac	2018-01-13 04:06:27.193388900 -0500
@@ -631,7 +631,7 @@ case "${host}" in
 	  link_whole_archive="1"
 	fi
 	DUMP_SYMS="dumpbin /SYMBOLS"
-	a="lib"
+	a="a"
 	libprefix=""
 	SOREV="${so}"
 	PIC_CFLAGS=""
--- jemalloc-5.0.1/configure.orig	2018-01-13 03:45:41.349461900 -0500
+++ jemalloc-5.0.1/configure	2018-01-13 04:09:33.169636600 -0500
@@ -7311,7 +7299,7 @@ fi
 	  link_whole_archive="1"
 	fi
 	DUMP_SYMS="dumpbin /SYMBOLS"
-	a="lib"
+	a="a"
 	libprefix=""
 	SOREV="${so}"
 	PIC_CFLAGS=""
