--- jemalloc-5.1.0/configure.ac.orig	2018-01-13 03:45:41.340437700 -0500
+++ jemalloc-5.1.0/configure.ac	2018-01-13 04:06:27.193388900 -0500
@@ -669,8 +669,8 @@ case "${host}" in
 	  *)
 	    ;;
 	esac
-	a="lib"
-	libprefix=""
+	a="a"
+	libprefix="lib"
 	SOREV="${so}"
 	PIC_CFLAGS=""
 	;;
--- jemalloc-5.1.0/configure.orig	2018-01-13 03:45:41.349461900 -0500
+++ jemalloc-5.1.0/configure	2018-01-13 04:09:33.169636600 -0500
@@ -7523,8 +7523,8 @@ fi
 	  *)
 	    ;;
 	esac
-	a="lib"
-	libprefix=""
+	a="a"
+	libprefix="lib"
 	SOREV="${so}"
 	PIC_CFLAGS=""
 	;;
