diff --git a/e/temp/tzdata2025b/Makefile b/Makefile
index 2130582c2d..9fe033d1c7 100644
--- a/Makefile
+++ b/Makefile
@@ -671,22 +671,6 @@ INSTALL:	ALL install date.1
 		cp date '$(DESTDIR)$(BINDIR)/.'
 		cp -f date.1 '$(DESTDIR)$(MANDIR)/man1/.'
 
-# Calculate version number from git, if available.
-# Otherwise, use $(VERSION) unless it is "unknown" and there is already
-# a 'version' file, in which case reuse the existing 'version' contents
-# and append "-dirty" if the contents do not already end in "-dirty".
-version:	$(VERSION_DEPS)
-		{ (type git) >/dev/null 2>&1 && \
-		  V=$$(git describe --match '[0-9][0-9][0-9][0-9][a-z]*' \
-				--abbrev=7 --dirty) || \
-		  if test '$(VERSION)' = unknown && read -r V <$@; then \
-		    V=$${V%-dirty}-dirty; \
-		  else \
-		    V='$(VERSION)'; \
-		  fi; } && \
-		printf '%s\n' "$$V" >$@.out
-		mv $@.out $@
-
 # These files can be tailored by setting BACKWARD, PACKRATDATA, PACKRATLIST.
 vanguard.zi main.zi rearguard.zi: $(DSTDATA_ZI_DEPS)
 		$(AWK) \
