diff -urN OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/OpenCASCADE/ReaderWriterOpenCASCADE.cpp.orig OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/OpenCASCADE/ReaderWriterOpenCASCADE.cpp
--- OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/OpenCASCADE/ReaderWriterOpenCASCADE.cpp.orig	2020-01-31 12:03:07.000000000 +0100
+++ OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/OpenCASCADE/ReaderWriterOpenCASCADE.cpp	2022-06-15 18:01:51.140786000 +0200
@@ -30,6 +30,7 @@
 #include <Standard_ErrorHandler.hxx>
 #include <Standard_CString.hxx>
 #include <Standard_Macro.hxx>
+#include <Standard_Version.hxx>
 
 #include <TopoDS.hxx>
 #include <TopoDS_Face.hxx>
@@ -211,7 +212,11 @@
                 {
                     // populate vertex list
                     // Ref: http://www.opencascade.org/org/forum/thread_16694/?forum=3
+#if OCC_VERSION_HEX < 0x070600
                     gp_Pnt pt = (triangulation->Nodes())(j).Transformed(transformation * location.Transformation());
+#else
+                    gp_Pnt pt = triangulation->Node(j).Transformed(transformation * location.Transformation());
+#endif
                     vertexList->push_back(osg::Vec3(pt.X(), pt.Y(), pt.Z()));
 
                     // populate color list
