--- OpenSceneGraph-3.5.1/CMakeLists.txt.orig	2015-11-09 18:52:06.000000000 +0300
+++ OpenSceneGraph-3.5.1/CMakeLists.txt	2015-11-11 14:14:03.911230800 +0300
@@ -668,6 +668,9 @@
     FIND_PACKAGE(OurDCMTK)
     FIND_PACKAGE(FFmpeg)
     FIND_PACKAGE(GStreamer COMPONENTS app pbutils)
+    IF(GStreamer_FOUND)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GStreamer_CFLAGS}")
+    ENDIF()
     FIND_PACKAGE(GLIB COMPONENTS gobject)
     FIND_PACKAGE(DirectShow)
     FIND_PACKAGE(SDL2)
--- OpenSceneGraph-3.5.1/CMakeModules/FindGStreamer.cmake.orig	2015-11-11 14:26:56.738940300 +0300
+++ OpenSceneGraph-3.5.1/CMakeModules/FindGStreamer.cmake	2015-11-11 14:27:12.773336700 +0300
@@ -65,7 +65,7 @@
 #    )
 #endmacro()
 
-if (WIN32)
+if (WIN32 AND NOT MINGW)
     macro(FIND_GSTREAMER_COMPONENT _component_prefix _pkgconfig_name _header _library)
         find_path(${_component_prefix}_INCLUDE_DIRS
             NAMES ${_header}
@@ -83,7 +83,7 @@
     find_package(PkgConfig)    
 
     macro(FIND_GSTREAMER_COMPONENT _component_prefix _pkgconfig_name _header _library)
-        pkg_check_modules(${_component_prefix} QUIET ${_pkgconfig_name})
+        pkg_check_modules(${_component_prefix} ${_pkgconfig_name})
 
 #        find_path(${_component_prefix}_INCLUDE_DIRS
 #            NAMES ${_header}
