--- OpenSceneGraph-OpenSceneGraph-3.6.5/src/osg/DisplaySettings.cpp.orig	2020-01-31 12:03:07.000000000 +0100
+++ OpenSceneGraph-OpenSceneGraph-3.6.5/src/osg/DisplaySettings.cpp	2021-10-29 08:18:06.345095100 +0200
@@ -22,9 +22,6 @@
 #include <algorithm>
 #include <string.h>
 
-using namespace osg;
-using namespace std;
-
 #if defined(WIN32) && !defined(__CYGWIN__)
 #include<windows.h>
 extern "C" { OSG_EXPORT DWORD NvOptimusEnablement=0x00000001; }
@@ -32,6 +29,9 @@
 extern "C" { int NvOptimusEnablement=0x00000001; }
 #endif
 
+using namespace osg;
+using namespace std;
+
 void DisplaySettings::setNvOptimusEnablement(int value)
 {
     NvOptimusEnablement = value;
diff --git a/src/osgPlugins/cfg/ConfigParser.cpp b/src/osgPlugins/cfg/ConfigParser.cpp
index d421710043d..8a6807eafce 100644
--- a/src/osgPlugins/cfg/ConfigParser.cpp
+++ b/src/osgPlugins/cfg/ConfigParser.cpp
@@ -235,7 +235,6 @@
 #include "CameraConfig.h"
 
 
-using namespace std;
 using namespace osgProducer;
 
 static void ConfigParser_error( const char * );
diff --git a/src/osgPlugins/cfg/ConfigParser.y b/src/osgPlugins/cfg/ConfigParser.y
index cf9adf507ca..f6a28f766ca 100644
--- a/src/osgPlugins/cfg/ConfigParser.y
+++ b/src/osgPlugins/cfg/ConfigParser.y
@@ -34,7 +34,6 @@
 #include <Producer/CameraConfig>
 
 
-using namespace std;
 using namespace Producer;
 
 static void ConfigParser_error( const char * );
