diff --git a/CMakeModules/FindFFmpeg.cmake b/CMakeModules/FindFFmpeg.cmake
index d548e7a98..7a53cfe25 100644
--- a/CMakeModules/FindFFmpeg.cmake
+++ b/CMakeModules/FindFFmpeg.cmake
@@ -43,7 +43,7 @@ MACRO(FFMPEG_FIND varname shortname headername)
         /opt/csw/include # Blastwave
         /opt/include
         /usr/freeware/include
-        PATH_SUFFIXES ffmpeg
+        PATH_SUFFIXES ffmpeg ffmpeg4.4
         DOC "Location of FFMPEG Headers"
     )
 
@@ -62,7 +62,7 @@ MACRO(FFMPEG_FIND varname shortname headername)
         /opt/csw/include # Blastwave
         /opt/include
         /usr/freeware/include
-        PATH_SUFFIXES ffmpeg
+        PATH_SUFFIXES ffmpeg ffmpeg4.4
         DOC "Location of FFMPEG Headers"
     )
 
@@ -84,6 +84,7 @@ MACRO(FFMPEG_FIND varname shortname headername)
         /opt/csw/lib
         /opt/lib
         /usr/freeware/lib64
+        PATH_SUFFIXES ffmpeg4.4
         DOC "Location of FFMPEG Libraries"
     )
 
