--- OpenSceneGraph-3.2.1-rc1/CMakeModules/FindCOLLADA.cmake.orig	2013-12-25 00:28:48.219200000 +0400
+++ OpenSceneGraph-3.2.1-rc1/CMakeModules/FindCOLLADA.cmake	2013-12-25 01:05:55.671600000 +0400
@@ -19,7 +19,7 @@
 IF ( COLLADA_ENV_VAR_AVAILABLE )
     SET(COLLADA_DOM_ROOT "$ENV{COLLADA_DIR}/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
 ELSE ( COLLADA_ENV_VAR_AVAILABLE )
-    SET(COLLADA_DOM_ROOT "${ACTUAL_3DPARTY_DIR}/include/1.4/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
+    SET(COLLADA_DOM_ROOT "${ACTUAL_3DPARTY_DIR}/include/collada-dom2.4/1.4/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
 ENDIF( COLLADA_ENV_VAR_AVAILABLE )
 
 
@@ -66,7 +66,7 @@
 )
 
 FIND_LIBRARY(COLLADA_DYNAMIC_LIBRARY 
-    NAMES collada_dom collada14dom Collada14Dom libcollada14dom21 libcollada14dom22
+    NAMES collada_dom collada14dom Collada14Dom libcollada14dom21 libcollada14dom22 libcolladadom141 libcollada-dom141
     PATHS
     ${COLLADA_DOM_ROOT}/build/${COLLADA_BUILDNAME}-1.4
     $ENV{COLLADA_DIR}/build/${COLLADA_BUILDNAME}-1.4
