--- OpenSceneGraph-3.2.1-rc1/src/osgPlugins/freetype/FreeTypeLibrary.cpp.orig	2013-12-25 07:02:01.426200000 +0400
+++ OpenSceneGraph-3.2.1-rc1/src/osgPlugins/freetype/FreeTypeLibrary.cpp	2013-12-25 07:02:17.993400000 +0400
@@ -17,7 +17,7 @@
 
 //#define PRINT_OUT_FONT_DETAILS
 #ifdef PRINT_OUT_FONT_DETAILS
-    #include <freetype/ftsnames.h>
+    #include <freetype2/ftsnames.h>
 #endif
 
 #include FT_TRUETYPE_IDS_H
