--- OpenSceneGraph-OpenSceneGraph-3.6.3/CMakeLists.txt.orig	2019-05-18 13:16:17.352829800 +0300
+++ OpenSceneGraph-OpenSceneGraph-3.6.3/CMakeLists.txt	2019-05-18 13:18:15.007038100 +0300
@@ -802,8 +802,7 @@
     FIND_PACKAGE(FLTK)
     FIND_PACKAGE(FOX)
 
-    SET(wxWidgets_USE_LIBS base core gl net)
-    FIND_PACKAGE(wxWidgets)
+    find_package(wxWidgets COMPONENTS gl core base net)
 
 ENDIF(BUILD_OSG_EXAMPLES AND NOT ANDROID)
 
