--- OpenSceneGraph-OpenSceneGraph-3.5.5/src/osgPlugins/3ds/ReaderWriter3DS.cpp	2017-02-24 13:09:52.785969600 +0100
+++ OpenSceneGraph-OpenSceneGraph-3.5.5/src/osgPlugins/3ds/ReaderWriter3DS.cpp.orig	2017-02-24 13:01:55.130353400 +0100
@@ -619,7 +619,7 @@
         {
             // add our geometry to group (where our children already are)
             // creates geometry under modifier node
-            processMesh(drawStateMap,group,mesh,meshAppliedMatPtr);
+            processMesh(drawStateMap,meshTransform,mesh,meshAppliedMatPtr);
             return group;
         }
         else
