--- OpenSceneGraph-3.2.1-rc3/src/osgPlugins/gif/ReaderWriterGIF.cpp.orig	2013-06-03 18:04:02.000000000 +0400
+++ OpenSceneGraph-3.2.1-rc3/src/osgPlugins/gif/ReaderWriterGIF.cpp	2014-05-30 07:29:09.498200000 +0400
@@ -561,7 +561,7 @@
     *width_ret = giffile->SWidth;
     *height_ret = giffile->SHeight;
     *numComponents_ret = 4;
-    DGifCloseFile(giffile);
+    DGifCloseFile(giffile, NULL);
     return buffer;
 }
 
