--- OpenSceneGraph-OpenSceneGraph-3.5.8/CMakeModules/FindLIBLAS.cmake.orig	2017-11-15 11:24:11.034629000 +0300
+++ OpenSceneGraph-OpenSceneGraph-3.5.8/CMakeModules/FindLIBLAS.cmake	2017-11-15 11:24:37.455282000 +0300
@@ -99,7 +99,7 @@
 
 set(LIBLAS_FOUND "NO")
 if(LIBLAS_LIBRARY AND LIBLAS_INCLUDE_DIR)
-    FIND_PACKAGE(Boost) # used by LIBLAS
+    FIND_PACKAGE(Boost COMPONENTS filesystem system) # used by LIBLAS
     if(Boost_FOUND)
         set(LIBLAS_LIBRARIES ${LIBLAS_LIBRARY} )
         set(LIBLAS_FOUND "YES")
--- OpenSceneGraph-OpenSceneGraph-3.5.8/src/osgPlugins/las/CMakeLists.txt.orig	2017-11-15 11:24:11.034629000 +0300
+++ OpenSceneGraph-OpenSceneGraph-3.5.8/src/osgPlugins/las/CMakeLists.txt	2017-11-15 11:24:37.455282000 +0300
@@ -3,7 +3,7 @@
 
 SET(TARGET_SRC ReaderWriterLAS.cpp )
 
-SET(TARGET_LIBRARIES_VARS LIBLAS_LIBRARY LIBLASC_LIBRARY)
+SET(TARGET_LIBRARIES_VARS LIBLAS_LIBRARY LIBLASC_LIBRARY Boost_LIBRARIES)
 
 #### end var setup  ###
 SETUP_PLUGIN(las)
