--- src/osgQt/CMakelists.txt.orig	2013-12-25 08:53:20.348600000 +0400
+++ src/osgQt/CMakeLists.txt	2013-12-25 08:54:29.987000000 +0400
@@ -13,7 +13,11 @@
 )
 
 IF ( Qt5Widgets_FOUND )
-    QT5_WRAP_CPP( SOURCES_H_MOC ${SOURCES_H} OPTIONS "-f" )
+    if (Qt5Widgets_VERSION VERSION_LESS 5.2.0)
+        QT5_WRAP_CPP( SOURCES_H_MOC ${SOURCES_H} OPTIONS "-f" )
+    else()
+        QT5_WRAP_CPP( SOURCES_H_MOC ${SOURCES_H} )
+    endif()
 ELSE()
     QT4_WRAP_CPP( SOURCES_H_MOC ${SOURCES_H} OPTIONS "-f" )
 ENDIF()
