--- kio-5.50.0/CMakeLists.txt.orig	2018-10-08 09:21:53.221182000 +0300
+++ kio-5.50.0/CMakeLists.txt	2018-10-08 09:22:22.771223300 +0300
@@ -77,12 +77,14 @@
 set(REQUIRED_QT_VERSION 5.8.0)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets DBus Network Concurrent Xml Test)
 
+if (NOT MINGW)
 find_package(GSSAPI)
 set_package_properties(GSSAPI PROPERTIES DESCRIPTION "Allows KIO to make use of certain HTTP authentication services"
                        URL "http://web.mit.edu/kerberos/www"
                        TYPE OPTIONAL
                        PURPOSE "A MIT or HEIMDAL flavor of GSSAPI can be used"
                       )
+endif ()
 
 if (NOT APPLE AND NOT WIN32)
     find_package(X11)
