--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,12 +112,14 @@
     set(_qt6_core_compat_found ${Qt6Core5Compat_FOUND})
 endif()
 
+if (NOT MINGW)
 find_package(GSSAPI)
 set_package_properties(GSSAPI PROPERTIES DESCRIPTION "Allows KIO to make use of certain HTTP authentication services"
                        URL "https://web.mit.edu/kerberos/www/"
                        TYPE OPTIONAL
                        PURPOSE "A MIT or HEIMDAL flavor of GSSAPI can be used"
                       )
+endif ()
 
 if (NOT APPLE AND NOT WIN32 AND NOT ANDROID)
     option(WITH_X11 "Build with X11 integration" ON)
