Index: src/core/kcoredirlister.cpp
===================================================================
--- src/core/kcoredirlister.cpp
+++ src/core/kcoredirlister.cpp
@@ -73,10 +73,10 @@
     connect(kdirnotify, SIGNAL(FilesChanged(QStringList)), SLOT(slotFilesChanged(QStringList)));
     connect(kdirnotify, SIGNAL(FilesRemoved(QStringList)), SLOT(slotFilesRemoved(QStringList)));
 
-    // Probably not needed in KF5 anymore:
-    // The use of KUrl::url() in ~DirItem (sendSignal) crashes if the static for QRegExpEngine got deleted already,
-    // so we need to destroy the KCoreDirListerCache before that.
-    //qAddPostRoutine(kDirListerCache.destroy);
+    qAddPostRoutine([]() {
+        delete kDirListerCache->kdirnotify;
+        kDirListerCache->kdirnotify = nullptr;
+    });
 }
 
 KCoreDirListerCache::~KCoreDirListerCache()
Index: src/core/ksslcertificatemanager.cpp
===================================================================
--- src/core/ksslcertificatemanager.cpp
+++ src/core/ksslcertificatemanager.cpp
@@ -27,6 +27,7 @@
 #include <kconfiggroup.h>
 #include <klocalizedstring.h>
 
+#include <QCoreApplication>
 #include <QDebug>
 #include <qstandardpaths.h>
 #include <QFile>
@@ -204,6 +205,12 @@
 {
     // set Qt's set to empty; this is protected by the lock in K_GLOBAL_STATIC.
     QSslSocket::setDefaultCaCertificates(QList<QSslCertificate>());
+
+    qAddPostRoutine([]() {
+        auto d = KSslCertificateManager::self()->d;
+        delete d->iface;
+        d->iface = nullptr;
+    });
 }
 
 KSslCertificateManagerPrivate::~KSslCertificateManagerPrivate()
