diff -urN kio-5.24.0.orig/CMakeLists.txt kio-5.24.0/CMakeLists.txt
--- kio-5.24.0.orig/CMakeLists.txt	2016-07-03 11:16:00.000000000 +0200
+++ kio-5.24.0/CMakeLists.txt	2016-08-01 10:12:07.045824800 +0200
@@ -71,12 +71,12 @@
 set(REQUIRED_QT_VERSION 5.4.0)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets DBus Network Concurrent Xml Test)
 
-find_package(GSSAPI)
-set_package_properties(GSSAPI PROPERTIES DESCRIPTION "Allows KIO to make use of certain HTTP authentication services"
-                       URL "http://web.mit.edu/kerberos/www"
-                       TYPE OPTIONAL
-                       PURPOSE "A MIT or HEIMDAL flavor of GSSAPI can be used"
-                      )
+# find_package(GSSAPI)
+# set_package_properties(GSSAPI PROPERTIES DESCRIPTION "Allows KIO to make use of certain HTTP authentication services"
+                       # URL "http://web.mit.edu/kerberos/www"
+                       # TYPE OPTIONAL
+                       # PURPOSE "A MIT or HEIMDAL flavor of GSSAPI can be used"
+                      # )
 
 if (NOT APPLE AND NOT WIN32)
     find_package(X11)
