--- libu2f-emu/meson.build.orig	2026-03-08 17:38:34.551606900 +0100
+++ libu2f-emu/meson.build	2026-03-08 17:38:24.158743700 +0100
@@ -39,7 +39,7 @@
 
 if not gtest.found()
   warning('Skip test: gtest not found.')
-else
+elif host_machine.system() != 'windows'
   setup_dir = custom_target('setup_dir',
     output: '.u2f.tmp',
     command: ['@0@/scripts/setup.sh'.format(meson.source_root()), '@OUTPUT@'])
