diff -bur irrlicht-1.8.5-o/source/Irrlicht/Makefile irrlicht-1.8.5/source/Irrlicht/Makefile
--- irrlicht-1.8.5-o/source/Irrlicht/Makefile	2025-04-20 19:14:14.318395200 -0600
+++ irrlicht-1.8.5/source/Irrlicht/Makefile	2025-04-20 19:14:42.227736400 -0600
@@ -54,6 +54,10 @@
 LIBAESGM = aesGladman/aescrypt.o aesGladman/aeskey.o aesGladman/aestab.o aesGladman/fileenc.o aesGladman/hmac.o aesGladman/prng.o aesGladman/pwd2key.o aesGladman/sha1.o aesGladman/sha2.o
 BZIP2OBJ = bzip2/blocksort.o bzip2/huffman.o bzip2/crctable.o bzip2/randtable.o bzip2/bzcompress.o bzip2/decompress.o bzip2/bzlib.o
 
+ifeq ($(MSYSTEM_CARCH),aarch64)
+	LIBPNGOBJ += libpng/arm/arm_init.o libpng/arm/filter_neon_intrinsics.o libpng/arm/palette_neon_intrinsics.o libpng/arm/filter_neon.o
+endif
+
 # Next variable is for additional scene nodes etc. of customized Irrlicht versions
 EXTRAOBJ =
 LINKOBJ = $(IRRMESHOBJ) $(IRROBJ) $(IRRPARTICLEOBJ) $(IRRANIMOBJ) \
