From 5989542ea6b92c8486a901b47caea390e2a72cd6 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 17 Feb 2021 16:38:27 +0100
Subject: [PATCH 3/7] mingw static libs

---
 config.mingw | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.mingw b/config.mingw
index 29a03c5..283fb6f 100644
--- a/config.mingw
+++ b/config.mingw
@@ -9,8 +9,8 @@ OBJ =                  o
 LINK =                 $(CXX) -o 
 LINK_OPTS =            -L.
 CONSOLE_LINK_OPTS =    $(LINK_OPTS)
-LIBRARY_LINK =         $(LD) -o 
-LIBRARY_LINK_OPTS =    $(LINK_OPTS) -r -Bstatic
+LIBRARY_LINK =         $(AR) cr 
+LIBRARY_LINK_OPTS =
 LIB_SUFFIX =                   a
 LIBS_FOR_CONSOLE_APPLICATION = -lws2_32 -lssl -lcrypto
 LIBS_FOR_GUI_APPLICATION = -lws2_32
-- 
2.30.1.windows.1

