From 57341aff4178aed4a3af42747bed68b0dd1d64f9 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 17 Feb 2021 16:59:03 +0100
Subject: [PATCH 6/7] mingw NO_GETIFADDRS

---
 config.mingw | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.mingw b/config.mingw
index 283fb6f..a0601ff 100644
--- a/config.mingw
+++ b/config.mingw
@@ -1,4 +1,4 @@
-COMPILE_OPTS =         $(INCLUDES) -I/usr/local/include -I. -O -DSOCKLEN_T=int -DLOCALE_NOT_USED
+COMPILE_OPTS =         $(INCLUDES) -I/usr/local/include -I. -O -DSOCKLEN_T=int -DLOCALE_NOT_USED -DNO_GETIFADDRS
 C =                    c
 C_COMPILER =           $(CC)
 C_FLAGS =              $(COMPILE_OPTS) -DUSE_OUR_BZERO=1 -D__MINGW32__
-- 
2.30.1.windows.1

