diff -urN scotch_6.0.8.orig/src/libscotch/graph_coarsen.c scotch_6.0.8/src/libscotch/graph_coarsen.c
--- scotch_6.0.8.orig/src/libscotch/graph_coarsen.c	2019-05-18 16:53:25.000000000 -0700
+++ scotch_6.0.8/src/libscotch/graph_coarsen.c	2019-10-08 04:36:55.269120100 -0700
@@ -354,7 +354,7 @@
 
 #ifdef GRAPHCOARSENTHREAD
   thrdnbr                  =
-  coarptr->thrddat.thrdnbr = SCOTCH_PTHREAD_NUMBER; /* Required by graphMatchInit */
+  coarptr->thrddat.thrdnbr = 1/*SCOTCH_PTHREAD_NUMBER*/; /* Required by graphMatchInit */
 #else /* GRAPHCOARSENTHREAD */
 #ifdef COMMON_PTHREAD
   coarptr->thrddat.thrdnbr = 1;
