--- a/meson.build
+++ b/meson.build
@@ -227,9 +227,9 @@
 # Install Configuration
 #===============================================================================
 if get_option('portable') or host_machine.system() == 'windows'
-    pragtical_bindir = '/'
-    pragtical_docdir = '/doc'
-    pragtical_datadir = '/data'
+    pragtical_bindir = 'bin'
+    pragtical_docdir = 'share/doc/pragtical'
+    pragtical_datadir = 'share/pragtical'
     configure_file(
         input: 'resources/windows/pragtical.exe.manifest.in',
         output: 'pragtical.exe.manifest',
