diff -Naur tk8.6.5-orig/unix/installManPage tk8.6.5/unix/installManPage
--- tk8.6.5-orig/unix/installManPage	2016-02-29 22:13:21.000000000 +0300
+++ tk8.6.5/unix/installManPage	2016-03-03 09:21:52.517037200 +0300
@@ -103,7 +103,7 @@
     Target=$Target.$Section$Suffix
     rm -f $Dir/$Target $Dir/$Target.*
     if test -z "$First" ; then
-	First=$Target
+	First=$(echo $Target | sed -e 's/:/-/g')
 	sed -e "/man\.macros/r $SrcDir/man.macros" -e "/man\.macros/d" \
 	    $ManPage > $Dir/$First
 	chmod 444 $Dir/$First
