diff -Naur tk8.6.5-orig/win/Makefile.in tk8.6.5/win/Makefile.in
--- tk8.6.5-orig/win/Makefile.in	2016-03-03 09:21:52.427037200 +0300
+++ tk8.6.5/win/Makefile.in	2016-03-03 09:21:52.896037200 +0300
@@ -590,6 +590,27 @@
 	    done;
 
 install-doc: doc
+	@for i in "$(MAN_INSTALL_DIR)" "$(MAN1_INSTALL_DIR)" "$(MAN3_INSTALL_DIR)" "$(MANN_INSTALL_DIR)" ; \
+	    do \
+	    if [ ! -d "$$i" ] ; then \
+		echo "Making directory $$i"; \
+		mkdir -p "$$i"; \
+		chmod 755 "$$i"; \
+		else true; \
+		fi; \
+	    done;
+	@echo "Installing and cross-linking top-level (.1) docs";
+	@for i in $(ROOT_DIR)/doc/*.1; do \
+	    $(SHELL) $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i "$(MAN1_INSTALL_DIR)"; \
+	done
+	@echo "Installing and cross-linking C API (.3) docs";
+	@for i in $(ROOT_DIR)/doc/*.3; do \
+	    $(SHELL) $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i "$(MAN3_INSTALL_DIR)"; \
+	done
+	@echo "Installing and cross-linking command (.n) docs";
+	@for i in $(ROOT_DIR)/doc/*.n; do \
+	    $(SHELL) $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i "$(MANN_INSTALL_DIR)"; \
+	done
 
 # Optional target to install private headers
 install-private-headers: libraries
