--- tk8.6.11/win/tkWin32Dll.c.orig	2021-01-20 20:00:07.218455000 -0800
+++ tk8.6.11/win/tkWin32Dll.c	2021-01-20 20:02:10.905928500 -0800
@@ -123,7 +123,7 @@
 	 */
 
 #ifdef HAVE_NO_SEH
-#   ifdef __WIN64
+#   ifdef __x86_64__
 	__asm__ __volatile__ (
 
 	    /*
@@ -193,7 +193,7 @@
 	    "%rax", "%rbx", "%rcx", "%rdx", "%rsi", "%rdi", "memory"
 	);
 
-#   else
+#   elif defined(__i386__)
 	__asm__ __volatile__ (
 
 	    /*
@@ -264,6 +264,8 @@
 	    "%eax", "%ebx", "%ecx", "%edx", "%esi", "%edi", "memory"
 	);
 
+#   else
+	TkFinalize(NULL);
 #   endif
 #else /* HAVE_NO_SEH */
 	__try {
