--- GDCM-3.0.0.orig/Source/Common/gdcmCAPICryptographicMessageSyntax.cxx	2019-05-06 07:50:26.947094900 -0400
+++ GDCM-3.0.0/Source/Common/gdcmCAPICryptographicMessageSyntax.cxx	2019-05-06 07:55:00.164340000 -0400
@@ -192,11 +192,14 @@
   unsigned char* cek = NULL;
   HCRYPTMSG hMsg = NULL;
   PCMSG_CMS_RECIPIENT_INFO recipientInfo = NULL;
-  DWORD dwMessageType, cbMessageTypeLen = sizeof(DWORD);
+  DWORD dwMessageType, cbMessageTypeLen;
+  cbMessageTypeLen = sizeof(DWORD);
   PCRYPT_ALGORITHM_IDENTIFIER cekAlg = NULL;
   ALG_ID kekAlg;
-  DWORD kekAlgLen = sizeof(ALG_ID);
-  DWORD nrOfRecipeints, nrOfRecipientsLen = sizeof(DWORD);
+  DWORD kekAlgLen;
+  kekAlgLen = sizeof(ALG_ID);
+  DWORD nrOfRecipeints, nrOfRecipientsLen;
+  nrOfRecipientsLen = sizeof(DWORD);
   unsigned char* bareContent = NULL;
   struct {
     BLOBHEADER header;
