--- x86_64-w64-mingw32/src/src.pro.orig	2014-12-16 07:06:13.178400000 +0300
+++ x86_64-w64-mingw32/src/src.pro	2014-12-16 07:07:03.301200000 +0300
@@ -9,7 +9,12 @@
 CONFIG += opengl warn_on release
 INCLUDEPATH += ../include ../ $$INSTALLDIR/include
 CONFIG -= qt
-LIBS += -L$$INSTALLDIR/lib -lGLEW
+win32 {
+  LIBS += -L$$INSTALLDIR/lib -lglew32
+}
+else {
+  LIBS += -L$$INSTALLDIR/lib -lGLEW
+}
 
 DESTDIR = ../lib
 headers.files = ../include/opencsg.h
--- x86_64-w64-mingw32/example/example.pro.orig	2014-12-16 07:03:05.367000000 +0300
+++ x86_64-w64-mingw32/example/example.pro	2014-12-16 07:05:29.422400000 +0300
@@ -4,7 +4,12 @@
 CONFIG += opengl warn_on release
 CONFIG -= qt
 INCLUDEPATH += ../include
-LIBS += -L../lib -lopencsg -lGLEW
+win32 {
+  LIBS += -L../lib -lopencsg -lglew32
+}
+else {
+  LIBS += -L../lib -lopencsg -lGLEW
+}
 
 INSTALLDIR = /usr/local
 INSTALLS += target
@@ -14,6 +19,9 @@
   INCLUDEPATH += /opt/local/include
   LIBS += -framework GLUT -L/opt/local/lib
 }
+win32 {
+  LIBS += -lfreeglut
+}
 else {
   LIBS += -lGLU -lglut
 }
