Subject: [PATCH 7/7] ChFi3d_Builder: PerformIntersectionAtEnd: Check max number of faces
From: blobfish <blobfish@gmx.com>

Date: Mon, 18 Jan 2021 22:47:56 -0500


---
 src/ChFi3d/ChFi3d_Builder_C1.cxx | 1 +
 1 file changed, 1 insertion(+)

--- a/src/ChFi3d/ChFi3d_Builder_C1.cxx
+++ b/src/ChFi3d/ChFi3d_Builder_C1.cxx
@@ -2018,6 +2018,7 @@
       if ((possible1 && possible2) || (!possible1 && !possible2) || (nbarete > 4)) {
 	while (!trouve) {
 	  nb++;
+    if (nb>=nn) throw Standard_Failure("IntersectionAtEnd : the max number of faces reached");
 	  if (nb!=1) F3=Face[nb-2];
 	  Face[nb-1]=F3;
 	  if (CV1.Arc().IsSame(edgelibre1))
