--- src/CMakeLists.txt.orig	2013-09-03 18:43:03.000000000 +0300
+++ src/CMakeLists.txt	2014-08-12 23:27:43.647686700 +0300
@@ -95,6 +95,9 @@
     VERSION ${LASTFM_VERSION_STRING}
     SOVERSION ${LASTFM_SOVERSION}
     COMPILE_DEFINITIONS LASTFM_LIB
+    OUTPUT_NAME lastfm5
+    RUNTIME_OUTPUT_NAME lastfm5-${LASTFM_SOVERSION}
+    ARCHIVE_OUTPUT_NAME lastfm5
 )
 
 install(TARGETS lastfm

--- src/fingerprint/CMakeLists.txt.orig	2013-09-03 18:43:03.000000000 +0300
+++ src/fingerprint/CMakeLists.txt	2014-08-12 23:34:25.204654400 +0300
@@ -38,6 +38,9 @@
     COMPILE_DEFINITIONS LASTFM_FINGERPRINT_LIB
     VERSION ${LASTFM_VERSION_STRING}
     SOVERSION ${LASTFM_SOVERSION}
+    OUTPUT_NAME lastfm_fingerprint5
+    RUNTIME_OUTPUT_NAME lastfm_fingerprint5-${LASTFM_SOVERSION}
+    ARCHIVE_OUTPUT_NAME lastfm_fingerprint5
 )
 
 qt5_use_modules(lastfm_fingerprint Network Sql Xml) 
