--- src/CMakeLists.txt	2020-04-11 20:55:03.963875300 +0800
+++ src/CMakeLists.txt-new	2020-04-11 20:57:08.889894800 +0800
@@ -76,7 +76,10 @@
   if(XCODE_VERSION)
     set_target_properties(rime PROPERTIES INSTALL_NAME_DIR "@rpath")
   endif()
-  install(TARGETS rime DESTINATION ${LIB_INSTALL_DIR})
+  install(TARGETS rime
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
   if(BUILD_SEPARATE_LIBS)
     add_library(rime-dict ${rime_dict_module_src})
@@ -139,5 +142,8 @@
   add_library(rime-static STATIC ${rime_src})
   target_link_libraries(rime-static ${rime_deps})
   set_target_properties(rime-static PROPERTIES OUTPUT_NAME "rime" PREFIX "lib")
-  install(TARGETS rime-static DESTINATION ${LIB_INSTALL_DIR})
+  install(TARGETS rime-static
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
