--- a/src/xslt.cpp
+++ b/src/xslt.cpp
@@ -20,6 +20,7 @@
 #include <libxslt/xsltutils.h>
 
 #include <QByteArray>
+#include <QCoreApplication>
 #include <QDir>
 #include <QFile>
 #include <QStandardPaths>
@@ -145,7 +146,9 @@
         defaultEntityLoader = xmlGetExternalEntityLoader();
         xmlSetExternalEntityLoader(xsltprocExternalEntityLoader);
 
-        replaceURLList[QLatin1String("http://www.oasis-open.org/docbook/xml/4.5")] = QString("file:///%1").arg(DOCBOOK_XML_CURRDTD);
+        QDir dir(qApp->applicationDirPath());
+        dir.cd("../share/xml/docbook/xml-dtd-4.5");
+        replaceURLList[QLatin1String("http://www.oasis-open.org/docbook/xml/4.5")] = QString("file:///%1").arg(dir.absolutePath());
     }
 #endif
 
@@ -437,7 +440,9 @@
     QStringList result = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, fileName, option);
 
     // fallback to stuff installed with KDocTools
-    const QFileInfo fileInInstallDataDir(QStringLiteral(KDOCTOOLS_INSTALL_DATADIR_KF) + QStringLiteral("/kdoctools/") + file);
+    QDir dir(qApp->applicationDirPath());
+    dir.cd("../share/kf5");
+    const QFileInfo fileInInstallDataDir(dir.absolutePath() + QStringLiteral("/kdoctools/") + file);
     if (fileInInstallDataDir.exists()) {
         if ((option == QStandardPaths::LocateFile) && fileInInstallDataDir.isFile()) {
             result.append(fileInInstallDataDir.absoluteFilePath());
