--- src/kdoctools-5.26.0/cmake/FindDocBookXML4.cmake	2016-09-06 14:02:41.000000000 +0200
+++ src/kdoctools-5.26.0/cmake/FindDocBookXML4.cmake	2017-07-01 17:14:44.649925000 +0200
@@ -55,6 +55,7 @@
         ${CMAKE_INSTALL_DATAROOTDIR}/xml/docbook/xml-dtd-${version}
         ${CMAKE_INSTALL_DATAROOTDIR}/sgml/docbook/xml-dtd-${version}
         ${CMAKE_INSTALL_DATAROOTDIR}/xml/docbook/${version}
+	share/xml/docbook/xml-dtd-${version}
     )
 
     find_path (searched_dir docbookx.dtd
