diff -rupN a/config-kdoctools.h.cmake b/config-kdoctools.h.cmake
--- a/config-kdoctools.h.cmake	2017-09-13 15:09:36.293815700 -0700
+++ b/config-kdoctools.h.cmake	2017-09-13 15:11:00.881032300 -0700
@@ -2,4 +2,4 @@
 
 #define DOCBOOK_XML_CURRDTD "@DocBookXML4_DTD_DIR@"
 
-#define KDOCTOOLS_INSTALL_DATADIR_KF5 "${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_DATADIR_KF5}"
+#define KDOCTOOLS_INSTALL_DATADIR_KF5 "${MSYS_REAL_INSTALL_DIR}/${KDE_INSTALL_DATADIR_KF5}"
