--- kdoctools-5.24.0.orig/src/xslt.cpp	2016-07-03 11:11:07 +0200
+++ kdoctools-5.24.0/src/xslt.cpp	2016-08-03 23:25:52 +0200
@@ -422,6 +422,25 @@
             result.append(fileInInstallDataDir.absoluteFilePath());
         }
     }
+    // MSYS2 HACK:
+    // At some point of porting a change to KDE Frameworks caused absoluteFilePath change to "D:/mingw64/share/kf5/kdoctools/customization"
+    // TODO: find the reason and use proper fix
+    else {
+        {
+            const QFileInfo fileInInstallDataDirMsys(QStringLiteral("/msys32") + QStringLiteral(KDOCTOOLS_INSTALL_DATADIR_KF5) + QStringLiteral("/kdoctools/") + file);
+            if ((option == QStandardPaths::LocateFile) && fileInInstallDataDirMsys.isFile())
+                result.append(fileInInstallDataDirMsys.absoluteFilePath());
+            else if ((option == QStandardPaths::LocateDirectory) && fileInInstallDataDirMsys.isDir())
+                result.append(fileInInstallDataDirMsys.absoluteFilePath());
+        }
+        {
+            const QFileInfo fileInInstallDataDirMsys(QStringLiteral("/msys64") + QStringLiteral(KDOCTOOLS_INSTALL_DATADIR_KF5) + QStringLiteral("/kdoctools/") + file);
+            if ((option == QStandardPaths::LocateFile) && fileInInstallDataDirMsys.isFile())
+                result.append(fileInInstallDataDirMsys.absoluteFilePath());
+            else if ((option == QStandardPaths::LocateDirectory) && fileInInstallDataDirMsys.isDir())
+                result.append(fileInInstallDataDirMsys.absoluteFilePath());
+        }
+    }
 
     if (result.isEmpty()) {
         qDebug() << "Could not locate file" << fileName << "in" << QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
