--- a/CMakeLists.txt.new	2019-02-23 14:57:10.916277725 +0700
+++ b/CMakeLists.txt	2019-02-23 14:58:39.583668715 +0700
@@ -81,7 +81,7 @@
 
 
 # install the conversion tool, .so, .a, and all the header files
-INSTALL ( PROGRAMS ${CMAKE_BINARY_DIR}/yuvconvert			DESTINATION bin )
+INSTALL ( PROGRAMS ${CMAKE_BINARY_DIR}/yuvconvert.exe			DESTINATION bin )
 INSTALL ( TARGETS ${ly_lib_static}						DESTINATION lib )
 INSTALL ( TARGETS ${ly_lib_shared} LIBRARY				DESTINATION lib RUNTIME DESTINATION bin )
 INSTALL ( DIRECTORY ${PROJECT_SOURCE_DIR}/include/		DESTINATION include )
