Index: shout.m4
===================================================================
--- a/m4/shout.m4	(revision 18912)
+++ b/m4/shout.m4	(working copy)
@@ -27,8 +27,8 @@ export PKG_CONFIG_PATH
 AC_PATH_PROG([PKGCONFIG], [pkg-config], [no])
 if test "$PKGCONFIG" != "no" && `$PKGCONFIG --exists shout`
 then
-  SHOUT_CFLAGS=`$PKGCONFIG --variable=cflags_only shout`
-  SHOUT_CPPFLAGS=`$PKGCONFIG --variable=cppflags shout`
+  SHOUT_CFLAGS=`$PKGCONFIG --cflags-only-other shout`
+  SHOUT_CPPFLAGS=`$PKGCONFIG --cflags-only-I shout`
   SHOUT_LIBS=`$PKGCONFIG --libs shout`
   xt_have_shout="maybe"
 else
