--- cegui-0.8.7/CMakeLists.txt.orig	2018-07-20 22:36:58.046612500 +0300
+++ cegui-0.8.7/CMakeLists.txt	2018-07-20 22:37:21.350945400 +0300
@@ -71,8 +71,8 @@
 endif()
 
 if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++03")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu++11")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
 elseif (NOT CMAKE_VERSION VERSION_LESS "3.1")
     set (CMAKE_C_STANDARD_REQUIRED TRUE)
     set (CMAKE_C_STANDARD "99")
