diff -Naur cegui-0.8.7.a/cegui/src/ScriptModules/Lua/Functor.cpp cegui-0.8.7.b/cegui/src/ScriptModules/Lua/Functor.cpp
--- cegui-0.8.7.a/cegui/src/ScriptModules/Lua/Functor.cpp	2016-04-23 20:19:40.000000000 +0200
+++ cegui-0.8.7.b/cegui/src/ScriptModules/Lua/Functor.cpp	2020-08-15 17:40:43.764150400 +0200
@@ -35,9 +35,9 @@
 
 // include Lua libs and tolua++
 extern "C" {
-#include "lua.h"
-#include "lualib.h"
-#include "lauxlib.h"
+#include "lua5.1/lua.h"
+#include "lua5.1/lualib.h"
+#include "lua5.1/lauxlib.h"
 }
 
 #include "tolua++.h"
diff -Naur cegui-0.8.7.a/cegui/src/ScriptModules/Lua/ScriptModule.cpp cegui-0.8.7.b/cegui/src/ScriptModules/Lua/ScriptModule.cpp
--- cegui-0.8.7.a/cegui/src/ScriptModules/Lua/ScriptModule.cpp	2016-04-23 20:19:40.000000000 +0200
+++ cegui-0.8.7.b/cegui/src/ScriptModules/Lua/ScriptModule.cpp	2020-08-15 17:40:43.764150400 +0200
@@ -37,9 +37,9 @@
 
 // include Lua libs and tolua++
 extern "C" {
-#include "lua.h"
-#include "lualib.h"
-#include "lauxlib.h"
+#include "lua5.1/lua.h"
+#include "lua5.1/lualib.h"
+#include "lua5.1/lauxlib.h"
 }
 
 #include "tolua++.h"
diff -Naur cegui-0.8.7.a/cegui/src/ScriptModules/Lua/support/tolua++bin/tolua.c cegui-0.8.7.b/cegui/src/ScriptModules/Lua/support/tolua++bin/tolua.c
--- cegui-0.8.7.a/cegui/src/ScriptModules/Lua/support/tolua++bin/tolua.c	2016-04-23 20:18:29.000000000 +0200
+++ cegui-0.8.7.b/cegui/src/ScriptModules/Lua/support/tolua++bin/tolua.c	2020-08-15 17:40:43.748524800 +0200
@@ -12,11 +12,11 @@
 ** enhancements, or modifications.
 */
 
-#include "tolua++.h"
+#include "lua5.1/lua.h"
+#include "lua5.1/lualib.h"
+#include "lua5.1/lauxlib.h"
 
-#include "lua.h"
-#include "lualib.h"
-#include "lauxlib.h"
+#include "tolua++.h"
 
 #include <stdio.h>
 #include <stdlib.h>
