--- cegui-0.8.3/cmake/FindGLEW.cmake.orig	2014-04-21 10:27:03.881600000 +0400
+++ cegui-0.8.3/cmake/FindGLEW.cmake	2014-04-21 10:27:56.016800000 +0400
@@ -4,12 +4,12 @@
 include(FindPackageHandleStandardArgs)
 
 find_path(GLEW_H_PATH NAMES GL/glew.h)
-find_library(GLEW_LIB NAMES GLEW libGLEW glew libglew PATH_SUFFIXES dynamic)
+find_library(GLEW_LIB NAMES GLEW libGLEW libglew32 glew32 PATH_SUFFIXES dynamic)
 find_library(GLEW_LIB_DBG NAMES GLEW_d libGLEW_d glew_d libglew_d PATH_SUFFIXES dynamic)
 mark_as_advanced(GLEW_H_PATH GLEW_LIB GLEW_LIB_DBG)
 
 if (WIN32 OR APPLE)
-    find_library(GLEW_LIB_STATIC NAMES GLEW libGLEW glew libglew PATH_SUFFIXES static)
+    find_library(GLEW_LIB_STATIC NAMES GLEW libGLEW libglew32 glew32 PATH_SUFFIXES static)
     find_library(GLEW_LIB_STATIC_DBG NAMES GLEW_d libGLEW_d glew_d libglew_d PATH_SUFFIXES static)
     mark_as_advanced(GLEW_LIB_STATIC GLEW_LIB_STATIC_DBG)
 endif()
