--- cegui-0.8.3/samples_framework/CMakeLists.txt.orig	2014-04-21 11:55:10.372200000 +0400
+++ cegui-0.8.3/samples_framework/CMakeLists.txt	2014-04-21 11:56:23.489400000 +0400
@@ -129,7 +129,7 @@
     endif()
 
     # This works around an issue in the prebuilt deps and can be removed once those are fixed :)
-    if (WIN32)
+    if (MSVC)
         set_target_properties(${CEGUI_TARGET_NAME}_Static PROPERTIES LINK_FLAGS_DEBUG /NODEFAULTLIB:freeglut.lib)
     endif()
 