--- cegui-0.8.3/cmake/FindGLFW.cmake.orig	2014-04-21 11:23:06.043000000 +0400
+++ cegui-0.8.3/cmake/FindGLFW.cmake	2014-04-21 11:23:53.077000000 +0400
@@ -3,9 +3,9 @@
 ################################################################################
 include(FindPackageHandleStandardArgs)
 
-find_path(GLFW_H_PATH NAMES GL/glfw.h glfw.h)
-find_library(GLFW_LIB NAMES glfw PATH_SUFFIXES dynamic)
-find_library(GLFW_LIB_DBG NAMES glfw_d PATH_SUFFIXES dynamic)
+find_path(GLFW_H_PATH NAMES GLFW/glfw3.h glfw3.h)
+find_library(GLFW_LIB NAMES glfw3 PATH_SUFFIXES dynamic)
+find_library(GLFW_LIB_DBG NAMES glfw3_d PATH_SUFFIXES dynamic)
 mark_as_advanced(GLFW_H_PATH GLFW_LIB GLFW_LIB_DBG)
 
 if (WIN32)
@@ -14,8 +14,8 @@
 endif()
 
 if (WIN32 OR APPLE)
-    find_library(GLFW_LIB_STATIC NAMES glfw PATH_SUFFIXES static)
-    find_library(GLFW_LIB_STATIC_DBG NAMES glfw_d PATH_SUFFIXES static)
+    find_library(GLFW_LIB_STATIC NAMES glfw3 PATH_SUFFIXES static)
+    find_library(GLFW_LIB_STATIC_DBG NAMES glfw3_d PATH_SUFFIXES static)
     mark_as_advanced(GLFW_LIB_STATIC GLFW_LIB_STATIC_DBG)
 endif()
 
