diff -r 319182022f9f cmake/FindFreetype.cmake
--- a/cmake/FindFreetype.cmake	Sat Dec 07 09:00:48 2013 +0000
+++ b/cmake/FindFreetype.cmake	Tue Dec 17 05:02:30 2013 +0100
@@ -3,8 +3,8 @@
 ################################################################################
 include(FindPackageHandleStandardArgs)
 
-find_path(FREETYPE_H_PATH_ft2build NAMES ft2build.h)
-find_path(FREETYPE_H_PATH_ftconfig NAMES freetype/config/ftconfig.h PATH_SUFFIXES freetype2)
+find_path(FREETYPE_H_PATH_ft2build NAMES ft2build.h PATH_SUFFIXES freetype2)
+find_path(FREETYPE_H_PATH_ftconfig NAMES config/ftconfig.h PATH_SUFFIXES freetype2)
 find_library(FREETYPE_LIB NAMES freetype2 freetype libfreetype PATH_SUFFIXES dynamic)
 find_library(FREETYPE_LIB_DBG NAMES freetype_d libfreetype_d PATH_SUFFIXES dynamic)
 mark_as_advanced(FREETYPE_H_PATH_ft2build FREETYPE_H_PATH_ftconfig FREETYPE_LIB FREETYPE_LIB_DBG)
