diff -Naur a/cegui/src/CMakeLists.txt b/cegui/src/CMakeLists.txt
--- a/cegui/src/CMakeLists.txt	2013-12-23 14:17:45.132800000 +0400
+++ b/cegui/src/CMakeLists.txt	2013-12-07 13:37:08.000000000 +0400
@@ -85,7 +85,7 @@
 endif ()
 
 if (WIN32)
-    cegui_target_link_libraries(${CEGUI_TARGET_NAME} winmm debug DbgHelp)
+    cegui_target_link_libraries(${CEGUI_TARGET_NAME} winmm shlwapi debug DbgHelp)
 elseif (UNIX AND NOT APPLE)
     # This is intentionally not using 'cegui_target_link_libraries' 
     target_link_libraries(${CEGUI_TARGET_NAME} ${CMAKE_DL_LIBS})
diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-12-23 14:07:01.199000000 +0400
+++ b/CMakeLists.txt	2013-12-07 13:37:07.000000000 +0400
@@ -71,9 +71,7 @@
 find_package(Freetype)
 find_package(Minizip)
 find_package(Fribidi)
-if (NOT WIN32)
-    find_package(Iconv REQUIRED)
-endif()
+find_package(Iconv REQUIRED)
 
 find_package(OpenGL)
 find_package(GLEW)
