--- libharu-RELEASE_2_3_0/CMakeLists.txt.orig	2013-10-30 14:11:59.000000000 +0100
+++ libharu-RELEASE_2_3_0/CMakeLists.txt	2015-10-11 15:24:05.099693992 +0200
@@ -209,15 +209,6 @@
 # install header files
 install(FILES ${haru_HDRS} DESTINATION include)
 
-# install various files
-install(FILES README CHANGES INSTALL DESTINATION .)
-if(NOT DEVPAK)
-  install(DIRECTORY if DESTINATION .)
-endif(NOT DEVPAK)
-if(DEVPAK)
-  install(FILES ${CMAKE_BINARY_DIR}/libharu.DevPackage DESTINATION .)
-endif(DEVPAK)
-
 # =======================================================================
 # print out some information
 # =======================================================================
