--- a/wx-config.in
+++ b/wx-config.in
@@ -401,6 +401,9 @@
 
 # Determine the base directories we require.
 prefix=${input_option_prefix-${this_prefix:-@prefix@}}
+if [ "x${MSYSTEM}" = "xMINGW32" ] || [ "x${MSYSTEM}" = "xMINGW64" ] || [ "x${MSYSTEM}" = "xUCRT64" ] || [ "x${MSYSTEM}" = "xCLANG64" ] || [ "x${MSYSTEM}" = "xCLANG32" ] || [ "x${MSYSTEM}" = "xCLANGARM64" ]; then
+    prefix=$(cygpath -m ${prefix})
+fi
 exec_prefix=${input_option_exec_prefix-${input_option_prefix-${this_exec_prefix:-@exec_prefix@}}}
 wxconfdir="@libdir@/wx/config"
 
