diff -uNr wxWidgets-3.0.2.orig/src/stc/scintilla/src/Editor.cxx wxWidgets-3.0.2.mod/src/stc/scintilla/src/Editor.cxx
--- wxWidgets-3.0.2.orig/src/stc/scintilla/src/Editor.cxx	2016-07-22 11:35:57.000000000 +0500
+++ wxWidgets-3.0.2.mod/src/stc/scintilla/src/Editor.cxx	2016-07-22 11:30:51.000000000 +0500
@@ -11,6 +11,7 @@
 #include <ctype.h>
 #include <assert.h>
 
+#include <cmath>
 #include <string>
 #include <vector>
 #include <map>
@@ -5841,9 +5842,9 @@
 }
 
 static bool Close(Point pt1, Point pt2) {
-	if (abs(pt1.x - pt2.x) > 3)
+	if (std::abs(pt1.x - pt2.x) > 3)
 		return false;
-	if (abs(pt1.y - pt2.y) > 3)
+	if (std::abs(pt1.y - pt2.y) > 3)
 		return false;
 	return true;
 }
