--- a/Makefile.system
+++ b/Makefile.system
@@ -420,8 +420,14 @@
 # Ensure the correct stack alignment on Win32
 # http://permalink.gmane.org/gmane.comp.lib.openblas.general/97
 ifeq ($(ARCH), x86)
+ifeq ($(C_COMPILER), GCC)
 CCOMMON_OPT += -mincoming-stack-boundary=2
+else ifeq ($(C_COMPILER), CLANG)
+CCOMMON_OPT += -mstack-alignment=2
+endif
+ifeq ($(F_COMPILER), GFORTRAN)
 FCOMMON_OPT += -mincoming-stack-boundary=2
+endif
 endif
 
 endif
