diff --git a/src/gnuwin32/MkRules.rules b/src/gnuwin32/MkRules.rules
index 7252d1d449..aa8b1b4aad 100644
--- a/src/gnuwin32/MkRules.rules
+++ b/src/gnuwin32/MkRules.rules
@@ -161,7 +161,7 @@ else
   FC=$(BINPREF)flang $(M_ARCH)
   ## Fortran_main is needed/available only in some builds of llvm/flang-new
   LIBFM = $(or $(and $(wildcard $(EXT_LIBS)/lib/libFortran_main.a),-lFortran_main),)
-  FLIBS=-lFortranRuntime -lFortranDecimal $(LIBFM) -lc++ -lm
+  FLIBS=-lflang_rt.runtime -lFortranDecimal $(LIBFM) -lc++ -lm
 endif
 LINKER=$(MAIN_LD)
 MAIN_LD=$(CC)
diff --git a/src/include/Makefile.win b/src/include/Makefile.win
index 38816ca294..4ba799f35c 100644
--- a/src/include/Makefile.win
+++ b/src/include/Makefile.win
@@ -39,8 +39,8 @@ fixh: $(GW32_HEADERS) $(OBJ_HEADERS)
 	@$(ECHO) done > fixh
 
 config.h: ../gnuwin32/fixed/h/config.h ../../VERSION
-	@$(SED) -e 's/@VERSION@/$(VER)/' \
-                -e 's/@CC_VER@/$(CC_VER)/' -e 's/@FC_VER@/$(FC_VER)/' $< > $@
+	@$(SED) -e 's|@VERSION@|$(VER)|' \
+                -e 's|@CC_VER@|$(CC_VER)|' -e 's|@FC_VER@|$(FC_VER)|' $< > $@
 
 Rconfig.h: ../gnuwin32/fixed/h/Rconfig.h
 	@cp $< $@
