--- srt-1.3.1/CMakeLists.txt.orig	2018-09-25 10:49:17.307164900 +0300
+++ srt-1.3.1/CMakeLists.txt	2018-09-25 10:57:40.362248600 +0300
@@ -461,8 +461,12 @@
     set_target_properties (${TARGET_srt}_shared PROPERTIES VERSION ${SRT_VERSION} SOVERSION ${SRT_VERSION_MAJOR})
 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_shared)
 	target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES})
-	if (MICROSOFT)
-		target_link_libraries(${TARGET_srt}_shared PRIVATE ws2_32.lib)
+	if (WIN32)
+		if (MSVC)
+			target_link_libraries(${TARGET_srt}_shared PRIVATE Ws2_32.lib)
+		else ()
+			target_link_libraries(${TARGET_srt}_shared PRIVATE ws2_32)
+		endif()
 	endif()
 endif()
 
@@ -506,8 +510,12 @@
 
 set (SRT_LIBS_PRIVATE ${SSL_LIBRARIES})
 
-if (MICROSOFT)
-	set (SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ws2_32.lib)
+if (WIN32)
+	if (MSVC)
+		set (SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ws2_32.lib)
+	else ()
+		set (SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ws2_32)
+	endif()
 endif()
 
 # ---
@@ -541,8 +549,12 @@
 endif()
 
 if (srt_libspec_shared)
-if (MICROSOFT)
-    target_link_libraries(${TARGET_srt}_shared PUBLIC Ws2_32.lib)
+if (WIN32)
+	if (MSVC)
+		target_link_libraries(${TARGET_srt}_shared PUBLIC Ws2_32.lib)
+	else ()
+		target_link_libraries(${TARGET_srt}_shared PUBLIC ws2_32)
+	endif()
 endif()
 endif()
 
