--- arrow-adbc-apache-arrow-adbc-22/c/cmake_modules/GoUtils.cmake.orig	2026-01-20 18:40:47.649032600 +0100
+++ arrow-adbc-apache-arrow-adbc-22/c/cmake_modules/GoUtils.cmake	2026-01-20 18:40:57.586629100 +0100
@@ -181,7 +181,7 @@
       set(EXTLDFLAGS ",--version-script=${REPOSITORY_ROOT}/c/symbols.map")
     endif()
 
-    if(NOT APPLE)
+    if(NOT APPLE AND NOT WIN32)
       set(EXTLDFLAGS "${EXTLDFLAGS},-soname,${LIB_NAME_SHARED}.${ADBC_SO_VERSION}")
     endif()
 
