--- arrow-adbc-23/c/cmake_modules/GoUtils.cmake
+++ arrow-adbc-23/c/cmake_modules/GoUtils.cmake
@@ -184,13 +184,15 @@
     set(LIB_NAME_SHARED
         "${CMAKE_SHARED_LIBRARY_PREFIX}${GO_LIBNAME}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 
-    set(ADBC_VERSION_SCRIPT_LINK_FLAG
-        "-Wl,--version-script=${REPOSITORY_ROOT}/c/symbols.map")
+    if(NOT WIN32)
+      set(ADBC_VERSION_SCRIPT_LINK_FLAG
+          "-Wl,--version-script=${REPOSITORY_ROOT}/c/symbols.map")
 
-    check_linker_flag(CXX ${ADBC_VERSION_SCRIPT_LINK_FLAG}
-                      CXX_LINKER_SUPPORTS_VERSION_SCRIPT)
-    if(CXX_LINKER_SUPPORTS_VERSION_SCRIPT)
-      set(EXTLDFLAGS ",--version-script=${REPOSITORY_ROOT}/c/symbols.map")
+      check_linker_flag(CXX ${ADBC_VERSION_SCRIPT_LINK_FLAG}
+                        CXX_LINKER_SUPPORTS_VERSION_SCRIPT)
+      if(CXX_LINKER_SUPPORTS_VERSION_SCRIPT)
+        set(EXTLDFLAGS ",--version-script=${REPOSITORY_ROOT}/c/symbols.map")
+      endif()
     endif()
 
     if(NOT APPLE AND NOT WIN32)
