unchanged:
--- libblocksruntime-0.4.1/configure-orig.ac	2017-11-08 18:52:55.961703800 -0500
+++ libblocksruntime-0.4.1/configure.ac	2017-11-08 18:53:12.298880700 -0500
@@ -3,7 +3,7 @@
 
 AC_INIT([libBlocksRuntime], [0.4.1])
 AC_PROG_CC([clang])
-LT_INIT
+LT_INIT[(win32-dll)]
 AM_INIT_AUTOMAKE([foreign subdir-objects])
 AC_CONFIG_SRCDIR([configure.ac])
 AC_CONFIG_HEADERS([config.h])
only in patch2:
unchanged:
--- libblocksruntime-0.4.1/Makefile-orig.am	2017-11-08 18:58:16.324977000 -0500
+++ libblocksruntime-0.4.1/Makefile.am	2017-11-08 18:58:19.477580300 -0500
@@ -6,6 +6,8 @@
 
 libBlocksRuntime_la_CFLAGS = -std=c99 -Wall -Wextra -W -pedantic
 
+libBlocksRuntime_la_LDFLAGS = -no-undefined
+
 libBlocksRuntime_la_SOURCES = runtime.c data.c config2.h
 
 check_PROGRAMS = blockrt-test
