--- gdal-2.1.3/configure.in.orig	2017-01-31 15:53:29.143937100 +0300
+++ gdal-2.1.3/configure.in	2017-01-31 15:53:45.180854300 +0300
@@ -3639,7 +3639,7 @@
         else
             EXTRA_INCLUDES="-I/usr/local/include/qhull $EXTRA_INCLUDES"
         fi
-        LIBS="-lqhull $LIBS"
+        LIBS="-lqhullstatic $LIBS"
     fi
   fi
   AC_CHECK_HEADERS([libqhull/libqhull.h])
@@ -3652,7 +3652,7 @@
         else
             EXTRA_INCLUDES="-I/usr/local/include/libqhull $EXTRA_INCLUDES"
         fi
-        LIBS="-lqhull $LIBS"
+        LIBS="-lqhullstatic $LIBS"
     fi
   fi
 
