--- gdal-2.2.0/configure.ac.orig	2017-01-31 15:53:29.143937100 +0300
+++ gdal-2.2.0/configure.ac	2017-01-31 15:53:45.180854300 +0300
@@ -3976,7 +3976,7 @@
     if test "$QHULL_SETTING" = "yes"; then
         QHULL_SETTING=external
         QHULL_INCLUDE_SUBDIR_IS_LIBQHULL=0
-        LIBS="-lqhull $LIBS"
+        LIBS="-lqhullstatic $LIBS"
     fi
   fi
  AC_CHECK_HEADERS([libqhull/libqhull.h])
@@ -3985,7 +3985,7 @@
     if test "$QHULL_SETTING" = "yes"; then
         QHULL_SETTING=external
         QHULL_INCLUDE_SUBDIR_IS_LIBQHULL=1
-        LIBS="-lqhull $LIBS"
+        LIBS="-lqhullstatic $LIBS"
     fi
   fi
 
