--- gdal-2.2.0/ogr/ogrct.cpp.ori	2017-04-28 18:15:25.000000000 +0200
+++ gdal-2.2.0/ogr/ogrct.cpp	2017-06-23 14:18:19.231068600 +0200
@@ -96,7 +96,7 @@
 // build) it can be named either way, so use PROJSO environment variable to
 // specify the right library name. By default assume that in Cygwin/MinGW all
 // components were built in the same way.
-#  define LIBNAME "libproj-9.dll"
+#  define LIBNAME "libproj-12.dll"
 #elif defined(__CYGWIN__)
 #  define LIBNAME "cygproj-9.dll"
 #elif defined(__APPLE__)
