--- gdal-2.2.4/frmts/pdf/pdfobject.cpp	2018-03-19 16:26:54.000000000 +0300
+++ gdal-2.2.4/frmts/pdf/pdfobject.cpp	2018-04-24 11:09:49.685426800 +0300
@@ -1055,7 +1055,7 @@
 {
     if (GetType() == PDFObjectType_String)
     {
-        GooString* gooString = m_po->getString();
+        const GooString* gooString = m_po->getString();
         return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->getCString()),
                                                       static_cast<int>(gooString->getLength())));
     }
