diff -bur gdal-3.6.3-orig/swig/python/CMakeLists.txt gdal-3.6.3/swig/python/CMakeLists.txt
--- gdal-3.6.3-orig/swig/python/CMakeLists.txt	2023-03-28 13:21:24.576109900 -0600
+++ gdal-3.6.3/swig/python/CMakeLists.txt	2023-03-28 13:22:45.069307000 -0600
@@ -459,7 +459,11 @@
         set(INSTALL_ARGS "${INSTALL_ARGS} \"--install-lib=${GDAL_PYTHON_INSTALL_LIB}\"")
       endif ()
     else ()
+      if (MINGW)
+        set(INSTALL_ARGS "${INSTALL_ARGS} \"--prefix=${CMAKE_INSTALL_PREFIX}\"")
+      else()
       set(INSTALL_ARGS "${INSTALL_ARGS} \"--prefix=${NATIVE_CMAKE_INSTALL_PREFIX}\"")
+      endif()
     endif ()
   endif ()
 
