diff -Naur gdal-2.4.0-orig/configure.ac gdal-2.4.0/configure.ac
--- gdal-2.4.0-orig/configure.ac	2018-12-15 00:34:19.000000000 +0300
+++ gdal-2.4.0/configure.ac	2018-12-25 11:58:47.052523400 +0300
@@ -4499,6 +4499,8 @@
 POPPLER_0_23_OR_LATER=no
 POPPLER_0_58_OR_LATER=no
 POPPLER_0_69_OR_LATER=no
+POPPLER_0_72_OR_LATER=no
+POPPLER_0_73_OR_LATER=no
 
 AC_MSG_CHECKING([for poppler])
 
@@ -4540,6 +4541,53 @@
         CHECK_OTHER_POPPLER_VERSION=yes
 
         if test "$CHECK_OTHER_POPPLER_VERSION" = "yes"; then
+            AC_MSG_CHECKING([if poppler have gtypes.h header])
+            rm -f testpoppler.*
+            echo '#include <poppler/goo/gtypes.h>' > testpoppler.cpp
+            echo 'int main(int argc, char** argv) {' >> testpoppler.cpp
+            echo 'return 0; }' >> testpoppler.cpp
+            if test -z "`${CXX} ${CXXFLAGS} ${CPPFLAGS} testpoppler.cpp -c ${POPPLER_INC} 2>&1`" ; then
+                AC_MSG_RESULT([yes])
+            else
+                POPPLER_0_73_OR_LATER=yes
+                POPPLER_0_72_OR_LATER=yes
+                POPPLER_0_71_OR_LATER=yes
+                POPPLER_0_69_OR_LATER=yes
+                POPPLER_0_58_OR_LATER=yes
+                POPPLER_0_23_OR_LATER=yes
+                POPPLER_0_20_OR_LATER=yes
+                POPPLER_BASE_STREAM_HAS_TWO_ARGS=yes
+                POPPLER_HAS_OPTCONTENT=yes
+                CHECK_OTHER_POPPLER_VERSION=no
+                AC_MSG_RESULT([no])
+            fi
+        fi
+
+        if test "$CHECK_OTHER_POPPLER_VERSION" = "yes"; then
+            AC_MSG_CHECKING([if GooString has API >= 0.72.0])
+            rm -f testpoppler.*
+            echo '#include <poppler/goo/GooString.h>' > testpoppler.cpp
+            echo 'int main(int argc, char** argv) {' >> testpoppler.cpp
+            echo 'GooString *goostr = new GooString("Test");' >> testpoppler.cpp
+            echo 'const char* data = (const char*)goostr->c_str();' >> testpoppler.cpp
+            echo 'return 0; }' >> testpoppler.cpp
+            if test -z "`${CXX} ${CXXFLAGS} ${CPPFLAGS} testpoppler.cpp -c ${POPPLER_INC} 2>&1`" ; then
+                POPPLER_0_72_OR_LATER=yes
+                POPPLER_0_71_OR_LATER=yes
+                POPPLER_0_69_OR_LATER=yes
+                POPPLER_0_58_OR_LATER=yes
+                POPPLER_0_23_OR_LATER=yes
+                POPPLER_0_20_OR_LATER=yes
+                POPPLER_BASE_STREAM_HAS_TWO_ARGS=yes
+                POPPLER_HAS_OPTCONTENT=yes
+                CHECK_OTHER_POPPLER_VERSION=no
+                AC_MSG_RESULT([yes])
+            else
+                AC_MSG_RESULT([no])
+            fi
+        fi
+
+        if test "$CHECK_OTHER_POPPLER_VERSION" = "yes"; then
             AC_MSG_CHECKING([if OptionalContent has API >= 0.69.0])
             rm -f testpoppler.*
             echo '#include <poppler/OptionalContent.h>' > testpoppler.cpp
@@ -4663,6 +4688,8 @@
 AC_SUBST(POPPLER_0_23_OR_LATER, $POPPLER_0_23_OR_LATER)
 AC_SUBST(POPPLER_0_58_OR_LATER, $POPPLER_0_58_OR_LATER)
 AC_SUBST(POPPLER_0_69_OR_LATER, $POPPLER_0_69_OR_LATER)
+AC_SUBST(POPPLER_0_72_OR_LATER, $POPPLER_0_72_OR_LATER)
+AC_SUBST(POPPLER_0_73_OR_LATER, $POPPLER_0_73_OR_LATER)
 AC_SUBST(POPPLER_INC, $POPPLER_INC)
 AC_SUBST(POPPLER_PLUGIN_LIB, $POPPLER_PLUGIN_LIB)
 
diff -Naur gdal-2.4.0-orig/frmts/pdf/GNUmakefile gdal-2.4.0/frmts/pdf/GNUmakefile
--- gdal-2.4.0-orig/frmts/pdf/GNUmakefile	2018-12-15 00:34:20.000000000 +0300
+++ gdal-2.4.0/frmts/pdf/GNUmakefile	2018-12-25 11:59:44.039423500 +0300
@@ -38,6 +38,14 @@
 CPPFLAGS +=  -DPOPPLER_0_69_OR_LATER
 endif
 
+ifeq ($(POPPLER_0_72_OR_LATER),yes)
+CPPFLAGS +=  -DPOPPLER_0_72_OR_LATER
+endif
+
+ifeq ($(POPPLER_0_73_OR_LATER),yes)
+CPPFLAGS +=  -DPOPPLER_0_73_OR_LATER
+endif
+
 ifeq ($(HAVE_PODOFO),yes)
 CPPFLAGS +=  -DHAVE_PODOFO
 endif
diff -Naur gdal-2.4.0-orig/frmts/pdf/pdfdataset.cpp gdal-2.4.0/frmts/pdf/pdfdataset.cpp
--- gdal-2.4.0-orig/frmts/pdf/pdfdataset.cpp	2018-12-15 00:37:15.000000000 +0300
+++ gdal-2.4.0/frmts/pdf/pdfdataset.cpp	2018-12-25 12:03:40.855376100 +0300
@@ -3459,7 +3459,11 @@
 #endif
             if( ocg != nullptr && ocg->getName() != nullptr )
             {
+#ifdef POPPLER_0_72_OR_LATER
+                const char* pszLayerName = (const char*)ocg->getName()->c_str();
+#else
                 const char* pszLayerName = (const char*)ocg->getName()->getCString();
+#endif
                 AddLayer(pszLayerName);
                 oLayerOCGMapPoppler[pszLayerName] = ocg;
             }
@@ -4787,7 +4791,11 @@
     GooString* poMetadata = poCatalogPoppler->readMetadata();
     if (poMetadata)
     {
+#ifdef POPPLER_0_72_OR_LATER
+        const char* pszContent = poMetadata->c_str();
+#else
         const char* pszContent = poMetadata->getCString();
+#endif
         if (pszContent != nullptr &&
             STARTS_WITH(pszContent, "<?xpacket begin="))
         {
diff -Naur gdal-2.4.0-orig/frmts/pdf/pdfobject.cpp gdal-2.4.0/frmts/pdf/pdfobject.cpp
--- gdal-2.4.0-orig/frmts/pdf/pdfobject.cpp	2018-12-15 00:37:15.000000000 +0300
+++ gdal-2.4.0/frmts/pdf/pdfobject.cpp	2018-12-25 12:05:19.681549600 +0300
@@ -1061,7 +1061,11 @@
 #else
         GooString* gooString = m_po->getString();
 #endif
+#ifdef POPPLER_0_72_OR_LATER
+        return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->c_str()),
+#else
         return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->getCString()),
+#endif
                                                       static_cast<int>(gooString->getLength())));
     }
     else
@@ -1422,7 +1426,11 @@
         char* pszContent = (char*) VSIMalloc(m_nLength + 1);
         if (pszContent)
         {
+#ifdef POPPLER_0_72_OR_LATER
+            memcpy(pszContent, gstr->c_str(), m_nLength);
+#else
             memcpy(pszContent, gstr->getCString(), m_nLength);
+#endif
             pszContent[m_nLength] = '\0';
         }
         delete gstr;
diff -Naur gdal-2.4.0-orig/frmts/pdf/pdfsdk_headers.h gdal-2.4.0/frmts/pdf/pdfsdk_headers.h
--- gdal-2.4.0-orig/frmts/pdf/pdfsdk_headers.h	2018-12-15 00:37:15.000000000 +0300
+++ gdal-2.4.0/frmts/pdf/pdfsdk_headers.h	2018-12-25 12:05:19.681549600 +0300
@@ -50,7 +50,11 @@
 #pragma warning( disable : 4244 ) /* conversion from 'const int' to 'Guchar', possible loss of data */
 #endif
 
+#if defined(POPPLER_0_73_OR_LATER)
+typedef unsigned char Guchar;
+#else
 #include <goo/gtypes.h>
+#endif
 #include <goo/GooList.h>
 
 /* begin of poppler xpdf includes */
diff -Naur gdal-2.4.0-orig/GDALmake.opt.in gdal-2.4.0/GDALmake.opt.in
--- gdal-2.4.0-orig/GDALmake.opt.in	2018-12-15 00:34:19.000000000 +0300
+++ gdal-2.4.0/GDALmake.opt.in	2018-12-25 11:55:57.757409400 +0300
@@ -482,6 +482,8 @@
 POPPLER_0_23_OR_LATER = @POPPLER_0_23_OR_LATER@
 POPPLER_0_58_OR_LATER = @POPPLER_0_58_OR_LATER@
 POPPLER_0_69_OR_LATER = @POPPLER_0_69_OR_LATER@
+POPPLER_0_72_OR_LATER = @POPPLER_0_72_OR_LATER@
+POPPLER_0_73_OR_LATER = @POPPLER_0_73_OR_LATER@
 POPPLER_INC = @POPPLER_INC@
 POPPLER_PLUGIN_LIB = @POPPLER_PLUGIN_LIB@
 
