diff -urN gdal-3.3.3-orig/port/cpl_port.h gdal-3.3.3/port/cpl_port.h
--- gdal-3.3.3-orig/port/cpl_port.h	2021-11-11 23:24:19 +0800
+++ gdal-3.3.3/port/cpl_port.h	2021-11-13 15:06:41 +0800
@@ -353,7 +353,7 @@
 #else
 #  if defined(USE_GCC_VISIBILITY_FLAG)
 #    define CPL_DLL     __attribute__ ((visibility("default")))
-#    if !defined(__MINGW32__)
+#    if !defined(__MINGW32__) && !defined(__clang__)
 #        define CPL_INTERNAL __attribute__((visibility("hidden")))
 #    else
 #        define CPL_INTERNAL
