--- gdal-3.9.3/frmts/pdf/pdfsdk_headers.h.orig	2024-10-07 18:02:06.000000000 +0200
+++ gdal-3.9.3/frmts/pdf/pdfsdk_headers.h	2025-01-03 11:38:41.758861300 +0100
@@ -37,6 +37,11 @@
 
 #ifdef HAVE_POPPLER
 
+// The "#define private public" hacks we do below do not play well with the
+// sstream header.
+// Cf https://github.com/conda-forge/gdal-feedstock/pull/1019#issuecomment-2528710051
+#include <sstream>
+
 /* Horrible hack because there's a conflict between struct FlateDecode of */
 /* include/poppler/Stream.h and the FlateDecode() function of */
 /* pdfium/core/include/fpdfapi/fpdf_parser.h. */
