diff -urN gdal-3.4.0-orig/port/cpl_port.h gdal-3.4.0/port/cpl_port.h
--- gdal-3.4.0-orig/port/cpl_port.h	2021-11-11 23:24:19 +0800
+++ gdal-3.4.0/port/cpl_port.h	2021-11-19 23:02:46 +0800
@@ -326,12 +326,16 @@
 #endif
 
 #ifndef CPL_DLL
-#if defined(_MSC_VER) && !defined(CPL_DISABLE_DLL)
+#if (defined(_MSC_VER) || defined(__MINGW32__)) && !defined(CPL_DISABLE_DLL)
+# ifdef DLL_EXPORT
 #  ifdef GDAL_COMPILATION
 #    define CPL_DLL __declspec(dllexport)
 #  else
-#    define CPL_DLL
+#    define CPL_DLL __declspec(dllimport)
 #  endif
+# else
+#  define CPL_DLL
+# endif
 #  define CPL_INTERNAL
 #else
 #  if defined(USE_GCC_VISIBILITY_FLAG)
