--- a/apps/gdalalg_raster_tile.h
+++ b/apps/gdalalg_raster_tile.h
@@ -17,6 +17,7 @@
 
 #include "cpl_string.h"
 
+#include <atomic>
 #include <limits>
 
 //! @cond Doxygen_Suppress
--- a/port/cpl_vsil_curl.cpp
+++ b/port/cpl_vsil_curl.cpp
@@ -14,6 +14,7 @@
 #include "cpl_vsil_curl_priv.h"
 #include "cpl_vsil_curl_class.h"
 
+#include <cstdlib>
 #include <algorithm>
 #include <array>
 #include <limits>
