diff -urN gdal-3.4.0-orig/GDALmake.opt.in gdal-3.4.0/GDALmake.opt.in
--- gdal-3.4.0-orig/GDALmake.opt.in	2021-11-04 19:55:37 +0800
+++ gdal-3.4.0/GDALmake.opt.in	2021-11-16 23:09:53 +0800
@@ -27,9 +27,10 @@
 LD	=	$(LIBTOOL_LINK) @CXX@
 RM	=	$(LIBTOOL_CLEAN) /bin/rm -f *.lo
 INSTALL = 	$(LIBTOOL_INSTALL) $(GDAL_ROOT)/install-sh -c
-INSTALL_LIB 	=	$(LIBTOOL_INSTALL) $(GDAL_ROOT)/install-sh -c
+INSTALL_BIN 	=	$(LIBTOOL_INSTALL) $(GDAL_ROOT)/install-sh -c -m 0755
+INSTALL_LIB 	=	$(LIBTOOL_INSTALL) $(GDAL_ROOT)/install-sh -c -m 0644
 INSTALL_DATA	= 	$(GDAL_ROOT)/install-sh -c -m 0644
-INSTALL_DIR	= 	$(GDAL_ROOT)/install-sh -d
+INSTALL_DIR	= 	mkdir -p
 
 LIBS	=	@LIBS@ $(KAK_LIBS) $(DWG_LIBS) $(CURL_LIB) \
 		$(MRSID_LIBS) $(MRSID_LIDAR_LIBS) $(ECW_LIBS) $(INGRES_LIB) \
@@ -73,7 +74,7 @@
 INST_MAN	=	@mandir@
 INST_WEB	=	$(HOME)/www/gdal
 
-CPPFLAGS	:= @CPPFLAGS@ -iquote $(GDAL_ROOT)/port @EXTRA_INCLUDES@ -DGDAL_COMPILATION
+CPPFLAGS	:= @CPPFLAGS@ -iquote $(GDAL_ROOT)/port -iquote $(GDAL_ROOT)/alg/marching_squares @EXTRA_INCLUDES@ -DGDAL_COMPILATION
 CFLAGS		= @CFLAGS@ @C_WFLAGS@ $(USER_DEFS)
 CXXFLAGS	= @CXXFLAGS@ @CXX_WFLAGS@ $(USER_DEFS)
 CFLAGS_NOFTRAPV          = @CFLAGS_NOFTRAPV@ @C_WFLAGS@ $(USER_DEFS)
@@ -140,9 +141,9 @@
 GDAL_VERSION_REV   =    @GDAL_VERSION_REV@
 
 GDAL_LIB	= 	$(GDAL_ROOT)/libgdal.a
-GDAL_SLIB	=	$(GDAL_ROOT)/libgdal.$(SO_EXT)
-GDAL_SLIB_LINK	=	-L$(GDAL_ROOT) -lgdal
-#GDAL_SLIB_SONAME = -Wl,-soname,libgdal.$(SO_EXT).@GDAL_VERSION_MAJOR@
+GDAL_SLIB	=	$(GDAL_ROOT)/libgdal-$(LIBGDAL_CURRENT).$(SO_EXT)
+GDAL_SLIB_LINK	=	$(GDAL_ROOT)/$(GDAL_SLIB_SONAME)
+GDAL_SLIB_SONAME =	libgdal.$(SO_EXT).a
 
 # Mac OS X Framework definition
 MACOSX_FRAMEWORK = @MACOSX_FRAMEWORK@
@@ -630,17 +631,16 @@
 else # HAVE_LIBTOOL
 
 ifeq ($(HAVE_LD_SHARED),yes)
-CONFIG_LIBS	=	$(GDAL_SLIB_LINK)
+CONFIG_LIBS	=	$(GDAL_SLIB_LINK) $(LIBS)
 ifeq ($(MACOSX_FRAMEWORK),yes)
 CONFIG_LIBS_INS	=	-L$(INST_LIB)/unix/lib -lgdal
 else
-CONFIG_LIBS_INS	=	-L$(INST_LIB) -lgdal
+CONFIG_LIBS_INS	=	-L$(INST_LIB) -lgdal $(LIBS)
 endif
 EXE_DEP_LIBS	=	$(GDAL_SLIB)
 else
-CONFIG_LIBS	=	$(GDAL_LIBS) $(LIBS)
-CONFIG_LIBS_INS	=	$(foreach LF,$(GDAL_LIBS),$(INST_LIB)/$(notdir $(LF)))\
-			 $(LIBS)
+CONFIG_LIBS	=	$(GDAL_LIB) $(LIBS)
+CONFIG_LIBS_INS	=	-L$(INST_LIB) -lgdal $(LIBS)
 EXE_DEP_LIBS	=	$(GDAL_LIB)
 endif
 
diff -urN gdal-3.4.0-orig/GNUmakefile gdal-3.4.0/GNUmakefile
--- gdal-3.4.0-orig/GNUmakefile	2021-11-04 19:55:37 +0800
+++ gdal-3.4.0/GNUmakefile	2021-11-16 23:14:24 +0800
@@ -3,7 +3,10 @@
 GDAL_OBJ	=	frmts/o/*.o \
 			gcore/*.o \
 			port/*.o \
-			alg/*.o \
+			alg/*.o
+
+GDAL_BIN_OBJ	=	alg/contour.o \
+			alg/viewshed.o \
 			apps/commonutils.o \
 			apps/gdalinfo_lib.o \
 			apps/gdalmdiminfo_lib.o \
@@ -56,16 +59,15 @@
 
 static-lib-stage2: $(GDAL_OBJ)
 	rm -f libgdal.a
-	$(AR) r $(GDAL_LIB) $(GDAL_OBJ)
+	$(AR) r $(GDAL_LIB) $(GDAL_OBJ) $(GDAL_BIN_OBJ)
 
 $(GDAL_LIB):	$(GDAL_OBJ) GDALmake.opt
 	rm -f libgdal.a
 	$(AR) r $(GDAL_LIB) $(GDAL_OBJ)
 	$(RANLIB) $(GDAL_LIB)
 
-$(GDAL_SLIB):	$(GDAL_OBJ) $(GDAL_LIB)
-	$(LD_SHARED) $(GDAL_SLIB_SONAME) $(GDAL_OBJ) $(GDAL_LIBS) $(LDFLAGS) $(LIBS) \
-		-o $(GDAL_SLIB)
+$(GDAL_SLIB):	$(GDAL_LIB) $(GDAL_BIN_OBJ) $(GDAL_OBJ)
+	for f in $(GDAL_OBJ); do $(LD_SHARED) $$f $(GDAL_BIN_OBJ) $(GDAL_LIB) $(LDFLAGS) $(LIBS) -Wl,--out-implib,$(GDAL_SLIB_LINK) -o $(GDAL_SLIB) ; done
 
 #  split potentially long lines
 SORTED  := $(sort $(wildcard $(GDAL_OBJ:.o=.lo)))
@@ -275,14 +277,10 @@
 ifeq ($(MACOSX_FRAMEWORK),yes)
 	$(INSTALL_LIB) $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)/GDAL
 else
-	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B)
-	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR)
-	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
-	$(INSTALL_LIB) $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
-	(cd $(DESTDIR)$(INST_LIB) ; \
-	 ln -s $(GDAL_SLIB_B).$(GDAL_VER) $(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR))
-	(cd $(DESTDIR)$(INST_LIB) ; \
-	 ln -s $(GDAL_SLIB_B).$(GDAL_VERSION_MAJOR) $(GDAL_SLIB_B))
+	rm -f $(DESTDIR)$(INST_BIN)/$(GDAL_SLIB_B)
+	rm -f $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_SONAME)
+	$(INSTALL_BIN) $(GDAL_SLIB) $(DESTDIR)$(INST_BIN)/$(GDAL_SLIB_B)
+	$(INSTALL_LIB) $(GDAL_ROOT)/$(GDAL_SLIB_SONAME) $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_SONAME)
 	$(INSTALL_DIR) $(DESTDIR)$(INST_LIB)/gdalplugins
 endif
 
diff -urN gdal-3.4.0-orig/alg/GNUmakefile gdal-3.4.0/alg/GNUmakefile
--- gdal-3.4.0-orig/alg/GNUmakefile	2021-11-04 19:55:37 +0800
+++ gdal-3.4.0/alg/GNUmakefile	2021-11-16 22:25:34 +0800
@@ -8,9 +8,9 @@
 		gdalgrid.o gdalcutline.o gdalproximity.o rasterfill.o \
 		gdalrasterpolygonenumerator.o \
 		gdalsievefilter.o gdalwarpkernel_opencl.o polygonize.o \
-		contour.o gdaltransformgeolocs.o gdallinearsystem.o \
+		gdaltransformgeolocs.o gdallinearsystem.o \
 		gdal_octave.o gdal_simplesurf.o gdalmatching.o delaunay.o \
-		gdalpansharpen.o gdalapplyverticalshiftgrid.o viewshed.o
+		gdalpansharpen.o gdalapplyverticalshiftgrid.o
 
 ifeq ($(HAVE_GEOS),yes)
 CPPFLAGS 	:=	-DHAVE_GEOS=1 $(GEOS_CFLAGS) $(CPPFLAGS)
@@ -45,6 +45,12 @@
 gdalgridsse.$(OBJ_EXT):   gdalgridsse.cpp
 	$(CXX) $(GDAL_INCLUDE) $(CXXFLAGS) $(WARN_OLD_STYLE_CAST) $(SSEFLAGS) $(CPPFLAGS) -c -o $@ $<
 
+contour.$(OBJ_EXT):   contour.cpp
+	$(CXX) $(GDAL_INCLUDE) $(CXXFLAGS) $(CPPFLAGS) -c -o $@ $<
+
+viewshed.$(OBJ_EXT):   viewshed.cpp
+	$(CXX) $(GDAL_INCLUDE) $(CXXFLAGS) $(CPPFLAGS) -c -o $@ $<
+
 clean:
 	$(RM) *.o $(O_OBJ)
 
