diff --git a/gdal/ogr/ogrsf_frmts/wfs/ogrwfslayer.cpp b/gdal/ogr/ogrsf_frmts/wfs/ogrwfslayer.cpp
index b82f7a3..d08ae58 100644
--- a/gdal/ogr/ogrsf_frmts/wfs/ogrwfslayer.cpp
+++ b/gdal/ogr/ogrsf_frmts/wfs/ogrwfslayer.cpp
@@ -1195,6 +1195,8 @@ OGRFeature *OGRWFSLayer::GetNextFeature()
             if (poBaseDS)
             {
                 poBaseLayer = poBaseDS->GetLayer(0);
+                if(poBaseLayer == NULL)
+                    return NULL;
                 poBaseLayer->ResetReading();
 
                 /* Check that the layer field definition is consistent with the one */
-- 
