diff -Naur glbinding-3.1.0.orig/source/glbinding/source/getProcAddress.cpp glbinding-3.1.0/source/glbinding/source/getProcAddress.cpp
--- glbinding-3.1.0.orig/source/glbinding/source/getProcAddress.cpp	2019-05-24 13:36:30.490322100 -0400
+++ glbinding-3.1.0/source/glbinding/source/getProcAddress.cpp	2019-05-26 10:29:14.896609400 -0400
@@ -18,10 +18,10 @@
 
 ProcAddress getProcAddress(const char * name)
 {
-    static auto module = LoadLibrary(_T("OPENGL32.DLL"));
+    static auto module = LoadLibrary(_T("opengl32.dll"));
 
 	// Prevent static linking of opengl32
-	static auto wglGetProcAddress_ = reinterpret_cast<void * (__stdcall *)(const char *)>(::GetProcAddress(module, "wglGetProcAddress"));
+	static auto wglGetProcAddress_ = reinterpret_cast<void * (__stdcall *)(const char *)>((intptr_t) ::GetProcAddress(module, "wglGetProcAddress"));
 	assert(wglGetProcAddress_ != nullptr);
 
 	auto procAddress = wglGetProcAddress_(name);
@@ -30,7 +30,7 @@
 		return reinterpret_cast<ProcAddress>(procAddress);
 	}
 
-	procAddress = ::GetProcAddress(module, name);
+	procAddress = (void *) ::GetProcAddress(module, name);
     return reinterpret_cast<ProcAddress>(procAddress);
 }
 
