--- opencollada/common/libBuffer/include/CommonFWriteBufferFlusher.h.orig	2016-10-14 12:42:00.737418400 +0300
+++ opencollada/common/libBuffer/include/CommonFWriteBufferFlusher.h	2016-10-14 12:44:02.532358500 +0300
@@ -40,6 +40,9 @@
 #endif
 #endif
 
+#ifdef __MINGW32__
+#include <inttypes.h>
+#else
 #if __cplusplus > 199711L
 typedef int64_t __int64;
 #else
@@ -47,6 +50,7 @@
 typedef int64_t __int64;
 #endif
 #endif
+#endif
 
 namespace Common
 {
