--- opencollada/Externals/MathMLSolver/include/MathMLPlatform.h.orig	2013-12-31 00:26:36.296400000 +0400
+++ opencollada/Externals/MathMLSolver/include/MathMLPlatform.h	2013-12-31 00:27:33.751200000 +0400
@@ -127,19 +127,7 @@
 #       undef MML_DEBUG_MODE
 #   endif
 
-    // Disable unicode support on MingW at the moment, poorly supported in stdlibc++
-    // STLPORT fixes this though so allow if found
-    // MinGW C++ Toolkit supports unicode and sets the define __MINGW32_TOOLKIT_UNICODE__ in _mingw.h
-#if defined( __MINGW32__ ) && !defined(_STLPORT_VERSION) && !defined( __MINGW64__ )
-#   include<_mingw.h>
-#   if defined(__MINGW32_TOOLBOX_UNICODE__)
-#     define MML_UNICODE_SUPPORT 1
-#   else
-#       define MML_UNICODE_SUPPORT 0
-#   endif
-#else
-# define MML_UNICODE_SUPPORT 1
-#endif
+#define MML_UNICODE_SUPPORT 1
 
 #endif
     //----------------------------------------------------------------------------
