From 1a3b5bf2f4204453a72c66b608149cd7ed55e26d Mon Sep 17 00:00:00 2001
From: Thomas Dickerson <elfprince13@gmail.com>
Date: Mon, 4 Mar 2019 17:15:15 -0500
Subject: [PATCH] Please use C++ Headers for C++ Files

---
 GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp b/GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
index 1f9a3eef..c3450e9d 100644
--- a/GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
+++ b/GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
@@ -9,9 +9,9 @@
 */
 
 #include "GeneratedSaxParserUtils.h"
-#include <math.h>
+#include <cmath>
 #include <memory>
-#include <string.h>
+#include <cstring>
 #include <limits>
 #include <algorithm>
 
