diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/common/g++-win32.conf qt-everywhere-src-5.12.4/qtbase/mkspecs/common/g++-win32.conf
--- qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/common/g++-win32.conf	2019-06-15 14:54:16.704696800 +0300
+++ qt-everywhere-src-5.12.4/qtbase/mkspecs/common/g++-win32.conf	2019-06-15 15:06:52.137023700 +0300
@@ -51,6 +51,7 @@
 QMAKE_LFLAGS_CONSOLE    = -Wl,-subsystem,console
 QMAKE_LFLAGS_WINDOWS    = -Wl,-subsystem,windows
 QMAKE_LFLAGS_DLL        = -shared
+QMAKE_LFLAGS_STATIC_LIB = -static
 QMAKE_LFLAGS_GCSECTIONS = -Wl,--gc-sections
 equals(QMAKE_HOST.os, Windows) {
     QMAKE_LINK_OBJECT_MAX = 10
diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/default_post.prf qt-everywhere-src-5.12.4/qtbase/mkspecs/features/default_post.prf
--- qt-everywhere-src-5.12.4-orig/qtbase/mkspecs/features/default_post.prf	2019-06-12 23:59:14.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtbase/mkspecs/features/default_post.prf	2019-06-15 15:06:52.137023700 +0300
@@ -83,8 +83,8 @@
     enable_gdb_index: QMAKE_LFLAGS += $$QMAKE_LFLAGS_GDB_INDEX
 }
 
+static: QMAKE_LFLAGS += $$QMAKE_LFLAGS_STATIC_LIB
 dll:win32: QMAKE_LFLAGS += $$QMAKE_LFLAGS_DLL
-static:mac: QMAKE_LFLAGS += $$QMAKE_LFLAGS_STATIC_LIB
 staticlib:unix {
     QMAKE_CFLAGS += $$QMAKE_CFLAGS_STATIC_LIB
     QMAKE_CXXFLAGS += $$QMAKE_CXXFLAGS_STATIC_LIB
