--- amqp-cpp-4.1.6/CMakeLists.txt.orig	2020-05-07 09:58:26.355425900 +0300
+++ amqp-cpp-4.1.6/CMakeLists.txt	2020-05-07 10:00:51.572604100 +0300
@@ -111,11 +111,11 @@
         FILES_MATCHING PATTERN "*.h")
 install(FILES include/amqpcpp.h DESTINATION include)
 
-install(EXPORT ${PROJECT_NAME}Config DESTINATION cmake)
+install(EXPORT ${PROJECT_NAME}Config DESTINATION lib/cmake/${PROJECT_NAME})
 export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_NAME}Config.cmake)
 
 set(DEST_DIR "${CMAKE_INSTALL_PREFIX}")
-set(PRIVATE_LIBS "-llibamqpcc")
+set(PRIVATE_LIBS "-lws2_32")
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/amqpcpp.pc.in"
                "${CMAKE_CURRENT_BINARY_DIR}/amqpcpp.pc" @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/amqpcpp.pc" DESTINATION lib/pkgconfig)
--- amqp-cpp-4.1.6/amqpcpp.pc.in.orig	2020-05-07 11:24:28.140401500 +0300
+++ amqp-cpp-4.1.6/amqpcpp.pc.in	2020-05-07 11:25:04.787766700 +0300
@@ -7,4 +7,5 @@
 Version: @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@
 
 Libs: -L${libdir} -lamqpcpp
+Libs.private:  @PRIVATE_LIBS@
 Cflags: -I${includedir}
