--- amqp-cpp-4.1.6/CMakeLists.txt.orig	2020-05-07 11:20:36.573553800 +0300
+++ amqp-cpp-4.1.6/CMakeLists.txt	2020-05-07 11:21:52.837693500 +0300
@@ -100,7 +100,7 @@
     install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config
             ARCHIVE DESTINATION lib
             LIBRARY DESTINATION lib
-            RUNTIME DESTINATION lib
+            RUNTIME DESTINATION bin
     )
 else()
     # copy static lib
