--- glade-3.16.1/gladeui/Makefile.am.orig	2013-07-28 21:11:04.599868800 +0000
+++ glade-3.16.1/gladeui/Makefile.am	2013-07-28 21:18:26.664003800 +0000
@@ -259,6 +259,6 @@
 
 Gladeui-2.0.gir: libgladeui-2.la
 Gladeui_2_0_gir_INCLUDES = Gtk-3.0
-Gladeui_2_0_gir_CFLAGS = $(INCLUDES)
+Gladeui_2_0_gir_CFLAGS = $(INCLUDES) $(GTK_CFLAGS)
 Gladeui_2_0_gir_LIBS = libgladeui-2.la
 Gladeui_2_0_gir_FILES = $(introspection_sources)
 INTROSPECTION_GIRS += Gladeui-2.0.gir
