--- glade-3.16.1/configure.ac.orig	2014-02-24 11:46:43.302800000 +0400
+++ glade-3.16.1/configure.ac	2014-02-24 11:47:07.124000000 +0400
@@ -206,12 +206,6 @@
 AM_CONDITIONAL(BUILD_PYTHON, test x"$have_python" = "xyes")
 
 # ==================================================================
-# Glade User Manual (requires yelp-tools)
-# ==================================================================
-
-YELP_HELP_INIT
-
-# ==================================================================
 # Man pages (requires xsltproc)
 # ==================================================================
 
@@ -337,7 +331,6 @@
 po/Makefile.in
 doc/Makefile
 doc/version.xml
-help/Makefile
 man/Makefile
 tests/Makefile
 build/mingw-w64/glade.nsi
--- glade-3.16.1/Makefile.am.orig	2014-02-24 11:46:43.302800000 +0400
+++ glade-3.16.1/Makefile.am	2014-02-24 11:47:07.124000000 +0400
@@ -1,10 +1,10 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = po data gladeui src plugins tests doc help
+SUBDIRS = po data gladeui src plugins tests doc
 if BUILD_MAN_PAGES
 SUBDIRS += man
 endif
-DIST_SUBDIRS = po data gladeui src plugins tests doc help man
+DIST_SUBDIRS = po data gladeui src plugins tests doc man
 
 EXTRA_DIST = \
 	autogen.sh \
