--- glade-3.18.3/gladeui/glade-previewer.rc.in.orig	2015-01-26 22:54:17.967200000 +0300
+++ glade-3.18.3/gladeui/glade-previewer.rc.in	2015-01-26 22:56:00.094800000 +0300
@@ -1,6 +1,6 @@
 #include <winver.h>                      
 
-GLADE_ICON		ICON		"../data/icons/glade.ico"
+GLADE_ICON		ICON		"@top_srcdir@/data/icons/glade.ico"
 
 VS_VERSION_INFO VERSIONINFO
   FILEVERSION     @GLADE_MAJOR_VERSION@,@GLADE_MINOR_VERSION@,@GLADE_MICRO_VERSION@,0
--- glade-3.18.3/src/glade.rc.in.orig	2015-01-26 23:03:56.444600000 +0300
+++ glade-3.18.3/src/glade.rc.in	2015-01-26 23:04:06.054200000 +0300
@@ -1,6 +1,6 @@
 #include <winver.h>                      
 
-GLADE_ICON		ICON		"../data/icons/glade.ico"
+GLADE_ICON		ICON		"@top_srcdir@/data/icons/glade.ico"
 
 VS_VERSION_INFO VERSIONINFO
   FILEVERSION     @GLADE_MAJOR_VERSION@,@GLADE_MINOR_VERSION@,@GLADE_MICRO_VERSION@,0
