--- nsis-3.03-src/Source/scriptpp.cpp.orig	2018-11-02 08:07:46.395481800 +0300
+++ nsis-3.03-src/Source/scriptpp.cpp	2018-11-02 08:15:51.260933500 +0300
@@ -1051,7 +1051,7 @@
   TCHAR* cmdstr = line.gettoken_str(1);
   int validparams = false;
   struct postbuild_cmd *newcmd, *prevcmd;
-  newcmd = (struct postbuild_cmd*) (new BYTE[FIELD_OFFSET(struct postbuild_cmd, cmd[_tcsclen(cmdstr)+1])]);
+  newcmd = (struct postbuild_cmd*) (new BYTE[FIELD_OFFSET(struct postbuild_cmd, cmd)*(_tcsclen(cmdstr)+1)]);
   newcmd->next = NULL, _tcscpy(newcmd->cmd, cmdstr);
   newcmd->cmpop = line.gettoken_enum(2, _T("<\0>\0<>\0=\0ignore\0")), newcmd->cmpval = line.gettoken_int(3, &validparams);
   if (line.getnumtokens() == 1+1)
