diff -aurN 004/SCons/Config/gnu 005/SCons/Config/gnu
--- 004/SCons/Config/gnu	2015-04-29 18:20:50.150418600 -0300
+++ 005/SCons/Config/gnu	2015-05-03 18:59:14.093116800 -0300
@@ -16,8 +16,8 @@
 
 ### flags
 
-def entry(x,u):
-	if x == 'NSISWinMainNOCRT':
+def entry(x,u,a):
+	if x == 'NSISWinMainNOCRT' and a == 'x86':
 		x = '_' + x
 	elif x == 'DllMain':
 		x = '_DllMain@12'
diff -aurN 004/SConstruct 005/SConstruct
--- 004/SConstruct	2015-04-29 19:29:54.008843900 -0300
+++ 005/SConstruct	2015-05-03 18:58:11.889884700 -0300
@@ -307,7 +307,7 @@
 		unicodestr = "None"
 		if 'UNICODE' in env['CPPDEFINES']:
 			unicodestr = "True"
-		env.Append(LINKFLAGS = ['${ENTRY_FLAG("%s",%s)}' % (entry,unicodestr)])
+		env.Append(LINKFLAGS = ['${ENTRY_FLAG("%s",%s,"%s")}' % (entry,unicodestr,env['TARGET_ARCH'])])
 
 	if nodeflib:
 		env.Append(LINKFLAGS = ['$NODEFLIBS_FLAG']) # no default libraries
