diff -urN nsis-3.08-src-orig/Source/build.cpp nsis-3.08-src/Source/build.cpp
--- nsis-3.08-src-orig/Source/build.cpp	2022-02-10 18:47:42.561286500 +0800
+++ nsis-3.08-src/Source/build.cpp	2022-02-10 19:11:51.278148400 +0800
@@ -4142,7 +4161,7 @@
 
 CEXEBuild::postbuild_cmd* CEXEBuild::postbuild_cmd::make(const TCHAR *cmdstr, int cmpop, int cmpval)
 {
-  postbuild_cmd *p = (postbuild_cmd*) (new BYTE[FIELD_OFFSET(postbuild_cmd, cmd[_tcsclen(cmdstr)+!0])]);
+  postbuild_cmd *p = (postbuild_cmd*) (new BYTE[FIELD_OFFSET(postbuild_cmd, cmd)*_tcsclen(cmdstr)+!0]);
   p->next = NULL, _tcscpy(p->cmd, cmdstr);
   p->cmpop = cmpop, p->cmpval = cmpval;
   return p;
