diff -urN nsis-3.08-src-orig/Source/build.cpp nsis-3.08-src/Source/build.cpp
--- nsis-3.08-src-orig/Source/build.cpp	2022-02-10 18:47:42.561286500 +0800
+++ nsis-3.08-src/Source/build.cpp	2022-02-12 02:02:21.759035000 +0800
@@ -3654,7 +3656,7 @@
     CharEncConv cec;
     if (!g_outputenc.IsUTF16LE())
     {
-      size_t cbConv;
+      size_t cbConv = 0;
       if (!cec.Initialize(g_outputenc.GetCodepage(), -1) || !(data = (const TCHAR*) cec.Convert(data, cb, &cbConv)))
         PrintColorFmtMsg_ERR(_T("conversion failed!\n")), quit(); // Cannot use ERROR_MSG() here!
       cb = (DWORD) (cbConv + NStreamEncoding::GetCodeUnitSize(g_outputenc.GetCodepage())); // cbConv does not include the \0.
diff -urN nsis-3.08-src-orig/Source/ResourceEditor.cpp nsis-3.08-src/Source/ResourceEditor.cpp
--- nsis-3.08-src-orig/Source/ResourceEditor.cpp	2022-02-10 18:47:42.686293600 +0800
+++ nsis-3.08-src/Source/ResourceEditor.cpp	2022-02-11 15:43:15.537108900 +0800
@@ -913,7 +913,7 @@
   }
 
   FILEVIEW &map = *(FILEVIEW*) &X.Map;
-  size_t datasize;
+  size_t datasize = 0;
   char *filedata = create_file_view_readonly(File, map), *data = 0, *dataalloc = 0;
   if (filedata) {
     if (resproto) {
diff -urN nsis-3.08-src-orig/Source/util.h nsis-3.08-src/Source/util.h
--- nsis-3.08-src-orig/Source/util.h	2022-02-10 18:47:32.341702000 +0800
+++ nsis-3.08-src/Source/util.h	2022-02-11 08:17:01.592693500 +0800
@@ -131,7 +131,8 @@
   friend int fseek(S_t*s, long int offset, int origin)
   {
     if ((unsigned long) offset != (size_t) offset) return 1; // long int will usually fit in our size_t
-    size_t newpos, invalid = 0;
+    size_t newpos = 0;
+    size_t invalid = 0;
     switch(origin)
     {
     case SEEK_SET: newpos = (size_t) offset, invalid = offset < 0; break;
