diff -urN nsis-3.08-src-orig/Contrib/System/Source/Call-amd64.S nsis-3.08-src/Contrib/System/Source/Call-amd64.S
--- nsis-3.08-src-orig/Contrib/System/Source/Call-amd64.S	2021-01-02 04:27:52.000000000 +0800
+++ nsis-3.08-src/Contrib/System/Source/Call-amd64.S	2022-02-17 18:52:01.030814500 +0800
@@ -51,11 +51,9 @@
 
 .macro FUNC_DECL name
 .global \name
-.func   \name
 \name:
 .endm
 .macro FUNC_END name
-.endfunc
 .endm
 
 ;# ~GNU
diff -urN nsis-3.08-src-orig/Contrib/System/Source/Call.S nsis-3.08-src/Contrib/System/Source/Call.S
--- nsis-3.08-src-orig/Contrib/System/Source/Call.S	2021-01-02 04:27:52.000000000 +0800
+++ nsis-3.08-src/Contrib/System/Source/Call.S	2022-02-17 19:29:19.591852800 +0800
@@ -118,12 +118,10 @@
 
 .macro FUNC_DECL name
 .global \name
-.func   \name
 \name:
 .endm
 
 .macro FUNC_END name
-.endfunc
 .endm
 
 ;# /*
