diff -ur a/Contrib/System/Source/Call-amd64.S b/Contrib/System/Source/Call-amd64.S
--- a/Contrib/System/Source/Call-amd64.S	2023-01-26 21:51:19.076000000 +0800
+++ b/Contrib/System/Source/Call-amd64.S	2024-05-24 23:31:03.951271500 +0800
@@ -51,11 +51,9 @@
 
 .macro FUNC_DECL name
 .global \name
-.func   \name
 \name:
 .endm
 .macro FUNC_END name
-.endfunc
 .endm
 
 ;# ~GNU
diff -ur a/Contrib/System/Source/Call.S b/Contrib/System/Source/Call.S
--- a/Contrib/System/Source/Call.S	2023-01-26 21:51:19.000000000 +0800
+++ b/Contrib/System/Source/Call.S	2024-05-24 23:31:04.013780200 +0800
@@ -118,12 +118,10 @@
 
 .macro FUNC_DECL name
 .global \name
-.func   \name
 \name:
 .endm
 
 .macro FUNC_END name
-.endfunc
 .endm
 
 ;# /*
