--- nsis-3.11-src/Contrib/Makensisw/resource.rc.orig	2024-06-12 13:03:58 +0200
+++ nsis-3.11-src/Contrib/Makensisw/resource.rc	2025-04-24 22:59:33 +0200
@@ -63,35 +63,35 @@
 // Menu
 //
 
-IDM_MENU MENUEX
+IDM_MENU MENU
 BEGIN
-    POPUP "&File", IDM_FILE
+    POPUP "&File"
     BEGIN
         MENUITEM "&Save Output...",             IDM_SAVE
         MENUITEM "&Load Script...\tCtrl+L",     IDM_LOADSCRIPT
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "E&xit\tAlt+X",                IDM_EXIT
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "",                            IDM_MRU_FILE
     END
-    POPUP "&Edit", IDM_EDIT
+    POPUP "&Edit"
     BEGIN
         MENUITEM "&Copy\tCtrl+C",               IDM_COPYSELECTED
         MENUITEM "Copy &All",                   IDM_COPYALL
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "&Find...\tCtrl+F",            IDM_FIND
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "&Select All\tCtrl+A",         IDM_SELECTALL
         MENUITEM "Clear &Log Window\tCtrl+W",   IDM_CLEARLOG
     END
-    POPUP "&Script", IDM_SCRIPT
+    POPUP "&Script"
     BEGIN
         MENUITEM "&Recompile\tCtrl+R",          IDM_RECOMPILE
         MENUITEM "Test &Installer\tCtrl+T",     IDM_TEST
         MENUITEM "Reco&mpile and Test\tCtrl+M", IDM_RECOMPILE_TEST
         MENUITEM "C&ancel compilation\tCtrl+K", IDM_CANCEL
-        MENUITEM "", -1, MFT_SEPARATOR
-        POPUP "Set &Compressor", IDM_COMPRESSOR_SUBMENU
+        MENUITEM SEPARATOR
+        POPUP "Set &Compressor"
         BEGIN
             MENUITEM "&Defined in Script/Compiler Default", IDM_COMPRESSOR_SCRIPT
             MENUITEM "&ZLIB",                       IDM_ZLIB
@@ -102,30 +102,30 @@
             MENUITEM "LZMA (solid)",                IDM_LZMA_SOLID
             MENUITEM "&Best Compressor",            IDM_BEST
         END
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "&Edit Script\tCtrl+E",        IDM_EDITSCRIPT
         MENUITEM "Open Script &Folder\tCtrl+B", IDM_BROWSESCR
     END
-    POPUP "&Tools", IDM_TOOLS
+    POPUP "&Tools"
     BEGIN
         MENUITEM "&Settings\tCtrl+S",         IDM_SETTINGS
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "&Window Info",                IDM_WNDSPY
         MENUITEM "&Lookup",                     IDM_LOOKUP
         MENUITEM "Generate &GUID",              IDM_GUIDGEN
         MENUITEM "&Add/Remove Programs",        IDM_ARP
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "Clear Recent &Files List",    IDM_CLEAR_MRU_LIST
     END
-    POPUP "&Help", IDM_HELP
+    POPUP "&Help"
     BEGIN
         MENUITEM "NSIS Users &Manual\tF1",      IDM_DOCS
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "Check for &Updates",          IDM_NSISUPDATE
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "NSIS &Homepage",              IDM_NSISHOME
         MENUITEM "NSIS &Forum",                 IDM_FORUM
-        MENUITEM "", -1, MFT_SEPARATOR
+        MENUITEM SEPARATOR
         MENUITEM "&About MakeNSISW",            IDM_ABOUT
     END
 END
@@ -173,7 +173,6 @@
     WS_MAXIMIZEBOX | WS_POPUP | WS_VISIBLE | WS_CLIPCHILDREN | WS_CAPTION | 
     WS_SYSMENU | WS_THICKFRAME
 CAPTION "MakeNSISW"
-MENU IDM_MENU
 FONT 8, "MS Shell Dlg"
 BEGIN
     CONTROL         "",IDC_LOGWIN,RICHEDIT_CLASS,ES_MULTILINE | ES_AUTOVSCROLL | 
