--- nsis-3.11-src/Source/Tests/SConscript.orig	2025-04-24 23:16:32 +0200
+++ nsis-3.11-src/Source/Tests/SConscript	2025-04-24 23:24:57 +0200
@@ -141,7 +141,7 @@
 
 	# build test program
 	tests = env.Program(target, tests + required_obj)
-	if env['PLATFORM'] == 'win32' and 'ZLIB_W32_DLL' in env:
+	if env['PLATFORM'] == 'win32' and 'ZLIB_W32_DLL' in env and env['ZLIB_W32_DLL']:
 		import os.path
 		env.Depends(tests, env.InstallAs(
 			os.path.basename(str(env['ZLIB_W32_DLL'])), 
