diff --git a/CMakeLists.txt b/CMakeLists.txt
index 50dd412e..e1ad4c86 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,6 +66,8 @@ endfunction()
 
 # Set install parameters
 get_all_targets(targets .)
+include(GNUInstallDirs)
 install(TARGETS ${targets}
-		RUNTIME DESTINATION ${PROJECT_NAME}
-		LIBRARY DESTINATION ${PROJECT_NAME})
+		RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+		LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
