diff --git a/libs/compiler/codegen.c b/libs/compiler/codegen.c
index e7b41670..ed7eba65 100644
--- a/libs/compiler/codegen.c
+++ b/libs/compiler/codegen.c
@@ -226,7 +226,7 @@ static int print_table(const encoder *const enc, const vector *const table)
  */
 static int enc_export(const encoder *const enc)
 {
-	uni_printf(enc->sx->io, "#!/usr/bin/ruc-vm\n");
+	uni_printf(enc->sx->io, "#!/usr/bin/env ruc-vm\n");
 
 	uni_printf(enc->sx->io, "%zi %zi %zi %zi %zi %" PRIitem " 0\n"
 		, vector_size(&enc->memory)
