diff -bur qqc2-desktop-style-6.2.0-orig/kirigami-plasmadesktop-integration/animationspeedprovider.cpp qqc2-desktop-style-6.2.0/kirigami-plasmadesktop-integration/animationspeedprovider.cpp
--- qqc2-desktop-style-6.2.0-orig/kirigami-plasmadesktop-integration/animationspeedprovider.cpp	2024-06-08 15:07:33.121344800 -0600
+++ qqc2-desktop-style-6.2.0/kirigami-plasmadesktop-integration/animationspeedprovider.cpp	2024-06-08 15:09:35.319028000 -0600
@@ -67,7 +67,7 @@
 
 void WindowsAnimationSpeedProvider::update()
 {
-    bool isAnimated = true;
+    BOOL isAnimated = TRUE;
     if (SystemParametersInfoW(SPI_GETCLIENTAREAANIMATION, 0, &isAnimated, 0)) {
         m_animationSpeedModifier = isAnimated ? 1.0 : 0.0;
     }
