diff -Naur graphviz-2.40.1-orig/lib/edgepaint/lab_gamut.def graphviz-2.40.1/lib/edgepaint/lab_gamut.def
--- graphviz-2.40.1-orig/lib/edgepaint/lab_gamut.def	2018-01-26 21:42:07.550918600 +0300
+++ graphviz-2.40.1/lib/edgepaint/lab_gamut.def	2018-01-26 21:54:31.849490100 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"lab_gamut"
+LIBRARY	"liblab_gamut-1"
 
 
 EXPORTS
diff -Naur graphviz-2.40.1-orig/lib/gvc.def graphviz-2.40.1/lib/gvc.def
--- graphviz-2.40.1-orig/lib/gvc.def	2018-01-26 21:43:31.455717700 +0300
+++ graphviz-2.40.1/lib/gvc.def	2018-01-26 21:54:41.418037300 +0300
@@ -1,4 +1,4 @@
-LIBRARY    "gvc"
+LIBRARY    "libgvc-6"
 
 EXPORTS
 add_box    
diff -Naur graphviz-2.40.1-orig/lib/pathplan/pathplan.def graphviz-2.40.1/lib/pathplan/pathplan.def
--- graphviz-2.40.1-orig/lib/pathplan/pathplan.def	2018-01-26 21:42:10.411082200 +0300
+++ graphviz-2.40.1/lib/pathplan/pathplan.def	2018-01-26 21:49:37.814672200 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"Pathplan"
+LIBRARY	"libpathplan-4"
 
 
 EXPORTS
