--- a/src/libs/CMakeLists.txt
+++ b/src/libs/CMakeLists.txt
@@ -97,7 +97,7 @@
   add_library(midi MODULE "tools/midi.c")
   target_link_libraries (midi ${PORTMIDI_LIBRARY})
 endif()
-
+#[[
 find_package(SDL2)
 if(SDL2_INCLUDE_DIRS)
  add_definitions("-DHAVE_SDL")
@@ -106,7 +106,7 @@
   add_library(gamepad MODULE "tools/gamepad.c")
   target_link_libraries(gamepad ${SDL2_LIBRARIES})
 endif()
-
+]]
 if(BUILD_BATTERY_INDICATOR)
   add_library(battery_indicator MODULE "tools/battery_indicator.c")
 endif()
