--- a/src/external/LibRaw/libraw/libraw.h
+++ b/src/external/LibRaw/libraw/libraw.h
@@ -71,6 +71,10 @@ it under the terms of the one of two licenses as you choose:
 #  ifndef LIBRAW_WIN32_UNICODEPATHS
 #    define LIBRAW_WIN32_UNICODEPATHS
 #  endif
+# elif defined(_LIBCPP_HAS_OPEN_WITH_WCHAR)
+#  ifndef LIBRAW_WIN32_UNICODEPATHS
+#    define LIBRAW_WIN32_UNICODEPATHS
+#  endif
 # endif
 
 #endif
--- a/src/external/LibRaw/src/utils/open.cpp
+++ b/src/external/LibRaw/src/utils/open.cpp
@@ -178,7 +178,7 @@ int LibRaw::open_file(const char *fname)
 
 #if defined(WIN32) || defined(_WIN32)
 #ifndef LIBRAW_WIN32_UNICODEPATHS
-int LibRaw::open_file(const wchar_t *, INT64)
+int LibRaw::open_file(const wchar_t *)
 {
     return LIBRAW_NOT_IMPLEMENTED;
 }
