diff -Nur darktable-3.2.1/data/CMakeLists.txt darktable-3.2.1-new/data/CMakeLists.txt
--- darktable-3.2.1/data/CMakeLists.txt	2020-08-04 13:51:32.000000000 +0200
+++ darktable-3.2.1-new/data/CMakeLists.txt	2020-11-13 09:02:26.203082400 +0100
@@ -38,7 +38,7 @@
 add_custom_command(
 		OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop
 		SOURCE ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop.in
-		COMMAND ${intltool_merge_BIN} --desktop-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop
+		COMMAND ${perl_BIN} ${intltool_merge_BIN} --desktop-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop
 		MAIN_DEPENDENCY ${CMAKE_CURRENT_BINARY_DIR}/darktable.desktop.in
 		DEPENDS ${PO_FILES}
 )
@@ -60,7 +60,7 @@
   add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
     SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in
-    COMMAND ${intltool_merge_BIN} --xml-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
+    COMMAND ${perl_BIN} ${intltool_merge_BIN} --xml-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
     COMMAND ${appstream_util_BIN} validate --nonet ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
     MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in
     DEPENDS ${PO_FILES}
@@ -69,7 +69,7 @@
   add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
     SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in
-    COMMAND ${intltool_merge_BIN} --xml-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
+    COMMAND ${perl_BIN} ${intltool_merge_BIN} --xml-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in ${CMAKE_CURRENT_BINARY_DIR}/darktable.appdata.xml
     MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/darktable.appdata.xml.in
     DEPENDS ${PO_FILES}
   )
diff -Nur darktable-3.2.1/doc/man/CMakeLists.txt darktable-3.2.1-new/doc/man/CMakeLists.txt
--- darktable-3.2.1/doc/man/CMakeLists.txt	2020-08-04 13:51:32.000000000 +0200
+++ darktable-3.2.1-new/doc/man/CMakeLists.txt	2020-11-13 09:03:43.058556600 +0100
@@ -57,7 +57,7 @@
       # create translated .pod
       add_custom_command(
         OUTPUT ${tmppod}
-        COMMAND ${po4a-translate_BIN} -f pod -m ${pod} -p ${pofile} -l ${tmppod} -k 0
+        COMMAND ${perl_BIN} ${po4a-translate_BIN} -f pod -m ${pod} -p ${pofile} -l ${tmppod} -k 0
         DEPENDS ${pofile} ${pod}
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
       )
diff -Nur darktable-3.2.1/doc/man/po/CMakeLists.txt darktable-3.2.1-new/doc/man/po/CMakeLists.txt
--- darktable-3.2.1/doc/man/po/CMakeLists.txt	2020-08-04 13:51:32.000000000 +0200
+++ darktable-3.2.1-new/doc/man/po/CMakeLists.txt	2020-11-13 09:03:14.459722300 +0100
@@ -19,7 +19,7 @@
     set(pofile "${language}.po")
     add_custom_command(
       OUTPUT ${pofile}
-      COMMAND ${po4a-updatepo_BIN} -f pod ${files} -p po/${pofile}
+      COMMAND ${perl_BIN} ${po4a-updatepo_BIN} -f pod ${files} -p po/${pofile}
       DEPENDS ${deps}
       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../
     )
--- darktable-3.2.1/doc/man/CMakeLists.txt.orig	2020-11-13 09:21:48.994876000 +0100
+++ darktable-3.2.1/doc/man/CMakeLists.txt	2020-11-13 09:21:05.400853600 +0100
@@ -25,7 +25,7 @@
   set(target ${CMAKE_CURRENT_BINARY_DIR}/${manpage})
   add_custom_command(
     OUTPUT ${target}
-    COMMAND ${CMAKE_SOURCE_DIR}/tools/makeman.sh ${source} ${CMAKE_CURRENT_SOURCE_DIR}/../../AUTHORS ${target}
+    COMMAND bash ${CMAKE_SOURCE_DIR}/tools/makeman.sh ${source} ${CMAKE_CURRENT_SOURCE_DIR}/../../AUTHORS ${target}
     DEPENDS ${source} ${CMAKE_CURRENT_SOURCE_DIR}/../../AUTHORS ${CMAKE_SOURCE_DIR}/tools/makeman.sh
   )
   add_custom_target(manpage_${manpage} ALL DEPENDS ${target})
@@ -64,7 +64,7 @@
       # turn the .pod into a manpage
       add_custom_command(
         OUTPUT ${target}
-        COMMAND ${CMAKE_SOURCE_DIR}/tools/makeman.sh ${tmppod} ${CMAKE_CURRENT_SOURCE_DIR}/../../AUTHORS ${target}
+        COMMAND bash ${CMAKE_SOURCE_DIR}/tools/makeman.sh ${tmppod} ${CMAKE_CURRENT_SOURCE_DIR}/../../AUTHORS ${target}
         DEPENDS ${tmppod} ${CMAKE_CURRENT_SOURCE_DIR}/../../AUTHORS ${CMAKE_SOURCE_DIR}/tools/makeman.sh
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
       )
