--- glib-networking-2.42.0/configure.ac.orig	2014-09-23 15:51:38.068600000 +0500
+++ glib-networking-2.42.0/configure.ac	2014-09-23 15:52:46.084600000 +0500
@@ -37,6 +37,7 @@
 GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40"
 
 GIO_MODULE_DIR=$($PKG_CONFIG --variable giomoduledir gio-2.0)
+GIO_MODULE_DIR=`cygpath -u $GIO_MODULE_DIR`
 AS_IF([test "$GIO_MODULE_DIR" = ""],
       [AC_MSG_FAILURE(GIO_MODULE_DIR is missing from gio-2.0.pc)])
 AC_SUBST(GIO_MODULE_DIR)
