--- glib-networking-2.32.0/configure.ac.orig	2012-03-26 23:07:26 +0400
+++ glib-networking-2.32.0/configure.ac	2012-04-09 18:53:11 +0400
@@ -35,7 +35,7 @@
 AM_PATH_GLIB_2_0(2.39.1,,AC_MSG_ERROR(GLIB not found),gio)
 GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40"
 
-GIO_MODULE_DIR=$($PKG_CONFIG --variable giomoduledir gio-2.0)
+GIO_MODULE_DIR=$($PKG_CONFIG --define-variable=prefix= --variable giomoduledir gio-2.0)
 AS_IF([test "$GIO_MODULE_DIR" = ""],
       [AC_MSG_FAILURE(GIO_MODULE_DIR is missing from gio-2.0.pc)])
 AC_SUBST(GIO_MODULE_DIR)
