From 1139d143b1fdafc2685ad0568b2dd0cb04a52794 Mon Sep 17 00:00:00 2001
From: Cyril Arnould <cyril.arnould@outlook.com>
Date: Tue, 13 Jun 2023 23:46:27 +0200
Subject: [PATCH 07/17] Modify tty_stream names

Adapted from ezwinports
---
 xargs/xargs.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/xargs/xargs.c b/xargs/xargs.c
index 16e828ff..94b512e8 100644
--- a/xargs/xargs.c
+++ b/xargs/xargs.c
@@ -1156,7 +1156,11 @@ print_args (bool ask)
 
       if (!tty_stream)
 	{
+#if defined(__MINGW32__) || defined(_MSC_VER)
+	  tty_stream = fopen_cloexec_for_read_only ("CON");
+#else
 	  tty_stream = fopen_cloexec_for_read_only ("/dev/tty");
+#endif
 	  if (!tty_stream)
 	    die (EXIT_FAILURE, errno,
 		 _("failed to open /dev/tty for reading"));
@@ -1235,7 +1239,11 @@ prep_child_for_exec (void)
   if (!keep_stdin || open_tty)
     {
       int fd;
+#if defined(__MINGW32__) || defined(_MSC_VER)
+      const char *inputfile = open_tty ? "CON" : "NUL";
+#else
       const char *inputfile = open_tty ? "/dev/tty" : "/dev/null";
+#endif
 
       close (0);
       if ((fd = open (inputfile, O_RDONLY)) < 0)
-- 
2.41.0

