From 8f22e357d0f99b26e2ca599db555c2fea71d0d1f Mon Sep 17 00:00:00 2001
From: Cyril Arnould <cyril.arnould@outlook.com>
Date: Wed, 14 Jun 2023 00:11:24 +0200
Subject: [PATCH 14/17] Require signals to be defined

---
 find/exec.c   |  2 ++
 xargs/xargs.c | 10 ++++++++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/find/exec.c b/find/exec.c
index 6c2a179c..36a1e932 100644
--- a/find/exec.c
+++ b/find/exec.c
@@ -394,7 +394,9 @@ launch (struct buildcmd_control *ctl, void *usercontext, int argc, char **argv)
   if (first_time)
     {
       first_time = 0;
+#ifdef SIGCHLD
       signal (SIGCHLD, SIG_DFL);
+#endif
     }
   
 #if defined(__MINGW32__) || defined(_MSC_VER) || defined(__MSDOS__)
diff --git a/xargs/xargs.c b/xargs/xargs.c
index 72a51f88..c29b43be 100644
--- a/xargs/xargs.c
+++ b/xargs/xargs.c
@@ -421,8 +421,12 @@ main (int argc, char **argv)
   void (*act_on_init_result)(void) = noop;
   enum BC_INIT_STATUS bcstatus;
   enum { XARGS_POSIX_HEADROOM = 2048u };
+#ifdef SIGUSR1
+# ifdef SIGUSR2
   struct sigaction sigact;
-
+# endif /* SIGUSR2 */
+#endif /* SIGUSR1 */
+  
   /* We #define __STDC_LIMIT_MACROS above for its side effect on
    * <limits.h>, but we use it here to avoid getting what would
    * otherwise be a spurious compiler warning. */
@@ -827,8 +831,10 @@ main (int argc, char **argv)
   bc_state.argbuf = xmalloc (bc_ctl.arg_max + 1);
 
   /* Make sure to listen for the kids.  */
+#ifdef SIGCHLD
   signal (SIGCHLD, SIG_DFL);
-
+#endif
+  
   if (!bc_ctl.replace_pat)
     {
       for (; optind < argc; optind++)
-- 
2.41.0

