--- a/lib/regextype.c
+++ b/lib/regextype.c
@@ -49,15 +49,15 @@
   {
    { "findutils-default",     CONTEXT_FINDUTILS, RE_SYNTAX_EMACS|RE_DOT_NEWLINE  },
    { "ed",                    CONTEXT_GENERIC,   RE_SYNTAX_ED                    },
-   { "emacs",                 CONTEXT_ALL,       RE_SYNTAX_EMACS                 },
-   { "gnu-awk",               CONTEXT_ALL,       RE_SYNTAX_GNU_AWK               },
-   { "grep",                  CONTEXT_ALL,       RE_SYNTAX_GREP                  },
-   { "posix-awk",             CONTEXT_ALL,       RE_SYNTAX_POSIX_AWK             },
-   { "awk",                   CONTEXT_ALL,       RE_SYNTAX_AWK                   },
-   { "posix-basic",           CONTEXT_ALL,       RE_SYNTAX_POSIX_BASIC           },
-   { "posix-egrep",           CONTEXT_ALL,       RE_SYNTAX_POSIX_EGREP           },
-   { "egrep",                 CONTEXT_ALL,       RE_SYNTAX_EGREP                 },
-   { "posix-extended",        CONTEXT_ALL,       RE_SYNTAX_POSIX_EXTENDED        },
+   { "emacs",                 CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_EMACS                 },
+   { "gnu-awk",               CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_GNU_AWK               },
+   { "grep",                  CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_GREP                  },
+   { "posix-awk",             CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_POSIX_AWK             },
+   { "awk",                   CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_AWK                   },
+   { "posix-basic",           CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_POSIX_BASIC           },
+   { "posix-egrep",           CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_POSIX_EGREP           },
+   { "egrep",                 CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_EGREP                 },
+   { "posix-extended",        CONTEXT_FINDUTILS_OR_GENERIC,       RE_SYNTAX_POSIX_EXTENDED        },
    { "posix-minimal-basic",   CONTEXT_GENERIC,   RE_SYNTAX_POSIX_MINIMAL_BASIC   },
    { "sed",                   CONTEXT_GENERIC,   RE_SYNTAX_SED                   },
    /*    ,{ "posix-common",   CONTEXT_GENERIC,  _RE_SYNTAX_POSIX_COMMON   } */
--- a/lib/regextype.h
+++ b/lib/regextype.h
@@ -29,7 +29,7 @@
 enum {
   CONTEXT_FINDUTILS = 1u,
   CONTEXT_GENERIC   = 2u,
-  CONTEXT_ALL = CONTEXT_GENERIC|CONTEXT_FINDUTILS,
+  CONTEXT_FINDUTILS_OR_GENERIC = CONTEXT_GENERIC|CONTEXT_FINDUTILS,
 };
 
 
@@ -52,7 +52,7 @@
  */
 int get_regex_type_synonym(unsigned int ix, unsigned int context);
 
-/* Returns one of CONTEXT_FINDUTILS, CONTEXT_GENERIC or CONTEXT_ALL.
+/* Returns one of CONTEXT_FINDUTILS, CONTEXT_GENERIC or CONTEXT_FINDUTILS_OR_GENERIC.
  * This identifies whether this regular expression type index is relevant for,
  * respectively, findutils, general callers, or all callers.
  */
