From dab04e936fd8c195843ac0b4a40201c46142da4a Mon Sep 17 00:00:00 2001
From: Junjie Mao <eternal.n08@gmail.com>
Date: Sun, 5 Apr 2015 16:46:03 +0800
Subject: [PATCH] Guess the libraries are located at lib/coq/ instead of lib/

Signed-off-by: Junjie Mao <eternal.n08@gmail.com>
---
 lib/envars.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/envars.ml b/lib/envars.ml
index 3040dd4..3a25e9c 100644
--- a/lib/envars.ml
+++ b/lib/envars.ml
@@ -34,7 +34,7 @@ let reldir instdir testfile oth =
 
 let guess_coqlib () =
   let file = "states/initial.coq" in
-    reldir (if Coq_config.arch = "win32" then ["lib"] else ["lib";"coq"]) file
+    reldir (if Coq_config.arch = "win32" then ["lib";"coq"] else ["lib";"coq"]) file
       (fun () ->
         let coqlib = match Coq_config.coqlib with
           | Some coqlib -> coqlib
-- 
2.3.4

