--- a/src/core/processing/models/qgsprocessingmodelalgorithm.cpp
+++ b/src/core/processing/models/qgsprocessingmodelalgorithm.cpp
@@ -1117,7 +1117,7 @@
 
   auto safeName = []( const QString &name ) -> QString {
     QString s = name;
-    const thread_local QRegularExpression safeNameRe( u"[\\s'\"\\(\\):\\.]"_s );
+    const QRegularExpression safeNameRe( u"[\\s'\"\\(\\):\\.]"_s );
     return s.replace( safeNameRe, u"_"_s );
   };
 
--- a/src/core/processing/qgsprocessingalgorithm.cpp
+++ b/src/core/processing/qgsprocessingalgorithm.cpp
@@ -341,7 +341,7 @@
 
   auto escapeIfNeeded = []( const QString &input ) -> QString {
     // play it safe and escape everything UNLESS it's purely alphanumeric characters (and a very select scattering of other common characters!)
-    const thread_local QRegularExpression nonAlphaNumericRx( u"[^a-zA-Z0-9.\\-/_]"_s );
+    const QRegularExpression nonAlphaNumericRx( u"[^a-zA-Z0-9.\\-/_]"_s );
     if ( nonAlphaNumericRx.match( input ).hasMatch() )
     {
       QString escaped = input;
--- a/src/core/processing/qgsprocessingcontext.cpp
+++ b/src/core/processing/qgsprocessingcontext.cpp
@@ -209,7 +209,7 @@
 {
   auto escapeIfNeeded = []( const QString &input ) -> QString {
     // play it safe and escape everything UNLESS it's purely alphanumeric characters (and a very select scattering of other common characters!)
-    const thread_local QRegularExpression nonAlphaNumericRx( u"[^a-zA-Z0-9.\\-/_]"_s );
+    const QRegularExpression nonAlphaNumericRx( u"[^a-zA-Z0-9.\\-/_]"_s );
     if ( nonAlphaNumericRx.match( input ).hasMatch() )
     {
       QString escaped = input;
--- a/src/core/qgsogrutils.cpp
+++ b/src/core/qgsogrutils.cpp
@@ -1601,7 +1601,7 @@
   const QVariantMap styles = parseStyleString( string );
 
   auto convertSize = []( const QString &size, double &value, Qgis::RenderUnit &unit ) -> bool {
-    const thread_local QRegularExpression sUnitRx = QRegularExpression( u"^([\\d\\.]+)(g|px|pt|mm|cm|in)$"_s );
+    const QRegularExpression sUnitRx = QRegularExpression( u"^([\\d\\.]+)(g|px|pt|mm|cm|in)$"_s );
     const QRegularExpressionMatch match = sUnitRx.match( size );
     if ( match.hasMatch() )
     {
@@ -1656,7 +1656,7 @@
     if ( string.isEmpty() )
       return QColor();
 
-    const thread_local QRegularExpression sColorWithAlphaRx = QRegularExpression( u"^#([0-9a-fA-F]{6})([0-9a-fA-F]{2})$"_s );
+    const QRegularExpression sColorWithAlphaRx = QRegularExpression( u"^#([0-9a-fA-F]{6})([0-9a-fA-F]{2})$"_s );
     const QRegularExpressionMatch match = sColorWithAlphaRx.match( string );
     if ( match.hasMatch() )
     {
@@ -1677,7 +1677,7 @@
     convertSize( lineStyle.value( u"w"_s ).toString(), lineWidth, lineWidthUnit );
 
     // if the pen is a mapinfo pen, use dedicated converter for more accurate results
-    const thread_local QRegularExpression sMapInfoId = QRegularExpression( u"mapinfo-pen-(\\d+)"_s );
+    const QRegularExpression sMapInfoId = QRegularExpression( u"mapinfo-pen-(\\d+)"_s );
     const QRegularExpressionMatch match = sMapInfoId.match( string );
     if ( match.hasMatch() )
     {
@@ -1695,7 +1695,7 @@
     const QString pattern = lineStyle.value( u"p"_s ).toString();
     if ( !pattern.isEmpty() )
     {
-      const thread_local QRegularExpression sPatternUnitRx = QRegularExpression( u"^([\\d\\.\\s]+)(g|px|pt|mm|cm|in)$"_s );
+      const QRegularExpression sPatternUnitRx = QRegularExpression( u"^([\\d\\.\\s]+)(g|px|pt|mm|cm|in)$"_s );
       const QRegularExpressionMatch match = sPatternUnitRx.match( pattern );
       if ( match.hasMatch() )
       {
@@ -1775,7 +1775,7 @@
     const QString id = brushStyle.value( u"id"_s ).toString();
 
     // if the pen is a mapinfo brush, use dedicated converter for more accurate results
-    const thread_local QRegularExpression sMapInfoId = QRegularExpression( u"mapinfo-brush-(\\d+)"_s );
+    const QRegularExpression sMapInfoId = QRegularExpression( u"mapinfo-brush-(\\d+)"_s );
     const QRegularExpressionMatch match = sMapInfoId.match( id );
     if ( match.hasMatch() )
     {
@@ -1786,7 +1786,7 @@
         return res;
     }
 
-    const thread_local QRegularExpression sOgrId = QRegularExpression( u"ogr-brush-(\\d+)"_s );
+    const QRegularExpression sOgrId = QRegularExpression( u"ogr-brush-(\\d+)"_s );
     const QRegularExpressionMatch ogrMatch = sOgrId.match( id );
 
     Qt::BrushStyle style = Qt::SolidPattern;
@@ -1863,7 +1863,7 @@
     const QString id = symbolStyle.value( u"id"_s ).toString();
 
     // if the symbol is a mapinfo symbol, use dedicated converter for more accurate results
-    const thread_local QRegularExpression sMapInfoId = QRegularExpression( u"mapinfo-sym-(\\d+)"_s );
+    const QRegularExpression sMapInfoId = QRegularExpression( u"mapinfo-sym-(\\d+)"_s );
     const QRegularExpressionMatch match = sMapInfoId.match( id );
     if ( match.hasMatch() )
     {
@@ -1880,7 +1880,7 @@
 
     std::unique_ptr< QgsMarkerSymbolLayer > markerLayer;
 
-    const thread_local QRegularExpression sFontId = QRegularExpression( u"font-sym-(\\d+)"_s );
+    const QRegularExpression sFontId = QRegularExpression( u"font-sym-(\\d+)"_s );
     const QRegularExpressionMatch fontMatch = sFontId.match( id );
     if ( fontMatch.hasMatch() )
     {
@@ -1933,7 +1933,7 @@
 
     if ( !markerLayer )
     {
-      const thread_local QRegularExpression sOgrId = QRegularExpression( u"ogr-sym-(\\d+)"_s );
+      const QRegularExpression sOgrId = QRegularExpression( u"ogr-sym-(\\d+)"_s );
       const QRegularExpressionMatch ogrMatch = sOgrId.match( id );
 
       Qgis::MarkerShape shape;
--- a/src/core/textrenderer/qgsfontmanager.cpp
+++ b/src/core/textrenderer/qgsfontmanager.cpp
@@ -57,8 +57,8 @@
 
 QString QgsFontDownloadDetails::standardizeFamily( const QString &family )
 {
-  const thread_local QRegularExpression charsToRemove( u"[^a-z]"_s );
-  const thread_local QRegularExpression styleNames( u"(?:normal|regular|light|bold|black|demi|italic|oblique|medium|thin)"_s );
+  const QRegularExpression charsToRemove( u"[^a-z]"_s );
+  const QRegularExpression styleNames( u"(?:normal|regular|light|bold|black|demi|italic|oblique|medium|thin)"_s );
 
   QString processed = family.toLower();
   processed.replace( styleNames, QString() );
@@ -3252,7 +3252,7 @@
       if ( contentDispositionFilename.isEmpty() )
       {
         const QUrl originalUrl = details.fontUrls().value( i );
-        const thread_local QRegularExpression rxExtension( u"^.*\\.(\\w+?)$"_s );
+        const QRegularExpression rxExtension( u"^.*\\.(\\w+?)$"_s );
         extension = rxExtension.match( originalUrl.toString() ).captured( 1 );
       }
       QString thisLicenseDetails;
--- a/src/gui/editorwidgets/qgshtmlwidgetwrapper.cpp
+++ b/src/gui/editorwidgets/qgshtmlwidgetwrapper.cpp
@@ -113,7 +113,7 @@
   mHtmlCode = htmlCode;
 
   bool ok = false;
-  const thread_local QRegularExpression expRe( QStringLiteral( R"re(expression.evaluate\s*\(\s*"(.*)"\))re" ), QRegularExpression::PatternOption::MultilineOption | QRegularExpression::PatternOption::DotMatchesEverythingOption );
+  const QRegularExpression expRe( QStringLiteral( R"re(expression.evaluate\s*\(\s*"(.*)"\))re" ), QRegularExpression::PatternOption::MultilineOption | QRegularExpression::PatternOption::DotMatchesEverythingOption );
   QRegularExpressionMatchIterator matchIt = expRe.globalMatch( mHtmlCode );
   while ( !ok && matchIt.hasNext() )
   {
--- a/src/gui/editorwidgets/qgstextwidgetwrapper.cpp
+++ b/src/gui/editorwidgets/qgstextwidgetwrapper.cpp
@@ -97,7 +97,7 @@
   mText = text;
 
   bool ok = false;
-  const thread_local QRegularExpression sRegEx( u"\\[%(.*?)%\\]"_s, QRegularExpression::MultilineOption | QRegularExpression::DotMatchesEverythingOption );
+  const QRegularExpression sRegEx( u"\\[%(.*?)%\\]"_s, QRegularExpression::MultilineOption | QRegularExpression::DotMatchesEverythingOption );
   QRegularExpressionMatchIterator matchIt = sRegEx.globalMatch( mText );
   while ( !ok && matchIt.hasNext() )
   {
