--- a/src/app/pluginmanager/qgspluginmanager.h
+++ b/src/app/pluginmanager/qgspluginmanager.h
@@ -56,7 +56,7 @@
     static const QgsSettingsEntryVariant *settingsCheckOnStartLastDate;
     static const QgsSettingsEntryStringList *settingsSeenPlugins;
 
-    static inline QgsSettingsTreeNode *sTreeUi = sTreePluginManager->createChildNode( u"UI"_s );
+    static inline QgsSettingsTreeNode *sTreeUi = QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s )->createChildNode( u"UI"_s );
     static const QgsSettingsEntryString *settingsLastZipDirectory;
     static const QgsSettingsEntryBool *settingsShowInstallFromZipWarning;
 
--- a/src/app/pluginmanager/qgspluginmanager.cpp
+++ b/src/app/pluginmanager/qgspluginmanager.cpp
@@ -71,13 +71,13 @@
 #endif
 #endif
 
-const QgsSettingsEntryBool *QgsPluginManager::settingsAutomaticallyCheckForPluginUpdates = new QgsSettingsEntryBool( u"automatically-check-for-updates"_s, sTreePluginManager, true, u"Automatically check for plugin updates on startup"_s );
-const QgsSettingsEntryBool *QgsPluginManager::settingsAllowExperimental = new QgsSettingsEntryBool( u"allow-experimental"_s, sTreePluginManager, false, u"Allow experimental plugins."_s );
-const QgsSettingsEntryBool *QgsPluginManager::settingsAllowDeprecated = new QgsSettingsEntryBool( u"allow-deprecated"_s, sTreePluginManager, false, u"Allow deprecated plugins."_s );
-const QgsSettingsEntryVariant *QgsPluginManager::settingsCheckOnStartLastDate = new QgsSettingsEntryVariant( u"check-on-start-last-date"_s, sTreePluginManager, QgsVariantUtils::createNullVariant( QMetaType::Type::QDate ), u"Date last time the check was performed."_s );
-const QgsSettingsEntryStringList *QgsPluginManager::settingsSeenPlugins = new QgsSettingsEntryStringList( u"seen-plugins"_s, sTreePluginManager, {}, u"Date last time the check was performed."_s );
-const QgsSettingsEntryString *QgsPluginManager::settingsLastZipDirectory = new QgsSettingsEntryString( u"last-zip-directory"_s, sTreePluginManager, QString(), u"Last ZIP directory."_s );
-const QgsSettingsEntryBool *QgsPluginManager::settingsShowInstallFromZipWarning = new QgsSettingsEntryBool( u"show-install-from-zip-warning"_s, sTreePluginManager, true );
+const QgsSettingsEntryBool *QgsPluginManager::settingsAutomaticallyCheckForPluginUpdates = new QgsSettingsEntryBool( u"automatically-check-for-updates"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), true, u"Automatically check for plugin updates on startup"_s );
+const QgsSettingsEntryBool *QgsPluginManager::settingsAllowExperimental = new QgsSettingsEntryBool( u"allow-experimental"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), false, u"Allow experimental plugins."_s );
+const QgsSettingsEntryBool *QgsPluginManager::settingsAllowDeprecated = new QgsSettingsEntryBool( u"allow-deprecated"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), false, u"Allow deprecated plugins."_s );
+const QgsSettingsEntryVariant *QgsPluginManager::settingsCheckOnStartLastDate = new QgsSettingsEntryVariant( u"check-on-start-last-date"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), QgsVariantUtils::createNullVariant( QMetaType::Type::QDate ), u"Date last time the check was performed."_s );
+const QgsSettingsEntryStringList *QgsPluginManager::settingsSeenPlugins = new QgsSettingsEntryStringList( u"seen-plugins"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), {}, u"Date last time the check was performed."_s );
+const QgsSettingsEntryString *QgsPluginManager::settingsLastZipDirectory = new QgsSettingsEntryString( u"last-zip-directory"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), QString(), u"Last ZIP directory."_s );
+const QgsSettingsEntryBool *QgsPluginManager::settingsShowInstallFromZipWarning = new QgsSettingsEntryBool( u"show-install-from-zip-warning"_s, QgsSettingsTree::treeRoot()->createChildNode( u"plugin-manager"_s ), true );
 
 
 QgsPluginManager::QgsPluginManager( QWidget *parent, bool pluginsAreEnabled, Qt::WindowFlags fl )
