--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -418,10 +418,10 @@
   #############################################################
   # search for dependencies
 
-  if(NOT WIN32 AND NOT ANDROID)
+  if(NOT MSVC AND NOT ANDROID)
     include(CheckFunctionExists)
     CHECK_FUNCTION_EXISTS(openpty OPENPTY_IN_LIBC)
-    if(NOT OPENPTY_IN_LIBC)
+    if(UNIX AND NOT OPENPTY_IN_LIBC)
       set(CMAKE_REQUIRED_INCLUDES util.h)
       set(CMAKE_REQUIRED_LIBRARIES util)
       CHECK_FUNCTION_EXISTS(openpty NEED_LIBUTIL)
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -2294,7 +2294,7 @@
   )
 else()
   if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND (CMAKE_BUILD_TYPE MATCHES Release OR CMAKE_BUILD_TYPE MATCHES RelWithDebInfo))
-    set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/qgsexpression_texts.cpp PROPERTIES COMPILE_FLAGS "-O1")
+    set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/qgsexpression_texts.cpp textrenderer/qgsfontmanager.cpp PROPERTIES COMPILE_FLAGS "-O0 -Wa,-mbig-obj")
     set_source_files_properties(proj/qgscoordinatereferencesystem.cpp PROPERTIES COMPILE_FLAGS "-mllvm -inline-threshold=128")
   endif()
 endif()
