diff -urN dino-0.5.1.org/meson.build dino-0.5.1/meson.build
--- dino-0.5.1.org/meson.build	2025-11-16 01:30:00.000000000 +0500
+++ dino-0.5.1/meson.build	2026-05-03 10:30:54.732057000 +0500
@@ -1,4 +1,4 @@
-project('Dino', 'c', 'cpp', 'vala', license: 'GPL-3.0', meson_version: '>=1.3.0')
+project('Dino', 'c', 'cpp', 'vala', license: 'GPL-3.0', meson_version: '>=1.3.0', version: '0.5.1')
 
 fs = import('fs')
 gnome = import('gnome')
