diff -urN dino-0.5.1.org/main/src/ui/util/helper.vala dino-0.5.1/main/src/ui/util/helper.vala
--- dino-0.5.1.org/main/src/ui/util/helper.vala	2025-11-16 01:30:00.000000000 +0500
+++ dino-0.5.1/main/src/ui/util/helper.vala	2026-05-03 10:59:18.117751500 +0500
@@ -135,16 +135,12 @@
 }
 
 private static int8 is24h = 0;
-public static bool is_24h_format() {
-    if (is24h == 0) {
-        Regex has_ampm = /(^|[^%])%[pP]/;
-        Regex has_t_fmt_ampm = /(^|[^%])%r/;
-        unowned string t_fmt = Posix.nl_langinfo(Posix.NLItem.T_FMT);
-        unowned string t_fmt_ampm = Posix.nl_langinfo(Posix.NLItem.T_FMT_AMPM);
-        bool has_am_str = Posix.nl_langinfo(Posix.NLItem.AM_STR).strip() != "";
-        bool has_pm_str = Posix.nl_langinfo(Posix.NLItem.PM_STR).strip() != "";
-        is24h = ((has_ampm.match(t_fmt) || has_t_fmt_ampm.match(t_fmt) && has_ampm.match(t_fmt_ampm)) && (has_am_str || has_pm_str)) ? -1 : 1;
-    }
+public bool is_24h_format() {
+    if (is24h != 0) return is24h == 1;
+    var now = new DateTime.now_local();
+    string with_ampm = now.format("%I:%M %p");
+    string with_24h = now.format("%H:%M");
+    is24h = "%p" in now.format("%X") || now.format("%p").strip().length > 0 ? -1 : 1;
     return is24h == 1;
 }
 
