diff -Nur mlt-6.24.0-orig/src/swig/CMakeLists.txt mlt-6.24.0/src/swig/CMakeLists.txt
--- mlt-6.24.0-orig/src/swig/CMakeLists.txt	2020-12-07 06:37:27.000000000 +0100
+++ mlt-6.24.0/src/swig/CMakeLists.txt	2021-03-14 09:41:07.112648700 +0100
@@ -9,7 +9,7 @@
 if(Python3_FOUND)
   swig_add_library(pymlt LANGUAGE python SOURCES mlt.i)
   target_include_directories(pymlt PUBLIC ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/src/mlt++ ${Python3_INCLUDE_DIRS})
-  target_link_libraries(pymlt mlt mlt++)
+  target_link_libraries(pymlt mlt mlt++ ${Python3_LIBRARIES})
   set_target_properties(pymlt PROPERTIES PREFIX "_" OUTPUT_NAME "mlt")
 
   if(NOT DEFINED PYTHON_MODULE_PATH)
