From 44b01fae91c9db55388bea32fafefc8d24f17cf5 Mon Sep 17 00:00:00 2001
From: TheShermanTanker <tanksherman27@gmail.com>
Date: Wed, 19 Jul 2023 10:12:28 +0800
Subject: [PATCH] Disable ms_printf and related overrides for ucrt64

---
 gcc/config/i386/mingw32.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc/config/i386/mingw32.h b/gcc/config/i386/mingw32.h
index a1ee001983a..6a96e6aea9f 100644
--- a/gcc/config/i386/mingw32.h
+++ b/gcc/config/i386/mingw32.h
@@ -284,11 +284,11 @@ do {						         \
    overrides of format attributes.  See c-format.h for structure
    definition.  */
 #undef TARGET_OVERRIDES_FORMAT_ATTRIBUTES
-#define TARGET_OVERRIDES_FORMAT_ATTRIBUTES mingw_format_attribute_overrides
+// #define TARGET_OVERRIDES_FORMAT_ATTRIBUTES mingw_format_attribute_overrides
 
 /* Specify the count of elements in TARGET_OVERRIDES_ATTRIBUTE.  */
 #undef TARGET_OVERRIDES_FORMAT_ATTRIBUTES_COUNT
-#define TARGET_OVERRIDES_FORMAT_ATTRIBUTES_COUNT 3
+// #define TARGET_OVERRIDES_FORMAT_ATTRIBUTES_COUNT 3
 
 /* Custom initialization for warning -Wpedantic-ms-format for c-format.  */
 #undef TARGET_OVERRIDES_FORMAT_INIT
-- 
2.35.1.windows.2

