--- a/libgcc/Makefile.in	2020-08-29 23:18:54.246729700 -0700
+++ b/libgcc/Makefile.in	2020-08-29 23:23:26.754578500 -0700
@@ -84,6 +84,7 @@
 
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 RANLIB = @RANLIB@
 LN_S = @LN_S@
 
@@ -991,7 +992,7 @@
 	# @multilib_dir@ is not really necessary, but sometimes it has
 	# more uses than just a directory name.
 	$(mkinstalldirs) $(MULTIDIR)
-	$(subst @multilib_flags@,$(CFLAGS) -B./,$(subst \
+	$(subst @multilib_flags@,$(CFLAGS) -B./ $(LDFLAGS),$(subst \
 		@multilib_dir@,$(MULTIDIR),$(subst \
 		@shlib_objs@,$(objects) libgcc.a,$(subst \
 		@shlib_base_name@,libgcc_s,$(subst \
