diff -urN gcc-4.8.2.orig/gcc/config.gcc gcc-4.8.2/gcc/config.gcc
--- gcc-4.8.2.orig/gcc/config.gcc	2014-01-11 16:50:54.401307700 +0000
+++ gcc-4.8.2/gcc/config.gcc	2014-01-11 17:03:21.806216000 +0000
@@ -1504,7 +1504,10 @@
 			tmake_file="${tmake_file} i386/t-mingw-w32"
 			;;
 	esac
-        native_system_header_dir=/mingw/include
+	# Don't ignore values passed in to configure via --native-system-header-dir
+	if test x$native_system_header_dir = x ; then
+		native_system_header_dir=/mingw/include
+	fi
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	extra_options="${extra_options} i386/cygming.opt i386/mingw.opt"
 	case ${target} in
