diff -urN gcc-4.9.2.orig/gcc/config/i386/cygming.opt gcc-4.9.2/gcc/config/i386/cygming.opt
--- gcc-4.9.2.orig/gcc/config/i386/cygming.opt	2015-02-08 17:42:18.991804700 +0000
+++ gcc-4.9.2/gcc/config/i386/cygming.opt	2015-02-08 18:30:04.004615200 +0000
@@ -22,6 +22,9 @@
 Target RejectNegative
 Create console application
 
+mcrtdll=
+Target RejectNegative Joined
+
 mdll
 Target RejectNegative
 Generate code for a DLL
diff -urN gcc-4.9.2.orig/gcc/config/i386/mingw32.h gcc-4.9.2/gcc/config/i386/mingw32.h
--- gcc-4.9.2.orig/gcc/config/i386/mingw32.h	2015-02-08 17:42:19.021308500 +0000
+++ gcc-4.9.2/gcc/config/i386/mingw32.h	2015-02-08 18:31:49.685034900 +0000
@@ -142,7 +142,7 @@
 #define REAL_LIBGCC_SPEC \
   "%{mthreads:-lmingwthrd} -lmingw32 \
    "SHARED_LIBGCC_SPEC" \
-   -lmoldname -lmingwex -lmsvcrt"
+   -lmoldname -lmingwex %{!mcrtdll=*:-lmsvcrt} %{mcrtdll=*:-l%*}"
 
 #undef STARTFILE_SPEC
 #define STARTFILE_SPEC "%{shared|mdll:dllcrt2%O%s} \
