Index: gcc-4.9.0/libgcc/gthr-posix.h
===================================================================
--- a/libgcc/gthr-posix.h        (Revision 210070)
+++ b/libgcc/gthr-posix.h        (Arbeitskopie)
@@ -34,6 +34,10 @@ see the files COPYING3 and COPYING.RUNTIME respect

 #include <pthread.h>

+#if defined (_WIN32) && !defined (__CYGWIN__)
+#define _GTHREAD_USE_MUTEX_INIT_FUNC 1
+#endif
+
 #if ((defined(_LIBOBJC) || defined(_LIBOBJC_WEAK)) \
      || !defined(_GTHREAD_USE_MUTEX_TIMEDLOCK))
 # include <unistd.h>
