From 5a8232f8d3d3425869f7a8eb9f1b36824af516ce Mon Sep 17 00:00:00 2001
From: dragon-archer <dragon-archer@outlook.com>
Date: Fri, 8 May 2026 16:30:15 +0800
Subject: [PATCH] jit: set missing JIT_LDFLAGS for MinGW

---
 gcc/jit/Make-lang.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/jit/Make-lang.in b/gcc/jit/Make-lang.in
index eb1c9966062..b0c1294aa4a 100644
--- a/gcc/jit/Make-lang.in
+++ b/gcc/jit/Make-lang.in
@@ -154,6 +154,7 @@ jit_OBJS = $(JIT_OBJS) $(JIT_TARGET_OBJS)
 jit-warn = $(STRICT_WARN)
 
 ifneq (,$(findstring mingw,$(target)))
+JIT_LDFLAGS = $(LDFLAGS)
 # Create import library
 LIBGCCJIT_EXTRA_OPTS = -Wl,--out-implib,$(LIBGCCJIT_IMPORT_LIB)
 else
-- 
2.54.0.windows.1

