--- origsrc/gcc-4.8-20130310/libatomic/Makefile.am	2013-01-14 12:16:23.000000000 -0600
+++ src/gcc-4.8-20130310/libatomic/Makefile.am	2013-03-17 05:10:55.958313400 -0500
@@ -66,7 +66,7 @@
 endif
 libatomic_version_info = -version-info $(libtool_VERSION)
 
-libatomic_la_LDFLAGS = $(libatomic_version_info) $(libatomic_version_script)
+libatomic_la_LDFLAGS = $(libatomic_version_info) $(libatomic_version_script) $(lt_host_flags)
 libatomic_la_SOURCES = gload.c gstore.c gcas.c gexch.c glfree.c lock.c init.c
 
 SIZEOBJS = load store cas exch fadd fsub fand fior fxor fnand tas
--- origsrc/gcc-4.8-20130310/libatomic/Makefile.in	2013-03-10 17:37:36.000000000 -0500
+++ src/gcc-4.8-20130310/libatomic/Makefile.in	2013-03-17 05:13:32.732280300 -0500
@@ -244,6 +244,7 @@
 libtool_VERSION = @libtool_VERSION@
 localedir = @localedir@
 localstatedir = @localstatedir@
+lt_host_flags = @lt_host_flags@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
 multi_basedir = @multi_basedir@
@@ -285,7 +286,7 @@
 @LIBAT_BUILD_VERSIONED_SHLIB_GNU_TRUE@@LIBAT_BUILD_VERSIONED_SHLIB_TRUE@libatomic_version_dep = $(top_srcdir)/libatomic.map
 @LIBAT_BUILD_VERSIONED_SHLIB_SUN_TRUE@@LIBAT_BUILD_VERSIONED_SHLIB_TRUE@libatomic_version_dep = libatomic.map-sun
 libatomic_version_info = -version-info $(libtool_VERSION)
-libatomic_la_LDFLAGS = $(libatomic_version_info) $(libatomic_version_script)
+libatomic_la_LDFLAGS = $(libatomic_version_info) $(libatomic_version_script) $(lt_host_flags)
 libatomic_la_SOURCES = gload.c gstore.c gcas.c gexch.c glfree.c lock.c init.c
 SIZEOBJS = load store cas exch fadd fsub fand fior fxor fnand tas
 EXTRA_libatomic_la_SOURCES = $(addsuffix _n.c,$(SIZEOBJS))
--- origsrc/gcc-4.8-20130310/libatomic/aclocal.m4	2013-03-10 17:37:36.000000000 -0500
+++ src/gcc-4.8-20130310/libatomic/aclocal.m4	2013-03-17 05:09:13.598458700 -0500
@@ -993,6 +993,7 @@
 m4_include([../config/acx.m4])
 m4_include([../config/depstand.m4])
 m4_include([../config/lead-dot.m4])
+m4_include([../config/lthostflags.m4])
 m4_include([../config/multi.m4])
 m4_include([../config/override.m4])
 m4_include([../config/stdint.m4])
--- origsrc/gcc-4.8-20130310/libatomic/configure	2013-03-10 17:37:36.000000000 -0500
+++ src/gcc-4.8-20130310/libatomic/configure	2013-03-17 05:29:38.288507000 -0500
@@ -627,6 +627,7 @@
 MAINTAINER_MODE_TRUE
 enable_static
 enable_shared
+lt_host_flags
 CPP
 OTOOL64
 OTOOL
@@ -11013,7 +11014,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 11016 "configure"
+#line 11017 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -11119,7 +11120,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 11122 "configure"
+#line 11123 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -11357,6 +11358,28 @@
 
 
 
+
+
+case $host in
+  *-cygwin* | *-mingw*)
+    # 'host' will be top-level target in the case of a target lib,
+    # we must compare to with_cross_host to decide if this is a native
+    # or cross-compiler and select where to install dlls appropriately.
+    if test -n "$with_cross_host" &&
+	test x"$with_cross_host" != x"no"; then
+      lt_host_flags='-no-undefined -bindir "$(toolexeclibdir)"';
+    else
+      lt_host_flags='-no-undefined -bindir "$(bindir)"';
+    fi
+    ;;
+  *)
+    lt_host_flags=
+    ;;
+esac
+
+
+
+
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to enable maintainer-specific portions of Makefiles" >&5
--- origsrc/gcc-4.8-20130310/libatomic/configure.ac	2013-01-14 12:16:23.000000000 -0600
+++ src/gcc-4.8-20130310/libatomic/configure.ac	2013-03-17 05:29:30.655070400 -0500
@@ -143,6 +143,7 @@
 
 # Configure libtool
 AM_PROG_LIBTOOL
+ACX_LT_HOST_FLAGS
 AC_SUBST(enable_shared)
 AC_SUBST(enable_static)
 AM_MAINTAINER_MODE
--- origsrc/gcc-4.8-20130310/libatomic/configure.tgt	2013-01-14 12:16:23.000000000 -0600
+++ src/gcc-4.8-20130310/libatomic/configure.tgt	2013-03-17 04:50:09.727033000 -0500
@@ -108,7 +108,7 @@
   *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu \
   | *-*-netbsd* | *-*-freebsd* | *-*-openbsd* \
   | *-*-solaris2* | *-*-sysv4* | *-*-irix6* | *-*-osf* | *-*-hpux11* \
-  | *-*-darwin* | *-*-aix*)
+  | *-*-darwin* | *-*-aix* | *-*-cygwin* | *-*-mingw*)
 	# POSIX system.  The OS is supported.
 	config_path="${config_path} posix"
 	;;
