--- gcc-4.8.2/libitm/configure.tgt.orig	2013-08-15 00:39:49.000000000 +0400
+++ gcc-4.8.2/libitm/configure.tgt	2014-01-20 05:11:24.335047841 +0400
@@ -141,6 +141,10 @@
 	# POSIX system.  The OS is supported.
 	;;
 
+  *-*-mingw*)
+	# POSIX system.  The OS is supported.
+	;;
+
   *)	# Non-POSIX, or embedded system
 	UNSUPPORTED=1
 	;;
--- gcc-4.8.2/libitm/configure.ac.orig	2013-08-02 19:41:10.000000000 +0400
+++ gcc-4.8.2/libitm/configure.ac	2014-01-20 05:43:44.742256732 +0400
@@ -147,6 +147,7 @@
 
 # Configure libtool
 AM_PROG_LIBTOOL
+ACX_LT_HOST_FLAGS
 AC_SUBST(enable_shared)
 AC_SUBST(enable_static)
 
--- gcc-4.8.2/libitm/configure.orig	2013-10-16 11:27:58.000000000 +0400
+++ gcc-4.8.2/libitm/configure	2014-01-20 05:44:03.297290868 +0400
@@ -630,6 +630,7 @@
 MAINTAINER_MODE_TRUE
 enable_static
 enable_shared
+lt_host_flags
 CXXCPP
 CPP
 OTOOL64
@@ -15138,6 +15139,22 @@
 # Only expand once:
 
 
+case $host in
+  *-cygwin* | *-mingw*)
+    # 'host' will be top-level target in the case of a target lib,
+    # we must compare to with_cross_host to decide if this is a native
+    # or cross-compiler and select where to install dlls appropriately.
+    if test -n "$with_cross_host" &&
+	test x"$with_cross_host" != x"no"; then
+      lt_host_flags='-no-undefined -bindir "$(toolexeclibdir)"';
+    else
+      lt_host_flags='-no-undefined -bindir "$(bindir)"';
+    fi
+    ;;
+  *)
+    lt_host_flags=
+    ;;
+esac
 
 
 
--- gcc-4.8.2/libitm/Makefile.am.orig	2012-02-14 17:14:27.000000000 +0400
+++ gcc-4.8.2/libitm/Makefile.am	2014-01-20 05:44:33.339355008 +0400
@@ -54,7 +54,7 @@
 # want or need libstdc++.
 libitm_la_DEPENDENCIES = $(libitm_version_dep)
 libitm_la_LINK = $(LINK) $(libitm_la_LDFLAGS)
-libitm_la_LDFLAGS = $(libitm_version_info) $(libitm_version_script)
+libitm_la_LDFLAGS = $(libitm_version_info) $(libitm_version_script) $(lt_host_flags)
 
 libitm_la_SOURCES = \
 	aatree.cc alloc.cc alloc_c.cc alloc_cpp.cc barrier.cc beginend.cc \
--- gcc-4.8.2/libitm/Makefile.in.orig	2013-10-16 11:27:58.000000000 +0400
+++ gcc-4.8.2/libitm/Makefile.in	2014-01-20 05:44:57.043190317 +0400
@@ -279,6 +279,7 @@
 link_itm = @link_itm@
 localedir = @localedir@
 localstatedir = @localstatedir@
+lt_host_flags = @lt_host_flags@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
 multi_basedir = @multi_basedir@
@@ -329,7 +330,7 @@
 # want or need libstdc++.
 libitm_la_DEPENDENCIES = $(libitm_version_dep)
 libitm_la_LINK = $(LINK) $(libitm_la_LDFLAGS)
-libitm_la_LDFLAGS = $(libitm_version_info) $(libitm_version_script)
+libitm_la_LDFLAGS = $(libitm_version_info) $(libitm_version_script) $(lt_host_flags)
 libitm_la_SOURCES = aatree.cc alloc.cc alloc_c.cc alloc_cpp.cc \
 	barrier.cc beginend.cc clone.cc eh_cpp.cc local.cc query.cc \
 	retry.cc rwlock.cc useraction.cc util.cc sjlj.S tls.cc \
