--- a/libgcc/config/i386/cygming-crtbegin.c	2014-01-02 23:25:22.000000000 +0100
+++ b/libgcc/config/i386/cygming-crtbegin.c	2014-04-21 21:47:42.328125000 +0200
@@ -54,27 +54,11 @@
 				   TARGET_ATTRIBUTE_WEAK;
 extern void *__deregister_frame_info (__attribute__((unused)) const void *)
 				      TARGET_ATTRIBUTE_WEAK;
-TARGET_ATTRIBUTE_WEAK void
-__register_frame_info (__attribute__((unused)) const void *p,
-		       __attribute__((unused)) struct object *o)
-{
-}
-
-TARGET_ATTRIBUTE_WEAK void *
-__deregister_frame_info (__attribute__((unused)) const void *p)
-{
-  return (void*) 0;
-}
 #endif /* DWARF2_UNWIND_INFO */
 
 #if TARGET_USE_JCR_SECTION
 extern void _Jv_RegisterClasses (__attribute__((unused)) const void *)
   TARGET_ATTRIBUTE_WEAK;
-
-TARGET_ATTRIBUTE_WEAK void
-_Jv_RegisterClasses (__attribute__((unused)) const void *p)
-{
-}
 #endif /* TARGET_USE_JCR_SECTION */
 
 #if defined(HAVE_LD_RO_RW_SECTION_MIXING)
--- a/libgcc/config/i386/cygming-crtend.c	2014-01-02 23:25:22.000000000 +0100
+++ b/libgcc/config/i386/cygming-crtend.c	2014-04-21 22:02:54.921875000 +0200
@@ -37,6 +37,27 @@
 #include "libgcc_tm.h"
 #include "unwind-dw2-fde.h"
 
+#if DWARF2_UNWIND_INFO
+TARGET_ATTRIBUTE_WEAK void
+__register_frame_info (__attribute__((unused)) const void *p,
+		       __attribute__((unused)) struct object *o)
+{
+}
+
+TARGET_ATTRIBUTE_WEAK void *
+__deregister_frame_info (__attribute__((unused)) const void *p)
+{
+  return (void*) 0;
+}
+#endif /* DWARF2_UNWIND_INFO */
+
+#if TARGET_USE_JCR_SECTION
+TARGET_ATTRIBUTE_WEAK void
+_Jv_RegisterClasses (__attribute__((unused)) const void *p)
+{
+}
+#endif /* TARGET_USE_JCR_SECTION */
+
 #if defined(HAVE_LD_RO_RW_SECTION_MIXING)
 # define EH_FRAME_SECTION_CONST const
 #else
