--- gcc-4.8.2/gcc/config/i386/xm-mingw32.h.orig	2013-01-11 00:38:27.000000000 +0400
+++ gcc-4.8.2/gcc/config/i386/xm-mingw32.h	2014-04-14 22:32:15.278200000 +0400
@@ -29,6 +29,10 @@
 /*  The st_ino field of struct stat is always 0.  */
 #define HOST_LACKS_INODE_NUMBERS
 
+#ifndef __USE_MINGW_ANSI_STDIO
 /* MSVCRT does not support the "ll" format specifier for printing
    "long long" values.  Instead, we use "I64".  */
 #define HOST_LONG_LONG_FORMAT "I64"
+#else
+#define HOST_LONG_LONG_FORMAT "ll"
+#endif
