diff -urN Theano-rel-1.0.4.orig/theano/configdefaults.py Theano-rel-1.0.4/theano/configdefaults.py
--- Theano-rel-1.0.4.orig/theano/configdefaults.py	2019-08-23 00:44:48.986000000 +0800
+++ Theano-rel-1.0.4/theano/configdefaults.py	2019-08-23 00:46:02.791000000 +0800
@@ -1277,7 +1277,7 @@
     global numpy
     warn_record = []
     try:
-        if (hasattr(np.distutils, '__config__') and
+        if (False and hasattr(np.distutils, '__config__') and
                 np.distutils.__config__):
             # If the old private interface is available use it as it
             # don't print information to the user.
