From d1ce42e72cfdfcadc47560d3461a1a72c96ca741 Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Wed, 8 Sep 2021 19:39:53 +0530
Subject: [PATCH] cmake: Install pkgconfig files in mingw

---
 lib/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 85b43d59c..646402185 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -278,7 +278,7 @@ install(TARGETS ${LWS_LIBRARIES}
 set(CPACK_COMPONENT_DEV_DISPLAY_NAME "Development files" PARENT_SCOPE)
 
 
-if (UNIX)
+if (UNIX OR MINGW)
 
 # figure out pkfcfg required libs here
 
@@ -340,7 +340,7 @@ endif()
 	install(FILES "${PROJECT_BINARY_DIR}/libwebsockets_static.pc"
 		DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 
-endif(UNIX)
+endif(UNIX OR MINGW)
 
 
 # Keep explicit parent scope exports at end
