--- level-zero-1.15.8/source/loader/windows/driver_discovery_win.cpp.orig	2024-01-11 12:23:46.264191800 +0300
+++ level-zero-1.15.8/source/loader/windows/driver_discovery_win.cpp	2024-01-11 12:28:11.994358000 +0300
@@ -10,6 +10,7 @@
 
 #include <windows.h>
 
+#include <algorithm>
 #include <cassert>
 #include <cfgmgr32.h>
 #include <devpkey.h>
diff -bur level-zero-1.21.9-o/third_party/xla/graphcycles.h level-zero-1.21.9/third_party/xla/graphcycles.h
--- level-zero-1.21.9-o/third_party/xla/graphcycles.h	2025-04-27 14:28:43.014133600 -0600
+++ level-zero-1.21.9/third_party/xla/graphcycles.h	2025-04-27 14:29:16.331558700 -0600
@@ -50,6 +50,7 @@
 // #include "absl/types/span.h"
 // #include <span>
 #include <string>
+#include <cstdint>
 
 namespace xla {
 
