--- a/squirrel/CMakeLists.txt
+++ b/squirrel/CMakeLists.txt
@@ -35,6 +35,9 @@
   add_library(squirrel_static STATIC ${SQUIRREL_SRC})
   add_library(squirrel::squirrel_static ALIAS squirrel_static)
   set_property(TARGET squirrel_static PROPERTY EXPORT_NAME squirrel_static)
+  if(MINGW)
+    set_property(TARGET squirrel_static PROPERTY OUTPUT_NAME squirrel)
+  endif()
   if(NOT SQ_DISABLE_INSTALLER)
     install(TARGETS squirrel_static EXPORT squirrel ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Development)
   endif()
--- a/sqstdlib/CMakeLists.txt
+++ b/sqstdlib/CMakeLists.txt
@@ -32,6 +32,9 @@
   add_library(sqstdlib_static STATIC ${SQSTDLIB_SRC})
   add_library(squirrel::sqstdlib_static ALIAS sqstdlib_static)
   set_property(TARGET sqstdlib_static PROPERTY EXPORT_NAME sqstdlib_static)
+  if(MINGW)
+    set_property(TARGET sqstdlib_static PROPERTY OUTPUT_NAME sqstdlib)
+  endif()
   if(NOT SQ_DISABLE_INSTALLER)
     install(TARGETS sqstdlib_static EXPORT squirrel ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Development)
   endif()
