diff -Naur libwpg-0.3.3-orig/configure.ac libwpg-0.3.3/configure.ac
--- libwpg-0.3.3-orig/configure.ac	2019-01-01 19:21:42.000000000 +0300
+++ libwpg-0.3.3/configure.ac	2019-06-03 14:04:27.716673100 +0300
@@ -138,10 +138,10 @@
         AC_MSG_RESULT([$have_visibility])
         CXXFLAGS="$saved_CXXFLAGS"
         AX_GCC_FUNC_ATTRIBUTE([visibility])
-        AM_CONDITIONAL([HAVE_VISIBILITY], [
-            test "$have_visibility" = "yes" && test "$ax_cv_have_func_attribute_visibility" = "yes"])
     ]
 )
+AM_CONDITIONAL([HAVE_VISIBILITY], [
+    test "$have_visibility" = "yes" && test "$ax_cv_have_func_attribute_visibility" = "yes"])
 
 # ================
 # Check for cflags
