--- a/fpm-0.13.0.F90
+++ b/fpm-0.13.0.F90
@@ -13108,14 +13108,10 @@
 
     select case (get_os_type())
 
-    case (OS_UNKNOWN, OS_LINUX, OS_MACOS, OS_CYGWIN, OS_SOLARIS, OS_FREEBSD, OS_OPENBSD)
+    case (OS_UNKNOWN, OS_LINUX, OS_MACOS, OS_CYGWIN, OS_WINDOWS, OS_SOLARIS, OS_FREEBSD, OS_OPENBSD)
         call run( "test -d " // dir , &
                 & exitstat=stat,echo=.false.,verbose=.false.)
 
-    case (OS_WINDOWS)
-        call run('cmd /c "if not exist ' // windows_path(dir) // '\ exit /B 1"', &
-                & exitstat=stat,echo=.false.,verbose=.false.)
-
     end select
 
     is_dir = (stat == 0)
