diff -Naur a/src/liolib.c b/src/liolib.c
--- a/src/liolib.c	2020-09-14 12:34:00.000000000 -0300
+++ b/src/liolib.c	2024-06-27 18:06:27.801668564 -0300
@@ -111,8 +111,8 @@
 #define l_ftell(f)		ftello(f)
 #define l_seeknum		off_t
 
-#elif defined(LUA_USE_WINDOWS) && !defined(_CRTIMP_TYPEINFO) \
-   && defined(_MSC_VER) && (_MSC_VER >= 1400)	/* }{ */
+#elif defined(LUA_USE_WINDOWS) && (defined(__MINGW32__) || !defined(_CRTIMP_TYPEINFO) \
+   && defined(_MSC_VER) && (_MSC_VER >= 1400))	/* }{ */
 
 /* Windows (but not DDK) and Visual C++ 2005 or higher */
 #define l_fseek(f,o,w)		_fseeki64(f,o,w)
