--- src/Makefile.orig	2023-02-15 08:58:22.205723900 +0100
+++ src/Makefile	2023-02-15 09:01:51.544746000 +0100
@@ -114,7 +114,7 @@
 
 mingw:
 	$(MAKE) "LUA_A=lua53.dll" "LUA_T=lua5.3.exe" \
-	"AR=$(CC) -shared -o" "RANLIB=strip --strip-unneeded" \
+	"AR=$(CC) -shared -Wl,--out-implib=liblua5.3.dll.a -o" "RANLIB=strip --strip-unneeded" \
 	"SYSCFLAGS=-DLUA_BUILD_AS_DLL" "SYSLIBS=" "SYSLDFLAGS=-s" lua5.3.exe
 	$(MAKE) "LUAC_T=luac5.3.exe" luac.exe
 
