--- PyQt5-5.14.1/configure.py.orig	2020-02-21 11:20:07.361753100 +0300
+++ PyQt5-5.14.1/configure.py	2020-02-21 11:27:47.538788100 +0300
@@ -2148,15 +2148,22 @@
     if target_config.py_platform == 'win32':
         if target_config.qmake_spec == 'win32-g++':
             make = 'mingw32-make'
+
+            if target_config.debug:
+                makefile_target = 'debug'
+                platform_exe = os.path.join('debug', exe + '.exe')
+            else:
+                makefile_target = 'all'
+                platform_exe = os.path.join('./', exe + '.exe')
         else:
             make = 'nmake'
 
-        if target_config.debug:
-            makefile_target = 'debug'
-            platform_exe = os.path.join('debug', exe + '.exe')
-        else:
-            makefile_target = 'release'
-            platform_exe = os.path.join('release', exe + '.exe')
+            if target_config.debug:
+                makefile_target = 'debug'
+                platform_exe = os.path.join('debug', exe + '.exe')
+            else:
+                makefile_target = 'release'
+                platform_exe = os.path.join('release', exe + '.exe')
     else:
         make = 'make'
         makefile_target = ''
