--- oidn-1.4.3/cmake/FindTBB.cmake.orig	2025-06-10 12:42:42.122501800 +0300
+++ oidn-1.4.3/cmake/FindTBB.cmake	2025-06-10 12:42:46.477048300 +0300
@@ -93,7 +93,7 @@
   # TBB_FIND_REQUIRED_<C> variables based on the invocation
   # of find_package.
   if (TBB_FIND_COMPONENTS STREQUAL "")
-    set(_REQUIRED_COMPONENTS "tbb")
+    set(_REQUIRED_COMPONENTS "tbb12")
     set(_OPTIONAL_COMPONENTS "tbbmalloc"
                              "tbbmalloc_proxy"
                              "tbbbind"
