--- lame-3.100/libmp3lame/Makefile.am.orig	2016-01-29 22:06:26.000000000 +0100
+++ lame-3.100/libmp3lame/Makefile.am	2017-10-15 11:40:06.357621100 +0200
@@ -22,7 +22,6 @@
 libmp3lame_la_LIBADD =	$(cpu_ldadd) $(vector_ldadd) $(decoder_ldadd) \
 			$(CONFIG_MATH_LIB)
 libmp3lame_la_LDFLAGS = -version-info @LIB_MAJOR_VERSION@:@LIB_MINOR_VERSION@ \
-			-export-symbols $(top_srcdir)/include/libmp3lame.sym \
 			-no-undefined
 
 INCLUDES = @INCLUDES@ -I$(top_srcdir)/mpglib -I$(top_builddir)
