diff -r 0c0093754b62 c/misc_win32.h
--- a/c/misc_win32.h	Tue Nov 17 10:36:39 2015 +0100
+++ b/c/misc_win32.h	Wed Nov 18 10:34:34 2015 +0100
@@ -238,4 +238,3 @@
 /************************************************************/
 /* obscure */
 
-#define ffi_prep_closure(a,b,c,d)  ffi_prep_closure_loc(a,b,c,d,a)
diff -r 0c0093754b62 setup.py
--- a/setup.py	Tue Nov 17 10:36:39 2015 +0100
+++ b/setup.py	Wed Nov 18 10:34:34 2015 +0100
@@ -86,10 +86,7 @@
         os.environ.get('PKG_CONFIG_PATH', '') + ':' + pkgconfig)
 
 
-if sys.platform == 'win32':
-    COMPILE_LIBFFI = 'c/libffi_msvc'    # from the CPython distribution
-else:
-    COMPILE_LIBFFI = None
+COMPILE_LIBFFI = None
 
 if COMPILE_LIBFFI:
     assert os.path.isdir(COMPILE_LIBFFI), "directory not found!"
