--- OpenSceneGraph/CMakeLists.txt.orig	2015-01-25 14:09:28.105400000 +0300
+++ OpenSceneGraph/CMakeLists.txt	2015-01-25 14:09:37.496600000 +0300
@@ -508,7 +508,7 @@
         SET(OPENGL_HEADER2 "" CACHE STRING "#include<> line for additional OpenGL Headers if required")
     ELSE()
         SET(OPENGL_HEADER1 "#include <GL/gl.h>" CACHE STRING "#include<> line for OpenGL Header")
-        SET(OPENGL_HEADER2 "" CACHE STRING "#include<> line for additional OpenGL Headers if required")
+        SET(OPENGL_HEADER2 "#include <GL/glext.h>" CACHE STRING "#include<> line for additional OpenGL Headers if required")
     ENDIF()
 ENDIF()
 
