--- OpenSceneGraph-3.2.1-rc1/CMakeModules/FindCOLLADA.cmake.orig	2013-12-25 00:28:48.219200000 +0400
+++ OpenSceneGraph-3.2.1-rc1/CMakeModules/FindCOLLADA.cmake	2013-12-25 01:05:55.671600000 +0400
@@ -17,7 +17,7 @@
 # Check if COLLADA_DIR is set, otherwise use ACTUAL_3DPARTY_DIR:
 SET( COLLADA_ENV_VAR_AVAILABLE $ENV{COLLADA_DIR} )
 IF ( COLLADA_ENV_VAR_AVAILABLE )
-    SET(COLLADA_DOM_ROOT "$ENV{COLLADA_DIR}/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
+    SET(COLLADA_DOM_ROOT "$ENV{COLLADA_DIR}/include/collada-dom2.4/1.4/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
 ELSE ()
     SET(COLLADA_DOM_ROOT "${ACTUAL_3DPARTY_DIR}/include/1.4/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
 ENDIF()
@@ -46,6 +46,7 @@
 FIND_PATH(COLLADA_INCLUDE_DIR dae.h
     ${COLLADA_DOM_ROOT}/include
     $ENV{COLLADA_DIR}/include
+    $ENV{COLLADA_DIR}/include/collada-dom2.4
     $ENV{COLLADA_DIR}
     ~/Library/Frameworks
     /Library/Frameworks
