--- OpenSceneGraph/src/osgPlugins/osgjs/WriteVisitor.cpp.orig	2015-05-08 12:19:06.415757900 +0300
+++ OpenSceneGraph/src/osgPlugins/osgjs/WriteVisitor.cpp	2015-05-08 12:19:00.109388600 +0300
@@ -225,7 +225,7 @@
             // no image file so use this inline name image and create a file
             std::stringstream ss;
             ss << osgDB::getFilePath(baseName) << osgDB::getNativePathSeparator();
-            ss << (long int)image << ".inline_conv_generated.png"; // write the pointer location
+            ss << (intptr_t)image << ".inline_conv_generated.png"; // write the pointer location
             std::string filename = ss.str();
             if (osgDB::writeImageFile(*image, filename)) {
                 image->setFileName(filename);
