--- h5py-2.5.0/setup_build.py.orig	2015-07-21 19:33:54.858023400 +0300
+++ h5py-2.5.0/setup_build.py	2015-07-21 19:37:16.980584200 +0300
@@ -17,6 +17,7 @@
 from functools import reduce
 import api_gen
 
+from sysconfig import _POSIX_BUILD
 
 def localpath(*args):
     return op.abspath(reduce(op.join, (op.dirname(__file__),)+args))
@@ -36,7 +37,14 @@
               localpath("lzf/lzf/lzf_d.c")]} 
 
 
-if sys.platform.startswith('win'):
+if _POSIX_BUILD:
+    COMPILER_SETTINGS = {
+        'libraries'     : ['hdf5', 'hdf5_hl'],
+        'include_dirs'  : [localpath('lzf'), '/opt/local/include', '/usr/local/include'],
+        'library_dirs'  : ['/opt/local/lib', '/usr/local/lib'],
+        'define_macros' : [('H5_USE_16_API', None), ('_HDF5USEDLL_', None)] }
+        
+elif sys.platform.startswith('win'):
     COMPILER_SETTINGS = {
         'libraries'     : ['h5py_hdf5', 'h5py_hdf5_hl'],
         'include_dirs'  : [localpath('lzf'), localpath('windows')],
