--- a/setup_configure.py
+++ b/setup_configure.py
@@ -224,8 +224,12 @@
             default_path = 'libhdf5.dylib'
             regexp = re.compile(r'^libhdf5.dylib')
         elif sys.platform.startswith('win'):
-            default_path = 'hdf5.dll'
-            regexp = re.compile(r'^hdf5.dll')
+            if 'GCC' in sys.version:
+                default_path = 'libhdf5-0.dll'
+                regexp = re.compile(r'^libhdf5-[0-9].dll')
+            else:
+                default_path = 'hdf5.dll'
+                regexp = re.compile(r'^hdf5.dll')
             if sys.version_info >= (3, 8):
                 # To overcome "difficulty" loading the library on windows
                 # https://bugs.python.org/issue42114
