--- ./m4/intltool.m4.orig	2013-05-15 13:07:27.000000000 +0200
+++ ./m4/intltool.m4	2015-03-14 17:51:28.198407000 +0100
@@ -155,31 +155,6 @@
 # Substitute ALL_LINGUAS so we can use it in po/Makefile
 AC_SUBST(ALL_LINGUAS)
 
-# Set DATADIRNAME correctly if it is not set yet
-# (copied from glib-gettext.m4)
-if test -z "$DATADIRNAME"; then
-  AC_LINK_IFELSE(
-    [AC_LANG_PROGRAM([[]],
-                     [[extern int _nl_msg_cat_cntr;
-                       return _nl_msg_cat_cntr]])],
-    [DATADIRNAME=share],
-    [case $host in
-    *-*-solaris*)
-    dnl On Solaris, if bind_textdomain_codeset is in libc,
-    dnl GNU format message catalog is always supported,
-    dnl since both are added to the libc all together.
-    dnl Hence, we'd like to go with DATADIRNAME=share
-    dnl in this case.
-    AC_CHECK_FUNC(bind_textdomain_codeset,
-      [DATADIRNAME=share], [DATADIRNAME=lib])
-    ;;
-    *)
-    [DATADIRNAME=lib]
-    ;;
-    esac])
-fi
-AC_SUBST(DATADIRNAME)
-
 IT_PO_SUBDIR([po])
 
 ])
--- ./src/dh-book-manager.c.orig	2013-02-05 19:18:57.000000000 +0100
+++ ./src/dh-book-manager.c	2015-03-14 17:51:23.006765300 +0100
@@ -528,8 +528,7 @@
 
                 /* Build the path of the directory where the final
                  * devhelp book resides */
-                book_dir_path = g_build_filename (G_DIR_SEPARATOR_S,
-                                                  dir_path,
+                book_dir_path = g_build_filename (dir_path,
                                                   name,
                                                   NULL);
 
--- ./src/dh-link.c.orig	2011-08-21 12:00:05.000000000 +0200
+++ ./src/dh-link.c	2015-03-14 17:56:00.731796500 +0100
@@ -234,7 +234,8 @@
         else
                 base = link->book->base;
 
-	uri = g_strconcat ("file://", base, "/", link->filename, NULL, NULL);
+	uri = g_strconcat ("file:///", base, "/", link->filename, NULL, NULL);
+	uri = g_strjoinv("/", g_strsplit(uri, "\\", -1));
 
 	return uri;
 }
--- ./src/dh-main.c.orig	2013-02-01 15:35:06.000000000 +0100
+++ ./src/dh-main.c	2015-03-14 17:51:23.019765900 +0100
@@ -103,7 +103,13 @@
         gint     status;
 
         setlocale (LC_ALL, "");
+#ifdef G_OS_WIN32
+		gchar *dir = g_win32_get_package_installation_directory_of_module (NULL);
+		bindtextdomain (GETTEXT_PACKAGE, g_build_filename (dir, "share", "locale", NULL));
+		g_free (dir);
+#else
         bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+#endif
         bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
         textdomain (GETTEXT_PACKAGE);
 
--- ./src/dh-util.c.orig	2013-05-15 13:05:20.000000000 +0200
+++ ./src/dh-util.c	2015-03-14 17:51:23.025767100 +0100
@@ -131,7 +131,13 @@
         va_start (args, first_part);
 
         if (datadir == NULL) {
+#ifdef G_OS_WIN32
+		gchar *dir = g_win32_get_package_installation_directory_of_module (NULL);
+		datadir = g_build_filename (dir, "share", NULL);
+		g_free (dir);
+#else
                 datadir = g_strdup (DATADIR);
+#endif
         }
 
         /* 2 = 1 initial component + terminating NULL element. */
