diff -urN scotch_6.0.8.orig/src/libscotch/common_file_compress.c scotch_6.0.8/src/libscotch/common_file_compress.c
--- scotch_6.0.8.orig/src/libscotch/common_file_compress.c	2019-05-18 16:40:45.000000000 -0700
+++ scotch_6.0.8/src/libscotch/common_file_compress.c	2019-10-08 04:32:19.642967800 -0700
@@ -209,8 +209,8 @@
   if (typeval <= FILECOMPRESSTYPENONE)            /* If nothing to do */
     return (0);
 
-  if (pipe (filetab) != 0) {
-    errorPrint ("fileCompress: cannot create pipe");
+  if (1/*pipe (filetab) != 0*/) {
+    errorPrint ("fileCompress: cannot create pipe: unsupported call");
     return (1);
   }
 
diff -urN scotch_6.0.8.orig/src/libscotch/common_file_decompress.c scotch_6.0.8/src/libscotch/common_file_decompress.c
--- scotch_6.0.8.orig/src/libscotch/common_file_decompress.c	2019-05-10 08:34:57.000000000 -0700
+++ scotch_6.0.8/src/libscotch/common_file_decompress.c	2019-10-08 04:32:48.017415600 -0700
@@ -186,8 +186,8 @@
   if (typeval <= FILECOMPRESSTYPENONE)            /* If nothing to do */
     return (0);
 
-  if (pipe (filetab) != 0) {
-    errorPrint ("fileDecompress: cannot create pipe");
+  if (1/*pipe (filetab) != 0*/) {
+    errorPrint ("fileDecompress: cannot create pipe: unsupported call");
     return (1);
   }
 
