diff -urN scotch_6.0.8.orig/src/Makefile scotch_6.0.8/src/Makefile
--- scotch_6.0.8.orig/src/Makefile	2019-05-18 16:53:25.000000000 -0700
+++ scotch_6.0.8/src/Makefile	2019-10-08 04:43:46.271261600 -0700
@@ -94,14 +94,14 @@
 
 scotch				:	libscotch
 					(cd scotch ;         $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) scotch && $(MAKE) install)
-					(cd libscotchmetis ; $(MAKE)                                                                scotch && $(MAKE) install)
+#					(cd libscotchmetis ; $(MAKE)                                                                scotch && $(MAKE) install)
 
 libptscotch			:	libscotch
 					(cd libscotch ;      $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) ptscotch && $(MAKE) ptinstall)
 
 ptscotch			:	libptscotch
 					(cd scotch ;         $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) ptscotch && $(MAKE) ptinstall)
-					(cd libscotchmetis ; $(MAKE)                                                                ptscotch && $(MAKE) ptinstall)
+#					(cd libscotchmetis ; $(MAKE)                                                                ptscotch && $(MAKE) ptinstall)
 
 check				:	scotch
 					(cd check ; $(MAKE) check)
@@ -127,14 +127,14 @@
 clean				:	required
 					(cd libscotch ;      $(MAKE) clean)
 					(cd scotch ;         $(MAKE) clean)
-					(cd libscotchmetis ; $(MAKE) clean)
+#					(cd libscotchmetis ; $(MAKE) clean)
 					(cd check ;          $(MAKE) clean)
 					(cd esmumps ;        $(MAKE) clean)
 
 realclean			:	required
 					(cd libscotch ;      $(MAKE) realclean)
 					(cd scotch ;         $(MAKE) realclean)
-					(cd libscotchmetis ; $(MAKE) realclean)
+#					(cd libscotchmetis ; $(MAKE) realclean)
 					(cd check ;          $(MAKE) realclean)
 					(cd esmumps ;        $(MAKE) realclean)
 					-$(RM) ../bin/* ../include/* ../lib/*
