$ diff -urN src/libscotch/context.c.orig src/libscotch/context.c
--- a/src/libscotch/context.c.orig	2022-02-14 18:14:58.000000000 +0100
+++ b/src/libscotch/context.c	2022-02-20 17:39:35.603086300 +0100
@@ -47,7 +47,9 @@
 **  The defines and includes.
 */
 
+#ifndef _WIN32
 #define CONTEXT
+#endif
 
 #include "module.h"
 #include "common.h"

diff -urN src/libscotch/module.h.orig src/libscotch/module.h
--- a/src/libscotch/module.h.orig	2022-02-14 18:14:58.000000000 +0100
+++ b/src/libscotch/module.h	2022-02-20 18:03:54.990997900 +0100
@@ -68,7 +68,7 @@
 #define SCOTCH_COPYRIGHT_STRING     "Copyright 1992-2022 IPB, Universite de Bordeaux, INRIA & CNRS, France"
 #define SCOTCH_LICENSE_STRING       "This software is libre/free software under CeCILL-C -- see the user's manual for more information"
 #ifdef SCOTCH_CODENAME
-#define SCOTCH_VERSION_STRING       STRINGIFY (SCOTCH_VERSION) "." STRINGIFY (SCOTCH_RELEASE) "." STRINGIFY (SCOTCH_PATCHLEVEL) " (" SCOTCH_CODENAME ")"
+#define SCOTCH_VERSION_STRING       STRINGIFY (SCOTCH_VERSION) "." STRINGIFY (SCOTCH_RELEASE) "." STRINGIFY (SCOTCH_PATCHLEVEL) " (" STRINGIFY (SCOTCH_CODENAME) ")"
 #else /* SCOTCH_CODENAME */
 #define SCOTCH_VERSION_STRING       STRINGIFY (SCOTCH_VERSION) "." STRINGIFY (SCOTCH_RELEASE) "." STRINGIFY (SCOTCH_PATCHLEVEL)
 #endif /* SCOTCH_CODENAME */
