https://gitlab.inria.fr/scotch/scotch/-/issues/48

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6c84b7b..46e5fcd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -130,7 +130,7 @@ endif()
 # Detect and remove GCC _FORTIFY_SOURCE
 if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
   execute_process(COMMAND ${CMAKE_C_COMPILER} ${CMAKE_C_FLAGS} -E -Og -dM -x c ${dev_null} OUTPUT_VARIABLE gcc_macros)
-  string(FIND ${gcc_macros} "_FORTIFY_SOURCE" fortify)
+  string(FIND "${gcc_macros}" "_FORTIFY_SOURCE" fortify)
   if(fortify GREATER -1)
     message(WARNING "GCC _FORTIFY_SOURCE is known to cause problem with Scotch and has been disabled")
     set(CMAKE_C_FLAGS "-U_FORTIFY_SOURCE ${CMAKE_C_FLAGS}")
diff --git a/src/scotch/amk_m2.h b/src/scotch/amk_m2.h
index 9349cb8..b170b46 100644
--- a/src/scotch/amk_m2.h
+++ b/src/scotch/amk_m2.h
@@ -91,7 +91,7 @@ typedef int (* DomnBipartFunc) (const ArchMesh2 * const, const ArchMesh2Dom * co
 **  The function prototypes.
 */
 
-void                        C_domnBipart        (ArchMesh2 *, ArchMesh2Dom *, Anum, Anum *, Anum *, int (*) ());
+void                        C_domnBipart        (ArchMesh2 *, ArchMesh2Dom *, Anum, Anum *, Anum *, DomnBipartFunc);
 int                         C_domnBipartOne     (const ArchMesh2 * const, const ArchMesh2Dom * const, ArchMesh2Dom * restrict const, ArchMesh2Dom * restrict const);
 
 /*
