diff -urN scotch_6.0.4.orig/src/libscotch/common_file_compress.c scotch_6.0.4/src/libscotch/common_file_compress.c
--- scotch_6.0.4.orig/src/libscotch/common_file_compress.c	2014-01-06 04:21:58.000000000 +0600
+++ scotch_6.0.4/src/libscotch/common_file_compress.c	2018-02-12 14:05:25.879539000 +0500
@@ -179,8 +179,8 @@
   if (typeval <= FILECOMPRESSTYPENONE)            /* If uncompressed stream, return original stream pointer */
     return (stream);
 
-  if (pipe (filetab) != 0) {
-    errorPrint ("fileCompress: cannot create pipe");
+  if (1/*pipe (filetab) != 0*/) {
+    errorPrint ("fileCompress: cannot create pipe: unsupported call");
     return (NULL);
   }
 
diff -urN scotch_6.0.4.orig/src/libscotch/common_file_uncompress.c scotch_6.0.4/src/libscotch/common_file_uncompress.c
--- scotch_6.0.4.orig/src/libscotch/common_file_uncompress.c	2014-01-06 04:21:58.000000000 +0600
+++ scotch_6.0.4/src/libscotch/common_file_uncompress.c	2018-02-12 14:05:58.254539000 +0500
@@ -182,8 +182,8 @@
   if (typeval <= FILECOMPRESSTYPENONE)            /* If uncompressed stream, return original stream pointer */
     return (stream);
 
-  if (pipe (filetab) != 0) {
-    errorPrint ("fileUncompress: cannot create pipe");
+  if (1/*pipe (filetab) != 0*/) {
+    errorPrint ("fileUncompress: cannot create pipe: unsupported call");
     return (NULL);
   }
 
