--- gtkglext-1.0.6/configure.in	2005-09-10 16:45:06.000000000 +0200
+++ gtkglext-1.0.6.az/configure.in	2005-09-10 16:46:01.000000000 +0200
@@ -59,6 +59,10 @@
 m4_define([pango_pkg], [pango])
 m4_define([pango_required_version], [1.0.0])
 
+# Pangox
+m4_define([pangox_pkg], [pangox])
+m4_define([pangox_required_version], [1.0.0])
+
 # PangoFT2
 m4_define([pangoft2_pkg], [pangoft2])
 m4_define([pangoft2_required_version], [1.0.0])
@@ -345,6 +349,7 @@
 gtk_pkg >= gtk_required_version \
 gdk_pkg >= gdk_required_version \
 pango_pkg >= pango_required_version \
+pangox_pkg >= pangox_required_version \
 gmodule_pkg >= gmodule_required_version \
 ])
 
@@ -789,7 +794,7 @@
 # CFLAGS and LIBS
 ##################################################
 
-GDKGLEXT_PACKAGES="gdk_pkg pango_pkg gmodule_pkg"
+GDKGLEXT_PACKAGES="gdk_pkg pango_pkg pangox_pkg gmodule_pkg"
 GDKGLEXT_EXTRA_CFLAGS="$GL_CFLAGS $GDKGLEXT_WIN_CFLAGS"
 GDKGLEXT_EXTRA_LIBS="$GL_LIBS $GDKGLEXT_WIN_LIBS"
 GDKGLEXT_DEP_CFLAGS="$GDKGLEXT_EXTRA_CFLAGS `$PKG_CONFIG --cflags $GDKGLEXT_PACKAGES`"
