diff --git a/buildconfig/config_msys2.py b/buildconfig/config_msys2.py
index 1fab5b90..ba594be8 100644
--- a/buildconfig/config_msys2.py
+++ b/buildconfig/config_msys2.py
@@ -470,7 +470,7 @@ def main(auto_config=False):
     if download_prebuilt:
         download_prebuilt = os.environ['PYGAME_DOWNLOAD_PREBUILT'] == '1'
     else:
-        download_prebuilt = True
+        download_prebuilt = False
 
     try:
         from . import download_msys2_prebuilt
@@ -481,7 +481,7 @@ def main(auto_config=False):
     if download_prebuilt:
-        download_msys2_prebuilt.update()
+        pass
 
     # MSYS2 config only supports setup with prebuilt dependencies
     # The prebuilt dir is the MinGW root from the MSYS2
