diff -bur pygame-2.6.1-o/src_c/alphablit.c pygame-2.6.1/src_c/alphablit.c
--- pygame-2.6.1-o/src_c/alphablit.c	2025-04-13 18:52:15.879513700 -0600
+++ pygame-2.6.1/src_c/alphablit.c	2025-04-13 18:52:43.769349900 -0600
@@ -47,7 +47,7 @@
 
 #if PG_ENABLE_ARM_NEON
 // sse2neon.h is from here: https://github.com/DLTcollab/sse2neon
-#include "include/sse2neon.h"
+#include <sse2neon/sse2neon.h>
 #endif /* PG_ENABLE_ARM_NEON */
 
 /* This defines PG_ENABLE_SSE_NEON as True if either SSE or NEON is available
diff -bur pygame-2.6.1-o/src_c/simd_blitters_sse2.c pygame-2.6.1/src_c/simd_blitters_sse2.c
--- pygame-2.6.1-o/src_c/simd_blitters_sse2.c	2025-04-13 18:52:15.900212100 -0600
+++ pygame-2.6.1/src_c/simd_blitters_sse2.c	2025-04-13 18:52:40.608100200 -0600
@@ -2,7 +2,7 @@
 
 #if PG_ENABLE_ARM_NEON
 // sse2neon.h is from here: https://github.com/DLTcollab/sse2neon
-#include "include/sse2neon.h"
+#include <sse2neon/sse2neon.h>
 #endif /* PG_ENABLE_ARM_NEON */
 
 /* See if we are compiled 64 bit on GCC or MSVC */
diff -bur pygame-2.6.1-o/buildconfig/Setup.SDL2.in pygame-2.6.1/buildconfig/Setup.SDL2.in
--- pygame-2.6.1-o/buildconfig/Setup.SDL2.in	2025-04-13 19:13:32.407008700 -0600
+++ pygame-2.6.1/buildconfig/Setup.SDL2.in	2025-04-13 19:18:58.455823000 -0600
@@ -67,7 +67,7 @@
 joystick src_c/joystick.c $(SDL) $(DEBUG)
 draw src_c/draw.c $(SDL) $(DEBUG)
 image src_c/image.c $(SDL) $(DEBUG)
-transform src_c/transform.c src_c/rotozoom.c src_c/scale2x.c src_c/scale_mmx.c $(SDL) $(DEBUG)
+transform src_c/transform.c src_c/rotozoom.c src_c/scale2x.c src_c/scale_mmx.c $(SDL) $(DEBUG) -D_NO_MMX_FOR_X86_64
 mask src_c/mask.c src_c/bitmask.c $(SDL) $(DEBUG)
 bufferproxy src_c/bufferproxy.c $(SDL) $(DEBUG)
 pixelarray src_c/pixelarray.c $(SDL) $(DEBUG)
diff -bur pygame-2.6.1-o/src_c/camera.h pygame-2.6.1/src_c/camera.h
--- pygame-2.6.1-o/src_c/camera.h	2025-04-13 19:25:38.008567000 -0600
+++ pygame-2.6.1/src_c/camera.h	2025-04-13 19:25:45.373221400 -0600
@@ -60,8 +60,8 @@
 
 #ifdef __MINGW32__
 #undef WINVER
-/** _WIN32_WINNT_WINBLUE sets minimum platform SDK to Windows 8.1. */
-#define WINVER _WIN32_WINNT_WINBLUE
+/** _WIN32_WINNT_WIN10 sets minimum platform SDK to Windows 10. */
+#define WINVER _WIN32_WINNT_WIN10
 #endif
 
 #include <mfapi.h>
