From 109a2275d8f1615b967441877f6ed367b4f709d6 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Fri, 22 Aug 2014 15:43:59 +0200
Subject: [PATCH] Do not make gedit api calls before the private lib is loaded

---
 gedit/gedit.c | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/gedit/gedit.c b/gedit/gedit.c
index 619446f..4ada6d3 100644
--- a/gedit/gedit.c
+++ b/gedit/gedit.c
@@ -98,17 +98,6 @@ main (int argc, char *argv[])
 	gint status;
 	const gchar *dir;
 
-	gedit_dirs_init ();
-
-	/* Setup locale/gettext */
-	setlocale (LC_ALL, "");
-
-	dir = gedit_dirs_get_gedit_locale_dir ();
-	bindtextdomain (GETTEXT_PACKAGE, dir);
-
-	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
-	textdomain (GETTEXT_PACKAGE);
-
 #ifdef OS_OSX
 	type = GEDIT_TYPE_APP_OSX;
 #else
@@ -125,6 +114,19 @@ main (int argc, char *argv[])
 #endif
 #endif
 
+	/* NOTE: we should not make any calls to the gedit api before the
+	 * private library is loaded */
+	gedit_dirs_init ();
+
+	/* Setup locale/gettext */
+	setlocale (LC_ALL, "");
+
+	dir = gedit_dirs_get_gedit_locale_dir ();
+	bindtextdomain (GETTEXT_PACKAGE, dir);
+
+	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+	textdomain (GETTEXT_PACKAGE);
+
 	app = g_object_new (type,
 	                    "application-id", "org.gnome.gedit",
 	                    "flags", G_APPLICATION_HANDLES_COMMAND_LINE | G_APPLICATION_HANDLES_OPEN,
-- 
1.9.3

