From f4c62c2a86aeec1b50d39c420c8a09414e8d43b2 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Tue, 22 Jul 2014 11:46:02 +0200
Subject: [PATCH] Specify INTROSPECTION_SCANNER_ENV to be able to build on
 windows

---
 gedit/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gedit/Makefile.am b/gedit/Makefile.am
index ed4d761..8aeb435 100644
--- a/gedit/Makefile.am
+++ b/gedit/Makefile.am
@@ -250,6 +250,7 @@ EXTRA_DIST += 					\
 if HAVE_INTROSPECTION
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS = gedit/Gedit-3.0.gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 gedit/Gedit-3.0.gir: gedit/gedit$(EXEEXT)
 INTROSPECTION_SCANNER_ARGS = -I$(top_srcdir) --warn-all
-- 
1.9.3

