From 8fe6c730443b0e9255ce8842880acb1927b547e1 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Tue, 22 Jul 2014 11:25:53 +0200
Subject: [PATCH] Fix the windows build

---
 gedit/gedit.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gedit/gedit.c b/gedit/gedit.c
index 14e4bf2..5d2806a 100644
--- a/gedit/gedit.c
+++ b/gedit/gedit.c
@@ -44,7 +44,7 @@ static GModule *libgedit_dll = NULL;
 /* This code must live in gedit.exe, not in libgedit.dll, since the whole
  * point is to find and load libgedit.dll.
  */
-static void
+static gboolean
 gedit_w32_load_private_dll (void)
 {
 	gchar *dllpath;
@@ -73,7 +73,7 @@ gedit_w32_load_private_dll (void)
 		libgedit_dll = g_module_open ("libgedit.dll", 0);
 	}
 
-	return (libgedit_dll != NULL)
+	return (libgedit_dll != NULL);
 }
 
 static void
@@ -100,7 +100,7 @@ main (int argc, char *argv[])
 #ifdef G_OS_WIN32
 	if (!gedit_w32_load_private_dll ())
 	{
-		g_printerr ("Failed to load libgedit DLL\n")
+		g_printerr ("Failed to load libgedit DLL\n");
 		return 1;
 	}
 
-- 
1.9.3

