diff -Naur a/gedit/Makefile.am b/gedit/Makefile.am
--- a/gedit/Makefile.am
+++ b/gedit/Makefile.am
@@ -266,7 +266,7 @@
 gedit_Gedit_3_0_gir_PROGRAM = $(builddir)/gedit/gedit$(EXEEXT)
 gedit_Gedit_3_0_gir_INCLUDES = Gtk-3.0 GtkSource-3.0
 gedit_Gedit_3_0_gir_EXPORT_PACKAGES = gedit
-gedit_Gedit_3_0_gir_SCANNERFLAGS = $(foreach header,$(gedit_INST_H_FILES),--c-include="$(header)")
+gedit_Gedit_3_0_gir_SCANNERFLAGS = $(GEDIT_CFLAGS) $(foreach header,$(gedit_INST_H_FILES),--c-include="$(header)")
 gedit_Gedit_3_0_gir_FILES = 		\
 	$(gedit_INST_H_FILES) 		\
 	$(gedit_libgedit_c_files)	\
