--- gedit-3.32.2/plugins/checkupdate/meson.build.orig	2019-05-15 02:59:29.000000000 +0200
+++ gedit-3.32.2/plugins/checkupdate/meson.build	2019-06-05 09:36:47.260881700 +0200
@@ -24,6 +24,17 @@
   )
 )
 
+configure_file(
+  input: 'org.gnome.gedit.plugins.checkupdate.gschema.xml.in',
+  output: 'org.gnome.gedit.plugins.checkupdate.gschema.xml',
+  configuration: gschema_xml,
+  install: true,
+  install_dir: join_paths(
+    glibdir,
+    'schemas',
+  )
+)
+
 custom_target(
   'checkupdate.plugin',
   input: 'checkupdate.plugin.desktop.in',
