--- nanogui/CMakeLists.txt.orig	2025-06-16 14:56:58.689929800 +0300
+++ nanogui/CMakeLists.txt	2025-06-16 14:57:03.182056800 +0300
@@ -79,7 +79,7 @@
 # default. You may want to set MTS_NATIVE_FLAGS to "-march=native" to use all
 # instructions available on a particular machine.
 
-if (MSVC)
+if (WIN32)
   set(NANOGUI_NATIVE_FLAGS_DEFAULT "")
 elseif (APPLE AND ((CMAKE_OSX_ARCHITECTURES MATCHES "arm64") OR
                    (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "arm64")))
