diff -ur httplib2-0.8.orig/python2/httplib2/__init__.py httplib2-0.8/python2/httplib2/__init__.py
--- httplib2-0.8.orig/python2/httplib2/__init__.py	2014-05-01 23:25:52.459852279 +0200
+++ httplib2-0.8/python2/httplib2/__init__.py	2014-05-01 23:28:08.822391994 +0200
@@ -190,9 +190,8 @@
     import ca_certs_locater
     CA_CERTS = ca_certs_locater.get()
 except ImportError:
-    # Default CA certificates file bundled with httplib2.
-    CA_CERTS = os.path.join(
-        os.path.dirname(os.path.abspath(__file__ )), "cacerts.txt")
+    # Use system CA certificates
+    CA_CERTS = os.path.join(sys.exec_prefix, "ssl", "certs", "ca-bundle.crt")
 
 # Which headers are hop-by-hop headers by default
 HOP_BY_HOP = ['connection', 'keep-alive', 'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'transfer-encoding', 'upgrade']
diff -ur httplib2-0.8.orig/python3/httplib2/__init__.py httplib2-0.8/python3/httplib2/__init__.py
--- httplib2-0.8.orig/python3/httplib2/__init__.py	2013-03-06 21:40:54.000000000 +0100
+++ httplib2-0.8/python3/httplib2/__init__.py	2014-05-01 23:28:36.822913468 +0200
@@ -123,9 +123,8 @@
 # Which headers are hop-by-hop headers by default
 HOP_BY_HOP = ['connection', 'keep-alive', 'proxy-authenticate', 'proxy-authorization', 'te', 'trailers', 'transfer-encoding', 'upgrade']
 
-# Default CA certificates file bundled with httplib2.
-CA_CERTS = os.path.join(
-        os.path.dirname(os.path.abspath(__file__ )), "cacerts.txt")
+# Use system CA certificates
+CA_CERTS = os.path.join(sys.exec_prefix, "ssl", "certs", "ca-bundle.crt")
 
 def _get_end2end_headers(response):
     hopbyhop = list(HOP_BY_HOP)
