diff -ur httplib2-0.8.orig/python2/httplib2/certs.py httplib2-0.8/python2/httplib2/certs.py
--- httplib2-0.8.orig/python2/httplib2/certs.py	2014-05-01 23:25:52.459852279 +0200
+++ httplib2-0.8/python2/httplib2/certs.py	2014-05-01 23:28:08.822391994 +0200
@@ -1,6 +1,7 @@
 """Utilities for certificate management."""
 
 import os
+import sys
 
 certifi_available = False
 certifi_where = None
@@ -20,7 +20,7 @@
 
 
 BUILTIN_CA_CERTS = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "cacerts.txt"
+    sys.exec_prefix, "ssl", "certs", "ca-bundle.crt"
 )
 
 
diff -ur httplib2-0.8.orig/python3/httplib2/certs.py httplib2-0.8/python3/httplib2/certs.py
--- httplib2-0.8.orig/python3/httplib2/certs.py	2013-03-06 21:40:54.000000000 +0100
+++ httplib2-0.8/python3/httplib2/certs.py	2014-05-01 23:28:36.822913468 +0200
@@ -1,6 +1,7 @@
 """Utilities for certificate management."""
 
 import os
+import sys
 
 certifi_available = False
 certifi_where = None
@@ -20,7 +20,7 @@
 
 
 BUILTIN_CA_CERTS = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "cacerts.txt"
+    sys.exec_prefix, "ssl", "certs", "ca-bundle.crt"
 )
 
 
