diff -urN a/config/baseconfig.mk b/config/baseconfig.mk
--- a/config/baseconfig.mk	2015-07-13 22:05:44.022749700 +0100
+++ b/config/baseconfig.mk	2015-07-13 22:16:09.001405100 +0100
@@ -31,9 +31,6 @@
 $(error Pymake is no longer supported. Please upgrade to MozillaBuild 1.9 or newer and build with 'mach' or 'mozmake')
 endif
 
-ifeq (a,$(firstword a$(subst /, ,$(abspath .))))
-$(error MSYS make is not supported)
-endif
 # 4.0- happens to be greater than 4.0, lower than the mozmake version,
 # and lower than 4.0.1 or 4.1, whatever next version of gnu make will
 # be released.
--- firefox-39.0/config/baseconfig.mk.orig	2015-07-14 21:57:46.627572700 +0100
+++ firefox-39.0/config/baseconfig.mk	2015-07-14 22:02:47.929566800 +0100
@@ -38,11 +38,6 @@
 $(error Make version too old. Only versions strictly greater than 4.0 are supported.)
 endif
 
-ifdef INCLUDED_AUTOCONF_MK
-ifeq (a,$(firstword a$(subst /, ,$(srcdir))))
-$(error MSYS-style srcdir are not supported for Windows builds.)
-endif
-endif
 endif # WINNT
 
 include_deps = $(eval $(if $(2),,-)include $(1))
