--- firefox-39.0/security/sandbox/chromium/base/strings/utf_string_conversion_utils.cc.orig	2015-06-30 23:57:54.000000000 +0100
+++ firefox-39.0/security/sandbox/chromium/base/strings/utf_string_conversion_utils.cc	2015-07-19 20:25:24.348811900 +0100
@@ -122,7 +122,9 @@
 
 // Instantiate versions we know callers will need.
 template void PrepareForUTF8Output(const wchar_t*, size_t, std::string*);
+#if !defined(WCHAR_T_IS_UTF16)
 template void PrepareForUTF8Output(const char16*, size_t, std::string*);
+#endif
 
 template<typename STRING>
 void PrepareForUTF16Or32Output(const char* src,
@@ -143,6 +145,8 @@
 
 // Instantiate versions we know callers will need.
 template void PrepareForUTF16Or32Output(const char*, size_t, std::wstring*);
+#if !defined(WCHAR_T_IS_UTF16)
 template void PrepareForUTF16Or32Output(const char*, size_t, string16*);
+#endif
 
 }  // namespace base
