--- firefox-39.0/security/sandbox/chromium/base/win/scoped_handle.h.orig	2015-07-19 20:52:07.176488400 +0100
+++ firefox-39.0/security/sandbox/chromium/base/win/scoped_handle.h	2015-07-19 20:57:44.812800100 +0100
@@ -18,8 +18,7 @@
 #include <intrin.h>
 #define BASE_WIN_GET_CALLER _ReturnAddress()
 #elif defined(COMPILER_GCC)
-#define BASE_WIN_GET_CALLER __builtin_extract_return_addr(\\
-    __builtin_return_address(0))
+#define BASE_WIN_GET_CALLER __builtin_extract_return_addr(__builtin_return_address(0))
 #endif
 
 namespace base {
