--- firefox-39.0/security/sandbox/chromium/base/time/time_win.cc.orig	2015-06-30 23:57:54.000000000 +0100
+++ firefox-39.0/security/sandbox/chromium/base/time/time_win.cc	2015-07-19 23:01:28.117321100 +0100
@@ -334,7 +334,7 @@
   // we keep last_seen_now stay correctly in sync.
   DWORD now = tick_function();
   if (now < last_seen_now)
-    rollover_ms += 0x100000000I64;  // ~49.7 days.
+    rollover_ms += 0x100000000ll;  // ~49.7 days.
   last_seen_now = now;
   return TimeDelta::FromMilliseconds(now + rollover_ms);
 }
