diff -urN firefox-39.0.orig/media/libstagefright/moz.build firefox-39.0/media/libstagefright/moz.build
--- firefox-39.0.orig/media/libstagefright/moz.build	2015-06-30 23:57:58.000000000 +0100
+++ firefox-39.0/media/libstagefright/moz.build	2015-07-20 18:53:20.523932000 +0100
@@ -16,9 +16,10 @@
         DEFINES['off64_t'] = 'int64_t'
         DEFINES['strcasecmp'] = 'stricmp'
         DEFINES['strncasecmp'] = 'strnicmp'
+        LOCAL_INCLUDES += [ 'ports/win32/include' ]
     DEFINES['HAVE_MS_C_RUNTIME'] = True
     DEFINES['__PRETTY_FUNCTION__'] = '__FUNCTION__'
-    LOCAL_INCLUDES += [ 'ports/win32/include' ]
+    LOCAL_INCLUDES += [ 'ports/mingw-w64/include' ]
 elif CONFIG['OS_TARGET'] == 'Darwin':
     DEFINES['HAVE_SYS_UIO_H'] = True
     DEFINES['off64_t'] = 'off_t'
diff -urN firefox-39.0.orig/media/libstagefright/ports/mingw-w64/include/arpa/inet.h firefox-39.0/media/libstagefright/ports/mingw-w64/include/arpa/inet.h
--- firefox-39.0.orig/media/libstagefright/ports/mingw-w64/include/arpa/inet.h	1970-01-01 01:00:00.000000000 +0100
+++ firefox-39.0/media/libstagefright/ports/mingw-w64/include/arpa/inet.h	2015-07-20 19:55:44.843115100 +0100
@@ -0,0 +1,9 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+#ifndef INET_H_
+#define INET_H_
+
+#include <netinet/in.h>
+
+#endif
diff -urN firefox-39.0.orig/media/libstagefright/ports/mingw-w64/include/byteswap.h firefox-39.0/media/libstagefright/ports/mingw-w64/include/byteswap.h
--- firefox-39.0.orig/media/libstagefright/ports/mingw-w64.orig/include/byteswap.h	1970-01-01 01:00:00.000000000 +0100
+++ firefox-39.0/media/libstagefright/ports/mingw-w64/include/byteswap.h	2015-07-20 20:22:01.796787900 +0100
@@ -0,0 +1,10 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+#ifndef BYTESWAP_H_
+#define BYTESWAP_H_
+
+#include <stdlib.h>
+#define bswap_16 _byteswap_ushort
+
+#endif
diff -urN firefox-39.0.orig/media/libstagefright/ports/mingw-w64/include/netinet/in.h firefox-39.0/media/libstagefright/ports/mingw-w64/include/netinet/in.h
--- firefox-39.0.orig/media/libstagefright/ports/mingw-w64/include/netinet/in.h	1970-01-01 01:00:00.000000000 +0100
+++ firefox-39.0/media/libstagefright/ports/mingw-w64/include/netinet/in.h	2015-07-20 19:55:44.858714700 +0100
@@ -0,0 +1,39 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+#ifndef IN_H_
+#define IN_H_
+
+#include <stdint.h>
+
+#if defined(_M_IX86) || defined(_M_AMD64)
+
+static uint32_t
+ntohl(uint32_t x)
+{
+  return x << 24 | (x << 8 & 0xff0000) | (x >> 8 & 0xff00) | x >> 24;
+}
+
+static uint16_t
+ntohs(uint16_t x)
+{
+  return x << 8 | x >> 8;
+}
+
+static uint32_t
+htonl(uint32_t x)
+{
+  return x << 24 | (x << 8 & 0xff0000) | (x >> 8 & 0xff00) | x >> 24;
+}
+
+static uint16_t
+htons(uint16_t x)
+{
+  return x << 8 | x >> 8;
+}
+
+#else
+#error Unsupported architecture
+#endif
+
+#endif
