--- firefox-41.0b5.orig/mozglue/misc/TimeStamp.h	2015-08-30 11:46:11.745214500 +0100
+++ firefox-41.0b5/mozglue/misc/TimeStamp.h	2015-08-30 11:46:41.480915300 +0100
@@ -426,8 +426,8 @@
    * lower precision, usually 15.6 ms, but with very good performance benefit.
    * Use it for measurements of longer times, like >200ms timeouts.
    */
-  static MFBT_API TimeStamp Now() { return Now(true); }
-  static MFBT_API TimeStamp NowLoRes() { return Now(false); }
+  static TimeStamp Now() { return Now(true); }
+  static TimeStamp NowLoRes() { return Now(false); }
 
   /**
    * Return a timestamp representing the time when the current process was
