--- firefox-41.0b5/widget/windows/WindowsUIUtils.cpp.orig	2015-08-30 16:03:42.591416200 +0100
+++ firefox-41.0b5/widget/windows/WindowsUIUtils.cpp	2015-08-30 16:03:45.043620700 +0100
@@ -20,6 +20,41 @@
 #include "nsString.h"
 #include "nsIWidget.h"
 
+#if defined(__MINGW64_VERSION_MAJOR)
+/* MinGW-w64 doesn't have support for this WRL stuff yet:
+   https://msdn.microsoft.com/en-us/library/hh438466(v=vs.110).aspx */
+
+WindowsUIUtils::WindowsUIUtils() :
+  mInTabletMode(eTabletModeUnknown)
+{
+}
+
+WindowsUIUtils::~WindowsUIUtils()
+{
+}
+
+/*
+ * Implement the nsISupports methods...
+ */
+NS_IMPL_ISUPPORTS(WindowsUIUtils,
+                  nsIWindowsUIUtils)
+
+NS_IMETHODIMP
+WindowsUIUtils::GetInTabletMode(bool* aResult)
+{
+  *aResult = false;
+  return NS_OK;
+}
+
+NS_IMETHODIMP
+WindowsUIUtils::UpdateTabletModeState()
+{
+  return NS_OK;
+}
+
+
+
+#else
 #include <windows.ui.viewmanagement.h>
 
 #pragma comment(lib, "runtimeobject.lib")
@@ -172,3 +207,4 @@
   return NS_OK;
 }
 
+#endif /* defined(__MINGW64_VERSION_MAJOR) */
