--- firefox-41.0b5/toolkit/mozapps/installer/packager.mk.orig	2015-08-27 22:30:13.000000000 +0100
+++ firefox-41.0b5/toolkit/mozapps/installer/packager.mk	2015-08-31 23:47:20.134091700 +0100
@@ -116,9 +116,9 @@
 # dist/sdk/lib -> prefix/lib/appname-devel-version/lib
 # prefix/lib/appname-devel-version/* symlinks to the above directories
 install:: prepare-package
-ifeq ($(OS_ARCH),WINNT)
-	$(error "make install" is not supported on this platform. Use "make package" instead.)
-endif
+# ifeq ($(OS_ARCH),WINNT)
+# 	$(error "make install" is not supported on this platform. Use "make package" instead.)
+# endif
 ifeq (bundle,$(MOZ_FS_LAYOUT))
 	$(error "make install" is not supported on this platform. Use "make package" instead.)
 endif
