--- firefox-41.0b5/xpcom/build/PoisonIOInterposerWin.cpp	2015-09-01 00:07:54.085669600 +0100
+++ firefox-41.0b5/xpcom/build/PoisonIOInterposerWin.cpp.orig	2015-09-01 00:06:43.475631000 +0100
@@ -445,7 +445,11 @@
 
   // Stdout and Stderr are OK.
   MozillaRegisterDebugFD(1);
-  MozillaRegisterDebugFD(2);
+  // Avoid a MOZ_ASSERT() when launching from MSYS2's mintty:
+  // https://bugzilla.mozilla.org/show_bug.cgi?id=1163452
+  if (_get_osfhandle(1) != _get_osfhandle(2)) {
+    MozillaRegisterDebugFD(2);
+  }
 
 #ifdef MOZ_REPLACE_MALLOC
   // The contract with InitDebugFd is that the given registry can be used
