--- firefox-42.0/dom/media/platforms/wmf/WMFVideoMFTManager.cpp.orig	2015-10-29 22:17:56.000000000 +0000
+++ firefox-42.0/dom/media/platforms/wmf/WMFVideoMFTManager.cpp	2015-11-25 00:04:09.460865800 +0000
@@ -30,7 +30,7 @@
 using mozilla::layers::LayerManager;
 using mozilla::layers::LayersBackend;
 
-#if MOZ_WINSDK_MAXVER < 0x0A000000
+#if MOZ_WINSDK_MAXVER < 0x0A000000 && !defined(__MINGW64_VERSION_MAJOR)
 // Windows 10+ SDK has VP80 and VP90 defines
 const GUID MFVideoFormat_VP80 =
 {
@@ -39,6 +39,7 @@
   0x0010,
   {0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71}
 };
+#endif /* MOZ_WINSDK_MAXVER < 0x0A000000 && !defined(__MINGW64_VERSION_MAJOR) */
 
 const GUID MFVideoFormat_VP90 =
 {
