diff -urN a/python/mozbuild/mozbuild/frontend/emitter.py b/python/mozbuild/mozbuild/frontend/emitter.py
--- a/python/mozbuild/mozbuild/frontend/emitter.py	2015-06-30 23:58:07.000000000 +0100
+++ b/python/mozbuild/mozbuild/frontend/emitter.py	2015-07-13 21:43:47.955204200 +0100
@@ -3,6 +3,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from __future__ import unicode_literals
+from __future__ import print_function
 
 import itertools
 import json
@@ -672,6 +673,8 @@
     def _process_sources(self, context, passthru):
         for symbol in ('SOURCES', 'HOST_SOURCES', 'UNIFIED_SOURCES'):
             for src in (context[symbol] or []):
+                print('sys.argv is %s' % sys.argv, file=sys.stderr)
+                print('checking if %s exists' % (mozpath.join(context.srcdir, src)), file=sys.stderr)
                 if not os.path.exists(mozpath.join(context.srcdir, src)):
                     raise SandboxValidationError('File listed in %s does not '
                         'exist: \'%s\'' % (symbol, src), context)
diff -urN a/python/mozbuild/mozbuild/config_status.py b/python/mozbuild/mozbuild/config_status.py
--- a/python/mozbuild/mozbuild/config_status.py	2015-06-30 23:58:07.000000000 +0100
+++ b/python/mozbuild/mozbuild/config_status.py	2015-07-14 20:21:48.954130100 +0100
@@ -145,7 +145,7 @@
     log_manager.add_terminal_logging(level=log_level)
     log_manager.enable_unstructured()
 
-    print('Reticulating splines...', file=sys.stderr)
+    print('Reticulating splines... sys.argv = %s' % sys.argv, file=sys.stderr)
     summary = the_backend.consume(definitions)
 
     for line in summary.summaries():
