diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user kiconthemes-6.1.0.orig/src/kicontheme.cpp kiconthemes-6.1.0/src/kicontheme.cpp
--- kiconthemes-6.1.0.orig/src/kicontheme.cpp	2024-04-05 12:56:43.000000000 +0200
+++ kiconthemes-6.1.0/src/kicontheme.cpp	2024-04-15 11:51:58.968290800 +0200
@@ -45,8 +45,8 @@
 // For this reason we use AppDataLocation: BINDIR/data on Windows, Resources on OS X
 void initRCCIconTheme()
 {
-#ifdef WITH_BREEZEICONS
-    BreezeIcons::initIcons()
+#ifdef WITH_BREEZEICONS_LIB
+    BreezeIcons::initIcons();
 #else
     const QString iconThemeRcc = QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("icontheme.rcc"));
     if (!iconThemeRcc.isEmpty()) {
