diff --git a/src/kiconengineplugin.json b/src/kiconengineplugin.json
index bc09e41..a5b4228 100644
--- a/src/kiconengineplugin.json
+++ b/src/kiconengineplugin.json
@@ -1,4 +1,4 @@
 {
-    "Keys": [ "KIconEngine" ]
+    "Keys": [ "KIconEngine", "svg", "svgz", "svg.gz" ]
 }
 
diff --git a/src/kiconloader.cpp b/src/kiconloader.cpp
index 3cebfc6..0eded28 100644
--- a/src/kiconloader.cpp
+++ b/src/kiconloader.cpp
@@ -147,6 +147,8 @@ QString KIconThemeNode::findIcon(const QString &name, int size, KIconLoader::Mat
 extern KICONTHEMES_EXPORT int kiconloader_ms_between_checks;
 KICONTHEMES_EXPORT int kiconloader_ms_between_checks = 5000;
 
+extern void initRCCIconTheme();
+
 class KIconLoaderGlobalData : public QObject
 {
     Q_OBJECT
@@ -154,6 +156,16 @@ class KIconLoaderGlobalData : public QObject
 public:
     KIconLoaderGlobalData()
     {
+        // ensure we load rcc files for application bundles (+ setup their theme)
+        initRCCIconTheme();
+
+        // Set the icon theme fallback to breeze
+        // Most of our apps use "lots" of icons that most of the times
+        // are only available with breeze, we still honour the user icon
+        // theme but if the icon is not found there, we go to breeze
+        // since it's almost sure it'll be there
+        QIcon::setFallbackThemeName(QStringLiteral("breeze"));
+
         const QStringList genericIconsFiles = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("mime/generic-icons"));
         // qCDebug(KICONTHEMES) << genericIconsFiles;
         for (const QString &file : genericIconsFiles) {
diff --git a/src/kicontheme.cpp b/src/kicontheme.cpp
index 4eee2ff..e94f10c 100644
--- a/src/kicontheme.cpp
+++ b/src/kicontheme.cpp
@@ -61,32 +61,7 @@ void initRCCIconTheme()
         }
     }
 }
-Q_COREAPP_STARTUP_FUNCTION(initRCCIconTheme)
-
-// Makes sure the icon theme fallback is set to breeze or one of its
-// variants. Most of our apps use "lots" of icons that most of the times
-// are only available with breeze, we still honour the user icon theme
-// but if the icon is not found there, we go to breeze since it's almost
-// sure it'll be there
-static void setBreezeFallback()
-{
-    if (const QPlatformTheme *theme = QGuiApplicationPrivate::platformTheme()) {
-        const QVariant themeHint = theme->themeHint(QPlatformTheme::SystemIconFallbackThemeName);
-        if (themeHint.isValid()) {
-            const QString iconTheme = themeHint.toString();
-            if (iconTheme.contains(QStringLiteral("breeze"), Qt::CaseInsensitive)) {
-                QIcon::setFallbackThemeName(iconTheme);
-                return;
-            }
-        }
-    }
 
-    QIcon::setFallbackThemeName(QStringLiteral("breeze"));
-}
-
-#ifndef Q_OS_ANDROID
-Q_COREAPP_STARTUP_FUNCTION(setBreezeFallback)
-#endif
 class KIconThemeDir;
 class KIconThemePrivate
 {
