diff -u -r webkitgtk-1.10.2/Source/ThirdParty/ANGLE/src/compiler/glslang.y webkitgtk-1.10.2-bison3/Source/ThirdParty/ANGLE/src/compiler/glslang.y
--- webkitgtk-1.10.2/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2012-11-23 21:12:16.000000000 +0100
+++ webkitgtk-1.10.2-bison3/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2013-10-12 20:42:54.531301735 +0200
@@ -47,6 +47,7 @@
 %expect 1 /* One shift reduce conflict because of if | else */
 %pure-parser
 %parse-param {TParseContext* context}
+%lex-param {YYLEX_PARAM}
 
 %union {
     struct {
