--- webkitgtk-1.10.2/configure.ac	2012-12-09 19:01:04.000000000 +0100
+++ webkitgtk-1.10.2/configure.ac	2013-05-07 08:27:18.636262304 +0200
@@ -1207,8 +1207,8 @@
 
 # Add '-g' flag to gcc to build with debug symbols
 if test "$enable_debug_symbols" = "yes"; then
-   CXXFLAGS="$CXXFLAGS -g"
-   CFLAGS="$CFLAGS -g"
+   CXXFLAGS="$CXXFLAGS -g1"
+   CFLAGS="$CFLAGS -g1"
 fi
 
 if test "$enable_debug_features" = "no"; then
--- webkitgtk-1.10.2/GNUmakefile.am	2012-11-27 13:27:36.000000000 +0100
+++ webkitgtk-1.10.2/GNUmakefile.am	2013-03-21 11:24:25.466255256 +0100
@@ -289,8 +289,6 @@
 include Source/WebKit2/UIProcess/API/gtk/tests/GNUmakefile.am
 include Tools/MiniBrowser/gtk/GNUmakefile.am
 include Tools/WebKitTestRunner/GNUmakefile.am
-include Source/ThirdParty/gtest/GNUmakefile.am
-include Tools/TestWebKitAPI/GNUmakefile.am
 # [GTK] Refactor the translations now that we have webkit2
 # https://bugs.webkit.org/show_bug.cgi?id=55153
 
--- webkitgtk-1.10.2/Tools/GNUmakefile.am	2012-11-23 21:13:23.000000000 +0100
+++ webkitgtk-1.10.2/Tools/GNUmakefile.am	2013-03-21 11:24:50.388662220 +0100
@@ -1,5 +1,4 @@
-noinst_PROGRAMS += \
-	Programs/DumpRenderTree \
+bin_PROGRAMS += \
 	Programs/GtkLauncher
 
 # GtkLauncher
@@ -343,7 +342,6 @@
 	  fi; \
 	fi
 endif
-	@$(AM_V_GEN)$(PYTHON) $(srcdir)/Tools/gtk/generate-gtkdoc --rebase --virtual-root=$${DESTDIR}
 
 uninstall-local:
 	@DOC_MODULE_VERSION=`cat ./Documentation/webkitgtk/version.xml`; \
