--- libsass-python-0.22.0/setup.py	2022-11-12 20:43:54.000000000 +0300
+++ python-build-MINGW32/setup.py	2023-06-05 22:21:34.989558300 +0300
@@ -15,7 +15,7 @@
 
 MACOS_FLAG = ['-mmacosx-version-min=10.7']
 FLAGS_POSIX = [
-    '-fPIC', '-std=gnu++0x', '-Wall', '-Wno-parentheses', '-Werror=switch',
+    '-fPIC', '-Wall', '-Wno-parentheses', '-Werror=switch',
 ]
 FLAGS_CLANG = ['-c', '-O3'] + FLAGS_POSIX + ['-stdlib=libc++']
 LFLAGS_POSIX = ['-fPIC',  '-lstdc++']
@@ -24,7 +24,7 @@
 sources = ['_sass.c']
 headers = []
 
-if sys.platform == 'win32':
+if sys.platform == 'win32' and not 'MSYSTEM' in os.environ:
     extra_compile_args = ['/Od', '/EHsc', '/MT']
     extra_link_args = []
 elif platform.system() == 'Darwin':
@@ -79,7 +79,7 @@
     # The version file should always exist at this point
     with open('.libsass-upstream-version', 'rb') as libsass_version_file:
         libsass_version = libsass_version_file.read().decode('UTF-8').strip()
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' and not 'MSYSTEM' in os.environ:
             # This looks wrong, but is required for some reason :(
             define = fr'/DLIBSASS_VERSION="\"{libsass_version}\""'
         else:
@@ -97,7 +97,7 @@
                 elif filename.endswith('.h'):
                     headers.append(filename)
 
-    if sys.platform == 'win32':
+    if sys.platform == 'win32' and not 'MSYSTEM' in os.environ:
         from distutils.msvc9compiler import get_build_version
         vscomntools_env = 'VS{}{}COMNTOOLS'.format(
             int(get_build_version()),
