diff -bur liblangtag-0.6.7-o/liblangtag/lt-macros.h liblangtag-0.6.7/liblangtag/lt-macros.h
--- liblangtag-0.6.7-o/liblangtag/lt-macros.h	2025-04-16 22:56:11.121608100 -0600
+++ liblangtag-0.6.7/liblangtag/lt-macros.h	2025-04-16 22:58:07.409723000 -0600
@@ -320,6 +320,9 @@
 #elif defined (__alpha__) && !defined(__osf__) && defined (__GNUC__) && __GNUC__ >= 2
 #  define LT_BREAKPOINT()						\
 	LT_STMT_START {__asm__ __volatile__ ("bpt");} LT_STMT_END
+#elif defined (__aarch64__)
+#  define LT_BREAKPOINT()						\
+	LT_STMT_START {__asm__ volatile(".inst 0xd4200000");} LT_STMT_END
 #else
 #include <signal.h>
 #  define LT_BREAKPOINT()				\
