Description: Fix some Py2 syntax errors
Author: Thomas Goirand <zigo@debian.org>
Forwarded: no
Last-Update: 2015-07-08

--- python-linecache2-1.0.0.orig/linecache2/tests/inspect_fodder2.py
+++ python-linecache2-1.0.0/linecache2/tests/inspect_fodder2.py
@@ -99,15 +99,15 @@ def f():
 method_in_dynamic_class = f().g
 
 #line 101
-def keyworded(*arg1, arg2=1):
+def keyworded(arg1, arg2=1):
     pass
 
 #line 105
-def annotated(arg1: list):
+def annotated(arg1):
     pass
 
 #line 109
-def keyword_only_arg(*, arg):
+def keyword_only_arg(arg):
     pass
 
 from functools import wraps
