--- a/setup.py
+++ b/setup.py
@@ -1004,1 +1004,1 @@
-        libs = ['mupdf']
+        libs = ['mupdf', 'jbig2dec', 'openjp2', 'jpeg', 'png', 'z', 'freetype', 'gumbo', 'harfbuzz']
@@ -1018,2 +1018,0 @@
-            if libpaths:
-                libpaths = libpaths.split(':')
@@ -1029,4 +1027,7 @@
         includes = list()
         pi = os.environ.get('PYMUPDF_INCLUDES')
+        mingw_prefix = os.getenv("MINGW_PREFIX", "/ucrt64")
+        includes.append(f'{mingw_prefix}/include')
+        includes.append(f'{mingw_prefix}/include/freetype2')
         if pi:
             includes += pi.split(':')
--- a/src/extra.i
+++ b/src/extra.i
@@ -228,1 +228,1 @@
-    PyObject* text_py = PyString_FromString(text);
+    PyObject* text_py = PyUnicode_FromString(text);
