diff -aur ImageMagick-7.0.1-3/MagickCore/distribute-cache.c.orig ImageMagick-7.0.1-3/MagickCore/distribute-cache.c
--- ImageMagick-7.0.1-3/MagickCore/distribute-cache.c.orig	2016-05-11 18:00:52 -0400
+++ ImageMagick-7.0.1-3/MagickCore/distribute-cache.c	2016-05-11 18:02:30 -0400
@@ -88,6 +88,8 @@
 #define LENGTH_TYPE size_t
 #define MAGICKCORE_HAVE_DISTRIBUTE_CACHE
 #elif defined(MAGICKCORE_WINDOWS_SUPPORT)
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #define CHAR_TYPE_CAST (char *)
 #define CLOSE_SOCKET(socket) (void) closesocket(socket)
 #define HANDLER_RETURN_TYPE DWORD WINAPI
diff -aur ImageMagick-7.0.4-1/MagickCore/nt-base.h.orig ImageMagick-7.0.4-1/MagickCore/nt-base.h
--- ImageMagick-7.0.4-1/MagickCore/nt-base.h.orig	2016-12-31 08:49:06.000000000 -0500
+++ ImageMagick-7.0.4-1/MagickCore/nt-base.h	2017-01-03 00:08:24.744499200 -0500
@@ -39,6 +39,8 @@
 #include <errno.h>
 #include <malloc.h>
 #include <sys/utime.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #if defined(_DEBUG) && !defined(__MINGW32__) && !defined(__MINGW64__)
 #include <crtdbg.h>
 #endif
@@ -95,7 +97,7 @@
 #if !defined(closedir)
 #  define closedir(directory)  NTCloseDirectory(directory)
 #endif
-#define MAGICKCORE_HAVE_ERF
+//#define MAGICKCORE_HAVE_ERF
 #if defined(_VISUALC_) && (_MSC_VER < 1700)
 #  define erf(x)  NTErf(x)
 #endif
diff -aur ImageMagick-7.0.1-3/MagickCore/nt-base.c.orig ImageMagick-7.0.1-3/MagickCore/nt-base.c > patch
--- ImageMagick-7.0.4-1/MagickCore/nt-base.c.orig	2017-01-02 23:52:31.225966200 -0500
+++ ImageMagick-7.0.4-1/MagickCore/nt-base.c	2017-01-03 00:06:00.213705700 -0500
@@ -1752,12 +1752,16 @@
   wchar_t
     file_specification[MagickPathExtent];
 
+  wchar_t WCDirectorySeparator[strlen(DirectorySeparator)+1];
+  MultiByteToWideChar(CP_UTF8,0,DirectorySeparator,-1,WCDirectorySeparator,
+                     strlen(DirectorySeparator)+1);
+
   assert(path != (const char *) NULL);
   length=MultiByteToWideChar(CP_UTF8,0,path,-1,file_specification,
     MagickPathExtent);
   if (length == 0)
     return((DIR *) NULL);
-  if (wcsncat(file_specification,(const wchar_t*) DirectorySeparator,
+  if(wcsncat(file_specification, WCDirectorySeparator,
         MagickPathExtent-wcslen(file_specification)-1) == (wchar_t*) NULL)
     return((DIR *) NULL);
   entry=(DIR *) AcquireMagickMemory(sizeof(DIR));

