--- ImageMagick-6.9.0-4/magick/distribute-cache.c.orig	2015-01-18 16:07:04.357600000 +0300
+++ ImageMagick-6.9.0-4/magick/distribute-cache.c	2015-01-18 16:11:40.668800000 +0300
@@ -86,6 +86,8 @@
 #define SOCKET_TYPE int
 #define MAGICKCORE_HAVE_DISTRIBUTE_CACHE
 #elif defined(MAGICKCORE_WINDOWS_SUPPORT)
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #define CHAR_TYPE_CAST (char *)
 #define CLOSE_SOCKET(socket) (void) closesocket(socket)
 #define HANDLER_RETURN_TYPE DWORD WINAPI
--- ImageMagick-6.9.0-4/magick/nt-base.h.orig	2015-01-18 16:39:35.182400000 +0300
+++ ImageMagick-6.9.0-4/magick/nt-base.h	2015-01-18 16:39:37.740800000 +0300
@@ -38,6 +38,8 @@
 #include <process.h>
 #include <errno.h>
 #include <malloc.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #if defined(_DEBUG) && !defined(__MINGW32__) && !defined(__MINGW64__)
 #include <crtdbg.h>
 #endif
--- ImageMagick-6.9.0-4/magick/nt-base.c.orig	2015-02-22 22:02:34.000000000 +0100
+++ ImageMagick-6.9.0-4/magick/nt-base.c	2015-05-12 13:31:02.627609000 +0200
@@ -1625,12 +1625,16 @@
   wchar_t
     file_specification[MaxTextExtent];

+  wchar_t WCDirectorySeparator[strlen(DirectorySeparator)+1];
+  MultiByteToWideChar(CP_UTF8,0,DirectorySeparator,-1,WCDirectorySeparator,
+                     strlen(DirectorySeparator)+1);
+
   assert(path != (const char *) NULL);
   length=MultiByteToWideChar(CP_UTF8,0,path,-1,file_specification,
     MaxTextExtent);
   if (length == 0)
     return((DIR *) NULL);
-  if(wcsncat(file_specification,(const wchar_t*) DirectorySeparator,
+  if(wcsncat(file_specification, WCDirectorySeparator,
        MaxTextExtent-wcslen(file_specification)-1) == (wchar_t*) NULL)
     return((DIR *) NULL);
   entry=(DIR *) AcquireMagickMemory(sizeof(DIR));
--- ImageMagick-6.9.1-2/magick/locale.c.orig	2015-05-12 14:33:17.733600000 +0300
+++ ImageMagick-6.9.1-2/magick/locale.c	2015-05-12 14:33:31.274400000 +0300
@@ -254,7 +254,7 @@
 #if defined(MAGICKCORE_HAVE_NEWLOCALE)
   if (c_locale != (locale_t) NULL)
     freelocale(c_locale);
-#elif defined(MAGICKCORE_WINDOWS_SUPPORT)
+#elif defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__MINGW32__)
   if (c_locale != (locale_t) NULL)
     _free_locale(c_locale);
 #endif
