--- ImageMagick-7.0.8-48/MagickCore/locale.c.orig	2019-06-04 10:15:56.942575100 +0300
+++ ImageMagick-7.0.8-48/MagickCore/locale.c	2019-06-04 10:16:13.172597800 +0300
@@ -66,7 +66,7 @@
 /*
   Define declarations.
 */
-#if defined(MAGICKCORE_HAVE_NEWLOCALE) || defined(MAGICKCORE_WINDOWS_SUPPORT)
+#if (defined(MAGICKCORE_HAVE_NEWLOCALE) || defined(MAGICKCORE_WINDOWS_SUPPORT)) && !defined(__MINGW32__)
 #  define MAGICKCORE_LOCALE_SUPPORT
 #endif
 #define LocaleFilename  "locale.xml"
