diff --git a/MagickCore/blob.c b/MagickCore/blob.c
index 1111111..2222222 100644
--- a/MagickCore/blob.c
+++ b/MagickCore/blob.c
@@ -385,7 +385,7 @@
   if (*filename == '\0')
     file=AcquireUniqueFileResource(filename);
   else
-    file=open_utf8(filename,O_RDWR | O_CREAT | O_EXCL | O_BINARY,P_MODE);
+    file=open_utf8(filename,O_RDWR | O_CREAT | O_EXCL | O_BINARY,S_MODE);
   if (file == -1)
     {
       ThrowFileException(exception,BlobError,"UnableToWriteBlob",filename);
@@ -2411,7 +2411,7 @@
     if (LocaleCompare(filename,"-") == 0)
       file=fileno(stdout);
     else
-      file=open_utf8(filename,O_RDWR | O_CREAT | O_EXCL | O_BINARY,P_MODE);
+      file=open_utf8(filename,O_RDWR | O_CREAT | O_EXCL | O_BINARY,S_MODE);
   if (file == -1)
     {
       ThrowFileException(exception,BlobError,"UnableToWriteBlob",filename);
@@ -3685,7 +3685,7 @@
               file;
 
             blob_info->file_info.file=(FILE *) NULL;
-            file=open_utf8(filename,flags,P_MODE);
+            file=open_utf8(filename,flags,S_MODE);
             if (file >= 0)
               blob_info->file_info.file=fdopen(file,type);
             if (blob_info->file_info.file != (FILE *) NULL)
diff --git a/MagickCore/studio.h b/MagickCore/studio.h
index 1111111..2222222 100644
--- a/MagickCore/studio.h
+++ b/MagickCore/studio.h
@@ -243,13 +243,10 @@ extern int vsnprintf(char *,size_t,const char *,va_list);
 #endif
 
 #if defined(S_IRUSR) && defined(S_IWUSR)
-# define P_MODE (S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH)
 # define S_MODE (S_IRUSR | S_IWUSR)
 #elif defined (MAGICKCORE_WINDOWS_SUPPORT)
-# define P_MODE (_S_IREAD | _S_IWRITE)
 # define S_MODE (_S_IREAD | _S_IWRITE)
 #else
-# define P_MODE  0666
 # define S_MODE  0600
 #endif
 
diff --git a/MagickWand/studio.h b/MagickWand/studio.h
index 1111111..2222222 100644
--- a/MagickWand/studio.h
+++ b/MagickWand/studio.h
@@ -224,13 +224,10 @@ extern int vsnprintf(char *,size_t,const char *,va_list);
 #endif
 
 #if defined(S_IRUSR) && defined(S_IWUSR)
-# define P_MODE (S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH)
 # define S_MODE (S_IRUSR | S_IWUSR)
 #elif defined (MAGICKWAND_WINDOWS_SUPPORT)
-# define P_MODE (_S_IREAD | _S_IWRITE)
 # define S_MODE (_S_IREAD | _S_IWRITE)
 #else
-# define P_MODE  0666
 # define S_MODE  0600
 #endif
 
