--- ImageMagick-6.8.8-1/coders/svg.c	2012-12-10 14:09:00.000000000 +0100
+++ ImageMagick-6.8.8-1/coders/svg.c	2013-01-30 09:36:16.075679148 +0100
@@ -76,9 +76,7 @@
 #include "magick/utility.h"
 #if defined(MAGICKCORE_XML_DELEGATE)
 #  if defined(MAGICKCORE_WINDOWS_SUPPORT)
-#    if defined(__MINGW32__) || defined(__MINGW64__)
-#      define _MSC_VER
-#    else
+#    if !defined(__MINGW32__) && !defined(__MINGW64__)
 #      include <win32config.h>
 #    endif
 #  endif
diff '--exclude=configure' '--exclude=*.m4' '--exclude=Makefile.in' -urEbwB ImageMagick-6.8.2-1/configure.ac ImageMagick-6.8.2-1.new/configure.ac
--- ImageMagick-6.8.8-1/magick/Makefile.am	2013-01-23 15:45:15.000000000 +0100
+++ ImageMagick-6.8.8-1.new/magick/Makefile.am	2013-01-30 09:05:59.828303824 +0100
@@ -44,10 +44,10 @@
 
 if WITH_MODULES
 magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_SOURCES = $(MAGICK_BASE_SRCS) $(MAGICK_PLATFORM_SRCS)
-magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS)
+magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS) -lws2_32
 else
 magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_SOURCES = $(MAGICK_BASE_SRCS) $(MAGICK_PLATFORM_SRCS) $(MAGICK_CODER_SRCS) $(MAGICK_FILTER_SRCS)
-magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS)
+magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS) -lws2_32
 endif # WITH_MODULES
 magick_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_CPPFLAGS = -I$(top_builddir)/ltdl -I$(top_srcdir)/ltdl $(LIBRARY_EXTRA_CPPFLAGS)
 
