--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,7 +90,7 @@ project (PUPNP VERSION ${PUPNP_VERSION_STRING} LANGUAGES C)
 
 include (GNUInstallDirs)
 
-if (WIN32)
+if (MSVC)
 	set (CMAKE_DEBUG_POSTFIX d)
 	set (STATIC_POSTFIX s)
 endif()
@@ -344,7 +344,7 @@ endif()
 #
 # Checks for POSIX Threads
 #
-if (NOT WIN32)
+if (NOT MSVC)
 	set (THREADS_PREFER_PTHREAD_FLAG TRUE)
 	include (FindThreads)
 	if (NOT DOWNLOAD_AND_BUILD_DEPS)
--- a/upnp/test/CMakeLists.txt
+++ b/upnp/test/CMakeLists.txt
@@ -1,4 +1,4 @@
-if (WIN32)
+if (MSVC)
 	set (TEST_ENV "PATH=$<TARGET_FILE_DIR:upnp_shared>\;$<TARGET_FILE_DIR:ixml_shared>\;")
 	string (APPEND TEST_ENV "$<TARGET_FILE_DIR:Threads::Threads>\;")
 	string (APPEND TEST_ENV "%PATH%")
