--- libsoundio-2.0.1-7/src/wasapi.c.orig	2025-10-24 06:17:35.860548600 +0200
+++ libsoundio-2.0.1-7/src/wasapi.c	2025-10-24 06:17:46.570132500 +0200
@@ -30,6 +30,7 @@
 #define E_NOTFOUND 0x80070490
 #endif //E_NOTFOUND
 
+#if 0
 // And some GUID are never implemented (Ignoring the INITGUID define)
 static const CLSID CLSID_MMDeviceEnumerator = {
     0xbcde0395, 0xe52f, 0x467c, {0x8e, 0x3d, 0xc4, 0x57, 0x92, 0x91, 0x69, 0x2e}
@@ -74,7 +75,7 @@
     //MIDL_INTERFACE("87ce5498-68d6-44e5-9215-6da47ef883d8")
     0x87ce5498, 0x68d6, 0x44e5,{ 0x92, 0x15, 0x6d, 0xa4, 0x7e, 0xf8, 0x83, 0xd8 }
 };
-
+#endif
 #ifdef __cplusplus
 // In C++ mode, IsEqualGUID() takes its arguments by reference
 #define IS_EQUAL_GUID(a, b) IsEqualGUID(*(a), *(b))
