--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@
     DESCRIPTION "libHaru is a free, cross platform, open source library for generating PDF files."
     LANGUAGES C)
 
-if (UNIX)
+if (UNIX OR MINGW)
     include(GNUInstallDirs)
 endif()
 
@@ -132,8 +132,8 @@
 install(FILES ${haru_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # install various files
-install(FILES README.md CHANGES INSTALL DESTINATION .)
-install(DIRECTORY bindings DESTINATION .)
+install(FILES README.md CHANGES INSTALL DESTINATION ${CMAKE_INSTALL_DOCDIR})
+install(DIRECTORY bindings DESTINATION ${CMAKE_INSTALL_DATADIR}/libharu)
 
 # =======================================================================
 # print out some information
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -88,7 +88,7 @@
 endif()
 
 install(
-   TARGETS
+   TARGETS haru
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
