diff -Nur libsignal-protocol-c-2.3.3.orig/src/CMakeLists.txt libsignal-protocol-c-2.3.3/src/CMakeLists.txt
--- libsignal-protocol-c-2.3.3.orig/src/CMakeLists.txt	2020-03-28 01:15:49.000000000 +0100
+++ libsignal-protocol-c-2.3.3/src/CMakeLists.txt	2021-05-11 09:14:36.351570000 +0200
@@ -72,7 +72,9 @@
 )
 
 if(BUILD_SHARED_LIBS)
-	target_link_libraries(signal-protocol-c ${M_LIB})
+	if(UNIX)
+		target_link_libraries(signal-protocol-c ${M_LIB})
+	endif()
 	set_target_properties(signal-protocol-c PROPERTIES
 		VERSION ${SIGNAL_PROTOCOL_C_VERSION}
 		SOVERSION ${SIGNAL_PROTOCOL_C_VERSION_MAJOR}
diff -Nur libsignal-protocol-c-2.3.3.orig/tests/CMakeLists.txt libsignal-protocol-c-2.3.3/tests/CMakeLists.txt
--- libsignal-protocol-c-2.3.3.orig/tests/CMakeLists.txt	2020-03-28 01:15:49.000000000 +0100
+++ libsignal-protocol-c-2.3.3/tests/CMakeLists.txt	2021-05-11 09:15:37.918894400 +0200
@@ -30,8 +30,11 @@
   set(TEST_PATH ${CMAKE_CURRENT_BINARY_DIR})
 endif()
 
+if(UNIX)
+	set(LIBS ${LIBS} ${M_LIB})
+endif()
+
 set(LIBS ${LIBS}
-	${M_LIB}
 	${CHECK_LDFLAGS}
 	${OPENSSL_LIBRARIES}
 	${CMAKE_THREAD_LIBS_INIT}
