--- zstd-1.5.7/contrib/seekable_format/zstdseek_decompress.c.orig	2025-02-20 07:21:39.707483000 +0100
+++ zstd-1.5.7/contrib/seekable_format/zstdseek_decompress.c	2025-02-20 07:24:35.134062600 +0100
@@ -80,7 +80,7 @@
 #if defined(LIBC_NO_FSEEKO)
 /* Some older libc implementations don't include these functions (e.g. Bionic < 24) */
 #   define LONG_SEEK fseek
-#elif defined(_MSC_VER) && _MSC_VER >= 1400
+#elif defined(_WIN32)
 #   define LONG_SEEK _fseeki64
 #elif !defined(__64BIT__) && (PLATFORM_POSIX_VERSION >= 200112L) /* No point defining Large file for 64 bit */
 #   define LONG_SEEK fseeko
--- zstd-1.5.7/programs/fileio_common.h.orig	2025-02-20 07:21:39.704979800 +0100
+++ zstd-1.5.7/programs/fileio_common.h	2025-02-20 07:24:23.667032100 +0100
@@ -79,7 +79,7 @@
 /* Some older libc implementations don't include these functions (e.g. Bionic < 24) */
 #   define LONG_SEEK fseek
 #   define LONG_TELL ftell
-#elif defined(_MSC_VER) && _MSC_VER >= 1400
+#elif defined(_WIN32)
 #   define LONG_SEEK _fseeki64
 #   define LONG_TELL _ftelli64
 #elif !defined(__64BIT__) && (PLATFORM_POSIX_VERSION >= 200112L) /* No point defining Large file for 64 bit */
