--- a/cmake/install-package-config-files.cmake
+++ b/cmake/install-package-config-files.cmake
@@ -4,7 +4,7 @@
 #
 function(osgearth_package_install_config_files INCLUDE_INSTALL_DIR LIBRARY_INSTALL_DIR)
 
-    set(PACKAGE_INSTALL_DIR share/osgearth)
+    set(PACKAGE_INSTALL_DIR lib/cmake/osgearth)
 
     include(CMakePackageConfigHelpers)
 
@@ -37,7 +37,7 @@
 #
 function(osgearth_package_install_library_target MY_TARGET)
 
-    set(PACKAGE_INSTALL_DIR share/osgearth)
+    set(PACKAGE_INSTALL_DIR lib/cmake/osgearth)
 
     install(
         EXPORT ${MY_TARGET}Targets
