--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@
 SET(OSGEARTH_SOVERSION 135)
 
 # C++11 is required for osgEarth 3.x
-set_property(GLOBAL PROPERTY CXX_STANDARD 11)
+set_property(GLOBAL PROPERTY CXX_STANDARD 17)
 set_property(GLOBAL PROPERTY CXX_STANDARD_REQUIRED ON)
 
 # Assign a custom prefix if necessary.
@@ -79,10 +79,6 @@
 
 # Third-party sources included in the osgEarth repo
 SET(OSGEARTH_EMBEDDED_THIRD_PARTY_DIR ${PROJECT_SOURCE_DIR}/src/third_party)
-
-# Ensure and configure C++11 compiler support.
-INCLUDE(CXX11)
-check_for_cxx11_compiler(CXX11_AVAILABLE)
 
 # Location of GLCORE headers if you are using GLCORE
 set(OSGEARTH_GLCORE_INCLUDE_DIR CACHE PATH "For OpenGL CORE Profile, location of the GLCORE header files")
