--- osgearth-osgearth-2.7/CMakeModules/FindMiniZip.cmake.orig	2016-07-08 08:57:17.739799400 +0300
+++ osgearth-osgearth-2.7/CMakeModules/FindMiniZip.cmake	2016-07-08 09:01:00.792799400 +0300
@@ -6,7 +6,7 @@
 #
 
 # prefer pkg-config
-IF(UNIX)
+IF(UNIX OR MINGW)
 	INCLUDE(FindPkgConfig)
 	IF(PKG_CONFIG_FOUND)
 		pkg_check_modules(MINIZIP QUIET minizip)
@@ -22,8 +22,9 @@
 
 # fallback logic
 IF(NOT MINIZIP_FOUND)
-	FIND_PATH(MINIZIP_INCLUDE_DIR zip.h
+	FIND_PATH(MINIZIP_INCLUDE_DIR unzip.h
 		${CMAKE_SOURCE_DIR}/src/3rdparty/minizip
+		$ENV{MINIZIP_DIR}/include/minizip
 		$ENV{MINIZIP_DIR}/include
 		$ENV{MINIZIP_DIR}
 		$ENV{OSGDIR}/include
