diff -Naur osgearth-osgearth-3.1.a/src/osgEarth/Config.cpp osgearth-osgearth-3.1.b/src/osgEarth/Config.cpp
--- osgearth-osgearth-3.1.a/src/osgEarth/Config.cpp	2020-12-03 14:25:21.000000000 +0100
+++ osgearth-osgearth-3.1.b/src/osgEarth/Config.cpp	2020-12-26 17:17:00.887245900 +0100
@@ -135,6 +135,10 @@
     // add in the new values.
     for( ConfigSet::const_iterator c = rhs._children.begin(); c != rhs._children.end(); ++c )
         add( *c );
+
+    // merge ref map
+    for( RefMap::const_iterator c = rhs._refMap.begin(); c != rhs._refMap.end(); ++c )
+        _refMap[c->first] = c->second;
 }
 
 const Config*
