From f3d3070851549875dfd833b2f0fc1f72793318e4 Mon Sep 17 00:00:00 2001
From: Ralf Habacker <ralf.haacker@freenet.de>
Date: Wed, 15 May 2019 13:29:10 +0200
Subject: [PATCH] Windows compile fix

---
 src/osgEarth/LayerShader | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/osgEarth/LayerShader b/src/osgEarth/LayerShader
index 536fc7070b..96bcaf1fd8 100644
--- a/src/osgEarth/LayerShader
+++ b/src/osgEarth/LayerShader
@@ -36,7 +36,7 @@ namespace osgEarth
 
     //! Serializable shader that supports samplers and uniforms
     //! in an earth file.
-    class ShaderOptions : public ConfigOptions
+    class OSGEARTH_EXPORT ShaderOptions : public ConfigOptions
     {
     public:
         META_ConfigOptions(osgEarth, ShaderOptions, ConfigOptions);
From 0ce9a9553232746f59cf75b53bf05c89a782c2a6 Mon Sep 17 00:00:00 2001
From: ZHANG SHI GUANG <andyzhshg@gmail.com>
Date: Thu, 18 Apr 2019 10:47:27 +0800
Subject: [PATCH] fix config ref map lost when merging

---
 src/osgEarth/Config.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/osgEarth/Config.cpp b/src/osgEarth/Config.cpp
index 5c3e9943ac..5dd3ec078f 100644
--- a/src/osgEarth/Config.cpp
+++ b/src/osgEarth/Config.cpp
@@ -135,6 +135,10 @@ Config::merge( const Config& rhs )
     // add in the new values.
     for( ConfigSet::const_iterator c = rhs._children.begin(); c != rhs._children.end(); ++c )
         add( *c );
+    
+    // merge ref map
+    for( RefMap::const_iterator c = rhs._refMap.begin(); c != rhs._refMap.end(); ++c )
+        _refMap[c->first] = c->second;
 }
 
 const Config*
