--- spyder-6.0.3.orig/setup.py	2024-12-23 21:58:45.651385800 +0300
+++ spyder-6.0.3/setup.py	2024-12-23 22:02:33.922833800 +0300
@@ -97,7 +97,7 @@
                       ('share/metainfo',
                        ['scripts/org.spyder_ide.spyder.appdata.xml'])]
     elif os.name == 'nt':
-        data_files = [('scripts', ['img_src/spyder.ico'])]
+        data_files = [('share/spyder', ['img_src/spyder.ico'])]
     else:
         data_files = []
 
@@ -244,8 +244,8 @@
 
 # Qt bindings requirements
 qt_requirements = {
-    'pyqt5': ['pyqt5>=5.15,<5.16', 'pyqtwebengine>=5.15,<5.16', 'pyqt5-sip<12.16; python_version=="3.8"'],
-    'pyqt6': ['pyqt6>=6.5,<7', 'pyqt6-webengine>=6.5,<7'],
+    'pyqt5': ['pyqt5>=5.15,<5.16', 'pyqt5-sip<12.16; python_version=="3.8"'],
+    'pyqt6': ['pyqt6>=6.5,<7'],
 }
 
 # Get the proper requirements for the selected Qt binding
