--- spyder-6.0.1.orig/setup.py	2024-10-29 22:23:20.239622300 +0300
+++ spyder-6.0.1/setup.py	2024-10-29 22:24:27.721910500 +0300
@@ -97,7 +97,7 @@
                       ('share/metainfo',
                        ['scripts/org.spyder_ide.spyder.appdata.xml'])]
     elif os.name == 'nt':
-        data_files = [('scripts', ['img_src/spyder.ico'])]
+        data_files = [('share/spyder', ['img_src/spyder.ico'])]
     else:
         data_files = []
 
@@ -233,7 +233,6 @@
     'pylint-venv>=3.0.2',
     'pyls-spyder>=0.4.0',
     'pyqt5>=5.15,<5.16',
-    'pyqtwebengine>=5.15,<5.16',
     'python-lsp-black>=2.0.0,<3.0.0',
     'python-lsp-server[all]>=1.12.0,<1.13.0',
     'pyuca>=1.2',
