diff -urN nanodbc-2.13.0-orig/CMakeLists.txt nanodbc-2.13.0/CMakeLists.txt
--- nanodbc-2.13.0-orig/CMakeLists.txt	2020-08-27 23:16:57.000000000 +0800
+++ nanodbc-2.13.0/CMakeLists.txt	2022-01-10 19:59:23.824793800 +0800
@@ -20,7 +20,7 @@
 ########################################
 ## nanodbc version
 ########################################
-file(STRINGS VERSION NANODBC_VERSION REGEX "[0-9]+\\.[0-9]+\\.[0-9]+")
+file(STRINGS VERSION_ NANODBC_VERSION REGEX "[0-9]+\\.[0-9]+\\.[0-9]+")
 string(REGEX REPLACE "^([0-9]+)\\.[0-9]+\\.[0-9]+" "\\1" NANODBC_VERSION_MAJOR "${NANODBC_VERSION}")
 string(REGEX REPLACE "^[0-9]+\\.([0-9])+\\.[0-9]+" "\\1" NANODBC_VERSION_MINOR "${NANODBC_VERSION}")
 string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.([0-9]+)" "\\1" NANODBC_VERSION_PATCH "${NANODBC_VERSION}")
@@ -89,7 +89,7 @@
 
 if(NANODBC_ENABLE_UNICODE)
   add_definitions(-DNANODBC_ENABLE_UNICODE)
-  if(MSVC)
+  if(WIN32)
     # Sets "Use Unicode Character Set" property in Visual Studio projects
     add_definitions(-DUNICODE -D_UNICODE)
   endif()
diff -urN nanodbc-2.13.0-orig/nanodbc/nanodbc.cpp nanodbc-2.13.0/nanodbc/nanodbc.cpp
--- nanodbc-2.13.0-orig/nanodbc/nanodbc.cpp	2020-08-27 23:16:57.000000000 +0800
+++ nanodbc-2.13.0/nanodbc/nanodbc.cpp	2022-01-10 18:45:43.605971700 +0800
@@ -23,6 +23,7 @@
 #include <cstring>
 #include <ctime>
 #include <iomanip>
+#include <limits>
 #include <map>
 #include <type_traits>
 
