From 416e59f75934cd1c42fcf043f118b2d1a4e4e8d4 Mon Sep 17 00:00:00 2001
From: aguswinger <agus.winger@yahoo.co.id>
Date: Wed, 11 Jun 2014 23:42:41 +0700
Subject: [PATCH] making it possible to build polipo using MinGW-w64

---
 polipo.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/polipo.h b/polipo.h
index c2058d7..5e28a56 100644
--- a/polipo.h
+++ b/polipo.h
@@ -60,7 +60,7 @@ THE SOFTWARE.
 #include <signal.h>
 #endif
 
-#ifdef __MINGW32_VERSION
+#ifdef __MINGW32__
 #define MINGW
 #endif
 
