From a532fc1ceee6cb6ecb5dd04cb360a757bb037dfe Mon Sep 17 00:00:00 2001
From: Paul Eipper <lkraider+git@gmail.com>
Date: Sun, 2 Nov 2014 20:12:30 -0200
Subject: [PATCH] Fix socklen_t warning

---
 io.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/io.c b/io.c
index e14d7ea..97c2505 100644
--- a/io.c
+++ b/io.c
@@ -689,7 +689,7 @@ do_scheduled_accept(int status, FdEventHandlerPtr event)
 {
     AcceptRequestPtr request = (AcceptRequestPtr)&event->data;
     int rc, done;
-    unsigned len;
+    socklen_t len;
     struct sockaddr_in addr;
 
     if(status) {
@@ -1128,7 +1128,7 @@ int
 netAddressMatch(int fd, NetAddressPtr list)
 {
     int rc;
-    unsigned int len;
+    socklen_t len;
     struct sockaddr_in sain;
 #ifdef HAVE_IPv6
     struct sockaddr_in6 sain6;
