diff --git a/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp b/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
index 2279e5a..d50c30f 100644
--- a/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
+++ b/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
@@ -370,8 +370,10 @@ lp_build_create_jit_compiler_for_module(LLVMExecutionEngineRef *OutJIT,
      * different strings for MinGW/MSVC, so better play it safe and be
      * explicit.
      */
-#  ifdef _WIN64
+#  if defined(PIPE_ARCH_X86_64)
     LLVMSetTarget(M, "x86_64-pc-win32-elf");
+#  elif defined(PIPE_ARCH_AARCH64)
+    LLVMSetTarget(M, "aarch64-pc-win32-elf");
 #  else
     LLVMSetTarget(M, "i686-pc-win32-elf");
 #  endif
