--- a/src/compiler/clc/meson.build
+++ b/src/compiler/clc/meson.build
@@ -29,7 +29,13 @@
 _libclc_sources = []
 
 if with_microsoft_clc
-  clang_resource_dir = join_paths(llvm_libdir, 'clang', dep_llvm.version(), 'include')
+  if dep_llvm.version().version_compare('< 16')
+    clang_resource_dir = join_paths(llvm_libdir, 'clang', dep_llvm.version(), 'include')
+  else
+    installed_llvm_version_array = dep_llvm.version().split('.')
+    installed_llvm_version_major =  installed_llvm_version_array[0]
+    clang_resource_dir = join_paths(llvm_libdir, 'clang', installed_llvm_version_major, 'include')
+  endif
   if dep_llvm.version().version_compare('< 15.0')
     opencl_c_h = custom_target(
       'opencl-c.h',
