--- mesa-17.3.3/src/gallium/targets/libgl-gdi/SConscript.orig	2018-01-21 17:12:22.323248900 -0500
+++ mesa-17.3.3/src/gallium/targets/libgl-gdi/SConscript	2018-01-21 17:13:45.667612200 -0500
@@ -15,6 +15,7 @@ env.Append(LIBS = [
     'gdi32',
     'user32',
     'kernel32',
+    'ole32',
     'ws2_32',
 ])
 
--- mesa-17.3.3/src/gallium/targets/graw-gdi/SConscript.orig	2018-01-21 17:36:04.571024700 -0500
+++ mesa-17.3.3/src/gallium/targets/graw-gdi/SConscript	2018-01-21 17:41:15.236917600 -0500
@@ -14,6 +14,7 @@ env.Prepend(LIBS = [
     gallium,
     'gdi32',
     'user32',
+    'ole32',
     'ws2_32',
 ])
 
--- mesa-17.3.3/src/glx/Makefile.am.orig	2018-01-21 18:20:29.452763400 -0500
+++ mesa-17.3.3/src/glx/Makefile.am	2018-01-21 18:21:49.227439400 -0500
@@ -196,7 +196,7 @@ lib@GL_LIB@_la_LIBADD = $(GL_LIBS)
 lib@GL_LIB@_la_LDFLAGS = $(GL_LDFLAGS)
 
 if HAVE_WINDOWSDRI
-lib@GL_LIB@_la_LDFLAGS += -lgdi32 -lopengl32 -Wl,--disable-stdcall-fixup
+lib@GL_LIB@_la_LDFLAGS += -lgdi32 -lole32 -lopengl32 -Wl,--disable-stdcall-fixup
 endif
 
 SUBDIRS += . tests
--- mesa-17.3.3/src/glx/Makefile.in.orig	2018-01-21 18:22:19.530664000 -0500
+++ mesa-17.3.3/src/glx/Makefile.in	2018-01-21 18:23:57.362006800 -0500
@@ -167,7 +167,7 @@ target_triplet = @target@
 @USE_LIBGLVND_FALSE@am__append_14 = \
 @USE_LIBGLVND_FALSE@	-DGL_LIB_NAME=\"lib@GL_LIB@.so.1\"
 
-@HAVE_WINDOWSDRI_TRUE@am__append_15 = -lgdi32 -lopengl32 -Wl,--disable-stdcall-fixup
+@HAVE_WINDOWSDRI_TRUE@am__append_15 = -lgdi32 -lole32 -lopengl32 -Wl,--disable-stdcall-fixup
 subdir = src/glx
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/ax_check_compile_flag.m4 \
--- mesa-17.3.3/scons/llvm.py.orig	2018-01-18 16:30:28.000000000 -0500
+++ mesa-17.3.3/scons/llvm.py	2018-01-22 07:16:54.450780000 -0500
@@ -126,7 +126,7 @@
             if env['platform'] == 'windows' and env['crosscompile']:
                 # LLVM 5.0 requires MinGW w/ pthreads due to use of std::thread and friends.
                 assert env['gcc']
-                env['CXX'] = env['CXX'] + '-posix'
+                env['CXX'] = env['CXX'] + ' -posix'
         elif llvm_version >= distutils.version.LooseVersion('4.0'):
             env.Prepend(LIBS = [
                 'LLVMX86Disassembler', 'LLVMX86AsmParser',
@@ -208,6 +208,10 @@ def generate(env):
                 'LLVMSupport', 'LLVMRuntimeDyld', 'LLVMObject'
             ])
         env.Append(LIBS = [
+            'z',
+            'ole32',
+            'version', 
+            'uuid',
             'imagehlp',
             'psapi',
             'shell32',
--- mesa-17.3.3/scons/gallium.py.orig	2018-01-22 01:38:20.096999500 -0500
+++ mesa-17.3.3/scons/gallium.py	2018-01-22 01:42:28.231168700 -0500
@@ -659,6 +659,7 @@ def generate(env):
 
     env.Tool('yacc')
     if host_platform.system() == 'Windows':
+        libs += ['shell32', 'version', 'uuid', 'ole32']
         if check_prog(env, 'win_bison'):
             env["YACC"] = 'win_bison'
 
