diff --git a/meson.build b/meson.build
index 78f2da6f..7a0593de 100644
--- a/meson.build
+++ b/meson.build
@@ -187,7 +187,6 @@ glib_dep = dependency('glib-2.0', version: '>= 2.62')
 gobject_dep = dependency('gobject-2.0', version: '>= 2.62')
 gio_dep = dependency('gio-2.0', version: '>= 2.62')
 gmodule_dep = dependency('gmodule-2.0', version: '>= 2.62')
-gio_unix_dep = dependency('gio-unix-2.0', version: '>= 2.62')
 curl_dep = dependency('libcurl', version: '>= 7.62')
 xml2_dep = dependency('libxml-2.0')
 fyaml_dep = dependency('libfyaml', version: '>= 0.8')
