diff -Naur simage-orig/CMakeLists.txt simage/CMakeLists.txt
--- simage-orig/CMakeLists.txt	2018-07-22 19:00:44.431650800 +0300
+++ simage/CMakeLists.txt	2018-07-22 19:03:18.463521800 +0300
@@ -460,6 +460,11 @@
 	${CMAKE_CURRENT_SOURCE_DIR}/tests/img.tif
 )
 
+if(UNIX OR MINGW)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/simage.pc.in ${CMAKE_BINARY_DIR}/simage.pc @ONLY)
+    install(FILES ${CMAKE_BINARY_DIR}/simage.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+endif()
+
 # ############################################################################
 # New CPACK section, please see the README file inside cpack.d directory.
 add_subdirectory(cpack.d)
diff -Naur simage-orig/simage.pc.in simage/simage.pc.in
--- simage-orig/simage.pc.in	2018-07-22 19:00:44.790451400 +0300
+++ simage/simage.pc.in	2018-07-22 19:01:44.551356900 +0300
@@ -1,20 +1,12 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-datarootdir=@datarootdir@
-datadir=${datarootdir}
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: simage
 Description: a library with image format loaders and front-ends to common import libraries
 Version: @VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} @SIMAGE_EXTRA_LDFLAGS@ @SIMAGE_EXTRA_LIBS@
-Cflags: -I${includedir} @SIMAGE_EXTRA_CFLAGS@ @SIMAGE_EXTRA_CPPFLAGS@
-
-simage_host="@host@"
-compiler=@COIN_COMPILER@
-compiler_is_gcc=@ac_compiler_gnu@
-frameworkdir=@frameworkdir@
-msvcrt=@SIMAGE_MSVC_LIBC@
+Libs: -L${libdir} -lsimage
+Cflags: -I${includedir}
