--- gst-plugins-base-1.2.4/configure.ac.orig	2014-04-30 15:10:52.750643300 +0000
+++ gst-plugins-base-1.2.4/configure.ac	2014-04-30 15:10:56.684142700 +0000
@@ -173,7 +173,7 @@
 
 dnl check for GCC specific SSE headers
 dnl these are used by the speex resampler code
-AC_CHECK_HEADERS([xmmintrin.h emmintrin.h])
+dnl AC_CHECK_HEADERS([xmmintrin.h emmintrin.h])
 
 dnl used in gst/tcp
 AC_CHECK_HEADERS([sys/socket.h],
