--- sip-6.15.1/sipbuild/installable.py.orig	2024-09-10 12:30:54.000000000 +0200
+++ sip-6.15.1/sipbuild/installable.py	2026-01-09 18:40:00.247918600 +0100
@@ -4,6 +4,7 @@
 
 
 import os
+import sys
 from shutil import copy2, copytree
 
 
@@ -27,11 +28,15 @@
         """ Return the full target directory name. """
 
         if self.target_subdir:
-            if os.path.isabs(self.target_subdir):
+            seps = (os.sep, os.altsep) if os.altsep else (os.sep,)
+            if os.path.isabs(self.target_subdir) or self.target_subdir.startswith(seps):
                 target_dir = self.target_subdir
             else:
                 target_dir = os.path.join(target_dir, self.target_subdir)
 
+        if sys.platform == 'win32' and "MSYSTEM" in os.environ:
+            target_dir = os.popen(' '.join(['cygpath', '--unix', target_dir])).readline().strip()
+
         return target_dir
 
     def install(self, target_dir, installed, *, do_install=True):
--- a/sipbuild/distinfo/distinfo.py
+++ b/sipbuild/distinfo/distinfo.py
@@ -79,6 +79,8 @@
         generator = os.path.basename(sys.argv[0])
 
     # The prefix directory corresponds to DESTDIR or INSTALL_ROOT.
+    if sys.platform == 'win32' and "MSYSTEM" in os.environ:
+        distinfo_dir = os.popen(' '.join(['cygpath', '--unix', distinfo_dir])).readline().strip()
     real_distinfo_dir = prefix_dir + distinfo_dir
 
     # Make sure we have an empty dist-info directory.  Handle exceptions as the
