--- webp-pixbuf-loader-0.0.5/meson.build.orig	2022-05-15 22:28:24.000000000 +0200
+++ webp-pixbuf-loader-0.0.5/meson.build	2022-06-28 13:22:27.961681900 +0200
@@ -2,7 +2,7 @@
 gdkpb = dependency('gdk-pixbuf-2.0', version: '>2.22.0', method: 'pkg-config')
 gdk_pb_moddir = get_option('gdk_pixbuf_moduledir')
 if gdk_pb_moddir == ''
-  gdk_pb_moddir = gdkpb.get_pkgconfig_variable('gdk_pixbuf_moduledir')
+  gdk_pb_moddir = gdkpb.get_pkgconfig_variable('gdk_pixbuf_moduledir', define_variable: ['prefix', get_option('prefix')])
 endif
 gdk_pb_query_loaders = gdkpb.get_pkgconfig_variable('gdk_pixbuf_query_loaders')
 webp = dependency('libwebp', version: '>0.4.3')
