--- engine/SCons/Platform/msys.py.orig	2016-06-30 11:36:19.000000000 -0700
+++ engine/SCons/Platform/msys.py	2016-06-30 11:34:12.000000000 -0700
@@ -35,6 +35,14 @@
 import posix, os, subprocess
 from SCons.Platform import TempFileMunge
 
+def exec_subprocess(l, env):
+    proc = subprocess.Popen(l, env = env, close_fds = True)
+    return proc.wait()
+
+def subprocess_spawn(sh, escape, cmd, args, env):
+    args = [x.replace('"', '') for x in args]
+    return exec_subprocess(args, env)
+
 def exec_popen3(l, env, stdout, stderr):
     proc = subprocess.Popen(l, env = env,
                             stdout = stdout,
@@ -73,10 +81,10 @@
     env['SHLIBSUFFIX'] = '.dll'
     env['LIBPREFIXES'] = [ '$LIBPREFIX', '$SHLIBPREFIX', '$IMPLIBPREFIX' ]
     env['LIBSUFFIXES'] = [ '$LIBSUFFIX', '$SHLIBSUFFIX', '$IMPLIBSUFFIX' ]
-    env['TEMPFILE']    = TempFileMunge
-    env['TEMPFILEPREFIX'] = '@'
-    env['MAXLINELENGTH']  = 2048
+    env['MAXLINELENGTH']  = 32300
     env['PSPAWN']         = piped_env_spawn
+    env['SPAWN']          = subprocess_spawn
+
 
 # Local Variables:
 # tab-width:4
