--- engine/SCons/Platform/msys.py-orig	2016-06-30 08:21:47.772625000 -0700
+++ engine/SCons/Platform/msys.py	2016-06-30 08:22:31.350750000 -0700
@@ -35,6 +35,19 @@
 import posix, os, subprocess
 from SCons.Platform import TempFileMunge
 
+def exec_popen3(l, env, stdout, stderr):
+    proc = subprocess.Popen(l, env = env,
+                            stdout = stdout,
+                            stderr = stderr)
+    return proc.wait()
+
+def piped_env_spawn(sh, escape, cmd, args, env, stdout, stderr):
+    # spawn using Popen3 combined with the env command
+    # the command name and the command's stdout is written to stdout
+    # the command's stderr is written to stderr
+    return exec_popen3([sh, '-c', ' '.join(args)],
+                       env, stdout, stderr)
+
 def generate(env):
     posix.generate(env)
 
@@ -54,8 +67,6 @@
             env['ENV'][var] = v
 
     env['ENV']['PATH'] = os.getenv('PATH')
-    env['SPAWN']       = (lambda sh, esc, cmd, args, env_param:
-                          subprocess.call(['sh', '-c', ' '.join(args)]))
     env['PROGPREFIX']  = ''
     env['PROGSUFFIX']  = '.exe'
     env['SHLIBPREFIX'] = ''
@@ -65,6 +76,7 @@
     env['TEMPFILE']    = TempFileMunge
     env['TEMPFILEPREFIX'] = '@'
     env['MAXLINELENGTH']  = 2048
+    env['PSPAWN']         = piped_env_spawn
 
 # Local Variables:
 # tab-width:4
