diff -Naur scons-2.3.4-orig/engine/SCons/Node/FS.py scons-2.3.4/engine/SCons/Node/FS.py
--- scons-2.3.4-orig/engine/SCons/Node/FS.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Node/FS.py	2014-11-05 01:00:41.503600000 +0300
@@ -365,7 +365,7 @@
     return DefaultRCSBuilder
 
 # Cygwin's os.path.normcase pretends it's on a case-sensitive filesystem.
-_is_cygwin = sys.platform == "cygwin"
+_is_cygwin = sys.platform in ["cygwin", "msys"]
 if os.path.normcase("TeSt") == os.path.normpath("TeSt") and not _is_cygwin:
     def _my_normcase(x):
         return x
@@ -1842,7 +1842,7 @@
                 for entry in map(_my_normcase, entries):
                     d[entry] = True
             self.on_disk_entries = d
-        if sys.platform == 'win32' or sys.platform == 'cygwin':
+        if sys.platform == 'win32' or sys.platform == 'cygwin' or sys.platform == 'msys':
             name = _my_normcase(name)
             result = d.get(name)
             if result is None:
diff -Naur scons-2.3.4-orig/engine/SCons/Platform/__init__.py scons-2.3.4/engine/SCons/Platform/__init__.py
--- scons-2.3.4-orig/engine/SCons/Platform/__init__.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Platform/__init__.py	2014-11-05 01:00:42.034000000 +0300
@@ -68,6 +68,8 @@
     if osname == 'posix':
         if sys.platform == 'cygwin':
             return 'cygwin'
+        elif sys.platform == 'msys':
+            return 'msys'
         elif sys.platform.find('irix') != -1:
             return 'irix'
         elif sys.platform.find('sunos') != -1:
diff -Naur scons-2.3.4-orig/engine/SCons/Platform/msys.py scons-2.3.4/engine/SCons/Platform/msys.py
--- scons-2.3.4-orig/engine/SCons/Platform/msys.py	1970-01-01 03:00:00.000000000 +0300
+++ scons-2.3.4/engine/SCons/Platform/msys.py	2014-11-05 01:00:42.034000000 +0300
@@ -0,0 +1,59 @@
+"""SCons.Platform.msys
+
+Platform-specific initialization for MSYS systems.
+
+There normally shouldn't be any need to import this module directly.  It
+will usually be imported through the generic SCons.Platform.Platform()
+selection method.
+"""
+
+#
+# Copyright (c) 2001 - 2014 The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+__revision__ = "src/engine/SCons/Platform/cygwin.py  2014/03/02 14:18:15 garyo"
+
+import posix, os, subprocess
+from SCons.Platform import TempFileMunge
+
+def generate(env):
+    posix.generate(env)
+
+    env['ENV']['PATH'] = os.getenv('PATH')
+    env['SPAWN']       = (lambda sh, esc, cmd, args, env_param:
+                          subprocess.call(['sh', '-c', ' '.join(args)]))
+    env['PROGPREFIX']  = ''
+    env['PROGSUFFIX']  = '.exe'
+    env['SHLIBPREFIX'] = ''
+    env['SHLIBSUFFIX'] = '.dll'
+    env['LIBPREFIXES'] = [ '$LIBPREFIX', '$SHLIBPREFIX', '$IMPLIBPREFIX' ]
+    env['LIBSUFFIXES'] = [ '$LIBSUFFIX', '$SHLIBSUFFIX', '$IMPLIBSUFFIX' ]
+    env['TEMPFILE']    = TempFileMunge
+    env['TEMPFILEPREFIX'] = '@'
+    env['MAXLINELENGTH']  = 2048
+
+# Local Variables:
+# tab-width:4
+# indent-tabs-mode:nil
+# End:
+# vim: set expandtab tabstop=4 shiftwidth=4:
+
diff -Naur scons-2.3.4-orig/engine/SCons/Script/Main.py scons-2.3.4/engine/SCons/Script/Main.py
--- scons-2.3.4-orig/engine/SCons/Script/Main.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Script/Main.py	2014-11-05 01:00:42.034000000 +0300
@@ -770,7 +770,7 @@
     def homedir(d):
         return os.path.expanduser('~/'+d)
 
-    if platform == 'win32' or platform == 'cygwin':
+    if platform == 'win32' or platform == 'cygwin' or platform == 'msys':
         # Note we use $ here instead of %...% because older
         # pythons (prior to 2.6?) didn't expand %...% on Windows.
         # This set of dirs should work on XP, Vista, 7 and later.
diff -Naur scons-2.3.4-orig/engine/SCons/Tool/__init__.py scons-2.3.4/engine/SCons/Tool/__init__.py
--- scons-2.3.4-orig/engine/SCons/Tool/__init__.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Tool/__init__.py	2014-11-05 01:00:42.034000000 +0300
@@ -761,6 +761,14 @@
         c_compilers = ['gcc', 'msvc', 'intelc', 'icc', 'cc']
         cxx_compilers = ['g++', 'msvc', 'intelc', 'icc', 'c++']
         assemblers = ['gas', 'nasm', 'masm']
+        fortran_compilers = ['gfortran', 'g77', 'ifort', 'ifl', 'f95', 'f90', 'f77']
+        ars = ['ar', 'mslib']
+    elif str(platform) == 'msys':
+        "prefer GNU tools on MSYS, except for a platform-specific linker"
+        linkers = ['msyslink', 'mslink', 'ilink']
+        c_compilers = ['gcc', 'msvc', 'intelc', 'icc', 'cc']
+        cxx_compilers = ['g++', 'msvc', 'intelc', 'icc', 'c++']
+        assemblers = ['gas', 'nasm', 'masm']
         fortran_compilers = ['gfortran', 'g77', 'ifort', 'ifl', 'f95', 'f90', 'f77']
         ars = ['ar', 'mslib']
     else:
diff -Naur scons-2.3.4-orig/engine/SCons/Tool/g77.py scons-2.3.4/engine/SCons/Tool/g77.py
--- scons-2.3.4-orig/engine/SCons/Tool/g77.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Tool/g77.py	2014-11-05 01:00:42.049600000 +0300
@@ -44,7 +44,7 @@
     add_f77_to_env(env)
 
     fcomp = env.Detect(compilers) or 'g77'
-    if env['PLATFORM'] in ['cygwin', 'win32']:
+    if env['PLATFORM'] in ['cygwin', 'msys', 'win32']:
         env['SHFORTRANFLAGS'] = SCons.Util.CLVar('$FORTRANFLAGS')
         env['SHF77FLAGS'] = SCons.Util.CLVar('$F77FLAGS')
     else:
diff -Naur scons-2.3.4-orig/engine/SCons/Tool/gcc.py scons-2.3.4/engine/SCons/Tool/gcc.py
--- scons-2.3.4-orig/engine/SCons/Tool/gcc.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Tool/gcc.py	2014-11-05 01:00:42.049600000 +0300
@@ -50,7 +50,7 @@
 
     cc.generate(env)
 
-    if env['PLATFORM'] in ['cygwin', 'win32']:
+    if env['PLATFORM'] in ['cygwin', 'msys', 'win32']:
         env['SHCCFLAGS'] = SCons.Util.CLVar('$CCFLAGS')
     else:
         env['SHCCFLAGS'] = SCons.Util.CLVar('$CCFLAGS -fPIC')
diff -Naur scons-2.3.4-orig/engine/SCons/Tool/gfortran.py scons-2.3.4/engine/SCons/Tool/gfortran.py
--- scons-2.3.4-orig/engine/SCons/Tool/gfortran.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Tool/gfortran.py	2014-11-05 01:00:42.049600000 +0300
@@ -46,7 +46,7 @@
     for dialect in ['F77', 'F90', 'FORTRAN', 'F95', 'F03']:
         env['%s' % dialect] = 'gfortran'
         env['SH%s' % dialect] = '$%s' % dialect
-        if env['PLATFORM'] in ['cygwin', 'win32']:
+        if env['PLATFORM'] in ['cygwin', 'msys', 'win32']:
             env['SH%sFLAGS' % dialect] = SCons.Util.CLVar('$%sFLAGS' % dialect)
         else:
             env['SH%sFLAGS' % dialect] = SCons.Util.CLVar('$%sFLAGS -fPIC' % dialect)
diff -Naur scons-2.3.4-orig/engine/SCons/Tool/hpc++.py scons-2.3.4/engine/SCons/Tool/hpc++.py
--- scons-2.3.4-orig/engine/SCons/Tool/hpc++.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Tool/hpc++.py	2014-11-05 01:00:42.049600000 +0300
@@ -69,7 +69,7 @@
         if line.find('aCC: HP ANSI C++') == 0:
             env['CXXVERSION'] = line.split()[-1]
 
-        if env['PLATFORM'] == 'cygwin':
+        if env['PLATFORM'] == 'cygwin' or env['PLATFORM'] == 'msys':
             env['SHCXXFLAGS'] = SCons.Util.CLVar('$CXXFLAGS')
         else:
             env['SHCXXFLAGS'] = SCons.Util.CLVar('$CXXFLAGS +Z')
diff -Naur scons-2.3.4-orig/engine/SCons/Tool/msyslink.py scons-2.3.4/engine/SCons/Tool/msyslink.py
--- scons-2.3.4-orig/engine/SCons/Tool/msyslink.py	1970-01-01 03:00:00.000000000 +0300
+++ scons-2.3.4/engine/SCons/Tool/msyslink.py	2014-11-05 01:18:06.061600000 +0300
@@ -0,0 +1,95 @@
+"""SCons.Tool.msyslink
+
+Customization of gnulink for MSYS
+
+There normally shouldn't be any need to import this module directly.
+It will usually be imported through the generic SCons.Tool.Tool()
+selection method.
+
+"""
+
+import SCons.Action
+import SCons.Util
+
+import gnulink
+
+def shlib_generator(target, source, env, for_signature):
+    cmd = SCons.Util.CLVar(['$SHLINK'])
+
+    dll = env.FindIxes(target, 'SHLIBPREFIX', 'SHLIBSUFFIX')
+    if dll: cmd.extend(['-o', dll])
+
+    cmd.extend(['$SHLINKFLAGS', '$__RPATH'])
+
+    implib = env.FindIxes(target, 'IMPLIBPREFIX', 'IMPLIBSUFFIX')
+    if implib:
+        cmd.extend([
+            '-Wl,--out-implib='+implib.get_string(for_signature),
+            '-Wl,--export-all-symbols',
+            '-Wl,--enable-auto-import',
+            '-Wl,--whole-archive', '$SOURCES',
+            '-Wl,--no-whole-archive', '$_LIBDIRFLAGS', '$_LIBFLAGS'
+            ])
+    else:
+        cmd.extend(['$SOURCES', '$_LIBDIRFLAGS', '$_LIBFLAGS'])
+    
+    return [cmd]
+
+def shlib_emitter(target, source, env):
+    dll = env.FindIxes(target, 'SHLIBPREFIX', 'SHLIBSUFFIX')
+    no_import_lib = env.get('no_import_lib', 0)
+
+    if not dll or len(target) > 1:
+        raise SCons.Errors.UserError("A shared library should have exactly one target with the suffix: %s" % env.subst("$SHLIBSUFFIX"))
+    
+    # Remove any "lib" after the prefix
+    pre = env.subst('$SHLIBPREFIX')
+    if dll.name[len(pre):len(pre)+3] == 'lib':
+        dll.name = pre + dll.name[len(pre)+3:]
+
+    orig_target = target
+    target = [env.fs.File(dll)]
+    target[0].attributes.shared = 1
+
+    # Append an import lib target
+    if not no_import_lib:
+        # Create list of target libraries as strings
+        target_strings = env.ReplaceIxes(orig_target[0],
+                                         'SHLIBPREFIX', 'SHLIBSUFFIX',
+                                         'IMPLIBPREFIX', 'IMPLIBSUFFIX')
+        
+        implib_target = env.fs.File(target_strings)
+        implib_target.attributes.shared = 1
+        target.append(implib_target)
+
+    return (target, source)
+                         
+
+shlib_action = SCons.Action.Action(shlib_generator, generator=1)
+
+def generate(env):
+    """Add Builders and construction variables for msyslink to an Environment."""
+    gnulink.generate(env)
+
+    env['LINKFLAGS']   = SCons.Util.CLVar('-Wl,-no-undefined')
+
+    env['SHLINKCOM'] = shlib_action
+    env['LDMODULECOM'] = shlib_action
+    env.Append(SHLIBEMITTER = [shlib_emitter])
+
+    env['SHLIBPREFIX']         = 'msys-'
+    env['SHLIBSUFFIX']         = '.dll'
+
+    env['IMPLIBPREFIX']        = 'lib'
+    env['IMPLIBSUFFIX']        = '.dll.a'
+
+def exists(env):
+    return gnulink.exists(env)
+
+
+# Local Variables:
+# tab-width:4
+# indent-tabs-mode:nil
+# End:
+# vim: set expandtab tabstop=4 shiftwidth=4:
+
diff -Naur scons-2.3.4-orig/engine/SCons/Util.py scons-2.3.4/engine/SCons/Util.py
--- scons-2.3.4-orig/engine/SCons/Util.py	2014-09-27 19:51:48.000000000 +0300
+++ scons-2.3.4/engine/SCons/Util.py	2014-11-05 01:00:42.049600000 +0300
@@ -889,7 +889,7 @@
     else:
         return sep.join(paths)
 
-if sys.platform == 'cygwin':
+if sys.platform in ['cygwin', 'msys']:
     def get_native_path(path):
         """Transforms an absolute path into a native path for the system.  In
         Cygwin, this converts from a Cygwin path to a Windows one."""
@@ -1020,7 +1020,7 @@
                     return None
 
 
-if sys.platform == 'cygwin':
+if sys.platform in ['cygwin', 'msys']:
     # On Cygwin, os.path.normcase() lies, so just report back the
     # fact that the underlying Windows OS is case-insensitive.
     def case_sensitive_suffixes(s1, s2):
diff -Naur scons-2.3.4-orig/MANIFEST scons-2.3.4/MANIFEST
--- scons-2.3.4-orig/MANIFEST	2014-09-27 19:52:20.000000000 +0300
+++ scons-2.3.4/MANIFEST	2014-11-05 01:00:42.065200000 +0300
@@ -31,6 +31,7 @@
 engine/SCons/Platform/darwin.py
 engine/SCons/Platform/hpux.py
 engine/SCons/Platform/irix.py
+engine/SCons/Platform/msys.py
 engine/SCons/Platform/os2.py
 engine/SCons/Platform/posix.py
 engine/SCons/Platform/sunos.py
@@ -84,6 +85,7 @@
 engine/SCons/Tool/bcc32.py
 engine/SCons/Tool/c++.py
 engine/SCons/Tool/cc.py
+engine/SCons/Tool/cyglink.py
 engine/SCons/Tool/cvf.py
 engine/SCons/Tool/cyglink.py
 engine/SCons/Tool/default.py
@@ -139,6 +141,7 @@
 engine/SCons/Tool/mssdk.py
 engine/SCons/Tool/msvc.py
 engine/SCons/Tool/msvs.py
+engine/SCons/Tool/msyslink.py
 engine/SCons/Tool/mwcc.py
 engine/SCons/Tool/mwld.py
 engine/SCons/Tool/nasm.py
