--- qemu-3.0.0/include/qemu/osdep.h.orig	2018-11-28 14:06:45.404308700 +0300
+++ qemu-3.0.0/include/qemu/osdep.h	2018-11-28 14:06:57.699130400 +0300
@@ -452,10 +452,13 @@
 int qemu_get_thread_id(void);
 
 #ifndef CONFIG_IOVEC
+#ifndef STRUCT_IOVEC_DEFINED
+#define STRUCT_IOVEC_DEFINED 1
 struct iovec {
     void *iov_base;
     size_t iov_len;
 };
+#endif
 /*
  * Use the same value as Linux for now.
  */
