--- a/Source/Lib/Decoder/pkg-config.pc.in
+++ b/Source/Lib/Decoder/pkg-config.pc.in
@@ -7,3 +7,4 @@
 Version: @DEC_VERSION_MAJOR@.@DEC_VERSION_MINOR@.@DEC_VERSION_PATCH@
 Libs: -L${libdir} -lSvtAv1Dec @LIBS_PRIVATE@
 Cflags: -I${includedir}@DEC_PKG_CONFIG_EXTRA_CFLAGS@
+Cflags.private: -UEB_DLL
--- a/Source/Lib/Encoder/pkg-config.pc.in
+++ b/Source/Lib/Encoder/pkg-config.pc.in
@@ -7,3 +7,4 @@
 Version: @ENC_VERSION_MAJOR@.@ENC_VERSION_MINOR@.@ENC_VERSION_PATCH@
 Libs: -L${libdir} -lSvtAv1Enc @LIBS_PRIVATE@
 Cflags: -I${includedir}@ENC_PKG_CONFIG_EXTRA_CFLAGS@
+Cflags.private: -UEB_DLL
