--- geoalchemy2-0.18.3.orig/geoalchemy2/admin/dialects/sqlite.py	2026-02-24 11:54:01.000000000 +0300
+++ geoalchemy2-0.18.3/geoalchemy2/admin/dialects/sqlite.py	2026-02-24 20:55:45.926879000 +0300
@@ -1,6 +1,8 @@
 """This module defines specific functions for SQLite dialect."""
 
 import os
+import sys
+import pathlib
 
 from sqlalchemy import text
 from sqlalchemy.dialects.sqlite.base import ischema_names as _sqlite_ischema_names
@@ -45,7 +47,13 @@
         *args: Additional arguments (unused).
     """
     if "SPATIALITE_LIBRARY_PATH" not in os.environ:
-        raise RuntimeError("The SPATIALITE_LIBRARY_PATH environment variable is not set.")
+        mod_path = None
+        if os.name == "nt":
+            mod_path = pathlib.Path(sys.executable).parent.joinpath('mod_spatialite.dll')
+        if mod_path and mod_path.exists():
+            os.environ["SPATIALITE_LIBRARY_PATH"] = str(mod_path.absolute())
+        else:
+            raise RuntimeError("The SPATIALITE_LIBRARY_PATH environment variable is not set.")
     dbapi_conn.enable_load_extension(True)
     dbapi_conn.load_extension(os.environ["SPATIALITE_LIBRARY_PATH"])
     dbapi_conn.enable_load_extension(False)
