--- pkg-config-0.29-orig/check/check-sysroot
+++ pkg-config-0.29/check/check-sysroot
@@ -9,7 +9,7 @@
 # MSYS mangles / paths to its own root in windows format. This probably
 # means sysroot doesn't work there, but match what pkg-config passes
 # back anyway.
-[ "$OSTYPE" = msys ] && root=$(cd / && pwd -W) || root=
+[ "$OSTYPE" = msys ] && root=$(cd / && pwd -W) && root=${root%/} || root=
 
 RESULT=""
 run_test --cflags simple
