--- SDL_sound/Makefile.am.orig	2014-06-11 08:38:01.168400000 +0400
+++ SDL_sound/Makefile.am	2014-06-11 08:41:57.330600000 +0400
@@ -1,12 +1,12 @@
-lib_LTLIBRARIES = libSDL_sound.la
+lib_LTLIBRARIES = libSDL2_sound.la
 
 SUBDIRS = decoders . playsound
 
-libSDL_soundincludedir = $(includedir)/SDL2
-libSDL_soundinclude_HEADERS =	\
+libSDL2_soundincludedir = $(includedir)/SDL2
+libSDL2_soundinclude_HEADERS =	\
 	SDL_sound.h
 
-libSDL_sound_la_SOURCES =	\
+libSDL2_sound_la_SOURCES =	\
 	SDL_sound.c	\
 	SDL_sound_internal.h	\
     audio_convert.c   \
@@ -25,10 +25,10 @@
 MPG123_LIB =
 endif
 
-libSDL_sound_la_LDFLAGS = 		\
+libSDL2_sound_la_LDFLAGS = 		\
 	-release $(LT_RELEASE)	\
 	-no-undefined \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
-libSDL_sound_la_LIBADD =	\
+libSDL2_sound_la_LIBADD =	\
 	decoders/libdecoders.la	\
 	$(TIMIDITY_LIB) $(MPG123_LIB)
 
--- SDL_sound/playsound/Makefile.am.orig	2014-06-11 08:45:45.231600000 +0400
+++ SDL_sound/playsound/Makefile.am	2014-06-11 08:45:54.264000000 +0400
@@ -12,9 +12,9 @@
 endif
 
 playsound_CFLAGS = $(PHYSFS_CFLG)
-playsound_LDADD = ../libSDL_sound.la $(PHYSFS_LIBS)
+playsound_LDADD = ../libSDL2_sound.la $(PHYSFS_LIBS)
 playsound_SOURCES = playsound.c physfsrwops.c physfsrwops.h
 
-playsound_simple_LDADD = ../libSDL_sound.la
+playsound_simple_LDADD = ../libSDL2_sound.la
 playsound_simple_SOURCES = playsound_simple.c
 
