--- a/CMakeLists.txt.origin
+++ b/CMakeLists.txt
@@ -135,17 +135,41 @@ if (UNIX OR MINGW)
 	SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.${BUILD_NUMBER}")
 	SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES SOVERSION "${PROJECT_VERSION_MAJOR}")
 endif()
-install(TARGETS ${PROJECT_NAME}
-	ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-	LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-	RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
-install(DIRECTORY Include/Model DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
-install(DIRECTORY Include/Common DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
-install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/Include/ DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+include(GNUInstallDirs)
+install(TARGETS ${PROJECT_NAME} 
+	COMPONENT bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	COMPONENT lib LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	COMPONENT run RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES "Include/Model/COM/NMR_DLLInterfaces.h"
+  COMPONENT inc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(FILES "Include/Model/Classes/NMR_ModelTypes.h"
+  COMPONENT inc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/Model/Classes)
+install(FILES "Include/Model/COM/NMR_COMFactory.h"
+  "Include/Model/COM/NMR_COMInterfaces.h"
+  COMPONENT inc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/Model/COM)
+install(FILES "Include/Common/Platform/NMR_COM_Native.h"
+  "Include/Common/Platform/NMR_SAL.h"
+  "Include/Common/Platform/NMR_WinTypes.h"
+  "Include/Common/Platform/NMR_COM_Emulation.h"
+  COMPONENT inc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/Common/Platform)
+install(FILES "Include/Common/3MF_ProgressTypes.h"
+  COMPONENT inc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/Common)
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/Include/
+  COMPONENT inc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+add_custom_target(install_bin
+  DEPENDS ${PROJECT_NAME}
+  COMMAND "${CMAKE_COMMAND}" -DCMAKE_INSTALL_COMPONENT=bin -P "${CMAKE_BINARY_DIR}/cmake_install.cmake")
+add_custom_target(install_devel
+  DEPENDS ${PROJECT_NAME}
+  COMMAND "${CMAKE_COMMAND}" -DCMAKE_INSTALL_COMPONENT=lib -P "${CMAKE_BINARY_DIR}/cmake_install.cmake"
+  COMMAND "${CMAKE_COMMAND}" -DCMAKE_INSTALL_COMPONENT=inc -P "${CMAKE_BINARY_DIR}/cmake_install.cmake"
+  COMMAND "${CMAKE_COMMAND}" -DCMAKE_INSTALL_COMPONENT=pkgconfig -P "${CMAKE_BINARY_DIR}/cmake_install.cmake")
 
 #########################################################
 configure_file(lib3MF.pc.in lib3MF.pc @ONLY)
-install(FILES ${CMAKE_BINARY_DIR}/lib3MF.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+install(FILES ${CMAKE_BINARY_DIR}/lib3MF.pc
+  COMPONENT pkgconfig DESTINATION ${CMAKE_INSTALL_LIBDIR}/lib)
 
 #########################################################
 option(LIB3MF_TESTS "Switch whether the tests of Lib3MF should be build" ON)
