--- geoipupdate-4.8.0/cmd/geoipupdate/main.go.orig	2022-02-01 08:23:33.340000000 +0100
+++ geoipupdate-4.8.0/cmd/geoipupdate/main.go	2022-02-01 08:08:48.823592300 +0100
@@ -21,11 +21,16 @@
 func main() {
 	log.SetFlags(0)
 
+	ex, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+
 	if defaultConfigFile == "" {
-		defaultConfigFile = geoipupdate.DefaultConfigFile
+		defaultConfigFile = filepath.Join(filepath.Dir(ex), "..", "etc", "GeoIP.conf")
 	}
 	if defaultDatabaseDirectory == "" {
-		defaultDatabaseDirectory = geoipupdate.DefaultDatabaseDirectory
+		defaultDatabaseDirectory = filepath.Join(filepath.Dir(ex), "..", "var", "lib", "GeoIP")
 	}
 
 	args := getArgs()
