--- a/pkg/geoipupdate/vars/defaults_windows.go
+++ b/pkg/geoipupdate/vars/defaults_windows.go
@@ -7,6 +7,6 @@
 var (
 	// I'm not sure these make sense. However they can be overridden at runtime
 	// and in the configuration, so we have some flexibility.
-	DefaultConfigFile        = os.Getenv("SYSTEMDRIVE") + `\ProgramData\MaxMind\GeoIPUpdate\GeoIP.conf`
-	DefaultDatabaseDirectory = os.Getenv("SYSTEMDRIVE") + `\ProgramData\MaxMind\GeoIPUpdate\GeoIP`
+	DefaultConfigFile        = os.Getenv("MINGW_PREFIX") + `/etc/GeoIP.conf`
+	DefaultDatabaseDirectory = os.Getenv("MINGW_PREFIX") + `/var/lib/GeoIP`
 )
