--- DirectX-Headers-1.610.0/CMakeLists.txt.bak	2023-05-18 20:00:56.042623700 -0400
+++ DirectX-Headers-1.610.0/CMakeLists.txt	2023-05-19 04:08:46.329015800 -0400
@@ -45,6 +45,8 @@ if (NOT WIN32)
         "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/wsl/stubs>"
         "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/wsl/stubs>"
     )
+elseif((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__REQUIRED_RPCNDR_H_VERSION__=475")
 endif()
 
 add_library(Microsoft::DirectX-Headers ALIAS DirectX-Headers)
