--- pygtk-2.24.0/examples/pygtk-demo/pygtk-demo.in.orig	2015-01-15 23:10:33.264600000 +0300
+++ pygtk-2.24.0/examples/pygtk-demo/pygtk-demo.in	2015-01-15 23:10:56.383800000 +0300
@@ -2,6 +2,6 @@
 
 import sys
 import os.path
-pygtklibdir = os.path.join("@libdir@", "pygtk", "2.0")
+pygtklibdir = os.path.join(sys.prefix, 'lib', "pygtk", "2.0")
 sys.path.insert(0, pygtklibdir)
 execfile(os.path.join(pygtklibdir, "pygtk-demo.py"))
