--- ghostscript-9.26/base/gp_mswin.c.orig	2018-11-20 23:18:07.972852000 -0500
+++ ghostscript-9.26/base/gp_mswin.c	2018-11-20 23:19:43.176239500 -0500
@@ -735,7 +735,7 @@ gp_fopen(const char *fname, const char *
     return file;
 }
 
-int gp_stat(const char *path, struct _stat64 *buf)
+int gp_stat(const char *path, struct stat64 *buf)
 {
     int len = utf8_to_wchar(NULL, path);
     wchar_t *uni;
