--- gdcm-2.8.7/CMakeLists.txt.orig	2018-08-21 16:51:28.148990200 +0300
+++ gdcm-2.8.7/CMakeLists.txt	2018-08-21 16:51:34.909376900 +0300
@@ -261,7 +261,7 @@
 endif()
 
 if(NOT GDCM_INSTALL_PACKAGE_DIR)
-  set(GDCM_INSTALL_PACKAGE_DIR ${GDCM_INSTALL_LIB_DIR}/${subdir}
+  set(GDCM_INSTALL_PACKAGE_DIR ${GDCM_INSTALL_LIB_DIR}/cmake/${subdir}
     CACHE INTERNAL "")
 endif()
 
--- gdcm-2.8.7/CMake/ExportConfiguration/GDCMConfig.cmake.in.orig	2018-08-21 16:52:05.911150100 +0300
+++ gdcm-2.8.7/CMake/ExportConfiguration/GDCMConfig.cmake.in	2018-08-21 16:52:33.914751800 +0300
@@ -37,9 +37,9 @@
 if(EXISTS ${SELF_DIR}/GDCMTargets.cmake)
   # This is an install tree
   include(${SELF_DIR}/GDCMTargets.cmake)
-  get_filename_component(GDCM_INCLUDE_ROOT "${SELF_DIR}/../../@GDCM_INSTALL_INCLUDE_DIR@" ABSOLUTE)
+  get_filename_component(GDCM_INCLUDE_ROOT "${SELF_DIR}/../../../@GDCM_INSTALL_INCLUDE_DIR@" ABSOLUTE)
   set(GDCM_INCLUDE_DIRS ${GDCM_INCLUDE_ROOT})
-  get_filename_component(GDCM_LIB_ROOT "${SELF_DIR}/../../@GDCM_INSTALL_LIB_DIR@" ABSOLUTE)
+  get_filename_component(GDCM_LIB_ROOT "${SELF_DIR}/../../../@GDCM_INSTALL_LIB_DIR@" ABSOLUTE)
   set(GDCM_LIBRARY_DIRS ${GDCM_LIB_ROOT})
 else()
   if(EXISTS ${SELF_DIR}/GDCMExports.cmake)
--- gdcm-2.8.7/Utilities/gdcmopenjpeg/CMakeLists.txt.orig	2018-08-21 16:55:15.706005700 +0300
+++ gdcm-2.8.7/Utilities/gdcmopenjpeg/CMakeLists.txt	2018-08-21 16:55:27.280667800 +0300
@@ -150,7 +150,7 @@
   # We could install *.cmake files in share/ however those files contains
   # hardcoded path to libraries on a multi-arch system (fedora/debian) those
   # path will be different (lib/i386-linux-gnu vs lib/x86_64-linux-gnu)
-  set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/${OPENJPEG_INSTALL_SUBDIR}")
+  set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/cmake/${OPENJPEG_INSTALL_SUBDIR}")
 endif()
 
 if (APPLE)
--- gdcm-2.8.7/Utilities/gdcmopenjpeg/cmake/OpenJPEGConfig.cmake.in.orig	2018-08-21 16:55:35.856158300 +0300
+++ gdcm-2.8.7/Utilities/gdcmopenjpeg/cmake/OpenJPEGConfig.cmake.in	2018-08-21 16:55:47.229808800 +0300
@@ -26,7 +26,7 @@
 if(EXISTS ${SELF_DIR}/OpenJPEGTargets.cmake)
   # This is an install tree
   include(${SELF_DIR}/OpenJPEGTargets.cmake)
-  get_filename_component(OPENJPEG_INCLUDE_ROOT "${SELF_DIR}/../../../@OPENJPEG_INSTALL_INCLUDE_DIR@" ABSOLUTE)
+  get_filename_component(OPENJPEG_INCLUDE_ROOT "${SELF_DIR}/../../../../@OPENJPEG_INSTALL_INCLUDE_DIR@" ABSOLUTE)
   set(OPENJPEG_INCLUDE_DIRS ${OPENJPEG_INCLUDE_ROOT})
 
 else()
