diff -Naur gdcm-2.8.4-orig/Source/Common/gdcmCryptoFactory.cxx gdcm-2.8.4/Source/Common/gdcmCryptoFactory.cxx
--- gdcm-2.8.4-orig/Source/Common/gdcmCryptoFactory.cxx	2017-11-09 11:09:58.000000000 +0300
+++ gdcm-2.8.4/Source/Common/gdcmCryptoFactory.cxx	2018-01-12 08:57:42.979622000 +0300
@@ -13,7 +13,7 @@
 =========================================================================*/
 #include "gdcmCryptoFactory.h"
 
-#ifdef WIN32
+#ifdef _WIN32
 #include "gdcmCAPICryptoFactory.h"
 #endif
 
@@ -29,7 +29,7 @@
 
 CryptoFactory* CryptoFactory::GetFactoryInstance(CryptoLib id)
 {
-#ifdef WIN32
+#ifdef _WIN32
   static CAPICryptoFactory capi(CryptoFactory::CAPI);
 #endif
 #ifdef GDCM_USE_SYSTEM_OPENSSL
@@ -50,9 +50,9 @@
 #endif // GDCM_HAVE_CMS_RECIPIENT_PASSWORD
 #endif // GDCM_USE_SYSTEM_OPENSSL
 // We always prefer native API (by default):
-#ifdef WIN32
+#ifdef _WIN32
     id = CryptoFactory::CAPI;
-#endif // WIN32
+#endif // _WIN32
     }
 
   std::map<CryptoLib, CryptoFactory*>::iterator it = getInstanceMap().find(id);
diff -Naur gdcm-2.8.4-orig/Source/Common/gdcmTerminal.cxx gdcm-2.8.4/Source/Common/gdcmTerminal.cxx
--- gdcm-2.8.4-orig/Source/Common/gdcmTerminal.cxx	2017-11-09 11:09:58.000000000 +0300
+++ gdcm-2.8.4/Source/Common/gdcmTerminal.cxx	2018-01-12 08:58:35.357304300 +0300
@@ -17,7 +17,7 @@
 #include <iostream>
 #include <fstream>
 
-#ifdef WIN32
+#ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h> /* SetConsoleTextAttribute */
 #endif
@@ -40,7 +40,7 @@
 {
 private:
 // console implementation details:
-#ifdef WIN32
+#ifdef _WIN32
   HANDLE hConsoleHandle;
   CONSOLE_SCREEN_BUFFER_INFO ConsoleInfo;
   WORD wNormalAttributes;
@@ -52,7 +52,7 @@
 public:
   ConsoleImp()
   {
-#ifdef WIN32
+#ifdef _WIN32
     hConsoleHandle = GetStdHandle(STD_OUTPUT_HANDLE);
     GetConsoleScreenBufferInfo(hConsoleHandle, &ConsoleInfo);
     wNormalAttributes = ConsoleInfo.wAttributes;
@@ -61,7 +61,7 @@
   }
   ~ConsoleImp()
   {
-#ifdef WIN32
+#ifdef _WIN32
     SetConsoleTextAttribute(hConsoleHandle, wNormalAttributes);
   }
   WORD get_attributes() {
@@ -85,7 +85,7 @@
     return command;
   }
   void set_attributes(int color) {
-#ifdef WIN32
+#ifdef _WIN32
   static const int colors[8] = { 0, 4, 2, 6, 1, 5, 3, 7 };
   WORD wAttributes;
 
@@ -154,7 +154,7 @@
   SetConsoleTextAttribute(hConsoleHandle, wAttributes);
 #else
   (void)color;
-#endif //WIN32
+#endif //_WIN32
 }
 
 };
diff -Naur gdcm-2.8.4-orig/Source/Common/gdcmWin32.h gdcm-2.8.4/Source/Common/gdcmWin32.h
--- gdcm-2.8.4-orig/Source/Common/gdcmWin32.h	2017-11-09 11:09:58.000000000 +0300
+++ gdcm-2.8.4/Source/Common/gdcmWin32.h	2018-01-12 08:56:34.851867800 +0300
@@ -20,7 +20,7 @@
 #endif
 //-----------------------------------------------------------------------------
 // http://gcc.gnu.org/wiki/Visibility
-#if defined(WIN32) && defined(GDCM_BUILD_SHARED_LIBS)
+#if defined(_WIN32) && defined(GDCM_BUILD_SHARED_LIBS)
   #if (defined(gdcmCommon_EXPORTS) || defined(gdcmDICT_EXPORTS) || defined(gdcmDSED_EXPORTS) || defined(gdcmIOD_EXPORTS) || defined(gdcmMSFF_EXPORTS) || defined(gdcmMEXD_EXPORTS)|| defined(_gdcmswig_EXPORTS)) || defined(vtkgdcm_EXPORTS)
     #define GDCM_EXPORT __declspec( dllexport )
   #else
diff -Naur gdcm-2.8.4-orig/Source/Common/zipstreamimpl.h gdcm-2.8.4/Source/Common/zipstreamimpl.h
--- gdcm-2.8.4-orig/Source/Common/zipstreamimpl.h	2017-11-09 11:09:58.000000000 +0300
+++ gdcm-2.8.4/Source/Common/zipstreamimpl.h	2018-01-12 08:59:04.739661700 +0300
@@ -64,7 +64,7 @@
 
 #include <gdcm_zlib.h>
 
-#ifdef WIN32 /* Window 95 & Windows NT */
+#ifdef _WIN32 /* Window 95 & Windows NT */
 #  define OS_CODE  0x0b
 #endif
 #if defined(MACOS) || defined(TARGET_OS_MAC)
diff -Naur gdcm-2.8.4-orig/Utilities/gdcmcharls/CMakeLists.txt gdcm-2.8.4/Utilities/gdcmcharls/CMakeLists.txt
--- gdcm-2.8.4-orig/Utilities/gdcmcharls/CMakeLists.txt	2017-11-09 11:09:58.000000000 +0300
+++ gdcm-2.8.4/Utilities/gdcmcharls/CMakeLists.txt	2018-01-12 10:11:32.600168700 +0300
@@ -52,7 +52,7 @@
 
 if(WIN32)
   if(BUILD_SHARED_LIBS)
-    add_definitions(-DCHARLS_DLL)
+    add_definitions(-DCHARLS_DLL_BUILD)
   else()
     add_definitions(-DCHARLS_STATIC)
   endif()
diff -Naur gdcm-2.8.4-orig/Utilities/gdcmcharls/interface.h gdcm-2.8.4/Utilities/gdcmcharls/interface.h
--- gdcm-2.8.4-orig/Utilities/gdcmcharls/interface.h	2017-11-09 11:09:58.000000000 +0300
+++ gdcm-2.8.4/Utilities/gdcmcharls/interface.h	2018-01-12 10:11:06.776376800 +0300
@@ -8,6 +8,11 @@
 
 #include "publictypes.h"
 
+// Windows and building CharLS DLL itself.
+#if defined(_WIN32) && defined(CHARLS_DLL_BUILD)
+#define CHARLS_IMEXPORT(returntype) __declspec(dllexport) returntype __stdcall
+#endif
+
 /* non-windows (static linking) */
 #if !defined(CHARLS_IMEXPORT) && !defined(_WIN32)
 #  define CHARLS_IMEXPORT(returntype) returntype
