--- gdcm-2.8.8/Source/Common/gdcmCAPICryptographicMessageSyntax.cxx.orig	2018-10-22 17:55:36.529896700 +0300
+++ gdcm-2.8.8/Source/Common/gdcmCAPICryptographicMessageSyntax.cxx	2018-10-22 18:09:01.999966900 +0300
@@ -218,7 +218,8 @@
     goto err;
     }
 
-  DWORD dwMessageType, cbMessageTypeLen = sizeof(DWORD);
+  DWORD dwMessageType, cbMessageTypeLen;
+  cbMessageTypeLen = sizeof(DWORD);
   if(! CryptMsgGetParam(hMsg, CMSG_TYPE_PARAM, 0, &dwMessageType, &cbMessageTypeLen)) 
     {
     gdcmErrorMacro( "CryptMsgGetParam CMSG_TYPE_PARAM failed with error 0x" << std::hex << GetLastError() );
@@ -232,7 +233,8 @@
     }
 
   ALG_ID kekAlg;
-  DWORD kekAlgLen = sizeof(ALG_ID);
+  DWORD kekAlgLen;
+  kekAlgLen = sizeof(ALG_ID);
   if(! CryptGetKeyParam(hRsaPrivK, KP_ALGID, (BYTE*)&kekAlg, &kekAlgLen, 0)) 
     {
     gdcmErrorMacro( "MsgGetParam KP_ALGID failed with error 0x" << std::hex << GetLastError() );
@@ -244,7 +246,8 @@
     goto err;
     }
 
-  DWORD nrOfRecipeints, nrOfRecipientsLen = sizeof(DWORD);
+  DWORD nrOfRecipeints, nrOfRecipientsLen;
+  nrOfRecipientsLen = sizeof(DWORD);
   if(! CryptMsgGetParam(hMsg, CMSG_RECIPIENT_COUNT_PARAM, 0, &nrOfRecipeints, &nrOfRecipientsLen))
     {
     gdcmErrorMacro( "Decode CMSG_RECIPIENT_COUNT_PARAM failed with error 0x" << std::hex << GetLastError() );
