--- a/Source/MediaStorageAndFileFormat/CMakeLists.txt
+++ b/Source/MediaStorageAndFileFormat/CMakeLists.txt
@@ -207,6 +207,11 @@
 if(GDCM_USE_JPEGLS)
   target_link_libraries(gdcmMSFF LINK_PRIVATE ${GDCM_CHARLS_LIBRARIES})
 endif()
+
+if(NOT MSVC)
+  find_package(Iconv REQUIRED)
+  target_link_libraries(gdcmMSFF PUBLIC Iconv::Iconv)
+endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX AND MINGW)
   # I am getting:
