--- GDCM-3.2.5/Applications/Cxx/gdcminfo.cxx.orig	2026-03-27 10:31:27.062505800 +0100
+++ GDCM-3.2.5/Applications/Cxx/gdcminfo.cxx	2026-03-27 10:36:40.672296300 +0100
@@ -514,7 +514,11 @@
 #endif
 
     PDFDoc *doc;
+#ifdef LIBPOPPLER_PDFDOC_HAS_OPTIONAL
+    doc = new PDFDoc(std::unique_ptr<BaseStream>(appearStream), ownerPW, userPW);
+#else
     doc = new PDFDoc(appearStream, ownerPW, userPW);
+#endif
 
     std::string title;
     std::string subject;
--- GDCM-3.2.5/Applications/Cxx/CMakeLists.txt.orig	2026-03-10 15:47:01.000000000 +0100
+++ GDCM-3.2.5/Applications/Cxx/CMakeLists.txt	2026-03-27 10:39:01.862932500 +0100
@@ -117,7 +117,7 @@
     list(APPEND libpoppler_flags -DLIBPOPPLER_UNICODEMAP_HAS_CONSTMAPUNICODE)
   endif()
   CHECK_CXX_SOURCE_COMPILES(
-    "\#include <poppler/PDFDoc.h>\nint main() { std::optional<GooString> ownerPW, userPW; PDFDoc d((BaseStream*)nullptr,ownerPW,userPW); return 0;}"
+    "\#include <poppler/PDFDoc.h>\nint main() { std::optional<GooString> ownerPW, userPW; PDFDoc d(std::unique_ptr<BaseStream>{},ownerPW,userPW); return 0;}"
     LIBPOPPLER_PDFDOC_HAS_OPTIONAL)
   if(LIBPOPPLER_PDFDOC_HAS_OPTIONAL)
     list(APPEND libpoppler_flags -DLIBPOPPLER_PDFDOC_HAS_OPTIONAL)
