--- libgusb-0.3.6/gusb/meson.build.orig	2021-07-11 15:05:09.764745600 -0700
+++ libgusb-0.3.6/gusb/meson.build	2021-07-11 15:07:34.171435400 -0700
@@ -44,7 +44,7 @@
 
 mapfile = 'libgusb.ver'
 vflag = []
-if host_machine.system() not in ['darwin']
+if host_machine.system() not in ['darwin'] and (host_machine.system() != 'windows' or meson.get_compiler('c').get_linker_id() != 'ld.lld')
   vflag += '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
 endif
 gusb = library(
