diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c3d0d4..e0537a9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,6 +123,7 @@ add_definitions(-DLIBRESSL_INTERNAL)
 add_definitions(-DOPENSSL_NO_HW_PADLOCK)
 add_definitions(-D__BEGIN_HIDDEN_DECLS=)
 add_definitions(-D__END_HIDDEN_DECLS=)
+add_definitions(-DOPENSSLBIN=\"${CMAKE_INSTALL_FULL_BINDIR}\")
 
 set(CMAKE_POSITION_INDEPENDENT_CODE true)
 
diff --git a/Makefile.am.common b/Makefile.am.common
index 5405704..a166c15 100644
--- a/Makefile.am.common
+++ b/Makefile.am.common
@@ -3,3 +3,4 @@ AM_CPPFLAGS = -I$(top_srcdir)/include
 AM_CPPFLAGS += -I$(abs_top_builddir)/include
 AM_CPPFLAGS += -I$(top_srcdir)/include/compat -DLIBRESSL_INTERNAL
 AM_CPPFLAGS += -D__BEGIN_HIDDEN_DECLS= -D__END_HIDDEN_DECLS=
+AM_CPPFLAGS += -DOPENSSLBIN=\"$(bindir)\"
diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
index 5140a7a..d5c216d 100644
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -204,6 +204,7 @@ set(
 	mem_clr.c
 	mem_dbg.c
 	o_fips.c
+	pathtools.c
 	aes/aes.c
 	aes/aes_core.c
 	aes/aes_ige.c
diff --git a/crypto/Makefile.am b/crypto/Makefile.am
index cd6fe29..8e20b62 100644
--- a/crypto/Makefile.am
+++ b/crypto/Makefile.am
@@ -223,6 +223,10 @@ noinst_HEADERS += crypto_internal.h
 noinst_HEADERS += crypto_local.h
 noinst_HEADERS += x86_arch.h
 
+# pathtools
+libcrypto_la_SOURCES += pathtools.c
+noinst_HEADERS += pathtools.h
+
 # aes
 libcrypto_la_SOURCES += aes/aes.c
 libcrypto_la_SOURCES += aes/aes_core.c
diff --git a/crypto/ct/ct_log.c b/crypto/ct/ct_log.c
index 514246f..63e0b0a 100644
--- a/crypto/ct/ct_log.c
+++ b/crypto/ct/ct_log.c
@@ -70,6 +70,7 @@
 #include "conf_local.h"
 
 #include "cryptlib.h"
+#include "pathtools.h"
 
 
 /*
@@ -200,7 +201,11 @@ ctlog_new_from_conf(CTLOG **ct_log, const CONF *conf, const char *section)
 int
 CTLOG_STORE_load_default_file(CTLOG_STORE *store)
 {
-	return CTLOG_STORE_load_file(store, CTLOG_FILE);
+	static char *reloc = NULL;
+	if (reloc == NULL) {
+		reloc = single_path_relocation_lib(OPENSSLBIN, CTLOG_FILE);
+	}
+	return CTLOG_STORE_load_file(store, reloc);
 }
 LCRYPTO_ALIAS(CTLOG_STORE_load_default_file);
 
diff --git a/crypto/x509/x509_def.c b/crypto/x509/x509_def.c
index f85781a..1f708a3 100644
--- a/crypto/x509/x509_def.c
+++ b/crypto/x509/x509_def.c
@@ -60,32 +60,49 @@
 #include "cryptlib.h"
 #include <openssl/crypto.h>
 #include <openssl/x509.h>
+#include "pathtools.h"
 
 const char *
 X509_get_default_private_dir(void)
 {
-	return (X509_PRIVATE_DIR);
+	static char *reloc = NULL;
+	if (reloc == NULL) {
+		reloc = single_path_relocation_lib(OPENSSLBIN, X509_PRIVATE_DIR);
+	}
+	return reloc;
 }
 LCRYPTO_ALIAS(X509_get_default_private_dir);
 
 const char *
 X509_get_default_cert_area(void)
 {
-	return (X509_CERT_AREA);
+	static char *reloc = NULL;
+	if (reloc == NULL) {
+		reloc = single_path_relocation_lib(OPENSSLBIN, X509_CERT_AREA);
+	}
+	return reloc;
 }
 LCRYPTO_ALIAS(X509_get_default_cert_area);
 
 const char *
 X509_get_default_cert_dir(void)
 {
-	return (X509_CERT_DIR);
+	static char *reloc = NULL;
+	if (reloc == NULL) {
+		reloc = single_path_relocation_lib(OPENSSLBIN, X509_CERT_DIR);
+	}
+	return reloc;
 }
 LCRYPTO_ALIAS(X509_get_default_cert_dir);
 
 const char *
 X509_get_default_cert_file(void)
 {
-	return (X509_CERT_FILE);
+	static char *reloc = NULL;
+	if (reloc == NULL) {
+		reloc = single_path_relocation_lib(OPENSSLBIN, X509_CERT_FILE);
+	}
+	return reloc;
 }
 LCRYPTO_ALIAS(X509_get_default_cert_file);
 
diff --git a/tls/CMakeLists.txt b/tls/CMakeLists.txt
index c81bbb0..36218bc 100644
--- a/tls/CMakeLists.txt
+++ b/tls/CMakeLists.txt
@@ -1,5 +1,6 @@
 set(
 	TLS_SRC
+	pathtools.c
 	tls.c
 	tls_bio_cb.c
 	tls_client.c
diff --git a/tls/Makefile.am b/tls/Makefile.am
index 22f3222..a00b1fe 100644
--- a/tls/Makefile.am
+++ b/tls/Makefile.am
@@ -59,3 +59,7 @@ libtls_la_SOURCES += compat/ftruncate.c
 libtls_la_SOURCES += compat/pread.c
 libtls_la_SOURCES += compat/pwrite.c
 endif
+
+# pathtools
+libtls_la_SOURCES += pathtools.c
+noinst_HEADERS += pathtools.h
diff --git a/tls/tls_config.c b/tls/tls_config.c
index 22fa845..90697c0 100644
--- a/tls/tls_config.c
+++ b/tls/tls_config.c
@@ -28,13 +28,18 @@
 #include <tls.h>
 
 #include "tls_internal.h"
+#include "pathtools.h"
 
 static const char default_ca_file[] = TLS_DEFAULT_CA_FILE;
 
 const char *
 tls_default_ca_cert_file(void)
 {
-	return default_ca_file;
+	static char *reloc = NULL;
+	if (reloc == NULL) {
+		reloc = single_path_relocation_lib(OPENSSLBIN, default_ca_file);
+	}
+	return reloc;
 }
 
 int
