diff --git a/src/kernel/kernel.cmake b/src/kernel/kernel.cmake
index 1234567..abcdefg 100644
--- a/src/kernel/kernel.cmake
+++ b/src/kernel/kernel.cmake
@@ -12,7 +12,8 @@ if(LINUX)
   set(_PREF_CONFIG_JSON /usr/share/dsg/configs/org.deepin.dtk.preference.json)
 else()
-  set(_PREF_CONFIG_JSON ${CMAKE_INSTALL_PREFIX}/share/dsg/configs/org.deepin.dtk.preference.json)
+  execute_process(COMMAND cygpath -w "${CMAKE_INSTALL_PREFIX}/share/dsg/configs/org.deepin.dtk.preference.json"
+    OUTPUT_VARIABLE _PREF_CONFIG_JSON OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 
 dtk_add_config_to_cpp(kernel_SRC ${_PREF_CONFIG_JSON}
