diff --git a/dtkgui.cmake b/dtkgui.cmake
index 50d374b2..073b167b 100644
--- a/dtkgui.cmake
+++ b/dtkgui.cmake
@@ -141,7 +141,7 @@ if(FreeImage_FOUND AND libraw_FOUND)
 endif()
 
 # Export all symbols on Windows to avoid dllimport issues
-if(WIN32 AND NOT MSVC)
+if(WIN32)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--export-all-symbols")
   add_definitions(-DLIBDTKGUI_LIBRARY)
 endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 82142aee..1c7b637f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -104,7 +104,11 @@ target_link_directories(${LIB_NAME} INTERFACE
     $<INSTALL_INTERFACE:${LIBRARY_INSTALL_DIR}>
 )
 
-install(TARGETS ${LIB_NAME} EXPORT Dtk${DTK_NAME_SUFFIX}GuiTargets DESTINATION "${LIBRARY_INSTALL_DIR}")
+install(TARGETS ${LIB_NAME} EXPORT Dtk${DTK_NAME_SUFFIX}GuiTargets
+    LIBRARY DESTINATION "${LIBRARY_INSTALL_DIR}"
+    ARCHIVE DESTINATION "${LIBRARY_INSTALL_DIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+)
 install(EXPORT Dtk${DTK_NAME_SUFFIX}GuiTargets NAMESPACE Dtk${DTK_NAME_SUFFIX}:: FILE Dtk${DTK_NAME_SUFFIX}GuiTargets.cmake DESTINATION "${CONFIG_INSTALL_DIR}")
 install(DIRECTORY ../include/util/ DESTINATION "${INCLUDE_INSTALL_DIR}" FILES_MATCHING PATTERN "*")
 install(DIRECTORY ../include/DtkGui/ DESTINATION "${INCLUDE_INSTALL_DIR}" FILES_MATCHING PATTERN "*")
