From e9dc59d772161149de30f79fce55d14a901055e9 Mon Sep 17 00:00:00 2001
From: LRN <lrn1986@gmail.com>
Date: Wed, 5 Aug 2015 19:03:42 +0100
Subject: [PATCH 16/20] MinGW-w64: Enable libsanitizer

---
 libsanitizer/configure.tgt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libsanitizer/configure.tgt b/libsanitizer/configure.tgt
index ac0eb7a..a7f7356 100644
--- a/libsanitizer/configure.tgt
+++ b/libsanitizer/configure.tgt
@@ -36,6 +36,8 @@ case "${target}" in
 	;;
   aarch64*-*-linux*)
 	;;
+  *-*-mingw*)
+	;;
   x86_64-*-darwin[1]* | i?86-*-darwin[1]*)
 	TSAN_SUPPORTED=no
 	;;
-- 
2.5.0

