--- expat-2.1.0/Makefile.in.orig	2015-04-11 23:40:32.731000000 +0300
+++ expat-2.1.0/Makefile.in	2015-04-11 23:40:38.456200000 +0300
@@ -79,7 +79,7 @@
 
 install: xmlwf/xmlwf@EXEEXT@ installlib
 	$(mkinstalldirs) $(DESTDIR)$(bindir) $(DESTDIR)$(man1dir)
-	$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) xmlwf/xmlwf@EXEEXT@ $(DESTDIR)$(bindir)/xmlwf
+	$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) xmlwf/xmlwf@EXEEXT@ $(DESTDIR)$(bindir)/xmlwf@EXEEXT@
 	$(INSTALL_DATA) $(MANFILE) $(DESTDIR)$(man1dir)
 
 installlib: $(LIBRARY) $(APIHEADER) expat.pc
