--- python3/python_support.c	2023-05-09 12:41:22.488385621 +1000
+++ python3/python_support.c-new	2023-05-09 12:41:59.191517548 +1000
@@ -814,13 +814,13 @@
 PyCodeObject*
 ada_pyframe_get_code (PyFrameObject* obj)
 {
-   return obj->f_code;
+   return PyFrame_GetCode (obj);
 }
 
 PyFrameObject*
 ada_pyframe_get_back (PyFrameObject* obj)
 {
-   return obj->f_back;
+   return PyFrame_GetBack (obj);
 }
 
 PyObject*
