diff -bur openssl-3.6.0-orig/Configurations/10-main.conf openssl-3.6.0/Configurations/10-main.conf
--- openssl-3.6.0-orig/Configurations/10-main.conf	2026-01-27 06:06:25.778001900 -0700
+++ openssl-3.6.0/Configurations/10-main.conf	2026-01-27 06:06:37.119230300 -0700
@@ -1744,7 +1744,7 @@
         uplink_arch      => undef,
         perlasm_scheme   => "mingw64",
         shared_rcflag    => "--target=pe-x86-64",
-        multilib         => "64",
+        multilib         => "-x64",
     },
 
     "mingwarm64" => {
diff -bur openssl-3.6.0-orig/Configurations/platform/mingw.pm openssl-3.6.0/Configurations/platform/mingw.pm
--- openssl-3.6.0-orig/Configurations/platform/mingw.pm	2026-01-27 06:06:25.780510300 -0700
+++ openssl-3.6.0/Configurations/platform/mingw.pm	2026-01-27 06:06:40.823174800 -0700
@@ -33,8 +33,7 @@
     return platform::BASE::__concat(platform::BASE->sharedname($_[1]),
                                     "-",
                                     $_[0]->shlib_version_as_filename(),
-                                    ($config{target} eq "mingw64"
-                                         ? "-x64" : ""));
+                                    ($target{multilib} // '' ));
 }
 
 # With Mingw and other DLL producers, there isn't any "simpler" shared
