--- giflib-5.1.6/Makefile.orig	2019-02-18 09:51:14.067307400 +0300
+++ giflib-5.1.6/Makefile	2019-02-18 09:55:49.687992300 +0300
@@ -60,21 +60,19 @@
 
 LDLIBS=libgif.a -lm
 
-all: libgif.so libgif.a $(UTILS)
+all: libgif-$(LIBMAJOR).dll libgif.a $(UTILS)
 	$(MAKE) -C doc
 
 $(UTILS):: libgif.a
 
-libgif.so: $(OBJECTS) $(HEADERS)
-	$(CC) $(CFLAGS) -shared $(OFLAGS) -o libgif.so $(OBJECTS)
+libgif-$(LIBMAJOR).dll: $(OBJECTS) $(HEADERS)
+	$(CC) $(CFLAGS) -shared $(OFLAGS) -Wl,--out-implib,libgif.dll.a -o libgif-$(LIBMAJOR).dll $(OBJECTS)
 
 libgif.a: $(OBJECTS) $(HEADERS)
 	ar rcs libgif.a $(OBJECTS)
 
 clean:
-	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a libgif.so
-	rm -f libgif.so.$(LIBMAJOR).$(LIBMINOR).$(LIBPOINT)
-	rm -f libgif.so.$(LIBMAJOR)
+	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a libgif-$(LIBMAJOR).dll
 	rm -fr doc/*.1 *.html doc/staging
 
 check: all
@@ -86,15 +84,14 @@
 install-bin: $(INSTALLABLE)
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
 	$(INSTALL) $^ "$(DESTDIR)$(BINDIR)"
+	$(INSTALL) -m 755 libgif-$(LIBMAJOR).dll "$(DESTDIR)$(BINDIR)/libgif-$(LIBMAJOR).dll"
 install-include:
 	$(INSTALL) -d "$(DESTDIR)$(INCDIR)"
 	$(INSTALL) -m 644 gif_lib.h "$(DESTDIR)$(INCDIR)"
 install-lib:
 	$(INSTALL) -d "$(DESTDIR)$(LIBDIR)"
+	$(INSTALL) -m 644 libgif.dll.a "$(DESTDIR)$(LIBDIR)/libgif.dll.a"
 	$(INSTALL) -m 644 libgif.a "$(DESTDIR)$(LIBDIR)/libgif.a"
-	$(INSTALL) -m 755 libgif.so "$(DESTDIR)$(LIBDIR)/libgif.so.$(LIBVER)"
-	ln -sf libgif.so.$(LIBVER) "$(DESTDIR)$(LIBDIR)/libgif.so.$(LIBMAJOR)"
-	ln -sf libgif.so.$(LIBMAJOR) "$(DESTDIR)$(LIBDIR)/libgif.so"
 install-man:
 	$(INSTALL) -d "$(DESTDIR)$(MANDIR)"
 	$(INSTALL) -m 644 doc/*.1 "$(DESTDIR)$(MANDIR)"
@@ -105,7 +102,7 @@
 	rm -f "$(DESTDIR)$(INCDIR)/gif_lib.h"
 uninstall-lib:
 	cd "$(DESTDIR)$(LIBDIR)" && \
-		rm -f libgif.a libgif.so libgif.so.$(LIBMAJOR) libgif.so.$(LIBVER)
+		rm -f libgif.a libgif.dll.a libgif-$(LIBMAJOR).dll
 uninstall-man:
 	cd "$(DESTDIR)$(MANDIR)" && rm -f $(shell cd doc >/dev/null && echo *.1)
 
