diff -rupN --no-dereference giflib-5.2.1/doc/Makefile giflib-5.2.1-new/doc/Makefile
--- giflib-5.2.1/doc/Makefile	2019-03-28 18:05:25.000000000 +0100
+++ giflib-5.2.1-new/doc/Makefile	2020-02-17 16:51:04.489397582 +0100
@@ -1,7 +1,7 @@
 .SUFFIXES: .xml .html .txt .adoc .1
 
 .xml.html:
-	xmlto xhtml-nochunks $<
+	xmlto --stringparam generate.consistent.ids=1 xhtml-nochunks $<
 
 .xml.1:
 	xmlto man $<
