--- libwebsockets-4.0.1/lib/core-net/output.c.orig	2020-04-12 10:16:02.179670900 +0300
+++ libwebsockets-4.0.1/lib/core-net/output.c	2020-04-12 10:16:05.007733000 +0300
@@ -366,7 +366,7 @@
 	} else
 #endif
 		if (wsi->role_ops->file_handle)
-			n = write((int)(long long)wsi->desc.filefd, buf, len);
+			n = write((int)(intptr_t)wsi->desc.filefd, buf, len);
 		else
 			n = send(wsi->desc.sockfd, (char *)buf, len, MSG_NOSIGNAL);
 //	lwsl_info("%s: sent len %d result %d", __func__, len, n);
