--- scikit-learn-0.21.2/sklearn/_build_utils/openmp_helpers.py.orig	2019-06-14 15:31:08.101027100 +0300
+++ scikit-learn-0.21.2/sklearn/_build_utils/openmp_helpers.py	2019-06-14 15:33:57.720728800 +0300
@@ -36,7 +36,7 @@
 
     if sys.platform == "win32" and ('icc' in compiler or 'icl' in compiler):
         return ['/Qopenmp']
-    elif sys.platform == "win32":
+    elif sys.platform == "win32" and not 'GCC' in sys.version:
         return ['/openmp']
     elif sys.platform == "darwin" and ('icc' in compiler or 'icl' in compiler):
         return ['-openmp']
