From 00a3839e26aff416872bf6c51cf7a8d905313871 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 10 Jul 2014 23:39:15 +0100
Subject: [PATCH] Debugging cmake

---
 Build/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Build/CMakeLists.txt b/Build/CMakeLists.txt
index 2b38364..289968f 100755
--- a/Build/CMakeLists.txt
+++ b/Build/CMakeLists.txt
@@ -81,11 +81,13 @@ mark_as_advanced(FREETYPE_INCLUDE_DIRS FREETYPE_LIBRARY FREETYPE_LINK_DIRECTORIE
 if(BUILD_PYTHON_BINDINGS)
     find_package(PythonInterp 2 REQUIRED)
     find_package(PythonLibs 2 REQUIRED)
+    message(STATUS "PYTHON_EXECUTABLE is ${PYTHON_EXECUTABLE}")
     execute_process(
         COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(1,0,prefix=''))"
         OUTPUT_VARIABLE PYTHON_INSTDIR
         OUTPUT_STRIP_TRAILING_WHITESPACE
     )
+    message(STATUS "PYTHON_INSTDIR is ${PYTHON_INSTDIR}")
     if(PYTHONLIBS_FOUND)
         include_directories(${PYTHON_INCLUDE_DIR})
     endif()
-- 
2.0.1

