diff -Naur librocket-orig/Include/Rocket/Controls/Header.h librocket/Include/Rocket/Controls/Header.h
--- librocket-orig/Include/Rocket/Controls/Header.h	2014-07-11 09:39:04.672400000 +0400
+++ librocket/Include/Rocket/Controls/Header.h	2014-07-11 09:56:53.265200000 +0400
@@ -30,18 +30,18 @@
 
 #include <Rocket/Core/Platform.h>
 
-#if !defined STATIC_LIB
-	#ifdef ROCKET_PLATFORM_WIN32
-		#ifdef RocketControls_EXPORTS
+#if defined(ROCKET_PLATFORM_WIN32)
+		#if defined(RocketControls_EXPORTS) || !defined(STATIC_LIB)
 			#define ROCKETCONTROLS_API __declspec(dllexport)
 		#else
 			#define ROCKETCONTROLS_API __declspec(dllimport)
 		#endif
-	#else
+#else
+	#if !defined(STATIC_LIB)
 		#define ROCKETCONTROLS_API __attribute__((visibility("default")))
+	#else
+		#define ROCKETCONTROLS_API
 	#endif
-#else
-	#define ROCKETCONTROLS_API
 #endif
 
 #endif
diff -Naur librocket-orig/Include/Rocket/Core/Python/Header.h librocket/Include/Rocket/Core/Python/Header.h
--- librocket-orig/Include/Rocket/Core/Python/Header.h	2014-07-11 09:39:04.703600000 +0400
+++ librocket/Include/Rocket/Core/Python/Header.h	2014-07-11 09:53:23.991200000 +0400
@@ -30,18 +30,18 @@
 
 #include <Rocket/Core/Platform.h>
 
-#if !defined STATIC_LIB
-#if defined ROCKET_PLATFORM_WIN32
-	#if defined RocketCorePython_EXPORTS
+#if defined(ROCKET_PLATFORM_WIN32)
+	#if defined(RocketCorePython_EXPORTS) || !defined(STATIC_LIB)
 		#define ROCKETCOREPYTHON_API __declspec(dllexport)
 	#else
 		#define ROCKETCOREPYTHON_API __declspec(dllimport)
 	#endif
 #else
-	#define ROCKETCOREPYTHON_API __attribute__((visibility("default")))
-#endif
-#else
-	#define ROCKETCOREPYTHON_API
+	#if !defined(STATIC_LIB)
+		#define ROCKETCOREPYTHON_API __attribute__((visibility("default")))
+	#else
+		#define ROCKETCOREPYTHON_API
+	#endif
 #endif
 
 #endif
diff -Naur librocket-orig/Include/Rocket/Debugger/Header.h librocket/Include/Rocket/Debugger/Header.h
--- librocket-orig/Include/Rocket/Debugger/Header.h	2014-07-11 09:39:04.719200000 +0400
+++ librocket/Include/Rocket/Debugger/Header.h	2014-07-11 09:55:06.233600000 +0400
@@ -30,18 +30,18 @@
 
 #include <Rocket/Core/Platform.h>
 
-#if !defined STATIC_LIB
-	#ifdef ROCKET_PLATFORM_WIN32
-		#ifdef RocketDebugger_EXPORTS
+#if defined(ROCKET_PLATFORM_WIN32)
+		#if defined(RocketDebugger_EXPORTS) || !defined(STATIC_LIB)
 			#define ROCKETDEBUGGER_API __declspec(dllexport)
 		#else
 			#define ROCKETDEBUGGER_API __declspec(dllimport)
 		#endif
-	#else
+#else
+	#if !defined(STATIC_LIB)
 		#define ROCKETDEBUGGER_API __attribute__((visibility("default")))
+	#else
+		#define ROCKETDEBUGGER_API
 	#endif
-#else
-	#define ROCKETDEBUGGER_API
 #endif
 
 #endif
--- librocket/Include/Rocket/Core/Header.h.orig	2014-09-24 22:10:37.808000000 +0500
+++ librocket/Include/Rocket/Core/Header.h	2014-09-24 22:21:41.423200000 +0500
@@ -33,9 +33,8 @@
 // Note: Changing a ROCKETCORE_API_INLINE method
 // breaks ABI compatibility!!
 
-#if !defined STATIC_LIB
-	#if defined ROCKET_PLATFORM_WIN32
-		#if defined RocketCore_EXPORTS
+#if defined(ROCKET_PLATFORM_WIN32)
+		#if (defined RocketCore_EXPORTS) || !defined(STATIC_LIB)
 			#define ROCKETCORE_API __declspec(dllexport)
 			// Note: Changing a ROCKETCORE_API_INLINE method
 			// breaks ABI compatibility!!
@@ -69,17 +68,18 @@
 				#define ROCKETCORE_API_INLINE inline
 			#endif
 		#endif
-	#else
+#else
+	#if !defined(STATIC_LIB)
 		#define ROCKETCORE_API __attribute__((visibility("default")))
 		// Note: Changing a ROCKETCORE_API_INLINE method
 		// breaks ABI compatibility!!
 		#define ROCKETCORE_API_INLINE __attribute__((visibility("default"))) inline
+	#else
+		#define ROCKETCORE_API
+		// Note: Changing a ROCKETCORE_API_INLINE method
+		// breaks ABI compatibility!!
+		#define ROCKETCORE_API_INLINE inline
 	#endif
-#else
-	#define ROCKETCORE_API
-	// Note: Changing a ROCKETCORE_API_INLINE method
-	// breaks ABI compatibility!!
-	#define ROCKETCORE_API_INLINE inline
 #endif
 
 #endif
