--- librocket/Samples/shell/src/win32/ShellRenderInterfaceExtensionsOpenGL_Win32.cpp.orig	2015-07-20 14:59:44.514453800 +0300
+++ librocket/Samples/shell/src/win32/ShellRenderInterfaceExtensionsOpenGL_Win32.cpp	2015-07-20 14:59:48.544010500 +0300
@@ -119,7 +119,7 @@
 	glMatrixMode(GL_PROJECTION);
 	glLoadIdentity();
 	RECT crect;
-	GetClientRect(this->window_handle, &crect)
+	GetClientRect(this->window_handle, &crect);
 	glOrtho(0, (crect.right - crect.left), (crect.bottom - crect.top), 0, -1, 1);
 
 	glMatrixMode(GL_MODELVIEW);
