--- librocket/Build/CMakeLists.txt.orig	2014-07-11 08:36:08.919800000 +0400
+++ librocket/Build/CMakeLists.txt	2014-07-11 08:41:33.611600000 +0400
@@ -95,7 +109,7 @@
     find_package(Boost 1.40.0 COMPONENTS python REQUIRED)
     if(Boost_FOUND)
         include_directories(${Boost_INCLUDE_DIR})
-        list(APPEND PY_BINDINGS_LINK_LIBS ${Boost_LIBRARIES})
+        list(APPEND PY_BINDINGS_LINK_LIBS ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
     endif()
 
 endif()
@@ -161,7 +175,7 @@
         set_target_properties(${NAME} PROPERTIES PREFIX "")
 
         install(TARGETS ${NAME}
-                LIBRARY DESTINATION ${PYTHON_INSTDIR}
+                DESTINATION ${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTDIR}
         )
     endforeach(library)
 endif()
@@ -297,14 +311,14 @@
 
 if(BUILD_PYTHON_BINDINGS)
     install(FILES ${PROJECT_SOURCE_DIR}/bin/rocket.py
-            DESTINATION ${PYTHON_INSTDIR}
+            DESTINATION ${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTDIR}
     )
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/Include/Rocket
-            DESTINATION include
+            DESTINATION ${CMAKE_INSTALL_PREFIX}/include
     )
 else()
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/Include/Rocket
-            DESTINATION include
+            DESTINATION ${CMAKE_INSTALL_PREFIX}/include
             PATTERN "Python" EXCLUDE
     )
 endif()
