diff --git a/src/configmanager.cpp b/src/configmanager.cpp
index fba59a912..8204bc293 100644
--- a/src/configmanager.cpp
+++ b/src/configmanager.cpp
@@ -895,9 +895,9 @@ QSettings *ConfigManager::readSettings(bool reread)
 #define PREFIX
 #endif
 			fallBackPaths << PREFIX"/share/hunspell" << PREFIX"/share/myspell"
-                          << "/usr/share/hunspell" << "/usr/share/myspell"
-                          << parseDir("[txs-app-dir]/../share/texstudio") ;
+                          << "/usr/share/hunspell" << "/usr/share/myspell";
 #endif
+            fallBackPaths << parseDir("[txs-app-dir]/../share/texstudio");
 #ifdef Q_OS_MAC
 			fallBackPaths << parseDir("[txs-app-dir]/Contents/Resources") << "/Applications/texstudio.app/Contents/Resources";
 #endif
@@ -935,8 +935,10 @@ QSettings *ConfigManager::readSettings(bool reread)
 		QStringList preferredPaths = QStringList() << parseDir("[txs-settings-dir]/dictionaries");
 		QStringList fallBackPaths;
 #ifdef Q_OS_LINUX
-        fallBackPaths << PREFIX"/share/mythes" << "/usr/share/mythes"
-                      << parseDir("[txs-app-dir]/../share/texstudio") ;
+        fallBackPaths << PREFIX"/share/mythes" << "/usr/share/mythes";
+#endif
+#if defined(Q_OS_LINUX) || defined(Q_OS_WIN32)
+        fallBackPaths << parseDir("[txs-app-dir]/../share/texstudio");
 #endif
 		thesaurus_database = findResourceFile("th_" + QString(QLocale::system().name()) + "_v2.dat", true, preferredPaths, fallBackPaths);
 		if (thesaurus_database == "") thesaurus_database = findResourceFile("th_en_US_v2.dat", true, preferredPaths, fallBackPaths);
diff --git a/src/templatemanager.cpp b/src/templatemanager.cpp
index 0eeafb90d..1ad1a9954 100644
--- a/src/templatemanager.cpp
+++ b/src/templatemanager.cpp
@@ -26,7 +26,11 @@ QString TemplateManager::builtinTemplateDir()
 	return fn;
 #endif
 #ifdef Q_OS_WIN
-	return QCoreApplication::applicationDirPath() + "/templates/";
+	QString fn = QCoreApplication::applicationDirPath() + "/../share/texstudio/";
+	if (!QDir(fn).isReadable()) { // fallback if program is not packaged as app (e.g. debug build )
+		return QCoreApplication::applicationDirPath() + "/templates/";
+	}
+	return fn;
 #else
 #if !defined(PREFIX)
 #define PREFIX ""
