--- texstudio-4.7.3.orig/CMakeLists.txt	2024-03-02 18:37:23.738496100 +0300
+++ texstudio-4.7.3/CMakeLists.txt	2024-03-02 18:47:40.601729500 +0300
@@ -26,8 +26,11 @@
 endif()
 include(cmake/project_version.cmake)
 
-if(MSVC)
+if(WIN32)
 	set(ADDITIONAL_DEFINITIONS ${ADDITIONAL_DEFINITIONS} -D_CRT_SECURE_NO_WARNINGS -D_UNICODE -DUNICODE)
+endif()
+
+if(MSVC)
 	add_compile_options(/wd4267 /wd4244)
 	add_compile_options($<$<CXX_COMPILER_ID:MSVC>:/MP>)
 else()
