--- gdb-7.6/bfd/Makefile.am.orig	2013-06-12 01:07:47 +0400
+++ gdb-7.6/bfd/Makefile.am	2013-06-12 01:06:59 +0400
@@ -826,6 +826,8 @@
 
 ofiles: stamp-ofiles ; @true
 
+LIBIBERTY = ../libiberty/libiberty.a
+
 # Since BFD64_LIBS is optional and we can't have substitution in
 # libbfd_la_SOURCES, we put BFD64_LIBS in OFILES instead.
 # However, list all sources in EXTRA_libbfd_la_SOURCES so the
@@ -833,6 +835,6 @@
 EXTRA_libbfd_la_SOURCES = $(CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
 libbfd_la_LIBADD = `cat ofiles` @SHARED_LIBADD@ $(LIBDL)
-libbfd_la_LDFLAGS += -release `cat libtool-soversion` @SHARED_LDFLAGS@
+libbfd_la_LDFLAGS += -release `cat libtool-soversion` @SHARED_LDFLAGS@ -Wl,--whole-archive -Wl,$(LIBIBERTY) -Wl,$(LIBINTL)  -Wl,--no-whole-archive
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
--- binutils-git-e6cf2ae8bb4c675ebfafb137bf13e92ba96225e8/bfd/Makefile.in.orig	2014-07-23 15:07:57.487590651 +0400
+++ binutils-git-e6cf2ae8bb4c675ebfafb137bf13e92ba96225e8/bfd/Makefile.in	2014-07-23 15:09:45.500342237 +0400
@@ -342,7 +342,7 @@
 SUBDIRS = doc po
 bfddocdir = doc
 libbfd_la_LDFLAGS = $(am__append_1) -release `cat libtool-soversion` \
-	@SHARED_LDFLAGS@ $(am__empty)
+	@SHARED_LDFLAGS@ $(am__empty) -no-undefined -Wl,--whole-archive -Wl,$(LIBIBERTY) -Wl,$(LIBINTL) -Wl,--no-whole-archive
 @INSTALL_LIBBFD_TRUE@bfdlib_LTLIBRARIES = libbfd.la
 @INSTALL_LIBBFD_FALSE@bfdinclude_HEADERS = $(am__append_2)
 @INSTALL_LIBBFD_TRUE@bfdinclude_HEADERS = $(BFD_H) \
@@ -1097,6 +1097,8 @@
 # bfd64_libs	Routines for 64bit support
 OFILES = $(BFD_BACKENDS) $(BFD_MACHINES) @COREFILE@ @bfd64_libs@
 
+LIBIBERTY = ../libiberty/libiberty.a
+
 # Since BFD64_LIBS is optional and we can't have substitution in
 # libbfd_la_SOURCES, we put BFD64_LIBS in OFILES instead.
 # However, list all sources in EXTRA_libbfd_la_SOURCES so the
@@ -1104,7 +1106,7 @@
 libbfd_la_SOURCES = $(BFD32_LIBS_CFILES)
 EXTRA_libbfd_la_SOURCES = $(CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
-libbfd_la_LIBADD = `cat ofiles` @SHARED_LIBADD@ $(LIBDL)
+libbfd_la_LIBADD = `cat ofiles` @SHARED_LIBADD@ $(LIBDL)
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
 # directory so that we don't have to convert all the programs that use
