diff -urN binutils-2.27.orig/libiberty/Makefile.in binutils-2.27/libiberty/Makefile.in
--- binutils-2.27.orig/libiberty/Makefile.in	2016-08-03 09:36:54.000000000 +0200
+++ binutils-2.27/libiberty/Makefile.in	2016-09-21 21:10:01.901622600 +0200
@@ -131,7 +131,7 @@
 	d-demangle.c dwarfnames.c dyn-string.c				\
 	fdmatch.c ffs.c fibheap.c filename_cmp.c floatformat.c		\
 	fnmatch.c fopen_unlocked.c					\
-	getcwd.c getopt.c getopt1.c getpagesize.c getpwd.c getruntime.c	\
+	getcwd.c getpagesize.c getpwd.c getruntime.c	\
          gettimeofday.c                                                 \
 	hashtab.c hex.c							\
 	index.c insque.c						\
@@ -156,9 +156,9 @@
 	 strtoll.c strtoul.c strtoull.c strndup.c strnlen.c             \
 	 strverscmp.c timeval-utils.c tmpnam.c				\
 	unlink-if-ordinary.c						\
-	vasprintf.c vfork.c vfprintf.c vprintf.c vprintf-support.c	\
+	vasprintf.c vfprintf.c vprintf.c vprintf-support.c	\
 	 vsnprintf.c vsprintf.c						\
-	waitpid.c							\
+	msdos.c							\
 	xasprintf.c xatexit.c xexit.c xmalloc.c xmemdup.c xstrdup.c	\
 	 xstrerror.c xstrndup.c xvasprintf.c

@@ -174,7 +174,7 @@
 	./fdmatch.$(objext) ./fibheap.$(objext)				\
 	./filename_cmp.$(objext) ./floatformat.$(objext)		\
 	./fnmatch.$(objext) ./fopen_unlocked.$(objext)			\
-	./getopt.$(objext) ./getopt1.$(objext) ./getpwd.$(objext)	\
+	./getpwd.$(objext)	\
 	./getruntime.$(objext) ./hashtab.$(objext) ./hex.$(objext)	\
 	./lbasename.$(objext) ./lrealpath.$(objext)			\
 	./make-relative-prefix.$(objext) ./make-temp-file.$(objext)	\
@@ -224,9 +224,9 @@
 	 ./strrchr.$(objext) ./strstr.$(objext) ./strtod.$(objext)	\
 	 ./strtol.$(objext) ./strtoul.$(objext) strtoll.$(objext)	\
 	./strtoull.$(objext) ./tmpnam.$(objext) ./strverscmp.$(objext)	\
-	./vasprintf.$(objext) ./vfork.$(objext) ./vfprintf.$(objext)	\
+	./vasprintf.$(objext) ./vfprintf.$(objext)	\
 	 ./vprintf.$(objext) ./vsnprintf.$(objext) ./vsprintf.$(objext)	\
-	./waitpid.$(objext)
+	./msdos.$(objext)

 # These files are installed if the library has been configured to do so.
 INSTALLED_HEADERS =                                                     \
