--- binutils-2.23.52-orig/libiberty/configure	2013-05-31 20:01:43.000000000 -0500
+++ binutils-2.23.52/libiberty/configure	2013-06-04 15:04:57.162176000 -0500
@@ -5507,7 +5507,6 @@ fi
 
 setobjs=
 CHECK=
-target_header_dir=
 if test -n "${with_target_subdir}"; then
 
   # We are being configured as a target library.  AC_REPLACE_FUNCS
--- binutils-2.23.52-orig/libiberty/configure.ac	2013-05-31 20:01:44.000000000 -0500
+++ binutils-2.23.52/libiberty/configure.ac	2013-06-04 15:04:48.269667400 -0500
@@ -405,7 +405,6 @@ fi
 
 setobjs=
 CHECK=
-target_header_dir=
 if test -n "${with_target_subdir}"; then
 
   # We are being configured as a target library.  AC_REPLACE_FUNCS
