--- binutils-2.34.orig/binutils/windres.c
+++ binutils-2.34/binutils/windres.c
@@ -962,8 +962,8 @@
 	  if (preprocargs == NULL)
 	    {
 	      quotedarg = quot (optarg);
-	      preprocargs = xmalloc (strlen (quotedarg) + 3);
-	      sprintf (preprocargs, "-I%s", quotedarg);
+	      preprocargs = xmalloc (strlen (quotedarg) + 5);
+	      sprintf (preprocargs, "-I\"%s\"", quotedarg);
 	    }
 	  else
 	    {
@@ -971,8 +971,8 @@
 	      char *n;
 
 	      quotedarg = quot (optarg);
-	      n = xmalloc (strlen (preprocargs) + strlen (quotedarg) + 4);
-	      sprintf (n, "%s -I%s", preprocargs, quotedarg);
+	      n = xmalloc (strlen (preprocargs) + strlen (quotedarg) + 6);
+	      sprintf (n, "%s -I\"%s\"", preprocargs, quotedarg);
 	      free (preprocargs);
 	      preprocargs = n;
 	    }
