--- a/gas/config/tc-i386.c	
+++ a/gas/config/tc-i386.c	
@@ -9145,12 +9145,6 @@ md_apply_fix (fixS *fixP, valueT *valP, segT seg ATTRIBUTE_UNUSED)
 	value += md_pcrel_from (fixP);
 #endif
     }
-#if defined (OBJ_COFF) && defined (TE_PE)
-  if (fixP->fx_addsy != NULL && S_IS_WEAK (fixP->fx_addsy))
-    {
-      value -= S_GET_VALUE (fixP->fx_addsy);
-    }
-#endif
 
   /* Fix a few things - the dynamic linker expects certain values here,
      and we must not disappoint it.  */
