--- ./ld/pdb.c.orig	2025-02-02 03:00:00 +0300
+++ ./ld/pdb.c	2025-05-05 13:20:14 +0300
@@ -1880,7 +1880,7 @@
       return false;
     }
 
-  version = bfd_getl32 (data + sizeof (uint32_t) + sizeof (uint32_t));
+  version = bfd_getl32 (data);
   if (version != CV_INLINEE_SOURCE_LINE_SIGNATURE)
     {
       einfo (_("%P: warning: unexpected DEBUG_S_INLINEELINES version %u\n"),
@@ -4362,7 +4362,7 @@
 			      struct source_files_info *source)
 {
   uint16_t dedupe_source_files_count = 0;
-  uint16_t source_files_count = 0;
+  uint32_t source_files_count = 0;
   uint32_t strings_len = 0;
   uint8_t *ptr;
 
