--- ./src/build-tools/generator.c.orig	2015-03-25 17:48:37.812068900 +0100
+++ ./src/build-tools/generator.c	2015-03-25 17:48:40.060777800 +0100
@@ -84,7 +84,7 @@
 
 			if (iter == strlen (full_flag)) {
 				full_comment = g_new (gchar, strlen (para->comment) - strlen (full_flag) + 1);
-				stpcpy (full_comment, para->comment + strlen(full_flag));
+				g_stpcpy (full_comment, para->comment + strlen(full_flag));
 				buffer = g_strconcat (res, "\n * ", full_comment, NULL);
 				g_free (res);
 				res = buffer;
--- ./src/libical-glib/Makefile.am.orig	2015-03-25 17:47:15.849595300 +0100
+++ ./src/libical-glib/Makefile.am	2015-03-25 17:47:22.483886100 +0100
@@ -131,6 +131,7 @@
 INTROSPECTION_COMPILER_ARGS = \
 	--includedir=$(srcdir) \
 	$(NULL)
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 if HAVE_INTROSPECTION
 introspection_sources = $(libical_glib_1_0_la_SOURCES)
