diff -aurN 005/source/pidgin/win32/gtkwin32dep.c 006/source/pidgin/win32/gtkwin32dep.c
--- 005/source/pidgin/win32/gtkwin32dep.c
+++ 006/source/pidgin/win32/gtkwin32dep.c
@@ -393,6 +393,7 @@
 void winpidgin_init(HINSTANCE hint) {
 	gchar *debug_dir, *locale_debug_dir;
 	INITCOMMONCONTROLSEX icc;
+	GtkSettings *settings;
 
 	purple_debug_info("winpidgin", "winpidgin_init start\n");
 
@@ -401,6 +402,10 @@
 	icc.dwICC = ICC_STANDARD_CLASSES;
 	InitCommonControlsEx(&icc);
 
+	/* Set GTK+ theme */
+	settings = gtk_settings_get_default();
+	gtk_settings_set_string_property(settings, "gtk-theme-name", APPLICATION_NAME, NULL);
+
 	exe_hInstance = hint;
 	debug_dir = g_build_filename(purple_user_dir(), "pidgin.rpt", NULL);
 	locale_debug_dir = g_locale_from_utf8(debug_dir, -1, NULL, NULL, NULL);
