diff -uNr wineditline-2.205.orig/wineditline.pc wineditline-2.205/wineditline.pc
--- wineditline-2.205.orig/wineditline.pc	1970-01-01 09:00:00.000000000 +0900
+++ wineditline-2.205/wineditline.pc	2019-07-03 00:29:29.029406918 +0900
@@ -0,0 +1,10 @@
+prefix=${prefix}
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: wineditline
+Description: EditLine API implementation for the native Windows Console
+Version: 2.205
+Cflags: -I${includedir}/editline
+Libs: -L${libdir} -ledit
