--- a/src/assistant/assistant/CMakeLists.txt
+++ b/src/assistant/assistant/CMakeLists.txt
@@ -196,3 +196,6 @@
     doc/qtassistant.qdocconf
 )
 
+set_target_properties(assistant PROPERTIES
+    OUTPUT_NAME "assistant-qt6"
+)
--- a/src/assistant/qhelpgenerator/CMakeLists.txt
+++ b/src/assistant/qhelpgenerator/CMakeLists.txt
@@ -25,3 +25,7 @@
 # QMAKE_TARGET_DESCRIPTION = "Qt Compressed Help File Generator"
 # QTPLUGIN.platforms = "qminimal"
 # QTPLUGIN.sqldrivers = "qsqlite"
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "qhelpgenerator-qt6"
+)
--- a/src/designer/src/designer/CMakeLists.txt
+++ b/src/designer/src/designer/CMakeLists.txt
@@ -145,3 +145,7 @@
 qt_internal_add_docs(designer
     doc/qtdesigner.qdocconf
 )
+
+set_target_properties(designer PROPERTIES
+    OUTPUT_NAME "designer-qt6"
+)
--- a/src/distancefieldgenerator/CMakeLists.txt
+++ b/src/distancefieldgenerator/CMakeLists.txt
@@ -34,3 +34,6 @@
     doc/qtdistancefieldgenerator.qdocconf
 )
 
+set_target_properties(qdistancefieldgenerator PROPERTIES
+    OUTPUT_NAME "qdistancefieldgenerator-qt6"
+)
--- a/src/linguist/lconvert/CMakeLists.txt
+++ b/src/linguist/lconvert/CMakeLists.txt
@@ -31,3 +31,7 @@
 #### Keys ignored in scope 1:.:.:lconvert.pro:<TRUE>:
 # QMAKE_TARGET_DESCRIPTION = "Qt Translation File Converter"
 # _OPTION = "host_build"
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "lconvert-qt6"
+)
--- a/src/linguist/linguist/CMakeLists.txt
+++ b/src/linguist/linguist/CMakeLists.txt
@@ -175,3 +175,6 @@
     doc/qtlinguist.qdocconf
 )
 
+set_target_properties(linguist PROPERTIES
+    OUTPUT_NAME "linguist-qt6"
+)
--- a/src/linguist/lrelease/CMakeLists.txt
+++ b/src/linguist/lrelease/CMakeLists.txt
@@ -40,3 +40,7 @@
 # _OPTION = "host_build"
 # qmake.name = "QMAKE"
 # qmake.value = "$$shell_path($$QMAKE_QMAKE)"
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "lrelease-qt6"
+)
--- a/src/linguist/lupdate/CMakeLists.txt
+++ b/src/linguist/lupdate/CMakeLists.txt
@@ -107,3 +107,7 @@
 
 #### Keys ignored in scope 9:.:.:lupdate.pro:MINGW:
 # RC_FILE = "lupdate.rc"
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME lupdate-qt6
+)
--- a/src/pixeltool/CMakeLists.txt
+++ b/src/pixeltool/CMakeLists.txt
@@ -28,3 +28,7 @@
         MACOSX_BUNDLE TRUE
     )
 endif()
+
+set_target_properties(pixeltool PROPERTIES
+    OUTPUT_NAME "pixeltool-qt6"
+)
--- a/src/qdbus/qdbus/CMakeLists.txt
+++ b/src/qdbus/qdbus/CMakeLists.txt
@@ -20,3 +20,7 @@
         WIN32_EXECUTABLE FALSE
     )
 endif()
+
+set_target_properties(qdbus PROPERTIES
+    OUTPUT_NAME "qdbus-qt6"
+)
--- a/src/qdbus/qdbusviewer/CMakeLists.txt
+++ b/src/qdbus/qdbusviewer/CMakeLists.txt
@@ -61,3 +61,7 @@
         QT_TARGET_RC_ICONS "${CMAKE_CURRENT_SOURCE_DIR}/images/qdbusviewer.ico"
     )
 endif()
+
+set_target_properties(qdbusviewer PROPERTIES
+    OUTPUT_NAME "qdbusviewer-qt6"
+)
--- a/src/qdoc/CMakeLists.txt
+++ b/src/qdoc/CMakeLists.txt
@@ -131,3 +131,6 @@
     doc/config/qdoc.qdocconf
 )
 
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "qdoc-qt6"
+)
--- a/src/qtdiag/CMakeLists.txt
+++ b/src/qtdiag/CMakeLists.txt
@@ -5,7 +5,6 @@
 #####################################################################
 
 qt_internal_add_app(qtdiag
-    INSTALL_VERSIONED_LINK
     SOURCES
         main.cpp
         qtdiag.cpp qtdiag.h
@@ -54,3 +53,7 @@
     PUBLIC_LIBRARIES
         Qt::Network
 )
+
+set_target_properties(qtdiag PROPERTIES
+    OUTPUT_NAME "qtdiag-qt6"
+)
--- a/src/qtplugininfo/CMakeLists.txt
+++ b/src/qtplugininfo/CMakeLists.txt
@@ -15,3 +15,7 @@
 #### Keys ignored in scope 1:.:.:qtplugininfo.pro:<TRUE>:
 # QT_FOR_CONFIG = "tools-private"
 # _REQUIREMENTS = "qtConfig(qtplugininfo)"
+
+set_target_properties(qtplugininfo PROPERTIES
+    OUTPUT_NAME "qtplugininfo-qt6"
+)
--- a/src/windeployqt/CMakeLists.txt
+++ b/src/windeployqt/CMakeLists.txt
@@ -43,3 +43,7 @@
     DEFINES
         QT_RELOCATABLE
 )
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "windeployqt-qt6"
+)
--- a/src/shared/winutils/utils.cpp
--- b/src/shared/winutils/utils.cpp
@@ -457,7 +457,7 @@
 
 QMap<QString, QString> queryQMakeAll(const QString &qmakeBinary, QString *errorMessage)
 {
-    const QString binary = !qmakeBinary.isEmpty() ? qmakeBinary : QStringLiteral("qmake");
+    const QString binary = !qmakeBinary.isEmpty() ? qmakeBinary : QStringLiteral("qmake6");
     QByteArray stdOut;
     QByteArray stdErr;
     unsigned long exitCode = 0;
--- a/src/windeployqt/main.cpp
--- b/src/windeployqt/main.cpp
@@ -1001,7 +1001,7 @@
     // Run lconvert to concatenate all files into a single named "qt_<prefix>.qm" in the application folder
     // Use QT_INSTALL_TRANSLATIONS as working directory to keep the command line short.
     const QString absoluteTarget = QFileInfo(target).absoluteFilePath();
-    const QString binary = QStringLiteral("lconvert");
+    const QString binary = QStringLiteral("lconvert-qt6");
     QStringList arguments;
     for (const QString &prefix : qAsConst(prefixes)) {
         arguments.clear();
