--- a/src/assistant/assistant/CMakeLists.txt
+++ b/src/assistant/assistant/CMakeLists.txt
@@ -175,3 +175,6 @@
     doc/qtassistant.qdocconf
 )
 
+set_target_properties(assistant PROPERTIES
+    OUTPUT_NAME "assistant-qt6"
+)
--- a/src/designer/src/designer/CMakeLists.txt
+++ b/src/designer/src/designer/CMakeLists.txt
@@ -149,3 +149,7 @@
 qt_internal_add_docs(designer
     doc/qtdesigner.qdocconf
 )
+
+set_target_properties(designer PROPERTIES
+    OUTPUT_NAME "designer-qt6"
+)
--- a/src/distancefieldgenerator/CMakeLists.txt
+++ b/src/distancefieldgenerator/CMakeLists.txt
@@ -32,3 +32,6 @@
     doc/qtdistancefieldgenerator.qdocconf
 )
 
+set_target_properties(qdistancefieldgenerator PROPERTIES
+    OUTPUT_NAME "qdistancefieldgenerator-qt6"
+)
--- a/src/linguist/lconvert/CMakeLists.txt
+++ b/src/linguist/lconvert/CMakeLists.txt
@@ -29,3 +29,7 @@
         Qt::CorePrivate
 )
 qt_internal_return_unless_building_tools()
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "lconvert-qt6"
+)
--- a/src/linguist/linguist/CMakeLists.txt
+++ b/src/linguist/linguist/CMakeLists.txt
@@ -176,3 +176,6 @@
     doc/qtlinguist.qdocconf
 )
 
+set_target_properties(linguist PROPERTIES
+    OUTPUT_NAME "linguist-qt6"
+)
--- a/src/linguist/lrelease/CMakeLists.txt
+++ b/src/linguist/lrelease/CMakeLists.txt
@@ -37,3 +37,7 @@
     # special case end
 )
 qt_internal_return_unless_building_tools()
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "lrelease-qt6"
+)
--- a/src/linguist/lupdate/CMakeLists.txt
+++ b/src/linguist/lupdate/CMakeLists.txt
@@ -107,3 +107,7 @@
 if(QT_FEATURE_clangcpp)
     set_property(SOURCE clangtoolastreader.cpp PROPERTY SKIP_AUTOMOC ON)
 endif()
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME lupdate-qt6
+)
--- a/src/pixeltool/CMakeLists.txt
+++ b/src/pixeltool/CMakeLists.txt
@@ -25,3 +25,7 @@
         MACOSX_BUNDLE TRUE
     )
 endif()
+
+set_target_properties(pixeltool PROPERTIES
+    OUTPUT_NAME "pixeltool-qt6"
+)
--- a/src/qdbus/qdbus/CMakeLists.txt
+++ b/src/qdbus/qdbus/CMakeLists.txt
@@ -21,3 +21,7 @@
         WIN32_EXECUTABLE FALSE
     )
 endif()
+
+set_target_properties(qdbus PROPERTIES
+    OUTPUT_NAME "qdbus-qt6"
+)
--- a/src/qdbus/qdbusviewer/CMakeLists.txt
+++ b/src/qdbus/qdbusviewer/CMakeLists.txt
@@ -60,3 +60,7 @@
         QT_TARGET_RC_ICONS "${CMAKE_CURRENT_SOURCE_DIR}/images/qdbusviewer.ico"
     )
 endif()
+
+set_target_properties(qdbusviewer PROPERTIES
+    OUTPUT_NAME "qdbusviewer-qt6"
+)
--- a/src/qdoc/qdoc/CMakeLists.txt
+++ b/src/qdoc/qdoc/CMakeLists.txt
@@ -114,3 +114,6 @@
 if(QT_BUILD_TESTS)
     add_subdirectory(tests)
 endif()
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "qdoc-qt6"
+)
--- a/src/qtdiag/CMakeLists.txt
+++ b/src/qtdiag/CMakeLists.txt
@@ -6,7 +6,6 @@
 #####################################################################
 
 qt_internal_add_app(qtdiag
-    INSTALL_VERSIONED_LINK
     SOURCES
         main.cpp
         qtdiag.cpp qtdiag.h
@@ -49,3 +48,7 @@
     PUBLIC_LIBRARIES
         Qt::Network
 )
+
+set_target_properties(qtdiag PROPERTIES
+    OUTPUT_NAME "qtdiag-qt6"
+)
--- a/src/qtplugininfo/CMakeLists.txt
+++ b/src/qtplugininfo/CMakeLists.txt
@@ -12,3 +12,7 @@
 set_target_properties(qtplugininfo PROPERTIES
     WIN32_EXECUTABLE FALSE
 )
+
+set_target_properties(qtplugininfo PROPERTIES
+    OUTPUT_NAME "qtplugininfo-qt6"
+)
