--- nsis-3.03-src/Source/util.cpp.orig	2018-11-01 14:19:48.740819800 +0300
+++ nsis-3.03-src/Source/util.cpp	2018-11-02 07:56:12.557924400 +0300
@@ -1030,7 +1030,8 @@
       BOOL okr = ReadFile(hSORd, iobuf+cbOfs, sizeof(iobuf)-cbOfs, &cbRead, 0);
       cbRead += cbOfs, cbOfs = 0;
       unsigned char cbTrail, cch;
-      for(DWORD i = 0; i < cbRead;)
+      DWORD i = 0;
+      for(i = 0; i < cbRead;)
       {
         cch = 0;
         if (utf8)
