--- a/setup.py
+++ b/setup.py
@@ -70,7 +70,7 @@
     'darwin': False,
     'linux': True,
     'freebsd': False, # not tested
-    'win32': False,   # no icu-config
+    'win32': True,
     'sunos5': False,  # not tested
     'cygwin': False,  # not tested
 }
@@ -79,7 +79,7 @@
     'darwin': False,  # no pkg-config ?
     'linux': True,
     'freebsd': False, # not tested
-    'win32': False,   # no pkg-config ?
+    'win32': True,
     'sunos5': False,  # not tested
     'cygwin': False,  # not tested
 }
@@ -111,7 +111,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['/usr/local/include'],
-    'win32': ['c:/icu/include'],
+    'win32': [],
     'sunos5': [],
     'cygwin': [],
 }
@@ -117,7 +117,7 @@
 }
 
 if sys.platform == 'win32' and sys.version_info < (3,9):
-    ver_flag = '/D%s=\\"%s\\"'
+    ver_flag = '-D%s=\"%s\"'
 else:
     ver_flag = '-D%s="%s"'
 
@@ -138,7 +138,7 @@
     'darwin': ['-std=c++17'],
     'linux': ['-std=c++17'],
     'freebsd': ['-std=c++17'],
-    'win32': ['/Zc:wchar_t', '/EHsc'],
+    'win32': ['-std=c++17'],
     'sunos5': ['-std=c++17'],
     'cygwin': ['-D_GNU_SOURCE=1', '-std=c++17'],
 }
@@ -148,7 +148,7 @@
     'darwin': ['-O0', '-g', '-DDEBUG'],
     'linux': ['-O0', '-g', '-DDEBUG'],
     'freebsd': ['-O0', '-g', '-DDEBUG'],
-    'win32': ['/Od', '/DDEBUG'],
+    'win32': ['-O0', '-g', '-DDEBUG'],
     'sunos5': ['-DDEBUG'],
     'cygwin': ['-Og', '-g', '-DDEBUG'],
 }
@@ -157,7 +157,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['-L/usr/local/lib'],
-    'win32': ['/LIBPATH:c:/icu/lib'],
+    'win32': [],
     'sunos5': [],
     'cygwin': [],
 }
@@ -166,7 +166,7 @@
     'darwin': [],
     'linux': [],
     'freebsd': ['icui18n', 'icuuc', 'icudata'],
-    'win32': ['icuin', 'icuuc', 'icudt'],
+    'win32': [],
     'sunos5': ['icui18n', 'icuuc', 'icudata'],
     'cygwin': ['icui18n', 'icuuc', 'icudata'],
 }
