diff -Naur showtime-50.0-orig/meson.build showtime-50.0/meson.build
--- showtime-50.0-orig/meson.build	2026-03-14 07:31:00.000000000 +0000
+++ showtime-50.0/meson.build	2026-09-06 00:00:00.000000000 +0000
@@ -36,5 +36,9 @@
 python_exec = get_option('python_exec')
 if python_exec == ''
-    conf.set('PYTHON', py_installation.full_path())
+    if host_machine.system() == 'windows'
+        conf.set('PYTHON', join_paths(get_option('prefix'), 'bin', 'pythonw.exe'))
+    else
+        conf.set('PYTHON', py_installation.full_path())
+    endif
 else
     conf.set('PYTHON', python_exec)
