--- l-smash-2.14.5.orig/configure	2021-07-03 21:44:02.294576100 -0400
+++ l-smash-2.14.5/configure	2021-07-03 21:44:48.668087200 -0400
@@ -211,7 +211,7 @@
         SHARED_EXT=".dll"
         DEFNAME="${SHARED_NAME}.def"
         IMPLIB="liblsmash.dll.a"
-        SO_LDFLAGS="-shared -Wl,--output-def,$DEFNAME -Wl,--out-implib,$IMPLIB -Wl,--version-script,liblsmash.ver"
+        SO_LDFLAGS="-shared -Wl,--output-def,$DEFNAME -Wl,--out-implib,$IMPLIB"
         CFLAGS="$CFLAGS -D__USE_MINGW_ANSI_STDIO=1"
         LIBARCH=i386
         if lib.exe --list > /dev/null 2>&1 ; then
