diff -ruN librsvg-2.45.5/Makefile.am librsvg-2.45.5-fixed/Makefile.am
--- librsvg-2.45.5/Makefile.am	2019-02-16 01:01:25.000000000 +0800
+++ librsvg-2.45.5-fixed/Makefile.am	2019-03-02 20:17:17.432974000 +0800
@@ -137,6 +137,7 @@
 cargo_verbose_1 = --verbose
 
 RUST_LIB=@abs_top_builddir@/target/@RUST_TARGET_SUBDIR@/librsvg_internals.a
+RUST_LIB_WIN=@abs_top_builddir@/target/@RUST_TARGET_SUBDIR@/rsvg_internals.lib
 CARGO_TARGET_DIR=@abs_top_builddir@/target
 
 check-local:
@@ -167,7 +168,7 @@
 	PKG_CONFIG_ALLOW_CROSS=1						\
 	PKG_CONFIG='$(PKG_CONFIG)'						\
 	CARGO_TARGET_DIR=$(CARGO_TARGET_DIR)					\
-	$(CARGO) --locked build $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS)
+	$(CARGO) --locked build $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS) && cp $(RUST_LIB_WIN) $(RUST_LIB)
 
 librsvg_@RSVG_API_MAJOR_VERSION@_la_CPPFLAGS = \
 	-I$(top_srcdir) 			\
