--- librsvg-2.58.93/rsvg/meson.build.orig	2024-08-02 02:33:14.000000000 +0200
+++ librsvg-2.58.93/rsvg/meson.build	2024-08-18 12:47:12.661963600 +0200
@@ -228,7 +228,7 @@
   librsvg_shared_lib = shared_library(
     rsvg_ver,
     rsvg_dummy,
-    link_whole: librsvg_c_lib,
+    link_with: librsvg_c_lib,
     link_args: link_args,
     link_depends: librsvg_ver,
     dependencies: library_dependencies,
@@ -342,7 +342,7 @@
       librsvg_gir = shared_library(
         'rsvg-gir',
         rsvg_dummy,
-        link_whole: librsvg_c_lib,
+        link_with: librsvg_c_lib,
         dependencies: library_dependencies,
         include_directories: [includeinc],
       )
