--- librsvg-2.40.3/rsvg-base.c
+++ librsvg-2.40.3/rsvg-base.c
@@ -2207,22 +2207,22 @@
     if (base == NULL)
         goto deny;
 
     dir = g_file_get_path (base);
     g_object_unref (base);
 
-    cdir = realpath (dir, NULL);
+    cdir = _fullpath (NULL, dir, _MAX_PATH);
     g_free (dir);
     if (cdir == NULL)
         goto deny;
 
     path = g_filename_from_uri (uri, NULL, NULL);
     if (path == NULL)
         goto deny;
 
-    cpath = realpath (path, NULL);
+    cpath = _fullpath (NULL, path, _MAX_PATH);
     g_free (path);
 
     if (cpath == NULL)
         goto deny;
 
     /* Now check that @cpath is below @cdir */
