--- librsvg-2.59.1/meson.build.orig	2024-09-27 23:55:37.000000000 +0200
+++ librsvg-2.59.1/meson.build	2024-10-15 00:45:09.387821900 +0200
@@ -257,9 +257,7 @@
   if is_msvc_style
     rustc_toolchain_cc = 'msvc'
   elif cc.get_id() == 'clang'
-    # https://github.com/msys2/MINGW-packages/issues/13388
-    is_msys2 = cc.get_define('__MINGW32__') == '1' and host_machine.cpu_family() != 'aarch64' and host_machine.system() in ['windows', 'cygwin']
-    rustc_toolchain_cc = is_msys2 ? 'gnu' : 'gnullvm'
+    rustc_toolchain_cc = 'gnullvm'
   else
     rustc_toolchain_cc = 'gnu'
   endif
