--- rsvg-base.c.old	2013-05-11 02:19:07.000000000 -0700
+++ rsvg-base.c	2013-10-31 15:22:09.451628741 -0700
@@ -54,6 +54,7 @@
 #include "rsvg-path.h"
 #include "rsvg-paint-server.h"
 #include "rsvg-xml.h"
+#include "realpath.c"
 
 /*
  * This is configurable at runtime
@@ -2191,7 +2192,7 @@
     g_object_unref (base);
 
     /* FIXME portability */
-    cdir = canonicalize_file_name (dir);
+    cdir = realpath(dir, NULL);
     g_free (dir);
     if (cdir == NULL)
         goto deny;
@@ -2201,7 +2202,7 @@
         goto deny;
 
     /* FIXME portability */
-    cpath = canonicalize_file_name (path);
+    cpath = realpath(path, NULL);
     g_free (path);
 
     if (cpath == NULL)
