From b669b211cd1a110a939989f560b8c4ad426eeb71 Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Sun, 3 Aug 2014 17:35:04 +0200
Subject: [PATCH 2/3] msys: fix introspection build

---
 clutter-gtk/Makefile.am | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/clutter-gtk/Makefile.am b/clutter-gtk/Makefile.am
index e9e4616..da41d0d 100644
--- a/clutter-gtk/Makefile.am
+++ b/clutter-gtk/Makefile.am
 GtkClutter-@CLUTTER_GTK_API_VERSION@.gir: libclutter-gtk-@CLUTTER_GTK_API_VERSION@.la Makefile
@@ -73,7 +74,9 @@ GtkClutter_@CLUTTER_GTK_API_VERSION_AM@_gir_SCANNERFLAGS = \
 	--identifier-prefix=GtkClutter \
 	--warn-all \
 	--pkg-export clutter-gtk-@CLUTTER_GTK_API_VERSION@ \
-	--c-include='clutter-gtk/clutter-gtk.h'
+	--c-include='clutter-gtk/clutter-gtk.h' \
+	`$(PKG_CONFIG) --cflags-only-I gtk+-3.0` \
+	`$(PKG_CONFIG) --cflags-only-I clutter-1.0`
 GtkClutter_@CLUTTER_GTK_API_VERSION_AM@_gir_NAMESPACE    = GtkClutter
 GtkClutter_@CLUTTER_GTK_API_VERSION_AM@_gir_VERSION      = 1.0
 GtkClutter_@CLUTTER_GTK_API_VERSION_AM@_gir_INCLUDES     = Gtk-3.0 Clutter-1.0
-- 
2.0.4

