From d76dc6b14c470f6d92cf6a599796496b97783d27 Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Sun, 3 Aug 2014 17:35:31 +0200
Subject: [PATCH 3/3] msys: update clutter-win32 header location

---
 clutter-gtk/gtk-clutter-actor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clutter-gtk/gtk-clutter-actor.c b/clutter-gtk/gtk-clutter-actor.c
index b5de571..e090a89 100644
--- a/clutter-gtk/gtk-clutter-actor.c
+++ b/clutter-gtk/gtk-clutter-actor.c
@@ -59,7 +59,7 @@
 #endif
 
 #ifdef CLUTTER_WINDOWING_WIN32
-#include <clutter/clutter-win32.h>
+#include <clutter/win32/clutter-win32.h>
 #endif
 #ifdef GDK_WINDOWING_WIN32
 #include <gdk/gdkwin32.h>
-- 
2.0.4

