--- mesonlsp-4.3.7/src/liblangserver/langserver.cpp
+++ mesonlsp-4.3.7/src/liblangserver/langserver.cpp
@@ -180,7 +180,7 @@
   pkgconf_client_set_sysroot_dir(&pkgClient, nullptr);
   pkgconf_client_init(&pkgClient,
                       (pkgconf_error_handler_func_t)pkgconfLogHandler, nullptr,
-                      personality);
+                      personality, nullptr, nullptr);
   pkgconf_client_set_trace_handler(
       &pkgClient, (pkgconf_error_handler_func_t)pkgconfLogHandler, nullptr);
   pkgconf_client_set_flags(&pkgClient, PKGCONF_PKG_PKGF_NONE);
