--- wt-4.4.0-orig/CMakeLists.txt	2020-08-03 12:46:02.000000000 +0300
+++ wt-4.4.0/CMakeLists.txt	2020-12-11 09:28:19.841828000 +0300
@@ -106,7 +106,7 @@
 SET(CMAKE_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake" CACHE STRING
   "Name for CMake package configuration directory within ${CMAKE_INSTALL_PREFIX}")
 
-IF(WIN32)
+IF(FALSE)
 
   SET(RUNDIR "c:/witty") # Does not apply to win32
 
@@ -119,7 +119,7 @@
     SET(BUILD_PARALLEL "/MP" CACHE STRING "MSVC option for parallel builds (/MP or /MPx)")
   ENDIF(MSVC)
 
-ELSE(WIN32)
+ELSE(FALSE)
 
   IF("${CMAKE_INSTALL_PREFIX}" STREQUAL "/usr")
     SET(CONFIG_PREFIX "")
@@ -144,7 +144,7 @@
     SET(USERLIB_PREFIX_DEFAULT "/usr")
   ENDIF()
 
-ENDIF(WIN32)
+ENDIF(FALSE)
 
 IF(DEFINED USERLIB_ROOT) # Deprecated <= 3.1.3
   SET(USERLIB_PREFIX ${USERLIB_ROOT} CACHE PATH
