diff --git a/src/Wt/WRasterImage-gm.C b/src/Wt/WRasterImage-gm.C
index 1885467c39..94915f938b 100644
--- a/src/Wt/WRasterImage-gm.C
+++ b/src/Wt/WRasterImage-gm.C
@@ -730,7 +730,7 @@ void WRasterImage::doDrawImage(const WRectF& rect, const WAbstractDataInfo* data
         imagePath = cpp17::filesystem::path(Wt::WApplication::instance()->docRoot()) / imagePath;
         LOG_WARN("drawImage failed on " << filePath
                   << " attempt to prefix docroot: " << imagePath.string());
-        strncpy(info.filename, imagePath.c_str(), 2048);
+        strncpy(info.filename, imagePath.string().c_str(), 2048);
         cImage = ReadImage(&info, &docrootCatchException);
 
         if (cImage == nullptr) {
