diff -u winico-ori/tkwinico.c winico-0.6/tkwinico.c
--- winico-ori/tkwinico.c	2005-09-27 16:02:54.000000000 +0200
+++ winico-0.6/tkwinico.c	2014-10-25 03:55:59.682882300 +0200
@@ -1658,10 +1658,10 @@
             result=SendMessage(h,WM_SETICON,iconsize,(LPARAM)hIcon);
         } else {
             if(iconsize==ICON_BIG){
-                result=SetClassLong(h,GCL_HICON,(LPARAM)hIcon);
+                result=SetClassLongPtr(h,GCLP_HICON,(LPARAM)hIcon);
             } else {
                 /*don't permit small icons,bcause they remove the big ones*/
-                result=GetClassLong(h,GCL_HICON);
+                result=GetClassLongPtr(h,GCLP_HICON);
             }
         }
         Tcl_SetObjResult(interp, Tcl_NewIntObj(result));
