--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -302,8 +302,8 @@
   # Provide pkg-config metadata.
   set(prefix "${CMAKE_INSTALL_PREFIX}")
   set(exec_prefix \${prefix})
-  set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
-  set(libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
+  set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+  set(libdir "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
   string(REPLACE ";" " " THREADDLLIBS "${THREADDLLIBS_LIST}")
   # PACKAGE_VERSION is defined above.
   configure_file(pkgconfig/atomic_ops.pc.in pkgconfig/atomic_ops.pc @ONLY)