--- a/rtree/finder.py
+++ b/rtree/finder.py
@@ -33,7 +33,10 @@
             arch = "64"
         else:
             arch = "32"
-        lib_name = f"spatialindex_c-{arch}.dll"
+        if "MSC" in sys.version:
+            lib_name = f"spatialindex_c-{arch}.dll"
+        else:
+            lib_name = "libspatialindex-7.dll"
 
         # add search paths for conda installs
         if (_sys_prefix / "conda-meta").exists() or "conda" in sys.version:
