--- hdf5-2.2.0/config/cmake/FindZLIBNG.cmake.orig	2026-07-29 06:06:33.000000000 +0800
+++ hdf5-2.2.0/config/cmake/FindZLIBNG.cmake	2026-08-04 02:14:37.315926600 +0800
@@ -87,6 +87,7 @@
 if (DEFINED ZLIBNG_ROOT AND NOT ZLIBNG_FIND_QUIETLY)
   message (VERBOSE "Looking for zlib-ng CMake configuration file at ${ZLIBNG_ROOT}")
 endif ()
+set (zlib-ng_USE_STATIC_LIBS ${ZLIBNG_USE_STATIC_LIBS})
 find_package (ZLIBNG QUIET CONFIG NAMES "zlib-ng")
 mark_as_advanced (ZLIBNG_DIR)
 
@@ -138,6 +139,9 @@
       if (NOT ZLIBNG_USE_STATIC_LIBS)
         set (ZLIBNG_LIBRARIES "zlib-ng::zlib")
       endif ()
+      if (ZLIBNG_USE_STATIC_LIBS AND NOT TARGET zlib-ng::zlibstatic)
+        add_library (zlib-ng::zlibstatic ALIAS zlib-ng::zlib)
+      endif ()
     endif ()
     if (TARGET zlib-ng::zlibstatic)
       set (ZLIBNG_static_FOUND TRUE)
