--- a/fortran/src/CMakeLists.txt
+++ b/fortran/src/CMakeLists.txt
@@ -406,7 +406,7 @@
 endif ()
 
 if (BUILD_SHARED_LIBS)
-  if(WIN32)
+  if(MSVC)
       set(DLLDEF ${HDF5_F90_SRC_BINARY_DIR}/hdf5_fortrandll.def)
   else()
       set(DLLDEF "")
--- a/hl/fortran/src/CMakeLists.txt
+++ b/hl/fortran/src/CMakeLists.txt
@@ -193,7 +193,7 @@
   add_dependencies(${HDF5_HL_F90_LIB_TARGET} H5HLgen)
 endif ()
 if (BUILD_SHARED_LIBS)
-  if(WIN32)
+  if(MSVC)
       set(DLLDEF ${HDF5_HL_F90_SRC_BINARY_DIR}/hdf5_hl_fortrandll.def)
   else()
       set(DLLDEF "")
