Link static libraries to static libaec (if available).

--- a/CMakeFilters.cmake
+++ b/CMakeFilters.cmake
@@ -225,7 +225,7 @@
       set (H5_SZIP_INCLUDE_DIR_GEN ${SZIP_INCLUDE_DIR})
       set (H5_SZIP_INCLUDE_DIRS ${H5_SZIP_INCLUDE_DIRS} ${SZIP_INCLUDE_DIR})
       if (LIBAEC_PACKAGE_NAME STREQUAL "libaec")
-        set (LINK_COMP_LIBS ${LINK_COMP_LIBS} libaec::sz libaec::aec)
+        set (LINK_LIBS ${LINK_LIBS} sz aec) # aec is needed only if built with autotools instead of CMake
       else ()
         set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
       endif ()
