Link static libraries to static libaec (if available).

--- a/CMakeFilters.cmake
+++ b/CMakeFilters.cmake
@@ -225,14 +225,30 @@
       set (H5_SZIP_INCLUDE_DIR_GEN ${SZIP_INCLUDE_DIR})
       set (H5_SZIP_INCLUDE_DIRS ${H5_SZIP_INCLUDE_DIRS} ${SZIP_INCLUDE_DIR})
       if (LIBAEC_PACKAGE_NAME STREQUAL "libaec")
+        if (TARGET libaec::sz-static AND TARGET libaec::aec-static)
+          set (LINK_COMP_STATIC_LIBS ${LINK_COMP_LIBS} libaec::sz-static libaec::aec-static)
+        else ()
+          set (LINK_COMP_STATIC_LIBS ${LINK_COMP_LIBS} libaec::sz-static libaec::aec-static)
+        endif ()
         set (LINK_COMP_LIBS ${LINK_COMP_LIBS} libaec::sz libaec::aec)
         # The targets do not set an OUTPUT_NAME on the target.
         get_filename_component (libname ${SZIP_LIBRARIES} NAME_WE)
         string (REGEX REPLACE "^lib" "" libname ${libname})
-        set_target_properties (libaec::sz PROPERTIES OUTPUT_NAME ${libname})
-        set_target_properties (libaec::aec PROPERTIES OUTPUT_NAME aec)
+        get_target_property(_aliased libaec::sz ALIASED_TARGET)
+        if (_aliased)
+          set_target_properties (${_aliased} PROPERTIES OUTPUT_NAME ${libname})
+        else ()
+          set_target_properties (libaec::sz PROPERTIES OUTPUT_NAME ${libname})
+        endif ()
+        get_target_property(_aliased libaec::aec ALIASED_TARGET)
+        if (_aliased)
+          set_target_properties (${_aliased} PROPERTIES OUTPUT_NAME aec)
+        else ()
+          set_target_properties (libaec::aec PROPERTIES OUTPUT_NAME aec)
+        endif ()
       else ()
         set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
+        set (LINK_COMP_STATIC_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
       endif ()
     endif ()
     message (VERBOSE "H5_SZIP_FOUND=${SZIP_FOUND} and LINK_COMP_LIBS=${LINK_COMP_LIBS}")
@@ -241,6 +257,7 @@
       EXTERNAL_SZIP_LIBRARY (${HDF5_ALLOW_EXTERNAL_SUPPORT} ${HDF5_ENABLE_SZIP_ENCODING})
       message (VERBOSE "Filter SZIP is built using library AEC")
       set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${H5_SZIP_STATIC_LIBRARY})
+      set (LINK_COMP_STATIC_LIBS ${LINK_COMP_LIBS} ${H5_SZIP_STATIC_LIBRARY})
     endif ()
   endif ()
   message (VERBOSE "LINK_COMP_LIBS=${LINK_COMP_LIBS}")
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1119,7 +1119,7 @@
 if (HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "GIT" OR HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "TGZ")
   if ((H5_ZLIB_FOUND AND ZLIB_USE_EXTERNAL) OR (H5_SZIP_FOUND AND SZIP_USE_EXTERNAL))
     if (BUILD_STATIC_LIBS)
-      add_dependencies (${HDF5_LIB_TARGET} ${LINK_COMP_LIBS})
+      add_dependencies (${HDF5_LIB_TARGET} ${LINK_COMP_STATIC_LIBS})
     endif ()
     if (BUILD_SHARED_LIBS)
       add_dependencies (${HDF5_LIBSH_TARGET} ${LINK_COMP_LIBS})
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1260,7 +1260,7 @@
   )
   TARGET_C_PROPERTIES (${HDF5_LIB_TARGET} STATIC)
   target_link_libraries (${HDF5_LIB_TARGET}
-      PRIVATE ${LINK_LIBS} ${LINK_COMP_LIBS}
+      PRIVATE ${LINK_LIBS} ${LINK_COMP_STATIC_LIBS}
       PUBLIC ${LINK_PUB_LIBS}
   )
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT ${HDF5_LIB_TARGET})
