--- a/config/ConfigureChecks.cmake.orig	2025-11-10 18:00:49.000000000 -0600
+++ b/config/ConfigureChecks.cmake	2026-03-03 09:57:07.943529600 -0600
@@ -642,7 +642,7 @@
       message (FATAL_ERROR "Found aws-c-s3 library but CMake target for library didn't exist")
     endif ()
 
-    list (APPEND LINK_LIBS AWS::aws-c-s3)
+    list (APPEND LINK_LIBS aws-c-s3 aws-c-auth aws-c-http aws-c-io aws-c-sdkutils aws-c-compression aws-c-cal aws-checksums aws-c-common shlwapi ws2_32 ncrypt crypt32 secur32)
 
     set (${HDF_PREFIX}_HAVE_ROS3_VFD 1)
 
