--- a/config/ConfigureChecks.cmake.orig	2025-11-10 18:00:49.000000000 -0600
+++ b/config/ConfigureChecks.cmake	2026-03-03 09:57:07.943529600 -0600
@@ -642,7 +642,7 @@
       message (FATAL_ERROR "Found aws-c-s3 library but CMake target for library didn't exist")
     endif ()
 
-    list (APPEND LINK_LIBS AWS::aws-c-s3)
+    list (APPEND LINK_LIBS AWS::aws-c-s3 crypt32 secur32)
 
     set (${HDF_PREFIX}_HAVE_ROS3_VFD 1)
 
--- src/src/CMakeLists.txt.orig	2026-03-06 10:10:30.077390500 -0600
+++ src/src/CMakeLists.txt	2026-03-06 10:11:49.515166200 -0600
@@ -1404,7 +1404,9 @@
 endforeach ()
 
 # Convert flag lists into strings for pkg-config files
+list (REVERSE pkg_config_lib_list)
 list (REMOVE_DUPLICATES pkg_config_lib_list)
+list (REVERSE pkg_config_lib_list)
 string (REPLACE ";" " " pkg_config_lib_list "${pkg_config_lib_list}")
 set (_PKG_CONFIG_LIBS_PRIVATE "${pkg_config_lib_list}")
 
