--- src/config/libh5cc.in.orig	2025-11-10 18:00:49.000000000 -0600
+++ src/config/libh5cc.in	2026-03-04 19:02:48.964844000 -0600
@@ -244,7 +244,7 @@
   escaped_libdir=$(echo "$libdir" | sed 's/\//\\\//g')
 
   # Replace flags of form '-lhdf5*' with path to matching static library
-  edited_pc_flags=$(echo "$pc_flags" | sed -E 's/-l(hdf5[a-z0-9_]*)/'"${escaped_libdir}\/"'lib\1.a/g')
+  edited_pc_flags=$(echo "$pc_flags" | sed -E 's/-l([[:alnum:]_-]+)/'"${escaped_libdir}\/"'lib\1.a/g')
   if test $? -ne 0; then
     echo "couldn't edit flags to compile against static HDF5 libraries" >&2
     exit $EXIT_FAILURE
