--- src/src/CMakeLists.txt.orig	2026-03-02 12:45:34.000000000 -0600
+++ src/src/CMakeLists.txt	2026-03-10 13:43:38.974598700 -0500
@@ -1292,6 +1292,8 @@
       PRIVATE ${LINK_LIBS} ${LINK_COMP_LIBS}
       PUBLIC ${LINK_PUB_LIBS}
   )
+  include(DelayLoad)
+  target_delay_load(${HDF5_LIBSH_TARGET} "aws-c|libsz|libz")
   set_global_variable (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_LIBSH_TARGET}")
   H5_SET_LIB_OPTIONS (${HDF5_LIBSH_TARGET} ${HDF5_LIB_NAME} SHARED "LIB")
   set_target_properties (${HDF5_LIBSH_TARGET} PROPERTIES FOLDER libraries)
