diff -Naur hdf5-1.12.0-orig/c++/src/CMakeLists.txt hdf5-1.12.0/c++/src/CMakeLists.txt
--- hdf5-1.12.0-orig/c++/src/CMakeLists.txt	2020-02-29 02:29:58.000000000 +0300
+++ hdf5-1.12.0/c++/src/CMakeLists.txt	2020-04-27 18:13:13.501802700 +0300
@@ -186,11 +186,11 @@
 
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf5.pc.in
-    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_CPP_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_CPP_LIB_CORENAME}.pc
     @ONLY
 )
 install (
-    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_CPP_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_CPP_LIB_CORENAME}.pc
     DESTINATION ${HDF5_INSTALL_LIB_DIR}/pkgconfig
     COMPONENT cpplibraries
 )
diff -Naur hdf5-1.12.0-orig/fortran/src/CMakeLists.txt hdf5-1.12.0/fortran/src/CMakeLists.txt
--- hdf5-1.12.0-orig/fortran/src/CMakeLists.txt	2020-02-29 02:29:58.000000000 +0300
+++ hdf5-1.12.0/fortran/src/CMakeLists.txt	2020-04-27 18:13:57.020714400 +0300
@@ -523,11 +523,11 @@
 
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf5.pc.in
-    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_F90_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_F90_LIB_CORENAME}.pc
     @ONLY
 )
 install (
-    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_F90_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_F90_LIB_CORENAME}.pc
     DESTINATION ${HDF5_INSTALL_LIB_DIR}/pkgconfig
     COMPONENT fortlibraries
 )
diff -Naur hdf5-1.12.0-orig/hl/c++/src/CMakeLists.txt hdf5-1.12.0/hl/c++/src/CMakeLists.txt
--- hdf5-1.12.0-orig/hl/c++/src/CMakeLists.txt	2020-02-29 02:29:58.000000000 +0300
+++ hdf5-1.12.0/hl/c++/src/CMakeLists.txt	2020-04-27 18:17:03.418274800 +0300
@@ -99,11 +99,11 @@
 
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf5.pc.in
-    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_CPP_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_CPP_LIB_CORENAME}.pc
     @ONLY
 )
 install (
-    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_CPP_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_CPP_LIB_CORENAME}.pc
     DESTINATION ${HDF5_INSTALL_LIB_DIR}/pkgconfig
     COMPONENT hlcpplibraries
 )
diff -Naur hdf5-1.12.0-orig/hl/src/CMakeLists.txt hdf5-1.12.0/hl/src/CMakeLists.txt
--- hdf5-1.12.0-orig/hl/src/CMakeLists.txt	2020-02-29 02:29:58.000000000 +0300
+++ hdf5-1.12.0/hl/src/CMakeLists.txt	2020-04-27 18:14:35.474650500 +0300
@@ -123,11 +123,11 @@
 
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf5.pc.in
-    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_LIB_CORENAME}.pc
     @ONLY
 )
 install (
-    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_HL_LIB_CORENAME}.pc
     DESTINATION ${HDF5_INSTALL_LIB_DIR}/pkgconfig
     COMPONENT hllibraries
 )
diff -Naur hdf5-1.12.0-orig/src/CMakeLists.txt hdf5-1.12.0/src/CMakeLists.txt
--- hdf5-1.12.0-orig/src/CMakeLists.txt	2020-02-29 02:29:58.000000000 +0300
+++ hdf5-1.12.0/src/CMakeLists.txt	2020-04-27 18:15:29.978638300 +0300
@@ -1262,11 +1262,11 @@
 
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf5.pc.in
-    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_LIB_CORENAME}.pc
     @ONLY
 )
 install (
-    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_LIB_CORENAME}-${HDF5_PACKAGE_VERSION}.pc
+    FILES ${HDF5_BINARY_DIR}/CMakeFiles/${HDF5_LIB_CORENAME}.pc
     DESTINATION ${HDF5_INSTALL_LIB_DIR}/pkgconfig
     COMPONENT libraries
 )
