--- a/config/cmake/HDFMacros.cmake	2023-01-13 04:34:01.443034400 -0500
+++ b/config/cmake/HDFMacros.cmake	2023-01-13 04:36:14.419475400 -0500
@@ -126,10 +126,12 @@
       OUTPUT_NAME_RELEASE        ${LIB_RELEASE_NAME}
       OUTPUT_NAME_MINSIZEREL     ${LIB_RELEASE_NAME}
       OUTPUT_NAME_RELWITHDEBINFO ${LIB_RELEASE_NAME}
+      RUNTIME_OUTPUT_NAME        ${LIB_RELEASE_NAME}-0
+      ARCHIVE_OUTPUT_NAME        ${LIB_RELEASE_NAME}
   )
 
   if (${libtype} MATCHES "STATIC")
-    if (WIN32)
+    if (MSVC)
       set_target_properties (${libtarget} PROPERTIES
           COMPILE_PDB_NAME_DEBUG          ${LIB_DEBUG_NAME}
           COMPILE_PDB_NAME_DEVELOPER      ${LIB_DEBUG_NAME}
