diff -Naur hdf5-1.8.19.org/config/cmake_ext_mod/ConfigureChecks.cmake hdf5-1.8.19/config/cmake_ext_mod/ConfigureChecks.cmake
--- hdf5-1.8.19.org/config/cmake_ext_mod/ConfigureChecks.cmake	2017-06-15 17:47:13.000000000 +0200
+++ hdf5-1.8.19/config/cmake_ext_mod/ConfigureChecks.cmake	2017-11-06 23:06:21.235526700 +0100
@@ -420,7 +420,7 @@
   if (NOT ${HDF_PREFIX}_SIZEOF_SSIZE_T)
     set (${HDF_PREFIX}_SIZEOF_SSIZE_T 0)
   endif ()
-  if (NOT WINDOWS)
+  if (NOT MSVC)
     HDF_CHECK_TYPE_SIZE (ptrdiff_t    ${HDF_PREFIX}_SIZEOF_PTRDIFF_T)
   endif ()
 endif ()
