--- hdf5-1.8.12/CMakeInstallation.cmake.orig	2013-11-21 19:14:04.000000000 +0400
+++ hdf5-1.8.12/CMakeInstallation.cmake	2014-04-30 12:23:14.248200000 +0400
@@ -144,7 +144,7 @@
   install (
       FILES
           ${HDF5_SOURCE_DIR}/COPYING
-      DESTINATION ${HDF5_INSTALL_DATA_DIR}
+      DESTINATION ${HDF5_INSTALL_DATA_DIR}/doc/hdf5
       COMPONENT hdfdocuments
   )
   if (EXISTS "${HDF5_SOURCE_DIR}/release_docs" AND IS_DIRECTORY "${HDF5_SOURCE_DIR}/release_docs")
@@ -188,7 +188,7 @@
     endif (HDF5_PACK_INSTALL_DOCS)
     install (
         FILES ${release_files}
-        DESTINATION ${HDF5_INSTALL_DATA_DIR}
+        DESTINATION ${HDF5_INSTALL_DATA_DIR}/doc/hdf5
         COMPONENT hdfdocuments
     )
   endif (EXISTS "${HDF5_SOURCE_DIR}/release_docs" AND IS_DIRECTORY "${HDF5_SOURCE_DIR}/release_docs")
