--- hdf5-1.8.19.orig/config/cmake_ext_mod/HDFMacros.cmake	2017-07-28 16:47:51.434642000 +0200
+++ hdf5-1.8.19/config/cmake_ext_mod/HDFMacros.cmake	2017-07-28 16:55:55.588824300 +0200
@@ -126,15 +126,6 @@
     endif ()
   endif ()
 
-  #----- Use MSVC Naming conventions for Shared Libraries
-  if (MINGW AND ${libtype} MATCHES "SHARED")
-    set_target_properties (${libtarget}
-        PROPERTIES
-        IMPORT_SUFFIX ".lib"
-        IMPORT_PREFIX ""
-        PREFIX ""
-    )
-  endif ()
 endmacro ()
 
 #-------------------------------------------------------------------------------
