--- hdf5-1.8.16/config/cmake_ext_mod/FindSZIP.cmake.orig	2016-01-19 08:26:45.389104700 +0300
+++ hdf5-1.8.16/config/cmake_ext_mod/FindSZIP.cmake	2016-01-19 08:27:01.478495600 +0300
@@ -90,13 +90,13 @@
     NO_DEFAULT_PATH
 )
 
-if (WIN32)
+if (MSVC)
     set (SZIP_SEARCH_DEBUG_NAMES "sz_d;libsz_d")
     set (SZIP_SEARCH_RELEASE_NAMES "sz;libsz;libszip")
-else (WIN32)
+else (MSVC)
     set (SZIP_SEARCH_DEBUG_NAMES "sz_d")
     set (SZIP_SEARCH_RELEASE_NAMES "sz;szip")
-endif (WIN32)
+endif (MSVC)
 
 # Look for the library.
 FIND_LIBRARY (SZIP_LIBRARY_DEBUG
--- hdf5-1.8.16/CMakeFilters.cmake.orig	2016-01-19 09:50:13.152040300 +0300
+++ hdf5-1.8.16/CMakeFilters.cmake	2016-01-19 09:51:23.955506900 +0300
@@ -31,11 +31,11 @@
       find_package (ZLIB NAMES ${ZLIB_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
       if (NOT ZLIB_FOUND)
         find_package (ZLIB) # Legacy find
-        if (ZLIB_FOUND)
-          set (LINK_LIBS ${LINK_LIBS} ${ZLIB_LIBRARIES})
-          set (LINK_SHARED_LIBS ${LINK_SHARED_LIBS} ${ZLIB_LIBRARIES})
-        endif (ZLIB_FOUND)
       endif (NOT ZLIB_FOUND)
+      if (ZLIB_FOUND)
+        set (LINK_LIBS ${LINK_LIBS} ${ZLIB_LIBRARIES})
+        set (LINK_SHARED_LIBS ${LINK_SHARED_LIBS} ${ZLIB_LIBRARIES})
+      endif (ZLIB_FOUND)
     endif (NOT ZLIB_USE_EXTERNAL)
     if (ZLIB_FOUND)
       set (H5_HAVE_FILTER_DEFLATE 1)
@@ -82,11 +82,11 @@
     find_package (SZIP NAMES ${SZIP_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
     if (NOT SZIP_FOUND)
       find_package (SZIP) # Legacy find
-      if (SZIP_FOUND)
-        set (LINK_LIBS ${LINK_LIBS} ${SZIP_LIBRARIES})
-        set (LINK_SHARED_LIBS ${LINK_SHARED_LIBS} ${SZIP_LIBRARIES})
-      endif (SZIP_FOUND)
     endif (NOT SZIP_FOUND)
+    if (SZIP_FOUND)
+      set (LINK_LIBS ${LINK_LIBS} ${SZIP_LIBRARIES})
+      set (LINK_SHARED_LIBS ${LINK_SHARED_LIBS} ${SZIP_LIBRARIES})
+    endif (SZIP_FOUND)
   endif (NOT SZIP_USE_EXTERNAL)
   if (SZIP_FOUND)
     set (H5_HAVE_FILTER_SZIP 1)
