diff -ur hdf5-1.8.12.orig/src/CMakeLists.txt hdf5-1.8.12/src/CMakeLists.txt
--- hdf5-1.8.12.orig/src/CMakeLists.txt	2013-11-22 22:01:31.874461256 +0100
+++ hdf5-1.8.12/src/CMakeLists.txt	2013-11-22 22:06:19.702407529 +0100
@@ -615,9 +615,9 @@
 #-----------------------------------------------------------------------------
 ADD_EXECUTABLE (H5detect ${HDF5_SRC_DIR}/H5detect.c)
 TARGET_C_PROPERTIES (H5detect " " " ")
-IF (MSVC)
-  TARGET_LINK_LIBRARIES (H5detect "ws2_32.lib")
-ENDIF (MSVC)
+IF (WIN32)
+  TARGET_LINK_LIBRARIES (H5detect ws2_32)
+ENDIF ()
 
 SET (CMD $<TARGET_FILE:H5detect>)
 ADD_CUSTOM_COMMAND (
@@ -629,9 +629,9 @@
 
 ADD_EXECUTABLE (H5make_libsettings ${HDF5_SRC_DIR}/H5make_libsettings.c)
 TARGET_C_PROPERTIES (H5make_libsettings " " " ")
-IF (MSVC)
-  TARGET_LINK_LIBRARIES (H5make_libsettings "ws2_32.lib")
-ENDIF (MSVC)
+IF (WIN32)
+  TARGET_LINK_LIBRARIES (H5make_libsettings ws2_32)
+ENDIF ()
 
 SET (CMD $<TARGET_FILE:H5make_libsettings>)
 ADD_CUSTOM_COMMAND (
