--- a/CMakeInstallation.cmake	2023-01-20 00:04:12.334144200 -0500
+++ b/CMakeInstallation.cmake	2023-01-20 00:03:49.497097700 -0500
@@ -126,6 +126,7 @@
 else ()
   set (BYTESEX little-endian)
 endif ()
+set (CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_INSTALL_PREFIX}/${HDF5_INSTALL_MODULE_DIR})
 configure_file (
     ${HDF_RESOURCES_DIR}/libhdf5.settings.cmake.in
     ${HDF5_SRC_BINARY_DIR}/libhdf5.settings ESCAPE_QUOTES @ONLY
