diff -ur hdf5-1.8.16.orig/config/cmake_ext_mod/HDFMacros.cmake hdf5-1.8.16/config/cmake_ext_mod/HDFMacros.cmake
--- hdf5-1.8.16.orig/config/cmake_ext_mod/HDFMacros.cmake	2013-11-22 22:35:38.454236491 +0100
+++ hdf5-1.8.16/config/cmake_ext_mod/HDFMacros.cmake	2013-11-22 22:37:27.717130034 +0100
@@ -108,15 +108,6 @@
       OUTPUT_NAME_RELWITHDEBINFO ${LIB_RELEASE_NAME}
   )
 
-  #----- Use MSVC Naming conventions for Shared Libraries
-  if (MINGW AND ${libtype} MATCHES "SHARED")
-    set_target_properties (${libtarget}
-        PROPERTIES
-        IMPORT_SUFFIX ".lib"
-        IMPORT_PREFIX ""
-        PREFIX ""
-    )
-  endif (MINGW AND ${libtype} MATCHES "SHARED")
 
 endmacro (HDF_SET_LIB_OPTIONS)
 
