--- hdf5-1.8.19.orig/src/CMakeLists.txt	2017-07-28 16:22:36.001558800 +0200
+++ hdf5-1.8.19/src/CMakeLists.txt	2017-07-28 16:23:54.234343200 +0200
@@ -613,8 +613,10 @@
 #-----------------------------------------------------------------------------
 add_executable (H5detect ${HDF5_SRC_DIR}/H5detect.c)
 TARGET_C_PROPERTIES (H5detect STATIC " " " ")
-if (MSVC OR MINGW)
+if (MSVC)
   target_link_libraries (H5detect "ws2_32.lib")
+elseif (MINGW)
+  target_link_libraries (H5detect "ws2_32")
 endif ()
 
 set (CMD $<TARGET_FILE:H5detect>)
@@ -627,8 +629,10 @@
 
 add_executable (H5make_libsettings ${HDF5_SRC_DIR}/H5make_libsettings.c)
 TARGET_C_PROPERTIES (H5make_libsettings STATIC " " " ")
-if (MSVC OR MINGW)
+if (MSVC)
   target_link_libraries (H5make_libsettings "ws2_32.lib")
+elseif (MINGW)
+  target_link_libraries (H5make_libsettings "ws2_32")
 endif ()
 
 set (CMD $<TARGET_FILE:H5make_libsettings>)
