--- hdf5-1.8.19.orig/config/cmake_ext_mod/HDFMacros.cmake	2017-07-28 16:41:41.840936100 +0200
+++ hdf5-1.8.19/config/cmake_ext_mod/HDFMacros.cmake	2017-07-28 16:45:12.960710400 +0200
@@ -88,7 +88,10 @@
 
 #-------------------------------------------------------------------------------
 macro (HDF_SET_LIB_OPTIONS libtarget libname libtype)
-  if (WIN32)
+  if (MINGW)
+    string(REGEX REPLACE "^lib([A-za-z]+)" "\\1" libname ${libname})
+  endif ()
+  if (MSVC)
     set (LIB_DEBUG_SUFFIX "_D")
   else ()
     set (LIB_DEBUG_SUFFIX "_debug")
@@ -97,7 +100,7 @@
       set (LIB_DEBUG_NAME "${libname}_debug")
     endif ()
   else ()
-    if (WIN32)
+    if (MSVC)
       set (LIB_RELEASE_NAME "lib${libname}")
       set (LIB_DEBUG_NAME "lib${libname}_D")
     else ()
@@ -116,6 +116,13 @@
          OUTPUT_NAME_RELWITHDEBINFO
                ${LIB_RELEASE_NAME}
   )
+ 
+  if (MINGW)
+    set_target_properties (${libtarget} PROPERTIES OUTPUT_NAME ${libname} ARCHIVE_OUTPUT_NAME ${libname})
+    if (${libtype} MATCHES "SHARED")
+      set_target_properties (${libtarget} PROPERTIES RUNTIME_OUTPUT_NAME ${libname}-0)
+    endif ()
+  endif ()
   if (${libtype} MATCHES "STATIC")
     if (WIN32)
       set_target_properties (${libtarget}
