--- hdf5-1.12.0/config/cmake_ext_mod/HDFMacros.cmake.orig	2020-04-26 10:08:44.715403300 +0300
+++ hdf5-1.12.0/config/cmake_ext_mod/HDFMacros.cmake	2020-04-26 10:34:09.889478600 +0300
@@ -125,6 +130,8 @@
       OUTPUT_NAME_RELEASE        ${LIB_RELEASE_NAME}
       OUTPUT_NAME_MINSIZEREL     ${LIB_RELEASE_NAME}
       OUTPUT_NAME_RELWITHDEBINFO ${LIB_RELEASE_NAME}
+      RUNTIME_OUTPUT_NAME        ${LIB_RELEASE_NAME}-0
+      ARCHIVE_OUTPUT_NAME        ${LIB_RELEASE_NAME}
   )
   #get_property (target_name TARGET ${libtarget} PROPERTY OUTPUT_NAME)
   #get_property (target_name_debug TARGET ${libtarget} PROPERTY OUTPUT_NAME_DEBUG)
@@ -132,7 +139,7 @@
   #message (STATUS "${target_name} : ${target_name_debug} : ${target_name_rwdi}")
 
   if (${libtype} MATCHES "STATIC")
-    if (WIN32)
+    if (MSVC)
       set_target_properties (${libtarget} PROPERTIES
           COMPILE_PDB_NAME_DEBUG          ${LIB_DEBUG_NAME}
           COMPILE_PDB_NAME_RELEASE        ${LIB_RELEASE_NAME}
