--- gtksourceview-3.20.0/configure.ac.orig	2014-08-22 00:42:28.532000000 +0400
+++ gtksourceview-3.20.0/configure.ac	2014-08-22 00:43:00.621200000 +0400
@@ -133,6 +133,12 @@
 	 AC_MSG_RESULT([$GLADE_CATALOG_DIR])
 	 AC_SUBST(GLADE_CATALOG_DIR)])
 
+case "$host" in
+  *-*-mingw*)
+    GLADE_CATALOG_DIR=`cygpath -u $GLADE_CATALOG_DIR`
+    ;;
+esac
+
 # i18n
 AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION([0.19.4])
