--- a/src/font.c
+++ b/src/font.c
@@ -24,7 +24,7 @@
  *	Jeffrey Stedfast <fejj@novell.com>
  */
 
-#ifdef WIN32
+#ifdef _MSC_VER
 #include "win32-private.h"
 #endif
 
@@ -1400,7 +1400,7 @@
 GdipPrivateAddMemoryFont(GpFontCollection *fontCollection, GDIPCONST void *memory, INT length)
 {
 	FcChar8	fontfile[256];
-#ifdef WIN32
+#ifdef _MSC_VER
 	FILE	*f;
 #else
 	int	f;
@@ -1411,7 +1411,7 @@
 	if (length <= 0)
 		return InvalidParameter;
 
-#ifdef WIN32
+#ifdef _MSC_VER
 	f = CreateTempFile (fontfile);
 	if (!f)
 		return FileNotFound;
