--- a/src/general.c
+++ b/src/general.c
@@ -32,7 +32,7 @@
 #include "font-private.h"
 #include "stringformat-private.h"
 #include "carbon-private.h"
-#ifdef WIN32
+#ifdef _MSC_VER
 #include "win32-private.h"
 #endif
 
@@ -75,7 +75,7 @@
 		   which we could use to avoid generating a temporary file. But meanwhile,
 		   we are stuck with this workaround. */
 		char namebuf[512];
-#ifdef WIN32
+#ifdef _MSC_VER
 		FILE *fi = CreateTempFile (namebuf);
 #else
 		strcpy ((char *) namebuf, "/tmp/ffXXXXXX");
@@ -86,7 +86,7 @@
 		if (fi) {
 			fprintf (fi, "<?xml version=\"1.0\"?>\n");
 			fprintf (fi, "<fontconfig>\n");
-#if defined(WIN32)
+#if defined(_MSC_VER)
 			fprintf (fi, "<dir>WINDOWSFONTDIR</dir>\n");
 #elif defined(__APPLE__)
 			fprintf (fi, "<dir>/System/Library/Fonts</dir>\n");
@@ -94,7 +94,7 @@
 			fprintf (fi, "<dir>~/.fonts</dir>\n");
 #endif
 
-#if defined(WIN32)
+#if defined(_MSC_VER)
 			fprintf (fi, "<cachedir>WINDOWSTEMPDIR_FONTCONFIG_CACHE</cachedir>\n");
 #else
 			fprintf (fi, "<cachedir>~/.fontconfig</cachedir>\n");
@@ -258,7 +258,7 @@
 		} else {
 			dpis = 96.0f;
 		}
-#elif WIN32
+#elif _MSC_VER
 		dpis = gdip_get_display_dpi_win32 ();
 #else
 		dpis = 96.0f;
