--- a/makefile
+++ b/makefile
@@ -24,7 +24,7 @@
 # By default, pick up the options from the PETSc configuration files
 ########################################################################################
 BLASLAPACK_TYPE  = F2CBLASLAPACK
-include ${PETSC_DIR}/lib/petsc/conf/base
+#include ${PETSC_DIR}/lib/petsc/conf/base
 
 ########################################################################################
 # compile the source files and create the blas and lapack libs
