diff -urN libdvdnav-4.2.0.orig/src/Makefile.am libdvdnav-4.2.0/src/Makefile.am
--- libdvdnav-4.2.0.orig/src/Makefile.am	2013-04-07 21:39:47 +0400
+++ libdvdnav-4.2.0/src/Makefile.am	2013-04-07 22:17:56 +0400
@@ -24,7 +24,7 @@
 	-export-symbols-regex "(^dvdnav.*|^nav.*|^ifo.*|^DVD.*)"
 #	-release $(DVDNAV_MAJOR).$(DVDNAV_MINOR).$(DVDNAV_SUB)
 
-libdvdnavmini_la_SOURCES = $(libdvdnav_la_SOURCES)
+libdvdnavmini_la_SOURCES = $(libdvdnav_la_SOURCES) dvdread_stump.c
 
 libdvdnavmini_la_LIBADD = $(THREAD_LIBS) \
 	$(top_builddir)/src/vm/libdvdvm.la
diff -urN libdvdnav-4.2.0.orig/src/dvdread_stump.c libdvdnav-4.2.0/src/dvdread_stump.c
--- libdvdnav-4.2.0.orig/src/dvdread_stump.c	1970-01-01 04:00:00 +0400
+++ libdvdnav-4.2.0/src/dvdread_stump.c	2013-04-07 22:40:24 +0400
@@ -0,0 +1,50 @@
+
+#include <inttypes.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <limits.h>
+#include <string.h>
+#include <sys/time.h>
+#include "dvdnav/dvdnav.h"
+
+void DVDCloseFile( dvd_file_t *dvd_file ) {}
+
+void navRead_DSI(dsi_t *dsi, unsigned char *buffer) {}
+
+void navRead_PCI(pci_t *pci, unsigned char *buffer) {}
+
+dvd_file_t *DVDOpenFile( dvd_reader_t *dvd, int titlenum,
+                         dvd_read_domain_t domain )
+{ return NULL; }
+
+ssize_t DVDReadBlocks( dvd_file_t *dvd_file, int offset,
+                       size_t block_count, unsigned char *data )
+{ return -1; }
+
+void ifoClose(ifo_handle_t *ifofile) {}
+
+
+ifo_handle_t *ifoOpenVTSI(dvd_reader_t *dvd, int title) { return NULL; }
+
+int ifoRead_VTS_PTT_SRPT(ifo_handle_t *ifofile) { return 0; }
+
+int ifoRead_PGCIT(ifo_handle_t *ifofile) { return 0; }
+
+int ifoRead_PGCI_UT(ifo_handle_t *ifofile) { return 0; }
+
+int ifoRead_VOBU_ADMAP(ifo_handle_t *ifofile) { return 0; }
+
+int ifoRead_TITLE_VOBU_ADMAP(ifo_handle_t *ifofile) { return 0; }
+
+ifo_handle_t *ifoOpenVMGI(dvd_reader_t *dvd) { return NULL; }
+
+int ifoRead_FP_PGC(ifo_handle_t *ifofile) { return 0; }
+
+int ifoRead_TT_SRPT(ifo_handle_t *ifofile) { return 0; }
+int ifoRead_PTL_MAIT(ifo_handle_t *ifofile) { return 0; }
+int ifoRead_VTS_ATRT(ifo_handle_t *ifofile) { return 0; }
+
+void DVDClose( dvd_reader_t *dvd ) { };
+dvd_reader_t *DVDOpen( const char *ppath ) { return NULL; }
+ifo_handle_t *ifoOpen(dvd_reader_t *dvd, int title) { return NULL; }
