--- a/openpgl/CMakeLists.txt
+++ b/openpgl/CMakeLists.txt
@@ -49,7 +49,7 @@
   target_compile_definitions(${PROJECT_NAME} PRIVATE OPENPGL_DEVICE_TYPE_CPU_16)
 endif()
 
-if(WIN32)
+if(MSVC)
   if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     set(OPENPGL_RELEASE_OPTIONS /Ox /Oi /fp:precise)
     #set(OPENPGL_RELEASE_OPTIONS ${OPENPGL_RELEASE_OPTIONS} -ftree-vectorize -mfpmath=sse -funsafe-math-optimizations -fno-rounding-math -fno-signaling-nans -fno-math-errno -fomit-frame-pointer )
