--- "gimp-2.10.24\\plug-ins\\script-fu\\tinyscheme\\scheme.c.orig"	2021-05-17 08:44:45.762000000 +0200
+++ "gimp-2.10.24\\plug-ins\\script-fu\\tinyscheme\\scheme.c"	2021-05-17 08:44:48.980561800 +0200
@@ -31,9 +31,6 @@
 #ifdef WIN32
 # include <io.h>
 #endif
-#ifdef WIN32
-#define snprintf _snprintf
-#endif
 #if USE_DL
 # include "dynload.h"
 #endif
--- "gimp-2.10.24\\plug-ins\\file-raw\\file-darktable.c.orig"	2021-05-17 08:44:26.785000000 +0200
+++ "gimp-2.10.24\\plug-ins\\file-raw\\file-darktable.c"	2021-05-17 08:44:31.141313000 +0200
@@ -401,7 +401,7 @@
 
   if (debug_prints)
     {
-      gchar **environ = g_get_environ ();
+      gchar **environ_ = g_get_environ ();
       gint    i;
 
       g_printf ("[%s] trying to call\n", __FILE__);
@@ -410,9 +410,9 @@
       g_printf ("\n");
 
       g_printf ("## Environment ##\n");
-      for (i = 0; environ[i]; i++)
-        g_printf ("- %s\n", environ[i]);
-      g_strfreev (environ) ;
+      for (i = 0; environ_[i]; i++)
+        g_printf ("- %s\n", environ_[i]);
+      g_strfreev (environ_) ;
     }
 
   if (g_spawn_sync (NULL,
