From 6d5d46985ed50035f1dc77d7669842215a6cb9b3 Mon Sep 17 00:00:00 2001
From: Yuta NAKAI <nak5124@live.jp>
Date: Sat, 11 Apr 2015 12:15:19 +0900
Subject: [PATCH 2/5] Makefile: Install windows DLL to BINDIR.

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 350c338..65d68cc 100644
--- a/Makefile
+++ b/Makefile
@@ -157,7 +157,11 @@ dcadec.pc: dcadec.pc.in
 
 install-lib: $(OUT_LIB) dcadec.pc
 	install -d -m 755 $(DESTDIR)$(LIBDIR) $(DESTDIR)$(LIBDIR)/pkgconfig $(DESTDIR)$(INCLUDEDIR)/libdcadec
+ifneq ($(and $(CONFIG_WINDOWS),$(CONFIG_SHARED)),)
+	install -m 755 $(OUT_LIB) $(DESTDIR)$(BINDIR)
+else
 	install -m 644 $(OUT_LIB) $(DESTDIR)$(LIBDIR)
+endif
 	install -m 644 $(addprefix $(SRC_DIR)/, $(INC_LIB)) $(DESTDIR)$(INCLUDEDIR)/libdcadec
 	install -m 644 dcadec.pc $(DESTDIR)$(LIBDIR)/pkgconfig
 
-- 
2.4.3

