From 65259407adf1eb5583284a93e2e4729bb8ae3678 Mon Sep 17 00:00:00 2001
From: Yuta NAKAI <nak5124@live.jp>
Date: Thu, 11 Jun 2015 20:49:45 +0900
Subject: [PATCH 5/5] Makefile: Add install-cut.

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 6a32ec3..e81f2fe 100644
--- a/Makefile
+++ b/Makefile
@@ -171,4 +171,8 @@ install-dec: $(OUT_DEC)
 	install -d -m 755 $(DESTDIR)$(BINDIR)
 	install -m 755 $(OUT_DEC) $(DESTDIR)$(BINDIR)
 
+install-cut: $(OUT_CUT)
+	install -d -m 755 $(DESTDIR)$(BINDIR)
+	install -m 755 $(OUT_CUT) $(DESTDIR)$(BINDIR)
+
 install: install-lib install-dec
-- 
2.4.3

