--- a/src/host/os_getversion.c
+++ b/src/host/os_getversion.c
@@ -55,7 +55,9 @@
 
 #if defined(PLATFORM_WINDOWS)
 
+#ifdef _MSC_VER
 #pragma comment(lib, "version.lib")
+#endif
 
 int getKernelVersion(struct OsVersionInfo* info)
 {
--- a/src/host/os_isdir.c
+++ b/src/host/os_isdir.c
@@ -9,7 +9,7 @@
 #include "premake.h"
 
 #ifdef _WIN32
-#include <Windows.h>
+#include <windows.h>
 #endif
 
 int os_isdir(lua_State* L)
@@ -17,7 +17,7 @@
 	struct stat buf;
 	const char* path = luaL_checkstring(L, 1);
 #ifdef _WIN32
-	int attr;
+	DWORD attr;
 
 	wchar_t wide_path[PATH_MAX];
 	if (MultiByteToWideChar(CP_UTF8, 0, path, -1, wide_path, PATH_MAX) == 0)
--- a/src/host/os_uuid.c
+++ b/src/host/os_uuid.c
@@ -7,7 +7,7 @@
 #include "premake.h"
 
 #if PLATFORM_WINDOWS
-#include <Objbase.h>
+#include <objbase.h>
 #endif
 
 
