--- a/Bootstrap.mak
+++ b/Bootstrap.mak
@@ -1,6 +1,6 @@
 MSDEV       = vs2012
-CONFIG      = release
-PLATFORM    = x86
+CONFIG     ?= release
+PLATFORM   ?= x86
 LUA_DIR     = contrib/lua/src
 LUASHIM_DIR = contrib/luashim
 
@@ -81,10 +81,10 @@
 	$(SILENT) if exist .\build rmdir /s /q .\build
 	$(SILENT) if exist .\obj rmdir /s /q .\obj
 	
-mingw-clean: windows-clean
+mingw-clean: nix-clean
 	
-mingw: mingw-clean
-	if not exist build\bootstrap (mkdir build\bootstrap)
+mingw:
+	mkdir -p build/bootstrap
 	$(CC) -o build/bootstrap/premake_bootstrap -DPREMAKE_NO_BUILTIN_SCRIPTS -I"$(LUA_DIR)" -I"$(LUASHIM_DIR)" $(SRC) -lole32 -lversion
 	./build/bootstrap/premake_bootstrap embed
 	./build/bootstrap/premake_bootstrap --os=windows --to=build/bootstrap --cc=mingw gmake2
