--- a/kde-modules/KDECMakeSettings.cmake
+++ b/kde-modules/KDECMakeSettings.cmake
@@ -222,6 +222,11 @@
    #   acts on all #include "ui_*.h" assuming *.ui exists
    set(CMAKE_AUTORCC ON)
 
+   # Enable autostaticplugins in MSYS2 qt5-static/cmake
+   # Since https://github.com/Alexpux/MINGW-packages/commit/e5517227778695a51bd56df07f1b05c43103ad3f
+   #   and https://github.com/Alexpux/MINGW-packages/commit/aa8f86ab267446002bcbb59e1455fbdf0d9f1cde
+   set(CMAKE_AUTOSTATICPLUGINS ON)
+
    # By default, create 'GUI' executables. This can be reverted on a per-target basis
    # using ECMMarkNonGuiExecutable
    # Since CMake 2.8.8
