diff --git a/Makefile b/Makefile
index 35741ac..bb56037 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,9 @@
 ifneq ($(wildcard .maint),)
-  include maint.mk
+  ifeq ($(OS),Windows_NT)
+    include maint_windows.mk
+  else
+    include maint.mk
+  endif
 endif
 
 ifeq ($(shell uname),Darwin)
diff --git a/maint_windows.mk b/maint_windows.mk
new file mode 100644
index 0000000..6088278
--- /dev/null
+++ b/maint_windows.mk
@@ -0,0 +1 @@
+CFLAGS=-std=gnu99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wbad-function-cast -Wcast-align -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Winline -Wdisabled-optimization -O2 -pipe
