--- nspr-4.39/pr/include/md/_win95.cfg.orig	2026-05-05 20:48:55.000000000 +0800
+++ nspr-4.39/pr/include/md/_win95.cfg	2026-07-09 23:21:54.697219600 +0800
@@ -67,7 +67,7 @@
 #define PR_BYTES_PER_WORD_LOG2	2
 #define PR_BYTES_PER_DWORD_LOG2	3
 
-#elif defined(_M_X64) || defined(_M_AMD64) || defined(_AMD64_)
+#elif defined(_M_X64) || defined(_M_AMD64) || defined(_AMD64_) || defined (_WIN64)
 
 #define IS_LITTLE_ENDIAN 1
 #undef  IS_BIG_ENDIAN
--- nspr-4.39/pr/include/md/_winnt.cfg.orig	2026-05-05 20:48:55.000000000 +0800
+++ nspr-4.39/pr/include/md/_winnt.cfg	2026-07-09 23:21:54.710200400 +0800
@@ -66,7 +66,7 @@
 #define PR_BYTES_PER_WORD_LOG2	2
 #define PR_BYTES_PER_DWORD_LOG2	2
 
-#elif defined(_M_X64) || defined(_M_AMD64) || defined(_AMD64_)
+#elif defined(_M_X64) || defined(_M_AMD64) || defined(_AMD64_) || defined (_WIN64)
 
 #define IS_LITTLE_ENDIAN 1
 #undef  IS_BIG_ENDIAN
--- nspr-4.39/pr/src/md/windows/ntgc.c.orig	2026-07-09 23:21:54.715175200 +0800
+++ nspr-4.39/pr/src/md/windows/ntgc.c	2026-07-09 23:24:05.830171700 +0800
@@ -10,7 +10,7 @@
 #include "primpl.h"
 
 PRWord* _MD_HomeGCRegisters(PRThread* t, int isCurrent, int* np) {
-#if defined(_X86_)
+#if defined(_X86_) && (!defined(_AMD64_))
   CONTEXT context;
   context.ContextFlags = CONTEXT_INTEGER;
 
