--- a/abinit.pc.in
+++ b/abinit.pc.in
@@ -1,8 +1,7 @@
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-includedir=@includedir@
-libdir=@libdir@
+exec_prefix=${prefix}/@exec_prefix@
+includedir=${prefix}/@includedir@
+libdir=${prefix}/@libdir@
 
 Name: abinit
 Description: Atomic-scale simulation software
--- a/cmake/generate_pkgconfig.cmake
+++ b/cmake/generate_pkgconfig.cmake
@@ -4,9 +4,9 @@
   # set variables as required by autotools
   set (VERSION ${PROJECT_VERSION})
   set (prefix ${CMAKE_INSTALL_PREFIX})
-  set (exec_prefix ${CMAKE_INSTALL_FULL_BINDIR})
-  set (libdir ${CMAKE_INSTALL_FULL_LIBDIR})
-  set (includedir ${CMAKE_INSTALL_FULL_INCLUDEDIR})
+  set (exec_prefix ${CMAKE_INSTALL_BINDIR})
+  set (libdir ${CMAKE_INSTALL_LIBDIR})
+  set (includedir ${CMAKE_INSTALL_INCLUDEDIR})
 
   # actually produce the pkg-config file
   configure_file(
