--- a/include/cppunit/plugin/TestPlugIn.h
+++ b/include/cppunit/plugin/TestPlugIn.h
@@ -146,7 +146,9 @@ typedef CppUnitTestPlugIn *(*TestPlugInSignature)();
 #define NOUSER
 #define NOKERNEL
 #define NOSOUND
+#ifndef NOMINMAX
 #define NOMINMAX
+#endif
 #define BLENDFUNCTION void    // for mingw & gcc
 #include <windows.h>
 #endif
--- a/src/cppunit/Win32DynamicLibraryManager.cpp
+++ b/src/cppunit/Win32DynamicLibraryManager.cpp
@@ -8,7 +8,9 @@
 #define NOUSER
 #define NOKERNEL
 #define NOSOUND
+#ifndef NOMINMAX
 #define NOMINMAX
+#endif
 #define BLENDFUNCTION void    // for mingw & gcc  
 #include <windows.h>
 
