diff -urN slepc-3.8.2.orig/include/slepcmath.h slepc-3.8.2/include/slepcmath.h
--- slepc-3.8.2.orig/include/slepcmath.h	2017-12-01 07:45:36.999998400 -0800
+++ slepc-3.8.2/include/slepcmath.h	2018-03-27 04:02:33.448686200 -0700
@@ -92,6 +92,7 @@
 PETSC_STATIC_INLINE PetscErrorCode SlepcSetFlushToZero(unsigned int *state)
 {
   PetscFunctionBegin;
+#if 0
 #if defined(PETSC_HAVE_XMMINTRIN_H)
 #if defined(_MM_FLUSH_ZERO_ON)
   *state = _MM_GET_FLUSH_ZERO_MODE();
@@ -100,6 +101,7 @@
   *state = 0;
 #endif
 #endif
+#endif
   PetscFunctionReturn(0);
 }
 
@@ -109,11 +111,13 @@
 PETSC_STATIC_INLINE PetscErrorCode SlepcResetFlushToZero(unsigned int *state)
 {
   PetscFunctionBegin;
+#if 0
 #if defined(PETSC_HAVE_XMMINTRIN_H)
 #if defined(_MM_FLUSH_ZERO_MASK)
   _MM_SET_FLUSH_ZERO_MODE(*state & _MM_FLUSH_ZERO_MASK);
 #endif
 #endif
+#endif
   PetscFunctionReturn(0);
 }
 
