From 4bcf004fc5dd31476ba83dcd99bee07071ac9e72 Mon Sep 17 00:00:00 2001
From: oltolm <oleg.tolmatcev@gmail.com>
Date: Thu, 2 Apr 2015 11:14:27 +0200
Subject: [PATCH] fix printf on mingw-w64

---
 qpdf/qpdf-ctest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qpdf/qpdf-ctest.c b/qpdf/qpdf-ctest.c
index 3c13836..b6efcf3 100644
--- a/qpdf/qpdf-ctest.c
+++ b/qpdf/qpdf-ctest.c
@@ -36,7 +36,7 @@ static FILE* safe_fopen(char const* filename, char const* mode)
 
 static void report_errors()
 {
-#ifdef _WIN32
+#if defined _WIN32 && !defined(__USE_MINGW_ANSI_STDIO)
 # define POS_FMT "  pos : %I64d\n"
 #else
 /* If your compiler doesn't support lld, change to ld and lose
-- 
2.3.4

