diff -urN qpdf-10.5.0-orig/configure.ac qpdf-10.5.0/configure.ac
--- qpdf-10.5.0-orig/configure.ac	2021-12-21 23:21:23.000000000 +0800
+++ qpdf-10.5.0/configure.ac	2022-01-16 00:01:04.124285500 +0800
@@ -247,7 +247,7 @@
     LDFLAGS="$oLDFLAGS"
     if test "$qpdf_USE_WMAIN" = "1"; then
       AC_MSG_RESULT(yes)
-      WINDOWS_WMAIN_COMPILE="-DWINDOWS_WMAIN $nLDFLAGS"
+      WINDOWS_WMAIN_COMPILE="-DWINDOWS_WMAIN"
       WINDOWS_WMAIN_LINK="$nLDFLAGS"
     else
       AC_MSG_RESULT(no)
@@ -284,7 +284,7 @@
 
 if test "x$qpdf_OS_SECURE_RANDOM" = "x1"; then
   OLIBS=$LIBS
-  LIBS="$LIBS Advapi32.lib"
+  LIBS="$LIBS -ladvapi32"
   AC_MSG_CHECKING(for Advapi32 library)
   AC_LINK_IFELSE([AC_LANG_PROGRAM(
     [[#pragma comment(lib, "crypt32.lib")
@@ -294,7 +294,7 @@
      [CryptAcquireContext(&cp, NULL, NULL, PROV_RSA_FULL, 0);]
      )],
      [AC_MSG_RESULT(yes)
-      LIBS="$OLIBS -lAdvapi32"],
+      LIBS=$OLIBS],
      [AC_MSG_RESULT(no)
       LIBS=$OLIBS])
 fi
@@ -926,9 +926,7 @@
    # Don't actually check for the presence of this -- we document that
    # the user can run this and then edit autoconf.mk if they have too
    # much trouble getting it to work with a different compiler.
-   CPPFLAGS="$CPPFLAGS -Iexternal-libs/include"
-   LDFLAGS="$LDFLAGS -Lexternal-libs/lib-$BUILDRULES$WINDOWS_WORDSIZE"
-   LIBS="$LIBS -lz -ljpeg -lssl -lcrypto -lmsvcrt -lws2_32 -lshell32 -ladvapi32 -lgdi32 -luser32 -lcrypt32"
+   LIBS="$LIBS -lz -ljpeg -lws2_32 -lshell32 -ladvapi32 -lgdi32 -luser32 -lcrypt32"
 fi
 
 AC_OUTPUT
