diff -Naur qpdf-7.1.0-orig/make/installwin.mk qpdf-7.1.0/make/installwin.mk
--- qpdf-7.1.0-orig/make/installwin.mk	2018-01-15 04:09:20.000000000 +0300
+++ qpdf-7.1.0/make/installwin.mk	2018-01-15 09:37:07.877351400 +0300
@@ -1,21 +1,20 @@
-DEST=$(INSTALL_DIR)/$(PACKAGE_TARNAME)-$(PACKAGE_VERSION)
+DEST=$(DESTDIR)/$(prefix)
 installwin: all
-	$(RM) -r $(INSTALL_DIR)
-	mkdir $(INSTALL_DIR)
 	mkdir $(DEST)
 	mkdir $(DEST)/bin
 	mkdir $(DEST)/lib
 	mkdir $(DEST)/include
 	mkdir $(DEST)/include/qpdf
-	mkdir $(DEST)/doc
+	mkdir -p $(DEST)/share/doc/qpdf
+	mkdir -p $(DEST)/share/licenses/qpdf
 	cp libqpdf/$(OUTPUT_DIR)/$(STATIC_LIB_NAME) $(DEST)/lib
 	cp libqpdf/$(OUTPUT_DIR)/qpdf*.dll $(DEST)/bin
-	perl copy_dlls libqpdf/$(OUTPUT_DIR)/qpdf*.dll $(DEST)/bin $(OBJDUMP) $(WINDOWS_WORDSIZE)
 	cp qpdf/$(OUTPUT_DIR)/qpdf.exe $(DEST)/bin
 	cp zlib-flate/$(OUTPUT_DIR)/zlib-flate.exe $(DEST)/bin
 	cp qpdf/fix-qdf $(DEST)/bin
 	cp include/qpdf/*.h $(DEST)/include/qpdf
 	cp include/qpdf/*.hh $(DEST)/include/qpdf
-	cp doc/stylesheet.css $(DEST)/doc
-	cp doc/qpdf-manual.html $(DEST)/doc
-	cp doc/qpdf-manual.pdf $(DEST)/doc
+	cp doc/stylesheet.css $(DEST)/share/doc/qpdf
+	cp doc/qpdf-manual.html $(DEST)/share/doc/qpdf
+	cp doc/qpdf-manual.pdf $(DEST)/share/doc/qpdf
+	cp Artistic-2.0 $(DEST)/share/licenses/qpdf
