--- a/libqpdf/CMakeLists.txt
+++ b/libqpdf/CMakeLists.txt
@@ -468,13 +467,6 @@
     # qpdf.dll. qpdf has always done this, and it gives us some
     # protection against binary incompatible DLLs being installed.
     set(SHARED_OUT qpdf${qpdf_SOVERSION})  # Put API version number in DLL
-    if(MINGW)
-      # Reference: Platform/Windows-GNU.cmake in the cmake installation
-      set(CMAKE_SHARED_LIBRARY_PREFIX "")    # libqpdf$v.dll -> qpdf$v.dll
-      set(CMAKE_IMPORT_LIBRARY_SUFFIX ".a")  # libqpdf.dll.a -> libqpdf.a
-      # Ensure the DLLs are striped in Release mode.
-      set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -s")
-    endif()
     if(MSVC)
       # Avoid linker warning from mixing libraries built with /MT and /MD.
       set_target_properties(${SHARED_LIB}
