--- a/source/utils/CMakeLists.txt
+++ b/source/utils/CMakeLists.txt
@@ -11,6 +11,9 @@
 
 # BPLS
 add_executable(bpls ./bpls/bpls.cpp)
 target_link_libraries(bpls adios2_core adios2sys adios2::thirdparty::pugixml)
+if(WIN32)
+  target_link_libraries(bpls shlwapi)
+endif()
 target_include_directories(bpls PRIVATE ${PROJECT_BINARY_DIR})
 set_property(TARGET bpls PROPERTY OUTPUT_NAME bpls${ADIOS2_EXECUTABLE_SUFFIX})
 install(TARGETS bpls EXPORT adios2
