--- a/source/utils/bpls/bpls.cpp
+++ b/source/utils/bpls/bpls.cpp
@@ -30,7 +30,7 @@
 
 #include <errno.h>
 
-#if defined(__GNUC__) && !(defined(__ICC) || defined(__INTEL_COMPILER))
+#if defined(__GNUC__) && !(defined(__ICC) || defined(__INTEL_COMPILER) || defined(__clang__))
 #if (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__) < 40900
 /* pre GCC 4.9 cannot handle the C++ regex implementation. Will use C-lib
  * regex"*/
