--- a/uvcview.h
+++ b/uvcview.h
@@ -34,10 +34,10 @@ Revision History:
 #include <usbioctl.h>
 #include <usbiodef.h>
 #include <intsafe.h>
-#include <strsafe.h>
 #include <specstrings.h>
 #include <usb.h>
 #include <usbuser.h>
+#include <usbspec.h>
 #include <basetyps.h>
 #include <wtypes.h>
 #include <objbase.h>
@@ -50,6 +50,8 @@ Revision History:
 #include <winioctl.h>
 #include <devpkey.h>
 #include <math.h>
+#include <strsafe.h>
+#include <sal.h>
 
 // This is mostly a private USB Audio descriptor header
 #include "usbdesc.h"
