--- gstreamer-1.8.3/libs/gst/check/libcheck/libcompat.h.orig	2016-09-24 21:41:09.835856800 +0300
+++ gstreamer-1.8.3/libs/gst/check/libcheck/libcompat.h	2016-09-24 21:41:14.077864000 +0300
@@ -101,7 +101,7 @@
 #endif /* !HAVE_DECL_STRDUP && HAVE__STRDUP */
 
 #if !HAVE_DECL_STRSIGNAL
-CK_DLL_EXP char *strsignal (int sig);
+CK_DLL_EXP const char *strsignal (int sig);
 #endif /* !HAVE_DECL_STRSIGNAL */
 
 /*
