--- common/m4/gst-x11.m4.orig	2012-03-27 11:24:15.000000000 +0400
+++ common/m4/gst-x11.m4	2014-07-10 20:46:24.623600000 +0400
@@ -9,7 +9,9 @@
 
   dnl now try to find the HEADER
   HAVE_X="no"
-  AC_CHECK_HEADER(X11/Xlib.h, HAVE_X="yes", HAVE_X="no")
+  HAVE_X="yes"
+  AC_CHECK_HEADER([X11/Xlib.h], [], [HAVE_X="no"], [AC_INCLUDES_DEFAULT])
+  AC_CHECK_HEADER([X11/XKBlib.h], [], [HAVE_X="no"], [AC_INCLUDES_DEFAULT])
 
   if test "x$HAVE_X" = "xno"
   then
