--- ./src/build_command_runner.vala.orig	2015-01-12 15:16:12.882554900 +0100
+++ ./src/build_command_runner.vala	2015-01-12 15:18:14.007981800 +0100
@@ -77,7 +77,7 @@
     {
         if (_child_pid != null)
         {
-            Posix.kill (_child_pid, Posix.SIGTERM);
+            Windows.TerminateProcess (_child_pid, 0);
             _child_pid = null;
         }
     }
--- ./src/Makefile.am.orig	2015-01-12 15:30:54.450523100 +0100
+++ ./src/Makefile.am	2015-01-12 15:31:07.342497200 +0100
@@ -69,7 +69,8 @@
 	$(vala_files)			\
 	$(vapidir)/config.vapi		\
 	$(vapidir)/gedit.vapi		\
-	$(vapidir)/menu_tool_action.vapi
+	$(vapidir)/menu_tool_action.vapi \
+	$(vapidir)/windows.vapi
 
 latexila_CPPFLAGS =		\
 	-I$(top_srcdir)		\
--- ./vapi/windows.vapi.orig	2015-01-12 15:42:31.000000000 +0100
+++ ./vapi/windows.vapi	2015-01-12 15:42:23.883452400 +0100
@@ -0,0 +1,5 @@
+[CCode (lower_case_cprefix="", cheader_filename="windows.h")]
+namespace Windows
+{
+	public bool TerminateProcess (GLib.Pid hProcess, uint uExitCode);
+}
