--- ./configure.ac.orig	2015-01-12 21:12:06.744867800 +0100
+++ ./configure.ac	2015-01-12 21:14:35.622919500 +0100
@@ -60,11 +60,13 @@
      echo "[$]_lcl_receval")`])
 
 adl_RECURSIVE_EVAL(["${datadir}/latexila"], [DATA_DIR])
+DATA_DIR=`cygpath -m $DATA_DIR`
 AC_DEFINE_UNQUOTED([DATA_DIR],
                    ["${DATA_DIR}"],
 		   [latexila data directory])
 
 adl_RECURSIVE_EVAL(["${datadir}/icons/hicolor"], [ICONS_DIR])
+ICONS_DIR=`cygpath -m $ICONS_DIR`
 AC_DEFINE_UNQUOTED([ICONS_DIR],
                    ["${ICONS_DIR}"],
 		   [icons directory])
@@ -130,6 +132,7 @@
                    [The domain to use with gettext])
 
 adl_RECURSIVE_EVAL(["${localedir}"], [LOCALE_DIR])
+LOCALE_DIR=`cygpath -m $LOCALE_DIR`
 AC_DEFINE_UNQUOTED([LOCALE_DIR],
                    ["${LOCALE_DIR}"],
 		   [locale directory])
