--- drmingw/CMakeLists.txt.orig	2015-06-21 14:26:32.215200000 +0300
+++ drmingw/CMakeLists.txt	2015-06-21 14:27:19.342800000 +0300
@@ -120,7 +120,7 @@
     FILES
         LICENSE.txt
         README.md
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
 )
 
 # cpack mistakenly detects Mingw-w64 as win32
--- drmingw/sample/CMakeLists.txt.orig	2015-06-21 14:27:55.165000000 +0300
+++ drmingw/sample/CMakeLists.txt	2015-06-21 14:28:26.833000000 +0300
@@ -6,5 +6,5 @@
 add_dependencies (sample exchndl)
 target_link_libraries (sample $<TARGET_FILE:exchndl>)
 
-install (FILES sample.cpp DESTINATION sample)
-install (FILES sample.mak DESTINATION sample RENAME Makefile)
+install (FILES sample.cpp DESTINATION share/drmingw/sample)
+install (FILES sample.mak DESTINATION share/drmingw/sample RENAME Makefile)
--- drmingw/thirdparty/dwarf/CMakeLists.txt.orig	2015-06-21 14:33:35.812200000 +0300
+++ drmingw/thirdparty/dwarf/CMakeLists.txt	2015-06-21 14:33:45.733800000 +0300
@@ -38,6 +38,6 @@
 
 install (
     FILES LIBDWARFCOPYRIGHT
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
     RENAME LICENSE-libdwarf.txt
 )
