--- gtksourceview-3.11.91/gtksourceview/Makefile.am.orig	2014-03-17 20:36:26.000000000 +0000
+++ gtksourceview-3.11.91/gtksourceview/Makefile.am	2014-03-18 20:37:32.781966700 +0000
@@ -166,7 +166,7 @@
 	$(AM_V_GEN) echo "#include \"gtksourceview-marshal.h\"" > $@ && \
 	$(GLIB_GENMARSHAL) $< --body --prefix=_gtksourceview_marshal >> $@
 
-gtksourceview-resources.c: gtksourceview.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --generate-dependencies $(srcdir)/gtksourceview.gresource.xml)
+gtksourceview-resources.c: gtksourceview.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/gtksourceview.gresource.xml)
 	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) --generate-source $(srcdir)/gtksourceview.gresource.xml
 
 gtksourceview-typebuiltins.c: stamp-gtksourceview-typebuiltins.c
