--- gtksourceview-3.10.2/gtksourceview/Makefile.am.orig	2013-07-28 21:11:04.599868800 +0000
+++ gtksourceview-3.10.2/gtksourceview/Makefile.am	2013-07-28 21:18:26.664003800 +0000
@@ -225,6 +225,7 @@
 if HAVE_INTROSPECTION
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS = GtkSource-3.0.gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 GtkSource-3.0.gir: libgtksourceview-3.0.la $(BUILT_SOURCES)
 INTROSPECTION_SCANNER_ARGS =			\
