--- a/sipbuild/installable.py
+++ b/sipbuild/installable.py
@@ -22,6 +22,7 @@
 
 
 import os
+import sys
 from shutil import copy2, copytree
 
 
@@ -49,6 +49,9 @@
                 target_dir = self.target_subdir
             else:
                 target_dir = os.path.join(target_dir, self.target_subdir)
+
+        if sys.platform == 'win32' and "MSYSTEM" in os.environ:
+            target_dir = os.popen(' '.join(['cygpath', '--unix', target_dir])).readline().strip()
 
         return target_dir
 
--- a/sipbuild/distinfo/distinfo.py
+++ b/sipbuild/distinfo/distinfo.py
@@ -79,6 +79,8 @@
         generator = os.path.basename(sys.argv[0])
 
     # The prefix directory corresponds to DESTDIR or INSTALL_ROOT.
+    if sys.platform == 'win32' and "MSYSTEM" in os.environ:
+        distinfo_dir = os.popen(' '.join(['cygpath', '--unix', distinfo_dir])).readline().strip()
     real_distinfo_dir = prefix_dir + distinfo_dir
 
     # Make sure we have an empty dist-info directory.  Handle exceptions as the
