--- a/libproxy/url.cpp	2014-03-28 09:36:01 +0400
+++ b/libproxy/url.cpp	2014-04-03 13:23:37 +0400
@@ -24,7 +24,6 @@
 #include <io.h>
 #define open _open
 #define O_RDONLY _O_RDONLY
-#define close _close
 #endif
 #include <fcntl.h> // For ::open()
 #include <cstring> // For memcpy()
@@ -36,7 +35,7 @@
 
 #ifdef WIN32
 #include <io.h>
-#define close _close
+#define close closesocket
 #define read _read
 #define SHUT_RDWR SD_BOTH
 #endif
