--- libproxy-0.4.15/cmake/FindPython3Interp.cmake.orig	2018-07-16 13:40:30.196949100 +0300
+++ libproxy-0.4.15/cmake/FindPython3Interp.cmake	2018-07-16 13:40:35.656958700 +0300
@@ -39,7 +39,7 @@
 
 unset(_Python3_NAMES)
 
-set(_Python3_VERSIONS 3.6 3.5 3.4 3.3 3.2 3.1 3.0)
+set(_Python3_VERSIONS 3.7 3.6 3.5 3.4 3.3 3.2 3.1 3.0)
 
 if(Python3Interp_FIND_VERSION)
     if(Python3Interp_FIND_VERSION_COUNT GREATER 1)
