diff -aur 002/startup/Makefile.in 003/startup/Makefile.in
--- 002/startup/Makefile.in	2012-02-07 12:36:47.000000000 -0200
+++ 003/startup/Makefile.in	2015-01-10 17:51:06.418945300 -0200
@@ -163,7 +163,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-startupdir = $(datadir)/startup
+startupdir = $(datadir)
 startup_DATA = startup.mk config.mk
 DIST_SUBDIRS = unix winnt
 SUBDIRS = @OS_TYPE@ 
diff -aur 002/startup/winnt/Makefile.in 003/startup/winnt/Makefile.in
--- 002/startup/winnt/Makefile.in	2012-02-07 12:36:46.000000000 -0200
+++ 003/startup/winnt/Makefile.in	2015-01-10 18:12:18.143554600 -0200
@@ -162,7 +162,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-startupdir = $(datadir)/startup/winnt
+startupdir = $(datadir)/winnt
 startup_DATA = macros.mk recipes.mk
 DIST_SUBDIRS = mingw msvc6
 SUBDIRS = @OS_VERSION@
diff -aur 002/startup/winnt/mingw/Makefile.in 003/startup/winnt/mingw/Makefile.in
--- 002/startup/winnt/mingw/Makefile.in	2012-02-07 12:36:46.000000000 -0200
+++ 003/startup/winnt/mingw/Makefile.in	2015-01-10 18:12:24.551757800 -0200
@@ -154,7 +154,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-startupdir = $(datadir)/startup/winnt/mingw
+startupdir = $(datadir)/winnt/mingw
 startup_DATA = macros.mk 
 all: all-am
 
diff -aur 002/winnt/startup.h 003/winnt/startup.h
--- 002/winnt/startup.h	2012-02-07 12:36:53.000000000 -0200
+++ 003/winnt/startup.h	2015-01-10 18:01:16.400390600 -0200
@@ -24,6 +24,6 @@
 --      Use cvs log to obtain detailed change logs.
 */
 
-"DMAKEROOT *= $(ABSMAKECMD:d)startup",
+"DMAKEROOT *= $(ABSMAKECMD:d)..\\etc\\dmake",
 "MAKESTARTUP := $(DMAKEROOT)\\startup.mk",
 
