diff -urN dolphin-16.04.3.orig/src/kitemviews/private/kdirectorycontentscounterworker.cpp dolphin-16.04.3/src/kitemviews/private/kdirectorycontentscounterworker.cpp
--- dolphin-16.04.3.orig/src/kitemviews/private/kdirectorycontentscounterworker.cpp	2016-08-04 23:33:45.015298900 +0200
+++ dolphin-16.04.3/src/kitemviews/private/kdirectorycontentscounterworker.cpp	2016-08-04 23:48:09.143457600 +0200
@@ -18,10 +18,11 @@
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA            *
  ***************************************************************************/
 
+#include <QtGlobal>
 #include "kdirectorycontentscounterworker.h"
 
 // Required includes for subItemsCount():
-#ifdef Q_WS_WIN
+#ifdef Q_OS_WIN
     #include <QDir>
 #else
     #include <dirent.h>
@@ -39,7 +40,7 @@
     const bool countHiddenFiles = options & CountHiddenFiles;
     const bool countDirectoriesOnly = options & CountDirectoriesOnly;
 
-#ifdef Q_WS_WIN
+#ifdef Q_OS_WIN
     QDir dir(path);
     QDir::Filters filters = QDir::NoDotAndDotDot | QDir::System;
     if (countHiddenFiles) {
