--- a/setup.py
+++ b/setup.py
@@ -20,9 +20,9 @@
     'click >= 7.0',
     # Temporary to suppress paramiko Blowfish warning which breaks CI.
     # Pinning cryptography should not be needed after paramiko 2.11.0.
-    'cryptography == 36.0.2',
+    'cryptography >= 41.0.4',
     # 'Pygments>=1.6,<=2.11.1',
-    'Pygments>=1.6',
+    'pygments>=1.6',
     'prompt_toolkit>=3.0.6,<4.0.0',
     'PyMySQL >= 0.9.2',
     'sqlparse>=0.3.0,<0.5.0',
