diff -urN fribidi-0.19.7.orig/lib/Makefile.am fribidi-0.19.7/lib/Makefile.am
--- fribidi-0.19.7.orig/lib/Makefile.am	2013-01-02 07:35:46.000000000 +1100
+++ fribidi-0.19.7/lib/Makefile.am	2015-08-06 16:38:22.122583500 +1000
@@ -8,11 +8,7 @@
 libfribidi_la_LIBADD = $(MISC_LIBS)
 libfribidi_la_DEPENDENCIES =
 
-if OS_WIN32
-libfribidi_la_LDFLAGS += -export-symbols $(srcdir)/fribidi.def
-else
 libfribidi_la_LDFLAGS += -export-symbols-regex "^fribidi_.*"
-endif # OS_WIN32
 
 if FRIBIDI_CHARSETS
 
