--- gsm-1.0-pl13/inc/config.h.orig	2006-04-26 23:14:26 +0400
+++ gsm-1.0-pl13/inc/config.h	2012-04-15 01:34:23 +0400
@@ -18,10 +18,10 @@
 #define	HAS_ERRNO_DECL	1		/* errno.h declares errno	*/
 
 #define	HAS_FSTAT 	1		/* fstat syscall		*/
-#define	HAS_FCHMOD 	1		/* fchmod syscall		*/
+/*efine	HAS_FCHMOD 	1		/* fchmod syscall		*/
 #define	HAS_CHMOD 	1		/* chmod syscall		*/
-#define	HAS_FCHOWN 	1		/* fchown syscall		*/
-#define	HAS_CHOWN 	1		/* chown syscall		*/
+/*efine	HAS_FCHOWN 	1		/* fchown syscall		*/
+/*efine	HAS_CHOWN 	1		/* chown syscall		*/
 /*efine	HAS__FSETMODE 	1		/* _fsetmode -- set file mode	*/
 
 #define	HAS_STRING_H 	1		/* /usr/include/string.h 	*/
