--- whisper.cpp-1.5.0/CMakeLists.txt.orig	2023-11-16 19:00:45.515603300 +0100
+++ whisper.cpp-1.5.0/CMakeLists.txt	2023-11-16 19:00:56.847646600 +0100
@@ -173,7 +173,7 @@
 endif()
 
 if (WHISPER_BLAS)
-    if (WIN32)
+    if (MSVC)
         if(DEFINED ENV{OPENBLAS_PATH})
             set(BLAS_LIBRARIES $ENV{OPENBLAS_PATH}/lib/libopenblas.dll.a)
             message(STATUS "Libraries ${BLAS_LIBRARIES}")
