--- whisper.cpp-1.5.5/models/download-ggml-model.sh.orig	2024-04-16 13:08:31.000000000 +0200
+++ whisper.cpp-1.5.5/models/download-ggml-model.sh	2024-04-19 20:56:05.804477900 +0200
@@ -22,7 +22,9 @@
     fi
 }
 
-models_path="${2:-$(get_script_path)}"
+default_path="$(pwd)/models"
+models_path="${2:-$default_path}"
+mkdir -p "$models_path"
 
 # Whisper models
 models="tiny
@@ -118,5 +120,5 @@
 
 printf "Done! Model '%s' saved in '%s/ggml-%s.bin'\n" "$model" "$models_path" "$model"
 printf "You can now use it like this:\n\n"
-printf "  $ ./main -m %s/ggml-%s.bin -f samples/jfk.wav\n" "$models_path" "$model"
+printf "  $ whisper.cpp -m %s/ggml-%s.bin -f samples/jfk.wav\n" "$models_path" "$model"
 printf "\n"
