--- whisper.cpp-1.5.3/models/download-ggml-model.sh.orig	2024-01-05 07:02:42.499063100 +0100
+++ whisper.cpp-1.5.3/models/download-ggml-model.sh	2024-01-05 07:01:53.561389400 +0100
@@ -19,7 +19,9 @@
     fi
 }
 
-models_path="${2:-$(get_script_path)}"
+default_path="$(pwd)/models"
+models_path="${2:-$default_path}"
+mkdir -p "$models_path"
 
 # Whisper models
 models=(
@@ -107,5 +109,5 @@
 
 printf "Done! Model '$model' saved in '$models_path/ggml-$model.bin'\n"
 printf "You can now use it like this:\n\n"
-printf "  $ ./main -m $models_path/ggml-$model.bin -f samples/jfk.wav\n"
+printf "  $ whisper.cpp -m $models_path/ggml-$model.bin -f samples/jfk.wav\n"
 printf "\n"
