--- configure.ac.orig	2014-06-08 22:43:12.416400000 +0400
+++ configure.ac	2014-06-08 22:43:55.800000000 +0400
@@ -591,10 +591,10 @@
   if test -n "$with_cross_host" &&
      test x"$with_cross_host" != x"no"; then
     toolexecdir='$(exec_prefix)/$(target_alias)'
-    toolexeclibdir='$(toolexecdir)/lib'
+    toolexeclibdir='${toolexecdir}/lib'
   else
     toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
-    toolexeclibdir='$(libdir)'
+    toolexeclibdir='${libdir}'
   fi
   multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
   case $multi_os_directory in
@@ -603,7 +603,7 @@
   esac
   AC_SUBST(toolexecdir)
 else
-  toolexeclibdir='$(libdir)'
+  toolexeclibdir='${libdir}'
 fi
 AC_SUBST(toolexeclibdir)
 
