--- a/src/ifcgeom/kernels/cgal/cgal_conversion_result.h
+++ b/src/ifcgeom/kernels/cgal/cgal_conversion_result.h
@@ -77,12 +77,12 @@
 typedef std::vector<kernel_::Point_3> cgal_curve;
 typedef std::vector<kernel_::Point_3> cgal_wire;
 
-namespace {
+namespace ifcopenshell { namespace geom {
 	struct cgal_face {
 		cgal_wire outer;
 		std::vector<cgal_wire> inner;
 	};
-}
+}}
 
 typedef CGAL::Polyhedron_3<kernel_> cgal_polyhedron;
 typedef boost::graph_traits<CGAL::Polyhedron_3<kernel_>>::vertex_descriptor cgal_vertex_descriptor;
--- a/src/serializers/geometry_obj_plugin.cpp
+++ b/src/serializers/geometry_obj_plugin.cpp
@@ -32,7 +32,7 @@
 std::string obj_mtl_filename(const std::string& output_filename) {
 	std::filesystem::path path(ifcopenshell::path::from_utf8(output_filename));
 	path.replace_extension(ifcopenshell::path::from_utf8(".mtl"));
-	return ifcopenshell::path::to_utf8(path.native());
+	return ifcopenshell::path::to_utf8(path.string());
 }
 
 }
