--- a/src/svgfill/CMakeLists.txt
+++ b/src/svgfill/CMakeLists.txt
@@ -36,7 +36,7 @@ find_package(Boost)
 find_package(LibXml2 REQUIRED)
 find_package(CGAL REQUIRED)

-set(SVGPP_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/svgpp/include")
+set(SVGPP_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/svgpp/include" CACHE PATH "Path to svgpp include directory")
 if(NOT EXISTS ${SVGPP_INCLUDE})
     message(
         FATAL_ERROR
