--- IfcOpenShell-bonsai-0.8.5-post1/src/ifcgeom/profile_helper.h.orig	2026-05-08 15:05:00.245608200 +0200
+++ IfcOpenShell-bonsai-0.8.5-post1/src/ifcgeom/profile_helper.h	2026-05-08 15:05:24.276299700 +0200
@@ -13,6 +13,9 @@
 			profile_point(const std::array<double, 2>& p, const boost::optional<double>& r = boost::none)
 				: xy(p), radius(r) {
 			}
+			profile_point(const std::array<double, 2>& p, double r)
+				: profile_point(p, boost::optional<double>(r)) {
+			}
 		};
 
 		struct profile_point_with_edges {
