diff -bur pdf2djvu-0.9.18.2/pdf2djvu.cc pdf2djvu-0.9.18.2-c/pdf2djvu.cc
--- pdf2djvu-0.9.18.2/pdf2djvu.cc	2021-11-22 10:38:31.000000000 -0700
+++ pdf2djvu-0.9.18.2-c/pdf2djvu.cc	2022-08-01 01:25:39.531678200 -0600
@@ -89,7 +89,7 @@
 #endif
   }
   else
-    dest.reset(orig_dest->copy());
+    dest = std::make_unique<LinkDest>(*goto_link->getDest());
   if (dest.get() != nullptr)
   {
     int page;
diff -bur pdf2djvu-0.9.18.2/pdf-backend.cc pdf2djvu-0.9.18.2-c/pdf-backend.cc
--- pdf2djvu-0.9.18.2/pdf-backend.cc	2021-11-22 10:38:31.000000000 -0700
+++ pdf2djvu-0.9.18.2-c/pdf-backend.cc	2022-08-01 01:50:04.200748400 -0600
@@ -130,7 +130,7 @@
  */
 
 pdf::Document::Document(const std::string &file_name)
-: ::PDFDoc(new pdf::String(file_name.c_str()), nullptr, nullptr)
+: ::PDFDoc(std::make_unique<GooString>(file_name))
 {
   if (!this->isOk())
     throw LoadError();
