diff -urN moab-5.2.1.orig/CMakeLists.txt moab-5.2.1/CMakeLists.txt
--- moab-5.2.1.orig/CMakeLists.txt	2020-08-23 04:59:23.000000000 +0200
+++ moab-5.2.1/CMakeLists.txt	2021-04-16 20:35:16.007525900 +0200
@@ -391,7 +391,7 @@
 add_subdirectory( src )
 message(STATUS "Adding itaps subdirectory.")
 add_subdirectory( itaps )
-if (NOT WIN32) # Need further work to prepare for windows
+if (NOT WIN32 OR MSYS OR MINGW) # Need further work to prepare for windows
   add_subdirectory( tools )
 endif ()
 if (ENABLE_PYMOAB)
diff -urN moab-5.2.1.orig/src/CMakeLists.txt moab-5.2.1/src/CMakeLists.txt
--- moab-5.2.1.orig/src/CMakeLists.txt	2020-08-22 06:11:37.000000000 +0200
+++ moab-5.2.1/src/CMakeLists.txt	2021-04-20 08:41:12.864382700 +0200
@@ -252,6 +252,7 @@
 set( MOAB_GENERATED_INSTALL_HEADERS
      ${CMAKE_CURRENT_BINARY_DIR}/moab/EntityHandle.hpp
      ${CMAKE_CURRENT_BINARY_DIR}/moab/MOABConfig.h
+     ${CMAKE_CURRENT_BINARY_DIR}/moab_export.h
   )
 
 if( MOAB_HAVE_CGM )
@@ -310,6 +311,7 @@
   "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>")
 
 include(GenerateExportHeader)
+#generate_export_header(MOAB EXPORT_FILE_NAME moab/moab_export.h)
 generate_export_header(MOAB)
 
 set_target_properties(
diff -urN moab-5.2.1.orig/tools/refiner/test_mesh_refiner.cpp moab-5.2.1/tools/refiner/test_mesh_refiner.cpp
--- moab-5.2.1.orig/tools/refiner/test_mesh_refiner.cpp	2020-07-28 18:47:21.000000000 +0200
+++ moab-5.2.1/tools/refiner/test_mesh_refiner.cpp	2021-04-16 20:35:16.023125900 +0200
@@ -17,6 +17,8 @@
 #include <map>
 
 #include <ctime>
+#include <time.h>
+#include <sys/time.h>
 
 using namespace moab;
 
