diff -urN moab-5.2.1.orig/CMakeLists.txt moab-5.2.1/CMakeLists.txt
--- moab-5.2.1.orig/CMakeLists.txt	2020-08-23 04:59:23.000000000 +0200
+++ moab-5.2.1/CMakeLists.txt	2021-02-03 15:57:13.020620400 +0100
@@ -391,7 +391,7 @@
 add_subdirectory( src )
 message(STATUS "Adding itaps subdirectory.")
 add_subdirectory( itaps )
-if (NOT WIN32) # Need further work to prepare for windows
+if (NOT WIN32 OR MSYS OR MINGW) # Need further work to prepare for windows
   add_subdirectory( tools )
 endif ()
 if (ENABLE_PYMOAB)
