diff -urN moab-5.3.0_orig/tools/refiner/test_mesh_refiner.cpp moab-5.3.0/tools/refiner/test_mesh_refiner.cpp
--- moab-5.3.0_orig/tools/refiner/test_mesh_refiner.cpp	2021-07-13 17:57:13.000000000 +0200
+++ moab-5.3.0/tools/refiner/test_mesh_refiner.cpp	2021-07-14 09:42:37.167185200 +0200
@@ -8,7 +8,7 @@
 #ifdef MOAB_HAVE_MPI
 #include "moab/ParallelComm.hpp"
 #include "moab_mpi.h"
-#if defined( _MINGW32_ ) || defined( _MINGW64_ )
+#if defined( __MINGW32__ ) || defined( __MINGW64__ )
 #include <time.h>
 #include <sys/time.h>
 #endif
