diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-11-12 07:17:17.000000000 +0100
+++ b/CMakeLists.txt	2022-02-20 21:11:47.292208900 +0100
@@ -245,8 +245,6 @@
 
 if ( LAPACK_FOUND )
 
-  SET (ENABLE_FORTRAN ON)
-
   # Eigen3 is optional if LAPACK has been found
   FIND_PACKAGE(Eigen3 OPTIONAL_COMPONENTS)
 

diff -urN src/moab/Matrix3.hpp.orig src/moab/Matrix3.hpp
--- a/src/moab/Matrix3.hpp	2021-11-12 07:17:18.000000000 +0100
+++ b/src/moab/Matrix3.hpp	2022-02-20 21:32:02.346125100 +0100
@@ -71,7 +71,7 @@
 
 // We will rely on LAPACK directly
 
-#ifdef WIN32
+#ifdef _MSC_VER
 
 // Should use second form below for windows but
 // needed to do this to make it work.
