From 57d1f7e6b49149dd0f48ce9c5891f499ab689925 Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Thu, 31 Jul 2014 11:50:50 +0200
Subject: [PATCH 1/2] msys2: python3-config returns 1, so use which

---
 acinclude.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/acinclude.m4 b/acinclude.m4
index c0d1a5e..4ecce7e 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -45,7 +45,7 @@ AC_MSG_CHECKING(for headers required to compile python extensions)
 dnl deduce PYTHON_INCLUDES
 py_prefix=`$PYTHON -c "import sys; print(sys.prefix)"`
 py_exec_prefix=`$PYTHON -c "import sys; print(sys.exec_prefix)"`
-if $PYTHON-config --help 1>/dev/null 2>/dev/null; then
+if which $PYTHON-config; then
   PYTHON_INCLUDES=`$PYTHON-config --includes 2>/dev/null`
   if $PYTHON-config --abiflags 1>/dev/null 2>/dev/null; then
     PYTHON_ABI_FLAGS=`$PYTHON-config --abiflags 2>/dev/null`
-- 
2.0.4

