--- gst-python-1.4.0/plugin/Makefile.am.orig	2015-07-21 14:56:42.175010500 +0300
+++ gst-python-1.4.0/plugin/Makefile.am	2015-07-21 14:57:19.383289300 +0300
@@ -1,4 +1,6 @@
-plugin_LTLIBRARIES = libgstpythonplugin.la
+plugindir = $(PLUGINDIR)
+pluginexecdir = $(PLUGINDIR)
+pluginexec_LTLIBRARIES = libgstpythonplugin.la
 
 INCLUDES = $(PYGOBJECT_CFLAGS) $(GST_CFLAGS)\
 -DPYTHON_VERSION=\"$(PYTHON_VERSION)\"		\
