From 1c56d65b20a354fa4c83e06524aee8e53d503c62 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 14 Jul 2015 16:29:12 +0100
Subject: [PATCH 1/6] GYP: Add missing files

---
 src/glsl_optimizer_lib.gyp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/glsl_optimizer_lib.gyp b/src/glsl_optimizer_lib.gyp
index 37ae920..980683a 100644
--- a/src/glsl_optimizer_lib.gyp
+++ b/src/glsl_optimizer_lib.gyp
@@ -114,7 +114,7 @@
         'glsl/lower_vec_index_to_swizzle.cpp',
         'glsl/lower_vector.cpp',
         'glsl/lower_vector_insert.cpp',
-        'glsl/lower_named_interface_blocks.cpp',
+        'glsl/lower_vertex_id.cpp',
         'glsl/opt_algebraic.cpp',
         'glsl/opt_array_splitting.cpp',
         'glsl/opt_constant_folding.cpp',
@@ -136,8 +136,9 @@
         'glsl/opt_swizzle_swizzle.cpp',
         'glsl/opt_tree_grafting.cpp',
         'glsl/opt_vectorize.cpp',
-        'glsl/opt_flip_matrices.cpp',
-        'glsl/opt_dead_builtin_varyings.cpp',
+        'glsl/opt_minmax.cpp',
+        'glsl/opt_rebalance_tree.cpp',
+        'glsl/opt_dead_builtin_variables.cpp',
         'glsl/program.h',
         'glsl/s_expression.cpp',
         'glsl/s_expression.h',
-- 
2.4.5

