diff -bur FreeCAD-1.1.0-orig/src/Base/UnlimitedUnsigned.h FreeCAD-1.1.0/src/Base/UnlimitedUnsigned.h
--- FreeCAD-1.1.0-orig/src/Base/UnlimitedUnsigned.h	2026-04-12 20:57:33.087401400 -0600
+++ FreeCAD-1.1.0/src/Base/UnlimitedUnsigned.h	2026-04-12 20:57:49.867634300 -0600
@@ -31,7 +31,7 @@
 #include <vector>
 #include <string>
 #include <stdexcept>
-
+#include <cstdint>
 // ----------------------------------------------------------------------------
 
 namespace Base
