diff -bur FreeCAD-1.1.0-orig/src/Mod/Sketcher/App/GeoEnum.h FreeCAD-1.1.0/src/Mod/Sketcher/App/GeoEnum.h
--- FreeCAD-1.1.0-orig/src/Mod/Sketcher/App/GeoEnum.h	2026-04-13 21:42:25.207342000 -0600
+++ FreeCAD-1.1.0/src/Mod/Sketcher/App/GeoEnum.h	2026-04-13 21:42:55.292229300 -0600
@@ -129,6 +129,11 @@
 
     int posIdAsInt() const;
 
+    inline bool operator<(const GeoElementId& rhs) const {
+        return (GeoId != rhs.GeoId) ? (GeoId < rhs.GeoId)
+                                    : (static_cast<int>(Pos) < static_cast<int>(rhs.Pos));
+    }
+
     /** @brief GeoElementId of the Root Point
      */
     static const GeoElementId RtPnt;
