diff -bur FreeCAD-1.1.0-orig/src/Mod/TechDraw/Gui/CommandCreateDims.cpp FreeCAD-1.1.0/src/Mod/TechDraw/Gui/CommandCreateDims.cpp
--- FreeCAD-1.1.0-orig/src/Mod/TechDraw/Gui/CommandCreateDims.cpp	2026-04-12 23:15:42.816307200 -0600
+++ FreeCAD-1.1.0/src/Mod/TechDraw/Gui/CommandCreateDims.cpp	2026-04-12 23:19:14.455106400 -0600
@@ -1097,7 +1097,7 @@
             TechDraw::pointPair pp = dim->getLinearPoints();
             float dx = pp.first().x - pp.second().x;
             float dy = pp.first().y - pp.second().y;
-            int alpha = std::round(Base::toDegrees(std::abs<float>(std::atan(type == "DistanceY" ? (dx / dy) : (dy / dx)))));
+            int alpha = std::round(Base::toDegrees(std::fabs<float>(std::atan(type == "DistanceY" ? (dx / dy) : (dy / dx)))));
             std::string sAlpha = std::to_string(alpha);
             std::string formatSpec = dim->FormatSpec.getStrValue();
             formatSpec = formatSpec + " x" + sAlpha + "°";
diff -bur FreeCAD-1.1.0-orig/src/Mod/TechDraw/Gui/CommandExtensionDims.cpp FreeCAD-1.1.0/src/Mod/TechDraw/Gui/CommandExtensionDims.cpp
--- FreeCAD-1.1.0-orig/src/Mod/TechDraw/Gui/CommandExtensionDims.cpp	2026-04-12 23:15:42.816307200 -0600
+++ FreeCAD-1.1.0/src/Mod/TechDraw/Gui/CommandExtensionDims.cpp	2026-04-12 23:19:02.672285800 -0600
@@ -1984,7 +1984,7 @@
         dim->Y.setValue(-yMax);
         float dx = allVertexes[0].point.x - allVertexes[1].point.x;
         float dy = allVertexes[0].point.y - allVertexes[1].point.y;
-        float alpha = std::round(Base::toDegrees(std::abs<float>(std::atan(dy / dx))));
+        float alpha = std::round(Base::toDegrees(std::fabs<float>(std::atan(dy / dx))));
         std::string sAlpha = std::to_string((int)alpha);
         std::string formatSpec = dim->FormatSpec.getStrValue();
         formatSpec = formatSpec + " x" + sAlpha + "°";
@@ -2050,7 +2050,7 @@
         dim->Y.setValue(-mid.y);
         float dx = allVertexes[0].point.x - allVertexes[1].point.x;
         float dy = allVertexes[0].point.y - allVertexes[1].point.y;
-        float alpha = std::round(Base::toDegrees(std::abs<float>(std::atan(dx / dy))));
+        float alpha = std::round(Base::toDegrees(std::fabs<float>(std::atan(dx / dy))));
         std::string sAlpha = std::to_string((int)alpha);
         std::string formatSpec = dim->FormatSpec.getStrValue();
         formatSpec = formatSpec + " x" + sAlpha + "°";
