--- a/src/Gui/PythonWrapper.cpp
+++ b/src/Gui/PythonWrapper.cpp
@@ -124,11 +128,24 @@
 # define HAVE_PYSIDE
 # define HAVE_SHIBOKEN_TYPE_FOR_TYPENAME
 # include <signalmanager.h>
+# include <sbkversion.h>
+# define SHIBOKEN_FULL_VERSION QT_VERSION_CHECK(SHIBOKEN_MAJOR_VERSION, SHIBOKEN_MINOR_VERSION, 0)
+# if (SHIBOKEN_FULL_VERSION >= QT_VERSION_CHECK(6, 7, 0))
+#  define HAVE_SHIBOKEN_TYPEINITSTRUCT
+# endif
+#ifdef HAVE_SHIBOKEN_TYPEINITSTRUCT
+Shiboken::Module::TypeInitStruct* SbkPySide6_QtCoreTypes           = nullptr;
+Shiboken::Module::TypeInitStruct* SbkPySide6_QtGuiTypes            = nullptr;
+Shiboken::Module::TypeInitStruct* SbkPySide6_QtWidgetsTypes        = nullptr;
+Shiboken::Module::TypeInitStruct* SbkPySide6_QtPrintSupportTypes   = nullptr;
+Shiboken::Module::TypeInitStruct* SbkPySide6_QtUiToolsTypes        = nullptr;
+#else
 PyTypeObject** SbkPySide6_QtCoreTypes = nullptr;
 PyTypeObject** SbkPySide6_QtGuiTypes = nullptr;
 PyTypeObject** SbkPySide6_QtWidgetsTypes = nullptr;
 PyTypeObject** SbkPySide6_QtPrintSupportTypes = nullptr;
 PyTypeObject** SbkPySide6_QtUiToolsTypes = nullptr;
+#endif
 # endif // HAVE_PYSIDE6
 #endif // HAVE_SHIBOKEN6
 
@@ -471,7 +484,11 @@
     return nullptr;
 }
 
+#ifdef HAVE_SHIBOKEN_TYPEINITSTRUCT
+bool loadPySideModule(const std::string& moduleName, Shiboken::Module::TypeInitStruct*& types)
+#else
 bool loadPySideModule(const std::string& moduleName, PyTypeObject**& types)
+#endif
 {
 #if defined (HAVE_SHIBOKEN) && defined(HAVE_PYSIDE)
     if (!types) {
