diff -urN readline-8.3/rlwinsize.h.orig readline-8.3/rlwinsize.h
--- readline-8.3/rlwinsize.h.orig	2025-06-05 15:22:28.000000000 +0200
+++ readline-8.3/rlwinsize.h	2025-07-08 14:17:34.166872700 +0200
@@ -55,6 +55,11 @@
 #  define tcflow(fd, action)	ioctl(fd, TCXONC, action)
 #endif
 
+#if defined (__MINGW32__)
+// On Windows, define dummy struct with no fields.
+struct winsize {};
+#endif
+
 extern int _rl_tcgetwinsize (int, struct winsize *);
 extern void _rl_tcsetwinsize (int, struct winsize *);
 
