--- a/cmake/modules/FindMySQL.cmake
+++ b/cmake/modules/FindMySQL.cmake
@@ -16,7 +16,7 @@
 
 include(CheckCXXSourceCompiles)
 
-if(WIN32)
+if(MSVC)
    find_path(MYSQL_INCLUDE_DIR mysql.h
       PATHS
       $ENV{MYSQL_INCLUDE_DIR}
@@ -25,7 +25,7 @@
       $ENV{SystemDrive}/MySQL/*/include
       $ENV{ProgramW6432}/MySQL/*/include
    )
-else(WIN32)
+else(MSVC)
    find_path(MYSQL_INCLUDE_DIR mysql.h
       PATHS
       $ENV{MYSQL_INCLUDE_DIR}
@@ -35,9 +35,9 @@
       PATH_SUFFIXES
       mysql
    )
-endif(WIN32)
+endif(MSVC)
 
-if(WIN32)
+if(MSVC)
    if (${CMAKE_BUILD_TYPE})
     string(TOLOWER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_TOLOWER)
    endif()
@@ -71,7 +71,7 @@
       PATHS
       ${MYSQL_LIB_PATHS}
    )
-else(WIN32)
+else(MSVC)
 #   find_library(MYSQL_LIBRARIES NAMES mysqlclient
    set(MYSQL_LIB_PATHS
       $ENV{MYSQL_DIR}/libmysql_r/.libs
@@ -91,7 +91,7 @@
       PATHS
       ${MYSQL_LIB_PATHS}
    )
-endif(WIN32)
+endif(MSVC)
 
 find_library(MYSQL_EMBEDDED_LIBRARIES NAMES mysqld
    PATHS
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -137,7 +137,7 @@
   message(FATAL_ERROR "No thread library found")
 endif()
 
-if(NOT MSVC)
+if(NOT WIN32)
   set(DL_FIND_QUIETLY TRUE)
   find_package(DL)
   if(DL_FOUND)
