diff -Naur openexr-2.2.0-orig/CMakeLists.txt openexr-2.2.0/CMakeLists.txt
--- openexr-2.2.0-orig/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/CMakeLists.txt	2014-05-28 16:55:20.527800000 +0400
@@ -40,7 +40,7 @@
 FIND_PACKAGE(ZLIB REQUIRED)
 INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
 
-IF (NOT WIN32)
+IF (NOT WIN32 OR MINGW)
   SET ( PTHREAD_LIB pthread )
 ENDIF()
 
@@ -107,7 +107,9 @@
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
-  OUTPUT_NAME "IlmImf${OPENEXR_LIBSUFFIX}"
+  OUTPUT_NAME IlmImf
+  RUNTIME_OUTPUT_NAME "IlmImf${OPENEXR_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmImf
   )
   
 ##########################
@@ -188,7 +188,9 @@
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
-  OUTPUT_NAME "IlmImfUtil${OPENEXR_LIBSUFFIX}"
+  OUTPUT_NAME IlmImfUtil
+  RUNTIME_OUTPUT_NAME "IlmImfUtil${OPENEXR_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmImfUtil
   )
   
 ##########################
@@ -177,3 +180,24 @@
   DESTINATION
   ${CMAKE_INSTALL_PREFIX}/share/doc/OpenEXR-${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}/examples
   )
+
+FILE ( WRITE ${CMAKE_BINARY_DIR}/OpenEXR.pc "prefix=${CMAKE_INSTALL_PREFIX}\n" )
+FILE ( APPEND ${CMAKE_BINARY_DIR}/OpenEXR.pc "exec_prefix=\${prefix}
+libdir=\${exec_prefix}/lib
+includedir=\${prefix}/include
+OpenEXR_includedir=\${prefix}/include/OpenEXR
+
+Name: OpenEXR
+Description: OpenEXR image library
+Version: ${OPENEXR_VERSION}
+Libs: -L\${libdir} -lIlmImf
+Cflags: -I\${OpenEXR_includedir}
+Requires: IlmBase
+Libs.private: -lz
+")
+
+INSTALL ( FILES
+  ${CMAKE_BINARY_DIR}/OpenEXR.pc
+  DESTINATION
+  ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig
+)
