From c1c82f7d2582fb74ad53e5cd1b6525e8dcdfa492 Mon Sep 17 00:00:00 2001
From: Jens Lindgren <lindgren_jens@hotmail.com>
Date: Tue, 22 Oct 2019 01:17:45 +0200
Subject: [PATCH] add boost to python module link library

---
 PyIlmBase/config/ModuleDefine.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/PyIlmBase/config/ModuleDefine.cmake b/PyIlmBase/config/ModuleDefine.cmake
index 43ffb516..e3749979 100644
--- a/PyIlmBase/config/ModuleDefine.cmake
+++ b/PyIlmBase/config/ModuleDefine.cmake
@@ -116,6 +116,7 @@ function(PYILMBASE_DEFINE_MODULE modname)
         ${extraDeps}
         ${PYILMBASE_CURMOD_DEPENDENCIES}
         ${PYILMBASE_CURMOD_PRIVATE_DEPS}
+        Boost::${PYILMBASE_BOOST_PY2_COMPONENT}
       )
     set_target_properties(${modname}_python2 PROPERTIES
       LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}/"
@@ -142,6 +143,7 @@ function(PYILMBASE_DEFINE_MODULE modname)
         ${libname} ${extraDeps}
         ${PYILMBASE_CURMOD_DEPENDENCIES}
         ${PYILMBASE_CURMOD_PRIVATE_DEPS}
+        Boost::${PYILMBASE_BOOST_PY3_COMPONENT}
       )
     set_target_properties(${modname}_python3 PROPERTIES
       LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python${Python3_VERSION_MAJOR}_${Python3_VERSION_MINOR}/"
