--- openexr-3.1.5/src/lib/OpenEXR/ImfExport.h.orig	2022-05-17 14:02:14.390495900 +0300
+++ openexr-3.1.5/src/lib/OpenEXR/ImfExport.h	2022-05-17 14:03:03.099166800 +0300
@@ -27,7 +27,11 @@
        // for mingw windows, we need to cause this to export the
        // typeinfo tables (but you don't need to have the
        // complementary import, because might be a local template too!)
-#      define IMF_EXPORT_TEMPLATE_TYPE IMF_EXPORT
+#      if defined(__clang__)
+#          define IMF_EXPORT_TEMPLATE_TYPE
+#      else
+#          define IMF_EXPORT_TEMPLATE_TYPE IMF_EXPORT
+#      endif
 #    else
        // for normal msvc, need to export the actual instantiation in
        // the cpp code, and none of the others
--- openexr-3.1.5/src/lib/OpenEXRUtil/ImfUtilExport.h.orig	2022-05-17 14:00:11.332897100 +0300
+++ openexr-3.1.5/src/lib/OpenEXRUtil/ImfUtilExport.h	2022-05-17 14:01:48.799323000 +0300
@@ -24,7 +24,11 @@
        // for mingw windows, we need to cause this to export the
        // typeinfo tables (but you don't need to have the
        // complementary import, because might be a local template too!)
-#      define IMFUTIL_EXPORT_TEMPLATE_TYPE IMFUTIL_EXPORT
+#      if defined(__clang__)
+#          define IMFUTIL_EXPORT_TEMPLATE_TYPE
+#      else
+#          define IMFUTIL_EXPORT_TEMPLATE_TYPE IMFUTIL_EXPORT
+#      endif
 #    else
        // for normal msvc, need to export the actual instantiation in
        // the cpp code, and none of the others
