--- openexr-2.2.0/exrenvmap/CMakeLists.txt.orig	2016-10-14 10:54:28.391519300 +0300
+++ openexr-2.2.0/exrenvmap/CMakeLists.txt	2016-10-14 10:54:50.002791100 +0300
@@ -10,6 +10,11 @@
   EnvmapImage.cpp
 )
 
+if (MINGW AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.1.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 6.1.0))  # 6.1.0 is the last known bad version
+      string (REPLACE -O3 -O2 CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+      string (REPLACE -O2 -O0 CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+endif ()
+
 TARGET_LINK_LIBRARIES ( exrenvmap
   IlmImf
   ${IlmBase_LIBRARIES}
--- openexr-2.2.0/IlmImfTest/CMakeLists.txt.orig	2016-10-14 10:55:06.631998400 +0300
+++ openexr-2.2.0/IlmImfTest/CMakeLists.txt	2016-10-14 10:55:13.237280500 +0300
@@ -61,6 +61,10 @@
   testYca.cpp
  )
 
+if (MINGW AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.1.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 6.1.0))  # 6.1.0 is the last known bad version
+      string (REPLACE -O3 -O2 CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+      string (REPLACE -O2 -O0 CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+endif ()
 
 ADD_TEST ( TestIlmImf IlmImfTest )
 TARGET_LINK_LIBRARIES ( IlmImfTest 
