From 2aa28bb574064749b434bef435c35bd2bd018b69 Mon Sep 17 00:00:00 2001
From: Marcus von Appen <mva@sysfault.org>
Date: Sun, 12 Jan 2014 13:43:29 +0100
Subject: [PATCH] - make library naming of libIlmImf consistent for cmake and
 the   autotools

---
 OpenEXR/configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/OpenEXR/configure.ac b/OpenEXR/configure.ac
index 90570d4..9b378dc 100644
--- a/OpenEXR/configure.ac
+++ b/OpenEXR/configure.ac
@@ -261,7 +261,7 @@ if test "x${library_namespace_versioning}" == xyes ; then
     AC_DEFINE(OPENEXR_IMF_INTERNAL_NAMESPACE_CUSTOM)
 
     lib_namespace="Imf_${OPENEXR_VERSION_API}"
-    LIB_SUFFIX="Imf_${OPENEXR_VERSION_API}"
+    LIB_SUFFIX="${OPENEXR_VERSION_API}"
     lib_suffix_valid="yes"
 elif test "x${library_namespace_versioning}" == xno ; then
     AC_DEFINE_UNQUOTED(OPENEXR_IMF_INTERNAL_NAMESPACE, Imf)
-- 
1.9.3
