--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -238,6 +238,10 @@
 find_package(Threads REQUIRED)
 target_link_libraries(rdkafka PUBLIC Threads::Threads)
 
+if(WITH_CURL)
+    target_link_libraries(rdkafka PUBLIC curl)
+endif()
+
 if(WITH_SASL_CYRUS)
   target_include_directories(rdkafka PRIVATE ${SASL_INCLUDE_DIRS})
   target_link_libraries(rdkafka PUBLIC ${SASL_LIBRARIES})
@@ -287,6 +291,10 @@
 if(NOT RDKAFKA_BUILD_STATIC)
   set(PKG_CONFIG_NAME "librdkafka")
   set(PKG_CONFIG_DESCRIPTION "The Apache Kafka C/C++ library")
+  if(WITH_CURL)
+    string(APPEND PKG_CONFIG_REQUIRES "curl ")
+  endif()
+
   if(WITH_ZLIB)
     string(APPEND PKG_CONFIG_REQUIRES "zlib ")
   endif()
