diff --git a/gtk/gtkimcontextsimple.c b/gtk/gtkimcontextsimple.c
index a090ea3e08..05d94d7d06 100644
--- a/gtk/gtkimcontextsimple.c
+++ b/gtk/gtkimcontextsimple.c
@@ -135,7 +135,7 @@ gtk_im_context_simple_class_init (GtkIMContextSimpleClass *class)
   im_context_class->get_preedit_string = gtk_im_context_simple_get_preedit_string;
   gobject_class->finalize = gtk_im_context_simple_finalize;
 
-  init_compose_table_async (NULL, NULL, NULL);
+  // init_compose_table_async (NULL, NULL, NULL);
 }
 
 static char *
