Export separate targets for shared and for static libraries.

diff -urN libaec-v1.1.3/cmake/libaec-config.cmake.in.orig libaec-v1.1.3/cmake/libaec-config.cmake.in
--- libaec-v1.1.3/cmake/libaec-config.cmake.in.orig	2025-05-06 16:15:50.757950300 +0200
+++ libaec-v1.1.3/cmake/libaec-config.cmake.in	2025-05-06 16:27:44.518380600 +0200
@@ -26,17 +26,21 @@
 
 find_path(libaec_INCLUDE_DIR NAMES libaec.h DOC "AEC include directory")
 find_path(SZIP_INCLUDE_DIR NAMES szlib.h DOC "SZIP include directory")
+find_library(libaec-shared_LIBRARY NAMES aec DOC "AEC library")
+find_library(SZIP-shared_LIBRARY NAMES sz szip DOC "SZIP compatible version of the AEC library")
+if (MSVC)
+  find_library(libaec-static_LIBRARY NAMES aec-static.lib DOC "AEC library")
+  find_library(SZIP-static_LIBRARY NAMES szip-static.lib DOC "SZIP compatible version of the AEC library")
+else ()
+  find_library(libaec-static_LIBRARY NAMES libaec.a DOC "AEC library")
+  find_library(SZIP-static_LIBRARY NAMES libsz.a DOC "SZIP compatible version of the AEC library")
+endif ()
 if (libaec_USE_STATIC_LIBS)
-  if (MSVC)
-    find_library(libaec_LIBRARY NAMES aec-static.lib DOC "AEC library")
-    find_library(SZIP_LIBRARY NAMES szip-static.lib DOC "SZIP compatible version of the AEC library")
-  else ()
-    find_library(libaec_LIBRARY NAMES libaec.a DOC "AEC library")
-    find_library(SZIP_LIBRARY NAMES libsz.a DOC "SZIP compatible version of the AEC library")
-  endif ()
+  set(libaec_LIBRARY ${libaec-static_LIBRARY})
+  set(SZIP_LIBRARY ${SZIP-static_LIBRARY})
 else ()
-  find_library(libaec_LIBRARY NAMES aec DOC "AEC library")
-  find_library(SZIP_LIBRARY NAMES sz szip DOC "SZIP compatible version of the AEC library")
+  set(libaec_LIBRARY ${libaec-shared_LIBRARY})
+  set(SZIP_LIBRARY ${SZIP-shared_LIBRARY})
 endif ()
 
 # Check version here
@@ -55,38 +59,58 @@
 )
 
 if (libaec_FOUND)
-  if (libaec_USE_STATIC_LIBS)
-    add_library(libaec::aec STATIC IMPORTED)
-  else ()
-    add_library(libaec::aec SHARED IMPORTED)
-    target_compile_definitions(libaec::aec INTERFACE LIBAEC_SHARED)
+  if (libaec-static_LIBRARY)
+    add_library(libaec::aec-static STATIC IMPORTED)
+    set_target_properties(libaec::aec-static PROPERTIES
+      IMPORTED_LOCATION "${libaec-static_LIBRARY}"
+      INTERFACE_INCLUDE_DIRECTORIES "${libaec_INCLUDE_DIR}"
+    )
+  endif ()
+  if (libaec-static_LIBRARY)
+    add_library(libaec::aec-shared STATIC IMPORTED)
+    target_compile_definitions(libaec::aec-shared INTERFACE LIBAEC_SHARED)
     if (WIN32)
-      set_target_properties(libaec::aec PROPERTIES
+      set_target_properties(libaec::aec-shared PROPERTIES
         IMPORTED_IMPLIB "${libaec_LIBRARY}"
       )
     endif ()
+    set_target_properties(libaec::aec-shared PROPERTIES
+      IMPORTED_LOCATION "${libaec-shared_LIBRARY}"
+      INTERFACE_INCLUDE_DIRECTORIES "${libaec_INCLUDE_DIR}"
+    )
+  endif ()
+  if (libaec_USE_STATIC_LIBS)
+    add_library(libaec::aec ALIAS libaec::aec-static)
+  else ()
+    add_library(libaec::aec ALIAS libaec::aec-shared)
   endif ()
-  set_target_properties(libaec::aec PROPERTIES
-    IMPORTED_LOCATION "${libaec_LIBRARY}"
-    INTERFACE_INCLUDE_DIRECTORIES "${libaec_INCLUDE_DIR}"
-  )
 
   # SZIP
-  if (libaec_USE_STATIC_LIBS)
-    add_library(libaec::sz STATIC IMPORTED)
-  else ()
-    add_library(libaec::sz SHARED IMPORTED)
-    target_compile_definitions(libaec::sz INTERFACE LIBAEC_SHARED)
+  if (SZIP-static_LIBRARY)
+    add_library(libaec::sz-static STATIC IMPORTED)
+    set_target_properties(libaec::sz-static PROPERTIES
+      IMPORTED_LOCATION "${SZIP-static_LIBRARY}"
+      INTERFACE_INCLUDE_DIRECTORIES "${SZIP_INCLUDE_DIR}"
+    )
+  endif ()
+  if (libaec-static_LIBRARY)
+    add_library(libaec::sz-shared SHARED IMPORTED)
+    target_compile_definitions(libaec::sz-shared INTERFACE LIBAEC_SHARED)
     if (WIN32)
-      set_target_properties(libaec::sz PROPERTIES
-        IMPORTED_IMPLIB "${SZIP_LIBRARY}"
+      set_target_properties(libaec::sz-shared PROPERTIES
+        IMPORTED_IMPLIB "${SZIP-shared_LIBRARY}"
       )
     endif ()
+    set_target_properties(libaec::sz-shared PROPERTIES
+      IMPORTED_LOCATION "${SZIP-shared_LIBRARY}"
+      INTERFACE_INCLUDE_DIRECTORIES "${SZIP_INCLUDE_DIR}"
+    )
+  endif ()
+  if (libaec_USE_STATIC_LIBS)
+    add_library(libaec::sz ALIAS libaec::sz-static)
+  else ()
+    add_library(libaec::sz ALIAS libaec::sz-shared)
   endif ()
-  set_target_properties(libaec::sz PROPERTIES
-    IMPORTED_LOCATION "${SZIP_LIBRARY}"
-    INTERFACE_INCLUDE_DIRECTORIES "${SZIP_INCLUDE_DIR}"
-  )
 
   # Set SZIP variables.
   set(SZIP_FOUND TRUE)
@@ -95,7 +119,11 @@
 
 mark_as_advanced(
   libaec_LIBRARY
+  libaec-shared_LIBRARY
+  libaec-static_LIBRARY
   libaec_INCLUDE_DIR
   SZIP_LIBRARY
+  SZIP-shared_LIBRARY
+  SZIP-static_LIBRARY
   SZIP_INCLUDE_DIR
 )
