diff -Naur libaec-1.0.4_orig/CMakeLists.txt libaec-1.0.4/CMakeLists.txt
--- libaec-1.0.4_orig/CMakeLists.txt	2019-02-11 09:37:18.000000000 +0100
+++ libaec-1.0.4/CMakeLists.txt	2020-10-06 09:35:14.522544300 +0200
@@ -17,7 +17,7 @@
 set(libaec_VERSION_MINOR 0)
 set(libaec_VERSION_PATCH 4)
 set(CMAKE_BUILD_TYPE Release)
-enable_testing()
+# enable_testing()
 
 check_include_files(malloc.h HAVE_MALLOC_H)
 test_big_endian(WORDS_BIGENDIAN)
@@ -55,7 +55,7 @@
 include_directories("${PROJECT_SOURCE_DIR}/src")
 
 add_subdirectory(src)
-add_subdirectory(tests)
+# add_subdirectory(tests)
 if(AEC_FUZZING)
   add_subdirectory(fuzzing)
 endif()
