--- libaec-v1.0.5_orig/CMakeLists.txt	2021-06-16 11:51:34.000000000 +0200
+++ libaec-v1.0.5/CMakeLists.txt	2021-06-21 14:18:24.980141400 +0200
@@ -7,7 +7,7 @@
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-enable_testing()
+# enable_testing()
 
 include(TestBigEndian)
 test_big_endian(WORDS_BIGENDIAN)
@@ -39,7 +39,7 @@
   "${CMAKE_CURRENT_BINARY_DIR}/config.h")
 
 add_subdirectory(src)
-add_subdirectory(tests)
+# add_subdirectory(tests)
 
 option(AEC_FUZZING "Enable build of fuzzing target" OFF)
 if(AEC_FUZZING)
