diff -Naur ocaml-4.11.1.orig/runtime/caml/misc.h ocaml-4.11.1/runtime/caml/misc.h
--- ocaml-4.11.1.orig/runtime/caml/misc.h	2020-09-01 14:47:58.000000000 +0200
+++ ocaml-4.11.1/runtime/caml/misc.h	2020-12-14 11:33:11.834706300 +0100
@@ -406,7 +406,7 @@
 #ifdef _WIN32
 #ifndef _UCRT
 extern int caml_snprintf(char * buf, size_t size, const char * format, ...);
-#define snprintf caml_snprintf
+//#define snprintf caml_snprintf
 #endif
 
 extern int caml_snwprintf(wchar_t * buf,
