

From: David Rothenberger <daveroth@acm.org>


---
 m4/sasldb.m4 |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/m4/sasldb.m4 b/m4/sasldb.m4
index b28b087..85e769a 100644
--- a/m4/sasldb.m4
+++ b/m4/sasldb.m4
@@ -111,7 +111,7 @@ AC_MSG_RESULT($dblib)
 SASL_DB_BACKEND="db_${dblib}.lo"
 SASL_DB_BACKEND_STATIC="db_${dblib}.o allockey.o"
 SASL_DB_BACKEND_STATIC_SRCS="\$(top_srcdir)/sasldb/db_${dblib}.c \$(top_srcdir)/sasldb/allockey.c"
-SASL_DB_UTILS="saslpasswd2 sasldblistusers2"
+SASL_DB_UTILS='saslpasswd2$(EXEEXT) sasldblistusers2$(EXEEXT)'
 SASL_DB_MANS="saslpasswd2.8 sasldblistusers2.8"
 
 case "$dblib" in
