diff --unified -r cyrus-sasl-2.1.27.orig/configure.ac cyrus-sasl-2.1.27/configure.ac
--- cyrus-sasl-2.1.27.orig/configure.ac	2018-10-09 20:28:04.000000000 +0530
+++ cyrus-sasl-2.1.27/configure.ac	2021-05-03 23:19:32.289742100 +0530
@@ -1095,13 +1095,16 @@
                           be found [[LIBDIR/sasl2]] ],
   plugindir=$withval,
   plugindir='${libdir}/sasl2')
+AC_DEFINE_UNQUOTED(PLUGINDIR, "$plugindir", [Runtime plugin location])
 AC_SUBST(plugindir)
 
 AC_ARG_WITH(configdir, [   --with-configdir=DIR    set the directory where config files will
                           be found [PLUGINDIR:SYSCONFDIR/sasl2] ],
   configdir=$withval,
   configdir='${plugindir}:${sysconfdir}/sasl2')
+AC_DEFINE_UNQUOTED(CONFIGDIR, "$configdir", [Runtime config file location])
 AC_SUBST(configdir)
+
 FULL_EXPAND_VARIABLE(sasl_bindir, "$bindir")
 AC_DEFINE_UNQUOTED(BINDIR, "$sasl_bindir", [Runtime executables location])
 
diff --unified -r cyrus-sasl-2.1.27.orig/lib/Makefile.am cyrus-sasl-2.1.27/lib/Makefile.am
--- cyrus-sasl-2.1.27.orig/lib/Makefile.am	2018-10-05 20:10:16.000000000 +0530
+++ cyrus-sasl-2.1.27/lib/Makefile.am	2021-05-03 23:20:45.644604600 +0530
@@ -46,7 +46,7 @@
 sasl_version = 3:0:0
 
 AM_CPPFLAGS=-DLIBSASL_EXPORTS=1 -I$(top_srcdir)/include -I$(top_srcdir)/plugins \
-               -I$(top_builddir)/include -I$(top_srcdir)/sasldb -I$(top_srcdir)/common -DCONFIGDIR='"${configdir}"' -DPLUGINDIR='"${plugindir}"'
+               -I$(top_builddir)/include -I$(top_srcdir)/sasldb -I$(top_srcdir)/common
 PLUGIN_COMMON_OBJS = $(top_builddir)/common/libplugin_common.la
 
 EXTRA_DIST = windlopen.c dlopen.c staticopen.h NTMakefile
diff --unified -r cyrus-sasl-2.1.27.orig/sample/Makefile.am cyrus-sasl-2.1.27/sample/Makefile.am
--- cyrus-sasl-2.1.27.orig/sample/Makefile.am	2018-10-05 20:10:16.000000000 +0530
+++ cyrus-sasl-2.1.27/sample/Makefile.am	2021-05-03 23:21:06.965457200 +0530
@@ -42,7 +42,7 @@
 #
 ################################################################
 
-AM_CPPFLAGS=-I$(top_srcdir)/include -DPLUGINDIR='"${plugindir}"'
+AM_CPPFLAGS=-I$(top_srcdir)/include
 
 noinst_PROGRAMS = client server http_digest_client
 EXTRA_PROGRAMS = sample-client sample-server
diff --unified -r cyrus-sasl-2.1.27.orig/utils/Makefile.am cyrus-sasl-2.1.27/utils/Makefile.am
--- cyrus-sasl-2.1.27.orig/utils/Makefile.am	2018-10-05 20:10:16.000000000 +0530
+++ cyrus-sasl-2.1.27/utils/Makefile.am	2021-05-03 23:21:35.991353300 +0530
@@ -89,7 +89,7 @@
 libsfsasl2_la_LIBADD = sfsasl.lo
 libsfsasl2_la_LDFLAGS = -version-info 1:0:0 -export-dynamic -rpath $(libdir)
 
-AM_CPPFLAGS=-DLIBSASL_EXPORTS=1 -I$(top_srcdir)/include -I$(top_builddir)/include @SASL_DB_INC@ -DPLUGINDIR='"${plugindir}"'
+AM_CPPFLAGS=-DLIBSASL_EXPORTS=1 -I$(top_srcdir)/include -I$(top_builddir)/include @SASL_DB_INC@
 
 EXTRA_DIST = saslpasswd2.8 sasldblistusers2.8 pluginviewer.8 sfsasl.h sfsasl.c smtptest.c testsuite.c pluginviewer.c NTMakefile
 
