--- a/waf_gensyms.py
+++ b/waf_gensyms.py
@@ -35,6 +35,4 @@
     sym_file_name = os.path.splitext(self.link_task.outputs[0].abspath())[0] + '.def'
     sym_file = self.path.find_or_declare(sym_file_name)
     symtask = self.create_task('gen_sym_file', self.link_task.outputs, sym_file)
-    self.add_install_files(install_to=self.link_task.inst_to, install_from=sym_file,
-        chmod=O644, task=self.link_task)
 
