--- jsonnet-0.20.0-orig/third_party/rapidyaml/rapidyaml/ext/c4core/CMakeLists.txt	2023-11-09 23:20:06.514630800 +0300
+++ jsonnet-0.20.0/third_party/rapidyaml/rapidyaml/ext/c4core/CMakeLists.txt	2023-11-10 01:41:36.746706300 +0300
@@ -77,11 +77,11 @@
 
 #-------------------------------------------------------
 
-c4_install_target(c4core)
-c4_install_exports()
-c4_pack_project()
+#c4_install_target(c4core)
+#c4_install_exports()
+#c4_pack_project()
 
-c4_add_dev_targets()
+#c4_add_dev_targets()
 
-c4_set_default_pack_properties(TYPE LIBRARY)
-include(CPack)
+#c4_set_default_pack_properties(TYPE LIBRARY)
+#include(CPack)
