--- jsonnet-0.22.0.orig/third_party/rapidyaml/rapidyaml-0.10.0.hpp	2026-03-24 17:14:29.000000000 +0300
+++ jsonnet-0.22.0/third_party/rapidyaml/rapidyaml-0.10.0.hpp	2026-03-24 22:06:07.858760100 +0300
@@ -8173,13 +8173,13 @@
     return answer;
   }
   value128 answer;
-#if defined(_M_ARM64) && !defined(__MINGW32__)
+#if defined(_M_ARM64) && !defined(__GNUC__)
   // ARM64 has native support for 64-bit multiplications, no need to emulate
   // But MinGW on ARM64 doesn't have native support for 64-bit multiplications
   answer.high = __umulh(a, b);
   answer.low = a * b;
 #elif defined(FASTFLOAT_32BIT) ||                                              \
-    (defined(_WIN64) && !defined(__clang__) && !defined(_M_ARM64))
+    (defined(_WIN64) && !defined(__clang__) && defined(_MSC_VER) && !defined(_M_ARM64))
   answer.low = _umul128(a, b, &answer.high); // _umul128 not available on ARM64
 #elif defined(FASTFLOAT_64BIT) && defined(__SIZEOF_INT128__)
   __uint128_t r = ((__uint128_t)a) * b;
@@ -20890,6 +20890,10 @@
 #    undef _c4_ARM_
 #    undef _ARM_
 #endif
+#ifdef _c4_ARM64_
+#    undef _c4_ARM64_
+#    undef _ARM64_
+#endif
 
 #ifdef _c4_NOMINMAX
 #    undef _c4_NOMINMAX
@@ -21352,7 +21356,7 @@
 #if defined(C4_MSVC) || defined(C4_MINGW)
 //included above:
 //#include <malloc.h>
-#else
+#elif !defined(__clang__)
 #include <alloca.h>
 #endif
 
