--- libpeas-1.10.0/libpeas/Makefile.am.orig	2014-02-09 23:55:50.000000000 +0400
+++ libpeas-1.10.0/libpeas/Makefile.am	2014-05-18 21:09:07.722800000 +0400
@@ -88,6 +88,7 @@
 if HAVE_INTROSPECTION
   -include $(INTROSPECTION_MAKEFILE)
   INTROSPECTION_GIRS = Peas-1.0.gir
+  INTROSPECTION_SCANNER_ENV = CC="$(CC)"
   INTROSPECTION_SCANNER_ARGS = \
 	-I$(top_srcdir) 		\
 	--c-include=libpeas/peas.h	\
--- libpeas-1.10.0/libpeas-gtk/Makefile.am.orig	1970-01-01 03:00:00.000000000 +0300
+++ libpeas-1.10.0/libpeas-gtk/Makefile.am	2014-05-18 21:09:34.204800000 +0400
@@ -47,6 +47,7 @@
 if HAVE_INTROSPECTION
   -include $(INTROSPECTION_MAKEFILE)
   INTROSPECTION_GIRS = PeasGtk-1.0.gir
+  INTROSPECTION_SCANNER_ENV = CC="$(CC)"
   INTROSPECTION_SCANNER_ARGS = \
 	--add-include-path=$(top_builddir)/libpeas	\
 	--c-include=libpeas-gtk/peas-gtk.h		\
--- libpeas-1.10.0/tests/libpeas/introspection/Makefile.am.orig	1970-01-01 03:00:00.000000000 +0300
+++ libpeas-1.10.0/tests/libpeas/introspection/Makefile.am	2014-05-18 21:09:34.204800000 +0400
@@ -31,6 +31,7 @@
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS = Introspection-1.0.gir
 INTROSPECTION_TYPELIBS = $(INTROSPECTION_GIRS:.gir=.typelib)
+  INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 INTROSPECTION_SCANNER_ARGS = \
 	--add-include-path=$(top_builddir)/libpeas	\
 	--warn-all
