diff --git a/loaders/python/peas-python-internal.py b/loaders/python/peas-python-internal.py
index 1f96ee1..029648f 100644
--- a/loaders/python/peas-python-internal.py
+++ b/loaders/python/peas-python-internal.py
@@ -48,7 +48,6 @@ class Hooks(object):
         sys.path.insert(0, PEAS_PYEXECDIR)
         gettext.install(GETTEXT_PACKAGE, PEAS_LOCALEDIR)
 
-        self.__idle_gc = 0
         self.__module_cache = {}
         self.__extension_cache = {}
 
@@ -152,30 +151,17 @@ class Hooks(object):
         module_gtypes[gtype] = None
         return None
 
-    def __run_gc(self):
-        gc.collect()
-
-        self.__idle_gc = 0
-        return False
-
     def garbage_collect(self):
         # We run the GC right now and we schedule
         # a further collection in the main loop
         gc.collect()
 
-        if self.__idle_gc == 0:
-            self.__idle_gc = GLib.idle_add(self.__run_gc)
-            GLib.source_set_name_by_id(self.__idle_gc, '[libpeas] run_gc')
-
     def all_plugins_unloaded(self):
         pass
 
     def exit(self):
         gc.collect()
 
-        if self.__idle_gc != 0:
-            GLib.source_remove(self.__idle_gc)
-
 
 if os.getenv('PEAS_PYTHON_PROFILE') is not None:
     import cProfile
