diff -Nur libpeas-1.24.0-orig/libpeas/meson.build libpeas-1.24.0/libpeas/meson.build
--- libpeas-1.24.0-orig/libpeas/meson.build	2019-09-10 02:41:29.000000000 +0200
+++ libpeas-1.24.0/libpeas/meson.build	2019-10-25 12:33:22.261464900 +0200
@@ -94,7 +94,6 @@
   c_args: libpeas_c_args + hidden_visibility_args,
   link_args: libpeas_link_args,
   install: true,
-  install_dir: libdir,
 )
 
 libpeas_dep_sources = [
diff -Nur libpeas-1.24.0-orig/libpeas-gtk/meson.build libpeas-1.24.0/libpeas-gtk/meson.build
--- libpeas-1.24.0-orig/libpeas-gtk/meson.build	2019-09-10 02:41:29.000000000 +0200
+++ libpeas-1.24.0/libpeas-gtk/meson.build	2019-10-25 11:17:22.192590200 +0200
@@ -61,7 +61,6 @@
   c_args: libpeas_gtk_c_args,
   link_args: libpeas_gtk_link_args,
   install: true,
-  install_dir: libdir,
 )
 
 libpeas_gtk_dep_sources = []
diff -Nur libpeas-1.24.0-orig/tests/libpeas/plugins/embedded/meson.build libpeas-1.24.0/tests/libpeas/plugins/embedded/meson.build
--- libpeas-1.24.0-orig/tests/libpeas/plugins/embedded/meson.build	2019-09-10 02:41:29.000000000 +0200
+++ libpeas-1.24.0/tests/libpeas/plugins/embedded/meson.build	2019-10-25 12:14:48.227458200 +0200
@@ -35,6 +35,7 @@
   include_directories: rootdir,
   dependencies: libembedded_deps,
   install: false,
+  link_args: ['-Wl,--export-all-symbols'],
 )
 
 libembedded_dep = declare_dependency(
diff -Nur libpeas-1.24.0-orig/tests/libpeas-gtk/testing/meson.build libpeas-1.24.0/tests/libpeas-gtk/testing/meson.build
--- libpeas-1.24.0-orig/tests/libpeas-gtk/testing/meson.build	2019-09-10 02:41:29.000000000 +0200
+++ libpeas-1.24.0/tests/libpeas-gtk/testing/meson.build	2019-10-25 12:15:59.910762300 +0200
@@ -25,6 +25,7 @@
 
 libpeas_gtk_testing_link_args = [
   '-Wl,--export-dynamic',
+  '-Wl,--export-all-symbols',
 ]
 
 libpeas_gtk_testing_lib = library(
