diff -Naur VTK-7.1.0-orig/CMake/vtkCompilerExtras.cmake VTK-7.1.0/CMake/vtkCompilerExtras.cmake
--- VTK-7.1.0-orig/Common/Core/vtkConfigure.h.in	2016-01-28 17:17:38.000000000 +0100
+++ VTK-7.1.0/Common/Core/vtkConfigure.h.in	2016-09-20 19:47:27.073608100 +0200
@@ -169,7 +169,7 @@
 
 /** extern template declarations for C++11
  */
-#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1700)
+#if ((__cplusplus >= 201103L) || (defined(_MSC_VER) && _MSC_VER >= 1700)) && !defined(__MINGW32__)
 #define VTK_USE_EXTERN_TEMPLATE
 #endif
 
