--- a/Utilities/Python/CMakeLists.txt
+++ b/Utilities/Python/CMakeLists.txt
@@ -68,7 +68,16 @@
 set(vtk_python_version "${Python${VTK_PYTHON_VERSION}_VERSION}")
 set(vtk_python_version_pair "${Python${VTK_PYTHON_VERSION}_VERSION_MAJOR}.${Python${VTK_PYTHON_VERSION}_VERSION_MINOR}")
 
-set_property(GLOBAL PROPERTY _vtk_python_soabi "${Python${VTK_PYTHON_VERSION}_SOABI}")
+if(MINGW)
+  execute_process( COMMAND ${Python3_EXECUTABLE} -c "import sysconfig; print(sysconfig.get_config_var('EXT_SUFFIX'))"
+    OUTPUT_VARIABLE _Python3_EXT_SUFFIX
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  string (REGEX REPLACE "^[.-](.+)\\.pyd$" "\\1" _Python3_EXT_SUFFIX "${_Python3_EXT_SUFFIX}")
+  set_property(GLOBAL PROPERTY _vtk_python_soabi "${_Python3_EXT_SUFFIX}")
+else()
+  set_property(GLOBAL PROPERTY _vtk_python_soabi "${Python${VTK_PYTHON_VERSION}_SOABI}")
+endif()
 set_property(GLOBAL PROPERTY _vtk_python_version_major "${Python${VTK_PYTHON_VERSION}_VERSION_MAJOR}")
 set_property(GLOBAL PROPERTY _vtk_python_version_minor "${Python${VTK_PYTHON_VERSION}_VERSION_MINOR}")
 
