--- a/CMake/FindODBC.cmake
+++ b/CMake/FindODBC.cmake
@@ -8,7 +8,7 @@
 
 # No .pc files are shipped with ODBC on Windows.
 set(_ODBC_use_pkgconfig 0)
-if (NOT WIN32)
+if (NOT MSVC)
   find_package(PkgConfig QUIET)
   if (PkgConfig_FOUND)
     set(_ODBC_use_pkgconfig 1)
@@ -36,6 +36,10 @@
       add_library(ODBC::ODBC INTERFACE IMPORTED)
       target_link_libraries(ODBC::ODBC
         INTERFACE "PkgConfig::${_odbc_target}")
+      if(MINGW AND (${_odbc_target} STREQUAL "_unixodbc"))
+        set_target_properties(ODBC::ODBC PROPERTIES
+          INTERFACE_COMPILE_DEFINITIONS SQL_WCHART_CONVERT)
+      endif()
     endif ()
   endif ()
   unset(_odbc_target)
