--- a/ThirdParty/ioss/vtkioss/Ioss_IOFactory.C
+++ b/ThirdParty/ioss/vtkioss/Ioss_IOFactory.C
@@ -12,6 +12,7 @@
 #include <cstddef> // for nullptr
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <map>     // for _Rb_tree_iterator, etc
 #include <ostream> // for basic_ostream, etc
 #include <set>
--- a/ThirdParty/ioss/vtkioss/Ioss_StructuredBlock.C
+++ b/ThirdParty/ioss/vtkioss/Ioss_StructuredBlock.C
@@ -15,6 +15,7 @@
 #include <Ioss_StructuredBlock.h>
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 
 #include <cstddef> // for size_t
 #include <numeric>
--- a/ThirdParty/ioss/vtkioss/Ioss_Utils.C
+++ b/ThirdParty/ioss/vtkioss/Ioss_Utils.C
@@ -21,6 +21,7 @@
 #include VTK_FMT(fmt/chrono.h)
 #include VTK_FMT(fmt/format.h)
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <fstream>
 #include <sstream>
 #include <string>
--- a/ThirdParty/ioss/vtkioss/Ioss_ZoneConnectivity.C
+++ b/ThirdParty/ioss/vtkioss/Ioss_ZoneConnectivity.C
@@ -9,6 +9,7 @@
 #include <cstddef> // for size_t
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <string> // for string
 #include <vector> // for vector
 
