Include missing header

diff -urN VTK-9.6.1/Wrapping/Tools/vtkParseSystem.c.orig VTK-9.6.1/Wrapping/Tools/vtkParseSystem.c
--- VTK-9.6.1/Wrapping/Tools/vtkParseSystem.c.orig	2026-03-26 17:23:37.000000000 +0100
+++ VTK-9.6.1/Wrapping/Tools/vtkParseSystem.c	2026-05-03 13:28:02.903638500 +0200
@@ -10,6 +10,7 @@
 
 #ifdef _WIN32
 #include <stddef.h>
+#include <errno.h>
 #include <windows.h>
 #else
 #include <dirent.h>
