From 83f7f1e7d321b9ab674e47bcd18d7e6770aacb81 Mon Sep 17 00:00:00 2001
From: Mehdi Chinoune <mehdi.chinoune@hotmail.com>
Date: Sat, 26 Mar 2022 22:51:04 +0100
Subject: [PATCH] Build GUISupportMFC only with MSVC

---
 GUISupport/MFC/vtk.module | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GUISupport/MFC/vtk.module b/GUISupport/MFC/vtk.module
index 77288e37a4..ae471d3d8e 100644
--- a/GUISupport/MFC/vtk.module
+++ b/GUISupport/MFC/vtk.module
@@ -3,7 +3,7 @@ NAME
 LIBRARY_NAME
   vtkGUISupportMFC
 CONDITION
-  WIN32
+  MSVC
 DEPENDS
   # If dependencies are added, update the list in the `CMakeLists.txt`.
   VTK::RenderingOpenGL2
-- 
2.35.1.windows.2

