--- VTK-8.1.2/Rendering/OpenGL2/vtkWin32RenderWindowInteractor.cxx.orig	2018-11-28 07:52:31.531104700 +0300
+++ VTK-8.1.2/Rendering/OpenGL2/vtkWin32RenderWindowInteractor.cxx	2018-11-28 07:53:04.823563300 +0300
@@ -56,6 +56,7 @@
 // so we define them and then do a runtime checks and function pointers
 // to avoid a link requirement on Windows 7
 #define MOUSEEVENTF_FROMTOUCH 0xFF515700
+#if _WIN32_WINNT < 0x601
 #define WM_TOUCH              0x0240
 #define TOUCH_COORD_TO_PIXEL(l)  ((l) / 100)
 typedef struct _TOUCHINPUT {
@@ -75,6 +76,7 @@
 #define TOUCHEVENTF_MOVE  0x0001
 #define TOUCHEVENTF_DOWN  0x0002
 #define TOUCHEVENTF_UP  0x0004
+#endif
 typedef bool (WINAPI *RegisterTouchWindowType)(HWND,ULONG);
 typedef bool (WINAPI *GetTouchInputInfoType)(HTOUCHINPUT,UINT,PTOUCHINPUT,int);
 typedef bool (WINAPI *CloseTouchInputHandleType)(HTOUCHINPUT);
