Boost.Math requires C++14

diff -urN VTK-9.4.1/CMake/vtkCompilerChecks.cmake.orig VTK-9.4.1/CMake/vtkCompilerChecks.cmake
--- VTK-9.4.1/CMake/vtkCompilerChecks.cmake.orig	2024-12-27 18:08:14.000000000 +0100
+++ VTK-9.4.1/CMake/vtkCompilerChecks.cmake	2025-01-14 15:41:43.252214400 +0100
@@ -33,7 +33,7 @@
   # Needed to make sure libraries and executables not built by the
   # vtkModuleMacros still have the C++11 compiler flags enabled
   # Wrap this in an escape hatch for unknown compilers
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 14)
   set(CMAKE_CXX_STANDARD_REQUIRED True)
   set(CMAKE_CXX_EXTENSIONS False)
 endif()
