diff -urN VTK-8.2.0-orig/ThirdParty/glew/vtkglew/CMakeLists.txt VTK-8.2.0/ThirdParty/glew/vtkglew/CMakeLists.txt
--- VTK-8.2.0-orig/ThirdParty/glew/vtkglew/CMakeLists.txt	2019-01-31 01:15:13.000000000 +0800
+++ VTK-8.2.0/ThirdParty/glew/vtkglew/CMakeLists.txt	2022-01-14 21:14:21.355404200 +0800
@@ -44,7 +44,7 @@
       # https://github.com/nigels-com/glew/issues/157
       target_link_libraries(vtkglew
         PRIVATE
-          -nostdlib)
+          -no-undefined)
     endif()
   endif ()
 else ()
