diff -Naur a/IO/Video/vtkWin32VideoSource.cxx b/IO/Video/vtkWin32VideoSource.cxx
--- a/IO/Video/vtkWin32VideoSource.cxx	2015-03-03 06:56:42.000000000 +0600
+++ b/IO/Video/vtkWin32VideoSource.cxx	2015-03-04 18:47:05.859427800 +0600
@@ -349,7 +349,7 @@
     }
 
   // set user data for callbacks
-  if (!capSetUserData(this->Internal->CapWnd,(long)this))
+  if (!capSetUserData(this->Internal->CapWnd,(vtkLONG)this))
     {
     vtkErrorMacro(<< "Initialize: couldn't set user data for callback"\
                     << " (" << GetLastError() << ")");
