--- a/tests/test_virtualenv.py
+++ b/tests/test_virtualenv.py
@@ -20,6 +20,7 @@ import re
 import shutil
 import subprocess
 import sys
+import sysconfig
 import types
 from importlib import metadata
 from pathlib import Path
@@ -39,6 +40,7 @@ if TYPE_CHECKING:
     from nox.virtualenv import CondaEnv, ProcessEnv, VirtualEnv
 
 IS_WINDOWS = sys.platform.startswith("win")
+IS_MINGW = sysconfig.get_platform().startswith("mingw")
 HAS_UV = shutil.which("uv") is not None
 RAISE_ERROR = "RAISE_ERROR"
 VIRTUALENV_VERSION = metadata.version("virtualenv")
@@ -591,10 +593,11 @@ def test_bin_paths(
     assert len(venv.bin_paths) == 1
     assert venv.bin_paths[0] == venv.bin
 
-    assert str(dir_.joinpath("Scripts" if IS_WINDOWS else "bin")) == venv.bin
+    assert str(dir_.joinpath("Scripts" if IS_WINDOWS and not IS_MINGW else "bin")) == venv.bin
 
 
 @mock.patch("nox.virtualenv._PLATFORM", new="win32")
+@mock.patch("nox.virtualenv._IS_MINGW", new=False)
 def test_bin_windows(
     make_one: Callable[..., tuple[VirtualEnv | ProcessEnv, Path]],
 ) -> None:
@@ -630,14 +633,14 @@ def test_create(
     assert venv.env["CONDA_PREFIX"] is None
     assert "NOT_CONDA_PREFIX" not in venv.env
 
-    if IS_WINDOWS:
+    if IS_WINDOWS and not IS_MINGW:
         assert dir_.joinpath("Scripts", "python.exe").exists()
         assert dir_.joinpath("Scripts", "pip.exe").exists()
         assert dir_.joinpath("Lib").exists()
         assert str(dir_.joinpath("Scripts")) in venv.bin_paths
     else:
-        assert dir_.joinpath("bin", "python").exists()
-        assert dir_.joinpath("bin", "pip").exists()
+        assert dir_.joinpath("bin", "python" if not IS_WINDOWS else "python.exe").exists()
+        assert dir_.joinpath("bin", "pip" if not IS_WINDOWS else "pip.exe").exists()
         assert dir_.joinpath("lib").exists()
         assert str(dir_.joinpath("bin")) in venv.bin_paths
 
