--- a/CMakeLists.txt
+++ a/CMakeLists.txt
@@ -17,5 +17,7 @@
 target_compile_definitions( ptest PRIVATE PCUBATURE=1 )
 
 include(GNUInstallDirs)
-install( TARGETS cubature DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+install( TARGETS cubature RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+                          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+                          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} )
 install( FILES cubature.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} )
