From c06828c9181985cd94b72b34e25d10991065b180 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sun, 15 Oct 2023 18:57:51 -0400
Subject: Add include of "cstdint"

---
 include/wxSVG/mediadec_ffmpeg.h | 1 +
 1 files changed, 1 insertions(+)

diff --git a/include/wxSVG/mediadec_ffmpeg.h b/include/wxSVG/mediadec_ffmpeg.h
index 706d495..8f55af3 100644
--- a/include/wxSVG/mediadec_ffmpeg.h
+++ b/include/wxSVG/mediadec_ffmpeg.h
@@ -15,6 +15,7 @@
 #include <wx/image.h>
 #include <vector>
 #include <map>
+#include <cstdint>
 using namespace std;
 
 struct AVFormatContext;
-- 
