--- wxsvg-1.5.19/libwxsvg.pc.in.orig	2019-08-01 17:53:44.236248500 +0000
+++ wxsvg-1.5.19/libwxsvg.pc.in	2019-08-01 17:57:21.174242500 +0000
@@ -4,8 +4,9 @@
 includedir=@includedir@
 
 Name: libwxsvg
-Description: C++ library to create, manipulate and render SVG files 
+Description: C++ library to create, manipulate and render SVG files
 Requires:
+Requires.private: pango, libexif
 Version: @VERSION@
 Libs: -L${libdir} -lwxsvg
 Cflags: -I${includedir}
