--- MySQL-python-1.2.5/setup_windows.py.orig	2018-07-15 17:06:49.215182100 -0400
+++ MySQL-python-1.2.5/setup_windows.py	2018-07-15 17:15:06.862619600 -0400
@@ -17,10 +17,10 @@ def get_config():
     else:
         client = "mysqlclient"
 
-    library_dirs = [ os.path.join(connector, r'lib\opt') ]
+    library_dirs = [ os.path.join(connector, r'lib') ]
     libraries = [ 'kernel32', 'advapi32', 'wsock32', client ]
-    include_dirs = [ os.path.join(connector, r'include') ]
-    extra_compile_args = [ '/Zl' ]
+    include_dirs = [ os.path.join(connector, r'include/mysql') ]
+    extra_compile_args = []
     
     name = "MySQL-python"
     if enabled(options, 'embedded'):
--- MySQL-python-1.2.5/site.cfg.orig	2018-07-15 17:06:49.215182100 -0400
+++ MySQL-python-1.2.5/site.cfg	2018-07-15 17:15:06.862619600 -0400
@@ -14,4 +14,3 @@
 
 # http://stackoverflow.com/questions/1972259/mysql-python-install-problem-using-virtualenv-windows-pip
 # Windows connector libs for MySQL. You need a 32-bit connector for your 32-bit Python build.
-connector = C:\Program Files (x86)\MySQL\MySQL Connector C 6.0.2
