--- MySQL-python-1.2.5/setup_windows.py.orig	2018-07-15 17:06:49.215182100 -0400
+++ MySQL-python-1.2.5/setup_windows.py	2018-07-15 17:15:06.862619600 -0400
@@ -17,7 +17,7 @@ def get_config():
     library_dirs = [ os.path.join(connector, r'lib\opt') ]
     libraries = [ 'kernel32', 'advapi32', 'wsock32', client ]
     include_dirs = [ os.path.join(connector, r'include') ]
-    extra_compile_args = [ '/Zl' ]
+    extra_compile_args = [ '-Z' ]
     
     name = "MySQL-python"
     if enabled(options, 'embedded'):
