--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -382,10 +382,11 @@
 string(TIMESTAMP BUILD_DATE "%Y%m%d")
 target_compile_definitions(${PROJECT_NAME} PRIVATE BUILD_DATE=\"${BUILD_DATE}\"
                                                    BUILD_VERSION=\"${GIT_COMMIT_HASH}\")
-if (UNIX)
+if (UNIX OR MINGW)
     # Compile-time protection against static sized buffer overflows.
     target_compile_definitions(${PROJECT_NAME} PRIVATE "_FORTIFY_SOURCE=2")
-else()
+endif()
+if(WIN32)
     target_compile_definitions(${PROJECT_NAME} PRIVATE NOMINMAX)
     if (NOT CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
         target_compile_definitions(${PROJECT_NAME} PRIVATE NDEBUG)
@@ -471,13 +472,15 @@
 endif()
 
 # Link options
-if (WIN32)
+if (MSVC)
     target_link_options(${PROJECT_NAME} PRIVATE /DYNAMICBASE)
     # Control flow guard
     target_link_options(${PROJECT_NAME} PRIVATE /GUARD:CF)
     if (MSVC AND NOT CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
         target_link_options(${PROJECT_NAME} PUBLIC /OPT:REF /OPT:ICF)
     endif()
+elseif (MINGW)
+    target_link_options(${PROJECT_NAME} PRIVATE "-fstack-protector")
 elseif (APPLE)
 else()
     # Link options for security hardening.
