--- h11-0.14.0/docs/source/conf.py	2022-09-25 18:32:15.000000000 +0300
+++ python-build-MINGW32/docs/source/conf.py	2023-05-15 21:56:49.943419000 +0300
@@ -59,9 +59,14 @@
 # Undocumented trick: if we def setup here in conf.py, it gets called just
 # like an extension's setup function.
 def setup(app):
-    app.add_javascript("show-code.js")
-    app.add_javascript("facebox.js")
-    app.add_stylesheet("facebox.css")
+    try:
+        app.add_javascript("show-code.js")
+        app.add_javascript("facebox.js")
+        app.add_stylesheet("facebox.css")
+    except AttributeError:
+        app.add_js_file("show-code.js")
+        app.add_js_file("facebox.js")
+        app.add_css_file("facebox.css")
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
