diff --git a/libraw/libraw.h b/libraw/libraw.h
index 50d4276d43..d4363fe01e 100644
--- a/libraw/libraw.h
+++ b/libraw/libraw.h
@@ -55,8 +55,8 @@ it under the terms of the one of two licenses as you choose:
 #   define LIBRAW_WIN32_CALLS
 #  endif
 
-/* DLLs: Microsoft or Intel compiler */
-# if defined(_MSC_VER) || defined(__INTEL_COMPILER)
+/* DLLs: MinGW, Microsoft or Intel compiler */
+# if defined(__GNUC__) || defined(_MSC_VER) || defined(__INTEL_COMPILER)
 # ifndef LIBRAW_WIN32_DLLDEFS
 #  define LIBRAW_WIN32_DLLDEFS
 # endif
