diff -urN parmetis-4.0.3.orig/metis/GKlib/gk_arch.h parmetis-4.0.3/metis/GKlib/gk_arch.h
--- parmetis-4.0.3.orig/metis/GKlib/gk_arch.h	2013-03-30 08:24:50.000000000 -0700
+++ parmetis-4.0.3/metis/GKlib/gk_arch.h	2019-11-13 22:02:07.318990500 -0800
@@ -41,7 +41,9 @@
 #endif
   #include <inttypes.h>
   #include <sys/types.h>
+#ifndef __MINGW32__
   #include <sys/resource.h>
+#endif
   #include <sys/time.h>
 #endif
 
