diff -urN parmetis-4.0.3.orig/CMakeLists.txt parmetis-4.0.3/CMakeLists.txt
--- parmetis-4.0.3.orig/CMakeLists.txt	2013-03-30 08:24:50.000000000 -0700
+++ parmetis-4.0.3/CMakeLists.txt	2019-11-12 00:35:22.652916600 -0800
@@ -9,12 +9,11 @@
 execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${METIS_PATH} metis)
 
 # Search for MPI.
-# GK commented this out as it seems to be creating problems
-# include(FindMPI)
-# if(NOT MPI_FOUND)
-#   message(FATAL_ERROR "mpi is not found")
-# endif()
-# set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MPI_COMPILE_FLAGS}")
+include(FindMPI)
+if(NOT MPI_FOUND)
+  message(FATAL_ERROR "mpi is not found")
+endif()
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MPI_COMPILE_FLAGS}")
 
 # Prepare libraries.
 if(SHARED)
