From dd5fa78166a2447fbf0a6c10990a622fb328b55e Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 15 Apr 2022 17:00:39 +0530
Subject: [PATCH] Don't download prebuild gn and emsdk

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 tools/git-sync-deps | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/tools/git-sync-deps b/tools/git-sync-deps
index 44d936b4..1b94bcde 100755
--- a/tools/git-sync-deps
+++ b/tools/git-sync-deps
@@ -259,12 +259,12 @@ def main(argv):
     return 1
 
   git_sync_deps(deps_file_path, argv, verbose)
-  subprocess.check_call(
-      [sys.executable,
-       os.path.join(os.path.dirname(deps_file_path), 'bin', 'fetch-gn')])
-  subprocess.check_call(
-      [sys.executable,
-       os.path.join(os.path.dirname(deps_file_path), 'bin', 'activate-emsdk')])
+  #subprocess.check_call(
+  #    [sys.executable,
+  #     os.path.join(os.path.dirname(deps_file_path), 'bin', 'fetch-gn')])
+  #subprocess.check_call(
+  #    [sys.executable,
+  #     os.path.join(os.path.dirname(deps_file_path), 'bin', 'activate-emsdk')])
   return 0
 
 
-- 
2.35.2.windows.1

