diff --git a/BUILD.gn b/BUILD.gn
index 4e97f10134..912a3d4315 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1595,6 +1595,30 @@ skia_component("skia") {
     sources += [ "src/ports/SkDebug_stdio.cpp" ]
   }
 
+  if (is_mingw) {
+    if (is_component_build && !skia_enable_svg) {
+      import("modules/skresources/skresources.gni")
+      import("modules/skshaper/skshaper.gni")
+      import("modules/svg/svg.gni")
+
+      include_dirs = [
+        "modules/svg/include",
+        "modules/skshaper/include",
+        "modules/skresources/include",
+      ]
+      public += skia_shaper_public
+      public += skia_skresources_public
+      public += skia_svg_public
+      sources += skia_shaper_primitive_sources
+      sources += skia_skresources_sources
+      sources += skia_svg_sources
+      defines += [
+        "SK_ENABLE_SVG",
+        "SK_SHAPER_PRIMITIVE_AVAILABLE",
+      ]
+    }
+  }
+
   if (skia_enable_spirv_validation) {
     deps += [ "//third_party/externals/spirv-tools:spvtools_val" ]
     defines += [ "SK_ENABLE_SPIRV_VALIDATION" ]
