--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1612,10 +1612,6 @@
   }
 }
 
-group("experimental") {
-  deps = [ "experimental/sktext" ]
-}
-
 config("our_vulkan_headers") {
   include_dirs = [ "include/third_party/vulkan" ]
 }
@@ -2162,7 +2158,6 @@
       ":skia",
       ":test",
       ":tool_utils",
-      "experimental/sktext:tests",
       "modules/skottie:tests",
       "modules/skparagraph:tests",
       "modules/sksg:tests",
@@ -2222,7 +2217,6 @@
       ":flags",
       ":gpu_tool_utils",
       ":xml",
-      "experimental/sktext:samples",
       "modules/audioplayer",
       "modules/skparagraph:samples",
       "modules/skshaper",
@@ -2651,7 +2645,6 @@
       deps = [
         ":sk_app",
         ":skia",
-        "experimental/sktext:sktext",
         "modules/skottie:skottie",
       ]
     }
@@ -2899,7 +2892,6 @@
         ":skia",
         ":tool_utils",
         ":trace",
-        "experimental/sktext",
         "modules/audioplayer",
         "modules/particles",
         "modules/skottie",
@@ -2968,10 +2960,6 @@
       is_shared_library = is_android
       deps = [ "modules/skplaintexteditor:editor_app" ]
     }
-    test_app("text_editor") {
-      is_shared_library = is_android
-      deps = [ "experimental/sktext:text_editor" ]
-    }
   }
 
   skia_executable("image_diff_metric") {
