diff -bur skia-orig/src/ports/SkScalerContext_win_dw.cpp skia/src/ports/SkScalerContext_win_dw.cpp
--- skia-orig/src/ports/SkScalerContext_win_dw.cpp	2026-01-17 16:03:50.166657600 -0700
+++ skia/src/ports/SkScalerContext_win_dw.cpp	2026-01-17 16:13:48.937880400 -0700
@@ -626,11 +626,11 @@
     auto drawChildren = [&](uint32_t childCount) -> bool {
         if (childCount != 0) {
             DWRITE_PAINT_ELEMENT childElement;
-            HRB(reader.MoveToFirstChild(&childElement));
+            HRB(reader.MoveToFirstChild(&childElement, sizeof(DWRITE_PAINT_ELEMENT)));
             this->drawColorV1Paint(canvas, reader, childElement);
 
             for (uint32_t i = 1; i < childCount; i++) {
-                HRB(reader.MoveToNextSibling(&childElement));
+                HRB(reader.MoveToNextSibling(&childElement, sizeof(DWRITE_PAINT_ELEMENT)));
                 this->drawColorV1Paint(canvas, reader, childElement);
             }
 
@@ -1182,13 +1182,13 @@
         DWRITE_PAINT_ELEMENT sourceElement;
         DWRITE_PAINT_ELEMENT backdropElement;
 
-        HRBM(reader.MoveToFirstChild(&sourceElement), "Could not move to child.");
-        HRBM(reader.MoveToNextSibling(&backdropElement), "Could not move to sibiling.");
+        HRBM(reader.MoveToFirstChild(&sourceElement, sizeof(DWRITE_PAINT_ELEMENT)), "Could not move to child.");
+        HRBM(reader.MoveToNextSibling(&backdropElement, sizeof(DWRITE_PAINT_ELEMENT)), "Could not move to sibiling.");
         canvas.saveLayer(nullptr, nullptr);
         this->drawColorV1Paint(canvas, reader, backdropElement);
 
         HRBM(reader.MoveToParent(), "Could not move to parent.");
-        HRBM(reader.MoveToFirstChild(&sourceElement), "Could not move to child.");
+        HRBM(reader.MoveToFirstChild(&sourceElement, sizeof(DWRITE_PAINT_ELEMENT)), "Could not move to child.");
         canvas.saveLayer(nullptr, &blendModePaint);
         this->drawColorV1Paint(canvas, reader, sourceElement);
 
@@ -1219,7 +1219,7 @@
     DWRITE_PAINT_ELEMENT paintElement;
     D2D_RECT_F clipBox;
     DWRITE_PAINT_ATTRIBUTES attributes;
-    HRBM(paintReader->SetCurrentGlyph(glyphIndex, &paintElement, &clipBox, &attributes),
+    HRBM(paintReader->SetCurrentGlyph(glyphIndex, &paintElement, sizeof(DWRITE_PAINT_ELEMENT), &clipBox, &attributes),
          "Could not set current glyph.");
 
     if (paintElement.paintType == DWRITE_PAINT_TYPE_NONE) {
@@ -1248,7 +1248,8 @@
 
     // The DirectWrite interface returns resolved colors if these are provided.
     // Indexes and alphas are reported but there is no reason to duplicate the color calculation.
-    paintReader->SetTextColor(dw_color_from(SkColor4f::FromColor(fRec.fForegroundColor)));
+    auto dwColor = dw_color_from(SkColor4f::FromColor(fRec.fForegroundColor));
+    paintReader->SetTextColor(&dwColor);
     paintReader->SetCustomColorPalette(typeface->fDWPalette.get(), typeface->fPaletteEntryCount);
 
     return this->drawColorV1Paint(canvas, *paintReader, paintElement);
@@ -1285,11 +1286,11 @@
             return true;
         }
         DWRITE_PAINT_ELEMENT childElement;
-        HRB(reader.MoveToFirstChild(&childElement));
+        HRB(reader.MoveToFirstChild(&childElement, sizeof(DWRITE_PAINT_ELEMENT)));
         this->generateColorV1PaintBounds(ctm, bounds, reader, childElement);
 
         for (uint32_t i = 1; i < childCount; ++i) {
-            HRB(reader.MoveToNextSibling(&childElement));
+            HRB(reader.MoveToNextSibling(&childElement, sizeof(DWRITE_PAINT_ELEMENT)));
             this->generateColorV1PaintBounds(ctm, bounds, reader, childElement);
         }
 
@@ -1435,7 +1436,7 @@
     D2D_RECT_F clipBox;
     DWRITE_PAINT_ATTRIBUTES attributes;
     // If the glyph is not color this will succeed but return paintType NONE.
-    HRBM(paintReader->SetCurrentGlyph(glyphIndex, &paintElement, &clipBox, &attributes),
+    HRBM(paintReader->SetCurrentGlyph(glyphIndex, &paintElement, sizeof(DWRITE_PAINT_ELEMENT), &clipBox, &attributes),
          "Could not set the current glyph.");
 
     if (paintElement.paintType == DWRITE_PAINT_TYPE_NONE) {
