diff -bur skia-orig/BUILD.gn skia/BUILD.gn
--- skia-orig/BUILD.gn	2026-01-17 23:22:07.878927200 -0700
+++ skia/BUILD.gn	2026-01-17 23:24:34.717081500 -0700
@@ -993,9 +993,7 @@
         "src/gpu/ganesh/gl/win/GrGLMakeNativeInterface_win.cpp",
         "src/gpu/ganesh/gl/win/GrGLMakeWinInterface.cpp",
       ]
-      if (target_cpu != "arm64") {
         libs += [ "opengl32" ]
-      }
     } else {
       sources += [ "src/gpu/ganesh/gl/GrGLMakeNativeInterface_none.cpp" ]
     }
@@ -2173,11 +2171,9 @@
             "tools/ganesh/gl/win/SkWGL_win.cpp",
           ]
           libs += [ "gdi32" ]
-          if (target_cpu != "arm64") {
             libs += [ "opengl32" ]
           }
         }
-      }
 
       if (skia_use_vulkan) {
         sources += [ "tools/gpu/vk/VkTestHelper.h" ]
