From e9b2e5e2c4e09a896e2ef9b6dc3d28d721fa21f5 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Tue, 30 Nov 2021 15:30:24 +0530
Subject: [PATCH] remove as-needed from ldflags

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 build/gen.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/gen.py b/build/gen.py
index 922bf46f..edcc0353 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -375,7 +375,8 @@ def WriteGNNinja(path, platform, host, options):
     ])
 
     if platform.is_linux() or platform.is_mingw() or platform.is_msys():
-      ldflags.append('-Wl,--as-needed')
+      if not platform.is_mingw():
+        ldflags.append('-Wl,--as-needed')
 
       if not options.no_static_libstdcpp:
         ldflags.append('-static-libstdc++')
-- 
2.34.1.windows.1

