--- mingw-w64/mingw-w64-tools/genstubdll-old/Makefile.am	1969-12-31 19:00:00 
+++ mingw-w64/mingw-w64-tools/genstubdll/Makefile.am	2017-05-31 05:36:00 -0400
@@ -0,0 +1,7 @@
+bin_PROGRAMS = genstubdll
+genstubdll_SOURCES = src/token.h src/lexdef.c src/output.c
+genstubdll_CPPFLAGS = -Isrc
+genstubdll_CFLAGS = $(AM_CFLAGS) -O3 -g -Wall -Wextra
+
+EXTRA_DIST = $(srcdir)/COPYING
+DISTCHECK_CONFIGURE_FLAGS = --host=$(host)
--- mingw-w64/mingw-w64-tools/genstubdll-old/configure.ac	1969-12-31 19:00:00 
+++ mingw-w64/mingw-w64-tools/genstubdll/configure.ac	2017-05-31 04:46:50 -0400
@@ -0,0 +1,39 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([mingw-w64-genstubdll],[1.0],[mingw-w64-public@lists.sourceforge.net])
+AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_SRCDIR([src/output.c])
+AC_CONFIG_HEADERS([config.h])
+
+AM_INIT_AUTOMAKE([foreign subdir-objects])
+AM_MAINTAINER_MODE
+
+AC_CANONICAL_HOST
+
+# Checks for programs.
+AC_PROG_CC
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([malloc.h memory.h stdint.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memset strchr strdup strrchr strstr])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
+
