--- a/src/mumps_scotch.c
+++ b/src/mumps_scotch.c
@@ -14,6 +14,7 @@
  *
  */
 /* Interfacing with SCOTCH and pt-SCOTCH */
+#include <windows.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -130,7 +130,7 @@
 #if defined(MUMPS_WIN32)
      ierr = _putenv("SCOTCH_PTHREAD_NUMBER=");
 #else
-     unsetenv("SCOTCH_PTHREAD_NUMBER");
+     SetEnvironmentVariable("SCOTCH_PTHREAD_NUMBER", NULL);
 #endif
   }
   else
@@ -140,7 +140,7 @@
     ierr = _putenv(param);
 #else
     sprintf(param, "%d", *PTHREAD_NUMBER);
-    setenv("SCOTCH_PTHREAD_NUMBER",param,1);
+    SetEnvironmentVariable("SCOTCH_PTHREAD_NUMBER", param);
 #endif
   }
 #endif
