diff -urN MUMPS_5.5.0.org/src/mumps_scotch.c MUMPS_5.5.0/src/mumps_scotch.c
--- MUMPS_5.5.0.org/src/mumps_scotch.c	2022-04-14 16:45:37.000000000 +0500
+++ MUMPS_5.5.0/src/mumps_scotch.c	2022-06-05 19:47:41.857341600 +0500
@@ -14,6 +14,7 @@
  *
  */
 /* Interfacing with SCOTCH and pt-SCOTCH */
+#include <windows.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -124,12 +125,12 @@
   char param[32];
   if (*PTHREAD_NUMBER == -1) 
   {
-     unsetenv("SCOTCH_PTHREAD_NUMBER");
+     SetEnvironmentVariable("SCOTCH_PTHREAD_NUMBER", NULL);
   }
   else
   {
     sprintf(param, "%d", *PTHREAD_NUMBER);
-    setenv("SCOTCH_PTHREAD_NUMBER",param,1);
+    SetEnvironmentVariable("SCOTCH_PTHREAD_NUMBER", param);
   }
 #endif
   return;
