diff -urN MUMPS_5.7.3.orig/include/mumps_compat.h MUMPS_5.7.3/include/mumps_compat.h
--- MUMPS_5.7.3.orig/include/mumps_compat.h	2024-07-15 16:44:26.000000000 +0500
+++ MUMPS_5.7.3/include/mumps_compat.h	2024-11-30 18:03:27.693658100 +0500
@@ -19,7 +19,7 @@
 #define MUMPS_COMPAT_H
 
 
-#if defined(_WIN32) && ! defined(__MINGW32__)
+#if 1 /*defined(_WIN32) && ! defined(__MINGW32__)*/
 # define MUMPS_WIN32 1
 #endif
 
diff -urN MUMPS_5.7.3.orig/src/mumps_c.c MUMPS_5.7.3/src/mumps_c.c
--- MUMPS_5.7.3.orig/src/mumps_c.c	2024-07-15 16:44:26.000000000 +0500
+++ MUMPS_5.7.3/src/mumps_c.c	2024-11-30 18:05:09.775659700 +0500
@@ -43,7 +43,7 @@
  *   CMUMPS_XXX in the code to get rid of any symbol convention annoyance.
  */
 #if MUMPS_ARITH == MUMPS_ARITH_s
-# if defined(UPPER) || defined(MUMPS_WIN32)
+# if 0 /*defined(UPPER) || defined(MUMPS_WIN32)*/
 #  define F_SYM_ARITH(lower_case,upper_case) SMUMPS_##upper_case
 # elif defined(Add_)
 #  define F_SYM_ARITH(lower_case,upper_case) smumps_##lower_case##_
@@ -53,7 +53,7 @@
 #  define F_SYM_ARITH(lower_case,upper_case) smumps_##lower_case
 # endif
 #elif MUMPS_ARITH == MUMPS_ARITH_d
-# if defined(UPPER) || defined(MUMPS_WIN32)
+# if 0 /*defined(UPPER) || defined(MUMPS_WIN32)*/
 #  define F_SYM_ARITH(lower_case,upper_case) DMUMPS_##upper_case
 # elif defined(Add_)
 #  define F_SYM_ARITH(lower_case,upper_case) dmumps_##lower_case##_
@@ -63,7 +63,7 @@
 #  define F_SYM_ARITH(lower_case,upper_case) dmumps_##lower_case
 # endif
 #elif MUMPS_ARITH == MUMPS_ARITH_c
-# if defined(UPPER) || defined(MUMPS_WIN32)
+# if 0 /*defined(UPPER) || defined(MUMPS_WIN32)*/
 #  define F_SYM_ARITH(lower_case,upper_case) CMUMPS_##upper_case
 # elif defined(Add_)
 #  define F_SYM_ARITH(lower_case,upper_case) cmumps_##lower_case##_
@@ -73,7 +73,7 @@
 #  define F_SYM_ARITH(lower_case,upper_case) cmumps_##lower_case
 # endif
 #elif MUMPS_ARITH == MUMPS_ARITH_z
-# if defined(UPPER) || defined(MUMPS_WIN32)
+# if 0 /*defined(UPPER) || defined(MUMPS_WIN32)*/
 #  define F_SYM_ARITH(lower_case,upper_case) ZMUMPS_##upper_case
 # elif defined(Add_)
 #  define F_SYM_ARITH(lower_case,upper_case) zmumps_##lower_case##_
diff -urN MUMPS_5.7.3.orig/src/mumps_common.h MUMPS_5.7.3/src/mumps_common.h
--- MUMPS_5.7.3.orig/src/mumps_common.h	2024-07-15 16:44:27.000000000 +0500
+++ MUMPS_5.7.3/src/mumps_common.h	2024-11-30 18:04:07.525754900 +0500
@@ -28,7 +28,7 @@
  *     knowledge, there is no way to perform the conversion with CPP
  *     directives only.
  */
-#if defined(UPPER) || defined(MUMPS_WIN32)
+#if 0 /*defined(UPPER) || defined(MUMPS_WIN32)*/
 # define F_SYMBOL(lower_case,upper_case) MUMPS_##upper_case
 #elif defined(Add_)
 # define F_SYMBOL(lower_case,upper_case) mumps_##lower_case##_
