diff -uNr mecab-0.996.orig/src/Makefile.am mecab-0.996/src/Makefile.am
--- mecab-0.996.orig/src/Makefile.am	2013-01-24 14:35:35.000000000 +0900
+++ mecab-0.996/src/Makefile.am	2019-11-07 16:16:41.036309898 +0900
@@ -3,7 +3,7 @@
 EXTRA_DIST = Makefile.msvc.in make.bat
 pkglibexecdir = ${libexecdir}/mecab
 INCLUDES = -DDIC_VERSION=$(DIC_VERSION) $(MECAB_NO_TLS) $(MECAB_USE_UTF8_ONLY) -DMECAB_DEFAULT_RC="\"$(MECAB_DEFAULT_RC)\""
-libmecab_la_LDFLAGS = -no-undefined -version-info $(LTVERSION)
+libmecab_la_LDFLAGS = -no-undefined -version-info $(LTVERSION) -liconv
 libmecab_la_SOURCES = viterbi.cpp tagger.cpp  utils.cpp utils.h eval.cpp iconv_utils.cpp iconv_utils.h \
                       dictionary_rewriter.h dictionary_rewriter.cpp dictionary_generator.cpp \
 	              dictionary_compiler.cpp context_id.h context_id.cpp \
