--- libgpg-error-1.10/potomo	2010-10-20 18:27:58 +0400
+++ libgpg-error-1.10/potomo	2011-04-03 15:05:06 +0400
@@ -57,7 +57,7 @@
         ;;   
     *)
         echo "potomo: '$infile' converting from $fromset to utf-8" >&2
-        iconv --silent --from-code=$fromset --to-code=utf-8 < "$infile" |\
+        iconv -s --from-code=$fromset --to-code=utf-8 < "$infile" |\
             sed "/^\"Content-Type:/ s/charset=[a-zA-Z0-9_-]*/charset=utf-8/"|\
             msgfmt --output-file="$outfile" -
         ;;
