diff -Naur blender-2.77a-orig/extern/glog/src/windows/port.h blender-2.77a/extern/glog/src/windows/port.h
--- blender-2.77a-orig/extern/glog/src/windows/port.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/extern/glog/src/windows/port.h	2016-07-12 13:40:19.613595000 +0300
@@ -74,17 +74,19 @@
 #pragma warning(disable:4244 4251 4355 4715 4800 4996)
 
 /* file I/O */
-#define PATH_MAX 1024
 #define access  _access
 #define getcwd  _getcwd
 #define open    _open
 #define read    _read
 #define write   _write
-#define lseek   _lseek
 #define close   _close
 #define popen   _popen
 #define pclose  _pclose
+#ifndef __MINGW64_VERSION_MAJOR
+#define lseek   _lseek
+#define PATH_MAX 1024
 #define R_OK    04           /* read-only (for access()) */
+#endif
 #define S_ISDIR(m)  (((m) & _S_IFMT) == _S_IFDIR)
 #ifndef __MINGW32__
 enum { STDIN_FILENO = 0, STDOUT_FILENO = 1, STDERR_FILENO = 2 };
@@ -130,12 +132,15 @@
 #define DEFAULT_TEMPLATE_ROOTDIR  ".."
 
 // ----------------------------------- SYSTEM/PROCESS
+#ifndef __MINGW64_VERSION_MAJOR
 typedef int pid_t;
+#endif
 #define getpid  _getpid
 
 #endif  // _MSC_VER
 
 // ----------------------------------- THREADS
+#if !defined(WIN_PTHREADS_H)
 typedef DWORD pthread_t;
 typedef DWORD pthread_key_t;
 typedef LONG pthread_once_t;
@@ -147,6 +152,7 @@
   localtime_s(result, timep);
   return result;
 }
+#endif
 
 inline char* strerror_r(int errnum, char* buf, size_t buflen) {
 #ifdef FREE_WINDOWS
diff -Naur blender-2.77a-orig/intern/cycles/util/util_logging.h blender-2.77a/intern/cycles/util/util_logging.h
--- blender-2.77a-orig/intern/cycles/util/util_logging.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/cycles/util/util_logging.h	2016-07-12 13:40:19.646095000 +0300
@@ -18,7 +18,9 @@
 #define __UTIL_LOGGING_H__
 
 #if defined(WITH_CYCLES_LOGGING) && !defined(__KERNEL_GPU__)
+#  define GLOG_NO_ABBREVIATED_SEVERITIES
 #  include <glog/logging.h>
+#  undef GLOG_NO_ABBREVIATED_SEVERITIES
 #endif
 
 #include <iostream>
diff -Naur blender-2.77a-orig/intern/cycles/util/util_optimization.h blender-2.77a/intern/cycles/util/util_optimization.h
--- blender-2.77a-orig/intern/cycles/util/util_optimization.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/cycles/util/util_optimization.h	2016-07-12 13:40:19.681095000 +0300
@@ -99,7 +99,6 @@
  * We assume __KERNEL_SSEX__ flags to have been defined at this point */
 
 /* SSE intrinsics headers */
-#ifndef FREE_WINDOWS64
 
 #ifdef _MSC_VER
 #include <intrin.h>
@@ -107,14 +106,6 @@
 #  include <x86intrin.h>
 #endif
 
-#else
-
-/* MinGW64 has conflicting declarations for these SSE headers in <windows.h>.
- * Since we can't avoid including <windows.h>, better only include that */
-#include "util/util_windows.h"
-
-#endif
-
 #endif
 
 #endif /* __UTIL_OPTIMIZATION_H__ */
diff -Naur blender-2.77a-orig/intern/cycles/util/util_types.h blender-2.77a/intern/cycles/util/util_types.h
--- blender-2.77a-orig/intern/cycles/util/util_types.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/cycles/util/util_types.h	2016-07-12 13:40:19.712595000 +0300
@@ -56,7 +56,7 @@
 #    define ccl_device_inline static inline __attribute__((always_inline))
 #    define ccl_device_forceinline static inline __attribute__((always_inline))
 #    define ccl_align(...) __attribute__((aligned(__VA_ARGS__)))
-#    ifndef FREE_WINDOWS64
+#    ifndef __MINGW64_VERSION_MAJOR
 #      define __forceinline inline __attribute__((always_inline))
 #    endif
 #    define ccl_try_align(...) __attribute__((aligned(__VA_ARGS__)))
diff -Naur blender-2.79-orig/intern/dualcon/intern/Projections.h blender-2.79/intern/dualcon/intern/Projections.h
--- blender-2.77a-orig/intern/dualcon/intern/Projections.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/dualcon/intern/Projections.h	2016-07-12 13:40:19.712595000 +0300
@@ -29,7 +29,7 @@
 #define CONTAINS_INDEX
 #define GRID_DIMENSION 20
 
-#if defined(_WIN32) && !(_MSC_VER >= 1900)
+#if defined(_MSC_VER) && !(_MSC_VER >= 1900)
 #define isnan(n) _isnan(n)
 #define LONG __int64
 #define int64_t __int64
diff -Naur blender-2.77a-orig/intern/ghost/intern/GHOST_SystemWin32.h blender-2.77a/intern/ghost/intern/GHOST_SystemWin32.h
--- blender-2.77a-orig/intern/ghost/intern/GHOST_SystemWin32.h	2016-04-05 00:58:19.000000000 +0300
+++ blender-2.77a/intern/ghost/intern/GHOST_SystemWin32.h	2016-07-12 13:40:19.757095000 +0300
@@ -38,8 +38,10 @@
 #endif // WIN32
 
 /* require Windows XP or newer */
+#ifndef __MINGW64_VERSION_MAJOR
 #undef _WIN32_WINNT
 #define _WIN32_WINNT 0x501
+#endif
 
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
diff -Naur blender-2.77a-orig/intern/ghost/intern/GHOST_TaskbarWin32.h blender-2.77a/intern/ghost/intern/GHOST_TaskbarWin32.h
--- blender-2.77a-orig/intern/ghost/intern/GHOST_TaskbarWin32.h	2016-04-05 00:58:19.000000000 +0300
+++ blender-2.77a/intern/ghost/intern/GHOST_TaskbarWin32.h	2016-07-12 13:40:19.763595000 +0300
@@ -9,8 +9,10 @@
 #endif // WIN32
 
 /* require Windows XP or newer */
+#ifndef __MINGW64_VERSION_MAJOR
 #undef _WIN32_WINNT
 #define _WIN32_WINNT 0x501
+#endif
 
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
diff -Naur blender-2.77a-orig/intern/ghost/intern/GHOST_WindowWin32.cpp blender-2.77a/intern/ghost/intern/GHOST_WindowWin32.cpp
--- blender-2.77a-orig/intern/ghost/intern/GHOST_WindowWin32.cpp	2016-03-23 11:49:43.000000000 +0300
+++ blender-2.77a/intern/ghost/intern/GHOST_WindowWin32.cpp	2016-07-12 13:40:19.800595000 +0300
@@ -31,6 +31,12 @@
 
 #define _USE_MATH_DEFINES
 
+#if defined(WITH_GL_EGL)
+#  include "GHOST_ContextEGL.h"
+#else
+#  include "GHOST_ContextWGL.h"
+#endif
+
 #include "GHOST_WindowWin32.h"
 #include "GHOST_SystemWin32.h"
 #include "GHOST_DropTargetWin32.h"
@@ -38,11 +44,6 @@
 #include "utfconv.h"
 #include "utf_winfunc.h"
 
-#if defined(WITH_GL_EGL)
-#  include "GHOST_ContextEGL.h"
-#else
-#  include "GHOST_ContextWGL.h"
-#endif
 #ifdef WIN32_COMPOSITING
 #include <Dwmapi.h>
 #endif
diff -Naur blender-2.77a-orig/intern/libmv/libmv/numeric/numeric.h blender-2.77a/intern/libmv/libmv/numeric/numeric.h
--- blender-2.77a-orig/intern/libmv/libmv/numeric/numeric.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/libmv/libmv/numeric/numeric.h	2016-07-12 13:40:19.817095000 +0300
@@ -33,7 +33,7 @@
 #include <Eigen/QR>
 #include <Eigen/SVD>
 
-#if !defined(__MINGW64__)
+#if !defined(__MINGW64_VERSION_MAJOR)
 #  if defined(_WIN32) || defined(__APPLE__) || \
       defined(__FreeBSD__) || defined(__NetBSD__)
 static void sincos(double x, double *sinx, double *cosx) {
@@ -41,7 +41,7 @@
   *cosx = cos(x);
 }
 #  endif
-#endif  // !__MINGW64__
+#endif  // !__MINGW64_VERSION_MAJOR
 
 #if (defined(WIN32) || defined(WIN64)) && !defined(__MINGW32__)
 inline long lround(double d) {
diff -Naur blender-2.79-orig/extern/carve/include/carve/win32.h blender-2.79/extern/carve/include/carve/win32.h
--- blender-2.79-orig/extern/carve/include/carve/win32.h	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.79/extern/carve/include/carve/win32.h	2016-07-12 13:40:19.817095000 +0300
@@ -8,9 +8,11 @@
 #include <string.h>
 #include <stdlib.h>
 
+#if defined(_MSC_VER)
 inline int strcasecmp(const char *a, const char *b) {
   return _stricmp(a,b);
 }
+#endif
 
 inline void srandom(unsigned long input) {
   srand(input);
diff --Naur  a/source/blender/blenlib/BLI_fileops.h b/source/blender/blenlib/BLI_fileops.h
--- a/source/blender/blenlib/BLI_fileops.h
+++ b/source/blender/blenlib/BLI_fileops.h
@@ -65,8 +65,10 @@ int    BLI_create_symlink(const char *path, const char *to) ATTR_NONNULL();
 
 /* keep in sync with the definition of struct direntry in BLI_fileops_types.h */
 #ifdef WIN32
-#  if defined(_MSC_VER)
+#  if defined(_MSC_VER) || defined(__MINGW64__)
 typedef struct _stat64 BLI_stat_t;
+#  elif defined(__MINGW32__)
+typedef struct _stati64 BLI_stat_t;
 #  else
 typedef struct _stat BLI_stat_t;
 #  endif
diff -Naur blender-2.79-orig/source/blender/blenlib/BLI_fileops_types.h blender-2.79/source/blender/blenlib/BLI_fileops_types.h
--- blender-2.79-orig/source/blender/blenlib/BLI_fileops_types.h	2016-03-23 11:49:43.000000000 +0300
+++ blender-2.79/source/blender/blenlib/BLI_fileops_types.h	2016-07-12 13:40:19.827095000 +0300
@@ -35,7 +35,7 @@
 
 #include <sys/stat.h>
 
-#if defined(WIN32)
+#if defined(_MSC_VER)
 typedef unsigned int mode_t;
 #endif
 
@@ -50,8 +50,10 @@ struct direntry {
 	const char   *relname;
 	const char   *path;
 #ifdef WIN32 /* keep in sync with the definition of BLI_stat_t in BLI_fileops.h */
-#  if defined(_MSC_VER)
+#  if defined(_MSC_VER) || defined(__MINGW64__)
 	struct _stat64 s;
+#  elif defined(__MINGW32__)
+ 	struct _stati64 s;
 #  else
 	struct _stat s;
 #  endif
diff -Naur blender-2.77a-orig/source/blender/blenlib/BLI_winstuff.h blender-2.77a/source/blender/blenlib/BLI_winstuff.h
--- blender-2.77a-orig/source/blender/blenlib/BLI_winstuff.h	2016-03-23 11:49:43.000000000 +0300
+++ blender-2.77a/source/blender/blenlib/BLI_winstuff.h	2016-07-12 13:40:19.827095000 +0300
@@ -90,7 +90,7 @@
 #endif
 
 /* defines for using ISO C++ conformant names */
-#if !defined(_MSC_VER) || _MSC_VER < 1900
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
 #  define snprintf _snprintf
 #endif
 
@@ -95,18 +95,23 @@
 #  define	F_OK	0
 #endif
 
+#if defined(_MSC_VER)
 typedef unsigned int mode_t;
+#endif
 
 /* use functions that take a 64 bit offset for files larger than 4GB */
 #include <stdio.h>
+#if defined(_MSC_VER)
 #define fseek(stream, offset, origin) _fseeki64(stream, offset, origin)
 #define ftell(stream) _ftelli64(stream)
 #define lseek(fd, offset, origin) _lseeki64(fd, offset, origin)
 #define tell(fd) _telli64(fd)
+#endif
 
 
-#ifndef _SSIZE_T_
-#  define _SSIZE_T_
+/* mingw using _SSIZE_T_DEFINED to declare ssize_t type */
+#ifndef _SSIZE_T_DEFINED
+#  define _SSIZE_T_DEFINED
 /* python uses HAVE_SSIZE_T */
 #  ifndef HAVE_SSIZE_T
 #    define HAVE_SSIZE_T 1
diff -Naur blender-2.78-orig/source/blender/blenlib/intern/math_base.c blender-2.78/source/blender/blenlib/intern/math_base.c
--- blender-2.78-orig/source/blender/blenlib/intern/math_base.c	2016-09-28 12:26:55.000000000 +0300
+++ blender-2.78/source/blender/blenlib/intern/math_base.c	2016-10-14 13:53:56.430617800 +0300
@@ -56,7 +56,7 @@
 		pow2 = 1.0;
 		y = (x * pow1) * pow2;
 		/* if y overflows, then rounded value is exactly x */
-		if (!isfinite(y))
+		if (!isfinite((float)y))
 			return x;
 	}
 	else {
diff -Naur blender-2.78-orig/source/blender/collada/MeshImporter.cpp blender-2.78/source/blender/collada/MeshImporter.cpp
--- blender-2.78-orig/source/blender/collada/MeshImporter.cpp	2016-09-28 12:26:55.000000000 +0300
+++ blender-2.78/source/blender/collada/MeshImporter.cpp	2016-10-14 13:53:56.430617800 +0300
@@ -27,7 +27,7 @@
 
 #include <algorithm>
 
-#if !defined(WIN32)
+#if !defined(_MSC_VER)
 #include <iostream>
 #endif
 
diff -Naur blender-2.78-orig/source/creator/creator.c blender-2.78/source/creator/creator.c
--- blender-2.78-orig/source/creator/creator.c	2016-09-28 12:26:55.000000000 +0300
+++ blender-2.78/source/creator/creator.c	2016-10-14 14:45:36.982849100 +0300
@@ -31,6 +31,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <time.h>
 
 #ifdef WIN32
 #  if defined(_MSC_VER) && defined(_M_X64)
diff -Naur blender-2.79-orig/source/gameengine/Expressions/intern/HashedPtr.cpp blender-2.79/source/gameengine/Expressions/intern/HashedPtr.cpp
--- blender-2.79-orig/source/gameengine/Expressions/intern/HashedPtr.cpp	2016-09-28 12:26:55.000000000 +0300
+++ blender-2.79/source/gameengine/Expressions/intern/HashedPtr.cpp	2016-10-14 14:45:36.982849100 +0300
@@ -40,8 +40,10 @@ CHashedPtr::CHashedPtr(void* val) : m_valptr(val)
 
 unsigned int CHashedPtr::hash() const
 {
-#if defined(_WIN64)
+#if defined(_WIN64) && !defined(FREE_WINDOWS64)
 	unsigned __int64 key = (unsigned __int64)m_valptr;
+#elif defined(FREE_WINDOWS64)
+	unsigned long long key = (unsigned long long)m_valptr;
 #else
 	unsigned long key = (unsigned long)m_valptr;
 #endif
--- blender-2.79/source/gameengine/Ketsji/KX_IPO_SGController.cpp.orig	2017-09-14 16:01:33.887152700 +0300
+++ blender-2.79/source/gameengine/Ketsji/KX_IPO_SGController.cpp	2017-09-14 16:02:15.131853400 +0300
@@ -31,8 +31,10 @@
  */
 
 
-#if defined(_WIN64)
+#if defined(_WIN64) && !defined(FREE_WINDOWS64)
 typedef unsigned __int64 uint_ptr;
+#elif defined(FREE_WINDOWS64)
+typedef unsigned long long uint_ptr;
 #else
 typedef unsigned long uint_ptr;
 #endif
diff --Naur a/source/blender/blenlib/intern/winstuff.c b/source/blender/blenlib/intern/winstuff.c
--- a/source/blender/blenlib/intern/winstuff.c
+++ b/source/blender/blenlib/intern/winstuff.c
@@ -160,6 +160,8 @@ void RegisterBlendExtension(void)
 	GetSystemDirectory(SysDir, FILE_MAXDIR);
 #ifdef _WIN64
 	ThumbHandlerDLL = "BlendThumb64.dll";
+#elif defined(__MINGW32__)
+	ThumbHandlerDLL = "BlendThumb.dll";
 #else
 	IsWow64Process(GetCurrentProcess(), &IsWOW64);
 	if (IsWOW64 == true)
diff -Naur a/source/blender/imbuf/intern/IMB_anim.h b/source/blender/imbuf/intern/IMB_anim.h
--- a/source/blender/imbuf/intern/IMB_anim.h
+++ b/source/blender/imbuf/intern/IMB_anim.h
@@ -40,7 +40,10 @@
 #  include <mmsystem.h>
 #  include <memory.h>
 #  include <commdlg.h>
-#  include <vfw.h>
+
+#  ifndef FREE_WINDOWS
+#    include <vfw.h>
+#  endif
 
 #  undef AVIIF_KEYFRAME // redefined in AVI_avi.h
 #  undef AVIIF_LIST // redefined in AVI_avi.h
@@ -130,7 +127,7 @@ struct anim {
 	/* avi */
 	struct _AviMovie *avi;
 
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(FREE_WINDOWS)
 	/* windows avi */
 	int avistreams;
 	int firstvideo;
diff -Naur a/source/blender/imbuf/intern/anim_movie.c b/source/blender/imbuf/intern/anim_movie.c
--- a/source/blender/imbuf/intern/anim_movie.c
+++ b/source/blender/imbuf/intern/anim_movie.c
@@ -37,7 +37,10 @@
 #include <mmsystem.h>
 #include <memory.h>
 #include <commdlg.h>
+
+#ifndef FREE_WINDOWS
 #include <vfw.h>
+#endif
 
 #undef AVIIF_KEYFRAME /* redefined in AVI_avi.h */
 #undef AVIIF_LIST /* redefined in AVI_avi.h */
@@ -173,7 +170,7 @@ static void an_stringenc(char *string, const char *head, const char *tail, unsig
 #ifdef WITH_AVI
 static void free_anim_avi(struct anim *anim)
 {
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(FREE_WINDOWS)
 	int i;
 #endif
 
@@ -184,7 +181,7 @@ static void free_anim_avi(struct anim *anim)
 	MEM_freeN(anim->avi);
 	anim->avi = NULL;
 
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(FREE_WINDOWS)
 
 	if (anim->pgf) {
 		AVIStreamGetFrameClose(anim->pgf);
@@ -283,7 +280,7 @@ static int startavi(struct anim *anim)
 {
 
 	AviError avierror;
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(FREE_WINDOWS)
 	HRESULT hr;
 	int i, firstvideo = -1;
 	int streamcount;
@@ -304,7 +301,7 @@ static int startavi(struct anim *anim)
 
 	avierror = AVI_open_movie(anim->name, anim->avi);
 
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(FREE_WINDOWS)
 	if (avierror == AVI_ERROR_COMPRESSION) {
 		AVIFileInit();
 		hr = AVIFileOpen(&anim->pfile, anim->name, OF_READ, 0L);
@@ -401,7 +398,7 @@ static ImBuf *avi_fetchibuf(struct anim *anim, int position)
 		return NULL;
 	}
 
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(FREE_WINDOWS)
 	if (anim->avistreams) {
 		LPBITMAPINFOHEADER lpbi;
 
diff -Naur a/source/blender/blenlib/intern/path_util.c b/source/blender/blenlib/intern/path_util.c
--- a/source/blender/blenlib/intern/path_util.c
+++ b/source/blender/blenlib/intern/path_util.c
@@ -1173,7 +1173,18 @@ bool BLI_path_program_search(
 {
 	/* free windows */
 
-#if (defined(WIN32) || defined(WIN64))
+#if defined(__MINGW32__)
+	char *envstr;
+
+	if (val)
+		envstr = BLI_sprintfN("%s=%s", env, val);
+	else
+		envstr = BLI_sprintfN("%s=", env);
+
+	putenv(envstr);
+	MEM_freeN(envstr);
+
+#elif defined(_MSC_VER) /* not free windows */
 	uputenv(env, val);
 
 #else
diff -Naur a/source/blender/blenlib/intern/storage.c b/source/blender/blenlib/intern/storage.c
--- a/source/blender/blenlib/intern/storage.c
+++ b/source/blender/blenlib/intern/storage.c
@@ -253,8 +253,10 @@ int BLI_stat(const char *path, BLI_stat_t *buffer)
 
 int BLI_wstat(const wchar_t *path, BLI_stat_t *buffer)
 {
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) || defined(__MINGW64__)
 	return _wstat64(path, buffer);
+#elif defined(__MINGW32__)
+	return _wstati64(path, buffer);
 #else
 	return _wstat(path, buffer);
 #endif
@@ -424,13 +422,22 @@ void BLI_file_free_lines(LinkNode *lines)
 bool BLI_file_older(const char *file1, const char *file2)
 {
 #ifdef WIN32
-	struct _stat st1, st2;
+#ifndef __MINGW32__
+	struct _stat st1, st2;
+#else
+	struct _stati64 st1, st2;
+#endif
 
 	UTF16_ENCODE(file1);
 	UTF16_ENCODE(file2);
 
-	if (_wstat(file1_16, &st1)) return false;
-	if (_wstat(file2_16, &st2)) return false;
+#ifndef __MINGW32__
+	if (_wstat(file1_16, &st1)) return false;
+	if (_wstat(file2_16, &st2)) return false;
+#else
+	if (_wstati64(file1_16, &st1)) return false;
+	if (_wstati64(file2_16, &st2)) return false;
+#endif
 
 	UTF16_UN_ENCODE(file2);
 	UTF16_UN_ENCODE(file1);
