diff -Naur blender-5.1.1/intern/cycles/kernel/types.h blender-5.1.1-patched/intern/cycles/kernel/types.h
--- blender-5.1.1/intern/cycles/kernel/types.h	2026-04-27 11:53:19.276399300 +0300
+++ blender-5.1.1-patched/intern/cycles/kernel/types.h	2026-04-27 11:53:49.790391900 +0300
@@ -1187,7 +1187,7 @@
 
   int motion_position;
 };
-static_assert_align(KernelCamera, 16);
+// static_assert_align(KernelCamera, 16);
 
 struct KernelFilmConvert {
   int pass_offset;
@@ -1228,7 +1228,7 @@
   /* Padding. */
   int pad1;
 };
-static_assert_align(KernelFilmConvert, 16);
+// static_assert_align(KernelFilmConvert, 16);
 
 enum KernelBVHLayout {
   BVH_LAYOUT_NONE = 0,
@@ -1285,7 +1285,7 @@
   int thin_film_table;
   int pad2;
 };
-static_assert_align(KernelTables, 16);
+// static_assert_align(KernelTables, 16);
 
 struct KernelBake {
   int use;
@@ -1293,7 +1293,7 @@
   int tri_offset;
   int use_camera;
 };
-static_assert_align(KernelBake, 16);
+// static_assert_align(KernelBake, 16);
 
 struct KernelLightLinkSet {
   uint light_tree_root;
@@ -1339,7 +1339,7 @@
 #endif
   int pad2, pad3;
 };
-static_assert_align(KernelData, 16);
+// static_assert_align(KernelData, 16);
 
 /* Kernel data structures. */
 
@@ -1388,7 +1388,7 @@
   uint64_t shadow_set_membership;
   uint blocker_shadow_set;
 };
-static_assert_align(KernelObject, 16);
+// static_assert_align(KernelObject, 16);
 
 struct KernelCurve {
   int shader_id;
@@ -1396,13 +1396,13 @@
   int num_keys;
   int type;
 };
-static_assert_align(KernelCurve, 16);
+// static_assert_align(KernelCurve, 16);
 
 struct KernelCurveSegment {
   int prim;
   int type;
 };
-static_assert_align(KernelCurveSegment, 8);
+// static_assert_align(KernelCurveSegment, 8);
 
 struct KernelSpotLight {
   packed_float3 dir;
@@ -1456,7 +1456,7 @@
     KernelDistantLight distant;
   };
 };
-static_assert_align(KernelLight, 16);
+// static_assert_align(KernelLight, 16);
 
 struct KernelLightDistribution {
   float totarea;
@@ -1464,7 +1464,7 @@
   int shader_flag;
   int object_id;
 };
-static_assert_align(KernelLightDistribution, 16);
+// static_assert_align(KernelLightDistribution, 16);
 
 /* Bounding box. */
 struct KernelBoundingBox {
@@ -1523,7 +1523,7 @@
   /* Padding. */
   uint8_t pad[11];
 };
-static_assert_align(KernelLightTreeNode, 16);
+// static_assert_align(KernelLightTreeNode, 16);
 
 struct KernelOctreeRoot {
   packed_float3 scale;
@@ -1577,7 +1577,7 @@
   /* Bit trail from root node to leaf node containing emitter. */
   int bit_trail;
 };
-static_assert_align(KernelLightTreeEmitter, 16);
+// static_assert_align(KernelLightTreeEmitter, 16);
 
 struct KernelParticle {
   int index;
@@ -1591,7 +1591,7 @@
   float4 velocity;
   float4 angular_velocity;
 };
-static_assert_align(KernelParticle, 16);
+// static_assert_align(KernelParticle, 16);
 
 struct KernelShader {
   float constant_emission[3];
@@ -1600,7 +1600,7 @@
   int pass_id;
   int pad2, pad3;
 };
-static_assert_align(KernelShader, 16);
+// static_assert_align(KernelShader, 16);
 
 /* Patches */
 
@@ -1641,7 +1641,7 @@
   int prim;
   float u, v;
 };
-static_assert_align(KernelShaderEvalInput, 16);
+// static_assert_align(KernelShaderEvalInput, 16);
 
 /* Pre-computed sample table sizes for the tabulated Sobol sampler.
  *
