diff -Naur blender-5.2.0/source/blender/makesdna/DNA_meta_types.h blender-5.2.0-patched/source/blender/makesdna/DNA_meta_types.h
--- blender-5.2.0/source/blender/makesdna/DNA_meta_types.h	2026-06-03 15:03:49.000000000 +0200
+++ blender-5.2.0-patched/source/blender/makesdna/DNA_meta_types.h	2026-08-09 00:21:07.113713500 +0200
@@ -19,6 +19,8 @@
 struct BoundBox;
 struct Material;
 
+#undef rad2
+
 /** #MetaBall::texspace_flag */
 enum eMetaBall_TexSpaceFlag : char {
   MB_TEXSPACE_FLAG_AUTO = 1 << 0,
