diff -Naur blender-5.1.1/source/blender/makesdna/DNA_meta_types.h blender-5.1.1-patched/source/blender/makesdna/DNA_meta_types.h
--- blender-5.1.1/source/blender/makesdna/DNA_meta_types.h	2026-04-27 11:53:46.227344800 +0300
+++ blender-5.1.1-patched/source/blender/makesdna/DNA_meta_types.h	2026-04-27 11:54:12.483192500 +0300
@@ -17,6 +17,8 @@
 struct BoundBox;
 struct Material;
 
+#undef rad2
+
 /** #MetaBall::texspace_flag */
 enum {
   MB_TEXSPACE_FLAG_AUTO = 1 << 0,
