diff -Naur blender-4.5.2/source/blender/makesdna/DNA_meta_types.h blender-4.5.2-patched/source/blender/makesdna/DNA_meta_types.h
--- blender-4.5.2/source/blender/makesdna/DNA_meta_types.h	2025-08-26 12:20:27.981478900 +0300
+++ blender-4.5.2-patched/source/blender/makesdna/DNA_meta_types.h	2025-08-26 12:21:09.233166800 +0300
@@ -17,6 +17,8 @@
 struct Ipo;
 struct Material;
 
+#undef rad2
+
 typedef struct MetaElem {
   struct MetaElem *next, *prev;
 
