diff -Naur blender-2.77a-orig/extern/clew/include/clew.h blender-2.77a/extern/clew/include/clew.h
--- blender-2.77a-orig/extern/clew/include/clew.h	2016-03-21 19:16:12.000000000 +0300
+++ blender-2.77a/extern/clew/include/clew.h	2016-07-12 13:40:29.182095000 +0300
@@ -493,7 +493,7 @@
 /* Define alignment keys */
 #if defined( __GNUC__ )
     #define CL_ALIGNED(_x)          __attribute__ ((aligned(_x)))
-#elif defined( _WIN32) && (_MSC_VER)
+#elif defined( _WIN32) && defined(_MSC_VER)
     /* Alignment keys neutered on windows because MSVC can't swallow function arguments with alignment requirements     */
     /* http://msdn.microsoft.com/en-us/library/373ak2y1%28VS.71%29.aspx                                                 */
     /* #include <crtdefs.h>                                                                                             */
