diff -Naur blender-4.5.2/intern/cycles/scene/CMakeLists.txt blender-4.5.2-patched/intern/cycles/scene/CMakeLists.txt
--- blender-4.5.2/intern/cycles/scene/CMakeLists.txt	2025-08-26 12:21:49.989896800 +0300
+++ blender-4.5.2-patched/intern/cycles/scene/CMakeLists.txt	2025-08-26 12:22:36.099083200 +0300
@@ -93,6 +93,9 @@
   volume.h
 )
 
+  # To avoid ..\volume.cpp.obj: too many sections (82336)
+  set_source_files_properties(volume.cpp PROPERTIES COMPILE_FLAGS "-g0")
+
 set(LIB
   cycles_bvh
   cycles_device
diff -Naur blender-4.5.2/source/blender/blenkernel/CMakeLists.txt blender-4.5.2-patched/source/blender/blenkernel/CMakeLists.txt
--- blender-4.5.2/source/blender/blenkernel/CMakeLists.txt	2025-08-26 12:21:44.089916400 +0300
+++ blender-4.5.2-patched/source/blender/blenkernel/CMakeLists.txt	2025-08-26 12:22:36.120080500 +0300
@@ -565,6 +565,11 @@
   intern/subdiv_inline.hh
 )
 
+  # To avoid : too many sections
+  set_source_files_properties(intern/volume_grid.cc PROPERTIES COMPILE_FLAGS "-g0")
+  set_source_files_properties(intern/volume_to_mesh.cc PROPERTIES COMPILE_FLAGS "-g0")
+  set_source_files_properties(intern/volume.cc PROPERTIES COMPILE_FLAGS "-g0")
+
 set(LIB
   PRIVATE bf::animrig
   PRIVATE bf::asset_system
