--- a/intern/libmv/CMakeLists.txt
+++ b/intern/libmv/CMakeLists.txt
@@ -195,6 +195,9 @@
     third_party/msinttypes/stdint.h
   )
 
+  # To avoid ..\fundamental.cc.obj: too many sections (35985)
+  set_source_files_properties(libmv/multiview/fundamental.cc PROPERTIES COMPILE_FLAGS "-g0")
+  set_source_files_properties(libmv/simple_pipeline/resect.cc PROPERTIES COMPILE_FLAGS "-g0")
 
   if(WITH_GTESTS)
     include(GTestTesting)
