diff -Naur blender-5.1.1/intern/cycles/scene/CMakeLists.txt blender-5.1.1-patched/intern/cycles/scene/CMakeLists.txt
--- blender-5.1.1/intern/cycles/scene/CMakeLists.txt	2026-04-27 11:54:30.004621700 +0300
+++ blender-5.1.1-patched/intern/cycles/scene/CMakeLists.txt	2026-04-27 11:55:02.944880200 +0300
@@ -87,6 +87,9 @@
   volume.h
 )
 
+  # To avoid ..\volume.cpp.obj: too many sections (82336)
+  set_source_files_properties(volume.cpp PROPERTIES COMPILE_FLAGS "-g0")
+
 set(LIB
   cycles_bvh
   cycles_device
diff -Naur blender-5.1.1/source/blender/blenkernel/CMakeLists.txt blender-5.1.1-patched/source/blender/blenkernel/CMakeLists.txt
--- blender-5.1.1/source/blender/blenkernel/CMakeLists.txt	2026-04-27 11:54:32.203084700 +0300
+++ blender-5.1.1-patched/source/blender/blenkernel/CMakeLists.txt	2026-04-27 11:55:02.948067300 +0300
@@ -557,6 +557,11 @@
   intern/tracking_private.hh
 )
 
+  # To avoid : too many sections
+  set_source_files_properties(intern/volume_grid.cc PROPERTIES COMPILE_FLAGS "-g0")
+  set_source_files_properties(intern/volume_to_mesh.cc PROPERTIES COMPILE_FLAGS "-g0")
+  set_source_files_properties(intern/volume.cc PROPERTIES COMPILE_FLAGS "-g0")
+
 set(LIB
   PRIVATE bf::animrig
   PRIVATE bf::asset_system
