diff -Naur blender-2.77a-orig/build_files/cmake/macros.cmake blender-2.77a/build_files/cmake/macros.cmake
--- blender-2.77a-orig/build_files/cmake/macros.cmake	2016-04-08 05:12:03.000000000 +0300
+++ blender-2.77a/build_files/cmake/macros.cmake	2016-07-12 13:40:35.162095000 +0300
@@ -328,7 +328,7 @@
 	if(WITH_PYTHON)  # AND NOT WITH_PYTHON_MODULE  # WIN32 needs
 		target_link_libraries(${target} ${PYTHON_LINKFLAGS})
 
-		if(WIN32 AND NOT UNIX)
+		if(WIN32 AND NOT UNIX AND NOT MINGW)
 			file_list_suffix(PYTHON_LIBRARIES_DEBUG "${PYTHON_LIBRARIES}" "_d")
 			target_link_libraries_debug(${target} "${PYTHON_LIBRARIES_DEBUG}")
 			target_link_libraries_optimized(${target} "${PYTHON_LIBRARIES}")
@@ -410,7 +410,7 @@
 		target_link_libraries(${target} ${FFMPEG_LIBRARIES})
 	endif()
 	if(WITH_OPENCOLLADA)
-		if(WIN32 AND NOT UNIX)
+		if(WIN32 AND NOT UNIX AND NOT MINGW)
 			file_list_suffix(OPENCOLLADA_LIBRARIES_DEBUG "${OPENCOLLADA_LIBRARIES}" "_d")
 			target_link_libraries_debug(${target} "${OPENCOLLADA_LIBRARIES_DEBUG}")
 			target_link_libraries_optimized(${target} "${OPENCOLLADA_LIBRARIES}")
