--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1537,7 +1537,7 @@
 
 #-----------------------------------------------------------------------------
 # Extra compile flags
-
+if(NOT MINGW)
 if(CMAKE_COMPILER_IS_GNUCC)
 
   ADD_CHECK_C_COMPILER_FLAG(C_WARNINGS C_WARN_ALL -Wall)
@@ -1762,7 +1762,7 @@
   set(CXX_WARNINGS "${_WARNINGS}")
   unset(_WARNINGS)
 endif()
-
+endif()
 # ensure python header is found since detection can fail, this could happen
 # with _any_ library but since we used a fixed python version this tends to
 # be most problematic.
