--- a/intern/ghost/intern/GHOST_Wintab.cc
+++ b/intern/ghost/intern/GHOST_Wintab.cc
@@ -5,6 +5,7 @@
  */
 
 #define _USE_MATH_DEFINES
+#include <cmath>
 
 #include "GHOST_Wintab.h"
 
--- a/extern/glog/src/symbolize.h
+++ b/extern/glog/src/symbolize.h
@@ -60,6 +60,8 @@
 
 #ifdef HAVE_SYMBOLIZE
 
+#include <cstdint>
+
 #if defined(__ELF__)  // defined by gcc
 #if defined(__OpenBSD__)
 #include <sys/exec_elf.h>
--- a/source/creator/creator.cc
+++ b/source/creator/creator.cc
@@ -8,6 +8,7 @@
 
 #include <cstdlib>
 #include <cstring>
+#include <time.h>
 
 #ifdef WIN32
 #  include "utfconv.h"
--- a/source/blender/blenlib/intern/winstuff_registration.cc
+++ b/source/blender/blenlib/intern/winstuff_registration.cc
@@ -9,7 +9,11 @@
 #  include <filesystem>
 #  include <propkey.h>
 #  include <propvarutil.h>
+#ifdef __MINGW32__
+#  include <shlobj.h>
+#else
 #  include <shlobj_core.h>
+#endif
 #  include <wrl.h>
 
 #  include "BLI_path_util.h"
--- a/source/blender/blenkernel/intern/ocean_spectrum.cc
+++ b/source/blender/blenkernel/intern/ocean_spectrum.cc
@@ -9,6 +9,7 @@
 #include "BKE_ocean.h"
 #include "ocean_intern.h"
 
+#define _USE_MATH_DEFINES
 #include <cmath>
 
 #ifdef WITH_OCEANSIM
