diff -Naur blender-5.2.0/extern/glog/src/symbolize.h blender-5.2.0-patched/extern/glog/src/symbolize.h
--- blender-5.2.0/extern/glog/src/symbolize.h	2026-06-03 15:03:48.000000000 +0200
+++ blender-5.2.0-patched/extern/glog/src/symbolize.h	2026-08-09 00:21:07.871307600 +0200
@@ -60,6 +60,8 @@
 
 #ifdef HAVE_SYMBOLIZE
 
+#include <cstdint>
+
 #if defined(__ELF__)  // defined by gcc
 #if defined(__OpenBSD__)
 #include <sys/exec_elf.h>
diff -Naur blender-5.2.0/intern/ghost/intern/GHOST_Wintab.cc blender-5.2.0-patched/intern/ghost/intern/GHOST_Wintab.cc
--- blender-5.2.0/intern/ghost/intern/GHOST_Wintab.cc	2026-06-03 15:03:48.000000000 +0200
+++ blender-5.2.0-patched/intern/ghost/intern/GHOST_Wintab.cc	2026-08-09 00:21:07.871307600 +0200
@@ -6,6 +6,9 @@
  * \ingroup GHOST
  */
 
+#define _USE_MATH_DEFINES
+#include <cmath>
+
 #include "GHOST_Wintab.hh"
 
 GHOST_Wintab *GHOST_Wintab::loadWintabUnsafe(HWND hwnd)
diff -Naur blender-5.2.0/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh blender-5.2.0-patched/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh
--- blender-5.2.0/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh	2026-06-03 15:03:48.000000000 +0200
+++ blender-5.2.0-patched/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh	2026-08-09 00:21:07.875337900 +0200
@@ -17,6 +17,7 @@
 #  include "GHOST_ContextVK.hh"
 #endif
 #include "GHOST_IXrGraphicsBinding.hh"
+#include <list>
 
 /**
  * Base class for bridging to an OpenXR platform that only supports Direct3D.
diff -Naur blender-5.2.0/intern/libmv/libmv/numeric/numeric.h blender-5.2.0-patched/intern/libmv/libmv/numeric/numeric.h
--- blender-5.2.0/intern/libmv/libmv/numeric/numeric.h	2026-08-09 00:21:06.675502100 +0200
+++ blender-5.2.0-patched/intern/libmv/libmv/numeric/numeric.h	2026-08-09 00:21:07.876856800 +0200
@@ -34,6 +34,7 @@
 #include <Eigen/LU>
 #include <Eigen/QR>
 #include <Eigen/SVD>
+#include <cassert>
 
 #if !defined(__MINGW32__)
 #  if defined(_WIN32) || defined(__APPLE__) || defined(__NetBSD__) ||          \
diff -Naur blender-5.2.0/source/blender/blenkernel/intern/ocean_spectrum.cc blender-5.2.0-patched/source/blender/blenkernel/intern/ocean_spectrum.cc
--- blender-5.2.0/source/blender/blenkernel/intern/ocean_spectrum.cc	2026-06-03 15:03:49.000000000 +0200
+++ blender-5.2.0-patched/source/blender/blenkernel/intern/ocean_spectrum.cc	2026-08-09 00:21:07.876856800 +0200
@@ -14,6 +14,7 @@
 #include "ocean_intern.h"
 
 #include <algorithm>
+#define _USE_MATH_DEFINES
 #include <cmath>
 
 namespace blender {
diff -Naur blender-5.2.0/source/blender/blenlib/BLI_index_range.hh blender-5.2.0-patched/source/blender/blenlib/BLI_index_range.hh
--- blender-5.2.0/source/blender/blenlib/BLI_index_range.hh	2026-06-03 15:03:49.000000000 +0200
+++ blender-5.2.0-patched/source/blender/blenlib/BLI_index_range.hh	2026-08-09 00:21:07.876856800 +0200
@@ -43,6 +43,7 @@
 
 #include <algorithm>
 #include <iosfwd>
+#include <cstdint>
 
 #include "BLI_assert.h"
 #include "BLI_random_access_iterator_mixin.hh"
diff -Naur blender-5.2.0/source/blender/blenlib/intern/winstuff_registration.cc blender-5.2.0-patched/source/blender/blenlib/intern/winstuff_registration.cc
--- blender-5.2.0/source/blender/blenlib/intern/winstuff_registration.cc	2026-06-03 15:03:49.000000000 +0200
+++ blender-5.2.0-patched/source/blender/blenlib/intern/winstuff_registration.cc	2026-08-09 00:21:07.882374800 +0200
@@ -16,7 +16,11 @@
 #  include <filesystem>
 #  include <propkey.h>
 #  include <propvarutil.h>
+#ifdef __MINGW32__
+#  include <shlobj.h>
+#else
 #  include <shlobj_core.h>
+#endif
 #  include <wrl.h>
 
 #  include "BLI_path_utils.hh"
diff -Naur blender-5.2.0/source/creator/creator.cc blender-5.2.0-patched/source/creator/creator.cc
--- blender-5.2.0/source/creator/creator.cc	2026-06-23 18:12:48.000000000 +0200
+++ blender-5.2.0-patched/source/creator/creator.cc	2026-08-09 00:21:07.882374800 +0200
@@ -8,6 +8,7 @@
 
 #include <cstdlib>
 #include <cstring>
+#include <time.h>
 
 #ifdef WIN32
 #  ifdef WIN32_LEAN_AND_MEAN
