diff -Naur blender-4.5.2/extern/ceres/internal/ceres/file.cc blender-4.5.2-patched/extern/ceres/internal/ceres/file.cc
--- blender-4.5.2/extern/ceres/internal/ceres/file.cc	2025-08-26 12:23:13.204800600 +0300
+++ blender-4.5.2-patched/extern/ceres/internal/ceres/file.cc	2025-08-26 12:23:40.166618900 +0300
@@ -34,6 +34,7 @@
 
 #include <cstdio>
 #include <string>
+#include <cstdint>
 
 #include "glog/logging.h"
 
diff -Naur blender-4.5.2/extern/glog/src/symbolize.h blender-4.5.2-patched/extern/glog/src/symbolize.h
--- blender-4.5.2/extern/glog/src/symbolize.h	2025-08-26 12:23:10.834415100 +0300
+++ blender-4.5.2-patched/extern/glog/src/symbolize.h	2025-08-26 12:23:40.131624500 +0300
@@ -60,6 +60,8 @@
 
 #ifdef HAVE_SYMBOLIZE
 
+#include <cstdint>
+
 #if defined(__ELF__)  // defined by gcc
 #if defined(__OpenBSD__)
 #include <sys/exec_elf.h>
diff -Naur blender-4.5.2/intern/ghost/intern/GHOST_Wintab.cc blender-4.5.2-patched/intern/ghost/intern/GHOST_Wintab.cc
--- blender-4.5.2/intern/ghost/intern/GHOST_Wintab.cc	2025-08-26 12:23:03.002466700 +0300
+++ blender-4.5.2-patched/intern/ghost/intern/GHOST_Wintab.cc	2025-08-26 12:23:40.123623400 +0300
@@ -7,6 +7,7 @@
  */
 
 #define _USE_MATH_DEFINES
+#include <cmath>
 
 #include "GHOST_Wintab.hh"
 
diff -Naur blender-4.5.2/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh blender-4.5.2-patched/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh
--- blender-4.5.2/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh	2025-08-26 12:23:02.988501800 +0300
+++ blender-4.5.2-patched/intern/ghost/intern/GHOST_XrGraphicsBindingD3D.hh	2025-08-26 12:23:40.163619800 +0300
@@ -17,6 +17,7 @@
 #  include "GHOST_ContextVK.hh"
 #endif
 #include "GHOST_IXrGraphicsBinding.hh"
+#include <list>
 
 /**
  * Base class for bridging to an OpenXR platform that only supports Direct3D.
diff -Naur blender-4.5.2/source/blender/blenkernel/intern/ocean_spectrum.cc blender-4.5.2-patched/source/blender/blenkernel/intern/ocean_spectrum.cc
--- blender-4.5.2/source/blender/blenkernel/intern/ocean_spectrum.cc	2025-08-26 12:23:07.657633600 +0300
+++ blender-4.5.2-patched/source/blender/blenkernel/intern/ocean_spectrum.cc	2025-08-26 12:23:40.151622200 +0300
@@ -12,6 +12,7 @@
 #include "ocean_intern.h"
 
 #include <algorithm>
+#define _USE_MATH_DEFINES
 #include <cmath>
 
 #ifdef WITH_OCEANSIM
diff -Naur blender-4.5.2/source/blender/blenlib/BLI_index_range.hh blender-4.5.2-patched/source/blender/blenlib/BLI_index_range.hh
--- blender-4.5.2/source/blender/blenlib/BLI_index_range.hh	2025-08-26 12:23:04.154194100 +0300
+++ blender-4.5.2-patched/source/blender/blenlib/BLI_index_range.hh	2025-08-26 12:23:40.157620100 +0300
@@ -39,6 +39,7 @@
 
 #include <algorithm>
 #include <iosfwd>
+#include <cstdint>
 
 #include "BLI_assert.h"
 #include "BLI_random_access_iterator_mixin.hh"
diff -Naur blender-4.5.2/source/blender/blenlib/intern/winstuff_registration.cc blender-4.5.2-patched/source/blender/blenlib/intern/winstuff_registration.cc
--- blender-4.5.2/source/blender/blenlib/intern/winstuff_registration.cc	2025-08-26 12:23:04.209304500 +0300
+++ blender-4.5.2-patched/source/blender/blenlib/intern/winstuff_registration.cc	2025-08-26 12:23:40.144657600 +0300
@@ -13,7 +13,11 @@
 #  include <filesystem>
 #  include <propkey.h>
 #  include <propvarutil.h>
+#ifdef __MINGW32__
+#  include <shlobj.h>
+#else
 #  include <shlobj_core.h>
+#endif
 #  include <wrl.h>
 
 #  include "BLI_path_utils.hh"
diff -Naur blender-4.5.2/source/creator/creator.cc blender-4.5.2-patched/source/creator/creator.cc
--- blender-4.5.2/source/creator/creator.cc	2025-08-26 12:23:08.105577200 +0300
+++ blender-4.5.2-patched/source/creator/creator.cc	2025-08-26 12:23:40.138654200 +0300
@@ -8,6 +8,7 @@
 
 #include <cstdlib>
 #include <cstring>
+#include <time.h>
 
 #ifdef WIN32
 #  include "utfconv.hh"
