diff -Naur blender-2.77a-orig/intern/cycles/kernel/osl/osl_closures.cpp blender-2.77a/intern/cycles/kernel/osl/osl_closures.cpp
--- blender-2.77a-orig/intern/cycles/kernel/osl/osl_closures.cpp	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/cycles/kernel/osl/osl_closures.cpp	2016-07-12 13:40:41.284595000 +0300
@@ -179,7 +179,7 @@
 	/* optimization: it's possible to not use a prepare function at all and
 	 * only initialize the actual class when accessing the closure component
 	 * data, but then we need to map the id to the class somehow */
-	ss->register_closure(name, id, params, prepare, NULL, 16);
+	ss->register_closure(name, id, params, prepare, NULL/*, 16*/);
 }
 
 void OSLShader::register_closures(OSLShadingSystem *ss_)
