diff -Naur blender-2.77a-orig/intern/ghost/intern/GHOST_ImeWin32.cpp blender-2.77a/intern/ghost/intern/GHOST_ImeWin32.cpp
--- blender-2.77a-orig/intern/ghost/intern/GHOST_ImeWin32.cpp	2016-03-23 11:49:49.000000000 +0300
+++ blender-2.77a/intern/ghost/intern/GHOST_ImeWin32.cpp	2016-07-12 13:40:41.284595000 +0300
@@ -64,7 +64,7 @@
 	 * while composing a text.
 	 */
 	HKL keyboard_layout = ::GetKeyboardLayout(0);
-	input_language_id_ = reinterpret_cast<LANGID>(keyboard_layout);
+	input_language_id_ = static_cast<LANGID>(reinterpret_cast<uintptr_t>(keyboard_layout));
 	ime_status_ = ::ImmIsIME(keyboard_layout);
 	return ime_status_;
 }
