diff -Naur blender-5.1.1/source/blender/blenlib/intern/math_geom.cc blender-5.1.1-patched/source/blender/blenlib/intern/math_geom.cc
--- blender-5.1.1/source/blender/blenlib/intern/math_geom.cc	2026-06-10 07:38:48.000000000 +0200
+++ blender-5.1.1-patched/source/blender/blenlib/intern/math_geom.cc	2026-08-08 22:20:00.000000000 +0200
@@ -23,6 +23,9 @@
 
 namespace blender {
 
+#undef near
+#undef far
+
 /* -------------------------------------------------------------------- */
 /** \name Polygons
  * \{ */
