diff -Naur blender-5.1.1/source/blender/blenlib/intern/math_geom.cc blender-5.1.1-patched/source/blender/blenlib/intern/math_geom.cc
--- blender-5.1.1/source/blender/blenlib/intern/math_geom.cc	2026-04-27 11:55:48.844727800 +0300
+++ blender-5.1.1-patched/source/blender/blenlib/intern/math_geom.cc	2026-04-27 11:56:16.273876600 +0300
@@ -23,6 +23,9 @@
 
 namespace blender {
 
+#undef near
+#undef far
+
 /********************************** Polygons *********************************/
 
 void cross_tri_v3(float n[3], const float v1[3], const float v2[3], const float v3[3])
