diff -Naur blender-4.5.2/source/blender/blenlib/intern/uuid.cc blender-4.5.2-patched/source/blender/blenlib/intern/uuid.cc
--- blender-4.5.2/source/blender/blenlib/intern/uuid.cc	2025-08-26 12:24:27.996670800 +0300
+++ blender-4.5.2-patched/source/blender/blenlib/intern/uuid.cc	2025-08-26 12:24:58.069119400 +0300
@@ -31,7 +31,7 @@
     static_assert(std::mt19937_64::max() == 0xffffffffffffffffLL);
 
     timespec ts;
-#ifdef __APPLE__
+#if defined(__APPLE__) || (defined(__MINGW32__) && !defined(_UCRT))
     /* `timespec_get()` is only available on macOS 10.15+, so until that's the minimum version
      * supported by Blender, use another function to get the timespec.
      *
