diff -Naur blender-5.2.2/scripts/addons_core/io_scene_gltf2/io/com/library.py blender-5.2.2-patched/scripts/addons_core/io_scene_gltf2/io/com/library.py
--- blender-5.2.2/scripts/addons_core/io_scene_gltf2/io/com/library.py	2026-06-11 16:38:45.000000000 +0200
+++ blender-5.2.2-patched/scripts/addons_core/io_scene_gltf2/io/com/library.py	2026-08-08 21:15:00.000000000 +0200
@@ -15,7 +15,7 @@ def dll_path(lib_name, lib_display_name) -> Path | None:
     import bpy
 
     if sys.platform == 'win32':
-        library_name = '{}.dll'.format(lib_name)
+        library_name = 'lib{}.dll'.format(lib_name)
         # Bundled beside the add-on's `__init__.py`.
         base = os.path.dirname(sys.modules['io_scene_gltf2'].__file__)
     elif sys.platform == 'darwin':
