--- a/scripts/addons_core/io_scene_gltf2/io/com/draco.py
+++ b/scripts/addons_core/io_scene_gltf2/io/com/draco.py
@@ -15,7 +15,7 @@
     """
     lib_name = 'extern_draco'
     library_name = {
-        'win32': '{}.dll'.format(lib_name),
+        'win32': 'lib{}.dll'.format(lib_name),
         'linux': 'lib{}.so'.format(lib_name),
         'darwin': 'lib{}.dylib'.format(lib_name)
     }.get(sys.platform)
