--- blender-2.79/source/blender/imbuf/intern/dds/dds_api.cpp.orig	2017-09-14 15:29:21.477036100 +0300
+++ blender-2.79/source/blender/imbuf/intern/dds/dds_api.cpp	2017-09-14 15:29:30.667254000 +0300
@@ -62,7 +62,7 @@
 	/* open file for writing */
 	std::ofstream fildes;
 
-#if defined (WIN32)
+#if defined (_MSC_VER)
 	wchar_t *wname = alloc_utf16_from_8(name, 0);
 	fildes.open(wname);
 	free(wname);
--- blender-2.79/source/blender/imbuf/intern/openexr/openexr_api.cpp.orig	2017-09-14 15:27:43.677657600 +0300
+++ blender-2.79/source/blender/imbuf/intern/openexr/openexr_api.cpp	2017-09-14 15:28:07.224301900 +0300
@@ -180,7 +180,7 @@
 	: IStream(filename)
 	{
 		/* utf-8 file path support on windows */
-#if defined (WIN32)
+#if defined (_MSC_VER)
 		wchar_t *wfilename = alloc_utf16_from_8(filename, 0);
 		ifs.open(wfilename, std::ios_base::binary);
 		free(wfilename);
@@ -243,7 +243,7 @@
 	: OStream(filename)
 	{
 		/* utf-8 file path support on windows */
-#if defined (WIN32)
+#if defined (_MSC_VER)
 		wchar_t *wfilename = alloc_utf16_from_8(filename, 0);
 		ofs.open(wfilename, std::ios_base::binary);
 		free(wfilename);
