--- a/source/blender/imbuf/intern/dds/dds_api.cpp
+++ b/source/blender/imbuf/intern/dds/dds_api.cpp
@@ -43,7 +43,7 @@
   /* open file for writing */
   std::ofstream fildes;
 
-#if defined(WIN32)
+#if defined (_MSC_VER)
   wchar_t *wname = alloc_utf16_from_8(name, 0);
   fildes.open(wname);
   free(wname);
--- a/source/blender/imbuf/intern/openexr/openexr_api.cpp
+++ b/source/blender/imbuf/intern/openexr/openexr_api.cpp
@@ -216,7 +216,7 @@
   IFileStream(const char *filepath) : IStream(filepath)
   {
     /* utf-8 file path support on windows */
-#if defined(WIN32)
+#if defined(_MSC_VER)
     wchar_t *wfilepath = alloc_utf16_from_8(filepath, 0);
     ifs.open(wfilepath, std::ios_base::binary);
     free(wfilepath);
@@ -322,7 +322,7 @@
   OFileStream(const char *filepath) : OStream(filepath)
   {
     /* utf-8 file path support on windows */
-#if defined(WIN32)
+#if defined(_MSC_VER)
     wchar_t *wfilepath = alloc_utf16_from_8(filepath, 0);
     ofs.open(wfilepath, std::ios_base::binary);
     free(wfilepath);
