--- a/source/blender/python/intern/bpy_app_handlers.c
+++ b/source/blender/python/intern/bpy_app_handlers.c
@@ -137,7 +137,7 @@
 /* dummy type because decorators can't be PyCFunctions */
 static PyTypeObject BPyPersistent_Type = {
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     PyVarObject_HEAD_INIT(NULL, 0)
 #else
     PyVarObject_HEAD_INIT(&PyType_Type, 0)
@@ -214,7 +214,7 @@
 {
   PyObject *ret;
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
   BPyPersistent_Type.ob_base.ob_base.ob_type = &PyType_Type;
 #endif
 
--- a/source/blender/python/intern/bpy_rna.c
+++ b/source/blender/python/intern/bpy_rna.c
@@ -6552,7 +6552,7 @@
     /*tp_methods*/ NULL,
     /*tp_members*/ NULL,
     /*tp_getset*/ NULL,
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     /*tp_base*/ NULL, /* Defer assignment. */
 #else
     /*tp_base*/ &PyType_Type,
@@ -7492,7 +7492,7 @@
 #endif
 
   /* For some reason MSVC complains of these. */
-#if defined(_MSC_VER)
+#if defined(_WIN32)
   pyrna_struct_meta_idprop_Type.tp_base = &PyType_Type;
 #endif
 
