--- blender-4.0.2-orig/source/blender/python/intern/bpy_app_handlers.cc	2023-09-27 18:54:39.000000000 +0300
+++ blender-4.0.2/source/blender/python/intern/bpy_app_handlers.cc	2024-01-11 14:44:50.063674100 +0300
@@ -159,7 +159,7 @@
 
 /** Dummy type because decorators can't be a #PyCFunction. */
 static PyTypeObject BPyPersistent_Type = {
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     /*ob_base*/ PyVarObject_HEAD_INIT(nullptr, 0)
 #else
     /*ob_base*/ PyVarObject_HEAD_INIT(&PyType_Type, 0)
@@ -248,7 +248,7 @@
 {
   PyObject *ret;
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
   BPyPersistent_Type.ob_base.ob_base.ob_type = &PyType_Type;
 #endif
 
--- blender-4.0.2-orig/source/blender/python/intern/bpy_rna.cc	2023-12-02 15:18:19.000000000 +0300
+++ blender-4.0.2/source/blender/python/intern/bpy_rna.cc	2024-01-11 14:45:43.703878400 +0300
@@ -6689,7 +6689,7 @@
     /*tp_methods*/ nullptr,
     /*tp_members*/ nullptr,
     /*tp_getset*/ nullptr,
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     /*tp_base*/ nullptr, /* Defer assignment. */
 #else
     /*tp_base*/ &PyType_Type,
@@ -7653,7 +7653,7 @@
 #endif
 
 /* For some reason MSVC complains of these. */
-#if defined(_MSC_VER)
+#if defined(_WIN32)
   pyrna_struct_meta_idprop_Type.tp_base = &PyType_Type;
 #endif
 
