diff -Naur blender-5.1.1/source/blender/python/intern/bpy_app_handlers.cc blender-5.1.1-patched/source/blender/python/intern/bpy_app_handlers.cc
--- blender-5.1.1/source/blender/python/intern/bpy_app_handlers.cc	2026-04-27 11:56:33.919472700 +0300
+++ blender-5.1.1-patched/source/blender/python/intern/bpy_app_handlers.cc	2026-04-27 11:57:07.362956500 +0300
@@ -243,7 +243,7 @@
 
 /** Dummy type because decorators can't be a #PyCFunction. */
 static PyTypeObject BPyPersistent_Type = {
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     /*ob_base*/ PyVarObject_HEAD_INIT(nullptr, 0)
 #else
     /*ob_base*/ PyVarObject_HEAD_INIT(&PyType_Type, 0)
@@ -332,7 +332,7 @@
 {
   PyObject *ret;
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
   BPyPersistent_Type.ob_base.ob_base.ob_type = &PyType_Type;
 #endif
 
diff -Naur blender-5.1.1/source/blender/python/intern/bpy_rna.cc blender-5.1.1-patched/source/blender/python/intern/bpy_rna.cc
--- blender-5.1.1/source/blender/python/intern/bpy_rna.cc	2026-04-27 11:56:33.899575500 +0300
+++ blender-5.1.1-patched/source/blender/python/intern/bpy_rna.cc	2026-04-27 11:57:07.367565200 +0300
@@ -7232,7 +7232,7 @@
     /*tp_methods*/ nullptr,
     /*tp_members*/ nullptr,
     /*tp_getset*/ nullptr,
-#if defined(_MSC_VER)
+#if defined(_WIN32)
     /*tp_base*/ nullptr, /* Defer assignment. */
 #else
     /*tp_base*/ &PyType_Type,
@@ -8760,7 +8760,7 @@
 #endif
 
 /* For some reason MSVC complains of these. */
-#if defined(_MSC_VER)
+#if defined(_WIN32)
   pyrna_struct_meta_idprop_Type.tp_base = &PyType_Type;
 #endif
 
