--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -140,10 +140,18 @@ if(WIN32 AND NOT UNIX)
     -DBLEN_VER_RC_3=${bver3}
     -DBLEN_VER_RC_4=0
   )
+  if(MINGW)
+    add_definitions(-DWINDRES)
+    if(CMAKE_C_COMPILER_ID MATCHES "GNU")
+      add_definitions(-DBLEN_VER_RC_STR_M="${BLENDER_VERSION}")
+    elseif(CMAKE_C_COMPILER_ID MATCHES "Clang")
+      add_definitions(-DLLVM_RC)
+    endif()
+  endif()
 
 
   list(APPEND SRC
     ${CMAKE_SOURCE_DIR}/release/windows/icons/winblender.rc
-    ${CMAKE_BINARY_DIR}/blender.exe.manifest
+    ${CMAKE_SOURCE_DIR}/release/windows/icons/blender.exe.manifest
   )
 endif()
 
--- a/release/windows/icons/winblender.rc
+++ b/release/windows/icons/winblender.rc
@@ -3,10 +3,13 @@
 
 #ifdef WINDRES
 	#include "winresrc.h"
+	1 RT_MANIFEST "blender.exe.manifest"
+	#ifndef LLVM_RC
 	#define IDC_STATIC (-1)
 	#define STRINGIFY_(x) #x
 	#define STRINGIFY(x) STRINGIFY_(x)
 	#define BLEN_VER_RC_STR STRINGIFY(BLEN_VER_RC_STR_M)
+	#endif
 #endif
 
 APPICON ICON DISCARDABLE "winblender.ico"
--- /dev/null	2020-02-18 15:56:11.000000000 +0300
+++ b/release/windows/icons/blender.exe.manifest	2020-02-18 15:56:02.230820700 +0300
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+    <security>
+      <requestedPrivileges>
+        <requestedExecutionLevel level="asInvoker" uiAccess="false"></requestedExecutionLevel>
+      </requestedPrivileges>
+    </security>
+  </trustInfo>
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1"> 
+        <application> 
+            <!-- Windows 10 --> 
+            <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/>
+            <!-- Windows 8.1 -->
+            <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}"/>
+            <!-- Windows 8 -->
+            <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}"/>
+            <!-- Windows 7 -->
+            <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}"/>
+            <!-- Windows Vista -->
+            <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}"/> 
+        </application> 
+  </compatibility>  
+  <dependency>
+    <dependentAssembly>
+        <assemblyIdentity
+            type="Win32"
+            name="Microsoft.Windows.Common-Controls"
+            version="6.0.0.0"
+            processorArchitecture="*"
+            publicKeyToken="6595b64144ccf1df"
+            language="*"
+        />
+    </dependentAssembly>
+  </dependency>
+</assembly>
