diff -Naur blender-4.5.2/source/blender/blenlib/intern/math_base.cc blender-4.5.2-patched/source/blender/blenlib/intern/math_base.cc
--- blender-4.5.2/source/blender/blenlib/intern/math_base.cc	2025-08-26 12:27:24.956157700 +0300
+++ blender-4.5.2-patched/source/blender/blenlib/intern/math_base.cc	2025-08-26 12:28:25.838045900 +0300
@@ -35,7 +35,7 @@
     pow2 = 1.0;
     y = (x * pow1) * pow2;
     /* if y overflows, then rounded value is exactly x */
-    if (!isfinite(y)) {
+    if (!isfinite((float)y)) {
       return x;
     }
   }
diff -Naur blender-4.5.2/source/blender/blenlib/intern/math_geom.cc blender-4.5.2-patched/source/blender/blenlib/intern/math_geom.cc
--- blender-4.5.2/source/blender/blenlib/intern/math_geom.cc	2025-08-26 12:27:24.961156600 +0300
+++ blender-4.5.2-patched/source/blender/blenlib/intern/math_geom.cc	2025-08-26 12:28:25.846046800 +0300
@@ -4056,7 +4056,7 @@
     const float dot = dot_v3v3(d_curr->dir, d_next->dir);
     const float len = d_curr->len * d_next->len;
     const float result = (len - dot) / area;
-    if (isfinite(result)) {
+    if (isfinite((float)result)) {
       return result;
     }
   }
diff -Naur blender-4.5.2/source/blender/blenlib/intern/math_solvers.cc blender-4.5.2-patched/source/blender/blenlib/intern/math_solvers.cc
--- blender-4.5.2/source/blender/blenlib/intern/math_solvers.cc	2025-08-26 12:27:24.939158600 +0300
+++ blender-4.5.2-patched/source/blender/blenlib/intern/math_solvers.cc	2025-08-26 12:28:25.855043500 +0300
@@ -84,7 +84,7 @@
 
   MEM_freeN(c1);
 
-  return isfinite(x_prev);
+  return isfinite((float)x_prev);
 }
 
 bool BLI_tridiagonal_solve_cyclic(
