--- blender-2.80/extern/audaspace/src/plugin/PluginManagerWindows.cpp.in.orig	2019-08-01 21:42:50.026016500 +0300
+++ blender-2.80/extern/audaspace/src/plugin/PluginManagerWindows.cpp.in	2019-08-01 21:43:11.476046500 +0300
@@ -29,7 +29,7 @@
 
 void *PluginManager::lookupLibrary(void *handle, const std::string &name)
 {
-	return GetProcAddress(reinterpret_cast<HMODULE>(handle), name.c_str());
+	return reinterpret_cast<void*>(GetProcAddress(reinterpret_cast<HMODULE>(handle), name.c_str()));
 }
 
 void PluginManager::closeLibrary(void *handle)
