diff -Naur blender-5.2.0/build_files/cmake/platform/dependency_targets.cmake blender-5.2.0-patched/build_files/cmake/platform/dependency_targets.cmake
--- blender-5.2.0/build_files/cmake/platform/dependency_targets.cmake	2026-08-09 00:21:05.937068300 +0200
+++ blender-5.2.0-patched/build_files/cmake/platform/dependency_targets.cmake	2026-08-09 00:21:09.259485300 +0200
@@ -425,7 +425,7 @@
   endif()
   target_include_directories(bf_deps_optional_audaspace SYSTEM INTERFACE ${AUDASPACE_C_INCLUDE_DIRS} ${AUDASPACE_PY_INCLUDE_DIRS})
   if(WITH_SYSTEM_AUDASPACE)
-    target_link_libraries(bf_deps_optional_audaspace INTERFACE ${AUDASPACE_LIBRARIES} ${AUDASPACE_PY_LIBRARIES})
+    target_link_libraries(bf_deps_optional_audaspace INTERFACE ${AUDASPACE_LIBRARIES} ${AUDASPACE_C_LIBRARIES} ${AUDASPACE_PY_LIBRARIES})
   else()
     target_link_libraries(bf_deps_optional_audaspace INTERFACE audaspace audaspace-py)
   endif()
diff -Naur blender-5.2.0/extern/glog/CMakeLists.txt blender-5.2.0-patched/extern/glog/CMakeLists.txt
--- blender-5.2.0/extern/glog/CMakeLists.txt	2026-06-03 15:03:48.000000000 +0200
+++ blender-5.2.0-patched/extern/glog/CMakeLists.txt	2026-08-09 00:21:09.247746500 +0200
@@ -75,6 +75,7 @@
   list(APPEND INC
     src/windows
   )
+  list(APPEND LIB dbghelp)
   if(MSVC)
     # Suppress warning about google::LogMessageFatal::~LogMessageFatal
     # not returning.
diff -Naur blender-5.2.0/intern/cycles/app/CMakeLists.txt blender-5.2.0-patched/intern/cycles/app/CMakeLists.txt
--- blender-5.2.0/intern/cycles/app/CMakeLists.txt	2026-08-09 00:21:05.937068300 +0200
+++ blender-5.2.0-patched/intern/cycles/app/CMakeLists.txt	2026-08-09 00:21:09.247746500 +0200
@@ -93,7 +93,7 @@
   add_executable(cycles ${SRC} ${INC} ${INC_SYS})
   unset(SRC)
 
-  target_link_libraries(cycles PRIVATE ${LIB})
+  target_link_libraries(cycles PRIVATE ${LIB} version)
 
   if(APPLE)
     if(WITH_CYCLES_STANDALONE_GUI)
diff -Naur blender-5.2.0/intern/ghost/CMakeLists.txt blender-5.2.0-patched/intern/ghost/CMakeLists.txt
--- blender-5.2.0/intern/ghost/CMakeLists.txt	2026-06-11 16:38:45.000000000 +0200
+++ blender-5.2.0-patched/intern/ghost/CMakeLists.txt	2026-08-09 00:21:09.247746500 +0200
@@ -446,6 +446,7 @@
     intern/GHOST_WindowWin32.hh
     intern/GHOST_Wintab.hh
   )
+  list(APPEND LIB comctl32 shcore dwmapi)
 
   list(APPEND SRC
     intern/GHOST_ContextWGL.cc
@@ -459,6 +460,7 @@
 
       intern/GHOST_ImeWin32.hh
     )
+    list(APPEND LIB imm32)
   endif()
 
   if(WITH_INPUT_NDOF)
diff -Naur blender-5.2.0/source/blender/blenlib/CMakeLists.txt blender-5.2.0-patched/source/blender/blenlib/CMakeLists.txt
--- blender-5.2.0/source/blender/blenlib/CMakeLists.txt	2026-06-23 18:12:48.000000000 +0200
+++ blender-5.2.0-patched/source/blender/blenlib/CMakeLists.txt	2026-08-09 00:21:09.247746500 +0200
@@ -464,6 +464,9 @@
   list(APPEND LIB
     bf_intern_uriconvert
     bf_intern_utfconv
+    version
+    dbghelp
+    shlwapi
     dxgi
   )
   list(APPEND SRC
diff -Naur blender-5.2.0/source/blender/editors/interface/CMakeLists.txt blender-5.2.0-patched/source/blender/editors/interface/CMakeLists.txt
--- blender-5.2.0/source/blender/editors/interface/CMakeLists.txt	2026-06-03 15:03:49.000000000 +0200
+++ blender-5.2.0-patched/source/blender/editors/interface/CMakeLists.txt	2026-08-09 00:21:09.253638100 +0200
@@ -155,6 +155,10 @@
   add_definitions(-DWITH_PYTHON)
 endif()
 
+if(WITH_TBB)
+  add_definitions(-DWITH_TBB)
+endif()
+
 if(WITH_INPUT_IME)
   add_definitions(-DWITH_INPUT_IME)
 endif()
diff -Naur blender-5.2.0/source/blender/editors/space_file/CMakeLists.txt blender-5.2.0-patched/source/blender/editors/space_file/CMakeLists.txt
--- blender-5.2.0/source/blender/editors/space_file/CMakeLists.txt	2026-06-09 15:52:11.000000000 +0200
+++ blender-5.2.0-patched/source/blender/editors/space_file/CMakeLists.txt	2026-08-09 00:21:09.259485300 +0200
@@ -86,6 +86,7 @@
 if(WIN32)
   list(APPEND LIB
     PRIVATE bf::intern::utfconv
+    mpr
   )
 endif()
 
diff -Naur blender-5.2.0/source/blender/editors/space_spreadsheet/CMakeLists.txt blender-5.2.0-patched/source/blender/editors/space_spreadsheet/CMakeLists.txt
--- blender-5.2.0/source/blender/editors/space_spreadsheet/CMakeLists.txt	2026-06-03 15:03:49.000000000 +0200
+++ blender-5.2.0-patched/source/blender/editors/space_spreadsheet/CMakeLists.txt	2026-08-09 00:21:09.255142600 +0200
@@ -56,6 +56,10 @@
   PRIVATE bf::windowmanager
 )
 
+if(WITH_TBB)
+  add_definitions(-DWITH_TBB)
+endif()
+
 blender_add_lib(bf_editor_space_spreadsheet "${SRC}" "${INC}" "${INC_SYS}" "${LIB}")
 
 # RNA_prototypes.hh dna_type_offsets.h
diff -Naur blender-5.2.0/source/creator/CMakeLists.txt blender-5.2.0-patched/source/creator/CMakeLists.txt
--- blender-5.2.0/source/creator/CMakeLists.txt	2026-08-09 00:21:08.595532200 +0200
+++ blender-5.2.0-patched/source/creator/CMakeLists.txt	2026-08-09 00:21:09.255142600 +0200
@@ -357,6 +357,9 @@
         ${CMAKE_SOURCE_DIR}/release/windows/icons/blender.exe.manifest
       )
     endif()
+    if(MINGW)
+      target_link_options(blender-launcher PRIVATE "-municode")
+    endif()
     target_compile_definitions(blender-launcher PRIVATE -D_UNICODE -DUNICODE)
     target_link_libraries(blender-launcher pathcch)
   endif()
