--- a/extern/glog/CMakeLists.txt
+++ b/extern/glog/CMakeLists.txt
@@ -66,6 +66,7 @@
   list(APPEND INC
     src/windows
   )
+  list(APPEND LIB dbghelp)
   if(MSVC)
     # Suppress warning about google::LogMessageFatal::~LogMessageFatal
     # not returning.
--- a/intern/cycles/app/CMakeLists.txt
+++ b/intern/cycles/app/CMakeLists.txt
@@ -80,7 +80,7 @@
   add_executable(cycles ${SRC} ${INC} ${INC_SYS})
   unset(SRC)
 
-  target_link_libraries(cycles PRIVATE ${LIB})
+  target_link_libraries(cycles PRIVATE ${LIB} version)
 
   if(APPLE)
     if(WITH_CYCLES_STANDALONE_GUI)
--- a/intern/cycles/subd/CMakeLists.txt
+++ b/intern/cycles/subd/CMakeLists.txt
@@ -25,7 +25,7 @@
 )
 
 set(LIB
-
+  ${OPENSUBDIV_LIBRARIES}
 )
 
 include_directories(${INC})
--- a/intern/cycles/scene/CMakeLists.txt
+++ b/intern/cycles/scene/CMakeLists.txt
@@ -112,7 +112,7 @@
   list(APPEND LIB
     ${OPENCOLORIO_LIBRARIES}
   )
-  if(WIN32 AND NOT USD_OVERRIDE_OPENCOLORIO)
+  if(MSVC AND NOT USD_OVERRIDE_OPENCOLORIO)
     add_definitions(-DOpenColorIO_SKIP_IMPORTS)
   endif()
 endif()
--- a/intern/cycles/util/CMakeLists.txt
+++ b/intern/cycles/util/CMakeLists.txt
@@ -32,6 +32,7 @@
 
 set(LIB
   ${TBB_LIBRARIES}
+  ${OPENIMAGEIO_LIBRARIES}
 )
 
 set(SRC_HEADERS
--- a/intern/ghost/CMakeLists.txt
+++ b/intern/ghost/CMakeLists.txt
@@ -387,6 +387,7 @@
     intern/GHOST_WindowWin32.h
     intern/GHOST_Wintab.h
   )
+  list(APPEND LIB comctl32 shcore dwmapi)
 
   if(NOT WITH_GL_EGL)
     list(APPEND SRC
@@ -404,6 +405,7 @@
 
       intern/GHOST_ImeWin32.h
     )
+    list(APPEND LIB imm32)
   endif()
 
   if(WITH_INPUT_NDOF)
--- a/source/blender/blenlib/CMakeLists.txt
+++ b/source/blender/blenlib/CMakeLists.txt
@@ -447,6 +447,8 @@
   )
   list(APPEND LIB
     bf_intern_utfconv
+    version
+    dbghelp
     dxguid
     dxgi
   )
--- a/source/blender/editors/interface/CMakeLists.txt
+++ b/source/blender/editors/interface/CMakeLists.txt
@@ -98,6 +98,10 @@
   add_definitions(-DWITH_PYTHON)
 endif()
 
+if(WITH_TBB)
+  add_definitions(-DWITH_TBB)
+endif()
+
 if(WIN32 OR APPLE)
   if(WITH_INPUT_IME)
     add_definitions(-DWITH_INPUT_IME)
--- a/source/blender/editors/space_spreadsheet/CMakeLists.txt
+++ b/source/blender/editors/space_spreadsheet/CMakeLists.txt
@@ -62,6 +62,10 @@
   add_definitions(-DWITH_OPENVDB ${OPENVDB_DEFINITIONS})
 endif()
 
+if(WITH_TBB)
+  add_definitions(-DWITH_TBB)
+endif()
+
 blender_add_lib(bf_editor_space_spreadsheet "${SRC}" "${INC}" "${INC_SYS}" "${LIB}")
 
 # RNA_prototypes.h
--- a/source/blender/python/generic/CMakeLists.txt
+++ b/source/blender/python/generic/CMakeLists.txt
@@ -40,6 +40,7 @@
 )
 
 set(LIB
+  ${OPENGL_gl_LIBRARY}
   ${Epoxy_LIBRARIES}
   ${PYTHON_LINKFLAGS}
   ${PYTHON_LIBRARIES}
--- a/source/blender/render/CMakeLists.txt
+++ b/source/blender/render/CMakeLists.txt
@@ -79,5 +79,10 @@
   add_definitions(-DWITH_FREESTYLE)
 endif()
 
+if(WITH_OPENCOLORIO)
+  list(APPEND LIB
+    ${OPENCOLORIO_LIBRARIES}
+  )
+endif()
 
 blender_add_lib_nolist(bf_render "${SRC}" "${INC}" "${INC_SYS}" "${LIB}")
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -305,6 +305,9 @@
       ${CMAKE_SOURCE_DIR}/release/windows/icons/winblender.rc
       ${CMAKE_SOURCE_DIR}/release/windows/icons/blender.exe.manifest
     )
+    if(MINGW)
+      target_link_options(blender-launcher PRIVATE "-municode")
+    endif()
     target_compile_definitions (blender-launcher PRIVATE -D_UNICODE -DUNICODE)
     target_link_libraries(blender-launcher pathcch)
   endif()
