diff -Naur blender-5.1.1/extern/mantaflow/CMakeLists.txt blender-5.1.1-patched/extern/mantaflow/CMakeLists.txt
--- blender-5.1.1/extern/mantaflow/CMakeLists.txt	2026-04-27 11:59:02.153194300 +0300
+++ blender-5.1.1-patched/extern/mantaflow/CMakeLists.txt	2026-04-27 11:59:32.500111800 +0300
@@ -77,7 +77,7 @@
 endif()
 
 if(WITH_OPENVDB)
-  if(WIN32)
+  if(MSVC)
     # OpenVDB emits lots of these, they should be suppressed through other
     # means but MSVC 16.8/16.9 has broken this functionality, so C4251 is
     # suppressed here explicitly. See
