--- blender-2.83.0/intern/ghost/intern/GHOST_XrContext.cpp.orig	2020-06-22 14:25:10.109252500 +0300
+++ blender-2.83.0/intern/ghost/intern/GHOST_XrContext.cpp	2020-06-22 14:25:52.778779600 +0300
@@ -178,7 +178,7 @@
   }
 }
 
-static XrBool32 debug_messenger_func(XrDebugUtilsMessageSeverityFlagsEXT /*messageSeverity*/,
+static XrBool32 __stdcall debug_messenger_func(XrDebugUtilsMessageSeverityFlagsEXT /*messageSeverity*/,
                                      XrDebugUtilsMessageTypeFlagsEXT /*messageTypes*/,
                                      const XrDebugUtilsMessengerCallbackDataEXT *callbackData,
                                      void * /*userData*/)
