diff -Naur blender-5.1.1/intern/ghost/intern/GHOST_XrContext.cc blender-5.1.1-patched/intern/ghost/intern/GHOST_XrContext.cc
--- blender-5.1.1/intern/ghost/intern/GHOST_XrContext.cc	2026-04-27 11:59:26.198995400 +0300
+++ blender-5.1.1-patched/intern/ghost/intern/GHOST_XrContext.cc	2026-04-27 11:59:57.380547100 +0300
@@ -197,7 +197,7 @@
   }
 }
 
-static XrBool32 debug_messenger_func(XrDebugUtilsMessageSeverityFlagsEXT /*messageSeverity*/,
+static XrBool32 __stdcall debug_messenger_func(XrDebugUtilsMessageSeverityFlagsEXT /*messageSeverity*/,
                                      XrDebugUtilsMessageTypeFlagsEXT /*messageTypes*/,
                                      const XrDebugUtilsMessengerCallbackDataEXT *callbackData,
                                      void * /*user_data*/)
