--- blender-2.90.1/intern/cycles/CMakeLists.txt.orig	2020-10-08 21:28:57.427384700 +0300
+++ blender-2.90.1/intern/cycles/CMakeLists.txt	2020-10-08 21:29:04.707761700 +0300
@@ -233,7 +233,6 @@
 
 if(WITH_CYCLES_EMBREE)
   add_definitions(-DWITH_EMBREE)
-  add_definitions(-DEMBREE_STATIC_LIB)
   include_directories(
     SYSTEM
     ${EMBREE_INCLUDE_DIRS}
--- blender-2.90.1/build_files/cmake/Modules/FindEmbree.cmake.orig	2020-10-08 22:57:05.690074300 +0300
+++ blender-2.90.1/build_files/cmake/Modules/FindEmbree.cmake	2020-10-08 22:59:36.200482300 +0300
@@ -40,14 +40,6 @@
 
 SET(_embree_FIND_COMPONENTS
   embree3
-  embree_sse42
-  embree_avx
-  embree_avx2
-  lexers
-  math
-  simd
-  sys
-  tasking
 )
 
 SET(_embree_LIBRARIES)
