diff -Naur blender-4.5.2/intern/cycles/app/opengl/display_driver.cpp blender-4.5.2-patched/intern/cycles/app/opengl/display_driver.cpp
--- blender-4.5.2/intern/cycles/app/opengl/display_driver.cpp	2025-08-26 12:35:26.232196100 +0300
+++ blender-4.5.2-patched/intern/cycles/app/opengl/display_driver.cpp	2025-08-26 18:33:20.350225800 +0300
@@ -188,7 +188,7 @@
  * Drawing.
  */
 
-void OpenGLDisplayDriver::clear()
+void OpenGLDisplayDriver::zero()
 {
   texture_.need_zero = true;
 }
diff -Naur blender-4.5.2/intern/cycles/app/opengl/display_driver.h blender-4.5.2-patched/intern/cycles/app/opengl/display_driver.h
--- blender-4.5.2/intern/cycles/app/opengl/display_driver.h	2025-08-26 12:35:26.234742300 +0300
+++ blender-4.5.2-patched/intern/cycles/app/opengl/display_driver.h	2025-08-26 18:32:58.949069200 +0300
@@ -24,7 +24,7 @@
   void graphics_interop_activate() override;
   void graphics_interop_deactivate() override;
 
-  void clear() override;
+  void zero() override;
 
   void set_zoom(const float zoom_x, const float zoom_y);
 
