--- a/build_files/cmake/Modules/FindTBB.cmake
+++ b/build_files/cmake/Modules/FindTBB.cmake
@@ -37,6 +37,10 @@
     include
 )
 
+find_package(TBB CONFIG)
+if(TBB_FOUND)
+  set(TBB_LIBRARY TBB::tbb)
+else()
 find_library(TBB_LIBRARY
   NAMES
     tbb
@@ -45,6 +49,7 @@
   PATH_SUFFIXES
     lib64 lib
   )
+endif()
 
 # handle the QUIETLY and REQUIRED arguments and set TBB_FOUND to TRUE if
 # all listed variables are TRUE
