--- a/intern/cycles/CMakeLists.txt
+++ b/intern/cycles/CMakeLists.txt
@@ -211,6 +211,9 @@
   set(RTTI_DISABLE_FLAGS "/GR- -DBOOST_NO_RTTI -DBOOST_NO_TYPEID")
 elseif(CMAKE_COMPILER_IS_GNUCC OR (CMAKE_C_COMPILER_ID MATCHES "Clang"))
   set(RTTI_DISABLE_FLAGS "-fno-rtti -DBOOST_NO_RTTI -DBOOST_NO_TYPEID")
+  if(MINGW)
+    set(RTTI_DISABLE_FLAGS "-DBOOST_NO_RTTI -DBOOST_NO_TYPEID")
+  endif()
 endif()
 
 # Definitions and Includes
