diff -Naur blender-4.5.2/intern/cycles/util/simd.h blender-4.5.2-patched/intern/cycles/util/simd.h
--- blender-4.5.2/intern/cycles/util/simd.h	2025-08-26 12:32:08.012792800 +0300
+++ blender-4.5.2-patched/intern/cycles/util/simd.h	2025-08-26 12:34:24.674170900 +0300
@@ -18,7 +18,9 @@
 #  include <intrin.h>
 #elif (defined(__x86_64__) || defined(__i386__))
 #  include <x86intrin.h>
-#elif defined(__KERNEL_NEON__)
+#endif
+
+#if defined(__KERNEL_NEON__)
 #  define SSE2NEON_PRECISE_MINMAX 1
 #  include <sse2neon.h>
 #endif
