From 19bef5e3211acc764d83f768ecb49f4b8dc0551e Mon Sep 17 00:00:00 2001
From: GH Cao <driver1998.ms@outlook.com>
Date: Sun, 4 Dec 2022 21:48:58 +0800
Subject: [PATCH 2/2] Enable SSE2NEON on Windows ARM

---
 intern/cycles/util/util_simd.h                          | 4 +++-
 1 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/intern/cycles/util/simd.h b/intern/cycles/util/simd.h
index e3e4127..6fe576b 100644
--- a/intern/cycles/util/simd.h
+++ b/intern/cycles/util/simd.h
@@ -31,7 +31,9 @@
 #  include <intrin.h>
 #elif (defined(__x86_64__) || defined(__i386__))
 #  include <x86intrin.h>
-#elif defined(__KERNEL_NEON__)
+#endif
+
+#if defined(__KERNEL_NEON__)
 #  define SSE2NEON_PRECISE_MINMAX 1
 #  include <sse2neon.h>
 #endif
-- 
2.38.1

