diff -Naur blender-5.1.1/intern/cycles/util/simd.h blender-5.1.1-patched/intern/cycles/util/simd.h
--- blender-5.1.1/intern/cycles/util/simd.h	2026-04-27 12:00:15.439262100 +0300
+++ blender-5.1.1-patched/intern/cycles/util/simd.h	2026-04-27 12:00:45.735116500 +0300
@@ -18,7 +18,9 @@
 #  include <intrin.h>
 #elif (defined(__x86_64__) || defined(__i386__))
 #  include <x86intrin.h>
-#elif defined(__KERNEL_NEON__)
+#endif
+
+#if defined(__KERNEL_NEON__)
 #  define SSE2NEON_PRECISE_MINMAX 1
 #  include <sse2neon.h>
 #endif
