--- blender-2.93.14/build_files/cmake/platform/platform_win32_mingw64.cmake.orig	2023-02-22 19:42:08.991589800 +0100
+++ blender-2.93.14/build_files/cmake/platform/platform_win32_mingw64.cmake	2023-02-22 19:55:59.658475000 +0100
@@ -205,13 +205,13 @@
 	mark_as_advanced(FFMPEG)
 
 	# lame, but until we have proper find module for ffmpeg
-	if(EXISTS "${FFMPEG}/include/ffmpeg/")
-		list(APPEND FFMPEG_INCLUDE_DIRS "${FFMPEG}/include/ffmpeg")
+	if(EXISTS "${FFMPEG}/include/ffmpeg4.4/")
+		list(APPEND FFMPEG_INCLUDE_DIRS "${FFMPEG}/include/ffmpeg4.4")
 	endif()
 	# end lameness
 
 	mark_as_advanced(FFMPEG_LIBRARIES)
-	set(FFMPEG_LIBPATH ${FFMPEG}/lib)
+	set(FFMPEG_LIBPATH ${FFMPEG}/lib/ffmpeg4.4)
 endif()
 
 if(WITH_FFTW3)
@@ -414,6 +414,10 @@
 	endif()
 endif()
 
+if(WITH_CPU_SIMD AND SUPPORT_NEON_BUILD)
+  find_package_wrapper(sse2neon)
+endif()
+
 if(WITH_LLVM)
 	find_package_wrapper(LLVM)
 
