--- blender-5.0.0/intern/ghost/intern/GHOST_Wintab.cc.orig	2025-12-03 12:47:20.433297500 +0300
+++ blender-5.0.0/intern/ghost/intern/GHOST_Wintab.cc	2025-12-03 12:49:12.158516700 +0300
@@ -161,6 +161,7 @@
   LPTOP_LEVEL_EXCEPTION_FILTER current_filter = SetUnhandledExceptionFilter(nullptr);
   SetUnhandledExceptionFilter(current_filter);
 
+#ifdef _MSC_VER
   /* __except and __finally cannot be used together, as such a second nested __try block is needed.
    */
   __try
@@ -180,6 +181,11 @@
      */
     SetUnhandledExceptionFilter(current_filter);
   }
+#else
+    try {
+      return GHOST_Wintab::loadWintabUnsafe(hwnd);
+    } catch (...) {}
+#endif
 
   return nullptr;
 }
