diff -Naur blender-5.1.1/intern/ghost/intern/GHOST_Wintab.cc blender-5.1.1-patched/intern/ghost/intern/GHOST_Wintab.cc
--- blender-5.1.1/intern/ghost/intern/GHOST_Wintab.cc	2026-04-27 12:01:07.610635200 +0300
+++ blender-5.1.1-patched/intern/ghost/intern/GHOST_Wintab.cc	2026-04-27 12:01:32.873661300 +0300
@@ -161,6 +161,7 @@
   LPTOP_LEVEL_EXCEPTION_FILTER current_filter = SetUnhandledExceptionFilter(nullptr);
   SetUnhandledExceptionFilter(current_filter);
 
+#ifdef _MSC_VER
   /* __except and __finally cannot be used together, as such a second nested __try block is needed.
    */
   __try
@@ -180,6 +181,11 @@
      */
     SetUnhandledExceptionFilter(current_filter);
   }
+#else
+    try {
+      return GHOST_Wintab::loadWintabUnsafe(hwnd);
+    } catch (...) {}
+#endif
 
   return nullptr;
 }
