--- blender-4.4.3/build_files/cmake/Modules/FindVulkan.cmake.orig	2025-06-19 07:40:37.158134800 +0200
+++ blender-4.4.3/build_files/cmake/Modules/FindVulkan.cmake	2025-06-19 07:44:20.564818700 +0200
@@ -34,15 +34,25 @@
   PATH_SUFFIXES
     include
 )
-
-find_library(VULKAN_LIBRARY
-  NAMES
-    vulkan
-  HINTS
-    ${_vulkan_SEARCH_DIRS}
-  PATH_SUFFIXES
-    lib
-)
+if(WIN32)
+  find_library(VULKAN_LIBRARY
+    NAMES
+      vulkan-1
+    HINTS
+      ${_vulkan_SEARCH_DIRS}
+    PATH_SUFFIXES
+      lib
+  )
+else()
+  find_library(VULKAN_LIBRARY
+    NAMES
+      vulkan
+    HINTS
+      ${_vulkan_SEARCH_DIRS}
+    PATH_SUFFIXES
+      lib
+  )
+endif()
 
 # handle the QUIETLY and REQUIRED arguments and set VULKAN_FOUND to TRUE if
 # all listed variables are TRUE
