diff -Naur blender-4.5.2/build_files/cmake/Modules/FindVulkan.cmake blender-4.5.2-patched/build_files/cmake/Modules/FindVulkan.cmake
--- blender-4.5.2/build_files/cmake/Modules/FindVulkan.cmake	2025-08-26 12:34:49.593611000 +0300
+++ blender-4.5.2-patched/build_files/cmake/Modules/FindVulkan.cmake	2025-08-26 12:35:30.735798300 +0300
@@ -34,15 +34,25 @@
   PATH_SUFFIXES
     include
 )
-
-find_library(VULKAN_LIBRARY
-  NAMES
-    vulkan
-  HINTS
-    ${_vulkan_SEARCH_DIRS}
-  PATH_SUFFIXES
-    lib
-)
+if(WIN32)
+  find_library(VULKAN_LIBRARY
+    NAMES
+      vulkan-1
+    HINTS
+      ${_vulkan_SEARCH_DIRS}
+    PATH_SUFFIXES
+      lib
+  )
+else()
+  find_library(VULKAN_LIBRARY
+    NAMES
+      vulkan
+    HINTS
+      ${_vulkan_SEARCH_DIRS}
+    PATH_SUFFIXES
+      lib
+  )
+endif()
 
 # handle the QUIETLY and REQUIRED arguments and set VULKAN_FOUND to TRUE if
 # all listed variables are TRUE
