diff -Naur blender-5.1.1/build_files/cmake/Modules/FindVulkan.cmake blender-5.1.1-patched/build_files/cmake/Modules/FindVulkan.cmake
--- blender-5.1.1/build_files/cmake/Modules/FindVulkan.cmake	2026-04-27 12:01:26.863543400 +0300
+++ blender-5.1.1-patched/build_files/cmake/Modules/FindVulkan.cmake	2026-04-27 12:01:56.810549500 +0300
@@ -34,15 +34,25 @@
   PATH_SUFFIXES
     include
 )
-
-find_library(VULKAN_LIBRARY
-  NAMES
-    vulkan
-  HINTS
-    ${_vulkan_SEARCH_DIRS}
-  PATH_SUFFIXES
-    lib
-)
+if(WIN32)
+  find_library(VULKAN_LIBRARY
+    NAMES
+      vulkan-1
+    HINTS
+      ${_vulkan_SEARCH_DIRS}
+    PATH_SUFFIXES
+      lib
+  )
+else()
+  find_library(VULKAN_LIBRARY
+    NAMES
+      vulkan
+    HINTS
+      ${_vulkan_SEARCH_DIRS}
+    PATH_SUFFIXES
+      lib
+  )
+endif()
 
 # handle the QUIETLY and REQUIRED arguments and set VULKAN_FOUND to TRUE if
 # all listed variables are TRUE
