--- blender-2.73/intern/cycles/kernel/osl/osl_closures.cpp.orig	2015-01-11 15:22:14.583200000 +0300
+++ blender-2.73/intern/cycles/kernel/osl/osl_closures.cpp	2015-01-11 15:22:23.334800000 +0300
@@ -189,7 +189,7 @@
 	/* optimization: it's possible to not use a prepare function at all and
 	 * only initialize the actual class when accessing the closure component
 	 * data, but then we need to map the id to the class somehow */
-	ss->register_closure(name, id, params, prepare, NULL, 16);
+	ss->register_closure(name, id, params, prepare, NULL/*, 16*/);
 }
 
 void OSLShader::register_closures(OSLShadingSystem *ss_)
