--- a/Makefile.PL
+++ b/Makefile.PL
@@ -13,7 +13,8 @@
 require './W32ATest.pm';
 
 my $is_64bit_build = ($Config{ptrsize} == 8);
-my $is_msvc_compiler = ($Config{cc} =~ /cl/);
+my $is_msvc_compiler = ($Config{cc} eq "cl");
+my $is_clang_compiler = ($Config{cc} eq "clang");
 
 #maybe this CC will banned in the future due to SEGV in DllMain
 #die "Mingw 3.4.5 will generate unloadble DLLs, upgrade your GCC"
@@ -134,7 +135,7 @@
 	'.($is_64bit_build ? 'ml64' : 'ml').' $(ASFLAGS) $(PERL_DEFINES)  -c $<
 
 .s$(OBJ_EXT):
-	$(AS) $(ASFLAGS) $(PERL_DEFINES) $< -o $*$(OBJ_EXT)
+	$(AS) $(ASFLAGS) $(PERL_DEFINES) '.($is_clang_compiler ? '-c ' : '').'$< -o $*$(OBJ_EXT)
 
 #target unused
 #make GCC issue "push" instructions rather than "mov" to esp+offset, this is the magic
@@ -200,7 +201,7 @@
     return $dlib.'PERL_DEFINES = '.(@defs?'-D':'').join(' -D', @defs)."\n" if $is_msvc_compiler;
 #convert -DFOO into GNU AS compatible defines for GNU AS macros
     my $defs = '';
-    $defs .= '-defsym '.$_.'=1 ' foreach @defs;
+    $defs .= ($is_clang_compiler?'-Wa,':'').'-defsym'.($is_clang_compiler?',':' ').$_.'=1 ' foreach @defs;
     return $dlib.'PERL_DEFINES = '.$defs."\n";
 }
 
