--- libunistring-0.9.9/lib/unictype/joininggroup_name.c.orig	2018-04-30 15:17:10.646207300 -0400
+++ libunistring-0.9.9/lib/unictype/joininggroup_name.c	2018-04-30 15:28:06.447600200 -0400
@@ -49,7 +49,12 @@ static const struct joining_group_string
 
 static const int joining_group_index[] =
   {
-#define ELEM(tag,string) (int)(long)&((struct joining_group_stringpool_t *)0)->tag,
+
+#ifdef _WIN32
+#define ELEM(tag,string) (intptr_t)&((struct joining_group_stringpool_t *)0)->tag,
+#else
+#define ELEM(tag,string) (int)(long)&((struct joining_group_stringpool_t *)0)->tag,
+#endif
 #include "unictype/joininggroup_name.h"
 #undef ELEM
   };
