diff -bur LuaJIT-2.1.0-beta3-orig/src/Makefile LuaJIT-2.1.0-beta3/src/Makefile
--- LuaJIT-2.1.0-beta3-orig/src/Makefile	2023-01-26 22:40:03.639259400 -0700
+++ LuaJIT-2.1.0-beta3/src/Makefile	2023-01-26 22:40:18.635694100 -0700
@@ -24,7 +24,7 @@
 # removing the '#' in front of them. Make sure you force a full recompile
 # with "make clean", followed by "make" if you change any options.
 #
-DEFAULT_CC = gcc
+DEFAULT_CC = cc
 #
 # LuaJIT builds as a native 32 or 64 bit binary by default.
 CC= $(DEFAULT_CC)
