diff -urN twisted-iocpsupport-1.0.2-orig/twisted_iocpsupport/winsock_pointers.h twisted-iocpsupport-1.0.2/twisted_iocpsupport/winsock_pointers.h
--- twisted-iocpsupport-1.0.2-orig/twisted_iocpsupport/winsock_pointers.h	2021-09-06 22:27:25.000000000 +0800
+++ twisted-iocpsupport-1.0.2/twisted_iocpsupport/winsock_pointers.h	2021-12-12 22:22:56.431463400 +0800
@@ -6,7 +6,7 @@
 #include <windows.h>
 
 int initWinsockPointers();
-BOOL
+static BOOL
 (PASCAL FAR * lpAcceptEx)(
     IN SOCKET sListenSocket,
     IN SOCKET sAcceptSocket,
@@ -17,7 +17,7 @@
     OUT LPDWORD lpdwBytesReceived,
     IN LPOVERLAPPED lpOverlapped
     );
-VOID
+static VOID
 (PASCAL FAR * lpGetAcceptExSockaddrs)(
     IN PVOID lpOutputBuffer,
     IN DWORD dwReceiveDataLength,
@@ -28,7 +28,7 @@
     OUT struct sockaddr **RemoteSockaddr,
     OUT LPINT RemoteSockaddrLength
     );
-BOOL
+static BOOL
 (PASCAL FAR * lpConnectEx) (
     IN SOCKET s,
     IN const struct sockaddr FAR *name,
