diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5787bd9..5bbe346 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -158,7 +158,7 @@ install (
     FILES
         LICENSE.txt
         README.md
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
 )
 
 # cpack mistakenly detects Mingw-w64 as win32
diff --git a/sample/CMakeLists.txt b/sample/CMakeLists.txt
index 50a5067..246e22a 100644
--- a/sample/CMakeLists.txt
+++ b/sample/CMakeLists.txt
@@ -6,5 +6,5 @@ add_executable (sample
 add_dependencies (sample exchndl_implib)
 target_link_libraries (sample PRIVATE exchndl_implib)
 
-install (FILES sample.cpp DESTINATION sample)
-install (FILES sample.mak DESTINATION sample RENAME Makefile)
+install (FILES sample.cpp DESTINATION share/drmingw/sample)
+install (FILES sample.mak DESTINATION share/drmingw/sample RENAME Makefile)
diff --git a/thirdparty/libdwarf.cmake b/thirdparty/libdwarf.cmake
index 6b7513d..bc1de9c 100644
--- a/thirdparty/libdwarf.cmake
+++ b/thirdparty/libdwarf.cmake
@@ -102,6 +102,6 @@ target_link_libraries (dwarf PRIVATE z)
 
 install (
     FILES libdwarf/src/lib/libdwarf/LIBDWARFCOPYRIGHT
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
     RENAME LICENSE-libdwarf.txt
 )
diff --git a/thirdparty/zlib.cmake b/thirdparty/zlib.cmake
index 44c4dd6..b1c58d1 100644
--- a/thirdparty/zlib.cmake
+++ b/thirdparty/zlib.cmake
@@ -30,6 +30,6 @@ target_include_directories (z PUBLIC zlib)
 
 install (
     FILES zlib/README
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
     RENAME LICENSE-zlib.txt
 )
