--- drmingw/CMakeLists.txt.orig	2015-06-21 14:48:57.330400000 +0300
+++ drmingw/CMakeLists.txt	2015-06-21 14:49:03.554800000 +0300
@@ -60,10 +60,6 @@
     set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -static-libstdc++")
 endif ()
 
-if (CMAKE_SIZEOF_VOID_P EQUAL 4)
-    set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--kill-at")
-endif ()
-
 # Put all executables into top-level bin subdirectory
 set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
--- drmingw/src/exchndl/CMakeLists.txt.orig	2015-06-20 21:46:52.998600000 +0300
+++ drmingw/src/exchndl/CMakeLists.txt	2015-06-20 21:46:57.741000000 +0300
@@ -1,6 +1,11 @@
-add_library (exchndl MODULE
+if (CMAKE_SIZEOF_VOID_P EQUAL 4)
+    set (EXCHNDL_DEF exchndl32.def)
+else ()
+    set (EXCHNDL_DEF exchndl.def)
+endif ()
+add_library (exchndl SHARED
     exchndl.c
-    exchndl.def
+    ${EXCHNDL_DEF}
     version.rc
 )
 
@@ -15,4 +15,8 @@
     PREFIX ""
 )
 
-install (TARGETS exchndl LIBRARY DESTINATION bin)
+install (TARGETS exchndl
+         RUNTIME DESTINATION bin
+         LIBRARY DESTINATION lib
+         ARCHIVE DESTINATION lib
+)
--- /dev/null	2015-06-21 14:20:14.000000000 +0300
+++ drmingw/src/exchndl/exchndl32.def	2015-06-21 14:18:57.437200000 +0300
@@ -0,0 +1,2 @@
+EXPORTS
+    SetLogFileNameA@4
--- drmingw/src/mgwhelp/CMakeLists.txt.orig	2015-06-20 21:56:28.342000000 +0300
+++ drmingw/src/mgwhelp/CMakeLists.txt	2015-06-20 22:08:56.832200000 +0300
@@ -1,7 +1,9 @@
 if (CMAKE_SIZEOF_VOID_P EQUAL 4)
     set (MGWHELP_DEF mgwhelp32.def)
+    set (DBGHELP_DEF dbghelp.def)
 else ()
     set (MGWHELP_DEF mgwhelp64.def)
+    set (DBGHELP_DEF dbghelp64.def)
 endif ()
 
 add_library (mgwhelp SHARED
@@ -23,7 +23,11 @@
     LINKER_LANGUAGE CXX # for __cxa_demangle
 )
 
-install (TARGETS mgwhelp RUNTIME DESTINATION bin)
+install (TARGETS mgwhelp
+         RUNTIME DESTINATION bin
+         LIBRARY DESTINATION lib
+         ARCHIVE DESTINATION lib
+)
 
 if (CMAKE_SIZEOF_VOID_P EQUAL 4)
     # XXX: We need to use a custom import lib to avoid link failures
@@ -42,8 +44,8 @@
     set (MGWHELP_IMPLIB ${CMAKE_CURRENT_BINARY_DIR}/libmgwhelp.a)
     add_custom_command (
         OUTPUT ${MGWHELP_IMPLIB}
-	COMMAND ${DLLTOOL} --output-lib ${MGWHELP_IMPLIB} --dllname mgwhelp.dll --kill-at --input-def=${CMAKE_CURRENT_SOURCE_DIR}/dbghelp.def
-	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/dbghelp.def
+	COMMAND ${DLLTOOL} --output-lib ${MGWHELP_IMPLIB} --dllname mgwhelp.dll --input-def=${CMAKE_CURRENT_SOURCE_DIR}/${DBGHELP_DEF}
+	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${DBGHELP_DEF}
     )
 else ()
     set (MGWHELP_IMPLIB mgwhelp)
--- /dev/null	2015-06-22 09:25:15.000000000 +0300
+++ drmingw/src/mgwhelp/dbghelp64.def	2015-06-22 09:24:34.433800000 +0300
@@ -0,0 +1,109 @@
+LIBRARY dbghelp.dll
+
+EXPORTS
+	EnumDirTree
+	EnumDirTreeW
+	EnumerateLoadedModules
+	EnumerateLoadedModules64
+	EnumerateLoadedModulesW64
+	ExtensionApiVersion
+	FindDebugInfoFile
+	FindDebugInfoFileEx
+	FindExecutableImage
+	FindExecutableImageEx
+	FindExecutableImageExW
+	GetTimestampForLoadedLibrary
+	ImageDirectoryEntryToData
+	ImageDirectoryEntryToDataEx
+	ImageNtHeader
+	ImageRvaToSection
+	ImageRvaToVa
+	ImagehlpApiVersion
+	ImagehlpApiVersionEx
+	MakeSureDirectoryPathExists
+	MapDebugInformation
+	MiniDumpReadDumpStream
+	MiniDumpWriteDump
+	SearchTreeForFile
+	SearchTreeForFileW
+	StackWalk
+	StackWalk64
+	SymAddSymbol
+	SymAddSymbolW
+	SymCleanup
+	SymEnumLines
+	SymEnumSourceFiles
+	SymEnumSymbols
+	SymEnumSymbolsW
+	SymEnumTypes
+	SymEnumTypesW
+	SymEnumerateModules
+	SymEnumerateModules64
+	SymEnumerateModulesW64
+	SymEnumerateSymbols
+	SymEnumerateSymbols64
+	SymFindFileInPath
+	SymFindFileInPathW
+	SymFromAddr
+	SymFromAddrW
+	SymFromName
+	SymFunctionTableAccess
+	SymFunctionTableAccess64
+	SymGetLineFromAddr
+	SymGetLineFromAddr64
+	SymGetLineFromAddrW64
+	SymGetLineNext
+	SymGetLineNext64
+	SymGetLinePrev
+	SymGetLinePrev64
+	SymGetModuleBase
+	SymGetModuleBase64
+	SymGetModuleInfo
+	SymGetModuleInfo64
+	SymGetModuleInfoW
+	SymGetModuleInfoW64
+	SymGetOptions
+	SymGetSearchPath
+	SymGetSearchPathW
+	SymGetSourceFileToken
+	SymGetSourceFileTokenW
+	SymGetSymFromAddr
+	SymGetSymFromAddr64
+	SymGetSymFromName
+	SymGetSymFromName64
+	SymGetSymNext
+	SymGetSymNext64
+	SymGetSymPrev
+	SymGetSymPrev64
+	SymGetTypeFromName
+	SymGetTypeInfo
+	SymInitialize
+	SymInitializeW
+	SymLoadModule
+	SymLoadModule64
+	SymLoadModuleEx
+	SymLoadModuleExW
+	SymMatchFileName
+	SymMatchFileNameW
+	SymMatchString
+	SymRefreshModuleList
+	SymRegisterCallback
+	SymRegisterCallback64
+	SymRegisterCallbackW64
+	SymRegisterFunctionEntryCallback
+	SymRegisterFunctionEntryCallback64
+	SymSearch
+	SymSearchW
+	SymSetContext
+	SymSetOptions
+	SymSetParentWindow
+	SymSetScopeFromAddr
+	SymSetSearchPath
+	SymSetSearchPathW
+	SymUnDName
+	SymUnDName64
+	SymUnloadModule
+	SymUnloadModule64
+	UnDecorateSymbolName
+	UnmapDebugInformation
+	WinDbgExtensionDllInit
