diff -aurN 000/CMakeLists.txt 001/CMakeLists.txt
--- 000/CMakeLists.txt	2015-06-26 13:00:48.000000000 -0300
+++ 001/CMakeLists.txt	2015-06-30 12:07:28.121666600 -0300
@@ -133,7 +133,7 @@
     FILES
         LICENSE.txt
         README.md
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
 )
 
 # cpack mistakenly detects Mingw-w64 as win32
diff -aurN 000/sample/CMakeLists.txt 001/sample/CMakeLists.txt
--- 000/sample/CMakeLists.txt	2015-06-26 13:00:48.000000000 -0300
+++ 001/sample/CMakeLists.txt	2015-06-30 12:07:28.121666600 -0300
@@ -6,5 +6,5 @@
 add_dependencies (sample exchndl_implib)
 target_link_libraries (sample ${EXCHNDL_IMPLIB})
 
-install (FILES sample.cpp DESTINATION sample)
-install (FILES sample.mak DESTINATION sample RENAME Makefile)
+install (FILES sample.cpp DESTINATION share/drmingw/sample)
+install (FILES sample.mak DESTINATION share/drmingw/sample RENAME Makefile)
diff -aurN 000/thirdparty/dwarf/CMakeLists.txt 001/thirdparty/dwarf/CMakeLists.txt
--- 000/thirdparty/dwarf/CMakeLists.txt	2015-06-26 13:00:48.000000000 -0300
+++ 001/thirdparty/dwarf/CMakeLists.txt	2015-06-30 12:07:28.121666600 -0300
@@ -38,6 +38,6 @@
 
 install (
     FILES LIBDWARFCOPYRIGHT
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
     RENAME LICENSE-libdwarf.txt
 )
diff -aurN 000/thirdparty/zlib/CMakeLists.txt 001/thirdparty/zlib/CMakeLists.txt
--- 000/thirdparty/zlib/CMakeLists.txt	2015-06-26 13:00:48.000000000 -0300
+++ 001/thirdparty/zlib/CMakeLists.txt	2015-06-30 12:07:28.121666600 -0300
@@ -38,6 +38,6 @@
 
 install (
     FILES README
-    DESTINATION doc
+    DESTINATION share/doc/drmingw
     RENAME LICENSE-zlib.txt
 )
