From 52d9003e807dc18c460bed6b3654a97c996ccfe4 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 08:56:28 +0300
Subject: [PATCH 03/27] Define AMD64 for 64-bit mingw.

---
 configure.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index fe2cc11d0a..21384bf391 100644
--- a/configure.in
+++ b/configure.in
@@ -464,7 +464,12 @@ dnl CPU_TYPE=ppc64
     EDITLINE_FLG=N
     RAW_DEVICES_FLG=N
     SHRLIB_EXT=dll
-	ICU_PLATFORM=MinGW
+    ICU_PLATFORM=MinGW
+    case "$target" in
+        x86_64-*-mingw*)
+            AC_DEFINE(AMD64, 1, [Define this if CPU is amd64])
+        ;;
+    esac
     ;;
 
   *)
-- 
2.13.0

