From 95d0091b613b32a904db451147e56fdde4f5c122 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 21:44:53 +0300
Subject: [PATCH 11/27] mingw-w64 have fenv.h

---
 src/common/classes/FpeControl.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/classes/FpeControl.h b/src/common/classes/FpeControl.h
index ecc334b805..f5587ea564 100644
--- a/src/common/classes/FpeControl.h
+++ b/src/common/classes/FpeControl.h
@@ -30,7 +30,7 @@
 #define CLASSES_FPE_CONTROL_H
 
 #include <math.h>
-#if defined(WIN_NT)
+#if defined(_MSC_VER)
 #include <float.h>
 #else
 #include <fenv.h>
@@ -78,7 +78,7 @@ public:
 		}
 	}
 
-#if defined(WIN_NT)
+#if defined(_MSC_VER)
 	static void maskAll() throw()
 	{
 		_clearfp(); // always call _clearfp() before setting control word
-- 
2.13.0

