From b44435148212da2bf91490e12722ef59c8efd3ea Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Fri, 21 Nov 2014 21:44:53 +0300
Subject: [PATCH 13/27] Fix source extension.

---
 builds/posix/prefix.mingw | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builds/posix/prefix.mingw b/builds/posix/prefix.mingw
index 7523670f6d..ac1327c46c 100644
--- a/builds/posix/prefix.mingw
+++ b/builds/posix/prefix.mingw
@@ -31,7 +31,7 @@ PLATFORM_PATH=os/win32
 
 OS_SPECIFIC_Files=thd_priority.cpp
 
-OS_ServerFiles= os/win32/srvr_w32 os/win32/window.cpp os/win32/chop.cpp os/win32/property.cpp os/win32/cntl.cpp 
+OS_ServerFiles= os/win32/srvr_w32.cpp os/win32/window.cpp os/win32/chop.cpp os/win32/property.cpp os/win32/cntl.cpp 
 
 # Override link command for implicit stdc++ linking
 LIB_LINK=$(LD)
-- 
2.13.0

