From bd4dc8916de1872a19b588bd9983f874167c35d5 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 26 Nov 2014 09:27:40 +0300
Subject: [PATCH 19/27] Don't install unix files for Windows platform

---
 configure.in | 66 ++++++++++++++++++++++++++++++++++++------------------------
 1 file changed, 40 insertions(+), 26 deletions(-)

diff --git a/configure.in b/configure.in
index 21384bf391..10ad6a2857 100644
--- a/configure.in
+++ b/configure.in
@@ -1314,32 +1314,39 @@ MAKE_SRC_DIR=builds/posix
 
 dnl common files for all posix hosts
 dnl TODO: fix "arch-specific/linux/" paths for common posix scripts with SVN
-AC_CONFIG_FILES([gen/install/misc/firebird.conf:builds/install/misc/firebird.conf.in
-gen/install/misc/aliases.conf:builds/install/misc/aliases.conf.in
-gen/install/misc/fbintl.conf:builds/install/misc/fbintl.conf
-gen/install/misc/fbtrace.conf:src/utilities/ntrace/fbtrace.conf
-gen/firebird/bin/fb_config:builds/install/misc/fb_config.in
-gen/firebird/bin/posixLibrary.sh:builds/install/misc/posixLibrary.sh.in
-gen/firebird/bin/classicLibrary.sh:builds/install/misc/classicLibrary.sh.in
-gen/firebird/bin/superLibrary.sh:builds/install/misc/superLibrary.sh.in
-gen/firebird/bin/changeRunUser.sh:builds/install/misc/changeRunUser.sh.in
-gen/firebird/bin/restoreRootRunUser.sh:builds/install/misc/restoreRootRunUser.sh.in
-gen/firebird/bin/tarMainInstall.sh:builds/install/arch-specific/linux/misc/tarMainInstall.sh.in
-gen/firebird/bin/tarinstall.sh:builds/install/arch-specific/linux/misc/tarinstall.sh.in
-gen/firebird/bin/preinstall.sh:builds/install/arch-specific/linux/misc/preinstall.sh.in
-gen/firebird/bin/postinstall.sh:builds/install/arch-specific/linux/misc/postinstall.sh.in
-gen/firebird/bin/preuninstall.sh:builds/install/arch-specific/linux/misc/preuninstall.sh.in
-gen/firebird/bin/postuninstall.sh:builds/install/arch-specific/linux/misc/postuninstall.sh.in
-gen/firebird/bin/taruninstall.sh:builds/install/arch-specific/linux/misc/taruninstall.sh.in
-gen/firebird/bin/tarMainUninstall.sh:builds/install/arch-specific/linux/misc/tarMainUninstall.sh.in
-gen/firebird/bin/changeDBAPassword.sh:builds/install/misc/changeDBAPassword.sh.in
-gen/firebird/bin/changeGdsLibraryCompatibleLink.sh:builds/install/misc/changeGdsLibraryCompatibleLink.sh.in
-gen/firebird/bin/createAliasDB.sh:builds/install/misc/createAliasDB.sh.in],
-[chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
-
-if test $FIREBIRD_ARCH_TYPE = classic; then
-	AC_CONFIG_FILES([gen/firebird/bin/changeMultiConnectMode.sh:builds/install/misc/changeMultiConnectMode.sh.in],
-	[chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+if test $PLATFORM = win32; then
+  AC_CONFIG_FILES(gen/firebird/firebird.conf:builds/install/misc/firebird.conf.in
+  gen/firebird/aliases.conf:builds/install/misc/aliases.conf.in
+  gen/firebird/intl/fbintl.conf:builds/install/misc/fbintl.conf
+  gen/firebird/fbtrace.conf:src/utilities/ntrace/fbtrace.conf)
+else
+  AC_CONFIG_FILES([gen/install/misc/firebird.conf:builds/install/misc/firebird.conf.in
+  gen/install/misc/aliases.conf:builds/install/misc/aliases.conf.in
+  gen/install/misc/fbintl.conf:builds/install/misc/fbintl.conf
+  gen/install/misc/fbtrace.conf:src/utilities/ntrace/fbtrace.conf
+  gen/firebird/bin/fb_config:builds/install/misc/fb_config.in
+  gen/firebird/bin/posixLibrary.sh:builds/install/misc/posixLibrary.sh.in
+  gen/firebird/bin/classicLibrary.sh:builds/install/misc/classicLibrary.sh.in
+  gen/firebird/bin/superLibrary.sh:builds/install/misc/superLibrary.sh.in
+  gen/firebird/bin/changeRunUser.sh:builds/install/misc/changeRunUser.sh.in
+  gen/firebird/bin/restoreRootRunUser.sh:builds/install/misc/restoreRootRunUser.sh.in
+  gen/firebird/bin/tarMainInstall.sh:builds/install/arch-specific/linux/misc/tarMainInstall.sh.in
+  gen/firebird/bin/tarinstall.sh:builds/install/arch-specific/linux/misc/tarinstall.sh.in
+  gen/firebird/bin/preinstall.sh:builds/install/arch-specific/linux/misc/preinstall.sh.in
+  gen/firebird/bin/postinstall.sh:builds/install/arch-specific/linux/misc/postinstall.sh.in
+  gen/firebird/bin/preuninstall.sh:builds/install/arch-specific/linux/misc/preuninstall.sh.in
+  gen/firebird/bin/postuninstall.sh:builds/install/arch-specific/linux/misc/postuninstall.sh.in
+  gen/firebird/bin/taruninstall.sh:builds/install/arch-specific/linux/misc/taruninstall.sh.in
+  gen/firebird/bin/tarMainUninstall.sh:builds/install/arch-specific/linux/misc/tarMainUninstall.sh.in
+  gen/firebird/bin/changeDBAPassword.sh:builds/install/misc/changeDBAPassword.sh.in
+  gen/firebird/bin/changeGdsLibraryCompatibleLink.sh:builds/install/misc/changeGdsLibraryCompatibleLink.sh.in
+  gen/firebird/bin/createAliasDB.sh:builds/install/misc/createAliasDB.sh.in],
+  [chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+
+  if test $FIREBIRD_ARCH_TYPE = classic; then
+    AC_CONFIG_FILES([gen/firebird/bin/changeMultiConnectMode.sh:builds/install/misc/changeMultiConnectMode.sh.in],
+    [chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+  fi
 fi
 
 dnl: Platform Specific Files
@@ -1439,6 +1446,13 @@ case "$PLATFORM" in
 
   DARWIN)
     ;;
+  
+  win32)
+    AC_CONFIG_FILES(gen/firebird/install_classic.bat:builds/install/arch-specific/win32/install_classic.bat
+                     gen/firebird/install_super.bat:builds/install/arch-specific/win32/install_super.bat
+                     gen/firebird/install_superclassic.bat:builds/install/arch-specific/win32/install_superclassic.bat
+                     gen/firebird/uninstall.bat:builds/install/arch-specific/win32/uninstall.bat)
+    ;;
 esac
 if test -f ${MAKE_SRC_DIR}/postfix.${MAKEFILE_POSTFIX};then
 	POSTFIX_FILE=gen/make.platform.postfix:${MAKE_SRC_DIR}/postfix.${MAKEFILE_POSTFIX}
-- 
2.13.0

