From ab52abf519b5121f1a7d820fc1bb86b253e908b6 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Tue, 11 Nov 2014 09:27:40 +0300
Subject: [PATCH 25/27] Fix building gstat

---
 src/utilities/gstat/dba.epp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utilities/gstat/dba.epp b/src/utilities/gstat/dba.epp
index 76a10c5b49..71e6cf5bf8 100644
--- a/src/utilities/gstat/dba.epp
+++ b/src/utilities/gstat/dba.epp
@@ -42,7 +42,7 @@
 #include "../jrd/ibase.h"
 #include "../jrd/ods.h"
 #include "../jrd/btn.h"
-#include "../jrd/svc.h"
+#include "../jrd/ThreadData.h"
 #include "../jrd/license.h"
 #include "../jrd/msg_encode.h"
 #include "../jrd/gdsassert.h"
-- 
2.13.0

