From 2b9a68b2a714f3cfa791e2dcb6b928b1b3a270a1 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 9 Jun 2017 15:01:35 +0300
Subject: [PATCH 29/29] Fix format warnings

---
diff --git a/src/include/fb_blk.h b/src/include/fb_blk.h
index f8fe95964a..efff81cc68 100644
--- a/src/include/fb_blk.h
+++ b/src/include/fb_blk.h
@@ -112,7 +112,7 @@
 	rem_type_rsr
 };
 
-static void * zero = (void *) 0;
+//static void * zero = (void *) 0;
 
 template<BlockType BLOCK_TYPE>
 class TypedHandle
@@ -172,5 +172,5 @@
 private:
     // These operators are off-limits
-	void* operator new(size_t s) { return zero; }
-    void* operator new[](size_t s) { return zero; }
+	void* operator new(size_t s) { return 0; }
+    void* operator new[](size_t s) { return 0; }
 };
@@ -195,13 +195,13 @@
 private:
     // These operations are not supported on static repeat-base objects
     void* operator new[](size_t s, MemoryPool& p)
-        { return zero; }
+        { return 0; }
     void operator delete[](void* mem, MemoryPool& p)
         { }
     void operator delete[](void* mem)
 		{ }
 
     // These operators are off-limits
-	void* operator new(size_t s) { return zero; }
-    void* operator new[](size_t s) { return zero; }
+	void* operator new(size_t s) { return 0; }
+    void* operator new[](size_t s) { return 0; }
 };
-- 
2.13.0

