From 2980d42e1b6164dddc62e78edaac8b99ccf8f83f Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 09:20:35 +0300
Subject: Fix calling SCL_check_access

---
 src/jrd/tra.cpp | 2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/jrd/tra.cpp b/src/jrd/tra.cpp
index e0e36acafa..ecc334b805 100644
--- a/src/jrd/tra.cpp	2019-05-21 14:02:46.952465100 +0300
+++ b/src/jrd/tra.cpp	2019-05-21 14:06:08.562747900 +0300
@@ -3857,7 +3857,7 @@
 				{
 					ThreadStatusGuard status_vector(tdbb);
 
-					SCL_check_access(tdbb, s_class, 0, NULL, NULL, SCL_read, object_table, false,
+					SCL_check_access(tdbb, s_class, 0, NULL, NULL, SCL_read, object_table,
 						blb_relation->rel_name);
 					s_class->scl_blb_access = SecurityClass::BA_SUCCESS;
 				}
-- 
2.21.0