--- build-x86_64-w64-mingw32/src/jrd/os/win32/mod_loader.cpp.orig	2015-06-15 12:43:29.538400000 +0300
+++ build-x86_64-w64-mingw32/src/jrd/os/win32/mod_loader.cpp	2015-06-15 12:45:03.824800000 +0300
@@ -74,7 +74,7 @@
 // if we don't use MSVC then we don't use MS CRT ?
 // NS: versions of MSVC before 2005 and, as preliminary reports suggest,
 // after 2008 do not need this hack
-#if !defined(_MSC_VER) || (_MSC_VER < 1400)
+#if (!defined(_MSC_VER) || (_MSC_VER < 1400)) && !defined(__MINGW32__)
 		return;
 #else
 
@@ -89,7 +89,9 @@
 		if ((*mFindActCtxSectionString)
 				(0, NULL,
 				ACTIVATION_CONTEXT_SECTION_DLL_REDIRECTION,
-#if _MSC_VER == 1400
+#if defined(__MINGW32__)
+                    "msvcrt.dll",
+#elif _MSC_VER == 1400
                     "msvcr80.dll",
 #elif _MSC_VER == 1500
                     "msvcr90.dll",
