From 3e38f9f7ec8f010fcbdab60b768d2c5e27b49ae7 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 07:39:37 +0300
Subject: [PATCH] Remove hardcoded march to allow 64-bit builds.

---
 builds/posix/prefix.mingw | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/builds/posix/prefix.mingw b/builds/posix/prefix.mingw
index fd36abe..04e0978 100644
--- a/builds/posix/prefix.mingw
+++ b/builds/posix/prefix.mingw
@@ -20,8 +20,8 @@
 # 
 
 # -Wno-unused-variable is used due to unused gpre generated variables
-PROD_FLAGS=-O2  -march=i586 -DMINGW -Wall -Wshadow -Wundef -Wno-long-long -Wno-unused-variable -Wno-sign-compare -Wno-parentheses -Wno-switch -fmessage-length=0 -Dlint -DWIN32_LEAN_AND_MEAN -MMD -mthreads
-DEV_FLAGS=-ggdb -march=i586 -DMINGW -Wall -Wshadow -Wundef -Wno-long-long -Wno-unused-variable -Wno-sign-compare -Wno-parentheses -Wno-switch -fmessage-length=0 -Dlint -DWIN32_LEAN_AND_MEAN -MMD -mthreads
+PROD_FLAGS=-O2  -DMINGW -Wall -Wshadow -Wundef -Wno-long-long -Wno-unused-variable -Wno-sign-compare -Wno-parentheses -Wno-switch -fmessage-length=0 -Dlint -DWIN32_LEAN_AND_MEAN -MMD -mthreads
+DEV_FLAGS=-ggdb -DMINGW -Wall -Wshadow -Wundef -Wno-long-long -Wno-unused-variable -Wno-sign-compare -Wno-parentheses -Wno-switch -fmessage-length=0 -Dlint -DWIN32_LEAN_AND_MEAN -MMD -mthreads
 
 PLATFORM_PATH=os/win32
 
-- 
2.4.3

From 0ed8f717543788f0c38324f4bcc550ff4e8025a0 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 06:18:02 +0300
Subject: [PATCH] Always use Win32 threads for mingw.

---
 src/jrd/common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jrd/common.h b/src/jrd/common.h
index 875040f..ca57c71 100644
--- a/src/jrd/common.h
+++ b/src/jrd/common.h
@@ -65,7 +65,7 @@
 #include <unistd.h>
 #endif
 
-#if defined(_POSIX_THREADS) && _POSIX_THREADS >= 200112L
+#if defined(_POSIX_THREADS) && _POSIX_THREADS >= 200112L && !defined(WIN_NT)
 // above check is generally true. However, we use pthreads on some platforms
 // where _POSIX_THREADS is defined to "1" or not even defined at all!
 #define USE_POSIX_THREADS
-- 
2.4.3

From 45d71e7fa21beeb444a5e3b8c60207db1b6cf38f Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 08:56:28 +0300
Subject: [PATCH] Define AMD64 for 64-bit mingw.

---
 configure.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 86e89f1..d345348 100644
--- a/configure.in
+++ b/configure.in
@@ -463,7 +463,12 @@ dnl CPU_TYPE=ppc64
     EDITLINE_FLG=N
     RAW_DEVICES_FLG=N
     SHRLIB_EXT=dll
-	ICU_PLATFORM=MinGW
+    ICU_PLATFORM=MinGW
+    case "$target" in
+        x86_64-*-mingw*)
+            AC_DEFINE(AMD64, 1, [Define this if CPU is amd64])
+        ;;
+    esac
     ;;
 
   *)
-- 
2.4.3

From 56c1e99e9eb1fe8c1e822132332c72a9e6fd97b6 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 09:20:35 +0300
Subject: [PATCH] mingw-w64 have isinf

---
 src/common/classes/FpeControl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/classes/FpeControl.h b/src/common/classes/FpeControl.h
index e0e36ac..363cad5 100644
--- a/src/common/classes/FpeControl.h
+++ b/src/common/classes/FpeControl.h
@@ -217,7 +217,7 @@ private:
 
 
 // getting a portable isinf() is harder than you would expect
-#ifdef WIN_NT
+#if defined(WIN_NT) && !defined(__MINGW32__)
 inline bool isinf(double x)
 {
 	return (!_finite (x) && !isnan(x));
-- 
2.4.3

From 68681019a2aa6c93f1ec99fa368df23578f3f6bc Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 09:57:27 +0300
Subject: [PATCH] Fix cast error.

---
 src/remote/inet.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/remote/inet.cpp b/src/remote/inet.cpp
index c3816a8..f05de04 100644
--- a/src/remote/inet.cpp
+++ b/src/remote/inet.cpp
@@ -1851,7 +1851,7 @@ static void wsaExitHandler(void*)
 }
 
 
-static int fork(SOCKET old_handle, USHORT flag)
+static int fork(SOCKET old_handle, DWORD flag)
 {
 /**************************************
  *
@@ -1908,7 +1908,7 @@ static int fork(SOCKET old_handle, USHORT flag)
 
 THREAD_ENTRY_DECLARE forkThread(THREAD_ENTRY_PARAM arg)
 {
-	const USHORT flag = (USHORT) arg;
+	const DWORD flag = (uintptr_t) arg;
 
 	while (!INET_shutting_down)
 	{
-- 
2.4.3

From e9aa5ea0fb2ed70f036c13eb17821242c3482c7d Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 10:08:42 +0300
Subject: [PATCH] mingw-w64 don't need hack for SecPkgContext_AccessToken

---
 src/auth/trusted/AuthSspi.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/auth/trusted/AuthSspi.cpp b/src/auth/trusted/AuthSspi.cpp
index d420041..15b859c 100644
--- a/src/auth/trusted/AuthSspi.cpp
+++ b/src/auth/trusted/AuthSspi.cpp
@@ -87,7 +87,7 @@ AuthSspi::~AuthSspi()
 
 bool AuthSspi::checkAdminPrivilege(PCtxtHandle phContext) const
 {
-#if defined (__GNUC__)
+#if defined (__GNUC__) && !defined(__MINGW64_VERSION_MAJOR)
 	// ASF: MinGW hack.
 	struct SecPkgContext_AccessToken
 	{
-- 
2.4.3

From 574e99459f6bec4857fcf459628651ca8f2b3b98 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 10:09:19 +0300
Subject: [PATCH] Use modern ICU libraries when link with system wide ICU.

---
 builds/posix/prefix.mingw | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/builds/posix/prefix.mingw b/builds/posix/prefix.mingw
index 04e0978..7523670 100644
--- a/builds/posix/prefix.mingw
+++ b/builds/posix/prefix.mingw
@@ -19,6 +19,10 @@
 # 2003-Sep-27 BRS Removed IPSERVER after XNET is included
 # 
 
+ifeq ($(STD_ICU),true)
+   ICU_LIBS= -licuuc -licudt -licuin
+endif
+
 # -Wno-unused-variable is used due to unused gpre generated variables
 PROD_FLAGS=-O2  -DMINGW -Wall -Wshadow -Wundef -Wno-long-long -Wno-unused-variable -Wno-sign-compare -Wno-parentheses -Wno-switch -fmessage-length=0 -Dlint -DWIN32_LEAN_AND_MEAN -MMD -mthreads
 DEV_FLAGS=-ggdb -DMINGW -Wall -Wshadow -Wundef -Wno-long-long -Wno-unused-variable -Wno-sign-compare -Wno-parentheses -Wno-switch -fmessage-length=0 -Dlint -DWIN32_LEAN_AND_MEAN -MMD -mthreads
-- 
2.4.3

From 57b07ada01bcfa4deb9da61563823b184e5bbb10 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 13:28:03 +0300
Subject: [PATCH] mingw-w64 don't have win98 support.

---
 src/common/classes/locks.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/classes/locks.cpp b/src/common/classes/locks.cpp
index bbb9440..de6b2f8 100644
--- a/src/common/classes/locks.cpp
+++ b/src/common/classes/locks.cpp
@@ -28,7 +28,7 @@
 #include "../../include/firebird.h"
 #include "../../jrd/common.h"
 
-#if defined(WIN_NT)
+#if defined(WIN_NT) && !defined(__MINGW64_VERSION_MAJOR)
 // minimum win32 version: win98 / winnt4 SP3
 #define _WIN32_WINNT 0x0403
 #endif
-- 
2.4.3

From f334249bd5d7ca5947940488d7b210f096e2d64d Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 15:47:34 +0300
Subject: [PATCH] Fix using function pointer.

---
 src/common/classes/locks.cpp | 2 +-
 src/common/classes/locks.h   | 7 +++----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/common/classes/locks.cpp b/src/common/classes/locks.cpp
index de6b2f8..321da36 100644
--- a/src/common/classes/locks.cpp
+++ b/src/common/classes/locks.cpp
@@ -124,7 +124,7 @@ BOOL WINAPI TryEnterCriticalSection_Win9X(CRITICAL_SECTION* cs)
 
 // On Win 98 and Win ME TryEnterCriticalSection is defined, but not implemented
 // So direct linking to it won't hurt and will signal our incompatibility with Win 95
-TryEnterCS::tTryEnterCriticalSection* TryEnterCS::m_funct = &TryEnterCriticalSection;
+TryEnterCS::tTryEnterCriticalSection TryEnterCS::m_funct = TryEnterCriticalSection;
 
 static TryEnterCS tryEnterCS;
 
diff --git a/src/common/classes/locks.h b/src/common/classes/locks.h
index 75ba194..d2ddb7d 100644
--- a/src/common/classes/locks.h
+++ b/src/common/classes/locks.h
@@ -60,14 +60,13 @@ public:
 
 	static bool tryEnter(LPCRITICAL_SECTION lpCS)
 	{
-		return ((*m_funct) (lpCS) == TRUE);
+		return ((m_funct) (lpCS) == TRUE);
 	}
 
 private:
-	typedef WINBASEAPI BOOL WINAPI tTryEnterCriticalSection
-		(LPCRITICAL_SECTION lpCriticalSection);
+	typedef BOOL WINAPI (*tTryEnterCriticalSection)(LPCRITICAL_SECTION lpCriticalSection);
 
-	static tTryEnterCriticalSection* m_funct;
+	static tTryEnterCriticalSection m_funct;
 };
 
 class Mutex
-- 
2.4.3

From 30e30e96daf43c5dcb42bd3c57d8df362589bb59 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 18:54:37 +0300
Subject: [PATCH] Fix redefinition of matherr.

---
 src/extlib/ib_udf.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/extlib/ib_udf.cpp b/src/extlib/ib_udf.cpp
index c5a620d..9017228 100644
--- a/src/extlib/ib_udf.cpp
+++ b/src/extlib/ib_udf.cpp
@@ -50,10 +50,12 @@ extern "C"
 #else
 #define exception_type __exception
 #endif
+#ifndef matherr
 int MATHERR(struct exception_type*)
 {
 	return 1;
 }
+#endif
 #undef exception_type
 #endif /* SOLARIS */
 #endif //__ICC
-- 
2.4.3

From e5e3455760e07ced13c9c91b7aaa2e3116ace50d Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 21:44:53 +0300
Subject: [PATCH] Don't export gds__print_pool as it not part of API according
 to gds_proto.h:135

---
 src/jrd/gds.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jrd/gds.cpp b/src/jrd/gds.cpp
index e3949fc..0b6852f 100644
--- a/src/jrd/gds.cpp
+++ b/src/jrd/gds.cpp
@@ -1226,7 +1226,7 @@ void API_ROUTINE gds__log(const TEXT* text, ...)
 #endif
 }
 
-void API_ROUTINE gds__print_pool(MemoryPool* pool, const TEXT* text, ...)
+void gds__print_pool(MemoryPool* pool, const TEXT* text, ...)
 {
 /**************************************
  *
-- 
2.4.3

From e3a94cfa6fe9686da6639b9605b7623c0aa960ac Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 19 Nov 2014 22:06:27 +0300
Subject: [PATCH] mingw-w64 have fenv.h

---
 src/common/classes/FpeControl.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/classes/FpeControl.h b/src/common/classes/FpeControl.h
index 363cad5..93e48be 100644
--- a/src/common/classes/FpeControl.h
+++ b/src/common/classes/FpeControl.h
@@ -30,7 +30,7 @@
 #define CLASSES_FPE_CONTROL_H
 
 #include <math.h>
-#if defined(WIN_NT)
+#if defined(_MSC_VER)
 #include <float.h>
 #else
 #include <fenv.h>
@@ -78,7 +78,7 @@ public:
 		}
 	}
 
-#if defined(WIN_NT)
+#if defined(_MSC_VER)
 	static void maskAll() throw()
 	{
 		_clearfp(); // always call _clearfp() before setting control word
-- 
2.4.3

From 3ac3bbf82501f0b70b07cb5a1834885524b19763 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Fri, 21 Nov 2014 12:38:29 +0300
Subject: [PATCH] Fix cast error for 64-bit.

---
 src/remote/os/win32/srvr_w32.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/remote/os/win32/srvr_w32.cpp b/src/remote/os/win32/srvr_w32.cpp
index fd0f75c..a0bf167 100644
--- a/src/remote/os/win32/srvr_w32.cpp
+++ b/src/remote/os/win32/srvr_w32.cpp
@@ -269,7 +269,7 @@ int WINAPI WinMain(HINSTANCE hThisInst, HINSTANCE /*hPrevInst*/, LPSTR lpszArgs,
 		else if (server_flag & SRVR_wnet)
 			port = WNET_reconnect(connection_handle, status_vector);
 		else if (server_flag & SRVR_xnet)
-			port = XNET_reconnect((ULONG) connection_handle, status_vector);
+			port = XNET_reconnect((ULONG_PTR) connection_handle, status_vector);
 
 		if (port) {
 			service_connection(port);
-- 
2.4.3

From ceda81a2ad529ef0711bfa7136b053637d97f760 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Fri, 21 Nov 2014 14:21:32 +0300
Subject: [PATCH] Fix source extension.

---
 builds/posix/prefix.mingw | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builds/posix/prefix.mingw b/builds/posix/prefix.mingw
index 7523670..ac1327c 100644
--- a/builds/posix/prefix.mingw
+++ b/builds/posix/prefix.mingw
@@ -31,7 +31,7 @@ PLATFORM_PATH=os/win32
 
 OS_SPECIFIC_Files=thd_priority.cpp
 
-OS_ServerFiles= os/win32/srvr_w32 os/win32/window.cpp os/win32/chop.cpp os/win32/property.cpp os/win32/cntl.cpp 
+OS_ServerFiles= os/win32/srvr_w32.cpp os/win32/window.cpp os/win32/chop.cpp os/win32/property.cpp os/win32/cntl.cpp 
 
 # Override link command for implicit stdc++ linking
 LIB_LINK=$(LD)
-- 
2.4.3

From 9405c09c70b553e49e7fe385386aba1fb3225d3e Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sat, 22 Nov 2014 12:03:41 +0300
Subject: [PATCH] Move extra mingw sources definitions for fbintl to
 Makefile.in.intl

---
 builds/posix/Makefile.in.intl | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/builds/posix/Makefile.in.intl b/builds/posix/Makefile.in.intl
index 57ecb34..8d1e4e4 100644
--- a/builds/posix/Makefile.in.intl
+++ b/builds/posix/Makefile.in.intl
@@ -52,6 +52,13 @@ INTL_Files1=	ld.cpp       cv_narrow.cpp \
 		cs_ksc.cpp     cv_ksc.cpp    lc_ksc.cpp \
 		cs_icu.cpp     cv_icu.cpp    lc_icu.cpp
 
+ifeq ($(PLATFORM),win32)
+INTL_Files2 += $(FBCONFIG_Sources) \
+				$(FBCLASSES_MsgSources) \
+				$(OS_SPECIFIC_Sources) \
+				jrd/isc.cpp jrd/gds.cpp \
+				common/utils.cpp
+endif
 # use INTL_Files2 in prefix file for host-specific additional files
 INTL_Sources= $(addprefix intl/, $(INTL_Files1)) $(INTL_Files2) \
 	jrd/IntlUtil.cpp jrd/unicode_util.cpp jrd/CharSet.cpp jrd/mod_loader.cpp
@@ -61,7 +68,7 @@ INTL_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(INTL_Sources)))
 # SD: Shared variables are overriden to exclude extra files and
 # this way fix 'unresolved symbol' errors when some unused routines call
 # other routines from not included modules
-FBCLASSES_ClientFiles=alloc.cpp locks.cpp fb_string.cpp timestamp.cpp
+FBCLASSES_ClientFiles=alloc.cpp locks.cpp fb_string.cpp timestamp.cpp TempFile.cpp
 
 
 AllObjects = $(INTL_Objects)
-- 
2.4.3

From a3a93057d7dec63d04e583080c47009466b9eb2a Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sat, 22 Nov 2014 22:59:59 +0300
Subject: [PATCH] Add missing sources for utils building.

---
 builds/posix/Makefile.in.client.gsec | 2 +-
 builds/posix/Makefile.in.client.util | 4 ++--
 builds/posix/Makefile.in.embed.util  | 8 ++++----
 builds/posix/Makefile.in.fbtrace     | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/builds/posix/Makefile.in.client.gsec b/builds/posix/Makefile.in.client.gsec
index 2efa520..803ea4f 100644
--- a/builds/posix/Makefile.in.client.gsec
+++ b/builds/posix/Makefile.in.client.gsec
@@ -29,7 +29,7 @@ include     $(ROOT)/gen/make.shared.variables
 
 @SET_MAKE@
 
-GSEC_Other_Sources = jrd/isc_file.cpp jrd/ThreadData.cpp common/UtilSvc.cpp jrd/enc.cpp \
+GSEC_Other_Sources = jrd/isc.cpp jrd/isc_file.cpp jrd/ThreadData.cpp common/UtilSvc.cpp jrd/enc.cpp \
 					 jrd/sha.cpp jrd/guid.cpp
 GSEC_Files =  gsec.cpp call_service.cpp gsecMain.cpp security.cpp
 GSEC_Sources = $(addprefix utilities/gsec/, $(GSEC_Files)) $(GSEC_Other_Sources)
diff --git a/builds/posix/Makefile.in.client.util b/builds/posix/Makefile.in.client.util
index dab8911..70bfab7 100644
--- a/builds/posix/Makefile.in.client.util
+++ b/builds/posix/Makefile.in.client.util
@@ -40,14 +40,14 @@ include     $(ROOT)/gen/make.shared.variables
 @SET_MAKE@
 
 
-LOCKPRINT_Other_Sources = jrd/isc_sync.cpp jrd/isc.cpp jrd/db_alias.cpp \
+LOCKPRINT_Other_Sources = jrd/isc_ipc.cpp jrd/isc_sync.cpp jrd/isc.cpp jrd/db_alias.cpp \
 common/config/config.cpp common/config/config_file.cpp $(OS_SPECIFIC_Sources)
 LOCKPRINT_Files= print.cpp
 LOCKPRINT_Sources = $(addprefix lock/, $(LOCKPRINT_Files)) $(LOCKPRINT_Other_Sources)
 LOCKPRINT_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(LOCKPRINT_Sources)))) \
 $(FBCONFIG_Objects)
 
-NBACKUP_Other_Sources = jrd/isc_file.cpp jrd/db_alias.cpp common/UtilSvc.cpp $(OS_SPECIFIC_Sources)
+NBACKUP_Other_Sources = jrd/isc.cpp jrd/isc_file.cpp jrd/db_alias.cpp common/UtilSvc.cpp $(OS_SPECIFIC_Sources)
 NBACKUP_Files= nbackup.cpp nbackup/nbkMain.cpp
 NBACKUP_Sources = $(addprefix utilities/, $(NBACKUP_Files)) $(NBACKUP_Other_Sources)
 NBACKUP_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(NBACKUP_Sources)))) \
diff --git a/builds/posix/Makefile.in.embed.util b/builds/posix/Makefile.in.embed.util
index 3d1a98f..8488a05 100644
--- a/builds/posix/Makefile.in.embed.util
+++ b/builds/posix/Makefile.in.embed.util
@@ -40,7 +40,7 @@ include     $(ROOT)/gen/make.shared.variables
 @SET_MAKE@
 
 
-LOCKPRINT_Other_Sources += jrd/isc_sync.cpp jrd/isc.cpp jrd/db_alias.cpp \
+LOCKPRINT_Other_Sources += jrd/isc_ipc.cpp jrd/isc_sync.cpp jrd/isc.cpp jrd/db_alias.cpp \
 $(OS_SPECIFIC_Sources)
 
 LOCKPRINT_Files= print.cpp
@@ -48,7 +48,7 @@ LOCKPRINT_Sources = $(addprefix lock/, $(LOCKPRINT_Files)) $(LOCKPRINT_Other_Sou
 LOCKPRINT_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(LOCKPRINT_Sources)))) \
 $(FBCONFIG_Objects)
 
-NBACKUP_Other_Sources = jrd/isc_file.cpp jrd/db_alias.cpp common/UtilSvc.cpp $(OS_SPECIFIC_Sources)
+NBACKUP_Other_Sources = jrd/isc.cpp jrd/isc_file.cpp jrd/db_alias.cpp common/UtilSvc.cpp $(OS_SPECIFIC_Sources)
 
 NBACKUP_Files= nbackup.cpp nbackup/nbkMain.cpp
 NBACKUP_Sources = $(addprefix utilities/, $(NBACKUP_Files)) $(NBACKUP_Other_Sources)
@@ -65,7 +65,7 @@ DROP_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(DROP_Sources)))
 
 
 GSEC_Other_Sources = jrd/enc.cpp jrd/sha.cpp jrd/guid.cpp \
-jrd/isc_file.cpp common/config/config.cpp common/config/config_file.cpp \
+jrd/isc.cpp jrd/isc_file.cpp common/config/config.cpp common/config/config_file.cpp \
 common/config/dir_list.cpp \
 common/UtilSvc.cpp jrd/ThreadData.cpp \
 $(OS_SPECIFIC_Sources)
@@ -76,7 +76,7 @@ GSEC_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(GSEC_Sources)))
 
 
 GSTAT_Other_Sources = jrd/btn.cpp jrd/db_alias.cpp common/utils.cpp jrd/ods.cpp \
-common/UtilSvc.cpp jrd/ThreadData.cpp jrd/isc_file.cpp $(OS_SPECIFIC_Sources)
+common/UtilSvc.cpp jrd/ThreadData.cpp jrd/isc.cpp jrd/isc_file.cpp $(OS_SPECIFIC_Sources)
 
 GSTAT_Files =  dba.epp ppg.cpp gstatMain.cpp
 GSTAT_Sources = $(addprefix utilities/gstat/, $(GSTAT_Files)) $(GSTAT_Other_Sources)
diff --git a/builds/posix/Makefile.in.fbtrace b/builds/posix/Makefile.in.fbtrace
index 0f1a09e..daee0d8 100644
--- a/builds/posix/Makefile.in.fbtrace
+++ b/builds/posix/Makefile.in.fbtrace
@@ -49,7 +49,7 @@ FBTRACE_UTIL_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(FBTRACE
 
 
 FBTRACE_OTHER_Sources = $(OS_SPECIFIC_Sources) $(TRACE_OS_Sources) \
-	jrd/isc.cpp jrd/isc_file.cpp jrd/CharSet.cpp jrd/TextType.cpp \
+	jrd/isc.cpp jrd/isc_file.cpp jrd/isc_sync.cpp jrd/CharSet.cpp jrd/TextType.cpp \
 	jrd/IntlUtil.cpp jrd/unicode_util.cpp common/classes/ClumpletReader.cpp common/utils.cpp
 
 FBTRACE_OTHER_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(FBTRACE_OTHER_Sources))))
-- 
2.4.3

From d671d8b1d4bec4683c3be258afacacbc47f640cc Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sun, 23 Nov 2014 00:13:24 +0300
Subject: [PATCH] smtp_server: Add win32 platform files.

---
 builds/posix/Makefile.in.inet_server | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/builds/posix/Makefile.in.inet_server b/builds/posix/Makefile.in.inet_server
index d59b07a..ad471fd 100644
--- a/builds/posix/Makefile.in.inet_server
+++ b/builds/posix/Makefile.in.inet_server
@@ -40,6 +40,9 @@ include     $(ROOT)/gen/make.shared.variables
 
 SERVER_Files = server_stub.cpp 
 SERVER_Sources = $(addprefix remote/, $(SERVER_Files))
+ifeq ($(PLATFORM),win32)
+SERVER_Sources += utilities/nbackup/nbkMain.cpp common/UtilSvc.cpp
+endif
 SERVER_Objects = $(addprefix $(OBJ)/, $(addsuffix .o, $(basename $(SERVER_Sources))))
 
 AllObjects = $(SERVER_Objects)
-- 
2.4.3

From 5cba22ce0e704841f7b808298a8d60a86aa10425 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sun, 23 Nov 2014 13:27:02 +0300
Subject: [PATCH] Fix building net_server and mp_server with mingw

---
 builds/posix/Makefile.in.inet_server | 4 ++--
 builds/posix/Makefile.in.smp_server  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/builds/posix/Makefile.in.inet_server b/builds/posix/Makefile.in.inet_server
index ad471fd..99ddd19 100644
--- a/builds/posix/Makefile.in.inet_server
+++ b/builds/posix/Makefile.in.inet_server
@@ -54,8 +54,8 @@ Dependencies = $(AllObjects:.o=.d)
 #     but we must ensure it's existance
 fb_inet_server : $(LIBFBEMBED_SO) $(FB_CLASSIC_SERVER)
 
-$(FB_CLASSIC_SERVER):	$(SERVER_Objects)
-	$(EXE_LINK) $(LINK_EMPTY_SYMBOLS) $(LINK_OPTS) $^ -o $@ $(FBEMBED_LINK) $(LIB_GUI) $(LINK_LIBS)
+$(FB_CLASSIC_SERVER):	$(SERVER_Objects) $(COMMON_Objects) $(CLUMPLETS_Objects) $(FBCONFIG_Objects) $(OS_SPECIFIC_Objects) $(DSQL_Objects) $(BURP_Objects) $(ALICE_Objects) $(LOCK_Objects) $(INTERFACE_Objects) $(NBACKUP_SERVER_Objects) $(NTRACE_SERVER_Objects) $(SECURITY_Objects) $(JRD_Objects)
+	$(EXE_LINK) $(LINK_EMPTY_SYMBOLS) $(LINK_OPTS) $^ -o $@ $(FBEMBED_LINK) $(LIB_GUI) $(ICU_LIBS) $(LINK_LIBS)
 
 include $(ROOT)/gen/make.shared.targets
 
diff --git a/builds/posix/Makefile.in.smp_server b/builds/posix/Makefile.in.smp_server
index 1bc3872..f386021 100644
--- a/builds/posix/Makefile.in.smp_server
+++ b/builds/posix/Makefile.in.smp_server
@@ -59,8 +59,8 @@ $(OS_SPECIFIC_Objects)
 
 fb_smp_server:	$(FB_DAEMON)
 
-$(FB_DAEMON):	$(SERVER_Objects)
-	$(EXE_LINK) $(LINK_EMPTY_SYMBOLS) $(LINK_OPTS) $^ -o $@ $(FBEMBED_LINK) $(LINK_LIBS)
+$(FB_DAEMON):	$(SERVER_Objects) $(DSQL_Objects) $(BURP_Objects) $(ALICE_Objects) $(JRD_Objects) $(LOCK_Objects) $(INTERFACE_Objects) $(SECURITY_Objects) $(NBACKUP_SERVER_Objects) $(NTRACE_SERVER_Objects)
+	$(EXE_LINK) $(LINK_EMPTY_SYMBOLS) $(LINK_OPTS) $^ -o $@ $(FBEMBED_LINK) $(LIB_GUI) $(LINK_LIBS) $(ICU_LIBS)
 
 AllObjects = $(SERVER_Objects)
 
-- 
2.4.3

From 715f7f59c04684720bd120c87a905052aececc92 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 26 Nov 2014 09:27:40 +0300
Subject: [PATCH] For mingw place dll's into BIN and generate import libraries
 in LIB. Don't link with def files.

---
 builds/posix/make.defaults | 13 +++++++------
 builds/posix/prefix.mingw  | 38 ++++++++++++++++++++++++--------------
 2 files changed, 31 insertions(+), 20 deletions(-)

diff --git a/builds/posix/make.defaults b/builds/posix/make.defaults
index bced0da..34fd3e9 100755
@@ -93,6 +93,11 @@
 
 CpuType=@CPU_TYPE@
 
+ifeq ($(CpuType), x86_64)
+  BITNESS_SUFFIX := 64
+else
+  BITNESS_SUFFIX := 32
+endif
 PLATFORM=@PLATFORM@
 ifeq (@FIREBIRD_ARCH_TYPE@,classic)
 	ArchPrefix=CS
--- a/builds/posix/make.defaults
+++ b/builds/posix/make.defaults
@@ -193,9 +193,9 @@ BLRTABLE	= $(BIN)/blrtable$(EXEC_EXT)
 #     Scold me, but I don't want library names to be in configure.in
 # 
 
-SharedLibraryName=libfbembed.${SHRLIB_EXT}.${FirebirdVersion}
-SharedLibrarySoName=libfbembed.${SHRLIB_EXT}.${MajorVer}.${MinorVer}
-SharedLibraryBaseName=libfbembed.${SHRLIB_EXT}
+SharedLibraryBaseName=$(LIB_PREFIX)fbembed.${SHRLIB_EXT}
+SharedLibraryName=${SharedLibraryBaseName}.${FirebirdVersion}
+SharedLibrarySoName=${SharedLibraryBaseName}.${MajorVer}.${MinorVer}
 
 LIBFBEMBED_SO = $(LIB)/$(SharedLibraryName)
 LIBFBEMBED_SOBASENAME = $(LIB)/$(SharedLibrarySoName)
@@ -261,6 +261,7 @@ LIB_LINK_RPATH = $(call LIB_PLATFORM_RPATH,'$$$$ORIGIN/../$(1)')
 endif
 LIB_PATH_OPTS = $(call LIB_LINK_RPATH,lib) $(call LIB_LINK_RPATH,intl)
 LIB_LINK_SONAME= -Wl,-soname,$(1)
+LIB_LINK_IMPLIB=
 LIB_LINK_MAPFILE= -Wl,--version-script,
 FBEMBED_LINK= -L$(LIB) -lfbembed
 
@@ -281,7 +282,7 @@ LINK_UDF = $(LIB_LINK) $(LIB_LINK_OPTIONS) $(call LIB_LINK_SONAME,$(1).$(SHRLIB_
 LINK_UDF_LIBS = $(THR_LIBS) -L$(LIB) -lib_util
 
 LINK_IB_UTIL = $(LIB_LINK) $(LINK_IBUTIL_SYMBOLS) $(LIB_LINK_OPTIONS)\
- $(call LIB_LINK_SONAME,$(IbUtilLibraryName)) $(call LIB_LINK_RPATH,lib)
+ $(call LIB_LINK_IMPLIB,$(LIB)/lib$(IbUtilLibraryName).a) $(call LIB_LINK_SONAME,$(IbUtilLibraryName)) $(call LIB_LINK_RPATH,lib)
 LINK_IB_UTIL_LIBS = $(THR_LIBS)
 
 LINK_INTL = $(LIB_LINK) $(LINK_FBINTL_SYMBOLS) $(LIB_LINK_OPTIONS)\
@@ -293,11 +294,11 @@ LINK_TRACE = $(LIB_LINK) $(LINK_TRACE_SYMBOLS) $(LIB_LINK_OPTIONS)\
 LINK_TRACE_LIBS = -L$(LIB) $(SO_LINK_LIBS)
 
 LINK_CLIENT = $(CLIENTLIB_LINK) $(LINK_FIREBIRD_CLIENT_SYMBOLS) $(LIB_LINK_OPTIONS)\
- $(LIB_LINK_IMPLIB) $(call LIB_LINK_SONAME,$(ClientLibrarySoName)) $(call LIB_LINK_RPATH,lib)
+ $(call LIB_LINK_IMPLIB,$(LIB)/lib$(ClientLibraryName).a) $(call LIB_LINK_SONAME,$(ClientLibrarySoName)) $(call LIB_LINK_RPATH,lib)
 LINK_CLIENT_LIBS = $(SO_LINK_LIBS) $(STATIC_CXXSUPPORT_LIB)
 
 LINK_EMBED = $(LIB_LINK) $(LINK_FIREBIRD_EMBED_SYMBOLS) $(LIB_LINK_OPTIONS) $(LIB_EMBED_LINK_OPTIONS)\
- $(call LIB_LINK_SONAME,$(SharedLibrarySoName)) $(call LIB_LINK_RPATH,lib)
+ $(call LIB_LINK_IMPLIB,$(LIB)/lib$(SharedLibraryBaseName).a) $(call LIB_LINK_SONAME,$(SharedLibrarySoName)) $(call LIB_LINK_RPATH,lib)
 LINK_EMBED_LIBS = -L$(LIB) $(LIB_GUI) $(SO_LINK_LIBS) $(ICU_LIBS)
 
 # From utilities 
diff --git a/builds/posix/prefix.mingw b/builds/posix/prefix.mingw
index ac1327c..09a5103 100644
--- a/builds/posix/prefix.mingw
+++ b/builds/posix/prefix.mingw
@@ -36,16 +36,12 @@ OS_ServerFiles= os/win32/srvr_w32.cpp os/win32/window.cpp os/win32/chop.cpp os/w
 # Override link command for implicit stdc++ linking
 LIB_LINK=$(LD)
 
-LIB_LINK_OPTIONS+=-Wl,--enable-stdcall-fixup
-
 # Strip symbols from release versions to decrease size
 ifeq ($(IsProdTypeBuild),Y)
 LINK_OPTS+=-Wl,-s
 LIB_LINK_OPTIONS+=-Wl,-s
 endif
 
-# Generation of fbclient_ms.lib
-LIB_LINK_IMPLIB:=-Wl,--out-implib,firebird/lib/fbclient_ms.lib
 LIB_GUI:= -mwindows -lcomctl32 -lgdi32
 
 
@@ -65,19 +61,33 @@ FilesToAddDialog2=$(IBGUARD)
 LIB_PREFIX=
 
 # Override library names to suppress version suffix
-ClientLibraryNameFull := $(ClientLibraryName)
-ClientLibrarySoName := $(ClientLibraryName)
+SharedLibraryName = $(SharedLibraryBaseName)
+SharedLibrarySoName = $(SharedLibraryBaseName)
+
+LIBFBEMBED_SO = $(BIN)/$(SharedLibraryName)
+LIBFBEMBED_SOBASENAME = $(BIN)/$(SharedLibrarySoName)
+LIBFBEMBED_SONAME = $(BIN)/$(SharedLibraryBaseName)
+
+ClientLibraryNameFull = $(ClientLibraryName)
+ClientLibrarySoName = $(ClientLibraryName)
+
+LIBFBCLIENT_SO = $(BIN)/$(ClientLibraryNameFull)
+LIBFBCLIENT_SOBASENAME = $(BIN)/$(ClientLibrarySoName)
+LIBFBCLIENT_SONAME = $(BIN)/$(ClientLibraryName)
 
-# Looks like MinGW 3 does not support version scripts but support def-files
-LINK_FIREBIRD_SYMBOLS = $(BLD_ROOT)/win32/defs/fbclient_s.def $(BLD_ROOT)/win32/defs/fbclient.def
+LIBIBUTIL_SO = $(BIN)/$(IbUtilLibraryName)
 
-# This is required for newly built executable to find newly built shared
-# libraries because on Win32 there is no such thing as LD_LIBRARY_PATH
-LIB=$(BIN)
+# Generation of import libraries
+LIB_LINK_IMPLIB= -Wl,--out-implib,$(1)
+LIB_LINK_SONAME=
 
-# Special options for UDF link
-LINK_UDF = $(LIB_LINK) $(LIB_LINK_OPTIONS) $(call LIB_LINK_SONAME,$(1).$(SHRLIB_EXT)) -L$(LIB) -lib_util
-LINK_UDF_LIBS = 
+LINK_FIREBIRD_SYMBOLS = $(BLD_ROOT)/win32/defs/fbclient$(BITNESS_SUFFIX).def
+#LINK_FIREBIRD_EMBED_SYMBOLS =
+#LINK_FIREBIRD_CLIENT_SYMBOLS =
+LINK_TRACE_SYMBOLS =
+LINK_FBINTL_SYMBOLS =
+LINK_IBUTIL_SYMBOLS =
+LINK_EMPTY_SYMBOLS =
 
 # Special options for trace plugin link
 LINK_TRACE_LIBS = -L$(LIB) $(SO_LINK_LIBS) $(ICU_LIBS)
diff --git a/builds/win32/defs/fbclient32.def b/builds/win32/defs/fbclient32.def
new file mode 100644
index d345348..346a94d 100644
--- /dev/null
+++ b/builds/win32/defs/fbclient32.def
@@ -0,0 +1,361 @@
+; The contents of this file are subject to the Interbase Public
+; License Version 1.0 (the "License"); you may not use this file
+; except in compliance with the License. You may obtain a copy
+; of the License at http://www.Inprise.com/IPL.html
+;
+; Software distributed under the License is distributed on an
+; "AS IS" basis, WITHOUT WARRANTY OF ANY KIND, either express
+; or implied. See the License for the specific language governing
+; rights and limitations under the License.
+;
+; The Original Code was created by Inprise Corporation
+; and its predecessors. Portions created by Inprise Corporation are
+; Copyright (C) Inprise Corporation.
+;
+; All Rights Reserved.
+; Contributor(s): ______________________________________.
+
+LIBRARY "fbclient.dll"
+EXPORTS
+
+;-------------------
+; PRIVATE INTERFACE
+;-------------------
+
+	gds__alloc@4	@1
+	gds__attach_database@24	@2
+	gds__blob_info@24	@3
+	gds__blob_size@16	@4
+	gds__cancel_blob@8	@5
+	gds__cancel_events@12	@6
+	gds__close@8	@7
+	gds__close_blob@8	@8
+	gds__commit_retaining@8	@9
+	gds__commit_transaction@8	@10
+	gds__compile_request@20	@11
+	gds__compile_request2@20	@12
+	gds__create_blob@20	@13
+	gds__create_blob2@28	@14
+	gds__create_database@28	@15
+	gds__database_cleanup@16	@16
+	gds__database_info@24	@17
+	gds__ddl@20	@18
+	gds__declare@12	@19
+	gds__decode@12	@20
+	gds__decode_date@8	@21
+	gds__describe@12	@22
+	gds__describe_bind@12	@23
+	gds__detach_database@8	@24
+	gds__disable_subsystem@4	@25
+	gds__dsql_finish@4	@26
+	gds__edit@8	@27
+	gds__enable_subsystem@4	@28
+	gds__encode@8	@29
+	gds__encode_date@8	@30
+	gds__event_block	@31
+	gds__event_counts@16	@32
+	gds__event_wait@20	@33
+	gds__execute@16	@34
+	gds__execute_immediate@20	@35
+	gds__fetch@12	@36
+	gds__free@4	@37
+	gds__ftof@16	@38
+	gds__get_prefix@8	@248
+	gds__get_segment@20	@39
+	gds__get_slice@44	@40
+	gds__interprete@8	@41
+	gds__log	@42
+	gds__log_status@8	@142
+	gds__map_blobs@8	@240
+	gds__msg_close@4	@223
+	gds__msg_format@40	@43
+	gds__msg_lookup@24	@44
+	gds__msg_open@8	@45
+	gds__msg_put@32	@46
+	gds__open@16	@47
+	gds__open_blob@20	@48
+	gds__open_blob2@28	@49
+	gds__parse_bpb@16	@50
+	gds__prefix@8	@51
+	gds__prefix_lock@8	@249
+	gds__prefix_msg@8	@250
+	gds__prepare@28	@52
+	gds__prepare_transaction@8	@53
+	gds__prepare_transaction2@16	@54
+	gds__print_blr@16	@55
+	gds__print_status@4	@56
+	gds__put_error@4	@57
+	gds__put_segment@16	@58
+	gds__put_slice@40	@59
+	gds__qtoq@8	@60
+	gds__que_events@28	@61
+	gds__receive@24	@62
+	gds__reconnect_transaction@20	@63
+	gds__register_cleanup@8	@64
+	gds__release_request@8	@65
+	gds__request_info@28	@66
+	gds__rollback_transaction@8	@67
+	gds__seek_blob@20	@68
+	gds__send@24	@69
+	gds__set_debug@4	@70
+	gds__sqlcode@4	@71
+	gds__start_and_send@28	@72
+	gds__start_multiple@16	@73
+	gds__start_request@16	@74
+	gds__start_transaction	@75
+	gds__temp_file@20	@231
+	gds__thread_enable@4	@76
+	gds__thread_enter@0	@77
+	gds__thread_exit@0	@78
+	gds__thread_start@20	@79
+	gds__to_sqlda@20	@80
+	gds__transaction_cleanup@16	@235
+	gds__transaction_info@24	@81
+	gds__unregister_cleanup@8	@224
+	gds__unwind_request@12	@82
+	gds__validate_lib_path@16	@255
+	gds__vax_integer@8	@83
+	gds__version@12	@84
+	gds__vtof@12	@85
+	gds__vtov@12	@86
+	gds__default_printer @87
+
+	gds__alloc_debug@12
+
+	gds_alloc_flag_unfreed@4
+	gds_alloc_report@12
+
+	fb_shutdown@8
+	fb_shutdown_callback@16
+	fb_cancel_operation@12
+	fb_disconnect_transaction@8
+
+	perf_format@20	@100
+	perf_get_info@8	@101
+	perf_report@16	@102
+
+	perf64_format@20
+	perf64_get_info@8
+	perf64_report@16
+
+;----------------------------
+; PUBLIC INTERFACE, i.e. API
+;----------------------------
+
+	; Array functions
+
+	isc_array_gen_sdl@20	@217
+	isc_array_get_slice@28	@218
+	isc_array_lookup_bounds@24	@219
+	isc_array_lookup_desc@24	@220
+	isc_array_put_slice@28	@221
+	isc_array_set_desc@28	@222
+	isc_get_slice@44	@140
+	isc_put_slice@40	@153
+
+	; BLOB functions
+
+	isc_blob_default_desc@12	@239
+	isc_blob_gen_bpb@24	@236
+	isc_blob_info@24	@108
+	isc_blob_lookup_desc@28	@237
+	isc_blob_set_desc@28	@238
+	isc_cancel_blob@8	@110
+	isc_close_blob@8	@113
+	isc_create_blob@20	@118
+	isc_create_blob2@28	@119
+	isc_get_segment@20	@139
+	isc_open_blob@20	@144
+	isc_open_blob2@28	@145
+	isc_put_segment@16	@152
+	
+	; Database functions
+
+	isc_attach_database@24	@103
+	isc_create_database@28	@120
+	isc_database_info@24	@122
+	isc_detach_database@8	@128
+	isc_drop_database@8	@229
+	isc_expand_dpb	@135
+	isc_modify_dpb@20	@244
+	isc_version@12	@174
+	isc_get_client_major_version@0	@268
+	isc_get_client_minor_version@0	@269
+	isc_get_client_version@4	@270
+
+	; Conversion functions
+
+	isc_decode_date@8	@125
+	isc_decode_sql_date@8	@261
+	isc_decode_sql_time@8	@262
+	isc_decode_timestamp@8	@263
+	isc_encode_date@8	@129
+	isc_encode_sql_date@8	@264
+	isc_encode_sql_time@8	@265
+	isc_encode_timestamp@8	@266
+	isc_portable_integer@8	@259
+	isc_vax_integer@8	@173
+
+	; DSQL functions
+
+	isc_dsql_allocate_statement@12	@178
+	isc_dsql_alloc_statement2@12	@179
+	isc_dsql_describe@16	@180
+	isc_dsql_describe_bind@16	@181
+	isc_dsql_execute@20	@182
+	isc_dsql_execute2@24	@183
+	isc_dsql_execute_m@32	@184
+	isc_dsql_execute2_m@52	@185
+	isc_dsql_execute_immediate@28	@186
+	isc_dsql_exec_immed2@32	@187
+	isc_dsql_execute_immediate_m@44	@188
+	isc_dsql_exec_immed2_m@64	@189
+	isc_dsql_exec_immed3_m@64	@230
+	isc_dsql_fetch@16	@190
+	isc_dsql_fetch_a@20	@191
+	isc_dsql_fetch_m@28	@192
+	isc_dsql_finish@4	@193
+	isc_dsql_free_statement@12	@194
+	isc_dsql_insert@16	@195
+	isc_dsql_insert_m@28	@196
+	isc_dsql_prepare@28	@197
+	isc_dsql_prepare_m@40	@198
+	isc_dsql_release@8	@199
+	isc_dsql_set_cursor_name@16	@200
+	isc_dsql_sql_info@24	@201
+
+	; ESQL functions
+
+	isc_embed_dsql_close@8	@202
+	isc_embed_dsql_declare@12	@203
+	isc_embed_dsql_describe@16	@204
+	isc_embed_dsql_describe_bind@16	@205
+	isc_embed_dsql_execute@20	@206
+	isc_embed_dsql_execute2@24	@207
+	isc_embed_dsql_execute_immed@28	@208
+	isc_embed_dsql_exec_immed2@32	@209
+	isc_embed_dsql_fetch@16	@210
+	isc_embed_dsql_fetch_a@20	@211
+	isc_embed_dsql_insert@16	@212
+	isc_embed_dsql_open@20	@213
+	isc_embed_dsql_open2@24	@214
+	isc_embed_dsql_prepare@32	@215
+	isc_embed_dsql_release@8	@216
+	isc_embed_dsql_length@8	@273
+
+	; Error-handling functions
+
+	isc_interprete@8	@141
+	fb_interpret@12
+	isc_print_sqlerror@8	@150
+	isc_print_status@4	@151
+	isc_sqlcode@4	@165
+	isc_sql_interprete@12	@164
+	isc_sqlcode_s@8		@278
+	fb_sqlstate@8			@279
+
+	; Event functions
+
+	isc_cancel_events@12	@111
+	isc_event_block	@130
+	isc_event_counts@16	@131
+	isc_que_events@28	@155
+	isc_wait_for_event@20	@177
+	isc_event_block_a@16	@274
+	isc_event_block_s@20	@275
+
+	; Security functions
+
+	isc_add_user@8	@256
+	isc_delete_user@8	@257
+	isc_modify_user@8	@258
+
+	; Services functions
+
+	isc_service_attach@24	@225
+	isc_service_detach@8	@226
+	isc_service_query@36	@227
+	isc_service_start@20	@228
+
+	; Transaction control functions
+
+	isc_commit_retaining@8	@114
+	isc_commit_transaction@8	@115
+	isc_prepare_transaction@8	@147
+	isc_prepare_transaction2@16	@148
+	isc_rollback_retaining@8	@260
+	isc_rollback_transaction@8	@160
+	isc_start_multiple@16	@167
+	isc_start_transaction	@169
+	isc_transaction_info@24	@171
+
+	; Other functions
+
+	isc_compile_request@20	@116
+	isc_compile_request2@20	@117
+	isc_ddl@20	@123
+	isc_prepare@28	@146
+	isc_receive@24	@156
+	isc_reconnect_transaction@20	@157
+	isc_release_request@8	@158
+	isc_request_info@28	@159
+	isc_seek_blob@20	@161
+	isc_send@24	@162
+	isc_start_and_send@28	@166
+	isc_start_request@16	@168
+	isc_transact_request@36	@132
+	isc_unwind_request@12	@172
+
+	; Other SQL functions
+
+	isc_close@8	@112
+	isc_declare@12	@124
+	isc_describe@12	@126
+	isc_describe_bind@12	@127
+	isc_execute@16	@133
+	isc_execute_immediate@20	@134
+	isc_fetch@12	@136
+	isc_fetch_a@16	@137
+	isc_open@16	@143
+
+	; Other BLOB functions
+
+	BLOB_close@4	@91
+	BLOB_display@16	@92
+	BLOB_dump@16	@93
+	BLOB_edit@16	@94
+	BLOB_get@4	@95
+	BLOB_load@16	@96
+	BLOB_open@12	@97
+	BLOB_put@8	@98
+	BLOB_text_dump@16	@242
+	BLOB_text_load@16	@243
+	Bopen@16	@99
+
+	; Other misc functions
+
+	isc_ftof@16	@138
+	isc_free@4	@241
+	isc_print_blr@16	@149
+	fb_print_blr@20	@280
+	isc_qtoq@8	@154
+	isc_reset_fpe@4
+	isc_set_debug@4	@163
+	isc_to_sqlda@20	@170
+	isc_vtof@12	@175
+	isc_vtov@12	@176
+	isc_baddress@4	@276
+	isc_baddress_s@8	@277
+
+	_isc_start_transaction=isc_start_transaction
+	_isc_expand_dpb=isc_expand_dpb
+	_isc_event_block=isc_event_block
+
+;-----------------
+; OTHER FUNCTIONS
+;-----------------
+
+	CVT_move@12	@267
+
+	KEYWORD_stringIsAToken@4	@271
+	KEYWORD_getTokens@0	@272
diff --git a/builds/win32/defs/fbclient64.def b/builds/win32/defs/fbclient64.def
new file mode 100644
index d345348..346a94d 100644
--- /dev/null
+++ b/builds/win32/defs/fbclient64.def
@@ -0,0 +1,361 @@
+; The contents of this file are subject to the Interbase Public
+; License Version 1.0 (the "License"); you may not use this file
+; except in compliance with the License. You may obtain a copy
+; of the License at http://www.Inprise.com/IPL.html
+;
+; Software distributed under the License is distributed on an
+; "AS IS" basis, WITHOUT WARRANTY OF ANY KIND, either express
+; or implied. See the License for the specific language governing
+; rights and limitations under the License.
+;
+; The Original Code was created by Inprise Corporation
+; and its predecessors. Portions created by Inprise Corporation are
+; Copyright (C) Inprise Corporation.
+;
+; All Rights Reserved.
+; Contributor(s): ______________________________________.
+
+LIBRARY "fbclient.dll"
+EXPORTS
+
+;-------------------
+; PRIVATE INTERFACE
+;-------------------
+
+	gds__alloc	@1
+	gds__attach_database	@2
+	gds__blob_info	@3
+	gds__blob_size	@4
+	gds__cancel_blob	@5
+	gds__cancel_events	@6
+	gds__close	@7
+	gds__close_blob	@8
+	gds__commit_retaining	@9
+	gds__commit_transaction	@10
+	gds__compile_request	@11
+	gds__compile_request2	@12
+	gds__create_blob	@13
+	gds__create_blob2	@14
+	gds__create_database	@15
+	gds__database_cleanup	@16
+	gds__database_info	@17
+	gds__ddl	@18
+	gds__declare	@19
+	gds__decode	@20
+	gds__decode_date	@21
+	gds__describe	@22
+	gds__describe_bind	@23
+	gds__detach_database	@24
+	gds__disable_subsystem	@25
+	gds__dsql_finish	@26
+	gds__edit	@27
+	gds__enable_subsystem	@28
+	gds__encode	@29
+	gds__encode_date	@30
+	gds__event_block	@31
+	gds__event_counts	@32
+	gds__event_wait	@33
+	gds__execute	@34
+	gds__execute_immediate	@35
+	gds__fetch	@36
+	gds__free	@37
+	gds__ftof	@38
+	gds__get_prefix	@248
+	gds__get_segment	@39
+	gds__get_slice	@40
+	gds__interprete	@41
+	gds__log	@42
+	gds__log_status	@142
+	gds__map_blobs	@240
+	gds__msg_close	@223
+	gds__msg_format	@43
+	gds__msg_lookup	@44
+	gds__msg_open	@45
+	gds__msg_put	@46
+	gds__open	@47
+	gds__open_blob	@48
+	gds__open_blob2	@49
+	gds__parse_bpb	@50
+	gds__prefix	@51
+	gds__prefix_lock	@249
+	gds__prefix_msg	@250
+	gds__prepare	@52
+	gds__prepare_transaction	@53
+	gds__prepare_transaction2	@54
+	gds__print_blr	@55
+	gds__print_status	@56
+	gds__put_error	@57
+	gds__put_segment	@58
+	gds__put_slice	@59
+	gds__qtoq	@60
+	gds__que_events	@61
+	gds__receive	@62
+	gds__reconnect_transaction	@63
+	gds__register_cleanup	@64
+	gds__release_request	@65
+	gds__request_info	@66
+	gds__rollback_transaction	@67
+	gds__seek_blob	@68
+	gds__send	@69
+	gds__set_debug	@70
+	gds__sqlcode	@71
+	gds__start_and_send	@72
+	gds__start_multiple	@73
+	gds__start_request	@74
+	gds__start_transaction	@75
+	gds__temp_file	@231
+	gds__thread_enable	@76
+	gds__thread_enter	@77
+	gds__thread_exit	@78
+	gds__thread_start	@79
+	gds__to_sqlda	@80
+	gds__transaction_cleanup	@235
+	gds__transaction_info	@81
+	gds__unregister_cleanup	@224
+	gds__unwind_request	@82
+	gds__validate_lib_path	@255
+	gds__vax_integer	@83
+	gds__version	@84
+	gds__vtof	@85
+	gds__vtov	@86
+	gds__default_printer @87
+
+    gds__alloc_debug
+
+    gds_alloc_flag_unfreed
+    gds_alloc_report
+
+	fb_shutdown
+	fb_shutdown_callback
+	fb_cancel_operation
+	fb_disconnect_transaction
+
+	perf_format	@100
+	perf_get_info	@101
+	perf_report	@102
+
+	perf64_format
+	perf64_get_info
+	perf64_report
+
+;----------------------------
+; PUBLIC INTERFACE, i.e. API
+;----------------------------
+
+	; Array functions
+
+	isc_array_gen_sdl	@217
+	isc_array_get_slice	@218
+	isc_array_lookup_bounds	@219
+	isc_array_lookup_desc	@220
+	isc_array_put_slice	@221
+	isc_array_set_desc	@222
+	isc_get_slice	@140
+	isc_put_slice	@153
+
+	; BLOB functions
+
+	isc_blob_default_desc	@239
+	isc_blob_gen_bpb	@236
+	isc_blob_info	@108
+	isc_blob_lookup_desc	@237
+	isc_blob_set_desc	@238
+	isc_cancel_blob	@110
+	isc_close_blob	@113
+	isc_create_blob	@118
+	isc_create_blob2	@119
+	isc_get_segment	@139
+	isc_open_blob	@144
+	isc_open_blob2	@145
+	isc_put_segment	@152
+	
+	; Database functions
+
+	isc_attach_database	@103
+	isc_create_database	@120
+	isc_database_info	@122
+	isc_detach_database	@128
+	isc_drop_database	@229
+	isc_expand_dpb	@135
+	isc_modify_dpb	@244
+	isc_version	@174
+	isc_get_client_major_version	@268
+	isc_get_client_minor_version	@269
+	isc_get_client_version	@270
+
+	; Conversion functions
+
+	isc_decode_date	@125
+	isc_decode_sql_date	@261
+	isc_decode_sql_time	@262
+	isc_decode_timestamp	@263
+	isc_encode_date	@129
+	isc_encode_sql_date	@264
+	isc_encode_sql_time	@265
+	isc_encode_timestamp	@266
+	isc_portable_integer	@259
+	isc_vax_integer	@173
+
+	; DSQL functions
+
+	isc_dsql_allocate_statement	@178
+	isc_dsql_alloc_statement2	@179
+	isc_dsql_describe	@180
+	isc_dsql_describe_bind	@181
+	isc_dsql_execute	@182
+	isc_dsql_execute2	@183
+	isc_dsql_execute_m	@184
+	isc_dsql_execute2_m	@185
+	isc_dsql_execute_immediate	@186
+	isc_dsql_exec_immed2	@187
+	isc_dsql_execute_immediate_m	@188
+	isc_dsql_exec_immed2_m	@189
+	isc_dsql_exec_immed3_m	@230
+	isc_dsql_fetch	@190
+	isc_dsql_fetch_a	@191
+	isc_dsql_fetch_m	@192
+	isc_dsql_finish	@193
+	isc_dsql_free_statement	@194
+	isc_dsql_insert	@195
+	isc_dsql_insert_m	@196
+	isc_dsql_prepare	@197
+	isc_dsql_prepare_m	@198
+	isc_dsql_release	@199
+	isc_dsql_set_cursor_name	@200
+	isc_dsql_sql_info	@201
+
+	; ESQL functions
+
+	isc_embed_dsql_close	@202
+	isc_embed_dsql_declare	@203
+	isc_embed_dsql_describe	@204
+	isc_embed_dsql_describe_bind	@205
+	isc_embed_dsql_execute	@206
+	isc_embed_dsql_execute2	@207
+	isc_embed_dsql_execute_immed	@208
+	isc_embed_dsql_exec_immed2	@209
+	isc_embed_dsql_fetch	@210
+	isc_embed_dsql_fetch_a	@211
+	isc_embed_dsql_insert	@212
+	isc_embed_dsql_open	@213
+	isc_embed_dsql_open2	@214
+	isc_embed_dsql_prepare	@215
+	isc_embed_dsql_release	@216
+	isc_embed_dsql_length	@273
+
+	; Error-handling functions
+
+	isc_interprete	@141
+	fb_interpret
+	isc_print_sqlerror	@150
+	isc_print_status	@151
+	isc_sqlcode	@165
+	isc_sql_interprete	@164
+	isc_sqlcode_s		@278
+	fb_sqlstate			@279
+
+	; Event functions
+
+	isc_cancel_events	@111
+	isc_event_block	@130
+	isc_event_counts	@131
+	isc_que_events	@155
+	isc_wait_for_event	@177
+	isc_event_block_a	@274
+	isc_event_block_s	@275
+
+	; Security functions
+
+	isc_add_user	@256
+	isc_delete_user	@257
+	isc_modify_user	@258
+
+	; Services functions
+
+	isc_service_attach	@225
+	isc_service_detach	@226
+	isc_service_query	@227
+	isc_service_start	@228
+
+	; Transaction control functions
+
+	isc_commit_retaining	@114
+	isc_commit_transaction	@115
+	isc_prepare_transaction	@147
+	isc_prepare_transaction2	@148
+	isc_rollback_retaining	@260
+	isc_rollback_transaction	@160
+	isc_start_multiple	@167
+	isc_start_transaction	@169
+	isc_transaction_info	@171
+
+	; Other functions
+
+	isc_compile_request	@116
+	isc_compile_request2	@117
+	isc_ddl	@123
+	isc_prepare	@146
+	isc_receive	@156
+	isc_reconnect_transaction	@157
+	isc_release_request	@158
+	isc_request_info	@159
+	isc_seek_blob	@161
+	isc_send	@162
+	isc_start_and_send	@166
+	isc_start_request	@168
+	isc_transact_request	@132
+	isc_unwind_request	@172
+
+	; Other SQL functions
+
+	isc_close	@112
+	isc_declare	@124
+	isc_describe	@126
+	isc_describe_bind	@127
+	isc_execute	@133
+	isc_execute_immediate	@134
+	isc_fetch	@136
+	isc_fetch_a	@137
+	isc_open	@143
+
+	; Other BLOB functions
+
+	BLOB_close	@91
+	BLOB_display	@92
+	BLOB_dump	@93
+	BLOB_edit	@94
+	BLOB_get	@95
+	BLOB_load	@96
+	BLOB_open	@97
+	BLOB_put	@98
+	BLOB_text_dump	@242
+	BLOB_text_load	@243
+	Bopen	@99
+
+	; Other misc functions
+
+	isc_ftof	@138
+	isc_free	@241
+	isc_print_blr	@149
+	fb_print_blr	@280
+	isc_qtoq	@154
+	isc_reset_fpe
+	isc_set_debug	@163
+	isc_to_sqlda	@170
+	isc_vtof	@175
+	isc_vtov	@176
+	isc_baddress	@276
+	isc_baddress_s	@277
+
+	_isc_start_transaction=isc_start_transaction
+	_isc_expand_dpb=isc_expand_dpb
+	_isc_event_block=isc_event_block
+
+;-----------------
+; OTHER FUNCTIONS
+;-----------------
+
+	CVT_move	@267
+
+	KEYWORD_stringIsAToken	@271
+	KEYWORD_getTokens	@272
-- 
2.4.3

From 8fc62a800a470a31ad36be09504310f765600f4f Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 26 Nov 2014 12:25:47 +0300
Subject: [PATCH] Don't install unix files for Windows platform

---
 configure.in | 66 ++++++++++++++++++++++++++++++++++++------------------------
 1 file changed, 40 insertions(+), 26 deletions(-)

diff --git a/configure.in b/configure.in
index d345348..346a94d 100644
--- a/configure.in
+++ b/configure.in
@@ -1313,32 +1313,39 @@ MAKE_SRC_DIR=builds/posix
 
 dnl common files for all posix hosts
 dnl TODO: fix "arch-specific/linux/" paths for common posix scripts with SVN
-AC_CONFIG_FILES([gen/install/misc/firebird.conf:builds/install/misc/firebird.conf.in
-gen/install/misc/aliases.conf:builds/install/misc/aliases.conf.in
-gen/install/misc/fbintl.conf:builds/install/misc/fbintl.conf
-gen/install/misc/fbtrace.conf:src/utilities/ntrace/fbtrace.conf
-gen/firebird/bin/fb_config:builds/install/misc/fb_config.in
-gen/firebird/bin/posixLibrary.sh:builds/install/misc/posixLibrary.sh.in
-gen/firebird/bin/classicLibrary.sh:builds/install/misc/classicLibrary.sh.in
-gen/firebird/bin/superLibrary.sh:builds/install/misc/superLibrary.sh.in
-gen/firebird/bin/changeRunUser.sh:builds/install/misc/changeRunUser.sh.in
-gen/firebird/bin/restoreRootRunUser.sh:builds/install/misc/restoreRootRunUser.sh.in
-gen/firebird/bin/tarMainInstall.sh:builds/install/arch-specific/linux/misc/tarMainInstall.sh.in
-gen/firebird/bin/tarinstall.sh:builds/install/arch-specific/linux/misc/tarinstall.sh.in
-gen/firebird/bin/preinstall.sh:builds/install/arch-specific/linux/misc/preinstall.sh.in
-gen/firebird/bin/postinstall.sh:builds/install/arch-specific/linux/misc/postinstall.sh.in
-gen/firebird/bin/preuninstall.sh:builds/install/arch-specific/linux/misc/preuninstall.sh.in
-gen/firebird/bin/postuninstall.sh:builds/install/arch-specific/linux/misc/postuninstall.sh.in
-gen/firebird/bin/taruninstall.sh:builds/install/arch-specific/linux/misc/taruninstall.sh.in
-gen/firebird/bin/tarMainUninstall.sh:builds/install/arch-specific/linux/misc/tarMainUninstall.sh.in
-gen/firebird/bin/changeDBAPassword.sh:builds/install/misc/changeDBAPassword.sh.in
-gen/firebird/bin/changeGdsLibraryCompatibleLink.sh:builds/install/misc/changeGdsLibraryCompatibleLink.sh.in
-gen/firebird/bin/createAliasDB.sh:builds/install/misc/createAliasDB.sh.in],
-[chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
-
-if test $FIREBIRD_ARCH_TYPE = classic; then
-	AC_CONFIG_FILES([gen/firebird/bin/changeMultiConnectMode.sh:builds/install/misc/changeMultiConnectMode.sh.in],
-	[chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+if test $PLATFORM = win32; then
+  AC_CONFIG_FILES(gen/firebird/firebird.conf:builds/install/misc/firebird.conf.in
+  gen/firebird/aliases.conf:builds/install/misc/aliases.conf.in
+  gen/firebird/intl/fbintl.conf:builds/install/misc/fbintl.conf
+  gen/firebird/fbtrace.conf:src/utilities/ntrace/fbtrace.conf)
+else
+  AC_CONFIG_FILES([gen/install/misc/firebird.conf:builds/install/misc/firebird.conf.in
+  gen/install/misc/aliases.conf:builds/install/misc/aliases.conf.in
+  gen/install/misc/fbintl.conf:builds/install/misc/fbintl.conf
+  gen/install/misc/fbtrace.conf:src/utilities/ntrace/fbtrace.conf
+  gen/firebird/bin/fb_config:builds/install/misc/fb_config.in
+  gen/firebird/bin/posixLibrary.sh:builds/install/misc/posixLibrary.sh.in
+  gen/firebird/bin/classicLibrary.sh:builds/install/misc/classicLibrary.sh.in
+  gen/firebird/bin/superLibrary.sh:builds/install/misc/superLibrary.sh.in
+  gen/firebird/bin/changeRunUser.sh:builds/install/misc/changeRunUser.sh.in
+  gen/firebird/bin/restoreRootRunUser.sh:builds/install/misc/restoreRootRunUser.sh.in
+  gen/firebird/bin/tarMainInstall.sh:builds/install/arch-specific/linux/misc/tarMainInstall.sh.in
+  gen/firebird/bin/tarinstall.sh:builds/install/arch-specific/linux/misc/tarinstall.sh.in
+  gen/firebird/bin/preinstall.sh:builds/install/arch-specific/linux/misc/preinstall.sh.in
+  gen/firebird/bin/postinstall.sh:builds/install/arch-specific/linux/misc/postinstall.sh.in
+  gen/firebird/bin/preuninstall.sh:builds/install/arch-specific/linux/misc/preuninstall.sh.in
+  gen/firebird/bin/postuninstall.sh:builds/install/arch-specific/linux/misc/postuninstall.sh.in
+  gen/firebird/bin/taruninstall.sh:builds/install/arch-specific/linux/misc/taruninstall.sh.in
+  gen/firebird/bin/tarMainUninstall.sh:builds/install/arch-specific/linux/misc/tarMainUninstall.sh.in
+  gen/firebird/bin/changeDBAPassword.sh:builds/install/misc/changeDBAPassword.sh.in
+  gen/firebird/bin/changeGdsLibraryCompatibleLink.sh:builds/install/misc/changeGdsLibraryCompatibleLink.sh.in
+  gen/firebird/bin/createAliasDB.sh:builds/install/misc/createAliasDB.sh.in],
+  [chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+
+  if test $FIREBIRD_ARCH_TYPE = classic; then
+    AC_CONFIG_FILES([gen/firebird/bin/changeMultiConnectMode.sh:builds/install/misc/changeMultiConnectMode.sh.in],
+    [chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+  fi
 fi
 
 dnl: Platform Specific Files
@@ -1438,6 +1445,13 @@ case "$PLATFORM" in
 
   DARWIN)
     ;;
+  
+  win32)
+    AC_CONFIG_FILES(gen/firebird/install_classic.bat:builds/install/arch-specific/win32/install_classic.bat
+                     gen/firebird/install_super.bat:builds/install/arch-specific/win32/install_super.bat
+                     gen/firebird/install_superclassic.bat:builds/install/arch-specific/win32/install_superclassic.bat
+                     gen/firebird/uninstall.bat:builds/install/arch-specific/win32/uninstall.bat)
+    ;;
 esac
 if test -f ${MAKE_SRC_DIR}/postfix.${MAKEFILE_POSTFIX};then
 	POSTFIX_FILE=gen/make.platform.postfix:${MAKE_SRC_DIR}/postfix.${MAKEFILE_POSTFIX}
-- 
2.4.3

From 4d220545d836947405485990a9f69c509ccd5278 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 26 Nov 2014 16:15:47 +0300
Subject: [PATCH] Fix rules to use variables.

---
 builds/posix/Makefile.in.examples     | 18 +++++++++---------
 builds/posix/Makefile.in.refDatabases |  8 ++++----
 builds/posix/make.defaults            |  2 +-
 3 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/builds/posix/Makefile.in.examples b/builds/posix/Makefile.in.examples
index 3137649..ade180b 100644
--- a/builds/posix/Makefile.in.examples
+++ b/builds/posix/Makefile.in.examples
@@ -117,9 +117,9 @@ $(EXAMPLES_DEST)% : $(EXAMPLES_SRC)%
 
 
 $(EXAMPLES_DEST)/employee.fdb:	$(EXAMPLES_DEST)/empbuild$(EXEC_EXT) $(INPUT_Sources) $(EXAMPLES_DEST)/isql$(EXEC_EXT)
-	-$(RM) employee.fdb
-	./empbuild employee.fdb
-	-$(CHMOD_6) employee.fdb
+	-$(RM) $(EXAMPLES_DEST)/employee.fdb
+	$(EXAMPLES_DEST)/empbuild$(EXEC_EXT) $(EXAMPLES_DEST)/employee.fdb
+	-$(CHMOD_6) $(EXAMPLES_DEST)/employee.fdb
 
 # To get past the fact isql is called from the programs, we create a local link
 # to the static one in this directory
@@ -134,16 +134,16 @@ $(EXAMPLES_DEST)/empbuild.c:	$(EXAMPLES_DEST)/empbuild.fdb $(EXAMPLES_DEST)/empb
 
 $(EXAMPLES_DEST)/empbuild.fdb : $(EXAMPLES_DEST)/empddl.sql $(EXAMPLES_DEST)/empbld.sql $(EXAMPLES_DEST)/isql$(EXEC_EXT)
 	-$(RM) $(EXAMPLES_DEST)/empbuild.fdb
-	$(EXAMPLES_DEST)/isql$(EXEC_EXT) -i empbld.sql
+	$(EXAMPLES_DEST)/isql$(EXEC_EXT) -i $(EXAMPLES_DEST)/empbld.sql
 
 # The chain for intlemp.fdb is the same a script file to create an empty database
 # to allow a .e program to be compiled, to then create and populate with data
 # the intlemp.fdb database.
 
 $(EXAMPLES_DEST)/intlemp.fdb:	$(EXAMPLES_DEST)/intlbld$(EXEC_EXT) $(INTL_Sources) $(EXAMPLES_DEST)/isql$(EXEC_EXT)
-	-$(RM) intlemp.fdb
-	./intlbld intlemp.fdb
-	-$(CHMOD_6) intlemp.fdb
+	-$(RM) $(EXAMPLES_DEST)/intlemp.fdb
+	$(EXAMPLES_DEST)/intlbld$(EXEC_EXT) $(EXAMPLES_DEST)/intlemp.fdb
+	-$(CHMOD_6) $(EXAMPLES_DEST)/intlemp.fdb
 
 $(EXAMPLES_DEST)/intlbld$(EXEC_EXT): $(INTLBLD_Objects) $(COMMON_Objects) $(LIBFBSTATIC_A) 
 	$(EXE_LINK) $(LINK_OPTS) $(INTLBLD_Objects) $(COMMON_Objects) -o $@ -L$(LIB) -lfbstatic $(LINK_LIBS) $(ICU_LIBS)
@@ -151,8 +151,8 @@ $(EXAMPLES_DEST)/intlbld$(EXEC_EXT): $(INTLBLD_Objects) $(COMMON_Objects) $(LIBF
 $(EXAMPLES_DEST)/intlbld.c:	$(EXAMPLES_DEST)/intlbuild.fdb $(EXAMPLES_DEST)/intlbld.e
 
 $(EXAMPLES_DEST)/intlbuild.fdb : $(EXAMPLES_DEST)/intlddl.sql $(EXAMPLES_DEST)/intlbld.sql $(EXAMPLES_DEST)/isql$(EXEC_EXT)
-	-$(RM) intlbuild.fdb
-	$(EXAMPLES_DEST)/isql$(EXEC_EXT) -i intlbld.sql
+	-$(RM) $(EXAMPLES_DEST)/intlbuild.fdb
+	$(EXAMPLES_DEST)/isql$(EXEC_EXT) -i $(EXAMPLES_DEST)/intlbld.sql
 
 
 $(EXAMPLES_DEST)/%.sql: $(EXAMPLES_SRC)/empbuild/%.sql
diff --git a/builds/posix/Makefile.in.refDatabases b/builds/posix/Makefile.in.refDatabases
index c1b61a6..68016cc 100644
--- a/builds/posix/Makefile.in.refDatabases
+++ b/builds/posix/Makefile.in.refDatabases
@@ -65,7 +65,7 @@ empty_db : empty.fdb yachts.lnk
 empty.fdb : $(CREATE_DB)
 	$(MAKE) -f $(GEN_ROOT)/Makefile.static.createdb
 	-$(RM) $@
-	$(BIN)/create_db $@
+	$(CREATE_DB) $@
 	$(TOUCH) $@
 	$(CHMOD) 444 $@
 
@@ -99,7 +99,7 @@ help.fdb:	$(FIREBIRD)/help/help.fdb
 	$(LN) -f $^ $@
 
 $(FIREBIRD)/help/help.fdb:	$(BLD_ROOT)/misc/help.gbak
-	$(BIN)/gbak_static -MODE read_only -R $< $@
+	$(GBAK_STATIC) -MODE read_only -R $< $@
 	$(TOUCH) $@
 	$(CHMOD) 444 $@
 
@@ -114,7 +114,7 @@ security2.fdb:	$(SRC_ROOT)/dbs/security.sql
 	$(CHMOD) a=rw $@
 
 metadata.fdb:	$(BLD_ROOT)/misc/metadata.gbak
-	$(BIN)/gbak_static -MODE read_only -R $< $@
+	$(GBAK_STATIC) -MODE read_only -R $< $@
 	$(TOUCH) $@
 	$(CHMOD) 444 $@
 
@@ -122,7 +122,7 @@ metadata.fdb:	$(BLD_ROOT)/misc/metadata.gbak
 # since it ensures the data is valid.
 
 metadata.fdb.x:	$(SRC_ROOT)/misc/metadata.sql
-	$(BIN)/isql -i $<
+	$(ISQL) -i $<
 	$(TOUCH) $@
 	$(CHMOD) 444 $@
 
diff --git a/builds/posix/make.defaults b/builds/posix/make.defaults
index 34fd3e9..0f1da2d 100755
--- a/builds/posix/make.defaults
+++ b/builds/posix/make.defaults
@@ -178,7 +178,7 @@ SHRLIB_FOREIGN_EXT=	$(SHRLIB_EXT)
 
 vpath %.so $(LIB)
 vpath %.a $(LIB)
-vpath %.dll $(LIB)
+vpath %.dll $(BIN)
 
 #_____________________________________________________________________________
 
-- 
2.4.3

From 3665f4d104ed9c57c5d4cda4c4213a82e373d83f Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Sun, 30 Nov 2014 15:08:19 +0300
Subject: [PATCH] Auto detect WINDRES with autoconf.

Fix building .rc files.
---
 builds/posix/make.defaults       |  4 +++-
 builds/posix/make.rules          |  4 ++--
 builds/posix/make.shared.targets | 14 +++++++-------
 configure.in                     | 10 ++++++++++
 4 files changed, 22 insertions(+), 10 deletions(-)

diff --git a/builds/posix/make.defaults b/builds/posix/make.defaults
index caa9930..65f8c6c 100755
--- a/builds/posix/make.defaults
+++ b/builds/posix/make.defaults
@@ -151,9 +151,11 @@ ECHO=			echo
 QUIET_ECHO=		@echo
 CD=			cd
 CAT=			cat
-AR=			ar @AR_OPTIONS@ crsu
+AR=			@AR@ @AR_OPTIONS@ crsu
 LN=			@LN_S@
+DLLTOOL=			@DLLTOOL@
 RANLIB=			@RANLIB@
+RC=			@RC@
 BTYACC=$(ROOT)/extern/btyacc/btyacc
 
 CC =	@CC@
diff --git a/builds/posix/make.rules b/builds/posix/make.rules
index 42c719d..601dc82 100644
--- a/builds/posix/make.rules
+++ b/builds/posix/make.rules
@@ -102,5 +102,5 @@ $(OBJ)/%.o: $(SRC_ROOT)/%.cpp
 
 # Rules for making resource files
 
-$(GEN_ROOT)/%.res: $(SRC_ROOT)/%.rc
-	windres --output-format=coff --include-dir=$(<D) $< $@
+$(OBJ)/%.res: $(SRC_ROOT)/%.rc
+	$(RC) --output-format=coff --include-dir=$(<D) $< $@
diff --git a/builds/posix/make.shared.targets b/builds/posix/make.shared.targets
index 07afa91..62097a7 100644
--- a/builds/posix/make.shared.targets
+++ b/builds/posix/make.shared.targets
@@ -75,16 +75,16 @@ $(OBJ)/dsql/PackageNodes.cpp: $(SRC_ROOT)/dsql/PackageNodes.epp
 
 # Adding resources as prerequisite for some files
 
-$(FilesToAddVersionInfo): $(GEN_ROOT)/jrd/version.res
-$(FilesToAddDialog): $(GEN_ROOT)/remote/os/win32/window.res
-$(FilesToAddDialog2): $(GEN_ROOT)/iscguard/iscguard.res
+$(FilesToAddVersionInfo): $(OBJ)/jrd/version.res
+$(FilesToAddDialog): $(OBJ)/remote/os/win32/window.res
+$(FilesToAddDialog2): $(OBJ)/iscguard/iscguard.res
 
 # Explicit dependence of resource script
-$(GEN_ROOT)/remote/os/win32/window.res: $(SRC_ROOT)/remote/os/win32/window.rc $(SRC_ROOT)/remote/os/win32/window.rh \
- $(SRC_ROOT)/jrd/version.rc $(SRC_ROOT)/jrd/build_no.h $(SRC_ROOT)/remote/os/win32/property.rc \
- $(SRC_ROOT)/remote/os/win32/property.rh
+$(OBJ)/remote/os/win32/window.res: $(SRC_ROOT)/remote/os/win32/window.rc $(SRC_ROOT)/remote/os/win32/window.rh \
+ $(SRC_ROOT)/jrd/version.rc $(SRC_ROOT)/jrd/build_no.h $(SRC_ROOT)/remote/os/win32/property.rc \
+ $(SRC_ROOT)/remote/os/win32/property.rh
 
-$(GEN_ROOT)/iscguard/iscguard.res: $(SRC_ROOT)/iscguard/iscguard.rc $(SRC_ROOT)/iscguard/iscguard.rh \
+$(OBJ)/iscguard/iscguard.res: $(SRC_ROOT)/iscguard/iscguard.rc $(SRC_ROOT)/iscguard/iscguard.rh \
  $(SRC_ROOT)/jrd/version.rc 
 
 .PHONY: FORCE
diff --git a/configure.in b/configure.in
index 3e65f10..2e81f32 100644
--- a/configure.in
+++ b/configure.in
@@ -679,6 +679,16 @@ AC_PROG_INSTALL
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_MAKE_SET
 AC_PROG_LIBTOOL
+if test $PLATFORM = win32; then
+  # Checks for mingw resource compiler.
+  AC_CHECK_PROGS([RC],
+     [$target-windres \
+      windres])
+
+  if test -z "${RC}"; then
+    AC_MSG_ERROR(windres command is not found.);
+  fi
+fi
 
 dnl Check for system header files
 AC_HEADER_DIRENT
-- 
2.4.3

From a792754a810b2fe9b4a85b23b0234301ae265961 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 26 Nov 2014 16:23:09 +0300
Subject: [PATCH] Create symlinks of fbembed only if names are different

---
 builds/posix/Makefile.in.libfbembed | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/builds/posix/Makefile.in.libfbembed b/builds/posix/Makefile.in.libfbembed
index d919ba2..2e70ab9 100644
--- a/builds/posix/Makefile.in.libfbembed
+++ b/builds/posix/Makefile.in.libfbembed
@@ -61,14 +61,18 @@ Dependencies = $(AllObjects:.o=.d)
 
 libfbembed : $(LIBIBUTIL_SO) $(LIBFBEMBED_SONAME)
 
-$(LIBFBEMBED_SO): $(LIBFBEMBED_Objects) $(SERVER_Objects) $(COMMON_Objects)
-	$(LINK_EMBED) -o $@ $^ $(LINK_EMBED_LIBS)
-
+ifneq ($(SharedLibraryName),$(SharedLibrarySoName))
 $(LIBFBEMBED_SOBASENAME): $(LIBFBEMBED_SO)
 	(cd $(LIB) && $(LN) -f $(SharedLibraryName) $(SharedLibrarySoName) )
+endif
 
+ifneq ($(SharedLibrarySoName),$(SharedLibraryBaseName))
 $(LIBFBEMBED_SONAME): $(LIBFBEMBED_SOBASENAME)
 	(cd $(LIB) && $(LN) -f $(SharedLibrarySoName) $(SharedLibraryBaseName) )
+endif
+
+$(LIBFBEMBED_SO): $(LIBFBEMBED_Objects) $(SERVER_Objects) $(COMMON_Objects)
+	$(LINK_EMBED) -o $@ $^ $(LINK_EMBED_LIBS)
 
 include $(ROOT)/gen/make.shared.targets
 
-- 
2.4.3

From 8373e05051d23ae15f3e455a879786e9f8f5d6f9 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Wed, 3 Dec 2014 14:23:31 +0300
Subject: [PATCH] MSVC compatibility fix.

---
 src/common/classes/locks.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/classes/locks.h b/src/common/classes/locks.h
index d2ddb7d..3586592 100644
--- a/src/common/classes/locks.h
+++ b/src/common/classes/locks.h
@@ -64,7 +64,7 @@ public:
 	}
 
 private:
-	typedef BOOL WINAPI (*tTryEnterCriticalSection)(LPCRITICAL_SECTION lpCriticalSection);
+	typedef BOOL (WINAPI *tTryEnterCriticalSection)(LPCRITICAL_SECTION lpCriticalSection);
 
 	static tTryEnterCriticalSection m_funct;
 };
-- 
2.4.3

