--- julia-1.1.1.orig/base/floatfuncs.jl	2019-05-16 12:10:48.000000000 +0800
+++ julia-1.1.1/base/floatfuncs.jl	2019-07-31 14:24:46.678800200 +0800
@@ -329,5 +329,5 @@
 # This is necessary at least on 32-bit Intel Linux, since fma_llvm may
 # have called glibc, and some broken glibc fma implementations don't
 # properly restore the rounding mode
-Rounding.setrounding_raw(Float32, Rounding.JL_FE_TONEAREST)
-Rounding.setrounding_raw(Float64, Rounding.JL_FE_TONEAREST)
+# Rounding.setrounding_raw(Float32, Rounding.JL_FE_TONEAREST)
+# Rounding.setrounding_raw(Float64, Rounding.JL_FE_TONEAREST)
