--- julia-1.1.1/Makefile	2019-08-07 06:30:31.338129300 +0800
+++ build-x86_64-w64-mingw32/Makefile	2019-08-08 22:45:09.645033500 +0800
@@ -330,7 +330,7 @@
 
 
 install:
-	@for subdir in $(bindir) $(datarootdir)/julia/stdlib/$(VERSDIR) $(docdir) $(man1dir) $(includedir)/julia $(libdir) $(private_libdir) $(sysconfdir); do \
+	@for subdir in $(bindir) $(bindir)/julia-$(JULIA_VERSION) $(datarootdir)/julia/stdlib/$(VERSDIR) $(docdir) $(man1dir) $(includedir)/julia $(libdir) $(private_libdir) $(sysconfdir); do \
 		mkdir -p $(DESTDIR)$$subdir; \
 	done
 
@@ -339,13 +339,16 @@
 	$(INSTALL_M) $(build_bindir)/julia-debug $(DESTDIR)$(bindir)/
 endif
 ifeq ($(OS),WINNT)
-	-$(INSTALL_M) $(filter-out $(build_bindir)/libjulia-debug.dll,$(wildcard $(build_bindir)/*.dll)) $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(build_bindir)/libjulia.dll $(DESTDIR)$(bindir)/
 	-$(INSTALL_M) $(build_libdir)/libjulia.dll.a $(DESTDIR)$(libdir)/
 ifeq ($(BUNDLE_DEBUG_LIBS),1)
 	-$(INSTALL_M) $(build_bindir)/libjulia-debug.dll $(DESTDIR)$(bindir)/
 	-$(INSTALL_M) $(build_libdir)/libjulia-debug.dll.a $(DESTDIR)$(libdir)/
 endif
-	-$(INSTALL_M) $(build_bindir)/libopenlibm.dll.a $(DESTDIR)$(libdir)/
+	-$(INSTALL_M) $(build_bindir)/libccalltest.dll $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(build_bindir)/libllvmcalltest.dll $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(filter-out $(foreach suffix,$(JL_TARGETS) ccalltest llvmcalltest,$(build_bindir)/lib$(suffix).dll),$(wildcard $(build_bindir)/*.dll)) $(DESTDIR)$(bindir)/julia-$(JULIA_VERSION)/
+	-$(INSTALL_M) $(build_bindir)/libopenlibm.dll.a $(DESTDIR)$(private_libdir)/
 else
 
 # Copy over .dSYM directories directly for Darwin
