--- a/Makefile
+++ b/Makefile
@@ -275,7 +275,7 @@ endif
 
 
 install:
-	@for subdir in $(bindir) $(datarootdir)/julia/stdlib/$(VERSDIR) $(docdir) $(man1dir) $(includedir)/julia $(libdir) $(private_libdir) $(sysconfdir) $(libexecdir); do \
+	@for subdir in $(bindir) $(bindir)/julia-$(JULIA_VERSION) $(datarootdir)/julia/stdlib/$(VERSDIR) $(docdir) $(man1dir) $(includedir)/julia $(libdir) $(private_libdir) $(sysconfdir) $(libexecdir); do \
 		mkdir -p $(DESTDIR)$$subdir; \
 	done
 
@@ -284,7 +284,7 @@ ifeq ($(BUNDLE_DEBUG_LIBS),1)
 	$(INSTALL_M) $(build_bindir)/julia-debug $(DESTDIR)$(bindir)/
 endif
 ifeq ($(OS),WINNT)
-	-$(INSTALL_M) $(filter-out $(build_bindir)/libjulia-debug.dll,$(wildcard $(build_bindir)/*.dll)) $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(build_bindir)/libjulia.dll $(DESTDIR)$(bindir)/
 	-$(INSTALL_M) $(build_libdir)/libjulia.dll.a $(DESTDIR)$(libdir)/
 
 	# We have a single exception; we want 7z.dll to live in libexec, not bin, so that 7z.exe can find it.
@@ -293,7 +293,11 @@ ifeq ($(BUNDLE_DEBUG_LIBS),1)
 	-$(INSTALL_M) $(build_bindir)/libjulia-debug.dll $(DESTDIR)$(bindir)/
 	-$(INSTALL_M) $(build_libdir)/libjulia-debug.dll.a $(DESTDIR)$(libdir)/
 endif
-	-$(INSTALL_M) $(build_bindir)/libopenlibm.dll.a $(DESTDIR)$(libdir)/
+	-$(INSTALL_M) $(build_bindir)/libccalltest.dll $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(build_bindir)/libllvmcalltest.dll $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(build_bindir)/LLVM.dll $(DESTDIR)$(bindir)/
+	-$(INSTALL_M) $(filter-out $(foreach suffix,$(JL_TARGETS) LLVM ccalltest llvmcalltest,$(build_bindir)/lib$(suffix).dll),$(wildcard $(build_bindir)/*.dll)) $(DESTDIR)$(bindir)/julia-$(JULIA_VERSION)/
+	-$(INSTALL_M) $(build_bindir)/libopenlibm.dll.a $(DESTDIR)$(private_libdir)/
 else
 
 # Copy over .dSYM directories directly for Darwin
-- 
