--- julia-1.1.1/deps/llvm.mk	2019-08-06 20:36:45.478162700 +0800
+++ build-x86_64-w64-mingw32/deps/llvm.mk	2019-08-06 20:36:37.322162700 +0800
@@ -469,8 +469,10 @@
     cp -r $$(LLVM_SRC_DIR)/utils/lit $2$$(build_depsbindir)/ && \
     $$(CMAKE) -DCMAKE_INSTALL_PREFIX="$2$$(build_prefix)" -P cmake_install.cmake
 ifeq ($(OS), WINNT)
+ifeq ($(USE_LLVM_SHLIB),1)
 LLVM_INSTALL += && cp $2$$(build_shlibdir)/LLVM.dll $2$$(build_depsbindir)
 endif
+endif
 ifeq ($(OS),Darwin)
 # https://github.com/JuliaLang/julia/issues/29981
 LLVM_INSTALL += && ln -s libLLVM.dylib $2$$(build_shlibdir)/libLLVM-$$(LLVM_VER_SHORT).dylib
