From 09fde5834422d113aa814047db73e957e507591c Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Mon, 11 May 2020 07:10:13 -0400
Subject: Add $(SRCCACHE)/$(LIBGIT2_SRC_DIR)/source-extracted target

---
 deps/libgit2.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/deps/libgit2.mk b/deps/libgit2.mk
index 845cfba273..3b8b8c8b0d 100644
--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -44,6 +44,13 @@ endif
 
 LIBGIT2_SRC_PATH := $(SRCCACHE)/$(LIBGIT2_SRC_DIR)
 
+$(SRCCACHE)/$(LIBGIT2_SRC_DIR)/source-extracted: $(SRCCACHE)/$(LIBGIT2_SRC_DIR).tar.gz
+	$(JLCHECKSUM) $<
+	mkdir -p $(dir $@) && \
+	$(TAR) -C $(dir $@) --strip-components 1 --exclude tests/resources/testrepo-worktree/link_to_new.txt -xf $<
+	touch -c $(SRCCACHE)/$(LIBGIT2_SRC_DIR)/CMakeLists.txt # old target
+	echo 1 > $@
+
 $(LIBGIT2_SRC_PATH)/libgit2-mbedtls.patch-applied: $(SRCCACHE)/$(LIBGIT2_SRC_DIR)/source-extracted
 	cd $(LIBGIT2_SRC_PATH) && \
 		patch -p1 -f < $(SRCDIR)/patches/libgit2-mbedtls.patch
-- 
