From e21c2a38259cd967bd29a889d10d8ddf7c2a50fe Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 25 Feb 2023 23:11:57 -0500
Subject: deps/tools: [MSYS2] Set CMAKE_C_COMPILER using "cygpath -m"

---
 deps/tools/common.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/deps/tools/common.mk b/deps/tools/common.mk
index c19886114c..9c204ff629 100644
--- a/deps/tools/common.mk
+++ b/deps/tools/common.mk
@@ -27,8 +27,12 @@ endif
 ifneq ($(VERBOSE), 0)
 CMAKE_COMMON += -DCMAKE_VERBOSE_MAKEFILE=ON
 endif
+ifeq ($(BUILD_OS),WINNT)
+CMAKE_COMMON += -DCMAKE_C_COMPILER="$$(cygpath -m $$(which $(CC_BASE)))"
+else
 # The call to which here is to work around https://cmake.org/Bug/view.php?id=14366
 CMAKE_COMMON += -DCMAKE_C_COMPILER="$$(which $(CC_BASE))"
+endif
 ifneq ($(strip $(CMAKE_CC_ARG)),)
 CMAKE_COMMON += -DCMAKE_C_COMPILER_ARG1="$(CMAKE_CC_ARG) $(SANITIZE_OPTS)"
 endif
-- 
