From f38d9d0fbf32cd8031fd6e06ed92e70d11ed96a7 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 25 Feb 2023 11:34:49 -0500
Subject: deps/libgit2: Add LIBSSH2 paths

---
 deps/libgit2.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/deps/libgit2.mk b/deps/libgit2.mk
index 9bd7bd555d..014fdc0108 100644
--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -16,6 +16,11 @@ endif
 LIBGIT2_OPTS := $(CMAKE_COMMON) -DCMAKE_BUILD_TYPE=Release -DUSE_THREADS=ON -DUSE_BUNDLED_ZLIB=ON -DUSE_SSH=ON -DBUILD_CLI=OFF
 ifeq ($(OS),WINNT)
 LIBGIT2_OPTS += -DWIN32=ON -DMINGW=ON
+ifeq ($(USE_SYSTEM_LIBSSH2), 0)
+LIBGIT2_OPTS += -DLIBSSH2_LIBRARIES=libssh2.dll
+LIBGIT2_OPTS += -DLIBSSH2_LIBRARY_DIRS=$(build_prefix)/lib
+LIBGIT2_OPTS += -DLIBSSH2_INCLUDE_DIRS=$(build_prefix)/include
+endif # USE_SYSTEM_LIBSSH2=0
 ifneq ($(ARCH),x86_64)
 ifneq ($(USECLANG),1)
 LIBGIT2_OPTS += -DCMAKE_C_FLAGS="-mincoming-stack-boundary=2"
-- 
2.39.1.windows.1

