From f7809870d97f363b482b6ef85b651d19fc565093 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sun, 28 May 2023 07:32:16 -0400
Subject: julia: Skip 2nd build in install target

---
 Makefile | 2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index ef0ade09e2..0f9f4f75d9 100644
--- a/Makefile
+++ b/Makefile
@@ -236,6 +236,6 @@ define stringreplace
 
 install: install-deps
-	@$(MAKE) $(QUIET_MAKE) $(JULIA_BUILD_MODE)
+	#@$(MAKE) $(QUIET_MAKE) $(JULIA_BUILD_MODE)
 	@for subdir in $(bindir) $(datarootdir)/julia/stdlib/$(VERSDIR) $(docdir) $(man1dir) $(includedir)/julia $(libdir) $(private_libdir) $(sysconfdir) $(private_libexecdir); do \
 		mkdir -p $(DESTDIR)$$subdir; \
 	done
-- 
