From 3cf3f6a303b1218c49e1070306208a0f478484f5 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 21 Sep 2021 20:52:42 +0200
Subject: [PATCH 016/N] sysconfig: MINGW build extensions with GCC

---
 Lib/sysconfig/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/sysconfig/__init__.py b/Lib/sysconfig/__init__.py
index 510c7b9..93abee5 100644
--- a/Lib/sysconfig/__init__.py
+++ b/Lib/sysconfig/__init__.py
@@ -609,6 +609,8 @@ def get_platform():
 
     """
     if os.name == 'nt':
+        if 'GCC' in sys.version:
+            return 'mingw'
         if 'amd64' in sys.version.lower():
             return 'win-amd64'
         if '(arm)' in sys.version.lower():
