From bcc2f8580a9d1c95a91878d1e1a011ca484a3972 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 2 Sep 2025 16:07:14 +0300
Subject: [PATCH 085/N] tests: Fix test for library name

---
 Lib/test/test_sysconfig.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/Lib/test/test_sysconfig.py b/Lib/test/test_sysconfig.py
index 691b84f..ddc138c 100644
--- a/Lib/test/test_sysconfig.py
+++ b/Lib/test/test_sysconfig.py
@@ -466,9 +466,13 @@ class TestSysConfig(unittest.TestCase):
         ldlibrary = sysconfig.get_config_var('LDLIBRARY')
         major, minor = sys.version_info[:2]
         if sys.platform == 'win32':
-            self.assertTrue(library.startswith(f'python{major}{minor}'))
-            self.assertTrue(library.endswith('.dll'))
-            self.assertEqual(library, ldlibrary)
+            if sys._is_mingw:
+                self.assertTrue(library.startswith(f'libpython{major}.{minor}'))
+                self.assertTrue(ldlibrary.endswith('.dll.a'))
+            else:
+                self.assertTrue(library.startswith(f'python{major}{minor}'))
+                self.assertTrue(library.endswith('.dll'))
+                self.assertEqual(library, ldlibrary)
         elif is_apple_mobile:
             framework = sysconfig.get_config_var('PYTHONFRAMEWORK')
             self.assertEqual(ldlibrary, f"{framework}.framework/{framework}")
