From c5eb13557ff35c6628c00f6ce5eaa181fb9a1af4 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Mon, 1 Jan 2024 16:43:41 +0530
Subject: [PATCH 107/N] Enable curses library on MINGW

---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 1d1b8f9..5f59ed1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7186,6 +7186,14 @@ dnl have_panel=[no, yes]
 have_curses=no
 have_panel=no
 
+case "$host_os" in
+	mingw*)
+	# On Mingw, include directory is different, so just assume
+  # it's there.
+	ac_cv_header_ncurses_h=yes
+	;;
+esac
+
 dnl PY_CHECK_CURSES(LIBCURSES, LIBPANEL)
 dnl Sets 'have_curses' and 'have_panel'.
 dnl For the PKG_CHECK_MODULES() calls, we can safely reuse the first variable
