From 52648cdd9a59e395fcd017255ecdb752d28fecd2 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Thu, 28 Aug 2025 12:13:51 +0300
Subject: [PATCH 121/N] Disable __try/__except for MINGW

---
 Modules/mmapmodule.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Modules/mmapmodule.c b/Modules/mmapmodule.c
index 4cc8880..220d2a8 100644
--- a/Modules/mmapmodule.c
+++ b/Modules/mmapmodule.c
@@ -289,7 +289,7 @@ filter_page_exception_method(mmap_object *self, EXCEPTION_POINTERS *ptrs,
 }
 #endif
 
-#if defined(MS_WINDOWS) && !defined(DONT_USE_SEH)
+#if defined(_MSC_VER) && !defined(DONT_USE_SEH)
 #define HANDLE_INVALID_MEM(sourcecode)                                     \
 do {                                                                       \
     EXCEPTION_RECORD record;                                               \
@@ -317,7 +317,7 @@ do {                                                                       \
 } while (0)
 #endif
 
-#if defined(MS_WINDOWS) && !defined(DONT_USE_SEH)
+#if defined(_MSC_VER) && !defined(DONT_USE_SEH)
 #define HANDLE_INVALID_MEM_METHOD(self, sourcecode)                           \
 do {                                                                          \
     EXCEPTION_RECORD record;                                                  \
