From b04c44abb0c1660973faa810d8591ea88d205c47 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 2 Sep 2025 11:48:19 +0300
Subject: [PATCH 137/N] Smoketests: Try fix build extension

---
 mingw_smoketests.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 112a931..7acf276 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -255,7 +255,7 @@ class Tests(unittest.TestCase):
             if not sysconfig.is_python_build():
                 op = subprocess.check_output(
                     [
-                        os.path.join(tmp, "bin", "python.exe"),
+                        os.path.join(tmp, "bin", 'python' + abiflags + '.exe'),
                         "-c",
                         "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))"
                     ],
@@ -294,6 +294,10 @@ class Tests(unittest.TestCase):
         import venv
         from pathlib import Path
 
+        if sysconfig.is_python_build():
+            pydir = os.path.dirname(sys._base_executable)
+            os.environ["LDFLAGS"] = "-L " + pydir
+
         with tempfile.TemporaryDirectory() as tmppro:
             builder = venv.EnvBuilder(with_pip=True)
             builder.create(tmppro)
