From 6fe520af6ec4a7594703647a9cd630de9e20fadf Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 2 Sep 2025 12:35:22 +0300
Subject: [PATCH 138/N] Try fix LDFLAGS passing via environment

---
 mingw_smoketests.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 7acf276..9a4f39e 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -295,7 +295,7 @@ class Tests(unittest.TestCase):
         from pathlib import Path
 
         if sysconfig.is_python_build():
-            pydir = os.path.dirname(sys._base_executable)
+            pydir = os.path.dirname(sys._base_executable).replace('\\', '/')
             os.environ["LDFLAGS"] = "-L " + pydir
 
         with tempfile.TemporaryDirectory() as tmppro:
