From 42aceea84071befb56910bd5ac720929beb3cb78 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 2 Sep 2025 22:09:04 +0300
Subject: [PATCH 142/N] Makefile: Remove duplicate object from link

---
 Makefile.pre.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 43c5d3f..ffb26f8 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1030,7 +1030,7 @@ $(PYTHONFRAMEWORKDIR)/$(PYTHONFRAMEWORK): \
 $(DLLLIBRARY) libpython$(LDVERSION).dll.a: $(LIBRARY_OBJS) python_nt.o
 	if test -n "$(DLLLIBRARY)"; then \
 		$(LDSHARED) -Wl,--out-implib=$@ -o $(DLLLIBRARY) $^ \
-			$(LIBS) $(LDFLAGS_NODIST) $(MODLIBS) $(SYSLIBS) python_nt.o; \
+			$(LIBS) $(LDFLAGS_NODIST) $(MODLIBS) $(SYSLIBS); \
 	else true; \
 	fi
 
