From 0fcaf960a0f42d3bfe5bd9930b68cd054bd27176 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 2 Sep 2025 22:50:10 +0300
Subject: [PATCH 146/N] smoketests: Adjust SOABI change

---
 mingw_smoketests.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 9a4f39e..2eb7ce8 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -226,7 +226,7 @@ class Tests(unittest.TestCase):
         self.assertEqual(sys.implementation.name, "cpython")
         self.assertEqual(sys.platform, "win32")
         self.assertTrue(sysconfig.get_platform().startswith("mingw"))
-        self.assertTrue(sysconfig.get_config_var('SOABI').startswith("cpython-"))
+        self.assertTrue(sysconfig.get_config_var('SOABI').startswith("cp" + f'{sys.version_info.major}{sys.version_info.minor}'))
         ext_suffix = sysconfig.get_config_var('EXT_SUFFIX')
         abiflags = sysconfig.get_config_var('ABIFLAGS')
         self.assertTrue(ext_suffix.endswith(".pyd"))
