From 5f29ad34764d5284bc296cb9b1b4d297d4cf8177 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 3 Sep 2025 09:16:51 +0300
Subject: [PATCH 151/N] ntpath: Try fix concatenate UNC paths

---
 Lib/ntpath.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/Lib/ntpath.py b/Lib/ntpath.py
index d3423ac..7d079bc 100644
--- a/Lib/ntpath.py
+++ b/Lib/ntpath.py
@@ -62,6 +62,12 @@ def _get_colon(path):
     else:
         return ':'
 
+def _get_colon_seps(path):
+    if isinstance(path, bytes):
+        return b':'+bsep+baltsep
+    else:
+        return ':'+sep+altsep
+
 def _get_unc_prefix(path):
     if isinstance(path, bytes):
         return b'\\\\?\\UNC\\'
@@ -125,7 +131,7 @@ def join(path, *paths):
     path = os.fspath(path)
     sep = _get_sep(path)
     seps = _get_bothseps(path)
-    colon_seps = _get_colon(path)
+    colon_seps = _get_colon_seps(path)
     try:
         result_drive, result_root, result_path = splitroot(path)
         for p in paths:
