From 24e3caad4e284b6697027f618b4ad2ea57cb7e31 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 3 Sep 2025 09:33:12 +0300
Subject: [PATCH 152/N] Revert fix warning for semaphore

---
 Modules/_multiprocessing/semaphore.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Modules/_multiprocessing/semaphore.c b/Modules/_multiprocessing/semaphore.c
index ae1ecc9..dc521b8 100644
--- a/Modules/_multiprocessing/semaphore.c
+++ b/Modules/_multiprocessing/semaphore.c
@@ -631,14 +631,14 @@ _multiprocessing_SemLock__get_value_impl(SemLockObject *self)
     PyErr_SetNone(PyExc_NotImplementedError);
     return NULL;
 #else
-    long sval;
+    int sval;
     if (SEM_GETVALUE(self->handle, &sval) < 0)
         return _PyMp_SetError(NULL, MP_STANDARD_ERROR);
     /* some posix implementations use negative numbers to indicate
        the number of waiting threads */
     if (sval < 0)
         sval = 0;
-    return PyLong_FromLong(sval);
+    return PyLong_FromLong((long)sval);
 #endif
 }
 
