From 19cd310d14e8843ddd470dd2b875670129d5dd13 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 3 Sep 2025 15:08:30 +0300
Subject: [PATCH 154/N] Fix format identifier

---
 Modules/_ctypes/_ctypes.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/_ctypes/_ctypes.c b/Modules/_ctypes/_ctypes.c
index cb306cb..632587d 100644
--- a/Modules/_ctypes/_ctypes.c
+++ b/Modules/_ctypes/_ctypes.c
@@ -5008,7 +5008,7 @@ PyCArrayType_from_ctype(ctypes_state *st, PyObject *itemtype, Py_ssize_t length)
         return NULL;
     }
 #ifdef MS_WIN64
-    sprintf(name, "%.200s_Array_%Illd",
+    sprintf(name, "%.200s_Array_%lld",
         ((PyTypeObject *)itemtype)->tp_name, length);
 #else
     sprintf(name, "%.200s_Array_%ld",
