From fe5cac185510c8cb58f0d3464ab3de800ad1dec9 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 3 Sep 2025 15:44:47 +0300
Subject: [PATCH 156/N] Pass profile LDFLAGS for C++ modules

---
 Makefile.pre.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 62668fd..084d9fd 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -181,7 +181,7 @@ EXT_SUFFIX=	@EXT_SUFFIX@
 PYD_PLATFORM_TAG = @PYD_PLATFORM_TAG@
 LDSHARED=	@LDSHARED@ $(PY_LDFLAGS)
 BLDSHARED=	@BLDSHARED@ $(PY_CORE_LDFLAGS)
-LDCXXSHARED=	@LDCXXSHARED@ $(PY_LDFLAGS)
+LDCXXSHARED=	@LDCXXSHARED@ $(PY_CORE_LDFLAGS)
 DESTSHARED=	$(BINLIBDEST)/lib-dynload
 
 # List of exported symbols for AIX
