From 49d0e50adbe1e0991a87f3fa3ce993018f2366df Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 3 Sep 2025 22:21:39 +0300
Subject: [PATCH 159/N] Cleanup USE_WIN32_MODULE in configure

---
 configure.ac | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9e20605..d3c9a8f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4852,13 +4852,6 @@ case $host in
     CFLAGS_NODIST="$CFLAGS_NODIST -D_WIN32_WINNT=0x0602";;
 esac
 
-# Determine if windows modules should be used.
-AC_SUBST(USE_WIN32_MODULE)
-USE_WIN32_MODULE='#'
-case $host in
-  *-*-mingw*) USE_WIN32_MODULE=;;
-esac
-
 # Templates for things AC_DEFINEd more than once.
 # For a single AC_DEFINE, no template is needed.
 AH_TEMPLATE([_REENTRANT],
