From be4f094bd348151a22203e1faf5f39b4f3f3d20a Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 4 Sep 2025 07:55:42 +0200
Subject: [PATCH 161/N] Fix include naming for cross build

---
 Modules/posixmodule.c | 2 +-
 PC/_wmimodule.cpp     | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index 00eadab..d81f59d 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -9337,7 +9337,7 @@ os_setpgrp_impl(PyObject *module)
 
 #ifdef MS_WINDOWS
 #include <winternl.h>
-#include <ProcessSnapshot.h>
+#include <processsnapshot.h>
 
 // The structure definition in winternl.h may be incomplete.
 // This structure is the full version from the MSDN documentation.
diff --git a/PC/_wmimodule.cpp b/PC/_wmimodule.cpp
index e05037d..6cf9856 100644
--- a/PC/_wmimodule.cpp
+++ b/PC/_wmimodule.cpp
@@ -14,9 +14,9 @@
 #endif
 
 #define _WIN32_DCOM
-#include <Windows.h>
+#include <windows.h>
 #include <comdef.h>
-#include <Wbemidl.h>
+#include <wbemidl.h>
 #include <propvarutil.h>
 
 #include <Python.h>
