From 14fbd6a4ef91affcb448e3467129e70e46875ab6 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 5 Sep 2025 16:14:26 +0300
Subject: [PATCH 167/N] _boostrap_external: Use new flag use_alt_sep

---
 Lib/importlib/_bootstrap_external.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/importlib/_bootstrap_external.py b/Lib/importlib/_bootstrap_external.py
index 520025e..cbe7330 100644
--- a/Lib/importlib/_bootstrap_external.py
+++ b/Lib/importlib/_bootstrap_external.py
@@ -43,7 +43,7 @@ if _MS_WINDOWS:
 else:
     path_separators = ['/']
 
-if _os.environ.get('MSYSTEM', ''):
+if sys.use_alt_sep:
     path_separators = path_separators[::-1]
 
 # Assumption made in _path_join()
