From f12c612b07fc35e693d704b28011e6b6934bbb28 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 13 Sep 2025 17:59:54 +0200
Subject: [PATCH 181/N] smoketests: add test for soabi fix

---
 mingw_smoketests.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 18db3b5..3e6533a 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -221,9 +221,11 @@ class Tests(unittest.TestCase):
         self.assertEqual(sys.implementation.name, "cpython")
         self.assertEqual(sys.platform, "win32")
         self.assertTrue(sysconfig.get_platform().startswith("mingw"))
-        self.assertTrue(sysconfig.get_config_var('SOABI').startswith("cp" + f'{sys.version_info.major}{sys.version_info.minor}'))
+        soabi = sysconfig.get_config_var('SOABI')
+        self.assertTrue(soabi.startswith("cp" + f'{sys.version_info.major}{sys.version_info.minor}'))
         ext_suffix = sysconfig.get_config_var('EXT_SUFFIX')
         abiflags = sysconfig.get_config_var('ABIFLAGS')
+        self.assertTrue("." + soabi + "." in ext_suffix)
         self.assertTrue(ext_suffix.endswith(".pyd"))
         self.assertTrue("mingw" in ext_suffix)
         self.assertEqual(sysconfig.get_config_var('SHLIB_SUFFIX'), ".pyd")
