diff -urN baloo-5.24.0.orig/src/CMakeLists.txt baloo-5.24.0/src/CMakeLists.txt
--- baloo-5.24.0.orig/src/CMakeLists.txt	2016-07-03 01:57:07.000000000 +0200
+++ baloo-5.24.0/src/CMakeLists.txt	2016-08-01 16:14:18.293240700 +0200
@@ -2,10 +2,10 @@
 add_subdirectory(engine)
 add_subdirectory(codecs)
 add_subdirectory(qml)
+add_subdirectory(dbus)
 
 if (BUILD_KINOTIFY)
     add_subdirectory(file)
     add_subdirectory(kioslaves)
     add_subdirectory(tools)
-    add_subdirectory(dbus)
 endif()
