diff -urN gputils-1.5.2/libgputils/gpsystem.c gputils-1.5.2-mod/libgputils/gpsystem.c
--- a/libgputils/gpsystem.c	2017-05-07 02:45:29.000000000 -0700
+++ b/libgputils/gpsystem.c	2025-11-15 13:29:50.762645600 -0800
@@ -52,7 +52,6 @@
   gp_lkr_path    = getenv("GPUTILS_LKR_PATH");
   gp_lib_path    = getenv("GPUTILS_LIB_PATH");
 
-  #ifndef HAVE_DOS_BASED_FILE_SYSTEM
     if (gp_header_path == NULL) {
       gp_header_path = GP_Strdup(HEADER_PATH);
     }
@@ -62,7 +61,6 @@
     if (gp_lib_path == NULL) {
       gp_lib_path = GP_Strdup(LIB_PATH);
     }
-  #endif
 
 #else
   gp_header_path = NULL;
