--- a/cmake/Findyaml-cpp.cmake
+++ b/cmake/Findyaml-cpp.cmake
@@ -8,6 +8,9 @@
 
 find_package(yaml-cpp 0.5 QUIET NO_MODULE)
 if (yaml-cpp_FOUND)
+  if(yaml-cpp_VERSION VERSION_GREATER_EQUAL "0.8.0")
+    add_library(yaml-cpp ALIAS yaml-cpp::yaml-cpp)
+  else()
   # target doesn't set include directory for some reason
   get_filename_component(yaml_cpp_include_dir "${YAML_CPP_INCLUDE_DIR}" ABSOLUTE)
   if (NOT EXISTS yaml_cpp_include_dir)
@@ -16,6 +19,7 @@
     find_path(yaml_cpp_include_dir yaml-cpp/yaml.h)
   endif()
   set_target_properties(yaml-cpp PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${yaml_cpp_include_dir}")
+  endif()
 else()
   if(TARGET yaml-cpp)
     return()
