--- a/src/plugins/clangformat/CMakeLists.txt
+++ b/src/plugins/clangformat/CMakeLists.txt
@@ -1,6 +1,6 @@
 add_qtc_plugin(ClangFormat
   CONDITION TARGET ${CLANG_FORMAT_LIB} AND LLVM_PACKAGE_VERSION VERSION_GREATER_EQUAL 14.0.0 AND (QTC_CLANG_BUILDMODE_MATCH OR CLANGTOOLING_LINK_CLANG_DYLIB)
-  DEPENDS Utils Qt::Widgets ${CLANG_FORMAT_LIB}
+  DEPENDS Utils Qt::Widgets ${CLANG_FORMAT_LIB} LLVM
   PLUGIN_DEPENDS Core TextEditor CppEditor ProjectExplorer
   SOURCES
     clangformatbaseindenter.cpp clangformatbaseindenter.h
