diff -Naur qt-creator-opensource-src-15.0.0-orig/src/libs/solutions/zip/CMakeLists.txt qt-creator-opensource-src-15.0.0/src/libs/solutions/zip/CMakeLists.txt
--- qt-creator-opensource-src-15.0.0-orig/src/libs/solutions/zip/CMakeLists.txt	2024-11-27 13:00:22.000000000 +0300
+++ qt-creator-opensource-src-15.0.0/src/libs/solutions/zip/CMakeLists.txt	2024-12-11 13:23:13.599640700 +0300
@@ -1,4 +1,4 @@
-add_qtc_library(Zip
+add_qtc_library(QZip
   DEPENDS Qt::Core
   DEFINES ZIP_LIB
   SOURCES
@@ -10,7 +10,7 @@
       ${CMAKE_CURRENT_LIST_DIR}/../../3rdparty/zlib/src
 )
 
-extend_qtc_library(Zip
+extend_qtc_library(QZip
   SOURCES_PREFIX ${CMAKE_CURRENT_LIST_DIR}/../../3rdparty/zlib/src
   SKIP_AUTOMOC
   SOURCES
@@ -43,5 +43,5 @@
 )
 
 if(MSVC)
-  target_compile_options(Zip PUBLIC /wd4996)
+  target_compile_options(QZip PUBLIC /wd4996)
 endif()
diff -Naur qt-creator-opensource-src-15.0.0-orig/src/plugins/qmldesigner/CMakeLists.txt qt-creator-opensource-src-15.0.0/src/plugins/qmldesigner/CMakeLists.txt
--- qt-creator-opensource-src-15.0.0-orig/src/plugins/qmldesigner/CMakeLists.txt	2024-11-27 13:00:24.000000000 +0300
+++ qt-creator-opensource-src-15.0.0/src/plugins/qmldesigner/CMakeLists.txt	2024-12-11 13:23:00.532821600 +0300
@@ -50,7 +50,7 @@
   PLUGIN_MANUAL_DEPENDS LicenseChecker ${IDE_VERSION} optional
   DEPENDS
     QmlJS LanguageUtils QmlEditorWidgets AdvancedDockingSystem
-    Qt::QuickWidgets Qt::CorePrivate Qt::Xml Qt::Svg Sqlite Zip Qt::GuiPrivate
+    Qt::QuickWidgets Qt::CorePrivate Qt::Xml Qt::Svg Sqlite QZip Qt::GuiPrivate
   PUBLIC_DEPENDS
     QmlDesignerUtils QmlPuppetCommunication QmlDesignerCore
   DEFINES
