--- a/src/plugins/ios/Ios.json.in
+++ b/src/plugins/ios/Ios.json.in
@@ -2,7 +2,6 @@
     \"Name\" : \"Ios\",
     \"Version\" : \"$$QTCREATOR_VERSION\",
     \"CompatVersion\" : \"$$QTCREATOR_COMPAT_VERSION\",
-    \"Platform\" : \"OS X.*\",
     \"Vendor\" : \"The Qt Company Ltd\",
     \"Copyright\" : \"(C) $$QTCREATOR_COPYRIGHT_YEAR The Qt Company Ltd\",
     \"License\" : [ \"Commercial Usage\",
