--- a/cmake/QtCreatorAPIInternal.cmake
+++ b/cmake/QtCreatorAPIInternal.cmake
@@ -74,9 +74,9 @@
   set(_IDE_DATA_PATH "share/qtcreator")
   set(_IDE_DOC_PATH "share/doc/qtcreator")
   set(_IDE_BIN_PATH "bin")
-  set(_IDE_LIBRARY_ARCHIVE_PATH "${_IDE_BIN_PATH}")
+  set(_IDE_LIBRARY_ARCHIVE_PATH "${_IDE_LIBRARY_PATH}")
 
-  set(_IDE_DEVEL_LIBRARY_ARCHIVE_INSTALL_PATH "${_IDE_BIN_PATH}")
+  set(_IDE_DEVEL_LIBRARY_ARCHIVE_INSTALL_PATH "${_IDE_LIBRARY_PATH}")
   set(_IDE_DEVEL_LIBRARY_INSTALL_PATH "${_IDE_LIBRARY_PATH}")
   set(_IDE_DEVEL_DATA_INSTALL_PATH "share/qtcreator")
   set(_IDE_DEVEL_HEADER_INSTALL_PATH "include/qtcreator")
