diff -Naur qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h
--- qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h	2016-01-25 12:44:53.642390800 +0200
+++ qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/projectpartsdonotexistexception.h	2016-01-25 12:45:46.577257300 +0200
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class ProjectPartDoNotExistException : public std::exception
diff -Naur qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitalreadyexistsexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitalreadyexistsexception.h
--- qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitalreadyexistsexception.h	2016-01-25 12:47:11.904653700 +0200
+++ qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitalreadyexistsexception.h	2016-01-25 12:47:16.069025400 +0200
@@ -32,7 +32,11 @@
 #define CLANGBACKEND_TRANSLATIONUNITALREADYEXISTS_H
 
 #include <filecontainer.h>
-
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
 namespace ClangBackEnd {
 
 class TranslationUnitAlreadyExistsException : public std::exception
diff -Naur qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h
--- qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h	2016-01-25 12:44:53.652151800 +0200
+++ qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitdoesnotexistexception.h	2016-01-25 12:45:46.578758600 +0200
@@ -33,6 +33,12 @@
 
 #include <filecontainer.h>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitDoesNotExistException : public std::exception
diff -Naur qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h
--- qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h	2016-01-25 12:44:53.653343300 +0200
+++ qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitfilenotexitexception.h	2016-01-25 12:45:46.579759500 +0200
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitFileNotExitsException : public std::exception
diff -Naur qt-creator-opensource-src-4.0.0-orig/src/tools/clangbackend/ipcsource/translationunitreparseerrorexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitreparseerrorexception.h
--- qt-creator-opensource-src-4.0.0-orig/src/tools/clangbackend/ipcsource/translationunitreparseerrorexception.h	2016-01-25 12:44:53.653343300 +0200
+++ qt-creator-opensource-src-4.0.0/src/tools/clangbackend/ipcsource/translationunitreparseerrorexception.h	2016-01-25 12:45:46.579759500 +0200
@@ -30,6 +30,12 @@
 #include <clang-c/Index.h>
 
 #include <exception>
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 
 namespace ClangBackEnd {
 
diff -Naur qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitisnullexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitisnullexception.h
--- qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitisnullexception.h	2016-01-25 12:44:53.654844600 +0200
+++ qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitisnullexception.h	2016-01-25 12:45:46.580760700 +0200
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitIsNullException : public std::exception
diff -Naur qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h
--- qt-creator-opensource-src-3.6.0-orig/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h	2016-01-25 12:44:53.656345900 +0200
+++ qt-creator-opensource-src-3.6.0/src/tools/clangbackend/ipcsource/translationunitparseerrorexception.h	2016-01-25 12:45:46.582271300 +0200
@@ -35,6 +35,12 @@
 
 #include <exception>
 
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GNUC_PREREQ)
+#  if defined(__MINGW64_VERSION_MAJOR)
+#    define __GNUC_PREREQ __MINGW_GNUC_PREREQ
+#  endif
+#endif
+
 namespace ClangBackEnd {
 
 class TranslationUnitParseErrorException : public std::exception
