diff -rupN qt-creator-opensource-src-4.10.0/src/plugins/clangformat/clangformatutils.cpp qt-creator-opensource-src-4.10.0-new/src/plugins/clangformat/clangformatutils.cpp
--- qt-creator-opensource-src-4.10.0/src/plugins/clangformat/clangformatutils.cpp	2019-09-03 16:36:38.000000000 +0200
+++ qt-creator-opensource-src-4.10.0-new/src/plugins/clangformat/clangformatutils.cpp	2019-09-25 13:33:27.230835047 +0200
@@ -60,7 +60,7 @@ static clang::format::FormatStyle qtcSty
     style.AllowShortBlocksOnASingleLine = false;
     style.AllowShortCaseLabelsOnASingleLine = false;
     style.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
-    style.AllowShortIfStatementsOnASingleLine = false;
+    style.AllowShortIfStatementsOnASingleLine = FormatStyle::SIS_Never;
     style.AllowShortLoopsOnASingleLine = false;
     style.AlwaysBreakAfterReturnType = FormatStyle::RTBS_None;
     style.AlwaysBreakBeforeMultilineStrings = false;
