diff -Naur qt-creator-opensource-src-5.0.1-orig/cmake/QtCreatorAPIInternal.cmake qt-creator-opensource-src-5.0.1/cmake/QtCreatorAPIInternal.cmake
--- qt-creator-opensource-src-5.0.1-orig/cmake/QtCreatorAPIInternal.cmake	2021-09-26 14:01:05.079629000 +0300
+++ qt-creator-opensource-src-5.0.1/cmake/QtCreatorAPIInternal.cmake	2021-09-26 16:43:46.107410400 +0300
@@ -61,7 +61,7 @@
   set(_IDE_DATA_PATH "share/qtcreator")
   set(_IDE_DOC_PATH "share/doc/qtcreator")
   set(_IDE_BIN_PATH "bin")
-  set(_IDE_LIBRARY_ARCHIVE_PATH "${_IDE_BIN_PATH}")
+  set(_IDE_LIBRARY_ARCHIVE_PATH "${_IDE_LIBRARY_PATH}")
 
   set(_IDE_HEADER_INSTALL_PATH "include/qtcreator")
   set(_IDE_CMAKE_INSTALL_PATH "lib/cmake")
