--- doc/doc.pri.orig	2013-12-06 09:21:59.313800000 +0400
+++ doc/doc.pri	2013-12-06 09:22:07.925000000 +0400
@@ -46,7 +46,7 @@
 html_docs_online.commands = $$qdoc($$OUT_PWD/doc/html) $$PWD/qtcreator-online$${COMPAT}.qdocconf
 html_docs_online.depends += $$HELP_DEP_FILES
 
-qch_docs.commands = $$HELPGENERATOR -o \"$$QCH_FILE\" $$QHP_FILE
+qch_docs.commands = $$HELPGENERATOR -o $$QCH_FILE $$QHP_FILE
 qch_docs.depends += html_docs
 
 DEV_QHP_FILE = $$OUT_PWD/doc/html-dev/qtcreator-dev.qhp
@@ -72,7 +72,7 @@
 dev_html_docs_online.commands = $$qdoc($$OUT_PWD/doc/html-dev) $$PWD/api/qtcreator-dev-online$${COMPAT}.qdocconf
 dev_html_docs_online.depends += $$DEV_HELP_DEP_FILES
 
-dev_qch_docs.commands = $$HELPGENERATOR -o \"$$DEV_QCH_FILE\" $$DEV_QHP_FILE
+dev_qch_docs.commands = $$HELPGENERATOR -o $$DEV_QCH_FILE $$DEV_QHP_FILE
 dev_qch_docs.depends += dev_html_docs
 
 !macx {
