--- qt-creator-opensource-src-3.2.0-rc1/src/plugins/qbsprojectmanager/qbsprojectmanager.pro.orig	2014-06-19 16:30:25.000000000 +0100
+++ qt-creator-opensource-src-3.2.0-rc1/src/plugins/qbsprojectmanager/qbsprojectmanager.pro	2014-07-06 11:19:45.267996400 +0100
@@ -12,7 +12,11 @@
     include($${QBS_INSTALL_DIR}/include/qbs/use_installed_qtprofilesetup.pri)
 }
 QBS_INSTALL_DIR_FWD_SLASHES = $$replace(QBS_INSTALL_DIR, \\\\, /)
-DEFINES += QBS_INSTALL_DIR=\\\"$$QBS_INSTALL_DIR_FWD_SLASHES\\\"
+# With this as it was, the Qbs plugin tries to load from C:\msys64r\..
+# since it bakes in the installation prefix. It seems that qbsproject.cpp
+# will do the right thing if QBS_INSTALL_DIR is empty.
+# DEFINES += QBS_INSTALL_DIR=\\\"$$QBS_INSTALL_DIR_FWD_SLASHES\\\"
+DEFINES += QBS_INSTALL_DIR=\\\"../\\\"
 
 DEFINES += \
     QBSPROJECTMANAGER_LIBRARY
--- qt-creator-opensource-src-3.3.0-beta1/src/plugins/qbsprojectmanager/qbsprojectparser.cpp.orig	2014-10-29 08:44:30.000000000 +0000
+++ qt-creator-opensource-src-3.3.0-beta1/src/plugins/qbsprojectmanager/qbsprojectparser.cpp	2014-10-31 20:32:47.988627200 +0000
@@ -158,8 +158,14 @@
 QString QbsProjectParser::resourcesBaseDirectory() const
 {
     const QString qbsInstallDir = QLatin1String(QBS_INSTALL_DIR);
-    if (!qbsInstallDir.isEmpty())
-        return qbsInstallDir;
+    if (!qbsInstallDir.isEmpty()) {
+        if (QDir(qbsInstallDir).isAbsolute()) {
+            return qbsInstallDir;
+        } else {
+            return QDir::cleanPath(QCoreApplication::applicationDirPath()
+                                   + QLatin1String("/") + qbsInstallDir);
+        }
+    }
     return Core::ICore::resourcePath() + QLatin1String("/qbs");
 }

@@ -211,7 +211,7 @@
     const QString qbsLibDirName = QLatin1String(IDE_LIBRARY_BASENAME);
 
     if (!qbsInstallDir.isEmpty())
-        return qbsInstallDir + QLatin1Char('/') + qbsLibDirName;
+        return resourcesBaseDirectory() + QLatin1Char('/') + qbsLibDirName;
     if (HostOsInfo::isMacHost())
         return QDir::cleanPath(QCoreApplication::applicationDirPath()
                                + QLatin1String("/../PlugIns"));
