diff -urN qt-creator-opensource-src-4.3.0.orig/qtcreator.pro qt-creator-opensource-src-4.3.0/qtcreator.pro
--- qt-creator-opensource-src-4.3.0.orig/qtcreator.pro	2017-06-11 12:50:07.594530300 +0100
+++ qt-creator-opensource-src-4.3.0/qtcreator.pro	2017-06-11 13:27:40.638530300 +0100
@@ -89,7 +89,10 @@
     }
     unset(DOC_FILES)
     unset(DOC_TARGET_PREFIX)
-}
+    } else:minQtVersion(5, 0, 0):!qbs_no_dev_install {
+        QBS_CONFIG_ADDITION = qbs_enable_project_file_updates
+        cache(CONFIG, add, QBS_CONFIG_ADDITION)
+    }
 
 contains(QT_ARCH, i386): ARCHITECTURE = x86
 else: ARCHITECTURE = $$QT_ARCH
