--- a/configure
+++ b/configure
@@ -4546,7 +4546,8 @@ $as_echo "yes" >&6; }
 $as_echo "yes" >&6; }
                         LIBMXML="mxml1.dll"
                         DSO="\$(CC)"
-                        DSOFLAGS="$DSOFLAGS -shared -Wl,--out-implib,libmxml1.a,--no-undefined,--enable-runtime-pseudo-reloc"
+                        DSOFLAGS="$DSOFLAGS -shared -Wl,--out-implib,libmxml1.a,--enable-runtime-pseudo-reloc"
+                        PICFLAG=0
                         ;;
 
 		*)
