--- a/rake.d/pch.rb
+++ b/rake.d/pch.rb
@@ -538,11 +538,7 @@
   end
 
   def self.execute(command, options={}, &block)
-    if STDOUT.tty?
-      execute_tty(command, options, &block)
-    else
-      execute_command(command, options, &block)
-    end
+    execute_command(command, options, &block)
   end
 
   def self.execute_command(command, options={})
@@ -557,33 +553,6 @@
     end_execute(lines, options)
   end
 
-  def self.execute_tty(command, options={})
-    lines = []
-    IO.pipe do |read,write|
-      master,slave = PTY.open
-      begin
-        pid = spawn(command, :in => read, [:out, :err] => slave)
-        read.close
-        slave.close
-        begin
-          if block_given?
-            master.each_line { |s| lines.push(s) if s = yield(s) }
-          else
-            master.each_line { |s| lines << s }
-          end
-        rescue
-          ;
-        ensure
-          Process.wait(pid)
-        end
-      ensure
-        master.close unless master.closed?
-        slave.close unless slave.closed?
-      end
-    end
-    end_execute(lines, options)
-  end
-
   def self.end_execute(lines, options)
     if options.fetch(:allow_failure, nil) != true
       exit(1) if $?.exitstatus == nil
--- a/rake.d/requirements.rb
+++ b/rake.d/requirements.rb
@@ -3,7 +3,6 @@
 require "fileutils"
 require "json"
 require "pathname"
-require "pty"
 require "tempfile"
 require "tmpdir"
 require "zlib"
