--- CMakeLists.txt.orig	2021-09-30 13:36:46.000000000 +0200
+++ CMakeLists.txt	2022-07-06 12:05:24.248187832 +0200
@@ -180,7 +180,9 @@
   "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 
 # staniek: don't install
-if(DATA_INSTALL_DIR)
+if(MINGW)
+    set(CMAKE_MODULES_INSTALL_PATH ${CMAKE_INSTALL_PREFIX}/share/cmake/Modules)
+elseif(DATA_INSTALL_DIR)
     set(CMAKE_MODULES_INSTALL_PATH ${DATA_INSTALL_DIR}/cmake/modules)
 else()
     set(CMAKE_MODULES_INSTALL_PATH ${CMAKE_ROOT}/Modules)
