diff --git a/configure b/configure
index 878381b..be39a38 100755
--- a/configure
+++ b/configure
@@ -11143,13 +11143,14 @@ find_lib()
     gcc_bin_path=`$CC -print-search-dirs 2>/dev/null | fgrep programs: | sed 's/[^=]*=\(.*\)/\1/' | sed 's/:/ /g'`
     gcc_lib_path=`$CC -print-search-dirs 2>/dev/null | fgrep libraries: | sed 's/[^=]*=\(.*\)/\1/' | sed 's/:/ /g'`
     env_lib_path=`echo $LIBS $LDFLAGS $* | sed 's/-L[ ]*//g'`
+    env_path=`echo $PATH | sed 's/:/ /g'`
     if test "$cross_compiling" = yes; then
         host_lib_path=""
     else
         host_lib_path="/usr/$base_libdir /usr/local/$base_libdir"
     fi
-    for path in $gcc_bin_path $gcc_lib_path $env_lib_path $host_lib_path; do
-        lib=`ls -- $path/$1 2>/dev/null | sed -e '/\.so\..*\./d' -e 's,.*/,,' | sort | tail -1`
+    for path in $gcc_bin_path $gcc_lib_path $env_lib_path $env_path $host_lib_path; do
+        lib=`ls -- $path/$1 2>/dev/null | sort | sed 's/.*\/\(.*\)/\1/; q'`
         if test x$lib != x; then
             echo $lib
             return
@@ -11938,7 +11939,7 @@ fi
             *-*-darwin*)
                 modplug_lib=`find_lib libmodplug.dylib`
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin* | *-*-mingw*)
                 modplug_lib=`find_lib "libmodplug*.dll"`
                 ;;
             *)
@@ -12070,7 +12071,7 @@ fi
             *-*-darwin*)
                 mikmod_lib=`find_lib libmikmod.dylib`
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin* | *-*-mingw*)
                 mikmod_lib=`find_lib "libmikmod*.dll"`
                 ;;
             *)
@@ -12132,7 +12133,7 @@ fi
             *mingw32ce*)
                 use_music_midi_native=no
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin* | *-*-mingw*)
                 use_music_midi_native=yes
                 EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lwinmm"
                 ;;
@@ -12221,8 +12222,11 @@ fi
                 *-*-darwin*)
                     fluidsynth_lib=`find_lib libfluidsynth.dylib`
                     ;;
-                *-*-cygwin* | *-*-mingw32*)
+                *-*-cygwin* | *-*-mingw*)
                     fluidsynth_lib=`find_lib "fluidsynth*.dll"`
+                    if test x$fluidsynth_lib = x; then
+                        fluidsynth_lib=`find_lib "libfluidsynth*.dll"`
+                    fi
                     ;;
                 *)
                     fluidsynth_lib=`find_lib "libfluidsynth[0-9]*.so.*"`
@@ -12331,8 +12335,11 @@ fi
                 *-*-darwin*)
                     ogg_lib=`find_lib libvorbisidec.dylib`
                     ;;
-                *-*-cygwin* | *-*-mingw32*)
+                *-*-cygwin* | *-*-mingw*)
                     ogg_lib=`find_lib "vorbisidec*.dll"`
+                    if test x$ogg_lib = x; then
+                        ogg_lib=`find_lib "libvorbisidec*.dll"`
+                    fi
                     ;;
                 *)
                     ogg_lib=`find_lib "libvorbisidec[0-9]*.so.*"`
@@ -12409,7 +12416,7 @@ fi
                 *-*-darwin*)
                     ogg_lib=`find_lib libvorbisfile.dylib`
                     ;;
-                *-*-cygwin* | *-*-mingw32*)
+                *-*-cygwin* | *-*-mingw*)
                     ogg_lib=`find_lib "libvorbisfile*.dll"`
                     ;;
                 *)
@@ -12545,7 +12552,7 @@ fi
                 *-*-darwin*)
                     flac_lib=`find_lib libFLAC.dylib`
                     ;;
-                *-*-cygwin* | *-*-mingw32*)
+                *-*-cygwin* | *-*-mingw*)
                     flac_lib=`find_lib "libFLAC-*.dll"`
                     ;;
                 *)
@@ -12841,7 +12848,7 @@ rm -f core conftest.err conftest.$ac_objext \
             *-*-darwin*)
                 smpeg_lib=`find_lib libsmpeg2.dylib`
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin* | *-*-mingw*)
                 smpeg_lib=`find_lib "smpeg2*.dll"`
                 ;;
             *)
