Index: SDL2_mixer-2.0.2/configure.in
===================================================================
--- SDL2_mixer-2.0.2.orig/configure.in
+++ SDL2_mixer-2.0.2/configure.in
@@ -55,6 +55,7 @@ else
     hostaliaswindres="$host_alias-windres"
 fi
 AC_CHECK_PROGS(WINDRES, [windres $hostaliaswindres $host_os-windres])
+LT_PROG_RC
 
 dnl Set up the compiler and linker flags
 case "$host" in
@@ -702,10 +703,10 @@ DEPENDS=`echo "$DEPENDS" | sed 's,\\$,\\
 
 VERSION_OBJECTS=`echo $VERSION_SOURCES`
 VERSION_DEPENDS=`echo $VERSION_SOURCES`
-VERSION_OBJECTS=`echo "$VERSION_OBJECTS" | sed 's,[[^ ]]*/\([[^ ]]*\)\.rc,$(objects)/\1.o,g'`
+VERSION_OBJECTS=`echo "$VERSION_OBJECTS" | sed 's,[[^ ]]*/\([[^ ]]*\)\.rc,$(objects)/\1.lo,g'`
 VERSION_DEPENDS=`echo "$VERSION_DEPENDS" | sed 's,\([[^ ]]*\)/\([[^ ]]*\)\.rc,\\
-$(objects)/\2.o: \1/\2.rc\\
-	\$(WINDRES) \$< \$@,g'`
+$(objects)/\2.lo: \1/\2.rc\\
+	\$(LIBTOOL) --tag=RC --mode=compile \$(WINDRES) \$< -o \$@,g'`
 VERSION_DEPENDS=`echo "$VERSION_DEPENDS" | sed 's,\\$,\\\\$,g'`
 
 PLAYWAVE_SOURCES="$srcdir/playwave.c"
