--- Tktable2.10/tclconfig/tcl.m4.orig	2015-07-29 09:08:46.206702700 +0300
+++ Tktable2.10/tclconfig/tcl.m4	2015-07-29 09:08:53.774459400 +0300
@@ -2952,7 +2952,7 @@
 	AC_MSG_RESULT([ok (TEA ${TEA_VERSION})])
     fi
     case "`uname -s`" in
-	*win32*|*WIN32*|*CYGWIN_NT*|*CYGWIN_9*|*CYGWIN_ME*|*MINGW32_*)
+	*win32*|*WIN32*|*CYGWIN_NT*|*CYGWIN_9*|*CYGWIN_ME*|*MINGW*)
 	    AC_CHECK_PROG(CYGPATH, cygpath, cygpath -w, echo)
 	    EXEEXT=".exe"
 	    TEA_PLATFORM="windows"
