--- sox_ng-14.7.0.4/src/Makefile.am.orig	2026-01-15 08:48:59.033006400 +0100
+++ sox_ng-14.7.0.4/src/Makefile.am	2026-01-15 08:50:55.962052800 +0100
@@ -118,7 +118,7 @@
 
 libsox_ng_la_CFLAGS = @WARN_CFLAGS@
 libsox_ng_la_LDFLAGS = @APP_LDFLAGS@ -version-info @SHLIB_VERSION@ \
-  -export-symbols-regex '^((sox|lsx)_.*|create_lpc10_(en|de)coder_state|lpc10_(en|de)code|gsm_(create|destroy|encode|decode))$$'
+  -export-symbols-regex '^(win32_utf16_to_utf8|win32_utf8_to_utf16|(sox|lsx)_.*|create_lpc10_(en|de)coder_state|lpc10_(en|de)code|gsm_(create|destroy|encode|decode))$$'
 
 if HAVE_WIN32_LTDL
   libsox_ng_la_SOURCES += win32-ltdl.c win32-ltdl.h
