--- a/buckygen.c
+++ b/buckygen.c
@@ -274,8 +274,8 @@ int comparenodes(unsigned char *canong, int codelength, int type, SPLAYNODE *lis
 #define PRESENT(p) old_splaynode(p, is_new_node)
 
 
-void new_splaynode();
-void old_splaynode();
+void new_splaynode(SPLAYNODE *el, unsigned char *canong, int codelength, int type, int *is_new_node);
+void old_splaynode(SPLAYNODE *el, int *is_new_node);
 
 #include "splay.c"
 
