From ddb8271cecc48b840e9612682c4c13257f922e5b Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Mon, 1 Jan 2024 14:14:45 +0530
Subject: [PATCH 074/N] build: add MINGW support for selectmodule

---
 Modules/selectmodule.c | 6 +++---
 configure.ac           | 7 ++++++-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/Modules/selectmodule.c b/Modules/selectmodule.c
index d234d50..8e58158 100644
--- a/Modules/selectmodule.c
+++ b/Modules/selectmodule.c
@@ -167,9 +167,9 @@ seq2set(PyObject *seq, fd_set *set, pylist fd2obj[FD_SETSIZE + 1])
         v = PyObject_AsFileDescriptor( o );
         if (v == -1) goto finally;
 
-#if defined(_MSC_VER)
+#if defined(MS_WIN32)
         max = 0;                             /* not used for Win32 */
-#else  /* !_MSC_VER */
+#else  /* !MS_WIN32 */
         if (!_PyIsSelectable_fd(v)) {
             PyErr_SetString(PyExc_ValueError,
                         "filedescriptor out of range in select()");
@@ -177,7 +177,7 @@ seq2set(PyObject *seq, fd_set *set, pylist fd2obj[FD_SETSIZE + 1])
         }
         if (v > max)
             max = v;
-#endif /* _MSC_VER */
+#endif /* MS_WIN32 */
         FD_SET(v, set);
 
         /* add object and its file descriptor to the list */
diff --git a/configure.ac b/configure.ac
index 1662e92..966c988 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4545,6 +4545,11 @@ AS_CASE([$MACHDEP],
   [MULTIPROCESSING_LIBS=""]
 )
 
+dnl On MINGW, you need to link againt ws2_32 for selectmodule
+AS_CASE([$MACHDEP],
+  [win32], [SELECTMODULE_LIBS="-lws2_32"],
+  [SELECTMODULE_LIBS=""]
+)
 
 dnl detect sqlite3 from Emscripten emport
 PY_CHECK_EMSCRIPTEN_PORT([LIBSQLITE3], [-sUSE_SQLITE3])
@@ -8325,7 +8330,7 @@ PY_STDLIB_MOD_SIMPLE([_pickle])
 PY_STDLIB_MOD_SIMPLE([_queue])
 PY_STDLIB_MOD_SIMPLE([_random])
 PY_STDLIB_MOD_SIMPLE([_remote_debugging])
-PY_STDLIB_MOD_SIMPLE([select])
+PY_STDLIB_MOD_SIMPLE([select], [], [$SELECTMODULE_LIBS])
 PY_STDLIB_MOD_SIMPLE([_struct])
 PY_STDLIB_MOD_SIMPLE([_types])
 PY_STDLIB_MOD_SIMPLE([_typing])
