From 1ec3f6c0667b8afc3163ad5651d9d3875e4609db Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 2 Sep 2025 16:07:14 +0300
Subject: [PATCH 084/N] tests: Fix test for library name

---
 Lib/test/test_sysconfig.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/Lib/test/test_sysconfig.py b/Lib/test/test_sysconfig.py
index 8c6308a..bae62be 100644
--- a/Lib/test/test_sysconfig.py
+++ b/Lib/test/test_sysconfig.py
@@ -469,8 +469,12 @@ class TestSysConfig(unittest.TestCase, VirtualEnvironmentMixin):
         major, minor = sys.version_info[:2]
         abiflags = sysconfig.get_config_var('ABIFLAGS')
         if sys.platform.startswith('win'):
-            self.assertEqual(library, f'python{major}{minor}{abiflags}.dll')
-            self.assertEqual(library, ldlibrary)
+            if sys._is_mingw:
+                self.assertEqual(library, f'libpython{major}.{minor}{abiflags}.a')
+                self.assertEqual(ldlibrary, f'libpython{major}.{minor}{abiflags}.dll.a')
+            else:
+                self.assertEqual(library, f'python{major}{minor}{abiflags}.dll')
+                self.assertEqual(library, ldlibrary)
         elif is_apple_mobile:
             framework = sysconfig.get_config_var('PYTHONFRAMEWORK')
             self.assertEqual(ldlibrary, f"{framework}.framework/{framework}")
