--- a/indicator/CMakeLists.txt
+++ b/indicator/CMakeLists.txt
@@ -46,6 +46,7 @@
 target_link_libraries(kdeconnect-indicator Qt::Widgets KF6::CoreAddons KF6::I18n KF6::Notifications KF6::DBusAddons KF6::KCMUtils KF6::Crash kdeconnectinterfaces kdeconnectcore kdeconnectversion)
 
 if (WIN32)
+    target_compile_options(kdeconnect-indicator PUBLIC -fexceptions)
     target_link_libraries(kdeconnect-indicator windowsapp)
 else()
     target_link_libraries(kdeconnect-indicator KF6::StatusNotifierItem)
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -51,18 +51,9 @@
 endif()
 
 if(WITH_PULSEAUDIO OR WIN32)
-    add_subdirectory(pausemusic)
-endif()
-
-if(WITH_PULSEAUDIO OR WIN32)
     add_subdirectory(findthisdevice)
 endif()
 
-if (WITH_PULSEAUDIO OR WIN32 OR APPLE)
-    add_subdirectory(systemvolume)
-endif()
-
-
 # If we split notifications per plugin, in several notifyrc files, they won't
 # appear in the same group in the Notifications KCM
 install(FILES kdeconnect.notifyrc DESTINATION ${KDE_INSTALL_KNOTIFYRCDIR})
--- a/plugins/sendnotifications/CMakeLists.txt
+++ b/plugins/sendnotifications/CMakeLists.txt
@@ -16,6 +16,7 @@
 
 
 if(WIN32)
+target_compile_options(kdeconnect_sendnotifications PUBLIC -fexceptions)
 target_sources(kdeconnect_sendnotifications PRIVATE windowsnotificationslistener.cpp)
 target_link_libraries(kdeconnect_sendnotifications runtimeobject windowsapp)
 else()
