--- a/plugins/systemvolume/PolicyConfig.h
+++ b/plugins/systemvolume/PolicyConfig.h
@@ -51,7 +51,7 @@
 
     virtual HRESULT STDMETHODCALLTYPE SetPropertyValue(PCWSTR, const PROPERTYKEY &, PROPVARIANT *);
 
-    virtual HRESULT STDMETHODCALLTYPE SetDefaultEndpoint(__in PCWSTR wszDeviceId, __in ERole eRole);
+    virtual HRESULT STDMETHODCALLTYPE SetDefaultEndpoint(PCWSTR wszDeviceId, ERole eRole);
 
     virtual HRESULT STDMETHODCALLTYPE SetEndpointVisibility(PCWSTR, INT);
 };
@@ -97,7 +97,7 @@
 
     virtual HRESULT STDMETHODCALLTYPE SetPropertyValue(PCWSTR, const PROPERTYKEY &, PROPVARIANT *);
 
-    virtual HRESULT STDMETHODCALLTYPE SetDefaultEndpoint(__in PCWSTR wszDeviceId, __in ERole eRole);
+    virtual HRESULT STDMETHODCALLTYPE SetDefaultEndpoint(PCWSTR wszDeviceId, ERole eRole);
 
     virtual HRESULT STDMETHODCALLTYPE SetEndpointVisibility(PCWSTR,
                                                             INT); // not available on Windows 7, use method from IPolicyConfig
