--- ogre-1.11.1/PlugIns/FreeImageCodec/CMakeLists.txt.orig	2018-07-20 15:54:05.023608000 +0300
+++ ogre-1.11.1/PlugIns/FreeImageCodec/CMakeLists.txt	2018-07-20 15:54:35.318861300 +0300
@@ -11,7 +11,9 @@
 list(APPEND HEADER_FILES ${CMAKE_BINARY_DIR}/include/OgreFreeImageCodecExports.h)
 set(SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreFreeImageCodec.cpp")
 
-add_definitions(-DFREEIMAGE_LIB)
+if (NOT MINGW)
+  add_definitions(-DFREEIMAGE_LIB)
+endif ()
 add_library(Codec_FreeImage ${OGRE_LIB_TYPE} ${HEADER_FILES} ${SOURCES})
 target_link_libraries(Codec_FreeImage PUBLIC OgreMain "${FreeImage_LIBRARIES}")
 target_include_directories(Codec_FreeImage PUBLIC 
