--- senet-1.3.13/configure.ac.orig	2016-01-14 08:49:48.966946100 +0300
+++ enet-1.3.13/configure.ac	2016-01-14 08:56:43.378946100 +0300
@@ -7,6 +7,14 @@
 AC_PROG_CC
 AC_PROG_LIBTOOL
 
+
+# Check system type
+case "$target" in
+*mingw*)
+     LIBS="$LIBS -lws2_32 -lwinmm"
+     ;;
+esac
+
 AC_CHECK_FUNC(getaddrinfo, [AC_DEFINE(HAS_GETADDRINFO)])
 AC_CHECK_FUNC(getnameinfo, [AC_DEFINE(HAS_GETNAMEINFO)])
 AC_CHECK_FUNC(gethostbyaddr_r, [AC_DEFINE(HAS_GETHOSTBYADDR_R)])
