--- ois/CMakeLists.txt.orig	2018-07-22 18:00:21.513004200 +0300
+++ ois/CMakeLists.txt	2018-07-22 18:00:29.609418400 +0300
@@ -222,7 +222,7 @@
   FRAMEWORK DESTINATION Library/Frameworks
 )
 
-if(UNIX)
+if(UNIX OR MINGW)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/OIS.pc.in ${CMAKE_BINARY_DIR}/OIS.pc @ONLY)
     install(FILES ${CMAKE_BINARY_DIR}/OIS.pc DESTINATION ${OIS_LIB_DIRECTORY}/pkgconfig)
 endif()
