--- ois/CMakeLists.txt.orig	2018-07-22 18:06:00.183599700 +0300
+++ ois/CMakeLists.txt	2018-07-22 18:08:30.755064200 +0300
@@ -155,11 +155,15 @@
         set(DIRECTX_ARCH x86)
     endif()
 
+    if(MINGW)
+        target_link_libraries(OIS "dinput8" "dxguid" "xinput")
+    else()
     if (WINDOWSSDK_FOUND)
         target_link_libraries(OIS "dinput8.lib" "dxguid.lib")
     else()
         target_link_libraries(OIS "${DXSDK_DIR}/Lib/${DIRECTX_ARCH}/dinput8.lib" "${DXSDK_DIR}/Lib/${DIRECTX_ARCH}/dxguid.lib")
     endif()
+    endif()
 endif()
 
 if(UNIX)
