From ad8df954ac5396b6d14e10c1f630a3bf5d61a16e Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Mon, 17 Nov 2014 07:50:04 +0300
Subject: [PATCH 02/10] Fix macros name

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 36ad204..b1a338d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,7 +5,7 @@ AC_INIT( [OIS], 1.4.0 )
 
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADERS([includes/config.h])
+AC_CONFIG_HEADERS([includes/config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
 case "$host" in
-- 
2.9.0

