From 4d621d46f8c0c7f3ea7381a85f6d1f7aff8f5900 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Mon, 17 Nov 2014 08:42:32 +0300
Subject: [PATCH 03/10] InputManager: Cannot have pure virtual methods.

---
 includes/OISInputManager.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/OISInputManager.h b/includes/OISInputManager.h
index 6794510..a17f244 100644
--- a/includes/OISInputManager.h
+++ b/includes/OISInputManager.h
@@ -171,7 +171,7 @@ namespace OIS
 		@remarks
 			Called from createInputSystem, gives derived input class a chance to setup after it is created
 		*/
-		virtual void _initialize(ParamList &paramList) = 0;
+		virtual void _initialize(ParamList &paramList) {};
 
 		/**
 		@remarks
-- 
2.9.0

