From 6472f08a3284d80562d0355e24537b773af5e1e4 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Mon, 17 Nov 2014 10:54:57 +0300
Subject: [PATCH 05/10] Fix typo.

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 24374fc..c9dfccd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,9 +42,9 @@ case "$host" in
                 DX_LIBS="-ldinput8 -ldxguid -lole32 -loleaut32 -luser32"
                 OIS_LIBS="$OIS_LIBS $DX_LIBS"
                 if test "${enable_shared}" = yes; then
-                    AC_DEFINE([OIS_DYNAMIC_LIB], [], [Import settings for Win32]);;
+                    AC_DEFINE([OIS_DYNAMIC_LIB], [], [Import settings for Win32])
                 fi
-                AC_DEFINE([OIS_NONCLIENT_BUILD], [], [Export settings for Win32]);;
+                AC_DEFINE([OIS_NONCLIENT_BUILD], [], [Export settings for Win32])
                 ;;
         *-*-linux*)
                 AC_DEFINE(PLATFORM_LINUX, 1, [Platform is Linux])
-- 
2.9.0

