From 90ea5b77d56c4a9b833dc37c9dcb8d0db2193262 Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Tue, 18 Nov 2014 00:04:14 +0300
Subject: [PATCH 08/10] Add macros directory to configure.

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 91ead1c..16888ed 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,6 +3,8 @@
 AC_PREREQ(2.69)
 AC_INIT( [OIS], 1.4.0 )
 
+AC_CONFIG_MACRO_DIR([m4])
+
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([includes/config.h])
-- 
2.9.0

