From 0a39b785d3a65ba183aa15e6611f5ebdceb2c794 Mon Sep 17 00:00:00 2001
From: ael <ael-code@users.noreply.github.com>
Date: Wed, 24 Oct 2018 00:42:01 +0200
Subject: [PATCH] fix include path in pkg-config template (#1276)

---
 capstone.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/capstone.pc.in b/capstone.pc.in
index 7715397ea..69541b248 100644
--- a/capstone.pc.in
+++ b/capstone.pc.in
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 libdir=${prefix}/lib@LIBSUFFIX@
-includedir=${prefix}/include
+includedir=${prefix}/include/capstone
 
 Name: capstone
 Description: Capstone disassembly engine
