--- assimp/tools/assimp_cmd/assimp_cmd.rc.orig	2014-06-12 21:50:48.160000000 +0400
+++ assimp/tools/assimp_cmd/assimp_cmd.rc	2014-06-12 21:54:56.986400000 +0400
@@ -1,7 +1,7 @@
 // Microsoft Visual C++ generated resource script.
 //
 #include "resource.h"
-#include "../../revision.h"
+#include "revision.h"
 
 #define APSTUDIO_READONLY_SYMBOLS
 /////////////////////////////////////////////////////////////////////////////
