--- a/clang_delta/RemoveUnusedStructField.cpp
+++ b/clang_delta/RemoveUnusedStructField.cpp
@@ -247,7 +247,7 @@ const Expr *RemoveUnusedStructField::getInitExprFromDesignatedInitExpr(
       }
       else {
         const DesignatedInitExpr::Designator *DS = DIE->getDesignator(0);
-        const FieldDecl *CurrFD = DS->getField();
+        const FieldDecl *CurrFD = DS->getFieldDecl();
         if ((CurrFD && FD == CurrFD) ||
             (CurrFD == NULL && DS->getFieldName() == FD->getIdentifier())) {
           IsFirstField = (I == 0);
