diff -bur creduce-o/clang_delta/Transformation.cpp creduce/clang_delta/Transformation.cpp
--- creduce-o/clang_delta/Transformation.cpp	2025-04-18 00:20:17.540740900 -0600
+++ creduce/clang_delta/Transformation.cpp	2025-04-18 00:20:29.202521300 -0600
@@ -94,6 +94,8 @@
 
 void Transformation::outputTransformedSource(llvm::raw_ostream &OutStream)
 {
+  using namespace llvm;
+
   FileID MainFileID = SrcManager->getMainFileID();
   const RewriteBuffer *RWBuf = TheRewriter.getRewriteBufferFor(MainFileID);
 
diff -bur creduce-o/clang_delta/TransformationManager.cpp creduce/clang_delta/TransformationManager.cpp
--- creduce-o/clang_delta/TransformationManager.cpp	2025-04-18 00:20:17.516237400 -0600
+++ creduce/clang_delta/TransformationManager.cpp	2025-04-18 00:31:08.114333100 -0600
@@ -91,7 +91,7 @@
   ClangInstance = new CompilerInstance();
   assert(ClangInstance);
   
-  ClangInstance->createDiagnostics();
+  ClangInstance->createDiagnostics(ClangInstance->getVirtualFileSystem());
 
   TargetOptions &TargetOpts = ClangInstance->getTargetOpts();
   PreprocessorOptions &PPOpts = ClangInstance->getPreprocessorOpts();
