Index: base/tkimg.h
===================================================================
--- a/base/tkimg.h	(revision 957)
+++ b/base/tkimg.h	(working copy)
@@ -24,7 +24,7 @@
 #    define tkimg_vsnprintf vsnprintf
 #endif /* _MSC_VER_ */
 
-#if defined(__MINGW32__)
+#if defined(__MINGW32__) && !defined(__aarch64__) && !defined(_UCRT)
 #    define SETJMP(jbuf)        __builtin_setjmp(jbuf)
 #    define LONGJMP(jbuf, code) __builtin_longjmp(jbuf, code)
 #else
Index: compat/libtiff/libtiff/tif_ojpeg.c
===================================================================
--- a/compat/libtiff/libtiff/tif_ojpeg.c	(revision 957)
+++ b/compat/libtiff/libtiff/tif_ojpeg.c	(working copy)
@@ -152,8 +152,13 @@
  */
 
 /* define LIBJPEG_ENCAP_EXTERNAL */
-#define SETJMP(jbuf) setjmp(jbuf)
-#define LONGJMP(jbuf, code) longjmp(jbuf, code)
+#if defined(__MINGW32__) && !defined(__aarch64__) && !defined(_UCRT)
+#    define SETJMP(jbuf)        __builtin_setjmp(jbuf)
+#    define LONGJMP(jbuf, code) __builtin_longjmp(jbuf, code)
+#else
+#    define SETJMP(jbuf)        setjmp(jbuf)
+#    define LONGJMP(jbuf, code) longjmp(jbuf, code)
+#endif
 #define JMP_BUF jmp_buf
 #define OJPEG_BUFFER 2048
 /* define EGYPTIANWALK */
