From d2d4fdd9d9f4be4ecfc5cf53dc1b591fda4ef099 Mon Sep 17 00:00:00 2001
From: Lubosz Sarnecki <lubosz@gmail.com>
Date: Sun, 3 Aug 2014 11:28:10 +0200
Subject: [PATCH] msys2: add header for BOOL defintion

---
 clutter/win32/clutter-backend-win32.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clutter/win32/clutter-backend-win32.c b/clutter/win32/clutter-backend-win32.c
index fc7c604..2415d94 100644
--- a/clutter/win32/clutter-backend-win32.c
+++ b/clutter/win32/clutter-backend-win32.c
@@ -24,6 +24,7 @@
 #endif
 
 #include <windows.h>
+#include <minwindef.h>
 
 #include "clutter-backend-win32.h"
 #include "clutter-stage-win32.h"
-- 
2.0.4

