--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,9 +70,10 @@
 
 if (NOT APPLE AND NOT ANDROID AND NOT WIN32 OR (WIN32 AND NOT WITH_SNORETOAST))
     find_package(Qt6 ${REQUIRED_QT_VERSION} CONFIG REQUIRED DBus)
-    find_package(Canberra REQUIRED)
+    find_package(Canberra)
     set_package_properties(Canberra PROPERTIES
-        PURPOSE "Needed to build audio notification support")
+        PURPOSE "Needed to build audio notification support"
+        TYPE OPTIONAL)
     if (TARGET Canberra::Canberra)
         add_definitions(-DHAVE_CANBERRA)
     endif()
