--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2337,12 +2337,7 @@
             OUTPUT_NAME "${LIBRARY_NAME}_static"
         )
     elseif(MINGW)
-        #
-        # For compatibility, build the shared library without the "lib" prefix on
-        # MinGW as well.
-        #
         set_target_properties(${LIBRARY_NAME} PROPERTIES
-            PREFIX ""
             OUTPUT_NAME "${LIBRARY_NAME}"
         )
         set_target_properties(${LIBRARY_NAME}_static PROPERTIES
