diff -bur meanwhile-1.1.1-orig/configure.ac meanwhile-1.1.1/configure.ac
--- meanwhile-1.1.1-orig/configure.ac	2025-02-23 17:05:11 -0700
+++ meanwhile-1.1.1/configure.ac	2025-02-23 17:05:18 -0700
@@ -117,13 +117,10 @@
 
 AC_CONFIG_FILES(
 	[Makefile src/Makefile src/mpi/Makefile]
-	[samples/Makefile]
 	[doc/Makefile doc/Doxyfile]
 	[meanwhile.spec meanwhile.pc]
 )
 
-AC_CONFIG_FILES([samples/build], [chmod +x samples/build])
-
 AC_OUTPUT()
 
 
--- meanwhile-1.1.1-orig/Makefile.am	2025-02-23 17:10:34 -0700
+++ meanwhile-1.1.1/Makefile.am	2025-02-23 17:10:39 -0700
@@ -1,6 +1,6 @@
 
 
-SUBDIRS = doc samples src
+SUBDIRS = doc src
 
 
 EXTRA_DIST =	meanwhile.spec \
