From 99226cd819cdd4b4e72f818e818b04e978b47ea4 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Wed, 15 Oct 2025 15:30:35 -0400
Subject: Add include of <cstdint>

---
 LiteEditor/cl_editor.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/LiteEditor/cl_editor.h b/LiteEditor/cl_editor.h
index e0356ff75..c010b47be 100644
--- a/LiteEditor/cl_editor.h
+++ b/LiteEditor/cl_editor.h
@@ -44,6 +44,7 @@
 #include "plugin.h"
 #include "stringhighlighterjob.h"
 
+#include <cstdint>
 #include <map>
 #include <unordered_map>
 #include <vector>
-- 
