From e5c7e55a9f64cd76212679f35f5a74b0914d424c Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Tue, 26 Jul 2022 19:31:36 -0400
Subject: Comment out wxCrafter plugin

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93044f01b..552b8b954 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -716,10 +716,10 @@ else()
         endif()
     endif(NOT NO_CORE_PLUGINS)
 
-    # wxCrafter is not a core plugin so we include it we find the directory
-    if(IS_DIRECTORY "${CL_SRC_ROOT}/wxcrafter" AND NOT DISABLE_CXX AND NOT NO_CORE_PLUGINS)
-        add_subdirectory(wxcrafter)
-    endif()
+    ## wxCrafter is not a core plugin so we include it we find the directory
+    #if(IS_DIRECTORY "${CL_SRC_ROOT}/wxcrafter" AND NOT DISABLE_CXX AND NOT NO_CORE_PLUGINS)
+        #add_subdirectory(wxcrafter)
+    #endif()
 
     ## Executables
     if(NOT SDK_ONLY)
-- 
2.37.1.windows.1

