--- a/LiteEditor/OutputDebugStringThread.h	2015-06-01 23:22:47.501600000 +0300
+++ b/Outline/OutputDebugStringThread.h	2015-06-01 23:39:02.750800000 +0300
@@ -3,6 +3,10 @@
 
 #include <wx/thread.h> // Base class: wxThread
 #include "cl_command_event.h"
+#ifdef __WXMSW__
+#include <windows.h>
+#include <wx/msw/winundef.h>
+#endif
 
 wxDECLARE_EVENT(wxEVT_OUTPUT_DEBUG_STRING, clCommandEvent);
 class OutputDebugStringThread : public wxThread
--- a/Outline/outline_tab.cpp	2015-06-01 23:22:47.501600000 +0300
+++ b/Outline/outline_tab.cpp	2015-06-01 23:39:02.750800000 +0300
@@ -25,6 +25,9 @@
 #include "bitmap_loader.h"
 #include "clToolBar.h"
 #include "cl_config.h"
+#ifdef __WXMSW__
+#include <wx/msw/winundef.h>
+#endif
 #include "event_notifier.h"
 #include "file_logger.h"
 #include "fileextmanager.h"
diff --git a/wxcrafter/wxcTreeView.cpp b/wxcrafter/wxcTreeView.cpp
index 3e7bb0bd1..326d14a75 100644
--- a/wxcrafter/wxcTreeView.cpp
+++ b/wxcrafter/wxcTreeView.cpp
@@ -12,6 +12,10 @@
 #include <wx/aui/auibar.h>
 #include "drawingutils.h"
 
+#if defined(AddForm)
+    #undef AddForm
+#endif
+
 const wxEventType wxEVT_SHOW_WXCRAFTER_DESIGNER = wxNewEventType();
 
 BEGIN_EVENT_TABLE(wxcTreeView, wxcTreeViewBaseClass)
--- a/wxcrafter/wxcrafter_plugin.cpp
+++ b/wxcrafter/wxcrafter_plugin.cpp
@@ -71,6 +71,10 @@
 #include <wx/stc/stc.h>
 #include <wx/xrc/xmlres.h>
 
+#if defined(AddForm)
+    #undef AddForm
+#endif
+
 static wxCrafterPlugin* thePlugin = NULL;
 
 // Define the plugin entry point
