From eaacaefc34e605570b90c72f2138899126a3acd5 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sun, 19 Oct 2025 07:57:56 -0400
Subject: Disable selected compiler warnings

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bedc6c69d..d0b6fbe4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -186,6 +186,10 @@ write_file("${CMAKE_BINARY_DIR}/post_install.cmake"
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations"
 )# Avoid very multiple warnings spam due to deprecated wx methods
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-attributes")
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-command-line-argument")
+endif()
 if(UNIX AND NOT APPLE)
   set(CMAKE_CXX_FLAGS
       "${CMAKE_CXX_FLAGS} -isystem /usr/include/harfbuzz -isystem /usr/local/include/harfbuzz"
-- 
2.48.1.windows.1

