From 9c922c63867669a1115d46e31cfa830c198fa6aa Mon Sep 17 00:00:00 2001
From: Eran Ifrah <eran.ifrah@gmail.com>
Date: Mon, 1 Mar 2021 12:21:25 +0200
Subject: fixed build error with recent gcc compilers

---
 CodeLite/fileutils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CodeLite/fileutils.cpp b/CodeLite/fileutils.cpp
index 4d8434147..fb7b5f711 100644
--- a/CodeLite/fileutils.cpp
+++ b/CodeLite/fileutils.cpp
@@ -631,7 +631,7 @@ std::string FileUtils::ToStdString(const wxString& str) { return StringUtils::To
 
 bool FileUtils::ReadBufferFromFile(const wxFileName& fn, wxString& data, size_t bufferSize)
 {
-    std::wifstream fin(fn.GetFullPath().c_str(), std::ios::binary);
+    std::wifstream fin(fn.GetFullPath().ToStdString(), std::ios::binary);
     if(fin.bad()) {
         clERROR() << "Failed to open file:" << fn;
         return false;
-- 
