From 37a33611265a39e64200e37d60f6182bc4c61b33 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Thu, 16 Oct 2025 20:17:53 -0400
Subject: Fix cmake splashscreen location

---
 LiteEditor/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/LiteEditor/CMakeLists.txt b/LiteEditor/CMakeLists.txt
index 8a1fdfcfe..1de66680e 100644
--- a/LiteEditor/CMakeLists.txt
+++ b/LiteEditor/CMakeLists.txt
@@ -227,6 +227,13 @@ if(NOT APPLE)
         install(
             CODE "execute_process(COMMAND rm -f \$ENV{DESTDIR}/${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/icon-theme.cache)"
         )
+    elseif(MINGW AND BUILD_FOR_MSYS2_MINGW_PACKAGE)
+        install(
+            FILES ${CL_SRC_ROOT}/art/splashscreen@2x.png
+            DESTINATION ${CMAKE_INSTALL_PREFIX}/share/codelite/images
+            RENAME splashscreen.png)
+
+        install(FILES ${CL_SRC_ROOT}/art/splashscreen@2x.png DESTINATION ${CMAKE_INSTALL_PREFIX}/share/codelite/images)
     elseif(MINGW)
         install(
             FILES ${CL_SRC_ROOT}/art/splashscreen@2x.png
-- 
