diff --git a/LiteEditor/app.cpp b/LiteEditor/app.cpp
index 3f0eb452d..adb9f48da 100644
--- a/LiteEditor/app.cpp
+++ b/LiteEditor/app.cpp
@@ -462,7 +462,7 @@ bool CodeLiteApp::OnInit()
 
     // Set the global log file verbosity. NB Doing this earlier seems to break wxGTK debug output when debugging
     // CodeLite itself :/
-    FileLogger::OpenLog("codelite.log", clConfig::Get().Read(kConfigLogVerbosity, FileLogger::Error));
+    FileLogger::OpenLog("codelite.log", clConfig::Get().Read(kConfigLogVerbosity, FileLogger::Developer));
     clDEBUG() << "Starting codelite..." << endl;
 
     // Copy gdb pretty printers from the installation folder to a writeable location
