--- a/wxcrafter/CMakeLists.txt
+++ b/wxcrafter/CMakeLists.txt
@@ -82,7 +82,11 @@ endif(NOT OFFICIAL_CL_BUILD)
 endif ( WXC_APP )
 
 FILE(GLOB SRCS "*.cpp")
+if ( NOT WXC_APP )
+FILE(GLOB WXCLIB_SRCS "wxcLib/json_node.cpp" "wxcLib/cJSON.cpp" "wxcLib/wxc*.cpp")
+else ( NOT WXC_APP )
 FILE(GLOB WXCLIB_SRCS "wxcLib/*.cpp")
+endif ( NOT WXC_APP )
 
 include_directories( ${CMAKE_CURRENT_LIST_DIR} )
 include_directories( "${CMAKE_CURRENT_LIST_DIR}/wxcLib" )
