From 5133c1cc528168d3ca5c84c8a3e1d5197f4a095d Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Fri, 5 Aug 2022 12:37:06 -0400
Subject: unnest StringHashOptionalCase and StringCompareOptionalCase

---
 SpellChecker/IHunSpell.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SpellChecker/IHunSpell.h b/SpellChecker/IHunSpell.h
index 0ae1dc185..1a5fc7b47 100644
--- a/SpellChecker/IHunSpell.h
+++ b/SpellChecker/IHunSpell.h
@@ -52,8 +52,6 @@ class CorrectSpellingDlg;
 class SpellCheck;
 class IEditor;
 // ------------------------------------------------------------
-class IHunSpell
-{
     class StringHashOptionalCase
     {
     public:
@@ -96,6 +94,8 @@ class IHunSpell
         bool m_isCaseSensitive;
     };
 
+class IHunSpell
+{
 public:
     IHunSpell();
     virtual ~IHunSpell();
-- 
2.37.1.windows.1

