--- codelite/SpellChecker/CMakeLists.txt.orig	2014-07-02 22:34:10.188600000 +0400
+++ codelite/SpellChecker/CMakeLists.txt	2014-07-02 22:38:48.105000000 +0400
@@ -1,13 +1,17 @@
 # define minimum cmake version
 cmake_minimum_required(VERSION 2.6.2)
 
-find_library(LIBHUNSPELL
-         NAMES libhunspell.so
-         HINTS  /usr/lib /usr/local/lib /usr/lib/x86_64-linux-gnu /usr/lib/i386-linux-gnu)
+find_library(LIBHUNSPELL 
+            NAMES hunspell
+            PATH_SUFFIXES lib)
 
 find_path(  HUNSPELL_H
             NAMES hunspell.h
-            HINTS  /usr/include /usr/local/include /usr/include/hunspell /usr/local/include/hunspell)
+            HINTS
+              ENV HUNSPELL_DIR
+            PATHS
+              /usr/include /usr/local/include /usr/include/hunspell /usr/local/include/hunspell
+            PATH_SUFFIXES include/hunspell include)
 
 if ( (LIBHUNSPELL STREQUAL "LIBHUNSPELL-NOTFOUND" OR HUNSPELL_H STREQUAL "HUNSPELL_H-NOTFOUND") AND NOT APPLE )
     message("    **")
