--- codelite/CodeFormatter/astyle_main.h.orig	2014-07-02 23:35:17.395600000 +0400
+++ codelite/CodeFormatter/astyle_main.h	2014-07-02 23:35:53.993200000 +0400
@@ -93,7 +93,7 @@
 #endif
 
 // for mingw BOM, UTF-16, and Unicode functions
-#if defined(__MINGW32__)
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR)
 #if (__MINGW32_MAJOR_VERSION > 3)  || ((__MINGW32_MAJOR_VERSION == 3) && (__MINGW32_MINOR_VERSION < 16))
 #error - Use MinGW compiler version 4 or higher
 #endif
