--- codelite/LiteEditor/CMakeLists.txt.orig	2014-07-03 22:22:02.622200000 +0400
+++ codelite/LiteEditor/CMakeLists.txt	2014-07-03 22:59:50.077400000 +0400
@@ -68,7 +68,12 @@
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
+if (MINGW)
+add_definitions(-DWINDRES)
+add_executable(codelite ${SRCS} code_parser.rc)
+else()
 add_executable(codelite ${SRCS})
+endif()
 
 set( ADDITIONAL_LIBRARIES "" )
 
--- codelite/codelite_terminal/CMakeLists.txt.orig	2014-07-03 22:22:02.450600000 +0400
+++ codelite/codelite_terminal/CMakeLists.txt	2014-07-03 23:03:01.598000000 +0400
@@ -27,7 +27,15 @@
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
+if (MINGW)
+add_definitions(-DWINDRES)
+execute_process(COMMAND sh ${CL_WX_CONFIG} --debug=no --rescomp OUTPUT_VARIABLE WX_RC_FLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
+string(REGEX REPLACE "windres" "" WX_RC_FLAGS ${WX_RC_FLAGS})
+set (CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} ${WX_RC_FLAGS}")
+add_executable(codelite-terminal ${SRCS} win_resources.rc)
+else()
 add_executable(codelite-terminal ${SRCS})
+endif()
 set ( UTIL_LIB "" )
 if ( UNIX )
     set( UTIL_LIB "-lutil")
