diff --git a/../gtkglarea-2.1.0/configure.ac b/configure.ac
index 2d58545..f6cc6a3 100644
--- a/../gtkglarea-2.1.0/configure.ac
+++ b/configure.ac
@@ -40,6 +40,7 @@ AC_SUBST(VERSION)
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
+AM_PROG_AR
 
 dnl Initialize libtool
 LT_PREREQ([2.2.6])
@@ -174,8 +175,8 @@ AC_MSG_RESULT($have_shaders)
 LIBS="$saved_LIBS"
 AM_CONDITIONAL(HAVE_SHADERS, test "$have_shaders" = "yes")
 
-AC_SUBST(GL_CFLAGS)
-AC_SUBST(GL_LIBS)
+AC_SUBST([GL_CFLAGS])
+AC_SUBST([GL_LIBS])
 
 AC_CONFIG_FILES([
   Makefile
