From 9f3ebed8efebc45b74fa3baf2a4295352465ade9 Mon Sep 17 00:00:00 2001
From: David Macek <david.macek.0@gmail.com>
Date: Tue, 13 Jan 2015 17:48:53 +0100
Subject: [PATCH] Use unixy filenames even on Windows

---
 compiler/platform.nim | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/compiler/platform.nim b/compiler/platform.nim
index c6cb3d2..b0e0862 100644
--- a/compiler/platform.nim
+++ b/compiler/platform.nim
@@ -43,9 +43,9 @@ const
       parDir: "..", dllFrmt: "$1.dll", altDirSep: "/", objExt: ".obj",
       newLine: "\x0D\x0A", pathSep: ";", dirSep: "\\", scriptExt: ".bat",
       curDir: ".", exeExt: ".exe", extSep: ".", props: {ospCaseInsensitive}),
-     (name: "Windows", parDir: "..", dllFrmt: "$1.dll", altDirSep: "/",
-      objExt: ".obj", newLine: "\x0D\x0A", pathSep: ";", dirSep: "\\",
-      scriptExt: ".bat", curDir: ".", exeExt: ".exe", extSep: ".",
+     (name: "Windows", parDir: "..", dllFrmt: "lib$1.dll", altDirSep: "/", 
+      objExt: ".a", newLine: "\x0D\x0A", pathSep: ";", dirSep: "\\", 
+      scriptExt: ".cmd", curDir: ".", exeExt: ".exe", extSep: ".", 
       props: {ospCaseInsensitive}),
      (name: "OS2", parDir: "..",
       dllFrmt: "$1.dll", altDirSep: "/",
-- 
1.9.4.msysgit.2

