--- nim-1.6.12/lib/system/inclrtl.nim.orig	2023-03-26 20:14:38.409292800 +0200
+++ nim-1.6.12/lib/system/inclrtl.nim	2023-03-26 20:15:15.131483100 +0200
@@ -34,7 +34,7 @@
   `{.rtl.}` should only be used for non-generic procs.
   ]#
   const nimrtl* =
-    when defined(windows): "nimrtl.dll"
+    when defined(windows): "libnimrtl.dll"
     elif defined(macosx): "libnimrtl.dylib"
     else: "libnimrtl.so"
   {.pragma: rtl, importc: "nimrtl_$1", dynlib: nimrtl, gcsafe.}
