--- nim-1.6.12/lib/wrappers/openssl.nim.orig	2023-03-09 20:19:29.000000000 +0100
+++ nim-1.6.12/lib/wrappers/openssl.nim	2023-03-26 21:23:55.296239000 +0200
@@ -71,12 +71,12 @@
         DLLUtilName* = "libeay32.dll"
   elif defined(cpu64):
     const
-      DLLSSLName* = "(libssl-1_1-x64|ssleay64|libssl64).dll"
-      DLLUtilName* = "(libcrypto-1_1-x64|libeay64).dll"
+      DLLSSLName* = "libssl-3-x64.dll"
+      DLLUtilName* = "libcrypto-3-x64.dll"
   else:
     const
-      DLLSSLName* = "(libssl-1_1|ssleay32|libssl32).dll"
-      DLLUtilName* = "(libcrypto-1_1|libeay32).dll"
+      DLLSSLName* = "libssl-3.dll"
+      DLLUtilName* = "libcrypto-3.dll"
 
   from winlean import SocketHandle
 else:
