diff -Naur poco-1.14.2-all-orig/Foundation/include/Poco/Platform.h poco-1.14.2-all/Foundation/include/Poco/Platform.h
--- poco-1.14.2-all-orig/Foundation/include/Poco/Platform.h	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/include/Poco/Platform.h	2025-10-09 12:52:37.445370900 +0300
@@ -245,6 +245,10 @@
 #endif
 
 
+#if defined (__MINGW32__)
+	#define POCO_COMPILER_MINGW
+#endif
+
 #if defined(__clang__)
 	#define POCO_COMPILER_CLANG
 	#if __has_include(<cxxabi.h>)
@@ -257,11 +261,6 @@
 	#if __has_include(<cxxabi.h>)
 		#define POCO_HAVE_CXXABI_H
 	#endif
-	#if defined (__MINGW32__) || defined (__MINGW64__)
-		#define POCO_COMPILER_MINGW
-	#endif
-#elif defined (__MINGW32__) || defined (__MINGW64__)
-	#define POCO_COMPILER_MINGW
 #elif defined (__INTEL_COMPILER) || defined(__ICC) || defined(__ECC) || defined(__ICL)
 	#define POCO_COMPILER_INTEL
 #elif defined (__SUNPRO_CC)
