--- a/Foundation/include/Poco/Platform.h
+++ b/Foundation/include/Poco/Platform.h
@@ -240,6 +240,10 @@
 #endif
 
 
+#if defined (__MINGW32__)
+	#define POCO_COMPILER_MINGW
+#endif
+
 #if defined(__clang__)
 	#define POCO_COMPILER_CLANG
 	#define POCO_HAVE_CXXABI_H
@@ -261,11 +264,6 @@
 	#if __has_include(<cxxabi.h>)
 		#define POCO_HAVE_CXXABI_H
 	#endif
-	#if defined (__MINGW32__) || defined (__MINGW64__)
-		#define POCO_COMPILER_MINGW
-	#endif
-#elif defined (__MINGW32__) || defined (__MINGW64__)
-	#define POCO_COMPILER_MINGW
 #elif defined (__INTEL_COMPILER) || defined(__ICC) || defined(__ECC) || defined(__ICL)
 	#define POCO_COMPILER_INTEL
 #elif defined (__SUNPRO_CC)
