--- poco-poco-1.7.7-release/cmake/DefinePlatformSpecifc.cmake.orig	2014-11-17 11:55:04.044400000 +0300
+++ poco-poco-1.7.7-release/cmake/DefinePlatformSpecifc.cmake	2014-11-17 11:55:44.370400000 +0300
@@ -108,7 +108,7 @@
 
 if (CMAKE_COMPILER_IS_MINGW)
   add_definitions(-DWC_NO_BEST_FIT_CHARS=0x400  -DPOCO_WIN32_UTF8)
-  add_definitions(-D_WIN32 -DMINGW32 -DWINVER=0x500 -DODBCVER=0x0300 -DPOCO_THREAD_STACK_SIZE)
+  add_definitions(-D_WIN32 -DMINGW32 -DWINVER=0x501 -DPOCO_NO_FPENVIRONMENT -DODBCVER=0x0300 -DPOCO_THREAD_STACK_SIZE)
 endif (CMAKE_COMPILER_IS_MINGW)
 
 # SunPro C++
--- poco-poco-1.7.7-release/cmake/PocoMacros.cmake.orig	2014-11-17 12:31:47.705800000 +0300
+++ poco-poco-1.7.7-release/cmake/PocoMacros.cmake	2014-11-17 12:32:39.209200000 +0300
@@ -24,8 +24,13 @@
       set(kit81_bindir "${kit81_dir}/bin/x86")
     endif (X64)
   endif ()
+  if(MINGW)
+  find_program(CMAKE_MC_COMPILER windmc.exe HINTS "${sdk_bindir}" "${kit_bindir}"
+    DOC "path to message compiler")
+  else()
   find_program(CMAKE_MC_COMPILER mc.exe HINTS "${sdk_bindir}" "${kit_bindir}" "${kit81_bindir}"
     DOC "path to message compiler")
+  endif()
   if (NOT CMAKE_MC_COMPILER)
     message(FATAL_ERROR "message compiler not found: required to build")
   endif (NOT CMAKE_MC_COMPILER)
