--- poco-poco-1.6.0-release/cmake/FindMySQL.cmake.orig	2015-03-02 09:27:14.895200000 +0300
+++ poco-poco-1.6.0-release/cmake/FindMySQL.cmake	2015-03-02 09:33:06.774600000 +0300
@@ -12,7 +12,9 @@
 		$ENV{MYSQL_DIR}/include
 		$ENV{ProgramFiles}/MySQL/*/include
 		${BINDIR32}/MySQL/include
-		$ENV{SystemDrive}/MySQL/*/include)
+		$ENV{SystemDrive}/MySQL/*/include
+		PATH_SUFFIXES
+			include include/mysql include/mariadb)
 
 if (WIN32)
 	if (CMAKE_BUILD_TYPE STREQUAL Debug)
@@ -20,7 +20,7 @@
 		add_definitions(-DDBUG_OFF)
 	endif (CMAKE_BUILD_TYPE STREQUAL Debug)
 
-	find_library(MYSQL_LIB NAMES mysqlclient
+	find_library(MYSQL_LIB NAMES mysqlclient mariadbclient
 				 PATHS
 				 $ENV{MYSQL_DIR}/lib/${libsuffixDist}
 				 $ENV{MYSQL_DIR}/libmysql/${libsuffixBuild}
