diff -Naur poco-1.14.2-all-orig/ActiveRecord/CMakeLists.txt poco-1.14.2-all/ActiveRecord/CMakeLists.txt
--- poco-1.14.2-all-orig/ActiveRecord/CMakeLists.txt	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/ActiveRecord/CMakeLists.txt	2025-10-02 11:51:54.927730300 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Crypto/CMakeLists.txt poco-1.14.2-all/Crypto/CMakeLists.txt
--- poco-1.14.2-all-orig/Crypto/CMakeLists.txt	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/Crypto/CMakeLists.txt	2025-10-02 11:51:14.279862600 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Encodings/CMakeLists.txt poco-1.14.2-all/Encodings/CMakeLists.txt
--- poco-1.14.2-all-orig/Encodings/CMakeLists.txt	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Encodings/CMakeLists.txt	2025-10-02 11:52:21.713363600 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS(SRCS Encodings ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Foundation/CMakeLists.txt poco-1.14.2-all/Foundation/CMakeLists.txt
--- poco-1.14.2-all-orig/Foundation/CMakeLists.txt	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/CMakeLists.txt	2025-10-02 11:24:38.192259500 +0300
@@ -16,7 +16,7 @@
 	)
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/JSON/CMakeLists.txt poco-1.14.2-all/JSON/CMakeLists.txt
--- poco-1.14.2-all-orig/JSON/CMakeLists.txt	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/JSON/CMakeLists.txt	2025-10-02 11:24:23.533259000 +0300
@@ -8,7 +8,7 @@
 POCO_SOURCES(SRCS pdjson src/pdjson.c)
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/JWT/CMakeLists.txt poco-1.14.2-all/JWT/CMakeLists.txt
--- poco-1.14.2-all-orig/JWT/CMakeLists.txt	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/JWT/CMakeLists.txt	2025-10-02 11:52:37.752773200 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/MongoDB/CMakeLists.txt poco-1.14.2-all/MongoDB/CMakeLists.txt
--- poco-1.14.2-all-orig/MongoDB/CMakeLists.txt	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/MongoDB/CMakeLists.txt	2025-10-02 11:52:48.504174900 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Net/CMakeLists.txt poco-1.14.2-all/Net/CMakeLists.txt
--- poco-1.14.2-all-orig/Net/CMakeLists.txt	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/Net/CMakeLists.txt	2025-10-02 11:53:03.797302900 +0300
@@ -33,7 +33,7 @@
 endif()
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/NetSSL_OpenSSL/CMakeLists.txt poco-1.14.2-all/NetSSL_OpenSSL/CMakeLists.txt
--- poco-1.14.2-all-orig/NetSSL_OpenSSL/CMakeLists.txt	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/NetSSL_OpenSSL/CMakeLists.txt	2025-10-02 11:53:15.092100300 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/PageCompiler/CMakeLists.txt poco-1.14.2-all/PageCompiler/CMakeLists.txt
--- poco-1.14.2-all-orig/PageCompiler/CMakeLists.txt	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/PageCompiler/CMakeLists.txt	2025-10-02 11:53:26.090749300 +0300
@@ -5,7 +5,7 @@
 POCO_SOURCES_AUTO(SRCS ${SRCS_G})
 
 # Version Resource
-if(MSVC)
+if(WIN32)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Prometheus/CMakeLists.txt poco-1.14.2-all/Prometheus/CMakeLists.txt
--- poco-1.14.2-all-orig/Prometheus/CMakeLists.txt	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/Prometheus/CMakeLists.txt	2025-10-02 11:53:38.940083100 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Redis/CMakeLists.txt poco-1.14.2-all/Redis/CMakeLists.txt
--- poco-1.14.2-all-orig/Redis/CMakeLists.txt	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/Redis/CMakeLists.txt	2025-10-02 11:53:49.590925600 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Util/CMakeLists.txt poco-1.14.2-all/Util/CMakeLists.txt
--- poco-1.14.2-all-orig/Util/CMakeLists.txt	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/Util/CMakeLists.txt	2025-10-02 11:54:00.662186000 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/XML/CMakeLists.txt poco-1.14.2-all/XML/CMakeLists.txt
--- poco-1.14.2-all-orig/XML/CMakeLists.txt	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/XML/CMakeLists.txt	2025-10-02 11:54:11.762542600 +0300
@@ -12,7 +12,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
diff -Naur poco-1.14.2-all-orig/Zip/CMakeLists.txt poco-1.14.2-all/Zip/CMakeLists.txt
--- poco-1.14.2-all-orig/Zip/CMakeLists.txt	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/Zip/CMakeLists.txt	2025-10-02 11:54:23.058987700 +0300
@@ -7,7 +7,7 @@
 POCO_HEADERS_AUTO(SRCS ${HDRS_G})
 
 # Version Resource
-if(MSVC AND BUILD_SHARED_LIBS)
+if(WIN32 AND BUILD_SHARED_LIBS)
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 endif()
