diff -Naur poco-1.14.2-all-orig/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h poco-1.14.2-all/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h
--- poco-1.14.2-all-orig/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h	2025-10-02 11:41:01.177867300 +0300
@@ -245,7 +245,7 @@
 // explicit instantiation definition
 #ifndef POCO_DOC
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 extern template class Diagnostics<SQLHENV, SQL_HANDLE_ENV>;
 extern template class Diagnostics<SQLHDBC, SQL_HANDLE_DBC>;
 extern template class Diagnostics<SQLHSTMT, SQL_HANDLE_STMT>;
diff -Naur poco-1.14.2-all-orig/Data/ODBC/include/Poco/Data/ODBC/Error.h poco-1.14.2-all/Data/ODBC/include/Poco/Data/ODBC/Error.h
--- poco-1.14.2-all-orig/Data/ODBC/include/Poco/Data/ODBC/Error.h	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/Data/ODBC/include/Poco/Data/ODBC/Error.h	2025-10-02 11:41:16.599999300 +0300
@@ -127,7 +127,7 @@
 // explicit instantiation definition
 #ifndef POCO_DOC
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 extern template class Error<SQLHENV, SQL_HANDLE_ENV>;
 extern template class Error<SQLHDBC, SQL_HANDLE_DBC>;
 extern template class Error<SQLHSTMT, SQL_HANDLE_STMT>;
diff -Naur poco-1.14.2-all-orig/Data/ODBC/include/Poco/Data/ODBC/ODBCException.h poco-1.14.2-all/Data/ODBC/include/Poco/Data/ODBC/ODBCException.h
--- poco-1.14.2-all-orig/Data/ODBC/include/Poco/Data/ODBC/ODBCException.h	2025-05-12 12:00:13.000000000 +0300
+++ poco-1.14.2-all/Data/ODBC/include/Poco/Data/ODBC/ODBCException.h	2025-10-02 11:41:33.079490400 +0300
@@ -141,7 +141,7 @@
 // explicit instantiation definition
 #ifndef POCO_DOC
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 extern template class HandleException<SQLHENV, SQL_HANDLE_ENV>;
 extern template class HandleException<SQLHDBC, SQL_HANDLE_DBC>;
 extern template class HandleException<SQLHSTMT, SQL_HANDLE_STMT>;
diff -Naur poco-1.14.2-all-orig/Foundation/include/Poco/BufferedBidirectionalStreamBuf.h poco-1.14.2-all/Foundation/include/Poco/BufferedBidirectionalStreamBuf.h
--- poco-1.14.2-all-orig/Foundation/include/Poco/BufferedBidirectionalStreamBuf.h	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/include/Poco/BufferedBidirectionalStreamBuf.h	2025-10-02 11:42:14.911675800 +0300
@@ -182,7 +182,7 @@
 // We provide an instantiation for char.
 //
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 extern template class BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
 #else
 extern template class Foundation_API BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
diff -Naur poco-1.14.2-all-orig/Foundation/include/Poco/BufferedStreamBuf.h poco-1.14.2-all/Foundation/include/Poco/BufferedStreamBuf.h
--- poco-1.14.2-all-orig/Foundation/include/Poco/BufferedStreamBuf.h	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/include/Poco/BufferedStreamBuf.h	2025-10-02 11:47:33.084313600 +0300
@@ -164,7 +164,7 @@
 // We provide an instantiation for char.
 //
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 extern template class BasicBufferedStreamBuf<char, std::char_traits<char>>;
 #else
 extern template class Foundation_API BasicBufferedStreamBuf<char, std::char_traits<char>>;
diff -Naur poco-1.14.2-all-orig/Foundation/include/Poco/Dynamic/Struct.h poco-1.14.2-all/Foundation/include/Poco/Dynamic/Struct.h
--- poco-1.14.2-all-orig/Foundation/include/Poco/Dynamic/Struct.h	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/include/Poco/Dynamic/Struct.h	2025-10-02 11:21:58.900162600 +0300
@@ -288,7 +288,7 @@
 };
 
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 
 extern template class Struct<std::string>;
 extern template class Struct<int>;
diff -Naur poco-1.14.2-all-orig/Foundation/include/Poco/UTFString.h poco-1.14.2-all/Foundation/include/Poco/UTFString.h
--- poco-1.14.2-all-orig/Foundation/include/Poco/UTFString.h	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/include/Poco/UTFString.h	2025-10-02 11:55:25.830434000 +0300
@@ -307,13 +307,24 @@
 } // namespace Poco
 
 
+
 #if defined(POCO_USE_STRING16)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
+// Explicitly instatiated shared pointer in JSON library
+extern template class std::basic_string<Poco::UTF16Char, Poco::UTF16CharTraits>;
+#else
 extern template class Foundation_API std::basic_string<Poco::UTF16Char, Poco::UTF16CharTraits>;
 #endif
+#endif
 
 #if defined(POCO_USE_STRING32)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
+// Explicitly instatiated shared pointer in JSON library
+extern template class std::basic_string<Poco::UTF32Char, Poco::UTF32CharTraits>;
+#else
 extern template class Foundation_API std::basic_string<Poco::UTF32Char, Poco::UTF32CharTraits>;
 #endif
+#endif
 
 
 #endif // Foundation_UTFString_INCLUDED
diff -Naur poco-1.14.2-all-orig/Foundation/include/Poco/UnbufferedStreamBuf.h poco-1.14.2-all/Foundation/include/Poco/UnbufferedStreamBuf.h
--- poco-1.14.2-all-orig/Foundation/include/Poco/UnbufferedStreamBuf.h	2025-05-12 12:00:11.000000000 +0300
+++ poco-1.14.2-all/Foundation/include/Poco/UnbufferedStreamBuf.h	2025-10-02 11:47:57.368456800 +0300
@@ -164,7 +164,7 @@
 // We provide an instantiation for char.
 //
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 extern template class BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
 #else
 extern template class Foundation_API BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
diff -Naur poco-1.14.2-all-orig/JSON/include/Poco/JSON/Array.h poco-1.14.2-all/JSON/include/Poco/JSON/Array.h
--- poco-1.14.2-all-orig/JSON/include/Poco/JSON/Array.h	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/JSON/include/Poco/JSON/Array.h	2025-10-02 11:49:34.258919600 +0300
@@ -34,7 +34,7 @@
 }
 
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 // Explicitly instantiated shared pointer in JSON library
 extern template class Poco::SharedPtr<Poco::JSON::Array>;
 #else
diff -Naur poco-1.14.2-all-orig/JSON/include/Poco/JSON/Object.h poco-1.14.2-all/JSON/include/Poco/JSON/Object.h
--- poco-1.14.2-all-orig/JSON/include/Poco/JSON/Object.h	2025-05-12 12:00:12.000000000 +0300
+++ poco-1.14.2-all/JSON/include/Poco/JSON/Object.h	2025-10-02 11:49:40.876010000 +0300
@@ -41,7 +41,7 @@
 
 }
 
-#if defined(POCO_OS_FAMILY_WINDOWS)
+#if defined(POCO_OS_FAMILY_WINDOWS) && !defined(__clang__)
 // Explicitly instatiated shared pointer in JSON library
 extern template class Poco::SharedPtr<Poco::JSON::Object>;
 #else
