--- poco-poco-1.6.0-release/Foundation/include/Poco/Config.h.orig	2015-03-02 09:52:02.328400000 +0300
+++ poco-poco-1.6.0-release/Foundation/include/Poco/Config.h	2015-03-02 09:53:15.274000000 +0300
@@ -23,7 +23,7 @@
 // Define to enable Windows Unicode (UTF-8) support
 // NOTE: As of POCO C++ Libraries release 1.6.0, compiling POCO
 // without POCO_WIN32_UTF8 defined on Windows is deprecated.
-#define POCO_WIN32_UTF8
+// #define POCO_WIN32_UTF8
 
 
 // Define to enable C++11 support
