--- poco-poco-1.4.7-release/build/config/MinGW.orig	2014-11-16 23:36:56.214400000 +0300
+++ poco-poco-1.4.7-release/build/config/MinGW	2014-11-16 23:37:56.180800000 +0300
@@ -66,9 +66,9 @@
 #
 # System Specific Flags
 #
-SYSFLAGS = -D_WIN32 -DMINGW32 -DWINVER=0x501 -DPOCO_NO_FPENVIRONMENT -DPCRE_STATIC -DPOCO_THREAD_STACK_SIZE -I/usr/local/include -I/usr/include
+SYSFLAGS = -D_WIN32 -DMINGW32 -DWINVER=0x501 -DPOCO_NO_FPENVIRONMENT -DPCRE_STATIC -DPOCO_THREAD_STACK_SIZE
 
 #
 # System Specific Libraries
 #
-SYSLIBS  = -L/usr/local/lib -L/usr/lib -liphlpapi -lssl -lcrypto -lws2_32
+SYSLIBS  = -liphlpapi -lssl -lcrypto -lws2_32
