diff -urN librasterlite2-1.1.0-beta1-orig/src/Makefile.am librasterlite2-1.1.0-beta1/src/Makefile.am
--- librasterlite2-1.1.0-beta1-orig/src/Makefile.am	2020-08-24 23:20:40 +0800
+++ librasterlite2-1.1.0-beta1/src/Makefile.am	2021-11-26 20:06:34 +0800
@@ -28,7 +28,7 @@
 
 if MINGW
 librasterlite2_la_LDFLAGS = -avoid-version -no-undefined
-librasterlite2_la_LIBADD += -lm
+librasterlite2_la_LIBADD += -lpthread -lm
 else
 librasterlite2_la_LDFLAGS = -version-info 1:0:0
 librasterlite2_la_LIBADD += -lpthread -lm
@@ -56,7 +56,7 @@
 
 if MINGW
 mod_rasterlite2_la_LDFLAGS = -module -avoid-version -no-undefined
-mod_rasterlite2_la_LIBADD += -lm
+mod_rasterlite2_la_LIBADD += -lpthread -lm
 else
 mod_rasterlite2_la_LDFLAGS = -module -version-info 1:0:0
 mod_rasterlite2_la_LIBADD += -lpthread -lm
diff -urN librasterlite2-1.1.0-beta1-orig/tools/Makefile.am librasterlite2-1.1.0-beta1/tools/Makefile.am
--- librasterlite2-1.1.0-beta1-orig/tools/Makefile.am	2020-08-24 23:20:40 +0800
+++ librasterlite2-1.1.0-beta1/tools/Makefile.am	2021-11-26 20:07:04 +0800
@@ -36,7 +36,7 @@
 	@LIBCURL_LIBS@ @LIBXML2_LIBS@ @LIBFONTCONFIG_LIBS@ \
 	@LIBFREETYPE2_LIBS@ @LIBPIXMAN_LIBS@ @LIBCAIRO_LIBS@ \
 	@LIBSPATIALITE_LIBS@ \
-	-lrasterlite2 -lws2_32
+	-lrasterlite2 -lws2_32 -lpthread
 else
 wmslite_LDADD = @LIBPNG_LIBS@ @LIBWEBP_LIBS@ @LIBLZMA_LIBS@ \
 	@LIBLZ4_LIBS@ @LIBZSTD_LIBS@ @LIBOPENJP2_LIBS@ \
