diff -Naur nodeeditor-2.1.3.orig/CMakeLists.txt nodeeditor-2.1.3/CMakeLists.txt
--- nodeeditor-2.1.3.orig/CMakeLists.txt	2019-05-01 13:11:24.000000000 +0200
+++ nodeeditor-2.1.3/CMakeLists.txt	2021-08-17 23:46:58.009857300 +0200
@@ -166,6 +166,7 @@
   EXPORT NodeEditorTargets
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
 