From 12d299345b51c1686c1d82986de6de020970bb57 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Sun, 25 Dec 2015 20:14:51 +0100
Subject: [PATCH] Globally provide DESTDIR for all platforms


diff --git a/cmake/modules/LLVMInstallSymlink.cmake b/cmake/modules/LLVMInstallSymlink.cmake
index 671a8ce..45c47e1 100644
--- a/cmake/modules/LLVMInstallSymlink.cmake
+++ b/cmake/modules/LLVMInstallSymlink.cmake
@@ -3,9 +3,9 @@
 # See PR8397.
 
 function(install_symlink name target outdir)
+set(DESTDIR $ENV{DESTDIR})
   if(UNIX)
     set(LINK_OR_COPY create_symlink)
-    set(DESTDIR $ENV{DESTDIR})
   else()
     set(LINK_OR_COPY copy)
   endif()
2.4.5
