From 53625b291503734db14e4a6146ceb06f288fd987 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Sun, 5 Jul 2015 00:54:34 +0100
Subject: [PATCH] libcxx: add support for mingw-w64


diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index d3d5f38..4820f2c 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -2,7 +2,7 @@ set(LIBCXX_LIB_CMAKEFILES_DIR "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTOR
 
 # Get sources
 file(GLOB LIBCXX_SOURCES ../src/*.cpp)
-if(WIN32)
+if(MSVC)
   file(GLOB LIBCXX_WIN32_SOURCES ../src/support/win32/*.cpp)
   list(APPEND LIBCXX_SOURCES ${LIBCXX_WIN32_SOURCES})
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS")
diff --git a/src/random.cpp b/src/random.cpp
index 4ab424e..d3062f5 100644
--- a/src/random.cpp
+++ b/src/random.cpp
@@ -7,7 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if defined(_LIBCPP_USING_WIN32_RANDOM)
+#if defined(_LIBCPP_USING_WIN32_RANDOM) || defined(__MINGW32__)
 // Must be defined before including stdlib.h to enable rand_s().
 #define _CRT_RAND_S
 #endif // defined(_LIBCPP_USING_WIN32_RANDOM)
-- 
2.4.5

