From b9e9319381d414d070ea477b7307f706343ee337 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 13 Nov 2013 18:27:28 +0000
Subject: [PATCH 2/5] program prefix ld

---
 autoconf/m4/link_options.m4 | 2 +-
 configure                   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/autoconf/m4/link_options.m4 b/autoconf/m4/link_options.m4
index b58d617..5fc465e 100644
--- a/autoconf/m4/link_options.m4
+++ b/autoconf/m4/link_options.m4
@@ -6,7 +6,7 @@
 AC_DEFUN([AC_LINK_GET_VERSION],
   [AC_CACHE_CHECK([for linker version],[llvm_cv_link_version],
   [
-   version_string="$(ld -v 2>&1 | head -1)"
+   version_string="$(${program_prefix}ld -v 2>&1 | head -1)"
 
    # Check for ld64.
    if (echo "$version_string" | grep -q "ld64"); then
diff --git a/configure b/configure
index ad63002..24901d9 100755
--- a/configure
+++ b/configure
@@ -7919,7 +7919,7 @@ if test "${llvm_cv_link_version+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-   version_string="$(ld -v 2>&1 | head -1)"
+   version_string="$(${program_prefix}ld -v 2>&1 | head -1)"
 
    # Check for ld64.
    if (echo "$version_string" | grep -q "ld64"); then
-- 
1.8.5.rc2

