From c3e65b5819be35557aead709dae34c5783ecd834 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 13 Nov 2013 18:01:13 +0000
Subject: [PATCH 1/5] cross runtime

---
 runtime/compiler-rt/Makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/runtime/compiler-rt/Makefile b/runtime/compiler-rt/Makefile
index e34a69f..7bfa18b 100644
--- a/runtime/compiler-rt/Makefile
+++ b/runtime/compiler-rt/Makefile
@@ -67,7 +67,7 @@ COMPILERRT_SRC_ROOT := $(LLVM_SRC_ROOT)/projects/compiler-rt
 #     information that compiler-rt would need to build libraries for all the
 #     requested targets. This might be a simple short term solution, but is
 #     likely to be unwieldy and irritating to maintain in the long term.
-ifneq ($(LLVM_CROSS_COMPILING),1)
+#ifneq ($(LLVM_CROSS_COMPILING),1)
 ifneq ($(CLANG_NO_RUNTIME),1)
 ifeq ($(shell test -d $(COMPILERRT_SRC_ROOT) && echo OK),OK)
 
@@ -77,7 +77,7 @@ ifeq ($(shell test -d $(COMPILERRT_SRC_ROOT) && echo OK),OK)
 # build/install runtime libraries for as many targets as clang was configured to
 # support.
 RuntimeDirs :=
-ifeq ($(OS),Darwin)
+ifeq ($(TARGET_OS),Darwin)
 RuntimeDirs += darwin darwin_embedded
 RuntimeLibrary.darwin.Configs := \
 	eprintf.a 10.4.a osx.a ios.a cc_kext.a cc_kext_ios5.a \
@@ -102,7 +102,7 @@ endif
 endif
 
 # On Linux, include a library which has all the runtime functions.
-ifeq ($(OS),Linux)
+ifeq ($(TARGET_OS),Linux)
 RuntimeDirs += linux
 RuntimeLibrary.linux.Configs :=
 
@@ -123,11 +123,11 @@ Runtime32BitConfigs = \
 	ubsan_cxx-i386.a
 
 # We currently only try to generate runtime libraries on x86.
-ifeq ($(ARCH),x86)
+ifeq ($(TARGET_ARCH),x86)
 RuntimeLibrary.linux.Configs += $(Runtime32BitConfigs)
 endif
 
-ifeq ($(ARCH),x86_64)
+ifeq ($(TARGET_ARCH),x86_64)
 RuntimeLibrary.linux.Configs += \
 	full-x86_64.a profile-x86_64.a san-x86_64.a asan-x86_64.a \
 	tsan-x86_64.a msan-x86_64.a ubsan-x86_64.a ubsan_cxx-x86_64.a \
@@ -213,7 +213,7 @@ $(ResourceLibDir)/$1/libclang_rt.%.dylib: \
 	$(Echo) Copying runtime library $1/$$* to build dir
 	$(Verb) cp $(PROJ_OBJ_DIR)/clang_$1/$$*/libcompiler_rt.dylib $$@
 	$(Echo) Fixing LC_ID_DYLIB of $$@
-	$(Verb) install_name_tool $$@ -id $$@
+	$(Verb) ${prefix}/bin/$(TARGET_TRIPLE)-install_name_tool $$@ -id $$@
 RuntimeLibrary.$1: \
 		$(RuntimeLibrary.$1.Configs:%=$(ResourceLibDir)/$1/libclang_rt.%)
 .PHONY: RuntimeLibrary.$1
@@ -260,4 +260,4 @@ clean-local:: CleanRuntimeLibraries
 
 endif
 endif
-endif
+#endif
-- 
1.8.5.rc2

