--- faad2-2.8.1/frontend/getopt.c.orig	2017-08-10 19:06:48 +0400
+++ faad2-2.8.1/frontend/getopt.c	2017-08-10 19:19:21 +0400
@@ -172,7 +172,9 @@
 #if __STDC__ || defined(PROTO)
 extern char *getenv(const char *name);
 extern int  strcmp (const char *s1, const char *s2);
+#if !defined(__MINGW64_VERSION_MAJOR)
 extern int  strncmp(const char *s1, const char *s2, unsigned int n);
+#endif
 
 static int my_strlen(const char *s);
 static char *my_index (const char *str, int chr);
