--- faad2-2.8.8/libfaad/decoder.c.orig	2018-06-02 07:58:04.852073000 -0400
+++ faad2-2.8.8/libfaad/decoder.c	2018-06-02 07:58:09.695726400 -0400
@@ -53,10 +53,6 @@
 uint16_t dbg_count;
 #endif
 
-#if defined WIN32 || defined _WIN32 || defined WIN64 || defined _WIN64
-#include "win32_ver.h"
-#endif
-
 /* static function declarations */
 static void* aac_frame_decode(NeAACDecStruct *hDecoder,
                               NeAACDecFrameInfo *hInfo,
