--- a/psycopg/win32_support.h
+++ b/psycopg/win32_support.h
@@ -37,11 +37,8 @@
 
 
 #ifdef _WIN32
-#ifndef __MINGW32__
 extern HIDDEN int gettimeofday(struct timeval * tp, void * tzp);
 extern HIDDEN void timeradd(struct timeval *a, struct timeval *b, struct timeval *c);
-#elif
-#endif
 
 extern HIDDEN void timersub(struct timeval *a, struct timeval *b, struct timeval *c);
 
--- a/psycopg/win32_support.c
+++ b/psycopg/win32_support.c
@@ -31,7 +31,6 @@
 
 #ifdef _WIN32
 
-#ifndef __MINGW32__
 /* millisecond-precision port of gettimeofday for Win32, taken from
    src/port/gettimeofday.c in PostgreSQL core */
 
@@ -73,7 +72,6 @@ timeradd(struct timeval *a, struct timev
     c->tv_sec += 1;
   }
 }
-#endif /* !defined(__MINGW32__) */
 
 /* timersub is missing on mingw & MS VC */
 void
