--- pango-1.30.0/pango/pangocairo.def.orig	2011-08-16 05:11:08 +0400
+++ pango-1.30.0/pango/pangocairo.def	2012-04-03 10:07:15 +0400
@@ -8,8 +8,8 @@
 	pango_cairo_create_context
 	pango_cairo_create_layout
 	pango_cairo_error_underline_path
-;	pango_cairo_fc_font_get_type
-;	pango_cairo_fc_font_map_get_type
+	pango_cairo_fc_font_get_type
+	pango_cairo_fc_font_map_get_type
 	pango_cairo_font_get_type
 	pango_cairo_font_get_scaled_font
 	pango_cairo_font_map_create_context
