--- pango-1.36.2/pango/pangocairo-fontmap.c.orig	2013-10-04 21:34:33.000000000 +0000
+++ pango-1.36.2/pango/pangocairo-fontmap.c	2014-02-08 02:25:22.257663500 +0000
@@ -66,10 +66,27 @@
 PangoFontMap *
 pango_cairo_font_map_new (void)
 {
+  const char *env_backend;
 #if !GLIB_CHECK_VERSION (2, 35, 3)
   /* Make sure that the type system is initialized */
   g_type_init ();
 #endif
+  env_backend = getenv ("PANGOCAIRO_BACKEND");
+  if (NULL != env_backend)
+  {
+#if defined(HAVE_CORE_TEXT) && defined (HAVE_CAIRO_QUARTZ)
+    if (0 == strcmp (env_backend, "quartz"))
+      return g_object_new (PANGO_TYPE_CAIRO_CORE_TEXT_FONT_MAP, NULL);
+#endif
+#if defined(HAVE_CAIRO_WIN32)
+    if (0 == strcmp (env_backend, "win32"))
+      return g_object_new (PANGO_TYPE_CAIRO_WIN32_FONT_MAP, NULL);
+#endif
+#if defined(HAVE_CAIRO_FREETYPE)
+    if (0 == strcmp (env_backend, "fontconfig"))
+      return g_object_new (PANGO_TYPE_CAIRO_FC_FONT_MAP, NULL);
+#endif
+  }
 #if defined(HAVE_CORE_TEXT) && defined (HAVE_CAIRO_QUARTZ)
   return g_object_new (PANGO_TYPE_CAIRO_CORE_TEXT_FONT_MAP, NULL);
 #elif defined(HAVE_CAIRO_WIN32)
