--- a/configure.ac	2011-06-20 22:39:08.377416539 +0200
+++ b/configure.ac	2011-06-20 22:40:46.959037254 +0200
@@ -145,17 +145,6 @@
 AC_LIBTOOL_WIN32_DLL
 AM_DISABLE_STATIC
 
-if test "$pango_os_win32" = "yes"; then
-  if test x$enable_static = xyes -o x$enable_static = x; then
-    AC_MSG_WARN([Disabling static library build, must build as DLL on Windows.])
-    enable_static=no
-  fi
-  if test x$enable_shared = xno; then
-    AC_MSG_WARN([Enabling shared library build, must build as DLL on Windows.])
-  fi
-  enable_shared=yes
-fi
-
 AM_PROG_LIBTOOL
 dnl when using libtool 2.x create libtool early, because it's used in configure
 m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
