--- pango-1.36.2/pango/Makefile.am.orig	2014-02-24 08:41:05.374400000 +0400
+++ pango-1.36.2/pango/Makefile.am	2014-02-24 08:52:49.792400000 +0400
@@ -371,7 +371,7 @@
 PangoCairo_1_0_gir_EXPORT_PACKAGES = pangocairo
 PangoCairo_1_0_gir_INCLUDES = GObject-2.0 cairo-1.0
 PangoCairo_1_0_gir_LIBS = $(Pango_1_0_gir_LIBS) libpangocairo-1.0.la
-PangoCairo_1_0_gir_CFLAGS = -I$(top_builddir) -I$(top_srcdir) $(PANGO_CFLAGS) $(FREETYPE_CFLAGS)
+PangoCairo_1_0_gir_CFLAGS = -I$(top_builddir) -I$(top_srcdir) $(PANGO_CFLAGS) $(FREETYPE_CFLAGS) $(CAIRO_CFLAGS)
 PangoCairo_1_0_gir_SCANNERFLAGS = --include-uninstalled=$(builddir)/Pango-1.0.gir --c-include "pango/pangocairo.h"
 if HAVE_CAIRO_FREETYPE
 PangoCairo_1_0_gir_SCANNERFLAGS += --include-uninstalled=$(builddir)/$(PANGOCAIRO_FONT_BACKEND_GI_MODULE)
