--- pango-1.42.1/pango-view/meson.build.orig	2018-04-07 12:00:40.000000000 +0200
+++ pango-1.42.1/pango-view/meson.build	2018-04-08 20:29:14.002005600 +0200
@@ -40,7 +40,7 @@
                         ])
 
 help2man = find_program('help2man', required: false)
-if help2man.found()
+if help2man.found() and get_option('default_library') != 'static'
   help2man_opts = [
     '--no-info',
     '--section=1',
