--- a/src/stardict_lib.cpp
+++ b/src/stardict_lib.cpp
@@ -551,7 +551,7 @@
     const std::list<std::string> vars = get_cache_variant(url);
 
     for (const std::string &item : vars) {
-        struct ::stat idxstat, cachestat;
+        GStatBuf idxstat, cachestat;
         if (g_stat(url.c_str(), &idxstat) != 0 || g_stat(item.c_str(), &cachestat) != 0)
             continue;
         if (cachestat.st_mtime < idxstat.st_mtime)
