--- a/lib/srchilite/Makefile.am
+++ b/lib/srchilite/Makefile.am
@@ -98,7 +98,7 @@ libsource_highlight_la_SOURCES = \
 libsource_highlight_la_LIBADD = $(top_builddir)/gl/libgnu.la \
 	@LTLIBOBJS@ $(BOOST_REGEX_LIB)
 
-libsource_highlight_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(LIBRARY_VERSION)
+libsource_highlight_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(LIBRARY_VERSION) -no-undefined
 
 library_includedir=$(includedir)/$(LIBRARY_NAMESPACE)
 library_include_HEADERS = $(h_sources)
