--- libebml-1.3.0/make/mingw32/Makefile.orig	2014-09-25 12:52:00.900800000 +0500
+++ libebml-1.3.0/make/mingw32/Makefile	2014-09-25 13:14:22.953200000 +0500
@@ -9,6 +9,11 @@
 # Compile with debug information?
 #DEBUG = yes
 
+prefix=/mingw
+bindir=$(prefix)/bin
+libdir=$(prefix)/lib
+includedir=$(prefix)/include/ebml
+INCLUDE_DIR=../../ebml
 
 #
 # Don't change anything below this line.
@@ -22,6 +26,11 @@
 WINDRES = $(CROSS)windres
 RANLIB  = $(CROSS)ranlib
 AR      = $(CROSS)ar
+INSTALL = install
+INSTALL_OPTS = -m 644
+INSTALL_OPTS_LIB = -m 644
+INSTALL_DIR_OPTS = -m 755
+
 RES  =
 SRC  = $(wildcard ../../src/*.cpp)
 OBJ  = $(patsubst %.cpp,%.o,$(SRC))
@@ -66,6 +75,32 @@
 		$(CXX) $(CXXFLAGS) -MM -MT $$o $$i >> .depend ; \
 	done
 
+ifeq (yes,$(SHARED))
+install: install_sharedlib install_headers
+else
+install: install_staticlib install_headers
+endif
+
+install_headers:
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(includedir)
+	for i in $(INCLUDE_DIR)/*.h; do \
+		$(INSTALL) $(INSTALL_OPTS) $$i $(includedir) ; \
+	done
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(includedir)/c
+	for i in $(INCLUDE_DIR)/c/*.h; do \
+		$(INSTALL) $(INSTALL_OPTS) $$i $(includedir)/c ; \
+	done
+
+install_staticlib: libebml.a
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(libdir)
+	$(INSTALL) $(INSTALL_OPTS_LIB) $(LIBS) $(libdir)
+
+install_sharedlib: libebml.dll
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(bindir)
+	$(INSTALL) $(INSTALL_DIR_OPTS) -d $(libdir)
+	$(INSTALL) $(INSTALL_OPTS_LIB) $(LIBS) $(bindir)
+	$(INSTALL) $(INSTALL_OPTS_LIB) $(LIBS).a $(libdir)
+
 #
 # include dependency files if they exist
 #
