--- libebml-1.3.6/CMakeLists.txt.orig	2018-04-26 09:50:46.289183100 +0300
+++ libebml-1.3.6/CMakeLists.txt	2018-04-26 09:53:57.777710900 +0300
@@ -92,6 +92,11 @@
     DEFINE_SYMBOL "EBML_DLL_EXPORT"
     C_VISIBILITY_PRESET hidden
     VISIBILITY_INLINES_HIDDEN 1)
+  if(MINGW)
+    set_target_properties(ebml
+      PROPERTIES
+      RUNTIME_OUTPUT_NAME ebml-4)
+  endif()
   target_compile_definitions(ebml
     PUBLIC EBML_DLL
     PRIVATE EBML_DLL_EXPORT)
