--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@ function(feature_info_on_off)
   endif()
 endfunction(feature_info_on_off)
 
-cmake_dependent_option(BUILD_SHARED_LIBS "Build libebml as a shared library (except Windows)" OFF "NOT WIN32" OFF)
+option(BUILD_SHARED_LIBS "Build libebml as a shared library" OFF)
 feature_info_on_off(BUILD_SHARED_LIBS "will build as a dynamic library" "will build as a static library")
 
 option(DISABLE_PKGCONFIG "Disable PkgConfig module generation" OFF)
