--- ./configure.ac.orig	2015-03-25 18:12:11.388179700 +0100
+++ ./configure.ac	2015-03-25 18:12:49.726012100 +0100
@@ -46,6 +46,7 @@
 	glib-2.0 >= 2.38.0
 	gtk+-3.0 >= 3.14.0
 	libpeas-1.0 >= 0.7.4
+	libpeas-gtk-1.0 >= 0.7.4
 	eog >= 3.15.90
 ])
 AC_SUBST(EOG_LIBS)
--- ./plugins/exif-display/Makefile.am.orig	2014-08-10 17:37:36.000000000 +0200
+++ ./plugins/exif-display/Makefile.am	2015-03-25 18:12:11.398181200 +0100
@@ -25,10 +25,11 @@
 	eog-exif-display-plugin-setup.h
 
 libexif_display_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
 libexif_display_la_LIBADD = \
 	$(EOG_LIBS)		\
+	-leog \
 	$(EXIFDISPLAY_LIBS)
 
 eog-exif-display-plugin-resources.c: eog-exif-display-plugin.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/eog-exif-display-plugin.gresource.xml)
--- ./plugins/fit-to-width/Makefile.am.orig	2014-08-09 20:39:08.000000000 +0200
+++ ./plugins/fit-to-width/Makefile.am	2015-03-25 18:12:11.448183500 +0100
@@ -15,9 +15,9 @@
 	eog-fit-to-width-plugin.c
 
 libfit_to_width_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libfit_to_width_la_LIBADD = $(EOG_LIBS)
+libfit_to_width_la_LIBADD = $(EOG_LIBS) -leog
 
 # Plugin Info
 
--- ./plugins/hide-titlebar/Makefile.am.orig	2014-08-09 20:39:08.000000000 +0200
+++ ./plugins/hide-titlebar/Makefile.am	2015-03-25 18:12:11.455184100 +0100
@@ -15,9 +15,9 @@
 	eog-hide-titlebar-plugin.c
 
 libhide_titlebar_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libhide_titlebar_la_LIBADD = $(EOG_LIBS)
+libhide_titlebar_la_LIBADD = $(EOG_LIBS) -leog
 
 # Plugin Info
 
--- ./plugins/light-theme/Makefile.am.orig	2014-08-09 20:39:08.000000000 +0200
+++ ./plugins/light-theme/Makefile.am	2015-03-25 18:12:11.461204300 +0100
@@ -15,9 +15,9 @@
 	eog-light-theme-plugin.c
 
 liblight_theme_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-liblight_theme_la_LIBADD = $(EOG_LIBS)
+liblight_theme_la_LIBADD = $(EOG_LIBS) -leog
 
 # Plugin Info
 
--- ./plugins/map/Makefile.am.orig	2014-08-09 20:39:08.000000000 +0200
+++ ./plugins/map/Makefile.am	2015-03-25 18:12:11.474185900 +0100
@@ -18,7 +18,9 @@
 libmap_la_LDFLAGS = \
 	$(PLUGIN_LIBTOOL_FLAGS)				\
 	$(EOG_LIBS) \
-	$(CHAMPLAIN_LIBS)
+	$(CHAMPLAIN_LIBS) \
+	-no-undefined \
+	-leog
 
 # Plugin Info
 
--- ./plugins/postasa/Makefile.am.orig	2014-08-09 20:39:08.000000000 +0200
+++ ./plugins/postasa/Makefile.am	2015-03-25 18:12:11.532189200 +0100
@@ -24,10 +24,12 @@
 
 libpostasa_la_LDFLAGS = \
 	$(PLUGIN_LIBTOOL_FLAGS)				\
-	$(EOG_LIBS)
+	$(EOG_LIBS) \
+	-no-undefined
 
 libpostasa_la_LIBADD = $(EOG_LIBS)			\
-	$(POSTASA_LIBS)
+	$(POSTASA_LIBS) \
+	-leog
 
 eog-postasa-plugin-resources.c: eog-postasa-plugin.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/eog-postasa-plugin.gresource.xml)
 	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --internal --target=$@ --sourcedir=$(srcdir) --generate-source $(srcdir)/eog-postasa-plugin.gresource.xml
--- ./plugins/postr/Makefile.am.orig	2015-02-15 14:53:30.000000000 +0100
+++ ./plugins/postr/Makefile.am	2015-03-25 18:12:11.563210600 +0100
@@ -15,9 +15,10 @@
 	eog-postr-plugin.c
 
 libpostr_la_LDFLAGS = \
-	$(PLUGIN_LIBTOOL_FLAGS)
+	$(PLUGIN_LIBTOOL_FLAGS) \
+	-no-undefined
 
-libpostr_la_LIBADD = $(EOG_LIBS)
+libpostr_la_LIBADD = $(EOG_LIBS) -leog
 
 
 # Plugin Info
--- ./plugins/send-by-mail/Makefile.am.orig	2014-08-09 20:39:08.000000000 +0200
+++ ./plugins/send-by-mail/Makefile.am	2015-03-25 18:12:11.593212500 +0100
@@ -15,9 +15,9 @@
 	eog-send-by-mail-plugin.c
 
 libsend_by_mail_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libsend_by_mail_la_LIBADD = $(EOG_LIBS)
+libsend_by_mail_la_LIBADD = $(EOG_LIBS) -leog
 
 # Plugin Info
 
