--- ./plugins/map/eog-map-plugin.h.orig	2014-10-19 19:10:03.188380100 +0200
+++ ./plugins/map/eog-map-plugin.h	2014-10-19 19:12:03.720793800 +0200
@@ -4,11 +4,10 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#include <champlain/champlain.h>
-#include <eog/eog-list-store.h>
 #include <eog/eog-window.h>
 #include <libpeas/peas-extension-base.h>
 #include <libpeas/peas-object-module.h>
+#include <champlain/champlain.h>
 
 G_BEGIN_DECLS
 
