--- a/qtbase/mkspecs/win32-g++/qmake.conf
+++ b/qtbase/mkspecs/win32-g++/qmake.conf
@@ -13,12 +13,12 @@
 # modifications to g++-win32.conf
 
 QMAKE_CC                = $${CROSS_COMPILE}gcc
-QMAKE_CFLAGS           += -fno-keep-inline-dllexport
-QMAKE_CFLAGS_WARN_ON   += -Wextra
+QMAKE_CFLAGS           +=
+QMAKE_CFLAGS_WARN_ON   += -Wextra -Wno-deprecated-declarations -Wno-implicit-fallthrough -Wno-unused-parameter
 
 QMAKE_CXX               = $${CROSS_COMPILE}g++
-QMAKE_CXXFLAGS         += -fno-keep-inline-dllexport
-QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON
+QMAKE_CXXFLAGS         += -Wa,-mbig-obj
+QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON -Wno-deprecated-copy
 
 QMAKE_LINK              = $${CROSS_COMPILE}g++
 QMAKE_LINK_C            = $${CROSS_COMPILE}gcc
--- a/qtbase/mkspecs/win32-clang-g++/qmake.conf
+++ b/qtbase/mkspecs/win32-clang-g++/qmake.conf
@@ -15,11 +15,11 @@
 
 QMAKE_CC                = $${CROSS_COMPILE}clang
 QMAKE_CFLAGS           +=
-QMAKE_CFLAGS_WARN_ON   += -Wextra -Wno-ignored-attributes
+QMAKE_CFLAGS_WARN_ON   += -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wno-missing-exception-spec -Wno-unused-but-set-variable
 
 QMAKE_CXX               = $${CROSS_COMPILE}clang++
-QMAKE_CXXFLAGS         +=
-QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON
+QMAKE_CXXFLAGS         += -Wa,-mbig-obj
+QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON -Wno-deprecated-copy
 
 QMAKE_LINK              = $${CROSS_COMPILE}clang++
 QMAKE_LINK_C            = $${CROSS_COMPILE}clang
