diff --git a/qtbase/src/corelib/tools/qregularexpression.cpp b/qtbase/src/corelib/tools/qregularexpression.cpp
index bdaa2d3243..90f8250c03 100644
--- a/qtbase/src/corelib/tools/qregularexpression.cpp
+++ b/qtbase/src/corelib/tools/qregularexpression.cpp
@@ -53,6 +53,10 @@
 #include <QtCore/qdatastream.h>
 
 #define PCRE2_CODE_UNIT_WIDTH 16
+#ifdef QT_STATIC
+#define PCRE_STATIC
+#define PCRE2_STATIC
+#endif
 
 #include <pcre2.h>
 
--- x86_64/qtbase/src/network/configure.json.orig	2017-01-26 11:44:31.552004100 +0300
+++ x86_64/qtbase/src/network/configure.json	2017-01-26 11:45:32.569724600 +0300
@@ -91,7 +91,7 @@
                     "condition": "config.msvc"
                 },
                 {
-                    "libs": "-lssl -lcrypto",
+                    "libs": "-lssl -lcrypto -lws2_32 -lcrypt32 -lgdi32",
                     "condition": "!config.msvc"
                 }
             ]