--- a/qtconnectivity/src/bluetooth/qbluetoothservicediscoveryagent_win.cpp
+++ b/qtconnectivity/src/bluetooth/qbluetoothservicediscoveryagent_win.cpp
@@ -206,12 +206,7 @@
     return sequence;
 }

-#if defined(Q_CC_MINGW)
-# define SDP_CALLBACK
-#else
-# define SDP_CALLBACK QT_WIN_CALLBACK
-#endif
-static BOOL SDP_CALLBACK bluetoothSdpCallback(ULONG attributeId, LPBYTE valueStream, ULONG streamSize, LPVOID param)
+static BOOL QT_WIN_CALLBACK bluetoothSdpCallback(ULONG attributeId, LPBYTE valueStream, ULONG streamSize, LPVOID param)
 {
     QBluetoothServiceInfo *result = static_cast<QBluetoothServiceInfo*>(param);
 
