diff -urN --ignore-all-space a/qtbase/src/angle/src/compiler/preprocessor/preprocessor.pro b/qtbase/src/angle/src/compiler/preprocessor/preprocessor.pro
--- a/qtbase/src/angle/src/compiler/preprocessor/preprocessor.pro	2014-05-02 23:06:13.969569900 +0100
+++ b/qtbase/src/angle/src/compiler/preprocessor/preprocessor.pro	2014-05-02 23:06:33.470685300 +0100
@@ -41,7 +41,7 @@
     $$ANGLE_DIR/src/compiler/preprocessor/Token.cpp
 
 # NOTE: 'win_flex' and 'bison' can be found in qt5/gnuwin32/bin
-flex.commands = $$addGnuPath(win_flex) --noline --nounistd --outfile=${QMAKE_FILE_BASE}.cpp ${QMAKE_FILE_NAME}
+flex.commands = $$addGnuPath(flex) --noline --nounistd --outfile=${QMAKE_FILE_BASE}.cpp ${QMAKE_FILE_NAME}
 flex.output = ${QMAKE_FILE_BASE}.cpp
 flex.input = FLEX_SOURCES
 flex.dependency_type = TYPE_C
diff -urN --ignore-all-space a/qtbase/src/angle/src/compiler/translator.pro b/qtbase/src/angle/src/compiler/translator.pro
--- a/qtbase/src/angle/src/compiler/translator.pro	2014-05-02 23:06:13.969569900 +0100
+++ b/qtbase/src/angle/src/compiler/translator.pro	2014-05-02 23:06:33.467685100 +0100
@@ -132,7 +132,7 @@
 }
 
 # NOTE: 'win_flex' and 'bison' can be found in qt5/gnuwin32/bin
-flex.commands = $$addGnuPath(win_flex) --noline --nounistd --outfile=${QMAKE_FILE_BASE}_lex.cpp ${QMAKE_FILE_NAME}
+flex.commands = $$addGnuPath(flex) --noline --nounistd --outfile=${QMAKE_FILE_BASE}_lex.cpp ${QMAKE_FILE_NAME}
 flex.output = ${QMAKE_FILE_BASE}_lex.cpp
 flex.input = FLEX_SOURCES
 flex.dependency_type = TYPE_C
diff -urN --ignore-all-space a/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf b/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf
--- a/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf	2014-05-02 23:06:28.221385000 +0100
+++ b/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf	2014-05-02 23:06:33.474685500 +0100
@@ -106,7 +106,6 @@
 # A newer version of flex is required on Windows. At the moment the only
 # one that appears to provide binaries and is not cygwin is winflex.
 FLEX = flex
-win32: FLEX = win_flex
 
 BIN_EXTENSION =
 win32: BIN_EXTENSION = .exe
diff -urN --ignore-all-space a/qtwebkit/Tools/Scripts/webkitdirs.pm b/qtwebkit/Tools/Scripts/webkitdirs.pm
--- a/qtwebkit/Tools/Scripts/webkitdirs.pm	2014-05-02 23:06:28.275388100 +0100
+++ b/qtwebkit/Tools/Scripts/webkitdirs.pm	2014-05-02 23:06:33.480685800 +0100
@@ -1496,11 +1496,7 @@
         }
     } elsif (isGtk() or isQt() or isEfl()) {
         my @cmds = qw(bison gperf);
-        if (isQt() and isWindows()) {
-            push @cmds, "win_flex";
-        } else {
-            push @cmds, "flex";
-        }
+        push @cmds, "flex";
         my @missing = ();
         my $oldPath = $ENV{PATH};
         if (isQt() and isWindows()) {
