--- a/qtbase/src/network/network.pro
+++ b/qtbase/src/network/network.pro
@@ -1,5 +1,6 @@
 TARGET   = QtNetwork
 QT = core-private
+CONFIG += metatypes install_metatypes
 
 DEFINES += QT_NO_USING_NAMESPACE QT_NO_FOREACH
 #DEFINES += QLOCALSERVER_DEBUG QLOCALSOCKET_DEBUG
--- a/qtbase/src/sql/sql.pro
+++ b/qtbase/src/sql/sql.pro
@@ -1,5 +1,6 @@
 TARGET	   = QtSql
 QT         = core-private
+CONFIG += metatypes install_metatypes
 
 DEFINES += QT_NO_USING_NAMESPACE
 msvc:equals(QT_ARCH, i386): QMAKE_LFLAGS += /BASE:0x62000000
--- a/qtbase/src/testlib/testlib.pro
+++ b/qtbase/src/testlib/testlib.pro
@@ -1,6 +1,6 @@
 TARGET = QtTest
 QT = core-private
-CONFIG += exceptions
+CONFIG += exceptions metatypes install_metatypes
 
 MODULE_CONFIG = console testlib_defines
 
