--- a/qtbase/mkspecs/common/g++-win32.conf
+++ b/qtbase/mkspecs/common/g++-win32.conf
@@ -45,7 +45,7 @@
 QMAKE_LFLAGS_CONSOLE    = -Wl,-subsystem,console
 QMAKE_LFLAGS_WINDOWS    = -Wl,-subsystem,windows
 QMAKE_LFLAGS_DLL        = -shared
-QMAKE_LFLAGS_STATIC_LIB = -static
+QMAKE_LFLAGS_STATIC_LIB = -static -static-libgcc -static-libstdc++
 QMAKE_LFLAGS_GCSECTIONS = -Wl,--gc-sections
 equals(QMAKE_HOST.os, Windows) {
     QMAKE_LINK_OBJECT_MAX = 10
--- a/qtbase/mkspecs/win32-g++/qmake.conf
+++ b/qtbase/mkspecs/win32-g++/qmake.conf
@@ -13,12 +13,12 @@
 # modifications to g++-win32.conf
 
 QMAKE_CC                = $${CROSS_COMPILE}gcc
-QMAKE_CFLAGS           += -fno-keep-inline-dllexport
-QMAKE_CFLAGS_WARN_ON   += -Wextra
+QMAKE_CFLAGS           +=
+QMAKE_CFLAGS_WARN_ON   += -Wextra -Wno-deprecated-declarations -Wno-implicit-fallthrough -Wno-unused-parameter
 
 QMAKE_CXX               = $${CROSS_COMPILE}g++
-QMAKE_CXXFLAGS         += -fno-keep-inline-dllexport
-QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON
+QMAKE_CXXFLAGS         += -Wa,-mbig-obj
+QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON -Wno-class-memaccess
 
 QMAKE_LINK              = $${CROSS_COMPILE}g++
 QMAKE_LINK_C            = $${CROSS_COMPILE}gcc
--- a/qtbase/mkspecs/win32-clang-g++/qmake.conf
+++ b/qtbase/mkspecs/win32-clang-g++/qmake.conf
@@ -15,11 +15,11 @@
 
 QMAKE_CC                = $${CROSS_COMPILE}clang
 QMAKE_CFLAGS           +=
-QMAKE_CFLAGS_WARN_ON   += -Wextra -Wno-ignored-attributes
+QMAKE_CFLAGS_WARN_ON   += -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wno-missing-exception-spec -Wno-unused-but-set-variable
 
 QMAKE_CXX               = $${CROSS_COMPILE}clang++
-QMAKE_CXXFLAGS         +=
-QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON
+QMAKE_CXXFLAGS         += -Wa,-mbig-obj
+QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON -Wno-deprecated-copy
 
 QMAKE_LINK              = $${CROSS_COMPILE}clang++
 QMAKE_LINK_C            = $${CROSS_COMPILE}clang
--- a/qttools/src/assistant/qcollectiongenerator/qcollectiongenerator.pro
+++ b/qttools/src/assistant/qcollectiongenerator/qcollectiongenerator.pro
@@ -2,6 +2,8 @@
 CONFIG -= qt app_bundle
 SOURCES += main.c
 
+win32-g++:QMAKE_CFLAGS_WARN_ON += -Wno-incompatible-pointer-types
+
 QMAKE_TARGET_DESCRIPTION = "Qt Help Collection File Generator"
 load(qt_tool)
 
