--- a/qtwebkit/Tools/qmake/mkspecs/features/win32/default_pre.prf.orig	2012-11-12 21:19:24.500379251 +0100
+++ b/qtwebkit/Tools/qmake/mkspecs/features/win32/default_pre.prf	2012-11-12 21:19:50.901564454 +0100
@@ -4,6 +4,9 @@
 # See 'Tools/qmake/README' for an overview of the build system
 # -------------------------------------------------------------------
 
+# Make sure we process the PKGCONFIG variable if applicable
+CONFIG += link_pkgconfig
+
 load(default_pre)
 
 debug: CONFIG += use_all_in_one_files
