diff -Naur qt-everywhere-src-5.12.4-orig/qtdeclarative/examples/qml/tutorials/extending-qml/chapter6-plugins/import/import.pro qt-everywhere-src-5.12.4/qtdeclarative/examples/qml/tutorials/extending-qml/chapter6-plugins/import/import.pro
--- qt-everywhere-src-5.12.4-orig/qtdeclarative/examples/qml/tutorials/extending-qml/chapter6-plugins/import/import.pro	2019-05-28 15:19:15.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtdeclarative/examples/qml/tutorials/extending-qml/chapter6-plugins/import/import.pro	2019-06-15 15:16:50.399074600 +0300
@@ -2,7 +2,7 @@
 CONFIG += plugin
 QT += qml quick
 
-DESTDIR = ../Charts
+DESTDIR = ../Charts/
 TARGET = $$qtLibraryTarget(chartsplugin)
 
 HEADERS += piechart.h \
diff -Naur qt-everywhere-src-5.12.4-orig/qtsensors/examples/sensors/grue/import/import.pro qt-everywhere-src-5.12.4/qtsensors/examples/sensors/grue/import/import.pro
--- qt-everywhere-src-5.12.4-orig/qtsensors/examples/sensors/grue/import/import.pro	2019-05-23 15:20:42.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtsensors/examples/sensors/grue/import/import.pro	2019-06-15 15:16:50.399074600 +0300
@@ -4,7 +4,7 @@
 TARGET = $$qtLibraryTarget(declarative_grue)
 
 macos: DESTDIR = ../grue_app.app/Contents/MacOS/Grue
-else: DESTDIR = ../Grue
+else: DESTDIR = ../Grue/
 
 QT = core gui qml sensors
 
diff -Naur qt-everywhere-src-5.12.4-orig/qtsensors/examples/sensors/sensor_explorer/import/import.pro qt-everywhere-src-5.12.4/qtsensors/examples/sensors/sensor_explorer/import/import.pro
--- qt-everywhere-src-5.12.4-orig/qtsensors/examples/sensors/sensor_explorer/import/import.pro	2019-05-23 15:20:42.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtsensors/examples/sensors/sensor_explorer/import/import.pro	2019-06-15 15:16:50.414674600 +0300
@@ -4,7 +4,7 @@
 TARGET = $$qtLibraryTarget(declarative_explorer)
 
 macos: DESTDIR = ../sensor_explorer.app/Contents/MacOS/Explorer
-else: DESTDIR = ../Explorer
+else: DESTDIR = ../Explorer/
 
 QT += qml sensors
 
