--- a/qtwebkit/Source/WebCore/WebCore.pri.orig	2013-01-05 01:39:51.909266162 +0100
+++ b/qtwebkit/Source/WebCore/WebCore.pri	2013-01-05 01:39:55.803294288 +0100
@@ -210,7 +210,7 @@
 
 use?(3D_GRAPHICS) {
     win32: {
-        win32-g++: {
+        win32-g++*: {
             # Make sure OpenGL libs are after the webcore lib so MinGW can resolve symbols
             contains(QT_CONFIG, opengles2) {
                 LIBS += $$QMAKE_LIBS_OPENGL_ES2
