--- a/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in
+++ b/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in
@@ -2,9 +2,9 @@
 !!IF !isEmpty(CMAKE_ANGLE_EGL_DLL_RELEASE)
 
 !!IF isEmpty(CMAKE_INCLUDE_DIR_IS_ABSOLUTE)
-set(Qt5Gui_EGL_INCLUDE_DIRS \"${_qt5$${CMAKE_MODULE_NAME}_install_prefix}/$$CMAKE_INCLUDE_DIR/QtANGLE\")
+set(Qt5Gui_EGL_INCLUDE_DIRS \"${_qt5$${CMAKE_MODULE_NAME}_install_prefix}/$$CMAKE_INCLUDE_DIR\")
 !!ELSE
-set(Qt5Gui_EGL_INCLUDE_DIRS \"$$CMAKE_INCLUDE_DIR/QtANGLE\")
+set(Qt5Gui_EGL_INCLUDE_DIRS \"$$CMAKE_INCLUDE_DIR\")
 !!ENDIF
 
 _qt5_Gui_check_file_exists(${Qt5Gui_EGL_INCLUDE_DIRS})
--- a/qtbase/src/gui/gui.pro
+++ b/qtbase/src/gui/gui.pro
@@ -19,12 +19,6 @@
     imageformats \
     egldeviceintegrations
 
-# This is here only because the platform plugin is no module, obviously.
-qtConfig(angle) {
-    MODULE_AUX_INCLUDES = \
-        \$\$QT_MODULE_INCLUDE_BASE/QtANGLE
-}
-
 # Code coverage with TestCocoon
 # The following is required as extra compilers use $$QMAKE_CXX instead of $(CXX).
 # Without this, testcocoon.prf is read only after $$QMAKE_CXX is used by the
@@ -36,6 +30,8 @@
 osx: LIBS_PRIVATE += -framework AppKit
 darwin: LIBS_PRIVATE += -framework CoreGraphics
 
+win32:qtConfig(angle): LIBS_PRIVATE += -lGLESv2
+
 CONFIG += simd optimize_full
 
 include(accessible/accessible.pri)
--- a/qtbase/src/opengl/opengl.pro
+++ b/qtbase/src/opengl/opengl.pro
@@ -10,6 +10,7 @@
 
 qtConfig(opengl): CONFIG += opengl
 qtConfig(opengles2): CONFIG += opengles2
+win32:qtConfig(angle): LIBS_PRIVATE += -lGLESv2
 
 HEADERS += qgl.h \
            qgl_p.h \
--- a/qtbase/src/plugins/platforms/windows/windows.pri
+++ b/qtbase/src/plugins/platforms/windows/windows.pri
@@ -4,6 +4,7 @@
 QT_FOR_CONFIG += gui
 
 qtConfig(opengl):!qtConfig(opengles2):!qtConfig(dynamicgl): LIBS *= -lopengl32
+qtConfig(angle):LIBS += -lGLESv2 -lEGL
 
 mingw: QMAKE_USE *= uuid
 # For the dialog helpers:
--- a/qtbase/src/src.pro
+++ b/qtbase/src/src.pro
@@ -203,10 +203,6 @@
         SUBDIRS += src_3rdparty_harfbuzzng
         src_gui.depends += src_3rdparty_harfbuzzng
     }
-    qtConfig(angle) {
-        SUBDIRS += src_angle
-        src_gui.depends += src_angle
-    }
     qtConfig(jpeg):!qtConfig(system-jpeg) {
         SUBDIRS += src_3rdparty_libjpeg
         src_plugins.depends = src_3rdparty_libjpeg
