--- x86_64/qtbase/src/corelib/tools/tools.pri.orig	2014-04-20 21:21:48.808906400 +0100
+++ x86_64/qtbase/src/corelib/tools/tools.pri	2014-04-20 21:20:41.677381800 +0100
@@ -159,6 +159,7 @@
         CONFIG(static, static|shared) {
             CONFIG(debug, debug|release) {
                 LIBS_PRIVATE += -lsicuind -lsicuucd -lsicudtd
+                DEFINES += "U_LIB_SUFFIX_C_NAME=d"
             } else {
                 LIBS_PRIVATE += -lsicuin -lsicuuc -lsicudt
             }
--- x86_64/qtbase/config.tests/unix/icu/icu.pro.orig	2014-04-20 21:24:31.107015700 +0100
+++ x86_64/qtbase/config.tests/unix/icu/icu.pro	2014-04-20 21:24:25.573813000 +0100
@@ -5,6 +5,7 @@
     CONFIG(static, static|shared) {
         CONFIG(debug, debug|release) {
             LIBS += -lsicuind -lsicuucd -lsicudtd
+            DEFINES += "U_LIB_SUFFIX_C_NAME=d"
         } else {
             LIBS += -lsicuin -lsicuuc -lsicudt
         }
