--- a/qtbase/mkspecs/features/create_cmake.prf	2014-09-11 11:48:07.000000000 +0100
+++ b/qtbase/mkspecs/features/create_cmake.prf	2014-10-13 17:14:33.345108800 +0100
@@ -134,7 +134,7 @@
 
     win32 {
         isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .dll
-        else: CMAKE_PlUGIN_EXT = .lib
+        else: CMAKE_PlUGIN_EXT = .$${QMAKE_EXTENSION_STATICLIB}
 
         CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/$${TARGET}$${CMAKE_PlUGIN_EXT}
         CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/$${TARGET}d$${CMAKE_PlUGIN_EXT}
