diff -Naur qt-everywhere-src-5.12.4-orig/qtbase/qmake/generators/win32/mingw_make.cpp qt-everywhere-src-5.12.4/qtbase/qmake/generators/win32/mingw_make.cpp
--- qt-everywhere-src-5.12.4-orig/qtbase/qmake/generators/win32/mingw_make.cpp	2019-06-12 23:59:14.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtbase/qmake/generators/win32/mingw_make.cpp	2019-06-15 15:34:10.657701800 +0300
@@ -209,7 +209,7 @@
 
     processVars();
 
-    project->values("LIBS") += project->values("RES_FILE");
+    project->values("OBJECTS") += project->values("RES_FILE");
 
     if (project->isActiveConfig("dll")) {
         QString destDir = "";
