--- i686/qtbase/mkspecs/features/create_cmake.prf.orig	2014-10-24 23:47:00.282911600 +0100
+++ i686/qtbase/mkspecs/features/create_cmake.prf	2014-10-24 23:52:53.666785700 +0100
@@ -175,7 +175,15 @@
 
 CMAKE_MKSPEC = $$[QMAKE_XSPEC]
 
-CMAKE_MODULE_DEPS = $$cmakeModuleList($$sort_depends(QT.$${MODULE}.depends, QT.))
+# Without adding the module itself as a dependency, qt5-static doesn't link ws2_32
+# when creating executables.  The result of doing this is that Qt5CoreConfig.cmake
+# ends up with line:
+# set(_Qt5Core_LIB_DEPENDENCIES "Qt5::Core")
+# instead of:
+# set(_Qt5Core_LIB_DEPENDENCIES "")
+SELF_AND_DEPS = $${MODULE}
+SELF_AND_DEPS += QT.$${MODULE}.depends
+CMAKE_MODULE_DEPS = $$cmakeModuleList($$sort_depends(SELF_AND_DEPS, QT.))
 CMAKE_PARTIAL_MODULE_DEPS = $$replace(CMAKE_MODULE_DEPS, ";", ";Qt5::")
 !isEmpty(CMAKE_PARTIAL_MODULE_DEPS):CMAKE_QT5_MODULE_DEPS = "Qt5::$${CMAKE_PARTIAL_MODULE_DEPS}"
 
