--- qt-everywhere-opensource-src-5.9.1.orig/qtlocation/src/plugins/geoservices/geoservices.pro	2017-06-07 19:05:58.022981900 +0200
+++ qt-everywhere-opensource-src-5.9.1/qtlocation/src/plugins/geoservices/geoservices.pro	2017-06-07 19:00:24.921062300 +0200
@@ -7,7 +7,7 @@
 qtConfig(geoservices_itemsoverlay): SUBDIRS += itemsoverlay
 qtConfig(geoservices_osm): SUBDIRS += osm
 
-qtConfig(geoservices_mapboxgl) {
+qtConfig(geoservices_mapboxgl):!mingw {
     !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
         warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
     } else {
