--- qt-everywhere-opensource-src-5.9.2.orig/qtlocation/src/plugins/geoservices/geoservices.pro.orig	2017-10-10 17:13:30.572784400 +0200
+++ qt-everywhere-opensource-src-5.9.2/qtlocation/src/plugins/geoservices/geoservices.pro	2017-10-10 17:41:09.230569000 +0200
@@ -8,11 +8,11 @@
 qtConfig(geoservices_itemsoverlay): SUBDIRS += itemsoverlay
 qtConfig(geoservices_osm): SUBDIRS += osm
 
-qtConfig(geoservices_mapboxgl) {
-    !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
-        warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
-    } else {
-        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
-        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
-    }
-}
+#qtConfig(geoservices_mapboxgl) {
+#    !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
+#        warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
+#    } else {
+#        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
+#        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
+#    }
+#}
