--- a/qtlocation/src/plugins/geoservices/geoservices.pro
+++ b/qtlocation/src/plugins/geoservices/geoservices.pro
@@ -8,7 +8,7 @@
 qtConfig(geoservices_itemsoverlay): SUBDIRS += itemsoverlay
 qtConfig(geoservices_osm): SUBDIRS += osm
 
-qtConfig(geoservices_mapboxgl) {
+qtConfig(geoservices_mapboxgl):!mingw {
     !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
         warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
     } else {
