--- qt-everywhere-src-5.11.2/qttools/mkspecs/features/qt_find_clang.prf.orig	2018-09-03 18:13:05.000000000 +0300
+++ qt-everywhere-src-5.11.2/qttools/mkspecs/features/qt_find_clang.prf	2018-10-08 14:26:00.160671900 +0300
@@ -70,16 +70,152 @@
     }
 
     # note: llvm_config only exits on unix
-    llvm_config = $$clangInstallDir/bin/llvm-config
+    llvm_config = $$clangInstallDir/bin/llvm-config.exe
     exists($$llvm_config) {
-        CLANG_LIBDIR = $$system("$$llvm_config --libdir 2>/dev/null")
-        CLANG_INCLUDEPATH = $$system("$$llvm_config --includedir 2>/dev/null")
-        output = $$system("$$llvm_config --version 2>/dev/null")
+        CLANG_LIBDIR = $$system("$$llvm_config --libdir")
+        CLANG_INCLUDEPATH = $$system("$$llvm_config --includedir")
+        output = $$system("$$llvm_config --version")
         CLANG_VERSION = $$extractVersion($$output)
+        LLVM_COMPONENTS = $$system("$$llvm_config --libs")
+        LLVM_SYSTEM_LIBS = $$system("$$llvm_config --system-libs")
     } else {
         CLANG_LIBDIR = $$clangInstallDir/lib
         CLANG_INCLUDEPATH = $$clangInstallDir/include
         CLANG_VERSION = $$findLLVMVersionFromLibDir($$CLANG_LIBDIR)
+        LLVM_COMPONENTS =  -lLLVMAArch64AsmParser \
+                           -lLLVMAArch64AsmPrinter \
+                           -lLLVMAArch64CodeGen \
+                           -lLLVMAArch64Desc \
+                           -lLLVMAArch64Disassembler \
+                           -lLLVMAArch64Info \
+                           -lLLVMAArch64Utils \
+                           -lLLVMAMDGPUAsmParser \
+                           -lLLVMAMDGPUAsmPrinter \
+                           -lLLVMAMDGPUCodeGen \
+                           -lLLVMAMDGPUDesc \
+                           -lLLVMAMDGPUDisassembler \
+                           -lLLVMAMDGPUInfo \
+                           -lLLVMAMDGPUUtils \
+                           -lLLVMAnalysis \
+                           -lLLVMARMAsmParser \
+                           -lLLVMARMAsmPrinter \
+                           -lLLVMARMCodeGen \
+                           -lLLVMARMDesc \
+                           -lLLVMARMDisassembler \
+                           -lLLVMARMInfo \
+                           -lLLVMARMUtils \
+                           -lLLVMAsmParser \
+                           -lLLVMAsmPrinter \
+                           -lLLVMBinaryFormat \
+                           -lLLVMBitReader \
+                           -lLLVMBitWriter \
+                           -lLLVMBPFAsmParser \
+                           -lLLVMBPFAsmPrinter \
+                           -lLLVMBPFCodeGen \
+                           -lLLVMBPFDesc \
+                           -lLLVMBPFDisassembler \
+                           -lLLVMBPFInfo \
+                           -lLLVMCodeGen \
+                           -lLLVMCore \
+                           -lLLVMCoroutines \
+                           -lLLVMCoverage \
+                           -lLLVMDebugInfoCodeView \
+                           -lLLVMDebugInfoDWARF \
+                           -lLLVMDebugInfoMSF \
+                           -lLLVMDebugInfoPDB \
+                           -lLLVMDemangle \
+                           -lLLVMDlltoolDriver \
+                           -lLLVMExecutionEngine \
+                           -lLLVMFuzzMutate \
+                           -lLLVMGlobalISel \
+                           -lLLVMHexagonAsmParser \
+                           -lLLVMHexagonCodeGen \
+                           -lLLVMHexagonDesc \
+                           -lLLVMHexagonDisassembler \
+                           -lLLVMHexagonInfo \
+                           -lLLVMInstCombine \
+                           -lLLVMInstrumentation \
+                           -lLLVMInterpreter \
+                           -lLLVMipo \
+                           -lLLVMIRReader \
+                           -lLLVMLanaiAsmParser \
+                           -lLLVMLanaiAsmPrinter \
+                           -lLLVMLanaiCodeGen \
+                           -lLLVMLanaiDesc \
+                           -lLLVMLanaiDisassembler \
+                           -lLLVMLanaiInfo \
+                           -lLLVMLibDriver \
+                           -lLLVMLineEditor \
+                           -lLLVMLinker \
+                           -lLLVMLTO \
+                           -lLLVMMC \
+                           -lLLVMMCDisassembler \
+                           -lLLVMMCJIT \
+                           -lLLVMMCParser \
+                           -lLLVMMipsAsmParser \
+                           -lLLVMMipsAsmPrinter \
+                           -lLLVMMipsCodeGen \
+                           -lLLVMMipsDesc \
+                           -lLLVMMipsDisassembler \
+                           -lLLVMMipsInfo \
+                           -lLLVMMIRParser \
+                           -lLLVMMSP430AsmPrinter \
+                           -lLLVMMSP430CodeGen \
+                           -lLLVMMSP430Desc \
+                           -lLLVMMSP430Info \
+                           -lLLVMNVPTXAsmPrinter \
+                           -lLLVMNVPTXCodeGen \
+                           -lLLVMNVPTXDesc \
+                           -lLLVMNVPTXInfo \
+                           -lLLVMObjCARCOpts \
+                           -lLLVMObject \
+                           -lLLVMObjectYAML \
+                           -lLLVMOption \
+                           -lLLVMOrcJIT \
+                           -lLLVMPasses \
+                           -lLLVMPowerPCAsmParser \
+                           -lLLVMPowerPCAsmPrinter \
+                           -lLLVMPowerPCCodeGen \
+                           -lLLVMPowerPCDesc \
+                           -lLLVMPowerPCDisassembler \
+                           -lLLVMPowerPCInfo \
+                           -lLLVMProfileData \
+                           -lLLVMRuntimeDyld \
+                           -lLLVMScalarOpts \
+                           -lLLVMSelectionDAG \
+                           -lLLVMSparcAsmParser \
+                           -lLLVMSparcAsmPrinter \
+                           -lLLVMSparcCodeGen \
+                           -lLLVMSparcDesc \
+                           -lLLVMSparcDisassembler \
+                           -lLLVMSparcInfo \
+                           -lLLVMSupport \
+                           -lLLVMSymbolize \
+                           -lLLVMSystemZAsmParser \
+                           -lLLVMSystemZAsmPrinter \
+                           -lLLVMSystemZCodeGen \
+                           -lLLVMSystemZDesc \
+                           -lLLVMSystemZDisassembler \
+                           -lLLVMSystemZInfo \
+                           -lLLVMTableGen \
+                           -lLLVMTarget \
+                           -lLLVMTransformUtils \
+                           -lLLVMVectorize \
+                           -lLLVMWindowsManifest \
+                           -lLLVMX86AsmParser \
+                           -lLLVMX86AsmPrinter \
+                           -lLLVMX86CodeGen \
+                           -lLLVMX86Desc \
+                           -lLLVMX86Disassembler \
+                           -lLLVMX86Info \
+                           -lLLVMX86Utils \
+                           -lLLVMXCoreAsmPrinter \
+                           -lLLVMXCoreCodeGen \
+                           -lLLVMXCoreDesc \
+                           -lLLVMXCoreDisassembler \
+                           -lLLVMXCoreInfo \
+                           -lLLVMXRay
+        LLVM_SYSTEM_LIBS = -lz -lpsapi -lshell32 -lole32 -luuid -ladvapi32
     }
     isEmpty(CLANG_VERSION) {
         !isEmpty(LLVM_INSTALL_DIR): \
@@ -102,7 +238,7 @@
 
     isEmpty(QDOC_USE_STATIC_LIBCLANG) {
         equals(QMAKE_HOST.os, Windows): \
-            CLANG_LIBS += -llibclang -ladvapi32 -lshell32
+            CLANG_LIBS += -lclang $${LLVM_SYSTEM_LIBS}
         else: \
             CLANG_LIBS += -lclang
     } else {
@@ -145,6 +281,7 @@
                         -lclangStaticAnalyzerCore \
                         -lclangStaticAnalyzerFrontend \
                         -lclangTidy \
+                        -lclangTidyAbseilModule \
                         -lclangTidyAndroidModule \
                         -lclangTidyBoostModule \
                         -lclangTidyBugproneModule \
@@ -160,149 +297,20 @@
                         -lclangTidyObjCModule \
                         -lclangTidyPerformanceModule \
                         -lclangTidyPlugin \
+                        -lclangTidyPortabilityModule \
                         -lclangTidyReadabilityModule \
                         -lclangTidyUtils \
+                        -lclangTidyZirconModule \
                         -lclangTooling \
                         -lclangToolingASTDiff \
                         -lclangToolingCore \
+                        -lclangToolingInclusions \
                         -lclangToolingRefactor \
                         -lfindAllSymbols \
-                        -lLLVMAArch64AsmParser \
-                        -lLLVMAArch64AsmPrinter \
-                        -lLLVMAArch64CodeGen \
-                        -lLLVMAArch64Desc \
-                        -lLLVMAArch64Disassembler \
-                        -lLLVMAArch64Info \
-                        -lLLVMAArch64Utils \
-                        -lLLVMAMDGPUAsmParser \
-                        -lLLVMAMDGPUAsmPrinter \
-                        -lLLVMAMDGPUCodeGen \
-                        -lLLVMAMDGPUDesc \
-                        -lLLVMAMDGPUDisassembler \
-                        -lLLVMAMDGPUInfo \
-                        -lLLVMAMDGPUUtils \
-                        -lLLVMAnalysis \
-                        -lLLVMARMAsmParser \
-                        -lLLVMARMAsmPrinter \
-                        -lLLVMARMCodeGen \
-                        -lLLVMARMDesc \
-                        -lLLVMARMDisassembler \
-                        -lLLVMARMInfo \
-                        -lLLVMARMUtils \
-                        -lLLVMAsmParser \
-                        -lLLVMAsmPrinter \
-                        -lLLVMBinaryFormat \
-                        -lLLVMBitReader \
-                        -lLLVMBitWriter \
-                        -lLLVMBPFAsmParser \
-                        -lLLVMBPFAsmPrinter \
-                        -lLLVMBPFCodeGen \
-                        -lLLVMBPFDesc \
-                        -lLLVMBPFDisassembler \
-                        -lLLVMBPFInfo \
-                        -lLLVMCodeGen \
-                        -lLLVMCore \
-                        -lLLVMCoroutines \
-                        -lLLVMCoverage \
-                        -lLLVMDebugInfoCodeView \
-                        -lLLVMDebugInfoDWARF \
-                        -lLLVMDebugInfoMSF \
-                        -lLLVMDebugInfoPDB \
-                        -lLLVMDemangle \
-                        -lLLVMDlltoolDriver \
-                        -lLLVMExecutionEngine \
-                        -lLLVMFuzzMutate \
-                        -lLLVMGlobalISel \
-                        -lLLVMHexagonAsmParser \
-                        -lLLVMHexagonCodeGen \
-                        -lLLVMHexagonDesc \
-                        -lLLVMHexagonDisassembler \
-                        -lLLVMHexagonInfo \
-                        -lLLVMInstCombine \
-                        -lLLVMInstrumentation \
-                        -lLLVMInterpreter \
-                        -lLLVMipo \
-                        -lLLVMIRReader \
-                        -lLLVMLanaiAsmParser \
-                        -lLLVMLanaiAsmPrinter \
-                        -lLLVMLanaiCodeGen \
-                        -lLLVMLanaiDesc \
-                        -lLLVMLanaiDisassembler \
-                        -lLLVMLanaiInfo \
-                        -lLLVMLibDriver \
-                        -lLLVMLineEditor \
-                        -lLLVMLinker \
-                        -lLLVMLTO \
-                        -lLLVMMC \
-                        -lLLVMMCDisassembler \
-                        -lLLVMMCJIT \
-                        -lLLVMMCParser \
-                        -lLLVMMipsAsmParser \
-                        -lLLVMMipsAsmPrinter \
-                        -lLLVMMipsCodeGen \
-                        -lLLVMMipsDesc \
-                        -lLLVMMipsDisassembler \
-                        -lLLVMMipsInfo \
-                        -lLLVMMIRParser \
-                        -lLLVMMSP430AsmPrinter \
-                        -lLLVMMSP430CodeGen \
-                        -lLLVMMSP430Desc \
-                        -lLLVMMSP430Info \
-                        -lLLVMNVPTXAsmPrinter \
-                        -lLLVMNVPTXCodeGen \
-                        -lLLVMNVPTXDesc \
-                        -lLLVMNVPTXInfo \
-                        -lLLVMObjCARCOpts \
-                        -lLLVMObject \
-                        -lLLVMObjectYAML \
-                        -lLLVMOption \
-                        -lLLVMOrcJIT \
-                        -lLLVMPasses \
-                        -lLLVMPowerPCAsmParser \
-                        -lLLVMPowerPCAsmPrinter \
-                        -lLLVMPowerPCCodeGen \
-                        -lLLVMPowerPCDesc \
-                        -lLLVMPowerPCDisassembler \
-                        -lLLVMPowerPCInfo \
-                        -lLLVMProfileData \
-                        -lLLVMRuntimeDyld \
-                        -lLLVMScalarOpts \
-                        -lLLVMSelectionDAG \
-                        -lLLVMSparcAsmParser \
-                        -lLLVMSparcAsmPrinter \
-                        -lLLVMSparcCodeGen \
-                        -lLLVMSparcDesc \
-                        -lLLVMSparcDisassembler \
-                        -lLLVMSparcInfo \
-                        -lLLVMSupport \
-                        -lLLVMSymbolize \
-                        -lLLVMSystemZAsmParser \
-                        -lLLVMSystemZAsmPrinter \
-                        -lLLVMSystemZCodeGen \
-                        -lLLVMSystemZDesc \
-                        -lLLVMSystemZDisassembler \
-                        -lLLVMSystemZInfo \
-                        -lLLVMTableGen \
-                        -lLLVMTarget \
-                        -lLLVMTransformUtils \
-                        -lLLVMVectorize \
-                        -lLLVMWindowsManifest \
-                        -lLLVMX86AsmParser \
-                        -lLLVMX86AsmPrinter \
-                        -lLLVMX86CodeGen \
-                        -lLLVMX86Desc \
-                        -lLLVMX86Disassembler \
-                        -lLLVMX86Info \
-                        -lLLVMX86Utils \
-                        -lLLVMXCoreAsmPrinter \
-                        -lLLVMXCoreCodeGen \
-                        -lLLVMXCoreDesc \
-                        -lLLVMXCoreDisassembler \
-                        -lLLVMXCoreInfo \
-                        -lLLVMXRay
+                        $${LLVM_COMPONENTS}
             !equals(QMAKE_HOST.os, Darwin): CLANG_LIBS+=-Wl,--end-group
             CLANG_LIBS += -lz
-            equals(QMAKE_HOST.os, Windows): CLANG_LIBS += -lpsapi -lshell32 -lole32 -luuid -ladvapi32 -lversion
+            equals(QMAKE_HOST.os, Windows): CLANG_LIBS += -lz3 -lgomp -lpsapi -lshell32 -lole32 -luuid -ladvapi32 -lversion
             else: CLANG_LIBS += -ldl
             equals(QMAKE_HOST.os, Darwin): CLANG_LIBS += -lcurses -lm -lxml2
         }
