--- x86_64/qtwinextras/src/winextras/doc/qtwinextras.qdocconf.orig	2018-04-25 13:28:03.000000000 +0300
+++ x86_64/qtwinextras/src/winextras/doc/qtwinextras.qdocconf	2018-06-16 13:46:08.607496600 +0300
@@ -12,8 +12,7 @@
 moduleheader = QtWinExtrasDoc
 
 includepaths = -I . \
-               -I .. \
-               -I $QT_INSTALL_HEADERS
+               -I .. 
 
 headerdirs = .. ../../imports/winextras
 sourcedirs = .. ../../imports/winextras
diff --git a/mkspecs/features/qt_docs.prf b/mkspecs/features/qt_docs.prf
index 3139c44..7f8b933 100644
--- a/qtbase/mkspecs/features/qt_docs.prf
+++ b/qtbase/mkspecs/features/qt_docs.prf
@@ -39,55 +39,70 @@
 load(qt_build_paths)
 QMAKE_DOCS_BASE_OUTDIR = $$MODULE_BASE_OUTDIR/doc
 
 QMAKE_DOCS_TARGET = $$replace(QMAKE_DOCS, ^(.*/)?(.*)\\.qdocconf$, \\2)
 isEmpty(QMAKE_DOCS_TARGETDIR): QMAKE_DOCS_TARGETDIR = $$QMAKE_DOCS_TARGET
 QMAKE_DOCS_OUTPUTDIR = $$QMAKE_DOCS_BASE_OUTDIR/$$QMAKE_DOCS_TARGETDIR
 
 QDOC += -outputdir $$shell_quote($$QMAKE_DOCS_OUTPUTDIR)
 !build_online_docs: \
     QDOC += -installdir $$shell_quote($$[QT_INSTALL_DOCS])
+
 PREP_DOC_INDEXES =
 DOC_INDEXES =
-!isEmpty(QTREPOS) {
+DOC_INCPATH =
+!isEmpty(QTREPOS) {
+    deps = $$replace(QT, -private$, )
+    deps = $$resolve_depends(deps, "QT.")
     prepare_docs {
         # This is not for linking, but for providing type information.
         mps =
-        deps = $$replace(QT, -private$, )
-        deps = $$resolve_depends(deps, "QT.")
         for (d, deps): \
             mps += $$dirname(QT.$${d}.libs)
         mps = $$unique(mps)
         for (mp, mps): \
             PREP_DOC_INDEXES += -indexdir $$shell_quote($$mp/doc)
     }
     for(qrep, QTREPOS): \
         DOC_INDEXES += -indexdir $$shell_quote($$qrep/doc)
+
+    # Some modules have dummy doc targets with empty INCLUDEPATH
+    # - resolve relevant paths manually.
+    isEmpty(INCLUDEPATH) {
+        incpaths = $$eval(QT.$${MODULE}.includes)
+        for (d, deps): \
+            incpaths += $$eval(QT.$${d}.includes)
+        incpaths = $$unique(incpaths)
+        for (ip, incpaths): \
+            DOC_INCPATH += -I$$shell_quote($$ip)
+    }
 } else {
     prepare_docs: \
         PREP_DOC_INDEXES += -indexdir $$shell_quote($$[QT_INSTALL_DOCS/get])
     DOC_INDEXES += -indexdir $$shell_quote($$[QT_INSTALL_DOCS/get])
+    isEmpty(INCLUDEPATH): \
+        DOC_INCPATH += -I$$shell_quote($$[QT_INSTALL_HEADERS/get])
 }
 
 qtattributionsscanner.target = qtattributionsscanner
 qtattributionsscanner.commands = $$QTATTRIBUTIONSSCANNER $$shell_quote($$MODULE_BASE_INDIR) \
     --filter "QDocModule=$$QMAKE_DOCS_TARGET" -o $$shell_quote($$OUT_PWD/codeattributions.qdoc)
 qtattributionsscanner.CONFIG += phony
 QMAKE_EXTRA_TARGETS += qtattributionsscanner
 
 doc_command = $$QDOC $$QMAKE_DOCS
 prepare_docs {
-    prepare_docs.commands += $$doc_command -prepare $$PREP_DOC_INDEXES -no-link-errors $(INCPATH)
-    generate_docs.commands += $$doc_command -generate $$DOC_INDEXES $(INCPATH)
+    prepare_docs.commands += $$doc_command -prepare $$PREP_DOC_INDEXES -no-link-errors $(INCPATH) $$DOC_INCPATH
+    generate_docs.commands += $$doc_command -generate $$DOC_INDEXES $(INCPATH) $$DOC_INCPATH
     prepare_docs.depends += qtattributionsscanner
 } else {
-    html_docs.commands += $$doc_command $$DOC_INDEXES $(INCPATH)
+    html_docs.commands += $$doc_command $$DOC_INDEXES $(INCPATH) $$DOC_INCPATH
     html_docs.depends += qtattributionsscanner
 }
 
 !build_online_docs {
     qch_docs.commands = $$QHELPGENERATOR $$shell_quote($$QMAKE_DOCS_OUTPUTDIR/$${QMAKE_DOCS_TARGET}.qhp) -o $$shell_quote($$QMAKE_DOCS_BASE_OUTDIR/$${QMAKE_DOCS_TARGET}.qch)
 
     inst_html_docs.files = $$QMAKE_DOCS_OUTPUTDIR
     inst_html_docs.path = $$[QT_INSTALL_DOCS]
     inst_html_docs.CONFIG += no_check_exist directory no_default_install no_build
     INSTALLS += inst_html_docs
--- x86_64/qt3d/src/doc/qt3d.qdocconf.orig	2018-04-25 13:28:03.000000000 +0300
+++ x86_64/qt3d/src/doc/qt3d.qdocconf	2018-06-16 13:46:08.607496600 +0300
@@ -5,14 +5,7 @@
 version                 = $QT_VERSION
 
 moduleheader            = Qt3DDoc
-includepaths            = -I . \
-                          -I $QT_INSTALL_HEADERS \
-                          -I $QT_INSTALL_HEADERS/Qt3DCore \
-                          -I $QT_INSTALL_HEADERS/Qt3DCore/$QT_VERSION \
-                          -I $QT_INSTALL_HEADERS/Qt3DAnimation \
-                          -I $QT_INSTALL_HEADERS/Qt3DAnimation/$QT_VERSION \
-                          -I $QT_INSTALL_HEADERS/Qt3DInput \
-                          -I $QT_INSTALL_HEADERS/Qt3DInput/$QT_VERSION
+includepaths            = -I .
 
 examplesinstallpath     = qt3d
 
--- x86_64/qtactiveqt/src/activeqt/doc/activeqt.qdocconf.orig	2018-04-25 13:28:03.000000000 +0300
+++ x86_64/qtactiveqt/src/activeqt/doc/activeqt.qdocconf	2018-06-16 13:46:08.607496600 +0300
@@ -11,8 +11,7 @@
 includepaths            = -I . \
                           -I ../container \
                           -I ../control \
-                          -I ../shared \
-                          -I $QT_INSTALL_HEADERS
+                          -I ../shared
 
 qhp.ActiveQt.file         = activeqt.qhp
 qhp.ActiveQt.namespace           = org.qt-project.activeqt.$QT_VERSION_TAG
--- x86_64/qtspeech/src/doc/qtspeech.qdocconf.orig	2018-04-25 13:28:03.000000000 +0300
+++ x86_64/qtspeech/src/doc/qtspeech.qdocconf	2018-06-16 13:46:08.607496600 +0300
@@ -14,9 +14,7 @@
 
 moduleheader = QtSpeechDoc
 
-includepaths = -I . \
-               -I $QT_INSTALL_HEADERS \
-               -I $QT_INSTALL_HEADERS/QtCore
+includepaths = -I .
 
 examplesinstallpath = speech
 
