--- x86_64/qtwinextras/src/winextras/doc/qtwinextras.qdocconf.orig	2018-04-25 13:28:03.000000000 +0300
+++ x86_64/qtwinextras/src/winextras/doc/qtwinextras.qdocconf	2018-06-16 13:46:08.607496600 +0300
@@ -12,8 +12,7 @@
 moduleheader = QtWinExtrasDoc
 
 includepaths = -I . \
-               -I .. \
-               -I $QT_INSTALL_HEADERS
+               -I .. 
 
 headerdirs = .. ../../imports/winextras
 sourcedirs = .. ../../imports/winextras
