--- a/qtbase/src/plugins/platformthemes/platformthemes.pro
+++ b/qtbase/src/plugins/platformthemes/platformthemes.pro
@@ -1,6 +1,6 @@
 TEMPLATE = subdirs
 QT_FOR_CONFIG += widgets-private
 
-qtConfig(dbus):qtConfig(regularexpression):qtConfig(mimetype): SUBDIRS += xdgdesktopportal
+qtConfig(dbus):qtConfig(regularexpression):qtConfig(mimetype):!win32: SUBDIRS += xdgdesktopportal
 
 qtHaveModule(widgets):qtConfig(gtk3): SUBDIRS += gtk3
