diff --git a/qtbase/mkspecs/features/qt_configure.prf b/qtbase/mkspecs/features/qt_configure.prf
index c45439c3ef..7953cdd19e 100644
--- a/qtbase/mkspecs/features/qt_configure.prf
+++ b/qtbase/mkspecs/features/qt_configure.prf
@@ -541,12 +541,12 @@ defineTest(qtConfResolveLibs) {
                 # Under Windows, we look only for static libraries, as even for DLLs
                 # one actually links against a static import library.
                 mingw {
-                    lcan += \
-                        # MinGW supports UNIX-style library naming in addition to
-                        # the MSVC style.
-                        lib$${lib}.dll.a lib$${lib}.a \
-                        # Fun fact: prefix-less libraries are also supported.
-                        $${lib}.dll.a $${lib}.a
+                    # prefer (import) libraries for dynamic linking unless we're doing a
+                    # static build where we solely rely on presence of static libraries
+                    shared {
+                        lcan += $${QMAKE_PREFIX_STATICLIB}$${lib}.$${QMAKE_EXTENSION_IMPORTLIB} $${lib}.$${QMAKE_EXTENSION_IMPORTLIB}
+                    }
+                    lcan += $${QMAKE_PREFIX_STATICLIB}$${lib}.$${QMAKE_EXTENSION_STATICLIB} $${lib}.$${QMAKE_EXTENSION_STATICLIB}
                 }
                 lcan += $${lib}.lib
             }
-- 
2.20.1
