--- a/qtwebkit/Source/WTF/WTF.pri	2015-03-23 19:21:26.013831236 +0100
+++ b/qtwebkit/Source/WTF/WTF.pri	2015-03-23 19:22:46.668582720 +0100
@@ -14,7 +14,7 @@
     INCLUDEPATH = $${ROOT_WEBKIT_DIR}/Source/WTF/icu $$INCLUDEPATH
     LIBS += -licucore
 } else:!use?(wchar_unicode): {
-    win32 {
+    win32:!mingw {
         CONFIG(static, static|shared) {
             CONFIG(debug, debug|release) {
                 LIBS += -lsicuind -lsicuucd -lsicudtd
