diff -Naur qt-everywhere-src-5.12.4-orig/qtmultimedia/src/plugins/directshow/directshow.pro qt-everywhere-src-5.12.4/qtmultimedia/src/plugins/directshow/directshow.pro
--- qt-everywhere-src-5.12.4-orig/qtmultimedia/src/plugins/directshow/directshow.pro	2019-05-31 14:09:49.000000000 +0300
+++ qt-everywhere-src-5.12.4/qtmultimedia/src/plugins/directshow/directshow.pro	2019-06-15 16:26:03.148968900 +0300
@@ -14,6 +14,7 @@
 mingw {
     DEFINES -= WINVER=0x0601 _WIN32_WINNT=0x0601
     DEFINES += NO_DSHOW_STRSAFE
+    LIBS_PRIVATE += -lamstrmid
 }
 
 include(common/common.pri)
