https://bugreports.qt.io/browse/QTBUG-84037
--- i686/qtquick3d/src/plugins/assetimporters/assimp/assimp.pro.orig	2020-10-27 09:02:12.000000000 +0100
+++ i686/qtquick3d/src/plugins/assetimporters/assimp/assimp.pro	2020-12-28 00:46:02.767974100 +0100
@@ -10,7 +10,7 @@
 include($$OUT_PWD/../qtassetimporters-config.pri)
 
 qtConfig(system-assimp):!if(cross_compile:host_build) {
-    QMAKE_USE_PRIVATE += assimp
+    QMAKE_USE_PRIVATE += quick3d-assimp
 } else {
     include(../../../3rdparty/assimp/assimp.pri)
 }
