--- a/qtbase/qmake/generators/makefile.cpp
+++ b/qtbase/qmake/generators/makefile.cpp
@@ -323,8 +323,10 @@
                                   vpath.join(QString("::")).toLatin1().constData());
                         if(flags & VPATH_RemoveMissingFiles)
                             remove_file = true;
-                        else if(flags & VPATH_WarnMissingFiles)
-                            warn_msg(WarnLogic, "Failure to find: %s", val.toLatin1().constData());
+                        else if(flags & VPATH_WarnMissingFiles) {
+                            if(!project->isActiveConfig("static"))
+                                warn_msg(WarnLogic, "Failure to find: %s", val.toLatin1().constData());
+                        }
                     } else {
                         l.removeAt(val_it);
                         QString a;
@@ -341,8 +343,10 @@
                               regex.toLatin1().constData(), real_dir.toLatin1().constData());
                     if(flags & VPATH_RemoveMissingFiles)
                         remove_file = true;
-                    else if(flags & VPATH_WarnMissingFiles)
-                        warn_msg(WarnLogic, "Failure to find: %s", val.toLatin1().constData());
+                    else if(flags & VPATH_WarnMissingFiles) {
+                        if(!project->isActiveConfig("static"))
+                            warn_msg(WarnLogic, "Failure to find: %s", val.toLatin1().constData());
+                    }
                 }
             }
         }
