--- a/qtbase/src/corelib/time/qdatetime.cpp
+++ b/qtbase/src/corelib/time/qdatetime.cpp
@@ -2728,7 +2728,7 @@
 #elif defined(Q_OS_INTEGRITY) || defined(Q_OS_RTEMS)
         return 0;
 #else
-        return timezone;
+        return _timezone;
 #endif // Q_OS_WIN
 }
 
@@ -2743,7 +2743,7 @@
         return QString();
     return QString::fromLocal8Bit(name);
 #else
-    return QString::fromLocal8Bit(tzname[isDst]);
+    return QString::fromLocal8Bit(_tzname[isDst]);
 #endif // Q_OS_WIN
 }
 
