--- a/qtactiveqt/tools/dumpcpp/dumpcpp.pro
+++ b/qtactiveqt/tools/dumpcpp/dumpcpp.pro
@@ -1,5 +1,6 @@
 QT += axcontainer widgets core-private
 DEFINES += QT_NO_CAST_TO_ASCII QT_ASCII_CAST_WARNINGS
+static: CONFIG -= import_plugins
 
 SOURCES = main.cpp
 
--- a/qtactiveqt/tools/dumpdoc/dumpdoc.pro
+++ b/qtactiveqt/tools/dumpdoc/dumpdoc.pro
@@ -1,4 +1,5 @@
 QT += axcontainer widgets
+static: CONFIG -= import_plugins
 
 SOURCES += main.cpp
 
