--- a/qtscript/src/3rdparty/javascriptcore/WebKit.pri
+++ b/qtscript/src/3rdparty/javascriptcore/WebKit.pri
@@ -60,6 +60,9 @@
 CONFIG -= warn_on
 *-g++*:QMAKE_CXXFLAGS += -Wall -Wreturn-type -Wcast-align -Wchar-subscripts -Wformat-security -Wreturn-type -Wno-unused-parameter -Wno-sign-compare -Wno-switch -Wno-switch-enum -Wundef -Wmissing-noreturn -Winit-self
 
+win32-clang-g++:QMAKE_CXXFLAGS += -Wno-deprecated-builtins -Wno-deprecated-declarations
+win32-g++:QMAKE_CXXFLAGS += -Wno-class-memaccess -Wno-deprecated-declarations
+
 # Enable GNU compiler extensions to the ARM compiler for all Qt ports using RVCT
 *-armcc {
     RVCT_COMMON_CFLAGS = --gnu --diag_suppress 68,111,177,368,830,1293
