diff -urN qtbase.orig/src/3rdparty/harfbuzz-ng/harfbuzz-ng.pro qtbase/src/3rdparty/harfbuzz-ng/harfbuzz-ng.pro
--- qtbase.orig/src/3rdparty/harfbuzz-ng/harfbuzz-ng.pro	2014-04-23 07:44:49.836276900 +0000
+++ qtbase/src/3rdparty/harfbuzz-ng/harfbuzz-ng.pro	2014-04-23 07:46:22.431276900 +0000
@@ -135,3 +135,6 @@
 }
 
 TARGET = $$TARGET$$qtPlatformTargetSuffix()
+
+target.path = $$[QT_INSTALL_LIBS]
+INSTALLS += target
