--- qt-stable/qtbase/mkspecs/win32-g++/qmake.conf.orig	2014-03-03 22:44:17.750353500 +0000
+++ qt-stable/qtbase/mkspecs/win32-g++/qmake.conf	2014-03-03 22:49:17.035471600 +0000
@@ -111,3 +111,9 @@
 QMAKE_OBJCOPY           = $${CROSS_COMPILE}objcopy
 QMAKE_NM                = $${CROSS_COMPILE}nm -P
 load(qt_config)
+
+# [QT-STATIC-PATCH] - http://qt-project.org/wiki/How-to-build-a-static-Qt-for-Windows-MinGW
+QMAKE_LFLAGS += -static -static-libgcc
+QMAKE_CFLAGS_RELEASE -= -O2
+QMAKE_CFLAGS_RELEASE += -Os -momit-leaf-frame-pointer
+DEFINES += QT_STATIC_BUILD
