diff -urN qt-everywhere-opensource-src-5.2.1/qtbase/qmake/generators/win32/mingw_make.cpp.orig qt-everywhere-opensource-src-5.2.1/qtbase/qmake/generators/win32/mingw_make.cpp
--- qt-everywhere-opensource-src-5.2.1/qtbase/qmake/generators/win32/mingw_make.cpp.orig	2014-02-01 20:37:29.000000000 +0000
+++ qt-everywhere-opensource-src-5.2.1/qtbase/qmake/generators/win32/mingw_make.cpp	2014-03-09 01:48:21.162177200 +0000
@@ -71,7 +71,8 @@
 
 QString MingwMakefileGenerator::getLibTarget()
 {
-    return QString("lib" + project->first("TARGET") + project->first("TARGET_VERSION_EXT") + ".a");
+    return QString("lib" + project->first("TARGET") + project->first("TARGET_VERSION_EXT")
+                  + (project->isActiveConfig("staticlib") ? ".a" : ".dll.a"));
 }
 
 bool MingwMakefileGenerator::findLibraries()
@@ -268,7 +269,7 @@
         if(!project->first("DESTDIR").isEmpty())
             destDir = Option::fixPathToTargetOS(project->first("DESTDIR") + Option::dir_sep, false, false);
         project->values("MINGW_IMPORT_LIB").prepend(destDir + "lib" + project->first("TARGET")
-                                                         + project->first("TARGET_VERSION_EXT") + ".a");
+                                                         + project->first("TARGET_VERSION_EXT") + ".dll.a");
 	project->values("QMAKE_LFLAGS").append(QString("-Wl,--out-implib,") + project->first("MINGW_IMPORT_LIB"));
     }
 
