--- ResIL-1.7.9/m4/devil-api_checks.m4.orig	2009-03-25 14:30:42.000000000 +0300
+++ ResIL-1.7.9/m4/devil-api_checks.m4	2013-12-24 12:23:53.847800000 +0400
@@ -58,7 +58,11 @@
                  AC_CHECK_LIB([alleg42],
 			      [main],
 			      [use_allegro="yes"
-			       ILUT_LIBS="-lalleg42 $ILUT_LIBS"]) ])
+			       ILUT_LIBS="-lalleg42 $ILUT_LIBS"])
+                 AC_CHECK_LIB([allegro],
+			      [main],
+			      [use_allegro="yes"
+			       ILUT_LIBS="-lallegro $ILUT_LIBS"])])
 AS_IF([test "x$use_allegro" != "xyes"],
       [AM_PATH_ALLEGRO(4.2.0, 
                        [use_allegro="yes"
