diff --git a/configure.ac b/configure.ac
index 501d933..557d492 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,6 +9,7 @@ dnl === automake >= 1.12 requires this for 'unusual archivers' support.
 dnl === it must occur before LT_INIT (AC_PROG_LIBTOOL).
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 AC_PROG_SED
 AM_PROG_CC_C_O
diff --git a/src/webp/types.h b/src/webp/types.h
index 0ce2622..5ec9c2d 100644
--- a/src/webp/types.h
+++ b/src/webp/types.h
@@ -39,11 +39,15 @@ typedef long long int int64_t;
 #ifndef WEBP_EXTERN
 // This explicitly marks library functions and allows for changing the
 // signature for e.g., Windows DLL builds.
+# if defined(_WIN32)
+#  define WEBP_EXTERN __declspec(dllexport)
+# else
 # if defined(__GNUC__) && __GNUC__ >= 4
 #  define WEBP_EXTERN extern __attribute__ ((visibility ("default")))
 # else
 #  define WEBP_EXTERN extern
 # endif  /* __GNUC__ >= 4 */
+# endif  /* _WIN32 */
 #endif  /* WEBP_EXTERN */
 
 // Macro to check ABI compatibility (same major revision number)
