--- libwebp-1.3.0/src/libwebp.pc.in.orig	2022-12-20 01:23:35.000000000 +0100
+++ libwebp-1.3.0/src/libwebp.pc.in	2023-01-16 12:46:48.183625800 +0100
@@ -6,7 +6,7 @@
 Name: libwebp
 Description: Library for the WebP graphics format
 Version: @PACKAGE_VERSION@
-Requires: libsharpyuv
+Requires.private: libsharpyuv
 Cflags: -I${includedir}
 Libs: -L${libdir} -l@webp_libname_prefix@webp
 Libs.private: -lm @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
