--- libwebp-0.6.1/configure.ac.orig	2018-01-17 18:06:58.784497500 +0300
+++ libwebp-0.6.1/configure.ac	2018-01-17 18:07:10.759182400 +0300
@@ -452,7 +452,7 @@
                   [AC_MSG_WARN(SDL library not available - no sdl.h)])])
   if test x"$sdl_header" != "xno"; then
     AC_CHECK_LIB(SDL, SDL_Init,
-                 [SDL_LIBS="-lSDL"
+                 [SDL_LIBS="-lmingw32 -lSDLmain -lSDL"
                   SDL_INCLUDES="-DWEBP_HAVE_SDL"
                   AC_DEFINE(WEBP_HAVE_SDL, [1],
                             [Set to 1 if SDL library is installed])
