--- a/makefile
+++ b/makefile
@@ -226,10 +226,10 @@ lrsnash:	lrsnash.c nashdemo.c lrsnashlib.c lrslib.c lrsnashlib.h lrslib.h ${ARIT
 # From here on the author is David Bremner <bremner@unb.ca> to whom you should turn for help             
 #
 # Shared library variables
-SONAME ?=liblrs.so.1
+SONAME ?=liblrs-1.dll
-SOMINOR ?=.0.0
+SOMINOR ?=
-SHLIB ?=$(SONAME)$(SOMINOR)
+SHLIB ?=liblrs.dll
-SHLINK ?=liblrs.so
+SHLINK ?=liblrs-1.0.0.dll
 
 SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
 
