--- libxml2-2.9.0/configure.in.orig	2013-03-31 03:49:49 +0400
+++ libxml2-2.9.0/configure.in	2013-03-31 03:55:47 +0400
@@ -464,6 +464,7 @@
 # include <sys/types.h>
 # endif
 ])
+AC_CHECK_HEADERS([ws2tcpip.h])
 AC_CHECK_HEADERS([netinet/in.h], [], [],
 [#if HAVE_SYS_TYPES_H
 # include <sys/types.h>
@@ -556,21 +557,33 @@
 AC_TRY_COMPILE2([
 #include <stddef.h>
 #include <sys/types.h>
-#include <sys/socket.h>],[
+#if HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
+#elif HAVE_WS2TCPIP_H
+#include <ws2tcpip.h>
+#endif],[
 (void)getsockopt (1, 1, 1, NULL, (socklen_t *)NULL)],[
   AC_MSG_RESULT(socklen_t *)
   XML_SOCKLEN_T=socklen_t],[
   AC_TRY_COMPILE2([
 #include <stddef.h>
 #include <sys/types.h>
-#include <sys/socket.h>],[
+#if HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
+#elif HAVE_WS2TCPIP_H
+#include <ws2tcpip.h>
+#endif],[
 (void)getsockopt (1, 1, 1, NULL, (size_t *)NULL)],[
     AC_MSG_RESULT(size_t *)
     XML_SOCKLEN_T=size_t],[
     AC_TRY_COMPILE2([
 #include <stddef.h>
 #include <sys/types.h>
-#include <sys/socket.h>],[
+#if HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
+#elif HAVE_WS2TCPIP_H
+#include <ws2tcpip.h>
+#endif],[
 (void)getsockopt (1, 1, 1, NULL, (int *)NULL)],[
       AC_MSG_RESULT(int *)
       XML_SOCKLEN_T=int],[
