--- libxml2-2.9.0/xmlmodule.c.orig	2013-03-31 14:50:57 +0400
+++ libxml2-2.9.0/xmlmodule.c	2013-03-31 15:36:37 +0400
@@ -340,6 +340,12 @@
      */
     *symbol = GetProcAddressA(handle, name);
 #else
+    /* FIXME: On the next ABI break change xmlModuleSymbol() prototype
+     * to use some dummy function pointer (doesn't matter what kind of
+     * function, as long as it's a function) instead of void*.
+     * ISO C does NOT require void* to be able to hold a function
+     * pointer, and any conformant compiler would warn about doing this:
+     */
     *symbol = GetProcAddress(handle, name);
 #endif
     return (NULL == *symbol) ? -1 : 0;
