--- libxml2-2.9.0/testThreadsWin32.c.orig	2013-03-31 15:44:28 +0400
+++ libxml2-2.9.0/testThreadsWin32.c	2013-03-31 15:45:53 +0400
@@ -83,13 +83,15 @@
 }
 
 int
-main()
+main(int argc, char **argv)
 {
 	unsigned int i, repeat;
 	unsigned int num_threads = sizeof(testfiles) / sizeof(testfiles[0]);
 	DWORD results[MAX_ARGC];
 	BOOL ret;
 
+        (void) argc;
+        (void) argv;
 	xmlInitParser();
 	for (repeat = 0;repeat < TEST_REPEAT_COUNT;repeat++)
 	{
