--- libxml2-2.9.0/timsort.h.orig	2013-03-31 15:49:18 +0400
+++ libxml2-2.9.0/timsort.h	2013-03-31 16:08:17 +0400
@@ -315,7 +315,15 @@
     SORT_TYPE *tempstore = (SORT_TYPE *)realloc(store->storage, new_size * sizeof(SORT_TYPE));
     if (tempstore == NULL)
     {
-      fprintf(stderr, "Error allocating temporary storage for tim sort: need %lu bytes", sizeof(SORT_TYPE) * new_size);
+      fprintf(stderr, "Error allocating temporary storage for tim sort: need "
+#if defined(_MSC_VER)
+          "%Iu"
+#elif defined(__MINGW32_MAJOR_VERSION)
+          "%u"
+#else
+          "%zu"
+#endif
+          " bytes",sizeof(SORT_TYPE) * new_size);
       exit(1);
     }
     store->storage = tempstore;
