--- libxml2-2.9.0/configure.in.orig	2013-06-16 00:54:13.616644200 +0000
+++ libxml2-2.9.0/configure.in	2013-06-16 00:54:48.285046500 +0000
@@ -1508,7 +1508,7 @@
  WIN32_EXTRA_LDFLAGS="-no-undefined"
  if test "${PYTHON}" != ""
  then
-   WIN32_EXTRA_PYTHON_LIBADD="-L${pythondir}/../../libs -lpython$(echo ${PYTHON_VERSION} | tr -d .)"
+   WIN32_EXTRA_PYTHON_LIBADD="-shrext .pyd -L${pythondir}/../../lib -lpython$(echo ${PYTHON_VERSION} | tr -d .)"
  fi
  ;;
  *-*-cygwin*)
