diff --git a/net.c b/net.c
index b514bab..07edb67 100644
--- a/net.c
+++ b/net.c
@@ -277,7 +277,7 @@ static int redisContextTimeoutMsec(redisContext *c, long *result)
 }
 
 static long redisPollMillis(void) {
-#ifndef _MSC_VER
+#if (!defined(_WIN32) && !defined(_MSC_VER))
     struct timespec now;
     clock_gettime(CLOCK_MONOTONIC, &now);
     return (now.tv_sec * 1000) + now.tv_nsec / 1000000;
diff --git a/ssl.c b/ssl.c
index aa5e3d4..32db0fe 100644
--- a/ssl.c
+++ b/ssl.c
@@ -294,7 +294,8 @@ redisSSLContext *redisCreateSSLContextWithOptions(redisSSLOptions *options, redi
                 goto error;
             }
             X509_STORE* store = SSL_CTX_get_cert_store(ctx->ssl_ctx);
-            while (win_ctx = CertEnumCertificatesInStore(win_store, win_ctx)) {
+            win_ctx = CertEnumCertificatesInStore(win_store, win_ctx);
+            while (win_ctx != NULL) {
                 X509* x509 = NULL;
                 x509 = d2i_X509(NULL, (const unsigned char**)&win_ctx->pbCertEncoded, win_ctx->cbCertEncoded);
                 if (x509) {
@@ -306,6 +307,7 @@ redisSSLContext *redisCreateSSLContextWithOptions(redisSSLOptions *options, redi
                     }
                     X509_free(x509);
                 }
+                win_ctx = CertEnumCertificatesInStore(win_store, win_ctx);
             }
             CertFreeCertificateContext(win_ctx);
             CertCloseStore(win_store, 0);
