From 98819905aad97d8643656c464c188f1e7d34cb98 Mon Sep 17 00:00:00 2001
From: taozuhong <taozuhong@gmail.com>
Date: Sat, 9 Jan 2021 14:09:10 +0800
Subject: [PATCH] add MSYS2 compile support

---
 Makefile | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a8d37a2eb..5463b56c8 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 # This file is released under the BSD license, see the COPYING file
 
 OBJ=alloc.o net.o hiredis.o sds.o async.o read.o sockcompat.o
-SSL_OBJ=ssl.o
+SSL_OBJ=alloc.o net.o hiredis.o sds.o async.o read.o sockcompat.o ssl.o
 EXAMPLES=hiredis-example hiredis-example-libevent hiredis-example-libev hiredis-example-glib hiredis-example-push
 ifeq ($(USE_SSL),1)
 EXAMPLES+=hiredis-example-ssl hiredis-example-libevent-ssl
@@ -84,6 +84,13 @@ else
   SSL_LDFLAGS+=-L$(OPENSSL_PREFIX)/lib -lssl -lcrypto
 endif
 
+ifeq ($(findstring MINGW64,$(uname_S)),MINGW64)
+  OPENSSL_PREFIX?=/mingw64
+  CFLAGS+=-I$(OPENSSL_PREFIX)/include/openssl
+  REAL_LDFLAGS+= -lwsock32 -lws2_32
+  SSL_LDFLAGS+=-L$(OPENSSL_PREFIX)/lib -lssl -lcrypto -lcrypt32
+endif
+
 ifeq ($(uname_S),SunOS)
   IS_SUN_CC=$(shell sh -c '$(CC) -V 2>&1 |egrep -i -c "sun|studio"')
   ifeq ($(IS_SUN_CC),1)
