diff -Naur libssh-0.6.4-orig/config.h.cmake libssh-0.6.4/config.h.cmake
--- libssh-0.6.4-orig/config.h.cmake	2014-12-19 11:11:17.000000000 +0300
+++ libssh-0.6.4/config.h.cmake	2014-12-25 14:10:49.834000000 +0300
@@ -5,7 +5,7 @@
 #cmakedefine VERSION "${APPLICATION_VERSION}"
 
 #cmakedefine LOCALEDIR "${LOCALE_INSTALL_DIR}"
-#cmakedefine DATADIR "${DATADIR}"
+#cmakedefine DATA_DIR "${DATA_DIR}"
 #cmakedefine LIBDIR "${LIBDIR}"
 #cmakedefine PLUGINDIR "${PLUGINDIR}"
 #cmakedefine SYSCONFDIR "${SYSCONFDIR}"
diff -Naur libssh-0.6.4-orig/ConfigureChecks.cmake libssh-0.6.4/ConfigureChecks.cmake
--- libssh-0.6.4-orig/ConfigureChecks.cmake	2014-12-19 11:11:17.000000000 +0300
+++ libssh-0.6.4/ConfigureChecks.cmake	2014-12-25 14:11:05.297000000 +0300
@@ -8,7 +8,7 @@
 
 set(PACKAGE ${APPLICATION_NAME})
 set(VERSION ${APPLICATION_VERSION})
-set(DATADIR ${DATA_INSTALL_DIR})
+set(DATA_DIR ${DATA_INSTALL_DIR})
 set(LIBDIR ${LIB_INSTALL_DIR})
 set(PLUGINDIR "${PLUGIN_INSTALL_DIR}-${LIBRARY_SOVERSION}")
 set(SYSCONFDIR ${SYSCONF_INSTALL_DIR})
