diff -Naur libssh-0.8.1.orig/cmake/Modules/DefineInstallationPaths.cmake libssh-0.8.1/cmake/Modules/DefineInstallationPaths.cmake
--- libssh-0.8.1.orig/cmake/Modules/DefineInstallationPaths.cmake	2018-08-21 08:30:18.455197300 -0400
+++ libssh-0.8.1/cmake/Modules/DefineInstallationPaths.cmake	2018-08-21 08:37:15.702630000 -0400
@@ -1,4 +1,4 @@
-if (UNIX OR OS2)
+if (UNIX OR OS2 OR MINGW)
   IF (NOT APPLICATION_NAME)
     MESSAGE(STATUS "${PROJECT_NAME} is used as APPLICATION_NAME")
     SET(APPLICATION_NAME ${PROJECT_NAME})
diff -Naur libssh-0.8.1.orig/CMakeLists.txt libssh-0.8.1/CMakeLists.txt
--- libssh-0.8.1.orig/CMakeLists.txt	2018-08-21 08:30:18.377098700 -0400
+++ libssh-0.8.1/CMakeLists.txt	2018-08-21 08:37:16.760077700 -0400
@@ -98,8 +98,8 @@
 add_subdirectory(src)
 
 # pkg-config file
-if (UNIX)
-configure_file(libssh.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/libssh.pc)
+if (UNIX OR MINGW)
+configure_file(libssh.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/libssh.pc @ONLY)
 install(
   FILES
     ${CMAKE_CURRENT_BINARY_DIR}/libssh.pc
@@ -108,7 +108,7 @@
   COMPONENT
     pkgconfig
 )
-endif (UNIX)
+endif (UNIX OR MINGW)
 
 # cmake config files
 set(LIBSSH_LIBRARY_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}ssh${CMAKE_SHARED_LIBRARY_SUFFIX})
diff -Naur libssh-0.8.1.orig/libssh.pc.cmake libssh-0.8.1/libssh.pc.cmake
--- libssh-0.8.1.orig/libssh.pc.cmake	2018-08-21 08:30:18.361515500 -0400
+++ libssh-0.8.1/libssh.pc.cmake	2018-08-21 08:37:17.353829600 -0400
@@ -1,6 +1,11 @@
-Name: ${PROJECT_NAME}
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}/bin
+libdir=@LIB_INSTALL_DIR@
+includedir=@INCLUDE_INSTALL_DIR@
+
+Name: @PROJECT_NAME@
 Description: The SSH Library
-Version: ${PROJECT_VERSION}
-Libs: -L${LIB_INSTALL_DIR} -lssh
-Cflags: -I${INCLUDE_INSTALL_DIR}
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lssh
+Cflags: -I${includedir}
 
