--- libtorrent-rasterbar-2.0.5/CMakeLists.txt.orig	2022-01-11 20:57:52.852999800 +0100
+++ libtorrent-rasterbar-2.0.5/CMakeLists.txt	2022-01-11 20:57:58.833204400 +0100
@@ -593,11 +593,17 @@
 		BOOST_ASIO_NO_DEPRECATED
 	PRIVATE
 		TORRENT_BUILDING_LIBRARY
-		_FILE_OFFSET_BITS=64
 		BOOST_EXCEPTION_DISABLE
 		BOOST_ASIO_HAS_STD_CHRONO
 )
 
+if (NOT WIN32)
+	target_compile_definitions(torrent-rasterbar
+		PRIVATE
+			_FILE_OFFSET_BITS=64
+	)
+endif()
+
 target_link_libraries(torrent-rasterbar
 	PUBLIC
 		Threads::Threads
--- libtorrent-rasterbar-2.0.5/include/libtorrent/config.hpp.orig	2022-01-11 21:25:10.602000000 +0100
+++ libtorrent-rasterbar-2.0.5/include/libtorrent/config.hpp	2022-01-11 21:25:16.564783800 +0100
@@ -42,7 +42,9 @@
 
 #include "libtorrent/aux_/disable_warnings_push.hpp"
 
+#ifndef _WIN32
 #define _FILE_OFFSET_BITS 64
+#endif
 
 #include <cstddef>
 
