From 64d6b4900448097b0157abb328621dd211e2947d Mon Sep 17 00:00:00 2001
From: arvidn <arvid@libtorrent.org>
Date: Wed, 11 Apr 2018 13:48:42 +0200
Subject: [PATCH] fix boost-1.67 build

---
 include/libtorrent/ip_filter.hpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/libtorrent/ip_filter.hpp b/include/libtorrent/ip_filter.hpp
index 4e47b39a1d..4455624449 100644
--- a/include/libtorrent/ip_filter.hpp
+++ b/include/libtorrent/ip_filter.hpp
@@ -41,10 +41,16 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <vector>
 
 #include <boost/limits.hpp>
-#include <boost/utility.hpp>
 #include <boost/cstdint.hpp>
 #include <boost/tuple/tuple.hpp>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106700
+#include <boost/next_prior.hpp>
+#else
+#include <boost/utility.hpp>
+#endif
+
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
 #include "libtorrent/address.hpp"
From 9cd0ae67e74a507c1b9ff9c057ee97dda38ccb81 Mon Sep 17 00:00:00 2001
From: arvidn <arvid@libtorrent.org>
Date: Fri, 13 Apr 2018 08:42:39 +0200
Subject: [PATCH] another boost-1.67 build fix

---
 src/kademlia/routing_table.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/kademlia/routing_table.cpp b/src/kademlia/routing_table.cpp
index a23500b69f..dfc3500930 100644
--- a/src/kademlia/routing_table.cpp
+++ b/src/kademlia/routing_table.cpp
@@ -54,6 +54,13 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/cstdint.hpp>
 #include <boost/bind.hpp>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106700
+#include <boost/next_prior.hpp>
+#else
+#include <boost/utility.hpp>
+#endif
+
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
 using boost::uint8_t;
