--- CMakeLists.txt.orig	2014-07-10 23:30:19.099760500 +0300
+++ CMakeLists.txt	2014-07-11 16:38:53.144106400 +0300
@@ -327,7 +327,7 @@
 	set (LIBDIR "lib")
 endif()
 
-install(TARGETS torrent-rasterbar DESTINATION ${LIBDIR})
+install(TARGETS torrent-rasterbar RUNTIME DESTINATION bin LIBRARY DESTINATION ${LIBDIR} ARCHIVE DESTINATION ${LIBDIR} CONFIGURATIONS release)
 install(DIRECTORY include/libtorrent
 	DESTINATION include
 	PATTERN ".svn" EXCLUDE)
