--- PDCursesMod-4.1.0/wingui/Makefile.mng.orig	2021-04-30 22:23:42.270025000 -0700
+++ PDCursesMod-4.1.0/wingui/Makefile.mng	2021-04-30 22:24:54.066906800 -0700
@@ -155,10 +155,10 @@
 
 firework.exe ozdemo.exe newtest.exe ptest.exe rain.exe testcurs.exe  \
 version.exe worm.exe xmas.exe: %.exe: $(demodir)/%.c
-	$(CC) $(CFLAGS) -mwindows -o$@ $< $(LIBCURSES) $(EXELIBS)
+	$(CC) $(CFLAGS) -mwindows -o$@ $< $(LIBARCHIVE) $(EXELIBS)
 
 tuidemo.exe: tuidemo.o tui.o
-	$(LINK) $(LDFLAGS) -mwindows -o$@ tuidemo.o tui.o $(LIBCURSES) $(EXELIBS)
+	$(LINK) $(LDFLAGS) -mwindows -o$@ tuidemo.o tui.o $(LIBARCHIVE) $(EXELIBS)
 
 tui.o: $(demodir)/tui.c $(demodir)/tui.h $(PDCURSES_CURSES_H)
 	$(CC) -c $(CFLAGS) -I$(demodir) -o$@ $<
