diff -Naur pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/cmake/Macros/PySideModules.cmake pyside-setup-opensource-src-5.15.0/sources/pyside2/cmake/Macros/PySideModules.cmake
--- pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/cmake/Macros/PySideModules.cmake	2020-05-26 10:58:24.000000000 -0700
+++ pyside-setup-opensource-src-5.15.0/sources/pyside2/cmake/Macros/PySideModules.cmake	2020-07-30 04:42:55.046927600 -0700
@@ -126,9 +126,9 @@
                         BYPRODUCTS ${${module_SOURCES}}
                         COMMAND Shiboken2::shiboken2 ${GENERATOR_EXTRA_FLAGS}
                         "${pyside2_BINARY_DIR}/${module_NAME}_global.h"
-                        --include-paths=${shiboken_include_dirs}
+                        --include-paths="${shiboken_include_dirs}"
                         ${shiboken_framework_include_dirs_option}
-                        --typesystem-paths=${pyside_binary_dir}${PATH_SEP}${pyside2_SOURCE_DIR}${PATH_SEP}${${module_TYPESYSTEM_PATH}}
+                        --typesystem-paths="${pyside_binary_dir}${PATH_SEP}${pyside2_SOURCE_DIR}${PATH_SEP}${${module_typesystem_path}}"
                         --output-directory=${CMAKE_CURRENT_BINARY_DIR}
                         --license-file=${CMAKE_CURRENT_SOURCE_DIR}/../licensecomment.txt
                         ${typesystem_path}
@@ -148,7 +148,9 @@
                           OUTPUT_NAME "${module_NAME}${SHIBOKEN_PYTHON_EXTENSION_SUFFIX}"
                           LIBRARY_OUTPUT_DIRECTORY ${pyside2_BINARY_DIR})
     if(WIN32)
+        if(MSVC)
         set_target_properties(${module_NAME} PROPERTIES SUFFIX ".pyd")
+        endif()
         # Sanitize windows.h as pulled by gl.h to prevent clashes with QAbstract3dAxis::min(), etc.
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DNOMINMAX")
     endif()
@@ -203,12 +206,14 @@
         list(APPEND generate_pyi_options "--quiet")
     endif()
 
+    if(NOT MINGW)
     # Add target to generate pyi file, which depends on the module target.
     add_custom_target("${module_NAME}_pyi" ALL
                       COMMAND ${CMAKE_COMMAND} -E env ${ld_prefix}
                       "${SHIBOKEN_PYTHON_INTERPRETER}"
                       "${CMAKE_CURRENT_SOURCE_DIR}/../support/generate_pyi.py" ${generate_pyi_options})
     add_dependencies("${module_NAME}_pyi" ${module_NAME})
+    endif()
 
     # install
     install(TARGETS ${module_NAME} LIBRARY DESTINATION "${PYTHON_SITE_PACKAGES}/PySide2")
diff -Naur pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/CMakeLists.txt pyside-setup-opensource-src-5.15.0/sources/pyside2/CMakeLists.txt
--- pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/CMakeLists.txt	2020-05-26 10:58:24.000000000 -0700
+++ pyside-setup-opensource-src-5.15.0/sources/pyside2/CMakeLists.txt	2020-07-30 04:42:52.592848300 -0700
@@ -243,7 +243,7 @@
 use_protected_as_public_hack()
 
 # Build with Address sanitizer enabled if requested. This may break things, so use at your own risk.
-if (SANITIZE_ADDRESS AND NOT MSVC)
+if (SANITIZE_ADDRESS AND NOT WIN32)
     setup_sanitize_address()
 endif()
 
diff -Naur pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/PySide2/QtNetwork/typesystem_network.xml pyside-setup-opensource-src-5.15.0/sources/pyside2/PySide2/QtNetwork/typesystem_network.xml
--- pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/PySide2/QtNetwork/typesystem_network.xml	2020-05-26 10:58:24.000000000 -0700
+++ pyside-setup-opensource-src-5.15.0/sources/pyside2/PySide2/QtNetwork/typesystem_network.xml	2020-07-30 04:42:56.905572400 -0700
@@ -272,6 +272,13 @@
         <enum-type name="Capability" flags="Capabilities" since="4.7"/>
     </object-type>
     <object-type name="QNetworkSession" since="4.7">
+        <modify-function signature="interface()const">
+        <inject-code class="target" position="beginning">
+            #undef interface
+            %RETURN_TYPE %0 = %CPPSELF->%FUNCTION_NAME();
+            %PYARG_0 = %CONVERTTOPYTHON[%RETURN_TYPE](%0);
+        </inject-code>
+        </modify-function>
         <enum-type name="SessionError" since="4.7"/>
         <enum-type name="State" since="4.7"/>
         <enum-type name="UsagePolicy" flags="UsagePolicies"/>
diff -Naur pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/tests/pysidetest/CMakeLists.txt pyside-setup-opensource-src-5.15.0/sources/pyside2/tests/pysidetest/CMakeLists.txt
--- pyside-setup-opensource-src-5.15.0-orig/sources/pyside2/tests/pysidetest/CMakeLists.txt	2020-05-26 10:58:24.000000000 -0700
+++ pyside-setup-opensource-src-5.15.0/sources/pyside2/tests/pysidetest/CMakeLists.txt	2020-07-30 04:43:00.343248600 -0700
@@ -86,9 +86,9 @@
         BYPRODUCTS ${testbinding_SRC}
         COMMAND Shiboken2::shiboken2 ${GENERATOR_EXTRA_FLAGS}
         ${CMAKE_CURRENT_SOURCE_DIR}/pysidetest_global.h
-        --include-paths=${testbinding_include_dirs}
+        --include-paths="${testbinding_include_dirs}"
         ${shiboken_framework_include_dirs_option}
-        --typesystem-paths=${testbinding_typesystem_path}
+        --typesystem-paths="${testbinding_typesystem_path}"
         --output-directory=${CMAKE_CURRENT_BINARY_DIR}
         ${CMAKE_CURRENT_SOURCE_DIR}/typesystem_pysidetest.xml
         --api-version=${SUPPORTED_QT_VERSION}
