--- a/sources/pyside2/CMakeLists.txt	2018-06-19 15:54:43.493795800 +0300
+++ b/sources/pyside2/CMakeLists.txt	2018-06-19 15:55:15.539628700 +0300
@@ -204,6 +204,9 @@
     # Qt5: this flag has changed from /Zc:wchar_t- in Qt4.X
     set(CMAKE_CXX_FLAGS "/Zc:wchar_t /GR /EHsc /DNOCOLOR /DWIN32 /D_WINDOWS /D_SCL_SECURE_NO_WARNINGS") # XXX
 else()
+    if(MINGW)
+        set(CMAKE_CXX_FLAGS "-DNOCOLOR -DWIN32 -D_WINDOWS -D_SCL_SECURE_NO_WARNINGS")
+    endif()
     if(CMAKE_HOST_UNIX AND NOT CYGWIN)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fvisibility=hidden -Wno-strict-aliasing")
     endif()
