diff -Naur gtkimageview-1.6.4-orig/configure.in gtkimageview-1.6.4/configure.in
--- gtkimageview-1.6.4-orig/configure.in	2019-05-29 10:11:23.407610600 +0300
+++ gtkimageview-1.6.4/configure.in	2019-05-29 10:16:24.349032000 +0300
@@ -17,9 +17,7 @@
 AC_PROG_INSTALL
 AM_PROG_LIBTOOL
 
-# Initialize GNOME environment
-GNOME_COMMON_INIT
-GNOME_COMPILE_WARNINGS(error)
+AX_COMPILER_FLAGS([WARN_CFLAGS],[WARN_LDFLAGS])
 DEPRECATED_FLAGS="-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
 AC_SUBST(DEPRECATED_FLAGS)
 
diff -Naur gtkimageview-1.6.4-orig/src/Makefile.am gtkimageview-1.6.4/src/Makefile.am
--- gtkimageview-1.6.4-orig/src/Makefile.am	2019-05-29 10:11:23.397610600 +0300
+++ gtkimageview-1.6.4/src/Makefile.am	2019-05-29 10:15:01.867916500 +0300
@@ -1,4 +1,4 @@
-INCLUDES = $(DEP_CFLAGS)
+AM_CPPFLAGS = $(DEP_CFLAGS)
 
 BUILT_SOURCES =				\
 	gtkimageview-marshal.c		\
diff -Naur gtkimageview-1.6.4-orig/tests/Makefile.am gtkimageview-1.6.4/tests/Makefile.am
--- gtkimageview-1.6.4-orig/tests/Makefile.am	2009-04-05 03:25:14.000000000 +0400
+++ gtkimageview-1.6.4/tests/Makefile.am	2019-05-29 10:15:38.568967900 +0300
@@ -4,7 +4,7 @@
 	$(top_builddir)/src/libgtkimageview.la	\
 	$(DEP_LIBS)				\
 	./testlib/libtest.la
-INCLUDES = $(DEP_CFLAGS) -I$(top_srcdir) -I.
+AM_CPPFLAGS = $(DEP_CFLAGS) -I$(top_srcdir) -I.
 
 noinst_PROGRAMS =	     \
 	ex-abssize	     \
diff -Naur gtkimageview-1.6.4-orig/tests/testlib/Makefile.am gtkimageview-1.6.4/tests/testlib/Makefile.am
--- gtkimageview-1.6.4-orig/tests/testlib/Makefile.am	2009-04-05 03:25:14.000000000 +0400
+++ gtkimageview-1.6.4/tests/testlib/Makefile.am	2019-05-29 10:15:48.208981400 +0300
@@ -1,4 +1,4 @@
-INCLUDES = $(DEP_CFLAGS)
+AM_CPPFLAGS = $(DEP_CFLAGS)
 
 noinst_LTLIBRARIES = libtest.la
 noinst_HEADERS = testlib.h
