diff -ur hdf5-1.8.12.orig/config/cmake/HDFMacros.cmake hdf5-1.8.12/config/cmake/HDFMacros.cmake
--- szip-2.1.orig/config/cmake/SZIPMacros.cmake	2013-11-22 22:35:38.454236491 +0100
+++ szip-2.1/config/cmake/SZIPMacros.cmake	2013-11-22 22:37:27.717130034 +0100
@@ -163,13 +163,4 @@
     endif ()
   endif ()
 
-  #----- Use MSVC Naming conventions for Shared Libraries
-  if (MINGW AND ${libtype} MATCHES "SHARED")
-    set_target_properties (${libtarget}
-        PROPERTIES
-        IMPORT_SUFFIX ".lib"
-        IMPORT_PREFIX ""
-        PREFIX ""
-    )
-  endif ()
 endmacro ()
