--- Vulkan-ValidationLayers-sdk-1.1.85.0/layers/CMakeLists.txt.orig	2018-11-18 00:01:08.401612900 +0300
+++ Vulkan-ValidationLayers-sdk-1.1.85.0/layers/CMakeLists.txt	2018-11-18 00:01:15.205002100 +0300
@@ -158,7 +158,7 @@
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${VulkanHeaders_INCLUDE_DIR})
 
-if(WIN32)
+if(MSVC)
     # Applies to all configurations
     add_definitions(-D_CRT_SECURE_NO_WARNINGS)
     # Avoid: fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj
