--- Vulkan-ValidationLayers/layers/CMakeLists.txt.orig	2018-11-18 00:38:59.130491200 +0300
+++ Vulkan-ValidationLayers/layers/CMakeLists.txt	2018-11-18 00:40:07.358393600 +0300
@@ -101,6 +101,7 @@ endif()
 # System-specific macros to create a library target.
 if(WIN32)
     macro(AddVkLayer target LAYER_COMPILE_DEFINITIONS)
+        if(MSVC)
         file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_${target}.def DEF_FILE)
         add_custom_target(copy-${target}-def-file ALL
                           COMMAND ${CMAKE_COMMAND} -E copy_if_different ${DEF_FILE} VkLayer_${target}.def
@@ -108,6 +109,9 @@ if(WIN32)
         set_target_properties(copy-${target}-def-file PROPERTIES FOLDER ${LAYERS_HELPER_FOLDER})
 
         add_library(VkLayer_${target} SHARED ${ARGN} VkLayer_${target}.def)
+        else()
+        add_library(VkLayer_${target} SHARED ${ARGN})
+        endif()
         target_compile_definitions(VkLayer_${target} PUBLIC ${LAYER_COMPILE_DEFINITIONS})
         target_link_libraries(VkLayer_${target} PRIVATE VkLayer_utils)
         add_dependencies(VkLayer_${target} VkLayer_utils)
--- Vulkan-ValidationLayers/layers/vk_format_utils.h.orig	2020-02-28 11:50:38.167672800 +0300
+++ Vulkan-ValidationLayers/layers/vk_format_utils.h	2020-02-28 11:51:10.078172600 +0300
@@ -25,7 +25,9 @@
 #include "vulkan/vulkan.h"
 
 #if !defined(VK_LAYER_EXPORT)
-#if defined(__GNUC__) && __GNUC__ >= 4
+#if defined(_WIN32)
+#define VK_LAYER_EXPORT __declspec(dllexport)
+#elif defined(__GNUC__) && __GNUC__ >= 4
 #define VK_LAYER_EXPORT __attribute__((visibility("default")))
 #elif defined(__SUNPRO_C) && (__SUNPRO_C >= 0x590)
 #define VK_LAYER_EXPORT __attribute__((visibility("default")))
