--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -326,7 +326,9 @@
 
 if (WIN32)
     install(FILES ${INTERMEDIATE_FILE} DESTINATION ${LAYER_INSTALL_DIR} RENAME ${OUTPUT_FILE_FINAL_NAME})
-    install(FILES $<TARGET_PDB_FILE:VkLayer_khronos_validation> DESTINATION ${LAYER_INSTALL_DIR})
+    if(MSVC)
+        install(FILES $<TARGET_PDB_FILE:VkLayer_khronos_validation> DESTINATION ${LAYER_INSTALL_DIR})
+    endif()
 endif()
 
 install(TARGETS VkLayer_khronos_validation DESTINATION ${LAYER_INSTALL_DIR})
