--- Vulkan-ValidationLayers-1.1.112/layers/CMakeLists.txt.orig	2019-07-01 14:07:17.696528300 +0300
+++ Vulkan-ValidationLayers-1.1.112/layers/CMakeLists.txt	2019-07-01 14:08:31.376925800 +0300
@@ -98,7 +99,10 @@
         set_target_properties(VkLayer_${target} PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libVkLayer_${target}.map,-Bsymbolic,--exclude-libs,ALL")
     endif()
 
-    install(TARGETS VkLayer_${target} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install(TARGETS VkLayer_${target}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 endmacro()
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/generated ${VulkanHeaders_INCLUDE_DIR})
