diff -bur dwarfstack-2.2-orig/src/dwst-exception.c dwarfstack-2.2/src/dwst-exception.c
--- dwarfstack-2.2-orig/src/dwst-exception.c	2025-01-31 20:35:01.142760700 -0700
+++ dwarfstack-2.2/src/dwst-exception.c	2025-01-31 20:35:07.981744200 -0700
@@ -54,7 +54,12 @@
 // csp ... current stack pointer
 // cip ... current instruction pointer
 // cfp ... current frame pointer
-#ifdef _WIN64
+#if defined(_M_ARM64)
+#define csp Sp
+#define cip Pc
+#define cfp Fp
+#define MACH_TYPE IMAGE_FILE_MACHINE_ARM64
+#elif defined(_WIN64)
 #define csp Rsp
 #define cip Rip
 #define cfp Rbp
