--- src/libGLESv2/libGLESv2.def.orig	2014-01-06 11:49:13.494061976 +0100
+++ src/libGLESv2/libGLESv2.def	2014-01-06 11:49:15.984075067 +0100
@@ -186,3 +186,13 @@ EXPORTS
 
     ; Setting up TRACE macro callbacks
     SetTraceFunctionPointers        @180
+    
+    ; WebKit dependencies
+    ShInitBuiltInResources          @200 NONAME
+    ShInitialize                    @201 NONAME
+    ShDestruct                      @202 NONAME
+    ShGetInfo                       @203 NONAME
+    ShGetInfoLog                    @204 NONAME
+    ShGetObjectCode                 @207 NONAME
+    ShCompile                       @208 NONAME
+    ShConstructCompiler             @209 NONAME
--- src/libGLESv2/libGLESv2_mingw32.def.orig	2014-01-06 13:17:51.686441162 +0100
+++ src/libGLESv2/libGLESv2_mingw32.def	2014-01-06 13:17:21.853045374 +0100
@@ -180,3 +180,15 @@ EXPORTS
 	glGetCurrentContext                      @147 NONAME
 	glGetProcAddress@4                       @148 NONAME
 	glBindTexImage@4                         @158 NONAME
+	glCreateRenderer                         @177 NONAME
+	glDestroyRenderer                        @178 NONAME
+
+	; WebKit dependencies
+	ShInitBuiltInResources                   @200 NONAME
+	ShInitialize                             @201 NONAME
+	ShDestruct                               @202 NONAME
+	ShGetInfo                                @203 NONAME
+	ShGetInfoLog                             @204 NONAME
+	ShGetObjectCode                          @207 NONAME
+	ShCompile                                @208 NONAME
+	ShConstructCompiler                      @209 NONAME
