--- angleproject/src/libEGL.gypi.orig	2018-08-01 10:29:47.147896800 +0300
+++ angleproject/src/libEGL.gypi	2018-08-01 10:57:25.538059500 +0300
@@ -52,7 +52,13 @@
                         'DYLIB_INSTALL_NAME_BASE': '@rpath',
                     },
                 }],
+                ['OS=="win"', {
+                    'ldflags': [ '-Wl,--out-implib,libEGL.dll.a ../src/libEGL/libEGL.def' ],
+                }],
             ],
+            'libraries': [
+                '-ldxguid -ld3d9 -lgdi32 -lsetupapi'
+            ]
         },
 
         {
--- angleproject/src/libGLESv2.gypi.orig	2018-08-01 10:29:47.600297600 +0300
+++ angleproject/src/libGLESv2.gypi	2018-08-01 10:56:24.218925200 +0300
@@ -1298,7 +1298,7 @@
         {
             'target_name': 'libGLESv2',
             'type': '<(angle_gl_library_type)',
-            'dependencies': [ 'libANGLE', 'angle_common' ],
+            'dependencies': [ 'translator', 'libANGLE', 'angle_common' ],
             'includes': [ '../gyp/common_defines.gypi', ],
             'sources':
             [
@@ -1315,7 +1315,13 @@
                 {
                     'msvs_requires_importlibrary' : 'true',
                 }],
+                ['OS=="win"', {
+                    'ldflags': [ '-Wl,--out-implib,libGLESv2.dll.a ../src/libGLESv2/libGLESv2_autogen.def' ],
+                }],
             ],
+            'libraries': [
+                '-ldxguid -ld3d9 -lgdi32 -lsetupapi'
+            ]
         },
 
         {
@@ -1367,6 +1373,7 @@
                         'GL_APICALL=',
                         'GL_API=',
                     ],
+                    'ldflags': [ '-Wl,--out-implib,libGLESv1_CM.dll.a ../src/libGLESv1_CM/libGLESv1_CM.def' ],
                 },
                 {
                     'defines':
