--- jellyfish-0.8.8/cjellyfish/porter.c.orig	2021-10-14 23:41:51 +0800
+++ jellyfish-0.8.8/cjellyfish/porter.c	2021-10-14 23:15:29 +0800
@@ -124,7 +124,8 @@
    int j = z->j;
    while(TRUE)
    {  if (i > j) return n;
-      if (! cons(z, i)) break; i++;
+      if (! cons(z, i)) break;
+      i++;
    }
    i++;
    while(TRUE)
@@ -336,21 +337,21 @@
 {  switch (z->b[z->k-1])
    {  case 'a': if (ends(z, 2, "al")) break; return;
       case 'c': if (ends(z, 4, "ance")) break;
-                if (ends(z, 4, "ence")) break; return;
+                if (ends(z, 4, "ence")) {break;} return;
       case 'e': if (ends(z, 2, "er")) break; return;
       case 'i': if (ends(z, 2, "ic")) break; return;
       case 'l': if (ends(z, 4, "able")) break;
-                if (ends(z, 4, "ible")) break; return;
+                if (ends(z, 4, "ible")) {break;} return;
       case 'n': if (ends(z, 3, "ant")) break;
                 if (ends(z, 5, "ement")) break;
                 if (ends(z, 4, "ment")) break;
-                if (ends(z, 3, "ent")) break; return;
+                if (ends(z, 3, "ent")) {break;} return;
       case 'o': if (ends(z, 3, "ion") && (z->b[z->j] == 's' || z->b[z->j] == 't')) break;
-                if (ends(z, 2, "ou")) break; return;
+                if (ends(z, 2, "ou")) {break;} return;
                 /* takes care of -ous */
       case 's': if (ends(z, 3, "ism")) break; return;
       case 't': if (ends(z, 3, "ate")) break;
-                if (ends(z, 3, "iti")) break; return;
+                if (ends(z, 3, "iti")) {break;} return;
       case 'u': if (ends(z, 3, "ous")) break; return;
       case 'v': if (ends(z, 3, "ive")) break; return;
       case 'z': if (ends(z, 3, "ize")) break; return;
