diff -rupN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-07-07 10:38:25.015023050 -0700
+++ b/CMakeLists.txt	2023-07-07 11:03:46.667536200 -0700
@@ -459,13 +459,20 @@ if(ENABLE_IWYU)
   endif()
 endif()
 
-set_target_properties(webm PROPERTIES PUBLIC_HEADER "${webm_public_headers}")
 install(
   TARGETS webm
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/webm)
+  )
+
+install(
+  FILES ${mkvmuxer_public_headers}
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/webm/mkvmuxer)
+
+install(
+  FILES ${mkvparser_public_headers}
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/webm/mkvparser)
 
 # Install common headers into a subdirectory to avoid breaking nested includes.
 install(FILES ${libwebm_common_public_headers}
