--- a/CMakeLists.txt	2015-12-09 08:12:34.000000000 -0800
+++ b/CMakeLists.txt	2021-04-04 13:57:11.379115900 -0700
@@ -30,6 +30,8 @@
   # webm and webm.lib).
   set_target_properties(webm PROPERTIES PROJECT_LABEL libwebm)
   set_target_properties(webm PROPERTIES PREFIX lib)
+  # Link GCC libs statically (uncomment in the next update for dynamic library)
+  # target_link_libraries(webm LINK_PUBLIC -static stdc++ -dynamic)
 endif(WIN32)
 
 include_directories("${LIBWEBM_SRC_DIR}")
@@ -62,3 +64,20 @@
                "${LIBWEBM_SRC_DIR}/webm2pes.cc"
                "${LIBWEBM_SRC_DIR}/webm2pes.h")
 target_link_libraries(webm2pes LINK_PUBLIC webm)
+
+# This stuff copied directly from the Arch MAKEPKG
+install(
+FILES
+  "mkvmuxer.hpp"
+  "mkvmuxertypes.hpp"
+  "mkvmuxerutil.hpp"
+  "mkvparser.hpp"
+  "mkvreader.hpp"
+  "mkvwriter.hpp"
+  "webmids.hpp"
+DESTINATION
+  "include/webm")
+
+install(TARGETS webm vttdemux webm2pes
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib)
