diff -bur mingw-w64-c/mingw-w64-crt/libarm64/Makefile.am mingw-w64/mingw-w64-crt/libarm64/Makefile.am
--- mingw-w64-c/mingw-w64-crt/libarm64/Makefile.am	2022-08-09 20:38:37.579024300 -0600
+++ mingw-w64/mingw-w64-crt/libarm64/Makefile.am	2022-08-09 20:53:43.611208300 -0600
@@ -56,6 +56,7 @@
 libarm64_DATA += %reldir%/libd3d10.a
 libarm64_DATA += %reldir%/libd3d11.a
 libarm64_DATA += %reldir%/libd3d12.a
+libarm64_DATA += %reldir%/libd3dxof.a
 libarm64_DATA += %reldir%/libd3dcompiler_47.a
 libarm64_DATA += %reldir%/libdavclnt.a
 libarm64_DATA += %reldir%/libdavhlpr.a
@@ -91,6 +92,8 @@
 libarm64_DATA += %reldir%/libdsuiext.a
 libarm64_DATA += %reldir%/libduser.a
 libarm64_DATA += %reldir%/libdwmapi.a
+libarm64_DATA += %reldir%/libd3dx11_42.a
+libarm64_DATA += %reldir%/libd3dx11_43.a
 libarm64_DATA += %reldir%/libdwrite.a
 libarm64_DATA += %reldir%/libdxgi.a
 libarm64_DATA += %reldir%/libdxva2.a
diff -bur mingw-w64-c/mingw-w64-crt/Makefile.am mingw-w64/mingw-w64-crt/Makefile.am
--- mingw-w64-c/mingw-w64-crt/Makefile.am	2022-08-09 20:38:36.395147800 -0600
+++ mingw-w64/mingw-w64-crt/Makefile.am	2022-08-09 21:14:02.140459100 -0600
@@ -1935,11 +1935,14 @@
 
 dx64_DATA =
 dx64_DATA += libarm64/libxinput.a
+dx64_DATA += libarm64/libd3dx11.a
 dx64_DATA += libarm64/libd3dcompiler.a
 
 libarm64/libxinput.a: lib-common/xinput1_4.def
 	$(DTDEFARM64) $<
-libarm64/libd3dcompiler.a: lib-common/d3dcompiler_47.def
+libarm64/libd3dx11.a: libarm64/$(d3dx11).def
+	$(DTDEFARM64) $<
+libarm64/libd3dcompiler.a: lib-common/$(d3dcompiler).def
 	$(DTDEFARM64) $<
 
 endif
diff -bur mingw-w64-c/mingw-w64-crt/Makefile.in mingw-w64/mingw-w64-crt/Makefile.in
--- mingw-w64-c/mingw-w64-crt/Makefile.in	2022-08-09 20:38:36.453134200 -0600
+++ mingw-w64/mingw-w64-crt/Makefile.in	2022-08-09 21:16:34.572562700 -0600
@@ -11909,7 +11909,8 @@
 @LIB64_TRUE@	lib64/libx3daudio.a lib64/libd3dx9.a \
 @LIB64_TRUE@	lib64/libd3dx10.a lib64/libd3dx11.a \
 @LIB64_TRUE@	lib64/libd3dcsxd.a lib64/libd3dcompiler.a
-@LIBARM64_TRUE@dx64_DATA = libarm64/libxinput.a \
+@LIBARM64_TRUE@dx64_DATA = libarm64/libxinput.a \
+@LIBARM64_TRUE@	libarm64/libd3dx11.a \
 @LIBARM64_TRUE@	libarm64/libd3dcompiler.a
 
 # End 64-bit runtime
@@ -12692,6 +12693,7 @@
 @LIBARM64_TRUE@	libarm64/libcscapi.a libarm64/libd2d1.a \
 @LIBARM64_TRUE@	libarm64/libd3d9.a libarm64/libd3d10.a \
 @LIBARM64_TRUE@	libarm64/libd3d11.a libarm64/libd3d12.a \
+@LIBARM64_TRUE@	libarm64/libd3dxof.a \
 @LIBARM64_TRUE@	libarm64/libd3dcompiler_47.a \
 @LIBARM64_TRUE@	libarm64/libdavclnt.a libarm64/libdavhlpr.a \
 @LIBARM64_TRUE@	libarm64/libdbgeng.a libarm64/libdbghelp.a \
@@ -12709,40 +12711,42 @@
 @LIBARM64_TRUE@	libarm64/libdsrole.a libarm64/libdssec.a \
 @LIBARM64_TRUE@	libarm64/libdssenh.a libarm64/libdsuiext.a \
 @LIBARM64_TRUE@	libarm64/libduser.a libarm64/libdwmapi.a \
-@LIBARM64_TRUE@	libarm64/libdwrite.a libarm64/libdxgi.a \
-@LIBARM64_TRUE@	libarm64/libdxva2.a libarm64/libeappcfg.a \
-@LIBARM64_TRUE@	libarm64/libeappgnui.a libarm64/libeapphost.a \
-@LIBARM64_TRUE@	libarm64/libeappprxy.a libarm64/libefsadu.a \
-@LIBARM64_TRUE@	libarm64/libelscore.a libarm64/libesentprf.a \
-@LIBARM64_TRUE@	libarm64/libevr.a libarm64/libfdeploy.a \
-@LIBARM64_TRUE@	libarm64/libfeclient.a libarm64/libfilemgmt.a \
-@LIBARM64_TRUE@	libarm64/libfltlib.a libarm64/libfmifs.a \
-@LIBARM64_TRUE@	libarm64/libfontsub.a libarm64/libfwpuclnt.a \
-@LIBARM64_TRUE@	libarm64/libgdi32.a libarm64/libgetuname.a \
-@LIBARM64_TRUE@	libarm64/libglu32.a libarm64/libgpedit.a \
-@LIBARM64_TRUE@	libarm64/libhbaapi.a libarm64/libhid.a \
-@LIBARM64_TRUE@	libarm64/libhlink.a libarm64/libhotplug.a \
-@LIBARM64_TRUE@	libarm64/libhtmlhelp.a libarm64/libhtui.a \
-@LIBARM64_TRUE@	libarm64/libiashlpr.a libarm64/libiassam.a \
-@LIBARM64_TRUE@	libarm64/libiassvcs.a libarm64/libicm32.a \
-@LIBARM64_TRUE@	libarm64/libicmp.a libarm64/libicmui.a \
-@LIBARM64_TRUE@	libarm64/libiernonce.a libarm64/libimagehlp.a \
-@LIBARM64_TRUE@	libarm64/libimgutil.a libarm64/libimm32.a \
-@LIBARM64_TRUE@	libarm64/libinetcomm.a libarm64/libinetmib1.a \
-@LIBARM64_TRUE@	libarm64/libinput.a libarm64/libinseng.a \
-@LIBARM64_TRUE@	libarm64/libiphlpapi.a libarm64/libipnathlp.a \
-@LIBARM64_TRUE@	libarm64/libjsproxy.a libarm64/libkdcom.a \
-@LIBARM64_TRUE@	libarm64/libkeymgr.a libarm64/libks.a \
-@LIBARM64_TRUE@	libarm64/libksecdd.a libarm64/libktmw32.a \
-@LIBARM64_TRUE@	libarm64/liblinkinfo.a libarm64/libloadperf.a \
-@LIBARM64_TRUE@	libarm64/libloghours.a libarm64/liblogoncli.a \
-@LIBARM64_TRUE@	libarm64/liblz32.a libarm64/libmapi32.a \
-@LIBARM64_TRUE@	libarm64/libmapistub.a libarm64/libmcicda.a \
-@LIBARM64_TRUE@	libarm64/libmciseq.a libarm64/libmciwave.a \
-@LIBARM64_TRUE@	libarm64/libmdminst.a libarm64/libmf3216.a \
-@LIBARM64_TRUE@	libarm64/libmf.a libarm64/libmfcore.a \
-@LIBARM64_TRUE@	libarm64/libmfplat.a libarm64/libmfplay.a \
-@LIBARM64_TRUE@	libarm64/libmfreadwrite.a \
+@LIBARM64_TRUE@	libarm64/libd3dx11_42.a \
+@LIBARM64_TRUE@	libarm64/libd3dx11_43.a \
+@LIBARM64_TRUE@	libarm64/libdwrite.a \
+@LIBARM64_TRUE@	libarm64/libdxgi.a libarm64/libdxva2.a \
+@LIBARM64_TRUE@	libarm64/libeappcfg.a libarm64/libeappgnui.a \
+@LIBARM64_TRUE@	libarm64/libeapphost.a libarm64/libeappprxy.a \
+@LIBARM64_TRUE@	libarm64/libefsadu.a libarm64/libelscore.a \
+@LIBARM64_TRUE@	libarm64/libesentprf.a libarm64/libevr.a \
+@LIBARM64_TRUE@	libarm64/libfdeploy.a libarm64/libfeclient.a \
+@LIBARM64_TRUE@	libarm64/libfilemgmt.a libarm64/libfltlib.a \
+@LIBARM64_TRUE@	libarm64/libfmifs.a libarm64/libfontsub.a \
+@LIBARM64_TRUE@	libarm64/libfwpuclnt.a libarm64/libgdi32.a \
+@LIBARM64_TRUE@	libarm64/libgetuname.a libarm64/libglu32.a \
+@LIBARM64_TRUE@	libarm64/libgpedit.a libarm64/libhbaapi.a \
+@LIBARM64_TRUE@	libarm64/libhid.a libarm64/libhlink.a \
+@LIBARM64_TRUE@	libarm64/libhotplug.a libarm64/libhtmlhelp.a \
+@LIBARM64_TRUE@	libarm64/libhtui.a libarm64/libiashlpr.a \
+@LIBARM64_TRUE@	libarm64/libiassam.a libarm64/libiassvcs.a \
+@LIBARM64_TRUE@	libarm64/libicm32.a libarm64/libicmp.a \
+@LIBARM64_TRUE@	libarm64/libicmui.a libarm64/libiernonce.a \
+@LIBARM64_TRUE@	libarm64/libimagehlp.a libarm64/libimgutil.a \
+@LIBARM64_TRUE@	libarm64/libimm32.a libarm64/libinetcomm.a \
+@LIBARM64_TRUE@	libarm64/libinetmib1.a libarm64/libinput.a \
+@LIBARM64_TRUE@	libarm64/libinseng.a libarm64/libiphlpapi.a \
+@LIBARM64_TRUE@	libarm64/libipnathlp.a libarm64/libjsproxy.a \
+@LIBARM64_TRUE@	libarm64/libkdcom.a libarm64/libkeymgr.a \
+@LIBARM64_TRUE@	libarm64/libks.a libarm64/libksecdd.a \
+@LIBARM64_TRUE@	libarm64/libktmw32.a libarm64/liblinkinfo.a \
+@LIBARM64_TRUE@	libarm64/libloadperf.a libarm64/libloghours.a \
+@LIBARM64_TRUE@	libarm64/liblogoncli.a libarm64/liblz32.a \
+@LIBARM64_TRUE@	libarm64/libmapi32.a libarm64/libmapistub.a \
+@LIBARM64_TRUE@	libarm64/libmcicda.a libarm64/libmciseq.a \
+@LIBARM64_TRUE@	libarm64/libmciwave.a libarm64/libmdminst.a \
+@LIBARM64_TRUE@	libarm64/libmf3216.a libarm64/libmf.a \
+@LIBARM64_TRUE@	libarm64/libmfcore.a libarm64/libmfplat.a \
+@LIBARM64_TRUE@	libarm64/libmfplay.a libarm64/libmfreadwrite.a \
 @LIBARM64_TRUE@	libarm64/libmfsensorgroup.a \
 @LIBARM64_TRUE@	libarm64/libmgmtapi.a libarm64/libmidimap.a \
 @LIBARM64_TRUE@	libarm64/libmincore.a libarm64/libmlang.a \
@@ -78465,7 +78483,9 @@
 
 @LIBARM64_TRUE@libarm64/libxinput.a: lib-common/xinput1_4.def
 @LIBARM64_TRUE@	$(DTDEFARM64) $<
-@LIBARM64_TRUE@libarm64/libd3dcompiler.a: lib-common/d3dcompiler_47.def
+@LIBARM64_TRUE@libarm64/libd3dx11.a: libarm64/$(d3dx11).def
+@LIBARM64_TRUE@	$(DTDEFARM64) $<
+@LIBARM64_TRUE@libarm64/libd3dcompiler.a: lib-common/$(d3dcompiler).def
 @LIBARM64_TRUE@	$(DTDEFARM64) $<
 
 # End ARM 64-bit runtime
