From 13d7142491390894f057da66adea4fd99c3c1fc0 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 24 Aug 2018 20:20:26 +0300
Subject: [PATCH] libamstrmid: Add evr stuff

---
 mingw-w64-crt/libsrc/amstrmid.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mingw-w64-crt/libsrc/amstrmid.c b/mingw-w64-crt/libsrc/amstrmid.c
index 36b2704b..5119f2c7 100644
--- a/mingw-w64-crt/libsrc/amstrmid.c
+++ b/mingw-w64-crt/libsrc/amstrmid.c
@@ -4,3 +4,4 @@
 
 #include <initguid.h>
 #include <amaudio.h>
+#include <evr.h>
-- 
2.18.0

