--- glbinding-2.0.0/glbinding-config.cmake.orig	2016-03-25 18:47:33.337953800 +0200
+++ glbinding-2.0.0/glbinding-config.cmake	2016-03-25 18:51:25.416871900 +0200
@@ -20,21 +20,17 @@
 endmacro()
 
 # Macro to search for all modules
-macro(find_modules PREFIX)
+macro(find_modules)
     foreach(module_name ${MODULE_NAMES})
-        find_module("${CMAKE_CURRENT_LIST_DIR}/${PREFIX}/${module_name}/${module_name}-export.cmake")
+        find_module("${CMAKE_CURRENT_LIST_DIR}/${module_name}/${module_name}-export.cmake")
     endforeach(module_name)
 endmacro()
 
 
 # Try install location
 set(MODULE_FOUND FALSE)
-find_modules("cmake")
+find_modules()
 
 if(MODULE_FOUND)
     return()
 endif()
-
-# Try common build locations
-find_modules("build/cmake")
-find_modules("build-debug/cmake")
