--- glbinding-2.0.0/CMakeLists.txt.orig	2016-03-24 19:15:07.000000000 +0200
+++ glbinding-2.0.0/CMakeLists.txt	2016-03-25 17:33:50.581788900 +0200
@@ -113,6 +113,17 @@
     set(INSTALL_SHORTCUTS "share/applications")     # /usr/[local]/share/applications
     set(INSTALL_ICONS     "share/pixmaps")          # /usr/[local]/share/pixmaps
     set(INSTALL_INIT      "/etc/init")              # /etc/init (upstart init scripts)
+elseif(MINGW)
+    set(SYSTEM_DIR_INSTALL ON)
+    set(INSTALL_ROOT      "lib/cmake")
+    set(INSTALL_CMAKE     "lib/cmake/${project}")
+    set(INSTALL_EXAMPLES  ".") # Examples are not built
+    set(INSTALL_DATA      "share/${project}")
+    set(INSTALL_BIN       "bin")
+    set(INSTALL_SHARED    "lib")
+    set(INSTALL_LIB       "lib")
+    set(INSTALL_INCLUDE   "include")
+    set(INSTALL_DOC       "share/doc/${project}")
 else()
     # Install into local directory
     set(INSTALL_ROOT      ".")                      # ./
@@ -159,15 +170,18 @@
 # 
 
 # Install version file
-install(FILES "${PROJECT_BINARY_DIR}/VERSION" DESTINATION ${INSTALL_ROOT} COMPONENT runtime)
+install(FILES "${PROJECT_BINARY_DIR}/VERSION" DESTINATION ${INSTALL_DOC} COMPONENT runtime)
 
 # Install cmake find script for the project
-install(FILES ${META_PROJECT_NAME}-config.cmake DESTINATION ${INSTALL_ROOT} COMPONENT dev)
+install(FILES ${META_PROJECT_NAME}-config.cmake DESTINATION ${INSTALL_CMAKE} COMPONENT dev)
 
 # Install the project meta files
-install(FILES AUTHORS   DESTINATION ${INSTALL_ROOT} COMPONENT runtime)
-install(FILES LICENSE   DESTINATION ${INSTALL_ROOT} COMPONENT runtime)
-install(FILES README.md DESTINATION ${INSTALL_ROOT} COMPONENT runtime)
+install(FILES AUTHORS   DESTINATION ${INSTALL_DOC} COMPONENT runtime)
+install(FILES LICENSE   DESTINATION ${INSTALL_DOC} COMPONENT runtime)
+install(FILES README.md DESTINATION ${INSTALL_DOC} COMPONENT runtime)
 
 # Install runtime data
+    
+if(OPTION_BUILD_EXAMPLES)
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/data DESTINATION ${INSTALL_DATA} COMPONENT examples_data)
+endif()
