--- glbinding-1.1.0/CMakeLists.txt.orig	2015-08-26 08:11:47.410400000 +0300
+++ glbinding-1.1.0/CMakeLists.txt	2015-08-26 08:11:51.669200000 +0300
@@ -97,7 +97,7 @@
 
 # Installation paths
 set(project ${META_PROJECT_NAME})
-if(WIN32)
+if(MSVC)
 
     # "%PROGRAMFILES%/<project>/"
     
@@ -185,7 +185,7 @@
 install(FILES AUTHORS                      DESTINATION ${INSTALL_ROOT} COMPONENT dev)
 install(FILES LICENSE                      DESTINATION ${INSTALL_ROOT} COMPONENT dev)
 install(FILES codegeneration/gl.revision   DESTINATION ${INSTALL_ROOT} COMPONENT dev)
-install(FILES glbinding-config.cmake       DESTINATION ${INSTALL_ROOT} COMPONENT dev)
+install(FILES glbinding-config.cmake       DESTINATION ${INSTALL_LIB}/cmake/${project} COMPONENT dev)
 
 # Data files
 if(OPTION_BUILD_EXAMPLES)
--- glbinding-1.1.0/glbinding-config.cmake.orig	2015-08-26 08:29:55.746400000 +0300
+++   glbinding-1.1.0/glbinding-config.cmake	2015-08-26 08:30:51.953200000 +0300
@@ -98,7 +98,7 @@
 
 find(glbinding glbinding/glbinding_api.h ${LIB_PATHS})
 
-if (GLBINDING_LIBRARY AND WIN32)
+if (GLBINDING_LIBRARY AND MSVC)
     set(GLBINDING_BINARIES "")
 
     find_file(GLBINDING_BINARY_RELEASE
