--- a/source/API/SBValue.cpp
+++ b/source/API/SBValue.cpp
@@ -75,7 +75,7 @@
   return *this;
 }
 
-SBValue::~SBValue() = default;
+LLDB_API SBValue::~SBValue() = default;
 
 bool SBValue::IsValid() {
   LLDB_INSTRUMENT_VA(this);
--- a/source/API/SBValueList.cpp
+++ b/source/API/SBValueList.cpp
@@ -152,7 +152,7 @@
   }
 }
 
-SBValue SBValueList::GetValueAtIndex(uint32_t idx) const {
+LLDB_API SBValue SBValueList::GetValueAtIndex(uint32_t idx) const {
   LLDB_INSTRUMENT_VA(this, idx);
 
   SBValue sb_value;
@@ -162,7 +162,7 @@
   return sb_value;
 }
 
-uint32_t SBValueList::GetSize() const {
+LLDB_API uint32_t SBValueList::GetSize() const {
   LLDB_INSTRUMENT_VA(this);
 
   uint32_t size = 0;
