--- libmimic-1.0.4/configure.ac.orig	2013-04-10 20:50:10 +0400
+++ libmimic-1.0.4/configure.ac	2013-04-10 20:50:49 +0400
@@ -2,7 +2,7 @@
 AC_PREREQ(2.53)
 AC_INIT(libmimic, 1.0.4)
 AC_CONFIG_SRCDIR(libmimic.pc.in)
-AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
+AM_INIT_AUTOMAKE([silent-rules])
 
 AM_MAINTAINER_MODE
 AC_PROG_MAKE_SET
@@ -10,7 +10,7 @@
 dnl Libtool.
 AM_PROG_LIBTOOL
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 dnl Increment on interface add, remove or change.
 MIMIC_CURRENT=0
