--- lasem-LASEM_0_4_4.orig/src/Makefile.am	2019-06-03 20:54:54.789589000 -0400
+++ lasem-LASEM_0_4_4/src/Makefile.am	2019-06-03 20:56:54.075516900 -0400
@@ -268,7 +268,7 @@
 liblasem_@LASEM_API_VERSION@_la_HEADERS = $(LASEM_DOM_HDRS)
 liblasem_@LASEM_API_VERSION@_la_HEADERS += lsmdomenumtypes.h
 
-liblasem_@LASEM_API_VERSION@_la_LDFLAGS = -version-info $(LASEM_LIBTOOL_VERSION)
+liblasem_@LASEM_API_VERSION@_la_LDFLAGS = -version-info $(LASEM_LIBTOOL_VERSION) -no-undefined
 
 lsmdomenumtypes.h: lsmdomenumtypes.h.template $(LASEM_DOM_HDRS) $(GLIB_MKENUMS)
 	$(AM_V_GEN) (cd $(srcdir) && $(GLIB_MKENUMS) --template lsmdomenumtypes.h.template $(LASEM_DOM_HDRS)) > $@
