diff --git a/src/librustc_llvm/build.rs b/src/librustc_llvm/build.rs
index caa2740408..52738a47f7 100644
--- a/src/librustc_llvm/build.rs
+++ b/src/librustc_llvm/build.rs
@@ -269,6 +269,7 @@
     // LLVM requires symbols from this library, but apparently they're not printed
     // during llvm-config?
     if target.contains("windows-gnu") {
+        println!("cargo:rustc-link-lib=static-nobundle=ffi");
         println!("cargo:rustc-link-lib=static-nobundle=gcc_s");
         println!("cargo:rustc-link-lib=static-nobundle=pthread");
     }
-- 
2.12.0
