--- rustc-1.62.0-src/compiler/rustc_llvm/build.rs.orig  2022-06-30 20:07:13.622789400 +0200
+++ rustc-1.62.0-src/compiler/rustc_llvm/build.rs       2022-06-30 20:08:33.409147400 +0200
@@ -367,6 +367,6 @@
     // Libstdc++ depends on pthread which Rust doesn't link on MinGW
     // since nothing else requires it.
     if target.ends_with("windows-gnu") {
-        println!("cargo:rustc-link-lib=static:-bundle=pthread");
+        println!("cargo:rustc-link-lib=static:-bundle=ffi");
     }
 }
