From f3b4afc073e0f96ee80297bbeb18396433dbcf34 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Wed, 14 Jun 2017 01:36:09 +0100
Subject: [PATCH] bootstrap: use stable release for stage0


diff --git a/src/bootstrap/bootstrap.py b/src/bootstrap/bootstrap.py
index 6cd9496b7a..644eb6f0f4 100644
--- a/src/bootstrap/bootstrap.py
+++ b/src/bootstrap/bootstrap.py
@@ -201,7 +201,7 @@ class RustBuild(object):
         if self.cargo().startswith(self.bin_root()) and \
                 (not os.path.exists(self.cargo()) or self.cargo_out_of_date()):
             self.print_what_it_means_to_bootstrap()
-            filename = "cargo-{}-{}.tar.gz".format('0.18.0', self.build)
+            filename = "cargo-{}-{}.tar.gz".format('0.19.0', self.build)
             url = "https://static.rust-lang.org/dist/" + self.stage0_rustc_date()
             tarball = os.path.join(rustc_cache, filename)
             if not os.path.exists(tarball):
diff --git a/src/stage0.txt b/src/stage0.txt
index 38cbfa0dec..75b3cb86e0 100644
--- a/src/stage0.txt
+++ b/src/stage0.txt
@@ -12,4 +12,4 @@
 # tarball for a stable release you'll likely see `1.x.0-$date` where `1.x.0` was
 # released on `$date`
 
-rustc: 1.17.0-2017-04-27
+rustc: 1.18.0-2017-06-08
-- 
2.13.0

