diff --git a/compiler/rustc_llvm/build.rs b/compiler/rustc_llvm/build.rs
index 943ce589c4f..ea524daa93c 100644
--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -302,6 +302,7 @@ fn main() {
     // Libstdc++ depends on pthread which Rust doesn't link on MinGW
     // since nothing else requires it.
     if target.contains("windows-gnu") {
+        println!("cargo:rustc-link-lib=static:-bundle=ffi");
         println!("cargo:rustc-link-lib=static:-bundle=pthread");
     }
 }
