--- a/src/bootstrap/src/utils/helpers.rs
+++ b/src/bootstrap/src/utils/helpers.rs
@@ -467,7 +467,10 @@
     stage: u32,
 ) -> Vec<String> {
     let mut args = vec![];
-    if !builder.is_lld_direct_linker(target) && builder.config.lld_mode.is_used() {
+    if !builder.is_lld_direct_linker(target)
+        && builder.config.lld_mode.is_used()
+        && use_host_linker(target)
+    {
         match builder.config.lld_mode {
             LldMode::External => {
                 // cfg(bootstrap) - remove after updating bootstrap compiler (#137498)
