--- rustc-1.72.0-src/compiler/rustc_codegen_ssa/src/back/link.rs.orig   2023-08-23 13:20:55.586469700 +0200
+++ rustc-1.72.0-src/compiler/rustc_codegen_ssa/src/back/link.rs        2023-08-23 13:33:55.439370400 +0200
@@ -1687,7 +1687,10 @@
 /// Various toolchain components used during linking are used from rustc distribution
 /// instead of being found somewhere on the host system.
 /// We only provide such support for a very limited number of targets.
+#[allow(warnings)]
 fn self_contained(sess: &Session, crate_type: CrateType) -> bool {
+    return false;
+
     if let Some(self_contained) = sess.opts.cg.link_self_contained.explicitly_set {
         if sess.target.link_self_contained == LinkSelfContainedDefault::False {
             sess.emit_err(errors::UnsupportedLinkSelfContained);
