--- rustc-beta-src/compiler/rustc_codegen_ssa/src/back/link.rs.orig     2025-04-20 11:16:06.616814900 -0700
+++ rustc-beta-src/compiler/rustc_codegen_ssa/src/back/link.rs  2025-04-20 11:16:29.066080000 -0700
@@ -1808,6 +1808,9 @@
     crate_type: CrateType,
     linker: &Path,
 ) -> LinkSelfContainedComponents {
+    if sess.target.options.os == "windows" && sess.target.options.env == "gnu" {
+        return LinkSelfContainedComponents::empty();
+    }
     // Turn the backwards compatible bool values for `self_contained` into fully inferred
     // `LinkSelfContainedComponents`.
     let self_contained =
