--- rustc-1.53.0-src/compiler/rustc_codegen_ssa/src/back/link.rs.orig	2021-07-04 01:49:38.816000000 +0200
+++ rustc-1.53.0-src/compiler/rustc_codegen_ssa/src/back/link.rs	2021-07-04 12:45:55.559928800 +0200
@@ -1420,7 +1420,10 @@
 
 /// Whether we link to our own CRT objects instead of relying on gcc to pull them.
 /// We only provide such support for a very limited number of targets.
+#[allow(warnings)]
 fn crt_objects_fallback(sess: &Session, crate_type: CrateType) -> bool {
+    return false;
+
     if let Some(self_contained) = sess.opts.cg.link_self_contained {
         return self_contained;
     }
