--- rustc-1.75.0-src/compiler/rustc_codegen_ssa/src/back/link.rs.orig	2023-12-21 17:55:28.000000000 +0100
+++ rustc-1.75.0-src/compiler/rustc_codegen_ssa/src/back/link.rs	2023-12-22 13:17:31.361178700 +0100
@@ -1708,7 +1708,9 @@
 /// Various toolchain components used during linking are used from rustc distribution
 /// instead of being found somewhere on the host system.
 /// We only provide such support for a very limited number of targets.
+#[allow(warnings)]
 fn self_contained_components(sess: &Session, crate_type: CrateType) -> LinkSelfContainedComponents {
+    return LinkSelfContainedComponents::empty();
     // Turn the backwards compatible bool values for `self_contained` into fully inferred
     // `LinkSelfContainedComponents`.
     let self_contained =
