--- rustc-1.81.0-src/src/tools/cargo/src/cargo/core/compiler/build_context/target_info.rs
+++ rustc-1.81.0-src/src/tools/cargo/src/cargo/core/compiler/build_context/target_info.rs
@@ -389,7 +389,10 @@ impl TargetInfo {
                     crate_type: Some(crate_type.clone()),
                     should_replace_hyphens: true,
                 });
-            } else if target_triple.ends_with("windows-gnu") && suffix == ".dll" {
+            } else if suffix == ".dll"
+                && (target_triple.ends_with("windows-gnu")
+                    || target_triple.ends_with("windows-gnullvm"))
+            {
                 // See https://cygwin.com/cygwin-ug-net/dll.html for more
                 // information about GNU import libraries.
                 // LD can link DLL directly, but LLD requires the import library.
