--- a/src/bootstrap/src/core/build_steps/llvm.rs
+++ b/src/bootstrap/src/core/build_steps/llvm.rs
@@ -284,7 +284,7 @@
             LlvmBuildStatus::ShouldBuild(m) => m,
         };
 
-        if builder.llvm_link_shared() && target.is_windows() && !target.is_windows_gnullvm() {
+        if builder.llvm_link_shared() && target.is_msvc() {
             panic!("shared linking to LLVM is not currently supported on {}", target.triple);
         }
 
