--- a/library/compiler-builtins/compiler-builtins/build.rs
+++ b/library/compiler-builtins/compiler-builtins/build.rs
@@ -611,7 +611,7 @@ pub fn compile(llvm_target: &[&str], target: &Target) {
             }
             if let Some(lib) = rt_builtins_ext.file_name() {
                 println!(
-                    "cargo::rustc-link-lib=static:+verbatim={}",
+                    "cargo::rustc-link-lib=static:-bundle,+verbatim={}",
                     lib.to_str().unwrap()
                 );
             }
