--- rustc-1.16.0-src/src/bootstrap/config.rs.orig      2017-03-10 23:55:21 +0100
+++ rustc-1.16.0-src/src/bootstrap/config.rs   2017-03-23 22:51:46 +0100
@@ -516,7 +516,7 @@
                 "CFG_JEMALLOC_ROOT" if value.len() > 0 => {
                     let target = self.target_config.entry(self.build.clone())
                                      .or_insert(Target::default());
-                    target.jemalloc = Some(parse_configure_path(value).join("libjemalloc_pic.a"));
+                    target.jemalloc = Some(parse_configure_path(value).join("libjemalloc.a"));
                 }
                 "CFG_ARM_LINUX_ANDROIDEABI_NDK" if value.len() > 0 => {
                     let target = "arm-linux-androideabi".to_string();
