commit 7be5e15a12b89168e4d096348848eaa9a9534858
Author: Zion Nimchuk <zionnimchuk@gmail.com>
Date:   Fri Oct 14 22:45:53 2016 -0700

    Force download to use curl, not powershell

diff --git a/src/etc/download.py b/src/etc/download.py
index 5ef0df3..9374262 100644
--- a/src/etc/download.py
+++ b/src/etc/download.py
@@ -7,12 +7,7 @@ import tarfile
 
 def get(url, path, quiet=False):
     # see http://serverfault.com/questions/301128/how-to-download
-    if sys.platform == 'win32':
-        run(["PowerShell.exe", "/nologo", "-Command",
-             "(New-Object System.Net.WebClient).DownloadFile('" + url +
-             "', '" + path + "')"], quiet=quiet)
-    else:
-        run(["curl", "-o", path, url], quiet=quiet)
+    run(["curl", "-o", path, url], quiet=quiet)
 
 
 def unpack(tarball, dst, quiet=False, strip=0):
