 main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main.c b/main.c
index b61ca34..c937efc 100644
--- a/main.c
+++ b/main.c
@@ -190,6 +190,7 @@ output_opt_cb (const char *opt, const char *arg, gpointer data,
         {
           fprintf (stderr, "Ignoring incompatible output option \"%s\"\n",
                    opt);
+          fflush (stderr);
           return TRUE;
         }
     }
@@ -322,6 +323,7 @@ process_package_args (const char *cmdline, GList **packages, FILE *log)
   if (reqs == NULL)
     {
       fprintf (stderr, "Must specify package names on the command line\n");
+      fflush (stderr);
       return FALSE;
     }
 
-- 1.8.1.4 