diff -bur gcc-15.1.0-orig/fixincludes/configure gcc-15.1.0/fixincludes/configure
--- gcc-15.1.0-orig/fixincludes/configure	2026-01-03 16:23:07.429903900 -0700
+++ gcc-15.1.0/fixincludes/configure	2026-01-03 16:24:13.401084500 -0700
@@ -4820,6 +4820,7 @@
 	i?86-*-msdosdjgpp* | \
 	i?86-*-mingw32* | \
 	x86_64-*-mingw32* | \
+  aarch64-*-mingw32* | \
 	*-*-beos* | \
         *-*-*vms*)
 		TARGET=twoprocess
