diff --git a/CMakeLists.txt b/CMakeLists.txt
index d72bf0c..55cd4f2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,6 +83,11 @@ if (ARGS_BUILD_UNITTESTS)
         target_compile_options(argstest PRIVATE -Wall -Wextra -Werror -pedantic -Wshadow -Wunused-parameter)
     endif ()
 
+    # https://github.com/catchorg/Catch2/issues/3076
+    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 22)
+        target_compile_options(argstest PRIVATE -Wno-c2y-extensions)
+    endif ()
+
     add_executable(argstest-multiple-inclusion test/multiple_inclusion_1.cxx test/multiple_inclusion_2.cxx)
 
     target_link_libraries(argstest-multiple-inclusion args)
