--- laszip-src-3.4.3/src/lasindex.cpp.orig	2019-11-11 23:31:12.000000000 +0300
+++ laszip-src-3.4.3/src/lasindex.cpp	2019-11-13 10:07:01.228278000 +0300
@@ -342,7 +342,7 @@
     name[strlen(name)-2] = 'a';
     name[strlen(name)-1] = 'x';
   }
-#ifdef _MSC_VER
+#ifdef _WIN32
   wchar_t* utf16_name = UTF8toUTF16(name);
   FILE* file = _wfopen(utf16_name, L"rb");
   delete [] utf16_name;
@@ -385,7 +385,7 @@
 
   lasreader->close();
 
-#ifdef _MSC_VER
+#ifdef _WIN32
   wchar_t* utf16_file_name = UTF8toUTF16(file_name);
   FILE* file = _wfopen(utf16_file_name, L"rb");
   delete [] utf16_file_name;
@@ -455,7 +455,7 @@
   fclose(file);
 
   ByteStreamOut* bytestreamout;
-#ifdef _MSC_VER
+#ifdef _WIN32
   utf16_file_name = UTF8toUTF16(file_name);
   file = _wfopen(utf16_file_name, L"rb+");
   delete [] utf16_file_name;
@@ -535,7 +535,7 @@
     name[strlen(name)-2] = 'a';
     name[strlen(name)-1] = 'x';
   }
-#ifdef _MSC_VER
+#ifdef _WIN32
   wchar_t* utf16_name = UTF8toUTF16(name);
   FILE* file = _wfopen(utf16_name, L"wb");
   delete [] utf16_name;
--- laszip-src-3.4.3/src/laszip_dll.cpp.orig	2019-11-13 10:03:27.784703000 +0300
+++ laszip-src-3.4.3/src/laszip_dll.cpp	2019-11-13 10:07:13.037498800 +0300
@@ -2915,7 +2915,7 @@
 
     // open the file
 
-#ifdef _MSC_VER
+#ifdef _WIN32
     wchar_t* utf16_file_name = UTF8toUTF16(file_name);
     laszip_dll->file = _wfopen(utf16_file_name, L"wb");
     delete [] utf16_file_name;
@@ -4330,7 +4330,7 @@
 
     // open the file
 
-#ifdef _MSC_VER
+#ifdef _WIN32
     wchar_t* utf16_file_name = UTF8toUTF16(file_name);
     laszip_dll->file = _wfopen(utf16_file_name, L"rb");
     delete [] utf16_file_name;
--- laszip-src-3.4.3/src/mydefs.hpp.orig	2019-11-13 10:13:17.362538800 +0300
+++ laszip-src-3.4.3/src/mydefs.hpp	2019-11-13 10:13:23.228149100 +0300
@@ -246,7 +246,7 @@
   to[7] = from[0];
 }
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #include <windows.h>
 wchar_t* UTF8toUTF16(const char* utf8);
 #endif
--- laszip-src-3.4.3/src/mydefs.cpp.orig	2019-11-13 10:14:41.619286800 +0300
+++ laszip-src-3.4.3/src/mydefs.cpp	2019-11-13 10:14:44.754892300 +0300
@@ -30,7 +30,7 @@
 */
 #include "mydefs.hpp"
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #include <windows.h>
 wchar_t* UTF8toUTF16(const char* utf8)
 {
