diff -c -r LASzip-2.2.0.orig/CMakeLists.txt LASzip-2.2.0/CMakeLists.txt
*** LASzip-2.2.0.orig/CMakeLists.txt	2013-08-05 15:05:19.000000000 -0300
--- LASzip-2.2.0/CMakeLists.txt	2016-03-13 19:42:53.853767500 -0300
***************
*** 78,84 ****
  ###############################################################################
  # Installation settings
  
! if(WIN32)
    set(DEFAULT_LIB_SUBDIR lib)
    set(DEFAULT_DATA_SUBDIR .)
    set(DEFAULT_INCLUDE_SUBDIR include)
--- 78,84 ----
  ###############################################################################
  # Installation settings
  
! if(WIN32 AND NOT MINGW)
    set(DEFAULT_LIB_SUBDIR lib)
    set(DEFAULT_DATA_SUBDIR .)
    set(DEFAULT_INCLUDE_SUBDIR include)
***************
*** 154,159 ****
  
  add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
  
! if (WIN32)
      include(BuildOSGeo4W)
! endif(WIN32)
--- 154,159 ----
  
  add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
  
! if (WIN32 AND NOT MINGW)
      include(BuildOSGeo4W)
! endif(WIN32 AND NOT MINGW)
