diff -bur sleef-3.9.0-orig/Configure.cmake sleef-3.9.0/Configure.cmake
--- sleef-3.9.0-orig/Configure.cmake	2026-04-10 23:04:49.737965400 -0600
+++ sleef-3.9.0/Configure.cmake	2026-04-10 23:05:43.497245600 -0600
@@ -194,7 +194,7 @@
   set(SLEEF_ARCH_X86 ON CACHE INTERNAL "True for x86 architecture.")
 
   set(CLANG_FLAGS_ENABLE_PURECFMA_SCALAR "-mavx2;-mfma")
-elseif(SLEEF_TARGET_PROCESSOR MATCHES "aarch64|arm64")
+elseif(SLEEF_TARGET_PROCESSOR MATCHES "aarch64|arm64|ARM64")
   set(SLEEF_ARCH_AARCH64 ON CACHE INTERNAL "True for Aarch64 architecture.")
   # Aarch64 requires support for advsimdfma4
   set(COMPILER_SUPPORTS_ADVSIMD 1)
