--- aspell6-en-7.1-0/configure.orig	2015-07-27 12:03:47.683663100 +0300
+++ aspell6-en-7.1-0/configure	2015-07-27 12:05:05.653865300 +0300
@@ -73,10 +73,12 @@
 
 echo $ECHO_N "Finding Dictionary file location ... $ECHO_C"
 dictdir=`$ASPELL dump config dict-dir`
+dictdir=`cygpath -u $dictdir`
 echo $dictdir
 
 echo $ECHO_N "Finding Data file location ... $ECHO_C"
 datadir=`$ASPELL dump config data-dir`
+datadir=`cygpath -u $datadir`
 echo $datadir
 
 echo "ASPELL = `which $ASPELL`" > Makefile
