From f99a97037c4ae95ec48a2ae3ac8a3f76d9220cba Mon Sep 17 00:00:00 2001
From: Alberto Fanjul <albertofanjul@gmail.com>
Date: Wed, 28 Apr 2021 16:10:08 +0200
Subject: [PATCH] Use gettext 0.19

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a500c1a..7fd1c74 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,7 +37,7 @@ AC_SUBST(PACKAGE_LIBS)
 
 # i18n stuff
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.20])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 GETTEXT_PACKAGE=goocanvas3
 AC_SUBST(GETTEXT_PACKAGE)
-- 
2.30.1
