diff -urN a/src/w32-util.c b/src/w32-util.c
--- a/src/w32-util.c	2015-08-04 22:35:22.574099100 +0100
+++ b/src/w32-util.c	2015-08-04 22:35:22.948499800 +0100
@@ -42,6 +42,7 @@
 #endif
 #include <fcntl.h>
 #include <io.h>
+#include <share.h>
 
 #if __MINGW64_VERSION_MAJOR >= 2
 # define _WIN32_IE 0x0501 /* Required by mingw64 toolkit.  */
@@ -715,6 +716,19 @@
       return -1;
     }
 
+  /* mingw-w64's mkstemp implementation opens files with _O_TEMPORARY
+     meaning they get deleted when the last handle to them is closed.
+     While ReOpenFile () could be used to work around this, it is not
+     available until Vista. Instead, close and recreate. */
+  close (*fd);
+  *fd = _sopen(tmpname,
+          _O_RDWR | _O_CREAT | _O_EXCL | _O_BINARY,
+          _SH_DENYRW, _S_IREAD | _S_IWRITE);  
+  if (*fd < 0)
+    {
+      free (tmpname);
+      return -1;
+    }
   *name = tmpname;
   return 0;
 }
