diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2015-08-04 22:35:23.556900900 +0100
+++ b/configure.ac	2015-08-04 22:35:24.087301800 +0100
@@ -161,10 +161,10 @@
     *-mingw32ce*|*-mingw32*)
 	have_dosish_system=yes
         have_w32_system=yes
-        GPG_DEFAULT='c:\\gnupg\\gpg.exe'
-	GPGSM_DEFAULT='c:\\gnupg\\gpgsm.exe'
-        GPGCONF_DEFAULT='c:\\gnupg\\gpgconf.exe'
-        G13_DEFAULT='c:\\gnupg\\g13.exe'
+        GPG_DEFAULT='gpg.exe'
+	GPGSM_DEFAULT='gpgsm.exe'
+        GPGCONF_DEFAULT='gpgconf.exe'
+        G13_DEFAULT='g13.exe'
         #component_system='COM+'
 
 	AM_PATH_GLIB_2_0
diff -urN a/src/w32-util.c b/src/w32-util.c
--- a/src/w32-util.c	2015-08-04 22:35:23.525700800 +0100
+++ b/src/w32-util.c	2015-08-04 22:35:24.087301800 +0100
@@ -476,6 +476,7 @@
       gpg = find_program_in_dir (inst_dir, name);
     }
 
+#ifndef __MINGW32__
   /* 2. Try to find gpg.exe using that ancient registry key.  */
   if (!gpg)
     {
@@ -497,6 +497,7 @@
       name = default_gpg_name? default_gpg_name : "GNU\\GnuPG\\gpg.exe";
       gpg = find_program_at_standard_place (name);
     }
+#endif
 
   /* 4. Print a debug message if not found.  */
   if (!gpg)
@@ -523,6 +523,7 @@
       gpgconf = find_program_in_dir (inst_dir, name);
     }
 
+#ifndef __MINGW32__
   /* 2. Try to find gpgconf.exe from GnuPG >= 2.1 below CSIDL_PROGRAM_FILES. */
   if (!gpgconf)
     {
@@ -552,6 +552,7 @@
     {
       gpgconf = find_program_at_standard_place ("GNU\\GnuPG\\gpgconf.exe");
     }
+#endif
 
   /* 5. Print a debug message if not found.  */
   if (!gpgconf)
