diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2015-08-04 22:35:23.556900900 +0100
+++ b/configure.ac	2015-08-04 22:35:24.087301800 +0100
@@ -161,10 +161,10 @@
     *-mingw32ce*|*-mingw32*)
 	have_dosish_system=yes
         have_w32_system=yes
-        GPG_DEFAULT='c:\\gnupg\\gpg.exe'
-	GPGSM_DEFAULT='c:\\gnupg\\gpgsm.exe'
-        GPGCONF_DEFAULT='c:\\gnupg\\gpgconf.exe'
-        G13_DEFAULT='c:\\gnupg\\g13.exe'
+        GPG_DEFAULT='gpg.exe'
+	GPGSM_DEFAULT='gpgsm.exe'
+        GPGCONF_DEFAULT='gpgconf.exe'
+        G13_DEFAULT='g13.exe'
         #component_system='COM+'
 
 	AM_PATH_GLIB_2_0
diff -urN a/src/w32-util.c b/src/w32-util.c
--- a/src/w32-util.c	2015-08-04 22:35:23.525700800 +0100
+++ b/src/w32-util.c	2015-08-04 22:35:24.087301800 +0100
@@ -419,6 +419,7 @@
         return result;
     }
 
+#ifndef __MINGW32__
   dir = read_w32_registry_string ("HKEY_LOCAL_MACHINE",
 				  "Software\\GNU\\GnuPG",
 				  "Install Directory");
@@ -428,6 +429,7 @@
       free (dir);
       return result;
     }
+#endif
   return NULL;
 }
 
@@ -504,13 +506,15 @@
   inst_dir = _gpgme_get_inst_dir ();
   gpg = find_program_in_inst_dir
     (inst_dir,
-     default_gpg_name? get_basename (default_gpg_name) : "gpg.exe");
+     default_gpg_name ? get_basename (default_gpg_name) : "gpg.exe");
   if (!gpg)
     {
-      name = (default_gpg_name? default_gpg_name
+#ifndef __MINGW32__
+      name = (default_gpg_name ? default_gpg_name
               /* */           : "GNU\\GnuPG\\gpg.exe");
       gpg = find_program_at_standard_place (name);
       if (!gpg)
+#endif
         _gpgme_debug (DEBUG_ENGINE, "_gpgme_get_gpg_path: '%s' not found",
                       name);
     }
@@ -530,13 +534,15 @@
   inst_dir = _gpgme_get_inst_dir ();
   gpgconf = find_program_in_inst_dir
     (inst_dir,
-     default_gpgconf_name? get_basename (default_gpgconf_name) : "gpgconf.exe");
+     default_gpgconf_name ? get_basename (default_gpgconf_name) : "gpgconf.exe");
   if (!gpgconf)
     {
-      name = (default_gpgconf_name? default_gpgconf_name
+#ifndef __MINGW32__
+      name = (default_gpgconf_name ? default_gpgconf_name
               /* */               : "GNU\\GnuPG\\gpgconf.exe");
       gpgconf = find_program_at_standard_place (name);
       if (!gpgconf)
+#endif
         _gpgme_debug (DEBUG_ENGINE, "_gpgme_get_gpgconf_path: '%s' not found",
                       name);
     }
@@ -554,9 +560,11 @@
   LOCK (get_path_lock);
   if (!w32spawn_program)
     w32spawn_program = find_program_in_inst_dir (inst_dir,"gpgme-w32spawn.exe");
+#ifndef __MINGW32__
   if (!w32spawn_program)
     w32spawn_program
       = find_program_at_standard_place ("GNU\\GnuPG\\gpgme-w32spawn.exe");
+#endif
   UNLOCK (get_path_lock);
   return w32spawn_program;
 }
