--- gpgme-1.13.0.orig/lang/python/setup.py.in	2019-04-07 20:10:01.767088400 -0400
+++ gpgme-1.13.0/lang/python/setup.py.in	2019-04-07 20:18:35.986816000 -0400
@@ -91,7 +91,7 @@
 
 # Adjust include and library locations in case of win32
 uname_s = os.popen('uname -s').read()
-if uname_s.startswith('MINGW32'):
+if uname_s.startswith('MINGW'):
     mnts = [
         x.split()[0:3:2] for x in os.popen('mount').read().split('\n') if x
     ]
