--- gpgme-1.24.0/configure.ac.orig	2024-11-11 07:40:12.665109100 +0100
+++ gpgme-1.24.0/configure.ac	2024-11-11 07:40:31.367107300 +0100
@@ -614,7 +614,7 @@
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
-                     [[2.7],[3.6],[3.8],[3.9],[3.10],[3.11],[3.12],[all]],
+                     [[3.6],[3.8],[3.9],[3.10],[3.11],[3.12],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS
