--- gpgme-1.14.0/configure.ac.orig	2020-09-05 15:02:43.720753300 +0200
+++ gpgme-1.14.0/configure.ac	2020-09-05 15:03:38.558261200 +0200
@@ -590,8 +590,8 @@
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
-                     [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],
-                      [3.11],[3.12],[all]],
+                     [[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],
+                      [3.11],[3.12],[3.13],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS