--- gpgme-1.24.0/lang/cpp/src/Makefile.am.orig	2024-11-11 07:48:27.905978800 +0100
+++ gpgme-1.24.0/lang/cpp/src/Makefile.am	2024-11-11 08:04:53.007311700 +0100
@@ -96,7 +96,7 @@
 
 $(copied_headers): Makefile.am
 	mkdir -p $(builddir)/gpgme++/interfaces
-	echo -n "#include \"$(abs_srcdir)" > "$@"
+	echo -n "#include \"$(shell cygpath -m "$(abs_srcdir)")" > "$@"
 	echo -n "$@" | sed "s/gpgme++//" >> "$@"
 	echo "\"" >> "$@"
 
 