diff --git a/configure.ac b/configure.ac
index 1813cc57..1d9ceb4b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1041,7 +1041,8 @@ AC_CONFIG_FILES(Makefile src/Makefile
                 src/versioninfo.rc
                 src/gpgme.pc
                 src/gpgme-glib.pc
-                src/gpgme.h)
+                src/gpgme.h
+                lang/cpp/src/data.h)
 AC_CONFIG_FILES(src/gpgme-config, chmod +x src/gpgme-config)
 AC_CONFIG_FILES(lang/cpp/Makefile lang/cpp/src/Makefile)
 AC_CONFIG_FILES(lang/cpp/tests/Makefile)
diff --git a/lang/cpp/src/Makefile.am b/lang/cpp/src/Makefile.am
index 1e6bdc28..ccfb16b0 100644
--- a/lang/cpp/src/Makefile.am
+++ b/lang/cpp/src/Makefile.am
@@ -20,7 +20,7 @@
 # 02111-1307, USA
 
 EXTRA_DIST = GpgmeppConfig.cmake.in.in GpgmeppConfigVersion.cmake.in \
-             gpgmepp_version.h.in GpgmeppConfig-w32.cmake.in.in
+             gpgmepp_version.h.in data.h.in GpgmeppConfig-w32.cmake.in.in
 
 lib_LTLIBRARIES = libgpgmepp.la
 
@@ -42,7 +42,7 @@ main_sources = \
     util.cpp
 
 gpgmepp_headers = \
-    configuration.h context.h data.h decryptionresult.h \
+    configuration.h context.h decryptionresult.h \
     defaultassuantransaction.h editinteractor.h encryptionresult.h \
     engineinfo.h error.h eventloopinteractor.h exception.h global.h \
     gpgadduserideditinteractor.h gpgagentgetinfoassuantransaction.h \
@@ -68,7 +68,7 @@ interface_headers= \
 gpgmeppincludedir = $(includedir)/gpgme++
 gpgmeppinclude_HEADERS = $(gpgmepp_headers)
 nobase_gpgmeppinclude_HEADERS = $(interface_headers)
-nodist_gpgmeppinclude_HEADERS = gpgmepp_version.h
+nodist_gpgmeppinclude_HEADERS = gpgmepp_version.h data.h
 
 libgpgmepp_la_SOURCES = $(main_sources) $(gpgmepp_headers) context_vanilla.cpp \
                         $(interface_headers) $(private_gpgmepp_headers)
@@ -116,4 +116,4 @@ install-data-local: install-cmake-files
 uninstall-local: uninstall-cmake-files
 
 CLEANFILES = GpgmeppConfig.cmake GpgmeppConfigVersion.cmake \
-             gpgmepp_version.h GpgmeppConfig.cmake.in
+             gpgmepp_version.h data.h GpgmeppConfig.cmake.in
diff --git a/lang/cpp/src/data.cpp b/lang/cpp/src/data.cpp
index 2782aa79..1da0df52 100644
--- a/lang/cpp/src/data.cpp
+++ b/lang/cpp/src/data.cpp
@@ -222,17 +222,17 @@ GpgME::Error GpgME::Data::setFileName(const std::string &name)
     return Error(gpgme_data_set_file_name(d->data, name.c_str()));
 }
 
-ssize_t GpgME::Data::read(void *buffer, size_t length)
+gpgme_ssize_t GpgME::Data::read(void *buffer, size_t length)
 {
     return gpgme_data_read(d->data, buffer, length);
 }
 
-ssize_t GpgME::Data::write(const void *buffer, size_t length)
+gpgme_ssize_t GpgME::Data::write(const void *buffer, size_t length)
 {
     return gpgme_data_write(d->data, buffer, length);
 }
 
-off_t GpgME::Data::seek(off_t offset, int whence)
+gpgme_off_t GpgME::Data::seek(gpgme_off_t offset, int whence)
 {
     return gpgme_data_seek(d->data, offset, whence);
 }
diff --git a/lang/cpp/src/data.h b/lang/cpp/src/data.h.in
similarity index 93%
rename from lang/cpp/src/data.h
rename to lang/cpp/src/data.h.in
index df8607e7..1c909ef8 100644
--- a/lang/cpp/src/data.h
+++ b/lang/cpp/src/data.h.in
@@ -26,12 +26,14 @@
 #include "global.h"
 #include "key.h"
 
-#include <sys/types.h> // for size_t, off_t
 #include <cstdint> // unit64_t
 #include <cstdio> // FILE
 #include <algorithm>
 #include <memory>
 
+/* System specific typedefs.  */
+@INSERT__TYPEDEFS_FOR_GPGME_H@
+
 namespace GpgME
 {
 
@@ -109,9 +111,9 @@
     Error setFileName(const char *name);
     Error setFileName(const std::string &name);
 
-    ssize_t read(void *buffer, size_t length);
-    ssize_t write(const void *buffer, size_t length);
-    off_t seek(off_t offset, int whence);
+    gpgme_ssize_t read(void *buffer, size_t length);
+    gpgme_ssize_t write(const void *buffer, size_t length);
+    gpgme_off_t seek(gpgme_off_t offset, int whence);
 
     /* Convenience function to do a seek (0, SEEK_SET).  */
     Error rewind();
diff --git a/lang/qt/src/Makefile.am b/lang/qt/src/Makefile.am
index 32251424..60f18b10 100644
--- a/lang/qt/src/Makefile.am
+++ b/lang/qt/src/Makefile.am
@@ -307,7 +307,7 @@ camelcaseinclude_HEADERS = $(camelcase_headers)
 nodist_qgpgmeinclude_HEADERS = qgpgme_version.h
 
 if WANT_QT5
-AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
+AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/lang/cpp/src -I$(top_builddir)/src \
               @GPGME_QT5_CFLAGS@ @GPG_ERROR_CFLAGS@ @LIBASSUAN_CFLAGS@ \
               -DBUILDING_QGPGME -Wsuggest-override \
               -Wzero-as-null-pointer-constant
@@ -320,7 +320,7 @@ libqgpgme_la_LDFLAGS = -no-undefined -version-info \
                        @LIBQGPGME_LT_CURRENT@:@LIBQGPGME_LT_REVISION@:@LIBQGPGME_LT_AGE@
 endif
 if WANT_QT6
-AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
+AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/lang/cpp/src -I$(top_builddir)/src \
               @GPGME_QT6_CFLAGS@ @GPG_ERROR_CFLAGS@ @LIBASSUAN_CFLAGS@ \
               -DBUILDING_QGPGME -Wsuggest-override \
               -Wzero-as-null-pointer-constant
diff --git a/lang/qt/tests/Makefile.am b/lang/qt/tests/Makefile.am
index bfe77ad5..7856a4f6 100644
--- a/lang/qt/tests/Makefile.am
+++ b/lang/qt/tests/Makefile.am
@@ -51,7 +51,7 @@ LDADD = ../../cpp/src/libgpgmepp.la ../src/libqgpgme.la \
         ../../../src/libgpgme.la @GPGME_QT5_LIBS@ @GPG_ERROR_LIBS@ \
         @GPGME_QT5TEST_LIBS@ @LDADD_FOR_TESTS_KLUDGE@ -lstdc++
 
-AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
+AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/lang/cpp/src -I$(top_builddir)/src \
               @GPG_ERROR_CFLAGS@ @GPGME_QT5_CFLAGS@ @GPG_ERROR_CFLAGS@ \
               @LIBASSUAN_CFLAGS@ @GPGME_QT5TEST_CFLAGS@ -DBUILDING_QGPGME \
               -I$(top_srcdir)/lang/qt/src \
@@ -62,7 +62,7 @@ LDADD = ../../cpp/src/libgpgmepp.la ../src/libqgpgmeqt6.la \
         ../../../src/libgpgme.la @GPGME_QT6_LIBS@ @GPG_ERROR_LIBS@ \
         @GPGME_QT6TEST_LIBS@ @LDADD_FOR_TESTS_KLUDGE@ -lstdc++
 
-AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/src \
+AM_CPPFLAGS = -I$(top_srcdir)/lang/cpp/src -I$(top_builddir)/lang/cpp/src -I$(top_builddir)/src \
               @GPG_ERROR_CFLAGS@ @GPGME_QT6_CFLAGS@ @GPG_ERROR_CFLAGS@ \
               @LIBASSUAN_CFLAGS@ @GPGME_QT6TEST_CFLAGS@ -DBUILDING_QGPGME \
               -I$(top_srcdir)/lang/qt/src \
