--- gpgme-1.5.4/src/w32-util.c.orig	2015-04-30 09:13:53.846200000 +0300
+++ gpgme-1.5.4/src/w32-util.c	2015-04-30 09:14:42.224800000 +0300
@@ -595,6 +595,7 @@
 static const char letters[] =
 "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
 
+#if __MINGW64_VERSION_MAJOR < 5
 /* Generate a temporary file name based on TMPL.  TMPL must match the
    rules for mk[s]temp (i.e. end in "XXXXXX").  The name constructed
    does not exist at the time of the call to mkstemp.  TMPL is
@@ -677,7 +678,7 @@
   gpg_err_set_errno (EEXIST);
   return -1;
 }
-
+#endif
 
 int
 _gpgme_mkstemp (int *fd, char **name)
