--- a/cmake/QtDocsHelpers.cmake
+++ b/cmake/QtDocsHelpers.cmake
@@ -152,7 +152,7 @@
         set(executable_suffix "")
     endif()
 
-    set(qdoc_bin "${doc_tools_bin}/qdoc${executable_suffix}")
+    set(qdoc_bin "${doc_tools_bin}/qdoc-qt6${executable_suffix}")
     set(qtattributionsscanner_bin "${doc_tools_libexec}/qtattributionsscanner${executable_suffix}")
     set(qhelpgenerator_bin "${doc_tools_libexec}/qhelpgenerator${executable_suffix}")
 
