diff -ur starpu-1.4.0/socl/src/cl_buildprogram.c starpu-1.4.0-new/socl/src/cl_buildprogram.c
--- starpu-1.4.0/socl/src/cl_buildprogram.c	2023-03-27 10:12:59.000000000 +0200
+++ starpu-1.4.0-new/socl/src/cl_buildprogram.c	2023-03-30 10:40:46.467421400 +0200
@@ -80,7 +80,7 @@
 		 cl_uint              num_devices,
 		 const cl_device_id * device_list,
 		 const char *         options,
-		 void (*pfn_notify)(cl_program program, void * user_data),
+		 void (CL_CALLBACK *pfn_notify)(cl_program program, void * user_data),
 		 void *               user_data)
 {
 	struct bp_data *data;
diff -ur starpu-1.4.0/socl/src/cl_createcontext.c starpu-1.4.0-new/socl/src/cl_createcontext.c
--- starpu-1.4.0/socl/src/cl_createcontext.c	2023-03-27 10:12:59.000000000 +0200
+++ starpu-1.4.0-new/socl/src/cl_createcontext.c	2023-03-30 10:33:26.239963400 +0200
@@ -38,7 +38,7 @@
 soclCreateContext(const cl_context_properties * properties,
 		  cl_uint                       num_devices,
 		  const cl_device_id *          devices,
-		  void (*pfn_notify)(const char *, const void *, size_t, void *),
+		  void (CL_CALLBACK *pfn_notify)(const char *, const void *, size_t, void *),
 		  void *                        user_data,
 		  cl_int *                      errcode_ret)
 {
diff -ur starpu-1.4.0/socl/src/cl_createcontextfromtype.c starpu-1.4.0-new/socl/src/cl_createcontextfromtype.c
--- starpu-1.4.0/socl/src/cl_createcontextfromtype.c	2023-03-27 10:12:59.000000000 +0200
+++ starpu-1.4.0-new/socl/src/cl_createcontextfromtype.c	2023-03-30 10:41:00.807622200 +0200
@@ -22,7 +22,7 @@
 CL_API_ENTRY cl_context CL_API_CALL
 soclCreateContextFromType(const cl_context_properties * properties,
 			  cl_device_type                device_type,
-			  void (*pfn_notify)(const char *, const void *, size_t, void *),
+			  void (CL_CALLBACK *pfn_notify)(const char *, const void *, size_t, void *),
 			  void *                        user_data,
 			  cl_int *                      errcode_ret)
 {
diff -ur starpu-1.4.0/socl/src/socl.h starpu-1.4.0-new/socl/src/socl.h
--- starpu-1.4.0/socl/src/socl.h	2023-03-27 10:12:59.000000000 +0200
+++ starpu-1.4.0-new/socl/src/socl.h	2023-03-30 10:40:36.315942400 +0200
@@ -113,7 +113,7 @@
 {
 	CL_ENTITY;
 
-	void (*pfn_notify)(const char *, const void *, size_t, void *);
+	void (CL_CALLBACK *pfn_notify)(const char *, const void *, size_t, void *);
 	void *user_data;
 
 	/* Associated devices */
@@ -330,14 +330,14 @@
 soclCreateContext(const cl_context_properties * /* properties */,
 		  cl_uint                       /* num_devices */,
 		  const cl_device_id *          /* devices */,
-		  void (*pfn_notify)(const char *, const void *, size_t, void *) /* pfn_notify */,
+		  void (CL_CALLBACK *pfn_notify)(const char *, const void *, size_t, void *) /* pfn_notify */,
 		  void *                        /* user_data */,
 		  cl_int *                      /* errcode_ret */) CL_API_SUFFIX__VERSION_1_0;
 
 extern CL_API_ENTRY cl_context CL_API_CALL
 soclCreateContextFromType(const cl_context_properties * /* properties */,
 			  cl_device_type                /* device_type */,
-			  void (*pfn_notify)(const char *, const void *, size_t, void *) /* pfn_notify */,
+			  void (CL_CALLBACK *pfn_notify)(const char *, const void *, size_t, void *) /* pfn_notify */,
 			  void *                        /* user_data */,
 			  cl_int *                      /* errcode_ret */) CL_API_SUFFIX__VERSION_1_0;
 
@@ -487,7 +487,7 @@
 		 cl_uint              /* num_devices */,
 		 const cl_device_id * /* device_list */,
 		 const char *         /* options */,
-		 void (*pfn_notify)(cl_program /* program */, void * /* user_data */),
+		 void (CL_CALLBACK *pfn_notify)(cl_program /* program */, void * /* user_data */),
 		 void *               /* user_data */) CL_API_SUFFIX__VERSION_1_0;
 
 extern CL_API_ENTRY cl_int CL_API_CALL
