diff --git a/socl/src/socl.h b/socl/src/socl.h
index e9c68b8e48..d481136664 100644
--- a/socl/src/socl.h
+++ b/socl/src/socl.h
@@ -749,17 +749,19 @@ soclEnqueueWaitForEvents(cl_command_queue /* command_queue */,
 extern CL_API_ENTRY cl_int CL_API_CALL
 soclEnqueueBarrier(cl_command_queue /* command_queue */) CL_API_SUFFIX__VERSION_1_0;
 
-extern CL_API_ENTRY cl_int soclEnqueueMarkerWithWaitList(cl_command_queue /* command_queue */,
-							 cl_uint           /* num_events_in_wait_list */,
-							 const cl_event *  /* event_wait_list */,
-							 cl_event *        /* event */
-							 ) CL_API_SUFFIX__VERSION_1_2;
-
-extern CL_API_ENTRY cl_int soclEnqueueBarrierWithWaitList(cl_command_queue /* command_queue */,
-							  cl_uint           /* num_events_in_wait_list */,
-							  const cl_event *  /* event_wait_list */,
-							  cl_event *        /* event */
-							  ) CL_API_SUFFIX__VERSION_1_2;
+extern CL_API_ENTRY cl_int CL_API_CALL
+soclEnqueueMarkerWithWaitList(cl_command_queue /* command_queue */,
+			      cl_uint           /* num_events_in_wait_list */,
+			      const cl_event *  /* event_wait_list */,
+			      cl_event *        /* event */
+			      ) CL_API_SUFFIX__VERSION_1_2;
+
+extern CL_API_ENTRY cl_int CL_API_CALL
+soclEnqueueBarrierWithWaitList(cl_command_queue /* command_queue */,
+			       cl_uint           /* num_events_in_wait_list */,
+			       const cl_event *  /* event_wait_list */,
+			       cl_event *        /* event */
+			       ) CL_API_SUFFIX__VERSION_1_2;
 
 /* Extension function access
  *
