--- graphite2-1.3.15/tests/comparerenderer/CMakeLists.txt.orig	2026-06-03 22:49:58.547282400 +0200
+++ graphite2-1.3.15/tests/comparerenderer/CMakeLists.txt	2026-06-03 22:50:06.939207600 +0200
@@ -27,7 +27,7 @@
     add_definitions(-D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS)
     add_custom_target(${PROJECT_NAME}_copy_dll ALL
     COMMAND ${CMAKE_COMMAND} -E copy_if_different ${graphite2_core_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}graphite2${CMAKE_SHARED_LIBRARY_SUFFIX} ${PROJECT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
-    add_dependencies(${PROJECT_NAME}_copy_dll graphite2 iconv comparerenderer)
+    add_dependencies(${PROJECT_NAME}_copy_dll graphite2 comparerenderer)
 endif()
 
 macro(comparetest TESTNAME FONTFILE TEXTFILE)
