Index: mpxout.w
===================================================================
diff --git a/texk/web2c/mplibdir/mpxout.w b/texk/web2c/mplibdir/mpxout.w
--- a/texk/web2c/mplibdir/mpxout.w	(revision 58686)
+++ b/texk/web2c/mplibdir/mpxout.w	(working copy)
@@ -3869,7 +3869,7 @@
   char *wrk, *p;
 #ifdef _WIN32
   struct _finddata_t c_file;
-  long hFile;
+  intptr_t hFile;
 #else
   struct dirent *entry;
   DIR *d;
