diff --git a/texk/dvisvgm/dvisvgm-src/src/DVIActions.hpp b/texk/dvisvgm/dvisvgm-src/src/DVIActions.hpp
index 8e2e2a08..ed3c012f 100644
--- a/texk/dvisvgm/dvisvgm-src/src/DVIActions.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/DVIActions.hpp
@@ -21,6 +21,7 @@
 #ifndef DVIACTIONS_HPP
 #define DVIACTIONS_HPP
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/texk/dvisvgm/dvisvgm-src/src/Pair.hpp b/texk/dvisvgm/dvisvgm-src/src/Pair.hpp
index c1fdd5b6..176183ec 100644
--- a/texk/dvisvgm/dvisvgm-src/src/Pair.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/Pair.hpp
@@ -22,6 +22,7 @@
 #define PAIR_HPP
 
 #include <cmath>
+#include <cstdint>
 #include <ostream>
 #include "macros.hpp"
 
