--- a/texk/dvisvgm/dvisvgm-src/libs/woff2/include/woff2/output.h
+++ b/texk/dvisvgm/dvisvgm-src/libs/woff2/include/woff2/output.h
@@ -13,6 +13,7 @@ FILE *Poptr;
 #include <cstring>
 #include <memory>
 #include <string>
+#include <cstdint>
 
 namespace woff2 {
 
--- a/texk/dvisvgm/dvisvgm-src/src/TFM.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/TFM.hpp
@@ -21,6 +21,7 @@
 #ifndef TFM_HPP
 #define TFM_HPP
 
+#include <cstdint>
 #include <istream>
 #include <string>
 #include <vector>
