--- a/texk/dvisvgm/dvisvgm-src/libs/woff2/include/woff2/output.h
+++ b/texk/dvisvgm/dvisvgm-src/libs/woff2/include/woff2/output.h
@@ -13,6 +13,7 @@ FILE *Poptr;
 #include <cstring>
 #include <memory>
 #include <string>
+#include <cstdint>
 
 namespace woff2 {
 
--- a/texk/dvisvgm/dvisvgm-src/src/Character.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/Character.hpp
@@ -21,6 +21,7 @@ FILE *Poptr;
 #ifndef CHARACTER_HPP
 #define CHARACTER_HPP
 
+#include <cstdint>
 
 class Character {
 	public:
--- a/texk/dvisvgm/dvisvgm-src/src/DVIActions.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/DVIActions.hpp
@@ -21,6 +21,7 @@
 #ifndef DVIACTIONS_HPP
 #define DVIACTIONS_HPP
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/texk/dvisvgm/dvisvgm-src/src/FontMetrics.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/FontMetrics.hpp
@@ -21,6 +21,7 @@
 #ifndef FONTMETRICS_HPP
 #define FONTMETRICS_HPP
 
+#include <cstdint>
 #include <istream>
 #include <memory>
 #include <string>
--- a/texk/dvisvgm/dvisvgm-src/src/HashFunction.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/HashFunction.hpp
@@ -21,6 +21,7 @@
 #ifndef HASHFUNCTION_HPP
 #define HASHFUNCTION_HPP
 
+#include <cstdint>
 #include <istream>
 #include <memory>
 #include <string>
--- a/texk/dvisvgm/dvisvgm-src/src/Pair.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/Pair.hpp
@@ -22,6 +22,7 @@
 #define PAIR_HPP
 
 #include <cmath>
+#include <cstdint>
 #include <ostream>
 #include "macros.hpp"
 
--- a/texk/dvisvgm/dvisvgm-src/src/TFM.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/TFM.hpp
@@ -21,6 +21,7 @@
 #ifndef TFM_HPP
 #define TFM_HPP
 
+#include <cstdint>
 #include <istream>
 #include <string>
 #include <vector>
--- a/texk/dvisvgm/dvisvgm-src/src/utility.hpp
+++ b/texk/dvisvgm/dvisvgm-src/src/utility.hpp
@@ -13,6 +13,7 @@ FILE *Poptr;
 #include <config.h>
 #endif
 
+#include <cstdint>
 #include <iomanip>
 #include <functional>
 #include <memory>
