From 76a45f58e4c84e35634b095821b3268091bce144 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Wed, 23 Dec 2020 19:53:17 +0100
Subject: [PATCH] pkg-config: add missing Requires.private field for
 dav1d/rav1e

---
 CMakeLists.txt | 6 ++++++
 configure.ac   | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05afd5da..961186fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -124,6 +124,12 @@ endif()
 if (AOM_DECODER_FOUND OR AOM_ENCODER_FOUND)
     list(APPEND REQUIRES_PRIVATE "aom")
 endif()
+if (DAV1D_FOUND)
+    list(APPEND REQUIRES_PRIVATE "dav1d")
+endif()
+if (RAV1E_FOUND)
+    list(APPEND REQUIRES_PRIVATE "rav1e")
+endif()
 if (AOM_DECODER_FOUND OR DAV1D_FOUND)
     set(have_avif_decoder yes)
 else()
diff --git a/configure.ac b/configure.ac
index c545c894..ef5974da 100644
--- a/configure.ac
+++ b/configure.ac
@@ -250,6 +250,7 @@ if eval "test x$enable_rav1e != xno"; then
         AC_SUBST(rav1e_CFLAGS)
         AC_SUBST(rav1e_LIBS)
         have_avif_encoder="yes"
+        REQUIRES_PRIVATE="$REQUIRES_PRIVATE rav1e"
         have_rav1e="yes"
     ], [have_rav1e="no"])
 else
@@ -263,6 +264,7 @@ PKG_CHECK_MODULES([dav1d], [dav1d], [
     AC_SUBST(dav1d_CFLAGS)
     AC_SUBST(dav1d_LIBS)
     have_avif_decoder="yes"
+    REQUIRES_PRIVATE="$REQUIRES_PRIVATE dav1d"
     have_dav1d="yes"
 ], [have_dav1d="no"])
 AM_CONDITIONAL([HAVE_DAV1D], [test "x$have_dav1d" = "xyes"])
