--- libheif-1.11.0/gdk-pixbuf/CMakeLists.txt.orig	2021-01-24 20:56:53.000000000 +0100
+++ libheif-1.11.0/gdk-pixbuf/CMakeLists.txt	2021-04-29 10:25:26.355235000 +0200
@@ -1,4 +1,4 @@
-if(UNIX)
+if(UNIX OR MINGW)
   find_package(PkgConfig)
   find_package(Threads)
   pkg_check_modules(GDKPIXBUF2 gdk-pixbuf-2.0)
@@ -6,7 +6,7 @@
   if(GDKPIXBUF2_FOUND)
     execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} gdk-pixbuf-2.0 --variable gdk_pixbuf_moduledir --define-variable=prefix=${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE GDKPIXBUF2_MODULE_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-    add_library(pixbufloader-heif MODULE pixbufloader-heif.c)
+    add_library(pixbufloader-heif pixbufloader-heif.c)
 
     target_include_directories(pixbufloader-heif
                                PRIVATE
@@ -15,6 +15,6 @@
                                    ${libheif_SOURCE_DIR})
     target_link_libraries(pixbufloader-heif PUBLIC ${GDKPIXBUF2_LIBRARIES} heif)
 
-    install(TARGETS pixbufloader-heif LIBRARY DESTINATION ${GDKPIXBUF2_MODULE_DIR})
+    install(TARGETS pixbufloader-heif DESTINATION ${GDKPIXBUF2_MODULE_DIR})
   endif()
 endif()
