--- qwt-6.2.0/qwtconfig.pri.bak	2021-07-27 05:59:33.324111100 +0100
+++ qwt-6.2.0/qwtconfig.pri	2021-07-27 04:41:15.668293500 +0100
@@ -170,7 +170,7 @@
 # See http://www.freedesktop.org/wiki/Software/pkg-config/
 ######################################################################
 
-unix {
+unix|win32 {
 
     QWT_CONFIG     += QwtPkgConfig
 }
--- qwt-6.2.0/src/src.pro.bak	2021-07-27 05:55:14.566409800 +0100
+++ qwt-6.2.0/src/src.pro	2021-07-27 05:47:25.643064600 +0100
@@ -73,6 +73,7 @@
 
     QMAKE_PKGCONFIG_NAME = Qwt$${QWT_VER_MAJ}
     QMAKE_PKGCONFIG_DESCRIPTION = Qt Widgets for Technical Applications
+    QMAKE_PKGCONFIG_VERSION = $$QWT_VERSION
 
     QMAKE_PKGCONFIG_LIBDIR = $${QWT_INSTALL_LIBS}
     QMAKE_PKGCONFIG_INCDIR = $${QWT_INSTALL_HEADERS}
