--- a/plugins/auth/CMakeLists.txt
+++ b/plugins/auth/CMakeLists.txt
@@ -69,9 +69,11 @@
                 INCLUDES ${REF10_DIR}
                 LIBRARIES ${CRYPT_LIBS}
                 COMPILE_OPTIONS -DMYSQL_CLIENT=1)
+  if(TARGET client_ed25519)
   SET_PROPERTY(TARGET client_ed25519 PROPERTY C_STANDARD 11)
+  endif()
 
-  IF(HAVE_C_THREAD_LOCAL)
+  IF(HAVE_C_THREAD_LOCAL AND TARGET client_ed25519)
     TARGET_COMPILE_DEFINITIONS(client_ed25519 PRIVATE -DHAVE_THREAD_LOCAL)
   ENDIF()
 
