--- mariadb-connector-c-3.1.7-src/include/mysql.h.orig	2020-04-25 18:52:37.817258900 +0300
+++ mariadb-connector-c-3.1.7-src/include/mysql.h	2020-04-25 18:52:43.342827200 +0300
@@ -444,7 +444,7 @@
   MYSQL_CLIENT_PLUGIN_HEADER
 };
 
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 struct st_mysql_client_plugin * STDCALL
--- mariadb-connector-c-3.1.7-src/include/mysql/client_plugin.h.orig	2020-04-25 18:51:10.870443800 +0300
+++ mariadb-connector-c-3.1.7-src/include/mysql/client_plugin.h	2020-04-25 18:51:14.620895400 +0300
@@ -180,7 +180,7 @@
   @retval
   a pointer to the loaded plugin, or NULL in case of a failure
 */
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 
