From b12d16a625ddd1a639470fbb8dcccf04ef4d992d Mon Sep 17 00:00:00 2001
From: Zoltan Gyarmati <mr.zoltan.gyarmati@gmail.com>
Date: Sat, 7 Apr 2018 11:44:34 +0200
Subject: [PATCH] check also for __MINGW64__ definition to decide target arch

Signed-off-by: Zoltan Gyarmati <mr.zoltan.gyarmati@gmail.com>
---
 include/my_global.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/my_global.h b/include/my_global.h
index e596d45..df4c03f 100644
--- a/include/my_global.h
+++ b/include/my_global.h
@@ -770,7 +770,7 @@ typedef char		bool;	/* Ordinary boolean values 0 1 */
 /* Optimized store functions for Intel x86 */
 #define int1store(T,A) *((int8*) (T)) = (A)
 #define uint1korr(A)   (*(((uint8*)(A))))
-#if defined(__i386__) || defined(_WIN32)
+#if defined(__i386__) || defined(_WIN32) && !defined(__MINGW64__)
 #define sint2korr(A)	(*((int16 *) (A)))
 #define sint3korr(A)	((int32) ((((uchar) (A)[2]) & 128) ? \
 				  (((uint32) 255L << 24) | \
-- 
2.7.4

