--- mariadb_client-2.0.0-src/libmariadb/CMakeLists.txt.orig	2014-04-04 14:21:01.000000000 +0400
+++ mariadb_client-2.0.0-src/libmariadb/CMakeLists.txt	2014-04-18 12:08:42.907600000 +0400
@@ -149,31 +149,32 @@
   TARGET_LINK_LIBRARIES(mariadbclient ${SSL_LIBRARIES})
 ENDIF()
 
-ADD_LIBRARY(libmariadb SHARED ${LIBMARIADB_SOURCES})
-TARGET_LINK_LIBRARIES(libmariadb ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS} ${MATH_LIBRARY} ${ICONV_LIBRARIES})
+ADD_LIBRARY(mariadb SHARED ${LIBMARIADB_SOURCES})
+TARGET_LINK_LIBRARIES(mariadb ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS} ${MATH_LIBRARY} ${ICONV_LIBRARIES})
 
 IF(OPENSSL_FOUND)
-  TARGET_LINK_LIBRARIES(libmariadb ${SSL_LIBRARIES})
+  TARGET_LINK_LIBRARIES(mariadb ${SSL_LIBRARIES})
 ENDIF()
 
 IF(WIN32)
-  TARGET_LINK_LIBRARIES(libmariadb ws2_32 advapi32 kernel32)
+  TARGET_LINK_LIBRARIES(mariadb ws2_32 advapi32 kernel32)
   TARGET_LINK_LIBRARIES(mariadbclient ws2_32 advapi32 kernel32)
 ELSE()
-  TARGET_LINK_LIBRARIES(libmariadb m)
+  TARGET_LINK_LIBRARIES(mariadb m)
   TARGET_LINK_LIBRARIES(mariadbclient m)
 ENDIF()
 
 IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  TARGET_LINK_LIBRARIES (libmariadb "-Wl,--no-undefined")
-  TARGET_LINK_LIBRARIES (libmariadb "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/version_script.txt")
+  TARGET_LINK_LIBRARIES (mariadb "-Wl,--no-undefined")
+  TARGET_LINK_LIBRARIES (mariadb "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/version_script.txt")
 ENDIF()
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES PREFIX "")
+SET_TARGET_PROPERTIES(mariadb PROPERTIES PREFIX "")
 
-SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION 
+SET_TARGET_PROPERTIES(mariadb PROPERTIES VERSION 
    ${CPACK_PACKAGE_VERSION_MAJOR}
-   SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
+   SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR}
+   RUNTIME_OUTPUT_NAME libmariadb)
 
 #
 # Installation
@@ -209,10 +210,10 @@
 
 
 INSTALL(TARGETS
-          libmariadb mariadbclient
-          RUNTIME DESTINATION "lib/mariadb"
-          LIBRARY DESTINATION "lib/mariadb"
-          ARCHIVE DESTINATION "lib/mariadb")
+          mariadb mariadbclient
+          RUNTIME DESTINATION "bin"
+          LIBRARY DESTINATION "lib"
+          ARCHIVE DESTINATION "lib")
 IF(NOT WIN32 AND WITH_MYSQLCOMPAT)
   INSTALL(FILES "${CMAKE_BINARY_DIR}/libmariadb/libmysql${CMAKE_SHARED_LIBRARY_SUFFIX}"
                 "${CMAKE_BINARY_DIR}/libmariadb/libmysqlclient${CMAKE_SHARED_LIBRARY_SUFFIX}"
