--- libmariadb/dbug.c	2014-04-04 04:21:01.000000000 -0600
+++ libmariadb/dbug.c	2015-01-24 16:47:13.316693150 -0700
@@ -2050,11 +2050,7 @@
       }
       else
       {
-#ifdef _WIN32
-        if (fopen_s(&fp, name, append ? "a+" : "w"))
-#else
         if (!(fp= fopen(name, append ? "a+" : "w")))
-#endif
         {
           (void) fprintf(stderr, ERR_OPEN, cs->process, name);
           perror("");
--- libmariadb/my_fopen.c	2014-04-04 04:21:01.000000000 -0600
+++ libmariadb/my_fopen.c	2015-01-24 16:46:10.735766900 -0700
@@ -36,11 +36,7 @@
 		   FileName, Flags, MyFlags));
 
   make_ftype(type,Flags);
-#ifdef _WIN32
-  if (fopen_s(&fd, FileName, type) == 0)
-#else
   if ((fd = fopen(FileName, type)) != 0)
-#endif
   {
     /*
       The test works if MY_NFILE < 128. The problem is that fileno() is char
