diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-05-28 18:09:16.000000000 -0300
+++ b/CMakeLists.txt	2024-06-04 11:54:39.862731257 -0300
@@ -1,7 +1,7 @@
 # The name of our project is "CMINPACK". CMakeLists files in this project can
 # refer to the root source directory of the project as ${CMINPACK_SOURCE_DIR} and
 # to the root binary directory of the project as ${CMINPACK_BINARY_DIR}.
-cmake_minimum_required (VERSION 3.0.0)
+cmake_minimum_required (VERSION 3.14)
 
 if (NOT DEFINED CMAKE_BUILD_TYPE)
   set (CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
diff -Naur a/examples/runtest.cmake b/examples/runtest.cmake
--- a/examples/runtest.cmake	2024-05-28 18:09:16.000000000 -0300
+++ b/examples/runtest.cmake	2024-06-04 11:54:39.870731572 -0300
@@ -27,7 +27,7 @@
 endif()
 # result with reference
 execute_process(COMMAND "${CMAKE_COMMAND}"
-  -E compare_files  "${OUTPUT}" "${REFERENCE}"
+  -E compare_files --ignore-eol "${OUTPUT}" "${REFERENCE}"
   RESULT_VARIABLE RET)
 # if the test does not return 0, then fail it
 if(NOT ${RET} EQUAL 0)
