From 4fcbcccf381ce0987faa297173e4008b0490918f Mon Sep 17 00:00:00 2001
From: Jack Mustacato <jmustacato@knights.ucf.edu>
Date: Fri, 27 Oct 2017 15:05:23 -0400
Subject: [PATCH] Update minumum pycodestyle version to 2.3.0

---
 setup.cfg | 2 +-
 setup.py  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index fadd0f9..286856a 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -14,5 +14,5 @@ requires-dist =
     enum34; python_version<"3.4"
     configparser; python_version<"3.2"
     pyflakes >= 1.5.0, < 1.7.0
-    pycodestyle >= 2.0.0, < 2.4.0
+    pycodestyle >= 2.3.0, < 2.4.0
     mccabe >= 0.6.0, < 0.7.0
diff --git a/setup.py b/setup.py
index aa45e43..4833ff6 100644
--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,7 @@ requires = [
     # And in which releases we will update those ranges here:
     # http://flake8.pycqa.org/en/latest/internal/releases.html#releasing-flake8
     "pyflakes >= 1.5.0, < 1.7.0",
-    "pycodestyle >= 2.0.0, < 2.4.0",
+    "pycodestyle >= 2.3.0, < 2.4.0",
     "mccabe >= 0.6.0, < 0.7.0",
     "setuptools >= 30",
 ]
-- 
2.18.0

