From 0273ca561f0ad03adff41ce5d95a1ec31b10fe5a Mon Sep 17 00:00:00 2001
From: Jimi Cullen <jimifhcullen@gmail.com>
Date: Sun, 15 Apr 2018 10:51:18 +0000
Subject: [PATCH 1/4] Update pycodestyle dependency version.

---
 setup.cfg | 2 +-
 setup.py  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index 286856a..2d48e18 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -14,5 +14,5 @@ requires-dist =
     enum34; python_version<"3.4"
     configparser; python_version<"3.2"
     pyflakes >= 1.5.0, < 1.7.0
-    pycodestyle >= 2.3.0, < 2.4.0
+    pycodestyle >= 2.4.0, < 2.5.0
     mccabe >= 0.6.0, < 0.7.0
diff --git a/setup.py b/setup.py
index 4833ff6..08b13cf 100644
--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,7 @@ requires = [
     # And in which releases we will update those ranges here:
     # http://flake8.pycqa.org/en/latest/internal/releases.html#releasing-flake8
     "pyflakes >= 1.5.0, < 1.7.0",
-    "pycodestyle >= 2.3.0, < 2.4.0",
+    "pycodestyle >= 2.4.0, < 2.5.0",
     "mccabe >= 0.6.0, < 0.7.0",
     "setuptools >= 30",
 ]
-- 
2.18.0


From 85c503de32f81ed9859d902cbe20eb4d2e4e8d55 Mon Sep 17 00:00:00 2001
From: Jimi Cullen <jimifhcullen@gmail.com>
Date: Sun, 15 Apr 2018 10:52:07 +0000
Subject: [PATCH 2/4] Remove broken hook.

Remove broken hook to pycodestyle.break_around_binary_operator.
---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 08b13cf..9883795 100644
--- a/setup.py
+++ b/setup.py
@@ -108,7 +108,6 @@ setuptools.setup(
             PEP8_PLUGIN('module_imports_on_top_of_file'),
             PEP8_PLUGIN('compound_statements'),
             PEP8_PLUGIN('explicit_line_join'),
-            PEP8_PLUGIN('break_around_binary_operator'),
             PEP8_PLUGIN('comparison_to_singleton'),
             PEP8_PLUGIN('comparison_negative'),
             PEP8_PLUGIN('comparison_type'),
-- 
2.18.0


From 68782675b7f00c5d24c24e424efd1fbcb0705224 Mon Sep 17 00:00:00 2001
From: Jimi Cullen <jimifhcullen@gmail.com>
Date: Sun, 15 Apr 2018 14:02:49 +0000
Subject: [PATCH 3/4] Add new pycodestyle check hooks.

Add new pycodestyle check hooks for breaks around binary ops.
---
 setup.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/setup.py b/setup.py
index 9883795..c4862e9 100644
--- a/setup.py
+++ b/setup.py
@@ -108,6 +108,8 @@ setuptools.setup(
             PEP8_PLUGIN('module_imports_on_top_of_file'),
             PEP8_PLUGIN('compound_statements'),
             PEP8_PLUGIN('explicit_line_join'),
+            PEP8_PLUGIN('break_after_binary_operator'),
+            PEP8_PLUGIN('break_before_binary_operator'),
             PEP8_PLUGIN('comparison_to_singleton'),
             PEP8_PLUGIN('comparison_negative'),
             PEP8_PLUGIN('comparison_type'),
-- 
2.18.0


From e61d9d24668a3f98baf09db1d07dacfb9fc83beb Mon Sep 17 00:00:00 2001
From: Jimi Cullen <jimifhcullen@gmail.com>
Date: Sun, 15 Apr 2018 14:17:26 +0000
Subject: [PATCH 4/4] Add W504 to default ignore list.

---
 tox.ini | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tox.ini b/tox.ini
index 687a273..2e6a5bb 100644
--- a/tox.ini
+++ b/tox.ini
@@ -143,7 +143,7 @@ commands =
 # defaults to selecting all other errors so we do not need select=E,F,W,I,D
 # Once Flake8 3.0 is released and in a good state, we can use both and it will
 # work well \o/
-ignore = D203
+ignore = D203, W504
 exclude =
     .tox,
     .git,
-- 
2.18.0

