diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2026-09-03 18:20:57.000000000 +0000
+++ b/configure.ac	2026-09-08 05:42:09.684041300 +0000
@@ -2733,6 +2733,8 @@
 
 AC_SUBST(EXAMPLE_LIBS)
 
+AC_DEFINE_DIR(SWIG_BINDIR, [bindir], [Swig binary instalation directory])
+
 AC_CONFIG_FILES([
     Makefile
     Examples/Makefile
diff -Naur a/Source/Makefile.am b/Source/Makefile.am
--- a/Source/Makefile.am	2026-09-03 18:20:57.000000000 +0000
+++ b/Source/Makefile.am	2026-09-08 05:42:09.684041300 +0000
@@ -89,6 +89,7 @@
 		Swig/misc.c			\
 		Swig/naming.c			\
 		Swig/parms.c			\
+		Swig/pathtools.c		\
 		Swig/scanner.c			\
 		Swig/stype.c			\
 		Swig/symbol.c			\
diff -Naur a/Source/Modules/main.cxx b/Source/Modules/main.cxx
--- a/Source/Modules/main.cxx	2026-09-03 18:20:57.000000000 +0000
+++ b/Source/Modules/main.cxx	2026-09-08 05:42:09.684041300 +0000
@@ -26,6 +26,10 @@
 #include <errno.h>
 #include <limits.h>  // for INT_MAX
 
+extern "C" {
+#include "pathtools.h"
+}
+
 // Global variables
 
 static Language *lang = 0;  // Language method
@@ -916,7 +920,7 @@
       SwigLib = NewStringf("");  // Unexpected error
     }
     if (Len(SWIG_LIB_WIN_UNIX) > 0)
-      SwigLibWinUnix = NewString(SWIG_LIB_WIN_UNIX);  // Unix installation path using a drive letter (for msys/mingw)
+      SwigLibWinUnix = NewString(single_path_relocation(SWIG_BINDIR, SWIG_LIB)); // Unix installation path using a drive letter (for msys/mingw)
 #else
     SwigLib = NewString(SWIG_LIB);
 #endif
 
