--- swig-3.0.12/configure.ac.orig	2017-01-28 02:52:02.000000000 +0300
+++ swig-3.0.12/configure.ac	2017-03-09 09:12:31.784759400 +0300
@@ -628,7 +628,13 @@
     PYEPREFIX=`($PYTHON -c "import sys; sys.stdout.write(sys.exec_prefix)") 2>/dev/null`
     AC_MSG_RESULT($PYEPREFIX)
 
-    if test x"$PYOSNAME" = x"nt"; then
+    case $host in
+        *-*-mingw*) PYTHON_SO=.pyd
+                    IS_MINGW=true
+        ;;
+        *)IS_MINGW=false;;
+    esac
+    if test x"$PYOSNAME" = x"nt" && test x"$IS_MINGW" = x"false"; then
       # Windows installations are quite different to posix installations
       PYPREFIX=`echo "$PYPREFIX" | sed -e 's,\\\\,/,g'` # Forward slashes are easier to use and even work on Windows most of the time
       PYTHON_SO=.pyd
@@ -777,7 +783,13 @@
     PY3OSNAME=`($PYTHON3 -c "import sys, os; sys.stdout.write(os.name)")`
     AC_MSG_RESULT($PY3OSNAME)
 
-    if test x"$PY3OSNAME" = x"nt"; then
+    case $host in
+        *-*-mingw*) PYTHON_SO=.pyd
+                    IS_MINGW=true
+        ;;
+        *)IS_MINGW=false;;
+    esac
+    if test x"$PY3OSNAME" = x"nt" && test x"$IS_MINGW" = x"false"; then
       # Windows installations are quite different to posix installations
       # There is no python-config to use
       AC_MSG_CHECKING(for Python 3.x prefix)
