--- swig-4.0.0.orig/configure.ac	2019-05-01 18:28:41.445071500 -0400
+++ swig-4.0.0/configure.ac	2019-05-01 18:32:33.610678900 -0400
@@ -622,7 +622,13 @@
     PYEPREFIX=`($PYTHON -c "import sys; sys.stdout.write(sys.exec_prefix)") 2>/dev/null`
     AC_MSG_RESULT($PYEPREFIX)
 
-    if test x"$PYOSNAME" = x"nt" -a x"$PYSEPARATOR" = x"\\"; then
+    case $host in
+        *-*-mingw*) PYTHON_SO=.pyd
+                    IS_MINGW=true
+        ;;
+        *)IS_MINGW=false;;
+    esac
+    if test x"$PYOSNAME" = x"nt" -a x"$PYSEPARATOR" = x"\\" && test x"$IS_MINGW" = x"false"; then
       # Windows installations are quite different to posix installations (MinGW path separator is a forward slash)
       PYPREFIX=`echo "$PYPREFIX" | sed -e 's,\\\\,/,g'` # Forward slashes are easier to use and even work on Windows most of the time
       PYTHON_SO=.pyd
@@ -774,7 +780,13 @@
     PYSEPARATOR=`($PYTHON3 -c "import sys, os; sys.stdout.write(os.sep)") 2>/dev/null`
     AC_MSG_RESULT($PYSEPARATOR)
 
-    if test x"$PY3OSNAME" = x"nt" -a x"$PYSEPARATOR" = x"\\"; then
+    case $host in
+        *-*-mingw*) PYTHON_SO=.pyd
+                    IS_MINGW=true
+        ;;
+        *)IS_MINGW=false;;
+    esac
+    if test x"$PY3OSNAME" = x"nt" -a x"$PYSEPARATOR" = x"\\" && test x"$IS_MINGW" = x"false"; then
       # Windows installations are quite different to posix installations
       # There is no python-config to use
       AC_MSG_CHECKING(for Python 3.x prefix)
