--- a/lib/nfkc.c
+++ b/lib/nfkc.c
@@ -462,13 +462,13 @@ g_utf8_to_ucs4_fast (const gchar * str, gssize len, gsize * items_written)
  **/
 static gchar *
 g_ucs4_to_utf8 (const gunichar * str,
-		glong len,
-		glong * items_read, glong * items_written)
+		gssize len,
+		gsize * items_read, gsize * items_written)
 {
-  gint result_length;
+  gsize result_length;
   gchar *result = NULL;
   gchar *p;
-  gint i;
+  gsize i;
 
   result_length = 0;
   for (i = 0; len < 0 || i < len; i++)
@@ -1035,8 +1035,7 @@ char *
 stringprep_ucs4_to_utf8 (const uint32_t * str, ssize_t len,
 			 size_t * items_read, size_t * items_written)
 {
-  return g_ucs4_to_utf8 (str, len, (glong *) items_read,
-			 (glong *) items_written);
+  return g_ucs4_to_utf8 (str, len, items_read, items_written);
 }
 
 /**

