--- opencv_contrib-3.2.0/modules/sfm/src/libmv_light/libmv/multiview/robust_estimation.h.orig	2017-02-27 12:15:55.172447000 +0300
+++ opencv_contrib-3.2.0/modules/sfm/src/libmv_light/libmv/multiview/robust_estimation.h	2017-02-27 12:16:02.097290200 +0300
@@ -54,10 +54,10 @@
   double threshold_;
 };
 
-static uint IterationsRequired(int min_samples,
+static unsigned int IterationsRequired(int min_samples,
                         double outliers_probability,
                         double inlier_ratio) {
-  return static_cast<uint>(
+  return static_cast<unsigned int>(
       log(outliers_probability) / log(1.0 - pow(inlier_ratio, min_samples)));
 }
 
