diff --git a/test/msgpack_basic.cpp b/test/msgpack_basic.cpp
index 15215b2..4cc1e92 100644
--- a/test/msgpack_basic.cpp
+++ b/test/msgpack_basic.cpp
@@ -11,7 +11,7 @@
 
 #include <gtest/gtest.h>
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define msgpack_rand() ((double)rand() / RAND_MAX)
 #else  // _MSC_VER
 #define msgpack_rand() drand48()
diff --git a/test/msgpack_c.cpp b/test/msgpack_c.cpp
index 3ce5e2b..71d5b22 100644
--- a/test/msgpack_c.cpp
+++ b/test/msgpack_c.cpp
@@ -6,7 +6,7 @@
 
 #include <gtest/gtest.h>
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define msgpack_rand() ((double)rand() / RAND_MAX)
 #else  // _MSC_VER
 #define msgpack_rand() drand48()
