diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a9b241..d568e43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -270,7 +270,11 @@ ADD_LIBRARY (msgpackc-static STATIC
 
 SET_TARGET_PROPERTIES (msgpackc-static PROPERTIES OUTPUT_NAME "msgpackc")
 IF (MSGPACK_ENABLE_SHARED)
-    SET_TARGET_PROPERTIES (msgpackc PROPERTIES IMPORT_SUFFIX "_import.lib")
+    IF (MINGW)
+      SET_TARGET_PROPERTIES (msgpackc PROPERTIES IMPORT_SUFFIX ".dll.a")
+    ELSE ()
+      SET_TARGET_PROPERTIES (msgpackc PROPERTIES IMPORT_SUFFIX "_import.lib")
+    ENDIF ()
     SET_TARGET_PROPERTIES (msgpackc PROPERTIES SOVERSION 2 VERSION 2.0.0)
 ENDIF ()
 
