--- indent-2.2.11/configure.ac.orig	2015-03-06 22:17:50.197484100 +0000
+++ indent-2.2.11/configure.ac	2015-03-06 22:18:02.402705900 +0000
@@ -65,7 +65,6 @@
    po/Makefile.in
    Makefile
    src/Makefile
-   doc/Makefile
    man/Makefile
 ],
 [touch Makefile.in])
--- indent-2.2.11/Makefile.am.orig	2008-05-21 21:18:40.000000000 +0100
+++ indent-2.2.11/Makefile.am	2015-03-06 22:23:57.682642900 +0000
@@ -1,5 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 AUTOMAKE_OPTIONS = no-texinfo.tex
-SUBDIRS = src doc po man regression
+SUBDIRS = src po man regression
 BUILT_SOURCES = 
 
