--- a/src/utils/pathinfo.cpp
+++ b/src/utils/pathinfo.cpp
@@ -20,7 +20,7 @@
 {
     QString binaryPath =
       QFileInfo(qApp->applicationDirPath()).absoluteFilePath();
-    QString trPath = QDir::toNativeSeparators(binaryPath + "/translations");
+    QString trPath = QDir::toNativeSeparators(binaryPath + "/../share/flameshot/translations/");
 #if defined(Q_OS_LINUX) || defined(Q_OS_UNIX)
     return QStringList()
            << QStringLiteral(APP_PREFIX) + "/share/flameshot/translations"
