--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -91,7 +91,11 @@
         PRIVATE
         windows-cli.cpp)
     set_target_properties(flameshot-cli PROPERTIES OUTPUT_NAME "flameshot-cli")
-    target_link_options(flameshot-cli PRIVATE /SUBSYSTEM:CONSOLE)
+    if(MSVC)
+      target_link_options(flameshot-cli PRIVATE /SUBSYSTEM:CONSOLE)
+    else() # MINGW
+      target_link_options(flameshot-cli PRIVATE -municode)
+    endif()
     set_property(TARGET flameshot PROPERTY WIN32_EXECUTABLE true)
     # Getting error D8016 - /utf-8 automatically set by current cmake/msvc ?
     #if (MSVC)
