--- gst-plugins-bad-1.10.2/ext/gl/Makefile.am.orig	2017-01-16 08:25:10.131688300 +0300
+++ gst-plugins-bad-1.10.2/ext/gl/Makefile.am	2017-01-16 08:25:29.154329000 +0300
@@ -144,14 +144,14 @@
 	$(top_builddir)/gst-libs/gst/gl/libgstgl-$(GST_API_VERSION).la \
 	$(top_builddir)/gst-libs/gst/base/libgstbadbase-$(GST_API_VERSION).la \
 	$(top_builddir)/gst-libs/gst/video/libgstbadvideo-$(GST_API_VERSION).la \
+	$(GRAPHENE_LIBS) \
 	$(GST_BASE_LIBS) \
 	$(GST_PLUGINS_BASE_LIBS) -lgstvideo-$(GST_API_VERSION) \
 	-lgstpbutils-$(GST_API_VERSION) \
 	$(GL_LIBS) \
 	$(LIBPNG_LIBS) \
 	$(JPEG_LIBS) \
-	$(LIBM) \
-	$(GRAPHENE_LIBS)
+	$(LIBM)
 
 libgstopengl_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstopengl_la_LIBTOOLFLAGS = $(GST_PLUGIN_LIBTOOLFLAGS) --tag=CC
