--- a/ext/opencv/gstcameracalibrate.cpp
+++ b/ext/opencv/gstcameracalibrate.cpp
@@ -85,6 +85,7 @@
 
 #include <opencv2/imgproc.hpp>
 #include <opencv2/calib3d.hpp>
+#include <opencv2/objdetect.hpp>
 
 #include <gst/opencv/gstopencvutils.h>
 
--- a/ext/opencv/gstfaceblur.h
+++ b/ext/opencv/gstfaceblur.h
@@ -51,7 +51,7 @@
 #include <gst/opencv/gstopencvvideofilter.h>
 
 #include <opencv2/core.hpp>
-#include <opencv2/objdetect.hpp>
+#include <opencv2/xobjdetect.hpp>
 
 G_BEGIN_DECLS
 /* #defines don't like whitespacey bits */
--- a/ext/opencv/gstfacedetect.h
+++ b/ext/opencv/gstfacedetect.h
@@ -51,7 +51,7 @@
 
 #include <gst/gst.h>
 #include <gst/opencv/gstopencvvideofilter.h>
-#include <opencv2/objdetect.hpp>
+#include <opencv2/xobjdetect.hpp>
 
 G_BEGIN_DECLS
 /* #defines don't like whitespacey bits */
--- a/ext/opencv/gsthanddetect.h
+++ b/ext/opencv/gsthanddetect.h
@@ -49,6 +49,7 @@
 /* opencv */
 #include <opencv2/core.hpp>
 #include <opencv2/objdetect.hpp>
+#include <opencv2/xobjdetect.hpp>
 
 G_BEGIN_DECLS
 /* #defines don't like whitespacey bits */
--- a/ext/opencv/gstsegmentation.cpp
+++ b/ext/opencv/gstsegmentation.cpp
@@ -91,6 +91,7 @@
 
 #include "gstsegmentation.h"
 #include <opencv2/imgproc.hpp>
+#include <opencv2/geometry/2d.hpp>
 
 GST_DEBUG_CATEGORY_STATIC (gst_segmentation_debug);
 #define GST_CAT_DEFAULT gst_segmentation_debug
--- a/gst-libs/gst/opencv/meson.build
+++ b/gst-libs/gst/opencv/meson.build
@@ -32,9 +32,12 @@
 gstopencv_cargs = []
 opencv_dep = dependency('opencv', version : ['>= 3.0.0', '< 3.5.0'], required : false, include_type: 'system' )
 if not opencv_dep.found()
-  opencv_dep = dependency('opencv4', version : ['>= 4.0.0'], required : opencv_opt, include_type: 'system')
+  opencv_dep = dependency('opencv4', version : ['>= 4.0.0'], required : false, include_type: 'system')
   if not opencv_dep.found()
-    subdir_done()
+    opencv_dep = dependency('opencv5', version : ['>= 5.0.0'], required : opencv_opt, include_type: 'system')
+    if not opencv_dep.found()
+      subdir_done()
+    endif
   endif
 endif
 
@@ -65,7 +68,7 @@
 # Use prefix from pkg-config to be compatible with cross-compilation
 fsmod = import('fs')
 opencv_path_name = ''
-foreach d : ['opencv', 'OpenCV', 'opencv4']
+foreach d : ['opencv', 'OpenCV', 'opencv4', 'opencv5']
   if fsmod.is_dir(opencv_prefix / 'share' / d)
     opencv_path_name = d
     break
