--- a/sys/wasapi2/meson.build
+++ b/sys/wasapi2/meson.build
@@ -65,7 +65,7 @@
   #include <wrl/wrappers/corewrappers.h>
   #include <audioclient.h>
   #include <mmdeviceapi.h>
-  #if !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+  #if !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP) && !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
   #error "not winrt"
   #endif
   int main (int argc, char ** argv) {