--- gst-plugins-bad-1.4.0/ext/openjpeg/gstopenjpeg.h.orig	2013-10-17 07:00:31.000000000 +0000
+++ gst-plugins-bad-1.4.0/ext/openjpeg/gstopenjpeg.h	2014-07-30 21:17:32.722123400 +0000
@@ -38,7 +38,7 @@
 #define OPJ_CPRL CPRL
 #else
 #include <stdio.h>
-#include <openjpeg-2.0/openjpeg.h>
+#include <openjpeg-2.1/openjpeg.h>
 #endif
 
 #endif /* __GST_OPENJPEG_H__ */
--- gst-plugins-bad-1.4.0/ext/openjpeg/gstopenjpegdec.c.orig	2014-02-27 14:45:16.000000000 +0000
+++ gst-plugins-bad-1.4.0/ext/openjpeg/gstopenjpegdec.c	2014-07-30 21:23:32.206272100 +0000
@@ -1036,7 +1036,7 @@
   opj_stream_set_write_function (stream, write_fn);
   opj_stream_set_skip_function (stream, skip_fn);
   opj_stream_set_seek_function (stream, seek_fn);
-  opj_stream_set_user_data (stream, &mstream);
+  opj_stream_set_user_data (stream, &mstream, NULL);
   opj_stream_set_user_data_length (stream, mstream.size);
 
   image = NULL;
--- gst-plugins-bad-1.4.0/ext/openjpeg/gstopenjpegenc.c.orig	2013-11-23 17:51:16.000000000 +0000
+++ gst-plugins-bad-1.4.0/ext/openjpeg/gstopenjpegenc.c	2014-07-30 21:23:36.508318400 +0000
@@ -899,7 +899,7 @@
   opj_stream_set_write_function (stream, write_fn);
   opj_stream_set_skip_function (stream, skip_fn);
   opj_stream_set_seek_function (stream, seek_fn);
-  opj_stream_set_user_data (stream, &mstream);
+  opj_stream_set_user_data (stream, &mstream, NULL);
   opj_stream_set_user_data_length (stream, mstream.size);
 
   if (!opj_start_compress (enc, image, stream))
