--- gst-plugins-bad-1.26.10/sys/mediafoundation/gstmfcapturedshow.cpp.orig	2026-01-01 09:27:06.464909400 +0100
+++ gst-plugins-bad-1.26.10/sys/mediafoundation/gstmfcapturedshow.cpp	2026-01-01 09:27:16.908504900 +0100
@@ -312,7 +312,7 @@
       gst_caps_unref (caps);
   }
 
-  bool operator< (const GstMFDShowPinInfo & other)
+  bool operator< (const GstMFDShowPinInfo & other) const
   {
     return gst_mf_source_object_caps_compare (caps, other.caps) < 0;
   }
