diff -urN gst-plugins-bad-1.6.1.orig/tests/examples/gl/sdl/sdlshare.c gst-plugins-bad-1.6.1/tests/examples/gl/sdl/sdlshare.c
--- gst-plugins-bad-1.6.1.orig/tests/examples/gl/sdl/sdlshare.c	2015-11-02 18:28:09.641044400 +0000
+++ gst-plugins-bad-1.6.1/tests/examples/gl/sdl/sdlshare.c	2015-11-02 18:28:12.358449300 +0000
@@ -26,6 +26,7 @@
 #endif
 
 #include <GL/gl.h>
+#include <GL/glext.h>
 #include "SDL/SDL.h"
 #include "SDL/SDL_opengl.h"
 
diff -urN gst-plugins-bad-1.6.1.orig/tests/examples/gl/sdl/sdlshare2.c gst-plugins-bad-1.6.1/tests/examples/gl/sdl/sdlshare2.c
--- gst-plugins-bad-1.6.1.orig/tests/examples/gl/sdl/sdlshare2.c	2015-11-02 18:28:09.641044400 +0000
+++ gst-plugins-bad-1.6.1/tests/examples/gl/sdl/sdlshare2.c	2015-11-02 18:28:12.358449300 +0000
@@ -26,6 +26,7 @@
 #endif
 
 #include <GL/gl.h>
+#include <GL/glext.h>
 #include "SDL/SDL.h"
 #include "SDL/SDL_opengl.h"
 
