diff --git a/src/obj/Makefile.am b/src/obj/Makefile.am
index 7195238..ad48bf6 100644
--- a/src/obj/Makefile.am
+++ b/src/obj/Makefile.am
@@ -10,6 +10,6 @@ libJudy_la_SOURCES =
 
 libJudy_la_LIBADD = ../JudyCommon/*.lo ../JudyL/*.lo ../Judy1/*.lo ../JudyHS/*.lo ../JudySL/*.lo 
 
-libJudy_la_LDFLAGS = @VERSION_INFO@
+libJudy_la_LDFLAGS = @VERSION_INFO@ -no-undefined
 
 DISTCLEANFILES = .deps Makefile
