--- c-ares-1.25.0/CMakeLists.txt.orig	2024-01-03 09:44:48.000000000 +0100
+++ c-ares-1.25.0/CMakeLists.txt	2024-01-05 13:22:20.955977500 +0100
@@ -721,11 +721,10 @@
 	INSTALL (FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake" "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake" COMPONENT Devel DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 
 	# pkgconfig support
-	IF (NOT CARES_SHARED)
 		FOREACH (LIB ${CARES_DEPENDENT_LIBS})
 			SET (CARES_PRIVATE_LIBS "${CARES_PRIVATE_LIBS} -l${LIB}")
 		ENDFOREACH ()
-	ENDIF ()
+
 	CONFIGURE_FILE("libcares.pc.cmake" "libcares.pc" @ONLY)
 	INSTALL (FILES "${CMAKE_CURRENT_BINARY_DIR}/libcares.pc" COMPONENT Devel DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 ENDIF ()
--- c-ares-1.25.0/libcares.pc.cmake.orig	2024-01-03 09:41:43.000000000 +0100
+++ c-ares-1.25.0/libcares.pc.cmake	2024-01-05 13:14:57.807581600 +0100
@@ -7,9 +7,9 @@
 # Copyright (C) The c-ares project and its contributors
 # SPDX-License-Identifier: MIT
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=@CMAKE_INSTALL_FULL_BINDIR@
-libdir=@CMAKE_INSTALL_FULL_LIBDIR@
-includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: c-ares
 URL: https://c-ares.org/
@@ -18,5 +18,6 @@
 Requires: 
 Requires.private: 
 Cflags: -I${includedir}
+Cflags.private: -DCARES_STATICLIB
 Libs: -L${libdir} -lcares
 Libs.private: @CARES_PRIVATE_LIBS@
