--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -724,12 +724,10 @@
 	INSTALL (FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake" "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake" COMPONENT Devel DESTINATION ${CMAKECONFIG_INSTALL_DIR})
 
 	# pkgconfig support
-	IF (NOT CARES_SHARED)
 		SET (CPPFLAG_CARES_STATICLIB "-DCARES_STATICLIB")
 		FOREACH (LIB ${CARES_DEPENDENT_LIBS})
 			SET (CARES_PRIVATE_LIBS "${CARES_PRIVATE_LIBS} -l${LIB}")
 		ENDFOREACH ()
-	ENDIF ()
 	CONFIGURE_FILE("libcares.pc.cmake" "libcares.pc" @ONLY)
 	INSTALL (FILES "${CMAKE_CURRENT_BINARY_DIR}/libcares.pc" COMPONENT Devel DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 ENDIF ()
--- a/libcares.pc.cmake
+++ b/libcares.pc.cmake
@@ -7,9 +7,9 @@
 # Copyright (C) The c-ares project and its contributors
 # SPDX-License-Identifier: MIT
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=@CMAKE_INSTALL_FULL_BINDIR@
-libdir=@CMAKE_INSTALL_FULL_LIBDIR@
-includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: c-ares
 URL: https://c-ares.org/
@@ -17,6 +17,7 @@
 Version: @CARES_VERSION@
 Requires: 
 Requires.private: 
-Cflags: -I${includedir} @CPPFLAG_CARES_STATICLIB@
+Cflags: -I${includedir}
+Cflags.private: @CPPFLAG_CARES_STATICLIB@
 Libs: -L${libdir} -lcares
 Libs.private: @CARES_PRIVATE_LIBS@
