--- a/setup.py
+++ b/setup.py
@@ -90,6 +90,12 @@
             targetFileName = os.path.join(dir, baseName)
             self.copy_file(command.importLibraryName, targetFileName)
             self.outfiles.append(targetFileName)
+            # https://github.com/anthony-tuininga/cx_Logging/pull/4/files
+            target_dir = os.path.join(self.install_dir, "Include")
+            self.mkpath(target_dir)
+            target_file_name = os.path.join(target_dir, "cx_Logging.h")
+            self.copy_file("src\\cx_Logging.h", target_file_name)
+            self.outfiles.append(target_file_name)
 
 
 # setup macros
