--- a/setup.py
+++ b/setup.py
@@ -84,7 +84,7 @@
         distutils.command.install_data.install_data.run(self)
         if sys.platform == "win32":
             command = self.get_finalized_command("build_ext")
-            dir = os.path.join(self.install_dir, "Libs")
+            dir = os.path.join(self.install_dir, "lib")
             self.mkpath(dir)
             baseName = os.path.basename(command.importLibraryName)
             targetFileName = os.path.join(dir, baseName)
@@ -166,5 +169,5 @@
 setup(
         cmdclass = dict(build_ext = build_ext, install_data = install_data),
         version = BUILD_VERSION,
-        data_files = [ ("cx_Logging-doc", ["LICENSE.txt", "README.txt"]) ],
+        data_files = [ ("share/doc/cx_Logging", ["LICENSE.txt", "README.txt"]) ],
         ext_modules = [extension])
