--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -295,7 +295,7 @@
     set(INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
 endif()
 
-if(BUILD_SHARED_LIBS)
+if(NOT MINGW AND BUILD_SHARED_LIBS)
     set(ARCHIVE_DIRECTORY "${BINARY_DIRECTORY}")
 else()
     set(ARCHIVE_DIRECTORY "${LIBRARY_DIRECTORY}")
--- a/cmake/AWSSDKConfig.cmake
+++ b/cmake/AWSSDKConfig.cmake
@@ -43,7 +43,7 @@
 
 # On Windows, dlls are treated as runtime target and installed in bindir
 if (WIN32 AND AWSSDK_INSTALL_AS_SHARED_LIBS)
-    set(AWSSDK_INSTALL_LIBDIR "${AWSSDK_INSTALL_BINDIR}")
+    set(AWSSDK_INSTALL_LIBDIR "${AWSSDK_INSTALL_LIBDIR}")
     # If installed CMake scripts are associated with dll library, define USE_IMPORT_EXPORT for customers
     add_definitions(-DUSE_IMPORT_EXPORT)
 endif()
