--- a/third-party/cmake/BuildAwsCCommon.cmake
+++ b/third-party/cmake/BuildAwsCCommon.cmake
@@ -38,8 +38,7 @@
 else()
     ExternalProject_Add(AwsCCommon
         PREFIX ${AWS_DEPS_BUILD_DIR}
-        GIT_REPOSITORY ${AWS_C_COMMON_URL}
-        GIT_TAG ${AWS_C_COMMON_TAG}
+        SOURCE_DIR AwsCCommon
         BUILD_IN_SOURCE 0
         UPDATE_COMMAND ""
         CMAKE_ARGS
--- a/third-party/cmake/BuildAwsChecksums.cmake
+++ b/third-party/cmake/BuildAwsChecksums.cmake
@@ -37,8 +37,7 @@
 else()
     ExternalProject_Add(AwsChecksums
         PREFIX ${AWS_DEPS_BUILD_DIR}
-        GIT_REPOSITORY ${AWS_CHECKSUMS_URL}
-        GIT_TAG ${AWS_CHECKSUMS_TAG}
+        SOURCE_DIR AwsChecksums
         BUILD_IN_SOURCE 0
         UPDATE_COMMAND ""
         CMAKE_ARGS
--- a/third-party/cmake/BuildAwsEventStream.cmake
+++ b/third-party/cmake/BuildAwsEventStream.cmake
@@ -45,8 +45,7 @@
 else()
     ExternalProject_Add(AwsCEventStream
         PREFIX ${AWS_DEPS_BUILD_DIR}
-        GIT_REPOSITORY ${AWS_EVENT_STREAM_URL}
-        GIT_TAG ${AWS_EVENT_STREAM_TAG}
+        SOURCE_DIR AwsCEventStream
         BUILD_IN_SOURCE 0
         UPDATE_COMMAND ""
         CMAKE_ARGS
