--- a/aws-cpp-sdk-sts/include/aws/sts/STSEndpointRules.h
+++ b/aws-cpp-sdk-sts/include/aws/sts/STSEndpointRules.h
@@ -12,7 +12,7 @@
 {
 namespace STS
 {
-class STSEndpointRules
+class AWS_STS_API STSEndpointRules
 {
 public:
     static Aws::String GetRulesAsString();
--- a/aws-cpp-sdk-cognito-identity/include/aws/cognito-identity/CognitoIdentityEndpointRules.h
+++ b/aws-cpp-sdk-cognito-identity/include/aws/cognito-identity/CognitoIdentityEndpointRules.h
@@ -12,7 +12,7 @@
 {
 namespace CognitoIdentity
 {
-class CognitoIdentityEndpointRules
+class AWS_COGNITOIDENTITY_API CognitoIdentityEndpointRules
 {
 public:
     static Aws::String GetRulesAsString();
