diff -aurp AwsChecksums/source/intel/cpuid.c AwsChecksums-orig/source/intel/cpuid.c
--- AwsChecksums/source/intel/cpuid.c	2020-08-08 21:25:59.314437100 +0000
+++ AwsChecksums-orig/source/intel/cpuid.c	2020-08-08 21:25:44.210636300 +0000
@@ -40,4 +40,9 @@ int aws_checksums_do_cpu_id(int32_t *cpu
 
+#else
+int aws_checksums_do_cpu_id(int32_t *cpuid) {
+    (void)cpuid;
+    return 0;
+}
 #endif /* defined(__x86_64__) && !(defined(__GNUC__) && defined(DEBUG_BUILD)) */
