diff --git a/configure.ac b/configure.ac
index bd98a31..3f199c4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,7 +31,7 @@ AM_INIT_AUTOMAKE([foreign -Wall -Werror])
 
 # Enable additional compiler warnings via -Wall and -Wextra. Use hidden
 # visibility for all non-static symbols by default with -fvisibility=hidden.
-JAYLINK_CFLAGS="-Wall -Wextra -Werror -fvisibility=hidden"
+JAYLINK_CFLAGS="-Wall -Wextra -Werror"
 
 # Checks for programs.
 AC_PROG_CC
