--- aribb24-1.0.3/Makefile.am.orig	2014-08-18 20:29:48.000000000 +0500
+++ aribb24-1.0.3/Makefile.am	2014-09-26 00:58:26.140400000 +0500
@@ -7,6 +7,7 @@
 	src/drcs.h src/convtable.h			\
 	src/decoder_macro.h
 libaribb24_la_LIBADD = $(PNG_LIBS)
+libaribb24_la_LDFLAGS = -no-undefined
 libaribb24_la_CFLAGS = -Wall -fvisibility=hidden
 
 pkginclude_HEADERS = src/aribb24/decoder.h src/aribb24/parser.h	\
