diff --git a/addons/image/pcx.c b/addons/image/pcx.c
index 85ec058db..3328959ef 100644
--- a/addons/image/pcx.c
+++ b/addons/image/pcx.c
@@ -55,7 +55,7 @@ ALLEGRO_BITMAP *_al_load_pcx_f(ALLEGRO_FILE *f, int flags)
 
    bytes_per_line = al_fread16le(f);
    /* It can only be this because we only support 8 bit planes */
-   if (bytes_per_line != width) {
+   if (bytes_per_line < width) {
       ALLEGRO_ERROR("Invalid bytes per line %d\n", bytes_per_line);
       return NULL;
    }
