diff -Naur bison-3.0.2-orig/lib/error.c bison-3.0.2/lib/error.c
--- bison-3.0.2-orig/lib/error.c	2013-10-23 12:25:42.000000000 +0400
+++ bison-3.0.2/lib/error.c	2014-07-31 10:25:23.799200000 +0400
@@ -68,11 +68,11 @@
 /* In GNU libc we want do not want to use the common name 'error' directly.
    Instead make it a weak alias.  */
 extern void __error (int status, int errnum, const char *message, ...)
-     __attribute__ ((__format__ (__printf__, 3, 4)));
+     __attribute__ ((__format__ (gnu_printf, 3, 4)));
 extern void __error_at_line (int status, int errnum, const char *file_name,
                              unsigned int line_number, const char *message,
                              ...)
-     __attribute__ ((__format__ (__printf__, 5, 6)));
+     __attribute__ ((__format__ (gnu_printf, 5, 6)));
 # define error __error
 # define error_at_line __error_at_line
 
diff -Naur bison-3.0.2-orig/lib/error.h bison-3.0.2/lib/error.h
--- bison-3.0.2-orig/lib/error.h	2013-08-02 19:03:13.000000000 +0400
+++ bison-3.0.2/lib/error.h	2014-07-31 10:27:52.529600000 +0400
@@ -40,11 +40,11 @@
    If STATUS is nonzero, terminate the program with 'exit (STATUS)'.  */
 
 extern void error (int __status, int __errnum, const char *__format, ...)
-     _GL_ATTRIBUTE_FORMAT ((__printf__, 3, 4));
+     _GL_ATTRIBUTE_FORMAT ((gnu_printf, 3, 4));
 
 extern void error_at_line (int __status, int __errnum, const char *__fname,
                            unsigned int __lineno, const char *__format, ...)
-     _GL_ATTRIBUTE_FORMAT ((__printf__, 5, 6));
+     _GL_ATTRIBUTE_FORMAT ((gnu_printf, 5, 6));
 
 /* If NULL, error will flush stdout, then print on stderr the program
    name, a colon and a space.  Otherwise, error will call this
diff -Naur bison-3.0.2-orig/lib/vasnprintf.h bison-3.0.2/lib/vasnprintf.h
--- bison-3.0.2-orig/lib/vasnprintf.h	2013-08-02 19:16:02.000000000 +0400
+++ bison-3.0.2/lib/vasnprintf.h	2014-07-31 10:28:11.748800000 +0400
@@ -68,9 +68,9 @@
 # define vasnprintf rpl_vasnprintf
 #endif
 extern char * asnprintf (char *resultbuf, size_t *lengthp, const char *format, ...)
-       _GL_ATTRIBUTE_FORMAT ((__printf__, 3, 4));
+       _GL_ATTRIBUTE_FORMAT ((gnu_printf, 3, 4));
 extern char * vasnprintf (char *resultbuf, size_t *lengthp, const char *format, va_list args)
-       _GL_ATTRIBUTE_FORMAT ((__printf__, 3, 0));
+       _GL_ATTRIBUTE_FORMAT ((gnu_printf, 3, 0));
 
 #ifdef __cplusplus
 }
diff -Naur bison-3.0.2-orig/src/complain.h bison-3.0.2/src/complain.h
--- bison-3.0.2-orig/src/complain.h	2013-12-03 18:15:16.000000000 +0400
+++ bison-3.0.2/src/complain.h	2014-07-31 10:29:29.436800000 +0400
@@ -108,7 +108,7 @@
 
 /** Make a complaint, with maybe a location.  */
 void complain (location const *loc, warnings flags, char const *message, ...)
-  __attribute__ ((__format__ (__printf__, 3, 4)));
+  __attribute__ ((__format__ (gnu_printf, 3, 4)));
 
 /** Likewise, but with an \a argc/argv interface.  */
 void complain_args (location const *loc, warnings w, unsigned *indent,
@@ -117,7 +117,7 @@
 /** Make a complaint with location and some indentation.  */
 void complain_indent (location const *loc, warnings flags, unsigned *indent,
                       char const *message, ...)
-  __attribute__ ((__format__ (__printf__, 4, 5)));
+  __attribute__ ((__format__ (gnu_printf, 4, 5)));
 
 
 /** Report an obsolete syntax, suggest the updated one.  */
