--- bison-3.0.2/lib/spawn-pipe.c.orig	2014-08-04 17:37:41.455792200 +0100
+++ bison-3.0.2/lib/spawn-pipe.c	2014-08-04 17:38:00.301792200 +0100
@@ -136,10 +136,10 @@
   prog_argv = prepare_spawn (prog_argv);
 
   if (pipe_stdout)
-    if (pipe2_safer (ifd, O_BINARY | O_CLOEXEC) < 0)
+    if (pipe2_safer (ifd, O_TEXT | O_CLOEXEC) < 0)
       error (EXIT_FAILURE, errno, _("cannot create pipe"));
   if (pipe_stdin)
-    if (pipe2_safer (ofd, O_BINARY | O_CLOEXEC) < 0)
+    if (pipe2_safer (ofd, O_TEXT | O_CLOEXEC) < 0)
       error (EXIT_FAILURE, errno, _("cannot create pipe"));
 /* Data flow diagram:
  *
