--- zlib-1.3.2.orig/zconf.h     2026-02-17 20:47:06.000000000 +0800
+++ zlib-1.3.2/zconf.h  2026-03-05 15:10:21.915014500 +0800
@@ -516,7 +516,7 @@
 #endif

 #ifndef z_off_t
-#  define z_off_t long long
+#  define z_off_t long
 #endif

 #if !defined(_WIN32) && defined(Z_LARGE64)
--- zlib-1.3.2.orig/zconf.h.in  2026-02-17 20:47:06.000000000 +0800
+++ zlib-1.3.2/zconf.h.in       2026-03-05 15:10:47.663570700 +0800
@@ -516,10 +516,10 @@
 #endif

 #ifndef z_off_t
-#  define z_off_t long long
+#  define z_off_t long
 #endif

-#if !defined(_WIN32) && defined(Z_LARGE64)
+#if (defined(_WIN32) && defined(__GNUC__) && defined(_LARGEFILE64_SOURCE)) || defined(Z_LARGE64)
 #  define z_off64_t off64_t
 #elif defined(__MINGW32__)
 #  define z_off64_t long long
