--- a/zlib-1.2.8/contrib/minizip/iowin32.c.orig	2013-04-28 22:57:11.000000000 +0000
+++ b/zlib-1.2.8/contrib/minizip/iowin32.c	2013-06-15 20:07:42.136103400 +0000
@@ -26,11 +26,9 @@
 #endif
 
 
-#if defined(WINAPI_FAMILY_PARTITION) && (!(defined(IOWIN32_USING_WINRT_API)))
-#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+#if _WIN32_WINNT >= 0x0602
 #define IOWIN32_USING_WINRT_API 1
 #endif
-#endif
 
 voidpf  ZCALLBACK win32_open_file_func  OF((voidpf opaque, const char* filename, int mode));
 uLong   ZCALLBACK win32_read_file_func  OF((voidpf opaque, voidpf stream, void* buf, uLong size));
