diff -ENwbur /home/Martell/test/MINGW-packages/mingw-w64-zlib/minizip-orig/unzip.c /home/Martell/test/MINGW-packages/mingw-w64-zlib/minizip/unzip.c
--- minizip-orig/unzip.c	2013-02-24 07:32:30.000000000 +0000
+++ minizip/unzip.c	2014-09-17 19:54:02.675900800 +0100
@@ -1090,8 +1091,9 @@
 
         if (lSeek!=0)
         {
-            if (ZSEEK64(s->z_filefunc, s->filestream,lSeek,ZLIB_FILEFUNC_SEEK_CUR)==0)
-                lSeek=0;
+            if (ZSEEK64(s->z_filefunc, s->filestream,lSeek,ZLIB_FILEFUNC_SEEK_CUR)==0) {
+              //lSeek=0;
+			}
             else
                 err=UNZ_ERRNO;
         }
@@ -1099,10 +1101,10 @@
         if ((file_info.size_file_comment>0) && (commentBufferSize>0))
             if (ZREAD64(s->z_filefunc, s->filestream,szComment,uSizeRead)!=uSizeRead)
                 err=UNZ_ERRNO;
-        lSeek+=file_info.size_file_comment - uSizeRead;
+        //lSeek+=file_info.size_file_comment - uSizeRead;
     }
-    else
-        lSeek+=file_info.size_file_comment;
+    //else
+    //    lSeek+=file_info.size_file_comment;
 
 
     if ((err==UNZ_OK) && (pfile_info!=NULL))
@@ -1145,7 +1147,7 @@
                                                 szFileName,fileNameBufferSize,
                                                 extraField,extraFieldBufferSize,
                                                 szComment,commentBufferSize);
-    if ((err==UNZ_OK) && (pfile_info != NULL))
+    if (err==UNZ_OK)
     {
         pfile_info->version = file_info64.version;
         pfile_info->version_needed = file_info64.version_needed;
@@ -1533,10 +1567,10 @@
 /* #ifdef HAVE_BZIP2 */
         (s->cur_file_info.compression_method!=Z_BZIP2ED) &&
 /* #endif */
-        (s->cur_file_info.compression_method!=Z_DEFLATED))
-
-        err=UNZ_BADZIPFILE;
+        (s->cur_file_info.compression_method!=Z_DEFLATED)) {
 
+        //err=UNZ_BADZIPFILE;
+    }
     pfile_in_zip_read_info->crc32_wait=s->cur_file_info.crc;
     pfile_in_zip_read_info->crc32=0;
     pfile_in_zip_read_info->total_out_64=0;
