From 3724005ffa45a6115f94f8361d0f08cb95af568b Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 1 Aug 2021 15:18:50 +0200
Subject: [PATCH 110/N] configure.ac: set _WIN32_WINNT version

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 12373f7..2455a05 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3477,6 +3477,11 @@ else
 fi])
 AC_MSG_RESULT($with_dbmliborder)
 
+case $host in
+  *-*-mingw*)
+    CFLAGS_NODIST="$CFLAGS_NODIST -D_WIN32_WINNT=0x0601";;
+esac
+
 # Determine if windows modules should be used.
 AC_SUBST(USE_WIN32_MODULE)
 USE_WIN32_MODULE='#'
-- 
2.33.0

