--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -725,10 +725,10 @@
     set(ZSHAUTOCOMPLETE_INSTALL_DIR "${DATAROOTDIR}/zsh/site-functions")
 
     if(WIN32)
-        set(PLUGIN_INSTALL_DIR "plugins/gammaray")
+        set(PLUGIN_INSTALL_DIR "share/qt${QT_VERSION_MAJOR}/plugins/gammaray")
         set(LIBEXEC_INSTALL_DIR "${BIN_INSTALL_DIR}")
-        set(DOC_INSTALL_DIR .)
-        set(TRANSLATION_INSTALL_DIR "translations")
+        set(DOC_INSTALL_DIR "share/doc/gammaray/")
+        set(TRANSLATION_INSTALL_DIR "share/qt${QT_VERSION_MAJOR}/translations")
     else()
         set(PLUGIN_INSTALL_DIR "${LIB_INSTALL_DIR}/gammaray")
         set(LIBEXEC_INSTALL_DIR "${LIB_INSTALL_DIR}/gammaray/libexec")
@@ -749,9 +749,9 @@
     else()
         set(PROBE_INSTALL_DIR ${LIB_INSTALL_DIR})
     endif()
-    set(PROBE_PLUGIN_INSTALL_DIR "plugins/gammaray")
+    set(PROBE_PLUGIN_INSTALL_DIR "share/qt${QT_VERSION_MAJOR}/plugins/gammaray")
     set(PLUGIN_INSTALL_DIR ${PROBE_PLUGIN_INSTALL_DIR})
-    set(TARGET_PLUGIN_INSTALL_DIR "plugins/gammaray-target")
+    set(TARGET_PLUGIN_INSTALL_DIR "share/qt${QT_VERSION_MAJOR}/plugins/gammaray-target")
 else()
     set(PROBE_INSTALL_DIR
         "${PLUGIN_INSTALL_DIR}/${GAMMARAY_PLUGIN_VERSION}/${GAMMARAY_PROBE_ABI}${GAMMARAY_PROBE_ABI_POSTFIX}"
