--- a/tools/assimp_view/assimp_view.rc
+++ b/tools/assimp_view/assimp_view.rc
@@ -56,7 +56,6 @@
 STYLE DS_SETFONT | DS_MODALFRAME | DS_CENTER | WS_MINIMIZEBOX | WS_MAXIMIZEBOX | WS_POPUP | WS_CAPTION | WS_SYSMENU
 EXSTYLE WS_EX_ACCEPTFILES | WS_EX_WINDOWEDGE
 CAPTION "Open Asset Import Library - Model Viewer "
-MENU IDR_MENU1
 FONT 8, "Microsoft Sans Serif", 400, 0, 0x0
 BEGIN
     CONTROL         "",IDC_RT,"Static",SS_OWNERDRAW,0,0,513,324
