--- a/code/AssetLib/IFC/IFCLoader.cpp
+++ b/code/AssetLib/IFC/IFCLoader.cpp
@@ -50,11 +50,7 @@
 #include <tuple>
 
 #ifndef ASSIMP_BUILD_NO_COMPRESSED_IFC
-#ifdef ASSIMP_USE_HUNTER
 #include <minizip/unzip.h>
-#else
-#include <unzip.h>
-#endif
 #endif
 
 #include "../STEPParser/STEPFileReader.h"
--- a/code/Common/ZipArchiveIOSystem.cpp
+++ b/code/Common/ZipArchiveIOSystem.cpp
@@ -51,11 +51,7 @@
 #include <map>
 #include <memory>
 
-#ifdef ASSIMP_USE_HUNTER
-#    include <minizip/unzip.h>
-#else
-#    include <unzip.h>
-#endif
+#include <minizip/unzip.h>
 
 namespace Assimp {
 
