diff -Naur assimp-4.0.1-orig/tools/assimp_view/assimp_view.h assimp-4.0.1/tools/assimp_view/assimp_view.h
--- assimp-4.0.1-orig/tools/assimp_cmd/assimp_cmd.rc.orig	2014-06-12 21:50:48.160000000 +0400
+++ assimp-4.0.1/tools/assimp_cmd/assimp_cmd.rc	2014-06-12 21:54:56.986400000 +0400
@@ -1,7 +1,7 @@
 // Microsoft Visual C++ generated resource script.
 //
 #include "resource.h"
-#include "../../revision.h"
+#include "revision.h"
 
 #define APSTUDIO_READONLY_SYMBOLS
 /////////////////////////////////////////////////////////////////////////////
diff -Naur assimp-4.0.1-orig/tools/assimp_view/assimp_view.h assimp-4.0.1/tools/assimp_view/assimp_view.h
--- assimp-4.0.1-orig/tools/assimp_view/assimp_view.h	2016-10-21 10:48:42.584965100 +0300
+++ assimp-4.0.1/tools/assimp_view/assimp_view.h	2016-10-21 10:49:26.377751400 +0300
@@ -183,7 +185,7 @@
 {
     // for unsigned types only ...
     intype mask = (0x1u << (sizeof(type)*8))-1;
-    return (type)max((intype)0,min(in,mask));
+    return (type)std::max((intype)0,std::min(in,mask));
 }
 
 
diff -Naur assimp-4.0.1-orig/tools/assimp_view/assimp_view.cpp assimp-4.0.1/tools/assimp_view/assimp_view.cpp
--- assimp-4.0.1-orig/tools/assimp_view/assimp_view.cpp	2016-10-21 10:48:42.584965100 +0300
+++ assimp-4.0.1/tools/assimp_view/assimp_view.cpp	2016-10-21 10:49:26.377751400 +0300
@@ -42,7 +42,11 @@
 
 
 #include "assimp_view.h"
+#ifdef __MINGW32__
+#include <mmsystem.h>
+#else
 #include <timeapi.h>
+#endif
 #include "StringUtils.h"
 #include <map>
 
diff -Naur assimp-4.0.1-orig/tools/assimp_view/Display.cpp assimp-4.0.1/tools/assimp_view/Display.cpp
--- assimp-4.0.1-orig/tools/assimp_view/Display.cpp	2016-10-21 10:48:46.191201000 +0300
+++ assimp-4.0.1/tools/assimp_view/Display.cpp	2016-10-21 10:49:17.437250000 +0300
@@ -2249,7 +2249,7 @@
         const float ny = (float)sRect.bottom;
         const float  x = (float)sDesc.Width;
         const float  y = (float)sDesc.Height;
-        float f = min((nx-30) / x,(ny-30) / y) * (m_fTextureZoom/1000.0f);
+        float f = std::min((nx-30) / x,(ny-30) / y) * (m_fTextureZoom/1000.0f);
 
         float fHalfX = (nx - (f * x)) / 2.0f;
         float fHalfY = (ny - (f * y)) / 2.0f;
diff -Naur assimp-4.0.1-orig/tools/assimp_view/Material.cpp assimp-4.0.1/tools/assimp_view/Material.cpp
--- assimp-4.0.1-orig/tools/assimp_view/Material.cpp	2016-07-08 20:29:51.000000000 +0300
+++ assimp-4.0.1/tools/assimp_view/Material.cpp	2016-10-21 10:49:47.462604500 +0300
@@ -275,7 +275,7 @@
                         for (unsigned int i = 0; i < iSizeFound;++i)
                             info.cFileName[i] = (CHAR)tolower(info.cFileName[i]);
 
-                        if (0 == memcmp(info.cFileName,szFile2, min(iSizeFound,iSize)))
+                        if (0 == memcmp(info.cFileName,szFile2, std::min(iSizeFound,iSize)))
                         {
                             // we have it. Build the full path ...
                             char* sz = strrchr(szTempB,'*');
diff -Naur assimp-4.0.1-orig/tools/assimp_view/MessageProc.cpp assimp-4.0.1/tools/assimp_view/MessageProc.cpp
--- assimp-4.0.1-orig/tools/assimp_view/MessageProc.cpp	2016-07-08 20:29:51.000000000 +0300
+++ assimp-4.0.1/tools/assimp_view/MessageProc.cpp	2016-10-21 10:50:42.072467800 +0300
@@ -45,7 +45,11 @@
 
 #include <windowsx.h>
 #include <commdlg.h>
+#ifdef __MINGW32__
+#include <mmsystem.h>
+#else
 #include <timeapi.h>
+#endif
 
 namespace AssimpView {
 
@@ -1056,9 +1056,9 @@
         ai_assert(strlen(szFileName) <= MAX_PATH);
 
         // invent a nice default file name
-        char* sz = max(strrchr(szFileName,'\\'),strrchr(szFileName,'/'));
+        char* sz = std::max(strrchr(szFileName,'\\'),strrchr(szFileName,'/'));
         if (sz) {
-            strncpy(sz,max(strrchr(g_szFileName,'\\'),strrchr(g_szFileName,'/')),MAX_PATH);
+            strncpy(sz,std::max(strrchr(g_szFileName,'\\'),strrchr(g_szFileName,'/')),MAX_PATH);
         }
     }
     else {
