--- mpv-0.35.0/video/out/vulkan/utils.c.orig	2022-11-12 14:52:40.000000000 +0100
+++ mpv-0.35.0/video/out/vulkan/utils.c	2023-01-08 15:15:16.468521700 +0100
@@ -34,7 +34,7 @@
     if (vk->surface) {
         assert(vk->vkinst);
         vkDestroySurfaceKHR(vk->vkinst->instance, vk->surface, NULL);
-        vk->surface = NULL;
+        vk->surface = VK_NULL_HANDLE;
     }
 
     pl_vk_inst_destroy(&vk->vkinst);
