--- glib-2.62.0/docs/reference/gio/meson.build.orig	2019-09-06 19:31:19.266531200 +0200
+++ glib-2.62.0/docs/reference/gio/meson.build	2019-09-06 19:31:31.986119500 +0200
@@ -144,8 +144,8 @@
   concat_files_helper = find_program('concat-files-helper.py')
   configure_file(
     output : 'gio-sections.txt',
-    input : sections_files,
-    command : [concat_files_helper, '@OUTPUT@', '@INPUT@'],
+    input : sections_files[0],
+    command : [concat_files_helper, '@OUTPUT@'] + sections_files,
   )
 
   configure_file(
