From e0adf46abad3c14f47d272f4137c2b597a726b8f Mon Sep 17 00:00:00 2001
From: Doronin Stanislav <mugisbrows@gmail.com>
Date: Tue, 2 Nov 2021 23:02:55 +0300
Subject: [PATCH] ext_to_mime_hash_table

---
 gio/gcontenttype-win32.c | 57 +++++++++++++++++++++++++++++++++++++++-
 1 file changed, 56 insertions(+), 1 deletion(-)

diff --git a/gio/gcontenttype-win32.c b/gio/gcontenttype-win32.c
index 2f07fa3..e284b9c 100644
--- a/gio/gcontenttype-win32.c
+++ b/gio/gcontenttype-win32.c
@@ -345,6 +345,9 @@ g_content_type_from_mime_type (const gchar *mime_type)
   return content_type;
 }
 
+G_LOCK_DEFINE_STATIC (_ext_to_mime_type);
+static GHashTable *_ext_to_mime_type = NULL;
+
 gchar *
 g_content_type_guess (const gchar  *filename,
                       const guchar *data,
@@ -364,12 +367,64 @@ g_content_type_guess (const gchar  *filename,
    * not documented and not allowed; guard against that */
   g_return_val_if_fail (data_size != (gsize) -1, g_strdup ("*"));
 
+  G_LOCK(_ext_to_mime_type);
+  if (_ext_to_mime_type == NULL) {
+    static const char* exts[111] = {".3ds",".ag",".arw",".astc",".avif",".avifs",
+    ".bmp",".cgm",".cr2",".crw",".cur",".dcr",".dds",".dib",".djv",".djvu",".dng",
+    ".dwg",".dxf",".emf",".eps",".epsf",".epsi",".exr",".fig",".fits",".g3",".gbr",
+    ".gif",".gih",".heic",".heif",".icb",".icns",".ico",".ief",".iff",".ilbm",".j2c",
+    ".j2k",".jng",".jp2",".jpc",".jpe",".jpeg",".jpf",".jpg",".jpg2",".jpgm",".jpm",
+    ".jpx",".k25",".kdc",".ktx",".ktx2",".lbm",".lwo",".lwob",".lws",".mdi",".mrw",
+    ".msod",".nef",".ora",".orf",".pat",".pbm",".pcd",".pct",".pcx",".pef",".pgm",
+    ".pict",".pict1",".pict2",".png",".pnm",".pntg",".ppm",".psd",".qif",".qtif",
+    ".raf",".ras",".raw",".rgb",".rle",".rp",".rw2",".sgi",".sk",".sk1",".sr2",
+    ".srf",".sun",".svg",".svgz",".tga",".tif",".tiff",".tpic",".vda",".vst",".wbmp",
+    ".webp",".wmf",".x3f",".xbm",".xcf",".xpm",".xwd"};
+    static const char* mimes[111] = {"image/x-3ds","image/x-applix-graphics",
+    "image/x-sony-arw","image/astc","image/avif","image/avif","image/bmp",
+    "image/cgm","image/x-canon-cr2","image/x-canon-crw","image/x-win-bitmap",
+    "image/x-kodak-dcr","image/x-dds","image/bmp","image/vnd.djvu+multipage",
+    "image/vnd.djvu+multipage","image/x-adobe-dng","image/vnd.dwg","image/vnd.dxf",
+    "image/emf","image/x-eps","image/x-eps","image/x-eps","image/x-exr",
+    "image/x-xfig","image/fits","image/g3fax","image/x-gimp-gbr","image/gif",
+    "image/x-gimp-gih","image/heif","image/heif","image/x-tga","image/x-icns",
+    "image/vnd.microsoft.icon","image/ief","image/x-ilbm","image/x-ilbm",
+    "image/x-jp2-codestream","image/x-jp2-codestream","image/x-jng","image/jp2",
+    "image/x-jp2-codestream","image/jpeg","image/jpeg","image/jpx","image/jpeg",
+    "image/jp2","image/jpm","image/jpm","image/jpx","image/x-kodak-k25",
+    "image/x-kodak-kdc","image/ktx","image/ktx2","image/x-ilbm","image/x-lwo",
+    "image/x-lwo","image/x-lws","image/vnd.ms-modi","image/x-minolta-mrw",
+    "image/x-msod","image/x-nikon-nef","image/openraster","image/x-olympus-orf",
+    "image/x-gimp-pat","image/x-portable-bitmap","image/x-photo-cd","image/x-pict",
+    "image/vnd.zbrush.pcx","image/x-pentax-pef","image/x-portable-graymap",
+    "image/x-pict","image/x-pict","image/x-pict","image/png",
+    "image/x-portable-anymap","image/x-macpaint","image/x-portable-pixmap",
+    "image/vnd.adobe.photoshop","image/x-quicktime","image/x-quicktime",
+    "image/x-fuji-raf","image/x-cmu-raster","image/x-panasonic-rw","image/x-rgb",
+    "image/rle","image/vnd.rn-realpix","image/x-panasonic-rw2","image/x-sgi",
+    "image/x-skencil","image/x-skencil","image/x-sony-sr2","image/x-sony-srf",
+    "image/x-sun-raster","image/svg+xml","image/svg+xml-compressed","image/x-tga",
+    "image/tiff","image/tiff","image/x-tga","image/x-tga","image/x-tga",
+    "image/vnd.wap.wbmp","image/webp","image/wmf","image/x-sigma-x3f",
+    "image/x-xbitmap","image/x-xcf","image/x-xpixmap","image/x-xwindowdump"};
+    _ext_to_mime_type = g_hash_table_new (g_str_hash, g_str_equal);
+    for(int i=0;i<111;i++)
+      g_hash_table_insert(_ext_to_mime_type, g_strdup(exts[i]), g_strdup(mimes[i]));
+  }
+  G_UNLOCK(_ext_to_mime_type);
+
   if (filename)
     {
       basename = g_path_get_basename (filename);
       dot = strrchr (basename, '.');
-      if (dot)
+      if (dot) {
+        char* mime;
         type = g_strdup (dot);
+        mime = g_hash_table_lookup(_ext_to_mime_type, g_utf8_strdown(type, -1));
+        if (mime) {
+          type = g_strdup(mime);
+        }
+      }
       g_free (basename);
     }
 
-- 
2.29.2.windows.2

