--- glib-2.78.0/meson.build.orig	2023-09-12 16:28:58.458355600 +0200
+++ glib-2.78.0/meson.build	2023-09-12 16:30:37.026111500 +0200
@@ -2390,8 +2390,6 @@
 Please use gcc >= 4.7 or clang >= 12: GLib will drop -mms-bitfields in the future.
 ''')
   endif
-  win32_cflags = ['-mms-bitfields']
-  add_project_arguments(win32_cflags, language : 'c')
 
   # Win32 API libs, used only by libglib and exposed in glib-2.0.pc
   win32_ldflags = ['-lws2_32', '-lole32', '-lwinmm', '-lshlwapi', '-luuid']
