--- a/meson.build
+++ b/meson.build
@@ -2316,12 +2316,6 @@
 win32_cflags = []
 win32_ldflags = []
 if host_system == 'windows' and cc.get_id() != 'msvc' and cc.get_id() != 'clang-cl'
-  # Ensure MSVC-compatible struct packing convention is used when
-  # compiling for Win32 with gcc. It is used for the whole project and exposed
-  # in glib-2.0.pc.
-  win32_cflags = ['-mms-bitfields']
-  add_project_arguments(win32_cflags, language : 'c')
-
   # Win32 API libs, used only by libglib and exposed in glib-2.0.pc
   win32_ldflags = ['-lws2_32', '-lole32', '-lwinmm', '-lshlwapi', '-luuid']
 elif host_system == 'cygwin'
