--- a/gio/glib-compile-schemas.c
+++ b/gio/glib-compile-schemas.c
@@ -21,6 +21,7 @@
 
 /* Prologue {{{1 */
 #include "config.h"
+#include <glib.h>
 
 #include <gstdio.h>
 #include <gi18n.h>
@@ -2169,6 +2170,7 @@
   gboolean strict = FALSE;
   gchar **schema_files = NULL;
   gchar **override_files = NULL;
+  gchar **command_line = NULL;
   GOptionContext *context = NULL;
   gint retval;
   GOptionEntry entries[] = {
@@ -2210,12 +2212,22 @@
        "and the cache file is called gschemas.compiled."));
   g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
 
+#ifdef G_OS_WIN32
+  command_line = g_win32_get_command_line ();
+  if (!g_option_context_parse_strv(context, &command_line, &error))
+    {
+      fprintf (stderr, "%s\n", error->message);
+      retval = 1;
+      goto done;
+    }
+#else
   if (!g_option_context_parse (context, &argc, &argv, &error))
     {
       fprintf (stderr, "%s\n", error->message);
       retval = 1;
       goto done;
     }
+#endif
 
   if (show_version_and_exit)
     {
@@ -2231,7 +2243,11 @@
       goto done;
     }
 
+#ifdef G_OS_WIN32
+  srcdir = command_line[1];
+#else
   srcdir = argv[1];
+#endif
 
   target = g_build_filename (targetdir ? targetdir : srcdir, "gschemas.compiled", NULL);
 
@@ -2320,6 +2336,7 @@
   g_free (target);
   g_strfreev (schema_files);
   g_strfreev (override_files);
+  g_strfreev (command_line);
   g_option_context_free (context);
 
 #ifdef G_OS_WIN32
