--- glib-2.39.90/glib/tests/fileutils.c.orig	2014-01-06 19:02:48.000000000 +0000
+++ glib-2.39.90/glib/tests/fileutils.c	2014-03-02 01:57:29.174087800 +0000
@@ -36,9 +36,9 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#endif
 #include <fcntl.h>
 #include <utime.h>
-#endif
 #ifdef G_OS_WIN32
 #include <windows.h>
 #endif
--- glib-2.39.90/gio/tests/filter-cat.c.orig	2014-02-03 17:40:41.000000000 +0000
+++ glib-2.39.90/gio/tests/filter-cat.c	2014-03-02 02:34:34.098117300 +0000
@@ -32,6 +32,7 @@
 #endif
 
 #ifdef G_OS_WIN32
+#include <io.h>
 #ifndef STDOUT_FILENO
 #define STDOUT_FILENO 1
 #endif
--- glib-2.39.90/gio/tests/gsubprocess-testprog.c.orig	2014-01-06 19:02:48.000000000 +0000
+++ glib-2.39.90/gio/tests/gsubprocess-testprog.c	2014-03-02 02:35:07.151314500 +0000
@@ -7,6 +7,8 @@
 #include <unistd.h>
 #include <gio/gunixinputstream.h>
 #include <gio/gunixoutputstream.h>
+#else
+#include <io.h>
 #endif
 
 static GOptionEntry options[] = {
--- glib-2.39.90/gio/tests/win32-streams.c.orig	2014-01-06 19:02:48.000000000 +0000
+++ glib-2.39.90/gio/tests/win32-streams.c	2014-03-02 02:44:01.223133000 +0000
@@ -339,7 +339,7 @@
   gchar name[256];
 
   g_snprintf (name, sizeof (name),
-              "\\\\.\\pipe\\gtest-io-overlap-%u", (guint) getpid ());
+              "\\\\.\\pipe\\gtest-io-overlap-%u", (guint) GetCurrentProcessId ());
 
   server = CreateNamedPipe (name,
                             PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
@@ -418,7 +418,7 @@
   gchar name[256], c;
 
   g_snprintf (name, sizeof (name),
-              "\\\\.\\pipe\\gtest-io-concurrent-%u", (guint) getpid ());
+              "\\\\.\\pipe\\gtest-io-concurrent-%u", (guint) GetCurrentProcessId ());
 
   server = CreateNamedPipe (name,
                             PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
@@ -490,7 +490,7 @@
   gchar name[256];
 
   g_snprintf (name, sizeof (name),
-              "\\\\.\\pipe\\gtest-io-cancel-%u", (guint) getpid ());
+              "\\\\.\\pipe\\gtest-io-cancel-%u", (guint) GetCurrentProcessId ());
 
   in_handle = CreateNamedPipe (name,
                                PIPE_ACCESS_INBOUND | FILE_FLAG_OVERLAPPED,
