--- glib-2.39.90/glib/gfileutils.c.orig	2014-02-15 15:31:32.000000000 +0000
+++ glib-2.39.90/glib/gfileutils.c	2014-03-02 16:30:35.862352400 +0000
@@ -607,16 +607,17 @@
 
 static char *
 format_error_message (const gchar  *filename,
-                      const gchar  *format_string) G_GNUC_FORMAT(2);
+                      const gchar  *format_string,
+                      int           saved_errno) G_GNUC_FORMAT(2);
 
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wformat-nonliteral"
 
 static char *
 format_error_message (const gchar  *filename,
-                      const gchar  *format_string)
+                      const gchar  *format_string,
+                      int           saved_errno)
 {
-  gint saved_errno = errno;
   gchar *display_name;
   gchar *msg;
 
@@ -637,10 +638,10 @@
 static void
 set_file_error (GError      **error,
                 const gchar  *filename,
-                const gchar  *format_string)
+                const gchar  *format_string,
+                int           saved_errno)
 {
-  int saved_errno = errno;
-  char *msg = format_error_message (filename, format_string);
+  char *msg = format_error_message (filename, format_string, saved_errno);
 
   g_set_error_literal (error, G_FILE_ERROR, g_file_error_from_errno (saved_errno),
                        msg);
@@ -860,7 +861,8 @@
     {
       set_file_error (error,
                       filename,
-                      _("Failed to open file '%s': %s"));
+                      _("Failed to open file '%s': %s"),
+                      errno);
 
       return FALSE;
     }
@@ -870,7 +872,8 @@
     {
       set_file_error (error,
                       filename,
-                      _("Failed to get attributes of file '%s': fstat() failed: %s"));
+                      _("Failed to get attributes of file '%s': fstat() failed: %s"),
+                      errno);
       close (fd);
 
       return FALSE;
@@ -898,7 +901,8 @@
         {
           set_file_error (error,
                           filename,
-                          _("Failed to open file '%s': fdopen() failed: %s"));
+                          _("Failed to open file '%s': fdopen() failed: %s"),
+                          errno);
 
           return FALSE;
         }
@@ -926,7 +930,8 @@
     {
       set_file_error (error,
                       filename,
-                      _("Failed to open file '%s': %s"));
+                      _("Failed to open file '%s': %s"),
+                      errno);
 
       return FALSE;
     }
@@ -1027,7 +1032,9 @@
 
   if (fd == -1)
     {
-      set_file_error (err, tmp_name, _("Failed to create file '%s': %s"));
+      set_file_error (err,
+                      tmp_name, _("Failed to create file '%s': %s"),
+                      errno);
       goto out;
     }
 
@@ -1051,7 +1058,9 @@
           if (errno == EINTR)
             continue;
 
-          set_file_error (err, tmp_name, _("Failed to write file '%s': write() failed: %s"));
+          set_file_error (err,
+                          tmp_name, _("Failed to write file '%s': write() failed: %s"),
+                          errno);
           close (fd);
           g_unlink (tmp_name);
 
@@ -1091,7 +1100,9 @@
      */
     if (g_lstat (dest_file, &statbuf) == 0 && statbuf.st_size > 0 && fsync (fd) != 0)
       {
-        set_file_error (err, tmp_name, _("Failed to write file '%s': fsync() failed: %s"));
+        set_file_error (err,
+                        tmp_name, _("Failed to write file '%s': fsync() failed: %s"),
+                        errno);
         close (fd);
         g_unlink (tmp_name);
 
@@ -1213,7 +1224,8 @@
 	{
           set_file_error (error,
                           filename,
-		          _("Existing file '%s' could not be removed: g_unlink() failed: %s"));
+		          _("Existing file '%s' could not be removed: g_unlink() failed: %s"),
+                          errno);
 	  g_unlink (tmp_filename);
 	  retval = FALSE;
 	  goto out;
@@ -1516,7 +1528,8 @@
     {
       set_file_error (error,
                       fulltemplate,
-                      _("Failed to create file '%s': %s"));
+                      _("Failed to create file '%s': %s"),
+                      errno);
       g_free (fulltemplate);
       return -1;
     }
@@ -2012,7 +2025,8 @@
         {
           set_file_error (error,
                           filename,
-                          _("Failed to read the symbolic link '%s': %s"));
+                          _("Failed to read the symbolic link '%s': %s"),
+                          errno);
 	
           return NULL;
         }
