--- glib-2.39.90/glib/tests/test-printf.c.orig	2014-03-02 17:25:28.535468800 +0000
+++ glib-2.39.90/glib/tests/test-printf.c	2014-03-02 17:24:39.981803200 +0000
@@ -24,6 +24,9 @@
 #include <string.h>
 #include "glib.h"
 #include "gstdio.h"
+#ifdef G_OS_WIN32
+#include <fcntl.h>
+#endif
 
 static void
 test_retval_and_trunc (void)
@@ -896,6 +899,9 @@
 main (int   argc,
       char *argv[])
 {
+#ifdef G_OS_WIN32
+  _setmode (fileno (stdout), _O_BINARY);
+#endif
   g_test_init (&argc, &argv, NULL);
 
   g_test_add_func ("/snprintf/retval-and-trunc", test_retval_and_trunc);
