--- glib-2.40.0/glib/gnulib/printf.c.orig	2014-02-03 17:40:41.000000000 +0000
+++ glib-2.40.0/glib/gnulib/printf.c	2014-05-20 00:11:04.036075100 +0000
@@ -88,16 +88,16 @@
 int _g_gnulib_vfprintf (FILE *file, char const *format, va_list args)
 {
   char *result;
-  size_t length;
+  size_t length, rlength;
 
   result = vasnprintf (NULL, &length, format, args);
   if (result == NULL) 
     return -1;
 
-  fwrite (result, 1, length, file);
+  rlength = fwrite (result, 1, length, file);
   free (result);
   
-  return length;
+  return rlength;
 }
 
 int _g_gnulib_vsprintf (char *string, char const *format, va_list args)
