--- glib-2.41.2/glib/gspawn-win32.c.orig	2014-06-28 17:02:43.000000000 +0000
+++ glib-2.41.2/glib/gspawn-win32.c	2014-07-29 18:27:41.843139500 +0000
@@ -590,7 +590,7 @@
     goto cleanup_and_fail;
   
   new_argv = g_new (char *, argc + 1 + ARG_COUNT);
-  if (GetConsoleCursorInfo (GetStdHandle (STD_OUTPUT_HANDLE), &cursor_info))
+  if (GetConsoleWindow () != NULL)
     helper_process = HELPER_PROCESS "-console.exe";
   else
     helper_process = HELPER_PROCESS ".exe";
