--- libtool-2.4.6/build-aux/ltmain.in.orig	2021-08-29 04:58:24.267126800 -0400
+++ libtool-2.4.6/build-aux/ltmain.in	2021-08-29 05:04:21.377405000 -0400
@@ -4270,6 +4270,11 @@ strendzap (char *str, const char *pat)
   return str;
 }
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-nonliteral"
+#endif
+
 void
 lt_debugprintf (const char *file, int line, const char *fmt, ...)
 {
@@ -4296,6 +4301,10 @@ lt_error_core (int exit_status, const ch
     exit (exit_status);
 }
 
+#ifdef __clang__
+#pragma clang diagnostic pop    
+#endif   
+
 void
 lt_fatal (const char *file, int line, const char *message, ...)
 {
