diff --git a/src/hdy-swipe-group.c b/src/hdy-swipe-group.c
index 18b167ab5a260093bc0e6cd8fc4deef86211cb0f..2779a312e066f698635f151a8df6c57b6ac8b125 100644
--- a/src/hdy-swipe-group.c
+++ b/src/hdy-swipe-group.c
@@ -104,7 +104,7 @@ hdy_swipe_group_new (void)
 
 static void
 child_switched_cb (HdySwipeGroup *self,
-                   uint           index,
+                   guint          index,
                    gint64         duration,
                    HdySwipeable  *swipeable)
 {
diff --git a/tests/test-carousel.c b/tests/test-carousel.c
index 538ca6e1a8f9589ebf10d0099bfdb56349f0a357..25ccb3e68700612dc8ed5f0c3e4debb6e8cbdc8e 100644
--- a/tests/test-carousel.c
+++ b/tests/test-carousel.c
@@ -158,7 +158,7 @@ static void
 test_hdy_carousel_indicator_spacing (void)
 {
   HdyCarousel *carousel = HDY_CAROUSEL (hdy_carousel_new ());
-  uint spacing;
+  guint spacing;
 
   notified = 0;
   g_signal_connect (carousel, "notify::indicator-spacing", G_CALLBACK (notify_cb), NULL);
@@ -210,7 +210,7 @@ static void
 test_hdy_carousel_spacing (void)
 {
   HdyCarousel *carousel = HDY_CAROUSEL (hdy_carousel_new ());
-  uint spacing;
+  guint spacing;
 
   notified = 0;
   g_signal_connect (carousel, "notify::spacing", G_CALLBACK (notify_cb), NULL);
@@ -236,7 +236,7 @@ static void
 test_hdy_carousel_animation_duration (void)
 {
   HdyCarousel *carousel = HDY_CAROUSEL (hdy_carousel_new ());
-  uint duration;
+  guint duration;
 
   notified = 0;
   g_signal_connect (carousel, "notify::animation-duration", G_CALLBACK (notify_cb), NULL);
