--- a/lib/gks/socket.c
+++ b/lib/gks/socket.c
@@ -74,7 +74,7 @@ static int wsa_initialized = 0;
 
 static DWORD WINAPI gksqt_thread(LPVOID parm)
 {
-  wchar_t *cmd = (char *)parm;
+  wchar_t *cmd = (wchar_t *)parm;
   wchar_t w_cmd_line[CMD_LINE_LEN];
   wchar_t buffer[BUFFER_LEN];
   wchar_t *currentDirectory = NULL;
--- a/lib/gr/stream.c
+++ b/lib/gr/stream.c
@@ -313,7 +313,7 @@ static DWORD WINAPI grplot_thread(LPVOID parm)
 
 static DWORD WINAPI grplot_thread(LPVOID parm)
 {
-  wchar_t *cmd = (char *)parm;
+  wchar_t *cmd = (wchar_t *)parm;
   wchar_t w_cmd_line[CMD_LINE_LEN];
   STARTUPINFOW startupInfo;
   PROCESS_INFORMATION processInformation;
--- a/lib/gr/gr.c
+++ b/lib/gr/gr.c
@@ -10117,8 +10117,8 @@ void gr_quiver(int nx, int ny, double *x, double *y, double *u, double *v, int c
         }
     }
 
-  if (!isnan(color_lim_min)) gmin = color_lim_min;
-  if (!isnan(color_lim_max)) gmax = color_lim_max;
+  if (!is_nan(color_lim_min)) gmin = color_lim_min;
+  if (!is_nan(color_lim_max)) gmax = color_lim_max;
   for (j = 0; j < ny; j++)
     for (i = 0; i < nx; i++)
       {
@@ -10707,8 +10707,8 @@ void gr_hexbin_2pass(int nx, int ny, double *x, double *y, int nbins)
           ycm = context->priv->ycm;
           gmax = cntmax;
 
-          if (!isnan(color_lim_min)) gmin = color_lim_min;
-          if (!isnan(color_lim_max)) gmax = color_lim_max;
+          if (!is_nan(color_lim_min)) gmin = color_lim_min;
+          if (!is_nan(color_lim_max)) gmax = color_lim_max;
 
           for (j = 0; j < 6; j++)
             {
@@ -15365,8 +15365,8 @@ void gr_volume_2pass(int nx, int ny, int nz, double *data, int algorithm, double
         }
     }
 
-  if (!isnan(color_lim_min)) dmin = color_lim_min;
-  if (!isnan(color_lim_max)) dmax = color_lim_max;
+  if (!is_nan(color_lim_min)) dmin = color_lim_min;
+  if (!is_nan(color_lim_max)) dmax = color_lim_max;
 
   colormap = (int *)gks_malloc((last_color - first_color + 1) * sizeof(int));
   for (i = first_color; i <= last_color; i++)
--- a/lib/gr3/gr3_gr.c
+++ b/lib/gr3/gr3_gr.c
@@ -2002,8 +2002,8 @@ static void gr3_drawmesh_grlike(int n, const float *positions, const float *dire
         }
 
       gr_inqcolorlimits(&color_lim_min, &color_lim_max);
-      if (!isnan(color_lim_min)) min = color_lim_min;
-      if (!isnan(color_lim_max)) max = color_lim_max;
+      if (!is_nan(color_lim_min)) min = color_lim_min;
+      if (!is_nan(color_lim_max)) max = color_lim_max;
 
       for (i = 0; i < width * height; i++)
         {
@@ -2629,8 +2629,8 @@ static void gr3_drawsurface_grlike(int nx, int ny, const float *px, const float
             }
 
           gr_inqcolorlimits(&color_lim_min, &color_lim_max);
-          if (!isnan(color_lim_min)) min = color_lim_min;
-          if (!isnan(color_lim_max)) max = color_lim_max;
+          if (!is_nan(color_lim_min)) min = color_lim_min;
+          if (!is_nan(color_lim_max)) max = color_lim_max;
 
           for (i = 0; i < width * height; i++)
             {
