diff -bur mbuild-orig/mbuild/env.py mbuild/mbuild/env.py
--- mbuild-orig/mbuild/env.py	2025-04-16 22:22:12.760930900 -0600
+++ mbuild/mbuild/env.py	2025-04-16 22:22:17.163333200 -0600
@@ -1244,7 +1244,7 @@
             return 'x86-64'
         elif name[0:3] == 'x86':
             return 'ia32'
-        elif name in ['aarch64', 'arm64']:
+        elif name in ['aarch64', 'arm64'] or os.environ['MSYSTEM_CARCH'] == 'aarch64':
             return 'aarch64'
         else:
             die("Unknown cpu " + name)
