--- oneTBB-2021.5.0/cmake/compilers/Clang.cmake.orig	2022-05-16 08:54:01.120758100 +0300
+++ oneTBB-2021.5.0/cmake/compilers/Clang.cmake	2022-05-16 08:54:05.693175900 +0300
@@ -54,10 +54,6 @@
     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=2>)
 endif()
 
-if (MINGW)
-    list(APPEND TBB_COMMON_COMPILE_FLAGS -U__STRICT_ANSI__)
-endif()
-
 # Enabling LTO on Android causes the NDK bug.
 # NDK throws the warning: "argument unused during compilation: '-Wa,--noexecstack'"
 if (NOT ANDROID_PLATFORM AND BUILD_SHARED_LIBS)
--- oneTBB-2021.5.0/cmake/compilers/GNU.cmake.orig	2022-05-16 08:53:46.898599900 +0300
+++ oneTBB-2021.5.0/cmake/compilers/GNU.cmake	2022-05-16 08:53:52.123864000 +0300
@@ -59,10 +59,6 @@
     set(TBB_TEST_COMPILE_FLAGS ${TBB_TEST_COMPILE_FLAGS} -DTBB_TEST_LOW_WORKLOAD $<$<CONFIG:DEBUG>:-mxgot>)
 endif()
 
-if (MINGW)
-    list(APPEND TBB_COMMON_COMPILE_FLAGS -U__STRICT_ANSI__)
-endif()
-
 # For some reason GCC does not instrument code with Thread Sanitizer when lto is enabled and C linker is used.
 if (NOT TBB_SANITIZE MATCHES "thread")
     set(TBB_IPO_COMPILE_FLAGS $<$<NOT:$<CONFIG:Debug>>:-flto>)
