--- oneTBB-2021.5.0/cmake/config_generation.cmake.orig	2022-05-13 15:38:42.367469900 +0300
+++ oneTBB-2021.5.0/cmake/config_generation.cmake	2022-05-13 15:40:14.968844600 +0300
@@ -89,9 +89,9 @@
         # Expand TBB_LIB_REL_PATH here in IMPORTED_IMPLIB property and
         # redefine it with TBB_DLL_REL_PATH value to properly fill IMPORTED_LOCATION property in TBBConfig.cmake.in template.
         set(TBB_IMPLIB_RELEASE "
-                                      IMPORTED_IMPLIB_RELEASE \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}.lib\"")
+                                      IMPORTED_IMPLIB_RELEASE \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}.dll.a\"")
         set(TBB_IMPLIB_DEBUG "
-                                      IMPORTED_IMPLIB_DEBUG \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}_debug.lib\"")
+                                      IMPORTED_IMPLIB_DEBUG \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}d.dll.a\"")
         set(TBB_LIB_REL_PATH ${TBB_DLL_REL_PATH})
 
         if (tbb_gen_cfg_HANDLE_SUBDIRS)
--- oneTBB-2021.5.0/src/tbb/CMakeLists.txt.orig	2022-05-13 16:00:17.804666400 +0300
+++ oneTBB-2021.5.0/src/tbb/CMakeLists.txt	2022-05-13 16:00:59.259080300 +0300
@@ -123,7 +123,7 @@
 
 tbb_install_target(tbb)
 
-if (WIN32)
+if (MSVC)
     # Create a copy of target linker file (tbb<ver>[_debug].lib) with legacy name (tbb[_debug].lib)
     # to support previous user experience for linkage.
     install(FILES
