diff --git a/build/windows.gcc.inc b/build/windows.gcc.inc
index eef1c1b8..ed82791e 100644
--- a/build/windows.gcc.inc
+++ b/build/windows.gcc.inc
@@ -59,7 +59,7 @@ TEST_WARNING_KEY = -Wextra -Wshadow -Wcast-qual -Woverloaded-virtual -Wnon-virtu
 WARNING_SUPPRESS = -Wno-parentheses -Wno-uninitialized -Wno-non-virtual-dtor
 DYLIB_KEY = -shared
 LIBDL =
-EXPORT_KEY = -Wl,--version-script,
+EXPORT_KEY = -Wl,--output-def=
 LIBS = -lpsapi
 BIGOBJ_KEY = -Wa,-mbig-obj
 
@@ -77,13 +77,6 @@ ifeq (ok,$(call detect_js,/minversion gcc 4.8))
     RTM_KEY = -mrtm
 endif
 
-# gcc 6.0 and later have -flifetime-dse option that controls
-# elimination of stores done outside the object lifetime
-ifeq (ok,$(call detect_js,/minversion gcc 6.0))
-    # keep pre-contruction stores for zero initialization
-    DSE_KEY = -flifetime-dse=1
-endif
-
 ifeq ($(cfg), release)
         CPLUS_FLAGS = -g -O2
 endif
