diff -bur oneTBB-2020.3-c/include/tbb/tbb_machine.h oneTBB-2020.3/include/tbb/tbb_machine.h
--- oneTBB-2020.3-c/include/tbb/tbb_machine.h	2020-07-10 07:18:01.000000000 -0600
+++ oneTBB-2020.3/include/tbb/tbb_machine.h	2022-08-06 23:25:30.185874300 -0600
@@ -196,7 +196,11 @@
         #if (TBB_USE_GCC_BUILTINS && __TBB_GCC_BUILTIN_ATOMICS_PRESENT)
             #include "machine/gcc_generic.h"
         #elif __MINGW64__
+            #if defined(__aarch64__)
+                #include "machine/gcc_arm.h"
+            #else
             #include "machine/linux_intel64.h"
+            #endif
         #elif __MINGW32__
             #include "machine/linux_ia32.h"
         #endif
diff -bur oneTBB-2020.3-c/src/tbbmalloc/TypeDefinitions.h oneTBB-2020.3/src/tbbmalloc/TypeDefinitions.h
--- oneTBB-2020.3-c/src/tbbmalloc/TypeDefinitions.h	2020-07-10 07:18:01.000000000 -0600
+++ oneTBB-2020.3/src/tbbmalloc/TypeDefinitions.h	2022-08-06 23:26:59.192388100 -0600
@@ -27,6 +27,8 @@
 #       define __ARCH_x86_32 1
 #   elif defined(_M_ARM)
 #       define __ARCH_other 1
+#   elif defined(__aarch64__)
+#       define __ARCH_other 1
 #   else
 #       error Unknown processor architecture for Windows
 #   endif
