--- pybind11-2.2.4/tools/FindPythonLibsNew.cmake.orig	2018-10-31 11:29:57.131339500 +0300
+++ pybind11-2.2.4/tools/FindPythonLibsNew.cmake	2018-10-31 11:30:08.082558800 +0300
@@ -138,7 +138,7 @@
 string(REGEX REPLACE "\\\\" "/" PYTHON_INCLUDE_DIR ${PYTHON_INCLUDE_DIR})
 string(REGEX REPLACE "\\\\" "/" PYTHON_SITE_PACKAGES ${PYTHON_SITE_PACKAGES})
 
-if(CMAKE_HOST_WIN32)
+if(MSVC)
     set(PYTHON_LIBRARY
         "${PYTHON_PREFIX}/libs/Python${PYTHON_LIBRARY_SUFFIX}.lib")
 
