--- a/cmake/rkcommon_macros.cmake
+++ b/cmake/rkcommon_macros.cmake
@@ -93,7 +93,7 @@
   endif()
 
   # setting DEPENDENTLOADFLAG:LOAD_LIBRARY_SAFE_CURRENT_DIRS on rkcommon DLL
-  if(WIN32)
+  if(WIN32 AND NOT MINGW)
     if(RKCOMMON_COMPILER_MSVC)
       set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEPENDENTLOADFLAG:0x2000")
       set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /DEPENDENTLOADFLAG:0x2000")
--- a/rkcommon/common.h
+++ b/rkcommon/common.h
@@ -16,11 +16,13 @@
 #define _USE_MATH_DEFINES
 #endif
 #include <cmath>
+#ifdef _MSC_VER
 #ifdef _M_X64
 typedef long long ssize_t;
 #else
 typedef int ssize_t;
 #endif
+#endif
 #else
 // ----------- NOT windows -----------
 #include "unistd.h"
--- a/rkcommon/math/rkmath.h
+++ b/rkcommon/math/rkmath.h
@@ -11,7 +11,7 @@
 
 // Include vector intrinsics
 #ifndef RKCOMMON_NO_SIMD
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(__aarch64__)
 #include <intrin.h>
 #elif defined(__ARM_NEON)
 #include "arm/emulation.h"
--- a/rkcommon/os/library.cpp
+++ b/rkcommon/os/library.cpp
@@ -216,7 +216,7 @@
   void *Library::getSymbol(const std::string &sym) const
   {
 #ifdef _WIN32
-    return GetProcAddress((HMODULE)lib, sym.c_str());
+    return reinterpret_cast<void*>(GetProcAddress((HMODULE)lib, sym.c_str()));
 #else
     return dlsym(lib, sym.c_str());
 #endif
--- a/rkcommon/platform.h
+++ b/rkcommon/platform.h
@@ -160,7 +160,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 /* windows does not have ssize_t */
-#ifdef __WIN32
+#ifdef _MSC_VER
 #ifdef __X86_64__
 typedef int64_t ssize_t;
 #else
--- a/rkcommon/tasking/detail/tasking_system_init.cpp
+++ b/rkcommon/tasking/detail/tasking_system_init.cpp
@@ -22,7 +22,7 @@
 #ifndef RKCOMMON_NO_SIMD
 #if !defined(__ARM_NEON)
 #include <xmmintrin.h>
-#elif !defined(_WIN32)
+#elif !defined(_MSC_VER)
 #include "math/arm/emulation.h"
 #endif
 /* normally defined in pmmintrin.h, but we always need this */
