diff --git a/make/configure.py b/make/configure.py
index a383cb3..6bc93ef 100644
--- a/make/configure.py
+++ b/make/configure.py
@@ -163,7 +163,7 @@ class Configure( object ):
         for dir in path.split( os.pathsep ):
             f = os.path.join( dir, name )
             if os.access( f, os.X_OK ):
-                return f
+                return os.path.normpath(f)
         return None
 
     ## taken from python2.6 -- we need it
@@ -499,7 +499,7 @@ class BuildTupleProbe( ShellProbe, list ):
     GNU_TUPLE_RE = '([^-]+)-?([^-]*)-([^0-9-]+)([^-]*)-?([^-]*)'
 
     def __init__( self ):
-        super( BuildTupleProbe, self ).__init__( 'build tuple', '%s/config.guess' % (cfg.dir), abort=True, head=True )
+        super( BuildTupleProbe, self ).__init__( 'build tuple', 'sh %s/config.guess' % (cfg.dir), abort=True, head=True )
 
     def _parseSession( self ):
         self.spec = self.session[0].decode('utf-8') if self.session else ''
@@ -1032,6 +1032,8 @@ class ToolProbe( Action ):
                 name = None
             if name is not None:
                 self.names.append( name )
+                if ".exe" not in name:
+                    self.names.append( name + ".exe" )
         self.name = self.names[0]
         self.pretext = self.name
         self.pathname = self.names[0]
@@ -1107,6 +1109,8 @@ class VersionProbe( Action ):
     def _action( self ):
         with open(os.devnull, 'w') as devnull:
             ## pipe and redirect stderr to dev/null; effects communicate result
+            if ".exe" not in self.command[0]:
+                self.command.insert(0, "sh")
             pipe = subprocess.Popen( self.command, stdout=subprocess.PIPE, stderr=devnull )
 
         ## read data into memory buffers
@@ -1678,14 +1682,14 @@ try:
         else:
             gmake  = ToolProbe( 'GMAKE.exe',      'make',       'gmake', 'make', abort=True )
 
-        autoconf   = ToolProbe( 'AUTOCONF.exe',   'autoconf',   'autoconf', abort=True, minversion=([2,71,0] if build_tuple.match('*-*-darwin*') else [2,69,0]) )
-        automake   = ToolProbe( 'AUTOMAKE.exe',   'automake',   'automake', abort=True, minversion=[1,13,0] )
-        libtool    = ToolProbe( 'LIBTOOL.exe',    'libtool',    'libtool', abort=True )
+        autoconf   = ToolProbe( 'AUTOCONF.exe',   'autoconf',   'autoconf', abort=False, minversion=([2,71,0] if build_tuple.match('*-*-darwin*') else [2,69,0]) )
+        automake   = ToolProbe( 'AUTOMAKE.exe',   'automake',   'automake', abort=False, minversion=[1,13,0] )
+        libtool    = ToolProbe( 'LIBTOOL.exe',    'libtool',    'libtool', abort=False )
         lipo       = ToolProbe( 'LIPO.exe',       'lipo',       'lipo', abort=False )
-        pkgconfig  = ToolProbe( 'PKGCONFIG.exe',  'pkgconfig',  'pkg-config', abort=True, minversion=[0,27,0] )
-        meson      = ToolProbe( 'MESON.exe',      'meson',      'meson', abort=True, minversion=[0,51,0] )
-        nasm       = ToolProbe( 'NASM.exe',       'asm',        'nasm', abort=True, minversion=[2,13,0] )
-        ninja      = ToolProbe( 'NINJA.exe',      'ninja',      'ninja-build', 'ninja', abort=True )
+        pkgconfig  = ToolProbe( 'PKGCONFIG.exe',  'pkgconfig',  'pkg-config', abort=False, minversion=[0,27,0] )
+        meson      = ToolProbe( 'MESON.exe',      'meson',      'meson', abort=False, minversion=[0,51,0] )
+        nasm       = ToolProbe( 'NASM.exe',       'asm',        'nasm', abort=False, minversion=[2,13,0] )
+        ninja      = ToolProbe( 'NINJA.exe',      'ninja',      'ninja-build', 'ninja', abort=False )
         cargo      = ToolProbe( 'CARGO.exe',      'cargo',        'cargo', abort=False )
         cargoc     = ToolProbe( 'CARGO-C.exe',    'cargo-cbuild', 'cargo-cbuild', abort=False )
 
@@ -1786,7 +1790,7 @@ try:
     ## Additional library and tool checks
     #####################################
     # Requires oneVPL which requires CMake 3.16.3 or later
-    Tools.cmake = ToolProbe('CMAKE.exe', 'cmake', 'cmake', abort=True, minversion=[3,16,3])
+    Tools.cmake = ToolProbe('CMAKE.exe', 'cmake', 'cmake', abort=False, minversion=[3,16,3])
     Tools.cmake.__init__( Tools.cmake.var, Tools.cmake.option, Tools.cmake.name, **Tools.cmake.kwargs )
     Tools.cmake.run()
     for action in Action.actions:
diff --git a/test/module.defs b/test/module.defs
index 9f803c4..ee6b87d 100644
--- a/test/module.defs
+++ b/test/module.defs
@@ -76,7 +76,7 @@ ifneq (,$(filter $(HOST.release),2.10))
     TEST.GCC.l += iconv
 endif
     TEST.GCC.D += _POSIX_C_SOURCE=200112L __EXTENSIONS__
-else ifeq (1-mingw,$(HOST.cross)-$(HOST.system))
+else ifeq (0-mingw,$(HOST.cross)-$(HOST.system))
 ifeq ($(HAS.dlfcn),1)
     TEST.GCC.l += dl
 endif
@@ -85,10 +85,9 @@ ifeq (1,$(HAS.pthread))
 else
     TEST.GCC.l += pthreadGC2
 endif
-    TEST.GCC.l += bcrypt iconv ws2_32 regex uuid ole32 gdi32 userenv ntdll
+    TEST.GCC.l += x265 bcrypt iconv ws2_32 regex uuid ole32 gdi32 userenv ntdll
     ifeq (1,$(FEATURE.mf))
         TEST.GCC.l += mfplat strmiids
     endif
     TEST.GCC.D += PTW32_STATIC_LIB
-    TEST.GCC.args.extra.exe++ += -static
-endif #   (1-mingw,$(HOST.cross)-$(HOST.system))
+endif #   (0-mingw,$(HOST.cross)-$(HOST.system))
