diff -bur HandBrake-1.9.2-orig/make/config.guess HandBrake-1.9.2/make/config.guess
--- HandBrake-1.9.2-orig/make/config.guess	2025-03-15 14:21:57 -0600
+++ HandBrake-1.9.2/make/config.guess	2025-03-15 14:22:40 -0600
@@ -144,7 +144,7 @@
 	PATH=$PATH:/.attbin ; export PATH
 fi
 
-UNAME_MACHINE=`(uname -m) 2>/dev/null` || UNAME_MACHINE=unknown
+UNAME_MACHINE=$MSYSTEM_CARCH
 UNAME_RELEASE=`(uname -r) 2>/dev/null` || UNAME_RELEASE=unknown
 UNAME_SYSTEM=`(uname -s) 2>/dev/null` || UNAME_SYSTEM=unknown
 UNAME_VERSION=`(uname -v) 2>/dev/null` || UNAME_VERSION=unknown
