diff --git a/test/module.defs b/test/module.defs
--- a/test/module.defs
+++ b/test/module.defs
@@ -21,6 +21,10 @@ TEST.pkgconfig_libs = libass libavformat libavfilter libavcodec libavutil libswres
 	dvdread libswscale theoraenc theoradec vorbis vorbisenc ogg x264 libbluray \
 	jansson libturbojpeg SvtAv1Enc
 
+ifeq (1,$(FEATURE.libdovi))
+TEST.pkgconfig_libs += dovi
+endif
+
 TEST.pkgconfig_libs += $(foreach m,$(MODULES.NAMES),$($m.OSL.libs))
 
 ifeq (1,$(FEATURE.flatpak))
