diff -Naur chrono-10.0.0.orig/CMakeLists.txt chrono-10.0.0/CMakeLists.txt
--- chrono-10.0.0.orig/CMakeLists.txt	2026-04-07 15:33:26.000000000 +0300
+++ chrono-10.0.0/CMakeLists.txt	2026-04-25 22:53:49.544293400 +0300
@@ -237,7 +237,7 @@
 # (relative to CMAKE_CURRENT_SOURCE_DIR)
 #------------------------------------------------------------------------
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+if(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     set(CH_INSTALL_DATA "data")
     set(CH_INSTALL_SENSOR_SHADERS "sensor_shaders")
     set(CH_INSTALL_DEMO "bin")
@@ -246,7 +246,7 @@
     set(CH_INSTALL_DATA "share/chrono/data")
     set(CH_INSTALL_SENSOR_SHADERS "share/chrono/sensor_shaders")
     set(CH_INSTALL_DEMO "share/chrono/bin")
-    set(CH_INSTALL_PYTHON "share/chrono/python")
+    set(CH_INSTALL_PYTHON "lib/python3.14/site-packages")
 endif()
 
 #------------------------------------------------------------
@@ -260,13 +260,13 @@
 
 install(DIRECTORY ${CMAKE_SOURCE_DIR}/data/ DESTINATION ${CH_INSTALL_DATA})
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+if(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
   install(DIRECTORY ${CMAKE_SOURCE_DIR}/template_project/ DESTINATION "template_project")
 else()
   install(DIRECTORY ${CMAKE_SOURCE_DIR}/template_project/ DESTINATION "share/chrono/template_project")
 endif()
 
-install(DIRECTORY ${CMAKE_SOURCE_DIR}/src/importer_blender/ DESTINATION "importer_blender")
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/src/importer_blender/ DESTINATION "share/chrono/importer_blender")
 
 #------------------------------------------------------------
 # Defer configuration of all Chrono libraries and programs
diff -Naur chrono-10.0.0.orig/src/CMakeLists.txt chrono-10.0.0/src/CMakeLists.txt
--- chrono-10.0.0.orig/src/CMakeLists.txt	2026-04-07 15:33:26.000000000 +0300
+++ chrono-10.0.0/src/CMakeLists.txt	2026-04-25 22:50:24.458723300 +0300
@@ -124,7 +124,7 @@
 # Create installation directory
 #-----------------------------------------------------------------------------
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+if(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     set(CH_CONFIG_INSTALL_PATH "cmake")
 else()
     set(CH_CONFIG_INSTALL_PATH "lib/cmake/Chrono")
diff -Naur chrono-10.0.0.orig/src/chrono/CMakeLists.txt chrono-10.0.0/src/chrono/CMakeLists.txt
--- chrono-10.0.0.orig/src/chrono/CMakeLists.txt	2026-04-07 15:33:26.000000000 +0300
+++ chrono-10.0.0/src/chrono/CMakeLists.txt	2026-04-25 22:50:24.474586400 +0300
@@ -1480,7 +1480,9 @@
 	elseif("${CH_COMPILER}" STREQUAL "COMPILER_MSVC_X64")
         target_link_libraries(Chrono_core PUBLIC Ws2_32.lib)
 	elseif("${CH_COMPILER}" STREQUAL "COMPILER_GCC")
+        target_link_libraries(Chrono_core PUBLIC ws2_32 winmm)
 	elseif("${CH_COMPILER}" STREQUAL "COMPILER_GCC_X64")
+        target_link_libraries(Chrono_core PUBLIC ws2_32 winmm)
 	endif()
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
diff -Naur chrono-10.0.0.orig/src/chrono/utils/ChSocket.cpp chrono-10.0.0/src/chrono/utils/ChSocket.cpp
--- chrono-10.0.0.orig/src/chrono/utils/ChSocket.cpp	2026-04-07 15:33:26.000000000 +0300
+++ chrono-10.0.0/src/chrono/utils/ChSocket.cpp	2026-04-25 22:50:24.474586400 +0300
@@ -16,6 +16,8 @@
 
 #include "chrono/utils/ChSocket.h"
 
+#include <cstdint>
+
 namespace chrono {
 namespace utils {
 
