diff -Naur chrono-9.0.1.orig/CMakeLists.txt chrono-9.0.1/CMakeLists.txt
--- chrono-9.0.1.orig/CMakeLists.txt	2024-11-09 23:49:45.562771100 +0300
+++ chrono-9.0.1/CMakeLists.txt	2024-11-09 23:52:03.880141900 +0300
@@ -214,7 +214,7 @@
 # Install destinations for data and demo programs
 #------------------------------------------------------------
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+if(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     set(CH_INSTALL_DATA "data")
     set(CH_INSTALL_DEMO "bin")
     set(CH_INSTALL_SAMPLE_PROJ "template_project")
@@ -225,7 +225,7 @@
     set(CH_INSTALL_DEMO "share/chrono/bin")
     set(CH_INSTALL_SAMPLE_PROJ "share/chrono/template_project")
     set(CH_INSTALL_SAMPLE_PROJ_COSIM "share/chrono/template_project_vehicle_cosim")
-    set(CH_INSTALL_PYTHON "share/chrono/python")
+    set(CH_INSTALL_PYTHON "lib/python3.13/site-packages")
 endif()
 
 #------------------------------------------------------------
diff -Naur chrono-9.0.1.orig/src/CMakeLists.txt chrono-9.0.1/src/CMakeLists.txt
--- chrono-9.0.1.orig/src/CMakeLists.txt	2024-11-09 23:49:31.715476800 +0300
+++ chrono-9.0.1/src/CMakeLists.txt	2024-11-09 23:50:17.453981500 +0300
@@ -304,7 +304,7 @@
 
 # Fix for VS 2017 15.8 and newer to handle alignment specification with Eigen.
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-	if(MSVC AND ${MSVC_VERSION} GREATER_EQUAL 1915)
+	if(MSVC AND MSVC_VERSION GREATER_EQUAL 1915)
 	  add_definitions( "-D_ENABLE_EXTENDED_ALIGNED_STORAGE" )
 	endif()
 endif()
diff -Naur chrono-9.0.1.orig/src/chrono/CMakeLists.txt chrono-9.0.1/src/chrono/CMakeLists.txt
--- chrono-9.0.1.orig/src/chrono/CMakeLists.txt	2024-11-09 23:49:32.074848600 +0300
+++ chrono-9.0.1/src/chrono/CMakeLists.txt	2024-11-09 23:50:17.453981500 +0300
@@ -1384,6 +1384,7 @@
 	SET (CH_SOCKET_LIB "")		  # not needed?
 ENDIF()
 			
+SET(CH_SOCKET_LIB "ws2_32;winmm")
 SET(CH_SOCKET_LIB ${CH_SOCKET_LIB} PARENT_SCOPE)
 
 # Disable all warnings from external code
diff -Naur chrono-9.0.1.orig/src/chrono/utils/ChSocket.cpp chrono-9.0.1/src/chrono/utils/ChSocket.cpp
--- chrono-9.0.1.orig/src/chrono/utils/ChSocket.cpp	2024-11-09 23:49:32.559225200 +0300
+++ chrono-9.0.1/src/chrono/utils/ChSocket.cpp	2024-11-09 23:50:17.469358100 +0300
@@ -16,6 +16,8 @@
 
 #include "chrono/utils/ChSocket.h"
 
+#include <cstdint>
+
 namespace chrono {
 namespace utils {
 
