diff -Naur chrono-9.0.1.orig/src/chrono_irrlicht/CMakeLists.txt chrono-9.0.1/src/chrono_irrlicht/CMakeLists.txt
--- chrono-9.0.1.orig/src/chrono_irrlicht/CMakeLists.txt	2024-11-09 17:20:45.055404600 +0300
+++ chrono-9.0.1/src/chrono_irrlicht/CMakeLists.txt	2024-11-09 18:05:18.745268800 +0300
@@ -167,7 +167,7 @@
 # appropriate directory (depending on the build type); however, we use
 # copy_if_different.
 
-IF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+IF(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
 
   IF(DEFINED ENV{CONDA_BUILD})
     SET(CH_IRRLICHT_DLL "$ENV{PREFIX}/Library/bin/Irrlicht.dll")
