--- chrono-10.0.0.orig/src/chrono_swig/chrono_python/CMakeLists.txt	2026-04-07 15:33:26.000000000 +0300
+++ chrono-10.0.0/src/chrono_swig/chrono_python/CMakeLists.txt	2026-04-25 21:59:39.529939400 +0300
@@ -781,7 +781,7 @@
   endif()
 
 
-  IF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  IF(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     target_compile_definitions(cascade PRIVATE "/DWNT")
   ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     target_compile_definitions(cascade PRIVATE "-DHAVE_IOSTREAM")
@@ -878,7 +878,7 @@
     target_link_libraries(${CHPY_VEHICLE} PRIVATE Chrono_fsi Chrono_fsisph)
   endif()
 
-  IF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  IF(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     target_compile_definitions(${CHPY_VEHICLE} PRIVATE "/DWNT")
   ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     target_compile_definitions(${CHPY_VEHICLE} PRIVATE "-DHAVE_IOSTREAM")
@@ -1047,7 +1047,7 @@
   swig_add_library(${CHPY_RS} LANGUAGE python SOURCES ${CHPY_ROBOT_MODULE_FILE})
   target_link_libraries(${CHPY_RS} PRIVATE Python3::Python Chrono_core ChronoModels_robot)
 
-  IF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  IF(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     target_compile_definitions(${CHPY_RS} PRIVATE "/DWNT")
   ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     target_compile_definitions(${CHPY_RS} PRIVATE "-DHAVE_IOSTREAM")
@@ -1120,7 +1120,7 @@
   swig_add_library(parsers LANGUAGE python SOURCES ${CHPY_PARSERS_MODULE_FILE})
   target_link_libraries(parsers PRIVATE Python3::Python Chrono_core Chrono_parsers)
 
-  IF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  IF(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     target_compile_definitions(parsers PRIVATE "/DWNT")
   ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     target_compile_definitions(parsers PRIVATE "-DHAVE_IOSTREAM")
@@ -1202,7 +1202,7 @@
   swig_add_library(ros LANGUAGE python SOURCES ${CHPY_ROS_MODULE_FILE})
   target_link_libraries(ros PRIVATE Python3::Python Chrono_core Chrono_ros)
 
-  IF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  IF(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     target_compile_definitions(ros PRIVATE "/DWNT")
   ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     target_compile_definitions(ros PRIVATE "-DHAVE_IOSTREAM")
