diff -Naur chrono-9.0.1.orig/doxygen/Doxyfile chrono-9.0.1/doxygen/Doxyfile
--- chrono-9.0.1.orig/doxygen/Doxyfile	2024-11-10 14:14:50.559288400 +0300
+++ chrono-9.0.1/doxygen/Doxyfile	2024-11-10 14:25:10.984962400 +0300
@@ -58,7 +58,7 @@
 # entered, it will be relative to the location where doxygen was started. If
 # left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = /tmp/chrono_doxygen/
+OUTPUT_DIRECTORY       = ../../build-doxygen/
 
 # If the CREATE_SUBDIRS tag is set to YES then doxygen will create 4096 sub-
 # directories (in 2 levels) under the output directory of each output format and
@@ -1603,7 +1603,7 @@
 # The default value is: https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/.
 # This tag requires that the tag USE_MATHJAX is set to YES.
 
-MATHJAX_RELPATH        = https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/
+MATHJAX_RELPATH        = ../mathjax
 
 # The MATHJAX_EXTENSIONS tag can be used to specify one or more MathJax
 # extension names that should be enabled during MathJax rendering. For example
