diff -Naur chrono-9.0.1.orig/src/chrono_opengl/CMakeLists.txt chrono-9.0.1/src/chrono_opengl/CMakeLists.txt
--- chrono-9.0.1.orig/src/chrono_opengl/CMakeLists.txt	2024-11-17 21:15:41.217132300 +0300
+++ chrono-9.0.1/src/chrono_opengl/CMakeLists.txt	2024-11-17 21:16:39.280402200 +0300
@@ -310,7 +310,7 @@
 # On Windows, copy the GLEW and GLFW DLLs (if specified)
 #-------------------------------------------------------------------------------
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+if(FALSE) #${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     if(EXISTS "${GLEW_DLL}")
       ADD_CUSTOM_COMMAND(TARGET ChronoEngine_opengl POST_BUILD
                          COMMAND ${CMAKE_COMMAND} -E copy_if_different
