diff -Naur chrono-9.0.1.orig/src/chrono_vsg/ChVisualSystemVSG.cpp chrono-9.0.1/src/chrono_vsg/ChVisualSystemVSG.cpp
--- chrono-9.0.1.orig/src/chrono_vsg/ChVisualSystemVSG.cpp	2024-11-18 22:36:02.514857100 +0300
+++ chrono-9.0.1/src/chrono_vsg/ChVisualSystemVSG.cpp	2024-11-18 21:47:17.545116300 +0300
@@ -716,8 +716,10 @@
     directionalLight->name = "sun light";
     directionalLight->color.set(1.0f, 1.0f, 1.0f);
     directionalLight->intensity = m_lightIntensity;
-    if (m_use_shadows)
-        directionalLight->shadowMaps = 10;
+    if (m_use_shadows) {
+        directionalLight->shadowSettings = vsg::ShadowSettings::create();
+        directionalLight->shadowSettings->shadowMapCount = 10;
+    }
 
     double se = std::sin(m_elevation);
     double ce = std::cos(m_elevation);
