diff -bur chrono-9.0.1-orig/cmake/FindEigen3.cmake chrono-9.0.1/cmake/FindEigen3.cmake
--- chrono-9.0.1-orig/cmake/FindEigen3.cmake	2026-01-03 09:10:21.627357900 -0700
+++ chrono-9.0.1/cmake/FindEigen3.cmake	2026-01-03 09:10:39.499144900 -0700
@@ -36,7 +36,11 @@
 endif(NOT Eigen3_FIND_VERSION)
 
 macro(_eigen3_check_version)
+  if(EXISTS "${EIGEN3_INCLUDE_DIR}/Eigen/Version")
+    file(READ "${EIGEN3_INCLUDE_DIR}/Eigen/Version" _eigen3_version_header)
+  else()
   file(READ "${EIGEN3_INCLUDE_DIR}/Eigen/src/Core/util/Macros.h" _eigen3_version_header)
+  endif()
 
   string(REGEX MATCH "define[ \t]+EIGEN_WORLD_VERSION[ \t]+([0-9]+)" _eigen3_world_version_match "${_eigen3_version_header}")
   set(EIGEN3_WORLD_VERSION "${CMAKE_MATCH_1}")
