diff -bur chrono-9.0.1-orig/src/chrono_vsg/shapes/ShaderUtils.cpp chrono-9.0.1/src/chrono_vsg/shapes/ShaderUtils.cpp
--- chrono-9.0.1-orig/src/chrono_vsg/shapes/ShaderUtils.cpp	2026-01-03 09:30:46.679143000 -0700
+++ chrono-9.0.1/src/chrono_vsg/shapes/ShaderUtils.cpp	2026-01-03 10:08:28.892267500 -0700
@@ -134,9 +134,9 @@
     shaderSet->addPushConstantRange("pc", "", VK_SHADER_STAGE_VERTEX_BIT, 0, 128);
 
     shaderSet->definesArrayStates.push_back(vsg::DefinesArrayState{
-        {"VSG_INSTANCE_POSITIONS", "VSG_DISPLACEMENT_MAP"}, vsg::PositionAndDisplacementMapArrayState::create()});
+        {"VSG_INSTANCE_POSITIONS", "VSG_DISPLACEMENT_MAP"}, vsg::TranslationAndDisplacementMapArrayState::create()});
     shaderSet->definesArrayStates.push_back(
-        vsg::DefinesArrayState{{"VSG_INSTANCE_POSITIONS"}, vsg::PositionArrayState::create()});
+        vsg::DefinesArrayState{{"VSG_INSTANCE_POSITIONS"}, vsg::TranslationArrayState::create()});
     shaderSet->definesArrayStates.push_back(
         vsg::DefinesArrayState{{"VSG_DISPLACEMENT_MAP"}, vsg::DisplacementMapArrayState::create()});
     shaderSet->definesArrayStates.push_back(
diff -bur chrono-9.0.1-orig/src/chrono_vsg/utils/ChUtilsVSG.cpp chrono-9.0.1/src/chrono_vsg/utils/ChUtilsVSG.cpp
--- chrono-9.0.1-orig/src/chrono_vsg/utils/ChUtilsVSG.cpp	2026-01-03 09:30:46.680143300 -0700
+++ chrono-9.0.1/src/chrono_vsg/utils/ChUtilsVSG.cpp	2026-01-03 10:05:20.998217600 -0700
@@ -238,7 +238,7 @@
     // create texture image and associated DescriptorSets and binding
     auto sampler = vsg::Sampler::create();
     const auto layout = data->getLayout();
-    sampler->maxLod = layout.maxNumMipmaps;
+    sampler->maxLod = layout.mipLevels;
 
     auto texture = vsg::DescriptorImage::create(sampler, data, 0, 0, VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER);
 
