diff -bur portsmf-orig/configure.ac portsmf/configure.ac
--- portsmf-orig/configure.ac	2023-02-01 02:00:58.257411500 -0700
+++ portsmf/configure.ac	2023-02-01 02:03:22.187179700 -0700
@@ -23,7 +23,7 @@
 dnl keep autools files in a subdirectory
 AC_CONFIG_AUX_DIR(autotools)
 # which also contains a subdir containing macros
-AC_CONFIG_MACRO_DIR([autotools/m4])
+dnl AC_CONFIG_MACRO_DIR([autotools/m4])
 dnl set up automake
 AM_INIT_AUTOMAKE([-Wall foreign])
 dnl Audacity policy: don't enable automatic rebuild of configure et al if 
@@ -49,7 +49,8 @@
 dnl we will need an "install" program to be available
 AC_PROG_INSTALL
 dnl and ranlib for static libraries
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
+LT_INIT([win32-dll])
 
 dnl Make the install prefix available to the program so that the pkg-config file
 dnl can be created correctly
diff -bur portsmf-orig/Makefile.am portsmf/Makefile.am
--- portsmf-orig/Makefile.am	2023-02-01 02:00:58.241785900 -0700
+++ portsmf/Makefile.am	2023-02-01 02:03:55.411446900 -0700
@@ -6,10 +6,11 @@
 ACLOCAL_AMFLAGS = -I autotools/m4
 
 # define the output library (link with -lportSMF)
-lib_LIBRARIES = libportSMF.a
+lib_LTLIBRARIES = libportSMF.la
 
 # define the source files to use
-libportSMF_a_SOURCES = allegro.cpp \
+libportSMF_la_LDFLAGS = -no-undefined
+libportSMF_la_SOURCES = allegro.cpp \
 	allegrosmfwr.cpp \
 	allegrord.cpp \
 	allegrowr.cpp \
