Index: DAEValidator/CMakeLists.txt
===================================================================
--- DAEValidator/CMakeLists.txt.orig
+++ DAEValidator/CMakeLists.txt
@@ -91,16 +91,16 @@
 target_link_libraries(DAEValidatorExecutable ${Libraries})
 set_target_properties(DAEValidatorExecutable PROPERTIES OUTPUT_NAME DAEValidator)
 
-if (WIN32)
+if (MSVC)
 # C4505: 'function' : unreferenced local function has been removed
 # C4514: 'function' : unreferenced inline function has been removed
 # C4592: symbol will be dynamically initialized (implementation limitation)
 # C4710: 'function' : function not inlined
 # C4711: function 'function' selected for inline expansion
 # C4820: 'bytes' bytes padding added after construct 'member_name'
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP /Wall /WX /wd4505 /wd4514 /wd4592 /wd4710 /wd4711 /wd4820")
 else ()
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Werror")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")
 endif ()
 
 add_custom_command(
@@ -128,7 +128,7 @@
     endforeach(templateFile)
 endmacro(configure_files)
 
-if (WIN32 AND ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION} GREATER 3.0.1)
+if (MSVC AND ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION} GREATER 3.0.1)
 	set(DAEValidatorTestsSources
 		tests/src/ArgumentParserTests.cpp
 		tests/src/Common.cpp
