Index: common/libBuffer/CMakeLists.txt
===================================================================
--- common/libBuffer/CMakeLists.txt.orig
+++ common/libBuffer/CMakeLists.txt
@@ -28,7 +28,10 @@ set(SRC
 	include/performanceTest/performanceTest.h
 )
 
-set(TARGET_LIBS ftoa)
+set(TARGET_LIBS
+	ftoa
+	UTF
+)
 
 include_directories(
 	${libBuffer_include_dirs}
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -270,10 +270,10 @@ else ()  # if pcre not found building it
 endif ()
 
 # building required libs
+add_subdirectory(${EXTERNAL_LIBRARIES}/UTF)                                    
+add_subdirectory(${EXTERNAL_LIBRARIES}/MathMLSolver)
 add_subdirectory(common/libftoa)
 add_subdirectory(common/libBuffer)
-add_subdirectory(${EXTERNAL_LIBRARIES}/UTF)
-add_subdirectory(${EXTERNAL_LIBRARIES}/MathMLSolver)
 
 # building OpenCOLLADA libs
 add_subdirectory(COLLADABaseUtils)
