--- boost_1_77_0/boost/process/detail/windows/file_descriptor.hpp.orig	2021-09-24 11:48:59.101187700 +0300
+++ boost_1_77_0/boost/process/detail/windows/file_descriptor.hpp	2021-09-24 11:49:17.484092500 +0300
@@ -102,6 +102,7 @@
         if (_handle != ::boost::winapi::INVALID_HANDLE_VALUE_)
             ::boost::winapi::CloseHandle(_handle);
         _handle = boost::exchange(other._handle, ::boost::winapi::INVALID_HANDLE_VALUE_);
+        return *this;
     }
 
     ~file_descriptor()
