diff -rupN faac-1.28.orig/libfaac/Makefile.am faac-1.28/libfaac/Makefile.am
--- faac-1.28.orig/libfaac/Makefile.am	2007-08-20 14:03:23.000000000 -0400
+++ faac-1.28/libfaac/Makefile.am	2014-10-30 16:05:10.042800000 -0400
@@ -7,6 +7,7 @@ endif
 libfaac_la_SOURCES = $(main_SOURCES) $(drm_SOURCES)
 libfaac_la_INCLUDES = aacquant.h channels.h filtbank.h hufftab.h psych.h backpred.h coder.h frame.h midside.h tns.h bitstream.h fft.h huffman.h ltp.h util.h
 libfaac_la_LIBADD = -lm
+libfaac_la_LDFLAGS = -no-undefined
 
 INCLUDES = -I$(top_srcdir)/include
 
diff -rupN faac-1.28.orig/libfaac/Makefile.in faac-1.28/libfaac/Makefile.in
--- faac-1.28.orig/libfaac/Makefile.in	2009-02-10 03:37:01.000000000 -0500
+++ faac-1.28/libfaac/Makefile.in	2014-10-30 16:05:43.057800000 -0400
@@ -191,6 +191,7 @@ main_SOURCES = aacquant.c bitstream.c ff
 libfaac_la_SOURCES = $(main_SOURCES) $(drm_SOURCES)
 libfaac_la_INCLUDES = aacquant.h channels.h filtbank.h hufftab.h psych.h backpred.h coder.h frame.h midside.h tns.h bitstream.h fft.h huffman.h ltp.h util.h
 libfaac_la_LIBADD = -lm
+libfaac_la_LDFLAGS = -no-undefined
 INCLUDES = -I$(top_srcdir)/include
 all: all-am
 
@@ -252,7 +253,7 @@ clean-libLTLIBRARIES:
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
-libfaac.la: $(libfaac_la_OBJECTS) $(libfaac_la_DEPENDENCIES) 
+libfaac.la: $(libfaac_la_OBJECTS) $(libfaac_la_DEPENDENCIES)
 	$(LINK) -rpath $(libdir) $(libfaac_la_LDFLAGS) $(libfaac_la_OBJECTS) $(libfaac_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
