diff -r -c xxHash-0.6.0.orig/cmake_unofficial/CMakeLists.txt xxHash-0.6.0/cmake_unofficial/CMakeLists.txt
*** xxHash-0.6.0.orig/cmake_unofficial/CMakeLists.txt	2016-06-02 05:45:37.000000000 -0300
--- xxHash-0.6.0/cmake_unofficial/CMakeLists.txt	2016-06-02 13:11:49.892088500 -0300
***************
*** 21,24 ****
  
  
  INSTALL(FILES ../xxhash.h DESTINATION include)
! INSTALL(TARGETS ${install_libs} DESTINATION lib)
--- 21,27 ----
  
  
  INSTALL(FILES ../xxhash.h DESTINATION include)
! INSTALL(TARGETS ${install_libs}
!         RUNTIME DESTINATION bin
!         LIBRARY DESTINATION lib
!         ARCHIVE DESTINATION lib)
