--- xxHash-0.6.5/cmake_unofficial/CMakeLists.txt.orig	2018-05-17 13:14:35.479246800 +0300
+++ xxHash-0.6.5/cmake_unofficial/CMakeLists.txt	2018-05-17 13:16:24.270765500 +0300
@@ -91,6 +91,7 @@
   install(TARGETS xxhsum
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
   install(TARGETS xxhash
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
   install(FILES "${XXHASH_DIR}/xxhash.h"
