diff -c -r xxHash-0.5.1.orig/cmake_unofficial/CMakeLists.txt xxHash-0.5.1/cmake_unofficial/CMakeLists.txt
*** xxHash-0.5.1.orig/cmake_unofficial/CMakeLists.txt	2016-03-01 09:22:23.000000000 -0300
--- xxHash-0.5.1/cmake_unofficial/CMakeLists.txt	2016-02-11 08:37:24.000000000 -0200
***************
*** 21,24 ****
  
  
  INSTALL(FILES ../xxhash.h DESTINATION include)
! INSTALL(TARGETS ${install_libs} DESTINATION lib)
--- 21,27 ----
  
  
  INSTALL(FILES ../xxhash.h DESTINATION include)
! INSTALL(TARGETS ${install_libs}
!         RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
!         LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
!         ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
\ Falta o caracter nova linha no final do arquivo
