--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@
 MANDIR=$(PREFIX)/share/man
 MAN3DIR=$(MANDIR)/man3
 
-all: $(LIBRARY) $(BINFILES)
+all: $(LIBRARY)
 
 $(LIBRARY): $(OBJECTS)
 	@echo LINK $@
@@ -91,7 +91,7 @@
 	$(LIBTOOL) --mode=clean rm -f $(LIBRARY) $(BINFILES)
 
 .PHONY: install
-install: install-inc install-lib install-bin
+install: install-inc install-lib
 
 install-inc:
 	install -d $(DESTDIR)$(INCDIR)
