--- a/mk/cmake/ConfigureChecks.cmake
+++ b/mk/cmake/ConfigureChecks.cmake
@@ -5,7 +5,7 @@
 	"
 	#include <iconv.h>
 	// this declaration will fail when there already exists a non const char** version which returns size_t
-	double iconv(iconv_t cd,  char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
+	size_t iconv(iconv_t cd,  char **inbuf, size_t *inbytesleft, char **outbuf, size_t *outbytesleft);
 	int main() { return 0; }
 	"
 	HAVE_ICONV_CONST
