--- src/ConEmu/ConEmu.cpp.orig	2014-11-09 02:31:01.456287600 +0100
+++ src/ConEmu/ConEmu.cpp	2014-11-09 02:29:15.901296900 +0100
@@ -938,7 +937,7 @@
 		}
 	}
 
-	if (*szRequired || *szRecommended)
+	if (*szRequired)
 	{
 		size_t cchMax = _tcslen(szRequired) + _tcslen(szRecommended) + _tcslen(ms_ConEmuExe) + 255;
 		wchar_t* pszMsg = (wchar_t*)calloc(cchMax, sizeof(*pszMsg));
@@ -951,12 +950,6 @@
 				_wcscat_c(pszMsg, cchMax, szRequired);
 				_wcscat_c(pszMsg, cchMax, L"\n\n");
 			}
-			if (*szRecommended)
-			{
-				_wcscat_c(pszMsg, cchMax, L"Recommended files not found!\n");
-				_wcscat_c(pszMsg, cchMax, szRecommended);
-				_wcscat_c(pszMsg, cchMax, L"\n\n");
-			}
 			_wcscat_c(pszMsg, cchMax, L"ConEmu was started from:\n");
 			_wcscat_c(pszMsg, cchMax, ms_ConEmuExe);
 			_wcscat_c(pszMsg, cchMax, L"\n");
