--- webkitgtk-2.2.3/Tools/TestWebKitAPI/Tests/WTF/MediaTime.cpp.orig	2013-12-05 08:22:08.604832230 +0100
+++ webkitgtk-2.2.3/Tools/TestWebKitAPI/Tests/WTF/MediaTime.cpp	2013-12-05 08:23:27.257593978 +0100
@@ -44,6 +44,11 @@
 #endif
 #endif
 
+#if COMPILER(MINGW) && !defined(M_PI)
+// M_PI is only defined in MinGW in non-c++11 builds
+#define M_PI		3.14159265358979323846
+#endif
+
 namespace WTF {
 
 std::ostream& operator<<(std::ostream& out, const MediaTime& val)
