--- a/Source/WebCore/GNUmakefile.list.am.orig	2013-12-05 18:16:06.814020249 +0100
+++ b/Source/WebCore/GNUmakefile.list.am	2013-12-05 19:08:48.105560181 +0100
@@ -6261,22 +6261,25 @@
 	Source/WebCore/platform/graphics/glx/GLContextGLX.h
 endif # END USE_GLX
 else
+if !TARGET_WIN32
 webcore_sources += \
 	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/PluginViewNone.cpp
+endif # END !TARGET_WIN32
 endif # END TARGET_X11
 
+if USE_OPENGL
+platformgtk_sources += \
+        Source/WebCore/platform/graphics/GLContext.cpp \
+        Source/WebCore/platform/graphics/GLContext.h \
+        Source/WebCore/platform/gtk/RedirectedXCompositeWindow.cpp \
+        Source/WebCore/platform/gtk/RedirectedXCompositeWindow.h
+endif # END USE_OPENGL
+
 if TARGET_X11_OR_WAYLAND
 platformgtk_sources += \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.h \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.cpp
-if USE_OPENGL
-platformgtk_sources += \
-	Source/WebCore/platform/graphics/GLContext.cpp \
-	Source/WebCore/platform/graphics/GLContext.h \
-	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.cpp \
-	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.h
-endif # END USE_OPENGL
 if USE_EGL
 webcoregtk_sources += \
 	Source/WebCore/platform/graphics/egl/GLContextEGL.cpp \
