--- webkitgtk-2.4.8/Source/WebCore/GNUmakefile.list.am.orig	2015-03-24 06:14:46.327573500 +0000
+++ webkitgtk-2.4.8/Source/WebCore/GNUmakefile.list.am	2015-03-24 06:17:07.769534300 +0000
@@ -6263,6 +6263,13 @@
 platformgtk_sources += \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.h \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.cpp
+if USE_EGL
+webcoregtk_sources += \
+	Source/WebCore/platform/graphics/egl/GLContextEGL.cpp \
+	Source/WebCore/platform/graphics/egl/GLContextEGL.h
+endif # END USE_EGL
+endif # END TARGET_X11_OR_WAYLAND
+
 if USE_OPENGL
 platformgtk_sources += \
 	Source/WebCore/platform/graphics/GLContext.cpp \
@@ -6270,12 +6277,6 @@
 	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.cpp \
 	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.h
 endif # END USE_OPENGL
-if USE_EGL
-webcoregtk_sources += \
-	Source/WebCore/platform/graphics/egl/GLContextEGL.cpp \
-	Source/WebCore/platform/graphics/egl/GLContextEGL.h
-endif # END USE_EGL
-endif # END TARGET_X11_OR_WAYLAND
 
 if TARGET_WIN32
 webcore_sources += \
