--- webkitgtk-2.2.5/Tools/TestWebKitAPI/GNUmakefile.am.orig	2014-03-05 07:24:20.597608800 +0000
+++ webkitgtk-2.2.5/Tools/TestWebKitAPI/GNUmakefile.am	2014-03-05 07:29:56.239729900 +0000
@@ -114,7 +114,6 @@
 noinst_PROGRAMS += \
 	Programs/TestWebKitAPI/WTF/TestWTF \
 	Programs/TestWebKitAPI/JavaScriptCore/TestJavaScriptCore \
-	Programs/TestWebKitAPI/WebCore/TestWebCore \
 	Programs/TestWebKitAPI/WebCoreGtk/TestWebCoreGtk
 
 if ENABLE_WEBKIT1
