--- webkitgtk-2.2.5/Source/WebCore/platform/gtk/FileSystemGtk.cpp.orig	2014-02-17 11:24:05.000000000 +0000
+++ webkitgtk-2.2.5/Source/WebCore/platform/gtk/FileSystemGtk.cpp	2014-03-05 09:17:20.497546700 +0000
@@ -148,7 +148,7 @@
     if (filename.isNull())
         return false;
 
-    struct stat statResult;
+    GStatBuf statResult;
     gint result = g_stat(filename.data(), &statResult);
     if (result)
         return false;
