--- webkitgtk-2.2.5/Source/WebKit/gtk/GNUmakefile.am.orig	2014-02-17 11:24:13.000000000 +0000
+++ webkitgtk-2.2.5/Source/WebKit/gtk/GNUmakefile.am	2014-03-07 02:38:18.647631200 +0000
@@ -301,8 +301,8 @@
 	     --include=Gtk-@GTK_API_VERSION@ \
 	     --include=JavaScriptCore-@WEBKITGTK_API_VERSION@ \
 	     --include=Soup-2.4 \
-	     --library=webkitgtk-@WEBKITGTK_API_VERSION@ \
-	     --library=javascriptcoregtk-@WEBKITGTK_API_VERSION@ \
+	     --library=libwebkitgtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
+	     --library=libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
 	     --libtool="$(LIBTOOL)" \
 	     --pkg=gobject-2.0 \
 	     --pkg=gtk+-@GTK_API_VERSION@ \
