--- webkitgtk-2.2.5/Source/autotools/SetupCompilerFlags.m4.orig	2014-02-17 11:24:13.000000000 +0000
+++ webkitgtk-2.2.5/Source/autotools/SetupCompilerFlags.m4	2014-03-07 12:21:56.378818300 +0000
@@ -32,8 +32,8 @@
     CXXFLAGS="$CXXFLAGS -g1"
     CFLAGS="$CFLAGS -g1"
 elif test "$enable_debug_symbols" != "no"; then
-    CXXFLAGS="$CXXFLAGS -g"
-    CFLAGS="$CFLAGS -g"
+    CXXFLAGS="$CXXFLAGS -gstabs"
+    CFLAGS="$CFLAGS -gstabs"
 fi
 
 # Add the appropriate 'O' level for optimized builds.
