--- webkitgtk-2.2.5/Tools/TestWebKitAPI/Tests/WTF/IntegerToStringConversion.cpp.orig	2014-03-08 18:03:07.528800000 +0400
+++ webkitgtk-2.2.5/Tools/TestWebKitAPI/Tests/WTF/IntegerToStringConversion.cpp	2014-03-08 18:05:39.769200000 +0400
@@ -42,7 +42,7 @@
 const char PrintfFormatTrait<long>::format[] = "%ld";
 
 template<> struct PrintfFormatTrait<long long> { static const char format[]; };
-#if OS(WINDOWS) && !PLATFORM(QT)
+#if OS(WINDOWS) && !PLATFORM(QT) && !defined(__USE_MINGW_ANSI_STDIO)
 const char PrintfFormatTrait<long long>::format[] = "%I64i";
 #else
 const char PrintfFormatTrait<long long>::format[] = "%lli";
@@ -58,7 +58,7 @@
 const char PrintfFormatTrait<unsigned long>::format[] = "%lu";
 
 template<> struct PrintfFormatTrait<unsigned long long> { static const char format[]; };
-#if OS(WINDOWS) && !PLATFORM(QT)
+#if OS(WINDOWS) && !PLATFORM(QT) && !defined(__USE_MINGW_ANSI_STDIO)
 const char PrintfFormatTrait<unsigned long long>::format[] = "%I64u";
 #else
 const char PrintfFormatTrait<unsigned long long>::format[] = "%llu";
--- webkitgtk-2.2.5/Source/WebCore/loader/FTPDirectoryParser.cpp.orig	2014-03-08 18:53:11.563600000 +0400
+++ webkitgtk-2.2.5/Source/WebCore/loader/FTPDirectoryParser.cpp	2014-03-08 18:55:30.965200000 +0400
@@ -188,7 +188,7 @@
               if (pos < linelen && line[pos] == ',')
               {
                 unsigned long long seconds = 0;
-#if OS(WINDOWS)
+#if OS(WINDOWS) && !defined(__USE_MINGW_ANSI_STDIO)
                 sscanf(p + 1, "%I64u", &seconds);
 #else
                 sscanf(p + 1, "%llu", &seconds);
--- webkitgtk-2.2.5/Source/WebCore/loader/icon/IconDatabase.cpp.orig	2014-03-08 19:04:04.267000000 +0400
+++ webkitgtk-2.2.5/Source/WebCore/loader/icon/IconDatabase.cpp	2014-03-08 19:04:17.386600000 +0400
@@ -1711,7 +1711,7 @@
         SQLiteStatement pageDeleteSQL(m_syncDB, "DELETE FROM PageURL WHERE rowid = (?);");
         pageDeleteSQL.prepare();
         for (size_t i = 0; i < numToDelete; ++i) {
-#if OS(WINDOWS)
+#if OS(WINDOWS) && !defined(__USE_MINGW_ANSI_STDIO)
             LOG(IconDatabase, "Pruning page with rowid %I64i from disk", static_cast<long long>(pageIDsToDelete[i]));
 #else
             LOG(IconDatabase, "Pruning page with rowid %lli from disk", static_cast<long long>(pageIDsToDelete[i]));
--- webkitgtk-2.2.5/Source/WebCore/Modules/webdatabase/DatabaseTracker.cpp.orig	2014-03-08 19:07:17.561400000 +0400
+++ webkitgtk-2.2.5/Source/WebCore/Modules/webdatabase/DatabaseTracker.cpp	2014-03-08 19:07:28.621800000 +0400
@@ -767,7 +767,7 @@
         }
 
         if (error)
-#if OS(WINDOWS)
+#if OS(WINDOWS) && !defined(__USE_MINGW_ANSI_STDIO)
             LOG_ERROR("Failed to set quota %I64u in tracker database for origin %s", quota, origin->databaseIdentifier().ascii().data());
 #else
             LOG_ERROR("Failed to set quota %llu in tracker database for origin %s", quota, origin->databaseIdentifier().ascii().data());
--- webkitgtk-2.2.5/Source/WebCore/platform/sql/SQLiteDatabase.cpp.orig	2014-03-08 19:09:22.623000000 +0400
+++ webkitgtk-2.2.5/Source/WebCore/platform/sql/SQLiteDatabase.cpp	2014-03-08 19:09:26.913000000 +0400
@@ -178,7 +178,7 @@
     SQLiteStatement statement(*this, "PRAGMA max_page_count = " + String::number(newMaxPageCount));
     statement.prepare();
     if (statement.step() != SQLResultRow)
-#if OS(WINDOWS)
+#if OS(WINDOWS) && !defined(__USE_MINGW_ANSI_STDIO)
         LOG_ERROR("Failed to set maximum size of database to %I64i bytes", static_cast<long long>(size));
 #else
         LOG_ERROR("Failed to set maximum size of database to %lli bytes", static_cast<long long>(size));
