--- webkitgtk-2.2.6/Source/WTF/wtf/ThreadingWin.cpp.orig	2014-03-19 07:53:02.000000000 +0000
+++ webkitgtk-2.2.6/Source/WTF/wtf/ThreadingWin.cpp	2014-04-15 10:18:07.793755700 +0000
@@ -212,14 +212,26 @@
 
 static unsigned __stdcall wtfThreadEntryPoint(void* param)
 {
-    OwnPtr<ThreadFunctionInvocation> invocation = adoptPtr(static_cast<ThreadFunctionInvocation*>(param));
-    invocation->function(invocation->data);
+    if (getenv ("WEBKIT_DONTFIX_INVOCATION_DESTRUCTOR"))
+    {
+        OwnPtr<ThreadFunctionInvocation> invocation = adoptPtr(static_cast<ThreadFunctionInvocation*>(param));
+        invocation->function(invocation->data);
 
 #if !USE(PTHREADS) && OS(WINDOWS)
+        // Do the TLS cleanup.
+        ThreadSpecificThreadExit();
+#endif
+        return 0;
+    }
+    else
+    {
+        OwnPtr<ThreadFunctionInvocation> invocation = adoptPtr(static_cast<ThreadFunctionInvocation*>(param));
+        invocation->function(invocation->data);
+    }
+#if !USE(PTHREADS) && OS(WINDOWS)
     // Do the TLS cleanup.
     ThreadSpecificThreadExit();
 #endif
-
     return 0;
 }
 
