--- webkitgtk-2.2.5/Source/WebCore/platform/graphics/GLContext.h.orig	2013-12-11 22:37:44.176510725 +0100
+++ webkitgtk-2.2.5/Source/WebCore/platform/graphics/GLContext.h	2013-12-11 22:52:27.280475698 +0100
@@ -25,8 +25,8 @@
 #include <wtf/Noncopyable.h>
 #include <wtf/PassOwnPtr.h>
 
-#if USE(EGL) && !PLATFORM(GTK)
-#include "eglplatform.h"
+#if USE(EGL)
+#include <EGL/eglplatform.h>
 typedef EGLNativeWindowType GLNativeWindowType;
 #else
 typedef uint64_t GLNativeWindowType;
