--- webkitgtk-2.2.5/GNUmakefile.in.orig	2014-03-28 14:01:52.271600000 +0400
+++ webkitgtk-2.2.5/GNUmakefile.in	2014-03-28 14:06:34.709600000 +0400
@@ -78633,7 +78633,9 @@
 
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@WebKit2-@WEBKITGTK_API_VERSION@.gir: $(G_IR_SCANNER) libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	$(AM_V_GEN) \
+@ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     CC=$(CC) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     CFLAGS="$(CFLAGS) -Wno-deprecated-declarations" \
+@ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     $(PYTHON) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     $(G_IR_SCANNER) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     --quiet \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     --warn-all \
@@ -78656,6 +78658,8 @@
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     --add-include-path=$(WebKit2) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     --add-include-path=$(top_builddir) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     --c-include="webkit2/webkit2.h" \
+@ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     $(GTK_CFLAGS) \
+@ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     $(LIBSOUP_CFLAGS) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     -I$(srcdir)/Source \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     -I$(WebKit2) \
 @ENABLE_INTROSPECTION_TRUE@@ENABLE_WEBKIT2_TRUE@	     -I$(GENSOURCES) \
--- webkitgtk-2.2.5/Source/WebKit/gtk/GNUmakefile.am.orig	2014-03-28 14:02:42.675200000 +0400
+++ webkitgtk-2.2.5/Source/WebKit/gtk/GNUmakefile.am	2014-03-28 14:05:16.538000000 +0400
@@ -289,7 +289,9 @@
 if ENABLE_INTROSPECTION
 WebKit-@WEBKITGTK_API_VERSION@.gir: $(G_IR_SCANNER) JavaScriptCore-@WEBKITGTK_API_VERSION@.gir libwebkitgtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la
 	$(AM_V_GEN) \
+	     CC=$(CC) \
 	     CFLAGS="$(CFLAGS) -Wno-deprecated-declarations" \
+	     $(PYTHON) \
 	     $(G_IR_SCANNER) \
 	     --quiet \
 	     --warn-all \
@@ -313,6 +315,8 @@
 	     --add-include-path=$(top_builddir) \
 	     --c-include="webkit/webkit.h" \
 	     -DBUILDING_WEBKIT \
+	     $(GTK_CFLAGS) \
+	     $(LIBSOUP_CFLAGS) \
 	     -I$(srcdir)/Source \
 	     -I$(WebKit) \
 	     -I$(GENSOURCES) \
--- webkitgtk-2.2.5/Source/WebKit2/GNUmakefile.am.orig	2014-03-28 14:03:01.629200000 +0400
+++ webkitgtk-2.2.5/Source/WebKit2/GNUmakefile.am	2014-03-28 14:05:00.111200000 +0400
@@ -281,7 +281,9 @@
 
 WebKit2-@WEBKITGTK_API_VERSION@.gir: $(G_IR_SCANNER) libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la
 	$(AM_V_GEN) \
+	     CC=$(CC) \
 	     CFLAGS="$(CFLAGS) -Wno-deprecated-declarations" \
+	     $(PYTHON) \
 	     $(G_IR_SCANNER) \
 	     --quiet \
 	     --warn-all \
@@ -304,6 +306,8 @@
 	     --add-include-path=$(WebKit2) \
 	     --add-include-path=$(top_builddir) \
 	     --c-include="webkit2/webkit2.h" \
+	     $(GTK_CFLAGS) \
+	     $(LIBSOUP_CFLAGS) \
 	     -I$(srcdir)/Source \
 	     -I$(WebKit2) \
 	     -I$(GENSOURCES) \
