--- webkitgtk-2.2.5/Source/WebKit2/GNUmakefile.am.orig	2014-03-28 14:03:01.629200000 +0400
+++ webkitgtk-2.2.5/Source/WebKit2/GNUmakefile.am	2014-03-28 14:05:00.111200000 +0400
@@ -304,6 +306,8 @@
 	     --add-include-path=$(WebKit2) \
 	     --add-include-path=$(top_builddir) \
 	     --c-include="webkit2/webkit2.h" \
+	     $(GTK_CFLAGS) \
+	     $(LIBSOUP_CFLAGS) \
 	     -I$(srcdir)/Source \
 	     -I$(WebKit2) \
 	     -I$(GENSOURCES) \
