--- webkitgtk-2.4.1.old/Source/autotools/FindDependencies.m4	2014-05-16 19:07:53 +0000
+++ webkitgtk-2.4.1/Source/autotools/FindDependencies.m4	2014-05-17 07:38:01 +0000
@@ -109,8 +109,14 @@ case "$host" in
         UNICODE_LIBS="-licucore"
         ;;
     *-*-mingw*)
-        UNICODE_CFLAGS=""
-        UNICODE_LIBS="-licui18n -licuuc"
+	PKG_CHECK_MODULES(ICU, icu-i18n, ,)
+	if test "x$ICU_LIBS" = "x" ; then
+        	UNICODE_CFLAGS=""
+        	UNICODE_LIBS="-licui18n -licuuc"
+	else
+        	UNICODE_CFLAGS="$ICU_CFLAGS"
+        	UNICODE_LIBS="$ICU_LIBS"
+	fi
         AC_CHECK_HEADERS([unicode/uchar.h], [], [AC_MSG_ERROR([Could not find ICU headers.])])
         ;;
     *)
@@ -455,7 +461,14 @@ if test "$found_opengl" = "yes"; then
         OPENGL_LIBS="-lGLESv2"
     else
         acceleration_description="$acceleration_description (gl"
-        OPENGL_LIBS="-lGL"
+        case "$host" in
+            *-*-mingw*)
+                OPENGL_LIBS="-lopengl32"
+                ;;
+            *)
+                OPENGL_LIBS="-lGL"
+                ;;
+        esac
     fi
     if test "$enable_egl" = "yes"; then
         acceleration_description="$acceleration_description, egl"
