https://github.com/jphilbert/shim_executable/pull/3
diff --git a/shim.cpp b/shim.cpp
index 2515d5b..6aa0e0d 100644
--- a/shim.cpp
+++ b/shim.cpp
@@ -184,8 +184,9 @@ All other argument are passed to the parent executable.
                         (alias --shimgen-gui)
 
     --shim-UseTargetWorkingDirectory
-                    Set the working directory to the target path. By default
-                        the application will use the shim's path. Useful when
+                    Set the working directory to the target path. By default 
+                        the application will use the shim's path for GUI and 
+                        shell current path for console. This option is useful when
                         programs need to be running from where they are located
                         (usually indicates programs that have issues being run
                         globally).
@@ -337,7 +338,10 @@ int ShimMain() {
     appArgs += L" ";
   appArgs += calling_args;
 
-  wstring working_dir = shimArgUseTarget ? appDir : shimDir;
+  // for console application, set wd to where the shim was called from
+  wstring fallback_dir = shimType == L"CONSOLE" ? currDir : shimDir;
+
+  wstring working_dir = shimArgUseTarget ? appDir : fallback_dir;
   
   
   
