--- src/shim.cpp.orig	2026-04-14 21:58:05.831925700 -0500
+++ src/shim.cpp	2026-04-14 23:17:03.820642200 -0500
@@ -278,7 +278,7 @@
   wstring appArgs   = L"";
   wstring shimType  = L"";
   
-  if (!GetResourceData("SHIM_PATH", appPath)) {
+  if (!GetResourceData(L"SHIM_PATH", appPath)) {
     LOG(1)  << "Shim has no application path. ";
     LOG(-1) << "Shim is no longer valid and must be regenerated.";
     return exitCode;
@@ -297,8 +297,8 @@
     exitCode = 0;
 
   appDir = filesystem::path(appPath).parent_path().c_str();
-  GetResourceData("SHIM_ARGS", appArgs);
-  GetResourceData("SHIM_TYPE", shimType);
+  GetResourceData(L"SHIM_ARGS", appArgs);
+  GetResourceData(L"SHIM_TYPE", shimType);
 
   // Here forward we'll just use shimArgWait
   if (shimType == L"CONSOLE")
--- src/include/resource_functions.h.orig	2026-04-14 17:29:19.830815300 -0500
+++ src/include/resource_functions.h	2026-04-14 23:13:18.526186500 -0500
@@ -31,13 +31,13 @@
 
 
 // ---------------------------- Read Resources ----------------------------- // 
-bool HasResourceData(LPCSTR name) {
+bool HasResourceData(LPCTSTR name) {
   if (FindResource(NULL, name, RT_RCDATA)) 
     return true;
   return false;
 }
 
-bool GetResourceData(LPCSTR name, wstring& arg) {
+bool GetResourceData(LPCTSTR name, wstring& arg) {
   // Get the resource handle if it exists 
   HRSRC     resource    = FindResource(NULL, name, RT_RCDATA);
   if (!resource) return false;
@@ -52,7 +52,7 @@
   return true;
 }
 
-bool GetResourceFile(string name, const filesystem::path& path) {
+bool GetResourceFile(wstring name, const filesystem::path& path) {
   // Get the resource handle if it exists 
   HRSRC     resource    = FindResource(NULL, name.c_str(), RT_RCDATA);
   if (!resource) return false;
@@ -79,7 +79,7 @@
 
 
 // ----------------------------- Add Resources ----------------------------- // 
-BOOL AddResourceData (filesystem::path target, LPCSTR name, wstring arg) {
+BOOL AddResourceData (filesystem::path target, LPCTSTR name, wstring arg) {
   HANDLE resource   = BeginUpdateResourceW(target.c_str(), FALSE);
   BOOL bUpdate      = UpdateResource(
       resource,                 // Handle
@@ -104,7 +104,7 @@
 HANDLE resource_handle;
 
 BOOL CALLBACK enumLangsFunc(HMODULE hModule, LPCTSTR lpType, LPCTSTR lpName,
-                   WORD wLang, LONG lParam) {
+                   WORD wLang, LONG_PTR lParam) {
     
   HRSRC hRes =          FindResourceEx(hModule, lpType, lpName, wLang);
   HGLOBAL hResLoad =    LoadResource(hModule, hRes);
@@ -114,20 +114,20 @@
                  lpResLock,                        // ptr to resource info
                  SizeofResource(hModule, hRes));   // size of resource info
 
-  string log_str = "Copied ";
+  wstring log_str = L"Copied ";
   
   if (lpType == RT_ICON)
-    log_str += "ICON ";
+    log_str += L"ICON ";
   else if (lpType == RT_VERSION)
-    log_str += "VERSION ";
+    log_str += L"VERSION ";
   else if (lpType == RT_GROUP_ICON)
-    log_str += "ICON GROUP ";
+    log_str += L"ICON GROUP ";
 
-  log_str += "resource ";
+  log_str += L"resource ";
   if (!IS_INTRESOURCE(lpName))
     log_str += lpName;
   else
-    log_str += to_string((USHORT)lpName);
+    log_str += to_wstring((uintptr_t)lpName);
 
   LOG(3) << log_str;
 
@@ -135,12 +135,12 @@
 }
 
 BOOL CALLBACK enumNamesFunc(HMODULE hModule, LPCTSTR lpType, LPTSTR lpName,
-                   LONG lParam) {
+                   LONG_PTR lParam) {
   EnumResourceLanguages(hModule, lpType, lpName, enumLangsFunc, 0);
   return TRUE;
 }
 
-BOOL CALLBACK enumTypesFunc(HMODULE hModule, LPTSTR lpType, LONG lParam) {
+BOOL CALLBACK enumTypesFunc(HMODULE hModule, LPTSTR lpType, LONG_PTR lParam) {
   // Only Copy Icons and Version Info
   if(lpType == RT_ICON || lpType == RT_VERSION || lpType == RT_GROUP_ICON)
     EnumResourceNames(hModule, lpType, enumNamesFunc, 0);
--- src/include/utility_functions.h.orig	2026-04-14 17:29:19.833812000 -0500
+++ src/include/utility_functions.h	2026-04-14 23:14:24.821586600 -0500
@@ -120,7 +120,7 @@
 
 
 filesystem::path GetExecPath() {
-  CHAR cExePath[MAX_PATH];  
+  TCHAR cExePath[MAX_PATH];  
   GetModuleFileName(NULL, cExePath, MAX_PATH);
   return filesystem::path(cExePath);
 }  
--- src/shim_executable.cpp.orig	2026-04-14 23:43:28.322259900 -0500
+++ src/shim_executable.cpp	2026-04-14 23:25:09.887076800 -0500
@@ -33,7 +33,7 @@
 
 // ----------------- Unpack the Shim from this Application ----------------- // 
 BOOL UnpackShim(const filesystem::path& path, wstring shim_type) {
-  return GetResourceFile("SHIM_" + NarrowString(shim_type), path);
+  return GetResourceFile(L"SHIM_" + shim_type, path);
 }
 
 
@@ -535,10 +535,10 @@
   CopyResources(output_path, input_path);    
 
   // Add Shim Arguments
-  AddResourceData(output_path, "SHIM_PATH", input_path);
-  AddResourceData(output_path, "SHIM_TYPE", shim_type);  
+  AddResourceData(output_path, L"SHIM_PATH", input_path);
+  AddResourceData(output_path, L"SHIM_TYPE", shim_type);  
   if (!command_args.empty()) 
-    AddResourceData(output_path, "SHIM_ARGS", command_args);
+    AddResourceData(output_path, L"SHIM_ARGS", command_args);
 
 
   // -------------------------------- Done --------------------------------- // 
