--- openvr-1.0.15/src/CMakeLists.txt.orig	2018-05-15 14:56:17.412227200 +0300
+++ openvr-1.0.15/src/CMakeLists.txt	2018-05-15 14:57:40.076772400 +0300
@@ -15,9 +15,12 @@
     set(EXTRA_LIBS ${EXTRA_LIBS} ${FOUNDATION_FRAMEWORK})
   endif(BUILD_SHARED OR BUILD_FRAMEWORK)
 elseif(WIN32)
+  add_definitions( -DWIN32 )
   if(CMAKE_SIZEOF_VOID_P EQUAL 8)
     add_definitions( -DWIN64 )
-    set( LIBNAME "openvr_api64" )
+    if (MSVC)
+      set( LIBNAME "openvr_api64" )
+    endif()
   endif()
 endif()
 
