From 328b9a1f20187d3bc9c5301e8e4ac873f5db9da4 Mon Sep 17 00:00:00 2001
From: Szymon Piotr Krasuski <krasuski.szymon.piotr@gmail.com>
Date: Mon, 7 Dec 2020 20:22:22 +0100
Subject: [PATCH] Remove Polish letters from setup.py

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index a7fb925..074f2bd 100644
--- a/setup.py
+++ b/setup.py
@@ -9,7 +9,7 @@
 from setuptools.command.install import install
 from setuptools.command.test import test
 
-__author__ = 'Paweł Zadrożny'
+__author__ = 'Pawel Zadrozny'
 __copyright__ = 'Copyright (c) 2018, Pawelzny'
 
 with open('README.rst', 'r') as readme_file:
@@ -49,7 +49,7 @@ def run(self):
     description="Dictionary wrapper for quick access to deeply nested keys.",
     long_description=readme,
     license="MIT license",
-    author="Paweł Zadrożny @pawelzny",
+    author="Pawel Zadrozny @pawelzny",
     author_email='pawel.zny@gmail.com',
     url='https://github.com/pawelzny/dotty_dict',
     packages=find_packages(exclude=('tests', 'docs', 'bin', 'example')),
