--- a/setup.py
+++ b/setup.py
@@ -12,14 +12,14 @@
 __author__ = 'Paweł Zadrożny'
 __copyright__ = 'Copyright (c) 2018, Pawelzny'
 
-with open('README.rst', 'r') as readme_file:
+with open('README.rst', 'r', encoding='utf-8') as readme_file:
     readme = readme_file.read()
 
 
 def get_version(*file_paths):
     """Retrieves the version from project/__init__.py"""
     filename = os.path.join(os.path.dirname(__file__), *file_paths)
-    version_file = open(filename).read()
+    version_file = open(filename, encoding='utf-8').read()
     version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]",
                               version_file, re.M)
     if version_match:
