--- a/portable.h
+++ b/portable.h
@@ -146,7 +146,7 @@
 #endif
 
 /* 64 bit IO */
-#ifdef __WIN32__
+#if defined(__WIN32__) && !defined(_UCRT)
 #define off_t off64_t /* This must be after including stdio.h */
 off64_t rpl_ftello(FILE* f);
 int rpl_fseeko(FILE* f, off64_t offset, int origin);
--- a/portable.c
+++ b/portable.c
@@ -30,7 +30,7 @@
 
 #include "portable.h"
 
-#ifdef __WIN32__
+#if defined(__WIN32__) && !defined(_UCRT)
 off64_t rpl_ftello(FILE* f)
 {
 	fpos_t fpos;
