--- oqs-provider-0.6.1/test/oqs_test_tlssig.c.orig	2024-06-14 11:48:31.000000000 +0200
+++ oqs-provider-0.6.1/test/oqs_test_tlssig.c	2024-08-15 11:34:20.039749900 +0200
@@ -13,6 +13,10 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
+#ifdef _WIN32
+#include <direct.h>
+#endif
+
 static OSSL_LIB_CTX *libctx = NULL;
 static char *modulename = NULL;
 static char *configfile = NULL;
@@ -40,7 +44,11 @@
     sprintf(certpath, "%s%s%s%s", certsdir, sep, sig_name, "_srv.crt");
     sprintf(privkeypath, "%s%s%s%s", certsdir, sep, sig_name, "_srv.key");
     /* ensure certsdir exists */
+#ifdef _WIN32
+    if (_mkdir(certsdir) == -1) {
+#else
     if (mkdir(certsdir, 0700)) {
+#endif
         if (errno != EEXIST) {
             fprintf(stderr, "Couldn't create certsdir %s: Err = %d\n", certsdir,
                     errno);
