diff -Naur libyaml-0.2.1.orig/include/yaml.h libyaml-0.2.1/include/yaml.h
--- libyaml-0.2.1.orig/include/yaml.h	2018-08-11 22:56:41.072390100 -0400
+++ libyaml-0.2.1/include/yaml.h	2018-08-11 23:21:03.942460800 -0400
@@ -28,7 +28,7 @@
 
 #if defined(__MINGW32__)
 #   define  YAML_DECLARE(type)  type
-#elif defined(WIN32)
+#elif defined(_MSC_VER)
 #   if defined(YAML_DECLARE_STATIC)
 #       define  YAML_DECLARE(type)  type
 #   elif defined(YAML_DECLARE_EXPORT)
