diff -Naur c-blosc-1.20.1-orig/blosc/blosc-export.h c-blosc-1.20.1/blosc/blosc-export.h
--- c-blosc-1.20.1-orig/blosc/blosc-export.h	2020-09-08 18:23:32.000000000 +0300
+++ c-blosc-1.20.1/blosc/blosc-export.h	2020-10-22 19:53:24.417616400 +0300
@@ -30,7 +30,7 @@
   #define BLOSC_EXPORT
 #endif  /* defined(BLOSC_SHARED_LIBRARY) */
 
-#if defined(__GNUC__) || defined(__clang__)
+#if !defined(__MINGW32__) && (defined(__GNUC__) || defined(__clang__))
   #define BLOSC_NO_EXPORT __attribute__((visibility("hidden")))
 #else
   #define BLOSC_NO_EXPORT
diff -Naur c-blosc-1.20.1-orig/blosc/shuffle.c c-blosc-1.20.1/blosc/shuffle.c
--- c-blosc-1.20.1-orig/blosc/shuffle.c	2020-09-08 18:23:32.000000000 +0300
+++ c-blosc-1.20.1/blosc/shuffle.c	2020-10-22 19:54:48.744360700 +0300
@@ -14,7 +14,7 @@
 #include "blosc-comp-features.h"
 #include <stdio.h>
 
-#if defined(_WIN32)
+#if !defined(__MINGW32__)
 #include "win32/pthread.h"
 #else
 #include <pthread.h>
