--- Pillow-2.5.1/mp_compile.py.orig	2014-07-10 17:52:44.000000000 +0400
+++ Pillow-2.5.1/mp_compile.py	2014-07-12 01:13:13.373200000 +0400
@@ -1,7 +1,7 @@
 # A monkey patch of the base distutils.ccompiler to use parallel builds
 # Tested on 2.7, looks to be identical to 3.3.
 
-from multiprocessing import Pool, cpu_count
+from multiprocessing import Pool, cpu_count, freeze_support
 from distutils.ccompiler import CCompiler
 import os
 
@@ -54,8 +54,10 @@
     try:
         # bug, only enable if we can make a Pool. see issue #790 and
         # http://stackoverflow.com/questions/6033599/oserror-38-errno-38-with-multiprocessing
-        pool = Pool(2)
-        CCompiler.compile = _mp_compile
+        if __name__ == '__main__':
+            freeze_support()
+            pool = Pool(2)
+            CCompiler.compile = _mp_compile
     except Exception as msg:
         print("Exception installing mp_compile, proceeding without: %s" %msg)
 else:
