--- a/meson.build
+++ b/meson.build
@@ -180,9 +180,9 @@
 # Install Configuration
 #===============================================================================
 if get_option('portable') or host_machine.system() == 'windows'
-    lite_bindir = '/'
-    lite_docdir = '/doc'
-    lite_datadir = '/data'
+    lite_bindir = 'bin'
+    lite_docdir = 'share/doc/lite-xl'
+    lite_datadir = 'share/lite-xl'
     configure_file(
         input: 'resources' / 'windows' / 'lite-xl.exe.manifest.in',
         output: 'lite-xl.exe.manifest',
