diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index dbad37f6..dc8395c6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -194,7 +194,10 @@ install(DIRECTORY ../include/filesystem/ DESTINATION "${INCLUDE_INSTALL_DIR}" FI
 install(DIRECTORY ../include/global/ DESTINATION "${INCLUDE_INSTALL_DIR}" FILES_MATCHING PATTERN "*.h")
 
 install(TARGETS ${LIB_NAME} EXPORT ${DtkCore}Targets
- DESTINATION ${LIBRARY_INSTALL_DIR})
+    LIBRARY DESTINATION ${LIBRARY_INSTALL_DIR}
+    ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
 
 install(EXPORT ${DtkCore}Targets
   FILE ${DtkCore}Targets.cmake
