--- sqlite-src-3490100/main.mk.orig	2025-02-18 15:09:33.000000000 +0100
+++ sqlite-src-3490100/main.mk	2025-05-04 09:57:47.680155900 +0200
@@ -2044,7 +2044,7 @@
 # How to build sqldiff$(T.exe) depends on $(LINK_TOOLS_DYNAMICALLY)
 #
 sqldiff.0.deps = $(TOP)/tool/sqldiff.c $(TOP)/ext/misc/sqlite3_stdio.h sqlite3.o sqlite3.h
-sqldiff.0.rules = $(T.link) -o $@ $(TOP)/tool/sqldiff.c sqlite3.o $(LDFLAGS.libsqlite3)
+sqldiff.0.rules = $(T.link) -o $@ $(TOP)/tool/sqldiff.c $(TOP)/ext/misc/sqlite3_stdio.c sqlite3.o $(LDFLAGS.libsqlite3)
 sqldiff.1.deps = $(TOP)/tool/sqldiff.c $(TOP)/ext/misc/sqlite3_stdio.h $(libsqlite3.SO)
 sqldiff.1.rules = $(T.link) -o $@ $(TOP)/tool/sqldiff.c -L. -lsqlite3 $(LDFLAGS.configure)
 sqldiff$(T.exe): $(sqldiff.$(LINK_TOOLS_DYNAMICALLY).deps)
