Index: src/shell.c.in
==================================================================
--- a/src/shell.c.in	
+++ b/src/shell.c.in	
@@ -12774,11 +12774,11 @@
 
 /* Use the wmain() entry point on Windows.  Translate arguments to
 ** UTF8, then invoke the traditional main() entry point which is
 ** renamed using a #define to utf8_main() .
 */
-#if defined(_WIN32) && !defined(main)
+#if defined(_WIN32) && !defined(__MINGW32__) && !defined(main)
 #  define main utf8_main                 /* Rename entry point to utf_main() */
 int SQLITE_CDECL utf8_main(int,char**);  /* Forward declaration */  
 int SQLITE_CDECL wmain(int argc, wchar_t **wargv){
   int rc, i;
   char **argv = malloc( sizeof(char*) * (argc+1) );
