--- sqlite-src-3490100/main.mk.orig	2025-02-18 15:09:33.000000000 +0100
+++ sqlite-src-3490100/main.mk	2025-05-02 18:07:01.901993900 +0200
@@ -1053,7 +1053,7 @@
 	touch .target_source
 
 libsqlite3.LIB = libsqlite3$(T.lib)
-libsqlite3.SO = libsqlite3$(T.dll)
+libsqlite3.SO = libsqlite3-0$(T.dll)
 
 # Rules to build the LEMON compiler generator
 #
@@ -1467,10 +1467,10 @@
 # In either case, libsqlite3.la, if found, is deleted because it would
 # contain stale state, refering to non-libtool-generated libraries.
 #
-install-so-1: $(install-dir.lib) $(libsqlite3.SO)
-	$(INSTALL) $(libsqlite3.SO) "$(install-dir.lib)"
-	@if [ -f $(libsqlite3.SO).a ]; then \
-		$(INSTALL) $(libsqlite3.SO).a "$(install-dir.lib)"; \
+install-so-1: $(install-dir.lib) $(install-dir.bin) $(libsqlite3.SO)
+	$(INSTALL) $(libsqlite3.SO) "$(install-dir.bin)"
+	@if [ -f libsqlite3.dll.a ]; then \
+		$(INSTALL) libsqlite3.dll.a "$(install-dir.lib)"; \
 	fi
 	@echo "Setting up $(libsqlite3.SO) version symlinks..."; \
 	if [ x.dll = x$(T.dll) ]; then \
