--- ncurses-6.2/misc/ncurses-config.in.orig	2020-02-03 00:34:34.000000000 +0100
+++ ncurses-6.2/misc/ncurses-config.in	2020-09-24 07:39:24.972343500 +0200
@@ -36,7 +36,7 @@
 LC_ALL=C;	export LC_ALL
 LC_CTYPE=C;	export LC_CTYPE
 
-prefix="@prefix@"
+prefix="$(cygpath -m "@prefix@")"
 exec_prefix="@exec_prefix@"
 
 bindir="@bindir@"
@@ -44,7 +44,7 @@
 libdir="@libdir@"
 datarootdir="@datarootdir@"
 datadir="@datadir@"
-mandir="@mandir@"
+mandir="$(cygpath -m "@mandir@")"
 
 THIS="@LIB_NAME@@USE_LIB_SUFFIX@"
 TINFO_LIB="@TINFO_ARG_SUFFIX@"
@@ -116,6 +116,9 @@
 		@LD_SEARCHPATH@) # skip standard libdir
 			continue
 			;;
+		$libdir) # skip standard libdir
+			continue
+			;;
 		*)
 			found=no
 			for check in $lib_flags
@@ -241,13 +244,13 @@
 		echo "${mandir}"
 		;;
 	--terminfo)
-		echo "@TERMINFO@"
+		echo "$(cygpath -m "@TERMINFO@")"
 		;;
 	--terminfo-dirs)
-		echo "@TERMINFO_DIRS@"
+		echo "$(cygpath -m "@TERMINFO_DIRS@")"
 		;;
 	--termpath)
-		echo "@TERMPATH@"
+		echo "$(cygpath -m "@TERMPATH@")"
 		;;
 	# general info
 	--help)
