--- ncurses-6.3/include/ncurses_dll.h.in.orig	2021-12-11 13:47:24.101612300 +0100
+++ ncurses-6.3/include/ncurses_dll.h.in	2021-12-11 13:47:24.572486300 +0100
@@ -78,6 +78,11 @@
 #define NCURSES_EXPORT(type) NCURSES_IMPEXP type NCURSES_API
 #define NCURSES_EXPORT_VAR(type) NCURSES_IMPEXP type
 
+/* XXX: MSYS2 hack to keep static builds working by default
+ * https://github.com/msys2/MINGW-packages/issues/10312
+ */
+#define NCURSES_STATIC
+
 /*
  * These symbols hide dllimport/dllexport, for compilers which care about it.
  */
