diff -bur librt-0.7.5-orig/setup.py librt-0.7.5/setup.py
--- librt-0.7.5-orig/setup.py	2025-12-28 15:26:38.583904200 -0700
+++ librt-0.7.5/setup.py	2025-12-28 15:27:15.247808700 -0700
@@ -130,7 +130,7 @@
                     "base64/arch/neon64/codec.c",
                 ],
                 include_dirs=[".", "base64"],
-                extra_compile_args=cflags,
+                extra_compile_args=cflags + ["-mavx2"],
             ),
         ]
     )
