--- OpenShadingLanguage-Release-1.11.8.0/src/cmake/modules/FindLLVM.cmake.orig	2020-10-08 11:14:41.035584100 +0300
+++ OpenShadingLanguage-Release-1.11.8.0/src/cmake/modules/FindLLVM.cmake	2020-10-08 11:15:12.426896100 +0300
@@ -142,6 +142,9 @@
   endif ()
 endif ()
 
+if (MINGW)
+  set (LLVM_LIBRARIES "${LLVM_LIBRARIES};-lz;-lversion")
+endif()
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args (LLVM
