From adeda71f4e7893de035563a68f518370844cfe92 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 2 Jun 2016 15:17:59 +0100
Subject: [PATCH 1/2] Less of the silly nonsense

---
 Makefile | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Makefile b/Makefile
index e3ebbee..9c15d4d 100644
--- a/Makefile
+++ b/Makefile
@@ -4,10 +4,6 @@ all: c99conv$(EXT) c99wrap$(EXT)
 
 OBJS = convert.o
 
-CC=/opt/local/bin/clang-mp-3.2
-LD=$(CC)
-CFLAGS=-I/opt/local/libexec/llvm-3.2/include -g
-LDFLAGS=-L/opt/local/libexec/llvm-3.2/lib -g
 LIBS=-lclang
 
 clean:
-- 
2.8.2

