--- libgdata-0.17.11/meson.build.orig	2019-08-21 12:05:55.000000000 +0200
+++ libgdata-0.17.11/meson.build	2020-02-24 09:20:39.410688800 +0100
@@ -78,7 +78,7 @@
   dependency('gobject-2.0'),
   dependency('glib-2.0', version: '>= 2.44.0'),
   dependency('gio-2.0', version: '>= 2.44.0'),
-  dependency('gio-unix-2.0'),
+  #dependency('gio-unix-2.0'),
   dependency('json-glib-1.0', version: '>= 0.15'),
   dependency('libxml-2.0'),
 ]
