--- gtkmm-3.24.3/untracked/gtk/gtkmm/treepath.h.orig	2020-12-13 18:49:01.923754200 +0100
+++ gtkmm-3.24.3/untracked/gtk/gtkmm/treepath.h	2021-02-12 12:01:47.398551600 +0100
@@ -360,7 +360,7 @@
  * @param rhs The right-hand side
  * @result The result
  */
-
+GTKMM_API
 bool operator==(const TreePath& lhs, const TreePath& rhs);
 
 /** @relates Gtk::TreePath
@@ -368,7 +368,7 @@
  * @param rhs The right-hand side
  * @result The result
  */
-
+GTKMM_API
 bool operator!=(const TreePath& lhs, const TreePath& rhs);
 
 /** @relates Gtk::TreePath
@@ -376,7 +376,7 @@
  * @param rhs The right-hand side
  * @result The result
  */
-
+GTKMM_API
 bool operator<(const TreePath& lhs, const TreePath& rhs);
 
 /** @relates Gtk::TreePath
@@ -384,7 +384,7 @@
  * @param rhs The right-hand side
  * @result The result
  */
-
+GTKMM_API
 bool operator>(const TreePath& lhs, const TreePath& rhs);
 
 /** @relates Gtk::TreePath
@@ -392,7 +392,7 @@
  * @param rhs The right-hand side
  * @result The result
  */
-
+GTKMM_API
 bool operator<=(const TreePath& lhs, const TreePath& rhs);
 
 /** @relates Gtk::TreePath
@@ -400,7 +400,7 @@
  * @param rhs The right-hand side
  * @result The result
  */
-
+GTKMM_API
 bool operator>=(const TreePath& lhs, const TreePath& rhs);
 
 
