--- guile-2.0.9/libguile/stime.c.orig	2013-05-03 02:11:33 +0400
+++ guile-2.0.9/libguile/stime.c	2013-05-03 03:01:29 +0400
@@ -127,7 +127,7 @@
      ts.tv_nsec - posix_real_time_base.tv_nsec);
 }
 
-#if defined _POSIX_CPUTIME && defined CLOCK_PROCESS_CPUTIME_ID
+#if defined _POSIX_CPUTIME && defined CLOCK_PROCESS_CPUTIME_ID && !defined(WINPTHREAD_API)
 /* You see, FreeBSD defines _POSIX_CPUTIME but not
    CLOCK_PROCESS_CPUTIME_ID.  */
 #define HAVE_POSIX_CPUTIME 1
