--- a/src/spice-widget-egl.c
+++ b/src/spice-widget-egl.c
@@ -67,16 +66,16 @@                         \
 ";
 
 static void apply_ortho(guint mproj, float left, float right,
-                        float bottom, float top, float near, float far)
+                        float bottom, float top, float near_spice, float far_spice)
 
 {
     float a = 2.0f / (right - left);
     float b = 2.0f / (top - bottom);
-    float c = -2.0f / (far - near);
+    float c = -2.0f / (far_spice - near_spice);
 
     float tx = - (right + left) / (right - left);
     float ty = - (top + bottom) / (top - bottom);
-    float tz = - (far + near) / (far - near);
+    float tz = - (far_spice + near_spice) / (far_spice - near_spice);
 
     float ortho[16] = {
         a, 0, 0, 0,
