diff --git a/src/meson.build b/src/meson.build
index 50e27d9..d3b43e8 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -186,9 +186,8 @@ spice_client_glib_syms_path = meson.current_source_dir() / 'map-file'
 spice_gtk_version_script = '-Wl,--version-script=@0@'.format(spice_client_glib_syms_path)
 spice_gtk_has_version_script = compiler.has_link_argument(spice_gtk_version_script)
 if not spice_gtk_has_version_script
-  spice_client_glib_syms = files('spice-glib-sym-file')
-  spice_client_glib_syms_path = meson.current_source_dir() / 'spice-glib-sym-file'
-  spice_gtk_version_script = ['-export-symbols', spice_client_glib_syms_path]
+  spice_client_glib_syms = []
+  spice_gtk_version_script = []
 endif
 
 # soversion
@@ -207,7 +206,8 @@ spice_client_glib_lib = library('spice-client-glib-2.0', spice_client_glib_sourc
                                 include_directories : spice_gtk_include,
                                 link_args : [spice_gtk_version_script],
                                 link_depends : spice_client_glib_syms,
-                                dependencies : spice_glib_deps)
+                                dependencies : spice_glib_deps,
+                                vs_module_defs : 'spice-glib-sym-file.def')
 
 spice_client_glib_dep = declare_dependency(sources : [spice_marshals[1], spice_client_glib_enums[1]],
                                            link_with : spice_client_glib_lib,
@@ -386,11 +386,6 @@ if spice_gtk_has_gtk
   # libspice-client-gtk.so
   #
   spice_client_gtk_syms = spice_client_glib_syms
-  if not spice_gtk_has_version_script
-    spice_client_gtk_syms = files('spice-gtk-sym-file')
-    spice_client_gtk_syms_path = meson.current_source_dir() / 'spice-gtk-sym-file'
-    spice_gtk_version_script = ['-export-symbols', spice_client_gtk_syms_path]
-  endif
 
   # soversion
   # http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
@@ -407,7 +402,8 @@ if spice_gtk_has_gtk
                                  install : true,
                                  link_args : [spice_gtk_version_script],
                                  link_depends : spice_client_gtk_syms,
-                                 dependencies : [spice_client_glib_dep, spice_gtk_deps, spice_wayland_deps])
+                                 dependencies : [spice_client_glib_dep, spice_gtk_deps, spice_wayland_deps],
+                                 vs_module_defs : 'spice-gtk-sym-file.def')
 
   spice_client_gtk_dep = declare_dependency(sources : spice_widget_enums[1],
                                             link_with : spice_client_gtk_lib,
diff --git a/src/spice-glib-sym-file.def b/src/spice-glib-sym-file.def
index 2cc80aa..baa63bd 100644
--- a/src/spice-glib-sym-file.def
+++ b/src/spice-glib-sym-file.def
@@ -1,3 +1,4 @@
+EXPORTS
 spice_audio_get
 spice_audio_get_type
 spice_audio_new
@@ -151,6 +153,7 @@
 spice_usb_device_manager_can_redirect_device
 spice_usb_device_manager_connect_device_async
 spice_usb_device_manager_connect_device_finish
+spice_usb_device_manager_create_shared_cd_device
 spice_usb_device_manager_disconnect_device
 spice_usb_device_manager_disconnect_device_async
 spice_usb_device_manager_disconnect_device_finish
@@ -159,6 +162,7 @@
 spice_usb_device_manager_get_devices_with_filter
 spice_usb_device_manager_get_type
 spice_usb_device_manager_is_device_connected
+spice_usb_device_manager_is_device_shared_cd
 spice_usb_device_manager_is_redirecting
 spice_usbredir_channel_get_type
 spice_util_get_debug
diff --git a/src/spice-gtk-sym-file.def b/src/spice-gtk-sym-file.def
index 5ba57cb..34ad225 100644
--- a/src/spice-gtk-sym-file.def
+++ b/src/spice-gtk-sym-file.def
@@ -1,3 +1,4 @@
+EXPORTS
 spice_display_get_grab_keys
 spice_display_get_pixbuf
 spice_display_get_type
