--- a/src/usb-backend.c
+++ b/src/usb-backend.c
@@ -214,7 +214,7 @@
     return num_changed;
 }
 
-static LRESULT subclass_proc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam,
+static LRESULT CALLBACK subclass_proc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam,
                              UINT_PTR uIdSubclass, DWORD_PTR dwRefData)
 {
     SpiceUsbBackend *be = (SpiceUsbBackend *)dwRefData;
