--- spice-gtk-0.41/src/meson.build.orig	2022-07-24 19:25:34.615147100 +0200
+++ spice-gtk-0.41/src/meson.build	2022-07-24 19:25:12.502127800 +0200
@@ -312,7 +312,7 @@
   foreach keymap : keymaps
     varname = 'keymap_@0@2xtkbd'.format(keymap)
     target = 'vncdisplay@0@.h'.format(varname)
-    cmd = [python, keymapgen, 'code-map', '--lang', 'glib2', '--varname', varname, keymapcsv, keymap, 'xtkbd']
+    cmd = [python, keymapgen.path(), 'code-map', '--lang', 'glib2', '--varname', varname, keymapcsv, keymap, 'xtkbd']
     spice_client_gtk_sources += custom_target(target,
                                               output : target,
                                               capture : true,
