--- a/src/base/tinyxml/Makefile.am
+++ b/src/base/tinyxml/Makefile.am
@@ -3,6 +3,10 @@
 				-I$(top_srcdir)/src/include \
 				-I$(top_srcdir)/src/include/tinyxml
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -DEXPORT_LIB
+endif
+
 noinst_LTLIBRARIES = libtinyxml.la
 
 libtinyxml_la_SOURCES = tinyxml.cpp tinyxmlerror.cpp tinyxmlparser.cpp
--- a/src/include/manager.h
+++ b/src/include/manager.h
@@ -201,7 +201,7 @@
     cbSearchResultsLog *m_SearchResultLog;
 };
 
-template <class MgrT> class DLLIMPORT Mgr
+template <class MgrT> class Mgr
 {
     static MgrT *instance;
     static bool isShutdown;s
--- a/src/plugins/codecompletion/Makefile.am
+++ b/src/plugins/codecompletion/Makefile.am
@@ -11,8 +11,7 @@
 libcodecompletion_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libcodecompletion_la_LIBADD = ../../sdk/libcodeblocks.la\
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libcodecompletion_la_SOURCES = ccoptionsdlg.cpp \
 			ccoptionsprjdlg.cpp \
--- a/src/plugins/contrib/AutoVersioning/Makefile.am
+++ b/src/plugins/contrib/AutoVersioning/Makefile.am
@@ -9,8 +9,7 @@
 libAutoVersioning_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libAutoVersioning_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libAutoVersioning_la_SOURCES = AutoVersioning.cpp \
             avChangesDlg.cpp \
--- a/src/plugins/contrib/BrowseTracker/Makefile.am
+++ b/src/plugins/contrib/BrowseTracker/Makefile.am
@@ -9,8 +9,7 @@
 libBrowseTracker_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libBrowseTracker_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libBrowseTracker_la_SOURCES = BrowseTracker.cpp \
 			Version.cpp \
--- a/src/plugins/contrib/CppCheck/Makefile.am
+++ b/src/plugins/contrib/CppCheck/Makefile.am
@@ -11,8 +11,7 @@
 libCppCheck_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libCppCheck_la_LIBADD = ../../../sdk/libcodeblocks.la \
-		$(WX_LIBS) \
-		$(CB_TINYXML_LIBS)
+		$(WX_LIBS)
 
 libCppCheck_la_SOURCES = CppCheck.cpp \
             CppCheckListLog.cpp \
--- a/src/plugins/contrib/DoxyBlocks/Makefile.am
+++ b/src/plugins/contrib/DoxyBlocks/Makefile.am
@@ -9,8 +9,7 @@
 libDoxyBlocks_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libDoxyBlocks_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libDoxyBlocks_la_SOURCES = AutoDoc.cpp \
 				Config.cpp \
--- a/src/plugins/contrib/EditorConfig/Makefile.am
+++ b/src/plugins/contrib/EditorConfig/Makefile.am
@@ -9,8 +9,7 @@
 libEditorConfig_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libEditorConfig_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libEditorConfig_la_SOURCES = \
 			EditorConfigUI.cpp \
--- a/src/plugins/contrib/SpellChecker/Makefile.am
+++ b/src/plugins/contrib/SpellChecker/Makefile.am
@@ -15,8 +15,7 @@
 libSpellChecker_la_LIBADD = ../../../sdk/libcodeblocks.la \
 							wxspellchecker/libwxSpellChecker.la \
 							$(WX_LIBS) \
-							$(CB_HUNSPELL_LIBS) \
-							$(CB_TINYXML_LIBS)
+							$(CB_HUNSPELL_LIBS)
 
 libSpellChecker_la_SOURCES = DictionariesNeededDialog.cpp \
                             MySpellingDialog.cpp \
--- a/src/plugins/contrib/Valgrind/Makefile.am
+++ b/src/plugins/contrib/Valgrind/Makefile.am
@@ -11,8 +11,7 @@
 libValgrind_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libValgrind_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libValgrind_la_SOURCES = Valgrind.cpp \
 			ValgrindListLog.cpp \
--- a/src/plugins/contrib/clangd_client/Makefile.am
+++ b/src/plugins/contrib/clangd_client/Makefile.am
@@ -33,9 +33,11 @@
 
 libclangd_client_la_LIBADD = \
 			../../../sdk/libcodeblocks.la \
-			-lutil \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
+
+if ! CODEBLOCKS_NT
+libclangd_client_la_LIBADD += -lutil
+endif
 
 libclangd_client_la_SOURCES = \
 			src/ClangLocator.cpp \
--- a/src/plugins/contrib/codesnippets/Makefile.am
+++ b/src/plugins/contrib/codesnippets/Makefile.am
@@ -16,8 +16,7 @@
 libcodesnippets_la_LIBADD = ../../../sdk/libcodeblocks.la \
 		$(WX_LIBS) \
 		$(WX_GTK_LIBS) \
-		$(WX_X11_LIBS) \
-		$(CB_TINYXML_LIBS)
+		$(WX_X11_LIBS)
 
 libcodesnippets_la_SOURCES = codesnippets.cpp \
 		codesnippetstreectrl.cpp \
--- a/src/plugins/contrib/envvars/Makefile.am
+++ b/src/plugins/contrib/envvars/Makefile.am
@@ -11,8 +11,7 @@
 libenvvars_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libenvvars_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libenvvars_la_SOURCES = envvars.cpp \
                         envvars_cfgdlg.cpp \
--- a/src/plugins/contrib/help_plugin/Makefile.am
+++ b/src/plugins/contrib/help_plugin/Makefile.am
@@ -19,8 +19,7 @@
 libhelp_plugin_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libhelp_plugin_la_LIBADD = ../../../sdk/libcodeblocks.la \
-					$(WX_LIBS) \
-					$(CB_SQUIRREL_LIBS)
+					$(WX_LIBS)
 
 
 if ! HAVE_BZIP2
--- a/src/plugins/contrib/lib_finder/Makefile.am
+++ b/src/plugins/contrib/lib_finder/Makefile.am
@@ -15,8 +15,7 @@
 
 liblib_finder_la_LIBADD = ../../../sdk/libcodeblocks.la \
 		../wxContribItems/wxFlatNotebook/libwxflatnotebook.la \
-		$(WX_LIBS) \
-		$(CB_TINYXML_LIBS)
+		$(WX_LIBS)
 
 liblib_finder_la_SOURCES = \
 		librariesdlg.cpp \
--- a/src/plugins/contrib/wxContribItems/KWIC/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/KWIC/Makefile.am
@@ -7,7 +7,7 @@
 
 wxcontribitems_LTLIBRARIES = libwxkwic.la
 
-libwxkwic_la_LDFLAGS = -version-info 0:1:0
+libwxkwic_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxkwic_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxFlatNotebook/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxFlatNotebook/Makefile.am
@@ -5,7 +5,7 @@
 
 wxcontribitems_LTLIBRARIES = libwxflatnotebook.la
 
-libwxflatnotebook_la_LDFLAGS = -version-info 0:1:0
+libwxflatnotebook_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxflatnotebook_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxImagePanel/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxImagePanel/Makefile.am
@@ -5,7 +5,7 @@
 
 wxcontribitems_LTLIBRARIES = libwximagepanel.la
 
-libwximagepanel_la_LDFLAGS = -version-info 0:1:0
+libwximagepanel_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwximagepanel_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxSpeedButton/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxSpeedButton/Makefile.am
@@ -5,7 +5,7 @@
 
 wxcontribitems_LTLIBRARIES = libwxspeedbutton.la
 
-libwxspeedbutton_la_LDFLAGS = -version-info 0:1:0
+libwxspeedbutton_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxspeedbutton_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxchart/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxchart/Makefile.am
@@ -1,12 +1,15 @@
 AM_CPPFLAGS = $(WX_CXXFLAGS) \
 		-I$(srcdir)/include
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -DWXMAKINGDLL_CHART
+endif
 
 wxcontribitemsdir=$(libdir)/codeblocks/wxContribItems
 
 wxcontribitems_LTLIBRARIES = libwxchartctrl.la
 
-libwxchartctrl_la_LDFLAGS = -version-info 0:1:0
+libwxchartctrl_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxchartctrl_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxchart/include/wx/axis.h
+++ b/src/plugins/contrib/wxContribItems/wxchart/include/wx/axis.h
@@ -16,6 +16,7 @@
 // Headers
 //----------------------------------------------------------------------------
 #include "wx/charttypes.h"
+#include "wx/chartdef.h"
 
 // External Classes
 //-----------------
--- a/src/plugins/contrib/wxContribItems/wxchart/include/wx/bar3dchartpoints.h
+++ b/src/plugins/contrib/wxContribItems/wxchart/include/wx/bar3dchartpoints.h
@@ -22,6 +22,7 @@
 #include "wx/points.h"
 #include "wx/chartcolors.h"
 #include "wx/chartpoints.h"
+#include "wx/chartdef.h"
 
 // External Classes
 //-----------------
--- a/src/plugins/contrib/wxContribItems/wxchart/include/wx/barchartpoints.h
+++ b/src/plugins/contrib/wxContribItems/wxchart/include/wx/barchartpoints.h
@@ -22,6 +22,7 @@
 #include "wx/points.h"
 #include "wx/chartcolors.h"
 #include "wx/chartpoints.h"
+#include "wx/chartdef.h"
 
 // External Classes
 //-----------------
--- a/src/plugins/contrib/wxContribItems/wxchart/include/wx/chart.h
+++ b/src/plugins/contrib/wxContribItems/wxchart/include/wx/chart.h
@@ -27,6 +27,7 @@
 #endif
 
 #include "wx/chartpoints.h"
+#include "wx/chartdef.h"
 
 // External Classes
 //-----------------
--- a/src/plugins/contrib/wxContribItems/wxchart/include/wx/pie3dchartpoints.h
+++ b/src/plugins/contrib/wxContribItems/wxchart/include/wx/pie3dchartpoints.h
@@ -22,6 +22,7 @@
 #include "wx/points.h"
 #include "wx/chartcolors.h"
 #include "wx/chartpoints.h"
+#include "wx/chartdef.h"
 
 // External Classes
 //-----------------
--- a/src/plugins/contrib/wxContribItems/wxled/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxled/Makefile.am
@@ -10,7 +10,7 @@
 
 wxcontribitems_LTLIBRARIES = libwxled.la
 
-libwxled_la_LDFLAGS = -version-info 0:1:0
+libwxled_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxled_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxmathplot/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxmathplot/Makefile.am
@@ -4,7 +4,7 @@
 
 wxcontribitems_LTLIBRARIES = libwxmathplot.la
 
-libwxmathplot_la_LDFLAGS = -version-info 0:1:0
+libwxmathplot_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxmathplot_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxContribItems/wxthings/Makefile.am
+++ b/src/plugins/contrib/wxContribItems/wxthings/Makefile.am
@@ -1,11 +1,15 @@
 AM_CPPFLAGS = $(WX_CXXFLAGS) \
 		-I$(srcdir)/include
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -DWXMAKINGDLL_THINGS
+endif
+
 wxcontribitemsdir=$(libdir)/codeblocks/wxContribItems
 
 wxcontribitems_LTLIBRARIES = libwxcustombutton.la
 
-libwxcustombutton_la_LDFLAGS = -version-info 0:1:0
+libwxcustombutton_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libwxcustombutton_la_LIBADD = $(WX_LIBS)
 
--- a/src/plugins/contrib/wxSmithAui/Makefile.am
+++ b/src/plugins/contrib/wxSmithAui/Makefile.am
@@ -14,8 +14,7 @@
 
 libwxSmithAui_la_LIBADD = ../../../sdk/libcodeblocks.la \
 		       ../wxSmith/libwxsmithlib.la \
-		       $(WX_LIBS) \
-		       $(CB_TINYXML_LIBS)
+		       $(WX_LIBS)
 
 EXTRA_DIST = $(srcdir)/images/*.xpm \
 		$(srcdir)/images/*.xcf \
--- a/src/plugins/contrib/wxSmithContribItems/Makefile.am
+++ b/src/plugins/contrib/wxSmithContribItems/Makefile.am
@@ -38,8 +38,7 @@
 			../wxContribItems/wxthings/libwxcustombutton.la \
 			../wxContribItems/wxled/libwxled.la \
 			../wxContribItems/wxmathplot/libwxmathplot.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 EXTRA_DIST = $(srcdir)/wxchart/*.xpm \
 		$(srcdir)/wxchart/*.png \
--- a/src/plugins/debuggergdb/Makefile.am
+++ b/src/plugins/debuggergdb/Makefile.am
@@ -13,8 +13,7 @@
 
 libdebugger_la_LIBADD = ../../sdk/libcodeblocks.la \
 			$(WX_LIBS) \
-			$(WX_GTK_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_GTK_LIBS)
 
 libdebugger_la_SOURCES = cdb_driver.cpp \
 			databreakpointdlg.cpp \
--- a/src/plugins/openfileslist/Makefile.am
+++ b/src/plugins/openfileslist/Makefile.am
@@ -9,8 +9,7 @@
 libopenfileslist_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libopenfileslist_la_LIBADD = ../../sdk/libcodeblocks.la \
-			$(WX_LIBS) \
-			$(CB_TINYXML_LIBS)
+			$(WX_LIBS)
 
 libopenfileslist_la_SOURCES = openfileslistplugin.cpp
 noinst_HEADERS = openfileslistplugin.h
--- a/src/plugins/projectsimporter/Makefile.am
+++ b/src/plugins/projectsimporter/Makefile.am
@@ -11,8 +11,7 @@
 libprojectsimporter_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libprojectsimporter_la_LIBADD = ../../sdk/libcodeblocks.la \
-				$(WX_LIBS) \
-				$(CB_TINYXML_LIBS)
+				$(WX_LIBS)
 
 libprojectsimporter_la_SOURCES = projectsimporter.cpp \
 				devcpploader.cpp \
--- a/src/sdk/Makefile.am
+++ b/src/sdk/Makefile.am
@@ -13,9 +13,13 @@
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/xpcom/base \
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/xpcom/glue
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -DEXPORT_LIB -DEXPORT_EVENTS -DWXMAKINGDLL_SCI
+endif
+
 lib_LTLIBRARIES = libcodeblocks.la
 
-libcodeblocks_la_LDFLAGS = -version-info @CODEBLOCKS_VERSION_INFO@
+libcodeblocks_la_LDFLAGS = -no-undefined -version-info @CODEBLOCKS_VERSION_INFO@
 
 libcodeblocks_la_LIBADD = scripting/bindings/libsqbindings.la \
 			wxscintilla/libwxscintilla.la \
@@ -25,6 +29,10 @@
 			$(CB_TINYXML_LIBS) \
 			$(CB_SQUIRREL_LIBS)
 
+if CODEBLOCKS_NT
+libcodeblocks_la_LIBADD += -lgdi32
+endif
+
 libcodeblocks_la_SOURCES = annoyingdialog.cpp \
 						autodetectcompilers.cpp \
 						base64.cpp \
--- a/src/sdk/configmanager.cpp
+++ b/src/sdk/configmanager.cpp
@@ -53,8 +53,8 @@
 #include <glib.h>
 #endif // __linux__
 
-template<> CfgMgrBldr* Mgr<CfgMgrBldr>::instance = nullptr;
-template<> bool  Mgr<CfgMgrBldr>::isShutdown = false;
+template<> CfgMgrBldr* DLLIMPORT Mgr<CfgMgrBldr>::instance = nullptr;
+template<> bool DLLIMPORT Mgr<CfgMgrBldr>::isShutdown = false;
 
 wxString ConfigManager::alternate_user_data_path;
 bool ConfigManager::has_alternate_user_data_path=false;
--- a/src/sdk/editormanager.cpp
+++ b/src/sdk/editormanager.cpp
@@ -49,8 +49,8 @@
 #include "projectfileoptionsdlg.h"
 #include "filegroupsandmasks.h"
 
-template<> EditorManager* Mgr<EditorManager>::instance = nullptr;
-template<> bool  Mgr<EditorManager>::isShutdown = false;
+template<> EditorManager* DLLIMPORT Mgr<EditorManager>::instance = nullptr;
+template<> bool DLLIMPORT Mgr<EditorManager>::isShutdown = false;
 
 int ID_NBEditorManager        = wxNewId();
 int ID_EditorManager          = wxNewId();
--- a/src/sdk/logmanager.cpp
+++ b/src/sdk/logmanager.cpp
@@ -21,8 +21,8 @@
 #include "logmanager.h"
 #include "loggers.h"
 
-template<> LogManager* Mgr<LogManager>::instance = nullptr;
-template<> bool  Mgr<LogManager>::isShutdown = false;
+template<> LogManager* DLLIMPORT Mgr<LogManager>::instance = nullptr;
+template<> bool DLLIMPORT Mgr<LogManager>::isShutdown = false;
 
 static NullLogger g_null_log;
 
--- a/src/sdk/projectmanager.cpp
+++ b/src/sdk/projectmanager.cpp
@@ -36,8 +36,8 @@
 //#include "filefilters.h"
 #include "filegroupsandmasks.h"
 
-template<> ProjectManager* Mgr<ProjectManager>::instance = nullptr;
-template<> bool  Mgr<ProjectManager>::isShutdown = false;
+template<> ProjectManager* DLLIMPORT Mgr<ProjectManager>::instance = nullptr;
+template<> bool DLLIMPORT Mgr<ProjectManager>::isShutdown = false;
 
 // static
 bool ProjectManager::s_CanShutdown = true;
--- a/src/sdk/scripting/bindings/Makefile.am
+++ b/src/sdk/scripting/bindings/Makefile.am
@@ -3,6 +3,10 @@
 			-I$(top_srcdir)/src/include \
 			-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -DEXPORT_LIB -DWXMAKINGDLL_SCI
+endif
+
 noinst_LTLIBRARIES = libsqbindings.la
 
 libsqbindings_la_SOURCES = \
--- a/src/sdk/scripting/squirrel/Makefile.am
+++ b/src/sdk/scripting/squirrel/Makefile.am
@@ -1,5 +1,9 @@
 AM_CPPFLAGS = -I$(top_srcdir)/src/include/scripting/include
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -D__WXMSW__ -DEXPORT_LIB
+endif
+
 noinst_LTLIBRARIES = libsquirrel.la
 
 noinst_HEADERS = \
--- a/src/sdk/scriptingmanager.cpp
+++ b/src/sdk/scriptingmanager.cpp
@@ -45,8 +45,8 @@
 
 static_assert(std::is_same<cbHSQUIRRELVM, HSQUIRRELVM>::value, "Types don't match");
 
-template<> ScriptingManager* Mgr<ScriptingManager>::instance = nullptr;
-template<> bool  Mgr<ScriptingManager>::isShutdown = false;
+template<> ScriptingManager* DLLIMPORT Mgr<ScriptingManager>::instance = nullptr;
+template<> bool DLLIMPORT Mgr<ScriptingManager>::isShutdown = false;
 
 static wxString s_ScriptOutput;
 static wxString s_ScriptErrors;
--- a/src/sdk/templatemanager.cpp
+++ b/src/sdk/templatemanager.cpp
@@ -32,8 +32,8 @@
 #include "filefilters.h"
 #include "newfromtemplatedlg.h"
 
-template<> TemplateManager* Mgr<TemplateManager>::instance = nullptr;
-template<> bool  Mgr<TemplateManager>::isShutdown = false;
+template<> TemplateManager* DLLIMPORT Mgr<TemplateManager>::instance = nullptr;
+template<> bool DLLIMPORT Mgr<TemplateManager>::isShutdown = false;
 
 TemplateManager::TemplateManager()
 {
--- a/src/sdk/wxscintilla/Makefile.am
+++ b/src/sdk/wxscintilla/Makefile.am
@@ -9,6 +9,10 @@
 			-DGTK \
 			-D__WX__
 
+if CODEBLOCKS_NT
+AM_CPPFLAGS += -DWXMAKINGDLL_SCI
+endif
+
 noinst_LTLIBRARIES = libwxscintilla.la
 
 libwxscintilla_la_SOURCES = src/PlatWX.cpp \
--- a/src/src/Makefile.am
+++ b/src/src/Makefile.am
@@ -25,7 +25,8 @@
 # codeblocks_LDFLAGS = $(WX_GTK_LIBS)
 
 codeblocks_LDADD =  ../sdk/libcodeblocks.la \
 			$(WX_LIBS) \
-			$(WX_GTK_LIBS)
+			$(WX_GTK_LIBS) \
+			$(CB_SQUIRREL_LIBS)
 
 codeblocks_SOURCES = app.cpp \
 			appglobals.cpp \
@@ -70,7 +71,7 @@
 			watchesdlg.cpp
 
 if CODEBLOCKS_NT
-codeblocks_LDADD += -lexchndl -lcomctl32 -lgdi32
+codeblocks_LDADD += -lexchndl -lcomctl32
 codeblocks_SOURCES += associations.cpp
 endif
 
