--- a/src/sdk/cbeditor.cpp
+++ b/src/sdk/cbeditor.cpp
@@ -80,7 +80,7 @@
 #define C_CHANGEBAR_MARGIN 2
 #define C_FOLDING_MARGIN   3
 
-const int foldingMarginBaseWidth = 16;
+const int foldingMarginBaseWidth = 22;
 const int changeBarMarginBaseWidth = 4;
 
 /* This struct holds private data for the cbEditor class.
@@ -1796,7 +1796,11 @@
         control->SetProperty(_T("fold.preprocessor"), mgr->ReadBool(_T("/folding/fold_preprocessor"), false) ? _T("1") : _T("0"));
 
         control->SetMarginType(C_FOLDING_MARGIN, wxSCI_MARGIN_SYMBOL);
-        control->SetMarginWidth(C_FOLDING_MARGIN, foldingMarginBaseWidth);
+#if wxCHECK_VERSION(3, 1, 0)
+        control->SetMarginWidth(C_FOLDING_MARGIN, control->FromDIP(foldingMarginBaseWidth));
+#else
+        control->SetMarginWidth(C_FOLDING_MARGIN, wxRound(foldingMarginBaseWidth*cbGetContentScaleFactor(*control)));
+#endif
         // use "|" here or we might break plugins that use the margin (none at the moment)
         control->SetMarginMask(C_FOLDING_MARGIN,
                                  control->GetMarginMask(C_FOLDING_MARGIN)
