--- a/src/plugins/codecompletion/Makefile.am
+++ b/src/plugins/codecompletion/Makefile.am
@@ -11,12 +11,9 @@
 libcodecompletion_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libcodecompletion_la_LIBADD = ../../sdk/libcodeblocks.la\
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
-if ! CODEBLOCKS_NT
-libcodecompletion_la_LIBADD += $(CB_TINYXML_LIBS)
-endif
-
 libcodecompletion_la_SOURCES = ccoptionsdlg.cpp \
 			ccoptionsprjdlg.cpp \
 			cctreectrl.cpp \
--- a/src/plugins/contrib/AutoVersioning/Makefile.am
+++ b/src/plugins/contrib/AutoVersioning/Makefile.am
@@ -9,7 +9,8 @@
 libAutoVersioning_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libAutoVersioning_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 libAutoVersioning_la_SOURCES = AutoVersioning.cpp \
             avChangesDlg.cpp \
--- a/src/plugins/contrib/BrowseTracker/Makefile.am
+++ b/src/plugins/contrib/BrowseTracker/Makefile.am
@@ -9,7 +9,8 @@
 libBrowseTracker_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libBrowseTracker_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 libBrowseTracker_la_SOURCES = BrowseTracker.cpp \
 			Version.cpp \
--- a/src/plugins/contrib/CppCheck/Makefile.am
+++ b/src/plugins/contrib/CppCheck/Makefile.am
@@ -11,7 +11,8 @@
 libCppCheck_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libCppCheck_la_LIBADD = ../../../sdk/libcodeblocks.la \
-		$(WX_LIBS)
+		$(WX_LIBS) \
+		$(CB_TINYXML_LIBS)
 
 libCppCheck_la_SOURCES = CppCheck.cpp \
             CppCheckListLog.cpp \
--- a/src/plugins/contrib/DoxyBlocks/Makefile.am
+++ b/src/plugins/contrib/DoxyBlocks/Makefile.am
@@ -9,7 +9,8 @@
 libDoxyBlocks_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libDoxyBlocks_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 libDoxyBlocks_la_SOURCES = AutoDoc.cpp \
 				Config.cpp \
--- a/src/plugins/contrib/EditorConfig/Makefile.am
+++ b/src/plugins/contrib/EditorConfig/Makefile.am
@@ -9,7 +9,8 @@
 libEditorConfig_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libEditorConfig_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 libEditorConfig_la_SOURCES = \
 			EditorConfigUI.cpp \
--- a/src/plugins/contrib/SpellChecker/Makefile.am
+++ b/src/plugins/contrib/SpellChecker/Makefile.am
@@ -15,7 +15,8 @@
 libSpellChecker_la_LIBADD = ../../../sdk/libcodeblocks.la \
 							wxspellchecker/libwxSpellChecker.la \
 							$(WX_LIBS) \
-							$(CB_HUNSPELL_LIBS)
+							$(CB_HUNSPELL_LIBS) \
+							$(CB_TINYXML_LIBS)
 
 libSpellChecker_la_SOURCES = DictionariesNeededDialog.cpp \
                             MySpellingDialog.cpp \
--- a/src/plugins/contrib/Valgrind/Makefile.am
+++ b/src/plugins/contrib/Valgrind/Makefile.am
@@ -11,7 +11,8 @@
 libValgrind_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libValgrind_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 libValgrind_la_SOURCES = Valgrind.cpp \
 			ValgrindListLog.cpp \
--- a/src/plugins/contrib/clangd_client/Makefile.am
+++ b/src/plugins/contrib/clangd_client/Makefile.am
@@ -33,7 +33,8 @@
 
 libclangd_client_la_LIBADD = \
 			../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 if ! CODEBLOCKS_NT
 libclangd_client_la_LIBADD += -lutil
 endif
--- a/src/plugins/contrib/codesnippets/Makefile.am
+++ b/src/plugins/contrib/codesnippets/Makefile.am
@@ -16,7 +16,8 @@
 libcodesnippets_la_LIBADD = ../../../sdk/libcodeblocks.la \
 		$(WX_LIBS) \
 		$(WX_GTK_LIBS) \
-		$(WX_X11_LIBS)
+		$(WX_X11_LIBS) \
+		$(CB_TINYXML_LIBS)
 
 libcodesnippets_la_SOURCES = codesnippets.cpp \
 		codesnippetstreectrl.cpp \
--- a/src/plugins/contrib/envvars/Makefile.am
+++ b/src/plugins/contrib/envvars/Makefile.am
@@ -11,7 +11,8 @@
 libenvvars_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libenvvars_la_LIBADD = ../../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 libenvvars_la_SOURCES = envvars.cpp \
                         envvars_cfgdlg.cpp \
--- a/src/plugins/contrib/lib_finder/Makefile.am
+++ b/src/plugins/contrib/lib_finder/Makefile.am
@@ -15,7 +15,8 @@
 
 liblib_finder_la_LIBADD = ../../../sdk/libcodeblocks.la \
 		../wxContribItems/wxFlatNotebook/libwxflatnotebook.la \
-		$(WX_LIBS)
+		$(WX_LIBS) \
+		$(CB_TINYXML_LIBS)
 
 liblib_finder_la_SOURCES = \
 		librariesdlg.cpp \
--- a/src/plugins/contrib/wxSmithAui/Makefile.am
+++ b/src/plugins/contrib/wxSmithAui/Makefile.am
@@ -14,7 +14,8 @@
 
 libwxSmithAui_la_LIBADD = ../../../sdk/libcodeblocks.la \
 		       ../wxSmith/libwxsmithlib.la \
-		       $(WX_LIBS)
+		       $(WX_LIBS) \
+		       $(CB_TINYXML_LIBS)
 
 EXTRA_DIST = $(srcdir)/images/*.xpm \
 		$(srcdir)/images/*.xcf \
--- a/src/plugins/contrib/wxSmithContribItems/Makefile.am
+++ b/src/plugins/contrib/wxSmithContribItems/Makefile.am
@@ -38,7 +38,8 @@
 			../wxContribItems/wxthings/libwxcustombutton.la \
 			../wxContribItems/wxled/libwxled.la \
 			../wxContribItems/wxmathplot/libwxmathplot.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
 EXTRA_DIST = $(srcdir)/wxchart/*.xpm \
 		$(srcdir)/wxchart/*.png \
--- a/src/plugins/debuggergdb/Makefile.am
+++ b/src/plugins/debuggergdb/Makefile.am
@@ -13,12 +13,9 @@
 
 libdebugger_la_LIBADD = ../../sdk/libcodeblocks.la \
 			$(WX_LIBS) \
-			$(WX_GTK_LIBS)
+			$(WX_GTK_LIBS) \
+			$(CB_TINYXML_LIBS)
 
-if ! CODEBLOCKS_NT
-libdebugger_la_LIBADD += $(CB_TINYXML_LIBS)
-endif
-
 libdebugger_la_SOURCES = cdb_driver.cpp \
 			databreakpointdlg.cpp \
 			debugger_defs.cpp \
--- a/src/plugins/openfileslist/Makefile.am
+++ b/src/plugins/openfileslist/Makefile.am
@@ -9,12 +9,9 @@
 libopenfileslist_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libopenfileslist_la_LIBADD = ../../sdk/libcodeblocks.la \
-			$(WX_LIBS)
+			$(WX_LIBS) \
+			$(CB_TINYXML_LIBS)
 
-if ! CODEBLOCKS_NT
-libopenfileslist_la_LIBADD += $(CB_TINYXML_LIBS)
-endif
-
 libopenfileslist_la_SOURCES = openfileslistplugin.cpp
 noinst_HEADERS = openfileslistplugin.h
 
--- a/src/plugins/projectsimporter/Makefile.am
+++ b/src/plugins/projectsimporter/Makefile.am
@@ -11,12 +11,9 @@
 libprojectsimporter_la_LDFLAGS = @MODULE_SHARED_LDFLAGS@ -version-info 0:1:0 -no-undefined -avoid-version
 
 libprojectsimporter_la_LIBADD = ../../sdk/libcodeblocks.la \
-				$(WX_LIBS)
+				$(WX_LIBS) \
+				$(CB_TINYXML_LIBS)
 
-if ! CODEBLOCKS_NT
-libprojectsimporter_la_LIBADD += $(CB_TINYXML_LIBS)
-endif
-
 libprojectsimporter_la_SOURCES = projectsimporter.cpp \
 				devcpploader.cpp \
 				msvc10loader.cpp \
