From 1bbb0f52c354732118ec11039b09c6fc336f7145 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Wed, 28 Feb 2018 04:29:41 -0500
Subject: [PATCH 11/11] Make win32 pch files

---
 src/include/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/include/Makefile.am b/src/include/Makefile.am
index 828d76e6e..8fe6778f6 100644
--- a/src/include/Makefile.am
+++ b/src/include/Makefile.am
@@ -180,11 +180,11 @@ sdk_HEADERS = $(ALLHEADERS)
 
 sdk.h.gch: sdk_precomp.h.gch $(ALLHEADERS)
 	-rm -f sdk.h.gch
-	$(CXXCOMPILE) -o sdk.h.gch -xc++-header $(srcdir)/sdk.h
+	$(CXXCOMPILE) -DCB_PRECOMP -DDLL_EXPORT -DBUILDING_PLUGIN -o sdk.h.gch -xc++-header $(srcdir)/sdk.h
 
 sdk_precomp.h.gch: $(ALLHEADERS)
 	-rm -f sdk_precomp.h.gch
-	$(CXXCOMPILE) -o sdk_precomp.h.gch -xc++-header $(srcdir)/sdk_precomp.h
+	$(CXXCOMPILE) -DCB_PRECOMP -DDLL_EXPORT -DEXPORT_LIB -DEXPORT_EVENTS -o sdk_precomp.h.gch -xc++-header $(srcdir)/sdk_precomp.h
 
 clean-local:
 	-rm -f sdk.h.gch
-- 
2.16.2.windows.1

