From 8e501d7ebf569f96964ac5b384325b73f4d0a320 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Wed, 28 Feb 2018 04:25:20 -0500
Subject: [PATCH 08/11] Fix sdk.h pch issue

---
 src/include/sdk.h | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/include/sdk.h b/src/include/sdk.h
index 46361c730..3ac9ae459 100644
--- a/src/include/sdk.h
+++ b/src/include/sdk.h
@@ -10,11 +10,6 @@
 #ifndef SDK_H
 #define SDK_H
 
-#ifdef __WXMSW__
-	#include "sdk_common.h"
-#else
-	// for non-windows platforms, one PCH is enough
-	#include "sdk_precomp.h"
-#endif
+    #include "sdk_common.h"
 
 #endif // SDK_H
-- 
2.16.2.windows.1

