--- ./m4/intltool.m4.orig	2014-07-23 14:02:09.000000000 +0200
+++ ./m4/intltool.m4	2015-04-01 21:14:20.543805300 +0200
@@ -155,31 +155,6 @@
 # Substitute ALL_LINGUAS so we can use it in po/Makefile
 AC_SUBST(ALL_LINGUAS)
 
-# Set DATADIRNAME correctly if it is not set yet
-# (copied from glib-gettext.m4)
-if test -z "$DATADIRNAME"; then
-  AC_LINK_IFELSE(
-    [AC_LANG_PROGRAM([[]],
-                     [[extern int _nl_msg_cat_cntr;
-                       return _nl_msg_cat_cntr]])],
-    [DATADIRNAME=share],
-    [case $host in
-    *-*-solaris*)
-    dnl On Solaris, if bind_textdomain_codeset is in libc,
-    dnl GNU format message catalog is always supported,
-    dnl since both are added to the libc all together.
-    dnl Hence, we'd like to go with DATADIRNAME=share
-    dnl in this case.
-    AC_CHECK_FUNC(bind_textdomain_codeset,
-      [DATADIRNAME=share], [DATADIRNAME=lib])
-    ;;
-    *)
-    [DATADIRNAME=lib]
-    ;;
-    esac])
-fi
-AC_SUBST(DATADIRNAME)
-
 IT_PO_SUBDIR([po])
 
 ])
--- ./plparse/Makefile.am.orig	2014-01-13 10:17:47.000000000 +0100
+++ ./plparse/Makefile.am	2015-04-01 21:14:20.570806300 +0200
@@ -194,6 +194,7 @@
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir) --add-include-path=$(builddir) --warn-all
 INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 if HAVE_INTROSPECTION
 introspection_sources = \
--- ./plparse/plparser.symbols.orig	2011-09-18 12:33:29.000000000 +0200
+++ ./plparse/plparser.symbols	2015-04-01 21:14:20.578806900 +0200
@@ -4,7 +4,6 @@
 totem_cd_get_human_readable_name
 totem_cd_has_medium
 totem_cd_mrl_from_type
-totem_disc_media_type_get_type
 totem_disc_media_type_quark
 totem_pl_parser_add_ignored_mimetype
 totem_pl_parser_add_ignored_scheme
@@ -14,7 +13,6 @@
 totem_pl_parser_error_get_type
 totem_pl_parser_error_quark
 totem_pl_parser_get_type
-totemplparser_marshal_VOID__STRING_STRING_STRING
 totem_pl_parser_new
 totem_pl_parser_parse
 totem_pl_parser_parse_async
--- ./plparse/totem-pl-parser.c.orig	2015-02-17 15:27:48.000000000 +0100
+++ ./plparse/totem-pl-parser.c	2015-04-01 21:16:17.484317500 +0200
@@ -662,8 +662,14 @@
 static gpointer
 totem_pl_parser_real_init_i18n (gpointer data)
 {
-	bindtextdomain (GETTEXT_PACKAGE, GNOMELOCALEDIR);
+	gchar *dir;
+	gchar *path;
+	dir = g_win32_get_package_installation_directory_of_module (NULL);
+	path = g_build_filename (dir, "share", "locale", NULL);
+	bindtextdomain (GETTEXT_PACKAGE, path);
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+	g_free (path);
+	g_free (dir);
 	return NULL;
 }
 
--- ./po/Makefile.in.in.orig	2014-07-23 14:02:09.000000000 +0200
+++ ./po/Makefile.in.in	2015-04-01 21:14:20.596808200 +0200
@@ -33,8 +33,7 @@
 datadir = @datadir@
 datarootdir = @datarootdir@
 libdir = @libdir@
-DATADIRNAME = @DATADIRNAME@
-itlocaledir = $(prefix)/$(DATADIRNAME)/locale
+localedir = @localedir@
 subdir = po
 install_sh = @install_sh@
 # Automake >= 1.8 provides @mkdir_p@.
@@ -108,7 +107,7 @@
 install-data-yes: all
 	linguas="$(USE_LINGUAS)"; \
 	for lang in $$linguas; do \
-	  dir=$(DESTDIR)$(itlocaledir)/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$(localedir)/$$lang/LC_MESSAGES; \
 	  $(mkdir_p) $$dir; \
 	  if test -r $$lang.gmo; then \
 	    $(INSTALL_DATA) $$lang.gmo $$dir/$(GETTEXT_PACKAGE).mo; \
@@ -142,8 +141,8 @@
 uninstall:
 	linguas="$(USE_LINGUAS)"; \
 	for lang in $$linguas; do \
-	  rm -f $(DESTDIR)$(itlocaledir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo; \
-	  rm -f $(DESTDIR)$(itlocaledir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo.m; \
+	  rm -f $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo; \
+	  rm -f $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE).mo.m; \
 	done
 
 check: all $(GETTEXT_PACKAGE).pot
