--- libnotify-0.8.8/meson.build.orig	2026-01-09 11:16:29.000000000 +0800
+++ libnotify-0.8.8/meson.build	2026-07-25 15:18:38.800017300 +0800
@@ -46,15 +46,11 @@
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
 glib_dep = dependency('glib-2.0', version: glib_req_version)
 gio_dep = dependency('gio-2.0', version: glib_req_version)
-gio_unix_dep = dependency('gio-unix-2.0', version: glib_req_version)
 gtk_dep = dependency('gtk4', version: '>= 4.0', required: get_option('tests'))
 
-have_gio_desktop_app_info = cc.check_header('gio/gdesktopappinfo.h',
-  dependencies: gio_unix_dep,
-  required: host_machine.system() == 'linux',
-)
+have_gio_desktop_app_info = false
 
-libnotify_deps = [gdk_pixbuf_dep, gio_dep, glib_dep, gio_unix_dep]
+libnotify_deps = [gdk_pixbuf_dep, gio_dep, glib_dep]
 tests_deps = [gtk_dep]
 
 conf = configuration_data()
