diff -Naur libnotify-0.7.9.orig/libnotify/notification.c libnotify-0.7.9/libnotify/notification.c
--- libnotify-0.7.9.orig/libnotify/notification.c	2020-02-26 20:07:28.392563800 +0100
+++ libnotify-0.7.9/libnotify/notification.c	2022-04-13 11:15:41.995249600 +0200
@@ -28,6 +28,7 @@
 #include "notify.h"
 #include "internal.h"
 
+#include <stdlib.h>
 
 /**
  * SECTION:notification
@@ -442,7 +443,7 @@
                 if (base_path && base_path[0] == G_DIR_SEPARATOR) {
                         path_filename = g_strdup (base_path);
                 } else {
-                        path_filename = realpath (base_path, NULL);
+                        path_filename = _fullpath (base_path, NULL, _MAX_PATH);
                 }
         }
 
