--- libnotify-0.7.6/libnotify/Makefile.am	Fri Jan 27 02:52:36 2012
+++ libnotify-0.7.6/libnotify/Makefile.am	Mon Oct 20 11:08:31 2014
@@ -44,7 +44,8 @@

 libnotify_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-export-symbols-regex "^notify_.*"
+	-export-symbols-regex "^notify_.*" \
+	-no-undefined

 notify-marshal.h: notify-marshal.list
 	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=$(MARSHAL_PREFIX) $(srcdir)/notify-marshal.list \
@@ -86,6 +87,7 @@

 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS = Notify-0.7.gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 INTROSPECTION_SCANNER_ARGS = --add-include-path=$(top_srcdir) --add-include-path=$(top_builddir) --warn-all -I$(top_srcdir) -I$(top_builddir)
 INTROSPECTION_COMPILER_ARGS = --includedir=$(top_srcdir)

