--- libxslt-1.1.26/configure.in.orig	2009-09-24 18:27:30 +0400
+++ libxslt-1.1.26/configure.in	2012-09-06 14:34:36 +0400
@@ -632,6 +632,10 @@
  AC_DEFINE([_WINSOCKAPI_],1,[Using the Win32 Socket implementation])
  AC_DEFINE([snprintf],[_snprintf],[Win32 Std C name mangling work-around])
  AC_DEFINE([vsnprintf],[_vsnprintf],[Win32 Std C name mangling work-around])
+ if test "x${PYTHON}" != "x"
+ then
+   WIN32_EXTRA_PYTHON_LIBADD="-shrext .pyd " #-L${pythondir}/../../libs -lpython$(echo ${PYTHON_VERSION} | tr -d .)
+ fi
  ;;
 esac
 AC_SUBST(WIN32_EXTRA_LIBADD)
--- libxslt-1.1.26/configure.in.orig	2012-09-06 14:40:22 +0400
+++ libxslt-1.1.26/configure.in	2012-09-06 15:23:05 +0400
@@ -640,6 +640,7 @@
 esac
 AC_SUBST(WIN32_EXTRA_LIBADD)
 AC_SUBST(WIN32_EXTRA_LDFLAGS)
+AC_SUBST(WIN32_EXTRA_PYTHON_LIBADD)
 
 
 AC_SUBST(XSLTPROCDV)
--- libxslt-1.1.27/python/Makefile.am.orig	2012-09-26 17:04:41 +0400
+++ libxslt-1.1.27/python/Makefile.am	2012-09-26 17:18:52 +0400
@@ -38,7 +38,7 @@
 
 libxsltmod_la_SOURCES = libxslt.c types.c
 nodist_libxsltmod_la_SOURCES = libxslt-py.c
-libxsltmod_la_LIBADD = $(mylibs) $(PYTHON_LIBS)
+libxsltmod_la_LIBADD = $(mylibs) $(PYTHON_LIBS) @WIN32_EXTRA_PYTHON_LIBADD@
 
 libxslt.py: $(srcdir)/libxsl.py libxsltclass.py
 	cat $(srcdir)/libxsl.py libxsltclass.py > $@
