--- libxslt-1.1.27/libxslt/xsltlocale.h.orig	2012-09-26 17:54:17 +0400
+++ libxslt-1.1.27/libxslt/xsltlocale.h	2012-09-26 18:08:49 +0400
@@ -12,6 +12,7 @@
 #define __XML_XSLTLOCALE_H__
 
 #include <libxml/xmlstring.h>
+#include "xsltexports.h"
 
 #ifdef XSLT_LOCALE_XLOCALE
 
@@ -53,6 +53,6 @@
 void xsltFreeLocale(xsltLocale locale);
 xsltLocaleChar *xsltStrxfrm(xsltLocale locale, const xmlChar *string);
 int xsltLocaleStrcmp(xsltLocale locale, const xsltLocaleChar *str1, const xsltLocaleChar *str2);
-void xsltFreeLocales(void);
+XSLTPUBFUN void XSLTCALL xsltFreeLocales (void);
 
 #endif /* __XML_XSLTLOCALE_H__ */
