diff -Naur a/Dependencies/GenericTextEditor/include/generictexteditor.hxx b/Dependencies/GenericTextEditor/include/generictexteditor.hxx
--- a/Dependencies/GenericTextEditor/include/generictexteditor.hxx	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/GenericTextEditor/include/generictexteditor.hxx	2013-12-20 08:47:13.456400000 +0400
@@ -61,6 +61,11 @@
 public:
     GenericTextEditor(QString editorName, QWidget *parent = 0);
     virtual ~GenericTextEditor();
+	
+    /// Get reference to singleton instance; or crash if N/A.
+    static GenericTextEditor& getSingleton(void);
+    /// Get pointer to singleton instance; or pointer if N/A.
+    static GenericTextEditor* getSingletonPtr(void);
 
     bool displayTextFromFile(QString filePath, QString optionalData = "");
     bool displayText(QString docName, QString text, QString extension = "", QString optionalData = "");
diff -Naur a/Dependencies/GenericTextEditor/src/generictexteditor.cpp b/Dependencies/GenericTextEditor/src/generictexteditor.cpp
--- a/Dependencies/GenericTextEditor/src/generictexteditor.cpp	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/GenericTextEditor/src/generictexteditor.cpp	2013-12-20 08:48:01.005200000 +0400
@@ -49,6 +49,15 @@
 template<> GenericTextEditor* Ogre::Singleton<GenericTextEditor>::msSingleton = 0;
 TextCodecExtensionFactoryMap GenericTextEditor::mRegisteredCodecFactories = TextCodecExtensionFactoryMap();
 
+GenericTextEditor* GenericTextEditor::getSingletonPtr () {
+    return msSingleton;
+}
+
+GenericTextEditor& GenericTextEditor::getSingleton () {  
+    assert (msSingleton);
+    return *msSingleton;  
+}
+
 //-----------------------------------------------------------------------------------------
 GenericTextEditor::GenericTextEditor(QString editorName, QWidget *parent) : QMdiArea(parent)
 {
