diff -Naur a/Dependencies/Caelum-0.5.0/main/CMakeLists.txt b/Dependencies/Caelum-0.5.0/main/CMakeLists.txt
--- a/Dependencies/Caelum-0.5.0/main/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/Caelum-0.5.0/main/CMakeLists.txt	2013-12-19 09:05:54.898200000 +0400
@@ -80,7 +80,7 @@
 
 if(WIN32)
 	add_custom_command(TARGET Caelum POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:Caelum>" ../../../RunPath/bin)
 
 	install(TARGETS Caelum
 		RUNTIME DESTINATION bin
diff -Naur a/Dependencies/GenericImageEditor/CMakeLists.txt b/Dependencies/GenericImageEditor/CMakeLists.txt
--- a/Dependencies/GenericImageEditor/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/GenericImageEditor/CMakeLists.txt	2013-12-19 09:05:00.501000000 +0400
@@ -87,9 +87,9 @@
 
     if(WIN32)
         add_custom_command(TARGET GenericImageEditor PRE_BUILD
-            COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
         add_custom_command(TARGET GenericImageEditor POST_BUILD
-            COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:GenericImageEditor>" ../../RunPath/bin)
 
         install(TARGETS GenericImageEditor
             RUNTIME DESTINATION bin
diff -Naur a/Dependencies/GenericTextEditor/CMakeLists.txt b/Dependencies/GenericTextEditor/CMakeLists.txt
--- a/Dependencies/GenericTextEditor/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/GenericTextEditor/CMakeLists.txt	2013-12-19 09:07:14.879400000 +0400
@@ -85,9 +85,9 @@
 
     if(WIN32)
         add_custom_command(TARGET GenericTextEditor PRE_BUILD
-            COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
         add_custom_command(TARGET GenericTextEditor POST_BUILD
-            COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:GenericTextEditor>" ../../RunPath/bin)
 
         install(TARGETS GenericTextEditor
             RUNTIME DESTINATION bin
diff -Naur a/Dependencies/HYDRAX/src/CMakeLists.txt b/Dependencies/HYDRAX/src/CMakeLists.txt
--- a/Dependencies/HYDRAX/src/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/HYDRAX/src/CMakeLists.txt	2013-12-19 10:15:57.043600000 +0400
@@ -73,9 +73,9 @@
 
 IF(WIN32)
 	add_custom_command( TARGET Hydrax PRE_BUILD
-		COMMAND if not exist ..\\..\\..\\..\\..\\RunPath\\bin mkdir ..\\..\\..\\..\\..\\RunPath\\bin )
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../../../../RunPath/bin )
 	add_custom_command( TARGET Hydrax POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\..\\..\\..\\RunPath\\bin )
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:Hydrax>" ../../../../../RunPath/bin )
 ENDIF(WIN32)
 
 set_target_properties(Hydrax PROPERTIES SOLUTION_FOLDER Dependencies)
diff -Naur a/Dependencies/ImageConverter/CMakeLists.txt b/Dependencies/ImageConverter/CMakeLists.txt
--- a/Dependencies/ImageConverter/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/ImageConverter/CMakeLists.txt	2013-12-19 09:10:23.202600000 +0400
@@ -66,15 +66,14 @@
         add_custom_command(TARGET ImageConverter PRE_BUILD
             COMMAND mkdir -p ../../RunPath/lib)
         add_custom_command(TARGET ImageConverter POST_BUILD
-            COMMAND cp -r ../lib/libImageConverter_Ogitor.*
-../../RunPath/lib/)
+            COMMAND cp -r ../lib/libImageConverter_Ogitor.* ../../RunPath/lib/)
     endif(UNIX AND NOT APPLE)
 
     if(WIN32)
         add_custom_command(TARGET ImageConverter PRE_BUILD
-            COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
         add_custom_command(TARGET ImageConverter POST_BUILD
-            COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:ImageConverter>" ../../RunPath/bin)
 
         install(TARGETS ImageConverter
             RUNTIME DESTINATION bin
diff -Naur a/Dependencies/MeshMagick/CMakeLists.txt b/Dependencies/MeshMagick/CMakeLists.txt
--- a/Dependencies/MeshMagick/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/MeshMagick/CMakeLists.txt	2013-12-19 09:11:33.995400000 +0400
@@ -99,9 +99,9 @@
 
 if(WIN32)
 	add_custom_command(TARGET meshmagick_shared_lib PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
 	add_custom_command(TARGET meshmagick_shared_lib POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:meshmagick_shared_lib>" ../../RunPath/bin)
 
 	install(TARGETS meshmagick_bin meshmagick_shared_lib
 		RUNTIME DESTINATION bin
diff -Naur a/Dependencies/OFS/CMakeLists.txt b/Dependencies/OFS/CMakeLists.txt
--- a/Dependencies/OFS/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/OFS/CMakeLists.txt	2013-12-19 08:59:09.267000000 +0400
@@ -42,9 +42,9 @@
 
 if(WIN32)
 	add_custom_command(TARGET OFS PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
 	add_custom_command(TARGET OFS POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OFS>" ../../RunPath/bin)
 
 	install(TARGETS OFS RUNTIME DESTINATION bin CONFIGURATIONS Debug Release RelWithDebInfo MinSizeRel)
 endif(WIN32)
diff -Naur a/Dependencies/OgreTerrainConverter/CMakeLists.txt b/Dependencies/OgreTerrainConverter/CMakeLists.txt
--- a/Dependencies/OgreTerrainConverter/CMakeLists.txt	2013-12-18 22:35:28.000000000 +0400
+++ b/Dependencies/OgreTerrainConverter/CMakeLists.txt	2013-12-19 08:59:09.267000000 +0400
@@ -31,9 +31,9 @@
 
 if(WIN32)
 	add_custom_command(TARGET OgreTerrainConverter PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
 	add_custom_command(TARGET OgreTerrainConverter POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgreTerrainConverter>" ../../RunPath/bin)
 
 	install(TARGETS OgreTerrainConverter RUNTIME DESTINATION bin CONFIGURATIONS Debug Release RelWithDebInfo MinSizeRel)
 endif(WIN32)
diff -Naur a/Dependencies/SkyX/CMakeLists.txt b/Dependencies/SkyX/CMakeLists.txt
--- a/Dependencies/SkyX/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Dependencies/SkyX/CMakeLists.txt	2013-12-19 12:01:22.264000000 +0400
@@ -71,9 +71,9 @@
 
 if(WIN32)
 	add_custom_command(TARGET SkyX PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\bin mkdir ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/bin)
 	add_custom_command(TARGET SkyX POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:SkyX>" ../../RunPath/bin)
 
 	install(TARGETS SkyX
 		RUNTIME DESTINATION bin
diff -Naur a/Ogitor/CMakeLists.txt b/Ogitor/CMakeLists.txt
--- a/Ogitor/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Ogitor/CMakeLists.txt	2013-12-19 11:59:54.990000000 +0400
@@ -181,9 +181,9 @@
 
 if(WIN32)
 	add_custom_command(TARGET Ogitor PRE_BUILD
-		COMMAND if not exist ..\\RunPath\\bin mkdir ..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../RunPath/bin)
 	add_custom_command(TARGET Ogitor POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:Ogitor>" ../RunPath/bin)
 
 	install(TARGETS Ogitor RUNTIME DESTINATION bin CONFIGURATIONS Debug Release RelWithDebInfo MinSizeRel)
 endif(WIN32)
diff -Naur a/OgreOfsPlugin/CMakeLists.txt b/OgreOfsPlugin/CMakeLists.txt
--- a/OgreOfsPlugin/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/OgreOfsPlugin/CMakeLists.txt	2013-12-19 09:14:49.463400000 +0400
@@ -22,9 +22,9 @@
 
 if(WIN32)
 	add_custom_command(TARGET OgreOfsPlugin PRE_BUILD
-		COMMAND if not exist ..\\RunPath\\bin mkdir ..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../RunPath/bin)
 	add_custom_command(TARGET OgreOfsPlugin POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\RunPath\\bin)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgreOfsPlugin>" ../RunPath/bin)
 
 	install(TARGETS OgreOfsPlugin RUNTIME DESTINATION bin CONFIGURATIONS Debug Release RelWithDebInfo MinSizeRel)
 endif(WIN32)
diff -Naur a/Plugins/CustomSerializer/CMakeLists.txt b/Plugins/CustomSerializer/CMakeLists.txt
--- a/Plugins/CustomSerializer/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/CustomSerializer/CMakeLists.txt	2013-12-19 09:40:22.731200000 +0400
@@ -45,11 +45,17 @@
 
 if(WIN32)
 	add_custom_command(TARGET CustomSerializer PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 	add_custom_command(TARGET CustomSerializer POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-	add_custom_command(TARGET CustomSerializer POST_BUILD
-		COMMAND if exist ..\\..\\RunPath\\Plugins\\CustomSerializer_d.dll move ..\\..\\RunPath\\Plugins\\CustomSerializer_d.dll ..\\..\\RunPath\\Plugins\\debug)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:CustomSerializer>" ../../RunPath/Plugins)
+	set(FileToCheck "../../RunPath/Plugins/CustomSerializer_d.dll")
+	set(MoveToDir "../../RunPath/Plugins/debug")
+	if(EXISTS ${FileToCheck})
+		add_custom_command(TARGET CustomSerializer POST_BUILD
+			COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+	endif()
 
 	install(TARGETS CustomSerializer
 		RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/DotSceneSerializer/CMakeLists.txt b/Plugins/DotSceneSerializer/CMakeLists.txt
--- a/Plugins/DotSceneSerializer/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/DotSceneSerializer/CMakeLists.txt	2013-12-19 09:48:58.124000000 +0400
@@ -49,11 +49,17 @@
 
 if(WIN32)
 	add_custom_command(TARGET DotSceneSerializer PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 	add_custom_command(TARGET DotSceneSerializer POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-	add_custom_command(TARGET DotSceneSerializer POST_BUILD
-		COMMAND if exist ..\\..\\RunPath\\Plugins\\DotSceneSerializer_d.dll move ..\\..\\RunPath\\Plugins\\DotSceneSerializer_d.dll ..\\..\\RunPath\\Plugins\\debug)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:DotSceneSerializer>" ../../RunPath/Plugins)
+	set(FileToCheck "../../RunPath/Plugins/DotSceneSerializer_d.dll")
+	set(MoveToDir "../../RunPath/Plugins/debug")
+	if(EXISTS ${FileToCheck})
+		add_custom_command(TARGET DotSceneSerializer POST_BUILD
+			COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+	endif()
 
 	install(TARGETS DotSceneSerializer
 		RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgAngelDialogManager/CMakeLists.txt b/Plugins/OgAngelDialogManager/CMakeLists.txt
--- a/Plugins/OgAngelDialogManager/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgAngelDialogManager/CMakeLists.txt	2013-12-19 09:48:51.119600000 +0400
@@ -88,11 +88,17 @@
 
 	if(WIN32)
 		add_custom_command(TARGET OgAngelDialogManager PRE_BUILD
-			COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 		add_custom_command(TARGET OgAngelDialogManager POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-		add_custom_command(TARGET OgAngelDialogManager POST_BUILD
-			COMMAND if exist ..\\..\\RunPath\\Plugins\\OgAngelDialogManager_d.dll move ..\\..\\RunPath\\Plugins\\OgAngelDialogManager_d.dll ..\\..\\RunPath\\Plugins\\debug)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgAngelDialogManager>" ../../RunPath/Plugins)
+		set(FileToCheck "../../RunPath/Plugins/OgAngelDialogManager_d.dll")
+		set(MoveToDir "../../RunPath/Plugins/debug")
+		if(EXISTS ${FileToCheck})
+			add_custom_command(TARGET OgAngelDialogManager POST_BUILD
+				COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+		endif()
 
 		install(TARGETS OgAngelDialogManager
 			RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgAngelScript/CMakeLists.txt b/Plugins/OgAngelScript/CMakeLists.txt
--- a/Plugins/OgAngelScript/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgAngelScript/CMakeLists.txt	2013-12-19 09:51:48.039200000 +0400
@@ -70,9 +70,17 @@
 
 if(WIN32)
 	add_custom_command(TARGET OgAngelScript PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 	add_custom_command(TARGET OgAngelScript POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins & if exist ..\\..\\RunPath\\Plugins\\OgAngelScript_d.dll move ..\\..\\RunPath\\Plugins\\OgAngelScript_d.dll ..\\..\\RunPath\\Plugins\\debug)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgAngelScript>" ../../RunPath/Plugins)
+	set(FileToCheck "../../RunPath/Plugins/OgAngelScript_d.dll")
+	set(MoveToDir "../../RunPath/Plugins/debug")
+	if(EXISTS ${FileToCheck})
+		add_custom_command(TARGET OgAngelScript POST_BUILD
+			COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+	endif()
 
 	install(TARGETS OgAngelScript
 		RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgCaelum/CMakeLists.txt b/Plugins/OgCaelum/CMakeLists.txt
--- a/Plugins/OgCaelum/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgCaelum/CMakeLists.txt	2013-12-19 09:51:35.387600000 +0400
@@ -48,9 +48,17 @@
 
 if(WIN32)
 	add_custom_command(TARGET OgCaelum PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 	add_custom_command(TARGET OgCaelum POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins & if exist ..\\..\\RunPath\\Plugins\\OgCaelum_d.dll move ..\\..\\RunPath\\Plugins\\OgCaelum_d.dll ..\\..\\RunPath\\Plugins\\debug)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgCaelum>" ../../RunPath/Plugins)
+	set(FileToCheck "../../RunPath/Plugins/OgCaelum_d.dll")
+	set(MoveToDir "../../RunPath/Plugins/debug")
+	if(EXISTS ${FileToCheck})
+		add_custom_command(TARGET OgCaelum POST_BUILD
+			COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+	endif()
 
 	install(TARGETS OgCaelum
 		RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgHydrax/CMakeLists.txt b/Plugins/OgHydrax/CMakeLists.txt
--- a/Plugins/OgHydrax/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgHydrax/CMakeLists.txt	2013-12-19 09:54:48.266000000 +0400
@@ -48,9 +48,17 @@
 
 if(WIN32)
 	add_custom_command(TARGET OgHydrax PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 	add_custom_command(TARGET OgHydrax POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins & if exist ..\\..\\RunPath\\Plugins\\OgHydrax_d.dll move ..\\..\\RunPath\\Plugins\\OgHydrax_d.dll ..\\..\\RunPath\\Plugins\\debug)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgHydrax>" ../../RunPath/Plugins)
+	set(FileToCheck "../../RunPath/Plugins/OgHydrax_d.dll")
+	set(MoveToDir "../../RunPath/Plugins/debug")
+	if(EXISTS ${FileToCheck})
+		add_custom_command(TARGET OgHydrax POST_BUILD
+			COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+	endif()
 
 	install(TARGETS OgHydrax
 		RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgMaterialEditor/CMakeLists.txt b/Plugins/OgMaterialEditor/CMakeLists.txt
--- a/Plugins/OgMaterialEditor/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgMaterialEditor/CMakeLists.txt	2013-12-19 09:57:44.842400000 +0400
@@ -97,11 +97,17 @@
 
 	if(WIN32)
 		add_custom_command(TARGET OgMaterialEditor PRE_BUILD
-			COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 		add_custom_command(TARGET OgMaterialEditor POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-		add_custom_command(TARGET OgMaterialEditor POST_BUILD
-			COMMAND if exist ..\\..\\RunPath\\Plugins\\OgMaterialEditor_d.dll move ..\\..\\RunPath\\Plugins\\OgMaterialEditor_d.dll ..\\..\\RunPath\\Plugins\\debug)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgMaterialEditor>" ../../RunPath/Plugins)
+		set(FileToCheck "../../RunPath/Plugins/OgMaterialEditor_d.dll")
+		set(MoveToDir "../../RunPath/Plugins/debug")
+		if(EXISTS ${FileToCheck})
+			add_custom_command(TARGET OgMaterialEditor POST_BUILD
+				COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+		endif()
 
 		install(TARGETS OgMaterialEditor
 			RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgModularZone/CMakeLists.txt b/Plugins/OgModularZone/CMakeLists.txt
--- a/Plugins/OgModularZone/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgModularZone/CMakeLists.txt	2013-12-19 09:59:54.525200000 +0400
@@ -100,11 +100,17 @@
 
 	if(WIN32)
 		add_custom_command(TARGET OgModularZone PRE_BUILD
-			COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 		add_custom_command(TARGET OgModularZone POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-		add_custom_command(TARGET OgModularZone POST_BUILD
-			COMMAND if exist ..\\..\\RunPath\\Plugins\\OgModularZone_d.dll move ..\\..\\RunPath\\Plugins\\OgModularZone_d.dll ..\\..\\RunPath\\Plugins\\debug)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgModularZone>" ../../RunPath/Plugins)
+		set(FileToCheck "../../RunPath/Plugins/OgModularZone_d.dll")
+		set(MoveToDir "../../RunPath/Plugins/debug")
+		if(EXISTS ${FileToCheck})
+			add_custom_command(TARGET OgModularZone POST_BUILD
+				COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+		endif()
 
 		install(TARGETS OgModularZone
 			RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgQtExample/CMakeLists.txt b/Plugins/OgQtExample/CMakeLists.txt
--- a/Plugins/OgQtExample/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgQtExample/CMakeLists.txt	2013-12-19 10:02:05.705600000 +0400
@@ -76,11 +76,17 @@
 
 	if(WIN32)
 		add_custom_command(TARGET ogQtExample PRE_BUILD
-			COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 		add_custom_command(TARGET ogQtExample POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-		add_custom_command(TARGET ogQtExample POST_BUILD
-			COMMAND if exist ..\\..\\RunPath\\Plugins\\ogQtExample_d.dll move ..\\..\\RunPath\\Plugins\\ogQtExample_d.dll ..\\..\\RunPath\\Plugins\\debug)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:ogQtExample>" ../../RunPath/Plugins)
+		set(FileToCheck "../../RunPath/Plugins/ogQtExample_d.dll")
+		set(MoveToDir "../../RunPath/Plugins/debug")
+		if(EXISTS ${FileToCheck})
+			add_custom_command(TARGET ogQtExample POST_BUILD
+				COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+		endif()
 
 		install(TARGETS ogQtExample
 			RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgScreenEffectEditor/CMakeLists.txt b/Plugins/OgScreenEffectEditor/CMakeLists.txt
--- a/Plugins/OgScreenEffectEditor/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgScreenEffectEditor/CMakeLists.txt	2013-12-19 10:03:41.489600000 +0400
@@ -86,11 +86,17 @@
 
 	if(WIN32)
 		add_custom_command(TARGET OgScreenEffectEditor PRE_BUILD
-			COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 		add_custom_command(TARGET OgScreenEffectEditor POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-		add_custom_command(TARGET OgScreenEffectEditor POST_BUILD
-			COMMAND if exist ..\\..\\RunPath\\Plugins\\OgScreenEffectEditor_d.dll move ..\\..\\RunPath\\Plugins\\OgScreenEffectEditor_d.dll ..\\..\\RunPath\\Plugins\\debug)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgScreenEffectEditor>" ../../RunPath/Plugins)
+		set(FileToCheck "../../RunPath/Plugins/OgScreenEffectEditor_d.dll")
+		set(MoveToDir "../../RunPath/Plugins/debug")
+		if(EXISTS ${FileToCheck})
+			add_custom_command(TARGET OgScreenEffectEditor POST_BUILD
+				COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+		endif()
 
 		install(TARGETS OgScreenEffectEditor
 			RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgScriptEditor/CMakeLists.txt b/Plugins/OgScriptEditor/CMakeLists.txt
--- a/Plugins/OgScriptEditor/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgScriptEditor/CMakeLists.txt	2013-12-19 10:04:49.552400000 +0400
@@ -90,11 +90,17 @@
 
 	if(WIN32)
 		add_custom_command(TARGET OgScriptEditor PRE_BUILD
-			COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 		add_custom_command(TARGET OgScriptEditor POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins)
-		add_custom_command(TARGET OgScriptEditor POST_BUILD
-			COMMAND if exist ..\\..\\RunPath\\Plugins\\OgScriptEditor_d.dll move ..\\..\\RunPath\\Plugins\\OgScriptEditor_d.dll ..\\..\\RunPath\\Plugins\\debug)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgScriptEditor>" ../../RunPath/Plugins)
+		set(FileToCheck "../../RunPath/Plugins/OgScriptEditor_d.dll")
+		set(MoveToDir "../../RunPath/Plugins/debug")
+		if(EXISTS ${FileToCheck})
+			add_custom_command(TARGET OgScriptEditor POST_BUILD
+				COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+				COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+		endif()
 
 		install(TARGETS OgScriptEditor
 			RUNTIME DESTINATION Plugins
diff -Naur a/Plugins/OgSkyX/CMakeLists.txt b/Plugins/OgSkyX/CMakeLists.txt
--- a/Plugins/OgSkyX/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/Plugins/OgSkyX/CMakeLists.txt	2013-12-19 10:06:22.154000000 +0400
@@ -48,9 +48,17 @@
 
 if(WIN32)
 	add_custom_command(TARGET OgSkyX PRE_BUILD
-		COMMAND if not exist ..\\..\\RunPath\\Plugins mkdir ..\\..\\RunPath\\Plugins)
+		COMMAND "${CMAKE_COMMAND}" -E make_directory ../../RunPath/Plugins)
 	add_custom_command(TARGET OgSkyX POST_BUILD
-		COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\Plugins & if exist ..\\..\\RunPath\\Plugins\\OgSkyX_d.dll move ..\\..\\RunPath\\Plugins\\OgSkyX_d.dll ..\\..\\RunPath\\Plugins\\debug)
+		COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:OgSkyX>" ../../RunPath/Plugins)
+	set(FileToCheck "../../RunPath/Plugins/OgSkyX_d.dll")
+	set(MoveToDir "../../RunPath/Plugins/debug")
+	if(EXISTS ${FileToCheck})
+		add_custom_command(TARGET OgSkyX POST_BUILD
+			COMMAND "${CMAKE_COMMAND}" -E make_directory "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E copy "${FileToCheck}" "${MoveToDir}"
+			COMMAND "${CMAKE_COMMAND}" -E remove -f "${FileToCheck}")
+	endif()
 
 	install(TARGETS OgSkyX
 		RUNTIME DESTINATION Plugins
diff -Naur a/qtOfs/CMakeLists.txt b/qtOfs/CMakeLists.txt
--- a/qtOfs/CMakeLists.txt	2013-12-18 22:35:30.000000000 +0400
+++ b/qtOfs/CMakeLists.txt	2013-12-19 10:10:59.412800000 +0400
@@ -73,9 +73,9 @@
 
 	if(WIN32)
 		add_custom_command(TARGET qtOfs PRE_BUILD
-			COMMAND if not exist ..\\RunPath\\bin mkdir ..\\RunPath\\bin)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../RunPath/bin)
 		add_custom_command(TARGET qtOfs POST_BUILD
-			COMMAND copy \"$(TargetPath)\" ..\\RunPath\\bin)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:qtOfs>" ../RunPath/bin)
 	endif(WIN32)
 
 	if(UNIX)
diff -Naur a/qtOgitor/CMakeLists.txt b/qtOgitor/CMakeLists.txt
--- a/qtOgitor/CMakeLists.txt	2013-12-18 22:35:30.000000000 +0400
+++ b/qtOgitor/CMakeLists.txt	2013-12-19 10:10:34.796000000 +0400
@@ -255,9 +255,9 @@
         set_target_properties(qtOgitor PROPERTIES DEBUG_POSTFIX _d)
 		set_property(TARGET qtOgitor PROPERTY WIN32_EXECUTABLE 0)
         add_custom_command(TARGET qtOgitor PRE_BUILD
-            COMMAND if not exist ..\\RunPath\\bin mkdir ..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E make_directory ../RunPath/bin)
         add_custom_command(TARGET qtOgitor POST_BUILD
-            COMMAND copy \"$(TargetPath)\" ..\\RunPath\\bin)
+            COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:qtOgitor>" ../RunPath/bin)
     endif(WIN32)
 	
 	qt5_use_modules(qtOgitor Widgets)
diff -Naur a/qtOgitor/qtpropertybrowser-2.5/CMakeLists.txt b/qtOgitor/qtpropertybrowser-2.5/CMakeLists.txt
--- a/qtOgitor/qtpropertybrowser-2.5/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/qtOgitor/qtpropertybrowser-2.5/CMakeLists.txt	2013-12-19 10:10:27.776000000 +0400
@@ -172,7 +172,7 @@
 
     IF(WIN32)
         add_custom_command( TARGET QtSolutions_PropertyBrowser POST_BUILD
-            COMMAND copy \"$(TargetPath)\" ..\\..\\RunPath\\bin )
+            COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:QtSolutions_PropertyBrowser>" ../../RunPath/bin)
     ENDIF(WIN32)
 
     IF(WIN32)
diff -Naur a/SceneLoader/CMakeLists.txt b/SceneLoader/CMakeLists.txt
--- a/SceneLoader/CMakeLists.txt	2013-12-18 22:35:29.000000000 +0400
+++ b/SceneLoader/CMakeLists.txt	2013-12-19 10:10:45.123200000 +0400
@@ -86,9 +86,9 @@
 
 	if(WIN32)
 		add_custom_command(TARGET SampleLoadApp PRE_BUILD 
-			COMMAND if not exist ..\\RunPath\\bin mkdir ..\\RunPath\\bin)
+			COMMAND "${CMAKE_COMMAND}" -E make_directory ../RunPath/bin)
 		add_custom_command(TARGET SampleLoadApp POST_BUILD 
-			COMMAND copy \"$(TargetPath)\" ..\\RunPath\\bin)
+			COMMAND "${CMAKE_COMMAND}" -E copy "$<TARGET_FILE:SampleLoadApp>" ../RunPath/bin)
 
 		install(FILES ${OGRE_PLUGIN_DIR_REL}/OIS.dll
 			DESTINATION bin
