diff -Naur a/Ogitor/include/OgitorsClipboardManager.h b/Ogitor/include/OgitorsClipboardManager.h
--- a/Ogitor/include/OgitorsClipboardManager.h	2013-12-18 22:35:29.000000000 +0400
+++ b/Ogitor/include/OgitorsClipboardManager.h	2013-12-19 11:05:50.277000000 +0400
@@ -32,6 +32,8 @@
 
 #pragma once
 
+#include "tinyxml.h"
+
 #include "OgitorsSingleton.h"
 
 namespace Ogitors
diff -Naur a/Ogitor/src/OgitorsClipboardManager.cpp b/Ogitor/src/OgitorsClipboardManager.cpp
--- a/Ogitor/src/OgitorsClipboardManager.cpp	2013-12-18 22:35:29.000000000 +0400
+++ b/Ogitor/src/OgitorsClipboardManager.cpp	2013-12-19 11:05:56.111400000 +0400
@@ -31,7 +31,6 @@
 ////////////////////////////////////////////////////////////////////////////////*/
 
 #include "OgitorsPrerequisites.h"
-#include "tinyxml.h"
 #include "OgitorsClipboardManager.h"
 #include "OgitorsRoot.h"
 #include "OgitorsSystem.h"
