From f3eec857361c15f9a685d608fc6ff8855fef0baa Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 29 Nov 2014 18:20:12 +0000
Subject: [PATCH 04/10] Use .dll.a instead of .a for mingw* import libraries

---
 src/lib/qtprofilesetup/qtmoduleinfo.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/qtprofilesetup/qtmoduleinfo.cpp b/src/lib/qtprofilesetup/qtmoduleinfo.cpp
index 7ae49fa..4165805 100644
--- a/src/lib/qtprofilesetup/qtmoduleinfo.cpp
+++ b/src/lib/qtprofilesetup/qtmoduleinfo.cpp
@@ -253,7 +253,7 @@ void QtModuleInfo::setupLibraries(const QtEnvironment &qtEnv, bool debugBuild,
             if (qtEnv.mkspecName.contains(QLatin1String("msvc")))
                 libFilePath += QLatin1String(".lib");
             else if (isMingw)
-                libFilePath += QLatin1String(".a");
+                libFilePath += isStaticLibrary ? QLatin1String(".a") : QLatin1String(".dll.a");
             continue;
         }
         if (!simplifiedLine.startsWith("QMAKE_PRL_LIBS"))
-- 
2.3.5

