From 7f31d5bf93bdbc10ace282b0736d9db68ec4662c Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 28 Apr 2015 23:10:54 +0100
Subject: [PATCH 09/10] win32: Don't specify LIBS += Psapi.lib, just psapi

---
 src/lib/corelib/corelib.pro | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lib/corelib/corelib.pro b/src/lib/corelib/corelib.pro
index 5118cf7..bfb8db9 100644
--- a/src/lib/corelib/corelib.pro
+++ b/src/lib/corelib/corelib.pro
@@ -21,7 +21,8 @@ include(logging/logging.pri)
 include(parser/parser.pri)
 include(tools/tools.pri)
 
-win32:LIBS += Psapi.lib
+win32-msvc*:LIBS += Psapi.lib
+win32-g++:LIBS += -lpsapi
 
 HEADERS += \
     qbs.h
-- 
2.3.5

