--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -4,6 +4,7 @@
 
 nobase_include_HEADERS = \
 	mpcdec/config_types.h \
+	mpcdec/config_win32.h \
 	mpcdec/decoder.h \
 	mpcdec/huffman.h \
 	mpcdec/math.h \
--- a/include/Makefile.in
+++ b/include/Makefile.in
@@ -172,6 +172,7 @@
 AUTOMAKE_OPTIONS = foreign
 nobase_include_HEADERS = \
 	mpcdec/config_types.h \
+	mpcdec/config_win32.h \
 	mpcdec/decoder.h \
 	mpcdec/huffman.h \
 	mpcdec/math.h \
