Index: libmpc-0.1~r459/configure.in
===================================================================
--- libmpc-0.1~r459.orig/configure.in	2010-05-31 16:32:02.551447078 +0200
+++ libmpc-0.1~r459/configure.in	2010-05-31 16:32:09.578951197 +0200
@@ -4,6 +4,7 @@
 AC_CONFIG_SRCDIR([libmpcdec/mpc_reader.c])
 AM_CONFIG_HEADER([include/config.h])
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE
 
 AC_LANG_C
 AC_PROG_CC
