Subject: Add extern keyword to global variable declaration.
Origin: upstream, commit:r479
Bug-Debian: http://bugs.debian.org/665974
---
 libmpcdec/requant.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- libmpc.orig/libmpcdec/requant.h
+++ libmpc/libmpcdec/requant.h
@@ -47,9 +47,9 @@ extern "C" {
 
 
 /* C O N S T A N T S */
-const mpc_uint8_t      Res_bit [18];     ///< Bits per sample for chosen quantizer
-const MPC_SAMPLE_FORMAT __Cc    [1 + 18]; ///< Requantization coefficients
-const mpc_int16_t       __Dc    [1 + 18]; ///< Requantization offset
+extern const mpc_uint8_t      Res_bit [18];     ///< Bits per sample for chosen quantizer
+extern const MPC_SAMPLE_FORMAT __Cc    [1 + 18]; ///< Requantization coefficients
+extern const mpc_int16_t       __Dc    [1 + 18]; ///< Requantization offset
 
 #define Cc (__Cc + 1)
 #define Dc (__Dc + 1)
