Description: avoid automake error
Author: Hideki Yamane <henrich@debian.org>
Forwarded: no
Last-Update: 2013-10-23

--- libmpc-0.1~r459.orig/configure.in
+++ libmpc-0.1~r459/configure.in
@@ -3,7 +3,7 @@ AC_CONFIG_AUX_DIR([config])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([libmpcdec/mpc_reader.c])
 AM_CONFIG_HEADER([include/config.h])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE(subdir-objects)
 AM_MAINTAINER_MODE
 
 AC_LANG_C
