--- sox/src/formats.c.orig	2021-04-16 20:44:08.479849800 +0200
+++ sox/src/formats.c	2021-04-16 20:49:01.077926700 +0200
@@ -464,7 +464,7 @@
   fp->_r += PIPE_AUTO_DETECT_SIZE;
 #elif defined __GLIBC__
   fp->_IO_read_ptr = fp->_IO_read_base;
-#elif defined _MSC_VER && _MSC_VER >= 1900
+#elif (defined _MSC_VER && _MSC_VER >= 1900) || defined _UCRT
   #define NO_REWIND_PIPE
 #elif defined _MSC_VER || defined _WIN32 || defined _WIN64 || \
       defined _ISO_STDIO_ISO_H || defined __sgi
