--- netcdf-4.4.1.1/CMakeLists.txt.orig	2017-02-01 08:16:34.358615500 +0300
+++ netcdf-4.4.1.1/CMakeLists.txt	2017-02-01 08:20:43.083093300 +0300
@@ -163,7 +163,7 @@
 
 # Set Memory test program for non-MSVC based builds.
 # Assume valgrind for now.
-IF(NOT MSVC)
+IF(NOT WIN32)
   SET(CTEST_MEMORYCHECK_COMMAND valgrind CACHE STRING "")
 ENDIF()
 
@@ -245,7 +245,7 @@
 
 # Suppress CRT Warnings.
 # Only necessary for Windows
-IF(MSVC)
+IF(WIN32)
   ADD_DEFINITIONS(-D_CRT_SECURE_NO_WARNINGS)
 ENDIF()
 
@@ -267,7 +267,7 @@
 # Configuration for post-install RPath
 # Adapted from http://www.cmake.org/Wiki/CMake_RPATH_handling
 ##
-IF(NOT MSVC)
+IF(NOT WIN32)
   # use, i.e. don't skip the full RPATH for the build tree
   SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
 
@@ -839,7 +839,7 @@
   SET(NC_CTEST_DROP_LOC_PREFIX "" CACHE STRING "Prefix for Dashboard location on remote server when using CTest-based testing.")
 
   FIND_PROGRAM(HOSTNAME_CMD NAMES hostname)
-  IF(NOT MSVC)
+  IF(NOT WIN32)
     SET(HOSTNAME_ARG "-s")
   ENDIF()
   IF(HOSTNAME_CMD)
