--- netcdf-c-4.10.1/include/nc4dispatch.h.orig	2026-07-08 06:08:43.000000000 +0800
+++ netcdf-c-4.10.1/include/nc4dispatch.h	2026-09-16 15:10:58.857428000 +0800
@@ -119,12 +119,12 @@
                  const size_t *start, const size_t *count,
                  void *value, nc_type);
 
-    extern int
+    EXTERNL int
     NC4_put_vars(int ncid, int varid,
                  const size_t *start, const size_t *count, const ptrdiff_t* stride,
                  const void *value, nc_type);
 
-    extern int
+    EXTERNL int
     NC4_get_vars(int ncid, int varid,
                  const size_t *start, const size_t *count, const ptrdiff_t* stride,
                  void *value, nc_type);
--- netcdf-c-4.10.1/libsrc4/nc4internal.c.orig	2026-07-08 06:08:43.000000000 +0800
+++ netcdf-c-4.10.1/libsrc4/nc4internal.c	2026-09-16 15:10:58.858428100 +0800
@@ -24,6 +24,7 @@
 #include "nc4internal.h"
 #include "nc.h" /* from libsrc */
 #include "ncdispatch.h" /* from libdispatch */
+#include "nc4dispatch.h" /* for the EXTERNL declarations of the NC4_* functions defined here */
 #include "ncutf8.h"
 #include "ncrc.h"
 #include "nc4internal.h"
