--- netcdf-4.6.0/libdispatch/dutil.c.old	2018-01-25 18:32:02.000000000 +0100
+++ netcdf-4.6.0/libdispatch/dutil.c	2018-02-28 18:22:09.019723300 +0100
@@ -225,7 +225,7 @@
 	_mktemp_s(tmp,sizeof(tmp)-1);
 #else /*!_MSC_VER*/
         mktemp(tmp);
-	tmo[sizeof[tmp]-1] = '\0';
+	tmp[sizeof(tmp)-1] = '\0';
 #endif
 #else /* !HAVE_MKTEMP */
 	/* Need to simulate by using some kind of pseudo-random number */
