--- netcdf-4.6.0/include/ncconfigure.h.orig	2018-03-05 20:39:39.189518100 +0100
+++ netcdf-4.6.0/include/ncconfigure.h	2018-03-05 20:44:07.802389700 +0100
@@ -47,7 +47,16 @@
 #endif
 
 #ifndef strlcat
+#if defined(DLL_NETCDF)
+# if defined(DLL_EXPORT)
+#  define GTOPT_EXTRA __declspec(dllexport)
+# else
+#  define GTOPT_EXTRA __declspec(dllimport)
+# endif
+GTOPT_EXTRA extern size_t strlcat(char* dst, const char* src, size_t dsize);
+#else
 extern size_t strlcat(char*,const char*,size_t);
+#endif
 #endif
 
 #ifndef snprintf
