--- netcdf-4.6.0/ncgen/cvt.c.orig	2018-03-13 21:50:55.762607800 +0100
+++ netcdf-4.6.0/ncgen/cvt.c	2018-03-13 21:52:12.171941700 +0100
@@ -18,7 +18,7 @@
     Constvalue tmp;
     unsigned char* bytes = NULL;
     size_t bytelen;
-#ifdef _MSC_VER
+#ifdef HAVE_WIN_SSCANF
     int byteval;
 #endif
     dst->lineno = src->lineno;
@@ -407,7 +407,7 @@
     break;
 
 /* Conversion of a string to e.g. an integer should be what?*/
-#ifdef _MSC_VER
+#ifdef HAVE_WIN_SSCANF
 case CASE(NC_STRING,NC_BYTE):
     sscanf(src->value.stringv.stringv,"%d",&byteval); tmp.int8v = (char)byteval; break;
 case CASE(NC_STRING,NC_UBYTE):
