--- netcdf-c-4.6.2/libdispatch/dwinpath.c.orig	2018-12-20 14:10:40.029978200 +0300
+++ netcdf-c-4.6.2/libdispatch/dwinpath.c	2018-12-20 14:10:42.947183300 +0300
@@ -14,8 +14,9 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#ifdef _MSC_VER
+#ifdef _WIN32
 #include <io.h>
+#include <errno.h>
 #endif
 
 #include "ncexternl.h"
@@ -214,7 +215,7 @@
     int status = 0;
     char* cvtname = NCpathcvt(path);
     if(cvtname == NULL) return -1;
-#ifdef _MSC_VER
+#ifdef _WIN32
     status = _access(cvtname,mode);
 #else
     status = access(cvtname,mode);
