--- netcdf-c-4.7.0/include/XGetopt.h.orig	2019-06-02 21:50:20.186839000 +0300
+++ netcdf-c-4.7.0/include/XGetopt.h	2019-06-02 21:50:23.891050800 +0300
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <tchar.h>
 
+#if !defined(__GETOPT_H__)
 #if defined(DLL_NETCDF)
 # if defined(DLL_EXPORT)
 #  define GTOPT_EXTRA __declspec(dllexport)
@@ -35,5 +35,6 @@
 extern TCHAR *optarg;
 
 int getopt(int argc, TCHAR *argv[], TCHAR *optstring);
+#endif
 
 #endif //XGETOPT_H
