--- netcdf-c-4.7.4/libhdf5/hdf5var.c.orig	2020-04-28 14:00:45.167090100 +0300
+++ netcdf-c-4.7.4/libhdf5/hdf5var.c	2020-04-28 14:00:50.414367600 +0300
@@ -2290,7 +2290,7 @@
     float real_preemption = CHUNK_CACHE_PREEMPTION;
 
     LOG((1, "%s: ncid 0x%x varid %d size %d nelems %d preemption %d",
-	 __func__, ncid, varid, size, nelems, preemptions));
+	 __func__, ncid, varid, size, nelems, preemption));
     
     if (size >= 0)
         real_size = ((size_t) size) * MEGABYTE;
