--- a/tools/meshdebug/CMakeLists.txt
+++ b/tools/meshdebug/CMakeLists.txt
@@ -15,3 +15,7 @@
         Qt::Quick3DAssetImportPrivate
 )
 qt_internal_return_unless_building_tools()
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "meshdebug-qt6"
+)
--- a/tools/balsam/CMakeLists.txt
+++ b/tools/balsam/CMakeLists.txt
@@ -16,3 +16,7 @@
         Qt::Quick3DIblBakerPrivate
 )
 qt_internal_return_unless_building_tools()
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME "balsam-qt6"
+)
