--- a/postgis/gserialized_estimate.c
+++ b/postgis/gserialized_estimate.c
@@ -1321,7 +1321,6 @@
 *
 * joinsel = estimated_nrows / (totalrows1 * totalrows2)
 */
-PG_FUNCTION_INFO_V1(gserialized_gist_joinsel);
 Datum gserialized_gist_joinsel(PG_FUNCTION_ARGS)
 {
 	PlannerInfo *root = (PlannerInfo *) PG_GETARG_POINTER(0);
@@ -2270,7 +2269,6 @@
 	return selectivity;
 }
 
-PG_FUNCTION_INFO_V1(gserialized_gist_sel);
 Datum gserialized_gist_sel(PG_FUNCTION_ARGS)
 {
 	PlannerInfo *root = (PlannerInfo *) PG_GETARG_POINTER(0);
--- a/postgis/lwgeom_geos.c
+++ b/postgis/lwgeom_geos.c
@@ -841,7 +841,6 @@
  *      'POLYGON((0 0, 10 0, 10 10, 0 10, 0 0))',
  *      'POLYGON((5 5, 15 5, 15 7, 5 7, 5 5))');
  */
-PG_FUNCTION_INFO_V1(ST_SymDifference);
 Datum ST_SymDifference(PG_FUNCTION_ARGS)
 {
 	GSERIALIZED *geom1;
