diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8115507..d1f74ee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,6 +99,7 @@ option(OPTION_VERBOSE "Verbose build " OFF)
 option(SIMVOLEON_BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(SIMVOLEON_BUILD_DOCUMENTATION "Build and install API documentation (requires Doxygen)." OFF)
 option(SIMVOLEON_BUILD_TESTS "Build test code" OFF)
+option(SIMVOLEON_USE_CPACK "If enabled the cpack subrepo is mandatory" OFF)
 cmake_dependent_option(SIMVOLEON_BUILD_INTERNAL_DOCUMENTATION "Document internal code not part of the API." OFF "SIMVOLEON_BUILD_DOCUMENTATION" OFF)
 cmake_dependent_option(SIMVOLEON_BUILD_DOC_MAN "Build So${Gui} man pages." OFF "SIMVOLEON_BUILD_DOCUMENTATION" OFF)
 cmake_dependent_option(SIMVOLEON_BUILD_DOC_QTHELP "Build QtHelp documentation." OFF "SIMVOLEON_BUILD_DOCUMENTATION" OFF)
@@ -228,4 +229,6 @@ endif()
 
 ############################################################################
 # New CPACK section, please see the README file inside cpack.d directory.
-add_subdirectory(cpack.d)
+if(SIMVOLEON_USE_CPACK)
+  add_subdirectory(cpack.d)
+endif()
