--- a/configure.ac
+++ b/configure.ac
@@ -1915,10 +1915,10 @@
 if test "x$with_gdiplus" != "xyes"; then
   use_gdiplus="No (disabled by default - Windows only)"
 else
-  if test -f "$PLATFORMSDKINCLUDE\GdiPlus.h" -a -f "$PLATFORMSDKLIB\GdiPlus.lib"; then
+  if test -f "$PLATFORMSDKINCLUDE\gdiplus.h" -a -f "$PLATFORMSDKLIB\libgdiplus.a"; then
     use_gdiplus="Yes"
     AC_DEFINE_UNQUOTED(HAVE_GDIPLUS,1,[Define if you have the GDI+ framework for Windows])
-    GDIPLUS_HEADS=`cd "$PLATFORMSDKINCLUDE" && echo GdiPlus*.h`
+    GDIPLUS_HEADS=`cd "$PLATFORMSDKINCLUDE" && echo gdiplus*.h`
     GDIPLUS_CFLAGS=''
     GDIPLUS_LIBS=''
     AC_SUBST([GDIPLUS_HEADS])
--- a/plugin/gdiplus/Makefile.am
+++ b/plugin/gdiplus/Makefile.am
@@ -20,7 +20,7 @@
 
 noinst_HEADERS = FileStream.h gvplugin_gdiplus.h
 
-nodist_libgvplugin_gdiplus_C_la_SOURCES = GdiPlus*.h
+nodist_libgvplugin_gdiplus_C_la_SOURCES = gdiplus*.h
 libgvplugin_gdiplus_C_la_SOURCES = \
 	FileStream.cpp \
 	gvdevice_gdiplus.cpp \
@@ -30,8 +30,8 @@
 	gvtextlayout_gdiplus.cpp
 
 # libtool doesn't want to run with actual Windows import libs, so we force GdiPlus.lib through to the linker
-libgvplugin_gdiplus_la_LDFLAGS = -version-info $(GVPLUGIN_VERSION_INFO) -Wl,"$(PLATFORMSDKLIB)\GdiPlus.lib"
-nodist_libgvplugin_gdiplus_la_SOURCES = GdiPlus*.h
+libgvplugin_gdiplus_la_LDFLAGS = -version-info $(GVPLUGIN_VERSION_INFO) -Wl,"$(PLATFORMSDKLIB)\libgdiplus.a"
+nodist_libgvplugin_gdiplus_la_SOURCES = gdiplus*.h
 libgvplugin_gdiplus_la_SOURCES = $(libgvplugin_gdiplus_C_la_SOURCES)
 libgvplugin_gdiplus_la_LIBADD = -lgdi32 -lole32 -luuid $(top_builddir)/lib/gvc/libgvc.la
 
