--- a/tclpkg/gv/Makefile.am
+++ b/tclpkg/gv/Makefile.am
@@ -179,7 +179,7 @@
 nodist_libgv_python3_la_SOURCES = gv_python3.cpp $(PYTHON3_data)
 libgv_python3_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_python3_la_LIBADD = $(BASELIBS) $(PYTHON3_LIBS)
-libgv_python3_la_LDFLAGS = -module -avoid-version
+libgv_python3_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_python3_la_CPPFLAGS = $(BASECPPFLAGS) $(PYTHON3_INCLUDES)
 $(PYTHON3_data): gv_python3.cpp
 gv_python3.cpp: gv.i
