diff -Naur graphviz-2.40.1-orig/cmd/dot/dot.c graphviz-2.40.1/cmd/dot/dot.c
--- graphviz-2.40.1-orig/cmd/dot/dot.c	2016-12-14 08:45:02.000000000 +0300
+++ graphviz-2.40.1/cmd/dot/dot.c	2017-03-16 12:00:22.040710500 +0300
@@ -20,7 +20,7 @@
 #include "gvc.h"
 #include "gvio.h"
 
-#ifdef WIN32_DLL
+#ifdef _MSC_VER
 __declspec(dllimport) boolean MemTest;
 __declspec(dllimport) int GvExitOnUsage;
 /*gvc.lib cgraph.lib*/
diff -Naur graphviz-2.40.1-orig/cmd/edgepaint/edgepaintmain.c graphviz-2.40.1/cmd/edgepaint/edgepaintmain.c
--- graphviz-2.40.1-orig/cmd/edgepaint/edgepaintmain.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/cmd/edgepaint/edgepaintmain.c	2017-03-16 12:06:07.578259600 +0300
@@ -10,7 +10,7 @@
 
 #include "config.h"
 
-#ifdef WIN32 /*dependencies*/
+#ifdef _MSC_VER /*dependencies*/
     #pragma comment( lib, "cgraph.lib" )
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "ingraphs.lib" )
diff -Naur graphviz-2.40.1-orig/cmd/gvmap/cluster.c graphviz-2.40.1/cmd/gvmap/cluster.c
--- graphviz-2.40.1-orig/cmd/gvmap/cluster.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/cmd/gvmap/cluster.c	2017-03-16 12:06:36.067221900 +0300
@@ -31,7 +31,7 @@
 #include "DotIO.h"
 #include "colorutil.h"
 
-#ifdef WIN32 /*dependencies*/
+#ifdef _MSC_VER /*dependencies*/
     #pragma comment( lib, "cgraph.lib" )
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "ingraphs.lib" )
diff -Naur graphviz-2.40.1-orig/cmd/gvmap/gvmap.c graphviz-2.40.1/cmd/gvmap/gvmap.c
--- graphviz-2.40.1-orig/cmd/gvmap/gvmap.c	2016-10-29 07:44:55.000000000 +0300
+++ graphviz-2.40.1/cmd/gvmap/gvmap.c	2017-03-16 12:06:53.633995900 +0300
@@ -40,7 +40,7 @@
 enum {MAX_GRPS = 10000};
 static char swork[maxlen];
 
-#ifdef WIN32
+#ifdef _MSC_VER
     #pragma comment( lib, "cgraph.lib" )
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "ingraphs.lib" )
diff -Naur graphviz-2.40.1-orig/lib/common/geomprocs.h graphviz-2.40.1/lib/common/geomprocs.h
--- graphviz-2.40.1-orig/lib/common/geomprocs.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/geomprocs.h	2017-03-15 12:24:49.724951600 +0300
@@ -24,13 +24,15 @@
 
 #include "geom.h"
 
-#ifdef _WIN32
 #ifdef GVDLL
 #define extern __declspec(dllexport)
-#else
+#elif defined(_MSC_VER)
+#ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
-#endif	
+#else
+#define extern
+#endif
 
 extern box mkbox(point p, point q);
 extern boxf mkboxf(pointf p, pointf q);
diff -Naur graphviz-2.40.1-orig/lib/common/globals.h graphviz-2.40.1/lib/common/globals.h
--- graphviz-2.40.1-orig/lib/common/globals.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/globals.h	2017-03-15 12:24:11.419760600 +0300
@@ -38,21 +38,21 @@
 #if !defined(_BLD_dotneato) && defined(__IMPORT__)
 #   define external __IMPORT__
 #endif
-#if defined(GVDLL)
+
+#ifdef GVDLL
 #if !defined(_BLD_gvc)
-#define extern	__declspec(dllimport)
+#define extern __declspec(dllimport)
 #else
 #define extern __declspec(dllexport)
 #endif
-#endif
-#endif
-/*visual studio*/
-#ifdef _WIN32
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
+#endif
 #endif
-/*end visual studio*/
 
 #ifndef external
 #   define external   extern
diff -Naur graphviz-2.40.1-orig/lib/common/memory.h graphviz-2.40.1/lib/common/memory.h
--- graphviz-2.40.1-orig/lib/common/memory.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/memory.h	2017-03-15 12:14:33.349158300 +0300
@@ -23,6 +23,16 @@
 extern "C" {
 #endif
 
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
+#ifndef GVC_EXPORTS
+#define extern __declspec(dllimport)
+#endif
+#else
+#define extern
+#endif
+
 #ifdef DMALLOC
 #define NEW(t)           (t*)calloc(1,sizeof(t))
 #define N_NEW(n,t)       (t*)calloc((n),sizeof(t))
@@ -42,16 +52,6 @@
 #define RALLOC(size,ptr,type) ((type*)grealloc(ptr,(size)*sizeof(type)))
 #define ZALLOC(size,ptr,type,osize) (ptr? (type*)zrealloc(ptr,size,sizeof(type),osize):(type*)zmalloc((size)*sizeof(type)))
 #endif
-#ifdef GVDLL
-#define extern __declspec(dllexport)
-#else
-#ifdef _WIN32
-#ifndef GVC_EXPORTS
-#define extern __declspec(dllimport)
-#endif
-#endif
-
-#endif
 
     extern void *zmalloc(size_t);
     extern void *zrealloc(void *, size_t, size_t, size_t);
diff -Naur graphviz-2.40.1-orig/lib/common/pointset.h graphviz-2.40.1/lib/common/pointset.h
--- graphviz-2.40.1-orig/lib/common/pointset.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/pointset.h	2017-03-15 12:22:42.156655100 +0300
@@ -23,20 +23,16 @@
 
     typedef Dict_t PointSet;
     typedef Dict_t PointMap;
+
 #ifdef GVDLL
 #define extern __declspec(dllexport)
-#else
-#define extern
-#endif
-
-/*visual studio*/
-#ifdef _WIN32
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
-#undef extern
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
 	extern PointSet *newPS(void);
     extern void freePS(PointSet *);
diff -Naur graphviz-2.40.1-orig/lib/common/render.h graphviz-2.40.1/lib/common/render.h
--- graphviz-2.40.1-orig/lib/common/render.h	2016-12-22 08:44:41.000000000 +0300
+++ graphviz-2.40.1/lib/common/render.h	2017-03-15 12:21:40.252114300 +0300
@@ -58,13 +58,15 @@
 
     typedef void (*nodesizefn_t) (Agnode_t *, boolean);
 
-/*visual studio*/
-#ifdef _WIN32
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
 	extern void add_box(path *, boxf);
     extern void arrow_flags(Agedge_t * e, int *sflag, int *eflag);
diff -Naur graphviz-2.40.1-orig/lib/common/utils.h graphviz-2.40.1/lib/common/utils.h
--- graphviz-2.40.1-orig/lib/common/utils.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/utils.h	2017-03-15 12:22:05.404553000 +0300
@@ -19,13 +19,15 @@
 #endif
 
 
-/*visual studio*/
-#ifdef _WIN32
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
 #ifndef HAVE_STRCASECMP
     extern int strcasecmp(const char *s1, const char *s2);
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvc.h graphviz-2.40.1/lib/gvc/gvc.h
--- graphviz-2.40.1-orig/lib/gvc/gvc.h	2016-12-18 15:58:37.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvc.h	2017-03-15 12:22:21.932498300 +0300
@@ -23,18 +23,13 @@
 
 #ifdef GVDLL
 #define extern __declspec(dllexport)
-#else
-#define extern
-#endif
-
-/*visual studio*/
-#ifdef WIN32
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
-#undef extern
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 	
 #define LAYOUT_DONE(g) (agbindrec(g, "Agraphinfo_t", 0, TRUE) && GD_drawing(g))
 
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvcext.h graphviz-2.40.1/lib/gvc/gvcext.h
--- graphviz-2.40.1-orig/lib/gvc/gvcext.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvcext.h	2017-03-15 11:12:58.636377400 +0300
@@ -64,7 +64,7 @@
     typedef struct gvplugin_available_s gvplugin_available_t;
 
 /*visual studio*/
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 __declspec(dllimport) lt_symlist_t lt_preloaded_symbols[];
 #else
@@ -77,11 +77,11 @@
 /*end visual studio*/
 
 
-#ifndef WIN32
+#ifndef _MSC_VER
+#if !defined(LTDL_H)
 #if defined(GVDLL)
 	__declspec(dllexport) lt_symlist_t lt_preloaded_symbols[];
 #else
-#if !defined(LTDL_H)
 	extern lt_symlist_t lt_preloaded_symbols[];
 #endif
 #endif
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvcint.h graphviz-2.40.1/lib/gvc/gvcint.h
--- graphviz-2.40.1-orig/lib/gvc/gvcint.h	2017-03-15 14:50:51.528603900 +0300
+++ graphviz-2.40.1/lib/gvc/gvcint.h	2017-03-15 14:51:13.207271100 +0300
@@ -24,6 +24,16 @@
 #include "gvcommon.h"
 #include "color.h"
 
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
+#ifndef VMALLOC_EXPORTS
+#define extern __declspec(dllimport)
+#endif
+#else
+#define extern
+#endif
+
     /* active plugin headers */
     typedef struct gvplugin_active_layout_s {
         gvlayout_engine_t *engine;
@@ -146,6 +156,8 @@
 extern GVC_t* gvCloneGVC (GVC_t *);
 extern void gvFreeCloneGVC (GVC_t *);
 
+#undef extern
+
 #ifdef WIN32
 #define DIRSEP "\\"
 #else
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvio.h graphviz-2.40.1/lib/gvc/gvio.h
--- graphviz-2.40.1-orig/lib/gvc/gvio.h	2016-09-08 07:45:03.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvio.h	2017-03-15 12:26:35.091978200 +0300
@@ -22,18 +22,13 @@
 
 #ifdef GVDLL
 #define extern __declspec(dllexport)
-#else
-#define extern
-#endif
-
-/*visual studio*/
-#ifdef WIN32
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
-#undef extern
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
     extern size_t gvwrite (GVJ_t * job, const char *s, size_t len);
     extern size_t gvfwrite (const void *ptr, size_t size, size_t nmemb, FILE *stream);
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvplugin_loadimage.h graphviz-2.40.1/lib/gvc/gvplugin_loadimage.h
--- graphviz-2.40.1-orig/lib/gvc/gvplugin_loadimage.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvplugin_loadimage.h	2017-03-15 12:27:00.020404000 +0300
@@ -23,16 +23,14 @@
 #endif
 
 #ifdef GVDLL
-#  define extern __declspec(dllexport)
-#endif
-
-/*visual studio*/
-#ifdef WIN32
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
 extern boolean gvusershape_file_access(usershape_t *us);
 extern void gvusershape_file_release(usershape_t *us);
diff -Naur graphviz-2.40.1-orig/lib/gvpr/gprstate.c graphviz-2.40.1/lib/gvpr/gprstate.c
--- graphviz-2.40.1-orig/lib/gvpr/gprstate.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvpr/gprstate.c	2017-03-16 12:10:00.361932900 +0300
@@ -17,7 +17,7 @@
  *
  */
 
-#ifdef WIN32
+#ifdef _WIN32
 #include "windows.h"
 #include "shlwapi.h"
 #endif
@@ -139,7 +139,7 @@
     free (state);
 }
 
-#ifdef WIN32_DLL
+#if defined(WIN32_DLL) || defined(GVDLL)
 int pathisrelative (char* path)
 {
     return PathIsRelative(path);
diff -Naur graphviz-2.40.1-orig/lib/gvpr/gvpr.c graphviz-2.40.1/lib/gvpr/gvpr.c
--- graphviz-2.40.1-orig/lib/gvpr/gvpr.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/lib/gvpr/gvpr.c	2017-03-16 12:05:22.617541200 +0300
@@ -219,7 +219,7 @@
     Sfio_t *pathp = NULL;
     size_t sz;
 
-#ifdef WIN32_DLL
+#if defined(WIN32_DLL) || defined(GVDLL)
     if (!pathisrelative(arg))
 #else
     if (strchr(arg, '/'))
diff -Naur graphviz-2.40.1-orig/lib/gvpr/Makefile.am graphviz-2.40.1/lib/gvpr/Makefile.am
--- graphviz-2.40.1-orig/lib/gvpr/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvpr/Makefile.am	2017-03-16 12:11:43.357174300 +0300
@@ -51,7 +51,7 @@
 libgvpr_la_DEPENDENCIES = $(libgvpr_C_la_DEPENDENCIES)
 
 if WITH_WIN32
-libgvpr_la_LDFLAGS += -export-symbols-regex gvpr
+libgvpr_la_LDFLAGS += -export-symbols-regex gvpr -lshlwapi
 endif
 
 mkdefs_SOURCES = mkdefs.c
diff -Naur graphviz-2.40.1-orig/lib/pathplan/pathutil.h graphviz-2.40.1/lib/pathplan/pathutil.h
--- graphviz-2.40.1-orig/lib/pathplan/pathutil.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/pathplan/pathutil.h	2017-03-15 12:27:39.732675500 +0300
@@ -31,13 +31,16 @@
 #define TRUE	(NOT(FALSE))
 #endif
 
-/*visual studio*/
-#ifdef _WIN32
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef PATHPLAN_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
+
 	typedef double COORD;
     extern COORD area2(Ppoint_t, Ppoint_t, Ppoint_t);
     extern int wind(Ppoint_t a, Ppoint_t b, Ppoint_t c);
diff -Naur graphviz-2.40.1-orig/lib/pathplan/vis.h graphviz-2.40.1/lib/pathplan/vis.h
--- graphviz-2.40.1-orig/lib/pathplan/vis.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/pathplan/vis.h	2017-03-15 12:28:57.877145100 +0300
@@ -46,12 +46,17 @@
 	/* this is computed from the above */
 	array2 vis;
     };
-#ifdef _WIN32
+
+
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef PATHPLAN_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
 	extern COORD *ptVis(vconfig_t *, int, Ppoint_t);
     extern int directVis(Ppoint_t, int, Ppoint_t, int, vconfig_t *);
diff -Naur graphviz-2.40.1-orig/lib/pack/pack.h graphviz-2.40.1/lib/pack/pack.h
--- graphviz-2.40.1-orig/lib/pack/pack.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/pack/pack.h	2017-03-15 13:56:32.529203100 +0300
@@ -57,13 +57,15 @@
 	int flags;       
     } pack_info;
 
-/*visual studio*/
-#ifdef WIN32
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/
 
     extern point *putRects(int ng, boxf* bbs, pack_info* pinfo);
     extern int packRects(int ng, boxf* bbs, pack_info* pinfo);
diff -Naur graphviz-2.40.1-orig/lib/vmalloc/vmalloc.h graphviz-2.40.1/lib/vmalloc/vmalloc.h
--- graphviz-2.40.1-orig/lib/vmalloc/vmalloc.h	2016-12-22 08:44:41.000000000 +0300
+++ graphviz-2.40.1/lib/vmalloc/vmalloc.h	2017-03-15 13:58:19.217305300 +0300
@@ -109,16 +109,16 @@
 #define extern	__IMPORT__
 #endif
 
-/*visual studio*/
-#ifdef WIN32
-#undef extern
+#ifdef GVDLL
+#define extern __declspec(dllexport)
+#elif defined(_MSC_VER)
 #ifndef VMALLOC_EXPORTS
 #define extern __declspec(dllimport)
-#else
-#define extern __declspec(dllexport)
 #endif
+#else
+#define extern
 #endif
-/*end visual studio*/		 
+
 		 extern Vmethod_t *Vmbest;	/* best allocation              */
     extern Vmethod_t *Vmlast;	/* last-block allocation        */
     extern Vmethod_t *Vmpool;	/* pool allocation              */
diff -Naur graphviz-2.40.1-orig/plugin/core/gvplugin_core.c graphviz-2.40.1/plugin/core/gvplugin_core.c
--- graphviz-2.40.1-orig/plugin/core/gvplugin_core.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/core/gvplugin_core.c	2017-03-16 12:03:00.226254500 +0300
@@ -88,7 +88,7 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_CORE_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_core_LTX_library = { "core", apis };
 #else
@@ -98,7 +98,7 @@
 
 
 
-#ifndef WIN32_DLL
+#ifndef _MSC_VER
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_core_LTX_library = { "core", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/core/gvrender_core_dot.c graphviz-2.40.1/plugin/core/gvrender_core_dot.c
--- graphviz-2.40.1-orig/plugin/core/gvrender_core_dot.c	2016-12-05 08:43:27.000000000 +0300
+++ graphviz-2.40.1/plugin/core/gvrender_core_dot.c	2017-03-16 08:37:04.823805300 +0300
@@ -46,7 +46,7 @@
 	FORMAT_XDOT14,
 } format_type;
 
-#ifdef WIN32 /*dependencies*/
+#ifdef _MSC_VER /*dependencies*/
     #pragma comment( lib, "cgraph.lib" )
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "xdot.lib" )
diff -Naur graphviz-2.40.1-orig/plugin/dot_layout/gvlayout_dot_layout.c graphviz-2.40.1/plugin/dot_layout/gvlayout_dot_layout.c
--- graphviz-2.40.1-orig/plugin/dot_layout/gvlayout_dot_layout.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/plugin/dot_layout/gvlayout_dot_layout.c	2017-03-16 08:26:11.984929800 +0300
@@ -18,7 +18,7 @@
 
 typedef enum { LAYOUT_DOT, } layout_type;
 
-#ifdef WIN32 /*dependencies*/
+#ifdef _MSC_VER /*dependencies*/
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "ingraphs.lib" )
     #pragma comment( lib, "cdt.lib" )
diff -Naur graphviz-2.40.1-orig/plugin/dot_layout/gvplugin_dot_layout.c graphviz-2.40.1/plugin/dot_layout/gvplugin_dot_layout.c
--- graphviz-2.40.1-orig/plugin/dot_layout/gvplugin_dot_layout.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/dot_layout/gvplugin_dot_layout.c	2017-03-16 12:02:36.994886200 +0300
@@ -20,7 +20,7 @@
     {(api_t)0, 0},
 };
 /*visual studio*/
-#ifdef WIN32_DLL
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_DOT_LAYOUT_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_dot_layout_LTX_library = { "dot_layout", apis };
 #else
@@ -33,7 +33,7 @@
 /*end visual studio*/
 
 
-#ifndef WIN32_DLL
+#ifndef _MSC_VER
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_dot_layout_LTX_library = { "dot_layout", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/dot_layout/Makefile.am graphviz-2.40.1/plugin/dot_layout/Makefile.am
--- graphviz-2.40.1-orig/plugin/dot_layout/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/dot_layout/Makefile.am	2017-03-16 08:28:18.601173900 +0300
@@ -22,7 +22,8 @@
 	gvplugin_dot_layout.c \
 	gvlayout_dot_layout.c
 libgvplugin_dot_layout_C_la_LIBADD = \
-	$(top_builddir)/lib/dotgen/libdotgen_C.la
+	$(top_builddir)/lib/dotgen/libdotgen_C.la \
+	$(top_builddir)/lib/ortho/libortho_C.la
 
 libgvplugin_dot_layout_la_LDFLAGS = -version-info $(GVPLUGIN_VERSION_INFO)
 libgvplugin_dot_layout_la_SOURCES = $(libgvplugin_dot_layout_C_la_SOURCES)
diff -Naur graphviz-2.40.1-orig/plugin/gd/gvdevice_gd.c graphviz-2.40.1/plugin/gd/gvdevice_gd.c
--- graphviz-2.40.1-orig/plugin/gd/gvdevice_gd.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/plugin/gd/gvdevice_gd.c	2017-03-16 08:36:36.616150400 +0300
@@ -14,7 +14,7 @@
 #include "config.h"
 
 
-#ifdef WIN32 /*dependencies*/
+#ifdef _MSC_VER /*dependencies*/
     #pragma comment( lib, "cgraph.lib" )
     #pragma comment( lib, "libpng12.lib" )
     #pragma comment( lib, "libgd.lib" )
diff -Naur graphviz-2.40.1-orig/plugin/gdk/gvloadimage_gdk.c graphviz-2.40.1/plugin/gdk/gvloadimage_gdk.c
--- graphviz-2.40.1-orig/plugin/gdk/gvloadimage_gdk.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/plugin/gdk/gvloadimage_gdk.c	2017-03-16 08:35:44.311849500 +0300
@@ -23,7 +23,7 @@
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <gdk/gdkcairo.h>
 
-#ifdef WIN32 //*dependencies
+#ifdef _MSC_VER //*dependencies
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "glib-2.0.lib" )
     #pragma comment( lib, "cairo.lib" )
diff -Naur graphviz-2.40.1-orig/plugin/neato_layout/gvlayout_neato_layout.c graphviz-2.40.1/plugin/neato_layout/gvlayout_neato_layout.c
--- graphviz-2.40.1-orig/plugin/neato_layout/gvlayout_neato_layout.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/plugin/neato_layout/gvlayout_neato_layout.c	2017-03-16 08:34:30.284909100 +0300
@@ -27,7 +27,7 @@
 #include "globals.h"
 
 
-#ifdef WIN32 //*dependencies
+#ifdef _MSC_VER //*dependencies
     #pragma comment( lib, "cgraph.lib" )
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "pathplan.lib" )
diff -Naur graphviz-2.40.1-orig/plugin/gd/gvplugin_gd.c graphviz-2.40.1/plugin/gd/gvplugin_gd.c
--- graphviz-2.40.1-orig/plugin/gd/gvplugin_gd.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gd/gvplugin_gd.c	2017-03-16 12:02:03.025335200 +0300
@@ -32,7 +32,7 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL /*visual studio*/
+#ifdef _MSC_VER /*visual studio*/
 #ifndef GVPLUGIN_GD_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_gd_LTX_library = { "gd", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/gdiplus/gvplugin_gdiplus.cpp graphviz-2.40.1/plugin/gdiplus/gvplugin_gdiplus.cpp
--- graphviz-2.40.1-orig/plugin/gdiplus/gvplugin_gdiplus.cpp	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gdiplus/gvplugin_gdiplus.cpp	2017-03-16 12:01:28.689777700 +0300
@@ -89,7 +89,7 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL /*visual studio*/
+#ifdef _MSC_VER /*visual studio*/
 #ifndef GVPLUGIN_GDIPLUS_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_gdiplus_LTX_library = { "gdiplus", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/neato_layout/Makefile.am graphviz-2.40.1/plugin/neato_layout/Makefile.am
--- graphviz-2.40.1-orig/plugin/neato_layout/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/neato_layout/Makefile.am	2017-03-16 08:28:18.601173900 +0300
@@ -22,6 +22,7 @@
 	$(top_builddir)/lib/patchwork/libpatchwork_C.la \
 	$(top_builddir)/lib/osage/libosage_C.la \
 	$(top_builddir)/lib/fdpgen/libfdpgen_C.la \
+	$(top_builddir)/lib/ortho/libortho_C.la \
 	$(top_builddir)/lib/sparse/libsparse_C.la \
 	$(top_builddir)/lib/rbtree/librbtree_C.la \
 	$(top_builddir)/lib/circogen/libcircogen_C.la
diff -Naur graphviz-2.40.1-orig/plugin/neato_layout/gvplugin_neato_layout.c graphviz-2.40.1/plugin/neato_layout/gvplugin_neato_layout.c
--- graphviz-2.40.1-orig/plugin/neato_layout/gvplugin_neato_layout.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/neato_layout/gvplugin_neato_layout.c	2017-03-16 12:00:48.209627600 +0300
@@ -20,7 +20,7 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL /*visual studio*/
+#ifdef _MSC_VER /*visual studio*/
 #ifndef GVPLUGIN_NEATO_LAYOUT_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_neato_layout_LTX_library = { "neato_layout", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/pango/gvloadimage_pango.c graphviz-2.40.1/plugin/pango/gvloadimage_pango.c
--- graphviz-2.40.1-orig/plugin/pango/gvloadimage_pango.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/plugin/pango/gvloadimage_pango.c	2017-03-16 08:33:32.347998200 +0300
@@ -21,7 +21,7 @@
 #ifdef HAVE_PANGOCAIRO
 #include <cairo.h>
 
-#ifdef WIN32 //*dependencies
+#ifdef _MSC_VER //*dependencies
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "glib-2.0.lib" )
     #pragma comment( lib, "pango-1.0.lib" )

diff -Naur graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.c graphviz-2.40.1/plugin/pango/gvplugin_pango.c
--- graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/pango/gvplugin_pango.c	2017-03-16 11:59:45.952632400 +0300
@@ -26,7 +26,7 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL   /*visual studio*/
+#ifdef _MSC_VER   /*visual studio*/
 #ifndef GVPLUGIN_PANGO_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_pango_LTX_library = { "cairo", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.c graphviz-2.40.1/plugin/visio/gvplugin_visio.c
--- graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/visio/gvplugin_visio.c	2017-03-16 12:00:10.041517000 +0300
@@ -11,7 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
-#ifdef WIN32 //*dependencies
+#ifdef _MSC_VER //*dependencies
     #pragma comment( lib, "gvc.lib" )
 //    #pragma comment( lib, "glib-2.0.lib" )
 //    #pragma comment( lib, "pango-1.0.lib" )
@@ -33,7 +33,7 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL   /*visual studio*/
+#ifdef _MSC_VER   /*visual studio*/
 #ifndef GVPLUGIN_VISIO_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_visio_LTX_library = { "visio", apis };
 #else
