diff -Naur graphviz-2.40.1-orig/lib/cdt/Makefile.am graphviz-2.40.1/lib/cdt/Makefile.am
--- graphviz-2.40.1-orig/lib/cdt/Makefile.am	2016-12-22 08:44:41.000000000 +0300
+++ graphviz-2.40.1/lib/cdt/Makefile.am	2018-01-22 13:05:19.169484900 +0300
@@ -24,6 +24,10 @@
 libcdt_la_LDFLAGS = -version-info $(CDT_VERSION) -no-undefined
 libcdt_la_SOURCES = $(libcdt_C_la_SOURCES)
 
+if WITH_WIN32
+libcdt_la_LDFLAGS += -export-symbols $(top_srcdir)/lib/cdt/cdt.def
+endif
+
 cdt.3.pdf: $(srcdir)/cdt.3
 	- @GROFF@ -Tps -man $(srcdir)/cdt.3 | @PS2PDF@ - - >cdt.3.pdf
 
diff -Naur graphviz-2.40.1-orig/lib/cgraph/Makefile.am graphviz-2.40.1/lib/cgraph/Makefile.am
--- graphviz-2.40.1-orig/lib/cgraph/Makefile.am	2016-10-08 20:56:00.000000000 +0300
+++ graphviz-2.40.1/lib/cgraph/Makefile.am	2018-01-22 13:05:26.699526400 +0300
@@ -26,6 +26,10 @@
 libcgraph_la_SOURCES = $(libcgraph_C_la_SOURCES)
 libcgraph_la_LIBADD = $(top_builddir)/lib/cdt/libcdt.la
 
+if WITH_WIN32
+libcgraph_la_LDFLAGS += -export-symbols $(top_srcdir)/lib/cgraph/cgraph.def
+endif
+
 scan.o scan.lo: scan.c grammar.h
 
 scan.c: $(top_srcdir)/lib/cgraph/scan.l
diff -Naur graphviz-2.40.1-orig/lib/edgepaint/Makefile.am graphviz-2.40.1/lib/edgepaint/Makefile.am
--- graphviz-2.40.1-orig/lib/edgepaint/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/edgepaint/Makefile.am	2018-01-22 09:36:57.619764100 +0300
@@ -28,6 +28,10 @@
 
 liblab_gamut_la_LDFLAGS = -version-info $(LAB_GAMUT_VERSION)
 liblab_gamut_la_SOURCES = lab_gamut.c
+if WITH_WIN32
+liblab_gamut_la_LDFLAGS += -no-undefined
+liblab_gamut_la_LDFLAGS += -export-symbols $(top_srcdir)/lib/edgepaint/lab_gamut.def
+endif
 
 lab_gamut.3.pdf: $(srcdir)/lab_gamut.3
 	- @GROFF@ -Tps -man $(srcdir)/lab_gamut.3 | @PS2PDF@ - - >lab_gamut.3.pdf
diff -Naur graphviz-2.40.1-orig/lib/pathplan/Makefile.am graphviz-2.40.1/lib/pathplan/Makefile.am
--- graphviz-2.40.1-orig/lib/pathplan/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/pathplan/Makefile.am	2018-01-22 09:36:57.698764300 +0300
@@ -26,6 +26,10 @@
 libpathplan_la_SOURCES = $(libpathplan_C_la_SOURCES)
 libpathplan_la_LIBADD = @MATH_LIBS@
 
+if WITH_WIN32
+libpathplan_la_LDFLAGS += -export-symbols $(top_srcdir)/lib/pathplan/pathplan.def
+endif
+
 pathplan.3.pdf: $(srcdir)/pathplan.3
 	- @GROFF@ -Tps -man $(srcdir)/pathplan.3 | @PS2PDF@ - - >pathplan.3.pdf
 
diff -Naur graphviz-2.40.1-orig/tclpkg/tcldot/Makefile.am graphviz-2.40.1/tclpkg/tcldot/Makefile.am
--- graphviz-2.40.1-orig/tclpkg/tcldot/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/tclpkg/tcldot/Makefile.am	2018-01-22 09:36:57.619764100 +0300
@@ -99,6 +99,10 @@
 libtcldot_builtin_la_LIBADD += $(top_builddir)/plugin/gd/libgvplugin_gd_C.la
 libtcldot_builtin_la_LIBADD += $(GD_LIBS)
 endif
+
+if WITH_WIN32
+libtcldot_builtin_la_LDFLAGS += -no-undefined
+endif
 
 libtcldot_builtin_la_LIBADD += $(EXPAT_LIBS) $(LIBGEN_LIBS) $(SOCKET_LIBS) $(IPSEPCOLA_LIBS) $(MATH_LIBS)
 
