diff -Naur graphviz-2.40.1-orig/configure.ac graphviz-2.40.1/configure.ac
--- graphviz-2.40.1-orig/configure.ac	2016-12-22 08:44:41.000000000 +0300
+++ graphviz-2.40.1/configure.ac	2017-03-15 09:24:50.495509200 +0300
@@ -292,7 +292,7 @@
   TCLSH=$withval,)
 
 if test "x$TCLSH" = "x"; then
-  AC_PATH_PROGS(TCLSH,[tclsh8.6 tclsh8.5 tclsh8.4 tclsh8.3 tclsh])
+  AC_PATH_PROGS(TCLSH,[tclsh8.6 tclsh86 tclsh8.5 tclsh85 tclsh8.4 tclsh84 tclsh8.3 tclsh83 tclsh])
 #  if test "x$TCLSH" = "x"; then
 #    AC_MSG_ERROR([Unable to find a tclsh. Tclsh is a required program for building graphviz, independent of wether tcl-based graphviz products are built])
 #    use_tcl="No (tclsh unavailable)"
@@ -1625,7 +1625,7 @@
           RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb vendorarchdir`"
 	fi
       ],[
-        PKG_CHECK_MODULES([RUBY], [ruby-1.9],[
+        PKG_CHECK_MODULES([RUBY], [ruby-2.5],[
 	  AC_CHECK_PROG(RUBY,ruby,ruby)
           if test "x$RUBY" = "x"; then
             use_ruby="No (ruby not available)"
@@ -2808,8 +2808,8 @@
   LDFLAGS="$LDFLAGS $GLUT_LIBS"
   
   AC_CHECK_HEADER(GL/glut.h,
-    AC_CHECK_LIB(glut,main,
-      [GLUT_LIBS="$GLUT_LIBS -lglut"
+    AC_CHECK_LIB(freeglut,main,
+      [GLUT_LIBS="$GLUT_LIBS -lfreeglut"
       use_glut="Yes"
       AC_DEFINE_UNQUOTED(HAVE_GLUT,1,[Define if you have the GLUT library])],
       use_glut="No (missing libglut)"
