diff -Naur graphviz-2.40.1-orig/lib/common/geomprocs.h graphviz-2.40.1/lib/common/geomprocs.h
--- graphviz-2.40.1-orig/lib/common/geomprocs.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/geomprocs.h	2018-01-23 10:25:03.844181100 +0300
@@ -24,13 +24,15 @@
 
 #include "geom.h"
 
-#ifdef _WIN32
-#ifdef GVDLL
+#ifdef _MSC_VER
+#ifndef GVC_EXPORTS
+#define extern __declspec(dllimport)
+#endif
+#elif defined(GVDLL)
 #define extern __declspec(dllexport)
 #else
-#define extern __declspec(dllimport)
+#define extern
 #endif
-#endif	
 
 extern box mkbox(point p, point q);
 extern boxf mkboxf(pointf p, pointf q);
diff -Naur graphviz-2.40.1-orig/lib/common/globals.h graphviz-2.40.1/lib/common/globals.h
--- graphviz-2.40.1-orig/lib/common/globals.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/globals.h	2018-01-23 10:24:14.974596900 +0300
@@ -38,19 +38,21 @@
 #if !defined(_BLD_dotneato) && defined(__IMPORT__)
 #   define external __IMPORT__
 #endif
-#if defined(GVDLL)
-#if !defined(_BLD_gvc)
-#define extern	__declspec(dllimport)
-#else
-#define extern __declspec(dllexport)
-#endif
-#endif
 #endif
+
 /*visual studio*/
-#ifdef _WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#if !defined(_BLD_gvc)
+#define extern	__declspec(dllimport)
+#else
+#define extern __declspec(dllexport)
+#endif
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/common/memory.h graphviz-2.40.1/lib/common/memory.h
--- graphviz-2.40.1-orig/lib/common/memory.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/memory.h	2018-01-23 10:23:24.065991300 +0300
@@ -42,15 +42,15 @@
 #define RALLOC(size,ptr,type) ((type*)grealloc(ptr,(size)*sizeof(type)))
 #define ZALLOC(size,ptr,type,osize) (ptr? (type*)zrealloc(ptr,size,sizeof(type),osize):(type*)zmalloc((size)*sizeof(type)))
 #endif
-#ifdef GVDLL
-#define extern __declspec(dllexport)
-#else
-#ifdef _WIN32
+
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
-#endif
-
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 
     extern void *zmalloc(size_t);
diff -Naur graphviz-2.40.1-orig/lib/common/pointset.h graphviz-2.40.1/lib/common/pointset.h
--- graphviz-2.40.1-orig/lib/common/pointset.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/pointset.h	2018-01-23 10:23:05.753309100 +0300
@@ -23,18 +23,16 @@
 
     typedef Dict_t PointSet;
     typedef Dict_t PointMap;
-#ifdef GVDLL
-#define extern __declspec(dllexport)
-#else
-#define extern
-#endif
 
 /*visual studio*/
-#ifdef _WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
-#undef extern
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/common/render.h graphviz-2.40.1/lib/common/render.h
--- graphviz-2.40.1-orig/lib/common/render.h	2016-12-22 08:44:41.000000000 +0300
+++ graphviz-2.40.1/lib/common/render.h	2018-01-23 10:22:28.726960600 +0300
@@ -58,13 +58,16 @@
 
     typedef void (*nodesizefn_t) (Agnode_t *, boolean);
 
-/*visual studio*/
-#ifdef _WIN32
+
+#if defined(_MSC_VER)
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
-/*end visual studio*/
 
 	extern void add_box(path *, boxf);
     extern void arrow_flags(Agedge_t * e, int *sflag, int *eflag);
diff -Naur graphviz-2.40.1-orig/lib/common/utils.h graphviz-2.40.1/lib/common/utils.h
--- graphviz-2.40.1-orig/lib/common/utils.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/common/utils.h	2018-01-23 10:21:21.231508300 +0300
@@ -20,10 +20,14 @@
 
 
 /*visual studio*/
-#ifdef _WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvc.h graphviz-2.40.1/lib/gvc/gvc.h
--- graphviz-2.40.1-orig/lib/gvc/gvc.h	2016-12-18 15:58:37.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvc.h	2018-01-23 10:25:52.228393600 +0300
@@ -21,18 +21,15 @@
 extern "C" {
 #endif
 
-#ifdef GVDLL
-#define extern __declspec(dllexport)
-#else
-#define extern
-#endif
-
 /*visual studio*/
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
-#undef extern
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 	
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvcext.h graphviz-2.40.1/lib/gvc/gvcext.h
--- graphviz-2.40.1-orig/lib/gvc/gvcext.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvcext.h	2018-01-23 09:40:58.273501000 +0300
@@ -64,7 +64,7 @@
     typedef struct gvplugin_available_s gvplugin_available_t;
 
 /*visual studio*/
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 __declspec(dllimport) lt_symlist_t lt_preloaded_symbols[];
 #else
@@ -77,11 +77,11 @@
 /*end visual studio*/
 
 
-#ifndef WIN32
+#ifndef _MSC_VER
+#if !defined(LTDL_H)
 #if defined(GVDLL)
 	__declspec(dllexport) lt_symlist_t lt_preloaded_symbols[];
 #else
-#if !defined(LTDL_H)
 	extern lt_symlist_t lt_preloaded_symbols[];
 #endif
 #endif
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvcint.h graphviz-2.40.1/lib/gvc/gvcint.h
--- graphviz-2.40.1-orig/lib/gvc/gvcint.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvcint.h	2018-01-23 10:27:04.653316800 +0300
@@ -24,6 +24,16 @@
 #include "gvcommon.h"
 #include "color.h"
 
+#ifdef _MSC_VER
+#ifndef GVC_EXPORTS
+#define extern __declspec(dllimport)
+#endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
+#endif
+
     /* active plugin headers */
     typedef struct gvplugin_active_layout_s {
         gvlayout_engine_t *engine;
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvio.h graphviz-2.40.1/lib/gvc/gvio.h
--- graphviz-2.40.1-orig/lib/gvc/gvio.h	2016-09-08 07:45:03.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvio.h	2018-01-23 10:27:43.497868100 +0300
@@ -20,18 +20,15 @@
 extern "C" {
 #endif
 
-#ifdef GVDLL
-#define extern __declspec(dllexport)
-#else
-#define extern
-#endif
-
 /*visual studio*/
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
-#undef extern
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/gvc/gvplugin_loadimage.h graphviz-2.40.1/lib/gvc/gvplugin_loadimage.h
--- graphviz-2.40.1-orig/lib/gvc/gvplugin_loadimage.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvc/gvplugin_loadimage.h	2018-01-23 10:28:02.997557700 +0300
@@ -22,15 +22,15 @@
 extern "C" {
 #endif
 
-#ifdef GVDLL
-#  define extern __declspec(dllexport)
-#endif
-
 /*visual studio*/
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/gvpr/gprstate.c graphviz-2.40.1/lib/gvpr/gprstate.c
--- graphviz-2.40.1-orig/lib/gvpr/gprstate.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/gvpr/gprstate.c	2018-01-23 09:09:14.250538500 +0300
@@ -17,6 +17,8 @@
  *
  */
 
+#include "config.h"
+
 #ifdef WIN32
 #include "windows.h"
 #include "shlwapi.h"
@@ -139,7 +141,7 @@
     free (state);
 }
 
-#ifdef WIN32_DLL
+#if defined(WIN32_DLL) || defined(GVDLL)
 int pathisrelative (char* path)
 {
     return PathIsRelative(path);
diff -Naur graphviz-2.40.1-orig/lib/gvpr/gvpr.c graphviz-2.40.1/lib/gvpr/gvpr.c
--- graphviz-2.40.1-orig/lib/gvpr/gvpr.c	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/lib/gvpr/gvpr.c	2018-01-23 09:10:13.915402100 +0300
@@ -219,7 +219,7 @@
     Sfio_t *pathp = NULL;
     size_t sz;
 
-#ifdef WIN32_DLL
+#if defined(WIN32_DLL) || defined(GVDLL)
     if (!pathisrelative(arg))
 #else
     if (strchr(arg, '/'))
diff -Naur graphviz-2.40.1-orig/lib/pack/pack.h graphviz-2.40.1/lib/pack/pack.h
--- graphviz-2.40.1-orig/lib/pack/pack.h	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/lib/pack/pack.h	2018-01-23 10:28:28.999274600 +0300
@@ -58,10 +58,14 @@
     } pack_info;
 
 /*visual studio*/
-#ifdef WIN32
+#ifdef _MSC_VER
 #ifndef GVC_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/pathplan/pathutil.h graphviz-2.40.1/lib/pathplan/pathutil.h
--- graphviz-2.40.1-orig/lib/pathplan/pathutil.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/pathplan/pathutil.h	2018-01-23 10:29:52.419556600 +0300
@@ -32,10 +32,14 @@
 #endif
 
 /*visual studio*/
-#ifdef _WIN32
+#ifdef _MSC_VER
 #ifndef PATHPLAN_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 	typedef double COORD;
diff -Naur graphviz-2.40.1-orig/lib/pathplan/vis.h graphviz-2.40.1/lib/pathplan/vis.h
--- graphviz-2.40.1-orig/lib/pathplan/vis.h	2016-12-08 08:43:20.000000000 +0300
+++ graphviz-2.40.1/lib/pathplan/vis.h	2018-01-23 10:30:37.277179400 +0300
@@ -46,10 +46,16 @@
 	/* this is computed from the above */
 	array2 vis;
     };
-#ifdef _WIN32
+
+/*visual studio*/
+#ifdef _MSC_VER
 #ifndef PATHPLAN_EXPORTS
 #define extern __declspec(dllimport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/
 
diff -Naur graphviz-2.40.1-orig/lib/vmalloc/vmalloc.h graphviz-2.40.1/lib/vmalloc/vmalloc.h
--- graphviz-2.40.1-orig/lib/vmalloc/vmalloc.h	2016-12-22 08:44:41.000000000 +0300
+++ graphviz-2.40.1/lib/vmalloc/vmalloc.h	2018-01-23 10:31:24.766384500 +0300
@@ -110,15 +110,17 @@
 #endif
 
 /*visual studio*/
-#ifdef WIN32
-#undef extern
+#ifdef _MSC_VER
 #ifndef VMALLOC_EXPORTS
 #define extern __declspec(dllimport)
-#else
-#define extern __declspec(dllexport)
 #endif
+#elif defined(GVDLL)
+#define extern __declspec(dllexport)
+#else
+#define extern
 #endif
 /*end visual studio*/		 
+
 		 extern Vmethod_t *Vmbest;	/* best allocation              */
     extern Vmethod_t *Vmlast;	/* last-block allocation        */
     extern Vmethod_t *Vmpool;	/* pool allocation              */
diff -Naur graphviz-2.40.1-orig/tclpkg/gdtclft/gdtclft.c graphviz-2.40.1/tclpkg/gdtclft/gdtclft.c
--- graphviz-2.40.1-orig/tclpkg/gdtclft/gdtclft.c	2018-01-23 11:35:08.100504600 +0300
+++ graphviz-2.40.1/tclpkg/gdtclft/gdtclft.c	2018-01-23 11:35:11.352690600 +0300
@@ -1270,7 +1270,8 @@
 #ifdef __CYGWIN__
 int Gdtclft_Init(Tcl_Interp * interp)
 #else
-#ifdef __WIN32__
+#ifdef _MSC_VER
+
 EXPORT(int, Gdtclft_Init) (interp)
 #else
 int Gdtclft_Init(Tcl_Interp * interp)
@@ -1296,7 +1297,7 @@
 #ifdef __CYGWIN__
 int Gdtclft_SafeInit(Tcl_Interp * interp)
 #else
-#ifdef __WIN32__
+#ifdef _MSC_VER
 EXPORT(int, Gdtclft_SafeInit) (interp)
 #else
 int Gdtclft_SafeInit(Tcl_Interp * interp)
