diff -Naur graphviz-2.40.1-orig/plugin/core/gvplugin_core.c graphviz-2.40.1/plugin/core/gvplugin_core.c
--- graphviz-2.40.1-orig/plugin/core/gvplugin_core.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/core/gvplugin_core.c	2018-01-23 08:02:03.964079300 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_dot_types[];
@@ -88,22 +89,16 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_CORE_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_core_LTX_library = { "core", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_core_LTX_library = { "core", apis };
 #endif
-#endif
-
-
-
-#ifndef WIN32_DLL
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_core_LTX_library = { "core", apis };
 #else
 gvplugin_library_t gvplugin_core_LTX_library = { "core", apis };
 #endif
 #endif
-
-
diff -Naur graphviz-2.40.1-orig/plugin/core/Makefile.am graphviz-2.40.1/plugin/core/Makefile.am
--- graphviz-2.40.1-orig/plugin/core/Makefile.am	2016-09-20 07:45:02.000000000 +0300
+++ graphviz-2.40.1/plugin/core/Makefile.am	2018-01-23 08:12:35.710656500 +0300
@@ -51,7 +51,7 @@
 	$(AWK) -f $(top_srcdir)/awk/stringize.awk $(srcdir)/ps.txt > ps.h
 
 if WITH_WIN32
-libgvplugin_core_la_LDFLAGS += -no-undefined
+libgvplugin_core_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/core/gvplugin_core.def
 libgvplugin_core_la_LIBADD += $(top_builddir)/lib/cgraph/libcgraph.la
 endif
 
diff -Naur graphviz-2.40.1-orig/plugin/devil/gvplugin_devil.c graphviz-2.40.1/plugin/devil/gvplugin_devil.c
--- graphviz-2.40.1-orig/plugin/devil/gvplugin_devil.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/devil/gvplugin_devil.c	2018-01-23 08:04:14.088209400 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_devil_types[];
@@ -20,4 +21,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_DEVIL_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_devil_LTX_library = { "devil", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_devil_LTX_library = { "devil", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_devil_LTX_library = { "devil", apis };
+#else
 gvplugin_library_t gvplugin_devil_LTX_library = { "devil", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/devil/gvplugin_devil.def graphviz-2.40.1/plugin/devil/gvplugin_devil.def
--- graphviz-2.40.1-orig/plugin/devil/gvplugin_devil.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/devil/gvplugin_devil.def	2018-01-23 08:11:13.401536800 +0300
@@ -0,0 +1,4 @@
+LIBRARY	"gvplugin_devil"
+EXPORTS
+gvplugin_devil_LTX_library	
+
diff -Naur graphviz-2.40.1-orig/plugin/devil/Makefile.am graphviz-2.40.1/plugin/devil/Makefile.am
--- graphviz-2.40.1-orig/plugin/devil/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/devil/Makefile.am	2018-01-23 08:11:52.942287300 +0300
@@ -28,7 +28,7 @@
 libgvplugin_devil_la_LIBADD = @DEVIL_LIBS@
 
 if WITH_WIN32
-libgvplugin_devil_la_LDFLAGS += -no-undefined
+libgvplugin_devil_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/devil/gvplugin_devil.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/dot_layout/gvplugin_dot_layout.c graphviz-2.40.1/plugin/dot_layout/gvplugin_dot_layout.c
--- graphviz-2.40.1-orig/plugin/dot_layout/gvplugin_dot_layout.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/dot_layout/gvplugin_dot_layout.c	2018-01-23 08:07:12.062433200 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvlayout_dot_layout[];
@@ -19,26 +20,17 @@
     {API_layout, gvlayout_dot_layout},
     {(api_t)0, 0},
 };
-/*visual studio*/
-#ifdef WIN32_DLL
+
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_DOT_LAYOUT_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_dot_layout_LTX_library = { "dot_layout", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_dot_layout_LTX_library = { "dot_layout", apis };
 #endif
-#endif
-
-
-
-/*end visual studio*/
-
-
-#ifndef WIN32_DLL
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_dot_layout_LTX_library = { "dot_layout", apis };
 #else
 gvplugin_library_t gvplugin_dot_layout_LTX_library = { "dot_layout", apis };
 #endif
 #endif
-
-
diff -Naur graphviz-2.40.1-orig/plugin/dot_layout/Makefile.am graphviz-2.40.1/plugin/dot_layout/Makefile.am
--- graphviz-2.40.1-orig/plugin/dot_layout/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/dot_layout/Makefile.am	2018-01-23 08:10:02.459053100 +0300
@@ -33,7 +33,7 @@
 	$(top_builddir)/lib/pathplan/libpathplan.la $(MATH_LIBS)
 	
 if WITH_WIN32
-libgvplugin_dot_layout_la_LDFLAGS += -no-undefined
+libgvplugin_dot_layout_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/dot_layout/gvplugin_dot_layout.def
 libgvplugin_dot_layout_la_LIBADD += $(top_builddir)/lib/cgraph/libcgraph.la
 endif
 
diff -Naur graphviz-2.40.1-orig/plugin/gd/gvplugin_gd.c graphviz-2.40.1/plugin/gd/gvplugin_gd.c
--- graphviz-2.40.1-orig/plugin/gd/gvplugin_gd.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gd/gvplugin_gd.c	2018-01-23 08:07:35.375314300 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvrender_gd_types[];
@@ -32,18 +33,16 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL /*visual studio*/
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_GD_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_gd_LTX_library = { "gd", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_gd_LTX_library = { "gd", apis };
 #endif
-#else /*end visual studio*/
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_gd_LTX_library = { "gd", apis };
 #else
 gvplugin_library_t gvplugin_gd_LTX_library = { "gd", apis };
 #endif
 #endif
-
-
diff -Naur graphviz-2.40.1-orig/plugin/gd/Makefile.am graphviz-2.40.1/plugin/gd/Makefile.am
--- graphviz-2.40.1-orig/plugin/gd/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gd/Makefile.am	2018-01-23 08:20:09.217596000 +0300
@@ -39,7 +39,7 @@
 libgvplugin_gd_la_DEPENDENCIES =  $(top_builddir)/lib/gvc/libgvc.la
 
 if WITH_WIN32
-libgvplugin_gd_la_LDFLAGS += -no-undefined
+libgvplugin_gd_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/gd/gvplugin_gd.def
 libgvplugin_gd_la_LIBADD += $(top_builddir)/lib/cgraph/libcgraph.la $(top_builddir)/lib/pathplan/libpathplan.la
 endif
 
diff -Naur graphviz-2.40.1-orig/plugin/gdiplus/gvplugin_gdiplus.cpp graphviz-2.40.1/plugin/gdiplus/gvplugin_gdiplus.cpp
--- graphviz-2.40.1-orig/plugin/gdiplus/gvplugin_gdiplus.cpp	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gdiplus/gvplugin_gdiplus.cpp	2018-01-23 08:15:12.039264600 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 #include "gvplugin_gdiplus.h"
@@ -89,17 +90,16 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL /*visual studio*/
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_GDIPLUS_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_gdiplus_LTX_library = { "gdiplus", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_gdiplus_LTX_library = { "gdiplus", apis };
 #endif
-#else /*end visual studio*/
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_gdiplus_LTX_library = { "gdiplus", apis };
 #else
 extern "C" gvplugin_library_t gvplugin_gdiplus_LTX_library = { "gdiplus", apis };
 #endif
 #endif
-
diff -Naur graphviz-2.40.1-orig/plugin/gdiplus/Makefile.am graphviz-2.40.1/plugin/gdiplus/Makefile.am
--- graphviz-2.40.1-orig/plugin/gdiplus/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gdiplus/Makefile.am	2018-01-23 08:14:05.571784700 +0300
@@ -39,7 +39,7 @@
 libgvplugin_gdiplus_la_LIBADD = -lgdi32 -lole32 -lstdc++ -luuid $(top_builddir)/lib/gvc/libgvc.la
 
 if WITH_WIN32
-libgvplugin_gdiplus_la_LDFLAGS += -no-undefined
+libgvplugin_gdiplus_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/gdiplus/gvplugin_gdiplus.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/gdk/gvplugin_gdk.c graphviz-2.40.1/plugin/gdk/gvplugin_gdk.c
--- graphviz-2.40.1-orig/plugin/gdk/gvplugin_gdk.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gdk/gvplugin_gdk.c	2018-01-23 08:17:03.275743900 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_gdk_types[];
@@ -22,4 +23,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_GDK_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_gdk_LTX_library = { "gdk", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_gdk_LTX_library = { "gdk", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_gdk_LTX_library = { "gdk", apis };
+#else
 gvplugin_library_t gvplugin_gdk_LTX_library = { "gdk", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/gdk/gvplugin_gdk.def graphviz-2.40.1/plugin/gdk/gvplugin_gdk.def
--- graphviz-2.40.1-orig/plugin/gdk/gvplugin_gdk.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/gdk/gvplugin_gdk.def	2018-01-23 08:17:20.259996900 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_gdk"
+EXPORTS
+gvplugin_gdk_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/gdk/Makefile.am graphviz-2.40.1/plugin/gdk/Makefile.am
--- graphviz-2.40.1-orig/plugin/gdk/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gdk/Makefile.am	2018-01-23 08:16:07.200675500 +0300
@@ -34,7 +34,7 @@
 	$(top_builddir)/lib/gvc/libgvc.la @GDK_PIXBUF_LIBS@ @PANGOCAIRO_LIBS@ @GDK_LIBS@
 	
 if WITH_WIN32
-libgvplugin_gdk_la_LDFLAGS += -no-undefined
+libgvplugin_gdk_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/gdk/gvplugin_gdk.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/glitz/gvplugin_glitz.c graphviz-2.40.1/plugin/glitz/gvplugin_glitz.c
--- graphviz-2.40.1-orig/plugin/glitz/gvplugin_glitz.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/glitz/gvplugin_glitz.c	2018-01-23 08:19:06.248401200 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_types_glitz[];
@@ -20,4 +21,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_GLITZ_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_glitz_LTX_library = { "glitz", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_glitz_LTX_library = { "glitz", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_glitz_LTX_library = { "glitz", apis };
+#else
 gvplugin_library_t gvplugin_glitz_LTX_library = { "glitz", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/glitz/gvplugin_glitz.def graphviz-2.40.1/plugin/glitz/gvplugin_glitz.def
--- graphviz-2.40.1-orig/plugin/glitz/gvplugin_glitz.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/glitz/gvplugin_glitz.def	2018-01-23 08:18:50.011725200 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_glitz"
+EXPORTS
+gvplugin_glitz_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/glitz/Makefile.am graphviz-2.40.1/plugin/glitz/Makefile.am
--- graphviz-2.40.1-orig/plugin/glitz/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/glitz/Makefile.am	2018-01-23 08:19:50.859738100 +0300
@@ -30,7 +30,7 @@
 libgvplugin_glitz_la_LIBADD = @PANGOCAIRO_LIBS@ @GLITZ_LIBS@
 
 if WITH_WIN32
-libgvplugin_glitz_la_LDFLAGS += -no-undefined
+libgvplugin_glitz_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/glitz/gvplugin_glitz.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/gs/gvplugin_gs.c graphviz-2.40.1/plugin/gs/gvplugin_gs.c
--- graphviz-2.40.1-orig/plugin/gs/gvplugin_gs.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gs/gvplugin_gs.c	2018-01-23 08:21:57.037950500 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvloadimage_gs_types[];
@@ -20,4 +21,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_GS_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_gs_LTX_library = { "gs", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_gs_LTX_library = { "gs", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_gs_LTX_library = { "gs", apis };
+#else
 gvplugin_library_t gvplugin_gs_LTX_library = { "gs", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/gs/gvplugin_gs.def graphviz-2.40.1/plugin/gs/gvplugin_gs.def
--- graphviz-2.40.1-orig/plugin/gs/gvplugin_gs.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/gs/gvplugin_gs.def	2018-01-23 08:21:45.625916200 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_gs"
+EXPORTS
+gvplugin_gs_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/gs/Makefile.am graphviz-2.40.1/plugin/gs/Makefile.am
--- graphviz-2.40.1-orig/plugin/gs/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gs/Makefile.am	2018-01-23 08:21:00.107564900 +0300
@@ -31,7 +31,7 @@
 	$(top_builddir)/lib/gvc/libgvc.la $(PANGOCAIRO_LIBS) $(GS_LIBS)
 
 if WITH_WIN32
-libgvplugin_gs_la_LDFLAGS += -no-undefined
+libgvplugin_gs_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/gs/gvplugin_gs.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/gtk/gvplugin_gtk.c graphviz-2.40.1/plugin/gtk/gvplugin_gtk.c
--- graphviz-2.40.1-orig/plugin/gtk/gvplugin_gtk.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gtk/gvplugin_gtk.c	2018-01-23 08:23:47.288735700 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_types_gtk[];
@@ -20,4 +21,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_GTK_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_gtk_LTX_library = { "gtk", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_gtk_LTX_library = { "gtk", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_gtk_LTX_library = { "gtk", apis };
+#else
 gvplugin_library_t gvplugin_gtk_LTX_library = { "gtk", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/gtk/gvplugin_gtk.def graphviz-2.40.1/plugin/gtk/gvplugin_gtk.def
--- graphviz-2.40.1-orig/plugin/gtk/gvplugin_gtk.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/gtk/gvplugin_gtk.def	2018-01-23 08:23:40.781511100 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_gtk"
+EXPORTS
+gvplugin_gtk_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/gtk/Makefile.am graphviz-2.40.1/plugin/gtk/Makefile.am
--- graphviz-2.40.1-orig/plugin/gtk/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/gtk/Makefile.am	2018-01-23 08:22:55.144538100 +0300
@@ -34,7 +34,7 @@
 libgvplugin_gtk_la_LIBADD = $(GTK_LIBS)
 
 if WITH_WIN32
-libgvplugin_gtk_la_LDFLAGS += -no-undefined
+libgvplugin_gtk_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/gtk/gvplugin_gtk.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/lasi/gvplugin_lasi.c graphviz-2.40.1/plugin/lasi/gvplugin_lasi.c
--- graphviz-2.40.1-orig/plugin/lasi/gvplugin_lasi.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/lasi/gvplugin_lasi.c	2018-01-23 08:26:00.230259700 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_lasi_types[];
@@ -24,4 +25,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_LASI_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_lasi_LTX_library = { "lasi", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_lasi_LTX_library = { "lasi", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_lasi_LTX_library = { "lasi", apis };
+#else
 gvplugin_library_t gvplugin_lasi_LTX_library = { "lasi", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/lasi/gvplugin_lasi.def graphviz-2.40.1/plugin/lasi/gvplugin_lasi.def
--- graphviz-2.40.1-orig/plugin/lasi/gvplugin_lasi.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/lasi/gvplugin_lasi.def	2018-01-23 08:26:08.558687200 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_lasi"
+EXPORTS
+gvplugin_lasi_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/lasi/Makefile.am graphviz-2.40.1/plugin/lasi/Makefile.am
--- graphviz-2.40.1-orig/plugin/lasi/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/lasi/Makefile.am	2018-01-23 08:25:26.223527400 +0300
@@ -38,7 +38,7 @@
 	$(AWK) -f $(top_srcdir)/awk/stringize.awk $(top_srcdir)/plugin/core/ps.txt > ps.h
 
 if WITH_WIN32
-libgvplugin_lasi_la_LDFLAGS += -no-undefined
+libgvplugin_lasi_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/lasi/gvplugin_lasi.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/ming/gvplugin_ming.c graphviz-2.40.1/plugin/ming/gvplugin_ming.c
--- graphviz-2.40.1-orig/plugin/ming/gvplugin_ming.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/ming/gvplugin_ming.c	2018-01-23 08:26:58.189040800 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvrender_ming_types[];
@@ -22,4 +23,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_MING_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_ming_LTX_library = { "ming", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_ming_LTX_library = { "ming", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_ming_LTX_library = { "ming", apis };
+#else
 gvplugin_library_t gvplugin_ming_LTX_library = { "ming", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/ming/gvplugin_ming.def graphviz-2.40.1/plugin/ming/gvplugin_ming.def
--- graphviz-2.40.1-orig/plugin/ming/gvplugin_ming.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/ming/gvplugin_ming.def	2018-01-23 08:27:48.151221700 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_ming"
+EXPORTS
+gvplugin_ming_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/ming/Makefile.am graphviz-2.40.1/plugin/ming/Makefile.am
--- graphviz-2.40.1-orig/plugin/ming/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/ming/Makefile.am	2018-01-23 08:27:27.756351800 +0300
@@ -31,7 +31,7 @@
 libgvplugin_ming_la_LIBADD = @MING_LIBS@ @MATH_LIBS@
 
 if WITH_WIN32
-libgvplugin_ming_la_LDFLAGS += -no-undefined
+libgvplugin_ming_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/ming/gvplugin_ming.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/neato_layout/gvplugin_neato_layout.c graphviz-2.40.1/plugin/neato_layout/gvplugin_neato_layout.c
--- graphviz-2.40.1-orig/plugin/neato_layout/gvplugin_neato_layout.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/neato_layout/gvplugin_neato_layout.c	2018-01-23 08:29:11.100307200 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvlayout_neato_types[];
@@ -20,17 +21,16 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL /*visual studio*/
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_NEATO_LAYOUT_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_neato_layout_LTX_library = { "neato_layout", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_neato_layout_LTX_library = { "neato_layout", apis };
 #endif
-#else /*end visual studio*/
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_neato_layout_LTX_library = { "neato_layout", apis };
 #else
 gvplugin_library_t gvplugin_neato_layout_LTX_library = { "neato_layout", apis };
 #endif
 #endif
-
diff -Naur graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.c graphviz-2.40.1/plugin/pango/gvplugin_pango.c
--- graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/pango/gvplugin_pango.c	2018-01-23 08:30:13.337119300 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvrender_pango_types[];
@@ -26,17 +27,16 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL   /*visual studio*/
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_PANGO_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_pango_LTX_library = { "cairo", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_pango_LTX_library = { "cairo", apis };
 #endif
-#else /*end visual studio*/
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_pango_LTX_library = { "cairo", apis };
 #else
 gvplugin_library_t gvplugin_pango_LTX_library = { "cairo", apis };
 #endif
 #endif
-
diff -Naur graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.def graphviz-2.40.1/plugin/pango/gvplugin_pango.def
--- graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/pango/gvplugin_pango.def	2018-01-23 08:29:59.899874600 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_pango"
+EXPORTS
+gvplugin_pango_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/pango/Makefile.am graphviz-2.40.1/plugin/pango/Makefile.am
--- graphviz-2.40.1-orig/plugin/pango/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/pango/Makefile.am	2018-01-23 08:30:27.883759100 +0300
@@ -35,7 +35,7 @@
 	$(top_builddir)/lib/gvc/libgvc.la @PANGOCAIRO_LIBS@ @PANGOFT2_LIBS@ @MATH_LIBS@
 
 if WITH_WIN32
-libgvplugin_pango_la_LDFLAGS += -no-undefined
+libgvplugin_pango_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/pango/gvplugin_pango.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/poppler/gvplugin_poppler.c graphviz-2.40.1/plugin/poppler/gvplugin_poppler.c
--- graphviz-2.40.1-orig/plugin/poppler/gvplugin_poppler.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/poppler/gvplugin_poppler.c	2018-01-23 08:31:17.432530300 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvloadimage_poppler_types[];
@@ -20,4 +21,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_POPPLER_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_poppler_LTX_library = { "poppler", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_poppler_LTX_library = { "poppler", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_poppler_LTX_library = { "poppler", apis };
+#else
 gvplugin_library_t gvplugin_poppler_LTX_library = { "poppler", apis };
+#endif
+#endif
\ В конце файла нет новой строки
diff -Naur graphviz-2.40.1-orig/plugin/poppler/gvplugin_poppler.def graphviz-2.40.1/plugin/poppler/gvplugin_poppler.def
--- graphviz-2.40.1-orig/plugin/poppler/gvplugin_poppler.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/poppler/gvplugin_poppler.def	2018-01-23 08:31:28.105759600 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_poppler"
+EXPORTS
+gvplugin_poppler_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/poppler/Makefile.am graphviz-2.40.1/plugin/poppler/Makefile.am
--- graphviz-2.40.1-orig/plugin/poppler/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/poppler/Makefile.am	2018-01-23 08:31:46.733830400 +0300
@@ -31,7 +31,7 @@
 	$(top_builddir)/lib/gvc/libgvc.la $(PANGOCAIRO_LIBS) $(POPPLER_LIBS)
 
 if WITH_WIN32
-libgvplugin_poppler_la_LDFLAGS += -no-undefined
+libgvplugin_poppler_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/poppler/gvplugin_poppler.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/rsvg/gvplugin_rsvg.c graphviz-2.40.1/plugin/rsvg/gvplugin_rsvg.c
--- graphviz-2.40.1-orig/plugin/rsvg/gvplugin_rsvg.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/rsvg/gvplugin_rsvg.c	2018-01-23 08:33:33.619596600 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvloadimage_rsvg_types[];
@@ -19,9 +20,17 @@
     {API_loadimage, gvloadimage_rsvg_types},
     {(api_t)0, 0},
 };
+
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_RSVG_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_rsvg_LTX_library = { "rsvg", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_rsvg_LTX_library = { "rsvg", apis };
+#endif
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_rsvg_LTX_library = { "rsvg", apis };
 #else
 gvplugin_library_t gvplugin_rsvg_LTX_library = { "rsvg", apis };
 #endif
-
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/rsvg/gvplugin_rsvg.def graphviz-2.40.1/plugin/rsvg/gvplugin_rsvg.def
--- graphviz-2.40.1-orig/plugin/rsvg/gvplugin_rsvg.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/rsvg/gvplugin_rsvg.def	2018-01-23 08:33:16.327343400 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_rsvg"
+EXPORTS
+gvplugin_rsvg_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/rsvg/Makefile.am graphviz-2.40.1/plugin/rsvg/Makefile.am
--- graphviz-2.40.1-orig/plugin/rsvg/Makefile.am	2016-12-05 02:18:35.000000000 +0300
+++ graphviz-2.40.1/plugin/rsvg/Makefile.am	2018-01-23 08:33:49.681659100 +0300
@@ -30,7 +30,7 @@
 	@RSVG_LIBS@ @PANGOCAIRO_LIBS@ @XRENDER_LIBS@
 
 if WITH_WIN32
-libgvplugin_rsvg_la_LDFLAGS += -no-undefined
+libgvplugin_rsvg_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/rsvg/gvplugin_rsvg.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.c graphviz-2.40.1/plugin/visio/gvplugin_visio.c
--- graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/visio/gvplugin_visio.c	2018-01-23 08:35:15.453312500 +0300
@@ -21,7 +21,7 @@
     #pragma comment( lib, "graph.lib" )
 #endif
 
-
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_vdx_types[];
@@ -33,13 +33,13 @@
     {(api_t)0, 0},
 };
 
-#ifdef WIN32_DLL   /*visual studio*/
+#ifdef _MSC_VER
 #ifndef GVPLUGIN_VISIO_EXPORTS
 __declspec(dllimport) gvplugin_library_t gvplugin_visio_LTX_library = { "visio", apis };
 #else
 __declspec(dllexport) gvplugin_library_t gvplugin_visio_LTX_library = { "visio", apis };
 #endif
-#else /*end visual studio*/
+#else
 #ifdef GVDLL
 __declspec(dllexport) gvplugin_library_t gvplugin_visio_LTX_library = { "visio", apis };
 #else
diff -Naur graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.def graphviz-2.40.1/plugin/visio/gvplugin_visio.def
--- graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/visio/gvplugin_visio.def	2018-01-23 08:34:56.547260400 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_visio"
+EXPORTS
+gvplugin_visio_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/visio/Makefile.am graphviz-2.40.1/plugin/visio/Makefile.am
--- graphviz-2.40.1-orig/plugin/visio/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/visio/Makefile.am	2018-01-23 08:35:31.750961700 +0300
@@ -34,7 +34,7 @@
 	$(top_builddir)/lib/gvc/libgvc.la @VISIO_LIBS@ 
 
 if WITH_WIN32
-libgvplugin_visio_la_LDFLAGS += -no-undefined
+libgvplugin_visio_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/visio/gvplugin_visio.def
 endif
 
 if WITH_DARWIN9
diff -Naur graphviz-2.40.1-orig/plugin/webp/gvplugin_webp.c graphviz-2.40.1/plugin/webp/gvplugin_webp.c
--- graphviz-2.40.1-orig/plugin/webp/gvplugin_webp.c	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/webp/gvplugin_webp.c	2018-01-23 08:36:33.828777100 +0300
@@ -11,6 +11,7 @@
  * Contributors: See CVS logs. Details at http://www.graphviz.org/
  *************************************************************************/
 
+#include "config.h"
 #include "gvplugin.h"
 
 extern gvplugin_installed_t gvdevice_webp_types[];
@@ -22,4 +23,16 @@
     {(api_t)0, 0},
 };
 
+#ifdef _MSC_VER
+#ifndef GVPLUGIN_WEBP_EXPORTS
+__declspec(dllimport) gvplugin_library_t gvplugin_webp_LTX_library = { "webp", apis };
+#else
+__declspec(dllexport) gvplugin_library_t gvplugin_webp_LTX_library = { "webp", apis };
+#endif
+#else
+#ifdef GVDLL
+__declspec(dllexport) gvplugin_library_t gvplugin_webp_LTX_library = { "webp", apis };
+#else
 gvplugin_library_t gvplugin_webp_LTX_library = { "webp", apis };
+#endif
+#endif
diff -Naur graphviz-2.40.1-orig/plugin/webp/gvplugin_webp.def graphviz-2.40.1/plugin/webp/gvplugin_webp.def
--- graphviz-2.40.1-orig/plugin/webp/gvplugin_webp.def	1970-01-01 03:00:00.000000000 +0300
+++ graphviz-2.40.1/plugin/webp/gvplugin_webp.def	2018-01-23 08:36:20.911122400 +0300
@@ -0,0 +1,5 @@
+LIBRARY	"gvplugin_webp"
+EXPORTS
+gvplugin_webp_LTX_library	
+
+
diff -Naur graphviz-2.40.1-orig/plugin/webp/Makefile.am graphviz-2.40.1/plugin/webp/Makefile.am
--- graphviz-2.40.1-orig/plugin/webp/Makefile.am	2016-08-10 00:02:10.000000000 +0300
+++ graphviz-2.40.1/plugin/webp/Makefile.am	2018-01-23 08:36:53.207895300 +0300
@@ -29,7 +29,7 @@
 libgvplugin_webp_la_LIBADD = @WEBP_LIBS@ @PANGOCAIRO_LIBS@
 
 if WITH_WIN32
-libgvplugin_webp_la_LDFLAGS += -no-undefined
+libgvplugin_webp_la_LDFLAGS += -no-undefined -export-symbols $(top_srcdir)/plugin/webp/gvplugin_webp.def
 endif
 
 if WITH_DARWIN9
