diff -Naur graphviz-2.44.1.a/cmd/smyrna/main.c graphviz-2.44.1.b/cmd/smyrna/main.c
--- graphviz-2.44.1.a/cmd/smyrna/main.c	2020-06-29 10:47:18.000000000 +0200
+++ graphviz-2.44.1.b/cmd/smyrna/main.c	2020-11-27 03:03:01.433414500 +0100
@@ -210,7 +210,7 @@
     create_window(glconfig, gladewidget);
     change_cursor(GDK_TOP_LEFT_ARROW);
 
-#ifndef _WIN32
+#ifdef _WIN32 /* yes this is still needed gah */
     glutInit(&argc, argv);
 #endif
 
