diff -Naur graphviz-2.40.1-orig/lib/cdt/cdt.def graphviz-2.40.1/lib/cdt/cdt.def
--- graphviz-2.40.1-orig/lib/cdt/cdt.def	2018-01-26 21:42:30.602237100 +0300
+++ graphviz-2.40.1/lib/cdt/cdt.def	2018-01-26 21:50:54.406053000 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"cdt"
+LIBRARY	"libcdt-5"
 
 EXPORTS
 Dtbag	
diff -Naur graphviz-2.40.1-orig/lib/cgraph/cgraph.def graphviz-2.40.1/lib/cgraph/cgraph.def
--- graphviz-2.40.1-orig/lib/cgraph/cgraph.def	2018-01-26 21:42:30.610237500 +0300
+++ graphviz-2.40.1/lib/cgraph/cgraph.def	2018-01-26 21:50:45.038517200 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"cgraph"
+LIBRARY	"libcgraph-6"
 EXPORTS
 Ag_G_global	
 Ag_mainedge_id_disc	
diff -Naur graphviz-2.40.1-orig/lib/edgepaint/lab_gamut.def graphviz-2.40.1/lib/edgepaint/lab_gamut.def
--- graphviz-2.40.1-orig/lib/edgepaint/lab_gamut.def	2018-01-26 21:42:07.550918600 +0300
+++ graphviz-2.40.1/lib/edgepaint/lab_gamut.def	2018-01-26 21:54:31.849490100 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"lab_gamut"
+LIBRARY	"liblab_gamut-1"
 
 
 EXPORTS
diff -Naur graphviz-2.40.1-orig/lib/graph/graph.def graphviz-2.40.1/lib/graph/graph.def
--- graphviz-2.40.1-orig/lib/graph/graph.def	2018-01-26 21:42:08.248958500 +0300
+++ graphviz-2.40.1/lib/graph/graph.def	2018-01-26 21:56:11.380182900 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"graph"
+LIBRARY	"libgraph-5"
 EXPORTS
 AG	
 agattr	
diff -Naur graphviz-2.40.1-orig/lib/graph/Makefile.am graphviz-2.40.1/lib/graph/Makefile.am
--- graphviz-2.40.1-orig/lib/graph/Makefile.am	2018-01-26 21:42:08.192955300 +0300
+++ graphviz-2.40.1/lib/graph/Makefile.am	2018-01-26 21:58:27.498968400 +0300
@@ -26,6 +26,10 @@
 libgraph_la_SOURCES = $(libgraph_C_la_SOURCES)
 libgraph_la_LIBADD = $(top_builddir)/lib/cdt/libcdt.la
 
+if WITH_WIN32
+libgvc_la_LDFLAGS += -export-symbols $(top_srcdir)/lib/graph/graph.def
+endif
+
 lexer.o lexer.lo trie.o trie.lo: parser.h
 
 parser.c: y.tab.c
diff -Naur graphviz-2.40.1-orig/lib/gvc.def graphviz-2.40.1/lib/gvc.def
--- graphviz-2.40.1-orig/lib/gvc.def	2018-01-26 21:43:31.455717700 +0300
+++ graphviz-2.40.1/lib/gvc.def	2018-01-26 21:54:41.418037300 +0300
@@ -1,4 +1,4 @@
-LIBRARY    "gvc"
+LIBRARY    "libgvc-6"
 
 EXPORTS
 add_box    
diff -Naur graphviz-2.40.1-orig/lib/pathplan/pathplan.def graphviz-2.40.1/lib/pathplan/pathplan.def
--- graphviz-2.40.1-orig/lib/pathplan/pathplan.def	2018-01-26 21:42:10.411082200 +0300
+++ graphviz-2.40.1/lib/pathplan/pathplan.def	2018-01-26 21:49:37.814672200 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"Pathplan"
+LIBRARY	"libpathplan-4"
 
 
 EXPORTS
diff -Naur graphviz-2.40.1-orig/plugin/core/gvplugin_core.def graphviz-2.40.1/plugin/core/gvplugin_core.def
--- graphviz-2.40.1-orig/plugin/core/gvplugin_core.def	2018-01-26 21:42:31.419283800 +0300
+++ graphviz-2.40.1/plugin/core/gvplugin_core.def	2018-01-26 21:50:12.111633900 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_core"
+LIBRARY	"libgvplugin_core-6"
 EXPORTS
 gvplugin_core_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/devil/gvplugin_devil.def graphviz-2.40.1/plugin/devil/gvplugin_devil.def
--- graphviz-2.40.1-orig/plugin/devil/gvplugin_devil.def	2018-01-26 21:43:29.767621100 +0300
+++ graphviz-2.40.1/plugin/devil/gvplugin_devil.def	2018-01-26 21:54:09.228196200 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_devil"
+LIBRARY	"libgvplugin_devil-6"
 EXPORTS
 gvplugin_devil_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/dot_layout/gvplugin_dot_layout.def graphviz-2.40.1/plugin/dot_layout/gvplugin_dot_layout.def
--- graphviz-2.40.1-orig/plugin/dot_layout/gvplugin_dot_layout.def	2018-01-26 21:42:32.563349200 +0300
+++ graphviz-2.40.1/plugin/dot_layout/gvplugin_dot_layout.def	2018-01-26 21:51:40.822707900 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_dot_layout"
+LIBRARY	"libgvplugin_dot_layout-6"
 EXPORTS
 gvplugin_dot_layout_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/gd/gvplugin_gd.def graphviz-2.40.1/plugin/gd/gvplugin_gd.def
--- graphviz-2.40.1-orig/plugin/gd/gvplugin_gd.def	2018-01-26 21:42:32.919369600 +0300
+++ graphviz-2.40.1/plugin/gd/gvplugin_gd.def	2018-01-26 21:51:27.024918700 +0300
@@ -1,3 +1,3 @@
-LIBRARY	"gvplugin_gd"
+LIBRARY	"libgvplugin_gd-6"
 EXPORTS
 gvplugin_gd_LTX_library
diff -Naur graphviz-2.40.1-orig/plugin/gdiplus/gvplugin_gdiplus.def graphviz-2.40.1/plugin/gdiplus/gvplugin_gdiplus.def
--- graphviz-2.40.1-orig/plugin/gdiplus/gvplugin_gdiplus.def	2018-01-26 21:42:33.413397900 +0300
+++ graphviz-2.40.1/plugin/gdiplus/gvplugin_gdiplus.def	2018-01-26 21:51:15.258245700 +0300
@@ -1,3 +1,3 @@
-LIBRARY	"gvplugin_gdiplus"
+LIBRARY	"libgvplugin_gdiplus-6"
 EXPORTS
 gvplugin_gdiplus_LTX_library 
diff -Naur graphviz-2.40.1-orig/plugin/gdk/gvplugin_gdk.def graphviz-2.40.1/plugin/gdk/gvplugin_gdk.def
--- graphviz-2.40.1-orig/plugin/gdk/gvplugin_gdk.def	2018-01-26 21:43:29.794622700 +0300
+++ graphviz-2.40.1/plugin/gdk/gvplugin_gdk.def	2018-01-26 21:53:58.037556100 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_gdk"
+LIBRARY	"libgvplugin_gdk-6"
 EXPORTS
 gvplugin_gdk_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/glitz/gvplugin_glitz.def graphviz-2.40.1/plugin/glitz/gvplugin_glitz.def
--- graphviz-2.40.1-orig/plugin/glitz/gvplugin_glitz.def	2018-01-26 21:43:29.801623100 +0300
+++ graphviz-2.40.1/plugin/glitz/gvplugin_glitz.def	2018-01-26 21:53:44.518782900 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_glitz"
+LIBRARY	"libgvplugin_glitz-6"
 EXPORTS
 gvplugin_glitz_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/gs/gvplugin_gs.def graphviz-2.40.1/plugin/gs/gvplugin_gs.def
--- graphviz-2.40.1-orig/plugin/gs/gvplugin_gs.def	2018-01-26 21:43:29.809623500 +0300
+++ graphviz-2.40.1/plugin/gs/gvplugin_gs.def	2018-01-26 21:53:34.040183600 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_gs"
+LIBRARY	"libgvplugin_gs-6"
 EXPORTS
 gvplugin_gs_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/gtk/gvplugin_gtk.def graphviz-2.40.1/plugin/gtk/gvplugin_gtk.def
--- graphviz-2.40.1-orig/plugin/gtk/gvplugin_gtk.def	2018-01-26 21:43:29.816623900 +0300
+++ graphviz-2.40.1/plugin/gtk/gvplugin_gtk.def	2018-01-26 21:53:24.016610200 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_gtk"
+LIBRARY	"libgvplugin_gtk-6"
 EXPORTS
 gvplugin_gtk_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/lasi/gvplugin_lasi.def graphviz-2.40.1/plugin/lasi/gvplugin_lasi.def
--- graphviz-2.40.1-orig/plugin/lasi/gvplugin_lasi.def	2018-01-26 21:43:29.823624300 +0300
+++ graphviz-2.40.1/plugin/lasi/gvplugin_lasi.def	2018-01-26 21:53:11.720907000 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_lasi"
+LIBRARY	"libgvplugin_lasi-6"
 EXPORTS
 gvplugin_lasi_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/ming/gvplugin_ming.def graphviz-2.40.1/plugin/ming/gvplugin_ming.def
--- graphviz-2.40.1-orig/plugin/ming/gvplugin_ming.def	2018-01-26 21:43:29.831624800 +0300
+++ graphviz-2.40.1/plugin/ming/gvplugin_ming.def	2018-01-26 21:52:57.457091100 +0300
@@ -1,5 +1,5 @@
-LIBRARY	"gvplugin_ming"
-EXPORTS
+LIBRARY	"gvplugin_ming-6"
+EXPORTSlib
 gvplugin_ming_LTX_library	
 
 
diff -Naur graphviz-2.40.1-orig/plugin/neato_layout/gvplugin_neato_layout.def graphviz-2.40.1/plugin/neato_layout/gvplugin_neato_layout.def
--- graphviz-2.40.1-orig/plugin/neato_layout/gvplugin_neato_layout.def	2018-01-26 21:42:34.016432400 +0300
+++ graphviz-2.40.1/plugin/neato_layout/gvplugin_neato_layout.def	2018-01-26 21:51:52.450373000 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_neato_layout"
+LIBRARY	"libgvplugin_neato_layout-6"
 EXPORTS
 allow_edits	
 checkStart	
diff -Naur graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.def graphviz-2.40.1/plugin/pango/gvplugin_pango.def
--- graphviz-2.40.1-orig/plugin/pango/gvplugin_pango.def	2018-01-26 21:43:29.842625400 +0300
+++ graphviz-2.40.1/plugin/pango/gvplugin_pango.def	2018-01-26 21:52:46.635472200 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_pango"
+LIBRARY	"libgvplugin_pango-6"
 EXPORTS
 gvplugin_pango_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/poppler/gvplugin_poppler.def graphviz-2.40.1/plugin/poppler/gvplugin_poppler.def
--- graphviz-2.40.1-orig/plugin/poppler/gvplugin_poppler.def	2018-01-26 21:43:29.849625800 +0300
+++ graphviz-2.40.1/plugin/poppler/gvplugin_poppler.def	2018-01-26 21:52:37.363941900 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_poppler"
+LIBRARY	"libgvplugin_poppler-6"
 EXPORTS
 gvplugin_poppler_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/rsvg/gvplugin_rsvg.def graphviz-2.40.1/plugin/rsvg/gvplugin_rsvg.def
--- graphviz-2.40.1-orig/plugin/rsvg/gvplugin_rsvg.def	2018-01-26 21:43:29.856626200 +0300
+++ graphviz-2.40.1/plugin/rsvg/gvplugin_rsvg.def	2018-01-26 21:52:24.827224800 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_rsvg"
+LIBRARY	"libgvplugin_rsvg-6"
 EXPORTS
 gvplugin_rsvg_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.def graphviz-2.40.1/plugin/visio/gvplugin_visio.def
--- graphviz-2.40.1-orig/plugin/visio/gvplugin_visio.def	2018-01-26 21:43:29.863626600 +0300
+++ graphviz-2.40.1/plugin/visio/gvplugin_visio.def	2018-01-26 21:52:14.322624000 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_visio"
+LIBRARY	"libgvplugin_visio-6"
 EXPORTS
 gvplugin_visio_LTX_library	
 
diff -Naur graphviz-2.40.1-orig/plugin/webp/gvplugin_webp.def graphviz-2.40.1/plugin/webp/gvplugin_webp.def
--- graphviz-2.40.1-orig/plugin/webp/gvplugin_webp.def	2018-01-26 21:43:29.870627000 +0300
+++ graphviz-2.40.1/plugin/webp/gvplugin_webp.def	2018-01-26 21:52:03.058979700 +0300
@@ -1,4 +1,4 @@
-LIBRARY	"gvplugin_webp"
+LIBRARY	"libgvplugin_webp-6"
 EXPORTS
 gvplugin_webp_LTX_library	
 
