diff -Naur graphviz-2.44.1-orig/lib/sfio/sfwrite.c graphviz-2.44.1/lib/sfio/sfwrite.c
--- graphviz-2.44.1-orig/lib/sfio/sfwrite.c	2020-06-29 11:47:18.000000000 +0300
+++ graphviz-2.44.1/lib/sfio/sfwrite.c	2020-10-30 13:16:53.424084700 +0300
@@ -48,13 +48,13 @@
 	f->mode &= ~SF_PEEK;
 
 	if (f->mode & SF_PKRD) {	/* read past peeked data */
-	    char buf[16];
+	    char tmpbuf[16];
 	    reg ssize_t r;
 
 	    for (w = n; w > 0;) {
-		if ((r = w) > sizeof(buf))
-		    r = sizeof(buf);
-		if ((r = read(f->file, buf, r)) <= 0) {
+		if ((r = w) > sizeof(tmpbuf))
+		    r = sizeof(tmpbuf);
+		if ((r = read(f->file, tmpbuf, r)) <= 0) {
 		    n -= w;
 		    break;
 		} else