diff -Naur graphviz-2.44.1-orig/lib/sfio/sfhdr.h graphviz-2.44.1/lib/sfio/sfhdr.h
--- graphviz-2.44.1-orig/lib/sfio/sfhdr.h	2020-06-29 11:47:18.000000000 +0300
+++ graphviz-2.44.1/lib/sfio/sfhdr.h	2020-10-30 13:16:53.411119900 +0300
@@ -774,7 +774,9 @@
 #endif /*HAVE_UNISTD_H*/
 #endif /* _WIN32 */
 
+#ifndef WIN32
     extern time_t time(time_t *);
+#endif
     extern int waitpid(int, int *, int);
 #ifndef _WIN32
     extern void _exit(int);
