--- a/builder/cc_builder.rs
+++ b/builder/cc_builder.rs
@@ -393,10 +393,16 @@ impl CcBuilder {
 
     #[allow(clippy::unused_self)]
     fn add_defines(&self, build_options: &mut Vec<BuildOption>, is_like_msvc: bool) {
-        if is_like_msvc {
-            build_options.push(BuildOption::define("_HAS_EXCEPTIONS", "0"));
+        // WIN32_LEAN_AND_MEAN and NOMINMAX are needed for all Windows targets to avoid
+        // header type definition errors, no matter the compiler. This matches the behavior
+        // in aws-lc/CMakeLists.txt, which defines these for all WIN32 targets
+        if target_os() == "windows" {
             build_options.push(BuildOption::define("WIN32_LEAN_AND_MEAN", ""));
             build_options.push(BuildOption::define("NOMINMAX", ""));
+        }
+
+        if is_like_msvc {
+            build_options.push(BuildOption::define("_HAS_EXCEPTIONS", "0"));
             build_options.push(BuildOption::define("_CRT_SECURE_NO_WARNINGS", "0"));
             build_options.push(BuildOption::define(
                 "_STL_EXTRA_DISABLED_WARNINGS",
