--- configure.ac	0.10.36
+++ configure.ac	fixed
@@ -740,6 +740,8 @@
     "$GST_PLUGIN_SCANNER_INSTALLED", [location of the installed gst-plugin-scanner])
 AC_SUBST(GST_PLUGIN_SCANNER_INSTALLED)
 
+AC_DEFINE(GLIB_DISABLE_DEPRECATION_WARNINGS, , [Disable GLib deprecation warnings])
+
 dnl things for our internal libcheck (must be called even if building
 dnl libcheck is disabled because it defines conditionals)
 AG_GST_CHECK_CHECKS()
--- gst/gstpoll.c	0.10.36
+++ gst/gstpoll.c	fixed
@@ -72,7 +72,9 @@
 
 #ifdef G_OS_WIN32
 #include <winsock2.h>
-#define EINPROGRESS WSAEINPROGRESS
+#ifndef EINPROGRESS
+#  define EINPROGRESS WSAEINPROGRESS
+#endif
 #else
 #define _GNU_SOURCE 1
 #ifdef HAVE_SYS_POLL_H
--- gst/gststructure.c	0.10.36
+++ gst/gststructure.c	fixed
@@ -2672,7 +2672,7 @@
   guint _flags = (flags);                                                     \
   GType _value_type = G_VALUE_TYPE (_value);                                  \
   GTypeValueTable *_vtable = g_type_value_table_peek (_value_type);           \
-  gchar *_lcopy_format = _vtable->lcopy_format;                               \
+  const gchar *_lcopy_format = _vtable->lcopy_format;                         \
   GTypeCValue _cvalues[G_VALUE_COLLECT_FORMAT_MAX_LENGTH] = { { 0, }, };      \
   guint _n_values = 0;                                                        \
                                                                               \
--- gst/gstsystemclock.c	0.10.36
+++ gst/gstsystemclock.c	fixed
@@ -52,7 +52,9 @@
 #  define WIN32_LEAN_AND_MEAN   /* prevents from including too many things */
 #  include <windows.h>          /* QueryPerformance* stuff */
 #  undef WIN32_LEAN_AND_MEAN
-#  define EWOULDBLOCK EAGAIN    /* This is just to placate gcc */
+#  ifndef EWOULDBLOCK
+#    define EWOULDBLOCK EAGAIN    /* This is just to placate gcc */
+#  endif
 #endif /* G_OS_WIN32 */
 
 #define GET_ENTRY_STATUS(e)          ((GstClockReturn) g_atomic_int_get(&GST_CLOCK_ENTRY_STATUS(e)))
--- plugins/elements/gstfdsrc.c	0.10.36
+++ plugins/elements/gstfdsrc.c	fixed
@@ -408,22 +408,22 @@
   GstBuffer *buf;
   gssize readbytes;
   guint blocksize;
-  GstClockTime timeout;
 
 #ifndef HAVE_WIN32
+  GstClockTime timeout;
   gboolean try_again;
   gint retval;
 #endif
 
   src = GST_FD_SRC (psrc);
 
+#ifndef HAVE_WIN32
   if (src->timeout > 0) {
     timeout = src->timeout * GST_USECOND;
   } else {
     timeout = GST_CLOCK_TIME_NONE;
   }
 
-#ifndef HAVE_WIN32
   do {
     try_again = FALSE;
 
--- plugins/elements/gstfilesrc.c	0.10.36
+++ plugins/elements/gstfilesrc.c	fixed
@@ -46,7 +46,9 @@
 #include <io.h>                 /* lseek, open, close, read */
 /* On win32, stat* default to 32 bit; we need the 64-bit
  * variants, so explicitly define it that way. */
+#undef stat
 #define stat __stat64
+#undef fstat
 #define fstat _fstat64
 #undef lseek
 #define lseek _lseeki64
