unchanged:
--- libgdiplus-5.4/src/Makefile-orig.am	2017-11-10 15:58:17.262916800 -0500
+++ libgdiplus-5.4/src/Makefile.am	2017-11-10 15:58:45.805357800 -0500
@@ -122,3 +122,5 @@
 libgdiplus_la_LIBADD = $(GDIPLUS_LIBS)
 
 INCLUDES = $(GDIPLUS_CFLAGS) -Wall -Wno-unused -Wno-format
+
+libgdiplus_la_LDFLAGS = -no-undefined
only in patch2:
unchanged:
--- libgdiplus-5.4/configure-orig.ac	2017-11-10 15:57:31.480608900 -0500
+++ libgdiplus-5.4/configure.ac	2017-11-10 15:57:40.720647900 -0500
@@ -8,6 +8,7 @@
 AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([config.h])
 AM_MAINTAINER_MODE
+AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 
 AC_C_BIGENDIAN
