--- libxslt-1.1.32/configure.ac.orig	2017-11-01 09:46:32.000000000 +0100
+++ libxslt-1.1.32/configure.ac	2017-11-23 22:23:38.273437800 +0100
@@ -591,10 +591,15 @@
  AC_DEFINE([_WINSOCKAPI_],1,[Using the Win32 Socket implementation])
  AC_DEFINE([snprintf],[_snprintf],[Win32 Std C name mangling work-around])
  AC_DEFINE([vsnprintf],[_vsnprintf],[Win32 Std C name mangling work-around])
+ if test "x${PYTHON}" != "x"
+ then
+   WIN32_EXTRA_PYTHON_LIBADD="-shrext .pyd " #-L${pythondir}/../../libs -lpython$(echo ${PYTHON_VERSION} | tr -d .)
+ fi
  ;;
 esac
 AC_SUBST(WIN32_EXTRA_LIBADD)
 AC_SUBST(WIN32_EXTRA_LDFLAGS)
+AC_SUBST(WIN32_EXTRA_PYTHON_LIBADD)
 
 
 AC_SUBST(XSLTPROCDV)

--- libxslt-1.1.32/python/Makefile.am.orig	2015-05-10 16:11:30.000000000 +0200
+++ libxslt-1.1.32/python/Makefile.am	2017-11-23 22:26:44.469861000 +0100
@@ -37,7 +37,7 @@
 	-I../libexslt
 libxsltmod_la_SOURCES = libxslt.c types.c
 nodist_libxsltmod_la_SOURCES = libxslt-py.c
-libxsltmod_la_LIBADD = $(mylibs) $(PYTHON_LIBS)
+libxsltmod_la_LIBADD = $(mylibs) $(PYTHON_LIBS) @WIN32_EXTRA_PYTHON_LIBADD@
 
 libxslt.py: $(srcdir)/libxsl.py libxsltclass.py
 	cat $(srcdir)/libxsl.py libxsltclass.py > $@
