diff -Naur libxslt-1.1.28-orig/configure.in libxslt-1.1.28/configure.in
--- libxslt-1.1.28-orig/configure.in	2014-04-07 08:54:05.401200000 +0400
+++ libxslt-1.1.28/configure.in	2014-04-07 09:01:04.362600000 +0400
@@ -629,9 +629,6 @@
  *-*-mingw*)
  WIN32_EXTRA_LIBADD="-lws2_32 -lmswsock"
  WIN32_EXTRA_LDFLAGS="-no-undefined"
- AC_DEFINE([_WINSOCKAPI_],1,[Using the Win32 Socket implementation])
- AC_DEFINE([snprintf],[_snprintf],[Win32 Std C name mangling work-around])
- AC_DEFINE([vsnprintf],[_vsnprintf],[Win32 Std C name mangling work-around])
  if test "x${PYTHON}" != "x"
  then
    WIN32_EXTRA_PYTHON_LIBADD="-shrext .pyd " #-L${pythondir}/../../libs -lpython$(echo ${PYTHON_VERSION} | tr -d .)
diff -Naur libxslt-1.1.28-orig/libxslt/win32config.h libxslt-1.1.28/libxslt/win32config.h
--- libxslt-1.1.28-orig/libxslt/win32config.h	2012-09-12 09:58:15.000000000 +0400
+++ libxslt-1.1.28/libxslt/win32config.h	2014-04-07 09:00:54.363000000 +0400
@@ -79,10 +79,6 @@
 #include <direct.h>
 #if defined(_MSC_VER) || defined(__MINGW32__)
 #define mkdir(p,m) _mkdir(p)
-#define snprintf _snprintf
-#if _MSC_VER < 1500
-#define vsnprintf(b,c,f,a) _vsnprintf(b,c,f,a)
-#endif
 #endif
 
 #define HAVE_SYS_STAT_H
diff -Naur libxslt-1.1.28-orig/python/libxslt.c libxslt-1.1.28/python/libxslt.c
--- libxslt-1.1.28-orig/python/libxslt.c	2012-09-12 09:58:38.000000000 +0400
+++ libxslt-1.1.28/python/libxslt.c	2014-04-07 09:00:35.455800000 +0400
@@ -22,9 +22,7 @@
 
 #include <stdio.h>
 
-#if (defined(_MSC_VER) || defined(__MINGW32__)) && !defined(vsnprintf)
-#define vsnprintf(b,c,f,a) _vsnprintf(b,c,f,a)
-#elif defined(XSLT_NEED_TRIO)
+#if defined(XSLT_NEED_TRIO)
 #include "trio.h"
 #define vsnprintf trio_vsnprintf
 #endif
diff -Naur libxslt-1.1.28-orig/xsltproc/xsltproc.c libxslt-1.1.28/xsltproc/xsltproc.c
--- libxslt-1.1.28-orig/xsltproc/xsltproc.c	2012-09-04 18:26:23.000000000 +0400
+++ libxslt-1.1.28/xsltproc/xsltproc.c	2014-04-07 09:00:07.625400000 +0400
@@ -57,7 +57,6 @@
 #if defined(_MSC_VER) || defined(__MINGW32__)
 #include <winsock2.h>
 #define gettimeofday(p1,p2)
-#define snprintf _snprintf
 #endif /* _MS_VER */
 #else /* WIN32 */
 #if defined(HAVE_SYS_TIME_H)
