--- libxslt-1.1.32.orig/libexslt/crypto.c	2017-10-30 08:49:55.000000000 +0100
+++ libxslt-1.1.32/libexslt/crypto.c	2017-11-23 23:19:58.609456500 +0100
@@ -230,8 +230,13 @@
     }
 
     hHash = exsltCryptoCryptoApiCreateHash (ctxt, hCryptProv,
+#ifdef __MINGW32__
+					    CALG_SHA1, (LPVOID) key,
+					    RC4_KEY_LENGTH, (LPVOID) hash,
+#else
 					    CALG_SHA1, key,
 					    RC4_KEY_LENGTH, hash,
+#endif
 					    HASH_DIGEST_LENGTH);
 
     if (!CryptDeriveKey
@@ -281,8 +286,13 @@
     }
 
     hHash = exsltCryptoCryptoApiCreateHash (ctxt, hCryptProv,
+#ifdef __MINGW32__
+					    CALG_SHA1, (LPVOID) key,
+					    RC4_KEY_LENGTH, (LPVOID) hash,
+#else
 					    CALG_SHA1, key,
 					    RC4_KEY_LENGTH, hash,
+#endif
 					    HASH_DIGEST_LENGTH);
 
     if (!CryptDeriveKey
--- libxslt-1.1.32.orig/libexslt/date.c	2017-10-30 08:49:55.000000000 +0100
+++ libxslt-1.1.32/libexslt/date.c	2017-11-23 23:19:58.611965400 +0100
@@ -748,6 +748,14 @@
     time_t secs;
     int local_s, gm_s;
     exsltDateValPtr ret;
+#ifdef __MINGW32__
+    gmTm.tm_year = 0;
+    gmTm.tm_mon = 0;
+    gmTm.tm_mday = 0;
+    gmTm.tm_hour = 0;
+    gmTm.tm_min = 0;
+    gmTm.tm_sec = 0;
+#endif
 #ifdef HAVE_ERRNO_H
     char *source_date_epoch;
 #endif /* HAVE_ERRNO_H */
--- libxslt-1.1.32.orig/libxslt/security.c	2017-10-30 08:49:55.000000000 +0100
+++ libxslt-1.1.32/libxslt/security.c	2017-11-23 23:22:29.111827000 +0100
@@ -387,14 +395,16 @@
 #if defined(_WIN32) && !defined(__CYGWIN__)
     if ((uri->path)&&(uri->path[0]=='/')&&
         (uri->path[1]!='\0')&&(uri->path[2]==':'))
-    ret = xsltCheckWritePath(sec, ctxt, uri->path+1);
+        ret = xsltCheckWritePath(sec, ctxt, uri->path+1);
     else
-#endif
+        ret = xsltCheckWritePath(sec, ctxt, uri->path);
+#else
 
 	/*
 	 * Check if we are allowed to write this file
 	 */
 	ret = xsltCheckWritePath(sec, ctxt, uri->path);
+#endif
 	if (ret <= 0) {
 	    xmlFreeURI(uri);
 	    return(ret);
