diff --git a/PKG-INFO b/PKG-INFO.modified
index da2af3e..132b16b 100644
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -29,7 +29,6 @@ Classifier: Programming Language :: Python :: Implementation :: CPython
 Classifier: Topic :: Scientific/Engineering :: Mathematics
 Requires-Python: <3.15,>=3.10
 Description-Content-Type: text/x-rst
-Requires-Dist: cysignals<1.12.4; sys_platform == "win32"
 Requires-Dist: cysignals!=1.12.0,>=1.11.2
 Provides-Extra: test
 Requires-Dist: passagemath-graphs; extra == "test"
diff --git a/pyproject.toml b/pyproject.toml.modified
index a60bce7..7810941 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ requires = [
     'passagemath-environment ~= 10.6.39.0',
     'passagemath-objects ~= 10.6.39.0',
     'cython >=3.0.8, <3.2.0', 'cython >=3.0.8,<3.2.0',
-    'cysignals <1.12.4; sys_platform == "win32"', 'cysignals >=1.11.2, != 1.12.0',
+    'cysignals >=1.11.2, != 1.12.0',
     'pkgconfig',
 ]
 build-backend = "setuptools.build_meta"
@@ -16,7 +16,7 @@ build-backend = "setuptools.build_meta"
 name = "passagemath-homfly"
 description = "passagemath: Homfly polynomials of knots/links with libhomfly"
 dependencies = [
-    'cysignals <1.12.4; sys_platform == "win32"', 'cysignals >=1.11.2, != 1.12.0',
+    'cysignals >=1.11.2, != 1.12.0',
 ]
 dynamic = ["version"]
 license = "GPL-2.0-or-later"
diff --git a/passagemath_homfly.egg-info/requires.txt b/passagemath_homfly.egg-info/requires.txt.modified
index 38f48ce..c252984 100644
--- a/passagemath_homfly.egg-info/requires.txt
+++ b/passagemath_homfly.egg-info/requires.txt
@@ -1,8 +1,5 @@
 cysignals!=1.12.0,>=1.11.2

-[:sys_platform == "win32"]
-cysignals<1.12.4
-
 [test]
 passagemath-graphs
 passagemath-repl
diff --git a/passagemath_homfly.egg-info/PKG-INFO b/passagemath_homfly.egg-info/PKG-INFO.modified
index da2af3e..132b16b 100644
--- a/passagemath_homfly.egg-info/PKG-INFO
+++ b/passagemath_homfly.egg-info/PKG-INFO
@@ -29,7 +29,6 @@ Classifier: Programming Language :: Python :: Implementation :: CPython
 Classifier: Topic :: Scientific/Engineering :: Mathematics
 Requires-Python: <3.15,>=3.10
 Description-Content-Type: text/x-rst
-Requires-Dist: cysignals<1.12.4; sys_platform == "win32"
 Requires-Dist: cysignals!=1.12.0,>=1.11.2
 Provides-Extra: test
 Requires-Dist: passagemath-graphs; extra == "test"
