--- libdsm-0.3.0/compat/compat.h.orig	2018-05-29 11:28:34.233160300 +0300
+++ libdsm-0.3.0/compat/compat.h	2018-05-29 11:28:46.869182400 +0300
@@ -67,6 +67,7 @@
 
 #if !defined(HAVE_PIPE) && defined(HAVE__PIPE)
 #define HAVE_PIPE
+#define NOTHAVE_PIPE
 int pipe(int fds[2]);
 #endif
 
--- libdsm-0.3.0/compat/compat.c.orig	2018-05-29 11:29:42.490880700 +0300
+++ libdsm-0.3.0/compat/compat.c	2018-05-29 11:29:50.353294500 +0300
@@ -1,6 +1,6 @@
 #include "config.h"
 
-#if !defined(HAVE_PIPE) && defined(HAVE__PIPE)
+#if defined(NOTHAVE_PIPE) && defined(HAVE__PIPE)
 
 #include <fcntl.h>
 
