diff -bur sqlitestudio-orig/SQLiteStudio3/common.pri sqlitestudio/SQLiteStudio3/common.pri
--- sqlitestudio-orig/SQLiteStudio3/common.pri	2024-08-23 00:52:12.631874200 -0600
+++ sqlitestudio/SQLiteStudio3/common.pri	2024-08-23 00:55:18.873484800 -0600
@@ -1,7 +1,7 @@
-OUTPUT_DIR_NAME = output
+OUTPUT_DIR_NAME = ../build-$$(MSYSTEM)
 export(OUTPUT_DIR_NAME)
 
-DESTDIR = $$PWD/../$$OUTPUT_DIR_NAME/SQLiteStudio
+DESTDIR = $$PWD/../$$OUTPUT_DIR_NAME/bin
 OBJECTS_DIR = $$PWD/../$$OUTPUT_DIR_NAME/build
 MOC_DIR = $$PWD/../$$OUTPUT_DIR_NAME/build
 UI_DIR = $$PWD/../$$OUTPUT_DIR_NAME/build
