diff -bur sqlitestudio-orig/SQLiteStudio3/sqlitestudiocli/sqlitestudiocli.pro sqlitestudio/SQLiteStudio3/sqlitestudiocli/sqlitestudiocli.pro
--- sqlitestudio-orig/SQLiteStudio3/sqlitestudiocli/sqlitestudiocli.pro	2025-01-31 00:55:26.969138700 -0700
+++ sqlitestudio/SQLiteStudio3/sqlitestudiocli/sqlitestudiocli.pro	2025-01-31 00:55:44.404002900 -0700
@@ -64,7 +64,7 @@
 
 win32: {
     INCLUDEPATH += $$PWD/../../../include
-    LIBS += -L$$PWD/../../../lib -ledit_static
+    LIBS += -L$$PWD/../../../lib -ledit
 }
 
 unix: {
