From 7640ff2320c061d26b94e50051e4992dea3b044e Mon Sep 17 00:00:00 2001
From: Hernan Martinez <hernan.c.martinez@gmail.com>
Date: Fri, 31 Jan 2025 14:54:02 -0700
Subject: [PATCH] Enable C++ deprecated exceptions handlers when building with
 modern Clang

---
 SQLiteStudio3/coreSQLiteStudio/coreSQLiteStudio.pro | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SQLiteStudio3/coreSQLiteStudio/coreSQLiteStudio.pro b/SQLiteStudio3/coreSQLiteStudio/coreSQLiteStudio.pro
index f040741fc8..31c54063c6 100644
--- a/SQLiteStudio3/coreSQLiteStudio/coreSQLiteStudio.pro
+++ b/SQLiteStudio3/coreSQLiteStudio/coreSQLiteStudio.pro
@@ -28,6 +28,10 @@ win32: {
         THE_DEST ~= s,/,\\,g
         QMAKE_POST_LINK += "$$QMAKE_COPY $$THE_FILE $$THE_DEST $$escape_expand(\\n\\t);"
     }
+
+    equals(QMAKE_CXX, clang++) {
+        QMAKE_CXXFLAGS += -D_LIBCPP_ENABLE_CXX17_REMOVED_UNEXPECTED_FUNCTIONS
+    }
 }
 
 linux: {
