diff -Naur evince-45.0-orig/libdocument/ev-file-helpers.c evince-45.0/libdocument/ev-file-helpers.c
--- evince-45.0-orig/libdocument/ev-file-helpers.c	2019-01-31 06:03:59.000000000 +0300
+++ evince-45.0/libdocument/ev-file-helpers.c	2019-06-11 22:40:13.048270000 +0300
@@ -39,6 +39,10 @@
 #define O_BINARY 0
 #endif
 
+#if !defined O_CLOEXEC && defined O_NOINHERIT
+# define O_CLOEXEC O_NOINHERIT
+#endif
+
 /*
  * ev_dir_ensure_exists:
  * @dir: the directory name
@@ -595,6 +599,7 @@
 #define N_ARGS      4
 #define BUFFER_SIZE 1024
 
+#ifndef G_OS_WIN32
 static void
 compression_child_setup_cb (gpointer fd_ptr)
 {
@@ -607,6 +612,7 @@
                 fcntl (fd, F_SETFD, flags);
         }
 }
+#endif
 
 static gchar *
 compression_run (const gchar       *uri,
@@ -655,7 +661,11 @@
 
 	if (g_spawn_async_with_pipes (NULL, argv, NULL,
 				      G_SPAWN_STDERR_TO_DEV_NULL,
+#ifdef G_OS_WIN32
+                                      NULL, NULL,
+#else
                                       compression_child_setup_cb, GINT_TO_POINTER (fd),
+#endif
                                       NULL,
 				      NULL, &pout, NULL, &err)) {
 		GIOChannel *in, *out;
diff -Naur evince-45.0-orig/libdocument/ev-document.c evince-45.0/libdocument/ev-document.c
--- evince-45.0-orig/libdocument/ev-document.c	2019-01-31 06:03:59.000000000 +0300
+++ evince-45.0/libdocument/ev-document.c	2019-06-11 22:40:13.048270000 +0300
@@ -595,8 +595,12 @@ ev_document_load_fd (EvDocument         *document,
                 return FALSE;
         }

+#ifdef G_OS_WIN32
+        if (fstat(fd, &statbuf) == -1) {
+#else
         if (fstat(fd, &statbuf) == -1 ||
             (fd_flags = fcntl (fd, F_GETFL, &flags)) == -1) {
+#endif
                 int errsv = errno;
                 g_set_error_literal (error, G_FILE_ERROR,
                                      g_file_error_from_errno (errsv),
diff -Naur evince-45.0-orig/libview/ev-jobs.c evince-45.0/libview/ev-jobs.c
--- evince-45.0-orig/libview/ev-jobs.c	2019-01-31 06:03:59.000000000 +0300
+++ evince-45.0/libview/ev-jobs.c	2019-06-11 22:40:13.048270000 +0300
@@ -1458,7 +1458,11 @@ ev_dupfd (int fd,
 {
         int new_fd;
 
+#ifdef G_OS_WIN32
+        new_fd = -1;
+#else
         new_fd = fcntl (fd, F_DUPFD_CLOEXEC, 3);
+#endif
         if (new_fd == -1) {
                 int errsv = errno;
                 g_set_error_literal (error, G_FILE_ERROR, g_file_error_from_errno (errsv),
diff -Naur evince-45.0-orig/previewer/ev-previewer-window.c evince-45.0/previewer/ev-previewer-window.c
--- evince-45.0-orig/previewer/ev-previewer-window.c	2019-01-31 06:03:59.000000000 +0300
+++ evince-45.0/previewer/ev-previewer-window.c	2019-06-11 22:40:13.048270000 +0300
@@ -648,7 +648,11 @@ ev_previewer_window_set_source_fd (EvPreviewerWindow *window,
         g_return_val_if_fail (EV_IS_PREVIEWER_WINDOW (window), FALSE);
         g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
 
+#ifdef G_OS_WIN32
+        nfd = -1;
+#else
         nfd = fcntl (fd, F_DUPFD_CLOEXEC, 3);
+#endif
         if (nfd == -1) {
                 int errsv = errno;
                 g_set_error (error, G_IO_ERROR, g_io_error_from_errno (errsv),
diff -Naur evince-45.0-orig/previewer/ev-previewer.c evince-45.0/previewer/ev-previewer.c
--- evince-45.0-orig/previewer/ev-previewer.c	2019-01-31 06:03:59.000000000 +0300
+++ evince-45.0/previewer/ev-previewer.c	2019-06-11 22:40:13.048270000 +0300
@@ -166,8 +166,12 @@ check_arguments (int argc,
                 struct stat statbuf;
                 int flags;
 
+#ifdef G_OS_WIN32
+                if (fstat (input_fd, &statbuf) == -1 ) {
+#else
                 if (fstat (input_fd, &statbuf) == -1 ||
                     (flags = fcntl (input_fd, F_GETFL, &flags)) == -1) {
+#endif
                         int errsv = errno;
                         g_set_error_literal (error, G_FILE_ERROR,
                                              g_file_error_from_errno(errsv),
