diff -Naur evince-3.32.0-orig/previewer/ev-previewer-window.c evince-3.32.0/previewer/ev-previewer-window.c
--- evince-3.32.0-orig/previewer/ev-previewer-window.c	2019-02-09 06:36:23.000000000 +0300
+++ evince-3.32.0/previewer/ev-previewer-window.c	2019-06-12 00:13:30.339416700 +0300
@@ -272,7 +272,9 @@
 #endif
 
 static const GActionEntry actions[] = {
+#if GTKUNIXPRINT_ENABLED
 	{ "print", ev_previewer_window_print },
+#endif
 	{ "go-previous-page", ev_previewer_window_previous_page },
 	{ "go-next-page", ev_previewer_window_next_page },
 	{ "select-page", ev_previewer_window_focus_page_selector },
