--- evince-3.14.0/libdocument/Makefile.am.orig	2014-09-24 08:52:47.575000000 +0500
+++ evince-3.14.0/libdocument/Makefile.am	2014-09-24 08:52:59.727400000 +0500
@@ -167,6 +167,7 @@
 	-I$(builddir)
 EvinceDocument_@EV_API_VERSION_U@_gir_EXPORT_PACKAGES = evince-document-$(EV_API_VERSION)
 INTROSPECTION_GIRS = EvinceDocument-$(EV_API_VERSION).gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 girdir = $(datadir)/gir-1.0
 gir_DATA = EvinceDocument-$(EV_API_VERSION).gir
--- evince-3.14.0/libview/Makefile.am.orig	2014-09-24 08:53:51.044400000 +0500
+++ evince-3.14.0/libview/Makefile.am	2014-09-24 08:54:11.090400000 +0500
@@ -150,6 +150,7 @@
 	--identifier-prefix=Ev				\
 	--symbol-prefix=ev
 INTROSPECTION_GIRS = EvinceView-$(EV_API_VERSION).gir
+INTROSPECTION_SCANNER_ENV = CC="$(CC)"
 
 girdir = $(datadir)/gir-1.0
 gir_DATA = EvinceView-$(EV_API_VERSION).gir
