diff -Naur evince-49.alpha.1-orig/meson.build evince-49.alpha.1/meson.build
--- evince-49.alpha.1-orig/meson.build	2026-09-06 00:00:00.000000000 +0000
+++ evince-49.alpha.1/meson.build	2026-09-06 00:00:00.000000000 +0000
@@ -155 +155 @@
-exempi_dep = dependency('exempi-2.0', version: exempi_req_version)
+exempi_dep = dependency('exempi-2.0', version: exempi_req_version, required: host_machine.system() != 'windows')
diff -Naur evince-49.alpha.1-orig/libdocument/ev-init.c evince-49.alpha.1/libdocument/ev-init.c
--- evince-49.alpha.1-orig/libdocument/ev-init.c	2026-09-06 00:00:00.000000000 +0000
+++ evince-49.alpha.1/libdocument/ev-init.c	2026-09-06 00:00:00.000000000 +0000
@@ -28 +28,3 @@
-#include <exempi/xmp.h>
+#ifndef G_OS_WIN32
+#include <exempi/xmp.h>
+#endif
@@ -119 +121,3 @@
-	xmp_init ();
+#ifndef G_OS_WIN32
+	xmp_init ();
+#endif
@@ -146 +150,3 @@
-	xmp_terminate ();
+#ifndef G_OS_WIN32
+	xmp_terminate ();
+#endif
diff -Naur evince-49.alpha.1-orig/libdocument/ev-xmp.c evince-49.alpha.1/libdocument/ev-xmp.c
--- evince-49.alpha.1-orig/libdocument/ev-xmp.c	2026-09-06 00:00:00.000000000 +0000
+++ evince-49.alpha.1/libdocument/ev-xmp.c	2026-09-06 00:00:00.000000000 +0000
@@ -27,0 +28,10 @@
+#ifdef G_OS_WIN32
+gboolean
+ev_xmp_parse (const char    *metadata,
+              gsize          size,
+              EvDocumentInfo *info)
+{
+        return FALSE;
+}
+#else
+
@@ -352,0 +363 @@
+#endif
