diff -Naur evince-49.alpha.1-orig/meson.build evince-49.alpha.1/meson.build
--- evince-49.alpha.1-orig/meson.build	2026-09-06 00:00:00.000000000 +0000
+++ evince-49.alpha.1/meson.build	2026-09-06 00:00:00.000000000 +0000
@@ -265 +265,4 @@
-if get_option('internal_sysprof') == 'true'
+if host_machine.system() == 'windows'
+  external_sysprof = true
+  libsysprof_capture_dep = declare_dependency()
+elif get_option('internal_sysprof') == 'true'
diff -Naur evince-49.alpha.1-orig/libdocument/ev-debug.h evince-49.alpha.1/libdocument/ev-debug.h
--- evince-49.alpha.1-orig/libdocument/ev-debug.h	2026-09-06 00:00:00.000000000 +0000
+++ evince-49.alpha.1/libdocument/ev-debug.h	2026-09-06 00:00:00.000000000 +0000
@@ -96 +96,3 @@
-#include <sysprof-capture.h>
+#ifndef G_OS_WIN32
+#include <sysprof-capture.h>
+#endif
@@ -107,0 +110,6 @@
+#ifdef G_OS_WIN32
+#undef EV_PROFILER_START
+#undef EV_PROFILER_STOP
+#define EV_PROFILER_START(job_type)
+#define EV_PROFILER_STOP()
+#endif
