--- evince-48.1/libview/ev-print-operation.c.orig	2025-06-30 06:36:12.275982200 +0200
+++ evince-48.1/libview/ev-print-operation.c	2025-06-30 06:37:52.806731500 +0200
@@ -1908,7 +1908,8 @@
         g_object_class->constructed = ev_print_operation_export_portal_constructed;
         g_object_class->finalize = ev_print_operation_export_portal_finalize;
 }
-
+#else
+#define ev_should_use_portal() (FALSE)
 #endif /* PORTAL_ENABLED */
 
 /* Export with unix print dialogue */
