--- opencolorio/src/bindings/python/CMakeLists.txt.orig	2018-01-10 08:13:42.931606200 +0300
+++ opencolorio/src/bindings/python/CMakeLists.txt	2018-01-10 08:15:55.689556300 +0300
@@ -87,7 +87,7 @@
 	PRIVATE
 		PYOCIO_NAME=PyOpenColorIO
 )
-if(WIN32)
+if(MSVC)
        set(PYTHON_VARIANT_PATH "lib${LIB_SUFFIX}/site-packages")
 else()
        set(PYTHON_VARIANT_PATH "lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages")
