diff -bur mysql-connector-c++-9.6.0-src-orig/CMakeLists.txt mysql-connector-c++-9.6.0-src/CMakeLists.txt
--- mysql-connector-c++-9.6.0-src-orig/CMakeLists.txt	2026-01-25 04:56:28.851600700 -0700
+++ mysql-connector-c++-9.6.0-src/CMakeLists.txt	2026-01-25 04:56:32.582952200 -0700
@@ -517,7 +517,7 @@
 # system-wide). Here we arrange for the OpenSSL DLLs to be copied to the build
 # location of the connector library after building it.
 
-if(WIN32 AND OPENSSL_LIB_DIR)
+if(MSVC AND OPENSSL_LIB_DIR)
 
   # Note: For simplicity we just copy any DLLs we can find at the given
   # OpenSSL location.
