--- a/libs/process/build/Jamfile
+++ b/libs/process/build/Jamfile
@@ -46,7 +46,8 @@
   lib advapi32 ;
   lib ntdll ;
   lib user32 ;
-  explicit shell32 advapi32 ntdll user32 ;
+  lib ws2_32 ;
+  explicit shell32 advapi32 ntdll user32 ws2_32 ;
 }
 
 lib boost_process
@@ -58,6 +59,7 @@
      <target-os>windows:<library>user32
      <target-os>windows:<library>ntdll
      <target-os>windows:<library>advapi32
+     <target-os>windows:<library>ws2_32
    : usage-requirements
      <link>shared:<define>BOOST_PROCESS_DYN_LINK=1
      <boost.process.fs>boost:<library>/boost/filesystem//boost_filesystem
