--- a/tools/boost_install/BoostConfig.cmake
+++ b/tools/boost_install/BoostConfig.cmake
@@ -43,7 +43,7 @@
 #                           The default is multithreaded.
 # Boost_USE_STATIC_LIBS:    When ON, uses static Boost libraries; when OFF,
 #                           uses shared Boost libraries; when not set, uses
-#                           static on Windows, shared otherwise.
+#                           static on Windows/MSVC, shared otherwise.
 # Boost_USE_STATIC_RUNTIME: When ON, uses Boost libraries linked against the
 #                           static runtime. The default is shared runtime.
 # Boost_USE_DEBUG_RUNTIME:  When ON, uses Boost libraries linked against the
--- a/tools/boost_install/boost-install.jam
+++ b/tools/boost_install/boost-install.jam
@@ -253,7 +253,7 @@
             "    return()"
             "  endif()"
             "else()"
-            "  if(NOT WIN32 AND NOT _BOOST_SINGLE_VARIANT)"
+            "  if(NOT MSVC AND NOT _BOOST_SINGLE_VARIANT)"
             "    _BOOST_SKIPPED(\"$(fname)\" \"static, default is shared, set Boost_USE_STATIC_LIBS=ON to override\")"
             "    return()"
             "  endif()"
@@ -271,8 +271,8 @@
             "    return()"
             "  endif()"
             "else()"
-            "  if(WIN32 AND NOT _BOOST_SINGLE_VARIANT)"
-            "    _BOOST_SKIPPED(\"$(fname)\" \"shared, default on Windows is static, set Boost_USE_STATIC_LIBS=OFF to override\")"
+            "  if(MSVC AND NOT _BOOST_SINGLE_VARIANT)"
+            "    _BOOST_SKIPPED(\"$(fname)\" \"shared, default on Windows/MSVC is static, set Boost_USE_STATIC_LIBS=OFF to override\")"
             "    return()"
             "  endif()"
             "endif()"
