--- a/libs/log/CMakeLists.txt
+++ b/libs/log/CMakeLists.txt
@@ -631,6 +631,12 @@
         PRIVATE
             ${boost_log_common_private_linkflags}
     )
+    if(MINGW)
+        target_link_libraries(boost_log_setup
+            PRIVATE
+                ws2_32
+        )
+    endif()
 else()
     add_library(boost_log_setup INTERFACE)
     target_include_directories(boost_log_setup INTERFACE include)
