--- boost_1_75_0/tools/build/src/engine/build.sh.orig	2021-01-31 23:59:07.083918500 -0800
+++ boost_1_75_0/tools/build/src/engine/build.sh	2021-01-31 23:59:52.849850400 -0800
@@ -221,7 +221,7 @@
             exit 1
         fi
         case $machine in
-        *mingw*)
+        *mingw*|*windows-gnu*)
         # MinGW insists that its bin directory be in PATH.
         if test -r ${B2_TOOLSET_ROOT}bin/gcc ; then
             export PATH=${B2_TOOLSET_ROOT}bin:$PATH
