--- boost_1_77_0/boostcpp.jam.orig	2021-08-05 12:41:03.000000000 +0300
+++ boost_1_77_0/boostcpp.jam	2021-09-10 10:34:07.605536000 +0300
@@ -163,7 +163,7 @@ else if $(layout) = tagged
             <base> <threading> <runtime> ;
     case 1.69 :
         .format-name-args =
-            <base> <threading> <runtime> <arch-and-model> ;
+            <base> <threading> <runtime> ;
     }
 }
 else if $(layout) = system
