diff -urp boost_1_54_0-orig/boost/property_tree/detail/xml_parser_write.hpp boost_1_54_0/boost/property_tree/detail/xml_parser_write.hpp
--- boost_1_54_0-orig/boost/property_tree/detail/xml_parser_write.hpp	2013-07-23 00:47:43.731888449 +0200
+++ boost_1_54_0/boost/property_tree/detail/xml_parser_write.hpp	2013-07-23 00:50:40.597960319 +0200
@@ -35,7 +35,6 @@ namespace boost { namespace property_tre
                            const xml_writer_settings<Ch> & settings
                            )
     {
-        typedef typename std::basic_string<Ch> Str;
         if (separate_line)
             write_xml_indent(stream,indent,settings);
         stream << Ch('<') << Ch('!') << Ch('-') << Ch('-');
@@ -69,7 +68,6 @@ namespace boost { namespace property_tre
     {
 
         typedef typename Ptree::key_type::value_type Ch;
-        typedef typename std::basic_string<Ch> Str;
         typedef typename Ptree::const_iterator It;
 
         bool want_pretty = settings.indent_count > 0;
