Index: boost/mpl/print.hpp
===================================================================
--- a/boost/mpl/print.hpp	(revision 83411)
+++ b/boost/mpl/print.hpp	(working copy)
@@ -45,22 +45,21 @@
     : mpl::identity<T>
 #if defined(__MWERKS__)
     , aux::print_base
-#endif 
+#endif
 {
 #if defined(BOOST_MSVC)
     enum { n = sizeof(T) + -1 };
 #elif defined(__MWERKS__)
     void f(int);
-#else 
-    enum {
-        n =
-# if defined(__EDG_VERSION__)
-           aux::dependent_unsigned<T>::value > -1
-# else 
-           sizeof(T) > -1
-# endif 
-        };
-#endif 
+#elif defined(__EDG_VERSION__)
+    enum { n = aux::dependent_unsigned<T>::value > -1 };
+#elif defined(BOOST_GCC)
+    enum { n1 };
+    enum { n2 };
+    enum { n = n1 != n2 };
+#else
+    enum { n = sizeof(T) > -1 };
+#endif
 };
 
 #if defined(BOOST_MSVC)
