diff -urp boost_1_54_0-orig/libs/mpi/src/communicator.cpp boost_1_54_0/libs/mpi/src/communicator.cpp
--- boost_1_54_0-orig/libs/mpi/src/communicator.cpp	2013-07-23 00:48:17.317091987 +0200
+++ boost_1_54_0/libs/mpi/src/communicator.cpp	2013-07-23 00:50:40.722961078 +0200
@@ -118,7 +118,6 @@ optional<status> communicator::iprobe(in
 
 status communicator::probe(int source, int tag) const
 {
-  typedef optional<status> result_type;
 
   status stat;
   BOOST_MPI_CHECK_RESULT(MPI_Probe,
