--- a/boost/interprocess/detail/win32_api.hpp	2014-09-04 14:54:48.000000000 +0400
+++ b/boost/interprocess/detail/win32_api.hpp	2014-09-30 23:06:13.567000000 +0400
@@ -651,7 +652,7 @@
 
 typedef FILETIME interprocess_filetime;
 
-typedef WIN32_FIND_DATA win32_find_data;
+typedef WIN32_FIND_DATAA win32_find_data;
 
 typedef SECURITY_ATTRIBUTES interprocess_security_attributes;
 
@@ -1522,7 +1523,7 @@
    }
 
    public:
-   static void *get(const unsigned int id)
+   static farproc_t get(const unsigned int id)
    {
       BOOST_ASSERT(id < (unsigned int)NumFunction);
       for(unsigned i = 0; FunctionStates[id] < 2; ++i){
