diff -up boost_1_54_0/boost/date_time/gregorian/greg_facet.hpp\~ boost_1_54_0/boost/date_time/gregorian/greg_facet.hpp
--- boost_1_54_0/boost/date_time/gregorian/greg_facet.hpp~	2008-11-23 12:13:35.000000000 +0100
+++ boost_1_54_0/boost/date_time/gregorian/greg_facet.hpp	2013-07-22 23:44:42.834012788 +0200
@@ -215,7 +215,6 @@ namespace gregorian {
   {
     std::istream_iterator<std::basic_string<charT>, charT> beg(is), eos;
     
-    typedef boost::date_time::all_date_names_put<greg_facet_config, charT> facet_def;
     d = from_stream(beg, eos);
     return is;
   }

Diff finished.  Mon Jul 22 23:44:50 2013
