From cdd83345585161f0b1152f9be5669f88f563b224 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 21 Sep 2021 20:52:42 +0200
Subject: [PATCH 022/N] sysconfig: MINGW build extensions with GCC

---
 Lib/sysconfig.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index eadb99c..6381bcd 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -709,6 +709,8 @@ def get_platform():
 
     """
     if os.name == 'nt':
+        if 'GCC' in sys.version:
+            return 'mingw'
         if 'amd64' in sys.version.lower():
             return 'win-amd64'
         if '(arm)' in sys.version.lower():
