From 4d557c506ebc15f45e5d459faa416ea557651d46 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 7 Apr 2023 19:13:42 +0530
Subject: [PATCH 025/N] setup.py: add export_symbols to fix compile of ctypes

---
 setup.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/setup.py b/setup.py
index b55ee94..af2ada6 100644
--- a/setup.py
+++ b/setup.py
@@ -1434,6 +1434,10 @@ def detect_ctypes(self):
                         extra_compile_args=extra_compile_args,
                         extra_link_args=extra_link_args,
                         libraries=(['ole32', 'oleaut32', 'uuid'] if MS_WINDOWS else []),
+                        export_symbols=(
+                            ['DllGetClassObject PRIVATE', 'DllCanUnloadNow PRIVATE']
+                            if MS_WINDOWS else None
+                        ),
                         sources=sources)
         self.add(ext)
         # function my_sqrt() needs libm for sqrt()
