From 0fb5deb81af53ced738488bd62b9026ddac8c1ca Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 29 Aug 2025 14:03:35 +0300
Subject: [PATCH 049/N] venvlauncher: Skip find python in Registry for mingw
 builds

---
 PC/launcher.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/PC/launcher.c b/PC/launcher.c
index 15d2559..8dc0416 100644
--- a/PC/launcher.c
+++ b/PC/launcher.c
@@ -499,6 +499,7 @@ locate_all_pythons(void)
 {
     /* venv Python is highest priority */
     locate_venv_python();
+#ifndef __MINGW32__
 #if defined(_M_X64)
     /* If we are a 64bit process, first hit the 32bit keys. */
     debug(L"locating Pythons in 32bit registry\n");
@@ -519,6 +520,7 @@ locate_all_pythons(void)
     locate_pythons_for_key(HKEY_LOCAL_MACHINE, KEY_READ);
     /* Store-installed Python is lowest priority */
     locate_store_pythons();
+#endif
     qsort(installed_pythons, num_installed_pythons, sizeof(INSTALLED_PYTHON),
           compare_pythons);
 }
