From 03bbbb4036c00ae4d29f1c27acbedf68d29a7915 Mon Sep 17 00:00:00 2001
From: jeremyd2019 <4524874+jeremyd2019@users.noreply.github.com>
Date: Thu, 17 Jun 2021 18:52:37 +0530
Subject: [PATCH 064/N] configure.ac: set MINGW stack reserve

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index b679ad4..5032eaf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3545,6 +3545,11 @@ then
 	VxWorks*)
 		LINKFORSHARED='-Wl,-export-dynamic';;
 	esac
+	case $host in
+	*-*-mingw*)
+		# for https://bugs.python.org/issue40458 on MINGW
+		LINKFORSHARED="-Wl,--stack,2000000";;
+	esac
 fi
 AC_MSG_RESULT($LINKFORSHARED)
 
