From d80d623487d08f06bfcca468ce1685bc45c576dd Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 17 Jun 2021 18:52:21 +0530
Subject: [PATCH 066/N] sqlite3 module defines

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 41c463a..c381fb4 100644
--- a/setup.py
+++ b/setup.py
@@ -1660,6 +1660,7 @@ class PyBuildExt(build_ext):
                 '_sqlite/statement.c',
                 '_sqlite/util.c', ]
             sqlite_defines = []
+            sqlite_defines.append(('MODULE_NAME', '"sqlite3"'))
 
             # Enable support for loadable extensions in the sqlite3 module
             # if --enable-loadable-sqlite-extensions configure option is used.
