From 4ea1e178da214e7e2906e85e1d00868a2bfe66d4 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 1 Jan 2026 18:40:13 +0100
Subject: [PATCH 070/N] dynload_win: don't run _Py_CheckPython3 for MINGW
 builds

---
 Python/dynload_win.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Python/dynload_win.c b/Python/dynload_win.c
index bac1b42..9a17866 100644
--- a/Python/dynload_win.c
+++ b/Python/dynload_win.c
@@ -219,9 +219,9 @@ dl_funcptr _PyImport_FindSharedFuncptrWindows(const char *prefix,
     dl_funcptr p;
     char funcname[258], *import_python;
 
-#ifdef Py_ENABLE_SHARED
+#if defined(Py_ENABLE_SHARED) && defined(_MSC_VER)
     _Py_CheckPython3();
-#endif /* Py_ENABLE_SHARED */
+#endif /* defined(Py_ENABLE_SHARED) && defined(_MSC_VER) */
 
     int use_legacy = 0;
     DWORD load_library_flags = 0;
