From a8f5a4150864944df096ef7301d8d3faf267355c Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Mon, 1 Jan 2024 14:19:41 +0530
Subject: [PATCH 075/N] configure: disable various modules on MINGW

---
 configure.ac | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/configure.ac b/configure.ac
index 052fed3..095aedf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8178,6 +8178,20 @@ AS_CASE([$ac_sys_system],
     )
   ],
   [CYGWIN*], [PY_STDLIB_MOD_SET_NA([_scproxy])],
+  [MINGW*], [
+    PY_STDLIB_MOD_SET_NA(
+      [readline],
+      [_posixshmem],
+      [_posixsubprocess],
+      [fcntl],
+      [grp],
+      [pwd],
+      [resource],
+      [syslog],
+      [termios],
+      [_scproxy],
+    )
+  ],
   [QNX*], [PY_STDLIB_MOD_SET_NA([_scproxy])],
   [FreeBSD*], [PY_STDLIB_MOD_SET_NA([_scproxy])],
   [Emscripten], [
