From 1c451265815865f8d620361d85eb9835c45e602f Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Sun, 18 Jun 2023 14:20:44 +0530
Subject: [PATCH 077/N] link with bcrypt

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2258bb5..01344fe 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7231,7 +7231,7 @@ AC_MSG_RESULT([done])
 # For mingw build need additional library for linking
 case $host in
   *-*-mingw*)
-    LIBS="$LIBS -lversion -lshlwapi -lpathcch"
+    LIBS="$LIBS -lversion -lshlwapi -lpathcch -lbcrypt"
     AC_PROG_AWK
     if test "$AWK" = "gawk"; then
       awk_extra_flag="--non-decimal-data"
