From 1222380ac230162490b1803ba2a8ef54890877f7 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Tue, 20 Jun 2023 23:00:43 +0530
Subject: [PATCH 084/N] Always normalize path in abspath

---
 Lib/ntpath.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/ntpath.py b/Lib/ntpath.py
index e634c5a..8e4f579 100644
--- a/Lib/ntpath.py
+++ b/Lib/ntpath.py
@@ -593,7 +593,7 @@ else:  # use native Windows method on Windows
     def abspath(path):
         """Return the absolute version of a path."""
         try:
-            return _getfullpathname(normpath(path))
+            return normpath(_getfullpathname(normpath(path)))
         except (OSError, ValueError):
             # See gh-75230, handle outside for cleaner traceback
             pass
