From f974ea135776c79da322abdadbfbd7902e37d681 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 3 Sep 2025 00:03:20 +0300
Subject: [PATCH 085/N] dictobject: MINGW support

---
 Objects/dictobject.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Objects/dictobject.c b/Objects/dictobject.c
index c2223af..51ec9b0 100644
--- a/Objects/dictobject.c
+++ b/Objects/dictobject.c
@@ -549,7 +549,7 @@ calculate_log2_keysize(Py_ssize_t minsize)
 #if SIZEOF_LONG == SIZEOF_SIZE_T
     minsize = Py_MAX(minsize, PyDict_MINSIZE);
     return _Py_bit_length(minsize - 1);
-#elif defined(_MSC_VER)
+#elif defined(MS_WINDOWS)
     // On 64bit Windows, sizeof(long) == 4. We cannot use _Py_bit_length.
     minsize = Py_MAX(minsize, PyDict_MINSIZE);
     unsigned long msb;
