From fc0dd6b35c87cabe69757117506e3035b923f922 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 23 Jun 2023 21:47:32 +0200
Subject: [PATCH 090/N] configure.ac: set BUILDEXEEXT and EXEEXT

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 305a401..b3c3ab0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1409,6 +1409,7 @@ AC_ARG_WITH([suffix],
     [Emscripten/browser*], [EXEEXT=.js],
     [Emscripten/node*], [EXEEXT=.js],
     [WASI/*], [EXEEXT=.wasm],
+    [MINGW*], [EXEEXT=.exe],
     [EXEEXT=]
   )
 ])
@@ -1435,6 +1436,10 @@ else
 fi
 rmdir CaseSensitiveTestDir
 
+AS_CASE([$ac_sys_system],
+  [MINGW], [BUILDEXEEXT=".exe"]
+)
+
 case $ac_sys_system in
 hp*|HP*)
     case $CC in
