From dbf53f043462b05922b37affbde1ca46f794f4b2 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 1 Aug 2021 15:18:50 +0200
Subject: [PATCH 094/N] configure.ac: set _WIN32_WINNT version

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index cf4b742..c0e7703 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4277,6 +4277,11 @@ AS_VAR_IF([with_dbmliborder], [error], [
 ])
 AC_MSG_RESULT($with_dbmliborder)
 
+case $host in
+  *-*-mingw*)
+    CFLAGS_NODIST="$CFLAGS_NODIST -D_WIN32_WINNT=0x0602";;
+esac
+
 # Determine if windows modules should be used.
 AC_SUBST(USE_WIN32_MODULE)
 USE_WIN32_MODULE='#'
