From 6968f18b6fa6b662cc76e3b32e8fcd2467a1de5b Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Mon, 1 Jan 2024 14:19:41 +0530
Subject: [PATCH 106/N] readline: disable on MINGW

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 353ff02..e15e19a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7760,6 +7760,7 @@ AS_CASE([$ac_sys_system],
   [VxWorks*], [PY_STDLIB_MOD_SET_NA([_scproxy], [_crypt], [termios], [grp])],
   [Darwin], [PY_STDLIB_MOD_SET_NA([ossaudiodev], [spwd])],
   [CYGWIN*], [PY_STDLIB_MOD_SET_NA([_scproxy], [nis])],
+  [MINGW*], [PY_STDLIB_MOD_SET_NA([readline])],
   [QNX*], [PY_STDLIB_MOD_SET_NA([_scproxy], [nis])],
   [FreeBSD*], [PY_STDLIB_MOD_SET_NA([_scproxy], [spwd])],
   [Emscripten|WASI], [
