From f291913c893e9e6300f662b2bcc7283f332d0ccb Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Mon, 1 Jan 2024 16:43:41 +0530
Subject: [PATCH 110/N] Enable curses library on MINGW

---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 2575149..c4dfe3b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6868,6 +6868,14 @@ dnl have_panel=[no, panelw, panel]
 have_curses=no
 have_panel=no
 
+case "$host_os" in
+	mingw*)
+	# On Mingw, include directory is different, so just assume
+  # it's there.
+	ac_cv_header_ncurses_h=yes
+	;;
+esac
+
 AH_TEMPLATE([HAVE_NCURSESW], [Define to 1 if you have the `ncursesw' library.])
 AC_CHECK_HEADERS([curses.h ncurses.h])
 
