From 53f904e06dea49a6a6d981c65e40782549d8185d Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 2 Feb 2024 14:43:29 +0530
Subject: [PATCH 111/N] mingw_smoketests: disable distutils related tests

---
 mingw_smoketests.py | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 4f81c68..0ab5d3b 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -61,24 +61,24 @@ class Tests(unittest.TestCase):
                 self.assertEqual(str(Path("~").expanduser()), expanduser("~"))
                 self.assertEqual(str(Path.home()), expanduser("~"))
 
-    def test_sysconfig_schemes(self):
-        # https://github.com/msys2/MINGW-packages/issues/9319
-        import sysconfig
-        from distutils.dist import Distribution
-        from distutils.command.install import install
-
-        names = ['scripts', 'purelib', 'platlib', 'data', 'include']
-        for scheme in ["nt", "nt_user"]:
-            for name in names:
-                c = install(Distribution({"name": "foobar"}))
-                c.user = (scheme == "nt_user")
-                c.finalize_options()
-                if name == "include":
-                    dist_path = os.path.dirname(getattr(c, "install_" + "headers"))
-                else:
-                    dist_path = getattr(c, "install_" + name)
-                sys_path = sysconfig.get_path(name, scheme)
-                self.assertEqual(dist_path, sys_path, (scheme, name))
+    # def test_sysconfig_schemes(self):
+    #     # https://github.com/msys2/MINGW-packages/issues/9319
+    #     import sysconfig
+    #     from distutils.dist import Distribution
+    #     from distutils.command.install import install
+
+    #     names = ['scripts', 'purelib', 'platlib', 'data', 'include']
+    #     for scheme in ["nt", "nt_user"]:
+    #         for name in names:
+    #             c = install(Distribution({"name": "foobar"}))
+    #             c.user = (scheme == "nt_user")
+    #             c.finalize_options()
+    #             if name == "include":
+    #                 dist_path = os.path.dirname(getattr(c, "install_" + "headers"))
+    #             else:
+    #                 dist_path = getattr(c, "install_" + name)
+    #             sys_path = sysconfig.get_path(name, scheme)
+    #             self.assertEqual(dist_path, sys_path, (scheme, name))
 
     def test_ctypes_find_library(self):
         from ctypes.util import find_library
