From 0e379708a37c54ba35113c578b755717dd0d5f01 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Sun, 18 Jun 2023 13:12:56 +0530
Subject: [PATCH 115/N] def VPATH when compiling `Python/sysmodule.c`

---
 Makefile.pre.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 57709e3..08a3e36 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1325,6 +1325,7 @@ Python/dynload_win.o: $(srcdir)/Python/dynload_win.c Makefile
 Python/sysmodule.o: $(srcdir)/Python/sysmodule.c Makefile $(srcdir)/Include/pydtrace.h
 	$(CC) -c $(PY_CORE_CFLAGS) \
 		-DABIFLAGS='"$(ABIFLAGS)"' \
+		-DVPATH='"$(VPATH)"' \
 		$(MULTIARCH_CPPFLAGS) \
 		-o $@ $(srcdir)/Python/sysmodule.c
 
