From 6e2e92dd38f7e0baee17f3a955095b19204ce9ff Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Sun, 18 Jun 2023 13:12:56 +0530
Subject: [PATCH 115/N] def VPATH when compiling `Python/sysmodule.c`

---
 Makefile.pre.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 1b3f0f7..2f7fc8c 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1323,6 +1323,7 @@ Python/dynload_win.o: $(srcdir)/Python/dynload_win.c Makefile
 Python/sysmodule.o: $(srcdir)/Python/sysmodule.c Makefile $(srcdir)/Include/pydtrace.h
 	$(CC) -c $(PY_CORE_CFLAGS) \
 		-DABIFLAGS='"$(ABIFLAGS)"' \
+		-DVPATH='"$(VPATH)"' \
 		$(MULTIARCH_CPPFLAGS) \
 		-o $@ $(srcdir)/Python/sysmodule.c
 
