From b61f8d8fabf93ec5bc345d03f2b8bd3c2de18b95 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Sun, 18 Jun 2023 14:20:44 +0530
Subject: [PATCH 117/N] link with bcrypt

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2c48b24..8d9ff39 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6780,7 +6780,7 @@ AC_MSG_RESULT(done)
 # For mingw build need additional library for linking
 case $host in
   *-*-mingw*)
-    LIBS="$LIBS -lversion -lshlwapi -lpathcch"
+    LIBS="$LIBS -lversion -lshlwapi -lpathcch -lbcrypt"
     AC_PROG_AWK
     if test "$AWK" = "gawk"; then
       awk_extra_flag="--non-decimal-data"
