From 1e212039792aa9d80c671475e0160b16a840653b Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 1 Nov 2024 11:47:01 +0530
Subject: [PATCH 117/N] Add ucrt to version string

---
 Python/getcompiler.c | 5 +++++
 configure.ac         | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/Python/getcompiler.c b/Python/getcompiler.c
index 4b0b9b3..dd4d3c8 100644
--- a/Python/getcompiler.c
+++ b/Python/getcompiler.c
@@ -32,8 +32,13 @@
 #if defined(__clang__)
 #define str(x) #x
 #define xstr(x) str(x)
+#if defined(_UCRT)
+#define COMPILER COMP_SEP "[GCC UCRT Clang " xstr(__clang_major__) "." \
+        xstr(__clang_minor__) "." xstr(__clang_patchlevel__) ARCH_SUFFIX "]"
+#else
 #define COMPILER COMP_SEP "[GCC Clang " xstr(__clang_major__) "." \
         xstr(__clang_minor__) "." xstr(__clang_patchlevel__) ARCH_SUFFIX "]"
+#endif
 #else
 #if defined(_UCRT)
 #define COMPILER COMP_SEP "[GCC UCRT " __VERSION__ ARCH_SUFFIX "]"
diff --git a/configure.ac b/configure.ac
index 264bae6..10a6e84 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6377,7 +6377,7 @@ case $host_os in
   if test $linking_to_ucrt = no; then
     PYD_PLATFORM_TAG+="_msvcrt"
   else
-    PYD_PLATFORM_TAG += "_ucrt"
+    PYD_PLATFORM_TAG+="_ucrt"
   fi
   if test -n "${cc_is_clang}"; then
     # it is CLANG32
