From bc6f2af9f0a79d2b7c2a38f08abf25e72cfd2594 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Fri, 23 Jun 2023 20:15:38 +0530
Subject: [PATCH 128/N] configure: build `mmap` module on win32

---
 configure.ac | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 98bef51..1863975 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7490,7 +7490,9 @@ PY_STDLIB_MOD([fcntl],
   [], [test "$ac_cv_header_sys_ioctl_h" = "yes" -a "$ac_cv_header_fcntl_h" = "yes"],
   [], [$FCNTL_LIBS])
 PY_STDLIB_MOD([mmap],
-  [], [test "$ac_cv_header_sys_mman_h" = "yes" -a "$ac_cv_header_sys_stat_h" = "yes"])
+  [], m4_flatten([test "$ac_cv_header_sys_mman_h" = "yes"
+                    -a "$ac_cv_header_sys_stat_h" = "yes"
+                    -o "$MACHDEP" = "win32"]))
 PY_STDLIB_MOD([_socket],
   [], m4_flatten([test "$ac_cv_header_sys_socket_h" = "yes"
                     -a "$ac_cv_header_sys_types_h" = "yes"
