From 02d7ab7ae63fdfd05cd6f8d6270606f22b3434d1 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 23 Jun 2023 21:47:32 +0200
Subject: [PATCH 131/N] configure.ac: set BUILDEXEEXT and EXEEXT

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 482aec5..5fefb1d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1389,6 +1389,7 @@ AC_ARG_WITH([suffix],
     [Emscripten/browser*], [EXEEXT=.js],
     [Emscripten/node*], [EXEEXT=.js],
     [WASI/*], [EXEEXT=.wasm],
+    [MINGW*], [EXEEXT=.exe],
     [EXEEXT=]
   )
 ])
@@ -1412,6 +1413,10 @@ else
 fi
 rmdir CaseSensitiveTestDir
 
+AS_CASE([$ac_sys_system],
+  [MINGW], [BUILDEXEEXT=".exe"]
+)
+
 case $ac_sys_system in
 hp*|HP*)
     case $CC in
