diff -Naur Python-3.8.0-orig/Include/pyport.h Python-3.8.0/Include/pyport.h
--- Python-3.8.0-orig/Include/pyport.h	2019-10-22 10:01:16.988171900 +0300
+++ Python-3.8.0/Include/pyport.h	2019-10-22 10:03:09.574569700 +0300
@@ -177,6 +177,8 @@
 #       define PY_FORMAT_SIZE_T "l"
 #   elif defined(MS_WINDOWS)
 #       define PY_FORMAT_SIZE_T "I"
+#   elif defined(__MINGW32__) && defined(__USE_MINGW_ANSI_STDIO)
+#       define PY_FORMAT_SIZE_T "z"
 #   else
 #       error "This platform's pyconfig.h needs to define PY_FORMAT_SIZE_T"
 #   endif
