diff -Naur Python-3.8.0-orig/Makefile.pre.in Python-3.8.0/Makefile.pre.in
--- Python-3.8.0-orig/Makefile.pre.in	2019-10-22 10:03:33.473811700 +0300
+++ Python-3.8.0/Makefile.pre.in	2019-10-22 10:04:23.315899200 +0300
@@ -1404,6 +1404,7 @@
 		distutils distutils/command distutils/tests $(XMLLIBSUBDIRS) \
 		test/test_tools test/test_warnings test/test_warnings/data \
 		turtledemo \
+		msilib \
 		multiprocessing multiprocessing/dummy \
 		unittest unittest/test unittest/test/testmock \
 		venv venv/scripts venv/scripts/common venv/scripts/posix \
