--- Python-3.8.7/Python/initconfig.c.orig	2020-12-21 17:25:24.000000000 +0100
+++ Python-3.8.7/Python/initconfig.c	2021-01-06 18:07:12.498508100 +0100
@@ -1,4 +1,5 @@
 #include "Python.h"
+#include "iscygpty.h"
 #include "osdefs.h"       /* DELIM */
 #include "pycore_fileutils.h"
 #include "pycore_getopt.h"
@@ -1598,6 +1599,22 @@
         }
     }
 
+    #ifdef MS_WINDOWS
+    /* In case of is_cygpty() we still want the console encoding.
+       XXX: we ignore GetConsoleCP() here as Python always uses one encoding
+       in init_sys_streams(), and GetConsoleOutputCP() seems more important */
+    if (config->stdio_encoding == NULL) {
+        if ((isatty((int)fileno(stdout)) || is_cygpty((int)fileno(stdout)))) {
+            char encoding[20];
+            PyOS_snprintf(encoding, sizeof(encoding), "cp%u", GetConsoleOutputCP());
+            status = PyConfig_SetBytesString(config, &config->stdio_encoding, encoding);
+            if (_PyStatus_EXCEPTION(status)) {
+                return status;
+            }
+        }
+    }
+    #endif
+
     /* Choose the default error handler based on the current locale. */
     if (config->stdio_encoding == NULL) {
         status = config_get_locale_encoding(config, &config->stdio_encoding);
