diff -Naur Python-3.8.0-orig/Include/fileobject.h Python-3.8.0/Include/fileobject.h
--- Python-3.8.0-orig/Include/fileobject.h	2019-10-14 16:34:47.000000000 +0300
+++ Python-3.8.0/Include/fileobject.h	2019-10-22 10:05:02.627968300 +0300
@@ -23,7 +23,7 @@
 PyAPI_DATA(int) Py_HasFileSystemDefaultEncoding;
 
 /* A routine to check if a file descriptor can be select()-ed. */
-#ifdef _MSC_VER
+#ifdef MS_WINDOWS
     /* On Windows, any socket fd can be select()-ed, no matter how high */
     #define _PyIsSelectable_fd(FD) (1)
 #else
