--- Python-3.8.0/Modules/_xxsubinterpretersmodule.c.orig	2019-10-22 13:34:28.145872500 +0300
+++ Python-3.8.0/Modules/_xxsubinterpretersmodule.c	2019-10-22 13:34:49.299509600 +0300
@@ -1749,7 +1749,7 @@
 "A channel ID identifies a channel and may be used as an int.");
 
 static PyTypeObject ChannelIDtype = {
-    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    PyVarObject_HEAD_INIT(NULL, 0)
     "_xxsubinterpreters.ChannelID", /* tp_name */
     sizeof(channelid),              /* tp_basicsize */
     0,                              /* tp_itemsize */
