--- Python-3.8.7/setup.py.orig	2020-12-26 07:15:03.663309900 +0100
+++ Python-3.8.7/setup.py	2020-12-26 07:51:28.765752400 +0100
@@ -892,7 +892,11 @@
                            depends=['testcapi_long.h']))
 
         # Python Internal C API test module
+        macros = []
+        if MS_WINDOWS:
+            macros.append(('PY3_DLLNAME', 'L"%s"' % sysconfig.get_config_var('DLLLIBRARY')))
         self.add(Extension('_testinternalcapi', ['_testinternalcapi.c'],
+                           define_macros=macros,
                            extra_compile_args=['-DPy_BUILD_CORE_MODULE']))
 
         # Python PEP-3118 (buffer protocol) test module