--- graphene-1.6.0/configure.ac	2017-03-02 18:21:17.000000000 +0100
+++ build-x86_64-w64-mingw32/configure.ac	2017-06-02 22:42:10.103817600 +0200
@@ -126,7 +126,7 @@
 AM_CONDITIONAL(OS_LINUX, [test "$platform_linux" = "yes"])
 AM_CONDITIONAL(OS_WIN32, [test "$platform_win32" = "yes"])
 
-AC_CHECK_FUNCS([aligned_alloc posix_memalign memalign])
+AC_CHECK_FUNCS([aligned_alloc posix_memalign memalign _aligned_malloc])
 
 saved_LIBS="$LIBS"
 LIBS=$LIBM
