--- p11-kit-0.23.21/common/compat.h.orig	2020-10-25 22:40:35.843040200 +0100
+++ p11-kit-0.23.21/common/compat.h	2020-10-25 22:41:06.882100100 +0100
@@ -294,7 +294,7 @@
 
 #endif /* HAVE_STRDUP */
 
-#if defined HAVE_DECL_REALLOCARRAY && !HAVE_DECL_REALLOCARRAY
+#if !defined(HAVE_DECL_REALLOCARRAY) || !HAVE_DECL_REALLOCARRAY
 
 void *     reallocarray     (void *ptr,
                              size_t nmemb,
