From 442b9a641d10e4de462e59300b7e1711448d5530 Mon Sep 17 00:00:00 2001
From: Gavin Smith <gavinsmith0123@gmail.com>
Date: Tue, 10 Sep 2024 16:29:21 +0100
Subject: [PATCH] * info/terminal.c (terminal_prep_terminal_hook): Return void not int.

---
diff --git a/info/terminal.c b/info/terminal.c
index 5340434..69d1f0e 100644
--- a/info/terminal.c
+++ b/info/terminal.c
@@ -56,7 +56,7 @@ extern int tputs ();
 
 void (*terminal_initialize_terminal_hook) (char *terminal_name) = NULL;
 void (*terminal_get_screen_size_hook) (void) = NULL;
-int (*terminal_prep_terminal_hook) (void) = NULL;
+void (*terminal_prep_terminal_hook) (void) = NULL;
 void (*terminal_unprep_terminal_hook) (void) = NULL;
 void (*terminal_new_terminal_hook) (char *terminal_name) = NULL;
 void (*terminal_goto_xy_hook) (int x, int y) = NULL;
diff --git a/info/terminal.h b/info/terminal.h
index 3439250..28a6d07 100644
--- a/info/terminal.h
+++ b/info/terminal.h
@@ -59,7 +59,7 @@ extern void (*terminal_get_screen_size_hook) (void);
 extern int terminal_prep_terminal (void);
 extern void terminal_unprep_terminal (void);
 
-extern int (*terminal_prep_terminal_hook) (void);
+extern void (*terminal_prep_terminal_hook) (void);
 extern void (*terminal_unprep_terminal_hook) (void);
 
 /* Re-initialize the terminal to TERMINAL_NAME. */
-- 
2.46.0

