--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -259,7 +259,7 @@
 add_library(objlib OBJECT ${LIBKQUEUE_SOURCES} ${LIBKQUEUE_HEADERS})
 add_library(kqueue SHARED $<TARGET_OBJECTS:objlib>)
 add_library(kqueue_static STATIC $<TARGET_OBJECTS:objlib>)
-if (CMAKE_SYSTEM_NAME MATCHES Windows)
+if (CMAKE_SYSTEM_NAME MATCHES Windows AND NOT MINGW)
   set_target_properties(kqueue_static
           PROPERTIES
           OUTPUT_NAME kqueue_static
