--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -335,7 +335,9 @@
           kqueue kqueue_static
         DESTINATION
           "${CMAKE_INSTALL_FULL_LIBDIR}"
-        COMPONENT libraries)
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 install(FILES
           "${CMAKE_SOURCE_DIR}/kqueue.2"
         DESTINATION
