--- ffmpeg-7.0.1/fftools/ffplay_renderer.c.orig	2024-04-05 01:22:59.000000000 +0200
+++ ffmpeg-7.0.1/fftools/ffplay_renderer.c	2024-06-21 22:35:50.291061800 +0200
@@ -765,7 +765,7 @@
         vkDestroySurfaceKHR = (PFN_vkDestroySurfaceKHR)
                 ctx->get_proc_addr(ctx->inst, "vkDestroySurfaceKHR");
         vkDestroySurfaceKHR(ctx->inst, ctx->vk_surface, NULL);
-        ctx->vk_surface = NULL;
+        ctx->vk_surface = VK_NULL_HANDLE;
     }
 
     av_buffer_unref(&ctx->hw_device_ref);
