--- taglib-1.9.1/taglib/toolkit/trefcounter.h.orig	2014-04-20 20:30:13.837600000 +0400
+++ taglib-1.9.1/taglib/toolkit/trefcounter.h	2014-04-20 20:31:11.167600000 +0400
@@ -33,7 +33,6 @@
 #  include <libkern/OSAtomic.h>
 #  define TAGLIB_ATOMIC_MAC
 #elif defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__CYGWIN__)
-#  define NOMINMAX
 #  include <windows.h>
 #  define TAGLIB_ATOMIC_WIN
 #elif defined (__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 401)    \
