diff --git a/cmake/FindICONV.cmake b/cmake/FindICONV.cmake
index 56b6ad320..fd624daf4 100644
--- a/cmake/FindICONV.cmake
+++ b/cmake/FindICONV.cmake
@@ -36,7 +36,7 @@ find_path(ICONV_INCLUDE_DIR iconv.h
    PATHS /sw /opt/local
 )
 
-set(POTENTIAL_ICONV_LIBS iconv libiconv libiconv2)
+set(POTENTIAL_ICONV_LIBS iconv libiconv libiconv2 libiconv-2)
 
 find_library(ICONV_LIBRARY
    NAMES ${POTENTIAL_ICONV_LIBS}
@@ -44,7 +44,7 @@ find_library(ICONV_LIBRARY
 )
 
 if(WIN32)
-	set(ICONV_DLL_NAMES iconv.dll  libiconv.dll libiconv2.dll)
+	set(ICONV_DLL_NAMES iconv.dll  libiconv.dll libiconv2.dll libiconv-2.dll)
 	find_file(ICONV_DLL   
 					NAMES ${ICONV_DLL_NAMES}
 					PATHS ENV PATH
