diff --git a/CMakeLists.txt b/CMakeLists.txt
index 949900c55..f21115330 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -282,8 +282,10 @@ if(WIN32)
   set(REGEX_MALLOC 1)
   set(USE_GENERIC_MS_NINT 1)
   set(HAVE_STRING_H 0)
-  # Suppress warnings for regex.c
-  set_source_files_properties(${REGEX_SOURCES} PROPERTIES COMPILE_FLAGS /w)
+  if(MSVC)
+    # Suppress warnings for regex.c
+    set_source_files_properties(${REGEX_SOURCES} PROPERTIES COMPILE_FLAGS /w)
+  endif()
 else(WIN32)
   set(REGEX_SOURCES "")
 endif(WIN32)
