diff -urN kservice-5.25.0.orig/src/CMakeLists.txt kservice-5.25.0/src/CMakeLists.txt
--- kservice-5.25.0.orig/src/CMakeLists.txt	2016-08-07 15:57:40.000000000 +0200
+++ kservice-5.25.0/src/CMakeLists.txt	2016-08-18 13:58:34.982616200 +0200
@@ -144,7 +144,7 @@
    DESTINATION ${KDE_INSTALL_KSERVICETYPES5DIR}
 )
 
-if (WIN32)
+if (WIN32 AND NOT MSYS)
 install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_DATAROOTDIR}/xdg/menus  RENAME ${APPLICATIONS_MENU_NAME} )
 else ()
 install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/menus  RENAME ${APPLICATIONS_MENU_NAME} )
