From 3273a1af4c4e03559528301a4e4681a453709c26 Mon Sep 17 00:00:00 2001
From: nulano <nulano@nulano.eu>
Date: Sun, 4 Oct 2020 22:55:53 +0100
Subject: [PATCH 2/2] fix tk on latest msys2-mingw64

---
 src/Tk/tkImaging.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Tk/tkImaging.c b/src/Tk/tkImaging.c
index 161a835fbd..fdb3a1e1de 100644
--- a/src/Tk/tkImaging.c
+++ b/src/Tk/tkImaging.c
@@ -63,7 +63,7 @@ ImagingFind(const char* name)
     Py_ssize_t id;
 
     /* FIXME: use CObject instead? */
-#if defined(_MSC_VER) && defined(_WIN64)
+#if defined(_WIN64)
     id = _atoi64(name);
 #else
     id = atol(name);
