--- Pillow-5.0.0/setup.py.orig	2018-01-01 10:03:06.000000000 -0500
+++ Pillow-5.0.0/setup.py	2018-02-17 22:54:51.672853100 -0500
@@ -597,8 +597,6 @@ class pil_build_ext(build_ext):
         if feature.jpeg2000:
             libs.append(feature.jpeg2000)
             defs.append(("HAVE_OPENJPEG", None))
-            if sys.platform == "win32":
-                defs.append(("OPJ_STATIC", None))
         if feature.zlib:
             libs.append(feature.zlib)
             defs.append(("HAVE_LIBZ", None))
