--- Pillow-4.1.1/setup.py.orig	2017-06-17 16:27:09.000000000 +0100
+++ Pillow-4.1.1/setup.py	2017-06-17 23:34:17.226557000 +0100
@@ -562,8 +562,6 @@
         if feature.jpeg2000:
             libs.append(feature.jpeg2000)
             defs.append(("HAVE_OPENJPEG", None))
-            if sys.platform == "win32":
-                defs.append(("OPJ_STATIC", None))
         if feature.zlib:
             libs.append(feature.zlib)
             defs.append(("HAVE_LIBZ", None))
