--- a/src/main-application.c
+++ b/src/main-application.c
@@ -110,12 +110,6 @@
	ocontext = g_option_context_new (_("[FILE ...]"));
 	g_option_context_add_main_entries (ocontext, gnumeric_options, GETTEXT_PACKAGE);
 	g_option_context_add_group	  (ocontext, gnm_get_option_group ());
-
-#if defined(G_OS_WIN32)
-	/* we have already translated to utf8, do not do it again.
-	 * http://bugzilla.gnome.org/show_bug.cgi?id=361321 */
-	g_option_context_set_delocalize   (ocontext, FALSE);
-#endif
 
 	g_option_context_add_group (ocontext, gtk_get_option_group (TRUE));
 	g_option_context_parse (ocontext, &argc, &argv, &error);

