--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -1,10 +1,10 @@
-SUBDIRS_FILE_FORMATS = excel lotus-123 oleo sc sylk xbase html dif qpro \
+SUBDIRS_FILE_FORMATS = lotus-123 oleo sc sylk xbase html dif qpro \
 	plan-perfect applix openoffice mps
 SUBDIRS_FUNCTIONS = fn-numtheory fn-christian-date fn-database fn-date \
 	fn-eng fn-erlang fn-financial fn-flt \
-	fn-hebrew-date fn-info fn-logical fn-complex fn-lookup \
-	fn-math fn-r fn-stat fn-string fn-random fn-tsa fn-derivatives
-SUBDIRS_LOADERS = excelplugins
+	fn-hebrew-date fn-info fn-logical fn-lookup \
+	fn-math fn-r fn-stat fn-string fn-random fn-derivatives
+SUBDIRS_LOADERS =
 SUBDIRS_DATA_SOURCES = sample_datasource
 SUBDIRS_SOLVERS = nlsolve lpsolve glpk
 SUBDIRS_EXAMPLES = uihello
