--- a/configure.ac
+++ b/configure.ac
@@ -945,11 +945,7 @@
 	PY_PREFIX=`$PYTHON -c 'import sys ; sys.stdout.write(sys.prefix)'`
     fi
     if test "x$PY_INCLUDE_DIR" = x; then
-	if test "x$with_native_win32" = xyes ; then
-	    PY_INCLUDES="-I$PY_PREFIX/include"
-	else
-	    PY_INCLUDES=`$PYTHON-config --includes`
-	fi
+	PY_INCLUDES=`$PYTHON-config --includes`
     else
 	PY_INCLUDES="-I$PY_INCLUDE_DIR"
     fi
