--- lpeg-0.12/lptypes.h	2015-04-08 08:39:37.670848100 +0200
+++ lpeg-0.12/lptypes.h	2015-04-08 08:41:08.649094600 +0200
@@ -27,9 +27,9 @@
 
 
 /*
-** compatibility with Lua 5.2
+** compatibility with Lua 5.2 and higher
 */
-#if (LUA_VERSION_NUM == 502)
+#if (LUA_VERSION_NUM >= 502)
 
 #undef lua_equal
 #define lua_equal(L,idx1,idx2)  lua_compare(L,(idx1),(idx2),LUA_OPEQ)
@@ -48,6 +48,15 @@
 
 #endif
 
+/*
+** compatibility with Lua 5.3 and higher
+*/
+#if (LUA_VERSION_NUM >= 503)
+
+#undef luaL_checkint
+#define luaL_checkint luaL_checkinteger
+
+#endif
 
 /* default maximum size for call/backtrack stack */
 #if !defined(MAXBACK)
