diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config.mk	2017-04-30 19:09:25.629000800 +0100
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk	2017-04-30 19:09:52.490907400 +0100
@@ -429,10 +429,8 @@
 
 ifeq ($(UNAME),Windows)
     # Cygwin Make on Windows (untested)
-    AR_TARGET = $(LIBRARY).lib
-    SO_PLAIN  = $(LIBRARY).dll
-    SO_MAIN   = $(LIBRARY).$(SO_VERSION).dll
-    SO_TARGET = $(LIBRARY).$(VERSION).dll
+    AR_TARGET = $(LIBRARY).a
+    SO_TARGET = $(LIBRARY).dll
     SO_INSTALL_NAME = echo
 else
     # Mac or Linux/Unix