diff -urN SuiteSparse.orig/GPUQREngine/Demo/Makefile SuiteSparse/GPUQREngine/Demo/Makefile
--- SuiteSparse.orig/GPUQREngine/Demo/Makefile	2015-02-01 12:07:54.875166100 +0000
+++ SuiteSparse/GPUQREngine/Demo/Makefile	2015-02-01 12:09:29.881730400 +0000
@@ -8,7 +8,7 @@
     ../../SuiteSparse_GPURuntime/Lib/libSuiteSparse_GPURuntime.a \
     ../../CHOLMOD/Lib/libcholmod.a \
     ../../SuiteSparse_config/libsuitesparseconfig.a \
-    $(CUDART_LIB) $(CUBLAS_LIB) -lm -lrt
+    $(CUDART_LIB) $(CUBLAS_LIB) -lm
 
 INC  = -I../../SuiteSparse_GPURuntime/Include -I../Include \
        -I../../SuiteSparse_config -I../../CHOLMOD/Include
diff -urN SuiteSparse.orig/KLU/Tcov/Makefile SuiteSparse/KLU/Tcov/Makefile
--- SuiteSparse.orig/KLU/Tcov/Makefile	2015-02-01 12:07:54.916671400 +0000
+++ SuiteSparse/KLU/Tcov/Makefile	2015-02-01 12:09:35.275415300 +0000
@@ -23,7 +23,7 @@
     ../../CAMD/Lib/libcamd.a ../../CCOLAMD/Lib/libccolamd.a \
     ../../metis-4.0/libmetis.a \
     ../../SuiteSparse_config/libsuitesparseconfig.a \
-    $(LAPACK) $(BLAS) $(CUDART_LIB) $(CUBLAS_LIB) -lm -lrt
+    $(LAPACK) $(BLAS) $(CUDART_LIB) $(CUBLAS_LIB) -lm
 
 I = -I../../SuiteSparse_config -I../../AMD/Include -I../../COLAMD/Include \
     -I../../BTF/Include -I../../CHOLMOD/Include -I../../CAMD/Include \
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config.mk	2015-02-01 12:07:55.130698600 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk	2015-02-01 12:10:51.118546200 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_backslash.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_backslash.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_backslash.mk	2015-02-01 12:07:55.131198600 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_backslash.mk	2015-02-01 12:12:19.955827100 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_debug.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_debug.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_debug.mk	2015-02-01 12:07:55.131198600 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_debug.mk	2015-02-01 12:10:36.268660500 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_gcc.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_gcc.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_gcc.mk	2015-02-01 12:07:55.131698700 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_gcc.mk	2015-02-01 12:11:08.608767200 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_icc.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_icc.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_icc.mk	2015-02-01 12:07:55.131698700 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_icc.mk	2015-02-01 12:11:49.473956400 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_icc10.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_icc10.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_GPU_icc10.mk	2015-02-01 12:07:55.132198800 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_GPU_icc10.mk	2015-02-01 12:12:02.795148000 +0000
@@ -91,9 +91,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_linux.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_linux.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_linux.mk	2015-02-01 12:07:55.132198800 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_linux.mk	2015-02-01 12:12:27.206247800 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_Mac.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config_Mac.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config_Mac.mk	2015-02-01 12:07:55.132698800 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config_Mac.mk	2015-02-01 12:12:12.022319700 +0000
@@ -90,9 +90,9 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+# LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
-# LIB = -lm
+  LIB = -lm
 
 # For "make install"
 INSTALL_LIB = /usr/local/lib
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.1 SuiteSparse/UMFPACK/Tcov/Make.1
--- SuiteSparse.orig/UMFPACK/Tcov/Make.1	2015-02-02 11:16:00.669134800 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.1	2015-02-02 11:16:54.447426900 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O0 -g -ftest-coverage -fprofile-arcs -O0
 UMFPACK_CONFIG = -DNBLAS -DTEST_FOR_INTEGER_OVERFLOW -DTESTING
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.2 SuiteSparse/UMFPACK/Tcov/Make.2
--- SuiteSparse.orig/UMFPACK/Tcov/Make.2	2015-02-02 11:16:00.669134800 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.2	2015-02-02 11:16:56.771588500 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O0 -g -ftest-coverage -fprofile-arcs -O0
 UMFPACK_CONFIG = -DTESTING
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.3 SuiteSparse/UMFPACK/Tcov/Make.3
--- SuiteSparse.orig/UMFPACK/Tcov/Make.3	2015-02-02 11:16:00.669134800 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.3	2015-02-02 11:17:00.795975700 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O3 -fPIC
 UMFPACK_CONFIG = -DNBLAS -DTEST_FOR_INTEGER_OVERFLOW -DTESTING
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.4 SuiteSparse/UMFPACK/Tcov/Make.4
--- SuiteSparse.orig/UMFPACK/Tcov/Make.4	2015-02-02 11:16:00.669134800 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.4	2015-02-02 11:17:04.245221900 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O3 -fPIC
 UMFPACK_CONFIG = -DTESTING
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.5 SuiteSparse/UMFPACK/Tcov/Make.5
--- SuiteSparse.orig/UMFPACK/Tcov/Make.5	2015-02-02 11:16:00.669134800 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.5	2015-02-02 11:17:08.800954600 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O0 -g -ftest-coverage -fprofile-arcs -O0
 UMFPACK_CONFIG = -DNBLAS -DTEST_FOR_INTEGER_OVERFLOW -DTESTING -DNRECIPROCAL
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.6 SuiteSparse/UMFPACK/Tcov/Make.6
--- SuiteSparse.orig/UMFPACK/Tcov/Make.6	2015-02-02 11:16:00.669134800 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.6	2015-02-02 11:17:13.386884200 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O0 -g -ftest-coverage -fprofile-arcs -O0
 UMFPACK_CONFIG = -DTESTING -DNTIMER
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.7 SuiteSparse/UMFPACK/Tcov/Make.7
--- SuiteSparse.orig/UMFPACK/Tcov/Make.7	2015-02-02 11:16:00.623939400 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.7	2015-02-02 11:17:17.614050600 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O0 -g -ftest-coverage -fprofile-arcs -O0
 UMFPACK_CONFIG = -DNBLAS -DTEST_FOR_INTEGER_OVERFLOW -DTESTING -DNO_DIVIDE_BY_ZERO
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
diff -urN SuiteSparse.orig/UMFPACK/Tcov/Make.8 SuiteSparse/UMFPACK/Tcov/Make.8
--- SuiteSparse.orig/UMFPACK/Tcov/Make.8	2015-02-02 11:16:00.624939300 +0000
+++ SuiteSparse/UMFPACK/Tcov/Make.8	2015-02-02 11:17:22.796719000 +0000
@@ -5,7 +5,7 @@
 CC = gcc
 CF = -O0 -g -ftest-coverage -fprofile-arcs -O0
 UMFPACK_CONFIG = -DTESTING -DNO_DIVIDE_BY_ZERO
-LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm -lrt
+LIB = -llapack -lblas -lgfortran -lgfortranbegin -lm
 
 RANLIB = ranlib
 MV = mv -f
