--- SuiteSparse/GraphBLAS/CMakeLists.txt.orig	2018-06-07 09:26:39.422576800 +0300
+++ SuiteSparse/GraphBLAS/CMakeLists.txt	2018-06-07 09:27:04.422611800 +0300
@@ -121,6 +121,7 @@
 
 # graphblas installation location
 install ( TARGETS graphblas graphblas_static
+    RUNTIME       DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY       DESTINATION ${CMAKE_INSTALL_LIBDIR}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     ARCHIVE       DESTINATION ${CMAKE_INSTALL_LIBDIR} )
