diff -urN SuiteSparse.orig/Makefile SuiteSparse/Makefile
--- SuiteSparse.orig/Makefile	2019-02-01 23:53:56.506337700 -0600
+++ SuiteSparse/Makefile	2019-02-06 16:04:11.116867000 -0600
@@ -13,7 +13,6 @@
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
 	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
@@ -39,7 +38,6 @@
 install: metisinstall
 	( cd SuiteSparse_config && $(MAKE) install )
 	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
 	( cd AMD && $(MAKE) install )
 	( cd BTF && $(MAKE) install )
 	( cd CAMD && $(MAKE) install )
@@ -85,7 +83,6 @@
 	( cd SuiteSparse_config && $(MAKE) uninstall )
 	- ( cd metis-5.1.0 && $(MAKE) uninstall )
 	- ( cd GraphBLAS && $(MAKE) uninstall )
-	- ( cd Mongoose  && $(MAKE) uninstall )
 	( cd AMD && $(MAKE) uninstall )
 	( cd CAMD && $(MAKE) uninstall )
 	( cd COLAMD && $(MAKE) uninstall )
@@ -112,12 +109,11 @@
 endif
 	$(RM) -r $(INSTALL_DOC)
 
-# compile the dynamic libraries.  For GraphBLAS and Mongoose, this also builds
+# compile the dynamic libraries.  For GraphBLAS, this also builds
 # the static library
 library: metis
 	( cd SuiteSparse_config && $(MAKE) )
 	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
 	( cd AMD && $(MAKE) library )
 	( cd BTF && $(MAKE) library )
 	( cd CAMD && $(MAKE) library )
@@ -138,13 +134,12 @@
 #	( cd PIRO_BAND && $(MAKE) library )
 #	( cd SKYLINE_SVD && $(MAKE) library )
 
-# compile the static libraries (except for metis, GraphBLAS, and Mongoose).
-# metis is only dynamic, and the 'make static' for GraphBLAS and Mongoose makes
+# compile the static libraries (except for metis and GraphBLAS).
+# metis is only dynamic, and the 'make static' for GraphBLAS makes
 # both the dynamic and static libraries.
 static: metis
 	( cd SuiteSparse_config && $(MAKE) static )
 	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' static )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' static )
 	( cd AMD && $(MAKE) static )
 	( cd BTF && $(MAKE) static )
 	( cd CAMD && $(MAKE) static )
@@ -171,7 +166,6 @@
 	- ( cd metis-5.1.0 && $(MAKE) distclean )
 	- ( cd AMD && $(MAKE) purge )
 	- ( cd GraphBLAS && $(MAKE) purge )
-	- ( cd Mongoose  && $(MAKE) purge )
 	- ( cd CAMD && $(MAKE) purge )
 	- ( cd COLAMD && $(MAKE) purge )
 	- ( cd BTF && $(MAKE) purge )
@@ -198,7 +192,6 @@
 	- ( cd SuiteSparse_config && $(MAKE) clean )
 	- ( cd metis-5.1.0 && $(MAKE) clean )
 	- ( cd GraphBLAS && $(MAKE) clean )
-	- ( cd Mongoose  && $(MAKE) clean )
 	- ( cd AMD && $(MAKE) clean )
 	- ( cd CAMD && $(MAKE) clean )
 	- ( cd COLAMD && $(MAKE) clean )
@@ -220,7 +213,6 @@
 # Create the PDF documentation
 docs:
 	( cd GraphBLAS && $(MAKE) docs )
-	( cd Mongoose  && $(MAKE) docs )
 	( cd AMD && $(MAKE) docs )
 	( cd CAMD && $(MAKE) docs )
 	( cd KLU && $(MAKE) docs )
