diff -urN SuiteSparse.orig/Makefile SuiteSparse/Makefile
--- SuiteSparse.orig/Makefile	2019-02-06 22:15:12.282897100 -0600
+++ SuiteSparse/Makefile	2019-02-06 22:17:22.234177700 -0600
@@ -12,7 +12,6 @@
 # Compile the default rules for each package
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
@@ -37,7 +36,6 @@
 # (note that CSparse is not installed; CXSparse is installed instead)
 install: metisinstall
 	( cd SuiteSparse_config && $(MAKE) install )
-	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
 	( cd AMD && $(MAKE) install )
 	( cd BTF && $(MAKE) install )
 	( cd CAMD && $(MAKE) install )
@@ -82,7 +80,6 @@
 	$(RM) $(INSTALL_DOC)/SuiteSparse_README.txt
 	( cd SuiteSparse_config && $(MAKE) uninstall )
 	- ( cd metis-5.1.0 && $(MAKE) uninstall )
-	- ( cd GraphBLAS && $(MAKE) uninstall )
 	( cd AMD && $(MAKE) uninstall )
 	( cd CAMD && $(MAKE) uninstall )
 	( cd COLAMD && $(MAKE) uninstall )
@@ -109,11 +106,9 @@
 endif
 	$(RM) -r $(INSTALL_DOC)
 
-# compile the dynamic libraries.  For GraphBLAS, this also builds
-# the static library
+# compile the dynamic libraries.
 library: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
 	( cd AMD && $(MAKE) library )
 	( cd BTF && $(MAKE) library )
 	( cd CAMD && $(MAKE) library )
@@ -134,12 +129,10 @@
 #	( cd PIRO_BAND && $(MAKE) library )
 #	( cd SKYLINE_SVD && $(MAKE) library )
 
-# compile the static libraries (except for metis and GraphBLAS).
-# metis is only dynamic, and the 'make static' for GraphBLAS makes
-# both the dynamic and static libraries.
+# compile the static libraries (except for metis).
+# metis is dynamic only.
 static: metis
 	( cd SuiteSparse_config && $(MAKE) static )
-	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' static )
 	( cd AMD && $(MAKE) static )
 	( cd BTF && $(MAKE) static )
 	( cd CAMD && $(MAKE) static )
@@ -165,7 +158,6 @@
 	- ( cd SuiteSparse_config && $(MAKE) purge )
 	- ( cd metis-5.1.0 && $(MAKE) distclean )
 	- ( cd AMD && $(MAKE) purge )
-	- ( cd GraphBLAS && $(MAKE) purge )
 	- ( cd CAMD && $(MAKE) purge )
 	- ( cd COLAMD && $(MAKE) purge )
 	- ( cd BTF && $(MAKE) purge )
@@ -191,7 +183,6 @@
 clean:
 	- ( cd SuiteSparse_config && $(MAKE) clean )
 	- ( cd metis-5.1.0 && $(MAKE) clean )
-	- ( cd GraphBLAS && $(MAKE) clean )
 	- ( cd AMD && $(MAKE) clean )
 	- ( cd CAMD && $(MAKE) clean )
 	- ( cd COLAMD && $(MAKE) clean )
@@ -212,7 +203,6 @@
 
 # Create the PDF documentation
 docs:
-	( cd GraphBLAS && $(MAKE) docs )
 	( cd AMD && $(MAKE) docs )
 	( cd CAMD && $(MAKE) docs )
 	( cd KLU && $(MAKE) docs )
@@ -267,13 +257,3 @@
 	@echo 'Using pre-installed METIS 5.1.0 library at ' '[$(MY_METIS_LIB)]'
 endif
 
-# just compile GraphBLAS
-gb:
-	echo $(CMAKE_OPTIONS)
-	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
-
-# just install GraphBLAS
-gbinstall:
-	echo $(CMAKE_OPTIONS)
-	( cd GraphBLAS && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
-
