--- a/cmake/MSVCRuntime.cmake
+++ b/cmake/MSVCRuntime.cmake
@@ -1,4 +1,4 @@
-if (WIN32)
+if (MSVC)
 	if (CMAKE_VERSION VERSION_LESS 3.15.0)
 		message(FATAL_ERROR "windows builds require CMake >= 3.15")
 	endif()
--- a/server/Windows/cli/CMakeLists.txt
+++ b/server/Windows/cli/CMakeLists.txt
@@ -16,7 +16,7 @@
 # limitations under the License.
 
 set(MODULE_NAME "wfreerdp-server-cli")
-set(OUTPUT_NAME " wfreerdp-server")
+set(OUTPUT_NAME "wfreerdp-server")
 set(MODULE_PREFIX "FREERDP_SERVER_WINDOWS_CLI")
 
 include_directories(..)
