diff -bur FreeRDP-3.12.0-orig/server/Windows/wf_mirage.c FreeRDP-3.12.0/server/Windows/wf_mirage.c
--- FreeRDP-3.12.0-orig/server/Windows/wf_mirage.c	2025-02-24 14:29:50 -0700
+++ FreeRDP-3.12.0/server/Windows/wf_mirage.c	2025-02-24 14:33:37 -0700
@@ -229,8 +229,8 @@
 		deviceMode->dmPelsWidth = wfi->virtscreen_width;
 		deviceMode->dmPelsHeight = wfi->virtscreen_height;
 		deviceMode->dmBitsPerPel = wfi->bitsPerPixel;
-		deviceMode->u.s2.dmPosition.x = wfi->servscreen_xoffset;
-		deviceMode->u.s2.dmPosition.y = wfi->servscreen_yoffset;
+		deviceMode->dmPosition.x = wfi->servscreen_xoffset;
+		deviceMode->dmPosition.y = wfi->servscreen_yoffset;
 	}
 
 	deviceMode->dmFields = DM_BITSPERPEL | DM_PELSWIDTH | DM_PELSHEIGHT | DM_POSITION;
