diff -bur FreeRDP-3.12.0-orig/channels/remdesk/client/CMakeLists.txt FreeRDP-3.12.0/channels/remdesk/client/CMakeLists.txt
--- FreeRDP-3.12.0-orig/channels/remdesk/client/CMakeLists.txt	2025-02-24 15:03:24 -0700
+++ FreeRDP-3.12.0/channels/remdesk/client/CMakeLists.txt	2025-02-24 15:03:29 -0700
@@ -19,6 +19,6 @@
 
 set(${MODULE_PREFIX}_SRCS remdesk_main.c remdesk_main.h)
 
-set(${MODULE_PREFIX}_LIBS winpr freerdp remdesk-common)
+set(${MODULE_PREFIX}_LIBS winpr winpr-tools freerdp remdesk-common)
 
 add_channel_client_library(${MODULE_PREFIX} ${MODULE_NAME} ${CHANNEL_NAME} FALSE "VirtualChannelEntryEx")
