diff --git a/server/Windows/wf_interface.c b/server/Windows/wf_interface.c
index 9542e9cf2a..ad9912e917 100644
--- a/server/Windows/wf_interface.c
+++ b/server/Windows/wf_interface.c
@@ -172,7 +172,6 @@ BOOL wfreerdp_server_stop(wfServer* server)
 wfServer* wfreerdp_server_new()
 {
 	WSADATA wsaData = WINPR_C_ARRAY_INIT;
-	wfServer* server;
 
 	if (WSAStartup(MAKEWORD(2, 2), &wsaData) != 0)
 		return nullptr;
