diff -bur FreeRDP-3.31.1-orig/server/Windows/cli/CMakeLists.txt FreeRDP-3.31.1/server/Windows/cli/CMakeLists.txt
--- FreeRDP-3.31.1-orig/server/Windows/cli/CMakeLists.txt
+++ FreeRDP-3.31.1/server/Windows/cli/CMakeLists.txt
@@ -25,7 +25,7 @@
 
 addtargetwithresourcefile(${MODULE_NAME} TRUE "${FREERDP_VERSION}" ${MODULE_PREFIX}_SRCS)
 
-set(${MODULE_PREFIX}_LIBS wfreerdp-server)
+set(${MODULE_PREFIX}_LIBS wfreerdp-server freerdp-server)
 
 target_link_libraries(${MODULE_NAME} PRIVATE ${${MODULE_PREFIX}_LIBS})
