--- a/winpr/include/winpr/nt.h
+++ b/winpr/include/winpr/nt.h
@@ -1327,7 +1327,7 @@
 	FileMailslotSetInformation,
 	FileCompressionInformation,
 	FileObjectIdInformation,
-	FileUnknownInformation1,
+	FileCompletionInformation,
 	FileMoveClusterInformation,
 	FileQuotaInformation,
 	FileReparsePointInformation,
--- a/winpr/libwinpr/nt/nt.c
+++ b/winpr/libwinpr/nt/nt.c
@@ -125,8 +125,8 @@
 			return "FileCompressionInformation";
 		case FileObjectIdInformation:
 			return "FileObjectIdInformation";
-		case FileUnknownInformation1:
-			return "FileUnknownInformation1";
+		case FileCompletionInformation:
+			return "FileCompletionInformation";
 		case FileMoveClusterInformation:
 			return "FileMoveClusterInformation";
 		case FileQuotaInformation:
