--- gsm-1.0-pl13/Makefile.orig	2013-07-31 07:00:01.650417600 +0000
+++ gsm-1.0-pl13/Makefile	2013-07-31 07:03:23.928603700 +0000
@@ -98,7 +98,7 @@
 #  Other tools
 
 SHELL		= /bin/sh
-LN		= ln
+LN		= ln -s
 BASENAME 	= basename
 AR		= ar
 ARFLAGS		= cr
@@ -361,16 +361,15 @@
 
 $(TOAST_INSTALL_BIN)/toast$(EXEEXT):	$(TOAST)
 		-rm $@
-		cp $(TOAST) $@
-		chmod 755 $@
+		/usr/bin/install $(TOAST) $@
 
 $(TOAST_INSTALL_BIN)/untoast$(EXEEXT):	$(TOAST_INSTALL_BIN)/toast$(EXEEXT)
 		-rm $@
-		ln $? $@
+		ln -s toast$(EXEEXT) $@
 
 $(TOAST_INSTALL_BIN)/tcat$(EXEEXT):	$(TOAST_INSTALL_BIN)/toast$(EXEEXT)
 		-rm $@
-		ln $? $@
+		ln -s toast$(EXEEXT) $@
 
 $(TOAST_INSTALL_MAN)/toast.1:	$(MAN)/toast.1
 		-rm $@
--- gsm-1.0-pl13/Makefile.orig	2013-07-31 07:05:29.540054300 +0000
+++ gsm-1.0-pl13/Makefile	2013-07-31 07:07:15.369492900 +0000
@@ -321,11 +321,11 @@
 
 $(UNTOAST):	$(BIN) $(TOAST)
 		-rm $(RMFLAGS) $(UNTOAST)
-		$(LN) $(TOAST) $(UNTOAST)
+		$(LN) $(notdir $(TOAST)) $(UNTOAST)
 
 $(TCAT):	$(BIN) $(TOAST)
 		-rm $(RMFLAGS) $(TCAT)
-		$(LN) $(TOAST) $(TCAT)
+		$(LN) $(notdir $(TOAST)) $(TCAT)
 
 
 # The local bin and lib directories
