--- unixODBC-2.3.4/include/sqltypes.h.orig	2013-12-06 05:51:46.000000000 -0500
+++ unixODBC-2.3.4/include/sqltypes.h	2017-12-20 12:08:21.025954400 -0500
@@ -174,9 +174,15 @@ typedef unsigned long   SQLSETPOSIROW;
 #else
 typedef long            SQLINTEGER;
 typedef unsigned long   SQLUINTEGER;
+#ifdef _WIN64
+  typedef long long SQLLEN;
+  typedef unsigned long long SQLULEN;
+  typedef unsigned long long SQLSETPOSIROW;
+#else
 #define SQLLEN          SQLINTEGER
 #define SQLULEN         SQLUINTEGER
 #define SQLSETPOSIROW   SQLUSMALLINT
+#endif
 typedef SQLULEN         SQLROWCOUNT;
 typedef SQLULEN         SQLROWSETSIZE;
 typedef SQLULEN         SQLTRANSID;
--- unixODBC-2.3.4/DriverManager/__attribute.c.orig	2017-12-22 08:20:12.747089400 -0500
+++ unixODBC-2.3.4/DriverManager/__attribute.c	2017-12-22 16:15:59.293200100 -0500
@@ -1302,7 +1302,7 @@ void *__attr_override( void *handle, int
 
         if ( as -> is_int_type )
         {
-            return (void*)(long) as -> int_value;
+            return (void*)(ptrdiff_t) as -> int_value;
         }
         else
         {
@@ -1364,7 +1364,7 @@ void *__attr_override_wide( void *handle
 
         if ( as -> is_int_type )
         {
-            return (void*)(long) as -> int_value;
+            return (void*)(ptrdiff_t) as -> int_value;
         }
         else
         {
@@ -1397,9 +1397,9 @@ void *__attr_override_wide( void *handle
 
 int dm_check_connection_attrs( DMHDBC connection, SQLINTEGER attribute, SQLPOINTER value )
 {
-    SQLINTEGER ival;
+    ptrdiff_t ival;
 
-    ival = (SQLINTEGER) value;
+    ival = (ptrdiff_t) value;
 
     switch( attribute ) 
     {
@@ -1534,9 +1534,9 @@ int dm_check_connection_attrs( DMHDBC co
 
 int dm_check_statement_attrs( DMHSTMT statement, SQLINTEGER attribute, SQLPOINTER value )
 {
-    SQLUINTEGER ival;
+    ptrdiff_t ival;
 
-    ival = (SQLUINTEGER) value;
+    ival = (ptrdiff_t) value;
 
     switch( attribute ) 
     {
--- unixodbc-2.3.4/DriverManager/SQLSetEnvAttr.c.orig	2017-12-22 18:27:24.730554800 -0500
+++ unixodbc-2.3.4/DriverManager/SQLSetEnvAttr.c	2017-12-22 18:31:11.050932000 -0500
@@ -198,7 +198,7 @@ SQLRETURN SQLSetEnvAttr( SQLHENV environ
     {
       case SQL_ATTR_CONNECTION_POOLING:
         {
-            SQLUINTEGER ptr = (SQLUINTEGER) value;
+            ptrdiff_t ptr = (ptrdiff_t) value;
 
             if ( ptr != SQL_CP_OFF &&
                 ptr != SQL_CP_ONE_PER_DRIVER &&
@@ -223,7 +223,7 @@ SQLRETURN SQLSetEnvAttr( SQLHENV environ
 
       case SQL_ATTR_CP_MATCH:
         {
-            SQLUINTEGER ptr = (SQLUINTEGER) value;
+            ptrdiff_t ptr = (ptrdiff_t) value;
 
             if ( ptr != SQL_CP_STRICT_MATCH &&
                 ptr != SQL_CP_RELAXED_MATCH )
@@ -247,7 +247,7 @@ SQLRETURN SQLSetEnvAttr( SQLHENV environ
 
       case SQL_ATTR_ODBC_VERSION:
         {
-            SQLUINTEGER ptr = (SQLUINTEGER) value;
+            ptrdiff_t ptr = (ptrdiff_t) value;
 
             if ( ptr != SQL_OV_ODBC2 &&
                     ptr != SQL_OV_ODBC3 &&
@@ -289,7 +289,7 @@ SQLRETURN SQLSetEnvAttr( SQLHENV environ
 
       case SQL_ATTR_OUTPUT_NTS:
         {
-            SQLUINTEGER ptr = (SQLUINTEGER) value;
+            ptrdiff_t ptr = (ptrdiff_t) value;
 
             /*
              * this must be one of the most brain dead atribute,
--- unixodbc-2.3.4/DriverManager/SQLSetStmtAttrW.c.orig	2017-12-22 18:51:12.772199700 -0500
+++ unixodbc-2.3.4/DriverManager/SQLSetStmtAttrW.c	2017-12-22 18:54:20.916200500 -0500
@@ -433,7 +433,7 @@ SQLRETURN SQLSetStmtAttrW( SQLHSTMT stat
 
     if ( attribute == SQL_ATTR_METADATA_ID )
     {
-        statement -> metadata_id = (SQLINTEGER) value;
+        statement -> metadata_id = (ptrdiff_t) value;
     }
 
     if ( attribute == SQL_ATTR_IMP_ROW_DESC || 
--- unixodbc-2.3.4/extras/snprintf.c.orig	2017-12-22 18:56:59.179392700 -0500
+++ unixodbc-2.3.4/extras/snprintf.c	2017-12-22 18:58:31.942243400 -0500
@@ -344,7 +344,7 @@ static int dopr (char *buffer, size_t ma
 	break;
       case 'p':
 	strvalue = (char*) va_arg (args, void *);
-	fmtint (buffer, &currlen, maxlen, (long) strvalue, 16, min, max, flags);
+	fmtint (buffer, &currlen, maxlen, (ptrdiff_t) strvalue, 16, min, max, flags);
 	break;
       case 'n':
 	if (cflags == DP_C_SHORT) 
