--- a/src/ptex/CMakeLists.txt
+++ b/src/ptex/CMakeLists.txt
@@ -44,7 +44,11 @@ if(PTEX_BUILD_SHARED_LIBS)
       PUBLIC Threads::Threads
       PRIVATE libdeflate::libdeflate_shared
     )
-    install(TARGETS Ptex_dynamic EXPORT Ptex DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install(TARGETS Ptex_dynamic
+            EXPORT Ptex
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 install(FILES
