--- ptex-2.1.28/src/ptex/CMakeLists-orig.txt	2017-11-23 09:19:24.775350700 -0500
+++ ptex-2.1.28/src/ptex/CMakeLists.txt	2017-11-23 09:21:26.591618800 -0500
@@ -22,7 +22,10 @@
   add_library(Ptex_dynamic SHARED ${SRCS})
   set_target_properties(Ptex_dynamic PROPERTIES OUTPUT_NAME Ptex)
   target_link_libraries(Ptex_dynamic ${CMAKE_THREAD_LIBS_INIT} ${ZLIB_LIBRARIES})
-  install(TARGETS Ptex_dynamic DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS Ptex_dynamic
+          RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 install(FILES