diff -bur rcedit-2.0.0-o/CMakeLists.txt rcedit-2.0.0/CMakeLists.txt
--- rcedit-2.0.0-o/CMakeLists.txt	2025-04-13 09:51:58.081962300 -0600
+++ rcedit-2.0.0/CMakeLists.txt	2025-04-13 09:52:03.525370400 -0600
@@ -1,11 +1,6 @@
 cmake_minimum_required(VERSION 3.15)
 
-set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded")
-
-# /Ox, full optimization
-# /Os, favour small code
-add_compile_options(/Ox /Os)
-
 project(rcedit)
 
 add_executable(rcedit src/main.cc src/rescle.cc src/rcedit.rc)
+target_link_options(rcedit PUBLIC -mconsole -municode)
