From ae2b809cec3fd845f05508e288a0f048f93d98e5 Mon Sep 17 00:00:00 2001
From: James Ross-Gowan <rossymiles@gmail.com>
Date: Tue, 19 Jan 2016 18:31:29 +1100
Subject: [PATCH] Makefile: Add API version suffix to DLL Name

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 20354a7..051b898 100644
--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,7 @@ else
 endif
 
 ifdef CONFIG_SHARED
-    OUT_LIB ?= libdcadec/libdcadec$(DLLSUF)$(SONAMESUF)
+    OUT_LIB ?= libdcadec/libdcadec-$(API_MAJOR)$(DLLSUF)
 else
     OUT_LIB ?= libdcadec/libdcadec$(LIBSUF)
 endif
-- 
2.6.3

