diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f4722c3..45d2184 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -185,6 +185,11 @@ endif()
 
 if (LIBCXXABI_ENABLE_SHARED)
   add_library(cxxabi_shared SHARED)
+  if (LIBCXXABI_USE_LLVM_UNWINDER AND NOT LIBCXXABI_STATICALLY_LINK_UNWINDER_IN_SHARED_LIBRARY)
+    target_link_libraries(cxxabi_shared PUBLIC cxx_shared unwind_shared)
+  else()
+    target_link_libraries(cxxabi_shared PUBLIC cxx_shared)
+  endif()
   set_target_properties(cxxabi_shared
     PROPERTIES
       LINK_FLAGS "${LIBCXXABI_LINK_FLAGS}"
