diff -Naur global-6.6.2/libutil/path.c src/global-6.6.2/libutil/path.c
--- global-6.6.2/libutil/path.c	2018-02-10 04:31:40.000000000 +0800
+++ src/global-6.6.2/libutil/path.c	2018-02-21 00:39:20.671989400 +0800
@@ -185,8 +185,9 @@
 	} else
 		_fixpath(in_path, out_path);
 #else
-	_fullpath(out_path, in_path, MAXPATHLEN);
-	canonpath(out_path);
+	out_path = _fullpath(out_path, in_path, MAXPATHLEN);
+	if (out_path)
+		canonpath(out_path);
 #endif
 	return out_path;
 }
diff -Naur global-6.6.2/libutil/secure_popen.c src/global-6.6.2/libutil/secure_popen.c
--- global-6.6.2/libutil/secure_popen.c	2018-02-10 04:31:40.000000000 +0800
+++ src/global-6.6.2/libutil/secure_popen.c	2018-02-21 00:09:04.850266100 +0800
@@ -23,7 +23,9 @@
 #endif
 #include <stdio.h>
 #include <errno.h>
+#if !defined(__MSDOS) && !defined(_WIN32)
 #include <sys/wait.h>
+#endif
 #ifdef STDC_HEADERS
 #include <stdlib.h>
 #endif
