From 0cb4ccb1f09d39820682e052fc106f5fd7fa1309 Mon Sep 17 00:00:00 2001
From: Stefan Sauer <ensonic@users.sf.net>
Date: Fri, 24 Feb 2012 20:37:00 +0000
Subject: interfaces: fix ABI class padding after the recent changes

---
(limited to 'gst-libs/gst/interfaces/colorbalance.h')

diff --git a/gst-libs/gst/interfaces/colorbalance.h b/gst-libs/gst/interfaces/colorbalance.h
index fd8ceaa..9f0a1cd 100644
--- a/gst-libs/gst/interfaces/colorbalance.h
+++ b/gst-libs/gst/interfaces/colorbalance.h
@@ -102,7 +102,7 @@ struct _GstColorBalanceClass {
   GstColorBalanceType (*get_balance_type)  (GstColorBalance *balance);
 
   /*< private >*/
-  gpointer _gst_reserved[GST_PADDING];
+  gpointer _gst_reserved[GST_PADDING-1];
 };
 
 GType   gst_color_balance_get_type      (void);
--
cgit v0.9.0.2-2-gbebe
