--- gst-plugins-base/win32/common/libgstaudio.def.orig	2013-12-10 12:09:41.280012610 +0000
+++ gst-plugins-base/win32/common/libgstaudio.def	2013-12-10 12:10:12.253127720 +0000
@@ -24,6 +24,7 @@
 	gst_audio_decoder_get_plc_aware
 	gst_audio_decoder_get_tolerance
 	gst_audio_decoder_get_type
+	gst_audio_decoder_merge_tags
 	gst_audio_decoder_set_byte_time
 	gst_audio_decoder_set_drainable
 	gst_audio_decoder_set_latency
