--- ufraw-0.19.2/configure.ac.orig	2014-02-27 09:46:30.949200000 +0400
+++ ufraw-0.19.2/configure.ac	2014-02-27 09:46:52.898400000 +0400
@@ -210,10 +210,12 @@
     echo cannot execute: pod2man], $PATH)
 AC_FUNC_FSEEKO
 
+if test $ufraw_win32 = no; then
 # Ensure that getopt_long is available. It is included in GNU libc and
 # in at least most BSD libcs. If not found, search for it in libgnugetopt.
 AC_SEARCH_LIBS(getopt_long, gnugetopt, ,
   AC_MSG_ERROR(can not build UFRaw without getopt_long))
+fi
 
 # Make sure that pow is available, trying libm if necessary.
 AC_SEARCH_LIBS(pow, m)
