--- a/vendor/boringssl/crypto/curve25519/internal.h
+++ b/vendor/boringssl/crypto/curve25519/internal.h
@@ -32,7 +32,7 @@
 #endif
 
 #if !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_SMALL) && \
-    defined(__GNUC__) && defined(__x86_64__)
+    defined(__GNUC__) && defined(__x86_64__) && !defined(OPENSSL_WINDOWS)
 #define BORINGSSL_FE25519_ADX
 
 // fiat_curve25519_adx_mul is defined in
--- a/vendor/boringssl/include/openssl/target.h
+++ b/vendor/boringssl/include/openssl/target.h
@@ -66,8 +66,13 @@
 #define OPENSSL_APPLE
 #endif
 
+// Unfortunate workaround to avoid symbol conflict with wincrypt.h
+// See https://github.com/openssl/openssl/issues/9981
 #if defined(_WIN32)
 #define OPENSSL_WINDOWS
+#undef PKCS7_SIGNER_INFO
+#undef X509_EXTENSIONS
+#undef X509_NAME
 #endif
 
 // Trusty and Android baremetal aren't't Linux but currently define __linux__.
--- a/vendor/libziparchive/zip_cd_entry_map.h
+++ b/vendor/libziparchive/zip_cd_entry_map.h
@@ -90,7 +90,7 @@
   static constexpr size_t offset_max = (1u << 20) - 1;
   static constexpr size_t length_max = (1u << 12) - 1;
   uint32_t name_offset : 20;
-  uint16_t name_length : 12;
+  uint32_t name_length : 12;
 };
 
 static_assert(sizeof(struct ZipStringOffset20) == 4);
