--- android-tools-35.0.2/vendor/adb/sysdeps/win32/stat.cpp.orig	2025-11-11 08:57:16.178051100 +0100
+++ android-tools-35.0.2/vendor/adb/sysdeps/win32/stat.cpp	2025-11-11 09:12:44.269015400 +0100
@@ -28,7 +28,7 @@
 // Version of stat() that takes a UTF-8 path.
 int adb_stat(const char* path, struct adb_stat* s) {
 // This definition of wstat seems to be missing from <sys/stat.h>.
-#if defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64)
+#if defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64) && !defined(__MINGW32__)
 #ifdef _USE_32BIT_TIME_T
 #define wstat _wstat32i64
 #else
