--- android-tools-35.0.2/vendor/core/fastboot/tcp.cpp.orig	2024-08-29 19:46:57.000000000 +0200
+++ android-tools-35.0.2/vendor/core/fastboot/tcp.cpp	2026-05-03 13:12:06.286391900 +0200
@@ -30,6 +30,7 @@
 
 #include <android-base/parseint.h>
 #include <android-base/stringprintf.h>
+#include <cstdint>
 
 namespace tcp {
 
--- android-tools-35.0.2/vendor/core/fastboot/udp.h.orig	2024-08-29 19:46:57.000000000 +0200
+++ android-tools-35.0.2/vendor/core/fastboot/udp.h	2026-05-03 13:11:20.189760000 +0200
@@ -30,6 +30,7 @@
 
 #include <memory>
 #include <string>
+#include <cstdint>
 
 #include "socket.h"
 #include "transport.h"
--- android-tools-35.0.2/vendor/core/fastboot/usb.h.orig	2026-05-03 13:12:37.703259500 +0200
+++ android-tools-35.0.2/vendor/core/fastboot/usb.h	2026-05-03 13:12:40.217044800 +0200
@@ -30,6 +30,7 @@
 
 #include <functional>
 #include <memory>
+#include <cstdint>
 
 #include "transport.h"
 
--- android-tools-35.0.2/vendor/libbase/hex.cpp.orig	2026-05-03 13:10:18.609867000 +0200
+++ android-tools-35.0.2/vendor/libbase/hex.cpp	2026-05-03 13:10:25.159466500 +0200
@@ -17,6 +17,7 @@
 #include "android-base/hex.h"
 
 #include "android-base/logging.h"
+#include <cstdint>
 
 namespace android {
 namespace base {
--- android-tools-35.0.2/vendor/libziparchive/include/ziparchive/zip_writer.h.orig	2024-08-29 19:46:57.000000000 +0200
+++ android-tools-35.0.2/vendor/libziparchive/include/ziparchive/zip_writer.h	2026-05-03 13:11:07.910753000 +0200
@@ -18,6 +18,7 @@
 
 #include <cstdio>
 #include <ctime>
+#include <cstdint>
 
 #include <gtest/gtest_prod.h>
 #include <memory>
