--- srecord-1.64.orig/Makefile.in	2014-06-22 11:01:46.000000000 +0530
+++ srecord-1.64/Makefile.in	2020-03-17 15:24:03.179100900 +0530
@@ -4093,7 +4093,7 @@
 all: all-bin all-doc
 
 all-bin: bin/srec_cat bin/srec_cmp bin/srec_info bin/test_arglex_ambiguous \
-		bin/test_crc16 bin/test_fletcher16 bin/test_gecos \
+		bin/test_crc16 bin/test_fletcher16 \
 		bin/test_hyphen bin/test_prelude bin/test_url_decode
 
 all-doc: etc/BUILDING.pdf etc/README.pdf etc/change_log.pdf etc/reference.pdf \
