diff -bur srecord-1.65.0-Source-o/doc/CMakeLists.txt srecord-1.65.0-Source/doc/CMakeLists.txt
--- srecord-1.65.0-Source-o/doc/CMakeLists.txt	2025-04-13 22:40:36.957682400 -0600
+++ srecord-1.65.0-Source/doc/CMakeLists.txt	2025-04-13 22:42:03.145318600 -0600
@@ -163,25 +163,25 @@
 # Standalone Documents
 
 # Transform man page(s) (ie troff format files) into a PDF
-FUNCTION(ADD_DOC TARGET SOURCE DEPS)
-  add_custom_command(OUTPUT ${TARGET}
-    COMMAND ${GROFF} -Tps -s -I${CMAKE_CURRENT_SOURCE_DIR_UNIX_FORMAT} -I${CMAKE_CURRENT_BINARY_DIR_UNIX_FORMAT} -t -man ${SOURCE} > ${TARGET}.ps.tmp
-    COMMAND ${PAGE_SELECT} ${CMAKE_CURRENT_SOURCE_DIR}/etc/page-list.awk ${TARGET}.ps.tmp ${TARGET}.ps
-    COMMAND ${PS2PDFEXE} ${TARGET}.ps ${TARGET}
-    COMMAND rm ${TARGET}.ps ${TARGET}.ps.tmp
-    DEPENDS ${SOURCE} ${DEPS}
-    COMMENT "Building ${TARGET}"
-    VERBATIM)
-  list(APPEND DOC_TARGETS ${TARGET})
-  set(DOC_TARGETS ${DOC_TARGETS} PARENT_SCOPE)
-ENDFUNCTION()
+# FUNCTION(ADD_DOC TARGET SOURCE DEPS)
+#   add_custom_command(OUTPUT ${TARGET}
+#     COMMAND ${GROFF} -Tps -s -I${CMAKE_CURRENT_SOURCE_DIR_UNIX_FORMAT} -I${CMAKE_CURRENT_BINARY_DIR_UNIX_FORMAT} -t -man ${SOURCE} > ${TARGET}.ps.tmp
+#     COMMAND ${PAGE_SELECT} ${CMAKE_CURRENT_SOURCE_DIR}/etc/page-list.awk ${TARGET}.ps.tmp ${TARGET}.ps
+#     COMMAND ${PS2PDFEXE} ${TARGET}.ps ${TARGET}
+#     COMMAND rm ${TARGET}.ps ${TARGET}.ps.tmp
+#     DEPENDS ${SOURCE} ${DEPS}
+#     COMMENT "Building ${TARGET}"
+#     VERBATIM)
+#   list(APPEND DOC_TARGETS ${TARGET})
+#   set(DOC_TARGETS ${DOC_TARGETS} PARENT_SCOPE)
+# ENDFUNCTION()
 
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/BUILDING.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/BUILDING.man "")
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/README.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/README.man "${NEW_CHANGES}")
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/change_log.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/change_log.man "${NEW_CHANGES}")
-ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/reference-${srecord_VERSION_MAJOR}.${srecord_VERSION_MINOR}.pdf
-        ${CMAKE_CURRENT_SOURCE_DIR}/etc/reference.man "${REF_PARTS};${REF_INDEX};${NEW_CHANGES}")
-add_custom_target(doco ALL DEPENDS ${DOC_TARGETS})
+# ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/BUILDING.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/BUILDING.man "")
+# ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/README.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/README.man "${NEW_CHANGES}")
+# ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/change_log.pdf ${CMAKE_CURRENT_SOURCE_DIR}/etc/change_log.man "${NEW_CHANGES}")
+# ADD_DOC(${CMAKE_CURRENT_BINARY_DIR}/reference-${srecord_VERSION_MAJOR}.${srecord_VERSION_MINOR}.pdf
+#         ${CMAKE_CURRENT_SOURCE_DIR}/etc/reference.man "${REF_PARTS};${REF_INDEX};${NEW_CHANGES}")
+# add_custom_target(doco ALL DEPENDS ${DOC_TARGETS})
 
 # Installation
 # A copy of the generated PDFs In the docdir for local reference
