diff -bur srecord-1.65.0-Source-o/doc/CMakeLists.txt srecord-1.65.0-Source/doc/CMakeLists.txt
--- srecord-1.65.0-Source-o/doc/CMakeLists.txt	2025-04-16 18:36:48.154915300 -0600
+++ srecord-1.65.0-Source/doc/CMakeLists.txt	2025-04-16 18:37:11.643088400 -0600
@@ -194,6 +194,6 @@
 # Install all the man pages on not-windows
-if(NOT WIN32)
+if((NOT WIN32) OR MINGW)
   set(sections 1 3 5)
   foreach(i ${sections})
     install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/man${i}/
