--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -53,6 +53,9 @@ foreach(target ${WSLAY_TARGETS})
 		$<BUILD_INTERFACE:${GEN_INCLUDE_DIR}>
 		$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 	target_compile_definitions(${target} PRIVATE "HAVE_CONFIG_H")
+	if(HAVE_WINSOCK2_H)
+		target_link_libraries(${target} PRIVATE ws2_32)
+	endif()
 endforeach()
 
 # install
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -46,5 +46,6 @@ add_executable(wslay_tests ${SOURCES} ${HEADERS}
 target_include_directories(wslay_tests PRIVATE ${WSLAY_DIR} ${WSLAY_INCLUDE_DIRS} ${CUNIT_INCLUDE_DIRS})
 target_compile_definitions(wslay_tests PRIVATE ${WSLAY_COMPILE_DEFINITIONS})
 target_link_libraries(wslay_tests ${CUNIT_LIBRARIES})
+target_link_libraries(wslay_tests ws2_32)
 add_dependencies(wslay_tests ${WSLAY_TARGET})
 add_test(NAME wslay_tests COMMAND wslay_tests)
