--- ./libvirt-gobject/Makefile.am.orig	2016-07-18 10:03:46.534356500 +0200
+++ ./libvirt-gobject/Makefile.am	2016-07-18 10:06:01.550863500 +0200
@@ -94,7 +94,8 @@
 			$(COVERAGE_CFLAGS:-f%=-Wc,f%) \
 			$(CYGWIN_EXTRA_LDFLAGS) $(MINGW_EXTRA_LDFLAGS) \
 			-Wl,--version-script=$(srcdir)/libvirt-gobject.sym \
-			-version-info $(LIBVIRT_GLIB_VERSION_INFO)
+			-version-info $(LIBVIRT_GLIB_VERSION_INFO) \
+			-no-undefined
 
 libvirt-gobject-enums.c: $(libvirt_gobject_1_0_la_HEADERS)
 	$(AM_V_GEN)glib-mkenums \
--- ./python/Makefile.am.orig	2016-07-18 10:09:19.492771200 +0200
+++ ./python/Makefile.am	2016-07-18 10:09:31.617963900 +0200
@@ -16,6 +16,7 @@
 libvirtglibmod_la_CFLAGS = -Wno-redundant-decls
 
 libvirtglibmod_la_LDFLAGS = -module -avoid-version \
+    -no-undefined \
 	$(CYGWIN_EXTRA_LDFLAGS)
 libvirtglibmod_la_LIBADD = ../libvirt-glib/libvirt-glib-1.0.la
 	$(CYGWIN_EXTRA_LIBADD) $(CYGWIN_EXTRA_PYTHON_LIBADD)
