--- a/meson.build	2021-09-22 11:56:17.941233800 -0700
+++ b/meson.build	2021-09-22 12:20:40.171597100 -0700
@@ -381,7 +381,7 @@
 
 # on aarch64 error: -fstack-protector not supported for this target
 if host_machine.cpu_family() != 'aarch64'
-  if host_machine.system() in [ 'linux', 'freebsd', 'windows' ]
+  if host_machine.system() in [ 'linux', 'freebsd' ]
     # we prefer -fstack-protector-strong but fallback to -fstack-protector-all
     fstack_cflags = cc.first_supported_argument([
       '-fstack-protector-strong',
