--- a/meson.build
+++ b/meson.build
@@ -336,7 +336,7 @@
 if get_option('warning_level') == '2'
   # on aarch64 error: -fstack-protector not supported for this target
   if host_machine.cpu_family() != 'aarch64'
-    if host_machine.system() in [ 'linux', 'freebsd', 'windows' ]
+    if host_machine.system() in [ 'linux', 'freebsd' ]
       # we prefer -fstack-protector-strong but fallback to -fstack-protector-all
       fstack_cflags = cc.first_supported_argument([
         '-fstack-protector-strong',
