From 9083a42bab0dac997e1c00180671e80828a3b359 Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen@syrusdark.website>
Date: Thu, 28 Jan 2021 16:55:23 +0530
Subject: [PATCH] Fix column spacing with xterm reset escape sequence

---
 src/prettytable/prettytable.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/prettytable/prettytable.py b/src/prettytable/prettytable.py
index b84e4e9..d07f698 100644
--- a/src/prettytable/prettytable.py
+++ b/src/prettytable/prettytable.py
@@ -58,7 +58,7 @@ MARKDOWN = 13
 ORGMODE = 14
 RANDOM = 20
 
-_re = re.compile(r"\033\[[0-9;]*m")
+_re = re.compile(r"\033(\[[0-9;]*m|\(B)")
 
 
 def _get_size(text):
-- 
2.29.2.windows.2

