--- z3-z3-4.7.1/src/api/python/CMakeLists.txt.orig	2018-06-06 07:41:30.619809100 +0300
+++ z3-z3-4.7.1/src/api/python/CMakeLists.txt	2018-06-06 07:42:00.540851100 +0300
@@ -99,7 +99,7 @@
     message(STATUS "CMAKE_INSTALL_PYTHON_PKG_DIR not set. Trying to guess")
     execute_process(
       COMMAND "${PYTHON_EXECUTABLE}" "-c"
-        "import distutils.sysconfig; print(distutils.sysconfig.get_python_lib())"
+        "import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(prefix=''))"
       RESULT_VARIABLE exit_code
       OUTPUT_VARIABLE CMAKE_INSTALL_PYTHON_PKG_DIR
       OUTPUT_STRIP_TRAILING_WHITESPACE
