--- z3-z3-4.8.9/src/util/hwf.cpp.orig	2021-02-04 20:15:45.155087700 -0800
+++ z3-z3-4.8.9/src/util/hwf.cpp	2021-02-04 20:16:28.030101000 -0800
@@ -48,7 +48,7 @@
 // clear to the compiler what instructions should be used. E.g., for sqrt(), the Windows compiler selects
 // the x87 FPU, even when /arch:SSE2 is on.
 // Luckily, these are kind of standardized, at least for Windows/Linux/macOS.
-#if defined(__clang__) || defined(_M_ARM) && defined(_M_ARM64)
+#if defined(_M_ARM) && defined(_M_ARM64)
 #undef USE_INTRINSICS
 #endif
 
