--- z3-z3-4.8.1/src/api/z3_api.h.orig	2018-11-01 13:17:37.893619100 +0300
+++ z3-z3-4.8.1/src/api/z3_api.h	2018-11-01 13:18:25.879303400 +0300
@@ -5325,6 +5325,8 @@
        def_API('Z3_get_error_msg', STRING, (_in(CONTEXT), _in(ERROR_CODE)))
     */
     Z3_string Z3_API Z3_get_error_msg(Z3_context c, Z3_error_code err);
+    
+    Z3_string Z3_API Z3_get_error_msg_ex(Z3_context c, Z3_error_code err);
 
     /*@}*/
 
