diff -bur sbcl-2.6.0-orig/src/runtime/hopscotch.c sbcl-2.6.0/src/runtime/hopscotch.c
--- sbcl-2.6.0-orig/src/runtime/hopscotch.c	2025-12-31 08:25:08.518909600 -0700
+++ sbcl-2.6.0/src/runtime/hopscotch.c	2025-12-31 08:25:35.547275300 -0700
@@ -23,7 +23,11 @@
 #include <stdio.h>
 #ifdef LISP_FEATURE_WIN32
 /* I don't know where ffs() is prototyped */
-extern int ffs(int);
+  #ifdef __clang__
+    #define ffs __builtin_ffs
+  #else
+    extern int ffs(int);
+  #endif
 #else
 /* https://www.freebsd.org/cgi/man.cgi?query=fls&sektion=3&manpath=FreeBSD+7.1-RELEASE
    says strings.h */
