diff --git a/src/gn/ninja_binary_target_writer.cc b/src/gn/ninja_binary_target_writer.cc
index b960cc01..6c45c2d0 100644
--- a/src/gn/ninja_binary_target_writer.cc
+++ b/src/gn/ninja_binary_target_writer.cc
@@ -361,8 +361,9 @@ void NinjaBinaryTargetWriter::WriteLinkerFlags(
   // Then the library search path
   WriteLibrarySearchPath(out, tool);
 
+  // Add a space before DEF file path
   if (optional_def_file) {
-    out_ << " /DEF:";
+    out_ << " ";
     path_output_.WriteFile(out, *optional_def_file);
   }
 }
