diff --git a/src/util/sys_info.cc b/src/util/sys_info.cc
index 20bbdf72..e05ef653 100644
--- a/src/util/sys_info.cc
+++ b/src/util/sys_info.cc
@@ -83,6 +83,8 @@ std::string OperatingSystemArchitecture() {
   SYSTEM_INFO system_info = {};
   ::GetNativeSystemInfo(&system_info);
   switch (system_info.wProcessorArchitecture) {
+    case PROCESSOR_ARCHITECTURE_ARM64:
+      return "arm64";
     case PROCESSOR_ARCHITECTURE_INTEL:
       return "x86";
     case PROCESSOR_ARCHITECTURE_AMD64:
