diff -Naur hexchat-2.14.1.orig/data/misc/meson.build hexchat-2.14.1/data/misc/meson.build
--- hexchat-2.14.1.orig/data/misc/meson.build	2018-03-13 22:27:13.000000000 -0400
+++ hexchat-2.14.1/data/misc/meson.build	2018-05-27 23:27:32.448482500 -0400
@@ -2,21 +2,6 @@
 desktop_utils = find_program('desktop-file-validate', required: false)
 
 if get_option('with-gtk')
-  hexchat_appdata = i18n.merge_file(
-    input: 'io.github.Hexchat.appdata.xml.in',
-    output: 'io.github.Hexchat.appdata.xml',
-    po_dir: '../../po',
-    install: true,
-    install_dir: join_paths(get_option('datadir'), 'metainfo')
-  )
-
-  appstream_util = find_program('appstream-util', required: false)
-  if appstream_util.found()
-    test('Validate io.github.Hexchat.appdata.xml', appstream_util,
-      args: ['validate-relax', hexchat_appdata]
-    )
-  endif
-
   desktop_conf = configuration_data()
   if get_option('with-dbus')
     desktop_conf.set('exec_command', 'hexchat --existing %U')
