diff -ruN yajl-2.1.0-orig/test/parsing/run_tests.sh yajl-2.1.0/test/parsing/run_tests.sh
--- yajl-2.1.0-orig/test/parsing/run_tests.sh   2014-03-19 04:58:29 +0000
+++ yajl-2.1.0/test/parsing/run_tests.sh        2016-11-04 10:05:47 +0000
@@ -68,7 +68,7 @@
   # parse with a read buffer size ranging from 1-31 to stress stream parsing
   while [ $iter -lt 32  ] && [ $success = "SUCCESS" ] ; do
     $testBin $allowPartials $allowComments $allowGarbage $allowMultiple -b $iter < $file > ${file}.test  2>&1
-    diff ${DIFF_FLAGS} ${file}.gold ${file}.test > ${file}.out
+    diff -a ${DIFF_FLAGS} ${file}.gold ${file}.test > ${file}.out
     if [ $? -eq 0 ] ; then
       if [ $iter -eq 31 ] ; then testsSucceeded=$(( $testsSucceeded + 1 )) ; fi
     else

diff -ruN yajl-2.1.0-orig/test/parsing/cases/doubles_in_array.json.gold yajl-2.1.0/test/parsing/cases/doubles_in_array.json.gold
--- yajl-2.1.0-orig/test/parsing/cases/doubles_in_array.json.gold       2014-03-19 04:58:29 +0000
+++ yajl-2.1.0/test/parsing/cases/doubles_in_array.json.gold    2016-11-04 09:52:17 +0000
@@ -1,8 +1,8 @@
 array open '['
 double: 0.00012
-double: 6e-06
-double: 6e-06
-double: 1e-06
-double: 1e-06
+double: 6e-006
+double: 6e-006
+double: 1e-006
+double: 1e-006
 array close ']'
 memory leaks:  0
