--- build-x86_64-perl-5.32.1/makedef.pl.orig	2021-07-17 20:22:00.208254900 -0700
+++ build-x86_64-perl-5.32.1/makedef.pl	2021-07-17 20:22:17.880187400 -0700
@@ -1320,7 +1320,7 @@
     print "LIBRARY $dll\n";
     # The DESCRIPTION module definition file statement is not supported
     # by VC7 onwards.
-    if ($ARGS{CCTYPE} eq 'GCC') {
+    if (0 && $ARGS{CCTYPE} eq 'GCC') {
 	print "DESCRIPTION 'Perl interpreter'\n";
     }
     print "EXPORTS\n";
