--- Python-3.7.2/Lib/sysconfig.py.orig	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2/Lib/sysconfig.py	2019-03-15 15:58:58.818000800 +0100
@@ -251,6 +251,7 @@
     # if the expansion uses the name without a prefix.
     renamed_variables = ('CFLAGS', 'LDFLAGS', 'CPPFLAGS')
 
+    done['prefix']='${SYS_PREFIX}'
     while len(variables) > 0:
         for name in tuple(variables):
             value = notdone[name]
@@ -402,7 +403,20 @@
         f.write('# system configuration generated and used by'
                 ' the sysconfig module\n')
         f.write('build_time_vars = ')
-        pprint.pprint(vars, stream=f)
+        pprint.pprint(vars, stream=f, width=999999)
+
+    # Now reload the file and replace:
+    replacements = {": '${SYS_PREFIX}'" : ": sys.prefix",
+                    ": '${SYS_PREFIX}"  : ": sys.prefix + '",
+                     "${SYS_PREFIX}'" : "' + sys.prefix",
+                     "${SYS_PREFIX}"  : "' + sys.prefix + '"}
+
+    contents = open(destfile).read()
+    for rep in replacements.keys():
+        contents = contents.replace(rep, replacements[rep])
+    with open(destfile, 'w', encoding='utf8') as f:
+        f.write('import sys\n')
+        f.write(contents)
 
     # Create file used for sys.path fixup -- see Modules/getpath.c
     with open('pybuilddir.txt', 'w', encoding='ascii') as f:
