From 8418fabcc0b418be9b5531d04263ee014014b8b6 Mon Sep 17 00:00:00 2001
From: Roumen Petrov <local@example.net>
Date: Sat, 23 Feb 2013 13:53:16 +0200
Subject: [PATCH 03/16] MINGW: use header in lowercase

---
 Modules/signalmodule.c | 2 +-
 Objects/exceptions.c   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Modules/signalmodule.c b/Modules/signalmodule.c
index 0e91450..8b6c1c2 100644
--- a/Modules/signalmodule.c
+++ b/Modules/signalmodule.c
@@ -9,7 +9,7 @@
 #endif
 
 #ifdef MS_WINDOWS
-#include <Windows.h>
+#include <windows.h>
 #ifdef HAVE_PROCESS_H
 #include <process.h>
 #endif
diff --git a/Objects/exceptions.c b/Objects/exceptions.c
index 6b04700..79bbb8f 100644
--- a/Objects/exceptions.c
+++ b/Objects/exceptions.c
@@ -2327,7 +2327,7 @@ PyObject *PyExc_RecursionErrorInst = NULL;
     }
 
 #ifdef MS_WINDOWS
-#include <Winsock2.h>
+#include <winsock2.h>
 /* The following constants were added to errno.h in VS2010 but have
    preferred WSA equivalents. */
 #undef EADDRINUSE
-- 
1.7.12.1
