From 67d90c03230878f5720e1f02fa0c487dd2b5ee03 Mon Sep 17 00:00:00 2001
From: Roumen Petrov <local@example.net>
Date: Sat, 23 Feb 2013 19:47:15 +0200
Subject: [PATCH 07/16] MINGW: add wincrypt.h in Python/random.c

---
 Python/random.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Python/random.c b/Python/random.c
index 1ad4c3d..090cb90 100644
--- a/Python/random.c
+++ b/Python/random.c
@@ -1,6 +1,12 @@
 #include "Python.h"
 #ifdef MS_WINDOWS
 #include <windows.h>
+#ifdef __MINGW32__
+/* NOTE: All sample MSDN wincrypt programs include header below as it
+ * is required to ensure that application could be build properly in
+ * all cases. */
+#  include <wincrypt.h>
+#endif
 #else
 #include <fcntl.h>
 #endif
-- 
1.7.12.1