From aeeae4a9913a06d211ea2607c076d3a016dc1514 Mon Sep 17 00:00:00 2001
From: Roumen Petrov <local@example.net>
Date: Sat, 23 Feb 2013 19:59:56 +0200
Subject: [PATCH 14/16] MINGW: AC_LIBOBJ replacement of fileblocks

---
 Python/fileblocks.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 Python/fileblocks.c

diff --git a/Python/fileblocks.c b/Python/fileblocks.c
new file mode 100644
index 0000000..a869239
--- /dev/null
+++ b/Python/fileblocks.c
@@ -0,0 +1,17 @@
+/*
+-- Macro: AC_STRUCT_ST_BLOCKS
+    If 'struct stat' contains an 'st_blocks' member, define
+    'HAVE_STRUCT_STAT_ST_BLOCKS'.  Otherwise, require an 'AC_LIBOBJ'
+    replacement of 'fileblocks'.
+*/
+
+#if !HAVE_STRUCT_STAT_ST_BLOCKS
+/* If necessary you may see gnulib for replacement function:
+ * off_t st_blocks (off_t size).
+ * You may found code available under GPL2 or GPL3.
+ */
+
+/* This declaration is solely to ensure that after preprocessing
+   this file is never empty. */
+typedef int textutils_fileblocks_unused;
+#endif
-- 
1.7.12.1