diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-04-24 23:35:39.103741067 +0100
+++ b/setup.py	2013-04-24 23:35:39.448743357 +0100
@@ -2232,8 +2232,9 @@
                 newfilename = filename + fullversion
             else:
                 newfilename = filename + minoronly
-            log.info('renaming {} to {}'.format(filename, newfilename))
-            os.rename(filename, newfilename)
+            if not os.path.exists(newfilename):
+                log.info('renaming {} to {}'.format(filename, newfilename))
+                os.rename(filename, newfilename)
             newoutfiles.append(newfilename)
             if filename in updated_files:
                 newupdated_files.append(newfilename)
