diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2013-04-24 23:35:40.388749595 +0100
+++ b/configure.ac	2013-04-24 23:35:40.727751844 +0100
@@ -1678,7 +1678,9 @@
     ;;
 esac
 ac_save_cppflags="$CPPFLAGS"
+if test yes != "$cross_compiling"; then
 CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+fi
 AC_CHECK_HEADERS(asm/types.h conio.h curses.h direct.h dlfcn.h errno.h \
 fcntl.h grp.h \
 ieeefp.h io.h langinfo.h libintl.h ncurses.h process.h \
@@ -4660,7 +4662,9 @@
 fi
 
 ac_save_cppflags="$CPPFLAGS"
+if test yes != "$cross_compiling"; then
 CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+fi
 # On HP/UX 11.0, mvwdelch is a block with a return statement
 AC_MSG_CHECKING(whether mvwdelch is an expression)
 AC_CACHE_VAL(ac_cv_mvwdelch_is_expression,
