diff -urN a/Include/pyport.h b/Include/pyport.h
--- a/Include/pyport.h	2013-12-26 18:42:47.645850000 +0000
+++ b/Include/pyport.h	2013-12-26 18:42:51.374063300 +0000
@@ -13,6 +13,21 @@
 #include <stdint.h>
 #endif
 
+#ifdef __MINGW32__
+/* Translate GCC[mingw*] platform specific defines to those
+ * used in python code.
+ */
+#if !defined(MS_WIN64) && defined(_WIN64)
+#  define MS_WIN64
+#endif
+#if !defined(MS_WIN32) && defined(_WIN32)
+#  define MS_WIN32
+#endif
+#if !defined(MS_WINDOWS) && defined(MS_WIN32)
+#  define MS_WINDOWS
+#endif
+#endif /*def __MINGW32__*/
+
 /**************************************************************************
 Symbols and macros to supply platform-independent interfaces to basic
 C language & library operations whose spellings vary across platforms.
