diff -urN a/Modules/signalmodule.c b/Modules/signalmodule.c
--- a/Modules/signalmodule.c	2013-12-26 18:42:53.880206600 +0000
+++ b/Modules/signalmodule.c	2013-12-26 18:42:56.812374300 +0000
@@ -9,7 +9,7 @@
 #endif
 
 #ifdef MS_WINDOWS
-#include <Windows.h>
+#include <windows.h>
 #ifdef HAVE_PROCESS_H
 #include <process.h>
 #endif
diff -urN a/Objects/exceptions.c b/Objects/exceptions.c
--- a/Objects/exceptions.c	2013-12-26 18:42:54.115220000 +0000
+++ b/Objects/exceptions.c	2013-12-26 18:42:56.814374400 +0000
@@ -2327,7 +2327,7 @@
     }
 
 #ifdef MS_WINDOWS
-#include <Winsock2.h>
+#include <winsock2.h>
 /* The following constants were added to errno.h in VS2010 but have
    preferred WSA equivalents. */
 #undef EADDRINUSE
