diff -urN a/Python/random.c b/Python/random.c
--- a/Python/random.c	2013-12-26 18:43:16.696511600 +0000
+++ b/Python/random.c	2013-12-26 18:43:18.801632000 +0000
@@ -1,6 +1,12 @@
 #include "Python.h"
 #ifdef MS_WINDOWS
 #include <windows.h>
+#ifdef __MINGW32__
+/* NOTE: All sample MSDN wincrypt programs include header below as it
+ * is required to ensure that application could be build properly in
+ * all cases. */
+#  include <wincrypt.h>
+#endif
 #else
 #include <fcntl.h>
 #endif
