diff -Naur Python-3.5.2-orig/configure.ac Python-3.5.2/configure.ac
--- Python-3.6.1/configure.ac.orig	2017-06-13 12:46:54.007793900 +0200
+++ Python-3.6.1/configure.ac	2017-06-13 12:48:29.585260700 +0200
@@ -2995,6 +2995,14 @@
 fi])
 AC_MSG_RESULT($with_dbmliborder)
 
+
+# Determine if windows modules should be used.
+AC_SUBST(USE_WIN32_MODULE)
+USE_WIN32_MODULE='#'
+case $host in
+  *-*-mingw*) USE_WIN32_MODULE=;;
+esac
+
 # This is used to generate Setup.config
 AC_SUBST(USE_THREAD_MODULE)
 USE_THREAD_MODULE=""
diff -Naur Python-3.5.2-orig/Modules/Setup.config.in Python-3.5.2/Modules/Setup.config.in
--- Python-3.6.1/Modules/Setup.config.in.orig	2017-06-13 12:22:21.218555200 +0200
+++ Python-3.6.1/Modules/Setup.config.in	2017-06-13 12:51:10.275451600 +0200
@@ -9,5 +9,8 @@
 # Threading
 @USE_THREAD_MODULE@_thread _threadmodule.c
 
+# build-in modules for windows platform:
+@USE_WIN32_MODULE@winreg ../PC/winreg.c
+
 # The rest of the modules previously listed in this file are built
 # by the setup.py script in Python 2.1 and later.
diff -Naur Python-3.5.2-orig/PC/winreg.c Python-3.5.2/PC/winreg.c
--- Python-3.5.2-orig/PC/winreg.c	2016-06-26 00:38:38.000000000 +0300
+++ Python-3.5.2/PC/winreg.c	2016-07-12 14:20:50.352300700 +0300
@@ -16,6 +16,25 @@
 #include "structmember.h"
 #include "windows.h"
 
+#ifndef SIZEOF_HKEY
+/* used only here */
+#if defined(MS_WIN64)
+#  define SIZEOF_HKEY 8
+#elif defined(MS_WIN32)
+#  define SIZEOF_HKEY 4
+#else
+#  error "SIZEOF_HKEY is not defined"
+#endif
+#endif
+
+#ifndef REG_LEGAL_CHANGE_FILTER
+#define REG_LEGAL_CHANGE_FILTER        (\
+          REG_NOTIFY_CHANGE_NAME       |\
+          REG_NOTIFY_CHANGE_ATTRIBUTES |\
+          REG_NOTIFY_CHANGE_LAST_SET   |\
+          REG_NOTIFY_CHANGE_SECURITY   )
+#endif
+
 static BOOL PyHKEY_AsHKEY(PyObject *ob, HKEY *pRes, BOOL bNoneOK);
 static BOOL clinic_HKEY_converter(PyObject *ob, void *p);
 static PyObject *PyHKEY_FromHKEY(HKEY h);
