diff -Naur Python-3.5.0-orig/configure.ac Python-3.5.0/configure.ac
--- Python-3.5.0-orig/configure.ac	2015-09-21 13:41:16.957153700 +0300
+++ Python-3.5.0/configure.ac	2015-09-21 13:41:17.094140000 +0300
@@ -2750,6 +2750,13 @@
   *-*-mingw*) USE_WIN32_MODULE=;;
 esac
 
+# Determine if pwdmodule should be used.
+AC_SUBST(USE_PWD_MODULE)
+USE_PWD_MODULE=
+case $host in
+  *-*-mingw*) USE_PWD_MODULE='#';;
+esac
+
 # Determine if signalmodule should be used.
 AC_SUBST(USE_SIGNAL_MODULE)
 AC_SUBST(SIGNAL_OBJS)
diff -Naur Python-3.5.0-orig/Modules/Setup.config.in Python-3.5.0/Modules/Setup.config.in
--- Python-3.5.0-orig/Modules/Setup.config.in	2015-09-21 13:41:16.975151900 +0300
+++ Python-3.5.0/Modules/Setup.config.in	2015-09-21 13:41:17.109138500 +0300
@@ -6,6 +6,9 @@
 # init system calls(posix/nt/...) for INITFUNC (used by makesetup)
 @INITSYS@ posixmodule.c
 
+# This is needed to find out the user's home dir if $HOME is not set
+@USE_PWD_MODULE@pwd pwdmodule.c
+
 # Threading
 @USE_THREAD_MODULE@_thread _threadmodule.c
 
diff -Naur Python-3.5.0-orig/Modules/Setup.dist Python-3.5.0/Modules/Setup.dist
--- Python-3.5.0-orig/Modules/Setup.dist	2015-09-21 13:41:14.307418700 +0300
+++ Python-3.5.0/Modules/Setup.dist	2015-09-21 13:41:17.134136000 +0300
@@ -106,8 +106,6 @@
 # setup.py script in the root of the Python source tree.
 
 errno errnomodule.c		# posix (UNIX) errno values
-pwd pwdmodule.c			# this is needed to find out the user's home dir
-				# if $HOME is not set
 _sre _sre.c			# Fredrik Lundh's new regular expressions
 _codecs _codecsmodule.c		# access to the builtin codecs and codec registry
 _weakref _weakref.c		# weak references
