diff -Naur Python-3.6.5-orig/Lib/distutils/unixccompiler.py Python-3.6.5/Lib/distutils/unixccompiler.py
--- Python-3.6.5-orig/Lib/distutils/unixccompiler.py	2018-03-28 12:19:31.000000000 +0300
+++ Python-3.6.5/Lib/distutils/unixccompiler.py	2018-04-16 09:53:21.316521300 +0300
@@ -242,9 +242,13 @@
                 # -Wl whenever gcc was used in the past it is probably
                 # safest to keep doing so.
                 if sysconfig.get_config_var("GNULD") == "yes":
-                    # GNU ld needs an extra option to get a RUNPATH
+                    # GNU ELF ld needs an extra option to get a RUNPATH
                     # instead of just an RPATH.
-                    return "-Wl,--enable-new-dtags,-R" + dir
+                    if sys.platform in ["win32", "cygwin"] or \
+                       "mingw" in compiler:
+                        return []
+                    else:
+                        return "-Wl,--enable-new-dtags,-R" + dir
                 else:
                     return "-Wl,-R" + dir
             else:
