diff -Naur Python-3.6.5-orig/configure.ac Python-3.6.5/configure.ac
--- Python-3.6.5-orig/configure.ac	2018-04-16 09:53:28.897970900 +0300
+++ Python-3.6.5/configure.ac	2018-04-16 09:53:30.628384200 +0300
@@ -5523,6 +5523,16 @@
     ;;
 esac
 
+dnl Python interpreter main program for frozen scripts
+AC_SUBST(PYTHON_OBJS_FROZENMAIN)
+PYTHON_OBJS_FROZENMAIN="Python/frozenmain.o"
+case $host in
+  *-*-mingw*)
+    dnl 'PC/frozen_dllmain.c' - not yet
+    PYTHON_OBJS_FROZENMAIN=
+    ;;
+esac
+
 AC_SUBST(SRCDIRS)
 SRCDIRS="Parser Objects Python Modules Programs"
 case $host in
diff -Naur Python-3.6.5-orig/Makefile.pre.in Python-3.6.5/Makefile.pre.in
--- Python-3.6.5-orig/Makefile.pre.in	2018-04-16 09:53:28.897970900 +0300
+++ Python-3.6.5/Makefile.pre.in	2018-04-16 09:53:30.633384500 +0300
@@ -326,7 +326,7 @@
 		Python/codecs.o \
 		Python/dynamic_annotations.o \
 		Python/errors.o \
-		Python/frozenmain.o \
+		@PYTHON_OBJS_FROZENMAIN@ \
 		Python/future.o \
 		Python/getargs.o \
 		Python/getcompiler.o \
