diff -Naur Python-3.6.5-orig/setup.py Python-3.6.5/setup.py
--- Python-3.6.5-orig/setup.py	2018-04-16 09:53:25.770154600 +0300
+++ Python-3.6.5/setup.py	2018-04-16 09:53:32.180998700 +0300
@@ -1355,7 +1355,7 @@
             missing.append('_gdbm')
 
         # Unix-only modules
-        if host_platform != 'win32':
+        if not host_platform.startswith(('mingw', 'win')):
             # Steen Lumholt's termios module
             exts.append( Extension('termios', ['termios.c']) )
             # Jeremy Hylton's rlimit interface
