diff -Naur Python-3.7.0-orig/setup.py Python-3.7.0/setup.py
--- Python-3.7.0-orig/setup.py	2018-07-12 10:21:05.491125700 +0300
+++ Python-3.7.0/setup.py	2018-07-12 10:21:11.075935500 +0300
@@ -1321,7 +1321,7 @@
             missing.append('_gdbm')
 
         # Unix-only modules
-        if host_platform != 'win32':
+        if not host_platform.startswith(('mingw', 'win')):
             # Steen Lumholt's termios module
             exts.append( Extension('termios', ['termios.c']) )
             # Jeremy Hylton's rlimit interface
