diff -Naur Python-3.7.0-orig/setup.py Python-3.7.0/setup.py
--- Python-3.7.0-orig/setup.py	2018-06-30 11:16:13.307417300 +0300
+++ Python-3.7.0/setup.py	2018-06-30 11:16:13.525817600 +0300
@@ -1584,7 +1584,7 @@
             macros = dict()
             libraries = ['rt']
 
-        if host_platform == 'win32':
+        if host_platform.startswith(('mingw', 'win')):
             multiprocessing_srcs = [ '_multiprocessing/multiprocessing.c',
                                      '_multiprocessing/semaphore.c',
                                    ]
@@ -1596,8 +1596,12 @@
                 sysconfig.get_config_var('POSIX_SEMAPHORES_NOT_ENABLED')):
                 multiprocessing_srcs.append('_multiprocessing/semaphore.c')
 
+        multiprocessing_libs = []
+        if host_platform.startswith(('mingw', 'win')):
+            multiprocessing_libs += ['ws2_32']
         exts.append ( Extension('_multiprocessing', multiprocessing_srcs,
                                 define_macros=list(macros.items()),
+                                libraries=multiprocessing_libs,
                                 include_dirs=["Modules/_multiprocessing"]))
         # End multiprocessing
 
