diff -Naur Python-3.8.0-orig/setup.py Python-3.8.0/setup.py
--- Python-3.8.0-orig/setup.py	2019-10-22 10:01:17.378172600 +0300
+++ Python-3.8.0/setup.py	2019-10-22 10:01:29.904994600 +0300
@@ -1667,7 +1667,12 @@
                                    libraries=libs,
                                    include_dirs=["Modules/_multiprocessing"]))
 
+        multiprocessing_libs = []
+        if MS_WINDOWS:
+            multiprocessing_libs += ['ws2_32']
         self.add(Extension('_multiprocessing', multiprocessing_srcs,
+                           define_macros={},
+                           libraries=multiprocessing_libs,
                            include_dirs=["Modules/_multiprocessing"]))
 
     def detect_uuid(self):
