diff -Naur Python-3.5.0-orig/Lib/distutils/cygwinccompiler.py Python-3.5.0/Lib/distutils/cygwinccompiler.py
--- Python-3.5.0-orig/Lib/distutils/cygwinccompiler.py	2015-09-21 13:41:44.031446000 +0300
+++ Python-3.5.0/Lib/distutils/cygwinccompiler.py	2015-09-21 13:41:44.133435800 +0300
@@ -93,6 +93,7 @@
     obj_extension = ".o"
     static_lib_extension = ".a"
     shared_lib_extension = ".dll"
+    dylib_lib_extension = ".dll.a"
     static_lib_format = "lib%s%s"
     shared_lib_format = "%s%s"
     exe_extension = ".exe"
