diff -Naur Python-3.7.0-orig/Lib/distutils/cygwinccompiler.py Python-3.7.0/Lib/distutils/cygwinccompiler.py
--- Python-3.7.0-orig/Lib/distutils/cygwinccompiler.py	2018-07-12 10:21:26.379562400 +0300
+++ Python-3.7.0/Lib/distutils/cygwinccompiler.py	2018-07-12 10:21:26.707163000 +0300
@@ -93,6 +93,7 @@
     obj_extension = ".o"
     static_lib_extension = ".a"
     shared_lib_extension = ".dll"
+    dylib_lib_extension = ".dll.a"
     static_lib_format = "lib%s%s"
     shared_lib_format = "%s%s"
     exe_extension = ".exe"
