diff -Naur Python-3.5.2-orig/setup.py Python-3.5.2/setup.py
--- Python-3.5.2-orig/setup.py	2016-07-12 14:21:28.099300700 +0300
+++ Python-3.5.2/setup.py	2016-07-12 14:21:34.084800700 +0300
@@ -856,10 +856,13 @@
 
         if (ssl_incs is not None and
             ssl_libs is not None):
+            _ssl_libs = ['ssl', 'crypto']
+            if host_platform.startswith(('mingw', 'win')):
+                _ssl_libs += ['ws2_32']
             exts.append( Extension('_ssl', ['_ssl.c'],
                                    include_dirs = ssl_incs,
                                    library_dirs = ssl_libs,
-                                   libraries = ['ssl', 'crypto'],
+                                   libraries = _ssl_libs,
                                    depends = ['socketmodule.h']), )
         else:
             missing.append('_ssl')
