diff -Naur a/Include/node.h b/Include/node.h
--- a/Include/node.h	2014-03-17 06:31:29.000000000 +0400
+++ b/Include/node.h	2014-03-18 00:06:56.012600000 +0400
@@ -21,7 +21,7 @@
                                       char *str, int lineno, int col_offset);
 PyAPI_FUNC(void) PyNode_Free(node *n);
 #ifndef Py_LIMITED_API
-Py_ssize_t _PyNode_SizeOf(node *n);
+PyAPI_FUNC(Py_ssize_t) _PyNode_SizeOf(node *n);
 #endif
 
 /* Node access functions */
