From dbfbe78295fed05c495878ba70ee40960937090b Mon Sep 17 00:00:00 2001
From: Roumen Petrov <local@example.net>
Date: Sun, 24 Mar 2013 02:30:38 +0200
Subject: [PATCH 4/5] use replace instead rename to avoid failure on windows

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 49d2ec2..69e5020 100644
--- a/setup.py
+++ b/setup.py
@@ -2194,7 +2194,7 @@ class PyBuildScripts(build_scripts):
             else:
                 newfilename = filename + minoronly
             log.info('renaming {} to {}'.format(filename, newfilename))
-            os.rename(filename, newfilename)
+            os.replace(filename, newfilename)
             newoutfiles.append(newfilename)
             if filename in updated_files:
                 newupdated_files.append(newfilename)
-- 
1.7.12.1
