diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-03-18 00:06:55.841000000 +0400
+++ b/setup.py	2014-03-18 00:07:01.878200000 +0400
@@ -2193,7 +2193,7 @@
             else:
                 newfilename = filename + minoronly
             log.info('renaming {} to {}'.format(filename, newfilename))
-            os.rename(filename, newfilename)
+            os.replace(filename, newfilename)
             newoutfiles.append(newfilename)
             if filename in updated_files:
                 newupdated_files.append(newfilename)
