diff -Naur a/Lib/distutils/cygwinccompiler.py b/Lib/distutils/cygwinccompiler.py
--- a/Lib/distutils/cygwinccompiler.py	2014-03-18 00:06:53.953400000 +0400
+++ b/Lib/distutils/cygwinccompiler.py	2014-03-18 00:07:03.875000000 +0400
@@ -48,7 +48,6 @@
 import os
 import sys
 import copy
-from subprocess import Popen, PIPE, check_output
 import re
 
 from distutils.ccompiler import gen_preprocess_options, gen_lib_options
@@ -385,6 +384,7 @@
     executable = cmd.split()[0]
     if find_executable(executable) is None:
         return None
+    from subprocess import Popen, PIPE
     out = Popen(cmd, shell=True, stdout=PIPE).stdout
     try:
         out_string = out.read()
