diff -Naur Python-3.6.5-orig/Lib/site.py Python-3.6.5/Lib/site.py
--- Python-3.6.5-orig/Lib/site.py	2018-03-28 12:19:31.000000000 +0300
+++ Python-3.6.5/Lib/site.py	2018-04-16 09:53:56.915590500 +0300
@@ -268,6 +268,12 @@
             USER_SITE = get_path('purelib', 'osx_framework_user')
             return USER_SITE
 
+    if sys.platform == 'win32':
+        from sysconfig import _POSIX_BUILD
+        if _POSIX_BUILD:
+            USER_SITE = get_path('purelib', 'posix_user')
+            return USER_SITE
+
     USER_SITE = get_path('purelib', '%s_user' % os.name)
     return USER_SITE
 
@@ -298,12 +304,13 @@
     if prefixes is None:
         prefixes = PREFIXES
 
+    from sysconfig import _POSIX_BUILD
     for prefix in prefixes:
         if not prefix or prefix in seen:
             continue
         seen.add(prefix)
 
-        if os.sep == '/':
+        if _POSIX_BUILD:
             sitepackages.append(os.path.join(prefix, "lib",
                                         "python%d.%d" % sys.version_info[:2],
                                         "site-packages"))
