diff -Naur Python-3.7.0-orig/setup.py Python-3.7.0/setup.py
--- Python-3.7.0-orig/setup.py	2018-06-30 11:16:41.395866700 +0300
+++ Python-3.7.0/setup.py	2018-06-30 11:16:54.421889600 +0300
@@ -1967,6 +1967,10 @@
         return True
 
     def configure_ctypes(self, ext):
+        if host_platform == 'win32':
+            ext.libraries.extend(['ole32', 'oleaut32', 'uuid'])
+            ext.export_symbols.extend(['DllGetClassObject PRIVATE',
+                                       'DllCanUnloadNow PRIVATE'])
         if not self.use_system_libffi:
             if host_platform == 'darwin':
                 return self.configure_ctypes_darwin(ext)
