diff -Naur Python-3.6.5-orig/configure.ac Python-3.6.5/configure.ac
--- Python-3.6.5-orig/configure.ac	2018-04-16 09:54:02.684643700 +0300
+++ Python-3.6.5/configure.ac	2018-04-16 09:54:04.531856800 +0300
@@ -608,6 +608,16 @@
 fi
 AC_MSG_RESULT($MACHDEP)
 
+# Windows uses ; to separate paths, everything else uses :
+AC_MSG_CHECKING(DELIM)
+DELIM=:
+if test "$MACHDEP" = "win"
+then
+	DELIM=\;
+fi
+AC_MSG_RESULT([$DELIM])
+AC_SUBST(DELIM)
+
 AC_MSG_CHECKING([for init system calls])
 AC_SUBST(INITSYS)
 case $host in
diff -Naur Python-3.6.5-orig/Include/osdefs.h Python-3.6.5/Include/osdefs.h
--- Python-3.6.5-orig/Include/osdefs.h	2018-03-28 12:19:31.000000000 +0300
+++ Python-3.6.5/Include/osdefs.h	2018-04-16 09:54:04.547456800 +0300
@@ -12,6 +12,7 @@
 #define ALTSEP L'/'
 #define MAXPATHLEN 256
 #define DELIM L';'
+#define DELIMSTR L";"
 #endif
 
 /* Filename separator */
@@ -39,6 +40,7 @@
 /* Search path entry delimiter */
 #ifndef DELIM
 #define DELIM L':'
+#define DELIMSTR L":"
 #endif
 
 #ifdef __cplusplus
diff -Naur Python-3.6.5-orig/Makefile.pre.in Python-3.6.5/Makefile.pre.in
--- Python-3.6.5-orig/Makefile.pre.in	2018-04-16 09:53:31.957197100 +0300
+++ Python-3.6.5/Makefile.pre.in	2018-04-16 09:54:04.547456800 +0300
@@ -110,7 +110,8 @@
 # Strict or non-strict aliasing flags used to compile dtoa.c, see above
 CFLAGS_ALIASING=@CFLAGS_ALIASING@
 
-
+# ; on Windows otherwise :
+DELIM=		@DELIM@
 # Machine-dependent subdirectories
 MACHDEP=	@MACHDEP@
 
diff -Naur Python-3.6.5-orig/Modules/Setup.dist Python-3.6.5/Modules/Setup.dist
--- Python-3.6.5-orig/Modules/Setup.dist	2018-04-16 09:53:15.174261400 +0300
+++ Python-3.6.5/Modules/Setup.dist	2018-04-16 09:54:04.563056900 +0300
@@ -84,14 +84,14 @@
 # Empty since this is now just the runtime prefix.
 DESTPATH=
 
-# Site specific path components -- should begin with : if non-empty
+# Site specific path components -- should begin with $(DELIM) if non-empty
 SITEPATH=
 
 # Standard path components for test modules
 TESTPATH=
 
 # Path components for machine- or system-dependent modules and shared libraries
-MACHDEPPATH=:$(PLATDIR)
+MACHDEPPATH=$(DELIM)$(PLATDIR)
 EXTRAMACHDEPPATH=
 
 COREPYTHONPATH=$(DESTPATH)$(SITEPATH)$(TESTPATH)$(MACHDEPPATH)$(EXTRAMACHDEPPATH)
