diff -Naur Python-3.5.2-orig/configure.ac Python-3.5.2/configure.ac
--- Python-3.5.2-orig/configure.ac	2016-07-12 14:21:52.030300700 +0300
+++ Python-3.5.2/configure.ac	2016-07-12 14:21:54.212300700 +0300
@@ -603,6 +603,16 @@
 fi
 AC_MSG_RESULT($MACHDEP)
 
+# Windows uses ; to separate paths, everything else uses :
+AC_MSG_CHECKING(DELIM)
+DELIM=:
+if test "$MACHDEP" = "win"
+then
+	DELIM=\;
+fi
+AC_MSG_RESULT([$DELIM])
+AC_SUBST(DELIM)
+
 AC_MSG_CHECKING([for init system calls])
 AC_SUBST(INITSYS)
 case $host in
diff -Naur Python-3.5.2-orig/Include/osdefs.h Python-3.5.2/Include/osdefs.h
--- Python-3.5.2-orig/Include/osdefs.h	2016-06-26 00:38:35.000000000 +0300
+++ Python-3.5.2/Include/osdefs.h	2016-07-12 14:21:54.227300700 +0300
@@ -12,6 +12,7 @@
 #define ALTSEP L'/'
 #define MAXPATHLEN 256
 #define DELIM L';'
+#define DELIMSTR L";"
 #endif
 
 /* Filename separator */
@@ -39,6 +40,7 @@
 /* Search path entry delimiter */
 #ifndef DELIM
 #define DELIM L':'
+#define DELIMSTR L":"
 #endif
 
 #ifdef __cplusplus
--- Python-3.6.2rc1/Makefile.pre.in.orig	2017-06-18 14:53:39.660917400 +0200
+++ Python-3.6.2rc1/Makefile.pre.in	2017-06-18 14:54:31.765009000 +0200
@@ -110,7 +110,8 @@
 # Strict or non-strict aliasing flags used to compile dtoa.c, see above
 CFLAGS_ALIASING=@CFLAGS_ALIASING@
 
-
+# ; on Windows otherwise :
+DELIM=		@DELIM@
 # Machine-dependent subdirectories
 MACHDEP=	@MACHDEP@
 
diff -Naur Python-3.5.2-orig/Modules/Setup.dist Python-3.5.2/Modules/Setup.dist
--- Python-3.5.2-orig/Modules/Setup.dist	2016-07-12 14:20:53.918300700 +0300
+++ Python-3.5.2/Modules/Setup.dist	2016-07-12 14:21:54.256800700 +0300
@@ -84,14 +84,14 @@
 # Empty since this is now just the runtime prefix.
 DESTPATH=
 
-# Site specific path components -- should begin with : if non-empty
+# Site specific path components -- should begin with $(DELIM) if non-empty
 SITEPATH=
 
 # Standard path components for test modules
 TESTPATH=
 
 # Path components for machine- or system-dependent modules and shared libraries
-MACHDEPPATH=:$(PLATDIR)
+MACHDEPPATH=$(DELIM)$(PLATDIR)
 EXTRAMACHDEPPATH=
 
 COREPYTHONPATH=$(DESTPATH)$(SITEPATH)$(TESTPATH)$(MACHDEPPATH)$(EXTRAMACHDEPPATH)
