diff -Naur Python-3.6.5-orig/Modules/Setup.config.in Python-3.6.5/Modules/Setup.config.in
--- Python-3.6.5-orig/Modules/Setup.config.in	2018-04-16 09:53:15.158661400 +0300
+++ Python-3.6.5/Modules/Setup.config.in	2018-04-16 09:54:11.159312400 +0300
@@ -14,6 +14,7 @@
 
 # build-in modules for windows platform:
 @USE_WIN32_MODULE@winreg ../PC/winreg.c
+@USE_WIN32_MODULE@msvcrt ../PC/msvcrtmodule.c
 
 # The rest of the modules previously listed in this file are built
 # by the setup.py script in Python 2.1 and later.
