diff -Naur Python-3.5.2-orig/Modules/Setup.config.in Python-3.5.2/Modules/Setup.config.in
--- Python-3.5.2-orig/Modules/Setup.config.in	2016-07-12 14:20:53.904300700 +0300
+++ Python-3.5.2/Modules/Setup.config.in	2016-07-12 14:22:04.057800700 +0300
@@ -17,6 +17,7 @@
 
 # build-in modules for windows platform:
 @USE_WIN32_MODULE@winreg ../PC/winreg.c
+@USE_WIN32_MODULE@msvcrt ../PC/msvcrtmodule.c
 
 # The rest of the modules previously listed in this file are built
 # by the setup.py script in Python 2.1 and later.
