diff -Naur a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2014-03-17 10:46:18.126400000 +0400
+++ b/Makefile.pre.in	2014-03-17 10:46:26.394400000 +0400
@@ -969,6 +969,7 @@
 	if test -f $(LDLIBRARY); then \
 		if test -n "$(DLLLIBRARY)" ; then \
 			$(INSTALL_SHARED) $(DLLLIBRARY) $(DESTDIR)$(BINDIR); \
+			mkdir -p $(DESTDIR)$(LIBPL); $(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBPL); \
 		else \
 			$(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBDIR)/$(INSTSONAME); \
 			if test $(LDLIBRARY) != $(INSTSONAME); then \
