diff -Naur Python-3.6.5-orig/Lib/distutils/cygwinccompiler.py Python-3.6.5/Lib/distutils/cygwinccompiler.py
--- Python-3.6.5-orig/Lib/distutils/cygwinccompiler.py	2018-04-16 09:53:55.144373100 +0300
+++ Python-3.6.5/Lib/distutils/cygwinccompiler.py	2018-04-16 09:54:15.851368900 +0300
@@ -58,6 +58,7 @@
 from distutils import log
 from distutils.version import LooseVersion
 from distutils.spawn import find_executable
+from subprocess import Popen, PIPE, check_output
 
 def get_msvcr():
     """Include the appropriate MSVC runtime library if Python was built
@@ -373,7 +374,7 @@
         return (CONFIG_H_UNCERTAIN,
                 "couldn't read '%s': %s" % (fn, exc.strerror))
 
-RE_VERSION = re.compile(br'(\d+\.\d+(\.\d+)*)')
+RE_VERSION = re.compile(br'[\D\s]*(\d+\.\d+(\.\d+)*)[\D\s]*$')
 
 def _find_exe_version(cmd):
     """Find the version of an executable by running `cmd` in the shell.
@@ -402,7 +403,16 @@
 
     If not possible it returns None for it.
     """
-    commands = ['gcc -dumpversion', 'ld -v', 'dllwrap --version']
+    gcc = os.environ.get('CC') or 'gcc'
+    ld = 'ld'
+    out = Popen(gcc+' --print-prog-name ld', shell=True, stdout=PIPE).stdout
+    try:
+        ld = test=str(out.read(),encoding='utf-8').strip()
+    finally:
+        out.close()
+    dllwrap = os.environ.get('DLLWRAP') or 'dllwrap'
+    # MinGW64 doesn't have i686-w64-mingw32-ld, so instead we ask gcc.
+    commands = [gcc+' -dumpversion', ld+' -v', dllwrap+' --version']
     return tuple([_find_exe_version(cmd) for cmd in commands])
 
 def is_cygwingcc():
diff -Naur Python-3.6.5-orig/Modules/Setup.config.in Python-3.6.5/Modules/Setup.config.in
--- Python-3.6.5-orig/Modules/Setup.config.in	2018-04-16 09:54:12.403719600 +0300
+++ Python-3.6.5/Modules/Setup.config.in	2018-04-16 09:54:15.851368900 +0300
@@ -15,6 +15,7 @@
 # build-in modules for windows platform:
 @USE_WIN32_MODULE@winreg ../PC/winreg.c
 @USE_WIN32_MODULE@msvcrt ../PC/msvcrtmodule.c
+@USE_WIN32_MODULE@_winapi _winapi.c
 
 # The rest of the modules previously listed in this file are built
 # by the setup.py script in Python 2.1 and later.
diff -Naur Python-3.6.5-orig/setup.py Python-3.6.5/setup.py
--- Python-3.6.5-orig/setup.py	2018-04-16 09:54:04.234456200 +0300
+++ Python-3.6.5/setup.py	2018-04-16 09:54:15.866968900 +0300
@@ -1007,7 +1007,9 @@
             exts.append( Extension('msvcrt', [os.path.join(pc_srcdir, p)
                 for p in ['msvcrtmodule.c']]) )
 
-            exts.append( Extension('_winapi', ['_winapi.c']) )
+            # Added to Setup.config.in as now needed earlier since I
+            # use subprocess (which uses Popen) in cygwinccompiler.py
+            # exts.append( Extension('_winapi', ['_winapi.c']) )
 
             exts.append( Extension('_msi', [os.path.join(pc_srcdir, p)
                 for p in ['_msi.c']],
