diff -Naur a/Lib/distutils/cygwinccompiler.py b/Lib/distutils/cygwinccompiler.py
--- a/Lib/distutils/cygwinccompiler.py	2014-03-18 00:07:05.669000000 +0400
+++ b/Lib/distutils/cygwinccompiler.py	2014-03-18 00:07:36.900200000 +0400
@@ -58,6 +58,7 @@
 from distutils import log
 from distutils.version import LooseVersion
 from distutils.spawn import find_executable
+from subprocess import Popen, PIPE, check_output
 
 def get_msvcr():
     """Include the appropriate MSVC runtime library if Python was built
@@ -373,7 +374,7 @@
         return (CONFIG_H_UNCERTAIN,
                 "couldn't read '%s': %s" % (fn, exc.strerror))
 
-RE_VERSION = re.compile(b'(\d+\.\d+(\.\d+)*)')
+RE_VERSION = re.compile(b'[\D\s]*(\d+\.\d+(\.\d+)*)[\D\s]*$')
 
 def _find_exe_version(cmd):
     """Find the version of an executable by running `cmd` in the shell.
@@ -402,7 +403,16 @@
 
     If not possible it returns None for it.
     """
-    commands = ['gcc -dumpversion', 'ld -v', 'dllwrap --version']
+    gcc = os.environ.get('CC') or 'gcc'
+    ld = 'ld'
+    out = Popen(gcc+' --print-prog-name ld', shell=True, stdout=PIPE).stdout
+    try:
+        ld = test=str(out.read(),encoding='utf-8').strip()
+    finally:
+        out.close()
+    dllwrap = os.environ.get('DLLWRAP') or 'dllwrap'
+    # MinGW64 doesn't have i686-w64-mingw32-ld, so instead we ask gcc.
+    commands = [gcc+' -dumpversion', ld+' -v', dllwrap+' --version']
     return tuple([_find_exe_version(cmd) for cmd in commands])
 
 def is_cygwingcc():
diff -Naur a/Modules/Setup.config.in b/Modules/Setup.config.in
--- a/Modules/Setup.config.in	2014-03-18 00:07:26.853800000 +0400
+++ b/Modules/Setup.config.in	2014-03-18 00:07:36.900200000 +0400
@@ -19,6 +19,7 @@
 @USE_WIN32_MODULE@winreg ../PC/winreg.c
 @USE_WIN32_MODULE@time timemodule.c
 @USE_WIN32_MODULE@msvcrt ../PC/msvcrtmodule.c
+@USE_WIN32_MODULE@_winapi _winapi.c
 
 # The rest of the modules previously listed in this file are built
 # by the setup.py script in Python 2.1 and later.
diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-03-18 00:07:17.259800000 +0400
+++ b/setup.py	2014-03-18 00:07:36.900200000 +0400
@@ -890,7 +890,9 @@
             exts.append( Extension('msvcrt', [os.path.join(pc_srcdir, p)
                 for p in ['msvcrtmodule.c']]) )
 
-            exts.append( Extension('_winapi', ['_winapi.c']) )
+            # Added to Setup.config.in as now needed earlier since I
+            # use subprocess (which uses Popen) in cygwinccompiler.py
+            # exts.append( Extension('_winapi', ['_winapi.c']) )
 
         def allow_db_ver(db_ver):
             """Returns a boolean if the given BerkeleyDB version is acceptable.
