diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-03-17 10:46:35.972800000 +0400
+++ b/setup.py	2014-03-17 10:46:38.078800000 +0400
@@ -444,10 +444,10 @@
         if not cross_compiling:
             add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
+            self.add_multiarch_paths()
         # only change this for cross builds for 3.3, issues on Mageia
         if cross_compiling:
             self.add_gcc_paths()
-        self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
