diff -Naur Python-3.6.5-orig/setup.py Python-3.6.5/setup.py
--- Python-3.6.5-orig/setup.py	2018-04-16 09:54:17.167383600 +0300
+++ Python-3.6.5/setup.py	2018-04-16 09:54:18.863993300 +0300
@@ -521,10 +521,10 @@
         if not cross_compiling:
             add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
+            self.add_multiarch_paths()
         # only change this for cross builds for 3.3, issues on Mageia
         if cross_compiling:
             self.add_gcc_paths()
-        self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
