diff -Naur Python-3.8.0-orig/Lib/compileall.py Python-3.8.0/Lib/compileall.py
--- Python-3.8.0-orig/Lib/compileall.py	2019-10-14 16:34:47.000000000 +0300
+++ Python-3.8.0/Lib/compileall.py	2019-10-22 10:03:15.206179600 +0300
@@ -36,6 +36,8 @@
         if name == '__pycache__':
             continue
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
