diff -Naur Python-3.6.5-orig/Lib/compileall.py Python-3.6.5/Lib/compileall.py
--- Python-3.6.5-orig/Lib/compileall.py	2018-03-28 12:19:31.000000000 +0300
+++ Python-3.6.5/Lib/compileall.py	2018-04-16 09:54:20.372803500 +0300
@@ -40,6 +40,8 @@
         if name == '__pycache__':
             continue
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
