diff -urN a/Lib/compileall.py b/Lib/compileall.py
--- a/Lib/compileall.py	2013-12-26 18:48:37.780876600 +0000
+++ b/Lib/compileall.py	2013-12-26 18:48:42.628153800 +0000
@@ -47,6 +47,8 @@
         if name == '__pycache__':
             continue
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
