diff -urN a/Lib/compileall.py b/Lib/compileall.py
--- a/Lib/compileall.py	2014-10-11 14:22:16.521542500 +0100
+++ b/Lib/compileall.py	2014-10-11 14:22:18.904678800 +0100
@@ -46,6 +46,8 @@
         if name == '__pycache__':
             continue
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
