diff -Naur Python-3.5.2-orig/Lib/compileall.py Python-3.5.2/Lib/compileall.py
--- Python-3.5.2-orig/Lib/compileall.py	2016-06-26 00:38:35.000000000 +0300
+++ Python-3.5.2/Lib/compileall.py	2016-07-12 14:22:17.360300700 +0300
@@ -38,6 +38,8 @@
         if name == '__pycache__':
             continue
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
