diff -Naur a/Lib/compileall.py b/Lib/compileall.py
--- a/Lib/compileall.py	2014-03-17 06:31:29.000000000 +0400
+++ b/Lib/compileall.py	2014-03-18 00:07:43.093400000 +0400
@@ -47,6 +47,8 @@
         if name == '__pycache__':
             continue
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
