diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-12-26 18:48:53.261762100 +0000
+++ b/setup.py	2013-12-26 18:48:56.395941300 +0000
@@ -2197,8 +2197,9 @@
                 newfilename = filename + fullversion
             else:
                 newfilename = filename + minoronly
-            log.info('renaming {} to {}'.format(filename, newfilename))
-            os.rename(filename, newfilename)
+            if not os.path.exists(newfilename):
+                log.info('renaming {} to {}'.format(filename, newfilename))
+                os.rename(filename, newfilename)
             newoutfiles.append(newfilename)
             if filename in updated_files:
                 newupdated_files.append(newfilename)
