diff -Naur Python-3.5.0-orig/configure.ac Python-3.5.0/configure.ac
--- Python-3.5.0-orig/configure.ac	2015-09-21 13:42:11.881910400 +0300
+++ Python-3.5.0/configure.ac	2015-09-21 13:42:23.013007200 +0300
@@ -4969,7 +4969,9 @@
 
 # first curses header check
 ac_save_cppflags="$CPPFLAGS"
+if test yes != "$cross_compiling"; then
 CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+fi
 
 AC_CHECK_HEADERS(curses.h ncurses.h)
 
