diff -Naur Python-3.7.0-orig/Include/py_curses.h Python-3.7.0/Include/py_curses.h
--- Python-3.7.0-orig/Include/py_curses.h	2018-06-27 06:07:35.000000000 +0300
+++ Python-3.7.0/Include/py_curses.h	2018-06-30 11:17:25.122743500 +0300
@@ -36,6 +36,13 @@
 #include <curses.h>
 #endif
 
+#if defined(__MINGW32__)
+#include <windows.h>
+#if !defined(_ISPAD)
+#define _ISPAD 0x10
+#endif
+#endif
+
 #ifdef HAVE_NCURSES_H
 /* configure was checking <curses.h>, but we will
    use <ncurses.h>, which has some or all these features. */
