diff -Naur Python-3.5.2-orig/Modules/parsermodule.c Python-3.5.2/Modules/parsermodule.c
--- Python-3.5.2-orig/Modules/parsermodule.c	2016-06-26 00:38:38.000000000 +0300
+++ Python-3.5.2/Modules/parsermodule.c	2016-07-12 14:22:29.674300700 +0300
@@ -42,7 +42,7 @@
 #undef Yield
 #include "ast.h"
 
-extern grammar _PyParser_Grammar; /* From graminit.c */
+PyAPI_DATA(grammar) _PyParser_Grammar; /* From graminit.c */
 
 #ifdef lint
 #include <note.h>
diff -Naur Python-3.5.2-orig/Parser/metagrammar.c Python-3.5.2/Parser/metagrammar.c
--- Python-3.5.2-orig/Parser/metagrammar.c	2016-06-26 00:38:38.000000000 +0300
+++ Python-3.5.2/Parser/metagrammar.c	2016-07-12 14:22:29.718800700 +0300
@@ -139,7 +139,7 @@
     {7, 0},
     {8, 0},
 };
-static grammar _PyParser_Grammar = {
+static grammar _PyParser_MetaGrammar = {
     6,
     dfas,
     {19, labels},
@@ -149,7 +149,7 @@
 grammar *
 meta_grammar(void)
 {
-    return &_PyParser_Grammar;
+    return &_PyParser_MetaGrammar;
 }
 
 grammar *
