diff -Naur a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2014-03-18 00:07:46.884200000 +0400
+++ b/Makefile.pre.in	2014-03-18 00:07:59.255000000 +0400
@@ -672,13 +672,13 @@
 ############################################################################
 # Importlib
 
-Modules/_freeze_importlib: Modules/_freeze_importlib.o $(LIBRARY_OBJS_OMIT_FROZEN)
+_freeze_importlib$(EXE): Modules/_freeze_importlib.o $(LIBRARY_OBJS_OMIT_FROZEN)
 	$(LINKCC) $(PY_LDFLAGS) -o $@ Modules/_freeze_importlib.o $(LIBRARY_OBJS_OMIT_FROZEN) $(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST)
 
-Python/importlib.h: $(srcdir)/Lib/importlib/_bootstrap.py Modules/_freeze_importlib.c
-	$(MAKE) Modules/_freeze_importlib
-	./Modules/_freeze_importlib \
-		$(srcdir)/Lib/importlib/_bootstrap.py Python/importlib.h
+$(srcdir)/Python/importlib.h: $(srcdir)/Lib/importlib/_bootstrap.py $(srcdir)/Modules/_freeze_importlib.c
+	$(MAKE) _freeze_importlib$(EXE)
+	./_freeze_importlib$(EXE) \
+		$(srcdir)/Lib/importlib/_bootstrap.py $(srcdir)/Python/importlib.h
 
 
 ############################################################################
@@ -1545,7 +1545,7 @@
 	find build -name '*.py[co]' -exec rm -f {} ';' || true
 	-rm -f pybuilddir.txt
 	-rm -f Lib/lib2to3/*Grammar*.pickle
-	-rm -f Modules/_testembed Modules/_freeze_importlib
+	-rm -f Modules/_testembed _freeze_importlib$(EXE)
 
 profile-removal:
 	find . -name '*.gc??' -exec rm -f {} ';'
