diff -Naur a/Modules/getpath.c b/Modules/getpath.c
--- a/Modules/getpath.c	2014-03-18 00:07:25.106600000 +0400
+++ b/Modules/getpath.c	2014-03-18 00:08:05.089400000 +0400
@@ -795,7 +795,10 @@
 
     bufsz += wcslen(zip_path) + 1;
     bufsz += wcslen(exec_prefix) + 1;
-
+#ifdef MS_WINDOWS
+    if (_exec_prefix)
+        bufsz += wcslen(_exec_prefix) + 1;
+#endif
     buf = (wchar_t *)PyMem_Malloc(bufsz * sizeof(wchar_t));
     if (buf == NULL) {
         Py_FatalError(
@@ -871,6 +874,13 @@
     }
     else
         wcsncpy(exec_prefix, _exec_prefix, MAXPATHLEN);
+#ifdef MS_WINDOWS
+    if (_exec_prefix) {
+        /* Next add bin folder where the exe was found, for System DLLs e.g. tcl86.dll, tk86.dll */
+        wcscat(buf, delimiter);
+        wcscat(buf, _exec_prefix);
+      }
+#endif
 
 #ifdef MS_WINDOWS
     if (_path) {
