diff -Naur Python-3.6.5-orig/Modules/getpath.c Python-3.6.5/Modules/getpath.c
--- Python-3.6.5-orig/Modules/getpath.c	2018-04-16 09:54:10.929109600 +0300
+++ Python-3.6.5/Modules/getpath.c	2018-04-16 09:54:34.743149100 +0300
@@ -771,7 +771,10 @@
 
     bufsz += wcslen(zip_path) + 1;
     bufsz += wcslen(exec_prefix) + 1;
-
+#ifdef MS_WINDOWS
+    if (_exec_prefix)
+        bufsz += wcslen(_exec_prefix) + 1;
+#endif
     buf = PyMem_RawMalloc(bufsz * sizeof(wchar_t));
     if (buf == NULL) {
         Py_FatalError(
@@ -850,6 +853,13 @@
     }
     else
         wcsncpy(exec_prefix, _exec_prefix, MAXPATHLEN);
+#ifdef MS_WINDOWS
+    if (_exec_prefix) {
+        /* Next add bin folder where the exe was found, for System DLLs e.g. tcl86.dll, tk86.dll */
+        wcscat(buf, delimiter);
+        wcscat(buf, _exec_prefix);
+      }
+#endif
 
 #ifdef MS_WINDOWS
     if (_path) {
