diff -urN a/Include/osdefs.h b/Include/osdefs.h
--- a/Include/osdefs.h	2014-10-11 14:23:05.655352800 +0100
+++ b/Include/osdefs.h	2014-10-11 14:23:07.572462400 +0100
@@ -17,7 +17,6 @@
 #if defined(MS_WINDOWS) || defined(__BORLANDC__) || defined(__WATCOMC__) || defined(__DJGPP__)
 #define SEP L'\\'
 #define ALTSEP L'/'
-#define MAXPATHLEN 256
 #define DELIM L';'
 #define DELIMSTR ";"
 #endif
diff -urN a/Modules/main.c b/Modules/main.c
--- a/Modules/main.c	2014-10-11 14:23:05.736357400 +0100
+++ b/Modules/main.c	2014-10-11 14:23:07.573462500 +0100
@@ -9,7 +9,6 @@
 #include <windows.h>
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
-#define PATH_MAX MAXPATHLEN
 #endif
 #endif
 
diff -urN a/Python/pythonrun.c b/Python/pythonrun.c
--- a/Python/pythonrun.c	2014-10-11 14:23:04.818304900 +0100
+++ b/Python/pythonrun.c	2014-10-11 14:23:07.574462500 +0100
@@ -32,7 +32,6 @@
 #ifdef MS_WINDOWS
 #undef BYTE
 #include "windows.h"
-#define PATH_MAX MAXPATHLEN
 #endif
 
 #ifdef __gnu_hurd__
