diff -r 20dc8d6430eb Modules/main.c
--- a/Modules/main.c	Mon Feb 10 19:17:46 2014 +0100
+++ b/Modules/main.c	Tue Feb 11 11:50:12 2014 -0500
@@ -9,7 +9,6 @@
 #include <windows.h>
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
-#define PATH_MAX MAXPATHLEN
 #endif
 #endif
 
diff -r 20dc8d6430eb Python/pythonrun.c
--- a/Python/pythonrun.c	Mon Feb 10 19:17:46 2014 +0100
+++ b/Python/pythonrun.c	Tue Feb 11 11:50:12 2014 -0500
@@ -32,7 +32,6 @@
 #ifdef MS_WINDOWS
 #undef BYTE
 #include "windows.h"
-#define PATH_MAX MAXPATHLEN
 #endif
 
 _Py_IDENTIFIER(builtins);
