diff -Naur Python-3.8.0-orig/Modules/_gdbmmodule.c Python-3.8.0/Modules/_gdbmmodule.c
--- Python-3.8.0-orig/Modules/_gdbmmodule.c	2019-10-14 16:34:47.000000000 +0300
+++ Python-3.8.0/Modules/_gdbmmodule.c	2019-10-22 10:03:53.161046300 +0300
@@ -11,7 +11,7 @@
 #include <fcntl.h>
 #include "gdbm.h"
 
-#if defined(WIN32) && !defined(__CYGWIN__)
+#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #include "gdbmerrno.h"
 extern const char * gdbm_strerror(gdbm_error);
 #endif
