diff -Naur Python-3.5.2-orig/Modules/_gdbmmodule.c Python-3.5.2/Modules/_gdbmmodule.c
--- Python-3.5.2-orig/Modules/_gdbmmodule.c	2016-06-26 00:38:38.000000000 +0300
+++ Python-3.5.2/Modules/_gdbmmodule.c	2016-07-12 14:22:44.958300700 +0300
@@ -11,7 +11,7 @@
 #include <fcntl.h>
 #include "gdbm.h"
 
-#if defined(WIN32) && !defined(__CYGWIN__)
+#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #include "gdbmerrno.h"
 extern const char * gdbm_strerror(gdbm_error);
 #endif
