diff -Naur a/Include/pythonrun.h b/Include/pythonrun.h
--- a/Include/pythonrun.h	2014-05-19 09:19:37.000000000 +0400
+++ b/Include/pythonrun.h	2014-05-20 00:57:08.242000000 +0400
@@ -214,7 +214,7 @@
 PyAPI_FUNC(wchar_t *) Py_GetPath(void);
 PyAPI_FUNC(void)      Py_SetPath(const wchar_t *);
 #ifdef MS_WINDOWS
-int _Py_CheckPython3();
+int _Py_CheckPython3(void);
 #endif
 
 /* In their own files */
