diff -Naur Python-3.6.5-orig/Include/pylifecycle.h Python-3.6.5/Include/pylifecycle.h
--- Python-3.6.5-orig/Include/pylifecycle.h	2018-04-16 09:54:07.672880800 +0300
+++ Python-3.6.5/Include/pylifecycle.h	2018-04-16 09:54:45.464450300 +0300
@@ -66,7 +66,7 @@
 PyAPI_FUNC(wchar_t *) Py_GetPath(void);
 PyAPI_FUNC(void)      Py_SetPath(const wchar_t *);
 #ifdef MS_WINDOWS
-int _Py_CheckPython3();
+int _Py_CheckPython3(void);
 #endif
 
 /* In their own files */
