diff -Naur Python-3.5.0-orig/Include/pylifecycle.h Python-3.5.0/Include/pylifecycle.h
--- Python-3.5.0-orig/Include/pylifecycle.h	2015-09-21 13:42:04.489399200 +0300
+++ Python-3.5.0/Include/pylifecycle.h	2015-09-21 13:42:45.489672800 +0300
@@ -65,7 +65,7 @@
 PyAPI_FUNC(wchar_t *) Py_GetPath(void);
 PyAPI_FUNC(void)      Py_SetPath(const wchar_t *);
 #ifdef MS_WINDOWS
-int _Py_CheckPython3();
+int _Py_CheckPython3(void);
 #endif
 
 /* In their own files */
