diff -Naur Python-3.5.2-orig/Include/pylifecycle.h Python-3.5.2/Include/pylifecycle.h
--- Python-3.5.2-orig/Include/pylifecycle.h	2016-07-12 14:21:59.900800700 +0300
+++ Python-3.5.2/Include/pylifecycle.h	2016-07-12 14:22:48.365800700 +0300
@@ -65,7 +65,7 @@
 PyAPI_FUNC(wchar_t *) Py_GetPath(void);
 PyAPI_FUNC(void)      Py_SetPath(const wchar_t *);
 #ifdef MS_WINDOWS
-int _Py_CheckPython3();
+int _Py_CheckPython3(void);
 #endif
 
 /* In their own files */
