diff -urN a/Include/pythonrun.h b/Include/pythonrun.h
--- a/Include/pythonrun.h	2014-10-11 14:23:24.506431000 +0100
+++ b/Include/pythonrun.h	2014-10-11 14:23:26.506545400 +0100
@@ -214,7 +214,7 @@
 PyAPI_FUNC(wchar_t *) Py_GetPath(void);
 PyAPI_FUNC(void)      Py_SetPath(const wchar_t *);
 #ifdef MS_WINDOWS
-int _Py_CheckPython3();
+int _Py_CheckPython3(void);
 #endif
 
 /* In their own files */
