diff -Naur Python-3.5.2-orig/Lib/ssl.py Python-3.5.2/Lib/ssl.py
--- Python-3.5.2-orig/Lib/ssl.py	2016-06-26 00:38:36.000000000 +0300
+++ Python-3.5.2/Lib/ssl.py	2016-07-12 14:22:50.005300700 +0300
@@ -138,7 +138,7 @@
 except NameError:
     _SSLv2_IF_EXISTS = None
 
-if sys.platform == "win32":
+if sys.platform == "win32" and sys.version.find("GCC") == -1:
     from _ssl import enum_certificates, enum_crls
 
 from socket import socket, AF_INET, SOCK_STREAM, create_connection
@@ -421,7 +421,7 @@
     def load_default_certs(self, purpose=Purpose.SERVER_AUTH):
         if not isinstance(purpose, _ASN1Object):
             raise TypeError(purpose)
-        if sys.platform == "win32":
+        if sys.platform == "win32" and sys.version.find("GCC") == -1:
             for storename in self._windows_cert_stores:
                 self._load_windows_store_certs(storename, purpose)
         self.set_default_verify_paths()
