diff -urN Python-3.4.1.orig/Lib/ssl.py Python-3.4.1/Lib/ssl.py
--- Python-3.4.1.orig/Lib/ssl.py	2014-06-19 17:43:06.631281400 +0100
+++ Python-3.4.1/Lib/ssl.py	2014-06-19 17:50:29.817281400 +0100
@@ -144,7 +144,7 @@
     _PROTOCOL_NAMES[PROTOCOL_TLSv1_1] = "TLSv1.1"
     _PROTOCOL_NAMES[PROTOCOL_TLSv1_2] = "TLSv1.2"
 
-if sys.platform == "win32":
+if sys.platform == "win32" and sys.version.find("GCC") == -1:
     from _ssl import enum_certificates, enum_crls
 
 from socket import socket, AF_INET, SOCK_STREAM, create_connection
@@ -387,7 +387,7 @@
     def load_default_certs(self, purpose=Purpose.SERVER_AUTH):
         if not isinstance(purpose, _ASN1Object):
             raise TypeError(purpose)
-        if sys.platform == "win32":
+        if sys.platform == "win32" and sys.version.find("GCC") == -1:
             for storename in self._windows_cert_stores:
                 self._load_windows_store_certs(storename, purpose)
         else:
