--- a/Lib/distutils/cygwinccompiler.py.orig	2014-08-05 23:01:05.598200000 +0400
+++ b/Lib/distutils/cygwinccompiler.py	2014-08-05 23:02:24.355000000 +0400
@@ -167,8 +167,12 @@
                 raise CompileError(msg)
         else: # for other files use the C-compiler
             try:
-                self.spawn(self.compiler_so + cc_args + [src, '-o', obj] +
-                           extra_postargs)
+                if "MSYSTEM" in os.environ:
+                    self.spawn(['sh', '-c', ' '.join(self.compiler_so + cc_args + [src, '-o', obj] +
+                               extra_postargs).replace("\\", "\\\\")])
+                else:
+                    self.spawn(self.compiler_so + cc_args + [src, '-o', obj] +
+                               extra_postargs)
             except DistutilsExecError as msg:
                 raise CompileError(msg)
 
