--- a/Lib/distutils/cygwinccompiler.py.orig	2014-05-19 10:19:38.000000000 +0500
+++ b/Lib/distutils/cygwinccompiler.py	2014-09-23 11:09:54.211200000 +0500
@@ -299,9 +299,9 @@
             raise CCompilerError(
                 'Cygwin gcc cannot be used with --compiler=mingw32')
 
-        self.set_executables(compiler='gcc -O -Wall',
-                             compiler_so='gcc -mdll -O -Wall',
-                             compiler_cxx='g++ -O -Wall',
+        self.set_executables(compiler='gcc -O2 -Wall',
+                             compiler_so='gcc -mdll -O2 -Wall',
+                             compiler_cxx='g++ -O2 -Wall',
                              linker_exe='gcc',
                              linker_so='%s %s %s'
                                         % (self.linker_dll, shared_option,
