diff -Naur Python-3.6.5-orig/Lib/distutils/cygwinccompiler.py Python-3.6.5/Lib/distutils/cygwinccompiler.py
--- Python-3.6.5-orig/Lib/distutils/cygwinccompiler.py	2018-04-16 09:54:49.754677800 +0300
+++ Python-3.6.5/Lib/distutils/cygwinccompiler.py	2018-04-16 09:54:49.985083200 +0300
@@ -238,7 +238,7 @@
         # (On my machine: 10KB < stripped_file < ??100KB
         #   unstripped_file = stripped_file + XXX KB
         #  ( XXX=254 for a typical python extension))
-        if not debug:
+        if not debug and not hasattr(sys, 'gettotalrefcount'):
             extra_preargs.append("-s")
 
         UnixCCompiler.link(self, target_desc, objects, output_filename,
