--- a/Lib/distutils/cygwinccompiler.py	2014-10-05 22:38:20.354004400 +0100
+++ b/Lib/distutils/cygwinccompiler.py	2014-10-05 22:44:36.855314000 +0100
@@ -238,7 +238,7 @@
         # (On my machine: 10KB < stripped_file < ??100KB
         #   unstripped_file = stripped_file + XXX KB
         #  ( XXX=254 for a typical python extension))
-        if not debug:
+        if not debug and not hasattr(sys, 'gettotalrefcount'):
             extra_preargs.append("-s")
 
         UnixCCompiler.link(self, target_desc, objects, output_filename,
