diff -Naur Python-3.5.0-orig/Python/sysmodule.c Python-3.5.0/Python/sysmodule.c
--- Python-3.5.0-orig/Python/sysmodule.c	2015-09-21 13:42:04.576316800 +0300
+++ Python-3.5.0/Python/sysmodule.c	2015-09-21 13:42:51.105452800 +0300
@@ -26,7 +26,7 @@
 #include <windows.h>
 #endif /* MS_WINDOWS */
 
-#ifdef MS_COREDLL
+#if defined(MS_WINDOWS) && defined(Py_ENABLE_SHARED)
 extern void *PyWin_DLLhModule;
 /* A string loaded from the DLL at startup: */
 extern const char *PyWin_DLLVersionString;
@@ -1803,7 +1803,7 @@
                         PyUnicode_FromString("little"));
 #endif
 
-#ifdef MS_COREDLL
+#if defined(MS_WINDOWS) && defined(Py_ENABLE_SHARED)
     SET_SYS_FROM_STRING("dllhandle",
                         PyLong_FromVoidPtr(PyWin_DLLhModule));
     SET_SYS_FROM_STRING("winver",
