diff -Naur Python-3.5.2-orig/Lib/distutils/command/build_ext.py Python-3.5.2/Lib/distutils/command/build_ext.py
--- Python-3.5.2-orig/Lib/distutils/command/build_ext.py	2016-07-12 14:22:11.418300700 +0300
+++ Python-3.5.2/Lib/distutils/command/build_ext.py	2016-07-12 14:22:56.843300700 +0300
@@ -216,7 +216,7 @@
 
         # for extensions under Cygwin and AtheOS Python's library directory must be
         # appended to library_dirs
-        if sys.platform[:6] == 'cygwin' or sys.platform[:6] == 'atheos':
+        if sys.platform[:6] == 'cygwin' or sys.platform[:6] == 'atheos' or self.plat_name.startswith(('mingw')):
             if sys.executable.startswith(os.path.join(sys.exec_prefix, "bin")):
                 # building third party extensions
                 config_dir_name = os.path.basename(sysconfig.get_config_var('LIBPL'))
