diff -Naur Python-3.5.2-orig/setup.py Python-3.5.2/setup.py
--- Python-3.5.2-orig/setup.py	2016-07-12 14:22:36.178300700 +0300
+++ Python-3.5.2/setup.py	2016-07-12 14:22:58.502800700 +0300
@@ -1658,6 +1658,10 @@
                            '-framework', 'CoreFoundation',
                         ]))
 
+        if host_platform.startswith(('mingw', 'win')):
+            exts.append( Extension('_overlapped', ['overlapped.c'],
+                                   libraries=['ws2_32']))
+
         self.extensions.extend(exts)
 
         # Call the method for detecting whether _tkinter can be compiled
