diff -Naur Python-3.7.0-orig/setup.py Python-3.7.0/setup.py
--- Python-3.7.0-orig/setup.py	2018-06-30 11:17:35.605961900 +0300
+++ Python-3.7.0/setup.py	2018-06-30 11:18:01.268007000 +0300
@@ -1678,6 +1678,10 @@
                            '-framework', 'CoreFoundation',
                         ]))
 
+        if host_platform.startswith(('mingw', 'win')):
+            exts.append( Extension('_overlapped', ['overlapped.c'],
+                                   libraries=['ws2_32']))
+
         self.extensions.extend(exts)
 
         # Call the method for detecting whether _tkinter can be compiled
