diff -Naur Python-3.6.5-orig/Include/pyport.h Python-3.6.5/Include/pyport.h
--- Python-3.6.5-orig/Include/pyport.h	2018-04-16 09:54:18.658792000 +0300
+++ Python-3.6.5/Include/pyport.h	2018-04-16 09:54:56.032557500 +0300
@@ -759,7 +759,7 @@
 #define PY_LITTLE_ENDIAN 1
 #endif
 
-#ifdef Py_BUILD_CORE
+#if defined Py_BUILD_CORE || defined Py_BUILD_CORE_MODULE
 /*
  * Macros to protect CRT calls against instant termination when passed an
  * invalid parameter (issue23524).
