diff -Naur Python-3.5.2-orig/Include/pyport.h Python-3.5.2/Include/pyport.h
--- Python-3.5.2-orig/Include/pyport.h	2016-07-12 14:22:15.265300700 +0300
+++ Python-3.5.2/Include/pyport.h	2016-07-12 14:23:00.197800700 +0300
@@ -896,7 +896,7 @@
 #define PY_LITTLE_ENDIAN 1
 #endif
 
-#ifdef Py_BUILD_CORE
+#if defined Py_BUILD_CORE || defined Py_BUILD_CORE_MODULE
 /*
  * Macros to protect CRT calls against instant termination when passed an
  * invalid parameter (issue23524).
