diff -Naur Python-3.7.0-orig/Include/pyport.h Python-3.7.0/Include/pyport.h
--- Python-3.7.0-orig/Include/pyport.h	2018-07-12 10:22:00.418822200 +0300
+++ Python-3.7.0/Include/pyport.h	2018-07-12 10:22:37.468887300 +0300
@@ -772,7 +772,7 @@
 #define PY_LITTLE_ENDIAN 1
 #endif
 
-#if defined(Py_BUILD_CORE) || defined(Py_BUILD_CORE_BUILTIN)
+#if defined(Py_BUILD_CORE) || defined(Py_BUILD_CORE_BUILTIN) || defined(Py_BUILD_CORE_MODULE)
 /*
  * Macros to protect CRT calls against instant termination when passed an
  * invalid parameter (issue23524).
