diff -Naur Python-3.6.5-orig/Python/getcompiler.c Python-3.6.5/Python/getcompiler.c
--- Python-3.6.5-orig/Python/getcompiler.c	2018-03-28 12:19:31.000000000 +0300
+++ Python-3.6.5/Python/getcompiler.c	2018-04-16 09:55:00.617392300 +0300
@@ -6,7 +6,22 @@
 #ifndef COMPILER
 
 #ifdef __GNUC__
-#define COMPILER "\n[GCC " __VERSION__ "]"
+/* To not break compatibility with things that determine
+   CPU arch by calling get_build_version in msvccompiler.py
+   (such as NumPy) add "32 bit" or "64 bit (AMD64)" on Windows
+   and also use a space as a separator rather than a newline. */
+#if defined(_WIN32)
+#define COMP_SEP " "
+#if defined(__x86_64__)
+#define ARCH_SUFFIX " 64 bit (AMD64)"
+#else
+#define ARCH_SUFFIX " 32 bit"
+#endif
+#else
+#define COMP_SEP "\n"
+#define ARCH_SUFFIX ""
+#endif
+#define COMPILER COMP_SEP "[GCC " __VERSION__ ARCH_SUFFIX "]"
 #endif
 
 #endif /* !COMPILER */
