diff -Naur Python-3.7.0-orig/configure.ac Python-3.7.0/configure.ac
--- Python-3.7.0-orig/configure.ac	2018-06-30 11:17:44.107976800 +0300
+++ Python-3.7.0/configure.ac	2018-06-30 11:18:09.692021800 +0300
@@ -5116,9 +5116,16 @@
         [Define if you have readline 4.0]), ,$READLINE_LIBS)
 
 # also in 4.0, but not in editline
+case $host in
+  *-*-mingw*)
+  dnl Windows don't have resize terminal signal.
+  ;;
+  *)
 AC_CHECK_LIB(readline, rl_resize_terminal,
 	AC_DEFINE(HAVE_RL_RESIZE_TERMINAL, 1,
         [Define if you have readline 4.0]), ,$READLINE_LIBS)
+  ;;
+esac
 
 # check for readline 4.2
 AC_CHECK_LIB(readline, rl_completion_matches,
