diff -Naur Python-3.5.0-orig/Modules/posixmodule.c Python-3.5.0/Modules/posixmodule.c
--- Python-3.5.0-orig/Modules/posixmodule.c	2015-09-21 13:42:06.149864000 +0300
+++ Python-3.5.0/Modules/posixmodule.c	2015-09-21 13:42:55.398279200 +0300
@@ -1370,10 +1370,10 @@
 
     if(!SetCurrentDirectoryW(path))
         return FALSE;
-    result = GetCurrentDirectoryW(Py_ARRAY_LENGTH(new_path), new_path);
+    result = GetCurrentDirectoryW(Py_ARRAY_LENGTH(_new_path), new_path);
     if (!result)
         return FALSE;
-    if (result > Py_ARRAY_LENGTH(new_path)) {
+    if (result > Py_ARRAY_LENGTH(_new_path)) {
         new_path = PyMem_RawMalloc(result * sizeof(wchar_t));
         if (!new_path) {
             SetLastError(ERROR_OUTOFMEMORY);
