diff -Naur Python-3.5.0-orig/Makefile.pre.in Python-3.5.0/Makefile.pre.in
--- Python-3.5.0-orig/Makefile.pre.in	2015-09-21 13:42:31.879731200 +0300
+++ Python-3.5.0/Makefile.pre.in	2015-09-21 13:42:56.820567200 +0300
@@ -1219,6 +1219,7 @@
 		test/test_importlib/extension test/test_importlib/frozen \
 		test/test_importlib/import_ test/test_importlib/source \
 		turtledemo \
+		msilib \
 		multiprocessing multiprocessing/dummy \
 		unittest unittest/test unittest/test/testmock \
 		venv venv/scripts venv/scripts/posix \
