diff -Naur Python-3.5.2-orig/Makefile.pre.in Python-3.5.2/Makefile.pre.in
--- Python-3.5.2-orig/Makefile.pre.in	2016-07-12 14:22:32.838800700 +0300
+++ Python-3.5.2/Makefile.pre.in	2016-07-12 14:23:07.018300700 +0300
@@ -1260,6 +1260,7 @@
 		test/test_importlib/extension test/test_importlib/frozen \
 		test/test_importlib/import_ test/test_importlib/source \
 		turtledemo \
+		msilib \
 		multiprocessing multiprocessing/dummy \
 		unittest unittest/test unittest/test/testmock \
 		venv venv/scripts venv/scripts/posix \
