--- Python-3.6.1/Makefile.pre.in.orig	2017-06-13 18:54:47.196308100 +0200
+++ Python-3.6.1/Makefile.pre.in	2017-06-13 18:55:21.584275000 +0200
@@ -1264,6 +1264,7 @@
 		test/test_importlib/import_ test/test_importlib/source \
 		test/test_tools test/test_warnings test/test_warnings/data \
 		turtledemo \
+		msilib \
 		multiprocessing multiprocessing/dummy \
 		unittest unittest/test unittest/test/testmock \
 		venv venv/scripts venv/scripts/common venv/scripts/posix \
