--- Python-3.7.3/configure.ac.orig	2019-05-28 11:29:47.194040000 +0300
+++ Python-3.7.3/configure.ac	2019-05-28 11:32:34.312240800 +0300
@@ -5383,9 +5383,15 @@
 
 # first curses header check
 ac_save_cppflags="$CPPFLAGS"
-if test "$cross_compiling" = no; then
-  CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
-fi
+case $host_os in
+    mingw*)
+    ;;
+    *)
+      if test "$cross_compiling" = no; then
+        CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+      fi
+    ;;
+esac
 
 AC_CHECK_HEADERS(curses.h ncurses.h)
 
