--- Python-3.6.1/setup.py.orig	2017-06-13 16:41:16.611128800 +0200
+++ Python-3.6.1/setup.py	2017-06-13 16:42:21.729853400 +0200
@@ -942,7 +942,7 @@
         blake2_deps.append('hashlib.h')
 
         blake2_macros = []
-        if not cross_compiling and os.uname().machine == "x86_64":
+        if not cross_compiling and not host_platform.startswith(('mingw', 'win')):
             # Every x86_64 machine has at least SSE2.
             blake2_macros.append(('BLAKE2_USE_SSE', '1'))
 
