diff -Naur Python-3.7.0-orig/Modules/posixmodule.c Python-3.7.0/Modules/posixmodule.c
--- Python-3.7.0-orig/Modules/posixmodule.c	2018-06-30 11:18:07.352017600 +0300
+++ Python-3.7.0/Modules/posixmodule.c	2018-06-30 11:18:18.194036700 +0300
@@ -180,6 +180,7 @@
 #define HAVE_GETPPID    1
 #define HAVE_GETLOGIN   1
 #define HAVE_SPAWNV	1
+#define HAVE_WSPAWNV    1
 /*#define HAVE_EXECV	1 - detected by configure*/
 #define HAVE_PIPE	1
 #define HAVE_POPEN	1
