diff -Naur Python-3.6.5-orig/Modules/Setup.dist Python-3.6.5/Modules/Setup.dist
--- Python-3.6.5-orig/Modules/Setup.dist	2018-04-16 09:54:05.842862600 +0300
+++ Python-3.6.5/Modules/Setup.dist	2018-04-16 09:55:11.276677000 +0300
@@ -114,7 +114,7 @@
 _collections _collectionsmodule.c # Container types
 itertools itertoolsmodule.c    # Functions creating iterators for efficient looping
 atexit atexitmodule.c      # Register functions to be run at interpreter-shutdown
-_signal signalmodule.c
+_signal signalmodule.c -lws2_32
 _stat _stat.c			# stat.h interface
 time timemodule.c	# -lm # time operations and variables
 
