diff -Naur Python-3.7.0-orig/Modules/Setup.dist Python-3.7.0/Modules/Setup.dist
--- Python-3.7.0-orig/Modules/Setup.dist	2018-07-12 10:20:58.876714100 +0300
+++ Python-3.7.0/Modules/Setup.dist	2018-07-12 10:22:44.098898900 +0300
@@ -114,7 +114,7 @@
 _abc _abc.c				# Abstract base classes
 itertools itertoolsmodule.c		# Functions creating iterators for efficient looping
 atexit atexitmodule.c			# Register functions to be run at interpreter-shutdown
-_signal -DPy_BUILD_CORE signalmodule.c
+_signal -DPy_BUILD_CORE signalmodule.c -lws2_32
 _stat _stat.c				# stat.h interface
 time -DPy_BUILD_CORE timemodule.c	# -lm # time operations and variables
 _thread -DPy_BUILD_CORE _threadmodule.c	# low-level threading interface
