diff -Naur Python-3.8.0-orig/Modules/Setup Python-3.8.0/Modules/Setup
--- Python-3.8.0-orig/Modules/Setup	2019-10-22 10:04:31.147113000 +0300
+++ Python-3.8.0/Modules/Setup	2019-10-22 10:04:32.223514900 +0300
@@ -120,7 +120,7 @@
 _locale -DPy_BUILD_CORE_BUILTIN _localemodule.c  # -lintl
 
 # Standard I/O baseline
-_io -DPy_BUILD_CORE_BUILTIN -I$(srcdir)/Include/internal -I$(srcdir)/Modules/_io _io/_iomodule.c _io/iobase.c _io/fileio.c _io/bytesio.c _io/bufferedio.c _io/textio.c _io/stringio.c
+_io -DPy_BUILD_CORE_BUILTIN -I$(srcdir)/Include/internal -I$(srcdir)/Modules/_io _io/_iomodule.c _io/iobase.c _io/fileio.c _io/bytesio.c _io/bufferedio.c _io/textio.c _io/stringio.c _io/winconsoleio.c
 
 # faulthandler module
 faulthandler faulthandler.c
