diff -Naur Python-3.7.0-orig/Modules/Setup.dist Python-3.7.0/Modules/Setup.dist
--- Python-3.7.0-orig/Modules/Setup.dist	2018-07-12 10:22:45.440501300 +0300
+++ Python-3.7.0/Modules/Setup.dist	2018-07-12 10:22:45.736901800 +0300
@@ -123,7 +123,7 @@
 _locale _localemodule.c  # -lintl
 
 # Standard I/O baseline
-_io -DPy_BUILD_CORE -I$(srcdir)/Modules/_io _io/_iomodule.c _io/iobase.c _io/fileio.c _io/bytesio.c _io/bufferedio.c _io/textio.c _io/stringio.c
+_io -DPy_BUILD_CORE -I$(srcdir)/Modules/_io _io/_iomodule.c _io/iobase.c _io/fileio.c _io/bytesio.c _io/bufferedio.c _io/textio.c _io/stringio.c _io/winconsoleio.c
 
 # The zipimport module is always imported at startup. Having it as a
 # builtin module avoids some bootstrapping problems and reduces overhead.
