diff -Naur Python-3.6.5-orig/Modules/_io/_iomodule.c Python-3.6.5/Modules/_io/_iomodule.c
--- Python-3.6.5-orig/Modules/_io/_iomodule.c	2018-03-28 12:19:31.000000000 +0300
+++ Python-3.6.5/Modules/_io/_iomodule.c	2018-04-16 09:55:14.166902200 +0300
@@ -20,7 +20,7 @@
 #include <sys/stat.h>
 #endif /* HAVE_SYS_STAT_H */
 
-#ifdef MS_WINDOWS
+#if defined(MS_WINDOWS) && !defined(__MINGW32__)
 #include <consoleapi.h>
 #endif
 
