--- Python-3.6.2rc1/Modules/_multiprocessing/multiprocessing.c.orig	2017-06-22 13:35:38.307651400 +0200
+++ Python-3.6.2rc1/Modules/_multiprocessing/multiprocessing.c	2017-06-22 13:40:12.238319300 +0200
@@ -128,7 +128,7 @@
     {"recv", multiprocessing_recv, METH_VARARGS, ""},
     {"send", multiprocessing_send, METH_VARARGS, ""},
 #endif
-#if !defined(POSIX_SEMAPHORES_NOT_ENABLED) && !defined(__ANDROID__)
+#if defined(MS_WINDOWS) || (!defined(POSIX_SEMAPHORES_NOT_ENABLED) && !defined(__ANDROID__))
     {"sem_unlink", _PyMp_sem_unlink, METH_VARARGS, ""},
 #endif
     {NULL}
