diff -Naur Python-3.7.0-orig/setup.py Python-3.7.0/setup.py
--- Python-3.7.0-orig/setup.py	2018-07-12 10:22:37.188086800 +0300
+++ Python-3.7.0/setup.py	2018-07-12 10:22:50.869310800 +0300
@@ -1306,7 +1306,7 @@
             if dbm_args:
                 dbm_order = [arg.split('=')[-1] for arg in dbm_args][-1].split(":")
             else:
-                dbm_order = "ndbm:gdbm:bdb".split(":")
+                dbm_order = []
             dbmext = None
             for cand in dbm_order:
                 if cand == "ndbm":
