diff -Naur Python-3.6.5-orig/setup.py Python-3.6.5/setup.py
--- Python-3.6.5-orig/setup.py	2018-04-16 09:55:20.108348800 +0300
+++ Python-3.6.5/setup.py	2018-04-16 09:55:20.311351800 +0300
@@ -1345,7 +1345,7 @@
             if dbm_args:
                 dbm_order = [arg.split('=')[-1] for arg in dbm_args][-1].split(":")
             else:
-                dbm_order = "ndbm:gdbm:bdb".split(":")
+                dbm_order = []
             dbmext = None
             for cand in dbm_order:
                 if cand == "ndbm":
