--- a/setup.py
+++ b/setup.py
@@ -2090,7 +2090,7 @@ class PyBuildExt(build_ext):
         undef_macros = []
         if '--with-system-libmpdec' in sysconfig.get_config_var("CONFIG_ARGS"):
             include_dirs = []
-            libraries = [':libmpdec.so.2']
+            libraries = [':libmpdec.dll.a']
             sources = ['_decimal/_decimal.c']
             depends = ['_decimal/docstrings.h']
         else:
