--- Python-3.7.0/Modules/makesetup.orig	2018-07-01 18:35:00.747322100 +0300
+++ Python-3.7.0/Modules/makesetup	2018-07-01 18:35:16.737350200 +0300
@@ -233,7 +233,7 @@
 			case $doconfig in
 			no)	cc="$cc \$(CCSHARED) \$(PY_CFLAGS) \$(PY_CPPFLAGS)";;
 			*)
-				cc="$cc \$(PY_STDMODULE_CFLAGS)";;
+				cc="$cc \$(PY_CORE_CFLAGS)";;
 			esac
 			rule="$obj: $src; $cc $cpps -c $src -o $obj"
 			echo "$rule" >>$rulesf
