diff -Naur Python-3.7.0-orig/Modules/makesetup Python-3.7.0/Modules/makesetup
--- Python-3.7.0-orig/Modules/makesetup	2018-06-27 06:07:35.000000000 +0300
+++ Python-3.7.0/Modules/makesetup	2018-07-12 10:22:57.639722700 +0300
@@ -233,7 +233,7 @@
 			case $doconfig in
 			no)	cc="$cc \$(CCSHARED) \$(PY_CFLAGS) \$(PY_CPPFLAGS)";;
 			*)
-				cc="$cc \$(PY_STDMODULE_CFLAGS)";;
+				cc="$cc \$(PY_CORE_CFLAGS)";;
 			esac
 			rule="$obj: $src; $cc $cpps -c $src -o $obj"
 			echo "$rule" >>$rulesf
