--- Python-3.7.0/setup.py.orig	2018-07-06 12:59:49.823002800 +0200
+++ Python-3.7.0/setup.py	2018-07-06 13:28:16.321187300 +0200
@@ -807,7 +807,10 @@
                                libraries=['m']) )
 
         # readline
-        do_readline = self.compiler.find_library_file(lib_dirs, 'readline')
+        if not host_platform.startswith(('mingw', 'win')):
+            do_readline = self.compiler.find_library_file(lib_dirs, 'readline')
+        else:
+            do_readline = False
         readline_termcap_library = ""
         curses_library = ""
         # Cannot use os.popen here in py3k.
