diff -Naur Python-3.8.0-orig/setup.py Python-3.8.0/setup.py
--- Python-3.8.0-orig/setup.py	2019-10-22 10:04:55.763956200 +0300
+++ Python-3.8.0/setup.py	2019-10-22 10:04:59.804363300 +0300
@@ -911,7 +911,10 @@
 
     def detect_readline_curses(self):
         # readline
-        do_readline = self.compiler.find_library_file(self.lib_dirs, 'readline')
+        if not MS_WINDOWS:
+            do_readline = self.compiler.find_library_file(self.lib_dirs, 'readline')
+        else:
+            do_readline = False
         readline_termcap_library = ""
         curses_library = ""
         # Cannot use os.popen here in py3k.
