--- Python-3.7.0/Include/fileobject.h.orig	2018-07-06 16:32:54.532737200 +0200
+++ Python-3.7.0/Include/fileobject.h	2018-07-06 16:31:01.342311200 +0200
@@ -42,7 +42,7 @@
 #endif /* Py_LIMITED_API */
 
 /* A routine to check if a file descriptor can be select()-ed. */
-#ifdef _MSC_VER
+#ifdef MS_WINDOWS
     /* On Windows, any socket fd can be select()-ed, no matter how high */
     #define _PyIsSelectable_fd(FD) (1)
 #else
