--- Python-3.7.0/setup.py.orig	2018-07-05 02:42:31.000000000 -0500
+++ Python-3.7.0/setup.py	2018-07-05 02:42:56.000000000 -0500
@@ -874,6 +874,8 @@
                 readline_extra_link_args = ()
 
             readline_libs = ['readline']
+            if host_platform.startswith(('mingw', 'win')):
+                readline_libs.append('ws2_32')
             if readline_termcap_library:
                 pass # Issue 7384: Already linked against curses or tinfo.
             elif curses_library:
--- ./Python-3.7.0/Modules/readline.c.orig	2018-07-05 02:43:06.000000000 -0500
+++ ./Python-3.7.0/Modules/readline.c	2018-07-05 02:43:46.000000000 -0500
@@ -11,6 +11,9 @@
 #include <signal.h>
 #include <errno.h>
 #include <sys/time.h>
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 
 #if defined(HAVE_SETLOCALE)
 /* GNU readline() mistakenly sets the LC_CTYPE locale.
