--- Python-3.7.3/Modules/getpath.c.orig	2019-04-16 08:23:18.111566000 +0300
+++ Python-3.7.3/Modules/getpath.c	2019-04-16 08:23:20.781569700 +0300
@@ -341,7 +341,7 @@
     /* Check for already have an executable suffix */
     size_t n = wcslen(progpath);
     size_t s = wcslen(EXE_SUFFIX);
-    if (wcsncasecmp(EXE_SUFFIX, progpath+n-s, s) != 0) {
+    if (_wcsnicmp(EXE_SUFFIX, progpath+n-s, s) != 0) {
         if (n + s > MAXPATHLEN) {
             Py_FatalError("progpath overflow in getpath.c's add_exe_suffix()");
         }
