--- Python-3.7.4/Makefile.pre.in.orig	2019-07-22 09:15:44.386184900 +0200
+++ Python-3.7.4/Makefile.pre.in	2019-07-22 08:57:45.643259600 +0200
@@ -713,7 +713,7 @@
 $(DLLLIBRARY) libpython$(LDVERSION).dll.a: $(LIBRARY_OBJS) python_nt.o
 	if test -n "$(DLLLIBRARY)"; then \
 		$(LDSHARED) -Wl,--out-implib=$@ -o $(DLLLIBRARY) $^ \
-			$(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST) python_nt.o; \
+			$(LIBS) $(LDFLAGS_NODIST) $(MODLIBS) $(SYSLIBS) $(LDLAST) python_nt.o; \
 	else true; \
 	fi
 
