--- Python-3.8.0/configure.ac.orig	2019-10-24 13:55:46.684382200 +0300
+++ Python-3.8.0/configure.ac	2019-10-24 13:55:55.334394400 +0300
@@ -4990,7 +4990,7 @@
 
 # On Android and Cygwin the shared libraries must be linked with libpython.
 AC_SUBST(LIBPYTHON)
-if test -n "$ANDROID_API_LEVEL" -o "$MACHDEP" = "cygwin"; then
+if test -n "$ANDROID_API_LEVEL" -o "$MACHDEP" = "cygwin" -o "$MACHDEP" = "win"; then
   LIBPYTHON="-lpython${VERSION}${ABIFLAGS}"
 else
   LIBPYTHON=''
--- Python-3.8.0/Misc/python.pc.in.orig	2019-10-24 13:55:46.684382200 +0300
+++ Python-3.8.0/Misc/python.pc.in	2019-10-24 13:55:55.334394400 +0300
@@ -9,5 +9,5 @@
 Requires:
 Version: @VERSION@
 Libs.private: @LIBS@
-Libs:
+Libs: -L${libdir} -lpython@VERSION@@ABIFLAGS@
 Cflags: -I${includedir}/python@VERSION@@ABIFLAGS@
