--- Python-3.7.4/Lib/shutil.py.orig	2019-08-13 11:08:32.602736300 +0300
+++ Python-3.7.4/Lib/shutil.py	2019-08-13 11:08:58.952773200 +0300
@@ -10,6 +10,7 @@
 import fnmatch
 import collections
 import errno
+import time
 
 try:
     import zlib
@@ -399,6 +400,7 @@
             except OSError:
                 onerror(os.unlink, fullname, sys.exc_info())
     try:
+        time.sleep(2)
         os.rmdir(path)
     except OSError:
         onerror(os.rmdir, path, sys.exc_info())
