diff -Naur openmp-14.0.0.src/runtime/cmake/LibompDefinitions.cmake src/openmp-14.0.0.src/runtime/cmake/LibompDefinitions.cmake
--- openmp-14.0.0.src/runtime/cmake/LibompDefinitions.cmake	2022-04-21 01:17:04.574642900 -0400
+++ src/openmp-14.0.0.src/runtime/cmake/LibompDefinitions.cmake	2022-04-21 01:17:49.954060500 -0400
@@ -26,5 +26,7 @@
   endif()
 
   # CMake doesn't include CPPFLAGS from environment, but we will.
-  set(${cppflags} ${cppflags_local} ${LIBOMP_CPPFLAGS} $ENV{CPPFLAGS} PARENT_SCOPE)
+  string(REPLACE " " ";" ENV_CPPFLAGS $ENV{CPPFLAGS})
+  set(${cppflags} ${cppflags_local} ${LIBOMP_CPPFLAGS} ${ENV_CPPFLAGS} PARENT_SCOPE)
+  unset(ENV_CPPFLAGS)
 endfunction()
diff -Naur openmp-14.0.0.src/runtime/CMakeLists.txt src/openmp-14.0.0.src/runtime/CMakeLists.txt
--- openmp-14.0.0.src/runtime/CMakeLists.txt	2022-03-14 05:44:55.000000000 -0400
+++ src/openmp-14.0.0.src/runtime/CMakeLists.txt	2022-04-21 01:11:05.925977700 -0400
@@ -291,10 +291,6 @@
 set(LIBOMP_ENABLE_SHARED TRUE CACHE BOOL
   "Shared library instead of static library?")
 
-if(WIN32 AND NOT LIBOMP_ENABLE_SHARED)
-  libomp_error_say("Static libraries requested but not available on Windows")
-endif()
-
 if(LIBOMP_USE_ITT_NOTIFY AND NOT LIBOMP_ENABLE_SHARED)
   message(STATUS "ITT Notify not supported for static libraries - forcing ITT Notify off")
   set(LIBOMP_USE_ITT_NOTIFY FALSE)
diff -Naur openmp-14.0.0.src/runtime/src/CMakeLists.txt src/openmp-14.0.0.src/runtime/src/CMakeLists.txt
--- openmp-14.0.0.src/runtime/src/CMakeLists.txt	2022-03-14 05:44:55.000000000 -0400
+++ src/openmp-14.0.0.src/runtime/src/CMakeLists.txt	2022-04-21 01:11:05.931526900 -0400
@@ -251,14 +251,14 @@
       set(LIBOMP_IMP_LIB_FILE ${LIBOMP_DEFAULT_LIB_NAME}${CMAKE_IMPORT_LIBRARY_SUFFIX})
       set(LIBOMP_GENERATED_IMP_LIB_FILENAME ${LIBOMP_DEFAULT_LIB_NAME}${LIBOMP_LIBRARY_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX})
     endif()
-  else()
+  elseif(LIBOMP_ENABLE_SHARED)
     set(LIBOMP_IMP_LIB_FILE ${LIBOMP_LIB_NAME}${CMAKE_IMPORT_LIBRARY_SUFFIX})
     set(LIBOMP_GENERATED_IMP_LIB_FILENAME ${LIBOMP_LIB_FILE}${CMAKE_STATIC_LIBRARY_SUFFIX})
+    set_target_properties(omp PROPERTIES ARCHIVE_OUTPUT_NAME ${LIBOMP_GENERATED_IMP_LIB_FILENAME})
   endif()
   set_target_properties(omp PROPERTIES
     VERSION ${LIBOMP_VERSION_MAJOR}.${LIBOMP_VERSION_MINOR} # uses /version flag
     IMPORT_PREFIX "" IMPORT_SUFFIX "" # control generated import library name when building omp
-    ARCHIVE_OUTPUT_NAME ${LIBOMP_GENERATED_IMP_LIB_FILENAME}
   )
 
   if(MSVC)
