--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -335,10 +335,6 @@
   set(LIBOMP_ENABLE_SHARED FALSE)
 endif()
 
-if(WIN32 AND NOT LIBOMP_ENABLE_SHARED)
-  libomp_error_say("Static libraries requested but not available on Windows")
-endif()
-
 if(LIBOMP_USE_ITT_NOTIFY AND NOT LIBOMP_ENABLE_SHARED)
   message(STATUS "ITT Notify not supported for static libraries - forcing ITT Notify off")
   set(LIBOMP_USE_ITT_NOTIFY FALSE)
--- a/runtime/src/CMakeLists.txt
+++ b/runtime/src/CMakeLists.txt
@@ -284,14 +284,14 @@
       set(LIBOMP_IMP_LIB_FILE ${LIBOMP_DEFAULT_LIB_NAME}${CMAKE_IMPORT_LIBRARY_SUFFIX})
       set(LIBOMP_GENERATED_IMP_LIB_FILENAME ${LIBOMP_DEFAULT_LIB_NAME}${LIBOMP_LIBRARY_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX})
     endif()
-  else()
+  elseif(LIBOMP_ENABLE_SHARED)
     set(LIBOMP_IMP_LIB_FILE ${LIBOMP_LIB_NAME}${CMAKE_IMPORT_LIBRARY_SUFFIX})
     set(LIBOMP_GENERATED_IMP_LIB_FILENAME ${LIBOMP_LIB_FILE}${CMAKE_STATIC_LIBRARY_SUFFIX})
+    set_target_properties(omp PROPERTIES ARCHIVE_OUTPUT_NAME ${LIBOMP_GENERATED_IMP_LIB_FILENAME})
   endif()
   set_target_properties(omp PROPERTIES
     VERSION ${LIBOMP_VERSION_MAJOR}.${LIBOMP_VERSION_MINOR} # uses /version flag
     IMPORT_PREFIX "" IMPORT_SUFFIX "" # control generated import library name when building omp
-    ARCHIVE_OUTPUT_NAME ${LIBOMP_GENERATED_IMP_LIB_FILENAME}
   )
 
   set(LIBOMP_IMP_LIB_TARGET omp)
