--- a/docs/manpages/meson.build
+++ b/docs/manpages/meson.build
@@ -49,7 +49,7 @@
     input: keymap_src_file,
     output: 'virkeycode-@0@.rst'.format(name),
     command: [
-      meson_python_prog, python3_prog, keymap_gen_prog, 'code-docs',
+      meson_python_prog, python3_prog, keymap_gen_prog.path(), 'code-docs',
       '--lang', 'rst',
       '--title', 'virkeycode-@0@'.format(name),
       '--subtitle', 'Key code values for @0@'.format(name),
@@ -70,7 +70,7 @@
     input: keymap_src_file,
     output: 'virkeyname-@0@.rst'.format(name),
     command: [
-      meson_python_prog, python3_prog, keymap_gen_prog, 'name-docs',
+      meson_python_prog, python3_prog, keymap_gen_prog.path(), 'name-docs',
       '--lang', 'rst',
       '--title', 'virkeyname-@0@'.format(name),
       '--subtitle', 'Key name values for @0@'.format(name),
--- a/docs/meson.build
+++ b/docs/meson.build
@@ -124,7 +124,7 @@
   input: access_perm_h,
   output: 'aclperms.htmlinc',
   command: [
-    meson_python_prog, python3_prog, genaclperms_prog, '@INPUT@',
+    meson_python_prog, python3_prog, genaclperms_prog.path(), '@INPUT@',
   ],
   capture: true,
 )
@@ -146,7 +146,7 @@
     'libvirt-admin-api.xml',
   ],
   command: [
-    meson_python_prog, python3_prog, apibuild_prog,
+    meson_python_prog, python3_prog, apibuild_prog.path(),
     meson.current_source_dir(),
     meson.current_build_dir(),
   ],
@@ -264,7 +264,7 @@
   command: [
     meson_python_prog,
     python3_prog,
-    hvsupport_prog,
+    hvsupport_prog.path(),
     meson.project_source_root(),
     meson.project_build_root(),
   ],
--- a/src/admin/meson.build
+++ b/src/admin/meson.build
@@ -68,7 +68,7 @@
   ],
   output: 'libvirt_admin.syms',
   command: [
-    meson_python_prog, python3_prog, meson_gen_sym_prog,
+    meson_python_prog, python3_prog, meson_gen_sym_prog.path(),
     '@OUTPUT@', 'LIBVIRT_ADMIN_PRIVATE_' + meson.project_version(), '@INPUT@',
   ],
 )
@@ -79,7 +79,7 @@
     input: libvirt_admin_syms,
     output: 'libvirt_admin.def',
     command: [
-      meson_python_prog, python3_prog, meson_gen_def_prog,
+      meson_python_prog, python3_prog, meson_gen_def_prog.path(),
       '@INPUT@', '@OUTPUT@',
     ],
   )
--- a/src/meson.build
+++ b/src/meson.build
@@ -75,7 +75,7 @@
     input: infile,
     output: out_stp,
     command: [
-      meson_python_prog, python3_prog, dtrace2systemtap_prog,
+      meson_python_prog, python3_prog, dtrace2systemtap_prog.path(),
       bindir, sbindir, libdir, '@INPUT@'
     ],
     capture: true,
@@ -350,7 +350,7 @@
   input: [ public_sym_file ] + used_sym_files + generated_sym_files,
   output: 'libvirt.syms',
   command: [
-    meson_python_prog, python3_prog, meson_gen_sym_prog,
+    meson_python_prog, python3_prog, meson_gen_sym_prog.path(),
     '@OUTPUT@', 'LIBVIRT_PRIVATE_' + meson.project_version(), '@INPUT@',
   ],
 )
@@ -361,7 +361,7 @@
     input: libvirt_syms,
     output: 'libvirt.def',
     command: [
-      meson_python_prog, python3_prog, meson_gen_def_prog,
+      meson_python_prog, python3_prog, meson_gen_def_prog.path(),
       '@INPUT@', '@OUTPUT@',
     ],
   )
@@ -417,7 +417,7 @@
     input: libvirt_qemu_syms,
     output: 'libvirt_qemu.def',
     command: [
-      meson_python_prog, python3_prog, meson_gen_def_prog,
+      meson_python_prog, python3_prog, meson_gen_def_prog.path(),
       '@INPUT@', '@OUTPUT@',
     ],
   )
@@ -476,7 +476,7 @@
     input: libvirt_lxc_syms,
     output: 'libvirt_lxc.def',
     command: [
-      meson_python_prog, python3_prog, meson_gen_def_prog,
+      meson_python_prog, python3_prog, meson_gen_def_prog.path(),
       '@INPUT@', '@OUTPUT@',
     ],
   )
@@ -692,7 +692,7 @@
     input: [ data['conf'], data['aug'] ],
     output: data['name'],
     command: [
-      meson_python_prog, python3_prog, augeas_gentest_prog,
+      meson_python_prog, python3_prog, augeas_gentest_prog.path(),
       '@INPUT@',
     ],
     capture: true,
@@ -771,7 +771,7 @@
     input: [ conf_out, test_aug_tmp ],
     output: test_aug_out,
     command: [
-      meson_python_prog, python3_prog, augeas_gentest_prog,
+      meson_python_prog, python3_prog, augeas_gentest_prog.path(),
       '@INPUT@',
     ],
     capture: true,
@@ -889,7 +889,7 @@
     input: rpc_probe_files,
     output: 'libvirt_functions.stp',
     command: [
-      meson_python_prog, python3_prog, gensystemtap_prog,
+      meson_python_prog, python3_prog, gensystemtap_prog.path(),
       '@INPUT@',
     ],
     capture: true,
--- a/src/util/meson.build
+++ b/src/util/meson.build
@@ -146,7 +146,7 @@
     input: keymap_src_file,
     output: 'virkeycodetable_@0@.h'.format(name),
     command: [
-      meson_python_prog, python3_prog, keymap_gen_prog, 'code-table',
+      meson_python_prog, python3_prog, keymap_gen_prog.path(), 'code-table',
       '--lang', 'stdc',
       '--varname', 'virKeyCodeTable_@0@'.format(name),
       '@INPUT@', name,
@@ -161,7 +161,7 @@
     input: keymap_src_file,
     output: 'virkeynametable_@0@.h'.format(name),
     command: [
-      meson_python_prog, python3_prog, keymap_gen_prog, 'name-table',
+      meson_python_prog, python3_prog, keymap_gen_prog.path(), 'name-table',
       '--lang', 'stdc',
       '--varname', 'virKeyNameTable_@0@'.format(name),
       '@INPUT@', name,
