--- libvirt-7.0.0/src/rpc/genprotocol.pl.orig	2021-01-09 13:04:12.963609000 +0300
+++ libvirt-7.0.0/src/rpc/genprotocol.pl	2021-01-09 13:04:26.081292200 +0300
@@ -57,6 +57,8 @@
     # We only want to fixup the GLibc rpcgen output
     # So just print data unchanged, if non-Linux
     unless ($fixup) {
+        # Fix VPATH builds
+        s,#include ".*/([^/]+)protocol\.h",#include "${1}protocol.h",;
         print TARGET;
         next;
     }
