--- avrdude-6.4/configure.ac
+++ avrdude-6.4/configure.ac
@@ -34,7 +34,7 @@
 AC_CONFIG_HEADERS(ac_cfg.h)
 AC_CONFIG_MACRO_DIR([m4])
 
-LT_INIT()
+LT_INIT([win32-dll])
 
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
--- avrdude-6.4/Makefile.am
+++ avrdude-6.4/Makefile.am
@@ -194,7 +194,7 @@
 	xbee.h \
 	xbee.c
 libavrdude_la_SOURCES = $(libavrdude_a_SOURCES)
-libavrdude_la_LDFLAGS = -version-info 1:0
+libavrdude_la_LDFLAGS = -no-undefined -version-info 1:0
 
 include_HEADERS = libavrdude.h
 
