--- avrdude-6.4/serbb_win32.c
+++ avrdude-6.4/serbb_win32.c
@@ -308,8 +308,8 @@
                         progname, port);
                 return -1;
 	}
-        avrdude_message(MSG_DEBUG, "%s: ser_open(): opened comm port \"%s\", handle 0x%x\n",
-                        progname, port, (int)hComPort);
+        avrdude_message(MSG_DEBUG, "%s: ser_open(): opened comm port \"%s\", handle 0x%p\n",
+                        progname, port, hComPort);
 
         pgm->fd.pfd = (void *)hComPort;
 
@@ -326,8 +326,8 @@
 		pgm->setpin(pgm, PIN_AVR_RESET, 1);
 		CloseHandle (hComPort);
 	}
-        avrdude_message(MSG_DEBUG, "%s: ser_close(): closed comm port handle 0x%x\n",
-                                progname, (int)hComPort);
+        avrdude_message(MSG_DEBUG, "%s: ser_close(): closed comm port handle 0x%p\n",
+                                progname, hComPort);
 
 	hComPort = INVALID_HANDLE_VALUE;
 }
