--- gdb-7.7/gdb/common/format.h.orig	2014-03-11 08:18:16.200800000 +0400
+++ gdb-7.7/gdb/common/format.h	2014-03-11 08:18:59.334800000 +0400
@@ -18,7 +18,11 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 
 #if defined(__MINGW32__) && !defined(PRINTF_HAS_LONG_LONG)
-# define USE_PRINTF_I64 1
+# if !defined(__USE_MINGW_ANSI_STDIO)
+#  define USE_PRINTF_I64 1
+# else
+#  define USE_PRINTF_I64 0
+# endif
 # define PRINTF_HAS_LONG_LONG
 #else
 # define USE_PRINTF_I64 0
