diff -Naur a/gdb/gdbtk/generic/gdbtk.c b/gdb/gdbtk/generic/gdbtk.c
--- a/gdb/gdbtk/generic/gdbtk.c	2013-09-20 15:38:18.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk.c	2014-03-12 11:16:02.324400000 +0400
@@ -48,14 +48,14 @@
 #include "gdbtk.h"
 
 #include <fcntl.h>
-#include "gdb_stat.h"
+#include <sys/stat.h>
 #ifdef HAVE_SYS_IOCTL_H
 #include <sys/ioctl.h>
 #endif
 #include <sys/time.h>
 #include <signal.h>
 
-#include "gdb_string.h"
+#include <string.h>
 #include "dis-asm.h"
 #include "gdbcmd.h"
 
diff -Naur a/gdb/gdbtk/generic/gdbtk-bp.c b/gdb/gdbtk/generic/gdbtk-bp.c
--- a/gdb/gdbtk/generic/gdbtk-bp.c	2013-02-16 11:05:36.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-bp.c	2014-03-12 11:14:31.407600000 +0400
@@ -25,7 +25,7 @@
 #include "linespec.h"
 #include "breakpoint.h"
 #include "tracepoint.h"
-#include "gdb_string.h"
+#include <string.h>
 #include <tcl.h>
 #include "gdbtk.h"
 #include "gdbtk-cmds.h"
diff -Naur a/gdb/gdbtk/generic/gdbtk-cmds.c b/gdb/gdbtk/generic/gdbtk-cmds.c
--- a/gdb/gdbtk/generic/gdbtk-cmds.c	2013-08-15 18:26:28.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-cmds.c	2014-03-12 11:17:02.758800000 +0400
@@ -65,9 +65,9 @@
 #include <sys/ioctl.h>
 #endif
 #include <sys/time.h>
-#include "gdb_stat.h"
+#include <sys/stat.h>
 
-#include "gdb_string.h"
+#include <string.h>
 #include "dis-asm.h"
 #include "gdbcmd.h"
 
diff -Naur a/gdb/gdbtk/generic/gdbtk-hooks.c b/gdb/gdbtk/generic/gdbtk-hooks.c
--- a/gdb/gdbtk/generic/gdbtk-hooks.c	2014-01-09 22:52:28.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-hooks.c	2014-03-12 11:17:41.743200000 +0400
@@ -56,7 +56,7 @@
 #endif
 #include <sys/time.h>
 
-#include "gdb_string.h"
+#include <string.h>
 #include "dis-asm.h"
 #include "gdbcmd.h"
 
diff -Naur a/gdb/gdbtk/generic/gdbtk-interp.c b/gdb/gdbtk/generic/gdbtk-interp.c
--- a/gdb/gdbtk/generic/gdbtk-interp.c	2013-09-05 16:55:57.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-interp.c	2014-03-12 11:18:28.917600000 +0400
@@ -25,7 +25,7 @@
 #include "ui-file.h"
 #include "ui-out.h"
 #include "cli-out.h"
-#include "gdb_string.h"
+#include <string.h>
 #include "cli/cli-cmds.h"
 #include "cli/cli-decode.h"
 #include "exceptions.h"
diff -Naur a/gdb/gdbtk/generic/gdbtk-main.c b/gdb/gdbtk/generic/gdbtk-main.c
--- a/gdb/gdbtk/generic/gdbtk-main.c	2014-03-12 11:23:40.032600000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-main.c	2014-03-12 11:24:07.242200000 +0400
@@ -22,7 +22,7 @@
 
 #include "defs.h"
 #include "main.h"
-#include "gdb_string.h"
+#include <string.h>
 
 int
 main (int argc, char **argv)
diff -Naur a/gdb/gdbtk/generic/gdbtk-register.c b/gdb/gdbtk/generic/gdbtk-register.c
--- a/gdb/gdbtk/generic/gdbtk-register.c	2013-09-20 15:49:16.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-register.c	2014-03-12 11:18:57.450000000 +0400
@@ -24,7 +24,7 @@
 #include "reggroups.h"
 #include "value.h"
 #include "target.h"
-#include "gdb_string.h"
+#include <string.h>
 #include "language.h"
 #include "valprint.h"
 #include "arch-utils.h"
diff -Naur a/gdb/gdbtk/generic/gdbtk-varobj.c b/gdb/gdbtk/generic/gdbtk-varobj.c
--- a/gdb/gdbtk/generic/gdbtk-varobj.c	2010-04-30 01:00:52.000000000 +0400
+++ b/gdb/gdbtk/generic/gdbtk-varobj.c	2014-03-12 11:19:24.750000000 +0400
@@ -21,7 +21,7 @@
 
 #include "defs.h"
 #include "value.h"
-#include "gdb_string.h"
+#include <string.h>
 #include "varobj.h"
 #include "exceptions.h"
 
