diff -c -r edd-dbg-4bf707ee0f03.orig/src/windows/frames.cpp edd-dbg-4bf707ee0f03/src/windows/frames.cpp
*** edd-dbg-4bf707ee0f03.orig/src/windows/frames.cpp	2013-03-14 00:46:30.000000000 -0300
--- edd-dbg-4bf707ee0f03/src/windows/frames.cpp	2016-03-12 14:33:52.977002100 -0300
***************
*** 30,36 ****
          context.ContextFlags = CONTEXT_FULL;
  
          // RtlCaptureContext() crashes with heavy optimizations on MinGW 4.7.
! #if defined(__MINGW32__)
          DWORD eip_val = 0;
          DWORD esp_val = 0;
          DWORD ebp_val = 0;
--- 30,36 ----
          context.ContextFlags = CONTEXT_FULL;
  
          // RtlCaptureContext() crashes with heavy optimizations on MinGW 4.7.
! #ifndef _WIN64
          DWORD eip_val = 0;
          DWORD esp_val = 0;
          DWORD ebp_val = 0;
