--- qtbase-everywhere-src-6.1.2/qmake/CMakeLists.txt
+++ qtbase-everywhere-src-6.1.2/qmake/CMakeLists.txt
@@ -9,7 +9,6 @@
     NO_QT # special case
     TOOLS_TARGET Core # special case
     USER_FACING
-    INSTALL_VERSIONED_LINK
     SOURCES
         ../src/3rdparty/pcre2/src/config.h
         ../src/3rdparty/pcre2/src/pcre2.h
@@ -287,3 +286,6 @@
 )
 # special case end
 
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME qmake-qt6
+)
--- qtbase-everywhere-src-6.1.2/src/tools/qdbuscpp2xml/CMakeLists.txt
+++ qtbase-everywhere-src-6.1.2/src/tools/qdbuscpp2xml/CMakeLists.txt
@@ -47,3 +47,7 @@
 # qt_internal_extend_target(qdbuscpp2xml CONDITION force_bootstrap [...])
 # qt_internal_extend_target(qdbuscpp2xml CONDITION NOT force_bootstrap [...])
 # special case end
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME qdbuscpp2xml-qt6
+)
--- qtbase-everywhere-src-6.1.2/src/tools/qdbusxml2cpp/CMakeLists.txt
+++ qtbase-everywhere-src-6.1.2/src/tools/qdbusxml2cpp/CMakeLists.txt
@@ -35,3 +35,7 @@
 # qt_internal_extend_target(qdbusxml2cpp CONDITION NOT QT_FEATURE_commandlineparser AND NOT force_bootstrap [...])
 # qt_internal_extend_target(qdbusxml2cpp CONDITION force_bootstrap [...])
 # special case end
+
+set_target_properties(${target_name} PROPERTIES
+    OUTPUT_NAME qdbusxml2cpp-qt6
+)
