--- a/src/network/kernel/qhostaddress.cpp
+++ b/src/network/kernel/qhostaddress.cpp
@@ -63,6 +63,11 @@
 #endif
 
 QT_BEGIN_NAMESPACE
+
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv4Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv6Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::AnyIPProtocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::UnknownNetworkLayerProtocol;
 
 QHostAddressPrivate::QHostAddressPrivate()
     : a(0), protocol(QHostAddress::UnknownNetworkLayerProtocol)
