--- a/src/corelib/time/qtimezone.cpp
+++ b/src/corelib/time/qtimezone.cpp
@@ -17,6 +17,9 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr int QTimeZone::MinUtcOffsetSecs;
+constexpr int QTimeZone::MaxUtcOffsetSecs;
+
 static_assert(!std::is_constructible_v<QTimeZone, Qt::TimeSpec>);
 using namespace Qt::StringLiterals;
 
--- a/src/corelib/kernel/qdeadlinetimer.cpp
+++ b/src/corelib/kernel/qdeadlinetimer.cpp
@@ -7,6 +7,8 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr QDeadlineTimer::ForeverConstant QDeadlineTimer::Forever;
+
 QT_IMPL_METATYPE_EXTERN(QDeadlineTimer)
 
 using namespace std::chrono;
--- a/src/gui/rhi/qrhi.cpp
+++ b/src/gui/rhi/qrhi.cpp
@@ -25,6 +25,9 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr int QRhi::MAX_MIP_LEVELS;
+constexpr int QRhiShaderResourceBinding::LAYOUT_DESC_ENTRIES_PER_BINDING;
+
 // Play nice with QSG_INFO since that is still the most commonly used
 // way to get graphics info printed from Qt Quick apps, and the Quick
 // scenegraph is our primary user.
--- a/src/network/kernel/qhostaddress.cpp
+++ b/src/network/kernel/qhostaddress.cpp
@@ -29,6 +29,11 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv4Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv6Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::AnyIPProtocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::UnknownNetworkLayerProtocol;
+
 #if QT_VERSION < QT_VERSION_CHECK(7, 0, 0)
 const QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv4Protocol;
 const QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv6Protocol;
