--- a/src/network/kernel/qhostaddress.cpp
+++ b/src/network/kernel/qhostaddress.cpp
@@ -63,6 +63,11 @@
 #endif
 
 QT_BEGIN_NAMESPACE
+
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv4Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv6Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::AnyIPProtocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::UnknownNetworkLayerProtocol;
 
 QHostAddressPrivate::QHostAddressPrivate()
     : a(0), protocol(QHostAddress::UnknownNetworkLayerProtocol)
--- qtbase-everywhere-src-6.3.0/src/corelib/global/qoperatingsystemversion.cpp.orig	2022-04-13 20:51:42.574017300 +0200
+++ qtbase-everywhere-src-6.3.0/src/corelib/global/qoperatingsystemversion.cpp	2022-04-13 20:42:14.498020700 +0200
@@ -685,4 +685,13 @@
 }
 #endif // !QT_NO_DEBUG_STREAM
 
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_1809;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_1903;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_1909;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_2004;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_20H2;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_21H1;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows10_21H2;
+constexpr QOperatingSystemVersionBase QOperatingSystemVersion::Windows11;
+
 QT_END_NAMESPACE
