--- a/src/corelib/time/qtimezone.cpp
+++ b/src/corelib/time/qtimezone.cpp
@@ -15,6 +15,9 @@
 #include <algorithm>
 
 QT_BEGIN_NAMESPACE
+
+constexpr int QTimeZone::MinUtcOffsetSecs;
+constexpr int QTimeZone::MaxUtcOffsetSecs;
 
 using namespace Qt::StringLiterals;
 
--- a/src/corelib/kernel/qdeadlinetimer.cpp
+++ b/src/corelib/kernel/qdeadlinetimer.cpp
@@ -5,6 +5,8 @@
 #include "private/qnumeric_p.h"
 
 QT_BEGIN_NAMESPACE
+
+constexpr QDeadlineTimer::ForeverConstant QDeadlineTimer::Forever;
 
 QT_IMPL_METATYPE_EXTERN(QDeadlineTimer)
 
--- a/src/gui/rhi/qrhi.cpp
+++ b/src/gui/rhi/qrhi.cpp
@@ -23,6 +23,9 @@
 #include <memory>
 
 QT_BEGIN_NAMESPACE
+
+constexpr int QRhi::MAX_MIP_LEVELS;
+constexpr int QRhiShaderResourceBinding::LAYOUT_DESC_ENTRIES_PER_BINDING;
 
 Q_LOGGING_CATEGORY(QRHI_LOG_INFO, "qt.rhi.general")
 
--- a/src/network/kernel/qhostaddress.cpp
+++ b/src/network/kernel/qhostaddress.cpp
@@ -27,6 +27,11 @@
 #endif
 
 QT_BEGIN_NAMESPACE
+
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv4Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::IPv6Protocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::AnyIPProtocol;
+constexpr QAbstractSocket::NetworkLayerProtocol QHostAddress::UnknownNetworkLayerProtocol;
 
 QHostAddressPrivate::QHostAddressPrivate()
     : a(0), protocol(QHostAddress::UnknownNetworkLayerProtocol)
