--- a/src/tools/windeployqt/utils.cpp.orig	Thu Jul  6 12:02:03 2023
+++ b/src/tools/windeployqt/utils.cpp	Mon Aug  7 19:27:50 2023
@@ -101,12 +101,12 @@
     const QFileInfoList &dlls = directory.entryInfoList(QStringList(nameFilter), QDir::Files);
     for (const QFileInfo &dllFi : dlls) {
         const QString dllPath = dllFi.absoluteFilePath();
+        bool isMinGW = platform == WindowsDesktopMinGW || platform == WindowsDesktopClangMinGW;
         bool matches = true;
         if (debugMatchMode != MatchDebugOrRelease && (platform & WindowsBased)) {
             bool debugDll;
-            if (readPeExecutable(dllPath, &errorMessage, 0, 0, &debugDll,
-                                 (platform == WindowsDesktopMinGW))) {
-                matches = debugDll == (debugMatchMode == MatchDebug);
+            if (readPeExecutable(dllPath, &errorMessage, 0, 0, &debugDll, isMinGW)) {
+                matches = debugDll == (debugMatchMode == MatchDebug) || isMinGW;
             } else {
                 std::wcerr << "Warning: Unable to read " << QDir::toNativeSeparators(dllPath)
                            << ": " << errorMessage;
