diff --git a/src/libespeak-ng/speech.c b/src/libespeak-ng/speech.c
index 3c36e9a5..beec848b 100644
--- a/src/libespeak-ng/speech.c
+++ b/src/libespeak-ng/speech.c
@@ -879,15 +879,15 @@ ESPEAK_API const char *espeak_TextToPhonemes(const void **textptr, int textmode,
 
 ESPEAK_NG_API espeak_ng_STATUS espeak_ng_Cancel(void)
 {
-#if USE_ASYNC
-	fifo_stop();
-	event_clear_all();
-#endif
-
 #if USE_LIBPCAUDIO
 	if ((my_mode & ENOUTPUT_MODE_SPEAK_AUDIO) == ENOUTPUT_MODE_SPEAK_AUDIO)
 		audio_object_flush(my_audio);
 #endif
+
+#if USE_ASYNC
+	fifo_stop();
+	event_clear_all();
+#endif
 	embedded_value[EMBED_T] = 0; // reset echo for pronunciation announcements
 
 	for (int i = 0; i < N_SPEECH_PARAM; i++)
diff --git a/src/libespeak-ng/speech.h b/src/libespeak-ng/speech.h
index ee81d052..97cd7dd2 100644
--- a/src/libespeak-ng/speech.h
+++ b/src/libespeak-ng/speech.h
@@ -55,7 +55,12 @@ extern "C"
 #if defined(_WIN32) || defined(_WIN64) // Windows
 
 #define PLATFORM_WINDOWS 1
-#define PATHSEP '\\'
+
+#ifdef __MINGW32__
+    #define PATHSEP '/'
+#else
+    #define PATHSEP '\\'
+#endif
 #define N_PATH_HOME_DEF  230
 #define NO_VARIADIC_MACROS
 
diff --git a/src/libespeak-ng/voices.c b/src/libespeak-ng/voices.c
index 756b2472..68c0ae37 100644
--- a/src/libespeak-ng/voices.c
+++ b/src/libespeak-ng/voices.c
@@ -80,7 +80,7 @@ static const char variants_male[N_VOICE_VARIANTS] = { 1, 2, 3, 4, 5, 6, 0 };
 static const char variants_female[N_VOICE_VARIANTS] = { 11, 12, 13, 14, 0 };
 static const char *const variant_lists[3] = { variants_either, variants_male, variants_female };
 
-#ifdef _WIN32
+#if !defined(__MINGW32__) && defined(_WIN32)
 #define MB_PREFIX "mb\\"
 #else
 #define MB_PREFIX "mb/"
