From 6f07580b8dd6ec0cf3419d5bceff097de2e213df Mon Sep 17 00:00:00 2001
From: mcarans <rans@email.com>
Date: Sun, 21 Sep 2025 17:31:01 +1200
Subject: [PATCH 1/5] audio_object_flush before fifo_stop

---
 src/libespeak-ng/speech.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/libespeak-ng/speech.c b/src/libespeak-ng/speech.c
index 3c36e9a5a..beec848b8 100644
--- a/src/libespeak-ng/speech.c
+++ b/src/libespeak-ng/speech.c
@@ -879,15 +879,15 @@ ESPEAK_API const char *espeak_TextToPhonemes(const void **textptr, int textmode,
 
 ESPEAK_NG_API espeak_ng_STATUS espeak_ng_Cancel(void)
 {
-#if USE_ASYNC
-	fifo_stop();
-	event_clear_all();
-#endif
-
 #if USE_LIBPCAUDIO
 	if ((my_mode & ENOUTPUT_MODE_SPEAK_AUDIO) == ENOUTPUT_MODE_SPEAK_AUDIO)
 		audio_object_flush(my_audio);
 #endif
+
+#if USE_ASYNC
+	fifo_stop();
+	event_clear_all();
+#endif
 	embedded_value[EMBED_T] = 0; // reset echo for pronunciation announcements
 
 	for (int i = 0; i < N_SPEECH_PARAM; i++)

From 247839fa8ee9b3cf327c4b836441a0d4be0e6e0d Mon Sep 17 00:00:00 2001
From: Mike <rans@email.com>
Date: Tue, 30 Sep 2025 09:50:14 +1300
Subject: [PATCH 2/5] Change path separator for mingw64

---
 src/libespeak-ng/speech.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/libespeak-ng/speech.h b/src/libespeak-ng/speech.h
index ee81d0526..bf57d3b16 100644
--- a/src/libespeak-ng/speech.h
+++ b/src/libespeak-ng/speech.h
@@ -55,10 +55,15 @@ extern "C"
 #if defined(_WIN32) || defined(_WIN64) // Windows
 
 #define PLATFORM_WINDOWS 1
-#define PATHSEP '\\'
 #define N_PATH_HOME_DEF  230
 #define NO_VARIADIC_MACROS
 
+#if defined(__MINGW32__) || defined(__MINGW64__)
+#define PATHSEP  '/'
+#else
+#define PATHSEP '\\'
+#endif
+
 #else
 
 #define PLATFORM_POSIX 1

From 553d189ee384a303846b0e2e2bf5cdfccff640e6 Mon Sep 17 00:00:00 2001
From: Mike <rans@email.com>
Date: Wed, 1 Oct 2025 09:49:15 +1300
Subject: [PATCH 3/5] Change back speech.h to original Add msys2path conversion
 to CMakeLists.txt for mingw only

---
 cmake/msys2path.cmake           | 9 +++++++++
 src/libespeak-ng/CMakeLists.txt | 9 ++++++++-
 src/libespeak-ng/speech.h       | 7 +------
 3 files changed, 18 insertions(+), 7 deletions(-)
 create mode 100644 cmake/msys2path.cmake

diff --git a/cmake/msys2path.cmake b/cmake/msys2path.cmake
new file mode 100644
index 000000000..9f6b1b6d2
--- /dev/null
+++ b/cmake/msys2path.cmake
@@ -0,0 +1,9 @@
+# Usage: msys2_to_windows("${msys_path}" win_path)
+
+function(msys_to_windows MsysPath WinPath)
+    execute_process(
+            COMMAND cygpath -w "${MsysPath}"
+            OUTPUT_VARIABLE converted_path
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+endfunction()
\ No newline at end of file
diff --git a/src/libespeak-ng/CMakeLists.txt b/src/libespeak-ng/CMakeLists.txt
index bb39dc281..4c12244cd 100644
--- a/src/libespeak-ng/CMakeLists.txt
+++ b/src/libespeak-ng/CMakeLists.txt
@@ -69,7 +69,14 @@ if (NOT BUILD_SHARED_LIBS)
   target_compile_definitions(espeak-ng INTERFACE "LIBESPEAK_NG_EXPORT=1")
 endif()
 
-target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=\"${CMAKE_INSTALL_PREFIX}/share/espeak-ng-data\"")
+set(espeakngdata "\"${CMAKE_INSTALL_PREFIX}/share/espeak-ng-data\"")
+if (MINGW)
+  include("${CMAKE_SOURCE_DIR}/cmake/msys2path.cmake")
+  msys_to_windows("${espeakngdata}" espeakngdata)
+  message(STATUS "Converted espeak-ng data path: ${espeakngdata}")
+endif()
+
+target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=${espeakngdata}")
 
 if (USE_ASYNC)
   target_sources(espeak-ng PRIVATE
diff --git a/src/libespeak-ng/speech.h b/src/libespeak-ng/speech.h
index bf57d3b16..ee81d0526 100644
--- a/src/libespeak-ng/speech.h
+++ b/src/libespeak-ng/speech.h
@@ -55,15 +55,10 @@ extern "C"
 #if defined(_WIN32) || defined(_WIN64) // Windows
 
 #define PLATFORM_WINDOWS 1
+#define PATHSEP '\\'
 #define N_PATH_HOME_DEF  230
 #define NO_VARIADIC_MACROS
 
-#if defined(__MINGW32__) || defined(__MINGW64__)
-#define PATHSEP  '/'
-#else
-#define PATHSEP '\\'
-#endif
-
 #else
 
 #define PLATFORM_POSIX 1

From 31373ca6a54be87247f0c84e874c1e37ed08a10d Mon Sep 17 00:00:00 2001
From: Mike <rans@email.com>
Date: Wed, 1 Oct 2025 10:22:56 +1300
Subject: [PATCH 4/5] Fix msys2path

---
 cmake/msys2path.cmake           | 8 ++++++++
 src/libespeak-ng/CMakeLists.txt | 8 +++++---
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/cmake/msys2path.cmake b/cmake/msys2path.cmake
index 9f6b1b6d2..0d015f2a9 100644
--- a/cmake/msys2path.cmake
+++ b/cmake/msys2path.cmake
@@ -5,5 +5,13 @@ function(msys_to_windows MsysPath WinPath)
             COMMAND cygpath -w "${MsysPath}"
             OUTPUT_VARIABLE converted_path
             OUTPUT_STRIP_TRAILING_WHITESPACE
+            ERROR_VARIABLE err
+            RESULT_VARIABLE res
     )
+    if(NOT res EQUAL 0)
+        message(WARNING "cygpath failed: ${err}")
+        set(${WinPath} "${MsysPath}" PARENT_SCOPE)
+    else ()
+        set(${WinPath} "${converted_path}" PARENT_SCOPE)
+    endif ()
 endfunction()
\ No newline at end of file
diff --git a/src/libespeak-ng/CMakeLists.txt b/src/libespeak-ng/CMakeLists.txt
index 4c12244cd..281908db6 100644
--- a/src/libespeak-ng/CMakeLists.txt
+++ b/src/libespeak-ng/CMakeLists.txt
@@ -69,14 +69,16 @@ if (NOT BUILD_SHARED_LIBS)
   target_compile_definitions(espeak-ng INTERFACE "LIBESPEAK_NG_EXPORT=1")
 endif()
 
-set(espeakngdata "\"${CMAKE_INSTALL_PREFIX}/share/espeak-ng-data\"")
+set(espeakngdata_in "${CMAKE_INSTALL_PREFIX}/share/espeak-ng-data")
 if (MINGW)
   include("${CMAKE_SOURCE_DIR}/cmake/msys2path.cmake")
-  msys_to_windows("${espeakngdata}" espeakngdata)
+  msys_to_windows("${espeakngdata_in}" espeakngdata)
   message(STATUS "Converted espeak-ng data path: ${espeakngdata}")
+else ()
+  set(espeakngdata "${espeakngdata_in}")
 endif()
 
-target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=${espeakngdata}")
+target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=\"${espeakngdata}\"")
 
 if (USE_ASYNC)
   target_sources(espeak-ng PRIVATE

From 67d8e09ae8c3378a6e9862fc319d9991a4a9dd88 Mon Sep 17 00:00:00 2001
From: Mike <rans@email.com>
Date: Wed, 1 Oct 2025 10:35:05 +1300
Subject: [PATCH 5/5] Fix msys2path

---
 cmake/msys2path.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/msys2path.cmake b/cmake/msys2path.cmake
index 0d015f2a9..7c6cb491c 100644
--- a/cmake/msys2path.cmake
+++ b/cmake/msys2path.cmake
@@ -12,6 +12,7 @@ function(msys_to_windows MsysPath WinPath)
         message(WARNING "cygpath failed: ${err}")
         set(${WinPath} "${MsysPath}" PARENT_SCOPE)
     else ()
-        set(${WinPath} "${converted_path}" PARENT_SCOPE)
+        string(REPLACE "\\" "/" norm_path "${converted_path}")
+        set(${WinPath} "${norm_path}" PARENT_SCOPE)
     endif ()
 endfunction()
\ No newline at end of file
