--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,7 +89,6 @@ if(MSVC)
 
     set(BUILD_SHARED_LIBS ON)
     set(GENERATE_MAN_PAGES OFF)
-    set(LIBCRYPTO_LIBRARIES ${LIBCRYPTO_LIBRARIES} bcrypt)
 else()
     find_package (PkgConfig REQUIRED)
 
@@ -105,6 +104,10 @@ else()
     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer")
 endif()
 
+if(WIN32)
+    set(LIBCRYPTO_LIBRARIES ${LIBCRYPTO_LIBRARIES} bcrypt)
+endif(WIN32)
+
 # Use -Wshorten-64-to-32 if available.
 check_c_compiler_flag("-Wshorten-64-to-32" HAVE_SHORTEN_64_TO_32)
 if(HAVE_SHORTEN_64_TO_32)
--- a/cmake/check.cmake
+++ b/cmake/check.cmake
@@ -26,7 +26,7 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 macro (find_check)
-    if(WIN32)
+    if(MSVC)
 
         if(NOT check_FOUND)
             find_package(check CONFIG PATHS ${CHECK_PATH})
@@ -50,7 +50,7 @@ macro (find_check)
             endif(check_FOUND)
         endif(NOT check_FOUND)
 
-    else(WIN32)
+    else(MSVC)
 
         if(NOT LIBCHECK_FOUND)
             pkg_check_modules(LIBCHECK REQUIRED check)
@@ -74,7 +74,7 @@ macro (find_check)
             endif(LIBCHECK_FOUND)
         endif(NOT LIBCHECK_FOUND)
 
-    endif(WIN32)
+    endif(MSVC)
 
     include_directories(${LIBCHECK_INCLUDE_DIRS})
 
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -85,11 +85,11 @@ set_target_properties(ykpiv_shared PROPERTIES SOVERSION ${SO_VERSION} VERSION ${
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set_target_properties(ykpiv_shared PROPERTIES INSTALL_RPATH "${YKPIV_INSTALL_LIB_DIR}")
 endif()
-if(WIN32)
+if(MSVC)
     set_target_properties(ykpiv_shared PROPERTIES OUTPUT_NAME libykpiv)
-else(WIN32)
+else(MSVC)
     set_target_properties(ykpiv_shared PROPERTIES OUTPUT_NAME ykpiv)
-endif(WIN32)
+endif(MSVC)
 add_coverage(ykpiv_shared)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ykpiv.pc.in ${CMAKE_CURRENT_SOURCE_DIR}/ykpiv.pc @ONLY)
--- a/tool/CMakeLists.txt
+++ b/tool/CMakeLists.txt
@@ -35,7 +35,7 @@ set (SOURCE
         ../common/openssl-compat.c
         ../common/util.c)
 
-if(WIN32)
+if(MSVC)
     add_definitions (-DPACKAGE="yubico-piv-tool")
     add_definitions (-DVERSION="${yubico_piv_tool_VERSION_MAJOR}.${yubico_piv_tool_VERSION_MINOR}.${yubico_piv_tool_VERSION_PATCH}")
 
@@ -45,12 +45,12 @@ if(WIN32)
     include_directories(${GETOPT_INCLUDE_DIR})
 
     set(LINK_LIBS_WIN ${GETOPT})
-else(WIN32)
+else(MSVC)
     include(${CMAKE_SOURCE_DIR}/cmake/gengetopt.cmake)
     find_gengetopt()
     add_gengetopt_files(cmdline)
     set(SOURCE ${SOURCE} ${GGO_C})
-endif(WIN32)
+endif(MSVC)
 
 include_directories (
         ${CMAKE_SOURCE_DIR}/lib
@@ -62,9 +62,9 @@

     find_library(ZLIB zlib PATHS ${ZLIB_LIB_DIR})
     include_directories(${ZLIB_INCL_DIR})
-    if (WIN32)
+    if (MSVC)
         set(LINK_LIBS_WIN ${LINK_LIBS_WIN} ${ZLIB})
-    endif(WIN32)
+    endif(MSVC)

     find_package(ZLIB REQUIRED)
 endif()
--- a/tool/tests/test_inout.c
+++ b/tool/tests/test_inout.c
@@ -38,7 +38,11 @@
 #include "../../common/util.h"
 
 #ifdef _WIN32
+#ifdef _MSC_VER
 #include <openssl/applink.c>
+#else
+#include <io.h>
+#endif
 #define pipe(fds) _pipe(fds,4096, 0)
 #else
 #include <unistd.h>
--- a/tool/yubico-piv-tool.c
+++ b/tool/yubico-piv-tool.c
@@ -39,7 +39,7 @@
 
 #include "ykpiv.h"
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 #include <windows.h>
 #include <openssl/applink.c>
 #else
--- a/ykcs11/CMakeLists.txt
+++ b/ykcs11/CMakeLists.txt
@@ -78,11 +78,11 @@ set_target_properties(ykcs11_shared PROPERTIES SOVERSION ${SO_VERSION} VERSION $
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set_target_properties(ykcs11_shared PROPERTIES INSTALL_RPATH "${YKPIV_INSTALL_LIB_DIR}")
 endif()
-if(WIN32)
+if(MSVC)
     set_target_properties(ykcs11_shared PROPERTIES OUTPUT_NAME libykcs11)
-else(WIN32)
+else(MSVC)
     set_target_properties(ykcs11_shared PROPERTIES OUTPUT_NAME ykcs11)
-endif(WIN32)
+endif(MSVC)
 add_coverage(ykcs11_shared)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ykcs11.pc.in ${CMAKE_CURRENT_SOURCE_DIR}/ykcs11.pc @ONLY)
