--- blender-2.82/build_files/cmake/Modules/FindLLVM.cmake.orig	2020-02-17 20:27:31.502040900 +0300
+++ blender-2.82/build_files/cmake/Modules/FindLLVM.cmake	2020-02-17 20:31:51.327226500 +0300
@@ -50,6 +50,7 @@
           OUTPUT_STRIP_TRAILING_WHITESPACE)
   set(LLVM_ROOT_DIR ${LLVM_ROOT_DIR} CACHE PATH "Path to the LLVM installation")
 endif()
+file(TO_CMAKE_PATH "${LLVM_ROOT_DIR}" LLVM_ROOT_DIR)
 if(NOT LLVM_LIBPATH)
   execute_process(COMMAND ${LLVM_CONFIG} --libdir
           OUTPUT_VARIABLE LLVM_LIBPATH
@@ -57,6 +58,7 @@
   set(LLVM_LIBPATH ${LLVM_LIBPATH} CACHE PATH "Path to the LLVM library path")
   mark_as_advanced(LLVM_LIBPATH)
 endif()
+file(TO_CMAKE_PATH "${LLVM_LIBPATH}" LLVM_LIBPATH)
 
 if(LLVM_STATIC)
   find_library(LLVM_LIBRARY
@@ -69,7 +71,7 @@
                  LLVMAnalysis  # check for the static library as a fall-back
                PATHS ${LLVM_LIBPATH})
 endif()
-
+file(TO_CMAKE_PATH "${LLVM_LIBRARY}" LLVM_LIBRARY)
 
 if(LLVM_LIBRARY AND LLVM_ROOT_DIR AND LLVM_LIBPATH)
   if(LLVM_STATIC)
@@ -82,7 +84,7 @@
     string(REPLACE " " ";" LLVM_LIBRARY "${LLVM_LIBRARY}")
   endif()
 endif()
-
+file(TO_CMAKE_PATH "${LLVM_LIBRARY}" LLVM_LIBRARY)
 
 # handle the QUIETLY and REQUIRED arguments and set SDL2_FOUND to TRUE if
 # all listed variables are TRUE
