--- random/rndw32.c.orig	2011-07-04 19:11:18.137525831 +0200
+++ random/rndw32.c	2011-07-04 19:11:40.785404249 +0200
@@ -887,7 +887,7 @@
   {
     HANDLE handle;
     FILETIME creationTime, exitTime, kernelTime, userTime;
-    DWORD minimumWorkingSetSize, maximumWorkingSetSize;
+    SIZE_T minimumWorkingSetSize, maximumWorkingSetSize;
 
     handle = GetCurrentThread ();
     GetThreadTimes (handle, &creationTime, &exitTime,
