--- VTK-8.0.1/ThirdParty/libharu/vtk_libharu.h.in.orig	2017-09-05 19:09:32.245588300 +0300
+++ VTK-8.0.1/ThirdParty/libharu/vtk_libharu.h.in	2017-09-05 19:09:48.414513100 +0300
@@ -27,7 +27,7 @@
 
 /* HPDF requires that HPDF_DLL be defined when linking against a shared library
  * on MSVC. */
-#if defined(VTK_COMPILER_MSVC)
+#if defined(WIN32)
 
 #ifdef VTK_USE_SYSTEM_LIBHARU
 
--- VTK-8.1.0/IO/Export/CMakeLists.txt.orig	2018-01-16 14:40:52.435101000 +0300
+++ VTK-8.1.0/IO/Export/CMakeLists.txt	2018-01-16 14:43:22.684067900 +0300
@@ -36,6 +36,11 @@
   ABSTRACT
   )
 
+if(VTK_USE_SYSTEM_LIBHARU AND VTK_EXTERNAL_LIBHARU_IS_SHARED)
+  add_definitions(-DHPDF_DLL)
+elseif(VTK_BUILD_SHARED_LIBS)
+  add_definitions(-DHPDF_DLL)
+endif()
 vtk_module_library(vtkIOExport ${Module_SRCS})
 include(vtkOpenGL)
 vtk_opengl_link(vtkIOExport)
