From 4a7840f43a8e05e592794c1eb358623bc86b4d4e Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 21 Jul 2014 21:11:52 +0100
Subject: [PATCH 4/5] fix mingw64 32bit time temp

---
 git-compat-util.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/git-compat-util.h b/git-compat-util.h
index e0f258c..670c5e3 100644
--- a/git-compat-util.h
+++ b/git-compat-util.h
@@ -100,6 +100,9 @@ enum git_target_type {
 #  define _WIN32_WINNT 0x0502
 # endif
 #define WIN32_LEAN_AND_MEAN  /* stops windows.h including winsock.h */
+# ifdef __MINGW64__
+#  undef _USE_32BIT_TIME_T
+# endif
 #include <winsock2.h>
 #include <windows.h>
 #define GIT_WINDOWS_NATIVE
-- 
2.0.1

