From 7841c243d2b382fd24076f669abaab5224e5c9e3 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 21 Jul 2014 23:52:34 +0100
Subject: [PATCH 06/16] Include ntdef.h on MinGW-w64 for REPARSE_DATA_BUFFER

---
 compat/mingw.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/compat/mingw.c b/compat/mingw.c
index 013c938..0604c62 100644
--- a/compat/mingw.c
+++ b/compat/mingw.c
@@ -6,6 +6,10 @@
 #include "../run-command.h"
 #include "../cache.h"
 
+#ifdef __MINGW64_VERSION_MAJOR
+#include <ntdef.h>
+#endif
+
 static const int delay[] = { 0, 1, 10, 20, 40 };
 unsigned int _CRT_fmode = _O_BINARY;
 
-- 
2.1.2

