From 6614d3e004096a8ae99424476a2a91e3db46f960 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 21 Jul 2014 21:10:25 +0100
Subject: [PATCH 8/9] fix mingw64 compat wide char

---
 compat/mingw.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/compat/mingw.h b/compat/mingw.h
index 2dd1279..1b07f17 100644
--- a/compat/mingw.h
+++ b/compat/mingw.h
@@ -1,14 +1,13 @@
+#include <stdint.h>
+#include <wchar.h>
 #include <winsock2.h>
 #include <ws2tcpip.h>
-
 /*
  * things that are not available in header files
  */
 
-typedef int pid_t;
 typedef int uid_t;
 typedef int socklen_t;
-#define hstrerror strerror
 
 #define S_IFLNK    0120000 /* Symbolic link */
 #define S_ISLNK(x) (((x) & S_IFMT) == S_IFLNK)
-- 
2.0.1

