From 991c718c4309c0e2ada2ccd5803abf0b25267bf2 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 21 Jul 2014 21:11:03 +0100
Subject: [PATCH 9/9] fix fork for mingw64

---
 compat/mingw.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/compat/mingw.h b/compat/mingw.h
index 1b07f17..1f9b4ef 100644
--- a/compat/mingw.h
+++ b/compat/mingw.h
@@ -85,8 +85,8 @@ struct itimerval {
 
 static inline int fchmod(int fildes, mode_t mode)
 { errno = ENOSYS; return -1; }
-static inline pid_t fork(void)
-{ errno = ENOSYS; return -1; }
+//static inline pid_t fork(void)
+//{ errno = ENOSYS; return -1; }
 static inline unsigned int alarm(unsigned int seconds)
 { return 0; }
 static inline int fsync(int fd)
-- 
2.0.1

