From 026c6f2f760ed005c964e23367735f228e524fc7 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Thu, 24 Jul 2014 00:15:28 +0100
Subject: [PATCH 14/16] Do not compile dirent on mingw64

---
 config.mak.uname | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/config.mak.uname b/config.mak.uname
index 548e377..3285490 100644
--- a/config.mak.uname
+++ b/config.mak.uname
@@ -512,8 +512,7 @@ ifneq (,$(findstring MINGW,$(uname_S)))
 	endif
 	COMPAT_CFLAGS += -DSTRIP_EXTENSION=\".exe\"
 	COMPAT_OBJS += compat/mingw.o compat/winansi.o \
-		compat/win32/syslog.o \
-		compat/win32/dirent.o compat/win32/fscache.o
+		compat/win32/syslog.o compat/win32/fscache.o
 	SOCKET_LIB = -lws2_32
 	GITLIBS += git.res
 	PTHREAD_LIBS =
-- 
2.1.2

