diff --git a/compat/mingw.h b/compat/mingw.h
index 9dbba85..28090d2 100644
--- a/compat/mingw.h
+++ b/compat/mingw.h
@@ -1,14 +1,13 @@
+#include <stdint.h>
+#include <wchar.h>
 #include <winsock2.h>
 #include <ws2tcpip.h>
-
 /*
  * things that are not available in header files
  */
 
-typedef int pid_t;
 typedef int uid_t;
 typedef int socklen_t;
-#define hstrerror strerror
 
 #define S_IFLNK    0120000 /* Symbolic link */
 #define S_ISLNK(x) (((x) & S_IFMT) == S_IFLNK)
@@ -86,8 +85,8 @@ struct itimerval {
 
 static inline int fchmod(int fildes, mode_t mode)
 { errno = ENOSYS; return -1; }
-static inline pid_t fork(void)
-{ errno = ENOSYS; return -1; }
+//static inline pid_t fork(void)
+//{ errno = ENOSYS; return -1; }
 static inline unsigned int alarm(unsigned int seconds)
 { return 0; }
 static inline int fsync(int fd)
diff --git a/configure.ac b/configure.ac
index 5a3d531..faf9296 100644
--- a/configure.ac
+++ b/configure.ac
@@ -742,11 +742,11 @@ GIT_CONF_SUBST([OLD_ICONV])
 ## Checks for typedefs, structures, and compiler characteristics.
 AC_MSG_NOTICE([CHECKS for typedefs, structures, and compiler characteristics])
 #
-TYPE_SOCKLEN_T
-case $ac_cv_type_socklen_t in
-  yes)	SOCKLEN_T='';;
-  *)  	SOCKLEN_T=$git_cv_socklen_t_equiv;;
-esac
+#TYPE_SOCKLEN_T
+#case $ac_cv_type_socklen_t in
+SOCKLEN_T='size_t'
+#  *)  	SOCKLEN_T=$git_cv_socklen_t_equiv;;
+#esac
 GIT_CONF_SUBST([SOCKLEN_T])
 
 # Define NO_D_INO_IN_DIRENT if you don't have d_ino in your struct dirent.
diff --git a/git-compat-util.h b/git-compat-util.h
index e0f258c..670c5e3 100644
--- a/git-compat-util.h
+++ b/git-compat-util.h
@@ -100,6 +100,9 @@ enum git_target_type {
 #  define _WIN32_WINNT 0x0502
 # endif
 #define WIN32_LEAN_AND_MEAN  /* stops windows.h including winsock.h */
+# ifdef __MINGW64__
+#  undef _USE_32BIT_TIME_T
+# endif
 #include <winsock2.h>
 #include <windows.h>
 #define GIT_WINDOWS_NATIVE
