diff -ur b/CMakeLists.txt a/CMakeLists.txt
--- b/CMakeLists.txt	2011-07-29 04:37:48.000000000 -0400
+++ a/CMakeLists.txt	2021-05-21 22:13:46.569418300 -0400
@@ -214,8 +214,6 @@
     IF(SNDFILE_FOUND)
         SET(HAS_SNDFILE 1)
         LINK_DIRECTORIES(${SNDFILE_LIBRARY_DIRS})
-        SET_SOURCE_FILES_PROPERTIES(src/codec_sndfile.cpp PROPERTIES
-                                    COMPILE_FLAGS "${SNDFILE_CFLAGS}")
     ELSE(SNDFILE_FOUND)
         FIND_PACKAGE(SndFile)
         IF(SNDFILE_FOUND)
diff -ur b/src/codec_dumb.cpp a/src/codec_dumb.cpp
--- b/src/codec_dumb.cpp	2011-07-29 04:37:48.000000000 -0400
+++ a/src/codec_dumb.cpp	2021-05-21 22:14:22.436515100 -0400
@@ -272,7 +272,7 @@
 
 private:
     // DUMBFILE iostream callbacks
-    static int skip(void *user_data, long offset)
+    static int skip(void *user_data, long long offset)
     {
         std::istream *stream = static_cast<dumbStream*>(user_data)->fstream;
         stream->clear();
@@ -282,7 +282,7 @@
         return -1;
     }
 
-    static long read(char *ptr, long size, void *user_data)
+    static dumb_ssize_t read(char *ptr, size_t size, void *user_data)
     {
         std::istream *stream = static_cast<dumbStream*>(user_data)->fstream;
         stream->clear();
