--- SDL2-2.0.3-orig/configure.in	2014-03-16 06:31:42.000000000 +0400
+++ SDL2-2.0.3/configure.in	2014-06-10 08:37:51.605200000 +0400
@@ -2859,12 +2859,15 @@
             have_loadso=yes
         fi
         # Set up the system libraries we need
-        if test -f /lib/w32api/libuuid.a; then
+        case "$host" in
+          *-*-cygwin*)
             LIBUUID=/lib/w32api/libuuid.a
-        else
+          ;;
+          *-*-mingw32*)
             LIBUUID=-luuid
-        fi
-        EXTRA_LDFLAGS="$EXTRA_LDFLAGS -luser32 -lgdi32 -lwinmm -limm32 -lole32 -loleaut32 -lshell32 -lversion $LIBUUID -XCClinker -static-libgcc"
+          ;;
+        esac
+        EXTRA_LDFLAGS="$EXTRA_LDFLAGS -luser32 -lgdi32 -lwinmm -limm32 -lole32 -loleaut32 -lshell32 -lversion $LIBUUID -static-libgcc"
         # The Windows platform requires special setup
         VERSION_SOURCES="$srcdir/src/main/windows/*.rc"
         SDLMAIN_SOURCES="$srcdir/src/main/windows/*.c"
