--- kicad/pcbnew/github/CMakeLists.txt.orig	2014-09-30 07:49:18.828000000 +0400
+++ kicad/pcbnew/github/CMakeLists.txt	2014-09-30 07:49:34.880400000 +0400
@@ -26,10 +26,6 @@
 #################################################
 include( download_avhttp )
 
-if( MINGW AND NOT OPENSSL_ROOT_DIR )
-    # download, compile and install to scratch dir a recent OPENSSL library and headers
-    include( download_openssl )
-else()
     find_package( OpenSSL REQUIRED )
     #message( STATUS "OPENSSL_FOUND:${OPENSSL_FOUND}  OPENSSL_LIBRARIES:${OPENSSL_LIBRARIES}" )
 
@@ -37,7 +33,6 @@
     # do it here incase some other link image needs these libraries
     set( OPENSSL_LIBRARIES   "${OPENSSL_LIBRARIES}"   CACHE FILEPATH "OpenSSL link libraries" )
     set( OPENSSL_INCLUDE_DIR "${OPENSSL_INCLUDE_DIR}" CACHE FILEPATH "OpenSSL include dir" )
-endif()
 
 
 # These are additions to any inherited from pcbnew dir:
