--- kicad/CMakeLists.txt.orig	2014-10-02 09:27:42.914800000 +0400
+++ kicad/CMakeLists.txt	2014-10-02 09:31:37.975600000 +0400
@@ -312,7 +312,7 @@
 set( KICAD_FP_LIB_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}"
      CACHE PATH "Default path where footprint libraries are installed." )
 
-if( UNIX )
+if( UNIX OR MINGW )
     # Everything without leading / is relative to CMAKE_INSTALL_PREFIX.
     set( KICAD_PLUGINS lib/kicad/plugins
         CACHE PATH "Location of KiCad plugins." )
@@ -323,17 +323,6 @@
     set( KICAD_FP_LIB_INSTALL_PATH "${KICAD_FP_LIB_INSTALL_PATH}/share/kicad/modules" )
 endif()
 
-if( MINGW )
-    # Everything without leading / is relative to CMAKE_INSTALL_PREFIX.
-    set( KICAD_PLUGINS ${KICAD_BIN}/plugins
-        CACHE PATH "Location of KiCad plugins." )
-    set( KICAD_DATA share
-        CACHE PATH "Location of KiCad data files." )
-    set( KICAD_DOCS doc
-        CACHE PATH "Location of KiCad documentation files." )
-    set( KICAD_FP_LIB_INSTALL_PATH "${KICAD_FP_LIB_INSTALL_PATH}/modules" )
-endif()
-
 set( KICAD_DEMOS ${KICAD_DATA}/demos
     CACHE PATH "Location of KiCad demo files." )
 set( KICAD_TEMPLATE ${KICAD_DATA}/template
