diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3dda778..953bfec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -309,7 +309,7 @@ set( KICAD_BIN bin
 set( KICAD_FP_LIB_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}"
      CACHE PATH "Default path where footprint libraries are installed." )
 
-if( UNIX AND NOT APPLE )
+if( NOT APPLE )
     # Everything without leading / is relative to CMAKE_INSTALL_PREFIX.
     set( KICAD_PLUGINS lib/kicad/plugins
         CACHE PATH "Location of KiCad plugins." )
@@ -320,17 +320,6 @@ if( UNIX AND NOT APPLE )
     set( KICAD_FP_LIB_INSTALL_PATH "${KICAD_FP_LIB_INSTALL_PATH}/share/kicad/modules" )
 endif()
 
-if( MINGW )
-    # Everything without leading / is relative to CMAKE_INSTALL_PREFIX.
-    set( KICAD_PLUGINS ${KICAD_BIN}/plugins
-        CACHE PATH "Location of KiCad plugins." )
-    set( KICAD_DATA share
-        CACHE PATH "Location of KiCad data files." )
-    set( KICAD_DOCS doc
-        CACHE PATH "Location of KiCad documentation files." )
-    set( KICAD_FP_LIB_INSTALL_PATH "${KICAD_FP_LIB_INSTALL_PATH}/modules" )
-endif()
-
 set( KICAD_DEMOS ${KICAD_DATA}/demos
     CACHE PATH "Location of KiCad demo files." )
 set( KICAD_TEMPLATE ${KICAD_DATA}/template
