--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1776,9 +1776,13 @@
   set_target_properties(shared PROPERTIES OUTPUT_NAME gmsh
      VERSION ${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION}.${GMSH_PATCH_VERSION}
      SOVERSION ${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION})
-  if(WIN32)
+  if(MSVC)
     set_target_properties(shared PROPERTIES PREFIX "" IMPORT_PREFIX ""
-      IMPORT_SUFFIX ".dll.lib" COMPILE_FLAGS "-DGMSH_DLL -DGMSH_DLL_EXPORT")
+      IMPORT_SUFFIX ".dll.lib")
+  endif()
+  if(WIN32)
+    set_target_properties(shared PROPERTIES
+      COMPILE_FLAGS "-DGMSH_DLL -DGMSH_DLL_EXPORT")
   elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     if(NOT ENABLE_PRIVATE_API)
       # hide all symbols by default (if not exporting the private API); symbols to
