diff -urN gmsh-gmsh_4_10_2/CMakeLists.txt.orig gmsh-gmsh_4_10_2/CMakeLists.txt
--- gmsh-gmsh_4_10_2/CMakeLists.txt.orig	2022-05-24 13:33:03.906684600 +0200
+++ gmsh-gmsh_4_10_2/CMakeLists.txt	2022-05-24 13:35:19.558104800 +0200
@@ -1673,8 +1673,12 @@
      VERSION ${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION}.${GMSH_PATCH_VERSION}
      SOVERSION ${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION})
   if(WIN32)
-    set_target_properties(shared PROPERTIES PREFIX "" IMPORT_PREFIX ""
-      IMPORT_SUFFIX ".lib" COMPILE_FLAGS "-DGMSH_DLL -DGMSH_DLL_EXPORT")
+    if(MSVC)
+      set_target_properties(shared PROPERTIES PREFIX "" IMPORT_PREFIX ""
+        IMPORT_SUFFIX ".lib")
+    endif()
+    set_target_properties(shared PROPERTIES
+      COMPILE_FLAGS "-DGMSH_DLL -DGMSH_DLL_EXPORT")
   elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang" AND NOT ENABLE_PRIVATE_API)
     # hide all symbols by default (if not exporting the private API); symbols to
     # export should be marked with __attribute__ ((visibility("default")))
