diff -Naur gmsh-gmsh_4_7_1.orig/CMakeLists.txt gmsh-gmsh_4_7_1/CMakeLists.txt
--- gmsh-gmsh_4_7_1.orig/CMakeLists.txt	2020-12-08 15:39:11.485922000 +0100
+++ gmsh-gmsh_4_7_1/CMakeLists.txt	2020-12-08 15:54:52.189447300 +0100
@@ -1332,7 +1332,13 @@
                      lib ${OCC_SYS_NAME}/lib ${OCC_SYS_NAME}/vc8/lib
                      ${OCC_SYS_NAME}/gcc/lib ${OCC_SYS_NAME}/gcc/bin)
         if(OCC_CAF_LIB)
-          list(APPEND OCC_CAF_LIBS ${OCC_CAF_LIB})
+			if(OCC MATCHES "TKService")
+				find_library(FREEIMAGE_LIB freeimage)
+				find_library(OPENVR_LIB openvr_api)
+				list(APPEND OCC_CAF_LIBS ${FREEIMAGE_LIB} ${OPENVR_LIB} ${OCC_CAF_LIB})
+			else()
+				list(APPEND OCC_CAF_LIBS ${OCC_CAF_LIB})
+			endif()
         else()
           message(STATUS "OCC CAF lib " ${OCC} " not Found")
         endif()
