diff -Naur gmsh-gmsh_4_7_1.orig/CMakeLists.txt gmsh-gmsh_4_7_1/CMakeLists.txt
--- gmsh-gmsh_4_7_1.orig/CMakeLists.txt	2020-11-28 17:14:38.673558700 +0100
+++ gmsh-gmsh_4_7_1/CMakeLists.txt	2020-11-28 17:16:43.698764500 +0100
@@ -212,7 +212,9 @@
       set(WIN32 1)
       add_definitions(-DWIN32)
+      set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc -static-libstdc++ -static")
+      set(CMAKE_SHARED_LINKER_FLAGS "-static-libgcc -static-libstdc++ -static")
       set(CMAKE_FIND_LIBRARY_PREFIXES "lib" "")
-      set(CMAKE_FIND_LIBRARY_SUFFIXES ".a" ".so" ".lib" ".LIB" ".dll" ".DLL" ".dll.a")
+      set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
     endif()
   endif()
 else()
