diff -bur gmsh-4.13.1-source-o/CMakeLists.txt gmsh-4.13.1-source/CMakeLists.txt
--- gmsh-4.13.1-source-o/CMakeLists.txt	2025-04-09 05:37:34.446557200 -0600
+++ gmsh-4.13.1-source/CMakeLists.txt	2025-04-09 05:37:51.048837600 -0600
@@ -1812,7 +1812,7 @@
 
 # OS specific linker options
 if(WIN32 AND NOT MSVC)
-  set(FLAGS "-Wl,--stack,16777216 -Wl,--image-base -Wl,0x10000000")
+  set(FLAGS "-Wl,--stack,16777216 -Wl")
   if(HAVE_FLTK)
     set(APPFLAGS "${FLAGS} -municode -mwindows")
     if(HAVE_64BIT_SIZE_T)
