--- a/Makefile.in
+++ b/Makefile.in
@@ -99,7 +99,7 @@
 # Additional flags for making a plugin.  This version script tells the
 # GNU linker to make all symbols in the library hidden except for the
 # public entry point.  It's not essential, but makes a tidier library.
-PLUGIN_LDFLAGS		= $(DYNAMIC_LDFLAGS) -Wl,--version-script=build/vamp-plugin.map
+PLUGIN_LDFLAGS		= $(DYNAMIC_LDFLAGS)
 
 
 # Adapt install details when found to be running on OSX (thanks to David O)
