--- gnutls-3.1.5/gl/m4/strtok_r.m4.orig	2013-05-03 03:34:40 +0400
+++ gnutls-3.1.5/gl/m4/strtok_r.m4	2013-05-03 04:34:01 +0400
@@ -13,6 +13,7 @@
   AC_REQUIRE([gl_HEADER_STRING_H_DEFAULTS])
   AC_REQUIRE([AC_CANONICAL_HOST]) dnl for cross-compiles
   AC_CHECK_FUNCS([strtok_r])
+  ac_cv_func_strtok_r=yes
   if test $ac_cv_func_strtok_r = yes; then
     HAVE_STRTOK_R=1
     dnl glibc 2.7 has a bug in strtok_r that causes a segmentation fault
--- gnutls-3.1.5/lib/verify-tofu.c.orig	2013-05-03 04:35:09 +0400
+++ gnutls-3.1.5/lib/verify-tofu.c	2013-05-03 05:21:50 +0400
@@ -34,6 +34,7 @@
 #include <gnutls/abstract.h>
 #include <system.h>
 #include <locks.h>
+#include <pthread.h>
 
 struct gnutls_tdb_int {
   gnutls_tdb_store_func store;
