--- gnutls-3.8.1/src/benchmark.h.orig	2023-08-15 12:08:01.059509200 +0200
+++ gnutls-3.8.1/src/benchmark.h	2023-08-15 12:10:19.840993900 +0200
@@ -31,19 +31,6 @@
 #include <stdint.h>
 #include "timespec.h"
 
-#if defined(HAVE_CLOCK_GETTIME) && defined(CLOCK_PROCESS_CPUTIME_ID)
-#undef gettime
-#define gettime(x) clock_gettime(CLOCK_PROCESS_CPUTIME_ID, x)
-#else
-inline static void gettime(struct timespec *ts)
-{
-	struct timeval tv;
-	gettimeofday(&tv, NULL);
-	ts->tv_sec = tv.tv_sec;
-	ts->tv_nsec = tv.tv_usec * 1000;
-}
-#endif
-
 typedef void (*sighandler_t)(int);
 
 void benchmark_cipher(int debug_level);
