diff -rupN gnutls-3.4.0.orig/lib/gnutls_supplemental.c gnutls-3.4.0/lib/gnutls_supplemental.c
--- gnutls-3.4.0.orig/lib/gnutls_supplemental.c	2015-04-04 12:36:46.000000000 +0200
+++ gnutls-3.4.0/lib/gnutls_supplemental.c	2015-04-09 21:58:51.870344400 +0200
@@ -42,7 +42,7 @@
  * append the 2-byte length of its data, and the data.  If it doesn't
  * want to send any data, it is fine to return without doing anything.
  */
-
+#define GNUTLS_INTERNAL_BUILD 1
 #include <gnutls/gnutls.h>
 #include "gnutls_int.h"
 #include "gnutls_supplemental.h"
diff -rupN gnutls-3.4.0.orig/gl/Makefile.am gnutls-3.4.0/gl/Makefile.am
--- gnutls-3.4.0.orig/gl/Makefile.am	2015-04-09 20:42:38.770341600 +0200
+++ gnutls-3.4.0/gl/Makefile.am	2015-04-09 20:50:43.063325300 +0200
@@ -1299,48 +1299,6 @@ EXTRA_DIST += sys_uio.in.h
 
 ## begin gnulib module time
 
-BUILT_SOURCES += time.h
-
-# We need the following in order to create <time.h> when the system
-# doesn't have one that works with the given compiler.
-time.h: time.in.h $(top_builddir)/config.status $(CXXDEFS_H) $(ARG_NONNULL_H) $(WARN_ON_USE_H)
-	$(AM_V_GEN)rm -f $@-t $@ && \
-	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */' && \
-	  sed -e 's|@''GUARD_PREFIX''@|GL|g' \
-	      -e 's|@''INCLUDE_NEXT''@|$(INCLUDE_NEXT)|g' \
-	      -e 's|@''PRAGMA_SYSTEM_HEADER''@|@PRAGMA_SYSTEM_HEADER@|g' \
-	      -e 's|@''PRAGMA_COLUMNS''@|@PRAGMA_COLUMNS@|g' \
-	      -e 's|@''NEXT_TIME_H''@|$(NEXT_TIME_H)|g' \
-	      -e 's/@''GNULIB_GETTIMEOFDAY''@/$(GNULIB_GETTIMEOFDAY)/g' \
-	      -e 's/@''GNULIB_MKTIME''@/$(GNULIB_MKTIME)/g' \
-	      -e 's/@''GNULIB_NANOSLEEP''@/$(GNULIB_NANOSLEEP)/g' \
-	      -e 's/@''GNULIB_STRPTIME''@/$(GNULIB_STRPTIME)/g' \
-	      -e 's/@''GNULIB_TIMEGM''@/$(GNULIB_TIMEGM)/g' \
-	      -e 's/@''GNULIB_TIME_R''@/$(GNULIB_TIME_R)/g' \
-	      -e 's|@''HAVE_DECL_LOCALTIME_R''@|$(HAVE_DECL_LOCALTIME_R)|g' \
-	      -e 's|@''HAVE_NANOSLEEP''@|$(HAVE_NANOSLEEP)|g' \
-	      -e 's|@''HAVE_STRPTIME''@|$(HAVE_STRPTIME)|g' \
-	      -e 's|@''HAVE_TIMEGM''@|$(HAVE_TIMEGM)|g' \
-	      -e 's|@''REPLACE_GMTIME''@|$(REPLACE_GMTIME)|g' \
-	      -e 's|@''REPLACE_LOCALTIME''@|$(REPLACE_LOCALTIME)|g' \
-	      -e 's|@''REPLACE_LOCALTIME_R''@|$(REPLACE_LOCALTIME_R)|g' \
-	      -e 's|@''REPLACE_MKTIME''@|$(REPLACE_MKTIME)|g' \
-	      -e 's|@''REPLACE_NANOSLEEP''@|$(REPLACE_NANOSLEEP)|g' \
-	      -e 's|@''REPLACE_TIMEGM''@|$(REPLACE_TIMEGM)|g' \
-	      -e 's|@''PTHREAD_H_DEFINES_STRUCT_TIMESPEC''@|$(PTHREAD_H_DEFINES_STRUCT_TIMESPEC)|g' \
-	      -e 's|@''SYS_TIME_H_DEFINES_STRUCT_TIMESPEC''@|$(SYS_TIME_H_DEFINES_STRUCT_TIMESPEC)|g' \
-	      -e 's|@''TIME_H_DEFINES_STRUCT_TIMESPEC''@|$(TIME_H_DEFINES_STRUCT_TIMESPEC)|g' \
-	      -e 's|@''UNISTD_H_DEFINES_STRUCT_TIMESPEC''@|$(UNISTD_H_DEFINES_STRUCT_TIMESPEC)|g' \
-	      -e '/definitions of _GL_FUNCDECL_RPL/r $(CXXDEFS_H)' \
-	      -e '/definition of _GL_ARG_NONNULL/r $(ARG_NONNULL_H)' \
-	      -e '/definition of _GL_WARN_ON_USE/r $(WARN_ON_USE_H)' \
-	      < $(srcdir)/time.in.h; \
-	} > $@-t && \
-	mv $@-t $@
-MOSTLYCLEANFILES += time.h time.h-t
-
-EXTRA_DIST += time.in.h
-
 ## end   gnulib module time
 
 ## begin gnulib module time_r
diff -rupN gnutls-3.4.0.orig/Makefile.am gnutls-3.4.0/Makefile.am
--- gnutls-3.4.0.orig/Makefile.am	2015-03-06 18:01:22.000000000 +0100
+++ gnutls-3.4.0/Makefile.am	2015-04-09 23:09:22.520335600 +0200
@@ -34,10 +34,6 @@ else
 SUBDIRS += src/gl
 endif
 
-if ENABLE_TESTS
-SUBDIRS += tests
-endif
-
 if HAVE_GUILE
 SUBDIRS += guile
 endif
diff -rupN gnutls-3.4.0.orig/src/Makefile.am gnutls-3.4.0/src/Makefile.am
--- gnutls-3.4.0.orig/src/Makefile.am	2015-01-19 18:38:09.000000000 +0100
+++ gnutls-3.4.0/src/Makefile.am	2015-04-09 23:08:36.728332600 +0200
@@ -30,12 +30,8 @@ if ENABLE_CRYWRAP
 SUBDIRS += crywrap
 endif
 
-if NEED_LIBOPTS
 LIBOPTS = libopts/libopts.la
 SUBDIRS += libopts
-else
-LIBOPTS = $(LIBOPTS_LDADD)
-endif
 
 EXTRA_DIST = args-std.def
 
diff -rupN gnutls-3.4.0.orig/doc/manpages/Makefile.am gnutls-3.4.0/doc/manpages/Makefile.am
--- gnutls-3.4.0.orig/doc/manpages/Makefile.am	2015-04-06 10:48:30.000000000 +0200
+++ gnutls-3.4.0/doc/manpages/Makefile.am	2015-04-09 23:45:26.084339500 +0200
@@ -134,11 +134,8 @@ APIMANS += gnutls_certificate_get_peers.
 APIMANS += gnutls_certificate_get_peers_subkey_id.3
 APIMANS += gnutls_certificate_get_trust_list.3
 APIMANS += gnutls_certificate_get_verify_flags.3
-APIMANS += gnutls_certificate_get_verify_flags.3
-APIMANS += gnutls_certificate_get_x509_crt.3
 APIMANS += gnutls_certificate_get_x509_crt.3
 APIMANS += gnutls_certificate_get_x509_key.3
-APIMANS += gnutls_certificate_get_x509_key.3
 APIMANS += gnutls_certificate_send_x509_rdn_sequence.3
 APIMANS += gnutls_certificate_server_set_request.3
 APIMANS += gnutls_certificate_set_dh_params.3
