--- a/setup.py.orig       2022-01-19 16:37:13.715599300 +0530
+++ b/setup.py    2022-01-19 16:37:53.038319700 +0530
@@ -1,5 +1,6 @@
 #!/usr/bin/env python

+import sys
 from platform import architecture
 from shutil import which
 from os import curdir
@@ -14,7 +15,7 @@
 library_dirs = [curdir, "build"]

 is_portable = False
-if is_win:
+if is_win and "gcc" not in sys.version.lower():
     if is_64:
         library_dirs.append(join("msvc_project", "x64", "Release"))
         lib_suffix = "lib64"


