--- pycurl-REL_7_45_1-orig/setup.py	2022-03-13 15:13:55.000000000 +0800
+++ pycurl-REL_7_45_1/setup.py	2022-10-30 01:46:17.984959200 +0800
@@ -701,7 +701,7 @@
 def get_data_files():
     # a list of tuples with (path to install to, a list of local files)
     data_files = []
-    if sys.platform == "win32":
+    if str.find(sys.version, "MSC") >= 0:
         datadir = os.path.join("doc", PACKAGE)
     else:
         datadir = os.path.join("share", "doc", PACKAGE)
