--- graphite2-1.2.1/graphite2.pc.in	2013-02-27 20:32:04.000000000 +0100
+++ graphite2-1.2.1/graphite2.pc.in	2013-05-09 22:52:58.990959979 +0200
@@ -1,7 +1,11 @@
+prefix=@prefix@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: Graphite2
 Description: Font rendering engine for Complex Scripts
-Version: ${version}
+Version: @version@
 Libs: -L${libdir} -lgraphite2
 Cflags: -I${includedir}
 
--- graphite2-1.2.1/CMakeLists.txt	2013-02-27 20:32:04.000000000 +0100
+++ graphite2-1.2.1/CMakeLists.txt	2013-06-06 21:11:28.241391998 +0200
@@ -89,9 +89,10 @@
 endif()
 
 set(version 3.0.1)
+set(prefix ${CMAKE_INSTALL_PREFIX})
 set(libdir ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 set(includedir ${CMAKE_INSTALL_PREFIX}/include)
 
-configure_file(graphite2.pc.in graphite2.pc)
+configure_file(graphite2.pc.in graphite2.pc @ONLY)
 
 install(FILES ${PROJECT_BINARY_DIR}/graphite2.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
