diff --git a/extras/Makefile.in b/extras/Makefile.in
index 857dea6..4cfc0a9 100644
--- a/extras/Makefile.in
+++ b/extras/Makefile.in
@@ -612,9 +612,9 @@ uninstall-am: uninstall-extrasDATA
 .PRECIOUS: Makefile
 
 .cob.$(COB_MODULE_EXT):
-	("$(top_builddir)/pre-inst-env" $(COBC) -m -Wall -O2 -o "$@" "$<" || \
-	 "$(top_builddir)/pre-inst-env" $(COBC) -m -Wall     -o "$@" "$<" || \
-	 "$(top_builddir)/pre-inst-env" $(COBC) -m -Wall -vv -o "$@" "$<")
+	("$(top_builddir)/pre-inst-env" $(COBC) -I$(top_srcdir) -m -Wall -O2 -o "$@" "$<" || \
+	 "$(top_builddir)/pre-inst-env" $(COBC) -I$(top_srcdir) -m -Wall     -o "$@" "$<" || \
+	 "$(top_builddir)/pre-inst-env" $(COBC) -I$(top_srcdir) -m -Wall -vv -o "$@" "$<")
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
