diff -Naur ./libcob/common.h.orig ./libcob/common.h
--- ./libcob/common.h.orig	2017-09-06 20:37:28.000000000 +0200
+++ ./libcob/common.h	2017-10-25 23:26:58.208945700 +0200
@@ -50,7 +50,7 @@
 
 #endif
 
-#if	defined(_WIN32)
+#if	defined(_WIN32) && !defined(__MINGW32__) /* hack for msys2 patch */
 
 #define	CB_FMT_LLD		"%I64d"
 #define	CB_FMT_LLU		"%I64u"
@@ -389,12 +389,12 @@
 
 /* Also OK for icc which defines __GNUC__ */
 
-#if	defined(__GNUC__) || (defined(__xlc__) && __IBMC__ >= 700)
+#if	defined(__GNUC__) && defined(__MINGW32__) /* hack for msys2 patch */
 #define	COB_A_NORETURN	__attribute__((noreturn))
-#define	COB_A_FORMAT12	__attribute__((format(printf, 1, 2)))
-#define	COB_A_FORMAT23	__attribute__((format(printf, 2, 3)))
-#define	COB_A_FORMAT34	__attribute__((format(printf, 3, 4)))
-#define	COB_A_FORMAT45	__attribute__((format(printf, 4, 5)))
+#define	COB_A_FORMAT12	__attribute__((format(__MINGW_PRINTF_FORMAT, 1, 2)))
+#define	COB_A_FORMAT23	__attribute__((format(__MINGW_PRINTF_FORMAT, 2, 3)))
+#define	COB_A_FORMAT34	__attribute__((format(__MINGW_PRINTF_FORMAT, 3, 4)))
+#define	COB_A_FORMAT45	__attribute__((format(__MINGW_PRINTF_FORMAT, 4, 5)))
 #define	DECLNORET
 #else
 #define	COB_A_NORETURN
