diff -ur a/src/include/winpty.h b/src/include/winpty.h
--- a/src/include/winpty.h	2017-05-17 14:47:20.000000000 +0800
+++ b/src/include/winpty.h	2024-01-29 00:19:51.221579600 +0800
@@ -218,6 +218,11 @@
 winpty_set_size(winpty_t *wp, int cols, int rows,
                 winpty_error_ptr_t *err /*OPTIONAL*/);
 
+/* Gets a list of processes attached to the console. */
+WINPTY_API int
+winpty_get_console_process_list(winpty_t *wp, int *processList, const int processCount,
+                                winpty_error_ptr_t *err /*OPTIONAL*/);
+
 /* Frees the winpty_t object and the OS resources contained in it.  This
  * call breaks the connection with the agent, which should then close its
  * console, terminating the processes attached to it.
diff -ur a/src/libwinpty/winpty.cc b/src/libwinpty/winpty.cc
--- a/src/libwinpty/winpty.cc	2017-05-17 14:47:20.000000000 +0800
+++ b/src/libwinpty/winpty.cc	2024-01-29 00:19:51.282214900 +0800
@@ -935,6 +935,33 @@
     } API_CATCH(FALSE)
 }
 
+WINPTY_API int
+winpty_get_console_process_list(winpty_t *wp, int *processList, const int processCount,
+                                winpty_error_ptr_t *err /*OPTIONAL*/) {
+    API_TRY {
+        ASSERT(wp != nullptr);
+        ASSERT(processList != nullptr);
+        LockGuard<Mutex> lock(wp->mutex);
+        RpcOperation rpc(*wp);
+        auto packet = newPacket();
+        packet.putInt32(AgentMsg::GetConsoleProcessList);
+        writePacket(*wp, packet);
+        auto reply = readPacket(*wp);
+
+        auto actualProcessCount = reply.getInt32();
+
+        if (actualProcessCount <= processCount) {
+            for (auto i = 0; i < actualProcessCount; i++) {
+                processList[i] = reply.getInt32();
+            }
+        }
+
+        reply.assertEof();
+        rpc.success();
+        return actualProcessCount;
+    } API_CATCH(0)
+}
+
 WINPTY_API void winpty_free(winpty_t *wp) {
     // At least in principle, CloseHandle can fail, so this deletion can
     // fail.  It won't throw an exception, but maybe there's an error that
diff -ur a/src/shared/AgentMsg.h b/src/shared/AgentMsg.h
--- a/src/shared/AgentMsg.h	2017-05-17 14:47:20.000000000 +0800
+++ b/src/shared/AgentMsg.h	2024-01-29 00:23:38.303283300 +0800
@@ -26,6 +26,7 @@
     enum Type {
         StartProcess,
         SetSize,
+        GetConsoleProcessList,
     };
 };
 
