--- uhd-4.6.0.0/host/lib/usrp_clock/octoclock/octoclock_image_loader.cpp.orig	2024-05-29 15:03:01.763825200 -0400
+++ uhd-4.6.0.0/host/lib/usrp_clock/octoclock/octoclock_image_loader.cpp	2024-05-29 15:05:32.619093800 -0400
@@ -95,7 +95,7 @@
                                      % session.image_filepath));
     }
 
-    std::string extension = fs::extension(session.image_filepath);
+    const std::string extension = fs::path(session.image_filepath).extension().string();
     if (extension == ".bin") {
         octoclock_read_bin(session);
     } else if (extension == ".hex") {
