diff -upr uhd-3.11.1.0.orig/host/lib/usrp/e300/e300_fifo_config.cpp uhd-3.11.1.0/host/lib/usrp/e300/e300_fifo_config.cpp
--- uhd-3.11.1.0.orig/host/lib/usrp/e300/e300_fifo_config.cpp	2018-05-16 22:25:48.000000000 +0300
+++ uhd-3.11.1.0/host/lib/usrp/e300/e300_fifo_config.cpp	2018-06-03 00:16:53.119018076 +0300
@@ -107,7 +107,7 @@ struct e300_fifo_poll_waiter
         boost::mutex::scoped_lock l(_mutex);
         if (_poll_claimed)
         {
-            _cond.timed_wait(l, boost::posix_time::microseconds(timeout*1000000));
+            _cond.timed_wait(l, boost::posix_time::microseconds(long(timeout*1000000)));
         }
         else
         {
