diff --git a/crates/languages/src/deno.rs b/crates/languages/src/deno.rs
index fc7fd49d0..8f49b2376 100644
--- a/crates/languages/src/deno.rs
+++ b/crates/languages/src/deno.rs
@@ -70,7 +70,7 @@ impl LspAdapter for DenoLspAdapter {
         let os = match consts::OS {
             "macos" => "apple-darwin",
             "linux" => "unknown-linux-gnu",
-            "windows" => "pc-windows-msvc",
+            "windows" => "pc-windows-gnu",
             other => bail!("Running on unsupported os: {other}"),
         };
         let asset_name = format!("deno-{}-{os}.zip", consts::ARCH);
diff --git a/crates/languages/src/rust.rs b/crates/languages/src/rust.rs
index 91965ada1..84da6e869 100644
--- a/crates/languages/src/rust.rs
+++ b/crates/languages/src/rust.rs
@@ -74,7 +74,7 @@ impl LspAdapter for RustLspAdapter {
         let os = match consts::OS {
             "macos" => "apple-darwin",
             "linux" => "unknown-linux-gnu",
-            "windows" => "pc-windows-msvc",
+            "windows" => "pc-windows-gnu",
             other => bail!("Running on unsupported os: {other}"),
         };
         let asset_name = format!("rust-analyzer-{}-{os}.gz", consts::ARCH);
diff --git a/extensions/gleam/src/gleam.rs b/extensions/gleam/src/gleam.rs
index a5b594759..0b2d2b427 100644
--- a/extensions/gleam/src/gleam.rs
+++ b/extensions/gleam/src/gleam.rs
@@ -46,7 +46,7 @@ impl GleamExtension {
             os = match platform {
                 zed::Os::Mac => "apple-darwin",
                 zed::Os::Linux => "unknown-linux-musl",
-                zed::Os::Windows => "pc-windows-msvc",
+                zed::Os::Windows => "pc-windows-gnu",
             },
         );
 
