diff --git a/.cargo/config.toml b/.cargo/config.toml
index 07cbc23195..fcc5fd7b3d 100644
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,6 +1,6 @@
 [build]
 # v0 mangling scheme provides more detailed backtraces around closures
-rustflags = ["-C", "symbol-mangling-version=v0", "--cfg", "tokio_unstable"]
+rustflags = ["-C", "symbol-mangling-version=v0", "--cfg", "tokio_unstable", "-Cforce-frame-pointers=yes", "-Clink-arg=-fuse-ld=lld"]
 
 [alias]
 xtask = "run --package xtask --"
@@ -23,6 +23,4 @@ rustflags = ["-C", "link-args=-Objc -all_load"]
 rustflags = [
     "--cfg",
     "windows_slim_errors",        # This cfg will reduce the size of `windows::core::Error` from 16 bytes to 4 bytes
-    "-C",
-    "target-feature=+crt-static", # This fixes the linking issue when compiling livekit on Windows
 ]
