--- a/crates/languages/src/rust.rs
+++ b/crates/languages/src/rust.rs
@@ -77,7 +77,7 @@ impl LspAdapter for RustLspAdapter {
         let os = match consts::OS {
             "macos" => "apple-darwin",
             "linux" => "unknown-linux-gnu",
-            "windows" => "pc-windows-msvc",
+            "windows" => "pc-windows-gnu",
             other => bail!("Running on unsupported os: {other}"),
         };
         let asset_name = format!("rust-analyzer-{}-{os}.gz", consts::ARCH);
--- a/extensions/deno/src/deno.rs
+++ b/extensions/deno/src/deno.rs
@@ -47,7 +47,7 @@ impl DenoExtension {
             os = match platform {
                 zed::Os::Mac => "apple-darwin",
                 zed::Os::Linux => "unknown-linux-gnu",
-                zed::Os::Windows => "pc-windows-msvc",
+                zed::Os::Windows => "pc-windows-gnu",
             },
         );
