--- nlopt-2.11.0/CMakeLists.txt.orig	2026-06-06 02:42:06.000000000 +0800
+++ nlopt-2.11.0/CMakeLists.txt	2026-06-07 00:57:17.311970400 +0800
@@ -230,7 +230,7 @@
   target_compile_definitions (${nlopt_lib} PRIVATE NLOPT_LUKSAN)
 endif ()
 if (M_LIBRARY)
-  target_link_libraries (${nlopt_lib} PRIVATE ${M_LIBRARY})
+  target_link_libraries (${nlopt_lib} PRIVATE m)
 endif ()
 set_target_properties (${nlopt_lib} PROPERTIES SOVERSION ${SO_MAJOR})
 set_target_properties (${nlopt_lib} PROPERTIES VERSION "${SO_MAJOR}.${SO_MINOR}.${SO_PATCH}")
@@ -309,7 +309,7 @@
   endif ()
 endif ()
 
-if (WIN32)
+if (MSVC)
   set (INSTALL_PYTHON_DIR Lib/site-packages CACHE STRING "site packages dir")
 else ()
   set (INSTALL_PYTHON_DIR ${NLOPT_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages CACHE STRING "site packages dir")
