--- nlopt-2.10.1/CMakeLists.txt.orig	2026-02-09 21:30:18.000000000 +0100
+++ nlopt-2.10.1/CMakeLists.txt	2026-05-03 10:23:48.994075700 +0200
@@ -230,7 +230,7 @@
   target_compile_definitions (${nlopt_lib} PRIVATE NLOPT_LUKSAN)
 endif ()
 if (M_LIBRARY)
-  target_link_libraries (${nlopt_lib} ${M_LIBRARY})
+  target_link_libraries (${nlopt_lib} m)
 endif ()
 set_target_properties (${nlopt_lib} PROPERTIES SOVERSION ${SO_MAJOR})
 set_target_properties (${nlopt_lib} PROPERTIES VERSION "${SO_MAJOR}.${SO_MINOR}.${SO_PATCH}")
