--- nlopt-2.11.0/test/CMakeLists.txt.orig	2026-06-06 02:42:06.000000000 +0800
+++ nlopt-2.11.0/test/CMakeLists.txt	2026-06-07 00:42:07.062599300 +0800
@@ -8,7 +8,7 @@
     target_include_directories (${test_name} PRIVATE ${NLOPT_PRIVATE_INCLUDE_DIRS})
     foreach(arg IN ITEMS ${ARGN})
       add_test (NAME check_${test_name}_${arg} COMMAND ${test_name} ${arg})
-      if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+      if (MSVC)
         set_tests_properties (check_${test_name}_${arg}
           PROPERTIES ENVIRONMENT "PATH=${PROJECT_BINARY_DIR}\\${CMAKE_BUILD_TYPE};$ENV{PATH}")  # to load dll
       endif ()
@@ -51,7 +51,7 @@
     endif ()
     if (enable_)
       add_test (NAME testopt_algo${algo_index}_obj${obj_index} COMMAND testopt -r 0 -a ${algo_index} -o ${obj_index})
-      if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+      if (MSVC)
         set_tests_properties (testopt_algo${algo_index}_obj${obj_index} PROPERTIES ENVIRONMENT "PATH=${PROJECT_BINARY_DIR}\\${CMAKE_BUILD_TYPE};$ENV{PATH}")  # to load dll
       endif ()
       # Check if LUKSAN targets are available.
