From 6e73246acdd7937c3c4724b8c0005d431163c427 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 17 Jun 2021 18:52:31 +0530
Subject: [PATCH 059/N] pathlib path sep

---
 Lib/pathlib.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/pathlib.py b/Lib/pathlib.py
index 9d40d88..0b86bc0 100644
--- a/Lib/pathlib.py
+++ b/Lib/pathlib.py
@@ -115,6 +115,8 @@ class _WindowsFlavour(_Flavour):
 
     sep = '\\'
     altsep = '/'
+    if 'MSYSTEM' in os.environ:
+        sep, altsep = altsep, sep
     has_drv = True
     pathmod = ntpath
 
