diff -ur a/src/add_reed_solomon_checkwords.cpp b/src/add_reed_solomon_checkwords.cpp
--- a/src/add_reed_solomon_checkwords.cpp	2009-07-24 19:15:30.000000000 +0300
+++ b/src/add_reed_solomon_checkwords.cpp	2016-12-21 11:51:30.234404000 +0200
@@ -1,6 +1,6 @@
 #include <cstddef>
 #include <cassert>
-#include "base\rs_encode.h"
+#include "base/rs_encode.h"
 #include "add_reed_solomon_checkwords.h"
 
 
diff -ur a/src/common_definitions.h b/src/common_definitions.h
--- a/src/common_definitions.h	2012-02-17 23:57:40.000000000 +0200
+++ b/src/common_definitions.h	2016-12-21 11:51:50.023331600 +0200
@@ -1,6 +1,6 @@
 #pragma once
 #include <vector>
-#include "base\matrix.h"
+#include "base/matrix.h"
 
 
 typedef unsigned char bit_type;
diff -ur a/src/generate_bit_stream.h b/src/generate_bit_stream.h
--- a/src/generate_bit_stream.h	2012-03-28 18:34:30.000000000 +0300
+++ b/src/generate_bit_stream.h	2016-12-21 11:52:03.079031200 +0200
@@ -4,7 +4,7 @@
 #include <vector>
 #include <iterator>
 #include "common_definitions.h"
-#include "base\bin_dec_conversion.h"
+#include "base/bin_dec_conversion.h"
 
 
 template <typename DataIterator>
diff -ur a/src/generate_codeword_stream.cpp b/src/generate_codeword_stream.cpp
--- a/src/generate_codeword_stream.cpp	2012-03-28 18:34:46.000000000 +0300
+++ b/src/generate_codeword_stream.cpp	2016-12-21 11:52:16.512717500 +0200
@@ -1,7 +1,7 @@
 #include <cstddef>
 #include <cassert>
 #include <algorithm>
-#include "base\bin_dec_conversion.h"
+#include "base/bin_dec_conversion.h"
 #include "generate_codeword_stream.h"
 
 
diff -ur a/src/generate_data_message.cpp b/src/generate_data_message.cpp
--- a/src/generate_data_message.cpp	2012-03-28 18:34:12.000000000 +0300
+++ b/src/generate_data_message.cpp	2016-12-21 11:52:27.807189900 +0200
@@ -1,7 +1,7 @@
 #include <cstddef>
 #include <cassert>
 #include <algorithm>
-#include "base\bin_dec_conversion.h"
+#include "base/bin_dec_conversion.h"
 #include "add_reed_solomon_checkwords.h"
 #include "generate_data_message.h"
 
diff -ur a/src/generate_mode_message.cpp b/src/generate_mode_message.cpp
--- a/src/generate_mode_message.cpp	2012-03-28 18:34:04.000000000 +0300
+++ b/src/generate_mode_message.cpp	2016-12-21 11:52:39.632065700 +0200
@@ -1,6 +1,6 @@
 #include <cstddef>
 #include <cassert>
-#include "base\bin_dec_conversion.h"
+#include "base/bin_dec_conversion.h"
 #include "add_reed_solomon_checkwords.h"
 #include "generate_mode_message.h"
 
