diff -ur a/src/generate_bit_stream.h b/src/generate_bit_stream.h
--- a/src/generate_bit_stream.h	2016-12-21 12:17:04.492705000 +0200
+++ b/src/generate_bit_stream.h	2016-12-21 12:17:58.442851300 +0200
@@ -33,7 +33,7 @@
 
 	// Генерация потока бит из входных данных
 	typedef std::vector<
-		std::iterator_traits<DataIterator>::value_type> buffer_type;
+		typename std::iterator_traits<DataIterator>::value_type> buffer_type;
 	enum {bits_on_small_size = 5};
 	enum {bits_on_big_size = 11};
 
@@ -74,8 +74,8 @@
 		}
 
 		// Генерация данных
-		buffer_type::const_iterator b = buffer.begin();
-		buffer_type::const_iterator e = buffer.end();
+		typename buffer_type::const_iterator b = buffer.begin();
+		typename buffer_type::const_iterator e = buffer.end();
 		while (b != e)
 		{
 			enum {bits_on_byte = 8};
