--- a/src/io/stream/inkscapestream.h
+++ b/src/io/stream/inkscapestream.h
@@ -12,6 +12,10 @@
 
 #include <cstdio>
 #include <glibmm/ustring.h>
+
+#ifdef printf
+#undef printf
+#endif
 
 namespace Inkscape
 {
--- a/src/inkscape-application.cpp
+++ b/src/inkscape-application.cpp
@@ -41,6 +41,7 @@
 #include <cerrno>  // History file
 #include <regex>
 #include <numeric>
+#include <unistd.h>
 
 // checking if dithering is supported
 #ifdef  WITH_PATCHED_CAIRO
