--- inkscape-1.0.1/CMakeLists.txt.orig	2020-10-28 10:36:52.521819700 +0300
+++ inkscape-1.0.1/CMakeLists.txt	2020-10-28 10:38:53.329212100 +0300
@@ -207,10 +207,6 @@
 # -----------------------------------------------------------------------------
 add_subdirectory(share)
 
-if(WIN32)
-  include(CMakeScripts/InstallMSYS2.cmake)
-endif()
-
 # -----------------------------------------------------------------------------
 # Uninstall Target
 # -----------------------------------------------------------------------------
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,7 +133,7 @@
 mark_as_advanced(WITH_FUZZ)
 option(WITH_MANPAGE_COMPRESSION "gzips manpages if gzip is available" ON)
 mark_as_advanced(WITH_MANPAGE_COMPRESSION)
-cmake_dependent_option(WITH_INTERNAL_CAIRO "Use a patched version of cairo" OFF "UNIX" ON)
+cmake_dependent_option(WITH_INTERNAL_CAIRO "Use a patched version of cairo" OFF "MINGW" ON)
 
 option(ENABLE_BINRELOC "Enable relocatable binaries" OFF)
 
