--- inkscape-0.48.4-orig/configure.ac.orig	2012-12-15 20:50:19.205918100 +0400
+++ inkscape-0.48.4-orig/configure.ac	2014-03-02 10:33:02.384000000 +0400
@@ -780,7 +780,7 @@
 fi
 
 dnl Shouldn't we test for libpng and libz?
-INKSCAPE_LIBS="$INKSCAPE_LIBS -lpng -lz -lxml2 -ldl"
+INKSCAPE_LIBS="$INKSCAPE_LIBS -lpng -lz -lxml2"
 if test "x$openmp_ok" = "xyes"; then
 	INKSCAPE_LIBS="$INKSCAPE_LIBS -lgomp"
 fi
