--- inkscape-1.0.1/CMakeScripts/ConfigPaths.cmake.orig	2020-10-28 11:06:48.890928800 +0300
+++ inkscape-1.0.1/CMakeScripts/ConfigPaths.cmake	2020-10-28 11:06:55.587717400 +0300
@@ -1,6 +1,6 @@
 message(STATUS "Creating build files in: ${CMAKE_CURRENT_BINARY_DIR}")
 
-if(WIN32)
+if(MSVC)
   if(${CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT})
     set(CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/inkscape"
         CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
