--- inkscape-0.48.4-orig/src/extension/internal/pdfinput/pdf-parser.h.orig	2014-03-02 17:32:26.275800000 +0400
+++ inkscape-0.48.4-orig/src/extension/internal/pdfinput/pdf-parser.h	2014-03-02 17:32:33.233400000 +0400
@@ -27,7 +27,7 @@
 using Inkscape::Extension::Internal::SvgBuilder;
 
 #include "goo/gtypes.h"
-#include "Object.h"
+#include <poppler/Object.h>
 
 class GooString;
 class XRef;
--- inkscape-0.48.4-orig/src/extension/internal/pdfinput/pdf-parser.cpp.orig	2012-12-13 21:00:46.726633000 +0400
+++ inkscape-0.48.4-orig/src/extension/internal/pdfinput/pdf-parser.cpp	2014-03-02 17:31:00.838000000 +0400
@@ -36,20 +36,20 @@
 #include "glib/poppler-features.h"
 #include "goo/gmem.h"
 #include "goo/GooString.h"
-#include "GlobalParams.h"
-#include "CharTypes.h"
-#include "Object.h"
-#include "Array.h"
-#include "Dict.h"
-#include "Stream.h"
-#include "Lexer.h"
-#include "Parser.h"
-#include "GfxFont.h"
-#include "GfxState.h"
-#include "OutputDev.h"
-#include "Page.h"
-#include "Annot.h"
-#include "Error.h"
+#include <poppler/GlobalParams.h>
+#include <poppler/CharTypes.h>
+#include <poppler/Object.h>
+#include <poppler/Array.h>
+#include <poppler/Dict.h>
+#include <poppler/Stream.h>
+#include <poppler/Lexer.h>
+#include <poppler/Parser.h>
+#include <poppler/GfxFont.h>
+#include <poppler/GfxState.h>
+#include <poppler/OutputDev.h>
+#include <poppler/Page.h>
+#include <poppler/Annot.h>
+#include <poppler/Error.h>
 
 // the MSVC math.h doesn't define this
 #ifndef M_PI
