--- inkscape-0.92.3/src/libavoid/connector.cpp.orig	2017-09-06 11:42:09.353060000 +0300
+++ inkscape-0.92.3/src/libavoid/connector.cpp	2017-09-06 11:42:40.064816600 +0300
@@ -885,7 +885,7 @@
             {
                 // Check for consecutive points on opposite 
                 // corners of two touching shapes.
-                COLA_ASSERT(abs((long)i->pathNext->id.objID - i->id.objID) != 2);
+                COLA_ASSERT(abs((long)(i->pathNext->id.objID - i->id.objID)) != 2);
             }
         }
     }
