--- inkscape-0.92.1/src/libavoid/connector.cpp.orig	2017-09-06 11:42:09.353060000 +0300
+++ inkscape-0.92.1/src/libavoid/connector.cpp	2017-09-06 11:42:40.064816600 +0300
@@ -885,7 +885,7 @@
             {
                 // Check for consecutive points on opposite 
                 // corners of two touching shapes.
-                COLA_ASSERT(abs(i->pathNext->id.objID - i->id.objID) != 2);
+                COLA_ASSERT(abs((int)(i->pathNext->id.objID - i->id.objID)) != 2);
             }
         }
     }
--- inkscape-0.92.1/src/ui/tools/flood-tool.cpp.orig	2017-09-14 12:35:28.815818000 +0300
+++ inkscape-0.92.1/src/ui/tools/flood-tool.cpp	2017-09-14 12:36:29.374736400 +0300
@@ -232,13 +232,13 @@
     
     switch (method) {
         case FLOOD_CHANNELS_ALPHA:
-            return abs(static_cast<int>(ac) - ao) <= threshold;
+            return abs(static_cast<int>(ac - ao)) <= threshold;
         case FLOOD_CHANNELS_R:
-            return abs(static_cast<int>(ac ? unpremul_alpha(rc, ac) : 0) - (ao ? unpremul_alpha(ro, ao) : 0)) <= threshold;
+            return abs(static_cast<int>((ac ? unpremul_alpha(rc, ac) : 0) - (ao ? unpremul_alpha(ro, ao) : 0))) <= threshold;
         case FLOOD_CHANNELS_G:
-            return abs(static_cast<int>(ac ? unpremul_alpha(gc, ac) : 0) - (ao ? unpremul_alpha(go, ao) : 0)) <= threshold;
+            return abs(static_cast<int>((ac ? unpremul_alpha(gc, ac) : 0) - (ao ? unpremul_alpha(go, ao) : 0))) <= threshold;
         case FLOOD_CHANNELS_B:
-            return abs(static_cast<int>(ac ? unpremul_alpha(bc, ac) : 0) - (ao ? unpremul_alpha(bo, ao) : 0)) <= threshold;
+            return abs(static_cast<int>((ac ? unpremul_alpha(bc, ac) : 0) - (ao ? unpremul_alpha(bo, ao) : 0))) <= threshold;
         case FLOOD_CHANNELS_RGB:
             guint32 amc, rmc, bmc, gmc;
             //amc = 255*255 - (255-ac)*(255-ad); amc = (amc + 127) / 255;
@@ -248,9 +248,9 @@
             gmc = (255-ac)*gd + 255*gc; gmc = (gmc + 127) / 255;
             bmc = (255-ac)*bd + 255*bc; bmc = (bmc + 127) / 255;
 
-            diff += abs(static_cast<int>(amc ? unpremul_alpha(rmc, amc) : 0) - (amop ? unpremul_alpha(rmop, amop) : 0));
-            diff += abs(static_cast<int>(amc ? unpremul_alpha(gmc, amc) : 0) - (amop ? unpremul_alpha(gmop, amop) : 0));
-            diff += abs(static_cast<int>(amc ? unpremul_alpha(bmc, amc) : 0) - (amop ? unpremul_alpha(bmop, amop) : 0));
+            diff += abs(static_cast<int>((amc ? unpremul_alpha(rmc, amc) : 0) - (amop ? unpremul_alpha(rmop, amop) : 0)));
+            diff += abs(static_cast<int>((amc ? unpremul_alpha(gmc, amc) : 0) - (amop ? unpremul_alpha(gmop, amop) : 0)));
+            diff += abs(static_cast<int>((amc ? unpremul_alpha(bmc, amc) : 0) - (amop ? unpremul_alpha(bmop, amop) : 0)));
             return ((diff / 3) <= ((threshold * 3) / 4));
         
         case FLOOD_CHANNELS_H:
