--- inkscape-0.92.0/src/main.cpp.orig	2017-01-02 12:44:59.000000000 +0300
+++ inkscape-0.92.0/src/main.cpp	2017-01-16 12:53:06.744038300 +0300
@@ -546,33 +546,31 @@
     gchar const *path = g_getenv("PATH");
     gchar const *pythonpath = g_getenv("PYTHONPATH");
 
-    gchar *python = g_build_filename(exe, "python", NULL);
-    gchar *scripts = g_build_filename(exe, "python", "Scripts", NULL);
-    gchar *perl = g_build_filename(exe, "python", NULL);
-    gchar *pythonlib = g_build_filename(exe, "python", "Lib", NULL);
-    gchar *pythondll = g_build_filename(exe, "python", "DLLs", NULL);
+    gchar *prefixpath = g_strdup(win32_getPrefix().data());
+    gchar *pythonlib = g_build_filename(prefixpath, "lib", "python2.7", NULL);
+    gchar *pythondll = g_build_filename(prefixpath, "lib", "python2.7", "lib-dynload", "DLLs", NULL);
 
     // Python 2.x needs short paths in PYTHONPATH.
     // Otherwise it doesn't work when Inkscape is installed in Unicode directories.
     // g_win32_locale_filename_from_utf8 is the GLib wrapper for GetShortPathName.
     // Remove this once we move to Python 3.0.
-    gchar *python_s = g_win32_locale_filename_from_utf8(python);
+    gchar *python_s = g_win32_locale_filename_from_utf8(exe);
     gchar *pythonlib_s = g_win32_locale_filename_from_utf8(pythonlib);
     gchar *pythondll_s = g_win32_locale_filename_from_utf8(pythondll);
 
     gchar *new_path;
     gchar *new_pythonpath;
     if (path) {
-        new_path = g_strdup_printf("%s;%s;%s;%s;%s", exe, python, scripts, perl, path);
+        new_path = g_strdup_printf("%s;%s", exe, path);
     } else {
-        new_path = g_strdup_printf("%s;%s;%s;%s", exe, python, scripts, perl);
+        new_path = g_strdup_printf("%s", exe);
     }
     if (pythonpath) {
-        new_pythonpath = g_strdup_printf("%s;%s;%s;%s",
-             python_s, pythonlib_s, pythondll_s, pythonpath);
+        new_pythonpath = g_strdup_printf("%s;%s;%s;%s;%s",
+             exe, python_s, pythonlib_s, pythondll_s, pythonpath);
     } else {
-        new_pythonpath = g_strdup_printf("%s;%s;%s",
-            python_s, pythonlib_s, pythondll_s);
+        new_pythonpath = g_strdup_printf("%s;%s;%s;%s",
+            exe, python_s, pythonlib_s, pythondll_s);
     }
 
     g_setenv("PATH", new_path, TRUE);
@@ -597,9 +595,7 @@
     // prevent "please insert disk" messages. fixes bug #950781
     SetErrorMode(SEM_FAILCRITICALERRORS | SEM_NOGPFAULTERRORBOX | SEM_NOOPENFILEERRORBOX);
 
-    g_free(python);
-    g_free(scripts);
-    g_free(perl);
+    g_free(prefixpath);
     g_free(pythonlib);
     g_free(pythondll);
 
@@ -673,19 +669,21 @@
     // obtain short path to executable dir and pass it
     // to bindtextdomain (it doesn't understand UTF-8)
     gchar *shortexedir = g_win32_locale_filename_from_utf8(exedir);
-    gchar *localepath = g_build_filename(shortexedir, PACKAGE_LOCALE_DIR, NULL);
+    gchar *localepath = g_strdup(win32_getLocaleDir().data());
     bindtextdomain(GETTEXT_PACKAGE, localepath);
     g_free(shortexedir);
     g_free(localepath);
 # endif
     g_free(exedir);
 
+#ifndef __MINGW32__
     // Don't touch the registry (works fine without it) for Inkscape Portable
     gchar const *val = g_getenv("INKSCAPE_PORTABLE_PROFILE_DIR");
     if (!val) {
         RegistryTool rt;
         rt.setPathInfo();
     }
+#endif
 #elif defined(ENABLE_NLS)
 # ifdef ENABLE_BINRELOC
     bindtextdomain(GETTEXT_PACKAGE, BR_LOCALEDIR(""));
--- inkscape-0.92.0/src/prefix.h.orig	2014-03-04 07:07:25.022000000 +0400
+++ inkscape-0.92.0/src/prefix.h	2014-03-04 09:31:46.705000000 +0400
@@ -124,6 +124,8 @@
 #include <glibmm/ustring.h>
 
 Glib::ustring win32_getExePath();
+Glib::ustring win32_getPrefix();
+Glib::ustring win32_getLocaleDir();
 char *win32_relative_path(const char *childPath);
 
 #define WIN32_DATADIR(suffix) (win32_relative_path(suffix))
--- inkscape-0.92.0/src/prefix.cpp.orig	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.92.0/src/prefix.cpp	2014-03-04 09:31:35.395000000 +0400
@@ -447,23 +447,49 @@
     return ret;
 }
 
+/**
+ * Return the relocatable version of the prefix,
+ * probably c:\inkscape 
+ */
+Glib::ustring win32_getPrefix()
+{
+    gchar *dir = g_strdup(win32_getExePath().data());
+    gchar *dir2 = g_path_get_dirname(dir);
+    Glib::ustring ret = dir2;
+    g_free(dir);
+    g_free(dir2);
+    return ret;
+}
 
 /**
  * Return the relocatable version of the datadir,
- * probably c:\inkscape 
+ * probably c:\inkscape\share
  */
 static Glib::ustring win32_getDataDir()
 {
-    Glib::ustring dir = win32_getExePath();
-    if (INKSCAPE_DATADIR && *INKSCAPE_DATADIR &&
-        strcmp(INKSCAPE_DATADIR, ".") != 0)
-        {
-        dir += "\\";
-        dir += INKSCAPE_DATADIR;
-        }
+    Glib::ustring dir = win32_getPrefix();
+    dir += "\\share";
     return dir;
 }
 
+/**
+ * Return the relocatable version of the locale,
+ * probably c:\inkscape\share\locale
+ */
+Glib::ustring win32_getLocaleDir()
+{
+    gchar *data_dir = g_strdup(win32_getDataDir().data());
+    gchar *dir = g_build_filename(data_dir, PACKAGE_LOCALE_DIR, NULL);
+    Glib::ustring ret = dir;
+    g_free(data_dir);
+    g_free(dir);
+    return ret;
+}
+
+/**
+ * Return the relocatable version of the locale,
+ * probably c:\inkscape\share\locale
+ */
 static Glib::ustring win32_getResourcePath(const Glib::ustring &childPath)
 {
     Glib::ustring dir = win32_getDataDir();
--- inkscape-0.92.0/src/path-prefix.h.orig	2014-03-04 07:02:21.005800000 +0400
+++ inkscape-0.92.0/src/path-prefix.h	2014-03-04 07:04:57.427800000 +0400
@@ -43,22 +43,22 @@
 #else
 #  ifdef WIN32
 #    define INKSCAPE_APPICONDIR   WIN32_DATADIR("pixmaps")
-#    define INKSCAPE_ATTRRELDIR   WIN32_DATADIR("share\\attributes")
-#    define INKSCAPE_BINDDIR      WIN32_DATADIR("share\\bind")
-#    define INKSCAPE_EXAMPLESDIR  WIN32_DATADIR("share\\examples")
-#    define INKSCAPE_EXTENSIONDIR WIN32_DATADIR("share\\extensions")
-#    define INKSCAPE_FILTERDIR    WIN32_DATADIR("share\\filters")
-#    define INKSCAPE_GRADIENTSDIR WIN32_DATADIR("share\\gradients")
-#    define INKSCAPE_KEYSDIR      WIN32_DATADIR("share\\keys")
-#    define INKSCAPE_PIXMAPDIR    WIN32_DATADIR("share\\icons")
-#    define INKSCAPE_MARKERSDIR   WIN32_DATADIR("share\\markers")
-#    define INKSCAPE_PALETTESDIR  WIN32_DATADIR("share\\palettes")
-#    define INKSCAPE_PATTERNSDIR  WIN32_DATADIR("share\\patterns")
-#    define INKSCAPE_SCREENSDIR   WIN32_DATADIR("share\\screens")
-#    define INKSCAPE_SYMBOLSDIR   WIN32_DATADIR("share\\symbols")
-#    define INKSCAPE_TUTORIALSDIR WIN32_DATADIR("share\\tutorials")
-#    define INKSCAPE_TEMPLATESDIR WIN32_DATADIR("share\\templates")
-#    define INKSCAPE_UIDIR        WIN32_DATADIR("share\\ui")
+#    define INKSCAPE_ATTRRELDIR   WIN32_DATADIR("inkscape\\attributes")
+#    define INKSCAPE_BINDDIR      WIN32_DATADIR("inkscape\\bind")
+#    define INKSCAPE_EXAMPLESDIR  WIN32_DATADIR("inkscape\\examples")
+#    define INKSCAPE_EXTENSIONDIR WIN32_DATADIR("inkscape\\extensions")
+#    define INKSCAPE_FILTERDIR    WIN32_DATADIR("inkscape\\filters")
+#    define INKSCAPE_GRADIENTSDIR WIN32_DATADIR("inkscape\\gradients")
+#    define INKSCAPE_KEYSDIR      WIN32_DATADIR("inkscape\\keys")
+#    define INKSCAPE_PIXMAPDIR    WIN32_DATADIR("inkscape\\icons")
+#    define INKSCAPE_MARKERSDIR   WIN32_DATADIR("inkscape\\markers")
+#    define INKSCAPE_PALETTESDIR  WIN32_DATADIR("inkscape\\palettes")
+#    define INKSCAPE_PATTERNSDIR  WIN32_DATADIR("inkscape\\patterns")
+#    define INKSCAPE_SCREENSDIR   WIN32_DATADIR("inkscape\\screens")
+#    define INKSCAPE_SYMBOLSDIR   WIN32_DATADIR("inkscape\\symbols")
+#    define INKSCAPE_TUTORIALSDIR WIN32_DATADIR("inkscape\\tutorials")
+#    define INKSCAPE_TEMPLATESDIR WIN32_DATADIR("inkscape\\templates")
+#    define INKSCAPE_UIDIR        WIN32_DATADIR("inkscape\\ui")
 //CREATE V0.1  WIN32 support
 #    define CREATE_GRADIENTSDIR   WIN32_DATADIR("create\\gradients\\gimp")
 #    define CREATE_PALETTESDIR    WIN32_DATADIR("create\\swatches")
