--- inkscape-0.48.4-orig/src/color-profile.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/color-profile.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -13,6 +13,7 @@
 #include <glibmm/checksum.h>
 #include <glib/gstdio.h>
 #include <fcntl.h>
+#include <glibmm.h>
 #include <glib/gi18n.h>
 
 #ifdef DEBUG_LCMS
@@ -840,7 +841,7 @@
 static bool isIccFile( gchar const *filepath )
 {
     bool isIccFile = false;
-    struct stat st;
+    GStatBuf st;
     if ( g_stat(filepath, &st) == 0 && (st.st_size > 128) ) {
         //0-3 == size
         //36-39 == 'acsp' 0x61637370
--- inkscape-0.48.4-orig/src/display/cairo-utils.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/display/cairo-utils.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -291,7 +291,7 @@
     if (!g_file_test(fn.c_str(), G_FILE_TEST_EXISTS)) { 
         return NULL;
     }
-    struct stat stdir;
+    GStatBuf stdir;
     int val = g_stat(fn.c_str(), &stdir);
     if (val == 0 && stdir.st_mode & S_IFDIR){
         return NULL;
--- inkscape-0.48.4-orig/src/inkscape.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/inkscape.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -339,7 +339,7 @@
         if (doc->isModifiedSinceSave()) {
             gchar *oldest_autosave = 0;
             const gchar  *filename = 0;
-            struct stat sb;
+            GStatBuf sb;
             time_t min_time = 0;
             gint count = 0;
 
--- inkscape-0.48.4-orig/src/sp-image.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/sp-image.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -804,7 +804,7 @@
     if ( image->href && image->pixbuf && image->pixbuf->modificationTime()) {
         // It *might* change
 
-        struct stat st;
+        GStatBuf st;
         memset(&st, 0, sizeof(st));
         int val = 0;
         if (g_file_test (image->pixbuf->originalPath().c_str(), G_FILE_TEST_EXISTS)){ 
--- inkscape-0.48.4-orig/src/io/sys.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/io/sys.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -226,7 +226,7 @@
             filename = g_filename_from_utf8 ( utf8name, -1, NULL, NULL, NULL );
         }
         if ( filename ) {
-            struct stat st;
+            GStatBuf st;
             if (g_file_test (filename, G_FILE_TEST_EXISTS)){ 
                 if (g_lstat (filename, &st) == 0) {
                     success = ((st.st_mode & S_IWRITE) != 0);
--- inkscape-0.48.4-orig/src/ui/dialog/filedialogimpl-gtkmm.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/ui/dialog/filedialogimpl-gtkmm.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -513,7 +513,7 @@
         Glib::ustring fileNameUtf8 = Glib::filename_to_utf8(fileName);
         gchar *fName = const_cast<gchar *>(
             fileNameUtf8.c_str()); // const-cast probably not necessary? (not necessary on Windows version of stat())
-        struct stat info;
+        GStatBuf info;
         if (g_stat(fName, &info)) // stat returns 0 upon success
         {
             g_warning("SVGPreview::set() : %s : %s", fName, strerror(errno));
