diff -Naur inkscape-0.48.4-orig/share/extensions/guides_creator.py inkscape-0.48.4/share/extensions/guides_creator.py
--- inkscape-0.48.4-orig/share/extensions/guides_creator.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/guides_creator.py	2014-03-04 23:34:38.522000000 +0400
@@ -41,7 +41,7 @@
 # These two lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-sys.path.append('/usr/share/inkscape/extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_autoTexts.py inkscape-0.48.4/share/extensions/jessyInk_autoTexts.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_autoTexts.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_autoTexts.py	2014-03-04 23:26:49.586000000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_effects.py inkscape-0.48.4/share/extensions/jessyInk_effects.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_effects.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_effects.py	2014-03-04 23:27:10.115600000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_export.py inkscape-0.48.4/share/extensions/jessyInk_export.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_export.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_export.py	2014-03-04 23:33:23.002400000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 import inkex, os.path
 import subprocess
@@ -158,9 +153,8 @@
 	# Function to try and find the correct command to invoke Inkscape.
 	def findInkscapeCommand(self):
 		commands = []
-		commands.append("inkscape")
-		commands.append("C:\Program Files\Inkscape\inkscape.exe")
-		commands.append("/Applications/Inkscape.app/Contents/Resources/bin/inkscape")
+		commands.append("inkscape.exe")
+		commands.append(sys.path.append(sys.prefix+os.sep+'bin'+os.sep+'inkscape.exe'))
 
 		for command in commands:
 			proc = subprocess.Popen([command + " --without-gui --version"], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_install.py inkscape-0.48.4/share/extensions/jessyInk_install.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_install.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_install.py	2014-03-04 23:27:44.170400000 +0400
@@ -18,12 +18,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_keyBindings.py inkscape-0.48.4/share/extensions/jessyInk_keyBindings.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_keyBindings.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_keyBindings.py	2014-03-04 23:28:06.930800000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_masterSlide.py inkscape-0.48.4/share/extensions/jessyInk_masterSlide.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_masterSlide.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_masterSlide.py	2014-03-04 23:28:20.300000000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_mouseHandler.py inkscape-0.48.4/share/extensions/jessyInk_mouseHandler.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_mouseHandler.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_mouseHandler.py	2014-03-04 23:28:40.704800000 +0400
@@ -18,12 +18,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_summary.py inkscape-0.48.4/share/extensions/jessyInk_summary.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_summary.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_summary.py	2014-03-04 23:28:59.315600000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_transitions.py inkscape-0.48.4/share/extensions/jessyInk_transitions.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_transitions.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_transitions.py	2014-03-04 23:29:24.603200000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_uninstall.py inkscape-0.48.4/share/extensions/jessyInk_uninstall.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_uninstall.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_uninstall.py	2014-03-04 23:29:37.192400000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_video.py inkscape-0.48.4/share/extensions/jessyInk_video.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_video.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_video.py	2014-03-04 23:29:53.853200000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/jessyInk_view.py inkscape-0.48.4/share/extensions/jessyInk_view.py
--- inkscape-0.48.4-orig/share/extensions/jessyInk_view.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/jessyInk_view.py	2014-03-04 23:30:04.976000000 +0400
@@ -16,12 +16,7 @@
 # These lines are only needed if you don't put the script directly into
 # the installation directory
 import sys
-# Unix
-sys.path.append('/usr/share/inkscape/extensions')
-# OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
-# Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
diff -Naur inkscape-0.48.4-orig/share/extensions/pixelsnap.py inkscape-0.48.4/share/extensions/pixelsnap.py
--- inkscape-0.48.4-orig/share/extensions/pixelsnap.py	2011-07-08 22:25:09.468790000 +0400
+++ inkscape-0.48.4/share/extensions/pixelsnap.py	2014-03-04 23:30:55.754000000 +0400
@@ -74,6 +74,7 @@
 #sys.path += ['/usr/share/inkscape/extensions']                     # If you're using a standard Linux installation
 #sys.path += ['/usr/local/share/inkscape/extensions']               # If you're using a custom Linux installation
 #sys.path += ['C:\\Program Files\\Inkscape\\share\\extensions']     # If you're using a standard Windows installation
+sys.path.append(sys.prefix+os.sep+'share'+os.sep+'inkscape'+os.sep+'extensions')
 
 try:
     import inkex
