--- inkscape-0.48.4-orig/src/color-profile.cpp	2012-02-19 05:41:36.476182000 +0400
+++ inkscape-0.48.4/src/color-profile.cpp	2014-03-02 00:11:22.365600000 +0400
@@ -6,6 +6,7 @@
 
 #include <glib/gstdio.h>
 #include <sys/fcntl.h>
+#include <glibmm.h>
 #include <gdkmm/color.h>
 #include <glib/gi18n.h>
 
@@ -840,7 +841,7 @@
 static bool isIccFile( gchar const *filepath )
 {
     bool isIccFile = false;
-    struct stat st;
+    GStatBuf st;
     if ( g_stat(filepath, &st) == 0 && (st.st_size > 128) ) {
         //0-3 == size
         //36-39 == 'acsp' 0x61637370
