--- blender/build_files/cmake/Modules/FindGLEW.cmake.orig	2014-07-28 22:50:06.466400000 +0400
+++ blender/build_files/cmake/Modules/FindGLEW.cmake	2014-07-28 22:50:22.581200000 +0400
@@ -43,7 +43,7 @@
 
 FIND_LIBRARY(GLEW_LIBRARY
   NAMES
-    GLEW
+    GLEW glew32
   HINTS
     ${_glew_SEARCH_DIRS}
   PATH_SUFFIXES
@@ -53,7 +53,7 @@
 
 FIND_LIBRARY(GLEW_MX_LIBRARY
   NAMES
-    GLEWmx
+    GLEWmx glew32mx
   HINTS
     ${_glew_SEARCH_DIRS}
   PATH_SUFFIXES
