diff -Naur blender/CMakeLists.txt blender-pp/CMakeLists.txt
--- blender/CMakeLists.txt	2014-01-14 08:17:56.083000000 +0400
+++ blender-pp/CMakeLists.txt	2014-01-14 08:16:01.706800000 +0400
@@ -177,6 +177,8 @@
 	if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
 		option(WITH_GHOST_XDND   "Enable drag'n'drop support on Linux using XDND protocol" ON)
 	endif()
+elseif(MINGW)
+	option(WITH_SYSTEM_GLEW "Use GLEW OpenGL wrapper library provided by the operating system" ON)
 else()
 	# not an option for other OS's
 	set(WITH_SYSTEM_GLEW OFF)
@@ -1281,7 +1283,10 @@
 		include(CheckCSourceCompiles)
 		# Setup 64bit and 64bit windows systems
 		CHECK_C_SOURCE_COMPILES("
-			#ifndef __MINGW64__
+			#ifdef __MINGW32__
+			#include <_mingw.h>
+			#endif
+			#ifndef __MINGW64_VERSION_MAJOR
 			#error
 			#endif
 			int main(void) { return 0; }
@@ -1290,10 +1295,10 @@
 		
 		if(WITH_MINGW64)
 			message(STATUS "Compiling for 64 bit with MinGW-w64.")
-			set(LIBDIR ${CMAKE_SOURCE_DIR}/../lib/mingw64)
+			set(LIBDIR ${MINGW_LIBS})
 		else()
 			message(STATUS "Compiling for 32 bit with MinGW-w32.")
-			set(LIBDIR ${CMAKE_SOURCE_DIR}/../lib/mingw32)
+			set(LIBDIR ${MINGW_LIBS})
 			
 			if(WITH_RAYOPTIMIZATION)
 				message(WARNING "MinGW-w32 is known to be unstable with 'WITH_RAYOPTIMIZATION' option enabled.")
@@ -1316,7 +1321,7 @@
 
 		add_definitions(-DFREE_WINDOWS)
 
-		set(PNG "${LIBDIR}/png")
+		set(PNG "${LIBDIR}")
 		set(PNG_INCLUDE_DIR "${PNG}/include")
 		set(PNG_LIBPATH ${PNG}/lib) # not cmake defined
 
@@ -1327,12 +1332,12 @@
 		endif()
 		set(PNG_LIBRARIES png)
 
-		set(ZLIB ${LIBDIR}/zlib)
+		set(ZLIB ${LIBDIR})
 		set(ZLIB_INCLUDE_DIRS ${ZLIB}/include)
 		set(ZLIB_LIBPATH ${ZLIB}/lib)
 		set(ZLIB_LIBRARIES z)
 
-		set(JPEG "${LIBDIR}/jpeg")
+		set(JPEG "${LIBDIR}")
 		set(JPEG_INCLUDE_DIR "${JPEG}/include")
 		set(JPEG_LIBPATH ${JPEG}/lib) # not cmake defined
 		
@@ -1344,20 +1349,20 @@
 			set(PTHREADS_LIBRARIES pthreadGC2)
 		endif()
 		
-		set(FREETYPE ${LIBDIR}/freetype)
+		set(FREETYPE ${LIBDIR})
 		set(FREETYPE_INCLUDE_DIRS ${FREETYPE}/include ${FREETYPE}/include/freetype2)
 		set(FREETYPE_LIBPATH ${FREETYPE}/lib)
 		set(FREETYPE_LIBRARY freetype)
 
 		if(WITH_FFTW3)
-			set(FFTW3 ${LIBDIR}/fftw3)
+			set(FFTW3 ${LIBDIR})
 			set(FFTW3_LIBRARIES fftw3)
 			set(FFTW3_INCLUDE_DIRS ${FFTW3}/include)
 			set(FFTW3_LIBPATH ${FFTW3}/lib)
 		endif()
 
 		if(WITH_OPENCOLLADA)
-			set(OPENCOLLADA ${LIBDIR}/opencollada)
+			set(OPENCOLLADA ${LIBDIR})
 			set(OPENCOLLADA_INCLUDE_DIRS
 				${OPENCOLLADA}/include/opencollada/COLLADAStreamWriter
 				${OPENCOLLADA}/include/opencollada/COLLADABaseUtils
@@ -1366,23 +1371,19 @@
 				${OPENCOLLADA}/include/opencollada/GeneratedSaxParser
 			)
 			set(OPENCOLLADA_LIBPATH ${OPENCOLLADA}/lib/opencollada)
-			set(OPENCOLLADA_LIBRARIES OpenCOLLADAStreamWriter OpenCOLLADASaxFrameworkLoader OpenCOLLADAFramework OpenCOLLADABaseUtils GeneratedSaxParser UTF MathMLSolver buffer ftoa xml)
+			set(OPENCOLLADA_LIBRARIES OpenCOLLADAStreamWriter OpenCOLLADASaxFrameworkLoader OpenCOLLADAFramework OpenCOLLADABaseUtils GeneratedSaxParser UTF MathMLSolver buffer ftoa xml2)
 			set(PCRE_LIBRARIES pcre)
 		endif()
 
 		if(WITH_CODEC_FFMPEG)
-			set(FFMPEG ${LIBDIR}/ffmpeg)
+			set(FFMPEG ${LIBDIR})
 			set(FFMPEG_INCLUDE_DIRS ${FFMPEG}/include)
-			if(WITH_MINGW64)
-				set(FFMPEG_LIBRARIES avcodec.dll avformat.dll avdevice.dll avutil.dll swscale.dll swresample.dll)
-			else()
-				set(FFMPEG_LIBRARIES avcodec-53 avformat-53 avdevice-53 avutil-51 swscale-2)
-			endif()
+			set(FFMPEG_LIBRARIES avcodec avformat avdevice avutil swscale swresample)
 			set(FFMPEG_LIBPATH ${FFMPEG}/lib)
 		endif()
 
 		if(WITH_IMAGE_OPENEXR)
-			set(OPENEXR ${LIBDIR}/openexr)
+			set(OPENEXR ${LIBDIR})
 			set(OPENEXR_INCLUDE_DIR ${OPENEXR}/include)
 			set(OPENEXR_INCLUDE_DIRS ${OPENEXR}/include/OpenEXR)
 			set(OPENEXR_LIBRARIES Half IlmImf Imath IlmThread Iex)
@@ -1390,14 +1391,14 @@
 		endif()
 
 		if(WITH_IMAGE_TIFF)
-			set(TIFF ${LIBDIR}/tiff)
+			set(TIFF ${LIBDIR})
 			set(TIFF_LIBRARY tiff)
 			set(TIFF_INCLUDE_DIR ${TIFF}/include)
 			set(TIFF_LIBPATH ${TIFF}/lib)
 		endif()
 
 		if(WITH_JACK)
-			set(JACK ${LIBDIR}/jack)
+			set(JACK ${LIBDIR})
 			set(JACK_INCLUDE_DIRS ${JACK}/include/jack ${JACK}/include)
 			set(JACK_LIBRARIES jack)
 			set(JACK_LIBPATH ${JACK}/lib)
@@ -1409,8 +1410,8 @@
 		if(WITH_PYTHON)
 			# normally cached but not since we include them with blender
 			set(PYTHON_VERSION 3.3) #  CACHE STRING)
-			set(PYTHON_INCLUDE_DIR "${LIBDIR}/python/include/python${PYTHON_VERSION}")  # CACHE PATH)
-			set(PYTHON_LIBRARY "${LIBDIR}/python/lib/python33mw.lib")  # CACHE FILEPATH)
+			set(PYTHON_INCLUDE_DIR "${LIBDIR}/include/python${PYTHON_VERSION}m")  # CACHE PATH)
+			set(PYTHON_LIBRARY "${LIBDIR}/lib/libpython3.3m.dll.a")  # CACHE FILEPATH)
 
 			# uncached vars
 			set(PYTHON_INCLUDE_DIRS "${PYTHON_INCLUDE_DIR}")
@@ -1418,15 +1419,10 @@
 		endif()
 
 		if(WITH_BOOST)
-			set(BOOST ${LIBDIR}/boost)
+			set(BOOST ${LIBDIR})
 			set(BOOST_INCLUDE_DIR ${BOOST}/include)
-			if(WITH_MINGW64)
-				set(BOOST_POSTFIX "mgw47-mt-s-1_49")
-				set(BOOST_DEBUG_POSTFIX "mgw47-mt-sd-1_49")
-			else()
-				set(BOOST_POSTFIX "mgw46-mt-s-1_49")
-				set(BOOST_DEBUG_POSTFIX "mgw46-mt-sd-1_49")
-			endif()
+			set(BOOST_POSTFIX "mt")
+			set(BOOST_DEBUG_POSTFIX "mt")
 			set(BOOST_LIBRARIES
 				optimized boost_date_time-${BOOST_POSTFIX} boost_filesystem-${BOOST_POSTFIX}
 				boost_regex-${BOOST_POSTFIX}
@@ -1449,7 +1445,7 @@
 		endif()
 			
 		if(WITH_OPENIMAGEIO)
-			set(OPENIMAGEIO ${LIBDIR}/openimageio)
+			set(OPENIMAGEIO ${LIBDIR})
 			set(OPENIMAGEIO_INCLUDE_DIRS ${OPENIMAGEIO}/include)
 			set(OPENIMAGEIO_LIBRARIES OpenImageIO)
 			set(OPENIMAGEIO_LIBPATH ${OPENIMAGEIO}/lib)
@@ -1457,11 +1453,11 @@
 		endif()
 		
 		if(WITH_LLVM)
-			set(LLVM_DIRECTORY ${LIBDIR}/llvm CACHE PATH	"Path to the LLVM installation")
+			set(LLVM_DIRECTORY ${LIBDIR} CACHE PATH	"Path to the LLVM installation")
 			set(LLVM_LIBPATH ${LLVM_DIRECTORY}/lib)
 			# Explicitly set llvm lib order.
 			#---- WARNING ON GCC ORDER OF LIBS IS IMPORTANT, DO NOT CHANGE! ---------
-			set(LLVM_LIBRARY LLVMSelectionDAG LLVMCodeGen LLVMScalarOpts LLVMAnalysis LLVMArchive
+			set(LLVM_LIBRARY LLVMSelectionDAG LLVMCodeGen LLVMScalarOpts LLVMAnalysis LLVMIRReader
 				LLVMAsmParser LLVMAsmPrinter
 				LLVMBitReader LLVMBitWriter
 				LLVMDebugInfo LLVMExecutionEngine
@@ -1469,8 +1465,8 @@
 				LLVMInterpreter LLVMJIT
 				LLVMLinker LLVMMC
 				LLVMMCDisassembler LLVMMCJIT
-				LLVMMCParser LLVMObject
-				LLVMRuntimeDyld 
+				LLVMMCParser LLVMObject LLVMObjCARCOpts
+				LLVMOption LLVMRuntimeDyld 
 				LLVMSupport
 				LLVMTableGen LLVMTarget
 				LLVMTransformUtils LLVMVectorize
@@ -1484,7 +1480,7 @@
 		endif()
 		
 		if(WITH_OPENCOLORIO)
-			set(OPENCOLORIO ${LIBDIR}/opencolorio)
+			set(OPENCOLORIO ${LIBDIR})
 			set(OPENCOLORIO_INCLUDE_DIRS ${OPENCOLORIO}/include)
 			set(OPENCOLORIO_LIBRARIES OpenColorIO)
 			set(OPENCOLORIO_LIBPATH ${OPENCOLORIO}/lib)
@@ -1501,14 +1497,16 @@
 	
 	# Things common to both mingw and MSVC  should go here
 
-	set(WINTAB_INC ${LIBDIR}/wintab/include)
+	set(WINTAB_INC ${LIBDIR}/include)
 
 	if(WITH_OPENAL)
-		set(OPENAL ${LIBDIR}/openal)
+		set(OPENAL ${LIBDIR})
 		set(OPENALDIR ${LIBDIR}/openal)
         set(OPENAL_INCLUDE_DIR ${OPENAL}/include)
         if(MSVC11 OR MSVC12)
             set(OPENAL_LIBRARY openal32)
+		elseif(MINGW)
+			set(OPENAL_LIBRARY OpenAL32)
         else()
             set(OPENAL_LIBRARY wrap_oal)
         endif()
@@ -1516,15 +1514,19 @@
 	endif()
 
 	if(WITH_CODEC_SNDFILE)
-		set(SNDFILE ${LIBDIR}/sndfile)
+		set(SNDFILE ${LIBDIR})
 		set(SNDFILE_INCLUDE_DIRS ${SNDFILE}/include)
-		set(SNDFILE_LIBRARIES libsndfile-1)
+		if(MINGW)
+			set(SNDFILE_LIBRARIES sndfile)
+		else()
+			set(SNDFILE_LIBRARIES libsndfile-1)
+		endif()
 		set(SNDFILE_LIBPATH ${SNDFILE}/lib) # TODO, deprecate
 	endif()
 
 	if(WITH_SDL)
-		set(SDL ${LIBDIR}/sdl)
-		set(SDL_INCLUDE_DIR ${SDL}/include)
+		set(SDL ${LIBDIR})
+		set(SDL_INCLUDE_DIR ${SDL}/include ${SDL}/include/SDL)
 		set(SDL_LIBRARY SDL)
 		set(SDL_LIBPATH ${SDL}/lib)
 	endif()
@@ -1534,7 +1536,7 @@
 	endif()
 
 	if(WITH_CYCLES_OSL)
-		set(CYCLES_OSL ${LIBDIR}/osl CACHE PATH "Path to OpenShadingLanguage installation")
+		set(CYCLES_OSL ${LIBDIR} CACHE PATH "Path to OpenShadingLanguage installation")
 	
 		find_library(OSL_LIB_EXEC NAMES oslexec PATHS ${CYCLES_OSL}/lib)
 		find_library(OSL_LIB_COMP NAMES oslcomp PATHS ${CYCLES_OSL}/lib)
@@ -1860,7 +1862,7 @@
 #-----------------------------------------------------------------------------
 # Common.
 
-if(APPLE OR WIN32)
+if(APPLE)
 	if(NOT EXISTS "${LIBDIR}/")
 		message(FATAL_ERROR "Apple and Windows require pre-compiled libs at: '${LIBDIR}'")
 	endif()
