--- blender-2.72/build_files/cmake/macros.cmake.orig	2014-10-07 11:53:48.545534700 +0100
+++ blender-2.72/build_files/cmake/macros.cmake	2014-10-07 11:55:10.141534700 +0100
@@ -294,7 +294,7 @@
 	if(WITH_PYTHON)  # AND NOT WITH_PYTHON_MODULE  # WIN32 needs
 		target_link_libraries(${target} ${PYTHON_LINKFLAGS})
 
-		if(WIN32 AND NOT UNIX)
+		if(WIN32 AND NOT UNIX AND NOT MINGW)
 			file_list_suffix(PYTHON_LIBRARIES_DEBUG "${PYTHON_LIBRARIES}" "_d")
 			target_link_libraries_debug(${target} "${PYTHON_LIBRARIES_DEBUG}")
 			target_link_libraries_optimized(${target} "${PYTHON_LIBRARIES}")
@@ -366,7 +366,7 @@
 		target_link_libraries(${target} ${FFMPEG_LIBRARIES})
 	endif()
 	if(WITH_OPENCOLLADA)
-		if(WIN32 AND NOT UNIX)
+		if(WIN32 AND NOT UNIX AND NOT MINGW)
 			file_list_suffix(OPENCOLLADA_LIBRARIES_DEBUG "${OPENCOLLADA_LIBRARIES}" "_d")
 			target_link_libraries_debug(${target} "${OPENCOLLADA_LIBRARIES_DEBUG}")
 			target_link_libraries_optimized(${target} "${OPENCOLLADA_LIBRARIES}")
