--- a/include/orcus/env.hpp
+++ b/include/orcus/env.hpp
@@ -11,10 +11,7 @@
 // orcus
 
 #if defined _WIN32 || defined __CYGWIN__
-  #if defined __MINGW32__
-    #define ORCUS_DLLPUBLIC
-    #define ORCUS_DLLLOCAL
-  #elif defined __ORCUS_BUILDING_DLL
+  #if defined __ORCUS_BUILDING_DLL
     #ifdef __GNUC__
       #define ORCUS_DLLPUBLIC __attribute__ ((dllexport))
     #else
@@ -43,10 +40,7 @@
 // orcus-parser
 
 #if defined _WIN32 || defined __CYGWIN__
-  #if defined __MINGW32__
-    #define ORCUS_PSR_DLLPUBLIC
-    #define ORCUS_PSR_DLLLOCAL
-  #elif defined __ORCUS_PSR_BUILDING_DLL
+  #if defined __ORCUS_PSR_BUILDING_DLL
     #ifdef __GNUC__
       #define ORCUS_PSR_DLLPUBLIC __attribute__ ((dllexport))
     #else
@@ -75,10 +69,7 @@
 // orcus-spreadsheet-model
 
 #if defined _WIN32 || defined __CYGWIN__
-  #if defined __MINGW32__
-    #define ORCUS_SPM_DLLPUBLIC
-    #define ORCUS_SPM_DLLLOCAL
-  #elif defined __ORCUS_SPM_BUILDING_DLL
+  #if defined __ORCUS_SPM_BUILDING_DLL
     #ifdef __GNUC__
       #define ORCUS_SPM_DLLPUBLIC __attribute__ ((dllexport))
     #else
@@ -107,10 +98,7 @@
 // orcus-mso
 
 #if defined _WIN32 || defined __CYGWIN__
-  #if defined __MINGW32__
-    #define ORCUS_MSO_DLLPUBLIC
-    #define ORCUS_MSO_DLLLOCAL
-  #elif defined __ORCUS_MSO_BUILDING_DLL
+  #if defined __ORCUS_MSO_BUILDING_DLL
     #ifdef __GNUC__
       #define ORCUS_MSO_DLLPUBLIC __attribute__ ((dllexport))
     #else
--- a/src/python/Makefile.am
+++ b/src/python/Makefile.am
@@ -29,7 +29,7 @@
 	gnumeric.hpp \
 	gnumeric.cpp
 
-_orcus_la_LDFLAGS = -module -avoid-version -export-symbols-regex PyInit__orcus
+_orcus_la_LDFLAGS = -no-undefined -module -avoid-version -export-symbols-regex PyInit__orcus
 _orcus_la_LIBADD = \
 	../liborcus/liborcus-@ORCUS_API_VERSION@.la \
 	../parser/liborcus-parser-@ORCUS_API_VERSION@.la \
@@ -64,7 +64,7 @@
 _orcus_json_la_SOURCES = \
 	json.cpp
 
-_orcus_json_la_LDFLAGS = -module -avoid-version -export-symbols-regex PyInit__orcus_json
+_orcus_json_la_LDFLAGS = -no-undefined -module -avoid-version -export-symbols-regex PyInit__orcus_json
 _orcus_json_la_LIBADD = \
 	../liborcus/liborcus-@ORCUS_API_VERSION@.la \
 	../parser/liborcus-parser-@ORCUS_API_VERSION@.la \
--- a/src/python/json.cpp
+++ b/src/python/json.cpp
@@ -279,7 +279,7 @@
 
 extern "C" {
 
-ORCUS_DLLPUBLIC PyObject* PyInit__orcus_json()
+PyObject* PyInit__orcus_json()
 {
     PyObject* m = PyModule_Create(&orcus::python::moduledef);
     return m;
--- a/src/python/python.cpp
+++ b/src/python/python.cpp
@@ -146,7 +146,7 @@
 
 extern "C" {
 
-ORCUS_DLLPUBLIC PyObject* PyInit__orcus()
+PyObject* PyInit__orcus()
 {
     PyObject* m = PyModule_Create(&orcus::python::moduledef);
     if (!orcus::python::populate_module_attributes(m))
