From e83040c34dfbd908a784fe3bde9f948eee82cfc1 Mon Sep 17 00:00:00 2001
From: Hugo <hugovk@users.noreply.github.com>
Date: Thu, 12 Oct 2017 22:31:59 +0300
Subject: [PATCH] Update classifiers

---
 setup.cfg | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/setup.cfg b/setup.cfg
index 2ad0da56..b201b856 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -6,8 +6,23 @@ description-file = doc/overview.rst
 author = Jonathan M. Lange
 author-email = jml+testtools@mumak.net
 classifier = 
+    Development Status :: 5 - Production/Stable
+    Intended Audience :: Developers
     License :: OSI Approved :: MIT License
+    Operating System :: OS Independent
+    Programming Language :: Python
+    Programming Language :: Python :: 2
+    Programming Language :: Python :: 2.7
     Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.3
+    Programming Language :: Python :: 3.4
+    Programming Language :: Python :: 3.5
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: Implementation :: CPython
+    Programming Language :: Python :: Implementation :: PyPy
+    Topic :: Software Development :: Libraries
+    Topic :: Software Development :: Libraries :: Python Modules
+    Topic :: Software Development :: Testing
 
 [extras]
 test =
