From f41ee7224121a9b74dab1c91b3fede44c9c34029 Mon Sep 17 00:00:00 2001
From: Hugo <hugovk@users.noreply.github.com>
Date: Tue, 9 Jan 2018 14:56:38 +0200
Subject: [PATCH] Ignore IDE metadata

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 305dc6f7..b1fafa81 100644
--- a/.gitignore
+++ b/.gitignore
@@ -19,3 +19,4 @@ testtools.egg-info
 /.eggs/
 AUTHORS
 ChangeLog
+.idea
