diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0209c56..0b605e6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -347,7 +347,7 @@ if(WIN32)
 			COMMAND windres "-I${CMAKE_CURRENT_SOURCE_DIR}" "-i${CMAKE_CURRENT_SOURCE_DIR}/src/winapp.rc" -o "${CMAKE_CURRENT_BINARY_DIR}/sqlbicon.o" VERBATIM)
 		set(SQLB_SRC ${SQLB_SRC} "${CMAKE_CURRENT_BINARY_DIR}/sqlbicon.o")
 		set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-subsystem,windows")
-		set(WIN32_STATIC_LINK -Wl,-Bstatic -lssl -lcrypto -lws2_32)
+		set(WIN32_LINK -lssl -lcrypto -lws2_32)
 		set(ADDITIONAL_LIBS lzma)
 	ELSE( MINGW )
 		set(SQLB_SRC ${SQLB_SRC} "${CMAKE_CURRENT_SOURCE_DIR}/src/winapp.rc")
@@ -452,7 +452,7 @@ set(QT_LIBS Qt5::Gui Qt5::Test Qt5::PrintSupport Qt5::Widgets Qt5::Network Qt5::
 target_link_libraries(${PROJECT_NAME}
     ${LPTHREAD}
     ${QT_LIBS}
-    ${WIN32_STATIC_LINK}
+    ${WIN32_LINK}
     ${LIBSQLITE}
     ${ADDITIONAL_LIBS})
 if(QHexEdit_FOUND)
