From 0cfc23197c6cb3badd525e3ebc1c6342f755bde9 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Thu, 12 May 2022 19:15:00 -0400
Subject: Fix compiler problem

---
 sip/cpp/sip_corewxMenuItem.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sip/cpp/sip_corewxMenuItem.cpp b/sip/cpp/sip_corewxMenuItem.cpp
index 03366128..917fe137 100644
--- a/sip/cpp/sip_corewxMenuItem.cpp
+++ b/sip/cpp/sip_corewxMenuItem.cpp
@@ -38,7 +38,7 @@
     const wxBitmap* _wxMenuItem_GetDisabledBitmap(const wxMenuItem* self)
     {
-        #ifdef __WXMSW__
+        #if defined( __WXMSW__ ) && !defined( __clang__ )
             return &self->GetDisabledBitmap();
         #else
             return &wxNullBitmap;
         #endif
-- 
