--- a/buildtools/config.py
+++ b/buildtools/config.py
@@ -1065,7 +1065,7 @@ def updateLicenseFiles(cfg):
     # Combine the relevant files into a single LICENSE.txt file
     text = ''
     for filename in ['preamble.txt', 'licence.txt', 'lgpl.txt', 'sip-license.txt']:
-        with open(opj('license', filename), 'r') as f:
+        with open(opj('license', filename), 'r', encoding='utf-8') as f:
             text += f.read() + '\n\n'
     with open('LICENSE.txt', 'w') as f:
         f.write(text)
--- a/wscript
+++ b/wscript
@@ -570,7 +570,6 @@ def build(bld):
         features = 'c cxx cshlib cxxshlib pyext',
         target   = makeTargetName(bld, 'siplib'),
         source   = ['sip/siplib/apiversions.c',
-                    'sip/siplib/bool.cpp',
                     'sip/siplib/descriptors.c',
                     'sip/siplib/int_convertors.c',
                     'sip/siplib/objmap.c',
