--- imagecodecs-2022.2.22/setup.py.orig	2022-06-19 20:09:02.175217700 +0200
+++ imagecodecs-2022.2.22/setup.py	2022-06-19 20:15:53.590230600 +0200
@@ -148,7 +148,7 @@
     'brunsli': ext(libraries=['brunslidec-c', 'brunslienc-c']),
     'bz2': ext(libraries=['bz2']),
     'cms': ext(libraries=['lcms2']),
-    'deflate': ext(libraries=['deflate']),
+    'deflate': ext(libraries=['deflate'],define_macros=[('LIBDEFLATE_DLL', 1)]),
     'gif': ext(libraries=['gif']),
     # 'heif': ext(libraries=['heif']),
     'jpeg2k': ext(
