--- imagecodecs-2022.7.27/setup.py.orig	2022-07-27 21:09:37.000000000 +0200
+++ imagecodecs-2022.7.27/setup.py	2022-07-28 18:09:46.847904400 +0200
@@ -142,7 +142,7 @@
     'brunsli': ext(libraries=['brunslidec-c', 'brunslienc-c']),
     'bz2': ext(libraries=['bz2']),
     'cms': ext(libraries=['lcms2']),
-    'deflate': ext(libraries=['deflate']),
+    'deflate': ext(libraries=['deflate'],define_macros=[('LIBDEFLATE_DLL', 1)]),
     'gif': ext(libraries=['gif']),
     'heif': ext(libraries=['heif']),
     'jetraw': ext(libraries=['jetraw', 'dpcore']),
