diff -Naur imagecodecs-2021.1.11_orig/setup.py imagecodecs-2021.1.11/setup.py
--- imagecodecs-2021.1.11_orig/setup.py	2021-01-12 05:06:22.000000000 +0100
+++ imagecodecs-2021.1.11/setup.py	2021-01-12 15:05:51.063607600 +0100
@@ -179,7 +179,7 @@
     """Customize build for common platforms: recent Debian, arch..."""
     import platform
 
-    del EXTENSIONS['avif']  # libavif library not commonly available
+    # del EXTENSIONS['avif']  # libavif library not commonly available
     del EXTENSIONS['jpeg12']  # jpeg12 requires custom build
     del EXTENSIONS['lerc']  # LERC library not commonly available
     del EXTENSIONS['lz4f']  # requires static linking
@@ -187,17 +187,14 @@
     if 'arch' in platform.platform():
         del EXTENSIONS['zopfli']  # zopfli/zopfli.h does not exist
     else:
-        del EXTENSIONS['jpegls']  # CharLS 2.1 library not commonly available
+        # del EXTENSIONS['jpegls']  # CharLS 2.1 library not commonly available
         del EXTENSIONS['jpegxl']  # Brunsli library not commonly available
         del EXTENSIONS['zfp']  # ZFP library not commonly available
 
-    if sys.platform == 'win32':
-        EXTENSIONS['bz2']['libraries'] = ['libbz2']
-    else:
-        EXTENSIONS['jpeg2k']['include_dirs'].extend(
-            ('/usr/include/openjpeg-2.3', '/usr/include/openjpeg-2.4')
-        )
-        EXTENSIONS['jpegxr']['include_dirs'].append('/usr/include/jxrlib')
+    EXTENSIONS['jpeg2k']['include_dirs'].extend(
+        (sys.prefix + '/include/openjpeg-2.3', sys.prefix + '/include/openjpeg-2.4')
+    )
+    EXTENSIONS['jpegxr']['include_dirs'].append(sys.prefix + '/include/jxrlib')
 
 
 def customize_build_cg(EXTENSIONS, OPTIONS):
