--- a/imagecodecs/_blosc2.pyx
+++ b/imagecodecs/_blosc2.pyx
@@ -113,7 +113,7 @@
 
 def blosc2_version():
     """Return Blosc library version string."""
-    return 'c-blosc2 ' + BLOSC_VERSION_STRING.decode()
+    return 'c-blosc2 ' + BLOSC2_VERSION_STRING.decode()
 
 
 def blosc2_check(data):
@@ -162,7 +162,7 @@
 
     srcsize = src.size
 
-    if srcsize > 2147483647 - BLOSC_MAX_OVERHEAD:
+    if srcsize > 2147483647 - BLOSC2_MAX_OVERHEAD:
         raise ValueError('data size larger than 2 GB')
 
     if blocksize is None:
@@ -201,7 +201,7 @@
 
     if out is None:
         if dstsize < 0:
-            dstsize = srcsize + BLOSC_MAX_OVERHEAD
+            dstsize = srcsize + BLOSC2_MAX_OVERHEAD
         if dstsize < 17:
             dstsize = 17
         out = _create_output(outtype, dstsize)
@@ -210,17 +210,17 @@
     dstsize = dst.size
 
     with nogil:
-        ret = blosc_set_compressor(compname)
+        ret = blosc1_set_compressor(compname)
         if ret < 0:
-            raise Blosc2Error(f'blosc_set_compressor', None, ret)
+            raise Blosc2Error(f'blosc1_set_compressor', None, ret)
 
         if nthreads == 0:
-            nthreads = blosc_get_nthreads()
-        ret = blosc_set_nthreads(nthreads)
+            nthreads = blosc2_get_nthreads()
+        ret = blosc2_set_nthreads(nthreads)
         if ret < 0:
-            raise Blosc2Error(f'blosc_set_nthreads', None, ret)
+            raise Blosc2Error(f'blosc2_set_nthreads', None, ret)
 
-        blosc_set_blocksize(cblocksize)
+        blosc1_set_blocksize(cblocksize)
 
         ret = blosc2_compress(
             clevel,
@@ -281,10 +281,10 @@
 
     with nogil:
         if nthreads == 0:
-            nthreads = blosc_get_nthreads()
-        ret = blosc_set_nthreads(nthreads)
+            nthreads = blosc2_get_nthreads()
+        ret = blosc2_set_nthreads(nthreads)
         if ret < 0:
-            raise Blosc2Error(f'blosc_set_nthreads', None, ret)
+            raise Blosc2Error(f'blosc2_set_nthreads', None, ret)
 
         ret = blosc2_decompress(
             <const void*> &src[0],
--- a/imagecodecs/blosc2.pxd
+++ b/imagecodecs/blosc2.pxd
@@ -16,7 +16,7 @@
     int  BLOSC_VERSION_MINOR
     int  BLOSC_VERSION_RELEASE
 
-    char* BLOSC_VERSION_STRING
+    char* BLOSC2_VERSION_STRING
     char* BLOSC_VERSION_DATE
 
     int BLOSC2_MAX_DIM
@@ -42,7 +42,7 @@
 
     int BLOSC_MIN_HEADER_LENGTH
     int BLOSC_EXTENDED_HEADER_LENGTH
-    int BLOSC_MAX_OVERHEAD
+    int BLOSC2_MAX_OVERHEAD
     int BLOSC_MAX_BUFFERSIZE
     int BLOSC_MAX_TYPESIZE
     int BLOSC_MIN_BUFFERSIZE
@@ -232,15 +232,15 @@
         void* callback_data
     ) nogil
 
-    int16_t blosc_get_nthreads() nogil
+    int16_t blosc2_get_nthreads() nogil
 
-    int16_t blosc_set_nthreads(
+    int16_t blosc2_set_nthreads(
         int16_t nthreads
     ) nogil
 
     const char* blosc_get_compressor() nogil
 
-    int blosc_set_compressor(
+    int blosc1_set_compressor(
         const char* compname
     ) nogil
 
@@ -827,7 +827,7 @@
     int blosc_get_blocksize(
     ) nogil
 
-    void blosc_set_blocksize(
+    void blosc1_set_blocksize(
         size_t blocksize
     ) nogil
 
