--- configure.py.orig	2014-07-15 16:41:19.233000000 +0200
+++ configure.py	2014-07-15 16:41:40.357000000 +0200
@@ -2131,7 +2131,7 @@
 
     # Create the output file, first making sure it doesn't exist.
     remove_file(out_file)
-    run_command(exe_file)
+    run_command("sh -c " + exe_file)
 
     if not os.access(out_file, os.F_OK):
         sipconfig.error("%s failed to create %s. Make sure your Qt installation is correct." % (exe_file, out_file))
