diff --git a/helix-loader/src/lib.rs b/helix-loader/src/lib.rs
index 0e7c134d..2b2f0728 100644
--- a/helix-loader/src/lib.rs
+++ b/helix-loader/src/lib.rs
@@ -39,7 +39,7 @@ pub fn initialize_log_file(specified_file: Option<PathBuf>) {
 ///
 /// Postcondition: returns at least two paths (they might not exist).
 fn prioritize_runtime_dirs() -> Vec<PathBuf> {
-    const RT_DIR: &str = "runtime";
+    const RT_DIR: &str = "../lib/helix/runtime";
     // Adding higher priority first
     let mut rt_dirs = Vec::new();
     if let Ok(dir) = std::env::var("CARGO_MANIFEST_DIR") {
