--- matplotlib-3.1.0/setupext.py.orig	2019-05-18 03:55:28.000000000 +0200
+++ matplotlib-3.1.0/setupext.py	2019-05-19 11:31:01.633173900 +0200
@@ -24,6 +24,7 @@
 
 _log = logging.getLogger(__name__)
 
+MSYS = "MSYSTEM" in os.environ
 
 def _get_xdg_cache_dir():
     """
@@ -220,7 +221,7 @@
     def __init__(self):
         """Determines whether pkg-config exists on this machine."""
         self.pkg_config = None
-        if sys.platform != 'win32':
+        if sys.platform != 'win32' or MSYS:
             pkg_config = os.environ.get('PKG_CONFIG', 'pkg-config')
             if shutil.which(pkg_config) is not None:
                 self.pkg_config = pkg_config
@@ -276,7 +277,7 @@
 
         # conda Windows header and library paths.
         # https://github.com/conda/conda/issues/2312 re: getting the env dir.
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' and not MSYS:
             conda_env_path = (os.getenv('CONDA_PREFIX')  # conda >= 4.1
                               or os.getenv('CONDA_DEFAULT_ENV'))  # conda < 4.1
             if conda_env_path and os.path.isdir(conda_env_path):
