--- matplotlib-3.0.0/setupext.py.orig	2018-09-18 05:17:52.000000000 +0200
+++ matplotlib-3.0.0/setupext.py	2018-09-20 12:51:06.529116400 +0200
@@ -22,6 +22,7 @@
 import setuptools
 import versioneer
 
+MSYS = "MSYSTEM" in os.environ
 
 def _get_xdg_cache_dir():
     """
@@ -267,7 +268,7 @@
         """
         Determines whether pkg-config exists on this machine.
         """
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' and not MSYS:
             self.has_pkgconfig = False
         else:
             self.pkg_config = os.environ.get('PKG_CONFIG', 'pkg-config')
@@ -946,7 +947,7 @@
         if options.get('local_freetype'):
             return "Using local version for testing"
 
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' and not MSYS:
             try:
                 check_include_file(get_include_dirs(), 'ft2build.h', 'freetype')
             except CheckFailed:
@@ -1176,7 +1177,7 @@
         }
 
     def check(self):
-        if sys.platform == 'win32':
+        if sys.platform == 'win32' and not MSYS:
             check_include_file(get_include_dirs(), 'png.h', 'png')
             return 'Using unknown version found on system.'
 
