--- ./diffutils-3.6/lib/mktime.c.orig	2023-12-17 19:42:03.475896700 +0100
+++ ./diffutils-3.6/lib/mktime.c	2023-12-17 19:42:05.149366700 +0100
@@ -54,6 +54,7 @@
 
 #include <intprops.h>
 #include <verify.h>
+#include <stdlib.h>
 
 #if DEBUG_MKTIME
 # include <stdio.h>
--- ./diffutils-3.6/lib/tzset.orig.c	2023-12-17 19:31:27.044167900 +0100
+++ ./diffutils-3.6/lib/tzset.c	2023-12-17 19:32:02.390247800 +0100
@@ -21,6 +21,8 @@
 
 /* Specification.  */
 #include <time.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "localtime-buffer.h"
 
--- ./diffutils-3.6/src/system.h.orig	2023-12-17 19:37:07.267051800 +0100
+++ ./diffutils-3.6/src/system.h	2023-12-17 19:40:25.529946000 +0100
@@ -70,7 +70,9 @@
 
 #include <string.h>
 #if ! HAVE_STRCASECOLL
-# if HAVE_STRICOLL || defined stricoll
+# if _WIN32
+#  define strcasecoll(a, b) _stricoll (a, b)
+# elif HAVE_STRICOLL || defined stricoll
 #  define strcasecoll(a, b) stricoll (a, b)
 # else
 #  define strcasecoll(a, b) strcasecmp (a, b) /* best we can do */
