--- lib/config.hin.orig	2013-03-24 17:55:37.000000000 +0000
+++ lib/config.hin	2015-04-13 13:38:17.351409000 +0100
@@ -599,7 +599,9 @@
 
 /* Define to 1 on MSVC platforms that have the "invalid parameter handler"
    concept. */
-#undef HAVE_MSVC_INVALID_PARAMETER_HANDLER
+/* Commented out due to a gnulib bug - see
+   http://lists.gnu.org/archive/html/bug-sed/2015-01/msg00003.html */
+/* #undef HAVE_MSVC_INVALID_PARAMETER_HANDLER */
 
 /* Define to 1 if you have the `nanotime' function. */
 #undef HAVE_NANOTIME
