diff -bur diffutils-3.10-orig/lib/cmpbuf.c diffutils-3.10/lib/cmpbuf.c
--- diffutils-3.10-orig/lib/cmpbuf.c	2025-02-01 15:09:12.633436600 -0700
+++ diffutils-3.10/lib/cmpbuf.c	2025-02-01 15:10:07.563027900 -0700
@@ -35,6 +35,10 @@
 #undef MIN
 #define MIN(a, b) ((a) <= (b) ? (a) : (b))

+#ifndef SA_RESTART 
+# define SA_RESTART 0
+#endif
+
 /* Read NBYTES bytes from descriptor FD into BUF.
    NBYTES must not be SIZE_MAX.
    Return the number of characters successfully read.
diff -bur diffutils-3.10-orig/src/util.c diffutils-3.10/src/util.c
--- diffutils-3.10-orig/src/util.c	2025-02-01 15:09:12.712562500 -0700
+++ diffutils-3.10/src/util.c	2025-02-01 15:10:26.525072700 -0700
@@ -45,6 +45,15 @@
 # define SA_RESTART 0
 #endif
 
+/* Out-of-range substitutes make a good fallback for uncatchable
+   signals.  */
+#ifndef SIGKILL
+# define SIGKILL (-1)
+#endif
+#ifndef SIGSTOP
+# define SIGSTOP (-1)
+#endif
+
 char const pr_program[] = PR_PROGRAM;
 
 /* Queue up one-line messages to be printed at the end,
@@ -307,7 +316,15 @@
 #ifdef SIGALRM
     SIGALRM,
 #endif
-    SIGHUP, SIGINT, SIGPIPE,
+#ifdef SIGHUP
+    SIGHUP, 
+#endif
+#ifdef SIGINT
+    SIGINT, 
+#endif
+#ifdef SIGPIPE
+    SIGPIPE, 
+#endif
 #ifdef SIGQUIT
     SIGQUIT,
 #endif
@@ -374,7 +391,11 @@
       if (h != SIG_IGN && h != SIG_ERR)
 	{
 	  xsigaddset (&caught_signals, sig[j]);
+#ifdef SIGTSTP    
 	  xsignal (sig[j], is_tstp_index (j) ? stophandler : sighandler);
+#else
+	  xsignal (sig[j], sighandler);
+#endif
 	  some_signals_caught = true;
 	  if (siginterrupt (sig[j], 0) != 0)
 	    pfatal_with_name ("siginterrupt");
Only in diffutils-3.10/src: util.c.orig
