--- vine-5.1.0/t/unit/test_synchronization.py	2021-11-19 10:03:20.000000000 +0300
+++ python-build-UCRT64/t/unit/test_synchronization.py	2024-08-27 20:36:12.782232900 +0300
@@ -8,7 +8,7 @@
 
 class test_barrier:
 
-    def setup(self):
+    def setup_method(self):
         self.m1, self.m2, self.m3 = Mock(), Mock(), Mock()
         self.ps = [promise(self.m1), promise(self.m2), promise(self.m3)]
 
