--- tclx8.4/configure.orig	2015-07-31 17:24:18.840987200 +0200
+++ tclx8.4/configure	2015-07-31 17:24:41.606710900 +0200
@@ -2361,7 +2361,7 @@
     fi
 
     case "`uname -s`" in
-	*win32*|*WIN32*|*MINGW32_*)
+	*win32*|*WIN32*|*MINGW*)
 	    # Extract the first word of "cygpath", so it can be a program name with args.
 set dummy cygpath; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
