diff -bur apache-arrow-9.0.0-c/cpp/src/arrow/vendored/fast_float/float_common.h apache-arrow-9.0.0/cpp/src/arrow/vendored/fast_float/float_common.h
--- apache-arrow-9.0.0-c/cpp/src/arrow/vendored/fast_float/float_common.h	2022-07-29 10:44:37.000000000 -0600
+++ apache-arrow-9.0.0/cpp/src/arrow/vendored/fast_float/float_common.h	2022-08-09 01:21:29.097022100 -0600
@@ -183,7 +183,7 @@
 fastfloat_really_inline value128 full_multiplication(uint64_t a,
                                                      uint64_t b) {
   value128 answer;
-#ifdef _M_ARM64
+#if defined(_M_ARM64) && !defined(__MINGW32__)
   // ARM64 has native support for 64-bit multiplications, no need to emulate
   answer.high = __umulh(a, b);
   answer.low = a * b;
