diff --git a/os.hpp b/os.hpp
index 7df3c6c..af2759e 100644
--- a/os.hpp
+++ b/os.hpp
@@ -39,10 +39,15 @@
 #define _UNICODE // Set _T() macro to convert from narrow to wide strings.
 #endif
 
+#ifndef WINVER
 #define WINVER _WIN32_WINNT_WINXP
+#endif
+
+#ifndef _WIN32_WINNT
 #define _WIN32_WINNT _WIN32_WINNT_WINXP
+#endif
 
-#if !defined(ZIPSFX)
+#if !defined(ZIPSFX) && !defined(RAR_SMP)
 #define RAR_SMP
 #endif
 
@@ -51,9 +56,7 @@
 #include <windows.h>
 #include <prsht.h>
 #include <shlwapi.h>
-#pragma comment(lib, "Shlwapi.lib")
-#include <PowrProf.h>
-#pragma comment(lib, "PowrProf.lib")
+#include <powrprof.h>
 #include <shellapi.h>
 #include <shlobj.h>
 #include <winioctl.h>
