diff --git a/isnt.cpp b/isnt.cpp
index 3cc876b..27ed6ef 100644
--- a/isnt.cpp
+++ b/isnt.cpp
@@ -25,9 +25,9 @@ DWORD WinNT()
 
 // Replace it with documented Windows 11 check when available.
 #include <comdef.h>
-#include <Wbemidl.h>
-#pragma comment(lib, "wbemuuid.lib")
+#include <wbemidl.h>
 
+/*
 static bool WMI_IsWindows10()
 {
   IWbemLocator *pLoc = NULL;
@@ -95,7 +95,7 @@ static bool WMI_IsWindows10()
 
   return Win10;
 }
-
+*/
 
 // Replace it with actual check when available.
 bool IsWindows11OrGreater()
@@ -107,7 +107,7 @@ bool IsWindows11OrGreater()
     WinVer.dwOSVersionInfoSize=sizeof(WinVer);
     GetVersionEx(&WinVer);
     IsWin11=WinVer.dwMajorVersion>10 || 
-          WinVer.dwMajorVersion==10 && WinVer.dwBuildNumber >= 22000 && !WMI_IsWindows10();
+          WinVer.dwMajorVersion==10 && WinVer.dwBuildNumber >= 22000; // && !WMI_IsWindows10();
     IsSet=true;
   }
   return IsWin11;
