diff -bur flashrom-v1.5.1-o/meson.build flashrom-v1.5.1/meson.build
--- flashrom-v1.5.1-o/meson.build	2025-04-19 22:34:16.179277800 -0600
+++ flashrom-v1.5.1/meson.build	2025-04-19 22:34:29.178030600 -0600
@@ -661,7 +661,9 @@
 if host_machine.system() == 'darwin'
   vflag = ''
 else
-  vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
+  vflag = cc.get_supported_link_arguments([
+    '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile),
+  ])
 endif
 libflashrom = library(
   'flashrom',
