--- a/scintilla/gtk/makefile
+++ b/scintilla/gtk/makefile
@@ -63,7 +63,7 @@ endif
 
 ifdef windir
 CC = gcc
-DEL = del /q
+DEL = rm -f
 else
 DEL = rm -f
 endif
--- a/scite/win32/makefile
+++ b/scite/win32/makefile
@@ -44,23 +44,9 @@ WINDRES ?= windres
 	ifndef CLANG
 		CC = gcc
 	endif
-	DEL = $(if $(wildcard $(dir $(SHELL))rm.exe), $(dir $(SHELL))rm.exe -f, del)
-	COPY = $(if $(wildcard $(dir $(SHELL))cp.exe), $(dir $(SHELL))cp.exe -a, xcopy /Y)
-	VER = cmd /c ver
-	# Discover Windows version by running 'VER' command and parsing output
-	# For Windows 2000 looks like:Microsoft Windows 2000 [Version 5.00.2195]
-	WINVERWORDS:=$(wordlist 1,2,$(subst ., ,$(lastword $(shell $(VER)))))
-	WINVER:=$(firstword $(WINVERWORDS)).$(lastword $(WINVERWORDS))
-	# Windows NT 4 and Windows 2000 do not support themes so turn off
-ifeq '$(WINVER)' '4.0'
-	DEFINES += -DDISABLE_THEMES -DWIN_TARGET=0x0400
-else
-ifeq '$(WINVER)' '5.00'
-	DEFINES += -DDISABLE_THEMES
-else
+	DEL = rm -f
+	COPY = cp -a
 	LIBS += -luxtheme
-endif
-endif
 else
 	DEL = rm -f
 	COPY = cp -a
