--- a/lexilla/src/makefile
+++ b/lexilla/src/makefile
@@ -20,7 +20,7 @@
 WARNINGS = -Wpedantic -Wall -Wextra
 
 ifdef windir
-    SHARED_NAME = lexilla
+    SHARED_NAME = liblexilla
     SHAREDEXTENSION = dll
     WINDRES ?= windres
     VERSION_RESOURCE = $(DIR_O)/LexillaVersion.o
@@ -128,7 +128,7 @@
 LEXILLA_OBJS := $(addprefix $(DIR_O)/, $(OBJS) $(LEXLIB_OBJS) $(LEXERS:.cxx=.o))
 
 $(LEXILLA): $(LEXILLA_OBJS) $(VERSION_RESOURCE)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) $^ -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -Wl,--out-implib,$(LEXILLA).a $^ -o $@
 
 $(LIBLEXILLA):  $(LEXILLA_OBJS)
 ifeq ($(SHAREDEXTENSION),dylib)
--- a/scintilla/gtk/makefile
+++ b/scintilla/gtk/makefile
@@ -67,8 +67,8 @@
 else
 DEL = rm -f
 endif
-COMPLIB=$(basedir)/bin/scintilla.a
-COMPONENT=$(basedir)/bin/libscintilla.$(SHAREDEXTENSION)
+COMPLIB=$(basedir)/bin/libscintillaGtk.a
+COMPONENT=$(basedir)/bin/libscintillaGtk.$(SHAREDEXTENSION)
 
 vpath %.h $(srcdir) $(basedir)/src $(basedir)/include
 vpath %.c $(srcdir)
@@ -165,7 +165,7 @@
 	$(RANLIB) $@
 
 $(COMPONENT): $(SRC_OBJS) $(GTK_OBJS) $(MARSHALLER)
-	$(CXX) $(CXX_ALL_FLAGS) $(CXXFLAGS) $(LDFLAGS) $^ -o $@ $(CONFIGLIB)
+	$(CXX) $(CXX_ALL_FLAGS) $(CXXFLAGS) $(LDFLAGS) -Wl,--out-implib,$(COMPONENT).a $^ -o $@ $(CONFIGLIB)
 
 # Automatically generate header dependencies with "make depend"
 include deps.mak
--- a/scintilla/win32/makefile
+++ b/scintilla/win32/makefile
@@ -12,7 +12,7 @@
 DIR_O=.
 DIR_BIN=../bin
 
-COMPONENT = $(DIR_BIN)/Scintilla.dll
+COMPONENT = $(DIR_BIN)/libScintilla.dll
 LIBSCI = $(DIR_BIN)/libscintilla.a
 
 WARNINGS = -Wpedantic -Wall -Wextra
@@ -44,7 +44,7 @@
 vpath %.h ../src ../include
 vpath %.cxx ../src
 
-LDFLAGS=-shared -static -mwindows
+LDFLAGS=-shared -mwindows
 LIBS=-lgdi32 -luser32 -limm32 -lole32 -luuid -loleaut32 $(LIBSMINGW)
 
 INCLUDES=-I ../include -I ../src
@@ -125,7 +125,7 @@
 	$(DIR_O)/ScintRes.o
 
 $(COMPONENT): $(COMPONENT_OBJS) $(SHARED_OBJS)
-	$(CXX) $(LDFLAGS) -o $@ $(STRIPFLAG) $^ $(CXXFLAGS) $(LIBS)
+	$(CXX) $(LDFLAGS) -Wl,-out-implib,$(COMPONENT).a -o $@ $(STRIPFLAG) $^ $(CXXFLAGS) $(LIBS)
 
 $(LIBSCI): $(COMPONENT_OBJS)
 	$(AR) $(ARFLAGS) $@ $^
--- a/scite/win32/SciTEWin.cxx
+++ b/scite/win32/SciTEWin.cxx
@@ -27,7 +27,7 @@
 const GUI::gui_char appName[] = GUI_TEXT("Sc1");
 #else
 const GUI::gui_char appName[] = GUI_TEXT("SciTE");
-static const GUI::gui_char scintillaName[] = GUI_TEXT("Scintilla.DLL");
+static const GUI::gui_char scintillaName[] = GUI_TEXT("libScintilla.DLL");
 #endif
 
 static GUI::gui_string GetErrorMessage(DWORD nRet) {
--- a/scite/win32/makefile
+++ b/scite/win32/makefile
@@ -12,7 +12,7 @@
 
 OBJS = Credits.o $(OTHER_OBJS)
 
-DLLS=../bin/Scintilla.dll ../bin/lexilla.dll
+DLLS=../bin/libScintilla.dll ../bin/liblexilla.dll
 
 NOTBINPROPS=SciTE.properties Embedded.properties
 PROPS=$(addprefix ../bin/,$(filter-out $(NOTBINPROPS), $(notdir $(wildcard ../src/*.properties))))
