--- scite/src/SciTEIO.cxx.orig	2014-11-27 21:47:34.000000000 +0100
+++ scite/src/SciTEIO.cxx	2014-12-20 18:52:57.887040400 +0100
@@ -41,14 +41,7 @@
 #include "SciTEBase.h"
 #include "Utf8_16.h"
 
-#if defined(GTK)
 const GUI::gui_char propUserFileName[] = GUI_TEXT(".SciTEUser.properties");
-#elif defined(__APPLE__)
-const GUI::gui_char propUserFileName[] = GUI_TEXT("SciTEUser.properties");
-#else
-// Windows
-const GUI::gui_char propUserFileName[] = GUI_TEXT("SciTEUser.properties");
-#endif
 const GUI::gui_char propGlobalFileName[] = GUI_TEXT("SciTEGlobal.properties");
 const GUI::gui_char propAbbrevFileName[] = GUI_TEXT("abbrev.properties");
 
--- scite/win32/SciTEWin.cxx.orig	2014-09-30 02:14:42.000000000 +0200
+++ scite/win32/SciTEWin.cxx	2014-12-20 18:52:54.939587600 +0100
@@ -413,8 +413,29 @@
 			return FilePath();
 		// Remove the SciTE.exe
 		GUI::gui_char *lastSlash = wcsrchr(path, pathSepChar);
-		if (lastSlash)
+		if (lastSlash) {
 			*lastSlash = '\0';
+		}
+		// MSYS2: go to $(bindir)/../share/scite
+		lastSlash = wcsrchr(path, pathSepChar);
+		if (lastSlash) {
+			if (lastSlash - path + sizeof("share/scite") >= MAX_PATH) {
+				return FilePath(path); // don't risk overflow, return early
+			}
+			*(lastSlash++) = pathSepChar;
+			*(lastSlash++) = 's';
+			*(lastSlash++) = 'h';
+			*(lastSlash++) = 'a';
+			*(lastSlash++) = 'r';
+			*(lastSlash++) = 'e';
+			*(lastSlash++) = pathSepChar;
+			*(lastSlash++) = 's';
+			*(lastSlash++) = 'c';
+			*(lastSlash++) = 'i';
+			*(lastSlash++) = 't';
+			*(lastSlash++) = 'e';
+			*(lastSlash++) = '\0';
+		}
 		return FilePath(path);
 	}
 }
@@ -432,6 +452,8 @@
 FilePath SciTEWin::GetSciteUserHome() {
 	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_HOME"));
 	if (!home)
+		home = _wgetenv(GUI_TEXT("HOME"));
+	if (!home)
 		home = _wgetenv(GUI_TEXT("USERPROFILE"));
 	return GetSciTEPath(home);
 }
