--- a/api/gmxapi/include/gmxapi/gmxapicompat.h
+++ b/api/gmxapi/include/gmxapi/gmxapicompat.h
@@ -49,6 +49,7 @@
 
 #include <map>
 #include <string>
+#include <cstdint>
 
 #include "gmxapi/exceptions.h"
 #include "gmxapi/gmxapi.h"
--- a/src/external/thread_mpi/src/type.cpp
+++ b/src/external/thread_mpi/src/type.cpp
@@ -163,6 +163,7 @@
 #ifdef _MSC_VER
 #define TYPE __int64
 #else
+#include <cstdint>
 #define TYPE int64_t
 #endif
 #define TYPENM INT64_T
