diff -bur gromacs-2025.1-o/src/external/thread_mpi/src/type.cpp gromacs-2025.1/src/external/thread_mpi/src/type.cpp
--- gromacs-2025.1-o/src/external/thread_mpi/src/type.cpp	2025-04-10 08:29:16.515520500 -0600
+++ gromacs-2025.1/src/external/thread_mpi/src/type.cpp	2025-04-10 08:29:26.557646000 -0600
@@ -163,6 +163,7 @@
 #ifdef _MSC_VER
 #define TYPE __int64
 #else
+#include <cstdint>
 #define TYPE int64_t
 #endif
 #define TYPENM INT64_T
