diff -aur 000/src/download.cpp 001/src/download.cpp
--- 000/src/download.cpp	2015-04-12 07:10:48.357386800 -0300
+++ 001/src/download.cpp	2015-04-12 07:11:26.623061500 -0300
@@ -152,7 +152,7 @@
                           NULL, // lpszHeaders
                           -1,   // dwHeadersLength
                           dwFlags,
-                          NULL  // dwContext
+                          0     // dwContext
                       );
     if ( !conn )
         throw Win32Exception();
diff -aur 000/src/ui.cpp 001/src/ui.cpp
--- 000/src/ui.cpp	2015-04-12 07:10:48.388636400 -0300
+++ 001/src/ui.cpp	2015-04-12 07:12:39.286851600 -0300
@@ -1339,7 +1339,6 @@
     SignalReady();
 
     // Run the app:
-    wxMSWDisableSettingHighDPIAware();
     wxEntry(ms_hInstance);
 }
 
diff -aur 000/src/utils.h 001/src/utils.h
--- 000/src/utils.h	2015-04-12 07:10:48.404261800 -0300
+++ 001/src/utils.h	2015-04-12 07:11:26.638687500 -0300
@@ -55,7 +55,7 @@
     std::basic_string<TOut> out;
     out.reserve(s.length());
 
-    for ( std::basic_string<TIn>::const_iterator i = s.begin(); i != s.end(); ++i )
+    for ( typename std::basic_string<TIn>::const_iterator i = s.begin(); i != s.end(); ++i )
     {
         out += static_cast<TOut>(*i);
     }
