

From: David Rothenberger <daveroth@acm.org>


---
 common/plugin_common.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/common/plugin_common.h b/common/plugin_common.h
index 0f75897..3e844c7 100644
--- a/common/plugin_common.h
+++ b/common/plugin_common.h
@@ -63,7 +63,7 @@
 #include <saslutil.h>
 #include <saslplug.h>
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 #define PLUG_API __declspec(dllexport)
 #else
 #define PLUG_API extern
