diff -urN a/libsasl2.pc.in b/libsasl2.pc.in
--- a/libsasl2.pc.in	2014-08-12 22:38:23.568094200 +0100
+++ b/libsasl2.pc.in	2014-08-12 22:39:00.721219300 +0100
@@ -1,4 +1,7 @@
-libdir = @libdir@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
 
 Name: Cyrus SASL
 Description: Cyrus SASL implementation
@@ -6,3 +9,4 @@
 Version: @VERSION@
 Libs: -L${libdir} -lsasl2
 Libs.private: @LIB_DOOR@ @SASL_DL_LIB@ @LIBS@
+Cflags: -I${includedir}
