diff -urN a/lib/saslutil.c b/lib/saslutil.c
--- a/lib/saslutil.c	2014-08-12 22:40:28.481238900 +0100
+++ b/lib/saslutil.c	2014-08-12 22:40:29.324287100 +0100
@@ -607,7 +607,7 @@
     return (0);
 }
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW64_VERSION_MAJOR)
 /***************************************************************************** 
  * 
  *  MODULE NAME : GETOPT.C 
diff -urN a/utils/sasldblistusers.c b/utils/sasldblistusers.c
--- a/utils/sasldblistusers.c	2014-08-12 22:40:28.537242100 +0100
+++ b/utils/sasldblistusers.c	2014-08-12 22:40:29.325287100 +0100
@@ -57,9 +57,13 @@
 
 #ifdef WIN32
 #include <saslutil.h>
+#if defined(__MINGW64_VERSION_MAJOR)
+#include <getopt.h>
+#else
 __declspec(dllimport) char *optarg;
 __declspec(dllimport) int optind;
 #endif
+#endif
 
 /* Cheating to make the utils work out right */
 LIBSASL_VAR const sasl_utils_t *sasl_global_utils;
diff -urN a/utils/saslpasswd.c b/utils/saslpasswd.c
--- a/utils/saslpasswd.c	2014-08-12 22:40:28.539242200 +0100
+++ b/utils/saslpasswd.c	2014-08-12 22:40:29.326287200 +0100
@@ -58,9 +58,12 @@
 #include <io.h>
 
 #include <saslutil.h>
+#if defined(__MINGW64_VERSION_MAJOR)
+#include <getopt.h>
+#else
 __declspec(dllimport) char *optarg;
 __declspec(dllimport) int optind;
-
+#endif
 /* perror can't be used on Windows system calls, so we define a new macro to underline this */
 void p_oserror (const char *string);
 #endif /*WIN32*/
