diff -urN cyrus-sasl-2.1.26.orig/cmulocal/bsd_sockets.m4 cyrus-sasl-2.1.26/cmulocal/bsd_sockets.m4
--- cyrus-sasl-2.1.26.orig/cmulocal/bsd_sockets.m4	2014-08-13 22:23:19.991836200 +0100
+++ cyrus-sasl-2.1.26/cmulocal/bsd_sockets.m4	2014-08-13 22:23:35.042697100 +0100
@@ -8,7 +8,26 @@
 
 AC_DEFUN([CMU_SOCKETS], [
 	save_LIBS="$LIBS"
-	LIB_SOCKET=""
+	AC_CHECK_HEADERS([sys/socket.h ws2tcpip.h])
+	AC_CHECK_FUNC(socket, , [
+		AC_CHECK_LIB(socket, socket, [LIB_SOCKET=-lsocket], [
+			LIBS="$LIBS -lws2_32"
+			AC_LINK_IFELSE([
+				AC_LANG_PROGRAM([[
+					#ifdef HAVE_SYS_SOCKET_H
+					#	include <sys/socket.h>
+					#endif
+					#ifdef HAVE_WS2TCPIP_H
+					#	include <ws2tcpip.h>
+					#endif
+				]], [[return socket(0, 0, 0);]])
+			],
+			[LIB_SOCKET=-lws2_32
+			 AC_MSG_RESULT(yes)],
+			[AC_MSG_ERROR([socket not found])])
+		])
+	])
+	LIBS="$save_LIBS"
 	AC_CHECK_FUNC(connect, :,
 		[AC_CHECK_LIB(nsl, gethostbyname,
 			     LIB_SOCKET="-lnsl $LIB_SOCKET")
diff -urN cyrus-sasl-2.1.26.orig/plugins/Makefile.am cyrus-sasl-2.1.26/plugins/Makefile.am
--- cyrus-sasl-2.1.26.orig/plugins/Makefile.am	2014-08-13 22:23:20.118843500 +0100
+++ cyrus-sasl-2.1.26/plugins/Makefile.am	2014-08-13 22:23:35.043697100 +0100
@@ -71,11 +71,11 @@
 
 libplain_la_SOURCES = plain.c plain_init.c $(common_sources)
 libplain_la_DEPENDENCIES = $(COMPAT_OBJS)
-libplain_la_LIBADD = $(PLAIN_LIBS) $(COMPAT_OBJS)
+libplain_la_LIBADD = $(PLAIN_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libanonymous_la_SOURCES = anonymous.c anonymous_init.c $(common_sources)
 libanonymous_la_DEPENDENCIES = $(COMPAT_OBJS)
-libanonymous_la_LIBADD = $(COMPAT_OBJS)
+libanonymous_la_LIBADD = $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libkerberos4_la_SOURCES = kerberos4.c kerberos4_init.c $(common_sources)
 libkerberos4_la_DEPENDENCIES = $(COMPAT_OBJS)
@@ -91,7 +91,7 @@
 
 libcrammd5_la_SOURCES = cram.c crammd5_init.c $(common_sources)
 libcrammd5_la_DEPENDENCIES = $(COMPAT_OBJS)
-libcrammd5_la_LIBADD = $(COMPAT_OBJS)
+libcrammd5_la_LIBADD = $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libdigestmd5_la_SOURCES = digestmd5.c digestmd5_init.c $(common_sources)
 libdigestmd5_la_DEPENDENCIES = $(COMPAT_OBJS)
@@ -99,32 +99,32 @@
 
 libscram_la_SOURCES = scram.c scram_init.c $(common_sources)
 libscram_la_DEPENDENCIES = $(COMPAT_OBJS)
-libscram_la_LIBADD = $(SCRAM_LIBS) $(COMPAT_OBJS)
+libscram_la_LIBADD = $(SCRAM_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 liblogin_la_SOURCES = login.c login_init.c $(common_sources)
 liblogin_la_DEPENDENCIES = $(COMPAT_OBJS)
-liblogin_la_LIBADD = $(PLAIN_LIBS) $(COMPAT_OBJS)
+liblogin_la_LIBADD = $(PLAIN_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libsrp_la_SOURCES = srp.c srp_init.c $(common_sources)
 libsrp_la_DEPENDENCIES = $(COMPAT_OBJS)
-libsrp_la_LIBADD = $(SRP_LIBS) $(COMPAT_OBJS)
+libsrp_la_LIBADD = $(SRP_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libotp_la_SOURCES = otp.c otp_init.c otp.h $(common_sources)
 libotp_la_DEPENDENCIES = $(COMPAT_OBJS)
-libotp_la_LIBADD = $(OTP_LIBS) $(COMPAT_OBJS)
+libotp_la_LIBADD = $(OTP_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libntlm_la_SOURCES = ntlm.c ntlm_init.c $(common_sources)
 libntlm_la_DEPENDENCIES = $(COMPAT_OBJS)
-libntlm_la_LIBADD = $(NTLM_LIBS) $(COMPAT_OBJS)
+libntlm_la_LIBADD = $(NTLM_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libpassdss_la_SOURCES = passdss.c passdss_init.c $(common_sources)
 libpassdss_la_DEPENDENCIES = $(COMPAT_OBJS)
-libpassdss_la_LIBADD = $(PASSDSS_LIBS) $(COMPAT_OBJS)
+libpassdss_la_LIBADD = $(PASSDSS_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 # Auxprop Plugins
 libsasldb_la_SOURCES = sasldb.c sasldb_init.c $(common_sources)
 libsasldb_la_DEPENDENCIES = $(COMPAT_OBJS)
-libsasldb_la_LIBADD = ../sasldb/libsasldb.la $(SASL_DB_LIB) $(COMPAT_OBJS)
+libsasldb_la_LIBADD = ../sasldb/libsasldb.la $(SASL_DB_LIB) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libldapdb_la_SOURCES = ldapdb.c ldapdb_init.c $(common_sources)
 libldapdb_la_DEPENDENCIES = $(COMPAT_OBJS)
