diff -urN cyrus-sasl-2.1.26.orig/configure.in cyrus-sasl-2.1.26/configure.in
--- cyrus-sasl-2.1.26.orig/configure.in	2014-08-10 21:45:35.049028300 +0100
+++ cyrus-sasl-2.1.26/configure.in	2014-08-10 21:45:47.641748500 +0100
@@ -1045,6 +1045,14 @@
 esac
 AM_CONDITIONAL(MACOSX, test "$building_for_macosx" = yes)
 
+building_on_msys2_cygwin=no
+case "$build_os" in
+        mingw*|cygwin*)
+          building_on_msys2_cygwin=yes
+        ;;
+esac
+AM_CONDITIONAL(MSYS2_CYGWIN, test "$building_on_msys2_cygwin" = yes)
+
 dnl dmalloc tests
 AC_MSG_CHECKING(for dmalloc library)
 AC_ARG_WITH(dmalloc, [  --with-dmalloc=DIR      with DMALLOC support (for test applications) [[no]] ],
diff -urN cyrus-sasl-2.1.26.orig/lib/Makefile.am cyrus-sasl-2.1.26/lib/Makefile.am
--- cyrus-sasl-2.1.26.orig/lib/Makefile.am	2014-08-10 21:45:35.042027900 +0100
+++ cyrus-sasl-2.1.26/lib/Makefile.am	2014-08-10 21:45:47.670750200 +0100
@@ -100,5 +100,5 @@
 $(SASL_STATIC_SRCS): linksrcs
 
 linksrcs:
-	-ln -s $(SASL_STATIC_SRCS) .
+	-$(LN_S) $(SASL_STATIC_SRCS) .
 
diff -urN cyrus-sasl-2.1.26.orig/plugins/Makefile.am cyrus-sasl-2.1.26/plugins/Makefile.am
--- cyrus-sasl-2.1.26.orig/plugins/Makefile.am	2014-08-10 21:45:35.018026500 +0100
+++ cyrus-sasl-2.1.26/plugins/Makefile.am	2014-08-10 21:48:51.712276800 +0100
@@ -53,6 +53,11 @@
 AM_LDFLAGS = -module -export-dynamic -rpath $(plugindir) -version-info $(plugin_version) -no-undefined
 
 COMPAT_OBJS = @LTGETADDRINFOOBJS@ @LTGETNAMEINFOOBJS@ @LTSNPRINTFOBJS@
+if MSYS2_CYGWIN
+COMPAT_OBJS_REAL = @GETADDRINFOOBJS@ @GETNAMEINFOOBJS@ @SNPRINTFOBJS@
+else
+COMPAT_OBJS_REAL =
+endif
 
 EXTRA_DIST = makeinit.sh NTMakefile
 noinst_SCRIPTS = makeinit.sh
@@ -153,6 +158,7 @@
 $(COMPAT_OBJS):
 	rm -f $(COMPAT_OBJS)
 	cd ../lib; $(MAKE) $(COMPAT_OBJS)
-	for file in $(COMPAT_OBJS); do ln -s ../lib/$$file .; done
-
+	for file in $(COMPAT_OBJS); do $(LN_S) ../lib/$$file .; done
+	-[ -d .libs ] || mkdir .libs
+	for file in $(COMPAT_OBJS_REAL); do $(LN_S) ../lib/.libs/$$file .libs/; done
 
