diff -urN a/configure.in b/configure.in
--- a/configure.in	2014-08-12 22:40:33.195508500 +0100
+++ b/configure.in	2014-08-12 22:40:33.842545500 +0100
@@ -1045,6 +1045,14 @@
 esac
 AM_CONDITIONAL(MACOSX, test "$building_for_macosx" = yes)
 
+building_on_msys2_cygwin=no
+case "$build_os" in
+        mingw*|cygwin*)
+          build_os_msys2_cygwin=yes
+        ;;
+esac
+AM_CONDITIONAL(BUILD_OS_MSYS2_CYGWIN, test "$build_os_msys2_cygwin" = yes)
+
 dnl dmalloc tests
 AC_MSG_CHECKING(for dmalloc library)
 AC_ARG_WITH(dmalloc, [  --with-dmalloc=DIR      with DMALLOC support (for test applications) [[no]] ],
diff -urN a/lib/Makefile.am b/lib/Makefile.am
--- a/lib/Makefile.am	2014-08-12 22:40:33.204509000 +0100
+++ b/lib/Makefile.am	2014-08-12 22:40:33.844545600 +0100
@@ -100,5 +100,5 @@
 $(SASL_STATIC_SRCS): linksrcs
 
 linksrcs:
-	-ln -s $(SASL_STATIC_SRCS) .
+	-$(LN_S) $(SASL_STATIC_SRCS) .
 
diff -urN a/plugins/Makefile.am b/plugins/Makefile.am
--- a/plugins/Makefile.am	2014-08-12 22:40:33.277513200 +0100
+++ b/plugins/Makefile.am	2014-08-12 22:40:33.845545700 +0100
@@ -53,6 +53,11 @@
 AM_LDFLAGS = -module -export-dynamic -rpath $(plugindir) -version-info $(plugin_version) -no-undefined
 
 COMPAT_OBJS = @LTGETADDRINFOOBJS@ @LTGETNAMEINFOOBJS@ @LTSNPRINTFOBJS@
+if BUILD_OS_MSYS2_CYGWIN
+COMPAT_OBJS_REAL = @GETADDRINFOOBJS@ @GETNAMEINFOOBJS@ @SNPRINTFOBJS@
+else
+COMPAT_OBJS_REAL =
+endif
 
 EXTRA_DIST = makeinit.sh NTMakefile
 noinst_SCRIPTS = makeinit.sh
@@ -153,6 +158,7 @@
 $(COMPAT_OBJS):
 	rm -f $(COMPAT_OBJS)
 	cd ../lib; $(MAKE) $(COMPAT_OBJS)
-	for file in $(COMPAT_OBJS); do ln -s ../lib/$$file .; done
-
+	for file in $(COMPAT_OBJS); do $(LN_S) ../lib/$$file .; done
+	-[ -d .libs ] || mkdir .libs
+	for file in $(COMPAT_OBJS_REAL); do $(LN_S) ../lib/.libs/$$file .libs/; done
 
