diff -urN cyrus-sasl-2.1.26.orig/sasldb/db_berkeley.c cyrus-sasl-2.1.26/sasldb/db_berkeley.c
--- cyrus-sasl-2.1.26.orig/sasldb/db_berkeley.c	2012-01-27 23:31:36.000000000 +0000
+++ cyrus-sasl-2.1.26/sasldb/db_berkeley.c	2014-08-12 00:42:11.645836800 +0100
@@ -358,7 +358,7 @@
   return result;
 }
 
-int _sasl_check_db(const sasl_utils_t *utils,
+LIBSASL_API int _sasl_check_db(const sasl_utils_t *utils,
 		   sasl_conn_t *conn)
 {
     const char *path = SASL_DB_PATH;
@@ -412,7 +412,7 @@
     DBC *cursor;
 } berkleyhandle_t;
 
-sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
+LIBSASL_API sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
 				   sasl_conn_t *conn) 
 {
     int ret;
@@ -447,7 +447,7 @@
     return (sasldb_handle)handle;
 }
 
-int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
+LIBSASL_API int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
 		       sasldb_handle handle, char *out,
 		       const size_t max_out, size_t *out_len) 
 {
@@ -501,7 +501,7 @@
 }
 
 
-int _sasldb_releasekeyhandle(const sasl_utils_t *utils,
+LIBSASL_API int _sasldb_releasekeyhandle(const sasl_utils_t *utils,
 			     sasldb_handle handle) 
 {
     berkleyhandle_t *dbh = (berkleyhandle_t *)handle;
diff -urN cyrus-sasl-2.1.26.orig/sasldb/db_gdbm.c cyrus-sasl-2.1.26/sasldb/db_gdbm.c
--- cyrus-sasl-2.1.26.orig/sasldb/db_gdbm.c	2012-01-27 23:31:36.000000000 +0000
+++ cyrus-sasl-2.1.26/sasldb/db_gdbm.c	2014-08-12 00:42:11.645836800 +0100
@@ -221,7 +221,7 @@
   return result;
 }
 
-int _sasl_check_db(const sasl_utils_t *utils,
+LIBSASL_API int _sasl_check_db(const sasl_utils_t *utils,
 		   sasl_conn_t *conn)
 {
     const char *path = SASL_DB_PATH;
@@ -270,7 +270,7 @@
     int first;
 } handle_t;
 
-sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
+LIBSASL_API sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
 				   sasl_conn_t *conn) 
 {
     const char *path = SASL_DB_PATH;
@@ -316,7 +316,7 @@
     return (sasldb_handle)handle;
 }
 
-int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
+LIBSASL_API int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
 		       sasldb_handle handle, char *out,
 		       const size_t max_out, size_t *out_len) 
 {
@@ -346,7 +346,7 @@
     return SASL_CONTINUE;
 }
 
-int _sasldb_releasekeyhandle(const sasl_utils_t *utils,
+LIBSASL_API int _sasldb_releasekeyhandle(const sasl_utils_t *utils,
 			     sasldb_handle handle) 
 {
     handle_t *dbh = (handle_t *)handle;
diff -urN cyrus-sasl-2.1.26.orig/sasldb/db_ndbm.c cyrus-sasl-2.1.26/sasldb/db_ndbm.c
--- cyrus-sasl-2.1.26.orig/sasldb/db_ndbm.c	2014-08-11 17:22:15.615410300 +0100
+++ cyrus-sasl-2.1.26/sasldb/db_ndbm.c	2014-08-11 17:22:24.933410300 +0100
@@ -221,7 +221,7 @@
 #define SUFLEN 5
 #endif
 
-int _sasl_check_db(const sasl_utils_t *utils,
+LIBSASL_API int _sasl_check_db(const sasl_utils_t *utils,
 		   sasl_conn_t *conn)
 {
     const char *path = SASL_DB_PATH;
@@ -294,7 +294,7 @@
     int first;
 } handle_t;
 
-sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
+LIBSASL_API sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
 				   sasl_conn_t *conn) 
 {
     const char *path = SASL_DB_PATH;
@@ -339,7 +339,7 @@
     return (sasldb_handle)handle;
 }
 
-int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
+LIBSASL_API int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
 		       sasldb_handle handle, char *out,
 		       const size_t max_out, size_t *out_len) 
 {
@@ -369,7 +369,7 @@
     return SASL_CONTINUE;
 }
 
-int _sasldb_releasekeyhandle(const sasl_utils_t *utils,
+LIBSASL_API int _sasldb_releasekeyhandle(const sasl_utils_t *utils,
 			     sasldb_handle handle) 
 {
     handle_t *dbh = (handle_t *)handle;
diff -urN cyrus-sasl-2.1.26.orig/sasldb/db_none.c cyrus-sasl-2.1.26/sasldb/db_none.c
--- cyrus-sasl-2.1.26.orig/sasldb/db_none.c	2014-08-11 17:22:15.613410300 +0100
+++ cyrus-sasl-2.1.26/sasldb/db_none.c	2014-08-11 17:22:24.926410300 +0100
@@ -73,21 +73,21 @@
     return SASL_FAIL;
 }
 
-int _sasl_check_db(const sasl_utils_t *utils,
+LIBSASL_API int _sasl_check_db(const sasl_utils_t *utils,
 		   sasl_conn_t *conn)
 {
     if(conn) utils->seterror(conn, 0, "No Database Driver");
     return SASL_FAIL;
 }
 
-sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
+LIBSASL_API sasldb_handle _sasldb_getkeyhandle(const sasl_utils_t *utils,
                                    sasl_conn_t *conn) 
 {
     if(conn) utils->seterror(conn, 0, "No Database Driver");
     return NULL;
 }
 
-int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
+LIBSASL_API int _sasldb_getnextkey(const sasl_utils_t *utils __attribute__((unused)),
                        sasldb_handle handle __attribute__((unused)),
 		       char *out __attribute__((unused)),
                        const size_t max_out __attribute__((unused)),
@@ -96,7 +96,7 @@
     return SASL_FAIL;
 }
 
-int _sasldb_releasekeyhandle(const sasl_utils_t *utils __attribute__((unused)),
+LIBSASL_API int _sasldb_releasekeyhandle(const sasl_utils_t *utils __attribute__((unused)),
                              sasldb_handle handle __attribute__((unused)))  
 {
     return SASL_FAIL;
