diff -bur vrpn-07.35-orig/vrpn_Vality.C vrpn-07.35/vrpn_Vality.C
--- vrpn-07.35-orig/vrpn_Vality.C	2024-02-28 22:30:45.245106200 -0700
+++ vrpn-07.35/vrpn_Vality.C	2024-02-28 22:30:47.112562100 -0700
@@ -3,6 +3,7 @@
 #include <stdio.h>                      // for fprintf, stderr, NULL
 #include <string.h>                     // for memset
 #include <math.h>                       // for fabs
+#include <stdint.h>                     // for uint8_t
 
 #include "vrpn_Vality.h"
 #include "vrpn_MessageMacros.h"         // for VRPN_MSG_INFO, VRPN_MSG_WARNING, VRPN_MSG_ERROR
