diff --git a/mingw-w64-headers/include/windows.ui.composition.h b/mingw-w64-headers/include/windows.ui.composition.h
index faf73ea0b..76b4c2172 100644
--- a/mingw-w64-headers/include/windows.ui.composition.h
+++ b/mingw-w64-headers/include/windows.ui.composition.h
@@ -4916,10 +4916,10 @@ namespace ABI {
                 ICompositionDrawingSurface : public IInspectable
                 {
                     virtual HRESULT STDMETHODCALLTYPE get_AlphaMode(
-                        enum DirectXAlphaMode *value) = 0;
+                        enum Windows::Graphics::DirectX::DirectXAlphaMode *value) = 0;
 
                     virtual HRESULT STDMETHODCALLTYPE get_PixelFormat(
-                        enum DirectXPixelFormat *value) = 0;
+                        enum Windows::Graphics::DirectX::DirectXPixelFormat *value) = 0;
 
                     virtual HRESULT STDMETHODCALLTYPE get_Size(
                         struct Size *value) = 0;
@@ -5704,8 +5704,8 @@ namespace ABI {
                 {
                     virtual HRESULT STDMETHODCALLTYPE CreateDrawingSurface(
                         struct Size pixels,
-                        enum DirectXPixelFormat format,
-                        enum DirectXAlphaMode mode,
+                        enum Windows::Graphics::DirectX::DirectXPixelFormat format,
+                        enum Windows::Graphics::DirectX::DirectXAlphaMode mode,
                         ABI::Windows::UI::Composition::ICompositionDrawingSurface **result) = 0;
 
                     virtual HRESULT STDMETHODCALLTYPE add_RenderingDeviceReplaced(
