diff --git a/mingw-w64-headers/include/xaudio2.h b/mingw-w64-headers/include/xaudio2.h
index ca611de33..f26e03911 100644
--- a/mingw-w64-headers/include/xaudio2.h
+++ b/mingw-w64-headers/include/xaudio2.h
@@ -1570,12 +1570,12 @@ IXAudio2 : public IUnknown
 
     virtual HRESULT STDMETHODCALLTYPE CreateMasteringVoice(
         IXAudio2MasteringVoice **ppMasteringVoice,
-        UINT32 InputChannels,
-        UINT32 InputSampleRate,
-        UINT32 Flags,
-        LPCWSTR DeviceId,
-        const XAUDIO2_EFFECT_CHAIN *pEffectChain,
-        AUDIO_STREAM_CATEGORY category) = 0;
+        UINT32 InputChannels = XAUDIO2_DEFAULT_CHANNELS,
+        UINT32 InputSampleRate = XAUDIO2_DEFAULT_SAMPLERATE,
+        UINT32 Flags = 0,
+        LPCWSTR DeviceId = 0,
+        const XAUDIO2_EFFECT_CHAIN *pEffectChain = 0,
+        AUDIO_STREAM_CATEGORY category = AudioCategory_GameEffects) = 0;
 
     virtual HRESULT STDMETHODCALLTYPE StartEngine(
         ) = 0;
diff --git a/mingw-w64-headers/include/xaudio2.idl b/mingw-w64-headers/include/xaudio2.idl
index d0e781fe1..277fc4504 100644
--- a/mingw-w64-headers/include/xaudio2.idl
+++ b/mingw-w64-headers/include/xaudio2.idl
@@ -555,7 +555,7 @@ interface IXAudio2 : IUnknown
 #endif
         [in, defaultvalue(NULL)] const XAUDIO2_EFFECT_CHAIN* pEffectChain
 #if XAUDIO2_VER >= 8
-        , AUDIO_STREAM_CATEGORY category
+        , [in, defaultvalue(AudioCategory_GameEffects)] AUDIO_STREAM_CATEGORY category
 #endif
         );
 
