--- kdstatemachineeditor-1.2.8/cmake/FindGraphviz.cmake.orig	2021-11-18 12:18:52.899526300 +0300
+++ kdstatemachineeditor-1.2.8/cmake/FindGraphviz.cmake	2021-11-18 12:19:44.470646700 +0300
@@ -59,7 +59,7 @@
   HINTS ${_GRAPHVIZ_INCLUDE_DIR}
   ${_GRAPHVIZ_FIND_OPTS})
 
-if(WIN32)
+if(MSVC)
   if(CMAKE_BUILD_TYPE STREQUAL "Release")
     set(GRAPHVIZ_LIB_PATH_SUFFIX "release/lib")
   else()
@@ -101,7 +101,7 @@
 
 # Ok, now compute the version and make sure its greater then the min required
 if(GRAPHVIZ_FOUND)
-  if(NOT WIN32)
+  if(NOT MSVC)
     set(FIND_GRAPHVIZ_VERSION_SOURCE
       "#include <graphviz/graphviz_version.h>\n#include <stdio.h>\n int main()\n {\n printf(\"%s\",PACKAGE_VERSION);return 1;\n }\n")
     set(FIND_GRAPHVIZ_VERSION_SOURCE_FILE ${CMAKE_BINARY_DIR}/CMakeTmp/FindGRAPHVIZ.cxx)
