From eb298d1405cc604984061a66b19e0d561d0d68ab Mon Sep 17 00:00:00 2001
From: Melven Roehrig-Zoellner <Melven.Roehrig-Zoellner@DLR.de>
Date: Wed, 3 Feb 2021 23:31:13 +0100
Subject: [PATCH] Dont build and install the examples

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f34e0c..255abb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,8 +99,8 @@ if(TIXI_BUILD_TESTS)
 endif(TIXI_BUILD_TESTS)
 
 #demos
-add_subdirectory(examples/Demo)
-add_subdirectory(examples/fortran77)
+#add_subdirectory(examples/Demo)
+#add_subdirectory(examples/fortran77)
 
 # create the doc
 include(createDoc)
-- 
2.30.0.windows.2

