--- libgpg-error-1.11/src/gpg-error-config.in.orig	2013-04-21 11:42:47 +0400
+++ libgpg-error-1.11/src/gpg-error-config.in	2013-04-21 11:43:51 +0400
@@ -58,7 +58,7 @@
 	    exit 0
 	    ;;
         --cflags)
-	    if test "x$includedir" != "x/usr/include" -a "x$includedir" != "x/include"; then
+	    if test "x$includedir" != "x/usr/include" -a "x$includedir" != "x/include" -a "x$includedir" != "x/mingw/include"; then
 		output="$output -I$includedir"
 	    fi
             # Note: -idirafter is a gcc extension.  It is only used on
@@ -70,7 +70,7 @@
 	    ;;
 	--libs)
 	    case "$libdir" in
-		/usr/lib|/usr/lib64|/lib|/lib64) ;;
+		/usr/lib|/usr/lib64|/lib|/lib64|/mingw/lib|/mingw/lib64) ;;
 		*)
 		    output="$output -L$libdir"
 		    ;;
