--- libgpg-error-1.13/configure.ac.orig	2014-04-15 12:25:16.000000000 +0000
+++ libgpg-error-1.13/configure.ac	2014-05-02 23:52:02.087073700 +0000
@@ -169,27 +169,8 @@
 
 
 # Check for thread library.
-#
-# Windows has always thread support; thus we don't bother to test for
-# it as it may lead to false results when cross building.
-if test "$have_w32_system" = yes; then
-  AC_DEFINE([USE_WINDOWS_THREADS], [1])
-  LIBTHREAD=
-  LTLIBTHREAD=
-  LIBMULTITHREAD=
-  LTLIBMULTITHREAD=
-  THREADLIB_CPPFLAGS=""
-  AC_SUBST([LIBTHREAD])
-  AC_SUBST([LTLIBTHREAD])
-  AC_SUBST([LIBMULTITHREAD])
-  AC_SUBST([LTLIBMULTITHREAD])
-else
-  gl_LOCK
-  if test "$gl_threads_api" = posix; then
     AC_CHECK_SIZEOF(pthread_mutex_t,,[AC_INCLUDES_DEFAULT
                                       #include <pthread.h>])
-  fi
-fi
 
 
 # Substitution used for gpg-error-config
--- libgpg-error-1.13/src/gen-posix-lock-obj.c.orig	2014-01-28 22:47:12.000000000 +0000
+++ libgpg-error-1.13/src/gen-posix-lock-obj.c	2014-05-02 23:55:32.376277000 +0000
@@ -21,10 +21,6 @@
 #include <config.h>
 #endif
 
-#ifdef HAVE_W32_SYSTEM
-# error This module may not be build for Windows.
-#endif
-
 #include <stdlib.h>
 #include <stdio.h>
 #include <errno.h>
--- libgpg-error-1.13/src/posix-lock.c.orig	2014-01-28 22:47:12.000000000 +0000
+++ libgpg-error-1.13/src/posix-lock.c	2014-05-02 23:55:32.376277000 +0000
@@ -25,10 +25,6 @@
 #include <config.h>
 #endif
 
-#ifdef HAVE_W32_SYSTEM
-# error This module may not be build for Windows.
-#endif
-
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
--- libgpg-error-1.13/src/posix-thread.c.orig	2014-01-28 22:47:12.000000000 +0000
+++ libgpg-error-1.13/src/posix-thread.c	2014-05-02 23:55:32.376277000 +0000
@@ -25,10 +25,6 @@
 #include <config.h>
 #endif
 
-#ifdef HAVE_W32_SYSTEM
-# error This module may not be build for Windows.
-#endif
-
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
