diff --git a/CMakeLists.txt b/CMakeLists.txt
index 033544fc..4506fec1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,7 +114,7 @@ if (NOT USE_INTERNAL_SHA256)
 endif()
 
 if (WIN32)
-  if(${CMAKE_GENERATOR_PLATFORM} MATCHES "ARM64")
+  if(${CMAKE_GENERATOR_PLATFORM} MATCHES "aarch64|ARM64")
     # https://discourse.cmake.org/t/visual-studio-error-unable-to-deploy-local-file-c-x64-release-zero-check/2072
     # Target ZERO_CHECK blocks remote debugger of ARM64 in Visual Studio
     set(CMAKE_SUPPRESS_REGENERATION ON)
