--- glm-0.9.8.5/glm/simd/platform.h.orig	2018-04-30 19:33:51.796181400 -0400
+++ glm-0.9.8.5/glm/simd/platform.h	2018-04-30 19:35:26.668333700 -0400
@@ -283,6 +283,8 @@
 #		define GLM_COMPILER (GLM_COMPILER_GCC71)
 #	elif (__GNUC__ == 7) && (__GNUC_MINOR__ == 2)
 #		define GLM_COMPILER (GLM_COMPILER_GCC72)
+#	elif (__GNUC__ == 7) && (__GNUC_MINOR__ == 3)
+#		define GLM_COMPILER (GLM_COMPILER_GCC72)
 #	elif (__GNUC__ >= 8)
 #		define GLM_COMPILER (GLM_COMPILER_GCC80)
 #	else
