Index: src/lib/openjp3d/libopenjp3d.pc.cmake.in
===================================================================
--- a/src/lib/openjp3d/libopenjp3d.pc.cmake.in	(revision 0)
+++ b/src/lib/openjp3d/libopenjp3d.pc.cmake.in	(revision 2350)
@@ -0,0 +1,14 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+bindir=${prefix}/@OPENJPEG_INSTALL_BIN_DIR@
+mandir=${prefix}/@OPENJPEG_INSTALL_MAN_DIR@
+docdir=${prefix}/@OPENJPEG_INSTALL_DOC_DIR@
+libdir=${prefix}/@OPENJPEG_INSTALL_LIB_DIR@
+includedir=${prefix}/@OPENJPEG_INSTALL_INCLUDE_DIR@
+
+Name: openjp3d
+Description: JPEG2000 Extensions for three-dimensional data (Part 10)
+URL: http://www.openjpeg.org/
+Version: @OPENJPEG_VERSION@
+Libs: -L${libdir} -lopenjp3d
+Libs.private: -lm
+Cflags: -I${includedir}
Index: src/lib/openjpip/libopenjpip.pc.cmake.in
===================================================================
--- a/src/lib/openjpip/libopenjpip.pc.cmake.in	(revision 0)
+++ b/src/lib/openjpip/libopenjpip.pc.cmake.in	(revision 2350)
@@ -0,0 +1,15 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+bindir=${prefix}/@OPENJPEG_INSTALL_BIN_DIR@
+mandir=${prefix}/@OPENJPEG_INSTALL_MAN_DIR@
+docdir=${prefix}/@OPENJPEG_INSTALL_DOC_DIR@
+libdir=${prefix}/@OPENJPEG_INSTALL_LIB_DIR@
+includedir=${prefix}/@OPENJPEG_INSTALL_INCLUDE_DIR@
+
+Name: openjpip
+Description: JPEG2000 Interactivity tools, APIs and protocols (Part 9)
+URL: http://www.openjpeg.org/
+Version: @OPENJPEG_VERSION@
+Requires: openjp2
+Libs: -L${libdir} -lopenjpip
+Libs.private: -lm -lcurl -lfcgi -lpthread
+Cflags: -I${includedir}
Index: src/lib/openjp2/libopenjp2.pc.cmake.in
===================================================================
--- a/src/lib/openjp2/libopenjp2.pc.cmake.in	(revision 0)
+++ b/src/lib/openjp2/libopenjp2.pc.cmake.in	(revision 2350)
@@ -0,0 +1,13 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+bindir=${prefix}/@OPENJPEG_INSTALL_BIN_DIR@
+mandir=${prefix}/@OPENJPEG_INSTALL_MAN_DIR@
+docdir=${prefix}/@OPENJPEG_INSTALL_DOC_DIR@
+libdir=${prefix}/@OPENJPEG_INSTALL_LIB_DIR@
+includedir=${prefix}/@OPENJPEG_INSTALL_INCLUDE_DIR@
+
+Name: openjp2
+Description: JPEG2000 library (Part 1 and 2)
+URL: http://www.openjpeg.org/
+Version: @OPENJPEG_VERSION@
+Libs: -L${libdir} -lopenjp2
+Cflags: -I${includedir}
Index: src/lib/openjpwl/libopenjpwl.pc.cmake.in
===================================================================
--- a/src/lib/openjpwl/libopenjpwl.pc.cmake.in	(revision 0)
+++ b/src/lib/openjpwl/libopenjpwl.pc.cmake.in	(revision 2350)
@@ -0,0 +1,15 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+bindir=${prefix}/@OPENJPEG_INSTALL_BIN_DIR@
+mandir=${prefix}/@OPENJPEG_INSTALL_MAN_DIR@
+docdir=${prefix}/@OPENJPEG_INSTALL_DOC_DIR@
+libdir=${prefix}/@OPENJPEG_INSTALL_LIB_DIR@
+includedir=${prefix}/@OPENJPEG_INSTALL_INCLUDE_DIR@
+
+Name: openjpwl
+Description: JPEG2000 Wireless library (Part 11)
+URL: http://www.openjpeg.org/
+Version: @OPENJPEG_VERSION@
+Requires: openjp2
+Libs: -L${libdir} -lopenjpwl
+Libs.private: -lm
+Cflags: -I${includedir}
Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt	(revision 2226)
+++ b/CMakeLists.txt	(revision 2350)
@@ -287,3 +285,38 @@
 install(FILES LICENSE DESTINATION ${OPENJPEG_INSTALL_DOC_DIR})
 
 include (cmake/OpenJPEGCPack.cmake)
+
+#-----------------------------------------------------------------------------
+# pkgconfig support
+
+  # install in lib and not share (see multi-arch note above)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/lib/openjp2/libopenjp2.pc.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/libopenjp2.pc @ONLY)
+  install( FILES  ${CMAKE_CURRENT_BINARY_DIR}/libopenjp2.pc DESTINATION
+    ${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig )
+#
+  if(BUILD_JPWL)
+  # install in lib and not share (see multi-arch note above)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/lib/openjpwl/libopenjpwl.pc.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/libopenjpwl.pc @ONLY)
+  install( FILES  ${CMAKE_CURRENT_BINARY_DIR}/libopenjpwl.pc DESTINATION
+    ${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig )
+  endif()
+#
+  if(BUILD_JPIP)
+  # install in lib and not share (see multi-arch note above)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/lib/openjpip/libopenjpip.pc.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/libopenjpip.pc @ONLY)
+  install( FILES  ${CMAKE_CURRENT_BINARY_DIR}/libopenjpip.pc DESTINATION
+    ${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig )
+  endif()
+#
+  if(BUILD_JP3D)
+  # install in lib and not share (see multi-arch note above)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/lib/openjp3d/libopenjp3d.pc.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/libopenjp3d.pc @ONLY)
+  install( FILES  ${CMAKE_CURRENT_BINARY_DIR}/libopenjp3d.pc DESTINATION
+    ${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig )
+  endif()
+
+#-----------------------------------------------------------------------------
