--- openjpeg-2.3.0/CMakeLists.txt.orig	2018-08-02 12:10:38.105864300 -0400
+++ openjpeg-2.3.0/CMakeLists.txt	2018-08-02 12:12:33.009046900 -0400
@@ -150,7 +150,7 @@
   # We could install *.cmake files in share/ however those files contains
   # hardcoded path to libraries on a multi-arch system (fedora/debian) those
   # path will be different (lib/i386-linux-gnu vs lib/x86_64-linux-gnu)
-  set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/${OPENJPEG_INSTALL_SUBDIR}")
+  set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/cmake/${OPENJPEG_INSTALL_SUBDIR}")
 endif()
 
 if (APPLE)
--- openjpeg-2.3.0/cmake/OpenJPEGConfig.cmake.in.orig	2018-08-21 10:42:11.707314700 +0300
+++ openjpeg-2.3.0/cmake/OpenJPEGConfig.cmake.in	2018-08-21 10:42:18.490702700 +0300
@@ -26,7 +26,7 @@
 if(EXISTS ${SELF_DIR}/OpenJPEGTargets.cmake)
   # This is an install tree
   include(${SELF_DIR}/OpenJPEGTargets.cmake)
-  get_filename_component(OPENJPEG_INCLUDE_ROOT "${SELF_DIR}/../../@OPENJPEG_INSTALL_INCLUDE_DIR@" ABSOLUTE)
+  get_filename_component(OPENJPEG_INCLUDE_ROOT "${SELF_DIR}/../../../@OPENJPEG_INSTALL_INCLUDE_DIR@" ABSOLUTE)
   set(OPENJPEG_INCLUDE_DIRS ${OPENJPEG_INCLUDE_ROOT})
 
 else()
