--- openjpeg-2.0.0/src/lib/openjp2/CMakeLists.txt.orig	2012-11-19 14:26:08.000000000 +0000
+++ openjpeg-2.0.0/src/lib/openjp2/CMakeLists.txt	2014-02-15 03:07:56.149288400 +0000
@@ -53,7 +53,7 @@
 if(UNIX)
   target_link_libraries(${OPENJPEG_LIBRARY_NAME} m)
 endif()
-set_target_properties(${OPENJPEG_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
+set_target_properties(${OPENJPEG_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES} OUTPUT_NAME openjp2 RUNTIME_OUTPUT_NAME openjp2-${OPENJPEG_SOVERSION} ARCHIVE_OUTPUT_NAME openjp2)
 
 # Install library
 install(TARGETS ${OPENJPEG_LIBRARY_NAME}
--- openjpeg-2.0.0/src/lib/openjp3d/CMakeLists.txt.orig	2014-02-15 02:49:31.533520000 +0000
+++ openjpeg-2.0.0/src/lib/openjp3d/CMakeLists.txt	2014-02-15 03:07:20.589772900 +0000
@@ -26,7 +26,7 @@
 if(UNIX)
   target_link_libraries(${OPENJP3D_LIBRARY_NAME} m)
 endif()
-set_target_properties(${OPENJP3D_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
+set_target_properties(${OPENJP3D_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES} OUTPUT_NAME openjp3d RUNTIME_OUTPUT_NAME openjp3d-${OPENJPEG_SOVERSION} ARCHIVE_OUTPUT_NAME openjp3d)
 
 # Install library
 install(TARGETS ${OPENJP3D_LIBRARY_NAME}
--- openjpeg-2.0.0/src/lib/openjpip/CMakeLists.txt.orig	2012-11-19 14:26:08.000000000 +0000
+++ openjpeg-2.0.0/src/lib/openjpip/CMakeLists.txt	2014-02-15 03:07:49.780979700 +0000
@@ -60,7 +60,7 @@
 endif()
 add_library(openjpip ${OPENJPIP_SRCS} ${LOCAL_SRCS})
 set_target_properties(openjpip
-  PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
+  PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES} OUTPUT_NAME openjpip RUNTIME_OUTPUT_NAME openjpip-${OPENJPEG_SOVERSION} ARCHIVE_OUTPUT_NAME openjpip)
 target_link_libraries(openjpip ${OPENJPEG_LIBRARY_NAME})
 if(WIN32)
   # add Winsock on windows+mingw
--- openjpeg-2.0.0/src/lib/openjpwl/CMakeLists.txt.orig	2012-11-19 14:26:08.000000000 +0000
+++ openjpeg-2.0.0/src/lib/openjpwl/CMakeLists.txt	2014-02-15 03:08:14.749150300 +0000
@@ -50,7 +50,7 @@
   target_link_libraries(openjpwl m)
 endif()
 set_target_properties(openjpwl
-  PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
+  PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES} OUTPUT_NAME openjpwl RUNTIME_OUTPUT_NAME openjpwl-${OPENJPEG_SOVERSION} ARCHIVE_OUTPUT_NAME openjpwl)
 
 # Install library
 install(TARGETS openjpwl
--- openjpeg-2.0.0/src/lib/openmj2/CMakeLists.txt.orig	2014-02-15 02:49:31.530519700 +0000
+++ openjpeg-2.0.0/src/lib/openmj2/CMakeLists.txt	2014-02-15 03:08:51.649836100 +0000
@@ -45,7 +45,7 @@
 if(UNIX)
   target_link_libraries(${OPENMJ2_LIBRARY_NAME} m)
 endif()
-set_target_properties(${OPENMJ2_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
+set_target_properties(${OPENMJ2_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES} OUTPUT_NAME openmj2 RUNTIME_OUTPUT_NAME openmj2-${OPENJPEG_SOVERSION} ARCHIVE_OUTPUT_NAME openmj2)
 
 # Install library
 install(TARGETS ${OPENMJ2_LIBRARY_NAME}
