--- a/src/bin/jp3d/convert.c
+++ b/src/bin/jp3d/convert.c
@@ -581,7 +581,7 @@
     f = fopen(filename, "rb");
     if (!f) {
         fprintf(stdout, "[ERROR] Failed to open %s for reading !!\n", filename);
-        opj_free(volume);
+        free(volume);
         return 0;
     }
 
@@ -919,7 +919,7 @@
     f = fopen(filename, "rb");
     if (!f) {
         fprintf(stderr, "[ERROR] Failed to open %s for reading !!\n", filename);
-        opj_free(volume);
+        free(volume);
         return 0;
     }
 
