--- openjpeg-2.3.0/CMakeLists.txt.orig	2018-08-23 01:11:20.224502000 -0400
+++ openjpeg-2.3.0/CMakeLists.txt	2018-08-23 01:13:11.209812900 -0400
@@ -265,6 +265,7 @@ option(BUILD_JPWL "Build the JPWL librar
 option(BUILD_JPIP "Build the JPIP library and executables." OFF)
 if(BUILD_JPIP)
   option(BUILD_JPIP_SERVER "Build the JPIP server." OFF)
+  option(BUILD_JPIP_CLIENT "Build the JPIP client (Java)." OFF)
 endif()
 option(BUILD_VIEWER "Build the OPJViewer executable (C++)" OFF)
 option(BUILD_JAVA "Build the openjpeg jar (Java)" OFF)
--- openjpeg-2.3.0/src/bin/jpip/CMakeLists.txt.orig	2018-08-23 01:13:49.781712900 -0400
+++ openjpeg-2.3.0/src/bin/jpip/CMakeLists.txt	2018-08-23 01:15:11.096739600 -0400
@@ -56,6 +56,7 @@ add_executable(${exe} ${exe}.c)
     )
 endforeach()
 
+if(BUILD_JPIP_CLIENT)
 # Build the two java clients:
 find_package(Java 1.5 COMPONENTS Development) # javac, jar
 
@@ -159,3 +160,4 @@ if(Java_Development_FOUND AND Java_JAVAC
 else()
   message(WARNING "No java compiler found. Wont be able to build java viewer")
 endif()
+endif()
