diff -urN --ignore-all-space a/qtbase/src/sql/drivers/ibase/qsql_ibase.pri b/qtbase/src/sql/drivers/ibase/qsql_ibase.pri
--- a/qtbase/src/sql/drivers/ibase/qsql_ibase.pri	2014-05-02 22:55:53.748095200 +0100
+++ b/qtbase/src/sql/drivers/ibase/qsql_ibase.pri	2014-05-02 22:56:51.598404100 +0100
@@ -5,6 +5,6 @@
     !contains(LIBS, .*gds.*):!contains(LIBS, .*libfb.*):LIBS += -lgds
 } else {
     !contains(LIBS, .*gds.*):!contains(LIBS, .*fbclient.*) {
-        LIBS += -lgds32_ms
+        LIBS += -lfbclient
     }
 }
