--- a/qtbase/configure.bat
+++ b/qtbase/configure.bat
@@ -157,12 +157,12 @@ rem Determine host spec
 if "%PLATFORM%" == "" (
     if not "%icl.exe%" == "" (
         set PLATFORM=win32-icc
-    ) else if not "%clang-cl.exe%" == "" (
-        set PLATFORM=win32-clang-msvc
     ) else if not "%cl.exe%" == "" (
         set PLATFORM=win32-msvc
     ) else if not "%g++.exe%" == "" (
         set PLATFORM=win32-g++
+    ) else if not "%clang-cl.exe%" == "" (
+        set PLATFORM=win32-clang-msvc
     ) else (
         echo Cannot detect host toolchain. Please use -platform. Aborting. >&2
         exit /b 1
-- 
2.13.0.windows.1
