--- x86_64/qtbase/configure.bat.orig	2017-01-26 10:49:27.923099300 +0300
+++ x86_64/qtbase/configure.bat	2017-01-26 10:50:56.168562200 +0300
@@ -150,7 +150,7 @@
 :doneargs
 
 rem Find various executables
-for %%C in (clang-cl.exe cl.exe icl.exe g++.exe perl.exe jom.exe) do set %%C=%%~$PATH:C
+for %%C in (g++.exe clang-cl.exe cl.exe icl.exe perl.exe jom.exe) do set %%C=%%~$PATH:C
 
 rem Determine host spec
 
