--- a/configure	2014-05-03 01:45:34.788281700 +0100
+++ b/configure	2014-05-03 02:50:46.800036000 +0100
@@ -3874,6 +3874,12 @@
 
 if [ -z "$PKG_CONFIG" ]; then
     QT_CONFIG="$QT_CONFIG no-pkg-config"
+else
+    if [ "$CFG_SHARED" = "no" ]; then
+        # --static only matters for --libs*
+        # and should be harmless otherwise.
+        PKG_CONFIG="$PKG_CONFIG"" --static"
+    fi
 fi
 
 #-------------------------------------------------------------------------------
@@ -4555,6 +4561,12 @@
 if [ "$CFG_DBUS" != "no" ]; then
     if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --atleast-version="$MIN_DBUS_1_VERSION" dbus-1 2>/dev/null; then
         QT_CFLAGS_DBUS=`$PKG_CONFIG --cflags dbus-1 2>/dev/null`
+        if [ "$CFG_SHARED" = "no" -a "$XPLATFORM_MINGW" = "yes" ] ; then
+            # There's no way of returning different
+            # cflags from pkg-config when --static
+            # is specified unfortunately ..
+            QT_CFLAGS_DBUS="$QT_CFLAGS_DBUS"" -DDBUS_STATIC_BUILD"
+        fi
         QT_LIBS_DBUS=`$PKG_CONFIG --libs dbus-1 2>/dev/null`
     fi
     if compileTest unix/dbus "D-Bus" $QT_CFLAGS_DBUS $QT_LIBS_DBUS; then
--- a/src/dbus/dbus.pro	2014-05-03 01:45:34.788281700 +0100
+++ b/src/dbus/dbus.pro	2014-05-03 02:50:46.800036000 +0100
@@ -4,8 +4,7 @@
 MODULE_CONFIG = dbusadaptors dbusinterfaces
 
 !contains(QT_LIBS_DBUS, .*dbus-1.*) {
-    win32:CONFIG(debug, debug|release):QT_LIBS_DBUS += -ldbus-1d
-    else:QT_LIBS_DBUS += -ldbus-1
+    QT_LIBS_DBUS += -ldbus-1
 }
 
 DEFINES += DBUS_API_SUBJECT_TO_CHANGE