--- a/configure	2014-05-03 04:00:16.473527600 +0100
+++ b/configure	2014-05-03 04:14:11.844308100 +0100
@@ -4640,25 +4640,6 @@
     CFG_QGTKSTYLE=no
 fi
 
-# auto-detect libicu support
-if [ "$CFG_ICU" != "no" ]; then
-    if compileTest unix/icu "ICU"; then
-        [ "$CFG_ICU" = "auto" ] && CFG_ICU=yes
-    else
-        if [ "$CFG_ICU" = "auto" ]; then
-            CFG_ICU=no
-        elif [ "$CFG_CONFIGURE_EXIT_ON_ERROR" = "yes" ]; then
-            # CFG_ICU is "yes"
-
-            echo "The ICU library support cannot be enabled."
-            echo " Turn on verbose messaging (-v) to $0 to see the final report."
-            echo " If you believe this message is in error you may use the continue"
-            echo " switch (-continue) to $0 to continue."
-            exit 101
-        fi
-    fi
-fi
-
 # Auto-detect PulseAudio support
 if [ "$CFG_PULSEAUDIO" != "no" ]; then
     if [ -n "$PKG_CONFIG" ]; then
@@ -5551,6 +5532,29 @@
     QTCONFIG_CONFIG="$QTCONFIG_CONFIG static"
 fi

+if [ "$CFG_SHARED" = "no" ]; then
+    QMAKE_CONFIG="$QMAKE_CONFIG static"
+fi
+
+# auto-detect libicu support (must be done post static/shared)
+if [ "$CFG_ICU" != "no" ]; then
+    if compileTest unix/icu "ICU"; then
+        [ "$CFG_ICU" = "auto" ] && CFG_ICU=yes
+    else
+        if [ "$CFG_ICU" = "auto" ]; then
+            CFG_ICU=no
+        elif [ "$CFG_CONFIGURE_EXIT_ON_ERROR" = "yes" ]; then
+            # CFG_ICU is "yes"
+
+            echo "The ICU library support cannot be enabled."
+            echo " Turn on verbose messaging (-v) to $0 to see the final report."
+            echo " If you believe this message is in error you may use the continue"
+            echo " switch (-continue) to $0 to continue."
+            exit 101
+        fi
+    fi
+fi
+
 #FIXME: qpa is implicit this should all be removed
 QMAKE_CONFIG="$QMAKE_CONFIG qpa"
 QT_CONFIG="$QT_CONFIG qpa"
