--- a/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri	2014-04-08 22:03:26.386318500 +0100
+++ b/qtbase/src/platformsupport/fontdatabases/fontdatabases.pri	2014-04-08 22:01:12.262286900 +0100
@@ -1,8 +1,8 @@
-!win32|contains(QT_CONFIG, freetype):!mac {
+win32-g++|contains(QT_CONFIG, freetype):!mac {
     include($$PWD/basic/basic.pri)
 }
 
-unix:!mac {
+unix|win32-g++:!mac {
     CONFIG += qpa/genericunixfontdatabase
     include($$PWD/genericunix/genericunix.pri)
     contains(QT_CONFIG,fontconfig) {
