--- a/qtlocation/src/3rdparty/icu_dependency.pri
+++ b/qtlocation/src/3rdparty/icu_dependency.pri
@@ -2,11 +2,17 @@
     CONFIG(static, static|shared) {
         CONFIG(debug, debug|release) {
             LIBS_PRIVATE += -lsicuind -lsicuucd -lsicudtd
+            DEFINES += "U_LIB_SUFFIX_C_NAME=d"
         } else {
             LIBS_PRIVATE += -lsicuin -lsicuuc -lsicudt
         }
     } else {
-        LIBS_PRIVATE += -licuin -licuuc -licudt
+        CONFIG(debug, debug|release) {
+            LIBS_PRIVATE += -licuind -licuucd -licudtd
+            DEFINES += "U_LIB_SUFFIX_C_NAME=d"
+        } else {
+            LIBS_PRIVATE += -licuin -licuuc -licudt
+        }
     }
 } else {
     LIBS_PRIVATE += -licui18n -licuuc -licudata
