--- qt-stable/qtbase/config.tests/unix/ibase/ibase.pro.orig	2013-06-07 09:13:54 +0300
+++ qt-stable/qtbase/config.tests/unix/ibase/ibase.pro	2013-06-07 09:14:13 +0300
@@ -1,4 +1,4 @@
 SOURCES = ibase.cpp
 CONFIG -= qt dylib
 mac:CONFIG -= app_bundle
-LIBS += -lgds
+LIBS += -lfbclient_ms
--- qt-stable/qtbase/config.tests/unix/mysql/mysql.pro.orig	2013-06-07 09:14:27 +0300
+++ qt-stable/qtbase/config.tests/unix/mysql/mysql.pro	2013-06-07 09:14:32 +0300
@@ -1,4 +1,4 @@
 SOURCES = mysql.cpp
 CONFIG -= qt dylib
 mac:CONFIG -= app_bundle
-LIBS += -lmysqlclient
+LIBS += -lmysql
