From 6ceebb4d1ed8a4a7444cc20cb444c9a664a31652 Mon Sep 17 00:00:00 2001
From: Ben Wolsieffer <benwolsieffer@gmail.com>
Date: Sat, 18 Jan 2020 20:26:19 -0500
Subject: [PATCH] Fix build with OpenSceneGraph >=3.5.6

---
 src/osgbCollision/ComputeTriMeshVisitor.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/osgbCollision/ComputeTriMeshVisitor.cpp b/src/osgbCollision/ComputeTriMeshVisitor.cpp
index b9a9d58..7f8992b 100644
--- a/src/osgbCollision/ComputeTriMeshVisitor.cpp
+++ b/src/osgbCollision/ComputeTriMeshVisitor.cpp
@@ -44,7 +44,12 @@ struct ComputeTriMeshFunc
         vertices->clear();
     }
 
+    // https://github.com/openscenegraph/OpenSceneGraph/commit/c9545970ac3b6a805b994abd3bd76ed447e3e3f6
+    #if( OSGWORKS_OSG_VERSION >= 30506 )
+    void inline operator()( const osg::Vec3 v1, const osg::Vec3 v2, const osg::Vec3 v3 )
+    #else
     void inline operator()( const osg::Vec3 v1, const osg::Vec3 v2, const osg::Vec3 v3, bool _temp )
+    #endif
     {
         vertices->push_back( v1 );
         vertices->push_back( v2 );
