--- a/cmake/FindLFortranZLIB.cmake
+++ b/cmake/FindLFortranZLIB.cmake
@@ -13,7 +13,7 @@
 
 if (USE_DYNAMIC_ZLIB)
     if(WIN32)
-        set(CMAKE_FIND_LIBRARY_SUFFIXES .dll .lib .a)
+        set(CMAKE_FIND_LIBRARY_SUFFIXES .dll.a .lib .a)
     elseif(APPLE)
         set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so .a .lib)
     else()
--- a/cmake/FindLFortranZSTD.cmake
+++ b/cmake/FindLFortranZSTD.cmake
@@ -11,7 +11,7 @@
 if (USE_DYNAMIC_ZSTD)
     # Search dynamic library first, then static one
     if(WIN32)
-        set(CMAKE_FIND_LIBRARY_SUFFIXES .dll .lib .a)
+        set(CMAKE_FIND_LIBRARY_SUFFIXES .dll.a .lib .a)
     elseif(APPLE)
         set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so .a .lib)
     else()
