--- src/bin/lfortran.cpp.orig	2026-01-13 23:28:51.000000000 -0600
+++ src/bin/lfortran.cpp	2026-02-28 16:20:43.759281900 -0600
@@ -2175,7 +2175,7 @@
     if (backend == Backend::wasm) {
         // for node version less than 16, we need to also provide flag --experimental-wasm-bigint
         run_cmd = "node --experimental-wasi-unstable-preview1 " + outfile + ".js";
-    } else if (t == "x86_64-pc-windows-msvc") {
+    } else if (t == "x86_64-pc-windows-msvc" || t == "x86_64-w64-windows-gnu") {
         run_cmd = outfile;
     } else if (LCompilers::startswith(t, "wasm")) {
         if (LCompilers::endswith(t, "wasi")) {
