diff -ur occt-V7_5_0.orig/src/Aspect/Aspect_OpenVRSession.cxx occt-V7_5_0/src/Aspect/Aspect_OpenVRSession.cxx
--- occt-V7_5_0.orig/src/Aspect/Aspect_OpenVRSession.cxx	2020-11-26 11:27:09.822443300 +0100
+++ occt-V7_5_0/src/Aspect/Aspect_OpenVRSession.cxx	2020-11-26 11:30:38.912514000 +0100
@@ -168,7 +168,7 @@
   vr::IVRSystem*          System; //!< OpenVR session object
 
   //! Empty constructor.
-  Aspect_OpenVRSession::VRContext() : System (NULL)
+  VRContext() : System (NULL)
   {
     memset (TrackedPoses, 0, sizeof(TrackedPoses));
   }
