--- a/adm/cmake/occt_defs_flags.cmake
+++ b/adm/cmake/occt_defs_flags.cmake
@@ -200,9 +200,6 @@
   endif()
 endif()
 if(MINGW)
-  add_definitions(-D_WIN32_WINNT=0x0601)
-  # _WIN32_WINNT=0x0601 (use Windows 7 SDK)
-  #set (CMAKE_SYSTEM_VERSION "6.1")
   # workaround bugs in mingw with vtable export
   set (CMAKE_SHARED_LINKER_FLAGS "-Wl,--export-all-symbols")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wattributes")
