diff -Naur occt-V7_5_3.orig/src/Standard/Standard.cxx occt-V7_5_3/src/Standard/Standard.cxx
See https://gcc.gnu.org/onlinedocs/gcc-4.8.2/gcc/X86-Built-in-Functions.html
--- occt-V7_5_3.orig/src/Standard/Standard.cxx	2021-08-04 09:41:47.000000000 +0200
+++ occt-V7_5_3/src/Standard/Standard.cxx	2021-09-23 09:07:56.085053700 +0200
@@ -108,6 +108,7 @@
     // if not then use MMgrRaw instead of MMgrTBBalloc.
     // It is to avoid runtime crash when running on a CPU
     // that supports SSE but does not support SSE2 (some modifications of AMD Sempron).
+    #if defined(_MSC_VER)
     static const DWORD _SSE2_FEATURE_BIT(0x04000000);
     DWORD volatile dwFeature;
     _asm
@@ -129,6 +130,11 @@
     }
     if ((dwFeature & _SSE2_FEATURE_BIT) == 0)
       anAllocId = 0;
+    #elif defined(__GNUC__)
+    __builtin_cpu_init();
+    if (__builtin_cpu_supports("sse2"))
+      anAllocId = 0;
+    #endif
   }
 #endif
 
