From e0f014cb0cfdd06cf3f9ac8c5fe1bd9b5d0e7b5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Sat, 31 Dec 2022 16:43:49 +0100
Subject: [PATCH] Fix TBB detection

---
 adm/cmake/tbb.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/adm/cmake/tbb.cmake b/adm/cmake/tbb.cmake
index 4e5f724ac2..f69519e681 100644
--- a/adm/cmake/tbb.cmake
+++ b/adm/cmake/tbb.cmake
@@ -14,7 +14,7 @@ if (NOT DEFINED 3RDPARTY_TBB_DIR)
   set (3RDPARTY_TBB_DIR "" CACHE PATH "The directory containing tbb")
 endif()
 
-if (WIN32)
+if (MSVC)
   if (NOT DEFINED 3RDPARTY_DIR)
     message (FATAL_ERROR "3RDPARTY_DIR is not defined.")
   endif()
-- 
2.38.0.windows.1

