From fdd61fb135ed49ad5c35dfcfe346bd36fe965921 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Sat, 31 Dec 2022 17:57:26 +0100
Subject: [PATCH] Fix linking TKXDE

---
 src/TKXDE/EXTERNLIB | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/TKXDE/EXTERNLIB b/src/TKXDE/EXTERNLIB
index 58c24f0d95..a71aed682d 100644
--- a/src/TKXDE/EXTERNLIB
+++ b/src/TKXDE/EXTERNLIB
@@ -1 +1,2 @@
 TKernel
+TKMath
\ No newline at end of file
-- 
2.38.0.windows.1

