From 890d0b1a13f5f04a53494c0e1ab0787e7b55be19 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Thu, 24 Oct 2024 14:02:58 +0200
Subject: [PATCH] Fix linking TKDE

---
 src/TKDE/EXTERNLIB | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/TKDE/EXTERNLIB b/src/TKDE/EXTERNLIB
index 58c24f0d95..1a12b23444 100644
--- a/src/TKDE/EXTERNLIB
+++ b/src/TKDE/EXTERNLIB
@@ -1 +1,2 @@
 TKernel
+TKMath
-- 
2.44.0.windows.1

