From 287eea2762da5b5cf44016718e58f4a6eab6c89c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Sat, 31 Dec 2022 18:18:11 +0100
Subject: [PATCH] ViewerTest: include missing header

---
 src/ViewerTest/ViewerTest_CmdParser.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ViewerTest/ViewerTest_CmdParser.cxx b/src/ViewerTest/ViewerTest_CmdParser.cxx
index 04cb7d4b1c..a8619da144 100644
--- a/src/ViewerTest/ViewerTest_CmdParser.cxx
+++ b/src/ViewerTest/ViewerTest_CmdParser.cxx
@@ -21,6 +21,7 @@
 
 #include <algorithm>
 #include <iostream>
+#include <limits>
 
 namespace
 {
-- 
2.38.0.windows.1

