diff -urN extra-cmake-modules-1.5.0.orig/kde-modules/KDECMakeSettings.cmake extra-cmake-modules-1.5.0/kde-modules/KDECMakeSettings.cmake
--- extra-cmake-modules-1.5.0.orig/kde-modules/KDECMakeSettings.cmake	2015-01-04 01:35:09.800389800 +0000
+++ extra-cmake-modules-1.5.0/kde-modules/KDECMakeSettings.cmake	2015-01-04 01:37:00.535144400 +0000
@@ -151,6 +151,11 @@
    # Since CMake 2.8.6
    set(CMAKE_AUTOMOC ON)
 
+   # Enable autostaticplugins in MSYS2 qt5-static/cmake
+   # Since https://github.com/Alexpux/MINGW-packages/commit/e5517227778695a51bd56df07f1b05c43103ad3f
+   #   and https://github.com/Alexpux/MINGW-packages/commit/aa8f86ab267446002bcbb59e1455fbdf0d9f1cde
+   set(CMAKE_AUTOSTATICPLUGINS ON)
+
    # By default, create 'GUI' executables. This can be reverted on a per-target basis
    # using ECMMarkNonGuiExecutable
    # Since CMake 2.8.8
