--- a/kde-modules/KDEPackageAppTemplates.cmake
+++ b/kde-modules/KDEPackageAppTemplates.cmake
@@ -86,7 +86,7 @@
         # NOTE: we also pass `--sort=name` here to check if the tar exe supports that
         #       this feature was only added in gnu tar v1.28
         execute_process(
-            COMMAND ${_tar_executable} --sort=name --version
+            COMMAND ${_tar_executable} --sort=name --force-local --version
             TIMEOUT 3
             RESULT_VARIABLE _tar_exit
             OUTPUT_VARIABLE _tar_version
@@ -131,7 +131,7 @@
                    --numeric-owner --owner=0 --group=0
                    --mtime="@${TIMESTAMP}"
                    --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime
-                    -c -j -f ${_template} .
+                    -c -j --force-local -f ${_template} .
                 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${_templateName}
                 DEPENDS ${_subdirs_entries}
             )
