diff -urN extra-cmake-modules-5.25.0.orig/kde-modules/KDEInstallDirs.cmake extra-cmake-modules-5.25.0/kde-modules/KDEInstallDirs.cmake
--- extra-cmake-modules-5.25.0.orig/kde-modules/KDEInstallDirs.cmake	2016-08-18 16:35:08.439605000 +0200
+++ extra-cmake-modules-5.25.0/kde-modules/KDEInstallDirs.cmake	2016-08-18 16:32:55.840399200 +0200
@@ -406,7 +406,7 @@
     "object code libraries"
     LIB_INSTALL_DIR)
 
-if(WIN32)
+if(WIN32 AND NOT MSYS)
     _define_relative(LIBEXECDIR BINDIR ""
         "executables for internal use by programs and libraries"
         LIBEXEC_INSTALL_DIR)
@@ -429,7 +429,8 @@
 set(_default_KDE_INSTALL_USE_QT_SYS_PATHS OFF)
 if(NOT DEFINED KDE_INSTALL_USE_QT_SYS_PATHS)
     query_qmake(qt_install_prefix_dir QT_INSTALL_PREFIX)
-    if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}")
+    STRING(REGEX MATCH "/mingw64.*" qt_install_prefix_dir_msys ${qt_install_prefix_dir})
+    if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}" OR qt_install_prefix_dir_msys STREQUAL "${CMAKE_INSTALL_PREFIX}")
         message(STATUS "Installing in the same prefix as Qt, adopting their path scheme.")
         set(_default_KDE_INSTALL_USE_QT_SYS_PATHS ON)
     endif()
@@ -437,24 +438,48 @@
 
 option (KDE_INSTALL_USE_QT_SYS_PATHS "Install mkspecs files, Plugins and Imports to the Qt 5 install dir" "${_default_KDE_INSTALL_USE_QT_SYS_PATHS}")
 if(KDE_INSTALL_USE_QT_SYS_PATHS)
-    # Qt-specific vars
-    query_qmake(qt_plugins_dir QT_INSTALL_PLUGINS)
-
-    _define_absolute(QTPLUGINDIR ${qt_plugins_dir}
-        "Qt plugins"
-         QT_PLUGIN_INSTALL_DIR)
-
-    query_qmake(qt_imports_dir QT_INSTALL_IMPORTS)
-
-    _define_absolute(QTQUICKIMPORTSDIR ${qt_imports_dir}
-        "QtQuick1 imports"
-        IMPORTS_INSTALL_DIR)
-
-    query_qmake(qt_qml_dir QT_INSTALL_QML)
-
-    _define_absolute(QMLDIR ${qt_qml_dir}
-        "QtQuick2 imports"
-        QML_INSTALL_DIR)
+    if(NOT MSYS)
+        # Qt-specific vars
+        query_qmake(qt_plugins_dir QT_INSTALL_PLUGINS)
+
+        _define_absolute(QTPLUGINDIR ${qt_plugins_dir}
+            "Qt plugins"
+             QT_PLUGIN_INSTALL_DIR)
+
+        query_qmake(qt_imports_dir QT_INSTALL_IMPORTS)
+
+        _define_absolute(QTQUICKIMPORTSDIR ${qt_imports_dir}
+            "QtQuick1 imports"
+            IMPORTS_INSTALL_DIR)
+
+        query_qmake(qt_qml_dir QT_INSTALL_QML)
+
+        _define_absolute(QMLDIR ${qt_qml_dir}
+            "QtQuick2 imports"
+            QML_INSTALL_DIR)
+    else()
+        # Qt-specific vars
+        query_qmake(qt_plugins_dir QT_INSTALL_PLUGINS)
+        STRING(REGEX MATCH "/mingw64.*" qt_plugins_dir_msys ${qt_plugins_dir})
+
+        _define_absolute(QTPLUGINDIR ${qt_plugins_dir_msys}
+            "Qt plugins"
+             QT_PLUGIN_INSTALL_DIR)
+
+        query_qmake(qt_imports_dir QT_INSTALL_IMPORTS)
+        STRING(REGEX MATCH "/mingw64.*" qt_imports_dir_msys ${qt_imports_dir})
+
+        _define_absolute(QTQUICKIMPORTSDIR ${qt_imports_dir_msys}
+            "QtQuick1 imports"
+            IMPORTS_INSTALL_DIR)
+
+        query_qmake(qt_qml_dir QT_INSTALL_QML)
+        STRING(REGEX MATCH "/mingw64.*" qt_qml_dir_msys ${qt_qml_dir})
+
+        _define_absolute(QMLDIR ${qt_qml_dir_msys}
+            "QtQuick2 imports"
+            QML_INSTALL_DIR)
+    endif()
 else()
     _define_relative(QTPLUGINDIR LIBDIR "plugins"
         "Qt plugins"
@@ -489,7 +514,7 @@
 
 
 
-if (WIN32)
+if (WIN32 AND NOT MSYS)
     _define_relative(DATAROOTDIR BINDIR "data"
         "read-only architecture-independent data root"
         SHARE_INSTALL_PREFIX)
@@ -582,6 +607,8 @@
 set(_default_sysconf_dir "etc")
 if (CMAKE_INSTALL_PREFIX STREQUAL "/usr")
     set(_default_sysconf_dir "/etc")
+elseif (CMAKE_INSTALL_PREFIX STREQUAL "/mingw32" OR CMAKE_INSTALL_PREFIX STREQUAL "/mingw64")
+    set(_default_sysconf_dir "${CMAKE_INSTALL_PREFIX}/etc")
 endif()
 
 
diff -urN extra-cmake-modules-5.25.0.orig/modules/ECMGeneratePriFile.cmake extra-cmake-modules-5.25.0/modules/ECMGeneratePriFile.cmake
--- extra-cmake-modules-5.25.0.orig/modules/ECMGeneratePriFile.cmake	2016-07-27 01:44:31.000000000 +0200
+++ extra-cmake-modules-5.25.0/modules/ECMGeneratePriFile.cmake	2016-08-18 17:50:12.783964400 +0200
@@ -91,7 +91,8 @@
 if(NOT DEFINED KDE_INSTALL_USE_QT_SYS_PATHS)
     include(ECMQueryQmake)
     query_qmake(qt_install_prefix_dir QT_INSTALL_PREFIX)
-    if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}")
+    STRING(REGEX MATCH "/mingw64.*" qt_install_prefix_dir_msys ${qt_install_prefix_dir})
+    if(qt_install_prefix_dir STREQUAL "${CMAKE_INSTALL_PREFIX}" OR qt_install_prefix_dir_msys STREQUAL "${CMAKE_INSTALL_PREFIX}")
         set(_askqmake ON)
     endif()
 endif()
@@ -99,7 +100,12 @@
 if(KDE_INSTALL_USE_QT_SYS_PATHS OR _askqmake)
   include(ECMQueryQmake)
   query_qmake(qt_host_data_dir QT_HOST_DATA)
-  set(ECM_MKSPECS_INSTALL_DIR ${qt_host_data_dir}/mkspecs/modules CACHE PATH "The directory where mkspecs will be installed to.")
+  if(NOT MSYS)
+      set(ECM_MKSPECS_INSTALL_DIR ${qt_host_data_dir}/mkspecs/modules CACHE PATH "The directory where mkspecs will be installed to.")
+  else()
+      STRING(REGEX MATCH "/mingw64.*" qt_host_data_dir_msys ${qt_host_data_dir})
+      set(ECM_MKSPECS_INSTALL_DIR ${qt_host_data_dir_msys}/mkspecs/modules CACHE PATH "The directory where mkspecs will be installed to.")
+  endif()
 else()
   set(ECM_MKSPECS_INSTALL_DIR mkspecs/modules CACHE PATH "The directory where mkspecs will be installed to.")
 endif()
