--- newlib-2.1.0/newlib/libc/include/sys/config.h.orig	2014-11-03 14:00:09.151452400 -0500
+++ newlib-2.1.0/newlib/libc/include/sys/config.h	2014-11-03 13:59:47.444421100 -0500
@@ -8,6 +8,10 @@
 #define MALLOC_ALIGNMENT 16
 #endif
 
+#if defined(__arm__) || defined(__thumb__)
+#define __DYNAMIC_REENT__
+#endif
+
 /* exceptions first */
 #if defined(__H8500__) || defined(__W65__)
 #define __SMALL_BITFIELDS
