--- libjpeg-turbo-1.2.90/jmorecfg.h.orig	2013-04-04 15:08:10 +0400
+++ libjpeg-turbo-1.2.90/jmorecfg.h	2013-04-04 16:23:13 +0400
@@ -12,6 +12,12 @@
  * optimizations.  Most users will not need to touch this file.
  */
 
+/* prevents conflicts */
+#if defined(__MINGW32__)
+#include <shlwapi.h> /* typedefs INT16 and INT32 */
+
+#define HAVE_BOOLEAN
+#endif
 
 /*
  * Define BITS_IN_JSAMPLE as either
@@ -154,14 +154,18 @@
 /* INT16 must hold at least the values -32768..32767. */
 
 #ifndef XMD_H			/* X11/xmd.h correctly defines INT16 */
+#ifndef _BASETSD_H_		/* basestd.h from mingw-w64 defines INT16 */
 typedef short INT16;
 #endif
+#endif
 
 /* INT32 must hold at least signed 32-bit values. */
 
 #ifndef XMD_H			/* X11/xmd.h correctly defines INT32 */
+#ifndef _BASETSD_H_		/* basestd.h from mingw-w64 defines INT32 */
 typedef long INT32;
 #endif
+#endif
 
 /* Datatype used for image dimensions.  The JPEG standard only supports
  * images up to 64K*64K due to 16-bit fields in SOF markers.  Therefore
