--- poco-1.10.1-all/Util/CMakeLists.txt.orig	2020-02-21 09:10:46.711376800 +0300
+++ poco-1.10.1-all/Util/CMakeLists.txt	2020-02-21 09:55:09.318313700 +0300
@@ -27,6 +27,12 @@
 	DEFINE_SYMBOL Util_EXPORTS
 )
 
+if(MINGW)
+	add_definitions(-DUNICODE -D_UNICODE)
+	target_compile_options(Util PUBLIC -municode)
+	target_link_libraries(Util PUBLIC -municode)
+endif()
+
 target_link_libraries(Util PUBLIC Poco::Foundation)
 if(ENABLE_XML)
 	target_link_libraries(Util PUBLIC Poco::XML)
