--- poco-poco-1.10.1-release/CMakeLists.txt.orig	2014-11-17 12:31:47.705800000 +0300
+++ poco-poco-1.10.1-release/CMakeLists.txt	2014-11-17 12:32:39.209200000 +0300
@@ -440,7 +440,7 @@
 # Set config script install location in a location that find_package() will
 # look for, which is different on MS Windows than for UNIX
 # Note: also set in POCO_GENERATE_PACKAGE macro in cmake/PocoMacros.cmake
-if(WIN32)
+if(MSVC)
 	set(PocoConfigPackageLocation "cmake")
 else()
 	set(PocoConfigPackageLocation "lib${LIB_SUFFIX}/cmake/${PROJECT_NAME}")
--- poco-poco-1.10.1-release/cmake/PocoMacros.cmake.orig	2014-11-17 12:31:47.705800000 +0300
+++ poco-poco-1.10.1-release/cmake/PocoMacros.cmake	2014-11-17 12:32:39.209200000 +0300
@@ -40,8 +40,13 @@
 			endforeach()
 		endif(X64)
 	endif()
+	if(MINGW)
+		find_program(CMAKE_MC_COMPILER windmc.exe HINTS "${sdk_bindir}" "${kit_bindir}"
+		DOC "path to message compiler")
+	else()
 	find_program(CMAKE_MC_COMPILER mc.exe HINTS "${sdk_bindir}" "${kit_bindir}" "${kit81_bindir}" ${kit10_bindir}
 		DOC "path to message compiler")
+	endif()
 	if(NOT CMAKE_MC_COMPILER)
 		message(FATAL_ERROR "message compiler not found: required to build")
 	endif(NOT CMAKE_MC_COMPILER)
@@ -235,7 +235,7 @@
 # Set config script install location in a location that find_package() will
 # look for, which is different on MS Windows than for UNIX
 # Note: also set in root CMakeLists.txt
-if(WIN32)
+if(MSVC)
 	set(PocoConfigPackageLocation "cmake")
 else()
 	set(PocoConfigPackageLocation "lib${LIB_SUFFIX}/cmake/${PROJECT_NAME}")
--- poco-poco-1.10.1-release/Foundation/CMakeLists.txt.orig	2014-11-17 12:31:47.705800000 +0300
+++ poco-poco-1.10.1-release/Foundation/CMakeLists.txt	2014-11-17 12:32:39.209200000 +0300
@@ -187,7 +187,9 @@
 			POCO_WIN32_UTF8
 			_WIN32
 			MINGW32
-			WINVER=0x500
+			WINVER=0x501
+			UNICODE
+			_UNICODE
 			ODBCVER=0x0300
 			POCO_THREAD_STACK_SIZE
 	)
