From 239d8a7dc09515742779c2f8f24154da8fc9c80f Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 2 Jun 2016 15:58:32 +0100
Subject: [PATCH 2/2] Ignore trailing cr and spaces in tests

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9c15d4d..38b01d7 100644
--- a/Makefile
+++ b/Makefile
@@ -13,17 +13,17 @@ clean:
 test1: c99conv$(EXT)
 	$(CC) -E unit.c -o unit.prev.c
 	./c99conv unit.prev.c unit.post.c
-	diff -u unit.{prev,post}.c || :
+	diff -u --strip-trailing-cr --ignore-all-space unit.{prev,post}.c || :
 
 test2: c99conv$(EXT)
 	$(CC) -E unit2.c -o unit2.prev.c
 	./c99conv unit2.prev.c unit2.post.c
-	diff -u unit2.{prev,post}.c || :
+	diff -u --strip-trailing-cr --ignore-all-space unit2.{prev,post}.c || :
 
 test3: c99conv$(EXT)
 	$(CC) $(CFLAGS) -E -o convert.prev.c convert.c
 	./c99conv convert.prev.c convert.post.c
-	diff -u convert.{prev,post}.c
+	diff -u --strip-trailing-cr --ignore-all-space convert.{prev,post}.c
 
 c99conv$(EXT): $(OBJS)
 	$(CC) -o $@ $< $(LDFLAGS) $(LIBS)
-- 
2.8.2

