--- jbig2dec-0.16/autogen.sh.orig	2019-07-12 10:00:44.078119900 +0300
+++ jbig2dec-0.16/autogen.sh	2019-07-12 10:00:58.320944900 +0300
@@ -164,12 +164,3 @@
 
 echo "  autoconf"
 autoconf
-
-if test -z "$*"; then
-        echo "I am going to run ./configure with no arguments - if you wish "
-        echo "to pass any to it, please specify them on the $0 command line."
-else
-	echo "running ./configure $@"
-fi
-
-$srcdir/configure "$@" && echo
