--- jbig2dec-0.20/getopt.h.orig	2023-09-27 09:07:07.056641700 +0200
+++ jbig2dec-0.20/getopt.h	2023-09-27 09:07:24.616292300 +0200
@@ -141,17 +141,17 @@
 /* Many other libraries have conflicting prototypes for getopt, with
    differences in the consts, in stdlib.h.  To avoid compilation
    errors, only prototype getopt for the GNU C library.  */
-extern int getopt(int __argc, char *const *__argv, const char *__shortopts);
+extern int getopt(int argc, char *const *argv, const char *__shortopts);
 # else                          /* not __GNU_LIBRARY__ */
 extern int getopt();
 # endif                         /* __GNU_LIBRARY__ */
 
 # ifndef __need_getopt
-extern int getopt_long(int __argc, char *const *__argv, const char *__shortopts, const struct option *__longopts, int *__longind);
-extern int getopt_long_only(int __argc, char *const *__argv, const char *__shortopts, const struct option *__longopts, int *__longind);
+extern int getopt_long(int argc, char *const *argv, const char *__shortopts, const struct option *__longopts, int *__longind);
+extern int getopt_long_only(int argc, char *const *argv, const char *__shortopts, const struct option *__longopts, int *__longind);
 
 /* Internal only.  Users should not call this directly.  */
-extern int _getopt_internal(int __argc, char *const *__argv, const char *__shortopts, const struct option *__longopts, int *__longind, int __long_only);
+extern int _getopt_internal(int argc, char *const *argv, const char *__shortopts, const struct option *__longopts, int *__longind, int __long_only);
 # endif
 #else                           /* not __STDC__ */
 extern int getopt();
