--- keras-2.2.4.orig/keras/backend/__init__.py	2019-01-21 14:13:39.446288200 +0800
+++ keras-2.2.4/keras/backend/__init__.py	2019-01-21 14:14:43.204935000 +0800
@@ -23,8 +23,8 @@
         _keras_base_dir = '/tmp'
     _keras_dir = os.path.join(_keras_base_dir, '.keras')
 
-# Default backend: TensorFlow.
-_BACKEND = 'tensorflow'
+# Default backend: Theano.
+_BACKEND = 'theano'
 
 # Attempt to read Keras config file.
 _config_path = os.path.expanduser(os.path.join(_keras_dir, 'keras.json'))
