diff --git a/Makefile b/Makefile
index 3dfc08a0..0949fdb9 100644
--- a/Makefile
+++ b/Makefile
@@ -8,19 +8,19 @@ export CGO_LDFLAGS
 ## The following tasks delegate to `script/build.go` so they can be run cross-platform.
 
 .PHONY: bin/gh
-bin/gh: script/build
-	@script/build bin/gh
+bin/gh: script/build.exe
+	@script/build.exe bin/gh.exe
 
-script/build: script/build.go
-	go build -o script/build script/build.go
+script/build.exe: script/build.go
+	go build -o script/build.exe script/build.go
 
 .PHONY: clean
-clean: script/build
-	@script/build clean
+clean: script/build.exe
+	@script/build.exe clean
 
 .PHONY: manpages
-manpages: script/build
-	@script/build manpages
+manpages: script/build.exe
+	@script/build.exe manpages
 
 # just a convenience task around `go test`
 .PHONY: test
