--- librsvg-2.50.3/Makefile.am.orig	2021-07-10 17:47:36.531222200 +0200
+++ librsvg-2.50.3/Makefile.am	2021-07-10 17:47:38.774497600 +0200
@@ -340,7 +340,7 @@
 
 VAPIGEN_VAPIS = librsvg-$(RSVG_API_VERSION).vapi
 
-librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo
+librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo gdk-pixbuf-2.0
 librsvg_@RSVG_API_VERSION_U@_vapi_METADATADIRS = $(srcdir)
 librsvg_@RSVG_API_VERSION_U@_vapi_FILES = Rsvg-$(RSVG_API_VERSION).gir $(srcdir)/Rsvg-$(RSVG_API_VERSION)-custom.vala
 
