From 38c7c653cfddbc6c4fb9f9fb1744ab3e1ecbc9f4 Mon Sep 17 00:00:00 2001
From: phweyland <philippe.weyland@libertysurf.fr>
Date: Wed, 12 Jun 2019 11:38:44 -0300
Subject: [PATCH] patch makefile for msys2

---
 Makefile | 26 +++++++++++++++++++++-----
 1 file changed, 21 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 547cd70..e118b54 100644
--- a/Makefile
+++ b/Makefile
@@ -113,10 + 113,15 @@
 OS = Unix
 endif
 
-ifneq (, $(findstring MINGW, $(OS)))
+ifneq (, $(findstring MINGW64, $(OS)))
 OS = Windows
 USR = /mingw64
 endif
+
+ifneq (, $(findstring MINGW32, $(OS)))
+OS = Windows
+USR = /mingw32
+endif
 
 ifeq ($(OS),Darwin)
 ifeq (,$(wildcard /opt/local))
@@ -186,9 +186,6 @@ ifndef NO_STDLIB
 MANDATORY_CFLAGS += -stdlib=libc++
 endif
 endif
-ifeq ($(OS),Windows)
-MANDATORY_LIBS += -Wl,--stack,16777216
-endif
 
 # Enable optimizations.
 OPT_CFLAGS = -Ofast
@@ -288,8 +288,8 @@ CURL_LIBS = $(shell pkg-config --libs libcurl || echo -lcurl)
 
 # Enable native support of webcams and video streaming, using the OpenCV library.
 # (https://opencv.org/)
-OPENCV_CFLAGS = -Dcimg_use_opencv $(shell pkg-config opencv --cflags || echo -I/usr/include/opencv) -I/usr/include/opencv
-OPENCV_LIBS = $(shell pkg-config opencv --libs || echo -lopencv_core -lopencv_highgui)
+OPENCV_CFLAGS = -Dcimg_use_opencv $(shell pkg-config opencv4 --cflags)
+OPENCV_LIBS = $(shell pkg-config opencv4 --libs || echo -lopencv_core -lopencv_highgui)
 
 # Enable support of most classical image file formats, using the GraphicsMagick++ library.
 # (http://www.graphicsmagick.org/Magick++/)
@@ -403,12 +403,21 @@ lib:
 
 _lib: libgmic.o use_libgmic.cpp
 	ar rcs libgmic.a libgmic.o
+ifeq ($(OS),Windows)
+	$(CXX) -shared -o libgmic$(SO) \
+	-Wl,--out-implib=libgmic$(SO).a \
+	-Wl,--export-all-symbols \
+	-Wl,--enable-auto-import \
+	-Wl,--whole-archive libgmic.o \
+	-Wl,--no-whole-archive $(LIBS)
+else
 ifeq ($(OS),Darwin)
 	$(CXX) -shared -o libgmic.so libgmic.o $(LIBS)
 else
 	$(CXX) -shared -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so libgmic.o $(LIBS)
 	$(CXX) -o use_libgmic use_libgmic.cpp -L. -lgmic $(LIBS)
 endif
+endif
 
 libgmic.o: gmic.cpp gmic.h gmic_stdlib.h CImg.h
 	$(CXX) -o libgmic.o -c gmic.cpp $(PIC) $(CFLAGS)
@@ -708,6 +717,12 @@ install:
 	@if [ -f ../gmic-qt/gmic_gimp_qt ]; then cp -f ../gmic-qt/gmic_gimp_qt $(DESTDIR)$(PLUGINDIR)/; fi
 	@if [ -f ../gmic-qt/gmic_krita_qt ]; then cp -f ../gmic-qt/gmic_krita_qt $(DESTDIR)$(USR)/$(BIN)/; fi
 
+ifeq ($(OS),Windows)
+	mkdir -p $(DESTDIR)$(USR)/share
+	mkdir -p $(DESTDIR)$(USR)/$(LIB)
+	cp -f libgmic$(SO) $(DESTDIR)$(USR)/$(LIB)
+	cp -f libgmic$(SO).a $(DESTDIR)$(USR)/$(LIB)
+else
 ifneq ($(OS),Darwin)
 	mkdir -p $(DESTDIR)$(USR)/share
 	mkdir -p $(DESTDIR)$(USR)/$(LIB)
@@ -720,6 +735,7 @@ ifneq ($(OS),Darwin)
           ln -fs libcgmic.so.$(VERSION) $(DESTDIR)$(USR)/$(LIB)/libcgmic.so.$(VERSION1) ; \
           ln -fs libcgmic.so.$(VERSION1) $(DESTDIR)$(USR)/$(LIB)/libcgmic.so; \
         fi
+endif
 endif
 	mkdir -p $(DESTDIR)$(USR)/share/man/
 	mkdir -p $(DESTDIR)$(USR)/share/man/man1/
-- 
2.21.0

