--- gegl-0.4.66/gegl/gegl-config.c.orig	2025-12-11 21:43:07.000000000 +0100
+++ gegl-0.4.66/gegl/gegl-config.c	2025-12-12 08:42:25.497588900 +0100
@@ -331,7 +331,7 @@
     if (default_tile_cache_size < mem_min)
       default_tile_cache_size = mem_min;
 
-#ifndef _WIN64
+#if defined(__MINGW32__) && !defined(_UCRT)
     env = getenv ("GEGL_BUILD");
 #else
     _dupenv_s (&env, NULL, "GEGL_BUILD");
@@ -385,7 +385,7 @@
   _gegl_threads = g_get_num_processors ();
   _gegl_threads = MIN (_gegl_threads, GEGL_MAX_THREADS);
 
-#ifndef _WIN64
+#if defined(__MINGW32__) && !defined(_UCRT)
   env = getenv ("GEGL_BUILD");
 #else
   _dupenv_s (&env, NULL, "GEGL_BUILD");
