--- gegl-0.4.62/operations/external/ff-load.c.orig	2025-08-31 22:28:31.950778300 +0200
+++ gegl-0.4.62/operations/external/ff-load.c	2025-08-31 22:33:55.397609700 +0200
@@ -349,7 +349,7 @@
   if (frame < 2 || frame > prevframe + 64 || frame < prevframe )
   {
     int64_t seek_target = av_rescale_q (((frame) * AV_TIME_BASE * 1.0) / o->frame_rate
-, AV_TIME_BASE_Q, p->video_stream->time_base) / p->video_ctx->ticks_per_frame;
+, AV_TIME_BASE_Q, p->video_stream->time_base) / (p->video_ctx->codec_descriptor->props & AV_CODEC_PROP_FIELDS ? 2 : 1);
 
     if (av_seek_frame (p->video_fcontext, p->video_index, seek_target, (AVSEEK_FLAG_BACKWARD )) < 0)
       fprintf (stderr, "video seek error!\n");
@@ -403,7 +403,7 @@
                   break;
                 }
               got_picture = 1;
-              if ((pkt.dts == pkt.pts) || (p->lavc_frame->key_frame!=0))
+              if ((pkt.dts == pkt.pts) || (p->lavc_frame->flags & AV_FRAME_FLAG_KEY))
                 {
                   // cur_dts and first_dts are moved to libavformat/internal.h
                   /*
