--- gegl-0.4.68/meson.build.orig	2026-03-15 09:25:42.378899700 +0100
+++ gegl-0.4.68/meson.build	2026-03-15 09:25:47.235979200 +0100
@@ -661,7 +661,7 @@
 subdir('docs')
 
 
-if not os_osx and host_cpu_family != 'x86'
+if not os_osx and host_cpu_family != 'x86' and false
   # Verify .def files for Windows linking.
   # We check this on non-Windows platform on CI, and on Windows itself.
   custom_target('check-def-files',
