From 26acf187810ad49a00dfb5c5d2226c860fb9e1dd Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Fri, 14 Aug 2020 17:35:36 +0200
Subject: [PATCH] meson: Pass export_package for GIR generation

---
 gegl/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gegl/meson.build b/gegl/meson.build
index 514f62107..c92110349 100644
--- a/gegl/meson.build
+++ b/gegl/meson.build
@@ -122,6 +122,7 @@ if get_option('introspection')
     symbol_prefix: 'gegl',
     header: 'gegl.h',
     includes: [ 'GLib-2.0', 'GObject-2.0', 'Babl-0.1' ],
+    export_packages: 'gegl-' + api_version,
     install: true,
   )
 
