From ba9f9848a4a5e3f5a71c76cbc4009550878ec89c Mon Sep 17 00:00:00 2001
From: Niels De Graef <nielsdegraef@gmail.com>
Date: Sun, 7 Jun 2020 17:59:41 +0200
Subject: [PATCH] meson: Correctly add the VAPI metadata

---
 gegl/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gegl/meson.build b/gegl/meson.build
index 514f62107..f61dc9fe5 100644
--- a/gegl/meson.build
+++ b/gegl/meson.build
@@ -133,7 +133,7 @@ if get_option('introspection')
     )
 
     gnome.generate_vapi(api_name,
-      metadata_dirs: '.',
+      metadata_dirs: meson.current_source_dir(),
       sources: gegl_gir[0],
       install: true,
     )
-- 
2.26.2

