diff --git a/meson.build b/meson.build
index d83a7b1cb..4cc7d1cd7 100644
--- a/meson.build
+++ b/meson.build
@@ -73,23 +73,21 @@ if os_osx and cc.get_id() != 'clang'
 endif
 
 
-host_cpu = host_machine.cpu()
-if   host_cpu.startswith('i') and host_cpu.endswith('86')
+host_cpu_family = host_machine.cpu_family()
+if   host_cpu_family == 'x86'
   have_x86 = true
   config.set10('ARCH_X86',    true)
-elif host_cpu == 'x86_64'
+elif host_cpu_family == 'x86_64'
   have_x86 = true
   config.set10('ARCH_X86',    true)
   config.set10('ARCH_X86_64', true)
-elif host_cpu == 'ppc' or host_cpu == 'powerpc'
+elif host_cpu_family == 'ppc'
   have_ppc = true
   config.set10('ARCH_PPC',    true)
-elif host_cpu == 'ppc64' or host_cpu == 'powerpc64'
+elif host_cpu_family == 'ppc64'
   have_ppc = true
   config.set10('ARCH_PPC',    true)
   config.set10('ARCH_PPC64',  true)
-else
-  error('Unknown host architecture')
 endif
 
 ################################################################################
