--- a/script/bootstrap.py
+++ b/script/bootstrap.py
@@ -711,8 +711,12 @@
 def ffmpeg_configure_opts(install_dir, arch, validation):
     """configure options for ffmpeg build"""
     posix_install_dir = to_posix_path(install_dir)
+    cc = os.getenv('CC')
+    cxx = os.getenv('CXX')
     result = [
         f'--prefix={posix_install_dir}',
+        f'--cc={cc}',
+        f'--cxx={cxx}',
         '--enable-static',
         '--disable-shared',
         '--enable-pic',
