diff -rup lgi-0.9.1/lgi/lgi.h lgi-0.9.1-ricky/lgi/lgi.h
--- lgi-0.9.1/lgi/lgi.h	2016-05-28 03:56:36.000000000 +0800
+++ lgi-0.9.1-ricky/lgi/lgi.h	2016-10-26 21:13:00.262476600 +0800
@@ -8,8 +8,8 @@
 
 #define G_LOG_DOMAIN "Lgi"
 
-#include <lua.h>
-#include <lauxlib.h>
+#include <lua5.1/lua.h>
+#include <lua5.1/lauxlib.h>
 
 /* Lua 5.2 compatibility stuff. */
 #if LUA_VERSION_NUM >= 502
diff -rup lgi-0.9.1/lgi/Makefile lgi-0.9.1-ricky/lgi/Makefile
--- lgi-0.9.1/lgi/Makefile	2016-05-28 03:56:36.000000000 +0800
+++ lgi-0.9.1-ricky/lgi/Makefile	2016-10-26 21:14:47.106118700 +0800
@@ -19,7 +19,7 @@ VERSION_FILE = version.lua
 ifneq ($(filter cygwin% msys% mingw%, $(HOST_OS)),)
 CORE = corelgilua51.dll
 LIBFLAG = -shared
-LIBS += -llua
+LIBS += -llua5.1
 else
 ifeq ($(HOST_OS),darwin)
 CORE = corelgilua51.so

