From 15823c0a5a1c7718d095e3ecb983108272c8a9fb Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Tue, 25 Nov 2014 21:20:09 +0000
Subject: [PATCH 1/7] Use libdir for plugins on msys2

---
 src/win32/dirs.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/win32/dirs.c b/src/win32/dirs.c
index 8a8876a1a2..2343579375 100644
--- a/src/win32/dirs.c
+++ b/src/win32/dirs.c
@@ -200,7 +200,15 @@ char *config_GetLibDir (void)
         goto error;
     *file = L'\0';
 
-    return FromWide (wpath);
+    file = wcsrchr (wpath, L'\\');
+    if (file == NULL)
+        goto error;
+    *file = L'\0';
+
+    wchar_t wpathlib[MAX_PATH];
+    wsprintf(wpathlib,L"%s\\lib\\vlc", wpath);
+
+    return FromWide (wpathlib);
 error:
     abort ();
 #endif
-- 
2.15.0

