see https://code.videolan.org/videolan/vlc/-/issues/29712
diff --git a/share/lua/intf/modules/httprequests.lua b/share/lua/intf/modules/httprequests.lua
index f55b79a55f..a015261709 100644
--- a/share/lua/intf/modules/httprequests.lua
+++ b/share/lua/intf/modules/httprequests.lua
@@ -34,7 +34,7 @@ function round(what, precision)
         what = common.us_tonumber(what)
     end
     if type(what) == "number" then
-        return math.floor(what*math.pow(10,precision)+0.5) / math.pow(10,precision)
+        return math.floor(what * (10 ^ precision) + 0.5) / (10 ^ precision)
     end
     return nil
 end
diff --git a/share/lua/sd/fmc.lua b/share/lua/sd/fmc.lua
index 6271c479b7..0d30fd1bcd 100644
--- a/share/lua/sd/fmc.lua
+++ b/share/lua/sd/fmc.lua
@@ -32,7 +32,7 @@ function main()
         if tonumber( show_node.children_map["songcount"][1].children[1] ) > 0 then
             local songs_node = node:add_node( {title="Songs"} )
             for _, song_node in ipairs( show_node.children_map["songs"][1].children ) do
-                _, _, artist, title = string.find( song_node.children_map["name"][1].children[1], "(.+)%s*-%s*(.+)" )
+                local _, _, artist, title = string.find( song_node.children_map["name"][1].children[1], "(.+)%s*-%s*(.+)" )
                 local rank = song_node.children_map["rank"][1].children[1]
                 if rank ~= nil then
                     rank = "Rank: " .. rank
