--- vlc-3.0.21/modules/codec/avcodec/dxva2.c.orig	2024-08-11 12:46:25.457348500 +0200
+++ vlc-3.0.21/modules/codec/avcodec/dxva2.c	2024-08-11 12:46:16.220464600 +0200
@@ -146,6 +146,10 @@
 static int DxResetVideoDecoder(vlc_va_t *);
 static void SetupAVCodecContext(vlc_va_t *);
 
+#ifndef FF_DXVA2_WORKAROUND_INTEL_CLEARVIDEO
+#define FF_DXVA2_WORKAROUND_INTEL_CLEARVIDEO    2 ///< Work around for DXVA2 and old Intel GPUs with ClearVideo interface
+#endif
+
 void SetupAVCodecContext(vlc_va_t *va)
 {
     vlc_va_sys_t *sys = va->sys;
--- vlc-3.0.21/modules/codec/avcodec/d3d11va.c.orig	2024-08-11 12:46:09.062420500 +0200
+++ vlc-3.0.21/modules/codec/avcodec/d3d11va.c	2024-08-11 12:41:53.261423300 +0200
@@ -138,6 +138,10 @@
 static void DxDestroySurfaces(vlc_va_t *);
 static void SetupAVCodecContext(vlc_va_t *);
 
+#ifndef FF_DXVA2_WORKAROUND_INTEL_CLEARVIDEO
+#define FF_DXVA2_WORKAROUND_INTEL_CLEARVIDEO    2 ///< Work around for DXVA2 and old Intel GPUs with ClearVideo interface
+#endif
+
 void SetupAVCodecContext(vlc_va_t *va)
 {
     vlc_va_sys_t *sys = va->sys;
