diff -ruN ruby-4.0.6/win32/win32.c ruby-4.0.6/win32/win32.c
--- ruby-4.0.6/win32/win32.c	2026-07-14 08:22:10.000000000 +0800
+++ ruby-4.0.6/win32/win32.c	2026-07-26 20:22:25.780844600 +0800
@@ -2390,6 +2390,7 @@
     return 0;
 }
 
+#if RUBY_MSVCRT_VERSION >= 140
 typedef struct {
     union
     {
@@ -2409,11 +2410,17 @@
 #define FILE_COUNT(stream) ((vcruntime_file*)stream)->_cnt
 #define FILE_READPTR(stream) ((vcruntime_file*)stream)->_ptr
 #define FILE_FILENO(stream) ((vcruntime_file*)stream)->_file
+#else
+#define FILE_COUNT(stream) ((stream)->_cnt)
+#define FILE_READPTR(stream) ((stream)->_ptr)
+#define FILE_FILENO(stream) ((stream)->_file)
+#endif
 
 /* License: Ruby's */
 typedef char lowio_text_mode;
 typedef char lowio_pipe_lookahead[3];
 
+#if RUBY_MSVCRT_VERSION >= 140
 typedef struct {
     CRITICAL_SECTION           lock;
     intptr_t                   osfhnd;          // underlying OS file HANDLE
@@ -2427,14 +2434,32 @@
     uint8_t dbcsBufferUsed   : 1; // Is the dbcsBuffer in use?
     char    dbcsBuffer;           // Buffer for the lead byte of DBCS when converting from DBCS to Unicode
 } ioinfo;
+#else
+typedef struct {
+    intptr_t osfhnd;
+    char osfile;
+    char pipech;
+    int lockinitflag;
+    CRITICAL_SECTION lock;
+#if RUBY_MSVCRT_VERSION >= 80
+    char textmode;
+    char pipech2[2];
+#endif
+} ioinfo;
+#endif
 
 #if !defined _CRTIMP || defined __MINGW32__
 #undef _CRTIMP
 #define _CRTIMP __declspec(dllimport)
 #endif
 
+#if RUBY_MSVCRT_VERSION >= 140
 static ioinfo ** __pioinfo = NULL;
 #define IOINFO_L2E 6
+#else
+EXTERN_C _CRTIMP ioinfo * __pioinfo[];
+#define IOINFO_L2E 5
+#endif
 static inline ioinfo* _pioinfo(int);
 
 
@@ -2450,6 +2475,7 @@
 static void
 set_pioinfo_extra(void)
 {
+#if RUBY_MSVCRT_VERSION >= 140
 #define FUNCTION_RET 0xc3 /* ret */
 #ifdef _DEBUG
 # define UCRTBASE "ucrtbased.dll"
@@ -2604,6 +2630,7 @@
         /* not found, maybe something wrong... */
         pioinfo_extra = 0;
     }
+#endif
 }
 
 static inline ioinfo*
