--- ruby-2.5.0/configure.ac.orig	2015-12-23 14:29:36.000000000 +0900
+++ ruby-2.5.0/configure.ac	2016-02-13 19:58:24.567691100 +0900
@@ -1550,7 +1550,7 @@
 	    @%:@include <stddef.h>
             @%:@ifdef __GNUC__
             @%:@define PRINTF_ARGS(decl, string_index, first_to_check) \
-              decl __attribute__((format(printf, string_index, first_to_check)))
+              decl __attribute__((format(gnu_printf, string_index, first_to_check)))
             @%:@else
             @%:@define PRINTF_ARGS(decl, string_index, first_to_check) decl
             @%:@endif
