--- ruby-2.1.1/configure.in.orig	2014-03-09 23:39:54.888800000 +0400
+++ ruby-2.1.1/configure.in	2014-03-09 23:40:46.946000000 +0400
@@ -1275,7 +1275,7 @@
 	    @%:@include <stddef.h>
             @%:@ifdef __GNUC__
             @%:@define PRINTF_ARGS(decl, string_index, first_to_check) \
-              decl __attribute__((format(printf, string_index, first_to_check)))
+              decl __attribute__((format(gnu_printf, string_index, first_to_check)))
             @%:@else
             @%:@define PRINTF_ARGS(decl, string_index, first_to_check) decl
             @%:@endif
