--- ruby-2.1.4/configure.in.orig	2014-10-28 09:33:36.930600000 +0300
+++ ruby-2.1.4/configure.in	2014-10-28 09:34:26.585400000 +0300
@@ -1049,8 +1049,9 @@
 		ac_cv_func_malloc_usable_size=no
 		{ test "$target_cpu" = x64 && ac_cv_func___builtin_setjmp=no; }
 		AC_CHECK_TYPE([NET_LUID], [], [],
-			      [@%:@include <windows.h>
-		  	      @%:@include <iphlpapi.h>])
+			      [@%:@include <winsock2.h>
+			       @%:@include <windows.h>
+			       @%:@include <iphlpapi.h>])
 		if test x"$ac_cv_type_NET_LUID" = xyes; then
 		    AC_DEFINE(HAVE_TYPE_NET_LUID, 1)
 		fi
