--- ruby-2.3.0/configure.in.orig	2016-02-13 19:58:24.531685400 +0900
+++ ruby-2.3.0/configure.in	2016-02-13 19:58:24.567691100 +0900
@@ -1187,7 +1187,8 @@
 		{ test "$target_cpu" = x64 && ac_cv_func___builtin_setjmp=no; }
 		AC_CHECK_TYPE([NET_LUID], [], [],
 			      [@%:@include <winsock2.h>
-			      @%:@include <iphlpapi.h>])
+			       @%:@include <windows.h>
+			       @%:@include <iphlpapi.h>])
 		if test x"$ac_cv_type_NET_LUID" = xyes; then
 		    AC_DEFINE(HAVE_TYPE_NET_LUID, 1)
 		fi
