--- a/configure
+++ b/configure
@@ -10150,7 +10150,6 @@ fi
 		 -Werror=implicit-function-declaration \
 		 -Werror=implicit-int \
 		 -Werror=pointer-arith \
-		 -Werror=shorten-64-to-32 \
 		 -Werror=write-strings \
 		 -Werror=old-style-definition \
 		 -Wimplicit-fallthrough=0 \
@@ -31628,13 +31627,7 @@ else $as_nop
 	     ;; #(
   mingw*) :
 
-		RUBY_SO_NAME="${rb_cv_msvcrt}"'-$(RUBY_BASE_NAME)$(MAJOR)$(MINOR)0'
-		if test x"${target_cpu}" != xi386
-then :
-
-		    RUBY_SO_NAME="${target_cpu}-${RUBY_SO_NAME}"
-
-fi
+              RUBY_SO_NAME="${target_cpu}-${rb_cv_msvcrt}"'-$(RUBY_BASE_NAME)$(MAJOR)$(MINOR)0'
 	     ;; #(
   *) :
     RUBY_SO_NAME='$(RUBY_BASE_NAME)' ;;
@@ -34048,7 +34041,7 @@ else $as_nop
     if test "${target_os}-${rb_cv_msvcrt}" = "mingw32-ucrt"
 then :
 
-        arch="${target_cpu}-mingw-ucrt"
+        arch="${target_cpu}-${target_os}-ucrt"
 
 else $as_nop
 
