diff -bur Serial-Studio-3.2.1-orig/lib/OpenSSL/CMakeLists.txt Serial-studio-3.2.1/lib/OpenSSL/CMakeLists.txt
--- Serial-Studio-3.2.1-orig/lib/OpenSSL/CMakeLists.txt	2025-12-28 15:50:06.932188200 -0700
+++ Serial-studio-3.2.1/lib/OpenSSL/CMakeLists.txt	2025-12-28 15:50:36.192359700 -0700
@@ -59,7 +59,7 @@
       ${BASE_DIR}/dll/macOS/libcrypto.a
     )
 
-  elseif(UNIX)
+  eelseif(UNIX OR MINGW)
     find_package(PkgConfig REQUIRED)
     pkg_check_modules(OPENSSL REQUIRED openssl)
     target_link_libraries(${TARGET} PRIVATE ${OPENSSL_LIBRARIES})
