diff -bur serial-studio-3.2.1-orig/app/CMakeLists.txt Serial-Studio-3.2.1/app/CMakeLists.txt
--- serial-studio-3.2.1-orig/app/CMakeLists.txt	2025-12-28 16:10:12.791425700 -0700
+++ Serial-Studio-3.2.1/app/CMakeLists.txt	2025-12-28 16:10:24.728303000 -0700
@@ -500,24 +500,6 @@
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
 
-set(deploy_tool_options_arg "")
-if(APPLE)
-  set(deploy_tool_options_arg "-hardened-runtime -appstore-compliant")
-elseif(WIN32)
-  set(deploy_tool_options_arg "-force-openssl --release --no-translations --no-compiler-runtime")
-endif()
-
-qt_generate_deploy_qml_app_script(
-  TARGET ${PROJECT_EXECUTABLE}
-  OUTPUT_SCRIPT deploy_script
-  MACOS_BUNDLE_POST_BUILD
-  NO_UNSUPPORTED_PLATFORM_ERROR
-  DEPLOY_USER_QML_MODULES_ON_UNSUPPORTED_PLATFORM
-  DEPLOY_TOOL_OPTIONS ${deploy_tool_options_arg}
-)
-
-install(SCRIPT ${deploy_script})
-
 #-------------------------------------------------------------------------------
 # Packaging
 #-------------------------------------------------------------------------------
