diff -bur serial-studio-3.2.1-orig/CMakeLists.txt Serial-Studio-3.2.1/CMakeLists.txt
--- serial-studio-3.2.1-orig/CMakeLists.txt	2025-12-28 16:15:30.815334200 -0700
+++ Serial-Studio-3.2.1/CMakeLists.txt	2025-12-28 16:15:47.928331900 -0700
@@ -209,7 +209,7 @@
    if(WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND MINGW)
       add_compile_options(
          -O3                             # Aggressive optimization
-         -march=x86-64-v2                # Target modern CPUs (2012+) with SSE4.2
+         -march=armv8.2-a                # Target modern CPUs (2012+) with SSE4.2
          -ftree-vectorize                # Enable automatic vectorization
          -funroll-loops                  # Improve loop performance by unrolling
          -fomit-frame-pointer            # Optimize function call overhead
