--- a/gplugin/gplugin-manager.c	2026-09-21 02:36:30.407162000 +0200
+++ b/gplugin/gplugin-manager.c	2026-09-21 02:44:28.565498400 +0200
@@ -668,7 +668,11 @@
 
 	g_return_if_fail(GPLUGIN_IS_MANAGER(manager));
 
-	path = g_build_filename(PREFIX, LIBDIR, "gplugin", NULL);
+	gchar *dir;
+
+	dir = g_win32_get_package_installation_directory_of_module (NULL);
+	path = g_build_filename(dir, "lib", "gplugin", NULL);
+	g_free (dir);
 	gplugin_manager_prepend_path(manager, path);
 	g_free(path);
 
@@ -699,7 +703,11 @@
 	g_return_if_fail(GPLUGIN_IS_MANAGER(manager));
 	g_return_if_fail(appname != NULL);
 
-	path = g_build_filename(prefix, LIBDIR, appname, NULL);
+	gchar *dir;
+
+	dir = g_win32_get_package_installation_directory_of_module (NULL);
+	path = g_build_filename(dir, "lib", appname, NULL);
+	g_free (dir);
 	gplugin_manager_prepend_path(manager, path);
 	g_free(path);
 
