--- jemalloc-5.2.1/include/jemalloc/internal/bit_util.h.orig	2021-01-18 13:05:31.228770900 -0800
+++ jemalloc-5.2.1/include/jemalloc/internal/bit_util.h	2021-01-18 13:06:52.228991500 -0800
@@ -188,6 +188,8 @@
 	return ((8 << LG_SIZEOF_PTR) - 1) - __builtin_clz(x);
 #elif (LG_SIZEOF_PTR == LG_SIZEOF_LONG)
 	return ((8 << LG_SIZEOF_PTR) - 1) - __builtin_clzl(x);
+#elif (LG_SIZEOF_PTR == LG_SIZEOF_LONG_LONG)
+	return ((8 << LG_SIZEOF_PTR) - 1) - __builtin_clzll(x);
 #else
 #  error "Unsupported type size for lg_floor()"
 #endif
