diff --git a/Makefile b/Makefile
index 18e07a60..616fbf81 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ all:
 	scons $(JOBS)
 
 release:
-	scons $(JOBS) mode=release
+	${MINGW_PREFIX}/bin/scons mode=release werror=no
 
 profile:
 	scons $(JOBS) mode=profile
diff --git a/SConstruct b/SConstruct
index 82bcc449..0e013cba 100644
--- a/SConstruct
+++ b/SConstruct
@@ -29,9 +29,9 @@ vars.AddVariables(
     PathVariable('config_file', 'Config file to use', 'src/config.h'),
 )
 
-target_os = str(Platform())
+target_os = 'msys'
 
-env = Environment(variables = vars, ENV = os.environ)
+env = Environment(tools = ['mingw'], variables = vars, ENV = os.environ)
 conf = env.Configure()
 
 if env['mode'] == 'analyze':
@@ -99,12 +99,10 @@ if target_os == 'posix':
     env.Append(CXXFLAGS=['-Wno-attributes', '-Wno-unused-variable',
                          '-Wno-unused-function'])
     env.Append(CCFLAGS=['-Wno-error=address']) # To remove if possible.
-    env.Append(LIBS=['glfw3', 'opengl32', 'z', 'tre', 'gdi32', 'Comdlg32',
-                     'ole32'],
-               LINKFLAGS='--static')
-    sources += glob.glob('ext_src/glew/glew.c')
-    env.Append(CPPPATH=['ext_src/glew'])
-    env.Append(CPPDEFINES=['GLEW_STATIC', 'FREE_WINDOWS'])
+    env.Append(LIBS=['opengl32', 'z', 'tre', 'pthread', 'gdi32', 'Comdlg32',
+                     'ole32'])
+    env.ParseConfig('pkg-config --cflags --libs glfw3 glew')
+    env.Append(CPPDEFINES=['FREE_WINDOWS'])
 
 # OSX Compilation support.
 if target_os == 'darwin':
