diff -urN Helics-v3.1.1-source-orig/CMakeLists.txt Helics-v3.1.1-source/CMakeLists.txt
--- Helics-v3.1.1-source-orig/CMakeLists.txt	2021-12-16 11:47:56.000000000 +0800
+++ Helics-v3.1.1-source/CMakeLists.txt	2022-01-07 14:22:31.213639300 +0800
@@ -539,7 +539,7 @@
         target_link_libraries(zmq INTERFACE libzmq-static)
         target_compile_definitions(zmq INTERFACE -DZMQ_STATIC)
     else()
-        target_link_libraries(zmq INTERFACE libzmq)
+        target_link_libraries(zmq INTERFACE libzmq wsock32 ws2_32 iphlpapi)
     endif()
 endif()
 add_library(helics::zmq ALIAS zmq)
diff -urN Helics-v3.1.1-source-orig/src/helics/apps/CMakeLists.txt Helics-v3.1.1-source/src/helics/apps/CMakeLists.txt
--- Helics-v3.1.1-source-orig/src/helics/apps/CMakeLists.txt	2021-12-16 11:47:56.000000000 +0800
+++ Helics-v3.1.1-source/src/helics/apps/CMakeLists.txt	2022-01-07 14:22:31.280643100 +0800
@@ -73,7 +73,7 @@
     target_link_libraries(helics_apps PUBLIC HELICS::application_api)
     target_link_libraries(
         helics_apps PRIVATE HELICS::compile_flags_target spdlog::spdlog fmt::fmt helics::zmq
-                            gmlc::networking
+                            gmlc::networking wsock32 ws2_32 iphlpapi
     )
 
     if(BOOST_VERSION_LEVEL GREATER 69 AND NOT (HELICS_DISABLE_WEBSERVER OR HELICS_DISABLE_BOOST
@@ -160,7 +160,7 @@
         target_link_libraries(helicscpp-apps PUBLIC HELICS::helicscpp)
         target_link_libraries(
             helicscpp-apps PRIVATE HELICS::common spdlog::spdlog fmt::fmt gmlc::utilities
-                                   gmlc::networking
+                                   gmlc::networking wsock32 ws2_32 iphlpapi
         )
 
         # add and alias library to match the find_package
diff -urN Helics-v3.1.1-source-orig/src/helics/core/CMakeLists.txt Helics-v3.1.1-source/src/helics/core/CMakeLists.txt
--- Helics-v3.1.1-source-orig/src/helics/core/CMakeLists.txt	2021-12-16 11:47:56.000000000 +0800
+++ Helics-v3.1.1-source/src/helics/core/CMakeLists.txt	2022-01-07 14:22:31.253641600 +0800
@@ -100,7 +100,7 @@
 add_library(helics_core STATIC ${SRC_FILES} ${INCLUDE_FILES} ${PUBLIC_INCLUDE_FILES})
 
 target_link_libraries(
-    helics_core PUBLIC HELICS::common gmlc::networking PRIVATE spdlog::spdlog fmt::fmt
+    helics_core PUBLIC HELICS::common gmlc::networking wsock32 ws2_32 iphlpapi PRIVATE spdlog::spdlog fmt::fmt
                                                                compile_flags_target
 )
 
diff -urN Helics-v3.1.1-source-orig/src/helics/cxx_shared_library/CMakeLists.txt Helics-v3.1.1-source/src/helics/cxx_shared_library/CMakeLists.txt
--- Helics-v3.1.1-source-orig/src/helics/cxx_shared_library/CMakeLists.txt	2021-12-16 11:47:56.000000000 +0800
+++ Helics-v3.1.1-source/src/helics/cxx_shared_library/CMakeLists.txt	2022-01-07 14:22:31.302644400 +0800
@@ -41,7 +41,7 @@
                      ${helics_shared_private_headers}
 )
 
-target_link_libraries(helicscpp PRIVATE HELICS::network fmt::fmt)
+target_link_libraries(helicscpp PRIVATE HELICS::network wsock32 ws2_32 iphlpapi fmt::fmt)
 target_compile_definitions(helicscpp PRIVATE helicscpp_EXPORTS PUBLIC HELICS_SHARED_LIBRARY)
 
 target_compile_options(helicscpp PUBLIC $<$<CXX_COMPILER_ID:MSVC>:/wd4251 /wd4275>)
