diff -urN Helics-v3.1.1-source-orig/ThirdParty/asio/asio/include/asio/detail/config.hpp Helics-v3.1.1-source/ThirdParty/asio/asio/include/asio/detail/config.hpp
--- Helics-v3.1.1-source-orig/ThirdParty/asio/asio/include/asio/detail/config.hpp	2021-10-16 12:59:18.000000000 +0800
+++ Helics-v3.1.1-source/ThirdParty/asio/asio/include/asio/detail/config.hpp	2022-01-07 14:35:08.877975200 +0800
@@ -617,9 +617,11 @@
 #  if (__cplusplus >= 201703)
 #   if defined(__clang__)
 #    if defined(ASIO_HAS_CLANG_LIBCXX)
-#     if (_LIBCPP_STD_VER > 14) && defined(_LIBCPP_HAS_ALIGNED_ALLOC)
+#     if (_LIBCPP_STD_VER > 14) && defined(_LIBCPP_HAS_ALIGNED_ALLOC) \
+        && !defined(_LIBCPP_MSVCRT) && !defined(__MINGW32__)
 #      define ASIO_HAS_STD_ALIGNED_ALLOC 1
 #     endif // (_LIBCPP_STD_VER > 14) && defined(_LIBCPP_HAS_ALIGNED_ALLOC)
+            //   && !defined(_LIBCPP_MSVCRT) && !defined(__MINGW32__)
 #    elif defined(_GLIBCXX_HAVE_ALIGNED_ALLOC)
 #     define ASIO_HAS_STD_ALIGNED_ALLOC 1
 #    endif // defined(_GLIBCXX_HAVE_ALIGNED_ALLOC)
