diff -bur magnum-2020.06-o/src/Magnum/Math/Vector.h magnum-2020.06/src/Magnum/Math/Vector.h
--- magnum-2020.06-o/src/Magnum/Math/Vector.h	2025-04-17 13:26:28.969883900 -0600
+++ magnum-2020.06/src/Magnum/Math/Vector.h	2025-04-17 13:29:21.358623800 -0600
@@ -30,6 +30,7 @@
  */
 
 #include <utility>
+#include <vector>
 #include <Corrade/Utility/Assert.h>
 #ifndef CORRADE_NO_DEBUG
 #include <Corrade/Utility/Debug.h>
diff -bur magnum-2020.06-o/src/Magnum/MeshTools/CompressIndices.h magnum-2020.06/src/Magnum/MeshTools/CompressIndices.h
--- magnum-2020.06-o/src/Magnum/MeshTools/CompressIndices.h	2025-04-17 13:26:28.975882900 -0600
+++ magnum-2020.06/src/Magnum/MeshTools/CompressIndices.h	2025-04-17 13:28:39.729414800 -0600
@@ -30,6 +30,7 @@
  */
 
 #include <utility>
+#include <vector>
 #include <Corrade/Containers/Containers.h>
 
 #include "Magnum/Mesh.h"
diff -bur magnum-2020.06-o/src/Magnum/MeshTools/GenerateNormals.h magnum-2020.06/src/Magnum/MeshTools/GenerateNormals.h
--- magnum-2020.06-o/src/Magnum/MeshTools/GenerateNormals.h	2025-04-17 13:26:28.978882000 -0600
+++ magnum-2020.06/src/Magnum/MeshTools/GenerateNormals.h	2025-04-17 13:30:17.859605600 -0600
@@ -29,6 +29,7 @@
  * @brief Function @ref Magnum::MeshTools::generateFlatNormals(), @ref Magnum::MeshTools::generateFlatNormalsInto(), @ref Magnum::MeshTools::generateSmoothNormals(), @ref Magnum::MeshTools::generateSmoothNormalsInto()
  */
 
+#include <vector>
 #include "Magnum/Magnum.h"
 #include "Magnum/MeshTools/visibility.h"
 
