--- origsrc/libsrtp-1.5.3/crypto/test/env.c	2015-11-25 16:24:25.000000000 +0100
+++ src/libsrtp-1.5.3/crypto/test/env.c	2015-12-30 17:45:04.798739500 +0100
@@ -49,9 +49,6 @@
 int 
 main(void) {
   int err_count = 0;
-#ifndef OPENSSL
-  char *str;
-#endif
 
 #ifdef WORDS_BIGENDIAN
   printf("CPU set to big-endian\t\t\t(WORDS_BIGENDIAN == 1)\n");
@@ -82,19 +79,6 @@
   printf("using stdout for error reporting\t(ERR_REPORTING_STDOUT == 1)\n");
 #endif
 
-#ifndef OPENSSL
-#ifdef DEV_URANDOM
-  str = DEV_URANDOM;
-#else
-  str = "";
-#endif
-  printf("using %s as a random source\t(DEV_URANDOM == %s)\n",
-	 str, str);
-  if (strcmp("", str) == 0) {
-    err_count++;
-  }
-#endif
-  
   if (err_count)
     printf("warning: configuration is probably in error "
 	   "(found %d problems)\n", err_count);
