--- origsrc/libsrtp-1.5.3/Makefile.in	2015-11-25 16:24:25.000000000 +0100
+++ src/libsrtp-1.5.3/Makefile.in	2016-01-01 23:31:08.351520400 +0100
@@ -273,7 +273,9 @@
 	if [ -f libsrtp.$(SHAREDLIBSUFFIX) ]; then \
 		$(INSTALL) -d $(DESTDIR)$(SHAREDLIB_DIR); \
 		cp libsrtp.$(SHAREDLIBSUFFIX) $(DESTDIR)$(SHAREDLIB_DIR)/; \
-		ln -sfn libsrtp.$(SHAREDLIBSUFFIX) $(DESTDIR)$(SHAREDLIB_DIR)/libsrtp.$(SHAREDLIBSUFFIXNOVER); \
+		if [ "mingw" != "$(findstring mingw,@host@)" ]; then \
+			ln -sfn libsrtp.$(SHAREDLIBSUFFIX) $(DESTDIR)$(SHAREDLIB_DIR)/libsrtp.$(SHAREDLIBSUFFIXNOVER); \
+		fi \
 	fi
 	if [ "$(pkgconfig_DATA)" != "" ]; then \
 		$(INSTALL) -d $(DESTDIR)$(pkgconfigdir); \
