--- origsrc/libsrtp-2.2.0/Makefile.in	2015-11-25 16:24:25.000000000 +0100
+++ src/libsrtp-2.2.0/Makefile.in	2016-01-01 23:31:08.351520400 +0100
@@ -256,9 +256,6 @@
 		$(INSTALL) -d $(DESTDIR)$(SHAREDLIB_DIR); \
 		cp libsrtp2.$(SHAREDLIBSUFFIX) $(DESTDIR)$(SHAREDLIB_DIR)/; \
 		cp libsrtp2.$(SHAREDLIBSUFFIXNOVER) $(DESTDIR)$(SHAREDLIB_DIR)/; \
-		if [ -n "$(SHAREDLIBVERSION)" ]; then \
-			ln -sfn libsrtp2.$(SHAREDLIBSUFFIX) $(DESTDIR)$(SHAREDLIB_DIR)/libsrtp2.$(SHAREDLIBSUFFIXNOVER); \
-		fi; \
 	fi
 	$(INSTALL) -d $(DESTDIR)$(pkgconfigdir)
 	cp $(top_builddir)/$(pkgconfig_DATA) $(DESTDIR)$(pkgconfigdir)/
