--- origsrc/libsrtp-1.5.3/crypto/Makefile.in	2015-11-25 16:24:25.000000000 +0100
+++ src/libsrtp-1.5.3/crypto/Makefile.in	2016-07-31 20:29:24.564876100 +0200
@@ -82,8 +82,8 @@
 %.o: %.c
 	$(COMPILE) -c $< -o $@
 
-%$(EXE): %.c ../test/getopt_s.c
-	$(COMPILE) $(LDFLAGS) $< ../test/getopt_s.c -o $@ $(CRYPTOLIB) $(LIBS)
+%$(EXE): %.c $(srcdir)/../test/getopt_s.c
+	$(COMPILE) $(LDFLAGS) $< $(srcdir)/../test/getopt_s.c -o $@ $(CRYPTOLIB) $(LIBS)
 
 all: $(testapp)
 
