From 3afa3ea124c5360ad561d7acb7ba97ae7497841d Mon Sep 17 00:00:00 2001
From: Ozkan Sezer <sezeroz@gmail.com>
Date: Sun, 30 Mar 2025 02:11:24 +0300
Subject: [PATCH] use -static suffix not in MinGW but MSVC builds

---
 gme/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gme/CMakeLists.txt b/gme/CMakeLists.txt
index fc2db68..9479005 100644
--- a/gme/CMakeLists.txt
+++ b/gme/CMakeLists.txt
@@ -423,7 +423,7 @@ if(GME_BUILD_STATIC)
         set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
     endif()
 
-    if(WIN32 AND GME_BUILD_SHARED)
+    if(MSVC AND GME_BUILD_SHARED)
         set_target_properties(gme_static PROPERTIES OUTPUT_NAME gme-static)
     else()
         set_target_properties(gme_static PROPERTIES OUTPUT_NAME gme)
