--- yaml-cpp-0.5.1/CMakeLists.txt.orig	2013-12-27 23:42:24.378600000 +0400
+++ yaml-cpp-0.5.1/CMakeLists.txt	2013-12-27 23:52:45.109800000 +0400
@@ -208,7 +208,7 @@
 ###
 ### General install settings
 ###
-if(WIN32)
+if(WIN32 AND NOT MINGW)
 	set(_library_dir bin)	# .dll are in PATH, like executables
 else()
 	set(_library_dir lib)
@@ -269,7 +269,7 @@
 	FILES_MATCHING PATTERN "*.h"
 )
 
-if(UNIX)
+if(UNIX OR MINGW)
 	set(PC_FILE ${CMAKE_BINARY_DIR}/yaml-cpp.pc)
 	configure_file("yaml-cpp.pc.cmake" ${PC_FILE} @ONLY)
 	install(FILES ${PC_FILE} DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
