--- ccache-4.14/doc/scripts/generate-manpage.orig	2026-08-23 16:35:18.000000000 +0800
+++ ccache-4.14/doc/scripts/generate-manpage	2026-08-25 12:38:29.667212600 +0800
@@ -24,4 +24,4 @@
 
 # Convert monospace to bold since that's typically rendered better when viewing
 # the man page.
-perl -pe "'s!\\\\f\\(CR(.*?)\\\\fP!\\\\fB\\1\\\\fP!g'" "${tmpfile}" >"${output}"
+sed 's/\\f(CR/\\fB/g' "${tmpfile}" >"${output}"
