diff -ur orig/CMakeLists.txt liboqs-0.10.0/CMakeLists.txt
--- orig/CMakeLists.txt	2024-03-23 13:34:28.000000000 +0100
+++ liboqs-0.10.0/CMakeLists.txt	2024-04-25 08:06:03.093690600 +0200
@@ -114,7 +114,7 @@
 
 option(OQS_SPEED_USE_ARM_PMU "Use ARM Performance Monitor Unit during benchmarking" OFF)
 
-if(WIN32)
+if(WIN32 AND NOT MINGW)
     set(CMAKE_GENERATOR_CC cl)
 endif()
 
diff -ur orig/tests/CMakeLists.txt liboqs-0.10.0/tests/CMakeLists.txt
--- orig/tests/CMakeLists.txt	2024-03-23 13:34:28.000000000 +0100
+++ liboqs-0.10.0/tests/CMakeLists.txt	2024-04-25 08:07:50.657181900 +0200
@@ -64,7 +64,7 @@
 
 add_executable(kat_kem kat_kem.c test_helpers.c)
 target_link_libraries(kat_kem PRIVATE ${TEST_DEPS})
-if(CMAKE_CROSSCOMPILING AND CMAKE_SYSTEM_NAME STREQUAL "Windows" AND BUILD_SHARED_LIBS)
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND BUILD_SHARED_LIBS)
     # workaround for Windows .dll cross-compiling
     target_link_options(kat_kem PRIVATE -Wl,--allow-multiple-definition)
 endif()
@@ -86,7 +86,7 @@
 
 add_executable(kat_sig kat_sig.c test_helpers.c)
 target_link_libraries(kat_sig PRIVATE ${TEST_DEPS})
-if(CMAKE_CROSSCOMPILING AND CMAKE_SYSTEM_NAME STREQUAL "Windows" AND BUILD_SHARED_LIBS)
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND BUILD_SHARED_LIBS)
     # workaround for Windows .dll cross-compiling
     target_link_options(kat_sig PRIVATE -Wl,--allow-multiple-definition)
 endif()
