--- a/cmake/tribits/common_tpls/find_modules/FindHDF5.cmake
+++ b/cmake/tribits/common_tpls/find_modules/FindHDF5.cmake
@@ -258,7 +258,7 @@
   endforeach()
 
   # Now match all the -I flags
-  if (${_cflags})
+  if ("${_cflags}")
     string(REGEX MATCHALL "-I([^\" ]+|\"[^\"]+\")" _inc_path_flags ${_cflags})
 
     # Loop through each
--- a/cmake/FortranSettings.cmake
+++ b/cmake/FortranSettings.cmake
@@ -6,6 +6,8 @@
 
 IF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU")
   SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fdefault-real-8 -fdefault-integer-8 -fno-range-check")
+ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "LLVMFlang")
+  SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fdefault-integer-8")
 ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "XL")
   SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -qintsize=8 -qrealsize=8")
 ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "Cray")
--- a/packages/seacas/cmake/FortranSettings.cmake
+++ b/packages/seacas/cmake/FortranSettings.cmake
@@ -6,6 +6,8 @@
 
 IF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU")
   SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fdefault-real-8 -fdefault-integer-8 -fno-range-check")
+ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "LLVMFlang")
+  SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fdefault-integer-8")
 ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "XL")
   SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -qintsize=8 -qrealsize=8")
 ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "Cray")
