--- a/src/multimedia/video/qvideoframeformat.cpp
+++ b/src/multimedia/video/qvideoframeformat.cpp
@@ -19,6 +19,8 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr int QVideoFrameFormat::NPixelFormats;
+
 class QVideoFrameFormatPrivate : public QSharedData
 {
 public:
--- a/src/multimedia/qmediametadata.cpp
+++ b/src/multimedia/qmediametadata.cpp
@@ -14,6 +14,8 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr int QMediaMetaData::NumMetaData;
+
 /*!
     \class QMediaMetaData
     \inmodule QtMultimedia
--- a/src/spatialaudio/qaudioengine.cpp
+++ b/src/spatialaudio/qaudioengine.cpp
@@ -12,6 +12,9 @@
 
 QT_BEGIN_NAMESPACE
 
+constexpr float QAudioEngine::DistanceScaleCentimeter;
+constexpr float QAudioEngine::DistanceScaleMeter;
+
 QAudioEnginePrivate::QAudioEnginePrivate(int sampleRate)
     : m_sampleRate(sampleRate),
       resonanceAudio{
