--- librsvg-2.40.21/Makefile.am.orig	2021-08-24 21:35:42.096743000 -0700
+++ librsvg-2.40.21/Makefile.am	2021-08-24 21:36:48.924886600 -0700
@@ -259,7 +259,7 @@
 
 VAPIGEN_VAPIS = librsvg-$(RSVG_API_VERSION).vapi
 
-librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo
+librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo gdk-pixbuf-2.0
 librsvg_@RSVG_API_VERSION_U@_vapi_METADATADIRS = $(top_srcdir)
 librsvg_@RSVG_API_VERSION_U@_vapi_FILES = $(top_builddir)/Rsvg-$(RSVG_API_VERSION).gir $(top_srcdir)/Rsvg-$(RSVG_API_VERSION)-custom.vala
 
