diff -Naur tk8.6.5-orig/win/configure.in tk8.6.5/win/configure.in
--- tk8.6.5-orig/win/configure.in	2016-02-29 22:13:21.000000000 +0300
+++ tk8.6.5/win/configure.in	2016-03-03 09:21:51.294037200 +0300
@@ -166,7 +166,11 @@
 eval "TK_SRC_DIR=\"`cd $srcdir/..; pwd`\""
 
 eval "TK_DLL_FILE=tk$VER${DLLSUFFIX}"
-eval "TK_LIB_FILE=${LIBPREFIX}tk$VER${LIBSUFFIX}"
+if test ${SHARED_BUILD} = 0 ; then
+  eval "TK_LIB_FILE=${LIBPREFIX}tk$VER${LIBSUFFIX}"
+else
+  eval "TK_LIB_FILE=${LIBPREFIX}tk$VER${DLLSUFFIX}.a"
+fi
 
 eval "TK_STUB_LIB_FILE=${LIBPREFIX}tkstub${VER}${LIBSUFFIX}"
 # FIXME: All of this var junk needs to be done in tcl.m4 !!!!
diff -Naur tk8.6.5-orig/win/Makefile.in tk8.6.5/win/Makefile.in
--- tk8.6.5-orig/win/Makefile.in	2016-02-29 22:13:21.000000000 +0300
+++ tk8.6.5/win/Makefile.in	2016-03-03 09:21:51.324037200 +0300
@@ -135,7 +135,7 @@
 TK_LIB_FILE		= @TK_LIB_FILE@
 TK_DLL_FILE		= @TK_DLL_FILE@
 TEST_DLL_FILE		= tktest$(VER)${DLLSUFFIX}
-TEST_LIB_FILE		= @LIBPREFIX@tktest$(VER)${LIBSUFFIX}
+TEST_LIB_FILE		= @LIBPREFIX@tktest$(VER)${DLLSUFFIX}${LIBSUFFIX}
 
 SHARED_LIBRARIES 	= $(TK_DLL_FILE) $(TK_STUB_LIB_FILE)
 STATIC_LIBRARIES	= $(TK_LIB_FILE)
diff -Naur tk8.6.5-orig/win/tcl.m4 tk8.6.5/win/tcl.m4
--- tk8.6.5-orig/win/tcl.m4	2016-02-29 22:13:21.000000000 +0300
+++ tk8.6.5/win/tcl.m4	2016-03-03 09:21:51.337037200 +0300
@@ -715,7 +715,7 @@
 	SHLIB_LD='${CC} -shared'
 	SHLIB_LD_LIBS='${LIBS}'
 	MAKE_DLL="\${SHLIB_LD} \$(LDFLAGS) -o \[$]@ ${extra_ldflags} \
-	    -Wl,--out-implib,\$(patsubst %.dll,lib%.a,\[$]@)"
+	    -Wl,--out-implib,\$(patsubst %.dll,lib%.dll.a,\[$]@)"
 	# DLLSUFFIX is separate because it is the building block for
 	# users of tclConfig.sh that may build shared or static.
 	DLLSUFFIX="\${DBGX}.dll"
