diff -urN tk8.6.2.orig/win/tcl.m4 tk8.6.2/win/tcl.m4
--- tk8.6.2.orig/win/tcl.m4	2014-11-03 22:23:18.826074700 +0000
+++ tk8.6.2/win/tcl.m4	2014-11-03 22:24:21.781569100 +0000
@@ -635,7 +635,6 @@
 
     if test "${GCC}" = "yes" ; then
 	extra_cflags="-pipe"
-	extra_ldflags="-pipe -static-libgcc"
 	AC_CACHE_CHECK(for mingw32 version of gcc,
 	    ac_cv_win32,
 	    AC_TRY_COMPILE([
@@ -691,6 +690,7 @@
 	if test "${SHARED_BUILD}" = "0" ; then
 	    # static
             AC_MSG_RESULT([using static flags])
+            extra_ldflags="-pipe -static-libgcc"
 	    runtime=
 	    LIBRARIES="\${STATIC_LIBRARIES}"
 	    EXESUFFIX="s\${DBGX}.exe"
