From 32179f547aecfe594579f4797dc598f93c248cbc Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Sat, 27 Jun 2015 13:40:38 +0100
Subject: [PATCH] fix gmtime_r for mingw-w64


diff --git a/CMakeLists.txt b/CMakeLists.txt
index c847a2d..e473861 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,7 +102,7 @@ IF (NOT WSLAY_FOUND)
     SET(WSLAY_LIBRARIES -lwslay)
 ENDIF (NOT WSLAY_FOUND)
 
-SET(CMAKE_C_FLAGS "-O2 -g -Wall -Wno-unused-function ${CMAKE_C_FLAGS} -DH2O_ROOT=\"\\\"${CMAKE_INSTALL_PREFIX}\\\"\"")
+SET(CMAKE_C_FLAGS "-O2 -g -Wall -Wno-unused-function ${CMAKE_C_FLAGS} -D_POSIX_C_SOURCE -DH2O_ROOT=\"\\\"${CMAKE_INSTALL_PREFIX}\\\"\"")
 
 INCLUDE_DIRECTORIES(
     include
diff --git a/include/h2o/time_.h b/include/h2o/time_.h
index ab0db18..8098adc 100644
--- a/include/h2o/time_.h
+++ b/include/h2o/time_.h
@@ -22,6 +22,9 @@
 #ifndef h2o__time_h
 #define h2o__time_h
 
+#ifndef _WIN32
+#define _timezone localt.tm_gmtoff
+#endif
 #include <time.h>
 
 #ifdef __cplusplus
diff --git a/lib/common/time.c b/lib/common/time.c
index 5308848..42a7cbd 100644
--- a/lib/common/time.c
+++ b/lib/common/time.c
@@ -139,7 +139,7 @@ void h2o_time2str_log(char *buf, time_t time)
 {
     struct tm localt;
     localtime_r(&time, &localt);
-    int gmt_off = (int)(localt.tm_gmtoff / 60);
+    int gmt_off = (int)(_timezone / 60);
     int gmt_sign;
 
     if (gmt_off >= 0) {
diff --git a/lib/core/context.c b/lib/core/context.c
index 4ac9f97..7f3f27d 100644
--- a/lib/core/context.c
+++ b/lib/core/context.c
@@ -173,7 +173,7 @@ void h2o_get_timestamp(h2o_context_t *ctx, h2o_mem_pool_t *pool, h2o_timestamp_t
             if (ctx->_timestamp_cache.value != NULL)
                 h2o_mem_release_shared(ctx->_timestamp_cache.value);
             ctx->_timestamp_cache.value = h2o_mem_alloc_shared(NULL, sizeof(h2o_timestamp_string_t), NULL);
-            gmtime_r(&ctx->_timestamp_cache.tv_at.tv_sec, &gmt);
+            gmtime_r((time_t *)&ctx->_timestamp_cache.tv_at.tv_sec, &gmt);
             h2o_time2str_rfc1123(ctx->_timestamp_cache.value->rfc1123, &gmt);
             h2o_time2str_log(ctx->_timestamp_cache.value->log, ctx->_timestamp_cache.tv_at.tv_sec);
         }
-- 
2.4.4

