--- a/src/google/protobuf/metadata_lite.h
+++ b/src/google/protobuf/metadata_lite.h
@@ -70,7 +70,7 @@
     GOOGLE_DCHECK(!is_message_owned || arena != nullptr);
   }
 
-#if defined(NDEBUG) || defined(_MSC_VER)
+#if defined(NDEBUG) || defined(_WIN32)
   ~InternalMetadata() {
     if (HasMessageOwnedArenaTag()) {
       delete reinterpret_cast<Arena*>(ptr_ - kMessageOwnedArenaTagMask);
--- a/src/google/protobuf/message_lite.cc
+++ b/src/google/protobuf/message_lite.cc
@@ -521,7 +521,7 @@
 }
 
 // Non-inline implementations of InternalMetadata routines
-#if defined(NDEBUG) || defined(_MSC_VER)
+#if defined(NDEBUG) || defined(_WIN32)
 // for opt and MSVC builds, the destructor is defined in the header.
 #else
 // This is moved out of the header because the GOOGLE_DCHECK produces a lot of code.
