--- openscad-2014.03/boost.pri.orig	2014-12-21 12:00:21.682400000 +0300
+++ openscad-2014.03/boost.pri	2014-12-21 12:00:52.383200000 +0300
@@ -18,6 +18,13 @@
     BOOST_LINK_FLAGS = -lboost_thread_win32-mt -lboost_program_options-mt -lboost_filesystem-mt -lboost_system-mt -lboost_regex-mt -lboost_chrono-mt
   } 
 
+  win32-g++ {
+    DEFINES += BOOST_STATIC
+    DEFINES += BOOST_THREAD_USE_LIB
+    DEFINES += Boost_USE_STATIC_LIBS
+    BOOST_LINK_FLAGS = -lboost_thread-mt -lboost_program_options-mt -lboost_filesystem-mt -lboost_system-mt -lboost_regex-mt -lboost_chrono-mt
+  }
+
   isEmpty(BOOST_LINK_FLAGS):win* {
     BOOST_LINK_FLAGS = -llibboost_thread-vc90-mt-s-1_46_1 -llibboost_program_options-vc90-mt-s-1_46_1 -llibboost_filesystem-vc90-mt-s-1_46_1 -llibboost_system-vc90-mt-s-1_46_1 -llibboost_regex-vc90-mt-s-1_46_1
   } 
--- openscad-2014.03/cgal.pri.orig	2014-12-21 01:25:50.043200000 +0300
+++ openscad-2014.03/cgal.pri	2014-12-21 10:18:53.992000000 +0300
@@ -11,8 +11,8 @@
     message("CGAL location: $$CGAL_DIR")
   }
 
-  CONFIG(mingw-cross-env) {
-    LIBS += -lgmp -lmpfr -lCGAL
+  win32-g++|CONFIG(mingw-cross-env) {
+    LIBS += -lmpfr -lgmp -lCGAL
     QMAKE_CXXFLAGS += -frounding-math 
   } else {
     win* {
