diff --git a/gtk/meson.build b/gtk/meson.build
index 5a0b1547d8..b008eba4fd 100644
--- a/gtk/meson.build
+++ b/gtk/meson.build
@@ -682,7 +682,7 @@ gtk_gresources_xml = configure_file(output: 'gtk.gresources.xml',
 # Re-build the theme files if sassc is available
 theme_deps = []
 sassc = find_program('sassc', required: false)
-if sassc.found()
+if sassc.found() and false
   sassc_opts = [ '-a', '-M', '-t', 'compact' ]
 
   subdir('theme/Adwaita')
