--- gtk+-3.24.20/meson.build.orig	2020-04-27 16:35:59.000000000 +0200
+++ gtk+-3.24.20/meson.build	2020-05-16 18:19:45.010944100 +0200
@@ -987,16 +987,6 @@
     install_dir : join_paths(gtk_datadir, 'gtk-3.0', 'valgrind'))
 endif
 
-test(
-  'version-check',
-  find_program('check-version.py'),
-  args: [
-    join_paths(meson.current_source_dir(), 'configure.ac'),
-    join_paths(meson.current_source_dir(), 'meson.build'),
-  ],
-  suite: 'gtk',
-)
-
 summary = [
   '',
   '------',
