--- gtk+-2.24.20/gdk/Makefile.am.orig	2013-07-28 21:56:07.983654900 +0000
+++ gtk+-2.24.20/gdk/Makefile.am	2013-07-28 22:00:08.786733000 +0000
@@ -2,6 +2,8 @@
 include $(top_srcdir)/Makefile.decl
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
+INTROSPECTION_SCANNER_ENV = \
+   CC="$(CC)"
 INTROSPECTION_SCANNER_ARGS = \
    --add-include-path=../gdk
 INTROSPECTION_COMPILER_ARGS = \
--- gtk+-2.24.20/gtk/Makefile.am.orig	2013-02-03 20:54:15.000000000 +0000
+++ gtk+-2.24.20/gtk/Makefile.am	2013-07-28 22:06:06.408145200 +0000
@@ -2,6 +2,8 @@
 include $(top_srcdir)/Makefile.decl
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
+INTROSPECTION_SCANNER_ENV = \
+	CC="$(CC)"
 INTROSPECTION_SCANNER_ARGS = \
 	--add-include-path=../gdk
 INTROSPECTION_COMPILER_ARGS = \
