--- gtk+-3.11.8/gdk/win32/gdkdisplay-win32.c.orig	2014-03-08 16:52:14.647800000 +0400
+++ gtk+-3.11.8/gdk/win32/gdkdisplay-win32.c	2014-03-08 17:13:06.329400000 +0400
@@ -424,7 +424,7 @@
   retval = inner_clipboard_window_procedure (hwnd, message, wparam, lparam);
   debug_indent -= 2;
 
-  GDK_NOTE (EVENTS, g_print (" => %I64d%s", (gint64) retval, (debug_indent == 0 ? "\n" : "")));
+  GDK_NOTE (EVENTS, g_print (" => %"G_GINT64_FORMAT"%s", (gint64) retval, (debug_indent == 0 ? "\n" : "")));
 
   return retval;
 }
--- gtk+-3.11.8/gdk/win32/gdkevents-win32.c.orig	2014-03-08 16:52:40.356600000 +0400
+++ gtk+-3.11.8/gdk/win32/gdkevents-win32.c	2014-03-08 17:13:43.722600000 +0400
@@ -292,7 +292,7 @@
   retval = inner_window_procedure (hwnd, message, wparam, lparam);
   debug_indent -= 2;
 
-  GDK_NOTE (EVENTS, g_print (" => %I64d%s", (gint64) retval, (debug_indent == 0 ? "\n" : "")));
+  GDK_NOTE (EVENTS, g_print (" => %"G_GINT64_FORMAT"%s", (gint64) retval, (debug_indent == 0 ? "\n" : "")));
 
   return retval;
 }
@@ -3251,7 +3251,7 @@
       break;
 
     case WM_ACTIVATEAPP:
-      GDK_NOTE (EVENTS, g_print (" %s thread: %I64d",
+      GDK_NOTE (EVENTS, g_print (" %s thread: %"G_GINT64_FORMAT,
 				 msg->wParam ? "YES" : "NO",
 				 (gint64) msg->lParam));
       if (msg->wParam && GDK_WINDOW_IS_MAPPED (window))
