--- gtk+-9e11bb3/gdk/win32/gdkwindow-win32.c.orig	2014-07-25 11:05:40.393869900 +0000
+++ gtk+-9e11bb3/gdk/win32/gdkwindow-win32.c	2014-07-25 11:49:49.131216800 +0000
@@ -450,6 +450,7 @@
   gint x, y, real_x = 0, real_y = 0;
   /* check consistency of redundant information */
   guint remaining_mask = attributes_mask;
+  gboolean suitable_for_alpha;
 
   GDK_NOTE (MISC,
 	    g_print ("_gdk_window_impl_new: %s %s\n",
@@ -512,6 +513,7 @@
       GDK_NOTE (MISC, g_print ("... GDK_INPUT_ONLY\n"));
     }
 
+  suitable_for_alpha = TRUE;
   switch (window->window_type)
     {
     case GDK_WINDOW_TOPLEVEL:
@@ -524,6 +526,7 @@
       if (GDK_WINDOW_TYPE (real_parent) == GDK_WINDOW_FOREIGN)
 	{
 	  dwStyle = WS_CHILDWINDOW | WS_CLIPCHILDREN;
+          suitable_for_alpha = FALSE;
 	}
       else
 	{
@@ -538,12 +541,21 @@
       break;
 
     case GDK_WINDOW_CHILD:
+      suitable_for_alpha = FALSE;
       dwStyle = WS_CHILDWINDOW | WS_CLIPCHILDREN | WS_CLIPSIBLINGS;
       break;
 
     case GDK_WINDOW_TEMP:
       /* A temp window is not necessarily a top level window */
-      dwStyle = (_gdk_root == real_parent ? WS_POPUP : WS_CHILDWINDOW);
+      if (_gdk_root == real_parent)
+        {
+          dwStyle = WS_POPUP;
+        }
+      else
+        {
+          dwStyle = WS_CHILDWINDOW;
+          suitable_for_alpha = FALSE;
+        }
       dwStyle |= WS_CLIPCHILDREN | WS_CLIPSIBLINGS;
       dwExStyle |= WS_EX_TOOLWINDOW | WS_EX_TOPMOST;
       offset_x = _gdk_offset_x;
@@ -692,6 +704,7 @@
   /* Enable blurbehind, but give it an empty region, leaving us
      with all the transparency with none of the blur */
   if (dwmEnableBlurBehindWindow && \
+      suitable_for_alpha && \
       _gdk_win32_is_desktop_composition_enabled () && \
       GDK_WINDOW_HWND (window) != NULL)
     {
