--- a/Makefile.am.orig	2014-10-24 21:22:13.559889300 +0200
+++ b/Makefile.am	2014-10-24 21:22:37.105175200 +0200
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = util gdk gtk libgail-util modules demos tests testsuite examples
+SRC_SUBDIRS = util gdk gtk libgail-util modules
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
--- a/Makefile.in.orig	2014-10-24 21:22:20.402924300 +0200
+++ b/Makefile.in	2014-10-24 21:22:52.339700600 +0200
@@ -540,7 +540,7 @@ XVFB_START = \
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = util gdk gtk libgail-util modules demos tests testsuite examples
+SRC_SUBDIRS = util gdk gtk libgail-util modules
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 MAINTAINERCLEANFILES = \
