--- polars-py-1.43.0.orig/Cargo.toml	2026-07-20 18:54:27.000000000 +0300
+++ polars-py-1.43.0/Cargo.toml	2026-07-21 20:42:16.985303400 +0300
@@ -163,6 +163,7 @@
 question_mark = "allow" # https://github.com/rust-lang/rust-clippy/issues/17384
 
 [patch.crates-io]
+pyo3.path = "../pyo3"
 # packed_simd_2 = { git = "https://github.com/rust-lang/packed_simd", rev = "e57c7ba11386147e6d2cbad7c88f376aab4bdc86" }
 # simd-json = { git = "https://github.com/ritchie46/simd-json", branch = "alignment" }
 # custom DNS resolver - ref https://github.com/apache/arrow-rs-object-store/pull/728
