--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -444,9 +444,9 @@
       get_filename_component(NEW_ABS ${NEW} ABSOLUTE BASE_DIR \${PREFIX_ABS})
       get_filename_component(NEW_DIR \${NEW_ABS} DIRECTORY)
       file(RELATIVE_PATH OLD_REL \${NEW_DIR} \${OLD_ABS})
-      message(STATUS \"Installing symlink: \$ENV{DESTDIR}\${NEW_ABS} -> \${OLD_REL}\")
+      message(STATUS \"Copying file: \$ENV{DESTDIR}\${OLD_ABS} -> \$ENV{DESTDIR}\${NEW_ABS}\")
       file(MAKE_DIRECTORY \$ENV{DESTDIR}\${NEW_DIR})
-      file(CREATE_LINK \${OLD_REL} \$ENV{DESTDIR}\${NEW_ABS} SYMBOLIC)")
+      file(COPY_FILE \$ENV{DESTDIR}\${OLD_ABS} \$ENV{DESTDIR}\${NEW_ABS})")
   endfunction()
 
   mold_install_relative_symlink(${CMAKE_INSTALL_BINDIR}/mold${CMAKE_EXECUTABLE_SUFFIX}
