diff -Nru -x '*~' quassel-0.12.2.orig/cmake/QuasselInstallDirs.cmake quassel-0.12.2/cmake/QuasselInstallDirs.cmake
--- quassel-0.12.2/cmake/QuasselInstallDirs.cmake.orig	2015-04-23 22:47:17.000000000 +0200
+++ quassel-0.12.2/cmake/QuasselInstallDirs.cmake	2015-04-28 10:05:30.094112600 +0200
@@ -14,7 +14,7 @@
 # we make use of its settings.
 
 if (NOT WITH_KDE)
-    if (WIN32)
+    if (WIN32 AND NOT MINGW)
         # On Windows, we have to guess good paths
         # We must check if the variables are already defined on the command line
         if (NOT DEFINED CMAKE_INSTALL_BINDIR)
