diff -bur quassel-0.14.0-orig/src/common/logbacktrace_win.cpp quassel-0.14.0/src/common/logbacktrace_win.cpp
--- quassel-0.14.0-orig/src/common/logbacktrace_win.cpp	2024-01-14 18:27:33.788820600 -0700
+++ quassel-0.14.0/src/common/logbacktrace_win.cpp	2024-01-14 18:32:39.343518200 -0700
@@ -143,6 +143,14 @@
     StackFrame.AddrBStore.Mode = AddrModeFlat;
     StackFrame.AddrStack.Offset = Context.IntSp;
     StackFrame.AddrStack.Mode = AddrModeFlat;
+#elif defined(_M_ARM64)
+    MachineType = IMAGE_FILE_MACHINE_ARM64;
+    StackFrame.AddrPC.Offset = Context.Pc;
+    StackFrame.AddrPC.Mode = AddrModeFlat;
+    StackFrame.AddrFrame.Offset = Context.Fp;
+    StackFrame.AddrFrame.Mode = AddrModeFlat;
+    StackFrame.AddrStack.Offset = Context.Sp;
+    StackFrame.AddrStack.Mode = AddrModeFlat;
 #else
 #    error "Unsupported platform"
 #endif
