--- quassel-0.12.2/src/client/treemodel.cpp.orig	2015-04-23 21:47:17.000000000 +0100
+++ quassel-0.12.2/src/client/treemodel.cpp	2015-07-24 16:25:11.185062500 +0100
@@ -556,9 +556,9 @@
     ChildStatus cs = _childStatus;
 #ifndef QT_NO_DEBUG
     QModelIndex parent = indexByItem(parentItem);
-#endif
     Q_ASSERT(cs.parent == parent);
     Q_ASSERT(rowCount(parent) == cs.childCount + cs.end - cs.start + 1);
+#endif
 
     _aboutToRemoveOrInsert = false;
     for (int i = cs.start; i <= cs.end; i++) {
@@ -605,9 +605,9 @@
 #ifndef QT_NO_DEBUG
     ChildStatus cs = _childStatus;
     QModelIndex parent = indexByItem(parentItem);
-#endif
     Q_ASSERT(cs.parent == parent);
     Q_ASSERT(rowCount(parent) == cs.childCount - cs.end + cs.start - 1);
+#endif
     _aboutToRemoveOrInsert = false;
 
     endRemoveRows();
