--- quassel-0.12.2/cmake/FindQCA2.cmake.orig	2015-07-24 23:39:01.451283000 +0300
+++ quassel-0.12.2/cmake/FindQCA2.cmake	2015-07-24 23:39:06.568092000 +0300
@@ -24,7 +24,7 @@
 else (QCA2_INCLUDE_DIR AND QCA2_LIBRARIES)
 
 
-  if (NOT WIN32)
+  if (NOT MSVC)
     find_package(PkgConfig)
     pkg_check_modules(PC_QCA2 QUIET qca2)
     set(QCA2_DEFINITIONS ${PC_QCA2_CFLAGS_OTHER})
--- quassel-0.12.2/cmake/FindQCA2-QT5.cmake.orig	2015-07-24 23:38:36.241638800 +0300
+++ quassel-0.12.2/cmake/FindQCA2-QT5.cmake	2015-07-24 23:38:44.634453500 +0300
@@ -37,7 +37,7 @@
 else (QCA2-QT5_INCLUDE_DIR AND QCA2-QT5_LIBRARIES)
 
 
-  if (NOT WIN32)
+  if (NOT MSVC)
     find_package(PkgConfig)
     pkg_check_modules(PC_QCA2-QT5 QUIET qca2-qt5)
     set(QCA2-QT5_DEFINITIONS ${PC_QCA2-QT5_CFLAGS_OTHER})
