--- d-feet-0.3.15/src/d-feet.in.orig	2019-08-08 08:56:18.000000000 +0200
+++ d-feet-0.3.15/src/d-feet.in	2020-10-25 14:56:41.848215700 +0100
@@ -33,12 +33,10 @@
 
 import os
 import sys
-sys.path.insert(1, '@pythondir@')
 
 import gettext, locale
 from gettext import gettext as _
 gettext.textdomain("@GETTEXT_PACKAGE@")
-locale.textdomain("@GETTEXT_PACKAGE@")
 
 import gi
 gi.require_version('Gtk', '3.0')
@@ -47,7 +45,7 @@
 
 
 if __name__ == "__main__":
-    data_dir = "@pkgdatadir@"
+    data_dir = sys.prefix + "/share/d-feet"
     #use local paths when debugging
     if os.getenv("DFEET_DEBUG") is not None:
         data_dir = os.path.join(os.path.dirname(__file__), "..", "data")
