From 0d04d12861118bd21a8568babd3a26116b6b11c4 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 15 Feb 2021 17:44:10 +0100
Subject: [PATCH 3/3] Fix LUA_*DIR for MSYS2 FHS layout

---
 src/luaconf.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/luaconf.h b/src/luaconf.h
index d9cf18c..c4b04c3 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -85,8 +85,8 @@
 ** In Windows, any exclamation mark ('!') in the path is replaced by the
 ** path of the directory of the executable file of the current process.
 */
-#define LUA_LDIR	"!\\lua\\"
-#define LUA_CDIR	"!\\"
+#define LUA_LDIR	"!\\..\\share\\lua\\5.1\\"
+#define LUA_CDIR	"!\\..\\lib\\lua\\5.1\\"
 #define LUA_PATH_DEFAULT  \
 		".\\?.lua;"  LUA_LDIR"?.lua;"  LUA_LDIR"?\\init.lua;" \
 		             LUA_CDIR"?.lua;"  LUA_CDIR"?\\init.lua"
-- 
2.30.0.windows.2
