diff --git a/src/Makefile b/src/Makefile
index 23670a4..47d2fe7 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -23,6 +23,7 @@ MYLIBS=
 PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
 LUA_A=	liblua5.1.a
+LUA_LA=	liblua5.1.a
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
 	lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o  \
 	lundump.o lvm.o lzio.o
@@ -52,10 +53,10 @@ $(LUA_A): $(CORE_O) $(LIB_O)
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) iscygpty.o $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) iscygpty.o $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) iscygpty.o $(LUA_LA) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_LA) $(LIBS)
 
 clean:
 	$(RM) $(ALL_T) $(ALL_O)
@@ -104,7 +105,7 @@ macosx:
 #	$(MAKE) all MYCFLAGS=-DLUA_USE_MACOSX
 
 mingw:
-	$(MAKE) "LUA_A=lua51.dll" "LUA_T=lua5.1.exe" \
+	$(MAKE) "LUA_A=lua51.dll" "LUA_LA=liblua5.1.dll.a" "LUA_T=lua5.1.exe" \
 	"AR=$(CC) -shared -Wl,--out-implib,liblua5.1.dll.a -o" "RANLIB=strip --strip-unneeded" \
 	"MYCFLAGS=-DLUA_BUILD_AS_DLL" "MYLIBS=" "MYLDFLAGS=-s" lua5.1.exe
 	$(MAKE) "LUAC_T=luac5.1.exe" luac5.1.exe
