--- src/luaconf.h.orig	2008-02-11 17:25:08.000000000 +0100
+++ src/luaconf.h	2015-03-21 13:04:23.674094700 +0100
@@ -85,8 +85,8 @@
 ** In Windows, any exclamation mark ('!') in the path is replaced by the
 ** path of the directory of the executable file of the current process.
 */
-#define LUA_LDIR	"!\\lua\\"
-#define LUA_CDIR	"!\\"
+#define LUA_LDIR	"!\\..\\share\\lua\\5.1\\"
+#define LUA_CDIR	"!\\..\\lib\\lua\\5.1\\"
 #define LUA_PATH_DEFAULT  \
 		".\\?.lua;"  LUA_LDIR"?.lua;"  LUA_LDIR"?\\init.lua;" \
 		             LUA_CDIR"?.lua;"  LUA_CDIR"?\\init.lua"
