--- partio-1.10.1/src/lib/CMakeLists.txt.orig	2019-07-31 13:58:21.450415100 +0300
+++ partio-1.10.1/src/lib/CMakeLists.txt	2019-07-31 13:59:55.534180400 +0300
@@ -50,7 +50,10 @@
     target_link_libraries(partio PUBLIC ${ZLIB_LIBRARY})
 endif (ZLIB_FOUND)
 
-install(TARGETS partio DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS partio
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 file(GLOB public_includes "*.h")
 install(FILES ${public_includes} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
