
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,6 +91,7 @@ include(GNUInstallDirs)
 set(OpenGL_GL_PREFERENCE GLVND)
 
 ## Search for useful libraries
+find_package(Threads REQUIRED)
 find_package(GLUT REQUIRED)
 find_package(OpenGL REQUIRED)
 find_package(ZLIB)
diff --git a/src/lib/CMakeLists.txt b/src/lib/CMakeLists.txt
index ce7828b..15c8929 100644
--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -42,6 +42,7 @@ target_include_directories(partio
         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
 
+target_link_libraries(partio PUBLIC Threads::Threads)
 if (ZLIB_FOUND)
     target_include_directories(partio
         PUBLIC
