--- a/engauge.pro
+++ b/engauge.pro
@@ -802,7 +802,7 @@
   DESTDIR = bin
 }
 
-linux-* {
+linux-*|win32-g++* {
   QMAKE_CXXFLAGS += -Wunused-parameter
   QT += network
   DEFINES += "NETWORKING"
@@ -941,8 +941,6 @@
                src/Jpeg2000/Jpeg2000Callbacks.cpp \
                src/Jpeg2000/Jpeg2000Color.cpp \
                src/Jpeg2000/Jpeg2000Convert.cpp
-    QMAKE_LFLAGS += -Wl,-rpath=\'\$\$ORIGIN\'
-    QMAKE_POST_LINK += cp $$(OPENJPEG_LIB)/libopenjp2.so.7 bin
 
 } else {
     message("JPEG2000 support:   no")
