--- soqt/CMakeLists.txt.orig	2018-07-22 11:13:30.258458500 +0300
+++ soqt/CMakeLists.txt	2018-07-22 11:14:34.553135900 +0300
@@ -25,6 +25,7 @@
 option(USE_QT5 "Prefer Qt5 over Qt4 if available" ON)
 option(WITH_STATIC_DEFAULTS "Enable statically linked in default materials" ON)
 option(HAVE_SPACENAV_SUPPORT "Enable Space Navigator support" ON)
+option(USE_X11 "Enable X11 support" ON)
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
 
@@ -65,7 +66,9 @@
 	include(${QT_USE_FILE})
 endif()
 
-find_package(X11)
+if(USE_X11)
+	find_package(X11)
+endif()
 
 check_include_files(windows.h HAVE_WINDOWS_H)
 check_include_files(sys/types.h HAVE_SYS_TYPES_H)
@@ -202,12 +202,14 @@
 set(USE_EXCEPTIONS ON)
 set(X_DISPLAY_MISSING ON)
 set(HAVE_JOYSTICK_LINUX OFF)
+if(USE_X11)
 check_include_files(X11/Xlib.h HAVE_X11_AVAILABLE)
 if(HAVE_X11_AVAILABLE)
 	check_include_files(X11/extensions/SGIMisc.h HAVE_X11_EXTENSIONS_SGIMISC_H)
 	check_include_files(X11/extensions/XInput.h HAVE_X11_EXTENSIONS_XINPUT_H)
 	check_include_files(X11/Xproto.h HAVE_X11_XPROTO_H)
 endif()
+endif()
 
 set(PACKAGE ${PROJECT_NAME})
 set(PACKAGE_DESCRIPTION "A Qt Gui-toolkit binding for Coin")
