diff -Naur soqt-orig/CMakeLists.txt soqt/CMakeLists.txt
--- soqt-orig/CMakeLists.txt	2018-07-22 21:48:45.650048800 +0300
+++ soqt/CMakeLists.txt	2018-07-22 21:55:19.723542200 +0300
@@ -306,25 +311,36 @@
 	if(WIN32)
 		configure_file(src/${PROJECT_NAME}.rc.cmake.in src/${PROJECT_NAME}.rc)
 		set(SO${GUI}_RESSOURCE ${CMAKE_BINARY_DIR}/src/${PROJECT_NAME}.rc CACHE INTERNAL "SO${GUI}_RESSOURCE")
+		if (MSVC)
 		set(CMAKE_RELEASE_POSTFIX ${PROJECT_VERSION_MAJOR})
 		set(CMAKE_MINSIZEREL_POSTFIX ${PROJECT_VERSION_MAJOR})
 		set(CMAKE_RELWITHDEBINFO_POSTFIX ${PROJECT_VERSION_MAJOR})
+		endif()
 		set(CMAKE_DEBUG_POSTFIX ${PROJECT_VERSION_MAJOR}d)
 	endif()
 	add_library(${PROJECT_NAME} SHARED ${HDRS} ${COMMON_HDRS} ${SRCS} ${COMMON_SRCS} ${SO${GUI}_RESSOURCE} ${MOC_SRCS})
 	target_compile_definitions(${PROJECT_NAME} PRIVATE SO${GUI}_MAKE_DLL INTERFACE SO${GUI}_DLL)
 else()
 	if(WIN32)
+		if (MSVC)
 		set(CMAKE_RELEASE_POSTFIX ${PROJECT_VERSION_MAJOR}s)
 		set(CMAKE_MINSIZEREL_POSTFIX ${PROJECT_VERSION_MAJOR}s)
 		set(CMAKE_RELWITHDEBINFO_POSTFIX ${PROJECT_VERSION_MAJOR}s)
 		set(CMAKE_DEBUG_POSTFIX ${PROJECT_VERSION_MAJOR}sd)
+		else()
+		set(CMAKE_DEBUG_POSTFIX ${PROJECT_VERSION_MAJOR}d)
+		endif()
 	endif()
 	add_library(${PROJECT_NAME} STATIC ${HDRS} ${COMMON_HDRS} ${SRCS} ${COMMON_SRCS} ${MOC_SRCS})
 	target_compile_definitions(${PROJECT_NAME} INTERFACE SO${GUI}_NOT_DLL)
 endif()
 
-set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${SO${GUI}_VERSION} SOVERSION ${SO${GUI}_SO_VERSION})
+set_target_properties(${PROJECT_NAME} PROPERTIES
+                      VERSION ${SO${GUI}_VERSION}
+                      SOVERSION ${SO${GUI}_SO_VERSION}
+                      OUTPUT_NAME SoQt
+                      RUNTIME_OUTPUT_NAME SoQt
+                      ARCHIVE_OUTPUT_NAME SoQt)
 
 target_compile_definitions(${PROJECT_NAME} PRIVATE HAVE_CONFIG_H SO${GUI}_INTERNAL SO${GUI}_DEBUG=$<CONFIG:Debug>)
 if(MSVC)
@@ -500,6 +516,11 @@
 	COMPONENT development
 )
 
+if(UNIX OR MINGW)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/SoQt.pc.in ${CMAKE_BINARY_DIR}/SoQt.pc @ONLY)
+    install(FILES ${CMAKE_BINARY_DIR}/SoQt.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+endif()
+
 set(CPACK_COMPONENT_DEVELOPMENT_DEPENDS "runtime")
 set(CPACK_DEB_COMPONENT_INSTALL ON)
 set(CPACK_DEBIAN_DEVELOPMENT_PACKAGE_DEPENDS lib${PACKAGE_TARNAME}${QT_VERSION_MAJOR}-${SO${GUI}_SO_VERSION})
diff -Naur soqt-orig/SoQt.pc.in soqt/SoQt.pc.in
--- soqt-orig/SoQt.pc.in	2018-07-22 21:48:35.978031800 +0300
+++ soqt/SoQt.pc.in	2018-07-22 21:50:47.385063500 +0300
@@ -1,22 +1,12 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-datarootdir=@datarootdir@
-datadir=${datarootdir}
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include
 
 Name: SoQt
 Description: a Qt Gui-toolkit binding for Coin
 Version: @SOQT_VERSION@
 Requires: Coin
 Conflicts:
-Libs: -L${libdir} @SOGUI_EXTRA_LDFLAGS@ @SOGUI_EXTRA_LIBS@
-Cflags: -I${includedir} @SOGUI_EXTRA_CFLAGS@ @SOGUI_EXTRA_CPPFLAGS@
-
-soqt_host=@host@
-compiler=@COIN_COMPILER@
-compiler_is_gcc=@ac_compiler_gnu@
-frameworkdir=@frameworkdir@
-htmldir=@htmldir@
-msvcrt=@SOGUI_MSVC_LIBC@
-qt_version=@QT_VERSION@
+Libs: -L${libdir} -lSoQt
+Cflags: -I${includedir} 
