--- pip-26.0/src/pip/_internal/utils/misc.py.orig	2026-01-31 02:15:05.000000000 +0100
+++ pip-26.0/src/pip/_internal/utils/misc.py	2026-01-31 09:05:18.730562700 +0100
@@ -604,7 +604,12 @@
     marker = os.path.join(sysconfig.get_path("stdlib"), "EXTERNALLY-MANAGED")
     if not os.path.isfile(marker):
         return
-    raise ExternallyManagedEnvironment.from_config(marker)
+    exc = ExternallyManagedEnvironment.from_config(marker)
+    if "CI" in os.environ:
+        exc.kind = "warning"
+        logger.warning("%s", exc, extra={"rich": True})
+    else:
+        raise exc
 
 
 def is_console_interactive() -> bool:
