--- a/config/Imath.pc.in
+++ b/config/Imath.pc.in
@@ -7,11 +7,11 @@
 exec_prefix=${prefix}
 libdir=@PKG_CONFIG_INSTALL_LIBDIR@
 includedir=@PKG_CONFIG_INSTALL_INCLUDEDIR@
-libsuffix=@LIB_SUFFIX_DASH@
+
 Name: Imath
 Description: Imath library: vector/matrix and math operations, plus the half type.
 Version: @IMATH_VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -lImath${libsuffix}
+Libs: -L${libdir} -lImath
 Cflags: -I${includedir} -I${includedir}/Imath
+++ b/src/Imath/CMakeLists.cmake
--- a/src/Imath/CMakeLists.cmake
@@ -146,7 +146,7 @@
       COMMENT "Copying resource files to Imath.framework/Resources in build directory"
     )
 else()
-    set_target_properties(${IMATH_LIBRARY} PROPERTIES OUTPUT_NAME "Imath${IMATH_LIB_SUFFIX}" )
+    set_target_properties(${IMATH_LIBRARY} PROPERTIES OUTPUT_NAME "Imath" )
 endif()
   
 #
--- a/src/python/PyImath/CMakeLists.txt
+++ b/src/python/PyImath/CMakeLists.txt
@@ -123,7 +123,7 @@
     MACOSX_RPATH TRUE)
 else()
   set_target_properties(${PYIMATH_LIBRARY} PROPERTIES
-    OUTPUT_NAME "${PYIMATH_LIBRARY}${PYIMATH_LIB_SUFFIX}"
+    OUTPUT_NAME "${PYIMATH_LIBRARY}"
     SOVERSION ${IMATH_LIB_SOVERSION}
     VERSION ${IMATH_LIB_VERSION}
     POSITION_INDEPENDENT_CODE ON
--- a/src/python/PyImath/PyImath.pc.in
+++ b/src/python/PyImath/PyImath.pc.in
@@ -7,10 +7,9 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-libsuffix=@LIB_SUFFIX_DASH@
 
 Name: PyImath
 Description: Python bindings for the Imath libraries
 Version: @IMATH_VERSION@
-Libs: -L${libdir} -lImath${libsuffix} -lPyImath@PYIMATH_LIB_PYTHONVER_ROOT@@Python3_VERSION_MAJOR@_@Python3_VERSION_MINOR@${libsuffix}
+Libs: -L${libdir} -lImath -lPyImath@PYIMATH_LIB_PYTHONVER_ROOT@@Python3_VERSION_MAJOR@_@Python3_VERSION_MINOR@
 Cflags: -I${includedir} -I${includedir}/Imath
