--- Imath-3.2.1/src/Imath/ImathVec.h.orig	2025-08-13 20:54:12.000000000 +0200
+++ Imath-3.2.1/src/Imath/ImathVec.h	2025-08-23 22:22:47.357870400 +0200
@@ -1091,7 +1091,7 @@
 //----------------------------------------------------------------------------
 
 /// @cond Doxygen_Suppress
-
+/*
 // Vec2<short>
 template <>
 IMATH_HOSTDEVICE short Vec2<short>::length () const IMATH_NOEXCEPT = delete;
@@ -1109,7 +1109,7 @@
 template <>
 IMATH_HOSTDEVICE Vec2<short>
 Vec2<short>::normalizedNonNull () const IMATH_NOEXCEPT = delete;
-
+*/
 // Vec2<int>
 template <>
 IMATH_HOSTDEVICE int Vec2<int>::length () const IMATH_NOEXCEPT = delete;
