diff --git a/src/meson.build b/src/meson.build
index 7bd1e52..fa7c198 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -127,9 +127,9 @@ cui_security_link_args = []
 if host_machine.system() == 'windows'
   # binutils does not take into account entry point when
   # -pie is used so we need to provide it manually
-  # The prefix is empty for x86_64, underscore ("_") otherwise
+  # The prefix is underscore ("_") for i686, empty otherwise
   entry_prefix = ''
-  if host_machine.cpu() != 'x86_64'
+  if host_machine.cpu() == 'i686'
     entry_prefix = '_'
   endif
 
