--- a/data/meson.build
+++ b/data/meson.build
@@ -2,32 +2,29 @@ if host_machine.system() != 'windows'
   desktop = 'remote-viewer.desktop'
 
   i18n.merge_file (
-    desktop,
     type: 'desktop',
     input: desktop + '.in',
     output: desktop,
     po_dir: po_dir,
     install: true,
     install_dir: join_paths(datadir, 'applications')
   )
 
   mimetypes = 'virt-viewer-mime.xml'
 
   i18n.merge_file (
-    mimetypes,
     type: 'xml',
     input: mimetypes + '.in',
     output: mimetypes,
     data_dirs: i18n_itsdir,
     po_dir: po_dir,
     install: true,
     install_dir: join_paths(datadir, 'mime', 'packages')
   )
 
   metainfo = 'remote-viewer.appdata.xml'
 
   i18n.merge_file (
-    metainfo,
     type: 'xml',
     input: metainfo + '.in',
     output: metainfo,
--- a/meson.build
+++ b/meson.build
@@ -39,7 +39,7 @@ gtk_vnc_min_version='>=0.4.0'
 spice_gtk_min_version='>=0.35'
 spice_protocol_min_version='>=0.12.7'
 govirt_min_version='>=0.3.7'
-rest_min_version='>=0.8'
+rest_min_version='>=0.9'
 vte_min_version='>=0.46.0'
 bash_completion_version='2.0'
 
@@ -436,7 +436,7 @@ if spice_gtk_dep.found()
 endif
 
 govirt_dep = dependency('govirt-1.0', version: govirt_min_version, required: get_option('ovirt'))
-rest_dep = dependency('rest-0.7', version: rest_min_version, required: get_option('ovirt'))
+rest_dep = dependency('rest-1.0', version: rest_min_version, required: get_option('ovirt'))
 if get_option('ovirt').auto()
   if govirt_dep.found() and not rest_dep.found()
     govirt_dep = dependency('', required: false)
