--- a/demo/data/meson.build
+++ b/demo/data/meson.build
@@ -28,7 +28,8 @@ appdata_config = configuration_data()
 appdata_config.set('BUILD_VERSION', meson.project_version())
 appdata_config.set('BUILD_DATE', today)
 
-if target_system != 'windows'
+appstream_util = find_program('appstream-util', required: false)
+if appstream_util.found()
   appstream_file = i18n.merge_file(
     input: configure_file(
       input: 'org.gnome.Adwaita1.Demo.metainfo.xml.in.in',
@@ -40,10 +41,7 @@ if target_system != 'windows'
     install: true,
     install_dir: datadir / 'metainfo'
   )
-endif
 
-appstream_util = find_program('appstream-util', required: false)
-if appstream_util.found()
   test('Validate appstream file', appstream_util,
     args: ['validate', '--nonet', appstream_file]
   )
