--- libgphoto2-2.5.12/configure.ac.orig	2017-01-28 00:15:31.342691400 -0500
+++ libgphoto2-2.5.12/configure.ac	2017-01-28 00:45:08.154549500 -0500
@@ -286,11 +286,7 @@
 dnl check for libgd
 dnl they call it "gdlib" ... confusing.
 dnl ---------------------------------------------------------------------------
-GP_CHECK_LIBRARY([LIBGD],[gdlib],[>= 2.0],[gd.h],[gdImageCreateTrueColor],[
-	# gdlib-config --libs upstream does not include -lgd, but only the dependencies
-	LIBGD_LIBS="-lgd $LIBGD_LIBS"
-
-],[],[default-on],[http://www.libgd.org/])
+PKG_CHECK_MODULES(LIBGD, [gdlib])
 GP_CONFIG_MSG([libGD conversion support],[${have_LIBGD}])
 
 dnl ---------------------------------------------------------------------------
