--- root/camlibs/lumix/lumix.c.orig	2021-01-24 12:38:56.661000000 +0100
+++ root/camlibs/lumix/lumix.c	2021-01-24 12:40:57.984864300 +0100
@@ -36,10 +36,14 @@
 #include <libxml/xmlreader.h>
 
 
-#include <sys/socket.h>
 #include <stdlib.h>
+#ifndef WIN32
+#include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
+#else
+#include <winsock.h>
+#endif
 
 
 #include <gphoto2/gphoto2-library.h>
@@ -132,6 +136,7 @@
 static int
 camera_capture_preview (Camera *camera, CameraFile *file, GPContext *context)
 {
+#ifndef WIN32
 	int			valread;
 	struct sockaddr_in	serv_addr;
 	unsigned char		buffer[65536];
@@ -192,6 +197,10 @@
 		return gp_file_append (file, (char*)buffer+start, end-start);
 	}
 	return GP_ERROR;
+#else
+	GP_LOG_E ("not supported currently on Windows");
+	return GP_ERROR;
+#endif
 }
 
 static int camera_about (Camera *camera, CameraText *about, GPContext *context);
