--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,7 @@
 include(CMakeTest)
 
 # installation rules for a project
-set(BINDIR "${CMAKE_INSTALL_PREFIX}/bin")
+set(BINDIR "bin")
 install(TARGETS osslsigncode RUNTIME DESTINATION ${BINDIR})
 if(UNIX)
   include(CMakeDist)
--- osslsigncode-2.5/cmake/SetBashCompletion.cmake.orig	2022-09-06 09:09:36.005390200 +0200
+++ osslsigncode-2.5/cmake/SetBashCompletion.cmake	2022-09-06 09:09:35.015646800 +0200
@@ -1,5 +1,7 @@
 if(NOT MSVC)
-  find_package(bash-completion QUIET)
+  if(NOT MINGW)
+    find_package(bash-completion QUIET)
+  endif()
   if(NOT BASH_COMPLETION_COMPLETIONSDIR)
     if(BASH_COMPLETION_COMPATDIR)
       set(BASH_COMPLETION_COMPLETIONSDIR ${BASH_COMPLETION_COMPATDIR})
