--- osslsigncode-2.6/CMakeLists.txt	2023-05-29 23:10:39.000000000 +0200
+++ osslsigncode-2.6/CMakeLists.txt	2023-07-13 20:47:07.902389000 +0200
@@ -79,7 +79,7 @@
 include(CMakeTest)
 
 # installation rules for a project
-set(BINDIR "${CMAKE_INSTALL_PREFIX}/bin")
+set(BINDIR "bin")
 install(TARGETS osslsigncode RUNTIME DESTINATION ${BINDIR})
 if(UNIX)
     include(CMakeDist)
--- osslsigncode-2.6/cmake/SetBashCompletion.cmake	2023-05-29 23:10:39.000000000 +0200
+++ osslsigncode-2.6/cmake/SetBashCompletion.cmake	2023-07-13 20:49:06.998166600 +0200
@@ -11,7 +11,9 @@
     if(BASH_COMPLETION_USER_DIR)
         set(BASH_COMPLETION_COMPLETIONSDIR "${BASH_COMPLETION_USER_DIR}/bash-completion/completions")
     else(BASH_COMPLETION_USER_DIR)
-        find_package(bash-completion QUIET)
+        if(NOT MINGW)
+            find_package(bash-completion QUIET)
+        endif()
         if(NOT BASH_COMPLETION_FOUND)
             set(SHAREDIR "${CMAKE_INSTALL_PREFIX}/share")
             set(BASH_COMPLETION_COMPLETIONSDIR "${SHAREDIR}/bash-completion/completions")
