From 921caeeb1cc6d9649647a7c20fe7101176221f6c Mon Sep 17 00:00:00 2001
From: Philippe Renon <philippe_renon@yahoo.fr>
Date: Fri, 9 Feb 2018 10:54:08 +0100
Subject: [PATCH 3/3] add with gflags option

---
 CMakeLists.txt | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb502da..43ab9c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,17 @@ else()
     endif()
   endif()
 
+  # No config file for this
+  option(WITH_GFLAGS "build with GFlags" ON)
+  if(WITH_GFLAGS)
+    find_package(gflags)
+    if(gflags_FOUND)
+      add_definitions(-DGFLAGS=1)
+      include_directories(${gflags_INCLUDE_DIR})
+      list(APPEND THIRDPARTY_LIBS ${gflags_LIBRARIES})
+    endif()
+  endif()
+
   option(WITH_SNAPPY "build with SNAPPY" OFF)
   if(WITH_SNAPPY)
     find_package(snappy REQUIRED)
@@ -253,13 +264,6 @@ if(WITH_UBSAN)
   endif()
 endif()
 
-find_package(gflags)
-if(gflags_FOUND)
-  add_definitions(-DGFLAGS=1)
-  include_directories(${gflags_INCLUDE_DIR})
-  list(APPEND THIRDPARTY_LIBS ${gflags_LIBRARIES})
-endif()
-
 find_package(NUMA)
 if(NUMA_FOUND)
   add_definitions(-DNUMA)
-- 
2.10.2

