--- rocksdb-rocksdb-5.15.10/CMakeLists.txt.orig	2018-02-13 14:09:02.887740400 +0300
+++ rocksdb-rocksdb-5.15.10/CMakeLists.txt	2018-02-13 14:14:31.135665800 +0300
@@ -736,7 +736,11 @@
 endif()
 
 if(WIN32)
+  if(MSVC)
   set(SYSTEM_LIBS ${SYSTEM_LIBS} Shlwapi.lib Rpcrt4.lib)
+  else()
+  set(SYSTEM_LIBS ${SYSTEM_LIBS} shlwapi rpcrt4)
+  endif()
   set(LIBS ${ROCKSDB_STATIC_LIB} ${THIRDPARTY_LIBS} ${SYSTEM_LIBS})
 else()
   set(SYSTEM_LIBS ${CMAKE_THREAD_LIBS_INIT})
@@ -767,6 +771,7 @@
     ${THIRDPARTY_LIBS} ${SYSTEM_LIBS})
   set_target_properties(${ROCKSDB_IMPORT_LIB} PROPERTIES
     COMPILE_DEFINITIONS "ROCKSDB_DLL;ROCKSDB_LIBRARY_EXPORTS")
+  set_target_properties(${ROCKSDB_IMPORT_LIB} PROPERTIES ARCHIVE_OUTPUT_NAME "rocksdb")
   if(MSVC)
     set_target_properties(${ROCKSDB_STATIC_LIB} PROPERTIES
       COMPILE_FLAGS "/Fd${CMAKE_CFG_INTDIR}/${ROCKSDB_STATIC_LIB}.pdb")