--- lensfun-0.3.95/CMakeLists.txt.orig	2018-07-09 11:46:35.789079600 +0300
+++ lensfun-0.3.95/CMakeLists.txt	2018-07-09 11:47:42.701675200 +0300
@@ -55,7 +55,7 @@
 ENDIF()
 
 # overwrite some folders before GNUInstallDirs is loaded
-IF (WIN32)
+IF (MSVC)
     SET(CMAKE_INSTALL_DATAROOTDIR "$ENV{ALLUSERSPROFILE}" CACHE PATH "read-only architecture-independent data root (share)")
 ENDIF()
 
@@ -78,7 +78,7 @@
   ENDIF()
 ENDIF()
 
-IF(WIN32)
+IF(MSVC)
   # base path for searching for glib on windows
   IF(NOT GLIB2_BASE_DIR)
     SET(GLIB2_BASE_DIR "${SOURCE_BASE_DIR}/glib-2.28.1" CACHE STRING "Base path of glib2 dir." FORCE)
@@ -174,7 +174,7 @@
   ADD_SUBDIRECTORY(docs)
 ENDIF()
 
-IF(WIN32 AND NOT BUILD_STATIC)
+IF(MSVC AND NOT BUILD_STATIC)
   FIND_FILE(GLIB2_DLL
             NAMES glib-2.dll glib-2-vs9.dll
             PATHS "${GLIB2_BASE_DIR}/bin"
