diff -Naur lensfun-0.3.3.orig/CMakeLists.txt lensfun-0.3.3/CMakeLists.txt
--- lensfun-0.3.3.orig/CMakeLists.txt	2022-02-22 21:49:04.000000000 +0100
+++ lensfun-0.3.3/CMakeLists.txt	2022-02-28 11:18:22.881354000 +0100
@@ -81,6 +81,8 @@
 
 IF(WIN32)
   INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/libs/getopt)
+ENDIF()
+IF(MSVC)
   # base path for searching for glib on windows
   IF(NOT GLIB2_BASE_DIR)
     SET(GLIB2_BASE_DIR "${SOURCE_BASE_DIR}/glib-2.28.1" CACHE STRING "Base path of glib2 dir." FORCE)
@@ -175,7 +177,7 @@
   ADD_SUBDIRECTORY(docs)
 ENDIF()
 
-IF(WIN32 AND NOT BUILD_STATIC)
+IF(MSVC AND NOT BUILD_STATIC)
   FIND_FILE(GLIB2_DLL 
             NAMES glib-2.dll glib-2-vs9.dll
             PATHS "${GLIB2_BASE_DIR}/bin"
diff -Naur lensfun-0.3.3.orig/libs/CMakeLists.txt lensfun-0.3.3/libs/CMakeLists.txt
--- lensfun-0.3.3.orig/libs/CMakeLists.txt	2022-02-22 21:49:04.000000000 +0100
+++ lensfun-0.3.3/libs/CMakeLists.txt	2022-02-28 11:19:12.916447900 +0100
@@ -5,6 +5,6 @@
 ADD_SUBDIRECTORY(lensfun)
 
 # also build getopt on windows
-IF(WIN32 AND BUILD_TESTS)
+IF(WIN32 AND BUILD_LENSTOOL)
   ADD_SUBDIRECTORY(getopt)
 ENDIF()
diff -Naur lensfun-0.3.3.orig/libs/lensfun/CMakeLists.txt lensfun-0.3.3/libs/lensfun/CMakeLists.txt
--- lensfun-0.3.3.orig/libs/lensfun/CMakeLists.txt	2022-02-22 21:49:04.000000000 +0100
+++ lensfun-0.3.3/libs/lensfun/CMakeLists.txt	2022-02-28 11:20:16.412851600 +0100
@@ -24,6 +24,8 @@
 
 IF(NOT HAVE_REGEX_H)
   TARGET_LINK_LIBRARIES(lensfun tre_regex)
+ELSE()
+  TARGET_LINK_LIBRARIES(lensfun regex)
 ENDIF()
 TARGET_LINK_LIBRARIES(lensfun ${GLIB2_LIBRARIES})
 
diff -Naur lensfun-0.3.3.orig/apps/CMakeLists.txt lensfun-0.3.3/apps/CMakeLists.txt
--- lensfun-0.3.3.orig/apps/CMakeLists.txt	2022-02-22 21:49:04.000000000 +0100
+++ lensfun-0.3.3/apps/CMakeLists.txt	2022-02-28 14:34:39.465244700 +0100
@@ -42,5 +42,5 @@
     IF(NOT DEFINED SETUP_PY_INSTALL_PREFIX)
       SET(SETUP_PY_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
     ENDIF()
-    INSTALL(CODE "execute_process(COMMAND ${PYTHON} ${SETUP_PY} install --prefix=\$ENV{DESTDIR}${SETUP_PY_INSTALL_PREFIX})")
+    INSTALL(CODE "execute_process(COMMAND ${PYTHON} ${SETUP_PY} install --prefix=${SETUP_PY_INSTALL_PREFIX} --root=\$ENV{DESTDIR})")
 ENDIF(PYTHON)
