--- lensfun-0.3.95/apps/lensfun/__init__.py.in.orig	2018-07-09 11:52:25.741609300 +0300
+++ lensfun-0.3.95/apps/lensfun/__init__.py.in	2018-07-09 11:53:57.496739700 +0300
@@ -14,12 +14,13 @@
 """
 
 import os
+import sys
 
 
 user_local_db_path = os.path.join(
     os.environ.get("XDG_DATA_HOME", os.path.expanduser("~/.local/share")), "lensfun")
-system_db_path = os.path.join("${CMAKE_INSTALL_FULL_DATAROOTDIR}", "lensfun")
-system_db_update_path = os.path.join("/", "${CMAKE_INSTALL_LOCALSTATEDIR}", "lib", "lensfun-updates")
+system_db_path = os.path.join(sys.prefix, "share", "lensfun")
+system_db_update_path = os.path.join(sys.prefix, "var", "lib", "lensfun-updates")
 
 
 def get_database_version():
