diff -Naur lensfun-0.3.3.orig/apps/lensfun/__init__.py.in lensfun-0.3.3/apps/lensfun/__init__.py.in
--- lensfun-0.3.3.orig/apps/lensfun/__init__.py.in	2022-02-22 21:49:04.000000000 +0100
+++ lensfun-0.3.3/apps/lensfun/__init__.py.in	2022-02-28 11:38:35.240176900 +0100
@@ -14,12 +14,13 @@
 """
 
 import os
+import sys
 
 
 user_local_db_path = os.path.join(
     os.environ.get("XDG_DATA_HOME", os.path.expanduser("~/.local/share")), "lensfun")
 system_db_path = os.path.join("${CMAKE_INSTALL_FULL_DATAROOTDIR}", "lensfun")
-system_db_update_path = os.path.join("/", "${CMAKE_INSTALL_LOCALSTATEDIR}", "lib", "lensfun-updates")
+system_db_update_path = os.path.join(sys.prefix, "${CMAKE_INSTALL_LOCALSTATEDIR}", "lib", "lensfun-updates")
 
 
 def get_database_version():
