--- lensfun-0.3.2/apps/lensfun-update-data.orig	2015-11-15 18:07:26.000000000 +0100
+++ lensfun-0.3.2/apps/lensfun-update-data	2017-09-03 15:46:53.005426300 +0200
@@ -48,7 +48,8 @@
 
 database_version = lensfun.get_database_version()
 
-lensfun_updates_dir = "/var/lib/lensfun-updates"
+lensfun_updates_dir = os.path.join(sys.prefix, "var", "lib", "lensfun-updates")
+
 if os.name == "posix" and os.geteuid() != 0:
     lensfun_updates_dir = os.path.join(os.path.expanduser("~"),
                                        ".local", "share", "lensfun", "updates")
