--- configure.old	2013-12-06 16:55:09.098654166 +0100
+++ configure	2013-12-06 16:55:20.298676545 +0100
@@ -45,7 +45,7 @@
     }
 
 # Check for tools
-tibs.check_program ("GNU Make", "make --version", ".*?([0-9\.]+).*", "3.81", True)
+tibs.check_program ("GNU Make", "make --version", ".*?([0-9\.]+).*", "4.0", True)
 if tibs.check_program ("makedep", "makedep -V", ".*Version.*?([0-9\.]+).*", "0.1.0"):
     tibs.add_config_mak ("MAKEDEP", "makedep")
     makedep = "yes"
