--- src/luaconf.h.orig	2013-11-21 18:26:16.000000000 +0100
+++ src/luaconf.h	2015-03-21 13:02:03.550621100 +0100
@@ -89,8 +89,9 @@
 ** In Windows, any exclamation mark ('!') in the path is replaced by the
 ** path of the directory of the executable file of the current process.
 */
-#define LUA_LDIR	"!\\lua\\"
-#define LUA_CDIR	"!\\"
+#define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "\\"
+#define LUA_LDIR	"!\\..\\share\\lua\\" LUA_VDIR
+#define LUA_CDIR	"!\\..\\lib\\lua\\" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
 		LUA_LDIR"?.lua;"  LUA_LDIR"?\\init.lua;" \
 		LUA_CDIR"?.lua;"  LUA_CDIR"?\\init.lua;" ".\\?.lua"
