--- cegui-0.8.3/cmake/FindDirectXSDK.cmake.orig	2014-04-21 10:46:47.705000000 +0400
+++ cegui-0.8.3/cmake/FindDirectXSDK.cmake	2014-04-21 11:13:02.642600000 +0400
@@ -9,15 +9,15 @@
         set (DIRECTXSDK_ARCH x64)
     endif()
 
-    find_path(DIRECTXSDK_H_PATH NAMES d3d10.h PATHS ENV DXSDK_DIR PATH_SUFFIXES Include NO_DEFAULT_PATH)
-    find_path(DIRECTXSDK_LIB_PATH NAMES d3dx10d.lib PATHS ENV DXSDK_DIR PATH_SUFFIXES "Lib/${DIRECTXSDK_ARCH}")
+    find_path(DIRECTXSDK_H_PATH NAMES d3d10.h PATHS ENV DXSDK_DIR PATH_SUFFIXES include NO_DEFAULT_PATH)
+    find_path(DIRECTXSDK_LIB_PATH NAMES libd3dx10.a PATHS ENV DXSDK_DIR PATH_SUFFIXES lib)
 
     # now test for the specific d3d lib versions and set MAX_D3D accordingly
-    if (EXISTS "${DIRECTXSDK_LIB_PATH}/d3d11.lib")
+    if (EXISTS "${DIRECTXSDK_LIB_PATH}/libd3d11.a")
         set (DIRECTXSDK_MAX_D3D 11)
-    elseif (EXISTS "${DIRECTXSDK_LIB_PATH}/d3d10.lib")
+    elseif (EXISTS "${DIRECTXSDK_LIB_PATH}/libd3d10.a")
         set (DIRECTXSDK_MAX_D3D 10)
-    elseif (EXISTS "${DIRECTXSDK_LIB_PATH}/d3d9.lib")
+    elseif (EXISTS "${DIRECTXSDK_LIB_PATH}/libd3d9.a")
         set (DIRECTXSDK_MAX_D3D 9)
     else()
         set (DIRECTXSDK_MAX_D3D)
