--- zxing_cpp-3.0.0/setup.py.orig	2026-05-08 13:07:33.255171400 +0200
+++ zxing_cpp-3.0.0/setup.py	2026-05-08 13:08:02.156010300 +0200
@@ -23,7 +23,7 @@
 		]
 		build_args = ['--config', cfg, '-j', '8']
 
-		if platform.system() == 'Windows':
+		if platform.system() == 'Windows' and "MSC" in platform.python_compiler():
 			cmake_args += [f'-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{cfg.upper()}={extdir}']
 			cmake_args += ['-A', 'x64' if sys.maxsize > 2**32 else 'Win32']
 			build_args += ['--', '/m']
