Index: configure.ac
===================================================================
--- a/configure.ac	(revision 18912)
+++ b/configure.ac	(working copy)
@@ -176,16 +176,6 @@ fi
 AM_CONDITIONAL([HAVE_PKGCONFIG], [test "$PKGCONFIG" != "no"])
 
 # Collect flags for shout.pc
-
-# I hate myself for doing this.
-save_prefix="$prefix"
-if test "$prefix" = "NONE"
-then
-  prefix="$ac_default_prefix"
-fi
-eval shout_includedir="$includedir"
-prefix="$save_prefix"
-
 XIPH_PATH_OPENSSL([
     XIPH_VAR_APPEND([XIPH_CPPFLAGS],[$OPENSSL_CFLAGS])
     XIPH_VAR_APPEND([XIPH_LDFLAGS],[$OPENSSL_LDFLAGS])
@@ -203,13 +203,11 @@ fi
 SHOUT_CFLAGS="$PTHREAD_CFLAGS"
 SHOUT_LIBS="-lshout"
 
-XIPH_CLEAN_CCFLAGS([$SHOUT_CPPFLAGS], [SHOUT_CPPFLAGS])
 XIPH_CLEAN_CCFLAGS([$SHOUT_CFLAGS], [SHOUT_CFLAGS])
 XIPH_CLEAN_CCFLAGS([$VORBIS_LIBS $THEORA_LIBS $SPEEX_LIBS $PTHREAD_LIBS $OPENSSL_LIBS $OPENSSL_LIBS $LIBS], [SHOUT_LIBDEPS])
 AC_SUBST(PTHREAD_CPPFLAGS)
 AC_SUBST(SHOUT_LIBDEPS)
 AC_SUBST(SHOUT_REQUIRES)
-AC_SUBST(SHOUT_CPPFLAGS)
 AC_SUBST(SHOUT_CFLAGS)
 
 dnl Make substitutions
Index: shout.pc.in
===================================================================
--- a/shout.pc.in	(revision 18912)
+++ b/shout.pc.in	(working copy)
@@ -4,8 +4,6 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-cppflags=@SHOUT_CPPFLAGS@
-cflags_only=@SHOUT_CFLAGS@
 
 Name: Shout
 Description: Audio streaming library for icecast encoders
