--- libshout-2.2.2/src/Makefile.am.orig	2012-04-04 20:01:57 +0400
+++ libshout-2.2.2/src/Makefile.am	2012-04-04 20:07:34 +0400
@@ -18,7 +18,7 @@
 SUBDIRS = avl net timing httpp $(MAYBE_THREAD)
 
 lib_LTLIBRARIES = libshout.la
-libshout_la_LDFLAGS = -version-info 5:0:2
+libshout_la_LDFLAGS = -version-info 5:0:2 -no-undefined
 
 EXTRA_DIST = theora.c speex.c
 noinst_HEADERS = shout_ogg.h shout_private.h util.h
