diff --git a/cmake/system_check.cmake b/cmake/system_check.cmake
index 86ce3dfb..dc5e6eaa 100644
--- a/cmake/system_check.cmake
+++ b/cmake/system_check.cmake
@@ -24,7 +24,7 @@ if(MINGW)
     execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpmachine
               OUTPUT_VARIABLE OPENBLAS_MINGW_TARGET_MACHINE
               OUTPUT_STRIP_TRAILING_WHITESPACE)
-    if(OPENBLAS_MINGW_TARGET_MACHINE MATCHES "amd64|x86_64|AMD64")
+    if(OPENBLAS_MINGW_TARGET_MACHINE MATCHES "amd64|x86_64|AMD64|aarch64")
       set(MINGW64 1)
     endif()
 endif()
diff --git a/common.h b/common.h
index 00d1d0ba..0dd5dcfb 100644
--- a/common.h
+++ b/common.h
@@ -354,7 +354,7 @@ typedef int blasint;
 #endif
 #endif
 
-#if defined(ARMV7) || defined(ARMV6) || defined(ARMV8) || defined(ARMV5)
+#if !defined(OS_WINDOWS) && (defined(ARMV7) || defined(ARMV6) || defined(ARMV8) || defined(ARMV5))
 #define YIELDING        __asm__ __volatile__ ("nop;nop;nop;nop;nop;nop;nop;nop; \n");
 #endif
 
