--- a/cmake/OpenBLASConfig.cmake.in
+++ b/cmake/OpenBLASConfig.cmake.in
@@ -48,9 +48,13 @@
 
 set(PN OpenBLAS)
 
+include(CMakeFindDependencyMacro)
+
 # need to check that the @USE_*@ evaluate to something cmake can perform boolean logic upon
 if(@USE_OPENMP@)
     set(${PN}@SUFFIX64@_openmp_FOUND 1)
+    enable_language(C Fortran)
+    find_dependency(OpenMP COMPONENTS C Fortran REQUIRED)
 elseif(@USE_THREAD@)
     set(${PN}@SUFFIX64@_pthread_FOUND 1)
 else()
