--- qtwebkit-tp5/Source/cmake/WebKitMacros.cmake.orig	2017-02-08 00:27:15.676621600 +0300
+++ qtwebkit-tp5/Source/cmake/WebKitMacros.cmake	2017-02-08 00:27:36.682823100 +0300
@@ -374,7 +374,7 @@
 endmacro()
 
 macro(MAKE_JS_FILE_ARRAYS _output_cpp _output_h _scripts _scripts_dependencies)
-    if (WIN32)
+    if (MSVC)
         set(_python_path set "PYTHONPATH=${JavaScriptCore_SCRIPTS_DIR}" &&)
     else ()
         set(_python_path "PYTHONPATH=${JavaScriptCore_SCRIPTS_DIR}")
