--- qtwebkit-tp5/Source/WebKit/PlatformQt.cmake.orig	2017-01-27 00:27:15.000000000 +0300
+++ qtwebkit-tp5/Source/WebKit/PlatformQt.cmake	2017-02-10 09:29:21.665520300 +0300
@@ -473,6 +473,11 @@
     set(WebKit_OUTPUT_NAME QtWebKit)
 else ()
     set(WebKit_OUTPUT_NAME Qt5WebKit)
+    if(MINGW)
+        set(LIB_PREFIX "")
+        set(WebKit_RUNTIME_OUTPUT_NAME Qt5WebKit)
+        set(WebKit_ARCHIVE_OUTPUT_NAME Qt5WebKit)
+    endif()
 endif ()
 
 
@@ -671,6 +680,11 @@
     set(WebKitWidgets_OUTPUT_NAME QtWebKitWidgets)
 else ()
     set(WebKitWidgets_OUTPUT_NAME Qt5WebKitWidgets)
+    if(MINGW)
+        set(LIB_PREFIX "")
+        set(WebKitWidgets_RUNTIME_OUTPUT_NAME Qt5WebKitWidgets)
+        set(WebKitWidgets_ARCHIVE_OUTPUT_NAME Qt5WebKitWidgets)
+    endif()
 endif ()
 set(WebKitWidgets_PRIVATE_HEADERS_LOCATION Headers/${PROJECT_VERSION}/QtWebKitWidgets/Private)
 
--- qtwebkit-tp5/Source/cmake/WebKitMacros.cmake.orig	2017-02-10 10:36:28.138759600 +0300
+++ qtwebkit-tp5/Source/cmake/WebKitMacros.cmake	2017-02-10 10:37:48.748942100 +0300
@@ -272,6 +272,14 @@
         set_target_properties(${_target} PROPERTIES OUTPUT_NAME ${${_target}_OUTPUT_NAME})
     endif ()
 
+    if (${_target}_RUNTIME_OUTPUT_NAME)
+        set_target_properties(${_target} PROPERTIES RUNTIME_OUTPUT_NAME ${${_target}_RUNTIME_OUTPUT_NAME})
+    endif ()
+    
+    if (${_target}_ARCHIVE_OUTPUT_NAME)
+        set_target_properties(${_target} PROPERTIES ARCHIVE_OUTPUT_NAME ${${_target}_ARCHIVE_OUTPUT_NAME})
+    endif ()
+    
     if (${_target}_PRE_BUILD_COMMAND)
         add_custom_target(_${_target}_PreBuild COMMAND ${${_target}_PRE_BUILD_COMMAND} VERBATIM)
         add_dependencies(${_target} _${_target}_PreBuild)
