--- qtwebkit-5.212.0-alpha3/Source/WebCore/dom/Document.cpp.orig	2019-11-08 11:33:20.999642500 +0300
+++ qtwebkit-5.212.0-alpha3/Source/WebCore/dom/Document.cpp	2019-11-08 11:34:00.987712500 +0300
@@ -4407,12 +4407,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -4474,7 +4474,7 @@
 
     for (unsigned i = 0; i < length;) {
         UChar32 c;
-        U16_NEXT(qualifiedName, i, length, c)
+        U16_NEXT(qualifiedName, i, length, c);
         if (c == ':') {
             if (sawColon) {
                 ec = NAMESPACE_ERR;
