--- qtwebkit-5.212.0-alpha2\\Source\\JavaScriptCore\\inspector\\InjectedScriptManager.cpp.orig        2017-06-05 06:16:05 +0000
+++ qtwebkit-5.212.0-alpha2\\Source\\JavaScriptCore\\inspector\\InjectedScriptManager.cpp     2019-12-20 10:45:22 +0000
@@ -172,6 +172,12 @@
             return it1->value;
     }
 
+#ifdef __MINGW32__
+/* javascript debugging using external python scripts buggy for MinGW
+   for now just disable it. */
+    return InjectedScript();
+#endif
+
     if (!m_environment.canAccessInspectedScriptState(inspectedExecState))
         return InjectedScript();
