diff --git a/Source/cmake/OptionsQt.cmake b/Source/cmake/OptionsQt.cmake
index 1ee60b7..bf7997b 100644
--- a/Source/cmake/OptionsQt.cmake
+++ b/Source/cmake/OptionsQt.cmake
@@ -709,6 +709,15 @@ if (COMPILER_IS_GCC_OR_CLANG AND UNIX)
     endif ()
 endif ()
 
+if (COMPILER_IS_GCC_OR_CLANG)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations -Wno-nonnull -Wno-switch -Wno-unused-result")
+    if (COMPILER_IS_CLANG)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-builtins -Wno-implicit-int-float-conversion")
+    else ()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-class-memaccess -Wno-nonnull-compare -Wno-redundant-move -Wno-template-id-cdtor -Wno-uninitialized")
+    endif ()
+endif ()
+
 # Improvised backport of r222112 - not needed with current WebKit
 # -Wexpansion-to-defined produces false positives with GCC but not Clang
 # https://bugs.webkit.org/show_bug.cgi?id=167643#c13
diff --git a/Source/JavaScriptCore/CMakeLists.txt b/Source/JavaScriptCore/CMakeLists.txt
index afd76ca..df45077 100644
--- a/Source/JavaScriptCore/CMakeLists.txt
+++ b/Source/JavaScriptCore/CMakeLists.txt
@@ -1321,6 +1321,15 @@ endif ()
 
 QT_ADD_EXTRA_WEBKIT_TARGET_EXPORT(JavaScriptCore)
 
+if (COMPILER_IS_GCC_OR_CLANG)
+    ADD_TARGET_PROPERTIES(JavaScriptCore COMPILE_FLAGS "-Wno-sign-compare -Wno-unused -Wno-unused-parameter -Wno-unused-variable")
+    if (COMPILER_IS_CLANG)
+        ADD_TARGET_PROPERTIES(JavaScriptCore COMPILE_FLAGS "-Wno-unknown-pragmas -Wno-deprecated-copy-with-user-provided-copy")
+    else ()
+        ADD_TARGET_PROPERTIES(JavaScriptCore COMPILE_FLAGS "-Wno-cast-function-type -Wno-deprecated -Wno-implicit-fallthrough")
+    endif ()
+endif ()
+
 # Force staging of shared scripts, even if they aren't directly used to build JavaScriptCore.
 
 add_custom_target(stageSharedScripts DEPENDS ${JavaScriptCore_SCRIPTS})
diff --git a/Source/WebCore/CMakeLists.txt b/Source/WebCore/CMakeLists.txt
index 3b3a9ed..ad75533 100644
--- a/Source/WebCore/CMakeLists.txt
+++ b/Source/WebCore/CMakeLists.txt
@@ -3832,6 +3832,15 @@ if (ENABLE_TEST_SUPPORT)
     QT_ADD_EXTRA_WEBKIT_TARGET_EXPORT(WebCoreTestSupport)
 endif ()
 
+if (COMPILER_IS_GCC_OR_CLANG)
+    ADD_TARGET_PROPERTIES(WebCore COMPILE_FLAGS "-Wno-deprecated -Wno-sign-compare -Wno-unknown-pragmas -Wno-unused -Wno-unused-parameter -Wno-unused-variable")
+    if (COMPILER_IS_CLANG)
+        ADD_TARGET_PROPERTIES(WebCore COMPILE_FLAGS "-Wno-cast-align -Wno-reorder-ctor -Wno-switch -Wno-deprecated-copy-with-user-provided-copy")
+    else ()
+        ADD_TARGET_PROPERTIES(WebCore COMPILE_FLAGS "-Wno-cast-function-type -Wno-implicit-fallthrough -Wno-stringop-overflow")
+    endif ()
+endif ()
+
 # [ARM] Build SVGPathElement.cpp with -O2 due to a GCC bug
 # https://bugs.webkit.org/show_bug.cgi?id=145377
 if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND WTF_CPU_ARM AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.9") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9.3"))
diff --git a/Source/WebKit/CMakeLists.txt b/Source/WebKit/CMakeLists.txt
index c845856..63c26d7 100644
--- a/Source/WebKit/CMakeLists.txt
+++ b/Source/WebKit/CMakeLists.txt
@@ -54,6 +54,15 @@ set(WebKit_PRIVATE_HEADERS_LOCATION Headers/${PROJECT_VERSION}/QtWebKit/private)
 
 WEBKIT_FRAMEWORK(WebKit)
 
+if (COMPILER_IS_GCC_OR_CLANG)
+    ADD_TARGET_PROPERTIES(WebKit COMPILE_FLAGS "-Wno-unused-parameter -Wno-unused-variable")
+    if (COMPILER_IS_CLANG)
+        ADD_TARGET_PROPERTIES(WebKit COMPILE_FLAGS "-Wno-cast-align")
+    else ()
+        ADD_TARGET_PROPERTIES(WebKit COMPILE_FLAGS "-Wno-cast-function-type")
+    endif ()
+endif ()
+
 if (${PORT} STREQUAL "Qt")
     set_target_properties(WebKit PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
     install(TARGETS WebKit EXPORT WebKitTargets
diff --git a/Source/WebKit2/CMakeLists.txt b/Source/WebKit2/CMakeLists.txt
index aa7863c..07c7867 100644
--- a/Source/WebKit2/CMakeLists.txt
+++ b/Source/WebKit2/CMakeLists.txt
@@ -794,7 +794,12 @@ QT_ADD_EXTRA_WEBKIT_TARGET_EXPORT(WebKit2)
 
 # Suppress unused parameter warnings for sources in WebKit2.
 if (COMPILER_IS_GCC_OR_CLANG)
-    ADD_TARGET_PROPERTIES(WebKit2 COMPILE_FLAGS "-Wno-unused-parameter")
+    ADD_TARGET_PROPERTIES(WebKit2 COMPILE_FLAGS "-Wno-sign-compare -Wno-switch -Wno-unused -Wno-unused-parameter")
+    if (COMPILER_IS_CLANG)
+        ADD_TARGET_PROPERTIES(WebKit2 COMPILE_FLAGS "-Wno-cast-align")
+    else ()
+        ADD_TARGET_PROPERTIES(WebKit2 COMPILE_FLAGS "-Wno-cast-function-type")
+    endif ()
 endif ()
 
 if (WebKit2_VERSION_SCRIPT)
diff --git a/Source/WTF/wtf/CMakeLists.txt b/Source/WTF/wtf/CMakeLists.txt
index 297589b..c96deca 100644
--- a/Source/WTF/wtf/CMakeLists.txt
+++ b/Source/WTF/wtf/CMakeLists.txt
@@ -278,6 +278,15 @@ WEBKIT_INCLUDE_CONFIG_FILES_IF_EXISTS()
 WEBKIT_WRAP_SOURCELIST(${WTF_SOURCES})
 WEBKIT_FRAMEWORK(WTF)
 
+if (COMPILER_IS_GCC_OR_CLANG)
+    ADD_TARGET_PROPERTIES(WTF COMPILE_FLAGS "-Wno-sign-compare -Wno-unused -Wno-unused-parameter -Wno-unused-variable")
+    if (COMPILER_IS_CLANG)
+        ADD_TARGET_PROPERTIES(WTF COMPILE_FLAGS "-Wno-unknown-pragmas -Wno-deprecated-copy-with-user-provided-copy")
+    else ()
+        ADD_TARGET_PROPERTIES(WTF COMPILE_FLAGS "-Wno-cast-function-type -Wno-deprecated -Wno-implicit-fallthrough")
+    endif ()
+endif ()
+
 if (MSVC)
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /SUBSYSTEM:WINDOWS")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /SUBSYSTEM:WINDOWS")
