diff --git a/Source/WebCore/css/CSSGrammar.y.in b/Source/WebCore/css/CSSGrammar.y.in
index 0aad456..95bf6f6 100644
--- a/Source/WebCore/css/CSSGrammar.y.in
+++ b/Source/WebCore/css/CSSGrammar.y.in
@@ -21,7 +21,7 @@
  *
  */
 
-%pure-parser
+%define api.pure
 
 %parse-param { CSSParser* parser }
 %lex-param { CSSParser* parser }
diff --git a/Source/WebCore/xml/XPathGrammar.y b/Source/WebCore/xml/XPathGrammar.y
index 356e118..f064782 100644
--- a/Source/WebCore/xml/XPathGrammar.y
+++ b/Source/WebCore/xml/XPathGrammar.y
@@ -47,7 +47,7 @@ using namespace XPath;
 
 %}
 
-%pure-parser
+%define api.pure
 %lex-param { parser }
 %parse-param { Parser& parser }
 
