diff --git a/Source/WebCore/Modules/indexeddb/client/IDBObjectStoreImpl.h b/Source/WebCore/Modules/indexeddb/client/IDBObjectStoreImpl.h
index 575c72a..1cd0e2a 100644
--- a/Source/WebCore/Modules/indexeddb/client/IDBObjectStoreImpl.h
+++ b/Source/WebCore/Modules/indexeddb/client/IDBObjectStoreImpl.h
@@ -45,7 +45,7 @@ namespace IDBClient {
 class IDBRequest;
 class IDBTransaction;
 
-class IDBObjectStore : public WebCore::IDBObjectStore, public ActiveDOMObject {
+class IDBObjectStore final : public WebCore::IDBObjectStore, public ActiveDOMObject {
 public:
     static Ref<IDBObjectStore> create(ScriptExecutionContext*, const IDBObjectStoreInfo&, IDBTransaction&);
 
diff --git a/Source/WebCore/Modules/indexeddb/client/IDBOpenDBRequestImpl.h b/Source/WebCore/Modules/indexeddb/client/IDBOpenDBRequestImpl.h
index f242a79..ecc83b5 100644
--- a/Source/WebCore/Modules/indexeddb/client/IDBOpenDBRequestImpl.h
+++ b/Source/WebCore/Modules/indexeddb/client/IDBOpenDBRequestImpl.h
@@ -40,7 +40,7 @@ namespace IDBClient {
 
 class IDBConnectionToServer;
 
-class IDBOpenDBRequest : public IDBRequest {
+class IDBOpenDBRequest final : public IDBRequest {
 public:
     static Ref<IDBOpenDBRequest> createDeleteRequest(IDBConnectionToServer&, ScriptExecutionContext*, const IDBDatabaseIdentifier&);
     static Ref<IDBOpenDBRequest> createOpenRequest(IDBConnectionToServer&, ScriptExecutionContext*, const IDBDatabaseIdentifier&, uint64_t version);
diff --git a/Source/WebCore/Modules/indexeddb/client/IDBTransactionImpl.h b/Source/WebCore/Modules/indexeddb/client/IDBTransactionImpl.h
index 4438c84..06f9f95 100644
--- a/Source/WebCore/Modules/indexeddb/client/IDBTransactionImpl.h
+++ b/Source/WebCore/Modules/indexeddb/client/IDBTransactionImpl.h
@@ -57,7 +57,7 @@ class IDBIndex;
 class IDBOpenDBRequest;
 class TransactionOperation;
 
-class IDBTransaction : public WebCore::IDBTransaction {
+class IDBTransaction final : public WebCore::IDBTransaction {
 public:
     static Ref<IDBTransaction> create(IDBDatabase&, const IDBTransactionInfo&);
     static Ref<IDBTransaction> create(IDBDatabase&, const IDBTransactionInfo&, IDBOpenDBRequest&);
diff --git a/Source/WebCore/Modules/indexeddb/server/MemoryIDBBackingStore.h b/Source/WebCore/Modules/indexeddb/server/MemoryIDBBackingStore.h
index d487d3e..4b33613 100644
--- a/Source/WebCore/Modules/indexeddb/server/MemoryIDBBackingStore.h
+++ b/Source/WebCore/Modules/indexeddb/server/MemoryIDBBackingStore.h
@@ -39,7 +39,7 @@ namespace IDBServer {
 
 class MemoryObjectStore;
 
-class MemoryIDBBackingStore : public IDBBackingStore {
+class MemoryIDBBackingStore final : public IDBBackingStore {
 public:
     static std::unique_ptr<MemoryIDBBackingStore> create(const IDBDatabaseIdentifier&);
     
diff --git a/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.h b/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.h
index de66871..4d9e959 100644
--- a/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.h
+++ b/Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.h
@@ -45,7 +45,7 @@ namespace IDBServer {
 
 class SQLiteIDBCursor;
 
-class SQLiteIDBBackingStore : public IDBBackingStore {
+class SQLiteIDBBackingStore final : public IDBBackingStore {
 public:
     SQLiteIDBBackingStore(const IDBDatabaseIdentifier&, const String& databaseRootDirectory);
     
