--- fmt-5.0.0/CMakeLists.txt.orig	2018-06-06 08:43:15.164239900 +0300
+++ fmt-5.0.0/CMakeLists.txt	2018-06-06 08:47:37.265606900 +0300
@@ -186,7 +186,9 @@
 
   # Install the library and headers.
   install(TARGETS ${INSTALL_TARGETS} EXPORT ${targets_export_name}
-          DESTINATION ${FMT_LIB_DIR})
+          RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+          LIBRARY DESTINATION ${FMT_LIB_DIR}
+          ARCHIVE DESTINATION ${FMT_LIB_DIR})
   install(FILES ${FMT_HEADERS} DESTINATION ${FMT_INC_DIR})
 endif ()
 
