--- a/src/meson.build	2021-09-22 14:46:02.475944600 -0700
+++ b/src/meson.build	2021-09-22 16:47:13.993374500 -0700
@@ -137,15 +137,19 @@
   # --nxcompat is to enable NX protection
   # -pie as --dynamicbase requires relocations
   gui_security_link_args += [
-    '-Wl,--dynamicbase,-pie,--nxcompat',
+    '-Wl,--dynamicbase,--nxcompat',
     '-Wl,-e,@0@WinMainCRTStartup'.format(entry_prefix),
     '-mwindows',
   ]
   cui_security_link_args += [
-    '-Wl,--dynamicbase,-pie,--nxcompat',
+    '-Wl,--dynamicbase,--nxcompat',
     '-Wl,-e,@0@WinMainCRTStartup'.format(entry_prefix),
     '-mconsole',
   ]
+  if cc.has_link_argument('-Wl,-pie')
+    gui_security_link_args += ['-Wl,-pie']
+    cui_security_link_args += ['-Wl,-pie']
+  endif
 endif
 
 if libvirt_dep.found()
