--- sqlcipher-3.4.1/configure.ac.orig	2017-03-21 13:43:37.750719800 +0300
+++ sqlcipher-3.4.1/configure.ac	2017-03-21 13:43:56.258099500 +0300
@@ -120,7 +120,7 @@
 # if not, then we fall back to plain tclsh.
 # TODO: try other versions before falling back?
 # 
-AC_CHECK_PROGS(TCLSH_CMD, [tclsh8.6 tclsh8.5 tclsh], none)
+AC_CHECK_PROGS(TCLSH_CMD, [tclsh8.6 tclsh86 tclsh8.5 tclsh85 tclsh], none)
 if test "$TCLSH_CMD" = "none"; then
   # If we can't find a local tclsh, then building the amalgamation will fail.
   # We act as though --disable-amalgamation has been used.
