--- OpenColorIO-2.2.0/src/bindings/python/CMakeLists.txt.orig	2022-10-31 23:27:35.000000000 +0100
+++ OpenColorIO-2.2.0/src/bindings/python/CMakeLists.txt	2022-11-10 14:15:28.555728000 +0100
@@ -227,7 +227,7 @@
 		PY_VERSION_PATCH=${Python_VERSION_PATCH}
 )
 
-if(WIN32)
+if(WIN32 AND NOT MINGW)
     set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/site-packages")
 else()
     set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages")
