--- a/share/cmake/modules/Findyaml-cpp.cmake
+++ b/share/cmake/modules/Findyaml-cpp.cmake
@@ -50,7 +50,7 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
     endif()
 
     if(yaml-cpp_FOUND)
-        get_target_property(yaml-cpp_LIBRARY yaml-cpp LOCATION)
+        get_target_property(yaml-cpp_LIBRARY yaml-cpp::yaml-cpp LOCATION)
     else()
 
         # As yaml-cpp-config.cmake search fails, search an installed library
@@ -137,7 +137,7 @@ endif()
 ###############################################################################
 ### Create target
 
-if(yaml-cpp_FOUND AND NOT TARGET yaml-cpp)
+if(yaml-cpp_FOUND AND NOT TARGET yaml-cpp::yaml-cpp)
     add_library(yaml-cpp UNKNOWN IMPORTED GLOBAL)
     set(_yaml-cpp_TARGET_CREATE TRUE)
 endif()
