--- OpenColorIO-2.4.2/src/OpenColorIO/SSE2.h.orig	2025-03-19 04:20:12.000000000 +0100
+++ OpenColorIO-2.4.2/src/OpenColorIO/SSE2.h	2025-03-20 23:16:49.067427600 +0100
@@ -351,10 +351,10 @@
         float16x4_t rgba04_05 = vcvt_f16_f32(vreinterpretq_f32_m128(rgba2));
         float16x4_t rgba06_07 = vcvt_f16_f32(vreinterpretq_f32_m128(rgba3));
         rgba = vcombine_f16(rgba00_01, rgba03_03);
-        vst1q_f16((float16_t *)(out+0), rgba);
+        vst1q_f16((__fp16 *)(out+0), rgba);
 
         rgba = vcombine_f16(rgba04_05, rgba06_07);
-        vst1q_f16((float16_t *)(out+8), rgba);
+        vst1q_f16((__fp16 *)(out+8), rgba);
 #else
         __m128i rgba;
         __m128i rgba00_01 = sse2_cvtps_ph(rgba0);
