Index: tools/sqlcompattest.c
==================================================================
--- tools/sqlcompattest.c
+++ tools/sqlcompattest.c
@@ -62,11 +62,10 @@
   version=(major*1000000)+(minor*1000)+release;
 
   int i;
   static const char *zRequiredOpts[] = {
     "ENABLE_FTS4",        /* Required for repository search */
-    "ENABLE_JSON1",       /* Required for the check-in locking protocol */
     "ENABLE_DBSTAT_VTAB", /* Required by /repo-tabsize page */
   };
 
   /* Check minimum SQLite version number */
   if( sqlite3_libversion_number()<version ){

