--- fossil-src-2.27.orig/auto.def	2025-09-30 14:03:45.000000000 +0300
+++ fossil-src-2.27/auto.def	2026-01-11 01:10:59.061805700 +0300
@@ -26,6 +26,7 @@
                          => {print the minimum SQLite version number required, and exit}
     internal-sqlite=1    => {Don't use the internal SQLite, use the system one}
     static=0             => {Link a static executable}
+    lineedit=1           => {Disable line editing}
     fusefs=1             => {Disable the Fuse Filesystem}
     fossil-debug=0       => {Build with fossil debugging enabled}
     no-opt=0             => {Build without optimization}
@@ -441,6 +442,20 @@
 }; # handle-zlib
 handle-zlib
 
+if {[opt-bool lineedit]} {
+    # Need readline-compatible line editing
+    cc-with {-includes stdio.h} {
+        if {[cc-check-includes readline/readline.h] && [cc-check-function-in-lib readline readline]} {
+            define-append EXTRA_CFLAGS -DHAVE_READLINE
+            msg-result "Using readline for line editing"
+        } elseif {[cc-check-includes editline/readline.h] && [cc-check-function-in-lib readline edit]} {
+            define-feature editline
+            define-append EXTRA_CFLAGS -DHAVE_EDITLINE
+            msg-result "Using editline for line editing"
+        }
+    }
+}
+
 #
 # Handle the --with-openssl flag and, incidentally, update @LIBS@ for
 # zlib if openssl is _not_ used (if it is, we get zlib via libssl).
