--- fossil-2.8/auto.def.orig	2019-05-17 09:06:44.429683400 +0300
+++ fossil-2.8/auto.def	2019-05-17 09:06:53.850299800 +0300
@@ -52,7 +52,7 @@
 if {[string equal false [get-define TCLSH]]} {
     msg-result "WARNING: 'make test' will not run here."
 } else {
-    set v [exec /bin/sh -c "echo 'puts \$tcl_version' | tclsh"]
+    set v [exec sh -c "echo 'puts \$tcl_version' | tclsh"]
     if {[expr $v >= 8.6]} {
         msg-result "Found Tclsh version $v in the PATH."
         define TCLSH tclsh
