--- a/gui/translationhandler.cpp	2019-09-01 15:01:12.000000000 +0200
+++ b/gui/translationhandler.cpp	2019-09-20 23:24:54.233058500 +0200
@@ -114,7 +114,7 @@
         QSettings settings;
         QString datadir = settings.value("DATADIR").toString();
         if (datadir.isEmpty())
-            datadir = appPath;
+            datadir = appPath + "/../share/cppcheck/lang/";
 
         QString translationFile;
         if (QFile::exists(datadir + "/lang/" + mTranslations[index].mFilename + ".qm"))
