--- a/lib/library.cpp	2016-08-07 01:49:40.000000000 +0900
+++ b/lib/library.cpp	2016-09-30 04:16:08.760802800 +0900
@@ -80,7 +80,7 @@
 #endif
         if (exename) {
             const std::string exepath(Path::fromNativeSeparators(Path::getPathFromFilename(exename)));
-            cfgfolders.push_back(exepath + "cfg");
+            cfgfolders.push_back(exepath + "../share/cppcheck/cfg");
             cfgfolders.push_back(exepath);
         }
 
