--- scons-4.1.0/SCons/Tool/mingw.py.orig	2021-01-19 21:10:56.000000000 +0100
+++ scons-4.1.0/SCons/Tool/mingw.py	2021-01-30 11:06:14.868300400 +0100
@@ -198,6 +198,11 @@
     env['_SHLIBSUFFIX'] = '$SHLIBSUFFIX'
     env["SHLIBPREFIX"] = ""
 
+    # Scons tries to find various tools in hardcoded paths and then appends those
+    # to PATH, not handling the case where those might be shadowed leading to the
+    # wrong toolchain being used. In MSYS2 our PATH is good as is.
+    env["ENV"]["PATH"] = os.environ["PATH"]
+
 
 
 def exists(env):
