--- soundtouch/source/SoundTouch/Makefile.am.orig	2012-09-16 18:22:55 +0400
+++ soundtouch/source/SoundTouch/Makefile.am	2012-09-16 18:23:19 +0400
@@ -66,7 +66,7 @@
 endif
 
 # Modify the default 0.0.0 to LIB_SONAME.0.0
-libSoundTouch_la_LDFLAGS=-version-info @LIB_SONAME@
+libSoundTouch_la_LDFLAGS=-no-undefined -version-info @LIB_SONAME@
 
 # other linking flags to add
 # noinst_LTLIBRARIES = libSoundTouchOpt.la
