--- a/src/build.rs
+++ b/src/build.rs
@@ -232,7 +232,7 @@
             dlltool_command.arg("-D").arg(format!("{}.dll", name));
             dlltool_command
                 .arg("-l")
-                .arg(targetdir.join(format!("{}.dll.a", name)));
+                .arg(targetdir.join(format!("lib{}.dll.a", name)));
             dlltool_command
                 .arg("-d")
                 .arg(targetdir.join(format!("{}.def", name)));
--- a/src/build_targets.rs
+++ b/src/build_targets.rs
@@ -103,7 +103,7 @@
                 let impl_lib = if env == "msvc" {
                     targetdir.join(format!("{}.dll.lib", lib_name))
                 } else {
-                    targetdir.join(format!("{}.dll.a", lib_name))
+                    targetdir.join(format!("lib{}.dll.a", lib_name))
                 };
                 let def = targetdir.join(format!("{}.def", lib_name));
                 (shared_lib, static_lib, Some(impl_lib), Some(def))
