--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,11 +76,6 @@
     set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${_FLAGS}")
 endmacro ()
 
-# We require at least GCC 4.9 for decent C++11 support
-if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR
-    CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
-    message (FATAL_ERROR "GCC 4.9 or newer required")
-endif ()
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
 
@@ -116,7 +111,6 @@
 if (CMAKE_SIZEOF_VOID_P EQUAL 8)
     add_definitions (-DHAVE_WIN64=1)
 else ()
-    add_linker_flags (-Wl,--enable-stdcall-fixup)
     add_definitions (-DHAVE_WIN64=0)
 endif ()
 
--- a/src/exchndl/CMakeLists.txt
+++ b/src/exchndl/CMakeLists.txt
@@ -28,7 +28,7 @@
 
 add_custom_command (
     OUTPUT ${EXCHNDL_IMPLIB}
-    COMMAND ${DLLTOOL} --output-lib ${EXCHNDL_IMPLIB} --kill-at --input-def=${CMAKE_CURRENT_SOURCE_DIR}/${EXCHNDL_IMP_DEF}
+    COMMAND ${DLLTOOL} --output-lib ${EXCHNDL_IMPLIB} --kill-at --input-def ${CMAKE_CURRENT_SOURCE_DIR}/${EXCHNDL_IMP_DEF}
     DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${EXCHNDL_IMP_DEF}
 )
 add_custom_target (exchndl_implib ALL DEPENDS exchndl ${EXCHNDL_IMPLIB})
--- a/src/mgwhelp/CMakeLists.txt
+++ b/src/mgwhelp/CMakeLists.txt
@@ -29,7 +29,7 @@
 
 add_custom_command (
     OUTPUT ${MGWHELP_IMPLIB}
-    COMMAND ${DLLTOOL} --output-lib ${MGWHELP_IMPLIB} --dllname mgwhelp.dll --kill-at --input-def=${CMAKE_CURRENT_SOURCE_DIR}/${MGWHELP_IMP_DEF}
+    COMMAND ${DLLTOOL} --output-lib ${MGWHELP_IMPLIB} --dllname mgwhelp.dll --kill-at --input-def ${CMAKE_CURRENT_SOURCE_DIR}/${MGWHELP_IMP_DEF}
     DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${MGWHELP_IMP_DEF}
 )
 add_custom_target (mgwhelp_implib DEPENDS mgwhelp ${MGWHELP_IMPLIB})
