From 5f3064fd4176cce596f5673cf8347ce27a86c68f Mon Sep 17 00:00:00 2001
From: Wan-Teh Chang <wtc@google.com>
Date: Mon, 12 Jan 2026 16:12:42 -0800
Subject: [PATCH 1/2] Set SVT-AV1's avif option for v4.0.0 or later

---
 src/codec_svt.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/codec_svt.c b/src/codec_svt.c
index 3c775133..6f366ed1 100644
--- a/src/codec_svt.c
+++ b/src/codec_svt.c
@@ -233,8 +233,12 @@ static avifResult svtCodecEncodeImage(av
 
 #if SVT_AV1_CHECK_VERSION(3, 0, 0)
         svt_config->lossless = quantizer == AVIF_QUANTIZER_LOSSLESS;
-        // TODO: https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/2245 - Enable when resolved.
-        // svt_config->avif = (addImageFlags & AVIF_ADD_IMAGE_FLAG_SINGLE) != 0;
+#endif
+
+#if SVT_AV1_CHECK_VERSION(4, 0, 0)
+        // Although the `avif` option was added in v3.0.0, it had a serious bug that was not fixed
+        // until v4.0.0. See https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/2245.
+        svt_config->avif = (addImageFlags & AVIF_ADD_IMAGE_FLAG_SINGLE) != 0;
 #endif
 
         res = svt_av1_enc_set_parameter(codec->internal->svt_encoder, svt_config);
-- 
2.52.0

