From 6399121d92e8ce82c1a9c705f5dc0e4667f801e1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert-Andr=C3=A9=20Mauchin?= <eclipseo@mauchin.fr>
Date: Sat, 24 Jan 2026 17:04:24 +0100
Subject: [PATCH 2/2] Conditional support for aq_mode in version >= 4.0.0

---
 src/codec_svt.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/codec_svt.c b/src/codec_svt.c
index 6f366ed1..3d0a18fc 100644
--- a/src/codec_svt.c
+++ b/src/codec_svt.c
@@ -162,7 +162,11 @@ static avifResult svtCodecEncodeImage(av
 #else
         svt_config->logical_processors = encoder->maxThreads;
 #endif
+#if SVT_AV1_CHECK_VERSION(4, 0, 0)
+        svt_config->aq_mode = 2;
+#else
         svt_config->enable_adaptive_quantization = 2;
+#endif
         // disable 2-pass
 #if SVT_AV1_CHECK_VERSION(0, 9, 0)
         svt_config->rc_stats_buffer = (SvtAv1FixedBuf) { NULL, 0 };
@@ -171,7 +175,7 @@ static avifResult svtCodecEncodeImage(av
         svt_config->rc_twopass_stats_in = (SvtAv1FixedBuf) { NULL, 0 };
 #endif
 
-        svt_config->rate_control_mode = 0; // CRF because enable_adaptive_quantization is 2
+        svt_config->rate_control_mode = 0; // CRF because aq_mode is 2
         if (alpha) {
             svt_config->min_qp_allowed = AVIF_CLAMP(encoder->minQuantizerAlpha, 0, 63);
             svt_config->max_qp_allowed = AVIF_CLAMP(encoder->maxQuantizerAlpha, 0, 63);
-- 
2.52.0

