From 117246d5faa5bc3c4c217bf548c2727e07e9ff9b Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Sat, 29 Aug 2020 19:17:45 +0530
Subject: [PATCH] gdk-pixbuf/loader.c: Include stdlib.h for strtol

---
 contrib/gdk-pixbuf/loader.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/gdk-pixbuf/loader.c b/contrib/gdk-pixbuf/loader.c
index af0e775..6531ec0 100644
--- a/contrib/gdk-pixbuf/loader.c
+++ b/contrib/gdk-pixbuf/loader.c
@@ -3,6 +3,7 @@
 */
 
 #include <avif/avif.h>
+#include <stdlib.h>
 
 #define GDK_PIXBUF_ENABLE_BACKEND
 #include <gdk-pixbuf/gdk-pixbuf-io.h>
-- 
2.27.0

