--- src/pkcs11-provider-1.2.0/meson.build.orig	2026-02-25 23:20:14.543046800 -0600
+++ src/pkcs11-provider-1.2.0/meson.build	2026-02-25 23:33:35.179765600 -0600
@@ -49,7 +49,10 @@
 if not libcrypto.found()
   libcrypto = dependency('libcrypto3', version: '>= 3.0.7', required: true)
 endif
-provider_path = libcrypto.get_variable(pkgconfig: 'modulesdir')
+provider_path = get_option('modulesdir')
+if provider_path == '' and libcrypto.found()
+  provider_path = libcrypto.get_variable(pkgconfig: 'modulesdir')
+endif
 libssl = dependency('libssl', version: '>= 3.0.7', required: false)
 if not libssl.found()
   libssl = dependency('libssl3', version: '>= 3.0.7', required: true, method: 'pkg-config')
--- src/pkcs11-provider-1.2.0/meson_options.txt.orig	2026-02-19 07:48:59.000000000 -0600
+++ src/pkcs11-provider-1.2.0/meson_options.txt	2026-02-25 23:40:02.918433300 -0600
@@ -17,3 +17,8 @@
     type : 'string',
     value : '',
     description : 'Optional string with build information, like a build id')
+
+option('modulesdir',
+    type : 'string',
+    value : '',
+    description : 'Path to the OpenSSL providers')
