diff --git a/src/exact_cutset.hpp b/src/exact_cutset.hpp
index 782bb96..4929730 100755
--- a/src/exact_cutset.hpp
+++ b/src/exact_cutset.hpp
@@ -994,7 +994,7 @@ bool exact_cutset<G_t, config>::try_it(T_t &T, unsigned bagsize)
     }else{
      //   incomplete(); //no//
      // messes up random tests, send to cerr instead
-        std::cerr << "incomplete ../../src/exact_cutset.hpp:978:try_it\n";
+//         std::cerr << "incomplete ../../src/exact_cutset.hpp:978:try_it\n";
     }
 
     typename boost::graph_traits<G_t>::vertex_iterator vIt, vEnd;
