--- src/cmake/ImportBSON.cmake.orig	2026-02-03 10:55:47.000000000 -0600
+++ src/cmake/ImportBSON.cmake	2026-03-12 20:39:14.606805600 -0500
@@ -84,7 +84,7 @@
          IMPORTED_CONFIGURATIONS "Release"
          INTERFACE_INCLUDE_DIRECTORIES "${_MONGOCRYPT_SYSTEM_LIBBSON_INCLUDE_DIR}"
          )
-      set_property (TARGET bson_shared PROPERTY IMPORTED_LOCATION "${_MONGOCRYPT_SYSTEM_LIBBSON_SHARED}")
+      set_property (TARGET bson_shared PROPERTY IMPORTED_IMPLIB "${_MONGOCRYPT_SYSTEM_LIBBSON_SHARED}")
       set_property (TARGET bson_static PROPERTY IMPORTED_LOCATION "${_MONGOCRYPT_SYSTEM_LIBBSON_STATIC}")
       set_property (
          CACHE _MONGOCRYPT_SYSTEM_LIBBSON_SHARED
@@ -170,7 +170,7 @@
    target_link_libraries (_mongocrypt-libbson_for_shared INTERFACE $<BUILD_INTERFACE:bson_static>)
 endif ()
 # libbson_for_static always links to the static libbson:
-target_link_libraries (_mongocrypt-libbson_for_static INTERFACE $<BUILD_INTERFACE:bson_static>)
+target_link_libraries (_mongocrypt-libbson_for_static INTERFACE $<BUILD_INTERFACE:bson_static> ws2_32)

 if (TARGET mongoc_static)
    # And an alias to the mongoc target for use in some test cases
