--- src/src/mc-mlib/str.test.c.orig	2026-02-03 10:55:47.000000000 -0600
+++ src/src/mc-mlib/str.test.c	2026-03-12 23:08:05.132850900 -0500
@@ -112,7 +112,7 @@
     test_predicate(false, "foo", ends_with, "oof");
     test_predicate(false, "foo", ends_with, "bar");
 
-#ifdef _WIN32
+#ifdef _MSC_VER
     const wchar_t *wide = L"🕴️";
     mstr_narrow_result narrow = mstr_win32_narrow(wide);
     CHECK(narrow.error == 0);
