--- src/bindings/python/pymongocrypt/binding.py.orig	2026-02-03 10:55:47.000000000 -0600
+++ src/bindings/python/pymongocrypt/binding.py	2026-03-17 10:13:00.868882600 -0500
@@ -1529,9 +1529,9 @@
 # export PYMONGOCRYPT_LIB='/path/to/libmongocrypt.so'
 # If the PYMONGOCRYPT_LIB is not set then load the embedded library and
 # fallback to the relying on a system installed library.
-_base = Path(os.path.realpath(__file__)).parent
+_base = Path(os.path.realpath(__file__)).parents[4]
 if sys.platform == "win32":
-    _path = Path(_base) / "mongocrypt.dll"
+    _path = Path(_base) / "bin" / "libmongocrypt.dll"
 elif sys.platform == "darwin":
     _path = Path(_base) / "libmongocrypt.dylib"
 else:
