--- src/build-UCRT64/_deps/intel_dfp-src/LIBRARY/src/bid_functions.h.orig	2026-03-12 22:00:27.719387300 -0500
+++ src/build-UCRT64/_deps/intel_dfp-src/LIBRARY/src/bid_functions.h	2026-03-12 22:09:50.695899300 -0500
@@ -153,7 +153,7 @@
 typedef unsigned bid__int64 fexcept_t;
 #endif
 #else
-#ifdef __QNX__
+#if defined(__QNX__) || defined(__MINGW32__)
 #include <fenv.h>
 #else
 typedef unsigned short int fexcept_t;
