diff -Naur xapian-core-1.4.3.org/api/omenquire.cc xapian-core-1.4.3/api/omenquire.cc
--- xapian-core-1.4.3.org/api/omenquire.cc	2017-01-25 01:41:55.000000000 +0100
+++ xapian-core-1.4.3/api/omenquire.cc	2017-03-17 09:37:35.882225800 +0100
@@ -256,7 +256,7 @@
 	return e;
     }
 
-    Xapian::doccount r = Xapian::doccount(exp10(int(log10(D))) + 0.5);
+    Xapian::doccount r = Xapian::doccount(pow(10.0, (int(log10(D))) + 0.5));
     while (r > e) r /= 10;
 
     Xapian::doccount R = e / r * r;