--- a/cmake/tomlplusplusConfig.cmake.meson.in
+++ b/cmake/tomlplusplusConfig.cmake.meson.in
@@ -20,7 +20,7 @@
   # Set the path to the installed library so that users can link to it
   if (@compile_library@)
     set_target_properties(tomlplusplus::tomlplusplus PROPERTIES
-      IMPORTED_LOCATION "${PACKAGE_PREFIX_DIR}/@libdir@/@lib_name@"
+      IMPORTED_LOCATION "${PACKAGE_PREFIX_DIR}/@bindir@/@lib_name@"
     )
     # compile_options not quoted on purpose
     target_compile_options(tomlplusplus::tomlplusplus INTERFACE @compile_options@)
--- a/src/meson.build
+++ b/src/meson.build
@@ -81,6 +81,7 @@
 			'compile_library': true,
 			'compile_options': cmake_compile_options,
 			'includedir': get_option('includedir'),
+			'bindir': get_option('bindir'),
 			'libdir': get_option('libdir'),
 			'lib_name': lib_name
 		})
