From de08a7d3fbc47ef0caa3bd60c0dbbee4af1a06ca Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 17 Jun 2021 18:52:21 +0530
Subject: [PATCH 066/N] sqlite3 module defines

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index f67a8a2..1bb51a5 100644
--- a/setup.py
+++ b/setup.py
@@ -1642,6 +1642,7 @@ class PyBuildExt(build_ext):
                 '_sqlite/statement.c',
                 '_sqlite/util.c', ]
             sqlite_defines = []
+            sqlite_defines.append(('MODULE_NAME', '"sqlite3"'))
 
             # Enable support for loadable extensions in the sqlite3 module
             # if --enable-loadable-sqlite-extensions configure option is used.
-- 
2.34.1

