From 8801fd78e593cc54d666ef9edd307361ac9b5332 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 17 Jun 2021 18:52:31 +0530
Subject: [PATCH 073/N] pathlib path sep

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 Lib/pathlib.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/pathlib.py b/Lib/pathlib.py
index 621fba0..c930e1d 100644
--- a/Lib/pathlib.py
+++ b/Lib/pathlib.py
@@ -115,6 +115,8 @@ class _WindowsFlavour(_Flavour):
 
     sep = '\\'
     altsep = '/'
+    if 'MSYSTEM' in os.environ:
+        sep, altsep = altsep, sep
     has_drv = True
     pathmod = ntpath
 
-- 
2.34.1

