From 111574d55172fda8eed70a8390d2132f830a3a24 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 17 Jun 2021 18:52:34 +0530
Subject: [PATCH 075/N] fix build testinternalcapi

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 setup.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/setup.py b/setup.py
index 1bb51a5..6649f53 100644
--- a/setup.py
+++ b/setup.py
@@ -1063,7 +1063,11 @@ class PyBuildExt(build_ext):
                            depends=['testcapi_long.h']))
 
         # Python Internal C API test module
+        macros = []
+        if MS_WINDOWS:
+            macros.append(('PY3_DLLNAME', 'L"%s"' % sysconfig.get_config_var('DLLLIBRARY')))
         self.add(Extension('_testinternalcapi', ['_testinternalcapi.c'],
+                           define_macros=macros,
                            extra_compile_args=['-DPy_BUILD_CORE_MODULE']))
 
         # Python PEP-3118 (buffer protocol) test module
-- 
2.34.1

