From 7bbbdb552ef068fa5b4c4e38a4b72a32d679b754 Mon Sep 17 00:00:00 2001
From: Long Nguyen <nguyen.long.908132@gmail.com>
Date: Thu, 17 Jun 2021 18:52:34 +0530
Subject: [PATCH 076/N] extend MS_WINDOWS flag

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 6649f53..5eb9b07 100644
--- a/setup.py
+++ b/setup.py
@@ -92,7 +92,7 @@ if sys.platform == "win32" and "MSYSTEM" in os.environ:
 
 CROSS_COMPILING = ("_PYTHON_HOST_PLATFORM" in os.environ)
 HOST_PLATFORM = get_platform()
-MS_WINDOWS = (HOST_PLATFORM == 'win32')
+MS_WINDOWS = (HOST_PLATFORM == 'win32' or HOST_PLATFORM == 'mingw')
 CYGWIN = (HOST_PLATFORM == 'cygwin')
 MACOS = (HOST_PLATFORM == 'darwin')
 AIX = (HOST_PLATFORM.startswith('aix'))
-- 
2.34.1

