From f471e4d6b28f351e5ccaf760969bd239088d8fbb Mon Sep 17 00:00:00 2001
From: jeremyd2019 <4524874+jeremyd2019@users.noreply.github.com>
Date: Thu, 17 Jun 2021 18:52:37 +0530
Subject: [PATCH 078/N] configure.ac: set MINGW stack reserve

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index ecf88ff..10334e1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3053,6 +3053,11 @@ then
 	VxWorks*)
 		LINKFORSHARED='-Wl,-export-dynamic';;
 	esac
+	case $host in
+	*-*-mingw*)
+		# for https://bugs.python.org/issue40458 on MINGW
+		LINKFORSHARED="-Wl,--stack,2000000";;
+	esac
 fi
 AC_MSG_RESULT($LINKFORSHARED)
 
-- 
2.34.1

