From 2b83ecfaa7c76175966bdf0fdb2b1f6b3f9a8169 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 1 Aug 2021 15:18:50 +0200
Subject: [PATCH 108/N] configure.ac: set _WIN32_WINNT version

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 164147c..dd21dfd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3470,6 +3470,11 @@ else
 fi])
 AC_MSG_RESULT($with_dbmliborder)
 
+case $host in
+  *-*-mingw*)
+    CFLAGS_NODIST="$CFLAGS_NODIST -D_WIN32_WINNT=0x0601";;
+esac
+
 # Determine if windows modules should be used.
 AC_SUBST(USE_WIN32_MODULE)
 USE_WIN32_MODULE='#'
-- 
2.34.1

