From c6765ac66cdd2a518beb24fee89bf311c0f44ce5 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 23 Sep 2021 09:57:56 +0200
Subject: [PATCH 113/N] smoketests: test that _decimal exists

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 mingw_smoketests.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mingw_smoketests.py b/mingw_smoketests.py
index 70acbd1..527aafb 100644
--- a/mingw_smoketests.py
+++ b/mingw_smoketests.py
@@ -139,6 +139,9 @@ class Tests(unittest.TestCase):
         import ctypes
         import curses
 
+    def test_c_modules_import(self):
+        import _decimal
+
     def test_socket_inet_ntop(self):
         import socket
         self.assertTrue(hasattr(socket, "inet_ntop"))
-- 
2.34.1

