From b2ff1b79447c6b6cc7008dc5ee013e272ea791b8 Mon Sep 17 00:00:00 2001
From: Long Nguyen <nguyen.long.908132@gmail.com>
Date: Sun, 14 Nov 2021 14:55:35 +0700
Subject: [PATCH 119/N] _testconsole.c: Fix casing & path sep

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 PC/_testconsole.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/PC/_testconsole.c b/PC/_testconsole.c
index db84f73..eecf2f9 100644
--- a/PC/_testconsole.c
+++ b/PC/_testconsole.c
@@ -6,7 +6,7 @@
 
 #ifdef MS_WINDOWS
 
-#include "..\modules\_io\_iomodule.h"
+#include "../Modules/_io/_iomodule.h"
 
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
@@ -104,7 +104,7 @@ _testconsole_read_output_impl(PyObject *module, PyObject *file)
     Py_RETURN_NONE;
 }
 
-#include "clinic\_testconsole.c.h"
+#include "clinic/_testconsole.c.h"
 
 PyMethodDef testconsole_methods[] = {
     _TESTCONSOLE_WRITE_INPUT_METHODDEF
-- 
2.34.1

