From 474084e1fa77bc9211249ccf0387a4dddd6a0f5d Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Wed, 6 Dec 2017 08:29:08 +0300
Subject: [PATCH] Build lua as static library

---
 Lua.pro | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Lua.pro b/Lua.pro
index 2565a2e..c76ef86 100644
--- a/Lua.pro
+++ b/Lua.pro
@@ -2,7 +2,8 @@ TARGET = Lua
 include($$PWD/../../commonplatform.pri)
 CONFIG += console warn_off installed
 win32 {
-DEFINES += LUA_ANSI LUA_BUILD_AS_DLL LUA_DL_DLL
+CONFIG += staticlib
+DEFINES += LUA_ANSI
 }
 linux {
 CONFIG += staticlib
-- 
2.15.0

