From 45f900bd1974431bbae64538e0994290d051a5a2 Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Sat, 23 Apr 2011 14:57:49 -0500
Subject: Fix the SRFI 60 copy-bit documentation.

---
 srfi/srfi-60.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/srfi/srfi-60.c b/srfi/srfi-60.c
index a91cb50..2b7c27c 100644
--- a/srfi/srfi-60.c
+++ b/srfi/srfi-60.c
@@ -63,7 +63,7 @@ SCM_DEFINE (scm_srfi60_log2_binary_factors, "log2-binary-factors", 1, 0, 0,
 
 
 SCM_DEFINE (scm_srfi60_copy_bit, "copy-bit", 3, 0, 0,
-            (SCM index, SCM n, SCM bit),
+            (SCM index, SCM n, SCM newbit),
 	    "Return @var{n} with the bit at @var{index} set according to\n"
 	    "@var{newbit}.  @var{newbit} should be @code{#t} to set the bit\n"
 	    "to 1, or @code{#f} to set it to 0.  Bits other than at\n"
@@ -79,7 +79,7 @@ SCM_DEFINE (scm_srfi60_copy_bit, "copy-bit", 3, 0, 0,
   int bb;
 
   ii = scm_to_ulong (index);
-  bb = scm_to_bool (bit);
+  bb = scm_to_bool (newbit);
 
   if (SCM_I_INUMP (n))
     {
