From 1509fdf537d60c732660cbf6c1dd8c885797b7f8 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgads@gmail.com>
Date: Thu, 13 Feb 2020 09:25:02 +0200
Subject: [PATCH 5/7] Fix TUI with pdcurses

---
 gdb/tui/tui-win.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/gdb/tui/tui-win.c b/gdb/tui/tui-win.c
index a51e7b9f6da..8967bacca7f 100644
--- a/gdb/tui/tui-win.c
+++ b/gdb/tui/tui-win.c
@@ -118,15 +118,15 @@ struct tui_translate
    The list of values must be terminated by a NULL.
    After the NULL value, an entry defines the default.  */
 static struct tui_translate tui_border_mode_translate[] = {
-  { "normal",		A_NORMAL },
-  { "standout",		A_STANDOUT },
-  { "reverse",		A_REVERSE },
-  { "half",		A_DIM },
-  { "half-standout",	A_DIM | A_STANDOUT },
-  { "bold",		A_BOLD },
-  { "bold-standout",	A_BOLD | A_STANDOUT },
+  { "normal",		(int)A_NORMAL },
+  { "standout",		(int)A_STANDOUT },
+  { "reverse",		(int)A_REVERSE },
+  { "half",		(int)A_DIM },
+  { "half-standout",	(int)(A_DIM | A_STANDOUT) },
+  { "bold",		(int)A_BOLD },
+  { "bold-standout",	(int)(A_BOLD | A_STANDOUT) },
   { 0, 0 },
-  { "normal",		A_NORMAL }
+  { "normal",		(int)A_NORMAL }
 };
 
 /* Translation tables for border-kind, one for each border
-- 
2.33.1.windows.1

