diff -Naur gdb-7.7-orig/bfd/bfd-in.h gdb-7.7/bfd/bfd-in.h
--- gdb-7.7-orig/bfd/bfd-in.h	2014-02-06 06:21:29.000000000 +0400
+++ gdb-7.7/bfd/bfd-in.h	2014-03-14 10:39:32.085200000 +0400
@@ -137,7 +137,7 @@
 
 #if BFD_HOST_64BIT_LONG
 #define BFD_VMA_FMT "l"
-#elif defined (__MSVCRT__)
+#elif defined(__MSVCRT__) && !defined( __USE_MINGW_ANSI_STDIO)
 #define BFD_VMA_FMT "I64"
 #else
 #define BFD_VMA_FMT "ll"
diff -Naur gdb-7.7-orig/bfd/bfd-in2.h gdb-7.7/bfd/bfd-in2.h
--- gdb-7.7-orig/bfd/bfd-in2.h	2014-02-06 06:21:29.000000000 +0400
+++ gdb-7.7/bfd/bfd-in2.h	2014-03-14 10:39:32.085200000 +0400
@@ -144,7 +144,7 @@
 
 #if BFD_HOST_64BIT_LONG
 #define BFD_VMA_FMT "l"
-#elif defined (__MSVCRT__)
+#elif defined (__MSVCRT__) && !defined(__USE_MINGW_ANSI_STDIO)
 #define BFD_VMA_FMT "I64"
 #else
 #define BFD_VMA_FMT "ll"
diff -Naur gdb-7.7-orig/gdb/common/format.h gdb-7.7/gdb/common/format.h
--- gdb-7.7-orig/gdb/common/format.h	2014-01-08 13:23:36.000000000 +0400
+++ gdb-7.7/gdb/common/format.h	2014-03-14 10:39:32.412800000 +0400
@@ -18,7 +18,11 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 
 #if defined(__MINGW32__) && !defined(PRINTF_HAS_LONG_LONG)
-# define USE_PRINTF_I64 1
+# if !defined(__USE_MINGW_ANSI_STDIO)
+#  define USE_PRINTF_I64 1
+# else
+#  define USE_PRINTF_I64 0
+# endif
 # define PRINTF_HAS_LONG_LONG
 #else
 # define USE_PRINTF_I64 0
diff -Naur gdb-7.7-orig/gdb/gnulib/import/inttypes.in.h gdb-7.7/gdb/gnulib/import/inttypes.in.h
--- gdb-7.7-orig/gdb/gnulib/import/inttypes.in.h	2013-12-08 08:11:51.000000000 +0400
+++ gdb-7.7/gdb/gnulib/import/inttypes.in.h	2014-03-14 10:43:47.309400000 +0400
@@ -183,7 +183,7 @@
 #ifdef INT64_MAX
 # if (@APPLE_UNIVERSAL_BUILD@ ? defined _LP64 : @INT64_MAX_EQ_LONG_MAX@)
 #  define _PRI64_PREFIX "l"
-# elif defined _MSC_VER || defined __MINGW32__
+# elif (defined _MSC_VER || defined __MINGW32__) && !defined(__USE_MINGW_ANSI_STDIO)
 #  define _PRI64_PREFIX "I64"
 # elif @HAVE_LONG_LONG_INT@ && LONG_MAX >> 30 == 1
 #  define _PRI64_PREFIX _LONG_LONG_FORMAT_PREFIX
@@ -200,7 +200,7 @@
 #ifdef UINT64_MAX
 # if (@APPLE_UNIVERSAL_BUILD@ ? defined _LP64 : @UINT64_MAX_EQ_ULONG_MAX@)
 #  define _PRIu64_PREFIX "l"
-# elif defined _MSC_VER || defined __MINGW32__
+# elif (defined _MSC_VER || defined __MINGW32__) && !defined(__USE_MINGW_ANSI_STDIO)
 #  define _PRIu64_PREFIX "I64"
 # elif @HAVE_UNSIGNED_LONG_LONG_INT@ && ULONG_MAX >> 31 == 1
 #  define _PRIu64_PREFIX _LONG_LONG_FORMAT_PREFIX
@@ -676,7 +676,7 @@
 #ifdef INT64_MAX
 # if (@APPLE_UNIVERSAL_BUILD@ ? defined _LP64 : @INT64_MAX_EQ_LONG_MAX@)
 #  define _SCN64_PREFIX "l"
-# elif defined _MSC_VER || defined __MINGW32__
+# elif (defined _MSC_VER || defined __MINGW32__) && !defined(__USE_MINGW_ANSI_STDIO)
 #  define _SCN64_PREFIX "I64"
 # elif @HAVE_LONG_LONG_INT@ && LONG_MAX >> 30 == 1
 #  define _SCN64_PREFIX _LONG_LONG_FORMAT_PREFIX
@@ -693,7 +693,7 @@
 #ifdef UINT64_MAX
 # if (@APPLE_UNIVERSAL_BUILD@ ? defined _LP64 : @UINT64_MAX_EQ_ULONG_MAX@)
 #  define _SCNu64_PREFIX "l"
-# elif defined _MSC_VER || defined __MINGW32__
+# elif (defined _MSC_VER || defined __MINGW32__) && !defined(__USE_MINGW_ANSI_STDIO)
 #  define _SCNu64_PREFIX "I64"
 # elif @HAVE_UNSIGNED_LONG_LONG_INT@ && ULONG_MAX >> 31 == 1
 #  define _SCNu64_PREFIX _LONG_LONG_FORMAT_PREFIX
