--- gdb-8.3/gdb/tui/tui-win.c.orig	2019-05-31 11:01:21.886191600 +0300
+++ gdb-8.3/gdb/tui/tui-win.c	2019-05-31 11:01:25.916197200 +0300
@@ -134,15 +134,15 @@
    The list of values must be terminated by a NULL.
    After the NULL value, an entry defines the default.  */
 struct tui_translate tui_border_mode_translate[] = {
-  { "normal",		A_NORMAL },
-  { "standout",		A_STANDOUT },
-  { "reverse",		A_REVERSE },
-  { "half",		A_DIM },
-  { "half-standout",	A_DIM | A_STANDOUT },
-  { "bold",		A_BOLD },
-  { "bold-standout",	A_BOLD | A_STANDOUT },
+  { "normal",		static_cast<int>(A_NORMAL) },
+  { "standout",		static_cast<int>(A_STANDOUT) },
+  { "reverse",		static_cast<int>(A_REVERSE) },
+  { "half",		static_cast<int>(A_DIM) },
+  { "half-standout",	static_cast<int>(A_DIM | A_STANDOUT) },
+  { "bold",		static_cast<int>(A_BOLD) },
+  { "bold-standout",	static_cast<int>(A_BOLD | A_STANDOUT) },
   { 0, 0 },
-  { "normal",		A_NORMAL }
+  { "normal",		static_cast<int>(A_NORMAL) }
 };
 
 /* Translation tables for border-kind, one for each border
