From 12d299345b51c1686c1d82986de6de020970bb57 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Sun, 28 Jun 2015 20:14:51 +0100
Subject: [PATCH] Fix GetHostTriple for mingw-w64 in msys


diff --git a/cmake/modules/GetHostTriple.cmake b/cmake/modules/GetHostTriple.cmake
index 671a8ce..45c47e1 100644
--- a/cmake/modules/GetHostTriple.cmake
+++ b/cmake/modules/GetHostTriple.cmake
@@ -8,11 +8,11 @@ function( get_host_triple var )
     else()
       set( value "i686-pc-win32" )
     endif()
-  elseif( MINGW AND NOT MSYS )
+  elseif( MINGW )
     if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
       set( value "x86_64-w64-mingw32" )
     else()
-      set( value "i686-pc-mingw32" )
+      set( value "i686-w64-mingw32" )
     endif()
   else( MSVC )
     set(config_guess ${LLVM_MAIN_SRC_DIR}/autoconf/config.guess)
-- 
2.4.5

