--- a/cmake/modules/LLVMInstallSymlink.cmake.orig	2022-08-12 22:10:34.881379200 +0200
+++ b/cmake/modules/LLVMInstallSymlink.cmake	2022-08-13 00:04:43.413084900 +0200
@@ -6,20 +6,22 @@
 
 function(install_symlink name target outdir)
   set(DESTDIR $ENV{DESTDIR})
+
+  if(CMAKE_HOST_UNIX)
+    set(LINK_OR_COPY create_symlink)
+  else()
+    set(LINK_OR_COPY copy)
+  endif()
+
   if(NOT IS_ABSOLUTE "${outdir}")
     set(outdir "${CMAKE_INSTALL_PREFIX}/${outdir}")
   endif()
+
   set(outdir "${DESTDIR}${outdir}")
 
   message(STATUS "Creating ${name}")
 
   execute_process(
-    COMMAND "${CMAKE_COMMAND}" -E create_symlink "${target}" "${name}"
-    WORKING_DIRECTORY "${outdir}" ERROR_VARIABLE has_err)
-  if(CMAKE_HOST_WIN32 AND has_err)
-    execute_process(
-      COMMAND "${CMAKE_COMMAND}" -E copy "${target}" "${name}"
-      WORKING_DIRECTORY "${outdir}")
-  endif()
-
+    COMMAND "${CMAKE_COMMAND}" -E ${LINK_OR_COPY} "${target}" "${name}"
+    WORKING_DIRECTORY "${outdir}")
 endfunction()
