--- a/cmake/modules/AddLLVM.cmake	2015-12-25 11:45:24.447038200 +0300
+++ b/cmake/modules/AddLLVM.cmake	2015-12-25 12:14:26.848945700 +0300
@@ -399,10 +399,15 @@
   llvm_update_compile_flags(${name})
   add_link_opts( ${name} )
   if(ARG_OUTPUT_NAME)
-    set_target_properties(${name}
-      PROPERTIES
-      OUTPUT_NAME ${ARG_OUTPUT_NAME}
-      )
+    if(MINGW)
+      string(REGEX REPLACE "^lib([A-Za-z]+)" "\\1" LIB_OUTPUT_NAME ${ARG_OUTPUT_NAME})
+      set_target_properties(${name} PROPERTIES OUTPUT_NAME ${LIB_OUTPUT_NAME} RUNTIME_OUTPUT_NAME ${ARG_OUTPUT_NAME} ARCHIVE_OUTPUT_NAME ${LIB_OUTPUT_NAME})
+    else()
+      set_target_properties(${name}
+        PROPERTIES
+        OUTPUT_NAME ${ARG_OUTPUT_NAME}
+        )
+    endif()
   endif()
 
   if(ARG_MODULE)
