Index: lib/Object/COFFObjectFile.cpp
===================================================================
--- a/lib/Object/COFFObjectFile.cpp	(Revision 274558)
+++ b/lib/Object/COFFObjectFile.cpp	(Arbeitskopie)
@@ -217,6 +217,9 @@
   if (Symb.isExternal() || Symb.isWeakExternal())
     Result |= SymbolRef::SF_Global;
 
+  if (Symb.isExternal())
+    Result |= SymbolRef::SF_Exported;
+
   if (Symb.isWeakExternal()) {
     Result |= SymbolRef::SF_Weak;
 
