--- llvm-6.0.0.src/cmake/modules/AddLLVM.cmake.orig	2018-06-15 07:29:10.041514000 +0300
+++ llvm-6.0.0.src/cmake/modules/AddLLVM.cmake	2018-06-15 07:30:34.101631600 +0300
@@ -419,7 +419,7 @@
     # static
     set(name_static "${name}_static")
     if(ARG_OUTPUT_NAME)
-      set(output_name OUTPUT_NAME "${ARG_OUTPUT_NAME}")
+      set(output_name OUTPUT_NAME "${ARG_OUTPUT_NAME}_static")
     endif()
     # DEPENDS has been appended to LLVM_COMMON_LIBS.
     llvm_add_library(${name_static} STATIC
