From 57def79f5d6e97f5b353b22a5fb856e627cdf061 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Sun, 28 Jun 2015 19:12:19 +0100
Subject: [PATCH] Set the x86 arch name to i686 for mingw-w64


diff --git a/lib/Driver/Driver.cpp b/lib/Driver/Driver.cpp
index 5451132..a6852b6 100644
--- a/lib/Driver/Driver.cpp
+++ b/lib/Driver/Driver.cpp
@@ -1991,6 +1991,12 @@ static llvm::Triple computeTargetTriple(StringRef DefaultTargetTriple,
 
     if (AT != llvm::Triple::UnknownArch && AT != Target.getArch())
       Target.setArch(AT);
+
+  // Override the default ArchName of 'i386' for mingw-w64 targets when passed -m32.
+  if (AT == llvm::Triple::x86 && Target.isWindowsGNUEnvironment() &&
+      Target.getVendorName().equals(StringRef("w64")))
+    Target.setArchName(StringRef("i686"));
+
   }
 
   return Target;
-- 
2.4.5

