From 0d44d5c00062f57a17eec0ca61e6590ab09d816d Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Mon, 8 Jun 2015 02:14:25 +0100
Subject: [PATCH 1/4] COFF: add the mingw alias for ImageBase

---
 COFF/SymbolTable.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/COFF/SymbolTable.cpp b/COFF/SymbolTable.cpp
index a90e208..e301b6a 100644
--- a/COFF/SymbolTable.cpp
+++ b/COFF/SymbolTable.cpp
@@ -23,6 +23,7 @@ namespace coff {
 
 SymbolTable::SymbolTable() {
   resolve(new (Alloc) DefinedAbsolute("__ImageBase", Config->ImageBase));
+  resolve(new (Alloc) DefinedAbsolute("__image_base__", Config->ImageBase));
   if (!Config->EntryName.empty())
     resolve(new (Alloc) Undefined(Config->EntryName));
 }
-- 
2.3.3

