From 5042e15216190f11cfb12f0c81c187c7353fb1d5 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Sun, 28 Jun 2015 19:12:19 +0100
Subject: [PATCH 3/5] Set the x86 arch name to i686 for mingw-w64

---
 lib/Driver/Driver.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/Driver/Driver.cpp b/lib/Driver/Driver.cpp
index ba4d0e836b..9f8c850b22 100644
--- a/lib/Driver/Driver.cpp
+++ b/lib/Driver/Driver.cpp
@@ -429,6 +429,11 @@ static llvm::Triple computeTargetTriple(const Driver &D,
 
     if (AT != llvm::Triple::UnknownArch && AT != Target.getArch())
       Target.setArch(AT);
+
+  // Override the default ArchName of 'i386' for mingw-w64 targets when passed -m32.
+  if (AT == llvm::Triple::x86 && Target.isWindowsGNUEnvironment() &&
+      Target.getVendorName().equals(StringRef("w64")))
+    Target.setArchName(StringRef("i686"));
   }
 
   // Handle -miamcu flag.
-- 
2.13.3

