From 0e31fbf4e602f857cd33c1b5b446da354b64df11 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Sun, 28 Jun 2015 19:12:19 +0100
Subject: [PATCH 5/9] Set the x86 arch name to i686 for mingw-w64

---
 lib/Driver/Driver.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/Driver/Driver.cpp b/lib/Driver/Driver.cpp
index b8ae7e4c31..a76ef16c8a 100644
--- a/lib/Driver/Driver.cpp
+++ b/lib/Driver/Driver.cpp
@@ -381,6 +381,11 @@ static llvm::Triple computeTargetTriple(const Driver &D,
 
     if (AT != llvm::Triple::UnknownArch && AT != Target.getArch())
       Target.setArch(AT);
+
+  // Override the default ArchName of 'i386' for mingw-w64 targets when passed -m32.
+  if (AT == llvm::Triple::x86 && Target.isWindowsGNUEnvironment() &&
+      Target.getVendorName().equals(StringRef("w64")))
+    Target.setArchName(StringRef("i686"));
   }
 
   // Handle -miamcu flag.
-- 
2.12.0

