diff -urN cfe-3.9.0.src.orig/tools/libclang/CMakeLists.txt cfe-3.9.0.src/tools/libclang/CMakeLists.txt
--- cfe-3.9.0.src.orig/tools/libclang/CMakeLists.txt	2016-03-03 12:09:43.000000000 +0100
+++ cfe-3.9.0.src/tools/libclang/CMakeLists.txt	2016-09-26 19:40:33.887171200 +0200
@@ -97,6 +97,15 @@
   Support
   )
 
+if(MINGW)
+    set_target_properties(libclang
+                          PROPERTIES
+                          PREFIX ""
+                          OUTPUT_NAME clang
+                          RUNTIME_OUTPUT_NAME libclang
+                          ARCHIVE_OUTPUT_NAME clang)
+endif()
+
 if(ENABLE_SHARED)
   if(WIN32)
     set_target_properties(libclang
