diff -urN cfe-3.9.0.src.orig/lib/Driver/Tools.cpp cfe-3.9.0.src/lib/Driver/Tools.cpp
--- cfe-3.9.0.src.orig/lib/Driver/Tools.cpp	2016-08-13 22:43:56.000000000 +0200
+++ cfe-3.9.0.src/lib/Driver/Tools.cpp	2016-10-20 17:35:19.119164300 +0200
@@ -2608,7 +2608,7 @@
   case llvm::Triple::wasm32:
   case llvm::Triple::wasm64:
     getWebAssemblyTargetFeatures(Args, Features);
-    break; 
+    break;
   case llvm::Triple::sparc:
   case llvm::Triple::sparcel:
   case llvm::Triple::sparcv9:
@@ -5008,7 +5008,8 @@
   Args.AddLastArg(CmdArgs, options::OPT_fno_operator_names);
   // Emulated TLS is enabled by default on Android, and can be enabled manually
   // with -femulated-tls.
-  bool EmulatedTLSDefault = Triple.isAndroid() || Triple.isWindowsCygwinEnvironment();
+  bool EmulatedTLSDefault = Triple.isAndroid() || Triple.isWindowsCygwinEnvironment()
+                            || getToolChain().getTriple().isWindowsGNUEnvironment();
   if (Args.hasFlag(options::OPT_femulated_tls, options::OPT_fno_emulated_tls,
                    EmulatedTLSDefault))
     CmdArgs.push_back("-femulated-tls");
@@ -10490,7 +10491,8 @@
       if (Args.hasArg(options::OPT_pg))
         CmdArgs.push_back("-lgmon");
 
-      if (Args.hasArg(options::OPT_pthread))
+      if (!Args.hasArg(options::OPT_no_pthread) &&
+          (TC.GetCXXStdlibType(Args) != ToolChain::CST_Libcxx))
         CmdArgs.push_back("-lpthread");
 
       // add system libraries
