diff -urN lld-3.9.0.src.orig/lib/Config/CMakeLists.txt lld-3.9.0.src/lib/Config/CMakeLists.txt
--- lld-3.9.0.src.orig/lib/Config/CMakeLists.txt	2016-01-07 01:14:09.000000000 +0100
+++ lld-3.9.0.src/lib/Config/CMakeLists.txt	2016-11-14 23:51:29.013207900 +0100
@@ -5,5 +5,5 @@
   ${LLD_INCLUDE_DIR}/lld/Config
 
   LINK_LIBS
-    LLVMSupport
+    #LLVMSupport
   )
diff -urN lld-3.9.0.src.orig/lib/Core/CMakeLists.txt lld-3.9.0.src/lib/Core/CMakeLists.txt
--- lld-3.9.0.src.orig/lib/Core/CMakeLists.txt	2016-01-07 01:14:09.000000000 +0100
+++ lld-3.9.0.src/lib/Core/CMakeLists.txt	2016-11-14 23:51:38.957965300 +0100
@@ -12,5 +12,5 @@
   ${LLD_INCLUDE_DIR}/lld/Core
 
   LINK_LIBS
-    LLVMSupport
+    #LLVMSupport
   )
diff -urN lld-3.9.0.src.orig/lib/Driver/CMakeLists.txt lld-3.9.0.src/lib/Driver/CMakeLists.txt
--- lld-3.9.0.src.orig/lib/Driver/CMakeLists.txt	2016-03-02 20:08:05.000000000 +0100
+++ lld-3.9.0.src/lib/Driver/CMakeLists.txt	2016-11-14 23:51:52.765482800 +0100
@@ -14,9 +14,9 @@
     lldCore
     lldReaderWriter
     lldYAML
-    LLVMObject
-    LLVMOption
-    LLVMSupport
+    #LLVMObject
+    #LLVMOption
+    #LLVMSupport
   )
 
 add_dependencies(lldDriver DriverOptionsTableGen)
diff -urN lld-3.9.0.src.orig/lib/ReaderWriter/CMakeLists.txt lld-3.9.0.src/lib/ReaderWriter/CMakeLists.txt
--- lld-3.9.0.src.orig/lib/ReaderWriter/CMakeLists.txt	2016-03-02 01:37:50.000000000 +0100
+++ lld-3.9.0.src/lib/ReaderWriter/CMakeLists.txt	2016-11-14 23:52:06.134198600 +0100
@@ -14,6 +14,6 @@
   LINK_LIBS
     lldCore
     lldYAML
-    LLVMObject
-    LLVMSupport
+    #LLVMObject
+    #LLVMSupport
   )
diff -urN lld-3.9.0.src.orig/lib/ReaderWriter/MachO/CMakeLists.txt lld-3.9.0.src/lib/ReaderWriter/MachO/CMakeLists.txt
--- lld-3.9.0.src.orig/lib/ReaderWriter/MachO/CMakeLists.txt	2016-03-01 16:56:53.000000000 +0100
+++ lld-3.9.0.src/lib/ReaderWriter/MachO/CMakeLists.txt	2016-11-14 23:52:15.413279600 +0100
@@ -21,8 +21,8 @@
   LINK_LIBS
     lldCore
     lldYAML
-    LLVMObject
-    LLVMSupport
+    #LLVMObject
+    #LLVMSupport
     ${PTHREAD_LIB}
   )
 
diff -urN lld-3.9.0.src.orig/lib/ReaderWriter/YAML/CMakeLists.txt lld-3.9.0.src/lib/ReaderWriter/YAML/CMakeLists.txt
--- lld-3.9.0.src.orig/lib/ReaderWriter/YAML/CMakeLists.txt	2016-01-07 01:14:04.000000000 +0100
+++ lld-3.9.0.src/lib/ReaderWriter/YAML/CMakeLists.txt	2016-11-14 23:52:21.509464400 +0100
@@ -2,5 +2,5 @@
   ReaderWriterYAML.cpp
   LINK_LIBS
     lldCore
-    LLVMSupport
+    #LLVMSupport
   )
diff -urN lld-3.9.0.src.orig/tools/lld/CMakeLists.txt lld-3.9.0.src/tools/lld/CMakeLists.txt
--- lld-3.9.0.src.orig/tools/lld/CMakeLists.txt	2016-03-01 16:56:53.000000000 +0100
+++ lld-3.9.0.src/tools/lld/CMakeLists.txt	2016-11-14 23:51:12.405379700 +0100
@@ -6,7 +6,7 @@
   lldDriver
   lldCOFF
   lldELF
-  LLVMSupport
+  #LLVMSupport
   )
 
 install(TARGETS lld
