Index: lld/trunk/COFF/Driver.cpp
===================================================================
--- lld/trunk/COFF/Driver.cpp
+++ lld/trunk/COFF/Driver.cpp
@@ -167,7 +167,9 @@
       std::unique_ptr<Archive> File =
           CHECK(Archive::create(MBRef), Filename + ": failed to parse archive");
 
-      for (MemoryBufferRef M : getArchiveMembers(File.get()))
+      Archive *archive = File.get();
+      make<std::unique_ptr<Archive>>(std::move(File)); // take ownership
+      for (MemoryBufferRef M : getArchiveMembers(archive))
         addArchiveBuffer(M, "<whole-archive>", Filename);
       return;
     }
