--- lld/COFF/Driver.cpp.orig	2021-06-30 14:07:21.236743400 -0700
+++ lld/COFF/Driver.cpp	2021-06-30 14:07:57.752513500 -0700
@@ -149,9 +149,10 @@
 // Create a std::future that opens and maps a file using the best strategy for
 // the host platform.
 static std::future<MBErrPair> createFutureForFile(std::string path) {
-#if _WIN32
+#if _WIN64
   // On Windows, file I/O is relatively slow so it is best to do this
-  // asynchronously.
+  // asynchronously.  But 32-bit has issues with potentially launching tons
+  // of threads
   auto strategy = std::launch::async;
 #else
   auto strategy = std::launch::deferred;
