diff -urN libcxx-3.9.0.origlibcxx/include/__config libcxx-3.9.0libcxx/include/__config
--- libcxx-3.9.0.origlibcxx/include/__config	2016-09-21 16:28:09.452978000 +0200
+++ libcxx-3.9.0libcxx/include/__config	2016-09-25 01:19:05.170475000 +0200
@@ -150,8 +150,6 @@
 #  // If mingw not explicitly detected, assume using MS C runtime only.
 #  ifndef __MINGW32__
 #    define _LIBCPP_MSVCRT // Using Microsoft's C Runtime library
-#  else
-#    define _LIBCPP_HAS_NO_CONSTEXPR
 #  endif
 #endif  // _WIN32

@@ -496,8 +494,14 @@
 #endif

 // Determine if GCC supports relaxed constexpr
-#if !defined(__cpp_constexpr) || __cpp_constexpr < 201304L
-#define _LIBCPP_HAS_NO_CXX14_CONSTEXPR
+// GCC 6.X is retarded, it thinks it supports __cpp_constexpr 201304 but ...
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66297
+#if defined(__clang__) || __GNUC__ > 6
+# if !defined(__cpp_constexpr) || __cpp_constexpr < 201304L
+# define _LIBCPP_HAS_NO_CXX14_CONSTEXPR
+# endif
+#elif defined(__GNUC__)
+# define _LIBCPP_HAS_NO_CXX14_CONSTEXPR
 #endif

 // GCC 5 will support variable templates
