--- libcxx/src/support/win32/thread_win32.cpp.orig	2018-09-26 12:57:16.398685300 +0300
+++ libcxx/src/support/win32/thread_win32.cpp	2018-09-26 12:57:33.465115300 +0300
@@ -15,6 +15,7 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
+#if !defined(_LIBCPP_HAS_THREAD_API_PTHREAD)
 static_assert(sizeof(__libcpp_mutex_t) == sizeof(SRWLOCK), "");
 static_assert(alignof(__libcpp_mutex_t) == alignof(SRWLOCK), "");
 
@@ -272,5 +273,6 @@
     return GetLastError();
   return 0;
 }
+#endif
 
 _LIBCPP_END_NAMESPACE_STD
