--- lldb/cmake/modules/AddLLDB.cmake.orig	2018-06-01 13:47:02.305059800 +0300
+++ lldb/cmake/modules/AddLLDB.cmake	2018-06-01 13:49:34.796327700 +0300
@@ -44,7 +44,7 @@
   if (PARAM_OBJECT)
     add_library(${name} ${libkind} ${srcs})
   else()
-    llvm_add_library(${name} ${libkind} ${srcs} LINK_LIBS
+    llvm_add_library(${name} ${libkind} ${ARG_UNPARSED_ARGUMENTS} ${srcs} LINK_LIBS
                                 ${PARAM_LINK_LIBS}
                                 DEPENDS ${PARAM_DEPENDS})
 
--- lldb/source/API/CMakeLists.txt.orig	2018-06-01 13:43:09.433650000 +0300
+++ lldb/source/API/CMakeLists.txt	2018-06-01 13:50:13.094395000 +0300
@@ -14,7 +14,14 @@
 
 get_property(LLDB_ALL_PLUGINS GLOBAL PROPERTY LLDB_PLUGINS)
 
+if(MSVC)
+  set(output_name "liblldb")
+else()
+  set(output_name "lldb")
+endif()
+
 add_lldb_library(liblldb SHARED
+  OUTPUT_NAME ${output_name}
   SBAddress.cpp
   SBAttachInfo.cpp
   SBBlock.cpp
