--- lldb/include/lldb/lldb-defines.h.orig	2018-06-14 10:47:59.093121500 +0300
+++ lldb/include/lldb/lldb-defines.h	2018-06-14 10:48:12.399156100 +0300
@@ -12,15 +12,17 @@
 
 #include "lldb/lldb-types.h"
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
+#if defined(LLDB_STATIC_BUILD)
+#define LLDB_API
+#else
 #if defined(EXPORT_LIBLLDB)
 #define LLDB_API __declspec(dllexport)
-#elif defined(IMPORT_LIBLLDB)
+#else
 #define LLDB_API __declspec(dllimport)
-#else
-#define LLDB_API
+#endif
 #endif
-#else // defined (_MSC_VER)
+#else // defined (_WIN32)
 #define LLDB_API
 #endif
 
