--- clang-3.4/lib/Frontend/InitHeaderSearch.cpp.orig	2014-01-08 12:37:15.055400000 +0400
+++ clang-3.4/lib/Frontend/InitHeaderSearch.cpp	2014-01-08 12:38:22.743800000 +0400
@@ -405,6 +405,15 @@
     AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.6.2");
     AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.6.3");
     AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.7.0");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.7.1");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.7.2");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.7.3");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.7.4");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.0");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.1");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.2");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.8.3");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.9.0");
+    AddMinGW64CXXPaths(HSOpts.ResourceDir, "4.9.1");
     // mingw.org C++ include paths
     AddMinGWCPlusPlusIncludePaths("/mingw/lib/gcc", "mingw32", "4.5.2"); //MSYS
 #if defined(_WIN32)
