Index: llvm-toolchain-snapshot-3.5~svn211313/tools/bugpoint/Makefile
===================================================================
--- llvm-toolchain-snapshot-3.5~svn211313.orig/tools/bugpoint/Makefile
+++ llvm-toolchain-snapshot-3.5~svn211313/tools/bugpoint/Makefile
@@ -12,6 +12,9 @@ TOOLNAME := bugpoint
 LINK_COMPONENTS := asmparser instrumentation scalaropts ipo linker bitreader \
                    bitwriter irreader vectorize objcarcopts codegen
 
+# Crappy workaround to make sure it links correctly.
+LLVMLibsOptions := ../../lib/IR/Release*/Pass.o
+
 # Support plugins.
 NO_DEAD_STRIP := 1
 
Index: llvm-toolchain-snapshot-3.5~svn211313/tools/opt/Makefile
===================================================================
--- llvm-toolchain-snapshot-3.5~svn211313.orig/tools/opt/Makefile
+++ llvm-toolchain-snapshot-3.5~svn211313/tools/opt/Makefile
@@ -10,7 +10,9 @@
 LEVEL := ../..
 TOOLNAME := opt
 LINK_COMPONENTS := bitreader bitwriter asmparser irreader instrumentation scalaropts objcarcopts ipo vectorize all-targets codegen
+# Crappy workaround to make sure it links correctly.
 
+LLVMLibsOptions := ../../lib/IR/Release*/Pass.o
 # Support plugins.
 NO_DEAD_STRIP := 1
 
