--- llvm-3.3.src/test/lit.site.cfg.in.orig	2013-05-04 08:36:23.000000000 +0100
+++ llvm-3.3.src/test/lit.site.cfg.in	2014-01-04 04:48:08.506762500 +0000
@@ -9,7 +9,6 @@
 config.llvm_shlib_ext = "@SHLIBEXT@"
 config.lit_tools_dir = "@LLVM_LIT_TOOLS_DIR@"
 config.python_executable = "@PYTHON_EXECUTABLE@"
-config.ocamlopt_executable = "@OCAMLOPT@"
 config.enable_shared = @ENABLE_SHARED@
 config.enable_assertions = @ENABLE_ASSERTIONS@
 config.lto_is_enabled = "@LTO_IS_ENABLED@"
--- llvm-3.3.src/test/lit.cfg.orig	2013-05-06 17:51:08.000000000 +0100
+++ llvm-3.3.src/test/lit.cfg	2014-01-04 12:10:16.361101100 +0000
@@ -185,7 +185,6 @@
   config.substitutions.append( ('%lli', 'lli') )
 
 # Add site-specific substitutions.
-config.substitutions.append( ('%ocamlopt', config.ocamlopt_executable) )
 config.substitutions.append( ('%llvmshlibdir', config.llvm_shlib_dir) )
 config.substitutions.append( ('%shlibext', config.llvm_shlib_ext) )
 
