--- gstreamer-editing-services-1.8.2/ges/ges-track-element.c.orig   2016-08-18 18:25:02.825182100 +0200
+++ gstreamer-editing-services-1.8.2/ges/ges-track-element.c        2016-08-18 18:29:19.570504100 +0200
@@ -291,13 +291,13 @@
   /**
    * GESTrackElement::control-binding-removed:
    * @track_element: a #GESTrackElement
-   * @control_binding: the #GstControlBinding that has been added
+   * @control_binding: the #GstControlBinding that has been removed
    *
-   * The control-bunding-added  is emitted each time a control binding
-   * is added for a child property of @track_element
+   * The control-bunding-removed  is emitted each time a control binding
+   * is removed for a child property of @track_element
    */
   ges_track_element_signals[CONTROL_BINDING_REMOVED] =
-      g_signal_new ("control-binding-reomved", G_TYPE_FROM_CLASS (klass),
+      g_signal_new ("control-binding-removed", G_TYPE_FROM_CLASS (klass),
       G_SIGNAL_RUN_FIRST, 0, NULL, NULL, g_cclosure_marshal_generic,
       G_TYPE_NONE, 1, GST_TYPE_CONTROL_BINDING);
