--- gstreamer-editing-services-1.6.1/ges/ges-track-element.c.orig	2015-11-02 12:02:31.273729700 +0300
+++ gstreamer-editing-services-1.6.1/ges/ges-track-element.c	2015-11-02 12:04:23.608495100 +0300
@@ -264,15 +264,15 @@
       G_TYPE_NONE, 1, GST_TYPE_CONTROL_BINDING);
 
   /**
-   * GESTrackElement::control-binding-added:
+   * GESTrackElement::control-binding-removed:
    * @track_element: a #GESTrackElement
-   * @control_binding: the #GstControlBinding that has been added
+   * @control_binding: the #GstControlBinding that has been removed
    *
-   * The control-bunding-added  is emitted each time a control binding
-   * is added for a child property of @track_element
+   * The control-binding-removed  is emitted each time a control binding
+   * is removed from a child property of @track_element
    */
   ges_track_element_signals[CONTROL_BINDING_REMOVED] =
-      g_signal_new ("control-binding-reomved", G_TYPE_FROM_CLASS (klass),
+      g_signal_new ("control-binding-removed", G_TYPE_FROM_CLASS (klass),
       G_SIGNAL_RUN_FIRST, 0, NULL, NULL, g_cclosure_marshal_generic,
       G_TYPE_NONE, 1, GST_TYPE_CONTROL_BINDING);
 
--- gstreamer-editing-services-1.6.1/ges/ges-extractable.c.orig	2015-11-02 12:06:11.201734700 +0300
+++ gstreamer-editing-services-1.6.1/ges/ges-extractable.c	2015-11-02 12:07:18.658988300 +0300
@@ -136,7 +136,7 @@
  * ges_extractable_get_id:
  * @self: The #GESExtractable
  *
- * Returns: (transfer full) The #id of the associated #GESAsset, free with #g_free
+ * Returns: (transfer full): The #id of the associated #GESAsset, free with #g_free
  */
 gchar *
 ges_extractable_get_id (GESExtractable * self)
--- gstreamer-editing-services-1.6.1/ges/ges-timeline-element.c.orig	2015-11-02 12:15:07.607088800 +0300
+++ gstreamer-editing-services-1.6.1/ges/ges-timeline-element.c	2015-11-02 12:15:18.545994800 +0300
@@ -1259,7 +1259,7 @@
 }
 
 /**
- * ges_track_element_get_child_property_by_pspec:
+ * ges_timeline_element_get_child_property_by_pspec:
  * @self: a #GESTrackElement
  * @pspec: The #GParamSpec that specifies the property you want to get
  * @value: (out): return location for the value
