diff -Naur llvmlite-0.36.0.a/ffi/custom_passes.cpp llvmlite-0.36.0.b/ffi/custom_passes.cpp
--- llvmlite-0.36.0.a/ffi/custom_passes.cpp	2021-03-11 17:54:55.000000000 +0100
+++ llvmlite-0.36.0.b/ffi/custom_passes.cpp	2021-03-30 12:25:43.415085100 +0200
@@ -400,7 +400,7 @@
                 // LLVM 9 postdomtree.dominates takes basic blocks
                 if ( domtree.dominates(incref, decref)
                         && postdomtree.dominates(decref->getParent(), incref->getParent()) ){
-#elif LLVM_VERSION_MAJOR == 10
+#elif LLVM_VERSION_MAJOR == 11
                 // LLVM 10 postdomtree.dominates can handle instructions
                 if ( domtree.dominates(incref, decref)
                         && postdomtree.dominates(decref, incref) ){
diff -Naur llvmlite-0.36.0.a/ffi/transforms.cpp llvmlite-0.36.0.b/ffi/transforms.cpp
--- llvmlite-0.36.0.a/ffi/transforms.cpp	2021-03-11 17:54:55.000000000 +0100
+++ llvmlite-0.36.0.b/ffi/transforms.cpp	2021-03-30 12:26:12.121988800 +0200
@@ -5,7 +5,7 @@
 
 
 extern "C" {
-
+/*
 namespace llvm {
     inline PassManagerBuilder *unwrap(LLVMPassManagerBuilderRef P) {
         return reinterpret_cast<PassManagerBuilder*>(P);
@@ -15,7 +15,7 @@
         return reinterpret_cast<LLVMPassManagerBuilderRef>(P);
     }
 }
-
+*/
 
 API_EXPORT(LLVMPassManagerBuilderRef)
 LLVMPY_PassManagerBuilderCreate()
