--- llvmlite-0.34.0rc1/ffi/build.py.old	2020-08-05 13:33:58.000000000 -0700
+++ llvmlite-0.34.0rc1/ffi/build.py	2020-08-11 17:40:52.722273000 -0700
@@ -92,7 +92,7 @@
     # Run configuration step
     try_cmake(here_dir, build_dir, generator)
     subprocess.check_call(['cmake', '--build', build_dir, '--config', config])
-    shutil.copy(os.path.join(build_dir, config, 'llvmlite.dll'), target_dir)
+    shutil.copy(os.path.join(build_dir, 'libllvmlite.dll'), target_dir)
 
 
 def main_posix(kind, library_ext):
