--- x86_64/qttools/src/qdoc/configure.pri.orig	2019-06-18 13:42:33.146847400 +0300
+++ x86_64/qttools/src/qdoc/configure.pri	2019-06-18 13:48:47.103371000 +0300
@@ -38,6 +38,10 @@
 }
 
 defineTest(qtConfTest_libclang) {
+    isEmpty(QDOC_SKIP_BUILD): QDOC_SKIP_BUILD = $$(QDOC_SKIP_BUILD)
+    !isEmpty(QDOC_SKIP_BUILD) {
+        return(false)
+    }
     isEmpty(QDOC_USE_STATIC_LIBCLANG): QDOC_USE_STATIC_LIBCLANG = $$(QDOC_USE_STATIC_LIBCLANG)
 
     equals(QMAKE_HOST.os, Windows) {
