--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,11 +114,11 @@ ELSE ()
 ENDIF ()
 
 if (QT_MAJOR_VERSION GREATER_EQUAL 6)
-	find_package(KUserFeedbackQt6)
+	find_package(KF6UserFeedback)
 else ()
 	find_package(KUserFeedback)
 endif ()
-IF (KUserFeedback_FOUND OR KUserFeedbackQt6_FOUND)
+IF (KUserFeedback_FOUND OR KF6UserFeedback_FOUND)
 	MESSAGE (STATUS "Found KUserFeedback")
 	add_definitions (-DHAVE_KUSERFEEDBACK)
 ELSE ()
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -628,8 +628,8 @@ ENDIF ()
 if (KUserFeedback_FOUND)
 	target_link_libraries( labplot2lib KUserFeedbackWidgets )
 endif ()
-if (KUserFeedbackQt6_FOUND)
-	target_link_libraries( labplot2lib KUserFeedbackCoreQt6 KUserFeedbackWidgetsQt6 )
+if (KF6UserFeedback_FOUND)
+	target_link_libraries( labplot2lib KF6::UserFeedbackCore KF6::UserFeedbackWidgets )
 endif ()
 IF (Discount_FOUND)
 	target_link_libraries( labplot2lib Discount::Lib )
--- a/src/kdefrontend/MainWin.cpp
+++ b/src/kdefrontend/MainWin.cpp
@@ -62,12 +62,12 @@
 
 #ifdef HAVE_KUSERFEEDBACK
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-#include <KUserFeedbackQt6/ApplicationVersionSource>
-#include <KUserFeedbackQt6/PlatformInfoSource>
-#include <KUserFeedbackQt6/QtVersionSource>
-#include <KUserFeedbackQt6/ScreenInfoSource>
-#include <KUserFeedbackQt6/StartCountSource>
-#include <KUserFeedbackQt6/UsageTimeSource>
+#include <KUserFeedback/ApplicationVersionSource>
+#include <KUserFeedback/PlatformInfoSource>
+#include <KUserFeedback/QtVersionSource>
+#include <KUserFeedback/ScreenInfoSource>
+#include <KUserFeedback/StartCountSource>
+#include <KUserFeedback/UsageTimeSource>
 #else
 #include <KUserFeedback/ApplicationVersionSource>
 #include <KUserFeedback/PlatformInfoSource>
--- a/src/kdefrontend/MainWin.h
+++ b/src/kdefrontend/MainWin.h
@@ -52,7 +52,7 @@ class CDockWidget;
 
 #ifdef HAVE_KUSERFEEDBACK
 #if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
-#include <KUserFeedbackQt6/Provider>
+#include <KUserFeedback/Provider>
 #else
 #include <KUserFeedback/Provider>
 #endif
--- a/src/kdefrontend/SettingsDialog.cpp
+++ b/src/kdefrontend/SettingsDialog.cpp
@@ -31,7 +31,7 @@
 
 #ifdef HAVE_KUSERFEEDBACK
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-#include <KUserFeedbackQt6/FeedbackConfigWidget>
+#include <KUserFeedback/FeedbackConfigWidget>
 #else
 #include <KUserFeedback/FeedbackConfigWidget>
 #endif
