diff -Naur vigra-1.11.0.org/config/FindVIGRANUMPY_DEPENDENCIES.cmake vigra-1.11.0/config/FindVIGRANUMPY_DEPENDENCIES.cmake
--- vigra-1.11.0.org/config/FindVIGRANUMPY_DEPENDENCIES.cmake	2016-03-17 11:22:21.000000000 +0100
+++ vigra-1.11.0/config/FindVIGRANUMPY_DEPENDENCIES.cmake	2016-08-21 00:25:53.671689582 +0200
@@ -62,7 +62,7 @@
             CACHE FILEPATH "Python libraries"
             FORCE)
     ELSE()
-        IF(WIN32)
+        IF(WIN32 AND NOT MSYS)
             set(PYTHON_LIBRARY_NAME python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
         ELSE()
             execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
@@ -141,6 +141,10 @@
                          "from distutils.sysconfig import *; print(get_python_lib(1))"
                           OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
         FILE(TO_CMAKE_PATH ${PYTHON_SITE_PACKAGES} VIGRANUMPY_INSTALL_DIR)
+        IF(MSYS)
+            execute_process(COMMAND cygpath -u ${VIGRANUMPY_INSTALL_DIR}
+                              OUTPUT_VARIABLE VIGRANUMPY_INSTALL_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+        ENDIF()
     ENDIF()
     SET(VIGRANUMPY_INSTALL_DIR ${VIGRANUMPY_INSTALL_DIR}
         CACHE PATH "where to install the VIGRA Python package" FORCE)
