--- a/config/FindVIGRANUMPY_DEPENDENCIES.cmake
+++ b/config/FindVIGRANUMPY_DEPENDENCIES.cmake
@@ -37,6 +37,7 @@
             # define names for thread-safe library variants
             SET(BOOST_PYTHON_NAMES
                     boost_python-py${Python_VERSION_MAJOR}${Python_VERSION_MINOR}-mt
+                    boost_python${Python_VERSION_MAJOR}${Python_VERSION_MINOR}-mt
                     boost_python-${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}-mt
                     boost_python${Python_VERSION_MAJOR}-mt
                     boost_python-mt)
@@ -86,6 +87,8 @@
                          "import sysconfig; print(sysconfig.get_paths()['purelib'])"
                           OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
         FILE(TO_CMAKE_PATH ${PYTHON_SITE_PACKAGES} VIGRANUMPY_INSTALL_DIR)
+        execute_process(COMMAND cygpath -u ${VIGRANUMPY_INSTALL_DIR}
+                        OUTPUT_VARIABLE VIGRANUMPY_INSTALL_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
     ENDIF()
     SET(VIGRANUMPY_INSTALL_DIR ${VIGRANUMPY_INSTALL_DIR}
         CACHE PATH "where to install the VIGRA Python package" FORCE)
