diff -Naur build-x86_64-w64-mingw32/include/vigra/hdf5impex.hxx build-x86_64-w64-mingw32.patched/include/vigra/hdf5impex.hxx
--- build-x86_64-w64-mingw32/include/vigra/hdf5impex.hxx	2016-11-26 21:31:14.570702300 +0100
+++ build-x86_64-w64-mingw32.patched/include/vigra/hdf5impex.hxx	2016-11-26 21:27:11.031268900 +0100
@@ -709,10 +709,7 @@
             ordered as 'z', 'y', 'x', this function will return the shape in the order
             'x', 'y', 'z'.
          */
-    VIGRA_EXPORT ArrayVector<hsize_t> const & shape() const
-    {
-        return m_dims;
-    }
+    VIGRA_EXPORT ArrayVector<hsize_t> const & shape() const;
 
         /** Get the shape (length) of the dataset along dimension \a dim.
 
diff -Naur build-x86_64-w64-mingw32/src/impex/hdf5impex.cxx build-x86_64-w64-mingw32.patched/src/impex/hdf5impex.cxx
--- build-x86_64-w64-mingw32/src/impex/hdf5impex.cxx	2016-11-26 21:31:14.570702300 +0100
+++ build-x86_64-w64-mingw32.patched/src/impex/hdf5impex.cxx	2016-11-26 21:27:02.713454100 +0100
@@ -156,6 +156,11 @@
     return MultiArrayIndex(m_dims[dim]); 
 }
 
+ArrayVector<hsize_t> const & HDF5ImportInfo::shape() const
+{
+    return m_dims;
+}
+
 MultiArrayIndex HDF5ImportInfo::numDimensions() const 
 { 
     return MultiArrayIndex(m_dimensions); 
