From 2c09ab7fa5886efd9951ccf4ccb5d39b9e5d685e Mon Sep 17 00:00:00 2001
From: Ullrich Koethe <ullrich.koethe@iwr.uni-heidelberg.de>
Date: Tue, 16 Oct 2018 19:33:35 +0200
Subject: [PATCH] cmake: added  -DH5_BUILT_AS_DYNAMIC_LIB

---
 config/FindHDF5.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/config/FindHDF5.cmake b/config/FindHDF5.cmake
index a842c7ccb..29de62973 100644
--- a/config/FindHDF5.cmake
+++ b/config/FindHDF5.cmake
@@ -45,6 +45,9 @@ if(HDF5_INCLUDE_DIR)
             SET(HDF5_CFLAGS)
             SET(HDF5_CPPFLAGS)
         ENDIF()
+        SET(HDF5_CFLAGS "${HDF5_CPPFLAGS} -DH5_BUILT_AS_DYNAMIC_LIB")
+        SET(HDF5_CPPFLAGS "${HDF5_CPPFLAGS} -DH5_BUILT_AS_DYNAMIC_LIB")
+
 
         set(HDF5_USES_ZLIB FALSE)
         TRY_COMPILE(HDF5_USES_ZLIB
