diff -Naur vigra-1.11.1-orig/config/VIGRA_ADD_NUMPY_MODULE.cmake vigra-1.11.1/config/VIGRA_ADD_NUMPY_MODULE.cmake
--- vigra-1.11.1-orig/config/VIGRA_ADD_NUMPY_MODULE.cmake	2017-05-19 18:01:08.000000000 +0300
+++ vigra-1.11.1/config/VIGRA_ADD_NUMPY_MODULE.cmake	2020-06-10 11:07:31.988569500 +0300
@@ -71,8 +71,10 @@
     TARGET_LINK_LIBRARIES(${TARGET_NAME} ${VIGRANUMPY_LIBRARIES})
 
     IF(PYTHON_PLATFORM MATCHES "^windows$")
+        execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import distutils.sysconfig, sys; get = distutils.sysconfig.get_config_var; sys.stdout.write(get('EXT_SUFFIX') or get('SO') or '.pyd');"
+                        OUTPUT_VARIABLE PYTHON_MOD_EXT)
         SET_TARGET_PROPERTIES(${TARGET_NAME} PROPERTIES OUTPUT_NAME "${LIBRARY_NAME}"
-                                                           PREFIX "" SUFFIX  ".pyd")
+                                                           PREFIX "" SUFFIX  "${PYTHON_MOD_EXT}")
     ELSEIF(MACOSX)
         IF(${CMAKE_MAJOR_VERSION} LESS 3)
             SET_TARGET_PROPERTIES(${TARGET_NAME} PROPERTIES OUTPUT_NAME "${LIBRARY_NAME}" PREFIX ""
diff -Naur vigra-1.11.1-orig/config/VigranumpyConfig.cmake.in vigra-1.11.1/config/VigranumpyConfig.cmake.in
--- vigra-1.11.1-orig/config/VigranumpyConfig.cmake.in	2017-05-19 18:01:08.000000000 +0300
+++ vigra-1.11.1/config/VigranumpyConfig.cmake.in	2020-06-10 11:08:42.134660900 +0300
@@ -52,9 +52,11 @@
         ADD_DEFINITIONS(-DBOOST_AUTO_LINK_NOMANGLE)
     endif()
     
-    IF(WIN32 AND NOT MINGW)
+    IF(WIN32)
+        execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import distutils.sysconfig, sys; get = distutils.sysconfig.get_config_var; sys.stdout.write(get('EXT_SUFFIX') or get('SO') or '.pyd');"
+                        OUTPUT_VARIABLE PYTHON_MOD_EXT)
         SET_TARGET_PROPERTIES(${TARGET_NAME} PROPERTIES OUTPUT_NAME "${LIBRARY_NAME}" 
-                                                           PREFIX "" SUFFIX  ".pyd")
+                                                           PREFIX "" SUFFIX  "${PYTHON_MOD_EXT}")
     ELSEIF(APPLE)
         SET_TARGET_PROPERTIES(${TARGET_NAME} PROPERTIES OUTPUT_NAME "${LIBRARY_NAME}" PREFIX "" SUFFIX ".so")
     ELSE()
