--- vigra-1.11.1/CMakeLists.txt.orig	2020-06-10 11:21:22.125614100 +0300
+++ vigra-1.11.1/CMakeLists.txt	2020-06-10 11:22:11.574132700 +0300
@@ -369,10 +369,10 @@
 
 # export targets:
 INSTALL(EXPORT vigra-targets
-        DESTINATION lib${LIB_SUFFIX}/vigra)
+        DESTINATION lib${LIB_SUFFIX}/cmake/vigra)
 INSTALL(FILES ${PROJECT_BINARY_DIR}/lib/vigra/CMake/VigraConfig.cmake
             ${PROJECT_BINARY_DIR}/lib/vigra/CMake/VigraConfigVersion.cmake
-        DESTINATION lib${LIB_SUFFIX}/vigra)
+        DESTINATION lib${LIB_SUFFIX}/cmake/vigra)
 EXPORT(TARGETS vigraimpex FILE vigra-targets.cmake)
 
 ##################################################
--- vigra-1.11.1/vigranumpy/CMakeLists.txt.orig	2020-06-10 11:22:37.014851900 +0300
+++ vigra-1.11.1/vigranumpy/CMakeLists.txt	2020-06-10 11:22:55.614969000 +0300
@@ -59,7 +59,7 @@
                            COMMENT "Creating vigranumpy setup in ${CMAKE_CURRENT_BINARY_DIR}/dist")
     ENDIF()
 
-    set(VIGRANUMPY_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/vigranumpy)
+    set(VIGRANUMPY_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/cmake/vigranumpy)
 
     set(VIGRANUMPY_INCLUDE_RELATIVE "")
     foreach(PATH ${CMAKE_INSTALL_PREFIX}/include ${VIGRANUMPY_INCLUDE_DIRS})
