diff --git a/include/vigra/bessel.hxx b/include/vigra/bessel.hxx
index 2ec0733..cfa2d78 100644
--- a/include/vigra/bessel.hxx
+++ b/include/vigra/bessel.hxx
@@ -279,7 +279,7 @@ inline double besselJ(int n, double x)
         return n == 0 ? 1.0 : 0.0;
 #if defined(HasBoostMath)
     return boost::math::cyl_bessel_j((double)n, x);
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && !defined(__STRICT_ANSI__)
     return ::jn(n, x);
 #elif defined(_MSC_VER)
     return _jn(n, x);
@@ -315,7 +315,7 @@ inline double besselY(int n, double x)
         return -std::numeric_limits<double>::infinity();
 #if defined(HasBoostMath)
     return boost::math::cyl_neumann((double)n, x);
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && !defined(__STRICT_ANSI__)
     return ::yn(n, x);
 #elif defined(_MSC_VER)
     return _yn(n, x);
