--- origsrc/paho.mqtt.c-1.1.0/src/CMakeLists.txt	2016-06-20 01:00:10.000000000 +0200
+++ src/paho.mqtt.c-1.1.0/src/CMakeLists.txt	2016-10-16 21:08:35.069400000 +0200
@@ -65,7 +65,8 @@
     SOVERSION ${PAHO_VERSION_MAJOR})
 INSTALL(TARGETS paho-mqtt3c paho-mqtt3a MQTTVersion
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib)
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib)
 INSTALL(FILES MQTTAsync.h MQTTClient.h MQTTClientPersistence.h
     DESTINATION include)
 
