--- a/env/CMakeLists.txt
+++ b/env/CMakeLists.txt
@@ -33,7 +33,7 @@
     PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/${ENV_SCRIPT_NAME}"
     DESTINATION "${ONEAPI_INSTALL_ENVDIR}"
     COMPONENT dev)
-else()
+
   file(TO_CMAKE_PATH "${env_rel_prefix}" env_rel_prefix)
   file(TO_CMAKE_PATH "${CMAKE_INSTALL_INCLUDEDIR}" _install_includedir)
   file(TO_CMAKE_PATH "${CMAKE_INSTALL_LIBDIR}" _install_libdir)
