From 270786a12f18b7639cac4e59f36388d14cad7811 Mon Sep 17 00:00:00 2001
From: Benedikt Bitterli <mail@noobody.org>
Date: Sun, 22 Nov 2015 12:25:29 +0100
Subject: [PATCH 2/4] Add CMake check for MinGW

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 51ee620..b333c86 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ SET(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "The direct
 SET(LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE STRING "Directory where lib will install")
 SET(DOC_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/doc/${PROJECT_NAME}" CACHE PATH "Path to the documentation")
 
-IF(UNIX OR CYGWIN)
+IF(UNIX OR CYGWIN OR MINGW)
     SET(_CMAKE_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}")
 ELSEIF(WIN32)
     SET(_CMAKE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/cmake")
@@ -69,7 +69,7 @@ if(RAPIDJSON_BUILD_TESTS)
 endif()
 
 # pkg-config
-IF (UNIX OR CYGWIN)
+IF (UNIX OR CYGWIN OR MINGW)
   CONFIGURE_FILE (${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}.pc.in
                   ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc
                   @ONLY)
-- 
2.5.0

