From 0c3b35390aa8c9203e48fb938fea4ba59a43f7ea Mon Sep 17 00:00:00 2001
From: Benedikt Bitterli <mail@noobody.org>
Date: Sun, 22 Nov 2015 12:25:51 +0100
Subject: [PATCH 3/4] Exclude CMakeLists.txt from install targets

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b333c86..17e347c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,7 @@ install(DIRECTORY example/
     # Following patterns are for excluding the intermediate/object files
     # from an install of in-source CMake build.
     PATTERN "CMakeFiles" EXCLUDE
+    PATTERN "CMakeLists.txt" EXCLUDE
     PATTERN "Makefile" EXCLUDE
     PATTERN "cmake_install.cmake" EXCLUDE)
 
-- 
2.5.0

